diff --git a/old_test.java b/new_test.java
index 4ca323e..3066541 100644
--- a/old_test.java
+++ b/new_test.java
@@ -73,7 +73,7 @@
             .build(ruleResolver);
     e.getBuildable().getEnhancedDeps(ruleResolver);
 
-    ImmutableList<HasAndroidResourceDeps> deps2 = UberRDotJavaUtil.getAndroidResourceDeps(e);
+    ImmutableList<HasAndroidResourceDeps> deps2 = UberRDotJavaUtil.getAndroidResourceDeps(a);
     assertTrue(
         String.format(
             "Topological sort %s should be either %s or %s", deps, validResult1, validResult2),
