diff --git a/old_product.java b/new_product.java
index 1a18b55..5c955d7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,12 @@
-public static JavaRDD<Geometry> readToGeometryRDD(JavaSparkContext sc, String inputPath, final GeometryFactory geometryFactory)
+public static SpatialRDD<Geometry> readToGeometryRDD(JavaSparkContext sc, String inputPath, final GeometryFactory geometryFactory)
     {
-        return readShapefile(sc, inputPath, geometryFactory);
+        SpatialRDD<Geometry> spatialRDD = new SpatialRDD();
+        spatialRDD.rawSpatialRDD = readShapefile(sc, inputPath, geometryFactory);
+        try {
+            spatialRDD.fieldNames = readFieldNames(sc, inputPath);
+        }
+        catch (IOException e) {
+            e.printStackTrace();
+        }
+        return spatialRDD;
     }
\ No newline at end of file
