diff --git a/old_test.java b/new_test.java
index 3864a31..e28bc2b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,6 @@
         String inputLocation = getShapeFilePath("polygon");
         FeatureCollection<SimpleFeatureType, SimpleFeature> collection = loadFeatures(inputLocation);
         // load shapes with our tool
-        JavaRDD<Geometry> shapeRDD = ShapefileReader.readToGeometryRDD(sc, inputLocation);
-        Assert.assertEquals(shapeRDD.collect().size(), collection.size());
+        SpatialRDD shapeRDD = ShapefileReader.readToGeometryRDD(sc, inputLocation);
+        assertEquals(shapeRDD.rawSpatialRDD.collect().size(), collection.size());
     }
\ No newline at end of file
