diff --git a/old_product.java b/new_product.java
index e6e0245..d331f91 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-public Stream syn(SynInfo synInfo) throws InterruptedException, ExecutionException, TimeoutException;
\ No newline at end of file
+public Stream push(PushInfo pushInfo) throws InterruptedException, ExecutionException, TimeoutException;
\ No newline at end of file
