diff --git a/old_test.java b/new_test.java
index cd7d6b5..effc9a3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,8 +6,9 @@
         Set<Stream> streams = new HashSet<>();
         streams.add(stream);
         when(synStreamFrame.isClose()).thenReturn(false);
-        SynInfo synInfo = new SynInfo(new Fields(), false);
+        PushInfo pushInfo = new PushInfo(new Fields(), false);
         when(session.getStreams()).thenReturn(streams);
-        stream.syn(synInfo, new Promise.Adapter<Stream>());
-        verify(session).syn(argThat(new PushSynInfoMatcher(stream.getId(), synInfo)), any(StreamFrameListener.class), any(Promise.class));
+        stream.push(pushInfo, new Promise.Adapter<Stream>());
+        verify(session).syn(argThat(new PushSynInfoMatcher(stream.getId(), pushInfo)),
+                any(StreamFrameListener.class), any(Promise.class));
     }
\ No newline at end of file
