diff --git a/old_product.java b/new_product.java
index 1f1d105..6cf645c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -47,7 +47,7 @@ void learn(T[] x, int[] y, double[] weight) {
             }
 
             // train SVM in a stochastic order.
-            int[] index = Math.permutate(n);
+            int[] index = MathEx.permutate(n);
             for (int i = 0; i < n; i++) {
                 if (weight == null) {
                     process(x[index[i]], y[index[i]]);
