diff --git a/old_test.java b/new_test.java
index 959e843..e95fd73 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,6 @@
 			Assert.fail("Should not be here, an exception should have been thrown in the line above");
 		}
 		catch (APIException e) {
-			Assert.assertTrue(e.getMessage().startsWith("This method can only be called from the TimerSchedulerTask class"));
+			Assert.assertTrue(e.getMessage().startsWith("Scheduler.timer.task.only"));
 		}
 	}
\ No newline at end of file
