diff --git a/old_test.java b/new_test.java
index 2908944..66b37e8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
-@Test(groups={"tck"})
+@Test
     public void test_adjustInto() {
         assertEquals(DayOfWeek.MONDAY.adjustInto(LocalDate.of(2012, 9, 2)), LocalDate.of(2012, 8, 27));
         assertEquals(DayOfWeek.MONDAY.adjustInto(LocalDate.of(2012, 9, 3)), LocalDate.of(2012, 9, 3));
