diff --git a/old_product.java b/new_product.java
index afcdd23..1b8c37a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @Override
     public Temporal adjustInto(Temporal temporal) {
         return temporal
-                .with(OFFSET_SECONDS, getOffset().getTotalSeconds())
                 .with(EPOCH_DAY, toLocalDate().toEpochDay())
-                .with(NANO_OF_DAY, toLocalTime().toNanoOfDay());
+                .with(NANO_OF_DAY, toLocalTime().toNanoOfDay())
+                .with(OFFSET_SECONDS, getOffset().getTotalSeconds());
     }
\ No newline at end of file
