diff --git a/old_test.java b/new_test.java
index 9e7d6cf..6355e8b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,18 +1,13 @@
 @Test
     public void testGetProjectDescriptions() {
         // Create 2 projects.
-        Project foo = new Project();
-        foo.setPath("/foo");
-        foo.setName("foo");
-
-        Project bar = new Project();
-        bar.setPath("/bar");
-        bar.setName("bar");
+        Project foo = new Project("foo", "/foo");
+        Project bar = new Project("bar", "/bar");
 
         // Make the runtime environment aware of these two projects.
-        List<Project> projects = new ArrayList<>();
-        projects.add(foo);
-        projects.add(bar);
+        HashMap<String,Project> projects = new HashMap<>();
+        projects.put("foo", foo);
+        projects.put("bar", bar);
         RuntimeEnvironment env = RuntimeEnvironment.getInstance();
         env.setProjects(projects);
 
