diff --git a/old_test.java b/new_test.java
index c3a3307..cb695b9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -22,7 +22,7 @@
             if (!write.isDone()) {
                 val result1 = q.removeFinishedWrites();
                 Assert.assertEquals("Unexpected value from removeFinishedWrites when there were writes left in the queue.",
-                        WriteQueue.CleanupStatus.QueueNotEmpty, result1);
+                        WriteQueue.CleanupStatus.QueueNotEmpty, result1.getStatus());
                 val stats1 = q.getStatistics();
                 Assert.assertEquals("Unexpected size after removeFinishedWrites with no effect.", writes.size() + 1, stats1.getSize());
 
@@ -42,7 +42,7 @@
 
             val result2 = q.removeFinishedWrites();
             val expectedResult = writes.isEmpty() ? WriteQueue.CleanupStatus.QueueEmpty : WriteQueue.CleanupStatus.QueueNotEmpty;
-            Assert.assertEquals("Unexpected result from removeFinishedWrites.", expectedResult, result2);
+            Assert.assertEquals("Unexpected result from removeFinishedWrites.", expectedResult, result2.getStatus());
             val stats2 = q.getStatistics();
             Assert.assertEquals("Unexpected size after removeFinishedWrites.", writes.size(), stats2.getSize());
             Assert.assertEquals("Unexpected getExpectedProcessingTimeMillis after clear.", expectedElapsed, stats2.getExpectedProcessingTimeMillis());
@@ -54,6 +54,6 @@
         w3.fail(new IntentionalException(), true);
         val result3 = q.removeFinishedWrites();
         Assert.assertEquals("Unexpected value from removeFinishedWrites when there were failed writes.",
-                WriteQueue.CleanupStatus.WriteFailed, result3);
+                WriteQueue.CleanupStatus.WriteFailed, result3.getStatus());
 
     }
\ No newline at end of file
