diff --git a/old_product.java b/new_product.java
index 002bf78..9815263 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-synchronized CleanupStatus removeFinishedWrites() {
+synchronized CleanupResult removeFinishedWrites() {
         Exceptions.checkNotClosed(this.closed, this);
         long currentTime = this.timeSupplier.get();
         long totalElapsed = 0;
@@ -16,7 +16,8 @@ synchronized CleanupStatus removeFinishedWrites() {
             this.lastDurationMillis = (int) (totalElapsed / removedCount / AbstractTimer.NANOS_TO_MILLIS);
         }
 
-        return failedWrite
+        CleanupStatus status = failedWrite
                 ? CleanupStatus.WriteFailed
                 : this.writes.isEmpty() ? CleanupStatus.QueueEmpty : CleanupStatus.QueueNotEmpty;
+        return new CleanupResult(status, removedCount);
     }
\ No newline at end of file
