diff --git a/old_product.java b/new_product.java
index 8823815..a0a7335 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,8 +2,6 @@ private Iter list(final QueryContext ctx) throws QueryException {
     final String path = string(checkStr(expr[0], ctx));
     final File dir = new File(path);
 
-    // Check if directory exists
-    if(!dir.exists()) PATHNOTEXISTS.thrw(input, path);
     // Check if not a directory
     if(!dir.isDirectory()) NOTDIR.thrw(input, path);
 
