diff --git a/old_test.java b/new_test.java
index 802cff1..02d07b0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void testList() throws QueryException {
     final String fun = check(FunDef.FLIST);
-    error(fun + "('" + PATH1 + "')", Err.PATHNOTEXISTS);
+    error(fun + "('" + PATH1 + "')", Err.NOTDIR);
     query("file:write('" + PATH1 + "', ())");
     error(fun + "('" + PATH1 + "')", Err.NOTDIR);
     contains(fun + "('" + Prop.TMP + "')", NAME);
