diff --git a/old_product.java b/new_product.java
index fff4b0e..484f199 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public void initialize(DistributedLogNamespace namespace) throws DurableDataLogException {
+void initialize(DistributedLogNamespace namespace) throws DurableDataLogException {
         Preconditions.checkNotNull(namespace, "namespace");
         Preconditions.checkState(this.logManager == null, "LogHandle is already initialized.");
 
