diff --git a/old_test.java b/new_test.java
index 11a4741..ee42550 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,10 +4,10 @@
 		PackageResourceGuard guard = new PackageResourceGuard();
 
 		guard.setAllowAccessToRootResources(false);
-		assertFalse(guard.accept(Integer.TYPE, "test.gif"));
+		assertFalse(guard.accept("test.gif"));
 
 		guard.setAllowAccessToRootResources(true);
-		assertTrue(guard.accept(Integer.TYPE, "test.gif"));
+		assertTrue(guard.accept("test.gif"));
 
 
 	}
\ No newline at end of file
