diff --git a/old_test.java b/new_test.java
index c36c782..79f9029 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,4 +5,7 @@
     query("sum(1 to 3037000499)", "4611686016981624750");
     query("sum(1 to 3037000500)", "4611686020018625250");
     query("sum(1 to 4294967295)", "9223372034707292160");
+    query("sum((), ())", "");
+    error("sum(1, 'x')", SUM_X_X);
+    error("sum((), (1,2))", SEQFOUND_X);
   }
\ No newline at end of file
