diff --git a/old_product.java b/new_product.java
index 0e8f1ee..511fdd1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 Item sum(final Iter iter, final Item it, final boolean avg) throws QueryException {
-    Item rs = it.type.isUntyped() ? Dbl.get(it.string(info), info) : it;
+    Item rs = it.type.isUntyped() ? Dbl.get(it.dbl(info)) : it;
     final boolean num = rs instanceof ANum, dtd = rs.type == DTD, ymd = rs.type == YMD;
-    if(!num && (!(rs instanceof Dur) || rs.type == DUR)) throw SUM_X_X.get(info, rs.type, rs);
+    if(!num && !dtd && !ymd) throw SUM_X_X.get(info, rs.type, rs);
 
     int c = 1;
     for(Item i; (i = iter.next()) != null;) {
