diff --git a/old_test.java b/new_test.java
index 73508b0..08b435d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void test_toString_formatter() {
-        DateTimeFormatter f = DateTimeFormatters.pattern("M d");
+        DateTimeFormatter f = DateTimeFormatter.ofPattern("M d");
         String t = MonthDay.of(12, 3).toString(f);
         assertEquals(t, "12 3");
     }
\ No newline at end of file
