diff --git a/old_test.java b/new_test.java
index 8c93b3d..859d0b9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,6 @@
 @Test
-    public void testGetConcept() {
-        Map<String,Object> params = new HashMap<String, Object>();
-        params.put(ConceptInsights.ACCOUNT_ID, "wikipedia");
-        params.put("graph", "en-20120601");
-        params.put("concept", "IBM");
-        ConceptMetadata conceptMetaData =  service.getConcept(params);
-        Assert.assertNotNull(conceptMetaData);
+	public void testGetConcept() {
+		ConceptMetadata conceptMetaData = service.getConcept(EXAMPLE_CONCEPT);
+		Assert.assertNotNull(conceptMetaData);
 
-    }
\ No newline at end of file
+	}
\ No newline at end of file
