diff --git a/old_product.java b/new_product.java
index 7fd72d7..3bcce50 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public ConceptMetadata getConcept(Concept concept,String graphName) {
-        ConceptInsightsId.validateGenarate(concept, getAccountId(), graphName);
-        return executeRequest(version + concept.getId(), null, ConceptMetadata.class);
-    }
\ No newline at end of file
+public ConceptMetadata getConcept(final Concept concept) {
+		IDValidator.getConceptId(concept);
+		return executeRequest(API_VERSION + concept.getId(), null, ConceptMetadata.class);
+	}
\ No newline at end of file
