diff --git a/old_test.java b/new_test.java
index 20a777b..f685db3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
-	public void testGetDocumentRelationScores() {
-		List<Concept> concepts = new ArrayList<Concept>();
-		concepts.add(EXAMPLE_CONCEPT);
-		Scores scores = service.getDocumentRelationScores(EXAMPLE_DOCUMENT, concepts);
-		Assert.assertNotNull(scores);
-	}
\ No newline at end of file
+  public void testGetDocumentRelationScores() {
+    final List<Concept> concepts = new ArrayList<Concept>();
+    concepts.add(EXAMPLE_CONCEPT);
+    final Scores scores = service.getDocumentRelationScores(EXAMPLE_DOCUMENT, concepts);
+    Assert.assertNotNull(scores);
+  }
\ No newline at end of file
