diff --git a/old_test.java b/new_test.java
index ef095d0..a34c5fe 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,7 @@
 	public void getConcepts_shouldNotReturnConceptsWithMatchingNamesThatAreVoided() throws Exception {
 		Concept concept = dao.getConcept(7);
 		Context.getConceptService().updateConceptIndex(concept);
-		Assert.assertEquals(0, dao.getConcepts("VOIDED", null, false, new ArrayList<ConceptClass>(),
-		    new ArrayList<ConceptDatatype>()).size());
+		List<Concept> concepts = dao.getConcepts("VOIDED", null, false, new ArrayList<ConceptClass>(),
+		    new ArrayList<ConceptDatatype>());
+		Assert.assertEquals(0, concepts.size());
 	}
\ No newline at end of file
