diff --git a/old_product.java b/new_product.java
index ae5833d..9ed9ded 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,15 +12,12 @@
 				// throw new DAOException("Locale must be not null");
 				loc = Context.getLocale();
 			
-			String caseSensitiveNamesInConceptNameTable = Context.getAdministrationService().getGlobalProperty(
-			    OpenmrsConstants.GP_CASE_SENSITIVE_NAMES_IN_CONCEPT_NAME_TABLE, "true");
-			
 			criteria.createAlias("names", "names");
 			MatchMode matchmode = MatchMode.EXACT;
 			if (searchOnPhrase)
 				matchmode = MatchMode.ANYWHERE;
 			
-			if (Boolean.valueOf(caseSensitiveNamesInConceptNameTable)) {
+			if (Context.getAdministrationService().isDatabaseStringComparisonCaseSensitive()) {
 				criteria.add(Restrictions.ilike("names.name", name, matchmode));
 			} else {
 				if (searchOnPhrase) {
