diff --git a/old_product.java b/new_product.java
index 1cc7614..061063a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,4 @@
-@TestInfo(
-      level = TestLevel.COMPLETE,
-      purpose = "",
-      targets = {
-        @TestTarget(
-          methodName = "getInstance",
-          methodArgs = {String.class}
-        )
-    })
-    private void GetInstance01(int mode) throws NoSuchAlgorithmException,
+private void GetInstance01(int mode) throws NoSuchAlgorithmException,
             InvalidAlgorithmParameterException {
         try {
             KeyPairGenerator.getInstance(null);
