diff --git a/old_test.java b/new_test.java
index 36214f0..e600894 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,10 @@
-@TestInfo(
-      level = TestLevel.COMPLETE,
-      purpose = "",
-      targets = {
-        @TestTarget(
-          methodName = "getInstance",
-          methodArgs = {java.lang.String.class}
-        )
-    })
-    public void _testGetInstance01() throws NoSuchAlgorithmException,
+@TestTargetNew(
+        level = TestLevel.COMPLETE,
+        notes = "",
+        method = "getInstance",
+        args = {java.lang.String.class}
+    )
+    public void testGetInstance01() throws NoSuchAlgorithmException,
             InvalidAlgorithmParameterException, CertPathValidatorException {
         try {
             CertPathValidator.getInstance(null);
