diff --git a/old_product.java b/new_product.java
index 707f53f..a19e190 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,8 +8,8 @@ public static <T> double[] cv(int k, RegressionTrainer<T> trainer, T[] x, double
         
         CrossValidation cv = new CrossValidation(n, k);
         for (int i = 0; i < k; i++) {
-            T[] trainx = Math.slice(x, cv.train[i]);
-            double[] trainy = Math.slice(y, cv.train[i]);
+            T[] trainx = MathEx.slice(x, cv.train[i]);
+            double[] trainy = MathEx.slice(y, cv.train[i]);
             
             Regression<T> model = trainer.train(trainx, trainy);
 
