diff --git a/old_test.java b/new_test.java
index fea556d..b0a9db8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
             AttributeDataset data = parser.parse(smile.data.parser.IOUtils.getTestDataFile("weka/cpu.arff"));
             double[] y = data.toArray(new double[data.size()]);
             double[][] x = data.toArray(new double[data.size()][]);
-            Math.standardize(x);
+            MathEx.standardize(x);
 
             RBFNetwork.Trainer<double[]> trainer = new RBFNetwork.Trainer<>(new EuclideanDistance());
             trainer.setNumCenters(20);
