diff --git a/old_product.java b/new_product.java
index 1b08dd6..0ca4e57 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,10 @@
 @Override
-    public int deleteTerminatedClusters() {
-        return this.clusterRepository.deleteTerminatedClusters();
+    public long deleteTerminatedClusters() {
+        return this.clusterRepository.deleteByIdIn(
+            this.clusterRepository
+                .findTerminatedUnusedClusters()
+                .stream()
+                .map(Number::longValue)
+                .collect(Collectors.toSet())
+        );
     }
\ No newline at end of file
