diff --git a/old_test.java b/new_test.java
index f2a96d5..abcba3b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
 
         final int resultNumberOfSegmentsBeforeMerging = model.getSegments().size();
         final int resultNumberOfSmallSegmentsBeforeMerging =
-                model.countSmallSegments(SMALL_SEGMENT_TARGET_NUMBER_THRESHOLD);
+                countSmallSegments(model, SMALL_SEGMENT_TARGET_NUMBER_THRESHOLD);
         final int expectedNumberOfSegmentsBeforeMerging = 21;
         final int expectedNumberOfSmallSegmentsBeforeMerging = 15;
         Assert.assertEquals(resultNumberOfSegmentsBeforeMerging, expectedNumberOfSegmentsBeforeMerging);
