diff --git a/old_test.java b/new_test.java
index f56a168..6521969 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,13 @@
 @Test
 	public void down_2inputs() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
-		Class input[] = new Class[]{ImageUInt8.class, ImageUInt16.class,ImageFloat32.class, ImageFloat64.class};
-		Class middle[] = new Class[]{ImageFloat32.class, ImageFloat32.class,ImageFloat32.class, ImageFloat64.class};
+		Class input[] = new Class[]{GrayU8.class, GrayU16.class,GrayF32.class, GrayF64.class};
+		Class middle[] = new Class[]{GrayF32.class, GrayF32.class,GrayF32.class, GrayF64.class};
 
 		for (int i = 0; i < input.length; i++) {
-			ImageSingleBand in = GeneralizedImageOps.createSingleBand(input[i],17,14);
-			ImageSingleBand mid = GeneralizedImageOps.createSingleBand(middle[i],3,14);
-			ImageSingleBand found = GeneralizedImageOps.createSingleBand(input[i],3,4);
-			ImageSingleBand expected = GeneralizedImageOps.createSingleBand(input[i],3,4);
+			ImageGray in = GeneralizedImageOps.createSingleBand(input[i],17,14);
+			ImageGray mid = GeneralizedImageOps.createSingleBand(middle[i],3,14);
+			ImageGray found = GeneralizedImageOps.createSingleBand(input[i],3,4);
+			ImageGray expected = GeneralizedImageOps.createSingleBand(input[i],3,4);
 
 			GImageMiscOps.fillUniform(in,rand,0,100);
 
