diff --git a/old_test.java b/new_test.java
index 2d5ff15..9e4b9c7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
   public void modify() {
-    query("let $c := <x/> return $c !! ()", "<x/>");
-    query("let $c := <x/> return $c !! insert node <y/> into .", "<x><y/></x>");
+    query("let $c := <x/> return $c update ()", "<x/>");
+    query("let $c := <x/> return $c update insert node <y/> into .", "<x><y/></x>");
   }
\ No newline at end of file
