diff --git a/old_product.java b/new_product.java
index 581dead..c925bf0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 private Expr modify() throws QueryException {
     final Expr e = comparison();
     if(e != null) {
-      if(wsConsumeWs("!!")) {
+      if(wsConsumeWs(UPDATE)) {
         final int s = scope.open();
         final boolean u = ctx.updating;
         ctx.updating(false);
