diff --git a/old_product.java b/new_product.java
index 4496a6c..980f2da 100644
--- a/old_product.java
+++ b/new_product.java
@@ -21,6 +21,10 @@ public static String linkToHtml(final String markdownText) {
 
                 if (MARKED_AVAILABLE) {
                     ret = toHtmlByMarked(markdownText);
+
+                    if (!StringUtils.startsWith(ret, "<p>")) {
+                        ret = "<p>" + ret + "</p>";
+                    }
                 } else {
                     final PegDownProcessor pegDownProcessor
                             = new PegDownProcessor(Extensions.ALL_OPTIONALS | Extensions.ALL_WITH_OPTIONALS);
@@ -32,9 +36,9 @@ public static String linkToHtml(final String markdownText) {
                     if (!StringUtils.startsWith(ret, "<p>")) {
                         ret = "<p>" + ret + "</p>";
                     }
-                }
 
-                ret = formatMarkdown(ret);
+                    ret = formatMarkdown(ret);
+                }
 
                 // cache it
                 putHTML(markdownText, ret);
