diff --git a/old_test.java b/new_test.java
index 195aafd..5d611c6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -48,8 +48,8 @@
                 expectedDeletedSegmentNames.add(m.getStreamSegmentMetadata(transactionId).getName());
             }
 
-            Collection<String> deletedSegmentNames = m.deleteStreamSegment(name);
-            AssertExtensions.assertContainsSameElements("Unexpected StreamSegments were deleted.", expectedDeletedSegmentNames, deletedSegmentNames);
+            Map<Long, String> deletedSegmentNames = m.deleteStreamSegment(name);
+            AssertExtensions.assertContainsSameElements("Unexpected StreamSegments were deleted.", expectedDeletedSegmentNames, deletedSegmentNames.values());
         }
 
         // Delete Transactions.
@@ -60,8 +60,8 @@
             deletedStreamSegmentIds.add(transactionId);
             expectedDeletedSegmentNames.add(name);
 
-            Collection<String> deletedSegmentNames = m.deleteStreamSegment(name);
-            AssertExtensions.assertContainsSameElements("Unexpected StreamSegments were deleted.", expectedDeletedSegmentNames, deletedSegmentNames);
+            Map<Long, String> deletedSegmentNames = m.deleteStreamSegment(name);
+            AssertExtensions.assertContainsSameElements("Unexpected StreamSegments were deleted.", expectedDeletedSegmentNames, deletedSegmentNames.values());
         }
 
         // Verify deleted segments have not been actually removed from the metadata.
