diff --git a/old_product.java b/new_product.java
index 285e9f5..36779d6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,9 @@
 private Str textEntry(final QueryContext ctx) throws QueryException {
-    return Str.get(entry(ctx));
+    final String enc = expr.length < 3 ? null : string(checkStr(expr[2], ctx));
+    final IO io = new IOContent(entry(ctx));
+    try {
+      return Str.get(TextInput.content(io, enc).finish());
+    } catch(final IOException ex) {
+      throw ZIPFAIL.thrw(input, ex.getMessage());
+    }
   }
\ No newline at end of file
