diff --git a/old_test.java b/new_test.java
index 456420b..9024239 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,11 @@
 @Test
   public void testTextEntry() throws BaseXException {
-    final String fun = check(FunDef.TEXTENTRY, Uri.class, String.class);
-    query(fun + "(xs:anyURI('" + ZIP + "'), '" + ENTRY1 + "')");
+    final String fun = check(FunDef.TEXTENTRY,
+        String.class, String.class, String.class);
+    query(fun + "('" + ZIP + "', '" + ENTRY1 + "')");
+    query(fun + "('" + ZIP + "', '" + ENTRY1 + "', 'US-ASCII')");
+    error(fun + "('" + ZIP + "', '" + ENTRY1 + "', 'xyz')",
+        Err.ZIPFAIL);
     // newlines are removed from the result..
-    contains(fun + "(xs:anyURI('" + ZIP + "'), '" + ENTRY1 + "')", "aaboutab");
+    contains(fun + "('" + ZIP + "', '" + ENTRY1 + "')", "aaboutab");
   }
\ No newline at end of file
