diff --git a/old_test.java b/new_test.java
index a626340..512a462 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
-    public void testSetSource()
+    public void testSetSource() throws SQLException
     {
         String source = "new source";
-        bs.setSource(source);
+        bs.setSource(context, source);
         assertThat("testSetSource 0", bs.getSource(), notNullValue());
         assertThat("testSetSource 1", bs.getSource(), not(equalTo("")));
         assertThat("testSetSource 2", bs.getSource(), equalTo(source));
