diff --git a/old_product.java b/new_product.java
index c7c38dc..2ac4e68 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public void setSource(String n) {
-        setMetadataSingleValue(MetadataSchema.DC_SCHEMA, "source", null, null, n);
+public void setSource(Context context, String n) throws SQLException {
+        getBitstreamService().setMetadataSingleValue(context, this, MetadataSchema.DC_SCHEMA, "source", null, null, n);
     }
\ No newline at end of file
