diff --git a/old_product.java b/new_product.java
index aa26256..cc401ac 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,11 @@
 public void setJobStatus(final JobStatus jobStatus) {
         this.status = jobStatus;
 
-        if (jobStatus == Types.JobStatus.INIT) {
+        if (jobStatus == JobStatus.INIT) {
             setStartTime(System.currentTimeMillis());
-        } else if (jobStatus == Types.JobStatus.SUCCEEDED
-                || jobStatus == Types.JobStatus.KILLED
-                || jobStatus == Types.JobStatus.FAILED) {
+        } else if (jobStatus == JobStatus.SUCCEEDED
+                || jobStatus == JobStatus.KILLED
+                || jobStatus == JobStatus.FAILED) {
             setFinishTime(System.currentTimeMillis());
         }
     }
\ No newline at end of file
