diff --git a/old_product.java b/new_product.java
index 4f90d90..3b8ab4c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 @Override
-    public default D minus(long amountToSubtract, TemporalUnit unit) {
+    default D minus(long amountToSubtract, TemporalUnit unit) {
         return (D) getChronology().ensureChronoLocalDate(Temporal.super.minus(amountToSubtract, unit));
     }
\ No newline at end of file
