diff --git a/old_product.java b/new_product.java
index 9bffe2a..79fcc6c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@
 	public void purgeOrderFrequency(OrderFrequency orderFrequency) {
 		
 		if (dao.isOrderFrequencyInUse(orderFrequency)) {
-			throw new APIException("This order frequency cannot be deleted because it is already in use");
+			throw new APIException("Order.frequency.cannot.delete", (Object[]) null);
 		}
 		
 		dao.purgeOrderFrequency(orderFrequency);
