diff --git a/old_test.java b/new_test.java
index 5e0141a..2436b83 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,6 @@
 		assertNotNull(orderFrequency);
 		
 		expectedException.expect(APIException.class);
-		expectedException.expectMessage("This order frequency cannot be deleted because it is already in use");
+		expectedException.expectMessage("Order.frequency.cannot.delete");
 		orderService.purgeOrderFrequency(orderFrequency);
 	}
\ No newline at end of file
