diff --git a/old_test.java b/new_test.java
index 7b0ccf0..928e9e4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,8 +17,8 @@
 			assertTrue(isNotBlank(item.getLabel()));
 			assertNotNull(item.getKind() );
 			assertTrue(isNotBlank(item.getSortText()));
-			//text edits are not set during calls to "completion"
-			assertNull(item.getTextEdit());
+			//text edits are set during calls to "completion"
+			assertNotNull(item.getTextEdit());
 			assertTrue(isNotBlank(item.getInsertText()));
 			assertNotNull(item.getFilterText());
 			assertFalse(item.getFilterText().contains(" "));
