diff --git a/old_product.java b/new_product.java
index 0416d02..980bff3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,14 +1,14 @@
 protected Connection obtainInternalConnection() throws SQLException {
 		boolean tryAgain = false;
 		Connection result = null;
+		
 		int acquireRetryAttempts = this.pool.getConfig().getAcquireRetryAttempts();
 		int acquireRetryDelay = this.pool.getConfig().getAcquireRetryDelay();
 		this.connectionHook = this.pool.getConfig().getConnectionHook();
 		do{ 
 			try { 
 				// keep track of this hook.
-
-				this.connection = DriverManager.getConnection(this.url, this.username, this.password);
+				this.connection = this.pool.obtainRawInternalConnection();
 				tryAgain = false;
 
 				if (acquireRetryAttempts != this.pool.getConfig().getAcquireRetryAttempts()){
