diff --git a/old_product.java b/new_product.java
index cb6cfa2..02a82ac 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,6 +11,6 @@ public Instant truncatedTo(TemporalUnit unit) {
             throw new DateTimeException("Unit must divide into a standard day without remainder");
         }
         long nod = (seconds % LocalTime.SECONDS_PER_DAY) * LocalTime.NANOS_PER_SECOND + nanos;
-        long result = (nod / dur) * dur;
+        long result = Jdk8Methods.floorDiv(nod, dur) * dur;
         return plusNanos(result - nod);
     }
\ No newline at end of file
