diff --git a/old_test.java b/new_test.java
index b37da8e..cd52a75 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
-@Test(groups={"tck"})
+@Test
     public void test_truncatedTo_normal() {
         assertEquals(TEST_2008_6_30_11_30_59_000000500.truncatedTo(NANOS), TEST_2008_6_30_11_30_59_000000500);
         assertEquals(TEST_2008_6_30_11_30_59_000000500.truncatedTo(SECONDS), TEST_2008_6_30_11_30_59_000000500.withNano(0));
