diff --git a/old_product.java b/new_product.java
index a19985e..efb9532 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 protected Set<ServiceComponentResponse> getComponents(
       Set<ServiceComponentRequest> requests) throws AmbariException {
-    Set<ServiceComponentResponse> response =
-        new HashSet<ServiceComponentResponse>();
+
+    Set<ServiceComponentResponse> response = new HashSet<ServiceComponentResponse>();
     for (ServiceComponentRequest request : requests) {
       try {
         response.addAll(getComponents(request));
@@ -11,12 +11,7 @@ protected Set<ServiceComponentResponse> getComponents(
           // there will be > 1 request in case of OR predicate
           throw e;
         }
-      } catch (ParentObjectNotFoundException ee) {
-        if (requests.size() == 1) {
-          throw ee;
-        }
       }
-      
     }
     return response;
   }
\ No newline at end of file
