diff --git a/old_product.java b/new_product.java
index 745e5ee..35300e4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,18 +1,3 @@
-public LocalDate plus(PeriodProvider periodProvider) {
-        Period period = Period.ofDateFields(periodProvider);
-        long periodMonths = period.totalMonths();
-        long periodDays = period.getDays();
-        if (periodMonths == 0) {
-            return plusDays(periodDays);  // optimization that also returns this for zero
-        }
-        long monthCount = ((long) year) * 12 + (month.getValue() - 1);
-        long calcMonths = monthCount + periodMonths;  // safe overflow
-        int newYear = YEAR.checkValidIntValue(DateTimes.floorDiv(calcMonths, 12));
-        MonthOfYear newMonth = MonthOfYear.of(DateTimes.floorMod(calcMonths, 12) + 1);
-        int newMonthLen = newMonth.lengthInDays(isLeapYear(newYear));
-        int newDay = Math.min(day, newMonthLen);
-        if (periodDays < 0 && day > newMonthLen) {
-            periodDays = Math.min(periodDays + (day - newMonthLen), 0);  // adjust for invalid days
-        }
-        return LocalDate.of(newYear, newMonth, newDay).plusDays(periodDays);
+public LocalDate plus(Period period) {
+        return plus(period.getAmount(), period.getUnit());
     }
\ No newline at end of file
