diff --git a/old_product.java b/new_product.java
index 0bd1403..a6aca21 100644
--- a/old_product.java
+++ b/new_product.java
@@ -18,7 +18,7 @@ public synchronized void addFeature(TransactionTrace txn, String key, Object val
             }
             if (debug.get()) LOG.debug("Adding new attribute " + key + " [" + type + "]");
             this.attributes.put(key, type);
-            this.attribute_histograms.put(key, new Histogram());
+            this.attribute_histograms.put(key, new ObjectHistogram());
             this.attribute_types.put(key, VoltType.NULL);
         }
         // HACK
