diff --git a/old_test.java b/new_test.java
index 2cd2400..09c30c1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,7 +13,7 @@
 			HU2PackingItemsAllocator.builder()
 					.itemToPack(itemToPack)
 					.packingItems(packingItems)
-					.fromHUs(luHUs)
+					.pickFromHUs(luHUs)
 					.allocate();
 
 			// Validate
@@ -40,7 +40,7 @@
 			HU2PackingItemsAllocator.builder()
 					.itemToPack(itemToPack)
 					.packingItems(packingItems)
-					.fromHUs(luHUs)
+					.pickFromHUs(luHUs)
 					.allocate();
 
 			// Validate
@@ -67,7 +67,7 @@
 			HU2PackingItemsAllocator.builder()
 					.itemToPack(itemToPack)
 					.packingItems(packingItems)
-					.fromHUs(luHUs)
+					.pickFromHUs(luHUs)
 					.allocate();
 
 			// Validate
