diff --git a/old_test.java b/new_test.java
index f9a8186..4611f26 100644
--- a/old_test.java
+++ b/new_test.java
@@ -26,4 +26,8 @@
     // We also expect the spans to have been reported
     assertThat(spans).flatExtracting(Span::annotations).extracting(Annotation::value)
         .containsExactly("brave.flush", "brave.flush");
+
+    // We also expect the spans reported to have the endpoint of the tracer
+    assertThat(spans).extracting(Span::localEndpoint)
+        .containsExactly(endpoint, endpoint);
   }
\ No newline at end of file
