diff --git a/old_product.java b/new_product.java
index b7cdfae..2f8e24a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ void reportOrphanedSpans() {
       if (value == null || noop.get()) continue;
       try {
         value.annotate(value.clock.currentTimeMicroseconds(), "brave.flush");
-        reporter.report(value.toSpan());
+        reporter.report(value.toSpan(endpoint));
       } catch (RuntimeException e) {
         // don't crash the caller if there was a problem reporting an unrelated span.
         if (context != null && logger.isLoggable(Level.FINE)) {
