diff --git a/old_test.java b/new_test.java
index 9132b02..d8f4b18 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,8 @@
     when(mSessions.getTimedOutSessions()).thenReturn(sessions);
     Whitebox.invokeMethod(mBlockWorker, "setupSessionCleaner");
     SessionCleaner cleaner = Whitebox.getInternalState(mBlockWorker, "mSessionCleaner");
-    SessionCleanable cleanable = Whitebox.getInternalState(cleaner, "mSessionCleanable");
+    SessionCleanupCallback cleanable =
+        Whitebox.getInternalState(cleaner, "mSessionCleanupCallback");
     cleanable.cleanupSessions();
     verify(mSessions).removeSession(sessionId);
     verify(mBlockStore).cleanupSession(sessionId);
