diff --git a/old_product.java b/new_product.java
index 72df369..52e2780 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 private void setupSessionCleaner() {
-    mSessionCleaner = new SessionCleaner(new SessionCleanable() {
+    mSessionCleaner = new SessionCleaner(new SessionCleanupCallback() {
       /**
        * Cleans up after sessions, to prevent zombie sessions holding local resources.
        */
