diff --git a/old_product.java b/new_product.java
index f910650..204ee7f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,8 +11,8 @@ public RoutingRequestBuilder addTripParameters(TripParameters tripParams) {
             setArriveBy(tripParams.getArrive_by());
         }
 
-        setOrigin(tripParams.getOrigin().getLat_lng());
-        setDestination(tripParams.getDestination().getLat_lng());
+        setOrigin(tripParams.getOrigin());
+        setDestination(tripParams.getDestination());
 
         return this;
     }
\ No newline at end of file
