diff --git a/old_test.java b/new_test.java
index 5e2da4c..019803a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,5 +11,4 @@
         Assert.assertNotNull(this.job.getClusterCriteriasString());
         Assert.assertNotNull(this.job.getCommandCriteriaString());
         Assert.assertTrue(this.job.getTags().contains(this.job.getId()));
-        Assert.assertTrue(this.job.getTags().contains(NAME));
     }
\ No newline at end of file
