diff --git a/old_test.java b/new_test.java
index c7e44f0..0728a20 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
-	@Verifies(value = "should void membership for every patient voided", method = "patientVoided(Patient)")
 	public void patientVoided_shouldVoidMemberships() throws Exception {
 		executeDataSet(COHORT_XML);
 
@@ -10,9 +9,11 @@
 		voidedPatient.setVoidedBy(Context.getAuthenticatedUser());
 		voidedPatient.setVoidReason("Voided as a result of the associated patient getting voided");
 
-		CohortMembership newMemberContainingVoidedPatient = new CohortMembership(voidedPatient);
+		CohortMembership newMemberContainingVoidedPatient = new CohortMembership(voidedPatient.getPatientId());
 		cohort.addMembership(newMemberContainingVoidedPatient);
 		assertTrue(cohort.contains(voidedPatient));
+		
+		assertEquals(service.getCohortsContainingPatient(voidedPatient).size(), 1);
 
 		service.patientVoided(voidedPatient);
 		assertTrue(newMemberContainingVoidedPatient.getVoided());
