diff --git a/old_product.java b/new_product.java
index 3821fb8..3548778 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,15 @@
-public void patientVoided(Patient patient) throws APIException {
+@Override
+	public void patientVoided(Patient patient) throws APIException {
 		List<Cohort> cohorts = Context.getCohortService().getCohortsContainingPatient(patient);
 		for (Cohort cohort : cohorts) {
-			List<CohortMembership> membersToVoid = cohort.getMembers().stream()
-					.filter(m -> !m.getVoided() && patient.getVoided() && m.getPatient().getPatientId().equals(patient.getPatientId()))
-					.collect(Collectors.toList());
-			for (CohortMembership member : membersToVoid) {
-				member.setVoided(patient.getVoided());
-				member.setDateVoided(patient.getDateVoided());
-				member.setVoidedBy(patient.getVoidedBy());
-				member.setVoidReason(patient.getVoidReason());
+			if (patient.getVoided()) {
+				List<CohortMembership> memberships = cohort.getMemberships(false);
+				memberships.forEach(m -> {
+					m.setVoided(patient.getVoided());
+					m.setDateVoided(patient.getDateVoided());
+					m.setVoidedBy(patient.getVoidedBy());
+					m.setVoidReason(patient.getVoidReason());
+				});
 			}
-			cohort.removeMember(patient);
-			Context.getCohortService().saveCohort(cohort);
 		}
 	}
\ No newline at end of file
