diff --git a/old_product.java b/new_product.java
index 54cf446..d17abb4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,9 @@
-public List<TempBlockMeta> cleanupUser(long userId) {
-    List<TempBlockMeta> blocksToRemove = new ArrayList<TempBlockMeta>();
+public void cleanupUser(long userId) {
     Set<Long> userTempBlocks = mUserIdToTempBlockIdsMap.get(userId);
     if (userTempBlocks != null) {
       for (long blockId : userTempBlocks) {
         TempBlockMeta tempBlock = mBlockIdToTempBlockMap.remove(blockId);
         if (tempBlock != null) {
-          blocksToRemove.add(tempBlock);
           reclaimSpace(tempBlock.getBlockSize(), false);
         } else {
           LOG.error("Cannot find blockId {} when cleanup userId {}", blockId, userId);
@@ -13,5 +11,4 @@ public List<TempBlockMeta> cleanupUser(long userId) {
       }
       mUserIdToTempBlockIdsMap.remove(userId);
     }
-    return blocksToRemove;
   }
\ No newline at end of file
