diff --git a/old_product.java b/new_product.java
index 9fd3d61..54cf446 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ public List<TempBlockMeta> cleanupUser(long userId) {
         TempBlockMeta tempBlock = mBlockIdToTempBlockMap.remove(blockId);
         if (tempBlock != null) {
           blocksToRemove.add(tempBlock);
-          reclaimSpace(tempBlock.getBlockSize());
+          reclaimSpace(tempBlock.getBlockSize(), false);
         } else {
           LOG.error("Cannot find blockId {} when cleanup userId {}", blockId, userId);
         }
