diff --git a/old_test.java b/new_test.java
index 4b95167..3d73885 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,11 +10,11 @@
     long otherUserId = TEST_USER_ID + 1;
 
     TempBlockMeta tempBlockMeta1 =
-        new TempBlockMeta(TEST_USER_ID, tempBlockId1, TEST_BLOCK_SIZE, mDir);
+        new TempBlockMeta(TEST_USER_ID, tempBlockId1, TEST_TEMP_BLOCK_SIZE, mDir);
     TempBlockMeta tempBlockMeta2 =
-        new TempBlockMeta(TEST_USER_ID, tempBlockId2, TEST_BLOCK_SIZE, mDir);
+        new TempBlockMeta(TEST_USER_ID, tempBlockId2, TEST_TEMP_BLOCK_SIZE, mDir);
     TempBlockMeta tempBlockMeta3 =
-        new TempBlockMeta(otherUserId, tempBlockId3, TEST_BLOCK_SIZE, mDir);
+        new TempBlockMeta(otherUserId, tempBlockId3, TEST_TEMP_BLOCK_SIZE, mDir);
     mDir.addTempBlockMeta(tempBlockMeta1);
     mDir.addTempBlockMeta(tempBlockMeta2);
     mDir.addTempBlockMeta(tempBlockMeta3);
