diff --git a/old_test.java b/new_test.java
index 5631ee6..90cd53c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,10 @@
-@Test(groups="eventbus")
-   public final void testNotify()
+@Test(groups = "eventbus")
+   public final void testNotify() throws Exception
    {
-      //TODO Implement
+      AnObserver observerInstance = new AnObserver();
+      Observer<Event> observer = new MockObserverImpl<Event>(tuna, om, Event.class);
+      ((MockObserverImpl<Event>) observer).setInstance(observerInstance);
+      Event event = new Event();
+      observer.notify(manager, event);
+      assert observerInstance.notified;
    }
\ No newline at end of file
