diff --git a/old_product.java b/new_product.java
index 240210d..765b211 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,7 @@
-public void notify(Manager container, T event)
+public void notify(Manager manager, T event)
    {
       // Get the most specialized instance of the component
-      Object instance = container.getInstanceByType(compModel.getType(),
-            compModel.getBindingTypes().toArray(new Annotation[0]));
+      Object instance = getInstance(manager);
       if (instance != null)
-         this.observerMethod.invoke(container, instance, event);
+         this.observerMethod.invoke(manager, instance, event);
    }
\ No newline at end of file
