diff --git a/old_test.java b/new_test.java
index 3d8f49b..57aae24 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,21 +4,21 @@
 
 		SimpleMatrix E = SimpleMatrix.wrap(constructE(x,y,z,w));
 
-		DenseMatrix64F A = new DenseMatrix64F(10,10);
-		DenseMatrix64F B = new DenseMatrix64F(10,10);
+		RowMatrix_F64 A = new RowMatrix_F64(10,10);
+		RowMatrix_F64 B = new RowMatrix_F64(10,10);
 
 		HelperNister5 alg = new HelperNister5();
 		alg.setNullSpace(X,Y,Z,W);
 		alg.setupA1(A);
 		alg.setupA2(B);
 
-		DenseMatrix64F Y1 = new DenseMatrix64F(10,1);
-		DenseMatrix64F Y2 = new DenseMatrix64F(10,1);
+		RowMatrix_F64 Y1 = new RowMatrix_F64(10,1);
+		RowMatrix_F64 Y2 = new RowMatrix_F64(10,1);
 
 		CommonOps_D64.mult(A,createCoefsA(x,y,z),Y1);
 		CommonOps_D64.mult(B,createCoefsB(x, y, z),Y2);
 
-		DenseMatrix64F Y = new DenseMatrix64F(10,1);
+		RowMatrix_F64 Y = new RowMatrix_F64(10,1);
 
 		CommonOps_D64.add(Y1,Y2,Y);
 
@@ -26,7 +26,7 @@
 		SimpleMatrix EEt = E.mult(E.transpose());
 		SimpleMatrix EEtE = EEt.mult(E);
 		SimpleMatrix aE = E.scale(-0.5*EEt.trace());
-		DenseMatrix64F eq2 = EEtE.plus(aE).matrix_F64();
+		RowMatrix_F64 eq2 = EEtE.plus(aE).matrix_F64();
 
 		// check the solution
 		assertEquals(E.determinant(),Y.data[0],1e-8);
