diff --git a/old_test.java b/new_test.java
index 4e43f50..ed2ad6b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
-@Test(groups = "examples-testng", description = "Example of using TestNGCucumberRunner to invoke Cucumber")
-    public void runCukes() {
-        new TestNGCucumberRunner(getClass()).runCukes();
+@Test(groups = "cucumber", description = "Runs Cucumber Feature", dataProvider = "scenarios")
+    public void scenario(PickleEventWrapper pickleEvent, CucumberFeatureWrapper cucumberFeature) throws Throwable {
+        testNGCucumberRunner.runScenario(pickleEvent.getPickleEvent());
     }
\ No newline at end of file
