diff --git a/old_product.java b/new_product.java
index 03a7bae..8153964 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
 public void runCukes() {
+        System.err.println("WARNING: The TestNGCucumberRunner.runCukes() is deprecated. Please create a runner class by subclassing AbstractTestNGCucumberTest.");
         for (CucumberFeature cucumberFeature : getFeatures()) {
             reporter.uri(cucumberFeature.getPath());
             runtime.runFeature(cucumberFeature);
