diff --git a/old_test.java b/new_test.java
index 8907c62..ffde342 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,7 +18,7 @@
 		orderService.saveOrder(revise, null);
 		Thread.sleep(10);
 		
-		expectedException.expect(APIException.class);
+		expectedException.expect(CannotUnvoidOrderException.class);
 		expectedException.expectMessage(
 		    Context.getMessageSourceService().getMessage("Order.action.cannot.unvoid", new Object[] { "revision" }, null));
 		orderService.unvoidOrder(order);
