diff --git a/old_product.java b/new_product.java
index 813d174..e1698bb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ public Order unvoidOrder(Order order) throws APIException {
 		if (previousOrder != null && isDiscontinueOrReviseOrder(order)) {
 			if (!previousOrder.isActive()) {
 				final String action = DISCONTINUE == order.getAction() ? "discontinuation" : "revision";
-				throw new APIException("Order.action.cannot.unvoid", new Object[] { action });
+				throw new CannotUnvoidOrderException(action);
 			}
 			stopOrder(previousOrder, aMomentBefore(order.getDateActivated()), false);
 		}
