diff --git a/old_product.java b/new_product.java
index cb8897b..d06d6ae 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 public static <T> Write<T> write(Class<T> recordClass) {
-    return AvroIO.<T>defaultWriteBuilder()
-        .setRecordClass(recordClass)
-        .setSchema(ReflectData.get().getSchema(recordClass))
-        .build();
+    return new Write<>(
+        AvroIO.<T, T>defaultWriteBuilder()
+            .setGenericRecords(false)
+            .setSchema(ReflectData.get().getSchema(recordClass))
+            .build());
   }
\ No newline at end of file
