diff --git a/old_test.java b/new_test.java
index ac124e2..9e2c4cf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,9 +8,5 @@
             "Intimidating pigeon",
             "Pedantic gull",
             "Frisky finch");
-    runWrite(
-        inputs,
-        IDENTITY_MAP,
-        getBaseOutputFilename(),
-        WriteFiles.to(makeSimpleSink(), SerializableFunctions.<String>identity()));
+    runWrite(inputs, IDENTITY_MAP, getBaseOutputFilename(), WriteFiles.to(makeSimpleSink()));
   }
\ No newline at end of file
