diff --git a/old_product.java b/new_product.java
index f497831..d4d215d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,14 +3,8 @@ public QrCode numeric(int[] numbers ) {
 			if( numbers[i] < 0 || numbers[i] > 9 )
 				throw new IllegalArgumentException("All numbers must have a value from 0 to 9");
 		}
-		int lengthBits;
-
-		if (qr.version < 10)
-			lengthBits = 10;
-		else if (qr.version < 27)
-			lengthBits = 12;
-		else
-			lengthBits = 14;
+		qr.mode = QrCode.Mode.NUMERIC;
+		int lengthBits = getLengthBitsNumeric(qr.version);
 
 		packed.resize(lengthBits + 10*(numbers.length/3)); // predeclare memory
 		packed.size = 0; // set size to 0 so that append() starts from the front
