diff --git a/old_test.java b/new_test.java
index 3e5f7fd..a999d60 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,8 +14,8 @@
 
 		QrCodeEncoder.flipBits8(expected,expected.length);
 
-		assertEquals(qr.dataRaw.length,expected.length);
+		assertEquals(qr.rawbits.length,expected.length);
 		for (int i = 0; i < expected.length; i++) {
-			assertEquals(qr.dataRaw[i],expected[i]);
+			assertEquals(qr.rawbits[i],expected[i]);
 		}
 	}
\ No newline at end of file
