diff --git a/old_test.java b/new_test.java
index ce7fea5..b0d44a0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,7 +13,8 @@
         DecisionTreeRegressionTrainer trainer = new DecisionTreeRegressionTrainer(1, 0);
 
         DecisionTreeNode tree = trainer.fit(
-            new LocalDatasetBuilder<>(data, parts),
+            data,
+            parts,
             (k, v) -> Arrays.copyOf(v, v.length - 1),
             (k, v) -> v[v.length - 1]
         );
