diff --git a/old_product.java b/new_product.java
index 53030d3..f82ab16 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 public List<List<Span>> getTraces() {
-    List<Long> traceIds = store.traceIds();
+    List<Long> traceIds = storage.spanStore().traceIds();
     List<List<Span>> result = new ArrayList<>(traceIds.size());
     for (long traceId : traceIds) {
-      result.add(store.getTrace(traceId));
+      result.add(storage.spanStore().getTrace(traceId));
     }
     return result;
   }
\ No newline at end of file
