diff --git a/old_product.java b/new_product.java
index 4b7be73..c7d54bd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public boolean isIdentical (RunTable that)
+boolean isIdentical (RunTable that)
     {
         // Check null entities
         if (that == null) {
@@ -9,11 +9,11 @@ public boolean isIdentical (RunTable that)
             && (this.width == that.width)
             && (this.height == that.height)) {
             // Check run sequences, row by row
-            for (int row = 0; row < getSize(); row++) {
-                RunSequence thisSeq = getSequence(row);
-                RunSequence thatSeq = that.getSequence(row);
+            for (int row = 0, size = getSize(); row < size; row++) {
+                short[] thisSeq = getSequence(row);
+                short[] thatSeq = that.getSequence(row);
 
-                if (!thisSeq.equals(thatSeq)) {
+                if (!Arrays.equals(thisSeq, thatSeq)) {
                     return false;
                 }
             }
