diff --git a/old_test.java b/new_test.java
index 6665a56..2c3516a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,7 +10,7 @@
 				.endLine()
 				.build();
 
-		final List<PartitionerConfigReference> referencingLines = config.getReferences("mno"); // also need to work with different case
+		final List<PartitionerConfigReference> referencingLines = config.getReferences("MNO");
 
 		assertThat(referencingLines.size(), is(2));
 
