diff --git a/old_product.java b/new_product.java
index 33e8bee..71831df 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public List<PartitionerConfigReference> getReferences(final String tableName)
 	{
 		final List<PartitionerConfigReference> references = getLines().stream()
 				.flatMap(line -> line.getReferences().stream()) // get a stream of all references
-				.filter(ref -> tableName.equalsIgnoreCase(ref.getReferencedTableName())) // filter those who refer to 'tablename'
+				.filter(ref -> tableName.equals(ref.getReferencedTableName())) // filter those who refer to 'tablename'
 				.collect(Collectors.toList());
 		return references;
 	}
\ No newline at end of file
