diff --git a/old_test.java b/new_test.java
index e864af3..34d5611 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     model = createVoiceModel();
     final List<CustomTranslation> expected = instantiateCustomTranslations();
 
-    service.saveWords(model, expected.toArray(new CustomTranslation[] { })).execute();
+    service.addWords(model, expected.toArray(new CustomTranslation[] { })).execute();
 
     final CustomTranslation word = service.getWord(model, expected.get(0).getWord()).execute();
     assertEquals(expected.get(0).getTranslation(), word.getTranslation());
