diff --git a/old_product.java b/new_product.java
index d4fd0f6..a588280 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
-public ServiceCall<CustomTranslation> getWord(CustomVoiceModel model, String word) {
-    Validator.notNull(model.getId(), "model id cannot be null");
+public ServiceCall<CustomTranslation> getWord(final CustomVoiceModel model, final String word) {
+    Validator.notNull(model, "model cannot be null");
+    Validator.notEmpty(model.getId(), "model id must not be empty");
     Validator.notNull(word, "word cannot be null");
 
     final Request request = RequestBuilder.get(String.format(PATH_WORD, model.getId(), word)).build();
