diff --git a/old_test.java b/new_test.java
index 10487f5..697941d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test(groups={"tck"})
     public void test_minusSeconds_one() {
         LocalDateTime t = TEST_2007_07_15_12_30_40_987654321.with(LocalTime.MIDNIGHT);
-        LocalDate d = t.toLocalDate().minusDays(1);
+        LocalDate d = t.getDate().minusDays(1);
 
         int hour = 0;
         int min = 0;
@@ -24,7 +24,7 @@
                 }
             }
 
-            assertEquals(t.toLocalDate(), d);
+            assertEquals(t.getDate(), d);
             assertEquals(t.getHour(), hour);
             assertEquals(t.getMinute(), min);
             assertEquals(t.getSecond(), sec);
