diff --git a/old_product.java b/new_product.java
index 39bdc22..214ad69 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 ChronoZonedDateTime<C> minusSeconds(long seconds) {
-        ChronoDateTime newDT = dateTime.toDateTime().minusSeconds(seconds);
-        return (newDT == dateTime.toDateTime() ? this :
+        ChronoDateTime newDT = dateTime.getDateTime().minusSeconds(seconds);
+        return (newDT == dateTime.getDateTime() ? this :
             resolve(newDT, zone, dateTime, ZoneResolvers.retainOffset()));
     }
\ No newline at end of file
