diff --git a/old_test.java b/new_test.java
index 93b885d..1559231 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,7 +15,7 @@
       Assert.fail("exception should be raised for adding to buffer which does not have room");
     }
     catch (Exception bue) {
-      assert (bue instanceof BufferOverflowException);
+      assert (bue instanceof IllegalStateException);
     }
 
     instance = new CircularBuffer<Integer>(10);
@@ -33,7 +33,7 @@
       Assert.fail("exception should have been thrown");
     }
     catch (Exception e) {
-      assert (e instanceof BufferOverflowException);
+      assert (e instanceof IllegalStateException);
       instance.remove();
       instance.add(new Integer(0));
     }
