diff --git a/old_test.java b/new_test.java
index 202afa6..0c81b68 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,9 @@
 @Test
-	@Ignore
-	public void testFromMessage() {
-		fail("Not yet implemented");
+	public void testFromMessage() throws Exception {
+		String s = "test";
+		Message<String> message = MessageBuilder.withPayload(s).build();
+		SocketMessageMapper mapper = new SocketMessageMapper();
+		byte[] bArray = mapper.fromMessage(message);
+		assertEquals(s, new String(bArray));
+		
 	}
\ No newline at end of file
