diff --git a/old_product.java b/new_product.java
index a3c8236..6240a49 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,7 +10,8 @@ public boolean equals(Object o) {
 
       result = (getStart() == s.getStart()) && 
           (getEnd() == s.getEnd()) &&
-          (getType() != null ? type.equals(s.getType()) : true);
+          (getType() != null ? type.equals(s.getType()) : true) &&
+          (s.getType() != null ? s.getType().equals(getType()) : true);
     }
     else {
       result = false;
