diff --git a/old_test.java b/new_test.java
index cf1ad6b..c717b9b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,10 +5,17 @@
 
     assertTrue(a1.equals(a2));
     
+    // end is different
     Span b1 = new Span(100, 100, "test");
     assertFalse(a1.equals(b1));
     
+    // type is different
     Span c1 = new Span(100, 1000, "Test");
     assertFalse(a1.equals(c1));
     
+    Span d1 = new Span(100, 1000);
+    
+    assertFalse(d1.equals(a1));
+    assertFalse(a1.equals(d1));
+    
   }
\ No newline at end of file
