diff --git a/old_product.java b/new_product.java
index 233074a..f68d329 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,16 +2,16 @@ public static char boolsToHexDigit(boolean[] src, int srcPos) {
         if (src.length > srcPos + 3 && src[srcPos + 3]) {
             if (src.length > srcPos + 2 && src[srcPos + 2]) {
                 if (src.length > srcPos + 1 && src[srcPos + 1]) {
-                    if (src[srcPos]) return 'F';
-                    else return 'E';
+                    if (src[srcPos]) return 'f';
+                    else return 'e';
                 } else {
-                    if (src[srcPos]) return 'D';
-                    else return 'C';
+                    if (src[srcPos]) return 'd';
+                    else return 'c';
                 }
             } else {
                 if (src.length > srcPos + 1 && src[srcPos + 1]) {
-                    if (src[srcPos]) return 'B';
-                    else return 'A';
+                    if (src[srcPos]) return 'b';
+                    else return 'a';
                 } else {
                     if (src[srcPos]) return '9';
                     else return '8';
