diff --git a/old_test.java b/new_test.java
index 0653211..257459b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,17 +1,25 @@
 @Test
   public void PinTest() throws Exception {
-    mTfs.mkdir(new TachyonURI("/myFolder"));
-    int folderId = mTfs.getFileId(new TachyonURI("/myFolder"));
-    mTfs.setPinned(folderId, true);
-    int file0Id = mTfs.createFile(new TachyonURI("/myFolder/file0"), 64);
-    mTfs.setPinned(file0Id, false);
-    int file1Id = mTfs.createFile(new TachyonURI("/myFolder/file1"), 64);
-    ClientFileInfo folderInfo = mLocalTachyonCluster.getMasterInfo().getClientFileInfo(folderId);
-    ClientFileInfo file0Info = mLocalTachyonCluster.getMasterInfo().getClientFileInfo(file0Id);
-    ClientFileInfo file1Info = mLocalTachyonCluster.getMasterInfo().getClientFileInfo(file1Id);
+    mTfs.mkdirs(new TachyonURI("/myFolder"));
+    TachyonFile folder = mTfs.open(new TachyonURI("/myFolder"));
+    mTfs.setPin(folder, true);
+
+    TachyonURI file0Path = new TachyonURI("/myFolder/file0");
+    ClientOptions op = new ClientOptions.Builder(mMasterTachyonConf).setBlockSize(64).build();
+    mTfs.getOutStream(file0Path, op).close();
+    TachyonFile file0 = mTfs.open(file0Path);
+    mTfs.setPin(file0, false);
+
+    TachyonURI file1Path = new TachyonURI("/myFolder/file1");
+    mTfs.getOutStream(file1Path, op).close();
+
+    FileInfo folderInfo = mTfs.getInfo(folder);
+    FileInfo file0Info = mTfs.getInfo(file0);
+    FileInfo file1Info = mTfs.getInfo(mTfs.open(file1Path));
+
     mLocalTachyonCluster.stopTFS();
+
     PinTestUtil(folderInfo, file0Info, file1Info);
-    String editLogPath = mLocalTachyonCluster.getEditLogPath();
-    UnderFileSystem.get(editLogPath, mMasterTachyonConf).delete(editLogPath, true);
+    deleteFsMasterJournalLogs();
     PinTestUtil(folderInfo, file0Info, file1Info);
   }
\ No newline at end of file
