diff --git a/old_product.java b/new_product.java
index a0f934b..a1053be 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,7 @@
-public int pin(TachyonURI path) throws IOException {
-    TachyonFS tachyonClient = createFS(path);
-    int fileId = tachyonClient.getFileId(path);
-
+public int pin(TachyonURI path) {
     try {
-      tachyonClient.pinFile(fileId);
+      TachyonFile fd = mTfs.open(path);
+      mTfs.setPin(fd, true);
       System.out.println("File '" + path + "' was successfully pinned.");
       return 0;
     } catch (Exception e) {
