diff --git a/old_test.java b/new_test.java
index 2a460b6..35afc6c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void readAllPropertiesToMap(){
-        Map<String, String> map = ResourceBundleUtil.readAllPropertiesToMap(BASE_NAME, Locale.CHINA);
-        LOGGER.debug(JsonUtil.format(map));
+        LOGGER.debug(JsonUtil.format(ResourceBundleUtil.readAllPropertiesToMap(BASE_NAME, Locale.CHINA)));
     }
\ No newline at end of file
