diff --git a/old_product.java b/new_product.java
index 4956b8d..0807ee0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,5 @@
 public static HighAvailabilityMode fromConfig(Configuration config) {
-		String haMode = ConfigurationUtil.getStringWithDeprecatedKeys(
-				config,
-				ConfigConstants.HA_MODE,
-				null,
-				ConfigConstants.RECOVERY_MODE);
+		String haMode = config.getValue(HighAvailabilityOptions.HA_MODE);
 
 		if (haMode == null) {
 			return HighAvailabilityMode.NONE;
