diff --git a/old_product.java b/new_product.java
index de102fb..001117f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,10 +5,10 @@ public static String[] tokenizeToStringArray(String str,String delimiters,boolea
         //StringTokenizer implements Enumeration<Object>
         //其在 Enumeration接口的基础上,  定义了 hasMoreTokens nextToken两个方法
         //实现的Enumeration接口中的  hasMoreElements nextElement 调用了  hasMoreTokens nextToken
-        StringTokenizer st = new StringTokenizer(str, delimiters);
+        StringTokenizer stringTokenizer = new StringTokenizer(str, delimiters);
         List<String> tokens = new ArrayList<String>();
-        while (st.hasMoreTokens()){
-            String token = st.nextToken();
+        while (stringTokenizer.hasMoreTokens()){
+            String token = stringTokenizer.nextToken();
             if (trimTokens){
                 token = token.trim();
             }
