diff --git a/old_test.java b/new_test.java
index 6232692..5d409d6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,18 +1,9 @@
 @Test
     public void tokenizeToStringArray(){
-        String str = "jin.xin  h hhaha ,lala;feilong;jin.xin  h hhaha ,lala;feilong;jin.xin  h hhaha ,lala;feilong;jin.xin  h hhaha ,lala;feilong;jin.xin  h hhaha ,lala;feilong;jin.xin  h hhaha ,lala;feilong;jin.xin  h hhaha ,lala;feilong";
-        StringBuilder sb = new StringBuilder();
-        for (int i = 0; i < 1; ++i){
-            sb.append(str);
-        }
-        str = sb.toString();
-        String delimiters = ";, .";
-        String[] tokenizeToStringArray = StringUtil.tokenizeToStringArray(str, delimiters);
-        LOGGER.info(JsonUtil.format(tokenizeToStringArray));
         Date beginDate = new Date();
 
         for (int i = 0; i < 100000; ++i){
-            StringUtil.tokenizeToStringArray(str, delimiters);
+            tokenizeToStringArray1();
         }
 
         Date endDate = new Date();
