diff --git a/old_test.java b/new_test.java
index ad6919f..8353781 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,7 +12,7 @@
         
         
         String dbName = this.getClass().getSimpleName();
-        DummyXMinMaxNormalizer df = new DummyXMinMaxNormalizer(dbName, conf, new DummyXMinMaxNormalizer.TrainingParameters());
+        DummyXMinMaxNormalizer df = MLBuilder.create(new DummyXMinMaxNormalizer.TrainingParameters(), dbName, conf);
         
         df.fit_transform(trainingData);
         df.transform(validationData);
@@ -22,7 +22,7 @@
         param.setTotalIterations(100);
         param.setL2(0.001);
 
-        OrdinalRegression instance = new OrdinalRegression(dbName, conf, param);
+        OrdinalRegression instance = MLBuilder.create(param, dbName, conf);
         
         instance.fit(trainingData);
         
@@ -31,8 +31,8 @@
         //instance = null;
         //df = null;
         
-        df = new DummyXMinMaxNormalizer(dbName, conf);
-        instance = new OrdinalRegression(dbName, conf);
+        df = MLBuilder.load(DummyXMinMaxNormalizer.class, dbName, conf);
+        instance = MLBuilder.load(OrdinalRegression.class, dbName, conf);
         
         instance.predict(validationData);
 
