diff --git a/old_product.java b/new_product.java
index c0d4498..d92641b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,7 @@ public void predict(Dataframe newData) {
         boolean transformData = dtParams!=null;
         if(transformData) {
             if(dataTransformer==null) {
-                dataTransformer = Trainable.newInstance(dtParams.getTClass(), dbName, conf);
+                dataTransformer = MLBuilder.load(dtParams.getTClass(), dbName, conf);
             }
             setParallelized(dataTransformer);
             dataTransformer.transform(newData);
@@ -18,14 +18,14 @@ public void predict(Dataframe newData) {
         boolean selectFeatures = fsParams!=null;
         if(selectFeatures) {
             if(featureSelector==null) {
-                featureSelector = Trainable.newInstance(fsParams.getTClass(), dbName, conf);
+                featureSelector = MLBuilder.load(fsParams.getTClass(), dbName, conf);
             }
             setParallelized(featureSelector);
             featureSelector.transform(newData);
         }
 
         if(modeler==null) {
-            modeler = Trainable.newInstance(trainingParameters.getModelerTrainingParameters().getTClass(), dbName, conf);
+            modeler = MLBuilder.load(trainingParameters.getModelerTrainingParameters().getTClass(), dbName, conf);
         }
         setParallelized(modeler);
         modeler.predict(newData);
