diff --git a/old_product.java b/new_product.java
index 85afe8a..7e05787 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,5 @@
 public static void createStorageDirPath(String path) throws IOException {
     File dir = new File(path);
-    String absolutePath = dir.getAbsolutePath();
     if (dir.exists()) {
       return;
     }
@@ -11,6 +10,7 @@ public static void createStorageDirPath(String path) throws IOException {
       }
       throw new IOException("Failed to create folder " + path);
     }
+    String absolutePath = dir.getAbsolutePath();
     changeLocalFileToFullPermission(absolutePath);
     setLocalDirStickyBit(absolutePath);
     LOG.info("Folder {} was created!", path);
