diff --git a/old_test.java b/new_test.java
index 6b6b2c7..162c913 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
   public void createStorageDirPathTest() throws IOException {
-    File storageDir = new File(mTestFolder.getRoot(), "tmp");
+    File storageDir = new File(mTestFolder.getRoot(), "storageDir");
     File blockFile = new File(storageDir, "200");
 
-    // When storage dir not exists
+    // When storage dir doesn't exists
     FileUtils.createBlockPath(blockFile.getAbsolutePath());
     Assert.assertTrue(FileUtils.exists(storageDir.getAbsolutePath()));
     Assert.assertEquals(
