diff --git a/old_test.java b/new_test.java
index 789f07a..c9dad40 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
         {
             {
                 oneOf(documentAccessBridge).getDocumentURL(ATTACHMENT_REFERENCE.getDocumentReference(), "temp", null,
-                    null);
+                    null, true);
                 will(returnValue("/xwiki/bin/temp/Main/Test"));
             }
         });
