diff --git a/old_product.java b/new_product.java
index 0ea8f7c..73cc7bc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,10 @@
 protected String buildURL(AttachmentReference attachmentReference, String fileName)
     {
+        // We need the absolute URL because the gallery macro, which is used when viewing an office presentation, uses
+        // the syntax of the target document to parse its content and XWiki 2.0 syntax (unlike XWiki 2.1) doesn't
+        // support path image references (e.g. image:path:/one/two/three.png).
         String prefix =
-            documentAccessBridge.getDocumentURL(attachmentReference.getDocumentReference(), "temp", null, null);
+            documentAccessBridge.getDocumentURL(attachmentReference.getDocumentReference(), "temp", null, null, true);
         try {
             String encodedAttachmentName = URLEncoder.encode(attachmentReference.getName(), DEFAULT_ENCODING);
             String encodedFileName = URLEncoder.encode(fileName, DEFAULT_ENCODING);
