diff --git a/old_product.java b/new_product.java
index ebd8b10..e61ae2a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,22 +2,18 @@
   @ResponseBody
   public String startGatewaySender(
       @RequestParam(CliStrings.START_GATEWAYSENDER__ID) final String gatewaySenderId,
-      @RequestParam(value = CliStrings.START_GATEWAYSENDER__GROUP,
-          required = false) final String[] groups,
-      @RequestParam(value = CliStrings.START_GATEWAYSENDER__MEMBER,
-          required = false) final String[] members) {
+      @RequestParam(value = CliStrings.GROUP, required = false) final String[] groups,
+      @RequestParam(value = CliStrings.MEMBER, required = false) final String[] members) {
     CommandStringBuilder command = new CommandStringBuilder(CliStrings.START_GATEWAYSENDER);
 
     command.addOption(CliStrings.START_GATEWAYSENDER__ID, gatewaySenderId);
 
     if (hasValue(groups)) {
-      command.addOption(CliStrings.START_GATEWAYSENDER__GROUP,
-          StringUtils.join(groups, StringUtils.COMMA_DELIMITER));
+      command.addOption(CliStrings.GROUP, StringUtils.join(groups, StringUtils.COMMA_DELIMITER));
     }
 
     if (hasValue(members)) {
-      command.addOption(CliStrings.START_GATEWAYSENDER__MEMBER,
-          StringUtils.join(members, StringUtils.COMMA_DELIMITER));
+      command.addOption(CliStrings.MEMBER, StringUtils.join(members, StringUtils.COMMA_DELIMITER));
     }
 
     return processCommand(command.toString());
