diff --git a/old_product.java b/new_product.java
index f597011..120ff87 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public boolean isRunning(final String id, final InputInfo info) throws QueryException {
-    return get(id, info).qp != null;
+public boolean finished(final String id, final InputInfo info) throws QueryException {
+    return get(id, info).qp == null;
   }
\ No newline at end of file
