diff --git a/old_test.java b/new_test.java
index dde3dc3..f653929 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,7 @@
 @Test
-  public void isRunning() {
-    final String id = query(_ASYNC_EVAL.args("\"(1 to 1000000)[.=0]\""));
-    query(_ASYNC_IS_RUNNING.args(id), "true");
+  public void finished() {
+    final String id = query(_ASYNC_EVAL.args("\"(1 to 100000000)[.=0]\""));
+    query(_ASYNC_FINISHED.args(id), "false");
+    query(_ASYNC_STOP.args(id));
+    error(_ASYNC_FINISHED.args(id), ASYNC_UNKNOWN_X);
   }
\ No newline at end of file
