diff --git a/old_product.java b/new_product.java
index c7c8a39..c3b3121 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,8 @@ public static <K, V, C extends Serializable, D extends AutoCloseable> D getData(
         UpstreamTransformerBuilder transformerBuilder,
         String datasetCacheName, UUID datasetId,
         PartitionDataBuilder<K, V, C, D> partDataBuilder,
-        LearningEnvironment env) {
+        LearningEnvironment env,
+        boolean isKeepBinary) {
 
         PartitionDataStorage dataStorage = (PartitionDataStorage)ignite
             .cluster()
@@ -19,6 +20,9 @@ public static <K, V, C extends Serializable, D extends AutoCloseable> D getData(
 
             IgniteCache<K, V> upstreamCache = ignite.cache(upstreamCacheName);
 
+            if (isKeepBinary)
+                upstreamCache = upstreamCache.withKeepBinary();
+
             ScanQuery<K, V> qry = new ScanQuery<>();
             qry.setLocal(true);
             qry.setPartition(part);
