diff --git a/old_test.java b/new_test.java
index 1ff2542..26e44fc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -41,7 +41,8 @@
                         UpstreamEntry<Integer, Integer> e = upstream.next();
                         return new TestPartitionData(e.getKey() + e.getValue());
                     },
-                    TestUtils.testEnvBuilder().buildForWorker(part)
+                    TestUtils.testEnvBuilder().buildForWorker(part),
+                    false
                 ),
                 0
             );
