diff --git a/old_test.java b/new_test.java
index 78074da..b8d8ae2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,8 @@
     public void testInitialize() {
         FileNameAnalyzer instance = new FileNameAnalyzer();
         try {
-            instance.initializeSettings(getSettings());
-            instance.initialize(null);
+            instance.initialize(getSettings());
+            instance.prepare(null);
         } catch (InitializationException ex) {
             fail(ex.getMessage());
         }
