diff --git a/old_product.java b/new_product.java
index d35829b..7f8e8fc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,17 +1,4 @@
 @Override
-    public final void initialize(Engine engine) throws InitializationException {
-        final String key = getAnalyzerEnabledSettingKey();
-        try {
-            this.setEnabled(settings.getBoolean(key, true));
-        } catch (InvalidSettingException ex) {
-            final String msg = String.format("Invalid setting for property '%s'", key);
-            LOGGER.warn(msg);
-            LOGGER.debug(msg, ex);
-        }
-
-        if (isEnabled()) {
-            initializeAnalyzer(engine);
-        } else {
-            LOGGER.debug("{} has been disabled", getName());
-        }
+    public void initialize(Settings settings) {
+        this.settings = settings;
     }
\ No newline at end of file
