diff --git a/old_product.java b/new_product.java
index 15a07f3..3c49935 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public static byte[] stringToByteArray(String text, int length) {
-        return Arrays.copyOf(text.getBytes(), length);
+public static byte[] asciiStringToByteArray(String text, int length) {
+        return Arrays.copyOf(text.getBytes(StandardCharsets.US_ASCII), length);
     }
\ No newline at end of file
