diff --git a/old_test.java b/new_test.java
index d252bd1..7f95247 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
     public void testStringToByteArray() {
         byte[] temp = { (byte) '1', (byte) '2', (byte) 'a', (byte) 'f', (byte) 0 };
-        assertTrue(Arrays.equals(temp, ParseUtil.stringToByteArray("12af", 5)));
+        assertTrue(Arrays.equals(temp, ParseUtil.asciiStringToByteArray("12af", 5)));
     }
\ No newline at end of file
