diff --git a/old_product.java b/new_product.java
index 9edd374..b233849 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,6 @@
 @PrePersist
     @PreUpdate
     protected void onCreateOrUpdateJob() throws GeniePreconditionException {
-        this.clusterCriteriasString = clusterCriteriasToString(this.clusterCriterias);
-        this.commandCriteriaString = commandCriteriaToString(this.commandCriteria);
-        // Add the id to the tags
-        if (this.tags == null) {
-            this.tags = new HashSet<>();
-        }
+//        this.clusterCriteriasString = clusterCriteriasToString(this.clusterCriterias);
+//        this.commandCriteriaString = commandCriteriaToString(this.commandCriteria);
     }
\ No newline at end of file
