diff --git a/old_test.java b/new_test.java
index c26fce8..14e49cc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	@Verifies(value = "should fail validation if startDate after dateStopped", method = "validate(Object,Errors)")
-	public void validate_shouldFailValidationIfStartDateAfterDiscontinuedDate() throws Exception {
+	public void validate_shouldFailValidationIfStartDateAfterDateStopped() throws Exception {
 		Order order = new Order();
 		order.setConcept(Context.getConceptService().getConcept(88));
 		order.setPatient(Context.getPatientService().getPatient(2));
