diff --git a/old_product.java b/new_product.java
index bf65a7f..01217be 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public void validate(Object obj, Errors errors) {
 		Order order = (Order) obj;
 		if (order == null) {
-			errors.rejectValue("order", "error.general");
+			errors.reject("error.general");
 		} else {
 			// for the following elements Order.hbm.xml says: not-null="true"
 			ValidationUtils.rejectIfEmpty(errors, "voided", "error.null");
