diff --git a/old_test.java b/new_test.java
index 2da8d12..3cde215 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,6 +7,6 @@
         realm.commitTransaction();
         assertEquals(sizeBefore - 1, collection.size());
 
-        assertNotNull(collection.osList);
-        assertEquals(collection.osList.getTargetTable().getName(), snapshot.getTable().getName());
+        assertNotNull(collection.getOsList());
+        assertEquals(collection.getOsList().getTargetTable().getName(), snapshot.getTable().getName());
     }
\ No newline at end of file
