diff --git a/old_product.java b/new_product.java
index bf97f8d..dbdedea 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,17 +4,20 @@
             throw new UnsupportedOperationException(ONLY_IN_MANAGED_MODE_MESSAGE);
         }
         checkValidRealm();
+        if (!osListOperator.forRealmModel()) {
+            throw new IllegalStateException(ALLOWED_ONLY_FOR_REALM_MODEL_ELEMENT_MESSAGE);
+        }
         if (className != null) {
             return new OrderedRealmCollectionSnapshot<>(
                     realm,
-                    new io.realm.internal.Collection(realm.sharedRealm, osList, null),
+                    new io.realm.internal.Collection(realm.sharedRealm, osListOperator.getOsList(), null),
                     className);
         } else {
             // 'clazz' is non-null when 'dynamicClassName' is null.
             //noinspection ConstantConditions
             return new OrderedRealmCollectionSnapshot<>(
                     realm,
-                    new io.realm.internal.Collection(realm.sharedRealm, osList, null),
+                    new io.realm.internal.Collection(realm.sharedRealm, osListOperator.getOsList(), null),
                     clazz);
         }
     }
\ No newline at end of file
