diff --git a/old_test.java b/new_test.java
index 04485a7..9945767 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
 	public void getProviders_shouldFetchProviderByMatchingQueryStringWithAnyUnVoidedPersonsFamilyName() throws Exception {
-		assertEquals(2, service.getProviders("Che", 0, null, null).size());
+		assertEquals(2, service.getProviders("Che", 0, null, null, true).size());
 	}
\ No newline at end of file
