diff --git a/old_product.java b/new_product.java
index 5272575..12dda5d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,7 @@
-public static boolean test(TransposeDataList transposeDataList, boolean is_twoTailed, double aLevel) throws IllegalArgumentException { 
-        //Note! Despite the fact that this test seems very very easy, in fact it is SUPER HARD to calcuate.
-        
-        //validate the results of the test with the theory because the PHP to JAVA implementation changed a lot the logic!
+public static boolean test(TransposeDataList transposeDataList, boolean is_twoTailed, double aLevel) { 
         int k=transposeDataList.size();
         if(k!=2) {
-            throw new IllegalArgumentException();
+            throw new IllegalArgumentException("The collection must contain observations from 2 groups.");
         }
         
         Object[] keys = transposeDataList.keySet().toArray();
@@ -21,7 +18,7 @@ public static boolean test(TransposeDataList transposeDataList, boolean is_twoTa
             Object keyj = keys[j];
             int nj=transposeDataList.get(keyj).size();
             if(nj<=0) {
-                throw new IllegalArgumentException();
+                throw new IllegalArgumentException("The number of observations in each group but be larger than 0.");
             }
             
             
