diff --git a/old_test.java b/new_test.java
index 4daecfe..41dbef3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testTest() throws Exception {
+    public void testTest() {
         logger.info("test");
         //Example from Dimaki's Non-parametrics notes. It should NOT reject the null hypothesis and return false.
         FlatDataCollection flatDataCollection =  new FlatDataCollection(Arrays.asList(new Object[]{33.4, 33.3, 31.0, 31.4, 33.5, 34.4, 33.7, 36.2, 34.9, 37.0}));
