diff --git a/old_product.java b/new_product.java
index 6c798d9..2b61ecc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -27,6 +27,8 @@ public static String getModuleAddress (String ApplicationRoot, String moduleId,
 		catch(Exception e)
 		{
 			log.error("Module Hash Retrieval: " + e.toString());
+			log.error("moduleID = " + moduleId);
+			log.error("userID = " + userId);
 		}
 		Database.closeConnection(conn);
 		log.debug("*** END getModuleAddress() ***");
