diff --git a/old_test.java b/new_test.java
index 3091ee6..8707fe8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,18 +1,36 @@
 @Test
 	public void testGetModuleAddress() 
 	{
+		String userName = new String("userGetModuleAddress");
 		String insecureCryptoLesson = new String("201ae6f8c55ba3f3b5881806387fbf34b15c30c2");
-		//Open all Modules First so that the Module Can Be Opened
-		if(Setter.openAllModules(applicationRoot))
+		try
 		{
-			//Simulate user Opening Level
-			if(!Getter.getModuleAddress(applicationRoot, insecureCryptoLesson, adminUserId).isEmpty())
+			if(verifyTestUser(applicationRoot, userName, userName))
 			{
-				log.debug("PASS: Could mark level open when level was marked as open");
+				String userId = Getter.getUserIdFromName(applicationRoot, userName);
+				//Open all Modules First so that the Module Can Be Opened
+				if(Setter.openAllModules(applicationRoot))
+				{
+					//Simulate user Opening Level
+					if(!Getter.getModuleAddress(applicationRoot, insecureCryptoLesson, userId).isEmpty())
+					{
+						log.debug("PASS: Could mark level open when level was marked as open");
+						return;
+					}
+					else
+						fail("Could not Insecure Crypto Lesson as Opened by user");
+				}
+				else
+					fail("Could not Open All Modules");
 			}
 			else
-				fail("Could not Insecure Crypto Lesson as Opened by Default admin");
+			{
+				fail("Could not verify user (No Exception Failure)");
+			}
+		}
+		catch(Exception e)
+		{
+			log.fatal("Could not Verify User: " + e.toString());
+			fail("Could not Verify User " + userName);
 		}
-		else
-			fail("Could not Open All Modules");
 	}
\ No newline at end of file
