diff --git a/old_test.java b/new_test.java
index 6f8ac99..a731db2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
     array(_ARRAY_INSERT_BEFORE.args(" [1]", " 1", " 2"), "[2, 1]");
     array(_ARRAY_INSERT_BEFORE.args(" [1]", " 2", " 2"), "[1, 2]");
 
-    error(_ARRAY_INSERT_BEFORE.args(" []", " 0", " 1"), Err.ARRAYBOUNDS);
-    error(_ARRAY_INSERT_BEFORE.args(" []", " 2", " 1"), Err.ARRAYBOUNDS);
+    error(_ARRAY_INSERT_BEFORE.args(" []", " 0", " 1"), Err.ARRAYBOUNDS_X_X);
+    error(_ARRAY_INSERT_BEFORE.args(" []", " 2", " 1"), Err.ARRAYBOUNDS_X_X);
   }
\ No newline at end of file
