diff --git a/old_product.java b/new_product.java
index 4be1f54..fe00cc6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 private Array insertBefore(final QueryContext qc) throws QueryException {
-    final Array array = array(0, qc);
-    final int p = checkPos(array, checkItr(exprs[1], qc), true);
+    final Array array = toArray(exprs[0], qc);
+    final int p = checkPos(array, toLong(exprs[1], qc), true);
     final Value ins = qc.value(exprs[2]);
     final int as = array.arraySize();
     final ValueList vl = new ValueList(as + 1);
