diff --git a/old_product.java b/new_product.java
index 571a98b..dfa2e19 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,8 +2,6 @@ public Class<?> defineClass(byte[] bytes) throws IllegalAccessException {
             SecurityManager sm = System.getSecurityManager();
             if (sm != null)
                 sm.checkPermission(new RuntimePermission("defineClass"));
-            if (hasPrivateAccess())
-                throw new UnsupportedOperationException("PRIVATE access not supported");
             if ((lookupModes() & PACKAGE) == 0)
                 throw new IllegalAccessException("Lookup does not have PACKAGE access");
             assert (lookupModes() & (MODULE|PUBLIC)) != 0;
