diff --git a/old_test.java b/new_test.java
index 590e722..4502dcc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void getPath() throws Exception {
     try (LockedInodePath inodePath = mTree.lockFullInodePath(0, InodeTree.LockMode.READ)) {
-      Inode<?> root = inodePath.getInode();
+      InodeView root = inodePath.getInode();
       // test root path
       assertEquals(new AlluxioURI("/"), mTree.getPath(root));
     }
