diff --git a/old_test.java b/new_test.java
index 6f22d06..0eee397 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,6 +11,6 @@
 		Assert.assertEquals(8, patientIdentifiers.size());
 		
 		for (PatientIdentifier patientIdentifier : patientIdentifiers) {
-			Assert.assertFalse(patientIdentifier.isVoided());
+			Assert.assertFalse(patientIdentifier.getVoided());
 		}
 	}
\ No newline at end of file
