diff --git a/old_product.java b/new_product.java
index 55acac5..51b74da 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public List<PatientIdentifier> getPatientIdentifiers(PatientIdentifierType pit)
 		List<PatientIdentifier> ids = new Vector<PatientIdentifier>();
 		if (getIdentifiers() != null) {
 			for (PatientIdentifier pi : getIdentifiers()) {
-				if (!pi.isVoided() && pit.equals(pi.getIdentifierType())) {
+				if (!pi.getVoided() && pit.equals(pi.getIdentifierType())) {
 					ids.add(pi);
 				}
 			}
