diff --git a/old_test.java b/new_test.java
index 7d8e364..8b3bc1e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test
     public void testClear() {
         // Given
-        cacheManager.put(new Feature("ff", false, "Description"));
-        cacheManager.put(new Feature("ff2", false, "Description"));
-        cacheManager.put(new Feature("ff3", false, "Description"));
+        cacheManager.putFeature(new Feature("ff", false, "Description"));
+        cacheManager.putFeature(new Feature("ff2", false, "Description"));
+        cacheManager.putFeature(new Feature("ff3", false, "Description"));
         Assert.assertEquals(3, cacheManager.listCachedFeatureNames().size());
         // When
-        cacheManager.clear();
+        cacheManager.clearFeatures();
         // Then
         Assert.assertTrue(cacheManager.listCachedFeatureNames().isEmpty());
     }
\ No newline at end of file
