diff --git a/old_product.java b/new_product.java
index ea7d7d4..102596a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,10 @@
 @Override
-    public void clear() {
-        cache.clear();
+    public void putProperty(AbstractProperty<?> prop) {
+        if (prop == null) {
+            throw new IllegalArgumentException("ff4j-core: Cannot insert null property into cache");
+        }
+        if (prop.getName() == null || prop.getName().isEmpty()) {
+            throw new IllegalArgumentException("ff4j-core: Cannot insert property with null identifier into cache");
+        }
+        getPropertyCache().put(prop.getName(), new InMemoryCacheEntry<AbstractProperty<?>>(prop));
     }
\ No newline at end of file
