diff --git a/old_test.java b/new_test.java
index aebf09e..391339a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,16 +1,19 @@
 @Test
-    public void testSpatialJoinQueryUsingIndex() throws Exception {
+    public void testSpatialJoinQueryWithPolygonRDDUsingQuadTreeIndex() throws Exception {
+    	
+        PolygonRDD queryRDD = new PolygonRDD(sc, InputLocationQueryPolygon, splitter, true, numPartitions);
 
-    	PolygonRDD polygonRDD = new PolygonRDD(sc, InputLocation, offset, splitter, numPartitions);
-
-    	PolygonRDD objectRDD = new PolygonRDD(sc, InputLocation, offset, splitter, gridType, numPartitions);
-
-    	objectRDD.buildIndex("strtree");
-
-        JoinQuery joinQuery = new JoinQuery(sc,objectRDD,polygonRDD);
+        PolygonRDD spatialRDD = new PolygonRDD(sc, InputLocation, splitter, true, numPartitions);
+        
+        spatialRDD.spatialPartitioning(gridType);
+        
+        spatialRDD.buildIndex(IndexType.QUADTREE, true);
+        
+        queryRDD.spatialPartitioning(spatialRDD.grids);
+        
+        List<Tuple2<Polygon, HashSet<Polygon>>> result = JoinQuery.SpatialJoinQuery(spatialRDD,queryRDD,false).collect();
         
-        List<Tuple2<Polygon, HashSet<Polygon>>> result = joinQuery.SpatialJoinQueryUsingIndex(objectRDD,polygonRDD).collect();
-        assert result.get(0)._1().getUserData()!=null;
+        assert result.get(1)._1().getUserData()!=null;
         for(int i=0;i<result.size();i++)
         {
         	if(result.get(i)._2().size()!=0)
