diff --git a/old_test.java b/new_test.java
index 1de7922..2398651 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,9 @@
     AlluxioURI src = new AlluxioURI("/file");
     AlluxioURI dst = new AlluxioURI("/file2");
     RenamePOptions renameOptions = RenamePOptions.getDefaultInstance();
-    doNothing().when(mFileSystemMasterClient).rename(src, dst, renameOptions);
+    doNothing().when(mFileSystemMasterClient).rename(src, dst,
+        FileSystemOptions.renameDefaults(mConf).toBuilder().mergeFrom(renameOptions).build());
     mFileSystem.rename(src, dst, renameOptions);
-    verify(mFileSystemMasterClient).rename(src, dst, renameOptions);
+    verify(mFileSystemMasterClient).rename(src, dst,
+        FileSystemOptions.renameDefaults(mConf).toBuilder().mergeFrom(renameOptions).build());
   }
\ No newline at end of file
