diff --git a/old_product.java b/new_product.java
index a8cff35..adb42fb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,8 +8,7 @@
       public Void call() throws Exception {
         Preconditions.checkNotNull(dst, "required 'dst' parameter is missing");
         if (options == null) {
-          mFileSystem.rename(new AlluxioURI(path), new AlluxioURI(dst),
-              RenamePOptions.getDefaultInstance());
+          mFileSystem.rename(new AlluxioURI(path), new AlluxioURI(dst));
         } else {
           mFileSystem.rename(new AlluxioURI(path), new AlluxioURI(dst), options);
         }
