diff --git a/old_product.java b/new_product.java
index 16c571d..5e2acff 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,6 +2,6 @@ public ResourceBundle getDriverConfigBundle() {
     if(driverConfigBundle != null) {
       return driverConfigBundle;
     }
-    retrieveAndCacheDriverConfig();
+    retrieveAndCacheDriver();
     return driverConfigBundle;
   }
\ No newline at end of file
