diff --git a/old_test.java b/new_test.java
index 59f49ff..5883779 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
     public void testSetJars() {
-        Assert.assertNull(this.a.getJars());
+        Assert.assertNull(this.a.getDependencies());
         final Set<String> jars = new HashSet<>();
         jars.add("s3://netflix/jars/myJar.jar");
-        this.a.setJars(jars);
-        Assert.assertEquals(jars, this.a.getJars());
+        this.a.setDependencies(jars);
+        Assert.assertEquals(jars, this.a.getDependencies());
     }
\ No newline at end of file
