diff --git a/old_test.java b/new_test.java
index 6d07107..505039a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,6 +8,10 @@
     query(COUNT.args(COLLECTION.args(NAME + "/test")), 0);
     query(COUNT.args(COLLECTION.args(NAME + "/newtest")), NFLDR);
 
+    // rename paths
+    query(_DB_RENAME.args(NAME, "", "x"));
+    query(COUNT.args(COLLECTION.args(NAME + "/x/newtest")), NFLDR);
+
     // rename binary file
     query(_DB_STORE.args(NAME, "one", ""));
     query(_DB_RENAME.args(NAME, "one", "two"));
