diff --git a/old_product.java b/new_product.java
index 6e6a60b..3eaaf75 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,8 @@ private Item rename(final QueryContext ctx) throws QueryException {
     for(int i = 0, is = il.size(); i < is; i++) {
       final int pre = il.get(i);
       final String trg = Rename.target(data, pre, source, target);
-      if(trg.isEmpty()) BXDB_EMPTY.thrw(info, this);
+      if(trg.isEmpty() || trg.endsWith("/") || trg.endsWith("."))
+        BXDB_EMPTY.thrw(info, this);
       ctx.updates.add(new ReplaceValue(pre, data, info, token(trg)), ctx);
     }
     // rename files
