diff --git a/old_product.java b/new_product.java
index 26c5715..702d00c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,19 @@
-public void removeRun (int pos,
+public void removeRun (int index,
                            Run run)
     {
-        RunSequence seq = getSequence(pos);
+        // Find where this run lies in rle
+        Iterator<Run> iter = new Itr(index);
 
-        if (!seq.remove(run)) {
-            throw new RuntimeException(this + " Cannot find " + run + " at pos " + pos);
+        while (iter.hasNext()) {
+            Run r = iter.next();
+
+            if (r.isIdentical(run)) {
+                // We are located on the right run
+                iter.remove();
+
+                return;
+            }
         }
+
+        throw new RuntimeException(this + " Cannot find " + run + " at pos " + index);
     }
\ No newline at end of file
