diff --git a/old_test.java b/new_test.java
index e37e885..3a862f3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,17 +1,17 @@
 @Test
     public void testRemoveRun ()
     {
-        System.out.println("removeRun");
+        System.out.println("\n+++ removeRun");
 
         int pos = 3;
         Run run = new Run(4, 1);
         RunTable instance = createHorizontalInstance();
-        RunSequence seq1 = instance.getSequence(pos);
-        System.out.println("sequence before: " + seq1);
+        short[] seq1 = instance.getSequence(pos);
+        System.out.println("sequence before: " + Arrays.toString(seq1));
         System.out.println("table before:" + instance.dumpOf());
         instance.removeRun(pos, run);
 
-        RunSequence seq2 = instance.getSequence(pos);
-        System.out.println("sequence  after: " + seq2);
+        short[] seq2 = instance.getSequence(pos);
+        System.out.println("sequence  after: " + Arrays.toString(seq2));
         System.out.println("table after:" + instance.dumpOf());
     }
\ No newline at end of file
