diff --git a/old_product.java b/new_product.java
index 456188e..51eb007 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,8 +15,10 @@
         double varSum = 0.0;
 
         // d is always zero
+        double[] e = new double[dim];
         double[] f = new double[dim];
-        f[0] = GaussianDistribution.getInstance().cdf(v[0] / sigmaL.get(0, 0));
+        e[0] = GaussianDistribution.getInstance().cdf(v[0] / sigmaL.get(0, 0));
+        f[0] = e[0];
 
         double[] y = new double[dim];
 
@@ -25,13 +27,14 @@
         for (N = 1; err > errMax && N <= Nmax; N++) {
             double[] w = Math.random(dim - 1);
             for (int i = 1; i < dim; i++) {
-                y[i - 1] = GaussianDistribution.getInstance().quantile(w[i - 1] * f[i - 1]);
+                y[i - 1] = GaussianDistribution.getInstance().quantile(w[i - 1] * e[i - 1]);
                 double q = 0.0;
                 for (int j = 0; j < i; j++) {
                     q += sigmaL.get(i, j) * y[j];
                 }
 
-                f[i] = GaussianDistribution.getInstance().cdf((v[i] - q) / sigmaL.get(i, i)) * f[i - 1];
+                e[i] = GaussianDistribution.getInstance().cdf((v[i] - q) / sigmaL.get(i, i));
+                f[i] = e[i] * f[i - 1];
             }
 
             double del = (f[dim - 1] - p) / N;
