diff --git a/old_test.java b/new_test.java
index 02a7a57..69ee393 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
         System.out.println("cdf");
         MultivariateGaussianDistribution instance = new MultivariateGaussianDistribution(mu, sigma);
         for (int i = 0; i < x.length; i++) {
-            assertEquals(cdf[i], instance.cdf(x[i]), 1E-2);
+            assertEquals(cdf[i], instance.cdf(x[i]), 5E-3);
         }
     }
\ No newline at end of file
