diff --git a/old_test.java b/new_test.java
index 25f5494..d3c8657 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,16 +1,18 @@
 @Test
 	public void findErrorEvaluator() {
-		GrowQueue_I8 syndromes = array(64, 192, 93, 231, 52, 92, 228, 49, 83, 245);
-		GrowQueue_I8 errorLocator = array(3,1);
 
-		GrowQueue_I8 found = new GrowQueue_I8();
-		GrowQueue_I8 expected = array(0,64);
+		// one error
+		findErrorEvaluator(array(64, 192, 93, 231, 52, 92, 228, 49, 83, 2455),
+				array(3,1),
+				array(0,64));
 
-		ReidSolomonCodes alg = new ReidSolomonCodes(8,primitive8);
-		alg.findErrorEvaluator(syndromes,errorLocator,found);
+		// two errors
+		findErrorEvaluator(array(62, 101, 255, 19, 236, 196, 112, 227, 174, 215),
+				array(159,118,1),
+				array(0,62,142));
 
-		assertEquals(found.size,expected.size);
-		for (int j = 0; j < found.size; j++) {
-			assertEquals(found.get(j),expected.get(j));
-		}
+		// three errors
+		findErrorEvaluator(array(32, 188, 7, 92, 8, 39, 184, 32, 101, 213),
+				array(97, 138, 194, 1),
+				array(0, 32, 217, 182));
 	}
\ No newline at end of file
