diff --git a/old_product.java b/new_product.java
index 3bb422c..8a51694 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public boolean isDiscontinued(Date checkDate) {
 		if (dateStopped != null && autoExpireDate != null && dateStopped.after(autoExpireDate)) {
-			throw new APIException("The order has invalid dateStopped and autoExpireDate values");
+			throw new APIException("Order.error.invalidDateStoppedAndAutoExpireDate", (Object[]) null);
 		}
 		if (isVoided()) {
 			return false;
