diff --git a/old_test.java b/new_test.java
index b0429fb..a428bb6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,12 +8,6 @@
             String query = "SELECT * FROM " + table1.getEscapedTableName() + ";";
             rs = stmt.executeQuery(query);
             rs.verify(table1);
-            // bvt_rs = new bvt_ResultSet(rs);
-
-            // close and verify
-
-            // bvt_rs.verify();
-
         }
         finally {
 
