diff --git a/old_product.java b/new_product.java
index b12d1d1..7ed79d9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,5 @@
 public DBStatement createStatement(int type, int concurrency) throws SQLException {
-        Statement product = ((Connection) product()).createStatement(type, concurrency);
-        DBStatement wrapper = new DBStatement(this, product, type, concurrency, _holdability);
+        DBStatement dbstatement = new DBStatement(this);
+        return dbstatement.createStatement(type, concurrency);
 
-        // State
-        wrapper._cursortype = type;
-        wrapper._concurrency = concurrency;
-
-        return wrapper;
     }
\ No newline at end of file
