diff --git a/old_product.java b/new_product.java
index 7eb6586..8784d8d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -26,7 +26,10 @@ public void compile(PathCompilerCreateFlow<T> creator,
                     switch (type) {
                         case VLAN:
                             manageVlanEncap(creator, flows, devices, intent);
-                            // TODO: implement MPLS case here
+                            break;
+                        case MPLS:
+                             manageMplsEncap(creator, flows, devices, intent);
+                            break;
                         default:
                             // Nothing to do
                     }
