diff --git a/old_product.java b/new_product.java
index 8f68911..b477558 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,6 @@
-public void setMetadata(String field, String value) throws MissingResourceException
-    {
-        if ((field.trim()).equals("name")
-                && (value == null || value.trim().equals("")))
+@Deprecated
+    public void setMetadata(String field, String value) throws MissingResourceException {
+        if ((field.trim()).equals("name") && (value == null || value.trim().equals("")))
         {
             try
             {
@@ -13,6 +12,8 @@ public void setMetadata(String field, String value) throws MissingResourceExcept
             }
         }
 
+        String[] MDValue = getMDValueByLegacyField(field);
+
         /*
          * Set metadata field to null if null
          * and trim strings to eliminate excess
@@ -20,13 +21,13 @@ public void setMetadata(String field, String value) throws MissingResourceExcept
          */
 		if(value == null)
         {
-            collectionRow.setColumnNull(field);
+            clearMetadata(MDValue[0], MDValue[1], MDValue[2], Item.ANY);
+            modifiedMetadata = true;
         }
         else
         {
-            collectionRow.setColumn(field, value.trim());
+            setMetadataSingleValue(MDValue[0], MDValue[1], MDValue[2], null, value);
         }
 
-        modifiedMetadata = true;
         addDetails(field);
     }
\ No newline at end of file
