diff --git a/old_test.java b/new_test.java
index 8846978..f8afa01 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void time() {
-    query(_PROF_TIME.args("()"));
-    query("count(" + _PROF_TIME.args(" 1 to 100 ", false) + ")", 100);
-    query("count(" + _PROF_TIME.args(" 1 to 100 ", true) + ")", 100);
-    query("count(" + _PROF_TIME.args(" 1 to 100 ", true, "label") + ")", 100);
+    final Function func = _PROF_TIME;
+    query(func.args(" ()"));
+    query("count(" + func.args(" 1 to 100 ") + ")", 100);
+    query("count(" + func.args(" 1 to 100 ", "label") + ")", 100);
   }
\ No newline at end of file
