diff --git a/old_test.java b/new_test.java
index 9161e2d..a6064d4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,5 +2,5 @@
     public void testGetBigramFrequency() {
         System.out.println("getBigramFrequency");
         Bigram bigram = new Bigram("romantic", "comedy");
-        assertEquals(29, corpus.getBigramFrequency(bigram));
+        assertEquals(9, corpus.getBigramFrequency(bigram));
     }
\ No newline at end of file
