diff --git a/old_product.java b/new_product.java
index 527f709..7a24a90 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ protected void notify(URL url, NotifyListener listener, List<URL> urls) {
             throw new IllegalArgumentException("notify listener == null");
         }
         if ((CollectionUtils.isEmpty(urls))
-                && !Constants.ANY_VALUE.equals(url.getServiceInterface())) {
+                && !ANY_VALUE.equals(url.getServiceInterface())) {
             logger.warn("Ignore empty notify urls for subscribe url " + url);
             return;
         }
