diff --git a/old_test.java b/new_test.java
index e2f92c7..450d00f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
     public void test_previous_serialization() throws IOException, ClassNotFoundException {
-        DateTimeAdjuster previous = DateTimeAdjusters.previous(SUNDAY);
+        WithAdjuster previous = DateTimeAdjusters.previous(SUNDAY);
         assertTrue(previous instanceof Serializable);
 
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
