diff --git a/old_product.java b/new_product.java
index 63f9cf3..6907421 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public static byte[] hexStringToByteArray(String digits) {
         int len = digits.length();
         // Check if string is valid hex
         if (!VALID_HEX.matcher(digits).matches() || (len & 0x1) != 0) {
-            LOG.error("Invalid hexadecimal string: {}", digits);
+            LOG.warn("Invalid hexadecimal string: {}", digits);
             return null;
         }
         byte[] data = new byte[len / 2];
