diff --git a/old_product.java b/new_product.java
index 28f6644..447af78 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-void score( int x0 , int y0 , int dx , int dy ) {
+void scorePath(int x0 , int y0 , int dx , int dy ) {
 		// there is no previous disparity score so simply fill the cost for d=0
 		GrayU16 costXD = costYXD.getBand(y0);
 		int idxCost = costXD.getIndex(0,x0);   // C(0,0)
