diff --git a/old_test.java b/new_test.java
index 700daa8..66a11b0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,13 @@
 @Test
-	void score_SinglePath() {
+	void scorePath() {
 		Planar<GrayU16> costYXD = new Planar<>(GrayU16.class,rangeD,width,height);
 		GImageMiscOps.fillUniform(costYXD,rand,0,SgmDisparityCost.MAX_COST);
 
 		SgmCostAggregation alg = new SgmCostAggregation();
 
 		alg.init(costYXD);
-		int before = countNotZero(alg.aggregated);
 
-		alg.score(0,0,1,1);
+		alg.scorePath(0,0,1,1);
 
 		// the length is the number of elements to expect
 		int length = alg.computePathLength(0,0,1,1);
