diff --git a/old_test.java b/new_test.java
index e8331f7..15ba68b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testGetByBusinessKeys() {
-        MapParameters params = MapParameters.create()
+        NamedParameters params = NamedParameters.create()
                 .add("category", education)
                 .add("code", "02");
         Dictionary result = repository.getByBusinessKeys(Dictionary.class, params);
