diff --git a/old_product.java b/new_product.java
index 6257ddc..4e49201 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,6 +10,8 @@ public static DigestAlgorithm getDigestAlgorithm(ProtocolVersion protocolVersion
             result = DigestAlgorithm.SHA384;
         } else if (cipherSuite.usesGOSTR3411()) {
             result = DigestAlgorithm.GOSTR3411;
+        } else if (cipherSuite.usesGOSTR34112012()) {
+            result = DigestAlgorithm.GOSTR3411_2012_256;
         } else {
             result = DigestAlgorithm.SHA256;
         }
