diff --git a/old_test.java b/new_test.java
index 61bd95d..8f8ec66 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
-    public void test_plus_Period() {
-        SimplePeriod period = SimplePeriod.of(7, LocalPeriodUnit.MONTHS);
+    public void test_plus_Period_ISOPeriod() {
+        ISOPeriod period = ISOPeriod.of(7, MONTHS);
         ZonedDateTime t = ZonedDateTime.of(LocalDateTime.of(2008, 6, 1, 12, 30, 59, 500), ZONE_0100);
         ZonedDateTime expected = ZonedDateTime.of(LocalDateTime.of(2009, 1, 1, 12, 30, 59, 500), ZONE_0100);
         assertEquals(t.plus(period), expected);
