diff --git a/old_product.java b/new_product.java
index b4b872e..f346608 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,7 +11,7 @@ public Concepts getConceptRelatedConcepts(final Concept concept,
     if (parameters.get(CONCEPT_FIELDS) != null) {
       final RequestedFields fields = (RequestedFields) parameters.get(CONCEPT_FIELDS);
       if (fields != null && !fields.isEmpty())
-        queryParameters.put(CONCEPT_FIELDS, fields.toString());
+        queryParameters.put(CONCEPT_FIELDS, toJson(fields.getFields()));
     }
     return executeRequest(API_VERSION + conceptId + RELATED_CONCEPTS_PATH, queryParameters,
         Concepts.class);
