diff --git a/old_test.java b/new_test.java
index 173413a..4d02772 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,8 +9,7 @@
         */
         AlignmentInterval interval = getAlignmentInterval();
 
-        Map<String, List<Row>> result = (new AlignmentPacker()).packAlignments(Arrays.asList(interval),
-                new AlignmentTrack.RenderOptions());
+        Map<String, List<Row>> result = (new AlignmentPacker()).packAlignments(interval, new AlignmentTrack.RenderOptions());
         assertEquals(1, result.size());
         for (List<Row> alignmentrows : result.values()) {
             for (Row alignmentrow : alignmentrows) {
