diff --git a/old_test.java b/new_test.java
index dc3794c..878a49d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,11 @@
 @Test
   public void cancelBlockTest() throws Exception {
-    final int fileId = mTfs.createFile(new TachyonURI("/testFile"));
+    ClientOptions options = new ClientOptions.Builder(new TachyonConf()).build();
+    mTfs.getOutStream(new TachyonURI("/testFile"), options);
+    TachyonFile file = mTfs.open(new TachyonURI("/testFile"));
+
     final int blockSize = (int) WORKER_CAPACITY_BYTES / 2;
-    final long blockId = mTfs.getBlockId(fileId, 0);
+    final long blockId = BlockId.createBlockId(BlockId.getContainerId(file.getFileId()), 0);
 
     String filename = mWorkerServiceHandler.requestBlockLocation(USER_ID, blockId, blockSize);
     createBlockFile(filename, blockSize);
@@ -13,5 +16,5 @@
 
     // The master should not have recorded any used space after the block is cancelled
     waitForHeartbeat();
-    Assert.assertEquals(0, mMasterInfo.getUsedBytes());
+    Assert.assertEquals(0, mBlockMasterClient.getUsedBytes());
   }
\ No newline at end of file
