diff --git a/old_product.java b/new_product.java
index 8834a58..a93305a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public synchronized void cancelBlock(long blockId) throws IOException {
     mustConnect();
 
     try {
-      mClient.cancelBlock(mMasterClient.getUserId(), blockId);
+      mClient.cancelBlock(mUserId, blockId);
     } catch (TException e) {
       mConnected = false;
       throw new IOException(e);
