diff --git a/old_test.java b/new_test.java
index 1af5445..2379ee9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,9 +4,9 @@
 		configuration.setString(JobManagerOptions.ADDRESS, "localhost");
 		final WebMonitorExtension webMonitorExtension = WebMonitorUtils.loadWebSubmissionExtension(
 			CompletableFuture::new,
-			CompletableFuture.completedFuture("localhost:12345"),
 			Time.seconds(10),
 			Collections.emptyMap(),
+			CompletableFuture.completedFuture("localhost:12345"),
 			Paths.get("/tmp"),
 			Executors.directExecutor(),
 			configuration);
