diff --git a/old_product.java b/new_product.java
index d63df16..eef2e94 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 public static WebMonitorExtension loadWebSubmissionExtension(
 			GatewayRetriever<? extends DispatcherGateway> leaderRetriever,
-			CompletableFuture<String> restAddressFuture,
 			Time timeout,
 			Map<String, String> responseHeaders,
+			CompletableFuture<String> localAddressFuture,
 			java.nio.file.Path uploadDir,
 			Executor executor,
 			Configuration configuration) throws FlinkException {
@@ -13,18 +13,18 @@ public static WebMonitorExtension loadWebSubmissionExtension(
 					.forName("org.apache.flink.runtime.webmonitor.WebSubmissionExtension")
 					.getConstructor(
 						Configuration.class,
-						CompletableFuture.class,
 						GatewayRetriever.class,
 						Map.class,
+						CompletableFuture.class,
 						java.nio.file.Path.class,
 						Executor.class,
 						Time.class);
 
 				return (WebMonitorExtension) webSubmissionExtensionConstructor.newInstance(
 					configuration,
-					restAddressFuture,
 					leaderRetriever,
 					responseHeaders,
+					localAddressFuture,
 					uploadDir,
 					executor,
 					timeout);
