diff --git a/old_test.java b/new_test.java
index 20ec814..c639b42 100644
--- a/old_test.java
+++ b/new_test.java
@@ -50,7 +50,7 @@
 
         // Test for getScalingEvents for invalid scope/stream.
         final CompletableFuture<List<ScaleMetadata>> completableFuture1 = new CompletableFuture<>();
-        completableFuture1.completeExceptionally(new DataNotFoundException(""));
+        completableFuture1.completeExceptionally(StoreException.create(StoreException.Type.DATA_NOT_FOUND, ""));
         when(mockControllerService.getScaleRecords("scope1", "stream1")).thenReturn(completableFuture1);
         response = client.target(resourceURI).queryParam("from", fromDateTime).
                 queryParam("to", toDateTime).request().buildGet().invoke();
