diff --git a/old_product.java b/new_product.java
index 96147d9..c93069e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -35,10 +35,8 @@
             log.info("Successfully fetched required scaling events for scope: {}, stream: {}", scopeName, streamName);
             return Response.status(Status.OK).entity(finalScaleMetadataList).build();
         }).exceptionally(exception -> {
-            if (exception.getCause() instanceof DataNotFoundException
-                    || exception instanceof DataNotFoundException
-                    || exception.getCause() instanceof StoreException.NodeNotFoundException
-                    || exception instanceof StoreException.NodeNotFoundException) {
+            if (exception.getCause() instanceof StoreException.DataNotFoundException
+                    || exception instanceof StoreException.DataNotFoundException) {
                 log.warn("Stream/Scope name: {}/{} not found", scopeName, streamName);
                 return Response.status(Status.NOT_FOUND).build();
             } else {
