diff --git a/old_product.java b/new_product.java
index c021931..fbd7817 100644
--- a/old_product.java
+++ b/new_product.java
@@ -34,7 +34,8 @@ public LinkedHashMap<String, List<AlignmentInterval.Row>> packAlignments(
 
             // Now alphabetize (sort) and pack the groups
             List<String> keys = new ArrayList<String>(groupedAlignments.keySet());
-            Collections.sort(keys);
+            Comparator<String> groupComparator = getGroupComparator(renderOptions.groupByOption);
+            Collections.sort(keys, groupComparator);
             for (String key : keys) {
                 List<Row> alignmentRows = new ArrayList<Row>(10000);
                 List<Alignment> group = groupedAlignments.get(key);
