diff --git a/old_test.java b/new_test.java
index d5a3112..3226072 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,21 +1,13 @@
 @Test
     public void testPackAlignments() throws Exception {
 
-        //Retrieve data to pack
-        String path = "http://www.broadinstitute.org/igvdata/1KG/pilot2Bams/NA12878.SLX.bam";
-        String chr = "1";
-        int start = 557000;
-        int end = 558000;
-        boolean contained = false;
-
-        ResourceLocator rl = new ResourceLocator(path);
-        AlignmentReader samReader = AlignmentReaderFactory.getReader(rl);
-        CloseableIterator<Alignment> iter = samReader.query(chr, start, end, contained);
         ///////////////////////////
-
+        /*
         boolean showDuplicates = false;
         int qualityThreshold = 0;
         int maxLevels = 1000;
+        */
+        CloseableIterator<Alignment> iter = getAlignments();
 
         Map<String, List<AlignmentInterval.Row>> result = (new AlignmentPacker()).packAlignments(iter, end,
                 new AlignmentTrack.RenderOptions());
