diff --git a/old_product.java b/new_product.java
index 32fadb6..6c4f5ac 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@ void reportOrphanedSpans() {
     // flushing a span than hurt performance of unrelated operations by calling
     // currentTimeMicroseconds N times
     long flushTime = 0L;
-    boolean noop = zipkinHandler == FirehoseHandler.NOOP || this.noop.get();
+    boolean noop = zipkinHandler == FinishedSpanHandler.NOOP || this.noop.get();
     while ((contextKey = (RealKey) poll()) != null) {
       PendingSpan value = delegate.remove(contextKey);
       if (noop || value == null || !contextKey.sampled) continue;
