diff --git a/old_product.java b/new_product.java
index 7b7be34..13d0cb1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -28,7 +28,8 @@ public void validate(Object obj, Errors errors) {
 			}
 			ValidateUtil.validateFieldLengths(errors, obj.getClass(), "name", "description", "address1", "address2",
 			    "cityVillage", "stateProvince", "country", "postalCode", "latitude", "longitude", "countyDistrict",
-			    "address3", "address4", "address5", "address6", "retireReason");
+			    "address3", "address4", "address5", "address6", "address7", "address8", "address9", "address10",
+			    "address11", "address12", "address13", "address14", "address15", "retireReason");
 			super.validateAttributes(location, errors, Context.getLocationService().getAllLocationAttributeTypes());
 		}
 		
