diff --git a/old_product.java b/new_product.java
index 000c74d..9d1108d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
-static CurrencyValuesArray presentValue(
+static CurrencyScenarioArray presentValue(
       ResolvedCdsTrade trade,
       ScenarioMarketData marketData) {
 
-    return CurrencyValuesArray.of(
+    return CurrencyScenarioArray.of(
         marketData.getScenarioCount(),
         i -> calculatePresentValue(trade, marketData.scenario(i)));
   }
\ No newline at end of file
