diff --git a/old_test.java b/new_test.java
index 6cc72d4..9c1b276 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,8 +8,8 @@
 
     AssertJUnit.assertEquals(
         DsfTradeCalculations.DEFAULT.presentValue(RTRADE, RATES_LOOKUP, md),
-        CurrencyValuesArray.of(ImmutableList.of(expectedPv)));
+        CurrencyScenarioArray.of(ImmutableList.of(expectedPv)));
     AssertJUnit.assertEquals(
         DsfTradeCalculations.DEFAULT.currencyExposure(RTRADE, RATES_LOOKUP, md),
-        MultiCurrencyValuesArray.of(ImmutableList.of(expectedCurrencyExposure)));
+        MultiCurrencyScenarioArray.of(ImmutableList.of(expectedCurrencyExposure)));
   }
\ No newline at end of file
