diff --git a/old_test.java b/new_test.java
index 3fdcc65..79bcb76 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
     query(fun + "('" + PATH1 + "')");
     query(fun + "('" + PATH1 + "')");
     query(fun + "('" + PATH3 + "')");
-    query("file:delete('" + PATH1 + "', true())");
+    query("file:delete('" + PATH1 + "')");
     query("file:write('" + PATH1 + "', ())");
     error(fun + "('" + PATH1 + "')", Err.FILEEXISTS);
     error(fun + "('" + PATH3 + "')", Err.FILEEXISTS);
