diff --git a/old_product.java b/new_product.java
index 815a67a..5f4fa10 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-private StrStream content(final QueryContext ctx) throws QueryException {
+private StrStream text(final QueryContext ctx) throws QueryException {
     final byte[] uri = checkStr(expr[0], ctx);
     final String enc = encoding(1, BXFE_ENCODING, ctx);
     return new StrStream(IO.get(Token.string(uri)), enc, BXFE_IO);
