diff --git a/old_test.java b/new_test.java
index b1ca6c3..071d14a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,7 +11,9 @@
 		// this.uncompressedDataBuffer, null);
 
 		PowerMockito.mockStatic(CompressionLoader.class);
-		when(CompressionLoader.getDecompressorByCompressionLevel(Matchers.any(CompressionLevel.class))).thenReturn(
+		when(
+			CompressionLoader.getDecompressorByCompressionLevel(Matchers.any(CompressionLevel.class),
+				Matchers.any(FileInputChannel.class))).thenReturn(
 			decompressorMock);
 
 		@SuppressWarnings("unchecked")
