diff --git a/old_product.java b/new_product.java
index 7ca725d..d6352dc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-private T deserializeNextRecord() throws IOException {
+private T deserializeNextRecord(final T target) throws IOException {
 
 		if (this.bufferedRecord != null) {
 			final T record = this.bufferedRecord;
@@ -26,7 +26,7 @@ private T deserializeNextRecord() throws IOException {
 			}
 		}
 
-		final T nextRecord = this.deserializationBuffer.readData(this.uncompressedDataBuffer);
+		final T nextRecord = this.deserializationBuffer.readData(target, this.uncompressedDataBuffer);
 
 		if (this.uncompressedDataBuffer.remaining() == 0) {
 			releasedConsumedReadBuffer();
