diff --git a/old_test.java b/new_test.java
index 9cf26da..c0be2e6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	public void transformPixelToRectNorm_F64() {
-		IntrinsicParameters param = new IntrinsicParameters().
+		PinholeRadial param = new PinholeRadial().
 						fsetK(300, 320, 0, 150, 130, width, height).fsetRadial(0.1,1e-4);
 
 		DenseMatrix64F rect = new DenseMatrix64F(3,3,true,1.1,0,0,0,2,0,0.1,0,3);
