diff --git a/old_product.java b/new_product.java
index 805d641..cfc63c6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static PointTransform_F64 transformPixelToRectNorm_F64(IntrinsicParameters param,
+public static PointTransform_F64 transformPixelToRectNorm_F64(PinholeRadial param,
 																  DenseMatrix64F rectify,
 																  DenseMatrix64F rectifyK) {
 		if (rectifyK.get(0, 1) != 0)
