diff --git a/old_test.java b/new_test.java
index bbb23b2..2ed562b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,18 +1,20 @@
 @Test
   public void addCheckpointTest() throws Exception {
-    final int fileId = mTfs.createFile(new TachyonURI("/testFile"));
+    ClientOptions options = new ClientOptions.Builder(new TachyonConf()).build();
+    mTfs.getOutStream(new TachyonURI("/testFile"), options);
+    TachyonFile file = mTfs.open(new TachyonURI("/testFile"));
     final int blockSize = (int) WORKER_CAPACITY_BYTES / 10;
 
     String tmpFolder = mWorkerServiceHandler.getUserUfsTempFolder(USER_ID);
     UnderFileSystem ufs = UnderFileSystem.get(tmpFolder, mMasterTachyonConf);
     ufs.mkdirs(tmpFolder, true);
-    String filename = PathUtils.concatPath(tmpFolder, fileId);
+    String filename = PathUtils.concatPath(tmpFolder, file.getFileId());
     OutputStream out = ufs.create(filename);
     out.write(BufferUtils.getIncreasingByteArray(blockSize));
     out.close();
-    mWorkerServiceHandler.addCheckpoint(USER_ID, fileId);
+    mWorkerServiceHandler.addCheckpoint(USER_ID, (int) file.getFileId());
 
     // No space should be used in Tachyon, but the file should be complete
-    Assert.assertEquals(0, mMasterInfo.getUsedBytes());
-    Assert.assertTrue(mTfs.getFile(fileId).isComplete());
+    Assert.assertEquals(0, mBlockMasterClient.getUsedBytes());
+    Assert.assertTrue(mTfs.getInfo(file).isComplete);
   }
\ No newline at end of file
