diff --git a/old_product.java b/new_product.java
index af21b49..bce6b31 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,14 +3,14 @@ public static String concatPath(Object base, Object... paths) throws IllegalArgu
     Preconditions.checkArgument(paths != null, "Failed to concatPath: a null set of paths");
     List<String> trimmedPathList = new ArrayList<>();
     String trimmedBase =
-        CharMatcher.is(AlluxioURI.SEPARATOR.charAt(0)).trimTrailingFrom(base.toString().trim());
+        CharMatcher.is(AlluxioURI.SEPARATOR.charAt(0)).trimTrailingFrom(base.toString());
     trimmedPathList.add(trimmedBase);
     for (Object path : paths) {
       if (path == null) {
         continue;
       }
       String trimmedPath =
-          CharMatcher.is(AlluxioURI.SEPARATOR.charAt(0)).trimFrom(path.toString().trim());
+          CharMatcher.is(AlluxioURI.SEPARATOR.charAt(0)).trimFrom(path.toString());
       if (!trimmedPath.isEmpty()) {
         trimmedPathList.add(trimmedPath);
       }
