diff --git a/old_test.java b/new_test.java
index ad37b10..90257d6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,11 +20,6 @@
     assertEquals("/foo/bar", PathUtils.concatPath("/foo/", "/bar"));
     assertEquals("/foo/bar", PathUtils.concatPath("/foo/", "/bar/"));
 
-    // Whitespace must be trimmed.
-    assertEquals("/foo/bar", PathUtils.concatPath("/foo ", "bar  "));
-    assertEquals("/foo/bar", PathUtils.concatPath("/foo ", "  bar"));
-    assertEquals("/foo/bar", PathUtils.concatPath("/foo ", "  bar  "));
-
     // Redundant separator must be trimmed.
     assertEquals("/foo/bar", PathUtils.concatPath("/foo/", "bar//"));
 
