diff --git a/old_test.java b/new_test.java
index c2cda0a..a350e5f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,28 +1,28 @@
 @Test
-    public void testParseServiceRequestQueryParams() {
-        Map<String, String> paramMap = GatewayServlet.parseServiceRequestQueryParams(null);
+    public void testParseApiRequestQueryParams() {
+        Map<String, String> paramMap = GatewayServlet.parseApiRequestQueryParams(null);
         Assert.assertNotNull(paramMap);
 
-        paramMap = GatewayServlet.parseServiceRequestQueryParams("param1");
+        paramMap = GatewayServlet.parseApiRequestQueryParams("param1");
         Assert.assertNull(paramMap.get("param1"));
 
-        paramMap = GatewayServlet.parseServiceRequestQueryParams("param1=value1");
+        paramMap = GatewayServlet.parseApiRequestQueryParams("param1=value1");
         Assert.assertEquals("value1", paramMap.get("param1"));
 
-        paramMap = GatewayServlet.parseServiceRequestQueryParams("param1=value1&param2");
+        paramMap = GatewayServlet.parseApiRequestQueryParams("param1=value1&param2");
         Assert.assertEquals("value1", paramMap.get("param1"));
         Assert.assertNull(paramMap.get("param2"));
 
-        paramMap = GatewayServlet.parseServiceRequestQueryParams("param1=value1&param2=value2");
+        paramMap = GatewayServlet.parseApiRequestQueryParams("param1=value1&param2=value2");
         Assert.assertEquals("value1", paramMap.get("param1"));
         Assert.assertEquals("value2", paramMap.get("param2"));
 
-        paramMap = GatewayServlet.parseServiceRequestQueryParams("param1=value1&param2=value2&param3=value3");
+        paramMap = GatewayServlet.parseApiRequestQueryParams("param1=value1&param2=value2&param3=value3");
         Assert.assertEquals("value1", paramMap.get("param1"));
         Assert.assertEquals("value2", paramMap.get("param2"));
         Assert.assertEquals("value3", paramMap.get("param3"));
 
-        paramMap = GatewayServlet.parseServiceRequestQueryParams("param1=hello%20world&param2=hello+world&param3=hello world");
+        paramMap = GatewayServlet.parseApiRequestQueryParams("param1=hello%20world&param2=hello+world&param3=hello world");
         Assert.assertEquals("hello world", paramMap.get("param1"));
         Assert.assertEquals("hello world", paramMap.get("param2"));
         Assert.assertEquals("hello world", paramMap.get("param3"));
