diff --git a/old_product.java b/new_product.java
index a68ed99..3996599 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-protected static final Map<String, String> parseServiceRequestQueryParams(String queryString) {
+protected static final Map<String, String> parseApiRequestQueryParams(String queryString) {
         Map<String, String> rval = new LinkedHashMap<>();
         if (queryString != null) {
             try {
