diff --git a/old_test.java b/new_test.java
index 6bc708c..acfe4e4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testAddCommand() throws GeniePreconditionException {
         final CommandEntity commandEntity = new CommandEntity();
-        commandEntity.setId("commandId");
+        commandEntity.setUniqueId("commandId");
         Assert.assertNotNull(this.c.getCommands());
         Assert.assertTrue(this.c.getCommands().isEmpty());
         this.c.addCommand(commandEntity);
