diff --git a/old_product.java b/new_product.java
index 4c757c4..1cee544 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,6 @@
 public void addCommand(@NotNull final CommandEntity command) throws GeniePreconditionException {
-        if (
-            this.commands
-                .stream()
-                .map(CommandEntity::getId)
-                .filter(id -> id.equals(command.getId()))
-                .count() != 0
-            ) {
-            throw new GeniePreconditionException("A command with id " + command.getId() + " is already added");
+        if (this.commands.contains(command)) {
+            throw new GeniePreconditionException("A command with id " + command.getUniqueId() + " is already added");
         }
         this.commands.add(command);
         command.getClusters().add(this);
