diff --git a/old_test.java b/new_test.java
index 67e60df..25d9c48 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,8 +4,9 @@
         Job initial = new Job();
         UUID uuid = UUID.randomUUID();
         initial.setJobID(uuid.toString());
+        initial.setUserName("myUserName");
         pm.createEntity(initial);
         initial.setJobStatus(JobStatus.FAILED);
         Job updated = pm.updateEntity(initial);
-        Assert.assertEquals("FAILED", updated.getStatus());
+        Assert.assertEquals(JobStatus.FAILED, updated.getStatus());
     }
\ No newline at end of file
