diff --git a/old_product.java b/new_product.java
index 3953288..fa08ecc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public DateTimeParseContext parseToContext(CharSequence text, ParsePosition position) {
+public DateTimeBuilder parseToBuilder(CharSequence text, ParsePosition position) {
         DateTimes.checkNotNull(text, "Text must not be null");
         DateTimes.checkNotNull(position, "ParsePosition must not be null");
         DateTimeParseContext context = new DateTimeParseContext(locale, symbols);
@@ -9,5 +9,5 @@ public DateTimeParseContext parseToContext(CharSequence text, ParsePosition posi
             return null;
         }
         position.setIndex(pos);
-        return context;
+        return context.toBuilder();  // TODO: this can fail and throw CalendricalException, but should it?
     }
\ No newline at end of file
