diff --git a/old_product.java b/new_product.java
index 9583c80..3a19996 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static <K, O, V> Map<K, V> constructSubMap(Map<K, O> map,K[] includeKeys,String extractPropertyName,Class<K> keysClass){
+public static <K, O, V> Map<K, V> extractSubMap(Map<K, O> map,K[] includeKeys,String extractPropertyName,Class<K> keysClass){
         if (Validator.isNullOrEmpty(map)){
             throw new NullPointerException("the map is null or empty!");
         }
