diff --git a/old_product.java b/new_product.java
index d0a6f6a..8e69cab 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,5 +6,8 @@ public ServiceCall<EntityExport> getEntity(GetEntityOptions getEntityOptions) {
     if (getEntityOptions.export() != null) {
       builder.query("export", String.valueOf(getEntityOptions.export()));
     }
+    if (getEntityOptions.includeAudit() != null) {
+      builder.query("include_audit", String.valueOf(getEntityOptions.includeAudit()));
+    }
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(EntityExport.class));
   }
\ No newline at end of file
