diff --git a/old_test.java b/new_test.java
index 82544e1..c4f5d43 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void testGetEntity() {
 
-    String entity = "Hello" + UUID.randomUUID().toString();  // gotta be unique
+    String entity = "Hello" + UUID.randomUUID().toString(); // gotta be unique
     String entityDescription = "Description of " + entity;
     String entityValue = "Value of " + entity;
     List<CreateValue> entityValues = new ArrayList<CreateValue>();
@@ -17,7 +17,10 @@
     Date start = new Date();
 
     try {
-      GetEntityOptions getOptions = new GetEntityOptions.Builder(workspaceId, entity).export(true).build();
+      GetEntityOptions getOptions = new GetEntityOptions.Builder(workspaceId, entity)
+          .export(true)
+          .includeAudit(true)
+          .build();
       EntityExport response = service.getEntity(getOptions).execute();
       assertNotNull(response);
       assertNotNull(response.getEntityName());
