diff --git a/old_product.java b/new_product.java
index 922d760..dcf7174 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,11 +13,7 @@ public static Integer[] sort(final ValueList vl, final StandardFunc sf, final Co
             Item m = v1.itemAt(v), n = v2.itemAt(v);
             if(m == Dbl.NAN || m == Flt.NAN) m = null;
             if(n == Dbl.NAN || n == Flt.NAN) n = null;
-            if(m != null && n != null && !m.comparable(n)) {
-              throw m instanceof FItem ? FIEQ_X.get(sf.info, m.type) :
-                    n instanceof FItem ? FIEQ_X.get(sf.info, n.type) :
-                    diffError(m, n, sf.info);
-            }
+            if(m != null && n != null && !m.comparable(n)) throw diffError(m, n, sf.info);
             final int d = m == null ? n == null ? 0 : -1 : n == null ? 1 :
               m.diff(n, coll, sf.info);
             if(d != 0 && d != Item.UNDEF) return d;
