diff --git a/old_product.java b/new_product.java
index 5015ca5..0769197 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 public Builder<T> rendering(Rendering config)
 		{
 			checkForNull(config, "Rendering is null.");
+			updateStatus(Properties.RESIZER_FACTORY, Status.CANNOT_SET);
 			updateStatus(Properties.RENDERING, Status.ALREADY_SET);
 			rendering = config;
 			return this;
