diff --git a/old_test.java b/new_test.java
index d98e890..5ca83f6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,16 @@
 @Test
-	public void fullView() {
+	public void fullView_Transform() {
 		IntrinsicParameters param = new IntrinsicParameters().
 				fsetK(300, 320, 0, 150, 130, width, height).fsetRadial(0.1, 0.05);
 
-		PointTransform_F32 adjusted = LensDistortionOps.fullView(param,null,true);
-		checkBorderOutside(adjusted);
+		PointTransform_F32 adjToDist = LensDistortionOps.fullView(param, null, true);
+		PointTransform_F32 distToAdj = LensDistortionOps.fullView(param, null, false);
+
+		checkBorderOutside(adjToDist,distToAdj);
 
 		param = new IntrinsicParameters().
 				fsetK(300, 320, 0, 150, 130, width, height).fsetRadial(-0.1,-0.05);
-		adjusted = LensDistortionOps.fullView(param,null,true);
-		checkBorderOutside(adjusted);
-
+		adjToDist = LensDistortionOps.fullView(param, null, true);
+		distToAdj = LensDistortionOps.fullView(param, null, false);
+		checkBorderOutside(adjToDist,distToAdj);
 	}
\ No newline at end of file
