diff --git a/old_product.java b/new_product.java
index 8878a4d..cc9b9e4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public static PointTransform_F32 fullView( IntrinsicParameters param,
 											   IntrinsicParameters paramAdj ,
-											   boolean inputUndistorted ) {
+											   boolean adjToDistorted ) {
 
 		PointTransform_F32 remove_p_to_p = distortTransform(param).undistort_F32(true, true);
 
@@ -19,12 +19,5 @@ public static PointTransform_F32 fullView( IntrinsicParameters param,
 		// adjustment matrix
 		DenseMatrix64F A = new DenseMatrix64F(3,3,true,scale,0,deltaX,0,scale,deltaY,0,0,1);
 
-		if( inputUndistorted ) {
-			PointTransform_F32 add_p_to_p = distortTransform(param).distort_F32(true, true);
-			// by setting forward false here the matrix is applied first
-			return PerspectiveOps.adjustIntrinsic_F32(add_p_to_p, false, param, A, paramAdj);
-		} else {
-			CommonOps.invert(A);
-			return PerspectiveOps.adjustIntrinsic_F32(remove_p_to_p, true, param, A, paramAdj);
-		}
+		return adjustmentTransform(param, paramAdj, adjToDistorted, remove_p_to_p, A);
 	}
\ No newline at end of file
