diff --git a/old_product.java b/new_product.java
index 42e688e..0c384a3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -33,7 +33,7 @@ public Community[] getSubcommunities() throws SQLException
         {
             while (tri.hasNext())
             {
-                TableRow row = tri.next();
+                TableRow row = tri.next(ourContext);
 
                 // First check the cache
                 Community fromCache = (Community) ourContext.fromCache(
