diff --git a/old_test.java b/new_test.java
index c37017d..08ee9b7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,4 +5,7 @@
     count(_MAP_PUT.args(" map{}", "a", "b"), 1);
     count(_MAP_PUT.args(" map{ 'a': 'b' }", "c", "d"), 2);
     count(_MAP_PUT.args(" map{ 'a': 'b' }", "c", "d"), 2);
+
+    error(_MAP_PUT.args(" map{ xs:time('01:01:01'):'b' }", "xs:time('01:01:02+01:00')", "1"),
+        MAP_TZ);
   }
\ No newline at end of file
