diff --git a/old_product.java b/new_product.java
index 39d956e..be2c541 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@
     if (mPosition == mLength) { // at end of file
       return -1;
     }
-    CountingRetry retry = new CountingRetry(MAX_WORKERS_TO_RETRY);
+    CountingRetry retry = new CountingRetry(mBlockWorkerClientReadRetry);
     IOException lastException = null;
     while (retry.attempt()) {
       try {
