diff --git a/old_test.java b/new_test.java
index 8684a7b..84e549e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,12 +6,14 @@
       FileSystemTestUtils.createByteFile(mFileSystem, uri, mWriteAlluxio, k);
 
       URIStatus status = mFileSystem.getStatus(uri);
-      InStreamOptions options = new InStreamOptions(status);
+      InStreamOptions options = new InStreamOptions(status, ServerConfiguration.global());
       long blockId = status.getBlockIds().get(0);
-      BlockInfo info = AlluxioBlockStore.create().getInfo(blockId);
+      BlockInfo info =
+          AlluxioBlockStore.create(FileSystemContext.create(ServerConfiguration.global()))
+              .getInfo(blockId);
       WorkerNetAddress workerAddr = info.getLocations().get(0).getWorkerAddress();
       BlockInStream is =
-          BlockInStream.create(FileSystemContext.get(), options.getBlockInfo(blockId),
+          BlockInStream.create(mFsContext, options.getBlockInfo(blockId),
               workerAddr, BlockInStreamSource.REMOTE, options);
       byte[] ret = new byte[k];
       int read = is.read(ret);
