diff --git a/old_test.java b/new_test.java
index d0daf28..3bdf98c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,6 +16,9 @@
         String harvardStage = shibUtil.getDisplayNameFromDiscoFeed("https://stage.fed.huit.harvard.edu/idp/shibboleth", discoFeedExample);
         assertEquals("Harvard Test IdP", harvardStage);
 
+        String minimal = shibUtil.getDisplayNameFromDiscoFeed("https://minimal.com/shibboleth", discoFeedExample);
+        assertEquals(null, minimal);
+
         String unknown = shibUtil.getDisplayNameFromDiscoFeed("https://nosuchdomain.com/idp/shibboleth", discoFeedExample);
         assertEquals(null, unknown);
 
