diff --git a/old_test.java b/new_test.java
index 2801855..7fefd48 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,8 +10,8 @@
 			numFound++;
 
 			Class imageType = method.getParameterTypes()[0];
-			ImageInteger input = (ImageInteger) GeneralizedImageOps.createSingleBand(imageType, width, height);
-			ImageInteger output = (ImageInteger) GeneralizedImageOps.createSingleBand(imageType, width, height);
+			GrayI input = (GrayI) GeneralizedImageOps.createSingleBand(imageType, width, height);
+			GrayI output = (GrayI) GeneralizedImageOps.createSingleBand(imageType, width, height);
 
 			equalizeLocalNaive(input, output);
 
