diff --git a/old_product.java b/new_product.java
index 13e17ea..e64a997 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-public static void equalizeLocalNaive( ImageUInt8 input , int radius , ImageUInt8 output ,
-										   int histogram[] )
+public static void equalizeLocalNaive(GrayU8 input , int radius , GrayU8 output ,
+										  int histogram[] )
 	{
 		int width = 2*radius+1;
 		int maxValue = histogram.length-1;
