diff --git a/old_product.java b/new_product.java
index 391b784..e5faad1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public synchronized void cleanupUser(long userId) {
+public synchronized void cleanupUserTempBlocks(long userId, List<Long> tempBlockIds) {
     for (StorageTier tier : mTiers) {
       for (StorageDir dir : tier.getStorageDirs()) {
-        dir.cleanupUser(userId);
+        dir.cleanupUserTempBlocks(userId, tempBlockIds);
       }
     }
   }
\ No newline at end of file
