diff --git a/old_test.java b/new_test.java
index 7fd586f..167b9ed 100644
--- a/old_test.java
+++ b/new_test.java
@@ -21,13 +21,17 @@
 
     // Check the temporary blocks belonging to TEST_USER_ID
     List<TempBlockMeta> actual = mDir.getUserTempBlocks(TEST_USER_ID);
+    List<Long> actualBlockIds = new ArrayList<Long>(actual.size());
+    for (TempBlockMeta tempBlockMeta : actual) {
+      actualBlockIds.add(tempBlockMeta.getBlockId());
+    }
     Assert.assertEquals(Sets.newHashSet(tempBlockMeta1, tempBlockMeta2),
         new HashSet<TempBlockMeta>(actual));
     Assert.assertTrue(mDir.hasTempBlockMeta(tempBlockId1));
     Assert.assertTrue(mDir.hasTempBlockMeta(tempBlockId2));
 
     // Two temp blocks created by TEST_USER_ID are expected to be removed
-    mDir.cleanupUser(TEST_USER_ID);
+    mDir.cleanupUserTempBlocks(TEST_USER_ID, actualBlockIds);
     Assert.assertFalse(mDir.hasTempBlockMeta(tempBlockId1));
     Assert.assertFalse(mDir.hasTempBlockMeta(tempBlockId2));
     // Temp block created by otherUserId is expected to stay
