diff --git a/old_product.java b/new_product.java
index f9c1978..a7299ca 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,14 +8,15 @@
       // This aborts the block if the block is not fully read.
       updateBlockWriter(mBlockMeta.getBlockSize());
 
-      Closer closer = Closer.create();
-      if (mBlockWriter != null) {
-        closer.register(mBlockWriter);
-      }
       if (mUnderFileSystemInputStream != null) {
-        closer.register(mUnderFileSystemInputStream);
+        UnderFileInputStreamsManager.INSTANCE.checkIn(mUnderFileSystemInputStream);
+        mUnderFileSystemInputStream = null;
+      }
+
+      if(mBlockWriter!=null) {
+        mBlockWriter.close();
       }
-      closer.close();
+
     } finally {
       mClosed = true;
     }
