diff --git a/old_test.java b/new_test.java
index 0010ba6..bcd51d3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -93,8 +93,8 @@
         } catch(CommandLineException e) {
             final int relativeIndex = env.getInstalledImage().getJbossHome().getAbsolutePath().length() + 1;
             // TODO modules and bundles are not checked at the moment
-            //assertConflicts(e, bundleConflictName + ":main", moduleConflictName + ":main", conflictFile.getAbsolutePath().substring(relativeIndex));
-            assertConflicts(e, conflictFile.getAbsolutePath().substring(relativeIndex));
+            assertConflicts(e, bundleConflictName + ":main", moduleConflictName + ":main", conflictFile.getAbsolutePath().substring(relativeIndex));
+            //assertConflicts(e, conflictFile.getAbsolutePath().substring(relativeIndex));
         } finally {
             ctx.terminateSession();
         }
