diff --git a/old_product.java b/new_product.java
index 0a6acbd..3195c21 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
 static void rollback(final String patchId, final Collection<ContentModification> originalPatch, final Collection<ContentModification> rollbackPatch,
-                         final Map<Location, ContentTaskDefinition> modifications, final ContentItemFilter filter, final boolean originalsOnly) {
+                         final Map<Location, ContentTaskDefinition> modifications, final ContentItemFilter filter,
+                         final PatchingTaskContext.Mode mode) {
 
         // Process the original patch information
         final Map<Location, ContentModification> originalModifications = new HashMap<Location, ContentModification>();
@@ -16,15 +17,6 @@ static void rollback(final String patchId, final Collection<ContentModification>
             }
             final Location location = new Location(item);
             final ContentModification original = originalModifications.remove(location);
-            if (original == null) {
-                if (originalsOnly) {
-                    continue;
-                }
-                if (modification.getType() != ModificationType.ADD) {
-                    throw new IllegalStateException(item.toString()); // Only for development purpose
-                }
-            }
-
             final ContentEntry contentEntry = new ContentEntry(patchId, modification);
             ContentTaskDefinition definition = modifications.get(location);
             if (definition == null) {
@@ -44,7 +36,8 @@ static void rollback(final String patchId, final Collection<ContentModification>
                 //
                 definition.setTarget(contentEntry);
             }
-            if (original == null) {
+            if (original == null
+                    || mode == PatchingTaskContext.Mode.ROLLBACK) {
                 continue;
             }
 
