diff --git a/old_product.java b/new_product.java
index f951723..f1211f8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,8 @@
 public List<Voice> getVoices() {
-		HttpRequestBase request = Request.Get("/v1/voices").build();
-		try {
-			HttpResponse response = execute(request);
-			JsonObject jsonObject = ResponseUtil.getJsonObject(response);
-			List<Voice> voices = GsonSingleton.getGson().fromJson(
-					jsonObject.get("voices"), listVoiceType);
-			return voices;
-		} catch (IOException e) {
-			throw new RuntimeException(e);
-		}
-	}
\ No newline at end of file
+    final Request request = RequestBuilder.get("/v1/voices").build();
+    final Response response = execute(request);
+    final JsonObject jsonObject = ResponseUtil.getJsonObject(response);
+    final List<Voice> voices =
+        GsonSingleton.getGson().fromJson(jsonObject.get("voices"), listVoiceType);
+    return voices;
+  }
\ No newline at end of file
