diff --git a/old_product.java b/new_product.java
index 2752acb..f72fbc7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static TIntArrayList getNeighborsND(SpatialLattice l, int columnIndex, int radius, boolean wrapAround) {
+public TIntArrayList getNeighborsND(SpatialLattice l, int columnIndex, int radius, boolean wrapAround) {
         final int[] dimensions = l.getInputDimensions();
         int[] columnCoords = l.getInputMatrix().computeCoordinates(columnIndex);
         List<int[]> dimensionCoords = new ArrayList<int[]>();
