diff --git a/old_product.java b/new_product.java
index 3a31f21..5f6bc8a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public HashCodeBuilder append(final double[] array) {
         if (array == null) {
             iTotal = iTotal * iConstant;
         } else {
-            for (double element : array) {
+            for (final double element : array) {
                 append(element);
             }
         }
