diff --git a/old_test.java b/new_test.java
index 8652719..bd442ba 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,5 @@
 @Test
 	public void startModule_shouldCreateDwrModulesXmlIfNotExists() throws Exception {
-		partialMockWebModuleUtilForMessagesTests();
-		
 		// create dummy module and start it
 		Module mod = buildModuleForMessageTest();
 		ModuleFactory.getStartedModulesMap().put(mod.getModuleId(), mod);
