diff --git a/old_product.java b/new_product.java
index 04a332c..085db5f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -19,9 +19,6 @@ public static boolean startModule(Module mod, ServletContext servletContext, boo
 				webInf.mkdir();
 			}
 			
-			copyModuleMessagesIntoWebapp(mod, realPath);
-			log.debug("Done copying messages");
-			
 			// flag to tell whether we added any xml/dwr/etc changes that necessitate a refresh
 			// of the web application context
 			boolean moduleNeedsContextRefresh = false;
