diff --git a/old_test.java b/new_test.java
index 121e8c1..d56796b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-    public void testSetInput0() {
-        Source s = new Source();
-        List<String> values = s.getAllValues( OPT.INPUT );
+    public void testSetInput0() throws SchemagenOptionsConfigurationException {
+        SchemagenOptions so = new SchemagenOptions(null, new Source()); 
+        List<String> values = so.getAllValues( OPT.INPUT );
         assertListMatch( new String[] {}, new String[] {}, 0, values );
     }
\ No newline at end of file
