diff --git a/old_product.java b/new_product.java
index fd2a3c3..c9111ef 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,7 +12,7 @@
                 }
                 return field.range();
             }
-            throw new UnsupportedTemporalTypeException("Unsupported field: " + field.getName());
+            throw new UnsupportedTemporalTypeException("Unsupported field: " + field);
         }
         return field.rangeRefinedBy(this);
     }
\ No newline at end of file
