diff --git a/old_test.java b/new_test.java
index f063b36..d9ab86c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,8 @@
     public void test_range() {
         // eras may be added after release
         for (JapaneseEra era : JapaneseEra.values()) {
-            assertEquals(era.range(ERA).getMinimum(), -999);
-            assertEquals(era.range(ERA).getLargestMinimum(), -999);
+            assertEquals(era.range(ERA).getMinimum(), -1);
+            assertEquals(era.range(ERA).getLargestMinimum(), -1);
             assertEquals(era.range(ERA).getSmallestMaximum(), era.range(ERA).getMaximum());
             assertEquals(era.range(ERA).getMaximum() >= 2, true);
         }
