diff --git a/old_product.java b/new_product.java
index 26dc1f7..09cd47a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,18 +1,18 @@
 @Override public ResultSet getTables(String catalog, String schemaPtrn, String tblNamePtrn,
         String[] tblTypes) throws SQLException {
-        updateMetaData();
+        conn.ensureNotClosed();
 
-        List<List<?>> rows = new LinkedList<>();
+        List<List<?>> rows = Collections.emptyList();
 
-        if (isValidCatalog(catalog) && (tblTypes == null || Arrays.asList(tblTypes).contains("TABLE"))) {
-            for (Map.Entry<String, Map<String, Map<String, ColumnInfo>>> schema : meta.entrySet()) {
-                if (matches(schema.getKey(), schemaPtrn)) {
-                    for (String tbl : schema.getValue().keySet()) {
-                        if (matches(tbl, tblNamePtrn))
-                            rows.add(tableRow(schema.getKey(), tbl));
-                    }
-                }
-            }
+        boolean areTypesValid = tblTypes == null || Arrays.asList(tblTypes).contains("TABLE");
+
+        if (isValidCatalog(catalog) && areTypesValid) {
+            List<JdbcTableMeta> tabMetas = meta.getTablesMeta(schemaPtrn, tblNamePtrn);
+
+            rows = new ArrayList<>(tabMetas.size());
+
+            for (JdbcTableMeta m : tabMetas)
+                rows.add(tableRow(m));
         }
 
         return new JdbcResultSet(true, null,
