diff --git a/old_product.java b/new_product.java
index 28418d8..e2e4c29 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,5 @@
-public static int[] selectPyramidScale( int imageWidth , int imageHeight, int featureRadius ) {
+public static int[] selectPyramidScale( int imageWidth , int imageHeight, int minSize ) {
 		int w = Math.max(imageWidth,imageHeight);
-		int minSize = (featureRadius*2+1)*5;
 
 		int maxScale = w/minSize;
 		int n = 1;
