diff --git a/old_test.java b/new_test.java
index ef88e0a..04ba702 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,4 +2,7 @@
     public void testSetName() {
         QueueConfig queueConfig = new QueueConfig().setName("a test name");
         assertEquals("a test name", queueConfig.getName());
+        assertEquals("default", queueConfig.getBackingMapName());
+        queueConfig.setBackingMapName("backingMap");
+        assertEquals("backingMap", queueConfig.getBackingMapName());
     }
\ No newline at end of file
