diff --git a/old_test.java b/new_test.java
index 36793bf..23e3119 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,9 @@
 @Test
-    public void testGetTenant() throws Exception {
-        TenantUserDto tenant = createTenant();
-        
-        TenantUserDto storedTenant = client.getTenant(tenant.getId());
-        
-        Assert.assertNotNull(storedTenant);
-        assertTenantsEquals(tenant, storedTenant);
-        client.deleteTenant(tenant.getId());
-    }
\ No newline at end of file
+  public void testGetTenant() throws Exception {
+    TenantDto tenant = createTenant();
+
+    TenantDto storedTenant = client.getTenant(tenant.getId());
+
+    Assert.assertNotNull(storedTenant);
+    assertTenantsEquals(tenant, storedTenant);
+  }
\ No newline at end of file
