diff --git a/old_product.java b/new_product.java
index afec6ae..8fd7830 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
 public void getTenant(String tenantId,
-            final AsyncCallback<TenantUserDto> callback) {
-        rpcService.getTenant(tenantId,
-                new DataCallback<TenantUserDto>(callback) {
-                    @Override
-                    protected void onResult(TenantUserDto result) {
-                    }
-                });
-    }
\ No newline at end of file
+                        final AsyncCallback<TenantDto> callback) {
+    tenantRpcService.getTenant(tenantId,
+        new DataCallback<TenantDto>(callback) {
+          @Override
+          protected void onResult(TenantDto result) {
+          }
+        });
+  }
\ No newline at end of file
