diff --git a/old_product.java b/new_product.java
index e9a8589..8331ea5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,8 +4,6 @@ public static void export(final Data data, final String path, final SerializerOp
     final IOFile root = new IOFile(path);
     root.md();
 
-    final HashSet<String> exported = new HashSet<>();
-
     // XML documents
     final IntList il = data.resources.docs();
     // raw files
@@ -25,6 +23,7 @@ public static void export(final Data data, final String path, final SerializerOp
     }
 
     // XML documents
+    final HashSet<String> exported = new HashSet<>();
     final int is = il.size();
     for(int i = 0; i < is; i++) {
       final int pre = il.get(i);
