diff --git a/old_test.java b/new_test.java
index 6b66b0a..9fe2402 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,5 +7,5 @@
     min.complete( context, resultEntryCollector.iterator() );
     Tuple tuple = resultEntryCollector.iterator().next().getTuple();
 
-    assertEquals( "Got expected initial value on start", Double.MAX_VALUE, tuple.getDouble( 0 ), 0.0d );
+    assertEquals( "Got expected initial value on start", null, tuple.get( 0 ) );
     }
\ No newline at end of file
