diff --git a/old_product.java b/new_product.java
index 96063dc..45d84f2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @SuppressWarnings("unchecked")
-    public static <T> Map<String, T> toMap(String json,Class<T> clazz,Map<String, Class<?>> classMap){
-        LOGGER.debug("in json:{}", json);
+    public static <T> Map<String, T> toMap(String json,Class<T> rootClass,Map<String, Class<?>> classMap){
+        LOGGER.debug("in json:[{}],klass:[{}]", json, rootClass);
 
         if (Validator.isNullOrEmpty(json)){
             return Collections.emptyMap();
@@ -13,8 +13,8 @@
         while (keys.hasNext()){
             String key = keys.next();
             Object value = jsonObject.get(key);
-            LOGGER.debug("key:[{}], value:[{}]", key, value);
-            map.put(key, null == clazz ? (T) value : toBean(value, clazz, classMap));//如果clazz是null,表示不需要转换
+            LOGGER.debug("key:[{}], value:[{}],value type is:[{}]", key, value, value.getClass().getName());
+            map.put(key, null == rootClass ? (T) value : toBean(value, rootClass, classMap));//如果klass是null,表示不需要转换
         }
         return map;
     }
\ No newline at end of file
