diff --git a/old_test.java b/new_test.java
index b5aa5f6..8fcf388 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,8 @@
 @Test
     public void toMap(){
-        String json = "{'data1':{'name':'get'},'data2':{'name':'set'}}";
-        Map<String, Person> map = JsonUtil.toMap(json, Person.class);
+        Map<String, Person> map = JsonUtil.toMap("{'data1':{'name':'get'},'data2':{'name':'set'}}", Person.class);
         LOGGER.info(JsonUtil.format(map));
+
+        Map<String, String> map1 = JsonUtil.toMap("{'data1':{'name':'get'},'data2':{'name':'set'}}", String.class);
+        LOGGER.info(JsonUtil.format(map1));
     }
\ No newline at end of file
