diff --git a/old_product.java b/new_product.java
index d6d0dfe..4179a6a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
 @POST
   @Path(CREATE_FILE)
+  @ReturnType("java.lang.Void")
   public Response createFile(@QueryParam("path") String path,
       @QueryParam("persisted") Boolean persisted, @QueryParam("recursive") Boolean recursive,
       @QueryParam("blockSizeBytes") Long blockSizeBytes, @QueryParam("ttl") Long ttl) {
