diff --git a/old_test.java b/new_test.java
index 9c6a5ea..695b8a5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,11 @@
-@Test(expected = APIException.class)
+@Test
 	@Verifies(value = "should not allow editing an existing order frequency that is in use", method = "saveOrderFrequency(OrderFrequency)")
 	public void saveOrderFrequency_shouldNotAllowEditingAnExistingOrderFrequencyThatIsInUse() throws Exception {
 		OrderFrequency orderFrequency = Context.getOrderService().getOrderFrequency(1);
 		assertNotNull(orderFrequency);
 		
 		orderFrequency.setFrequencyPerDay(4d);
+		expectedException.expect(APIException.class);
+		expectedException.expectMessage("This order frequency cannot be edited because it is already in use");
 		Context.getOrderService().saveOrderFrequency(orderFrequency);
 	}
\ No newline at end of file
