diff --git a/old_test.java b/new_test.java
index 419d021..5f7cf47 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
             final String content = UserInfoJsonExpectation.SERVLET_OUTPUT_PREFIX + "{\n\r\"key\":\t \n\"value\"\r }";
             mockery.checking(new Expectations() {
                 {
-                    one(webResponse).getContentAsString();
+                    allowing(webResponse).getContentAsString();
                     will(returnValue(content));
                 }
             });
