diff --git a/old_product.java b/new_product.java
index 9fa378d..cee582d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,11 @@
 protected JsonObject readJsonFromContent(Object contentToValidate) throws Exception {
-        if (contentToValidate == null || !(contentToValidate instanceof String)) {
-            String className = (contentToValidate == null) ? null : contentToValidate.getClass().getName();
-            throw new Exception("Provided content is not a string as expected so cannot be validated. The content is of type " + className + ".");
+        if (contentToValidate == null) {
+            throw new Exception("Provided content is null so cannot be validated.");
         }
         JsonObject obj = null;
         try {
-            obj = Json.createReader(new StringReader((String) contentToValidate)).readObject();
+            String responseText = WebResponseUtils.getResponseText(contentToValidate);
+            obj = Json.createReader(new StringReader(responseText)).readObject();
         } catch (Exception e) {
             throw new Exception("Failed to read JSON data from the provided content. The exception was [" + e + "]. The content to validate was: [" + contentToValidate + "].");
         }
