diff --git a/old_product.java b/new_product.java
index c359204..934a47d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ public static void fullViewLeft(int imageWidth,int imageHeight,
 		Rectangle2D_F32 bound = DistortImageOps.boundBox_F32(imageWidth, imageHeight,
 				new PointToPixelTransform_F32(tranLeft));
 
-		double scaleX = imageHeight/bound.width;
+		double scaleX = imageWidth/bound.width;
 		double scaleY = imageHeight/bound.height;
 
 		double scale = Math.min(scaleX,scaleY);
