diff --git a/old_product.java b/new_product.java
index a0198c2..7f4dda3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@
             new Thread(new Runnable() {
                 @Override
                 public void run() {
-                    final Response result = Response.created(getUriBuilder().segment("responseFilterAsync").build()).build();
+                    final Response result = Response.created(URI.create("responseFilterAsync")).build();
                     asyncResponse.resume(result);
                 }
             }).start();
