diff --git a/old_test.java b/new_test.java
index ae9fe14..ef1bcd3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -22,4 +22,10 @@
         assertEquals(14, Conversion.hexDigitToInt('e'));
         assertEquals(15, Conversion.hexDigitToInt('F'));
         assertEquals(15, Conversion.hexDigitToInt('f'));
+        try {
+            Conversion.hexDigitToInt('G');
+            fail("Thrown " + IllegalArgumentException.class.getName() + " expected");
+        } catch (final IllegalArgumentException e) {
+            // OK
+        }
     }
\ No newline at end of file
