diff --git a/old_test.java b/new_test.java
index adb2b1e..6fda544 100644
--- a/old_test.java
+++ b/new_test.java
@@ -19,7 +19,6 @@
         
         BootstrapAggregating.TrainingParameters param = new BootstrapAggregating.TrainingParameters();
         param.setMaxWeakClassifiers(5);
-        param.setWeakClassifierClass(MultinomialNaiveBayes.class);
         
         
         MultinomialNaiveBayes.TrainingParameters trainingParameters = new MultinomialNaiveBayes.TrainingParameters();
