diff --git a/old_product.java b/new_product.java
index 0b36fe8..1ef3834 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
-    public <U, R> ManagedCompletableFuture<R> thenCombine(CompletionStage<? extends U> other, BiFunction<? super T, ? super U, ? extends R> action) {
+    public <U, R> CompletableFuture<R> thenCombine(CompletionStage<? extends U> other, BiFunction<? super T, ? super U, ? extends R> action) {
         // Reject ManagedTask so that we have the flexibility to decide later how to handle ManagedTaskListener and execution properties
         if (action instanceof ManagedTask)
             throw new IllegalArgumentException(ManagedTask.class.getName());
@@ -16,6 +16,6 @@
             CompletableFuture<R> dependentStage = completableFuture.thenCombine(other, action);
             return new ManagedCompletableFuture<R>(dependentStage, defaultExecutor, null);
         } else {
-            return (ManagedCompletableFuture<R>) super.thenCombine(other, action);
+            return super.thenCombine(other, action);
         }
     }
\ No newline at end of file
