diff --git a/old_product.java b/new_product.java
index 0cd6d59..d3dc567 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @Override
-    public AppUsagePerServiceBean getAppUsagePerService(String organizationId, String applicationId,
+    public AppUsagePerApiBean getAppUsagePerApi(String organizationId, String applicationId,
             String version, DateTime from, DateTime to) {
-        AppUsagePerServiceBean rval = new AppUsagePerServiceBean();
+        AppUsagePerApiBean rval = new AppUsagePerApiBean();
         return rval;
     }
\ No newline at end of file
