diff --git a/old_product.java b/new_product.java
index 41af89c..62549b8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static TachyonConf loadFromHadoopConfiguration(Configuration source) {
+public static Configuration loadFromHadoopConfiguration(org.apache.hadoop.conf.Configuration source) {
     // Load TachyonConf if any and merge to the one in TachyonFS
     // Push TachyonConf to the Job conf
     Properties tachyonConfProperties = null;
@@ -26,5 +26,5 @@ public static TachyonConf loadFromHadoopConfiguration(Configuration source) {
       }
     }
     LOG.info("Loading Tachyon properties from Hadoop configuration: {}", tachyonConfProperties);
-    return new TachyonConf(tachyonConfProperties);
+    return new Configuration(tachyonConfProperties);
   }
\ No newline at end of file
