diff --git a/old_test.java b/new_test.java
index 69f3ff3..1f346e3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,7 @@
       }
       tierAliasToBlockIds.put(tier.getTierAlias(), blockIdsOnTier);
     }
-    Map<String, List<Long>> actual = mBlockStoreMeta.getBlockList();
+    Map<String, List<Long>> actual = mBlockStoreMetaFull.getBlockList();
     Assert.assertEquals(TieredBlockStoreTestUtils.TIER_ALIAS.length, actual.keySet().size());
     Assert.assertEquals(tierAliasToBlockIds, actual);
   }
\ No newline at end of file
