diff --git a/old_test.java b/new_test.java
index 88c331f..f1a5ae6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,6 +10,7 @@
 		}
 		assertNotNull(answer);
 		assertTrue(answer);
+		verify(userEdit, times(1)).setEid(MOCK_PRINCIPAL);
 		verify(userEdit, times(1)).setFirstName(MOCK_FIRSTNAME);
 		verify(userEdit, times(1)).setLastName(MOCK_LASTNAME);
 		verify(userEdit, times(1)).setEmail(MOCK_EMAIL);
