diff --git a/old_product.java b/new_product.java
index 0a22cf3..accd6f5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,6 +13,7 @@ public boolean authenticateUser(String eid, UserEdit edit, String password) {
 				.getPrincipalLoggedIntoNakamura(getHttpServletRequest());
 		if (authInfo != null) {
 			if (eid.equalsIgnoreCase(authInfo.getPrincipal())) {
+				edit.setEid(authInfo.getPrincipal());
 				edit.setFirstName(authInfo.getFirstName());
 				edit.setLastName(authInfo.getLastName());
 				edit.setEmail(authInfo.getEmailAddress());
