diff --git a/old_product.java b/new_product.java
index 1ddaa7d..263e77b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,12 @@
 @Override
     public String toString() {
-        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd"); //$NON-NLS-1$
-        return dateFormat.format(this);
+        StringBuilder sb = new StringBuilder(10);
+
+        format((getYear() + 1900), 4, sb);
+        sb.append('-');
+        format((getMonth() + 1), 2, sb);
+        sb.append('-');
+        format(getDate(), 2, sb);
+
+        return sb.toString();
     }
\ No newline at end of file
