diff --git a/old_product.java b/new_product.java
index 5591636..dd67733 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public static <O> List<O> selectRejected(Collection<O> objectCollection,Predicate<O> predicate){
-        return Validator.isNullOrEmpty(objectCollection) ? Collections.<O> emptyList()
+        return isNullOrEmpty(objectCollection) ? Collections.<O> emptyList()
                         : (List<O>) CollectionUtils.selectRejected(objectCollection, predicate);
     }
\ No newline at end of file
