diff --git a/old_test.java b/new_test.java
index 884c83b..2112631 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,7 @@
     Method addNewConfigurationsFromXml = AbstractUpgradeCatalog.class.getDeclaredMethod
         ("addNewConfigurationsFromXml");
     Method updateTezConfiguration = UpgradeCatalog200.class.getDeclaredMethod("updateTezConfiguration");
+    Method updateFlumeEnvConfig = UpgradeCatalog200.class.getDeclaredMethod("updateFlumeEnvConfig");
     Method updateClusterEnvConfiguration = UpgradeCatalog200.class.getDeclaredMethod("updateClusterEnvConfiguration");
     Method updateConfigurationProperties = AbstractUpgradeCatalog.class.getDeclaredMethod
             ("updateConfigurationProperties", String.class, Map.class, boolean.class, boolean.class);
@@ -15,6 +16,7 @@
         .addMockedMethod(updateHiveDatabaseType)
         .addMockedMethod(addNewConfigurationsFromXml)
         .addMockedMethod(updateTezConfiguration)
+        .addMockedMethod(updateFlumeEnvConfig)
         .addMockedMethod(updateConfigurationProperties)
         .addMockedMethod(updateClusterEnvConfiguration)
         .addMockedMethod(persistHDPRepo)
@@ -31,6 +33,9 @@
     upgradeCatalog.updateTezConfiguration();
     expectLastCall().once();
 
+    upgradeCatalog.updateFlumeEnvConfig();
+    expectLastCall().once();
+
     upgradeCatalog.updateConfigurationProperties("hive-site",
             Collections.singletonMap("hive.server2.transport.mode", "binary"), false, false);
     expectLastCall();
