diff --git a/old_product.java b/new_product.java
index b8cf596..4453b5d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery notEqProp(String propName, String otherProp) {
-        criterion = criterion.and(criterionBuilder.notEqProp(propName, otherProp));
+        criterion = criterion.and(Criteria.notEqProp(propName, otherProp));
         return this;
     }
\ No newline at end of file
