diff --git a/old_test.java b/new_test.java
index bd84796..a72e867 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,23 +2,23 @@
     public void testValidate() {
         logger.info("validate");
         
-        DatabaseConfiguration dbConf = TestUtils.getDBConfig();
+        Configuration conf = TestUtils.getConfig();
         
         
-        Dataframe[] data = Datasets.heartDiseaseClusters(dbConf);
+        Dataframe[] data = Datasets.heartDiseaseClusters(conf);
         
         Dataframe trainingData = data[0];
         Dataframe validationData = data[1];
         
         
         String dbName = this.getClass().getSimpleName();
-        DummyXYMinMaxNormalizer df = new DummyXYMinMaxNormalizer(dbName, dbConf);
+        DummyXYMinMaxNormalizer df = new DummyXYMinMaxNormalizer(dbName, conf);
         df.fit_transform(trainingData, new DummyXYMinMaxNormalizer.TrainingParameters());
         
         df.transform(validationData);
         
         
-        Kmeans instance = new Kmeans(dbName, dbConf);
+        Kmeans instance = new Kmeans(dbName, conf);
         
         Kmeans.TrainingParameters param = new Kmeans.TrainingParameters();
         param.setK(2);
@@ -37,8 +37,8 @@
         //instance = null;
         //df = null;
         
-        df = new DummyXYMinMaxNormalizer(dbName, dbConf);
-        instance = new Kmeans(dbName, dbConf);
+        df = new DummyXYMinMaxNormalizer(dbName, conf);
+        instance = new Kmeans(dbName, conf);
         
         instance.validate(validationData);
         
