diff --git a/old_product.java b/new_product.java
index f9707ee..cf4d294 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,16 @@
 private Item unparsedText(final QueryContext qc, final boolean check) throws QueryException {
     checkCreate(qc);
-    final byte[] path = checkStr(exprs[0], qc);
+    final byte[] path = toToken(exprs[0], qc);
     final IO base = sc.baseIO();
 
     String enc = null;
     try {
       if(base == null) throw STBASEURI.get(info);
-      enc = checkEncoding(1, WHICHENC, qc);
+      enc = toEncoding(1, ENCODING_X, qc);
 
       final String p = string(path);
-      if(p.indexOf('#') != -1) throw FRAGID.get(info, p);
-      if(!Uri.uri(p).isValid()) throw INVURL.get(info, p);
+      if(p.indexOf('#') != -1) throw FRAGID_X.get(info, p);
+      if(!Uri.uri(p).isValid()) throw INVURL_X.get(info, p);
 
       IO io = base.merge(p);
       final String[] rp = qc.resources.texts.get(io.path());
@@ -18,7 +18,7 @@ private Item unparsedText(final QueryContext qc, final boolean check) throws Que
         io = IO.get(rp[0]);
         if(rp.length > 1) enc = rp[1];
       }
-      if(!io.exists()) throw RESNF.get(info, p);
+      if(!io.exists()) throw RESNF_X.get(info, p);
 
       final InputStream is = io.inputStream();
       try {
@@ -36,8 +36,8 @@ private Item unparsedText(final QueryContext qc, final boolean check) throws Que
       if(check) return Bln.FALSE;
       if(ex instanceof InputException) {
         final boolean inv = ex instanceof EncodingException || enc != null;
-        throw (inv ? INVCHARS : WHICHCHARS).get(info, ex);
+        throw (inv ? INVCHARS_X : WHICHCHARS_X).get(info, ex);
       }
-      throw RESNF.get(info, path);
+      throw RESNF_X.get(info, path);
     }
   }
\ No newline at end of file
