diff --git a/old_test.java b/new_test.java
index f4fc7f5..8f01dec 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,5 +2,5 @@
   public void unparsedText() {
     contains(UNPARSED_TEXT.args(TEXT), "&lt;html");
     contains(UNPARSED_TEXT.args(TEXT, "US-ASCII"), "&lt;html");
-    error(UNPARSED_TEXT.args(TEXT, "xyz"), Err.WHICHENC);
+    error(UNPARSED_TEXT.args(TEXT, "xyz"), Err.ENCODING_X);
   }
\ No newline at end of file
