diff --git a/old_product.java b/new_product.java
index 1e38763..a59d801 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public synchronized void release()
     {
         if (this.free == true)
         {
-            throw new RuntimeException("Trying to release an already released Server");
+            throw new IllegalStateException("Trying to release an already released Server");
         }
         this.free = true;
     }
\ No newline at end of file
