diff --git a/old_test.java b/new_test.java
index 62eb281..a575be2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,14 +13,14 @@
     runError("geo:pointN(<gml:MultiLineString><gml:LineString><gml:coordinates>" +
             "1,1 0,0 2,1</gml:coordinates></gml:LineString><gml:LineString>" +
             "<gml:coordinates>2,1 3,3 4,4</gml:coordinates></gml:LineString>" +
-            "</gml:MultiLineString>, 4)", GeoErrors.qname(4));
+            "</gml:MultiLineString>, 4)", GeoErrors.qname(3));
 
     runError("geo:pointN(" +
             "<gml:unknown><gml:coordinates>1,1</gml:coordinates></gml:unknown>,1)",
             GeoErrors.qname(1));
     runError("geo:pointN(<gml:LineString><gml:coordinates>11,10 20,1 20,20 12,13" +
-            "</gml:coordinates></gml:LineString>, 5)", GeoErrors.qname(6));
+            "</gml:coordinates></gml:LineString>, 5)", GeoErrors.qname(4));
     runError("geo:pointN(<gml:LineString><gml:coordinates>11,10 20,1 20,20 12,13" +
-            "</gml:coordinates></gml:LineString>, 0)", GeoErrors.qname(6));
+            "</gml:coordinates></gml:LineString>, 0)", GeoErrors.qname(4));
     runError("geo:pointN(text {'a'},3)", FUNCMP.qname());
   }
\ No newline at end of file
