diff --git a/old_product.java b/new_product.java
index 93d3120..6b8efb1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public Value pointN(final ANode node, final Int number) throws QueryException {
     final Geometry geo = geo(node, Q_GML_LINEARRING, Q_GML_LINESTRING);
     if(geo == null && checkGeo(node) != null)
-      throw GeoErrors.lineNeeded(node.qname().local());
+      throw GeoErrors.geoType(node.qname().local(), "Line");
 
     final int max = geo.getNumPoints();
     final long n = number.itr();
