diff --git a/old_product.java b/new_product.java
index 901fa7d..78fec97 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,3 @@
 public static <O> List<O> remove(Collection<O> objectCollection,O removeElement){
-        Collection<O> remove = new ArrayList<O>();
-        remove.add(removeElement);
-        return removeAll(objectCollection, remove);
+        return removeAll(objectCollection, ConvertUtil.toList(removeElement));
     }
\ No newline at end of file
