diff --git a/old_test.java b/new_test.java
index 7ab3a9a..b35340f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,11 +2,6 @@
     public void testRemove(){
         List<String> list = new ArrayList<String>(){
 
-            /**
-             * 
-             */
-            private static final long serialVersionUID = -9002323146501447769L;
-
             {
                 add("xinge");
                 add("feilong1");
@@ -15,6 +10,7 @@
             }
         };
 
-        LOGGER.info("list:{}", JsonUtil.format(CollectionsUtil.remove(list, "feilong2")));
-        LOGGER.info("list:{}", JsonUtil.format(list));
+        List<String> removeList = CollectionsUtil.remove(list, "feilong2");
+        assertArrayEquals(ConvertUtil.toArray("xinge", "feilong1"), ConvertUtil.toArray(removeList, String.class));
+        assertArrayEquals(ConvertUtil.toArray("xinge", "feilong1", "feilong2", "feilong2"), ConvertUtil.toArray(list, String.class));
     }
\ No newline at end of file
