diff --git a/old_test.java b/new_test.java
index 2a549de..69f5541 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,11 @@
 @Test
     public void testGetSymbolEmissionProbabilities() {
         System.out.println("getSymbolEmissionProbabilities");
-        HMM hmm = new HMM(pi, a, b);
-        double[][] expResult = b;
-        double[][] result = hmm.getSymbolEmissionProbabilities();
-        for (int i = 0; i < expResult.length; i++) {
-            for (int j = 0; j < expResult[i].length; j++) {
-                assertEquals(expResult[i][j], result[i][j], 1E-7);
+        HMM hmm = new HMM(pi, Matrix.of(a), Matrix.of(b));
+        DenseMatrix result = hmm.getSymbolEmissionProbabilities();
+        for (int i = 0; i < b.length; i++) {
+            for (int j = 0; j < b[i].length; j++) {
+                assertEquals(b[i][j], result.get(i, j), 1E-7);
             }
         }
     }
\ No newline at end of file
