diff --git a/old_test.java b/new_test.java
index 3590e4d..f1b64dc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -54,7 +54,12 @@
         BiConsumer<Long, Long> verifyReadResult = (startOffset, endOffset) -> {
             int readLength = (int) (endOffset - startOffset);
             while (readLength > 0) {
-                InputStream actualDataStream = context.readIndex.readDirect(segmentId, startOffset, readLength);
+                InputStream actualDataStream;
+                try {
+                    actualDataStream = context.readIndex.readDirect(segmentId, startOffset, readLength);
+                } catch (StreamSegmentNotExistsException ex) {
+                    throw new CompletionException(ex);
+                }
                 Assert.assertNotNull(
                         String.format("Unexpected result when data is readily available for Offset = %s, Length = %s.", startOffset, readLength),
                         actualDataStream);
