diff --git a/old_product.java b/new_product.java
index a1053be..f11596a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,8 @@
 public int pin(TachyonURI path) {
     try {
       TachyonFile fd = mTfs.open(path);
-      mTfs.setPin(fd, true);
+      SetStateOptions options = new SetStateOptions.Builder(mTachyonConf).setPinned(true).build();
+      mTfs.setState(fd, options);
       System.out.println("File '" + path + "' was successfully pinned.");
       return 0;
     } catch (Exception e) {
