diff --git a/old_test.java b/new_test.java
index 257459b..5f1a47b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-  public void PinTest() throws Exception {
+  public void pinTest() throws Exception {
     mTfs.mkdirs(new TachyonURI("/myFolder"));
     TachyonFile folder = mTfs.open(new TachyonURI("/myFolder"));
     mTfs.setPin(folder, true);
@@ -19,7 +19,7 @@
 
     mLocalTachyonCluster.stopTFS();
 
-    PinTestUtil(folderInfo, file0Info, file1Info);
+    pinTestUtil(folderInfo, file0Info, file1Info);
     deleteFsMasterJournalLogs();
-    PinTestUtil(folderInfo, file0Info, file1Info);
+    pinTestUtil(folderInfo, file0Info, file1Info);
   }
\ No newline at end of file
