diff --git a/old_test.java b/new_test.java
index 2292e04..d946f1a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
         Workbook wb = _testDataProvider.createWorkbook();
         Sheet sheet = wb.createSheet();
         
-        Map<Integer, CellRangeAddress> mergedRegions = new HashMap<Integer, CellRangeAddress>();
+        Map<Integer, CellRangeAddress> mergedRegions = new HashMap<>();
         for (int r=0; r<10; r++) {
             CellRangeAddress region = new CellRangeAddress(r, r, 0, 1);
             mergedRegions.put(r, region);
