diff --git a/old_product.java b/new_product.java
index f3b9036..3db7ca2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @Override
     public void removeMergedRegions(Collection<Integer> indices) {
-        for (int i : (new TreeSet<Integer>(indices)).descendingSet()) {
+        for (int i : (new TreeSet<>(indices)).descendingSet()) {
             _sheet.removeMergedRegion(i);
         }
     }
\ No newline at end of file
