diff --git a/old_product.java b/new_product.java
index 3de565d..4740e13 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public static Map<String, String> readAllPropertiesToMap(String baseName){
-        return readAllPropertiesToMap(baseName, null);
+public static Map<String, String> readPropertiesToMap(String baseName){
+        return readPropertiesToMap(baseName, null);
     }
\ No newline at end of file
