diff --git a/old_test.java b/new_test.java
index de464de..d111062 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
     public void test_nextOrCurrent_serialization() throws IOException, ClassNotFoundException {
-        DateTimeAdjuster nextOrCurrent = DateTimeAdjusters.nextOrCurrent(SUNDAY);
+        WithAdjuster nextOrCurrent = DateTimeAdjusters.nextOrCurrent(SUNDAY);
         assertTrue(nextOrCurrent instanceof Serializable);
 
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
