diff --git a/old_test.java b/new_test.java
index be54fdf..7a89fcc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,9 +6,11 @@
             {Double.NaN, Double.NaN, Double.NaN},
         };
 
-        ImputerPreprocessor<Integer, Vector> preprocessor = new ImputerPreprocessor<>(
+        Vectorizer<Integer, double[], Integer, Double> vectorizer = new DoubleArrayVectorizer<>(0, 1, 2);
+
+        ImputerPreprocessor<Integer,  double[]> preprocessor = new ImputerPreprocessor<>(
             VectorUtils.of(1.1, 10.1, 100.1),
-            (k, v) -> v
+            vectorizer
         );
 
         double[][] postProcessedData = new double[][]{
@@ -18,5 +20,5 @@
         };
 
        for (int i = 0; i < data.length; i++)
-           assertArrayEquals(postProcessedData[i], preprocessor.apply(i, VectorUtils.of(data[i])).asArray(), 1e-8);
+           assertArrayEquals(postProcessedData[i], preprocessor.apply(i, data[i]).features().asArray(), 1e-8);
     }
\ No newline at end of file
