diff --git a/old_product.java b/new_product.java
index 09f17ea..b3458d3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public LabeledVector<L> apply(K key, V value) {
+@Override public LabeledVector<L> apply(K key, V value) {
         L lbl = isLabeled() ? label(labelCoord(key, value), key, value) : zero();
 
         List<C> allCoords = null;
@@ -8,15 +8,15 @@ public LabeledVector<L> apply(K key, V value) {
                 .collect(Collectors.toList());
         }
 
-        int vectorLength = useAllValues ? allCoords.size() : extractionCoordinates.size();
-        A.ensure(vectorLength >= 0, "vectorLength >= 0");
+        int vectorLen = useAllValues ? allCoords.size() : extractionCoordinates.size();
+        A.ensure(vectorLen >= 0, "vectorLength >= 0");
 
         List<C> coordinatesForExtraction = useAllValues ? allCoords : extractionCoordinates;
-        Vector vector = createVector(vectorLength);
+        Vector vector = createVector(vectorLen);
         for (int i = 0; i < coordinatesForExtraction.size(); i++) {
-            Double feature = feature(coordinatesForExtraction.get(i), key, value);
+            Serializable feature = feature(coordinatesForExtraction.get(i), key, value);
             if (feature != null)
-                vector.set(i, feature);
+                vector.setRaw(i, feature);
         }
         return new LabeledVector<>(vector, lbl);
     }
\ No newline at end of file
