diff --git a/old_product.java b/new_product.java
index 2655742..bf6854b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public synchronized FileInfo getFileInfo(final long fileId) throws IOException {
-    return retryRPC(() -> FileInfo
-        .fromThrift(mClient.getFileInfo(fileId, new GetFileInfoTOptions()).getFileInfo()));
+public FileInfo getFileInfo(final long fileId) throws IOException {
+    return retryRPC(() -> GrpcUtils.fromProto(mClient
+        .getFileInfo(GetFileInfoPRequest.newBuilder().setFileId(fileId).build()).getFileInfo()));
   }
\ No newline at end of file
