diff --git a/old_product.java b/new_product.java
index 14fcb06..5247805 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,7 +14,13 @@ void commit(DataFrameBuilder.CommitArgs commitArgs) {
                 synchronized (stateLock) {
                     if (addressSequence <= this.highestCommittedDataFrame) {
                         // Ack came out of order (we already processed one with a higher SeqNo).
-                        log.debug("{}: CommitRejected ({}, HighestCommittedDataFrame = ).", traceObjectId, commitArgs, this.highestCommittedDataFrame);
+                        log.debug("{}: CommitRejected ({}, HighestCommittedDataFrame = {}).", traceObjectId, commitArgs, this.highestCommittedDataFrame);
+                        return;
+                    }
+
+                    if (state() != State.RUNNING) {
+                        // We are shutting down.
+                        log.debug("{}: CommitRejected ({}, Not Running, State = {}).", traceObjectId, commitArgs, state());
                         return;
                     }
 
