diff --git a/old_test.java b/new_test.java
index 129fe7d..1a83afb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
         logger.info("flesch_kincaid_grade_level");
         String strText = TEST_STRING;
         double expResult = 112.3;
-        double result = ReadabilityStatistics.flesch_kincaid_grade_level(strText);
+        double result = ReadabilityStatistics.fleschKincaidGradeLevel(strText);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
