diff --git a/old_product.java b/new_product.java
index 30dd9ff..e90e37d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
 private static void run(final String query, final String result) {
-    query("import module namespace geo='http://expath.org/ns/geo'; " +
-          "declare namespace gml='http://www.opengis.net/gml';" + query, result);
+    final String qu = "import module namespace geo='http://expath.org/ns/geo'; " +
+        "declare namespace gml='http://www.opengis.net/gml';" + query;
+    assertEquals(result, query(qu).replaceAll(Prop.NL + "\\s*", ""));
   }
\ No newline at end of file
