diff --git a/old_test.java b/new_test.java
index 4a71b58..62a3427 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,7 +16,7 @@
         Mockito.verify(jobEntity).setExecution(argument.capture());
         Mockito.verify(jobEntity).setStatus(argument1.capture());
 
-        Assert.assertEquals(hostname, argument.getValue().getHostname());
+        Assert.assertEquals(hostname, argument.getValue().getHostName());
         Assert.assertEquals(pid, argument.getValue().getProcessId());
         Assert.assertThat(argument.getValue().getCheckDelay(), Matchers.is(checkDelay));
         Assert.assertEquals(JOB_1_ID, argument.getValue().getId());
