diff --git a/old_test.java b/new_test.java
index e861418..1681789 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,9 +6,11 @@
 		List<ResidueRangeAndLength> rrs = ResidueRangeAndLength.parseMultiple(ranges, map);
 		int expectedLengthE = rrs.get(0).getLength();
 		Structure structure = cache.getStructureForDomain("d1i3oe_");
-		assertEquals(1, structure.getChains().size());
+		assertEquals(1, structure.getPolyChains().size());
 		Chain e = structure.getChainByPDB("E");
 		assertEquals(expectedLengthE, e.getAtomGroups().size());
-		List<Group> ligandsE = StructureTools.filterLigands(e.getAtomGroups());
+
+		Chain eligands = structure.getNonPolyChainByPDB("E");
+		List<Group> ligandsE = StructureTools.filterLigands(eligands.getAtomGroups());
 		assertEquals(1, ligandsE.size());
 	}
\ No newline at end of file
