diff --git a/old_product.java b/new_product.java
index b80cdbf..a3107b7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 private B64Stream binary(final QueryContext qc) throws QueryException {
-    final byte[] uri = checkStr(exprs[0], qc);
-    return new B64Stream(IO.get(Token.string(uri)), BXFE_IO);
+    final byte[] uri = toToken(exprs[0], qc);
+    return new B64Stream(IO.get(Token.string(uri)), BXFE_IO_X);
   }
\ No newline at end of file
