diff --git a/old_test.java b/new_test.java
index 1d49936..dcfef96 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(expectedExceptions=IndexOutOfBoundsException.class)
     public void test_parse_negativePosition() throws Exception {
         StringLiteralPrinterParser pp = new StringLiteralPrinterParser("hello");
-        pp.parse(context, "hello", -1);
+        pp.parse(parseContext, "hello", -1);
     }
\ No newline at end of file
