diff --git a/old_product.java b/new_product.java
index b342e0a..8ad1110 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,7 +15,6 @@ public int parse(DateTimeParseContext context, String parseText, int position) {
                     return position;
                 }
             }
-            return numberPrinterParser().parse(context, parseText, position);
         } else {
             for (TextStyle textStyle : TextStyle.values()) {
                 TextStore textStore = rule.getTextStore(context.getLocale(), textStyle);
@@ -28,6 +27,6 @@ public int parse(DateTimeParseContext context, String parseText, int position) {
                     }
                 }
             }
-            return numberPrinterParser().parse(context, parseText, position);
         }
+        return numberPrinterParser().parse(context, parseText, position);
     }
\ No newline at end of file
