diff --git a/old_product.java b/new_product.java
index b8bd0ee..7d7c251 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,5 @@
 private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void trainAndValidate(
-            Class<ML> modelerClass, 
             ML.AbstractTrainingParameters modelerTrainingParameters,
-            Class<FS> featureSelectorClass, 
             FS.AbstractTrainingParameters featureSelectorTrainingParameters,
             double expectedF1score) {
         Configuration conf = Configuration.getConfiguration();
@@ -38,6 +36,7 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
 
         TextClassifier instance = MLBuilder.create(trainingParameters, dbName, conf);
         instance.fit(dataset);
+        instance.save();
 
 
         ClassificationMetrics vm = instance.validate(dataset);
@@ -49,7 +48,7 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
         
         
         instance = MLBuilder.load(TextClassifier.class, dbName, conf);
-        Dataframe validationData = null;
+        Dataframe validationData;
         try {
             validationData = instance.predict(this.getClass().getClassLoader().getResource("datasets/sentimentAnalysis.unlabelled.txt").toURI());
         }
