diff --git a/old_test.java b/new_test.java
index ed181da..a702d72 100644
--- a/old_test.java
+++ b/new_test.java
@@ -30,6 +30,7 @@
 
         Modeler instance = MLBuilder.create(trainingParameters, dbName, conf);
         instance.fit(trainingData);
+        instance.save();
 
         instance.close();
 
@@ -42,6 +43,7 @@
         double expResult2 = 0.8;
         assertEquals(expResult2, vm.getMacroF1(), Constants.DOUBLE_ACCURACY_HIGH);
 
+        trainingData.delete();
         instance.close();
         //instance = null;
 
@@ -63,7 +65,6 @@
         assertEquals(expResult, result);
         
         instance.delete();
-        
-        trainingData.delete();
+
         validationData.delete();
     }
\ No newline at end of file
