diff --git a/old_product.java b/new_product.java
index 03397be..4ce28b2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
-private HornetQConnectionFactory createConnectionFactory()
+private ActiveMQConnectionFactory createConnectionFactory()
    {
-      HornetQConnectionFactory cf =
-         HornetQJMSClient.createConnectionFactoryWithoutHA(JMSFactoryType.CF,
-                                                           new TransportConfiguration(NETTY_CONNECTOR_FACTORY));
+      ActiveMQConnectionFactory cf =
+         ActiveMQJMSClient.createConnectionFactoryWithoutHA(JMSFactoryType.CF,
+                                                            new TransportConfiguration(NETTY_CONNECTOR_FACTORY));
 
       connectionFactories.add(cf);
       return cf;
