diff --git a/old_product.java b/new_product.java
index bbcb2e1..6852a7d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -37,14 +37,15 @@
     try {
       ClusterInformation clusterInformation = cluster.buildClusterInformation();
       UpgradeActionOperations upgradeActionOperations = upgradeAction.getOperations(
-          clusterInformation);
+          clusterInformation, upgradeContext.buildUpgradeInformation());
 
       // update configurations
-      changeConfigurations(cluster, upgradeActionOperations.getConfigurationChanges());
+      changeConfigurations(cluster, upgradeActionOperations.getConfigurationChanges(), upgradeContext);
+      removeConfigurationTypes(cluster, upgradeActionOperations.getConfigurationTypeRemovals());
 
       standardOutput = "Successfully executed " + pluginClassName;
       if(null != upgradeActionOperations.getStandardOutput()) {
-        standardOutput = upgradeActionOperations.getStandardOutput().toString();
+        standardOutput = upgradeActionOperations.getStandardOutput();
       }
     } catch (UpgradeActionException exception) {
       LOG.error("Unable to run the upgrade action {}", pluginClassName, exception);
