diff --git a/old_product.java b/new_product.java
index 9952bc7..acc63cd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public Bln isRing(final ANode node) throws QueryException {
     final Geometry geo = geo(node, Q_GML_LINEARRING, Q_GML_LINESTRING);
     if(geo == null && checkGeo(node) != null)
-      throw GeoErrors.lineNeeded(node.qname().local());
+      throw GeoErrors.geoType(node.qname().local(), "Line");
 
     return Bln.get(geo instanceof LineString ?
        ((LineString) geo).isRing() :
