diff --git a/old_test.java b/new_test.java
index 79b188e..07545c7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,10 +9,10 @@
     runError("geo:isRing(<gml:MultiLineString><gml:LineString><gml:coordinates>" +
             "1,1 0,0 2,1</gml:coordinates></gml:LineString><gml:LineString>" +
             "<gml:coordinates>2,1 3,3 4,4</gml:coordinates></gml:LineString>" +
-            "</gml:MultiLineString>)", GeoErrors.qname(4));
+            "</gml:MultiLineString>)", GeoErrors.qname(3));
     runError("geo:isClosed(" +
             "<gml:Point><gml:coordinates>2,3</gml:coordinates></gml:Point>)",
-            GeoErrors.qname(4));
+            GeoErrors.qname(3));
     runError("geo:isRing(" +
             "<gml:LineString><gml:coordinates>1,1</gml:coordinates></gml:LineString>)",
             GeoErrors.qname(2));
