diff --git a/old_product.java b/new_product.java
index b08019c..249c87c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,7 @@
 void query(final String path) throws IOException {
-    final XQuery xq;
     int c = 0;
     while(!(cmds.get(c) instanceof XQuery)) run(cmds.get(c++));
-    xq = (XQuery) cmds.get(c);
+    final XQuery xq = (XQuery) cmds.get(c);
 
     // create query instance
     if(value != null) xq.bind(null, value, NodeType.DOC.toString());
