diff --git a/old_test.java b/new_test.java
index d2bfbe4..8276c87 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public void query() {
     contains(_CLIENT_EXECUTE.args(conn(), new ShowUsers()), Text.USERHEAD[0]);
     query("let $a := " + conn() + ", $b := " + conn() + " return " +
-        _CLIENT_QUERY.args("$a", "1") + "+" + _CLIENT_QUERY.args("$b", "2"), "3");
+        _CLIENT_QUERY.args("$a", "1") + '+' + _CLIENT_QUERY.args("$b", "2"), "3");
     query(_CLIENT_QUERY.args(conn(), "\"$a*2\"", " { 'a':1 }"), "2");
     // query errors
     error(_CLIENT_QUERY.args(conn(), "x"), Err.NOCTX);
