diff --git a/old_test.java b/new_test.java
index 99ca90e..7912e2b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,6 +16,7 @@
 
         // Disabled discovery
         generator.setIncludeDiscovery(false);
+        generator.setIncludeDiscoveryExtension(false);
         extendedMetadata = generator.generateExtendedMetadata();
         assertFalse(extendedMetadata.isIdpDiscoveryEnabled());
         assertNull(extendedMetadata.getIdpDiscoveryURL());
@@ -23,11 +24,11 @@
 
         // Default extended metadata
         ExtendedMetadata defaultMetadata = new ExtendedMetadata();
-        defaultMetadata.setIdpDiscoveryEnabled(false);
-        defaultMetadata.setIdpDiscoveryResponseURL("http://testDisco.com/response");
         defaultMetadata.setRequireLogoutResponseSigned(true);
         generator.setExtendedMetadata(defaultMetadata);
         generator.setIncludeDiscovery(true);
+        generator.setIncludeDiscoveryExtension(true);
+        generator.setCustomDiscoveryResponseURL("http://testDisco.com/response");
         extendedMetadata = generator.generateExtendedMetadata();
         assertTrue(extendedMetadata.isIdpDiscoveryEnabled());
         assertEquals("http://localhost:8080/saml/discovery/alias/testAlias", extendedMetadata.getIdpDiscoveryURL());
