diff --git a/old_product.java b/new_product.java
index 8ef36a1..c1c4a55 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,19 +8,14 @@ public ExtendedMetadata generateExtendedMetadata() {
             metadata = new ExtendedMetadata();
         }
 
+        String entityBaseURL = getEntityBaseURL();
+        String entityAlias = getEntityAlias();
+
         metadata.setIdpDiscoveryEnabled(isIncludeDiscovery());
 
         if (isIncludeDiscovery()) {
-            if (metadata.getIdpDiscoveryURL() == null) {
-                metadata.setIdpDiscoveryURL(getServerURL(entityBaseURL, entityAlias, getSAMLDiscoveryPath()));
-            }
-            if (metadata.getIdpDiscoveryResponseURL() == null) {
-                if (customDiscoveryURL != null) {
-                    metadata.setIdpDiscoveryURL(customDiscoveryURL);
-                } else {
-                    metadata.setIdpDiscoveryResponseURL(getServerURL(entityBaseURL, entityAlias, getSAMLEntryPointPath()) + "?" + SAMLEntryPoint.DISCOVERY_RESPONSE_PARAMETER + "=true");
-                }
-            }
+            metadata.setIdpDiscoveryURL(getDiscoveryURL(entityBaseURL, entityAlias));
+            metadata.setIdpDiscoveryResponseURL(getDiscoveryResponseURL(entityBaseURL, entityAlias));
         } else {
             metadata.setIdpDiscoveryURL(null);
             metadata.setIdpDiscoveryResponseURL(null);
