diff --git a/old_test.java b/new_test.java
index 16b0ae7..cc99eae 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,9 @@
 @Test
 	public void decrypt_shouldDecryptShortAndLongText() {
+		final Decoder base64 = Base64.getDecoder();
 		// use specific IV and Key
-		byte[] initVector = Base64.decode("9wyBUNglFCRVSUhMfsTa3Q==");
-		byte[] secretKey = Base64.decode("dTfyELRrAICGDwzjHDjuhw==");
+		byte[] initVector = base64.decode("9wyBUNglFCRVSUhMfsTa3Q==");
+		byte[] secretKey = base64.decode("dTfyELRrAICGDwzjHDjuhw==");
 		
 		// perform decryption
 		String expected = "this is fantasmic";
