diff --git a/old_product.java b/new_product.java
index cbf4692..7b0c87b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ public static String decrypt(String text, byte[] initVector, byte[] secretKey) {
 		try {
 			Cipher cipher = Cipher.getInstance(OpenmrsConstants.ENCRYPTION_CIPHER_CONFIGURATION);
 			cipher.init(Cipher.DECRYPT_MODE, secret, initVectorSpec);
-			byte[] original = cipher.doFinal(Base64.decode(text));
+			byte[] original = cipher.doFinal(Base64.getDecoder().decode(text));
 			decrypted = new String(original, encoding);
 		}
 		catch (GeneralSecurityException e) {
