diff --git a/old_test.java b/new_test.java
index 51933d4..20a28d8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,8 +7,7 @@
 				.columns("value1", "value2")
 				.values(System.currentTimeMillis(), 5)
 				.build();
-		Serie[] series = new Serie[] { serie };
-		this.influxDB.write(dbName, series, TimeUnit.MILLISECONDS);
+		this.influxDB.write(dbName, TimeUnit.MILLISECONDS, serie);
 
 		this.influxDB.deleteDatabase(dbName);
 	}
\ No newline at end of file
