diff --git a/old_product.java b/new_product.java
index e63c4de..e74e097 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,13 @@
 public Collection<TopLevelItem> getAllItems() {
 
-        final Collection<TopLevelItem> items = new LinkedHashSet<TopLevelItem>(
-                getItems()
-        );
-
         if (this instanceof ViewGroup) {
+            final Collection<TopLevelItem> items = new LinkedHashSet<TopLevelItem>(getItems());
 
-            for(final View view: ((ViewGroup) this).getViews()) {
-
+            for(View view: ((ViewGroup) this).getViews()) {
                 items.addAll(view.getAllItems());
             }
+            return Collections.unmodifiableCollection(items);
+        } else {
+            return getItems();
         }
-
-        return Collections.unmodifiableCollection(items);
     }
\ No newline at end of file
