diff --git a/old_test.java b/new_test.java
index 63801f9..42b1a3a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testGet() {
         // Given
-        Assert.assertTrue(ff4j.getStore() instanceof InMemoryFeatureStore);
+        Assert.assertTrue(ff4j.getFeatureStore() instanceof InMemoryFeatureStore);
         // When
         ClientResponse resHttp = resourceStore().get(ClientResponse.class);
         String resEntity = resHttp.getEntity(String.class);
