diff --git a/old_product.java b/new_product.java
index 80a1c79..5080910 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,5 +6,5 @@
     @ApiResponses(@ApiResponse(code = 200, message= "status of current ff4j bean"))
     @Produces(MediaType.APPLICATION_JSON)
     public FeatureStoreApiBean get() {
-        return new FeatureStoreApiBean(ff4j.getStore());
+        return new FeatureStoreApiBean(ff4j.getFeatureStore());
     }
\ No newline at end of file
