diff --git a/old_product.java b/new_product.java
index acb9229..16bdcd5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,9 @@
 @Override
-    public List<byte[]> geoHash(byte[] key, byte[]... members) {
-        List<byte[]> result = delegate.geoHash(key, members);
-        if (isFutureConversion()){
-            addResultConverter(identityConverter);
-        }
-        return result;
-    }
\ No newline at end of file
+	public List<String> geoHash(byte[] key, byte[]... members) {
+
+		List<String> result = delegate.geoHash(key, members);
+		if (isFutureConversion()) {
+			addResultConverter(identityConverter);
+		}
+		return result;
+	}
\ No newline at end of file
