diff --git a/old_test.java b/new_test.java
index 384e0c9..00b06be 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,17 +1,18 @@
 @Test
-    public void testGeoHash() throws Exception {
-        K key = keyFactory.instance();
-        M v1 = valueFactory.instance();
-        M v2 = valueFactory.instance();
+	public void testGeoHash() {
 
-        geoOperations.geoAdd(key, 13.361389, 38.115556, v1);
-        geoOperations.geoAdd(key, 15.087269, 37.502669, v2);
+		K key = keyFactory.instance();
+		M v1 = valueFactory.instance();
+		M v2 = valueFactory.instance();
 
-        List<byte[]> result = geoOperations.geoHash(key, v1, v2);
-        assertEquals(result.size(), 2);
+		geoOperations.geoAdd(key, POINT_PALERMO, v1);
+		geoOperations.geoAdd(key, POINT_CATANIA, v2);
 
-        final RedisSerializer<String> serializer = new StringRedisSerializer();
+		List<String> result = geoOperations.geoHash(key, v1, v2);
+		assertThat(result, hasSize(2));
 
-        assertEquals(serializer.deserialize(result.get(0)), "sqc8b49rny0");
-        assertEquals(serializer.deserialize(result.get(1)), "sqdtr74hyu0");
-    }
\ No newline at end of file
+		final RedisSerializer<String> serializer = new StringRedisSerializer();
+
+		assertThat(result.get(0), is(equalTo("sqc8b49rny0")));
+		assertThat(result.get(1), is(equalTo("sqdtr74hyu0")));
+	}
\ No newline at end of file
