diff --git a/old_product.java b/new_product.java
index 6458ead..d6ce1fa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public synchronized void delete(final Id.Program programId, final SchedulableProgramType programType,
+public synchronized void delete(final ProgramId programId, final SchedulableProgramType programType,
                                   final String scheduleName) throws InterruptedException, TransactionFailureException {
     factory.createExecutor(ImmutableList.of((TransactionAware) table))
       .execute(new TransactionExecutor.Subroutine() {
