diff --git a/old_test.java b/new_test.java
index 6094c85..b5fc7f7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,11 @@
 @Test
-    public void testCreateUser() throws Exception {
-        UserDto user = createUser(KaaAuthorityDto.TENANT_DEVELOPER);
-        Assert.assertFalse(strIsEmpty(user.getId()));
-    }
\ No newline at end of file
+  public void testTenantAdminCreateOtherUser() throws Exception {
+    loginTenantAdmin(tenantAdminDto.getUsername());
+    UserDto user = createUser(KaaAuthorityDto.TENANT_DEVELOPER);
+
+    UserDto storedUser = client.getUser(user.getId());
+    Assert.assertEquals(storedUser.getAuthority(), KaaAuthorityDto.TENANT_DEVELOPER);
+    Assert.assertNotNull(storedUser);
+    assertUsersEquals(user, storedUser);
+    Assert.assertEquals(tenantAdminDto.getTenantId(), storedUser.getTenantId());
+  }
\ No newline at end of file
