diff --git a/old_product.java b/new_product.java
index e25083f..e260bd9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,15 +1,16 @@
-protected UserDto createUser(TenantUserDto tenant, KaaAuthorityDto authority) throws Exception {
-        UserDto user = new UserDto();
-        String username = generateString(USERNAME);
-        user.setUsername(username);
-        user.setMail(username + "@demoproject.org");
-        user.setFirstName(generateString("Test"));
-        user.setLastName(generateString("User"));
-        user.setAuthority(authority);
-        if (tenant == null) {
-            tenant = createTenant();
-        }
-        loginTenantAdmin(tenant.getUsername());
-        UserDto savedUser = client.editUser(user);
-        return savedUser;
-    }
\ No newline at end of file
+protected UserDto createUser(org.kaaproject.kaa.common.dto.admin.UserDto tenant, KaaAuthorityDto authority) throws Exception {
+    UserDto user = new UserDto();
+    String username = generateString(USERNAME);
+    user.setUsername(username);
+    user.setMail(username + "@demoproject.org");
+    user.setFirstName(generateString("Test"));
+    user.setLastName(generateString("User"));
+    user.setAuthority(authority);
+    if (tenant == null) {
+      tenant = createTenantAdmin();
+    }
+    user.setTenantId(tenantAdminDto.getTenantId());
+    loginTenantAdmin(tenant.getUsername());
+    UserDto savedUser = client.editUser(user);
+    return savedUser;
+  }
\ No newline at end of file
