diff --git a/old_product.java b/new_product.java
index 8d40036..8d4fe48 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
-  public int read(String table, String key, Set<String> fields,
+  public Status read(String table, String key, Set<String> fields,
       HashMap<String, ByteIterator> result) {
 
     try {
@@ -41,12 +41,12 @@
 
       }
 
-      return StatusCode.OK;
+      return Status.OK;
 
     } catch (Exception e) {
       e.printStackTrace();
       System.out.println("Error reading key: " + key);
-      return StatusCode.ERROR;
+      return Status.ERROR;
     }
 
   }
\ No newline at end of file
