diff --git a/old_test.java b/new_test.java
index 84b22b1..c5703e7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,6 @@
         System.out.println("read");
         Set<String> fields = MOCK_DATA.keySet();
         HashMap<String, ByteIterator> resultParam = new HashMap<String, ByteIterator>(10);
-        int expResult = 0;
-        int result = instance.read(MOCK_TABLE, MOCK_KEY1, fields, resultParam);
-        assertEquals(expResult, result);
+        Status result = instance.read(MOCK_TABLE, MOCK_KEY1, fields, resultParam);
+        assertEquals(Status.OK, result);
     }
\ No newline at end of file
