diff --git a/old_test.java b/new_test.java
index 674ca76..e0f95c8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testProcessOSPFMessage() throws Exception {
+    public void testProcessOspfMessage() throws Exception {
         ospfInterface.setIpAddress(Ip4Address.valueOf("11.11.11.11"));
         ospfInterface.setIpNetworkMask(Ip4Address.valueOf("225.225.225.225"));
         ospfInterface.setInterfaceType(2);
@@ -27,7 +27,7 @@
         buf = ChannelBuffers.copiedBuffer(checkArray);
         helloPacket.setChecksum(buf.readUnsignedShort());
         message = helloPacket;
-        ospfInterfaceChannelHandler.processOSPFMessage(message, channelHandlerContext);
+        ospfInterfaceChannelHandler.processOspfMessage(message, channelHandlerContext);
         ddPacket = new DdPacket();
         ddPacket.setSourceIp(Ip4Address.valueOf("1.1.1.1"));
         ddPacket.setRouterId(Ip4Address.valueOf("10.10.10.10"));
@@ -46,7 +46,7 @@
         buf = ChannelBuffers.copiedBuffer(checkArray);
         ddPacket.setChecksum(buf.readUnsignedShort());
         message = ddPacket;
-        ospfInterfaceChannelHandler.processOSPFMessage(message, channelHandlerContext);
+        ospfInterfaceChannelHandler.processOspfMessage(message, channelHandlerContext);
         lsRequest = new LsRequest();
         lsRequest.setSourceIp(Ip4Address.valueOf("1.1.1.1"));
         lsRequest.setRouterId(Ip4Address.valueOf("10.10.10.10"));
@@ -64,7 +64,7 @@
         buf = ChannelBuffers.copiedBuffer(checkArray);
         lsRequest.setChecksum(buf.readUnsignedShort());
         message = lsRequest;
-        ospfInterfaceChannelHandler.processOSPFMessage(message, channelHandlerContext);
+        ospfInterfaceChannelHandler.processOspfMessage(message, channelHandlerContext);
         lsUpdate = new LsUpdate();
         lsUpdate.setSourceIp(Ip4Address.valueOf("1.1.1.1"));
         lsUpdate.setRouterId(Ip4Address.valueOf("10.10.10.10"));
@@ -82,7 +82,7 @@
         buf = ChannelBuffers.copiedBuffer(checkArray);
         lsUpdate.setChecksum(buf.readUnsignedShort());
         message = lsUpdate;
-        ospfInterfaceChannelHandler.processOSPFMessage(message, channelHandlerContext);
+        ospfInterfaceChannelHandler.processOspfMessage(message, channelHandlerContext);
         lsAck = new LsAcknowledge();
         lsAck.setSourceIp(Ip4Address.valueOf("1.1.1.1"));
         lsAck.setRouterId(Ip4Address.valueOf("10.10.10.10"));
@@ -100,7 +100,7 @@
         buf = ChannelBuffers.copiedBuffer(checkArray);
         lsAck.setChecksum(buf.readUnsignedShort());
         message = lsAck;
-        ospfInterfaceChannelHandler.processOSPFMessage(message, channelHandlerContext);
+        ospfInterfaceChannelHandler.processOspfMessage(message, channelHandlerContext);
         assertThat(ospfInterfaceChannelHandler, is(notNullValue()));
 
     }
\ No newline at end of file
