diff --git a/old_product.java b/new_product.java
index 0dbc8fa..558b750 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public Map<String, Properties> subProperties(Properties prop, String regex) {
+public static Map<String, Properties> subProperties(Properties prop, String regex) {
     Map<String, Properties> subProperties = new HashMap<>();
     Pattern pattern = Pattern.compile(regex);
 
