diff --git a/old_test.java b/new_test.java
index d189a44..2ae4ff7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,5 +13,5 @@
     assertEquals(
         PathUtils.concatPath("/tmp", ".tmp_blocks", sessionId % 1024,
             String.format("%x-%x", sessionId, blockId)),
-        mBlockWorker.createBlock(sessionId, blockId, tierAlias, initialBytes));
+        mBlockWorker.createBlock(sessionId, blockId, tierAlias, "", initialBytes));
   }
\ No newline at end of file
