diff --git a/old_product.java b/new_product.java
index 15e0f33..726efc5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,3 @@
-String createBlock(long sessionId, long blockId, String tierAlias, long initialBytes)
+String createBlock(long sessionId, long blockId, String tierAlias,
+      String medium, long initialBytes)
       throws BlockAlreadyExistsException, WorkerOutOfSpaceException, IOException;
\ No newline at end of file
