diff --git a/old_test.java b/new_test.java
index ad4b373..37452ae 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     {
         mv.create(context);
         int fieldId = mv.getFieldId();
-        Collection found = MetadataValue.findByField(context, fieldId);
+        List<MetadataValue> found = MetadataValue.findByField(context, fieldId);
         assertThat("testFind 0",found, notNullValue());
         assertTrue("testFind 1",found.size() >= 1);        
     }
\ No newline at end of file
