diff --git a/old_product.java b/new_product.java
index e73d026..55c3f4b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static java.util.Collection findByField(Context context, int fieldId)
+public static List<MetadataValue> findByField(Context context, int fieldId)
             throws IOException, SQLException, AuthorizeException
     {
         // Grab rows from DB
@@ -7,7 +7,7 @@ public static java.util.Collection findByField(Context context, int fieldId)
                 fieldId);
 
         TableRow row = null;
-        java.util.Collection ret = new ArrayList();
+        List<MetadataValue> ret = new ArrayList<MetadataValue>();
         try
         {
             while (tri.hasNext())
