diff --git a/old_product.java b/new_product.java
index 6bd40a5..edb78d1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,12 @@
 @RequiresSession
-  public HostVersionEntity findByClusterStackVersionAndHost(String clusterName, String stack, String version, String hostName) {
+  public HostVersionEntity findByClusterStackVersionAndHost(String clusterName,
+      StackId stackId, String version, String hostName) {
+
     final TypedQuery<HostVersionEntity> query = entityManagerProvider.get()
         .createNamedQuery("hostVersionByClusterStackVersionAndHostname", HostVersionEntity.class);
     query.setParameter("clusterName", clusterName);
-    query.setParameter("stack", stack);
+    query.setParameter("stackName", stackId.getStackName());
+    query.setParameter("stackVersion", stackId.getStackVersion());
     query.setParameter("version", version);
     query.setParameter("hostName", hostName);
 
