diff --git a/old_product.java b/new_product.java
index cec3790..8a58377 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,14 @@
-protected Query parseQuery(String searchString) throws ParseException {
+public synchronized Query parseQuery(String searchString) throws ParseException, IndexException {
         if (searchString == null || searchString.trim().isEmpty()) {
             throw new ParseException("Query is null or empty");
         }
         LOGGER.debug(searchString);
+
         final Query query = queryParser.parse(searchString);
+        try {
+            resetAnalyzers();
+        } catch (IOException ex) {
+            throw new IndexException("Unable to reset the analyzer after parsing", ex);
+        }
         return query;
     }
\ No newline at end of file
