diff --git a/old_test.java b/new_test.java
index 35d9712..f410883 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,13 @@
 @Test
     public void testParse1() throws Exception {
-        String headerValue = "custno = 12345; comment=test; version=1," +
+        final String headerValue = "custno = 12345; comment=test; version=1," +
             " name=John; version=1; max-age=600; secure; domain=.apache.org";
 
-        Header header = new BasicHeader("set-cookie", headerValue);
+        final Header header = new BasicHeader("set-cookie", headerValue);
 
-        CookieSpec cookiespec = new BrowserCompatSpec();
-        CookieOrigin origin = new CookieOrigin("www.apache.org", 80, "/", false);
-        List<Cookie> cookies = cookiespec.parse(header, origin);
+        final CookieSpec cookiespec = new BrowserCompatSpec();
+        final CookieOrigin origin = new CookieOrigin("www.apache.org", 80, "/", false);
+        final List<Cookie> cookies = cookiespec.parse(header, origin);
         for (int i = 0; i < cookies.size(); i++) {
             cookiespec.validate(cookies.get(i), origin);
         }
