diff --git a/old_product.java b/new_product.java
index 8527fe9..0b31d98 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 public static List <NameValuePair> parse (final URI uri, final String encoding) {
         final String query = uri.getRawQuery();
         if (query != null && query.length() > 0) {
-            List<NameValuePair> result = new ArrayList<NameValuePair>();
-            Scanner scanner = new Scanner(query);
+            final List<NameValuePair> result = new ArrayList<NameValuePair>();
+            final Scanner scanner = new Scanner(query);
             parse(result, scanner, encoding);
             return result;
         } else {
