diff --git a/old_product.java b/new_product.java
index 0f1bdb9..cf48eab 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
-public static HttpResponse sendCoin(String httpNode, byte[] fromAddress, byte[] toAddress, Long amount,String fromKey) {
+public static HttpResponse sendCoin(String httpNode, byte[] fromAddress, byte[] toAddress, 
+      Long amount, String fromKey) {
     try {
-      String requestUrl = "http://" + httpNode + "/wallet/createtransaction";
+      final String requestUrl = "http://" + httpNode + "/wallet/createtransaction";
       JsonObject userBaseObj2 = new JsonObject();
       userBaseObj2.addProperty("to_address", ByteArray.toHexString(toAddress));
       userBaseObj2.addProperty("owner_address", ByteArray.toHexString(fromAddress));
@@ -8,7 +9,6 @@ public static HttpResponse sendCoin(String httpNode, byte[] fromAddress, byte[]
       response = createConnect(requestUrl, userBaseObj2);
       transactionString = EntityUtils.toString(response.getEntity());
       transactionSignString = gettransactionsign(httpNode,transactionString,fromKey);
-      logger.info("transactionSignString is " + transactionSignString);
       response = broadcastTransaction(httpNode,transactionSignString);
     } catch (Exception e) {
       e.printStackTrace();
