diff --git a/old_test.java b/new_test.java
index 5cd8844..f81da4e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,7 @@
 @Test(enabled = true, description = "SendCoin by http")
   public void test1SendCoin() {
     response = HttpMethed.sendCoin(httpnode,fromAddress,receiverAddress,amount,testKey002);
-    Assert.assertEquals(response.getStatusLine().getStatusCode(),200);
-    responseContent = HttpMethed.parseResponseContent(response);
-    HttpMethed.printJsonContent(responseContent);
-    Assert.assertTrue(Boolean.valueOf(responseContent.get("result").toString()).booleanValue());
+    Assert.assertTrue(HttpMethed.verificationResult(response));
     Assert.assertEquals(HttpMethed.getBalance(httpnode,receiverAddress),amount);
 
   }
\ No newline at end of file
