diff --git a/old_test.java b/new_test.java
index 3a06a38..4f6a3d2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,15 +2,15 @@
   public void testRunInBackground() throws Exception {
 
     // mock the RunnableAction and how it's created by the resource
-    final ActionResource.RunnableAction runnableAction = Mockito.spy( ActionResource.RunnableAction.class );
-    Mockito.doReturn( runnableAction ).when( resourceMock ).createRunnable( actionId, actionClassName, actionUser,
+    final ActionResource.CallableAction runnableAction = Mockito.spy( ActionResource.CallableAction.class );
+    Mockito.doReturn( runnableAction ).when( resourceMock ).createCallable( actionId, actionClassName, actionUser,
       actionParams );
 
     // call the runInBackground methos
-    resourceMock.runInBackground( actionId, actionClassName, actionUser, actionParams );
+    resourceMock.invokeAction( ActionUtil.INVOKER_DEFAULT_ASYNC_EXEC_VALUE, actionId, actionClassName, actionUser, actionParams );
 
     // verify that the createRunnable method was called with the expected parameters
-    Mockito.verify( resourceMock, Mockito.times( 1 ) ).createRunnable( actionId, actionClassName, actionUser,
+    Mockito.verify( resourceMock, Mockito.times( 1 ) ).createCallable( actionId, actionClassName, actionUser,
       actionParams );
 
     // verity that the executor submit method was called to execute the expected RunnableAction
