diff --git a/old_product.java b/new_product.java
index 32ac2fe..ee765fc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
-@Authorized(PrivilegeConstants.EDIT_ORDERS)
-	public Order discontinueOrder(Order order, String reason, User user, Date discontinueDate) throws APIException;
\ No newline at end of file
+@Transactional(readOnly = true)
+	public OrderType getOrderTypeByUuid(String uuid) throws APIException;
\ No newline at end of file
