diff --git a/old_test.java b/new_test.java
index ab8aa69..c734a19 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,8 +18,7 @@
 		Assert.assertTrue(testEncounter.isVoided());
 		
 		//void one of the unvoided orders for testing purposes
-		//TODO To be fixed in order entry sprint
-		/*Order testOrder = os.getOrdersByPatient(patient).get(0);
+		Order testOrder = os.getOrdersByPatient(patient).get(0);
 		Assert.assertNotNull(testOrder);
 		Assert.assertNull(testOrder.getDateVoided());
 		Assert.assertNull(testOrder.getVoidedBy());
@@ -46,6 +45,6 @@
 		Assert.assertTrue(testOrder.isVoided());
 		Assert.assertNotNull(testOrder.getDateVoided());
 		Assert.assertNotNull(testOrder.getVoidedBy());
-		Assert.assertNotNull(testOrder.getVoidReason());*/
-
+		Assert.assertNotNull(testOrder.getVoidReason());
+		
 	}
\ No newline at end of file
