diff --git a/old_test.java b/new_test.java
index 4d0e6f0..ed0f7a5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,7 +13,7 @@
 
         // Get a non-existent stream
         when(mockControllerService.getStream(scope1, stream2)).thenReturn(CompletableFuture.supplyAsync(() -> {
-            throw new DataNotFoundException("Stream Not Found");
+            throw StoreException.create(StoreException.Type.DATA_NOT_FOUND, stream2);
         }));
         response = client.target(resourceURI2).request().buildGet().invoke();
         assertEquals("Get Stream Config Status", 404, response.getStatus());
