diff --git a/old_product.java b/new_product.java
index a783f39..7f15cc5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,8 +8,8 @@
                         .entity(ModelHelper.encodeStreamResponse(streamConfig))
                         .build())
                 .exceptionally(exception -> {
-                    if (exception.getCause() instanceof DataNotFoundException
-                            || exception instanceof DataNotFoundException) {
+                    if (exception.getCause() instanceof StoreException.DataNotFoundException
+                            || exception instanceof StoreException.DataNotFoundException) {
                         log.warn("Stream: {}/{} not found", scopeName, streamName);
                         return Response.status(Status.NOT_FOUND).build();
                     } else {
