diff --git a/old_test.java b/new_test.java
index 9cd43f3..4e03ab9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,13 +2,13 @@
     @SuppressWarnings("ObjectEqualsNull")
     public void testEquals() throws SQLException, AuthorizeException
     {
-        new NonStrictExpectations(AuthorizeManager.class)
+        new NonStrictExpectations(authorizeService.getClass())
         {{
             // Allow full Admin perms (just to create top-level community)
-            AuthorizeManager.isAdmin((Context) any); result = true;
+            authorizeService.isAdmin((Context) any); result = true;
         }};
 
         assertFalse("testEquals 0",c.equals(null));
-        assertFalse("testEquals 1",c.equals(Community.create(null, context)));
+        assertFalse("testEquals 1",c.equals(communityService.create(null, context)));
         assertTrue("testEquals 2", c.equals(c));
     }
\ No newline at end of file
