diff --git a/old_product.java b/new_product.java
index b2db3f2..5cb348d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,42 +1,35 @@
 @Override
     public boolean equals(Object obj)
     {
-        try
+        if (obj == null)
         {
-            if (obj == null)
-            {
-                return false;
-            }
-            if (getClass() != obj.getClass())
-            {
-                return false;
-            }
-            final ResourcePolicy other = (ResourcePolicy) obj;         
-            if (this.getAction() != other.getAction())
-            {
-                return false;
-            }
-            if (this.getEPerson() != other.getEPerson() && (this.getEPerson() == null || !this.getEPerson().equals(other.getEPerson())))
-            {
-                return false;
-            }
-            if (this.getGroup() != other.getGroup() && (this.getGroup() == null || !this.getGroup().equals(other.getGroup())))
-            {
-                return false;
-            }
-            if (this.getStartDate() != other.getStartDate() && (this.getStartDate() == null || !this.getStartDate().equals(other.getStartDate())))
-            {
-                return false;
-            }
-            if (this.getEndDate() != other.getEndDate() && (this.getEndDate() == null || !this.getEndDate().equals(other.getEndDate())))
-            {
-                return false;
-            }    
-            return true;
+            return false;
         }
-        catch (SQLException ex)
+        Class<?> objClass = HibernateProxyHelper.getClassWithoutInitializingProxy(obj);
+        if (getClass() != objClass)
         {
-            log.error("Error while comparing ResourcePolicy objects", ex);
+            return false;
         }
-        return false;
+        final ResourcePolicy other = (ResourcePolicy) obj;
+        if (getAction() != other.getAction())
+        {
+            return false;
+        }
+        if (!ObjectUtils.equals(getEPerson(), other.getEPerson()))
+        {
+            return false;
+        }
+        if (!ObjectUtils.equals(getGroup(), other.getGroup()))
+        {
+            return false;
+        }
+        if (!ObjectUtils.equals(getStartDate(), other.getStartDate()))
+        {
+            return false;
+        }
+        if (!ObjectUtils.equals(getEndDate(), other.getEndDate()))
+        {
+            return false;
+        }
+        return true;
     }
\ No newline at end of file
