diff --git a/old_product.java b/new_product.java
index 6393392..4ccce7e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,12 +4,12 @@
             byte[] plaintext;
             if (useExplicitIv) {
                 decryptIv = new IvParameterSpec(Arrays.copyOf(data, decryptCipher.getBlockSize()));
-                if (tlsContext.getConnectionEnd().getConnectionEndType() == ConnectionEndType.CLIENT) {
-                    decryptCipher.init(Cipher.DECRYPT_MODE,
-                            new SecretKeySpec(keySet.getServerWriteKey(), bulkCipherAlg.getJavaName()), decryptIv);
+                if (context.getConnectionEnd().getConnectionEndType() == ConnectionEndType.CLIENT) {
+                    decryptCipher.init(Cipher.DECRYPT_MODE, new SecretKeySpec(getKeySet().getServerWriteKey(),
+                            bulkCipherAlg.getJavaName()), decryptIv);
                 } else {
-                    decryptCipher.init(Cipher.DECRYPT_MODE,
-                            new SecretKeySpec(keySet.getClientWriteKey(), bulkCipherAlg.getJavaName()), decryptIv);
+                    decryptCipher.init(Cipher.DECRYPT_MODE, new SecretKeySpec(getKeySet().getClientWriteKey(),
+                            bulkCipherAlg.getJavaName()), decryptIv);
                 }
                 plaintext = decryptCipher.doFinal(Arrays.copyOfRange(data, decryptCipher.getBlockSize(), data.length));
             } else {
