diff --git a/old_test.java b/new_test.java
index 1ac03bf..79b4b80 100644
--- a/old_test.java
+++ b/new_test.java
@@ -31,4 +31,6 @@
     query(func + "/annotation/@uri/data()", "http://www.w3.org/2012/xquery");
     query(func + "/return/@type/data()", "xs:integer");
     query(func + "/return/@occurrence/data()", "");
+
+    query(_INSPECT_FUNCTION.args(" %db:f function() {()}") + "/annotation/@name = 'db:f'", "true");
   }
\ No newline at end of file
