diff --git a/old_product.java b/new_product.java
index e62cb15..5f61336 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 private Item function(final QueryContext ctx) throws QueryException {
-    final FItem f = checkFunc(expr[0], ctx);
-    final StaticFunc sf = f.funcName() == null ? null :
-      ctx.funcs.get(f.funcName(), f.arity(), null, false);
-    return new PlainDoc(ctx, info).function(f.funcName(), sf, f.funcType(), null);
+    final FItem func = checkFunc(expr[0], ctx);
+    final QNm name = func.funcName();
+    final StaticFunc sf = name == null ? null : ctx.funcs.get(name, func.arity(), null, false);
+    return new PlainDoc(ctx, info).function(name, sf, func.funcType(), func.annotations(), null);
   }
\ No newline at end of file
