diff --git a/old_test.java b/new_test.java
index b8e3275..4534c2d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,11 @@
 @Test
-    public void testGetCweName() {
+    public void testGetName() {
         String cweId = "CWE-16";
         String expResult = "Configuration";
-        String result = CweDB.getCweName(cweId);
+        String result = CweDB.getName(cweId);
         assertEquals(expResult, result);
+
+        cweId = "CWE-260000";
+        result = CweDB.getName(cweId);
+        assertNull(result);
     }
\ No newline at end of file
