diff --git a/old_product.java b/new_product.java
index 546708b..b14f087 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ private long reinitializeFile(AlluxioURI path, CreateFileOptions options)
     LineageMasterClient masterClient = mLineageContext.acquireMasterClient();
     try {
       return masterClient.reinitializeFile(path.getPath(), options.getBlockSizeBytes(),
-          options.getTtl());
+          options.getTtl(), options.toThrift().getTtlExpiryAction());
     } finally {
       mLineageContext.releaseMasterClient(masterClient);
     }
