diff --git a/old_test.java b/new_test.java
index b3bb7d3..c1f8967 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,7 @@
     mLineageMaster.start(true);
     mLineageMaster.createLineage(new ArrayList<AlluxioURI>(),
         Lists.newArrayList(new AlluxioURI("/test1")), mJob);
-    mLineageMaster.reinitializeFile("/test1", 500L, 10L);
-    Mockito.verify(mFileSystemMaster).reinitializeFile(new AlluxioURI("/test1"), 500L, 10L);
+    mLineageMaster.reinitializeFile("/test1", 500L, 10L, TtlExpiryAction.DELETE);
+    Mockito.verify(mFileSystemMaster).reinitializeFile(new AlluxioURI("/test1"), 500L, 10L,
+        TtlExpiryAction.DELETE);
   }
\ No newline at end of file
