diff --git a/old_product.java b/new_product.java
index a25e72e..31ba5ce 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public Corpus getCorpus(Corpus corpus) {
-        ConceptInsightsId.validateGenarate(corpus, getAccountId());
-        return executeRequest(version + corpus.getId(), null, Corpus.class);
-    }
\ No newline at end of file
+public Corpus getCorpus(final Corpus corpus) {
+		String corpusId = IDValidator.getCorpusId(corpus, getAccountId());
+		return executeRequest(API_VERSION + corpusId, null, Corpus.class);
+	}
\ No newline at end of file
