diff --git a/old_test.java b/new_test.java
index c60bb60..2e4333d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,7 @@
 @Test
-    public void testGetCorpus() {
-        Map<String,Object> params = new HashMap<String, Object>();
-        params.put(ConceptInsights.ACCOUNT_ID, "public");
-        params.put(ConceptInsights.CORPUS, "TEDTalks");
-
-        Corpus corpus= service.getCorpus("public", "TEDTalks");
-        Assert.assertNotNull(corpus);
-    }
\ No newline at end of file
+	public void testGetCorpus() {
+		Corpus corpus = service.getCorpus(Corpus.IBM_RESEARCHERS);
+		Assert.assertNotNull(corpus);
+		corpus = service.getCorpus(Corpus.TED_TALKS);
+		Assert.assertNotNull(corpus);
+	}
\ No newline at end of file
