diff --git a/old_test.java b/new_test.java
index ce536f0..c603837 100644
--- a/old_test.java
+++ b/new_test.java
@@ -26,7 +26,7 @@
         request.addParameter("publicFlag", String.valueOf(KnowledgeLogic.PUBLIC_FLAG_PROTECT));
         request.addParameter("updateContent", "true");
         
-        LoginedUser user = super.getLoginUser(USER1);
+        AccessUser user = super.getLoginUser(USER1);
         request.addParameter("groups", TargetLogic.ID_PREFIX_USER + user.getUserId());
         
         JsonBoundary jsonBoundary = invoke(request, response, JsonBoundary.class);
