diff --git a/old_product.java b/new_product.java
index 51a8a78..7c88f54 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 @RequestMapping(value="profileFilterRecord", method=RequestMethod.GET)
     @ResponseBody
     public ProfileFilterRecordDto getProfileFilterRecord(
-            @RequestParam(value="endpointProfileSchemaId") String endpointProfileSchemaId,
-            @RequestParam(value="serverProfileSchemaId") String serverProfileSchemaId,
+            @RequestParam(value="endpointProfileSchemaId", required = false) String endpointProfileSchemaId,
+            @RequestParam(value="serverProfileSchemaId", required = false) String serverProfileSchemaId,
             @RequestParam(value="endpointGroupId") String endpointGroupId) throws KaaAdminServiceException {
         return kaaAdminService.getProfileFilterRecord(endpointProfileSchemaId, serverProfileSchemaId, endpointGroupId);
     }
\ No newline at end of file
