diff --git a/old_product.java b/new_product.java
index a591f83..53588a3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,7 @@
 @Override
     @Transactional(readOnly = true)
     public Set<String> getDependenciesForCommand(
-        @NotBlank(message = "No command id entered. Unable to get dependencies.")
-        final String id
+        @NotBlank(message = "No command id entered. Unable to get dependencies.") final String id
     ) throws GenieException {
-        return this.findCommand(id).getDependencies();
+        return this.findCommand(id).getDependencies().stream().map(FileEntity::getFile).collect(Collectors.toSet());
     }
\ No newline at end of file
