diff --git a/old_test.java b/new_test.java
index 3fd703c..b4c2a84 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(expected = GenieNotFoundException.class)
     public void testGetDependenciesForCommand() throws GenieException {
         final String id = UUID.randomUUID().toString();
-        Mockito.when(this.jpaCommandRepository.findOne(id)).thenReturn(null);
+        Mockito.when(this.jpaCommandRepository.findByUniqueId(id)).thenReturn(Optional.empty());
         this.service.getDependenciesForCommand(id);
     }
\ No newline at end of file
