diff --git a/old_product.java b/new_product.java
index d716832..534e9ac 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static int encodeFormatBits(QrCode.ErrorCorrectionLevel level , int mask ) {
+public static int encodeFormatBits(QrCode.ErrorLevel level , int mask ) {
 		int message = (level.value << 3) | (mask & 0xFFFFFFF7);
 		message = message << 10;
 		return message ^ bitPolyModulus(message, FORMAT_GENERATOR,15,5);
