diff --git a/old_test.java b/new_test.java
index 33d16b8..7444181 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,9 @@
 @Test
     public void testGetProfileMeasurements() throws Exception {
-        Collection<ProfileMeasurement> pms = profiler.getProfileMeasurements();
+        ProfileMeasurement pms[] = profiler.getProfileMeasurements();
         assertNotNull(pms);
-        assertEquals(4, pms.size());
+        assertEquals(4, pms.length);
+        for (int i = 0; i < pms.length; i++) {
+            assertNotNull(Integer.toString(i), pms[i]);
+        } // FOR
     }
\ No newline at end of file
