diff --git a/old_product.java b/new_product.java
index e5035ba..a23632e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -27,6 +27,6 @@ private void restore(String token, String region, Date startTime, Date endTime,
             config.setDC(origRegion);
             priamServer.getId().getInstance().setToken(origToken);
         }
-        tuneCassandra.updateYaml(false);
-        SystemUtils.startCassandra(true, config);
+        tuner.updateAutoBootstrap(config.getYamlLocation(), false);
+        cassProcess.start(true);
     }
\ No newline at end of file
