diff --git a/old_product.java b/new_product.java
index 519ab25..1a76a10 100644
--- a/old_product.java
+++ b/new_product.java
@@ -39,10 +39,7 @@ private boolean negotiate(final Request request, final HttpServletResponse respo
         try {
             if (securityContext.isContinue() || ntlmPost) {
                 response.setHeader("Connection", "keep-alive");
-                response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
-                String body = "Unauthorized";
-                response.getWriter().write(body);
-                response.setContentLength(body.length());
+                response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
                 response.flushBuffer();
                 return false;
             }
