diff --git a/old_test.java b/new_test.java
index b3d8590..6bc5e0f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,7 +13,7 @@
     localNode.addMountPaths(Collections.singletonList(mountFile.getAbsolutePath()));
     PartitionId newPartition1 =
         new MockPartitionId(10L, MockClusterMap.DEFAULT_PARTITION_CLASS, clusterMap.getDataNodes(), newMountPathIndex);
-    StorageManager storageManager = createStorageManager(localReplicas, metricRegistry, null);
+    StorageManager storageManager = createStorageManager(localNode, metricRegistry, null);
     storageManager.start();
     // test add store that already exists, which should fail
     assertFalse("Add store which is already existing should fail", storageManager.addBlobStore(localReplicas.get(0)));
@@ -35,7 +35,7 @@
     String diskToFail = mountPaths.get(0);
     File reservePoolDir = new File(diskToFail, diskManagerConfig.diskManagerReserveFileDirName);
     File storeReserveDir = new File(reservePoolDir, DiskSpaceAllocator.STORE_DIR_PREFIX + newPartition2.toString());
-    StorageManager storageManager2 = createStorageManager(localReplicas, new MetricRegistry(), null);
+    StorageManager storageManager2 = createStorageManager(localNode, new MetricRegistry(), null);
     storageManager2.start();
     Utils.deleteFileOrDirectory(storeReserveDir);
     assertTrue("File creation should succeed", storeReserveDir.createNewFile());
