diff --git a/old_product.java b/new_product.java
index 3339680..5a6b145 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 @Bean
-    @ConditionalOnMissingBean(JobDirectoryManifestService.class)
-    public JobDirectoryManifestService jobDirectoryManifestService(
+    @ConditionalOnMissingBean(JobDirectoryManifestCreatorService.class)
+    public JobDirectoryManifestCreatorServiceImpl jobDirectoryManifestCreatorService(
         final DirectoryManifest.Factory directoryManifestFactory,
         @Qualifier("jobDirectoryManifestCache") final Cache<Path, DirectoryManifest> cache
     ) {
-        return new JobDirectoryManifestServiceImpl(directoryManifestFactory, cache, false);
+        return new JobDirectoryManifestCreatorServiceImpl(directoryManifestFactory, cache, false);
     }
\ No newline at end of file
