diff --git a/old_test.java b/new_test.java
index e4f14b5..8f3561f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testJobDirectoryManifestService() {
         this.contextRunner.run(
             context -> {
-                Assertions.assertThat(context).hasSingleBean(JobDirectoryManifestService.class);
+                Assertions.assertThat(context).hasSingleBean(JobDirectoryManifestCreatorService.class);
             }
         );
     }
\ No newline at end of file
