diff --git a/old_test.java b/new_test.java
index b8b73ff..763600a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
 			EWAHCompressedBitmap[] x = i.next();
 			EWAHCompressedBitmap tanswer = x[0].xor(x[1]);
 			EWAHCompressedBitmap x1 = IteratorUtil
-					.materialize(IteratorAggregation.xor(
+					.materialize(IteratorAggregation.bufferedxor(
 							x[0].getIteratingRLW(), x[1].getIteratingRLW()));
 			assertTrue(x1.equals(tanswer));
 		}
