diff --git a/old_product.java b/new_product.java
index f90d21e..4588efe 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,8 @@
-public Optional<DataSet> getDataSet(@Nonnull final String id) {
+public Optional<DataSet> getDataSet(@Nonnull final String id, boolean encryptedCredentials) {
         try {
-            return Optional.of(get(path("catalog", "dataset", id), DataSet.class));
+            return Optional.of(get(path("catalog", "dataset", id), 
+                                   uri -> uri.queryParam("encrypt", encryptedCredentials),
+                                   DataSet.class));
         } catch (final HttpClientErrorException e) {
             if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
                 return Optional.empty();
