diff --git a/old_product.java b/new_product.java
index bd1ad8c..a498240 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 @Override
   public int putAll(Iterable<? extends KTypeVTypeCursor<? extends KType, ? extends VType>> iterable){
-    final int count = this.assigned;
+    final int count = size();
     for (KTypeVTypeCursor<? extends KType, ? extends VType> c : iterable) {
       put(c.key, c.value);
     }
-    return this.assigned - count;
+    return size() - count;
   }
\ No newline at end of file
