diff --git a/old_test.java b/new_test.java
index 93d8c93..1dc51bc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
-    public void removeFromRealm_removedFromResults() {
+    public void deleteFromRealm_removedFromResults() {
         realm.beginTransaction();
-        realm.clear(Dog.class);
+        realm.delete(Dog.class);
         Dog dogToAdd = realm.createObject(Dog.class);
         dogToAdd.setName("Rex");
         realm.commitTransaction();
@@ -11,7 +11,7 @@
         Dog dogToRemove = realm.where(Dog.class).findFirst();
         assertNotNull(dogToRemove);
         realm.beginTransaction();
-        dogToRemove.removeFromRealm();
+        dogToRemove.deleteFromRealm();
         realm.commitTransaction();
 
         assertEquals(0, realm.allObjects(Dog.class).size());
