diff --git a/old_product.java b/new_product.java
index d22919a..8f79797 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,4 @@
-public void removeFromRealm() {
-        if (row == null) {
-            throw new IllegalStateException("Object malformed: missing object in Realm. Make sure to instantiate RealmObjects with Realm.createObject()");
-        }
-        if (realm == null) {
-            throw new IllegalStateException("Object malformed: missing Realm. Make sure to instantiate RealmObjects with Realm.createObject()");
-        }
-        realm.checkIfValid();
-
-        row.getTable().moveLastOver(row.getIndex());
-        row = InvalidRow.INSTANCE;
+@Deprecated
+    public void removeFromRealm() {
+        deleteFromRealm();
     }
\ No newline at end of file
