diff --git a/old_product.java b/new_product.java
index 6f0c775..4a61cd9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 private void copy(String storeId, File src, File tgt, long capacityInBytes) throws Exception {
-    try (StoreCopier copier = new StoreCopier(storeId, src, tgt, capacityInBytes, fetchSizeInBytes, storeConfig,
-        new MetricRegistry(), storeKeyFactory, diskIOScheduler, transformers, time)) {
+    try (
+        StoreCopier copier = new StoreCopier(storeId, src, tgt, capacityInBytes, fetchSizeInBytes, storeConfig, metrics,
+            storeKeyFactory, diskIOScheduler, transformers, time)) {
       copier.copy(new StoreFindTokenFactory(storeKeyFactory).getNewFindToken());
     }
     // verify that the stores are equivalent
