diff --git a/old_product.java b/new_product.java
index 8c3fe9d..00449d1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-CompletableFuture<Integer> read(SegmentHandle segmentHandle, long offset, byte[] buffer, int bufferOffset, int length, Duration timeout);
\ No newline at end of file
+CompletableFuture<Integer> read(String streamSegmentName, long offset, byte[] buffer, int bufferOffset, int length, Duration timeout);
\ No newline at end of file
