diff --git a/old_product.java b/new_product.java
index 55f0b4d..a4a9a79 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,14 @@
-@Override
-    public void close() {
+public List<FutureReadResultEntry> close() {
+        List<FutureReadResultEntry> result;
         synchronized (this.reads) {
             if (this.closed) {
-                return;
+                result = Collections.emptyList();
+            } else {
+                result = new ArrayList<>(this.reads);
+                this.reads.clear();
+                this.closed = true;
             }
-
-            this.closed = true;
         }
 
-        cancelAll();
+        return result;
     }
\ No newline at end of file
