diff --git a/old_product.java b/new_product.java
index efb9532..3caa304 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,5 @@
-protected Set<ServiceComponentResponse> getComponents(
-      Set<ServiceComponentRequest> requests) throws AmbariException {
-
-    Set<ServiceComponentResponse> response = new HashSet<ServiceComponentResponse>();
+protected Set<ServiceComponentResponse> getComponents(Set<ServiceComponentRequest> requests) throws AmbariException {
+    Set<ServiceComponentResponse> response = new HashSet<>();
     for (ServiceComponentRequest request : requests) {
       try {
         response.addAll(getComponents(request));
