diff --git a/old_test.java b/new_test.java
index 2104289..73e5b83 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-  public void testGetComponents___ServiceComponentNotFoundException() throws Exception {
+  public void testGetComponents_ServiceComponentNotFoundException() throws Exception {
     // member state mocks
     Injector injector = createStrictMock(Injector.class);
     Capture<AmbariManagementController> controllerCapture = EasyMock.newCapture();
