diff --git a/old_test.java b/new_test.java
index b3f6be5..0c435ef 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,8 +7,8 @@
 
         CorfuRuntime corfuRuntime = createDefaultRuntime();
         Layout l = incrementClusterEpoch(corfuRuntime);
-        corfuRuntime.getRouter("localhost:9000").getClient(BaseClient.class).restart()
-                .get();
+        corfuRuntime.getLayoutView().getEpochedClient(l).getBaseClient("localhost:9000")
+                .restart().get();
 
         restartServer(corfuRuntime, DEFAULT_ENDPOINT);
 
