diff --git a/old_test.java b/new_test.java
index 9b8867f..5b19969 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,9 @@
 @Test
   public void getUnderFSClassTest() {
+    Whitebox.setInternalState(UnderFileSystemCluster.class, "sUnderFSClass",
+        (String) null);
     String underFSClass = UnderFileSystemCluster.getUnderFSClass();
-    Assert.assertEquals("alluxio.underfs.LocalFileSystemCluster", underFSClass);
+    Assert.assertNull(underFSClass);
 
     Whitebox.setInternalState(UnderFileSystemCluster.class, "sUnderFSClass",
         "alluxio.underfs.hdfs.LocalMiniDFSCluster");
