diff --git a/old_product.java b/new_product.java
index 14b6a72..69a4d59 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,20 +1,15 @@
 @Override
     @Transactional
-    public List<Cluster> chooseClusterForJob(
+    public List<Cluster> chooseClusterForJobRequest(
             @NotBlank(message = "No job id entered. Unable to continue.")
-            final String jobId
+            final JobRequest jobRequest
     ) throws GenieException {
         if (LOG.isDebugEnabled()) {
             LOG.debug("Called");
         }
-        final JobEntity jobEntity = this.jobRepo.findOne(jobId);
-        if (jobEntity == null) {
-            throw new GenieNotFoundException("No job with id " + jobId + " exists. Unable to continue."
-            );
-        }
 
-        final List<ClusterCriteria> clusterCriterias = jobEntity.getRequest().getClusterCriteriasAsList();
-        final Set<String> commandCriteria = jobEntity.getRequest().getCommandCriteriaAsSet();
+        final List<ClusterCriteria> clusterCriterias = jobRequest.getClusterCriterias();
+        final Set<String> commandCriteria = jobRequest.getCommandCriteria();
 
         for (final ClusterCriteria clusterCriteria : clusterCriterias) {
             @SuppressWarnings("unchecked")
