diff --git a/old_test.java b/new_test.java
index 40fcfe6..2fab390 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,13 @@
 @Test
   public void foldRightTest() {
     query("declare function local:before-first(" +
-        "  $input as item()*," +
-        "  $pred as function(item()) as item()*" +
-        ") as item()* {" +
-        "  fold-right(" +
-        "    function($x, $xs) { if($pred($x)) then () else ($x, $xs) }," +
-        "      (), $input)" +
-        "};" +
-        "local:before-first((<h1/>, <p/>, <h1/>, <h2/>, <h3/>)," +
-        "  function($it) { name($it) = 'h2' })",
-        "<h1/><p/><h1/>");
+      "  $input as item()*," +
+      "  $pred as function(item()) as item()*" +
+      ") as item()* {" +
+      "  fold-right($input, ()," +
+      "    function($x, $xs) { if($pred($x)) then () else ($x, $xs) })" +
+      "};" +
+      "local:before-first((<h1/>, <p/>, <h1/>, <h2/>, <h3/>)," +
+      "  function($it) { name($it) = 'h2' })",
+      "<h1/><p/><h1/>");
   }
\ No newline at end of file
