diff --git a/old_product.java b/new_product.java
index fc306f4..5baf20f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,10 @@
 private Iter foldRight(final QueryContext ctx) throws QueryException {
-    final FItem f = withArity(0, 2, ctx);
-    final Value xs = ctx.value(expr[2]);
+    final FItem f = withArity(2, 2, ctx);
+    final Value xs = ctx.value(expr[0]);
     // evaluate start value lazily if it's passed straight through
     if(xs.isEmpty()) return expr[1].iter(ctx);
 
     Value res = ctx.value(expr[1]);
-    for(long i = xs.size(); --i >= 0;)
-      res = f.invValue(ctx, info, xs.itemAt(i), res);
-
+    for(long i = xs.size(); --i >= 0;) res = f.invValue(ctx, info, xs.itemAt(i), res);
     return res.iter();
   }
\ No newline at end of file
