diff --git a/old_test.java b/new_test.java
index 1af501b..cc78160 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void test_parseBest_firstOption() throws Exception {
-        DateTimeFormatter test = DateTimeFormatter.ofPattern("yyyy-MM[-dd]");
+        DateTimeFormatter test = DateTimeFormatter.ofPattern("uuuu-MM[-dd]");
         TemporalAccessor result = test.parseBest("2011-06-30", LocalDate.FROM, YearMonth.FROM);
         assertEquals(result, LocalDate.of(2011, 6, 30));
     }
\ No newline at end of file
