diff --git a/old_test.java b/new_test.java
index 338b093..a0d9f9e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,4 @@
 @Test
-	@Verifies(value = "should unvoid membership for every patient unvoided", method = "patientUnvoided(Patient, User, "
-			+ "Date, String)")
 	public void patientUnvoided_shouldUnvoidMemberships() throws Exception {
 		executeDataSet(COHORT_XML);
 		
@@ -10,14 +8,14 @@
 		Date dateVoided = new Date();
 		String voidReason = "Associated patient is voided";
 		
-		CohortMembership voidedMembership = new CohortMembership(unvoidedPatient);
+		CohortMembership voidedMembership = new CohortMembership(unvoidedPatient.getPatientId());
+		cohort.addMembership(voidedMembership);
 		voidedMembership.setVoided(true);
 		voidedMembership.setVoidedBy(voidedBy);
 		voidedMembership.setDateVoided(dateVoided);
 		voidedMembership.setVoidReason(voidReason);
 		
-		cohort.addMembership(voidedMembership);
-		service.patientUnvoided(unvoidedPatient, voidedBy, dateVoided, voidReason);
+		service.patientUnvoided(unvoidedPatient, voidedBy, dateVoided);
 		
 		assertFalse(voidedMembership.getVoided());
 		assertNull(voidedMembership.getVoidedBy());
