diff --git a/old_product.java b/new_product.java
index 80e69f7..55b8324 100644
--- a/old_product.java
+++ b/new_product.java
@@ -22,19 +22,15 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
         TextClassifier.TrainingParameters trainingParameters = new TextClassifier.TrainingParameters();
         
         //Classifier configuration
-        trainingParameters.setModelerClass(modelerClass);
         trainingParameters.setModelerTrainingParameters(modelerTrainingParameters);
         
         //data transfomation configuration
-        trainingParameters.setDataTransformerClass(null);
         trainingParameters.setDataTransformerTrainingParameters(null);
         
         //feature selection configuration
-        trainingParameters.setFeatureSelectorClass(featureSelectorClass);
         trainingParameters.setFeatureSelectorTrainingParameters(featureSelectorTrainingParameters);
         
         //text extraction configuration
-        trainingParameters.setTextExtractorClass(NgramsExtractor.class);
         NgramsExtractor.Parameters exParams = new NgramsExtractor.Parameters();
         exParams.setMaxDistanceBetweenKwds(2);
         exParams.setExaminationWindowLength(6);
