diff --git a/old_test.java b/new_test.java
index db380a0..80dc533 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,19 +16,16 @@
         
         
         //Model Configuration
-        
-        trainingParameters.setModelerClass(MultinomialNaiveBayes.class);
+
         MultinomialNaiveBayes.TrainingParameters modelTrainingParameters = new MultinomialNaiveBayes.TrainingParameters();
         modelTrainingParameters.setMultiProbabilityWeighted(true);
         trainingParameters.setModelerTrainingParameters(modelTrainingParameters);
 
         //data transfomation configuration
-        trainingParameters.setDataTransformerClass(DummyXMinMaxNormalizer.class);
         DummyXMinMaxNormalizer.TrainingParameters dtParams = new DummyXMinMaxNormalizer.TrainingParameters();
         trainingParameters.setDataTransformerTrainingParameters(dtParams);
         
         //feature selection configuration
-        trainingParameters.setFeatureSelectorClass(null);
         trainingParameters.setFeatureSelectorTrainingParameters(null);
 
         Modeler instance = new Modeler(dbName, conf, trainingParameters);
