diff --git a/old_product.java b/new_product.java
index f1dc02d..c3c7dc2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,15 +4,17 @@ private static void put(String key) {
     for (int i = 0; i < NO_PUT_OPERATION; i++) {
       switch (i) {
         case 0:
-          val = new DeltaTestImpl(0, "0", new Double(0), new byte[0], new TestObject1("0", 0));
+          val = new DeltaTestImpl(0, "0", new Double(0), new byte[0],
+              new TestObjectWithIdentifier("0", 0));
           break;
         case 1:
-          val = new DeltaTestImpl(0, "0", new Double(0), new byte[0], new TestObject1("0", 0));
+          val = new DeltaTestImpl(0, "0", new Double(0), new byte[0],
+              new TestObjectWithIdentifier("0", 0));
           val.setStr((String) putDelta[0]);
           break;
         case 2:
           val = new DeltaTestImpl(0, (String) putDelta[1], new Double(0), new byte[0],
-              new TestObject1("0", 0));
+              new TestObjectWithIdentifier("0", 0));
           val.setIntVar(((Integer) putDelta[2]).intValue());
           break;
       }
