diff --git a/old_test.java b/new_test.java
index 9c18212..ad4df38 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	public void testGeneratePagingQueryStatement() {
 		String sqlExpected = "select * from DATA_SOURCES LIMIT 0,10";
-		String sql = mySqlPagingQueryDialect.generatePagingQueryStatement();
+		String sql = mySqlPagingQueryDialect.generatePagingQueryStatement().getStatment();
 		assertEquals(sqlExpected, sql);
 	}
\ No newline at end of file
