diff --git a/old_test.java b/new_test.java
index 1716728..ae3ec77 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void getFileInfoTest() throws Exception {
-    long fileId = mFileSystemMaster.create(ROOT_FILE_URI, sNestedFileOptions);
+    long fileId = mFileSystemMaster.createFile(ROOT_FILE_URI, sNestedFileOptions);
 
     mFileSystemMaster.getFileInfo(fileId);
 
