diff --git a/old_product.java b/new_product.java
index 06cc005..02b0375 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,6 @@
 public FileInfo getFileInfo(long fileId) throws FileDoesNotExistException {
     MasterContext.getMasterSource().incGetFileInfoOps(1);
-    synchronized (mInodeTree) {
-      Inode<?> inode = mInodeTree.getInodeById(fileId);
-      return getFileInfoInternal(inode);
+    try (InodePath inodePath = mInodeTree.getInodePath(fileId)) {
+      return getFileInfoInternal(inodePath.getInode());
     }
   }
\ No newline at end of file
