diff --git a/old_test.java b/new_test.java
index 74193cb..47af624 100644
--- a/old_test.java
+++ b/new_test.java
@@ -22,12 +22,14 @@
         null);
     ServiceComponentRequest request4 = new ServiceComponentRequest("cluster1", "service1", "component4",
         null);
+    ServiceComponentRequest request5 = new ServiceComponentRequest("cluster1", "service2", null, null);
 
     Set<ServiceComponentRequest> setRequests = new HashSet<ServiceComponentRequest>();
     setRequests.add(request1);
     setRequests.add(request2);
     setRequests.add(request3);
     setRequests.add(request4);
+    setRequests.add(request5);
 
     // expectations
     // constructor init
@@ -35,9 +37,10 @@
     expect(managementController.getAmbariMetaInfo()).andReturn(ambariMetaInfo).anyTimes();
 
     // getComponents
-    expect(clusters.getCluster("cluster1")).andReturn(cluster).times(4);
+    expect(clusters.getCluster("cluster1")).andReturn(cluster).times(5);
     expect(cluster.getDesiredStackVersion()).andReturn(stackId).anyTimes();
     expect(cluster.getService("service1")).andReturn(service).times(4);
+    expect(cluster.getService("service2")).andThrow(new ObjectNotFoundException("service2"));
 
     expect(service.getServiceComponent("component1")).andThrow(new ServiceComponentNotFoundException("cluster1", "service1", "component1"));
     expect(service.getServiceComponent("component2")).andThrow(new ServiceComponentNotFoundException("cluster1", "service1", "component2"));
