diff --git a/old_product.java b/new_product.java
index c52cad9..9f102f4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -52,7 +52,6 @@ private synchronized Set<ServiceComponentResponse> getComponents(
       ServiceComponent sc = s.getServiceComponent(request.getComponentName());
       ServiceComponentResponse serviceComponentResponse = sc.convertToResponse();
 
-
       ComponentInfo componentInfo = ambariMetaInfo.getComponentCategory(stackId.getStackName(),
           stackId.getStackVersion(), s.getName(), request.getComponentName());
       if (componentInfo != null) {
@@ -81,7 +80,11 @@ private synchronized Set<ServiceComponentResponse> getComponents(
     Set<Service> services = new HashSet<Service>();
     if (request.getServiceName() != null
         && !request.getServiceName().isEmpty()) {
-      services.add(cluster.getService(request.getServiceName()));
+      try {
+        services.add(cluster.getService(request.getServiceName()));
+      } catch (ObjectNotFoundException e) {
+        throw new ParentObjectNotFoundException("Could not find service", e);
+      }
     } else {
       services.addAll(cluster.getServices().values());
     }
