diff --git a/old_product.java b/new_product.java
index e3a541f..69a6f94 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,10 @@
-public void setCurrentSchema(Schema schema) {
+public void changeCurrentSchemaTo(Schema schema) {
+        if (schema.getName().equals(originalSchema) || !schema.exists()) {
+            return;
+        }
+
         try {
-            doSetCurrentSchema(schema);
+            doChangeCurrentSchemaTo(schema.toString());
         } catch (SQLException e) {
             throw new FlywayException("Error setting current schema to " + schema, e);
         }
