diff --git a/old_test.java b/new_test.java
index 869355d..c3d4cdf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
     public void test_previousOrCurrent_serialization() throws IOException, ClassNotFoundException {
-        DateTimeAdjuster previousOrCurrent = DateTimeAdjusters.previousOrCurrent(SUNDAY);
+        WithAdjuster previousOrCurrent = DateTimeAdjusters.previousOrCurrent(SUNDAY);
         assertTrue(previousOrCurrent instanceof Serializable);
 
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
