diff --git a/old_product.java b/new_product.java
index 3606faf..608e200 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public byte[] buildClass( ClassDefinition classDef ) throws IOException,
+public byte[] buildClass( ClassDefinition classDef, ClassLoader classLoader ) throws IOException,
             IntrospectionException,
             SecurityException,
             IllegalArgumentException,
@@ -9,7 +9,7 @@ public byte[] buildClass( ClassDefinition classDef ) throws IOException,
             InstantiationException,
             NoSuchFieldException {
 
-        ClassWriter cw = new ClassWriter( ClassWriter.COMPUTE_MAXS );
+        ClassWriter cw = new ClassGenerator.InternalClassWriter( classLoader, ClassWriter.COMPUTE_MAXS | ClassWriter.COMPUTE_FRAMES );
         //ClassVisitor cw = new CheckClassAdapter(cwr);
 
         this.buildClassHeader( cw,
