diff --git a/old_test.java b/new_test.java
index 23fd4a2..c303a5c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -25,7 +25,7 @@
                                                                stringDef.getName(),
                                                                classLoader ) );
 
-            byte[] d = builder.buildClass( classDef );
+            byte[] d = builder.buildClass( classDef, classLoader );
 
             assertSame( "Returned class should be the same",
                                clazz,
