diff --git a/old_test.java b/new_test.java
index 9dfa54d..e128b32 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,10 +16,10 @@
 		
 		List<PatientIdentifierType> identifierTypes = Collections.emptyList();
 		//we expect only one matching patient
-		int actualSize = dao.getPatients("%ca", null, identifierTypes, false, 0, null).size();
+		int actualSize = dao.getPatients("%ca", null, identifierTypes, false, 0, null, false).size();
 		Assert.assertEquals(1, actualSize);
 		
-		Patient actualPatient = dao.getPatients("%ca", null, identifierTypes, false, 0, null).get(0);
+		Patient actualPatient = dao.getPatients("%ca", null, identifierTypes, false, 0, null, false).get(0);
 		//if actually the search returned the matching patient
 		Assert.assertEquals(patient2, actualPatient);
 	}
\ No newline at end of file
