diff --git a/old_product.java b/new_product.java
index a168fbf..af70a0e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,3 @@
 public List<Patient> getPatients(String name, String identifier, List<PatientIdentifierType> identifierTypes,
-	        boolean matchIdentifierExactly, Integer start, Integer length) throws DAOException;
\ No newline at end of file
+	        boolean matchIdentifierExactly, Integer start, Integer length, boolean searchOnNamesOrIdentifiers)
+	        throws DAOException;
\ No newline at end of file
