diff --git a/old_product.java b/new_product.java
index 49fd7f9..3ea7147 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-protected Zhang99ParamAll initialParam( List<CalibrationObservation> observations )
+protected boolean initialParam(List<CalibrationObservation> observations , Zhang99AllParam param )
 	{
 		status("Estimating Homographies");
 		List<DMatrixRMaj> homographies = new ArrayList<>();
@@ -6,7 +6,7 @@ protected Zhang99ParamAll initialParam( List<CalibrationObservation> observation
 
 		for( CalibrationObservation obs : observations ) {
 			if( !computeHomography.computeHomography(obs) )
-				return null;
+				return false;
 
 			DMatrixRMaj H = computeHomography.getHomography();
 
@@ -28,6 +28,6 @@ protected Zhang99ParamAll initialParam( List<CalibrationObservation> observation
 
 		double distort[] = computeRadial.getParameters();
 
-		return convertIntoZhangParam(motions, K,optimized.assumeZeroSkew, distort,
-				optimized.includeTangential);
+		convertIntoZhangParam(motions, K,distort, param);
+		return true;
 	}
\ No newline at end of file
