diff --git a/old_test.java b/new_test.java
index 631cbd3..acde1a0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,17 @@
 @Test
 	public void initialParam() {
 		List<Point2D_F64> grid = GenericCalibrationGrid.standardLayout();
-		Zhang99ParamAll initial = GenericCalibrationGrid.createEasierParam(true, 2, false, 3, rand);
+		Zhang99AllParam initial = GenericCalibrationGrid.createEasierParam(true, 2, false, 3, rand);
 		// tangential can't be linearly estimated
 
 		List<CalibrationObservation> observations = GenericCalibrationGrid.createObservations(initial,grid);
 
 		Helper alg = new Helper(grid,true,2,false);
 
-		Zhang99ParamAll found = alg.initialParam(observations);
+		Zhang99AllParam found = initial.createLike();
+		alg.initialParam(observations,found);
 
-		checkIntrinsicOnly(initial, found, 0.01, 0.1, 0.1);
+		checkIntrinsicOnly(
+				(CameraPinholeRadial)initial.getIntrinsic().getCameraModel(),
+				(CameraPinholeRadial)found.getIntrinsic().getCameraModel(), 0.01, 0.1, 0.1);
 	}
\ No newline at end of file
