diff --git a/old_test.java b/new_test.java
index 27666cb..c1954c2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -21,16 +21,4 @@
     // MAPREDUCE2 installed
     services.put("MAPREDUCE2", Mockito.mock(Service.class));
     Assert.assertTrue(m_check.isApplicable(request));
-
-    // Should not be supported for any upgrade from version less than 2.3.0.0
-    request.setTargetStackId(new StackId("HDP", "2.2.0.1"));
-    Assert.assertFalse(m_check.isApplicable(request));
-
-    request.setSourceStackId(new StackId("HDP", "2.2.0.1"));
-    request.setTargetStackId(new StackId("HDP", "2.3.0.0"));
-    Assert.assertFalse(m_check.isApplicable(request));
-
-    request.setSourceStackId(new StackId("HDP", "2.2.0.1"));
-    request.setTargetStackId(new StackId("HDP", "2.2.1.0"));
-    Assert.assertFalse(m_check.isApplicable(request));
   }
\ No newline at end of file
