diff --git a/old_product.java b/new_product.java
index 1e69af7..ecc3128 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,22 +1,4 @@
 @Override
   public boolean isApplicable(PrereqCheckRequest request) throws AmbariException {
-    if (!super.isApplicable(request, Arrays.asList("MAPREDUCE2"), true)) {
-      return false;
-    }
-
-    final Cluster cluster = clustersProvider.get().getCluster(request.getClusterName());
-
-    // Applicable only if stack not defined in MinimumApplicableStackVersion, or
-    // version equals or exceeds the enumerated version.
-    for (MinimumApplicableStackVersion minimumStackVersion : MinimumApplicableStackVersion.values()) {
-      String stackName = cluster.getCurrentStackVersion().getStackName();
-      if (minimumStackVersion.getStackName().equals(stackName)){
-        String targetVersion = request.getTargetStackId().getStackVersion();
-        String sourceVersion = request.getSourceStackId().getStackVersion();
-        return VersionUtils.compareVersions(targetVersion, minimumStackVersion.getStackVersion()) >= 0 &&
-               VersionUtils.compareVersions(sourceVersion, minimumStackVersion.getStackVersion()) >= 0;
-      }
-    }
-
-    return true;
+    return super.isApplicable(request, Arrays.asList("MAPREDUCE2"), true);
   }
\ No newline at end of file
