diff --git a/old_product.java b/new_product.java
index 6d65076..d776b59 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,11 +7,12 @@ public double get(VectorEntry entry) {
         } else if (evec.keys != this.keys) {
             throw new IllegalArgumentException("entry does not have safe key domain");
         } else if (entry.getKey() != keys[eind]) {
+            // REVIEW Should this be an assertion?
             throw new IllegalArgumentException("entry does not have the correct key for its index");
         }
         if (usedKeys.get(eind)) {
             return values[eind];
         } else {
-            return Double.NaN;
+            throw new IllegalArgumentException("Key " + entry.getKey() + " is not set");
         }
     }
\ No newline at end of file
