diff --git a/old_product.java b/new_product.java
index 9ec8110..c44a8ff 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,5 @@
-public static <O> List<O> removeDuplicate(Collection<O> objectCollection){
-        if (Validator.isNullOrEmpty(objectCollection)){
-            return Collections.emptyList();
-        }
-        return new ArrayList<O>(new LinkedHashSet<O>(objectCollection));
+@SuppressWarnings("unchecked")
+    public static <O> List<O> removeDuplicate(Collection<O> objectCollection){
+        return Validator.isNullOrEmpty(objectCollection) ? (List<O>) Collections.emptyList()
+                        : new ArrayList<O>(new LinkedHashSet<O>(objectCollection));
     }
\ No newline at end of file
