diff --git a/old_test.java b/new_test.java
index e49973e..020640a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(groups={"tck"})
-    public void test_toMillisLong() {
+    public void test_toMillis() {
         Duration test = Duration.ofSeconds(321, 123456789);
-        assertEquals(test.toMillisLong(), 321000 + 123);
+        assertEquals(test.toMillis(), 321000 + 123);
     }
\ No newline at end of file
