diff --git a/old_test.java b/new_test.java
index e9bfe1f..9b6ec52 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,9 @@
 @Test public void create() {
     // allow empty passwords, overwriting existing users
     query(_USER_CREATE.args(NAME, ""));
-    // specify permission
+    // specify permissions
     query(_USER_CREATE.args(NAME, NAME, Perm.ADMIN));
+    query(_USER_CREATE.args(NAME, NAME, "('admin','none')", "('','x')"));
 
     // invalid permission
     error(_USER_CREATE.args(NAME, NAME, ""), USER_PERMISSION_X);
@@ -14,4 +15,6 @@
 
     // redundant operations
     error(_USER_CREATE.args(NAME, "") + ',' + _USER_CREATE.args(NAME, ""), USER_UPDATE_X_X);
+    error(_USER_CREATE.args(NAME, "", "('admin','admin')", "('','')"), USER_SAMEPERM_X_X);
+    error(_USER_CREATE.args(NAME, "", "('admin','admin')", "('x','x')"), USER_SAMEPAT_X);
   }
\ No newline at end of file
