diff --git a/old_product.java b/new_product.java
index 1f621d3..bfa02c8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public synchronized void create(final String name, final String password, final Perm perm) {
-    users.put(name, new User(name, password, perm == null ? Perm.NONE : perm));
+public synchronized void add(final User user) {
+    users.put(user.name(), user);
   }
\ No newline at end of file
