diff --git a/old_product.java b/new_product.java
index c417036..bce8b55 100644
--- a/old_product.java
+++ b/new_product.java
@@ -21,7 +21,7 @@ public static Item restoreItem(Context c, InProgressSubmission is,
         DCDate now = DCDate.getCurrent();
         
         // If the item doesn't have a date.accessioned, set it to today
-        DCValue[] dateAccessioned = item.getDC("date", "accessioned", Item.ANY);
+        Metadatum[] dateAccessioned = item.getDC("date", "accessioned", Item.ANY);
         if (dateAccessioned.length == 0)
         {
 	        item.addDC("date", "accessioned", null, now.toString());
@@ -31,9 +31,9 @@ public static Item restoreItem(Context c, InProgressSubmission is,
         // In the below loop, we temporarily clear all issued dates and re-add, one-by-one,
         // replacing "today" with today's date.
         // NOTE: As of DSpace 4.0, DSpace no longer sets an issue date by default
-        DCValue[] currentDateIssued = item.getDC("date", "issued", Item.ANY);
+        Metadatum[] currentDateIssued = item.getDC("date", "issued", Item.ANY);
         item.clearDC("date", "issued", Item.ANY);
-        for (DCValue dcv : currentDateIssued)
+        for (Metadatum dcv : currentDateIssued)
         {
             if(dcv.value!=null && dcv.value.equalsIgnoreCase("today"))
             {
