diff --git a/old_test.java b/new_test.java
index 31324c4..f6bda1f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void testGetClassifiers() throws InterruptedException {
     server.enqueue(jsonResponse(classifiers));
-    final Classifiers response = service.getClassifiers().execute();
+    final ClassifierList response = service.listClassifiers().execute();
     final RecordedRequest request = server.takeRequest();
 
     assertEquals(CLASSIFIERS_PATH, request.getPath());
