diff --git a/old_product.java b/new_product.java
index 2332525..7d3f267 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
-public ServiceCall<Classifiers> getClassifiers() {
-    final Request request = RequestBuilder.get(PATH_CLASSIFIERS).build();
-    return createServiceCall(request, ResponseConverterUtils.getObject(Classifiers.class));
-
+public ServiceCall<ClassifierList> listClassifiers(ListClassifiersOptions listClassifiersOptions) {
+    RequestBuilder builder = RequestBuilder.get("/v1/classifiers");
+    if (listClassifiersOptions != null) {
+    }
+    return createServiceCall(builder.build(), ResponseConverterUtils.getObject(ClassifierList.class));
   }
\ No newline at end of file
