diff --git a/old_test.java b/new_test.java
index 3d61ad2..94edf6c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -43,5 +43,6 @@
 		assertThat(supplyRecord.getMD_Candidate_BusinessCase()).isEqualTo(CandidateBusinessCase.PRODUCTION.toString());
 		assertThat(stockRecord.getQty()).isEqualByComparingTo(ELEVEN.add(TWENTY_THREE));
 
-		assertThat(supplyRecord.getSeqNo()).isEqualTo(stockRecord.getSeqNo() + 1); // when we sort by SeqNo, the stock needs to be first and thus have the smaller value
+		// note that now, the stock record shall have the same SeqNo as it's "actual" record
+		assertThat(supplyRecord.getSeqNo()).isEqualTo(stockRecord.getSeqNo());
 	}
\ No newline at end of file
