diff --git a/old_test.java b/new_test.java
index b1b0289..d484af6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,18 +2,18 @@
     public void testCsvCpe() {
         EscapeTool instance = new EscapeTool();
         Set<Identifier> ids = null;
-        String expResult = "";
+        String expResult = "\"\"";
         String result = instance.csvCpe(ids);
         assertEquals(expResult, result);
         
         ids = new HashSet<>();
-        expResult = "";
+        expResult = "\"\"";
         result = instance.csvCpe(ids);
         assertEquals(expResult, result);
         
         ids = new HashSet<>();
         ids.add(new Identifier("gav", "somegroup:something:1.0", ""));
-        expResult = "";
+        expResult = "\"\"";
         result = instance.csvCpe(ids);
         assertEquals(expResult, result);
         
