diff --git a/old_product.java b/new_product.java
index 26e8976..08e65e6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-public static void derivX_I8(ImageInt8 orig,
-								 ImageInt16 derivX) {
-		InputSanityCheck.checkSameShape(orig, false, derivX, true);
+public static void derivX_I8(ImageUInt8 orig,
+								 ImageSInt16 derivX) {
+		InputSanityCheck.checkSameShape(orig, derivX);
 		GradientThree_Standard.derivX_I8(orig, derivX);
 	}
\ No newline at end of file
