diff --git a/old_product.java b/new_product.java
index c166b7b..35b59d0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-public static void execute(final GUI gui, final BaseXDialog dialog, final Runnable post,
-      final Command... cmds) {
+private static void execute(final GUI gui, final BaseXDialog dialog, final Runnable post,
+                              final Command... cmds) {
 
     for(final Command cmd : cmds) {
       // reset views
