diff --git a/old_product.java b/new_product.java
index 2f4c29c..6e68f7e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,11 @@
-public boolean equals(Object object) {
+@Override
+    public boolean equals(Object object) {
 		if (!(object instanceof ConceptDescription)) {
 			return false;
 		}
 		ConceptDescription rhs = (ConceptDescription) object;
 		if (conceptDescriptionId != null && rhs.conceptDescriptionId != null)
-			return this.conceptDescriptionId == rhs.conceptDescriptionId;
+			return this.conceptDescriptionId.equals(rhs.conceptDescriptionId);
 		else
 			return this == object;
 	}
\ No newline at end of file
