diff --git a/old_product.java b/new_product.java
index f733357..a853d8f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,12 @@
 final FuncType type(final int arity, final AnnList anns) {
     final SeqType[] st = new SeqType[arity];
     if(arity != 0 && minMax[1] == Integer.MAX_VALUE) {
-      final int al = args.length;
-      System.arraycopy(args, 0, st, 0, al);
-      final SeqType var = args[al - 1];
-      for(int a = al; a < arity; a++) st[a] = var;
+      final int pl = params.length;
+      System.arraycopy(params, 0, st, 0, pl);
+      final SeqType var = params[pl - 1];
+      for(int p = pl; p < arity; p++) st[p] = var;
     } else {
-      System.arraycopy(args, 0, st, 0, arity);
+      System.arraycopy(params, 0, st, 0, arity);
     }
     return FuncType.get(anns, type, st);
   }
\ No newline at end of file
