diff --git a/old_test.java b/new_test.java
index b2dbabe..b6a018c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,8 +9,8 @@
         validKeys.sort(KEY_REVERSE_COMPARATOR);
 
         val validKeysIterator = validKeys.iterator();
-        Integer expectedValue = Integer.MAX_VALUE;
-        for (int testKey = maxKey; testKey >= 0; testKey--) {
+        Long expectedValue = (long) Integer.MAX_VALUE;
+        for (long testKey = maxKey; testKey >= 0; testKey--) {
             // Since both testKey and validKeysIterator increase with natural ordering, finding the next expected value
             // is a straightforward call to the iterator next() method.
             while (expectedValue != null && testKey < expectedValue) {
@@ -22,7 +22,7 @@
             }
 
             val ceilingEntry = index.getFloor(testKey);
-            Integer actualValue = ceilingEntry != null ? ceilingEntry.key() : null;
+            Long actualValue = ceilingEntry != null ? ceilingEntry.key() : null;
             Assert.assertEquals("Unexpected value for getCeiling for key " + testKey, expectedValue, actualValue);
         }
     }
\ No newline at end of file
