diff --git a/old_test.java b/new_test.java
index 074123f..8fee9f9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void createFile() throws Exception {
     // created nested file
-    createPath(mTree, NESTED_FILE_URI, sNestedFileOptions);
+    createPath(mTree, NESTED_FILE_URI, sNestedFileContext);
     InodeView nestedFile = getInodeByPath(mTree, NESTED_FILE_URI);
     assertEquals("file", nestedFile.getName());
     assertEquals(2, nestedFile.getParentId());
