diff --git a/old_product.java b/new_product.java
index 746671c..08b2d65 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
-FileOutStream createFile(AlluxioURI path, CreateFileOptions options)
+FileOutStream createFile(AlluxioURI path, CreateFilePOptions options)
       throws FileAlreadyExistsException, InvalidPathException, IOException, AlluxioException;
\ No newline at end of file
