diff --git a/old_test.java b/new_test.java
index cf4bf0c..c8a4baf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void matchUdpSrcTest() {
-        Criterion criterion = Criteria.matchUdpSrc((short) 22);
+        Criterion criterion = Criteria.matchUdpSrc((short) 40000);
         ObjectNode result = criterionCodec.encode(criterion, context);
         assertThat(result.get("type").textValue(), is(criterion.type().toString()));
-        assertThat(result.get("udpPort").asInt(), is(22));
+        assertThat(result.get("udpPort").asInt(), is(40000));
     }
\ No newline at end of file
