diff --git a/old_product.java b/new_product.java
index bbd7541..43c9e7d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,11 @@
 @Override
-	public void initEncoder(int w, double minVal, double maxVal, int n,
-			double radius, double resolution) {
-		this.setPeriodic(false);
-		this.encLearningEnabled = true;
-		if (this.periodic) {
-			throw new IllegalStateException(
-					"Adaptive scalar encoder does not encode periodic inputs");
-		}
-		assert n != 0;
-		super.initEncoder(w, minVal, maxVal, n, radius, resolution);
-	}
\ No newline at end of file
+    public void initEncoder(int w, double minVal, double maxVal, int n,
+        double radius, double resolution) {
+        this.encLearningEnabled = true;
+        if(this.periodic) {
+            throw new IllegalStateException(
+                "Adaptive scalar encoder does not encode periodic inputs");
+        }
+        assert n != 0;
+        super.initEncoder(w, minVal, maxVal, n, radius, resolution);
+    }
\ No newline at end of file
