diff --git a/old_test.java b/new_test.java
index 20ffc13..38c5f5d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,13 +3,13 @@
         final Random rnd = new Random(42);
         Peer peer = null;
         try {
-            peer = new PeerMaker(new Number160(rnd)).ports(4000).makeAndListen();
+            peer = new PeerBuilder(new Number160(rnd)).ports(4000).start();
 
             Collection<PeerAddress> bootstrapTo = new ArrayList<PeerAddress>(2);
             PeerAddress pa = new PeerAddress(new Number160(rnd), "192.168.77.77", 4000, 4000);
-            bootstrapTo.add(peer.getPeerAddress());
+            bootstrapTo.add(peer.peerAddress());
             bootstrapTo.add(pa);
-            FutureBootstrap tmp = peer.bootstrap().setBootstrapTo(bootstrapTo).start();
+            FutureBootstrap tmp = peer.bootstrap().bootstrapTo(bootstrapTo).start();
             tmp.awaitUninterruptibly();
             Assert.assertEquals(false, tmp.isSuccess());
         } finally {
