diff --git a/old_test.java b/new_test.java
index 56f7b5b..ec8496e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
       new RepoManager(context).delete(token("xyz"));
       fail("Not installed package not detected.");
     } catch(final QueryException ex) {
-      check(ex, Err.PKGNOTEXIST);
+      check(ex, Err.BXRE_WHICH);
     }
     // install a package without dependencies (pkg3)
     new RepoInstall(REPO + "pkg3.xar", null).execute(context);
@@ -38,7 +38,7 @@
       new RepoManager(context).delete(token(PKG3ID));
       fail("Package involved in a dependency was deleted.");
     } catch(final QueryException ex) {
-      check(ex, Err.PKGDEP);
+      check(ex, Err.BXRE_DEP);
     }
     // try to delete pkg4 (use package name)
     new RepoDelete(PKG4, null).execute(context);
