diff --git a/old_product.java b/new_product.java
index f4ea931..4827300 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 private Item delete(final File path, final QueryContext ctx) throws QueryException {
-    if(!path.exists()) PATHNOTEXISTS.thrw(info, path.getAbsolutePath());
+    if(!path.exists()) FL_WHICH.thrw(info, path.getAbsolutePath());
     if(optionalBool(1, ctx)) deleteRec(path);
-    else if(!path.delete()) CANNOTDEL.thrw(info, path);
+    else if(!path.delete()) FL_DEL.thrw(info, path);
     return null;
   }
\ No newline at end of file
