diff --git a/old_test.java b/new_test.java
index 7b560bf..30eb567 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,11 +10,11 @@
 		Assert.assertEquals(
 				"Expected picked but not delivered",
 				Arrays.asList(qp1, qp3),
-				dao.retrievePickedNotDeliveredRecords(ss, I_M_ShipmentSchedule_QtyPicked.class));
+				dao.retrieveNotOnShipmentLineRecords(ss, I_M_ShipmentSchedule_QtyPicked.class));
 
 		Assert.assertEquals(
 				"Expected picked AND delivered",
 				Arrays.asList(qp2, qp4),
-				dao.retrievePickedAndDeliveredRecordsQuery(ss).create().list());
+				dao.retrieveOnShipmentLineRecordsQuery(ss).create().list());
 
 	}
\ No newline at end of file
