diff --git a/old_product.java b/new_product.java
index 121e4e9..966f6ef 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-<T extends I_M_ShipmentSchedule_QtyPicked> List<T> retrievePickedNotDeliveredRecords(I_M_ShipmentSchedule shipmentSchedule, Class<T> clazz);
\ No newline at end of file
+<T extends I_M_ShipmentSchedule_QtyPicked> List<T> retrieveNotOnShipmentLineRecords(I_M_ShipmentSchedule shipmentSchedule, Class<T> clazz);
\ No newline at end of file
