diff --git a/old_test.java b/new_test.java
index 6a91ec9..74438d7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,16 @@
 @Test
   public void testExecuteDMLUpdates() throws Exception {
     Method removeNagiosService = UpgradeCatalog200.class.getDeclaredMethod("removeNagiosService");
+    Method updateHiveDatabaseType = UpgradeCatalog200.class.getDeclaredMethod("updateHiveDatabaseType");
 
     UpgradeCatalog200 upgradeCatalog = createMockBuilder(
-        UpgradeCatalog200.class).addMockedMethod(removeNagiosService).createMock();
+        UpgradeCatalog200.class).addMockedMethod(removeNagiosService).addMockedMethod(updateHiveDatabaseType).createMock();
 
     upgradeCatalog.removeNagiosService();
     expectLastCall().once();
+    upgradeCatalog.updateHiveDatabaseType();
+    expectLastCall().once();
+
 
     replay(upgradeCatalog);
 
