diff --git a/old_test.java b/new_test.java
index a3f6f87..6ee8351 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,4 @@
-@Test(groups={"tck"})
-    public void test_plusNanos() {
-        LocalDateTime ldt = LocalDateTime.of(2008, 6, 30, 23, 30, 59, 0);
-        ZonedDateTime base = ZonedDateTime.of(ldt, ZONE_0100);
-        ZonedDateTime test = base.plusNanos(1);
-        assertEquals(test, ZonedDateTime.of(ldt.plusNanos(1), ZONE_0100));
+@Test(groups={"tck"}, dataProvider="plusTime")
+    public void test_plusNanos(ZonedDateTime base, long amount, ZonedDateTime expected) {
+        assertEquals(base.plusNanos(amount * 3600_000_000_000L), expected);
     }
\ No newline at end of file
