diff --git a/old_product.java b/new_product.java
index f3d1fbf..ba05f5f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,10 +3,10 @@ public StrBuilder reverse() {
             return this;
         }
         
-        int half = size / 2;
-        char[] buf = buffer;
+        final int half = size / 2;
+        final char[] buf = buffer;
         for (int leftIdx = 0, rightIdx = size - 1; leftIdx < half; leftIdx++,rightIdx--) {
-            char swap = buf[leftIdx];
+            final char swap = buf[leftIdx];
             buf[leftIdx] = buf[rightIdx];
             buf[rightIdx] = swap;
         }
