diff --git a/old_test.java b/new_test.java
index cd35322..efc7cfd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
     public void testReverse() {
-        StringBuffer str1 = new StringBuffer("pick");
-        String str2 = "a";
-        String[] str3 = new String[] {"stick"};
-        String str4 = "up";
+        final StringBuffer str1 = new StringBuffer("pick");
+        final String str2 = "a";
+        final String[] str3 = new String[] {"stick"};
+        final String str4 = "up";
         
         Object[] array = new Object[] {str1, str2, str3};
         ArrayUtils.reverse(array);
