diff --git a/old_product.java b/new_product.java
index 5585bbf..e710a34 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,4 @@
 @Override
 	public Order saveOrder(Order order) throws APIException {
-		if (dao.isActivatedInDatabase(order))
-			throw new APIException("Cannot modify an activated order");
-		
 		return saveOrderWithLesserValidation(order);
 	}
\ No newline at end of file
