diff --git a/old_test.java b/new_test.java
index 9cef3de..3fcf50c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,4 @@
-@Ignore
-	// re-enable test when we allow orders to be persisted when not activated and signed
-	@Test
+@Test
 	public void saveOrder_shouldNotAllowYouToChangeTheOrderNumberOfASavedOrder() throws Exception {
 		Order existing = service.getOrder(1);
 		existing.setOrderNumber("New Number");
