diff --git a/old_product.java b/new_product.java
index 3a9214d..2ce10fa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ private HttpResponse grabLog(HttpRequest request, ApplicationId applicationId, T
         if (getBindingMatch(request).groupCount() != 7)
             throw new NotFoundException("Illegal POST log request '" + request.getUri() +
                     "': Must have 6 arguments but had " + ( getBindingMatch(request).groupCount()-1 ) );
-        final String response = applicationRepository.grabLog(tenant, applicationId);
+        final String response = "";
         return new HttpResponse(200) {
             @Override
             public void render(OutputStream outputStream) throws IOException {
