diff --git a/old_test.java b/new_test.java
index ee85a6b..0122883 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	public void testProcess() throws Exception {
-		System.out.println("process");
+		logger.info("process");
 		InputStream inStream = this.getClass().getResourceAsStream("/PF00104_small.fasta");
 		assertNotNull(inStream);
 
@@ -10,7 +10,7 @@
 		inStream.close();
 
 		//Should have 282 sequences
-		//System.out.println("Expecting 283 got " + proteinSequences.size());
+		//logger.debug("Expecting 283 got " + proteinSequences.size());
 		assertEquals(proteinSequences.size() ,  283 );
 
 		int seqNum = 0;
@@ -22,8 +22,8 @@
 				assertEquals(proteinSequence.getSequenceAsString(),"-----------------FK-N----LP-LED----------------Q----ITL--IQY-----------SWM----------------------CL-SSFA------LSWRSYK---HTNSQFLYFAPDLVF-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------");
 				break;
 			case 281:
-				//System.out.println(proteinSequence.getAccession());
-				//System.out.println(proteinSequence.getSequenceAsString());
+				//logger.debug("Get Accession: {}", proteinSequence.getAccession());
+				//logger.debug("Get Protein Sequence: {}", proteinSequence.getSequenceAsString());
 				assertEquals(proteinSequence.getAccession().getID(),"Q9PU76_CRONI/141-323");
 				assertEquals(proteinSequence.getSequenceAsString(),"VETVTELTEFAKSI-PGFS-N----LD-LND----------------Q----VTL--LKY-----------GVY----------------------EA-IFAM------LASVMNK---DGMPVAYGNGFITRE------------------------------------------------------------------------------------------------------------------------------------------------------------FLKSLRKPFCDIMEPKFDFA-MKF-NSL-E-LDDSDI--------------------SLFVA-AIIC-CGDRPG-------------------------------------------LVNV--GHIEKMQESIVHVLKL-H-----LQN---------NH---PD----------------------------DI------F--------LFP-KLLQKMAD-LRQLV-----------------TEH-AQLV--QIIKK---TESDAHLHPLL-------QEI---");
 				break;
