diff --git a/old_test.java b/new_test.java
index f1bb5cc..d107f21 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
         map.put("a13", 123);
         map.put("a2", 345);
 
-        Map<String, Integer> sort = sort(
+        Map<String, Integer> sort = sortMap(
                         map,
                         new PropertyComparator<Map.Entry<String, Integer>>("key", new RegexGroupNumberComparator("a(\\d*)")));
         assertThat(sort.keySet(), contains("a2", "a8", "a13"));
