diff --git a/old_product.java b/new_product.java
index be1140f..566cfca 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,15 +1,15 @@
-private void include (LineFilament pivot,
+private void include (StaffFilament pivot,
                           int pivotPos)
     {
         if (logger.isDebugEnabled() || pivot.isVip()) {
             logger.info("VIP {} include pivot:{} at pos:{}", this, pivot.getId(), pivotPos);
 
             if (pivot.isVip()) {
-                setVip();
+                setVip(true);
             }
         }
 
-        LineFilament ancestor = (LineFilament) pivot.getAncestor();
+        StaffFilament ancestor = (StaffFilament) pivot.getAncestor();
 
         // Loop on all combs that involve this filament
         // Use a copy to avoid concurrent modification error
@@ -27,23 +27,22 @@ private void include (LineFilament pivot,
 
             // Dispatch content of comb to proper lines
             for (int i = 0; i < comb.getCount(); i++) {
-                LineFilament fil = (LineFilament) comb.getFilament(i).getAncestor();
+                StaffFilament fil = (StaffFilament) comb.getFilament(i).getAncestor();
                 LineCluster cluster = fil.getCluster();
 
                 if (cluster == null) {
                     int pos = i + deltaPos;
-                    FilamentLine line = lines.get(pos);
+                    StaffFilament line = lines.get(pos);
 
                     if (line == null) {
-                        line = new FilamentLine(null);
-                        lines.put(pos, line);
+                        lines.put(pos, fil);
+                    } else {
+                        line.include(fil);
                     }
 
-                    line.add(fil);
-
                     if (fil.isVip()) {
                         logger.info("VIP adding {} to {} at pos {}", fil, this, pos);
-                        setVip();
+                        setVip(true);
                     }
 
                     fil.setCluster(this, pos);
