diff --git a/old_product.java b/new_product.java
index 7dd5a6e..6f770eb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 public void validate(Object obj, Errors errors) {
 		PersonAttributeType patObj = (PersonAttributeType) obj;
 		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "name", "PersonAttributeType.error.nameEmpty");
+		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "format", "PersonAttributeType.error.formatEmpty");
 		PersonService ps = Context.getPersonService();
 		PersonAttributeType pat = ps.getPersonAttributeTypeByName(patObj.getName());
 		if (pat != null && !pat.getUuid().equals(patObj.getUuid())) {
