diff --git a/old_test.java b/new_test.java
index 70e3b66..cece218 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,7 @@
 	public void validate_shouldPassValidationIfAllFieldsAreCorreect() throws Exception {
 		PersonAttributeType type = new PersonAttributeType();
 		type.setName("Zodiac");
+		type.setFormat("java.lang.String");
 		type.setDescription("Zodiac Description");
 		
 		Errors errors = new BindException(type, "patObj");
