diff --git a/old_product.java b/new_product.java
index a7f42f8..8ec8001 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,7 +14,7 @@ public boolean isConnectionHandleAlive(ConnectionHandle connection) {
 				rs = connection.getMetaData().getTables( null, null, KEEPALIVEMETADATA, METADATATABLE );
 			} else {
 				stmt = connection.createStatement();
-				rs = stmt.executeQuery(testStatement);
+				stmt.execute(testStatement);
 			}
 
 
