diff --git a/old_product.java b/new_product.java
index 759b1b3..e8ceb57 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,9 @@
 public static void fullViewLeft(CameraPinholeBrown paramLeft,
-									DMatrixRMaj rectifyLeft, DMatrixRMaj rectifyRight,
-									DMatrixRMaj rectifyK)
+									@Nullable DMatrixRMaj rectifiedR, DMatrixRMaj rectifyLeft, DMatrixRMaj rectifyRight,
+									DMatrixRMaj rectifyK, @Nullable ImageDimension rectifiedSize)
 	{
-		ImplRectifyImageOps_F64.fullViewLeft(paramLeft, rectifyLeft, rectifyRight, rectifyK);
+		if( rectifiedSize == null )
+			rectifiedSize = new ImageDimension();
+
+		ImplRectifyImageOps_F64.fullViewLeft(paramLeft,rectifiedR, rectifyLeft, rectifyRight, rectifyK,rectifiedSize);
 	}
\ No newline at end of file
