diff --git a/old_test.java b/new_test.java
index 3bc770c..2a55840 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,10 +9,10 @@
 		FMatrixRMaj rect2 = CommonOps_FDRM.identity(3);
 		FMatrixRMaj rectK = PerspectiveOps.pinholeToMatrix(param, (FMatrixRMaj)null);
 
-		RectifyImageOps.fullViewLeft(param,rect1,rect2,rectK);
+		RectifyImageOps.fullViewLeft(param,null,rect1,rect2,rectK,null);
 
 		// check left image
 		Point2Transform2_F32 tran = RectifyImageOps.transformPixelToRect(param, rect1);
-		checkInside(tran);
+		checkInside(tran,width,height);
 		// the right view is not checked since it is not part of the contract
 	}
\ No newline at end of file
