diff --git a/old_product.java b/new_product.java
index 80aee40..5d5c6c9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,5 @@
-public UpdateRequest script(String script, ScriptService.ScriptType scriptType, @Nullable Map<String, Object> scriptParams) {
-        this.script = script;
-        this.scriptType = scriptType;
-        if (this.scriptParams != null) {
-            this.scriptParams.putAll(scriptParams);
-        } else {
-            this.scriptParams = scriptParams;
-        }
+@Deprecated
+    public UpdateRequest script(String script, ScriptService.ScriptType scriptType, @Nullable Map<String, Object> scriptParams) {
+        this.script = new Script(script, scriptType, null, scriptParams);
         return this;
     }
\ No newline at end of file
