diff --git a/old_test.java b/new_test.java
index fbc26c4..687085a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,9 @@
         }
         PolyglotEngine.Value apply = findGlobalSymbol(id);
 
-        ObjectWithValueInterface object = JavaInterop.asJavaObject(ObjectWithValueInterface.class, (TruffleObject) apply.execute().get());
+        TruffleObject truffleObject = (TruffleObject) apply.execute().get();
+        assertIsObjectOfLanguage(truffleObject);
+        ObjectWithValueInterface object = JavaInterop.asJavaObject(ObjectWithValueInterface.class, truffleObject);
         object.add(20.0);
         object.add(22.0);
 
