diff --git a/old_product.java b/new_product.java
index 8ac4393..c0d92d1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
 protected String objectWithValueAndAddProperty() {
-        throw new UnsupportedOperationException("implement objectWithValueProperty() method");
+        throw new UnsupportedOperationException("implement objectWithValueAndAddProperty() method");
     }
\ No newline at end of file
