diff --git a/old_test.java b/new_test.java
index df7eafe..d0ad963 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
     public void testClearAssociation() {
         IsisContext.getTransactionManager().startTransaction();
-        ClearValueRequest request = new ClearValueRequest(session, "name", movieData);
-		ClearValueResponse response = server.clearValue(request );
-		final ObjectData[] updatesData = response.getUpdates();
+        final ClearValueRequest request = new ClearValueRequest(session, "name", movieData);
+        final ClearValueResponse response = server.clearValue(request);
+        final ObjectData[] updatesData = response.getUpdates();
         IsisContext.getTransactionManager().endTransaction();
 
         nameField.assertFieldEmpty(object);
