diff --git a/old_product.java b/new_product.java
index 1520140..384e7e1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
 @Override
     public ClearAssociationResponse clearAssociation(final ClearAssociationRequest request) {
 
-        AuthenticationSession session = request.getSession();
-        String fieldIdentifier = request.getFieldIdentifier();
-        IdentityData targetData = request.getTarget();
-        IdentityData associateData = request.getAssociate();
+        final AuthenticationSession session = request.getSession();
+        final String fieldIdentifier = request.getFieldIdentifier();
+        final IdentityData targetData = request.getTarget();
+        final IdentityData associateData = request.getAssociate();
 
         if (LOG.isDebugEnabled()) {
             LOG.debug("request clearAssociation " + fieldIdentifier + " on " + targetData + " of " + associateData
