diff --git a/old_test.java b/new_test.java
index 9002ae4..7c48378 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,12 +2,13 @@
 	@Verifies(value = "should fail if patient program end date comes before its enrolled date", method = "validate(Object,Errors)")
 	public void validate_shouldFailIfPatientProgramEndDateComesBeforeItsEnrolledDate() throws Exception {
 		PatientProgram program = Context.getProgramWorkflowService().getPatientProgram(1);
-		Calendar c = Calendar.getInstance();
-		c.set(2014, 3, 1);//set to an old date
-		program.setDateEnrolled(new Date());
-		program.setDateCompleted(c.getTime());
+		SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy");
+		Date dateEnrolled = sdf.parse("12/04/2014");
+		Date dateCompleted = sdf.parse("21/03/2014");
+		program.setDateEnrolled(dateEnrolled);
+		program.setDateCompleted(dateCompleted);
 		
 		BindException errors = new BindException(program, "");
 		new PatientProgramValidator().validate(program, errors);
-		Assert.assertTrue(errors.hasErrors());
+		Assert.assertTrue(errors.hasFieldErrors("dateCompleted"));
 	}
\ No newline at end of file
