diff --git a/old_product.java b/new_product.java
index fd2faef..514c72d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,8 +2,6 @@ public CreateDirectoryTOptions toThrift() {
     CreateDirectoryTOptions options = new CreateDirectoryTOptions();
     options.setAllowExists(mAllowExists);
     options.setRecursive(mRecursive);
-    options.setTtl(mTtl);
-    options.setTtlAction(TtlAction.toThrift(mTtlAction));
     options.setPersisted(mWriteType.isThrough());
     if (mMode != null) {
       options.setMode(mMode.toShort());
