diff --git a/old_product.java b/new_product.java
index d913d90..69299aa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,6 +4,14 @@
 		if (includeRetired == false)
 			criteria.add(Expression.eq("retired", false));
 		List<EncounterType> results = criteria.list();
-		Collections.sort(results, new MetadataNameComparator(true));
+		
+		// do java sorting on the return value of "getName()",
+		// because maybe both unlocalized and localized encounterTypes are in "results" list
+		Collections.sort(results, new Comparator<EncounterType>() {
+			@Override
+			public int compare(EncounterType left, EncounterType right) {
+				return left.getName().compareTo(right.getName());
+			}
+		});
 		return results;
 	}
\ No newline at end of file
