diff --git a/old_test.java b/new_test.java
index 09cbcea..98e81cf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
         DatabaseConfiguration dbConf = TestUtils.getDBConfig();
         
         
-        Map<String, TypeInference.DataType> headerDataTypes = new HashMap<>(); 
+        LinkedHashMap<String, TypeInference.DataType> headerDataTypes = new LinkedHashMap<>(); 
         headerDataTypes.put("city", TypeInference.DataType.CATEGORICAL);
         headerDataTypes.put("temperature", TypeInference.DataType.NUMERICAL);
         headerDataTypes.put("is_sunny", TypeInference.DataType.BOOLEAN);
