diff --git a/old_product.java b/new_product.java
index c07a61c..739306f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static Dataframe parseCSVFile(Reader reader, String yVariable, Map<String, TypeInference.DataType> headerDataTypes, 
+public static Dataframe parseCSVFile(Reader reader, String yVariable, LinkedHashMap<String, TypeInference.DataType> headerDataTypes, 
                                            char delimiter, char quote, String recordSeparator, Long skip, Long limit, DatabaseConfiguration dbConf) {
             Logger logger = LoggerFactory.getLogger(Dataframe.Builder.class);
             
