diff --git a/old_test.java b/new_test.java
index 09f8de6..406dcd8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,27 +1,25 @@
 @Test
     public void testWrite1() throws IOException {
-        ByteArrayOutputStream stream  = new ByteArrayOutputStream();
-        DocumentOutputStream  dstream = new DocumentOutputStream(stream, 25);
+        final byte[] expected = data(25);
 
-        for (int j = 0; j < 25; j++)
-        {
-            dstream.write(j);
-        }
-        try
-        {
-            dstream.write(0);
-            fail("Should have caught IOException");
-        }
-        catch (IOException ignored)
-        {
-        }
-        byte[] output = stream.toByteArray();
+        POIFSWriterListener l = (event) -> {
+            DocumentOutputStream dstream = event.getStream();
 
-        assertEquals(25, output.length);
-        for (int j = 0; j < 25; j++)
-        {
-            assertEquals(( byte ) j, output[ j ]);
-        }
-        dstream.close();
-        stream.close();
+            try {
+                for (byte b : expected) {
+                    dstream.write((int)b);
+                }
+            } catch (IOException ignored) {
+                fail("stream exhausted too early");
+            }
+
+            try {
+                dstream.write(0);
+                fail("Should have caught IOException");
+            }
+            catch (IOException ignored) {
+            }
+        };
+
+        compare(l, expected);
     }
\ No newline at end of file
