diff --git a/old_test.java b/new_test.java
index a7a7022..86c52e7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,16 +12,16 @@
     createEmptyFile(testDirNonEmptyChildFile);
     createEmptyFile(testDirNonEmptyChildDirFile);
     mUfs.delete(testDirEmpty, false);
-    Assert.assertFalse(mUfs.directoryExists(testDirEmpty));
+    Assert.assertFalse(mUfs.isDirectory(testDirEmpty));
     try {
       mUfs.delete(testDirNonEmpty, false);
     } catch (IOException e) {
       // Some File systems may throw IOException
     }
-    Assert.assertTrue(mUfs.directoryExists(testDirNonEmpty));
+    Assert.assertTrue(mUfs.isDirectory(testDirNonEmpty));
     mUfs.delete(testDirNonEmpty, true);
-    Assert.assertFalse(mUfs.directoryExists(testDirNonEmpty));
-    Assert.assertFalse(mUfs.directoryExists(testDirNonEmptyChildDir));
-    Assert.assertFalse(mUfs.fileExists(testDirNonEmptyChildFile));
-    Assert.assertFalse(mUfs.fileExists(testDirNonEmptyChildDirFile));
+    Assert.assertFalse(mUfs.isDirectory(testDirNonEmpty));
+    Assert.assertFalse(mUfs.isDirectory(testDirNonEmptyChildDir));
+    Assert.assertFalse(mUfs.isFile(testDirNonEmptyChildFile));
+    Assert.assertFalse(mUfs.isFile(testDirNonEmptyChildDirFile));
   }
\ No newline at end of file
