diff --git a/old_product.java b/new_product.java
index 0981be2..3c85f96 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public static void deleteDir(final String path) throws IOException {
     UnderFileSystem ufs = UnderFileSystem.get(path);
 
-    if (ufs.directoryExists(path) && !ufs.delete(path, true)) {
+    if (ufs.isDirectory(path) && !ufs.delete(path, true)) {
       throw new IOException("Folder " + path + " already exists but can not be deleted.");
     }
   }
\ No newline at end of file
