diff --git a/old_product.java b/new_product.java
index 50e9166..db1d643 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery sizeLt(String propName, int size) {
-        criterion = criterion.and(criterionBuilder.sizeLt(propName, size));
+        criterion = criterion.and(Criteria.sizeLt(propName, size));
         return this;
     }
\ No newline at end of file
