diff --git a/old_test.java b/new_test.java
index a7099f2..65b5c06 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
     public void test_withNanoOfSecond_normal() {
         OffsetTime base = OffsetTime.of(11, 30, 59, 1, OFFSET_PONE);
-        OffsetTime test = base.withNanoOfSecond(15);
+        OffsetTime test = base.withNano(15);
         assertEquals(test, OffsetTime.of(11, 30, 59, 15, OFFSET_PONE));
     }
\ No newline at end of file
