diff --git a/old_test.java b/new_test.java
index 92db031..e7bbb1e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
         File html = new File("src/test/resources/document_conversion/html-with-extra-content-input.htm");
 
         mockServer.when(
-                request().withMethod("POST").withPath(DOCUMENTS_PATH)
+                request().withMethod("POST").withPath(DocumentConversion.DOCUMENTS_PATH)
         ).respond((response().withBody(GsonSingleton.getGson().toJson(response))));
 
         // Call upload document
