diff --git a/old_product.java b/new_product.java
index 14cb43c..dbcd9b5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,7 @@ public Document uploadDocument(final File document, final String mediaType) {
         try {
             MultipartEntity reqEntity = new MultipartEntity();
             reqEntity.addPart("file", new FileBody(document, mediaType));
-            HttpRequestBase request = Request.Post("/v1/documents")
+            HttpRequestBase request = Request.Post(DocumentConversion.DOCUMENTS_PATH)
                                              .withEntity(reqEntity).build();
 
             HttpResponse response = docConversionService.execute(request);
