diff --git a/old_product.java b/new_product.java
index 4d0fed0..1f7cd27 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public void setDescription(String n) {
-        setMetadataSingleValue(MetadataSchema.DC_SCHEMA, "description", null, null, n);
+public void setDescription(Context context, String n) throws SQLException {
+        getBitstreamService().setMetadataSingleValue(context, this, MetadataSchema.DC_SCHEMA, "description", null, null, n);
     }
\ No newline at end of file
