diff --git a/old_test.java b/new_test.java
index f9b6c1e..d4391c3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
-    public void testSetDescription()
+    public void testSetDescription() throws SQLException
     {
         String description = "new description";
-        bs.setDescription(description);
+        bs.setDescription(context, description);
         assertThat("testSetDescription 0", bs.getDescription(), notNullValue());
         assertThat("testSetDescription 1", bs.getDescription(), not(equalTo("")));
         assertThat("testSetDescription 2", bs.getDescription(), equalTo(description));
