diff --git a/old_product.java b/new_product.java
index 9c28f6c..8c7f5cc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery ltProp(String propName, String otherProp) {
-        addCriterion(criterionBuilder.ltProp(propName, otherProp));
+    	criterion = criterion.and(criterionBuilder.ltProp(propName, otherProp));
         return this;
     }
\ No newline at end of file
