diff --git a/old_test.java b/new_test.java
index 68e81e9..86010f1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testLtProp() {
-        assertTrue(instance.ltProp("id", "name").getQueryCriterions()
-                .contains(new LtPropCriterion("id", "name")));
+    	assertEquals(new LtPropCriterion("id", "name"), instance.ltProp("id", "name").getQueryCriterion());
     }
\ No newline at end of file
