diff --git a/old_test.java b/new_test.java
index 0c8852d..697711e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,13 @@
 @Test
         public void testError2() {
             // we are using synchronous execution to test this exactly rather than non-deterministic concurrent behavior
-            final IObservable<String> o1 = new TestErrorWatchable("one", "two", "three");
-            final IObservable<String> o2 = new TestErrorWatchable("four", null, "six"); // we expect to lose "six"
-            final IObservable<String> o3 = new TestErrorWatchable("seven", "eight", null);// we expect to lose all of these since o2 is done first and fails
-            final IObservable<String> o4 = new TestErrorWatchable("nine");// we expect to lose all of these since o2 is done first and fails
+            final Observable<String> o1 = new TestErrorObservable("one", "two", "three");
+            final Observable<String> o2 = new TestErrorObservable("four", null, "six"); // we expect to lose "six"
+            final Observable<String> o3 = new TestErrorObservable("seven", "eight", null);// we expect to lose all of these since o2 is done first and fails
+            final Observable<String> o4 = new TestErrorObservable("nine");// we expect to lose all of these since o2 is done first and fails
 
             @SuppressWarnings("unchecked")
-            IObservable<String> m = merge(o1, o2, o3, o4);
+            Observable<String> m = merge(o1, o2, o3, o4);
             m.subscribe(stringObserver);
 
             verify(stringObserver, times(1)).onError(any(NullPointerException.class));
