diff --git a/old_product.java b/new_product.java
index 5448a00..ad9e640 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-synchronized void error(CombineWatcher<R, ?> w, Exception e) {
-            watcher.onError(e);
-            /* tell ourselves to stop processing onNext events, event if the watchers don't obey the unsubscribe we're about to send */
+synchronized void error(CombineObserver<R, ?> w, Exception e) {
+            Observer.onError(e);
+            /* tell ourselves to stop processing onNext events, event if the Observers don't obey the unsubscribe we're about to send */
             running.set(false);
-            /* tell all watchers to unsubscribe since we had an error */
+            /* tell all Observers to unsubscribe since we had an error */
             stop();
         }
\ No newline at end of file
