diff --git a/old_product.java b/new_product.java
index 56a193b..fe0ef3b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,8 @@
 public static CompletedCheckpoint loadAndValidateSavepoint(
 			JobID jobId,
 			Map<JobVertexID, ExecutionJobVertex> tasks,
-			String savepointPath) throws IOException {
+			String savepointPath,
+			boolean allowNonRestoredState) throws IOException {
 
 		// (1) load the savepoint
 		Savepoint savepoint = SavepointStore.loadSavepoint(savepointPath);
@@ -28,11 +29,14 @@ public static CompletedCheckpoint loadAndValidateSavepoint(
 
 					throw new IllegalStateException(msg);
 				}
+			} else if (allowNonRestoredState) {
+				LOG.info("Skipping savepoint state for operator {}.", taskState.getJobVertexID());
 			} else {
 				String msg = String.format("Failed to rollback to savepoint %s. " +
-								"Cannot map old state for task %s to the new program. " +
-								"This indicates that the program has been changed in a " +
-								"non-compatible way  after the savepoint.",
+								"Cannot map savepoint state for operator %s to the new program, " +
+								"because the operator is not available in the new program. If " +
+								"you want to allow to skip this, you can set the --allowNonRestoredState " +
+								"option on the CLI.",
 						savepointPath, taskState.getJobVertexID());
 				throw new IllegalStateException(msg);
 			}
