diff --git a/old_test.java b/new_test.java
index b56f47c..e75c640 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,16 +7,16 @@
             .connectionString(getConnectionString())
             .buildAsyncClient();
         final EventHubProducerAsyncClient producer = asyncClient.createProducer();
-        final EventHubConsumer receiver = client.createConsumer(EventHubAsyncClient.DEFAULT_CONSUMER_GROUP_NAME,
-            PARTITION_ID, EventPosition.earliest());
+        final EventHubConsumerClient receiver = client.createConsumer(EventHubClientBuilder.DEFAULT_CONSUMER_GROUP_NAME,
+            EventPosition.earliest());
 
         producer.send(TestUtils.getEvents(numberOfEvents, messageId), sendOptions).block();
 
         // Act
-        final IterableStream<EventData> receive = receiver.receive(15, Duration.ofSeconds(30));
+        final IterableStream<PartitionEvent> receive = receiver.receive(PARTITION_ID, 15, Duration.ofSeconds(30));
 
         // Assert
         Assert.assertNotNull(receive);
-        final List<EventData> results = receive.stream().collect(Collectors.toList());
+        final List<PartitionEvent> results = receive.stream().collect(Collectors.toList());
         Assert.assertEquals(numberOfEvents, results.size());
     }
\ No newline at end of file
