diff --git a/old_product.java b/new_product.java
index 7d80504..98f80a5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,7 +9,7 @@ public static <K, V, C extends Serializable> void initContext(
         int retries,
         int interval,
         boolean isKeepBinary,
-        DeployContext deployContext) {
+        DeployingContext deployingContext) {
 
         affinityCallWithRetries(ignite, Arrays.asList(datasetCacheName, upstreamCacheName), part -> {
             Ignite locIgnite = Ignition.localIgnite();
@@ -51,5 +51,5 @@ public static <K, V, C extends Serializable> void initContext(
             datasetCache.put(part, ctx);
 
             return part;
-        }, retries, interval, deployContext);
+        }, retries, interval, deployingContext);
     }
\ No newline at end of file
