diff --git a/old_test.java b/new_test.java
index d72fd49..b7802ad 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testGe() {
-        assertTrue(instance.ge("id", 5).getQueryCriterions()
-                .contains(new GeCriterion("id", 5)));
+    	assertEquals(new GeCriterion("id", 5), instance.ge("id", 5).getQueryCriterion());
     }
\ No newline at end of file
