diff --git a/old_product.java b/new_product.java
index 94b5935..7caabd9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -25,7 +25,6 @@
         commandEntity.setDescription(updateCommand.getDescription());
         commandEntity.setExecutable(updateCommand.getExecutable());
         commandEntity.setSetupFile(updateCommand.getSetupFile());
-        commandEntity.setJobType(updateCommand.getJobType());
         commandEntity.setConfigs(updateCommand.getConfigs());
         commandEntity.setCommandTags(updateCommand.getTags());
         this.commandRepo.save(commandEntity);
