diff --git a/old_test.java b/new_test.java
index 438af94..27769e5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -22,7 +22,6 @@
                 .withTags(tags)
                 .withConfigs(command.getConfigs())
                 .withSetupFile(command.getSetupFile())
-                .withJobType(command.getJobType())
                 .build();
 
         this.service.updateCommand(COMMAND_1_ID, updateCommand);
