diff --git a/old_product.java b/new_product.java
index b6aae0a..fc6382b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,9 @@
-public static String getGetterForInterface(String yangName, String returnType, boolean isList) {
+public static String getGetterForInterface(String yangName, String returnType, boolean isList,
+            int generatedJavaFiles) {
 
         if (!isList) {
-            return getGetterInterfaceString(returnType, yangName);
+            return getGetterInterfaceString(returnType, yangName, generatedJavaFiles);
         }
         String listAttr = getListString() + returnType + DIAMOND_CLOSE_BRACKET;
-        return getGetterInterfaceString(listAttr, yangName);
+        return getGetterInterfaceString(listAttr, yangName, generatedJavaFiles);
     }
\ No newline at end of file
