diff --git a/old_test.java b/new_test.java
index 58c86bc..6b82ae0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test(groups={"tck"})
-    public void test_withMonthOfYear() {
-        assertEquals(MonthDay.of(6, 30).withMonthOfYear(1), MonthDay.of(1, 30));
+    public void test_withMonth() {
+        assertEquals(MonthDay.of(6, 30).withMonth(1), MonthDay.of(1, 30));
     }
\ No newline at end of file
