diff --git a/old_product.java b/new_product.java
index daf2c4b..5771c5b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public ChronoDate withMonthOfYear(int monthOfYear) {
-        return with(LocalDateTimeField.MONTH_OF_YEAR, monthOfYear);
+public ChronoDate withMonth(int month) {
+        return with(LocalDateTimeField.MONTH_OF_YEAR, month);
     }
\ No newline at end of file
