diff --git a/old_test.java b/new_test.java
index a867177..9949167 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
       new RepoManager(ctx.repo).delete("xyz", null);
       fail("Not installed package not detected.");
     } catch(final QueryException ex) {
-      check(ex, Err.PKGNOTINST);
+      check(ex, Err.PKGNOTEXIST);
     }
     // install a package without dependencies (pkg3)
     new RepoInstall(REPO + "pkg3.xar", null).execute(ctx);
