diff --git a/old_test.java b/new_test.java
index 45fcb2a..ad601ee 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,14 @@
 @Test
   public void functionTest() throws Exception {
     createColl();
-    final String doc = _DB_OPEN.args(NAME);
+    // document access after inlining
+    check("declare function local:x($d) { collection($d)//text()[. = '1'] };"
+        + "local:x('" + NAME + "')", "1");
+    check("declare function local:x($d, $s) { collection($d)//text()[. = $s] };"
+        + "local:x('" + NAME + "', '1')", "1");
+
     // text: search term must be string
+    final String doc = _DB_OPEN.args(NAME);
     check("declare function local:x() {" + doc +
         "//text()[. = '1'] }; local:x()", "1");
     check("declare function local:x($x as xs:string) {" + doc +
