diff --git a/old_product.java b/new_product.java
index 5f61336..6d149e1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 private Item function(final QueryContext ctx) throws QueryException {
-    final FItem func = checkFunc(expr[0], ctx);
-    final QNm name = func.funcName();
-    final StaticFunc sf = name == null ? null : ctx.funcs.get(name, func.arity(), null, false);
-    return new PlainDoc(ctx, info).function(name, sf, func.funcType(), func.annotations(), null);
+    final FItem fn = checkFunc(exprs[0], ctx);
+    final QNm name = fn.funcName();
+    final StaticFunc sf = name == null ? null : ctx.funcs.get(name, fn.arity(), null, false);
+    return new PlainDoc(ctx, info).function(name, sf, fn.funcType(), fn.annotations(), null);
   }
\ No newline at end of file
