diff --git a/old_test.java b/new_test.java
index 8eb8bf4..e700a81 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,9 +3,9 @@
         Assert.assertNotNull(this.c.getCommands());
         Assert.assertTrue(this.c.getCommands().isEmpty());
         final CommandEntity one = new CommandEntity();
-        one.setId("one");
+        one.setUniqueId("one");
         final CommandEntity two = new CommandEntity();
-        two.setId("two");
+        two.setUniqueId("two");
         final List<CommandEntity> commands = new ArrayList<>();
         commands.add(one);
         commands.add(two);
