diff --git a/old_product.java b/new_product.java
index 12c09de..219bea3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,6 +7,8 @@ public static SdkGenerator createSdkGenerator(SdkPlatform sdkPlatform) {
             return new CppSdkGenerator();
         case C:
             return new CSdkGenerator();
+        case OBJC:
+            return new ObjCSdkGenerator();
         default:
             return null;
         }
