diff --git a/old_test.java b/new_test.java
index c7d45ff..ea9ca65 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,6 +8,8 @@
         Assert.assertNotNull(generator);
         generator = factory.createSdkGenerator(SdkPlatform.C);
         Assert.assertNotNull(generator);
+        generator = factory.createSdkGenerator(SdkPlatform.OBJC);
+        Assert.assertNotNull(generator);
 //
 //        String profileSchema = "{\"type\":\"record\",\"name\":\"SuperProfile\",\"namespace\":\"org.kaaproject.kaa.common.endpoint.gen.test\",\"fields\":[{\"name\":\"profileBody\",\"type\":{\"type\":\"string\",\"avro.java.string\":\"String\"}}]}";
 //
