diff --git a/old_product.java b/new_product.java
index baa4887..b3ef1e5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,34 +4,12 @@
 			throw new IllegalArgumentException("Image derivatives must be set");
 
 		setAllowedBounds(feature);
-		int regionWidth = feature.radius * 2 + 1;
-		int size = regionWidth * regionWidth;
 
-		if (!isFullyInside(feature.x, feature.y))
+		if (!isFullyInside(feature.x, feature.y)) {
 			return false;
-
-		float tl_x = feature.x - feature.radius;
-		float tl_y = feature.y - feature.radius;
-
-		interpInput.region(tl_x, tl_y, feature.pixel, regionWidth, regionWidth);
-		interpDeriv.setImage(derivX);
-		interpDeriv.region(tl_x, tl_y, feature.derivX, regionWidth, regionWidth);
-		interpDeriv.setImage(derivY);
-		interpDeriv.region(tl_x, tl_y, feature.derivY, regionWidth, regionWidth);
-
-		float Gxx = 0, Gyy = 0, Gxy = 0;
-		for (int i = 0; i < size; i++) {
-			float dX = feature.derivX[i];
-			float dY = feature.derivY[i];
-
-			Gxx += dX * dX;
-			Gyy += dY * dY;
-			Gxy += dX * dY;
 		}
 
-		feature.Gxx = Gxx;
-		feature.Gyy = Gyy;
-		feature.Gxy = Gxy;
+		internalSetDescription(feature);
 
 		return true;
 	}
\ No newline at end of file
