diff --git a/old_product.java b/new_product.java
index 2ef2397..756ca44 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,7 +10,7 @@ protected ModelAndView onSubmit(HttpServletRequest request, HttpServletResponse
 			Field field = (Field) obj;
 			field = setObjects(field, request);
 			
-			if (action.equals(Context.getMessageSourceService().getMessage("general.delete"))) {
+			if (action != null && action.equals(Context.getMessageSourceService().getMessage("general.delete"))) {
 				try {
 					Context.getFormService().purgeField(field);
 				} catch (DataIntegrityViolationException e) {
