diff --git a/old_product.java b/new_product.java
index 6d712c3..14cb0dc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public void registerTimerEventInNanoSeconds(long timerInNanoSecnods, Runnable task) {
-    nioLooper.registerTimerEventInNanoSeconds(timerInNanoSecnods, task);
+public void registerTimerEvent(Duration timer, Runnable task) {
+    nioLooper.registerTimerEvent(timer, task);
   }
\ No newline at end of file
