diff --git a/old_test.java b/new_test.java
index 217fe3e..0e439b0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
   public void testHasExecution() throws IOException, ExecutorManagerException,
-    InterruptedException {
-    QueuedExecutions queue = new QueuedExecutions(2);
-    List<Pair<ExecutionReference, ExecutableFlow>> dataList = getDummyData();
+      InterruptedException {
+    final QueuedExecutions queue = new QueuedExecutions(2);
+    final List<Pair<ExecutionReference, ExecutableFlow>> dataList = getDummyData();
     queue.enqueueAll(dataList);
-    for (Pair<ExecutionReference, ExecutableFlow> pair : dataList) {
+    for (final Pair<ExecutionReference, ExecutableFlow> pair : dataList) {
       Assert.assertTrue(queue.hasExecution(pair.getFirst().getExecId()));
     }
     Assert.assertFalse(queue.hasExecution(5));
