diff --git a/old_product.java b/new_product.java
index e0f1598..3093a3a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,7 +15,7 @@
 				return;
 			}
 			if (!collectorServer.isApplicationDataAvailable(application)
-					&& req.getParameter(ACTION_PARAMETER) == null) {
+					&& HttpParameter.ACTION.getParameterFrom(req) == null) {
 				CollectorController.writeDataUnavailableForApplication(application, resp);
 				return;
 			}
