diff --git a/old_test.java b/new_test.java
index 15e252d..2ff7583 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test
 	public void testDoGet() throws ServletException, IOException {
-		doGet(Collections.<String, String> emptyMap(), true);
+		doGet(Collections.<HttpParameter, String> emptyMap(), true);
 
 		setProperty(Parameter.ALLOWED_ADDR_PATTERN, "256.*");
 		try {
-			doGet(Collections.<String, String> emptyMap(), false);
+			doGet(Collections.<HttpParameter, String> emptyMap(), false);
 			setProperty(Parameter.ALLOWED_ADDR_PATTERN, ".*");
-			doGet(Collections.<String, String> emptyMap(), false);
+			doGet(Collections.<HttpParameter, String> emptyMap(), false);
 		} finally {
 			setProperty(Parameter.ALLOWED_ADDR_PATTERN, null);
 		}
