diff --git a/old_test.java b/new_test.java
index 8efa120..264ceca 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
   public void testClearFabric() {
-    final byte[] a = { 'a' };
+    final byte[] a = "tCFa".getBytes();
     final byte[] x = { 'x' };
-    final byte[] q = "queue://q".getBytes();
-    final byte[] s = "stream://s".getBytes();
+    final byte[] q = "queue://tCF/q".getBytes();
+    final byte[] s = "stream://tCF/s".getBytes();
 
     // write to a table, a queue, and a stream
     Assert.assertTrue(remote.execute(new Write(a, x)));
