diff --git a/old_product.java b/new_product.java
index 0558b0b..27c8fa5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,6 +7,6 @@ private NettyServer getNettyServer(Properties properties)
     VerifiableProperties verifiableProperties = new VerifiableProperties(properties);
     NettyConfig nettyConfig = new NettyConfig(verifiableProperties);
     NettyMetrics nettyMetrics = new NettyMetrics(new MetricRegistry());
-    RequestResponseHandlerController requestResponseHandlerController = getRequestHandlerController(properties);
-    return new NettyServer(nettyConfig, nettyMetrics, requestResponseHandlerController);
+    RestRequestHandler requestHandler = new MockRequestResponseHandler();
+    return new NettyServer(nettyConfig, nettyMetrics, requestHandler);
   }
\ No newline at end of file
