diff --git a/old_test.java b/new_test.java
index 7401809..a307807 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,12 @@
 @Test
-  public void getNettyServerTest()
-      throws InstantiationException, IOException {
+  public void getNettyServerTest() {
     // dud properties. server should pick up defaults
     Properties properties = new Properties();
     VerifiableProperties verifiableProperties = new VerifiableProperties(properties);
-    RequestResponseHandlerController requestResponseHandlerController = getRequestHandlerController();
+    RestRequestHandler restRequestHandler = new MockRequestResponseHandler();
 
-    NioServerFactory nioServerFactory =
-        new NettyServerFactory(verifiableProperties, new MetricRegistry(), requestResponseHandlerController);
+    NettyServerFactory nioServerFactory =
+        new NettyServerFactory(verifiableProperties, new MetricRegistry(), restRequestHandler);
     NioServer nioServer = nioServerFactory.getNioServer();
     assertNotNull("No NioServer returned", nioServer);
     assertEquals("Did not receive a NettyServer instance", NettyServer.class.getCanonicalName(),
