diff --git a/old_test.java b/new_test.java
index ea25def..2ac4af9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,11 +7,11 @@
         final Cluster cluster = new Cluster(
                 CLUSTER_1_NAME,
                 CLUSTER_1_USER,
+                CLUSTER_1_VERSION,
                 ClusterStatus.OUT_OF_SERVICE,
-                CLUSTER_1_TYPE,
-                configs,
-                CLUSTER_1_VERSION
+                CLUSTER_1_TYPE
         );
+        cluster.setConfigs(configs);
         final String id = UUID.randomUUID().toString();
         cluster.setId(id);
         final Cluster created = this.service.createCluster(cluster);
