diff --git a/old_product.java b/new_product.java
index ffd4238..709de9b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@ public Cluster createCluster(final Cluster cluster)
             throw new GeniePreconditionException(
                     "No cluster entered. Unable to validate.");
         }
-        cluster.validate();
+
         final HttpRequest request = BaseGenieClient.buildRequest(
                 Verb.POST,
                 BASE_CONFIG_CLUSTER_REST_URL,
