diff --git a/old_test.java b/new_test.java
index b41c7a9..057df17 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public void testRemoveExtension() {
     final MockImpl m = new MockImpl();
     final MockExtension extension = new MockExtension();
-    m.getExtensionPoint().addExtension(MockExtension.class, extension);
+    m.getExtensionPoint().addExtension(extension);
 
     final Iterable<Extension<MockInterface>> extensions = m.getExtensionPoint().getExtensions();
     assertNotNull(extensions);
@@ -21,7 +21,7 @@
       // ignore
     }
 
-    m.getExtensionPoint().removeExtension(MockExtension.class);
+    m.getExtensionPoint().removeExtension(extension);
     assertEquals(0, m.extensionPoint.extensions.size());
 
     // extensions should be empty
