diff --git a/old_test.java b/new_test.java
index 3763e14..f8ef3e7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -46,11 +46,11 @@
 
         assertContains(log, testOut.toString(), "Cluster tag: " + newTag);
 
-        ignite.cluster().readOnly(true);
+        ignite.cluster().state(ACTIVE_READ_ONLY);
 
         awaitPartitionMapExchange();
 
-        assertTrue(ignite.cluster().readOnly());
+        assertEquals(ACTIVE_READ_ONLY, ignite.cluster().state());
 
         assertEquals(EXIT_CODE_OK, execute("--state"));
 
