diff --git a/old_test.java b/new_test.java
index f8539be..b6ec28b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,8 @@
   public void mergeHadoopConfigurationTest() {
     Configuration.defaultInit();
     org.apache.hadoop.conf.Configuration hadoopConfig = new org.apache.hadoop.conf.Configuration();
-    hadoopConfig.set(Constants.S3_ACCESS_KEY, TEST_S3_ACCCES_KEY);
-    hadoopConfig.set(Constants.S3_SECRET_KEY, TEST_S3_SECRET_KEY);
+    hadoopConfig.set(Constants.S3N_ACCESS_KEY, TEST_S3_ACCCES_KEY);
+    hadoopConfig.set(Constants.S3N_SECRET_KEY, TEST_S3_SECRET_KEY);
     hadoopConfig.set(TEST_ALLUXIO_PROPERTY, TEST_ALLUXIO_VALUE);
 
     // This hadoop config will not be loaded into Alluxio configuration.
@@ -13,7 +13,7 @@
     ConfUtils.mergeHadoopConfiguration(hadoopConfig);
     Map<String, String> after = Configuration.toMap();
     Assert.assertEquals(before.size() + 3, after.size());
-    Assert.assertEquals(TEST_S3_ACCCES_KEY, Configuration.get(Constants.S3_ACCESS_KEY));
-    Assert.assertEquals(TEST_S3_SECRET_KEY, Configuration.get(Constants.S3_SECRET_KEY));
+    Assert.assertEquals(TEST_S3_ACCCES_KEY, Configuration.get(Constants.S3N_ACCESS_KEY));
+    Assert.assertEquals(TEST_S3_SECRET_KEY, Configuration.get(Constants.S3N_SECRET_KEY));
     Assert.assertEquals(TEST_ALLUXIO_VALUE, Configuration.get(TEST_ALLUXIO_PROPERTY));
   }
\ No newline at end of file
