diff --git a/old_product.java b/new_product.java
index bcfd2b8..a094d38 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,8 +7,8 @@ public static void mergeHadoopConfiguration(org.apache.hadoop.conf.Configuration
       String propertyName = entry.getKey();
       // TODO(gene): use a better way to enumerate every Alluxio configuration parameter
       if (propertyName.startsWith("alluxio.")
-          || propertyName.equals(Constants.S3_ACCESS_KEY)
-          || propertyName.equals(Constants.S3_SECRET_KEY)
+          || propertyName.equals(Constants.S3N_ACCESS_KEY)
+          || propertyName.equals(Constants.S3N_SECRET_KEY)
           || propertyName.equals(Constants.GCS_ACCESS_KEY)
           || propertyName.equals(Constants.GCS_SECRET_KEY)
           || propertyName.equals(Constants.SWIFT_API_KEY)
