diff --git a/old_product.java b/new_product.java
index 2762a8e..c029ea6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
 public void add(T toAdd)
   {
-    if (count == bufferlen) {
-      throw overflow;
+    if (head - tail <= buffermask) {
+      buffer[head++ & buffermask] = toAdd;
+      return;
     }
 
-    buffer[head++ % bufferlen] = toAdd;
-    count++;
+    throw overflow;
   }
\ No newline at end of file
