diff --git a/old_test.java b/new_test.java
index ed9bd90..b7c09d6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,6 +8,8 @@
 		final HashMap<InstanceType, InstanceTypeDescription> availableInstances = new HashMap<InstanceType, InstanceTypeDescription>();
 		availableInstances.put(type, desc);
 
+		final DeploymentManager deploymentManager = new TestDeploymentManager();
+
 		try {
 			whenNew(HashMap.class).withNoArguments().thenReturn(requiredInstanceTypes);
 		} catch (Exception e) {
@@ -15,10 +17,13 @@
 		}
 
 		when(this.executionGraph.getNumberOfStages()).thenReturn(1);
+		when(this.executionGraph.getStage(0)).thenReturn(this.stage1);
+		when(this.executionGraph.getCurrentExecutionStage()).thenReturn(this.stage1);
 		when(this.instanceManager.getMapOfAvailableInstanceTypes()).thenReturn(availableInstances);
+		when(this.stage1.getExecutionGraph()).thenReturn(this.executionGraph);
 
 		// correct walk through method
-		final QueueScheduler toTest = new QueueScheduler(this.instanceManager);
+		final QueueScheduler toTest = new QueueScheduler(deploymentManager, this.instanceManager);
 		try {
 			toTest.schedulJob(this.executionGraph);
 			final Deque<ExecutionGraph> jobQueue = Whitebox.getInternalState(toTest, "jobQueue");
