diff --git a/old_product.java b/new_product.java
index 4ecf8c8..8c9cbc1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -46,6 +46,17 @@
 			vertex.getEnvironment().registerExecutionListener(new QueueExecutionListener(this, vertex));
 			vertex.setExecutionState(ExecutionState.SCHEDULED);
 		}
+		
+		// Register the scheduler as an execution stage listener
+		executionGraph.registerExecutionStageListener(this);
+
+		// Request resources for the first stage of the job
+		final ExecutionStage executionStage = executionGraph.getCurrentExecutionStage();
+		try {
+			requestInstances(executionStage);
+		} catch (InstanceException e) {
+			throw new SchedulingException(StringUtils.stringifyException(e));
+		}
 
 		synchronized (this.jobQueue) {
 			this.jobQueue.add(executionGraph);
