diff --git a/old_product.java b/new_product.java
index e70315c..181ccf5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,3 @@
 public static int getFirstIntValue(String line) {
-        String[] split = line.split("=");
-        if (split.length < 2) {
-            return 0;
-        }
-        return parseIntOrDefault(split[1].trim().split("\\s+")[0], 0);
+        return getNthIntValue(line, 1);
     }
\ No newline at end of file
