diff --git a/old_product.java b/new_product.java
index 481bc93..a5ac908 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,12 @@
-@SuppressWarnings("unchecked")
-	public EncounterType getEncounterType(String name) throws DAOException {
-		return HibernateUtil.findMetadataExactlyInLocalizedColumn(name, "name", EncounterType.class, sessionFactory);
+public EncounterType getEncounterType(String name) throws DAOException {
+		Criteria crit = sessionFactory.getCurrentSession().createCriteria(EncounterType.class);
+		crit.add(Expression.eq("retired", false));
+		crit.add(Expression.sql("name = ?", LocalizedStringUtil.escapeDelimiter(name), Hibernate.STRING));
+		EncounterType encounterType = (EncounterType) crit.uniqueResult();
+		
+		if (encounterType == null) //search in those localized encounterTypes
+			encounterType = HibernateUtil.getUniqueMetadataByLocalizedColumn(name, "name", false, EncounterType.class,
+			    sessionFactory);
+		
+		return encounterType;
 	}
\ No newline at end of file
