diff --git a/old_product.java b/new_product.java
index c67d795..604ea98 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,10 +7,6 @@ public void validate(Object obj, Errors errors) throws APIException {
 		
 		Provider provider = (Provider) obj;
 		
-		if (StringUtils.isBlank(provider.getIdentifier())) {
-			errors.rejectValue("identifier", "Provider.error.identifier.required");
-		}
-		
 		if (provider.getPerson() == null && StringUtils.isBlank(provider.getName())) {
 			errors.rejectValue("name", "Provider.error.personOrName.required");
 		}
