diff --git a/old_test.java b/new_test.java
index 1745df7..cc328ad 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test
-	public void validate_shouldBeInvalidIfIdentifierIsNotSet() throws Exception {
+	public void validate_shouldBeValidIfIdentifierIsNotSet() throws Exception {
 		//given
 		provider.setIdentifier(null);
+		provider.setName("bcj");
 		
 		//when
 		providerValidator.validate(provider, errors);
 		
 		//then
-		Assert.assertTrue(errors.hasFieldErrors("identifier"));
-		Assert.assertEquals("Provider.error.identifier.required", errors.getFieldError("identifier").getCode());
+		Assert.assertFalse(errors.hasErrors());
 	}
\ No newline at end of file
