diff --git a/old_product.java b/new_product.java
index bb1a4a8..d2359d2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 @ServiceMethod(returns = ReturnType.COLLECTION)
-    public IterableResponse<String> getPartitionIds() {
-        return new IterableResponse<>(client.getPartitionIds());
+    public IterableStream<String> getPartitionIds() {
+        return new IterableStream<>(client.getPartitionIds());
     }
\ No newline at end of file
