diff --git a/old_test.java b/new_test.java
index 4472257..5a12a03 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void getPartitionIds() {
         // Act
-        final IterableResponse<String> response = client.getPartitionIds();
+        final IterableStream<String> response = client.getPartitionIds();
 
         // Assert
         Assert.assertNotNull(response);
