diff --git a/old_product.java b/new_product.java
index 5de8bc4..743c7e5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,8 +15,20 @@ public List<Point2D_I32> process( List<Point2D_I32> contour ) {
 		corners.add( contour.get(corner1));
 
 		// find points which maximize the inlier to a line model and are not close to existing points
-		findCorner(corner0,corner1,contour,corners);
-		findCorner(corner1,corner0,contour,corners);
+		int corner2 = findCorner(corner0,corner1,contour,corners);
+		int corner3 = findCorner(corner1,corner0,contour,corners);
+
+		// refine the corner estimates by maximizing the acute angle
+		corner0 = refineCorner(corner0, refinementRadius,contour);
+		corner1 = refineCorner(corner1, refinementRadius,contour);
+		corner2 = refineCorner(corner2, refinementRadius,contour);
+		corner3 = refineCorner(corner3, refinementRadius,contour);
+
+		corners.clear();
+		corners.add(contour.get(corner0));
+		corners.add(contour.get(corner1));
+		corners.add(contour.get(corner2));
+		corners.add(contour.get(corner3));
 
 		// sort the corners to make future calculations easier
 		sortByAngleCCW(center, corners);
