diff --git a/old_test.java b/new_test.java
index aee5238..ec8c10a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,5 +6,5 @@
         assertEquals(b.compareTo(a) > 0, true);
         assertEquals(a.compareTo(a) == 0, true);
         assertEquals(b.compareTo(b) == 0, true);
-        assertEquals(DATE.atTime(a).toInstant().compareTo(DATE.atTime(b).toInstant()) < 0, true);
+        assertEquals(convertInstant(a).compareTo(convertInstant(b)) < 0, true);
     }
\ No newline at end of file
