diff --git a/old_product.java b/new_product.java
index cd36c81..6ada3af 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,17 @@
 @Override
     public int compareTo(ChronoZonedDateTime<?> other) {
-        int cmp = getOffsetDateTime().compareTo(other.getOffsetDateTime());
+        int cmp = Long.compare(toEpochSecond(), other.toEpochSecond());
         if (cmp == 0) {
-            cmp = getZone().getId().compareTo(other.getZone().getId());
+            cmp = getTime().getNano() - other.getTime().getNano();
+            if (cmp == 0) {
+                cmp = getDateTime().compareTo(other.getDateTime());
+                if (cmp == 0) {
+                    cmp = getZone().getId().compareTo(other.getZone().getId());
+                    if (cmp == 0) {
+                        cmp = getDate().getChrono().compareTo(other.getDate().getChrono());
+                    }
+                }
+            }
         }
         return cmp;
     }
\ No newline at end of file
