diff --git a/old_product.java b/new_product.java
index 1a77a7e..afffaa2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
-@Theory
-    public void noProxySelected(@FromDataPoints("configurations") ProxyConfiguration configuration) {
+@ParameterizedTest
+    @MethodSource("getProxyConfigurations")
+    public void noProxySelected(ProxyConfiguration configuration) {
         // Arrange
         final String hostname = "foo.eventhubs.azure.com";
         when(proxySelector.select(argThat(u -> u.getHost().equals(hostname))))
@@ -10,6 +11,6 @@
             TransportType.AMQP_WEB_SOCKETS, configuration);
 
         // Act and Assert
-        Assert.assertEquals(PROXY_ADDRESS.getHostName(), handler.getHostname());
-        Assert.assertEquals(PROXY_ADDRESS.getPort(), handler.getProtocolPort());
+        Assertions.assertEquals(PROXY_ADDRESS.getHostName(), handler.getHostname());
+        Assertions.assertEquals(PROXY_ADDRESS.getPort(), handler.getProtocolPort());
     }
\ No newline at end of file
