diff --git a/old_test.java b/new_test.java
index 68677b3..c3f8805 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,6 +8,6 @@
             PROXY_CONFIGURATION);
 
         // Act and Assert
-        Assert.assertEquals(PROXY_ADDRESS.getHostName(), handler.getHostname());
-        Assert.assertEquals(PROXY_ADDRESS.getPort(), handler.getProtocolPort());
+        Assertions.assertEquals(PROXY_ADDRESS.getHostName(), handler.getHostname());
+        Assertions.assertEquals(PROXY_ADDRESS.getPort(), handler.getProtocolPort());
     }
\ No newline at end of file
