diff --git a/old_product.java b/new_product.java
index c944aaa..4b9a26b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,11 @@
-protected void putConnectionBackInPartition(ConnectionHandle connectionHandle) {
-		LinkedTransferQueue<ConnectionHandle> queue = connectionHandle.getOriginatingPartition().getFreeConnections();
+protected void putConnectionBackInPartition(ConnectionHandle connectionHandle) throws SQLException {
+		BoundedLinkedTransferQueue<ConnectionHandle> queue = connectionHandle.getOriginatingPartition().getFreeConnections();
 
 		if (!queue.tryTransfer(connectionHandle)){
-			queue.put(connectionHandle);
+			if (!queue.tryPut(connectionHandle)){
+				connectionHandle.internalClose();
+			}
 		}
 
-		connectionHandle.getOriginatingPartition().getAvailableConnections().incrementAndGet();
 
 	}
\ No newline at end of file
