diff --git a/old_test.java b/new_test.java
index de5e9af..173413a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,9 +7,9 @@
         int qualityThreshold = 0;
         int maxLevels = 1000;
         */
-        CloseableIterator<Alignment> iter = getAlignments();
+        AlignmentInterval interval = getAlignmentInterval();
 
-        Map<String, List<Row>> result = (new AlignmentPacker()).packAlignments(iter, ranges,
+        Map<String, List<Row>> result = (new AlignmentPacker()).packAlignments(Arrays.asList(interval),
                 new AlignmentTrack.RenderOptions());
         assertEquals(1, result.size());
         for (List<Row> alignmentrows : result.values()) {
