diff --git a/old_test.java b/new_test.java
index 4d617a1..a0d1983 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
   public void list() {
     query(_FILE_WRITE.args(PATH1, "abcd"));
-    error(_FILE_LIST.args(PATH1), Err.FILE_NO_DIR);
-    error(_FILE_LIST.args(PATH1 + NAME), Err.FILE_NOT_FOUND);
+    error(_FILE_LIST.args(PATH1), Err.FILE_NO_DIR_X);
+    error(_FILE_LIST.args(PATH1 + NAME), Err.FILE_NOT_FOUND_X);
     query(_FILE_WRITE.args(PATH1, "()"));
-    error(_FILE_LIST.args(PATH1), Err.FILE_NO_DIR);
+    error(_FILE_LIST.args(PATH1), Err.FILE_NO_DIR_X);
     query(_FILE_LIST.args(PATH), NAME);
     contains(_FILE_LIST.args(PATH, "false()"), NAME);
     contains(_FILE_LIST.args(PATH, "false()", NAME), NAME);
