diff --git a/old_product.java b/new_product.java
index 82a7be8..bdb5850 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@ private Iter list(final QueryContext qc) throws QueryException {
       for(final String s : qc.context.databases.listDBs()) tl.add(s);
     } else {
       final Data data = checkData(qc);
-      final String path = string(el == 1 ? Token.EMPTY : checkStr(exprs[1], qc));
+      final String path = string(el == 1 ? Token.EMPTY : toToken(exprs[1], qc));
       // add xml resources
       final Resources res = data.resources;
       final IntList il = res.docs(path);
