diff --git a/old_test.java b/new_test.java
index e9b9e27..7cf37d3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -38,13 +38,4 @@
         // make sure assertions were propagated
         assertEquals( 1,
                       sink.getAsserted().size() );
-
-        RuleRemovalContext removalContext = new RuleRemovalContext( mock( Rule.class ));
-        InternalWorkingMemory[] workingMemories = new InternalWorkingMemory[]{this.workingMemory};
-
-        // This use to throw ClassCastException JBRULES-1719
-        node.remove( removalContext,
-                     this.ruleBase.getReteooBuilder(),
-                     sink,
-                     workingMemories );
     }
\ No newline at end of file
