diff --git a/old_product.java b/new_product.java
index 5b0aa5b..fb13e5d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public TranslationResult translate(final String text, final String source, final String target) {
-		Map<String, Object> params = new HashMap<String, Object>();
-		params.put(TEXT, text);
-		params.put(SOURCE, source);
-		params.put(TARGET, target);
-		return translate(params);
-	}
\ No newline at end of file
+    final Map<String, Object> params = new HashMap<String, Object>();
+    params.put(TEXT, text);
+    params.put(SOURCE, source);
+    params.put(TARGET, target);
+    return translate(params);
+  }
\ No newline at end of file
