diff --git a/old_product.java b/new_product.java
index 8dc7017..e29f4f0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
-public Int SRID(final ANode node) throws QueryException {
-		
-	 // Check QName
-      if(checkNode(node, new QNm[0])) {
-        Geometry geom = gmlReader(node);      
-		    return Int.get(geom.getSRID());
-	    }
-	    return null;
-	  }
\ No newline at end of file
+public Int getSRID(final ANode node) throws QueryException {
+
+    // Check QName
+    if(checkNode(node, new QNm[0])) {
+      Geometry geom = gmlReader(node);
+      return Int.get(geom.getSRID());
+    }
+    return null;
+  }
\ No newline at end of file
