diff --git a/old_product.java b/new_product.java
index b51052f..b0331e8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,8 +5,6 @@ private Item dropBackup(final QueryContext ctx) throws QueryException {
     final StringList backups = ctx.context.databases.backups(name);
     if(backups.isEmpty()) throw BXDB_WHICHBACK.get(info, name);
 
-    for(final String backup : backups) {
-      ctx.updates.add(new BackupDrop(backup, info, ctx), ctx);
-    }
+    for(final String backup : backups) ctx.updates.add(new BackupDrop(backup, info, ctx), ctx);
     return null;
   }
\ No newline at end of file
