diff --git a/old_test.java b/new_test.java
index 04e0d8a..aead2e4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-	public void getHandler_shouldGetAHandlerForTheStringWithRegexLogicalType() throws Exception {
+	public void getHandler_shouldGetAHandlerForTheStringWithRegexDatatype() throws Exception {
 		// ideally we'd have independent web-layer and api-layer tests
 		AttributeHandler<?> handler = service.getHandler("regex-validated-string", null);
 		Assert.assertTrue(RegexValidatedStringAttributeHandler.class.isAssignableFrom(handler.getClass()));
