diff --git a/old_test.java b/new_test.java
index 9e5d41a..3728f6d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,8 @@
     public void testThreshholdSplit()
     {
         System.out.println("threshholdSplit");
-        Distribution dist1 = new Normal(0, 1);
-        Distribution dist2 = new Normal(3, 2);
+        ContinuousDistribution dist1 = new Normal(0, 1);
+        ContinuousDistribution dist2 = new Normal(3, 2);
         PairedReturn<Integer, Double> ret = DecisionStump.threshholdSplit(dist1, dist2);
         assertEquals(0, (int) ret.getFirstItem());
         assertEquals(1.418344988105127, ret.getSecondItem(), 1e-6);
