diff --git a/old_product.java b/new_product.java
index defae42..acbccb1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static PairedReturn<Integer, Double> threshholdSplit(final Distribution dist1, final Distribution dist2)
+public static PairedReturn<Integer, Double> threshholdSplit(final ContinuousDistribution dist1, final ContinuousDistribution dist2)
     {
         if(dist1 == null && dist2 == null)
             throw new ArithmeticException("No Distributions given");
