diff --git a/old_test.java b/new_test.java
index d0277dd..7f29fd7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test
-    public void testIsDiagonal() {
-        System.out.println("isDiagonal");
+    public void testDiagonal() {
+        System.out.println("diagonal");
         MultivariateGaussianDistribution instance = new MultivariateGaussianDistribution(mu, 1.0);
-        assertEquals(true, instance.isDiagonal());
+        assertEquals(true, instance.diagonal);
 
         instance = new MultivariateGaussianDistribution(mu, sigma[0]);
-        assertEquals(true, instance.isDiagonal());
+        assertEquals(true, instance.diagonal);
 
         instance = new MultivariateGaussianDistribution(mu, Matrix.of(sigma));
-        assertEquals(false, instance.isDiagonal());
+        assertEquals(false, instance.diagonal);
     }
\ No newline at end of file
