diff --git a/old_product.java b/new_product.java
index 84381f0..f6c2f41 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,7 @@
-public boolean isDiagonal() {
-        return diagonal;
+public double[][] rand(int n) {
+        double[][] data = new double[n][];
+        for (int i = 0; i < n; i++) {
+            data[i] = rand();
+        }
+        return data;
     }
\ No newline at end of file
