diff --git a/old_product.java b/new_product.java
index d745e7c..cd690f8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,4 @@
 public <C extends Serializable, D extends AutoCloseable> Dataset<C, D> build(
-        PartitionContextBuilder<K, V, C> partCtxBuilder, PartitionDataBuilder<K, V, C, D> partDataBuilder);
\ No newline at end of file
+        LearningEnvironmentBuilder envBuilder,
+        PartitionContextBuilder<K, V, C> partCtxBuilder,
+        PartitionDataBuilder<K, V, C, D> partDataBuilder);
\ No newline at end of file
