diff --git a/old_test.java b/new_test.java
index 6718319..3f1166f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void createUfsFileTest() throws Exception {
-    mManager.createFile(SESSION_ID, mUri, PermissionStatus.defaults());
+    mManager.createFile(SESSION_ID, mUri, Permission.defaults());
     Mockito.verify(mMockUfs).create(Mockito.contains(mUri.toString()),
         Mockito.any(CreateOptions.class));
     Mockito.verify(mMockUfs).connectFromWorker(Mockito.any(Configuration.class),
