diff --git a/old_product.java b/new_product.java
index 8ae5dcd..cf70c97 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public long createUfsFile(long sessionId, AlluxioURI ufsUri, PermissionStatus ps)
+public long createUfsFile(long sessionId, AlluxioURI ufsUri, Permission perm)
       throws FileAlreadyExistsException, IOException {
-    return mUnderFileSystemManager.createFile(sessionId, ufsUri, ps);
+    return mUnderFileSystemManager.createFile(sessionId, ufsUri, perm);
   }
\ No newline at end of file
