diff --git a/old_test.java b/new_test.java
index 0fe6b8c..aee1b51 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,12 +10,12 @@
 
     Mockito.when(hive.getServiceComponent("HIVE_METASTORE")).thenReturn(metastore);
 
-    Map<String, ServiceComponentHost> metastores = new HashMap<String, ServiceComponentHost>();
+    Map<String, ServiceComponentHost> metastores = new HashMap<>();
     Mockito.when(metastore.getServiceComponentHosts()).thenReturn(metastores);
 
     PrerequisiteCheck check = new PrerequisiteCheck(null, null);
     PrereqCheckRequest request = new PrereqCheckRequest("cluster");
-    request.setRepositoryVersion("2.3.0.0");
+    request.setTargetRepositoryVersion(m_repositoryVersion);
     m_check.perform(check, request);
 
     Assert.assertEquals(PrereqCheckStatus.WARNING, check.getStatus());
