diff --git a/old_test.java b/new_test.java
index 61c872b..78e96af 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,7 +15,7 @@
     BasicTable.drop(path, conf);
     try {
       BasicTable.Writer writer = new BasicTable.Writer(path, STR_SCHEMA,
-          STR_STORAGE, false, conf);
+          STR_STORAGE, conf);
       Assert.fail("should throw exception");
     } catch (Exception e) {
       System.out.println(e);
