diff --git a/old_product.java b/new_product.java
index b811df7..4e3cf2c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -37,7 +37,7 @@
 		 } else if (st == Partition.SplitType.MAP && keys != null) {
        String key;
        Iterator<String> it;
-
+       Object value;
 			 for (int i = 0; i < size; i++)
 			 {
 				 if (children.get(i).projIndex != -1) // a leaf: set projection directly
@@ -45,7 +45,10 @@
            for (it = keys.iterator(); it.hasNext(); )
            {
              key = it.next();
-			 		   ((Map<String, Object>) (((Tuple)children.get(i).leaf.field).get(children.get(i).projIndex))).put(key, ((Map<String, Object>) field).get(key));
+             value = ((Map<String, Object>) field).get(key);
+             if (value == null)
+               continue;
+			 		   ((Map<String, Object>) (((Tuple)children.get(i).leaf.field).get(children.get(i).projIndex))).put(key, value);
            }
          } else {
            for (it = keys.iterator(); it.hasNext(); )
