diff --git a/old_product.java b/new_product.java
index 3d21394..fa3a348 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 private Str integerToBase(final QueryContext qc, final InputInfo ii) throws QueryException {
-    final long num = checkItr(exprs[0], qc), base = checkItr(exprs[1], qc);
-    if(base < 2 || base > 36) throw INVBASE.get(ii, base);
+    final long num = toLong(exprs[0], qc), base = toLong(exprs[1], qc);
+    if(base < 2 || base > 36) throw INVBASE_X.get(ii, base);
 
     // use fast variant for powers of two
     for(int i = 1, p = 2; i < 6; i++, p <<= 1)
