diff --git a/old_test.java b/new_test.java
index 190e9e2..e5cf1fb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
     query(_CONVERT_INTEGER_TO_BASE.args(4, 10), 4);
     query(_CONVERT_INTEGER_TO_BASE.args(65535, 10), 65535);
     query(_CONVERT_INTEGER_TO_BASE.args(65536, 10), 65536);
-    error(_CONVERT_INTEGER_TO_BASE.args(1, 1), Err.INVBASE);
-    error(_CONVERT_INTEGER_TO_BASE.args(1, 100), Err.INVBASE);
-    error(_CONVERT_INTEGER_TO_BASE.args(1, 100), Err.INVBASE);
+    error(_CONVERT_INTEGER_TO_BASE.args(1, 1), Err.INVBASE_X);
+    error(_CONVERT_INTEGER_TO_BASE.args(1, 100), Err.INVBASE_X);
+    error(_CONVERT_INTEGER_TO_BASE.args(1, 100), Err.INVBASE_X);
   }
\ No newline at end of file
