diff --git a/old_test.java b/new_test.java
index 92ac7bd..042a5f6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
 
         Collection<ReferenceBean<?>> referenceBeans = beanPostProcessor.getReferenceBeans();
 
-        Assert.assertEquals(1, referenceBeans.size());
+        Assert.assertEquals(2, referenceBeans.size());
 
         ReferenceBean<?> referenceBean = referenceBeans.iterator().next();
 
