diff --git a/old_product.java b/new_product.java
index 4b9a26b..8717ef9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ protected void putConnectionBackInPartition(ConnectionHandle connectionHandle) t
 		BoundedLinkedTransferQueue<ConnectionHandle> queue = connectionHandle.getOriginatingPartition().getFreeConnections();
 
 		if (!queue.tryTransfer(connectionHandle)){
-			if (!queue.tryPut(connectionHandle)){
+			if (!queue.offer(connectionHandle)){
 				connectionHandle.internalClose();
 			}
 		}
