diff --git a/old_test.java b/new_test.java
index 21f7e36..cd7f94f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
 		
 		expect(mockConnection.getOriginatingPartition()).andReturn(mockPartition).anyTimes();
 		expect(mockConnectionHandles.tryTransfer(mockConnection)).andReturn(false).anyTimes();
-		expect(mockConnectionHandles.tryPut(mockConnection)).andReturn(true).once();
+		expect(mockConnectionHandles.offer(mockConnection)).andReturn(true).once();
 		replay(mockPartition, mockConnectionHandles, mockConnection);
 		testClass.putConnectionBackInPartition(mockConnection);
 		// FIXME
