diff --git a/old_test.java b/new_test.java
index bc6bbe1..c8af1b4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,4 @@
-@Test(groups={"tck"})
-    public void test_plusMinutes() {
-        LocalDateTime ldt = LocalDateTime.of(2008, 6, 30, 23, 30, 59, 0);
-        ZonedDateTime base = ZonedDateTime.of(ldt, ZONE_0100);
-        ZonedDateTime test = base.plusMinutes(30);
-        assertEquals(test, ZonedDateTime.of(ldt.plusMinutes(30), ZONE_0100));
+@Test(groups={"tck"}, dataProvider="plusTime")
+    public void test_plusMinutes(ZonedDateTime base, long amount, ZonedDateTime expected) {
+        assertEquals(base.plusMinutes(amount * 60), expected);
     }
\ No newline at end of file
