diff --git a/old_test.java b/new_test.java
index 3e63a5e..66afade 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,9 @@
   public void testFindByName() {
     AlertDefinitionEntity entity = new AlertDefinitionEntity();
     TypedQuery<AlertDefinitionEntity> query = createStrictMock(TypedQuery.class);
-    
+
+    expect(query.setParameter("clusterId", 12345L)).andReturn(query);
+
     expect(query.setParameter("definitionName", "alert-definition-1")).andReturn(
         query);
 
@@ -14,7 +16,7 @@
 
     replay(query, entityManager);
 
-    AlertDefinitionEntity result = dao.findByName("alert-definition-1");
+    AlertDefinitionEntity result = dao.findByName(12345L, "alert-definition-1");
 
     assertSame(result, entity);
     verify(entityManagerProvider, entityManager);
