diff --git a/old_product.java b/new_product.java
index ec6c2ca..a1bf81d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,9 @@
 @RequiresSession
-  public AlertDefinitionEntity findByName(String definitionName) {
+  public AlertDefinitionEntity findByName(long clusterId, String definitionName) {
     TypedQuery<AlertDefinitionEntity> query = entityManagerProvider.get().createNamedQuery(
         "AlertDefinitionEntity.findByName", AlertDefinitionEntity.class);
 
+    query.setParameter("clusterId", clusterId);
     query.setParameter("definitionName", definitionName);
 
     return daoUtils.selectSingle(query);
