diff --git a/old_product.java b/new_product.java
index 471f9f7..ff91cb0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
-@Authorized(PrivilegeConstants.GET_ORDERS)
-	public List<Order> getOrderHistoryByConcept(Patient patient, Concept concept);
\ No newline at end of file
+@Authorized(PrivilegeConstants.MANAGE_ORDER_TYPES)
+	public void createOrderType(OrderType orderType) throws APIException;
\ No newline at end of file
