diff --git a/old_product.java b/new_product.java
index 782bbea..721887d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,27 +1,26 @@
 @Override
-    public Long poll() {
-        Long retval = null;
-        while (true) {
-            if (this.state == QueueState.UNBLOCKED ||
-                (this.state == QueueState.BLOCKED_SAFETY && EstTime.currentTimeMillis() >= this.blockTime)) {
+    public AbstractTransaction poll() {
+        AbstractTransaction retval = null;
+        
+        // These invocations of poll() can return null if the next
+        // txn was speculatively executed
+        
+        if (this.state == QueueState.BLOCKED_SAFETY) {
+            if (EstTime.currentTimeMillis() >= this.blockTime) {
                 retval = super.poll();
-                if (retval != null && this.removed.contains(retval)) {
-                    this.removed.remove(super.poll());
-                    this.checkQueueState();
-                    continue;
-                }
             }
-            break;
-        } // WHILE
-        
+        } else if (this.state == QueueState.UNBLOCKED) {
+//            assert(checkQueueState() == QueueState.UNBLOCKED);
+            retval = super.poll();
+        }
         if (t) LOG.trace(String.format("Partition %d :: poll() -> %s", this.partitionId, retval));
         if (retval != null) {
-            assert(this.nextTxnId.equals(retval)) : 
+            assert(this.nextTxn.equals(retval)) : 
                 String.format("Partition %d :: Next txn is %s but our poll returned %s\n" +
                               StringUtil.SINGLE_LINE + "%s",
-                              this.partitionId, this.nextTxnId, retval, this.debug());
-            this.nextTxnId = null;
-            this.lastTxnIdPopped = retval;
+                              this.partitionId, this.nextTxn, retval, this.debug());
+            this.nextTxn = null;
+            this.lastTxnPopped = retval.getTransactionId();
         }
         
         if (d && retval != null)
