diff --git a/old_product.java b/new_product.java
index af4e273..fd516fd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 private Value foldLeft1(final QueryContext ctx) throws QueryException {
-    final FItem f = withArity(0, 2, ctx);
-    final Iter xs = expr[1].iter(ctx);
+    final FItem f = withArity(1, 2, ctx);
+    final Iter xs = expr[0].iter(ctx);
 
     Value sum = checkNoEmpty(xs.next());
     for(Item x; (x = xs.next()) != null;) sum = f.invValue(ctx, info, sum, x);
