diff --git a/old_test.java b/new_test.java
index c17a221..52174f5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
   public void foldLeft1Test() {
-    query("hof:fold-left1(function($x, $y) { $x + $y }, 1 to 10)", "55");
-    error("hof:fold-left1(function($x, $y) { $x + $y }, ())", Err.INVEMPTY);
+    query("hof:fold-left1(1 to 10, function($x, $y) { $x + $y })", "55");
+    error("hof:fold-left1((), function($x, $y) { $x + $y })", Err.INVEMPTY);
   }
\ No newline at end of file
