diff --git a/old_test.java b/new_test.java
index 306c15e..c51ea9a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,17 +1,8 @@
 @Test
-  public void getNettyServerTest() {
-    // dud properties. server should pick up defaults
+  public void getNettyServerTest() throws Exception {
     Properties properties = new Properties();
-    VerifiableProperties verifiableProperties = new VerifiableProperties(properties);
-    RestRequestHandler restRequestHandler = new MockRestRequestResponseHandler();
-    PublicAccessLogger publicAccessLogger = new PublicAccessLogger(new String[]{}, new String[]{});
-    RestServerState restServerState = new RestServerState("/healthCheck");
-
-    NettyServerFactory nettyServerFactory =
-        new NettyServerFactory(verifiableProperties, new MetricRegistry(), restRequestHandler, publicAccessLogger,
-            restServerState);
-    NioServer nioServer = nettyServerFactory.getNioServer();
-    assertNotNull("No NioServer returned", nioServer);
-    assertEquals("Did not receive a NettyServer instance", NettyServer.class.getCanonicalName(),
-        nioServer.getClass().getCanonicalName());
+    doGetNettyServerTest(properties, SSL_FACTORY);
+    doGetNettyServerTest(properties, null);
+    properties.setProperty("netty.server.enable.ssl", "true");
+    doGetNettyServerTest(properties, SSL_FACTORY);
   }
\ No newline at end of file
