diff --git a/old_test.java b/new_test.java
index 6ac5927..9ecbc64 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,13 +5,13 @@
 
         ClusterState clusterState = buildClusterState(
                 localEndpoint,
+                ImmutableList.of(),
                 NodeState.getUnavailableNodeState("a"),
-                nodeState("b", OK, OK, OK),
+                nodeState("b", epoch, OK, OK, OK),
                 NodeState.getUnavailableNodeState("c")
         );
 
-        List<String> unresponsiveServers = new ArrayList<>();
-        Optional<NodeRank> failedServer = advisor.failedServer(clusterState, unresponsiveServers);
+        Optional<NodeRank> failedServer = advisor.failedServer(clusterState);
         assertTrue(failedServer.isPresent());
         assertEquals(new NodeRank("c", 0), failedServer.get());
     }
\ No newline at end of file
