diff --git a/old_test.java b/new_test.java
index 336e431..95a83bf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,12 +17,14 @@
         // Verify the queue hasn't returned before we actually set the result.
         Assert.assertNull("Queue unblocked before result was set.", result.get());
         Thread.sleep(10);
-        queue.close();
+        List<Integer> queueContents = queue.close();
 
         // Verify result.
         AssertExtensions.assertThrows(
                 "Future was not cancelled with the correct exception.",
                 resultSet::join,
                 ex -> ex instanceof InterruptedException);
+
         Assert.assertNull("Queue returned an item even if it got closed.", result.get());
+        Assert.assertEquals("Queue.close() returned an item even though it was empty.", 0, queueContents.size());
     }
\ No newline at end of file
