diff --git a/old_product.java b/new_product.java
index 2cdfa05..e4feb63 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,14 @@
-@Override
-    public void close() {
-        if (!this.closed) {
-            this.closed = true;
-            lock.lock();
-            try {
+public List<T> close() {
+        lock.lock();
+        try {
+            if (!this.closed) {
+                this.closed = true;
                 this.notEmpty.signal();
-            } finally {
-                lock.unlock();
+                return swapContents();
             }
+        } finally {
+            lock.unlock();
         }
+
+        return new ArrayList<>();
     }
\ No newline at end of file
