diff --git a/old_product.java b/new_product.java
index ebac5ca..9022765 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,14 +1,11 @@
 @Override
     //TODO: Shares a lot of code with the add, should be able to refactor
     public List<Command> updateCommandsForCluster(
+            @NotBlank(message = "No cluster id entered. Unable to update commands.")
             final String id,
-            final List<Command> commands) throws GenieException {
-        if (StringUtils.isBlank(id)) {
-            throw new GeniePreconditionException("No cluster id entered. Unable to update commands.");
-        }
-        if (commands == null) {
-            throw new GeniePreconditionException("No commands entered. Unable to add commands.");
-        }
+            @NotEmpty(message = "No commands entered. Unable to add commands.")
+            final List<Command> commands
+    ) throws GenieException {
         final Cluster cluster = this.clusterRepo.findOne(id);
         if (cluster != null) {
             final List<Command> cmds = new ArrayList<>();
