diff --git a/old_test.java b/new_test.java
index f4f5721..6fdce27 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,6 @@
 @Test
     public void testGetResetTodayAndTomorrow(){
-        LOGGER.debug(JsonUtil.format(DateExtensionUtil.getResetTodayAndTomorrow()));
+        assertArrayEquals(
+                        ConvertUtil.toArray(DateUtil.getFirstDateOfThisDay(NOW), DateUtil.getFirstDateOfThisDay(DateUtil.addDay(NOW, 1))),
+                        DateExtensionUtil.getResetTodayAndTomorrow());
     }
\ No newline at end of file
