diff --git a/old_product.java b/new_product.java
index 5282541..7759338 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,6 @@ public void validate(Object obj, Errors errors) throws IllegalArgumentException
 			        + ConceptSource.class);
 		} else {
 			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "name", "error.name");
-			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "description", "error.description.required");
 		}
 		
 	}
\ No newline at end of file
