diff --git a/old_product.java b/new_product.java
index 8770f9e..ac07daf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,9 @@
-public void deleteFromZooKeeper(ZooKeeper zk, AbstractConfiguration conf,
-                                    BookieSocketAddress address, Version version)
-            throws KeeperException, InterruptedException, UnknownHostException {
+public void deleteFromRegistrationManager(RegistrationManager rm,
+                                              BookieSocketAddress address,
+                                              Version version) throws BookieException {
         if (!(version instanceof LongVersion)) {
             throw new IllegalArgumentException("Invalid version type, expected ZkVersion type");
         }
 
-        String zkPath = getZkPath(conf, address);
-        zk.delete(zkPath, (int) ((LongVersion) version).getLongVersion());
-        LOG.info("Removed cookie from {} for bookie {}.", conf.getZkLedgersRootPath(), address);
+        rm.removeCookie(address.toString(), version);
     }
\ No newline at end of file
