diff --git a/old_test.java b/new_test.java
index aca9ad0..e7c8c83 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(dataProvider="sampleTimes", groups={"implementation"})
-    public void test_toLocalTime(int h, int m, int s, int ns) {
+    public void test_getTime(int h, int m, int s, int ns) {
         LocalTime t = LocalTime.of(h, m, s, ns);
         LocalDateTime dt = LocalDateTime.of(LocalDate.of(2011, 7, 30), t);
-        assertSame(dt.toLocalTime(), t);
+        assertSame(dt.getTime(), t);
     }
\ No newline at end of file
