diff --git a/old_test.java b/new_test.java
index ab5ffe8..2897d35 100644
--- a/old_test.java
+++ b/new_test.java
@@ -25,6 +25,6 @@
         "return validate:xsd-info($doc, $schema)"), "true");
 
     // invalid arguments
-    error(_VALIDATE_XSD_INFO.args("unknown"), Err.WHICHRES);
-    error(_VALIDATE_XSD_INFO.args(FILE, "unknown.xsd"), Err.WHICHRES);
+    error(_VALIDATE_XSD_INFO.args("unknown"), Err.WHICHRES_X);
+    error(_VALIDATE_XSD_INFO.args(FILE, "unknown.xsd"), Err.WHICHRES_X);
   }
\ No newline at end of file
