diff --git a/old_product.java b/new_product.java
index 0d4d5c1..228f690 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,10 @@
-public static <T extends PluginInfo<?>> void getPluginsOfClass(
-		final String className, final List<? extends PluginInfo<?>> srcList,
-		final List<T> destList)
+@Override
+	public <PT extends ImageJPlugin, P extends PT> List<PluginInfo<PT>>
+		getPluginsOfClass(final Class<P> pluginClass, final Class<PT> type)
 	{
-		for (final PluginInfo<?> info : srcList) {
-			if (info.getClassName().equals(className)) {
-				@SuppressWarnings("unchecked")
-				final T match = (T) info;
-				destList.add(match);
-			}
-		}
+		final ArrayList<PluginInfo<PT>> result =
+			new ArrayList<PluginInfo<PT>>();
+		final String className = pluginClass.getName();
+		findPluginsOfClass(className, getPluginsOfType(type), result);
+		return result;
 	}
\ No newline at end of file
