diff --git a/old_test.java b/new_test.java
index 2081118..de28441 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,11 +14,11 @@
       "return validate:dtd($doc, $dtd) ", "");
 
     // invalid arguments
-    error(_VALIDATE_DTD.args("unknown"), Err.WHICHRES);
-    error(_VALIDATE_DTD.args(FILE, "unknown.dtd"), Err.WHICHRES);
-    error(_VALIDATE_DTD.args(FILE), Err.BXVA_FAIL);
+    error(_VALIDATE_DTD.args("unknown"), Err.WHICHRES_X);
+    error(_VALIDATE_DTD.args(FILE, "unknown.dtd"), Err.WHICHRES_X);
+    error(_VALIDATE_DTD.args(FILE), Err.BXVA_FAIL_X);
     error(
         "let $doc := <root/> " +
         "let $dtd := '<!ELEMENT unknown (#PCDATA)>' " +
-        "return validate:dtd($doc, $dtd) ", Err.BXVA_FAIL);
+        "return validate:dtd($doc, $dtd) ", Err.BXVA_FAIL_X);
   }
\ No newline at end of file
