diff --git a/old_product.java b/new_product.java
index 5e709ee..74e5ddf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,8 @@
 public static <T extends ImageBase>
-	void rotate( T input , T output , TypeInterpolate type ,
-				 float centerX , float centerY , float angle ) {
-		Class<T> inputType = (Class<T>)input.getClass();
+	void rotate( T input , T output , TypeInterpolate interpType , float angleInputToOutput ) {
 
-		PixelTransform model = DistortSupport.transformRotate(centerX,centerY,angle);
-		ImageDistort<T> distorter = DistortSupport.createDistort(inputType,model,type);
+		Class<T> inputType = (Class<T>)input.getClass();
+		InterpolatePixel<T> interp = FactoryInterpolation.createPixel(0, 255, interpType, inputType);
 
-		distorter.apply(input,output);
+		rotate(input, output, interp, angleInputToOutput);
 	}
\ No newline at end of file
