diff --git a/old_product.java b/new_product.java
index 2c441da..6e819ca 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
 public PatientIdentifierType unretirePatientIdentifierType(PatientIdentifierType patientIdentifierType)
 	        throws APIException {
+		checkIfPatientIdentifierTypesAreLocked();
 		patientIdentifierType.setRetired(false);
 		patientIdentifierType.setRetiredBy(null);
 		patientIdentifierType.setDateRetired(null);
