diff --git a/old_test.java b/new_test.java
index 69d8889..ca45428 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,22 +15,22 @@
     }
 
     CreateValueOptions createOptions = new CreateValueOptions.Builder(workspaceId, entity, entityValue)
-        .synonyms(new ArrayList<String>(Arrays.asList(synonym1, synonym2)))
-        .build();
+            .synonyms(new ArrayList<String>(Arrays.asList(synonym1, synonym2)))
+            .build();
     service.createValue(createOptions).execute();
 
     Date start = new Date();
 
     try {
       GetValueOptions getOptions =
-          new GetValueOptions.Builder(workspaceId, entity, entityValue)
-              .export(true)
-              .build();
+              new GetValueOptions.Builder(workspaceId, entity, entityValue)
+                      .export(true)
+                      .build();
       ValueExport response = service.getValue(getOptions).execute();
 
       assertNotNull(response);
-      assertNotNull(response.getValue());
-      assertEquals(response.getValue(), entityValue);
+      assertNotNull(response.getValueText());
+      assertEquals(response.getValueText(), entityValue);
       assertNotNull(response.getCreated());
       assertNotNull(response.getUpdated());
 
@@ -48,7 +48,7 @@
       fail(ex.getMessage());
     } finally {
       DeleteValueOptions deleteOptions =
-          new DeleteValueOptions.Builder(workspaceId, entity, entityValue).build();
+              new DeleteValueOptions.Builder(workspaceId, entity, entityValue).build();
       service.deleteValue(deleteOptions).execute();
     }
   }
\ No newline at end of file
