diff --git a/old_product.java b/new_product.java
index 19344d8..65f2625 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,8 @@
-public static ProtocolMessage getNextProtocolMessageFromPeer(TlsContext tlsContext, int position) {
+public static ProtocolMessage getNextReceiveProtocolMessage(TlsContext tlsContext, int position) {
 	ConnectionEnd peer = tlsContext.getMyConnectionEnd().getPeer();
-	for (int i = position; i < tlsContext.getWorkflowTrace().getProtocolMessages().size(); i++) {
-	    ProtocolMessage pm = tlsContext.getWorkflowTrace().getProtocolMessages().get(i);
-	    if (peer == pm.getMessageIssuer()) {
-		return pm;
-	    }
+	for (int i = position; i < tlsContext.getWorkflowTrace().getConfiguredReceivingMessages().size(); i++) {
+	    ProtocolMessage pm = tlsContext.getWorkflowTrace().getConfiguredReceivingMessages().get(i);
+	    return pm;
 	}
 	return null;
     }
\ No newline at end of file
