diff --git a/old_test.java b/new_test.java
index ccd4459..9d1cd07 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test
     public void testGetNextProtocolMessageFromPeer() {
 	WorkflowConfigurationFactory factory = WorkflowConfigurationFactory.createInstance(new ClientCommandConfig());
-	TlsContext context = factory.createFullTlsContext();
+	TlsContext context = factory.createFullServerResponseTlsContext(ConnectionEnd.CLIENT);
 	context.setMyConnectionEnd(ConnectionEnd.CLIENT);
 
-	ProtocolMessage pm = TlsContextAnalyzer.getNextProtocolMessageFromPeer(context, 1);
+	ProtocolMessage pm = TlsContextAnalyzer.getNextReceiveProtocolMessage(context, 1);
 	assertEquals(ProtocolMessageType.HANDSHAKE, pm.getProtocolMessageType());
 
-	pm = TlsContextAnalyzer.getNextProtocolMessageFromPeer(context, 5);
+	pm = TlsContextAnalyzer.getNextReceiveProtocolMessage(context, 3);
 	assertEquals(ProtocolMessageType.CHANGE_CIPHER_SPEC, pm.getProtocolMessageType());
     }
\ No newline at end of file
