diff --git a/old_product.java b/new_product.java
index c4feeea..5bf556e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public Object putValues(String name, String... values) 
+@SafeVarargs
+    public final List<V> putValues(String name, V... values) 
     {
-        Object list=null;
-        for (int i=0;i<values.length;i++)
-            list=LazyList.add(list,values[i]);
-        return put(name,list);
+        List<V> list = new ArrayList<>();
+        list.addAll(Arrays.asList(values));
+        return super.put(name,list);
     }
\ No newline at end of file
