diff --git a/old_product.java b/new_product.java
index 3e41753..3b8bb0c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public static boolean stop(final Context ctx, final String id) {
     // stop scheduled task
-    final TimerTask task = ctx.jobs.tasks.get(id);
+    final TimerTask task = ctx.jobs.tasks.remove(id);
     if(task != null) task.cancel();
     // send stop signal to job
     final Job job = ctx.jobs.active.get(id);
