diff --git a/old_product.java b/new_product.java
index e960cdf..f496cd7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@
             notes = "Remove the given tag from the command with given id.  Note that the genie name space tags"
                     + "prefixed with genie.id and genie.name cannot be deleted.",
             response = String.class,
-            responseContainer = "List"
+            responseContainer = "Set"
     )
     @ApiResponses(value = {
             @ApiResponse(
