diff --git a/old_product.java b/new_product.java
index b70f4bc..648a08c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,10 @@
 @Override
     public List<Command> addCommandsForCluster(
+            @NotBlank(message = "No cluster id entered. Unable to add commands.")
             final String id,
-            final List<Command> commands) throws GenieException {
-        if (StringUtils.isBlank(id)) {
-            throw new GeniePreconditionException("No cluster id entered. Unable to add commands.");
-        }
-        if (commands == null) {
-            throw new GeniePreconditionException("No commands entered. Unable to add commands.");
-        }
+            @NotEmpty(message = "No commands entered. Unable to add commands.")
+            final List<Command> commands
+    ) throws GenieException {
         final Cluster cluster = this.clusterRepo.findOne(id);
         if (cluster != null) {
             for (final Command detached : commands) {
