diff --git a/old_product.java b/new_product.java
index 79766aa..80d2b6b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -34,9 +34,7 @@ public void configure(Properties props) throws HibernateException {
 						Method method = BoneCPConfig.class.getDeclaredMethod("set"+upFirst(field.getName()), boolean.class);
 						String val = props.getProperty("bonecp."+field.getName());
 						if (val != null) {
-							if (val.toUpperCase().equals("TRUE") || val.toUpperCase().equals("FALSE")){
-								method.invoke(this.config, Boolean.parseBoolean(val));
-							} 
+							method.invoke(this.config, Boolean.parseBoolean(val));
 						}
 					}
 				}
