diff --git a/old_product.java b/new_product.java
index 2518848..aca0cca 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,8 @@
 public boolean hasChildTag(String tagName)
 	{
-		boolean hasChild = false;
+		Args.notEmpty(tagName, "tagName");
 
-		if (Strings.isEmpty(tagName))
-		{
-			throw new IllegalArgumentException("You need to provide a not empty/not null argument.");
-		}
+		boolean hasChild = false;
 
 		if (openTag.isOpen())
 		{
@@ -21,7 +18,7 @@ public boolean hasChildTag(String tagName)
 					XmlPullParser p = new XmlPullParser();
 					p.parse(markup);
 
-					XmlTag tag = null;
+					XmlTag tag;
 					while ((tag = p.nextTag()) != null)
 					{
 						if (tagName.equalsIgnoreCase(tag.getName()))
@@ -34,10 +31,8 @@ public boolean hasChildTag(String tagName)
 			}
 			catch (Exception e)
 			{
-				// NOTE: IllegalStateException(Throwable) only exists since Java 1.5
 				throw new WicketRuntimeException(e);
 			}
-
 		}
 
 		return hasChild;
