diff --git a/old_product.java b/new_product.java
index 7d2bde2..bae7731 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,9 @@
 public TransfVec toEnum() {
     if( isEnum() ) return adaptTo(domain()); // Use existing domain directly
     if( !isInt() ) throw new IllegalArgumentException("Enum conversion only works on integer columns");
+    // Right now, limited to small dense integers.
+    if( min() < 0 || max() > 1000000 ) 
+      throw new IllegalArgumentException("Enum conversion only works on small integers, but min="+min()+" and max = "+max());
     long[] domain= new CollectDomain().doAll(this).domain();
     if( domain.length > Enum.MAX_ENUM_SIZE )
       throw new IllegalArgumentException("Column domain is too large to be represented as an enum: " + domain.length + " > " + Enum.MAX_ENUM_SIZE);
