diff --git a/old_product.java b/new_product.java
index ff3770f..605e275 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,4 @@ public void handle(Order order, User creator, Date dateCreated, String other) {
 		if (order.getPatient() == null && order.getEncounter() != null) {
 			order.setPatient(order.getEncounter().getPatient());
 		}
-		if (order.getOrderNumber() == null) {
-			order.setOrderNumber(Context.getOrderService().getNewOrderNumber());
-		}
 	}
\ No newline at end of file
