diff --git a/old_product.java b/new_product.java
index 3a48777..6398aac 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,6 +16,7 @@
 
         return this.clusterService.getCommandsForCluster(id, enumStatuses)
             .stream()
+            .map(DtoAdapters::toV3Command)
             .map(this.commandResourceAssembler::toResource)
             .collect(Collectors.toList());
     }
\ No newline at end of file
