diff --git a/old_test.java b/new_test.java
index d1ec81a..a8e4456 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,18 @@
 @Test
 	public void testGenerateConditionStatment() {
-		String queryCondition = dynamicQueryCondition.generateConditionStatment();
-		assertEquals(" and name like '%test%'", queryCondition);
+		String queryCondition = dynamicQueryCondition.generateConditionStatment().getStatment();
+		assertEquals(" and name like ?", queryCondition);
+		
+		List<Object> values = new ArrayList<Object>();
+		values.add("%test%");
+		assertEquals(values, dynamicQueryCondition.generateConditionStatment().getValues());
 		
 		dynamicQueryCondition = this.createAndInitDynamicQueryConditionTwoValues();
-		queryCondition = dynamicQueryCondition.generateConditionStatment();
-		assertEquals(" and age between '18' and '30'", queryCondition);
+		queryCondition = dynamicQueryCondition.generateConditionStatment().getStatment();
+		assertEquals(" and age between ? and ?", queryCondition);
+		
+		values = new ArrayList<Object>();
+		values.add("18");
+		values.add("30");
+		assertEquals(values, dynamicQueryCondition.generateConditionStatment().getValues());
 	}
\ No newline at end of file
