diff --git a/old_product.java b/new_product.java
index ae1b994..5ca8bb0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,3 @@
 public String getHtmlCookieString(String name, String value, Map<String, String> cookieProperties) {
-        if (name == null || name.isEmpty()) {
-            if (tc.isDebugEnabled()) {
-                Tr.debug(tc, "Cannot create a cookie string because the cookie name [" + name + "] was null or empty.");
-            }
-            return "";
-        }
-        String cookieString = createHtmlCookiePropertyString(name, value);
-        cookieString += createHtmlCookiePropertiesString(cookieProperties);
-        return cookieString;
+        return createHtmlCookieString(name, value, cookieProperties, true);
     }
\ No newline at end of file
