diff --git a/old_test.java b/new_test.java
index 25ac489..06bcf04 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,14 @@
 @Test
-    public void test_getHtmlCookieString_nullPropertyMap() throws Exception {
+    public void test_getUnencodedJavaScriptHtmlCookieString_emptyCookieProperties() throws Exception {
         try {
-            String name = "some name";
-            String value = "a value";
-            Map<String, String> cookieProps = null;
+            String name = "<cookie name\"'>";
+            String value = ">cookie'<\" value ";
+            Map<String, String> cookieProps = new HashMap<String, String>();
 
-            String result = utils.getHtmlCookieString(name, value, cookieProps);
+            String result = utils.getUnencodedJavaScriptHtmlCookieString(name, value, cookieProps);
 
-            String expectedCookieString = name + "=" + value + ";";
-            verifyCaseInsensitiveQuotedPatternMatches(result, expectedCookieString, "Cookie string did not match expected pattern.");
+            String expectedCookieString = DOCUMENT_COOKIE_START + name + "=" + value + ";";
+            verifyPattern(result, Pattern.quote(expectedCookieString), "Expected cookie name and value did not appear in the result.");
 
         } catch (Throwable t) {
             outputMgr.failWithThrowable(testName.getMethodName(), t);
