diff --git a/old_product.java b/new_product.java
index edf4db9..4b52cd4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 private static <O extends Options> O parse(final O opts, final Ann ann) throws Exception {
-    for(final Item arg : ann.args) opts.parse(string(arg.string(ann.info)));
+    for(final Item arg : ann.args) opts.assign(string(arg.string(ann.info)));
     return opts;
   }
\ No newline at end of file
