diff --git a/old_test.java b/new_test.java
index 7af6714..96dd7c4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void parse() {
     set(MainOptions.STRIPNS, true);
-    set(MainOptions.SERIALIZER, SerializerOptions.get(false));
+    set(MainOptions.SERIALIZER, SerializerMode.NOINDENT.get());
 
     final String doc = "<e xmlns='A'><b:f xmlns:b='B'/></e>";
     for(final boolean b : new boolean[] { false, true }) {
