diff --git a/old_product.java b/new_product.java
index a6cafa0..748fc1e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public static String toJson(HistogramMetric metric) {
         GridStringBuilder json = new GridStringBuilder();
 
-        json.a("{\"bounds\": [");
+        json.a("{\"bounds\":[");
 
         long[] bounds = metric.bounds();
 
@@ -9,10 +9,10 @@ public static String toJson(HistogramMetric metric) {
             json.a(bounds[i]);
 
             if (i < bounds.length - 1)
-                json.a(", ");
+                json.a(",");
         }
 
-        json.a("], \"values\": [");
+        json.a("],\"values\":[");
 
         long[] values = metric.value();
 
@@ -21,18 +21,18 @@ public static String toJson(HistogramMetric metric) {
             long toInclusive = (i == values.length - 1 ? -1 : bounds[i]);
             long val = values[i];
 
-            json.a("{\"fromExclusive\": ").a(fromExclusive);
+            json.a("{\"fromExclusive\":").a(fromExclusive);
 
             if (toInclusive >= 0)
-                json.a(", \"toInclusive\": ").a(toInclusive);
+                json.a(",\"toInclusive\":").a(toInclusive);
 
             json
-                .a(", \"value\": ")
+                .a(",\"value\":")
                 .a(val)
                 .a("}");
 
             if (i < values.length - 1)
-                json.a(", ");
+                json.a(",");
         }
 
         json.a("]}");
