diff --git a/old_test.java b/new_test.java
index 4857d03..9d600a5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,8 +8,8 @@
 
         System.out.println(singleBucketJson);
 
-        singleBucketJson.equals("{\"bounds\": [1], \"values\": " +
-            "[{\"fromExclusive\": 0, \"toInclusive\": 1, \"value\": 1}, {\"fromExclusive\": 1, \"value\": 0}]}"
+        singleBucketJson.equals("{\"bounds\":[1],\"values\":" +
+            "[{\"fromExclusive\":0,\"toInclusive\":1,\"value\":1},{\"fromExclusive\":1,\"value\":0}]}"
         );
 
         HistogramMetric histo = new HistogramMetric("histo", "", new long[] { 1, 5, 10, 20, 50 });
@@ -24,12 +24,12 @@
 
         System.out.println(histoJson);
 
-        histoJson.equals("{\"bounds\": [1, 5, 10, 20, 50], \"values\": " +
-            "[{\"fromExclusive\": 0, \"toInclusive\": 1, \"value\": 1}, " +
-            "{\"fromExclusive\": 1, \"toInclusive\": 5, \"value\": 0}, " +
-            "{\"fromExclusive\": 5, \"toInclusive\": 10, \"value\": 0}, " +
-            "{\"fromExclusive\": 10, \"toInclusive\": 20, \"value\": 2}, " +
-            "{\"fromExclusive\": 20, \"toInclusive\": 50, \"value\": 1}, " +
-            "{\"fromExclusive\": 50, \"value\": 1}]}"
+        histoJson.equals("{\"bounds\":[1,5,10,20,50],\"values\":" +
+            "[{\"fromExclusive\":0,\"toInclusive\":1,\"value\":1}, " +
+            "{\"fromExclusive\":1,\"toInclusive\":5,\"value\":0}, " +
+            "{\"fromExclusive\":5,\"toInclusive\":10,\"value\":0}, " +
+            "{\"fromExclusive\":10,\"toInclusive\":20,\"value\":2}, " +
+            "{\"fromExclusive\":20,\"toInclusive\":50,\"value\":1}, " +
+            "{\"fromExclusive\":50,\"value\":1}]}"
         );
     }
\ No newline at end of file
