diff --git a/old_product.java b/new_product.java
index 83ca646..912648e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@
                 case DAY_OF_WEEK: return isoDate.getDayOfWeek().getValue();
                 case DAY_OF_MONTH: return isoDate.getDayOfMonth();
                 case DAY_OF_YEAR: return isoDate.getDayOfYear();
-                case MONTH_OF_YEAR: return isoDate.getMonthOfYear().getValue();
+                case MONTH_OF_YEAR: return isoDate.getMonth().getValue();
                 case YEAR_OF_ERA: return (isoDate.getYear() >= 1 ? isoDate.getYear() : 1 - isoDate.getYear());
                 case YEAR: return isoDate.getYear();
                 case ERA: return (isoDate.getYear() >= 1 ? 1 : 0);
