diff --git a/old_test.java b/new_test.java
index 05942cd..7a1b6b2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,6 @@
     public void test_get_dates(int y, int m, int d) {
         LocalDateTime a = LocalDateTime.of(y, m, d, 12, 30);
         assertEquals(a.getYear(), y);
-        assertEquals(a.getMonthOfYear(), Month.of(m));
+        assertEquals(a.getMonth(), Month.of(m));
         assertEquals(a.getDayOfMonth(), d);
     }
\ No newline at end of file
