diff --git a/old_test.java b/new_test.java
index 29af4b4..43115f9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
     public void testEquals_Object() {
-        CharRange rangea = CharRange.is('a');
-        CharRange rangeae = CharRange.isIn('a', 'e');
-        CharRange rangenotbf = CharRange.isIn('b', 'f');
+        final CharRange rangea = CharRange.is('a');
+        final CharRange rangeae = CharRange.isIn('a', 'e');
+        final CharRange rangenotbf = CharRange.isIn('b', 'f');
 
         assertFalse(rangea.equals(null));
 
