diff --git a/old_product.java b/new_product.java
index 0c268b3..206640f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,27 +5,27 @@ public static boolean equals(final Annotation a1, final Annotation a2) {
         if (a1 == null || a2 == null) {
             return false;
         }
-        Class<? extends Annotation> type = a1.annotationType();
-        Class<? extends Annotation> type2 = a2.annotationType();
+        final Class<? extends Annotation> type = a1.annotationType();
+        final Class<? extends Annotation> type2 = a2.annotationType();
         Validate.notNull(type, "Annotation %s with null annotationType()", a1);
         Validate.notNull(type2, "Annotation %s with null annotationType()", a2);
         if (!type.equals(type2)) {
             return false;
         }
         try {
-            for (Method m : type.getDeclaredMethods()) {
+            for (final Method m : type.getDeclaredMethods()) {
                 if (m.getParameterTypes().length == 0
                         && isValidAnnotationMemberType(m.getReturnType())) {
-                    Object v1 = m.invoke(a1);
-                    Object v2 = m.invoke(a2);
+                    final Object v1 = m.invoke(a1);
+                    final Object v2 = m.invoke(a2);
                     if (!memberEquals(m.getReturnType(), v1, v2)) {
                         return false;
                     }
                 }
             }
-        } catch (IllegalAccessException ex) {
+        } catch (final IllegalAccessException ex) {
             return false;
-        } catch (InvocationTargetException ex) {
+        } catch (final InvocationTargetException ex) {
             return false;
         }
         return true;
