diff --git a/old_product.java b/new_product.java
index b350a6d..6010bff 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,19 +1,21 @@
-@Override
-  protected void prepareConfiguration(String path,
-      org.apache.hadoop.conf.Configuration hadoopConf) {
+public static Configuration prepareConfiguration(String path, Map<String, String> ufsConf) {
     if (path.startsWith(SCHEME)) {
+      Configuration hadoopConf = new Configuration();
       // Configure for Gluster FS
-      hadoopConf.set("fs.glusterfs.impl", Configuration.get(PropertyKey.UNDERFS_GLUSTERFS_IMPL));
-      hadoopConf.set("mapred.system.dir", Configuration.get(PropertyKey.UNDERFS_GLUSTERFS_MR_DIR));
-      hadoopConf
-          .set("fs.glusterfs.volumes", Configuration.get(PropertyKey.UNDERFS_GLUSTERFS_VOLUMES));
-      hadoopConf.set(
-          "fs.glusterfs.volume.fuse." + Configuration.get(PropertyKey.UNDERFS_GLUSTERFS_VOLUMES),
-          Configuration.get(PropertyKey.UNDERFS_GLUSTERFS_MOUNTS));
+      hadoopConf.set("fs.glusterfs.impl",
+          UnderFileSystemUtils.getValue(PropertyKey.UNDERFS_GLUSTERFS_IMPL, ufsConf));
+      hadoopConf.set("mapred.system.dir",
+          UnderFileSystemUtils.getValue(PropertyKey.UNDERFS_GLUSTERFS_MR_DIR, ufsConf));
+      hadoopConf.set("fs.glusterfs.volumes",
+          UnderFileSystemUtils.getValue(PropertyKey.UNDERFS_GLUSTERFS_VOLUMES, ufsConf));
+      hadoopConf.set("fs.glusterfs.volume.fuse."
+          + UnderFileSystemUtils.getValue(PropertyKey.UNDERFS_GLUSTERFS_VOLUMES, ufsConf),
+          UnderFileSystemUtils.getValue(PropertyKey.UNDERFS_GLUSTERFS_MOUNTS, ufsConf));
+      return hadoopConf;
     } else {
       // If not Gluster FS fall back to default HDFS behavior
       // This should only happen if someone creates an instance of this directly rather than via the
       // registry and factory which enforces the GlusterFS prefix being present.
-      super.prepareConfiguration(path, hadoopConf);
+      return HdfsUnderFileSystem.createConfiguration(ufsConf);
     }
   }
\ No newline at end of file
