diff --git a/old_product.java b/new_product.java
index 471639d..85576b4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public static String getOwnerFromGrpcClient() {
+public static String getOwnerFromGrpcClient(AlluxioConfiguration conf) {
     try {
-      User user = AuthenticatedClientUser.get();
+      User user = AuthenticatedClientUser.get(conf);
       if (user == null) {
         return "";
       }
