diff --git a/old_product.java b/new_product.java
index 0af3cb0..62fe713 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
 public ServerBuilder decorator(
             DecoratingServiceFunction<HttpRequest, HttpResponse> decoratingServiceFunction) {
-        requireNonNull(decoratingServiceFunction, "decoratingServiceFunction");
-        return decorator(delegate -> new FunctionalDecoratingService<>(delegate, decoratingServiceFunction));
+        return decorator(Route.builder().catchAll().build(), decoratingServiceFunction);
     }
\ No newline at end of file
