diff --git a/old_test.java b/new_test.java
index 3eea486..3d33211 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,9 @@
         expect(mockService.requestRoleForSync(anyObject())).andReturn(role1).anyTimes();
         replay(mockService);
 
+        expect(mockDeviceService.getDevice(deviceId1)).andReturn(device1);
+        replay(mockDeviceService);
+
         final WebTarget wt = target();
         final String response = wt.path("mastership/" + deviceId1.toString() +
                 "/request").request().get(String.class);
