diff --git a/old_product.java b/new_product.java
index 4334ae0..4413294 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,8 +2,11 @@
     @Produces(MediaType.APPLICATION_JSON)
     @Path("{deviceId}/request")
     public Response requestRoleFor(@PathParam("deviceId") String deviceId) {
-        MastershipRole role = nullIsNotFound(mastershipService.requestRoleForSync(
-                                DeviceId.deviceId(deviceId)), MASTERSHIP_ROLE_NOT_FOUND);
+        DeviceId id = DeviceId.deviceId(deviceId);
+        nullIsNotFound(deviceService.getDevice(id), DEVICE_ID_NOT_FOUND);
+
+        MastershipRole role = nullIsNotFound(mastershipService.requestRoleForSync(id),
+                        MASTERSHIP_ROLE_NOT_FOUND);
         ObjectNode root = codec(MastershipRole.class).encode(role, this);
         return ok(root).build();
     }
\ No newline at end of file
