diff --git a/old_product.java b/new_product.java
index 4dfe8de..f416c08 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,9 @@ public Run lookupRun (Point point)
             return null;
         }
 
-        for (Run run : getSequence(oPt.y)) {
+        for (Itr it = new Itr(oPt.y); it.hasNext();) {
+            Run run = it.next();
+
             if (run.getStart() > oPt.x) {
                 return null;
             }
