diff --git a/old_product.java b/new_product.java
index 0850146..c34542a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,17 +1,17 @@
 @Override
   public void ensureCapacity(int capacity) {
+    Preconditions.checkArgument(capacity <= _maxCapacity);
+
     // Nothing to be done for map mode.
     if (_storageMode == StorageMode.MAP_STORAGE) {
       return;
     }
 
-    if (capacity > _maxCapacity) {
+    if (capacity > _trimSize) {
       switchToMapMode(capacity);
       return;
     }
 
-    Preconditions.checkArgument(capacity <= _maxCapacity);
-
     if (capacity > _resultHolderCapacity) {
       int copyLength = _resultHolderCapacity;
       _resultHolderCapacity = Math.max(_resultHolderCapacity * 2, capacity);
