diff --git a/old_product.java b/new_product.java
index 180c15d..6d334dd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static SyncUser currentUser() {
-        SyncUser user = SyncManager.getUserStore().get(UserStore.CURRENT_USER_KEY);
+        SyncUser user = SyncManager.getUserStore().get();
         if (user != null && user.isValid()) {
             return user;
         }
