diff --git a/old_test.java b/new_test.java
index 0223561..0018415 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
     query(conn());
     query(EXISTS.args(' ' + conn()));
     // BXCL0001: connection errors
-    error(_CLIENT_CONNECT.args(Text.S_LOCALHOST, 9999, Text.S_ADMIN, ""), Err.BXCL_CONN);
-    error(_CLIENT_CONNECT.args("xxx", 9999, Text.S_ADMIN, Text.S_ADMIN), Err.BXCL_CONN);
+    error(_CLIENT_CONNECT.args(Text.S_LOCALHOST, 9999, Text.S_ADMIN, ""), Err.BXCL_CONN_X);
+    error(_CLIENT_CONNECT.args("xxx", 9999, Text.S_ADMIN, Text.S_ADMIN), Err.BXCL_CONN_X);
   }
\ No newline at end of file
