diff --git a/old_product.java b/new_product.java
index 47d6bda..3d2be28 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public void clear() {
-        this.cache.clear();
+        this.readIndex.clear();
         this.inMemoryOperationLog.clear();
-        synchronized (this.cache) {
+        synchronized (this.readIndex) {
             this.recentStreamSegmentIds = new HashSet<>();
         }
     }
\ No newline at end of file
