diff --git a/old_product.java b/new_product.java
index f7f34dc..5fd95ce 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery lt(String propName, Comparable<?> value) {
-        criterion = criterion.and(criterionBuilder.lt(propName, value));
+        criterion = criterion.and(Criteria.lt(propName, value));
         return this;
     }
\ No newline at end of file
