diff --git a/old_product.java b/new_product.java
index 1a48e5a..3ef5505 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @VisibleForTesting
 	/* package */ void applyDeltaToLaterStockCandidates(
-			@NonNull final Integer productId,
+			@NonNull final ProductDescriptor productDescriptor,
 			@NonNull final Integer warehouseId,
 			@NonNull final Date date,
 			@NonNull final Integer groupId,
@@ -10,9 +10,10 @@
 				.type(Type.STOCK)
 				.materialDescr(MaterialDescriptor.builderForQuery()
 						.date(date)
-						.productId(productId)
-						.warehouseId(warehouseId).build())
-				.dateOperator(DateOperator.AFTER)
+						.productDescriptor(productDescriptor)
+						.warehouseId(warehouseId)
+						.dateOperator(DateOperator.AFTER)
+						.build())
 				.build();
 
 		final List<Candidate> candidatesToUpdate = candidateRepository.retrieveOrderedByDateAndSeqNo(segment);
