diff --git a/old_product.java b/new_product.java
index a379fa7..50a8897 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@
     public ListenableFuture<SessionResult> sendCustomCommand(@NonNull SessionCommand command,
             @Nullable Bundle args) {
         if (command == null) {
-            throw new IllegalArgumentException("command shouldn't be null");
+            throw new NullPointerException("command shouldn't be null");
         }
         if (command.getCommandCode() != SessionCommand.COMMAND_CODE_CUSTOM) {
             throw new IllegalArgumentException("command should be a custom command");
