diff --git a/old_test.java b/new_test.java
index d807b6a..e4c22ef 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,7 +17,7 @@
             @Override
             public SessionResult onCustomCommand(@NonNull MediaController controller,
                     @NonNull SessionCommand command, @Nullable Bundle args) {
-                if (TextUtils.equals(testCommand.getCustomCommand(), command.getCustomCommand())) {
+                if (TextUtils.equals(testCommand.getCustomAction(), command.getCustomAction())) {
                     latch.countDown();
                 }
                 return super.onCustomCommand(controller, command, args);
