diff --git a/old_product.java b/new_product.java
index b725ba5..25a4974 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,14 +2,5 @@
     public void removeAllTagsForCommand(
         @NotBlank(message = "No command id entered. Unable to remove tags.") final String id
     ) throws GenieException {
-        final Set<TagEntity> tags = this.findCommand(id).getTags();
-        // Remove all the tags except the ones that start with "genie."
-        tags.removeAll(
-            tags
-                .stream()
-                .filter(
-                    tagEntity -> !tagEntity.getTag().startsWith(JpaBaseService.GENIE_TAG_NAMESPACE)
-                )
-                .collect(Collectors.toSet())
-        );
+        this.findCommand(id).getTags().clear();
     }
\ No newline at end of file
