diff --git a/old_product.java b/new_product.java
index e059b01..80408de 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public static List<Class<?>> getAllSuperclasses(final Class<?> cls) {
         if (cls == null) {
             return null;
         }
-        List<Class<?>> classes = new ArrayList<Class<?>>();
+        final List<Class<?>> classes = new ArrayList<Class<?>>();
         Class<?> superclass = cls.getSuperclass();
         while (superclass != null) {
             classes.add(superclass);
