diff --git a/old_test.java b/new_test.java
index 46b4d70..957adaf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void test_getAllSuperclasses_Class() {
-        List<?> list = ClassUtils.getAllSuperclasses(CY.class);
+        final List<?> list = ClassUtils.getAllSuperclasses(CY.class);
         assertEquals(2, list.size());
         assertEquals(CX.class, list.get(0));
         assertEquals(Object.class, list.get(1));
