diff --git a/old_product.java b/new_product.java
index b846efd..57bf374 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,11 @@
-public synchronized List<URIStatus> listStatus(final AlluxioURI path)
-      throws IOException, AlluxioException {
+public synchronized List<URIStatus> listStatus(final AlluxioURI path,
+      final ListStatusOptions options) throws IOException, AlluxioException {
     return retryRPC(new RpcCallableThrowsAlluxioTException<List<URIStatus>>() {
       @Override
       public List<URIStatus> call() throws AlluxioTException, TException {
         List<URIStatus> result = new ArrayList<URIStatus>();
-        for (alluxio.thrift.FileInfo fileInfo : mClient.listStatus(path.getPath())) {
+        for (alluxio.thrift.FileInfo fileInfo : mClient
+            .listStatus(path.getPath(), options.toThrift())) {
           result.add(new URIStatus(ThriftUtils.fromThrift(fileInfo)));
         }
         return result;
