diff --git a/old_test.java b/new_test.java
index 37717b7..a426006 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,8 +3,8 @@
     AlluxioURI file = new AlluxioURI("/file");
     List<URIStatus> infos = new ArrayList<>();
     infos.add(new URIStatus(new FileInfo()));
-    Mockito.when(mFileSystemMasterClient.listStatus(file)).thenReturn(infos);
     ListStatusOptions listStatusOptions = ListStatusOptions.defaults();
+    Mockito.when(mFileSystemMasterClient.listStatus(file, listStatusOptions)).thenReturn(infos);
     Assert.assertSame(infos, mFileSystem.listStatus(file, listStatusOptions));
-    Mockito.verify(mFileSystemMasterClient).listStatus(file);
+    Mockito.verify(mFileSystemMasterClient).listStatus(file, listStatusOptions);
   }
\ No newline at end of file
