diff --git a/old_product.java b/new_product.java
index d563892..64f904b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -37,7 +37,7 @@
 
 			for (int i = 0; i < numberOfInitialCheckpoints - 1; i++) {
 				try {
-					removeFromZooKeeperAndDiscardCheckpoint(initialCheckpoints.get(i));
+					removeSubsumed(initialCheckpoints.get(i));
 				}
 				catch (Exception e) {
 					LOG.error("Failed to discard checkpoint", e);
