diff --git a/old_test.java b/new_test.java
index 373eba5..c9bb66b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,7 +15,7 @@
         Assert.assertThat(createdId, Matchers.is(id));
         final Command created = this.service.getCommand(id);
         Assert.assertNotNull(this.service.getCommand(id));
-        Assert.assertEquals(id, created.getId());
+        Assert.assertEquals(id, created.getId().orElseThrow(IllegalArgumentException::new));
         Assert.assertEquals(COMMAND_1_NAME, created.getName());
         Assert.assertEquals(COMMAND_1_USER, created.getUser());
         Assert.assertEquals(CommandStatus.ACTIVE, created.getStatus());
