diff --git a/old_test.java b/new_test.java
index eb3e335..79f2626 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,7 +17,7 @@
         for (int i = 1000; i < 2000; i++)
             data[i] = gamma.rand();
 
-        Vector<Mixture.Component> m = new Vector<Mixture.Component>();
+        List<Mixture.Component> m = new ArrayList<>();
         Mixture.Component c = new Mixture.Component();
         c.priori = 0.25;
         c.distribution = new GaussianDistribution(0.0, 1.0);
