diff --git a/old_product.java b/new_product.java
index ae7b684..ad93d97 100644
--- a/old_product.java
+++ b/new_product.java
@@ -59,7 +59,7 @@ double EM(List<Component> components, int[] x , double gamma, int maxIter) {
             }
 
             // Maximization step
-            List<Component> newConfig = new ArrayList<Component>();
+            List<Component> newConfig = new ArrayList<>();
             for (int i = 0; i < m; i++)
                 newConfig.add(((DiscreteExponentialFamily) components.get(i).distribution).M(x, posteriori[i]));
 
