diff --git a/old_product.java b/new_product.java
index 7b6cd17..8647891 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,29 @@
-public static PointRDD SpatialRangeQuery(PointRDD pointRDD, Envelope envelope, Integer condition) {
-		JavaRDD<Point> result = pointRDD.getRawPointRDD().filter(new PointRangeFilter(envelope, condition));
-		return new PointRDD(result);
+public static JavaRDD<Point> SpatialRangeQuery(PointRDD spatialRDD, Envelope queryWindow, Integer condition, boolean useIndex) throws Exception {
+		if(useIndex==true)
+		{
+	        if(spatialRDD.indexedRawRDD == null) {
+	            throw new Exception("[RangeQuery][SpatialRangeQuery] Index doesn't exist. Please build index on rawSpatialRDD.");
+	        }
+			JavaRDD<Object> result = spatialRDD.indexedRawRDD.mapPartitions(new RangeFilterUsingIndex(queryWindow));
+			return result.map(new Function<Object, Point>()
+			{
+
+				@Override
+				public Point call(Object spatialObject) throws Exception {
+					return (Point)spatialObject;
+				}
+				
+			});
+		}
+		else{
+			JavaRDD<Object> result = spatialRDD.getRawSpatialRDD().filter(new GeometryRangeFilter(queryWindow, condition));
+			return result.map(new Function<Object,Point>()
+			{
+				@Override
+				public Point call(Object spatialObject) throws Exception {
+					return (Point)spatialObject;
+				}
+				
+			});
+		}
 	}
\ No newline at end of file
