diff --git a/old_test.java b/new_test.java
index 6b558a3..2f51f81 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,11 @@
 @Test
     public void testSpatialRangeQuery() throws Exception {
-    	RectangleRDD rectangleRDD = new RectangleRDD(sc, InputLocation, offset, splitter);
+    	RectangleRDD spatialRDD = new RectangleRDD(sc, InputLocation, offset, splitter, true);
     	for(int i=0;i<loopTimes;i++)
     	{
-    		long resultSize = RangeQuery.SpatialRangeQuery(rectangleRDD, queryEnvelope, 0).getRawRectangleRDD().count();
+    		long resultSize = RangeQuery.SpatialRangeQuery(spatialRDD, queryEnvelope, 0,false).count();
     		assert resultSize>-1;
     	}
-    	assert RangeQuery.SpatialRangeQuery(rectangleRDD, queryEnvelope, 0).getRawRectangleRDD().take(10).get(1).getUserData().toString()!=null;
+    	assert RangeQuery.SpatialRangeQuery(spatialRDD, queryEnvelope, 0,false).take(10).get(1).getUserData().toString()!=null;
+        
     }
\ No newline at end of file
