diff --git a/old_test.java b/new_test.java
index f86ba98..6f2c1eb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,14 +18,14 @@
 
         SchemaVersions schemaVersions = client.getSchemaVersionsByApplicationId(application.getId());
         
-        List<SchemaDto> storedConfigurationSchemas = schemaVersions.getConfigurationSchemaVersions();
+        List<VersionDto> storedConfigurationSchemas = schemaVersions.getConfigurationSchemaVersions();
 
         Collections.sort(storedConfigurationSchemas, new IdComparator());
 
         Assert.assertEquals(configurationSchemas.size(), storedConfigurationSchemas.size());
         for (int i=0;i<configurationSchemas.size();i++) {
             ConfigurationSchemaDto configurationSchema = configurationSchemas.get(i);
-            SchemaDto storedConfigurationSchema = storedConfigurationSchemas.get(i);
+            VersionDto storedConfigurationSchema = storedConfigurationSchemas.get(i);
             assertSchemasEquals(configurationSchema, storedConfigurationSchema);
         }
     }
\ No newline at end of file
