diff --git a/old_product.java b/new_product.java
index 032d305..b05c201 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 @Override
-    public List<SchemaDto> getConfigurationSchemaVersionsByApplicationId(String applicationId) throws ControlServiceException {
+    public List<VersionDto> getConfigurationSchemaVersionsByApplicationId(String applicationId) throws ControlServiceException {
         return configurationService.findConfigurationSchemaVersionsByAppId(applicationId);
     }
\ No newline at end of file
