diff --git a/old_product.java b/new_product.java
index b856822..11206c0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-private static void union(final SeqType a, final SeqType b, final SeqType res) {
-    assertTrue(a + ", " + b, a.union(b).eq(res));
-    assertTrue(b + ", " + a, b.union(a).eq(res));
+private static void union(final SeqType st1, final SeqType st2, final SeqType expected) {
+    eq(st1.union(st2), expected);
+    eq(st2.union(st1), expected);
   }
\ No newline at end of file
