diff --git a/old_test.java b/new_test.java
index 055973d..950d470 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,7 +17,7 @@
             .sharedPool("secondary", "serverResource2", 8, MemoryUnit.MEGABYTES)
             .build())));
 
-    assertThat(activeEntity.getSharedResourcePoolIds(), containsInAnyOrder("primary", "secondary"));
+    assertThat(registry.getStoreManagerService().getSharedResourcePoolIds(), containsInAnyOrder("primary", "secondary"));
 
     assertThat(registry.getResource("serverResource1").getUsed(), is(MemoryUnit.MEGABYTES.toBytes(4L)));
     assertThat(registry.getResource("serverResource2").getUsed(), is(MemoryUnit.MEGABYTES.toBytes(8L)));
@@ -25,5 +25,5 @@
 
     assertThat(activeEntity.getConnectedClients().get(client), is(Matchers.<String>empty()));
     assertThat(activeEntity.getInUseStores().keySet(), is(Matchers.<String>empty()));
-    assertThat(activeEntity.getStores(), is(Matchers.<String>empty()));
+    assertThat(registry.getStoreManagerService().getStores(), is(Matchers.<String>empty()));
   }
\ No newline at end of file
