diff --git a/old_product.java b/new_product.java
index cbd9539..20f8e5f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,6 @@ private void configure(ClientDescriptor clientDescriptor, ConfigureStoreManager
     if (clientState == null) {
       throw new LifecycleException("Client " + clientDescriptor + " is not connected to the Clustered Tier Manager");
     }
-    serverStoreManager.configure(message);
+    ehcacheStateService.configure(message);
     clientState.attach();
   }
\ No newline at end of file
