diff --git a/old_product.java b/new_product.java
index 7ea6d90..6b319f6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-default T handleHttpResponseException(HttpResponseException responseException)
-      throws HttpResponseException, RegistryErrorException {
+default T handleHttpResponseException(ResponseException responseException)
+      throws ResponseException, RegistryErrorException {
     throw responseException;
   }
\ No newline at end of file
