diff --git a/old_test.java b/new_test.java
index 53996ee..28da6f7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test
-  public void testHandleHttpResponseException_dockerRegistry_tagInvalid()
-      throws HttpResponseException {
-    HttpResponseException exception = Mockito.mock(HttpResponseException.class);
+  public void testHandleHttpResponseException_dockerRegistry_tagInvalid() throws ResponseException {
+    ResponseException exception = Mockito.mock(ResponseException.class);
     Mockito.when(exception.getStatusCode()).thenReturn(HttpStatus.SC_BAD_REQUEST);
     Mockito.when(exception.getContent())
         .thenReturn(
