diff --git a/old_test.java b/new_test.java
index 4ae636a..ca967b9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
 		order.setDiscontinued(null);
 		order.setConcept(Context.getConceptService().getConcept(88));
 		order.setPatient(Context.getPatientService().getPatient(2));
-		order.setOrderType(Context.getOrderService().getOrderType(1));
+		//order.setOrderType(Context.getOrderService().getOrderType(1));
 		
 		Errors errors = new BindException(order, "order");
 		new OrderValidator().validate(order, errors);
