diff --git a/old_product.java b/new_product.java
index a2388e3..2b03c3b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,7 +14,8 @@ private void list(final int root, final File dir, final StringList list,
     // parse files. ignore directories if a pattern is specified
     for(final File f : ch) {
       if(pat == null || pat.matcher(f.getName()).matches()) {
-        list.add(f.getPath().substring(root));
+        final String file = f.getPath().substring(root);
+        list.add(f.isDirectory() ? dir(file) : file);
       }
     }
   }
\ No newline at end of file
