diff --git a/old_test.java b/new_test.java
index ffdb990..01db66e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,15 +5,4 @@
         Map<Integer, List<Integer>> group = ArrayUtil.group(array);
         LOGGER.debug(JsonUtil.format(group));
 
-        Collection<List<Integer>> values = group.values();
-        LOGGER.debug(JsonUtil.format(values));
-
-        //****************************************************************
-        String[] array1 = { "关羽", "feilong", "关羽", "基友团", "关羽" };
-
-        Map<String, List<String>> group1 = ArrayUtil.group(array1);
-        LOGGER.debug(JsonUtil.format(group1));
-
-        Collection<List<String>> values1 = group1.values();
-        LOGGER.debug(JsonUtil.format(values1));
     }
\ No newline at end of file
