diff --git a/old_product.java b/new_product.java
index cdd8265..db9b76d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,8 +2,7 @@ public static <T> Map<T, List<T>> group(T[] array){
         if (null == array){
             return Collections.emptyMap();
         }
-        //视需求  可以换成 HashMap 或者TreeMap
-        Map<T, List<T>> map = new WeakHashMap<T, List<T>>(array.length);
+        Map<T, List<T>> map = new LinkedHashMap<T, List<T>>(array.length);
         for (T t : array){
             List<T> valueList = map.get(t);
             if (null == valueList){
