diff --git a/old_test.java b/new_test.java
index 941d5d0..2d45ef6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,7 +17,7 @@
 
         final int size = RollingRandomAccessFileManager.DEFAULT_BUFFER_SIZE * 3 + 1;
         final byte[] data = new byte[size];
-        manager.write(data, 0, data.length); // no exception
+        manager.write(data, 0, data.length, flushNow); // no exception
         assertEquals(RollingRandomAccessFileManager.DEFAULT_BUFFER_SIZE * 3,
                 raf.length());
 
