diff --git a/old_test.java b/new_test.java
index 93c341c..a0a193c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(groups="tck")
-    public void test_createEra() {
-        assertEquals(ISOChronology.INSTANCE.createEra(0), ISO_BCE);
-        assertEquals(ISOChronology.INSTANCE.createEra(1), ISO_CE);
+    public void test_eraOf() {
+        assertEquals(ISOChronology.INSTANCE.eraOf(0), ISO_BCE);
+        assertEquals(ISOChronology.INSTANCE.eraOf(1), ISO_CE);
     }
\ No newline at end of file
