diff --git a/old_product.java b/new_product.java
index 12cbe70..bb680b5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -42,7 +42,7 @@
             
             try {
                 this.sendingCommand(commandString);
-                conn.sendStringToComm(commandString + this.getLineTerminator());
+                conn.sendStringToComm(commandString + "\n");
                 synchronized(nextCommandLock) {
                     nextCommand = null;
                 }
