diff --git a/old_test.java b/new_test.java
index b0fa76c..94ceb5d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
         System.out.println("streamCommands");
         MockConnection mc = new MockConnection(mg.in, mg.out);
         GrblCommunicator instance = new GrblCommunicator(cb, asl, mc);
-        String term = instance.getLineTerminator();
+        String term = "\n";
         String thirtyNineCharString = "thirty-nine character command here.....";
 
         boolean result;
