diff --git a/old_test.java b/new_test.java
index ca7e259..e7ca2f5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,9 +9,9 @@
     query("count(" + COLLECTION.args(NAME + "/newtest") + ")", XMLFILES);
 
     // invalid target
-    error(_DB_RENAME.args(NAME, "input.xml", " ''"), BXDB_PATH_X);
-    error(_DB_RENAME.args(NAME, "input.xml", " '/'"), BXDB_PATH_X);
-    error(_DB_RENAME.args(NAME, "input.xml", " '.'"), BXDB_PATH_X);
+    error(_DB_RENAME.args(NAME, "input.xml", " ''"), DB_PATH_X);
+    error(_DB_RENAME.args(NAME, "input.xml", " '/'"), DB_PATH_X);
+    error(_DB_RENAME.args(NAME, "input.xml", " '.'"), DB_PATH_X);
 
     // rename paths
     query(_DB_RENAME.args(NAME, "", "x"));
@@ -32,8 +32,8 @@
     query(_DB_STORE.args(NAME, "file4", ""));
     query(_DB_STORE.args(NAME, "dir3/file5", ""));
 
-    error(_DB_RENAME.args(NAME, "dir2", "file4"), BXDB_PATH_X);
-    error(_DB_RENAME.args(NAME, "file4", "dir2"), BXDB_PATH_X);
+    error(_DB_RENAME.args(NAME, "dir2", "file4"), DB_PATH_X);
+    error(_DB_RENAME.args(NAME, "file4", "dir2"), DB_PATH_X);
 
     // move files in directories
     query(_DB_RENAME.args(NAME, "dir2", "dir3"));
