diff --git a/old_test.java b/new_test.java
index 4a0328b..94fd61a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,13 @@
 @Test
     public void testGetLastDateOfThisDay(){
         logDate(DateUtil.getLastDateOfThisDay(NOW));
+        LOGGER.debug(StringUtils.repeat("*", 20));
+
+        logDate(DateUtils.ceiling(NOW, Calendar.DAY_OF_MONTH));
+        logDate(DateUtils.round(NOW, Calendar.DAY_OF_MONTH));
+        logDate(DateUtils.truncate(NOW, Calendar.DAY_OF_MONTH));
+        LOGGER.debug(StringUtils.repeat("*", 20));
+        logDate(DateUtils.ceiling(NOW, Calendar.MONTH));
+        logDate(DateUtils.round(NOW, Calendar.MONTH));
+        logDate(DateUtils.truncate(NOW, Calendar.MONTH));
     }
\ No newline at end of file
