diff --git a/old_product.java b/new_product.java
index 7c67910..2c643c7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static Date getLastDateOfThisDay(Date date){
         Validate.notNull(date, "date can't be null!");
-        Calendar calendar = DateUtil.toCalendar(date);
+        Calendar calendar = toCalendar(date);
         return CalendarUtil.toDate(CalendarUtil.resetDayEnd(calendar));
     }
\ No newline at end of file
