diff --git a/old_test.java b/new_test.java
index 2a7cbf2..4d617a1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,11 @@
 @Test
   public void list() {
-    error(_FILE_LIST.args(PATH1), Err.FILE_ND);
+    query(_FILE_WRITE.args(PATH1, "abcd"));
+    error(_FILE_LIST.args(PATH1), Err.FILE_NO_DIR);
+    error(_FILE_LIST.args(PATH1 + NAME), Err.FILE_NOT_FOUND);
     query(_FILE_WRITE.args(PATH1, "()"));
-    error(_FILE_LIST.args(PATH1), Err.FILE_ND);
-    contains(_FILE_LIST.args(PATH), NAME);
+    error(_FILE_LIST.args(PATH1), Err.FILE_NO_DIR);
+    query(_FILE_LIST.args(PATH), NAME);
     contains(_FILE_LIST.args(PATH, "false()"), NAME);
     contains(_FILE_LIST.args(PATH, "false()", NAME), NAME);
     query(_FILE_LIST.args(PATH, "false()", "XXX"), "");
