diff --git a/old_test.java b/new_test.java
index 0a8830d..7223a2c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
     public void testGetFieldValueMap() throws IllegalAccessException{
         User user = new User(12L);
-        LOGGER.info(JsonUtil.format(FieldUtil.getFieldValueMap(user)));
+        LOGGER.info(JsonUtil.format(FieldUtil.getAllFieldNameAndValueMap(user)));
     }
\ No newline at end of file
