diff --git a/old_test.java b/new_test.java
index 7c5ce8d..ae6d6a2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
         logger.info("automated_readability_index");
         String strText = TEST_STRING;
         double expResult = 143.2;
-        double result = ReadabilityStatistics.automated_readability_index(strText);
+        double result = ReadabilityStatistics.automatedReadabilityIndex(strText);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
