diff --git a/old_product.java b/new_product.java
index ae47e34..9f224d7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,6 +2,7 @@ protected void setRequest(NettyRequest request) {
     if (request != null) {
       if (this.request == null) {
         this.request = request;
+        HttpHeaders.setKeepAlive(responseMetadata, request.isKeepAlive());
       } else {
         throw new IllegalStateException(
             "Request has already been set inside NettyResponseChannel for channel {} " + ctx.channel());
