diff --git a/old_test.java b/new_test.java
index a869458..81ce8bf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,4 +6,5 @@
 
     HttpResponse response = (HttpResponse) channel.readOutbound();
     assertEquals("Unexpected response status", HttpResponseStatus.ACCEPTED, response.getStatus());
+    assertFalse("Channel not closed on the server", channel.isActive());
   }
\ No newline at end of file
