diff --git a/old_test.java b/new_test.java
index 81fa9f0..223a8a7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,6 @@
     Method updateHiveDatabaseType = UpgradeCatalog200.class.getDeclaredMethod("updateHiveDatabaseType");
     Method addNewConfigurationsFromXml = AbstractUpgradeCatalog.class.getDeclaredMethod
         ("addNewConfigurationsFromXml");
-    Method setSecurityType = UpgradeCatalog200.class.getDeclaredMethod("setSecurityType");
     Method updateTezConfiguration = UpgradeCatalog200.class.getDeclaredMethod("updateTezConfiguration");
     Method updateClusterEnvConfiguration = UpgradeCatalog200.class.getDeclaredMethod("updateClusterEnvConfiguration");
     Method updateConfigurationProperties = AbstractUpgradeCatalog.class.getDeclaredMethod
@@ -15,7 +14,6 @@
         .addMockedMethod(removeNagiosService)
         .addMockedMethod(updateHiveDatabaseType)
         .addMockedMethod(addNewConfigurationsFromXml)
-        .addMockedMethod(setSecurityType)
         .addMockedMethod(updateTezConfiguration)
         .addMockedMethod(updateConfigurationProperties)
         .addMockedMethod(updateClusterEnvConfiguration)
@@ -29,8 +27,6 @@
     
     upgradeCatalog.updateHiveDatabaseType();
     expectLastCall().once();
-    upgradeCatalog.setSecurityType();
-    expectLastCall().once();
 
     upgradeCatalog.updateTezConfiguration();
     expectLastCall().once();
