diff --git a/old_product.java b/new_product.java
index 693758f..f62522a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1 @@
-boolean changeProtectionKey(String locationKey, String contentKey, NetworkContent content,
-			KeyPair oldKey, KeyPair newKey);
\ No newline at end of file
+boolean changeProtectionKey(String locationKey, String contentKey, int ttl, KeyPair oldKey, KeyPair newKey);
\ No newline at end of file
