diff --git a/old_product.java b/new_product.java
index be3cf05..a6900be 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ public void dumpSequences ()
         System.out.println(toString());
 
         for (int i = 0; i < sequences.length; i++) {
-            RunSequence seq = sequences[i];
-            System.out.printf("%4d:%s%n", i, seq.toString());
+            short[] rle = sequences[i];
+            System.out.printf("%4d:%s%n", i, Arrays.toString(rle));
         }
     }
\ No newline at end of file
