diff --git a/old_test.java b/new_test.java
index e4c9a68..975a5c7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -26,7 +26,7 @@
         response().withHeaders(
             new Header(HttpHeaders.Names.CONTENT_TYPE, HttpMediaType.APPLICATION_JSON)).withBody(
             GsonSingleton.getGson().toJson(speechResults)));
-    final SpeechResults result = service.recognize(audio, HttpMediaType.AUDIO_WAV);
+    final SpeechResults result = service.recognize(audio);
     Assert.assertNotNull(result);
     Assert.assertEquals(result, speechResults);
   }
\ No newline at end of file
