diff --git a/old_product.java b/new_product.java
index 1c7d031..535914b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 private Expr type(final QueryContext qc) {
-    FnTrace.dump(Util.inf("{ type: %, size: %, exprSize: % }", exprs[0].seqType(), exprs[0].size(),
+    FnTrace.trace(Util.inf("{ type: %, size: %, exprSize: % }", exprs[0].seqType(), exprs[0].size(),
         exprs[0].exprSize()), token(exprs[0].toString()), qc);
     return exprs[0];
   }
\ No newline at end of file
