diff --git a/old_product.java b/new_product.java
index ded0df3..e63a43f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public static Date truncate(final Date date, final int field) {
         if (date == null) {
             throw new IllegalArgumentException("The date must not be null");
         }
-        Calendar gval = Calendar.getInstance();
+        final Calendar gval = Calendar.getInstance();
         gval.setTime(date);
         modify(gval, field, MODIFY_TRUNCATE);
         return gval.getTime();
