diff --git a/old_test.java b/new_test.java
index bbb9145..5209724 100644
--- a/old_test.java
+++ b/new_test.java
@@ -23,8 +23,7 @@
 		
 		Task clientTask = null;
 		
-		TimerSchedulerServiceImpl t = new TimerSchedulerServiceImpl();
-		clientTask = t.scheduleTask(taskDefinition);
+		clientTask = Context.getSchedulerService().scheduleTask(taskDefinition);
 		
 		// without this commit there seems to be a table lock left on the SCHEDULER_TASK_CONFIG table, see TRUNK-4212
 		Context.flushSession();
