diff --git a/old_product.java b/new_product.java
index 646d036..5d5e8e4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @Override
-    public AbstractTransaction poll() {
-        AbstractTransaction retval = null;
+    public Long poll() {
+        Long retval = null;
         
         // These invocations of poll() can return null if the next
         // txn was speculatively executed
@@ -15,13 +15,12 @@
         }
         if (t) LOG.trace(String.format("Partition %d :: poll() -> %s", this.partitionId, retval));
         if (retval != null) {
-            assert(this.nextTxn.equals(retval)) : 
+            assert(this.nextTxnId.equals(retval)) : 
                 String.format("Partition %d :: Next txn is %s but our poll returned %s\n" +
                               StringUtil.SINGLE_LINE + "%s",
-                              this.partitionId, this.nextTxn, retval, this.debug());
-            this.nextTxn = null;
-            this.txnsPopped++;
-            this.lastTxnPopped = retval.getTransactionId();
+                              this.partitionId, this.nextTxnId, retval, this.debug());
+            this.nextTxnId = null;
+            this.lastTxnIdPopped = retval;
         }
         
         if (d && retval != null)
