diff --git a/old_product.java b/new_product.java
index 18dd792..90cf18a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,8 +7,9 @@ public void validate(Object obj, Errors errors) {
 			
 			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "version", "error.null");
 			
-			if (form.getVersion() != null && !form.getVersion().matches("^\\d.*$"))
+			if (form.getVersion() != null && !form.getVersion().matches("^\\d.*$")) {
 				errors.rejectValue("version", "Form.version.invalid");
+			}
 			
 			if (form.isRetired()) {
 				ValidationUtils.rejectIfEmptyOrWhitespace(errors, "retireReason", "general.retiredReason.empty");
