diff --git a/old_test.java b/new_test.java
index 9164583..8458bca 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
   public void testGetConnector() {
-    when(requests.readConnector(1L)).thenReturn(connectorBean(connector(1)));
+    when(resourceRequests.readConnector(1L)).thenReturn(connectorBean(connector(1)));
     MConnector connector = client.getConnector(1);
     assertEquals(1, connector.getPersistenceId());
 
-    client.getResourceBundle(1L);
+    client.getConnectorConfigResourceBundle(1L);
 
-    verify(requests, times(1)).readConnector(1L);
+    verify(resourceRequests, times(1)).readConnector(1L);
   }
\ No newline at end of file
