diff --git a/old_product.java b/new_product.java
index de0f34d..4bec552 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,6 +5,6 @@ public void complete() {
 
         this.done = true;
         if (this.successHandler != null) {
-            CallbackHelpers.invokeSafely(this.successHandler, seqNo, cex -> log.error("Success Callback invocation failure.", cex));
+            CallbackHelpers.invokeSafely(this.successHandler, null, cex -> log.error("Success Callback invocation failure.", cex));
         }
     }
\ No newline at end of file
