diff --git a/old_test.java b/new_test.java
index 7854008..388168f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,18 +1,17 @@
 @Test(description = "PUT /vApp/{id}", dependsOnMethods = { "testGetVm" })
-   public void testModifyVm() {
-      Vm newVm = Vm.builder()
-            .name(name("new-name-"))
-            .description("New Description")
-            .build();
+   public void testEditVm() {
+      Vm newVm = Vm.builder().name(name("new-name-")).description("New Description").build();
 
       // The method under test
-      Task modifyVm = vmApi.modifyVm(vm.getHref(), newVm);
-      assertTrue(retryTaskSuccess.apply(modifyVm), String.format(TASK_COMPLETE_TIMELY, "modifyVm"));
+      Task editVm = vmApi.edit(vm.getHref(), newVm);
+      assertTrue(retryTaskSuccess.apply(editVm), String.format(TASK_COMPLETE_TIMELY, "editVm"));
 
-      // Get the updated Vm
-      vm = vmApi.getVm(vm.getHref());
+      // Get the edited Vm
+      vm = vmApi.get(vm.getHref());
 
       // Check the required fields are set
-      assertEquals(vm.getName(), newVm.getName(), String.format(OBJ_FIELD_EQ, VM, "Name", newVm.getName(), vm.getName()));
-      assertEquals(vm.getDescription(), newVm.getDescription(), String.format(OBJ_FIELD_EQ, VM, "Description", newVm.getDescription(), vm.getDescription()));
+      assertEquals(vm.getName(), newVm.getName(),
+               String.format(OBJ_FIELD_EQ, VM, "Name", newVm.getName(), vm.getName()));
+      assertEquals(vm.getDescription(), newVm.getDescription(),
+               String.format(OBJ_FIELD_EQ, VM, "Description", newVm.getDescription(), vm.getDescription()));
    }
\ No newline at end of file
