diff --git a/old_product.java b/new_product.java
index 28f738e..b1687a4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,12 +2,12 @@ public static String[] getRootCauseStackTrace(final Throwable throwable) {
         if (throwable == null) {
             return ArrayUtils.EMPTY_STRING_ARRAY;
         }
-        Throwable throwables[] = getThrowables(throwable);
-        int count = throwables.length;
-        List<String> frames = new ArrayList<String>();
+        final Throwable throwables[] = getThrowables(throwable);
+        final int count = throwables.length;
+        final List<String> frames = new ArrayList<String>();
         List<String> nextTrace = getStackFrameList(throwables[count - 1]);
         for (int i = count; --i >= 0;) {
-            List<String> trace = nextTrace;
+            final List<String> trace = nextTrace;
             if (i != 0) {
                 nextTrace = getStackFrameList(throwables[i - 1]);
                 removeCommonFrames(trace, nextTrace);
