diff --git a/old_product.java b/new_product.java
index f4aa800..370d6c8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public static int extractTopologyTimeout(TopologyAPI.Topology topology) {
+public static Duration extractTopologyTimeout(TopologyAPI.Topology topology) {
     for (TopologyAPI.Config.KeyValue keyValue : topology.getTopologyConfig().getKvsList()) {
       if (keyValue.getKey().equals("topology.message.timeout.secs")) {
-        return Integer.parseInt(keyValue.getValue());
+        return TypeUtils.getDuration(keyValue.getValue(), ChronoUnit.SECONDS);
       }
     }
 
