diff --git a/old_test.java b/new_test.java
index 0637cfc..04c71ce 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
   public void testExtractTopologyTimeout() throws Exception {
-    Assert.assertEquals(1, PhysicalPlanUtil.extractTopologyTimeout(topology));
+    Assert.assertEquals(Duration.ofSeconds(1), PhysicalPlanUtil.extractTopologyTimeout(topology));
   }
\ No newline at end of file
