diff --git a/old_product.java b/new_product.java
index 5c46ce0..e6d4a09 100644
--- a/old_product.java
+++ b/new_product.java
@@ -20,10 +20,12 @@
 		}
 		
 		Disjunction disjunction = Restrictions.disjunction();
+		MatchMode matchMode = personSearchCriteria.getAttributeMatchMode();
+		
 		for (String value : values) {
 			if (value != null && value.length() > 0) {
 				disjunction.add(personSearchCriteria.prepareCriterionForName(value, voided)).add(
-				    personSearchCriteria.prepareCriterionForAttribute(value, voided));
+				    personSearchCriteria.prepareCriterionForAttribute(value, voided, matchMode));
 			}
 		}
 		criteria.add(disjunction);
