diff --git a/old_product.java b/new_product.java
index 3c26432..a0e4c69 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,4 @@
 private Str name(final QueryContext ctx) throws QueryException {
-    return Str.get(checkFile(0, ctx).getName());
+    final Path path = checkPath(0, ctx).getFileName();
+    return path == null ? Str.ZERO : Str.get(path.toString());
   }
\ No newline at end of file
