diff --git a/old_test.java b/new_test.java
index d93d356..17f22b6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
     // check with a path ending with a directory separator
     query(_FILE_NAME.args(PATH1 + File.separator), NAME);
     // check with a path consisting only of directory separators
-    query(_FILE_NAME.args("//"), "");
+    query(_FILE_NAME.args("/"), "");
     // check with empty string path
     query(_FILE_NAME.args(""), "");
     // check using a suffix
