diff --git a/old_product.java b/new_product.java
index a4e1b54..3df4281 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static double gunning_fog_score(String strText) {
-        strText = clean_text(strText);
-        return PHPfunctions.round(((average_words_per_sentence(strText) + percentage_words_with_three_syllables(strText)) * 0.4), 1);
+public static double gunningFogScore(String strText) {
+        strText = cleanText(strText);
+        return PHPfunctions.round(((averageWordsPerSentence(strText) + percentageWordsWithThreeSyllables(strText)) * 0.4), 1);
     }
\ No newline at end of file
