diff --git a/old_test.java b/new_test.java
index d4f043f..8d8df28 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
         logger.info("gunning_fog_score");
         String strText = TEST_STRING;
         double expResult = 118.0;
-        double result = ReadabilityStatistics.gunning_fog_score(strText);
+        double result = ReadabilityStatistics.gunningFogScore(strText);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
