diff --git a/old_test.java b/new_test.java
index fe06cb1..99e2d84 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void sendHeartbeatRequest()
             throws Exception {
-        byte[] buffer = client.sendHeartbeatRequest().get();
-        assertThat(NodeMetrics.parseFrom(buffer)).isNotNull();
+        NodeView nodeView = client.sendHeartbeatRequest().get();
+        assertThat(nodeView).isNotNull();
     }
\ No newline at end of file
