diff --git a/old_test.java b/new_test.java
index b422a32..30092e7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test
   void testEnsureCapacity() {
-    GroupByResultHolder resultHolder =
-        new DoubleGroupByResultHolder(INITIAL_CAPACITY, MAX_CAPACITY, MAX_CAPACITY, DEFAULT_VALUE);
+    GroupByResultHolder resultHolder = new DoubleGroupByResultHolder(INITIAL_CAPACITY, MAX_CAPACITY, DEFAULT_VALUE);
 
     for (int i = 0; i < INITIAL_CAPACITY; i++) {
       resultHolder.setValueForKey(i, _expected[i]);
