diff --git a/old_product.java b/new_product.java
index c34542a..be6b1aa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,16 +2,6 @@
   public void ensureCapacity(int capacity) {
     Preconditions.checkArgument(capacity <= _maxCapacity);
 
-    // Nothing to be done for map mode.
-    if (_storageMode == StorageMode.MAP_STORAGE) {
-      return;
-    }
-
-    if (capacity > _trimSize) {
-      switchToMapMode(capacity);
-      return;
-    }
-
     if (capacity > _resultHolderCapacity) {
       int copyLength = _resultHolderCapacity;
       _resultHolderCapacity = Math.max(_resultHolderCapacity * 2, capacity);
