diff --git a/old_test.java b/new_test.java
index 4a74e2f..be8f9ff 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
-	public void testDilemmas() {
-		Dilemma dilemma = service.dilemmas(problem,false);
-		assertNotNull(dilemma);
-		assertNotNull(dilemma.getProblem());
-		assertNotNull(dilemma.getResolution());
-	}
\ No newline at end of file
+  public void testDilemmas() {
+    final Dilemma dilemma = service.dilemmas(problem, false);
+    assertNotNull(dilemma);
+    assertNotNull(dilemma.getProblem());
+    assertNotNull(dilemma.getResolution());
+  }
\ No newline at end of file
