diff --git a/old_test.java b/new_test.java
index a3fb81d..7010fde 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,13 +9,14 @@
                 {
                     allowing(request).getRequestURL();
                     will(returnValue(urlValue));
-                    allowing(request).getQueryString
-                        ();
+                    allowing(request).getQueryString();
+                    will(returnValue(null));
+                    allowing(request).getParameterMap();
                     will(returnValue(null));
                 }
             });
         String output = WebUtils.getRequestStringForTrace(request,secret);
         assertNotNull(output);
         assertFalse(output.contains("password"));
-        assertTrue(output.contains(url));
+        assertTrue(output.contains(url)); 
     }
\ No newline at end of file
