diff --git a/old_product.java b/new_product.java
index b94819b..69eb474 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,23 @@
 @Trivial
     public static String getRequestStringForTrace(HttpServletRequest request, String[] secretStrings) {
-        if (request == null) {
+        if (request==null || request.getRequestURL()==null) {
             return "[]";
         }
-        String url = ((request.getRequestURL()==null)?null:request.getRequestURL().toString());
-        String queryString = request.getQueryString();
 
-        return "["+stripSecretsFromUrl(url,secretStrings)+"], queryString["+((queryString==null)?"":stripSecretsFromUrl(queryString,secretStrings))+"]";
+        StringBuffer sb = new StringBuffer("["+stripSecretsFromUrl(request.getRequestURL().toString(),secretStrings)+"]");
+
+        String query = request.getQueryString();
+        if (query!=null) {
+            String queryString = stripSecretsFromUrl(query,secretStrings);
+            if (queryString!=null) {
+                sb.append(", queryString["+queryString+"]");
+            }
+        } else {
+            Map<String, String[]> pMap = request.getParameterMap();
+            String paramString = stripSecretsFromParameters(pMap, secretStrings);
+            if (paramString!=null) {
+                sb.append(", parameters["+paramString+"]");
+            }
+        }
+        return sb.toString();
     }
\ No newline at end of file
