diff --git a/old_test.java b/new_test.java
index 16ff463..abf0b16 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,13 +8,15 @@
             {"/aaa///bbb/","/aaa///bbb/"},
             {"/aaa/./bbb/","/aaa/bbb/"},
             {"/aaa/../bbb/","/bbb/"},
+            {"/aaa..bbb/","/aaa..bbb/"},
+            {"/aaa/..bbb/","/aaa/..bbb/"},
             {"/aaa/./../bbb/","/bbb/"},
             {"/aaa/bbb/ccc/../../ddd/","/aaa/ddd/"},
             {"./bbb/","bbb/"},
             {"./aaa/../bbb/","bbb/"},
             {"./",""},
-            {".//",".//"},
-            {".///",".///"},
+            {".//","/"},
+            {".///","//"},
             {"/.","/"},
             {"//.","//"},
             {"///.","///"},
@@ -49,12 +51,19 @@
             {"a/../..",null},
             {"/foo/../../bar",null},
             {"/foo/../bar//","/bar//"},
+            {"/....","/...."},
         };
 
         for (int t=0;t<canonical.length;t++)
+        {
             assertEquals( "canonical "+canonical[t][0],
-                          canonical[t][1],
-                          URIUtil.canonicalPath(canonical[t][0])
-                          );
+                canonical[t][1],
+                URIUtil.canonicalPath(canonical[t][0])
+                );
+            assertEquals( "canonical "+canonical[t][0]+"?a=1",
+                canonical[t][1]==null?null:(canonical[t][1]+"?a=1"),
+                URIUtil.canonicalPath(canonical[t][0]+"?a=1")
+                );
+        }
 
     }
\ No newline at end of file
