diff --git a/old_test.java b/new_test.java
index 964f706..7f1f502 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(expected = APIException.class)
 	@Verifies(value = "should throw error when privilege is core privilege", method = "purgePrivilege(Privilege)")
 	public void purgePrivilege_shouldThrowErrorWhenPrivilegeIsCorePrivilege() throws Exception {
-		Context.getUserService().purgePrivilege(new Privilege(OpenmrsConstants.PRIV_ADD_COHORTS));
+		Context.getUserService().purgePrivilege(new Privilege(PrivilegeConstants.ADD_COHORTS));
 	}
\ No newline at end of file
