diff --git a/old_product.java b/new_product.java
index 3eb79e4..02f2af8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,12 @@
 private synchronized Item writeText(final boolean append, final QueryContext ctx)
       throws QueryException, IOException {
 
-    final File path = check(checkFile(0, ctx));
+    final Path path = checkParentDir(checkPath(0, ctx));
     final byte[] s = checkStr(expr[1], ctx);
-    final String enc = encoding(2, FILE_UE, ctx);
+    final String enc = encoding(2, FILE_UNKNOWN_ENCODING, ctx);
     final Charset cs = enc == null || enc == UTF8 ? null : Charset.forName(enc);
 
-    try(final PrintOutput out = PrintOutput.get(new FileOutputStream(path, append))) {
+    try(final PrintOutput out = PrintOutput.get(new FileOutputStream(path.toFile(), append))) {
       out.write(cs == null ? s : string(s).getBytes(cs));
     }
     return null;
