diff --git a/old_product.java b/new_product.java
index 12f7c9d..ec6c253 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@
 	public void purgeOrderFrequency(OrderFrequency orderFrequency) {
 		
 		if (dao.isOrderFrequencyInUse(orderFrequency)) {
-			throw CannotDeleteOrderPropertyInUseException.withProperty("frequency");
+			throw new CannotDeleteObjectInUseException("Order.frequency.cannot.delete", (Object[]) null);
 		}
 		
 		dao.purgeOrderFrequency(orderFrequency);
