diff --git a/old_product.java b/new_product.java
index e73a82f..8be1abc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,5 +6,8 @@ public ServiceCall<ValueExport> getValue(GetValueOptions getValueOptions) {
     if (getValueOptions.export() != null) {
       builder.query("export", String.valueOf(getValueOptions.export()));
     }
+    if (getValueOptions.includeAudit() != null) {
+      builder.query("include_audit", String.valueOf(getValueOptions.includeAudit()));
+    }
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(ValueExport.class));
   }
\ No newline at end of file
