diff --git a/old_test.java b/new_test.java
index ca45428..699494b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,17 +15,17 @@
     }
 
     CreateValueOptions createOptions = new CreateValueOptions.Builder(workspaceId, entity, entityValue)
-            .synonyms(new ArrayList<String>(Arrays.asList(synonym1, synonym2)))
-            .build();
+        .synonyms(new ArrayList<String>(Arrays.asList(synonym1, synonym2)))
+        .build();
     service.createValue(createOptions).execute();
 
     Date start = new Date();
 
     try {
-      GetValueOptions getOptions =
-              new GetValueOptions.Builder(workspaceId, entity, entityValue)
-                      .export(true)
-                      .build();
+      GetValueOptions getOptions = new GetValueOptions.Builder(workspaceId, entity, entityValue)
+          .export(true)
+          .includeAudit(true)
+          .build();
       ValueExport response = service.getValue(getOptions).execute();
 
       assertNotNull(response);
@@ -47,8 +47,7 @@
     } catch (Exception ex) {
       fail(ex.getMessage());
     } finally {
-      DeleteValueOptions deleteOptions =
-              new DeleteValueOptions.Builder(workspaceId, entity, entityValue).build();
+      DeleteValueOptions deleteOptions = new DeleteValueOptions.Builder(workspaceId, entity, entityValue).build();
       service.deleteValue(deleteOptions).execute();
     }
   }
\ No newline at end of file
