diff --git a/old_test.java b/new_test.java
index 5a1b142..2cae7ff 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,5 +2,5 @@
     public void test_plus_Duration() {
         Duration dur = Duration.ofSeconds(62, 3);
         OffsetDateTime t = TEST_2008_6_30_11_30_59_000000500.plus(dur);
-        assertEquals(t, OffsetDateTime.of(2008, 6, 30, 11, 32, 1, 503, OFFSET_PONE));
+        assertEquals(t, OffsetDateTime.of(LocalDate.of(2008, 6, 30), LocalTime.of(11, 32, 1, 503), OFFSET_PONE));
     }
\ No newline at end of file
