diff --git a/old_product.java b/new_product.java
index 876b858..90f607c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@ public Duration plus(long amountToAdd, TemporalUnit unit) {
             return plus(Math.multiplyExact(amountToAdd, SECONDS_PER_DAY), 0);
         }
         if (unit.isDurationEstimated()) {
-            throw new DateTimeException("Unit must not have an estimated duration");
+            throw new UnsupportedTemporalTypeException("Unit must not have an estimated duration");
         }
         if (amountToAdd == 0) {
             return this;
