diff --git a/old_product.java b/new_product.java
index dfc4a17..d824686 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,15 +2,24 @@ public void handle(ConceptName conceptName, User currentUser, Date currentDate,
 		
 		// put Integer conceptNameTagIds onto ConceptNameTags that are missing them
 		if (conceptName.getTags() != null) {
-			for (ConceptNameTag tag : conceptName.getTags()) {
+			Collection<ConceptNameTag> replacementTags = new ArrayList<ConceptNameTag>();
+			
+			Iterator<ConceptNameTag> tagsIt = conceptName.getTags().iterator();
+			while (tagsIt.hasNext()) {
+				ConceptNameTag tag = tagsIt.next();
+				
 				if (tag.getConceptNameTagId() == null) {
-					ConceptNameTag possibleReplacementTag = Context.getConceptService()
-					        .getConceptNameTagByName(tag.getTag());
-					if (possibleReplacementTag != null) {
-						conceptName.removeTag(tag);
-						conceptName.addTag(possibleReplacementTag);
+					ConceptNameTag replacementTag = Context.getConceptService().getConceptNameTagByName(tag.getTag());
+					
+					if (replacementTag != null) {
+						tagsIt.remove();
+						replacementTags.add(replacementTag);
 					}
 				}
 			}
+			
+			if (!replacementTags.isEmpty()) {
+				conceptName.getTags().addAll(replacementTags);
+			}
 		}
 	}
\ No newline at end of file
