diff --git a/old_test.java b/new_test.java
index 5e50c62..15b8ab4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,13 @@
 		ConceptNameSaveHandler handler = new ConceptNameSaveHandler();
 		ConceptName name = new ConceptName();
 		name.addTag(ConceptNameTag.PREFERRED); // this tag has a null id
+		name.addTag(ConceptNameTag.SHORT); // this tag has a null id
 		handler.handle(name, null, null, null);
-		ConceptNameTag newTag = name.getTags().iterator().next();
-		Assert.assertEquals(4, newTag.getConceptNameTagId().intValue());
+		for (ConceptNameTag tag : name.getTags()) {
+			if (tag.getTag().equals(ConceptNameTag.PREFERRED)) {
+				Assert.assertEquals(4, tag.getConceptNameTagId().intValue());
+			} else if (tag.getTag().equals(ConceptNameTag.SHORT)) {
+				Assert.assertEquals(2, tag.getConceptNameTagId().intValue());
+			}
+		}
 	}
\ No newline at end of file
