diff --git a/old_product.java b/new_product.java
index 785df92..dc6a97d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,11 +5,11 @@ public StrBuilder appendFixedWidthPadRight(final Object obj, final int width, fi
             if (str == null) {
                 str = "";
             }
-            int strLen = str.length();
+            final int strLen = str.length();
             if (strLen >= width) {
                 str.getChars(0, width, buffer, size);
             } else {
-                int padLen = width - strLen;
+                final int padLen = width - strLen;
                 str.getChars(0, strLen, buffer, size);
                 for (int i = 0; i < padLen; i++) {
                     buffer[size + strLen + i] = padChar;
