diff --git a/old_test.java b/new_test.java
index 485f1a5..f2ca125 100644
--- a/old_test.java
+++ b/new_test.java
@@ -91,7 +91,14 @@
         assertEquals("Get Scaling Events response code", 404, response.getStatus());
 
         // Test for getScalingEvents for bad request.
-        // from > to is tested here
+        // 1. Missing query parameters are validated here.
+        response = addAuthHeaders(client.target(resourceURI).request()).buildGet().invoke();
+        assertEquals("Get Scaling Events response code", 400, response.getStatus());
+
+        response = addAuthHeaders(client.target(resourceURI).queryParam("from", fromDateTime).request()).buildGet().invoke();
+        assertEquals("Get Scaling Events response code", 400, response.getStatus());
+
+        // 2. from > to is tested here.
         doAnswer(x -> CompletableFuture.completedFuture(scaleMetadataList))
                 .when(mockControllerService).getScaleRecords(anyString(), anyString(), anyLong(), anyLong());
 
@@ -105,6 +112,8 @@
         doAnswer(x -> completableFuture)
                 .when(mockControllerService).getScaleRecords(anyString(), anyString(), anyLong(), anyLong());
 
-        response = addAuthHeaders(client.target(resourceURI).request()).buildGet().invoke();
+        response = addAuthHeaders(client.target(resourceURI)
+                                        .queryParam("from", fromDateTime)
+                                        .queryParam("to", toDateTime).request()).buildGet().invoke();
         assertEquals("Get Scaling Events response code", 500, response.getStatus());
     }
\ No newline at end of file
