diff --git a/old_test.java b/new_test.java
index 164ccdf..cd0e1a0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,9 +2,9 @@
 	public void vertical() {
 		Kernel1D_I32 kernel = new Kernel1D_I32(new int[]{1,2,3,4,5,6}, 6, 4);
 
-		ImageUInt8 input = new ImageUInt8(15,16);
+		GrayU8 input = new GrayU8(15,16);
 		ImageMiscOps.fillUniform(input, rand, 0, 50);
-		ImageUInt8 output = new ImageUInt8(15,16);
+		GrayU8 output = new GrayU8(15,16);
 
 		ConvolveNormalizedNaive.vertical(kernel, input, output);
 
