diff --git a/old_test.java b/new_test.java
index 97e2323..a7cc783 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
-    public void test_withYearOfEra_BE() {
+    public void test_withYear_BE() {
         ChronoDate base = ThaiBuddhistChronology.INSTANCE.date(2555, 8, 29);
-        ChronoDate test = base.withYearOfEra(2554);
+        ChronoDate test = base.withYear(2554);
         assertEquals(test, ThaiBuddhistChronology.INSTANCE.date(2554, 8, 29));
     }
\ No newline at end of file
