diff --git a/old_product.java b/new_product.java
index a2a15b1..893c0b4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,11 @@
 public OffsetDate with(DateTimeField field, long newValue) {
         if (field instanceof LocalDateTimeField) {
+            LocalDateTimeField f = (LocalDateTimeField) field;
+            switch (f) {
+                case OFFSET_SECONDS: {
+                    return with(date, ZoneOffset.ofTotalSeconds(f.checkValidIntValue(newValue)));
+                }
+            }
             return with(date.with(field, newValue), offset);
         }
         return field.doSet(this, newValue);
