diff --git a/old_test.java b/new_test.java
index 232c483..bbc6d90 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,5 @@
 @Test(groups={"tck"})
-    public void test_with_LocalDate() {
-        OffsetDate base = OffsetDate.of(2008, 6, 30, OFFSET_PONE);
-        OffsetDate test = base.with(Year.of(2007));
-        assertEquals(test.toLocalDate(), LocalDate.of(2007, 6, 30));
+    public void test_with_adjustment_OffsetDate() {
+        OffsetDate test = TEST_2007_07_15_PONE.with(OffsetDate.of(2008, 6, 30, OFFSET_PTWO));
+        assertEquals(test, OffsetDate.of(2008, 6, 30, OFFSET_PTWO));
     }
\ No newline at end of file
