diff --git a/old_product.java b/new_product.java
index 6ef4f58..8b64c09 100644
--- a/old_product.java
+++ b/new_product.java
@@ -21,7 +21,7 @@
 			nodeManager.removeEntity(6);
 			
 			// Add node 7 using the NONE user.
-			node = new Node(7, 16, buildDate("2008-01-02 18:19:20"), OsmUser.NONE, -11, -12);
+			node = new Node(7, 16, buildDate("2008-01-02 18:19:20"), OsmUser.NONE, 0, -11, -12);
 			node.getTags().addAll(
 					Arrays.asList(new Tag[]{new Tag("created_by", "Me7"), new Tag("change", "new node")}));
 			nodeManager.addEntity(node);
