diff --git a/old_test.java b/new_test.java
index 2a7b163..f11e93d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
 		testOsmWriter.process(
 				new NodeContainer(
 						new Node(
-								1234, 0, new Date(), new OsmUser(12, "OsmosisTest"), new ArrayList<Tag>(), 20, 20)));
+								1234, 0, new Date(), new OsmUser(12, "OsmosisTest"), 0, new ArrayList<Tag>(), 20, 20)));
 		// Nothing to assert; just expect no exception
 	}
\ No newline at end of file
