diff --git a/old_product.java b/new_product.java
index 551025e..fda79d8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -42,13 +42,11 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
         trainingParameters.setTextExtractorParameters(exParams);
         
         instance.fit(dataset, trainingParameters);
-        
 
-        /*
-        //TODO: restore this test
-        ClassificationMetrics.Metrics vm = instance.validate(dataset);
+
+        ClassificationMetrics vm = instance.validate(dataset);
         assertEquals(expectedF1score, vm.getMacroF1(), Constants.DOUBLE_ACCURACY_HIGH);
-        */
+
         instance.close();
         //instance = null;
         
