diff --git a/old_product.java b/new_product.java
index 1724e4b..f632101 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public static byte[] nextBytes(int count) {
+public static byte[] nextBytes(final int count) {
         Validate.isTrue(count >= 0, "Count cannot be negative.");
 
-        byte[] result = new byte[count];
+        final byte[] result = new byte[count];
         RANDOM.nextBytes(result);
         return result;
     }
\ No newline at end of file
