diff --git a/old_test.java b/new_test.java
index a3b87cb..ecf8b53 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,8 +9,9 @@
 		
 		String[] propertyNames = new String[] { "changedBy", "dateChanged" };
 		Object[] currentState = new Object[] { "", null };
+		Object[] previousState = new Object[] { "", null };
 		
-		interceptor.onFlushDirty(u, null, currentState, null, propertyNames, null);
+		interceptor.onFlushDirty(u, null, currentState, previousState, propertyNames, null);
 		
 		Assert.assertNotNull(currentState[0]);
 	}
\ No newline at end of file
