diff --git a/old_product.java b/new_product.java
index 01f2666..9aace3d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -33,8 +33,10 @@ public static void fetchFile(URL url, File outputPath, boolean unzip) throws Dow
         BufferedOutputStream writer = null;
         try {
             InputStream reader;
-            if (unzip) {
+            if (unzip || (encoding != null && "gzip".equalsIgnoreCase(encoding))) {
                 reader = new GZIPInputStream(conn.getInputStream());
+            } else if (encoding != null && "deflate".equalsIgnoreCase(encoding)) {
+                reader = new InflaterInputStream(conn.getInputStream());
             } else {
                 reader = conn.getInputStream();
             }
