diff --git a/old_product.java b/new_product.java
index a10139b..e5802fb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,7 +7,5 @@ private static void deletePerformance(final IdPreMap m, final DummyIdPreMap b) {
       b.delete(d[i][0], d[i][1], -1);
     }
     // perform the test:
-    final Performance p = new Performance();
     for(final int[] dd : d) m.delete(dd[0], dd[1], -1);
-    if(VERBOSE) Util.errln(d.length + " records deleted in: " + p);
   }
\ No newline at end of file
