diff --git a/old_test.java b/new_test.java
index 529e5b6..3a13a98 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testLoadComponentMetadata() throws Exception {
         copyDir("list1", registryPath);
-        ComponentInfo info = storage.loadComponentMetadata("fastr");
+        ComponentInfo info = loadLastComponent("fastr");
         assertEquals("org.graalvm.fastr", info.getId());
         assertEquals("1.0", info.getVersionString());
         assertEquals("0.32", info.getRequiredGraalValues().get("graalvm_version"));
