diff --git a/old_test.java b/new_test.java
index 30bb365..373eba5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,14 +2,15 @@
     public void testCreateCommand() throws GenieException {
         final String id = UUID.randomUUID().toString();
         final Command command = new Command.Builder(
-                COMMAND_1_NAME,
-                COMMAND_1_USER,
-                COMMAND_1_VERSION,
-                CommandStatus.ACTIVE,
-                COMMAND_1_EXECUTABLE
+            COMMAND_1_NAME,
+            COMMAND_1_USER,
+            COMMAND_1_VERSION,
+            CommandStatus.ACTIVE,
+            COMMAND_1_EXECUTABLE,
+            COMMAND_1_CHECK_DELAY
         )
-                .withId(id)
-                .build();
+            .withId(id)
+            .build();
         final String createdId = this.service.createCommand(command);
         Assert.assertThat(createdId, Matchers.is(id));
         final Command created = this.service.getCommand(id);
@@ -19,14 +20,15 @@
         Assert.assertEquals(COMMAND_1_USER, created.getUser());
         Assert.assertEquals(CommandStatus.ACTIVE, created.getStatus());
         Assert.assertEquals(COMMAND_1_EXECUTABLE, created.getExecutable());
+        Assert.assertThat(COMMAND_1_CHECK_DELAY, Matchers.is(created.getCheckDelay()));
         this.service.deleteCommand(id);
         try {
             this.service.getCommand(id);
             Assert.fail("Should have thrown exception");
         } catch (final GenieException ge) {
             Assert.assertEquals(
-                    HttpURLConnection.HTTP_NOT_FOUND,
-                    ge.getErrorCode()
+                HttpURLConnection.HTTP_NOT_FOUND,
+                ge.getErrorCode()
             );
         }
     }
\ No newline at end of file
