diff --git a/old_product.java b/new_product.java
index ebe1ce2..105e95f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,6 +16,7 @@
         commandEntity.setVersion(command.getVersion());
         commandEntity.setDescription(command.getDescription());
         commandEntity.setExecutable(command.getExecutable());
+        commandEntity.setCheckDelay(command.getCheckDelay());
         commandEntity.setConfigs(command.getConfigs());
         commandEntity.setSetupFile(command.getSetupFile());
         commandEntity.setStatus(command.getStatus());
