diff --git a/old_product.java b/new_product.java
index 11ebf94..6391119 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @Override
 	@Transactional(readOnly = true)
 	public List<Location> getLocationsHavingAnyTag(List<LocationTag> tags) throws APIException {
-		List<Location> locations = new ArrayList<Location>();
+		List<Location> locations = new ArrayList<>();
 		
 		for (Location loc : dao.getAllLocations(false)) {
 			for (LocationTag t : tags) {
