diff --git a/old_test.java b/new_test.java
index d0659cf..4642414 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
 	public void getLocationsHavingAnyTag_shouldReturnEmptyListWhenGivenAnEmptyTagList() {
 		LocationService ls = Context.getLocationService();
-		Assert.assertEquals(0, ls.getLocationsHavingAnyTag(new ArrayList<LocationTag>()).size());
+		Assert.assertEquals(0, ls.getLocationsHavingAnyTag(new ArrayList<>()).size());
 	}
\ No newline at end of file
