diff --git a/old_test.java b/new_test.java
index afa210f..d661c5a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test(groups={"tck"})
     public void test_get_DateTimeField() {
-        assertEquals(TEST_2008_06.get(LocalDateTimeField.YEAR), 2008);
-        assertEquals(TEST_2008_06.get(LocalDateTimeField.MONTH_OF_YEAR), 6);
-        assertEquals(TEST_2008_06.get(LocalDateTimeField.YEAR_OF_ERA), 2008);
-        assertEquals(TEST_2008_06.get(LocalDateTimeField.ERA), 1);
-        assertEquals(TEST_2008_06.get(LocalDateTimeField.EPOCH_MONTH), (2008 - 1970) * 12 + 6 - 1);
+        assertEquals(TEST_2008_06.getLong(LocalDateTimeField.YEAR), 2008);
+        assertEquals(TEST_2008_06.getLong(LocalDateTimeField.MONTH_OF_YEAR), 6);
+        assertEquals(TEST_2008_06.getLong(LocalDateTimeField.YEAR_OF_ERA), 2008);
+        assertEquals(TEST_2008_06.getLong(LocalDateTimeField.ERA), 1);
+        assertEquals(TEST_2008_06.getLong(LocalDateTimeField.EPOCH_MONTH), (2008 - 1970) * 12 + 6 - 1);
     }
\ No newline at end of file
