diff --git a/old_test.java b/new_test.java
index fdedfa7..44f83bf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,13 +11,14 @@
     GraphState<LongWritable, IntWritable, FloatWritable, IntWritable> gs =
         new GraphState<LongWritable, IntWritable,
         FloatWritable, IntWritable>();
+    ImmutableClassesGiraphConfiguration configuration =
+        new ImmutableClassesGiraphConfiguration(job.getConfiguration());
     Vertex<LongWritable, IntWritable, FloatWritable, IntWritable> vertex =
-        BspUtils.createVertex(job.getConfiguration());
-    vertex.initialize(null, null, null, null);
+        configuration.createVertex();
     System.out.println("testInstantiateVertex: Got vertex " + vertex +
         ", graphState" + gs);
     VertexInputFormat<LongWritable, IntWritable, FloatWritable, IntWritable>
-    inputFormat = BspUtils.createVertexInputFormat(job.getConfiguration());
+    inputFormat = configuration.createVertexInputFormat();
     /*if[HADOOP_NON_SASL_RPC]
       List<InputSplit> splitArray =
           inputFormat.getSplits(
