diff --git a/old_product.java b/new_product.java
index 27f2d2f..44a75ae 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public void setOriginToCurrent() {
 		IT currToWorld = (IT)worldToCurr.invert(null);
-		IT oldWorldToNewWorld = (IT)worldToInit.concat(currToWorld,null);
+		IT oldWorldToNewWorld = (IT) worldToInit.concat(currToWorld,null);
 
 		PixelTransform_F32 newToOld = converter.convertPixel(oldWorldToNewWorld,null);
 
@@ -17,6 +17,7 @@ public void setOriginToCurrent() {
 
 		// have motion estimates be relative to this frame
 		motion.setToFirst();
-		worldToCurr.reset();
 		first = true;
+
+		computeCurrToInit_PixelTran();
 	}
\ No newline at end of file
