diff --git a/old_product.java b/new_product.java
index 144d9bc..56bf3f9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static void throwIfClosed(boolean closed, Object targetObject) throws ObjectClosedException {
+public static void checkNotClosed(boolean closed, Object targetObject) throws ObjectClosedException {
         if (closed) {
             throw new ObjectClosedException(targetObject);
         }
