diff --git a/old_test.java b/new_test.java
index 835bf3b..e70007a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,4 +4,9 @@
         assertPeriod(Period.of(1, 2, 3).plusYears(10), 11, 2, 3);
         assertPeriod(Period.of(1, 2, 3).plusYears(-10), -9, 2, 3);
         assertPeriod(Period.of(1, 2, 3).plusYears(-1), 0, 2, 3);
+
+        assertPeriod(Period.of(1, 2, 3).plus(Period.ofYears(0)), 1, 2, 3);
+        assertPeriod(Period.of(1, 2, 3).plus(Period.ofYears(10)), 11, 2, 3);
+        assertPeriod(Period.of(1, 2, 3).plus(Period.ofYears(-10)), -9, 2, 3);
+        assertPeriod(Period.of(1, 2, 3).plus(Period.ofYears(-1)), 0, 2, 3);
     }
\ No newline at end of file
