diff --git a/old_product.java b/new_product.java
index 164a530..e42276f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,7 @@
-@SuppressWarnings("unchecked")
-  public List<IdentifiedLanguage> identify(final String text) {
+public ServiceCall<List<IdentifiedLanguage>> identify(final String text) {
     final Request request = RequestBuilder.post(PATH_IDENTIFY)
-        .withHeader(HttpHeaders.ACCEPT, HttpMediaType.APPLICATION_JSON)
-        .withBodyContent(text, HttpMediaType.TEXT_PLAIN).build();
+            .withHeader(HttpHeaders.ACCEPT, HttpMediaType.APPLICATION_JSON)
+            .withBodyContent(text, HttpMediaType.TEXT_PLAIN).build3();
 
-    final LanguageList languages = executeRequest(request, LanguageList.class);
-
-    return (List<IdentifiedLanguage>) (List<?>) languages.getLanguages();
+    return  createServiceCall(createCall(request), ResponseUtil.getLanguageListIdentifier());
   }
\ No newline at end of file
