diff --git a/old_test.java b/new_test.java
index 7739d10..d32a8e7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void closeTest()
       throws IOException, RestServiceException {
-    NettyRequest nettyRequest = createNettyRequest(HttpMethod.GET, "/", null);
+    NettyRequest nettyRequest = createNettyRequest(HttpMethod.POST, "/", null);
     Queue<HttpContent> httpContents = new LinkedBlockingQueue<HttpContent>();
     for (int i = 0; i < 5; i++) {
       ByteBuffer content = ByteBuffer.wrap(getRandomBytes(1024));
