diff --git a/old_test.java b/new_test.java
index 5987d1d..5d75bdf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,23 +1,30 @@
 @Test
-	public void testInit() throws IOException {
+	public void testAgentConfigInitialization() throws IOException {
+		// Given
 		AgentConfig config = new AgentConfig();
-		config.init();
+		config.init(true);
 		File homeDir = config.getHome().getDirectory();
-		System.out.println("Home:" + homeDir.getAbsolutePath());
-		
+		// When
 		config.saveAgentPidProperties("1000", "agent");
-		String pid  = config.getAgentPidProperties("agent");
-		assertThat(pid, is("1000"));
-		
+		String pid = config.getAgentPidProperties("agent");
+		// Then
+		assertThat(pid).isEqualTo("1000");
+
+		//  When
 		config.saveAgentPidProperties("1001", "monitor");
-		String monitorPid  = config.getAgentPidProperties("monitor");
-		assertThat(monitorPid, is("1001"));
+		String monitorPid = config.getAgentPidProperties("monitor");
 
-		assertThat(config.isTestMode(), is(false));
+		// Then
+		assertThat(monitorPid).isEqualTo("1001");
+		assertThat(config.isTestMode()).isEqualTo(false);
 
+		// When
 		System.setProperty("ngrinder.agent.home", "./tmp_agent_home");
-		config.init();
+		config.init(true);
+		// Then
 		homeDir = config.getHome().getDirectory();
-		assertTrue(homeDir.getAbsolutePath().contains("tmp_agent_home"));
+		assertThat(homeDir.getAbsolutePath()).contains(File.separator + "tmp_agent_home");
 		FileUtils.deleteDirectory(homeDir);
+		System.setProperty("ngrinder.agent.home", "");
+
 	}
\ No newline at end of file
