diff --git a/old_product.java b/new_product.java
index 4f5079a..3b2e64f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @PostConstruct
 	public void init() {
 		try {
-			CoreLogger.LOGGER.info("NGrinder is starting...");
+			CoreLogger.LOGGER.info("nGrinder is starting...");
 			home = resolveHome();
 			exHome = resolveExHome();
 			copyDefaultConfigurationFiles();
@@ -13,6 +13,7 @@
 			cluster = getSystemProperties().getPropertyBoolean(NGrinderConstants.NGRINDER_PROP_CLUSTER_MODE, false);
 			initLogger(isTestMode());
 			resolveLocalIp();
+			loadAnnouncement();
 			loadDatabaseProperties();
 			versionString = getVersion();
 		} catch (IOException e) {
