diff --git a/old_test.java b/new_test.java
index 98bb5f7..56d51a9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
-@Test(groups={"tck"}, dataProvider="plusDays")
+@Test(dataProvider="plusDays")
     public void test_minus_TemporalAmount_Period_days(ZonedDateTime base, int amount, ZonedDateTime expected) {
         assertEquals(base.minus(Period.ofDays(-amount)), expected);
     }
\ No newline at end of file
