diff --git a/old_product.java b/new_product.java
index 94afb42..0f1cddb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
-  public void assign(String variable, String expression, JSONObject message, Context stellarContext) {
-    Object result = execute(expression, message, stellarContext);
+  public void assign(String variable, String expression, Map<String, Object> message) {
+    Object result = execute(expression, message);
     state.put(variable, result);
   }
\ No newline at end of file
