diff --git a/old_product.java b/new_product.java
index e7c7dd4..43bd3c2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 public CreateFileTOptions toThrift() {
     CreateFileTOptions options = new CreateFileTOptions();
     options.setBlockSizeBytes(mBlockSizeBytes);
+    options.setMode(mMode.toShort());
     options.setPersisted(mWriteType.getUnderStorageType().isSyncPersist());
     options.setRecursive(mRecursive);
     options.setTtl(mTtl);
