diff --git a/old_product.java b/new_product.java
index 140b4f2..1a536fb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,7 +9,7 @@ public int rename(String[] argv) throws IOException {
         System.out.println("mv: Failed to rename " + srcPath + " to " + dstPath);
         return -1;
       }
-    } catch (TException e) {
+    } catch (TachyonException e) {
       throw new IOException(e.getMessage());
     }
   }
\ No newline at end of file
