diff --git a/old_product.java b/new_product.java
index ac25668..495432d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,6 @@
-public ServiceCall<WordData> getWord(String customizationId, String wordName) {
-    Validator.notNull(customizationId, "customizationId cannot be null");
-    Validator.notNull(wordName, "wordName cannot be null");
-
-    RequestBuilder requestBuilder = RequestBuilder.get(String.format(PATH_WORD, customizationId, wordName));
-    return createServiceCall(requestBuilder.build(), ResponseConverterUtils.getObject(WordData.class));
+public ServiceCall<Word> getWord(GetWordOptions getWordOptions) {
+    Validator.notNull(getWordOptions, "getWordOptions cannot be null");
+    RequestBuilder builder = RequestBuilder.get(String.format("/v1/customizations/%s/words/%s", getWordOptions
+        .customizationId(), getWordOptions.wordName()));
+    return createServiceCall(builder.build(), ResponseConverterUtils.getObject(Word.class));
   }
\ No newline at end of file
