diff --git a/old_test.java b/new_test.java
index 60ad465..90a8b5b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,11 @@
 
     server.enqueue(new MockResponse().addHeader(CONTENT_TYPE, HttpMediaType.APPLICATION_JSON).setBody(wordAsStr));
 
-    Word result = service.getWord(id, wordName).execute();
+    GetWordOptions getOptions = new GetWordOptions.Builder()
+        .customizationId(id)
+        .wordName(wordName)
+        .build();
+    Word result = service.getWord(getOptions).execute();
     final RecordedRequest request = server.takeRequest();
 
     assertEquals("GET", request.getMethod());
