diff --git a/old_test.java b/new_test.java
index 835a5c9..51cd39b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
 				EWAHCompressedBitmap32[] x = i.next();
 				EWAHCompressedBitmap32 tanswer = EWAHCompressedBitmap32.and(x);
 				EWAHCompressedBitmap32 x1 = IteratorUtil32
-						.materialize(IteratorAggregation32.and(IteratorUtil32
+						.materialize(IteratorAggregation32.bufferedand(IteratorUtil32
 								.toIterators(x)));
 				assertTrue(x1.equals(tanswer));
 			}
