diff --git a/old_product.java b/new_product.java
index c589101..92e708b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static IteratingRLW and(final int bufsize, final IteratingRLW... al) {
+public static IteratingRLW bufferedand(final int bufsize, final IteratingRLW... al) {
 		if (al.length == 0)
 			throw new IllegalArgumentException("Need at least one iterator");
 		if (al.length == 1)
@@ -6,5 +6,5 @@ public static IteratingRLW and(final int bufsize, final IteratingRLW... al) {
 		final LinkedList<IteratingRLW> basell = new LinkedList<IteratingRLW>();
 		for (IteratingRLW i : al) 
 			basell.add(i);
-		return new BufferedIterator(new AndIt(basell,bufsize));
+		return new BufferedIterator(new BufferedAndIterator(basell,bufsize));
 	}
\ No newline at end of file
