diff --git a/old_product.java b/new_product.java
index fd95622..815ea60 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,7 +9,7 @@
             view.set(location, proxy.realmGet$proxyState().getRow$realm().getIndex());
             return oldObject;
         } else {
-            oldObject = nonManagedList.set(location, object);
+            oldObject = unmanagedList.set(location, object);
         }
         return oldObject;
     }
\ No newline at end of file
