diff --git a/old_product.java b/new_product.java
index 365778f..3020338 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
-public DeviceId vRouterId() {
+public Optional<DeviceId> vRouterId() {
         if (!object.has(VROUTER_ID)) {
-            return null;
+            return Optional.empty();
         }
 
         try {
-            return DeviceId.deviceId(object.path(VROUTER_ID).asText());
+            return Optional.of(DeviceId.deviceId(object.path(VROUTER_ID).asText()));
         } catch (IllegalArgumentException e) {
             return null;
         }
