diff --git a/old_test.java b/new_test.java
index 32729ad..52e8ab3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,6 @@
 @Test
     public void testVRouterId() throws Exception {
-        assertThat(config.vRouterId(), is(VROUTER_ID_1));
+        Optional<DeviceId> vRouterId = config.vRouterId();
+        assertTrue(vRouterId.isPresent());
+        assertThat(vRouterId.get(), is(VROUTER_ID_1));
     }
\ No newline at end of file
