diff --git a/old_product.java b/new_product.java
index c1bbea7..b88f400 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,6 +4,7 @@ public void validate(Object obj, Errors errors) {
 			errors.rejectValue("program", "error.general");
 		} else {
 			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "name", "error.name");
+			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "description", "error.required");
 			List<Program> programs = Context.getProgramWorkflowService().getAllPrograms(false);
 			for (Program program : programs) {
 				if (program.getName().equals(p.getName()) && !program.getProgramId().equals(p.getProgramId())) {
