diff --git a/old_test.java b/new_test.java
index 6bc2c9b..d22fb6e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
 	@Verifies(value = "should fail validation if program name already in use", method = "validate(Object,Errors)")
 	public void validate_shouldFailValidationIfProgramNameAlreadyInUse() throws Exception {
 		Program prog = new Program();
-		prog.setName("MDR program");
+		prog.setName("MDR-TB PROGRAM");
 		prog.setConcept(Context.getConceptService().getConcept(3));
 		
 		Errors errors = new BindException(prog, "prog");
