diff --git a/old_product.java b/new_product.java
index 2964376..42ede6c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,8 +11,8 @@ protected Range findDomainBounds(XYDataset dataset,
                     visibleSeriesKeys.add(dataset.getSeriesKey(s));
                 }
             }
-            return DatasetUtilities.findDomainBounds(dataset,
+            return DatasetUtils.findDomainBounds(dataset,
                     visibleSeriesKeys, includeInterval);
         }
-        return DatasetUtilities.findDomainBounds(dataset, includeInterval);
+        return DatasetUtils.findDomainBounds(dataset, includeInterval);
     }
\ No newline at end of file
