diff --git a/old_test.java b/new_test.java
index fee616a..9eb92c7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,9 +8,6 @@
     Map<String, Service> services = new HashMap<>();
     Mockito.when(cluster.getServices()).thenReturn(services);
 
-    ClusterVersionEntity clusterVersionEntity = Mockito.mock(ClusterVersionEntity.class);
-    Mockito.when(cluster.getCurrentClusterVersion()).thenReturn(clusterVersionEntity);
-
     PrereqCheckRequest request = new PrereqCheckRequest("cluster");
     request.setTargetStackId(new StackId("HDP", "2.3.1.1"));
     request.setSourceStackId(new StackId("HDP", "2.3.0.0"));
