diff --git a/old_product.java b/new_product.java
index 4ab528b..14f79ec 100644
--- a/old_product.java
+++ b/new_product.java
@@ -26,10 +26,11 @@
       if(minStack.length == 2) {
         String minStackName = minStack[0];
         String minStackVersion = minStack[1];
-        String stackName = cluster.getCurrentStackVersion().getStackName();
+        Service yarnService = cluster.getService("YARN");
+        String stackName = yarnService.getDesiredStackId().getStackName();
         if (minStackName.equals(stackName)) {
-          String currentClusterRepositoryVersion = cluster.getCurrentClusterVersion().getRepositoryVersion().getVersion();
-          return VersionUtils.compareVersions(currentClusterRepositoryVersion, minStackVersion) >= 0;
+          String currentRepositoryVersion = yarnService.getDesiredRepositoryVersion().getVersion();
+          return VersionUtils.compareVersions(currentRepositoryVersion, minStackVersion) >= 0;
         }
       }
     }
