diff --git a/old_test.java b/new_test.java
index 42085d5..eb244ff 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,6 @@
                         new User("刘备", 30),
                         new User("赵云", 50));
 
-        Predicate<User> comparatorPredicate = BeanPredicateUtil.comparatorPredicate("age", 30, Criterion.LESS);
         Map<String, Integer> map = AggregateUtil.groupCount(list, "name", comparatorPredicate);
         assertThat(map, allOf(hasEntry("刘备", 1), hasEntry("赵云", 2)));
     }
\ No newline at end of file
