diff --git a/old_product.java b/new_product.java
index 1f75ebc..23aef0c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,12 +4,8 @@ public static <T, O> Map<T, Integer> groupCount(Iterable<O> beanIterable,String
         }
         Validate.notBlank(propertyName, "propertyName can't be null/empty!");
 
-        Map<T, Integer> map = new LinkedHashMap<>();
-        for (O obj : beanIterable){
-            if (null != includePredicate && !includePredicate.evaluate(obj)){
-                continue;
-            }
-            MapUtil.putSumValue(map, PropertyUtil.<T> getProperty(obj, propertyName), 1);
-        }
-        return map;
+        //---------------------------------------------------------------
+
+        Map<String, Map<T, Integer>> groupCount = groupCount(beanIterable, toArray(propertyName), includePredicate);
+        return groupCount.get(propertyName);
     }
\ No newline at end of file
