diff --git a/old_test.java b/new_test.java
index 5914171..cbe1805 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-	public void saveOrder_shouldFailIfAnActiveOrderForTheSameConceptExists() throws Exception {
+	public void saveOrder_shouldFailIfAnActiveOrderForTheSameConceptAndCareSettingExists() throws Exception {
 		final Patient patient = patientService.getPatient(2);
 		final Concept cd4Count = conceptService.getConcept(5497);
 		//sanity check that we have an active order for the same concept
@@ -13,8 +13,9 @@
 		order.setConcept(cd4Count);
 		order.setEncounter(encounterService.getEncounter(6));
 		order.setOrderer(providerService.getProvider(1));
+		order.setCareSetting(duplicateOrder.getCareSetting());
 		
 		expectedException.expect(APIException.class);
-		expectedException.expectMessage("Cannot have more than one active order for the same concept");
+		expectedException.expectMessage("Cannot have more than one active order for the same concept and care setting");
 		orderService.saveOrder(order, null);
 	}
\ No newline at end of file
