diff --git a/old_product.java b/new_product.java
index 1c1fcf8..50b236e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,10 +12,10 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 			concept = ((DrugOrder) order).getDrug().getConcept();
 		}
 		if (!isDiscontinueOrReviseOrder(order)) {
-			List<Order> activeOrders = getActiveOrders(order.getPatient(), null, null, null);
+			List<Order> activeOrders = getActiveOrders(order.getPatient(), null, order.getCareSetting(), null);
 			for (Order o : activeOrders) {
 				if (o.getConcept().equals(concept)) {
-					throw new APIException("Cannot have more than one active order for the same concept");
+					throw new APIException("Cannot have more than one active order for the same concept and care setting");
 				}
 			}
 		}
