diff --git a/old_product.java b/new_product.java
index 2d1a537..951c9e4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,7 +7,8 @@ public static String sanitizeBasicHTML(String unsafe){
         //Whitelist wl = Whitelist.basic().addTags("img", "h1", "h2", "h3", "kbd", "hr", "s", "del");  
 
         Whitelist wl = Whitelist.basicWithImages().addTags( "h1", "h2", "h3", "kbd", "hr", "s", "del","map","area").addAttributes("img", "usemap")
-                .addAttributes("map", "name").addAttributes("area", "shape","coords","href","title","alt").addAttributes("a", "target");
+                .addAttributes("map", "name").addAttributes("area", "shape","coords","href","title","alt")
+                .addEnforcedAttribute("a", "target", "_blank");
 
         return Jsoup.clean(unsafe, wl);
         
