diff --git a/old_test.java b/new_test.java
index c636da9..d0ea3d2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -43,9 +43,15 @@
         assertTrue(safeStr.equals(sanitized));
 
         unsafeStr = "the <a href=\"http://dataverse.org\" target=\"_blank\">Dataverse project</a> in a new window";
-        safeStr = "the \n<a href=\"http://dataverse.org\" target=\"_blank\" rel=\"nofollow\">Dataverse project</a> in a new window";
+        safeStr = "the \n<a href=\"http://dataverse.org\" rel=\"nofollow\" target=\"_blank\">Dataverse project</a> in a new window";
         sanitized = MarkupChecker.sanitizeBasicHTML(unsafeStr);
         this.msgu("safeStr: " + safeStr + "\nsanitized: " + sanitized);
         assertTrue(safeStr.equals(sanitized));
+        
+        unsafeStr = "the <a href=\"http://dataverse.org\">Dataverse project</a> in a new window";
+        safeStr = "the \n<a href=\"http://dataverse.org\" rel=\"nofollow\" target=\"_blank\">Dataverse project</a> in a new window";
+        sanitized = MarkupChecker.sanitizeBasicHTML(unsafeStr);
+        this.msgu("safeStr: " + safeStr + "\nsanitized: " + sanitized);
+        assertTrue(safeStr.equals(sanitized));        
 
     }
\ No newline at end of file
