diff --git a/old_product.java b/new_product.java
index e8a1ced..9fb9ca6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,14 +1,15 @@
-private Item writeBinary(final File path, final QueryContext ctx)
-      throws QueryException {
+private Item writeBinary(final File path, final QueryContext ctx,
+      final boolean append) throws QueryException {
 
-    final B64 b64 = (B64) checkType(expr[1].item(ctx, input), AtomType.B64);
-    final boolean append = optionalBool(2, ctx);
     if(path.isDirectory()) PATHISDIR.thrw(input, path);
 
     try {
       final FileOutputStream out = new FileOutputStream(path, append);
       try {
-        out.write(b64.toJava());
+        final Iter ir = expr[1].iter(ctx);
+        for(Item it; (it = ir.next()) != null;) {
+          out.write(((B64) checkType(it, AtomType.B64)).toJava());
+        }
       } finally {
         out.close();
       }
