diff --git a/old_product.java b/new_product.java
index 72c32ae..f09d202 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static void applyPixelNormalization(RowMatrix_F64 N, Point2D_F64 orig, Point2D_F64 normed) {
+public static void applyPixelNormalization(DMatrixRMaj N, Point2D_F64 orig, Point2D_F64 normed) {
 		normed.x = orig.x * N.data[0] + N.data[2];
 		normed.y = orig.y * N.data[4] + N.data[5];
 	}
\ No newline at end of file
