diff --git a/old_product.java b/new_product.java
index ad514b5..3e4c109 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
     public void removeFromGroup(String featureId, String groupName) {
-        getCacheManager().evict(featureId);
         getTarget().removeFromGroup(featureId, groupName);
+        getCacheManager().evict(featureId);
     }
\ No newline at end of file
