diff --git a/old_test.java b/new_test.java
index ecbf066..274b865 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,12 +2,12 @@
     @Transactional
     public void testRemoveFromGroup() {
         // Given
-        assertFf4j.assertGroupSize(2, G1);
+        assertFf4j.assertThatGroupHasSize(2, G1);
         // When
         testedStore.removeFromGroup(F3, G1);
         // Then
-        assertFf4j.assertGroupSize(1, G1);
+        assertFf4j.assertThatGroupHasSize(1, G1);
         // End, Return to initial state
         testedStore.addToGroup(F3, G1);
-        assertFf4j.assertGroupSize(2, G1);
+        assertFf4j.assertThatGroupHasSize(2, G1);
     }
\ No newline at end of file
