diff --git a/old_test.java b/new_test.java
index e0e5477..0751c71 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,7 +17,7 @@
 		order.setDrug(discontinuationOrderDrug);
 		order.setOrderReasonNonCoded("Discontinue this");
 		
-		expectedException.expect(APIException.class);
+		expectedException.expect(EditedOrderDoesNotMatchPreviousException.class);
 		expectedException.expectMessage("The orderable of the previous order and the new one order don't match");
 		orderService.saveOrder(order, null);
 	}
\ No newline at end of file
