diff --git a/old_test.java b/new_test.java
index cff63ba..02863c3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,13 +18,11 @@
         streamManager.createScope(scope);
         Assert.assertTrue("Stream creation is not successful ",
                 streamManager.createStream(scope, streamName, StreamConfiguration.builder()
-                                   .scope(scope)
-                                   .streamName(streamName)
                                    .scalingPolicy(ScalingPolicy.fixed(numSegments))
                                    .build()));
         log.info("Created stream: " + streamName);
 
-        ClientFactory clientFactory = ClientFactory.withScope(scope, clientConfig);
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scope, clientConfig);
         EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName,
                 new JavaSerializer<String>(),
                 EventWriterConfig.builder().build());
