diff --git a/old_product.java b/new_product.java
index 70d92cf..a4fd1a9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,8 +9,6 @@ public void createTestStream(final String streamName, final int numSegments)
         streamManager.createScope(scope);
         streamManager.createStream(scope, streamName,
                                    StreamConfiguration.builder()
-                                                      .scope(scope)
-                                                      .streamName(streamName)
                                                       .scalingPolicy(ScalingPolicy.fixed(numSegments))
                                                       .build());
         log.info("Created stream: " + streamName);
