diff --git a/old_test.java b/new_test.java
index a157b0b..feb4ffe 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,5 +3,5 @@
 	public void findEncounterTypes_shouldReturnTypesByPartialCaseInsensitiveMatch() throws Exception {
 		EncounterService encounterService = Context.getEncounterService();
 		List<EncounterType> types = encounterService.findEncounterTypes("Test ENC");
-		assertEquals(5, types.size());
+		assertEquals(3, types.size());
 	}
\ No newline at end of file
