diff --git a/old_product.java b/new_product.java
index 34ed389..ad6a069 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery eqProp(String propName, String otherProp) {
-        addCriterion(criterionBuilder.eqProp(propName, otherProp));
+    	criterion = criterion.and(criterionBuilder.eqProp(propName, otherProp));
         return this;
     }
\ No newline at end of file
