diff --git a/old_test.java b/new_test.java
index 16090c0..eaa370d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,17 @@
 @Test
-	@Verifies(value = "should delete property from database if not in list", method = "saveGlobalProperties(List<QGlobalProperty;>)")
-	public void saveGlobalProperties_shouldDeletePropertyFromDatabaseIfNotInList() throws Exception {
-		List<GlobalProperty> globalProperties = Context.getAdministrationService().getAllGlobalProperties();
-		GlobalProperty firstGlobalProperty = globalProperties.remove(0);
-		Context.getAdministrationService().saveGlobalProperties(globalProperties);
-		Assert.assertNull(Context.getAdministrationService().getGlobalProperty(firstGlobalProperty.getProperty()));
+	public void purgeGlobalProperties_shouldDeleteGlobalPropertiesFromDatabase() throws Exception {
+		int originalSize = adminService.getAllGlobalProperties().size();
+		
+		List<GlobalProperty> props = new ArrayList<GlobalProperty>();
+		props.add(new GlobalProperty("a.property.key", "something"));
+		props.add(new GlobalProperty("a.property.KEY", "somethingelse"));
+		adminService.saveGlobalProperties(props);
+		int afterSaveSize = adminService.getAllGlobalProperties().size();
+		
+		Assert.assertEquals(originalSize + 2, afterSaveSize);
+		
+		adminService.purgeGlobalProperties(props);
+		int afterPurgeSize = adminService.getAllGlobalProperties().size();
+		
+		Assert.assertEquals(originalSize, afterPurgeSize);
 	}
\ No newline at end of file
