diff --git a/old_product.java b/new_product.java
index 0a78605..81068b6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,13 +2,15 @@ public static void checkNearCacheConfig(String mapName, NearCacheConfig nearCach
                                             NativeMemoryConfig nativeMemoryConfig, boolean isClient) {
         checkNotNativeWhenOpenSource(nearCacheConfig.getInMemoryFormat());
         checkLocalUpdatePolicy(mapName, nearCacheConfig.getLocalUpdatePolicy());
-        checkEvictionConfig(nearCacheConfig.getEvictionConfig(), true);
+        EvictionConfig evictionConfig = nearCacheConfig.getEvictionConfig();
+        checkNearCacheEvictionConfig(evictionConfig.getEvictionPolicy(),
+                evictionConfig.getComparatorClassName(), evictionConfig.getComparator());
         checkOnHeapNearCacheMaxSizePolicy(nearCacheConfig);
         checkNearCacheNativeMemoryConfig(nearCacheConfig.getInMemoryFormat(),
                 nativeMemoryConfig, getBuildInfo().isEnterprise());
 
         if (isClient && nearCacheConfig.isCacheLocalEntries()) {
-            throw new IllegalArgumentException("The Near Cache option `cache-local-entries` is not supported in "
+            throw new InvalidConfigurationException("The Near Cache option `cache-local-entries` is not supported in "
                     + "client configurations.");
         }
         checkPreloaderConfig(nearCacheConfig, isClient);
