diff --git a/old_product.java b/new_product.java
index f0cca18..26287dc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
-public void connect() throws UnknownHostException, IOException
-    {
-        socket = new Socket(host, port);
-        out = new PrintWriter(socket.getOutputStream(), true);
-        in = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+public void connect(Socket aSocket) throws UnknownHostException, IOException
+   {
+      socket = aSocket;
+      out = new PrintWriter(socket.getOutputStream(), true);
+      in = new BufferedReader(new InputStreamReader(socket.getInputStream()));
 
-        remoteServerName = getData("name");
-    }
\ No newline at end of file
+      remoteServerName = getData("name");
+   }
\ No newline at end of file
