diff --git a/old_product.java b/new_product.java
index 214d538..5264e31 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,8 +4,4 @@
         @NotEmpty(message = "No tags entered. Unable to update.") final Set<String> tags
     ) throws GenieException {
         this.findCluster(id).setTags(this.createAndGetTagEntities(tags));
-        final ClusterEntity clusterEntity = this.findCluster(id);
-        final Set<TagEntity> newTags = this.createAndGetTagEntities(tags);
-        this.setFinalTags(newTags, clusterEntity.getUniqueId(), clusterEntity.getName());
-        clusterEntity.setTags(newTags);
     }
\ No newline at end of file
