diff --git a/old_test.java b/new_test.java
index 9223e30..42c8377 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,9 +4,9 @@
 	rule.onApply(result);
 	vector.addModification(new AddMessageModification(new ServerHelloDoneMessage(), new SendAction()));
 	rule.onApply(result);
-	HashMap<ModificationType, MutableInt> typeMap = rule.getTypeMap();
-	MutableInt val = typeMap.get(ModificationType.ADD_RECORD);
-	assertTrue(val.getValue() == 2);
-	val = typeMap.get(ModificationType.ADD_MESSAGE);
-	assertTrue(val.getValue() == 3);
+	List<ModificationCounter> counterList= rule.getCounterList();
+	ModificationCounter counter = counterList.get(1);
+	assertTrue(counter.getCounter() == 2);
+	counter = counterList.get(0);
+	assertTrue(counter.getCounter() == 3);
     }
\ No newline at end of file
