diff --git a/old_test.java b/new_test.java
index bfdf6da..98f7e67 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public void cloneByteBufferListTest() {
     final int bufferSize = 10;
     final int listLength = 10;
-    ArrayList<ByteBuffer> bufList = new ArrayList<ByteBuffer>(listLength);
+    ArrayList<ByteBuffer> bufList = new ArrayList<>(listLength);
     for (int k = 0; k < listLength; k++) {
       ByteBuffer buf = ByteBuffer.allocate(bufferSize);
       for (byte i = 0; i < bufferSize; i++) {
