diff --git a/old_product.java b/new_product.java
index 6f2acb3..4a4d16f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static List<ByteBuffer> cloneByteBufferList(List<ByteBuffer> source) {
-    List<ByteBuffer> ret = new ArrayList<ByteBuffer>(source.size());
+    List<ByteBuffer> ret = new ArrayList<>(source.size());
     for (ByteBuffer b : source) {
       ret.add(cloneByteBuffer(b));
     }
