diff --git a/old_product.java b/new_product.java
index 27a1004..f1883cb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,5 @@
 @Override
    public ListenableFuture<Void> removeBlob(final String container, final String key) {
-      if (getContainerToBlobs().containsKey(container)) {
-         getContainerToBlobs().get(container).remove(key);
-      }
+      storageStrategy.removeBlob(container, key);
       return immediateFuture(null);
    }
\ No newline at end of file
