diff --git a/old_test.java b/new_test.java
index 3fc108b..e9df456 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,11 @@
 @Test
-    public void testAddParameterSingleValueMap2(){
-        String beforeUrl = "www.baidu.com?a=b";
-        Map<String, String> singleValueMap = new LinkedHashMap<String, String>();
+    public void testAddParameterSingleValueMapWithParamByReplace(){
+        String beforeUrl = PATH + "?a=b&city=12345&name=feilong";
+        Map<String, String> singleValueMap = new LinkedHashMap<>();
 
         singleValueMap.put("province", "江苏省");
         singleValueMap.put("city", "南通市");
 
-        LOGGER.debug(ParamUtil.addParameterSingleValueMap(beforeUrl, singleValueMap, UTF8));
+        String expected = Slf4jUtil.format(PATH + "?a=b&city={}&name=feilong&province={}", encode("南通市", UTF8), encode("江苏省", UTF8));
+        assertEquals(expected, addParameterSingleValueMap(beforeUrl, singleValueMap, UTF8));
     }
\ No newline at end of file
