diff --git a/old_product.java b/new_product.java
index 0d2405f..e079fd6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,4 @@
 public static String addParameterSingleValueMap(String uriString,Map<String, String> singleValueMap,String charsetType){
+        Validate.notNull(singleValueMap, "singleValueMap can't be null!");
         return addParameterArrayValueMap(uriString, MapUtil.toArrayValueMap(singleValueMap), charsetType);
     }
\ No newline at end of file
