diff --git a/old_product.java b/new_product.java
index d6ce1fa..10de9fa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,10 +4,13 @@ public synchronized void delete(final ProgramId programId, final SchedulableProg
       .execute(new TransactionExecutor.Subroutine() {
         @Override
         public void apply() throws Exception {
-          byte[] rowKey = Bytes.toBytes(String.format("%s:%s", KEY_PREFIX,
-                                                      AbstractSchedulerService.scheduleIdFor(programId, programType,
-                                                                                             scheduleName)));
-          table.delete(rowKey);
+          String rowKey = getRowKey(programId, programType, scheduleName);
+
+          String versionLessRowKey = removeAppVersion(rowKey);
+          if (versionLessRowKey != null) {
+            table.delete(Bytes.toBytes(versionLessRowKey));
+          }
+          table.delete(Bytes.toBytes(rowKey));
         }
       });
   }
\ No newline at end of file
