diff --git a/old_test.java b/new_test.java
index 6ad6f25..ae146ed 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testGeProp() {
-        assertTrue(instance.geProp("id", "name").getQueryCriterions()
-                .contains(new GePropCriterion("id", "name")));
+    	assertEquals(new GePropCriterion("id", "name"), instance.geProp("id", "name").getQueryCriterion());
     }
\ No newline at end of file
