diff --git a/old_product.java b/new_product.java
index 0bb46a7..ce8b215 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public boolean belongTo(BlockStoreLocation location) {
+public boolean belongsTo(BlockStoreLocation location) {
     boolean tierInRange =
         tierAlias().equals(location.tierAlias()) || location.tierAlias().equals(ANY_TIER);
     boolean dirInRange = (dir() == location.dir()) || (location.dir() == ANY_DIR);
