diff --git a/old_product.java b/new_product.java
index 30bfc70..7a204d1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,12 @@
 public GlobalProperty saveGlobalProperty(GlobalProperty gp) throws DAOException {
-		sessionFactory.getCurrentSession().saveOrUpdate(gp);
-		return gp;
+		GlobalProperty gpObject = getGlobalPropertyObject(gp.getProperty());
+		if (gpObject != null) {
+			gpObject.setPropertyValue(gp.getPropertyValue());
+			gpObject.setDescription(gp.getDescription());
+			sessionFactory.getCurrentSession().update(gpObject);
+			return gpObject;
+		} else {
+			sessionFactory.getCurrentSession().save(gp);
+			return gp;
+		}
 	}
\ No newline at end of file
