diff --git a/old_test.java b/new_test.java
index eb0929e..10a0ccf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -32,7 +32,7 @@
 
     String dstPath = PathUtils.concatPath(ufsRoot, fileInfo.getPath());
     Mockito.when(ufs.create(dstPath)).thenReturn(outputStream);
-    Mockito.when(ufs.create(Mockito.anyString(), Mockito.any(PermissionStatus.class)))
+    Mockito.when(ufs.create(Mockito.anyString(), Mockito.any(UnderFileSystemCreateOptions.class)))
         .thenReturn(outputStream);
 
     manager.lockBlocks(fileId, blockIds);
