diff --git a/old_product.java b/new_product.java
index b7f4b5c..f4dec52 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,7 +15,8 @@ public static long persistFile(FileSystem fs, AlluxioURI uri, URIStatus status,
       URIStatus uriStatus = fs.getStatus(uri);
       PermissionStatus ps = new PermissionStatus(uriStatus.getUserName(), uriStatus.getGroupName(),
           (short) uriStatus.getPermission());
-      OutputStream out = closer.register(ufs.create(dstPath.getPath(), ps));
+      OutputStream out = closer.register(ufs.create(dstPath.getPath(),
+          UnderFileSystemCreateOptions.defaults().setPermissionStatus(ps)));
       ret = IOUtils.copyLarge(in, out);
     } catch (Exception e) {
       throw closer.rethrow(e);
