diff --git a/old_test.java b/new_test.java
index 127de74..c57349a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
 	public void allInside() {
 		IntrinsicParameters param = new IntrinsicParameters(300,320,0,150,130,
-				width,height,new double[]{0.1,1e-4});
+				width,height, false, new double[]{0.1,1e-4});
 
-		PointTransform_F32 adjusted = LensDistortionOps.allInside(param,false,null);
+		PointTransform_F32 adjusted = LensDistortionOps.allInside(param,null);
 
 		checkInside(adjusted);
 	}
\ No newline at end of file
