diff --git a/old_product.java b/new_product.java
index 5d7f958..f20f4d9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
 public static PointTransform_F32 allInside( IntrinsicParameters param,
-												boolean applyLeftToRight ,
 												IntrinsicParameters paramAdj ) {
 		RemoveRadialPtoP_F32 removeDistort = new RemoveRadialPtoP_F32();
 		AddRadialPtoP_F32 addDistort = new AddRadialPtoP_F32();
@@ -23,7 +22,7 @@ public static PointTransform_F32 allInside( IntrinsicParameters param,
 
 		PointTransform_F32 tranAdj = UtilIntrinsic.adjustIntrinsic_F32(addDistort, false, param, A, paramAdj);
 
-		if( applyLeftToRight) {
+		if( param.leftHanded) {
 			PointTransform_F32 l2r = new LeftToRightHanded_F32(param.height);
 			return new SequencePointTransform_F32(l2r,tranAdj,l2r);
 		} else
