diff --git a/old_test.java b/new_test.java
index c938590..13e5d54 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testGetCommunities() throws Exception
     {
-        assertThat("testGetCommunities 0",c.getCommunities(), notNullValue());
-        assertTrue("testGetCommunities 1",c.getCommunities().length == 0);
+        assertThat("testGetCommunities 0",collection.getCommunities(), notNullValue());
+        assertTrue("testGetCommunities 1",collection.getCommunities().size() == 1);
     }
\ No newline at end of file
