diff --git a/old_product.java b/new_product.java
index 0209ab1..e2a5ff9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -47,7 +47,7 @@ public static Dataset parseCSVFile(Reader reader, String yVariable, Map<String,
                 }
             } 
             catch (IOException ex) {
-                dataset.erase();
+                dataset.delete();
                 throw new RuntimeException(ex);
             }
             return dataset;
