diff --git a/old_test.java b/new_test.java
index 40cc819..9d1df5d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
 	public void buildAuthRequestUrl() {
 
-		String requestUri = urlBuilder.buildAuthRequestUrl(serverConfig, clientConfig, redirectUri, nonce, state);
+		String requestUri = urlBuilder.buildAuthRequestUrl(serverConfig, clientConfig, redirectUri, nonce, state, options);
 
 		// parsing the result
 		UriComponentsBuilder builder = null;
@@ -32,4 +32,7 @@
 		assertEquals(redirectUri, claims.getClaim("redirect_uri"));
 		assertEquals(nonce, claims.getClaim("nonce"));
 		assertEquals(state, claims.getClaim("state"));
+		for (String claim : options.keySet()) {
+	        assertEquals(options.get(claim), claims.getClaim(claim));
+        }
 	}
\ No newline at end of file
