diff --git a/old_test.java b/new_test.java
index b977eee..da7031d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,5 +15,5 @@
         Set<Authority> authoritiesAddedToProvider = EnumSet.of(Authority.ROLE_ADMIN);
 
         // verify interaction with provider
-        Mockito.verify(authorityProvider, Mockito.times(1)).setAuthorities(USER_DN_3, authoritiesAddedToProvider);
+        Mockito.verify(authorityProvider, Mockito.times(1)).setAuthorities(USER_IDENTITY_3, authoritiesAddedToProvider);
     }
\ No newline at end of file
