diff --git a/old_product.java b/new_product.java
index c3456e5..c8c96f0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-void setAuthorities(String dn, Set<Authority> authorities) throws UnknownIdentityException, AuthorityAccessException;
\ No newline at end of file
+void setAuthorities(String identity, Set<Authority> authorities) throws UnknownIdentityException, AuthorityAccessException;
\ No newline at end of file
