diff --git a/old_product.java b/new_product.java
index fb65a3c..c8b92c2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -25,7 +25,7 @@ private GridSqlInsert parseInsert(Insert insert) {
 
         res.columns(cols);
 
-        List<Expression[]> srcRows = INSERT_ROWS.get(insert);
+        List<Expression[]> srcRows = COMMAND_WITH_VALUES_VALS_EXPRESSIONS.get(insert);
         if (!srcRows.isEmpty()) {
             List<GridSqlElement[]> rows = new ArrayList<>(srcRows.size());
             for (Expression[] srcRow : srcRows) {
