diff --git a/old_test.java b/new_test.java
index 1ee4009..b718468 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,28 +4,28 @@
         TestContext context = new TestContext(DEFAULT_CONFIG);
 
         // Check behavior for non-existent segments (in Storage).
-        context.transactionAggregators[0].initialize(TIMEOUT).join();
+        context.transactionAggregators[0].initialize(TIMEOUT, executorService()).join();
         Assert.assertTrue("isDeleted() flag not set on metadata for deleted segment.", context.transactionAggregators[0].getMetadata().isDeleted());
 
         // Check behavior for already-sealed segments (in storage, but not in metadata)
         context.storage.create(context.transactionAggregators[1].getMetadata().getName(), TIMEOUT).join();
-        context.storage.seal(context.transactionAggregators[1].getMetadata().getName(), TIMEOUT).join();
+        context.storage.seal(writeHandle(context.transactionAggregators[1].getMetadata().getName()), TIMEOUT).join();
         AssertExtensions.assertThrows(
                 "initialize() succeeded on a Segment is sealed in Storage but not in the metadata.",
-                () -> context.transactionAggregators[1].initialize(TIMEOUT),
+                () -> context.transactionAggregators[1].initialize(TIMEOUT, executorService()),
                 ex -> ex instanceof DataCorruptionException);
 
         // Check behavior for already-sealed segments (in storage, in metadata, but metadata does not reflect Sealed in storage.)
         context.storage.create(context.transactionAggregators[2].getMetadata().getName(), TIMEOUT).join();
-        context.storage.seal(context.transactionAggregators[2].getMetadata().getName(), TIMEOUT).join();
+        context.storage.seal(writeHandle(context.transactionAggregators[2].getMetadata().getName()), TIMEOUT).join();
         ((UpdateableSegmentMetadata) context.transactionAggregators[2].getMetadata()).markSealed();
-        context.transactionAggregators[2].initialize(TIMEOUT).join();
+        context.transactionAggregators[2].initialize(TIMEOUT, executorService()).join();
         Assert.assertTrue("isSealedInStorage() flag not set on metadata for storage-sealed segment.", context.transactionAggregators[2].getMetadata().isSealedInStorage());
 
         // Check the ability to update Metadata.StorageOffset if it is different.
         final int writeLength = 10;
         context.storage.create(context.segmentAggregator.getMetadata().getName(), TIMEOUT).join();
-        context.storage.write(context.segmentAggregator.getMetadata().getName(), 0, new ByteArrayInputStream(new byte[writeLength]), writeLength, TIMEOUT).join();
-        context.segmentAggregator.initialize(TIMEOUT).join();
+        context.storage.write(writeHandle(context.segmentAggregator.getMetadata().getName()), 0, new ByteArrayInputStream(new byte[writeLength]), writeLength, TIMEOUT).join();
+        context.segmentAggregator.initialize(TIMEOUT, executorService()).join();
         Assert.assertEquals("SegmentMetadata.StorageLength was not updated after call to initialize().", writeLength, context.segmentAggregator.getMetadata().getStorageLength());
     }
\ No newline at end of file
