diff --git a/old_test.java b/new_test.java
index 8768494..ee06cf1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,8 @@
 	@Verifies(value = "should erase the visit from the database", method = "purgeVisit(Visit)")
 	public void purgeVisit_shouldEraseTheVisitFromTheDatabase() throws Exception {
 		VisitService vs = Context.getVisitService();
-		Integer originalSize = vs.getAllVisits().size();
+		Integer originalSize = vs.getVisits(null, null, null, null, null, null, null, null, true).size();
 		Visit visit = Context.getVisitService().getVisit(1);
 		vs.purgeVisit(visit);
-		Assert.assertEquals(originalSize - 1, vs.getAllVisits().size());
+		Assert.assertEquals(originalSize - 1, vs.getVisits(null, null, null, null, null, null, null, null, true).size());
 	}
\ No newline at end of file
