diff --git a/old_product.java b/new_product.java
index 71e19d3..14b6a72 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,8 +13,8 @@
             );
         }
 
-        final List<ClusterCriteria> clusterCriterias = jobEntity.getClusterCriterias();
-        final Set<String> commandCriteria = jobEntity.getCommandCriteria();
+        final List<ClusterCriteria> clusterCriterias = jobEntity.getRequest().getClusterCriteriasAsList();
+        final Set<String> commandCriteria = jobEntity.getRequest().getCommandCriteriaAsSet();
 
         for (final ClusterCriteria clusterCriteria : clusterCriterias) {
             @SuppressWarnings("unchecked")
@@ -26,13 +26,6 @@
             );
 
             if (!clusterEntities.isEmpty()) {
-                // Add the successfully criteria to the job object in string form.
-                jobEntity.setChosenClusterCriteriaString(
-                        StringUtils.join(
-                                clusterCriteria.getTags(),
-                                CRITERIA_DELIMITER
-                        )
-                );
                 return clusterEntities
                         .stream()
                         .map(ClusterEntity::getDTO)
