diff --git a/old_product.java b/new_product.java
index 9078666..dd2b308 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,11 @@
 public boolean testConnection(Long id) {
-		DataSourceVO dataSourceVO = this.getDataSource(id);
+		DataSourceVO dataSourceVO = this.getDataSourceVoById(id);
 		// 将domain转成VO
 		try {
 			DataSource dataSource = new DataSource();
 			BeanUtils.copyProperties(dataSource, dataSourceVO);
 			return dataSource.testConnection();
 		} catch (Exception e) {
-//			e.printStackTrace();
 			return false;
 		}
 	}
\ No newline at end of file
