diff --git a/old_product.java b/new_product.java
index 4f239c3..8a13a22 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,8 @@
 @Nullable
   public String getHost() {
-    return mUri.getHost();
+    if (mUri.getAuthority() instanceof SingleMasterAuthority) {
+      SingleMasterAuthority authority = (SingleMasterAuthority) mUri.getAuthority();
+      return authority.getHost();
+    }
+    return null;
   }
\ No newline at end of file
