diff --git a/old_test.java b/new_test.java
index 489216c..b2a93ff 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,10 +6,10 @@
 
         final Transaction transaction = jdbcRepository.beginTransaction();
         jdbcRepository.add(jsonObject);
-        jdbcRepository.remove(jsonObject.getString(JdbcRepositories.OID));
+        jdbcRepository.remove(jsonObject.getString(JdbcRepositories.getDefaultKeyName()));
         transaction.commit();
 
-        final JSONObject jsonObjectDB = jdbcRepository.get(jsonObject.getString(JdbcRepositories.OID));
+        final JSONObject jsonObjectDB = jdbcRepository.get(jsonObject.getString(JdbcRepositories.getDefaultKeyName()));
 
         assertNull(jsonObjectDB);
 
