diff --git a/old_product.java b/new_product.java
index 5b6f77c..fefe185 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
-public synchronized void addCheckpoint(int fileId) throws IOException {
+public synchronized void addCheckpoint(long fileId) throws IOException {
     mustConnect();
 
     try {
-      mClient.addCheckpoint(mMasterClient.getUserId(), fileId);
+      mClient.addCheckpoint(mUserId, fileId);
     } catch (FileDoesNotExistException e) {
       throw new IOException(e);
     } catch (SuspectedFileSizeException e) {
