diff --git a/old_test.java b/new_test.java
index c3f7152..553f063 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,15 @@
 @Test
   public void AddCheckpointTest() throws Exception {
-    TachyonFSTestUtils.createByteFile(mTfs, "/xyz", WriteType.THROUGH, 10);
-    FileInfo fInfo =
-        mLocalTachyonCluster.getMasterInfo().getClientFileInfo(new TachyonURI("/xyz"));
-    mTfs.createFile(new TachyonURI("/xyz_ck"), new TachyonURI(fInfo.getUfsPath()));
-    FileInfo ckFileInfo =
-        mLocalTachyonCluster.getMasterInfo().getClientFileInfo(new TachyonURI("/xyz_ck"));
+    ClientOptions options = new ClientOptions.Builder(mMasterTachyonConf)
+        .setCacheType(CacheType.NO_CACHE).setUnderStorageType(UnderStorageType.PERSIST).build();
+    TachyonFSTestUtils.createByteFile(mTfs, "/xyz", options, 10);
+    FileInfo fInfo = mTfs.getInfo(mTfs.open(new TachyonURI("/xyz")));
+    TachyonURI ckPath = new TachyonURI("/xyz_ck");
+    // TODO(cc): what's the counterpart in the new client API for this?
+    //mTfs.createFile(new TachyonURI("/xyz_ck"), new TachyonURI(fInfo.getUfsPath()));
+    FileInfo ckFileInfo = mTfs.getInfo(mTfs.open(ckPath));
     mLocalTachyonCluster.stopTFS();
     AddCheckpointTestUtil(fInfo, ckFileInfo);
-    String editLogPath = mLocalTachyonCluster.getEditLogPath();
-    UnderFileSystem.get(editLogPath, mMasterTachyonConf).delete(editLogPath, true);
+    deleteFsMasterJournalLogs();
     AddCheckpointTestUtil(fInfo, ckFileInfo);
   }
\ No newline at end of file
