diff --git a/old_test.java b/new_test.java
index dd5be4b..1d3ede4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
         builder.appendValue(MONTH_OF_YEAR, 1, 2, SignStyle.NORMAL).appendValue(DAY_OF_MONTH, 2);
         DateTimeFormatter f = builder.toFormatter();
         assertEquals(f.toString(), "Value(MonthOfYear,1,2,NORMAL)Value(DayOfMonth,2)");
-        DateTimeParseContext cal = f.parseToContext("123", new ParsePosition(0));
-        assertEquals(cal.getParsed(MONTH_OF_YEAR), Long.valueOf(1));
-        assertEquals(cal.getParsed(DAY_OF_MONTH), Long.valueOf(23));
+        DateTimeBuilder cal = f.parseToBuilder("123", new ParsePosition(0));
+        assertEquals(cal.getFieldValueMap().get(MONTH_OF_YEAR), Long.valueOf(1));
+        assertEquals(cal.getFieldValueMap().get(DAY_OF_MONTH), Long.valueOf(23));
     }
\ No newline at end of file
