diff --git a/old_product.java b/new_product.java
index 07cc1f6..dbe476d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,5 @@
 public DateTimeFormatterBuilder appendValue(DateTimeField field) {
         DateTimes.checkNotNull(field, "DateTimeField must not be null");
-        NumberPrinterParser pp = new NumberPrinterParser(field, 1, 19, SignStyle.NORMAL);
-        active.valueParserIndex = appendInternal(pp, pp);
+        active.valueParserIndex = appendInternal(new NumberPrinterParser(field, 1, 19, SignStyle.NORMAL));
         return this;
     }
\ No newline at end of file
