diff --git a/old_test.java b/new_test.java
index d43a8d3..82da3a7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,13 +13,15 @@
 
         // Create some Segments and Transaction and verify they are properly created and registered.
         for (int i = 0; i < segmentCount; i++) {
-            String name = getName(i);
-            context.mapper.createNewStreamSegment(name, TIMEOUT).join();
-            assertStreamSegmentCreated(name, context);
+            String segmentName = getName(i);
+            val segmentAttributes = createAttributes();
+            context.mapper.createNewStreamSegment(segmentName, segmentAttributes, TIMEOUT).join();
+            assertStreamSegmentCreated(segmentName, segmentAttributes, context);
 
             for (int j = 0; j < transactionsPerSegment; j++) {
-                String transactionName = context.mapper.createNewTransactionStreamSegment(name, UUID.randomUUID(), TIMEOUT).join();
-                assertTransactionCreated(transactionName, name, context);
+                val transactionAttributes = createAttributes();
+                String transactionName = context.mapper.createNewTransactionStreamSegment(segmentName, UUID.randomUUID(), transactionAttributes, TIMEOUT).join();
+                assertTransactionCreated(transactionName, segmentName, transactionAttributes, context);
             }
         }
     }
\ No newline at end of file
