diff --git a/old_test.java b/new_test.java
index 8b8d830..27b0fd6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
         networkLsa.addAttachedRouter(Ip4Address.valueOf("1.1.1.1"));
         networkLsa.addAttachedRouter(Ip4Address.valueOf("2.2.2.2"));
         networkLsa.addAttachedRouter(Ip4Address.valueOf("3.3.3.3"));
-        result1 = networkLsa.getLSABodyAsByteArray();
+        result1 = networkLsa.getLsaBodyAsByteArray();
         assertThat(result1, is(notNullValue()));
     }
\ No newline at end of file
