diff --git a/old_test.java b/new_test.java
index 6910f7e..bd2ba88 100644
--- a/old_test.java
+++ b/new_test.java
@@ -23,5 +23,4 @@
         String expResult = "This domain is established to be used for illustrative examples in documents. You may use this domain in examples without prior coordination or asking for permission.";
         String result = instance.extract(text, parameters);
         assertEquals(expResult, result);
-        instance=null;
     }
\ No newline at end of file
