diff --git a/old_product.java b/new_product.java
index 0ad39e1..7591b9f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -47,7 +47,7 @@
             }
             
             
-            wordCombinations = null;
+            //wordCombinations = null;
         }
         
         //initialize keyword map
@@ -69,10 +69,10 @@
                     keywordsMap.put(sb.toString().trim(), proximityScore);
                 }
                 
-                sb=null;
+                //sb=null;
             }
         }
-        keywordProximityScores=null;
+        //keywordProximityScores=null;
         
         return keywordsMap;
     }
\ No newline at end of file
