diff --git a/old_test.java b/new_test.java
index 4b8546b..4163c71 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test
     public void toMap(){
-        String json = "{'data1':{'name':'get'},'data2':{'name':'set'}}";
+        String json = "{'data1':{'name':'get'},'data2':{'name':'set'},'null':{'name':'set'}}";
         Map<String, Person> map = JsonUtil.toMap(json, Person.class);
-
         LOGGER.info(JsonUtil.toJSON(map).toString(4, 4));
     }
\ No newline at end of file
