diff --git a/old_product.java b/new_product.java
index 15fa682..f7385cb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,19 +1,16 @@
-public static Map<String, Object> toMap(String json){
-        JSONObject jsonObject = toJSONObject(json);
-
-        //TODO 处理不了key 是 null 的情况
-        //java.lang.ClassCastException: net.sf.json.JSONNull cannot be cast to java.lang.String
-        //Map<String, Object> map = (Map<String, Object>) JSONObject.toBean(jsonObject, Map.class);
+public static <T> Map<String, T> toMap(String json,Class<T> clazz,Map<String, Class<?>> classMap){
+        LOGGER.debug("in json:{}", json);
 
-        Map<String, Object> map = new HashMap<String, Object>();
+        Map<String, T> map = new HashMap<String, T>();
 
+        JSONObject jsonObject = toJSONObject(json);
         @SuppressWarnings("unchecked")
         Iterator<String> keys = jsonObject.keys();
-
         while (keys.hasNext()){
             String key = keys.next();
             Object value = jsonObject.get(key);
-            map.put(key, value);
+            LOGGER.debug("key:[{}], value:{}", key, value);
+            map.put(key, toBean(value, clazz, classMap));
         }
         return map;
     }
\ No newline at end of file
