diff --git a/old_test.java b/new_test.java
index d53c2b1..8493baa 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
       //Create a test annotation for the event and use it to construct the
       //event object
       Annotation[] annotations = new Annotation[] { new TameAnnotationLiteral() };
-      EventImpl<DangerCall> eventComponent = new EventImpl<DangerCall>(manager, DangerCall.class, annotations);
+      EventImpl<DangerCall> eventComponent = new EventImpl<DangerCall>(DangerCall.class, manager, annotations);
       Observer<DangerCall> observer = new AnObserver<DangerCall>();
       eventComponent.observe(observer, new SynchronousAnnotationLiteral());
       assert manager.getObservedEventType().equals(DangerCall.class);
