diff --git a/old_product.java b/new_product.java
index 31b6b10..528b084 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,4 @@
 public void observe(Observer<T> observer, Annotation... bindingTypes)
    {
-      Set<Annotation> bindingParameters = checkBindingTypes(bindingTypes);
-      bindingParameters.addAll(this.bindingTypes);
-      manager.addObserver(observer, eventType, bindingParameters.toArray(new Annotation[0]));
+      manager.addObserver(observer, type, mergeBindings(bindingTypes));
    }
\ No newline at end of file
