diff --git a/old_product.java b/new_product.java
index 523b30c..6aed831 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 void findErrorLocatorBM( int syndromes[] , int length , GrowQueue_I8 errorLocator ) {
 		GrowQueue_I8 C = errorLocator; // error polynomial
 		GrowQueue_I8 B = new GrowQueue_I8();  // previous error polynomial
+		// TODO remove new from this function
 
 		initToOne(C,length+1);
 		initToOne(B,length+1);
@@ -44,6 +45,5 @@ void findErrorLocatorBM( int syndromes[] , int length , GrowQueue_I8 errorLocato
 			}
 		}
 
-		// TODO drop leading zeros?
-		System.out.println();
+		removeZeros(C);
 	}
\ No newline at end of file
