diff --git a/old_product.java b/new_product.java
index da6b009..9237d4a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,9 @@
             @NotBlank(message = "No id entered. Unable to get job.")
             final String id
     ) throws GenieException {
-        LOG.debug("called for id: " + id);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("called for id: " + id);
+        }
 
         final Job job = this.jobRepo.findOne(id);
         if (job != null) {
