diff --git a/old_test.java b/new_test.java
index 0acc8d8..7a7e1d9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test(expected = IllegalArgumentException.class)
     public void checkNearCacheConfig_NATIVE() {
-        checkNearCacheConfig(MAP_NAME, getNearCacheConfig(NATIVE), false);
+        checkNearCacheConfig(MAP_NAME, getNearCacheConfig(NATIVE), null, false);
     }
\ No newline at end of file
