diff --git a/old_product.java b/new_product.java
index d2a435f..817adea 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,12 @@
-public static void checkNearCacheConfig(String mapName, NearCacheConfig nearCacheConfig, boolean isClient) {
-        checkLocalUpdatePolicy(mapName, nearCacheConfig);
-        checkNotNative(nearCacheConfig.getInMemoryFormat());
+public static void checkNearCacheConfig(String mapName, NearCacheConfig nearCacheConfig,
+                                            NativeMemoryConfig nativeMemoryConfig, boolean isClient) {
+        InMemoryFormat inMemoryFormat = nearCacheConfig.getInMemoryFormat();
+
+        checkNotNative(inMemoryFormat);
+        checkLocalUpdatePolicy(mapName, nearCacheConfig.getLocalUpdatePolicy());
         checkEvictionConfig(nearCacheConfig.getEvictionConfig(), true);
+        checkOnHeapNearCacheMaxSizePolicy(nearCacheConfig);
+        checkNearCacheNativeMemoryConfig(nearCacheConfig.getInMemoryFormat(), nativeMemoryConfig, getBuildInfo().isEnterprise());
 
         if (isClient && nearCacheConfig.isCacheLocalEntries()) {
             throw new IllegalArgumentException("The Near Cache option `cache-local-entries` is not supported in "
