diff --git a/old_product.java b/new_product.java
index fa48a00..3ddbc0a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,12 @@
-private synchronized void copy(final File src, final File trg)
+private synchronized void copy(final Path src, final Path trg)
       throws QueryException, IOException {
 
-    if(src.isDirectory()) {
-      if(!trg.mkdir()) throw FILE_IE_DIR.get(info, trg);
-      final File[] files = src.listFiles();
-      if(files == null) throw FILE_IE_ACCESS.get(info, src);
-      for(final File f : files) copy(f, new File(trg, f.getName()));
+    if(Files.isDirectory(src)) {
+      Files.createDirectory(trg);
+      try(DirectoryStream<Path> paths = Files.newDirectoryStream(src)) {
+        for(final Path p : paths) copy(p, trg.resolve(p.getFileName()));
+      }
     } else {
-      new IOFile(src).copyTo(new IOFile(trg));
+      Files.copy(src, trg, StandardCopyOption.REPLACE_EXISTING);
     }
   }
\ No newline at end of file
