diff --git a/old_test.java b/new_test.java
index cbcbb0b..5662b70 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,7 +10,7 @@
           deque.addLast(cast(i));
         }
 
-        final int additions = randomIntBetween(0, 5000);
+        final int additions = randomIntBetween(1, 5000);
         deque.ensureCapacity(additions + deque.size());
         final int before = resizer.growCalls;
         for (int i = 0; i < additions; i++) {
