diff --git a/old_test.java b/new_test.java
index 09a78ab..cf17378 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
-@Test
-  public void soundexVariations() {
+@Test public void soundexVariations() {
     soundexVariations("B650",
       "BARHAM", "BARONE", "BARRON", "BERNA", "BIRNEY", "BIRNIE", "BOOROM", "BOREN", "BORN", "BOURN",
       "BOURNE", "BOWRON", "BRAIN", "BRAME", "BRANN", "BRAUN", "BREEN", "BRIEN", "BRIM", "BRIMM",
@@ -13,7 +12,9 @@
       "KINGSMITH", "-KINGSMITH", "K-INGSMITH", "KI-NGSMITH", "KIN-GSMITH", "KING-SMITH",
       "KINGS-MITH", "KINGSM-ITH", "KINGSMI-TH", "KINGSMIT-H", "KINGSMITH-"
     );
-    query(_STRINGS_SOUNDEX.args("Williams"), "W452");
+    soundexVariations("W452",
+      "Williams"
+    );
     soundexVariations("S460",
       "Sgler", "Swhgler",
       "SAILOR", "SALYER", "SAYLOR", "SCHALLER", "SCHELLER", "SCHILLER", "SCHOOLER", "SCHULER",
