diff --git a/old_product.java b/new_product.java
index 87b0a22..a9f9d72 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @Nullable
   public Class<?> getCoreInterfaceRewritingTarget(
       int opcode, String owner, String name, String desc, boolean itf) {
-    if (owner.contains("$$Lambda$") || owner.endsWith("$$CC")) {
+    if (looksGenerated(owner)) {
       // Regular desugaring handles generated classes, no emulation is needed
       return null;
     }
