diff --git a/old_product.java b/new_product.java
index 7f28320..783d50f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
-AppUsagePerServiceBean getAppUsagePerService(String organizationId, String applicationId, String version,
+AppUsagePerApiBean getAppUsagePerApi(String organizationId, String applicationId, String version,
             DateTime from, DateTime to);
\ No newline at end of file
