diff --git a/old_product.java b/new_product.java
index 026e58d..63fdbce 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@
     {
         ScriptEngineManager manager = new ScriptEngineManager();
         ScriptEngine engine = manager.getEngineByName( "gremlin" );
-        ArrayList<EdgeRepresentation> results = new ArrayList<EdgeRepresentation>();
+        ArrayList<Edge> results = new ArrayList<Edge>();
         Neo4jGraph graph = new Neo4jGraph( graphDb );
         engine.getBindings( ScriptContext.ENGINE_SCOPE ).put( "g",
                 graph );
@@ -21,5 +21,5 @@
             // TODO Auto-generated catch block
             e.printStackTrace();
         }
-        return new ListRepresentation( RepresentationType.RELATIONSHIP, results );
+        return new PipeEdgeRepresentation(  results );
     }
\ No newline at end of file
