diff --git a/old_test.java b/new_test.java
index a8d8fc4..afec0e7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,8 +9,6 @@
 			curRepresentationObj=(Representation)curGremlinPluginObj.getEdges(script, curGraphDBServiceObj);
 			assertNotNull(curRepresentationObj);
 			tx.success();
-			
-			System.err.println("GremlinPluginTest::testGetEdges the contents of the representation object="+curRepresentationObj.toString());
 		}
 		catch (Throwable t) {
 			t.printStackTrace ();
@@ -19,5 +17,5 @@
 		{
 			tx.finish();
 		}
-		
+		System.out.println("Results of testGetEdges"+json.format( curRepresentationObj ));
 	}
\ No newline at end of file
