diff --git a/old_product.java b/new_product.java
index cf8f1de..7536f17 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,11 @@
-public void delete(final byte[] pkg) throws QueryException {
+public void delete(final String pkg) throws QueryException {
     boolean found = false;
     final TokenMap dict = repo.pkgDict();
+    final byte[] pp = token(pkg);
     for(final byte[] nextPkg : dict) {
       if(nextPkg == null) continue;
       // a package can be deleted by its name or the name suffixed with its version
-      if(eq(nextPkg, pkg) || eq(Package.name(nextPkg), pkg)) {
+      if(eq(nextPkg, pp) || eq(Package.name(nextPkg), pp)) {
         // check if package to be deleted participates in a dependency
         final byte[] primPkg = primary(nextPkg);
         if(primPkg != null) BXRE_DEP.thrw(info, string(primPkg), pkg);
