diff --git a/old_test.java b/new_test.java
index aa00639..0be5276 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public void delete() throws BaseXException {
     // try to delete a package which is not installed
     try {
-      new RepoManager(context).delete(token("xyz"));
+      new RepoManager(context).delete("xyz");
       fail("Not installed package not detected.");
     } catch(final QueryException ex) {
       check(null, ex, Err.BXRE_WHICH);
@@ -35,7 +35,7 @@
 
     // try to delete pkg3
     try {
-      new RepoManager(context).delete(token(PKG3ID));
+      new RepoManager(context).delete(PKG3ID);
       fail("Package involved in a dependency was deleted.");
     } catch(final QueryException ex) {
       check(null, ex, Err.BXRE_DEP);
