diff --git a/old_product.java b/new_product.java
index 31247bf..efd9b07 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public boolean isMetadataEquals(ProtectedStorageEntry protectedStorageEntry) {
+public boolean matchesRelevantPubKey(ProtectedStorageEntry protectedStorageEntry) {
         boolean result = protectedStorageEntry.getOwnerPubKey().equals(this.ownerPubKey);
 
         if (!result) {
