diff --git a/old_product.java b/new_product.java
index cdd6497..724401a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,17 @@
-public void releaseConnection(Connection conn) throws SQLException {
+public void releaseConnection(Connection connection) throws SQLException {
+		
 		try {
+			ConnectionHandle handle = (ConnectionHandle)connection;
+			
+			// hook calls
+			if (handle.getConnectionHook() != null){
+				handle.getConnectionHook().onCheckIn(handle);
+			}
+			
 			if (this.releaseHelperThreadsConfigured){
-				((ConnectionHandle)conn).getOriginatingPartition().getConnectionsPendingRelease().put((ConnectionHandle) conn);
+				handle.getOriginatingPartition().getConnectionsPendingRelease().put(handle);
 			} else {
-				internalReleaseConnection(conn);
+				internalReleaseConnection(handle);
 			}
 		}
 		catch (InterruptedException e) {
