diff --git a/old_test.java b/new_test.java
index 12596be..25e0d0b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
   public void prepareConfigurationTest() throws Exception {
-    Configuration tConf = new Configuration();
-    org.apache.hadoop.conf.Configuration hConf = new org.apache.hadoop.conf.Configuration();
-    mMockHdfsUnderFileSystem.prepareConfiguration("", tConf, hConf);
-    Assert.assertEquals("org.apache.hadoop.hdfs.DistributedFileSystem", hConf.get("fs.hdfs.impl"));
-    Assert.assertFalse(hConf.getBoolean("fs.hdfs.impl.disable.cache", false));
-    Assert.assertNotNull(hConf.get(Constants.UNDERFS_HDFS_CONFIGURATION));
+    Configuration.defaultInit();
+    org.apache.hadoop.conf.Configuration conf = new org.apache.hadoop.conf.Configuration();
+    mMockHdfsUnderFileSystem.prepareConfiguration("", conf);
+    Assert.assertEquals("org.apache.hadoop.hdfs.DistributedFileSystem", conf.get("fs.hdfs.impl"));
+    Assert.assertFalse(conf.getBoolean("fs.hdfs.impl.disable.cache", false));
+    Assert.assertNotNull(conf.get(Constants.UNDERFS_HDFS_CONFIGURATION));
   }
\ No newline at end of file
