diff --git a/old_product.java b/new_product.java
index 59ce02c..a542372 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-private static void create(final int... db) throws BaseXException {
+private static void create(final int... db) {
     for(final int d : db) {
       final String[] doc = DOCS[d - 1];
-      new CreateDB(doc[0], doc[1]).execute(context);
+      execute(new CreateDB(doc[0], doc[1]));
     }
   }
\ No newline at end of file
