diff --git a/old_test.java b/new_test.java
index 5200319..29a46ec 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(dataProvider="sampleDates")
     public void test_toCalendrical(int year, int month, int day, ZoneOffset offset) {
         OffsetDate t = OffsetDate.date(year, month, day, offset);
-        assertEquals(t.toCalendrical(), Calendrical.calendrical(t.getDate(), null, t.getOffset(), null));
+        assertEquals(t.toCalendrical(), new Calendrical(t.getDate(), null, t.getOffset(), null));
     }
\ No newline at end of file
