diff --git a/old_test.java b/new_test.java
index 705a7d4..80f08c8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test(expected = IllegalArgumentException.class)
-    public void test_checkInMemoryFormat_NATIVE() {
-        checkInMemoryFormat(InMemoryFormat.NATIVE);
+    public void test_checkNotNative_NATIVE() {
+        checkNotNative(InMemoryFormat.NATIVE);
     }
\ No newline at end of file
