diff --git a/old_test.java b/new_test.java
index 67e8aab..fe1e5a1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -31,12 +31,11 @@
             .withDisableLogArchival(true)
             .build();
 
-        Mockito.when(this.jobPersistenceService.createJobRequest(Mockito.eq(jobRequest))).thenReturn(jobRequest);
+        Mockito.when(this.jobPersistenceService.createJobRequest(jobRequest, clientHost)).thenReturn(jobRequest);
         final ArgumentCaptor<Job> argument = ArgumentCaptor.forClass(Job.class);
 
         this.jobCoordinatorService.coordinateJob(jobRequest, clientHost);
 
-        Mockito.verify(this.jobPersistenceService, Mockito.times(1)).addClientHostToJobRequest(JOB_1_ID, clientHost);
         Mockito.verify(this.jobSubmitterService, Mockito.times(1)).submitJob(jobRequest);
 
         Mockito.verify(this.jobPersistenceService).createJob(argument.capture());
