diff --git a/old_test.java b/new_test.java
index a0ac4ea..fb3b552 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
 		assertEquals("0 bytes", FormatUtil.formatBytes(0));
 		assertEquals("1 byte", FormatUtil.formatBytes(1));
 		assertEquals("532 bytes", FormatUtil.formatBytes(532));
-		assertEquals("1 KB", FormatUtil.formatBytes(1024));
-		assertEquals("1 GB", FormatUtil.formatBytes(1024 * 1024 * 1024));
+		assertEquals("1 KiB", FormatUtil.formatBytes(1024));
+		assertEquals("1 GiB", FormatUtil.formatBytes(1024 * 1024 * 1024));
 		assertEquals("1 TiB", FormatUtil.formatBytes(1099511627776L));
 	}
\ No newline at end of file
