diff --git a/old_test.java b/new_test.java
index 2a4b816..ca2252c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,5 +13,5 @@
     // updates an existing entry
     error(_ARCHIVE_CREATE.args("X", "X",
         "<archive:options><archive:format value='gzip'/></archive:options>") + " ! " +
-        _ARCHIVE_DELETE.args(" .", "X"), Err.ARCH_MODIFY);
+        _ARCHIVE_DELETE.args(" .", "X"), Err.ARCH_MODIFY_X);
   }
\ No newline at end of file
