diff --git a/old_product.java b/new_product.java
index a758409..43ed7eb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -18,7 +18,7 @@ public static List<Point> SpatialKnnQuery(PointRDD spatialRDD, Point queryCenter
 		}
 		else
 		{
-			JavaRDD<Object> tmp = spatialRDD.getRawSpatialRDD().mapPartitions(new GeometryKnnJudgement(queryCenter,k));
+			JavaRDD<Object> tmp = spatialRDD.getRawSpatialRDD().mapPartitions(new KnnJudgement(queryCenter,k));
 			List<Object> result = tmp.takeOrdered(k, new GeometryDistanceComparator(queryCenter,true));
 			List<Point> geometryResult = new ArrayList<Point>();
 			for(Object spatialObject:result)
