diff --git a/old_product.java b/new_product.java
index 9cfa836..594497b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -23,7 +23,7 @@ private synchronized Item writeBinary(final boolean append, final QueryContext c
       final RandomAccessFile raf = new RandomAccessFile(path, "rw");
       try {
         final long dlen = raf.length();
-        if(off < 0 || off > dlen) throw FILE_BOUNDS.get(info, off, dlen);
+        if(off < 0 || off > dlen) throw FILE_OOR.get(info, off, dlen);
         raf.seek(off);
         raf.write(bin.binary(info));
       } finally {
