diff --git a/old_product.java b/new_product.java
index ee99899..0272e0f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,8 +5,7 @@
         }
         if (otherInfo instanceof ZoneOffsetInfo) {
             ZoneOffsetInfo info = (ZoneOffsetInfo) otherInfo;
-            return dateTime.equals(info.dateTime) &&
-                    (transition != null ? transition.equals(info.transition) : offset.equals(info.offset));
+            return (transition != null ? transition.equals(info.transition) : offset.equals(info.offset));
         }
         return false;
     }
\ No newline at end of file
