diff --git a/old_test.java b/new_test.java
index 9891004..60c5691 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
 	public void testScale() {
-		ImageFloat32 input = createImage(width,height);
+		GrayF32 input = createImage(width,height);
 
-		DescribePointBriefSO<ImageFloat32> alg = createAlg();
+		DescribePointBriefSO<GrayF32> alg = createAlg();
 		TupleDesc_B desc1 = alg.createFeature();
 		TupleDesc_B desc2 = alg.createFeature();
 
