diff --git a/old_product.java b/new_product.java
index 0df42c3..dc76de6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,13 @@
 public void handle(Order order, User creator, Date dateCreated, String other) {
-		if (order.getPatient() == null && order.getEncounter() != null)
+		if (order.getPatient() == null && order.getEncounter() != null) {
 			order.setPatient(order.getEncounter().getPatient());
+		}
+		
+		if (order.getOrderType() == null) {
+			OrderType orderType = Context.getOrderService().getOrderTypeByConcept(order.getConcept());
+			if (orderType == null) {
+				throw new APIException("No order type matches the concept class");
+			}
+			order.setOrderType(orderType);
+		}
 	}
\ No newline at end of file
