diff --git a/old_product.java b/new_product.java
index 765474c..d6750d1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,21 +1,13 @@
 public Object execute(CommandLine commandLine)
     throws Exception
   {
-    if(version == null){
+    if(versions == null){
       PluginResources resources = Services.getPluginResources("org.eclim");
-      String eclim_name = "eclim";
-      String eclipse_name = "eclipse";
-      int pad = Math.max(eclim_name.length(), eclipse_name.length());
 
-      String eclim_version = StringUtils.rightPad(eclim_name, pad) + ' ' +
-        resources.getProperty("pluginVersion");
-
-      String eclipse_version = getVersion();
-      eclipse_version = StringUtils.rightPad(
-          eclipse_name, pad) + ' ' + eclipse_version;
-
-      version = eclim_version + '\n' + eclipse_version;
+      versions = new HashMap<String,String>();
+      versions.put("eclim", resources.getProperty("pluginVersion"));
+      versions.put("eclipse", getVersion());
     }
 
-    return version;
+    return versions;
   }
\ No newline at end of file
