diff --git a/old_test.java b/new_test.java
index 43439c8..97d310b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,15 +11,15 @@
       inserted.put("a", new ByteArrayByteIterator(new byte[] {
           (byte) (i & 0xFF), (byte) (i >> 8 & 0xFF), (byte) (i >> 16 & 0xFF),
           (byte) (i >> 24 & 0xFF) }));
-      int result = client.insert(table, padded(i), inserted);
-      assertThat("Insert did not return success (0).", result, is(0));
+      Status result = client.insert(table, padded(i), inserted);
+      assertThat("Insert did not return success (0).", result, is(Status.OK));
     }
 
     Set<String> keys = Collections.singleton("a");
     Vector<HashMap<String, ByteIterator>> results =
         new Vector<HashMap<String, ByteIterator>>();
-    int result = client.scan(table, "00050", 5, null, results);
-    assertThat("Read did not return success (0).", result, is(0));
+    Status result = client.scan(table, "00050", 5, null, results);
+    assertThat("Read did not return success (0).", result, is(Status.OK));
     assertThat(results.size(), is(5));
     for (int i = 0; i < 5; ++i) {
       HashMap<String, ByteIterator> read = results.get(i);
