diff --git a/old_product.java b/new_product.java
index f102b65..8bd47d7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
-  public int scan(String table, String startkey, int recordcount,
+  public Status scan(String table, String startkey, int recordcount,
       Set<String> fields, Vector<HashMap<String, ByteIterator>> result) {
 
     try {
@@ -59,12 +59,12 @@
         result.add(tuple);
       }
 
-      return StatusCode.OK;
+      return Status.OK;
 
     } catch (Exception e) {
       e.printStackTrace();
       System.out.println("Error scanning with startkey: " + startkey);
-      return StatusCode.ERROR;
+      return Status.ERROR;
     }
 
   }
\ No newline at end of file
