diff --git a/old_test.java b/new_test.java
index 5c866f5..25ffbca 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
     public void login_withAccessToken() {
-        AuthenticationServer authServer = Mockito.mock(AuthenticationServer.class);
+        RealmObjectServer authServer = Mockito.mock(RealmObjectServer.class);
         when(authServer.loginUser(any(SyncCredentials.class), any(URL.class))).thenThrow(new AssertionError("Server contacted."));
-        AuthenticationServer originalServer = SyncManager.getAuthServer();
+        RealmObjectServer originalServer = SyncManager.getAuthServer();
         SyncManager.setAuthServerImpl(authServer);
         try {
             SyncCredentials credentials = SyncCredentials.accessToken("foo", "bar");
