diff --git a/old_product.java b/new_product.java
index 8a0a549..86976e4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,7 +13,7 @@ public static SyncUser logIn(final SyncCredentials credentials, final String aut
                 boolean isAdmin = (Boolean) credentials.getUserInfo().get("_isAdmin");
                 result = AuthenticateResponse.createValidResponseWithUser(userIdentifier, token, isAdmin);
             } else {
-                final AuthenticationServer server = SyncManager.getAuthServer();
+                final RealmObjectServer server = SyncManager.getAuthServer();
                 result = server.loginUser(credentials, authUrl);
             }
             if (result.isValid()) {
