diff --git a/old_test.java b/new_test.java
index 01273f5..2675eb3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,6 @@
   public void user() throws IOException {
     run("-cexit", "-cdrop user " + NAME);
     equals("5", new String[] { "-U" + NAME, "-P" + NAME, "-q5" },
-        new String[] { "-ccreate user " + NAME + ' ' + Token.md5(NAME) });
+        new String[] { "-ccreate user " + NAME + ' ' + NAME });
     run("-cexit", "-cdrop user " + NAME);
   }
\ No newline at end of file
