diff --git a/old_test.java b/new_test.java
index 7eed752..a144515 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,12 @@
 @Test
     public void test_putFilename() {
-        DBObject properties = new BasicDBObject();
-        String expectedFilename = "myFilename";
-        BodyInjectorHandler.putFilename(expectedFilename, "defaultFilename", properties);
-        assertEquals(expectedFilename, properties.get(BodyInjectorHandler.FILENAME));
+        BsonDocument properties = new BsonDocument();
+        BsonString expectedFilename = new BsonString("myFilename");
+        BodyInjectorHandler.putFilename(
+                expectedFilename.getValue(),
+                "defaultFilename", properties);
+
+        assertEquals(
+                expectedFilename,
+                properties.get(BodyInjectorHandler.FILENAME));
     }
\ No newline at end of file
