diff --git a/old_test.java b/new_test.java
index d7f26a6..7847a3f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,11 +4,11 @@
         final int numberOfEvents = 15;
         final String messageId = UUID.randomUUID().toString();
         final EventHubProducerAsyncClient producer = new EventHubClientBuilder()
-            .connectionString(getConnectionString()).buildAsyncProducer();
+            .connectionString(getConnectionString()).buildAsyncProducerClient();
         final EventHubConsumerClient receiver = new EventHubClientBuilder()
                 .connectionString(getConnectionString())
                 .consumerGroup(EventHubClientBuilder.DEFAULT_CONSUMER_GROUP_NAME)
-                .buildConsumer();
+                .buildConsumerClient();
 
         producer.send(TestUtils.getEvents(numberOfEvents, messageId), sendOptions).block();
 
