diff --git a/old_product.java b/new_product.java
index e80e19a..8955cc3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,9 @@
-static Iter search(final Data data, final int i, final Fun fun,
+static Iter search(final Data data, final byte[] str, final Fun fun,
       final QueryContext ctx) throws QueryException {
 
     final IndexContext ic = new IndexContext(ctx, data, null, true);
     if(!data.meta.ftindex) NOIDX.thrw(fun.input, fun);
 
-    final byte[] str = fun.checkStr(fun.expr[i], ctx);
     final FTOpt tmp = ctx.ftopt;
     ctx.ftopt = new FTOpt();
     ctx.ftopt.set(CS, data.meta.casesens);
