diff --git a/old_test.java b/new_test.java
index f83e1db..498cafd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
-@SuppressWarnings("serial")
-    @Test
+@Test
     public void testCompute1() {
         setupParameters();
         parameters.setInputDimensions(new int[] { 9 });
@@ -43,8 +42,7 @@
         }
 
         for(int i = 0;i < mem.getNumColumns();i++) {
-//            System.out.println(Arrays.toString((int[])mem.getConnectedCounts().getSlice(i)));
-//            System.out.println(Arrays.toString(mem.getPotentialPools().get(i).getDensePermanences(mem)));
-            assertTrue(Arrays.equals(inputVector, ((int[])mem.getConnectedCounts().getSlice(i))));
+            int[] permanences = ArrayUtils.toIntArray(mem.getPotentialPools().get(i).getDensePermanences(mem));
+            assertTrue(Arrays.equals(inputVector, permanences));
         }
     }
\ No newline at end of file
