diff --git a/old_test.java b/new_test.java
index 967aff3..241a999 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,11 +5,11 @@
 		Order order = Context.getOrderService().getOrderByUuid(uuid);
 		String discontinuedReasonNonCoded = "Non coded discontinued reason";
 		
-		order.setDiscontinuedReasonNonCoded(discontinuedReasonNonCoded);
+		order.setOrderReasonNonCoded(discontinuedReasonNonCoded);
 		OrderService orderService = Context.getOrderService();
 		orderService.saveOrder(order);
 		
 		order = Context.getOrderService().getOrderByUuid(uuid);
 		
-		Assert.assertEquals(discontinuedReasonNonCoded, order.getDiscontinuedReasonNonCoded());
+		Assert.assertEquals(discontinuedReasonNonCoded, order.getOrderReasonNonCoded());
 	}
\ No newline at end of file
