diff --git a/old_product.java b/new_product.java
index d49dcab..3c39e20 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,7 +11,7 @@ public <K,V> List<Model<Vector, Double>> update(GDBTrainer.GDBModel mdlToUpdate,
         for (int i = 0; i < cntOfIterations; i++) {
             double[] weights = Arrays.copyOf(compositionWeights, models.size());
 
-            WeightedPredictionsAggregator aggregator = new WeightedPredictionsAggregator(weights, meanLabelValue);
+            WeightedPredictionsAggregator aggregator = new WeightedPredictionsAggregator(weights, meanLbVal);
             ModelsComposition currComposition = new ModelsComposition(models, aggregator);
             if (convCheck.isConverged(datasetBuilder, currComposition))
                 break;
