diff --git a/old_product.java b/new_product.java
index 8cee4d5..373133c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -25,10 +25,12 @@ public static String replace(final String p, final InputSource input,
 
     try {
       if(lock) ctx.register(true);
-      // delete documents
-      for(int d = docs.size() - 1; d >= 0; d--) data.delete(docs.get(d));
       // add new document
       Add.add(path, target, input, ctx, null, false);
+      // delete old documents if addition was successful
+      for(int d = docs.size() - 1; d >= 0; d--) data.delete(docs.get(d));
+      // flushes changes
+      data.flush();
     } finally {
       if(lock) ctx.unregister(true);
     }
