diff --git a/old_product.java b/new_product.java
index e31db3e..16796a0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,7 +13,7 @@ public CompletableFuture<Long> getOrAssignStreamSegmentId(String streamSegmentNa
         // See if anyone else is currently waiting to get this StreamSegment's id.
         CompletableFuture<Long> result;
         boolean needsAssignment = false;
-        synchronized (assignmentLock) {
+        synchronized (this.assignmentLock) {
             result = this.pendingRequests.getOrDefault(streamSegmentName, null);
             if (result == null) {
                 needsAssignment = true;
