diff --git a/old_product.java b/new_product.java
index cecaef9..45e4148 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,9 @@
             @NotBlank(message = "No id entered. Unable to get.")
             final String id
     ) throws GenieException {
-        LOG.debug("Called with id " + id);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Called with id " + id);
+        }
         final Cluster cluster = this.clusterRepo.findOne(id);
         if (cluster != null) {
             return cluster;
