diff --git a/old_product.java b/new_product.java
index 956ddd5..4d2ba69 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @PrePersist
     @PreUpdate
-    protected void onCreateOrUpdateApplication() throws GenieException {
-        this.validate(this.status);
+    protected void onCreateOrUpdateApplication() throws GeniePreconditionException {
+        this.validate(this.status, null);
         // Add the id to the tags
         if (this.tags == null) {
             this.tags = new HashSet<>();
