diff --git a/old_test.java b/new_test.java
index b6a8199..bfe7794 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testOnCreateOrUpdateApplication() throws GenieException {
+    public void testOnCreateOrUpdateApplication() throws GeniePreconditionException {
         this.a = new Application(NAME, USER, ApplicationStatus.ACTIVE, VERSION);
         this.a.onCreateOrUpdateApplication();
     }
\ No newline at end of file
