diff --git a/old_product.java b/new_product.java
index 125159c..20eefe5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,8 +11,8 @@ public void add(final int pre, final FTMatches all) {
       c = -c - 1;
       if(size == pos.length) pos = Arrays.copyOf(pos, size << 1);
       Array.move(pos, c, 1, size++ - c);
-      pos[c] = new FTPos(pre, ps.finish());
+      pos[c] = new FTPos(pre, ps.toArray());
     } else {
-      pos[c].union(ps.finish());
+      pos[c].union(ps.toArray());
     }
   }
\ No newline at end of file
