diff --git a/old_test.java b/new_test.java
index d68c79e..5a2994f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
     public void testToString() {
 //        System.out.println("toString");
-        ResidueNumber instance = new ResidueNumber("A", 42, " ");
-        String expResult = "Chain:A resNum:42 insCode: ";
+        ResidueNumber instance = new ResidueNumber("A", 42, ' ');
+        String expResult = "42";
         String result = instance.toString();
         assertEquals(expResult, result);
 
