diff --git a/old_product.java b/new_product.java
index 2e98625..7ca941f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,13 @@
 public String toString(){
 
 		String str = "AminoAcid "+ recordType + ":"+ pdb_name + " " + amino_char +
-		" " + pdb_code +  " "+ pdb_flag + " " + recordType  ;
+		" " + residueNumber +  " "+ pdb_flag + " " + recordType  ;
 		if (pdb_flag) {
 			str = str + " atoms: "+atoms.size();
 		}
+		if ( altLocs != null)
+			str += " has altLocs :" + altLocs.size(); 
+
 		return str ;
 
 	}
\ No newline at end of file
