diff --git a/old_product.java b/new_product.java
index 7a3c15e..a770542 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,10 +13,7 @@ public void validate(Object obj, Errors errors) {
 			ValidationUtils.rejectIfEmpty(errors, "voided", "error.null");
 			ValidationUtils.rejectIfEmpty(errors, "concept", "Concept.noConceptSelected");
 			ValidationUtils.rejectIfEmpty(errors, "patient", "error.null");
-			
-			//TODO commented out because of this commit:
-			//https://github.com/openmrs/openmrs-core/commit/6360b0e78ee98f75eef10bf37d7cbda2e67d5ce9
-			//ValidationUtils.rejectIfEmpty(errors, "orderType", "error.null");
+			ValidationUtils.rejectIfEmpty(errors, "orderType", "error.null");
 			
 			Date startDate = order.getStartDate();
 			if (startDate != null) {
