diff --git a/old_product.java b/new_product.java
index 523963e..a1cd7ae 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,7 +7,7 @@
         if (action instanceof ManagedTask)
             throw new IllegalArgumentException(ManagedTask.class.getName());
 
-        WSContextService contextSvc = defaultExecutor.getContextService();
+        WSContextService contextSvc = ((WSManagedExecutorService) defaultExecutor).getContextService();
 
         @SuppressWarnings("unchecked")
         ThreadContextDescriptor contextDescriptor = contextSvc.captureThreadContext(XPROPS_SUSPEND_TRAN);
