diff --git a/old_product.java b/new_product.java
index d94d2c0..5a1c4d8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,11 @@
-public synchronized void start() {
+public synchronized void start(String prefix) {
 
 	// You have to ooccupie a Server to start it
 	if (!this.isFree()) {
 	    if (p != null) {
 		p.destroy();
 	    }
-	    restart();
+	    restart(prefix);
 	} else {
 	    throw new IllegalStateException("Cant start a not marked Server. Occupie it first!");
 	}
