diff --git a/old_test.java b/new_test.java
index 1d68eef..ea094fd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test(description = "GET /vApp/{id}")
    public void testGetVm() {
       // The method under test
-      vm = vmApi.get(vmURI);
+      vm = vmApi.get(vmUrn);
 
       // Check the retrieved object is well formed
       checkVm(vm);
@@ -11,5 +11,5 @@
                String.format(OBJ_FIELD_EQ, VM, "deployed", "FALSE", vm.isDeployed().toString()));
 
       // Check status
-      assertVmStatus(vm.getHref(), Status.POWERED_OFF);
+      assertVmStatus(vmUrn, Status.POWERED_OFF);
    }
\ No newline at end of file
