diff --git a/old_product.java b/new_product.java
index 85cf8dc..09071cd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,4 +2,4 @@
    @Consumes(VM)
    @JAXBResponseParser
    @ExceptionParser(ReturnNullOnNotFoundOr404.class)
-   ListenableFuture<Vm> getVm(@EndpointParam URI vmURI);
\ No newline at end of file
+   ListenableFuture<Vm> get(@EndpointParam(parser = VmURNToHref.class) String vmUrn);
\ No newline at end of file
