diff --git a/old_product.java b/new_product.java
index f1fa120..5fa076c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -83,6 +83,9 @@ public Order saveOrder(Order order, OrderContext orderContext) throws APIExcepti
 				throw new APIException("The class does not match that of the previous order");
 			}
 		}
+		if (order.getConcept() == null && DrugOrder.class.isAssignableFrom(order.getClass())) {
+			order.setConcept(((DrugOrder) order).getDrug().getConcept());
+		}
 		
 		return saveOrderInternal(order, orderContext);
 	}
\ No newline at end of file
