diff --git a/old_product.java b/new_product.java
index d40a4f7..3bf312a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,11 @@ public void process(RelationContainer relationContainer) {
 		Relation inputEntity = relationContainer.getEntity();
 		Relation outputEntity;
 		
-		outputEntity = new Relation(inputEntity.getId(), timestamp, inputEntity.getUser());
+		outputEntity = new Relation(inputEntity.getId(),
+							timestamp,
+							inputEntity.getUserName(),
+							inputEntity.getUserId(),
+							inputEntity.getVersion());
 		
 		outputEntity.addTags(inputEntity.getTagList());
 		
