diff --git a/old_product.java b/new_product.java
index 1fcc404..557b94a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static boolean checkConsistency(Map<UUID, GridCacheVersion> oldKeyVer,
-        Map<UUID, GridCacheVersion> actualKeyVer) {
+        Map<UUID, VersionedValue> actualKeyVer) {
         assert oldKeyVer.size() > 1;
 
         if (actualKeyVer.isEmpty())
@@ -25,16 +25,16 @@ public static boolean checkConsistency(Map<UUID, GridCacheVersion> oldKeyVer,
         GridCacheVersion maxVer = oldKeyVer.get(maxVersions.iterator().next());
 
         for (UUID maxVerOwner : maxVersions) {
-            GridCacheVersion ver = actualKeyVer.get(maxVerOwner);
-            if (ver == null || maxVer.isLess(ver))
+            VersionedValue verVal = actualKeyVer.get(maxVerOwner);
+            if (verVal == null || maxVer.isLess(verVal.getVer()))
                 return true;
         }
 
         boolean allNonMaxChanged = true;
 
         for (Map.Entry<UUID, GridCacheVersion> entry : oldKeyVer.entrySet()) {
-            GridCacheVersion actualVer = actualKeyVer.get(entry.getKey());
-            if (!maxVersions.contains(entry.getKey()) && (actualVer == null || !actualVer.isGreaterEqual(maxVer))) {
+            VersionedValue actualVer = actualKeyVer.get(entry.getKey());
+            if (!maxVersions.contains(entry.getKey()) && (actualVer == null || !actualVer.getVer().isGreaterEqual(maxVer))) {
                 allNonMaxChanged = false;
 
                 break;
