diff --git a/old_test.java b/new_test.java
index e72ac99..fdcc644 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,9 +3,7 @@
         Dependency instance = new Dependency();
         MavenArtifact mavenArtifact = new MavenArtifact("group", "artifact", "version", "url");
         instance.addAsEvidence("pom", mavenArtifact, Confidence.HIGH);
-        assertTrue(instance.getEvidence().contains(Confidence.HIGH));
-        assertFalse(instance.getEvidence().getEvidence("pom", "groupid").isEmpty());
-        assertFalse(instance.getEvidence().getEvidence("pom", "artifactid").isEmpty());
-        assertFalse(instance.getEvidence().getEvidence("pom", "version").isEmpty());
+        assertTrue(instance.contains(EvidenceType.VENDOR, Confidence.HIGH));
+        assertTrue(instance.size()>1);
         assertFalse(instance.getIdentifiers().isEmpty());
     }
\ No newline at end of file
