diff --git a/old_product.java b/new_product.java
index 2899854..76666af 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-private void updateTrackLocation(ImageFloat32 image) {
+protected void updateTrackLocation(ImageFloat32 image) {
 		get_subwindow(image, region.tl_x, region.tl_y, subInput);
 
 		// calculate response of the classifier at all locations
@@ -23,12 +23,12 @@ private void updateTrackLocation(ImageFloat32 image) {
 		}
 
 		// peak in region's coordinate system
-		int cx = indexBest % tmpReal0.width;
-		int cy = indexBest / tmpReal0.width;
+		int deltaX = subInput.width/2 - (indexBest % tmpReal0.width);
+		int deltaY = subInput.height/2 - (indexBest / tmpReal0.width);
 
 		// convert peak location into image coordinate system
-		region.tl_x = (region.tl_x+cx) - subInput.width/2;
-		region.tl_y = (region.tl_y+cy) - subInput.height/2;
+		region.tl_x = region.tl_x + deltaX;
+		region.tl_y = region.tl_y + deltaY;
 
 		ensureInBounds(region,image.width,image.height);
 	}
\ No newline at end of file
