diff --git a/old_test.java b/new_test.java
index 8839dfa..1aa2a52 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,5 +8,5 @@
 		new LocationValidator().validate(location, errors);
 		
 		Assert.assertTrue(errors.hasFieldErrors("localizedName.unlocalizedValue"));
-		Assert.assertFalse(errors.hasFieldErrors("description"));
+		Assert.assertFalse(errors.hasFieldErrors("localizedDescription.unlocalizedValue"));
 	}
\ No newline at end of file
