diff --git a/old_product.java b/new_product.java
index 69f24f3..0d9cad4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,6 +5,7 @@ public void validate(Object obj, Errors errors) {
 		} else {
 			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "localizedName.unlocalizedValue",
 		    "LocalizedName.unlocalizedName.empty");
-			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "description", "error.description");
+			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "localizedDescription.unlocalizedValue",
+			    "LocalizedDescription.unlocalizedDescription.empty");
 		}
 	}
\ No newline at end of file
