diff --git a/old_test.java b/new_test.java
index 347cec7..cc46a2d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,6 @@
 		String[] currentState = new String[] { "new" };
 		ImmutableEntityInterceptor interceptor = new SomeImmutableEntityInterceptor();
 		expectedException.expect(APIException.class);
-		expectedException.expectMessage(is("Editing some fields on " + interceptor.getSupportedType().getSimpleName()
-		        + " is not allowed"));
+		expectedException.expectMessage(is("editing.fields.not.allowed"));
 		interceptor.onFlushDirty(new Order(), null, currentState, previousState, propertyNames, null);
 	}
\ No newline at end of file
