diff --git a/old_test.java b/new_test.java
index c292d01..d5d4b9c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-	public void allInside_intrinsic() {
+	public void transform_F32_shrink_intrinsic() {
 
 		// distorted pixel in original image
 		float pixelX = 12.5f,pixelY = height-3;
@@ -13,7 +13,7 @@
 		distToNorm.compute(pixelX, pixelY, norm);
 
 		IntrinsicParameters adjusted = new IntrinsicParameters();
-		PointTransform_F32 distToAdj = LensDistortionOps.allInside(orig,adjusted,false);
+		PointTransform_F32 distToAdj = LensDistortionOps.transform_F32(AdjustmentType.SHRINK, orig, adjusted, false);
 
 		Point2D_F32 adjPixel = new Point2D_F32();
 		Point2D_F32 normFound = new Point2D_F32();
