diff --git a/old_test.java b/new_test.java
index 3de8df5..b290830 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,29 @@
     HttpMethed.waitToProduceOneBlock(httpnode);
 
     //update brokerage
-    response = HttpMethed.updateBrokerage(httpnode, witnessAddress, 30L, witnessKey);
+    response = HttpMethed.updateBrokerage(httpnode, witnessAddress, 11L, witnessKey);
     Assert.assertTrue(HttpMethed.verificationResult(response));
     HttpMethed.waitToProduceOneBlock(httpnode);
+
+    response = HttpMethed.sendCoin(httpnode, fromAddress, witnessAddress2, amount, testKey002);
+    Assert.assertTrue(HttpMethed.verificationResult(response));
+    HttpMethed.waitToProduceOneBlock(httpnode);
+
+//    //update brokerage onvisible
+//    response = HttpMethed
+//        .updateBrokerageOnVisible(httpnode, witnessAddress2, 24L, witnessKey2, "true");
+//    Assert.assertTrue(HttpMethed.verificationResult(response));
+//    HttpMethed.waitToProduceOneBlock(httpnode);
+//
+//    //update brokerage onvisible
+//    response = HttpMethed
+//        .updateBrokerageOnVisible(httpnode, witnessAddress, 88L, witnessKey, "false");
+//    Assert.assertTrue(HttpMethed.verificationResult(response));
+//    HttpMethed.waitToProduceOneBlock(httpnode);
+//
+//    //update brokerage onvisible
+//    response = HttpMethed
+//        .updateBrokerageOnVisible(httpnode, fromAddress, 78L, testKey002, "true");
+////    Assert.assertTrue(HttpMethed.verificationResult(response));
+//    HttpMethed.waitToProduceOneBlock(httpnode);
   }
\ No newline at end of file
