diff --git a/old_product.java b/new_product.java
index c0810d4..aad8f25 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,6 +5,7 @@ public static HttpResponse updateBrokerage(String httpNode, byte[] ownerAddress,
       JsonObject userBaseObj2 = new JsonObject();
       userBaseObj2.addProperty("owner_address", ByteArray.toHexString(ownerAddress));
       userBaseObj2.addProperty("brokerage", brokerage);
+      logger.info("userBaseObj2:" + userBaseObj2);
       response = createConnect(requestUrl, userBaseObj2);
       transactionString = EntityUtils.toString(response.getEntity());
       transactionSignString = gettransactionsign(httpNode, transactionString, fromKey);
