diff --git a/old_product.java b/new_product.java
index 45dd5e0..523963e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,5 +14,5 @@
         action = new ContextualConsumer<>(contextDescriptor, action);
 
         CompletableFuture<Void> dependentStage = completableFuture.acceptEither(other, action);
-        return new ManagedCompletableFuture<Void>(dependentStage, defaultExecutor);
+        return new ManagedCompletableFuture<Void>(dependentStage, defaultExecutor, null);
     }
\ No newline at end of file
