diff --git a/old_product.java b/new_product.java
index 42e20a3..e05b73c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @Override
     public void deleteComponent(String id) throws IOException {
-        Path compFile = registryPath.resolve(Paths.get(id + COMPONENT_FILE_SUFFIX));
-        Path listFile = registryPath.resolve(Paths.get(id + LIST_FILE_SUFFIX));
+        Path compFile = registryPath.resolve(SystemUtils.fileName(id + COMPONENT_FILE_SUFFIX));
+        Path listFile = registryPath.resolve(SystemUtils.fileName(id + LIST_FILE_SUFFIX));
         Files.deleteIfExists(compFile);
         Files.deleteIfExists(listFile);
     }
\ No newline at end of file
