diff --git a/old_test.java b/new_test.java
index cdba713..313c73a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-	@Verifies(value = "should add order to non nul initial order set", method = "addOrder(Order)")
-	public void addOrder_shouldAddOrderToNonNulInitialOrderSet() throws Exception {
+	@Verifies(value = "should add order to non null initial order set", method = "addOrder(Order)")
+	public void addOrder_shouldAddOrderToNonNullInitialOrderSet() throws Exception {
 		Encounter encounter = new Encounter();
 		Set<Order> orderSet = new HashSet<Order>();
 		orderSet.add(new Order(1));
