diff --git a/old_test.java b/new_test.java
index 1b93acc..3f1f5d3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,14 @@
 @Test
-    public void testMakeMaskedAddress() {
-        Ip4Address ip4Address = new Ip4Address("1.2.3.5");
-        Ip4Address ip4AddressMasked =
-            Ip4Address.makeMaskedAddress(ip4Address, 24);
-        assertThat(ip4AddressMasked.toString(), is("1.2.3.0"));
+    public void testMakeMaskedAddressIPv4() {
+        Ip4Address ipAddress = Ip4Address.valueOf("1.2.3.5");
+        Ip4Address ipAddressMasked;
 
-        ip4AddressMasked = Ip4Address.makeMaskedAddress(ip4Address, 0);
-        assertThat(ip4AddressMasked.toString(), is("0.0.0.0"));
+        ipAddressMasked = Ip4Address.makeMaskedAddress(ipAddress, 24);
+        assertThat(ipAddressMasked.toString(), is("1.2.3.0"));
 
-        ip4AddressMasked = Ip4Address.makeMaskedAddress(ip4Address, 32);
-        assertThat(ip4AddressMasked.toString(), is("1.2.3.5"));
+        ipAddressMasked = Ip4Address.makeMaskedAddress(ipAddress, 0);
+        assertThat(ipAddressMasked.toString(), is("0.0.0.0"));
+
+        ipAddressMasked = Ip4Address.makeMaskedAddress(ipAddress, 32);
+        assertThat(ipAddressMasked.toString(), is("1.2.3.5"));
     }
\ No newline at end of file
