diff --git a/old_product.java b/new_product.java
index 5af818f..26bd0b9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,5 @@
-public static Ip4Address makeMaskedAddress(final Ip4Address addr,
-                                               int prefixLen) {
-        Ip4Address mask = Ip4Address.makeMaskPrefix(prefixLen);
-        long v = addr.value & mask.value;
-
-        return new Ip4Address((int) v);
+public static Ip4Address makeMaskedAddress(final Ip4Address address,
+                                               int prefixLength) {
+        byte[] net = makeMaskedAddressArray(address, prefixLength);
+        return Ip4Address.valueOf(net);
     }
\ No newline at end of file
