diff --git a/old_test.java b/new_test.java
index ca2274a..dc4dff5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,10 @@
 @Test
-    public void testProcess() {
+    public void testProcess() throws CpeValidationException {
         //File struts = new File(this.getClass().getClassLoader().getResource("struts2-core-2.1.2.jar").getPath());
         File struts = BaseTest.getResourceAsFile(this, "struts2-core-2.1.2.jar");
         Dependency dependency = new Dependency(struts);
-        dependency.addIdentifier("cpe", "cpe:/a:microsoft:.net_framework:4.5", "some url not needed for this test");
+        CpeIdentifier cpeId = new CpeIdentifier("microsoft", ".net_framework", "4.5", Confidence.HIGH);
+        dependency.addVulnerableSoftwareIdentifier(cpeId);
         String sha1 = dependency.getSha1sum();
         dependency.setSha1sum("384FAA82E193D4E4B0546059CA09572654BC3970");
         Vulnerability v = createVulnerability();
@@ -48,7 +49,7 @@
         pt.setValue("cpe:/a:microsoft:.net_framework:4.0");
         instance.addCpe(pt);
         instance.process(dependency);
-        assertTrue(dependency.getIdentifiers().size() == 1);
+        assertTrue(dependency.getVulnerableSoftwareIdentifiers().size() == 1);
         pt = new PropertyType();
         pt.setValue("cpe:/a:microsoft:.net_framework:4.5");
         instance.addCpe(pt);
@@ -57,20 +58,21 @@
         pt.setRegex(true);
         instance.setFilePath(pt);
         instance.process(dependency);
-        assertTrue(dependency.getIdentifiers().isEmpty());
+        assertTrue(dependency.getVulnerableSoftwareIdentifiers().isEmpty());
         assertEquals(1, dependency.getSuppressedIdentifiers().size());
 
         instance = new SuppressionRule();
-        dependency.addIdentifier("cpe", "cpe:/a:microsoft:.net_framework:4.0", "some url not needed for this test");
-        dependency.addIdentifier("cpe", "cpe:/a:microsoft:.net_framework:4.5", "some url not needed for this test");
-        dependency.addIdentifier("cpe", "cpe:/a:microsoft:.net_framework:5.0", "some url not needed for this test");
+
+        dependency.addVulnerableSoftwareIdentifier(new CpeIdentifier("microsoft", ".net_framework", "4.0", Confidence.HIGH));
+        dependency.addVulnerableSoftwareIdentifier(new CpeIdentifier("microsoft", ".net_framework", "4.5", Confidence.HIGH));
+        dependency.addVulnerableSoftwareIdentifier(new CpeIdentifier("microsoft", ".net_framework", "5.0", Confidence.HIGH));
         pt = new PropertyType();
         pt.setValue("cpe:/a:microsoft:.net_framework");
         instance.addCpe(pt);
         instance.setBase(true);
-        assertEquals(3, dependency.getIdentifiers().size());
+        assertEquals(3, dependency.getVulnerableSoftwareIdentifiers().size());
         assertEquals(1, dependency.getSuppressedIdentifiers().size());
         instance.process(dependency);
-        assertTrue(dependency.getIdentifiers().isEmpty());
+        assertTrue(dependency.getVulnerableSoftwareIdentifiers().isEmpty());
         assertEquals(1, dependency.getSuppressedIdentifiers().size());
     }
\ No newline at end of file
