diff --git a/old_product.java b/new_product.java
index 0c22529..2d8c231 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public static String getOriginalImageUrl(String wikiUrl) throws NoSuchAlgorithmException {
 	  MessageDigest md = MessageDigest.getInstance("MD5");
 	  
-	  String imageName = wikiUrl.substring(wikiUrl.indexOf("/wiki/File:") + "/wiki/File:".length());
+	  String imageName = wikiUrl.substring(wikiUrl.indexOf("/wiki/Special:FilePath/") + "/wiki/Special:FilePath/".length());
 	  
 	  StringBuffer hashedName = new StringBuffer();
 	  md.update(imageName.getBytes(StandardCharsets.UTF_8));
@@ -10,6 +10,5 @@ public static String getOriginalImageUrl(String wikiUrl) throws NoSuchAlgorithmE
 	  for (int i = 0; i < byteData.length; i++) {
 	    hashedName.append(Integer.toString((byteData[i] & 0xff) + 0x100, 16).substring(1));
 	  }
-	  
     return (IMAGE_ORIGINALURL_TEMPLATE + hashedName.charAt(0) + "/" + hashedName.charAt(0) + hashedName.charAt(1) + "/" + imageName);
 	}
\ No newline at end of file
