diff --git a/old_product.java b/new_product.java
index 29f6e18..cc650f7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,15 +5,16 @@ public void changeKeyFrame() {
 		List<PointTrack> spawned = tracker.getNewTracks(null);
 
 		for( PointTrack l : spawned ) {
-			AssociatedPair p = l.getCookie();
+			AssociatedPairTrack p = l.getCookie();
 			if( p == null ) {
-				l.cookie = p = new AssociatedPair();
+				l.cookie = p = new AssociatedPairTrack();
 				// little bit of trickery here.  Save the reference so that the point
 				// in the current frame is updated for free as PointTrack is
 				p.p2 = l;
 				l.cookie = p;
 			}
-			p.p1.set(p.p2);
+			p.p1.set(l);
+			p.lastUsed = totalProcessed;
 		}
 
 		totalSpawned = spawned.size();
