diff --git a/old_test.java b/new_test.java
index f7b6953..425b015 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
     
     assertThat(v1, is(equalTo(v1)));
     assertThat(v1, is(equalTo(v1_copy)));
-    assertFalse(Equals(unitVector(6, 2), v1));
+    assertThat(unitVector(6, 2), is(not(equalTo(v1))));
     
     
     // Equals(Matrix)
@@ -31,7 +31,7 @@
         
     assertThat(m1, is(equalTo(m1)));
     assertThat(m1, is(equalTo(m1_copy)));
-    assertFalse(Equals(identity(6, 3), m1));
+    assertThat(identity(6, 3), is(not(equalTo(m1))));
     
     // TODO: Question: more testcases?
     
