diff --git a/old_test.java b/new_test.java
index 34b97e7..8ba60af 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,8 +4,6 @@
         method = "equals",
         args = {java.lang.Object.class}
     )
-    @KnownFailure("equals(Onject o) method throws java.lang.ClassCastException " +
-            "for TreeMap objects with different key objects.")
     public void test_equals() throws Exception {
         // comparing TreeMaps with different object types
         Map m1 = new TreeMap();
