diff --git a/old_product.java b/new_product.java
index 0f3adb1..dd7f1dd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,9 @@
 @Override
-        public boolean equals(Object an) {
-            if (an == null) {
+        public boolean equals(Object object) {
+            if (object == null || object.getClass() != getClass()
+                    || object.hashCode() != hashCode()) {
                 return false;
             }
-            return an.getClass() == this.getClass()
-                    && name.equalsIgnoreCase(((Name) an).name);
+
+            return Util.equalsIgnoreCase(name, ((Name) object).name);
         }
\ No newline at end of file
