diff --git a/old_product.java b/new_product.java
index 50a203a..60d2f3b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @RestAPI
 	@PreAuthorize("hasAnyRole('A')")
 	@RequestMapping(value = "/{userId}", method = RequestMethod.GET)
-	public HttpEntity<String> getOne(@PathVariable("userId") String userId) {
-		return toJsonHttpEntity(userService.getOne(userId));
+	public User getOne(@PathVariable("userId") String userId) {
+		return userService.getOne(userId);
 	}
\ No newline at end of file
