diff --git a/old_product.java b/new_product.java
index fa61483..622154b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -23,7 +23,7 @@ public void process() {
 
 			boolean success = false;
 			if( mode == 2 ) {
-				RectangleCorner2D_F64 rect = new RectangleCorner2D_F64();
+				Rectangle2D_F64 rect = new Rectangle2D_F64();
 				rect.set(point0.x, point0.y, point1.x, point1.y);
 				UtilPolygons2D_F64.convert(rect, target);
 				success = tracker.initialize(input,target);
