diff --git a/old_test.java b/new_test.java
index 224584f..e742b94 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
 	public void process() {
 		TldRegionTracker alg = createAlg();
 
-		RectangleCorner2D_F64 rect = new RectangleCorner2D_F64(10,20,115,125);
+		Rectangle2D_F64 rect = new Rectangle2D_F64(10,20,115,125);
 
 		alg.initialize(pyramid);
 		assertTrue(alg.process(pyramid, rect));
