diff --git a/old_product.java b/new_product.java
index 6f7298c..637b6c8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,5 @@
     @ResponseStatus(HttpStatus.OK)
     public Set<String> getTagsForApplication(@PathVariable("id") final String id) throws GenieException {
         log.debug("Called with id {}", id);
-        return this.applicationService.getTagsForApplication(id);
+        return DtoAdapters.toV3Application(this.applicationService.getApplication(id)).getTags();
     }
\ No newline at end of file
