diff --git a/old_product.java b/new_product.java
index 338455b..165c1ca 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@ public NetworkInterfaces networkInterfaces() {
                 this.networkManagementClient.networkInterfaces(),
                 this.networks(),
                 this.publicIpAddresses(),
-                this.resourceManager.resourceGroups()
+                this.resourceManager
             );
         }
         return this.networkInterfaces;
