diff --git a/old_product.java b/new_product.java
index 4a7211b..ffa9db8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,22 +1,15 @@
-private static byte[] info(final CmdIndexInfo idx, final Data data) {
+public static byte[] info(final CmdIndexInfo idx, final Data data) {
     switch(idx) {
-      case TAG:
-        return info(INFOTAGS, IndexType.TAG, data);
-      case ATTNAME:
-        return info(INFOATTS, IndexType.ATTNAME, data);
-      case TEXT:
-        return !data.meta.textindex ? Token.EMPTY :
-          info(INFOTEXTINDEX, IndexType.TEXT, data);
-      case ATTRIBUTE:
-        return !data.meta.attrindex ? Token.EMPTY :
-          info(INFOATTRINDEX, IndexType.ATTRIBUTE, data);
-      case FULLTEXT:
-        return !data.meta.ftindex ? Token.EMPTY :
-          info(INFOFTINDEX, IndexType.FULLTEXT, data);
-      case PATH:
-        return !data.meta.pathindex ? Token.EMPTY :
-          info(INFOPATHINDEX, IndexType.PATH, data);
-      default:
-        return Token.EMPTY;
+      case TAG:       return info(INFOTAGS, IndexType.TAG, data, true);
+      case ATTNAME:   return info(INFOATTS, IndexType.ATTNAME, data, true);
+      case TEXT:      return info(INFOTEXTINDEX, IndexType.TEXT, data,
+          data.meta.textindex);
+      case ATTRIBUTE: return info(INFOATTRINDEX, IndexType.ATTRIBUTE, data,
+          data.meta.attrindex);
+      case FULLTEXT:  return info(INFOFTINDEX, IndexType.FULLTEXT, data,
+          data.meta.ftindex);
+      case PATH:      return info(INFOPATHINDEX, IndexType.PATH, data,
+          data.meta.pathindex);
+      default:        return Token.token(LI + INDNOTAVL);
     }
   }
\ No newline at end of file
