diff --git a/old_test.java b/new_test.java
index d35e221..e85f142 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
         assertNotNull(connection1);
         AnnoManagedConnectionFactory mcf = connection1.getMCF();
         assertNotNull(mcf);
-        log.info("MCF:" + mcf + "//1//" + mcf.getFirst() + "//2//"
+        log.trace("MCF:" + mcf + "//1//" + mcf.getFirst() + "//2//"
                 + mcf.getSecond());
         assertEquals((byte) 4, (byte) mcf.getFirst());
         assertEquals((short) 0, (short) mcf.getSecond());
