diff --git a/old_product.java b/new_product.java
index 2039623..5a1856f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
     public AnnoConnection getConnection() throws ResourceException {
-        log.finest("getConnection()");
+        log.trace("getConnection()");
         return (AnnoConnection) connectionManager.allocateConnection(mcf, null);
     }
\ No newline at end of file
