diff --git a/old_product.java b/new_product.java
index d7dee6d..6fbe146 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,6 +16,7 @@ public SpanId startNewSpan(@Nullable String requestName) {
             return null;
         }
 
-        currentSpan().setCurrentSpan(newSpan.setName(requestName));
+        recorder().name(newSpan, requestName);
+        currentSpan().setCurrentSpan(newSpan);
         return nextContext;
     }
\ No newline at end of file
