diff --git a/old_test.java b/new_test.java
index d99d3a4..5547081 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,11 +12,13 @@
                 .build()
         );
 
-        Span started = brave.localSpanThreadBinder().getCurrentLocalSpan();
+        assertThat(spans).isEmpty(); // doesn't flush on start
+        recorder.flush(brave.localSpanThreadBinder().get());
 
-        assertEquals(START_TIME_MICROSECONDS, started.getTimestamp().longValue());
-        assertEquals("lc", started.getBinary_annotations().get(0).getKey());
-        assertEquals(COMPONENT_NAME, new String(started.getBinary_annotations().get(0).getValue(), Util.UTF_8));
-        assertThat(started.getBinary_annotations().get(0).host).isEqualTo(endpoint);
-        assertEquals(OPERATION_NAME, started.getName());
+        zipkin.Span started = spans.get(0);
+        assertThat(started.timestamp).isEqualTo(START_TIME_MICROSECONDS);
+        assertThat(started.name).isEqualTo(OPERATION_NAME);
+        assertThat(started.binaryAnnotations).containsExactly(
+            zipkin.BinaryAnnotation.create("lc", COMPONENT_NAME, ZIPKIN_ENDPOINT)
+        );
     }
\ No newline at end of file
