diff --git a/old_product.java b/new_product.java
index 623cc8a..9d34e27 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,11 +3,11 @@ private void object(final JsonHandler h) throws QueryException {
     h.openObject();
     if(!consumeWs('}', false)) {
       do {
-        h.openPair(spec != Spec.LIBERAL || curr() == '"' ? string() : unquoted());
+        h.openPair(spec != JsonSpec.LIBERAL || curr() == '"' ? string() : unquoted());
         consumeWs(':', true);
         value(h);
         h.closePair();
-      } while(consumeWs(',', false) && !(spec == Spec.LIBERAL && curr() == '}'));
+      } while(consumeWs(',', false) && !(spec == JsonSpec.LIBERAL && curr() == '}'));
       consumeWs('}', true);
     }
     h.closeObject();
