diff --git a/old_test.java b/new_test.java
index d0d359a..0da10d1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testRemoveAllTagsForCluster() throws GenieException {
-        Assert.assertEquals(5, this.service.getTagsForCluster(CLUSTER_1_ID).size());
+        Assert.assertEquals(3, this.service.getTagsForCluster(CLUSTER_1_ID).size());
         this.service.removeAllTagsForCluster(CLUSTER_1_ID);
-        Assert.assertEquals(2, this.service.getTagsForCluster(CLUSTER_1_ID).size());
+        Assert.assertEquals(0, this.service.getTagsForCluster(CLUSTER_1_ID).size());
     }
\ No newline at end of file
