diff --git a/old_product.java b/new_product.java
index d2e5374..6868f18 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,4 @@
 public DateTimeFormatterBuilder appendZoneText(TextStyle textStyle) {
-        // TODO: parsing of zone text?
-//        * During parsing, either the textual zone name, the zone ID or the offset
-//        * is accepted.
-//        * If the zone cannot be parsed then an exception is thrown unless the
-//        * section of the formatter is optional.
-        appendInternal(new ZoneTextPrinterParser(textStyle));
+        appendInternal(new ZoneTextPrinterParser(textStyle, null));
         return this;
     }
\ No newline at end of file
