diff --git a/old_product.java b/new_product.java
index c6aeb79..538070b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,20 +1,26 @@
 @PreAuthorize("hasAnyRole('A')")
 	@RequestMapping({ "", "/" })
 	public String getUserList(ModelMap model, @RequestParam(required = false) String roleName,
+					@PageableDefaults(pageNumber = 0, value = 10) Pageable pageable,
 					@RequestParam(required = false) String keywords) {
 
-		List<User> userList = null;
+		PageRequest pageReq = ((PageRequest) pageable);
+		Sort sort = pageReq == null ? null : pageReq.getSort();
+		if (sort == null && pageReq != null) {
+			sort = new Sort(Direction.ASC, "userName");
+			pageable = new PageRequest(pageReq.getPageNumber(), pageReq.getPageSize(), sort);
+		}
+		Page<User> pagedUser = null;
 		if (StringUtils.isEmpty(keywords)) {
-			userList = userService.getAllUserByRole(roleName);
+			pagedUser = userService.getAllUserByRole(roleName, pageable);
 		} else {
-			userList = userService.getUserListByKeyWord(keywords);
+			pagedUser = userService.getUserListByKeyWord(keywords, pageable);
 			model.put("keywords", keywords);
 		}
-
-		model.addAttribute("userList", userList);
+		model.addAttribute("userPage", pagedUser);
 		EnumSet<Role> roleSet = EnumSet.allOf(Role.class);
 		model.addAttribute("roleSet", roleSet);
 		model.addAttribute("roleName", roleName);
-
+		model.addAttribute("page", pageable);
 		return "user/list";
 	}
\ No newline at end of file
