diff --git a/old_test.java b/new_test.java
index 3c7ca1c..4c46999 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,14 @@
 @Test
 	public void testGetUserList() {
+		Pageable page = new PageRequest(1, 10);
+
 		ModelMap model = new ModelMap();
-		userController.getUserList(model, null, null);
+		userController.getUserList(model, null, page, null);
 
 		model.clear();
-		userController.getUserList(model, "ADMIN", null);
+		userController.getUserList(model, "ADMIN", page, null);
 
 		model.clear();
-		userController.getUserList(model, null, "user");
+		userController.getUserList(model, null, page, "user");
 
 	}
\ No newline at end of file
