diff --git a/old_test.java b/new_test.java
index 9b1bca6..78b7e11 100644
--- a/old_test.java
+++ b/new_test.java
@@ -88,6 +88,11 @@
                 // TODO Auto-generated method stub
                 return null;
             }
+
+            @Override
+            protected IApiRequestPathParser createRequestPathParser(IPluginRegistry pluginRegistry) {
+                return new DefaultRequestPathParser(null);
+            }
         };
 
         IEngine engine = factory.createEngine();
