diff --git a/old_product.java b/new_product.java
index 37fba48..208baf9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -25,6 +25,11 @@ private IEngine createEngine() {
                     }
                 };
             }
+
+            @Override
+            protected IApiRequestPathParser createRequestPathParser(IPluginRegistry pluginRegistry) {
+                return new DefaultRequestPathParser(null);
+            }
         };
         return factory.createEngine();
     }
\ No newline at end of file
