diff --git a/old_product.java b/new_product.java
index 97d8cff..8d32e93 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,12 @@
-private Item backup(final QueryContext ctx) throws QueryException {
-    checkCreate(ctx);
+private Item dropBackup(final QueryContext ctx) throws QueryException {
     final String name = string(checkStr(expr[0], ctx));
     if(!Databases.validName(name)) throw BXDB_NAME.get(info, name);
-    if(!ctx.context.globalopts.dbexists(name)) throw BXDB_WHICH.get(info, name);
 
-    ctx.updates.add(new DBBackup(name, info, ctx), ctx);
+    final StringList files = DropBackup.backups(name, ctx.context);
+    if(files.isEmpty()) throw BXDB_WHICHBACK.get(info, name);
+
+    for(final String file : files) {
+      ctx.updates.add(new BackupDrop(file, info, ctx), ctx);
+    }
     return null;
   }
\ No newline at end of file
