diff --git a/old_test.java b/new_test.java
index a3f3109..d5cd371 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
-  public void backup() throws BaseXException {
+  public void createBackup() throws BaseXException {
     query(COUNT.args(_DB_BACKUPS.args(NAME)), "0");
-    query(_DB_BACKUP.args(NAME));
+    query(_DB_CREATE_BACKUP.args(NAME));
     query(COUNT.args(_DB_BACKUPS.args(NAME)), "1");
 
     // try to backup non-existing database
-    error(_DB_BACKUP.args(NAME + 'x'), Err.BXDB_WHICH);
+    error(_DB_CREATE_BACKUP.args(NAME + 'x'), Err.BXDB_WHICH);
     // cleanup
     new DropBackup(NAME).execute(context);
     query(COUNT.args(_DB_BACKUPS.args(NAME)), "0");
