diff --git a/old_product.java b/new_product.java
index 10bf4f3..ef304ee 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,12 @@
-public static GeoPoint projectGeoPoint(final double aLat, final double aLon, final int aZoom,
-			final GeoPoint aUseAsReturnValue) {
-		final GeoPoint out = aUseAsReturnValue != null ? aUseAsReturnValue : new GeoPoint(0, 0);
+public static Point projectGeoPoint(final double aLat, final double aLon, final int aZoom,
+			final Point aUseAsReturnValue) {
+		final Point p = aUseAsReturnValue != null ? aUseAsReturnValue : new Point(0, 0);
 
-		out.setLongitudeE6((int) Math.floor((aLon + 180) / 360 * (1 << aZoom)));
-		out.setLatitudeE6((int) Math.floor((1 - Math.log(Math.tan(aLat * DEG2RAD) + 1
+		p.x = ((int) Math.floor((aLon + 180) / 360 * (1 << aZoom)));
+		p.y = ((int) Math.floor((1 - Math.log(Math.tan(aLat * DEG2RAD) + 1
 				/ Math.cos(aLat * DEG2RAD))
 				/ Math.PI)
 				/ 2 * (1 << aZoom)));
 
-		return out;
+		return p;
 	}
\ No newline at end of file
