diff --git a/old_test.java b/new_test.java
index a65ddd5..fab9a7f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,8 +4,8 @@
 		final int lonE6 = 9735936;
 		final int zoom = 8;
 
-		final GeoPoint point = Mercator.projectGeoPoint(latE6, lonE6, zoom, null);
+		final Point point = Mercator.projectGeoPoint(latE6, lonE6, zoom, null);
 
-		assertEquals("TODO describe test", 84, point.getLatitudeE6());
-		assertEquals("TODO describe test", 134, point.getLongitudeE6());
+		assertEquals("TODO describe test", 84, point.y);
+		assertEquals("TODO describe test", 134, point.x);
 	}
\ No newline at end of file
