diff --git a/old_product.java b/new_product.java
index 5d15e3e..bfec4ff 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1,3 @@
-public void register(String name, DoubleSupplier supplier, @Nullable String description);
\ No newline at end of file
+public void register(String name, DoubleSupplier supplier, @Nullable String desc) {
+        addMetric(name, new DoubleGauge(metricName(grpName, name), desc, nonThrowableSupplier(supplier, log)));
+    }
\ No newline at end of file
