diff --git a/old_test.java b/new_test.java
index 5658887..2554863 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void serialize() {
-    contains(SERIALIZE.args("<x/>"), "&lt;x/&gt;");
-    contains(SERIALIZE.args("<x/>", serialParams("")), "&lt;x/&gt;");
+    contains(SERIALIZE.args("<x/>"), "<x/>");
+    contains(SERIALIZE.args("<x/>", serialParams("")), "<x/>");
     contains(SERIALIZE.args("<x>a</x>", serialParams("<method value='text'/>")), "a");
   }
\ No newline at end of file
