diff --git a/old_product.java b/new_product.java
index 3be6437..a4c312e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
 public void serialize(final Value value) throws IOException {
-    final boolean s = sep;
-    for(final Item it : value) serialize(it, false, false);
-    sep = s;
+    more = false;
+    for(final Item it : value) serialize(it);
   }
\ No newline at end of file
