diff --git a/old_product.java b/new_product.java
index 733b194..18ca424 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
-    public <U> ManagedCompletableFuture<Void> thenAcceptBoth(CompletionStage<? extends U> other, BiConsumer<? super T, ? super U> action) {
+    public <U> CompletableFuture<Void> thenAcceptBoth(CompletionStage<? extends U> other, BiConsumer<? super T, ? super U> action) {
         // Reject ManagedTask so that we have the flexibility to decide later how to handle ManagedTaskListener and execution properties
         if (action instanceof ManagedTask)
             throw new IllegalArgumentException(ManagedTask.class.getName());
@@ -16,6 +16,6 @@
             CompletableFuture<Void> dependentStage = completableFuture.thenAcceptBoth(other, action);
             return new ManagedCompletableFuture<Void>(dependentStage, defaultExecutor, null);
         } else {
-            return (ManagedCompletableFuture<Void>) super.thenAcceptBoth(other, action);
+            return super.thenAcceptBoth(other, action);
         }
     }
\ No newline at end of file
