diff --git a/old_product.java b/new_product.java
index beb2108..613b798 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-@Override
-  public boolean delete() {
-    if(isDir()) for(final IO ch : children()) if(!ch.delete()) return false;
-    return file.delete();
+public boolean delete() {
+    boolean ok = true;
+    if(isDir()) for(final IOFile ch : children()) ok &= ch.delete();
+    return file.delete() && ok;
   }
\ No newline at end of file
