diff --git a/old_product.java b/new_product.java
index c474537..e7bcac2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@ public StringList backups(final String db) {
     if(file.exists()) {
       backups.add(db);
     } else {
-      final Pattern regex = regex(db, DateTime.PATTERN + '\\' + IO.ZIPSUFFIX);
+      final Pattern regex = regex(db, '-' + DATE + '\\' + IO.ZIPSUFFIX);
       for(final IOFile f : soptions.dbpath().children()) {
         final String n = f.name();
         if(regex.matcher(n).matches()) backups.add(n.substring(0, n.lastIndexOf('.')));
