diff --git a/old_test.java b/new_test.java
index bc7f05a..4d61db9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,20 +14,23 @@
 
         Modeler.TrainingParameters trainingParameters = new Modeler.TrainingParameters();
         
-        
-        //Model Configuration
 
-        MultinomialNaiveBayes.TrainingParameters modelTrainingParameters = new MultinomialNaiveBayes.TrainingParameters();
-        modelTrainingParameters.setMultiProbabilityWeighted(true);
-        trainingParameters.setModelerTrainingParameters(modelTrainingParameters);
+        //numerical scaling configuration
+        MinMaxScaler.TrainingParameters nsParams = new MinMaxScaler.TrainingParameters();
+        trainingParameters.setNumericalScalerTrainingParameters(nsParams);
 
-        //data transfomation configuration
-        DummyXMinMaxNormalizer.TrainingParameters dtParams = new DummyXMinMaxNormalizer.TrainingParameters();
-        trainingParameters.setDataTransformerTrainingParameters(dtParams);
+        //categorical encoding configuration
+        CornerConstraintsEncoder.TrainingParameters ceParams = new CornerConstraintsEncoder.TrainingParameters();
+        trainingParameters.setCategoricalEncoderTrainingParameters(ceParams);
         
         //feature selection configuration
         trainingParameters.setFeatureSelectorTrainingParameters(null);
 
+        //model Configuration
+        MultinomialNaiveBayes.TrainingParameters modelTrainingParameters = new MultinomialNaiveBayes.TrainingParameters();
+        modelTrainingParameters.setMultiProbabilityWeighted(true);
+        trainingParameters.setModelerTrainingParameters(modelTrainingParameters);
+
         Modeler instance = MLBuilder.create(trainingParameters, configuration);
         instance.fit(trainingData);
         instance.save(storageName);
@@ -45,7 +48,6 @@
 
         trainingData.close();
         instance.close();
-        //instance = null;
 
 
         instance = MLBuilder.load(Modeler.class, storageName, configuration);
