diff --git a/old_product.java b/new_product.java
index a06a241..028f586 100644
--- a/old_product.java
+++ b/new_product.java
@@ -19,15 +19,18 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
         }
 
         TextClassifier.TrainingParameters trainingParameters = new TextClassifier.TrainingParameters();
-        
-        //Classifier configuration
-        trainingParameters.setModelerTrainingParameters(modelerTrainingParameters);
-        
-        //data transfomation configuration
-        trainingParameters.setDataTransformerTrainingParameters(null);
+
+        //numerical scaling configuration
+        trainingParameters.setNumericalScalerTrainingParameters(null);
+
+        //categorical encoding configuration
+        trainingParameters.setCategoricalEncoderTrainingParameters(null);
         
         //feature selection configuration
         trainingParameters.setFeatureSelectorTrainingParameters(featureSelectorTrainingParameters);
+
+        //classifier configuration
+        trainingParameters.setModelerTrainingParameters(modelerTrainingParameters);
         
         //text extraction configuration
         NgramsExtractor.Parameters exParams = new NgramsExtractor.Parameters();
@@ -44,7 +47,6 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
         assertEquals(expectedF1score, vm.getMacroF1(), Constants.DOUBLE_ACCURACY_HIGH);
 
         instance.close();
-        //instance = null;
         
         
         
