diff --git a/old_product.java b/new_product.java
index 4273980..d305b96 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public static synchronized UnderFileSystemCluster get(String baseDir, Configuration configuration)
+public static synchronized UnderFileSystemCluster get(String baseDir)
       throws IOException {
     if (sUnderFSCluster == null) {
-      sUnderFSCluster = getUnderFilesystemCluster(baseDir, configuration);
+      sUnderFSCluster = getUnderFilesystemCluster(baseDir);
     }
 
     if (!sUnderFSCluster.isStarted()) {
