diff --git a/old_test.java b/new_test.java
index a258940..e50c59f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,8 @@
   public void getTest() throws IOException {
     PowerMockito.spy(UnderFileSystemCluster.class);
 
-    Mockito.when(UnderFileSystemCluster.getUnderFilesystemCluster(BASE_DIR,
-        ALLUXIO_CONF)).thenReturn(mUnderFileSystemCluster);
+    Mockito.when(UnderFileSystemCluster.getUnderFilesystemCluster(BASE_DIR))
+        .thenReturn(mUnderFileSystemCluster);
 
     Whitebox.setInternalState(UnderFileSystemCluster.class, "sUnderFSCluster",
         (UnderFileSystemCluster) null);
@@ -11,7 +11,7 @@
     Mockito.when(mUnderFileSystemCluster.isStarted()).thenReturn(false);
 
     // execute test
-    UnderFileSystemCluster.get(BASE_DIR, ALLUXIO_CONF);
+    UnderFileSystemCluster.get(BASE_DIR);
 
     UnderFileSystemCluster underFSCluster = Whitebox.getInternalState(UnderFileSystemCluster
         .class, "sUnderFSCluster");
