diff --git a/old_product.java b/new_product.java
index 9eaba78..1102ad6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,7 @@
     @ResponseStatus(HttpStatus.OK)
     public ApplicationResource getApplication(@PathVariable("id") final String id) throws GenieException {
         log.debug("Called to get Application for id {}", id);
-        return this.applicationResourceAssembler.toResource(this.applicationService.getApplication(id));
+        return this.applicationResourceAssembler.toResource(
+            DtoAdapters.toV3Application(this.applicationService.getApplication(id))
+        );
     }
\ No newline at end of file
