diff --git a/old_product.java b/new_product.java
index bdefacb..1ee98b3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-public List<AlertDefinitionEntity> findAll(long clusterId) {
+@RequiresSession
+  public List<AlertDefinitionEntity> findAll(long clusterId) {
     TypedQuery<AlertDefinitionEntity> query = entityManagerProvider.get().createNamedQuery(
         "AlertDefinitionEntity.findAllInCluster", AlertDefinitionEntity.class);
 
