diff --git a/old_test.java b/new_test.java
index 9a613ef..72d8265 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,7 +12,7 @@
             // initial client security context
             clientContext = new WindowsSecurityContextImpl();
             clientContext.setPrincipalName(WindowsAccountImpl.getCurrentUsername());
-            clientContext.setCredentialsHandle(clientCredentials.getHandle());
+            clientContext.setCredentialsHandle(clientCredentials);
             clientContext.setSecurityPackage(securityPackage);
             clientContext.initialize(null, null, targetName);
             // accept on the server
