diff --git a/old_product.java b/new_product.java
index 655689c..75f9b26 100644
--- a/old_product.java
+++ b/new_product.java
@@ -32,7 +32,7 @@
                     pfClientContextAttr, null);
 
             sc = new WindowsSecurityContextImpl();
-            sc.setCredentialsHandle(serverCredential.getHandle());
+            sc.setCredentialsHandle(serverCredential);
             sc.setSecurityPackage(securityPackage);
             sc.setSecurityContext(phNewServerContext);
 
