diff --git a/old_product.java b/new_product.java
index fefde03..36caa40 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,6 +2,7 @@
     public static XContentBuilder marshall(ServiceContract bean) throws Exception {
         XContentBuilder builder = XContentFactory.jsonBuilder().startObject();
         builder.field("apiKey", bean.getApikey());
+        builder.field("plan", bean.getPlan());
         builder.field("application");
         marshallInto(bean.getApplication(), builder);
         builder.field("service");
