diff --git a/old_test.java b/new_test.java
index 48a810d..c1a389c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,9 @@
     public void testMarshall_ServiceContract() throws Exception {
         ServiceContract sc = new ServiceContract();
         sc.setApikey("12345");
+        sc.setPlan("Gold");
         sc.setPolicies(new ArrayList<Policy>());
-        
+
         Service service = new Service();
         service.setServicePolicies(null);
         service.setEndpoint("http://host/path/to/svc");
@@ -32,6 +33,7 @@
 
         Assert.assertEquals("{"
                 + "\"apiKey\":\"12345\","
+                + "\"plan\":\"Gold\","
                 + "\"application\":{"
                     + "\"organizationId\":\"test-org\","
                     + "\"applicationId\":\"app-id\","
