diff --git a/old_test.java b/new_test.java
index f69247d..3865c26 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,11 +2,11 @@
 	public void validate_shouldFailValidationIfTheSpecimenSourceIsInvalid() throws Exception {
 		ConceptService conceptService = Context.getConceptService();
 		Concept specimenSource = conceptService.getConcept(3);
-		assertThat(specimenSource, not(isIn(Context.getOrderService().getDrugRoutes())));
+		OrderService orderService = Context.getOrderService();
+		assertThat(specimenSource, not(isIn(orderService.getDrugRoutes())));
 		TestOrder order = new TestOrder();
 		Patient patient = new Patient(8);
 		order.setPatient(patient);
-		OrderService orderService = Context.getOrderService();
 		order.setOrderType(orderService.getOrderTypeByName("Test order"));
 		order.setConcept(conceptService.getConcept(5497));
 		order.setOrderer(new Provider());
@@ -20,5 +20,6 @@
 		Errors errors = new BindException(order, "order");
 		new TestOrderValidator().validate(order, errors);
 		Assert.assertTrue(errors.hasFieldErrors("specimenSource"));
-		Assert.assertEquals("TestOrder.error.notAmongAllowedConcepts", errors.getFieldError("specimenSource").getCode());
+		Assert.assertEquals("TestOrder.error.specimenSourceNotAmongAllowedConcepts", errors.getFieldError("specimenSource")
+		        .getCode());
 	}
\ No newline at end of file
