diff --git a/old_product.java b/new_product.java
index 2c4ac77..4f9b78f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ public void validate(Object obj, Errors errors) {
 		
 		DrugOrder order = (DrugOrder) obj;
 		if (order == null) {
-			errors.rejectValue("order", "error.general");
+			errors.reject("error.general");
 		} else {
 			// for the following elements Order.hbm.xml says: not-null="true"
 			ValidationUtils.rejectIfEmpty(errors, "asNeeded", "error.null");
