diff --git a/old_test.java b/new_test.java
index 721ae54..825ac92 100644
--- a/old_test.java
+++ b/new_test.java
@@ -30,6 +30,17 @@
 
 		CharSequence json = behavior.renderAjaxAttributes(component, attributes);
 
-		String expected = "{\"coh\":[function(attrs, jqXHR, textStatus){alert('Complete!');}],\"u\":\"some/url\",\"pre\":[function(attrs){return somePrecondition();}],\"fh\":[function(attrs, jqXHR, errorMessage, textStatus){alert('Failure!');}],\"bh\":[function(attrs){alert('Before!');}],\"sh\":[function(attrs, jqXHR, data, textStatus){alert('Success!');}],\"ah\":[function(attrs){alert('After!');}]}";
+		String expected = "{\"" + AjaxAttributeName.COMPLETE_HANDLER.jsonName() +
+			"\":[function(attrs, jqXHR, textStatus){alert('Complete!');}],\"" +
+			AjaxAttributeName.URL.jsonName() + "\":\"some/url\",\"" +
+			AjaxAttributeName.PRECONDITION.jsonName() +
+			"\":[function(attrs){return somePrecondition();}],\"" +
+			AjaxAttributeName.FAILURE_HANDLER.jsonName() +
+			"\":[function(attrs, jqXHR, errorMessage, textStatus){alert('Failure!');}],\"" +
+			AjaxAttributeName.BEFORE_HANDLER.jsonName() +
+			"\":[function(attrs){alert('Before!');}],\"" +
+			AjaxAttributeName.SUCCESS_HANDLER.jsonName() +
+			"\":[function(attrs, jqXHR, data, textStatus){alert('Success!');}],\"" +
+			AjaxAttributeName.AFTER_HANDLER.jsonName() + "\":[function(attrs){alert('After!');}]}";
 		assertEquals(expected, json);
 	}
\ No newline at end of file
