diff --git a/old_product.java b/new_product.java
index b11f563..9637dca 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,6 @@ public void validate(Object obj, Errors errors) {
 			errors.rejectValue("location", "error.general");
 		} else {
 			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "name", "error.name");
-			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "description", "error.description");
 			
 			if (location.isRetired()) {
 				if (!StringUtils.hasLength(location.getRetireReason())) {
