diff --git a/old_test.java b/new_test.java
index 26bb0a8..244bf9a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,13 @@
-@Test(expected = APIException.class)
+@Test
 	public void validate_shouldRejectAProviderIfItHasFewerThanMinOccursOfAnAttribute() throws Exception {
 		executeDataSet(PROVIDER_ATTRIBUTE_TYPES_XML);
+		ProviderAttributeType attributeType = providerService.getProviderAttributeType(1);
+		attributeType.setMinOccurs(2);
+		attributeType.setMaxOccurs(3);
+		providerService.saveProviderAttributeType(attributeType);
+		
 		provider.addAttribute(makeAttribute("one"));
-		ValidateUtil.validate(provider);
+		Errors errors = new BindException(provider, "provider");
+		new ProviderValidator().validate(provider, errors);
+		Assert.assertTrue(errors.hasFieldErrors("activeAttributes"));
 	}
\ No newline at end of file
