diff --git a/old_product.java b/new_product.java
index 6ccfd5e..ceacc2b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,6 @@
 private void functionDecl(final boolean up, final Ann ann)
       throws QueryException {
 
-    skipWS();
-
     final QNm name = eQName(FUNCNAME, ctx.sc.nsFunc);
     if(keyword(name)) error(RESERVED, name);
     if(module != null && !eq(name.uri(), module.uri())) error(MODNS, name);
@@ -12,9 +10,7 @@ private void functionDecl(final boolean up, final Ann ann)
     final Var[] args = paramList();
     wsCheck(PAR2);
 
-    final UserFunc func = new UserFunc(input(), name, args, optAsType(), ann, true);
-    func.updating = up;
-
+    final UserFunc func = new UserFunc(input(), name, args, optAsType(), ann, true, up);
     ctx.funcs.add(func, input());
     if(!wsConsumeWs(EXTERNAL)) func.expr = enclosed(NOFUNBODY);
     ctx.vars.reset(vl);
