diff --git a/old_test.java b/new_test.java
index 32b426b..96b25a7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,9 @@
 @Test
   public void functionDecl() {
+    query("declare namespace a='a';declare %a:a function local:x() {1}; local:x()", "1");
     query("declare %public function local:x() { 1 }; local:x()", "1");
     query("declare %fn:public function local:x() { 1 }; local:x()", "1");
     query("declare %private function local:x() { 1 }; local:x()", "1");
     query("declare %fn:private function local:x() { 1 }; local:x()", "1");
-    //error("declare %unknown function local:x() { 1 }; local:x()", Err.WHICHANN);
-    //error("declare %err:public function local:x() { 1 }; local:x()", Err.WHICHANN);
-    //error("declare %pfff:public function local:x() { 1 }; local:x()", Err.NOURI);
+    query("declare namespace a='a';declare %a:a function local:x() {1}; local:x()", "1");
   }
\ No newline at end of file
