diff --git a/old_test.java b/new_test.java
index 74f0a60..970e73e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,5 +2,5 @@
   public void delete() {
     for(final byte[] t : LIST) set.delete(t);
     for(final byte[] t : LIST)
-      assertTrue("Token should not be contained in list.", set.id(t) == 0);
+      assertEquals("Token should not be contained in list.", 0, set.id(t));
   }
\ No newline at end of file
