diff --git a/old_product.java b/new_product.java
index 0913cf8..d754ce1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,5 +3,8 @@ public ServiceCall<Counterexample> getCounterexample(GetCounterexampleOptions ge
     RequestBuilder builder = RequestBuilder.get(String.format("/v1/workspaces/%s/counterexamples/%s",
         getCounterexampleOptions.workspaceId(), getCounterexampleOptions.text()));
     builder.query(VERSION, versionDate);
+    if (getCounterexampleOptions.includeAudit() != null) {
+      builder.query("include_audit", String.valueOf(getCounterexampleOptions.includeAudit()));
+    }
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(Counterexample.class));
   }
\ No newline at end of file
