diff --git a/old_test.java b/new_test.java
index 46b9147..7730066 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,13 +4,14 @@
     Date start = new Date();
 
     String counterExampleText = "Make me a " + UUID.randomUUID().toString() + " sandwich"; // gotta be unique
-    CreateCounterexampleOptions createOptions =
-            new CreateCounterexampleOptions.Builder(workspaceId, counterExampleText).build();
+    CreateCounterexampleOptions createOptions = new CreateCounterexampleOptions.Builder(workspaceId, counterExampleText)
+        .build();
     service.createCounterexample(createOptions).execute();
 
     try {
-      GetCounterexampleOptions getOptions =
-              new GetCounterexampleOptions.Builder(workspaceId, counterExampleText).build();
+      GetCounterexampleOptions getOptions = new GetCounterexampleOptions.Builder(workspaceId, counterExampleText)
+          .includeAudit(true)
+          .build();
       Counterexample response = service.getCounterexample(getOptions).execute();
       assertNotNull(response);
       assertNotNull(response.getText());
@@ -28,8 +29,8 @@
       fail(ex.getMessage());
     } finally {
       // Clean up
-      DeleteCounterexampleOptions deleteOptions =
-              new DeleteCounterexampleOptions.Builder(workspaceId, counterExampleText).build();
+      DeleteCounterexampleOptions deleteOptions = new DeleteCounterexampleOptions.Builder(workspaceId,
+          counterExampleText).build();
       service.deleteCounterexample(deleteOptions).execute();
     }
   }
\ No newline at end of file
