diff --git a/old_product.java b/new_product.java
index 2567384..7c6193e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -21,7 +21,7 @@
         keyRanges.put(0.5, 1.0);
 
         Boolean status = controller.scaleStream(new StreamImpl(SCOPE, SCALE_DOWN_STREAM_NAME),
-                Collections.singletonList(0),
+                Collections.singletonList(0L),
                 keyRanges,
                 EXECUTOR_SERVICE).getFuture().get();
         assertTrue(status);
