diff --git a/old_test.java b/new_test.java
index 4611f26..108dc6b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,6 +9,8 @@
     TraceContext context4 = context.toBuilder().spanId(4).build();
     map.getOrCreate(context4);
 
+    int initialClockVal = clock.get();
+
     // By clearing strong references in this test, we are left with the weak ones in the map
     context1 = context2 = null;
     blockOnGC();
@@ -30,4 +32,8 @@
     // We also expect the spans reported to have the endpoint of the tracer
     assertThat(spans).extracting(Span::localEndpoint)
         .containsExactly(endpoint, endpoint);
+
+    // we also expect the clock to have been called only once
+    assertThat(spans).flatExtracting(Span::annotations).extracting(Annotation::timestamp)
+        .allSatisfy(t -> assertThat(t).isEqualTo((initialClockVal + 1) * 1000));
   }
\ No newline at end of file
