diff --git a/old_product.java b/new_product.java
index 509df8e..3ba5ea0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,7 +14,8 @@
         }
 
         try {
-            authenticateAuthorize(scopeName + "/" + streamName, READ_UPDATE);
+            restAuthHelper.authenticateAuthorize(getAuthorizationHeader(),
+                    AuthResourceRepresentation.ofStreamsInScope(scopeName), READ_UPDATE);
         } catch (AuthException e) {
             log.warn("Create stream for {} failed due to authentication failure.", streamName);
             asyncResponse.resume(Response.status(Status.fromStatusCode(e.getResponseCode())).build());
