diff --git a/old_product.java b/new_product.java
index ec50ec8..84db9df 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 private Array tail(final QueryContext qc) throws QueryException {
-    final Array array = array(0, qc);
+    final Array array = toArray(exprs[0], qc);
     return Array.get(array, checkPos(array, 1) + 1, array.arraySize() - 1);
   }
\ No newline at end of file
