diff --git a/old_test.java b/new_test.java
index 2e9951c..7aa9e1c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,8 @@
 @Test
 	public void testExpire() throws Exception {
 
-		populateBoundKey();
-
 		assertEquals(keyOps.getClass().getName() + " -> " + keyOps.getKey(), Long.valueOf(-1), keyOps.getExpire());
+
 		if (keyOps.expire(10, TimeUnit.SECONDS)) {
 			long expire = keyOps.getExpire().longValue();
 			assertTrue(expire <= 10 && expire > 5);
