diff --git a/old_test.java b/new_test.java
index 3309d22..b0de95f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -26,6 +26,6 @@
         TransposeDataList expResult = new TransposeDataList();
         expResult.put("Class1", new FlatDataList(Arrays.asList(new Object[]{188.0,189.0})));
         expResult.put("Class2", new FlatDataList(Arrays.asList(new Object[]{190.0})));
-        TransposeDataList result = instance.extractColumnValuesByY(column);
+        TransposeDataList result = instance.extractXColumnValuesByY(column);
         assertEquals(expResult, result);
     }
\ No newline at end of file
