diff --git a/old_test.java b/new_test.java
index 820412e..bc009d5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,8 @@
 	public void testMaybeSignalForMoreConnections() throws SecurityException, NoSuchMethodException, IllegalArgumentException, IllegalAccessException, InvocationTargetException{
 		expect(mockPartition.isUnableToCreateMoreTransactions()).andReturn(false).once();
 		expect(mockPartition.getFreeConnections()).andReturn(mockConnectionHandles).anyTimes();
-		expect(mockConnectionHandles.size()).andReturn(1).anyTimes();
+//		expect(mockConnectionHandles.size()).andReturn(1).anyTimes();
+		expect(mockPartition.getAvailableConnections()).andReturn(new AtomicInteger(1)).anyTimes();
 		expect(mockPartition.getMaxConnections()).andReturn(10).anyTimes();
 		mockPartition.lockAlmostFullLock();
 		expectLastCall().once();
@@ -22,6 +23,8 @@
 		expect(mockPartition.getFreeConnections()).andReturn(mockConnectionHandles).anyTimes();
 		expect(mockConnectionHandles.size()).andReturn(1).anyTimes();
 		expect(mockPartition.getMaxConnections()).andReturn(10).anyTimes();
+		expect(mockPartition.getAvailableConnections()).andReturn(new AtomicInteger(1)).anyTimes();
+
 		mockPartition.lockAlmostFullLock();
 		expectLastCall().once();
 		mockPartition.almostFullSignal();
