diff --git a/old_product.java b/new_product.java
index adb5557..0dfd417 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 private void maybeSignalForMoreConnections(ConnectionPartition connectionPartition) {
 
-		if (!this.poolShuttingDown && !connectionPartition.isUnableToCreateMoreTransactions() && connectionPartition.getFreeConnections().size()*100/connectionPartition.getMaxConnections() < this.poolAvailabilityThreshold){
+		if (!this.poolShuttingDown && !connectionPartition.isUnableToCreateMoreTransactions() && 
+				connectionPartition.getAvailableConnections().get()*100/connectionPartition.getMaxConnections() <= this.poolAvailabilityThreshold){
 			try{
 				connectionPartition.lockAlmostFullLock();
 				connectionPartition.almostFullSignal();
