diff --git a/old_product.java b/new_product.java
index 514c72d..fd2faef 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,6 +2,8 @@ public CreateDirectoryTOptions toThrift() {
     CreateDirectoryTOptions options = new CreateDirectoryTOptions();
     options.setAllowExists(mAllowExists);
     options.setRecursive(mRecursive);
+    options.setTtl(mTtl);
+    options.setTtlAction(TtlAction.toThrift(mTtlAction));
     options.setPersisted(mWriteType.isThrough());
     if (mMode != null) {
       options.setMode(mMode.toShort());
