diff --git a/old_test.java b/new_test.java
index 8e91a5a..25344d4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -23,5 +23,7 @@
     Assert.assertEquals(blockSize, thriftOptions.getBlockSizeBytes());
     Assert.assertEquals(recursive, thriftOptions.isRecursive());
     Assert.assertEquals(writeType.isThrough(), thriftOptions.isPersisted());
+    Assert.assertEquals(ttl, thriftOptions.getTtl());
+    Assert.assertEquals(alluxio.thrift.TTtlAction.Free, thriftOptions.getTtlAction());
     Assert.assertEquals(mode.toShort(), thriftOptions.getMode());
   }
\ No newline at end of file
