diff --git a/old_test.java b/new_test.java
index b4561e5..bf720a0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,7 +11,9 @@
         "fb4a   =   //java/com/example:fbandroid",
         "katana =   //java/com/example:fbandroid"));
     BuckConfig config1 = createWithDefaultFilesystem(reader1, parser);
-    assertEquals(ImmutableMap.of("java/com/example", "fb4a"), config1.getBasePathToAliasMap());
+    assertEquals(
+        ImmutableMap.of(Paths.get("java/com/example"), "fb4a"),
+        config1.getBasePathToAliasMap());
     assertEquals(
         ImmutableMap.of(
             "fb4a", "//java/com/example:fbandroid",
@@ -23,7 +25,9 @@
         "katana =   //java/com/example:fbandroid",
         "fb4a   =   //java/com/example:fbandroid"));
     BuckConfig config2 = createWithDefaultFilesystem(reader2, parser);
-    assertEquals(ImmutableMap.of("java/com/example", "katana"), config2.getBasePathToAliasMap());
+    assertEquals(
+        ImmutableMap.of(Paths.get("java/com/example"), "katana"),
+        config2.getBasePathToAliasMap());
     assertEquals(
         ImmutableMap.of(
             "fb4a", "//java/com/example:fbandroid",
