diff --git a/old_test.java b/new_test.java
index 6719a9e..29ccfa4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,10 +5,10 @@
         distrPceStore.activate();
 
         // TunnelId with device label store information
-        distrPceStore.addTunnelInfo(tunnelId1, pceccTunnelInfo1);
+        distrPceStore.addTunnelInfo(tunnelId1, tunnelConsumerId1);
         assertThat(distrPceStore.existsTunnelInfo(tunnelId1), is(true));
-        assertThat(distrPceStore.getTunnelInfo(tunnelId1), is(pceccTunnelInfo1));
-        distrPceStore.addTunnelInfo(tunnelId2, pceccTunnelInfo2);
+        assertThat(distrPceStore.getTunnelInfo(tunnelId1), is(tunnelConsumerId1));
+        distrPceStore.addTunnelInfo(tunnelId2, tunnelConsumerId2);
         assertThat(distrPceStore.existsTunnelInfo(tunnelId2), is(true));
-        assertThat(distrPceStore.getTunnelInfo(tunnelId2), is(pceccTunnelInfo2));
+        assertThat(distrPceStore.getTunnelInfo(tunnelId2), is(tunnelConsumerId2));
     }
\ No newline at end of file
