diff --git a/old_product.java b/new_product.java
index a9af79c..2bed236 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 private void maybeSignalForMoreConnections(ConnectionPartition connectionPartition) {
 
 		if (!this.poolShuttingDown && !connectionPartition.isUnableToCreateMoreTransactions() && 
-				connectionPartition.getAvailableConnections().get()*100/connectionPartition.getMaxConnections() <= this.poolAvailabilityThreshold){
+				connectionPartition.getAvailableConnections()*100/connectionPartition.getMaxConnections() <= this.poolAvailabilityThreshold){
 			connectionPartition.getPoolWatchThreadSignalQueue().offer(new Object()); // item being pushed is not important.
 		}
 	}
\ No newline at end of file
