diff --git a/old_test.java b/new_test.java
index 3b89181..32d43e1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
 		expect(mockPartition.isUnableToCreateMoreTransactions()).andReturn(false).once();
 		expect(mockPartition.getFreeConnections()).andReturn(mockConnectionHandles).anyTimes();
 //		expect(mockConnectionHandles.size()).andReturn(1).anyTimes();
-		expect(mockPartition.getAvailableConnections()).andReturn(new AtomicInteger(1)).anyTimes();
+		expect(mockPartition.getAvailableConnections()).andReturn(1).anyTimes();
 		expect(mockPartition.getMaxConnections()).andReturn(10).anyTimes();
 		BlockingQueue<Object> bq = new ArrayBlockingQueue<Object>(1);
 		expect(mockPartition.getPoolWatchThreadSignalQueue()).andReturn(bq).anyTimes();
