diff --git a/old_test.java b/new_test.java
index d88fe18..c311fa1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -46,7 +46,7 @@
         tp.setLearningRate(1.0);
         tp.setTotalIterations(50);
         
-        AbstractClassifier.ValidationMetrics vm = smr.kFoldCrossValidation(reducedTrainingData, tp, 1);
+        AbstractClassifier.AbstractValidationMetrics vm = smr.kFoldCrossValidation(reducedTrainingData, tp, 1);
         
         double expResult = 0.6859007513066202;
         double result = vm.getMacroF1();
