diff --git a/old_test.java b/new_test.java
index 59883b3..ee97873 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
   public void invoke() {
     query(_XQUERY_INVOKE.args("src/test/resources/input.xq"), "XML");
-    error(_XQUERY_INVOKE.args("src/test/resources/xxx.xq"), Err.WHICHRES);
+    error(_XQUERY_INVOKE.args("src/test/resources/xxx.xq"), Err.WHICHRES_X);
   }
\ No newline at end of file
