diff --git a/old_product.java b/new_product.java
index 7b885d6..e0d1c85 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,6 +4,6 @@ private ValueBuilder invoke(final QueryContext qc) throws QueryException {
     try {
       return eval(qc, io.read(), io.path(), false);
     } catch(final IOException ex) {
-      throw IOERR.get(info, ex);
+      throw IOERR_X.get(info, ex);
     }
   }
\ No newline at end of file
