diff --git a/old_product.java b/new_product.java
index a1d25e0..596c13e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -19,7 +19,7 @@
 		
 		PatientIdentifier preferredIdentifier = null;
 		PatientIdentifier possiblePreferredId = patient.getPatientIdentifier();
-		if (possiblePreferredId != null && possiblePreferredId.isPreferred() && !possiblePreferredId.isVoided()) {
+		if (possiblePreferredId != null && possiblePreferredId.getPreferred() && !possiblePreferredId.isVoided()) {
 			preferredIdentifier = possiblePreferredId;
 		}
 		
@@ -37,7 +37,7 @@
 		
 		PersonName preferredName = null;
 		PersonName possiblePreferredName = patient.getPersonName();
-		if (possiblePreferredName != null && possiblePreferredName.isPreferred() && !possiblePreferredName.isVoided()) {
+		if (possiblePreferredName != null && possiblePreferredName.getPreferred() && !possiblePreferredName.isVoided()) {
 			preferredName = possiblePreferredName;
 		}
 		
@@ -55,7 +55,7 @@
 		
 		PersonAddress preferredAddress = null;
 		PersonAddress possiblePreferredAddress = patient.getPersonAddress();
-		if (possiblePreferredAddress != null && possiblePreferredAddress.isPreferred()
+		if (possiblePreferredAddress != null && possiblePreferredAddress.getPreferred()
 		        && !possiblePreferredAddress.isVoided()) {
 			preferredAddress = possiblePreferredAddress;
 		}
