diff --git a/old_test.java b/new_test.java
index 098e3c2..6593f1e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,8 @@
 @Test
-   public void testGetOutUserCount()
-   {
-      System.out.println("getOutUserCount");
-      SteppingThreadGroup instance = new SteppingThreadGroup();
-      int expResult = 1;
-      int result = instance.getOutUserCount();
-      assertEquals(expResult, result);
-   }
\ No newline at end of file
+    public void testGetOutUserCount() {
+        System.out.println("getOutUserCount");
+        SteppingThreadGroup instance = new SteppingThreadGroup();
+        String expResult = "";
+        String result = instance.getOutUserCount();
+        assertEquals(expResult, result);
+    }
\ No newline at end of file
