diff --git a/old_test.java b/new_test.java
index 4a62786..97af7a7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,4 @@
 @Test(expected = IllegalArgumentException.class)
-    @Ignore("As of 3.8 a warning is logged when client configuration specifies cache-local-entries option. "
-            + "This test is to be enabled again in 3.9, when this configuration will throw an exception.")
     public void checkNearCacheConfig_withUnsupportedClientConfig() {
         checkNearCacheConfig(MAP_NAME, getNearCacheConfig(BINARY), true);
     }
\ No newline at end of file
