diff --git a/old_product.java b/new_product.java
index 6cf7e69..d2a435f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,8 +4,8 @@ public static void checkNearCacheConfig(String mapName, NearCacheConfig nearCach
         checkEvictionConfig(nearCacheConfig.getEvictionConfig(), true);
 
         if (isClient && nearCacheConfig.isCacheLocalEntries()) {
-            LOGGER.warning("The Near Cache option `cache-local-entries` is not supported in client configurations. "
-                    + "Remove this option from your client configuration, future versions may fail startup with an exception.");
+            throw new IllegalArgumentException("The Near Cache option `cache-local-entries` is not supported in "
+                    + "client configurations.");
         }
         checkPreloaderConfig(nearCacheConfig, isClient);
     }
\ No newline at end of file
