diff --git a/old_test.java b/new_test.java
index 64c7de6..d58b1b3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,7 @@
 	@Verifies(value = "should return negative if other attribute is voided", method = "compareTo(PersonAttribute)")
 	public void compareTo_shouldReturnNegativeIfOtherAttributeIsVoided() throws Exception {
 		PersonAttribute pa = new PersonAttribute();
+		pa.setAttributeType(new PersonAttributeType(1));
 		PersonAttribute other = new PersonAttribute();
 		other.setVoided(true);
 		Assert.assertTrue(pa.compareTo(other) < 0);
