diff --git a/old_test.java b/new_test.java
index a10174b..530d965 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,14 @@
 @Test
   public void insertTest() {
     final int n = 1_000;
-    Array seq = Array.empty();
+    XQArray seq = XQArray.empty();
 
     for(int i = 0; i < n; i++) seq = seq.snoc(Int.get(i));
     assertEquals(n, seq.arraySize());
 
     final Int val = Int.get(n);
     for(int i = 0; i <= n; i++) {
-      final Array seq2 = seq.insertBefore(i, val, qc);
+      final XQArray seq2 = seq.insertBefore(i, val, qc);
       assertEquals(n, ((Int) seq2.get(i)).itr());
       assertEquals(n + 1L, seq2.arraySize());
       for(int j = 0; j < n; j++) {
