diff --git a/old_product.java b/new_product.java
index 7a99965..dcd932d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,6 @@
 private boolean insert(final byte[] string, final int offset, final int rem) {
-    final int ts = size(), al = string.length;
-    final byte[] tmp = new byte[offset + al + ts - rem];
-    System.arraycopy(text, 0, tmp, 0, offset);
-    System.arraycopy(string, 0, tmp, offset, al);
-    System.arraycopy(text, rem, tmp, offset + al, ts - rem);
-    return text(tmp);
+    final int ts = size();
+    final ByteList bl = new ByteList(offset + string.length + ts - rem);
+    bl.add(text, 0, offset).add(string).add(text, rem, ts);
+    return text(bl.finish());
   }
\ No newline at end of file
