diff --git a/old_test.java b/new_test.java
index bd0c290..ae9cef3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test(expected = IllegalStateException.class)
   public void handleGetTest()
-      throws IOException, JSONException, RestServiceException, URISyntaxException {
-    AdminBlobStorageService adminBlobStorageService = getAdminBlobStorageService();
-    MessageInfo messageInfo = createMessageInfo(RestMethod.GET, "/", new JSONObject());
-    adminBlobStorageService.handleGet(messageInfo);
+      throws JSONException, RestServiceException, URISyntaxException {
+    RestRequestInfo restRequestInfo = createRestRequestInfo(RestMethod.GET, "/", new JSONObject());
+    adminBlobStorageService.handleGet(restRequestInfo);
   }
\ No newline at end of file
