diff --git a/old_product.java b/new_product.java
index 16e0516..a8ce242 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public void setApplications(final Set<ApplicationEntity> applications) {
+public void setApplications(final List<ApplicationEntity> applications) {
         //Clear references to this command in existing applications
         for (final ApplicationEntity application : this.applications) {
             application.getCommands().remove(this);
