diff --git a/old_product.java b/new_product.java
index bab1550..2903096 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public Vector<Object> findProvider(String name, boolean includeVoided, Integer start, Integer length) {
+public Vector<Object> findProvider(String name, boolean includeRetired, Integer start, Integer length) {
 		Vector<Object> providerListItem = new Vector<Object>();
 		List<Provider> providerList = Context.getProviderService().getProviders(name, start, length, null);
 		
@@ -7,7 +7,7 @@ public Vector<Object> findProvider(String name, boolean includeVoided, Integer s
 			providerListItem.add(mss.getMessage("Provider.noMatchesFound", new Object[] { name }, Context.getLocale()));
 		} else {
 			for (Provider p : providerList) {
-				if (!p.isRetired() || (p.isRetired() && includeVoided == true))
+				if (!p.isRetired() || (p.isRetired() && includeRetired == true))
 					providerListItem.add(new ProviderListItem(p));
 			}
 		}
