diff --git a/old_product.java b/new_product.java
index 5dc823a..214ddfd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,24 @@
-private Item move(final File src, final QueryContext ctx)
-      throws QueryException {
+private Item transfer(final File src, final QueryContext ctx,
+      final boolean copy) throws QueryException {
+
+    File trg = new File(string(checkStr(expr[1], ctx))).getAbsoluteFile();
+    if(!src.exists()) PATHNOTEXISTS.thrw(input, src);
 
-    File trg = new File(string(checkStr(expr[1], ctx)));
-    // attach file name if target is a directory
     if(trg.isDirectory()) {
+      // target is a directory: attach file name
       trg = new File(trg, src.getName());
     } else if(!trg.isFile()) {
-      final File par = trg.getParentFile();
-      if(!par.exists() || !par.isDirectory()) PATHINVALID.thrw(input, trg);
+      // target does not exist: ensure that parent exists
+      if(!trg.getParentFile().isDirectory()) NOTDIR.thrw(input, trg);
+    } else if(src.isDirectory()) {
+      // if target is file, source cannot be a directory
+      PATHISDIR.thrw(input, src);
     }
 
-    if(!src.exists()) PATHNOTEXISTS.thrw(input, src);
-    if(!src.renameTo(trg)) CANNOTMOVE.thrw(input, src, trg);
+    // ignore operations on same source and target path
+    if(!src.equals(trg)) {
+      if(copy) copy(src, trg);
+      else if(!src.renameTo(trg)) CANNOTMOVE.thrw(input, src, trg);
+    }
     return null;
   }
\ No newline at end of file
