diff --git a/old_test.java b/new_test.java
index 214593a..5a9f190 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,9 +7,8 @@
     query(fun + "('" + PATH1 + "', '" + PATH2 + "')");
     query(fun + "('" + PATH2 + "', '" + PATH1 + "')");
     query(fun + "('" + PATH1 + "', '" + PATH1 + "')");
-    error(fun + "('" + PATH1 + "', '" + PATH4 + "')", Err.PATHINVALID);
+    error(fun + "('" + PATH1 + "', '" + PATH4 + "')", Err.NOTDIR);
     query("file:size('" + PATH1 + "')", "1");
     query("file:exists('" + PATH2 + "')", "false");
-
     query("file:delete('" + PATH1 + "')");
   }
\ No newline at end of file
