diff --git a/old_test.java b/new_test.java
index 7d01e19..0daa430 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(expected = InvalidCharactersPasswordException.class)
-	@Verifies(value = "should fail without upper case char password", method = "validatePassword(String,String,String)")
-	public void validatePassword_shouldFailWithoutUpperCaseCharPassword() throws Exception {
+	@Verifies(value = "should fail without upper and lower case password by default", method = "validatePassword(String,String,String)")
+	public void validatePassword_shouldFailWithoutUpperAndLowerCasePasswordByDefault() throws Exception {
 		OpenmrsUtil.validatePassword("admin", "test0nl1", "1-8");
 	}
\ No newline at end of file
