diff --git a/old_product.java b/new_product.java
index 1ab3a48..deae48d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -18,6 +18,11 @@ public boolean process( I frame ) {
 			return false;
 		}
 
+		// mark that the track is in the inlier set
+		for( AssociatedPair p : modelMatcher.getMatchSet() ) {
+			((AssociatedPairTrack)p).lastUsed = totalProcessed;
+		}
+
 		// refine the motion estimate
 		if( modelRefiner == null ||
 				!modelRefiner.fitModel(modelMatcher.getMatchSet(),modelMatcher.getModel(),keyToCurr))
@@ -28,5 +33,16 @@ public boolean process( I frame ) {
 		// Update the motion
 		worldToKey.concat(keyToCurr, worldToCurr);
 
+		// prune tracks which aren't being used
+		List<PointTrack> all = tracker.getAllTracks(null);
+		for( PointTrack t : all ) {
+			AssociatedPairTrack p = t.getCookie();
+
+			if( totalProcessed - p.lastUsed >= pruneThreshold ) {
+				System.out.println("Delta "+(totalProcessed-p.lastUsed));
+				tracker.dropTrack(t);
+			}
+		}
+
 		return true;
 	}
\ No newline at end of file
