diff --git a/old_product.java b/new_product.java
index d824686..dfc4a17 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,24 +2,15 @@ public void handle(ConceptName conceptName, User currentUser, Date currentDate,
 		
 		// put Integer conceptNameTagIds onto ConceptNameTags that are missing them
 		if (conceptName.getTags() != null) {
-			Collection<ConceptNameTag> replacementTags = new ArrayList<ConceptNameTag>();
-			
-			Iterator<ConceptNameTag> tagsIt = conceptName.getTags().iterator();
-			while (tagsIt.hasNext()) {
-				ConceptNameTag tag = tagsIt.next();
-				
+			for (ConceptNameTag tag : conceptName.getTags()) {
 				if (tag.getConceptNameTagId() == null) {
-					ConceptNameTag replacementTag = Context.getConceptService().getConceptNameTagByName(tag.getTag());
-					
-					if (replacementTag != null) {
-						tagsIt.remove();
-						replacementTags.add(replacementTag);
+					ConceptNameTag possibleReplacementTag = Context.getConceptService()
+					        .getConceptNameTagByName(tag.getTag());
+					if (possibleReplacementTag != null) {
+						conceptName.removeTag(tag);
+						conceptName.addTag(possibleReplacementTag);
 					}
 				}
 			}
-			
-			if (!replacementTags.isEmpty()) {
-				conceptName.getTags().addAll(replacementTags);
-			}
 		}
 	}
\ No newline at end of file
