diff --git a/old_test.java b/new_test.java
index 03fd930..41cacc3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void listStatusTest() throws Exception {
-    TachyonURI file = new TachyonURI("/file");
+    AlluxioURI file = new AlluxioURI("/file");
     List<URIStatus> infos = Lists.newArrayList(new URIStatus(new FileInfo()));
     Mockito.when(mFileSystemMasterClient.listStatus(file)).thenReturn(infos);
     ListStatusOptions listStatusOptions = ListStatusOptions.defaults();
