diff --git a/old_product.java b/new_product.java
index e2758a4..2ec7eeb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
-List<URIStatus> listStatus(TachyonURI path)
-      throws FileDoesNotExistException, IOException, TachyonException;
\ No newline at end of file
+List<URIStatus> listStatus(AlluxioURI path)
+      throws FileDoesNotExistException, IOException, AlluxioException;
\ No newline at end of file
