diff --git a/old_product.java b/new_product.java
index dac4df3..3dd7bc2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,7 @@
 @Override
     public void updateDependenciesForCommand(
-        @NotBlank(message = "No command id entered. Unable to update dependencies.")
-        final String id,
-        @NotNull(message = "No dependencies entered. Unable to update.")
-        final Set<String> dependencies
+        @NotBlank(message = "No command id entered. Unable to update dependencies.") final String id,
+        @NotNull(message = "No dependencies entered. Unable to update.") final Set<String> dependencies
     ) throws GenieException {
-        this.findCommand(id).setDependencies(dependencies);
+        this.findCommand(id).setDependencies(this.createAndGetFileEntities(dependencies));
     }
\ No newline at end of file
