diff --git a/old_test.java b/new_test.java
index fe1671d..ab239ec 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,8 +10,8 @@
     run("geo:boundary(" +
         "<gml:Point><gml:coordinates>2,3</gml:coordinates></gml:Point>)",
         "<gml:MultiGeometry xmlns:gml=\"http://www.opengis.net/gml\"/>");
-    error("geo:boundary(text {'a'})", FUNTYPE.qname());
-    error("geo:boundary(a)", NOCTX.qname());
+    error("geo:boundary(text {'a'})", EXPTYPE_X_X_X.qname());
+    error("geo:boundary(a)", NOCTX_X.qname());
     error("geo:boundary(<gml:geo/>)", GeoErrors.qname(1));
     error("geo:boundary(<gml:Point><gml:pos>1 2</gml:pos></gml:Point>)",
         GeoErrors.qname(2));
