diff --git a/old_product.java b/new_product.java
index 22e7e52..82542d7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,7 +11,7 @@ void add(final QueryContext ctx) throws QueryException  {
     
     if(hashes.containsKey(chash)) {
       final IntList ps = hashes.get(cand.hash);
-      for(int pp : ps.toArray()) {
+      for(final int pp : ps.toArray()) {
         if(cand.eq(partitions.get(pp))) {
           found = true;
           p = pp;
