diff --git a/old_product.java b/new_product.java
index 1b28d6c..251915c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,14 +1,14 @@
-public final ArrayOutput serialize() throws QueryException {
+public final ArrayOutput serialize() throws QueryIOException {
     final ArrayOutput ao = new ArrayOutput();
     try {
       final Serializer ser = Serializer.get(ao);
       final ValueIter vi = iter();
       for(Item it; (it = vi.next()) != null;) ser.serialize(it);
       ser.close();
-    } catch(final QueryIOException ex) {
-      throw ex.getCause(null);
+    } catch(QueryIOException ex) {
+      throw ex;
     } catch(final IOException ex) {
-      SERANY.thrw(null, ex);
+      SERANY.thrwIO(ex);
     }
     return ao;
   }
\ No newline at end of file
