diff --git a/old_test.java b/new_test.java
index 4ccf525..8d7a1a4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test
     public void testDelete() {
         System.out.println("delete");
-        int expResult = 0;
-        int result = instance.delete(MOCK_TABLE, MOCK_KEY1);
-        assertEquals(expResult, result);
+        Status result = instance.delete(MOCK_TABLE, MOCK_KEY1);
+        assertEquals(Status.OK, result);
     }
\ No newline at end of file
