diff --git a/old_product.java b/new_product.java
index 82646a0..091a164 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
-  public int delete(String table, String key) {
+  public Status delete(String table, String key) {
 
     try {
       Statement stmt;
@@ -12,13 +12,13 @@
         System.out.println(stmt.toString());
       }
 
-      ResultSet rs = session.execute(stmt);
+      session.execute(stmt);
 
-      return StatusCode.OK;
+      return Status.OK;
     } catch (Exception e) {
       e.printStackTrace();
       System.out.println("Error deleting key: " + key);
     }
 
-    return StatusCode.ERROR;
+    return Status.ERROR;
   }
\ No newline at end of file
