diff --git a/old_test.java b/new_test.java
index 4477d0d..7d40afc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,14 +2,9 @@
     final List<Integer> ints = Arrays.asList(1, 10, 2);
     final List<Integer> empty = Collections.emptyList();
     Assert.assertFalse(
-        Functions.exists(ints,
-            new Predicate1<Integer>() {
-              public boolean apply(Integer v1) {
-                return v1 > 20;
-              }
-            }));
+        Functions.exists(ints, v1 -> v1 > 20));
     Assert.assertFalse(
-        Functions.exists(empty, Functions.<Integer>falsePredicate1()));
+        Functions.exists(empty, Functions.falsePredicate1()));
     Assert.assertFalse(
-        Functions.exists(empty, Functions.<Integer>truePredicate1()));
+        Functions.exists(empty, Functions.truePredicate1()));
   }
\ No newline at end of file
