diff --git a/old_test.java b/new_test.java
index a9d5bde..8dccd5f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,6 @@
         requestJSONObject.put(User.USER_EMAIL, "test@gmail.com");
         requestJSONObject.put(User.USER_NAME, "Admin");
         requestJSONObject.put(User.USER_PASSWORD, "pass");
-        requestJSONObject.put(Keys.LOCALE, Preference.Default.DEFAULT_LANGUAGE);
 
         initService.init(requestJSONObject);
 
