diff --git a/old_product.java b/new_product.java
index 1e05969..2b88ed4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -52,7 +52,7 @@ public void init(final JSONObject requestJSONObject) throws ServiceException {
         final Transaction transaction = userRepository.beginTransaction();
 
         try {
-            helloWorld(requestJSONObject.getString(Keys.LOCALE));
+            helloWorld();
             transaction.commit();
         } catch (final Exception e) {
             if (transaction.isActive()) {
