diff --git a/old_product.java b/new_product.java
index 4985f48..4cc43cb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,7 @@
 public AlluxioURI join(String suffix) {
+    if (suffix.isEmpty()) {
+      return new AlluxioURI(getScheme(), getAuthority(), getPath());
+    }
     // TODO(gpang): there should be other usage of join() which can use joinUnsafe() instead.
     String path = getPath();
     StringBuilder sb = new StringBuilder(path.length() + 1 + suffix.length());
