diff --git a/old_test.java b/new_test.java
index 8c546dd..3b33185 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void addListener_shouldBeCalledByWaitForChangeThenRefresh() {
+    public void addListener_triggeredByRefresh() {
         final CountDownLatch latch = new CountDownLatch(1);
         Collection collection = new Collection(sharedRealm, table.where());
         collection.size();
