diff --git a/old_test.java b/new_test.java
index 536c6b6..361efe9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public void text() {
     // run function without and with index
     execute(new DropIndex(CmdIndex.TEXT));
-    error(_DB_TEXT.args(NAME, "XML"), BXDB_INDEX_X);
+    error(_DB_TEXT.args(NAME, "XML"), DB_NOINDEX_X_X);
     execute(new CreateIndex(CmdIndex.TEXT));
     query(_DB_TEXT.args(NAME, "XML"), "XML");
     query(_DB_TEXT.args(NAME, "XXX"), "");
