diff --git a/old_product.java b/new_product.java
index 224c1b9..99396fe 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,11 +8,12 @@
         IConnectorFactory cfactory = createConnectorFactory(pluginRegistry);
         IPolicyFactory pfactory = createPolicyFactory(pluginRegistry);
         IMetrics metrics = createMetrics(pluginRegistry);
-        
+        IDelegateFactory logFactory = createLoggerFactory(pluginRegistry);
+
         List<IGatewayInitializer> initializers = createInitializers(pluginRegistry);
         for (IGatewayInitializer initializer : initializers) {
             initializer.initialize();
         }
 
-        return new EngineImpl(registry, pluginRegistry, componentRegistry, cfactory, pfactory, metrics);
+        return new EngineImpl(registry, pluginRegistry, componentRegistry, cfactory, pfactory, metrics, logFactory);
     }
\ No newline at end of file
