diff --git a/old_test.java b/new_test.java
index fee7639..ef59091 100644
--- a/old_test.java
+++ b/new_test.java
@@ -82,6 +82,12 @@
                     }
                 };
             }
+
+            @Override
+            protected IDelegateFactory createLoggerFactory(IPluginRegistry pluginRegistry) {
+                // TODO Auto-generated method stub
+                return null;
+            }
         };
 
         IEngine engine = factory.createEngine();
