diff --git a/old_test.java b/new_test.java
index a565d38..9451766 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,6 +7,13 @@
         streamResponseActual = response.get().readEntity(StreamProperty.class);
         testExpectedVsActualObject(streamResponseExpected, streamResponseActual);
 
+        // Test to create a stream which doesn't exist and have Retention Policy INFINITE
+        when(mockControllerService.createStream(any(), anyLong())).thenReturn(createStreamStatus);
+        response = target(streamResourceURI).request().async().post(Entity.json(createStreamRequest4));
+        assertEquals("Create Stream Status", 201, response.get().getStatus());
+        streamResponseActual = response.get().readEntity(StreamProperty.class);
+        testExpectedVsActualObject(streamResponseExpected2, streamResponseActual);
+
         // Test to create a stream that already exists
         when(mockControllerService.createStream(any(), anyLong())).thenReturn(createStreamStatus2);
         response = target(streamResourceURI).request().async().post(Entity.json(createStreamRequest));
