diff --git a/old_product.java b/new_product.java
index 1e41e90..5a6bd09 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-@Task(name = "createStream", version = "1.0", resource = "{scope}/{stream}")
+@Task(name = "checkStreamExists", version = "1.0", resource = "{scope}/{stream}")
     public CompletableFuture<CreateStreamStatus> createStream(String scope, String stream, StreamConfiguration config, long createTimestamp) {
         return execute(
                 new Resource(scope, stream),
