diff --git a/old_product.java b/new_product.java
index a6001cc..c0ed902 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,7 +10,11 @@ public void validate(Object obj, Errors errors) throws APIException, DuplicateCo
 			errors.reject("Concept.name.atLeastOneRequired");
 			return;
 		}
-		
+		if (conceptToValidate.getDescriptions().size() == 0) {
+			errors.rejectValue("descriptions","Concept.description.atLeastOneRequired");
+			return;
+		}
+
 		boolean hasFullySpecifiedName = false;
 		for (Locale conceptNameLocale : conceptToValidate.getAllConceptNameLocales()) {
 			boolean fullySpecifiedNameForLocaleFound = false;
