diff --git a/old_test.java b/new_test.java
index 0e25bdb..997f852 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,7 @@
 	public void validate_shouldPassValidationIfFieldLengthsAreCorrect() throws Exception {
 		Concept concept = new Concept();
 		concept.addName(new ConceptName("CD4", Context.getLocale()));
+		concept.addDescription(new ConceptDescription("some description",null));
 		concept.setVersion("version");
 		concept.setRetireReason("retireReason");
 		
