diff --git a/old_product.java b/new_product.java
index 626c82c..a7169b3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,15 +1,15 @@
-public Concepts getDocumentRelatedConcepts(Document document,String corpusName, RequestedFields conceptFields, int level, int limit) {
-        ConceptInsightsId.validateGenarate(document, getAccountId(), corpusName);
-        Map<String, Object> queryParams = new HashMap<String, Object>();
-        if (level >= 0) {
-            queryParams.put(LEVEL, level);
-        }
-        if (limit >= 0) {
-            queryParams.put(LIMIT, limit);
-        }
-        if (conceptFields != null) {
-            if (conceptFields != null && conceptFields.getFields() != null && !conceptFields.getFields().isEmpty())
-                queryParams.put(CONCEPT_FIELDS, conceptFields.toString());
-        }
-        return executeRequest(version + document.getId() + RELATED_CONCEPTS_PATH, queryParams, Concepts.class);
-    }
\ No newline at end of file
+public Concepts getDocumentRelatedConcepts(final Document document, final Map<String, Object> parameters) {
+		String documentId = IDValidator.getDocumentId(document);
+		String[] queryParameters = new String[] { LEVEL, LIMIT };
+		Map<String, Object> queryParams = new HashMap<String, Object>();
+		for (String param : queryParameters) {
+			if (parameters.containsKey(param))
+				queryParams.put(param, parameters.get(param));
+		}
+		if (parameters.get(CONCEPT_FIELDS) != null) {
+			RequestedFields fields = (RequestedFields) parameters.get(CONCEPT_FIELDS);
+			if (fields != null && !fields.isEmpty())
+				queryParams.put(CONCEPT_FIELDS, fields.toString());
+		}
+		return executeRequest(API_VERSION + documentId + RELATED_CONCEPTS_PATH, queryParams, Concepts.class);
+	}
\ No newline at end of file
