diff --git a/old_test.java b/new_test.java
index 2b9b423..ba5df44 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,8 @@
 @Test
-    public void testGetDocumentRelatedConcepts() {
-        Map<String,Object> params = new HashMap<String, Object>();
-        params.put(ConceptInsights.ACCOUNT_ID, "public");
-        params.put(ConceptInsights.CORPUS, "ibmresearcher");
-        params.put(ConceptInsights.DOCUMENT, "il-AHARONA");
-        params.put(ConceptInsights.LEVEL, "1");
-        params.put(ConceptInsights.LIMIT, 10);
-        Concepts concepts =  service.getDocumentRelatedConcepts(params);
-        Assert.assertNotNull(concepts);
-    }
\ No newline at end of file
+	public void testGetDocumentRelatedConcepts() {
+		Map<String, Object> params = new HashMap<String, Object>();
+		params.put(ConceptInsights.LEVEL, "1");
+		params.put(ConceptInsights.LIMIT, 10);
+		Concepts concepts = service.getDocumentRelatedConcepts(EXAMPLE_DOCUMENT, params);
+		Assert.assertNotNull(concepts);
+	}
\ No newline at end of file
