diff --git a/old_product.java b/new_product.java
index ae88079..10d660d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,12 @@
-private static AsyncRequestResponseHandler getAsyncRequestResponseHandler(BlobStorageService blobStorageService,
-      int requestWorkers, int responseWorkers)
+private static AsyncRequestResponseHandler getAsyncRequestResponseHandler(int requestWorkers, int responseWorkers)
       throws IOException {
     RestServerMetrics serverMetrics = new RestServerMetrics(new MetricRegistry());
     AsyncRequestResponseHandler handler = new AsyncRequestResponseHandler(serverMetrics);
+    if(blobStorageService == null) {
+      blobStorageService = new MockBlobStorageService(verifiableProperties, handler, router);
+    }
     handler.setRequestWorkersCount(requestWorkers);
     handler.setResponseWorkersCount(responseWorkers);
-    if (blobStorageService != null) {
-      handler.setBlobStorageService(blobStorageService);
-    }
+    handler.setBlobStorageService(blobStorageService);
     return handler;
   }
\ No newline at end of file
