diff --git a/old_test.java b/new_test.java
index 78c0033..9cfadb8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,6 @@
     Properties properties = new Properties();
     VerifiableProperties verifiableProperties = new VerifiableProperties(properties);
     Router router = new InMemoryRouter(verifiableProperties);
-    BlobStorageService blobStorageService = new MockBlobStorageService(verifiableProperties, router);
 
     // Get response handler.
     AsyncRequestResponseHandlerFactory responseHandlerFactory =
@@ -14,6 +13,8 @@
     assertEquals("Did not receive an AsyncRequestResponseHandler instance",
         AsyncRequestResponseHandler.class.getCanonicalName(), restResponseHandler.getClass().getCanonicalName());
 
+    BlobStorageService blobStorageService =
+        new MockBlobStorageService(verifiableProperties, restResponseHandler, router);
     // Get request handler.
     AsyncRequestResponseHandlerFactory requestHandlerFactory =
         new AsyncRequestResponseHandlerFactory(1, restServerMetrics, blobStorageService);
