diff --git a/old_test.java b/new_test.java
index fde897d..a9e1401 100644
--- a/old_test.java
+++ b/new_test.java
@@ -38,6 +38,6 @@
 				.build());
 
 		final List<HUTraceEvent> result2 = huTraceRepository.query(query);
-		assertThat(result2.size(), is(1)); // still just one..
-		assertThat(result.get(0), is(result2.get(0)));
+		assertThat(result2).hasSize(1); // still just one!
+		assertThat(result.get(0)).isEqualTo(result2.get(0));
 	}
\ No newline at end of file
