diff --git a/old_test.java b/new_test.java
index f6166da..32c20da 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
-    public void testAddv()
+    public void testAddVarArgs()
     {
-        set.addv(newArray(set.keys, 0, 1, 2, 1, 0));
+        set.add(newArray(set.keys, 0, 1, 2, 1, 0));
         assertEquals(3, set.size());
         assertSortedListEquals(set.toArray(), 0, 1, 2);
     }
\ No newline at end of file
