diff --git a/old_product.java b/new_product.java
index 5a438d5..32841ec 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,7 +7,7 @@ public JComponent learn() {
         IsotonicMDS isomds = IsotonicMDS.of(data, 2);
         System.out.format("Learn Kruskal's Nonmetric MDS (k=2) from %d samples in %dms\n", data.length, System.currentTimeMillis()-clock);
 
-        PlotCanvas plot = ScatterPlot.plot(isomds.getCoordinates(), labels);
+        PlotCanvas plot = ScatterPlot.plot(isomds.coordinates, labels);
         plot.setTitle("Kruskal's Nonmetric MDS (k = 2)");
         pane.add(plot);
 
@@ -15,7 +15,7 @@ public JComponent learn() {
         isomds = IsotonicMDS.of(data, 3);
         System.out.format("Learn Kruskal's Nonmetric MDS (k=3) from %d samples in %dms\n", data.length, System.currentTimeMillis()-clock);
 
-        plot = ScatterPlot.plot(isomds.getCoordinates(), labels);
+        plot = ScatterPlot.plot(isomds.coordinates, labels);
         plot.setTitle("Kruskal's Nonmetric MDS (k = 3)");
         pane.add(plot);
 
