diff --git a/old_test.java b/new_test.java
index b734d66..8214b70 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,4 +9,17 @@
         expResult = true;
         result = instance.equals(obj);
         assertEquals(expResult, result);
+        
+        instance = new DependencyVersion("2.0.0");
+        obj = new DependencyVersion("2");
+        expResult = false;
+        result = instance.equals(obj);
+        assertEquals(expResult, result);
+        
+        obj = new DependencyVersion("2.0");
+        expResult = true;
+        result = instance.equals(obj);
+        assertEquals(expResult, result);
+        
+        
     }
\ No newline at end of file
