diff --git a/old_test.java b/new_test.java
index d7ec927..76116e6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,9 +2,9 @@
   public void fileExists() throws IOException {
     String testFile = PathUtils.concatPath(mUnderfsAddress, "testFile");
     String testDir = PathUtils.concatPath(mUnderfsAddress, "testDir");
-    Assert.assertFalse(mUfs.fileExists(testFile));
+    Assert.assertFalse(mUfs.isFile(testFile));
     createEmptyFile(testFile);
     mUfs.mkdirs(testDir, false);
-    Assert.assertTrue(mUfs.fileExists(testFile));
-    Assert.assertFalse(mUfs.fileExists(testDir));
+    Assert.assertTrue(mUfs.isFile(testFile));
+    Assert.assertFalse(mUfs.isFile(testDir));
   }
\ No newline at end of file
