diff --git a/old_test.java b/new_test.java
index a3f380c..54e4e20 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test public void testTrim() {
-    assertEquals("", trim(""));
-    assertEquals("", trim("    "));
-    assertEquals("x", trim("   x  "));
-    assertEquals("x", trim("   x "));
-    assertEquals("x y", trim("   x y "));
-    assertEquals("x", trim("   x"));
-    assertEquals("x", trim("x"));
+    assertEquals("", trimSpacesBoth(""));
+    assertEquals("", trimSpacesBoth("    "));
+    assertEquals("x", trimSpacesBoth("   x  "));
+    assertEquals("x", trimSpacesBoth("   x "));
+    assertEquals("x y", trimSpacesBoth("   x y "));
+    assertEquals("x", trimSpacesBoth("   x"));
+    assertEquals("x", trimSpacesBoth("x"));
   }
\ No newline at end of file
