diff --git a/old_test.java b/new_test.java
index 490c6c7..d411198 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,8 @@
 
     // Need to use updated options to set the correct last mod time.
     CreateDirectoryOptions dirOptions =
-        CreateDirectoryOptions.defaults().setPermission(TEST_PERMISSION).setRecursive(true);
+        CreateDirectoryOptions.defaults().setOwner(TEST_OWNER).setGroup(TEST_GROUP)
+            .setMode(TEST_MODE).setRecursive(true);
 
     // create nested directory
     InodeTree.CreatePathResult createResult = createPath(mTree, NESTED_URI, dirOptions);
