diff --git a/old_product.java b/new_product.java
index 6c5871f..ae697c6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -79,7 +79,9 @@ public CreatePathResult createPath(LockedInodePath inodePath, CreatePathOptions<
     CreateDirectoryOptions missingDirOptions = CreateDirectoryOptions.defaults()
         .setMountPoint(false)
         .setPersisted(options.isPersisted())
-        .setPermission(options.getPermission())
+        .setOwner(options.getOwner())
+        .setGroup(options.getGroup())
+        .setMode(options.getMode())
         .setDefaultMode(true);
     for (int k = pathIndex; k < (pathComponents.length - 1); k++) {
       InodeDirectory dir =
@@ -158,9 +160,9 @@ public CreatePathResult createPath(LockedInodePath inodePath, CreatePathOptions<
       MountTable.Resolution resolution = mMountTable.resolve(getPath(inode));
       String ufsUri = resolution.getUri().toString();
       UnderFileSystem ufs = resolution.getUfs();
-      Permission permission = new Permission(inode.getOwner(), inode.getGroup(), inode.getMode());
-      MkdirsOptions mkdirsOptions = MkdirsOptions.defaults().setCreateParent(false)
-          .setPermission(permission);
+      MkdirsOptions mkdirsOptions =
+          MkdirsOptions.defaults().setCreateParent(false).setOwner(inode.getOwner())
+              .setGroup(inode.getGroup()).setMode(new Mode(inode.getMode()));
       if (ufs.isDirectory(ufsUri) || ufs.mkdirs(ufsUri, mkdirsOptions)) {
         inode.setPersistenceState(PersistenceState.PERSISTED);
       }
