diff --git a/old_product.java b/new_product.java
index 658baa4..bddbc92 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public char[] toCharArray() {
         if (size == 0) {
             return ArrayUtils.EMPTY_CHAR_ARRAY;
         }
-        char chars[] = new char[size];
+        final char chars[] = new char[size];
         System.arraycopy(buffer, 0, chars, 0, size);
         return chars;
     }
\ No newline at end of file
