diff --git a/old_product.java b/new_product.java
index 89786c0..92bdbff 100644
--- a/old_product.java
+++ b/new_product.java
@@ -18,9 +18,31 @@ protected boolean determineIdentifiers(Dependency dependency, String vendor, Str
                 if (evVer == null) {
                     continue;
                 }
+
+                int maxDepth = 0;
+                for (VulnerableSoftware vs : cpes) {
+                    final DependencyVersion dbVer = DependencyVersionUtil.parseVersion(vs.getVersion());
+                    if (dbVer != null) {
+                        final int count = dbVer.getVersionParts().size();
+                        if (count > maxDepth) {
+                            maxDepth = count;
+                        }
+                    }
+                }
+
+                DependencyVersion evBaseVer = null;
+                //Only semantic versions used in NVD and evidence may contain an update version
+                if (maxDepth == 3 && evVer.getVersionParts().size() == 4) {
+                    String update = evVer.getVersionParts().get(3);
+                    if (update.matches("^(v|beta|alpha|u|rc|m|20\\d\\d).*$")) {
+                        evBaseVer = new DependencyVersion();
+                        evBaseVer.setVersionParts(evVer.getVersionParts().subList(0, 3));
+                    }
+                }
+
                 for (VulnerableSoftware vs : cpes) {
                     final DependencyVersion dbVer;
-                    if (vs.getUpdate() != null && !vs.getUpdate().isEmpty()) {
+                    if (vs.getUpdate() != null && !vs.getUpdate().isEmpty() && !vs.getUpdate().startsWith("~")) {
                         dbVer = DependencyVersionUtil.parseVersion(vs.getVersion() + '.' + vs.getUpdate());
                     } else {
                         dbVer = DependencyVersionUtil.parseVersion(vs.getVersion());
@@ -34,6 +56,10 @@ protected boolean determineIdentifiers(Dependency dependency, String vendor, Str
                         final String url = String.format(NVD_SEARCH_URL, URLEncoder.encode(vs.getName(), StandardCharsets.UTF_8.name()));
                         final IdentifierMatch match = new IdentifierMatch("cpe", vs.getName(), url, IdentifierConfidence.EXACT_MATCH, conf);
                         collected.add(match);
+                    } else if (evBaseVer != null && evBaseVer.equals(dbVer)
+                            && (bestGuessConf == null || bestGuessConf.compareTo(conf) > 0)) {
+                        bestGuessConf = conf;
+                        bestGuess = dbVer;
 
                         //TODO the following isn't quite right is it? need to think about this guessing game a bit more.
                     } else if (evVer.getVersionParts().size() <= dbVer.getVersionParts().size()
