diff --git a/old_test.java b/new_test.java
index 31730fc..031f1cd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test
-	public void testGetConceptRelatedConcepts() {
-		Map<String, Object> params = new HashMap<String, Object>();
-		params.put(ConceptInsights.LIMIT, 10);
-		params.put(ConceptInsights.LEVEL, 1);
-		RequestedFields fs = new RequestedFields();
-		fs.include("abstract");
-		params.put("concept_fields", fs);
-		Concepts concepts = service.getConceptRelatedConcepts(EXAMPLE_CONCEPT, params);
-		Assert.assertNotNull(concepts);
-		
-	}
\ No newline at end of file
+  public void testGetConceptRelatedConcepts() {
+    final Map<String, Object> params = new HashMap<String, Object>();
+    params.put(ConceptInsights.LIMIT, 10);
+    params.put(ConceptInsights.LEVEL, 1);
+    final RequestedFields fs = new RequestedFields();
+    fs.include("abstract");
+    params.put("concept_fields", fs);
+    final Concepts concepts = service.getConceptRelatedConcepts(EXAMPLE_CONCEPT, params);
+    Assert.assertNotNull(concepts);
+
+  }
\ No newline at end of file
