diff --git a/old_product.java b/new_product.java
index 08ecaf5..4434f4c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static Decompressor getCompressorInstance(@NonNull Context context, @NonNull File file, @Nullable String password) {
+public static Decompressor getCompressorInstance(@NonNull Context context, @NonNull File file) {
         Decompressor decompressor;
         String type = getExtension(file.getPath());
 
@@ -17,7 +17,7 @@ public static Decompressor getCompressorInstance(@NonNull Context context, @NonN
         } else if(isLzippedTar(type)) {
             decompressor = new LzmaDecompressor(context);
         } else if(is7zip(type)) {
-            decompressor = new SevenZipDecompressor(context, password);
+            decompressor = new SevenZipDecompressor(context);
         } else {
             return null;
         }
