diff --git a/old_product.java b/new_product.java
index 45debc2..f1f424d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,15 +10,7 @@
                 // Even though we have the value in the metadata, we need to be very careful not to invoke this callback
                 // before any other existing callbacks are invoked. As such, verify if we have an existing PendingRequest
                 // for this segment - if so, tag onto it so we invoke these callbacks in the correct order.
-                QueuedCallback<T> queuedCallback = null;
-                synchronized (this.assignmentLock) {
-                    PendingRequest pendingRequest = this.pendingRequests.getOrDefault(streamSegmentName, null);
-                    if (pendingRequest != null) {
-                        queuedCallback = new QueuedCallback<>(thenCompose);
-                        pendingRequest.callbacks.add(queuedCallback);
-                    }
-                }
-
+                QueuedCallback<T> queuedCallback = checkConcurrentAssignment(streamSegmentName, thenCompose);
                 return queuedCallback == null ? thenCompose.apply(streamSegmentId) : queuedCallback.result;
             }
         }
