diff --git a/old_product.java b/new_product.java
index 373982c..365c349 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 ChronoZonedDateTime<C> minusNanos(long nanos) {
-        ChronoDateTime newDT = dateTime.toDateTime().minusNanos(nanos);
-        return (newDT == dateTime.toDateTime() ? this :
+        ChronoDateTime newDT = dateTime.getDateTime().minusNanos(nanos);
+        return (newDT == dateTime.getDateTime() ? this :
             resolve(newDT, zone, dateTime, ZoneResolvers.retainOffset()));
     }
\ No newline at end of file
