diff --git a/old_test.java b/new_test.java
index 494db8b..3b56f2d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,7 @@
         Assert.assertEquals(APP_1_USER, app.getUser());
         Assert.assertEquals(APP_1_VERSION, app.getVersion());
         Assert.assertEquals(APP_1_STATUS, app.getStatus());
+        Assert.assertThat(app.getType(), Matchers.nullValue());
         Assert.assertEquals(3, app.getTags().size());
         Assert.assertEquals(2, app.getConfigs().size());
         Assert.assertEquals(2, app.getDependencies().size());
@@ -16,6 +17,7 @@
         Assert.assertEquals(APP_2_USER, app2.getUser());
         Assert.assertEquals(APP_2_VERSION, app2.getVersion());
         Assert.assertEquals(APP_2_STATUS, app2.getStatus());
+        Assert.assertThat(app2.getType(), Matchers.is(APP_2_TYPE));
         Assert.assertEquals(4, app2.getTags().size());
         Assert.assertEquals(2, app2.getConfigs().size());
         Assert.assertEquals(1, app2.getDependencies().size());
@@ -26,6 +28,7 @@
         Assert.assertEquals(APP_3_USER, app3.getUser());
         Assert.assertEquals(APP_3_VERSION, app3.getVersion());
         Assert.assertEquals(APP_3_STATUS, app3.getStatus());
+        Assert.assertThat(app3.getType(), Matchers.is(APP_3_TYPE));
         Assert.assertEquals(3, app3.getTags().size());
         Assert.assertEquals(1, app3.getConfigs().size());
         Assert.assertEquals(2, app3.getDependencies().size());
