diff --git a/old_product.java b/new_product.java
index 17eddcd..5aaf859 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,11 @@
 @Override
     public final IEngine createEngine() {
-        IRegistry registry = createRegistry();
         IPluginRegistry pluginRegistry = createPluginRegistry();
-        IComponentRegistry componentRegistry = createComponentRegistry();
-        IConnectorFactory cfactory = createConnectorFactory();
-        IPolicyFactory pfactory = createPolicyFactory();
-        IMetrics metrics = createMetrics();
+        IRegistry registry = createRegistry(pluginRegistry);
+        IComponentRegistry componentRegistry = createComponentRegistry(pluginRegistry);
+        IConnectorFactory cfactory = createConnectorFactory(pluginRegistry);
+        IPolicyFactory pfactory = createPolicyFactory(pluginRegistry);
+        IMetrics metrics = createMetrics(pluginRegistry);
 
         IEngine engine = new EngineImpl(registry, pluginRegistry, componentRegistry, cfactory, pfactory, metrics);
         return engine;
