diff --git a/old_test.java b/new_test.java
index 07b11e6..eb18414 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testCreateEngine() throws InterruptedException, ExecutionException {
         DefaultEngineFactory factory = new DefaultEngineFactory() {
             @Override
-            protected IConnectorFactory createConnectorFactory() {
+            protected IConnectorFactory createConnectorFactory(IPluginRegistry pluginRegistry) {
                 return new IConnectorFactory() {
                     @Override
                     public IServiceConnector createConnector(ServiceRequest request, Service service, RequiredAuthType requiredAuthType) {
