diff --git a/old_product.java b/new_product.java
index e8b9e7f..af31485 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @RequestMapping("/save")
 	@PreAuthorize("hasAnyRole('A', 'S') or #user.id == #updatedUser.id")
 	public String saveOrUpdateUserDetail(User user, ModelMap model, @ModelAttribute("user") User updatedUser,
-			@RequestParam(required = false) String followers) {
+			@RequestParam(required = false) String followersStr) {
 		checkArgument(updatedUser.validate());
 		if (user.getRole() == Role.USER) {
 			// General user can not change their role.
@@ -14,7 +14,7 @@
 							updatedUser);
 		}
 		if (updatedUser.exist()) {
-			userService.modifyUser(updatedUser);
+			userService.modifyUser(updatedUser,followersStr);
 		} else {
 			userService.saveUser(updatedUser);
 		}
