diff --git a/old_test.java b/new_test.java
index 71a07ec..d2ee72d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
 		ModelMap model = new ModelMap();
 		User currUser = getTestUser();
 		currUser.setUserName("new name");
-		userController.saveOrUpdateUserDetail(currUser, model, currUser);
+		userController.saveOrUpdateUserDetail(currUser, model, currUser,null);
 		
 		userController.getUserDetail(getTestUser(), model, currUser.getUserId());
 		User user = (User)model.get("user");
