diff --git a/old_product.java b/new_product.java
index 3a8f0be..0b9b4c8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,7 +14,7 @@ public JobResponse createJob(final String name, final String batchId,
         if (config != null )
             contentJson.add("config", config);
 
-        HttpRequestBase request = Request.Post(DocumentConversion.JOBS_PATH)
+        HttpRequestBase request = Request.Post(ConfigConstants.JOBS_PATH)
                                          .withContent(contentJson).build();
 
         try {
