diff --git a/old_test.java b/new_test.java
index 6a8b84e..92f4f1c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,8 @@
 @Test(enabled = true, description = "Get account by http")
   public void getAccount() {
-    response = HttpMethed.getAccount(httpnode,fromAddress);
-    try {
-      entity = response.getEntity();
-    } catch (Exception e) {
-      e.printStackTrace();
-    }
+    response = HttpMethed.getAccount(httpnode, fromAddress);
     logger.info("code is " + response.getStatusLine().getStatusCode());
-    Assert.assertEquals(response.getStatusLine().getStatusCode(),200);
+    Assert.assertEquals(response.getStatusLine().getStatusCode(), 200);
     responseContent = HttpMethed.parseResponseContent(response);
     HttpMethed.printJsonContent(responseContent);
     Assert.assertTrue(responseContent.size() > 3);
