diff --git a/old_test.java b/new_test.java
index 00ac72a..1942fdc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,17 +1,6 @@
 @Test
     public void testLocalPeek() throws Exception {
-        if (cacheMode() == LOCAL) {
-            checkAffinityLocalCache();
+        checkAffinityPeek(0);
 
-            checkStorage(0);
-        }
-        else {
-            checkAffinityPeek(0);
-
-            checkAffinityPeek(1);
-
-            checkStorage(0);
-
-            checkStorage(1);
-        }
+        checkAffinityPeek(1);
     }
\ No newline at end of file
