diff --git a/old_test.java b/new_test.java
index 92a4b3c..9296551 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,10 +5,10 @@
         MetricResponse metricRes = getMetricResponse();
         srvc.onNodeMetrics(UUID.randomUUID(), metricRes, SYSTEM_POOL);
 
-        ArgumentCaptor<StompHeaders> destCaptor = ArgumentCaptor.forClass(StompHeaders.class);
-        ArgumentCaptor<Object> payloadCaptor = ArgumentCaptor.forClass(Object.class);
+        ArgumentCaptor<String> destCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<byte[]> payloadCaptor = ArgumentCaptor.forClass(byte[].class);
         verify(mgr, times(1)).send(destCaptor.capture(), payloadCaptor.capture());
 
-        Assert.assertEquals(StompDestinationsUtils.buildMetricsDest(metricRes.clusterId()), destCaptor.getValue().getDestination());
-        Assert.assertArrayEquals(new byte[]{1, 2, 3, 4}, (byte[]) payloadCaptor.getValue());
+        Assert.assertEquals(StompDestinationsUtils.buildMetricsDest(metricRes.clusterId()), destCaptor.getValue());
+        Assert.assertArrayEquals(new byte[]{1, 2, 3, 4}, payloadCaptor.getValue());
     }
\ No newline at end of file
