diff --git a/old_product.java b/new_product.java
index 14f9d3b..2fee627 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ public void setAttribute(A attribute) {
 		
 		if (getActiveAttributes(attribute.getAttributeType()).size() == 1) {
 			A existing = getActiveAttributes(attribute.getAttributeType()).get(0);
-			if (existing.getValueReference().equals(attribute.getValueReference())) {
+			if (existing.getValue().equals(attribute.getValue())) {
 				// do nothing, since the value is already as-specified
 			} else {
 				if (existing.getId() != null)
