diff --git a/old_test.java b/new_test.java
index a5f22c1..292306e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,13 +4,16 @@
 		Provider provider = new Provider();
 		provider.setIdentifier("test");
 		provider.setName("test provider");
-		ProviderAttributeType providerAttributeType = service.getProviderAttributeType(3);
-		provider.setAttribute(createProviderAttribute(providerAttributeType, "bangalore"));
-		provider.setAttribute(createProviderAttribute(providerAttributeType, "chennai"));
+		ProviderAttributeType place = service.getProviderAttributeType(3);
+		provider.setAttribute(buildProviderAttribute(place, "bangalore"));
+		provider.setAttribute(buildProviderAttribute(place, "chennai"));
+		
 		Assert.assertEquals(1, provider.getAttributes().size());
+		
 		service.saveProvider(provider);
 		Assert.assertNotNull(provider.getId());
-		provider.setAttribute(createProviderAttribute(providerAttributeType, "seattle"));
+		
+		provider.setAttribute(buildProviderAttribute(place, "seattle"));
 		Assert.assertEquals(2, provider.getAttributes().size());
 		ProviderAttribute lastAttribute = (ProviderAttribute) provider.getAttributes().toArray()[0];
 		Assert.assertTrue(lastAttribute.getVoided());
