diff --git a/old_product.java b/new_product.java
index 94c86ce..fce3e5c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,6 +2,7 @@ private Item add(final QueryContext ctx) throws QueryException {
     final Data data = checkData(ctx);
     final byte[] path = expr.length < 3 ? Token.EMPTY : token(path(2, ctx));
     final NewInput input = checkInput(checkItem(expr[1], ctx), path);
-    ctx.updates.add(new DBAdd(data, input, ctx, info), ctx);
+    final Options opts = checkOptions(3, Q_OPTIONS, new Options(), ctx);
+    ctx.updates.add(new DBAdd(data, input, opts, ctx, info), ctx);
     return null;
   }
\ No newline at end of file
