diff --git a/old_test.java b/new_test.java
index 10789ce..d8fd38f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -30,4 +30,10 @@
         _DB_ADD.args(NAME, "\"<root/>\"", "\"doc\" || $i"));
     query(COUNT.args(" for $i in 1 to 3 return " +
         COLLECTION.args('"' + NAME + "/doc\" || $i")), 3);
+
+    // specify parsing options
+    query(_DB_ADD.args(NAME, " '<a> </a>'", "chop.xml", " map { 'chop':true() }"));
+    query(_DB_OPEN.args(NAME, "chop.xml"), "<a/>");
+    query(_DB_ADD.args(NAME, " '<a> </a>'", "nochop.xml", " map { 'chop':false() }"));
+    query(_DB_OPEN.args(NAME, "nochop.xml"), "<a> </a>");
   }
\ No newline at end of file
