diff --git a/old_test.java b/new_test.java
index 2442923..fd3a225 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,8 +5,7 @@
         final Date updateTime = getApp.getUpdated().orElseThrow(IllegalArgumentException::new);
 
         final String patchString = "[{ \"op\": \"replace\", \"path\": \"/user\", \"value\": \"" + APP_2_USER + "\" }]";
-        final ObjectMapper mapper = new ObjectMapper();
-        final JsonPatch patch = JsonPatch.fromJson(mapper.readTree(patchString));
+        final JsonPatch patch = JsonPatch.fromJson(GenieObjectMapper.getMapper().readTree(patchString));
 
         this.appService.patchApplication(APP_1_ID, patch);
 
