diff --git a/old_product.java b/new_product.java
index 9624708..6f07288 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
 void retractObject(FactHandle handle,
                               PropagationContext context,
-                              ReteooWorkingMemory workingMemory) throws FactException;
\ No newline at end of file
+                              AbstractWorkingMemory workingMemory) throws FactException;
\ No newline at end of file
