diff --git a/old_product.java b/new_product.java
index e44ec6d..bdf75b8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,7 @@
 @Override
     public boolean existGroup(String groupName) {
+        if (groupName == null || groupName.isEmpty()) {
+            throw new IllegalArgumentException("Groupname cannot be null nor empty");
+        }
         return featureGroups.containsKey(groupName);
     }
\ No newline at end of file
