diff --git a/old_test.java b/new_test.java
index 39c8d68..79fe6b5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,9 @@
 @Test
     public void testExistGroup() {
+        // Given
+        assertFf4j.assertThatGroupExist(G1);
+        assertFf4j.assertThatGroupDoesNotExist(G_DOESNOTEXIST);
+        // Then
         Assert.assertTrue(testedStore.existGroup(G1));
         Assert.assertFalse(testedStore.existGroup(G_DOESNOTEXIST));
     }
\ No newline at end of file
