diff --git a/old_test.java b/new_test.java
index a5e7c4e..a9d8597 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,6 @@
 		
 		orderFrequency.setFrequencyPerDay(4d);
 		expectedException.expect(APIException.class);
-		expectedException.expectMessage("This order frequency cannot be edited because it is already in use");
+		expectedException.expectMessage("Order.frequency.cannot.edit");
 		orderService.saveOrderFrequency(orderFrequency);
 	}
\ No newline at end of file
