diff --git a/old_product.java b/new_product.java
index fbef03f..32704b0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,8 @@
 @Override
 	public OrderFrequency saveOrderFrequency(OrderFrequency orderFrequency) throws APIException {
 		
-		if (orderFrequency.getOrderFrequencyId() != null) {
-			if (dao.isOrderFrequencyInUse(orderFrequency)) {
+		if (orderFrequency.getOrderFrequencyId() != null && dao.isOrderFrequencyInUse(orderFrequency)) {
 				throw new APIException("This order frequency cannot be edited because it is already in use");
-			}
 		}
 		
 		return dao.saveOrderFrequency(orderFrequency);
