diff --git a/old_product.java b/new_product.java
index 1f6688a..9d74762 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
 @Timeout(duration = 180, timeUnit = TimeUnit.SECONDS)
-   ServerDetails cloneServer(String serverid, String hostname, CloneServerOptions... options);
\ No newline at end of file
+   ServerDetails clone(String serverid, String hostname, CloneServerOptions... options);
\ No newline at end of file
