diff --git a/old_test.java b/new_test.java
index 14c6bd8..046350f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(enabled=false)
    public void testCloneServer() throws Exception {
-      ServerDetails testServer2 = api.cloneServer(serverId, testHostName2, CloneServerOptions.Builder.cpucores(1));
+      ServerDetails testServer2 = api.clone(serverId, testHostName2, CloneServerOptions.Builder.cpucores(1));
 
       assertNotNull(testServer2.getId());
       assertEquals(testServer2.getHostname(), "jclouds-test2");
@@ -11,19 +11,19 @@
       RetryablePredicate<Server.State> cloneChecker = new ServerStatusChecker(api, testServerId2, 300, 10, TimeUnit.SECONDS);
       assertTrue(cloneChecker.apply(Server.State.STOPPED));
 
-      api.startServer(testServer2.getId());
+      api.start(testServer2.getId());
 
       // TODO ServerStatus==STOPPED suggests the previous call to start should have worked
       cloneChecker = new RetryablePredicate<Server.State>(
             new Predicate<Server.State>() {
 
                public boolean apply(Server.State value) {
-                  ServerStatus status = api.getServerStatus(testServerId2, ServerStatusOptions.Builder.state());
+                  ServerStatus status = api.getStatus(testServerId2, ServerStatusOptions.Builder.state());
                   if (status.getState() == value) {
                      return true;
                   }
 
-                  api.startServer(testServerId2);
+                  api.start(testServerId2);
                   return false;
                }
 
