diff --git a/old_product.java b/new_product.java
index 25d24b2..809bb41 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@
         init();
         
         if (stream == null) {
-            throw new IOException("Stream can't be null");
+            throw new IOException("Stream can't be null.");
         }
         TabularDataIngest ingesteddata = new TabularDataIngest();
         DataTable dataTable = new DataTable();
