diff --git a/old_test.java b/new_test.java
index 8e322d3..de05254 100644
--- a/old_test.java
+++ b/new_test.java
@@ -30,7 +30,7 @@
 
         // Test for list streams for invalid scope.
         final CompletableFuture<List<StreamConfiguration>> completableFuture1 = new CompletableFuture<>();
-        completableFuture1.completeExceptionally(new DataNotFoundException(""));
+        completableFuture1.completeExceptionally(StoreException.create(StoreException.Type.DATA_NOT_FOUND));
         when(mockControllerService.listStreamsInScope("scope1")).thenReturn(completableFuture1);
         response = client.target(resourceURI).request().buildGet().invoke();
         assertEquals("List Streams response code", 404, response.getStatus());
