diff --git a/old_product.java b/new_product.java
index 21782a5..fc25996 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public void assertObject(final InternalFactHandle factHandle,
                              final PropagationContext context,
                              final InternalWorkingMemory workingMemory) {
-        EntryPoint entryPoint = context.getEntryPoint();
+        EntryPointId entryPoint = context.getEntryPoint();
         EntryPointNode node = this.entryPoints.get( entryPoint );
         ObjectTypeConf typeConf = ((InternalWorkingMemoryEntryPoint) workingMemory.getWorkingMemoryEntryPoint( entryPoint.getEntryPointId() )).getObjectTypeConfigurationRegistry().getObjectTypeConf( entryPoint,
                                                                                                                                                                                                        factHandle.getObject() );
