diff --git a/old_test.java b/new_test.java
index d24deae..5d852f8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -60,6 +60,8 @@
 
     RepositoryVersionEntity repoVersion = new RepositoryVersionEntity();
     repoVersion.setOperatingSystems(operatingSystemsJson);
+    StackEntity newDesiredStack = stackDAO.find("HDP", "2.0.1");
+    repoVersion.setStack(newDesiredStack);
 
     final ServiceOsSpecific.Package hivePackage = new ServiceOsSpecific.Package();
     hivePackage.setName("hive");
@@ -147,4 +149,5 @@
 
     // verify
     verify(managementController, response);
+    Assert.assertEquals(clusterEntity.getDesiredStack(), newDesiredStack);
   }
\ No newline at end of file
