diff --git a/old_product.java b/new_product.java
index 61c70a9..61e5fc5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -39,6 +39,13 @@
       LOG.info("Initiating finalization for manual upgrade to version {} for cluster {}",
               desiredRepoVersion, clName);
 
+      // First, set desired cluster stack version to enable cross-stack upgrade
+      StackId stackId = rve.getStackId();
+      ClusterEntity cluster = clusterDAO.findByName(clName);
+      StackEntity stackEntity = stackDAO.find(stackId.getStackName(), stackId.getStackVersion());
+      cluster.setDesiredStack(stackEntity);
+      clusterDAO.merge(cluster);
+
       Map<String, String> args = new HashMap<String, String>();
       if (newStateStr.equals(RepositoryVersionState.CURRENT.toString())) {
         // Finalize upgrade workflow
@@ -53,7 +60,6 @@
       }
 
       // Get a host name to populate the hostrolecommand table's hostEntity.
-      ClusterEntity cluster = clusterDAO.findByName(clName);
       String defaultHostName = null;
       List<HostEntity> hosts = new ArrayList(cluster.getHostEntities());
       if (hosts != null && !hosts.isEmpty()) {
