diff --git a/old_test.java b/new_test.java
index 79fd95f..301177f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void testAdd() {
-    BytesRef ref = new BytesRef();
+    BytesRefBuilder ref = new BytesRefBuilder();
     BytesRef scratch = new BytesRef();
     int num = atLeast(2);
     for (int j = 0; j < num; j++) {
@@ -13,7 +13,7 @@
         } while (str.length() == 0);
         ref.copyChars(str);
         int count = hash.size();
-        int key = hash.add(ref);
+        int key = hash.add(ref.get());
 
         if (key >=0) {
           assertTrue(strings.add(str));
