diff --git a/old_test.java b/new_test.java
index 3de3a6c..a01cc3e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Ignore
 	@Test
-	public void test1SaveModel() throws ClassNotFoundException, LibrecException, IOException {
+	public void test01SaveModel() throws ClassNotFoundException, LibrecException, IOException {
 		Configuration conf = new Configuration();
 		Configuration.Resource resource = new Resource("rec/cf/userknn-test.properties");
 		conf.addResource(resource);
