diff --git a/old_test.java b/new_test.java
index c91c0b8..1c7205c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,6 @@
         CostConstraint costConstraint = new CostConstraint(TE_COST);
         constraints.add(costConstraint);
 
-        boolean result = pceManager.setupPath(D1.deviceId(), D2.deviceId(), "T123", constraints, WITH_SIGNALLING);
+        boolean result = pceManager.setupPath(D1.deviceId(), D2.deviceId(), "T123", constraints, WITH_SIGNALLING, null);
         assertThat(result, is(false));
     }
\ No newline at end of file
