diff --git a/old_product.java b/new_product.java
index ca853fd..14b8fa5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,20 +1,10 @@
-public int nextFree(final int i) {
-    // calculate the index of the word in the array: i div 2^6 = i >> 6
-    int wi = i >>> WORD_POWER;
-    // invert the word and skip the first i bits:
-    long word = ~words[wi] & WORD_MASK << i;
-
-    if(word != 0) {
-      return (wi << WORD_POWER) + numberOfTrailingZeros(word);
-    }
-
+public int nextFree() {
     final int wl = words.length;
-    while(++wi < wl) {
-      if((word = ~words[wi]) != 0) {
-        return (wi << WORD_POWER) + numberOfTrailingZeros(word);
+    for(int w = 0; w < wl; w++) {
+      final long word = ~words[w];
+      if(word != 0) {
+        return (w << WORD_POWER) + numberOfTrailingZeros(word);
       }
     }
-
-    // wi * 2^6:
-    return wi << WORD_POWER;
+    return wl << WORD_POWER;
   }
\ No newline at end of file
