diff --git a/old_test.java b/new_test.java
index eab8010..f1e8c49 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test public void nextFree() {
-    final BitArray ba = new BitArray();
-    ba.init(new long[] {-1L, 0L}, 64);
-    assertEquals("Incorrect next clear bit", 64, ba.nextFree(0));
+    final BitArray ba = new BitArray(new long[] { -1L, 0L }, 64);
+    assertEquals("Incorrect next clear bit", 64, ba.nextFree());
   }
\ No newline at end of file
