diff --git a/old_product.java b/new_product.java
index de71d89..75f4986 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @Nonnull
     @VisibleForTesting
-    String toScript(@Nonnull final TransformRequest request) {
+    String toTransformScript(@Nonnull final TransformRequest request) {
         final StringBuilder script = new StringBuilder();
         script.append("class Transform (sqlContext: org.apache.spark.sql.SQLContext, sparkContextService: com.thinkbiganalytics.spark.SparkContextService) extends ");
         script.append(transformScriptClass.getName());
