diff --git a/old_test.java b/new_test.java
index e0e91ee..fd54a06 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,10 +6,10 @@
 
         // Test converting request to script
         final TransformService service = new TransformService(TransformScript.class, Mockito.mock(SparkScriptEngine.class), Mockito.mock(SparkContextService.class),
-                                                              Mockito.mock(JobTrackerService.class), Mockito.mock(DataSetConverterService.class),Mockito.mock(KyloCatalogClientBuilder.class));
+                                                              Mockito.mock(JobTrackerService.class), Mockito.mock(DataSetConverterService.class), Mockito.mock(KyloCatalogClientBuilder.class));
 
         InputStream inputStream = getClass().getResourceAsStream("transform-service-script1.scala");
         final String expected = IOUtils.toString(inputStream, "UTF-8");
         inputStream.close();
-        Assert.assertEquals(expected, service.toScript(request));
+        Assert.assertEquals(expected, service.toTransformScript(request));
     }
\ No newline at end of file
