diff --git a/old_test.java b/new_test.java
index d5d24d4..1d68eef 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,14 @@
 @Test(description = "GET /vApp/{id}")
    public void testGetVm() {
       // The method under test
-      vm = vmApi.getVm(vmURI);
+      vm = vmApi.get(vmURI);
 
       // Check the retrieved object is well formed
       checkVm(vm);
 
       // Check the required fields are set
-      assertEquals(vm.isDeployed(), Boolean.FALSE, String.format(OBJ_FIELD_EQ, VM, "deployed", "FALSE", vm.isDeployed().toString()));
+      assertEquals(vm.isDeployed(), Boolean.FALSE,
+               String.format(OBJ_FIELD_EQ, VM, "deployed", "FALSE", vm.isDeployed().toString()));
 
       // Check status
       assertVmStatus(vm.getHref(), Status.POWERED_OFF);
