diff --git a/old_product.java b/new_product.java
index 178ad18..7176def 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static TTest pairedTest(double[] x, double[] y) {
+public static TTest testPaired(double[] x, double[] y) {
         if (x.length != y.length) {
             throw new IllegalArgumentException("Input vectors have different size");
         }
@@ -22,5 +22,5 @@ public static TTest pairedTest(double[] x, double[] y) {
         double t = (mu1 - mu2) / sd;
         double p = Beta.regularizedIncompleteBetaFunction(0.5 * df, 0.5, df / (df + t * t));
 
-        return new TTest(t, df, p);
+        return new TTest("Paired", t, df, p);
     }
\ No newline at end of file
