diff --git a/old_test.java b/new_test.java
index e484f67..2c49098 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,9 +10,9 @@
     query(_DB_DROP_BACKUP.args(query(_DB_BACKUPS.args(NAME))));
 
     // invalid name
-    error(_DB_DROP_BACKUP.args(" ''"), Err.BXDB_NAME);
+    error(_DB_DROP_BACKUP.args(" ''"), Err.BXDB_NAME_X);
     // backup file does not exist
-    error(_DB_DROP_BACKUP.args(NAME), Err.BXDB_WHICHBACK);
+    error(_DB_DROP_BACKUP.args(NAME), Err.BXDB_WHICHBACK_X);
     // check if drop is called before create
-    error(_DB_CREATE_BACKUP.args(NAME) + ',' + _DB_DROP_BACKUP.args(NAME), Err.BXDB_WHICHBACK);
+    error(_DB_CREATE_BACKUP.args(NAME) + ',' + _DB_DROP_BACKUP.args(NAME), Err.BXDB_WHICHBACK_X);
   }
\ No newline at end of file
