diff --git a/old_product.java b/new_product.java
index 5ff5747..fae1fd0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,11 @@
-private void findLargestAnglesForAllNodes() {
+void findLargestAnglesForAllNodes() {
 		for (int i = 0; i < listInfo.size(); i++) {
 			NodeInfo info = listInfo.get(i);
 
 			if( info.edges.size < 2 )
 				continue;
 
-			for (int j = 0, k = info.edges.size-1; j < info.edges.size; k=j,j++) {
+			for (int k = 0, j = info.edges.size-1; k < info.edges.size; j=k,k++) {
 				double angleA = info.edges.get(j).angle;
 				double angleB = info.edges.get(k).angle;
 
