diff --git a/old_test.java b/new_test.java
index 62a3427..e01d217 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,8 @@
         final String hostname = "hostname";
         final int pid = 123;
         final long checkDelay = 3000L;
-        final JobExecution jobExecution = new JobExecution.Builder(hostname, pid, checkDelay)
+        final Date timeout = new Date();
+        final JobExecution jobExecution = new JobExecution.Builder(hostname, pid, checkDelay, timeout)
             .withId(JOB_1_ID)
             .build();
         final JobEntity jobEntity = Mockito.mock(JobEntity.class);
@@ -19,6 +20,7 @@
         Assert.assertEquals(hostname, argument.getValue().getHostName());
         Assert.assertEquals(pid, argument.getValue().getProcessId());
         Assert.assertThat(argument.getValue().getCheckDelay(), Matchers.is(checkDelay));
+        Assert.assertThat(argument.getValue().getTimeout(), Matchers.is(timeout));
         Assert.assertEquals(JOB_1_ID, argument.getValue().getId());
 
         // verify the method sets the status code of the Job to RUNNING
