diff --git a/old_test.java b/new_test.java
index 4c0b18e..74dc795 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,6 @@
   public void getFqdnHostTest() throws UnknownHostException {
     Assert.assertEquals(NetworkAddressUtils.getFqdnHost(new InetSocketAddress("localhost", 0)),
         "localhost");
-    Assert.assertEquals(NetworkAddressUtils.getFqdnHost(new WorkerNetAddress("localhost", 0, 0, 0)),
+    Assert.assertEquals(NetworkAddressUtils.getFqdnHost(new NetAddress("localhost", 0, 0, 0)),
         "localhost");
   }
\ No newline at end of file
