diff --git a/old_test.java b/new_test.java
index 1bd6836..e1d4c56 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testSetConfigs() {
-        Assert.assertNull(this.c.getConfigs());
+        Assert.assertNotNull(this.c.getConfigs());
         this.c.setConfigs(this.configs);
         Assert.assertEquals(this.configs, this.c.getConfigs());
     }
\ No newline at end of file
