diff --git a/old_product.java b/new_product.java
index ae7b98a..5281ad7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,6 +4,6 @@
    @Consumes(APPLICATION_JSON)
    @Path("/upload/storage/v1/b/{bucket}/o")
    @MapBinder(UploadBinder.class)
-   GCSObject simpleUpload(@PathParam("bucket") String bucketName, @HeaderParam("Content-Type") String contentType,
+   GoogleCloudStorageObject simpleUpload(@PathParam("bucket") String bucketName, @HeaderParam("Content-Type") String contentType,
             @HeaderParam("Content-Length") Long contentLength, @PayloadParam("payload") Payload payload,
             InsertObjectOptions Options);
\ No newline at end of file
