diff --git a/old_test.java b/new_test.java
index 86477eb..e47ddca 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,7 +11,7 @@
 
       InsertObjectOptions options = new InsertObjectOptions().name(UPLOAD_OBJECT_NAME);
 
-      GCSObject gcsObject = api().simpleUpload(BUCKET_NAME, "text/plain",
+      GoogleCloudStorageObject gcsObject = api().simpleUpload(BUCKET_NAME, "text/plain",
                payload.getPayload().getContentMetadata().getContentLength(), payload.getPayload(), options);
 
       assertNotNull(gcsObject);
