diff --git a/old_test.java b/new_test.java
index 51981c5..39d20ed 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,14 +10,13 @@
 
         // Callback for when a frame is written.
         AtomicReference<DataFrame> writtenFrame = new AtomicReference<>();
-        AtomicLong seqNo = new AtomicLong(0);
-        ConsumerWithException<DataFrame, IOException> callback = df -> {
+        Consumer<DataFrame> callback = df -> {
             Assert.assertNull("A frame has already been written.", writtenFrame.get());
             writtenFrame.set(df);
         };
 
         ArrayList<byte[]> records = DataFrameTestHelpers.generateRecords(2, 0, 1024);
-        try (DataFrameOutputStream s = new DataFrameOutputStream(maxFrameSize, seqNo::getAndIncrement, callback)) {
+        try (DataFrameOutputStream s = new DataFrameOutputStream(maxFrameSize, callback)) {
             // Test 1: write record + reset + flush -> no frame.
             s.startNewRecord();
             s.write(records.get(0));
