diff --git a/old_product.java b/new_product.java
index ba7a0b6..d4c062c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,18 +1,18 @@
-public static final ServiceRequestPathInfo parseServiceRequestPath(String apiVersionHeader, String acceptHeader, String pathInfo) {
+public static final ApiRequestPathInfo parseApiRequestPath(String apiVersionHeader, String acceptHeader, String pathInfo) {
         //String pathInfo = request.getPathInfo();
-        ServiceRequestPathInfo info = new ServiceRequestPathInfo();
+        ApiRequestPathInfo info = new ApiRequestPathInfo();
 
         boolean versionFound = false;
 
         //String apiVersionHeader = request.getHeader("X-API-Version"); //$NON-NLS-1$
         if (apiVersionHeader != null && apiVersionHeader.trim().length() > 0) {
-            info.serviceVersion = apiVersionHeader;
+            info.apiVersion = apiVersionHeader;
             versionFound = true;
         } else {
             //String acceptHeader = request.getHeader("Accept"); //$NON-NLS-1$
             if (acceptHeader != null && acceptHeader.startsWith("application/apiman.")) { //$NON-NLS-1$
                 String [] split = acceptHeader.split("\\+"); //$NON-NLS-1$
-                info.serviceVersion = split[0].substring("application/apiman.".length()); //$NON-NLS-1$
+                info.apiVersion = split[0].substring("application/apiman.".length()); //$NON-NLS-1$
                 versionFound = true;
             }
         }
@@ -23,9 +23,9 @@ public static final ServiceRequestPathInfo parseServiceRequestPath(String apiVer
             String[] split = pathInfo.split("/"); //$NON-NLS-1$
             if (split.length >= minParts) {
                 info.orgId = split[1];
-                info.serviceId = split[2];
+                info.apiId = split[2];
                 if (!versionFound) {
-                    info.serviceVersion = split[3];
+                    info.apiVersion = split[3];
                 }
                 if (split.length > minParts) {
                     StringBuilder resource = new StringBuilder();
