diff --git a/old_test.java b/new_test.java
index 31a31bd..2d36dd1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,59 +1,59 @@
 @Test
-    public void testParseServiceRequestPath() {
-        ServiceRequestPathInfo info = parseServiceRequestPath(null);
+    public void testParseApiRequestPath() {
+        ApiRequestPathInfo info = parseApiRequestPath(null);
 
-        info = parseServiceRequestPath("/invalidpath");
+        info = parseApiRequestPath("/invalidpath");
         Assert.assertNull(info.orgId);
-        Assert.assertNull(info.serviceId);
-        Assert.assertNull(info.serviceVersion);
+        Assert.assertNull(info.apiId);
+        Assert.assertNull(info.apiVersion);
         Assert.assertNull(info.resource);
 
-        info = parseServiceRequestPath("/invalid/path");
+        info = parseApiRequestPath("/invalid/path");
         Assert.assertNull(info.orgId);
-        Assert.assertNull(info.serviceId);
-        Assert.assertNull(info.serviceVersion);
+        Assert.assertNull(info.apiId);
+        Assert.assertNull(info.apiVersion);
         Assert.assertNull(info.resource);
 
-        info = parseServiceRequestPath("/Org1/Service1/1.0");
+        info = parseApiRequestPath("/Org1/Api1/1.0");
         Assert.assertEquals("Org1", info.orgId);
-        Assert.assertEquals("Service1", info.serviceId);
-        Assert.assertEquals("1.0", info.serviceVersion);
+        Assert.assertEquals("Api1", info.apiId);
+        Assert.assertEquals("1.0", info.apiVersion);
         Assert.assertNull(info.resource);
 
-        info = parseServiceRequestPath("/MyOrg/Service-99/2.7");
+        info = parseApiRequestPath("/MyOrg/Api-99/2.7");
         Assert.assertEquals("MyOrg", info.orgId);
-        Assert.assertEquals("Service-99", info.serviceId);
-        Assert.assertEquals("2.7", info.serviceVersion);
+        Assert.assertEquals("Api-99", info.apiId);
+        Assert.assertEquals("2.7", info.apiVersion);
         Assert.assertNull(info.resource);
 
-        info = parseServiceRequestPath("/MyOrg/Service-99/2.7/resource");
+        info = parseApiRequestPath("/MyOrg/Api-99/2.7/resource");
         Assert.assertEquals("MyOrg", info.orgId);
-        Assert.assertEquals("Service-99", info.serviceId);
-        Assert.assertEquals("2.7", info.serviceVersion);
+        Assert.assertEquals("Api-99", info.apiId);
+        Assert.assertEquals("2.7", info.apiVersion);
         Assert.assertEquals("/resource", info.resource);
 
-        info = parseServiceRequestPath("/MyOrg/Service-99/2.7/path/to/resource");
+        info = parseApiRequestPath("/MyOrg/Api-99/2.7/path/to/resource");
         Assert.assertEquals("MyOrg", info.orgId);
-        Assert.assertEquals("Service-99", info.serviceId);
-        Assert.assertEquals("2.7", info.serviceVersion);
+        Assert.assertEquals("Api-99", info.apiId);
+        Assert.assertEquals("2.7", info.apiVersion);
         Assert.assertEquals("/path/to/resource", info.resource);
 
-        info = parseServiceRequestPath("/MyOrg/Service-99/2.7/path/to/resource?query=1234");
+        info = parseApiRequestPath("/MyOrg/Api-99/2.7/path/to/resource?query=1234");
         Assert.assertEquals("MyOrg", info.orgId);
-        Assert.assertEquals("Service-99", info.serviceId);
-        Assert.assertEquals("2.7", info.serviceVersion);
+        Assert.assertEquals("Api-99", info.apiId);
+        Assert.assertEquals("2.7", info.apiVersion);
         Assert.assertEquals("/path/to/resource?query=1234", info.resource);
 
-        info = parseServiceRequestPath("/MyOrg/Service-99/path/to/resource?query=1234", null, "2.7");
+        info = parseApiRequestPath("/MyOrg/Api-99/path/to/resource?query=1234", null, "2.7");
         Assert.assertEquals("MyOrg", info.orgId);
-        Assert.assertEquals("Service-99", info.serviceId);
-        Assert.assertEquals("2.7", info.serviceVersion);
+        Assert.assertEquals("Api-99", info.apiId);
+        Assert.assertEquals("2.7", info.apiVersion);
         Assert.assertEquals("/path/to/resource?query=1234", info.resource);
 
-        info = parseServiceRequestPath("/MyOrg/Service-99/path/to/resource?query=1234", "application/apiman.2.7+json", null);
+        info = parseApiRequestPath("/MyOrg/Api-99/path/to/resource?query=1234", "application/apiman.2.7+json", null);
         Assert.assertEquals("MyOrg", info.orgId);
-        Assert.assertEquals("Service-99", info.serviceId);
-        Assert.assertEquals("2.7", info.serviceVersion);
+        Assert.assertEquals("Api-99", info.apiId);
+        Assert.assertEquals("2.7", info.apiVersion);
         Assert.assertEquals("/path/to/resource?query=1234", info.resource);
 
     }
\ No newline at end of file
