diff --git a/old_test.java b/new_test.java
index 63bf45e..b2fbfc1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,27 +5,21 @@
 
         String server1 = "dns1";
         OperationsServerLoadHistory server1History = new OperationsServerLoadHistory(MAX_HISTORY_TIME_LIVE);
-        server1History.addOpsServerLoad(0, 10, 0);
-        Map<ZkChannelType, OperationsServerLoadHistory> server1OpsHistory = new HashMap<>();
-        server1OpsHistory.put(ZkChannelType.HTTP, server1History);
+        server1History.addOpsServerLoad(new LoadInfo(10));
 
         String server2 = "dns2";
         OperationsServerLoadHistory server2History = new OperationsServerLoadHistory(MAX_HISTORY_TIME_LIVE);
-        server2History.addOpsServerLoad(0, 30, 0);
-        Map<ZkChannelType, OperationsServerLoadHistory> server2OpsHistory = new HashMap<>();
-        server2OpsHistory.put(ZkChannelType.HTTP, server2History);
+        server2History.addOpsServerLoad(new LoadInfo(30));
 
         String server3 = "dns3";
         OperationsServerLoadHistory server3History = new OperationsServerLoadHistory(MAX_HISTORY_TIME_LIVE);
-        server3History.addOpsServerLoad(0, 55, 0);
-        Map<ZkChannelType, OperationsServerLoadHistory> server3OpsHistory = new HashMap<>();
-        server3OpsHistory.put(ZkChannelType.HTTP, server3History);
+        server3History.addOpsServerLoad(new LoadInfo(55));
 
-        Map<String, Map<ZkChannelType, OperationsServerLoadHistory>> serversHistory = new HashMap<>();
+        Map<String, OperationsServerLoadHistory> serversHistory = new HashMap<>();
 
-        serversHistory.put(server1, server1OpsHistory);
-        serversHistory.put(server2, server2OpsHistory);
-        serversHistory.put(server3, server3OpsHistory);
+        serversHistory.put(server1, server1History);
+        serversHistory.put(server2, server2History);
+        serversHistory.put(server3, server3History);
         Map<String,RedirectionRule> rules = rebalancer.recalculate(serversHistory );
         assertNotNull(rules);
 
