diff --git a/old_product.java b/new_product.java
index 1ea6214..e615b44 100644
--- a/old_product.java
+++ b/new_product.java
@@ -33,7 +33,6 @@ public void repairSurvivors()
 
         m_promotionResult.set(new RepairResult(
                 m_maxSeenTxnId,
-                m_maxSeenUniqueId,
                 m_maxSeenBinaryLogSequenceNumber,
                 m_maxSeenBinaryLogUniqueId));
     }
\ No newline at end of file
