diff --git a/old_test.java b/new_test.java
index 086dc52..3414942 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(expectedExceptions=NullPointerException.class)
     public void test_print_nullAppendable() throws Exception {
         CharLiteralPrinterParser pp = new CharLiteralPrinterParser('a');
-        pp.print(emptyDateTime, (Appendable) null, symbols);
+        pp.print(emptyCalendrical, (Appendable) null, symbols);
     }
\ No newline at end of file
