diff --git a/old_test.java b/new_test.java
index 9dcaedf..5c3f730 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
 		
 		AnyM<String> responses = LiftMFunctions.liftM2(this::response).apply(anyM(stream1), anyM(stream2));
 		
-		assertThat(responses.traversable().toList(),equalTo(Arrays.asList("all upper::MIXEDCASE", 
+		assertThat(responses.toSequence().toList(),equalTo(Arrays.asList("all upper::MIXEDCASE", 
 				"all upper::ALL LOWER", "mixed case::MIXEDCASE", "mixed case::ALL LOWER")));
 		
 	}
\ No newline at end of file
