diff --git a/old_test.java b/new_test.java
index 4ce784a..62caf54 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,12 +2,12 @@
     @Transactional
     public void testDisableGroup() {
         // Given
-        assertFf4j.assertEnable(F4);
+        assertFf4j.assertThatFeatureIsEnabled(F4);
         // When
         testedStore.disableGroup(G1);
         // Then
-        assertFf4j.assertDisable(F4);
+        assertFf4j.assertThatFeatureIsDisabled(F4);
         // Cancel modifications
         testedStore.enable(F4);
-        assertFf4j.assertEnable(F4);
+        assertFf4j.assertThatFeatureIsEnabled(F4);
     }
\ No newline at end of file
