diff --git a/old_product.java b/new_product.java
index d987b04..384f0f3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @Override
     public void disableGroup(String groupName) {
+        getTarget().disableGroup(groupName);
         // Cannot know wich feature to work with (exceptional event) : flush cache
         getCacheManager().clear();
-        getTarget().disableGroup(groupName);
     }
\ No newline at end of file
