diff --git a/old_test.java b/new_test.java
index 5d6ae92..6a699ba 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,7 @@
 @Test
-	public void testAnnotateText() {
-		Annotations annotations = service.annotateText(Graph.WIKIPEDIA,
-				"Nizar Magboul Alseddeg is currently living in Austin Texas");
-		Assert.assertNotNull(annotations);
-	}
\ No newline at end of file
+  public void testAnnotateText() {
+    final Annotations annotations =
+        service.annotateText(Graph.WIKIPEDIA,
+            "Nizar Magboul Alseddeg is currently living in Austin Texas");
+    Assert.assertNotNull(annotations);
+  }
\ No newline at end of file
