diff --git a/old_test.java b/new_test.java
index 4e2abb5..7880ad5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void startNewSpan() {
-        state.setCurrentServerSpan(ServerSpan.create(PARENT_SPAN_ID.toSpan().setName("name")));
+        state.setCurrentServerSpan(ServerSpan.create(PARENT_SPAN_ID, "name"));
 
         PowerMockito.when(System.currentTimeMillis()).thenReturn(1L);
         PowerMockito.when(System.nanoTime()).thenReturn(500L);
