diff --git a/old_product.java b/new_product.java
index 6194875..2db1eb8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,7 +15,7 @@ public SpanId startNewSpan(String requestName) {
             }
         }
 
-        Span newSpan = newSpanId.toSpan();
+        Span newSpan = Span.fromSpanId(newSpanId);
         newSpan.setName(requestName);
         spanAndEndpoint().state().setCurrentClientSpan(newSpan);
         return newSpanId;
