diff --git a/old_test.java b/new_test.java
index df7444c..f804679 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,7 +14,7 @@
 		Errors errors = new BindException(group, "group");
 		getValidator().validate(group, errors);
 		
-		Assert.assertTrue(errors.hasErrors());
+		Assert.assertTrue(errors.hasFieldErrors("members[0].concept"));
 		Assert.assertFalse(errors.hasFieldErrors("creator"));
 		Assert.assertFalse(errors.hasFieldErrors("patient"));
 		Assert.assertFalse(errors.hasFieldErrors("dateCreated"));
