diff --git a/old_test.java b/new_test.java
index ebac90f..e935829 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,6 +11,6 @@
     mLocalTachyonCluster.stopTFS();
     RenameTestUtil();
     String editLogPath = mLocalTachyonCluster.getEditLogPath();
-    UnderFileSystem.get(editLogPath).delete(editLogPath, true);
+    UnderFileSystem.get(editLogPath, mMasterTachyonConf).delete(editLogPath, true);
     RenameTestUtil();
   }
\ No newline at end of file
