diff --git a/old_product.java b/new_product.java
index f4dbbd3..8046e2f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-public static void rename(String src, String dst) throws IOException {
-    UnderFileSystem ufs = UnderFileSystem.get(src);
+public static void rename(String src, String dst, MasterInfo info) throws IOException {
+    UnderFileSystem ufs = UnderFileSystem.get(src, info.getTachyonConf());
     ufs.rename(src, dst);
     LOG.info("Renamed " + src + " to " + dst);
   }
\ No newline at end of file
