diff --git a/old_product.java b/new_product.java
index 783d763..1df9a61 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,13 @@
 @Override
-	public UserListResult searchUser(String loginName, String name) {
+	public UserListResponse searchUser(String loginName, String name) {
 		try {
 			List<User> entityList = accountManager.searchUser(loginName, name);
 
 			List<UserDTO> dtoList = BeanMapper.mapList(entityList, UserDTO.class);
 
-			return new UserListResult(dtoList);
+			return new UserListResponse(dtoList);
 		} catch (RuntimeException e) {
 			logger.error(e.getMessage(), e);
-			return new UserListResult().setDefaultError();
+			return new UserListResponse().setDefaultError();
 		}
 	}
\ No newline at end of file
