diff --git a/old_test.java b/new_test.java
index 22b27d7..4bd2e73 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
 	public void searchUser() {
 		AccountWebService accountWebService = creatClient();
 
-		UserListResult result = accountWebService.searchUser(null, null);
+		UserListResponse result = accountWebService.searchUser(null, null);
 
 		assertTrue(result.getUserList().size() >= 4);
 		assertEquals("Jack", result.getUserList().get(0).getName());
