diff --git a/old_test.java b/new_test.java
index 79fe79e..b4a096a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -21,7 +21,7 @@
 		}
 		Context.logout();
 		
-		for (int x = 1; x <= 5; x++) {
+		for (int x = 1; x <= 8; x++) {
 			// now we fail several login attempts 
 			try {
 				dao.authenticate("admin", "not the right password");
@@ -32,7 +32,7 @@
 			}
 		}
 		
-		// those were the first five, now the sixth request 
+		// those were the first seven, now the eigth request 
 		// (with the same user and right pw) should fail
 		dao.authenticate("admin", "test");
     }
\ No newline at end of file
