diff --git a/old_product.java b/new_product.java
index cc401ac..548de1e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,10 +2,10 @@ public void setJobStatus(final JobStatus jobStatus) {
         this.status = jobStatus;
 
         if (jobStatus == JobStatus.INIT) {
-            setStartTime(System.currentTimeMillis());
+            this.setStarted(new Date());
         } else if (jobStatus == JobStatus.SUCCEEDED
                 || jobStatus == JobStatus.KILLED
                 || jobStatus == JobStatus.FAILED) {
-            setFinishTime(System.currentTimeMillis());
+            setFinished(new Date());
         }
     }
\ No newline at end of file
