diff --git a/old_product.java b/new_product.java
index 501d4bf..8272e04 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,12 +16,8 @@
         if (LOG.isDebugEnabled()) {
             LOG.debug("Setting the command info for job with id " + id);
         }
-        final Job job = this.jobRepo.findOne(id);
-        if (job != null) {
-            //TODO: Should we check if this is valid
-            job.setCommandId(commandId);
-            job.setCommandName(commandName);
-        } else {
-            throw new GenieNotFoundException("No job with id " + id + " exists");
-        }
+        final JobEntity jobEntity = this.findJob(id);
+        //TODO: Should we check if this is valid
+        jobEntity.setCommandId(commandId);
+        jobEntity.setCommandName(commandName);
     }
\ No newline at end of file
