diff --git a/old_test.java b/new_test.java
index cc96744..dc3f6aa 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,6 +18,8 @@
     ckDBs(new XQuery(_DB_RETRIEVE.args(NAME, "foo")), false, NAME_LIST);
     ckDBs(new XQuery(_DB_TEXT.args(NAME, "foo")), false, NAME_LIST);
     ckDBs(new XQuery(_DB_TEXT_RANGE.args(NAME, 23, 42)), false, NAME_LIST);
+    ckDBs(new XQuery(_DB_TOKEN.args(NAME, "foo")), false, NAME_LIST);
+    ckDBs(new XQuery(_DB_TOKEN.args(NAME, 23, 42)), false, NAME_LIST);
 
     // Updates
     ckDBs(new XQuery(_DB_CREATE.args(NAME)), true, NAME_LIST);
