diff --git a/old_product.java b/new_product.java
index fab0716..1f27e0a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -30,10 +30,11 @@ public static String db(final MetaData meta, final boolean bold, final boolean i
         info(tb, UP_TO_DATE, meta.uptodate);
         info(tb, MainOptions.TEXTINDEX.name(), meta.textindex);
         info(tb, MainOptions.ATTRINDEX.name(), meta.attrindex);
-        info(tb, MainOptions.ATTRTOKENIZE.name(), meta.attrtokeninclude);
+        info(tb, MainOptions.TOKENINDEX.name(), meta.tokenindex);
         info(tb, MainOptions.FTINDEX.name(), meta.ftindex);
         info(tb, MainOptions.TEXTINCLUDE.name(), meta.textinclude);
         info(tb, MainOptions.ATTRINCLUDE.name(), meta.attrinclude);
+        info(tb, MainOptions.TOKENINCLUDE.name(), meta.tokeninclude);
         info(tb, MainOptions.FTINCLUDE.name(), meta.ftinclude);
         info(tb, MainOptions.LANGUAGE.name(), meta.language);
         info(tb, MainOptions.STEMMING.name(), meta.stemming);
