diff --git a/old_test.java b/new_test.java
index c7ac22e..f4a64df 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,7 @@
-@Test
-	public void setAsText_shouldSetUsingUuid() {
-		PersonEditor editor = new PersonEditor();
-		editor.setAsText("da7f524f-27ce-4bb2-86d6-6d1d05312bd5");
-		Assert.assertNotNull(editor.getValue());
+@Override
+	@Ignore("to investigate, this behavior deviates from most openmrs propertyeditors")
+	@Test(expected = IllegalArgumentException.class)
+	public void shouldFailToSetTheEditorValueIfGivenUuidDoesNotExist() {
+		
+		editor.setAsText(getNonExistingObjectUuid());
 	}
\ No newline at end of file
