diff --git a/old_product.java b/new_product.java
index 2148ade..06b8906 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,4 @@
-@Test
-	public void setAsText_shouldSetUsingUuid() {
-		DrugEditor drugEditor = new DrugEditor();
-		drugEditor.setAsText("3cfcf118-931c-46f7-8ff6-7b876f0d4202");
-		Assert.assertNotNull(drugEditor.getValue());
+@Override
+	protected Drug getExistingObject() {
+		return conceptService.getDrug(EXISTING_ID);
 	}
\ No newline at end of file
