diff --git a/old_test.java b/new_test.java
index a31ea80..f5b51dc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,12 +9,19 @@
     );
 
     assertThat(store().getTraces(
-      requestBuilder().serviceName("frontend").build()
+      requestBuilder().serviceName(CLIENT_SPAN.localServiceName()).build()
     ).execute()).isEmpty();
 
     assertThat(store().getTraces(
       requestBuilder()
-        .serviceName("frontend")
+        .serviceName(CLIENT_SPAN.localServiceName())
+        .spanName(CLIENT_SPAN.remoteServiceName())
+        .build()
+    ).execute()).isEmpty();
+
+    assertThat(store().getTraces(
+      requestBuilder()
+        .serviceName(CLIENT_SPAN.localServiceName())
         .spanName(CLIENT_SPAN.name())
         .build()
     ).execute()).isEmpty();
@@ -23,12 +30,19 @@
     accept(CLIENT_SPAN);
 
     assertThat(store().getTraces(
-      requestBuilder().serviceName("frontend").build()
+      requestBuilder().serviceName(CLIENT_SPAN.localServiceName()).build()
+    ).execute()).isNotEmpty();
+
+    assertThat(store().getTraces(
+      requestBuilder()
+        .serviceName(CLIENT_SPAN.localServiceName())
+        .remoteServiceName(CLIENT_SPAN.remoteServiceName())
+        .build()
     ).execute()).isNotEmpty();
 
     assertThat(store().getTraces(
       requestBuilder()
-        .serviceName("frontend")
+        .serviceName(CLIENT_SPAN.localServiceName())
         .spanName(CLIENT_SPAN.name())
         .build()
     ).execute()).isNotEmpty();
