diff --git a/old_test.java b/new_test.java
index 5eb010a..ac525f9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -70,7 +70,4 @@
             = argument.getValue().getDescription().orElseThrow(IllegalArgumentException::new);
         Assert.assertThat(actualDescription, Matchers.is(description));
         Assert.assertThat(argument.getValue().getApplicationsAsList(), Matchers.empty());
-        Assert.assertThat(argument.getValue().getJob(), Matchers.notNullValue());
-        Assert.assertThat(argument.getValue().getJob().getExecution(), Matchers.notNullValue());
-        Assert.assertThat(argument.getValue().getJobMetadata(), Matchers.notNullValue());
     }
\ No newline at end of file
