diff --git a/old_product.java b/new_product.java
index 2e271cc..02f6363 100644
--- a/old_product.java
+++ b/new_product.java
@@ -23,9 +23,12 @@
         final JobEntity jobEntity = this.jobDtoToEntity(job);
         final JobExecutionEntity jobExecutionEntity = this.jobExecutionDtoToEntity(jobExecution);
 
-        jobEntity.setExecution(jobExecutionEntity);
-        jobRequestEntity.setJobMetadata(metadataEntity);
-        jobRequestEntity.setJob(jobEntity);
-
         this.jobRequestRepo.save(jobRequestEntity);
+
+        jobEntity.setRequest(jobRequestEntity);
+        this.jobRepo.save(jobEntity);
+        metadataEntity.setRequest(jobRequestEntity);
+        this.jobMetadataRepository.save(metadataEntity);
+        jobExecutionEntity.setJob(jobEntity);
+        this.jobExecutionRepo.save(jobExecutionEntity);
     }
\ No newline at end of file
