diff --git a/old_test.java b/new_test.java
index e30daff..4c5425d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public void testDelete() throws BaseXException {
     // try to delete a package which is not installed
     try {
-      new RepoManager(ctx).delete("xyz", null);
+      new RepoManager(ctx.repo).delete("xyz", null);
       fail("Not installed package not detected.");
     } catch(final QueryException ex) {
       check(ex, Err.PKGNOTINST);
@@ -32,7 +32,7 @@
 
     // try to delete pkg3
     try {
-      new RepoManager(ctx).delete("pkg3", null);
+      new RepoManager(ctx.repo).delete("pkg3", null);
       fail("Package involved in a dependency was deleted.");
     } catch(final QueryException ex) {
       check(ex, Err.PKGDEP);
