diff --git a/old_test.java b/new_test.java
index 4005c40..feb073b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,6 +20,6 @@
     // deleting a backup passing the exact backup name as argument
     ok(new CreateDB(NAME));
     ok(new CreateBackup(NAME));
-    ok(new DropBackup(ShowBackups.list(NAME, false, CONTEXT).get(0)));
-    assertEquals(0, ShowBackups.list(NAME, false, CONTEXT).size());
+    ok(new DropBackup(CONTEXT.getDatabases().listBackups().get(0)));
+    assertEquals(0, CONTEXT.getDatabases().listBackups(NAME).size());
   }
\ No newline at end of file
