diff --git a/old_product.java b/new_product.java
index bb50362..2926eae 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
-private boolean restore(final IOFile file) {
+private boolean restore(final IOFile file, final String db) {
     try {
       progress(new Zip(file)).unzip(mprop.dbpath());
+      context.getDatabases().add(db);
       return true;
     } catch(final IOException ex) {
       Util.debug(ex);
