diff --git a/old_product.java b/new_product.java
index 5e7e70d..243c715 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 private StrStream text(final QueryContext qc) throws QueryException {
-    final byte[] uri = checkStr(exprs[0], qc);
-    final String enc = checkEncoding(1, BXFE_ENCODING, qc);
-    return new StrStream(IO.get(Token.string(uri)), enc, BXFE_IO, qc);
+    final byte[] uri = toToken(exprs[0], qc);
+    final String enc = toEncoding(1, BXFE_ENCODING_X, qc);
+    return new StrStream(IO.get(Token.string(uri)), enc, BXFE_IO_X, qc);
   }
\ No newline at end of file
