diff --git a/old_product.java b/new_product.java
index 5ed265a..d7ffa77 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1,3 @@
-Optional<JobStatus> getJobStatus(@NotBlank(message = "Job id is missing and is required") String id);
\ No newline at end of file
+JobStatus getJobStatus(
+        @NotBlank(message = "Job id is missing and is required") String id
+    ) throws GenieNotFoundException;
\ No newline at end of file
