diff --git a/old_product.java b/new_product.java
index 8ee6dc5..59e4be6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,9 +16,11 @@ public void handle(OpenmrsObject openmrsObject, User creator, Date dateCreated,
 			try {
 				Object value = PropertyUtils.getProperty(openmrsObject, field.getName());
 				if (value != null && value.toString().isEmpty()) {
-					//TODO Commenting this out until we resolve the cases where some non
-					//nullable fields have emptry string holders.
-					//PropertyUtils.setProperty(openmrsObject, field.getName(), null);
+					
+					//Set to null only if object is not already voided
+					if (!(openmrsObject instanceof Voidable && ((Voidable)openmrsObject).isVoided())) {
+						PropertyUtils.setProperty(openmrsObject, field.getName(), null);
+					}
 				}
 			}
 			catch (Exception ex) {
