diff --git a/old_test.java b/new_test.java
index 01f3bed..093ac6d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,9 @@
-@Ignore
-	@Test
+@Test
 	@Verifies(value = "set empty string properties to null", method = "handle(OpenmrsObject,User,Date,String)")
 	public void handle_shouldSetEmptyStringPropertiesToNull() {
 		Role role = new Role();
 		role.setName("");
-		role.setDescription(" ");
+		role.setDescription("");
 		role.setRole("");
 		
 		new OpenmrsObjectSaveHandler().handle(role, null, null, null);
