diff --git a/old_product.java b/new_product.java
index 661116b..d0952fb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,7 +12,6 @@ void replaceLastFile(FileDescriptor fileDescriptor) {
             Preconditions.checkArgument(fileDescriptor.getEpoch() >= expectedMinEpoch,
                     "Invalid epoch. Expected at least %s, actual %s.", expectedMinEpoch, fileDescriptor.getEpoch());
 
-            this.files.remove(lastIndex);
-            this.files.add(fileDescriptor);
+            this.files.set(lastIndex, fileDescriptor);
         }
     }
\ No newline at end of file
