diff --git a/old_product.java b/new_product.java
index a528f22..9033508 100644
--- a/old_product.java
+++ b/new_product.java
@@ -27,10 +27,10 @@ public Connection getConnection() throws SQLException {
 			maybeSignalForMoreConnections(connectionPartition);  // see if we need to create more
 		}
 
-		// we still didn't find an empty one, wait forever until our partition is free
+		// we still didn't find an empty one, wait forever (or as per config) until our partition is free
 		if (result == null) {
 			try {
-				result = connectionPartition.getFreeConnections().take();
+				result = connectionPartition.getFreeConnections().poll(this.connectionTimeout, TimeUnit.MILLISECONDS);
 			}
 			catch (InterruptedException e) {
 				throw new SQLException(e.getMessage());
