diff --git a/old_product.java b/new_product.java
index 61ab56c..e896877 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
 public static void checkNearCacheConfig(NearCacheConfig nearCacheConfig, boolean isClient) {
         checkNotNative(nearCacheConfig.getInMemoryFormat());
+        checkEvictionConfig(nearCacheConfig.getEvictionConfig(), true);
 
         if (isClient && nearCacheConfig.isCacheLocalEntries()) {
             throw new IllegalArgumentException(
