diff --git a/old_test.java b/new_test.java
index 6233873..d6b5668 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test(description = "PUT /vApp/{id}/networkConnectionSection", dependsOnMethods = { "testEditGuestCustomizationSection" })
    public void testEditNetworkConnectionSection() {
-      powerOffVm(vm.getHref());
+      powerOffVm(vmUrn);
       // Look up a network in the Vdc
       Set<Reference> networks = vdc.getAvailableNetworks();
       Reference network = Iterables.getLast(networks);
 
       // Copy existing section and edit fields
-      NetworkConnectionSection oldSection = vmApi.getNetworkConnectionSection(vm.getHref());
+      NetworkConnectionSection oldSection = vmApi.getNetworkConnectionSection(vmUrn);
       NetworkConnectionSection newSection = oldSection
                .toBuilder()
                .networkConnection(
@@ -14,12 +14,12 @@
                                  .network(network.getName()).build()).build();
 
       // The method under test
-      Task editNetworkConnectionSection = vmApi.editNetworkConnectionSection(vm.getHref(), newSection);
+      Task editNetworkConnectionSection = vmApi.editNetworkConnectionSection(vmUrn, newSection);
       assertTrue(retryTaskSuccess.apply(editNetworkConnectionSection),
                String.format(TASK_COMPLETE_TIMELY, "editNetworkConnectionSection"));
 
       // Retrieve the modified section
-      NetworkConnectionSection modified = vmApi.getNetworkConnectionSection(vm.getHref());
+      NetworkConnectionSection modified = vmApi.getNetworkConnectionSection(vmUrn);
 
       // Check the retrieved object is well formed
       checkNetworkConnectionSection(modified);
