diff --git a/old_product.java b/new_product.java
index 708a5be..a6be8a8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,6 @@
-private void connectNodes() {
+void connectNodes() {
+		setupSearch();
+
 		for (int i = 0; i < nodes.size(); i++) {
 			SquareNode n = nodes.get(i);
 			double[] point = searchPoints.get(i);
@@ -10,13 +12,13 @@ private void connectNodes() {
 
 			// find it's neighbors
 			searchResults.reset();
-			search.findNearest(point,neighborDistance,MAX_NEIGHBORS,searchResults);
+			search.findNearest(point, neighborDistance*neighborDistance, MAX_NEIGHBORS + 1, searchResults);
 
 			// try to attach it's closest neighbors
 			for (int j = 0; j < searchResults.size(); j++) {
 				NnData<SquareNode> neighbor = searchResults.get(j);
 				if( neighbor.data != n )
-					considerAttach(n,neighbor.data);
+					considerConnect(n, neighbor.data);
 			}
 		}
 	}
\ No newline at end of file
