diff --git a/old_product.java b/new_product.java
index 3b8abf2..8d46ece 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,6 @@
 public static String leftPad(String str, int len, String fill) {
-    StringBuilder sb = new StringBuilder(len);
-    if (str.length() < len) {
-      for (int i = str.length(); i < len; i++) {
-        sb.append(fill);
-      }
+    if (str != null && str.length() < len) {
+      return Strings.padStart(str, len, fill.charAt(0));
     }
-    sb.append(str);
-    return sb.toString();
+    return str;
   }
\ No newline at end of file
