diff --git a/old_product.java b/new_product.java
index f78dab8..90e9d0a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,13 +14,13 @@ private Expr function(final QNm name, final Expr... exprs) throws QueryException
       final Expr f = lit != null ? lit : unknownLit(name, card, ii);
       ret = new PartFunc(sc, ii, f, args, holes);
       if(lit != null && (lit instanceof XQFunctionExpr ? ((XQFunctionExpr) f).annotations() :
-        ((FuncLit) lit).annotations()).contains(Ann.Q_UPDATING)) qc.updating();
+        ((FuncLit) lit).annotations()).contains(Annotation.UPDATING)) qc.updating();
     } else {
       final TypedFunc f = Functions.get(name, args, qc, sc, ii);
       if(f == null) {
         ret = null;
       } else {
-        if(f.ann.contains(Ann.Q_UPDATING)) qc.updating();
+        if(f.anns.contains(Annotation.UPDATING)) qc.updating();
         ret = f.fun;
       }
     }
