diff --git a/old_test.java b/new_test.java
index 79b4b80..96adff7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -32,5 +32,7 @@
     query(func + "/return/@type/data()", "xs:integer");
     query(func + "/return/@occurrence/data()", "");
 
-    query(_INSPECT_FUNCTION.args(" %db:f function() {()}") + "/annotation/@name = 'db:f'", "true");
+    // unknown annotations disappear
+    query("declare namespace x='x';" +
+      _INSPECT_FUNCTION.args(" %x:x function() {()}") + "/annotation", "");
   }
\ No newline at end of file
