diff --git a/old_product.java b/new_product.java
index 961c802..6b97b93 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,23 +5,7 @@ public static void mergeHadoopConfiguration(org.apache.hadoop.conf.Configuration
     // Load any Alluxio configuration parameters existing in the Hadoop configuration.
     for (Map.Entry<String, String> entry : source) {
       String propertyName = entry.getKey();
-      // TODO(gene): use a better way to enumerate every Alluxio configuration parameter
-      if (propertyName.startsWith("alluxio.")
-          || propertyName.equals(PropertyKey.S3N_ACCESS_KEY)
-          || propertyName.equals(PropertyKey.S3N_SECRET_KEY)
-          || propertyName.equals(PropertyKey.S3A_ACCESS_KEY)
-          || propertyName.equals(PropertyKey.S3A_SECRET_KEY)
-          || propertyName.equals(PropertyKey.GCS_ACCESS_KEY)
-          || propertyName.equals(PropertyKey.GCS_SECRET_KEY)
-          || propertyName.equals(PropertyKey.SWIFT_API_KEY)
-          || propertyName.equals(PropertyKey.SWIFT_AUTH_METHOD_KEY)
-          || propertyName.equals(PropertyKey.SWIFT_AUTH_PORT_KEY)
-          || propertyName.equals(PropertyKey.SWIFT_AUTH_URL_KEY)
-          || propertyName.equals(PropertyKey.SWIFT_PASSWORD_KEY)
-          || propertyName.equals(PropertyKey.SWIFT_TENANT_KEY)
-          || propertyName.equals(PropertyKey.SWIFT_USE_PUBLIC_URI_KEY)
-          || propertyName.equals(PropertyKey.SWIFT_USER_KEY)
-          || propertyName.equals(PropertyKey.SWIFT_SIMULATION)) {
+      if (PropertyKey.isValid(propertyName)) {
         alluxioConfProperties.put(propertyName, entry.getValue());
       }
     }
