diff --git a/old_product.java b/new_product.java
index b559518..40cf3d8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,10 @@
-@RpcMethod
+@Override
 	public CompletableFuture<RegistrationResponse> registerTaskExecutor(
 			final UUID resourceManagerLeaderId,
 			final String taskExecutorAddress,
 			final ResourceID taskExecutorResourceId,
-			final SlotReport slotReport) {
+			final SlotReport slotReport,
+			final Time timeout) {
 
 		if (Objects.equals(leaderSessionId, resourceManagerLeaderId)) {
 			CompletableFuture<TaskExecutorGateway> taskExecutorGatewayFuture = getRpcService().connect(taskExecutorAddress, TaskExecutorGateway.class);
