diff --git a/old_product.java b/new_product.java
index 0b80dbc..86e3edd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public StrBuilder replace(final int startIndex, int endIndex, final String replaceStr) {
         endIndex = validateRange(startIndex, endIndex);
-        int insertLen = (replaceStr == null ? 0 : replaceStr.length());
+        final int insertLen = (replaceStr == null ? 0 : replaceStr.length());
         replaceImpl(startIndex, endIndex, endIndex - startIndex, replaceStr, insertLen);
         return this;
     }
\ No newline at end of file
