diff --git a/old_product.java b/new_product.java
index 4935a61..53464b2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,7 @@
-public CriteriaQuery or(QueryCriterion... queryCriterions) {
-        criterion = criterion.and(criterionBuilder.or(queryCriterions));
+public CriteriaQuery or(QueryCriterion aCriterion) {
+        if (aCriterion == null) {
+            return this;
+        }
+        criterion = criterion.or(aCriterion);
         return this;
     }
\ No newline at end of file
