diff --git a/old_product.java b/new_product.java
index 9165010..1e63422 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,9 @@
             @NotBlank(message = "No id entered unable to get.")
             final String id
     ) throws GenieException {
-        LOG.debug("called");
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("called");
+        }
         final Command command = this.commandRepo.findOne(id);
         if (command != null) {
             return command;
