diff --git a/old_product.java b/new_product.java
index 8b99800..5dba288 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public void clear() {
-    for (Pair<ExecutionReference, ExecutableFlow> pair : queuedFlowMap.values()) {
+    for (final Pair<ExecutionReference, ExecutableFlow> pair : this.queuedFlowMap.values()) {
       dequeue(pair.getFirst().getExecId());
     }
   }
\ No newline at end of file
