diff --git a/old_product.java b/new_product.java
index f24ccc4..bac1ee4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,30 +1,10 @@
 public boolean isObserverInterested(Annotation... bindings)
    {
-      // TODO This logic needs to be in injectable
+      // Simply check that all event bindings specified by the observer are
+      // in the list provided.
+      List<Annotation> bindingsArray = Arrays.asList(bindings);
       boolean result = true;
-      // Check each binding specified by this observer against those provided
-      if (this.eventBindings.length > 0)
-      {
-         if ((bindings != null) && (bindings.length > 0))
-         {
-            List<Annotation> bindingsArray = Arrays.asList(bindings);
-            for (Annotation annotation : this.eventBindings)
-            {
-               int eventBindingIndex = bindingsArray.indexOf(annotation);
-               if (eventBindingIndex >= 0)
-               {
-                  //result = annotationsMatch(annotation, bindingsArray.get(eventBindingIndex));
-                  result = annotation.equals(bindingsArray.get(eventBindingIndex));
-               } else
-               {
-                  result = false;
-                  break;
-               }
-            }
-         } else
-         {
-            result = false;
-         }
-      }
+      if (!this.eventBindings.isEmpty())
+         result = bindingsArray.containsAll(this.eventBindings);
       return result;
    }
\ No newline at end of file
