diff --git a/old_product.java b/new_product.java
index 40e2f4b..31f63a4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
 public CorpusProcessingState getCorpusProcessingState(final Corpus corpus) {
-		String corpusId = IDHelper.getCorpusId(corpus, getAccountId());
-		return executeRequest(API_VERSION + corpusId + PROCESSING_STATE_PATH, null, CorpusProcessingState.class);
-	}
\ No newline at end of file
+    final String corpusId = IDHelper.getCorpusId(corpus, getAccountId());
+    return executeRequest(API_VERSION + corpusId + PROCESSING_STATE_PATH, null,
+        CorpusProcessingState.class);
+  }
\ No newline at end of file
