diff --git a/old_test.java b/new_test.java
index bdcdc1f..1afaf4b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,6 @@
 @Test
-	public void testGetCorpusProcessingState() {
-		CorpusProcessingState corpusProcessingState = service.getCorpusProcessingState(Corpus.IBM_RESEARCHERS);
-		Assert.assertNotNull(corpusProcessingState);
-	}
\ No newline at end of file
+  public void testGetCorpusProcessingState() {
+    final CorpusProcessingState corpusProcessingState =
+        service.getCorpusProcessingState(Corpus.IBM_RESEARCHERS);
+    Assert.assertNotNull(corpusProcessingState);
+  }
\ No newline at end of file
