diff --git a/old_test.java b/new_test.java
index 529f3ef..42893f5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,7 @@
 @Test
     public void testGet() {
         Random random = new Random(0L);
-        double[] bounds = Arrays.asList(random.nextInt(10) - 5, random.nextInt(10) - 5)
-            .stream().sorted().mapToDouble(x -> x)
+        double[] bounds = Stream.of(random.nextInt(10) - 5, random.nextInt(10) - 5).sorted().mapToDouble(x -> x)
             .toArray();
 
         double min = Math.min(bounds[0], bounds[1]);
