diff --git a/old_product.java b/new_product.java
index 7e7c8d9..4b6f1ea 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @Override public Double get() {
-        double result = generator().nextDouble() * (to - from) + from;
-        if (result > to)
-            result = to;
+        double res = generator().nextDouble() * (to - from) + from;
+        if (res > to)
+            res = to;
 
-        return result;
+        return res;
     }
\ No newline at end of file
