diff --git a/old_product.java b/new_product.java
index d46f5c3..0510840 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,7 +16,6 @@ void usage(boolean error) {
     out.println("  --base <url>            To specify the base URL to use");
     out.println("  --host <name>           To specify the hostname to send to");
     out.println("  --port <number>         To specify the port to use");
-    out.println("  --connector <name>      To specify the name of the rest collector");
     out.println("  --apikey <apikey>       To specify an API key for authentication");
     out.println("  --stream <id>           To specify the destination event stream of the");
     out.println("                          form <flow> or <flow>/<stream>.");
