diff --git a/old_test.java b/new_test.java
index dda70d5..3e7d265 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,12 @@
 @Test
   public void testUsage() throws Exception {
+    CConfiguration configuration = CConfiguration.create();
+    String port = Integer.toString(GatewayFastTestsSuite.getPort());
 
     // argument combinations that should return success
     String[][] goodArgsList = {
         { "--help" }, // print help
-        { "create", "--stream", "firststream" }, // create simple stream
+        { "create", "--stream", "teststream", "--host", hostname, "--port", port }, // create simple stream
 //        { "id", "--stream", "firststream" }, // fetch from simple stream
 //        { "send", "--stream", "firststream", "--body", "funk" }, // send event to stream
 //        { "fetch", "--stream", "firststream", "--consumer", "firstconsumer" }, // fetch from simple stream
