diff --git a/old_product.java b/new_product.java
index 14602e3..f3ccc0a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public static String removeParameter(String uriString,String paramName,String ch
         if (Validator.isNullOrEmpty(uriString)){
             return StringUtils.EMPTY;
         }
-        if (null == paramName){
+        if (Validator.isNullOrEmpty(paramName)){
             return uriString;
         }
         List<String> list = new ArrayList<String>();
