diff --git a/old_test.java b/new_test.java
index a778076..c50d43c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void removeParameter(){
-        uri = "http://www.feilong.com:8888/search.htm?keyword=中国&page=&categoryCode=2-5-3-11&label=TopSeller";
+        uriString = "http://www.feilong.com:8888/search.htm?keyword=中国&page=&categoryCode=2-5-3-11&label=TopSeller";
         String pageParamName = "label";
-        LOGGER.info(ParamUtil.removeParameter(uri, pageParamName, CharsetType.ISO_8859_1));
+        LOGGER.info(ParamUtil.removeParameter(uriString, pageParamName, CharsetType.UTF8));
     }
\ No newline at end of file
