diff --git a/old_product.java b/new_product.java
index 562ec21..e2ffdb3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
-void createDirectory(AlluxioURI path, CreateDirectoryOptions options)
+void createDirectory(AlluxioURI path, CreateDirectoryPOptions options)
       throws FileAlreadyExistsException, InvalidPathException, IOException, AlluxioException;
\ No newline at end of file
