diff --git a/old_test.java b/new_test.java
index 1d22261..9b9f3c2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void createDirectory() throws Exception {
     // create directory
-    createPath(mTree, TEST_URI, sDirectoryOptions);
+    createPath(mTree, TEST_URI, sDirectoryContext);
     assertTrue(mTree.inodePathExists(TEST_URI));
     InodeView test = getInodeByPath(mTree, TEST_URI);
     assertEquals(TEST_PATH, test.getName());
@@ -11,7 +11,7 @@
     assertEquals(TEST_DIR_MODE.toShort(), test.getMode());
 
     // create nested directory
-    createPath(mTree, NESTED_URI, sNestedDirectoryOptions);
+    createPath(mTree, NESTED_URI, sNestedDirectoryContext);
     assertTrue(mTree.inodePathExists(NESTED_URI));
     InodeView nested = getInodeByPath(mTree, NESTED_URI);
     assertEquals(TEST_PATH, nested.getName());
