diff --git a/old_test.java b/new_test.java
index b1467ab..b058540 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test
     public void testIsTraceEnabled() {
-        ListAppender appender = new ListAppender("List");
+        final ListAppender appender = new ListAppender("List");
         appender.start();
-        Logger root = Logger.getRootLogger();
+        final Logger root = Logger.getRootLogger();
         root.getLogger().addAppender(appender);
         root.setLevel(Level.INFO);
 
-        Logger tracer = Logger.getLogger("com.example.Tracer");
+        final Logger tracer = Logger.getLogger("com.example.Tracer");
         tracer.setLevel(Level.TRACE);
 
         assertTrue(tracer.isTraceEnabled());
