diff --git a/old_test.java b/new_test.java
index 6ccacd9..25dcb2f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,19 +1,19 @@
 @Test
   public void testIsHashCached() {
-    assertFalse(m_hash.isHashCached(HOSTNAME));
+    assertFalse(m_hash.isHashCached(CLUSTERNAME, HOSTNAME));
     String hash = m_hash.getHash(CLUSTERNAME, HOSTNAME);
     assertNotNull(hash);
-    assertTrue(m_hash.isHashCached(HOSTNAME));
+    assertTrue(m_hash.isHashCached(CLUSTERNAME, HOSTNAME));
 
     m_hash.invalidate(HOSTNAME);
-    assertFalse(m_hash.isHashCached(HOSTNAME));
+    assertFalse(m_hash.isHashCached(CLUSTERNAME, HOSTNAME));
     hash = m_hash.getHash(CLUSTERNAME, HOSTNAME);
     assertNotNull(hash);
-    assertTrue(m_hash.isHashCached(HOSTNAME));
+    assertTrue(m_hash.isHashCached(CLUSTERNAME, HOSTNAME));
 
     m_hash.invalidateAll();
-    assertFalse(m_hash.isHashCached(HOSTNAME));
+    assertFalse(m_hash.isHashCached(CLUSTERNAME, HOSTNAME));
     hash = m_hash.getHash(CLUSTERNAME, HOSTNAME);
     assertNotNull(hash);
-    assertTrue(m_hash.isHashCached(HOSTNAME));
+    assertTrue(m_hash.isHashCached(CLUSTERNAME, HOSTNAME));
   }
\ No newline at end of file
