diff --git a/old_product.java b/new_product.java
index a188b55..a57b242 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,12 @@
-public boolean isHashCached(String hostName) {
-    if (null == hostName) {
+public boolean isHashCached(String clusterName, String hostName) {
+    if (null == clusterName || null == hostName) {
       return false;
     }
 
-    return m_hashes.containsKey(hostName);
+    Map<String, String> clusterMapping = m_hashes.get(hostName);
+    if (null == clusterMapping) {
+      return false;
+    }
+
+    return clusterMapping.containsKey(clusterName);
   }
\ No newline at end of file
