diff --git a/old_test.java b/new_test.java
index 343b729..c0a9467 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,13 +8,13 @@
 
         DatasetBuilder<Integer, double[]> datasetBuilder = new LocalDatasetBuilder<>(data, parts);
 
-        NormalizationTrainer<Integer, double[]> standardizationTrainer = new NormalizationTrainer<>();
+        NormalizationTrainer<Integer, double[]> normalizationTrainer = new NormalizationTrainer<Integer, double[]>()
+            .withP(3);
 
-        NormalizationPreprocessor<Integer, double[]> preprocessor = standardizationTrainer.fit(
+        NormalizationPreprocessor<Integer, double[]> preprocessor = normalizationTrainer.fit(
             datasetBuilder,
             (k, v) -> v
         );
 
-        assertArrayEquals(new double[] {0, 4, 1}, preprocessor.getMin(), 1e-8);
-        assertArrayEquals(new double[] {4, 22, 300}, preprocessor.getMax(), 1e-8);
+        assertArrayEquals(new double[] {0.125, 0.99, 0.125}, preprocessor.apply(5, new double[] {1, 8, 1}), 1e-2);
     }
\ No newline at end of file
