diff --git a/old_test.java b/new_test.java
index c1a7e98..1f02f92 100644
--- a/old_test.java
+++ b/new_test.java
@@ -21,7 +21,7 @@
 			upper.set(x, y, sign*80);
 			lower.set(x, y, sign*90);
 
-			alg.processFeatureCandidate(15, 16, 100, sign > 0);
+			alg.processFeatureCandidate(15, 16, sign*100, sign > 0);
 
 			ScalePoint p = alg.getDetections().get(0);
 			// make sure it is close
@@ -39,7 +39,7 @@
 			lower.set(x, y, sign*80);
 
 			alg.detections.reset();
-			alg.processFeatureCandidate(15, 16, 100, sign > 0);
+			alg.processFeatureCandidate(15, 16, sign*100, sign > 0);
 			assertTrue(Math.abs(5 - p.scale) < 2);
 			assertTrue( 5 < p.scale );
 
