diff --git a/old_test.java b/new_test.java
index 697d621..86cf604 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,6 @@
 @Test
     public void testGetResetYesterdayAndToday(){
-        LOGGER.debug(JsonUtil.format(DateExtensionUtil.getResetYesterdayAndToday()));
+        assertArrayEquals(
+                        ConvertUtil.toArray(DateUtil.getFirstDateOfThisDay(DateUtil.addDay(NOW, -1)), DateUtil.getFirstDateOfThisDay(NOW)),
+                        DateExtensionUtil.getResetYesterdayAndToday());
     }
\ No newline at end of file
