diff --git a/old_product.java b/new_product.java
index 253182f..3ecbd69 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
 public static Date[] getResetYesterdayAndToday(){
-        Calendar calendar = CalendarUtil.resetDayBegin(new Date());
-        Date today = calendar.getTime();
+        Date today = DateUtil.getFirstDateOfThisDay(new Date());
         return new Date[] { DateUtil.addDay(today, -1), today };
     }
\ No newline at end of file
