diff --git a/old_product.java b/new_product.java
index f405615..b4ad109 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 public boolean process( T image ) {
 		tracker.process(image);
 
+		tick++;
 		inlierTracks.clear();
 
 		if( first ) {
@@ -21,7 +22,6 @@ public boolean process( T image ) {
 
 //			System.out.println("  num inliers = "+N+"  num dropped "+numDropped+" total active "+tracker.getActivePairs().size());
 		}
-		tick++;
 
 		return true;
 	}
\ No newline at end of file
