diff --git a/old_product.java b/new_product.java
index c476087..88f2aa5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,5 @@
 public static ListenerAdapter[] createListenerAdapters(EntryListener listener) {
-        // We only care about these reference event types for backward compatibility.
-        EntryEventType[] values = new EntryEventType[]{ADDED, REMOVED, EVICTED, UPDATED, EVICT_ALL, CLEAR_ALL};
+        EntryEventType[] values = new EntryEventType[]{ADDED, REMOVED, EVICTED, EXPIRED, UPDATED, EVICT_ALL, CLEAR_ALL};
         ListenerAdapter[] listenerAdapters = new ListenerAdapter[values.length];
         for (EntryEventType eventType : values) {
             listenerAdapters[eventType.ordinal()] = createListenerAdapter(eventType, listener);
