diff --git a/old_test.java b/new_test.java
index f8a412f..f527d78 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     BspEvent event = new PredicateLock(getStubProgressable());
     Thread signalThread = new SignalThread(event);
     signalThread.start();
-    event.waitForever();
+    event.waitForTimeoutOrFail(5 * 60_000);
     try {
       signalThread.join();
     } catch (InterruptedException e) {
