diff --git a/old_product.java b/new_product.java
index 0a82a17..af68996 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,9 +5,10 @@ protected SQLException markPossiblyBroken(SQLException e) {
 			state = "08999"; 
 		}
 
-		if ((sqlStateDBFailureCodes.contains(state) || connectionState.equals(ConnectionState.TERMINATE_ALL_CONNECTIONS)) && this.pool != null){
+		if (((sqlStateDBFailureCodes.contains(state) || connectionState.equals(ConnectionState.TERMINATE_ALL_CONNECTIONS)) && this.pool != null) && this.pool.getDbIsDown().compareAndSet(false, true) ){
 			logger.error("Database access problem. Killing off all remaining connections in the connection pool. SQL State = " + state);
 			this.pool.connectionStrategy.terminateAllConnections();
+			this.pool.poisonAndRepopulatePartitions();
 		}
 
 		// SQL-92 says:
