diff --git a/old_test.java b/new_test.java
index efe73d0..d3caf72 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,6 +7,7 @@
 		Assert.assertTrue(field.getBoolean(this.testClass));
 
 		// Test that a db fatal error will lead to the pool being instructed to terminate all connections (+ log)
+		expect(this.mockPool.getDbIsDown()).andReturn(new AtomicBoolean()).anyTimes();
 		this.mockPool.connectionStrategy.terminateAllConnections();
 		this.mockLogger.error((String)anyObject(), anyObject());
 		replay(this.mockPool);
