diff --git a/old_product.java b/new_product.java
index 268da71..f3caa7f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@ private DataClip data(final NewInput ni, final String dbname, final MainOptions
     // add document node
     final Context ctx = qc.context;
     if(ni.node != null) {
-      final MemData mdata = (MemData) ni.node.dbCopy(options).data();
+      final MemData mdata = (MemData) ni.node.dbNodeCopy(options).data();
       mdata.update(0, Data.DOC, ni.path);
       return new DataClip(mdata);
     }
