diff --git a/old_product.java b/new_product.java
index 99396fe..129f439 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,11 +9,12 @@
         IPolicyFactory pfactory = createPolicyFactory(pluginRegistry);
         IMetrics metrics = createMetrics(pluginRegistry);
         IDelegateFactory logFactory = createLoggerFactory(pluginRegistry);
+        IApiRequestPathParser pathParser = createRequestPathParser(pluginRegistry);
 
         List<IGatewayInitializer> initializers = createInitializers(pluginRegistry);
         for (IGatewayInitializer initializer : initializers) {
             initializer.initialize();
         }
 
-        return new EngineImpl(registry, pluginRegistry, componentRegistry, cfactory, pfactory, metrics, logFactory);
+        return new EngineImpl(registry, pluginRegistry, componentRegistry, cfactory, pfactory, metrics, logFactory, pathParser);
     }
\ No newline at end of file
