diff --git a/old_test.java b/new_test.java
index a203787..b7a1e3c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,9 +3,9 @@
         final Command command = new Command(
                 COMMAND_1_NAME,
                 COMMAND_1_USER,
+                COMMAND_1_VERSION,
                 CommandStatus.ACTIVE,
-                COMMAND_1_EXECUTABLE,
-                COMMAND_1_VERSION
+                COMMAND_1_EXECUTABLE
         );
         final String id = UUID.randomUUID().toString();
         command.setId(id);
