diff --git a/old_product.java b/new_product.java
index 7200b4c..6fe7d98 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ public Command createCommand(final Command command)
         if (command == null) {
             throw new GeniePreconditionException("No command entered to validate");
         }
-        command.validate();
+
         final HttpRequest request = BaseGenieClient.buildRequest(
                 Verb.POST,
                 BASE_CONFIG_COMMAND_REST_URL,
