diff --git a/old_product.java b/new_product.java
index d3d9ee7..ddae785 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,4 @@
 public static long roundDownTimeStampSeconds(long timestamp,
       int roundDownSec) throws IllegalStateException {
-    Preconditions.checkArgument(roundDownSec > 0 && roundDownSec <= 60,
-        "RoundDownSec must be > 0 and <=60");
-    Calendar cal = roundDownField(timestamp, Calendar.SECOND, roundDownSec);
-    cal.set(Calendar.MILLISECOND, 0);
-    return cal.getTimeInMillis();
+    return roundDownTimeStampSeconds(timestamp, roundDownSec, null);
   }
\ No newline at end of file
