diff --git a/old_product.java b/new_product.java
index 06159a7..0d90311 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,8 +3,5 @@
         @NotBlank(message = "No command id entered. Unable to update tags.") final String id,
         @NotEmpty(message = "No tags entered. Unable to update.") final Set<String> tags
     ) throws GenieException {
-        final CommandEntity commandEntity = this.findCommand(id);
-        final Set<TagEntity> newTags = this.createAndGetTagEntities(tags);
-        this.setFinalTags(newTags, commandEntity.getUniqueId(), commandEntity.getName());
-        commandEntity.setTags(newTags);
+        this.findCommand(id).setTags(this.createAndGetTagEntities(tags));
     }
\ No newline at end of file
