diff --git a/old_product.java b/new_product.java
index d768f83..97351aa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -28,6 +28,6 @@ private Item create(final QueryContext ctx) throws QueryException {
     }
 
     final Options opts = checkOptions(3, Q_OPTIONS, new Options(), ctx);
-    ctx.updates.add(new DBCreate(info, name, inputs, opts, ctx), ctx);
+    ctx.updates.add(new DBCreate(name, inputs, opts, ctx, info), ctx);
     return null;
   }
\ No newline at end of file
