diff --git a/old_test.java b/new_test.java
index 57010bc..69cda43 100644
--- a/old_test.java
+++ b/new_test.java
@@ -63,7 +63,7 @@
     // eventually drop database
     query(_DB_DROP.args(NAME));
 
-    // specify additional index options
+    // specify index options
     for(final boolean b : new boolean[] { false, true }) {
       query(_DB_CREATE.args(NAME, "()", "()", " map { 'updindex':" + b + "() }"));
       query(_DB_INFO.args(NAME) + "//updindex/text()", b);
@@ -86,7 +86,15 @@
       query(_DB_CREATE.args(NAME, "()", "()", " map { '" + k + "':'' }"));
     }
 
+    // specify parsing options
+    query(_DB_CREATE.args(NAME, " '<a> </a>'", "a.xml", " map { 'chop':true() }"));
+    query(_DB_OPEN.args(NAME), "<a/>");
+    query(_DB_CREATE.args(NAME, " '<a> </a>'", "a.xml", " map { 'chop':false() }"));
+    query(_DB_OPEN.args(NAME), "<a> </a>");
+
+    // specify unknown or invalid options
     error(_DB_CREATE.args(NAME, "()", "()", " map { 'xyz':'abc' }"), Err.BASX_OPTIONS);
     error(_DB_CREATE.args(NAME, "()", "()", " map { 'maxlen':-1 }"), Err.BASX_VALUE);
     error(_DB_CREATE.args(NAME, "()", "()", " map { 'maxlen':'a' }"), Err.BASX_VALUE);
+    error(_DB_CREATE.args(NAME, "()", "()", " map { 'textindex':'nope' }"), Err.BASX_VALUE);
   }
\ No newline at end of file
