diff --git a/old_product.java b/new_product.java
index 4f042e5..e9c4713 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,5 @@
 @Override
   public int create(final String path, final int mode) {
-    try {
-      if(!isFile(mode)) return -1;
-      int pre = insert(path, mode);
-      return (pre == -1) ? -1 : data.id(pre);
-    } catch(QueryException e) {
-      e.printStackTrace();
-      return -1;
-    }
+    if(!isFile(mode)) return -1;
+    return createFile(path, mode);
   }
\ No newline at end of file
