diff --git a/old_test.java b/new_test.java
index 8e8fe86..9d116fd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,14 +12,13 @@
 		double angle = UnitQuaternions.orientationAngle(cloud, cloud2, false);
 		assertEquals(angle, 0, 0.001);
 		
-		// Apply a 30 degree rotation to cloud 2
+		// Apply a 30 degree rotation to cloud
 		AxisAngle4d axis = new AxisAngle4d(new Vector3d(1,1,1), Math.PI / 6);
 		Matrix4d transform = new Matrix4d();
 		transform.set(axis);
 		
-		CalcPoint.transform(transform, cloud2);
+		CalcPoint.transform(transform, cloud);
 		angle = UnitQuaternions.orientationAngle(cloud, cloud2, false);
-		angle = Math.min(Math.abs(2 * Math.PI - angle), angle);
 		
 		// Test that angle was recovered
 		assertEquals(angle, Math.PI / 6, 0.001);
