diff --git a/old_test.java b/new_test.java
index 6c00bbc..3aa79ab 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,8 +3,7 @@
             throws SQLException
     {
         System.out.println("getType");
-        EPerson instance = new EPerson(context, row1);
         int expResult = Constants.EPERSON;
-        int result = instance.getType();
+        int result = eperson.getType();
         assertEquals("Should return Constants.EPERSON", expResult, result);
     }
\ No newline at end of file
