diff --git a/old_test.java b/new_test.java
index a8974b7..020094e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,14 @@
 @Test
-    public void testAddIdentifier() {
-        String type = "cpe";
-        String value = "cpe:/a:apache:struts:2.1.2";
-        String url = "http://somewhere";
-        Identifier expResult = new Identifier(type, value, url);
+    public void testAddVulnerableSoftwareIIdentifier() throws Exception {
+        CpeBuilder builder = new CpeBuilder();
+        Cpe cpe = builder.part(Part.APPLICATION).vendor("apache").product("struts").version("2.1.2").build();
+        CpeIdentifier id = new CpeIdentifier(cpe, Confidence.HIGHEST);
+
+        cpe = builder.part(Part.APPLICATION).vendor("apache").product("struts").version("2.1.2").build();
+        CpeIdentifier expResult = new CpeIdentifier(cpe, Confidence.HIGHEST);
 
         Dependency instance = new Dependency();
-        instance.addIdentifier(type, value, url);
-        assertEquals(1, instance.getIdentifiers().size());
-        assertTrue("Identifier doesn't contain expected result.", instance.getIdentifiers().contains(expResult));
+        instance.addVulnerableSoftwareIdentifier(id);
+        assertEquals(1, instance.getVulnerableSoftwareIdentifiers().size());
+        assertTrue("Identifier doesn't contain expected result.", instance.getVulnerableSoftwareIdentifiers().contains(expResult));
     }
\ No newline at end of file
