diff --git a/old_product.java b/new_product.java
index f6e1c60..4c2149a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,6 +2,6 @@ public SpeechModel getModel(final String name) {
     if (name == null)
       throw new IllegalArgumentException("name was not specified");
 
-    final Request request = RequestBuilder.get("/v1/models/" + name).build();
+    final Request request = RequestBuilder.get(String.format(PATH_MODEL, name)).build();
     return executeRequest(request, SpeechModel.class);
   }
\ No newline at end of file
