diff --git a/old_product.java b/new_product.java
index 11b67bd..a656ad0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,9 +4,8 @@ public TemporalAccessor parseBest(CharSequence text, TemporalQuery<?>... queries
         if (queries.length < 2) {
             throw new IllegalArgumentException("At least two queries must be specified");
         }
-        String str = text.toString();  // parsing whole String, so this makes sense
         try {
-            DateTimeBuilder builder = parseToBuilder(str).resolve();
+            DateTimeBuilder builder = parseToBuilder(text, null).resolve();
             for (TemporalQuery<?> query : queries) {
                 try {
                     return (TemporalAccessor) builder.query(query);
@@ -18,6 +17,6 @@ public TemporalAccessor parseBest(CharSequence text, TemporalQuery<?>... queries
         } catch (DateTimeParseException ex) {
             throw ex;
         } catch (RuntimeException ex) {
-            throw createError(str, ex);
+            throw createError(text, ex);
         }
     }
\ No newline at end of file
