diff --git a/old_test.java b/new_test.java
index 8e14aeb..790c6d2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,16 +10,16 @@
         localJobEntity.setJobStatus(JobStatus.INIT);
         Assert.assertTrue(localJobEntity.getStarted().isPresent());
         Assert.assertFalse(localJobEntity.getFinished().isPresent());
-        final Date started = localJobEntity.getStarted().orElseThrow(IllegalArgumentException::new);
+        final Instant started = localJobEntity.getStarted().orElseThrow(IllegalArgumentException::new);
 
         // Shouldn't affect finish time
         localJobEntity.setJobStatus(JobStatus.RUNNING);
-        Assert.assertThat(localJobEntity.getStarted().orElseGet(RandomSuppliers.DATE), Matchers.is(started));
+        Assert.assertThat(localJobEntity.getStarted().orElseGet(RandomSuppliers.INSTANT), Matchers.is(started));
         Assert.assertFalse(localJobEntity.getFinished().isPresent());
 
         // finish time is non-zero on completion
         localJobEntity.setJobStatus(JobStatus.SUCCEEDED);
-        Assert.assertThat(localJobEntity.getStarted().orElseGet(RandomSuppliers.DATE), Matchers.is(started));
+        Assert.assertThat(localJobEntity.getStarted().orElseGet(RandomSuppliers.INSTANT), Matchers.is(started));
         Assert.assertTrue(localJobEntity.getFinished().isPresent());
 
         final JobEntity jobEntity2 = new JobEntity();
