diff --git a/old_product.java b/new_product.java
index 2e98681..0913aa7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,8 +2,8 @@ void setJobStatus(@NotNull final JobStatus jobStatus) {
         this.status = jobStatus;
 
         if (jobStatus == JobStatus.INIT) {
-            this.setStarted(new Date());
+            this.setStarted(Instant.now());
         } else if (jobStatus.isFinished()) {
-            this.setFinished(new Date());
+            this.setFinished(Instant.now());
         }
     }
\ No newline at end of file
