diff --git a/old_product.java b/new_product.java
index 5f9deaf..aa583a2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
-public void setMaxSize(final int maxSize) throws TwilioRestException {
-        final Map<String, String> vars = new HashMap<String, String>();
-        final String maxSizeString = Integer.toString(maxSize);
+public void setMaxSize(int maxSize) throws TwilioRestException {
+        Map<String, String> vars = new HashMap<String, String>();
+        String maxSizeString = Integer.toString(maxSize);
         vars.put(Queue.MAX_SIZE, maxSizeString);
-        final TwilioRestResponse response = this.getClient().safeRequest(this.getResourceLocation(), "POST", vars);
+        TwilioRestResponse response = this.getClient().safeRequest(this.getResourceLocation(), "POST", vars);
         if (response.isError()) {
             throw new IllegalStateException("Response indicated error:" + response.getResponseText());
         }
