diff --git a/old_test.java b/new_test.java
index 67a3bb8..27c8a6e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,7 +18,7 @@
         mockK8s.setCreateResponse(resourceName, null);
         LogContext logContext = LogContext.zkWatch("///", topicName.toString());
         Async async = context.async();
-        topicOperator.onTopicCreated(logContext, topicName, ar -> {
+        topicOperator.onTopicCreated(logContext, topicName).setHandler(ar -> {
             assertSucceeded(context, ar);
             context.assertEquals(4, counter.get());
             mockK8s.assertExists(context, resourceName);
