diff --git a/old_product.java b/new_product.java
index 2ae3f7c..0a0a3e2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,40 +1,4 @@
-public static String generateXYZ(String command, Units units,
-            String distance, String feedRate, int dirX, int dirY, int dirZ) {
-        StringBuilder sb = new StringBuilder();
-
-      // Add units.
-        sb.append(unitCommand(units));
-
-        // Set command.
-        sb.append(command);
-
-        if (dirX != 0) {
-            sb.append("X");
-            if (dirX < 0) {
-                sb.append("-");
-            }
-            sb.append(distance);
-        }
-
-        if (dirY != 0) {
-            sb.append("Y");
-            if (dirY < 0) {
-                sb.append("-");
-            }
-            sb.append(distance);
-        }
-
-        if (dirZ != 0) {
-            sb.append("Z");
-            if (dirZ < 0) {
-                sb.append("-");
-            }
-            sb.append(distance);
-        }
-
-        if (feedRate != null) {
-            sb.append("F").append(feedRate);
-        }
-
-        return sb.toString();
+public static String generateJogCommand(String command, Units units,
+                                            double distance, double feedRate, int dirX, int dirY, int dirZ) {
+        return generateJogCommand(unitCommand(units) + command, units, distance, feedRate, dirX, dirY, dirZ, units);
     }
\ No newline at end of file
