diff --git a/old_product.java b/new_product.java
index 8bc2553..c146397 100644
--- a/old_product.java
+++ b/new_product.java
@@ -40,7 +40,7 @@
 			
 			Date newDate = encounter.getEncounterDatetime();
 			Location newLocation = encounter.getLocation();
-			for (Obs obs : encounter.getAllFlattenObs(true)) {
+			for (Obs obs : encounter.getAllFlattenedObs(true)) {
 				// if the date was changed
 				if (OpenmrsUtil.compare(originalDate, newDate) != 0
 				        && OpenmrsUtil.compare(obs.getObsDatetime(), originalDate) == 0) {
