diff --git a/old_test.java b/new_test.java
index e4c8566..4bf523e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -32,7 +32,7 @@
 		assertNotNull("save succeeds without error", es.saveEncounter(enc));
 		Date encounterDatetime = DateUtil.truncateToSeconds(enc.getEncounterDatetime());
 
-		for (Obs o : enc.getAllFlattenObs(false)) {
+		for (Obs o : enc.getAllFlattenedObs(false)) {
 			assertEquals("encounter datetime propagated", DateUtil.truncateToSeconds(o.getObsDatetime()), encounterDatetime);
 		}
 		
