diff --git a/old_test.java b/new_test.java
index ef338fd..1af501b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void test_parseBest_firstOption() throws Exception {
         DateTimeFormatter test = DateTimeFormatter.ofPattern("yyyy-MM[-dd]");
-        TemporalAccessor result = test.parseBest("2011-06-30", LocalDate.class, YearMonth.class);
+        TemporalAccessor result = test.parseBest("2011-06-30", LocalDate.FROM, YearMonth.FROM);
         assertEquals(result, LocalDate.of(2011, 6, 30));
     }
\ No newline at end of file
