diff --git a/old_product.java b/new_product.java
index e84f9a9..577bb8b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public TemporalAccessor parseBest(CharSequence text, Class<?>... types) {
+public TemporalAccessor parseBest(CharSequence text, TemporalQuery<?>... types) {
         Objects.requireNonNull(text, "text");
         Objects.requireNonNull(types, "types");
         if (types.length < 2) {
@@ -6,7 +6,7 @@ public TemporalAccessor parseBest(CharSequence text, Class<?>... types) {
         }
         try {
             DateTimeBuilder builder = parseToBuilder(text, null).resolve();
-            for (Class<?> type : types) {
+            for (TemporalQuery<?> type : types) {
                 try {
                     return (TemporalAccessor) builder.build(type);
                 } catch (RuntimeException ex) {
