diff --git a/old_test.java b/new_test.java
index 431b6ae..6ffb474 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test public void exists() {
-    query(_USER_EXISTS.args(UserText.ADMIN), "true");
-    query(_USER_EXISTS.args(NAME), "true");
-    query(_USER_EXISTS.args("unknown"), "false");
+    query(_USER_EXISTS.args(UserText.ADMIN), true);
+    query(_USER_EXISTS.args(NAME), true);
+    query(_USER_EXISTS.args("unknown"), false);
 
     // invalid name
     error(_USER_EXISTS.args(""), USER_NAME_X);
