diff --git a/old_product.java b/new_product.java
index ad5d919..f1f8f7b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 private boolean exists(final String db, final String path) throws IOException {
-    final WebDAVQuery query = new WebDAVQuery(_DB_EXISTS.args("$db", "$path"));
+    final WebDAVQuery query = new WebDAVQuery(_DB_EXISTS.args(" $db", " $path"));
     query.bind("db", db);
     query.bind("path", path);
     return execute(query).equals(Text.TRUE);
