diff --git a/old_product.java b/new_product.java
index 81d7bee..2a11632 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,11 +13,16 @@ protected boolean setupWorkingDirectory() {
     String topologyPackageDestination = Paths.get(
         topologyWorkingDirectory, "topology.tar.gz").toString();
 
-    return SchedulerUtils.setupWorkingDirectory(
-        topologyWorkingDirectory,
-        coreReleasePackageURI,
-        coreReleaseFileDestination,
-        topologyPackageURI,
-        topologyPackageDestination,
-        Context.verbose(config));
+    if (!SchedulerUtils.createOrCleanDirectory(topologyWorkingDirectory)) {
+      return false;
+    }
+
+    final boolean isVerbose = Context.verbose(config);
+    if (!SchedulerUtils.extractPackage(topologyWorkingDirectory, coreReleasePackageURI,
+        coreReleaseFileDestination, true, isVerbose)) {
+      return false;
+    }
+
+    return SchedulerUtils.extractPackage(topologyWorkingDirectory, topologyPackageURI,
+        topologyPackageDestination, true, isVerbose);
   }
\ No newline at end of file
