diff --git a/old_test.java b/new_test.java
index 1466639..ef16849 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
 		Assert.assertTrue(order.getActivatedBy() == null);
 		Assert.assertTrue(order.getDateActivated() == null);
 		
-		Context.getOrderService().activateOrder(order, provider);
+		Context.getOrderService().activateOrder(order, provider, null);
 		
 		order = Context.getOrderService().getOrder(10);
 		Assert.assertTrue(order.getActivatedBy() != null);
