diff --git a/old_product.java b/new_product.java
index 5161f31..97083ed 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,4 @@
-public static byte[] delete(final Data data, final TokenList res) {
-    for(final byte[] key : res) {
-      final IOFile file = data.meta.binary(string(key));
-      if(file == null || !file.delete()) return key;
-    }
-    return null;
+public static boolean delete(final Data data, final String res) {
+    final IOFile file = data.meta.binary(res);
+    return file != null && file.delete();
   }
\ No newline at end of file
