diff --git a/old_test.java b/new_test.java
index cef7291..89c41ad 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,9 @@
 @Test
   public void createFile() throws Exception {
-    doNothing().when(mFileSystemMasterClient)
-        .createFile(any(AlluxioURI.class), any(CreateFilePOptions.class));
     URIStatus status = new URIStatus(new FileInfo());
     AlluxioURI file = new AlluxioURI("/file");
-    GetStatusPOptions getStatusOptions = FileSystemOptions.getStatusDefaults(mConf)
-        .toBuilder().setLoadMetadataType(LoadMetadataPType.NEVER).build();
-    when(mFileSystemMasterClient.getStatus(file, getStatusOptions)).thenReturn(status);
+    when(mFileSystemMasterClient.createFile(any(AlluxioURI.class), any(CreateFilePOptions.class)))
+        .thenReturn(status);
     FileOutStream out = mFileSystem.createFile(file, CreateFilePOptions.getDefaultInstance());
     verify(mFileSystemMasterClient).createFile(file, FileSystemOptions.createFileDefaults(mConf)
             .toBuilder().mergeFrom(CreateFilePOptions.getDefaultInstance()).build());
