diff --git a/old_product.java b/new_product.java
index 761844d..a35d30f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@ public void createTestStream(final String streamName, final int numSegments)
         Preconditions.checkArgument(numSegments > 0);
 
         @Cleanup
-        StreamManager streamManager = StreamManager.create(this.controllerUri);
+        StreamManager streamManager = StreamManager.create(ClientConfig.builder().controllerURI(this.controllerUri).build());
         streamManager.createScope(this.scope);
         streamManager.createStream(this.scope, streamName,
                 StreamConfiguration.builder()
