diff --git a/old_test.java b/new_test.java
index 999d0e5..cff63ba 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,10 +6,14 @@
         String streamName = "Stream";
         int numSegments = 10;
 
+        ClientConfig clientConfig = ClientConfig.builder()
+                                                .controllerURI(URI.create(localPravega.getInProcPravegaCluster().getControllerURI()))
+                                                .credentials(new DefaultCredentials("1111_aaaa", "admin"))
+                                                .trustStore("../config/cert.pem")
+                                                .validateHostName(false)
+                                                .build();
         @Cleanup
-        StreamManager streamManager = StreamManager.create(URI.create(
-                localPravega.getInProcPravegaCluster().getControllerURI()
-        ));
+        StreamManager streamManager = StreamManager.create(clientConfig);
 
         streamManager.createScope(scope);
         Assert.assertTrue("Stream creation is not successful ",
@@ -18,6 +22,14 @@
                                    .streamName(streamName)
                                    .scalingPolicy(ScalingPolicy.fixed(numSegments))
                                    .build()));
-
         log.info("Created stream: " + streamName);
+
+        ClientFactory clientFactory = ClientFactory.withScope(scope, clientConfig);
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName,
+                new JavaSerializer<String>(),
+                EventWriterConfig.builder().build());
+        log.info("Created writer for stream: " + streamName);
+
+        writer.writeEvent("hello").get();
+        log.info("Wrote data to the stream");
     }
\ No newline at end of file
