diff --git a/old_test.java b/new_test.java
index 101eb26..3090142 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,7 @@
     assertThat(activeEntity.getConnectedClients().keySet(), contains(client));
 
     assertSuccess(activeEntity.invoke(client,
-        EhcacheEntityMessage.configure(new ServerSideConfigBuilder()
+        MESSAGE_FACTORY.configureCacheManager(new ServerSideConfigBuilder()
             .defaultResource("defaultServerResource")
             .sharedPool("primary", "serverResource1", 4, MemoryUnit.MEGABYTES)
             .sharedPool("secondary", "serverResource2", 8, MemoryUnit.MEGABYTES)
