diff --git a/old_product.java b/new_product.java
index 0dc1d7f..c255632 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,15 +7,6 @@ public void release(Throwable cause) {
 				this.cause = cause;
 			}
 
-			// Release all subpartitions
-			for (ResultSubpartition subpartition : subpartitions) {
-				try {
-					subpartition.release();
-				}
-				// Catch this in order to ensure that release is called on all subpartitions
-				catch (Throwable t) {
-					LOG.error("Error during release of result subpartition: " + t.getMessage(), t);
-				}
-			}
+			releaseInternal();
 		}
 	}
\ No newline at end of file
