diff --git a/old_test.java b/new_test.java
index ce12b93..d9981da 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-	public void testGetDialogs(){
-		List<Dialog> dialogs = service.getDialogs();
-		assertNotNull(dialogs);
-		assertFalse(dialogs.isEmpty());
-	}
\ No newline at end of file
+  public void testGetDialogs() {
+    final List<Dialog> dialogs = service.getDialogs();
+    assertNotNull(dialogs);
+    assertFalse(dialogs.isEmpty());
+  }
\ No newline at end of file
