diff --git a/old_product.java b/new_product.java
index 88e3664..4c7b367 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,9 @@
 public List<Dialog> getDialogs() {
-		HttpRequestBase request = Request.Get("/v1/dialogs").build();
+    final Request request = RequestBuilder.get(PATH_DIALOGS).build();
 
-		try {
-			HttpResponse response = execute(request);
-			JsonObject jsonObject = ResponseUtil.getJsonObject(response);
-			List<Dialog> dialogs = GsonSingleton.getGson().fromJson(
-					jsonObject.get("dialogs"), listDialogType);
-			return dialogs;
-		} catch (IOException e) {
-			throw new RuntimeException(e);
-		}
-	}
\ No newline at end of file
+    final Response response = execute(request);
+    final JsonObject jsonObject = ResponseUtil.getJsonObject(response);
+    final List<Dialog> dialogs =
+        GsonSingleton.getGson().fromJson(jsonObject.get("dialogs"), listDialogType);
+    return dialogs;
+  }
\ No newline at end of file
