diff --git a/old_test.java b/new_test.java
index a14745a..9c3f208 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testSizeGt() {
-        assertTrue(instance.sizeGt("id", 3).getQueryCriterions()
-                .contains(new SizeGtCriterion("id", 3)));
+    	assertEquals(new SizeGtCriterion("id", 3), instance.sizeGt("id", 3).getQueryCriterion());
     }
\ No newline at end of file
