diff --git a/old_test.java b/new_test.java
index f1fc242..2a8a5ee 100644
--- a/old_test.java
+++ b/new_test.java
@@ -54,10 +54,11 @@
 		}
 
 		// do the test
-		final MaterialDescriptor materialDescriptor = MaterialDescriptor.builderForQuery()
+		final MaterialDescriptor materialDescriptor = MaterialDescriptor.builderForCompleteDescriptor()
 				.productDescriptor(createProductDescriptor())
 				.warehouseId(WAREHOUSE_ID)
 				.date(t2)
+				.quantity(BigDecimal.ZERO) // doesn't matter
 				.build();
 		stockCandidateService.applyDeltaToMatchingLaterStockCandidates(
 				materialDescriptor,
