diff --git a/old_product.java b/new_product.java
index b7738fc..bd5cc3a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,6 @@
-public void singletonCollectionToGetInstance()
-    {
+public void singletonCollectionToGetInstance() {
         final Predicate<T> predicate = createMockPredicate(null);
         final Predicate<T> allPredicate = getPredicateInstance(
-                Collections.<Predicate<? super T>>singleton(predicate));
+                Collections.<Predicate<T>>singleton(predicate));
         Assert.assertSame("expected argument to be returned by getInstance()", predicate, allPredicate);
     }
\ No newline at end of file
