diff --git a/old_test.java b/new_test.java
index 336da2f..bdcaf82 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,14 @@
 @Test
     public void testPutAll() throws Exception {
-        Map<String, List<ParameterizedHeader>> all = new HashMap<String, List<ParameterizedHeader>>();
+        Map<String, List<ParameterizedHeader>> all = new HashMap<>();
 
-        List<ParameterizedHeader> fooValues = new ArrayList<ParameterizedHeader>();
+        List<ParameterizedHeader> fooValues = new ArrayList<>();
         fooValues.add(new ParameterizedHeader("foo1"));
         fooValues.add(new ParameterizedHeader("foo2"));
 
         all.put("foo", fooValues);
 
-        List<ParameterizedHeader> barValues = new ArrayList<ParameterizedHeader>();
+        List<ParameterizedHeader> barValues = new ArrayList<>();
         barValues.add(new ParameterizedHeader("bar1"));
         barValues.add(new ParameterizedHeader("bar2"));
 
