diff --git a/old_test.java b/new_test.java
index 1f9769d..5c52f9a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
     public void testIsBefore(){
-        boolean isBefore = DateUtil.isBefore(FROMSTRING, TOSTRING, DatePattern.COMMON_DATE);
-        LOGGER.debug(String.valueOf(isBefore));
+        assertEquals(true, DateUtil.isBefore(FROMSTRING, TOSTRING, DatePattern.COMMON_DATE));
+        assertEquals(false, DateUtil.isBefore("2011-05-01", "2011-04-01", DatePattern.COMMON_DATE));
     }
\ No newline at end of file
