diff --git a/old_product.java b/new_product.java
index 502a565..1e319de 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static boolean isBefore(String dateBefore,String dateAfter,String datePattern){
-        Date before = string2Date(dateBefore, datePattern);
-        return isBefore(before, dateAfter, datePattern);
+public static boolean isBefore(String dateString,String whenDateString,String datePattern){
+        Date before = string2Date(dateString, datePattern);
+        return isBefore(before, whenDateString, datePattern);
     }
\ No newline at end of file
