diff --git a/old_test.java b/new_test.java
index 2d48b9e..c4c2806 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testIsFalse() {
-        assertTrue(instance.isFalse("name").getQueryCriterions()
-                .contains(new EqCriterion("name", false)));
+    	assertEquals(new EqCriterion("name", false), instance.isFalse("name").getQueryCriterion());
     }
\ No newline at end of file
