diff --git a/old_test.java b/new_test.java
index b1ef9d5..ec61efb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,7 +10,9 @@
 
         DatasetBuilder<Integer, String[]> datasetBuilder = new LocalDatasetBuilder<>(data, parts);
 
-        StringEncoderTrainer<Integer, String[]> strEncoderTrainer = new StringEncoderTrainer<>();
+        StringEncoderTrainer<Integer, String[]> strEncoderTrainer = new StringEncoderTrainer<Integer, String[]>()
+            .encodeFeature(0)
+            .encodeFeature(1);
 
         StringEncoderPreprocessor<Integer, String[]> preprocessor = strEncoderTrainer.fit(
             datasetBuilder,
