diff --git a/old_test.java b/new_test.java
index 79c1568..8520685 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,6 +13,10 @@
         double serversProccessedAtLeast = 0.2;
 
         try {
+            ViewChangeListener listener = context.lookupStateless(ViewChangeListenerBean.class, ViewChangeListener.class);
+            
+            this.establishView(listener, NODES);
+            
             Stateless bean = context.lookupStateless(StatelessBean.class, Stateless.class);
 
             String node = bean.getNodeName();
@@ -32,10 +36,16 @@
             validateBalancing(bean, numberOfCalls, numberOfServers, serversProccessedAtLeast);
 
             this.stop(0);
+            
+            this.establishView(listener, NODES[1]);
+            
             node = bean.getNodeName();
             log.info("Node called : " + node);
 
             this.start(0);
+            
+            this.establishView(listener, NODES);
+            
             node = bean.getNodeName();
             log.info("Node called : " + node);
 
