diff --git a/old_product.java b/new_product.java
index dd8b060..eed2d62 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-private void loadbalance(String jndi, String deployment1, String deployment2) throws Exception {
+private void loadbalance(EJBDirectory directory, String deployment1, String deployment2) throws Exception {
         this.start(CONTAINER_1);
         this.deploy(CONTAINER_1, deployment1);
         this.start(CONTAINER_2);
@@ -12,7 +12,7 @@ private void loadbalance(String jndi, String deployment1, String deployment2) th
         double serversProccessedAtLeast = 0.2;
 
         try {
-            StatelessRemoteHome home = (StatelessRemoteHome) context.lookup(jndi);
+            StatelessRemoteHome home = directory.lookupHome(StatelessBean.class, StatelessRemoteHome.class);
             StatelessRemote bean = home.create();
 
             String node = bean.getNodeName();
