diff --git a/old_product.java b/new_product.java
index 0396da9..c3320ae 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 ChronoZonedDateTime<C> withMinute(int minute) {
         ChronoOffsetDateTime<C> newDT = dateTime.withMinute(minute);
         return (newDT == dateTime ? this :
-            resolve(newDT.getDateTime(), zone, dateTime, ZoneResolvers.retainOffset()));
+            resolve(newDT.getDateTime(), zoneId, dateTime, ZoneResolvers.retainOffset()));
     }
\ No newline at end of file
