diff --git a/old_product.java b/new_product.java
index d4542bd..4f42011 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public void clearPersistedFiles(List<Long> persistedFiles) {
     synchronized (mLock) {
       for (long persistedId : persistedFiles) {
-        mPersistedFilesInfo.remove(persistedId);
+        mPersistedUfsFingerprints.remove(persistedId);
       }
     }
   }
\ No newline at end of file
