diff --git a/old_test.java b/new_test.java
index 093b899..370b712 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,6 @@
   public void clearPersistedFiles() throws Exception {
     writeFileWithBlocks(1L, ImmutableList.of(2L, 3L));
     mManager.clearPersistedFiles(ImmutableList.of(1L));
-    FileDataManager.PersistedFilesInfo info = mManager.getPersistedFilesInfo();
+    FileDataManager.PersistedFilesInfo info = mManager.getPersistedUfsFingerprints();
     assertEquals(Collections.emptyList(), info.idList());
   }
\ No newline at end of file
