diff --git a/old_product.java b/new_product.java
index dc0734a..faecb59 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,22 +11,19 @@
         // validate parameters
         job.setJobStatus(JobStatus.INIT, "Initializing job");
 
+        if (StringUtils.isEmpty(job.getId())) {
+            job.setId(UUID.randomUUID().toString());
+        }
+
         // Validation successful. init state in DB - return if job already exists
         try {
-            final Job persistedJob = this.jobRepo.save(job);
             // if job can be launched, update the URIs
             final String hostName = this.netUtil.getHostName();
-            persistedJob.setHostName(hostName);
-            persistedJob.setOutputURI(
-                    getEndPoint(hostName)
-                            + "/" + this.jobDirPrefix
-                            + "/" + persistedJob.getId()
-            );
-            persistedJob.setKillURI(
-                    getEndPoint(hostName)
-                            + "/" + this.jobResourcePrefix
-                            + "/" + persistedJob.getId()
-            );
+            job.setHostName(hostName);
+            final String endpoint = getEndPoint(hostName);
+            job.setOutputURI(endpoint + "/" + this.jobDirPrefix + "/" + job.getId());
+            job.setKillURI(endpoint + "/" + this.jobResourcePrefix + "/" + job.getId());
+            final Job persistedJob = this.jobRepo.save(job);
 
             // increment number of submitted jobs as we have successfully
             // persisted it in the database.
