diff --git a/old_test.java b/new_test.java
index 6b08324..6112f97 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public void textEntry() {
     query(_ZIP_TEXT_ENTRY.args(ZIP, ENTRY1));
     query(_ZIP_TEXT_ENTRY.args(ZIP, ENTRY1, "US-ASCII"));
-    error(_ZIP_TEXT_ENTRY.args(ZIP, ENTRY1, "xyz"), Err.ZIP_FAIL);
+    error(_ZIP_TEXT_ENTRY.args(ZIP, ENTRY1, "xyz"), Err.ZIP_FAIL_X);
     // newlines are removed from the result..
     contains(_ZIP_TEXT_ENTRY.args(ZIP, ENTRY1), "aaboutab");
   }
\ No newline at end of file
