diff --git a/old_product.java b/new_product.java
index dc420c6..ceac622 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
 private Str textEntry(final QueryContext qc) throws QueryException {
-    final String enc = exprs.length < 3 ? null : string(checkStr(exprs[2], qc));
+    final String enc = exprs.length < 3 ? null : string(toToken(exprs[2], qc));
     final IO io = new IOContent(entry(qc));
     final boolean val = qc.context.options.get(MainOptions.CHECKSTRINGS);
     try {
       return Str.get(new NewlineInput(io).encoding(enc).validate(val).content());
     } catch(final IOException ex) {
-      throw ZIP_FAIL.get(info, ex);
+      throw ZIP_FAIL_X.get(info, ex);
     }
   }
\ No newline at end of file
