diff --git a/old_product.java b/new_product.java
index c2d74e8..7482d43 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,4 @@
 public static long roundDownTimeStampMinutes(long timestamp,
       int roundDownMins) throws IllegalStateException {
-    Preconditions.checkArgument(roundDownMins > 0 && roundDownMins <= 60,
-        "RoundDown must be > 0 and <=60");
-    Calendar cal = roundDownField(timestamp, Calendar.MINUTE, roundDownMins);
-    cal.set(Calendar.SECOND, 0);
-    cal.set(Calendar.MILLISECOND, 0);
-    return cal.getTimeInMillis();
-
+    return roundDownTimeStampMinutes(timestamp, roundDownMins, null);
   }
\ No newline at end of file
