diff --git a/old_product.java b/new_product.java
index d14c0f3..3c12dcb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,6 +5,7 @@ public InodeTree.CreatePathResult mkdir(AlluxioURI path, CreateDirectoryOptions
     synchronized (mInodeTree) {
       try {
         checkPermission(FileSystemAction.WRITE, path, true);
+        mMountTable.checkWritable(path);
         CreatePathOptions createPathOptions = new CreatePathOptions.Builder(MasterContext.getConf())
             .setAllowExists(options.isAllowExists())
             .setDirectory(true)
