diff --git a/old_product.java b/new_product.java
index af44155..c2937dd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 boolean hardDelete() throws StoreException {
-    if (index.getCurrentEndOffset().compareTo(log.getStartOffset()) > 0) {
+    if (index.getCurrentEndOffset().compareTo(index.getStartOffset()) > 0) {
       final Timer.Context context = metrics.hardDeleteTime.time();
       try {
         FindInfo info =
