diff --git a/old_test.java b/new_test.java
index a03c09f..4e28910 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,8 +8,7 @@
     // IndexSegment still uses real time so advance time so that it goes 2 days past the real time.
     advanceTime(SystemTime.getInstance().milliseconds() + 2 * Time.MsPerSec * Time.SecsPerDay);
     assertTrue("Hard delete did not do any work", index.hardDeleter.hardDelete());
-
-    long expectedProgress = log.getDifference(logOrder.lastKey(), new Offset(log.getFirstSegment().getName(), 0));
+    long expectedProgress = index.getAbsolutePositionInLogForOffset(logOrder.lastKey());
     assertEquals("Hard delete did not make expected progress", expectedProgress, index.hardDeleter.getProgress());
     verifyEntriesForHardDeletes(deletedKeys);
   }
\ No newline at end of file
