diff --git a/old_test.java b/new_test.java
index 046f962..15ce933 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
         logger.info("coleman_liau_index");
         String strText = TEST_STRING;
         double expResult = 6.8;
-        double result = ReadabilityStatistics.coleman_liau_index(strText);
+        double result = ReadabilityStatistics.colemanLiauIndex(strText);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
