diff --git a/old_product.java b/new_product.java
index 79ca32b..04573bb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-public static double coleman_liau_index(String strText) {
-        strText = clean_text(strText);
-        int intWordCount = word_count(strText);
-        return PHPfunctions.round( ( (5.89 * (letter_count(strText) / (double)intWordCount)) - (0.3 * (sentence_count(strText) / (double)intWordCount)) - 15.8 ), 1);
+public static double colemanLiauIndex(String strText) {
+        strText = cleanText(strText);
+        int intWordCount = wordCount(strText);
+        return PHPfunctions.round( ( (5.89 * (letterCount(strText) / (double)intWordCount)) - (0.3 * (sentenceCount(strText) / (double)intWordCount)) - 15.8 ), 1);
     }
\ No newline at end of file
