diff --git a/old_test.java b/new_test.java
index 170e8c9..022b079 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,6 +9,7 @@
 		order.setStartDate(cal.getTime());
 		order.setAutoExpireDate(new Date());
 		order.setDrug(Context.getConceptService().getDrug(3));
+		order.setOrderNumber("orderNumber");
 		
 		Errors errors = new BindException(order, "order");
 		new DrugOrderValidator().validate(order, errors);
