diff --git a/old_product.java b/new_product.java
index d39c3be..2dca9b2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public void removeAllCommands() throws GenieException {
+public void removeAllCommands() throws GeniePreconditionException {
         if (this.commands != null) {
             final List<Command> locCommands = new ArrayList<>();
             locCommands.addAll(this.commands);
