diff --git a/old_product.java b/new_product.java
index 5994df3..e8a7b5b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,7 +12,7 @@ public ServiceCall<List<TranslationModel>> getModels(final Boolean showDefault,
       requestBuilder.withQuery(DEFAULT, showDefault);
 
     ResponseConverter<List<TranslationModel>> converter =
-        ResponseConverterUtils.getGenericObject(TYPE_LIST_TRANSLATION_MODEL, "models");
+        ResponseConverterUtils.getGenericObject(TYPE_LIST_TRANSLATION_MODEL, MODELS);
 
     return createServiceCall(requestBuilder.build(), converter);
   }
\ No newline at end of file
