diff --git a/old_product.java b/new_product.java
index 58fdc57..017a97a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,20 @@
-public DateTimeBuilder parseToBuilder(CharSequence text, ParsePosition position) {
-        Objects.requireNonNull(text, "text");
-        Objects.requireNonNull(position, "position");
-        DateTimeParseContext context = new DateTimeParseContext(this);
-        int pos = position.getIndex();
-        pos = printerParser.parse(context, text, pos);
-        if (pos < 0) {
-            position.setErrorIndex(~pos);
-            return null;
+private DateTimeBuilder parseToBuilder(final CharSequence text, final ParsePosition position) {
+        ParsePosition pos = (position != null ? position : new ParsePosition(0));
+        Parsed result = parseUnresolved0(text, pos);
+        if (result == null || pos.getErrorIndex() >= 0 || (position == null && pos.getIndex() < text.length())) {
+            String abbr = "";
+            if (text.length() > 64) {
+                abbr = text.subSequence(0, 64).toString() + "...";
+            } else {
+                abbr = text.toString();
+            }
+            if (pos.getErrorIndex() >= 0) {
+                throw new DateTimeParseException("Text '" + abbr + "' could not be parsed at index " +
+                        pos.getErrorIndex(), text, pos.getErrorIndex());
+            } else {
+                throw new DateTimeParseException("Text '" + abbr + "' could not be parsed, unparsed text found at index " +
+                        pos.getIndex(), text, pos.getIndex());
+            }
         }
-        position.setIndex(pos);
-        return context.toBuilder();
+        return result.resolveFields().toBuilder();
     }
\ No newline at end of file
