diff --git a/old_test.java b/new_test.java
index f2e8da7..6fc2d29 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,5 +2,6 @@
     public void testGetWorkflowGroup() {
         //null by default
         int step = 1;
-        assertThat("testGetWorkflowGroup 0", collectionService.getWorkflowGroup(collection, step), nullValue());
+        assertThat("testGetWorkflowGroup 0", collectionService.getWorkflowGroup(context, collection, step),
+                nullValue());
     }
\ No newline at end of file
