diff --git a/old_product.java b/new_product.java
index f2b560f..2f4f5e2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-public void setItemSimilarity(
-			Class<? extends Similarity<? super SparseVector>> itemSimilarity) {
-		this.itemSimilarity = itemSimilarity;
+@Override
+	public void setName(String name) {
+		super.setName(name);
+		knn.setName(name);
 	}
\ No newline at end of file
