diff --git a/old_product.java b/new_product.java
index dd9d9bd..870404f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,8 +3,13 @@ public RunTable copy (String name)
         RunTable clone = new RunTable(name, orientation, width, height);
 
         for (int i = 0; i < sequences.length; i++) {
-            BasicRunSequence seq = (BasicRunSequence) getSequence(i);
-            clone.sequences[i] = new BasicRunSequence(seq);
+            short[] seq = getSequence(i);
+
+            if (seq != null) {
+                short[] rle = new short[seq.length];
+                System.arraycopy(seq, 0, rle, 0, seq.length);
+                clone.sequences[i] = rle;
+            }
         }
 
         return clone;
