diff --git a/old_test.java b/new_test.java
index bfe7ba0..b2b4b34 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-  public final void testGetId() {
+  public void testGetId() {
     ShowMissingDiskStoresFunction smdsFunc = new ShowMissingDiskStoresFunction();
     assertEquals(ShowMissingDiskStoresFunction.class.getName(), smdsFunc.getId());
   }
\ No newline at end of file
