diff --git a/old_product.java b/new_product.java
index 8000dea..672a722 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public void insertGelfMessage(GELFMessage message) throws Exception {
         // Check if all required parameters are set.
-        if (message.shortMessage == null || message.shortMessage.length() == 0 || message.host == null || message.host.length() == 0) {
+        if (message.getShortMessage() == null || message.getShortMessage().length() == 0 || message.getHost() == null || message.getHost().length() == 0) {
             throw new Exception("Missing GELF message parameters. short_message and host are required.");
         }
         DBCollection coll = this.getMessagesColl();
@@ -8,14 +8,24 @@ public void insertGelfMessage(GELFMessage message) throws Exception {
         BasicDBObject dbObj = new BasicDBObject();
 
         dbObj.put("gelf", true);
-        dbObj.put("message", message.shortMessage);
-        dbObj.put("full_message", message.fullMessage);
-        dbObj.put("type", message.type);
-        dbObj.put("file", message.file);
-        dbObj.put("line", message.line);
-        dbObj.put("host", message.host);
+        dbObj.put("message", message.getShortMessage());
+        dbObj.put("full_message", message.getFullMessage());
+        dbObj.put("file", message.getFile());
+        dbObj.put("line", message.getLine());
+        dbObj.put("host", message.getHost());
         dbObj.put("facility", null);
-        dbObj.put("level", message.level);
+        dbObj.put("level", message.getLevel());
+
+        // Add additional fields.
+        Map<String,String> additionalFields = message.getAdditionalData();
+        Set<String> set = additionalFields.keySet();
+        Iterator<String> iter = set.iterator();
+        while(iter.hasNext()) {
+            String key = iter.next();
+            String value = additionalFields.get(key);
+            dbObj.put(key, value);
+        }
+
         dbObj.put("created_at", (int) (System.currentTimeMillis()/1000));
         // Documents in capped collections cannot grow so we have to do that now and cannot just add 'deleted => true' later.
         dbObj.put("deleted", false);
