diff --git a/old_test.java b/new_test.java
index f28d995..44c7b1d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,13 @@
 @Test
     public void testInsertGelfMessage() throws Exception {
         GELFMessage message = new GELFMessage();
-        message.shortMessage = "gelftest";
-        message.fullMessage = "full gelftest\nstuff";
-        message.level = 1;
-        message.type = 8;
-        message.host = "junit-test";
-        message.file = "junit-testfile";
-        message.line = 9001;
+        message.setShortMessage("gelftest");
+        message.setFullMessage("full gelftest\nstuff");
+        message.setLevel(1);
+        message.setHost("junit-test");
+        message.setFile("junit-testfile");
+        message.setLine(9001);
+        message.addAdditionalData("something", "yepp");
 
         // Insert the message.
         MongoBridge instance = new MongoBridge();
@@ -26,4 +26,5 @@
         assertEquals(res.get("host"), "junit-test");
         assertEquals(res.get("file"), "junit-testfile");
         assertEquals(res.get("line"), 9001);
+        assertEquals(res.get("something"), "yepp");
     }
\ No newline at end of file
