diff --git a/old_product.java b/new_product.java
index 5287f93..52fec80 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
 UsageHistogramBean getUsage(String organizationId, String serviceId, String version,
-            UsageHistogramIntervalType interval, DateTime from, DateTime to);
\ No newline at end of file
+            HistogramIntervalType interval, DateTime from, DateTime to);
\ No newline at end of file
