diff --git a/old_product.java b/new_product.java
index e7dbbd0..5d8f0e2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
-@RequiresSession
-  public List<AlertGroupEntity> findAllGroups() {
+public List<AlertGroupEntity> findAllGroups() {
     TypedQuery<AlertGroupEntity> query = entityManagerProvider.get().createNamedQuery(
         "AlertGroupEntity.findAll", AlertGroupEntity.class);
 
