diff --git a/old_product.java b/new_product.java
index 2ce78bf..d2f7550 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,18 +1,18 @@
 public Documents listDocuments(final Corpus corpus, final Map<String, Object> parameters) {
-		String corpusId = IDHelper.getCorpusId(corpus, getAccountId());
+    final String corpusId = IDHelper.getCorpusId(corpus, getAccountId());
 
-		Map<String, Object> queryParameters = new HashMap<String, Object>();
-		String[] queryParams = new String[] { CURSOR, LIMIT };
-		for (String param : queryParams) {
-			if (parameters.containsKey(param)) {
-				queryParameters.put(param, parameters.get(param));
-			}
-		}
-		if (parameters.get(QUERY) != null) {
-			// TODO we may need to work in the query format,for now we do expect
-			// the query parameter String formatted as documented in Concept Insights.
-			queryParameters.put(QUERY, parameters.get(QUERY));
-		}
+    final Map<String, Object> queryParameters = new HashMap<String, Object>();
+    final String[] queryParams = new String[] {CURSOR, LIMIT};
+    for (final String param : queryParams) {
+      if (parameters.containsKey(param)) {
+        queryParameters.put(param, parameters.get(param));
+      }
+    }
+    if (parameters.get(QUERY) != null) {
+      // TODO we may need to work in the query format,for now we do expect
+      // the query parameter String formatted as documented in Concept Insights.
+      queryParameters.put(QUERY, parameters.get(QUERY));
+    }
 
-		return executeRequest(API_VERSION + corpusId + DOCUMENTS_PATH, queryParameters, Documents.class);
-	}
\ No newline at end of file
+    return executeRequest(API_VERSION + corpusId + DOCUMENTS_PATH, queryParameters, Documents.class);
+  }
\ No newline at end of file
