diff --git a/old_test.java b/new_test.java
index 6fc9b48..ff12849 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
-	public void testListDocuments() {
-		Map<String, Object> params = new HashMap<String, Object>();
-		params.put(ConceptInsights.CURSOR, 0);
-		params.put(ConceptInsights.LIMIT, 20);
+  public void testListDocuments() {
+    final Map<String, Object> params = new HashMap<String, Object>();
+    params.put(ConceptInsights.CURSOR, 0);
+    params.put(ConceptInsights.LIMIT, 20);
 
-		Documents documents = service.listDocuments(Corpus.TED_TALKS, params);
-		Assert.assertNotNull(documents);
-	}
\ No newline at end of file
+    final Documents documents = service.listDocuments(Corpus.TED_TALKS, params);
+    Assert.assertNotNull(documents);
+  }
\ No newline at end of file
