diff --git a/old_product.java b/new_product.java
index 3ab86b7..d1680d1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public synchronized long reinitializeFile(String path, long blockSizeBytes, long
       throws InvalidPathException, FileDoesNotExistException {
     long fileId = mFileSystemMaster.getFileId(new TachyonURI(path));
     FileInfo fileInfo = mFileSystemMaster.getFileInfo(fileId);
-    if (!fileInfo.isIsCompleted() || mFileSystemMaster.getLostFiles().contains(fileId)) {
+    if (!fileInfo.isCompleted() || mFileSystemMaster.getLostFiles().contains(fileId)) {
       LOG.info("Recreate the file {} with block size of {} bytes", path, blockSizeBytes);
       return mFileSystemMaster.reinitializeFile(new TachyonURI(path), blockSizeBytes, ttl);
     }
