diff --git a/old_product.java b/new_product.java
index e0b3687..961c802 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,21 +7,21 @@ public static void mergeHadoopConfiguration(org.apache.hadoop.conf.Configuration
       String propertyName = entry.getKey();
       // TODO(gene): use a better way to enumerate every Alluxio configuration parameter
       if (propertyName.startsWith("alluxio.")
-          || propertyName.equals(Constants.S3N_ACCESS_KEY)
-          || propertyName.equals(Constants.S3N_SECRET_KEY)
-          || propertyName.equals(Constants.S3A_ACCESS_KEY)
-          || propertyName.equals(Constants.S3A_SECRET_KEY)
-          || propertyName.equals(Constants.GCS_ACCESS_KEY)
-          || propertyName.equals(Constants.GCS_SECRET_KEY)
-          || propertyName.equals(Constants.SWIFT_API_KEY)
-          || propertyName.equals(Constants.SWIFT_AUTH_METHOD_KEY)
-          || propertyName.equals(Constants.SWIFT_AUTH_PORT_KEY)
-          || propertyName.equals(Constants.SWIFT_AUTH_URL_KEY)
-          || propertyName.equals(Constants.SWIFT_PASSWORD_KEY)
-          || propertyName.equals(Constants.SWIFT_TENANT_KEY)
-          || propertyName.equals(Constants.SWIFT_USE_PUBLIC_URI_KEY)
-          || propertyName.equals(Constants.SWIFT_USER_KEY)
-          || propertyName.equals(Constants.SWIFT_SIMULATION)) {
+          || propertyName.equals(PropertyKey.S3N_ACCESS_KEY)
+          || propertyName.equals(PropertyKey.S3N_SECRET_KEY)
+          || propertyName.equals(PropertyKey.S3A_ACCESS_KEY)
+          || propertyName.equals(PropertyKey.S3A_SECRET_KEY)
+          || propertyName.equals(PropertyKey.GCS_ACCESS_KEY)
+          || propertyName.equals(PropertyKey.GCS_SECRET_KEY)
+          || propertyName.equals(PropertyKey.SWIFT_API_KEY)
+          || propertyName.equals(PropertyKey.SWIFT_AUTH_METHOD_KEY)
+          || propertyName.equals(PropertyKey.SWIFT_AUTH_PORT_KEY)
+          || propertyName.equals(PropertyKey.SWIFT_AUTH_URL_KEY)
+          || propertyName.equals(PropertyKey.SWIFT_PASSWORD_KEY)
+          || propertyName.equals(PropertyKey.SWIFT_TENANT_KEY)
+          || propertyName.equals(PropertyKey.SWIFT_USE_PUBLIC_URI_KEY)
+          || propertyName.equals(PropertyKey.SWIFT_USER_KEY)
+          || propertyName.equals(PropertyKey.SWIFT_SIMULATION)) {
         alluxioConfProperties.put(propertyName, entry.getValue());
       }
     }
