diff --git a/old_test.java b/new_test.java
index b6ec28b..9b68cbb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,10 +9,10 @@
     // This hadoop config will not be loaded into Alluxio configuration.
     hadoopConfig.set("hadoop.config.parameter", "hadoop config value");
 
-    Map<String, String> before = Configuration.toMap();
+    long beforeSize = Configuration.toMap().size();
     ConfUtils.mergeHadoopConfiguration(hadoopConfig);
-    Map<String, String> after = Configuration.toMap();
-    Assert.assertEquals(before.size() + 3, after.size());
+    long afterSize = Configuration.toMap().size();
+    Assert.assertEquals(beforeSize + 3, afterSize);
     Assert.assertEquals(TEST_S3_ACCCES_KEY, Configuration.get(Constants.S3N_ACCESS_KEY));
     Assert.assertEquals(TEST_S3_SECRET_KEY, Configuration.get(Constants.S3N_SECRET_KEY));
     Assert.assertEquals(TEST_ALLUXIO_VALUE, Configuration.get(TEST_ALLUXIO_PROPERTY));
