diff --git a/old_product.java b/new_product.java
index b6f3e88..7067fe5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
-void createDirectory(TachyonURI path)
-      throws FileAlreadyExistsException, InvalidPathException, IOException, TachyonException;
\ No newline at end of file
+void createDirectory(AlluxioURI path)
+      throws FileAlreadyExistsException, InvalidPathException, IOException, AlluxioException;
\ No newline at end of file
