diff --git a/old_test.java b/new_test.java
index b15f9bc..82bf224 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void createDirectoryTest() throws Exception {
-    TachyonURI dir = new TachyonURI("/dir");
+    AlluxioURI dir = new AlluxioURI("/dir");
     CreateDirectoryOptions createDirectoryOptions = CreateDirectoryOptions.defaults();
     Mockito.doNothing().when(mFileSystemMasterClient).createDirectory(dir, createDirectoryOptions);
     mFileSystem.createDirectory(dir, createDirectoryOptions);
