diff --git a/old_test.java b/new_test.java
index f9bbbcf..6132de7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
 	public void purgeOrderType_shouldNotAllowDeletingAnOrderTypeThatIsInUse() {
 		OrderType orderType = orderService.getOrderType(1);
 		assertNotNull(orderType);
-		expectedException.expect(APIException.class);
+		expectedException.expect(CannotDeleteOrderPropertyInUseException.class);
 		expectedException.expectMessage("Order.type.cannot.delete");
 		orderService.purgeOrderType(orderType);
 	}
\ No newline at end of file
