diff --git a/old_product.java b/new_product.java
index 8efb86d..b6ecfe3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @Override
 	public void purgeOrderType(OrderType orderType) {
 		if (dao.isOrderTypeInUse(orderType)) {
-			throw new APIException("Order.type.cannot.delete");
+			throw CannotDeleteOrderPropertyInUseException.withProperty("type");
 		}
 		dao.purgeOrderType(orderType);
 	}
\ No newline at end of file
