diff --git a/old_product.java b/new_product.java
index 74a1ea0..ba10803 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@
     final Data d = it.data();
     if(d != null && !d.inMemory()) {
       // convert database node to main memory data instance
-      it = ((ANode) it).dbCopy(context.context.prop);
+      it = ((ANode) it).dbCopy(context.context.options);
     } else if(it instanceof FItem) {
       throw SessionErrors.functionItem();
     }
