diff --git a/old_product.java b/new_product.java
index dcfa179..b7d37f1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,8 +15,7 @@ public void start() throws IOException, YarnException {
     mYarnClient.start();
 
     // Register with ResourceManager
-    String hostname = NetworkAddressUtils.getLocalHostName(
-        Configuration.createServerConf());
+    String hostname = NetworkAddressUtils.getLocalHostName();
     mRMClient.registerApplicationMaster(hostname, 0 /* port */, "" /* tracking url */);
     LOG.info("ApplicationMaster registered");
   }
\ No newline at end of file
