diff --git a/old_test.java b/new_test.java
index 830ec76..6beacfc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
   public void startTest() throws Exception {
-    String hostname = NetworkAddressUtils.getLocalHostName(new Configuration());
+    String hostname = NetworkAddressUtils.getLocalHostName();
     Mockito.verify(mRMClient).registerApplicationMaster(hostname, 0, "");
   }
\ No newline at end of file
