diff --git a/old_test.java b/new_test.java
index b7f0386..4ab96ff 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,7 +17,7 @@
 		final SlotStatus slotStatus = new SlotStatus(slotId, resourceProfile, jobId, allocationId);
 		final SlotReport slotReport = new SlotReport(slotStatus);
 
-		try (SlotManager slotManager = createSlotManager(resourceManagerId, resourceManagerActions)) {
+		try (SlotManagerImpl slotManager = createSlotManager(resourceManagerId, resourceManagerActions)) {
 
 			slotManager.registerTaskManager(
 				taskExecutorConnection,
