diff --git a/old_product.java b/new_product.java
index dc207ac..d854c46 100644
--- a/old_product.java
+++ b/new_product.java
@@ -45,7 +45,7 @@ public RealmAsyncTask executeTransactionAsync(final Transaction transaction, fin
                         if (bgRealm.isInTransaction()) {
                             bgRealm.cancelTransaction();
                         } else if (exception[0] != null) {
-                            RealmLog.w("Could not cancel transaction, not currently in a transaction.");
+                            RealmLog.warn("Could not cancel transaction, not currently in a transaction.");
                         }
                         bgRealm.close();
                     }
