diff --git a/old_test.java b/new_test.java
index b8eedbc..7da40cc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,6 +7,7 @@
 	assertFalse("This ClientHello message contains no modification",
 		TlsContextAnalyzer.containsModifiableVariableModification(ch));
 	ModifiableInteger length = new ModifiableInteger();
+	length.setOriginalValue(2);
 	length.setModification(IntegerModificationFactory.add(1));
 	ch.setCipherSuiteLength(length);
 	assertTrue("This ClientHello message contains a modification in the CipherSuite Length variable",
