diff --git a/old_product.java b/new_product.java
index eec982c..36065e0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ private static boolean containsModifiableVariableModification(ModifiableVariable
 	try {
 	    field.setAccessible(true);
 	    ModifiableVariable mv = (ModifiableVariable) field.get(object);
-	    return (mv != null && mv.getModification() != null);
+	    return (mv != null && mv.getModification() != null && mv.isOriginalValueModified());
 	} catch (IllegalAccessException | IllegalArgumentException ex) {
 	    throw new ModificationException(ex.getLocalizedMessage(), ex);
 	}
