diff --git a/old_test.java b/new_test.java
index 08982a1..563de77 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
-@Test(expected=UnsupportedFormatException.class)
+@Test(expected=IllegalArgumentException.class)
 	public void testCreateThumbnail_IOSII_Transcoding_Bmp_Gif() throws IOException
 	{
 		/*
@@ -15,10 +15,9 @@
 			Thumbnailator.createThumbnail(is, os, "gif", 50, 50);
 			fail();
 		}
-		catch (UnsupportedFormatException e)
+		catch (IllegalArgumentException e)
 		{
-			assertEquals("No suitable ImageWriter found for gif.", e.getMessage());
-			assertEquals("gif", e.getFormatName());
+			assertTrue(e.getMessage().contains("gif"));
 			throw e;
 		}
 	}
\ No newline at end of file
