diff --git a/old_product.java b/new_product.java
index 0639449..650ee89 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,21 +16,9 @@ public static void createThumbnail(
 		{
 			throw new NullPointerException("OutputStream is null.");
 		}
-		
-		ThumbnailParameter param = 
-			new ThumbnailParameter(
-					new Dimension(width, height),
-					null,
-					true,
-					format,
-					ThumbnailParameter.DEFAULT_FORMAT_TYPE,
-					ThumbnailParameter.DEFAULT_QUALITY,
-					ThumbnailParameter.DEFAULT_IMAGE_TYPE,
-					null,
-					DefaultResizerFactory.getInstance(),
-					true,
-					true
-			);
-		
-		Thumbnailator.createThumbnail(new StreamThumbnailTask(param, is, os));
+
+		Thumbnails.of(is)
+			.size(width, height)
+			.outputFormat(format)
+			.toOutputStream(os);
 	}
\ No newline at end of file
