diff --git a/old_product.java b/new_product.java
index 2aad4af..33567ab 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public Map<String, Map<String, String>> extractBrokerConfigurations(Map<String, Object> properties) {
+public Map<String, Config> extractBrokerConfigurations(Map<String, Object> properties) {
         Map<String, Map<String, String>> configPerBroker = new HashMap<String, Map<String, String>>();
         for (Entry<String, Object> entry : properties.entrySet()) {
             String key = entry.getKey();
@@ -33,5 +33,6 @@ public Map<String, Map<String, String>> extractBrokerConfigurations(Map<String,
             brokerConfig.put(subkeys[1], value);
         }
 
-        return configPerBroker;
+        return configPerBroker.entrySet().stream()
+                .collect(Collectors.toMap(entry -> entry.getKey(), entry -> new Config(entry.getValue())));
     }
\ No newline at end of file
