diff --git a/old_test.java b/new_test.java
index d5c5b74..b0fe3c0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,10 +5,10 @@
         Map<String, Object> properties = new Hashtable<>();
         properties.put("bam.name", "brokername");
         properties.put("bam.url", "tcp://123.123.123.123");
-        Map<String, Map<String, String>> map = service.extractBrokerConfigurations(properties);
+        Map<String, MqttService.Config> map = service.extractBrokerConfigurations(properties);
         assertEquals(map.size(), 1);
-        Map<String, String> data = map.get("bam");
+        MqttService.Config data = map.get("bam");
         assertNotNull(data);
-        assertEquals("brokername", data.get("name"));
-        assertEquals("tcp://123.123.123.123", data.get("url"));
+        assertEquals("brokername", data.name);
+        assertEquals("tcp://123.123.123.123", data.url);
     }
\ No newline at end of file
