diff --git a/old_product.java b/new_product.java
index 940c4ab..f2d6984 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@
         @NotBlank(message = "No command id entered. Unable to set applications.")
         final String id,
         @NotNull(message = "No application ids entered. Unable to set applications.")
-        final Set<String> applicationIds
+        final List<String> applicationIds
     ) throws GenieException {
         if (applicationIds.size() != applicationIds.stream().filter(this.appRepo::exists).count()) {
             throw new GeniePreconditionException("All applications need to exist to add to a command");
