diff --git a/old_test.java b/new_test.java
index 5f16b05..2ec4fe9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,7 @@
     public void testSetApplicationsForCommand() throws GenieException {
         Assert.assertTrue(this.service.getApplicationsForCommand(COMMAND_2_ID).isEmpty());
 
-        final Set<String> appIds = new HashSet<>();
-        appIds.add(APP_1_ID);
+        final List<String> appIds = Lists.newArrayList(APP_1_ID);
         final Set<Command> preCommands = this.appService.getCommandsForApplication(APP_1_ID, null);
         Assert.assertEquals(1, preCommands.size());
         Assert.assertEquals(1, preCommands
