diff --git a/old_test.java b/new_test.java
index b8ae302..71add54 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,4 @@
-@Test(dataProvider="sampleDates")
-    public void test_matchesDate_true(int y, int m, int d) {
-        LocalDate a = LocalDate.date(y, m, d);
-        LocalDate b = LocalDate.date(y, m, d);
-        assertEquals(a.matchesDate(b), true);
+@Test(expectedExceptions=NullPointerException.class)
+    public void test_matches_null() {
+        TEST_2007_07_15.matches(null);
     }
\ No newline at end of file
