diff --git a/old_product.java b/new_product.java
index adc3b55..9cef248 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public boolean matchesDate(LocalDate date) {
-        ISOChronology.checkNotNull(date, "LocalDate must not be null");
+public boolean matchesCalendrical(Calendrical calendrical) {
+        ISOChronology.checkNotNull(calendrical, "Calendrical must not be null");
         for (Entry<DateTimeFieldRule<?>, Integer> entry : fieldValueMap.entrySet()) {
-            Integer dateValue = entry.getKey().getInteger(date);
+            Integer dateValue = entry.getKey().getInteger(calendrical);
             if (dateValue != null && dateValue.equals(entry.getValue()) == false) {
                 return false;
             }
