diff --git a/old_product.java b/new_product.java
index 8b301f6..1f9dc9b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public Weld initialize()
+public WeldContainer initialize()
    {
 
       SEWeldDeployment deployment = new SEWeldDeployment()
@@ -13,7 +13,8 @@ public Weld initialize()
       bootstrap.validateBeans();
       bootstrap.endInitialization();
 
-      instanceManager = WeldManagerUtils.getInstanceByType(manager, InstanceManager.class);
+      InstanceManager instanceManager = WeldManagerUtils.getInstanceByType(manager, InstanceManager.class);
+
+      return new WeldContainer(instanceManager, manager);
 
-      return this;
    }
\ No newline at end of file
