diff --git a/old_product.java b/new_product.java
index 7c2464a..5a0c845 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,6 +2,12 @@ public boolean killTopology() {
     // Setup Connection
     String topologyURI = String.format("%s/v2/groups/%s", this.marathonURI, this.topologyName);
     HttpURLConnection conn = NetworkUtils.getHttpConnection(topologyURI);
+
+    // Attach a token if there is one specified
+    if (this.marathonAuthToken != null) {
+      conn.setRequestProperty("Authorization", String.format("token=%s", this.marathonAuthToken));
+    }
+
     if (conn == null) {
       LOG.log(Level.SEVERE, "Failed to find marathon scheduler");
       return false;
@@ -20,6 +26,9 @@ public boolean killTopology() {
       if (success) {
         LOG.log(Level.INFO, "Successfully killed topology");
         return true;
+      } else if (NetworkUtils.checkHttpResponseCode(conn, HttpURLConnection.HTTP_UNAUTHORIZED)) {
+        LOG.log(Level.SEVERE, "Marathon requires authentication");
+        return false;
       } else {
         LOG.log(Level.SEVERE, "Failed to kill topology");
         return false;
