diff --git a/old_test.java b/new_test.java
index fd200db..70a527b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,12 +6,12 @@
     Collection<MConnector> connectors = client.getConnectors();
     assertEquals(2, connectors.size());
 
-    client.getConnectorConfigBundle(1);
-    connector = client.getConnector(1);
+    client.getConnectorConfigBundle("A1");
+    connector = client.getConnector("A1");
     assertEquals(1, connector.getPersistenceId());
 
-    connector = client.getConnector(2);
-    client.getConnectorConfigBundle(2);
+    connector = client.getConnector("A2");
+    client.getConnectorConfigBundle("A2");
     assertEquals(2, connector.getPersistenceId());
 
     connectors = client.getConnectors();
