diff --git a/old_product.java b/new_product.java
index 133f71f..5447afc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ public Collection<MConnector> getConnectors() {
     ConnectorBean bean = resourceRequests.readConnector(null);
     isAllConnectors = true;
     for(MConnector connector : bean.getConnectors()) {
-      connectors.put(connector.getPersistenceId(), connector);
+      connectors.put(connector.getUniqueName(), connector);
     }
     connectorConfigBundles = bean.getResourceBundles();
 
