diff --git a/old_product.java b/new_product.java
index 5ccb6db..e05dbe8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,7 +16,11 @@ FieldAnalysisRequest resolveAnalysisRequest(SolrQueryRequest req) {
     }
     analysisRequest.setQuery(solrParams.get(AnalysisParams.QUERY, solrParams.get(CommonParams.Q)));
 
-    String value = solrParams.required().get(AnalysisParams.FIELD_VALUE);
+    String value = solrParams.get(AnalysisParams.FIELD_VALUE);
+    if (analysisRequest.getQuery() == null && value == null)  {
+      throw new SolrException(SolrException.ErrorCode.BAD_REQUEST,
+          "One of analysis.value or q or analysis.query parameters must be specified");
+    }
 
     Iterable<ContentStream> streams = req.getContentStreams();
     if (streams != null) {
