diff --git a/old_test.java b/new_test.java
index 9415016..1ebc629 100644
--- a/old_test.java
+++ b/new_test.java
@@ -48,4 +48,35 @@
     request = handler.resolveAnalysisRequest(req);
     assertNull(request.getQuery());
     req.close();
+
+    // test absence of index-time value and presence of q
+    params.remove(AnalysisParams.FIELD_VALUE);
+    params.add(CommonParams.Q, "quick lazy");
+    request = handler.resolveAnalysisRequest(req);
+    assertEquals("quick lazy", request.getQuery());
+    req.close();
+
+    // test absence of index-time value and presence of query
+    params.remove(CommonParams.Q);
+    params.add(AnalysisParams.QUERY, "quick lazy");
+    request = handler.resolveAnalysisRequest(req);
+    assertEquals("quick lazy", request.getQuery());
+    req.close();
+
+    // must fail if all of q, analysis.query or analysis.value are absent
+    params.remove(CommonParams.Q);
+    params.remove(AnalysisParams.QUERY);
+    params.remove(AnalysisParams.FIELD_VALUE);
+    try {
+      request = handler.resolveAnalysisRequest(req);
+      fail("Analysis request must fail if all of q, analysis.query or analysis.value are absent");
+    } catch (SolrException e) {
+      if (e.code() != SolrException.ErrorCode.BAD_REQUEST.code)  {
+        fail("Unexpected exception");
+      }
+    } catch (Exception e) {
+      fail("Unexpected exception");
+    }
+
+    req.close();
   }
\ No newline at end of file
