diff --git a/old_test.java b/new_test.java
index 5053df5..e6eb9da 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test(groups={"tck"})
     public void test_hashCode_floatingWeek_gap_notEndOfDay() {
-        OffsetDateTime odtA = OffsetDateTime.of(2010, 3, 31, 1, 0, OFFSET_0200);
-        ZoneOffsetTransition a1 = new ZoneOffsetTransition(odtA, OFFSET_0300);
-        ZoneOffsetTransition a2 = new ZoneOffsetTransition(odtA, OFFSET_0300);
-        OffsetDateTime odtB = OffsetDateTime.of(2010, 10, 31, 1, 0, OFFSET_0300);
-        ZoneOffsetTransition b = new ZoneOffsetTransition(odtB, OFFSET_0200);
+        LocalDateTime ldtA = LocalDateTime.of(2010, 3, 31, 1, 0);
+        ZoneOffsetTransition a1 = new ZoneOffsetTransition(ldtA, OFFSET_0200, OFFSET_0300);
+        ZoneOffsetTransition a2 = new ZoneOffsetTransition(ldtA, OFFSET_0200, OFFSET_0300);
+        LocalDateTime ldtB = LocalDateTime.of(2010, 10, 31, 1, 0);
+        ZoneOffsetTransition b = new ZoneOffsetTransition(ldtB, OFFSET_0300, OFFSET_0200);
 
         assertEquals(a1.hashCode(), a1.hashCode());
         assertEquals(a1.hashCode(), a2.hashCode());
