diff --git a/old_product.java b/new_product.java
index 8a490d1..d42ac46 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 @Override
     public int hashCode() {
-        return transition.hashCode() ^ transitionAfter.getOffset().hashCode();
+        return transition.hashCode() ^ offsetBefore.hashCode() ^ Integer.rotateLeft(offsetAfter.hashCode(), 16);
     }
\ No newline at end of file
