diff --git a/old_test.java b/new_test.java
index 588eb22..1323564 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,22 +2,22 @@
 	public void testIsSavepoint() throws Exception {
 		{
 			CheckpointProperties props = CheckpointProperties.forStandardCheckpoint();
-			assertFalse(CheckpointProperties.isSavepoint(props));
+			assertFalse(props.isSavepoint());
 		}
 
 		{
 			CheckpointProperties props = CheckpointProperties.forExternalizedCheckpoint(true);
-			assertFalse(CheckpointProperties.isSavepoint(props));
+			assertFalse(props.isSavepoint());
 		}
 
 		{
 			CheckpointProperties props = CheckpointProperties.forExternalizedCheckpoint(false);
-			assertFalse(CheckpointProperties.isSavepoint(props));
+			assertFalse(props.isSavepoint());
 		}
 
 		{
 			CheckpointProperties props = CheckpointProperties.forStandardSavepoint();
-			assertTrue(CheckpointProperties.isSavepoint(props));
+			assertTrue(props.isSavepoint());
 		}
 
 	}
\ No newline at end of file
