diff --git a/old_product.java b/new_product.java
index 05c13c8..5baac64 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,24 +1,19 @@
-void cancel(boolean forceQryCancel) {
-        if (cancelled)
-            return;
+void cancel() {
+        synchronized (this) {
+            if (cancelled)
+                return;
 
-        cancelled = true;
+            cancelled = true;
 
-        for (int i = 0; i < results.length(); i++) {
-            MapQueryResult res = results.get(i);
-
-            if (res != null) {
-                res.close();
-
-                continue;
-            }
-
-            // NB: Cancel is already safe even for lazy queries (see implementation of passed Runnable).
-            if (forceQryCancel) {
+            for (int i = 0; i < results.length(); i++) {
                 GridQueryCancel cancel = cancels[i];
 
                 if (cancel != null)
                     cancel.cancel();
             }
         }
+
+        // The closing result set is synchronized by themselves.
+        // Include to synchronize block may be cause deadlock on <this> and MapQueryResult#lock.
+        close();
     }
\ No newline at end of file
