diff --git a/old_product.java b/new_product.java
index 8c83891..9dd4559 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery sizeNotEq(String propName, int size) {
-        criterion = criterion.and(criterionBuilder.sizeNotEq(propName, size));
+        criterion = criterion.and(Criteria.sizeNotEq(propName, size));
         return this;
     }
\ No newline at end of file
