diff --git a/old_test.java b/new_test.java
index 596706a..0ba407d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,7 @@
 @Test
     public void testInitializeUnchecked() throws ConcurrentException {
         @SuppressWarnings("unchecked")
+        final
         ConcurrentInitializer<Object> init = EasyMock
                 .createMock(ConcurrentInitializer.class);
         final Object result = new Object();
