diff --git a/old_product.java b/new_product.java
index 8597d8f..268c1ed 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public static <T> T initializeUnchecked(final ConcurrentInitializer<T> initializer) {
         try {
             return initialize(initializer);
-        } catch (ConcurrentException cex) {
+        } catch (final ConcurrentException cex) {
             throw new ConcurrentRuntimeException(cex.getCause());
         }
     }
\ No newline at end of file
