diff --git a/old_product.java b/new_product.java
index 7d1880f..e63c9ab 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,11 +5,11 @@ public boolean equalsIgnoreCase(final StrBuilder other) {
         if (this.size != other.size) {
             return false;
         }
-        char thisBuf[] = this.buffer;
-        char otherBuf[] = other.buffer;
+        final char thisBuf[] = this.buffer;
+        final char otherBuf[] = other.buffer;
         for (int i = size - 1; i >= 0; i--) {
-            char c1 = thisBuf[i];
-            char c2 = otherBuf[i];
+            final char c1 = thisBuf[i];
+            final char c2 = otherBuf[i];
             if (c1 != c2 && Character.toUpperCase(c1) != Character.toUpperCase(c2)) {
                 return false;
             }
