diff --git a/old_test.java b/new_test.java
index 4f53551..d763e5e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testEqualsIgnoreCase() {
-        StrBuilder sb1 = new StrBuilder();
-        StrBuilder sb2 = new StrBuilder();
+        final StrBuilder sb1 = new StrBuilder();
+        final StrBuilder sb2 = new StrBuilder();
         assertTrue(sb1.equalsIgnoreCase(sb1));
         assertTrue(sb1.equalsIgnoreCase(sb2));
         assertTrue(sb2.equalsIgnoreCase(sb2));
