diff --git a/old_test.java b/new_test.java
index cf6df11..095ae88 100644
--- a/old_test.java
+++ b/new_test.java
@@ -34,7 +34,7 @@
 
     // remove the output path (its only used for testing)
     Path outputPath = builder.getOutputPath(conf);
-    HadoopUtil.overwriteOutput(outputPath);
+    HadoopUtil.delete(conf, outputPath);
 
     builder.build(NUM_TREES, new MockCallback(data));
   }
\ No newline at end of file
