diff --git a/old_product.java b/new_product.java
index 420bd6f..75cb411 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,34 +15,24 @@ protected static void processOutput(JobContext job,
     Path[] outfiles = DFUtils.listOutputFiles(fs, outputPath);
 
     // read all the outputs
-    TreeID key = new TreeID();
-    MapredOutput value = new MapredOutput();
-
     int index = 0;
     for (Path path : outfiles) {
-      Reader reader = new Reader(fs, path, conf);
-
-      try {
-        while (reader.next(key, value)) {
-          if (keys != null) {
-            keys[index] = key.clone();
-          }
-
-          if (trees != null) {
-            trees[index] = value.getTree();
-          }
-
-          processOutput(firstIds, key, value, callback);
-
-          index++;
+      for (Pair<TreeID,MapredOutput> record : new SequenceFileIterable<TreeID, MapredOutput>(path, conf)) {
+        TreeID key = record.getFirst();
+        MapredOutput value = record.getSecond();
+        if (keys != null) {
+          keys[index] = key;
+        }
+        if (trees != null) {
+          trees[index] = value.getTree();
         }
-      } finally {
-        reader.close();
+        processOutput(firstIds, key, value, callback);
+        index++;
       }
     }
 
     // make sure we got all the keys/values
-    if ((keys != null) && (index != keys.length)) {
+    if (keys != null && index != keys.length) {
       throw new IllegalStateException("Some key/values are missing from the output");
     }
   }
\ No newline at end of file
