diff --git a/old_test.java b/new_test.java
index fc66644..6c18dd9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-  public void testRecord() {
+  public void testRecordDouble() {
     viewManager.registerView(RpcViewConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW);
     View beforeView =
         viewManager.getView(
@@ -22,8 +22,9 @@
     StatsContext context =
         defaultStatsContext.with(
             RpcMeasurementConstants.RPC_CLIENT_METHOD, TagValue.create("myMethod"));
-    MeasurementMap measurements =
-        MeasurementMap.of(RpcMeasurementConstants.RPC_CLIENT_ROUNDTRIP_LATENCY, 5.1);
+    MeasureMap measurements =
+        MeasureMap.builder()
+            .set((DoubleMeasure) RpcMeasurementConstants.RPC_CLIENT_ROUNDTRIP_LATENCY, 5.1).build();
     context.record(measurements);
     View afterView =
         viewManager.getView(
