diff --git a/old_test.java b/new_test.java
index 7909489..fdf031a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,6 +15,15 @@
     // OK to create dir
     PowerMockito.when(FileUtils.createDirectory(Mockito.anyString())).thenReturn(true);
 
+    // Fail to cleanup
+    PowerMockito.when(FileUtils.cleanDir(Mockito.anyString())).thenReturn(false);
+    Assert.assertFalse(SchedulerUtils.setupWorkingDirectory(
+            WORKING_DIR, CORE_RELEASE_URI, CORE_RELEASE_DEST,
+            TOPOLOGY_URI, TOPOLOGY_DEST, isVerbose));
+
+    // Ok to cleanup
+    PowerMockito.when(FileUtils.cleanDir(Mockito.anyString())).thenReturn(true);
+
     PowerMockito.spy(SchedulerUtils.class);
     // OK to curl and extract core-release-package
     PowerMockito.doReturn(true).when(SchedulerUtils.class, "curlAndExtractPackage",
