diff --git a/old_product.java b/new_product.java
index baa86a6..d32d6a1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,6 +14,12 @@ public static boolean setupWorkingDirectory(
       }
     }
 
+    // Cleanup the directory
+    if (!FileUtils.cleanDir(workingDirectory)) {
+      LOG.severe("Failed to clean directory: " + workingDirectory);
+      return false;
+    }
+
     // Curl and extract heron core release package and topology package
     // And then delete the downloaded release package
     boolean ret =
