diff --git a/old_product.java b/new_product.java
index e61656a..765908d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,7 @@
 public final void replace(final int rpre, final Data data) {
     meta.update();
-    meta.docindex = false;
+    docindex.replace(rpre, data);
 
-    // check if attribute size of parent must be updated
     final int dsize = data.meta.size;
     buffer(dsize);
 
