diff --git a/old_product.java b/new_product.java
index 6a7c874..bf97f8d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,18 +3,18 @@
         if (!isManaged()) {
             throw new UnsupportedOperationException(ONLY_IN_MANAGED_MODE_MESSAGE);
         }
-        checkValidView();
+        checkValidRealm();
         if (className != null) {
             return new OrderedRealmCollectionSnapshot<>(
                     realm,
-                    new io.realm.internal.Collection(realm.sharedRealm, view, null),
+                    new io.realm.internal.Collection(realm.sharedRealm, osList, null),
                     className);
         } else {
             // 'clazz' is non-null when 'dynamicClassName' is null.
             //noinspection ConstantConditions
             return new OrderedRealmCollectionSnapshot<>(
                     realm,
-                    new io.realm.internal.Collection(realm.sharedRealm, view, null),
+                    new io.realm.internal.Collection(realm.sharedRealm, osList, null),
                     clazz);
         }
     }
\ No newline at end of file
