diff --git a/old_test.java b/new_test.java
index 223ae36..2da8d12 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,6 +7,6 @@
         realm.commitTransaction();
         assertEquals(sizeBefore - 1, collection.size());
 
-        assertNotNull(collection.view);
-        assertEquals(collection.view.getTargetTable().getName(), snapshot.getTable().getName());
+        assertNotNull(collection.osList);
+        assertEquals(collection.osList.getTargetTable().getName(), snapshot.getTable().getName());
     }
\ No newline at end of file
