diff --git a/old_test.java b/new_test.java
index 2b2a439..3241c23 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,7 +11,7 @@
         
         //instance.setQueryReferences(null);
         //instance.setDatabaseReferences(null);
-        ArrayList<Result> result = instance.createObjects(1e-10);
+        List<Result> result = instance.createObjects(1e-10);
         
         // test with random manual selected results
         BlastHsp hsp1hit1res1 = new BlastHspBuilder()
