diff --git a/old_product.java b/new_product.java
index e94b287..83d4ef1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,7 @@
-public int hashCode(){
-        String allInOne = hitId+hitLen;
-        return allInOne.hashCode();
+@Override
+    public int hashCode() {
+        int hash = 3;
+        hash = 89 * hash + this.hitLen;
+        hash = 89 * hash + (this.hsps != null ? this.hsps.hashCode() : 0);
+        return hash;
     }
\ No newline at end of file
