diff --git a/old_test.java b/new_test.java
index 014c110..a40a8da 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,13 +5,42 @@
         int expResult;
         int result;
         
-        instance = hspImpl;
-        expResult = 71782805;
+        instance = new BlastHspBuilder()
+                .setHspNum(1)
+                .setHspBitScore(377.211)
+                .setHspEvalue(8.04143e-093)
+                .setHspQueryFrom(1)
+                .setHspQueryTo(224)
+                .setHspHitFrom(1035)
+                .setHspHitTo(811)
+                .setHspQueryFrame(-1)
+                .setHspIdentity(213)
+                .setHspPositive(213)
+                .setHspGaps(5)
+                .setHspAlignLen(227)
+                .setHspQseq("CTGACGACAGCCATGCACCACCTGTCTCGACTTTCCCCCGAAGGGCACCTAATGTATCTCTACCTCGTTAGTCGGATGTCAAGACCTGGTAAGGTTTTTTCGCGTATCTTCGAATTAAACCACATACTCCACTGCTTGTGCGG-CCCCCGTCAATTCCTTTGAGTTTCAACCTTGCGGCCGTACTCCC-AGGTGGA-TACTTATTGTGTTAACTCCGGCACGGAAGG")
+                .setHspHseq("CTGACGACAACCATGCACCACCTGTCTCAACTTTCCCC-GAAGGGCACCTAATGTATCTCTACTTCGTTAGTTGGATGTCAAGACCTGGTAAGGTT-CTTCGCGTTGCTTCGAATTAAACCACATACTCCACTGCTTGTGCGGGCCCCCGTCAATTCCTTTGAGTTTCAACCTTGCGGTCGTACTCCCCAGGTGGATTACTTATTGTGTTAACTCCGGCACAGAAGG")
+                .setHspIdentityString("||||||||| |||||||||||||||||| ||||||||| |||||||||||||||||||||||| |||||||| |||||||||||||||||||||||  |||||||  |||||||||||||||||||||||||||||||||||| |||||||||||||||||||||||||||||||||| ||||||||| ||||||| |||||||||||||||||||||||| |||||")
+                .createBlastHsp();
+        
+        expResult = hspImpl.hashCode();
         result = instance.hashCode();
         assertEquals(expResult, result);
         
-        instance = uncompleteHsp;
-        expResult = 679;
+        instance = new BlastHspBuilder()
+                .setPercentageIdentity(100.00/100)
+                .setHspAlignLen(48)
+                .setMismatchCount(0)
+                .setHspGaps(0)
+                .setHspQueryFrom(1)
+                .setHspQueryTo(48)
+                .setHspHitFrom(344)
+                .setHspHitTo(391)
+                .setHspEvalue(4e-19)
+                .setHspBitScore(95.6)
+                .createBlastHsp();
+        
+        expResult = uncompleteHsp.hashCode();
         result = instance.hashCode();
         assertEquals(expResult, result);
         
@@ -28,5 +57,5 @@
                 .setHspBitScore(95.6)
                 .createBlastHsp();
         
-        assertFalse(uncompleteHsp.hashCode() == uncompleteHsp2.hashCode());
+        assertEquals(uncompleteHsp.hashCode(), uncompleteHsp2.hashCode());
     }
\ No newline at end of file
