diff --git a/old_test.java b/new_test.java
index 919b091..734ce4e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
 	public void process_simple() {
 		SquaresIntoCrossClusters alg = new SquaresIntoCrossClusters(0.05, -1);
 
-		List<Polygon2D_F64> squares = new ArrayList<Polygon2D_F64>();
+		List<Polygon2D_F64> squares = new ArrayList<>();
 		squares.add( createSquare(7,8));
 		squares.add( createSquare(9,8));
 		squares.add( createSquare(8,9));
