diff --git a/old_test.java b/new_test.java
index df45e04..9cd9a6c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -47,7 +47,7 @@
             assertFalse(cf3.isDone());
             try {
                 Object result = cf3.get(100, TimeUnit.MILLISECONDS);
-                fail("Dependent completion stage must not complete first");
+                fail("Dependent completion stage must not complete first: " + result);
             } catch (TimeoutException x) {
             }
 
