diff --git a/old_product.java b/new_product.java
index 1731c2f..929cbb8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,7 +12,7 @@
             if (other instanceof ManagedCompletableFuture)
                 other = ((ManagedCompletableFuture<? extends T>) other).completableFuture;
             CompletableFuture<Void> dependentStage = completableFuture.acceptEither(other, action);
-            return new ManagedCompletableFuture<Void>(dependentStage, defaultExecutor, null);
+            return newInstance(dependentStage, defaultExecutor, null);
         } else {
             return super.acceptEither(other, action);
         }
