diff --git a/old_test.java b/new_test.java
index 04cb0d6..d8f723f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,8 +5,7 @@
     verifyOperationFailuresOnInactiveStore(store);
     StorageManagerMetrics metrics = new StorageManagerMetrics(new MetricRegistry());
     StoreConfig config = new StoreConfig(new VerifiableProperties(properties));
-    store =
-        new BlobStore(storeId, config, scheduler, diskIOScheduler, metrics, tempDirStr, LOG_CAPACITY, STORE_KEY_FACTORY,
-            recovery, hardDelete, time);
+    store = new BlobStore(storeId, config, scheduler, storeStatsScheduler, diskIOScheduler, metrics, tempDirStr,
+        LOG_CAPACITY, STORE_KEY_FACTORY, recovery, hardDelete, time);
     verifyOperationFailuresOnInactiveStore(store);
   }
\ No newline at end of file
