diff --git a/old_product.java b/new_product.java
index 7123dde..4cd87ec 100644
--- a/old_product.java
+++ b/new_product.java
@@ -29,6 +29,10 @@ void shutdown() throws InterruptedException {
             "Could not shutdown " + numFailures.get() + " out of " + stores.size() + " stores on the disk " + disk);
       }
       compactionManager.awaitTermination();
+      longLivedTaskScheduler.shutdown();
+      if (!longLivedTaskScheduler.awaitTermination(30, TimeUnit.SECONDS)) {
+        logger.error("Could not terminate long live tasks after DiskManager shutdown");
+      }
     } finally {
       metrics.diskShutdownTimeMs.update(time.milliseconds() - startTimeMs);
     }
