diff --git a/old_test.java b/new_test.java
index 4bdd963..09068cf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testValues()
     {
         Set<PageLayout> pageLayoutSet = EnumSet.noneOf(PageLayout.class);
-        Set<String> stringSet = new HashSet<String>();
+        Set<String> stringSet = new HashSet<>();
         for (PageLayout pl : PageLayout.values())
         {
             String s = pl.stringValue();
