diff --git a/old_product.java b/new_product.java
index 9772cdd..828bce8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,7 +7,7 @@
             return Collections.emptySet();
         }
         
-        List<V> values = new ArrayList<V>(mapArr.length >> 1);
+        List<V> values = new ArrayList<>(mapArr.length >> 1);
         for (int vIdx = 1; vIdx < mapArr.length; vIdx+=2)
         {
             values.add((V)mapArr[vIdx]);
