diff --git a/old_product.java b/new_product.java
index 4bb4990..26bc053 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public synchronized List<String> getHosts() { // TODO: Check whether method should be synchronized
-    List<String> hlist = new ArrayList<String>();
+    List<String> hlist = new ArrayList<>();
     for (String h : hostRoleCommands.keySet()) {
       hlist.add(h);
     }
