diff --git a/old_product.java b/new_product.java
index ab7a6c3..1f10adf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -32,17 +32,46 @@ private static <T> String toStringImpl0(
                 else
                     first = false;
 
-                String name = fd.getName();
+                buf.a(fd.getName()).a('=');
 
-                Field field = cls.getDeclaredField(name);
+                switch (fd.type()) {
+                    case GridToStringFieldDescriptor.FIELD_TYPE_OBJECT:
+                        toString(buf, fd.fieldClass(), GridUnsafe.getObjectField(obj, fd.offset()));
 
-                field.setAccessible(true);
+                        break;
+                    case GridToStringFieldDescriptor.FIELD_TYPE_BYTE:
+                        buf.a(GridUnsafe.getByteField(obj, fd.offset()));
 
-                buf.a(name).a('=');
+                        break;
+                    case GridToStringFieldDescriptor.FIELD_TYPE_BOOLEAN:
+                        buf.a(GridUnsafe.getBooleanField(obj, fd.offset()));
 
-                Class<?> fieldType = field.getType();
+                        break;
+                    case GridToStringFieldDescriptor.FIELD_TYPE_CHAR:
+                        buf.a(GridUnsafe.getCharField(obj, fd.offset()));
 
-                toString(buf, fieldType, field.get(obj));
+                        break;
+                    case GridToStringFieldDescriptor.FIELD_TYPE_SHORT:
+                        buf.a(GridUnsafe.getShortField(obj, fd.offset()));
+
+                        break;
+                    case GridToStringFieldDescriptor.FIELD_TYPE_INT:
+                        buf.a(GridUnsafe.getIntField(obj, fd.offset()));
+
+                        break;
+                    case GridToStringFieldDescriptor.FIELD_TYPE_FLOAT:
+                        buf.a(GridUnsafe.getFloatField(obj, fd.offset()));
+
+                        break;
+                    case GridToStringFieldDescriptor.FIELD_TYPE_LONG:
+                        buf.a(GridUnsafe.getLongField(obj, fd.offset()));
+
+                        break;
+                    case GridToStringFieldDescriptor.FIELD_TYPE_DOUBLE:
+                        buf.a(GridUnsafe.getDoubleField(obj, fd.offset()));
+
+                        break;
+                }
             }
 
             appendVals(buf, first, addNames, addVals, addSens, addLen);
