diff --git a/old_product.java b/new_product.java
index 00269c0..03e55b6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@
             if (useExplicitIv) {
                 decryptIv = new IvParameterSpec(Arrays.copyOf(data, decryptCipher.getBlockSize()));
             }
-            if (tlsContext.getChooser().getConnectionEnd().getConnectionEndType() == ConnectionEndType.CLIENT) {
+            if (tlsContext.getChooser().getConnection().getLocalConnectionEndType() == ConnectionEndType.CLIENT) {
                 decryptCipher.init(Cipher.DECRYPT_MODE, new SecretKeySpec(serverWriteKey, bulkCipherAlg.getJavaName()),
                         decryptIv);
             } else {
