diff --git a/old_test.java b/new_test.java
index 7a76be4..2b6b3bd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
-    public void testDecrypt() {
+    public void testDecrypt() throws NoSuchAlgorithmException {
         context.setConnectionEnd(new ClientConnectionEnd());
-        this.cipher = new RecordAEADCipher(context);
+        this.cipher = new RecordAEADCipher(context, KeySetGenerator.generateKeySet(context));
         byte[] ciphertext = ArrayConverter.hexStringToByteArray("1BB3293A919E0D66F145AE830488E8D89BE5EC16688229");
         byte[] plaintext = cipher.decrypt(ciphertext);
         byte[] plaintext_correct = ArrayConverter.hexStringToByteArray("08000002000016");
