diff --git a/old_product.java b/new_product.java
index 14fa3d6..a4ca07d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public void setSpecialGroup(int groupID)
+public void setSpecialGroup(UUID groupID)
     {
-        specialGroups.add(Integer.valueOf(groupID));
+        specialGroups.add(groupID);
 
         // System.out.println("Added " + groupID);
     }
\ No newline at end of file
