diff --git a/old_test.java b/new_test.java
index 8636f42..ecf27f0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,12 +5,14 @@
         Context instance = new Context();
         
         // Pass in random integers (need not be valid group IDs)
-        instance.setSpecialGroup(10000);
-        instance.setSpecialGroup(10001);
+        UUID groupID1 = UUID.randomUUID();
+        UUID groupID2 = UUID.randomUUID();
+        instance.setSpecialGroup(groupID1);
+        instance.setSpecialGroup(groupID2);
         
-        assertThat("testSetSpecialGroup 0", instance.inSpecialGroup(10000), equalTo(true));
-        assertThat("testSetSpecialGroup 1", instance.inSpecialGroup(10001), equalTo(true));
-        assertThat("testSetSpecialGroup 2", instance.inSpecialGroup(20000), equalTo(false));
+        assertThat("testSetSpecialGroup 0", instance.inSpecialGroup(groupID1), equalTo(true));
+        assertThat("testSetSpecialGroup 1", instance.inSpecialGroup(groupID2), equalTo(true));
+        assertThat("testSetSpecialGroup 2", instance.inSpecialGroup(UUID.randomUUID()), equalTo(false));
 
         // Cleanup our context
         cleanupContext(instance);
