diff --git a/old_test.java b/new_test.java
index 7dad51f..152fa4a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testRank() {
         System.out.println("rank");
         SumSquaresRatio ssr = new SumSquaresRatio();
-        double[] ratio = ssr.rank(Iris.x, Iris.y);
+        double[] ratio = ssr.apply(Iris.x, Iris.y);
         assertEquals(4, ratio.length);
         assertEquals( 1.6226463, ratio[0], 1E-6);
         assertEquals( 0.6444144, ratio[1], 1E-6);
