diff --git a/old_product.java b/new_product.java
index 9e1fe8e..3ad9700 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 @Override
     synchronized public DatagramSocket socket() {
         if (null == socket) {
-            socket = new DatagramSocketAdapter(SocketImplProvider
-                    .getDatagramSocketImpl(fd, localPort), this);
+            socket = new DatagramSocketAdapter(
+                    new PlainDatagramSocketImpl(fd, localPort), this);
         }
         return socket;
     }
\ No newline at end of file
