diff --git a/old_test.java b/new_test.java
index b9227cc..eaa1db4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,7 @@
         method = "socket",
         args = {}
     )
+    @KnownFailure(value="bug 2155708")
     public void testSocket_BasicStatusBeforeConnect() throws SocketException {
         assertFalse(this.channel1.isConnected());// not connected
         DatagramSocket s1 = this.channel1.socket();
