diff --git a/old_test.java b/new_test.java
index dd43382..9129640 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,6 @@
                 Month.MARCH, 20, DayOfWeek.SUNDAY, TIME_0100, false, TimeDefinition.WALL,
                 OFFSET_0200, OFFSET_0200, OFFSET_0300);
         ZoneOffsetTransition trans = new ZoneOffsetTransition(
-                OffsetDateTime.of(2000, Month.MARCH, 26, 1, 0, OFFSET_0200), OFFSET_0300);
+                LocalDateTime.of(2000, Month.MARCH, 26, 1, 0), OFFSET_0200, OFFSET_0300);
         assertEquals(test.createTransition(2000), trans);
     }
\ No newline at end of file
