diff --git a/old_product.java b/new_product.java
index 86ac938..e0146f2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,6 +15,6 @@ public ZoneOffsetTransition createTransition(int year) {
             date = date.plusDays(1);
         }
         LocalDateTime localDT = LocalDateTime.of(date, time);
-        OffsetDateTime transition = timeDefinition.createDateTime(localDT, standardOffset, offsetBefore);
-        return new ZoneOffsetTransition(transition, offsetAfter);
+        LocalDateTime transition = timeDefinition.createDateTime(localDT, standardOffset, offsetBefore);
+        return new ZoneOffsetTransition(transition, offsetBefore, offsetAfter);
     }
\ No newline at end of file
