diff --git a/old_product.java b/new_product.java
index 65bdeaa..8eda7ae 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public static Expr get(final Op cmp, final Expr a, final Expr o,
       final InputInfo ii) throws QueryException {
 
     // position() = last() -> last()
-    if(a.isFun(FunDef.LAST) && cmp == Op.EQ) return a;
+    if(a.isFun(Function.LAST) && cmp == Op.EQ) return a;
 
     if(a.item()) {
       final Item it = (Item) a;
