diff --git a/old_test.java b/new_test.java
index df4e84d..c92b3f1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,10 @@
                 = (DataCiteIdentifierProvider)
                 sm.getServicesByType(DataCiteIdentifierProvider.class).get(0);
 
-        DSpaceObject object = item;
+        String identifier = UUID.randomUUID().toString();
+        DSpaceObject object = newItem(context);
+        instance.register(context, object, identifier);
+
         String result = instance.lookup(context, object);
         assertNotNull("Null returned", result);
     }
\ No newline at end of file
