diff --git a/old_product.java b/new_product.java
index c927153..eab5a33 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,10 @@
-public EZIDResponse lookup()
-            throws IdentifierException, IOException
+public EZIDResponse lookup(String name)
+            throws IdentifierException, IOException, URISyntaxException
     {
         // GET path
         HttpGet request;
-        request = new HttpGet(url);
+        URI uri = new URI(scheme, host, ID_PATH + authority + name, null);
+        request = new HttpGet(uri);
         HttpResponse response = client.execute(request);
         return new EZIDResponse(response);
     }
\ No newline at end of file
