diff --git a/old_test.java b/new_test.java
index 5f92733..ad5ce21 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,13 +8,11 @@
 		
 		Order orderThatCanDiscontinueTheOrder = anOrder.cloneForDiscontinuing();
 		
-		assertEquals(anOrder.getClass(), orderThatCanDiscontinueTheOrder.getClass());
-		
 		assertEquals(anOrder.getPatient(), orderThatCanDiscontinueTheOrder.getPatient());
 		
 		assertEquals(anOrder.getConcept(), orderThatCanDiscontinueTheOrder.getConcept());
 		
-		assertEquals("should set previous order to anOrder", orderThatCanDiscontinueTheOrder.getPreviousOrder(), anOrder);
+		assertEquals("should set previous order to anOrder", anOrder, orderThatCanDiscontinueTheOrder.getPreviousOrder());
 		
 		assertEquals("should set new order action to new", orderThatCanDiscontinueTheOrder.getAction(),
 		    Order.Action.DISCONTINUE);
