diff --git a/old_test.java b/new_test.java
index f1ce6b7..fbaa3d3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
             realm.createObject(testClazz, null);
             fail("Null value as primary key already exists.");
         } catch (RealmPrimaryKeyConstraintException expected) {
-            assertEquals("Value already exists: null", expected.getMessage());
+            assertEquals("Primary key value already exists: 'null' .", expected.getMessage());
         } finally {
             realm.cancelTransaction();
         }
