diff --git a/old_product.java b/new_product.java
index d3eede8..c3d2038 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public DynamicRealmObject createObject(String className, Object primaryKeyValue) {
         Table table = schema.getTable(className);
-        long index = table.addEmptyRowWithPrimaryKey(primaryKeyValue);
-        return new DynamicRealmObject(this, table.getCheckedRow(index));
+        return new DynamicRealmObject(this,
+                CheckedRow.getFromRow(OsObject.createWithPrimaryKey(sharedRealm, table, primaryKeyValue)));
     }
\ No newline at end of file
