diff --git a/old_product.java b/new_product.java
index c902743..f732258 100644
--- a/old_product.java
+++ b/new_product.java
@@ -51,7 +51,7 @@ void create() throws Exception {
         function.error(HEAD_METHOD);
 
       // serialize result
-      final SerializerProp sp = function.output;
+      final SerializerOptions sp = function.output;
       http.initResponse(sp);
       final Serializer ser = Serializer.get(http.res.getOutputStream(), sp);
       for(; item != null; item = iter.next()) ser.serialize(item);
