diff --git a/old_test.java b/new_test.java
index 238ea19..4281f11 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,10 @@
 @Test
     public void testGetEvidence() {
         Dependency instance = new Dependency();
-        EvidenceCollection result = instance.getEvidence();
+        Set<Evidence> result = instance.getEvidence(EvidenceType.VENDOR);
+        assertNotNull(result);
+        result = instance.getEvidence(EvidenceType.PRODUCT);
+        assertNotNull(result);
+        result = instance.getEvidence(EvidenceType.VERSION);
         assertNotNull(result);
     }
\ No newline at end of file
