diff --git a/old_test.java b/new_test.java
index de6da1d..e9fbdf6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void put() throws Exception {
-    try(final QueryProcessor qp = new QueryProcessor(_HTTP_SEND_REQUEST.args(
+    try(QueryProcessor qp = new QueryProcessor(_HTTP_SEND_REQUEST.args(
         "<http:request method='put' status-only='true'>"
         + "<http:body media-type='text/xml'>" + BOOKS + "</http:body>"
         + "</http:request>", RESTURL), ctx)) {
