diff --git a/old_product.java b/new_product.java
index 4c7de6c..aa6d472 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,10 +6,10 @@ protected static void put(final String u, final InputStream is, final MediaType
     conn.setDoOutput(true);
     conn.setRequestMethod(PUT.name());
     if(type != null) conn.setRequestProperty(HttpText.CONTENT_TYPE, type.toString());
-    try(final OutputStream bos = new BufferedOutputStream(conn.getOutputStream())) {
+    try(OutputStream bos = new BufferedOutputStream(conn.getOutputStream())) {
       if(is != null) {
         // send input stream if it not empty
-        try(final BufferedInputStream bis = new BufferedInputStream(is)) {
+        try(BufferedInputStream bis = new BufferedInputStream(is)) {
           for(int i; (i = bis.read()) != -1;) bos.write(i);
         }
       }
