diff --git a/old_product.java b/new_product.java
index dd207c6..b4521d4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -21,8 +21,7 @@ private Item replace(final byte[] val, final QueryContext ctx) throws QueryExcep
     try {
       return Str.get(p.matcher(string(val)).replaceAll(r));
     } catch(final Exception ex) {
-      final String m = ex.getMessage();
-      if(m.contains("No group")) REGROUP.thrw(info);
-      throw REGPAT.thrw(info, m);
+      if(ex.getMessage().contains("No group")) REGROUP.thrw(info);
+      throw REGPAT.thrw(info, ex);
     }
   }
\ No newline at end of file
