diff --git a/old_product.java b/new_product.java
index 36c6c9b..d12567f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
-protected boolean hasServiceSupport(String layerName, String serviceName) {
-        LayerInfo linfo = getGeoServer().getCatalog().getLayerByName(layerName);
-        if (linfo != null && linfo.getResource() != null && serviceName != null) {
-            List<String> disabledServices =
-                    DisabledServiceResourceFilter.disabledServices(linfo.getResource());
-            return disabledServices.stream().noneMatch(d -> d.equalsIgnoreCase(serviceName));
+private List<ExternalLink> commonFormatLinks(PreviewLayer layer) {
+        List<ExternalLink> links = new ArrayList<>();
+        List<CommonFormatLink> formats =
+                getGeoServerApplication().getBeansOfType(CommonFormatLink.class);
+        Collections.sort(formats);
+        for (CommonFormatLink link : formats) {
+            links.add(link.getFormatLink(layer));
         }
-        // layer group and backward compatibility
-        return true;
+        return links;
     }
\ No newline at end of file
