diff --git a/old_test.java b/new_test.java
index 2b4162c..3b8c135 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,8 +7,8 @@
         lg.getLayers().add(cat.getLayerByName(getLayerId(MockData.PRIMITIVEGEOFEATURE)));
         new CatalogBuilder(cat).calculateLayerGroupBounds(lg);
         cat.add(lg);
+        PreviewLayer layer = new PreviewLayer(lg);
         tester.startPage(MapPreviewPage.class);
         tester.assertRenderedPage(MapPreviewPage.class);
-        MapPreviewPage page = (MapPreviewPage) tester.getLastRenderedPage();
-        assertTrue(page.hasServiceSupport("sf:linkgroup", "WMS"));
+        assertTrue(layer.hasServiceSupport("WMS"));
     }
\ No newline at end of file
