diff --git a/old_test.java b/new_test.java
index 62513e9..0ac7d6f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
     public void test_plus_Period_positiveHours() {
-        Period period = Period.of(7, LocalDateTimeUnit.HOURS);
+        Period period = Period.of(7, LocalPeriodUnit.HOURS);
         LocalTime t = TEST_12_30_40_987654321.plus(period);
         assertEquals(t, LocalTime.of(19, 30, 40, 987654321));
     }
\ No newline at end of file
