diff --git a/old_product.java b/new_product.java
index 0d62ae3..35fa20f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,3 @@
-public static boolean migratePrimaryKeyTableIfNeeded(SharedRealm sharedRealm) {
-        if (sharedRealm == null || !sharedRealm.isInTransaction()) {
-            throwImmutable();
-        }
-        if (!sharedRealm.hasTable(PRIMARY_KEY_TABLE_NAME)) {
-            return false;
-        }
-        Table pkTable = sharedRealm.getTable(PRIMARY_KEY_TABLE_NAME);
-        return nativeMigratePrimaryKeyTableIfNeeded(sharedRealm.getGroupNative(), pkTable.nativePtr);
+public static void migratePrimaryKeyTableIfNeeded(SharedRealm sharedRealm) {
+        nativeMigratePrimaryKeyTableIfNeeded(sharedRealm.getNativePtr());
     }
\ No newline at end of file
