diff --git a/old_test.java b/new_test.java
index 9578be6..1dfa88b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,16 +1,14 @@
 @Test
   public void testAddBlockout() {
-    SimpleBlockoutTrigger trigger = new SimpleBlockoutTrigger("blockout", new Date(), null, -1, 1000000, 50000 );
+    SimpleBlockoutTrigger trigger = new SimpleBlockoutTrigger("blockout", new Date(), null, -1, 1000000, 50000); //$NON-NLS-1$
     try {
-      blockoutManager.addBlockout(trigger);
+      blockOutManager.addBlockout(trigger);
     } catch (SchedulerException e) {
-      // TODO Auto-generated catch block
-      e.printStackTrace();
+      throw new RuntimeException(e);
     }
     try {
-      assertEquals(blockoutManager.getBlockout("blockout"), trigger);
+      assertEquals(blockOutManager.getBlockout("blockout"), trigger); //$NON-NLS-1$
     } catch (SchedulerException e) {
-      // TODO Auto-generated catch block
-      e.printStackTrace();
+      throw new RuntimeException(e);
     }
   }
\ No newline at end of file
