diff --git a/old_test.java b/new_test.java
index ea8dd3a..c100712 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test
     public void testSetName() {
-        final Application a = new Application();
-        Assert.assertNull(a.getName());
-        a.setName(NAME);
-        Assert.assertEquals(NAME, a.getName());
+        Assert.assertNull(this.a.getName());
+        this.a.setName(NAME);
+        Assert.assertEquals(NAME, this.a.getName());
     }
\ No newline at end of file
