diff --git a/old_product.java b/new_product.java
index ff91385..d035c03 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,3 @@
-public void setName(final String name) throws GenieException {
-        if (StringUtils.isBlank(name)) {
-            throw new GenieException(
-                    HttpURLConnection.HTTP_BAD_REQUEST,
-                    "No name Entered.");
-        }
+public void setName(final String name) {
         this.name = name;
     }
\ No newline at end of file
