diff --git a/old_product.java b/new_product.java
index 1550a1b..3e25c9e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,8 @@
 public static void fullViewLeft(IntrinsicParameters paramLeft,
-									boolean applyLeftToRight,
 									DenseMatrix64F rectifyLeft, DenseMatrix64F rectifyRight,
 									DenseMatrix64F rectifyK)
 	{
-		PointTransform_F32 tranLeft = rectifyTransform(paramLeft, applyLeftToRight, rectifyLeft);
+		PointTransform_F32 tranLeft = rectifyTransform(paramLeft, rectifyLeft);
 
 		Rectangle2D_F32 bound = DistortImageOps.boundBox_F32(paramLeft.width, paramLeft.height,
 				new PointToPixelTransform_F32(tranLeft));
