diff --git a/old_product.java b/new_product.java
index 761c879..33105d4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public Map<String, ServiceInfo> getServices(String stackName, String version) {
+public Map<String, ServiceInfo> getServices(String stackName, String version) throws AmbariException {
 
         Map<String, ServiceInfo> servicesInfoResult = new HashMap<String, ServiceInfo>();
 
