diff --git a/old_product.java b/new_product.java
index 847e25e..2ecd499 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,20 +10,14 @@
       return false;
     }
 
-    if(null == _configuration)
-      _configuration = new Configuration();
-    
-    String rollingUpgradeStack = _configuration.getRollingUpgradeStack(); 
-    // not applicable if not HDP 2.2.4.2 or later    
-    String stackName = cluster.getCurrentStackVersion().getStackName();
-    if (!rollingUpgradeStack.equals(stackName)) {
-      return false;
-    }
-
-    String rollingUpgradeVersion = _configuration.getRollingUpgradeVersion();
-    String currentClusterRepositoryVersion = cluster.getCurrentClusterVersion().getRepositoryVersion().getVersion();
-    if (VersionUtils.compareVersions(currentClusterRepositoryVersion, rollingUpgradeVersion) < 0) {
-      return false;
+    // Applicable only if stack not defined in MinimumApplicableStackVersion, or
+    // version equals or exceeds the enumerated version.
+    for (MinimumApplicableStackVersion minimumStackVersion : MinimumApplicableStackVersion.values()) {
+      String stackName = cluster.getCurrentStackVersion().getStackName();
+      if (minimumStackVersion.getStackName().equals(stackName)){
+        String currentClusterRepositoryVersion = cluster.getCurrentClusterVersion().getRepositoryVersion().getVersion();
+        return VersionUtils.compareVersions(currentClusterRepositoryVersion, minimumStackVersion.getStackVersion()) >= 0;
+      }
     }
 
     return true;
