diff --git a/old_test.java b/new_test.java
index 6f865e6..01e283a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
   public final void replace() throws IOException {
     session.execute("create db " + DB);
-    check("0", session.query("count(" + _DB_OPEN.args(DB) + ")").execute());
+    check("0", session.query("count(" + _DB_OPEN.args(DB) + ')').execute());
     session.replace(DB, new ArrayInput("<X/>"));
-    check("1", session.query("count(" + _DB_OPEN.args(DB) + ")").execute());
-    session.replace(DB + "2", new ArrayInput("<X/>"));
-    check("2", session.query("count(" + _DB_OPEN.args(DB) + ")").execute());
-    session.replace(DB + "2", new ArrayInput("<X/>"));
-    check("2", session.query("count(" + _DB_OPEN.args(DB) + ")").execute());
+    check("1", session.query("count(" + _DB_OPEN.args(DB) + ')').execute());
+    session.replace(DB + '2', new ArrayInput("<X/>"));
+    check("2", session.query("count(" + _DB_OPEN.args(DB) + ')').execute());
+    session.replace(DB + '2', new ArrayInput("<X/>"));
+    check("2", session.query("count(" + _DB_OPEN.args(DB) + ')').execute());
   }
\ No newline at end of file
