diff --git a/old_product.java b/new_product.java
index 07f6ed1..5af085b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 private static String replace(final String cls, final char incl) {
     final int[] v = cls.startsWith("Is") ? MAP.get(cls.substring(2)) : null;
-    if(v == null) return "\\" + incl + "{" + cls + "}";
+    if(v == null) return "\\" + incl + '{' + cls + '}';
     final TokenBuilder tb = new TokenBuilder().add('[');
     if(incl == 'P') tb.add("!");
     for(int i = 0; i < v.length;) {
