diff --git a/old_product.java b/new_product.java
index 77237bf..9ab537a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,11 @@
-public static void  syncLocalCoreWithSharedStore( String collectionName, String coreName, String shardName, CoreContainer  coreContainer) throws SolrException
+public static void syncLocalCoreWithSharedStore(String collectionName, String coreName, String shardName, CoreContainer coreContainer) throws SolrException
   {
     assertTrue(coreContainer.isZooKeeperAware());
 
     ZkController zkController = coreContainer.getZkController();
-    SharedShardMetadataController sharedMetadataController = zkController.getSharedShardMetadataController();
+    SharedShardMetadataController sharedMetadataController = coreContainer.getSharedStoreManager().getSharedShardMetadataController();
     DocCollection collection = zkController.getClusterState().getCollection(collectionName);
-    CoreStorageClient blobClient = zkController.getBlobStorageProvider().getDefaultClient();
+    CoreStorageClient blobClient = coreContainer.getSharedStoreManager().getBlobStorageProvider().getDefaultClient();
     log.info("sync intialized for collection=" + collectionName + " shard=" + shardName + " coreName=" + coreName);
 
     CoreSyncStatus syncStatus = CoreSyncStatus.FAILURE;
@@ -61,7 +61,7 @@ public static void  syncLocalCoreWithSharedStore( String collectionName, String
 
         if (resolutionResult.getFilesToPull().size() > 0) {
 
-          BlobDeleteManager deleteManager = zkController.getBlobDeleteManager();
+          BlobDeleteManager deleteManager = coreContainer.getSharedStoreManager().getBlobDeleteManager();
           CorePushPull cp = new CorePushPull(blobClient, deleteManager, pushPullData, resolutionResult, serverMetadata, blobstoreMetadata);
           cp.pullUpdateFromBlob(/* waitForSearcher */ true);
         } else {
