diff --git a/old_test.java b/new_test.java
index 931d8b4..e378171 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,20 +1,12 @@
 @Test
   public void testSyncLocalCoreWithSharedStore_syncEquivalent() throws Exception {
-    CoreContainer ccSpy = Mockito.spy(cc);
-    ZkController zkSpy = Mockito.spy(zk);
-    BlobStorageProvider blobProvideMock = Mockito.mock(BlobStorageProvider.class);
-    CoreStorageClient blobClientMock = Mockito.mock(CoreStorageClient.class);
-    
-    Mockito.when(ccSpy.getZkController()).thenReturn(zkSpy);
-    Mockito.when(zkSpy.getBlobStorageProvider()).thenReturn(blobProvideMock);
-    Mockito.when(blobProvideMock.getDefaultClient()).thenReturn(blobClientMock);
-    
-    SharedShardMetadataController sharedMetadataController = zk.getSharedShardMetadataController();
+    CoreStorageClient blobClientSpy = Mockito.spy(storageClient);    
+    SharedShardMetadataController sharedMetadataController = cc.getSharedStoreManager().getSharedShardMetadataController();
     sharedMetadataController.ensureMetadataNodeExists(collectionName, shardName);
     try {
-      BlobStoreUtils.syncLocalCoreWithSharedStore(collectionName, newReplica.getCoreName(),shardName, ccSpy);
-      verify(blobClientMock,never()).pullCoreMetadata(anyString(),anyString());
+      BlobStoreUtils.syncLocalCoreWithSharedStore(collectionName, newReplica.getCoreName(), shardName, cc);
+      verify(blobClientSpy, never()).pullCoreMetadata(anyString(), anyString());
     } catch (Exception ex){
-      fail("syncLocalCoreWithSharedStore failed with exception: " + ex.getMessage() );
+      fail("syncLocalCoreWithSharedStore failed with exception: " + ex.getMessage());
     } 
   }
\ No newline at end of file
