diff --git a/old_test.java b/new_test.java
index fc5a6c8..6ed9ad4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,5 @@
 @Test
-    public void testIn_Array() {
+    public void testInArray() {
         Object[] criterions = new Object[] {"a", "b"};
-        assertTrue(instance.in("name", criterions).getQueryCriterions()
-                .contains(new InCriterion("name", criterions)));
+    	assertEquals(new InCriterion("name", criterions), instance.in("name", criterions).getQueryCriterion());
     }
\ No newline at end of file
