diff --git a/old_product.java b/new_product.java
index 6bcfa51..981042e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery in(String propName, Collection<? extends Object> value) {
-        addCriterion(criterionBuilder.in(propName, value));
+        criterion = criterion.and(criterionBuilder.in(propName, value));
         return this;
     }
\ No newline at end of file
