diff --git a/old_test.java b/new_test.java
index 1231aa3..5a979bb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	public void convertTo_SB() {
-		Class[] types = new Class[]{ImageUInt8.class, ImageUInt16.class, ImageFloat32.class};
+		Class[] types = new Class[]{GrayU8.class, GrayU16.class, GrayF32.class};
 
 		for (Class t : types) {
 			ImageBase image = GeneralizedImageOps.createSingleBand(t, imgWidth, imgHeight);
