diff --git a/old_test.java b/new_test.java
index 29e7793..0cbd773 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
         Mockito.when(dataSetProvider.find(Mockito.any(DataSet.ID.class))).thenReturn(Optional.of(dataSet));
         Mockito.when(dataSetId.toString()).thenReturn("dataSet1");
 
-        final Response response = controller.getDataSet("dataSet1", true);
+        final Response response = controller.getDataSet("dataSet1");
         
         assertThat(response.getEntity())
             .isNotNull()
