diff --git a/old_product.java b/new_product.java
index 4588efe..f90d21e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,6 @@
-public Optional<DataSet> getDataSet(@Nonnull final String id, boolean encryptedCredentials) {
+public Optional<DataSet> getDataSet(@Nonnull final String id) {
         try {
-            return Optional.of(get(path("catalog", "dataset", id), 
-                                   uri -> uri.queryParam("encrypt", encryptedCredentials),
-                                   DataSet.class));
+            return Optional.of(get(path("catalog", "dataset", id), DataSet.class));
         } catch (final HttpClientErrorException e) {
             if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
                 return Optional.empty();
