diff --git a/old_test.java b/new_test.java
index 9465d86..a2d0535 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,8 @@
 @Test
     public final void testToIterator(){
-
         // *************************逗号分隔的数组********************************
         LOGGER.info(StringUtils.center("逗号分隔的数组", 60, "*"));
-        Iterator<?> iterator = ObjectUtil.toIterator("1,2");
-        printIterator(iterator);
+        LOGGER.debug(JsonUtil.format(ObjectUtil.toIterator("1,2")));
 
         // ************************map*********************************
         LOGGER.info(StringUtils.center("map", 60, "*"));
@@ -13,22 +11,19 @@
         map.put("a", "1");
         map.put("b", "2");
 
-        iterator = ObjectUtil.toIterator(map);
-        printIterator(iterator);
+        LOGGER.debug(JsonUtil.format(ObjectUtil.toIterator(map)));
 
         // ***************************array******************************
         LOGGER.info(StringUtils.center("array", 60, "*"));
         Object[] array = { "5", 8 };
-        iterator = ObjectUtil.toIterator(array);
-        printIterator(iterator);
+        LOGGER.debug(JsonUtil.format(ObjectUtil.toIterator(array)));
         // ***************************collection******************************
         LOGGER.info(StringUtils.center("collection", 60, "*"));
         Collection<String> collection = new ArrayList<String>();
         collection.add("aaaa");
         collection.add("nnnnn");
 
-        iterator = ObjectUtil.toIterator(collection);
-        printIterator(iterator);
+        LOGGER.debug(JsonUtil.format(ObjectUtil.toIterator(collection)));
 
         // **********************enumeration***********************************
         LOGGER.info(StringUtils.center("enumeration", 60, "*"));
