diff --git a/old_product.java b/new_product.java
index 9176c65..a85886e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,10 +3,14 @@
         if (null == object){
             return null;
         }
-        // object 不是空
+
+        // 逗号分隔的字符串
+        if (object instanceof String){
+            return toIterator(ConvertUtil.toStrings(object));
+        }
         // 数组
-        if (object.getClass().isArray()){
-            return ArrayUtil.toIterator(object);
+        else if (object.getClass().isArray()){
+            return org.apache.commons.collections4.IteratorUtils.arrayIterator(object);
         }
         // Iterator
         else if (object instanceof Iterator){
@@ -26,11 +30,5 @@
             Set<T> keySet = ((Map<T, ?>) object).keySet();
             return keySet.iterator();
         }
-        // 逗号分隔的字符串
-        else if (object instanceof String){
-            String[] strings = object.toString().split(",");
-            return ArrayUtil.toIterator(strings);
-        }else{
-            throw new IllegalArgumentException("param object:[" + object + "] don't support convert to Iterator.");
-        }
+        throw new IllegalArgumentException("param object:[" + object + "] don't support convert to Iterator.");
     }
\ No newline at end of file
