diff --git a/old_product.java b/new_product.java
index aa81bf9..bd07c6a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -37,7 +37,7 @@ public NetworkInterface getNetworkInterface() throws SocketException {
                     && (InetAddress.preferIPv6Addresses() == true)) {
                 theAddresses[0] = Inet6Address.ANY;
             } else {
-                theAddresses[0] = InetAddress.ANY;
+                theAddresses[0] = Inet4Address.ANY;
             }
             return new NetworkInterface(null, null, theAddresses,
                     NetworkInterface.UNSET_INTERFACE_INDEX);
