diff --git a/old_product.java b/new_product.java
index c53d903..b17accf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,7 @@
 @SuppressWarnings("unchecked")
     public VM kFoldCrossValidation(Dataset trainingData, TP trainingParameters, int k) {
-        if(GeneralConfiguration.DEBUG) {
-            System.out.println("kFoldCrossValidation()");
-        }
+        logger.debug("kFoldCrossValidation()");
+        
         initializeTrainingConfiguration(trainingParameters);
         
         return modelValidator.kFoldCrossValidation(trainingData, k, dbName, knowledgeBase.getDbConf(), this.getClass(), knowledgeBase.getTrainingParameters());
