diff --git a/old_test.java b/new_test.java
index c1d105d..963d140 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,14 +2,14 @@
     public void testSubscribe() {
         Logger.getLogger(ControllerTest.class.getName()).info("subscribe");
 
-        String callback = "http://localhost/doNothing";
-        String topic = "http://feeds.feedburner.com/screaming-penguin";
-        long lease_seconds = -1;
-        String secret = null;
-        String verify_token = "MyVoiceIsMyPassport";
-        HubDAO dao = new InMemoryHubDAO();
-        Notifier notifier = null;
-        FeedFetcher fetcher = new HttpURLFeedFetcher(HashMapFeedInfoCache.getInstance());
+        final String callback = "http://localhost/doNothing";
+        final String topic = "http://feeds.feedburner.com/screaming-penguin";
+        final long lease_seconds = -1;
+        final String secret = null;
+        final String verify_token = "MyVoiceIsMyPassport";
+        final HubDAO dao = new InMemoryHubDAO();
+        final Notifier notifier = null;
+        final FeedFetcher fetcher = new HttpURLFeedFetcher(HashMapFeedInfoCache.getInstance());
         Hub instance = new Hub(dao, new AlwaysVerifier(), notifier, fetcher);
 
         Boolean result = instance.subscribe(callback, topic, "sync", lease_seconds, secret, verify_token);
@@ -26,7 +26,7 @@
         try {
             instance.subscribe(null, topic, "async", lease_seconds, secret, verify_token);
             fail();
-        } catch (HttpStatusCodeException e) {
+        } catch (final HttpStatusCodeException e) {
             assertEquals(400, e.getStatus());
             Logger.getLogger(ControllerTest.class.getName()).info(e.getMessage());
         }
@@ -34,7 +34,7 @@
         try {
             instance.subscribe(callback, null, "async", lease_seconds, secret, verify_token);
             fail();
-        } catch (HttpStatusCodeException e) {
+        } catch (final HttpStatusCodeException e) {
             assertEquals(400, e.getStatus());
             Logger.getLogger(ControllerTest.class.getName()).info(e.getMessage());
         }
@@ -42,7 +42,7 @@
         try {
             instance.subscribe(callback, topic, "foo", lease_seconds, secret, verify_token);
             fail();
-        } catch (HttpStatusCodeException e) {
+        } catch (final HttpStatusCodeException e) {
             assertEquals(400, e.getStatus());
             Logger.getLogger(ControllerTest.class.getName()).info(e.getMessage());
         }
@@ -53,7 +53,7 @@
         try {
             result = instance.subscribe(callback, topic, "sync", lease_seconds, secret, verify_token);
             fail();
-        } catch (HttpStatusCodeException e) {
+        } catch (final HttpStatusCodeException e) {
             assertEquals(500, e.getStatus());
         }
     }
\ No newline at end of file
