diff --git a/old_test.java b/new_test.java
index 2e80a6c..35a050d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,33 +1,14 @@
 @Test
 	public void testCheckCollision_BoundingBox_BoundingBox() {
-		BoundingBox a = new BoundingBox(
-			new Vector3(1, 1, 1),
-			new Vector3(3, 3, 3));
-		BoundingBox b = new BoundingBox(
-			new Vector3(2, 2, 2),
-			new Vector3(4, 4, 4));
-
-		boolean result = CollisionHelper.checkCollision(a, b);
-
-		assertTrue(result);
+		BoundingBox a = new BoundingBox(new Vector3(1, 1, 1), new Vector3(3, 3, 3));
+		BoundingBox b = new BoundingBox(new Vector3(2, 2, 2), new Vector3(4, 4, 4));
+		assertTrue(CollisionHelper.checkCollision(a, b));
 
 		//Check shared points
-
-		a = new BoundingBox(
-			new Vector3(4, 4, 4),
-			new Vector3(6, 6, 6));
-
-		result = CollisionHelper.checkCollision(a, b);
-
-		assertTrue(result);
+		a = new BoundingBox(new Vector3(4, 4, 4), new Vector3(6, 6, 6));
+		assertFalse(CollisionHelper.checkCollision(a, b));
 
 		//Check not intersecting
-
-		a = new BoundingBox(
-			new Vector3(6, 6, 6),
-			new Vector3(10, 10, 10));
-
-		result = CollisionHelper.checkCollision(a, b);
-
-		assertFalse(result);
+		a = new BoundingBox(new Vector3(6, 6, 6), new Vector3(10, 10, 10));
+		assertFalse(CollisionHelper.checkCollision(a, b));
 	}
\ No newline at end of file
