diff --git a/old_product.java b/new_product.java
index 8def430..fdc340c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,8 @@
 public static boolean checkCollision(BoundingBox a, BoundingBox b) {
-		return a.min.compareTo(b.max) <= 0 && a.max.compareTo(b.min) >= 0;
+		return a.min.getX() < b.max.getX() &&
+				a.min.getY() < b.max.getY() &&
+				a.min.getZ() < b.max.getZ() &&
+				a.max.getX() > b.min.getX() &&
+				a.max.getY() > b.min.getY() &&
+				a.max.getZ() > b.min.getZ();
 	}
\ No newline at end of file
