diff --git a/old_test.java b/new_test.java
index d765ac3..c7a9c3d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,22 @@
 @Test
   public void event() throws BaseXException {
-    event(RETURN_VALUE, true);
-    event(null, false);
+    // create the event
+    session.execute("create event " + NAME);
+    // watch event
+    for(final ClientSession cs : sessions) {
+      cs.watch(NAME, new EventNotifier() {
+        @Override
+        public void notify(final String data) {
+          assertEquals(RETURN, data);
+        }
+      });
+    }
+    // fire an event
+    session.query("db:event('" + NAME + "', '" + RETURN + "')").execute();
+
+    // all clients unwatch the events
+    for(final ClientSession cs : sessions) cs.unwatch(NAME);
+
+    // drop event
+    session.execute("drop event " + NAME);
   }
\ No newline at end of file
