diff --git a/old_product.java b/new_product.java
index 639006a..f660bcd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,22 @@
-private Iter event(final QueryContext ctx) throws QueryException {
-    String name = expr[0].toString();
-    String msg = "";
-    Value v = expr[1].value(ctx);
-    if(expr.length < 3) {
-      ctx.updating = true;
-      msg = ctx.value(expr[1]).toString();
-    } else {
-      msg = ctx.value(expr[2]).toString();
+private Item event(final QueryContext ctx) throws QueryException {
+    final byte[] name = checkStr(expr[0], ctx);
+    if(expr.length == 3) expr[2].value(ctx);
+
+    final ArrayOutput ao = new ArrayOutput();
+    try {
+      // run serialization
+      final XMLSerializer xml = new XMLSerializer(ao);
+      final ValueIter ir = expr[1].value(ctx).iter();
+      for(Item it; (it = ir.next()) != null;) it.serialize(xml);
+      xml.close();
+    } catch(final SerializerException ex) {
+      throw new QueryException(input, ex);
+    } catch(final IOException ex) {
+      SERANY.thrw(input, ex);
     }
-    ctx.context.events.notify(ctx.context.session, name, msg);
-    return v.iter();
+    // throw exception if event is unknown
+    if(!ctx.context.events.notify(ctx.context, name, ao.toArray())) {
+      NOEVENT.thrw(input, name);
+    }
+    return null;
   }
\ No newline at end of file
