diff --git a/old_test.java b/new_test.java
index 6c5745e..46bf43b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,15 +1,15 @@
 @Test(groups={"tck"})
-    public void test_maxLengthInDays() {
-        assertEquals(Month.JANUARY.maxLengthInDays(), 31);
-        assertEquals(Month.FEBRUARY.maxLengthInDays(), 29);
-        assertEquals(Month.MARCH.maxLengthInDays(), 31);
-        assertEquals(Month.APRIL.maxLengthInDays(), 30);
-        assertEquals(Month.MAY.maxLengthInDays(), 31);
-        assertEquals(Month.JUNE.maxLengthInDays(), 30);
-        assertEquals(Month.JULY.maxLengthInDays(), 31);
-        assertEquals(Month.AUGUST.maxLengthInDays(), 31);
-        assertEquals(Month.SEPTEMBER.maxLengthInDays(), 30);
-        assertEquals(Month.OCTOBER.maxLengthInDays(), 31);
-        assertEquals(Month.NOVEMBER.maxLengthInDays(), 30);
-        assertEquals(Month.DECEMBER.maxLengthInDays(), 31);
+    public void test_maxLength() {
+        assertEquals(Month.JANUARY.maxLength(), 31);
+        assertEquals(Month.FEBRUARY.maxLength(), 29);
+        assertEquals(Month.MARCH.maxLength(), 31);
+        assertEquals(Month.APRIL.maxLength(), 30);
+        assertEquals(Month.MAY.maxLength(), 31);
+        assertEquals(Month.JUNE.maxLength(), 30);
+        assertEquals(Month.JULY.maxLength(), 31);
+        assertEquals(Month.AUGUST.maxLength(), 31);
+        assertEquals(Month.SEPTEMBER.maxLength(), 30);
+        assertEquals(Month.OCTOBER.maxLength(), 31);
+        assertEquals(Month.NOVEMBER.maxLength(), 30);
+        assertEquals(Month.DECEMBER.maxLength(), 31);
     }
\ No newline at end of file
