diff --git a/old_test.java b/new_test.java
index c18d71d..58c426f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,6 +11,7 @@
 				anyInt())).thenReturn(samplePromise.future());
 
 		ExecutionGraph graph = mock(ExecutionGraph.class);
+		when(graph.getState()).thenReturn(JobStatus.RUNNING);
 
 		// Same Thread execution context
 		when(graph.getExecutionContext()).thenReturn(new ExecutionContext() {
@@ -51,7 +52,7 @@
 				sampleCoordinator, 9999, numSamples, delayBetweenSamples);
 
 		// Trigger
-		tracker.triggerStackTraceSample(jobVertex);
+		assertTrue("Failed to trigger", tracker.triggerStackTraceSample(jobVertex));
 
 		verify(sampleCoordinator).triggerStackTraceSample(
 				eq(taskVertices),
@@ -60,7 +61,7 @@
 				eq(BackPressureStatsTracker.MAX_STACK_TRACE_DEPTH));
 
 		// Trigger again for pending request, should not fire
-		tracker.triggerStackTraceSample(jobVertex);
+		assertFalse("Unexpected trigger", tracker.triggerStackTraceSample(jobVertex));
 
 		assertTrue(tracker.getOperatorBackPressureStats(jobVertex).isEmpty());
 
