diff --git a/old_product.java b/new_product.java
index ecd284c..ccd729f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,5 @@
-public ServiceCall<SpeechModel> getModel(final String modelName) {
-    Validator.notNull(modelName, "name cannot be null");
-
-    Request request = RequestBuilder.get(String.format(PATH_MODEL, modelName)).build();
-    return createServiceCall(request, ResponseConverterUtils.getObject(SpeechModel.class));
+public ServiceCall<SpeechModel> getModel(GetModelOptions getModelOptions) {
+    Validator.notNull(getModelOptions, "getModelOptions cannot be null");
+    RequestBuilder builder = RequestBuilder.get(String.format("/v1/models/%s", getModelOptions.modelId()));
+    return createServiceCall(builder.build(), ResponseConverterUtils.getObject(SpeechModel.class));
   }
\ No newline at end of file
