diff --git a/old_test.java b/new_test.java
index 0b3bcc3..f9b7344 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,17 @@
 @Test
   public void testGetModel() throws Exception {
-    final SpeechModel speechModel = new SpeechModel("not-a-real-Model");
+    final SpeechModel speechModel = new SpeechModel();
+    speechModel.setName("not-a-real-Model");
     speechModel.setRate(8000);
 
     final MockResponse mockResponse =
         new MockResponse().addHeader(CONTENT_TYPE, HttpMediaType.APPLICATION_JSON).setBody(GSON.toJson(speechModel));
 
     server.enqueue(mockResponse);
-    SpeechModel model = service.getModel("not-a-real-Model").execute();
+    GetModelOptions getOptionsString = new GetModelOptions.Builder()
+        .modelId("not-a-real-Model")
+        .build();
+    SpeechModel model = service.getModel(getOptionsString).execute();
     RecordedRequest request = server.takeRequest();
 
     assertNotNull(model);
@@ -15,7 +19,10 @@
     assertEquals(PATH_MODELS + "/" + speechModel.getName(), request.getPath());
 
     server.enqueue(mockResponse);
-    model = service.getModel(speechModel.getName()).execute();
+    GetModelOptions getOptionsGetter = new GetModelOptions.Builder()
+        .modelId(speechModel.getName())
+        .build();
+    model = service.getModel(getOptionsGetter).execute();
     request = server.takeRequest();
 
     assertNotNull(model);
