diff --git a/old_product.java b/new_product.java
index 5280c33..6bccf14 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1,3 @@
-public void register(String name, LongSupplier supplier, @Nullable String description);
\ No newline at end of file
+public void register(String name, LongSupplier supplier, @Nullable String desc) {
+        addMetric(name, new LongGauge(metricName(grpName, name), desc, nonThrowableSupplier(supplier, log)));
+    }
\ No newline at end of file
