diff --git a/old_product.java b/new_product.java
index 5218446..7538464 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-private Message2 encodeDecode(final Message2 m1) throws Exception {
-        AtomicReference<Message2> m2 = new AtomicReference<Message2>();
+private Message encodeDecode(final Message m1) throws Exception {
+        AtomicReference<Message> m2 = new AtomicReference<Message>();
         TomP2POutbound encoder = new TomP2POutbound(true, new DefaultSignatureFactory());
         ByteBuf buf = Unpooled.buffer();
         ChannelHandlerContext ctx = mockChannelHandlerContext(buf, m2);
