diff --git a/old_test.java b/new_test.java
index ebd665f..c07a462 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testEncodeDecode() throws Exception {
         // encode
-        Message2 m1 = Utils2.createDummyMessage();
-        Message2 m2 = encodeDecode(m1);
+        Message m1 = Utils2.createDummyMessage();
+        Message m2 = encodeDecode(m1);
         compareMessage(m1, m2);
     }
\ No newline at end of file
