diff --git a/old_product.java b/new_product.java
index 091d285..3f36864 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
-public static double Nbar(TransposeDataList clusterIdList) {
+public static double nBar(TransposeDataList clusterIdList) {
         int populationM = clusterIdList.size();
         
-        double Nbar = 0.0;
+        double nBar = 0.0;
         for(Map.Entry<Object, FlatDataList> entry : clusterIdList.entrySet()) {
-            Nbar += (double)entry.getValue().size()/populationM;
+            nBar += (double)entry.getValue().size()/populationM;
         }
 
-        return Nbar;
+        return nBar;
     }
\ No newline at end of file
