diff --git a/old_test.java b/new_test.java
index 7b495fe..cea3ed1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,6 @@
         TransposeDataList clusterIdList = generateClusterIdList();
         
         double expResult = 10.0;
-        double result = ClusterSampling.Nbar(clusterIdList);
+        double result = ClusterSampling.nBar(clusterIdList);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
