diff --git a/old_test.java b/new_test.java
index 2ed7ef5..c0a4d99 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,22 +4,29 @@
 		SecurePackageResourceGuard guard = new SecurePackageResourceGuard();
 		guard.setAllowAccessToWebInfResources(false);
 		guard.addPattern("+*.gif");
-		assertTrue(guard.accept(Application.class, "test.gif"));
-		assertTrue(guard.accept(Application.class, "mydir/test.gif"));
+		assertTrue(guard.accept(Application.class,
+			Packages.absolutePath(Application.class, "test.gif")));
+		assertTrue(guard.accept(Application.class,
+			Packages.absolutePath(Application.class, "mydir/test.gif")));
 		assertTrue(guard.accept(Application.class, "/root/mydir/test.gif"));
-		assertTrue(guard.accept(Application.class, "../test.gif"));
-		assertTrue(guard.accept(Application.class, "../../test.gif"));
+		assertTrue(guard.accept(Application.class,
+			Packages.absolutePath(Application.class, "../test.gif")));
+		assertTrue(guard.accept(Application.class,
+			Packages.absolutePath(Application.class, "../../test.gif")));
 
 		// web-inf (root package)
-		assertFalse(guard.accept(Application.class, "../../../test.gif"));
+		assertFalse(guard.accept(Application.class,
+			Packages.absolutePath(Application.class, "../../../test.gif")));
 		guard.setAllowAccessToWebInfResources(true);
-		assertTrue(guard.accept(Application.class, "../../../test.gif"));
+		assertTrue(guard.accept(Application.class,
+			Packages.absolutePath(Application.class, "../../../test.gif")));
 
 		boolean hit = false;
 		try
 		{
 			// you can not go below root
-			assertTrue(guard.accept(Application.class, "../../../../test.gif"));
+			assertTrue(guard.accept(Application.class,
+				Packages.absolutePath(Application.class, "../../../../test.gif")));
 		}
 		catch (IllegalArgumentException ex)
 		{
