diff --git a/old_product.java b/new_product.java
index 404e13b..9ef2416 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,4 +4,4 @@
    @Consumes(MediaType.APPLICATION_JSON)
    ListenableFuture<ServerDetails> cloneServer(@FormParam("serverid") String serverid,
                                                @FormParam("hostname") String hostname,
-                                               ServerCloneOptions... options);
\ No newline at end of file
+                                               CloneServerOptions... options);
\ No newline at end of file
