diff --git a/old_product.java b/new_product.java
index 907cfc0..c1ed7ab 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,22 +1,15 @@
-private synchronized Item createDir(final QueryContext ctx) throws QueryException {
-    // resolve symbolic links
-    final File path = checkFile(0, ctx);
-    File f;
-    try {
-      f = path.getCanonicalFile();
-    } catch(final IOException ex) {
-      throw FILE_IE_PATH.get(info, path);
-    }
+private synchronized Item createDir(final QueryContext ctx) throws QueryException, IOException {
+    final Path path = absolute(checkPath(0, ctx));
 
     // find lowest existing path
-    while(!f.exists()) {
-      f = f.getParentFile();
-      if(f == null) throw FILE_IE_PATH.get(info, path);
+    for(Path p = path; p != null;) {
+      if(Files.exists(p)) {
+        if(Files.isRegularFile(p)) throw FILE_EXISTS.get(info, p);
+        break;
+      }
+      p = p.getParent();
     }
-    // warn if lowest path points to a file
-    if(f.isFile()) throw FILE_E.get(info, path);
 
-    // only create directories if path does not exist yet
-    if(!path.exists() && !path.mkdirs()) throw FILE_IE_DIR.get(info, path);
+    Files.createDirectories(path);
     return null;
   }
\ No newline at end of file
