diff --git a/old_product.java b/new_product.java
index 387d559..8119873 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public BasicVertex<I, V, E, M> instantiateVertex(
+public Vertex<I, V, E, M> instantiateVertex(
       I vertexId, V vertexValue, Map<I, E> edges, Iterable<M> messages) {
     MutableVertex<I, V, E, M> mutableVertex =
         (MutableVertex<I, V, E, M>) BspUtils
