diff --git a/old_test.java b/new_test.java
index 4e78448..fdedfa7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,7 +11,7 @@
     GraphState<LongWritable, IntWritable, FloatWritable, IntWritable> gs =
         new GraphState<LongWritable, IntWritable,
         FloatWritable, IntWritable>();
-    BasicVertex<LongWritable, IntWritable, FloatWritable, IntWritable> vertex =
+    Vertex<LongWritable, IntWritable, FloatWritable, IntWritable> vertex =
         BspUtils.createVertex(job.getConfiguration());
     vertex.initialize(null, null, null, null);
     System.out.println("testInstantiateVertex: Got vertex " + vertex +
