diff --git a/old_product.java b/new_product.java
index c49af71..d9a4849 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public static List<Class<?>> convertClassNamesToClasses(final List<String> class
         if (classNames == null) {
             return null;
         }
-        final List<Class<?>> classes = new ArrayList<Class<?>>(classNames.size());
+        final List<Class<?>> classes = new ArrayList<>(classNames.size());
         for (final String className : classNames) {
             try {
                 classes.add(Class.forName(className));
