diff --git a/old_test.java b/new_test.java
index 406e029..5e27005 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void test_convertClassNamesToClasses_List() {
-        final List<String> list = new ArrayList<String>();
+        final List<String> list = new ArrayList<>();
         List<Class<?>> result = ClassUtils.convertClassNamesToClasses(list);
         assertEquals(0, result.size());
 
