diff --git a/old_test.java b/new_test.java
index 50c07ca..6bd4497 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,7 +13,7 @@
 			alg.segment(input,expected);
 
 			// provide an output which is a sub-image
-			ImageSInt32 found = new ImageSInt32(width+3,height+2).subimage(2,1,width+2,height+1,null);
+			ImageSInt32 found = new ImageSInt32(width+3,height+2).subimage(2,1,width+2,height+1);
 			alg.segment(input,found);
 			BoofTesting.assertEquals(expected,found,0);
 
