diff --git a/old_product.java b/new_product.java
index 52b2ef9..421b361 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,7 +15,7 @@ public void setCommands(final List<Command> commands) {
             for (final Command command : this.commands) {
                 Set<Cluster> clusters = command.getClusters();
                 if (clusters == null) {
-                    clusters = new HashSet<Cluster>();
+                    clusters = new HashSet<>();
                     command.setClusters(clusters);
                 }
                 if (!clusters.contains(this)) {
