diff --git a/old_test.java b/new_test.java
index 9f16b6f..17ffbad 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testOnCreateOrUpdateCluster() throws GenieException {
+    public void testOnCreateOrUpdateCluster() throws GeniePreconditionException {
         this.c = new Cluster(NAME, USER, ClusterStatus.UP, CLUSTER_TYPE, this.configs, VERSION);
         this.c.onCreateOrUpdateCluster();
     }
\ No newline at end of file
