diff --git a/old_product.java b/new_product.java
index 91c5cf7..105f553 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,18 +1,19 @@
 private boolean match(final int[] t, final int tp, final int qp) {
-    int ql = qp;
-    int tl = tp;
-    while(ql < wc.length) {
-      if(wc[ql] == DOT) {
-        int n = min[ql];
-        final int m = max[ql++];
+    int qi = qp;
+    int ti = tp;
+    final int tl = t.length;
+    final int wl = wc.length;
+    while(qi < wl) {
+      if(wc[qi] == DOT) {
+        int n = min[qi];
+        final int m = max[qi++];
         // recursively evaluates wildcards (non-greedy)
-        while(!match(t, tl + n, ql))
-          if(tl + ++n > t.length) return false;
+        while(!match(t, ti + n, qi)) if(ti + ++n > tl) return false;
         if(n > m) return false;
-        tl += n;
+        ti += n;
       } else {
-        if(tl >= t.length || t[tl++] != wc[ql++]) return false;
+        if(ti >= tl || t[ti++] != wc[qi++]) return false;
       }
     }
-    return tl == t.length;
+    return ti == tl;
   }
\ No newline at end of file
