diff --git a/old_product.java b/new_product.java
index fa16cd2..2de155c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-private void computeHistogram(int c_x, int c_y, double sigma) {
+void computeHistogram(int c_x, int c_y, double sigma) {
 		int r = (int)Math.ceil(sigma * sigmaToRadius);
 
 		// specify the area being sampled
@@ -27,8 +27,9 @@ private void computeHistogram(int c_x, int c_y, double sigma) {
 
 				// edge intensity and angle
 				double magnitude = Math.sqrt(dx*dx + dy*dy);
-				double theta = Math.atan2(dy,dx) + Math.PI;
-				// weight
+				double theta = UtilAngle.domain2PI(Math.atan2(dy,dx));
+
+				// weight from gaussian
 				double weight = computeWeight( x-c_x, y-c_y , sigma );
 
 				// histogram index
