diff --git a/old_test.java b/new_test.java
index cc16678..4ff8048 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,23 @@
 @Test
-	public void computeHistogram_single() {
-		fail("Implement");
+	public void computeHistogram() {
+		ImageFloat32 derivX = new ImageFloat32(width, height);
+		ImageFloat32 derivY = new ImageFloat32(width, height);
+		int N = 36;
+		OrientationHistogramSift2 alg = new OrientationHistogramSift2(N, 2.5, 1);
+		alg.setImageGradient(derivX, derivY);
+
+		for (int degrees = 5; degrees < 360; degrees+=10) {
+			double theta = UtilAngle.degreeToRadian(degrees);
+
+			int bin = degrees * N / 360;
+
+			ImageMiscOps.fill(derivX, (float) Math.cos(theta));
+			ImageMiscOps.fill(derivY, (float) Math.sin(theta));
+
+			alg.computeHistogram(20, 15, 3);
+			checkHistogram(N, alg, theta, bin);
+
+			alg.computeHistogram(0, 0, 3);
+			checkHistogram(N, alg, theta, bin);
+		}
 	}
\ No newline at end of file
