diff --git a/old_product.java b/new_product.java
index c1d0fdd..5dadafa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ long includeExistingKey(long segmentId, UUID keyHash, long segmentOffset) {
         int generation;
         synchronized (this.segmentCaches) {
             generation = this.currentCacheGeneration;
-            cache = this.segmentCaches.computeIfAbsent(segmentId, s -> new SegmentKeyCache(s, this.cache));
+            cache = this.segmentCaches.computeIfAbsent(segmentId, s -> new SegmentKeyCache(s, this.cacheStorage));
         }
 
         return cache.includeExistingKey(keyHash, segmentOffset, generation);
