diff --git a/old_test.java b/new_test.java
index a1c265f..2980ca8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,7 @@
 @Test
     public void testIncludeExistingKey() {
         @Cleanup
-        val cacheFactory = new InMemoryCacheFactory();
-        @Cleanup
-        val keyCache = new ContainerKeyCache(CONTAINER_ID, cacheFactory);
+        val keyCache = new ContainerKeyCache(this.cacheStorage);
         val expectedResult = new HashMap<TestKey, CacheBucketOffset>();
 
         // Insert.
