diff --git a/old_test.java b/new_test.java
index 3510f0e..8e937e5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -26,7 +26,7 @@ protected void publishApi(FrameworkMethod method) {
             if (backEnd == null) {
                 backEnd = getTestClass().getJavaClass().getAnnotation(BackEndApi.class);
             }
-            Class<? extends IPolicyTestBackEndApi> backEndApi = null;
+            Class<? extends IPolicyTestBackEndApi> backEndApi;
             if (backEnd == null) {
                 backEndApi = EchoBackEndApi.class;
             } else {
