diff --git a/old_test.java b/new_test.java
index ac7d829..265ab22 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
-  public void dirName() {
+  public void parent() {
     // check with a simple path
-    assertEquals(norm(PATH), norm(query(_FILE_DIR_NAME.args(PATH1))).toLowerCase(Locale.ENGLISH));
+    assertEquals(norm(PATH), norm(query(_FILE_PARENT.args(PATH1))).toLowerCase(Locale.ENGLISH));
     // check with an empty path
-    query(_FILE_DIR_NAME.args(""), '.' + File.separator);
+    query(EMPTY.args(_FILE_PARENT.args("")), "true");
     // check with a path without directory separators
-    query(_FILE_DIR_NAME.args(NAME), '.' + File.separator);
+    query(EMPTY.args(_FILE_PARENT.args(NAME)), "true");
     // check with a path without directory separators
-    query(_FILE_DIR_NAME.args("/"), File.separator);
+    query(EMPTY.args(_FILE_PARENT.args("/")), "true");
   }
\ No newline at end of file
