diff --git a/old_product.java b/new_product.java
index 2ff6a83..7ad9c16 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,4 @@
-private Str dirName(final QueryContext ctx) throws QueryException {
-    final File file = checkFile(0, ctx);
-    String par = file.getParent();
-    if(par == null) par = file.toString().contains(File.separator) ? "" : ".";
-    return Str.get(dir(par));
+private Str parent(final QueryContext ctx) throws QueryException {
+    final String parent = checkFile(0, ctx).getParent();
+    return parent == null ? null : Str.get(dir(parent));
   }
\ No newline at end of file
