diff --git a/old_product.java b/new_product.java
index d2574c3..871b1d4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,10 +8,8 @@ private Iter node(final QueryContext ctx, final boolean id)
       public Item next() throws QueryException {
         final Item it = ir.next();
         if(it == null) return null;
-        final Nod node = checkNode(it);
-        if(!(node instanceof DBNode)) NODBCTX.thrw(input, FNDb.this);
-        final DBNode dbnode = (DBNode) node;
-        return Itr.get(id ? dbnode.data.id(dbnode.pre) : dbnode.pre);
+        final DBNode node = checkDBNode(it);
+        return Itr.get(id ? node.data.id(node.pre) : node.pre);
       }
     };
   }
\ No newline at end of file
