diff --git a/old_test.java b/new_test.java
index bd0b0c9..32a4d66 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,10 @@
 @Test
   public void testNode() throws BaseXException {
     // wrong arguments
-    args("db:node-id", DBNode.class);
-    args("db:node-pre", DBNode.class);
+    args("db:node-id", (Class<?>) null);
+    args("db:node-pre", (Class<?>) null);
+    error("db:node-id(1)", "BASX0002");
+    error("db:node-pre(1)", "BASX0002");
 
     // test results
     query("db:node-id(/html)", "1");
