diff --git a/old_test.java b/new_test.java
index 3b2cdbe..6fc9b48 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,9 @@
 @Test
-    public void testListDocuments() {
-        Map<String,Object> params = new HashMap<String, Object>();
-        params.put(ConceptInsights.ACCOUNT_ID, "public");
-        params.put(ConceptInsights.CORPUS, "ibmresearcher");
-        params.put(ConceptInsights.CURSOR, 0);
-        params.put(ConceptInsights.LIMIT, 20);
+	public void testListDocuments() {
+		Map<String, Object> params = new HashMap<String, Object>();
+		params.put(ConceptInsights.CURSOR, 0);
+		params.put(ConceptInsights.LIMIT, 20);
 
-        Documents documents = service.listDocuments(params);
-        Assert.assertNotNull(documents);
-    }
\ No newline at end of file
+		Documents documents = service.listDocuments(Corpus.TED_TALKS, params);
+		Assert.assertNotNull(documents);
+	}
\ No newline at end of file
