diff --git a/old_test.java b/new_test.java
index ddb8db5..b7f8695 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,10 +12,10 @@
     assertSame(result, entity);
     verify(mockEntityManagerProvider, entityManager);
 
-    List<AlertDefinitionEntity> definitions = realDAO.findAll();
+    List<AlertDefinitionEntity> definitions = dao.findAll();
     Assert.assertNotNull(definitions);
     AlertDefinitionEntity definition = definitions.get(2);
-    AlertDefinitionEntity retrieved = realDAO.findById(definition.getDefinitionId());
+    AlertDefinitionEntity retrieved = dao.findById(definition.getDefinitionId());
 
     Assert.assertEquals(definition, retrieved);
   }
\ No newline at end of file
