diff --git a/old_product.java b/new_product.java
index 7d05eff..89ad084 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
-@RequiresSession
-  public AlertDefinitionEntity findById(long definitionId) {
+public AlertDefinitionEntity findById(long definitionId) {
     return entityManagerProvider.get().find(AlertDefinitionEntity.class,
         definitionId);
   }
\ No newline at end of file
