diff --git a/old_product.java b/new_product.java
index 6db5b31..55131d2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 public List<AlignmentInterval.Row> packAlignments(
             Iterator<Alignment> iter, int end, boolean pairAlignments,
-            AlignmentTrack.SortOption groupBy) {
+            AlignmentTrack.SortOption groupBy,
+            int maxLevels) {
 
 
         List<Row> alignmentRows = new ArrayList(1000);
@@ -9,7 +10,7 @@ public List<AlignmentInterval.Row> packAlignments(
         }
 
         if (groupBy == null) {
-            pack(iter, end, pairAlignments, lengthComparator, alignmentRows);
+            pack(iter, end, pairAlignments, lengthComparator, alignmentRows, maxLevels);
         } else {
             // Separate by group
             List<Alignment> nullGroup = new ArrayList();
@@ -31,9 +32,9 @@ public List<AlignmentInterval.Row> packAlignments(
             Collections.sort(keys);
             for (String key : keys) {
                 List<Alignment> group = groupedAlignments.get(key);
-                pack(group.iterator(), end, pairAlignments, lengthComparator, alignmentRows);
+                pack(group.iterator(), end, pairAlignments, lengthComparator, alignmentRows, maxLevels);
             }
-            pack(nullGroup.iterator(), end, pairAlignments, lengthComparator, alignmentRows);
+            pack(nullGroup.iterator(), end, pairAlignments, lengthComparator, alignmentRows, maxLevels);
         }
 
         return alignmentRows;
