diff --git a/old_test.java b/new_test.java
index c92f63f..d657640 100644
--- a/old_test.java
+++ b/new_test.java
@@ -24,6 +24,6 @@
         int maxLevels = 1000;
 
 
-        List<AlignmentInterval.Row> result = (new AlignmentPacker()).packAlignments(iter, end, false);
+        List<AlignmentInterval.Row> result = (new AlignmentPacker()).packAlignments(iter, end, false, null, 10000);
 
     }
\ No newline at end of file
