diff --git a/old_product.java b/new_product.java
index 6c8571a..d89cfce 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
-void free(TachyonURI path, FreeOptions options)
-      throws FileDoesNotExistException, IOException, TachyonException;
\ No newline at end of file
+void free(AlluxioURI path, FreeOptions options)
+      throws FileDoesNotExistException, IOException, AlluxioException;
\ No newline at end of file
