diff --git a/old_test.java b/new_test.java
index 6fda713..d875c1f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,8 +11,8 @@
 
     // create nested directory
     InodeTree.CreatePathResult createResult = createPath(mTree, NESTED_URI, dirOptions);
-    List<Inode<?>> modified = createResult.getModified();
-    List<Inode<?>> created = createResult.getCreated();
+    List<InodeView> modified = createResult.getModified();
+    List<InodeView> created = createResult.getCreated();
 
     // 1 modified directory
     assertEquals(1, modified.size());
