diff --git a/old_product.java b/new_product.java
index 6f78f4c..75d6040 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,3 @@
-@ServiceMethod(returns = ReturnType.COLLECTION)
-    public Flux<String> getPartitionIds() {
-        return getProperties().flatMapMany(properties -> Flux.fromArray(properties.getPartitionIds()));
+public Flux<String> getPartitionIds() {
+        return getProperties().flatMapMany(properties -> Flux.fromIterable(properties.getPartitionIds()));
     }
\ No newline at end of file
