diff --git a/old_product.java b/new_product.java
index d813fd8..e2498bd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
-public List<TranslationModel> getModels(final Boolean showDefault, final String source,
-      final String target) {
+public ServiceCall<List<TranslationModel>> getModels(final Boolean showDefault, final String source, final String target) {
     final RequestBuilder requestBuilder = RequestBuilder.get(PATH_MODELS);
 
     if (source != null && !source.isEmpty())
@@ -11,7 +10,5 @@ public List<TranslationModel> getModels(final Boolean showDefault, final String
     if (showDefault != null)
       requestBuilder.withQuery(DEFAULT, showDefault);
 
-    final TranslationModelList models =
-        executeRequest(requestBuilder.build(), TranslationModelList.class);
-    return models.getModels();
+    return createServiceCall(createCall(requestBuilder.build3()), ResponseUtil.getTranslationModelListConverter());
   }
\ No newline at end of file
