diff --git a/old_product.java b/new_product.java
index 4d2c84e..e15ad78 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 private Str resolvePath(final QueryContext ctx) throws QueryException {
     final File path = checkFile(0, ctx);
-    final String abs = path.getAbsolutePath();
-    return Str.get(path.isDirectory() ? dir(abs) : abs);
+    final File abs = path.getAbsoluteFile();
+    return Str.get(abs.isDirectory() ? dir(abs.getPath()) : abs.getPath());
   }
\ No newline at end of file
