diff --git a/old_test.java b/new_test.java
index 2193715..c09fc02 100644
--- a/old_test.java
+++ b/new_test.java
@@ -24,13 +24,13 @@
 						}
 						for (int i = 0; i < rows; i++) {
 							for (int j = 0; j < cols; j++) {
-								UtilPolygons2D_F64.shiftDown(grid.get(i,j).corners);
+								UtilPolygons2D_F64.shiftDown(grid.get(i,j).square);
 							}
 						}
 					}
 					for (int i = 0; i < rows; i++) {
 						for (int j = 0; j < cols; j++) {
-							UtilPolygons2D_F64.flip(grid.get(i, j).corners);
+							UtilPolygons2D_F64.flip(grid.get(i, j).square);
 						}
 					}
 				}
