diff --git a/old_test.java b/new_test.java
index d0a880a..a7fec54 100644
--- a/old_test.java
+++ b/new_test.java
@@ -53,5 +53,7 @@
         } catch (TimeoutException e) {
             // If we exceed the timeout, the call is hung and it's a failure
             Assert.fail();
+        } finally {
+            LockServiceLoggerTestUtils.cleanUpLogStateDir();
         }
     }
\ No newline at end of file
