diff --git a/old_test.java b/new_test.java
index bb3607f..bcb7af4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-  public void updateAttribute() throws Exception {
-    final String query = "S:update-attribute('a','b'), S:attribute('a')";
+  public void set() throws Exception {
+    final String query = "S:set('a','b'), S:get('a')";
     assertEquals("b", get("?query=" + request(query)));
   }
\ No newline at end of file
