diff --git a/old_product.java b/new_product.java
index f41077f..93f7fd2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,12 @@
-@ContextDependent
-  @Requires(Permission.NONE)
-  public void updateAttribute(final Str key, final Item item) throws QueryException {
+@Requires(Permission.NONE)
+  public void set(final Str key, final Item item) throws QueryException {
     Item it = item;
     final Data d = it.data();
     if(d != null && !d.inMemory()) {
       // convert database node to main memory data instance
       it = ((ANode) it).dbCopy(context.context.prop);
     } else if(it instanceof FItem) {
-      BXSE_FITEM.thrw(null, it);
+      BXSE_FITEM.thrw(null);
     }
     session().setAttribute(key.toJava(), it);
   }
\ No newline at end of file
