diff --git a/old_test.java b/new_test.java
index 43a1382..1d68d10 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
   public void testGetFlow() throws IOException, ExecutorManagerException,
-    InterruptedException {
-    QueuedExecutions queue = new QueuedExecutions(2);
-    List<Pair<ExecutionReference, ExecutableFlow>> dataList = getDummyData();
+      InterruptedException {
+    final QueuedExecutions queue = new QueuedExecutions(2);
+    final List<Pair<ExecutionReference, ExecutableFlow>> dataList = getDummyData();
     queue.enqueueAll(dataList);
-    for (Pair<ExecutionReference, ExecutableFlow> pair : dataList) {
+    for (final Pair<ExecutionReference, ExecutableFlow> pair : dataList) {
       Assert.assertEquals(pair.getSecond(),
-        queue.getFlow(pair.getFirst().getExecId()));
+          queue.getFlow(pair.getFirst().getExecId()));
     }
   }
\ No newline at end of file
