diff --git a/old_product.java b/new_product.java
index ac1095c..917a4f2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public ExecutableFlow getFlow(int executionId) {
+public ExecutableFlow getFlow(final int executionId) {
     if (hasExecution(executionId)) {
-      return queuedFlowMap.get(executionId).getSecond();
+      return this.queuedFlowMap.get(executionId).getSecond();
     }
     return null;
   }
\ No newline at end of file
