diff --git a/old_product.java b/new_product.java
index ad9b3c4..14c9816 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,4 @@
-public boolean contains(Point point)
-	{
-		if(this.center.distance(point)<this.radius)
-		{
-			return true;
-		}
-		else
-		{
-			return false;
-		}
+public boolean covers(double x, double y) {
+		double distance = Math.sqrt((x-this.centerPoint.x)*(x-this.centerPoint.x)+(y-this.centerPoint.y)*(y-this.centerPoint.y));
+		return distance<=this.radius?true:false;
 	}
\ No newline at end of file
