diff --git a/old_test.java b/new_test.java
index 247f54b..51c6103 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,6 @@
 @Test
-    public void testIsProfilableClass() throws IOException {
-
-        ProfilerConfig profilerConfig = new ProfilerConfig();
-        profilerConfig.setProfilableClass("com.nhn.pinpoint.testweb.controller.*,com.nhn.pinpoint.testweb.MyClass");
-
-        Assert.assertTrue(profilerConfig.isProfilableClass("com/nhn/pinpoint/testweb/MyClass"));
-        Assert.assertTrue(profilerConfig.isProfilableClass("com/nhn/pinpoint/testweb/controller/MyController"));
-        Assert.assertTrue(profilerConfig.isProfilableClass("com/nhn/pinpoint/testweb/controller/customcontroller/MyCustomController"));
-
-        Assert.assertFalse(profilerConfig.isProfilableClass("com/nhn/pinpoint/testweb/MyUnknownClass"));
-        Assert.assertFalse(profilerConfig.isProfilableClass("com/nhn/pinpoint/testweb/controller2/MyController"));
-    }
\ No newline at end of file
+	public void defaultProfilableClassFilter() throws IOException {
+		ProfilerConfig profilerConfig = new ProfilerConfig();
+		Filter<String> profilableClassFilter = profilerConfig.getProfilableClassFilter();
+		Assert.assertFalse(profilableClassFilter.filter("net/spider/king/wang/Jjang"));
+	}
\ No newline at end of file
