diff --git a/old_product.java b/new_product.java
index 7ab687a..453f193 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,3 @@
-public boolean isProfilableClass(String className) {
-        if (profileInclude.contains(className)) {
-            return true;
-        } else {
-            String packageName = className.substring(0, className.lastIndexOf("/") + 1);
-            for (String pkg : profileIncludeSub) {
-                if (packageName.startsWith(pkg)) {
-                    return true;
-                }
-            }
-        }
-        return false;
+public Filter<String> getProfilableClassFilter() {
+		return profilableClassFilter;
     }
\ No newline at end of file
