diff --git a/old_test.java b/new_test.java
index 86dbeab..b023708 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
 
     	for(int i=0;i<loopTimes;i++)
     	{
-    		List<Envelope> result = KNNQuery.SpatialKnnQuery(rectangleRDD, queryPoint, 5, false);
+    		List<Polygon> result = KNNQuery.SpatialKnnQuery(rectangleRDD, queryPoint, topK, false);
     		assert result.size()>-1;
     		assert result.get(0).getUserData().toString()!=null;
     		//System.out.println(result.get(0).getUserData().toString());
