diff --git a/old_test.java b/new_test.java
index 750c5be..971f453 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
 		String requestedPath = "/contacts/33445/addresses/12";
 		String result = URITemplateHelper.replaceURLTemplateIDs(path,types);
 		Assert.assertTrue("/contacts/(\\d+)/addresses/(\\d+)$".equals(result));
-		Map<Integer,String> positions = URITemplateHelper.extractURLParameterValues(requestedPath, Pattern.compile(result));
+		Map<Integer,String> positions = URITemplateHelper.extractURLParameterValuesFromRequest(requestedPath, Pattern.compile(result));
 		Assert.assertFalse(positions.isEmpty());
 		Assert.assertTrue(positions.get(new Integer(1)).equals("33445"));
 		Assert.assertTrue(positions.get(new Integer(2)).equals("12"));
