diff --git a/old_product.java b/new_product.java
index d71d2e3..9b63840 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static Map<Integer,String> extractURLParameterValues(String path,Pattern pattern) {
+public static Map<Integer,String> extractURLParameterValuesFromRequest(String path,Pattern pattern) {
 		Matcher matcher = pattern.matcher(path);
 		Set<String> names = new LinkedHashSet<String>();
 		int count = matcher.groupCount();
