diff --git a/old_test.java b/new_test.java
index 7310aae..009ab17 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,9 +14,10 @@
         ArrayList<FeatureMeta> meta = new ArrayList<>();
         for (int i = 0; i < 4; i++)
             meta.add(new FeatureMeta("", i, false));
-        RandomForestClassifierTrainer trainer = new RandomForestClassifierTrainer(meta)
+        DatasetTrainer<ModelsComposition, Double> trainer = new RandomForestClassifierTrainer(meta)
             .withAmountOfTrees(5)
-            .withFeaturesCountSelectionStrgy(x -> 2);
+            .withFeaturesCountSelectionStrgy(x -> 2)
+            .withEnvironmentBuilder(TestUtils.testEnvBuilder());
 
         ModelsComposition mdl = trainer.fit(sample, parts, (k, v) -> VectorUtils.of(k), (k, v) -> v);
 
