diff --git a/old_product.java b/new_product.java
index 6fe3e70..e6fbfdf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static double PoissonCdf(int k, double lamda) {
+public static double poissonCdf(int k, double lamda) {
         if(k<0 || lamda<0) {
             throw new IllegalArgumentException("All the parameters must be positive.");
         }
@@ -7,12 +7,12 @@ public static double PoissonCdf(int k, double lamda) {
         //Slow!
         $probabilitySum=0;
         for($i=0;$i<=$k;++$i) {
-            $probabilitySum+=self::Poisson($i,$lamda);
+            $probabilitySum+=self::poisson($i,$lamda);
         }
         */
 
         //Faster solution as described at: http://www.math.ucla.edu/~tom/distributions/poisson.html?
-        double probabilitySum = 1.0 - ContinuousDistributions.GammaCdf(lamda,k+1);
+        double probabilitySum = 1.0 - ContinuousDistributions.gammaCdf(lamda,k+1);
         
         return probabilitySum;
     }
\ No newline at end of file
