diff --git a/old_test.java b/new_test.java
index cf184d2..b43853a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,6 @@
         int k = 3;
         double lamda = 5.0;
         double expResult = 0.26502591533403;
-        double result = DiscreteDistributions.PoissonCdf(k, lamda);
+        double result = DiscreteDistributions.poissonCdf(k, lamda);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
