diff --git a/old_test.java b/new_test.java
index afa4f65..81fa9f0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,7 @@
         ("addNewConfigurationsFromXml");
     Method setSecurityType = UpgradeCatalog200.class.getDeclaredMethod("setSecurityType");
     Method updateTezConfiguration = UpgradeCatalog200.class.getDeclaredMethod("updateTezConfiguration");
+    Method updateClusterEnvConfiguration = UpgradeCatalog200.class.getDeclaredMethod("updateClusterEnvConfiguration");
     Method updateConfigurationProperties = AbstractUpgradeCatalog.class.getDeclaredMethod
             ("updateConfigurationProperties", String.class, Map.class, boolean.class, boolean.class);
     Method persistHDPRepo = UpgradeCatalog200.class.getDeclaredMethod("persistHDPRepo");
@@ -17,6 +18,7 @@
         .addMockedMethod(setSecurityType)
         .addMockedMethod(updateTezConfiguration)
         .addMockedMethod(updateConfigurationProperties)
+        .addMockedMethod(updateClusterEnvConfiguration)
         .addMockedMethod(persistHDPRepo)
         .createMock();
 
@@ -40,6 +42,9 @@
     upgradeCatalog.persistHDPRepo();
     expectLastCall().once();
 
+    upgradeCatalog.updateClusterEnvConfiguration();
+    expectLastCall();
+
     replay(upgradeCatalog);
 
     upgradeCatalog.executeDMLUpdates();
