diff --git a/old_test.java b/new_test.java
index 44d0934..be82034 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,4 @@
-@Test(groups={"tck"})
-    public void test_next() {
-        assertEquals(DayOfWeek.MONDAY.next(), DayOfWeek.TUESDAY);
-        assertEquals(DayOfWeek.TUESDAY.next(), DayOfWeek.WEDNESDAY);
-        assertEquals(DayOfWeek.WEDNESDAY.next(), DayOfWeek.THURSDAY);
-        assertEquals(DayOfWeek.THURSDAY.next(), DayOfWeek.FRIDAY);
-        assertEquals(DayOfWeek.FRIDAY.next(), DayOfWeek.SATURDAY);
-        assertEquals(DayOfWeek.SATURDAY.next(), DayOfWeek.SUNDAY);
-        assertEquals(DayOfWeek.SUNDAY.next(), DayOfWeek.MONDAY);
+@Test(dataProvider="plus", groups={"tck"})
+    public void test_plus_long(int base, long amount, int expected) {
+        assertEquals(DayOfWeek.of(base).plus(amount), DayOfWeek.of(expected));
     }
\ No newline at end of file
