diff --git a/old_product.java b/new_product.java
index 304d961..3355faf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-UserListResult searchUser(@WebParam(name = "loginName") String loginName, @WebParam(name = "name") String name);
\ No newline at end of file
+UserListResponse searchUser(@WebParam(name = "loginName") String loginName, @WebParam(name = "name") String name);
\ No newline at end of file
