diff --git a/old_product.java b/new_product.java
index a654116..8fdf6c2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,5 @@
 private Iter attribute(final QueryContext ctx) throws QueryException {
-    final Data data = ctx.resource.data(checkStr(expr[0], ctx), input);
-    final IndexContext ic = new IndexContext(ctx, data, null, true);
+    final IndexContext ic = new IndexContext(ctx, data(0, ctx), null, true);
     final IndexAccess ia = new IndexAccess(
         input, expr[1], IndexType.ATTRIBUTE, ic);
 
