diff --git a/old_product.java b/new_product.java
index a0d7fca..afd2eff 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 synchronized DeleteResult delete(int blockId) {
         validateBlockId(blockId, false);
-        ByteBuf metadataBuf = getBlockBuffer(0);
+        ByteBuf metadataBuf = getMetadataBlock();
         int deletedLength = 0;
         int predecessorAddress = CacheLayout.NO_ADDRESS;
         Stack<Integer> freedBlocks = new Stack<>(); // We're traversing backwards, but need these later in ascending order.
