diff --git a/old_test.java b/new_test.java
index ffccd33..39d07f8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testDelete() throws Exception {
+    public void testDelete() {
         final byte[] toWrite = new byte[LAYOUT.bufferSize()];
         rnd.nextBytes(toWrite);
         final int multiBlockLength = LAYOUT.blockSize() + 1;
@@ -8,12 +8,12 @@
         val b = newBuffer();
         int expectedUsedBlockCount = 1;
 
-        val empty = b.write(new ByteArrayInputStream(toWrite), 0, CacheLayout.NO_ADDRESS);
+        val empty = b.write(new ByteArraySegment(toWrite, 0, 0), CacheLayout.NO_ADDRESS);
         expectedUsedBlockCount += 1;
-        val multiBlock = b.write(new ByteArrayInputStream(toWrite), multiBlockLength, CacheLayout.NO_ADDRESS);
+        val multiBlock = b.write(new ByteArraySegment(toWrite, 0, multiBlockLength), CacheLayout.NO_ADDRESS);
         expectedUsedBlockCount += 2;
         val predecessorAddress = LAYOUT.calculateAddress(BUFFER_ID + 1, 2);
-        val multiBuf = b.write(new ByteArrayInputStream(toWrite), 2 * LAYOUT.blockSize(), predecessorAddress);
+        val multiBuf = b.write(new ByteArraySegment(toWrite, 0, 2 * LAYOUT.blockSize()), predecessorAddress);
         expectedUsedBlockCount += 2;
 
         val emptyDelete = b.delete(LAYOUT.getBlockId(empty.getLastBlockAddress()));
