diff --git a/old_product.java b/new_product.java
index 7e7073c..70a26a2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,5 +4,5 @@
    @Consumes(TASK)
    @JAXBResponseParser
    @ExceptionParser(ThrowVCloudErrorOn4xx.class)
-   ListenableFuture<Task> insertMedia(@EndpointParam URI vAppURI,
+   ListenableFuture<Task> insertMedia(@EndpointParam URI vmURI,
                                       @BinderParam(BindToXMLPayload.class) MediaInsertOrEjectParams mediaParams);
\ No newline at end of file
