diff --git a/old_test.java b/new_test.java
index 16e8f01..d3feb24 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,7 @@
 @Test
     public void testGetCenter() throws Exception {
-        Circle circle = new Circle(0.0, 0.0, 0.1);
-        assertEquals(circle.getCenter().getX(), 0.0, 0.01);
+    	
+    	
+        Circle circle = new Circle(geomFact.createPoint(new Coordinate(0.0,0.0)), 0.1);
+        assertEquals(circle.getCenterPoint().x, 0.0, 0.01);
     }
\ No newline at end of file
