diff --git a/old_product.java b/new_product.java
index a1fc25f..89d3293 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 void setLineTerminator(String terminator) {
-        if (terminator.length() < 1) {
+        if (terminator == null || terminator.length() < 1) {
             this.lineTerminator = DEFAULT_TERMINATOR;
         } else {
             this.lineTerminator = terminator;
