diff --git a/old_test.java b/new_test.java
index 7cbc02c..7f7567c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,18 @@
 @Test
     public void testSetLineTerminator() {
         System.out.println("setLineTerminator");
-        System.out.println("-not testing setter.");
+        GrblCommunicator instance = new GrblCommunicator();
+        String defaultTerminator = AbstractCommunicator.DEFAULT_TERMINATOR;
+        
+        // Initial value.
+        assertEquals(defaultTerminator, instance.getLineTerminator());
+        
+        instance.setLineTerminator("tada");
+        assertEquals("tada", instance.getLineTerminator());
+        
+        instance.setLineTerminator(null);
+        assertEquals(defaultTerminator, instance.getLineTerminator());
+        
+        instance.setLineTerminator("");
+        assertEquals(defaultTerminator, instance.getLineTerminator());
     }
\ No newline at end of file
