diff --git a/old_test.java b/new_test.java
index bb10ab5..6a91ec9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,3 +1,16 @@
 @Test
   public void testExecuteDMLUpdates() throws Exception {
+    Method removeNagiosService = UpgradeCatalog200.class.getDeclaredMethod("removeNagiosService");
+
+    UpgradeCatalog200 upgradeCatalog = createMockBuilder(
+        UpgradeCatalog200.class).addMockedMethod(removeNagiosService).createMock();
+
+    upgradeCatalog.removeNagiosService();
+    expectLastCall().once();
+
+    replay(upgradeCatalog);
+
+    upgradeCatalog.executeDMLUpdates();
+
+    verify(upgradeCatalog);
   }
\ No newline at end of file
