diff --git a/old_product.java b/new_product.java
index b02a5e6..0a92407 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,8 @@
 @Override
-    final public double getDerivative(double net) {
-        return (1d - output * output);
+    final public double getDerivative(double input) {
+        //output here is a*tanh^2(s*x)
+        double E_x = Math.exp(2 * this.slope * input);
+        double tanhsx = (E_x - 1d) / (E_x + 1d);
+        derivativeOutput = amplitude * slope * (1 - tanhsx * tanhsx);
+        return derivativeOutput;
     }
\ No newline at end of file
