diff --git a/old_test.java b/new_test.java
index 4c80e47..b0ee121 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,9 @@
 @Test
-    public void testGetDerivative() {
-        
-        instance.setSlope(5);
-        double out = instance.getOutput(input);
-        double result = instance.getDerivative(input);
-        assertEquals(expected_derivative, result, 0.00001);
-        
-        
-    }
\ No newline at end of file
+        public void testGetDerivative() {
+            // this must be called before getDerivative is called;
+            double out = instance.getOutput(input);
+            double result = instance.getDerivative(input);
+            assertEquals(expected_derivative, result, 0.00001);
+
+
+        }
\ No newline at end of file
