diff --git a/old_product.java b/new_product.java
index 19f21a2..89a7b2b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -35,7 +35,8 @@ public void adjOk(Channel ch) {
                     //setting destination ip
                     ddPacket.setDestinationIp(neighborIpAddr());
                     setLastSentDdPacket(ddPacket);
-                    ch.write(ddPacket);
+                    byte[] messageToWrite = getMessage(ddPacket);
+                    ch.write(messageToWrite);
                 }
             } else if (state.getValue() >= OspfNeighborState.EXSTART.getValue()) {
                 if (!formAdjacencyOrNot()) {
