diff --git a/old_test.java b/new_test.java
index 0fd731c..73ef465 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,7 +10,7 @@
 
         MpPromoteAlgo algo = new MpPromoteAlgo(masters, mailbox, "Test");
         long requestId = algo.getRequestId();
-        Future<Pair<Boolean, Long>> result = algo.start();
+        Future<Long> result = algo.start();
         verify(mailbox, times(1)).send(any(long[].class), any(Iv2RepairLogRequestMessage.class));
 
         // has a frag for txn 1000. MP handle is 1000L
@@ -40,8 +40,8 @@
         needsRepair.add(2L);
         needsRepair.add(3L);
         verify(mailbox, times(1)).repairReplicasWith(eq(needsRepair), any(Iv2RepairLogResponseMessage.class));
-        Pair<Boolean, Long> real_result = result.get();
-        assertEquals(txnEgo(1000L), (long)real_result.getSecond());
+        Long real_result = result.get();
+        assertEquals(txnEgo(1000L), (long)real_result);
 
         // check if the hashinator was updated to the newer version
         assertEquals(torv3.getFirst(), TheHashinator.getCurrentVersionedConfig().getFirst());
