diff --git a/old_test.java b/new_test.java
index f22fbca..b26e17d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void openFileTest() throws Exception {
-    TachyonURI file = new TachyonURI("/file");
+    AlluxioURI file = new AlluxioURI("/file");
     URIStatus status = new URIStatus(new FileInfo());
     Mockito.when(mFileSystemMasterClient.getStatus(file)).thenReturn(status);
     OpenFileOptions openOptions = OpenFileOptions.defaults();
