diff --git a/old_product.java b/new_product.java
index 04f2de3..d6f52b3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
-FileInStream openFile(TachyonURI path)
-      throws FileDoesNotExistException, IOException, TachyonException;
\ No newline at end of file
+FileInStream openFile(AlluxioURI path)
+      throws FileDoesNotExistException, IOException, AlluxioException;
\ No newline at end of file
