diff --git a/old_test.java b/new_test.java
index 1e4e365..2f8a9bb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testSpatialRangeQuery() throws Exception {
-    	LineStringRDD spatialRDD = new LineStringRDD(sc, InputLocation, splitter, true);
+    	LineStringRDD spatialRDD = new LineStringRDD(sc, InputLocation, splitter, true,StorageLevel.MEMORY_ONLY());
     	for(int i=0;i<loopTimes;i++)
     	{
     		long resultSize = RangeQuery.SpatialRangeQuery(spatialRDD, queryEnvelope, false,false).count();
