diff --git a/old_product.java b/new_product.java
index ef584f2..4e6e157 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,11 @@
-public static JavaRDD<Polygon> SpatialRangeQuery(PolygonRDD spatialRDD, Polygon queryWindow,boolean considerBoundaryIntersection,boolean useIndex) throws Exception
+public static JavaRDD<Polygon> SpatialRangeQuery(PolygonRDD spatialRDD, Polygon originalQueryWindow,boolean considerBoundaryIntersection,boolean useIndex) throws Exception
 	{
+		Polygon queryWindow = originalQueryWindow;
+		if(spatialRDD.getCRStransformation())
+		{
+			queryWindow = CRSTransformation.Transform(spatialRDD.getSourceEpsgCode(),spatialRDD.getTargetEpgsgCode(), originalQueryWindow);
+		}
+		
 		if(useIndex==true)
 		{
 	        if(spatialRDD.indexedRawRDD == null) {
