diff --git a/old_test.java b/new_test.java
index 0518b16..710cb3b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testLe() {
-        assertTrue(instance.le("id", 5).getQueryCriterions()
-                .contains(new LeCriterion("id", 5)));
+    	assertEquals(new LeCriterion("id", 5), instance.le("id", 5).getQueryCriterion());
     }
\ No newline at end of file
