diff --git a/old_test.java b/new_test.java
index c25402a..acdb8d0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,8 +5,6 @@
 
         @Cleanup
         TestContext context = new TestContext();
-        setupOperationLog(context);
-
         HashSet<String> storageSegments = new HashSet<>();
         setupStorageCreateHandler(context, storageSegments);
         setupStorageGetHandler(context, storageSegments, segmentName -> new StreamSegmentInformation(segmentName, 0, false, false, new ImmutableDate()));
@@ -16,12 +14,12 @@
             String segmentName = getName(i);
             val segmentAttributes = createAttributes(ATTRIBUTE_COUNT);
             context.mapper.createNewStreamSegment(segmentName, segmentAttributes, TIMEOUT).join();
-            assertStreamSegmentCreated(segmentName, segmentAttributes, context);
+            assertSegmentCreated(segmentName, segmentAttributes, context);
 
             for (int j = 0; j < transactionsPerSegment; j++) {
                 val transactionAttributes = createAttributes(ATTRIBUTE_COUNT);
                 String transactionName = context.mapper.createNewTransactionStreamSegment(segmentName, UUID.randomUUID(), transactionAttributes, TIMEOUT).join();
-                assertTransactionCreated(transactionName, segmentName, transactionAttributes, context);
+                assertSegmentCreated(transactionName, transactionAttributes, context);
             }
         }
     }
\ No newline at end of file
