diff --git a/old_test.java b/new_test.java
index 9a8363c..d1c6a21 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
-	@Verifies(value = "should return all cohorts and voided", method = "getAllCohorts(null)")
 	public void getAllCohorts_shouldReturnAllCohortsAndVoided() throws Exception {
 		executeDataSet(COHORT_XML);
 		
@@ -7,13 +6,13 @@
 		List<Cohort> allCohorts = service.getAllCohorts(true);
 		assertNotNull(allCohorts);
 		assertEquals(2, allCohorts.size());
-		assertTrue(allCohorts.get(0).isVoided());
-		assertFalse(allCohorts.get(1).isVoided());
+		assertTrue(allCohorts.get(0).getVoided());
+		assertFalse(allCohorts.get(1).getVoided());
 		
 		// if called with false parameter, should not return the voided one
 		allCohorts = service.getAllCohorts(false);
 		assertNotNull(allCohorts);
 		// only the non-voided cohort should be returned
 		assertEquals(1, allCohorts.size());
-		assertFalse(allCohorts.get(0).isVoided());
+		assertFalse(allCohorts.get(0).getVoided());
 	}
\ No newline at end of file
