diff --git a/old_product.java b/new_product.java
index b94f258..8ca5c29 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public void finishSpan(long duration) {
-        Span span = state().getCurrentClientSpan();
+        Span span = spanAndEndpoint().span();
         if (span == null) return;
 
         synchronized (span) {
@@ -7,6 +7,5 @@ public void finishSpan(long duration) {
             spanCollector().collect(span);
         }
 
-        state().setCurrentClientSpan(null);
-        state().setCurrentClientServiceName(null);
+        spanAndEndpoint().state().setCurrentLocalSpan(null);
     }
\ No newline at end of file
