diff --git a/old_product.java b/new_product.java
index 1227d02..a0611d3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 void error(ZipObserver<R, ?> w, Exception e) {
             if (running.compareAndSet(true, false)) {
                 // this thread succeeded in setting running=false so let's propagate the error
-                Observer.onError(e);
+                observer.onError(e);
                 /* since we receive an error we want to tell everyone to stop */
                 stop();
             }
