diff --git a/old_test.java b/new_test.java
index dd98128..2291ba4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
             final Observable<String> o2 = new TestErrorObservable("one", "two", "three"); // we expect to lose all of these since o1 is done first and fails
 
             @SuppressWarnings("unchecked")
-            Observable<String> m = merge(o1, o2);
+            Observable<String> m = Observable.create(merge(o1, o2));
             m.subscribe(stringObserver);
 
             verify(stringObserver, times(1)).onError(any(NullPointerException.class));
