diff --git a/old_product.java b/new_product.java
index 7986562..dad2a70 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,9 +2,19 @@ public void changeKeyFrame() {
 		tracker.dropAllTracks();
 		tracker.spawnTracks();
 
-		spawned = tracker.getNewTracks(null);
-		for( PointTrack t : spawned )
-			handleSpawnedTrack(t);
+		List<PointTrack> spawned = tracker.getNewTracks(null);
+
+		for( PointTrack l : spawned ) {
+			AssociatedPairTrack p = l.getCookie();
+			if( p == null ) {
+				l.cookie = p = new AssociatedPairTrack();
+				// little bit of trickery here.  Save the reference so that the point
+				// in the current frame is updated for free as PointTrack is
+				p.p2 = l;
+			}
+			p.p1.set(l);
+			p.lastUsed = totalFramesProcessed;
+		}
 
 		totalSpawned = spawned.size();
 		worldToKey.set(worldToCurr);
