diff --git a/old_product.java b/new_product.java
index b1232b3..4d94c38 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,19 +1,11 @@
-String execute(Collector collector, String counterName, String sessionId) throws IOException {
+String execute(Collector collector, String counterName, String sessionId, String threadId)
+			throws IOException {
 		String messageForReport;
 		switch (this) {
 		case CLEAR_COUNTER:
 			assert collector != null;
 			assert counterName != null;
-			if ("all".equalsIgnoreCase(counterName)) {
-				clearCounters(collector);
-				messageForReport = I18N.getFormattedString("Toutes_statistiques_reinitialisees",
-						counterName);
-			} else {
-				// l'action Réinitialiser a été appelée pour un compteur
-				collector.clearCounter(counterName);
-				messageForReport = I18N.getFormattedString("Statistiques_reinitialisees",
-						counterName);
-			}
+			messageForReport = clearCounter(collector, counterName);
 			break;
 		case GC:
 			if (GC_ENABLED) {
@@ -56,6 +48,10 @@ String execute(Collector collector, String counterName, String sessionId) throws
 			clearCaches();
 			messageForReport = I18N.getString("caches_purges");
 			break;
+		case KILL_THREAD:
+			assert threadId != null;
+			messageForReport = killThread(threadId);
+			break;
 		default:
 			throw new IllegalStateException(toString());
 		}
