diff --git a/old_test.java b/new_test.java
index d282b5b..8797e93 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,7 +11,7 @@
 
         // Test to get non-existent scope
         when(mockControllerService.getScope("scope2")).thenReturn(CompletableFuture.supplyAsync(() -> {
-            throw new StoreException.NodeNotFoundException();
+            throw new StoreException.DataNotFoundException();
         }));
         response = client.target(resourceURI2).request().buildGet().invoke();
         assertEquals("Get non existent scope", 404, response.getStatus());
