diff --git a/old_product.java b/new_product.java
index 9c11bac..b78c77f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,7 @@
                         return Response.status(Status.OK).entity(new ScopeProperty().scopeName(scope)).build();
                 })
                 .exceptionally( exception -> {
-                    if (exception.getCause() instanceof StoreException.NodeNotFoundException) {
+                    if (exception.getCause() instanceof StoreException.DataNotFoundException) {
                         log.warn("Scope: {} not found", scopeName);
                         return Response.status(Status.NOT_FOUND).build();
                     } else {
