diff --git a/old_test.java b/new_test.java
index 178dadc..66520d8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test(dataProvider="parseText")
-    public void test_parse_strict_caseSensitive_parseUpper(DateTimeFieldRule rule, TextStyle style, int dow, String input) throws Exception {
+    public void test_parse_strict_caseSensitive_parseUpper(DateTimeFieldRule<?> rule, TextStyle style, int dow, String input) throws Exception {
         DateTimeParseContext context = new DateTimeParseContext(symbols);
         context.setCaseSensitive(true);
         TextPrinterParser pp = new TextPrinterParser(rule, style);
         int newPos = pp.parse(context, input.toUpperCase(), 0);
         assertEquals(newPos, ~0);
-        assertEquals(context.toCalendrical().getFieldMap().contains(RULE_DOW), false);
+        assertEquals(context.toCalendricalMerger().getInputMap().containsKey(RULE_DOW), false);
     }
\ No newline at end of file
