diff --git a/old_product.java b/new_product.java
index 1fddae6..e7ef20d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,3 @@
 public static LocalDate parse(String text) {
-        ISOChronology.checkNotNull(text, "Text to parse must not be null");
-        return DateTimeFormatters.isoLocalDate().parse(text).mergeStrict().toLocalDate();
+        return DateTimeFormatters.isoLocalDate().parse(text, rule());
     }
\ No newline at end of file
