diff --git a/old_test.java b/new_test.java
index 15f1a13..3781cc6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
     public void testIsBlank() {
-        QueryCriterion criterion1 = new IsNullCriterion("name");
-        QueryCriterion criterion2 = new EqCriterion("name", "");
-        QueryCriterion criterion3 = new OrCriterion(criterion1, criterion2);
+        QueryCriterion criterion1 = Criteria.isNull("name");
+        QueryCriterion criterion2 = Criteria.eq("name", "");
+        QueryCriterion criterion3 = Criteria.or(criterion1, criterion2);
         
         assertEquals(criterion3, instance.isBlank("name").getQueryCriterion());
     }
\ No newline at end of file
