diff --git a/old_product.java b/new_product.java
index d6712c2..417a161 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,10 @@
 public long getNewBlockIdForFile(AlluxioURI path)
-      throws FileDoesNotExistException, InvalidPathException {
+      throws FileDoesNotExistException, InvalidPathException, AccessControlException {
     MasterContext.getMasterSource().incGetNewBlockOps(1);
     Inode<?> inode;
     synchronized (mInodeTree) {
-      inode = mInodeTree.getInodeByPath(path);
-    }
-    if (!inode.isFile()) {
-      throw new FileDoesNotExistException(ExceptionMessage.PATH_MUST_BE_FILE.getMessage(path));
+      mPermissionChecker.checkParentPermission(FileSystemAction.WRITE, path);
+      inode = mInodeTree.getInodeFileByPath(path);
     }
     MasterContext.getMasterSource().incNewBlocksGot(1);
     return ((InodeFile) inode).getNewBlockId();
