diff --git a/old_product.java b/new_product.java
index 42cb05c..6b0a557 100644
--- a/old_product.java
+++ b/new_product.java
@@ -39,7 +39,7 @@ private void copy() throws InterruptedException, IOException, StoreException {
       long segmentCountDiff =
           compactionLog.getCompactionDetails().getLogSegmentsUnderCompaction().size() - numSwapsUsed;
       long savedBytes = srcLog.getSegmentCapacity() * segmentCountDiff;
-      srcMetrics.aggregatedStoreMetrics.compactionBytesReclaimedCount.inc(savedBytes);
+      srcMetrics.compactionBytesReclaimedCount.inc(savedBytes);
     }
     tgtIndex.close();
     tgtLog.close();
