diff --git a/old_test.java b/new_test.java
index cfc2749..19db007 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,8 @@
 @Test
 	public void testQueryTodoList() {
-
+		
+		long i = getJBPMApplication().startProcess("defaultPackage.Trade", "aaa", null);
 		List<TaskVO> tasks = getJBPMApplication().queryTodoList("fhjl");
 		Assert.assertTrue(tasks.size() > 0);
-
+		//getJBPMApplication().removeProcessInstance(i);
 	}
\ No newline at end of file
