diff --git a/old_product.java b/new_product.java
index a035423..06ee155 100644
--- a/old_product.java
+++ b/new_product.java
@@ -18,7 +18,7 @@
 						LOG.debug("Triggering stack trace sample for tasks: " + Arrays.toString(vertex.getTaskVertices()));
 					}
 
-					Future<StackTraceSample> sample = coordinator.triggerStackTraceSample(
+					CompletableFuture<StackTraceSample> sample = coordinator.triggerStackTraceSample(
 							vertex.getTaskVertices(),
 							numSamples,
 							delayBetweenSamples,
