diff --git a/old_test.java b/new_test.java
index 225fde9..960fdea 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	public void testAddView() throws IOException, URISyntaxException, AnnotatorException {
-		SimpleGazetteerAnnotator sga = new SimpleGazetteerAnnotator(6, "/testgazetteers/", false);
+		SimpleGazetteerAnnotator sga = new SimpleGazetteerAnnotator(defaultRm);
 		assertTrue ("Wrong number of dictionaries loaded.",sga.dictionaries.size() == 1);
 		assertTrue ("Wrong number of dictionaries loaded.",sga.dictionariesIgnoreCase.size() == 1);
 		TextAnnotation ta = tab.createTextAnnotation("I hail from the university of illinois at champaign urbana.");
@@ -23,8 +23,12 @@
 		assertEquals(c3.getLabel(),"places(IC)");
 		assertEquals(c4.getLabel(),"places(IC)");
 		assertEquals(c5.getLabel(),"places(IC)");
-		
-		sga = new SimpleGazetteerAnnotator(4, "/testgazetteers", false);
+
+        Properties props = new Properties();
+        props.setProperty( SimpleGazetteerAnnotatorConfigurator.PHRASE_LENGTH.key, "4" );
+        props.setProperty(SimpleGazetteerAnnotatorConfigurator.PATH_TO_DICTIONARIES.key, "/testgazetteers/");
+        props.setProperty(SimpleGazetteerAnnotatorConfigurator.IS_LAZILY_INITIALIZED.key,  SimpleGazetteerAnnotatorConfigurator.FALSE );
+		sga = new SimpleGazetteerAnnotator(new ResourceManager( props ));
 		assertTrue ("Wrong number of dictionaries loaded.",sga.dictionaries.size() == 1);
 		assertTrue ("Wrong number of dictionaries loaded.",sga.dictionariesIgnoreCase.size() == 1);
 		ta = tab.createTextAnnotation("I hail from the university of illinois at champaign urbana.");
