diff --git a/old_test.java b/new_test.java
index be64478..704521d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,7 @@
 @Test
-	public void visitObject() throws Exception
+	public void visitStrings() throws Exception
 	{
-		ClassVisitFilter<String> filter = new ClassVisitFilter<String>(String.class);
+		ClassVisitFilter filter = new ClassVisitFilter(String.class);
 		assertTrue(filter.visitObject("a string"));
 		assertFalse(filter.visitObject(123));
-
-		filter = new ClassVisitFilter<String>(null);
-		assertTrue(filter.visitObject("a string"));
-		assertTrue(filter.visitObject(123));
 	}
\ No newline at end of file
