diff --git a/old_test.java b/new_test.java
index 20ae457..bf83125 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,5 +13,5 @@
     query(_FILE_CREATE_DIR.args(PATH3));
     query(_FILE_WRITE.args(PATH4, "()"));
     contains(_FILE_LIST.args(PATH1, "true()"), "y");
-    query(_FILE_LIST.args(PATH1, "true()", "x"), "x" + File.separator);
+    query(_FILE_LIST.args(PATH1, "true()", "x"), 'x' + File.separator);
   }
\ No newline at end of file
