diff --git a/old_product.java b/new_product.java
index c126acf..b47346a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,9 +11,8 @@ private Iter list(final QueryContext ctx) throws QueryException {
     if(!dir.isDirectory()) FILE_NODIR.thrw(info, dir);
 
     final boolean rec = optionalBool(1, ctx);
-    final Pattern pat = expr.length != 3 ? null :
-      Pattern.compile(IOFile.regex(string(checkStr(expr[2], ctx))),
-        Prop.CASE ? 0 : Pattern.CASE_INSENSITIVE);
+    final Pattern pat = expr.length == 3 ? Pattern.compile(IOFile.regex(
+        string(checkStr(expr[2], ctx))), Prop.CASE ? 0 : Pattern.CASE_INSENSITIVE) : null;
 
     final StringList list = new StringList();
     final String p = dir.getPath();
