diff --git a/old_test.java b/new_test.java
index e7f20cc..02ffc29 100644
--- a/old_test.java
+++ b/new_test.java
@@ -26,7 +26,7 @@
         response().withHeaders(
             new Header(HttpHeaders.Names.CONTENT_TYPE, HttpMediaType.APPLICATION_JSON)).withBody(
             GsonSingleton.getGsonWithoutPrettyPrinting().toJson(speechResults)));
-    final SpeechResults result = service.recognize(audio);
+    final SpeechResults result = service.recognize(audio).execute();
     Assert.assertNotNull(result);
     Assert.assertEquals(result, speechResults);
   }
\ No newline at end of file
