diff --git a/old_product.java b/new_product.java
index e062765..6489614 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public SpeechResults recognize(File audio, String contentType, RecognizeOptions options) {
+public ServiceCall<SpeechResults> recognize(File audio, String contentType, RecognizeOptions options) {
     RecognizeOptions opt = options;
     if (opt == null)
       opt = new RecognizeOptions().contentType(contentType);
