diff --git a/old_test.java b/new_test.java
index 0db8f2d..25b34c6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,5 +2,6 @@
     public void handleFailure()
             throws Exception {
         // Since the servers are started as single nodes thus already bootstrapped.
-        assertThat(client.handleFailure(0L, Collections.singleton("key")).get()).isEqualTo(true);
+        assertThat(client.handleFailure(0L, Collections.emptySet()).get())
+                .isEqualTo(true);
     }
\ No newline at end of file
