diff --git a/old_product.java b/new_product.java
index 352ea08..477b311 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public boolean handleFailure(IReconfigurationHandlerPolicy failureHandlerPolicy,
-                                 Layout currentLayout,
-                                 CorfuRuntime corfuRuntime,
-                                 Set<String> failedServers) {
+public boolean handleFailure(@Nonnull IReconfigurationHandlerPolicy failureHandlerPolicy,
+                                 @Nonnull Layout currentLayout,
+                                 @Nonnull CorfuRuntime corfuRuntime,
+                                 @Nonnull Set<String> failedServers) {
         try {
             corfuRuntime.getLayoutManagementView().handleFailure(failureHandlerPolicy,
                     currentLayout, failedServers);
