diff --git a/old_test.java b/new_test.java
index 1251cd2..119daee 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,8 @@
-@SuppressWarnings({ "unchecked", "rawtypes" })
-    @Test
+@Test
     public void getData() {
         for (Material material : Material.values()) {
-            Class clazz = material.getData();
+            Class<? extends MaterialData> clazz = material.getData();
 
-            assertThat(material.getNewData((byte) 0), isA(clazz));
+            assertThat(material.getNewData((byte) 0), is(instanceOf(clazz)));
         }
     }
\ No newline at end of file
