diff --git a/old_test.java b/new_test.java
index 877c80e..68bed1c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,8 +6,8 @@
         BigInteger[] bigBigIntegerArrayTest=new BigInteger[420000];
         Arrays.fill(bigBigIntegerArrayTest, new BigInteger("42"));
         assertEquals(MathUtils.average(bigBigIntegerArrayTest), new BigDecimal("42"));
-        ArrayList<BigDecimal> bigBigDecimalCollectionTest=new ArrayList<BigDecimal>();
-        BigDecimal[] bigBigDecimalArrayTest=new BigDecimal[210000];
+        ArrayList<Number> bigBigDecimalCollectionTest=new ArrayList<Number>();
+        Number[] bigBigDecimalArrayTest=new Number[210000];
         Arrays.fill(bigBigDecimalArrayTest, new BigDecimal("2112"));
         bigBigDecimalCollectionTest.addAll(Arrays.asList(bigBigDecimalArrayTest));
         assertEquals(MathUtils.average(bigBigDecimalCollectionTest), new BigDecimal("2112"));
