diff --git a/old_product.java b/new_product.java
index 72e5627..819666f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static BigDecimal average(Number[] numbers) {
-        if(numbers.length==0){
+        if(numbers==null||numbers.length==0){
             return null;
         }
         BigDecimal sum = new BigDecimal(0);
@@ -12,5 +12,11 @@ public static BigDecimal average(Number[] numbers) {
             }
         }
 
-        return sum.divide(new BigDecimal(numbersCount),RoundingMode.HALF_EVEN);
+        BigDecimal result;
+        try{
+            result=sum.divide(new BigDecimal(numbersCount));
+        }catch(ArithmeticException ex){
+            result=sum.divide(new BigDecimal(numbersCount),10,RoundingMode.HALF_EVEN);//Maximum of 10 decimal digits to avoid periodic number exception.
+        }
+        return result;
     }
\ No newline at end of file
