diff --git a/old_test.java b/new_test.java
index 068df16..caa9a95 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,10 +2,10 @@
 	public void testCreateSSLServerContext() throws Exception {
 
 		Configuration serverConfig = new Configuration();
-		serverConfig.setBoolean(ConfigConstants.SECURITY_SSL_ENABLED, true);
-		serverConfig.setString(ConfigConstants.SECURITY_SSL_KEYSTORE, "src/test/resources/local127.keystore");
-		serverConfig.setString(ConfigConstants.SECURITY_SSL_KEYSTORE_PASSWORD, "password");
-		serverConfig.setString(ConfigConstants.SECURITY_SSL_KEY_PASSWORD, "password");
+		serverConfig.setBoolean(SecurityOptions.SSL_ENABLED, true);
+		serverConfig.setString(SecurityOptions.SSL_KEYSTORE, "src/test/resources/local127.keystore");
+		serverConfig.setString(SecurityOptions.SSL_KEYSTORE_PASSWORD, "password");
+		serverConfig.setString(SecurityOptions.SSL_KEY_PASSWORD, "password");
 
 		SSLContext serverContext = SSLUtils.createSSLServerContext(serverConfig);
 		Assert.assertNotNull(serverContext);
