diff --git a/old_product.java b/new_product.java
index 581a10d..84202cf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,5 @@ public DateTimeFormatter withLocale(Locale locale) {
         if (this.locale.equals(locale)) {
             return this;
         }
-        return new DateTimeFormatter(printerParser, locale, symbols, chrono, zone);
+        return new DateTimeFormatter(printerParser, locale, decimalStyle, chrono, zone);
     }
\ No newline at end of file
