diff --git a/old_test.java b/new_test.java
index 17d003a..315ade0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
       fr = parse_test_file("./smalldata/airlines/allyears2k_headers.zip");
       double[] mins =new double[fr.numCols()];
       for (int i=0; i < mins.length; i++)
-        mins[i] = fr.vecs()[i].min();
+        mins[i] = fr.vecs().min(i);
       // Scribble into a freshly parsed frame
       new SetDoubleInt(mins).doAll(fr);
     } finally {
