diff --git a/old_product.java b/new_product.java
index ad96303..dfb3cc9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public final float set(int idx, float f) {
     setWrite();
     if( _chk2.set_impl(idx,f) ) return f;
-    (_chk2 = inflate_impl(new NewChunk(this))).set_impl(idx,f);
+    (_chk2 = inflate()).set_impl(idx,f);
     return f;
   }
\ No newline at end of file
