diff --git a/old_product.java b/new_product.java
index 92f3c22..97b3893 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@
       throws IOException, FileDoesNotExistException, TachyonException {
     FileSystemMasterClient masterClient = mContext.acquireMasterClient();
     try {
-      return masterClient.getFileInfoList(file.getFileId());
+      return masterClient.listStatus(file.getFileId());
     } finally {
       mContext.releaseMasterClient(masterClient);
     }
