diff --git a/old_test.java b/new_test.java
index fef0825..70cc12a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
   public void listStatusTest() throws Exception {
     List<FileInfo> infos = Lists.newArrayList(new FileInfo());
-    Mockito.when(mFileSystemMasterClient.getFileInfoList(FILE_ID)).thenReturn(infos);
+    Mockito.when(mFileSystemMasterClient.listStatus(FILE_ID)).thenReturn(infos);
     ListStatusOptions listStatusOptions = ListStatusOptions.defaults();
     Assert.assertSame(infos, mFileSystem.listStatus(new TachyonFile(FILE_ID), listStatusOptions));
-    Mockito.verify(mFileSystemMasterClient).getFileInfoList(FILE_ID);
+    Mockito.verify(mFileSystemMasterClient).listStatus(FILE_ID);
   }
\ No newline at end of file
