diff --git a/old_product.java b/new_product.java
index 091f83b..830d903 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,26 +1,23 @@
-static void bottomTwoColumns(NodeInfo corner, List<NodeInfo> column0, List<NodeInfo> column1) {
-		column0.add(corner);
-		column0.add(corner.right);
+static void bottomTwoColumns(NodeInfo first, NodeInfo second, List<NodeInfo> column0, List<NodeInfo> column1) {
+		column0.add(first);
+		column0.add(second);
 
-		NodeInfo a = selectClosest(corner.right,corner).target;
+		NodeInfo a = selectClosestN(first,second);
 		if( a == null ) {
 			return;
 		}
 		a.marked = true;
 		column1.add(a);
-		NodeInfo b = corner.right;
-
-		corner.marked = true;
-		corner.right.marked = true;
+		NodeInfo b = second;
 
 		while( true ) {
-			NodeInfo t = selectClosest(b,a).target;
+			NodeInfo t = selectClosestN(b,a);
 			if( t == null ) break;
 			t.marked = true;
 			column1.add(t);
 
 			a = t;
-			t = selectClosest(b,a).target;
+			t = selectClosestN(b,a);
 			if( t == null ) break;
 			t.marked = true;
 			column0.add(t);
