diff --git a/old_product.java b/new_product.java
index e7dfc25..d868db3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,9 @@
 @Override
-    public BaseMLregressor.ValidationMetrics kFoldCrossValidation(Dataframe trainingData, TrainingParameters trainingParameters, int k) {
-        throw new UnsupportedOperationException("K-fold Cross Validation is not supported. Run it directly to the wrapped regressor."); 
+    public AbstractRegressor.ValidationMetrics kFoldCrossValidation(Dataframe trainingData, TrainingParameters trainingParameters, int k) {
+        if(mlregressor == null) {
+            throw new RuntimeException("You need to train a Regressor before running k-fold cross validation.");
+        }
+        else {
+            return (ValidationMetrics) mlregressor.kFoldCrossValidation(trainingData, trainingParameters, k);
+        }
     }
\ No newline at end of file
