diff --git a/old_test.java b/new_test.java
index 3cca8f2..599935c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,6 +10,6 @@
         
         AssociativeArray result = FixedCombinationRules.median(classifierClassProbabilityMatrix);
         for(Object k: expResult.keySet()) {
-            assertEquals(TypeConversions.toDouble(expResult.get(k)), TypeConversions.toDouble(result.get(k)), TestConfiguration.DOUBLE_ACCURACY_HIGH);
+            assertEquals(TypeInference.toDouble(expResult.get(k)), TypeInference.toDouble(result.get(k)), TestConfiguration.DOUBLE_ACCURACY_HIGH);
         }
     }
\ No newline at end of file
