diff --git a/old_product.java b/new_product.java
index 8ab324f..4c611b6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -34,8 +34,8 @@ public static double median(AssociativeArray2D survivalFunction) throws IllegalA
             return (ApointTi!=null)?ApointTi:BpointTi;
         }
         
-        double ApointTiValue = TypeConversions.toDouble(survivalFunction.get2d(ApointTi.toString(), "Sti"));
-        double BpointTiValue = TypeConversions.toDouble(survivalFunction.get2d(BpointTi.toString(), "Sti"));
+        double ApointTiValue = TypeInference.toDouble(survivalFunction.get2d(ApointTi.toString(), "Sti"));
+        double BpointTiValue = TypeInference.toDouble(survivalFunction.get2d(BpointTi.toString(), "Sti"));
         double median=BpointTi-(BpointTiValue-0.5)*(BpointTi-ApointTi)/(BpointTiValue-ApointTiValue);
 
         return median;
