diff --git a/old_test.java b/new_test.java
index f77044d..5206fbf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,6 @@
 	@Verifies(value = "should not fail on update method with no arguments", method = "before(Method,null,Object)")
 	public void before_shouldNotFailOnUpdateMethodWithNoArguments() throws Throwable {
 		Method method = ConceptServiceImpl.class.getMethod("updateConceptWords", (Class[]) null);
-		new RequiredDataAdvice().before(method, null, new ConceptServiceImpl());
-		new RequiredDataAdvice().before(method, new Object[] {}, new ConceptServiceImpl());
+		requiredDataAdvice.before(method, null, new ConceptServiceImpl());
+		requiredDataAdvice.before(method, new Object[] {}, new ConceptServiceImpl());
 	}
\ No newline at end of file
