diff --git a/old_product.java b/new_product.java
index ef53117..0801720 100644
--- a/old_product.java
+++ b/new_product.java
@@ -22,6 +22,10 @@
 			Reflect reflect = new Reflect(OpenmrsObject.class);
 			
 			if (reflect.isSuperClass(mainArgument)) {
+				// fail early if the method name is not like saveXyz(Xyz)
+				if (!methodNameEndsWithClassName(method, mainArgument.getClass()))
+					return;
+				
 				// if a second argument exists, pass that to the save handler as well
 				// (with current code, it means we're either in an obs save or a user save)				
 				String other = null;
@@ -34,6 +38,10 @@
 			}
 			// if the first argument is a list of openmrs objects, handle them all now
 			else if (Reflect.isCollection(mainArgument) && isOpenmrsObjectCollection(mainArgument)) {
+				// ideally we would fail early if the method name is not like savePluralOfXyz(Collection<Xyz>)
+				// but this only occurs once in the API (AdministrationService.saveGlobalProperties
+				// so it is not worth handling this case
+				
 				// if a second argument exists, pass that to the save handler as well
 				// (with current code, it means we're either in an obs save or a user save)				
 				String other = null;
@@ -50,7 +58,7 @@
 				
 			}
 		} else {
-			// fail early if the method name is not like retirePatient or retireConcept when dealing 
+			// fail early if the method name is not like retirePatient or retireConcept when dealing
 			// with Patients or Concepts as the first argument
 			if (!methodNameEndsWithClassName(method, mainArgument.getClass()))
 				return;
