diff --git a/old_test.java b/new_test.java
index 2361b95..0933733 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,8 +10,8 @@
         Assert.assertNotNull(this.job.getId());
         Assert.assertNotNull(this.job.getClusterCriteriasString());
         Assert.assertNotNull(this.job.getCommandCriteriaString());
-        Assert.assertTrue(this.job.getTags().contains(
+        Assert.assertFalse(this.job.getTags().contains(
                 CommonEntityFields.GENIE_ID_TAG_NAMESPACE + this.job.getId()));
-        Assert.assertTrue(this.job.getTags().contains(
+        Assert.assertFalse(this.job.getTags().contains(
                 CommonEntityFields.GENIE_NAME_TAG_NAMESPACE + this.job.getName()));
     }
\ No newline at end of file
