diff --git a/old_product.java b/new_product.java
index 24f49d8..d9b7d40 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,5 +8,7 @@
         if (this.tags == null) {
             this.tags = new HashSet<>();
         }
-        this.addAndValidateSystemTags(this.tags);
+        if (ConfigurationManager.getConfigInstance().getBoolean("netflix.genie.server.jobs.tags.default", false)) {
+            this.addAndValidateSystemTags(this.tags);
+        }
     }
\ No newline at end of file
