diff --git a/old_test.java b/new_test.java
index fcadcfc..3fc5f8e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,21 +1,21 @@
 @Test
     public void testDeleteProfileFilterRecord() throws Exception {
         EndpointGroupDto endpointGroup = createEndpointGroup();
-        
-        ProfileFilterDto profileFilter1 = createProfileFilter(null, endpointGroup.getId());
-        ProfileFilterDto profileFilter2 = createProfileFilter(null, endpointGroup.getId());
-        
+
+        ProfileFilterDto profileFilter1 = createProfileFilter(null, null, endpointGroup.getId());
+        ProfileFilterDto profileFilter2 = createProfileFilter(null, null, endpointGroup.getId());
+
         client.activateProfileFilter(profileFilter2.getId());
-        
-        client.deleteProfileFilterRecord(profileFilter2.getSchemaId(), endpointGroup.getId());
-        
+
+        client.deleteProfileFilterRecord(profileFilter2.getEndpointProfileSchemaId(), profileFilter2.getServerProfileSchemaId(), endpointGroup.getId());
+
         List<ProfileFilterRecordDto> profileFilterRecords = client.getProfileFilterRecords(endpointGroup.getId(), false);
-        
+
         Assert.assertNotNull(profileFilterRecords);
         Assert.assertEquals(1, profileFilterRecords.size());
-        assertProfileFiltersEquals(profileFilter1, profileFilterRecords.get(0).getInactiveProfileFilter());
-        
-        client.deleteProfileFilterRecord(profileFilter1.getSchemaId(), endpointGroup.getId());
+        assertProfileFiltersEquals(profileFilter1, profileFilterRecords.get(0).getInactiveStructureDto());
+
+        client.deleteProfileFilterRecord(profileFilter1.getEndpointProfileSchemaId(), profileFilter1.getServerProfileSchemaId(), endpointGroup.getId());
         profileFilterRecords = client.getProfileFilterRecords(endpointGroup.getId(), false);
         Assert.assertNotNull(profileFilterRecords);
         Assert.assertEquals(0, profileFilterRecords.size());
