diff --git a/old_test.java b/new_test.java
index 906956d..8dacdce 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void testPut() {
-    final Request request = RequestBuilder.put(urlWithQuery).build();
+    final Request request = RequestBuilder.put(HttpUrl.parse(urlWithQuery)).build();
     assertEquals("PUT", request.method());
     assertEquals(urlWithQuery, request.url().toString());
   }
\ No newline at end of file
