diff --git a/old_test.java b/new_test.java
index 80dc13e..3fc23c5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,11 @@
 @Test
-	@Verifies(value = "should fail if reason is null", method = "voidCohort(Cohort,String)")
 	public void voidCohort_shouldFailIfReasonIsNull() throws Exception {
 		executeDataSet(COHORT_XML);
 		
 		// Get a non-voided, valid Cohort and try to void it with a null reason
 		Cohort exampleCohort = service.getCohort("Example Cohort");
 		assertNotNull(exampleCohort);
-		assertFalse(exampleCohort.isVoided());
+		assertFalse(exampleCohort.getVoided());
 		
 		try {
 			service.voidCohort(exampleCohort, null);
@@ -17,7 +16,7 @@
 		// Now get the Cohort and try to void it with an empty reason
 		exampleCohort = service.getCohort("Example Cohort");
 		assertNotNull(exampleCohort);
-		assertFalse(exampleCohort.isVoided());
+		assertFalse(exampleCohort.getVoided());
 		
 		try {
 			service.voidCohort(exampleCohort, "");
