diff --git a/old_product.java b/new_product.java
index dc914f1..0e1fcb9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,19 +4,9 @@
             if (isClosed()) {
                 throw new IOException(Msg.getString("K005b")); //$NON-NLS-1$
             }
-            // BEGIN android-changed
-            // Exception priorities (in case of multiple errors) differ from
-            // RI, but are spec-compliant.
-            // made implicit null check explicit, used (offset | length) < 0
-            // instead of (offset < 0) || (length < 0) to safe one operation
-            if (buffer == null) {
-                throw new NullPointerException(Msg.getString("K0047")); //$NON-NLS-1$
+            if (offset < 0 || offset > buffer.length - length || length < 0) {
+                throw new IndexOutOfBoundsException();
             }
-            if ((offset | length) < 0 || offset > buffer.length - length) {
-                throw new IndexOutOfBoundsException(Msg.getString("K002f")); //$NON-NLS-1$
-            }
-            // END android-changed
-
             int outstanding = length;
             while (outstanding > 0) {
 
