diff --git a/old_test.java b/new_test.java
index 9f75978..f7e9633 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,17 +1,4 @@
-@Test(groups={"tck"})
-    public void test_extract_Class() {
-        LocalDateTime test = LocalDateTime.of(2008, 6, 30, 12, 30, 40, 987654321);
-        assertEquals(test.extract(LocalDate.class), test.getDate());
-        assertEquals(test.extract(LocalTime.class), test.getTime());
-        assertEquals(test.extract(LocalDateTime.class), null);
-        assertEquals(test.extract(OffsetDate.class), null);
-        assertEquals(test.extract(OffsetTime.class), null);
-        assertEquals(test.extract(OffsetDateTime.class), null);
-        assertEquals(test.extract(ZonedDateTime.class), null);
-        assertEquals(test.extract(ZoneOffset.class), null);
-        assertEquals(test.extract(ZoneId.class), null);
-        assertEquals(test.extract(Instant.class), null);
-        assertEquals(test.extract(String.class), null);
-        assertEquals(test.extract(BigDecimal.class), null);
-        assertNull(test.extract(null));
+@Test(expectedExceptions=NullPointerException.class, groups={"tck"})
+    public void test_query_null() {
+        TEST_2007_07_15_12_30_40_987654321.query(null);
     }
\ No newline at end of file
