diff --git a/old_product.java b/new_product.java
index 96d3688..5bae047 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-CompletableFuture<Void> concat(SegmentHandle targetSegmentHandle, long offset, SegmentHandle sourceSegmentHandle, Duration timeout);
\ No newline at end of file
+CompletableFuture<Void> concat(String targetStreamSegmentName, long offset, String sourceStreamSegmentName, Duration timeout);
\ No newline at end of file
