diff --git a/old_test.java b/new_test.java
index a0728db..deab49a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,11 @@
 @Test
     public void testConcat() throws Exception {
+        final String context = "Concat";
         try (Storage s = createStorage()) {
-            HashMap<String, ByteArrayOutputStream> appendData = populate(s);
+            HashMap<String, ByteArrayOutputStream> appendData = populate(s, context);
 
             // Check invalid handle.
-            val firstSegmentHandle = s.open(getSegmentName(0), TIMEOUT).join();
+            val firstSegmentHandle = s.open(getSegmentName(0, context), TIMEOUT).join();
             AtomicLong firstSegmentLength = new AtomicLong(s.getStreamSegmentInfo(firstSegmentHandle, TIMEOUT).join().getLength());
             assertThrows("concat() did not throw invalid target StreamSegment handle.",
                     () -> s.concat(createInvalidHandle("foo1"), 0, firstSegmentHandle, TIMEOUT),
