diff --git a/old_test.java b/new_test.java
index b8135ac..668230f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
 	public void testGetOutput() {
-		Operator fixture = new OpImpl(0);
+		final Operator fixture = new OpImpl(0);
 
-		Operator.Output result = fixture.getOutput(0);
+		final Operator.Output result = fixture.getOutput(0);
 
 		assertNotNull(result);
 		assertEquals(0, result.getIndex());
