diff --git a/old_test.java b/new_test.java
index bb7e1d4..402ba5f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,9 +3,8 @@
     Props props = new Props(baseProps);
     URL validatorUrl = Resources.getResource("project/testValidators");
     URL configUrl = Resources.getResource("test-conf/azkaban-validators-test2.xml");
-    props.put(XmlValidatorManager.VALIDATOR_PLUGIN_DIR, validatorUrl.getPath());
-    props.put(XmlValidatorManager.XML_FILE_PARAM,
-        configUrl.getPath());
+    props.put(ValidatorConfigs.VALIDATOR_PLUGIN_DIR, validatorUrl.getPath());
+    props.put(ValidatorConfigs.XML_FILE_PARAM, configUrl.getPath());
 
     XmlValidatorManager manager = new XmlValidatorManager(props);
     assertEquals("XmlValidatorManager should contain 2 validators.", manager.getValidatorsInfo().size(), 2);
