diff --git a/old_test.java b/new_test.java
index c8a4d2e..6f0190c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,7 @@
     List<? extends ReplicaId> invalidPartitionReplicas = invalidPartition.getReplicaIds();
     StorageManager storageManager = createStorageManager(replicas, metricRegistry);
     storageManager.start();
-    for (int i = 0; i < replicas.size() - 1; i++) {
+    for (int i = 1; i < replicas.size() - 1; i++) {
       ReplicaId replica = replicas.get(i);
       PartitionId id = replica.getPartitionId();
       assertTrue("Shutdown should succeed on given store", storageManager.shutdownBlobStore(id));
@@ -18,7 +18,13 @@
     PartitionId id = replica.getPartitionId();
     Store store = storageManager.getStore(id);
     store.shutdown();
-    assertFalse("Shutdown should fail on given store", storageManager.shutdownBlobStore(id));
+    assertTrue("Shutdown should succeed on the store which is not started", storageManager.shutdownBlobStore(id));
+    // test start the store whose DiskManager is not running
+    replica = replicas.get(0);
+    id = replica.getPartitionId();
+    storageManager.getDiskManager(id).shutdown();
+    assertFalse("Shutdown should fail on given store whose DiskManager is not running",
+        storageManager.shutdownBlobStore(id));
     // test invalid partition
     replica = invalidPartitionReplicas.get(0);
     id = replica.getPartitionId();
