diff --git a/old_product.java b/new_product.java
index fe6fa1a..0eea1bf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,16 @@
 boolean shutdownBlobStore(PartitionId id) {
-    BlobStore store = (BlobStore) getStore(id);
-    if (store == null) {
+    BlobStore store = stores.get(id);
+    if (store == null || !running) {
       return false;
+    } else if (!store.isStarted()) {
+      return true;
+    } else {
+      try {
+        store.shutdown();
+      } catch (Exception e) {
+        logger.error("Exception while shutting down store {} on disk {}", id, disk, e);
+        return false;
+      }
+      return true;
     }
-    try {
-      store.shutdown();
-    } catch (Exception e) {
-      logger.error("Exception while shutting down store {} on disk {}", id, disk, e);
-      return false;
-    }
-    return true;
   }
\ No newline at end of file
