diff --git a/old_product.java b/new_product.java
index b9bd6b8..2a2e6a6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,7 +7,7 @@
 
 			// generate new workers into persistent state and launch associated actors
 			for (int i = 0; i < numWorkers; i++) {
-				MesosWorkerStore.Worker worker = MesosWorkerStore.Worker.newTask(workerStore.newTaskID());
+				MesosWorkerStore.Worker worker = MesosWorkerStore.Worker.newWorker(workerStore.newTaskID());
 				workerStore.putWorker(worker);
 				workersInNew.put(extractResourceID(worker.taskID()), worker);
 
