diff --git a/old_test.java b/new_test.java
index c09e867..5d5b316 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,7 +14,7 @@
 
 						// verify that a new worker was persisted, the internal state was updated, the task router was notified,
 						// and the launch coordinator was asked to launch a task
-						MesosWorkerStore.Worker expected = MesosWorkerStore.Worker.newTask(task1);
+						MesosWorkerStore.Worker expected = MesosWorkerStore.Worker.newWorker(task1);
 						verify(workerStore).putWorker(expected);
 						assertThat(resourceManagerInstance.workersInNew, hasEntry(extractResourceID(task1), expected));
 						resourceManagerInstance.taskRouter.expectMsgClass(TaskMonitor.TaskGoalStateUpdated.class);
