diff --git a/old_test.java b/new_test.java
index 20a5ec3..66d3b74 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,7 @@
 @Test
   public void getGroups() throws Throwable {
+    InstancedConfiguration conf = ConfigurationTestUtils.defaults();
+
     String userName = "alluxio-user1";
     String userGroup1 = "alluxio-user1-group1";
     String userGroup2 = "alluxio-user1-group2";
@@ -10,12 +12,12 @@
     PowerMockito.when(cachedGroupService.getGroups(Mockito.anyString())).thenReturn(
         Lists.newArrayList(userGroup1, userGroup2));
     PowerMockito.mockStatic(GroupMappingService.Factory.class);
-    Mockito.when(GroupMappingService.Factory.get()).thenReturn(cachedGroupService);
+    Mockito.when(GroupMappingService.Factory.get(conf)).thenReturn(cachedGroupService);
 
-    List<String> groups = CommonUtils.getGroups(userName);
+    List<String> groups = CommonUtils.getGroups(userName, conf);
     assertEquals(Arrays.asList(userGroup1, userGroup2), groups);
 
-    String primaryGroup = CommonUtils.getPrimaryGroupName(userName);
+    String primaryGroup = CommonUtils.getPrimaryGroupName(userName, conf);
     assertNotNull(primaryGroup);
     assertEquals(userGroup1, primaryGroup);
   }
\ No newline at end of file
