diff --git a/old_test.java b/new_test.java
index 2a1ee8f..73605ea 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
     final SpeechRecognitionResults speechResults = new SpeechRecognitionResults();
     speechResults.setResultIndex(0);
     final SpeechRecognitionResult transcript = new SpeechRecognitionResult();
-    transcript.setFinal(true);
+    transcript.setFinalResults(true);
     final SpeechRecognitionAlternative speechAlternative = new SpeechRecognitionAlternative();
     speechAlternative.setTranscript("thunderstorms could produce large hail isolated tornadoes and heavy rain");
 
