diff --git a/old_product.java b/new_product.java
index 7a351f9..70d2ea7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 @SuppressWarnings("unchecked")
 	public List<Order> getActiveOrders(Patient patient, List<OrderType> orderTypes, CareSetting careSetting, Date asOfDate) {
 		Criteria crit = createOrderCriteria(patient, careSetting, orderTypes, false, false);
+		crit.add(Restrictions.le("dateActivated", asOfDate));
 		
 		Disjunction dateStoppedAndAutoExpDateDisjunction = Restrictions.disjunction();
 		Criterion stopAndAutoExpDateAreBothNull = Restrictions.and(Restrictions.isNull("dateStopped"), Restrictions
