diff --git a/old_product.java b/new_product.java
index b68ac51..abcf998 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 @Deprecated
-    @Transactional(readOnly = true)
-	@Authorized( { OpenmrsConstants.PRIV_VIEW_USERS })
+	@Transactional(readOnly = true)
+	@Authorized( { PrivilegeConstants.VIEW_USERS })
 	public List<User> getAllUsers(List<Role> roles, boolean includeVoided) throws APIException;
\ No newline at end of file
