diff --git a/old_test.java b/new_test.java
index d415b93..4720978 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
 	public void getAllUsers_shouldNotContainsAnyDuplicateUsers() throws Exception {
 		executeDataSet(XML_FILENAME);
 		List<User> users = Context.getUserService().getAllUsers();
-		Assert.assertEquals(10, users.size());
+		Assert.assertEquals(11, users.size());
 		// TODO Need to test with duplicate data in the dataset (not sure if that's possible)
 		
 	}
\ No newline at end of file
