diff --git a/old_test.java b/new_test.java
index 58c426f..72e5b54 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,7 +14,7 @@
 		when(graph.getState()).thenReturn(JobStatus.RUNNING);
 
 		// Same Thread execution context
-		when(graph.getExecutionContext()).thenReturn(new ExecutionContext() {
+		when(graph.getFutureExecutionContext()).thenReturn(new ExecutionContext() {
 
 			@Override
 			public void execute(Runnable runnable) {
@@ -23,7 +23,7 @@
 
 			@Override
 			public void reportFailure(Throwable t) {
-				fail();
+				// do nothing
 			}
 
 			@Override
