diff --git a/old_product.java b/new_product.java
index e4ab954..85ab898 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,12 @@
-public static void insertStringInFile(File inputFile, String contentTobeAdded) throws IOException {
+public static void updateFileHandle(File inputFile, String contentTobeAdded, boolean isClose) throws IOException {
         FileWriter fileWriter = new FileWriter(inputFile, true);
         PrintWriter outputPrintWriter = new PrintWriter(fileWriter);
-        outputPrintWriter.write(contentTobeAdded);
-        outputPrintWriter.flush();
-        outputPrintWriter.close();
-
+        if (!isClose) {
+            outputPrintWriter.write(contentTobeAdded);
+            outputPrintWriter.flush();
+            outputPrintWriter.close();
+        } else {
+            fileWriter.flush();
+            fileWriter.close();
+        }
     }
\ No newline at end of file
