diff --git a/old_test.java b/new_test.java
index 3556b41..71773b6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
         System.out.println("\n+++ toString");
 
         RunTable instance = createHorizontalInstance();
-        String expResult = "RunTable{hori HORIZONTAL 10x5}";
+        String expResult = "RunTable{HORIZONTAL 10x5}";
         String result = instance.toString();
         assertEquals(expResult, result);
     }
\ No newline at end of file
