diff --git a/old_product.java b/new_product.java
index 0895151..23e6501 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,11 +3,22 @@
     {
         StringBuilder sb = new StringBuilder();
         sb.append("{action");
-        sb.append(" class:").append(className).append(" method:").append(methodName);
 
         sb.append(" domain:").append(domain);
         sb.append(" section:").append(section);
 
+        if (className != null) {
+            sb.append(" class:").append(className);
+        }
+
+        if (methodName != null) {
+            sb.append(" method:").append(methodName);
+        }
+
+        if (menuName != null) {
+            sb.append(" menu:").append(menuName);
+        }
+
         if (itemClassName != null) {
             sb.append(" item:").append(itemClassName);
         }
