diff --git a/old_product.java b/new_product.java
index c560c67..b9e4c24 100644
--- a/old_product.java
+++ b/new_product.java
@@ -69,7 +69,7 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 			if (previousOrder == null) {
 				throw new APIException("Previous Order is required for a revised order");
 			}
-			stopOrder(previousOrder, order.getDateActivated());
+			stopOrder(previousOrder, aMomentBefore(order.getDateActivated()));
 		} else if (DISCONTINUE == order.getAction()) {
 			discontinueExistingOrdersIfNecessary(order);
 		}
