diff --git a/old_test.java b/new_test.java
index 2acf674..0c085b7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,6 +10,6 @@
 		previousOrder.setDrug(newDrug);
 		
 		expectedException.expect(APIException.class);
-		expectedException.expectMessage("Cannot change the drug of a drug order");
+		expectedException.expectMessage("Order.cannot.change.drug");
 		orderService.saveOrder(order, null);
 	}
\ No newline at end of file
