diff --git a/old_product.java b/new_product.java
index 7288a38..1cac9be 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @Override
         public String getId() {
-            if (groupID.equals(GROUP_TZDB)) {
-                return regionID;
+            if (groupId.equals(GROUP_TZDB)) {
+                return regionId;
             }
-            return groupID + ':' + regionID;
+            return groupId + ':' + regionId;
         }
\ No newline at end of file
