diff --git a/old_product.java b/new_product.java
index bcb4e2c..aecb2c8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public RealmQuery<E> isNotEmpty(String fieldName) {
         realm.checkIfValid();
 
-        FieldDescriptor fd = schema.getColumnIndices(fieldName, RealmFieldType.STRING, RealmFieldType.BINARY, RealmFieldType.LIST);
+        FieldDescriptor fd = schema.getColumnIndices(fieldName, RealmFieldType.STRING, RealmFieldType.BINARY, RealmFieldType.LIST, RealmFieldType.LINKING_OBJECTS);
         this.query.isNotEmpty(fd.getColumnIndices(), fd.getNativeTablePointers());
 
         return this;
