diff --git a/old_test.java b/new_test.java
index 816a649..fea4bf8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,5 +6,5 @@
 	    User user = controller.formBackingObject(request, null);
 	    user.addName(new PersonName("This", "is", "Test"));
 	    user.getPerson().setGender("F");
-	    controller.handleSubmission(request, new MockHttpSession(), new ModelMap(), "Save User", "pass123", "pass123", new String[0], "true", user, new BindException(user, "user"));
+	    controller.handleSubmission(request, new MockHttpSession(), new ModelMap(), "Save User", "pass123", "pass123", null, new String[0], "true", user, new BindException(user, "user"));
     }
\ No newline at end of file
