diff --git a/old_product.java b/new_product.java
index a08208b..10a19d6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,6 +5,7 @@
 	                               @RequestParam(required=false, value="action") String action,
 	                               @RequestParam(required=false, value="userFormPassword") String password,
 	                               @RequestParam(required=false, value="confirm") String confirm,
+	                               @RequestParam(required=false, value="forcePassword") Boolean forcePassword,
 	                               @RequestParam(required=false, value="roleStrings") String[] roles,
 	                               @RequestParam(required=false, value="createNewPerson") String createNewPerson,
 	                               @ModelAttribute("user") User user, BindingResult errors) {
@@ -83,23 +84,7 @@
 			else
 				user.getRoles().retainAll(newRoles);
 			
-			Map<String, String> properties = user.getUserProperties();
-			if (properties == null)
-				properties = new HashMap<String, String>();
 
-			Boolean newChangePassword = false;
-			String chk = request.getParameter(OpenmrsConstants.USER_PROPERTY_CHANGE_PASSWORD);
-			
-			if (chk != null)
-				newChangePassword = true;
-			
-			if (!newChangePassword.booleanValue() && properties.containsKey(OpenmrsConstants.USER_PROPERTY_CHANGE_PASSWORD)) {
-				properties.remove(OpenmrsConstants.USER_PROPERTY_CHANGE_PASSWORD);
-			}
-			if (newChangePassword.booleanValue()) {
-				properties.put(OpenmrsConstants.USER_PROPERTY_CHANGE_PASSWORD, newChangePassword.toString());
-			}
-			
 			String[] keys = request.getParameterValues("property");
 			String[] values = request.getParameterValues("value");
 			
@@ -107,11 +92,11 @@
 				for (int x = 0; x < keys.length; x++) {
 					String key = keys[x];
 					String val = values[x];
-					properties.put(key, val);
+					user.setUserProperty(key, val);
 				}
 			}
-						
-			user.setUserProperties(properties);
+							
+			new UserProperties(user.getUserProperties()).setSupposedToChangePassword(forcePassword);
 			
 			UserValidator uv = new UserValidator();
 			uv.validate(user, errors);
