diff --git a/old_product.java b/new_product.java
index ba371ae..e5e7882 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
-@RequiresSession
-  public AlertGroupEntity findGroupByName(long clusterId, String groupName) {
+public AlertGroupEntity findGroupByName(long clusterId, String groupName) {
     TypedQuery<AlertGroupEntity> query = entityManagerProvider.get().createNamedQuery(
         "AlertGroupEntity.findByNameInCluster", AlertGroupEntity.class);
 
