diff --git a/old_test.java b/new_test.java
index 7c663ed..02297ab 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
-  public void testFindGroupByName() {
+  public void testFindGroupByName() throws Exception {
     List<AlertGroupEntity> groups = dao.findAllGroups();
-    Assert.assertNotNull(groups);
+    assertNotNull(groups);
     AlertGroupEntity group = groups.get(3);
 
     AlertGroupEntity actual = dao.findGroupByName(group.getClusterId(),
         group.getGroupName());
 
-    Assert.assertEquals(group, actual);
+    assertEquals(group, actual);
   }
\ No newline at end of file
