diff --git a/old_product.java b/new_product.java
index 68f0c07..08f0714 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 public Builder<T> alphaInterpolation(AlphaInterpolation config)
 		{
 			checkForNull(config, "Alpha interpolation is null.");
+			updateStatus(Properties.RESIZER_FACTORY, Status.CANNOT_SET);
 			updateStatus(Properties.ALPHA_INTERPOLATION, Status.ALREADY_SET);
 			alphaInterpolation = config;
 			return this;
