diff --git a/old_test.java b/new_test.java
index 9b53836..2f0c1ac 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,14 +16,16 @@
 
       Date start = new Date();
 
-      CreateDialogNodeOptions createOptions =
-              new CreateDialogNodeOptions.Builder(workspaceId, dialogNodeName).description(dialogNodeDescription)
-                      .build();
+      CreateDialogNodeOptions createOptions = new CreateDialogNodeOptions.Builder(workspaceId, dialogNodeName)
+          .description(dialogNodeDescription)
+          .build();
       service.createDialogNode(createOptions).execute();
 
       long count = response.getDialogNodes().size();
-      ListDialogNodesOptions listOptions2 =
-              new ListDialogNodesOptions.Builder(workspaceId).pageLimit(count + 1).build();
+      ListDialogNodesOptions listOptions2 = new ListDialogNodesOptions.Builder(workspaceId)
+          .pageLimit(count + 1)
+          .includeAudit(true)
+          .build();
       DialogNodeCollection response2 = service.listDialogNodes(listOptions2).execute();
       assertNotNull(response2);
       assertNotNull(response2.getDialogNodes());
