diff --git a/old_product.java b/new_product.java
index 5db3e56..4d11032 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,5 +15,8 @@ public ServiceCall<DialogNodeCollection> listDialogNodes(ListDialogNodesOptions
     if (listDialogNodesOptions.cursor() != null) {
       builder.query("cursor", listDialogNodesOptions.cursor());
     }
+    if (listDialogNodesOptions.includeAudit() != null) {
+      builder.query("include_audit", String.valueOf(listDialogNodesOptions.includeAudit()));
+    }
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(DialogNodeCollection.class));
   }
\ No newline at end of file
