diff --git a/old_test.java b/new_test.java
index 343ebad..643ca76 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
 	public void derivX_I8() {
 		ImageInt8 img = new ImageInt8(width, height);
-		UtilImageInt8.randomize(img, rand);
+		BasicDrawing_I8.randomize(img, rand);
 
-		ImageInt16 derivX = new ImageInt16(width, height);
+		ImageInt16 derivX = new ImageInt16(width, height, true);
 
 		GecvTesting.checkSubImage(this, "derivX_I8", true, img, derivX);
 	}
\ No newline at end of file
