diff --git a/old_product.java b/new_product.java
index 1c71096..4505eb0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static Itinerary generateItinerary(GraphPath path, boolean showIntermediateStops, Locale requestedLocale) {
+public static Itinerary generateItinerary(GraphPath path, boolean showIntermediateStops, boolean disableAlertFiltering, Locale requestedLocale) {
         Itinerary itinerary = new Itinerary();
 
         State[] states = new State[path.states.size()];
@@ -19,7 +19,7 @@ public static Itinerary generateItinerary(GraphPath path, boolean showIntermedia
         }
 
         for (State[] legStates : legsStates) {
-            itinerary.addLeg(generateLeg(graph, legStates, showIntermediateStops, requestedLocale));
+            itinerary.addLeg(generateLeg(graph, legStates, showIntermediateStops, disableAlertFiltering, requestedLocale));
         }
 
         addWalkSteps(graph, itinerary.legs, legsStates, requestedLocale);
