diff --git a/old_test.java b/new_test.java
index 252cd62..139121e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,9 +10,9 @@
     query(_DB_OPEN.args(NAME) + "//title/text()", "XML");
 
     // reference invalid path
-    if(Prop.WIN) error(_DB_OPEN.args(NAME, "*"), Err.RESINV);
+    if(Prop.WIN) error(_DB_OPEN.args(NAME, "*"), Err.RESINV_X);
 
     // run function on non-existing database
     new DropDB(NAME).execute(context);
-    error(_DB_OPEN.args(NAME), Err.BXDB_OPEN);
+    error(_DB_OPEN.args(NAME), Err.BXDB_OPEN_X);
   }
\ No newline at end of file
