diff --git a/old_product.java b/new_product.java
index 0088d6f..e6a6bc5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -28,7 +28,7 @@
         }
 
         //initialize modeler
-        AbstractModeler modeler = Trainable.newInstance(aClass, "kfold_" + RandomGenerator.getThreadLocalRandomUnseeded().nextLong(), conf);
+        AbstractModeler modeler = Trainable.newInstance(aClass, "kfold_" + RandomGenerator.getThreadLocalRandomUnseeded().nextLong(), conf, trainingParameters);
 
         List<VM> validationMetricsList = new LinkedList<>();
         for(int fold=0;fold<k;++fold) {
@@ -63,7 +63,7 @@
 
 
             Dataframe trainingData = dataset.getSubset(foldTrainingIds);
-            modeler.fit(trainingData, (AbstractTrainer.AbstractTrainingParameters) trainingParameters);
+            modeler.fit(trainingData);
             trainingData.delete();
 
 
