diff --git a/old_test.java b/new_test.java
index 8f29760..029731e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,18 +12,19 @@
         
         
         String dbName = this.getClass().getSimpleName();
-        DummyXMinMaxNormalizer df = new DummyXMinMaxNormalizer(dbName, conf);
+        DummyXMinMaxNormalizer df = new DummyXMinMaxNormalizer(dbName, conf, new DummyXMinMaxNormalizer.TrainingParameters());
         
-        df.fit_transform(trainingData, new DummyXMinMaxNormalizer.TrainingParameters());
+        df.fit_transform(trainingData);
         df.transform(validationData);
-        
-        OrdinalRegression instance = new OrdinalRegression(dbName, conf);
-        
+
+
         OrdinalRegression.TrainingParameters param = new OrdinalRegression.TrainingParameters();
         param.setTotalIterations(100);
         param.setL2(0.001);
+
+        OrdinalRegression instance = new OrdinalRegression(dbName, conf, param);
         
-        instance.fit(trainingData, param);
+        instance.fit(trainingData);
         
         instance.close();
         df.close();
