diff --git a/old_test.java b/new_test.java
index aa02d68..372284c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void testSynthesize() throws IOException {
-    String text = "This is an integration test";
+    String text = "This is an integration test; 1,2 !, @, #, $, %, ^, 20.";
     InputStream result = service.synthesize(text, Voice.EN_LISA, AudioFormat.WAV).execute();
     writeInputStreamToFile(result, File.createTempFile("tts-audio", "wav"));
   }
\ No newline at end of file
