diff --git a/old_product.java b/new_product.java
index 92cc61d..1ec1798 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,9 +3,12 @@ public ServiceCall<InputStream> synthesize(final String text, final Voice voice,
     Validator.isTrue((text != null) && !text.isEmpty(), "text cannot be null or empty");
     Validator.isTrue(voice != null, "voice cannot be null or empty");
 
-    String modifiedText = text.replace(";", "%3B");
-    final RequestBuilder request = RequestBuilder.get(PATH_SYNTHESIZE);
-    request.query(TEXT, modifiedText);
+    final RequestBuilder request = RequestBuilder.post(PATH_SYNTHESIZE);
+
+    JsonObject jsonText = new JsonObject();
+    jsonText.addProperty(TEXT, text);
+    request.bodyJson(jsonText);
+    
     request.query(VOICE, voice.getName());
     request.query(ACCEPT, audioFormat != null ? audioFormat : AudioFormat.WAV);
 
