diff --git a/old_product.java b/new_product.java
index a6d0d51..234ebf0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @RequestMapping("/new")
 	@PreAuthorize("hasAnyRole('A') or #user.userId == #userId")
-	public String getUserDetail(User user, final ModelMap model) {
+	public String openForm(User user, final ModelMap model) {
 		model.addAttribute("roleSet", EnumSet.allOf(Role.class));
 		return "user/detail";
 	}
\ No newline at end of file
