diff --git a/old_test.java b/new_test.java
index 917ca78..5d4c188 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
 	public void testGetUserDetail() {
 		ModelMap model = new ModelMap();
-		userController.getUserDetail(getTestUser(), model, getTestUser().getUserId());
+		userController.getOne(getTestUser(), model, getTestUser().getUserId());
 		User user = (User) model.get("user");
 		assertThat(user.getId(), is(getTestUser().getId()));
 	}
\ No newline at end of file
