diff --git a/old_test.java b/new_test.java
index 72e9647..102590e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -43,7 +43,6 @@
 
         // thenRunAsync on default execution facility
         assertNotNull(threadName = results.poll(TIMEOUT_NS, TimeUnit.NANOSECONDS).toString());
-        assertNotSame(currentThreadName, threadName);
         assertTrue(threadName, threadName.startsWith("Default Executor-thread-"));
         assertNotNull(lookupResult = results.poll(TIMEOUT_NS, TimeUnit.NANOSECONDS));
         if (lookupResult instanceof Throwable)
@@ -52,8 +51,7 @@
 
         // thenRunAsync on unmanaged executor
         assertNotNull(threadName = results.poll(TIMEOUT_NS, TimeUnit.NANOSECONDS).toString());
-        assertNotSame(currentThreadName, threadName);
-        assertTrue(threadName, !threadName.startsWith("Default Executor-thread-"));
+        assertTrue(threadName, threadName.equals(currentThreadName) || !threadName.startsWith("Default Executor-thread-")); // could run on current thread if previous stage is complete
         assertNotNull(lookupResult = results.poll(TIMEOUT_NS, TimeUnit.NANOSECONDS));
         if (lookupResult instanceof NamingException)
             ; // pass
@@ -64,8 +62,7 @@
 
         // thenRun on unmanaged thread (context should be applied from stage creation time)
         assertNotNull(threadName = results.poll(TIMEOUT_NS, TimeUnit.NANOSECONDS).toString());
-        assertNotSame(currentThreadName, threadName);
-        assertTrue(threadName, !threadName.startsWith("Default Executor-thread-"));
+        assertTrue(threadName, threadName.equals(currentThreadName) || !threadName.startsWith("Default Executor-thread-")); // could run on current thread if previous stage is complete
         assertNotNull(lookupResult = results.poll(TIMEOUT_NS, TimeUnit.NANOSECONDS));
         if (lookupResult instanceof Throwable)
             throw new Exception((Throwable) lookupResult);
@@ -73,7 +70,6 @@
 
         // thenRunAsync (second occurrence) on default execution facility
         assertNotNull(threadName = results.poll(TIMEOUT_NS, TimeUnit.NANOSECONDS).toString());
-        assertNotSame(currentThreadName, threadName);
         assertTrue(threadName, threadName.startsWith("Default Executor-thread-"));
         assertNotNull(lookupResult = results.poll(TIMEOUT_NS, TimeUnit.NANOSECONDS));
         if (lookupResult instanceof Throwable)
@@ -83,7 +79,6 @@
         // thenRunAsync requested from unmanaged thread
         assertNotNull(threadName = results.poll(TIMEOUT_NS, TimeUnit.NANOSECONDS).toString());
         assertNotSame(currentThreadName, threadName);
-        assertTrue(threadName, threadName.startsWith("Default Executor-thread-"));
         assertNotNull(lookupResult = results.poll(TIMEOUT_NS, TimeUnit.NANOSECONDS));
         if (lookupResult instanceof NamingException)
             ; // pass
