diff --git a/old_product.java b/new_product.java
index 0668a0e..43b34f9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,15 @@
 @Override
     public ManagedCompletableFuture<Void> thenRun(Runnable action) {
-        ContextualRunnable contextualAction = new ContextualRunnable(defaultExecutor.getContextService(), action);
-        CompletableFuture<Void> dependentStage = completableFuture.thenRun(contextualAction);
+        // Reject ManagedTask so that we have the flexibility to decide later how to handle ManagedTaskListener and execution properties
+        if (action instanceof ManagedTask)
+            throw new IllegalArgumentException(ManagedTask.class.getName());
+
+        WSContextService contextSvc = defaultExecutor.getContextService();
+
+        @SuppressWarnings("unchecked")
+        ThreadContextDescriptor contextDescriptor = contextSvc.captureThreadContext(XPROPS_SUSPEND_TRAN);
+        action = contextSvc.createContextualProxy(contextDescriptor, action, Runnable.class);
+
+        CompletableFuture<Void> dependentStage = completableFuture.thenRun(action);
         return new ManagedCompletableFuture<Void>(dependentStage, defaultExecutor);
     }
\ No newline at end of file
