diff --git a/old_product.java b/new_product.java
index bb8cdbf..d17e4e8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,5 @@
     @ResponseStatus(HttpStatus.OK)
     public Set<String> getTagsForCommand(@PathVariable("id") final String id) throws GenieException {
         log.debug("Called with id {}", id);
-        return this.commandService.getTagsForCommand(id);
+        return DtoAdapters.toV3Command(this.commandService.getCommand(id)).getTags();
     }
\ No newline at end of file
