diff --git a/old_test.java b/new_test.java
index 11a66a4..b796496 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,9 @@
 @Test
   public void parse() throws Exception {
     context.options.set(MainOptions.STRIPNS, true);
-    context.options.set(MainOptions.SERIALIZER, "indent=no");
+    final SerializerOptions sopts = new SerializerOptions();
+    sopts.set(SerializerOptions.INDENT, YesNo.NO);
+    context.options.set(MainOptions.SERIALIZER, sopts);
 
     final String doc = "<e xmlns='A'><b:f xmlns:b='B'/></e>";
     for(final boolean b : new boolean[] { false, true }) {
