diff --git a/old_product.java b/new_product.java
index d39616a..43d07de 100644
--- a/old_product.java
+++ b/new_product.java
@@ -59,9 +59,9 @@ boolean parse() throws QueryException {
         }
       } else if(eq(uri, QueryText.OUTPUTURI)) {
         // serialization parameters
-        final String key = string(local);
-        final String val = toString(value, name);
-        if(!output.set(key, val)) error(info, UNKNOWN_SER, key);
+        final Option opt = output.option(string(local));
+        if(opt == null) error(info, UNKNOWN_SER, local);
+        output.set(opt, toString(value, name));
       }
       found |= rexq;
     }
