diff --git a/old_test.java b/new_test.java
index 0a9f82c..6a26115 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
 	public void deriv_I8() {
-		ImageInt8 img = new ImageInt8(width, height);
+		ImageUInt8 img = new ImageUInt8(width, height);
 		BasicDrawing_I8.randomize(img, rand);
 
-		ImageInt16 derivX = new ImageInt16(width, height, true);
-		ImageInt16 derivY = new ImageInt16(width, height, true);
+		ImageSInt16 derivX = new ImageSInt16(width, height);
+		ImageSInt16 derivY = new ImageSInt16(width, height);
 
 		GecvTesting.checkSubImage(this, "deriv_I8", true, img, derivX, derivY);
 	}
\ No newline at end of file
