diff --git a/old_product.java b/new_product.java
index 189743f..f97f295 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-public String print(TemporalAccessor temporal) {
+public String format(TemporalAccessor temporal) {
         StringBuilder buf = new StringBuilder(32);
-        printTo(temporal, buf);
+        formatTo(temporal, buf);
         return buf.toString();
     }
\ No newline at end of file
