diff --git a/old_test.java b/new_test.java
index 01d9a2e..b8f3343 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
-@Test(expectedExceptions=NullPointerException.class, groups={"tck"})
+@Test(expectedExceptions=NullPointerException.class)
     public void test_print_nullCalendrical() {
         DateTimeFormatters.isoDate().print((TemporalAccessor) null);
     }
\ No newline at end of file
