diff --git a/old_product.java b/new_product.java
index 6af4bb1..63f9cf3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
 public static byte[] hexStringToByteArray(String digits) {
+        int len = digits.length();
         // Check if string is valid hex
-        if (!VALID_HEX.matcher(digits).matches()) {
+        if (!VALID_HEX.matcher(digits).matches() || (len & 0x1) != 0) {
             LOG.error("Invalid hexadecimal string: {}", digits);
             return null;
         }
-        int len = digits.length();
         byte[] data = new byte[len / 2];
         for (int i = 0; i < len; i += 2) {
             data[i / 2] = (byte) ((Character.digit(digits.charAt(i), 16) << 4)
