diff --git a/old_test.java b/new_test.java
index a1d821e..e0dae98 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,4 +2,6 @@
     public void testHexStringToByteArray() {
         byte[] temp = { (byte) 0x12, (byte) 0xaf };
         assertTrue(Arrays.equals(temp, ParseUtil.hexStringToByteArray("12af")));
+        assertEquals(null, ParseUtil.hexStringToByteArray("abcde"));
+        assertEquals(null, ParseUtil.hexStringToByteArray("not hex"));
     }
\ No newline at end of file
