diff --git a/old_test.java b/new_test.java
index c9e3817..595379e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -19,7 +19,7 @@
                 latch.countDown();
             }
         };
-        mController = createController(mToken2, true, callback);
+        mController = createController(mSession.getSessionToken(), true, callback);
 
         mSession.setQueue(testQueue);
         assertTrue(latch.await(TIMEOUT_MS, TimeUnit.MILLISECONDS));
