diff --git a/old_product.java b/new_product.java
index c4242dc..d0af91e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 private void degreesAndPersist(String resourceType) {
-        insertOntology(resourceType, Data.LONG);
+        insertOntology(resourceType, ResourceType.DataType.LONG);
         ComputerResult result = computer.compute(new DegreeAndPersistVertexProgram(keySpace,allTypes));
 
         // TODO: get rid of this in the future MASSIVE bottleneck
