diff --git a/old_product.java b/new_product.java
index a0265c3..0d21d48 100644
--- a/old_product.java
+++ b/new_product.java
@@ -24,7 +24,7 @@ final ItemList eval(final QueryContext qc, final byte[] qu, final String path,
             @Override
             public void run() {
               // limit reached: perform garbage collection and check again
-              if(Performance.memory() > limit) qctx.stop();
+              if(Performance.memory() > limit) qctx.memory();
             }
           }, 500, 500);
         }
@@ -33,7 +33,7 @@ final ItemList eval(final QueryContext qc, final byte[] qu, final String path,
           if(to == null) to = new Timer(true);
           to.schedule(new TimerTask() {
             @Override
-            public void run() { qctx.stop(); }
+            public void run() { qctx.timeout(); }
           }, ms);
         }
       }
@@ -64,7 +64,9 @@ final ItemList eval(final QueryContext qc, final byte[] qu, final String path,
         }
         return cache;
       } catch(final ProcException ex) {
-        throw BXXQ_STOPPED.get(info);
+        if(qctx.state == State.TIMEOUT) throw BXXQ_TIMEOUT.get(info);
+        if(qctx.state == State.MEMORY)  throw BXXQ_MEMORY.get(info);
+        throw ex;
       } catch(final QueryException ex) {
         throw ex.error() == BASX_PERM_X ? BXXQ_PERM_X.get(info, ex.getLocalizedMessage()) :
           ex.info(info);
