diff --git a/old_test.java b/new_test.java
index 9e5a933..540c18d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -29,7 +29,7 @@
     error(_XQUERY_EVAL.args(" '" + _FILE_EXISTS.args("x") + "'", " map { }",
         " map { 'permission': 'none' }"), BXXQ_PERM_X);
     error(_XQUERY_EVAL.args("\"(1 to 10000000000000)[. = 0]\"", " map { }",
-        " map { 'timeout': 1 }"), BXXQ_STOPPED);
+        " map { 'timeout': 1 }"), BXXQ_TIMEOUT);
     error(_XQUERY_EVAL.args("\"(1 to 10000000000000) ! <a/>\"", " map { }",
-        " map { 'memory': 10 }"), BXXQ_STOPPED);
+        " map { 'memory': 10 }"), BXXQ_MEMORY);
   }
\ No newline at end of file
