diff --git a/old_product.java b/new_product.java
index 0156efd..d04d449 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public boolean fit( List<Point2D_I32> contour , GrowQueue_I32 corners ,
+public void fit( List<Point2D_I32> contour , GrowQueue_I32 corners ,
 						Quadrilateral_F64 output )
 	{
 		// pick the 4 largest segments to act as the initial seeds
@@ -7,8 +7,13 @@ public boolean fit( List<Point2D_I32> contour , GrowQueue_I32 corners ,
 			int next = (i+1)%corners.size;
 			segments.grow().set(corners.get(i),corners.get(next),contour.size());
 		}
-		sorter.sort(segments.data,segments.size);
+		sorterLength.sort(segments.data, segments.size);
 
+		// order the lines so that they can be converted into a quad later on easily
+		// bubble sort below
+		bubbleSortLines(segments);
+
+		// now create the lines
 		for (int i = 0; i < 4; i++) {
 			createLine(segments.get(i), contour, lines[i]);
 		}
@@ -16,11 +21,6 @@ public boolean fit( List<Point2D_I32> contour , GrowQueue_I32 corners ,
 		// estimate line equations
 		performLineEM(contour);
 
-		// compute the quality of the fit and decide if it's valid or not
-		// TODO write this part
-
 		// convert from lines to quadrilateral
 		convert(lines,output);
-
-		return true;
 	}
\ No newline at end of file
