diff --git a/old_test.java b/new_test.java
index 5bd21dc..b9f9dad 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,5 +2,10 @@
 	@Verifies(value = "should properly set the date", method = "setDate(int,int,int,null,null,null)")
 	public void setDate_shouldProperlySetTheDate() throws Exception {
 		//TODO auto-generated
+		ApproximateDate approximateDate = new ApproximateDate();
+		approximateDate.setDate(1988, 9, 15, false, false, false);
+		Calendar cal = Calendar.getInstance();
+		cal.set(1988, 9, 15);
+		//		cal.ge
 		Assert.fail("Not yet implemented");
 	}
\ No newline at end of file
