diff --git a/old_product.java b/new_product.java
index dee016a..5af7d48 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,7 +7,7 @@ void process(Operation operation) throws DataCorruptionException {
         // Add entry to MemoryTransactionLog and ReadIndex/Cache. This callback is invoked from the QueueProcessor,
         // which always acks items in order of Sequence Number - so the entries should be ordered (but always check).
         if (operation instanceof StorageOperation) {
-            this.cacheUpdater.addToReadIndex((StorageOperation) operation);
+            addToReadIndex((StorageOperation) operation);
             if (operation instanceof StreamSegmentAppendOperation) {
                 // Transform a StreamSegmentAppendOperation into its corresponding Cached version.
                 try {
