diff --git a/old_test.java b/new_test.java
index bc1c974..8721cbe 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
         TestReadIndex readIndex = new TestReadIndex(methodInvocations::add);
         @Cleanup
         InMemoryCache cache = new InMemoryCache("0");
-        MemoryStateUpdater updater = new MemoryStateUpdater(opLog, new CacheUpdater(cache, readIndex));
+        MemoryStateUpdater updater = new MemoryStateUpdater(opLog, readIndex);
         ArrayList<Operation> operations = populate(updater, segmentCount, operationCountPerType);
 
         // Verify they were properly processed.
