diff --git a/old_test.java b/new_test.java
index f75a801..8e86182 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,6 +9,7 @@
 		p.addName(new PersonName("hor", null, "ty"));
 		p.addIdentifier(new PatientIdentifier("hiuh", new PatientIdentifierType(2), new Location(1)));
 		ShortPatientModel model = new ShortPatientModel(p);
+		model.setPersonAddress(new PersonAddress());
 		Errors errors = new BindException(model, "patientModel");
 		validator.validate(model, errors);
 		Assert.assertEquals(false, errors.hasErrors());
