diff --git a/old_product.java b/new_product.java
index 508ac02..dfec384 100644
--- a/old_product.java
+++ b/new_product.java
@@ -33,4 +33,37 @@ public void validate(Object object, Errors errors) {
 			        + "'" }, "The End Date for address '" + addressString + "' shouldn't be earlier than the Start Date");
 		}
 		
+		String xml = Context.getLocationService().getAddressTemplate();
+		List<String> requiredElements;
+		
+		try {
+			AddressTemplate addressTemplate = Context.getSerializationService().getDefaultSerializer().deserialize(xml,
+			    AddressTemplate.class);
+			requiredElements = addressTemplate.getRequiredElements();
+		}
+		catch (Exception e) {
+			errors.reject(Context.getMessageSourceService().getMessage("AddressTemplate.error"));
+			return;
+		}
+		
+		if (requiredElements != null) {
+			for (String fieldName : requiredElements) {
+				try {
+					Object value = PropertyUtils.getProperty(personAddress, fieldName);
+					if (StringUtils.isBlank((String) value)) {
+						//required field not found
+						errors.reject(Context.getMessageSourceService().getMessage(
+						    "AddressTemplate.error.requiredAddressFieldIsBlank", new Object[] { fieldName },
+						    Context.getLocale()));
+					}
+				}
+				catch (Exception e) {
+					//wrong field declared in template
+					errors
+					        .reject(Context.getMessageSourceService().getMessage(
+					            "AddressTemplate.error.fieldNotDeclaredInTemplate", new Object[] { fieldName },
+					            Context.getLocale()));
+				}
+			}
+		}
 	}
\ No newline at end of file
