diff --git a/old_test.java b/new_test.java
index ccd7856..06fb6c7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testStringToCommandCriteria() throws GenieException {
+    public void testStringToCommandCriteria() throws GeniePreconditionException {
         Assert.assertEquals(COMMAND_CRITERIA,
                 this.job.stringToCommandCriteria(
                         EXPECTED_COMMAND_CRITERIA_STRING));
