diff --git a/old_test.java b/new_test.java
index 17ffbad..f0aa8a2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,7 @@
 @Test
     public void testOnCreateOrUpdateCluster() throws GeniePreconditionException {
-        this.c = new Cluster(NAME, USER, ClusterStatus.UP, CLUSTER_TYPE, this.configs, VERSION);
+        this.c = new Cluster(NAME, USER, VERSION, ClusterStatus.UP, CLUSTER_TYPE);
+        Assert.assertNull(this.c.getTags());
         this.c.onCreateOrUpdateCluster();
+        Assert.assertEquals(2, this.c.getTags().size());
     }
\ No newline at end of file
