diff --git a/old_product.java b/new_product.java
index e0f6813..b74d04a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,6 @@
 @PrePersist
     @PreUpdate
     protected void onCreateOrUpdateCluster() throws GeniePreconditionException {
-        validate(this.status, this.clusterType, null);
         // Add the id to the tags
         if (this.tags == null) {
             this.tags = new HashSet<>();
