diff --git a/old_test.java b/new_test.java
index c1e0577..ae0b098 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,8 +7,8 @@
 		Assert.assertTrue(order.getFiller() == null);
 		Assert.assertTrue(order.getDateFilled() == null);
 		
-		Context.getOrderService().signOrder(order, provider);
-		Context.getOrderService().fillOrder(order, "url");
+		Context.getOrderService().signOrder(order, provider, null);
+		Context.getOrderService().fillOrder(order, "url", null);
 		
 		order = Context.getOrderService().getOrder(10);
 		Assert.assertTrue(order.getFiller() != null);
