diff --git a/old_product.java b/new_product.java
index 85f42f5..9242ecb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,18 @@
-private void setState(Ignite ignite, ClusterState state, String strState) {
+private void setState(Ignite ignite, ClusterState state, String strState, String... cacheNames) {
         assertEquals(EXIT_CODE_OK, execute("--set-state", strState));
 
         assertEquals(state, ignite.cluster().state());
+
+        assertContains(log, testOut.toString(), "Cluster state changed to " + strState);
+
+        List<IgniteEx> nodes = IntStream.range(0, 2)
+            .mapToObj(this::grid)
+            .collect(Collectors.toList());
+
+        ClusterStateTestUtils.putSomeDataAndCheck(log, nodes, cacheNames);
+
+        if (state == ACTIVE) {
+            for (String cacheName : cacheNames)
+                grid(0).cache(cacheName).clear();
+        }
     }
\ No newline at end of file
