diff --git a/old_product.java b/new_product.java
index 38e2f00..f099f6a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public void various() {
 		Helper detector = new Helper();
 		detector.maximum = true;
 		GeneralToInterestMulti<GrayF32,GrayF32> alg =
-				new GeneralToInterestMulti<GrayF32,GrayF32>(detector,2.5, GrayF32.class,GrayF32.class);
+				new GeneralToInterestMulti<>(detector, 2.5, GrayF32.class, GrayF32.class);
 
 		alg.detect(input);
 
