diff --git a/old_test.java b/new_test.java
index 3c0876f..437fa86 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,14 @@
 @Test
 	public void various() {
 
-		Dummy<Integer> nn = new Dummy<Integer>();
+		Dummy<Integer> nn = new Dummy<>();
 		// src = assoc[i] where src is the index of the source feature and i is the index of the dst feature
 		nn.assoc = new int[]{2,0,1,-1,4,-1,-1,2,2,1};
 
-		AssociateNearestNeighbor<TupleDesc_F64> alg = new AssociateNearestNeighbor<TupleDesc_F64>(nn,10);
+		AssociateNearestNeighbor<TupleDesc_F64> alg = new AssociateNearestNeighbor<>(nn, 10);
 
-		FastQueue<TupleDesc_F64> src = new FastQueue<TupleDesc_F64>(10,TupleDesc_F64.class,false);
-		FastQueue<TupleDesc_F64> dst = new FastQueue<TupleDesc_F64>(10,TupleDesc_F64.class,false);
+		FastQueue<TupleDesc_F64> src = new FastQueue<>(10, TupleDesc_F64.class, false);
+		FastQueue<TupleDesc_F64> dst = new FastQueue<>(10, TupleDesc_F64.class, false);
 
 		for( int i = 0; i < 5; i++ ) {
 			src.add( new TupleDesc_F64(10));
