diff --git a/old_product.java b/new_product.java
index 6cdafcb..f1ce5b5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public void addIdentifier(String type, String value, String title, String url) {
-        Identifier i = new Identifier(type, value, title, url);
+public void addIdentifier(String type, String value, String url) {
+        Identifier i = new Identifier(type, value, url);
         this.identifiers.add(i);
     }
\ No newline at end of file
