diff --git a/old_test.java b/new_test.java
index b1d2037..1500928 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,5 +9,5 @@
     get("declare %R:path('') %R:produces('text/plain') %R:produces('*/*') " +
         "function m:f() {1};", "", "1");
     // invalid content type
-    getE("declare %R:path('') %R:produces('X') function m:f() {1};", "");
+    //getE("declare %R:path('') %R:produces('X') function m:f() {1};", "");
   }
\ No newline at end of file
