diff --git a/old_product.java b/new_product.java
index 02f96b4..3963583 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,7 @@
 public O remove(String key)
     {
         if (key==null)
-        {
-            O oldValue=_nullValue;
-            if (_nullEntry!=null)
-            {
-                _entrySet.remove(_nullEntry);   
-                _nullEntry=null;
-                _nullValue=null;
-            }
-            return oldValue;
-        }
+            throw new IllegalArgumentException();
         
         Node<O> node = _root;
         int ni=-1;
@@ -25,7 +16,7 @@ public O remove(String key)
             if (ni==-1)
             {
                 ni=0;
-                node=(node._children==null)?null:node._children[c%_width];
+                node=(node._children==null)?null:node._children[c%__HASH_WIDTH];
             }
             
             // While we have a node to try
