diff --git a/old_test.java b/new_test.java
index 85add7d..6a4952c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,15 +6,13 @@
         m5.remove("aBc");
         m5.remove("bbb");
         m5i.remove("aBc");
-        m5i.remove(null);
 
         assertEquals(0, m0.size());
         assertEquals(0, m1.size());
         assertEquals(4, m5.size());
-        assertEquals(3, m5i.size());
+        assertEquals(2, m5i.size());
 
         assertEquals("2",m5.get("abc"));
         assertEquals(null,m5.get("bbb"));
         assertEquals(null,m5i.get("AbC"));
-        assertEquals(null,m5i.get(null));
     }
\ No newline at end of file
