diff --git a/old_product.java b/new_product.java
index 0147909..a95e4ce 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
 public boolean setDescription(PyramidKltFeature feature) {
-		boolean valid = false;
 		for (int layer = 0; layer < image.getNumLayers(); layer++) {
 			float scale = (float)image.getScale(layer);
 			float x = feature.x / scale;
@@ -9,9 +8,7 @@ public boolean setDescription(PyramidKltFeature feature) {
 
 			feature.desc[layer].setPosition(x, y);
 			if( !tracker.setDescription(feature.desc[layer]) )
-				break;
-			feature.maxLayer = layer;
-			valid = true;
+				return false;
 		}
-		return valid;
+		return true;
 	}
\ No newline at end of file
