diff --git a/old_test.java b/new_test.java
index d7a2d30..be14d97 100644
--- a/old_test.java
+++ b/new_test.java
@@ -19,7 +19,7 @@
 
         tracker.writeLockState();
 
-        IgniteInternalFuture<Map<GridCacheVersion, WALPointer>> fut;
+        IgniteInternalFuture<Set<GridCacheVersion>> fut;
         try {
             tracker.startTxFinishAwaiting(1_000, 10_000);
 
@@ -37,14 +37,14 @@
 
         long curTs = U.currentTimeMillis();
 
-        Map<GridCacheVersion, WALPointer> pendingTxs = fut.get();
+        Set<GridCacheVersion> pendingTxs = fut.get();
 
         assertTrue("Waiting for awaitFinishOfPreparedTxs future too long", U.currentTimeMillis() - curTs < 1_000);
 
         assertEquals(3, pendingTxs.size());
-        assertTrue(pendingTxs.keySet().contains(nearXidVersion(1)));
-        assertTrue(pendingTxs.keySet().contains(nearXidVersion(3)));
-        assertTrue(pendingTxs.keySet().contains(nearXidVersion(5)));
+        assertTrue(pendingTxs.contains(nearXidVersion(1)));
+        assertTrue(pendingTxs.contains(nearXidVersion(3)));
+        assertTrue(pendingTxs.contains(nearXidVersion(5)));
 
         txCommit(1);
         txCommit(3);
