diff --git a/old_product.java b/new_product.java
index 7d476eb..399c759 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,8 @@
-private Map<GridCacheVersion, WALPointer> awaitFinishOfPreparedTxs(long preparedTxsTimeout,
-        long committingTxsTimeout) throws IgniteCheckedException {
-        IgniteInternalFuture<Map<GridCacheVersion, WALPointer>> fut;
+private Set<GridCacheVersion> awaitFinishOfPreparedTxs(
+        long preparedTxsTimeout,
+        long committingTxsTimeout
+    ) throws IgniteCheckedException {
+        IgniteInternalFuture<Set<GridCacheVersion>> fut;
 
         tracker.writeLockState();
 
