diff --git a/old_product.java b/new_product.java
index 1ac43c3..e358837 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,10 +4,10 @@
         long traceId = LoggerHelpers.traceEnter(log, "createScope");
 
         controllerService.createScope(createScopeRequest.getScopeName()).thenApply(scopeStatus -> {
-            if (scopeStatus == CreateScopeStatus.SUCCESS) {
+            if (scopeStatus.getStatus() == CreateScopeStatus.Status.SUCCESS) {
                 log.info("Successfully created new scope: {}", createScopeRequest.getScopeName());
                 return Response.status(Status.CREATED).entity(createScopeRequest).build();
-            } else if (scopeStatus == CreateScopeStatus.SCOPE_EXISTS) {
+            } else if (scopeStatus.getStatus() == CreateScopeStatus.Status.SCOPE_EXISTS) {
                 log.warn("Scope name: {} already exists", createScopeRequest.getScopeName());
                 return Response.status(Status.CONFLICT).build();
             } else {
