diff --git a/old_product.java b/new_product.java
index 4b271ba..4ac3144 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 @GET
         @Path("timeout")
         public void asyncGetWithTimeout(@Suspended final AsyncResponse asyncResponse) {
+            LOGGER.info("Async long-running get with timeout called on thread " + Thread.currentThread().getName());
             asyncResponse.setTimeoutHandler(new TimeoutHandler() {
 
                 @Override
@@ -9,7 +10,7 @@
                             .entity("Operation time out.").build());
                 }
             });
-            asyncResponse.setTimeout(3, TimeUnit.SECONDS);
+            asyncResponse.setTimeout(1, TimeUnit.SECONDS);
 
             new Thread(new Runnable() {
 
@@ -20,13 +21,17 @@
                 }
 
                 private String veryExpensiveOperation() {
-                    // ... very expensive operation that typically finishes within 10 seconds, simulated using sleep()
+                    // very expensive operation that typically finishes within 1 second but can take up to 5 seconds,
+                    // simulated using sleep()
                     try {
-                        Thread.sleep(7000);
+                        Thread.sleep(5 * OPERATION_DURATION);
+                        return "DONE";
                     } catch (InterruptedException e) {
-                        // ignore
+                        Thread.currentThread().interrupt();
+                        return "INTERRUPTED";
+                    } finally {
+                        LOGGER.info("Async long-running get with timeout finished on thread " + Thread.currentThread().getName());
                     }
-                    return "DONE";
                 }
             }).start();
         }
\ No newline at end of file
