diff --git a/old_test.java b/new_test.java
index 09d5763..fcd5e96 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testAsyncGetWithTimeout() throws ExecutionException, InterruptedException, TimeoutException {
+    public void testAsyncGetWithTimeout() throws Exception {
         final Future<Response> responseFuture = target(PATH).path("timeout").request().async().get();
         // Request is being processed asynchronously.
         final Response response = responseFuture.get();
