diff --git a/old_test.java b/new_test.java
index 893c9b9..0113bcb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,13 +7,13 @@
 
       final String[] good = TEXTS_GOOD[i];
       for(final String element : good) {
-        assertTrue("\"" + q + "\" did NOT match \"" + element + "\"",
+        assertTrue('"' + q + "\" did NOT match \"" + element + '"',
             wc.match(token(element)));
       }
 
       final String[] bad = TEXTS_BAD[i];
       for(final String element : bad) {
-        assertFalse("\"" + q + "\" matched \"" + element + "\"",
+        assertFalse('"' + q + "\" matched \"" + element + '"',
             wc.match(token(element)));
       }
     }
