diff --git a/old_product.java b/new_product.java
index 82b061c..793fdea 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CorpusStats getCorpusStats(final Corpus corpus) {
-		String corpusId = IDHelper.getCorpusId(corpus, getAccountId());
-		return executeRequest(API_VERSION + corpusId + STATS_PATH, null, CorpusStats.class);
-	}
\ No newline at end of file
+    final String corpusId = IDHelper.getCorpusId(corpus, getAccountId());
+    return executeRequest(API_VERSION + corpusId + STATS_PATH, null, CorpusStats.class);
+  }
\ No newline at end of file
