diff --git a/old_product.java b/new_product.java
index 9d6bab2..1603805 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,6 @@
 CompletableFuture<RegistrationResponse> registerTaskExecutor(
 		String taskExecutorAddress,
 		ResourceID resourceId,
-		SlotReport slotReport,
 		int dataPort,
 		HardwareDescription hardwareDescription,
 		@RpcTimeout Time timeout);
\ No newline at end of file
