diff --git a/old_test.java b/new_test.java
index 108c48c..31f9a7b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,6 +16,7 @@
 		mockRequest.setParameter("action", "");
 		mockRequest.setParameter("namesByLocale[en].name", conceptName);
 		mockRequest.setParameter("concept.datatype", "1");
+		mockRequest.setParameter("concept.conceptClass", "1");
 		mockRequest.setParameter("conceptMappings[0].conceptReferenceTerm", "1");
 		mockRequest.setParameter("conceptMappings[0].conceptMapType", "3");
 		
