diff --git a/old_test.java b/new_test.java
index 19f5580..29fdfa0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,19 +2,19 @@
     public void testCountByExample() {
         SqlSession sqlSession = MybatisHelper.getSqlSession();
         try {
-            CommonMapper entityMapper = sqlSession.getMapper(CommonMapper.class);
-            EntityMapper baseMapper = new EntityMapper(entityMapper);
+            CommonMapper commonMapper = sqlSession.getMapper(CommonMapper.class);
+            EntityMapper entityMapper = new EntityMapper(commonMapper);
 
             Example example = new Example(Country.class);
             example.createCriteria().andGreaterThan("id", 100).andLessThanOrEqualTo("id", 150);
 
-            int count = baseMapper.countByExample(example);
+            int count = entityMapper.countByExample(example);
             Assert.assertEquals(50, count);
 
             example = new Example(Country.class);
-            example.createCriteria().andLike("countryname","A%");
+            example.createCriteria().andLike("countryname", "A%");
 
-            count = baseMapper.countByExample(example);
+            count = entityMapper.countByExample(example);
             Assert.assertEquals(12, count);
         } finally {
             sqlSession.close();
