diff --git a/old_test.java b/new_test.java
index aa474af..f8a6891 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testGetPathToRoot() {
         System.out.println("getPathToRoot");
-        LinkedList<Concept> expResult = new LinkedList<Concept>();
+        LinkedList<Concept> expResult = new LinkedList<>();
         expResult.addFirst(instance.getRoot());
         expResult.addFirst(ad);
         expResult.addFirst(a);
