diff --git a/old_product.java b/new_product.java
index a2e1217..e10202d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-public abstract void update(ByteBuffer rowKey, Cell oldCol, Cell col, OpOrder.Group opGroup);
\ No newline at end of file
+public abstract void update(ByteBuffer rowKey, Clustering clustering, Cell oldCell, Cell cell, OpOrder.Group opGroup, int nowInSec);
\ No newline at end of file
