diff --git a/old_test.java b/new_test.java
index f2c08a8..9b763e9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test
   public void free() throws Exception {
-    sNestedFileOptions.setPersisted(true);
+    mNestedFileOptions.setPersisted(true);
     long blockId = createFileWithSingleBlock(NESTED_FILE_URI);
     Assert.assertEquals(1, mBlockMaster.getBlockInfo(blockId).getLocations().size());
 
     // free the file
-    Assert.assertTrue(mFileSystemMaster.free(NESTED_FILE_URI,
-        FreeOptions.defaults().setForced(false).setRecursive(false)));
+    mFileSystemMaster.free(NESTED_FILE_URI,
+        FreeOptions.defaults().setForced(false).setRecursive(false));
     // Update the heartbeat of removedBlockId received from worker 1
     Command heartbeat2 =
         mBlockMaster.workerHeartbeat(mWorkerId1, ImmutableMap.of("MEM", (long) Constants.KB),
@@ -14,5 +14,4 @@
     // Verify the muted Free command on worker1
     Assert.assertEquals(new Command(CommandType.Nothing, ImmutableList.<Long>of()), heartbeat2);
     Assert.assertEquals(0, mBlockMaster.getBlockInfo(blockId).getLocations().size());
-    sNestedFileOptions.setPersisted(false);
   }
\ No newline at end of file
