diff --git a/old_product.java b/new_product.java
index d173347..3719a49 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,13 +2,14 @@
   @Path(FREE)
   @ReturnType("java.lang.Void")
   public Response free(@QueryParam("path") final String path,
-      @QueryParam("recursive") final boolean recursive) {
+      @QueryParam("recursive") final boolean recursive,
+      @QueryParam("forced") final boolean forced) {
     return RestUtils.call(new RestUtils.RestCallable<Void>() {
       @Override
       public Void call() throws Exception {
         Preconditions.checkNotNull(path, "required 'path' parameter is missing");
         mFileSystemMaster.free(new AlluxioURI(path),
-            FreeOptions.defaults().setRecursive(recursive).setForced(true));
+            FreeOptions.defaults().setRecursive(recursive).setForced(forced));
         return null;
       }
     });
