diff --git a/old_product.java b/new_product.java
index 61cd6e4..d9150bb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
-	protected boolean acceptAbsolutePath(String path)
+	public boolean accept(String path)
 	{
 		// First check the cache
 		if (cache != null)
@@ -12,7 +12,7 @@
 		}
 
 		// Check typical files such as log4j.xml etc.
-		if (super.acceptAbsolutePath(path) == false)
+		if (super.accept(path) == false)
 		{
 			return false;
 		}
