diff --git a/old_test.java b/new_test.java
index 53cbc6f..8bbf159 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,8 +3,8 @@
 	{
 		SecurePackageResourceGuard guard = new SecurePackageResourceGuard();
 		guard.addPattern("+*.gif");
-		assertFalse(guard.acceptAbsolutePath("test.gif"));
-		assertFalse(guard.acceptAbsolutePath("/test.gif"));
-		assertTrue(guard.acceptAbsolutePath("mydir/test.gif"));
-		assertTrue(guard.acceptAbsolutePath("/root/mydir/test.gif"));
+		assertFalse(guard.accept("test.gif"));
+		assertFalse(guard.accept("/test.gif"));
+		assertTrue(guard.accept("mydir/test.gif"));
+		assertTrue(guard.accept("/root/mydir/test.gif"));
 	}
\ No newline at end of file
