diff --git a/old_test.java b/new_test.java
index 0f1455a..21c10ef 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
       final String query = test[1].isEmpty() ? _CSV_SERIALIZE.args(test[0]) :
         _CSV_SERIALIZE.args(test[0], " {" + test[1] + "}");
       if(test.length == 2) {
-        error(query, Err.BXCS_CONFIG);
+        error(query, Err.BXCS_CONFSEP, Err.ELMOPTION);
       } else {
         query(query, test[2]);
       }
