diff --git a/old_test.java b/new_test.java
index fcca2ba..cc659ee 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void createDB() throws BaseXException {
-    new CreateDB(DBNAME).execute(CONTEXT);
+    new CreateDB(DB).execute(CONTEXT);
     // check if database name equals argument of create command
-    assertEquals(dbName(), DBNAME);
+    assertEquals(db(), DB);
   }
\ No newline at end of file
