diff --git a/old_test.java b/new_test.java
index 148f5c1..eb08f9f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
         public void testFindNMatches() {
-            List<File> files = new ArrayList<>();
+            List<Path> files = new ArrayList<>();
             files.add(new File(String.join(File.separator, "src", "test", "resources"),
-                    "logviewer-search-context-tests.log.test"));
+                    "logviewer-search-context-tests.log.test").toPath());
             files.add(new File(String.join(File.separator, "src", "test", "resources"),
-                    "logviewer-search-context-tests.log.gz"));
+                    "logviewer-search-context-tests.log.gz").toPath());
 
             final LogviewerLogSearchHandler handler = getSearchHandler();
 
