diff --git a/old_test.java b/new_test.java
index a60b8e5..d2983b2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,10 @@
 @Test
    public void testSetCmdRegExps() {
       System.out.println("setCmdRegExps");
-      HashMap<String, String> cmdRegExps = new HashMap<String, String>();
-      cmdRegExps.put("regKey", "regVal");
+      ArrayList<Object> data = new ArrayList<Object>();
+      data.add("regKey");
+      data.add("regVal");
+      data.add(true);
       PageDataExtractorOverTimeGui instance = new PageDataExtractorOverTimeGui();
-      instance.setCmdRegExps(cmdRegExps);
+      instance.setCmdRegExps(data);
    }
\ No newline at end of file
