diff --git a/old_test.java b/new_test.java
index bbfe63a..c0b1eb9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -28,7 +28,7 @@
     Scanner scanner = conn.createScanner(MetadataTable.NAME, Authorizations.EMPTY);
     scanner.setRange(new KeyExtent(new Text("1"), null, null).toMetadataRange());
 
-    HashSet<String> files = new HashSet<String>();
+    HashSet<String> files = new HashSet<>();
 
     int count = 0;
     for (Entry<Key,Value> entry : scanner) {
