diff --git a/old_product.java b/new_product.java
index 7d163de..b8980dd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static byte[][] split(byte[] data, int dataOffset, int len) {
-    ArrayList<Integer> offsets = new ArrayList<Integer>();
+    ArrayList<Integer> offsets = new ArrayList<>();
 
     for (int i = dataOffset; i < (dataOffset + len); i++) {
       if (data[i] == 0x00) {
