diff --git a/old_product.java b/new_product.java
index b34d85b..4525706 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,14 +1,7 @@
 public TranslationModel getModel(String modelId) {
-		if (modelId == null || modelId.isEmpty())
-			throw new IllegalArgumentException("model_id can not be null or empty");
+    if (modelId == null || modelId.isEmpty())
+      throw new IllegalArgumentException("model_id cannot be null or empty");
 
-		HttpRequestBase request = Request.Get("/v2/models/" + modelId).build();
-		try {
-			HttpResponse response = execute(request);
-			String modelAsString = ResponseUtil.getString(response);
-			TranslationModel model = GsonSingleton.getGson().fromJson(modelAsString, TranslationModel.class);
-			return model;
-		} catch (IOException e) {
-			throw new RuntimeException(e);
-		}
-	}
\ No newline at end of file
+    final Request request = RequestBuilder.get("/v2/models/" + modelId).build();
+    return executeRequest(request, TranslationModel.class);
+  }
\ No newline at end of file
