diff --git a/old_test.java b/new_test.java
index e2f07f2..076cac2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,24 +1,26 @@
 @Test
-	public void testGetModel() {
-		// Mock response
-		TranslationModel tm = new TranslationModel();
-		String model_id = "not-a-real-model";
-		tm.setModelId(model_id);
-		tm.setSource("en");
-		tm.setBaseModelId("en-es");
-		tm.setDomain("news");
-		tm.setCustomizable(false);
-		tm.setDefaultModel(false);
-		tm.setOwner("not-a-real-user");
-		tm.setStatus("error");
-		tm.setName("custom-english-to-spanish");
+  public void testGetModel() {
+    // Mock response
+    final TranslationModel tm = new TranslationModel();
+    final String model_id = "not-a-real-model";
+    tm.setModelId(model_id);
+    tm.setSource("en");
+    tm.setBaseModelId("en-es");
+    tm.setDomain("news");
+    tm.setCustomizable(false);
+    tm.setDefaultModel(false);
+    tm.setOwner("not-a-real-user");
+    tm.setStatus("error");
+    tm.setName("custom-english-to-spanish");
 
-		mockServer.when(request().withPath(GET_MODELS_PATH + "/" + model_id)).respond(
-				response().withHeaders(new Header(HttpHeaders.Names.CONTENT_TYPE, MediaType.APPLICATION_JSON))
-						.withBody(GsonSingleton.getGson().toJson(tm)));
+    mockServer.when(request().withPath(GET_MODELS_PATH + "/" + model_id)).respond(
+        response().withHeaders(
+            new Header(HttpHeaders.Names.CONTENT_TYPE, HttpMediaType.APPLICATION_JSON)).withBody(
+            GsonSingleton.getGson().toJson(tm)));
 
-		TranslationModel model = service.getModel("not-a-real-model");
+    final TranslationModel model = service.getModel("not-a-real-model");
 
-		mockServer.verify(request().withPath(GET_MODELS_PATH + "/" + model_id), VerificationTimes.exactly(1));
-		assertNotNull(model);
-	}
\ No newline at end of file
+    mockServer.verify(request().withPath(GET_MODELS_PATH + "/" + model_id),
+        VerificationTimes.exactly(1));
+    assertNotNull(model);
+  }
\ No newline at end of file
