diff --git a/old_test.java b/new_test.java
index 50ef055..799e505 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testGetSigningKey_nullConfig() {
         try {
-            Key result = consumerUtil.getSigningKey((JwtConsumerConfig) null, (JwtContext) null);
+            Key result = consumerUtil.getSigningKey((JwtConsumerConfig) null, (JwtContext) null, null);
             assertNull("Result was not null when it should have been. Result: " + result, result);
         } catch (Throwable t) {
             outputMgr.failWithThrowable(testName.getMethodName(), t);
