diff --git a/old_product.java b/new_product.java
index 9604b99..1cf03ba 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-Key getSigningKey(JwtConsumerConfig config, JwtContext jwtContext) throws KeyException {
+Key getSigningKey(JwtConsumerConfig config, JwtContext jwtContext, Map properties) throws KeyException {
         Key signingKey = null;
         if (config == null) {
             if (tc.isDebugEnabled()) {
@@ -6,7 +6,7 @@ Key getSigningKey(JwtConsumerConfig config, JwtContext jwtContext) throws KeyExc
             }
             return null;
         }
-        signingKey = getSigningKeyBasedOnSignatureAlgorithm(config, jwtContext);
+        signingKey = getSigningKeyBasedOnSignatureAlgorithm(config, jwtContext, properties);
         if (signingKey == null) {
             if (tc.isDebugEnabled()) {
                 Tr.debug(tc, "A signing key could not be found");
