diff --git a/old_product.java b/new_product.java
index 79c4ddf..d42187a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -33,7 +33,9 @@
 				if (log.isDebugEnabled()) {
 					log.debug("The following fields cannot be changed for " + getSupportedType() + ":" + changedProperties);
 				}
-				throw new APIException("Editing some fields " + changedProperties + " on " + getSupportedType().getSimpleName() + " is not allowed");
+				
+				throw new UnchangeableObjectException("editing.fields.not.allowed", new Object[] { changedProperties,
+				        getSupportedType().getSimpleName() });
 			}
 		}
 		
