diff --git a/old_test.java b/new_test.java
index 7896e5f..9153495 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,8 +4,9 @@
 		String[] previousState = new String[] { "old" };
 		String[] currentState = new String[] { "new" };
 		ImmutableEntityInterceptor interceptor = new SomeImmutableEntityInterceptor();
-		expectedException.expect(APIException.class);
-		expectedException.expectMessage(is(Context.getMessageSourceService().getMessage("editing.fields.not.allowed", new Object[] { "[immutable]", Order.class.getSimpleName() },  null)));
+		expectedException.expect(UnchangeableObjectException.class);
+		expectedException.expectMessage(is(Context.getMessageSourceService().getMessage("editing.fields.not.allowed",
+		    new Object[] { "[immutable]", Order.class.getSimpleName() }, null)));
 		Order order = new Order();
 		order.setVoided(true);
 		interceptor.onFlushDirty(order, null, currentState, previousState, propertyNames, null);
