diff --git a/old_test.java b/new_test.java
index 455eaf2..dc647d9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
-@Test(dataProvider="sampleToString", groups={"tck"})
+@Test(dataProvider="sampleToString")
     public void test_parse(int y, int month, int d, int h, int m, int s, int n, String offsetId, String text) {
         OffsetDateTime t = OffsetDateTime.parse(text);
         assertEquals(t.getYear(), y);
