diff --git a/old_test.java b/new_test.java
index b3c7843..37ce108 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,7 +15,8 @@
           throw new IllegalStateException("No ID was set before making a call to getBlobReadInfo()");
         }
         IndexValue value = state.getExpectedValue(id, true);
-        return new MessageInfo(id, value.getSize(), value.getExpiresAtMs());
+        return new MessageInfo(id, value.getSize(), value.getExpiresAtMs(), value.getAccountId(),
+            value.getContainerId(), value.getOperationTimeInMs());
       }
     };
     state.reloadIndex(true, false);
