diff --git a/old_product.java b/new_product.java
index 4b7b3c8..e5ec136 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,7 +16,7 @@ public static double min(DifferentiableMultivariateFunction func, double[] x, do
 
         // Calculate starting function value and gradient and initialize the
         // inverse Hessian to the unit matrix.
-        double f = func.f(x, g);
+        double f = func.applyWithGradient(x, g);
         logger.info(String.format("BFGS: initial function value: %.5g", f));
 
         double sum = 0.0;
@@ -59,7 +59,7 @@ public static double min(DifferentiableMultivariateFunction func, double[] x, do
             
             System.arraycopy(g, 0, dg, 0, n);
 
-            func.f(x, g);
+            func.applyWithGradient(x, g);
 
             // Test for convergence on zero gradient.
             den = Math.max(f, 1.0);
