diff --git a/old_product.java b/new_product.java
index 9d8bed9..8b18fd2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@ public String getLicense()
         if (license == null || license.trim().equals(""))
         {
             // Fallback to site-wide default
-            license = ConfigurationManager.getDefaultSubmissionLicense();
+            license = LicenseManager.getDefaultSubmissionLicense();
         }
 
         return license;
