diff --git a/old_product.java b/new_product.java
index 222339c..5aa9d76 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,14 +1,12 @@
-static public void setAgentPatterns(List<String> agentPatterns)
+static public void setAgentPatterns(List<AgentPatternList> agentPatternLists)
     {
-        if (null == agents)
-            agents = new ArrayList<Pattern>(agentPatterns.size());
+        clearAgentPatterns();
 
-        int nPatterns = 0;
-        for (String agentPattern : agentPatterns)
-        {
-            Pattern newPattern = Pattern.compile(agentPattern);
-            agents.add(newPattern);
-            nPatterns++;
-        }
-        log.info("Received " + String.valueOf(nPatterns) + " agent patterns.");
+        for (AgentPatternList agentPatterns : agentPatternLists)
+            for (String agentPattern : agentPatterns.getPatterns())
+            {
+                Pattern newPattern = Pattern.compile(agentPattern);
+                agents.add(newPattern);
+            }
+        log.info("Received " + String.valueOf(agents.size()) + " agent patterns.");
     }
\ No newline at end of file
