diff --git a/old_test.java b/new_test.java
index 0113bcb..cb58841 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
-  public void testMatch() throws QueryException {
+  public void testMatch() {
     for(int i = 0; i < VALIDWC.length; i++) {
-
       final String q = VALIDWC[i];
-      final FTWildcard wc = new FTWildcard(token(q), null);
+      final FTWildcard wc = new FTWildcard();
+      assertTrue(wc.parse(token(q)));
 
       final String[] good = TEXTS_GOOD[i];
       for(final String element : good) {
