diff --git a/old_product.java b/new_product.java
index 105f553..28eba2a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,8 +2,7 @@ private boolean match(final int[] t, final int tp, final int qp) {
     int qi = qp;
     int ti = tp;
     final int tl = t.length;
-    final int wl = wc.length;
-    while(qi < wl) {
+    while(qi < size) {
       if(wc[qi] == DOT) {
         int n = min[qi];
         final int m = max[qi++];
