diff --git a/old_product.java b/new_product.java
index cd2b907..1f5a3d2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-static Collection<Migration> collectMigrations(Collection<MigrationResolver> migrationResolvers) {
-        Set<Migration> migrations = new HashSet<Migration>();
+static Collection<ExecutableMigration> collectMigrations(Collection<MigrationResolver> migrationResolvers) {
+        Set<ExecutableMigration> migrations = new HashSet<ExecutableMigration>();
         for (MigrationResolver migrationResolver : migrationResolvers) {
             migrations.addAll(migrationResolver.resolveMigrations());
         }
