diff --git a/old_product.java b/new_product.java
index f188647..ce42a2f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,25 +1,4 @@
-public void setPreferredName(Locale locale, ConceptName preferredName) {
-		ConceptNameTag preferredLanguage = ConceptNameTag.preferredLanguageTagFor(locale);
-		ConceptNameTag preferredCountry = ConceptNameTag.preferredCountryTagFor(locale);
-		
-		ConceptName currentPreferredNameInLanguage = getPreferredNameInLanguage(locale.getLanguage());
-		
-		if (preferredCountry != null) {
-			if (currentPreferredNameInLanguage == null) {
-				preferredName.addTag(preferredLanguage);
-			}
-			
-			ConceptName currentPreferredForCountry = getPreferredNameForCountry(locale.getCountry());
-			if (currentPreferredForCountry != null) {
-				currentPreferredForCountry.removeTag(preferredCountry);
-			}
-			preferredName.addTag(preferredCountry);
-		} else {
-			if (currentPreferredNameInLanguage != null) {
-				currentPreferredNameInLanguage.removeTag(preferredLanguage);
-			}
-			preferredName.addTag(preferredLanguage);
-		}
-		
-		addName(preferredName);
+@Deprecated
+	public void setPreferredName(Locale locale, ConceptName preferredName) {
+		setPreferredName(preferredName);
 	}
\ No newline at end of file
