diff --git a/old_test.java b/new_test.java
index 5ecf32e..4160d20 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,6 @@
   public void createUfsFileTest() throws Exception {
     String uniqPath = PathUtils.uniqPath();
     UnderFileSystemManager manager = new UnderFileSystemManager();
-    manager.createFile(uniqPath);
+    manager.createFile(new AlluxioURI(uniqPath));
     Mockito.verify(mMockUfs).create(Mockito.contains(uniqPath));
   }
\ No newline at end of file
