diff --git a/old_product.java b/new_product.java
index 8fb2fa3..c9330a0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public long createUfsFile(String ufsPath) throws FileAlreadyExistsException, IOException {
-    return mUnderFileSystemManager.createFile(ufsPath);
+public long createUfsFile(AlluxioURI ufsUri) throws FileAlreadyExistsException, IOException {
+    return mUnderFileSystemManager.createFile(ufsUri);
   }
\ No newline at end of file
