diff --git a/old_product.java b/new_product.java
index 350858b..2abe819 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,8 +4,8 @@
     public Response getFlowByAppId(@PathParam("appId") String appId) {
         final ApplicationService appService = get(ApplicationService.class);
         final ApplicationId idInstant = nullIsNotFound(appService.getId(appId), APP_ID_NOT_FOUND);
-        final Iterable<FlowRule> flowRules = service.getFlowRulesById(idInstant);
+        final Iterable<FlowEntry> flowEntries = service.getFlowEntriesById(idInstant);
 
-        flowRules.forEach(flow -> flowsNode.add(codec(FlowRule.class).encode(flow, this)));
+        flowEntries.forEach(flow -> flowsNode.add(codec(FlowEntry.class).encode(flow, this)));
         return ok(root).build();
     }
\ No newline at end of file
