diff --git a/old_test.java b/new_test.java
index 9f31615..e7ae45f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testAdjustTLSContext() {
         PaddingExtensionMessage msg = new PaddingExtensionMessage();
-        msg.setPaddingLength(6);
+        msg.setPaddingBytes(new byte[6]);
         handler.adjustTLSContext(msg);
         assertEquals(context.getPaddingExtensionLength(), 6);
     }
\ No newline at end of file
