diff --git a/old_product.java b/new_product.java
index d4f3898..3674631 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,8 @@
 @Override
     public void adjustTLSContext(PaddingExtensionMessage message) {
-        if (message.getPaddingLength().getValue() <= 65535) {
-            context.setPaddingExtensionLength(message.getPaddingLength().getValue());
-        } else {
+        if (message.getPaddingBytes().getValue().length <= 65535) {
             LOGGER.warn("The Padding Extension length value exceeds the two bytes defined in RFC 7685.");
-            context.setPaddingExtensionLength(message.getPaddingLength().getValue());
         }
+        context.setPaddingExtensionLength(message.getPaddingBytes().getValue().length);
+
     }
\ No newline at end of file
