diff --git a/old_product.java b/new_product.java
index 39d2f3d..61b5751 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@
             f.checkValidValue(newValue);
             switch (f) {
                 case DAY_OF_MONTH: return withDayOfMonth((int) newValue);
-                case MONTH_OF_YEAR: return  withMonthOfYear((int) newValue);
+                case MONTH_OF_YEAR: return  withMonth((int) newValue);
             }
             throw new CalendricalException(field.getName() + " not valid for MonthDay");
         }
