diff --git a/old_product.java b/new_product.java
index 78ae350..473825a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,11 +6,10 @@
                 case INSTANT_SECONDS: return create(newValue, getNano(), zoneId);
                 case OFFSET_SECONDS: {
                     ZoneOffset offset = ZoneOffset.ofTotalSeconds(f.checkValidIntValue(newValue));
-                    OffsetDateTime odt = dateTime.withOffsetSameLocal(offset);
-                    return ofInstant(odt, zoneId);
+                    return resolveOffset(offset);
                 }
             }
-            return withDateTime(getDateTime().with(field, newValue));
+            return resolveLocal(dateTime.with(field, newValue));
         }
         return field.doWith(this, newValue);
     }
\ No newline at end of file
