diff --git a/old_test.java b/new_test.java
index d6448b8..ed181da 100644
--- a/old_test.java
+++ b/new_test.java
@@ -28,12 +28,12 @@
         //feature selection configuration
         trainingParameters.setFeatureSelectorTrainingParameters(null);
 
-        Modeler instance = new Modeler(dbName, conf, trainingParameters);
+        Modeler instance = MLBuilder.create(trainingParameters, dbName, conf);
         instance.fit(trainingData);
 
         instance.close();
 
-        instance = new Modeler(dbName, conf);
+        instance = MLBuilder.load(Modeler.class, dbName, conf);
 
         instance.predict(trainingData);
 
@@ -46,7 +46,7 @@
         //instance = null;
 
 
-        instance = new Modeler(dbName, conf);
+        instance = MLBuilder.load(Modeler.class, dbName, conf);
         
         instance.predict(validationData);
         
