diff --git a/old_product.java b/new_product.java
index 55b8324..b8bd0ee 100644
--- a/old_product.java
+++ b/new_product.java
@@ -36,7 +36,7 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
         exParams.setExaminationWindowLength(6);
         trainingParameters.setTextExtractorParameters(exParams);
 
-        TextClassifier instance = new TextClassifier(dbName, conf, trainingParameters);
+        TextClassifier instance = MLBuilder.create(trainingParameters, dbName, conf);
         instance.fit(dataset);
 
 
@@ -48,7 +48,7 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
         
         
         
-        instance = new TextClassifier(dbName, conf);
+        instance = MLBuilder.load(TextClassifier.class, dbName, conf);
         Dataframe validationData = null;
         try {
             validationData = instance.predict(this.getClass().getClassLoader().getResource("datasets/sentimentAnalysis.unlabelled.txt").toURI());
