diff --git a/old_test.java b/new_test.java
index fe90aff..079ae90 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testSizeNotEq() {
-        assertTrue(instance.sizeNotEq("id", 3).getQueryCriterions()
-                .contains(new SizeNotEqCriterion("id", 3)));
+    	assertEquals(new SizeNotEqCriterion("id", 3), instance.sizeNotEq("id", 3).getQueryCriterion());
     }
\ No newline at end of file
