diff --git a/old_test.java b/new_test.java
index b257021..fc82ac9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,18 +1,15 @@
 @Test
   public void testGetErrorCode_multipleErrors() {
-    HttpResponseException httpResponseException =
-        new HttpResponseException.Builder(
-                HttpStatus.SC_BAD_REQUEST, "Bad Request", new HttpHeaders())
-            .setContent(
-                "{\"errors\":["
-                    + "{\"code\":\"MANIFEST_INVALID\",\"message\":\"message 1\",\"detail\":{}},"
-                    + "{\"code\":\"TAG_INVALID\",\"message\":\"message 2\",\"detail\":{}}"
-                    + "]}")
-            .build();
+    Mockito.when(responseException.getContent())
+        .thenReturn(
+            "{\"errors\":["
+                + "{\"code\":\"MANIFEST_INVALID\",\"message\":\"message 1\",\"detail\":{}},"
+                + "{\"code\":\"TAG_INVALID\",\"message\":\"message 2\",\"detail\":{}}"
+                + "]}");
     try {
-      ErrorResponseUtil.getErrorCode(httpResponseException);
+      ErrorResponseUtil.getErrorCode(responseException);
       Assert.fail();
     } catch (HttpResponseException ex) {
-      Assert.assertSame(httpResponseException, ex);
+      Assert.assertSame(responseException, ex);
     }
   }
\ No newline at end of file
