diff --git a/old_product.java b/new_product.java
index b0a141f..0aa8f8b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
-public static ErrorCodes getErrorCode(HttpResponseException httpResponseException)
+public static ErrorCodes getErrorCode(HttpResponseException responseException)
       throws HttpResponseException {
     // Obtain the error response code.
-    String errorContent = httpResponseException.getContent();
+    String errorContent = responseException.getContent();
     if (errorContent == null) {
-      throw httpResponseException;
+      throw responseException;
     }
 
     try {
@@ -25,5 +25,5 @@ public static ErrorCodes getErrorCode(HttpResponseException httpResponseExceptio
     }
 
     // rethrow the original exception
-    throw httpResponseException;
+    throw responseException;
   }
\ No newline at end of file
