diff --git a/old_product.java b/new_product.java
index 79b5f0e..19aa9a3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-public String buildAuthRequestUrl(ServerConfiguration serverConfig, RegisteredClient clientConfig, String redirectUri, String nonce, String state, Map<String, String> options);
\ No newline at end of file
+public String buildAuthRequestUrl(ServerConfiguration serverConfig, RegisteredClient clientConfig, String redirectUri, String nonce, String state, Map<String, String> options, String loginHint);
\ No newline at end of file
