diff --git a/old_test.java b/new_test.java
index 39a235b..d4b6258 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,8 @@
 @Test
   public void execute ()
   {
-    String result = Eclim.execute(new String[]{"ping"});
+    String[] result = StringUtils.split(Eclim.execute(new String[]{"ping"}), '\n');
     assertEquals("Unexpected result",
-        "eclim " + System.getProperty("eclim.version"), result);
+        "eclim   " + System.getProperty("eclim.version"), result[0]);
+    assertTrue("Unexpected result", result[1].startsWith("eclipse 3."));
   }
\ No newline at end of file
