diff --git a/old_product.java b/new_product.java
index 6efbbfe..765c60d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,5 +16,8 @@ public ServiceCall<CounterexampleCollection> listCounterexamples(
     if (listCounterexamplesOptions.cursor() != null) {
       builder.query("cursor", listCounterexamplesOptions.cursor());
     }
+    if (listCounterexamplesOptions.includeAudit() != null) {
+      builder.query("include_audit", String.valueOf(listCounterexamplesOptions.includeAudit()));
+    }
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(CounterexampleCollection.class));
   }
\ No newline at end of file
