diff --git a/old_product.java b/new_product.java
index b79b111..dd47312 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,6 @@
-public static Integer[] sort(final ValueList vl, final StandardFunc sf) throws QueryException {
+public static Integer[] sort(final ValueList vl, final StandardFunc sf, final Collation coll)
+      throws QueryException {
+
     final int al = vl.size();
     final Integer[] order = new Integer[al];
     for(int o = 0; o < al; o++) order[o] = o;
@@ -19,7 +21,7 @@ public static Integer[] sort(final ValueList vl, final StandardFunc sf) throws Q
                       diffError(m, n, sf.info);
               }
               final int d = m == null ? n == null ? 0 : -1 : n == null ? 1 :
-                m.diff(n, sf.sc.collation, sf.info);
+                m.diff(n, coll, sf.info);
               if(d != 0 && d != Item.UNDEF) return d;
             }
             return (int) (s1 - s2);
