diff --git a/old_product.java b/new_product.java
index 2f6d7f9..88f5d51 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,5 @@
-public final int removeAllIn(Iterator<? extends ObjectCursor<? extends KType>> iterator)
+@Override
+    public int removeAllOccurrences(KType key)
     {
-        int count = 0;
-        while (iterator.hasNext())
-        {
-            if (remove((KType) iterator.next().value)) count++;
-        }
-
-        return count;
+        return remove(key) ? 1 : 0;
     }
\ No newline at end of file
