diff --git a/old_test.java b/new_test.java
index a41fe7d..de54b9a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test
-    public void testRemoveAllIn()
+    public void testRemoveAllFromLookupContainer()
     {
-        set.add(newArray(set.keys, 0, 1, 2));
-        
-        ObjectArrayList<Object> list2 = new ObjectArrayList<Object>();
-        list2.add(newArray(list2.buffer, 1, 3));
+        set.add(newArray(set.keys, 0, 1, 2, 3, 4));
 
-        set.removeAllIn(list2);
-        assertEquals(2, set.size());
-        assertSortedListEquals(set.toArray(), 0, 2);
+        ObjectOpenHashSet<Object> list2 = new ObjectOpenHashSet<Object>();
+        list2.add(newArray(list2.keys, 1, 3, 5));
+
+        assertEquals(2, set.removeAll(list2));
+        assertEquals(3, set.size());
+        assertSortedListEquals(set.toArray(), 0, 2, 4);
     }
\ No newline at end of file
