diff --git a/old_product.java b/new_product.java
index f865c15..a7ea9db 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public static LocalTime parse(String time) {
         ISOChronology.checkNotNull(time, "Text to parse must not be null");
-        return PARSER.parse(time).mergeStrict().toLocalTime();
+        return DateTimeFormatters.isoLocalTime().parse(time).mergeStrict().toLocalTime();
     }
\ No newline at end of file
