diff --git a/old_test.java b/new_test.java
index 9465e2d..38440ea 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,10 @@
-@Test(dataProvider="sampleToString")
-    public void test_parse(int h, int m, int s, int n, String offsetId, String text) {
-        OffsetTime t = OffsetTime.parse(text);
+@Test(dataProvider = "sampleToString")
+    public void factory_parse_validText(int h, int m, int s, int n, String offsetId, String parsable) {
+        OffsetTime t = OffsetTime.parse(parsable);
+        assertNotNull(t, parsable);
         assertEquals(t.getHourOfDay(), h);
         assertEquals(t.getMinuteOfHour(), m);
         assertEquals(t.getSecondOfMinute(), s);
         assertEquals(t.getNanoOfSecond(), n);
-        assertEquals(t.getOffset().toString(), offsetId);
+        assertEquals(t.getOffset(), ZoneOffset.zoneOffset(offsetId));
     }
\ No newline at end of file
