diff --git a/old_test.java b/new_test.java
index 50e2e14..a9e30df 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
 	public void findPeopleByRoles_shouldMatchOnPatientIdentifiers() throws Exception {
 		DWRPersonService dwrPersonService = new DWRPersonService();
 		
-		List<PersonListItem> persons = dwrPersonService.findPeopleByRoles("12345K", false, null);
+		List<Object> persons = dwrPersonService.findPeopleByRoles("12345K", false, null);
 		
 		Assert.assertEquals(1, persons.size());
 		Assert.assertEquals(new PersonListItem(6), persons.get(0));
