diff --git a/old_product.java b/new_product.java
index 9959a58..5babecf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public Value interiorRingN(final ANode node, final Int number) throws QueryException {
     final Geometry geo = geo(node, Q_GML_POLYGON);
     if(geo == null && checkGeo(node) != null)
-      throw GeoErrors.polygonNeeded(node.qname().local());
+      throw GeoErrors.geoType(node.qname().local(), "Polygon");
 
     final long n = number.itr();
     final int max = ((Polygon) geo).getNumInteriorRing();
