diff --git a/old_test.java b/new_test.java
index a8af2bc..11d2f38 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,12 +15,12 @@
     runError("geo:interiorRingN(<gml:Polygon><gml:outerBoundaryIs><gml:LinearRing>" +
             "<gml:coordinates>11,11 18,11 18,18 11,18 11,11</gml:coordinates>" +
             "</gml:LinearRing></gml:outerBoundaryIs></gml:Polygon>, 1)",
-            GeoErrors.qname(6));
+            GeoErrors.qname(4));
 
     runError("geo:interiorRingN(<gml:Polygon><gml:outerBoundaryIs><gml:LinearRing>" +
             "<gml:coordinates>11,11 18,11 18,18 11,18 11,11</gml:coordinates>" +
             "</gml:LinearRing></gml:outerBoundaryIs></gml:Polygon>, 0)",
-            GeoErrors.qname(6));
+            GeoErrors.qname(4));
 
     runError("geo:interiorRingN(" +
             "<gml:Point><gml:coordinates>2.0,3.0</gml:coordinates></gml:Point>, 1)",
