diff --git a/old_product.java b/new_product.java
index bd700ec..8b760d3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @SuppressWarnings("unchecked")
     public T[] query(QueryBuilder builder) {
-        logger.debug("called");
+        LOG.debug("called");
         EntityManager em = createEntityManager();
         try {
             String strQuery;
@@ -9,8 +9,8 @@
             strQuery = ((criteria == null) || criteria.isEmpty()) ? String
                     .format("select %s from %s %s", ENTITY_ALIAS, table,
                             ENTITY_ALIAS) : String.format(
-                    "select %s from %s %s where %s", ENTITY_ALIAS, table,
-                    ENTITY_ALIAS, criteria);
+                            "select %s from %s %s where %s", ENTITY_ALIAS, table,
+                            ENTITY_ALIAS, criteria);
 
             // order by update time, if need be
             boolean orderByUpdateTime = builder.isOrderByUpdateTime();
@@ -26,13 +26,13 @@
             }
 
             Query q = em.createQuery(strQuery);
-            logger.debug("Query string: " + strQuery);
+            LOG.debug("Query string: " + strQuery);
 
             // set max results to default, or requested limit (capped by MAX)
             // for paginated results, limit is per page
             Integer limit = builder.getLimit();
             if ((limit == null) || (limit > MAX_PAGE_SIZE)) {
-                limit = Integer.valueOf(MAX_PAGE_SIZE);
+                limit = MAX_PAGE_SIZE;
             }
             boolean paginate = builder.isPaginate();
             if (paginate) {
