diff --git a/old_test.java b/new_test.java
index 7844ba7..2a56291 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,10 +7,11 @@
         initial.setJobName("My test job");
         initial.setJobStatus(JobStatus.FAILED);
         initial.setUpdateTime(System.currentTimeMillis());
+        initial.setUserName("myUserName");
         pm.createEntity(initial);
         ClauseBuilder cb = new ClauseBuilder(ClauseBuilder.AND);
         cb.append("jobID='" + initial.getJobID() + "'");
-        cb.append("status='FAILED'");
+        cb.append("userName='myUserName'");
         QueryBuilder qb = new QueryBuilder().table("Job").clause(
                 cb.toString());
         Object[] results = pm.query(qb);
