diff --git a/old_test.java b/new_test.java
index 111b266..979e778 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,10 @@
 @Test
 	public void testAddEvaluation() {
-		cout("Test: Adding evaluation.");
+		cout("TestCoordEvaluation: Adding evaluation.");
 
+		gotoEvaluations();
 		addEvaluation(sc.evaluation);
+		
 		gotoEvaluations();
-		verifyEvaluationAdded(sc.course.courseId, sc.evaluation.name,
-				"AWAITING", "0 / " + sc.students.size());
+		verifyEvaluationAdded(sc.course.courseId, sc.evaluation.name, "AWAITING", "0 / " + sc.students.size());
 	}
\ No newline at end of file
