diff --git a/old_product.java b/new_product.java
index 0856e3e..90ed5d0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
 public void getApplication(String applicationId,
-            final AsyncCallback<ApplicationDto> callback) {
-        rpcService.getApplication(applicationId,
-                new DataCallback<ApplicationDto>(callback) {
-                    @Override
-                    protected void onResult(ApplicationDto result) {
-                    }
-                });
-    }
\ No newline at end of file
+                             final AsyncCallback<ApplicationDto> callback) {
+    applicationRpcService.getApplication(applicationId,
+        new DataCallback<ApplicationDto>(callback) {
+          @Override
+          protected void onResult(ApplicationDto result) {
+          }
+        });
+  }
\ No newline at end of file
