diff --git a/old_test.java b/new_test.java
index b6ee43a..5336118 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testGetApplication() throws Exception {
         ApplicationDto application = createApplication();
 
-        ApplicationDto storedApplication = client.getApplication(application.getId());
+        ApplicationDto storedApplication = client.getApplicationByApplicationToken(application.getApplicationToken());
 
         Assert.assertNotNull(storedApplication);
         assertApplicationsEquals(application, storedApplication);
