diff --git a/old_test.java b/new_test.java
index de2144a..20c7265 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testCreateJob() throws GenieException {
-        final Job job = new Job.Builder(JOB_1_NAME, JOB_1_USER, JOB_1_VERSION)
+        final Job job = new Job.Builder(JOB_1_NAME, JOB_1_USER, JOB_1_VERSION, JOB_1_COMMAND_ARGS)
             .withId(JOB_1_ID)
             .withStarted(new Date())
             .build();
@@ -16,4 +16,5 @@
         Assert.assertEquals(JOB_1_USER, argument.getValue().getUser());
         Assert.assertEquals(JOB_1_VERSION, argument.getValue().getVersion());
         Assert.assertEquals(JOB_1_ID, argument.getValue().getId());
+        Assert.assertEquals(JOB_1_COMMAND_ARGS, argument.getValue().getCommandArgs());
     }
\ No newline at end of file
