diff --git a/old_product.java b/new_product.java
index 7704384..aeaddfb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public Cohort removePatientFromCohort(Cohort cohort, Patient patient) {
-		if (cohort.contains(patient)) {
-			cohort.getMemberIds().remove(patient.getPatientId());
+		if (cohort.contains(patient.getPatientId())) {
+			cohort.removeMember(patient);
 			Context.getCohortService().saveCohort(cohort);
 		}
 		return cohort;
