diff --git a/old_test.java b/new_test.java
index 4d2dac4..16eef63 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,12 @@
 		// make a patient, add it using the method
 		Patient patientToAddThenRemove = new Patient(4);
 		service.addPatientToCohort(service.getCohort(2), patientToAddThenRemove);
-		assertTrue(service.getCohort(2).contains(patientToAddThenRemove));
+		assertTrue(service.getCohort(2).contains(patientToAddThenRemove.getPatientId()));
 		service.removePatientFromCohort(service.getCohort(2), patientToAddThenRemove);
-		assertFalse(service.getCohort(2).contains(patientToAddThenRemove));
+		List<CohortMembership> memberList = service.getCohort(2)
+				.getMembers().stream()
+				.filter(m -> m.getPatient().getPatientId().equals(patientToAddThenRemove.getPatientId()))
+				.collect(Collectors.toList());
+		CohortMembership memberWithPatientToRemove = memberList.get(0);
+		assertNotNull(memberWithPatientToRemove.getEndDate());
 	}
\ No newline at end of file
