diff --git a/old_product.java b/new_product.java
index f3d8026..2141b69 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ void setRequest(NettyRequest request) {
     if (request != null) {
       if (this.request == null) {
         this.request = request;
-        HttpHeaders.setKeepAlive(responseMetadata, request.isKeepAlive());
+        HttpUtil.setKeepAlive(responseMetadata, request.isKeepAlive());
       } else {
         throw new IllegalStateException(
             "Request has already been set inside NettyResponseChannel for channel {} " + ctx.channel());
