diff --git a/old_test.java b/new_test.java
index c668106..77e9e20 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
   public void setRequestTest() {
     HttpRequest request = createRequestWithHeaders(HttpMethod.GET, TestingUri.SetRequest.toString());
-    HttpHeaders.setKeepAlive(request, false);
+    HttpUtil.setKeepAlive(request, false);
     EmbeddedChannel channel = createEmbeddedChannel();
     channel.writeInbound(request);
 
     HttpResponse response = (HttpResponse) channel.readOutbound();
-    assertEquals("Unexpected response status", HttpResponseStatus.ACCEPTED, response.getStatus());
+    assertEquals("Unexpected response status", HttpResponseStatus.ACCEPTED, response.status());
     assertFalse("Channel not closed on the server", channel.isActive());
   }
\ No newline at end of file
