diff --git a/old_test.java b/new_test.java
index 61567c8..8be4093 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
 				.columns("value1", "value2")
 				.values(System.currentTimeMillis(), 5)
 				.build();
-		this.influxDB.writeUdp(8088, TimeUnit.MILLISECONDS, serie);
+		this.influxDB.writeUdp(8088, serie);
 		// FIXME we need to check the existence of the data written.
 		this.influxDB.deleteDatabase(dbName);
 	}
\ No newline at end of file
