diff --git a/old_test.java b/new_test.java
index 3131b86..46de20a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	public void testlike() {
-		String sqlPart = Condition.instance().like("default", "default", SQLlikeType.DEFAULT).like("left","left", SQLlikeType.LEFT).like("right","right", SQLlikeType.RIGHT).toString();
+		String sqlPart = Condition.instance().like("default", "default", SqlLike.DEFAULT).like("left","left", SqlLike.LEFT).like("right","right", SqlLike.RIGHT).toString();
 		System.out.println("sql ==> " + sqlPart);
 		Assert.assertEquals("WHERE (default LIKE '%default%' AND left LIKE '%left' AND right LIKE 'right%')", sqlPart);
 	}
\ No newline at end of file
