diff --git a/old_product.java b/new_product.java
index e852ec7..a35a843 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,14 @@
 public void onSupplyCandidateNewOrChange(final Candidate supplyCandidate)
 	{
-		final Candidate parentStockCandidateWithId = updateStock(supplyCandidate);
+		// store the supply candidate and get both it's ID and qty-delta
+		final Candidate supplyCandidateDeltaWithId = candidateRepository.addOrReplace(supplyCandidate);
+		
+		// update the stock with the delta
+		final Candidate parentStockCandidateWithId = updateStock(supplyCandidateDeltaWithId);
 
-		candidateRepository.addOrReplace(supplyCandidate.withParentId(parentStockCandidateWithId.getId()));
+		// set the stock candidate as parent for the supply candidate
+		candidateRepository.addOrReplace(
+				supplyCandidate.withParentId(parentStockCandidateWithId.getId()));
 
 		// e.g.
 		// supply-candidate with 23 (i.e. +23)
