diff --git a/old_test.java b/new_test.java
index 89d09e2..34fdb0d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
 	public void fullViewLeft_calibrated() {
 
-		IntrinsicParameters param =
-				new IntrinsicParameters().fsetK(300, 320, 0, 150, 130, width, height).fsetRadial(0.1,1e-4);
+		PinholeRadial param =
+				new PinholeRadial().fsetK(300, 320, 0, 150, 130, width, height).fsetRadial(0.1,1e-4);
 
 		// do nothing rectification
 		DenseMatrix64F rect1 = CommonOps.identity(3);
