diff --git a/old_product.java b/new_product.java
index 700610b..be0e777 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
-public static void fullViewLeft(IntrinsicParameters paramLeft,
+public static void fullViewLeft(PinholeRadial paramLeft,
 									DenseMatrix64F rectifyLeft, DenseMatrix64F rectifyRight,
 									DenseMatrix64F rectifyK)
 	{
 		// need to take in account the order in which image distort will remove rectification later on
-		paramLeft = new IntrinsicParameters(paramLeft);
+		paramLeft = new PinholeRadial(paramLeft);
 
 		PointTransform_F32 tranLeft = transformPixelToRect_F32(paramLeft, rectifyLeft);
 
