diff --git a/old_product.java b/new_product.java
index 5b68c01..dceb286 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,6 +4,11 @@ static boolean sanityCheckConvex( List<Point2D_I32> contour , int indexA , int i
 
 		int maxAllowed = (int)(2*Math.PI*d+0.5);
 
+		if( indexA > indexB ) {
+			int tmp = indexA;
+			indexA = indexB;
+			indexB = tmp;
+		}
 		if( indexB-indexA > maxAllowed )
 			return false;
 		if( indexA + contour.size()-indexB > maxAllowed )
