diff --git a/old_test.java b/new_test.java
index 531423a..30854a2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,8 @@
 @Test
     public void testGetItems() throws Exception
     {
-        ItemIterator items = c.getItems();
+        Iterator<Item> items = itemService.findByCollection(context, collection);
         assertThat("testGetItems 0", items, notNullValue());
         //by default is empty
         assertFalse("testGetItems 1", items.hasNext());
-        assertThat("testGetItems 2", items.next(), nullValue());
-        assertThat("testGetItems 3", items.nextID(), equalTo(-1));
     }
\ No newline at end of file
