diff --git a/old_test.java b/new_test.java
index a709cb6..cabe922 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,6 +7,6 @@
     query(_HASH_HASH.args("", "SHA-1"), "2jmj7l5rSw0yVb/vlWAYkK/YBwk=");
     query(_HASH_HASH.args("", "SHA-256"), "47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=");
     query(_HASH_HASH.args("xs:base64Binary('')", "md5"), "1B2M2Y8AsgTpgAmY7PhCfg==");
-    error(_HASH_HASH.args("xs:hexBinary('')", "md5"), Err.STRB64TYPE);
+    query(_HASH_HASH.args("xs:hexBinary('')", "md5"), "1B2M2Y8AsgTpgAmY7PhCfg==");
     error(_HASH_HASH.args("", ""), Err.HASH_ALG);
   }
\ No newline at end of file
