diff --git a/old_product.java b/new_product.java
index ccdca74..0c5b3c6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,3 @@
 private B64 hash(final String algo, final QueryContext ctx) throws QueryException {
-    final Item it = checkItem(expr[0], ctx);
-    final byte[] val;
-    if(it instanceof AStr) {
-      val = it.string(info);
-    } else if(it instanceof B64) {
-      val = ((B64) it).binary(info);
-    } else {
-      throw STRB64TYPE.thrw(info, it.type);
-    }
-    return hashBinary(val, algo);
+    return hashBinary(checkStrBin(checkItem(expr[0], ctx)), algo);
   }
\ No newline at end of file
