diff --git a/old_test.java b/new_test.java
index 8e0dea6..2d9981c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testBuildClass() {
         try {
-            ClassBuilder builder = new ClassBuilder();
+            ClassBuilder builder = ClassBuilderFactory.getBeanClassBuilderService();
 
             ClassDefinition classDef = new ClassDefinition( "org.drools.TestClass1",
                                                             null,
