diff --git a/old_test.java b/new_test.java
index 9c73288..7baf78e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
     public void testWithYear() {
         ThaiBuddhistDate date = testDate.withYear(ThaiBuddhistEra.BEFORE_BUDDHIST, 2010);
-        assertEquals(date, ThaiBuddhistDate.thaiBuddhistDate(ThaiBuddhistEra.BEFORE_BUDDHIST, 2010, testMonthOfYear, testDayOfMonth));
+        assertEquals(date, ThaiBuddhistDate.of(ThaiBuddhistEra.BEFORE_BUDDHIST, 2010, testMonthOfYear, testDayOfMonth));
     }
\ No newline at end of file
