diff --git a/old_test.java b/new_test.java
index 0eaf6ea..e702e5d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,6 +14,8 @@
     int retryIntervalMs = -1;
     int verifyCount = -1;
     boolean isVerbose = true;
+    NetworkUtils.TunnelConfig tunnelConfig = new NetworkUtils.TunnelConfig(
+        true, tunnelHost, timeout, retryCount, retryIntervalMs, verifyCount);
 
     // Can reach directly, no need to ssh tunnel
     PowerMockito.spy(NetworkUtils.class);
@@ -22,7 +24,7 @@
 
     Pair<InetSocketAddress, Process> ret =
         NetworkUtils.establishSSHTunnelIfNeeded(NetworkUtils.getInetSocketAddress(mockEndpoint),
-            tunnelHost, timeout, retryCount, retryIntervalMs, verifyCount, isVerbose);
+            tunnelConfig, NetworkUtils.TunnelType.PORT_FORWARD);
     Assert.assertEquals(mockHost, ret.first.getHostName());
     Assert.assertEquals(mockPort, ret.first.getPort());
     Assert.assertEquals(mockEndpoint, ret.first.toString());
@@ -40,7 +42,7 @@
     PowerMockito.spy(ShellUtils.class);
     PowerMockito.doReturn(process).when(ShellUtils.class, "establishSSHTunnelProcess",
         Mockito.anyString(), Mockito.anyInt(), Mockito.anyString(),
-        Mockito.anyInt(), Mockito.anyBoolean());
+        Mockito.anyInt());
 
     InetSocketAddress newAddress =
         NetworkUtils.getInetSocketAddress(
@@ -48,18 +50,16 @@
 
     PowerMockito.doReturn(false).when(NetworkUtils.class, "isLocationReachable",
         Mockito.eq(newAddress), Mockito.anyInt(), Mockito.anyInt(), Mockito.anyInt());
-    ret =
-        NetworkUtils.establishSSHTunnelIfNeeded(NetworkUtils.getInetSocketAddress(mockEndpoint),
-            tunnelHost, timeout, retryCount, retryIntervalMs, verifyCount, isVerbose);
+    ret = NetworkUtils.establishSSHTunnelIfNeeded(NetworkUtils.getInetSocketAddress(mockEndpoint),
+            tunnelConfig, NetworkUtils.TunnelType.PORT_FORWARD);
     Assert.assertNull(ret.first);
     Assert.assertNull(ret.second);
 
     // Can not reach directly, but can establish ssh tunnel to reach the destination
     PowerMockito.doReturn(true).when(NetworkUtils.class, "isLocationReachable",
         Mockito.eq(newAddress), Mockito.anyInt(), Mockito.anyInt(), Mockito.anyInt());
-    ret =
-        NetworkUtils.establishSSHTunnelIfNeeded(NetworkUtils.getInetSocketAddress(mockEndpoint),
-            tunnelHost, timeout, retryCount, retryIntervalMs, verifyCount, isVerbose);
+    ret = NetworkUtils.establishSSHTunnelIfNeeded(NetworkUtils.getInetSocketAddress(mockEndpoint),
+        tunnelConfig, NetworkUtils.TunnelType.PORT_FORWARD);
     Assert.assertEquals(NetworkUtils.LOCAL_HOST, ret.first.getHostName());
     Assert.assertEquals(mockFreePort, ret.first.getPort());
     Assert.assertEquals(process, ret.second);
