diff --git a/old_product.java b/new_product.java
index 37fdae2..37664ec 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,9 +4,9 @@
         try {
             final Command commandToPatch = JpaServiceUtils.toCommandDto(commandEntity);
             log.debug("Will patch command {}. Original state: {}", id, commandToPatch);
-            final JsonNode commandNode = MAPPER.readTree(commandToPatch.toString());
+            final JsonNode commandNode = GenieObjectMapper.getMapper().readTree(commandToPatch.toString());
             final JsonNode postPatchNode = patch.apply(commandNode);
-            final Command patchedCommand = MAPPER.treeToValue(postPatchNode, Command.class);
+            final Command patchedCommand = GenieObjectMapper.getMapper().treeToValue(postPatchNode, Command.class);
             log.debug("Finished patching command {}. New state: {}", id, patchedCommand);
             this.updateEntityWithDtoContents(commandEntity, patchedCommand);
         } catch (final JsonPatchException | IOException e) {
