diff --git a/old_test.java b/new_test.java
index 2869789..ba9b571 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,15 +1,15 @@
 @Test
 	public void testAsPactModule() throws Exception {
-		Operator input1 = new Source(PersistenceType.HDFS, "1");
-		Operator input2 = new Source(PersistenceType.HDFS, "2");
-		Operator input3 = new Source(PersistenceType.HDFS, "3");
-		CompositeOperator fixture = new CompositeOperatorImpl(1, input1, input2, input3);
-		EvaluationContext context = new EvaluationContext();
+		final Operator input1 = new Source(PersistenceType.HDFS, "1");
+		final Operator input2 = new Source(PersistenceType.HDFS, "2");
+		final Operator input3 = new Source(PersistenceType.HDFS, "3");
+		final CompositeOperator fixture = new CompositeOperatorImpl(1, input1, input2, input3);
+		final EvaluationContext context = new EvaluationContext();
 
-		PactModule result = fixture.asPactModule(context);
+		final PactModule result = fixture.asPactModule(context);
 
 		assertNotNull(result);
-		List<Level<Contract>> reachableNodes = GraphLevelPartitioner.getLevels(
+		final List<Level<Contract>> reachableNodes = GraphLevelPartitioner.getLevels(
 			result.getAllOutputs(), ContractNavigator.INSTANCE);
 		assertEquals(3, reachableNodes.get(0).getLevelNodes().size());
 		assertEquals(1, reachableNodes.get(1).getLevelNodes().size());
