diff --git a/old_product.java b/new_product.java
index 0c2256b..3d8c45f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,32 +1,26 @@
 public PipelineMdl<K, V> fit(DatasetBuilder datasetBuilder) {
-        assert lbExtractor != null;
-        assert featureExtractor != null;
-
         if (finalStage == null)
             throw new IllegalStateException("The Pipeline should be finished with the Training Stage.");
 
         // Reload for new fit
-        finalFeatureExtractor = featureExtractor;
+        finalPreprocessor = vectorizer;
 
         preprocessingTrainers.forEach(e -> {
 
-            finalFeatureExtractor = e.fit(
+            finalPreprocessor = e.fit(
                 envBuilder,
                 datasetBuilder,
-                finalFeatureExtractor
+                finalPreprocessor
             );
         });
 
-        //TODO: IGNITE-11481
         IgniteModel<Vector, Double> internalMdl = finalStage
             .fit(
                 datasetBuilder,
-                finalFeatureExtractor,
-                lbExtractor
+                finalPreprocessor
             );
 
         return new PipelineMdl<K, V>()
-            .withFeatureExtractor(finalFeatureExtractor)
-            .withLabelExtractor(lbExtractor)
+            .withPreprocessor(finalPreprocessor)
             .withInternalMdl(internalMdl);
     }
\ No newline at end of file
