diff --git a/old_product.java b/new_product.java
index f3c17ba..3a8f0be 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,8 +14,8 @@ public JobResponse createJob(final String name, final String batchId,
         if (config != null )
             contentJson.add("config", config);
 
-        HttpRequestBase request = Request.Post("/v1/jobs")
-                .withContent(contentJson).build();
+        HttpRequestBase request = Request.Post(DocumentConversion.JOBS_PATH)
+                                         .withContent(contentJson).build();
 
         try {
             HttpResponse response = docConversionService.execute(request);
