diff --git a/old_test.java b/new_test.java
index 9aa7377..2674eaa 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testInstance() {
         String val1 = "'''";
         String val2 = "\\";
-        String sqlPart = Condition.instance().between("test_type", val1, val2).toString();
+        String sqlPart = Condition.create().between("test_type", val1, val2).toString();
         System.out.println("sql ==> " + sqlPart);
         Assert.assertEquals("WHERE (test_type BETWEEN ? AND ?)", sqlPart);
     }
\ No newline at end of file
