diff --git a/old_product.java b/new_product.java
index 009c2fa..8f77db7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,6 @@
 public <T> void removeObserver(Observer<T> observer, Class<T> eventType, Annotation... bindings)
    {
       List<EventObserver<?>> observers = registeredObservers.get(eventType);
-      for (Iterator<EventObserver<?>> i = observers.iterator(); i.hasNext();)
-      {
-         if (observer.equals(i.next().getObserver()))
-         {
-            i.remove();
-            break;
-         }
-      }
+      EventObserver<T> eventObserver = new EventObserver<T>(observer, eventType, bindings);
+      observers.remove(eventObserver);
    }
\ No newline at end of file
