diff --git a/old_product.java b/new_product.java
index e75ca0d..85e7819 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-final IdentifiedObject createFromCodes(final IdentifiedObject object) throws FactoryException {
-        final Set/*<String>*/ codes = getCodeCandidates(object);
+final IdentifiedObject createFromCodes(final IdentifiedObject object, boolean specific) throws FactoryException {
+        final Set/*<String>*/ codes = specific ? getSpecificCodeCandidates(object) : getCodeCandidates(object);
         for (final Iterator it=codes.iterator(); it.hasNext();) {
             final String code = (String) it.next();
             IdentifiedObject candidate;
