diff --git a/old_product.java b/new_product.java
index ca2d4de..749e62e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 private IdentityHashMap<String, String> getExtensions(Element root, String configVersion) {
 		
-		IdentityHashMap<String, String> extensions = new IdentityHashMap<String, String>();
+		IdentityHashMap<String, String> extensions = new IdentityHashMap<>();
 		
 		NodeList extensionNodes = root.getElementsByTagName("extension");
 		if (extensionNodes.getLength() > 0) {
