diff --git a/old_test.java b/new_test.java
index 1aaf591..e44df1d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,13 @@
 @Test
 	public void getExtensions_shouldNotExpandExtensionNamesIfExtensionNamesIsNull() throws Exception {
-		ArrayList<Extension> extensions = new ArrayList<Extension>();
+		ArrayList<Extension> extensions = new ArrayList<>();
 
 		Extension mockExtension = new MockExtension();
 		extensions.add(mockExtension);
 
 		mockModule.setExtensions(extensions);
 		mockModule.setExtensionNames(null);
-		ArrayList<Extension> ret = new ArrayList<Extension>(mockModule.getExtensions());
+		ArrayList<Extension> ret = new ArrayList<>(mockModule.getExtensions());
 
 		verifyPrivate(mockModule, never()).invoke("expandExtensionNames");
 	}
\ No newline at end of file
