diff --git a/old_test.java b/new_test.java
index 494215a..ba8f4f1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,10 @@
 @Test
 	public void testHandleRequest() throws Exception {
-		MetricFetcher fetcher = new MetricFetcher(mock(ActorSystem.class), mock(JobManagerRetriever.class), mock(ExecutionContext.class));
+		MetricFetcher fetcher = new MetricFetcher(
+			mock(JobManagerRetriever.class),
+			mock(MetricQueryServiceRetriever.class),
+			Executors.directExecutor(),
+			TestingUtils.TIMEOUT());
 		MetricStoreTest.setupStore(fetcher.getMetricStore());
 
 		JobVertexMetricsHandler handler = new JobVertexMetricsHandler(fetcher);
