diff --git a/old_test.java b/new_test.java
index 368910e..0aad216 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,11 @@
 @Test
-	public void scale() {
+	public void scale_InterpTypeStyle() {
 		ImageFloat32 input = new ImageFloat32(width,height);
 		ImageFloat32 output = new ImageFloat32(width,height);
 		ImageFloat32 output2 = new ImageFloat32(width,height);
 
+		GeneralizedImageOps.randomize(input,rand,0,100);
+
 		InterpolatePixel<ImageFloat32> interp = FactoryInterpolation.bilinearPixel(input);
 
 		// check the two scale function
