diff --git a/old_product.java b/new_product.java
index 6ed4879..c436442 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,7 @@
 public static <T extends ImageBase>
-	void scale( T input , T output , TypeInterpolate type ) {
+	void scale( T input , T output , TypeInterpolate interpType ) {
 		Class<T> inputType = (Class<T>)input.getClass();
+		InterpolatePixel<T> interp = FactoryInterpolation.createPixel(0, 255, interpType, inputType);
 
-		PixelTransform model = DistortSupport.transformScale(output, input);
-		ImageDistort<T> distorter = DistortSupport.createDistort(inputType,model,type);
-
-		distorter.apply(input,output);
+		scale(input,output,interp);
 	}
\ No newline at end of file
