diff --git a/old_product.java b/new_product.java
index 07975e8..28f399d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,9 @@
 @Override
     @Transactional
     public int markZombies() {
-        LOG.debug("called");
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("called");
+        }
         final ProcessStatus zombie = ProcessStatus.ZOMBIE_JOB;
         final long currentTime = new Date().getTime();
 
