diff --git a/old_test.java b/new_test.java
index 12da6ad..fc52803 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,6 @@
 @Test
-    public void testGetProcessorNumber() {
+    public void testGetProcessor() {
         SystemInfo si = new SystemInfo();
         HardwareAbstractionLayer hal = si.getHardware();
-        for (int i = 0; i < hal.getProcessors().length; i++) {
-            assertEquals(i, hal.getProcessors()[i].getProcessorNumber());
-        }
+        assertTrue(hal.getProcessor().getLogicalProcessorCount() > 0);
     }
\ No newline at end of file
