diff --git a/old_product.java b/new_product.java
index 22163bd..ddf0eae 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 CompletableFuture<Result<List<LogData>, BatchProcessorFailure>> readRecords(List<Long> addresses,
-                                                                                AtomicInteger initRetries) {
+                                                                                int retries) {
         return CompletableFuture.supplyAsync(() ->
                 Result.of(() -> addressSpaceView.simpleProtocolRead(addresses, readOptions))
                         .mapError(BatchProcessorError::new))
@@ -9,7 +9,7 @@ CompletableFuture<Result<List<LogData>, BatchProcessorFailure>> readRecords(List
                 .thenCompose(checkedReadResult -> {
                     if (checkedReadResult.isError()) {
                         return retryReadRecords(checkedReadResult.getError().getAddresses(),
-                                initRetries);
+                                retries);
                     } else {
                         return CompletableFuture.completedFuture(checkedReadResult
                                 .mapError(e -> BatchProcessorFailure
