diff --git a/old_test.java b/new_test.java
index 63029a7..7eec388 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
     public void testSpatialRangeQuery() throws Exception {
-    	PointRDD pointRDD = new PointRDD(sc, InputLocation, offset, splitter);
+    	PointRDD spatialRDD = new PointRDD(sc, InputLocation, offset, splitter, true);
     	for(int i=0;i<loopTimes;i++)
     	{
-    		long resultSize = RangeQuery.SpatialRangeQuery(pointRDD, queryEnvelope, 0).getRawPointRDD().count();
+    		long resultSize = RangeQuery.SpatialRangeQuery(spatialRDD, queryEnvelope, 0,false).count();
     		assert resultSize>-1;
     	}
-    	assert RangeQuery.SpatialRangeQuery(pointRDD, queryEnvelope, 0).getRawPointRDD().take(10).get(1).getUserData().toString()!=null;
+    	assert RangeQuery.SpatialRangeQuery(spatialRDD, queryEnvelope, 0,false).take(10).get(1).getUserData().toString()!=null;
         
     }
\ No newline at end of file
