diff --git a/old_product.java b/new_product.java
index 05664fd..1f3cd27 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,20 +1,3 @@
-protected void releaseInAnyFreePartition(ConnectionHandle connectionHandle, ConnectionPartition activePartition) throws InterruptedException  {
-
-		ConnectionPartition workingPartition = activePartition;
-		if (!workingPartition.getFreeConnections().offer(connectionHandle)){
-			// we ran out of space on this partition, pick another free one
-			boolean released = false;
-			for (int i=0; i < this.partitionCount; i++){
-				if (this.partitions[i].getFreeConnections().offer(connectionHandle)){
-					released=true;
-					break;
-				}
-			}
-
-			if (!released)	{
-				// we still didn't find an empty one, wait forever until our partition is free
-				connectionHandle.getOriginatingPartition().getFreeConnections().put(connectionHandle);
-			}
-		}
-
+protected void putConnectionBackInPartition(ConnectionHandle connectionHandle) throws InterruptedException {
+		connectionHandle.getOriginatingPartition().getFreeConnections().put(connectionHandle);
 	}
\ No newline at end of file
