diff --git a/old_product.java b/new_product.java
index 358860b..2c9f515 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 private Item assertEquals(final QueryContext qc) throws QueryException {
-    final Item it = exprs.length < 3 ? null : checkItem(exprs[2], qc);
+    final Item it = exprs.length < 3 ? null : toItem(exprs[2], qc);
     final Iter iter1 = qc.iter(exprs[0]), iter2 = qc.iter(exprs[1]);
     final DeepCompare comp = new DeepCompare(info);
     Item it1, it2;
@@ -12,5 +12,5 @@ private Item assertEquals(final QueryContext qc) throws QueryException {
       if(empty1 || empty2 || !comp.equal(it1.iter(), it2.iter())) break;
       c++;
     }
-    throw new UnitException(info, UNIT_ASSERT_EQUALS, it1, it2, c).value(it);
+    throw new UnitException(info, UNIT_ASSERT_EQUALS_X_X_X, it1, it2, c).value(it);
   }
\ No newline at end of file
