diff --git a/old_test.java b/new_test.java
index b933b48..982f4e0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
   public void assertEquals() {
     query(_UNIT_ASSERT_EQUALS.args("1", "1"), "");
-    error(_UNIT_ASSERT_EQUALS.args("1", "2"), Err.UNIT_ASSERT_EQUALS);
+    error(_UNIT_ASSERT_EQUALS.args("1", "2"), Err.UNIT_ASSERT_EQUALS_X_X_X);
   }
\ No newline at end of file
