diff --git a/old_test.java b/new_test.java
index d9d59eb..cd418e0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -25,7 +25,7 @@
        cache.put("1234", session);
        assertTrue(cache.contains("1234"));
 
-       cache.renewSessionId("1234", "5678");
+       cache.renewSessionId("1234", "5678", "1234.foo", "5678.foo");
        
        assertTrue(cache.contains("5678"));
        assertFalse(cache.contains("1234"));
