diff --git a/old_test.java b/new_test.java
index 8f1bede..ae06635 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,5 @@
         QueryCriterion criterion2 = new GtCriterion("id", 5);
         QueryCriterion criterion3 = new AndCriterion(criterion1, criterion2);
         
-        assertTrue(instance.and(criterion1, criterion2).getQueryCriterions()
-                .contains(criterion3));
+        assertEquals(criterion3, instance.and(criterion1, criterion2).getQueryCriterion());
     }
\ No newline at end of file
