diff --git a/old_product.java b/new_product.java
index ca3d263..991a511 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-protected void internalReleaseConnection(ConnectionHandle connectionHandle) throws InterruptedException, SQLException {
+protected void internalReleaseConnection(ConnectionHandle connectionHandle) throws SQLException {
 		connectionHandle.clearStatementCaches(false);
 
 		if (connectionHandle.getReplayLog() != null){
