diff --git a/old_test.java b/new_test.java
index c48ce0c..6674382 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,6 @@
     public void getLogSchemaVersionsByApplicationIdTest() throws Exception {
         LogSchemaDto logSchemaDto = createLogSchema();
         SchemaVersions schemaVersions = client.getSchemaVersionsByApplicationId(logSchemaDto.getApplicationId());
-        List<SchemaDto> found = schemaVersions.getLogSchemaVersions();
+        List<VersionDto> found = schemaVersions.getLogSchemaVersions();
         Assert.assertEquals(2, found.size());
     }
\ No newline at end of file
