diff --git a/old_product.java b/new_product.java
index 5a0e9aa..160c781 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 @Override
-    public List<SchemaDto> getLogSchemaVersionsByApplicationId(String applicationId) throws ControlServiceException {
+    public List<VersionDto> getLogSchemaVersionsByApplicationId(String applicationId) throws ControlServiceException {
         return logSchemaService.findLogSchemaVersionsByApplicationId(applicationId);
     }
\ No newline at end of file
