diff --git a/old_test.java b/new_test.java
index 6aefaf7..4f38644 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,5 +3,5 @@
     AuthenticatedRegistrarAccessor registrarAccessor =
         new AuthenticatedRegistrarAccessor(UNAUTHORIZED_USER, ADMIN_CLIENT_ID);
 
-    assertThat(registrarAccessor.getAllClientIdWithAccess()).isEmpty();
+    assertThat(registrarAccessor.getAllClientIdWithRoles()).isEmpty();
   }
\ No newline at end of file
