diff --git a/old_product.java b/new_product.java
index 3483243..24b1fb2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,10 +10,6 @@ public void validate(Object obj, Errors errors) throws APIException, DuplicateCo
 			errors.reject("Concept.name.atLeastOneRequired");
 			return;
 		}
-		if (conceptToValidate.getDescriptions().size() == 0) {
-			errors.rejectValue("descriptions","Concept.description.atLeastOneRequired");
-			return;
-		}
 
 		ValidationUtils.rejectIfEmpty(errors, "datatype", "Concept.datatype.empty");
 		ValidationUtils.rejectIfEmpty(errors, "conceptClass", "Concept.conceptClass.empty");
