diff --git a/old_test.java b/new_test.java
index 21e3f7b..78074da 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,8 @@
     public void testInitialize() {
         FileNameAnalyzer instance = new FileNameAnalyzer();
         try {
-            instance.initialize();
+            instance.initializeSettings(getSettings());
+            instance.initialize(null);
         } catch (InitializationException ex) {
             fail(ex.getMessage());
         }
