diff --git a/old_product.java b/new_product.java
index 8cbcf21..13f9980 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,15 +1,16 @@
 @Override
-    public final void initialize() throws InitializationException {
+    public final void initialize(Engine engine) throws InitializationException {
         final String key = getAnalyzerEnabledSettingKey();
         try {
-            this.setEnabled(Settings.getBoolean(key, true));
+            this.setEnabled(settings.getBoolean(key, true));
         } catch (InvalidSettingException ex) {
-            LOGGER.warn("Invalid setting for property '{}'", key);
-            LOGGER.debug("", ex);
+            String msg = String.format("Invalid setting for property '{}'", key);
+            LOGGER.warn(msg);
+            LOGGER.debug(msg, ex);
         }
 
         if (isEnabled()) {
-            initializeAnalyzer();
+            initializeAnalyzer(engine);
         } else {
             LOGGER.debug("{} has been disabled", getName());
         }
