diff --git a/old_test.java b/new_test.java
index 287c18c..073f7df 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,10 @@
 @Test
-	public void toString_shouldNotFailIfUuidIsNull() throws Exception {
+	public void toString_shouldIncludeHashCodeIfUuidIsNull() throws Exception {
 		//given
 		BaseOpenmrsObject o = new BaseOpenmrsObjectMock();
-		
-		//when
 		o.setUuid(null);
 		
+		//when
 		//then
-		o.toString();
+		assertEquals("BaseOpenmrsObjectMock{hashCode=" + Integer.toHexString(o.hashCode()) + "}", o.toString());
 	}
\ No newline at end of file
