diff --git a/old_product.java b/new_product.java
index 428c7a6..23b3989 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,5 @@
 @Override
 	public String toString() {
-		if (getUuid() != null) {
-			return getClass().getName() + "[" + getUuid() + "]";
-		} else {
-			return super.toString();
-		}
+		return Objects.toStringHelper(this).add("hashCode", Integer.toHexString(hashCode())).add("uuid", getUuid())
+		        .omitNullValues().toString();
 	}
\ No newline at end of file
