diff --git a/old_product.java b/new_product.java
index e9fc4b3..25754e4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,8 @@
 void scoreDirection(int dx , int dy ) {
+		// TODO fully support disparityMin here
 		if( dx > 0 ) {
 			for (int y = 0; y < lengthY; y++) {
-				scorePath(0,y,dx,dy);
+				scorePath(disparityMin,y,dx,dy);
 			}
 		} else if( dx < 0 ) {
 			for (int y = 0; y < lengthY; y++) {
