diff --git a/old_test.java b/new_test.java
index 9a96ad4..2953ae7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,16 +6,16 @@
     Date start = new Date();
 
     CreateDialogNodeOptions createOptions =
-        new CreateDialogNodeOptions.Builder(workspaceId, dialogNodeName).description(dialogNodeDescription).build();
+            new CreateDialogNodeOptions.Builder(workspaceId, dialogNodeName).description(dialogNodeDescription).build();
     service.createDialogNode(createOptions).execute();
 
     try {
       GetDialogNodeOptions getOptions =
-          new GetDialogNodeOptions.Builder().workspaceId(workspaceId).dialogNode(dialogNodeName).build();
+              new GetDialogNodeOptions.Builder().workspaceId(workspaceId).dialogNode(dialogNodeName).build();
       DialogNode response = service.getDialogNode(getOptions).execute();
       assertNotNull(response);
-      assertNotNull(response.getDialogNode());
-      assertEquals(response.getDialogNode(), dialogNodeName);
+      assertNotNull(response.getDialogNodeId());
+      assertEquals(response.getDialogNodeId(), dialogNodeName);
       assertNotNull(response.getDescription());
       assertEquals(response.getDescription(), dialogNodeDescription);
       assertNotNull(response.getCreated());
