diff --git a/old_test.java b/new_test.java
index 1f68e64..2731658 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,9 +4,9 @@
         TreeFilter filter2 = features -> features[1] > 2;
         TreeFilter filterAnd = filter1.and(features -> features[1] > 2);
 
-        TreeDataIndex filtered1 = index.filter(filter1);
+        TreeDataIndex filtered1 = idx.filter(filter1);
         TreeDataIndex filtered2 = filtered1.filter(filter2);
-        TreeDataIndex filtered3 = index.filter(filterAnd);
+        TreeDataIndex filtered3 = idx.filter(filterAnd);
 
         assertEquals(2, filtered1.rowsCount());
         assertEquals(4, filtered1.columnsCount());
