diff --git a/old_product.java b/new_product.java
index 81a2faa..fd825f6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -21,5 +21,5 @@ public DecisionTreeData filter(TreeFilter filter) {
             }
         }
 
-        return new DecisionTreeData(newFeatures, newLabels, buildIndex);
+        return new DecisionTreeData(newFeatures, newLabels, buildIdx);
     }
\ No newline at end of file
