diff --git a/old_test.java b/new_test.java
index f061e33..1d907b9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test(expected = AmbariException.class)
   public void testCreateUser_Duplicate() throws Exception {
     UserEntity existing = new UserEntity();
-    existing.setUserName(SERVICEOP_USER_NAME);
+    existing.setUserName(UserName.fromString(SERVICEOP_USER_NAME));
     existing.setUserType(UserType.LDAP);
     existing.setUserId(1);
     existing.setMemberEntities(Collections.<MemberEntity>emptySet());
