diff --git a/old_product.java b/new_product.java
index 7e00f1d..75a3a92 100644
--- a/old_product.java
+++ b/new_product.java
@@ -24,7 +24,7 @@ public synchronized void createUser(String userName, String password, UserType u
     principalDAO.create(principalEntity);
 
     UserEntity userEntity = new UserEntity();
-    userEntity.setUserName(userName);
+    userEntity.setUserName(UserName.fromString(userName));
     if (userType == UserType.LOCAL) {
       //passwords should be stored for local users only
       userEntity.setUserPassword(passwordEncoder.encode(password));
