diff --git a/old_test.java b/new_test.java
index aab6f7f..8ba1b4c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,7 @@
 @Test
 	public void testGetById() throws SolrServerException, IOException {
-		ArgumentCaptor<QueryRequest> captor = ArgumentCaptor.forClass(QueryRequest.class);
+
+		ArgumentCaptor<SolrRequest> captor = ArgumentCaptor.forClass(SolrRequest.class);
 		QueryResponse responseMock = Mockito.mock(QueryResponse.class);
 		SolrDocumentList resultList = new SolrDocumentList();
 		Mockito.when(responseMock.getResults()).thenReturn(resultList);
@@ -11,5 +12,5 @@
 		Mockito.verify(solrServerMock, Mockito.times(1)).request(captor.capture());
 		Assert.assertNull(result);
 		Assert.assertEquals("myId", captor.getValue().getParams().get("ids"));
-		Assert.assertEquals("/get", captor.getValue().getParams().get(CommonParams.QT));
+		Assert.assertEquals("/get", captor.getValue().getPath());
 	}
\ No newline at end of file
