diff --git a/old_product.java b/new_product.java
index 5688384..bf8064a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ void reportOrphanedSpans() {
     while ((reference = poll()) != null) {
       TraceContext context = reference.get();
       MutableSpan value = delegate.remove(reference);
-      if (value == null) continue;
+      if (value == null || noop.get()) continue;
       try {
         value.annotate(clock.currentTimeMicroseconds(), "brave.flush");
         reporter.report(value.toSpan());
