diff --git a/old_test.java b/new_test.java
index 45680c2..d033a61 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     MutableSpanMap map = new MutableSpanMap(localEndpoint, () -> 0, span ->
     {
       throw new RuntimeException("die!");
-    });
+    }, new AtomicBoolean(true));
 
     // We drop the reference to the context, which means the next GC should attempt to flush it
     map.getOrCreate(context.toBuilder().build());
