diff --git a/old_test.java b/new_test.java
index 73680c2..aa8b34e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,7 +20,7 @@
                                 .Builder(getVaultUri(), certificateName + i, certificateContent)
                                 .withPassword(certificatePassword)
                                 .withPolicy(certificatePolicy)
-                                .build()).getBody();
+                                .build());
                     CertificateIdentifier id = certificateBundle.certificateIdentifier();
                     certificates.add(id.baseIdentifier());
                     break;
@@ -36,7 +36,7 @@
             }
         }
 
-        PagedList<CertificateItem> listResult = keyVaultClient.listCertificates(getVaultUri(), PAGELIST_MAX_CERTS).getBody();
+        PagedList<CertificateItem> listResult = keyVaultClient.listCertificates(getVaultUri(), PAGELIST_MAX_CERTS);
         Assert.assertTrue(PAGELIST_MAX_CERTS >= listResult.currentPage().getItems().size());
 
         HashSet<String> toDelete = new HashSet<String>();
