diff --git a/old_product.java b/new_product.java
index c9beb89..9bd0ba0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,17 +8,17 @@ public void recalculate() {
             }
         }
         if (dynamicMgmt != null) {
-            Map<String, OperationsServerLoadHistory> opsServerHistory = new HashMap<String, OperationsServerLoadHistory>();
-            for (String dnsName : opsServersMap.keySet()) {
-                opsServerHistory.put(dnsName, opsServersMap.get(dnsName).history);
+            Map<Integer, OperationsServerLoadHistory> opsServerHistory = new HashMap<Integer, OperationsServerLoadHistory>();
+            for (Integer accessPointId : opsServersMap.keySet()) {
+                opsServerHistory.put(accessPointId, opsServersMap.get(accessPointId).history);
             }
-            Map<String, RedirectionRule> rules = dynamicMgmt.recalculate(opsServerHistory);
+            Map<Integer, RedirectionRule> rules = dynamicMgmt.recalculate(opsServerHistory);
             LOG.trace("DynamicLoadManager recalculate() got {} redirection rules", rules.size());
-            for (String dnsName : rules.keySet()) {
-                if (opsServersMap.containsKey(dnsName)) {
-                    sendRedirectionRule(dnsName, opsServersMap.get(dnsName).nodeInfo, rules.get(dnsName));
+            for (Integer accessPointId : rules.keySet()) {
+                if (opsServersMap.containsKey(accessPointId)) {
+                    sendRedirectionRule(accessPointId, opsServersMap.get(accessPointId).nodeInfo, rules.get(accessPointId));
                 } else {
-                    LOG.error("Operations server {} redirection rule exist, but NO server available, skip setting rule.", dnsName);
+                    LOG.error("Operations server {} redirection rule exist, but NO server available, skip setting rule.", accessPointId);
                 }
             }
         }
