diff --git a/old_test.java b/new_test.java
index b2fbfc1..08f521c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,30 +3,30 @@
         DefaultRebalancer rebalancer = new DefaultRebalancer();
         assertNotNull(rebalancer);
 
-        String server1 = "dns1";
+        Integer server1 = "dns1".hashCode();
         OperationsServerLoadHistory server1History = new OperationsServerLoadHistory(MAX_HISTORY_TIME_LIVE);
         server1History.addOpsServerLoad(new LoadInfo(10));
 
-        String server2 = "dns2";
+        Integer server2 = "dns2".hashCode();
         OperationsServerLoadHistory server2History = new OperationsServerLoadHistory(MAX_HISTORY_TIME_LIVE);
         server2History.addOpsServerLoad(new LoadInfo(30));
 
-        String server3 = "dns3";
+        Integer server3 = "dns3".hashCode();
         OperationsServerLoadHistory server3History = new OperationsServerLoadHistory(MAX_HISTORY_TIME_LIVE);
         server3History.addOpsServerLoad(new LoadInfo(55));
 
-        Map<String, OperationsServerLoadHistory> serversHistory = new HashMap<>();
+        Map<Integer, OperationsServerLoadHistory> serversHistory = new HashMap<>();
 
         serversHistory.put(server1, server1History);
         serversHistory.put(server2, server2History);
         serversHistory.put(server3, server3History);
-        Map<String,RedirectionRule> rules = rebalancer.recalculate(serversHistory );
+        Map<Integer, RedirectionRule> rules = rebalancer.recalculate(serversHistory);
         assertNotNull(rules);
 
         assertEquals(1, rules.size());
 
         assertNotNull(rules.get(server3));
 
-        assertEquals(server1, rules.get(server3).getDnsName());
+        assertEquals(server1.intValue(), rules.get(server3).getAccessPointId());
         assertEquals(new Double(0.2), new Double(rules.get(server3).getRedirectionProbability()));
     }
\ No newline at end of file
