diff --git a/old_test.java b/new_test.java
index 7705778..d728ad1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testNotNull() {
-        assertTrue(instance.notNull("name").getQueryCriterions()
-                .contains(new NotNullCriterion("name")));
+    	assertEquals(new NotNullCriterion("name"), instance.notNull("name").getQueryCriterion());
     }
\ No newline at end of file
