diff --git a/old_product.java b/new_product.java
index e5973b7..e562a4a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -28,9 +28,9 @@ public static String getIntfNameFromPciAddress(Port port) {
         String vendorInfoForPort = String.valueOf(port.getProfile().get(PCI_VENDOR_INFO));
 
         if (!portNamePrefixMap().containsKey(vendorInfoForPort)) {
-            log.error("Failed to retrieve the interface name because of no port name prefix for vendor ID {}",
+            log.warn("Failed to retrieve the interface name because of unsupported prefix for vendor ID {}",
                     vendorInfoForPort);
-            return null;
+            return UNSUPPORTED_VENDOR;
         }
         String portNamePrefix = portNamePrefixMap().get(vendorInfoForPort);
 
