diff --git a/old_test.java b/new_test.java
index 5519823..da4e2d9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,5 +7,5 @@
         assertNull(getIntfNameFromPciAddress(openstackPort));
         assertEquals(expectedIntfName1, getIntfNameFromPciAddress(openstackSriovPort1));
         assertEquals(expectedIntfName2, getIntfNameFromPciAddress(openstackSriovPort2));
-        assertNull(getIntfNameFromPciAddress(openstackSriovPort3));
+        assertEquals(UNSUPPORTED_VENDOR, getIntfNameFromPciAddress(openstackSriovPort3));
     }
\ No newline at end of file
