diff --git a/old_test.java b/new_test.java
index 0e63552..183fe2f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testIsInstance2(){
-        assertEquals(true, ClassUtil.isInstance(new User(), new Class<?>[] { Comparable.class, CharSequence.class }));
-        assertEquals(false, ClassUtil.isInstance(new User(), new Class<?>[] { Integer.class, CharSequence.class }));
-        assertEquals(true, ClassUtil.isInstance("1234", new Class<?>[] { Comparable.class, CharSequence.class }));
+        assertEquals(true, ClassUtil.isInstanceAnyClass(new User(), new Class<?>[] { Comparable.class, CharSequence.class }));
+        assertEquals(false, ClassUtil.isInstanceAnyClass(new User(), new Class<?>[] { Integer.class, CharSequence.class }));
+        assertEquals(true, ClassUtil.isInstanceAnyClass("1234", new Class<?>[] { Comparable.class, CharSequence.class }));
     }
\ No newline at end of file
