diff --git a/old_product.java b/new_product.java
index 29a97e4..cd054b7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,5 +5,5 @@
         @RequestBody final Command updateCommand
     ) throws GenieException {
         log.debug("Called to update command {}", updateCommand);
-        this.commandService.updateCommand(id, updateCommand);
+        this.commandService.updateCommand(id, DtoAdapters.toV4Command(updateCommand));
     }
\ No newline at end of file
