diff --git a/old_product.java b/new_product.java
index ebe88a0..a38a467 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,8 +10,11 @@ public void validate(Object obj, Errors errors) throws APIException, DuplicateCo
 			errors.reject("Concept.name.atLeastOneRequired");
 			return;
 		}
-		
-		boolean hasFullySpecifiedName = false;
+
+		ValidationUtils.rejectIfEmpty(errors, "datatype", "Concept.datatype.empty");
+		ValidationUtils.rejectIfEmpty(errors, "conceptClass", "Concept.conceptClass.empty");
+
+        boolean hasFullySpecifiedName = false;
 		for (Locale conceptNameLocale : conceptToValidate.getAllConceptNameLocales()) {
 			boolean fullySpecifiedNameForLocaleFound = false;
 			boolean preferredNameForLocaleFound = false;
