diff --git a/old_product.java b/new_product.java
index fc17aa7..3e99006 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,6 +5,6 @@ public Mono<CosmosDatabaseResponse> createDatabase(CosmosDatabaseSettings databa
         }
         Database wrappedDatabase = new Database();
         wrappedDatabase.id(databaseSettings.id());
-        return RxJava2Adapter.singleToMono(RxJavaInterop.toV2Single(asyncDocumentClient.createDatabase(wrappedDatabase, options.toRequestOptions()).map(databaseResourceResponse ->
-                new CosmosDatabaseResponse(databaseResourceResponse, this)).toSingle()));
+        return asyncDocumentClient.createDatabase(wrappedDatabase, options.toRequestOptions()).map(databaseResourceResponse ->
+                new CosmosDatabaseResponse(databaseResourceResponse, this)).single();
     }
\ No newline at end of file
