diff --git a/old_test.java b/new_test.java
index eb4f273..8637a7f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups = "samples", timeOut = TIMEOUT)
     public void createDatabase_Async() throws Exception {
-        Observable<ResourceResponse<Database>> createDatabaseObservable = client.createDatabase(getDatabaseDefinition(),
+        Flux<ResourceResponse<Database>> createDatabaseObservable = client.createDatabase(getDatabaseDefinition(),
                                                                                                      null);
 
         final CountDownLatch completionLatch = new CountDownLatch(1);
