diff --git a/old_test.java b/new_test.java
index 0c29bed..11738ef 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testIsTrue() {
-        assertTrue(instance.isTrue("name").getQueryCriterions()
-                .contains(new EqCriterion("name", true)));
+    	assertEquals(new EqCriterion("name", true), instance.isTrue("name").getQueryCriterion());
     }
\ No newline at end of file
