diff --git a/old_product.java b/new_product.java
index 2694d37..15f0aa9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,4 @@
-public Registrar getRegistrarForUser(String clientId, AuthResult authResult) {
-    return getAndAuthorize(Registrar::loadByClientId, clientId, authResult);
+public Registrar getRegistrarForUser(
+      String clientId, AccessType accessType, AuthResult authResult) {
+    return getAndAuthorize(Registrar::loadByClientId, clientId, accessType, authResult);
   }
\ No newline at end of file
