diff --git a/old_product.java b/new_product.java
index 53363ea..fa402bf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,5 +15,8 @@ public ServiceCall<ExampleCollection> listExamples(ListExamplesOptions listExamp
     if (listExamplesOptions.cursor() != null) {
       builder.query("cursor", listExamplesOptions.cursor());
     }
+    if (listExamplesOptions.includeAudit() != null) {
+      builder.query("include_audit", String.valueOf(listExamplesOptions.includeAudit()));
+    }
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(ExampleCollection.class));
   }
\ No newline at end of file
