diff --git a/old_test.java b/new_test.java
index 26cb320..6ac81c6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,11 +2,10 @@
 	@Test
 	public void testProcessImportedGrades() {
 		final List<Assignment> assignments = mockAssignments();
-		final List<GbStudentGradeInfo> grades = mockStudentGrades();
-		final ImportedGradeWrapper importedGradeWrapper = mockImportedGrades();
+		final List<GbStudentGradeInfo> existingGrades = mockExistingStudentGrades();
+		final ImportedSpreadsheetWrapper importedSpreadsheetWrapper = mockImportedSpreadsheetData();
 
-		final List<ProcessedGradeItem> processedGradeItems = ImportGradesHelper.processImportedGrades(importedGradeWrapper, assignments,
-				grades);
+		final List<ProcessedGradeItem> processedGradeItems = ImportGradesHelper.processImportedGrades(importedSpreadsheetWrapper, assignments, existingGrades);
 
 		Assert.assertNotNull(processedGradeItems);
 
