diff --git a/old_test.java b/new_test.java
index b9d8ed0..f1775a1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -19,7 +19,7 @@
 		int minCostPrev = 6;
 
 		// Compute the cost using this algorithm
-		alg.computeCostInnerD(costXD,idxCost,idxWork,minCostPrev);
+		alg.computeCostInnerD(costXD,idxCost,idxWork,minCostPrev, rangeD);
 
 		// Now compare it to a brute force solution
 		for (int d = 1; d < rangeD-1; d++) {
