diff --git a/old_test.java b/new_test.java
index d9bd7bd..6dfc6b3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void testGetPublicKeysetHandle() throws Exception {
-    KeysetHandle privateHandle = CleartextKeysetHandle.generateNew(
+    KeysetHandle privateHandle = KeysetHandle.generateNew(
         SignatureKeyTemplates.ECDSA_P256);
     KeyData privateKeyData = privateHandle.getKeyset().getKey(0).getKeyData();
     EcdsaPrivateKey privateKey = EcdsaPrivateKey.parseFrom(privateKeyData.getValue());
