diff --git a/old_product.java b/new_product.java
index f2bff62..5f3cbe3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,6 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 		if (order.getDateActivated() == null) {
 			order.setDateActivated(new Date());
 		}
-		//Reject if there is an active order for the same orderable with overlapping schedule
 		boolean isDrugOrder = DrugOrder.class.isAssignableFrom(getActualType(order));
 		Concept concept = order.getConcept();
 		if (concept == null && isDrugOrder) {
@@ -116,9 +115,8 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 		if (DISCONTINUE != order.getAction()) {
 			List<Order> activeOrders = getActiveOrders(order.getPatient(), null, order.getCareSetting(), null);
 			for (Order activeOrder : activeOrders) {
-				if (order.hasSameOrderableAs(activeOrder)
-				        && !OpenmrsUtil.nullSafeEquals(order.getPreviousOrder(), activeOrder)
-				        && OrderUtil.checkScheduleOverlap(order, activeOrder)) {
+				//Reject if there is an active drug order for the same orderable with overlapping schedule
+				if (areDrugOrdersOfSameOrderableAndOverlappingSchedule(order, activeOrder)) {
 					throw new APIException("Order.cannot.have.more.than.one", (Object[]) null);
 				}
 			}
