diff --git a/old_test.java b/new_test.java
index 9a16932..cdd0bc6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-	public void saveOrder_shouldFailIfAnActiveOrderForTheSameConceptAndCareSettingExists() throws Exception {
+	public void saveOrder_shouldPassIfAnActiveTestOrderForTheSameConceptAndCareSettingExists() throws Exception {
 		final Patient patient = patientService.getPatient(2);
 		final Concept cd4Count = conceptService.getConcept(5497);
 		//sanity check that we have an active order for the same concept
@@ -14,8 +14,8 @@
 		order.setEncounter(encounterService.getEncounter(6));
 		order.setOrderer(providerService.getProvider(1));
 		order.setCareSetting(duplicateOrder.getCareSetting());
-		
-		expectedException.expect(APIException.class);
-		expectedException.expectMessage("Order.cannot.have.more.than.one");
-		orderService.saveOrder(order, null);
+
+		Order savedOrder = orderService.saveOrder(order, null);
+
+		assertNotNull(orderService.getOrder(savedOrder.getOrderId()));
 	}
\ No newline at end of file
