diff --git a/old_product.java b/new_product.java
index 3961b06..d61b307 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 protected Connection obtainInternalConnection(ConnectionHandle connectionHandle) throws SQLException {
 		boolean tryAgain = false;
 		Connection result = null;
+		Connection oldRawConnection = connectionHandle.getInternalConnection();
 		String url = this.getConfig().getJdbcUrl();
 		
 		int acquireRetryAttempts = this.getConfig().getAcquireRetryAttempts();
@@ -23,6 +24,8 @@ protected Connection obtainInternalConnection(ConnectionHandle connectionHandle)
 				
 				this.getDbIsDown().set(false);
 				
+				connectionHandle.setInternalConnection(result);
+				
 				// call the hook, if available.
 				if (connectionHook != null){
 					connectionHook.onAcquire(connectionHandle);
@@ -47,6 +50,13 @@ protected Connection obtainInternalConnection(ConnectionHandle connectionHandle)
 					}
 				}
 				if (!tryAgain){
+					if (oldRawConnection != null) {
+						oldRawConnection.close();
+					}
+					if (result != null) {
+						result.close();
+					}
+					connectionHandle.setInternalConnection(oldRawConnection);
 					throw e;
 				}
 			}
