diff --git a/old_test.java b/new_test.java
index ada5524..66a80b2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,6 @@
         int r = 4;
         double p = 0.5;
         double expResult = 0.12705078125;
-        double result = DiscreteDistributions.NegativeBinomialCdf(n, r, p);
+        double result = DiscreteDistributions.negativeBinomialCdf(n, r, p);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
