diff --git a/old_product.java b/new_product.java
index 7c8fb4b..6923b4d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static double NegativeBinomialCdf(int n, int r, double p) {
+public static double negativeBinomialCdf(int n, int r, double p) {
         if(n<0 || r<0 || p<0) {
             throw new IllegalArgumentException("All the parameters must be positive.");
         }
@@ -6,7 +6,7 @@ public static double NegativeBinomialCdf(int n, int r, double p) {
         
         double probabilitySum = 0.0;
         for(int i=0;i<=r;++i) {
-            probabilitySum += NegativeBinomial(n, i, p);
+            probabilitySum += negativeBinomial(n, i, p);
         }
         
         return probabilitySum;
