diff --git a/old_product.java b/new_product.java
index 0a16c7c..977c3c1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery sizeGt(String propName, int size) {
-        criterion = criterion.and(criterionBuilder.sizeGt(propName, size));
+        criterion = criterion.and(Criteria.sizeGt(propName, size));
         return this;
     }
\ No newline at end of file
