diff --git a/old_product.java b/new_product.java
index 8bc5464..ff0051c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
-ResponseStatsPerAppBean getResponseStatsPerApp(String organizationId, String apiId, String version,
+ResponseStatsPerClientBean getResponseStatsPerClient(String organizationId, String apiId, String version,
             DateTime from, DateTime to);
\ No newline at end of file
