diff --git a/old_test.java b/new_test.java
index fcdf799..306c15e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,11 +5,11 @@
     VerifiableProperties verifiableProperties = new VerifiableProperties(properties);
     RestRequestHandler restRequestHandler = new MockRestRequestResponseHandler();
     PublicAccessLogger publicAccessLogger = new PublicAccessLogger(new String[]{}, new String[]{});
-    VIPHealthCheckService vipHealthCheckService = new VIPHealthCheckService("/ambry-frontend/admin");
+    RestServerState restServerState = new RestServerState("/healthCheck");
 
     NettyServerFactory nettyServerFactory =
         new NettyServerFactory(verifiableProperties, new MetricRegistry(), restRequestHandler, publicAccessLogger,
-            vipHealthCheckService);
+            restServerState);
     NioServer nioServer = nettyServerFactory.getNioServer();
     assertNotNull("No NioServer returned", nioServer);
     assertEquals("Did not receive a NettyServer instance", NettyServer.class.getCanonicalName(),
