diff --git a/old_product.java b/new_product.java
index a3aa971..9bf8d26 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,6 +9,6 @@ private NettyServer getNettyServer(Properties properties)
     NettyMetrics nettyMetrics = new NettyMetrics(new MetricRegistry());
     RestRequestHandler requestHandler = new MockRestRequestResponseHandler();
     PublicAccessLogger publicAccessLogger = new PublicAccessLogger(new String[]{}, new String[]{});
-    VIPHealthCheckService vipHealthCheckService = new VIPHealthCheckService("/ambry-frontend/admin");
-    return new NettyServer(nettyConfig, nettyMetrics, requestHandler, publicAccessLogger, vipHealthCheckService);
+    RestServerState restServerState = new RestServerState("/healthCheck");
+    return new NettyServer(nettyConfig, nettyMetrics, requestHandler, publicAccessLogger, restServerState);
   }
\ No newline at end of file
