diff --git a/old_product.java b/new_product.java
index 2ee0c87..fb718b6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,12 @@
-public List<EventClassDto> getEventClassesByFamilyIdVersionAndType(String eventClassFamilyId, int version, EventClassType type)
-            throws Exception {
-        ParameterizedTypeReference<List<EventClassDto>> typeRef = new ParameterizedTypeReference<List<EventClassDto>>() {
-        };
-        ResponseEntity<List<EventClassDto>> entity = restTemplate.exchange(restTemplate.getUrl()
-                        + "eventClasses?eventClassFamilyId={eventClassFamilyId}&version={version}&type={type}", HttpMethod.GET, null, typeRef,
-                eventClassFamilyId, version, type);
-        return entity.getBody();
-    }
\ No newline at end of file
+public List<EventClassDto> getEventClassesByFamilyIdVersionAndType(String eventClassFamilyId,
+                                                                     int version,
+                                                                     EventClassType type)
+      throws Exception {
+    ResponseEntity<List<EventClassDto>> entity = restTemplate.exchange(restTemplate.getUrl()
+        + "eventClasses?eventClassFamilyId={eventClassFamilyId}&"
+        + "version={version}"
+        + "&type={type}",
+        HttpMethod.GET, null,  new ParameterizedTypeReference<List<EventClassDto>>() {},
+        eventClassFamilyId, version, type);
+    return entity.getBody();
+  }
\ No newline at end of file
