diff --git a/old_product.java b/new_product.java
index 41f61a9..f9795ed 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@ public static Date toDate(String dateString,String...parsePatterns){
         try{
             return DateUtils.parseDate(dateString, parsePatterns);
         }catch (ParseException e){
-            String pattern = "input dateString:[{}],parsePatterns:[{}] parseDate exception";
-            throw new IllegalArgumentException(Slf4jUtil.format(pattern, dateString, parsePatterns), e);
+            String pattern = "parse dateString [{}] use patterns:[{}] to date exception,message:[{}]";
+            throw new IllegalArgumentException(Slf4jUtil.format(pattern, dateString, parsePatterns, e.getMessage()), e);
         }
     }
\ No newline at end of file
