diff --git a/old_test.java b/new_test.java
index 09073b8..51de835 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,16 +1,12 @@
 @Test
     public void testGetTemporaryDirectory() throws Exception
     {
-        final Container container = getComponentManager().lookup(Container.class);
-        final ApplicationContext applicationContext = getMockery().mock(ApplicationContext.class);
+        final Environment environment = getComponentManager().lookup(Environment.class);
 
         getMockery().checking(new Expectations()
         {
             {
-                oneOf(container).getApplicationContext();
-                will(returnValue(applicationContext));
-
-                oneOf(applicationContext).getTemporaryDirectory();
+                oneOf(environment).getTemporaryDirectory();
                 will(returnValue(new File(System.getProperty("java.io.tmpdir"))));
             }
         });
