diff --git a/old_test.java b/new_test.java
index 52246a6..888380e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,4 +7,5 @@
         assertEquals(a.compareTo(a) == 0, true);
         assertEquals(b.compareTo(b) == 0, true);
         assertEquals(a.toInstant().compareTo(b.toInstant()) < 0, true);
+        assertEquals(OffsetDateTime.timeLineOrder().compare(a, b) < 0, true);
     }
\ No newline at end of file
