diff --git a/old_test.java b/new_test.java
index 5afbb73..7a645bb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,6 @@
         double x = 3.0;
         int df = 10;
         double expResult = 0.018575928421771;
-        double result = ContinuousDistributions.ChisquareCdf(x, df);
+        double result = ContinuousDistributions.chisquareCdf(x, df);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
