diff --git a/old_product.java b/new_product.java
index 57e1c7d..3d49795 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public static double ChisquareCdf(double x, int df) {
+public static double chisquareCdf(double x, int df) {
         if(df<=0) {
             throw new IllegalArgumentException("The degrees of freedom need to be positive.");
         }
         
-        return GammaCdf(x/2.0, df/2.0);
+        return ContinuousDistributions.gammaCdf(x/2.0, df/2.0);
     }
\ No newline at end of file
