diff --git a/old_test.java b/new_test.java
index 6aa44b7..4f26961 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testReconcile_noCm_withKafka_withPrivate(TestContext context) {
+    public void testReconcile_noResource_withKafka_withPrivate(TestContext context) {
         Topic kubeTopic = null;
         Topic kafkaTopic = new Topic.Builder(topicName.toString(), 10, (short) 2, map("cleanup.policy", "bar")).build();
         Topic privateTopic = kafkaTopic;
