diff --git a/old_test.java b/new_test.java
index 3ff30b1..df4b8bb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,11 @@
 @Test
-	@Verifies(value = "should return true if given middle and family name are equal", method = "equalsContent(PersonName)")
-	public void equalsContent_shouldReturnTrueIfGivenMiddleAndFamilyNameAreEqual() throws Exception {
-		PersonName pn = new PersonName(1); // a different person name id than below
-		pn.setGivenName("Adam");
-		pn.setMiddleName("Alex");
-		pn.setFamilyName("Jones");
-		PersonName other = new PersonName(2); // a different person name id than above
-		other.setGivenName("Adam");
-		other.setMiddleName("Alex");
-		other.setFamilyName("Jones");
+	@Verifies(value = "should return false if family name 2 is not equal", method = "equalsContent(PersonName)")
+	public void equalsContent_shouldReturnFalseIfFamilyName2IsNotEqual() throws Exception {
+		PersonName name1 = new PersonName(1);
+		PersonName name2 = new PersonName(2);
 		
-		Assert.assertTrue(pn.equalsContent(other));
+		name1.setFamilyName2("van der");
+		name2.setFamilyName2("de");
+		
+		assertThat(name1.equalsContent(name2), is(false));
 	}
\ No newline at end of file
