diff --git a/old_product.java b/new_product.java
index cb3442e..34ac96b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,37 +1,17 @@
 @SuppressWarnings("unchecked")
 	public boolean equalsContent(PersonAddress otherAddress) {
-		boolean returnValue = true;
-		
-		// these are the methods to compare. All are expected to be Strings
-		String[] methods = { "getAddress1", "getAddress2", "getAddress3", "getAddress4", "getAddress5", "getAddress6",
-		        "getCityVillage", "getCountyDistrict", "getStateProvince", "getCountry", "getPostalCode", "getLatitude",
-		        "getLongitude" };
-		
-		Class addressClass = this.getClass();
-		
-		// loop over all of the selected methods and compare this and other
-		for (String methodName : methods) {
-			try {
-				Method method = addressClass.getMethod(methodName, new Class[] {});
-				
-				String thisValue = (String) method.invoke(this);
-				String otherValue = (String) method.invoke(otherAddress);
-				
-				if (otherValue != null && otherValue.length() > 0)
-					returnValue &= otherValue.equals(thisValue);
-				
-			}
-			catch (NoSuchMethodException e) {
-				log.warn("No such method for comparison " + methodName, e);
-			}
-			catch (IllegalAccessException e) {
-				log.error("Error while comparing addresses", e);
-			}
-			catch (InvocationTargetException e) {
-				log.error("Error while comparing addresses", e);
-			}
-			
-		}
-		
-		return returnValue;
+		return new EqualsBuilder().append(defaultString(otherAddress.getAddress1()), defaultString(address1)).append(
+		    defaultString(otherAddress.getAddress2()), defaultString(address2)).append(
+		    defaultString(otherAddress.getAddress3()), defaultString(address3)).append(
+		    defaultString(otherAddress.getAddress4()), defaultString(address4)).append(
+		    defaultString(otherAddress.getAddress5()), defaultString(address5)).append(
+		    defaultString(otherAddress.getAddress6()), defaultString(address6)).append(
+		    defaultString(otherAddress.getCityVillage()), defaultString(cityVillage)).append(
+		    defaultString(otherAddress.getCountyDistrict()), defaultString(countyDistrict)).append(
+		    defaultString(otherAddress.getStateProvince()), defaultString(stateProvince)).append(
+		    defaultString(otherAddress.getCountry()), defaultString(country)).append(
+		    defaultString(otherAddress.getPostalCode()), defaultString(postalCode)).append(
+		    defaultString(otherAddress.getLatitude()), defaultString(latitude)).append(
+		    defaultString(otherAddress.getLongitude()), defaultString(longitude)).append(otherAddress.getStartDate(),
+		    startDate).append(otherAddress.getEndDate(), endDate).isEquals();
 	}
\ No newline at end of file
