diff --git a/old_test.java b/new_test.java
index 197408e..34cd56a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,5 +3,5 @@
     {
         assertThat("testGetBundles 0", bs.getBundles(), notNullValue());
         //by default no bundles
-        assertTrue("testGetBundles 1", bs.getBundles().length == 0);
+        assertTrue("testGetBundles 1", bs.getBundles().size() == 0);
     }
\ No newline at end of file
