diff --git a/old_product.java b/new_product.java
index f5e39ea..fb4369f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,5 @@
-public String toString(int precition) {
+public String toString(int precision) {
         StringBuilder sb = new StringBuilder();
-        values.forEach((k, v) -> {
-            sb.append(String.format("%s = %." + precition + "f\n", k.getPrettyName(), v));
-        });
+        values.forEach((k, v) -> sb.append(String.format("%s = %." + precision + "f\n", k.getPrettyName(), v)));
         return sb.toString();
     }
\ No newline at end of file
