diff --git a/old_test.java b/new_test.java
index 40fa13e..86b3af0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
 	public void process() {
-		ImageUInt8 image = new ImageUInt8(20,25);
+		GrayU8 image = new GrayU8(20,25);
 		ImageMiscOps.fillRectangle(image,100,0,0,10,25);
-		ImageSInt32 output = new ImageSInt32(20,25);
+		GrayS32 output = new GrayS32(20,25);
 
 		// normal images
 		process(image, output);
