diff --git a/old_product.java b/new_product.java
index 7510598..b1e86cc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
 @Override
-    public void getStreamConfig(String scope, String stream, final AsyncResponse asyncResponse) {
+    public void getStream(final String scope, final String stream, final SecurityContext securityContext,
+            final AsyncResponse asyncResponse) {
         long traceId = LoggerHelpers.traceEnter(log, "getStreamConfig");
 
         StreamMetadataStore streamStore = controllerService.getStreamStore();
