diff --git a/old_test.java b/new_test.java
index 29f98d4..738d5f3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
-    public void testGetStreamConfig() throws ExecutionException, InterruptedException {
+    public void testGetStream() throws ExecutionException, InterruptedException {
         when(mockControllerService.getStreamStore()).thenReturn(mockStreamStore);
 
         // Test to get an existing stream
         when(mockStreamStore.getConfiguration(stream1)).thenReturn(streamConfigFuture);
         response = target(resourceURI).request().async().get();
-        streamResponseActual = response.get().readEntity(StreamResponse.class);
+        streamResponseActual = response.get().readEntity(StreamProperty.class);
         assertEquals("Get Stream Config Status", 200, response.get().getStatus());
         testExpectedVsActualObject(streamResponseExpected, streamResponseActual);
 
@@ -14,6 +14,6 @@
             throw new DataNotFoundException("Stream Not Found");
         }));
         response = target(resourceURI2).request().async().get();
-        streamResponseActual = response.get().readEntity(StreamResponse.class);
+        streamResponseActual = response.get().readEntity(StreamProperty.class);
         assertEquals("Get Stream Config Status", 404, response.get().getStatus());
     }
\ No newline at end of file
