diff --git a/old_product.java b/new_product.java
index 563c751..e705d4f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,5 +3,5 @@
     // return true if a deterministic expression returns at least one node
     final SeqType st = seqType();
     return st.type instanceof NodeType && st.oneOrMore() && !has(Flag.UPD) && !has(Flag.NDT) ?
-      cc.replaceWith(this, Bln.TRUE) : this;
+      cc.replaceEbv(this, Bln.TRUE) : this;
   }
\ No newline at end of file
