diff --git a/old_test.java b/new_test.java
index 30d1304..7ff74aa 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,6 +8,8 @@
     query("empty(<a/>[b = 'c'])", "true");
     query("exists(<a/>[b = 'c'])", "false");
 
+    query("let $n := <n/> where $n[<a><b/><b/></a>/*] return $n", "<n/>");
+
     check("empty(<a>X</a>[text()])", null, "//@axis = 'child'");
     check("exists(<a>X</a>[text()])", null, "//@axis = 'child'");
     check("boolean(<a>X</a>[text()])", null, "//@axis = 'child'");
