diff --git a/old_product.java b/new_product.java
index 1794228..dacfc14 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,9 +6,7 @@
 		} else {
 			
 			// for the following elements OrderGroup.hbm.xml says: not-null="true"
-			ValidationUtils.rejectIfEmpty(errors, "creator", "error.null");
 			ValidationUtils.rejectIfEmpty(errors, "patient", "error.null");
-			ValidationUtils.rejectIfEmpty(errors, "dateCreated", "error.null");
 			
 			if (group.getMembers() == null || group.getMembers().isEmpty())
 				errors.rejectValue("members", "OrderGroup.noMembersPresent");
