diff --git a/old_test.java b/new_test.java
index 355eb34..1eed8a4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,7 +12,5 @@
 		Errors errors = new BindException(group, "group");
 		getValidator().validate(group, errors);
 		
-		Assert.assertTrue(errors.hasFieldErrors("creator"));
 		Assert.assertTrue(errors.hasFieldErrors("patient"));
-		Assert.assertTrue(errors.hasFieldErrors("dateCreated"));
 	}
\ No newline at end of file
