diff --git a/old_test.java b/new_test.java
index 0a530d4..15bc2cc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testUpdatesNeeded() throws Exception {
         StandardUpdate instance = getStandardUpdateTask();
-        Updateable result = instance.updatesNeeded();
+        UpdateableNvdCve result = instance.updatesNeeded();
         assertNotNull(result);
     }
\ No newline at end of file
