diff --git a/old_product.java b/new_product.java
index a7e3b12..42471ff 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-protected final Updateable updatesNeeded() throws MalformedURLException, DownloadFailedException, UpdateException {
-        Updateable updates = null;
+protected final UpdateableNvdCve updatesNeeded() throws MalformedURLException, DownloadFailedException, UpdateException {
+        UpdateableNvdCve updates = null;
         try {
             updates = retrieveCurrentTimestampsFromWeb();
         } catch (InvalidDataException ex) {
