diff --git a/old_test.java b/new_test.java
index 8675d44..4e419c6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
 
         // Test converting request to script
         final TransformService service = new TransformService(TransformScript.class, Mockito.mock(SparkScriptEngine.class), Mockito.mock(SparkContextService.class),
-                                                              Mockito.mock(TransformJobTracker.class));
+                                                              Mockito.mock(JobTrackerService.class));
 
         final String expected = IOUtils.toString(getClass().getResourceAsStream("transform-service-script1.scala"), "UTF-8");
         Assert.assertEquals(expected, service.toScript(request));
