diff --git a/old_product.java b/new_product.java
index fe2979b..9a5c287 100644
--- a/old_product.java
+++ b/new_product.java
@@ -30,12 +30,12 @@ private Item updateEntries(final QueryContext ctx) throws QueryException {
         zos.close();
       } catch(final IOException ex) {
         ok = false;
-        ZIP_FAIL.thrw(info, ex.getMessage());
+        ZIP_FAIL.thrw(info, ex);
       } finally {
         if(fos != null) try { fos.close(); } catch(final IOException ex) { }
       }
     } catch(final IOException ex) {
-      throw ZIP_FAIL.thrw(info, ex.getMessage());
+      throw ZIP_FAIL.thrw(info, ex);
     } finally {
       if(zf != null) try { zf.close(); } catch(final IOException e) { }
       if(ok) {
