diff --git a/old_test.java b/new_test.java
index 48de935..81fc8b1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test(groups={"tck"})
     public void test_withZoneLocked() {
         ZonedDateTime base = ZonedDateTime.of(TEST_LOCAL_2008_06_30_11_30_59_500, ZONE_PARIS);
-        ZonedDateTime test = base.withZoneLocked();
+        ZonedDateTime test = base.withLockedOffset();
         ZonedDateTime expected = ZonedDateTime.of(TEST_LOCAL_2008_06_30_11_30_59_500, ZONE_0200);
         assertEquals(test, expected);
     }
\ No newline at end of file
