diff --git a/old_test.java b/new_test.java
index 3f406c1..ada947a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,17 +1,24 @@
 @Test
-    public void getDrugRoutes_shouldGetDrugRoutesAssociatedConceptPrividedInGlobalProperties() throws Exception {
-        Concept drugRoutes = new Concept();
-        ConceptName cn = new ConceptName("Test concept for drug routes", Locale.US);
-        drugRoutes.addName(cn);
-        drugRoutes.setSet(true);
-        drugRoutes.addSetMember(conceptService.getConcept(3));
-        drugRoutes.addSetMember(conceptService.getConcept(4));
-        drugRoutes.addSetMember(conceptService.getConcept(5));
-        drugRoutes=conceptService.saveConcept(drugRoutes);
-        AdministrationService as = Context.getAdministrationService();
-        List<GlobalProperty> globalProperties = Context.getAdministrationService().getAllGlobalProperties();
-        globalProperties.add(new GlobalProperty(OpenmrsConstants.DRUG_ROUTE_CONCEPT_UUID, drugRoutes.getUuid(), "test"));
-        as.saveGlobalProperties(globalProperties);
-        List<Concept> drugRoutesList=orderService.getDrugRoutes();
-        assertEquals(3,drugRoutesList.size());
-    }
\ No newline at end of file
+	public void getDrugRoutes_shouldGetDrugRoutesAssociatedConceptPrividedInGlobalProperties() throws Exception {
+		Concept drugRoutes = new Concept();
+		ConceptName cn = new ConceptName("Test concept for drug routes", Locale.US);
+		drugRoutes.addName(cn);
+		drugRoutes.setSet(true);
+		Concept concept3 = conceptService.getConcept(3);
+		Concept concept4 = conceptService.getConcept(4);
+		Concept concept5 = conceptService.getConcept(5);
+		drugRoutes.addSetMember(concept3);
+		drugRoutes.addSetMember(concept4);
+		drugRoutes.addSetMember(concept5);
+		drugRoutes = conceptService.saveConcept(drugRoutes);
+		
+		AdministrationService as = Context.getAdministrationService();
+		List<GlobalProperty> globalProperties = as.getAllGlobalProperties();
+		globalProperties.add(new GlobalProperty(OpenmrsConstants.DRUG_ROUTE_CONCEPT_UUID, drugRoutes.getUuid(), "test"));
+		as.saveGlobalProperties(globalProperties);
+		List<Concept> drugRoutesList = orderService.getDrugRoutes();
+		assertEquals(3, drugRoutesList.size());
+		assertTrue(drugRoutesList.contains(concept3));
+		assertTrue(drugRoutesList.contains(concept4));
+		assertTrue(drugRoutesList.contains(concept5));
+	}
\ No newline at end of file
