diff --git a/old_product.java b/new_product.java
index 195ea97..02ee5b5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,10 @@
 @Override
-    @Transactional(readOnly = true)
-    public List<Concept> getDrugRoutes(){
-        String conceptUuid = Context.getAdministrationService().getGlobalProperty(OpenmrsConstants.DRUG_ROUTE_CONCEPT_UUID);
-        return Context.getConceptService().getConceptByUuid(conceptUuid).getSetMembers();
-    }
\ No newline at end of file
+	@Transactional(readOnly = true)
+	public List<Concept> getDrugRoutes() {
+		String conceptUuid = Context.getAdministrationService().getGlobalProperty(OpenmrsConstants.DRUG_ROUTE_CONCEPT_UUID);
+		Concept concept = Context.getConceptService().getConceptByUuid(conceptUuid);
+		if (concept != null && concept.isSet()) {
+			return concept.getSetMembers();
+		}
+		return Collections.EMPTY_LIST;
+	}
\ No newline at end of file
