diff --git a/old_test.java b/new_test.java
index 5593e7f..5c0ca0c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,7 +10,7 @@
         int n = 100;
         boolean withReplacement = true;
         double expResult = n;
-        FlatDataCollection sampledIds = SRS.weightedProbabilitySampling(frequencyTable, n, withReplacement);
+        FlatDataCollection sampledIds = SRS.weightedSampling(frequencyTable, n, withReplacement);
         double result = sampledIds.size();
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
