diff --git a/old_product.java b/new_product.java
index ef554a7..4011fc5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,7 +9,7 @@ public static TransposeDataCollection weightedProbabilitySampling(AssociativeArr
                 continue;
             }
             
-            sampledIds.put(strata, SRS.weightedProbabilitySampling(entry.getValue(), sampleN.intValue(), withReplacement));
+            sampledIds.put(strata, SRS.weightedSampling(entry.getValue(), sampleN.intValue(), withReplacement));
         }
         
         return sampledIds;
