diff --git a/old_test.java b/new_test.java
index 402ba5f..3e38de7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,15 @@
 @Test
   public void testLoadValidators() {
-    Props props = new Props(baseProps);
-    URL validatorUrl = Resources.getResource("project/testValidators");
-    URL configUrl = Resources.getResource("test-conf/azkaban-validators-test2.xml");
+    final Props props = new Props(this.baseProps);
+    final URL validatorUrl = Resources.getResource("project/testValidators");
+    final URL configUrl = Resources.getResource("test-conf/azkaban-validators-test2.xml");
     props.put(ValidatorConfigs.VALIDATOR_PLUGIN_DIR, validatorUrl.getPath());
     props.put(ValidatorConfigs.XML_FILE_PARAM, configUrl.getPath());
 
-    XmlValidatorManager manager = new XmlValidatorManager(props);
-    assertEquals("XmlValidatorManager should contain 2 validators.", manager.getValidatorsInfo().size(), 2);
-    assertEquals("XmlValidatorManager should contain the validator specified in the xml configuration file.",
+    final XmlValidatorManager manager = new XmlValidatorManager(props);
+    assertEquals("XmlValidatorManager should contain 2 validators.",
+        manager.getValidatorsInfo().size(), 2);
+    assertEquals(
+        "XmlValidatorManager should contain the validator specified in the xml configuration file.",
         manager.getValidatorsInfo().get(1), "Test");
   }
\ No newline at end of file
