diff --git a/old_test.java b/new_test.java
index a2a7332..38cffa3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testClone() throws CloneNotSupportedException {
         MixtureComponent gaussian = new MixtureComponent(lm, new float[]{2}, new float[][]{{3}}, new float[]{4}, new float[]{5}, new float[][]{{6}}, new float[]{7});
 
-        MixtureComponent clonedGaussian = (MixtureComponent) gaussian.clone();
+        MixtureComponent clonedGaussian = gaussian.clone();
 
         assertTrue(!clonedGaussian.equals(gaussian));
 
