diff --git a/old_product.java b/new_product.java
index 4567f27..e8568ac 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
-public Object clone() throws CloneNotSupportedException {
+@Override
+    public DoubleData clone() throws CloneNotSupportedException {
         try {
-            DoubleData data = (DoubleData) super.clone();
+            DoubleData data = (DoubleData)super.clone();
             data.sampleRate = sampleRate;
             data.collectTime = collectTime;
             data.firstSampleNumber = firstSampleNumber;
