diff --git a/old_test.java b/new_test.java
index dc38f14..c4f20f3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,12 +12,11 @@
 
         RandomForestRegressionTrainer trainer = new RandomForestRegressionTrainer(4, 3, 5, 0.3, 4, 0.1);
         ModelsComposition model = trainer.fit(sample, parts, (k, v) -> v, (k, v) -> k);
+        model.getModels().forEach(m -> {
+            assertTrue(m instanceof ModelOnFeaturesSubspace);
+            assertTrue(((ModelOnFeaturesSubspace) m).getMdl() instanceof DecisionTreeConditionalNode);
+        });
 
         assertTrue(model.getPredictionsAggregator() instanceof MeanValuePredictionsAggregator);
         assertEquals(5, model.getModels().size());
-
-        for (ModelsComposition.ModelOnFeaturesSubspace tree : model.getModels()) {
-            assertTrue(tree.getMdl() instanceof DecisionTreeConditionalNode);
-            assertEquals(3, tree.getFeaturesMapping().size());
-        }
     }
\ No newline at end of file
