diff --git a/old_product.java b/new_product.java
index d9fd70e..8dc5551 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,31 +2,9 @@
         IgniteBiFunction<K, V, double[]> featureExtractor,
         IgniteBiFunction<K, V, Double> lbExtractor) {
 
-        List<ModelsComposition.ModelOnFeaturesSubspace> learnedModels = new ArrayList<>();
-        List<Future<ModelsComposition.ModelOnFeaturesSubspace>> futures = new ArrayList<>();
-
-        for (int i = 0; i < ensembleSize; i++) {
-            if (threadPool == null)
-                learnedModels.add(learnModel(datasetBuilder, featureExtractor, lbExtractor));
-            else {
-                Future<ModelsComposition.ModelOnFeaturesSubspace> fut = threadPool.submit(() -> {
-                    return learnModel(datasetBuilder, featureExtractor, lbExtractor);
-                });
-
-                futures.add(fut);
-            }
-        }
-
-        if (threadPool != null) {
-            for (Future<ModelsComposition.ModelOnFeaturesSubspace> future : futures) {
-                try {
-                    learnedModels.add(future.get());
-                }
-                catch (InterruptedException | ExecutionException e) {
-                    throw new RuntimeException(e);
-                }
-            }
-        }
+        List<ModelOnFeaturesSubspace> learnedModels = new ArrayList<>();
+        for (int i = 0; i < ensembleSize; i++)
+            learnedModels.add(learnModel(datasetBuilder, featureExtractor, lbExtractor));
 
         return new ModelsComposition(learnedModels, predictionsAggregator);
     }
\ No newline at end of file
