diff --git a/old_product.java b/new_product.java
index d57ace4..b3035ab 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,4 @@
-public void handleHead(MessageInfo messageInfo) {
+@Override
+  public void handleHead(RestRequestInfo restRequestInfo) {
     throw new IllegalStateException("handleHead() not implemented in " + this.getClass().getSimpleName());
   }
\ No newline at end of file
