diff --git a/old_test.java b/new_test.java
index 69b25d9..f4623e2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test(expected = IllegalStateException.class)
   public void handleHeadTest()
-      throws IOException, JSONException, RestServiceException, URISyntaxException {
-    AdminBlobStorageService adminBlobStorageService = getAdminBlobStorageService();
-    MessageInfo messageInfo = createMessageInfo(RestMethod.HEAD, "/", new JSONObject());
-    adminBlobStorageService.handleHead(messageInfo);
+      throws JSONException, RestServiceException, URISyntaxException {
+    RestRequestInfo restRequestInfo = createRestRequestInfo(RestMethod.HEAD, "/", new JSONObject());
+    adminBlobStorageService.handleHead(restRequestInfo);
   }
\ No newline at end of file
