diff --git a/old_test.java b/new_test.java
index e0569f3..d1debf5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
     org.apache.hadoop.conf.Configuration conf = new org.apache.hadoop.conf.Configuration();
     mMockHdfsUnderFileSystem.prepareConfiguration("", conf);
     Assert.assertEquals("org.apache.hadoop.hdfs.DistributedFileSystem", conf.get("fs.hdfs.impl"));
-    Assert.assertFalse(conf.getBoolean("fs.hdfs.impl.disable.cache", false));
+    Assert.assertTrue(conf.getBoolean("fs.hdfs.impl.disable.cache",false));
     Assert.assertNotNull(conf.get(PropertyKey.UNDERFS_HDFS_CONFIGURATION.toString()));
   }
\ No newline at end of file
