diff --git a/old_test.java b/new_test.java
index 7826a53..710a3f4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,7 +13,7 @@
         when(routerVO2.getIsRedundantRouter()).thenReturn(true);
         when(routerVO2.getState()).thenReturn(VirtualMachine.State.Stopped);
         // If this deployment is not redundant nothing will be executed
-        this.deployment.isRedundant = true;
+        when(this.mockNw.isRedundant()).thenReturn(true);
 
         // Execute
         this.deployment.setupPriorityOfRedundantRouter();
