diff --git a/old_product.java b/new_product.java
index 601eb67..f30139d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public void contextInitialized(ServletContextEvent event) {
-		final String contextPath = event.getServletContext().getContextPath();
+		final String contextPath = Parameters.getContextPath(event.getServletContext());
 		if (!instanceEnabled) {
 			if (!CONTEXT_PATHS.contains(contextPath)) {
 				// si jars dans tomcat/lib, il y a plusieurs instances mais dans des webapps différentes (issue 193)
