diff --git a/old_test.java b/new_test.java
index f448b0e..1632462 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,8 @@
 @Test
-  public void optimize() {
+  public void optimize() throws BaseXException {
     query(_DB_OPTIMIZE.args(NAME));
     query(_DB_OPTIMIZE.args(NAME));
+    error(_DB_OPTIMIZE.args(NAME, "true()"), Err.UPDBOPTERR);
+    new Close().execute(context);
     query(_DB_OPTIMIZE.args(NAME, "true()"));
   }
\ No newline at end of file
