diff --git a/old_test.java b/new_test.java
index 4e4df7c..25a3f8c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,19 +5,19 @@
 
 		final Candidate olderStockCandidate = Candidate.builder()
 				.type(Type.STOCK)
-				.orgId(1)
+				.orgId(org.getAD_Org_ID())
 				.productId(product.getM_Product_ID())
 				.warehouseId(warehouse.getM_Warehouse_ID())
 				.quantity(olderStockQty)
 				.date(t1)
 				.build();
-		candidateChangeHandler.updateStock(olderStockCandidate);
+		candidateChangeHandler.addOrUpdateStock(olderStockCandidate);
 
 		final BigDecimal supplyQty = new BigDecimal("23");
 
 		final Candidate candidate = Candidate.builder()
 				.type(Type.SUPPLY)
-				.orgId(1)
+				.orgId(org.getAD_Org_ID())
 				.subType(SubType.PRODUCTION)
 				.productId(product.getM_Product_ID())
 				.warehouseId(warehouse.getM_Warehouse_ID())
