diff --git a/old_test.java b/new_test.java
index 87a5cec..48e6b71 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,8 @@
 @Test
     public void setRequired_nullValueBecomesDefaultValue() {
+        if (type == ObjectSchemaType.IMMUTABLE) {
+            return;
+        }
         for (FieldType fieldType : FieldType.values()) {
             String fieldName = fieldType.name();
             switch (fieldType) {
@@ -13,7 +16,7 @@
                         break;
                     }
                     schema.addField(fieldName, fieldType.getType());
-                    DynamicRealmObject object = realm.createObject(schema.getClassName());
+                    DynamicRealmObject object = ((DynamicRealm)realm).createObject(schema.getClassName());
                     assertTrue(object.isNull(fieldName));
                     schema.setRequired(fieldName, true);
                     assertFalse(object.isNull(fieldName));
