diff --git a/old_test.java b/new_test.java
index 4086bcc..5780153 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
 
 		for( int i = 6; i <= 6; i += 2 ) {
 
-			WaveletCoefficient_F32 desc = FactoryWaveletCoiflet.generate_F32(i);
+			WlCoef_F32 desc = FactoryWaveletCoiflet.generate_F32(i).forward;
 
 			double sumScaling = UtilWavelet.sumCoefficients(desc.scaling);
 			double sumWavelet = UtilWavelet.sumCoefficients(desc.wavelet);
