diff --git a/old_product.java b/new_product.java
index 61c413d..2fb681b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,11 +8,15 @@ public long createFile(AlluxioURI path, CreateFileOptions options)
         mMountTable.checkUnderWritableMountPoint(path);
       }
       InodeTree.CreatePathResult createResult = createFileInternal(path, options);
-      List<Inode<?>> created = createResult.getCreated();
+      try {
+        List<Inode<?>> created = createResult.getCreated();
 
-      writeJournalEntry(mDirectoryIdGenerator.toJournalEntry());
-      journalCreatePathResult(createResult);
-      flushJournal();
-      return created.get(created.size() - 1).getId();
+        writeJournalEntry(mDirectoryIdGenerator.toJournalEntry());
+        journalCreatePathResult(createResult);
+        flushJournal();
+        return created.get(created.size() - 1).getId();
+      } finally {
+        createResult.unlock();
+      }
     }
   }
\ No newline at end of file
