diff --git a/old_test.java b/new_test.java
index d164b0a..9c21849 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public void createFileTest() throws Exception {
     // created nested file
     mTree.createPath(NESTED_FILE_URI, sNestedFileOptions);
-    Inode<?> nestedFile = mTree.getInodeByPath(NESTED_FILE_URI);
+    Inode<?> nestedFile = getInodeByPath(mTree, NESTED_FILE_URI);
     Assert.assertEquals("file", nestedFile.getName());
     Assert.assertEquals(2, nestedFile.getParentId());
     Assert.assertTrue(nestedFile.isFile());
