diff --git a/old_test.java b/new_test.java
index 0d38520..64f9ab8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,13 @@
 @Test
     public void testRelinquishMastership() {
-        // TODO: will be added when CompletableFuture is removed
+        mockService.relinquishMastershipSync(anyObject());
+        expectLastCall();
+        replay(mockService);
+
+        final WebTarget wt = target();
+        final Response response = wt.path("mastership/" + deviceId1.toString() +
+                "/relinquish").request().get();
+        assertThat(response.getStatus(), is(HttpURLConnection.HTTP_CREATED));
+        String location = response.getLocation().toString();
+        assertThat(location, Matchers.startsWith(deviceId1.toString()));
     }
\ No newline at end of file
