diff --git a/old_product.java b/new_product.java
index 2bf8b31..af3e4b4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public Mode applyUMask(Mode umask) {
-    return new Mode(mUserBits.and(umask.mUserBits.not()), mGroupBits.and(umask.mGroupBits.not()),
+    return new Mode(mOwnerBits.and(umask.mOwnerBits.not()), mGroupBits.and(umask.mGroupBits.not()),
         mOtherBits.and(umask.mOtherBits.not()));
   }
\ No newline at end of file
