diff --git a/old_test.java b/new_test.java
index a34b03a..5924006 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
     Permission permission = new Permission("user1", "group1", Mode.getDefault());
     permission.applyUMask(umaskMode);
 
-    Assert.assertEquals(Mode.Bits.ALL, permission.getMode().getUserBits());
+    Assert.assertEquals(Mode.Bits.ALL, permission.getMode().getOwnerBits());
     Assert.assertEquals(Mode.Bits.READ_EXECUTE, permission.getMode().getGroupBits());
     Assert.assertEquals(Mode.Bits.READ_EXECUTE, permission.getMode().getOtherBits());
     verifyPermission("user1", "group1", (short) 0755, permission);
