diff --git a/old_test.java b/new_test.java
index 0a7fe29..f12e891 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,7 @@
         Job initial = new Job();
         UUID uuid = UUID.randomUUID();
         initial.setJobID(uuid.toString());
+        initial.setUserName("myUserName");
         pm.createEntity(initial);
         Job deleted = pm.deleteEntity(uuid.toString(),
                 Job.class);
