diff --git a/old_test.java b/new_test.java
index 616dac3..00f4875 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,9 @@
 @Test
     public void testSetDependencies() {
         Assert.assertNotNull(this.c.getDependencies());
-        final Set<String> dependencies = Sets.newHashSet("s3://netflix/jars/myJar.jar");
+        final FileEntity dependency = new FileEntity();
+        dependency.setFile("s3://netflix/jars/myJar.jar");
+        final Set<FileEntity> dependencies = Sets.newHashSet(dependency);
         this.c.setDependencies(dependencies);
         Assert.assertEquals(dependencies, this.c.getDependencies());
 
