diff --git a/old_product.java b/new_product.java
index c39456d..1830f14 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public void setDependencies(final Set<String> dependencies) {
+public void setDependencies(@Nullable final Set<FileEntity> dependencies) {
         this.dependencies.clear();
         if (dependencies != null) {
             this.dependencies.addAll(dependencies);
