diff --git a/old_test.java b/new_test.java
index b549053..a9b6d11 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,13 +2,13 @@
     public void testGetSubnetIps() {
         String bClassCidr = "10.10.0.0/16";
         Set<IpAddress> bClassIps = getSubnetIps(bClassCidr);
-        assertEquals(((Double) Math.pow(2, 16)).intValue() - 2, bClassIps.size());
+        assertEquals(((Double) Math.pow(2, 16)).intValue() - 4, bClassIps.size());
 
         String cClassCidr = "10.10.10.0/24";
         Set<IpAddress> cClassIps = getSubnetIps(cClassCidr);
-        assertEquals(((Double) Math.pow(2, 8)).intValue() - 2, cClassIps.size());
+        assertEquals(((Double) Math.pow(2, 8)).intValue() - 4, cClassIps.size());
 
         String dClassCidr = "10.10.10.10/32";
         Set<IpAddress> dClassIps = getSubnetIps(dClassCidr);
-        assertEquals(1, dClassIps.size());
+        assertEquals(0, dClassIps.size());
     }
\ No newline at end of file
