diff --git a/old_product.java b/new_product.java
index da784eb..d79ca74 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,13 @@
 public static Set<IpAddress> getSubnetIps(String cidr) {
         SubnetUtils utils = new SubnetUtils(cidr);
-        utils.setInclusiveHostCount(true);
+        utils.setInclusiveHostCount(false);
         SubnetUtils.SubnetInfo info = utils.getInfo();
         Set<String> allAddresses =
                 new HashSet<>(Arrays.asList(info.getAllAddresses()));
 
         if (allAddresses.size() > 2) {
-            allAddresses.remove(info.getBroadcastAddress());
-            allAddresses.remove(info.getNetworkAddress());
+            allAddresses.remove(info.getLowAddress());
+            allAddresses.remove(info.getHighAddress());
         }
 
         return allAddresses.stream()
