diff --git a/old_test.java b/new_test.java
index 30dc9c2..e6f6cd2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,18 +2,18 @@
     public void testCsvIdentifiers() {
         EscapeTool instance = new EscapeTool();
         Set<Identifier> ids = null;
-        String expResult = "";
+        String expResult = "\"\"";
         String result = instance.csvIdentifiers(ids);
         assertEquals(expResult, result);
         
         ids = new HashSet<>();
-        expResult = "";
+        expResult = "\"\"";
         result = instance.csvIdentifiers(ids);
         assertEquals(expResult, result);
         
         ids = new HashSet<>();
         ids.add(new Identifier("cpe", "cpe:/a:somegroup:something:1.0", ""));
-        expResult = "";
+        expResult = "\"\"";
         result = instance.csvIdentifiers(ids);
         assertEquals(expResult, result);
         
