diff --git a/old_test.java b/new_test.java
index 05c1611..f95b140 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,20 +1,20 @@
 @Test(groups = "observerMethod")
    public void testAddObserver()
    {
-      EventManager eventManager = new EventManager();
+      EventManager eventManager = new EventManager(manager);
       Observer<DangerCall> observer = new AnObserver<DangerCall>();
       eventManager.addObserver(observer, DangerCall.class);
       DangerCall event = new DangerCall();
 
-      Set<Observer<DangerCall>> observerSet = eventManager.getObservers(new MetaDataCache(), event);
+      Set<Observer<DangerCall>> observerSet = eventManager.getObservers(event);
       assert observerSet.size() == 1;
       assert observerSet.iterator().next().equals(observer);
 
       // Add another observer for the same event, but with an event binding
       observer = new AnObserver<DangerCall>();
       eventManager.addObserver(observer, DangerCall.class, new TameAnnotationLiteral());
-      observerSet = eventManager.getObservers(new MetaDataCache(), event);
+      observerSet = eventManager.getObservers(event);
       assert observerSet.size() == 1;
-      observerSet = eventManager.getObservers(new MetaDataCache(), event, new TameAnnotationLiteral());
+      observerSet = eventManager.getObservers(event, new TameAnnotationLiteral());
       assert observerSet.size() == 2;
    }
\ No newline at end of file
