diff --git a/old_product.java b/new_product.java
index 49fd49b..2236587 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,4 +2,5 @@ public <T> void addObserver(Observer<T> observer, Class<T> eventType, Annotation
    {
       EventObserver<T> eventObserver = new EventObserver<T>(observer, eventType, bindings);
       registeredObservers.put(eventType, eventObserver);
+      log.debug("Added observer " + observer + " observing event type " + eventType);
    }
\ No newline at end of file
