diff --git a/old_product.java b/new_product.java
index 8e9881a..5d0b88c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
 protected boolean submitTopology(String[] appConfs) {
-
     if (!this.topologyName.equals(this.topologyName.toLowerCase())) {
       throw new TopologySubmissionException("K8S scheduler does not allow upper case topologies.");
     }
@@ -8,8 +7,10 @@ protected boolean submitTopology(String[] appConfs) {
       try {
         deployContainer(appConf);
       } catch (IOException ioe) {
-        LOG.log(Level.SEVERE, "Problem deploying container with config: " + appConf);
-        return false;
+        final String message = ioe.getMessage();
+        LOG.log(Level.SEVERE, "Problem deploying container: " + ioe.getMessage(), ioe);
+        LOG.log(Level.SEVERE, "Container config: " + appConf);
+        throw new TopologySubmissionException(message);
       }
     }
     return true;
