diff --git a/old_test.java b/new_test.java
index f6d01f1..9dc17ec 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,16 +1,10 @@
 @Test
-  public void testSubmitTopology() throws Exception {
+  public void testSubmitTopologySuccess() throws Exception {
 
     HttpJsonClient httpJsonClient = PowerMockito.spy(new HttpJsonClient(""));
     PowerMockito.whenNew(HttpJsonClient.class).withAnyArguments().thenReturn(httpJsonClient);
 
-    // Test a bad POST
-    PowerMockito.doThrow(new IOException()).when(httpJsonClient).post(Mockito.anyString(),
-        Mockito.anyInt());
-    Assert.assertFalse(controller.submitTopology(DEPLOY_CONFS));
-
     // Test a good path
     PowerMockito.doNothing().when(httpJsonClient).post(Mockito.anyString(), Mockito.anyInt());
     Assert.assertTrue(controller.submitTopology(DEPLOY_CONFS));
-
   }
\ No newline at end of file
