diff --git a/old_product.java b/new_product.java
index 27b89d8..44e3eb3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,8 @@
-protected boolean openCommPort(String name, int baud) throws Exception {
+protected boolean openCommPort(ConnectionDriver connectionDriver, String name, int baud) throws Exception {
         if (conn == null) {
-            conn = ConnectionFactory.getConnection();
-            logger.info("Connecting to controller using class: " + conn.getClass().getSimpleName());
+            String url = connectionDriver.getProtocol() + name + ":" + baud;
+            conn = ConnectionFactory.getConnection(url);
+            logger.info("Connecting to controller using class: " + conn.getClass().getSimpleName() + " with url " + url);
         }
 
         if (conn != null) {
@@ -9,12 +10,12 @@ protected boolean openCommPort(String name, int baud) throws Exception {
         }
         
         if (conn==null) {
-            throw new Exception(Localization.getString("communicator.exception.port") + ": "+name);
+            throw new Exception(Localization.getString("communicator.exception.port") + ": " + name);
         }
         
         // Handle all events in a single thread.
         this.eventThread.start();
 
         //open it
-        return conn.openPort(name, baud);
+        return conn.openPort();
     }
\ No newline at end of file
