diff --git a/old_test.java b/new_test.java
index 5916de7..16da7fb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
         Boolean expResult = true;
         Boolean result = false;
         try {
-            result = instance.openCommPort(port, portRate);
+            result = instance.openCommPort(getSettings().getConnectionDriver(), port, portRate);
         } catch (Exception e) {
             fail("Unexpected exception from GrblController: " + e.getMessage());
         }
@@ -19,7 +19,7 @@
         String exception = "";
         // Check exception trying to open the comm port twice.
         try {
-            instance.openCommPort(port, portRate);
+            instance.openCommPort(getSettings().getConnectionDriver(), port, portRate);
         } catch (Exception e) {
             exception = e.getMessage();
         }
