diff --git a/old_test.java b/new_test.java
index 380aef3..af4908f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,13 @@
 @Test
-  public void removeByFieldTest() {
-    Assert.assertEquals(3, mSet.getByField(mIntIndex, 1).size());
+  public void nonUniqueRemoveByFieldTest() {
+    Assert.assertEquals(3, mSet.getByField(mNonUniqueIntIndex, 1).size());
     Assert.assertEquals(9, mSet.size());
-    Assert.assertEquals(3, mSet.removeByField(mIntIndex, 1));
+    Assert.assertEquals(3, mSet.removeByField(mNonUniqueIntIndex, 1));
     Assert.assertEquals(6, mSet.size());
-    Assert.assertEquals(0, mSet.getByField(mIntIndex, 1).size());
-    Assert.assertEquals(3, mSet.getByField(mIntIndex, 0).size());
-    Assert.assertEquals(3, mSet.getByField(mIntIndex, 2).size());
-    for (long l = 0; l < 3; l++) {
-      Assert.assertEquals(2, mSet.getByField(mLongIndex, l).size());
+    Assert.assertEquals(0, mSet.getByField(mNonUniqueIntIndex, 1).size());
+    Assert.assertEquals(3, mSet.getByField(mNonUniqueIntIndex, 0).size());
+    Assert.assertEquals(3, mSet.getByField(mNonUniqueIntIndex, 2).size());
+    for (long l = 3; l < 6; l++) {
+      Assert.assertEquals(0, mSet.getByField(mUniqueLongIndex, l).size());
     }
   }
\ No newline at end of file
