diff --git a/old_product.java b/new_product.java
index 40406c9..e8475e1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,23 +1,12 @@
-public int removeByField(FieldIndex<T> index, Object value) {
+public int removeByField(UniqueFieldIndex<T> index, Object value) {
     int removed = 0;
-    if (index instanceof UniqueFieldIndex) {
-      T toRemove = getByFieldInternalUnique(index, value);
-      if (toRemove == null) {
-        return 0;
-      }
-      if (remove(toRemove)) {
-        removed++;
-      }
-    } else {
-      ConcurrentHashSet<T> toRemove = getByFieldInternalNonUnique(index, value);
-      if (toRemove == null) {
-        return 0;
-      }
-      for (T o : toRemove) {
-        if (remove(o)) {
-          removed++;
-        }
-      }
+    T toRemove = getByFieldInternalUnique(index, value);
+
+    if (toRemove == null) {
+      return 0;
+    }
+    if (remove(toRemove)) {
+      removed++;
     }
 
     return removed;
