diff --git a/old_test.java b/new_test.java
index 722eed2..f398741 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,15 +1,15 @@
 @Test(groups={"tck"})
-    public void test_lengthInDays_boolean_notLeapYear() {
-        assertEquals(Month.JANUARY.lengthInDays(false), 31);
-        assertEquals(Month.FEBRUARY.lengthInDays(false), 28);
-        assertEquals(Month.MARCH.lengthInDays(false), 31);
-        assertEquals(Month.APRIL.lengthInDays(false), 30);
-        assertEquals(Month.MAY.lengthInDays(false), 31);
-        assertEquals(Month.JUNE.lengthInDays(false), 30);
-        assertEquals(Month.JULY.lengthInDays(false), 31);
-        assertEquals(Month.AUGUST.lengthInDays(false), 31);
-        assertEquals(Month.SEPTEMBER.lengthInDays(false), 30);
-        assertEquals(Month.OCTOBER.lengthInDays(false), 31);
-        assertEquals(Month.NOVEMBER.lengthInDays(false), 30);
-        assertEquals(Month.DECEMBER.lengthInDays(false), 31);
+    public void test_length_boolean_notLeapYear() {
+        assertEquals(Month.JANUARY.length(false), 31);
+        assertEquals(Month.FEBRUARY.length(false), 28);
+        assertEquals(Month.MARCH.length(false), 31);
+        assertEquals(Month.APRIL.length(false), 30);
+        assertEquals(Month.MAY.length(false), 31);
+        assertEquals(Month.JUNE.length(false), 30);
+        assertEquals(Month.JULY.length(false), 31);
+        assertEquals(Month.AUGUST.length(false), 31);
+        assertEquals(Month.SEPTEMBER.length(false), 30);
+        assertEquals(Month.OCTOBER.length(false), 31);
+        assertEquals(Month.NOVEMBER.length(false), 30);
+        assertEquals(Month.DECEMBER.length(false), 31);
     }
\ No newline at end of file
