diff --git a/old_test.java b/new_test.java
index 4456e0f..27d7c9d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
     XmlWriteOperation<Bird> writeOp =
         XmlFileBasedSink.writeOf(testClass, testRootElement, testFilePrefix)
             .createWriteOperation(options);
-    XmlWriter<Bird> writer = writeOp.createWriter();
+    XmlWriter<Bird> writer = writeOp.createWriter(options);
     assertEquals(testFilePrefix, writer.getWriteOperation().baseTemporaryFilename);
     assertEquals(testRootElement, writer.getWriteOperation().getSink().rootElementName);
     assertNotNull(writer.marshaller);
