diff --git a/old_test.java b/new_test.java
index e60b439..ff65ff9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,7 +12,7 @@
 		RectifyImageOps.fullViewLeft(param,rect1,rect2,rectK);
 
 		// check left image
-		PointTransform_F32 tran = RectifyImageOps.transformPixelToRect_F32(param, rect1);
+		Point2Transform2_F32 tran = RectifyImageOps.transformPixelToRect_F32(param, rect1);
 		checkInside(tran);
 		// the right view is not checked since it is not part of the contract
 	}
\ No newline at end of file
