diff --git a/old_product.java b/new_product.java
index f29b0bb..da2a0b3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,15 @@
             final AsyncResponse asyncResponse) {
         long traceId = LoggerHelpers.traceEnter(log, "deleteScope");
 
+        try {
+            authenticate(scopeName, READ_UPDATE);
+        } catch (AuthenticationException e) {
+            log.warn("Delete scope for {} failed due to authentication failure.", scopeName);
+            asyncResponse.resume(Response.status(Status.UNAUTHORIZED).build());
+            LoggerHelpers.traceLeave(log, "createStream", traceId);
+            return;
+        }
+
         controllerService.deleteScope(scopeName).thenApply(scopeStatus -> {
             if (scopeStatus.getStatus() == DeleteScopeStatus.Status.SUCCESS) {
                 log.info("Successfully deleted scope: {}", scopeName);
