diff --git a/old_test.java b/new_test.java
index eadd936..87d1a25 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,17 +1,5 @@
 @Test
     public void testFlatten() {
-        assert "a b c".equals(StringUtils.flatten(new String[]{"a", "b", "c"}));
-        assert "a b c".equals(StringUtils.flatten(new String[]{"a", "b", "c"}, " "));
-
-        assert "a*b*c".equals(StringUtils.flatten(new String[]{"a", "b", "c"}, "*"));
-
-        assert "a*c".equals(StringUtils.flatten(new String[]{"a", "", "c"}, "*"));
-
-        assert "a*c".equals(StringUtils.flatten(new String[]{"a", null, "c"}, "*"));
-
-        assert "b".equals(StringUtils.flatten(new String[]{null, "b", null}, "*"));
-
-        assert "".equals(StringUtils.flatten(new String[]{null, null, null}, "*"));
-
         assert null == StringUtils.flatten(null, "*");
+        assert null == StringUtils.flatten(null);
     }
\ No newline at end of file
