diff --git a/old_test.java b/new_test.java
index cd47ab3..db588fa 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,9 @@
     public void testSetConfigs() {
         Assert.assertNotNull(this.c.getConfigs());
         Assert.assertTrue(this.c.getConfigs().isEmpty());
-        final Set<String> configs = Sets.newHashSet("s3://netflix.configFile");
+        final FileEntity config = new FileEntity();
+        config.setFile("s3://netflix.configFile");
+        final Set<FileEntity> configs = Sets.newHashSet(config);
         this.c.setConfigs(configs);
         Assert.assertEquals(configs, this.c.getConfigs());
 
