diff --git a/old_product.java b/new_product.java
index e6ebbf4..37134e2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,5 +3,5 @@ public DateTimeFormatter withLocale(Locale locale) {
         if (locale.equals(this.locale)) {
             return this;
         }
-        return new DateTimeFormatter(locale, symbols, printerParser);
+        return new DateTimeFormatter(printerParser, locale, symbols);
     }
\ No newline at end of file
