diff --git a/old_product.java b/new_product.java
index e4f73c7..8e0f562 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public boolean isValidCCN(String rawCCN) {
+static public boolean isValidCCN(String rawCCN) {
         //check for a valid separator
         boolean hasSpace = StringUtils.contains(rawCCN, ' ');
         boolean hasDash = StringUtils.contains(rawCCN, '-');
