diff --git a/old_test.java b/new_test.java
index 9cff04f..a42c4f0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,7 +14,7 @@
 		ppOrderBOMLine.setQtyDelivered(BigDecimal.ZERO);
 
 		Assert.assertThat("Invalid QtyRequired projected",
-				ppOrderBOMBL.calculateQtyRequiredBasedOnFinishedGoodReceipt(ppOrderBOMLine),
+				ppOrderBOMBL.calculateQtyToIssueBasedOnFinishedGoodReceipt(ppOrderBOMLine, ppOrderBOMLine.getC_UOM()).getQty(),
 				// Expected: 50(finished goods received) x 260(mm/finished good) x (scrap=1 + 10/100)
 				Matchers.comparesEqualTo(new BigDecimal("8580"))
 				);
