diff --git a/old_test.java b/new_test.java
index ecc94c1..b35907c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,8 +9,8 @@
 
       Cluster clusterClient = client.getClusterClient();
       MemberListResponse response = clusterClient.listMember().get();
-      String[] newPeerUrl = peerUrls.subList(0, 1).toArray(new String[]{});
-      clusterClient.updateMember(response.getMembers().get(0).getId(), Arrays.asList(newPeerUrl))
+      List<URI> newPeerUrl = peerUrls.subList(0, 1);
+      clusterClient.updateMember(response.getMembers().get(0).getId(), newPeerUrl)
           .get();
     } catch (Exception e) {
       System.out.println(e);
