diff --git a/old_product.java b/new_product.java
index 633b875..3401f96 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,14 +1,14 @@
-public static List<PointIndex_I32> fitPolygon(List<Point2D_I32> sequence,  boolean loop,
-												  double splitFraction, double minimumSideFraction, int iterations) {
+public static List<PointIndex_I32> fitPolygon(List<Point2D_I32> sequence, boolean loop,
+												  double splitFraction, ConfigLength minimumSideLength, int iterations) {
 		GrowQueue_I32 splits = new GrowQueue_I32();
 
 		if( loop ) {
-			SplitMergeLineFitLoop alg = new SplitMergeLineFitLoop(splitFraction,minimumSideFraction,iterations);
+			SplitMergeLineFitLoop alg = new SplitMergeLineFitLoop(splitFraction,minimumSideLength,iterations);
 			alg.process(sequence,splits);
 			RefinePolyLineCorner refine = new RefinePolyLineCorner(true,10);
 			refine.fit(sequence,splits);
 		} else {
-			SplitMergeLineFitSegment alg = new SplitMergeLineFitSegment(splitFraction,minimumSideFraction,iterations);
+			SplitMergeLineFitSegment alg = new SplitMergeLineFitSegment(splitFraction,minimumSideLength,iterations);
 			alg.process(sequence,splits);
 			RefinePolyLineCorner refine = new RefinePolyLineCorner(false,10);
 			refine.fit(sequence,splits);
