diff --git a/old_test.java b/new_test.java
index 2e85fe3..58fa187 100644
--- a/old_test.java
+++ b/new_test.java
@@ -30,7 +30,7 @@
 		// fetch the configuration object and check that everything is as we passed
 		BoneCPConfig config = partialTestClass.getConfig();
 
-		assertEquals(40, config.getPreparedStatementsCacheSize());
+		assertEquals(40, config.getStatementsCacheSize());
 		assertEquals(30, config.getStatementsCachedPerConnection());
 		assertEquals(20, config.getMinConnectionsPerPartition());
 		assertEquals(50, config.getMaxConnectionsPerPartition());
