diff --git a/old_product.java b/new_product.java
index 4857f39..259ccd1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -41,7 +41,7 @@ public void configure(Properties props)
 			this.config.setIdleConnectionTestPeriod(idleConnectionTestPeriod);
 			this.config.setIdleMaxAge(idleMaxAge);
 			this.config.setConnectionTestStatement(connectionTestStatement);
-			this.config.setPreparedStatementsCacheSize(preparedStatementCacheSize);
+			this.config.setStatementsCacheSize(preparedStatementCacheSize);
 			this.config.setStatementsCachedPerConnection(statementsCachedPerConnection);
 			this.config.setInitSQL(initSQL);
 			this.config.setCloseConnectionWatch(closeConnectionWatch);
