diff --git a/old_test.java b/new_test.java
index b32f4ba..1d14247 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
 	public void testIsSecure() {
 
-		RawData rawData = RawData.inbound(new byte[]{0x01, 0x02}, SOURCE, null, getSecureCorrelationContext(), false);
+		RawData rawData = RawData.inbound(new byte[]{0x01, 0x02}, SOURCE, null, getSecureEndpointContext(), false);
 		assertTrue(rawData.isSecure());
 
-		rawData = RawData.inbound(new byte[]{0x01, 0x02}, SOURCE, null, getNonSecureCorrelationContext(), false);
+		rawData = RawData.inbound(new byte[]{0x01, 0x02}, SOURCE, null, getNonSecureEndpointContext(), false);
 		assertFalse(rawData.isSecure());
 
 		rawData = RawData.inbound(new byte[]{0x01, 0x02}, SOURCE, null, null, false);
