diff --git a/old_product.java b/new_product.java
index 8822821..ef04d24 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Nonnull
-    public DataSet createDataSet(@Nonnull final DataSet source) {
+    public DataSet createDataSet(@Nonnull final DataSet source, boolean encryptedCredentials) {
         // Find data source
         final DataSource dataSource = Optional.of(source).map(DataSet::getDataSource).map(DataSource::getId).flatMap(dataSourceProvider::findDataSource)
             .orElseThrow(() -> new CatalogException("catalog.dataset.datasource.invalid"));
@@ -11,9 +11,12 @@
 
         // Create and store data set
         return metadataService.commit(() -> {
+            // Require admin permission if the results should include unencrypted credentials.
+            accessController.checkPermission(AccessController.SERVICES, encryptedCredentials ? FeedServicesAccessControl.EDIT_DATASOURCES : FeedServicesAccessControl.ADMIN_DATASOURCES);
+
             // Resolve the real data set if possible, otherwise create
             com.thinkbiganalytics.metadata.api.catalog.DataSource.ID dataSourceId = dataSourceMetadataProvider.resolveId(dataSource.getId());
             com.thinkbiganalytics.metadata.api.catalog.DataSet ds = modelTransform.buildDataSet(source, metadataProvider.build(dataSourceId));
-            return modelTransform.dataSetToRestModel().apply(ds);
+            return modelTransform.dataSetToRestModel(encryptedCredentials).apply(ds);
         });
     }
\ No newline at end of file
