diff --git a/old_test.java b/new_test.java
index ff19492..451d868 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,9 +5,9 @@
         @Cleanup
         val storage = new InMemoryStorage();
         storage.initialize(DEFAULT_EPOCH);
-        storage.create(segmentName, TIMEOUT).get(TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+        storage.create(segmentName);
 
-        val handle = storage.openWrite(segmentName).join();
+        val handle = storage.openWrite(segmentName);
         ByteArrayOutputStream writeStream = new ByteArrayOutputStream();
 
         for (int j = 0; j < APPENDS_PER_SEGMENT; j++) {
@@ -19,7 +19,7 @@
 
         byte[] expectedData = writeStream.toByteArray();
         byte[] readBuffer = new byte[expectedData.length];
-        int bytesRead = storage.read(handle, 0, readBuffer, 0, readBuffer.length, TIMEOUT).join();
+        int bytesRead = storage.read(handle, 0, readBuffer, 0, readBuffer.length);
         Assert.assertEquals("Unexpected number of bytes read.", readBuffer.length, bytesRead);
         AssertExtensions.assertArrayEquals("Unexpected read result.", expectedData, 0, readBuffer, 0, bytesRead);
     }
\ No newline at end of file
