diff --git a/old_product.java b/new_product.java
index 7dfdba1..1a07283 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,6 @@
-public void append(SegmentHandle handle, InputStream data, int length) {
+@SneakyThrows(StreamSegmentException.class)
+    public void append(SegmentHandle handle, InputStream data, int length) {
         ensurePreconditions();
         Preconditions.checkArgument(!handle.isReadOnly(), "Cannot append using a read-only handle.");
         getStreamSegmentData(handle.getSegmentName()).append(data, length);
-        this.executor.execute(
-                () -> {
-                    long segmentLength = getStreamSegmentData(handle.getSegmentName()).getInfo().getLength();
-                    fireOffsetTriggers(handle.getSegmentName(), segmentLength);
-                });
     }
\ No newline at end of file
