diff --git a/old_product.java b/new_product.java
index 542939c..6d21204 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,6 @@
 @Override
   public void loadValidators(final Props props, final Logger log) {
     this.validators = new LinkedHashMap<>();
-    // Add the default validator
-    final DirectoryFlowLoader flowLoader = new DirectoryFlowLoader(props, log);
-    this.validators.put(flowLoader.getValidatorName(), flowLoader);
-
     if (!props.containsKey(ValidatorConfigs.XML_FILE_PARAM)) {
       logger.warn(
           "Azkaban properties file does not contain the key " + ValidatorConfigs.XML_FILE_PARAM);
