diff --git a/old_test.java b/new_test.java
index fea28f2..1aba2d4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,12 +2,12 @@
 	public void testGetOutputs() {
 		final Operator<?> fixture = new OpImpl(0);
 
-		final List<Operator<?>.Output> result = fixture.getOutputs();
+		final List<JsonStream> result = fixture.getOutputs();
 
 		assertNotNull(result);
 		assertEquals(1, result.size());
 
-		final List<Operator<?>.Output> expectedResults = new ArrayList<Operator<?>.Output>();
+		final List<JsonStream> expectedResults = new ArrayList<JsonStream>();
 		expectedResults.add(fixture.getOutput(0));
 		assertEquals(expectedResults, result);
 	}
\ No newline at end of file
