diff --git a/old_product.java b/new_product.java
index 6752978..44915cd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public TranslationModel getModel(String modelId) {
+public ServiceCall<TranslationModel> getModel(String modelId) {
     if (modelId == null || modelId.isEmpty())
       throw new IllegalArgumentException("modelId cannot be null or empty");
 
-    final Request request = RequestBuilder.get(String.format(PATH_MODEL, modelId)).build();
-    return executeRequest(request, TranslationModel.class);
+    return createServiceCall(createCall(RequestBuilder.get(String.format(PATH_MODEL, modelId)).build3()),
+            ResponseUtil.getObjectConverter(TranslationModel.class));
   }
\ No newline at end of file
