diff --git a/old_test.java b/new_test.java
index 194bdd4..ec4e132 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,5 @@
 @Test(dependsOnMethods = "testWatchDelete")
     public void testCancelWatch() throws ExecutionException, InterruptedException, TimeoutException {
-        watchClient.cancelWatch(watcher);
-        WatchResponse watchResponse = cancelResponse.get(5, TimeUnit.SECONDS);
-        test.assertTrue(watchResponse.getCanceled());
+        CompletableFuture<Boolean> future = watcher.cancel();
+        test.assertTrue(future.get(5, TimeUnit.SECONDS));
     }
\ No newline at end of file
