diff --git a/old_test.java b/new_test.java
index 52f488d..793a3d5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
         final GeoPoint hannover = new GeoPoint(52370816, 9735936);
 		final int zoom = 8;
 
-		final Point point = Mercator.projectGeoPoint(hannover, zoom, null);
+		final BasicPoint point = Mercator.projectGeoPoint(hannover, zoom, null);
 		
 		assertEquals("TODO describe test", 134, point.x);
 		assertEquals("TODO describe test", 84, point.y);
