diff --git a/old_product.java b/new_product.java
index 6425df0..bc4e0a4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-public static Point projectGeoPoint(final GeoPoint aGeoPoint, final int aZoom, final Point aUseAsReturnValue) {
-		final Point p = (aUseAsReturnValue != null) ? aUseAsReturnValue : new Point();
+public static BasicPoint projectGeoPoint(final GeoPoint aGeoPoint, final int aZoom, final BasicPoint aUseAsReturnValue) {
+		final BasicPoint p = (aUseAsReturnValue != null) ? aUseAsReturnValue : new BasicPoint();
 		
 		final double aLon = aGeoPoint.getLongitudeE6()*1E-6;
 		final double aLat = aGeoPoint.getLatitudeE6()*1E-6; 
