diff --git a/old_test.java b/new_test.java
index 28e5086..9c3afdd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,7 +20,8 @@
                                                          Integer.valueOf(VALIDITY),
                                                          new DefaultSubjectDN(null, SERVER_NAME).getSubjectDN(),
                                                          DefaultSSLCertificateCreator.DEFAULT_SIZE,
-                                                         DefaultSSLCertificateCreator.SIGALG);
+                                                         DefaultSSLCertificateCreator.SIGALG,
+                                                         null);
             }
         });
 
