diff --git a/old_product.java b/new_product.java
index 1e42fa7..a5e2e37 100644
--- a/old_product.java
+++ b/new_product.java
@@ -17,11 +17,9 @@ public static String diffLines(List<String> lines1, List<String> lines2) {
           if (de > ds) {
             sw.append(",").append(String.valueOf(de));
           }
-          sw.append("d").append(String.valueOf(as - 1)).append(
-              TestUtil.NL);
+          sw.append("d").append(String.valueOf(as - 1)).append('\n');
           for (int i = ds - 1; i < de; ++i) {
-            sw.append("< ").append(lines1.get(i)).append(
-                TestUtil.NL);
+            sw.append("< ").append(lines1.get(i)).append('\n');
           }
         }
       } else {
@@ -32,10 +30,9 @@ public static String diffLines(List<String> lines1, List<String> lines2) {
           if (ae > as) {
             sw.append(",").append(String.valueOf(ae));
           }
-          sw.append(TestUtil.NL);
+          sw.append('\n');
           for (int i = as - 1; i < ae; ++i) {
-            sw.append("> ").append(lines2.get(i)).append(
-                TestUtil.NL);
+            sw.append("> ").append(lines2.get(i)).append('\n');
           }
         } else {
           // a change: "<ds>,<de>c<as>,<ae>
@@ -47,15 +44,13 @@ public static String diffLines(List<String> lines1, List<String> lines2) {
           if (ae > as) {
             sw.append(",").append(String.valueOf(ae));
           }
-          sw.append(TestUtil.NL);
+          sw.append('\n');
           for (int i = ds - 1; i < de; ++i) {
-            sw.append("< ").append(lines1.get(i)).append(
-                TestUtil.NL);
+            sw.append("< ").append(lines1.get(i)).append('\n');
           }
-          sw.append("---").append(TestUtil.NL);
+          sw.append("---\n");
           for (int i = as - 1; i < ae; ++i) {
-            sw.append("> ").append(lines2.get(i)).append(
-                TestUtil.NL);
+            sw.append("> ").append(lines2.get(i)).append('\n');
           }
           offset = offset + (ae - as) - (de - ds);
         }
