diff --git a/old_test.java b/new_test.java
index 8a7c22c..6e269d3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,9 +18,8 @@
         DiffTestCase.diffLines(
             Arrays.asList(before),
             Arrays.asList(after));
-    assertEquals(
-        diff,
-        TestUtil.fold(
+    assertThat(Util.toLinux(diff),
+        equalTo(
             "1a2\n"
             + "> (they call her \"Polythene Pam\")\n"
             + "3c4,5\n"
