diff --git a/old_test.java b/new_test.java
index 8effa55..f184dc7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
-    public void testNpara() {
-        System.out.println("npara");
+    public void testLength() {
+        System.out.println("length");
         GeometricDistribution instance = new GeometricDistribution(0.3);
         instance.rand();
-        assertEquals(1, instance.npara());
+        assertEquals(1, instance.length());
     }
\ No newline at end of file
