diff --git a/old_test.java b/new_test.java
index 420d06a..ef28d37 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
     public void testGetThrowableList_Throwable_null() {
-        List<?> throwables = ExceptionUtils.getThrowableList(null);
+        final List<?> throwables = ExceptionUtils.getThrowableList(null);
         assertEquals(0, throwables.size());
     }
\ No newline at end of file
