diff --git a/old_product.java b/new_product.java
index a31570e..590fbb8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 private Item assrt(final QueryContext ctx) throws QueryException {
     final byte[] str = expr.length < 2 ? null : checkStr(expr[1], ctx);
     if(expr[0].ebv(ctx, info).bool(info)) return null;
-    throw str == null ? BXUN_ASSERT.thrw(info) : BXUN_ERROR.thrw(info, str);
+    throw str == null ? UNIT_ASSERT.thrw(info) : UNIT_MESSAGE.thrw(info, str);
   }
\ No newline at end of file
