diff --git a/old_test.java b/new_test.java
index 18028cc..74773f7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,6 @@
   public void assrt() {
     query(_XQUNIT_ASSERT.args("1"), "");
     query(_XQUNIT_ASSERT.args("(<a/>,<b/>)"), "");
-    error(_XQUNIT_ASSERT.args("()"), Err.BXUN_ASSERT);
-    error(_XQUNIT_ASSERT.args("()", "X"), Err.BXUN_ERROR);
+    error(_XQUNIT_ASSERT.args("()"), Err.UNIT_ASSERT);
+    error(_XQUNIT_ASSERT.args("()", "X"), Err.UNIT_MESSAGE);
   }
\ No newline at end of file
