diff --git a/old_test.java b/new_test.java
index 4f068bf..0004361 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,11 @@
     ok(new Replace(FN, "<a/>"));
     ok(new Replace(FN, "<a/>"));
     no(new Replace(FN, ""));
+    // create binary file
+    ok(new XQuery("db:put('" + NAME + "', 'a', 'a')"));
+    ok(new Replace("a", "<b/>"));
+    assertTrue(!ok(new XQuery("db:open('" + NAME + "')")).isEmpty());
+    ok(new XQuery("db:get('" + NAME + "', 'a')"));
     // a failing replace should not remove existing documents
     no(new Replace(FN, "<a>"));
     assertTrue(!ok(new XQuery("doc('" + NAME + "')")).isEmpty());
