diff --git a/old_product.java b/new_product.java
index 5284ed3..2aa6ac1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -99,7 +99,8 @@ public static void main(String[] args) throws Exception {
       if (classificationMethod.equalsIgnoreCase("sequential")) {
         runFPGrowth(params);
       } else if (classificationMethod.equalsIgnoreCase("mapreduce")) {
-        HadoopUtil.overwriteOutput(outputDir);
+        Configuration conf = new Configuration();
+        HadoopUtil.delete(conf, outputDir);
         PFPGrowth.runPFPGrowth(params);
       }
     } catch (OptionException e) {
