diff --git a/old_product.java b/new_product.java
index 4db41b7..a6001cc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -154,5 +154,12 @@ public void validate(Object obj, Errors errors) throws APIException, DuplicateCo
 				index++;
 			}
 		}
+		if (CollectionUtils.isNotEmpty(conceptToValidate.getAnswers())) {
+			for (ConceptAnswer conceptAnswer : conceptToValidate.getAnswers()) {
+				if (conceptAnswer.getAnswerConcept().equals(conceptToValidate)) {
+					errors.reject("Concept.contains.itself.as.answer");
+				}
+			}
+		}
 		ValidateUtil.validateFieldLengths(errors, obj.getClass(), "version", "retireReason");
 	}
\ No newline at end of file
