diff --git a/old_test.java b/new_test.java
index 9997638..8b98512 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,6 +11,7 @@
 		
 		concept.addName(conceptFullySpecifiedName);
 		concept.addName(conceptShortName);
+		concept.addDescription(new ConceptDescription("some description",null));
 		
 		Errors errors = new BindException(concept, "concept");
 		new ConceptValidator().validate(concept, errors);
