diff --git a/old_product.java b/new_product.java
index eb0fe7d..7290a5a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
-Flux<ResourceResponse<DocumentCollection>> createCollection(String databaseLink, DocumentCollection collection,
+Mono<ResourceResponse<DocumentCollection>> createCollection(String databaseLink, DocumentCollection collection,
                                                                 RequestOptions options);
\ No newline at end of file
