diff --git a/old_test.java b/new_test.java
index 0eff9b5..33a62c7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups = "samples", timeOut = TIMEOUT)
     public void createCollection_toBlocking() {
-        Flux<ResourceResponse<DocumentCollection>> createCollectionObservable = client
+        Mono<ResourceResponse<DocumentCollection>> createCollectionObservable = client
                 .createCollection(getDatabaseLink(), collectionDefinition, null);
 
         // single() converts the flux to a mono.
