diff --git a/old_test.java b/new_test.java
index 5e77156..0487d3b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,20 @@
 @Test
   public void readEOFReturnsNegativeTest() {
     Whitebox.setInternalState(UnderFileSystemCluster.class, "sUnderFSClass",
-            (String) null);
-    Assert.assertFalse(UnderFileSystemCluster.readEOFReturnsNegative());
+        (String) null);
+    String underFSClass = UnderFileSystemCluster.getUnderFSClass();
+    Assert.assertFalse(underFSClass != null
+        && underFSClass.equals("alluxio.underfs.hdfs.LocalMiniDFSCluster"));
 
     Whitebox.setInternalState(UnderFileSystemCluster.class, "sUnderFSClass",
         "XXXX");
-    Assert.assertFalse(UnderFileSystemCluster.readEOFReturnsNegative());
+    underFSClass = UnderFileSystemCluster.getUnderFSClass();
+    Assert.assertFalse(underFSClass != null
+        && underFSClass.equals("alluxio.underfs.hdfs.LocalMiniDFSCluster"));
 
     Whitebox.setInternalState(UnderFileSystemCluster.class, "sUnderFSClass",
         "alluxio.underfs.hdfs.LocalMiniDFSCluster");
-    Assert.assertTrue(UnderFileSystemCluster.readEOFReturnsNegative());
+    underFSClass = UnderFileSystemCluster.getUnderFSClass();
+    Assert.assertTrue(underFSClass != null
+        && underFSClass.equals("alluxio.underfs.hdfs.LocalMiniDFSCluster"));
   }
\ No newline at end of file
