diff --git a/old_product.java b/new_product.java
index b7cc51a..e45e2a6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,3 @@
-public static boolean readEOFReturnsNegative() {
-    // TODO(hy): Should be dynamically determined - may need additional method on UnderFileSystem.
-    return sUnderFSClass != null
-        && sUnderFSClass.equals("alluxio.underfs.hdfs.LocalMiniDFSCluster");
+public static synchronized String getUnderFSClass() {
+    return sUnderFSClass;
   }
\ No newline at end of file
