diff --git a/old_test.java b/new_test.java
index fdeedcc..1a01e54 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
         expect(expectLastCall()).once();
         mockMessageService.dispatchMessage(anyObject(), anyString());
         expect(expectLastCall()).once();
-        expect(mockCommunicator.openCommPort(ConnectionDriver.JSSC, port, portRate)).andReturn(true).once();
+        mockCommunicator.connect(ConnectionDriver.JSSC, port, portRate);
         replay(instance, mockCommunicator, mockListener);
 
         Boolean expResult = true;
