diff --git a/old_test.java b/new_test.java
index 88783cc..21c791a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,16 +6,16 @@
     testOneAddGet(false, "b", "q", "c", "z", "a", "1", null, null);
 
     // clear account a, app p
-    mds.clear("a", "p");
-    Assert.assertNull(mds.get("a", "p", "a", "x"));
-    Assert.assertNotNull(mds.get("a", "q", "b", "y"));
-    Assert.assertNotNull(mds.get("a", null, "c", "z"));
+    mds.clear(context, "a", "p");
+    Assert.assertNull(mds.get(context, "a", "p", "a", "x"));
+    Assert.assertNotNull(mds.get(context, "a", "q", "b", "y"));
+    Assert.assertNotNull(mds.get(context, "a", null, "c", "z"));
 
     // clear all for account a
-    mds.clear("a", null);
-    Assert.assertNull(mds.get("a", "q", "b", "y"));
-    Assert.assertNull(mds.get("a", null, "c", "z"));
+    mds.clear(context, "a", null);
+    Assert.assertNull(mds.get(context, "a", "q", "b", "y"));
+    Assert.assertNull(mds.get(context, "a", null, "c", "z"));
 
     // make sure account b is still there
-    Assert.assertNotNull(mds.get("b", "q", "c", "z"));
+    Assert.assertNotNull(mds.get(context, "b", "q", "c", "z"));
   }
\ No newline at end of file
