diff --git a/old_test.java b/new_test.java
index 0933733..9463e66 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,6 @@
         Assert.assertNull(this.job.getCommandCriteriaString());
         //Simulate the call stack JPA will make on persist
         this.job.onCreateAuditable();
-        this.job.onCreateOrUpdateCommonEntityFields();
         this.job.onCreateOrUpdateJob();
         Assert.assertNotNull(this.job.getId());
         Assert.assertNotNull(this.job.getClusterCriteriasString());
