diff --git a/old_product.java b/new_product.java
index 4219b1f..1dab86d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 private ANode entries(final QueryContext ctx) throws QueryException {
-    final Uri uri = (Uri) checkType(expr[0].item(ctx, input), Type.URI);
-    final String file = IOFile.file(string(uri.atom()));
+    //final Uri uri = (Uri) checkType(expr[0].item(ctx, input), Type.URI);
+    final String file = IOFile.file(string(checkStr(expr[0], ctx)));
     if(!IO.get(file).exists()) ZIPNOTFOUND.thrw(input, file);
 
     // traverse all zip entries and create map (zip entries are not sorted)
