diff --git a/old_test.java b/new_test.java
index c6ad71f..023154d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
   public void testEntries() throws BaseXException {
-    final String fun = check(FunDef.ENTRIES, Uri.class);
-    query(fun + "(xs:anyURI('" + ZIP + "'))");
+    final String fun = check(FunDef.ENTRIES, String.class);
+    query(fun + "('" + ZIP + "')");
   }
\ No newline at end of file
