diff --git a/old_test.java b/new_test.java
index 9a9996a..1931586 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,8 @@
     PropertySet base = PropertySet.of(ImmutableListMultimap.of("a", "x", "a", "y", "b", "y", "c", "z"));
     PropertySet other = PropertySet.of(ImmutableListMultimap.of("a", "aa", "c", "cc", "d", "dd", "e", "ee"));
     PropertySet expected =
-        PropertySet.of(ImmutableListMultimap.of("a", "aa", "b", "y", "c", "cc", "d", "dd", "e", "ee"));
-    assertThat(base.overrideWith(other)).isEqualTo(expected);
+        PropertySet.of(ImmutableListMultimap.of("a", "aa", "c", "cc", "d", "dd", "e", "ee", "b", "y"));
+    PropertySet actual = base.overrideWith(other);
+    assertThat(actual).isEqualTo(expected);
+    assertThat(actual.asMultimap().entries().asList()).isEqualTo(expected.asMultimap().entries().asList());
   }
\ No newline at end of file
