diff --git a/old_test.java b/new_test.java
index d672938..e13b1e3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,13 +4,18 @@
         val count = new AtomicInteger();
         val wasInvoked = new Semaphore(0);
         val waitOn = new CompletableFuture<Void>();
+        val retry = Retry.withExpBackoff(1, 2, 3)
+                         .retryWhen(t -> true)
+                         .throwingOn(Exception.class);
+        val error = new AtomicReference<Throwable>();
         val p = new SequentialAsyncProcessor(
                 () -> {
                     count.incrementAndGet();
                     wasInvoked.release();
                     waitOn.join();
                 },
-                (ex, c) -> true,
+                retry,
+                error::set,
                 executorService());
 
         // Invoke it a number of times.
