diff --git a/old_test.java b/new_test.java
index dc2ab5b..85e169f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,11 +5,11 @@
 	    MockWebRequest request = new MockWebRequest(url);
 	    int score = encoder.getCompatibilityScore(request);
 	    
-	    assertEquals(2, score);
+	    assertEquals(4, score);
 	    
 	    url = Url.parse(MOUNT_PATH + "/foo/bar/" + "MyPage");
 	    request = new MockWebRequest(url);
 	    score = namedParametersEncoder.getCompatibilityScore(request);
 	    
-	    assertEquals(4, score);
+	    assertEquals(6, score);
 	}
\ No newline at end of file
