diff --git a/old_product.java b/new_product.java
index 8295e81..3f332ab 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,20 @@
 @Override
-	public abstract int getCompatibilityScore(Request request);
\ No newline at end of file
+	public int getCompatibilityScore(Request request)
+	{
+		if (urlStartsWith(request.getUrl(), mountSegments))
+		{
+			/* see WICKET-5056 - alter score with pathSegment type */
+			int countOptional = 0;
+			int fixedSegments = 0;
+			for (MountPathSegment pathSegment : pathSegments)
+			{
+				fixedSegments += pathSegment.getFixedPartSize();
+				countOptional += pathSegment.getOptionalParameters();
+			}
+			return mountSegments.length - countOptional + fixedSegments;
+		}
+		else
+		{
+			return 0;
+		}
+	}
\ No newline at end of file
