diff --git a/old_product.java b/new_product.java
index ae697c6..8a93d43 100644
--- a/old_product.java
+++ b/new_product.java
@@ -80,9 +80,7 @@ public CreatePathResult createPath(LockedInodePath inodePath, CreatePathOptions<
         .setMountPoint(false)
         .setPersisted(options.isPersisted())
         .setOwner(options.getOwner())
-        .setGroup(options.getGroup())
-        .setMode(options.getMode())
-        .setDefaultMode(true);
+        .setGroup(options.getGroup());
     for (int k = pathIndex; k < (pathComponents.length - 1); k++) {
       InodeDirectory dir =
           InodeDirectory.create(mDirectoryIdGenerator.getNewDirectoryId(),
