diff --git a/old_product.java b/new_product.java
index 3ef5505..85539fe 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,7 +7,7 @@
 			@NonNull final BigDecimal delta)
 	{
 		final CandidatesQuery segment = CandidatesQuery.builder()
-				.type(Type.STOCK)
+				.type(CandidateType.STOCK)
 				.materialDescr(MaterialDescriptor.builderForQuery()
 						.date(date)
 						.productDescriptor(productDescriptor)
