diff --git a/old_product.java b/new_product.java
index 345e607..86e232c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public static <T> Key<T> get(TypeLiteral<T> typeLiteral,
                                  Class<? extends Annotation> annotationType) {
-        return new Key<T>(typeLiteral, strategyFor(annotationType));
+        return new Key<>(typeLiteral, strategyFor(annotationType));
     }
\ No newline at end of file
