diff --git a/old_product.java b/new_product.java
index 1d18d3d..e8add97 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,29 +14,35 @@ private boolean negotiate(final Request request, final HttpServletResponse respo
             this.auth.resetSecurityToken(connectionId);
         }
 
+        final byte[] tokenBuffer = authorizationHeader.getTokenBytes();
+        this.log.debug("token buffer: {} byte(s)", Integer.valueOf(tokenBuffer.length));
+
         // log the user in using the token
         IWindowsSecurityContext securityContext;
-
         try {
-            final byte[] tokenBuffer = authorizationHeader.getTokenBytes();
-            this.log.debug("token buffer: {} byte(s)", Integer.valueOf(tokenBuffer.length));
             securityContext = this.auth.acceptSecurityToken(connectionId, tokenBuffer, securityPackage);
-            this.log.debug("continue required: {}", Boolean.valueOf(securityContext.isContinue()));
+        } catch (final Win32Exception e) {
+            this.log.warn("error logging in user: {}", e.getMessage());
+            this.log.trace("", e);
+            this.sendUnauthorized(response);
+            return false;
+        }
+        this.log.debug("continue required: {}", Boolean.valueOf(securityContext.isContinue()));
 
-            final byte[] continueTokenBytes = securityContext.getToken();
-            if (continueTokenBytes != null && continueTokenBytes.length > 0) {
-                final String continueToken = BaseEncoding.base64().encode(continueTokenBytes);
-                this.log.debug("continue token: {}", continueToken);
-                response.addHeader("WWW-Authenticate", securityPackage + " " + continueToken);
-            }
+        final byte[] continueTokenBytes = securityContext.getToken();
+        if (continueTokenBytes != null && continueTokenBytes.length > 0) {
+            final String continueToken = BaseEncoding.base64().encode(continueTokenBytes);
+            this.log.debug("continue token: {}", continueToken);
+            response.addHeader("WWW-Authenticate", securityPackage + " " + continueToken);
+        }
 
+        try {
             if (securityContext.isContinue() || ntlmPost) {
                 response.setHeader("Connection", "keep-alive");
                 response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
                 response.flushBuffer();
                 return false;
             }
-
         } catch (final IOException e) {
             this.log.warn("error logging in user: {}", e.getMessage());
             this.log.trace("", e);
