diff --git a/old_test.java b/new_test.java
index 886d870..52b0d77 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,7 +17,7 @@
             boolean authenticated = false;
             final SimpleHttpRequest request = new SimpleHttpRequest();
             while (true) {
-                final String clientToken = BaseEncoding.base64().encode(clientContext.getToken());
+                final String clientToken = Base64.getEncoder().encodeToString(clientContext.getToken());
                 request.addHeader("Authorization", securityPackage + " " + clientToken);
 
                 final SimpleHttpResponse response = new SimpleHttpResponse();
@@ -41,7 +41,7 @@
                 Assert.assertEquals(401, response.getStatus());
                 final String continueToken = response.getHeader("WWW-Authenticate").substring(
                         securityPackage.length() + 1);
-                final byte[] continueTokenBytes = BaseEncoding.base64().decode(continueToken);
+                final byte[] continueTokenBytes = Base64.getDecoder().decode(continueToken);
                 Assertions.assertThat(continueTokenBytes.length).isGreaterThan(0);
                 final SecBufferDesc continueTokenBuffer = new SecBufferDesc(Sspi.SECBUFFER_TOKEN, continueTokenBytes);
                 clientContext.initialize(clientContext.getHandle(), continueTokenBuffer,
