diff --git a/old_test.java b/new_test.java
index ec61efb..b2ae853 100644
--- a/old_test.java
+++ b/new_test.java
@@ -19,5 +19,5 @@
             (k, v) -> v
         );
 
-        assertArrayEquals(new double[] {0.0, 2.0}, preprocessor.apply(7, new String[] {"Monday", "September"}), 1e-8);
+        assertArrayEquals(new double[] {0.0, 2.0}, preprocessor.apply(7, new String[] {"Monday", "September"}).asArray(), 1e-8);
     }
\ No newline at end of file
