diff --git a/old_product.java b/new_product.java
index b569526..2684c83 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
 public void addCheckpoint(long userId, int fileId) throws TException, IOException {
     // TODO This part needs to be changed.
-    String srcPath = CommonUtils.concatPath(getUserUfsTmpFolder(userId), fileId);
+    String srcPath = PathUtils.concatPath(getUserUfsTmpFolder(userId), fileId);
     String ufsDataFolder =
         mTachyonConf.get(Constants.UNDERFS_DATA_FOLDER, Constants.DEFAULT_DATA_FOLDER);
-    String dstPath = CommonUtils.concatPath(ufsDataFolder, fileId);
+    String dstPath = PathUtils.concatPath(ufsDataFolder, fileId);
     try {
       if (!mUfs.rename(srcPath, dstPath)) {
         throw new FailedToCheckpointException("Failed to rename " + srcPath + " to " + dstPath);
