diff --git a/old_test.java b/new_test.java
index d94146f..3d58dfd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
     String tmpFolder = mWorkerServiceHandler.getUserUfsTempFolder(USER_ID);
     UnderFileSystem ufs = UnderFileSystem.get(tmpFolder, mMasterTachyonConf);
     ufs.mkdirs(tmpFolder, true);
-    String filename = CommonUtils.concatPath(tmpFolder, fileId);
+    String filename = PathUtils.concatPath(tmpFolder, fileId);
     OutputStream out = ufs.create(filename);
     out.write(TestUtils.getIncreasingByteArray(blockSize));
     out.close();
