diff --git a/old_test.java b/new_test.java
index 9f4d603..3d5f8fd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
     public void test_firstDayOfYear_serialization() throws IOException, ClassNotFoundException {
-        DateTimeAdjuster firstDayOfYear = DateTimeAdjusters.firstDayOfYear();
+        WithAdjuster firstDayOfYear = DateTimeAdjusters.firstDayOfYear();
         assertTrue(firstDayOfYear instanceof Serializable);
         
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
