diff --git a/old_test.java b/new_test.java
index 08761a2..5643596 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,14 +5,9 @@
         System.out.println("before: " + instance.getSequenceString());
 
         Point point = new Point(11, 3);
-        instance.setStickyDistance(2);
         assertEquals(p1, instance.findSegment(point));
 
-        instance.setStickyDistance(0);
-        assertEquals(null, instance.findSegment(point));
-
         instance = new BrokenLine();
-        instance.setStickyDistance(2);
         assertEquals(null, instance.findSegment(point));
         instance.addPoint(p1);
         assertEquals(null, instance.findSegment(point));
