diff --git a/old_product.java b/new_product.java
index b4c3a64..ae0c10c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
 public Point findSegment (Point point)
     {
+        final int sqrStickyDistance = getStickyDistance() * getStickyDistance();
         Point bestPoint = null;
         double bestDistSq = java.lang.Double.MAX_VALUE;
 
@@ -26,7 +27,7 @@ public Point findSegment (Point point)
             prevPt = pt;
         }
 
-        if (bestDistSq <= (stickyDistance * stickyDistance)) {
+        if (bestDistSq <= sqrStickyDistance) {
             return bestPoint;
         } else {
             return null;
