diff --git a/old_test.java b/new_test.java
index 5547081..6fde1df 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,6 @@
     public void startNewSpan() {
         brave.serverTracer().setStateCurrentTrace(PARENT_CONTEXT, "name");
 
-        PowerMockito.when(System.nanoTime()).thenReturn(500L);
-
         SpanId newContext = brave.localTracer().startNewSpan(COMPONENT_NAME, OPERATION_NAME);
         assertThat(newContext).isEqualTo(
             PARENT_CONTEXT.toBuilder()
