diff --git a/old_product.java b/new_product.java
index 2c7937d..8e6a051 100644
--- a/old_product.java
+++ b/new_product.java
@@ -49,14 +49,7 @@ public void configure(Properties props)
 		}
 		catch (NullPointerException e) {
 			throw new HibernateException(e);
-		} catch (ClassNotFoundException e) {
+		} catch (Exception e) {
 			throw new HibernateException(e);
-		} catch (InstantiationException e) {
-			throw new HibernateException(e);
-		} catch (IllegalAccessException e) {
-			throw new HibernateException(e);
-		} catch (ClassCastException e) {
-			throw new HibernateException(e);
-		}
-
+		} 
 	}
\ No newline at end of file
