diff --git a/old_product.java b/new_product.java
index 51d987d..5262fac 100644
--- a/old_product.java
+++ b/new_product.java
@@ -42,7 +42,7 @@ protected Object formBackingObject(HttpServletRequest request) throws ServletExc
 			String age = request.getParameter("addAge");
 			
 			p = new Patient();
-			UserFormController.getMiniPerson(p, name, gender, date, age);
+			PersonFormController.getMiniPerson(p, name, gender, date, age);
 			
 			patient = new ShortPatientModel(p);
 		}
