diff --git a/old_test.java b/new_test.java
index 70f77d1..fdea2ae 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,51 +1,51 @@
 @Test
-    public void testHexDigitToBools() {
+    public void testHexDigitToBoolArray() {
         assertBoolArrayEquals(
-            new boolean[]{false, false, false, false}, Conversion.hexDigitToBools('0'));
+            new boolean[]{false, false, false, false}, Conversion.hexDigitToBoolArray('0'));
         assertBoolArrayEquals(
-            new boolean[]{true, false, false, false}, Conversion.hexDigitToBools('1'));
+            new boolean[]{true, false, false, false}, Conversion.hexDigitToBoolArray('1'));
         assertBoolArrayEquals(
-            new boolean[]{false, true, false, false}, Conversion.hexDigitToBools('2'));
+            new boolean[]{false, true, false, false}, Conversion.hexDigitToBoolArray('2'));
         assertBoolArrayEquals(
-            new boolean[]{true, true, false, false}, Conversion.hexDigitToBools('3'));
+            new boolean[]{true, true, false, false}, Conversion.hexDigitToBoolArray('3'));
         assertBoolArrayEquals(
-            new boolean[]{false, false, true, false}, Conversion.hexDigitToBools('4'));
+            new boolean[]{false, false, true, false}, Conversion.hexDigitToBoolArray('4'));
         assertBoolArrayEquals(
-            new boolean[]{true, false, true, false}, Conversion.hexDigitToBools('5'));
+            new boolean[]{true, false, true, false}, Conversion.hexDigitToBoolArray('5'));
         assertBoolArrayEquals(
-            new boolean[]{false, true, true, false}, Conversion.hexDigitToBools('6'));
+            new boolean[]{false, true, true, false}, Conversion.hexDigitToBoolArray('6'));
         assertBoolArrayEquals(
-            new boolean[]{true, true, true, false}, Conversion.hexDigitToBools('7'));
+            new boolean[]{true, true, true, false}, Conversion.hexDigitToBoolArray('7'));
         assertBoolArrayEquals(
-            new boolean[]{false, false, false, true}, Conversion.hexDigitToBools('8'));
+            new boolean[]{false, false, false, true}, Conversion.hexDigitToBoolArray('8'));
         assertBoolArrayEquals(
-            new boolean[]{true, false, false, true}, Conversion.hexDigitToBools('9'));
+            new boolean[]{true, false, false, true}, Conversion.hexDigitToBoolArray('9'));
         assertBoolArrayEquals(
-            new boolean[]{false, true, false, true}, Conversion.hexDigitToBools('A'));
+            new boolean[]{false, true, false, true}, Conversion.hexDigitToBoolArray('A'));
         assertBoolArrayEquals(
-            new boolean[]{false, true, false, true}, Conversion.hexDigitToBools('a'));
+            new boolean[]{false, true, false, true}, Conversion.hexDigitToBoolArray('a'));
         assertBoolArrayEquals(
-            new boolean[]{true, true, false, true}, Conversion.hexDigitToBools('B'));
+            new boolean[]{true, true, false, true}, Conversion.hexDigitToBoolArray('B'));
         assertBoolArrayEquals(
-            new boolean[]{true, true, false, true}, Conversion.hexDigitToBools('b'));
+            new boolean[]{true, true, false, true}, Conversion.hexDigitToBoolArray('b'));
         assertBoolArrayEquals(
-            new boolean[]{false, false, true, true}, Conversion.hexDigitToBools('C'));
+            new boolean[]{false, false, true, true}, Conversion.hexDigitToBoolArray('C'));
         assertBoolArrayEquals(
-            new boolean[]{false, false, true, true}, Conversion.hexDigitToBools('c'));
+            new boolean[]{false, false, true, true}, Conversion.hexDigitToBoolArray('c'));
         assertBoolArrayEquals(
-            new boolean[]{true, false, true, true}, Conversion.hexDigitToBools('D'));
+            new boolean[]{true, false, true, true}, Conversion.hexDigitToBoolArray('D'));
         assertBoolArrayEquals(
-            new boolean[]{true, false, true, true}, Conversion.hexDigitToBools('d'));
+            new boolean[]{true, false, true, true}, Conversion.hexDigitToBoolArray('d'));
         assertBoolArrayEquals(
-            new boolean[]{false, true, true, true}, Conversion.hexDigitToBools('E'));
+            new boolean[]{false, true, true, true}, Conversion.hexDigitToBoolArray('E'));
         assertBoolArrayEquals(
-            new boolean[]{false, true, true, true}, Conversion.hexDigitToBools('e'));
+            new boolean[]{false, true, true, true}, Conversion.hexDigitToBoolArray('e'));
         assertBoolArrayEquals(
-            new boolean[]{true, true, true, true}, Conversion.hexDigitToBools('F'));
+            new boolean[]{true, true, true, true}, Conversion.hexDigitToBoolArray('F'));
         assertBoolArrayEquals(
-            new boolean[]{true, true, true, true}, Conversion.hexDigitToBools('f'));
+            new boolean[]{true, true, true, true}, Conversion.hexDigitToBoolArray('f'));
         try {
-            Conversion.hexDigitToBools('G');
+            Conversion.hexDigitToBoolArray('G');
             fail("Thrown " + IllegalArgumentException.class.getName() + " expected");
         } catch (final IllegalArgumentException e) {
             // OK
