diff --git a/old_product.java b/new_product.java
index d187e46..e98555f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@
         @NotBlank(message = "No command id entered. Unable to add applications.")
         final String id,
         @NotEmpty(message = "No application ids entered. Unable to add applications.")
-        final Set<String> applicationIds
+        final List<String> applicationIds
     ) throws GenieException {
         if (applicationIds.size() != applicationIds.stream().filter(this.appRepo::exists).count()) {
             throw new GeniePreconditionException("All applications need to exist to add to a command");
