diff --git a/old_test.java b/new_test.java
index 311b69e..c46f07e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testAddApplicationsForCommand() throws GenieException {
         Assert.assertTrue(this.service.getApplicationsForCommand(COMMAND_2_ID).isEmpty());
 
-        final Set<String> appIds = new HashSet<>();
+        final List<String> appIds = new ArrayList<>();
         appIds.add(APP_1_ID);
         final Set<Command> preCommands = this.appService.getCommandsForApplication(APP_1_ID, null);
         Assert.assertEquals(1, preCommands.size());
@@ -16,11 +16,5 @@
 
         final Set<Command> savedCommands = this.appService.getCommandsForApplication(APP_1_ID, null);
         Assert.assertEquals(2, savedCommands.size());
-        Assert.assertEquals(
-            1,
-            this.service.getApplicationsForCommand(COMMAND_2_ID)
-                .stream()
-                .filter(application -> APP_1_ID.equals(application.getId()))
-                .count()
-        );
+        Assert.assertThat(this.service.getApplicationsForCommand(COMMAND_2_ID).get(0).getId(), Matchers.is(APP_1_ID));
     }
\ No newline at end of file
