diff --git a/old_test.java b/new_test.java
index c3c82a7..bee7c96 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,5 +8,5 @@
 		encounter.setVisit(visit);
 		Errors errors = new BindException(encounter, "encounter");
 		new EncounterValidator().validate(encounter, errors);
-		Assert.assertTrue(errors.hasFieldErrors("visit"));
+		Assert.assertEquals("Encounter.visit.patients.dontMatch", errors.getFieldError("visit").getCode());
 	}
\ No newline at end of file
