diff --git a/old_product.java b/new_product.java
index 177600d..4d36cc3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,7 +9,9 @@ public void validate(Object obj, Errors errors) throws APIException {
 		
 		if (encounter != null) {
 			ValidationUtils.rejectIfEmpty(errors, "patient", "Encounter.error.patient.required", "Patient is required");
-			if (encounter.getVisit() != null && !encounter.getVisit().getPatient().equals(encounter.getPatient())) {
+            ValidationUtils.rejectIfEmpty(errors, "encounterType", "Encounter.error.encounterType.required",
+                    "Encounter type is Required");
+            if (encounter.getVisit() != null && !encounter.getVisit().getPatient().equals(encounter.getPatient())) {
 				errors.rejectValue("visit", "Encounter.visit.patients.dontMatch",
 				    "The patient for the encounter and visit should be the same");
 			}
