diff --git a/old_product.java b/new_product.java
index 7f61457..636845f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-TestItem iter() throws IOException {
-      final byte[] f = ((TestSession) cs).exec(ServerCmd.RESULTS, id);
-      final TestItem ti = new TestItem();
-      ti.type = f[0];
-      ti.result = Arrays.copyOfRange(f, 1, f.length);
-      return ti;
+TestResult iter() throws IOException {
+      final byte[] result = ((TestSession) cs).exec(ServerCmd.RESULTS, id);
+      final TestResult tr = new TestResult();
+      tr.type = result[0];
+      tr.result = Arrays.copyOfRange(result, 1, result.length);
+      return tr;
     }
\ No newline at end of file
