diff --git a/old_test.java b/new_test.java
index f2fffdb..b960ce0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test
   public void testIter() throws IOException {
-    for(final Object[] t : TYPES) {
-      if(t.length < 2) continue;
-      try(final TestQuery tq = session.query(t[1].toString())) {
-        final TestItem ti = tq.iter();
-        assertSame("Types are different.\nExpected: " + t[0] +
-            "\nFound: " + TYPES[ti.type][0], t, TYPES[ti.type]);
-        assertEquals(Token.string(ti.result), TYPES[ti.type][2]);
+    for(final Object[] exp : TYPES) {
+      if(exp.length < 2) continue;
+      try(final TestQuery tq = session.query(exp[1].toString())) {
+        final TestResult tr = tq.iter();
+        final Object[] type = TYPES[tr.type];
+        assertSame("Types are different.\nExpected: " + exp[0] + "\nFound: " + type[0], exp, type);
+        assertEquals(Token.string(tr.result), type[2]);
       }
     }
   }
\ No newline at end of file
