diff --git a/old_product.java b/new_product.java
index aeaddfb..69cb9c7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,8 @@
-public Cohort removePatientFromCohort(Cohort cohort, Patient patient) {
+@Override
+	public Cohort removePatientFromCohort(Cohort cohort, Patient patient) {
 		if (cohort.contains(patient.getPatientId())) {
-			cohort.removeMember(patient);
-			Context.getCohortService().saveCohort(cohort);
+			CohortMembership membership = cohort.getActiveMembership(patient);
+			Context.getCohortService().removeMembershipFromCohort(cohort, membership);
 		}
 		return cohort;
 	}
\ No newline at end of file
