diff --git a/old_product.java b/new_product.java
index 25b7998..d5a15ce 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-private byte[] number() throws QueryException {
+private byte[] number() throws QueryIOException {
     tb.reset();
 
     // integral part
@@ -6,13 +6,13 @@ private byte[] number() throws QueryException {
     tb.addByte((byte) c);
     if(c == '-') {
       c = consume();
-      if(c < '0' || c > '9') throw error("Number expected after '-'.");
+      if(c < '0' || c > '9') throw error("Number expected after '-'");
       tb.addByte((byte) c);
     }
 
     final boolean zero = c == '0';
     c = curr();
-    if(zero && c >= '0' && c <= '9') throw error("No digit allowed after '0'.");
+    if(zero && c >= '0' && c <= '9') throw error("No digit allowed after '0'");
     loop: while(true) {
       switch(c) {
         case '0':
@@ -42,7 +42,7 @@ private byte[] number() throws QueryException {
     if(consume('.')) {
       tb.addByte((byte) '.');
       c = curr();
-      if(c < '0' || c > '9') throw error("Number expected after '.'.");
+      if(c < '0' || c > '9') throw error("Number expected after '.'");
       do {
         tb.addByte((byte) c);
         pos++;
@@ -62,7 +62,7 @@ private byte[] number() throws QueryException {
       c = curr();
     }
 
-    if(c < '0' || c > '9') throw error("Exponent expected.");
+    if(c < '0' || c > '9') throw error("Exponent expected");
     do tb.addByte((byte) consume());
     while((c = curr()) >= '0' && c <= '9');
     skipWs();
