diff --git a/old_product.java b/new_product.java
index 1a6d29f..e235e74 100644
--- a/old_product.java
+++ b/new_product.java
@@ -33,5 +33,6 @@ public void validate(Object obj, Errors errors) {
 			}
 			validateFieldsForOutpatientCareSettingType(order, errors);
 			validatePairedFields(order, errors);
+			validateUnitsConceptClassIsUnitsOfMeasure(errors, order);
 		}
 	}
\ No newline at end of file
