diff --git a/old_test.java b/new_test.java
index 218e24b..2f94c6b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -19,14 +19,10 @@
 		order.setOrderType(Context.getOrderService().getOrderTypeByName("Drug order"));
 		order.setDrug(Context.getConceptService().getDrug(3));
 		order.setCareSetting(Context.getOrderService().getCareSetting(1));
-		double quantity = 2.00;
-		order.setQuantity(quantity);
-		
-		Concept quantityUnitsConcept = new Concept();
-		quantityUnitsConcept.setConceptId(101);
-		order.setQuantityUnits(quantityUnitsConcept);
-		Assert.assertTrue(order.getQuantityUnits().getConceptId().equals(quantityUnitsConcept.getConceptId()));
+		order.setQuantity(2.00);
+		order.setQuantityUnits(Context.getConceptService().getConcept(50));
 		order.setNumRefills(10);
+		
 		Errors errors = new BindException(order, "order");
 		new DrugOrderValidator().validate(order, errors);
 		Assert.assertFalse(errors.hasErrors());
