diff --git a/old_test.java b/new_test.java
index 417f7b6..89d829e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,11 @@
 @Test
     public void testSpatialRangeQuery() throws Exception {
-    	PointRDD spatialRDD = new PointRDD(sc, InputLocation, offset, splitter, true);
+    	PointRDD spatialRDD = new PointRDD(sc, InputLocation, offset, splitter, true,StorageLevel.MEMORY_ONLY());
     	for(int i=0;i<loopTimes;i++)
     	{
     		long resultSize = RangeQuery.SpatialRangeQuery(spatialRDD, queryEnvelope, false,false).count();
-    		assert resultSize>-1;
+    		System.out.println("testSpatialRangeQuery: "+resultSize);
+    		assert resultSize==3157;
     	}
     	assert RangeQuery.SpatialRangeQuery(spatialRDD, queryEnvelope, false,false).take(10).get(1).getUserData().toString()!=null;
         
