diff --git a/old_product.java b/new_product.java
index a02d0a2..62cc420 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,11 +5,11 @@ public StrBuilder appendFixedWidthPadLeft(final Object obj, final int width, fin
             if (str == null) {
                 str = "";
             }
-            int strLen = str.length();
+            final int strLen = str.length();
             if (strLen >= width) {
                 str.getChars(strLen - width, strLen, buffer, size);
             } else {
-                int padLen = width - strLen;
+                final int padLen = width - strLen;
                 for (int i = 0; i < padLen; i++) {
                     buffer[size + i] = padChar;
                 }
