diff --git a/old_product.java b/new_product.java
index 831b0e7..dcd27e2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,6 +5,5 @@
     ) throws GenieException {
         final ApplicationEntity applicationEntity = this.findApplication(id);
         final Set<TagEntity> newTags = this.createAndGetTagEntities(tags);
-        this.setFinalTags(newTags, applicationEntity.getUniqueId(), applicationEntity.getName());
         applicationEntity.setTags(newTags);
     }
\ No newline at end of file
