diff --git a/old_test.java b/new_test.java
index be7b377..82544e1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,8 +20,8 @@
       GetEntityOptions getOptions = new GetEntityOptions.Builder(workspaceId, entity).export(true).build();
       EntityExport response = service.getEntity(getOptions).execute();
       assertNotNull(response);
-      assertNotNull(response.getEntity());
-      assertEquals(response.getEntity(), entity);
+      assertNotNull(response.getEntityName());
+      assertEquals(response.getEntityName(), entity);
       assertNotNull(response.getDescription());
       assertEquals(response.getDescription(), entityDescription);
       assertNotNull(response.getValues());
@@ -36,7 +36,7 @@
 
       List<ValueExport> values = response.getValues();
       assertTrue(values.size() == 1);
-      assertEquals(values.get(0).getValue(), entityValue);
+      assertEquals(values.get(0).getValueText(), entityValue);
       assertTrue(fuzzyBefore(values.get(0).getCreated(), now));
       assertTrue(fuzzyAfter(values.get(0).getCreated(), start));
       assertTrue(fuzzyBefore(values.get(0).getUpdated(), now));
