diff --git a/old_test.java b/new_test.java
index 4517a3a..dfbb0de 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,7 @@
 @Test
-    public void testGetTemporaryFile()
+    public void testGetTemporaryFile() throws Exception
     {
         getContext().setDatabase("wiki");
+        createEmptyFile("temp/module/wiki/Space/Page/file.txt");
         Assert.assertNotNull(action.getTemporaryFile("/xwiki/bin/temp/Space/Page/module/file.txt", getContext()));
     }
\ No newline at end of file
