diff --git a/old_product.java b/new_product.java
index 8ca2ffa..f789adb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,9 +4,14 @@ protected File getTemporaryFile(String uri, XWikiContext context)
         File result = null;
         if (matcher.find()) {
             String wiki = context.getDatabase();
-            String space = matcher.group(1);
-            String page = matcher.group(2);
-            String module = matcher.group(3);
+            try {
+                wiki = URLEncoder.encode(wiki, URL_ENCODING);
+            } catch (UnsupportedEncodingException e) {
+                // This should never happen;
+            }
+            String space = withMinimalURLEncoding(matcher.group(1));
+            String page = withMinimalURLEncoding(matcher.group(2));
+            String module = withMinimalURLEncoding(matcher.group(3));
             String filePath = matcher.group(4);
             String prefix = String.format("temp/%s/%s/%s/%s/", module, wiki, space, page);
             String path = URI.create(prefix + filePath).normalize().toString();
