diff --git a/old_test.java b/new_test.java
index 79a57b6..21e3545 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,7 @@
 @Test
-    public void testGetCorpusRelationScores() {
-        Map<String,Object> params = new HashMap<String, Object>();
-        params.put(ConceptInsights.ACCOUNT_ID, "public");
-        params.put(ConceptInsights.CORPUS, "ibmresearcher");
-        List<String> concepts = new ArrayList<String>();
-        concepts.add("/graphs/wikipedia/en-20120601/concepts/IBM_Watson");
-        params.put(ConceptInsights.CONCEPTS, concepts);
-        params.put(ConceptInsights.CONCEPT, "IBM");
-        Scores scores = service.getCorpusRelationScores(params);
-        Assert.assertNotNull(scores);
-    }
\ No newline at end of file
+	public void testGetCorpusRelationScores() {
+		List<Concept> concepts = new ArrayList<Concept>();
+		concepts.add(EXAMPLE_CONCEPT);
+		Scores scores = service.getCorpusRelationScores(Corpus.IBM_RESEARCHERS, concepts);
+		Assert.assertNotNull(scores);
+	}
\ No newline at end of file
