diff --git a/old_product.java b/new_product.java
index 0c0eb58..7614ee8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -19,8 +19,12 @@ public static CandidatesQuery fromCandidate(
 		final DemandDetailsQuery demandDetailsQuery = DemandDetailsQuery
 				.ofDemandDetailOrNull(DemandDetail.castOrNull(candidate.getBusinessCaseDetail()));
 
+		final MaterialDescriptorQuery materialDescriptorQuery = MaterialDescriptorQuery.forDescriptor(
+				candidate.getMaterialDescriptor(),
+				DateAndSeqNo.ofCandidate(candidate));
+
 		final CandidatesQueryBuilder builder = CandidatesQuery.builder()
-				.materialDescriptorQuery(MaterialDescriptorQuery.forDescriptor(candidate.getMaterialDescriptor()))
+				.materialDescriptorQuery(materialDescriptorQuery)
 				.matchExactStorageAttributesKey(true)
 				.demandDetailsQuery(demandDetailsQuery)
 				.distributionDetailsQuery(distributionDetailsQuery)
