diff --git a/old_test.java b/new_test.java
index de3e2cc..5beb721 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,15 +1,10 @@
 @Test
 	public void fromCandidate()
 	{
-		final Timestamp date = new Timestamp(NOW.getTime());
-
 		final Candidate cand = Candidate.builder().type(CandidateType.STOCK)
-				.materialDescriptor(createMaterialDescriptor().withDate(date))
+				.materialDescriptor(createMaterialDescriptor().withDate(NOW))
 				.build();
 		final CandidatesQuery query = CandidatesQuery.fromCandidate(cand, false);
-
-		assertThat(query.getMaterialDescriptorQuery().getDate()).isEqualTo(date);
-		assertThat(query.getMaterialDescriptorQuery().getDateOperator()).isEqualTo(DateOperator.AT);
 		assertThat(query.getMaterialDescriptorQuery().getProductId()).isEqualTo(PRODUCT_ID);
 		assertThat(query.getMaterialDescriptorQuery().getStorageAttributesKey()).isEqualTo(STORAGE_ATTRIBUTES_KEY);
 		assertThat(query.getMaterialDescriptorQuery().getWarehouseId()).isEqualTo(WAREHOUSE_ID);
