diff --git a/old_test.java b/new_test.java
index 47c7e71..a7cca3c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,4 @@
 @Test
   public void isAbsolute() {
-    assertUriIsAbsolute("x:", true);
-
-    // absolute URIs always have schema
-    assertUriIsAbsolute("x", false);
-    assertUriIsAbsolute("", false);
-    // [DP] #928
-    //assertUriIsAbsolute("//localhost:80", false);
-
-    // absolute URIs don't have fragments
-    // [DP] #928
-    //assertUriIsAbsolute("http://localhost:80/html#f", false);
+    assertEquals("Uri absolute check failed", absolute, Uri.uri(uri).isAbsolute());
   }
\ No newline at end of file
