diff --git a/old_product.java b/new_product.java
index 5f97940..8f1e954 100644
--- a/old_product.java
+++ b/new_product.java
@@ -22,17 +22,26 @@ public String toPDB() {
         PDBHeader header = structure.getPDBHeader();
         header.toPDB(str);
 
-        for (SSBond ssbond : structure.getSSBonds()){
-        	ssbond.toPDB(str);
-        	str.append(newline);
+       
+        //REMARK 800
+        if (!structure.getSites().isEmpty()) {
+            str.append("REMARK 800                                                                      " + newline);
+            str.append("REMARK 800 SITE                                                                 " + newline);
+            for (Site site : structure.getSites()) {
+                site.remark800toPDB(str);
+            }
         }
-
+        //DBREF
         for (DBRef dbref : structure.getDBRefs()){
         	dbref.toPDB(str);
         	str.append(newline);
-
         }
-
+        //SSBOND
+        for (SSBond ssbond : structure.getSSBonds()){
+        	ssbond.toPDB(str);
+        	str.append(newline);
+        }
+        //SITE
         for (Site site : structure.getSites()) {
             site.toPDB(str);           
         }
