diff --git a/old_test.java b/new_test.java
index ece6fe2..b015f42 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,7 +15,7 @@
                 }
             }
 
-            try (GcodeStreamReader gsr = new GcodeStreamReader(f)) {
+            try (IGcodeStreamReader gsr = new GcodeStreamReader(f)) {
                 openInstanceExpectUtility(port, rate, false);
                 streamInstanceExpectUtility();
 
