diff --git a/old_product.java b/new_product.java
index 4c7860b..d6b91cd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,4 @@
-public static MonthDay from(Calendrical... calendricals) {
-        return CalendricalEngine.merge(calendricals).deriveChecked(rule());
+public static MonthDay from(CalendricalObject calendrical) {
+        LocalDate date = LocalDate.from(calendrical);
+        return MonthDay.of(date.getMonthOfYear(), date.getDayOfMonth());
     }
\ No newline at end of file
