diff --git a/old_product.java b/new_product.java
index 7df542d..382b983 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @VisibleForTesting
-    SortedSet<File> getDeadWorkerDirs(int nowSecs, Set<File> logDirs) throws Exception {
+    SortedSet<Path> getDeadWorkerDirs(int nowSecs, Set<Path> logDirs) throws Exception {
         if (logDirs.isEmpty()) {
             return new TreeSet<>();
         } else {
