diff --git a/old_test.java b/new_test.java
index 7956a88..abc0e4d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,5 @@
 @Test
 	public void parsePersonName_shouldNotFailWhenEndingWithWhitespace() throws Exception {
-		Context.getPersonService().parsePersonName("John ");
+		PersonName pname = Context.getPersonService().parsePersonName("John ");
+		assertEquals("John", pname.getGivenName());
 	}
\ No newline at end of file
