diff --git a/old_product.java b/new_product.java
index 208d751..dd6fde8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,6 +2,5 @@
     public Token recreateTokenFromBytes(byte[] tokenBytes, String... removeAttributes) throws InvalidTokenException, TokenExpiredException {
         TokenFactory tokenFactory = ltpaConfig.getTokenFactory();
         Token token = tokenFactory.validateTokenBytes(tokenBytes, removeAttributes);
-        validateRecreatedToken(token);
         return token;
     }
\ No newline at end of file
