diff --git a/old_product.java b/new_product.java
index b6e69e5..e6ac4e0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,12 +9,12 @@
 
         // Parse the cookie
         final String user;
-        final List<String> groups;
+        final List<String> principalsClaim;
 
         try {
             final JwtClaims claims = consumer.processToClaims(cookie);
             user = claims.getSubject();
-            groups = claims.getStringListClaimValue(PRINCIPALS);
+            principalsClaim = claims.getStringListClaimValue(PRINCIPALS);
         } catch (final InvalidJwtException e) {
             log.debug("JWT cookie is invalid: ", e);
             throw new InvalidCookieException("JWT cookie is invalid: " + e);
@@ -29,6 +29,6 @@
 
         // Build the token array
         final Stream<String> userStream = Stream.of(user);
-        final Stream<String> groupStream = groups.stream();
+        final Stream<String> groupStream = principalsClaim.stream();
         return Stream.concat(userStream, groupStream).toArray(String[]::new);
     }
\ No newline at end of file
