diff --git a/old_product.java b/new_product.java
index ebe1fa4..755a354 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public boolean changeState(StateVertix nextState) {
 		if (nextState == null) {
-			// TODO Stefan; Throw nullpointer exception??
+			LOGGER.info("nextState given is null");
 			return false;
 		}
 		LOGGER.debug("AFTER: sm.current: " + currentState.getName() + " hold.current: "
