diff --git a/old_test.java b/new_test.java
index 031a660..80c83a2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,7 +12,7 @@
 		 * Add index.
 		 */
 		Eventable c = new Eventable(new Identification(How.xpath, "/bla"), EventType.click);
-		assertTrue(sm.update(c, state2, dummyBrowser, new CrawlSession(dummyBrowser)));
+		assertTrue(sm.update(c, state2, dummyBrowser, new CrawlSession(dummyFactory)));
 
 		/**
 		 * Name is correctly changed
