diff --git a/old_product.java b/new_product.java
index 13a4086..b559518 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,12 @@
 @RpcMethod
-	public Future<RegistrationResponse> registerTaskExecutor(
+	public CompletableFuture<RegistrationResponse> registerTaskExecutor(
 			final UUID resourceManagerLeaderId,
 			final String taskExecutorAddress,
 			final ResourceID taskExecutorResourceId,
 			final SlotReport slotReport) {
 
 		if (Objects.equals(leaderSessionId, resourceManagerLeaderId)) {
-			Future<TaskExecutorGateway> taskExecutorGatewayFuture = getRpcService().connect(taskExecutorAddress, TaskExecutorGateway.class);
+			CompletableFuture<TaskExecutorGateway> taskExecutorGatewayFuture = getRpcService().connect(taskExecutorAddress, TaskExecutorGateway.class);
 
 			return taskExecutorGatewayFuture.handleAsync(
 				(TaskExecutorGateway taskExecutorGateway, Throwable throwable) -> {
@@ -26,9 +26,9 @@
 					"not equal the received leader session ID  {}",
 				taskExecutorResourceId, taskExecutorAddress, leaderSessionId, resourceManagerLeaderId);
 
-			return FutureUtils.toFlinkFuture(CompletableFuture.<RegistrationResponse>completedFuture(
+			return CompletableFuture.completedFuture(
 				new RegistrationResponse.Decline("Discard registration because the leader id " +
 					resourceManagerLeaderId + " does not match the expected leader id " +
-					leaderSessionId + '.')));
+					leaderSessionId + '.'));
 		}
 	}
\ No newline at end of file
