diff --git a/old_test.java b/new_test.java
index 98d9555..4062b89 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,16 +10,16 @@
     Date start = new Date();
 
     CreateIntentOptions createOptions = new CreateIntentOptions.Builder().workspaceId(workspaceId).intent(intentName)
-        .description(intentDescription).examples(intentExamples).build();
+            .description(intentDescription).examples(intentExamples).build();
     service.createIntent(createOptions).execute();
 
     try {
       GetIntentOptions getOptions =
-          new GetIntentOptions.Builder().workspaceId(workspaceId).intent(intentName).export(true).build();
+              new GetIntentOptions.Builder().workspaceId(workspaceId).intent(intentName).export(true).build();
       IntentExport response = service.getIntent(getOptions).execute();
       assertNotNull(response);
-      assertNotNull(response.getIntent());
-      assertEquals(response.getIntent(), intentName);
+      assertNotNull(response.getIntentName());
+      assertEquals(response.getIntentName(), intentName);
       assertNotNull(response.getDescription());
       assertEquals(response.getDescription(), intentDescription);
       assertNotNull(response.getExamples());
@@ -34,7 +34,7 @@
 
       List<Example> examples = response.getExamples();
       assertTrue(examples.size() == 1);
-      assertEquals(examples.get(0).getText(), intentExample);
+      assertEquals(examples.get(0).getExampleText(), intentExample);
       assertTrue(fuzzyBefore(examples.get(0).getCreated(), now));
       assertTrue(fuzzyAfter(examples.get(0).getCreated(), start));
       assertTrue(fuzzyBefore(examples.get(0).getUpdated(), now));
