diff --git a/old_test.java b/new_test.java
index ebc4d21..3bc3eb4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,5 +8,5 @@
 
         final String articleId = articles.get(0).getString(Keys.OBJECT_ID);
 
-        Assert.assertNotNull(articleQueryService.getArticleContent(articleId));
+        Assert.assertNotNull(articleQueryService.getArticleContent(null, articleId));
     }
\ No newline at end of file
