diff --git a/old_product.java b/new_product.java
index c29e84f..359d0ff 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@
         if (Strings.isEmptyOrNull(articleId)) {
             return;
         }
-
+        
         final TextHTMLRenderer renderer = new TextHTMLRenderer();
 
         context.setRenderer(renderer);
@@ -13,7 +13,7 @@
         String content;
 
         try {
-            content = articleQueryService.getArticleContent(articleId);
+            content = articleQueryService.getArticleContent(request, articleId);
         } catch (final ServiceException e) {
             LOGGER.log(Level.ERROR, "Can not get article content", e);
             return;
