diff --git a/old_test.java b/new_test.java
index 5417d74..a92263e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,9 +2,13 @@
   public void createDirectory() throws Exception {
     AlluxioURI dir = new AlluxioURI("/dir");
     CreateDirectoryPOptions createDirectoryOptions = CreateDirectoryPOptions.getDefaultInstance();
-    doNothing().when(mFileSystemMasterClient).createDirectory(dir, createDirectoryOptions);
+    doNothing().when(mFileSystemMasterClient).createDirectory(dir,
+        FileSystemOptions.createDirectoryDefaults(mConf)
+            .toBuilder().mergeFrom(createDirectoryOptions).build());
     mFileSystem.createDirectory(dir, createDirectoryOptions);
-    verify(mFileSystemMasterClient).createDirectory(dir, createDirectoryOptions);
+    verify(mFileSystemMasterClient).createDirectory(dir,
+        FileSystemOptions.createDirectoryDefaults(mConf)
+            .toBuilder().mergeFrom(createDirectoryOptions).build());
 
     verifyFilesystemContextAcquiredAndReleased();
   }
\ No newline at end of file
