diff --git a/old_product.java b/new_product.java
index e8cfec3..3e69393 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,8 +6,7 @@
       final CreateDirectoryPOptions options) {
     return RestUtils.call((RestUtils.RestCallable<Void>) () -> {
       if (options == null) {
-        mFileSystem.createDirectory(new AlluxioURI(path),
-                CreateDirectoryPOptions.getDefaultInstance());
+        mFileSystem.createDirectory(new AlluxioURI(path));
       } else {
         mFileSystem.createDirectory(new AlluxioURI(path), options);
       }
