diff --git a/old_product.java b/new_product.java
index a6be8a8..fddcd1b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,7 @@ void connectNodes() {
 
 		for (int i = 0; i < nodes.size(); i++) {
 			SquareNode n = nodes.get(i);
+
 			double[] point = searchPoints.get(i);
 
 			// distance between center when viewed head on will be space + 0.5*2*width.
@@ -12,7 +13,7 @@ void connectNodes() {
 
 			// find it's neighbors
 			searchResults.reset();
-			search.findNearest(point, neighborDistance*neighborDistance, MAX_NEIGHBORS + 1, searchResults);
+			search.findNearest(point, neighborDistance*neighborDistance, maxNeighbors + 1, searchResults);
 
 			// try to attach it's closest neighbors
 			for (int j = 0; j < searchResults.size(); j++) {
