diff --git a/old_product.java b/new_product.java
index 7de5571..fd5b6f6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,6 +10,8 @@ public void createAndAddEvents(
 			for (final I_M_HU_Assignment huAssignment : huAssignments)
 			{
 				final int topLevelHuId = huAccessService.retrieveTopLevelHuId(huAssignment.getM_HU());
+				Check.errorIf(topLevelHuId <= 0, "topLevelHuId returned by HUAccessService.retrieveTopLevelHuId has to be >0, but is {}; huAssignment={}", topLevelHuId, huAssignment);
+
 				builder.topLevelHuId(topLevelHuId);
 				builder.eventTime(huAssignment.getUpdated().toInstant());
 
