diff --git a/old_test.java b/new_test.java
index 741dd91..a00350b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -24,7 +24,7 @@
                 try {
                     return super.clone();
                 } catch (CloneNotSupportedException e) {
-                    return null;
+                    throw new AssertionError(e); // android-changed
                 }
             }
         }
