diff --git a/old_product.java b/new_product.java
index f4592c8..a91fceb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
 @Authorized(PrivilegeConstants.MANAGE_ORDER_TYPES)
-	public List<OrderType> getOrderSubtypes(OrderType orderType, boolean includeRetired);
\ No newline at end of file
+	public List<OrderType> getSubtypes(OrderType orderType, boolean includeRetired);
\ No newline at end of file
