diff --git a/old_product.java b/new_product.java
index d20363e..87370e6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public ServiceCall<Classifier> updateClassifier(UpdateClassifierOptions updateClassifierOptions) {
     Validator.notNull(updateClassifierOptions, "updateClassifierOptions cannot be null");
-    Validator.isTrue((updateClassifierOptions.classNames() != null) || (updateClassifierOptions
+    Validator.isTrue((updateClassifierOptions.classNames().size() > 0) || (updateClassifierOptions
         .negativeExamples() != null),
         "At least one of classnamePositiveExamples or negativeExamples must be supplied.");
     RequestBuilder builder = RequestBuilder.post(String.format("/v3/classifiers/%s", updateClassifierOptions
@@ -8,14 +8,12 @@ public ServiceCall<Classifier> updateClassifier(UpdateClassifierOptions updateCl
     builder.query(VERSION, versionDate);
     MultipartBody.Builder multipartBuilder = new MultipartBody.Builder();
     multipartBuilder.setType(MultipartBody.FORM);
-    if (updateClassifierOptions.classNames() != null) {
-      // Classes
-      for (String className : updateClassifierOptions.classNames()) {
-        String dataName = className + "_positive_examples";
-        File positiveExamples = updateClassifierOptions.positiveExamplesByClassName(className);
-        RequestBody body = RequestUtils.fileBody(positiveExamples, "application/octet-stream");
-        multipartBuilder.addFormDataPart(dataName, positiveExamples.getName(), body);
-      }
+    // Classes
+    for (String className : updateClassifierOptions.classNames()) {
+      String dataName = className + "_positive_examples";
+      File positiveExamples = updateClassifierOptions.positiveExamplesByClassName(className);
+      RequestBody body = RequestUtils.fileBody(positiveExamples, "application/octet-stream");
+      multipartBuilder.addFormDataPart(dataName, positiveExamples.getName(), body);
     }
     if (updateClassifierOptions.negativeExamples() != null) {
       RequestBody negativeExamplesBody = RequestUtils.inputStreamBody(updateClassifierOptions.negativeExamples(),
