diff --git a/old_test.java b/new_test.java
index 862d595..d590aa7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,7 +17,7 @@
     // first request
     String path = String.format(PATH_CLASSIFIER, classifierId);
     RecordedRequest request = server.takeRequest();
-    path += "?" + VERSION_DATE + "=" + VisualRecognition.VERSION_DATE_2016_05_20 + "&api_key=" + API_KEY;
+    path += "?" + VERSION_DATE + "=2016-05-20&api_key=" + API_KEY;
 
     assertEquals(path, request.getPath());
     assertEquals("POST", request.getMethod());
