diff --git a/old_product.java b/new_product.java
index c9f9004..7611c3d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
 public StrBuilder replaceFirst(final String searchStr, final String replaceStr) {
-        int searchLen = (searchStr == null ? 0 : searchStr.length());
+        final int searchLen = (searchStr == null ? 0 : searchStr.length());
         if (searchLen > 0) {
-            int index = indexOf(searchStr, 0);
+            final int index = indexOf(searchStr, 0);
             if (index >= 0) {
-                int replaceLen = (replaceStr == null ? 0 : replaceStr.length());
+                final int replaceLen = (replaceStr == null ? 0 : replaceStr.length());
                 replaceImpl(index, index + searchLen, searchLen, replaceStr, replaceLen);
             }
         }
