diff --git a/old_test.java b/new_test.java
index debc191..2d468ab 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,5 +7,5 @@
 		visit.setStartDatetime(new Date());
 		Errors errors = new BindException(visit, "visit");
 		new VisitValidator().validate(visit, errors);
-		Assert.assertTrue(errors.hasFieldErrors("patient"));
+		assertTrue(errors.hasFieldErrors("patient"));
 	}
\ No newline at end of file
