diff --git a/old_product.java b/new_product.java
index d6d4fb6..05591ef 100644
--- a/old_product.java
+++ b/new_product.java
@@ -30,4 +30,13 @@
 		
 		// check attributes
 		super.validateAttributes(visit, errors, Context.getVisitService().getAllVisitAttributeTypes());
+		
+		// check start and end dates
+		if (disallowOverlappingVisits()) {
+			List<Visit> otherVisitList = Context.getVisitService().getVisitsByPatient(visit.getPatient());
+			for (Visit otherVisit : otherVisitList) {
+				validateStartDatetime(visit, otherVisit, errors);
+				validateStopDatetime(visit, otherVisit, errors);
+			}
+		}
 	}
\ No newline at end of file
