diff --git a/old_test.java b/new_test.java
index 7be2b00..5cf9643 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testSendChallengeInitiateNegotiate() {
+    public void testSendChallengeInitiateNegotiate() throws IOException {
 
         this.out = new byte[1];
         this.out[0] = -1;
@@ -11,8 +11,7 @@
 
         Assertions.assertEquals("keep-alive", this.response.headers.get("Connection"));
 
-        Assertions.assertEquals(HttpServletResponse.SC_UNAUTHORIZED, this.response.sc);
-        Assertions.assertEquals(0, this.response.errorCode);
+        Assertions.assertEquals(HttpServletResponse.SC_UNAUTHORIZED, this.response.errorCode);
 
         Assertions.assertFalse(this.response.isFlushed);
     }
\ No newline at end of file
