diff --git a/old_test.java b/new_test.java
index 3a06fcb..7c918be 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void exist_filled() {
         // When-Then
-        Assert.assertTrue(testedStore.exist("a"));
-        Assert.assertFalse(testedStore.exist("k"));
+        Assert.assertTrue(testedStore.existProperty("a"));
+        Assert.assertFalse(testedStore.existProperty("k"));
     }
\ No newline at end of file
