diff --git a/old_product.java b/new_product.java
index 7d7c251..a0c114a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -34,9 +34,9 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
         exParams.setExaminationWindowLength(6);
         trainingParameters.setTextExtractorParameters(exParams);
 
-        TextClassifier instance = MLBuilder.create(trainingParameters, dbName, conf);
+        TextClassifier instance = MLBuilder.create(trainingParameters, conf);
         instance.fit(dataset);
-        instance.save();
+        instance.save(dbName);
 
 
         ClassificationMetrics vm = instance.validate(dataset);
