diff --git a/old_test.java b/new_test.java
index a702d72..9ca164d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -28,9 +28,9 @@
         //feature selection configuration
         trainingParameters.setFeatureSelectorTrainingParameters(null);
 
-        Modeler instance = MLBuilder.create(trainingParameters, dbName, conf);
+        Modeler instance = MLBuilder.create(trainingParameters, conf);
         instance.fit(trainingData);
-        instance.save();
+        instance.save(dbName);
 
         instance.close();
 
