diff --git a/old_product.java b/new_product.java
index 50e7ed4..30924d9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -29,7 +29,6 @@
                     log.warn("createStream for {}/{} failed {}: ", scopeName, streamConfiguration.getStreamName(),
                              exception);
                     return Response.status(Status.INTERNAL_SERVER_ERROR).build();
-                }).thenApply(asyncResponse::resume);
-
-        LoggerHelpers.traceLeave(log, "createStream", traceId);
+                }).thenApply(asyncResponse::resume)
+                .thenAccept(x -> LoggerHelpers.traceLeave(log, "createStream", traceId));
     }
\ No newline at end of file
