diff --git a/old_product.java b/new_product.java
index c27324a..d326c08 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,7 @@
 public int lsr(TachyonURI path) throws IOException {
-    TachyonFS tachyonClient = createFS(path);
-    List<ClientFileInfo> files = tachyonClient.listStatus(path);
-    Collections.sort(files);
+    List<FileInfo> files = listStatusSortedByIncreasingCreationTime(path);
     String format = "%-10s%-25s%-15s%-5s%n";
-    for (ClientFileInfo file : files) {
+    for (FileInfo file : files) {
       String inMemory = "";
       if (!file.isFolder) {
         if (100 == file.inMemoryPercentage) {
