diff --git a/old_product.java b/new_product.java
index 0a975cd..f5c1fca 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public static ANode stripNS(final ANode node, final byte[] ns, final Context ctx) {
     if(node.type != NodeType.ELM) return node;
 
-    final MemData md = new MemData(ctx.prop);
+    final MemData md = new MemData(ctx.options);
     final DataBuilder db = new DataBuilder(md);
     db.build(node);
 
