diff --git a/old_product.java b/new_product.java
index a85b738..125eae2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -107,6 +107,24 @@ void cacheConfiguration(ClientCacheConfiguration cfg, BinaryOutputStream out, Pr
                 )
             );
 
+            if (ver.compareTo(V1_6_0) >= 0) {
+                itemWriter.accept(CfgItem.EXPIRE_POLICY, w -> {
+                    ExpiryPolicy expiryPlc = cfg.getExpiryPolicy();
+                    if (expiryPlc == null)
+                        w.writeBoolean(false);
+                    else {
+                        w.writeBoolean(true);
+                        w.writeLong(convertDuration(expiryPlc.getExpiryForCreation()));
+                        w.writeLong(convertDuration(expiryPlc.getExpiryForUpdate()));
+                        w.writeLong(convertDuration(expiryPlc.getExpiryForAccess()));
+                    }
+                });
+            }
+            else if (cfg.getExpiryPolicy() != null) {
+                throw new ClientProtocolError(String.format("Expire policies have not supported by the server " +
+                    "version %s, required version %s", ver, V1_6_0));
+            }
+
             writer.writeInt(origPos, out.position() - origPos - 4); // configuration length
             writer.writeInt(origPos + 4, propCnt.get()); // properties count
         }
