diff --git a/old_test.java b/new_test.java
index ba2e2ce..d0618b6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
         ZoneRules rules = ZoneRulesProvider.getRules("Europe/London", false);
         assertEquals(versions.lastEntry().getValue(), rules);
 
-        NavigableMap<String, ZoneRules> copy = new TreeMap<>(versions);
+        NavigableMap<String, ZoneRules> copy = new TreeMap<String, ZoneRules>(versions);
         versions.clear();
         assertEquals(versions.size(), 0);
         NavigableMap<String, ZoneRules> versions2 = ZoneRulesProvider.getVersions("Europe/London");
