diff --git a/old_product.java b/new_product.java
index 4b81073..7f46382 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public static NavigableMap<String, ZoneRules> getVersions(String zoneId) {
-        Objects.requireNonNull(zoneId, "zoneId");
+        Jdk8Methods.requireNonNull(zoneId, "zoneId");
         return getProvider(zoneId).provideVersions(zoneId);
     }
\ No newline at end of file
