diff --git a/old_test.java b/new_test.java
index fe4f373..598eadb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,12 +8,25 @@
 			new CancelCheckpointMarker(287087987329842L)
 		};
 
+		Class[] expectedClasses = Arrays.stream(events)
+			.map(AbstractEvent::getClass)
+			.toArray(Class[]::new);
+
 		for (AbstractEvent evt : events) {
-			for (AbstractEvent evt2 : events) {
-				if (evt == evt2) {
-					assertTrue(checkIsEvent(evt, evt2.getClass()));
+			for (Class<?> expectedClass: expectedClasses) {
+				if (expectedClass.equals(TestTaskEvent.class)) {
+					try {
+						checkIsEvent(evt, expectedClass);
+						fail("This should fail");
+					}
+					catch (UnsupportedOperationException ex) {
+						// expected
+					}
+				}
+				else if (evt.getClass().equals(expectedClass)) {
+					assertTrue(checkIsEvent(evt, expectedClass));
 				} else {
-					assertFalse(checkIsEvent(evt, evt2.getClass()));
+					assertFalse(checkIsEvent(evt, expectedClass));
 				}
 			}
 		}
