diff --git a/old_test.java b/new_test.java
index b0527e0..8b03373 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-	@Verifies(value = "should return all order history", method = "getOrderHistoryByOrderNumber(String)")
-	public void getOrderHistoryByOrderNumber_shouldReturnAllOrderHistory() throws Exception {
+	@Verifies(value = "should return all order history for given order number", method = "getOrderHistoryByOrderNumber(String)")
+	public void getOrderHistoryByOrderNumber_shouldReturnAllOrderHistoryForGivenOrderNumber() throws Exception {
 		List<Order> orders = Context.getOrderService().getOrderHistoryByOrderNumber("111");
 		Assert.assertEquals(2, orders.size());
 		Assert.assertEquals(111, orders.get(0).getOrderId().intValue());
