diff --git a/old_test.java b/new_test.java
index c699c60..35cb32a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,8 +9,8 @@
         validKeys.sort(KEY_COMPARATOR);
 
         val validKeysIterator = validKeys.iterator();
-        Integer expectedValue = -1;
-        for (int testKey = 0; testKey < maxKey; testKey++) {
+        Long expectedValue = -1L;
+        for (long testKey = 0; testKey < maxKey; testKey++) {
             // Since both testKey and validKeysIterator increase with natural ordering, finding the next expected value
             // is a straightforward call to the iterator next() method.
             while (expectedValue != null && testKey > expectedValue) {
@@ -22,7 +22,7 @@
             }
 
             val ceilingEntry = index.getCeiling(testKey);
-            Integer actualValue = ceilingEntry != null ? ceilingEntry.key() : null;
+            Long actualValue = ceilingEntry != null ? ceilingEntry.key() : null;
             Assert.assertEquals("Unexpected value for getCeiling for key " + testKey, expectedValue, actualValue);
         }
     }
\ No newline at end of file
