diff --git a/old_product.java b/new_product.java
index 58aaf58..ab94b07 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,6 +6,9 @@ public ToneAnalysis getTone(final String text) {
     final JsonObject contentJson = new JsonObject();
     contentJson.addProperty(TEXT, text);
 
-    final Request request = RequestBuilder.post(PATH_TONE).withBodyJson(contentJson).build();
+    final Request request = RequestBuilder.post(PATH_TONE)
+        .withQuery(VERSION_DATE, versionDate)
+        .withBodyJson(contentJson)
+        .build();
     return executeRequest(request, ToneAnalysis.class);
   }
\ No newline at end of file
