diff --git a/old_product.java b/new_product.java
index 9f58171..02ca024 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,7 +13,9 @@
             @NotBlank(message = "No cluster name entered. Unable to update cluster info for job.")
             final String clusterName
     ) throws GenieException {
-        LOG.debug("Setting the application info for job with id " + id);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Setting the application info for job with id " + id);
+        }
         final Job job = this.jobRepo.findOne(id);
         if (job != null) {
             job.setExecutionClusterId(clusterId);
