diff --git a/old_product.java b/new_product.java
index 0ad9dff..5af6dcd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,6 +7,6 @@ public void hugeFolder() throws Exception {
         IntStream.range(0, 2000).forEach(i -> names.add(randomString()));
 
         for (String filename: names) {
-            userRoot.mkdir(filename, context, false, context.crypto.random);
+            userRoot.mkdir(filename, context.network, false, context.crypto.random, context.entrie);
         }
     }
\ No newline at end of file
