diff --git a/old_product.java b/new_product.java
index 8de98c4..65a0b3a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -21,9 +21,14 @@ public Map<String,String[]> toStringArrayMap()
             }
         };
         
-        for(Map.Entry<String,Object> entry: entrySet())
+        for(Map.Entry<String,List<V>> entry: entrySet())
         {
-            String[] a = LazyList.toStringArray(entry.getValue());
+            String[] a = null;
+            if (entry.getValue() != null)
+            {
+                a = new String[entry.getValue().size()];
+                a = entry.getValue().toArray(a);
+            }
             map.put(entry.getKey(),a);
         }
         return map;
