diff --git a/old_test.java b/new_test.java
index d7af545..ae2e236 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	public void detect_concave() {
-		List<Polygon2D_F64> polygons = new ArrayList<Polygon2D_F64>();
+		List<Polygon2D_F64> polygons = new ArrayList<>();
 		Polygon2D_F64 expected = new Polygon2D_F64(20, 20, 20, 80, 40, 40, 80, 80, 80, 20);
 		polygons.add(expected);
 
