diff --git a/old_product.java b/new_product.java
index 2689b15..524f0f8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,9 +8,9 @@
 			intensities[2] = new GrayF32(1, 1);
 
 			maximums = new List[3];
-			maximums[0] = new ArrayList<Point2D_I16>();
-			maximums[1] = new ArrayList<Point2D_I16>();
-			maximums[2] = new ArrayList<Point2D_I16>();
+			maximums[0] = new ArrayList<>();
+			maximums[1] = new ArrayList<>();
+			maximums[2] = new ArrayList<>();
 		}
 		foundPoints.clear();
 
