diff --git a/old_test.java b/new_test.java
index bf83125..2a7cbf2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
   public void list() {
-    error(_FILE_LIST.args(PATH1), Err.FILE_NODIR);
+    error(_FILE_LIST.args(PATH1), Err.FILE_ND);
     query(_FILE_WRITE.args(PATH1, "()"));
-    error(_FILE_LIST.args(PATH1), Err.FILE_NODIR);
+    error(_FILE_LIST.args(PATH1), Err.FILE_ND);
     contains(_FILE_LIST.args(PATH), NAME);
     contains(_FILE_LIST.args(PATH, "false()"), NAME);
     contains(_FILE_LIST.args(PATH, "false()", NAME), NAME);
