diff --git a/old_product.java b/new_product.java
index 0eac51e..dfdf2f2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,11 +4,11 @@ private Iter list(final QueryContext ctx) throws QueryException {
     try {
       dir = new File(dir.getCanonicalPath());
     } catch(final IOException ex) {
-      throw FILE_PATH.get(info, dir);
+      throw FILE_IE_PATH.get(info, dir);
     }
 
     // check if the addresses path is a directory
-    if(!dir.isDirectory()) throw FILE_NODIR.get(info, dir);
+    if(!dir.isDirectory()) throw FILE_ND.get(info, dir);
 
     final boolean rec = optionalBool(1, ctx);
     final Pattern pat = expr.length == 3 ? Pattern.compile(IOFile.regex(
