diff --git a/old_test.java b/new_test.java
index d8aee71..3903573 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	@Category(Smoke.class)
 	public void getUser() {
-		GetUserResponse response = accountWebServiceClient.getUser(1L);
+		GetUserResult response = accountWebServiceClient.getUser(1L);
 		assertEquals("admin", response.getUser().getLoginName());
 	}
\ No newline at end of file
