diff --git a/old_product.java b/new_product.java
index 1d1ff22..6cbd301 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,15 @@
                                  final SecurityContext securityContext, final AsyncResponse asyncResponse) {
         long traceId = LoggerHelpers.traceEnter(log, "getScalingEvents");
 
+        try {
+            authenticate(scopeName + "/" + streamName, READ);
+        } catch (AuthenticationException e) {
+            log.warn("Get scaling events for {} failed due to authentication failure.", scopeName + "/" + streamName);
+            asyncResponse.resume(Response.status(Status.UNAUTHORIZED).build());
+            LoggerHelpers.traceLeave(log, "Get scaling events", traceId);
+            return;
+        }
+
         if (from < 0 || to < 0 || from > to) {
             log.warn("Received invalid request from client for scopeName/streamName: {}/{} ", scopeName, streamName);
             asyncResponse.resume(Response.status(Status.BAD_REQUEST).build());
