diff --git a/old_product.java b/new_product.java
index c97e3dd..7aaff77 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,15 +1,8 @@
-public static void dump(final Item it, final byte[] label, final InputInfo info,
+public static void trace(final Item it, final byte[] label, final InputInfo info,
       final QueryContext qc) throws QueryException {
     try {
-      final byte[] value;
-      if(it == null) {
-        value = token(SeqType.EMP.toString());
-      } else if(it instanceof FItem || it.type == NodeType.ATT || it.type == NodeType.NSP) {
-        value = token(it.toString());
-      } else {
-        value = it.serialize(SerializerOptions.get(false)).finish();
-      }
-      dump(value, label, qc);
+      trace(it == null ? token(SeqType.EMP.toString()) :
+        it.serialize(SerializerMode.DEBUG.get()).finish(), label, qc);
     } catch(final QueryIOException ex) {
       throw ex.getCause(info);
     }
