diff --git a/old_product.java b/new_product.java
index 27f4bcd..a5a5fd2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -26,11 +26,18 @@
 			if (allergy.getAllergyId() == null && allergy.getPatient() != null) {
 				Allergies existingAllergies = patientService.getAllergies(allergy.getPatient());
 				if (existingAllergies.containsAllergen(allergy)) {
-					String key = "ui.i18n.Concept.name." + allergen.getCodedAllergen().getUuid();
-					String name = Context.getMessageSourceService().getMessage(key);
-					if (key.equals(name)) {
-						name = allergen.toString();
+					String name;
+					if (allergen.isCoded()) {
+						String key = "ui.i18n.Concept.name." + allergen.getCodedAllergen().getUuid();
+						name = Context.getMessageSourceService().getMessage(key);
+						if (key.equals(name)) {
+							name = allergen.toString();
+						}
 					}
+					else {
+						name = allergen.getNonCodedAllergen();
+					}
+						
 					errors.rejectValue("allergen", "allergyapi.message.duplicateAllergen", new Object[] { name }, null);
 				}
 			}
