diff --git a/old_product.java b/new_product.java
index 722fc35..2926a4f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,7 +7,7 @@
       @Override
       public LoadMetadataTResponse call() throws AlluxioException, IOException {
         return new LoadMetadataTResponse(mFileSystemMaster.loadMetadata(new AlluxioURI(alluxioPath),
-            LoadMetadataOptions.defaults().setCreateAncestors(true).setLoadDirectChildren(true)));
+            LoadMetadataOptions.defaults().setCreateAncestors(true).setLoadDescendantLevels(1)));
       }
 
       @Override
