diff --git a/old_test.java b/new_test.java
index a75d1f8..becbb5e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -25,7 +25,7 @@
 
     // This should not throw file exists exception those a/f1 is loaded.
     mFileSystemMaster.loadMetadata(new AlluxioURI("alluxio:/a"),
-        LoadMetadataOptions.defaults().setCreateAncestors(true).setLoadDirectChildren(true));
+        LoadMetadataOptions.defaults().setCreateAncestors(true).setLoadDescendantLevels(1));
 
     // TODO(peis): Avoid this hack by adding an option in getFileInfo to skip loading metadata.
     try {
@@ -38,5 +38,5 @@
     }
 
     mFileSystemMaster.loadMetadata(new AlluxioURI("alluxio:/a"),
-        LoadMetadataOptions.defaults().setCreateAncestors(true).setLoadDirectChildren(true));
+        LoadMetadataOptions.defaults().setCreateAncestors(true).setLoadDescendantLevels(1));
   }
\ No newline at end of file
