diff --git a/old_product.java b/new_product.java
index 25a43a9..0453376 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ private synchronized Item writeText(final boolean append, final QueryContext qc)
 
     final Path path = checkParentDir(checkPath(0, qc));
     final byte[] s = checkStr(exprs[1], qc);
-    final String enc = encoding(2, FILE_UNKNOWN_ENCODING, qc);
+    final String enc = checkEncoding(2, FILE_UNKNOWN_ENCODING, qc);
     final Charset cs = enc == null || enc == UTF8 ? null : Charset.forName(enc);
 
     try(final PrintOutput out = PrintOutput.get(new FileOutputStream(path.toFile(), append))) {
