diff --git a/old_test.java b/new_test.java
index effe717..e0d0620 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void writeText() {
     error(_FILE_WRITE_TEXT.args(PATH, "x"), Err.FILE_IS_DIR);
-    error(_FILE_WRITE_TEXT.args(PATH1, " 123"), Err.INVCAST);
+    error(_FILE_WRITE_TEXT.args(PATH1, " 123"), Err.INVCASTEX);
 
     query(_FILE_WRITE_TEXT.args(PATH1, "x"));
     query(_FILE_SIZE.args(PATH1), "1");
