diff --git a/old_test.java b/new_test.java
index 5653a8e..2e454d0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,12 +5,8 @@
 		
 		OrderGroup group = new OrderGroup(null, patient);
 		group.setCreator(provider);
-		group.setDateCreated(new Date());
 		Order order = new Order();
-		order.setOrderNumber("1");
-		order.setDateCreated(new Date());
 		order.setConcept(Context.getConceptService().getConcept(23));
-		order.setPatient(patient);
 		group.addOrder(order);
 		
 		group = Context.getOrderService().saveOrderGroup(group);
