diff --git a/old_test.java b/new_test.java
index 3eb3f53..ffbe58b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testToStringArrayMap()
     {
-        MultiMap<String> mm = new MultiMap<String>();
+        MultiMap mm = new MultiMap();
         mm.putValues("food","apple","cherry","raspberry");
         mm.put("color","red");
         mm.putValues("amount","bushel","pint");
