diff --git a/old_product.java b/new_product.java
index ed65d84..7ca8e19 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,18 +1,19 @@
-public Map<K,String[]> toStringArrayMap()
+public Map<String,String[]> toStringArrayMap()
     {
-        HashMap<K,String[]> map = new HashMap<K,String[]>(_map.size()*3/2)
+        HashMap<String,String[]> map = new HashMap<String,String[]>(_map.size()*3/2)
         {
+            @Override
             public String toString()
             {
                 StringBuilder b=new StringBuilder();
                 b.append('{');
-                for (K k:keySet())
+                for (String k:super.keySet())
                 {
                     if(b.length()>1)
                         b.append(',');
                     b.append(k);
                     b.append('=');
-                    b.append(Arrays.asList(get(k)));
+                    b.append(Arrays.asList(super.get(k)));
                 }
 
                 b.append('}');
@@ -20,7 +21,7 @@ public Map<K,String[]> toStringArrayMap()
             }
         };
         
-        for(Map.Entry<K,Object> entry: _map.entrySet())
+        for(Map.Entry<String,Object> entry: _map.entrySet())
         {
             String[] a = LazyList.toStringArray(entry.getValue());
             map.put(entry.getKey(),a);
