diff --git a/old_product.java b/new_product.java
index edf48ec..4f425db 100644
--- a/old_product.java
+++ b/new_product.java
@@ -43,7 +43,9 @@ public ServiceCall<DialogNode> updateDialogNode(UpdateDialogNodeOptions updateDi
     if (updateDialogNodeOptions.newParent() != null) {
       contentJson.addProperty("parent", updateDialogNodeOptions.newParent());
     }
-    contentJson.addProperty("dialog_node", updateDialogNodeOptions.newDialogNode());
+    if (updateDialogNodeOptions.newDialogNode() != null) {
+      contentJson.addProperty("dialog_node", updateDialogNodeOptions.newDialogNode());
+    }
     builder.bodyJson(contentJson);
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(DialogNode.class));
   }
\ No newline at end of file
