diff --git a/old_test.java b/new_test.java
index 3cd284a..cd7d6b5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,8 +6,8 @@
         Set<Stream> streams = new HashSet<>();
         streams.add(stream);
         when(synStreamFrame.isClose()).thenReturn(false);
-        SynInfo synInfo = new SynInfo(false);
+        SynInfo synInfo = new SynInfo(new Fields(), false);
         when(session.getStreams()).thenReturn(streams);
-        stream.syn(synInfo);
-        verify(session).syn(argThat(new PushSynInfoMatcher(stream.getId(), synInfo)), any(StreamFrameListener.class), anyLong(), any(TimeUnit.class), any(Promise.class));
+        stream.syn(synInfo, new Promise.Adapter<Stream>());
+        verify(session).syn(argThat(new PushSynInfoMatcher(stream.getId(), synInfo)), any(StreamFrameListener.class), any(Promise.class));
     }
\ No newline at end of file
