diff --git a/old_product.java b/new_product.java
index 5a95a99..5d33b1a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-public Future<Stream> syn(SynInfo synInfo, StreamFrameListener listener);
\ No newline at end of file
+public Stream syn(SynInfo synInfo, StreamFrameListener listener) throws ExecutionException, InterruptedException, TimeoutException;
\ No newline at end of file
