diff --git a/old_test.java b/new_test.java
index 512deb0..c693730 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,14 +2,14 @@
   public void unmount() throws Exception {
     AlluxioURI alluxioURI = new AlluxioURI("/hello");
     AlluxioURI ufsURI = createTempUfsDir("ufs/hello");
-    mFileSystemMaster.mount(alluxioURI, ufsURI, MountOptions.defaults());
-    mFileSystemMaster.createDirectory(alluxioURI.join("dir"),
-        CreateDirectoryOptions.defaults().setPersisted(true));
+    mFileSystemMaster.mount(alluxioURI, ufsURI, MountContext.defaults());
+    mFileSystemMaster.createDirectory(alluxioURI.join("dir"), CreateDirectoryContext
+        .defaults().setPersisted(true));
     mFileSystemMaster.unmount(alluxioURI);
     // after unmount, ufs path under previous mount point should still exist
     File file = new File(ufsURI.join("dir").toString());
     assertTrue(file.exists());
     // after unmount, alluxio path under previous mount point should not exist
     mThrown.expect(FileDoesNotExistException.class);
-    mFileSystemMaster.getFileInfo(alluxioURI.join("dir"), GET_STATUS_OPTIONS);
+    mFileSystemMaster.getFileInfo(alluxioURI.join("dir"), GET_STATUS_CONTEXT);
   }
\ No newline at end of file
