diff --git a/old_product.java b/new_product.java
index e40c4b4..0f2b212 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
 public int rebalanceNode(RebalancePartitionsInfo stealInfo) {
+        List<ROStoreVersionMap> readOnlyStoreVersionsList = decodeROStoreVersionMap(stealInfo.getReadOnlyStoreVersions());
         VAdminProto.InitiateRebalanceNodeRequest rebalanceNodeRequest = VAdminProto.InitiateRebalanceNodeRequest.newBuilder()
                                                                                                                 .setAttempt(stealInfo.getAttempt())
                                                                                                                 .setDonorId(stealInfo.getDonorId())
@@ -6,6 +7,9 @@ public int rebalanceNode(RebalancePartitionsInfo stealInfo) {
                                                                                                                 .addAllPartitions(stealInfo.getPartitionList())
                                                                                                                 .addAllUnbalancedStore(stealInfo.getUnbalancedStoreList())
                                                                                                                 .addAllDeletePartitions(stealInfo.getDeletePartitionsList())
+                                                                                                                .addAllStealMasterPartitions(stealInfo.getStealMasterPartitions())
+                                                                                                                .addAllRoStoreVersions(readOnlyStoreVersionsList)
+                                                                                                                .setDeleteAfterRebalance(stealInfo.getDeleteAfterRebalance())
                                                                                                                 .build();
         VAdminProto.VoldemortAdminRequest adminRequest = VAdminProto.VoldemortAdminRequest.newBuilder()
                                                                                           .setType(VAdminProto.AdminRequestType.INITIATE_REBALANCE_NODE)
