diff --git a/old_test.java b/new_test.java
index bd5dc45..0f70e81 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,9 +17,11 @@
         RebalancePartitionsInfo stealInfo = new RebalancePartitionsInfo(1,
                                                                         0,
                                                                         rebalancePartitionList,
-                                                                        new ArrayList<Integer>(0),
+                                                                        rebalancePartitionList,
+                                                                        rebalancePartitionList,
                                                                         Arrays.asList(testStoreName),
-                                                                        0);
+                                                                        0,
+                                                                        true);
         int asyncId = adminClient.rebalanceNode(stealInfo);
         assertNotSame("Got a valid rebalanceAsyncId", -1, asyncId);
 
