diff --git a/old_test.java b/new_test.java
index 3e0d3c7..1613006 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-  public void swallow() {
-    query(_PROF_SWALLOW.args("()"), "");
-    query(_PROF_SWALLOW.args("1"), "");
-    query(_PROF_SWALLOW.args("1,2"), "");
+  public void voidd() {
+    query(_PROF_VOID.args("()"), "");
+    query(_PROF_VOID.args("1"), "");
+    query(_PROF_VOID.args("1,2"), "");
   }
\ No newline at end of file
