diff --git a/old_test.java b/new_test.java
index 6535232..b468106 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
     mFileSystemMaster.scheduleAsyncPersistence(ROOT_FILE_URI);
 
     FileSystemCommand command = mFileSystemMaster
-        .workerHeartbeat(mWorkerId1, Lists.newArrayList(fileId), WorkerHeartbeatOptions.defaults());
-    assertEquals(CommandType.Persist, command.getCommandType());
-    assertEquals(0, command.getCommandOptions().getPersistOptions().getPersistFiles().size());
+        .workerHeartbeat(mWorkerId1, Lists.newArrayList(fileId), WorkerHeartbeatContext.defaults());
+    assertEquals(alluxio.wire.CommandType.Persist, command.getCommandType());
+    assertEquals(0, command.getCommandOptions().getPersistOptions().getFilesToPersist().size());
   }
\ No newline at end of file
