diff --git a/old_test.java b/new_test.java
index a98b381..8ed7a0e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,21 +1,23 @@
 @Test
-	public void testGetIdentifiableLanguages() {
+  public void testGetIdentifiableLanguages() {
 
-		Map<String, Object> response = new HashMap<String, Object>();
-		List<IdentifiableLanguage> langs = new ArrayList<IdentifiableLanguage>();
-		langs.add(new IdentifiableLanguage("af", "Afrikaans"));
-		langs.add(new IdentifiableLanguage("ar", "Arabic"));
-		langs.add(new IdentifiableLanguage("az", "Azerbaijani"));
-		langs.add(new IdentifiableLanguage("zh", "Chinese"));
-		response.put("languages", langs);
+    final Map<String, Object> response = new HashMap<String, Object>();
+    final List<IdentifiableLanguage> langs = new ArrayList<IdentifiableLanguage>();
+    langs.add(new IdentifiableLanguage("af", "Afrikaans"));
+    langs.add(new IdentifiableLanguage("ar", "Arabic"));
+    langs.add(new IdentifiableLanguage("az", "Azerbaijani"));
+    langs.add(new IdentifiableLanguage("zh", "Chinese"));
+    response.put("languages", langs);
 
-		mockServer.when(request().withPath(IDENTIFIABLE_LANGUAGES_PATH)).respond(
-				response().withHeaders(new Header(HttpHeaders.Names.CONTENT_TYPE, MediaType.APPLICATION_JSON))
-						.withBody(GsonSingleton.getGson().toJson(response)));
+    mockServer.when(request().withPath(IDENTIFIABLE_LANGUAGES_PATH)).respond(
+        response().withHeaders(
+            new Header(HttpHeaders.Names.CONTENT_TYPE, HttpMediaType.APPLICATION_JSON)).withBody(
+            GsonSingleton.getGson().toJson(response)));
 
-		List<IdentifiableLanguage> languages = service.getIdentifiableLanguages();
+    final List<IdentifiableLanguage> languages = service.getIdentifiableLanguages();
 
-		mockServer.verify(request().withPath(IDENTIFIABLE_LANGUAGES_PATH), VerificationTimes.exactly(1));
-		assertNotNull(languages);
-		assertEquals(languages,langs);
-	}
\ No newline at end of file
+    mockServer
+        .verify(request().withPath(IDENTIFIABLE_LANGUAGES_PATH), VerificationTimes.exactly(1));
+    assertNotNull(languages);
+    assertEquals(languages, langs);
+  }
\ No newline at end of file
