diff --git a/old_test.java b/new_test.java
index 877e82d..ef372bf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testParse() throws GenieException {
+    public void testParse() throws GeniePreconditionException {
         Assert.assertEquals(ProcessStatus.JOB_INTERRUPTED,
                 ProcessStatus.parse(-1));
         Assert.assertEquals(ProcessStatus.SUCCESS,
