diff --git a/old_product.java b/new_product.java
index 8aee7da..881eafc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static ApplicationStatus parse(final String value) throws GenieException {
+public static ApplicationStatus parse(final String value) throws GeniePreconditionException {
         if (StringUtils.isNotBlank(value)) {
             for (final ApplicationStatus status : ApplicationStatus.values()) {
                 if (value.equalsIgnoreCase(status.toString())) {
@@ -6,6 +6,6 @@ public static ApplicationStatus parse(final String value) throws GenieException
                 }
             }
         }
-        throw new GenieException(HttpURLConnection.HTTP_NOT_ACCEPTABLE,
+        throw new GeniePreconditionException(
                 "Unacceptable application status. Must be one of {ACTIVE, DEPRECATED, INACTIVE}");
     }
\ No newline at end of file
