diff --git a/old_test.java b/new_test.java
index 5c754e3..e42c787 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,8 @@
 @Test
     public void testStart() {
 	server.occupie();
-	server.start("AFL/afl-showmap -m none -o [output]/[id] ");
+	CertificateMutator mut = new FixedCertificateMutator();
+	server.start("", mut.getServerCertificateKeypair().getCertificateFile(), mut.getServerCertificateKeypair()
+		.getKeyFile());
 	server.serverIsRunning();
     }
\ No newline at end of file
