diff --git a/old_product.java b/new_product.java
index 5a1c4d8..0bd384f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,11 @@
-public synchronized void start(String prefix) {
+public synchronized void start(String prefix, File certificateFile, File keyFile) {
 
 	// You have to ooccupie a Server to start it
 	if (!this.isFree()) {
 	    if (p != null) {
-		p.destroy();
+		stop();
 	    }
-	    restart(prefix);
+	    restart(prefix, certificateFile, keyFile);
 	} else {
 	    throw new IllegalStateException("Cant start a not marked Server. Occupie it first!");
 	}
