diff --git a/old_product.java b/new_product.java
index d02687c..cdc4bfd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,14 @@
                             final SecurityContext securityContext, final AsyncResponse asyncResponse) {
         long traceId = LoggerHelpers.traceEnter(log, "listStreams");
 
+        try {
+            authenticate(scopeName, READ);
+        } catch (AuthenticationException e) {
+            log.warn("List streams for {} failed due to authentication failure.", scopeName);
+            asyncResponse.resume(Response.status(Status.UNAUTHORIZED).build());
+            LoggerHelpers.traceLeave(log, "listStreams", traceId);
+            return;
+        }
         boolean showOnlyInternalStreams = showInternalStreams != null && showInternalStreams.equals("true");
         controllerService.listStreamsInScope(scopeName)
                 .thenApply(streamsList -> {
