diff --git a/old_product.java b/new_product.java
index 1c5001f..5dab87a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,5 +12,15 @@
 					result.add(new GenericDrug(concept));
 			}
 		}
+		
+		// and next to try to find drugs by name
+		List<Drug> drugs = Context.getConceptService().getDrugs(query);
+		if (drugs != null) {
+			for (Drug drug : drugs) {
+				if (!drug.isRetired())
+					result.add(drug);
+			}
+		}
+		
 		return result;
 	}
\ No newline at end of file
