diff --git a/old_test.java b/new_test.java
index 108afa2..8a185f0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,5 +9,12 @@
 		
 		Assert.assertNotNull(result);
 		
-		Assert.assertEquals(2, result.size());
+		Assert.assertEquals(3, result.size());
+		
+		Boolean isExpected = result.get(0).getClass().equals(GenericDrug.class);
+		Assert.assertTrue(isExpected);
+		isExpected = result.get(1).getClass().equals(GenericDrug.class);
+		Assert.assertTrue(isExpected);
+		isExpected = result.get(2).getClass().equals(Drug.class);
+		Assert.assertTrue(isExpected);
 	}
\ No newline at end of file
