diff --git a/old_product.java b/new_product.java
index 4ebd59f..7e2072e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,13 @@
 @Override
     public ChronoOffsetDateTimeImpl<C> with(DateTimeField field, long newValue) {
-        if (field instanceof LocalDateTimeField) {
-            LocalDateTimeField f = (LocalDateTimeField) field;
+        if (field instanceof ChronoField) {
+            ChronoField f = (ChronoField) field;
             switch (f) {
                 case INSTANT_SECONDS:
                     long epochDays = Jdk8Methods.floorDiv(newValue, SECONDS_PER_DAY);
-                    ChronoOffsetDateTimeImpl<C> odt = with(LocalDateTimeField.EPOCH_DAY, epochDays);
+                    ChronoOffsetDateTimeImpl<C> odt = with(ChronoField.EPOCH_DAY, epochDays);
                     int secsOfDay = Jdk8Methods.floorMod(newValue, SECONDS_PER_DAY);
-                    odt  = odt.with(LocalDateTimeField.SECOND_OF_DAY, secsOfDay);
+                    odt  = odt.with(ChronoField.SECOND_OF_DAY, secsOfDay);
                     return odt;
 
                 case OFFSET_SECONDS: {
