diff --git a/old_test.java b/new_test.java
index 25a3f8c..753272d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,7 @@
 
 		final Candidate olderStockCandidate = Candidate.builder()
 				.type(Type.STOCK)
+				.clientId(org.getAD_Client_ID())
 				.orgId(org.getAD_Org_ID())
 				.productId(product.getM_Product_ID())
 				.warehouseId(warehouse.getM_Warehouse_ID())
@@ -17,6 +18,7 @@
 
 		final Candidate candidate = Candidate.builder()
 				.type(Type.SUPPLY)
+				.clientId(org.getAD_Client_ID())
 				.orgId(org.getAD_Org_ID())
 				.subType(SubType.PRODUCTION)
 				.productId(product.getM_Product_ID())
