diff --git a/old_product.java b/new_product.java
index f6fb7a0..6cefb5b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,8 +8,10 @@ public SqlNode updateByPrimaryKeySelective(MappedStatement ms) {
         List<SqlNode> ifNodes = new ArrayList<SqlNode>();
         //全部的if property!=null and property!=''
         for (EntityHelper.EntityColumn column : columnList) {
-            StaticTextSqlNode columnNode = new StaticTextSqlNode(column.getColumn() + " = #{" + column.getProperty() + "}, ");
-            ifNodes.add(getIfNotNull(column, columnNode));
+            if (!column.isId()) {
+                StaticTextSqlNode columnNode = new StaticTextSqlNode(column.getColumn() + " = #{" + column.getProperty() + "}, ");
+                ifNodes.add(getIfNotNull(column, columnNode));
+            }
         }
         sqlNodes.add(new SetSqlNode(ms.getConfiguration(), new MixedSqlNode(ifNodes)));
         //获取全部的主键的列
