diff --git a/old_test.java b/new_test.java
index d822ba3..6f9cf25 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,6 @@
 @Test
   public void testSystem() throws BaseXException {
     // wrong arguments
-    args("db:system");
-
-    // standard test
-    contains("db:system()", "ON");
+    final String fun = check(FunDef.SYSTEM);
+    contains(fun + "()", "ON");
   }
\ No newline at end of file
