diff --git a/old_test.java b/new_test.java
index 50bf9e2..df50398 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
     public void testContains_Char() {
-        CharSet btod = CharSet.getInstance("b-d");
-        CharSet dtob = CharSet.getInstance("d-b");
-        CharSet bcd = CharSet.getInstance("bcd");
-        CharSet bd = CharSet.getInstance("bd");
-        CharSet notbtod = CharSet.getInstance("^b-d");
+        final CharSet btod = CharSet.getInstance("b-d");
+        final CharSet dtob = CharSet.getInstance("d-b");
+        final CharSet bcd = CharSet.getInstance("bcd");
+        final CharSet bd = CharSet.getInstance("bd");
+        final CharSet notbtod = CharSet.getInstance("^b-d");
         
         assertFalse(btod.contains('a'));
         assertTrue(btod.contains('b'));
@@ -36,7 +36,7 @@
         assertTrue(dtob.contains('d'));
         assertFalse(dtob.contains('e'));
       
-        CharRange[] array = dtob.getCharRanges();
+        final CharRange[] array = dtob.getCharRanges();
         assertEquals("[b-d]", dtob.toString());
         assertEquals(1, array.length);
     }
\ No newline at end of file
