diff --git a/old_test.java b/new_test.java
index db20d86..7ec6896 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
     public void testProcessFilter() {
-        entryPoint.setFilterSuffix("/saml/sso");
-        expect(request.getRequestURI()).andReturn("/web/saml/sso");
-        expect(request.getRequestURI()).andReturn("/saml/sso");
+
+        expect(request.getRequestURI()).andReturn("/web/saml/login");
+        expect(request.getRequestURI()).andReturn("/saml/login");
         expect(request.getRequestURI()).andReturn("/saml");
-        expect(request.getRequestURI()).andReturn("/sso/");
-        expect(request.getRequestURI()).andReturn("/saml/sso/");
+        expect(request.getRequestURI()).andReturn("/login/");
+        expect(request.getRequestURI()).andReturn("/saml/login/");
 
         replayMock();
         assertTrue(entryPoint.processFilter(request));
