diff --git a/old_test.java b/new_test.java
index 519b114..2a091d0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,6 +11,10 @@
         Assert.assertNotNull(localJobEntity.getStarted());
         Assert.assertTrue(dt.compareTo(localJobEntity.getFinished()) == 0);
 
+        // Shouldn't affect finish time
+        localJobEntity.setJobStatus(JobStatus.RUNNING);
+        Assert.assertTrue(dt.compareTo(localJobEntity.getFinished()) == 0);
+
         // finish time is non-zero on completion
         localJobEntity.setJobStatus(JobStatus.SUCCEEDED);
         Assert.assertFalse(dt.compareTo(localJobEntity.getFinished()) == 0);
