diff --git a/old_product.java b/new_product.java
index da34a7d..e3e9f80 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,9 +3,7 @@ public void setJobStatus(final JobStatus jobStatus) {
 
         if (jobStatus == JobStatus.INIT) {
             this.setStarted(new Date());
-        } else if (jobStatus == JobStatus.SUCCEEDED
-            || jobStatus == JobStatus.KILLED
-            || jobStatus == JobStatus.FAILED) {
+        } else if (jobStatus != JobStatus.RUNNING) {
             setFinished(new Date());
         }
     }
\ No newline at end of file
