diff --git a/old_product.java b/new_product.java
index cec24dd..8552411 100644
--- a/old_product.java
+++ b/new_product.java
@@ -33,9 +33,10 @@ public Dependency insepct(File file) throws IOException {
                 fileNameEvidence, Evidence.Confidence.HIGH);
         dependency.getVendorEvidence().addEvidence("jar", "file name",
                 fileNameEvidence, Evidence.Confidence.HIGH);
-        dependency.getVersionEvidence().addEvidence("jar", "file name",
-                fileNameEvidence, Evidence.Confidence.HIGH);
-
+        if (fileNameEvidence.matches(".*\\d.*")) {
+            dependency.getVersionEvidence().addEvidence("jar", "file name",
+                    fileNameEvidence, Evidence.Confidence.HIGH);
+        }
         String md5 = null;
         String sha1 = null;
         try {
@@ -52,10 +53,5 @@ public Dependency insepct(File file) throws IOException {
         parseManifest(dependency);
         analyzePackageNames(dependency);
 
-        //TODO - can we get "version" information from the filename?  add it as medium confidence?
-        //   strip extension. find first numeric, chop off the first part. consider replacing [_-] with .
-        //dependency.getVersionEvidence().addEvidence("jar", "file name",
-        //                      version from file, Evidence.Confidence.MEDIUM);
-
         return dependency;
     }
\ No newline at end of file
