diff --git a/old_test.java b/new_test.java
index 20e415a..4f2cb0e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,6 @@
         int k = 1;
         double p = 0.5;
         double expResult = 1.0;
-        double result = DiscreteDistributions.BernoulliCdf(k, p);
+        double result = DiscreteDistributions.bernoulliCdf(k, p);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
