diff --git a/old_test.java b/new_test.java
index 593ca4e..b11920c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test
-	@Verifies(value = "should fail validation if complex is null", method = "validate(Object,Errors)")
-	public void validate_shouldFailValidationIfComplexIsNull() throws Exception {
+	@Verifies(value = "should fail validation if dosingType is null", method = "validate(Object,Errors)")
+	public void validate_shouldFailValidationIfDosingTypeIsNull() throws Exception {
 		DrugOrder order = new DrugOrder();
-		order.setComplex(null);
+		order.setDosingType(null);
 		order.setDrug(Context.getConceptService().getDrug(3));
 		
 		Errors errors = new BindException(order, "order");
 		new DrugOrderValidator().validate(order, errors);
 		
-		Assert.assertTrue(errors.hasFieldErrors("complex"));
+		Assert.assertTrue(errors.hasFieldErrors("dosingType"));
 	}
\ No newline at end of file
