diff --git a/old_product.java b/new_product.java
index 3d4fcd1..bd0fe01 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,8 +6,8 @@ public void validate(Object obj, Errors errors) {
 			errors.rejectValue("order", "error.general");
 		} else {
 			// for the following elements Order.hbm.xml says: not-null="true"
-			ValidationUtils.rejectIfEmpty(errors, "prn", "error.null");
-			ValidationUtils.rejectIfEmpty(errors, "complex", "error.null");
+			ValidationUtils.rejectIfEmpty(errors, "asNeeded", "error.null");
+			ValidationUtils.rejectIfEmpty(errors, "dosingType", "error.null");
 			//ValidationUtils.rejectIfEmpty(errors, "drug", "error.null");
 			
 			if (order.getDrug() != null)
