diff --git a/old_test.java b/new_test.java
index f19c971..4aa5fe0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
       final String query = test[1].isEmpty() ? _CSV_PARSE.args(test[0]) :
         _CSV_PARSE.args(test[0], " {" + test[1] + "}");
       if(test.length == 2) {
-        error(query, Err.BXCS_CONFSEP, Err.INVALIDOPT);
+        error(query, Err.BXCS_CONFIG, Err.INVALIDOPT);
       } else if(test[2].startsWith("...")) {
         contains(query, test[2].substring(3));
       } else {
