diff --git a/old_product.java b/new_product.java
index a4d7d69..a82ef77 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,8 +3,8 @@ private Item parse(final QueryContext ctx) throws QueryException {
     final JsonOptions opts = checkOptions(1, Q_OPTIONS, new JsonOptions(), ctx);
 
     try {
-      return JsonConverter.get(opts, info).convert(string(input)).item(ctx, info);
+      return JsonConverter.get(opts).convert(string(input)).item(ctx, info);
     } catch(final QueryIOException ex) {
-      throw ex.getCause();
+      throw ex.getCause(info);
     }
   }
\ No newline at end of file
