diff --git a/old_test.java b/new_test.java
index 4e0471e..62ab726 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,8 +3,8 @@
         final Application app = new Application(
                 APP_1_NAME,
                 APP_1_USER,
-                ApplicationStatus.ACTIVE,
-                APP_1_VERSION
+                APP_1_VERSION,
+                ApplicationStatus.ACTIVE
         );
         final String id = UUID.randomUUID().toString();
         app.setId(id);
