diff --git a/old_product.java b/new_product.java
index 5cdf029..144fc95 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,6 @@ public Application createApplication(final Application application)
             throw new GeniePreconditionException("No application passed in. Unable to validate.");
         }
 
-        application.validate();
         final HttpRequest request = BaseGenieClient.buildRequest(
                 Verb.POST,
                 BASE_CONFIG_APPLICATION_REST_URL,
