diff --git a/old_product.java b/new_product.java
index deae48d..0d734d4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public boolean process( I frame ) {
 		// update the feature tracker
 		tracker.process(frame);
-		totalProcessed++;
+		totalFramesProcessed++;
 
 		List<PointTrack> tracks = tracker.getActiveTracks(null);
 
@@ -20,7 +20,7 @@ public boolean process( I frame ) {
 
 		// mark that the track is in the inlier set
 		for( AssociatedPair p : modelMatcher.getMatchSet() ) {
-			((AssociatedPairTrack)p).lastUsed = totalProcessed;
+			((AssociatedPairTrack)p).lastUsed = totalFramesProcessed;
 		}
 
 		// refine the motion estimate
@@ -38,8 +38,7 @@ public boolean process( I frame ) {
 		for( PointTrack t : all ) {
 			AssociatedPairTrack p = t.getCookie();
 
-			if( totalProcessed - p.lastUsed >= pruneThreshold ) {
-				System.out.println("Delta "+(totalProcessed-p.lastUsed));
+			if( totalFramesProcessed - p.lastUsed >= pruneThreshold ) {
 				tracker.dropTrack(t);
 			}
 		}
