diff --git a/old_test.java b/new_test.java
index 6fc5b4e..dca8df1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
     public void testWithMonthOfYear() {
-        ThaiBuddhistDate date = testDate.withMonthOfYear(4);
-        assertEquals(date, ThaiBuddhistDate.thaiBuddhistDate(testYear, 4, testDayOfMonth));
+        ThaiBuddhistDate date = testDate.withMonthOfYear(MonthOfYear.APRIL);
+        assertEquals(date, ThaiBuddhistDate.thaiBuddhistDate(testYear, MonthOfYear.APRIL, testDayOfMonth));
     }
\ No newline at end of file
