diff --git a/old_product.java b/new_product.java
index fdb86d2..93badba 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public JapaneseDate withMonthOfYear(int monthOfYear) {
-        JapaneseChronology.monthOfYearRule().checkValue(monthOfYear);
-        return JapaneseDate.japaneseDate(date.withMonthOfYear(monthOfYear));
+public JapaneseDate withMonthOfYear(MonthOfYear monthOfYear) {
+        I18NUtil.checkNotNull(monthOfYear, "MonthOfYear must not be null");
+        return JapaneseDate.japaneseDate(date.with(monthOfYear));
     }
\ No newline at end of file
