diff --git a/old_product.java b/new_product.java
index f981c98..2be3b10 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,7 +7,7 @@ public static Map create(final String s) throws QueryException {
     boolean quote = false, error = false;
     final int pl = s.length();
     for(int p = 0; p < pl; p++) {
-      char ch = s.charAt(p);
+      final char ch = s.charAt(p);
       if(quote) {
         if(ch == '"' && (p + 1 == pl || s.charAt(p + 1) != '"')) {
           quote = false;
