diff --git a/old_test.java b/new_test.java
index d209447..3c67c86 100644
--- a/old_test.java
+++ b/new_test.java
@@ -30,8 +30,5 @@
     t1.join();
     t2.join();
 
-    // opens DB in session2 for further tests
-    session2.execute(new Open(NAME));
-
-    if(server.context.datas.pins(NAME) != 2) fail("test failed conCreate");
+    assertEquals(1, server.context.datas.pins(NAME));
   }
\ No newline at end of file
