diff --git a/old_test.java b/new_test.java
index c25c8cd..f4ef8a2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
         ospfRouter = new OspfRouterImpl();
 
         ospfController.agent.addLink(ospfRouter, new OspfLinkTedImpl());
-        ospfController.agent.deleteLink(ospfRouter);
+        ospfController.agent.deleteLink(ospfRouter, new OspfLinkTedImpl());
         assertThat(ospfController, is(notNullValue()));
     }
\ No newline at end of file
