diff --git a/old_product.java b/new_product.java
index 145b2a2..aae309e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static float nextFloat(float startInclusive, float endInclusive) {
+public static float nextFloat(final float startInclusive, final float endInclusive) {
         Validate.isTrue(endInclusive >= startInclusive,
                 "Start value must be smaller or equal to end value.");
         Validate.isTrue(startInclusive >= 0, "Both range values must be non-negative.");
