diff --git a/old_product.java b/new_product.java
index 02fcb81..1541cbc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public void print(Calendrical calendrical, Appendable appendable, DateTimeFormatSymbols symbols) throws IOException {
-        int value = calendrical.getRawValue(fieldRule);
+        int value = calendrical.getValue(fieldRule, false);
         String str = (value == Integer.MIN_VALUE ? "2147483648" : Integer.toString(Math.abs(value)));
         if (str.length() > maxWidth) {
             throw new CalendricalFormatFieldException(fieldRule, value, maxWidth);
