diff --git a/old_product.java b/new_product.java
index a56ebbf..68b5fea 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,5 +14,8 @@ public String csvCpeConfidence(Set<Identifier> ids) {
                 sb.append(id.getConfidence());
             }
         }
+        if (sb.length()==0) {
+            return "\"\"";
+        }
         return StringEscapeUtils.escapeCsv(sb.toString());
     }
\ No newline at end of file
