diff --git a/old_test.java b/new_test.java
index 74e7f1a..4f19586 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,7 +12,7 @@
         assertEquals(expResult, result);
 
         ids = new HashSet<>();
-        ids.add(new Identifier("gav", "somegroup:something:1.0", ""));
+        ids.add(new Identifier("maven", "somegroup:something:1.0", ""));
         expResult = "\"\"";
         result = instance.csvCpeConfidence(ids);
         assertEquals(expResult, result);
@@ -32,7 +32,7 @@
         Identifier i2 = new Identifier("cpe", "cpe:/a:somegroup:something2:1.0", "");
         i2.setConfidence(Confidence.MEDIUM);
         ids.add(i2);
-        Identifier i3 = new Identifier("gav", "somegroup:something:1.0", "");
+        Identifier i3 = new Identifier("maven", "somegroup:something:1.0", "");
         i3.setConfidence(Confidence.LOW);
         ids.add(i3);
 
