diff --git a/old_test.java b/new_test.java
index b142b81..98aedd6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,8 @@
   public void testDelete() throws QueryException, BaseXException {
     final String fun = check(Function.DELETE);
 
-    // add documents with certain prefix
     new Add("etc/test/dir", "docs", "test").execute(CONTEXT);
-    query(fun + "('db', 'test')", "");
-    query("count(collection('db/newtest')) eq 0", "true");
+
+    query(fun + "('db/test')", "");
+    query("count(collection('db/test')) eq 0", "true");
   }
\ No newline at end of file
