diff --git a/old_product.java b/new_product.java
index 5597744..6c558c6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,15 @@
 private Item delete(final QueryContext ctx) throws QueryException {
-    final Data data = ctx.resource.data(checkStr(expr[0], ctx), input);
-    final String path = string(checkStr(expr[1], ctx));
+    final String path = path(string(checkStr(expr[0], ctx)));
 
-    final int[] docs = data.doc(path);
+    // the first step of the path should be the database name
+    final int pos = path.indexOf('/');
+    if(pos <= 0) NODB.thrw(input, path);
+    final byte[] db = token(path.substring(0, pos));
+    final Data data = ctx.resource.data(db, input);
+
+    final String trg = path.substring(pos + 1);
+
+    final int[] docs = data.doc(trg);
     for(final int pre : docs)
       ctx.updates.add(new DeleteNode(pre, data, input), ctx);
 
