diff --git a/old_product.java b/new_product.java
index 4cb232c..b12655c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -65,5 +65,5 @@ public static CompletedCheckpoint loadAndValidateSavepoint(
 		// (3) convert to checkpoint so the system can fall back to it
 		CheckpointProperties props = CheckpointProperties.forStandardSavepoint();
 		return new CompletedCheckpoint(jobId, savepoint.getCheckpointId(), 0L, 0L,
-				taskStates, props, metadataHandle, savepointPath);
+				taskStates, savepoint.getMasterStates(), props, metadataHandle, savepointPath);
 	}
\ No newline at end of file
