diff --git a/old_test.java b/new_test.java
index d42ef20..a051768 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,20 +1,14 @@
 @Test
-	public void testGetStatement()
-	{
-		try
-		{
+	public void testGetStatement() {
+		try {
 			getSecuredReifiedStatement().getStatement();
-			if (!securityEvaluator.evaluate(Action.Read))
-			{
-				Assert.fail("Should have thrown AccessDenied Exception");
+			if (!securityEvaluator.evaluate(Action.Read)) {
+				Assert.fail("Should have thrown ReadDeniedException Exception");
 			}
-		}
-		catch (final AccessDeniedException e)
-		{
-			if (securityEvaluator.evaluate(Action.Read))
-			{
+		} catch (final ReadDeniedException e) {
+			if (securityEvaluator.evaluate(Action.Read)) {
 				Assert.fail(String
-						.format("Should not have thrown AccessDenied Exception: %s - %s",
+						.format("Should not have thrown ReadDeniedException Exception: %s - %s",
 								e, e.getTriple()));
 			}
 		}
