diff --git a/old_product.java b/new_product.java
index d23f046..606824d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public boolean disableCompactionForBlobStore(PartitionId id) {
+public boolean controlCompactionForBlobStore(PartitionId id, boolean enabled) {
     DiskManager diskManager = partitionToDiskManager.get(id);
-    return diskManager != null && diskManager.disableCompactionForBlobStore(id);
+    return diskManager != null && diskManager.controlCompactionForBlobStore(id, enabled);
   }
\ No newline at end of file
