diff --git a/old_test.java b/new_test.java
index 139d3e5..bd49800 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
     query(_FILE_COPY.args(PATH2, PATH2));
     query(_FILE_SIZE.args(PATH1), "1");
     query(_FILE_SIZE.args(PATH2), "1");
-    error(_FILE_COPY.args(PATH1, PATH3), Err.FILE_NODIR);
+    error(_FILE_COPY.args(PATH1, PATH3), Err.FILE_ND);
 
     query(_FILE_DELETE.args(PATH1));
     query(_FILE_DELETE.args(PATH2));
