diff --git a/old_test.java b/new_test.java
index c9babc5..790c158 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,5 +2,5 @@
     public void test_minus_MinusAdjuster() {
         MockSimplePeriod period = MockSimplePeriod.of(7, ChronoUnit.MINUTES);
         OffsetTime t = TEST_11_30_59_500_PONE.minus(period);
-        assertEquals(t, OffsetTime.of(11, 23, 59, 500, OFFSET_PONE));
+        assertEquals(t, OffsetTime.of(LocalTime.of(11, 23, 59, 500), OFFSET_PONE));
     }
\ No newline at end of file
