diff --git a/old_product.java b/new_product.java
index 0a6699e..0d8520c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -61,8 +61,7 @@ boolean parse() throws QueryException {
         // serialization parameters
         final String key = string(local);
         final String val = toString(value, name);
-        if(output.get(key) == null) error(info, UNKNOWN_SER, key);
-        output.set(key, val);
+        if(output.set(key, val) == null) error(info, UNKNOWN_SER, key);
       }
       found |= rexq;
     }
