diff --git a/old_test.java b/new_test.java
index 84e12d5..11a66a4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
   public void parse() throws Exception {
-    context.options.set(Options.STRIPNS, true);
-    context.options.set(Options.SERIALIZER, "indent=no");
+    context.options.set(MainOptions.STRIPNS, true);
+    context.options.set(MainOptions.SERIALIZER, "indent=no");
 
     final String doc = "<e xmlns='A'><b:f xmlns:b='B'/></e>";
     for(final boolean b : new boolean[] { false, true }) {
-      context.options.set(Options.INTPARSE, b);
+      context.options.set(MainOptions.INTPARSE, b);
       new CreateDB(NAME, doc).execute(context);
       String result = new XQuery(".").execute(context);
       assertEquals("<e><f/></e>", result);
