diff --git a/old_product.java b/new_product.java
index ff51760..2becfe6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,9 +2,6 @@ protected boolean shouldUpdate(final long lastChecked, final long now, final Dat
             String currentVersion) throws UpdateException {
         //check every 30 days if we know there is an update, otherwise check every 7 days
         int checkRange = 30;
-        if (updateToVersion == null || updateToVersion.isEmpty()) {
-            checkRange = 7;
-        }
         if (!DateUtil.withinDateRange(lastChecked, now, checkRange)) {
             LOGGER.debug("Checking web for new version.");
             final String currentRelease = getCurrentReleaseVersion();
