diff --git a/old_test.java b/new_test.java
index 06f5ebf..1bcf55e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -77,7 +77,7 @@
         updateToVersion = "";
         currentVersion = "1.2.5";
         lastChecked = df.parse("2014-12-01").getTime();
-        now = df.parse("2014-12-08").getTime();
+        now = df.parse("2015-12-08").getTime();
         expResult = true;
         instance.setUpdateToVersion(updateToVersion);
         result = instance.shouldUpdate(lastChecked, now, properties, currentVersion);
