diff --git a/old_test.java b/new_test.java
index 5a061e7..a1a64df 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,7 +12,7 @@
 		u.setUsername("bwolfe");
 		u.getPerson().setGender("M");
 		
-		User createdUser = us.saveUser(u, "Openmr5xy");
+		User createdUser = us.createUser(u, "Openmr5xy");
 		
 		// if we're returning the object from create methods, check validity
 		assertTrue("The user returned by the create user method should equal the passed in user", createdUser.equals(u));
