diff --git a/old_product.java b/new_product.java
index d895507..58cdb83 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,2 @@
-@Deprecated
-	@Authorized( { PrivilegeConstants.ADD_USERS, PrivilegeConstants.EDIT_USERS })
-	@Logging(ignoredArgumentIndexes = { 1 })
-	public User saveUser(User user, String password) throws APIException;
\ No newline at end of file
+@Authorized( { PrivilegeConstants.EDIT_USERS })
+	public User saveUser(User user) throws APIException;
\ No newline at end of file
