diff --git a/old_product.java b/new_product.java
index eb5449c..884f81a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-HandlerResult authenticate(Credentials credential) throws GeneralSecurityException, PreventedException;
\ No newline at end of file
+HandlerResult authenticate(Credential credential) throws GeneralSecurityException, PreventedException;
\ No newline at end of file
