diff --git a/old_test.java b/new_test.java
index 7c6cdb6..aed34b8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testAuthenticate() {
-        LegacyCasTrustedCredentials trustedCredentials = new LegacyCasTrustedCredentials();
+        LegacyCasTrustedCredential trustedCredentials = new LegacyCasTrustedCredential();
 
         ServletRequest request = new MockHttpServletRequest();
 
