diff --git a/old_product.java b/new_product.java
index 0022bbd..de32a3a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,10 @@
 private Item event(final QueryContext qc) throws QueryException {
-    final byte[] name = checkStr(exprs[0], qc);
+    final byte[] name = toToken(exprs[0], qc);
     try {
       final ArrayOutput ao = qc.value(exprs[1]).serialize(SerializerOptions.get(false));
       // throw exception if event is unknown
-      if(!qc.context.events.notify(qc.context, name, ao.finish())) throw BXDB_EVENT.get(info, name);
+      if(!qc.context.events.notify(qc.context, name, ao.finish()))
+        throw BXDB_EVENT_X.get(info, name);
       return null;
     } catch(final QueryIOException ex) {
       throw ex.getCause(info);
