diff --git a/old_product.java b/new_product.java
index 9246c01..d5922fd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-FullHttpResponse handleRequest(Map<String, String> pathParams, Map<String, String> queryParams, JobManagerGateway jobManagerGateway) throws Exception;
\ No newline at end of file
+CompletableFuture<FullHttpResponse> handleRequest(Map<String, String> pathParams, Map<String, String> queryParams, JobManagerGateway jobManagerGateway);
\ No newline at end of file
