diff --git a/old_product.java b/new_product.java
index 3df60a4..64115e2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public Object eval(final ScriptObject initialScope, final String string, final O
         final String  file       = (location == UNDEFINED || location == null) ? "<eval>" : location.toString();
         final Source  source     = new Source(file, string);
         final boolean directEval = location != UNDEFINED; // is this direct 'eval' call or indirectly invoked eval?
-        final ScriptObject global = Context.getGlobalTrusted();
+        final Global  global = Context.getGlobal();
 
         ScriptObject scope = initialScope;
 
@@ -34,7 +34,7 @@ public Object eval(final ScriptObject initialScope, final String string, final O
         // in the caller's environment. A new environment is created!
         if (strictFlag) {
             // Create a new scope object
-            final ScriptObject strictEvalScope = ((GlobalObject)global).newObject();
+            final ScriptObject strictEvalScope = global.newObject();
 
             // bless it as a "scope"
             strictEvalScope.setIsScope();
