diff --git a/old_test.java b/new_test.java
index 6fc799e..f96ce2c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
         final Options options = new Options("");
         final ErrorManager errors = new ErrorManager();
         final Context cx = new Context(options, errors, Thread.currentThread().getContextClassLoader());
-        final ScriptObject oldGlobal = Context.getGlobal();
+        final Global oldGlobal = Context.getGlobal();
         Context.setGlobal(cx.createGlobal());
         try {
             String code = "22 + 10";
