diff --git a/old_test.java b/new_test.java
index e325257..76a83f5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,5 +3,5 @@
     query(_UNIT_ASSERT.args("1"), "");
     query(_UNIT_ASSERT.args("(<a/>,<b/>)"), "");
     error(_UNIT_ASSERT.args("()"), Err.UNIT_ASSERT);
-    error(_UNIT_ASSERT.args("()", "X"), Err.UNIT_MESSAGE);
+    error(_UNIT_ASSERT.args("()", "X"), Err.UNIT_MESSAGE_X);
   }
\ No newline at end of file
