diff --git a/old_product.java b/new_product.java
index 6761960..de07145 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 private Item assrt(final QueryContext qc) throws QueryException {
-    final Item it = exprs.length < 2 ? null : checkItem(exprs[1], qc);
+    final Item it = exprs.length < 2 ? null : toItem(exprs[1], qc);
     if(exprs[0].ebv(qc, info).bool(info)) return null;
     throw error(it);
   }
\ No newline at end of file
