diff --git a/old_test.java b/new_test.java
index 4b704a5..6837b3d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,6 @@
    {
       System.out.println("iterator");
       AbstractGraphRow instance = new AbstractGraphRowImpl();
-      Iterator expResult = null;
       Iterator result = instance.iterator();
-      assertEquals(expResult, result);
+      assertNotNull(result);
    }
\ No newline at end of file
