diff --git a/old_product.java b/new_product.java
index f137d74..c5f98d5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -50,8 +50,7 @@ public WeldContainer initialize() {
         bootstrap.endInitialization();
 
         // Set up the ShutdownManager for later
-        this.shutdownManager = getInstanceByType(bootstrap.getManager(deployment.loadBeanDeploymentArchive(ShutdownManager.class)), ShutdownManager.class);
-        this.shutdownManager.setBootstrap(bootstrap);
+        this.shutdownManager = new ShutdownManager(bootstrap);
 
         WeldContainer container = getInstanceByType(bootstrap.getManager(deployment.loadBeanDeploymentArchive(WeldContainer.class)), WeldContainer.class);
 
