diff --git a/old_test.java b/new_test.java
index 1bd6e0a..40c6d80 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,15 @@
 @Test
     public void testInitialize() {
 
-        WeldContainer weld = new Weld().initialize();
+        Weld weld = new Weld();
+        WeldContainer container = weld.initialize();
 
-        MainTestBean mainTestBean = weld.instance().select(MainTestBean.class).get();
+        MainTestBean mainTestBean = container.instance().select(MainTestBean.class).get();
         assertNotNull(mainTestBean);
 
         ParametersTestBean paramsBean = mainTestBean.getParametersTestBean();
         assertNotNull(paramsBean);
         assertNotNull(paramsBean.getParameters());
 
-        shutdownManager(weld);
+        weld.shutdown();
     }
\ No newline at end of file
