diff --git a/old_product.java b/new_product.java
index 98edf06..84463c5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,9 @@
 void usage(boolean error) {
     PrintStream out = (error ? System.err : System.out);
     String name = "data-client";
-    if (System.getProperty("script")!=null) name = System.getProperty("script").replaceAll("[./]", "");
+    if (System.getProperty("script") != null) {
+      name = System.getProperty("script").replaceAll("[./]", "");
+    }
     Copyright.print(out);
     out.println("Usage: ");
     out.println("  " + name + " clear ( --all | --queues | --streams | --tables | --meta)");
