diff --git a/old_test.java b/new_test.java
index 734f899..d923f92 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,5 +3,5 @@
     // check if the info string is not empty
     query("let $a := " + conn() + " return (" +
         _CLIENT_EXECUTE.args("$a", "xquery 123") + " and " +
-        _CLIENT_INFO.args("$a") + ")", "true");
+        _CLIENT_INFO.args("$a") + ')', "true");
   }
\ No newline at end of file
