diff --git a/old_product.java b/new_product.java
index 766f047..441cf06 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public synchronized String info() {
     final TokenBuilder tb = new TokenBuilder();
-    tb.addExt(EVENTS_X, size()).add(size() != 0 ? COL : DOT);
+    tb.addExt(EVENTS_X, size()).add(size() == 0 ? DOT : COL);
 
     final String[] names = keySet().toArray(new String[size()]);
     Arrays.sort(names);
