diff --git a/old_test.java b/new_test.java
index 399b80d..ef6e431 100644
--- a/old_test.java
+++ b/new_test.java
@@ -36,10 +36,26 @@
                     break;
                 }
 
+                Assertions.assertEquals(401, response.getStatus());
+
+                // security package requested is one negotiate continues with
                 Assertions.assertTrue(response.getHeader("WWW-Authenticate").startsWith(securityPackage + " "));
+
+                // keep-alive, NTLM is a connection-oriented protocol
                 Assertions.assertEquals("keep-alive", response.getHeader("Connection"));
-                Assertions.assertEquals(2, response.getHeaderNamesSize());
-                Assertions.assertEquals(401, response.getStatus());
+
+                // Connection: keep-alive
+                // WWW-Authenticate: ...
+                // Content-Length: ...
+                Assertions.assertEquals(3, response.getHeaderNamesSize());
+
+                // response has a body and a content length (.NET clients require this)
+                int contentLength = Integer.parseInt(response.getHeader("Content-Length"));
+                Assertions.assertTrue(contentLength > 0);
+                String content = response.getOutputText();
+                Assertions.assertEquals(contentLength, content.length());
+
+                // continue token
                 final String continueToken = response.getHeader("WWW-Authenticate")
                         .substring(securityPackage.length() + 1);
                 final byte[] continueTokenBytes = Base64.getDecoder().decode(continueToken);
