diff --git a/old_product.java b/new_product.java
index 414bc15..6659f88 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,12 +6,10 @@ protected void putConnectionBackInPartition(ConnectionHandle connectionHandle) t
 			connectionHandle.inUseInThreadLocalContext.set(false);
 			((CachedConnectionStrategy)this.connectionStrategy).tlConnections.set(connectionHandle);
 		} else {
-			TransferQueue<ConnectionHandle> queue = connectionHandle.getOriginatingPartition().getFreeConnections();
-			if (!queue.tryTransfer(connectionHandle)){
-				if (!queue.offer(connectionHandle)){
+			BlockingQueue<ConnectionHandle> queue = connectionHandle.getOriginatingPartition().getFreeConnections();
+				if (!queue.offer(connectionHandle)){ // this shouldn't fail
 					connectionHandle.internalClose();
 				}
-			}
 		}
 
 
