diff --git a/old_test.java b/new_test.java
index a0931bf..42ac2c4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,8 @@
   public void close() {
     query(conn() + " ! " + _CLIENT_CLOSE.args(" ."));
     // BXCL0002: session not available
-    error(_CLIENT_CLOSE.args("xs:anyURI('unknown')"), Err.BXCL_NOTAVL);
+    error(_CLIENT_CLOSE.args("xs:anyURI('unknown')"), Err.BXCL_NOTAVL_X);
     // BXCL0002: session has already been closed
     error(conn() + " ! (" + _CLIENT_CLOSE.args(" .") + ", " + _CLIENT_CLOSE.args(" .") + ')',
-        Err.BXCL_NOTAVL);
+        Err.BXCL_NOTAVL_X);
   }
\ No newline at end of file
