diff --git a/old_product.java b/new_product.java
index 6855353..9c92847 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,6 @@ private Item close(final QueryContext qc) throws QueryException {
       session(qc, true).close();
       return null;
     } catch(final IOException ex) {
-      throw BXCL_COMMAND.get(info, ex);
+      throw BXCL_COMMAND_X.get(info, ex);
     }
   }
\ No newline at end of file
