diff --git a/old_product.java b/new_product.java
index 8a25c83..2c4ac77 100644
--- a/old_product.java
+++ b/new_product.java
@@ -32,6 +32,6 @@ public void validate(Object obj, Errors errors) {
 			}
 			validateFieldsForOutpatientCareSettingType(order, errors);
 			validatePairedFields(order, errors);
-			validateUnitsConceptClassIsUnitsOfMeasure(errors, order);
+			validateUnitsAreAmongAllowedConcepts(errors, order);
 		}
 	}
\ No newline at end of file
