diff --git a/old_test.java b/new_test.java
index e031ba7..4408562 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,7 @@
 @Test
-	public void validate_shouldFailValidationIfClassOfQuantityUnitsdoseUnitsOrDurationUnitsIsNotUnitsOfMeasureConceptClass()
-	        throws Exception {
+	public void validate_shouldFailValidationIfDoseUnitsIsNotADoseUnitConcept() throws Exception {
 		Concept concept = Context.getConceptService().getConcept(3);
-		Assert.assertFalse(concept.getConceptClass().getUuid().equals(ConceptClass.UNIT_OF_MEASUREMENT_UUID));
+		assertThat(concept, not(isIn(Context.getOrderService().getDrugDosingUnits())));
 		
 		DrugOrder order = new DrugOrder();
 		order.setDosingType(DrugOrder.DosingType.FREE_TEXT);
@@ -16,6 +15,4 @@
 		Errors errors = new BindException(order, "order");
 		new DrugOrderValidator().validate(order, errors);
 		Assert.assertTrue(errors.hasFieldErrors("doseUnits"));
-		Assert.assertTrue(errors.hasFieldErrors("quantityUnits"));
-		Assert.assertTrue(errors.hasFieldErrors("durationUnits"));
 	}
\ No newline at end of file
