diff --git a/old_product.java b/new_product.java
index 518d3bf..3df2024 100644
--- a/old_product.java
+++ b/new_product.java
@@ -244,12 +244,8 @@ public static GridCoverage2D reproject(GridCoverage2D            sourceCoverage,
             /*
              * Computes the final transform.
              */
-            if (step1.equals(step3.inverse())) {
-                allSteps = step2;
-            } else {
-                allSteps = mtFactory.createConcatenatedTransform(
-                           mtFactory.createConcatenatedTransform(step1, step2), step3);
-            }
+            allSteps = mtFactory.createConcatenatedTransform(
+                       mtFactory.createConcatenatedTransform(step1, step2), step3);
         }
         allSteps2D = toMathTransform2D(allSteps, mtFactory, targetGG);
         if (!(allSteps2D instanceof MathTransform2D)) {
