diff --git a/old_product.java b/new_product.java
index dc26d47..2ff6a83 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,6 @@
 private Str dirName(final QueryContext ctx) throws QueryException {
-    final String path = checkFile(0, ctx).getParent();
-    return Str.get(dir(path == null ? "." : path));
+    final File file = checkFile(0, ctx);
+    String par = file.getParent();
+    if(par == null) par = file.toString().contains(File.separator) ? "" : ".";
+    return Str.get(dir(par));
   }
\ No newline at end of file
