diff --git a/old_test.java b/new_test.java
index f7ff2d9..523ecbd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,4 +7,6 @@
     query(_FILE_DIR_NAME.args(""), "." + File.separator);
     // check with a path without directory separators
     query(_FILE_DIR_NAME.args(NAME), "." + File.separator);
+    // check with a path without directory separators
+    query(_FILE_DIR_NAME.args("/"), File.separator);
   }
\ No newline at end of file
