diff --git a/old_test.java b/new_test.java
index ab2b4ce..d4f72aa 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
 
         versions.addEvidence("util", "version", "33.3", Confidence.HIGHEST);
         versions.addEvidence("other", "version", "alpha", Confidence.HIGHEST);
-        versions.addEvidence("manifest", "implementation-version", "1.2.3", Confidence.HIGHEST);
+        versions.addEvidence("other", "Implementation-Version", "1.2.3", Confidence.HIGHEST);
 
         VersionFilterAnalyzer instance = new VersionFilterAnalyzer();
 
@@ -21,13 +21,17 @@
         instance.analyzeDependency(dependency, null);
         assertEquals(2, versions.size());
 
+        versions.addEvidence("Manifest", "Implementation-Version", "1.2.3", Confidence.HIGHEST);
+        instance.analyzeDependency(dependency, null);
+        assertEquals(3, versions.size());
+
         versions.addEvidence("nexus", "version", "1.2.3", Confidence.HIGHEST);
         versions.addEvidence("other", "version", "alpha", Confidence.HIGHEST);
         instance.analyzeDependency(dependency, null);
-        assertEquals(3, versions.size());
+        assertEquals(4, versions.size());
 
         versions.addEvidence("central", "version", "1.2.3", Confidence.HIGHEST);
         versions.addEvidence("other", "version", "alpha", Confidence.HIGHEST);
         instance.analyzeDependency(dependency, null);
-        assertEquals(4, versions.size());
+        assertEquals(5, versions.size());
     }
\ No newline at end of file
