diff --git a/old_product.java b/new_product.java
index d6ef118..8735686 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
-public LocalDateTime withDate(int year, MonthOfYear monthOfYear, int dayOfMonth) {
+public LocalDateTime withDate(int year, Month month, int dayOfMonth) {
         if (year == getYear() &&
-                monthOfYear == getMonthOfYear() &&
+                month == getMonthOfYear() &&
                 dayOfMonth == getDayOfMonth()) {
             return this;
         }
-        LocalDate newDate = LocalDate.of(year, monthOfYear, dayOfMonth);
+        LocalDate newDate = LocalDate.of(year, month, dayOfMonth);
         return with(newDate, time);
     }
\ No newline at end of file
