diff --git a/old_product.java b/new_product.java
index a58be98..ad0ab1f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
 @Authorized( { PrivilegeConstants.EDIT_ORDERS, PrivilegeConstants.ADD_ORDERS })
-	public Order saveOrder(Order order) throws APIException;
\ No newline at end of file
+	public Order saveOrder(Order order, OrderContext orderContext) throws APIException;
\ No newline at end of file
