diff --git a/old_test.java b/new_test.java
index f5a9bf5..5b5a483 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
 		QrCodeEncoder encoder = new QrCodeEncoder();
 		encoder.setVersion(1).setError(QrCode.ErrorLevel.M).
 				setMask(QrCodeMaskPattern.M011).
-				kanji("阿ん鞠ぷへ≦Ｋ").fixate();
+				addKanji("阿ん鞠ぷへ≦Ｋ").fixate();
 
 		byte expected[] = new byte[]{
 				0x01,0x4E,(byte)0x8B,(byte)0xA0,0x23,
