diff --git a/old_product.java b/new_product.java
index 1be3b36..2435141 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,20 +1,13 @@
-public QrCodeEncoder kanji(String message) {
+private void encodeKanji( byte []bytes , int length ) {
 		qr.mode = QrCode.Mode.KANJI;
 
 		int lengthBits = getLengthBitsKanji(qr.version);
 
-		byte[] bytes;
-		try {
-			bytes = message.getBytes("Shift_JIS");
-		} catch (UnsupportedEncodingException ex) {
-			throw new IllegalArgumentException(ex);
-		}
-
 		// specify the mode
 		packed.append(0b1000, 4, false);
 
 		// Specify the number of characters
-		packed.append(message.length(), lengthBits, false);
+		packed.append(length, lengthBits, false);
 
 		for (int i = 0; i < bytes.length; i += 2) {
 			int byte1 = bytes[i] & 0xFF;
@@ -26,11 +19,9 @@ public QrCodeEncoder kanji(String message) {
 			} else if (code >= 0xe040 && code <= 0xebbf) {
 				adjusted = code - 0xc140;
 			} else {
-				throw new IllegalArgumentException("Invalid byte sequence. " + message.charAt(i / 2));
+				throw new IllegalArgumentException("Invalid byte sequence. At " +(i / 2));
 			}
 			int encoded = ((adjusted >> 8) * 0xc0) + (adjusted & 0xff);
 			packed.append(encoded, 13, false);
 		}
-
-		return this;
 	}
\ No newline at end of file
