diff --git a/old_test.java b/new_test.java
index bb446c8..a217738 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,9 +5,12 @@
         assertThat("testIsReadOnly 0", context.isReadOnly(), equalTo(false));
         
         // Create a new read-only context
-        Context instance = new Context(Context.READ_ONLY);
+        Context instance = new Context(Context.Mode.READ_ONLY);
         assertThat("testIsReadOnly 1", instance.isReadOnly(), equalTo(true));
 
+        //When in read-only, we only support abort().
+        instance.abort();
+
         // Cleanup our context
         cleanupContext(instance);
     }
\ No newline at end of file
