diff --git a/old_test.java b/new_test.java
index 5827944..22ae91d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(expectedExceptions=NullPointerException.class)
     public void test_print_nullAppendable() throws Exception {
-        ZoneOffsetPrinterParser pp = new ZoneOffsetPrinterParser("Z", true, true);
+        ZoneOffsetPrinterParser pp = new ZoneOffsetPrinterParser("Z", "+HH:MM:ss");
         Calendrical cal = OFFSET_0130;
         pp.print(cal, (Appendable) null, symbols);
     }
\ No newline at end of file
