diff --git a/old_product.java b/new_product.java
index 479f491..01294ab 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
 private synchronized Item delete(final QueryContext ctx) throws QueryException {
     final File path = checkFile(0, ctx);
-    if(!path.exists()) throw FILE_WHICH.get(info, path.getAbsolutePath());
+    if(!path.exists()) throw FILE_NF.get(info, path.getAbsolutePath());
     if(optionalBool(1, ctx)) {
       deleteRec(path);
     } else if(!path.delete()) {
-      throw (path.isDirectory() ? FILE_NEDIR : FILE_DEL).get(info, path);
+      throw (path.isDirectory() ? FILE_ID_FULL : FILE_IE_DEL).get(info, path);
     }
     return null;
   }
\ No newline at end of file
