diff --git a/old_product.java b/new_product.java
index 3180b97..b5a0499 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@
                 @Override
                 public void run() {
                     final Response result =
-                            Response.created(getUriBuilder().segment("responseFilterAsyncExecutor").build()).build();
+                            Response.created(URI.create("responseFilterAsyncExecutor")).build();
                     asyncResponse.resume(result);
                 }
             });
