diff --git a/old_test.java b/new_test.java
index 714d73c..a1ab4d1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
     public void testResponseFilterAsyncExecutor() {
-        checkResponseFilter("ResponseTest/responseFilterAsyncExecutor");
+        checkResponseFilter("ResponseTest/responseFilterAsyncExecutor", "responseFilterAsyncExecutor");
     }
\ No newline at end of file
