diff --git a/old_product.java b/new_product.java
index e9c4fed..fb34987 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1,2 @@
-public void setWorkflowGroup(Collection collection, int step, Group group);
\ No newline at end of file
+public void setWorkflowGroup(Context context, Collection collection, int step, Group group)
+            throws SQLException, AuthorizeException;
\ No newline at end of file
