diff --git a/old_test.java b/new_test.java
index 44f1eb2..eeb30a6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
         int step = 1;
         Group g = groupService.create(context);
         context.restoreAuthSystemState();
-        collection.setWorkflowGroup(step, g);
+        collection.setWorkflowGroup(context, step, g);
         assertThat("testSetWorkflowGroup 0",collectionService.getWorkflowGroup(collection, step), notNullValue());
         assertThat("testSetWorkflowGroup 1",collectionService.getWorkflowGroup(collection, step), equalTo(g));
     }
\ No newline at end of file
