diff --git a/old_test.java b/new_test.java
index 6af7ede..c30d204 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,12 +3,7 @@
 		// replay
 		Context.logout();
 		Context.authenticate("admin", "test");
-		String userUuid = Context.getAuthenticatedUser().getUuid();
-		
-		Context.logout();
-		Authenticated authenticated = Context.authenticate(new UsernamePasswordAuthenticationScheme(), new UsernamePasswordCredentials("admin", "test"));
 		
 		// verif
-		Assert.assertEquals(UsernamePasswordCredentials.SCHEME, authenticated.getAuthenticationScheme());
-		Assert.assertEquals(userUuid, Context.getAuthenticatedUser().getUuid());
+		Assert.assertEquals("admin", Context.getAuthenticatedUser().getUsername());
 	}
\ No newline at end of file
