diff --git a/old_product.java b/new_product.java
index 03aba1d..cc4be95 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,23 +1,14 @@
-public static Authenticated authenticate(AuthenticationScheme authenticationScheme, Credentials credentials)
-			throws ContextAuthenticationException {
-		
+public static Authenticated authenticate(Credentials credentials) throws ContextAuthenticationException {
+
 		if (Daemon.isDaemonThread()) {
 			log.error("Authentication attempted while operating on a "
 					+ "daemon thread, authenticating is not necessary or allowed");
 			return new BasicAuthenticated(Daemon.getDaemonThreadUser(), "No auth scheme used by Context - Daemon user is always authenticated.");
 		}
-		
+
 		if (credentials == null) {
 			throw new ContextAuthenticationException("Context cannot authenticate with null credentials.");
 		}
-			
-		if (authenticationScheme == null) {
-			throw new ContextAuthenticationException("Context cannot authenticate with null authentication scheme.");
-		}
-		
-		if (authenticationScheme instanceof DaoAuthenticationScheme) {
-			((DaoAuthenticationScheme) authenticationScheme).setContextDao(getContextDAO());
-		}
-		
-		return getUserContext().authenticate(authenticationScheme, credentials);
+
+		return getUserContext().authenticate(credentials);
 	}
\ No newline at end of file
