diff --git a/old_product.java b/new_product.java
index cb8f5e5..9c71156 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
 public static byte[] expandLabel(String macAlgorithm, byte[] prk, String labelIn, byte[] hashValue, int outLen) {
         byte[] info = labelEncoder(hashValue, labelIn, outLen);
-        byte[] result = expand(macAlgorithm, prk, info, outLen);
-        return result;
+        return expand(macAlgorithm, prk, info, outLen);
     }
\ No newline at end of file
