diff --git a/old_product.java b/new_product.java
index 2c6a1cd..3ff6db3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,8 +2,8 @@ protected static String post(final String query, final String request, final Str
       throws IOException {
 
     // create connection
-    final URL url = new URL(root + query);
-    final HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+    final IOUrl url = new IOUrl(root + query);
+    final HttpURLConnection conn = (HttpURLConnection) url.connection();
     conn.setDoOutput(true);
     conn.setRequestMethod(POST.name());
     conn.setRequestProperty(HttpText.CONTENT_TYPE, type);
