diff --git a/old_product.java b/new_product.java
index c5bd83d..eb19d74 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-@Override public <K, V, C extends Serializable> IgniteModel<I, O> update(IgniteModel<I, O> mdl,
+@Override public <K, V> IgniteModel<I, O> update(IgniteModel<I, O> mdl,
                 DatasetBuilder<K, V> datasetBuilder,
-                Vectorizer<K, V, C, L> extractor) {
+                Preprocessor<K, V> extractor) {
                 DatasetTrainer<IgniteModel<I, O>, L> trainer1 = (DatasetTrainer<IgniteModel<I, O>, L>)trainer;
                 return trainer1.update(mdl, datasetBuilder, extractor);
             }
\ No newline at end of file
