diff --git a/old_test.java b/new_test.java
index a049d87..34fe757 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,7 +20,7 @@
 
         LinearRegressionModel originalMdl = trainer.withSeed(0).fit(
             data, parts,
-            new ArraysVectorizer<Integer>().labeled(Vectorizer.LabelCoordinate.LAST)
+            new DoubleArrayVectorizer<Integer>().labeled(Vectorizer.LabelCoordinate.LAST)
         );
 
 
@@ -28,14 +28,14 @@
             originalMdl,
             data,
             parts,
-            new ArraysVectorizer<Integer>().labeled(Vectorizer.LabelCoordinate.LAST)
+            new DoubleArrayVectorizer<Integer>().labeled(Vectorizer.LabelCoordinate.LAST)
         );
 
         LinearRegressionModel updatedOnEmptyDS = trainer.withSeed(0).update(
             originalMdl,
             new HashMap<>(),
             parts,
-            new ArraysVectorizer<Integer>().labeled(Vectorizer.LabelCoordinate.LAST)
+            new DoubleArrayVectorizer<Integer>().labeled(Vectorizer.LabelCoordinate.LAST)
         );
 
         assertArrayEquals(
