diff --git a/old_test.java b/new_test.java
index 0072899..fd3abec 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,7 @@
 @Test
     public void testReadFile() {
         List<String> thisFile = null;
-        try {
-            thisFile = FileUtil.readFile(THISCLASS);
-        } catch (IOException e) {
-            e.printStackTrace();
-            return;
-        }
+        thisFile = FileUtil.readFile(THISCLASS);
         // Comment ONE line
         int lineOne = 0;
         // Comment TWO line
