diff --git a/old_test.java b/new_test.java
index 683c7c6..853d01e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,8 @@
 				.build();
 		
 		assertEquals(null, param.getSize());
-		assertTrue(Double.compare(0.5, param.getScalingFactor()) == 0);
+		assertTrue(Double.compare(0.5, param.getWidthScalingFactor()) == 0);
+		assertTrue(Double.compare(0.5, param.getHeightScalingFactor()) == 0);
 		assertEquals(ThumbnailParameter.ORIGINAL_FORMAT, param.getOutputFormat());
 		assertEquals(ThumbnailParameter.DEFAULT_FORMAT_TYPE, param.getOutputFormatType());
 		assertTrue(Float.isNaN(param.getOutputQuality()));
