diff --git a/old_product.java b/new_product.java
index 1fd86cd..44b4bd5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,11 @@
 public ThumbnailParameter build()
 	{
-		if (!Double.isNaN(scalingFactor))
+		if (!Double.isNaN(widthScalingFactor))
 		{
 			// If scaling factor has been set.
 			return new ThumbnailParameter(
-					scalingFactor,
+					widthScalingFactor,
+					heightScalingFactor,
 					sourceRegion,
 					keepAspectRatio,
 					thumbnailFormat,
