diff --git a/old_product.java b/new_product.java
index c087219..417c0a4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
             Class<FS> featureSelectorClass, 
             FS.AbstractTrainingParameters featureSelectorTrainingParameters,
             double expectedF1score) {
-        Configuration conf = TestUtils.getConfig();
+        Configuration conf = Configuration.getConfiguration();
         
         
         String dbName = this.getClass().getSimpleName();
