diff --git a/old_product.java b/new_product.java
index 939f9e2..714c6d9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public double get(long key) {
-        final int idx = findIndex(key);
-        if (idx >= 0 && usedKeys.get(idx)) {
+        final int idx = keys.getIndexIfActive(key);
+        if (idx >= 0) {
             return values[idx];
         } else {
             throw new IllegalArgumentException("Key " + key + " is not in the key set");
