diff --git a/old_product.java b/new_product.java
index 08dcb55..3f5a8db 100644
--- a/old_product.java
+++ b/new_product.java
@@ -18,13 +18,19 @@ public boolean process( T input , ImageUInt8 binary ) {
 			if( grid.rows == numRows && grid.columns == numCols ) {
 				// this detector requires that the (0,0) grid cell has a square inside of it
 				if( grid.get(0,0) == null ){
-					if( grid.get(0,grid.columns-1) != null ) {
+					if( grid.get(0,-1) != null ) {
 						tools.flipColumns(grid);
+					} else if( grid.get(-1,0) != null ) {
+						tools.flipRows(grid);
 					} else {
 						continue;
 					}
 				}
 
+				// make sure its in the expected orientation
+				if( !ensureCCW(grid) )
+					continue;
+
 				// If symmetric, ensure that the (0,0) is closest to top-left image corner
 				putIntoCanonical(grid);
 
