diff --git a/old_product.java b/new_product.java
index 6daec7d..9dbd6cd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,5 @@
 @Override
   public int mkdir(final String path, final int mode) {
-    try {
-      if(!isDir(mode)) return -1;
-      return insert(path, mode) == -1 ? -1 : 0;
-    } catch(QueryException e) {
-      e.printStackTrace();
-      return -1;
-    }
+    if(!isDir(mode)) return -1;
+    return createFile(path, mode);
   }
\ No newline at end of file
