diff --git a/old_test.java b/new_test.java
index 2953ae7..4e3946e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,13 +5,16 @@
 
     Date start = new Date();
 
-    CreateDialogNodeOptions createOptions =
-            new CreateDialogNodeOptions.Builder(workspaceId, dialogNodeName).description(dialogNodeDescription).build();
+    CreateDialogNodeOptions createOptions = new CreateDialogNodeOptions.Builder(workspaceId, dialogNodeName)
+        .description(dialogNodeDescription).build();
     service.createDialogNode(createOptions).execute();
 
     try {
-      GetDialogNodeOptions getOptions =
-              new GetDialogNodeOptions.Builder().workspaceId(workspaceId).dialogNode(dialogNodeName).build();
+      GetDialogNodeOptions getOptions = new GetDialogNodeOptions.Builder()
+          .workspaceId(workspaceId)
+          .dialogNode(dialogNodeName)
+          .includeAudit(true)
+          .build();
       DialogNode response = service.getDialogNode(getOptions).execute();
       assertNotNull(response);
       assertNotNull(response.getDialogNodeId());
