diff --git a/old_product.java b/new_product.java
index 6fabd3d..a7b51f8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,5 +3,8 @@ public ServiceCall<DialogNode> getDialogNode(GetDialogNodeOptions getDialogNodeO
     RequestBuilder builder = RequestBuilder.get(String.format("/v1/workspaces/%s/dialog_nodes/%s", getDialogNodeOptions
         .workspaceId(), getDialogNodeOptions.dialogNode()));
     builder.query(VERSION, versionDate);
+    if (getDialogNodeOptions.includeAudit() != null) {
+      builder.query("include_audit", String.valueOf(getDialogNodeOptions.includeAudit()));
+    }
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(DialogNode.class));
   }
\ No newline at end of file
