diff --git a/old_product.java b/new_product.java
index 4181474..305ae2e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,7 @@
 @Override
     public void addDependenciesForCommand(
-        @NotBlank(message = "No command id entered. Unable to add dependencies.")
-        final String id,
-        @NotEmpty(message = "No dependencies entered. Unable to add dependencies.")
-        final Set<String> dependencies
+        @NotBlank(message = "No command id entered. Unable to add dependencies.") final String id,
+        @NotEmpty(message = "No dependencies entered. Unable to add dependencies.") final Set<String> dependencies
     ) throws GenieException {
-        this.findCommand(id).getDependencies().addAll(dependencies);
+        this.findCommand(id).getDependencies().addAll(this.createAndGetFileEntities(dependencies));
     }
\ No newline at end of file
