diff --git a/old_test.java b/new_test.java
index fe0029a..6049162 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(expected = GenieNotFoundException.class)
     public void testAddDependenciesForCommand() throws GenieException {
         final String id = UUID.randomUUID().toString();
-        Mockito.when(this.jpaCommandRepository.findOne(id)).thenReturn(null);
+        Mockito.when(this.jpaCommandRepository.findByUniqueId(id)).thenReturn(Optional.empty());
         this.service.addDependenciesForCommand(id, Sets.newHashSet());
     }
\ No newline at end of file
