diff --git a/old_test.java b/new_test.java
index a9633f5..5136994 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,9 @@
 @Test public void head() throws Exception {
     // correct return type
-    head("declare %R:HEAD %R:path('') function m:f() { <R:response/> };", "", "");
-    head("declare %R:HEAD %R:path('') function m:f() as element(R:response) " +
-        "{ <R:response/> };", "", "");
+    headR("declare %R:HEAD %R:path('') function m:f() { <R:response/> };");
+    headR("declare %R:HEAD %R:path('') function m:f() as element(R:response) { <R:response/> };");
     // wrong type
-    headE("declare %R:HEAD %R:path('') function m:f() { () };", "");
-    headE("declare %R:HEAD %R:path('') function m:f() { <response/> };", "");
-    headE("declare %R:HEAD %R:path('') function m:f() as element(R:response)* {()};", "");
+    headE("declare %R:HEAD %R:path('') function m:f() { () };");
+    headE("declare %R:HEAD %R:path('') function m:f() { <response/> };");
+    headE("declare %R:HEAD %R:path('') function m:f() as element(R:response)* {()};");
   }
\ No newline at end of file
