diff --git a/old_test.java b/new_test.java
index c27ee52..5c49707 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	public void process_simple() {
-		SquaresIntoCrossClusters alg = new SquaresIntoCrossClusters(5,0.5, -1);
+		SquaresIntoCrossClusters alg = new SquaresIntoCrossClusters(0.05,0.5, -1);
 
 		List<Polygon2D_F64> squares = new ArrayList<Polygon2D_F64>();
 		squares.add( createSquare(7,8));
