diff --git a/old_test.java b/new_test.java
index 3e4787c..b5a9569 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
 
         while ((entry = zstream.getNextEntry()) != null) {
             String expected = readInput(zstream);
-            InputStream sccs = SCCSget.getRevision(sccsfile, entry.getName());
+            InputStream sccs = SCCSget.getRevision("sccs",sccsfile, entry.getName());
             String got = readInput(sccs);
             sccs.close();
             zstream.closeEntry();
