diff --git a/old_product.java b/new_product.java
index 2f05552..a639a16 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,7 +14,7 @@ public InputStream getRevision(String parent, String basename, String rev) throw
             case SCCS :
                 history = SCCSHistoryParser.getSCCSFile(parent, basename);
                 if (history != null && history.canRead()) {
-                    in = SCCSget.getRevision(history, rev);
+                    in = SCCSget.getRevision(SCCSCommand,history, rev);
                     in.mark(32);
                     in.read();
                     in.reset();
