diff --git a/old_test.java b/new_test.java
index 62f1cf2..de5b818 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test
   public void testSynthesize() throws IOException {
-    final String text = "This is an integration test";
-    final File audio = File.createTempFile("tts-audio", "wav");
-
-    synthesize(text, audio);
+    String text = "This is an integration test";
+    InputStream result = service.synthesize(text, Voice.EN_LISA, HttpMediaType.AUDIO_WAV);
+    writeInputStreamToFile(result, File.createTempFile("tts-audio", "wav"));
   }
\ No newline at end of file
