diff --git a/old_product.java b/new_product.java
index f1d93a3..edcc98c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,19 +1,21 @@
-private void synthesize(String text, File audio) {
-    final InputStream is = service.synthesize(text, Voice.EN_LISA, HttpMediaType.AUDIO_WAV);
-    Assert.assertNotNull(is);
+private void writeInputStreamToFile(InputStream inputStream, File audio) {
     OutputStream outStream = null;
     try {
       outStream = new FileOutputStream(audio);
 
-      final byte[] buffer = new byte[8 * 1024];
+      byte[] buffer = new byte[8 * 1024];
       int bytesRead;
-      while ((bytesRead = is.read(buffer)) != -1) {
+      while ((bytesRead = inputStream.read(buffer)) != -1) {
         outStream.write(buffer, 0, bytesRead);
       }
-    } catch (final Exception e) {
+    } catch (Exception e) {
       fail();
     } finally {
-      IOUtils.closeQuietly(is);
-      IOUtils.closeQuietly(outStream);
+      try {
+        inputStream.close();
+        outStream.close();
+      } catch (Exception e) {
+        fail();
+      }
     }
   }
\ No newline at end of file
