diff --git a/old_product.java b/new_product.java
index 7cbffd5..689dcda 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public java.sql.ResultSet getColumnPrivileges(String catalog,
             String schema,
             String table,
-            String col) throws SQLServerException {
+            String col) throws SQLServerException, SQLTimeoutException {
         if (loggerExternal.isLoggable(Level.FINER) && Util.IsActivityTraceOn()) {
             loggerExternal.finer(toString() + " ActivityId: " + ActivityCorrelator.getNext().toString());
         }
