diff --git a/old_test.java b/new_test.java
index 04a584a..a13941a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testGetColumnPrivileges() throws SQLServerException, SQLException {
+    public void testGetColumnPrivileges() throws SQLException {
         DatabaseMetaData databaseMetaData = connection.getMetaData();
         String[] types = {"TABLE"};
         ResultSet rsTables = databaseMetaData.getTables(null, null, "%", types);
