diff --git a/old_test.java b/new_test.java
index c0d399b..98632e7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
     public void testLogFactorial() {
         System.out.println("logFactorial");
-        assertEquals(0.0, MathEx.logFactorial(0), 1E-7);
-        assertEquals(0.0, MathEx.logFactorial(1), 1E-7);
-        assertEquals(Math.log(2.0), MathEx.logFactorial(2), 1E-7);
-        assertEquals(Math.log(6.0), MathEx.logFactorial(3), 1E-7);
-        assertEquals(Math.log(24.0), MathEx.logFactorial(4), 1E-7);
+        assertEquals(0.0, MathEx.lfactorial(0), 1E-7);
+        assertEquals(0.0, MathEx.lfactorial(1), 1E-7);
+        assertEquals(Math.log(2.0), MathEx.lfactorial(2), 1E-7);
+        assertEquals(Math.log(6.0), MathEx.lfactorial(3), 1E-7);
+        assertEquals(Math.log(24.0), MathEx.lfactorial(4), 1E-7);
     }
\ No newline at end of file
