diff --git a/old_test.java b/new_test.java
index a61cd78..a56a97b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void testSearch() throws BaseXException, QueryException {
     // test arguments
-    final String fun = check(FunDef.SEARCH);
+    final String fun = check(Function.SEARCH);
 
     // check index results
     query(fun + "(., 'assignments')", "Assignments");
