diff --git a/old_product.java b/new_product.java
index 8955cc3..fcda00c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-static Iter search(final Data data, final byte[] str, final Fun fun,
-      final QueryContext ctx) throws QueryException {
+static Iter search(final Data data, final byte[] str,
+      final FuncCall fun, final QueryContext ctx) throws QueryException {
 
     final IndexContext ic = new IndexContext(ctx, data, null, true);
     if(!data.meta.ftindex) NOIDX.thrw(fun.input, fun);
