diff --git a/old_test.java b/new_test.java
index 549e9f9..fc3bbcd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,6 +20,8 @@
 		
 		program.getStates().add(state1);
 		program.getStates().add(state2);
+		state1.setPatientProgram(program);
+		state2.setPatientProgram(program);
 		
 		//when
 		program.voidLastState(workflow, new User(), new Date(), "");
@@ -27,4 +29,5 @@
 		//then
 		Assert.assertTrue(state1.isVoided());
 		Assert.assertFalse(state2.isVoided());
+		Assert.assertTrue(state2.getEndDate() == null);
 	}
\ No newline at end of file
