diff --git a/old_product.java b/new_product.java
index 4ed2c8b..078457f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -18,7 +18,9 @@ public void voidLastState(ProgramWorkflow workflow, User voidBy, Date voidDate,
 			last.setVoidReason(voidReason);
 		}
 		if (nextToLast != null && nextToLast.getEndDate() != null) {
-			nextToLast.setEndDate(null);
+			nextToLast.setEndDate(nextToLast.getPatientProgram() != null
+			        && nextToLast.getPatientProgram().getDateCompleted() != null ? nextToLast.getPatientProgram()
+			        .getDateCompleted() : null);
 			nextToLast.setDateChanged(voidDate);
 			nextToLast.setChangedBy(voidBy);
 		}
