diff --git a/old_product.java b/new_product.java
index 5cd83af..68210f1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public Map<String, String> getNestedProperties(String prefix) {
     Map<String, String> subProps = new HashMap<String, String>();
     for(Map.Entry<String, String> entry : options.entrySet()) {
       if(entry.getKey().startsWith(prefix)) {
-        subProps.put(entry.getKey().substring(prefix.length()), entry.getKey());
+        subProps.put(entry.getKey().substring(prefix.length()), entry.getValue());
       }
     }
 
