diff --git a/old_test.java b/new_test.java
index 232bef3..dca746f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
   public void getInfoTest() throws Exception {
     FileInfo info = new FileInfo();
-    Mockito.when(mFileSystemMasterClient.getFileInfo(FILE_ID)).thenReturn(info);
+    Mockito.when(mFileSystemMasterClient.getStatus(FILE_ID)).thenReturn(info);
     GetInfoOptions getInfoOptions = GetInfoOptions.defaults();
     Assert.assertSame(info, mFileSystem.getInfo(new TachyonFile(FILE_ID), getInfoOptions));
-    Mockito.verify(mFileSystemMasterClient).getFileInfo(FILE_ID);
+    Mockito.verify(mFileSystemMasterClient).getStatus(FILE_ID);
   }
\ No newline at end of file
