diff --git a/old_product.java b/new_product.java
index e83aa33..b845764 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,13 +3,11 @@ public String compressStringifiedRules() {
         final ByteArrayOutputStream out = new ByteArrayOutputStream();
         String encodedResult = null;
         try {
-            //Note : not using GZipOutputStream since that is for files
-            //GZipOutputStream gives a different header, although the compression is the same
             final DeflaterOutputStream dzip = new DeflaterOutputStream(out);
             dzip.write(stringified.getBytes());
             dzip.close();
             encodedResult = Base64.encodeBase64String(out.toByteArray());
-        } catch (IOException e) {
+        } catch (final IOException e) {
             LOGGER.warn("Exception while compressing security group rules");
         }
         return encodedResult;
