diff --git a/old_test.java b/new_test.java
index 0115d7b..d901498 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testPut() throws Exception {
-        List<ParameterizedHeader> fooValues1 = new ArrayList<ParameterizedHeader>();
+        List<ParameterizedHeader> fooValues1 = new ArrayList<>();
         fooValues1.add(new ParameterizedHeader("foo1"));
         fooValues1.add(new ParameterizedHeader("foo2"));
 
@@ -12,7 +12,7 @@
         assertTrue(map.containsValue(fooValues1));
         assertTrue(map.get("foo") == fooValues1);
 
-        List<ParameterizedHeader> fooValues2 = new ArrayList<ParameterizedHeader>();
+        List<ParameterizedHeader> fooValues2 = new ArrayList<>();
         fooValues2.add(new ParameterizedHeader("foo3"));
         fooValues2.add(new ParameterizedHeader("foo4"));
 
