diff --git a/old_test.java b/new_test.java
index e01d217..e85d7a8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,6 +17,8 @@
         Mockito.verify(jobEntity).setExecution(argument.capture());
         Mockito.verify(jobEntity).setStatus(argument1.capture());
 
+        Mockito.verify(jobEntity, Mockito.times(1)).setStarted(Mockito.any());
+
         Assert.assertEquals(hostname, argument.getValue().getHostName());
         Assert.assertEquals(pid, argument.getValue().getProcessId());
         Assert.assertThat(argument.getValue().getCheckDelay(), Matchers.is(checkDelay));
