diff --git a/old_product.java b/new_product.java
index 392042e..0902f9b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,6 +11,7 @@
             throw new GeniePreconditionException("Cannot create a job execution entry with id blank or null");
         }
 
+        this.updateJobStatus(jobExecution.getId(), JobStatus.RUNNING, "Job is Running.");
         final JobEntity jobEntity = jobRepo.findOne(jobExecution.getId());
         if (jobEntity == null) {
             throw new GenieNotFoundException("Cannot find the job for the id of the jobExecution specified.");
@@ -25,9 +26,4 @@
         jobExecutionEntity.setTimeout(jobExecution.getTimeout());
 
         jobEntity.setExecution(jobExecutionEntity);
-        jobEntity.setStatus(JobStatus.RUNNING);
-        jobEntity.setStatusMsg("Job is Running");
-
-        // TODO: This is quite a bit (in terms of code) after it actually started...
-        jobEntity.setStarted(new Date());
     }
\ No newline at end of file
