diff --git a/old_test.java b/new_test.java
index 3b02af6..05aa86c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
-	public void testGetTitle() {
-		Map<String, Object> params = new HashMap<String, Object>();
-		params.put(AlchemyLanguage.URL, "http://www.techcrunch.com/");
-		DocumentTitle title = service.getTitle(params);
-		Assert.assertNotNull(title);
-	}
\ No newline at end of file
+  public void testGetTitle() {
+    final Map<String, Object> params = new HashMap<String, Object>();
+    params.put(AlchemyLanguage.URL, "http://www.techcrunch.com/");
+    final DocumentTitle title = service.getTitle(params);
+    Assert.assertNotNull(title);
+  }
\ No newline at end of file
