diff --git a/old_product.java b/new_product.java
index 1ef0720..cea0fcd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
 public DocumentTitle getTitle(Map<String, Object> params) {
-		return executeRequest(params, AlchemyAPI.title, DocumentTitle.class, "html", "url");
-	}
\ No newline at end of file
+    return executeRequest(params, AlchemyAPI.title, DocumentTitle.class, HTML, URL);
+  }
\ No newline at end of file
