diff --git a/old_test.java b/new_test.java
index 2389f33..93476b1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testIsNull() {
-        assertTrue(instance.isNull("name").getQueryCriterions()
-                .contains(new IsNullCriterion("name")));
+    	assertEquals(new IsNullCriterion("name"), instance.isNull("name").getQueryCriterion());
     }
\ No newline at end of file
