diff --git a/old_product.java b/new_product.java
index bb0c82f..6d07c5c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ void query(final String path) throws IOException {
     // set base path and serialization parameters
     final HTTPContext http = session.http;
     context.options.set(MainOptions.QUERYPATH, path);
-    context.options.set(MainOptions.SERIALIZER, serial(http));
+    context.options.set(MainOptions.SERIALIZER, http.sopts());
     http.initResponse();
 
     final int cs = cmds.size();
