diff --git a/old_product.java b/new_product.java
index 5094344..75d1158 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,7 +15,6 @@ CompletableFuture<Void> initialize(Duration timeout, Executor executor) {
 
                         // It is very important to keep this value up-to-date and correct.
                         this.metadata.setStorageLength(segmentInfo.getLength());
-                        this.dataSource.notifyStorageLengthUpdated(this.metadata.getId());
                     }
 
                     // Check if the Storage segment is sealed, but it's not in metadata (this is 100% indicative of some data corruption happening).
