diff --git a/old_test.java b/new_test.java
index 084c50c..5ce031b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,5 +5,5 @@
         t1.set(0, "2010-04-15");
 
         // Time zone is preserved.
-        assertEquals(ISOHelper.parseDateTime(t1), new DateTime(2010, 4, 15, 0, 0, 0, 0, DateTimeZone.UTC));        
+        assertEquals(new DateTime(2010, 4, 15, 0, 0, 0, 0, DateTimeZone.getDefault()), ISOHelper.parseDateTime(t1));  
     }
\ No newline at end of file
