diff --git a/old_product.java b/new_product.java
index 5df86a9..1250fa2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,7 @@
 public static DateTime parseDateTime(Tuple input) throws ExecException {	
-	        
-	    // Save previous default time zone for restore later.
-	    DateTimeZone previousDefaultTimeZone = DateTimeZone.getDefault();
-
-	    // Temporarily set default time zone to UTC, for this parse.
-	    DateTimeZone.setDefault(DEFAULT_DATE_TIME_ZONE);
 
 	    String isoDateString = input.get(0).toString();
 	    DateTime dt = ISODateTimeFormat.dateTimeParser().withOffsetParsed().parseDateTime(isoDateString);			
 
-	    // restore previous default TimeZone.
-	    DateTimeZone.setDefault(previousDefaultTimeZone);
-
 	    return dt;
 	}
\ No newline at end of file
