diff --git a/old_product.java b/new_product.java
index 3aaa8a5..b7ce1cd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,11 @@
 @PreAuthorize("hasAnyRole('A')")
 	@RequestMapping("/delete")
-	public String delete(ModelMap model, @RequestParam String userIds) {
+	public String delete(User user, @RequestParam String userIds, ModelMap model) {
 		String[] ids = userIds.split(",");
 		for (String eachId : Arrays.asList(ids)) {
-			userService.delete(eachId);
+			if (!user.getUserId().equals(eachId)) {
+				userService.delete(eachId);
+			}
 		}
 		model.clear();
 		return "redirect:/user/";
