diff --git a/old_test.java b/new_test.java
index 05cc273..66b4d33 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,7 +16,7 @@
 
 		// test to delete one
 		model.clear();
-		userController.delete(model, "NewUserId1");
+		userController.delete(testUser, "NewUserId1", model);
 		model.clear();
 		userController.getAll(model, Role.USER, page, "NewUserName");
 		userList = (PageImpl<User>) model.get("users");
@@ -24,7 +24,7 @@
 
 		// test to delete more
 		model.clear();
-		userController.delete(model, "NewUserId2,NewUserId3");
+		userController.delete(testUser, "NewUserId2,NewUserId3", model);
 		model.clear();
 		userController.getAll(model, Role.USER, page, "NewUserName");
 		userList = (PageImpl<User>) model.get("users");
