diff --git a/old_test.java b/new_test.java
index 086ccef..f773624 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,8 +3,7 @@
 	public void getName_shouldReturnUnlocalizedNameWhenNoLocalizationIsAdded() throws Exception {
 		FieldType type = new FieldType();
 		String expected = "Test";
-		OpenmrsSerializer serializer = new LocalizedStringSerializer();
-		LocalizedString ls = serializer.deserialize(expected, LocalizedString.class);
+		LocalizedString ls = LocalizedString.valueOf(expected);
 		type.setLocalizedName(ls);
 		Assert.assertEquals(expected, type.getName());
 	}
\ No newline at end of file
