diff --git a/old_test.java b/new_test.java
index 8a55226..d0ca8b2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,14 +2,14 @@
   public void addKeyTest() {
     String key = "key";
     org.apache.hadoop.conf.Configuration conf = new org.apache.hadoop.conf.Configuration();
-    Configuration configuration = new Configuration();
-    configuration.set(key, "alluxioKey");
+    Configuration.defaultInit();
+    Configuration.set(key, "alluxioKey");
 
     System.setProperty(key, "systemKey");
-    HdfsUnderFileSystemUtils.addKey(conf, configuration, key);
+    HdfsUnderFileSystemUtils.addKey(conf, key);
     Assert.assertEquals("systemKey", conf.get(key));
 
     System.clearProperty(key);
-    HdfsUnderFileSystemUtils.addKey(conf, configuration, key);
+    HdfsUnderFileSystemUtils.addKey(conf, key);
     Assert.assertEquals("alluxioKey", conf.get(key));
   }
\ No newline at end of file
