diff --git a/old_product.java b/new_product.java
index af23e02..41b8d15 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
-public MonthOfYear getFirstMonthOfQuarter() {
+public Month getFirstMonthOfQuarter() {
         switch (this) {
-            case Q1: return MonthOfYear.JANUARY;
-            case Q2: return MonthOfYear.APRIL;
-            case Q3: return MonthOfYear.JULY;
-            case Q4: return MonthOfYear.OCTOBER;
+            case Q1: return Month.JANUARY;
+            case Q2: return Month.APRIL;
+            case Q3: return Month.JULY;
+            case Q4: return Month.OCTOBER;
             default: throw new IllegalStateException("Unreachable");
         }
     }
\ No newline at end of file
