diff --git a/old_product.java b/new_product.java
index a3c8c9a..f745d49 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 private static void calculateNetworkBufNew(final Configuration config) {
 		// (2) fixed size memory
-		config.setString(TaskManagerOptions.NETWORK_BUFFERS_MEMORY_MIN, String.valueOf(1L << 20)); // 1MB
-		config.setString(TaskManagerOptions.NETWORK_BUFFERS_MEMORY_MAX, String.valueOf(1L << 20)); // 1MB
+		config.setString(NetworkEnvironmentOptions.NETWORK_BUFFERS_MEMORY_MIN, String.valueOf(1L << 20)); // 1MB
+		config.setString(NetworkEnvironmentOptions.NETWORK_BUFFERS_MEMORY_MAX, String.valueOf(1L << 20)); // 1MB
 
 
 		// note: actual network buffer memory size is independent of the totalJavaMemorySize
@@ -13,14 +13,14 @@ private static void calculateNetworkBufNew(final Configuration config) {
 		Random ran = new Random();
 		for (int i = 0; i < 1_000; ++i){
 			float frac = Math.max(ran.nextFloat(), Float.MIN_VALUE);
-			config.setFloat(TaskManagerOptions.NETWORK_BUFFERS_MEMORY_FRACTION, frac);
+			config.setFloat(NetworkEnvironmentOptions.NETWORK_BUFFERS_MEMORY_FRACTION, frac);
 
 			long min = Math.max(MemorySize.parse(TaskManagerOptions.MEMORY_SEGMENT_SIZE.defaultValue()).getBytes(), ran.nextLong());
-			config.setString(TaskManagerOptions.NETWORK_BUFFERS_MEMORY_MIN, String.valueOf(min));
+			config.setString(NetworkEnvironmentOptions.NETWORK_BUFFERS_MEMORY_MIN, String.valueOf(min));
 
 
 			long max = Math.max(min, ran.nextLong());
-			config.setString(TaskManagerOptions.NETWORK_BUFFERS_MEMORY_MAX, String.valueOf(max));
+			config.setString(NetworkEnvironmentOptions.NETWORK_BUFFERS_MEMORY_MAX, String.valueOf(max));
 
 			long javaMem = Math.max(max + 1, ran.nextLong());
 
