diff --git a/old_product.java b/new_product.java
index 0950ec1..ef86712 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,5 @@
-@Deprecated
-    public static Date[] getExtentYesterday(){
+public static Date[] getResetYesterdayAndToday(){
         Calendar calendar = CalendarUtil.resetCalendarByDay(new Date());
         Date today = calendar.getTime();
-        calendar.add(Calendar.DATE, -1);
-        Date yesterday = calendar.getTime();
-        return new Date[] { yesterday, today };
+        return new Date[] { DateUtil.getYesterday(today), today };
     }
\ No newline at end of file
