diff --git a/old_product.java b/new_product.java
index 9311561..5b410c3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public long toNanos() {
-        long millis = Math.multiplyExact(seconds, NANOS_PER_SECOND);
-        millis = Math.addExact(millis, nanos);
-        return millis;
+        long totalNanos = Math.multiplyExact(seconds, NANOS_PER_SECOND);
+        totalNanos = Math.addExact(totalNanos, nanos);
+        return totalNanos;
     }
\ No newline at end of file
