diff --git a/old_test.java b/new_test.java
index 9176a28..ece49e7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void reportLostFileTest() throws Exception {
     AlluxioURI path = new AlluxioURI("test");
-    mTachyonLineageFileSystem.reportLostFile(path);
+    mAlluxioLineageFileSystem.reportLostFile(path);
     Mockito.verify(mLineageMasterClient).reportLostFile("test");
     // verify client is released
     Mockito.verify(mLineageContext).releaseMasterClient(mLineageMasterClient);
