diff --git a/old_product.java b/new_product.java
index 0b3bc08..31b6b10 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,5 @@ public void observe(Observer<T> observer, Annotation... bindingTypes)
    {
       Set<Annotation> bindingParameters = checkBindingTypes(bindingTypes);
       bindingParameters.addAll(this.bindingTypes);
-      ManagerImpl.instance().addObserver(observer, eventType, bindingParameters.toArray(new Annotation[0]));
+      manager.addObserver(observer, eventType, bindingParameters.toArray(new Annotation[0]));
    }
\ No newline at end of file
