diff --git a/old_product.java b/new_product.java
index 02ca024..8df1930 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,11 +16,7 @@
         if (LOG.isDebugEnabled()) {
             LOG.debug("Setting the application info for job with id " + id);
         }
-        final Job job = this.jobRepo.findOne(id);
-        if (job != null) {
-            job.setExecutionClusterId(clusterId);
-            job.setExecutionClusterName(clusterName);
-        } else {
-            throw new GenieNotFoundException("No job with id " + id + " exists");
-        }
+        final JobEntity jobEntity = this.findJob(id);
+        jobEntity.setExecutionClusterId(clusterId);
+        jobEntity.setExecutionClusterName(clusterName);
     }
\ No newline at end of file
