diff --git a/old_test.java b/new_test.java
index e7f2e0e..65bad47 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
         context.setConnection(new OutboundConnection());
         this.cipher = new RecordAEADCipher(context, KeySetGenerator.generateKeySet(context));
         byte[] ciphertext = ArrayConverter.hexStringToByteArray("1BB3293A919E0D66F145AE830488E8D89BE5EC16688229");
-        byte[] plaintext = cipher.decrypt(ciphertext);
+        byte[] plaintext = cipher.decrypt(ciphertext).getDecryptedCipherText();
         byte[] plaintext_correct = ArrayConverter.hexStringToByteArray("08000002000016");
         assertArrayEquals(plaintext, plaintext_correct);
     }
\ No newline at end of file
