diff --git a/old_product.java b/new_product.java
index 57e4285..e2caf3a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,8 @@
 @Override
-    public byte[] decrypt(byte[] data) {
+    public DecryptionResult decrypt(byte[] data) {
         try {
             byte[] plaintext;
+            byte[] usedIv;
             ConnectionEndType localConEndType = context.getConnection().getLocalConnectionEndType();
             if (useExplicitIv) {
                 decryptIv = new IvParameterSpec(Arrays.copyOf(data, decryptCipher.getBlockSize()));
@@ -10,6 +11,7 @@
                 decryptCipher.init(Cipher.DECRYPT_MODE, new SecretKeySpec(getKeySet().getReadKey(localConEndType),
                         bulkCipherAlg.getJavaName()), decryptIv);
                 plaintext = decryptCipher.doFinal(Arrays.copyOfRange(data, decryptCipher.getBlockSize(), data.length));
+                usedIv = decryptIv.getIV();
             } else {
                 decryptIv = new IvParameterSpec(getDecryptionIV());
                 LOGGER.debug("decryptionIV: " + ArrayConverter.bytesToHexString(decryptIv.getIV()));
@@ -17,15 +19,15 @@
                 decryptCipher.init(Cipher.DECRYPT_MODE, new SecretKeySpec(getKeySet().getReadKey(localConEndType),
                         bulkCipherAlg.getJavaName()), decryptIv);
                 plaintext = decryptCipher.doFinal(data);
-
+                usedIv = decryptIv.getIV();
                 // Set next IV
                 setNextDecryptIv(data);
             }
 
-            return plaintext;
+            return new DecryptionResult(usedIv, plaintext, useExplicitIv);
         } catch (BadPaddingException | IllegalBlockSizeException | InvalidAlgorithmParameterException
                 | InvalidKeyException | UnsupportedOperationException ex) {
             LOGGER.warn("Could not decrypt Data with the provided parameters. Returning undecrypted data.", ex);
-            return data;
+            return new DecryptionResult(null, data, useExplicitIv);
         }
     }
\ No newline at end of file
