diff --git a/old_test.java b/new_test.java
index 48be32a..4273053 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
         final String dateStr = "02 942, 1996";
         final String[] parsers = new String[] {"MM DDD, yyyy"};
         
-        Date date = DateUtils.parseDate(dateStr, parsers);
+        final Date date = DateUtils.parseDate(dateStr, parsers);
         assertEquals(cal.getTime(), date);
         
         try {
