diff --git a/old_product.java b/new_product.java
index c1d2d14..4a7ccd3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -43,11 +43,17 @@ public List<Block> execute(WikiMacroParameters parameters, String macroContent,
             xwikiContext.put(MACRO_KEY, macroBinding);
 
             MacroBlock wikiMacroBlock = context.getCurrentMacroBlock();
+
             MacroMarkerBlock wikiMacroMarker =
                 new MacroMarkerBlock(wikiMacroBlock.getId(), wikiMacroBlock.getParameters(),
                     wikiMacroBlock.getContent(), xdom.getChildren(), wikiMacroBlock.isInline());
+
+            // make sure to use provided metadatas
+            MetaDataBlock metaDataBlock =
+                new MetaDataBlock(Collections.<Block> singletonList(wikiMacroMarker), xdom.getMetaData());
+
             // otherwise the inner macros will not be able to access the parent DOM
-            wikiMacroMarker.setParent(wikiMacroBlock.getParent());
+            metaDataBlock.setParent(wikiMacroBlock.getParent());
 
             if (observation != null) {
                 observation.notify(STARTEXECUTION_EVENT, this, macroBinding);
