diff --git a/old_test.java b/new_test.java
index 46b8224..a70837d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testExecute() throws Exception
     {
-        registerWikiMacro("wikimacro1", "This is **bold**");
+        registerWikiMacro("wikimacro1", "This is **bold**", Syntax.XWIKI_2_0);
 
         Converter converter = getComponentManager().lookup(Converter.class);
 
