diff --git a/old_test.java b/new_test.java
index 73d759a..f138466 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,9 @@
 @Test
     public void testCreate_OptionalGetter() {
-        Map<String, Object> values = Maps.newHashMap();
-        values.put("name", Optional.ofNullable("MyName"));
-
-        ClassGeneratorProvider provider = createProvider();
-        EventFactory<OptionalGetter> factory = provider.create(OptionalGetter.class, Object.class, SpongeEventFactoryUtils.plugins);
-        OptionalGetter getter = factory.apply(values);
+        OptionalGetter getter = this.getFactory().createOptionalGetter(Optional.ofNullable("MyName"));
 
         assertThat(getter.getName().isPresent(), is(true));
-        assertThat(getter.getName().get(), is(Matchers.equalTo("MyName")));
+        assertThat(getter.getName().get(), is(equalTo("MyName")));
 
         getter.setName(null);
         assertThat(getter.getName().isPresent(), is(false));
@@ -16,5 +11,5 @@
         getter.setName("Aaron");
 
         assertThat(getter.getName().isPresent(), is(true));
-        assertThat(getter.getName().get(), is(Matchers.equalTo("Aaron")));
+        assertThat(getter.getName().get(), is(equalTo("Aaron")));
     }
\ No newline at end of file
