diff --git a/old_product.java b/new_product.java
index 9d9fec0..92343a8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-public void apply(ServiceResponse response, IPolicyContext context, Object config, IPolicyChain<ServiceResponse> chain);
\ No newline at end of file
+public void apply(ApiResponse response, IPolicyContext context, Object config, IPolicyChain<ApiResponse> chain);
\ No newline at end of file
