diff --git a/old_test.java b/new_test.java
index d5538d5..7183cc1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,7 @@
                 "  \"userHeader\" : \"X-Identity\"\r\n" +
                 "}";
         Object config = policy.parseConfiguration(json);
-        ServiceRequest request = new ServiceRequest();
+        ApiRequest request = new ApiRequest();
         request.setContract(createTestContract());
         request.setType("GET");
         request.setApiKey("12345");
@@ -24,7 +24,7 @@
             }
         });
         Mockito.when(context.getComponent(IRateLimiterComponent.class)).thenReturn(new InMemoryRateLimiterComponent());
-        IPolicyChain<ServiceRequest> chain = null;
+        IPolicyChain<ApiRequest> chain = null;
 
         for (int count = 0; count < 10; count++) {
             chain = Mockito.mock(IPolicyChain.class);
