diff --git a/old_test.java b/new_test.java
index 32bea97..01e371c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -48,7 +48,6 @@
 		        + p.getIdentifiers().size(), p.getIdentifiers().size() == 2);
 		
 		pa3.setIdentifier(pa3.getIdentifier() + "some new string to make sure it gets added");
-		pa3.setVoided(true);
 		p.addIdentifier(pa3);
 		// make sure the identifier IS added
 		assertTrue("There should be 3 identifiers in the patient object but there is actually : "
