diff --git a/old_product.java b/new_product.java
index fcb0261..3881c76 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,15 @@
 public void addIdentifier(PatientIdentifier patientIdentifier) {
 		patientIdentifier.setPatient(this);
 		if (getIdentifiers() == null)
-			identifiers = new TreeSet<PatientIdentifier>();
-		if (patientIdentifier != null && !OpenmrsUtil.collectionContains(identifiers, patientIdentifier))
-			identifiers.add(patientIdentifier);
+			identifiers = new LinkedHashSet<PatientIdentifier>();
+		if (patientIdentifier != null) {
+			// make sure the set doesn't already contain an identifier with the same
+			// identifier, identifierType
+			for (PatientIdentifier currentId : getIdentifiers()) {
+				if (currentId.equalsContent(patientIdentifier)) {
+					return; // fail silently if someone tries to add a duplicate
+				}
+			}
+		}
+		identifiers.add(patientIdentifier);
 	}
\ No newline at end of file
