diff --git a/old_product.java b/new_product.java
index 6274494..311a658 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,13 +6,15 @@ public boolean process( I frame ) {
 		// set up data structures and spawn tracks
 		if( totalProcessed == 1 ) {
 			tracker.spawnTracks();
+			tracker.setKeyFrame();
+
 			worldToKey.set(worldToInit);
 			worldToCurr.set(worldToInit);
 			return true;
 		}
 
 		// fit the motion model to the feature tracks
-		List<AssociatedPair> pairs = tracker.getActiveTracks();
+		List<AssociatedPair> pairs = tracker.getPairs();
 		if( !modelMatcher.process(pairs,null) ) {
 			return false;
 		}
