diff --git a/old_test.java b/new_test.java
index 649970d..e467aa3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,20 +1,23 @@
 @Test
   public void DeleteTest() throws Exception {
     for (int i = 0; i < 10; i ++) {
-      mTfs.mkdir(new TachyonURI("/i" + i));
+      String dirPath = "/i" + i;
+      mTfs.mkdirs(new TachyonURI(dirPath));
       for (int j = 0; j < 10; j ++) {
-        mTfs.createFile(new TachyonURI("/i" + i + "/j" + j), (i + j + 1) * 64);
+        ClientOptions option =
+            new ClientOptions.Builder(mMasterTachyonConf).setBlockSize((i + j + 1) * 64).build();
+        String filePath = dirPath + "/j" + j;
+        mTfs.getOutStream(new TachyonURI(filePath), option).close();
         if (j >= 5) {
-          mTfs.delete(new TachyonURI("/i" + i + "/j" + j), false);
+          mTfs.delete(mTfs.open(new TachyonURI(filePath)));
         }
       }
       if (i >= 5) {
-        mTfs.delete(new TachyonURI("/i" + i), true);
+        mTfs.delete(mTfs.open(new TachyonURI(dirPath)));
       }
     }
     mLocalTachyonCluster.stopTFS();
     DeleteTestUtil();
-    String editLogPath = mLocalTachyonCluster.getEditLogPath();
-    UnderFileSystem.get(editLogPath, mMasterTachyonConf).delete(editLogPath, true);
+    deleteFsMasterJournalLogs();
     DeleteTestUtil();
   }
\ No newline at end of file
