diff --git a/old_test.java b/new_test.java
index 0d17451..0ce613a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testGetCoordinates() throws MalformedURLException, IOException {
-        String expResult = "org.scala-lang:scala-library:2.11.2";
+        String expResult = "org/scala-lang/scala-library/2.11.2";
         String result = instance.getCoordinates();
         assertEquals(expResult, result);
 
@@ -8,11 +8,11 @@
         InputStream stream = BaseTest.getResourceAsStream(this, "bintray/spring-core-3.0.0.RELEASE.json");
         BintrayArtifact[] ba = searcher.parseResponse(stream);
 
-        expResult = "org.springframework:spring-core:3.0.0.RELEASE";
+        expResult = "org/springframework/spring-core/3.0.0.RELEASE";
         result = ba[0].getCoordinates();
         assertEquals(expResult, result);
 
-        expResult = "org.sonatype.aether:aether-util:1.7";
+        expResult = "org/sonatype/aether/aether-util/1.7";
         stream = BaseTest.getResourceAsStream(this, "bintray/aether-util-1.7.json");
         ba = searcher.parseResponse(stream);
         result = ba[0].getCoordinates();
