diff --git a/old_product.java b/new_product.java
index b37fe7d..b510053 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,21 +1,27 @@
-public void setApplication(final Application application) {
+public void setApplications(final Set<Application> applications) {
         //Clear references to this command in existing applications
-        if (this.application != null
-                && this.application.getCommands() != null) {
-            this.application.getCommands().remove(this);
+        if (this.applications != null) {
+            this.applications.stream()
+                    .filter(application -> application.getCommands() != null)
+                    .forEach(application -> application.getCommands().remove(this));
         }
         //set the application for this command
-        this.application = application;
+        if (this.applications == null) {
+            this.applications = new HashSet<>();
+        }
+        this.applications.clear();
+        if (applications != null) {
+            this.applications.addAll(applications);
+        }
 
         //Add the reverse reference in the new applications
-        if (this.application != null) {
-            Set<Command> commands = this.application.getCommands();
-            if (commands == null) {
-                commands = new HashSet<>();
-                this.application.setCommands(commands);
-            }
-            if (!commands.contains(this)) {
-                commands.add(this);
-            }
-        }
+        this.applications.stream()
+                .forEach(
+                        application -> {
+                            if (application.getCommands() == null) {
+                                application.setCommands(new HashSet<>());
+                            }
+                            application.getCommands().add(this);
+                        }
+                );
     }
\ No newline at end of file
