diff --git a/old_product.java b/new_product.java
index 8c186a2..fc0669d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,11 @@
 private Item restore(final QueryContext qc) throws QueryException {
     // extract database name from backup file
-    final String name = string(checkStr(exprs[0], qc));
-    if(!Databases.validName(name)) throw BXDB_NAME.get(info, name);
+    final String name = string(toToken(exprs[0], qc));
+    if(!Databases.validName(name)) throw BXDB_NAME_X.get(info, name);
 
     // find backup with or without date suffix
     final StringList backups = qc.context.databases.backups(name);
-    if(backups.isEmpty()) throw BXDB_NOBACKUP.get(info, name);
+    if(backups.isEmpty()) throw BXDB_NOBACKUP_X.get(info, name);
 
     final String backup = backups.get(0);
     final String db = Databases.name(backup);
