diff --git a/old_product.java b/new_product.java
index 2803688..bfe2137 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public ServiceCall<Void> deleteClassifier(String classifierId) {
-    Validator.isTrue((classifierId != null) && !classifierId.isEmpty(), "classifierId cannot be null or empty");
-
-    final Request request = RequestBuilder.delete(String.format(PATH_CLASSIFIER, classifierId)).build();
-    return createServiceCall(request, ResponseConverterUtils.getVoid());
+public ServiceCall<Void> deleteClassifier(DeleteClassifierOptions deleteClassifierOptions) {
+    Validator.notNull(deleteClassifierOptions, "deleteClassifierOptions cannot be null");
+    RequestBuilder builder = RequestBuilder.delete(String.format("/v1/classifiers/%s", deleteClassifierOptions
+        .classifierId()));
+    return createServiceCall(builder.build(), ResponseConverterUtils.getVoid());
   }
\ No newline at end of file
