diff --git a/old_product.java b/new_product.java
index 8d32f61..e5eed20 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-public static String getValue(ResourceBundle resourceBundle,String key){
+public static String getValue(ResourceBundle resourceBundle,String key,Object...arguments){
+        Validate.notNull(resourceBundle, "resourceBundle can't be null!");
         Validate.notBlank(key, "key can't be null/empty!");
 
         if (!resourceBundle.containsKey(key)){
@@ -10,5 +11,5 @@ public static String getValue(ResourceBundle resourceBundle,String key){
         if (isNullOrEmpty(value)){
             LOGGER.trace("resourceBundle has key:[{}],but value is null/empty", key);
         }
-        return value;
+        return isNullOrEmpty(value) ? StringUtils.EMPTY : MessageFormatUtil.format(value, arguments);// 支持 arguments 为null,原样返回
     }
\ No newline at end of file
