diff --git a/old_test.java b/new_test.java
index 1dc574a..8e5fa90 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void testConnect() throws Exception {
     Class.forName("org.apache.drill.jdbc.Driver");
-    final Connection connection = DriverManager.getConnection("jdbc:drillref:schema=DONUTS");
+    final Connection connection = DriverManager.getConnection("jdbc:drill:zk=local");
     connection.close();
   }
\ No newline at end of file
