diff --git a/old_product.java b/new_product.java
index ef5e051..be268e9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public VM kFoldCrossValidation(Dataset dataset, int k, String dbName, DatabaseConfiguration dbConf, Class<? extends BaseMLmodel> aClass, TP trainingParameters) {
-        int n = dataset.getRecordNumber();
+        int n = dataset.size();
         if(k<=0 || n<=k) {
             throw new IllegalArgumentException("Invalid number of folds");
         }
@@ -59,7 +59,7 @@ public VM kFoldCrossValidation(Dataset dataset, int k, String dbName, DatabaseCo
             
             Dataset trainingData = dataset.generateNewSubset(foldTrainingIds);
             mlmodel.fit(trainingData, trainingParameters); 
-            trainingData.erase();
+            trainingData.delete();
             trainingData = null;
                         
             
@@ -67,11 +67,11 @@ public VM kFoldCrossValidation(Dataset dataset, int k, String dbName, DatabaseCo
             
             //fetch validation metrics
             VM entrySample = mlmodel.validate(validationData);
-            validationData.erase();
+            validationData.delete();
             validationData = null;
             
             //delete algorithm
-            mlmodel.erase();
+            mlmodel.delete();
             mlmodel = null;
             
             //add the validationMetrics in the list
