diff --git a/old_test.java b/new_test.java
index 09a9f81..edff7c8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,8 +8,10 @@
 		}
 
 		ofy().transact(() -> {
-			for (int i=1; i<10; i++)
-				ofy().transactionless().load().type(Thing.class).id(i).now();
+			for (int i=1; i<10; i++) {
+				final int index = i;
+				ofy().transactionless(() -> ofy().load().type(Thing.class).id(index).now());
+			}
 
 			ofy().save().entity(new Thing(99));
 		});
