diff --git a/old_test.java b/new_test.java
index 1a3ab80..8416aca 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,5 @@
 @Test(expectedExceptions = IllegalArgumentException.class)
-  public void test_putAll_arraysMismatch() {
+  public void test_putAll_collectionsMismatch() {
     LocalDateDoubleTimeSeriesBuilder test = LocalDateDoubleTimeSeries.builder();
-    LocalDate[] dates = {date(2014, 1, 1)};
-    double[] values = {2d, 3d};
-    test.putAll(dates, values);
+    test.putAll(Arrays.asList(date(2014, 1, 1)), Doubles.asList(2d, 3d));
   }
\ No newline at end of file
