diff --git a/old_product.java b/new_product.java
index d650bd3..45cdac7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,10 @@
-public LocalDateDoubleTimeSeriesBuilder putAll(LocalDate[] dates, double[] values) {
-    ArgChecker.isTrue(dates.length == values.length,
-        "Arrays are of different sizes - dates: {}, values: {}", dates.length, values.length);
-    for (int i = 0; i < dates.length; i++) {
-      entries.put(dates[i], values[i]);
+public LocalDateDoubleTimeSeriesBuilder putAll(Collection<LocalDate> dates, Collection<Double> values) {
+    ArgChecker.isTrue(dates.size() == values.size(),
+        "Arrays are of different sizes - dates: {}, values: {}", dates.size(), values.size());
+    Iterator<LocalDate> itDate = dates.iterator();
+    Iterator<Double> itValue = values.iterator();
+    for (int i = 0; i < dates.size(); i++) {
+      entries.put(itDate.next(), itValue.next());
     }
     return this;
   }
\ No newline at end of file
