diff --git a/old_product.java b/new_product.java
index 841ec24..0ed3907 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public synchronized boolean rename(int fileId, TachyonURI dstPath) throws IOException {
+public synchronized boolean rename(long fileId, TachyonURI dstPath) throws IOException {
     return rename(fileId, TachyonURI.EMPTY_URI, dstPath);
   }
\ No newline at end of file
