diff --git a/old_test.java b/new_test.java
index e935829..567ea77 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,16 +1,18 @@
 @Test
   public void RenameTest() throws Exception {
     for (int i = 0; i < 10; i ++) {
-      mTfs.mkdir(new TachyonURI("/i" + i));
+      mTfs.mkdirs(new TachyonURI("/i" + i));
       for (int j = 0; j < 10; j ++) {
-        mTfs.createFile(new TachyonURI("/i" + i + "/j" + j), (i + j + 1) * 64);
-        mTfs.rename(new TachyonURI("/i" + i + "/j" + j), new TachyonURI("/i" + i + "/jj" + j));
+        ClientOptions option = new ClientOptions.Builder(mMasterTachyonConf).setBlockSize(
+            (i + j + 1) * 64).build();
+        TachyonURI path = new TachyonURI("/i" + i + "/j" + j);
+        mTfs.getOutStream(path, option).close();
+        mTfs.rename(mTfs.open(path), new TachyonURI("/i" + i + "/jj" + j));
       }
-      mTfs.rename(new TachyonURI("/i" + i), new TachyonURI("/ii" + i));
+      mTfs.rename(mTfs.open(new TachyonURI("/i" + i)), new TachyonURI("/ii" + i));
     }
     mLocalTachyonCluster.stopTFS();
     RenameTestUtil();
-    String editLogPath = mLocalTachyonCluster.getEditLogPath();
-    UnderFileSystem.get(editLogPath, mMasterTachyonConf).delete(editLogPath, true);
+    deleteFsMasterJournalLogs();
     RenameTestUtil();
   }
\ No newline at end of file
