diff --git a/old_test.java b/new_test.java
index e90db56..ce3a828 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,8 @@
     public void testSetStringIfNotEmpty() {
         String key = "optionalProperty";
         String value = "someValue";
-        Settings.setString(key, value);
-        Settings.setStringIfNotEmpty(key, ""); // NO-OP
-        String expResults = Settings.getString(key);
+        getSettings().setString(key, value);
+        getSettings().setStringIfNotEmpty(key, ""); // NO-OP
+        String expResults = getSettings().getString(key);
         Assert.assertEquals(expResults, value);
     }
\ No newline at end of file
