diff --git a/old_product.java b/new_product.java
index 55ec44a..e4ccc30 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,7 +16,7 @@ public void addMissingPartition(PartitionWrapper partition, String actionId) thr
             else { // new partition from different hcat server/db
                 _addNewEntry(hcatInstanceMap, prefix, tableName, partition, actionId);
             }
-            _registerMessageReceiver(_getTopic(partition));
+            _registerMessageReceiver(partition);
         }
         catch (ClassCastException e) {
             throw new MetadataServiceException(ErrorCode.E1501, e.getCause());
