diff --git a/old_test.java b/new_test.java
index d946f2b..9dd65e7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testAddMissingPartition() throws MetadataServiceException, URISyntaxException {
         Services services = Services.get();
         PartitionDependencyManagerService pdms = services.get(PartitionDependencyManagerService.class);
-        String newHCatDependency = "hcat://hcat.yahoo.com:5080/mydb/clicks/?datastamp=12&region=us";
+        String newHCatDependency = "hcat://hcat.yahoo.com:5080/database/mydb/table/clicks/partition/datastamp=12,region=us";
         String actionId = "myAction";
         pdms.addMissingPartition(newHCatDependency, actionId);
 
