diff --git a/old_test.java b/new_test.java
index 832de33..7b59a2a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,11 +14,13 @@
     mMaster.commitBlock(worker2, 1L, "MEM", 1L, 1L);
     mMaster.commitBlock(worker2, 2L, "MEM", 2L, 1L);
     mMaster.commitBlock(worker2, 3L, "MEM", 3L, 1L);
-    mMaster.removeBlocks(workerBlocks, false /* deleteBlocksMetadata */);
+    mMaster.removeBlocks(workerBlocks, false /* delete */);
     Assert.assertEquals(1L, mMaster.getBlockInfo(1L).getBlockId());
+    Assert.assertTrue(mMaster.getLostBlocks().contains(1L));
 
-    // Test removeBlocks with deleteBlocksMetadata
-    mMaster.removeBlocks(workerBlocks, true /* deleteBlocksMetadata */);
+    // Test removeBlocks with delete
+    mMaster.removeBlocks(workerBlocks, true /* delete */);
     mThrown.expect(BlockInfoException.class);
     mMaster.getBlockInfo(1L);
+    Assert.assertFalse(mMaster.getLostBlocks().contains(1L));
   }
\ No newline at end of file
