diff --git a/old_product.java b/new_product.java
index ef64727..a9fa4d7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public void removeBlocks(List<Long> blockIds, boolean deleteBlocksMetadata) {
+public void removeBlocks(List<Long> blockIds, boolean delete) {
     synchronized (mBlocks) {
       synchronized (mWorkers) {
         for (long blockId : blockIds) {
@@ -13,9 +13,10 @@ public void removeBlocks(List<Long> blockIds, boolean deleteBlocksMetadata) {
               worker.updateToRemovedBlock(true, blockId);
             }
           }
-          mLostBlocks.remove(blockId);
-          if (deleteBlocksMetadata) {
+          if (delete) {
             mBlocks.remove(blockId);
+          } else {
+            mLostBlocks.add(blockId);
           }
         }
       }
