diff --git a/old_product.java b/new_product.java
index f879fa7..c2ea441 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,6 +10,8 @@ public static PRFAlgorithm getPRFAlgorithm(ProtocolVersion protocolVersion, Ciph
             result = PRFAlgorithm.TLS_PRF_SHA384;
         } else if (cipherSuite.usesGOSTR3411()) {
             result = PRFAlgorithm.TLS_PRF_GOSTR3411;
+        } else if (cipherSuite.usesGOSTR34112012()) {
+            result = PRFAlgorithm.TLS_PRF_GOSTR3411_2012_256;
         } else {
             result = PRFAlgorithm.TLS_PRF_SHA256;
         }
