diff --git a/old_test.java b/new_test.java
index 08f7943..4450300 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void append() {
-    error(_FILE_APPEND.args(PATH, "()"), Err.FILE_IS_DIR);
-    error(_FILE_APPEND.args(PATH4, "()"), Err.FILE_NO_DIR);
+    error(_FILE_APPEND.args(PATH, "()"), Err.FILE_IS_DIR_X);
+    error(_FILE_APPEND.args(PATH4, "()"), Err.FILE_NO_DIR_X);
 
     query(_FILE_APPEND.args(PATH1, "0"));
     query(_FILE_SIZE.args(PATH1), "1");
