diff --git a/old_product.java b/new_product.java
index b933afa..a362dd3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 private Array append(final QueryContext qc) throws QueryException {
-    final Array array = array(0, qc);
+    final Array array = toArray(exprs[0], qc);
     final int as = array.arraySize();
     final ValueList vl = new ValueList(as + 1);
     for(final Value v : array.members()) vl.add(v);
