diff --git a/old_test.java b/new_test.java
index ce35ec3..23d34b6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,11 +3,9 @@
         context.setParsedField(YEAR, 2008);
         context.setParsedField(MONTH_OF_YEAR, 6);
         context.setParsed(ZoneOffset.ofHours(16));
-        context.setParsed(ZoneId.of(ZoneOffset.ofHours(18)));
 
         String str = context.toString();
         assertEquals(str.contains("MonthOfYear 6"), true);
         assertEquals(str.contains("Year 2008"), true);
-        assertEquals(str.contains("UTC:+18:00"), true);
         assertEquals(str.contains("+16:00"), true);
     }
\ No newline at end of file
