diff --git a/old_product.java b/new_product.java
index 324f806..00d1dd6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,8 @@
 @Override  // override for Javadoc
     public String toString() {
-        return dateTime.toString() + '[' + zone.toString() + ']';
+        String str = dateTime.toString() + offset.toString();
+        if (offset != zone) {
+            str += '[' + zone.toString() + ']';
+        }
+        return str;
     }
\ No newline at end of file
