diff --git a/old_test.java b/new_test.java
index 0b0634c..0cc43a1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,11 @@
 		Locale localeToSearch = new Locale("en", "US");
 		
 		Concept concept = new Concept();
-		concept.addName(new ConceptName("some name", definedNameLocale));
+		ConceptName fullySpecifiedName = new ConceptName("some name", definedNameLocale);
+		fullySpecifiedName.setConceptNameId(1);
+		fullySpecifiedName.setConceptNameType(ConceptNameType.FULLY_SPECIFIED);
+		fullySpecifiedName.setLocalePreferred(false);
+		concept.addName(fullySpecifiedName);
 		Assert.assertNotNull(concept.getName(localeToSearch, true));
 		Assert.assertEquals("some name", concept.getName(localeToSearch, true).getName());
 	}
\ No newline at end of file
