diff --git a/old_product.java b/new_product.java
index 9546763..203d377 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,12 +9,21 @@ public ConceptName getName(Locale locale, boolean exact) {
 		
 		if (log.isDebugEnabled())
 			log.debug("Getting conceptName for locale: " + locale);
-		
-		// matches on any name in the current locale, or first name available
-		ConceptName bestName = getBestName(locale);
-		
-		if (exact && !bestName.getLocale().equals(locale))
-			return null; // no exact match found
-		else
-			return bestName;
+		if (exact && locale != null) {
+			ConceptName preferredName = getPreferredName(locale);
+			if (preferredName != null)
+				return preferredName;
+			
+			ConceptName fullySpecifiedName = getFullySpecifiedName(locale);
+			if (fullySpecifiedName != null)
+				return fullySpecifiedName;
+			else if (getSynonyms(locale).size() > 0)
+				return getSynonyms(locale).iterator().next();
+			
+			return null;
+			
+		} else {
+			//just get any name
+			return getName();
+		}
 	}
\ No newline at end of file
