diff --git a/old_product.java b/new_product.java
index 9c3f154..bbfdf5b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,11 @@
-public static String getGetterForClass(JavaAttributeInfo attr) {
+public static String getGetterForClass(JavaAttributeInfo attr, int generatedJavaFiles) {
 
         String attrQuaifiedType = getReturnType(attr);
         String attributeName = getSmallCase(attr.getAttributeName());
 
         if (!attr.isListAttr()) {
-            return getGetter(attrQuaifiedType, attributeName);
+            return getGetter(attrQuaifiedType, attributeName, generatedJavaFiles);
         }
         String listAttr = getListString() + attrQuaifiedType + DIAMOND_CLOSE_BRACKET;
-        return getGetter(listAttr, attributeName);
+        return getGetter(listAttr, attributeName, generatedJavaFiles);
     }
\ No newline at end of file
