diff --git a/old_test.java b/new_test.java
index 8162952..6822031 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void getGetterForClassTest() {
         JavaAttributeInfo testAttr = getTestAttribute();
-        String method = getGetterForClass(testAttr);
+        String method = getGetterForClass(testAttr, GENERATE_SERVICE_AND_MANAGER);
         assertThat(true, is(method.contains(PUBLIC + SPACE + STRING_DATA_TYPE + SPACE + GET_METHOD_PREFIX)));
     }
\ No newline at end of file
