diff --git a/old_product.java b/new_product.java
index 754770a..e3cf071 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 private Item write(final QueryContext ctx) throws QueryException {
-    final File path = checkFile(0, ctx);
+    final java.nio.file.Path path = checkPath(0, ctx);
     final B64 archive = (B64) checkType(checkItem(expr[1], ctx), AtomType.B64);
     final TokenSet hs = entries(2, ctx);
 
@@ -9,12 +9,12 @@ private Item write(final QueryContext ctx) throws QueryException {
         final ZipEntry ze = in.entry();
         final String name = ze.getName();
         if(hs == null || hs.delete(token(name)) != 0) {
-          final IOFile file = new IOFile(path.getPath(), name);
+          final java.nio.file.Path file = path.resolve(name);
           if(ze.isDirectory()) {
-            file.md();
+            Files.createDirectory(file);
           } else {
-            file.dir().md();
-            file.write(in.read());
+            Files.createDirectory(file.getParent());
+            Files.write(file, in.read());
           }
         }
       }
