diff --git a/old_test.java b/new_test.java
index c945246..28e0b86 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void write() {
-    error(_FILE_WRITE.args(PATH, "()"), Err.FILE_ID);
-    error(_FILE_WRITE.args(PATH4, "()"), Err.FILE_ND);
+    error(_FILE_WRITE.args(PATH, "()"), Err.FILE_IS_DIR);
+    error(_FILE_WRITE.args(PATH4, "()"), Err.FILE_NO_DIR);
 
     query(_FILE_WRITE.args(PATH1, "0"));
     query(_FILE_SIZE.args(PATH1), "1");
