diff --git a/old_test.java b/new_test.java
index 793f929..f085ef9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,12 +6,12 @@
         SAMLMessageStorage store = token.getMessageStore();
         SAMLCredential result = new SAMLCredential(nameID, assertion, "IDP");
 
-        expect(consumer.processResponse(context, store)).andReturn(result);
+        expect(consumer.processAuthenticationResponse(context, store)).andReturn(result);
         expect(nameID.getValue()).andReturn("Name");
 
         DateTime expiry = new DateTime().plusHours(4);
         AuthnStatement as = createMock(AuthnStatement.class);
-        expect(assertion.getAuthnStatements()).andReturn(Arrays.asList(as));
+        expect(assertion.getAuthnStatements()).andReturn(Arrays.asList(as)).anyTimes();
         expect(as.getSessionNotOnOrAfter()).andReturn(expiry);
 
         replay(as);
