diff --git a/old_product.java b/new_product.java
index 094ba1d..4f8845a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,17 +10,21 @@ public Authentication authenticate(Authentication authentication) throws Authent
         SAMLCredential credential;
 
         try {
-            credential = consumer.processResponse(context, store);
+            credential = consumer.processAuthenticationResponse(context, store);
         } catch (SAMLException e) {
+            samlLogger.log(SAMLConstants.AUTH_N_RESPONSE, SAMLConstants.FAILURE, context, e);
             throw new AuthenticationServiceException("Error validating SAML message", e);
         } catch (ValidationException e) {
             log.debug("Error validating signature", e);
+            samlLogger.log(SAMLConstants.AUTH_N_RESPONSE, SAMLConstants.FAILURE, context);
             throw new AuthenticationServiceException("Error validating SAML message signature", e);
         } catch (org.opensaml.xml.security.SecurityException e) {
             log.debug("Error validating signature", e);
+            samlLogger.log(SAMLConstants.AUTH_N_RESPONSE, SAMLConstants.FAILURE, context);
             throw new AuthenticationServiceException("Error validating SAML message signature", e);
         } catch (DecryptionException e) {
             log.debug("Error decrypting SAML message", e);
+            samlLogger.log(SAMLConstants.AUTH_N_RESPONSE, SAMLConstants.FAILURE, context);
             throw new AuthenticationServiceException("Error decrypting SAML message", e);
         }
 
@@ -31,5 +35,9 @@ public Authentication authenticate(Authentication authentication) throws Authent
         Date expiration = getExpirationDate(credential);
         ExpiringUsernameAuthenticationToken result = new ExpiringUsernameAuthenticationToken(expiration, principal, credential, entitlements);
         result.setDetails(userDetails);
+
+        samlLogger.log(SAMLConstants.AUTH_N_RESPONSE, SAMLConstants.SUCCESS, context, result, null);
+
         return result;
+
     }
\ No newline at end of file
