diff --git a/old_test.java b/new_test.java
index 6fc9db4..3de7f32 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
     LayoutState current = mComponentTree.getMainThreadLayoutState();
     DefaultInternalNode layout = (DefaultInternalNode) current.getLayoutRoot();
 
-    mComponentTree.updateStateSync("root,B,D", createStateUpdate(), "test");
+    mComponentTree.updateStateSync("root,B,D", createStateUpdate(), "test", false);
 
     Set<String> set = new HashSet<>();
     set.add("root");
