diff --git a/old_product.java b/new_product.java
index c36c9e6..8398a91 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,5 +5,6 @@ public void updateStateSync(StateUpdate stateUpdate, String attribution) {
       return;
     }
 
-    mComponentTree.updateStateSync(mComponentScope.getGlobalKey(), stateUpdate, attribution);
+    mComponentTree.updateStateSync(
+        mComponentScope.getGlobalKey(), stateUpdate, attribution, isCreateLayoutInProgress());
   }
\ No newline at end of file
