diff --git a/old_product.java b/new_product.java
index 3376a30..b8d46e4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,8 +3,12 @@ public SetAttributeTOptions toThrift() {
     if (mPinned != null) {
       options.setPinned(mPinned);
     }
+
     if (mTtl != null) {
       options.setTtl(mTtl);
+    }
+
+    if (mTtlAction != null) {
       options.setTtlAction(TtlAction.toThrift(mTtlAction));
     }
 
