diff --git a/old_test.java b/new_test.java
index d369745..17e0c71 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,6 +9,7 @@
     options.setPersisted(persisted);
     options.setPinned(pinned);
     options.setTtl(ttl);
+    options.setTtlAction(TtlAction.FREE);
     SetAttributeTOptions thriftOptions = options.toThrift();
 
     assertTrue(thriftOptions.isSetPersisted());
@@ -16,6 +17,6 @@
     assertTrue(thriftOptions.isSetPinned());
     assertEquals(pinned, thriftOptions.isPinned());
     assertTrue(thriftOptions.isSetTtl());
-    assertEquals(alluxio.thrift.TTtlAction.Delete, thriftOptions.getTtlAction());
+    assertEquals(TTtlAction.Free, thriftOptions.getTtlAction());
     assertEquals(ttl, thriftOptions.getTtl());
   }
\ No newline at end of file
