diff --git a/old_test.java b/new_test.java
index b1202b3..cc75c9f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -65,4 +65,31 @@
 
     // eventually drop database
     query(_DB_DROP.args(dbname));
+
+    // specify additional index options
+    for(final boolean b : new boolean[] { false, true }) {
+      query(_DB_CREATE.args(dbname, "()", "()", " map { 'updindex':=" + b + "() }"));
+      query(_DB_INFO.args(dbname) + "//updindex/text()", b ? "ON" : "OFF");
+    }
+    assertEquals(context.prop.is(Prop.UPDINDEX), false);
+
+    final String[] nopt = { "maxcats", "maxlen", "indexsplitsize", "ftindexsplitsize" };
+    for(final String k : nopt) {
+      query(_DB_CREATE.args(dbname, "()", "()", " map { '" + k + "':=1 }"));
+    }
+    final String[] bopt = { "textindex", "attrindex", "ftindex", "stemming",
+        "casesens", "diacritics" };
+    for(final String k : bopt) {
+      for(final boolean v : new boolean[] { true, false }) {
+        query(_DB_CREATE.args(dbname, "()", "()", " map { '" + k + "':=" + v + "() }"));
+      }
+    }
+    final String[] sopt = { "language", "stopwords" };
+    for(final String k : sopt) {
+      query(_DB_CREATE.args(dbname, "()", "()", " map { '" + k + "':='' }"));
+    }
+
+    error(_DB_CREATE.args(dbname, "()", "()", " map { 'xyz':='abc' }"), Err.BASX_OPTIONS);
+    error(_DB_CREATE.args(dbname, "()", "()", " map { 'maxlen':=-1 }"), Err.BASX_VALUE);
+    error(_DB_CREATE.args(dbname, "()", "()", " map { 'maxlen':='a' }"), Err.BASX_VALUE);
   }
\ No newline at end of file
