diff --git a/old_product.java b/new_product.java
index 66e5e03..e961e32 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static Point2Transform2_F32 transformPixelToRectNorm(CameraPinholeRadial param,
-																RowMatrix_F32 rectify,
-																RowMatrix_F32 rectifyK) {
+																FMatrixRMaj rectify,
+																FMatrixRMaj rectifyK) {
 		return ImplRectifyImageOps_F32.transformPixelToRectNorm(param, rectify, rectifyK);
 	}
\ No newline at end of file
