diff --git a/old_test.java b/new_test.java
index a990700..889d7ee 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
     public void removeSendedMessage() {
-        if (null == messageCacheService.getSendMessage(blockMessage.getHash().getDigestHex())){
+        if (null == messageCacheService.getSendMessage(blockMessage.getHash())) {
             messageCacheService.cacheSendedMessage(blockMessage);
         }
-        assertNotNull(messageCacheService.getSendMessage(blockMessage.getHash().getDigestHex()));
+        assertNotNull(messageCacheService.getSendMessage(blockMessage.getHash()));
         messageCacheService.removeSendedMessage(blockMessage);
-        assertNull(messageCacheService.getSendMessage(blockMessage.getHash().getDigestHex()));
+        assertNull(messageCacheService.getSendMessage(blockMessage.getHash()));
     }
\ No newline at end of file
