diff --git a/old_test.java b/new_test.java
index 6396038..1ab58dc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,6 @@
         boolean k = true;
         double p = 0.5;
         double expResult = 0.5;
-        double result = DiscreteDistributions.Bernoulli(k, p);
+        double result = DiscreteDistributions.bernoulli(k, p);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
