diff --git a/old_product.java b/new_product.java
index 546a232..e77c824 100644
--- a/old_product.java
+++ b/new_product.java
@@ -32,7 +32,7 @@
 		}
 		
 		// do the checks for voided attributes (also in PersonVoidHandler)
-		if (person.isPersonVoided()) {
+		if (person.getPersonVoided()) {
 			
 			if (!StringUtils.hasLength(person.getPersonVoidReason())) {
 				throw new APIException("Person.voided.bit", new Object[] { person });
