diff --git a/old_test.java b/new_test.java
index e4a0e58..fd1be72 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
 	public void handle_shouldUnsetThePersonVoidedBit() throws Exception {
 		UnvoidHandler<Person> handler = new PersonUnvoidHandler();
 		Person person = new Person();
-		person.setPersonVoided(true); // make sure isPersonVoided is set
+		person.setPersonVoided(true); // make sure personVoided is set
 		handler.handle(person, null, null, null);
-		Assert.assertFalse(person.isPersonVoided());
+		Assert.assertFalse(person.getPersonVoided());
 	}
\ No newline at end of file
