diff --git a/old_test.java b/new_test.java
index 70e69a3..5a2ef5c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,14 +5,14 @@
         NulsDigestData hash = NulsDigestData.calcDigestData("abcdefg".getBytes());
         header.setHash(hash);
         smallBlock.setHeader(header);
-        manager.cacheSmallBlock(smallBlock);
+        manager.cacheSmallBlock(requestHash, smallBlock);
         assertTrue(true);
 
         this.getSmallBlock(hash, smallBlock);
 
         this.removeSmallBlock(hash);
 
-        manager.cacheSmallBlock(smallBlock);
+        manager.cacheSmallBlock(requestHash, smallBlock);
 
         this.clear();
     }
\ No newline at end of file
