diff --git a/old_product.java b/new_product.java
index 5a5d820..4132f06 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,3 @@
-public void cacheSmallBlock(SmallBlock smallBlock) {
-        smallBlockCacheMap.put(smallBlock.getHeader().getHash(), smallBlock);
-        smallBlockCacheMap.remove(smallBlock.getHeader().getPreHash());
+public void cacheSmallBlock(NulsDigestData requestHash, SmallBlock smallBlock) {
+        smallBlockCacheMap.put(requestHash, smallBlock);
     }
\ No newline at end of file
