diff --git a/old_product.java b/new_product.java
index 2adc32d..2a580b4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,8 @@ private Item delete(final QueryContext ctx) throws QueryException {
 
     // delete XML resources
     final IntList docs = data.resources.docs(path);
-    for(int i = 0, is = docs.size(); i < is; i++) {
+    final int is = docs.size();
+    for(int i = 0; i < is; i++) {
       ctx.updates.add(new DeleteNode(docs.get(i), data, info), ctx);
     }
     // delete raw resources
