diff --git a/old_test.java b/new_test.java
index 81bef2e..aa00639 100644
--- a/old_test.java
+++ b/new_test.java
@@ -46,11 +46,11 @@
     assertFalse(context.repo.pkgDict().contains(token(PKG4ID)));
 
     // check if pkg4 directory was deleted
-    assertTrue(!dir(pkg4Dir));
+    assertFalse(dir(pkg4Dir));
     // try to delete pkg3 (use package dir)
     new RepoDelete(PKG3ID, null).execute(context);
     // check if pkg3 is unregistered from the repo
     assertFalse(context.repo.pkgDict().contains(token(PKG3ID)));
     // check if pkg3 directory was deleted
-    assertTrue(!dir(pkg3Dir));
+    assertFalse(dir(pkg3Dir));
   }
\ No newline at end of file
