diff --git a/old_product.java b/new_product.java
index 7c4f8c1..4308626 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,11 +5,11 @@ public void move(int oldPos, int newPos) {
         } else {
             checkIndex(oldPos);
             checkIndex(newPos);
-            E object = nonManagedList.remove(oldPos);
+            E object = unmanagedList.remove(oldPos);
             if (newPos > oldPos) {
-                nonManagedList.add(newPos - 1, object);
+                unmanagedList.add(newPos - 1, object);
             } else {
-                nonManagedList.add(newPos, object);
+                unmanagedList.add(newPos, object);
             }
         }
     }
\ No newline at end of file
