diff --git a/old_product.java b/new_product.java
index e0fc2c3..855a48b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
 public void flush() {
         HashSet<Long> elements;
-        synchronized (this.cache) {
+        synchronized (this.readIndex) {
             elements = this.recentStreamSegmentIds;
             this.recentStreamSegmentIds = new HashSet<>();
         }
 
-        this.cache.triggerFutureReads(elements);
+        this.readIndex.triggerFutureReads(elements);
     }
\ No newline at end of file
