diff --git a/old_test.java b/new_test.java
index 317f344..feb1ce4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,16 +5,16 @@
 
         // Add to MTL + Add to ReadIndex (append; beginMerge).
         MemoryOperationLog opLog = new MemoryOperationLog();
-        ArrayList<TestCache.MethodInvocation> methodInvocations = new ArrayList<>();
-        TestCache cache = new TestCache(methodInvocations::add);
+        ArrayList<TestReadIndex.MethodInvocation> methodInvocations = new ArrayList<>();
+        TestReadIndex cache = new TestReadIndex(methodInvocations::add);
         MemoryLogUpdater updater = new MemoryLogUpdater(opLog, cache);
         ArrayList<Operation> operations = populate(updater, segmentCount, operationCountPerType);
 
         methodInvocations.clear(); // We've already tested up to here.
         updater.flush();
         Assert.assertEquals("Unexpected number of calls to the ReadIndex.", 1, methodInvocations.size());
-        TestCache.MethodInvocation mi = methodInvocations.get(0);
-        Assert.assertEquals("No call to ReadIndex.triggerFutureReads() after call to flush().", TestCache.TRIGGER_FUTURE_READS, mi.methodName);
+        TestReadIndex.MethodInvocation mi = methodInvocations.get(0);
+        Assert.assertEquals("No call to ReadIndex.triggerFutureReads() after call to flush().", TestReadIndex.TRIGGER_FUTURE_READS, mi.methodName);
         Collection<Long> triggerSegmentIds = (Collection<Long>) mi.args.get("streamSegmentIds");
         HashSet<Long> expectedSegmentIds = new HashSet<>();
         for (Operation op : operations) {
