diff --git a/old_product.java b/new_product.java
index 9a3533f..0aa0cd6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,8 @@
 @Override
 	public List<OrderFrequency> getOrderFrequencies(boolean includeRetired) {
 		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(OrderFrequency.class);
-		if (!includeRetired)
+		if (!includeRetired) {
 			criteria.add(Restrictions.eq("retired", false));
+		}
 		return criteria.list();
 	}
\ No newline at end of file
