diff --git a/old_product.java b/new_product.java
index 4648a27..716fcca 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 @Override
-    public List<SchemaDto> getUserNotificationSchemasByAppId(String applicationId) throws ControlServiceException {
+    public List<VersionDto> getUserNotificationSchemasByAppId(String applicationId) throws ControlServiceException {
         return notificationService.findUserNotificationSchemasByAppId(applicationId);
     }
\ No newline at end of file
