diff --git a/old_product.java b/new_product.java
index 6065b2d..4607487 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,8 @@ public static String formatDisplayText(String displayText) {
         if (displayText == null) {
             return null;
         }
-        if (MyTextUtils.isAllCaps(displayText) && displayText.contains(" ")) {
+        // See #883 for "SPLC" logic
+        if (MyTextUtils.isAllCaps(displayText) && displayText.contains(" ") && !displayText.contains("SPLC")) {
             return MyTextUtils.toTitleCase(displayText);
         } else {
             return displayText;
