diff --git a/old_product.java b/new_product.java
index fa4af4b..318e092 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
-private void clientClientMessage(boolean async) throws Exception {
+private void clientClientMessage() throws Exception {
         Ignite ignite = grid(CLIENT_NODE_IDX);
 
         ClusterGroup grp = ignite.cluster().forClients();
 
         assert !grp.nodes().isEmpty();
 
-        registerListenerAndSendMessages(ignite, grp, async);
+        registerListenerAndSendMessages(ignite, grp);
     }
\ No newline at end of file
