diff --git a/old_product.java b/new_product.java
index 5176cc9..7825606 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,5 +5,8 @@ public ServiceCall<WorkspaceExport> getWorkspace(GetWorkspaceOptions getWorkspac
     if (getWorkspaceOptions.export() != null) {
       builder.query("export", String.valueOf(getWorkspaceOptions.export()));
     }
+    if (getWorkspaceOptions.includeAudit() != null) {
+      builder.query("include_audit", String.valueOf(getWorkspaceOptions.includeAudit()));
+    }
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(WorkspaceExport.class));
   }
\ No newline at end of file
