diff --git a/old_product.java b/new_product.java
index 397feff..b31beab 100644
--- a/old_product.java
+++ b/new_product.java
@@ -17,13 +17,16 @@ public void add(Permission permission) {
                 permission));
         } else { 
             // this is the first element provided that another thread did not add
-            synchronized (items) {
+            // BEGIN android-changed
+            // copied from a newer version of harmony
+            synchronized (this) {
                 if (permClass != null && inClass != permClass) {
                     throw new IllegalArgumentException(Messages.getString("security.16", //$NON-NLS-1$
                         permission));
                 }
                 permClass = inClass;
             }
+            // END android-changed
         }
 
         String name = permission.getName();
