diff --git a/old_test.java b/new_test.java
index 3c0b5df..53266a0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,9 @@
-@TestInfo(
-      level = TestLevel.COMPLETE,
-      purpose = "",
-      targets = {
-        @TestTarget(
-          methodName = "add",
-          methodArgs = {Object.class}
-        )
-    })
+@TestTargetNew(
+        level = TestLevel.COMPLETE,
+        notes = "",
+        method = "add",
+        args = {java.lang.Object.class}
+    )
     @SuppressWarnings("unchecked")
     public void test_add_E() {
         Set<EnumFoo> set = EnumSet.noneOf(EnumFoo.class);
