diff --git a/old_product.java b/new_product.java
index 1406d28..771a3f5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,19 +6,17 @@ public boolean changeState(StateVertex nextState) {
 		LOGGER.debug("Trying to change to state: '{}' from: '{}'", nextState.getName(),
 		        currentState.getName());
 
-		synchronized (stateLock) {
-			if (stateFlowGraph.canGoTo(currentState, nextState)) {
+		if (stateFlowGraph.canGoTo(currentState, nextState)) {
 
-				LOGGER.debug("Changed to state: '{}' from: '{}'", nextState.getName(),
-				        currentState.getName());
+			LOGGER.debug("Changed to state: '{}' from: '{}'", nextState.getName(),
+			        currentState.getName());
 
-				currentState = nextState;
+			currentState = nextState;
 
-				return true;
-			} else {
-				LOGGER.info("Cannot go to state: '{}' from: '{}'", nextState.getName(),
-				        currentState.getName());
-				return false;
-			}
+			return true;
+		} else {
+			LOGGER.info("Cannot go to state: '{}' from: '{}'", nextState.getName(),
+			        currentState.getName());
+			return false;
 		}
 	}
\ No newline at end of file
