diff --git a/old_test.java b/new_test.java
index f0502e6..d946f2b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
         pdms.addMissingPartition(newHCatDependency, actionId);
 
         HCatURI hcatUri = new HCatURI(newHCatDependency);
-        Map<String, PartitionsGroup> tablePartitionsMap = pdms.getHCatMap().get(hcatUri.getServer() + "#" +
+        Map<String, PartitionsGroup> tablePartitionsMap = pdms.getHCatMap().get(hcatUri.getServerEndPoint() + "#" +
                                                                             hcatUri.getDb()); // clicks
         assertNotNull(tablePartitionsMap);
         assertTrue(tablePartitionsMap.containsKey("clicks"));
