diff --git a/old_product.java b/new_product.java
index 0dfe19f..b1f8bea 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ public void addMissingPartition(String hcatURI, String actionId) throws Metadata
         catch (URISyntaxException e) {
             throw new MetadataServiceException(ErrorCode.E1503, e.getMessage());
         }
-        PartitionWrapper partition = new PartitionWrapper(uri.getServer(), uri.getDb(), uri.getTable(),
+        PartitionWrapper partition = new PartitionWrapper(uri.getServerEndPoint(), uri.getDb(), uri.getTable(),
                 uri.getPartitionMap());
         addMissingPartition(partition, actionId);
     }
\ No newline at end of file
