diff --git a/old_product.java b/new_product.java
index 95dec8d..252f114 100644
--- a/old_product.java
+++ b/new_product.java
@@ -26,6 +26,19 @@ private void processServiceComponentHosts(Cluster cluster, KerberosDescriptor ke
 
             KerberosServiceDescriptor serviceDescriptor = kerberosDescriptor.getService(serviceName);
 
+            if (!StringUtils.isEmpty(hostName)) {
+              // Update the configurations with the relevant hostname
+              Map<String, String> generalProperties = currentConfigurations.get("");
+              if (generalProperties == null) {
+                generalProperties = new HashMap<>();
+                currentConfigurations.put("", generalProperties);
+              }
+
+              // Add the current hostname under "host" and "hostname"
+              generalProperties.put("host", hostName);
+              generalProperties.put("hostname", hostName);
+            }
+
             if (serviceDescriptor != null) {
               List<KerberosIdentityDescriptor> serviceIdentities = serviceDescriptor.getIdentities(true, filterContext);
 
