diff --git a/old_test.java b/new_test.java
index 0ab4e3f..89bc9a7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test public void testEmpty() throws Exception {
     that()
-        .with(OptiqAssert.Config.JDBC_FOODMART)
+        .with(CalciteAssert.Config.JDBC_FOODMART)
         .doWithConnection(
-            new Function<OptiqConnection, Object>() {
-              public Object apply(OptiqConnection a0) {
+            new Function<CalciteConnection, Object>() {
+              public Object apply(CalciteConnection a0) {
                 try {
                   ResultSet rset =
                       a0.getMetaData().getPrimaryKeys(
