diff --git a/old_product.java b/new_product.java
index 2db5cb8..397e8e2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -18,7 +18,7 @@ protected void internalReleaseConnection(ConnectionHandle connectionHandle) thro
 		}
 
 
-		connectionHandle.setConnectionLastUsed(System.currentTimeMillis());
+		connectionHandle.setConnectionLastUsedInMs(System.currentTimeMillis());
 		if (!this.poolShuttingDown){
 
 			putConnectionBackInPartition(connectionHandle);
