diff --git a/old_test.java b/new_test.java
index fa941d8..48fe16d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,8 +5,8 @@
 		for( int i = 0; i < 12; i++ ) {
 			AssociatedPair p = new AssociatedPair();
 
-			p.currLoc.set(rand.nextDouble()*5,rand.nextDouble()*5);
-			p.keyLoc.set(rand.nextDouble() * 5, rand.nextDouble() * 5);
+			p.p2.set(rand.nextDouble()*5,rand.nextDouble()*5);
+			p.p1.set(rand.nextDouble() * 5, rand.nextDouble() * 5);
 
 			list.add(p);
 		}
