diff --git a/old_product.java b/new_product.java
index 3aec1b9..ce88c01 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,8 +4,8 @@ public static void computeNormalization(List<AssociatedPair> points, DenseMatrix
 		double meanX2 = 0; double meanY2 = 0;
 
 		for( AssociatedPair p : points ) {
-			meanX1 += p.keyLoc.x;  meanY1 += p.keyLoc.y;
-			meanX2 += p.currLoc.x; meanY2 += p.currLoc.y;
+			meanX1 += p.p1.x;  meanY1 += p.p1.y;
+			meanX2 += p.p2.x; meanY2 += p.p2.y;
 		}
 
 		meanX1 /= points.size(); meanY1 /= points.size();
@@ -15,13 +15,13 @@ public static void computeNormalization(List<AssociatedPair> points, DenseMatrix
 		double stdX2 = 0; double stdY2 = 0;
 
 		for( AssociatedPair p : points ) {
-			double dx = p.keyLoc.x - meanX1;
-			double dy = p.keyLoc.y - meanY1;
+			double dx = p.p1.x - meanX1;
+			double dy = p.p1.y - meanY1;
 			stdX1 += dx*dx;
 			stdY1 += dy*dy;
 
-			dx = p.currLoc.x - meanX2;
-			dy = p.currLoc.y - meanY2;
+			dx = p.p2.x - meanX2;
+			dy = p.p2.y - meanY2;
 			stdX2 += dx*dx;
 			stdY2 += dy*dy;
 		}
