diff --git a/old_product.java b/new_product.java
index 54bdac3..924df2c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,29 +2,27 @@ public void replace(String pathInZooKeeper, int expectedVersion, T state) throws
 		checkNotNull(pathInZooKeeper, "Path in ZooKeeper");
 		checkNotNull(state, "State");
 
-		StateHandle<T> oldStateHandle = get(pathInZooKeeper);
+		RetrievableStateHandle<T> oldStateHandle = get(pathInZooKeeper);
 
-		StateHandle<T> stateHandle = storage.store(state);
+		RetrievableStateHandle<T> newStateHandle = storage.store(state);
 
 		boolean success = false;
 
 		try {
 			// Serialize the new state handle. This writes the state to the backend.
-			byte[] serializedStateHandle = InstantiationUtil.serializeObject(stateHandle);
+			byte[] serializedStateHandle = InstantiationUtil.serializeObject(newStateHandle);
 
 			// Replace state handle in ZooKeeper.
 			client.setData()
 					.withVersion(expectedVersion)
 					.forPath(pathInZooKeeper, serializedStateHandle);
-
 			success = true;
-		}
-		finally {
-			if (success) {
+		} finally {
+			if(success) {
 				oldStateHandle.discardState();
-			}
-			else {
-				stateHandle.discardState();
+			} else {
+				newStateHandle.discardState();
 			}
 		}
+
 	}
\ No newline at end of file
