diff --git a/old_test.java b/new_test.java
index 0e5e590..7edecde 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,6 +10,7 @@
 
         DefaultSessionCacheFactory cacheFactory = new DefaultSessionCacheFactory();
         cacheFactory.setEvictionPolicy(SessionCache.NEVER_EVICT);
+        cacheFactory.setSaveOnCreate(true);
         DefaultSessionCache cache = (DefaultSessionCache)cacheFactory.getSessionCache(context.getSessionHandler());
 
         TestSessionDataStore store = new TestSessionDataStore();
@@ -31,9 +32,8 @@
         assertFalse(cache.contains("1234"));
 
         //test remove of session in both store and cache
-        data = store.newSessionData("1234", now - 20, now - 10, now - 20, TimeUnit.MINUTES.toMillis(10));
-        session = cache.newSession(data);
-        cache.put("1234", session);
+        session = cache.newSession(null, "1234",now - 20 ,TimeUnit.MINUTES.toMillis(10));//saveOnCreate ensures write to store
+        cache.add("1234", session);
         assertTrue(store.exists("1234"));
         assertTrue(cache.contains("1234"));
         session = cache.delete("1234");
