diff --git a/old_product.java b/new_product.java
index e94032a..790df41 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@
     public Session delete(String id) throws Exception
     {
         //get the session, if its not in memory, this will load it
-        Session session = get(id);
+        Session session = getAndEnter(id, false);
 
         //Always delete it from the backing data store
         if (_sessionDataStore != null)
