diff --git a/old_product.java b/new_product.java
index 7ed1f0b..f3983ed 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,21 +1,20 @@
-public InodeTree.CreatePathResult mkdir(AlluxioURI path, CreateDirectoryOptions options)
+public InodeTree.CreatePathResult createDirectory(AlluxioURI path, CreateDirectoryOptions options)
       throws InvalidPathException, FileAlreadyExistsException, IOException, AccessControlException,
       FileDoesNotExistException {
-    LOG.debug("mkdir {} ", path);
+    LOG.debug("createDirectory {} ", path);
     MasterContext.getMasterSource().incCreateDirectoriesOps(1);
     synchronized (mInodeTree) {
       try {
-        checkPermission(FileSystemAction.WRITE, path, true);
+        mPermissionChecker.checkParentPermission(FileSystemAction.WRITE, path);
         if (!options.isMetadataLoad()) {
           mMountTable.checkUnderWritableMountPoint(path);
         }
         InodeTree.CreatePathResult createResult = mInodeTree.createPath(path, options);
-
-        LOG.debug("writing journal entry for mkdir {}", path);
+        LOG.debug("writing journal entry for createDirectory {}", path);
         writeJournalEntry(mDirectoryIdGenerator.toJournalEntry());
         journalCreatePathResult(createResult);
         flushJournal();
-        LOG.debug("flushed journal for mkdir {}", path);
+        LOG.debug("flushed journal for createDirectory {}", path);
         MasterContext.getMasterSource().incDirectoriesCreated(1);
         return createResult;
       } catch (BlockInfoException e) {
