diff --git a/old_product.java b/new_product.java
index b558ac5..d77cd64 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@
       @QueryParam("blockSizeBytes") Long blockSizeBytes, @QueryParam("ttl") Long ttl) {
     try {
       Preconditions.checkNotNull(path, "required 'path' parameter is missing");
-      CreateFileOptions options = CreateFileOptions.defaults();
+      CreatePathOptions options = CreatePathOptions.defaults();
       if (persisted != null) {
         options.setPersisted(persisted);
       }
