diff --git a/old_product.java b/new_product.java
index 5a33c9a..bde5f11 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,10 @@
 @Override
     public ChronoZonedDateTime<C> withEarlierOffsetAtOverlap() {
         ZoneOffsetTransition trans = getZone().getRules().getTransition(LocalDateTime.from(this));
-        if (trans != null) {
-            ZoneOffset offset = trans.getOffsetBefore();
-            if (offset.equals(getOffset()) == false) {
-                ChronoOffsetDateTimeImpl<C> newDT = dateTime.withOffsetSameLocal(offset);
-                return new ChronoZonedDateTimeImpl<C>(newDT, zone);
+        if (trans != null && trans.isOverlap()) {
+            ZoneOffset earlierOffset = trans.getOffsetBefore();
+            if (earlierOffset.equals(offset) == false) {
+                return new ChronoZonedDateTimeImpl<C>(dateTime, earlierOffset, zoneId);
             }
         }
         return this;
