diff --git a/old_test.java b/new_test.java
index e2bff62..691f84e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,8 @@
 @Test(groups={"tck"})
-    public void test_withEarlierOffsetAtOverlap() {
-        OffsetDateTime odt = OffsetDateTime.of(2008, 10, 26, 2, 30, OFFSET_0100);
+    public void test_withEarlierOffsetAtOverlap_atOverlap() {
+        OffsetDateTime odt = OffsetDateTime.of(TEST_PARIS_OVERLAP_2008_10_26_02_30, OFFSET_0100);
         ZonedDateTime base = ZonedDateTime.of(odt, ZONE_PARIS);
         ZonedDateTime test = base.withEarlierOffsetAtOverlap();
         assertEquals(test.getOffset(), OFFSET_0200);  // offset changed to earlier
+        assertEquals(test.getDateTime(), base.getDateTime());  // date-time not changed
     }
\ No newline at end of file
