diff --git a/old_product.java b/new_product.java
index 317860f..7054fca 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,4 @@
 @Override
-    public Double geoDist(String key, String member1, String member2, GeoUnit geoUnit) {
-        Double result = delegate.geoDist(serialize(key), serialize(member1), serialize(member2), geoUnit);
-        if (isFutureConversion()){
-            addResultConverter(identityConverter);
-        }
-        return result;
-    }
\ No newline at end of file
+	public Distance geoDist(String key, String member1, String member2, Metric metric) {
+		return geoDist(serialize(key), serialize(member1), serialize(member2), metric);
+	}
\ No newline at end of file
