diff --git a/old_test.java b/new_test.java
index caca1f6..01f5222 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,6 @@
 @Test
-	public void testGeoDist(){
-		doReturn(Arrays.asList(new Object[] { 102121.12d })).when(nativeConnection).closePipeline();
+	public void testGeoDist() {
+		doReturn(Arrays.asList(new Distance(102121.12d, DistanceUnit.METERS))).when(nativeConnection).closePipeline();
 		super.testGeoDist();
+
 	}
\ No newline at end of file
