diff --git a/old_test.java b/new_test.java
index 415895b..15e002c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test
     public void testListLogFiles() throws IOException {
-        FileAttribute[] attrs = new FileAttribute[0];
-        String rootPath = Files.createTempDirectory("workers-artifacts", attrs).toFile().getCanonicalPath();
+        String rootPath = Files.createTempDirectory("workers-artifacts").toFile().getCanonicalPath();
         File file1 = new File(String.join(File.separator, rootPath, "topoA", "1111"), "worker.log");
         File file2 = new File(String.join(File.separator, rootPath, "topoA", "2222"), "worker.log");
         File file3 = new File(String.join(File.separator, rootPath, "topoB", "1111"), "worker.log");
@@ -16,7 +15,7 @@
         String origin = "www.origin.server.net";
         Map<String, Object> stormConf = Utils.readStormConfig();
         StormMetricsRegistry metricsRegistry = new StormMetricsRegistry();
-        LogviewerLogPageHandler handler = new LogviewerLogPageHandler(rootPath, null,
+        LogviewerLogPageHandler handler = new LogviewerLogPageHandler(rootPath, rootPath,
                 new WorkerLogs(stormConf, Paths.get(rootPath), metricsRegistry), new ResourceAuthorizer(stormConf), metricsRegistry);
 
         final Response expectedAll = LogviewerResponseBuilder.buildSuccessJsonResponse(
