diff --git a/old_product.java b/new_product.java
index c205982..62a2d53 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public ServiceResponse<PagedList<CertificateItem>> listCertificateVersions(final String vaultBaseUrl, final String certificateName) 
+public PagedList<CertificateItem> listCertificateVersions(final String vaultBaseUrl, final String certificateName) 
             throws KeyVaultErrorException, IOException, IllegalArgumentException {
         return innerKeyVaultClient.getCertificateVersions(vaultBaseUrl, certificateName);
     }
\ No newline at end of file
