diff --git a/old_test.java b/new_test.java
index c77f106..9043cee 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,12 +4,12 @@
 		final Operator<?> input2 = new OpImpl(1);
 		final Operator<?> fixture = new OpImpl(0).withInputs(input1, input2);
 
-		final List<Operator<?>.Output> result = fixture.getInputs();
+		final List<JsonStream> result = fixture.getInputs();
 
 		assertNotNull(result);
 		assertEquals(2, result.size());
 
-		final List<Operator<?>.Output> expectedResults = new ArrayList<Operator<?>.Output>();
+		final List<JsonStream> expectedResults = new ArrayList<JsonStream>();
 		expectedResults.add(input1.getOutput(0));
 		expectedResults.add(input2.getOutput(0));
 		assertEquals(expectedResults, result);
