diff --git a/old_test.java b/new_test.java
index 8d3abfc..14bfff5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,5 +6,5 @@
       Assert.assertFalse(mAlluxioOutStreamMap.get(streamIndex).isCanceled());
       Assert.assertTrue(mAlluxioOutStreamMap.get(streamIndex).isClosed());
     }
-    verify(mFileSystemMasterClient).completeFile(eq(FILE_NAME), any(CompleteFileOptions.class));
+    verify(mFileSystemMasterClient).completeFile(eq(FILE_NAME), any(CompleteFilePOptions.class));
   }
\ No newline at end of file
