diff --git a/old_product.java b/new_product.java
index 1a45a3f..57ecf99 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery le(String propName, Comparable<?> value) {
-        criterion = criterion.and(criterionBuilder.le(propName, value));
+        criterion = criterion.and(Criteria.le(propName, value));
         return this;
     }
\ No newline at end of file
