diff --git a/old_product.java b/new_product.java
index d43fab7..ab32854 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,8 @@
-public MFramework getFramework() {
-    if(framework != null) {
-      return framework.clone(false);
+public MDriverConfig getDriverConfig() {
+    if(driverConfig != null) {
+      return driverConfig.clone(false);
     }
-
-    retrieveFramework();
-    return framework.clone(false);
+    retrieveAndCacheDriverConfig();
+    return driverConfig.clone(false);
 
   }
\ No newline at end of file
