diff --git a/old_test.java b/new_test.java
index 5111bdc..ae694fe 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
-  public void testGetFramework() {
-    when(requests.readFramework()).thenReturn(frameworkBean(framework()));
+  public void testGetDriverConfig() {
+    when(resourceRequests.readDriverConfig()).thenReturn(driverConfigBean(driverConfig()));
 
-    client.getFramework();
-    client.getFrameworkResourceBundle();
+    client.getDriverConfig();
+    client.getDriverConfigBundle();
 
-    verify(requests, times(1)).readFramework();
+    verify(resourceRequests, times(1)).readDriverConfig();
   }
\ No newline at end of file
