diff --git a/old_test.java b/new_test.java
index 3b165df..f069e06 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,8 @@
     Assert.assertFalse(thriftOptions.isReadOnly());
 
     options.setReadOnly(true);
+    options.setShared(true);
     thriftOptions = options.toThrift();
     Assert.assertTrue(thriftOptions.isReadOnly());
+    Assert.assertTrue(thriftOptions.isShared());
   }
\ No newline at end of file
