diff --git a/old_test.java b/new_test.java
index 6175c9b..57552bf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,13 +7,14 @@
     }
 
     // test one level
-    InodeTree.CreatePathResult createResult = createPath(mTree, TEST_URI, sDirectoryOptions);
-    List<Inode<?>> created = createResult.getCreated();
-    Assert.assertEquals(new AlluxioURI("/test"), mTree.getPath(created.get(created.size() - 1)));
+    createPath(mTree, TEST_URI, sDirectoryOptions);
+    try (LockedInodePath inodePath = mTree.lockFullInodePath(TEST_URI, InodeTree.LockMode.READ)) {
+      Assert.assertEquals(new AlluxioURI("/test"), mTree.getPath(inodePath.getInode()));
+    }
 
     // test nesting
-    createResult = createPath(mTree, NESTED_URI, sNestedDirectoryOptions);
-    created = createResult.getCreated();
-    Assert.assertEquals(new AlluxioURI("/nested/test"),
-        mTree.getPath(created.get(created.size() - 1)));
+    createPath(mTree, NESTED_URI, sNestedDirectoryOptions);
+    try (LockedInodePath inodePath = mTree.lockFullInodePath(NESTED_URI, InodeTree.LockMode.READ)) {
+      Assert.assertEquals(new AlluxioURI("/nested/test"), mTree.getPath(inodePath.getInode()));
+    }
   }
\ No newline at end of file
