diff --git a/old_product.java b/new_product.java
index f118df1..61c476f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,3 @@
 public static Write write() {
-    return new AutoValue_TextIO_Write.Builder()
-        .setFilenamePrefix(null)
-        .setShardTemplate(null)
-        .setFilenameSuffix(null)
-        .setFilenamePolicy(null)
-        .setWritableByteChannelFactory(FileBasedSink.CompressionType.UNCOMPRESSED)
-        .setWindowedWrites(false)
-        .setNumShards(0)
-        .build();
+    return new TextIO.Write();
   }
\ No newline at end of file
