diff --git a/old_product.java b/new_product.java
index 6cd90f5..94ea56f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,9 @@
-public static int getInt(String key) {
-        return Integer.parseInt(Settings.getString(key));
+public static int getInt(String key) throws InvalidSettingException {
+        int value;
+        try {
+            value = Integer.parseInt(Settings.getString(key));
+        } catch (NumberFormatException ex) {
+            throw new InvalidSettingException("Could not convert property '" + key + "' to an int.", ex);
+        }
+        return value;
     }
\ No newline at end of file
