diff --git a/old_test.java b/new_test.java
index 6eef767..83f3c5e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,17 +1,17 @@
 @Test
     public void testCompareTo() throws Exception {
         doTestComparisons(
-            HijrahDate.hijrahDate(1, 1, 1),
-            HijrahDate.hijrahDate(1, 1, 2),
-            HijrahDate.hijrahDate(1, 1, 30),
-            HijrahDate.hijrahDate(1, 2, 1),
-            HijrahDate.hijrahDate(1, 2, 29),
-            HijrahDate.hijrahDate(1, 12, 29),
-            HijrahDate.hijrahDate(2, 1, 1),
-            HijrahDate.hijrahDate(2, 12, 29),
-            HijrahDate.hijrahDate(3, 1, 1),
-            HijrahDate.hijrahDate(3, 12, 29),
-            HijrahDate.hijrahDate(4500, 1, 1),
-            HijrahDate.hijrahDate(4500, 12, 29)
+            HijrahDate.of(1, 1, 1),
+            HijrahDate.of(1, 1, 2),
+            HijrahDate.of(1, 1, 30),
+            HijrahDate.of(1, 2, 1),
+            HijrahDate.of(1, 2, 29),
+            HijrahDate.of(1, 12, 29),
+            HijrahDate.of(2, 1, 1),
+            HijrahDate.of(2, 12, 29),
+            HijrahDate.of(3, 1, 1),
+            HijrahDate.of(3, 12, 29),
+            HijrahDate.of(4500, 1, 1),
+            HijrahDate.of(4500, 12, 29)
         );
     }
\ No newline at end of file
