diff --git a/old_test.java b/new_test.java
index c762b22..0478e65 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
     when(mFileDataManager.getPersistedFileInfos()).thenReturn(filesInfo);
     // first time fails, second time passes
     when(mFileSystemMasterClient.heartbeat(anyLong(), eq(persistedFiles)))
-        .thenReturn(new FileSystemCommand());
+        .thenReturn(FileSystemCommand.newBuilder().build());
     mFileWorkerMasterSyncExecutor.heartbeat();
     verify(mFileDataManager).clearPersistedFiles(persistedFiles);
   }
\ No newline at end of file
