diff --git a/old_product.java b/new_product.java
index 14bb967..9ed748c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,5 @@
 @PrePersist
     @PreUpdate
     protected void onCreateOrUpdateApplication() throws GeniePreconditionException {
-        if (this.tags == null) {
-            this.tags = new HashSet<>();
-        }
         this.addAndValidateSystemTags(this.tags);
     }
\ No newline at end of file
