diff --git a/old_test.java b/new_test.java
index 528b3b0..288052f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testOnCreateOrUpdateApplication() throws GeniePreconditionException {
         this.a = new Application(NAME, USER, VERSION, ApplicationStatus.ACTIVE);
-        Assert.assertNull(this.a.getTags());
+        Assert.assertNotNull(this.a.getTags());
         this.a.onCreateOrUpdateApplication();
         Assert.assertEquals(2, this.a.getTags().size());
     }
\ No newline at end of file
