diff --git a/old_product.java b/new_product.java
index 9e92781..afcdcae 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-public Cohort addMembershipToCohort(Cohort cohort, CohortMembership cohortMembership) throws APIException {
+@Override
+	public Cohort addMembershipToCohort(Cohort cohort, CohortMembership cohortMembership) throws APIException {
 		cohort.addMembership(cohortMembership);
-		Context.getCohortService().saveCohort(cohort);
-		return cohort;
+		return Context.getCohortService().saveCohort(cohort);
 	}
\ No newline at end of file
