diff --git a/old_test.java b/new_test.java
index e9a5918..b701126 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,6 @@
         initial.setId(uuid.toString());
         initial.setJobName("My test job");
         initial.setJobStatus(JobStatus.FAILED);
-        initial.setUpdateTime(System.currentTimeMillis());
         initial.setUserName("myUserName");
         initial.setCmdArgs("commandArg");
         pm.createEntity(initial);
