diff --git a/old_product.java b/new_product.java
index 8b760d3..5faa580 100644
--- a/old_product.java
+++ b/new_product.java
@@ -17,10 +17,10 @@
             if (orderByUpdateTime) {
                 boolean desc = builder.isDesc();
                 if (desc) {
-                    strQuery += String.format(" order by %s.updateTime desc",
+                    strQuery += String.format(" order by %s.updated desc",
                             ENTITY_ALIAS);
                 } else {
-                    strQuery += String.format(" order by %s.updateTime asc",
+                    strQuery += String.format(" order by %s.updated asc",
                             ENTITY_ALIAS);
                 }
             }
