diff --git a/old_product.java b/new_product.java
index 58ce8cf..3148f32 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,6 @@
-public Period plus(Period other) {
+public Period plus(Period amountToAdd) {
         return create(
-                Jdk8Methods.safeAdd(years, other.years),
-                Jdk8Methods.safeAdd(months, other.months),
-                Jdk8Methods.safeAdd(days, other.days),
-                Jdk8Methods.safeAdd(nanos, other.nanos));
+                Jdk8Methods.safeAdd(years, amountToAdd.years),
+                Jdk8Methods.safeAdd(months, amountToAdd.months),
+                Jdk8Methods.safeAdd(days, amountToAdd.days));
     }
\ No newline at end of file
