diff --git a/old_product.java b/new_product.java
index a7963b3..6128ae9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,3 @@
-public synchronized boolean drop(final User user, final String pattern) {
-    if(pattern.isEmpty()) {
-      if(users.remove(user.name()) == null) return false;
-    } else {
-      user.remove(pattern);
-    }
-    return true;
+public synchronized boolean drop(final User user) {
+    return users.remove(user.name()) != null;
   }
\ No newline at end of file
