diff --git a/old_test.java b/new_test.java
index 1b3084a..4852d6f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,14 +9,14 @@
         map.put(4L, new User(4L));
 
         if (LOGGER.isDebugEnabled()){
-            LOGGER.debug(JsonUtil.format(MapUtil.constructSubMap(map, "id")));
+            LOGGER.debug(JsonUtil.format(MapUtil.constructSubMap(map, "id", Long.class)));
         }
         if (LOGGER.isDebugEnabled()){
             Long[] includeKeys = { 5L, 4L };
-            LOGGER.debug(JsonUtil.format(MapUtil.constructSubMap(map, includeKeys, "id")));
+            LOGGER.debug(JsonUtil.format(MapUtil.constructSubMap(map, includeKeys, "id", Long.class)));
         }
         if (LOGGER.isDebugEnabled()){
             Long[] includeKeys = { 5L, 4L };
-            LOGGER.debug(JsonUtil.format(MapUtil.constructSubMap(map, includeKeys, "userInfo.age")));
+            LOGGER.debug(JsonUtil.format(MapUtil.constructSubMap(map, includeKeys, "userInfo.age", Long.class)));
         }
     }
\ No newline at end of file
