diff --git a/old_product.java b/new_product.java
index d3c4b53..8274954 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static <T> Object selectableConvert(
-      Type type, Object x, Object what, @Nullable Label context)
+      Type<T> type, Object x, Object what, LabelConversionContext context)
       throws ConversionException {
     if (x instanceof com.google.devtools.build.lib.syntax.SelectorList) {
       return new SelectorList<T>(
