diff --git a/old_test.java b/new_test.java
index cab58c3..e729f6f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,14 +10,16 @@
         ImmutableList.of(Label.create("@//a", "a1"), Label.create("@//a", "a2"));
 
     // Conversion to direct type:
-    Object converted = BuildType
-        .selectableConvert(BuildType.LABEL_LIST, nativeInput, null, currentRule);
+    Object converted =
+        BuildType.selectableConvert(
+            BuildType.LABEL_LIST, nativeInput, null, labelConversionContext);
     assertThat(converted instanceof List<?>).isTrue();
     assertThat((List<Label>) converted).containsExactlyElementsIn(expectedLabels);
 
     // Conversion to selectable type:
-    converted = BuildType
-        .selectableConvert(BuildType.LABEL_LIST, selectableInput, null, currentRule);
+    converted =
+        BuildType.selectableConvert(
+            BuildType.LABEL_LIST, selectableInput, null, labelConversionContext);
     BuildType.SelectorList<?> selectorList = (BuildType.SelectorList<?>) converted;
     assertThat(((Selector<Label>) selectorList.getSelectors().get(0)).getEntries().entrySet())
         .containsExactlyElementsIn(
