diff --git a/old_test.java b/new_test.java
index 1e0f3bc..a9211d0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     AuthenticationProvider authenticationProvider = systemResource.getAuthenticationProvider();
 
     // default configuration should be JCR
-    AuthenticationProvider expectedResult = new AuthenticationProvider(PentahoSpringBeansConfig.AuthenticationProvider.JCR_BASED_AUTHENTICATION.toString());
+    AuthenticationProvider expectedResult = new AuthenticationProvider("jackrabbit");
 
     Assert.assertTrue(authenticationProvider.toString().equals(expectedResult.toString()));
   }
\ No newline at end of file
