diff --git a/old_product.java b/new_product.java
index 38c35f3..e2ce09f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,11 +3,9 @@
   @Produces({MediaType.APPLICATION_JSON})
   public AuthenticationProvider getAuthenticationProvider() throws Exception {
     try{
-      File configFile = new File(PentahoSystem.getApplicationContext().getSolutionPath("system/pentaho-spring-beans.xml"));
-      // File configFile = new File("/home/pminutillo/IntelliJProjects/pentaho-platform/extensions/test-res/solution1-no-config/system/pentaho-spring-beans.xml");
-      PentahoSpringBeansConfig config = new PentahoSpringBeansConfig(configFile);
-
-      return new AuthenticationProvider(config.getAuthenticationProvider().toString());
+      IConfiguration config = this.systemConfig.getConfiguration("security");
+      String provider = config.getProperties().getProperty("provider");
+      return new AuthenticationProvider(provider);
     }
     catch (Throwable t) {
       logger.error(Messages.getInstance().getString("SystemResource.GENERAL_ERROR"), t); //$NON-NLS-1$
