diff --git a/old_product.java b/new_product.java
index 8b3a5cf..c62bda5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,15 +1,8 @@
-public static Pair<String, List<Process>> setupZkTunnel(Config config) {
+public static Pair<String, List<Process>> setupZkTunnel(Config config,
+                                                          NetworkUtils.TunnelConfig tunnelConfig) {
     // Remove all spaces
     String connectionString = Context.stateManagerConnectionString(config).replaceAll("\\s+", "");
 
-    // Read values from config
-    String tunnelHost = ZkContext.tunnelHost(config);
-    int timeout = ZkContext.tunnelConnectionTimeoutMs(config);
-    int retry = ZkContext.tunnelConnectionRetryCount(config);
-    int retryInterval = ZkContext.tunnelRetryIntervalMs(config);
-    int verifyCount = ZkContext.tunnelVerifyCount(config);
-    boolean isVerbose = Context.verbose(config);
-
     List<Pair<InetSocketAddress, Process>> ret = new ArrayList<>();
 
     // For zookeeper, connection String can be a list of host:port, separated by comma
@@ -20,7 +13,7 @@ public static Pair<String, List<Process>> setupZkTunnel(Config config) {
       // Get the tunnel process if needed
       Pair<InetSocketAddress, Process> pair =
           NetworkUtils.establishSSHTunnelIfNeeded(
-              address, tunnelHost, timeout, retry, retryInterval, verifyCount, isVerbose);
+              address, tunnelConfig, NetworkUtils.TunnelType.PORT_FORWARD);
 
       ret.add(pair);
     }
