diff --git a/old_test.java b/new_test.java
index 557fb0d..7c965c4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
         logger.info("Uniform");
         int n = 10;
         double expResult = 0.1;
-        double result = DiscreteDistributions.Uniform(n);
+        double result = DiscreteDistributions.uniform(n);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
