diff --git a/old_product.java b/new_product.java
index 3bf280c..d6ee52c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,8 +9,8 @@ public static <T> double[][] bootstrap(int k, RegressionTrainer<T> trainer, T[]
         
         Bootstrap bootstrap = new Bootstrap(n, k);
         for (int i = 0; i < k; i++) {
-            T[] trainx = Math.slice(x, bootstrap.train[i]);
-            double[] trainy = Math.slice(y, bootstrap.train[i]);
+            T[] trainx = MathEx.slice(x, bootstrap.train[i]);
+            double[] trainy = MathEx.slice(y, bootstrap.train[i]);
             
             Regression<T> model = trainer.train(trainx, trainy);
 
