diff --git a/old_test.java b/new_test.java
index fb59c11..593d40b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -19,7 +19,7 @@
         instance.fit(trainingData, param);
         
         instance.close();
-        instance = null;
+        //instance = null;
         instance = new CollaborativeFiltering(dbName, dbConf);
         
         CollaborativeFiltering.ValidationMetrics vm = instance.validate(validationData);
