diff --git a/old_product.java b/new_product.java
index dd7aab1..f702f69 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public VM validate(Dataframe testingData) {  
         logger.info("validate()");
         
-        knowledgeBase.load();
+        kb().load();
 
         //validate the model with the testing data and update the validationMetrics
         VM validationMetrics = validateModel(testingData);
