diff --git a/old_test.java b/new_test.java
index 372292f..6834905 100644
--- a/old_test.java
+++ b/new_test.java
@@ -38,7 +38,7 @@
       fail("testCreateGatewayReceiver failed as did not get CommandResult");
     }
     vm3.invoke(() -> verifyReceiverCreationWithAttributes(false, 10000, 11000, "localhost", 100000,
-        512000, null));
+        512000, null, "localhost"));
     vm4.invoke(() -> verifyReceiverCreationWithAttributes(false, 10000, 11000, "localhost", 100000,
-        512000, null));
+        512000, null, "localhost"));
   }
\ No newline at end of file
