diff --git a/old_product.java b/new_product.java
index 6f7acf7..255ec56 100644
--- a/old_product.java
+++ b/new_product.java
@@ -43,5 +43,10 @@ private static GatewayReceiver createGatewayReceiver(Cache cache,
                 CliStrings.CREATE_GATEWAYRECEIVER__GATEWAYTRANSPORTFILTER));
       }
     }
+
+    String hostnameForSenders = gatewayReceiverCreateArgs.getHostnameForSenders();
+    if (hostnameForSenders != null) {
+      gatewayReceiverFactory.setHostnameForSenders(hostnameForSenders);
+    }
     return gatewayReceiverFactory.create();
   }
\ No newline at end of file
