diff --git a/old_test.java b/new_test.java
index b1b39a5..9176a28 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void reportLostFileTest() throws Exception {
-    TachyonURI path = new TachyonURI("test");
+    AlluxioURI path = new AlluxioURI("test");
     mTachyonLineageFileSystem.reportLostFile(path);
     Mockito.verify(mLineageMasterClient).reportLostFile("test");
     // verify client is released
