diff --git a/old_product.java b/new_product.java
index 45f60e5..85993b4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-public void reportLostFile(TachyonURI path)
-      throws IOException, FileDoesNotExistException, TachyonException {
+public void reportLostFile(AlluxioURI path)
+      throws IOException, FileDoesNotExistException, AlluxioException {
     LineageMasterClient masterClient = mLineageContext.acquireMasterClient();
     try {
       masterClient.reportLostFile(path.getPath());
