diff --git a/old_product.java b/new_product.java
index 62039df..644d9b0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public Authentication authenticate(Authentication authentication) throws AuthenticationException {
 
         if (!supports(authentication.getClass())) {
-            throw new IllegalArgumentException("Only SAMLAuthenticationToken is supported, "+authentication.getClass()+" was attempted");
+            throw new IllegalArgumentException("Only SAMLAuthenticationToken is supported, " + authentication.getClass() + " was attempted");
         }
 
         SAMLAuthenticationToken token = (SAMLAuthenticationToken) authentication;
@@ -22,7 +22,8 @@ public Authentication authenticate(Authentication authentication) throws Authent
         }
 
         String name = credential.getNameID().getValue();
-        UsernamePasswordAuthenticationToken result = new UsernamePasswordAuthenticationToken(name, credential, new GrantedAuthority[0]);
+        Date expiration = getExpirationDate(credential);
+        ExpiringUsernameAuthenticationToken result = new ExpiringUsernameAuthenticationToken(expiration, name, credential, new GrantedAuthority[0]);
         processUserDetails(result, credential);
         return result;
     }
\ No newline at end of file
