diff --git a/old_test.java b/new_test.java
index aa2f1a2..03bbef8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,12 +2,16 @@
     public void testApplyDelegate() {
         Config config = Config.createConfig();
         config.setHighestProtocolVersion(ProtocolVersion.SSL2);
-        config.setTransportHandlerType(TransportHandlerType.EAP_TLS);
+        config.setDefaultTransportHandlerType(TransportHandlerType.EAP_TLS);
         args = new String[2];
         args[0] = "-version";
         args[1] = "TLS12";
         jcommander.parse(args);
+        assertTrue(config.getHighestProtocolVersion() == ProtocolVersion.SSL2);
+        assertTrue(config.getDefaultTransportHandlerType() == TransportHandlerType.EAP_TLS);
+        assertTrue(config.getConnectionEnd().getTransportHandlerType() == TransportHandlerType.EAP_TLS);
         delegate.applyDelegate(config);
         assertTrue(config.getHighestProtocolVersion() == ProtocolVersion.TLS12);
-        assertTrue(config.getTransportHandlerType() == TransportHandlerType.TCP);
+        assertTrue(config.getDefaultTransportHandlerType() == TransportHandlerType.TCP);
+        assertTrue(config.getConnectionEnd().getTransportHandlerType() == TransportHandlerType.TCP);
     }
\ No newline at end of file
