diff --git a/old_test.java b/new_test.java
index 1401905..58ac90b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testKFoldCrossValidation() {
-        TestConfiguration.getLogger().debug("kFoldCrossValidation");
+        TestUtils.log(this.getClass(), "kFoldCrossValidation");
         RandomValue.randomGenerator = new Random(42); 
         int k = 5;
         
