diff --git a/old_product.java b/new_product.java
index b17accf..87f5269 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @SuppressWarnings("unchecked")
     public VM kFoldCrossValidation(Dataset trainingData, TP trainingParameters, int k) {
-        logger.debug("kFoldCrossValidation()");
+        logger.info("kFoldCrossValidation()");
         
         initializeTrainingConfiguration(trainingParameters);
         
