diff --git a/old_test.java b/new_test.java
index cef48ec..4c38155 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(expectedExceptions=NullPointerException.class)
-    public void test_print_nullAppendable() throws Exception {
+    public void test_print_nullStringBuilder() throws Exception {
         ReducedPrinterParser pp = new ReducedPrinterParser(YEAR, 2, 2010);
-        pp.print(calendrical2012, (Appendable) null, symbols);
+        pp.print(calendrical2012, (StringBuilder) null, symbols);
     }
\ No newline at end of file
