diff --git a/old_test.java b/new_test.java
index 3b783f9..642e21b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
 		QrCode qr = new QrCodeEncoder().setVersion(1).
 				setError(QrCode.ErrorLevel.M).
 				setMask(new QrCodeMaskPattern.NONE(0b011)).
-				numeric("01234567").fixate();
+				addNumeric("01234567").fixate();
 
 		byte[] expected = new byte[]{0b00010000,
 		0b00100000, 0b00001100, 0b01010110, 0b01100001 ,(byte)0b10000000, (byte)0b11101100, 0b00010001,
