diff --git a/old_test.java b/new_test.java
index 2c45cf5..4e1c4f8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testReceiveString() throws Exception {
         transportHandler.setFetchableByte(ArrayConverter.concatenate(new byte[] { 0x17, 0x03, 0x03, 0x00, 0x04 },
-                "test".getBytes()));
+                "test".getBytes(Charset.forName("ASCII"))));
         String receivedString = socket.receiveString();
         assertEquals("test", receivedString);
     }
\ No newline at end of file
