diff --git a/old_test.java b/new_test.java
index 893a15f..ed66f58 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test
   public void testXMLEntry() throws BaseXException {
-    final String fun = check(FunDef.XMLENTRY, Uri.class, String.class);
-    query(fun + "(xs:anyURI('" + ZIP + "'), '" + ENTRY2 + "')");
-    query(fun + "(xs:anyURI('" + ZIP + "'), '" +
-        ENTRY2 + "')//title/text()", "XML");
+    final String fun = check(FunDef.XMLENTRY, String.class, String.class);
+    query(fun + "('" + ZIP + "', '" + ENTRY2 + "')");
+    query(fun + "('" + ZIP + "', '" + ENTRY2 + "')//title/text()", "XML");
   }
\ No newline at end of file
