diff --git a/old_product.java b/new_product.java
index 081f4c7..83156ae 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,11 @@
-private ANode xmlEntry(final QueryContext ctx) throws QueryException {
+private ANode xmlEntry(final QueryContext ctx, final boolean html)
+      throws QueryException {
+
     final Prop prop = ctx.context.prop;
     final IO io = new IOContent(entry(ctx));
     try {
-      final Parser p = Parser.fileParser(io, prop, "");
+      final Parser p = html ? new HTMLParser(io, "", prop) :
+        Parser.xmlParser(io, prop, "");
       return new DBNode(MemBuilder.build(p, prop, ""), 0);
     } catch(final IOException ex) {
       throw SAXERR.thrw(input, ex);
