diff --git a/old_test.java b/new_test.java
index b18917c..9d61c53 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void getStatusTest() throws Exception {
-    TachyonURI file = new TachyonURI("/file");
+    AlluxioURI file = new AlluxioURI("/file");
     URIStatus status = new URIStatus(new FileInfo());
     Mockito.when(mFileSystemMasterClient.getStatus(file)).thenReturn(status);
     GetStatusOptions getStatusOptions = GetStatusOptions.defaults();
