diff --git a/old_product.java b/new_product.java
index 35ba8b0..f4991a6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
-public synchronized URIStatus getStatus(final TachyonURI path) throws IOException,
-      TachyonException {
+public synchronized URIStatus getStatus(final AlluxioURI path) throws IOException,
+      AlluxioException {
     return retryRPC(new RpcCallableThrowsTachyonTException<URIStatus>() {
       @Override
-      public URIStatus call() throws TachyonTException, TException {
+      public URIStatus call() throws AlluxioTException, TException {
         return new URIStatus(ThriftUtils.fromThrift(mClient.getStatus(path.getPath())));
       }
     });
