diff --git a/old_product.java b/new_product.java
index 08bcd0e..7c52f55 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 public Pair<ExecutionReference, ExecutableFlow> fetchHead()
-    throws InterruptedException {
-    Pair<ExecutionReference, ExecutableFlow> pair = queuedFlowList.take();
+      throws InterruptedException {
+    final Pair<ExecutionReference, ExecutableFlow> pair = this.queuedFlowList.take();
     if (pair != null && pair.getFirst() != null) {
-      queuedFlowMap.remove(pair.getFirst().getExecId());
+      this.queuedFlowMap.remove(pair.getFirst().getExecId());
     }
     return pair;
   }
\ No newline at end of file
