diff --git a/old_test.java b/new_test.java
index 47da624..318c90b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
   public void testFetchHead() throws IOException, ExecutorManagerException,
-    InterruptedException {
-    QueuedExecutions queue = new QueuedExecutions(5);
-    List<Pair<ExecutionReference, ExecutableFlow>> dataList = getDummyData();
+      InterruptedException {
+    final QueuedExecutions queue = new QueuedExecutions(5);
+    final List<Pair<ExecutionReference, ExecutableFlow>> dataList = getDummyData();
     Assert.assertTrue(queue.isEmpty());
     queue.enqueueAll(dataList);
     Assert.assertEquals(queue.fetchHead(), dataList.get(0));
