diff --git a/old_test.java b/new_test.java
index 81fe3e0..cfff1a0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test
     public void testInsert() {
         System.out.println("insert");
-        int expResult = 0;
-        int result = instance.insert(MOCK_TABLE, MOCK_KEY0, MOCK_DATA);
-        assertEquals(expResult, result);
+        Status result = instance.insert(MOCK_TABLE, MOCK_KEY0, MOCK_DATA);
+        assertEquals(Status.OK, result);
     }
\ No newline at end of file
