diff --git a/old_product.java b/new_product.java
index d249f5b..230dcd3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
-  public int insert(String table, String key,
+  public Status insert(String table, String key,
       HashMap<String, ByteIterator> values) {
 
     try {
@@ -25,10 +25,10 @@
 
       ResultSet rs = session.execute(insertStmt);
 
-      return StatusCode.OK;
+      return Status.OK;
     } catch (Exception e) {
       e.printStackTrace();
     }
 
-    return StatusCode.ERROR;
+    return Status.ERROR;
   }
\ No newline at end of file
