diff --git a/old_test.java b/new_test.java
index e9467ec..1ef8bbc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,8 +4,8 @@
         System.out.println("dump");
 
         RunsTable instance = createHorizontalInstance();
-        instance.dump(System.out);
+        System.out.println(instance.dumpOf());
 
         instance = createVerticalInstance();
-        instance.dump(System.out);
+        System.out.println(instance.dumpOf());
     }
\ No newline at end of file
