diff --git a/old_product.java b/new_product.java
index c693a74..7e995b3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static String format(Number value,String numberPattern) throws NullPointerException{
+public static String format(Number value,String numberPattern){
         // 如果不设置, DecimalFormat默认使用的是 RoundingMode.HALF_EVEN
         RoundingMode roundingMode = RoundingMode.HALF_UP;
         return format(value, numberPattern, roundingMode);
