diff --git a/old_test.java b/new_test.java
index 2f4ed77..82aa511 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public final void format() throws ParseException{
+    public final void format(){
         LOGGER.info(MessageFormatUtil.format("name=张三{0}a{1}", "jin", "xin"));
         LOGGER.info(MessageFormatUtil.format("name=张三{0,number}a{1}", 5, "xin"));
         LOGGER.info(MessageFormatUtil.format("name=张三{0,date}a{1}", 15, "xin"));
