diff --git a/old_test.java b/new_test.java
index fef816a..f32f429 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,4 +7,12 @@
         int expResult = -2;
         int result = instance.compareTo(vs);
         assertEquals(expResult, result);
+
+        vs = new VulnerableSoftware();
+        vs.setCpe("cpe:/a:some:dep:9.2.0.0-20090116170000");
+        instance = new VulnerableSoftware();
+        instance.setCpe("cpe:/a:some:dep:9.2.0.0-20090116170001");
+        expResult = 1;
+        result = instance.compareTo(vs);
+        assertEquals(expResult, result);
     }
\ No newline at end of file
