diff --git a/old_test.java b/new_test.java
index d49d9c8..5f3e363 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void handleAllServersAtOnceTest() throws MalformedObjectNameException {
-        JmxRequest request = new JmxRequestBuilder(READ, testBeanName).
+        JmxReadRequest request = new JmxRequestBuilder(READ, testBeanName).
                 attribute("attr").
                 build();
         assertFalse(handler.handleAllServersAtOnce(request));
