diff --git a/old_product.java b/new_product.java
index 4336f6f..f373ac5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,23 +7,33 @@ public static List<String> mergeValues(Class<?> type, String cfg, List<String> d
                 }
             }
         }
+        
 	    List<String> names = new ArrayList<String>();
+	    
+	    // 加入初始值
+        String[] configs = (cfg == null || cfg.trim().length() == 0) ? new String[0] : Constants.COMMA_SPLIT_PATTERN.split(cfg);
+        for (String config : configs) {
+            if(config != null && config.trim().length() > 0) {
+                names.add(config);
+            }
+        }
+
+        // 不包含 -default
         if (! names.contains(Constants.REMOVE_VALUE_PREFIX + Constants.DEFAULT_KEY)) {
+            // 加入 插入缺省扩展点
             int i = names.indexOf(Constants.DEFAULT_KEY);
             if (i > 0) {
                 names.addAll(i, defaults);
             } else {
-                names.addAll(defaults);
+                names.addAll(0, defaults);
             }
             names.remove(Constants.DEFAULT_KEY);
         }
-        String[] configs = cfg == null ? new String[0] : Constants.COMMA_SPLIT_PATTERN.split(cfg);
-        for (String config : configs) {
-            if(config != null && config.length() > 0) {
-                String[] fs = Constants.COMMA_SPLIT_PATTERN.split(config);
-                names.addAll(Arrays.asList(fs));
-            }
+        else {
+            names.remove(Constants.DEFAULT_KEY);
         }
+        
+        // 合并-的配置项
         for (String name : new ArrayList<String>(names)) {
             if (name.startsWith(Constants.REMOVE_VALUE_PREFIX)) {
                 names.remove(name);
