diff --git a/old_test.java b/new_test.java
index 151440e..480dfe4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testMergeValues() {
         List<String> merged = ConfigUtils.mergeValues(Serialization.class, "aaa,bbb,default.cunstom",
-                                                      Arrays.asList(new String[]{"dubbo","default.hessian2","json"}));
-        Assert.assertEquals("[dubbo, json, aaa, bbb, default.cunstom]",merged.toString());
+                toArray("dubbo","default.hessian2","json"));
+        Assert.assertEquals(toArray("dubbo", "json", "aaa", "bbb", "default.cunstom"), merged);
     }
\ No newline at end of file
