diff --git a/old_test.java b/new_test.java
index 02ffc29..5d0a3cd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -25,7 +25,7 @@
     .respond(
         response().withHeaders(
             new Header(HttpHeaders.Names.CONTENT_TYPE, HttpMediaType.APPLICATION_JSON)).withBody(
-            GsonSingleton.getGsonWithoutPrettyPrinting().toJson(speechResults)));
+            GSON.toJson(speechResults)));
     final SpeechResults result = service.recognize(audio).execute();
     Assert.assertNotNull(result);
     Assert.assertEquals(result, speechResults);
