diff --git a/old_product.java b/new_product.java
index a0429b2..66e68b9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,14 +1,3 @@
-public void recognize(InputStream stream, RecognizeOptions options) {
-      if (connected != true)
-        connect(options);
-
-      try {
-        socket.sendMessage(RequestBody.create(WebSocket.TEXT, buildStartMessage(options).body().toString()));
-        sendInputSteam(stream);
-        socket.sendMessage(RequestBody.create(WebSocket.TEXT, buildStopMessage().body().toString()));
-      } catch (IOException e) {
-        e.printStackTrace();
-      }
-
-
-    }
\ No newline at end of file
+public void recognize(final InputStream stream, final RecognizeOptions options, RecognizeCallback delegate) {
+    createConnection(options).enqueue(new SpeechToTextWebSocketListener(stream, options, delegate));
+  }
\ No newline at end of file
