diff --git a/old_test.java b/new_test.java
index 212f1c9..d53eb74 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public final void create() throws IOException {
     session.create(NAME, new ArrayInput(""));
-    check("", session.query("doc('" + NAME + "')").execute());
+    assertEqual("", session.query("doc('" + NAME + "')").execute());
     session.create(NAME, new ArrayInput("<X/>"));
-    check("<X/>", session.query("doc('" + NAME + "')").execute());
+    assertEqual("<X/>", session.query("doc('" + NAME + "')").execute());
   }
\ No newline at end of file
