diff --git a/old_test.java b/new_test.java
index b6528e6..5640863 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,10 +5,12 @@
 		tp.addToAllowed_modes(TravelMode.CAR);
 
 		LatLng originLatLng = new LatLng(1.0, 2.5);
-		Location origin = new Location(originLatLng);
+		Location origin = new Location();
+		origin.setLat_lng(originLatLng);
 
 		LatLng destLatLng = new LatLng(-3.0, 9.7);
-		Location dest = new Location(destLatLng);
+		Location dest = new Location();
+		dest.setLat_lng(destLatLng);
 
 		tp.setOrigin(origin);
 		tp.setDestination(dest);
