diff --git a/old_product.java b/new_product.java
index 9b24916..574e688 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,6 +4,13 @@ public RoutingRequestBuilder addTripParameters(TripParameters tripParams) {
 			setTravelModes(new TravelModeSet(allowedModes));
 		}
 
+		// Set trip timing information
+		if (tripParams.isSetStart_time()) {
+			setStartTime(tripParams.getStart_time());
+		} else if (tripParams.isSetArrive_by()) {
+			setArriveBy(tripParams.getArrive_by());
+		}
+		
 		setOrigin(tripParams.getOrigin().getLat_lng());
 		setDestination(tripParams.getDestination().getLat_lng());
 
