diff --git a/old_product.java b/new_product.java
index ff7bdf6..9d9f381 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public ServiceCall<Corpus> getCorpus(String customizationId, String corpusName) {
-    Validator.notNull(customizationId, "customizationId cannot be null");
-    Validator.notNull(corpusName, "corpusName cannot be null");
-    RequestBuilder requestBuilder = RequestBuilder.get(String.format(PATH_CORPUS, customizationId, corpusName));
-    return createServiceCall(requestBuilder.build(), ResponseConverterUtils.getObject(Corpus.class));
+public ServiceCall<Corpus> getCorpus(GetCorpusOptions getCorpusOptions) {
+    Validator.notNull(getCorpusOptions, "getCorpusOptions cannot be null");
+    RequestBuilder builder = RequestBuilder.get(String.format("/v1/customizations/%s/corpora/%s", getCorpusOptions
+        .customizationId(), getCorpusOptions.corpusName()));
+    return createServiceCall(builder.build(), ResponseConverterUtils.getObject(Corpus.class));
   }
\ No newline at end of file
