diff --git a/old_test.java b/new_test.java
index 767a875..29839e5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,11 @@
 
     server.enqueue(new MockResponse().addHeader(CONTENT_TYPE, HttpMediaType.APPLICATION_JSON).setBody("{}"));
 
-    service.getCorpus(id, corpus).execute();
+    GetCorpusOptions getOptions = new GetCorpusOptions.Builder()
+        .customizationId(id)
+        .corpusName(corpus)
+        .build();
+    service.getCorpus(getOptions).execute();
     final RecordedRequest request = server.takeRequest();
 
     assertEquals("GET", request.getMethod());
