diff --git a/old_test.java b/new_test.java
index 209168a..c2e00e0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,13 +8,13 @@
 
 		// Trigger
 		sampleFutures.add(coord.triggerStackTraceSample(
-				vertices, 1, new FiniteDuration(100, TimeUnit.MILLISECONDS), 0));
+				vertices, 1, Time.milliseconds(100L), 0));
 
 		sampleFutures.add(coord.triggerStackTraceSample(
-				vertices, 1, new FiniteDuration(100, TimeUnit.MILLISECONDS), 0));
+				vertices, 1, Time.milliseconds(100L), 0));
 
 		for (Future<StackTraceSample> future : sampleFutures) {
-			assertFalse(future.isCompleted());
+			assertFalse(future.isDone());
 		}
 
 		// Shut down
@@ -22,13 +22,20 @@
 
 		// Verify all completed
 		for (Future<StackTraceSample> future : sampleFutures) {
-			assertTrue(future.isCompleted());
+			assertTrue(future.isDone());
 		}
 
 		// Verify new trigger returns failed future
 		Future<StackTraceSample> future = coord.triggerStackTraceSample(
-				vertices, 1, new FiniteDuration(100, TimeUnit.MILLISECONDS), 0);
+				vertices, 1, Time.milliseconds(100L), 0);
+
+		assertTrue(future.isDone());
+
+		try {
+			future.get();
+			fail("Expected exception.");
+		} catch (ExecutionException e) {
+			// we expected an exception here :-)
+		}
 
-		assertTrue(future.isCompleted());
-		assertTrue(future.failed().isCompleted());
 	}
\ No newline at end of file
