diff --git a/old_product.java b/new_product.java
index cb018ae..5a438d5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@ public JComponent learn() {
         String[] labels = dataset[datasetIndex].names();
 
         long clock = System.currentTimeMillis();
-        IsotonicMDS isomds = new IsotonicMDS(data, 2);
+        IsotonicMDS isomds = IsotonicMDS.of(data, 2);
         System.out.format("Learn Kruskal's Nonmetric MDS (k=2) from %d samples in %dms\n", data.length, System.currentTimeMillis()-clock);
 
         PlotCanvas plot = ScatterPlot.plot(isomds.getCoordinates(), labels);
@@ -12,7 +12,7 @@ public JComponent learn() {
         pane.add(plot);
 
         clock = System.currentTimeMillis();
-        isomds = new IsotonicMDS(data, 3);
+        isomds = IsotonicMDS.of(data, 3);
         System.out.format("Learn Kruskal's Nonmetric MDS (k=3) from %d samples in %dms\n", data.length, System.currentTimeMillis()-clock);
 
         plot = ScatterPlot.plot(isomds.getCoordinates(), labels);
