diff --git a/old_test.java b/new_test.java
index 500c7aa..b684fd7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -51,7 +51,7 @@
             {-7.62133456223296, 30.975890997191055}
         };
 
-        IsotonicMDS mds = new IsotonicMDS(swiss, 2);
+        IsotonicMDS mds = IsotonicMDS.of(swiss);
 
         double sign = Math.signum(points[0][0] / mds.getCoordinates()[0][0]);
         for (int i = 0; i < points.length; i++) {
@@ -64,16 +64,5 @@
         }
 
         assertEquals(0.023190, mds.getStress(), 1E-6);
-        for (int i = 0; i < 5; i++) {
-            for (int j = 0; j < points[i].length; j++)
-                System.out.print(points[i][j] + " ");
-            System.out.println();
-        }
-        System.out.println("==============");
-        for (int i = 0; i < 5; i++) {
-            for (int j = 0; j < mds.getCoordinates()[i].length; j++)
-                System.out.print(mds.getCoordinates()[i][j] + " ");
-            System.out.println();
-        }
         assertTrue(MathEx.equals(points, mds.getCoordinates(), 1E-6));
     }
\ No newline at end of file
