diff --git a/old_test.java b/new_test.java
index de5b818..f4ce211 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void testSynthesize() throws IOException {
     String text = "This is an integration test";
-    InputStream result = service.synthesize(text, Voice.EN_LISA, HttpMediaType.AUDIO_WAV);
+    InputStream result = service.synthesize(text, Voice.EN_LISA, HttpMediaType.AUDIO_WAV).execute();
     writeInputStreamToFile(result, File.createTempFile("tts-audio", "wav"));
   }
\ No newline at end of file
