diff --git a/old_test.java b/new_test.java
index 996d4ec..7dae9a7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
             public DateRange getVal() { return super.getVal(); }
             public void setVal(final DateRange val) { super.setVal(val); }
         };
-        final DynamoDBMapperTableModel<Object> model = models.getTableModel((Class<Object>)obj.getClass());
+        final DynamoDBMapperTableModel<Object> model = getTableModel(obj);
         assertEquals(3, model.fields().size());
         assertEquals("DateRangeStart", model.field("DateRangeStart").name());
         assertEquals("DateRangeEnd", model.field("DateRangeEnd").name());
