diff --git a/old_product.java b/new_product.java
index d18d89d..03ddb47 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,6 +12,12 @@ public boolean submitTopology(String appConf) {
     // Setup Connection
     String schedulerURI = String.format("%s/v2/groups", this.marathonURI);
     HttpURLConnection conn = NetworkUtils.getHttpConnection(schedulerURI);
+
+    // Attach a token if there is one specified
+    if (this.marathonAuthToken != null) {
+      conn.setRequestProperty("Authorization", String.format("token=%s", this.marathonAuthToken));
+    }
+
     if (conn == null) {
       LOG.log(Level.SEVERE, "Failed to find marathon scheduler");
       return false;
@@ -30,6 +36,9 @@ public boolean submitTopology(String appConf) {
       if (success) {
         LOG.log(Level.INFO, "Topology submitted successfully");
         return true;
+      } else if (NetworkUtils.checkHttpResponseCode(conn, HttpURLConnection.HTTP_UNAUTHORIZED)) {
+        LOG.log(Level.SEVERE, "Marathon requires authentication");
+        return false;
       } else {
         LOG.log(Level.SEVERE, "Failed to submit topology");
         return false;
