diff --git a/old_test.java b/new_test.java
index ce81003..afa4f65 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,6 +8,7 @@
     Method updateTezConfiguration = UpgradeCatalog200.class.getDeclaredMethod("updateTezConfiguration");
     Method updateConfigurationProperties = AbstractUpgradeCatalog.class.getDeclaredMethod
             ("updateConfigurationProperties", String.class, Map.class, boolean.class, boolean.class);
+    Method persistHDPRepo = UpgradeCatalog200.class.getDeclaredMethod("persistHDPRepo");
 
     UpgradeCatalog200 upgradeCatalog = createMockBuilder(UpgradeCatalog200.class)
         .addMockedMethod(removeNagiosService)
@@ -16,6 +17,7 @@
         .addMockedMethod(setSecurityType)
         .addMockedMethod(updateTezConfiguration)
         .addMockedMethod(updateConfigurationProperties)
+        .addMockedMethod(persistHDPRepo)
         .createMock();
 
     upgradeCatalog.removeNagiosService();
@@ -35,6 +37,9 @@
             Collections.singletonMap("hive.server2.transport.mode", "binary"), false, false);
     expectLastCall();
 
+    upgradeCatalog.persistHDPRepo();
+    expectLastCall().once();
+
     replay(upgradeCatalog);
 
     upgradeCatalog.executeDMLUpdates();
