diff --git a/old_test.java b/new_test.java
index c6e21a7..9d95abc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
     public void testSetSequence ()
     {
-        System.out.println("setSequence");
+        System.out.println("\n+++ setSequence");
 
         int index = 1;
-        RunSequence seq = new BasicRunSequence(new short[]{0, 3, 7});
+        short[] seq = new short[]{0, 3, 7};
         RunTable instance = createHorizontalInstance();
         System.out.println("table before:" + instance.dumpOf());
         instance.setSequence(index, seq);
