diff --git a/old_test.java b/new_test.java
index 728d5eb..c7a658b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,9 +5,9 @@
 
         RunTable instance = createHorizontalInstance();
         RunTable expResult = createHorizontalInstance();
-        RunTable result = instance.copy("hori");
+        RunTable result = instance.copy();
 
-        if (!expResult.isIdentical(result) || !expResult.getName().equals(result.getName())) {
+        if (!expResult.isIdentical(result)) {
             fail("Copy not identical to original");
         }
     }
\ No newline at end of file
