diff --git a/old_test.java b/new_test.java
index 8947d4d..1fedde6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     recorder.start(span, START_TIME_MICROSECONDS);
     brave.serverSpanThreadBinder().setCurrentSpan(serverSpan);
 
-    PowerMockito.when(System.nanoTime()).thenReturn(500L);
+    timestamp = START_TIME_MICROSECONDS; // no time passed!
 
     brave.serverTracer().setServerSend();
 
