diff --git a/old_product.java b/new_product.java
index 51f812b..12ff58d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,15 +1,16 @@
 public void assignToNode(long processInstanceId, long nodeId) {
+		UserTransaction owner = null;
 		try {
-			this.getJbpmSupport().startTransaction();
+			owner = startUserTransaction();
 			this.assignToNodeCall(processInstanceId, nodeId);
-			this.getJbpmSupport().commitTransaction();
+			this.commitUserTransaction(owner);
 		} catch (RuntimeException e) {
 			e.printStackTrace();
-			getJbpmSupport().rollbackTransaction();
+			rollbackUserTransaction(owner);
 			throw e;
 		} catch (Exception e) {
 			e.printStackTrace();
-			this.getJbpmSupport().rollbackTransaction();
+			this.rollbackUserTransaction(owner);
 			throw new RuntimeException(e.getCause());
 		}
 	}
\ No newline at end of file
