diff --git a/old_product.java b/new_product.java
index cab7329..809ad82 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,9 +10,12 @@ public synchronized void paintRow(Graphics2D g2d, AbstractGraphRow row, Color co
 
       mustDrawFirstZeroingLine = chartSettings.isDrawFinalZeroingLines();
 
+      olStroke = g2d.getStroke();
+
       if (row.isDrawThickLines()) {
-         olStroke = g2d.getStroke();
-         lineStroke = chartSettings.getThickStroke();
+         g2d.setStroke(chartSettings.getThickStroke());
+      } else {
+         g2d.setStroke(getLineStroke());
       }
 
       g2d.setColor(color);
@@ -88,7 +91,5 @@ public synchronized void paintRow(Graphics2D g2d, AbstractGraphRow row, Color co
 
       postPaintRow(row, g2d);
 
-      if(olStroke != null) {
-         g2d.setStroke(olStroke);
-      }
+      g2d.setStroke(olStroke);
    }
\ No newline at end of file
