diff --git a/old_product.java b/new_product.java
index 94d8811..fcecf34 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,11 +9,11 @@ public String makeAppSecretProof() {
       String out = new String(hex, "UTF-8");
       return out;
     } catch (NoSuchAlgorithmException e) {
-      System.out.println(e);
+      logger.info("Appsecret_proof creation fialed: " + e);
     } catch (InvalidKeyException e) {
-      System.out.println(e);
+      logger.warning("Appsecret_proof creation fialed: " + e);
     } catch (UnsupportedEncodingException e) {
-      System.out.println(e);
+      logger.info("Appsecret_proof creation fialed: " + e);
     }
     throw new NullPointerException("AppSecretProof creation has failed");
   }
\ No newline at end of file
