diff --git a/old_test.java b/new_test.java
index 9be1979..1b524c7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testGetLong() throws InvalidSettingException {
         String key = "SomeNumber";
         long expResult = 300L;
-        Settings.setString(key, "300");
-        long result = Settings.getLong(key);
+        getSettings().setString(key, "300");
+        long result = getSettings().getLong(key);
         Assert.assertEquals(expResult, result);
     }
\ No newline at end of file
