diff --git a/old_product.java b/new_product.java
index 69c47ff..b57b3fa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public static long getLong(String key) throws InvalidSettingException {
+public long getLong(String key) throws InvalidSettingException {
         try {
-            return Long.parseLong(Settings.getString(key));
+            return Long.parseLong(getString(key));
         } catch (NumberFormatException ex) {
             throw new InvalidSettingException("Could not convert property '" + key + "' to a long.", ex);
         }
