diff --git a/old_product.java b/new_product.java
index 82b59b2..1c8ad88 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,24 +1,25 @@
 public void applyDeltaToMatchingLaterStockCandidates(
-			@NonNull final MaterialDescriptor materialDescriptor,
-			@NonNull final Integer groupId,
-			@NonNull final BigDecimal delta)
+			@NonNull final Candidate stockWithDelta)
 	{
-		final MaterialDescriptorQuery materialDescriptorQuery = MaterialDescriptorQuery
-				.forDescriptor(materialDescriptor, DateOperator.AFTER);
-
-		final CandidatesQuery query = CandidatesQuery.builder()
-				.type(CandidateType.STOCK)
-				.materialDescriptorQuery(materialDescriptorQuery)
-				.parentId(CandidatesQuery.UNSPECIFIED_PARENT_ID)
-				.matchExactStorageAttributesKey(true)
-				.build();
+		final CandidatesQuery query = createStockQueryBuilderWithDateOperator(
+				stockWithDelta,
+				DateOperator.AT_OR_AFTER);
 
 		final List<Candidate> candidatesToUpdate = candidateRepositoryRetrieval.retrieveOrderedByDateAndSeqNo(query);
 		for (final Candidate candidate : candidatesToUpdate)
 		{
+			final boolean sameDateButLowerSeqNo = //
+					candidate.getDate().equals(stockWithDelta.getDate())
+							&& candidate.getSeqNo() <= stockWithDelta.getSeqNo();
+			if (sameDateButLowerSeqNo)
+			{
+				continue;
+			}
+
+			final BigDecimal delta = stockWithDelta.getQuantity();
 			final BigDecimal newQty = candidate.getQuantity().add(delta);
 			candidateRepositoryWriteService.updateCandidateById(candidate
 					.withQuantity(newQty)
-					.withGroupId(groupId));
+					.withGroupId(stockWithDelta.getGroupId()));
 		}
 	}
\ No newline at end of file
