diff --git a/old_test.java b/new_test.java
index 9116cd3..e4b8918 100644
--- a/old_test.java
+++ b/new_test.java
@@ -58,12 +58,15 @@
 				.productDescriptor(createProductDescriptor())
 				.warehouseId(WAREHOUSE_ID)
 				.date(t2)
-				.quantity(BigDecimal.ZERO) // doesn't matter
+				.quantity(THREE)
 				.build();
-		stockCandidateService.applyDeltaToMatchingLaterStockCandidates(
-				materialDescriptor,
-				earlierCandidate.getGroupId(),
-				new BigDecimal("3"));
+		final Candidate candidateWithDelta = Candidate.builder()
+				.type(CandidateType.STOCK)
+				.clientId(CLIENT_ID)
+				.orgId(ORG_ID)
+				.materialDescriptor(materialDescriptor)
+				.groupId(earlierCandidate.getGroupId()).build();
+		stockCandidateService.applyDeltaToMatchingLaterStockCandidates(candidateWithDelta);
 
 		// assert that every stock record got some groupId
 		assertThat(DispoTestUtils.retrieveAllRecords()).allSatisfy(r -> assertThatModel(r).hasValueGreaterThanZero(I_MD_Candidate.COLUMN_MD_Candidate_GroupId));
