diff --git a/old_test.java b/new_test.java
index 114a227..47c9d60 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,10 +5,10 @@
 		DrugOrder order = new DrugOrder();
 		order.setAction(Order.Action.DISCONTINUE);
 		order.setOrderReasonNonCoded("Discontinue this");
+		order.setDrug(conceptService.getDrug(3));
 		order.setEncounter(encounterService.getEncounter(5));
 		order.setPatient(Context.getPatientService().getPatient(7));
 		order.setOrderer(Context.getProviderService().getProvider(1));
-		order.setConcept(Context.getConceptService().getConcept(88));
 		order.setCareSetting(orderService.getCareSetting(1));
 		order.setEncounter(encounterService.getEncounter(3));
 		order.setOrderType(orderService.getOrderType(1));
