diff --git a/old_test.java b/new_test.java
index ca60643..d5d154f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,8 @@
 @Test
-   public void testGetThreadGroupDelay()
-   {
-      System.out.println("getThreadGroupDelay");
-      SteppingThreadGroup instance = new SteppingThreadGroup();
-      int expResult = 0;
-      int result = instance.getThreadGroupDelay();
-      assertEquals(expResult, result);
-   }
\ No newline at end of file
+    public void testGetThreadGroupDelay() {
+        System.out.println("getThreadGroupDelay");
+        SteppingThreadGroup instance = new SteppingThreadGroup();
+        String expResult = "";
+        String result = instance.getThreadGroupDelay();
+        assertEquals(expResult, result);
+    }
\ No newline at end of file
