diff --git a/old_product.java b/new_product.java
index 9f049b4..e7f682f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,3 @@
-public int getThreadGroupDelay()
-   {
-      return getPropertyAsInt(THREAD_GROUP_DELAY, 0);
-   }
\ No newline at end of file
+public String getThreadGroupDelay() {
+        return getPropertyAsString(THREAD_GROUP_DELAY);
+    }
\ No newline at end of file
