diff --git a/old_test.java b/new_test.java
index 3b9ece9..0f067e2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,18 +10,13 @@
 
     for(final IOFile f : new IOFile("etc").children()) {
       final String n = f.name();
-      final StringList missing = new StringList();
-      final StringList obsolete = new StringList();
-      if(n.endsWith(".bat")) libraries(n, libs, missing, obsolete);
-      if(!missing.isEmpty()) {
-        final StringBuilder sb = new StringBuilder();
-        for(final String l : missing) sb.append(";%LIB%").append(l);
-        fail("Library not found in '" + n + "':\n" + sb.substring(1));
-      }
-      if(!obsolete.isEmpty()) {
-        final StringBuilder sb = new StringBuilder();
-        for(final String l : obsolete) sb.append(";%LIB%").append(l);
-        fail("Library obsolete in '" + n + "':\n" + sb.substring(1));
+      if(n.endsWith(".bat") && !libraries(n, libs)) {
+        final TokenBuilder tb = new TokenBuilder("set CP=%CP%");
+        for(final IOFile l : new IOFile("lib").children()) {
+          if(!l.name().contains("basex")) tb.add(";%LIB%/").add(l.name());
+        }
+        Util.errln(tb.toString());
+        fail(n + ": see STDERR output");
       }
     }
   }
\ No newline at end of file
