diff --git a/old_product.java b/new_product.java
index 9d23e74..376ea65 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-private void libraries(final String name, final HashSet<String> libs,
-      final StringList missing, final StringList obsolete) throws Exception {
+private boolean libraries(final String name, final HashSet<String> libs)
+      throws Exception {
 
     final HashSet<String> sl = new HashSet<String>();
     final NewlineInput nli = new NewlineInput(IO.get("etc/" + name));
@@ -13,11 +13,15 @@ private void libraries(final String name, final HashSet<String> libs,
       nli.close();
     }
 
+    final StringList mis = new StringList();
     for(final String l : libs) {
-      if(l.contains("basex")) continue;
-      if(!sl.remove(l)) missing.add(l);
+      if(!l.contains("basex") && !sl.remove(l)) mis.add(l);
     }
+    final StringList obs = new StringList();
     for(final String l : sl) {
-      if(l.endsWith(".jar")) obsolete.add(l);
+      if(l.endsWith(".jar")) obs.add(l);
     }
+    if(!mis.isEmpty()) Util.errln("Missing: " + Arrays.toString(mis.toArray()));
+    if(!obs.isEmpty()) Util.errln("Obsolete: " + Arrays.toString(obs.toArray()));
+    return mis.isEmpty() && obs.isEmpty();
   }
\ No newline at end of file
