diff --git a/old_product.java b/new_product.java
index 1ad8a02..f49dbf6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @Override
     public ChronoDateImpl<C> plus(long amountToAdd, PeriodUnit unit) {
-        if (unit instanceof LocalPeriodUnit) {
-            LocalPeriodUnit f = (LocalPeriodUnit) unit;
+        if (unit instanceof ChronoUnit) {
+            ChronoUnit f = (ChronoUnit) unit;
             switch (f) {
                 case DAYS: return plusDays(amountToAdd);
                 case WEEKS: return plusDays(Jdk8Methods.safeMultiply(amountToAdd, 7));
