diff --git a/old_test.java b/new_test.java
index 47b2177..3d1badb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,8 +4,8 @@
 		transform.set(1.2,0,0,1.2,0,0);
 		UtilAffine.convert(transform,a);
 
-		PixelTransform_F32 tranFrom = new PixelTransformAffine_F32(a);
-		PixelTransform_F32 tranTo = new PixelTransformAffine_F32(a.invert(null));
+		PixelTransform2_F32 tranFrom = new PixelTransformAffine_F32(a);
+		PixelTransform2_F32 tranTo = new PixelTransformAffine_F32(a.invert(null));
 
 		BinaryPolygonDetector alg = createDetector(GrayU8.class, true, 4,4);
 		alg.setLensDistortion(width, height, tranTo, tranFrom);
