diff --git a/old_product.java b/new_product.java
index bc506c5..d869f17 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public void setTransform( PixelTransform_F32 undistToDist ) {
+public void setTransform( PixelTransform2_F32 undistToDist ) {
 		if( undistToDist != null ) {
 			InterpolatePixelS<T> interpolate = FactoryInterpolation.bilinearPixelS(imageType, BorderType.EXTENDED);
 			integralImage = new GImageGrayDistorted<>(undistToDist, interpolate);
