diff --git a/old_test.java b/new_test.java
index b633ad2..3e70de9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,15 +1,15 @@
 @Test
-    public void testBoolsToHexDigit_2args() {
+    public void testBoolArrayToHexDigit_2args() {
         boolean[] shortArray = new boolean[]{false, true, true};
-        assertEquals('6', Conversion.boolsToHexDigit(shortArray, 0));
-        assertEquals('3', Conversion.boolsToHexDigit(shortArray, 1));
-        assertEquals('1', Conversion.boolsToHexDigit(shortArray, 2));
+        assertEquals('6', Conversion.boolArrayToHexDigit(shortArray, 0));
+        assertEquals('3', Conversion.boolArrayToHexDigit(shortArray, 1));
+        assertEquals('1', Conversion.boolArrayToHexDigit(shortArray, 2));
         boolean[] longArray = new boolean[]{true, false, true, false, false, true, true};
-        assertEquals('5', Conversion.boolsToHexDigit(longArray, 0));
-        assertEquals('2', Conversion.boolsToHexDigit(longArray, 1));
-        assertEquals('9', Conversion.boolsToHexDigit(longArray, 2));
-        assertEquals('c', Conversion.boolsToHexDigit(longArray, 3));
-        assertEquals('6', Conversion.boolsToHexDigit(longArray, 4));
-        assertEquals('3', Conversion.boolsToHexDigit(longArray, 5));
-        assertEquals('1', Conversion.boolsToHexDigit(longArray, 6));
+        assertEquals('5', Conversion.boolArrayToHexDigit(longArray, 0));
+        assertEquals('2', Conversion.boolArrayToHexDigit(longArray, 1));
+        assertEquals('9', Conversion.boolArrayToHexDigit(longArray, 2));
+        assertEquals('c', Conversion.boolArrayToHexDigit(longArray, 3));
+        assertEquals('6', Conversion.boolArrayToHexDigit(longArray, 4));
+        assertEquals('3', Conversion.boolArrayToHexDigit(longArray, 5));
+        assertEquals('1', Conversion.boolArrayToHexDigit(longArray, 6));
     }
\ No newline at end of file
