diff --git a/old_test.java b/new_test.java
index 7618e0c..ab25bcf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,7 @@
     // correct syntax
     get("declare %R:path('') %R:produces('text/plain') function m:f() {1};", "", "1");
     get("declare %R:path('') %R:produces('*/*') function m:f() {1};", "", "1");
+    get("declare %R:path('') %R:produces('text/plain;bla=blu') function m:f() {1};", "", "1");
     // multiple types
     get("declare %R:path('') %R:produces('text/plain','*/*') function m:f() {1};", "", "1");
     get("declare %R:path('') %R:produces('text/plain') %R:produces('*/*') " +
