diff --git a/old_product.java b/new_product.java
index 11df6fd..4a05fd1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,8 +3,8 @@ private boolean produces(final HTTPContext http) {
     if(produces.isEmpty()) return true;
     // check if any combination matches
     for(final HTTPAccept accept : http.accepts()) {
-      for(final String p : produces) {
-        if(MimeTypes.matches(p, accept.type)) return true;
+      for(final String produce : produces) {
+        if(MimeTypes.matches(produce, accept.type)) return true;
       }
     }
     return false;
