diff --git a/old_product.java b/new_product.java
index 94628d7..222bd15 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,13 @@
-public boolean remove(KType key)
-    {
-        final int mask = this.mask;
-        int slot = hashKey(key) & mask; 
-        while (allocated[slot])
-        {
-            if (Intrinsics.equalsKType(key, keys[slot]))
-             {
-                assigned--;
-                shiftConflictingKeys(slot);
-                return true;
-             }
-             slot = (slot + 1) & mask;
-        }
-        return false;
-    }
\ No newline at end of file
+public boolean remove(KType key) {
+    final int mask = this.mask;
+    int slot = hashKey(key) & mask;
+    while (allocated[slot]) {
+      if (Intrinsics.equalsKType(key, keys[slot])) {
+        shiftConflictingKeys(slot);
+        assigned--;
+        return true;
+      }
+      slot = (slot + 1) & mask;
+    }
+    return false;
+  }
\ No newline at end of file
