diff --git a/old_test.java b/new_test.java
index ccd8145..c4de871 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,8 +5,7 @@
 
         Map<String, Long> properties = toMap("id", 8L);
 
-        BeanUtil.populate(user, properties);
-        LOGGER.debug(JsonUtil.format(user));
+        LOGGER.debug(JsonUtil.format(BeanUtil.populate(user, properties)));
 
         //********************************************************
         user = new User();
