diff --git a/old_product.java b/new_product.java
index c5657a9..db029a8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,7 +10,9 @@
             @NotBlank(message = "No job id entered. Unable to set update time.")
             final String id
     ) throws GenieException {
-        LOG.debug("Updating db for job: " + id);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Updating db for job: " + id);
+        }
         final Job job = this.jobRepo.findOne(id);
         if (job == null) {
             throw new GenieNotFoundException(
