diff --git a/old_test.java b/new_test.java
index 111a5d1..8f582d7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void unmountTest() throws Exception {
-    TachyonURI path = new TachyonURI("/");
+    AlluxioURI path = new AlluxioURI("/");
     UnmountOptions unmountOptions = UnmountOptions.defaults();
     Mockito.doNothing().when(mFileSystemMasterClient).unmount(path);
     mFileSystem.unmount(path, unmountOptions);
