diff --git a/old_test.java b/new_test.java
index 9ded033..befaff0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,6 @@
 
         // More or less the same thing as the above test, so just make sure the
         // line number is applied.
-        List<PointSegment> results = instance.addCommand("G20 G0X1F150", 123);
+        List<GcodeMeta> results = instance.addCommand("G20 G0X1F150", 123);
         testCommand(results, 1, 150, 1., 0., 0., true, false, false, false, false, 123);
     }
\ No newline at end of file
