diff --git a/old_product.java b/new_product.java
index 3ca3344..d65a985 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,18 +3,16 @@
         List<PointSegment> results = new ArrayList<>();
         // Add command get meta doesn't update the state, so we need to do that
         // manually.
-        List<String> processedCommands = this.preprocessCommand(command);
-        for (String processedCommand : processedCommands) {
-            Collection<GcodeMeta> metaObjects = addCommandGetMeta(processedCommand, line, state);
-            if (metaObjects != null) {
-                for (GcodeMeta c : metaObjects) {
-                    if (c.point != null)
-                        results.add(c.point);
-                    if (c.points != null)
-                        results.addAll(c.points);
-                    if (c.state != null)
-                        this.state = c.state;
-                }
+        //List<String> processedCommands = this.preprocessCommand(command);
+        Collection<GcodeMeta> metaObjects = addCommandGetMeta(command, line, state);
+        if (metaObjects != null) {
+            for (GcodeMeta c : metaObjects) {
+                if (c.point != null)
+                    results.add(c.point);
+                if (c.points != null)
+                    results.addAll(c.points);
+                if (c.state != null)
+                    this.state = c.state;
             }
         }
 
