diff --git a/old_product.java b/new_product.java
index d044a10..1f87630 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,7 +12,6 @@
     ret.setPinned(isPinned());
     ret.setCacheable(false);
     ret.setPersisted(isPersisted());
-    ret.setBlockIds(null);
     ret.setLastModificationTimeMs(getLastModificationTimeMs());
     ret.setTtl(Constants.NO_TTL);
     ret.setUserName(getUserName());
