diff --git a/old_test.java b/new_test.java
index 92efc82..b25e560 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testListExtensions() throws Exception {
        for (String zoneId : zones) {
           ExtensionApi api = novaContext.getApi().getExtensionApiForZone(zoneId);
-          Set<? extends Extension> response = api.listExtensions();
+          Set<? extends Extension> response = api.list();
           assertNotNull(response);
           assertFalse(response.isEmpty());
            for (Extension extension : response) {
