diff --git a/old_test.java b/new_test.java
index efe9873..3f4ec42 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,6 +11,6 @@
 			Assert.assertTrue("Should not hit this line, since the previous needed to throw an exception", false);
 		}
 		catch (APIAuthenticationException ex) {
-			Assert.assertEquals("Can only be called from a Daemon thread", ex.getMessage());
+			Assert.assertEquals("Only daemon threads can spawn new daemon threads", ex.getMessage());
 		}
 	}
\ No newline at end of file
