diff --git a/old_product.java b/new_product.java
index 7dfaec5..9ed7406 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public static Thread runInNewDaemonThread(final Runnable runnable) {
 		// make sure we're already in a daemon thread
 		if (!isDaemonThread())
-			throw new APIAuthenticationException("Can only be called from a Daemon thread");
+			throw new APIAuthenticationException("Only daemon threads can spawn new daemon threads");
 		
 		// we should consider making DaemonThread public, so the caller can access returnedObject and exceptionThrown
 		DaemonThread thread = new DaemonThread() {
