diff --git a/old_product.java b/new_product.java
index fe17d70..d04806e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static String[] splitCmdLine(String input)
-            throws CloudServiceException {
+            throws GenieException {
         LOG.debug("Command line: " + input);
         if (input == null) {
             return new String[0];
@@ -13,7 +13,7 @@ public static String[] splitCmdLine(String input)
         } catch (Exception e) {
             String msg = "Invalid argument: " + input;
             LOG.error(msg, e);
-            throw new CloudServiceException(
+            throw new GenieException(
                     HttpURLConnection.HTTP_INTERNAL_ERROR, msg, e);
         }
 
