diff --git a/old_test.java b/new_test.java
index a86a225..ed9f0c1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testSplitCmdLine() throws CloudServiceException {
+    public void testSplitCmdLine() throws GenieException {
         final String input = "-f ch_survey_response_f.pig -p vhs_window_start_date=20120912 "
                 + "-p survey_date=20120919 -p now_epoch_ts=1348125009 "
                 + "-p rap_comment='/**/' -p no_rap_comment='--'";
