diff --git a/old_product.java b/new_product.java
index a10aa2e..9b07941 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 @Override
-  public void distribute(JSONObject message, long timestamp, MessageRoute route, Context context) {
+  public void distribute(MessageRoute route, Context context) {
     try {
       ProfileBuilder builder = getBuilder(route, context);
-      builder.apply(message, timestamp);
+      builder.apply(route.getMessage(), route.getTimestamp());
 
     } catch(ExecutionException e) {
       LOG.error("Unexpected error", e);
