diff --git a/old_test.java b/new_test.java
index 00d0ad1..deafb7b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
   public void addEdge_nodesNotInGraph() {
     graph.addNode(N1);
-    assertTrue(graph.addEdge(N1, N5));
-    assertTrue(graph.addEdge(N4, N1));
-    assertTrue(graph.addEdge(N2, N3));
+    assertTrue(graph.putEdge(N1, N5));
+    assertTrue(graph.putEdge(N4, N1));
+    assertTrue(graph.putEdge(N2, N3));
     assertThat(graph.nodes()).containsExactly(N1, N5, N4, N2, N3).inOrder();
     assertThat(graph.adjacentNodes(N1)).containsExactly(N4, N5);
     assertThat(graph.adjacentNodes(N2)).containsExactly(N3);
