diff --git a/old_product.java b/new_product.java
index e2a0b24..a0b76dd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -21,7 +21,10 @@
             this.sentBufferSize += commandString.length();
             
             // Newlines are embedded when they get queued so just send it.
-            this.sendStringToComm(commandString);
+        
+            // Command already has a newline attached.
+            this.sendMessageToConsoleListener(">>> " + commandString);
+            conn.sendStringToComm(commandString);
             
             GcodeCommand command = new GcodeCommand(commandString);
             command.setSent(true);
