diff --git a/old_test.java b/new_test.java
index 6472755..1934918 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,8 @@
 @Test
     public void testStreamCommands() {
         System.out.println("streamCommands");
-        GrblCommunicator instance = new GrblCommunicator(mg.in, mg.out, cb, asl);
+        MockConnection mc = new MockConnection(mg.in, mg.out);
+        GrblCommunicator instance = new GrblCommunicator(cb, asl, mc);
         String thirtyNineCharString = "thirty-nine character command here.....";
 
         boolean result;
@@ -47,7 +48,8 @@
         }
         
         // Tell the instance that we have made data available.
-        instance.serialEvent(null);
+        //instance.serialEvent(null);
+        instance.responseMessage(null);
 
         instance.streamCommands();
         // Make sure the queued command was sent.
@@ -56,7 +58,8 @@
 
         
         // Tell the instance that we have made data available.
-        instance.serialEvent(null);
+        //instance.serialEvent(null);
+        instance.responseMessage(null);
         
         // Wrap up.
         try {
@@ -68,7 +71,8 @@
         }
         
         // Tell the instance that we have made data available.
-        instance.serialEvent(null);
+        //instance.serialEvent(null);
+        instance.responseMessage(null);
 
         expResult = false;
         result = instance.areActiveCommands();
