diff --git a/old_product.java b/new_product.java
index 693fa04..6f64153 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,7 @@
 public Candidate onSupplyCandidateNewOrChange(final Candidate supplyCandidate)
 	{
+		Preconditions.checkArgument(supplyCandidate.getType() == Type.SUPPLY, "Given parameter 'supplyCandidate' has type=%s; supplyCandidate=%s", supplyCandidate.getType(), supplyCandidate);
+
 		// store the supply candidate and get both it's ID and qty-delta
 		final Candidate supplyCandidateDeltaWithId = candidateRepository.addOrReplace(supplyCandidate);
 
@@ -9,14 +11,20 @@ public Candidate onSupplyCandidateNewOrChange(final Candidate supplyCandidate)
 		}
 
 		// update the stock with the delta
-		final Candidate parentStockCandidateWithId = updateStock(supplyCandidateDeltaWithId);
+		final Candidate parentStockCandidateWithId = updateStock(supplyCandidateDeltaWithId
+				.withSeqNo(null) // don't provide the supply's SeqNo, because there might already be a stock record which we might override; plus, the supply's seqNo shall depend on the stock's anyways
+		);
 
 		// set the stock candidate as parent for the supply candidate
 		// the return value would have qty=0, but in the repository we updated the parent-ID
 		candidateRepository.addOrReplace(
-				supplyCandidate.withParentId(parentStockCandidateWithId.getId()));
+				supplyCandidate
+						.withParentId(parentStockCandidateWithId.getId())
+						.withSeqNo(parentStockCandidateWithId.getSeqNo() + 1));
 
-		return supplyCandidateDeltaWithId;
+		return supplyCandidateDeltaWithId
+				.withParentId(parentStockCandidateWithId.getId())
+				.withSeqNo(parentStockCandidateWithId.getSeqNo() + 1);
 
 		// e.g.
 		// supply-candidate with 23 (i.e. +23)
