diff --git a/old_product.java b/new_product.java
index 67a5a61..dac28cd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,8 +3,8 @@ private B64 encodeString(final QueryContext ctx) throws QueryException {
     final String enc = encoding(1, BIN_UE_X, ctx);
     if(str == null) return null;
     try {
-      return new B64(FNConvert.toBinary(str, enc));
+      return new B64(enc == null || enc == UTF8 ? str : FNConvert.toBinary(str, enc));
     } catch(final CharacterCodingException ex) {
-      throw BIN_EE.get(info);
+      throw BIN_CE.get(info, ex);
     }
   }
\ No newline at end of file
