diff --git a/old_test.java b/new_test.java
index b71d19d..65a6872 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test(dataProvider="dateProvider")
     public void test_matchesCalendrical(int startYear, int startMonth, int startDay, int endYear, int endMonth, int endDay) {
-        LocalDate date = LocalDate.date(startYear, startMonth, startDay);
-        long offset = LocalDate.date(endYear, endMonth, endDay).toModifiedJulianDays() - date.toModifiedJulianDays();
+        LocalDate date = LocalDate.of(startYear, startMonth, startDay);
+        long offset = LocalDate.of(endYear, endMonth, endDay).toModifiedJulianDays() - date.toModifiedJulianDays();
         int week = 0;
 
         for (long l = 0; l < offset; l++) {
