diff --git a/old_product.java b/new_product.java
index 3d5929c..094c3c2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @DataProvider(name="matchesCalendrical")
     Iterator<Object[]> matchesCalendrical() {
         return new Iterator<Object[]>() {
-            private LocalDate date = LocalDate.date(2008, 1, 1);
+            private LocalDate date = LocalDate.of(2008, 1, 1);
 
             public boolean hasNext() {
                 return date.getYear() != 2009;
