diff --git a/old_test.java b/new_test.java
index 205b638..12adc3d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,6 +8,6 @@
 		order.setEncounter(encounterService.getEncounter(6));
 		order.setStartDate(new Date());
 		expectedException.expect(APIException.class);
-		expectedException.expectMessage("No order type matches the concept class");
+		expectedException.expectMessage("Cannot determine the order type of the order");
 		orderService.saveOrder(order, null);
 	}
\ No newline at end of file
