diff --git a/old_product.java b/new_product.java
index 61780c2..4cdb145 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@ public void createDirectory(AlluxioURI path, CreateDirectoryOptions options)
     LOG.debug("createDirectory {} ", path);
     MasterContext.getMasterSource().incCreateDirectoriesOps(1);
     long flushCounter = AsyncJournalWriter.INVALID_FLUSH_COUNTER;
-    try (InodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.WRITE)) {
+    try (LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.WRITE)) {
       mPermissionChecker.checkParentPermission(FileSystemAction.WRITE, inodePath);
       mMountTable.checkUnderWritableMountPoint(path);
       flushCounter = createDirectoryAndJournal(inodePath, options);
