diff --git a/old_test.java b/new_test.java
index 835bc38..152f2bd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void createDirectoryTest() throws Exception {
     // create directory
-    mTree.createPath(TEST_URI, sDirectoryOptions);
+    createPath(mTree, TEST_URI, sDirectoryOptions);
     Assert.assertTrue(mTree.inodePathExists(TEST_URI));
     Inode<?> test = getInodeByPath(mTree, TEST_URI);
     Assert.assertEquals(TEST_PATH, test.getName());
@@ -11,7 +11,7 @@
     Assert.assertEquals((short) 0755, test.getPermission());
 
     // create nested directory
-    mTree.createPath(NESTED_URI, sNestedDirectoryOptions);
+    createPath(mTree, NESTED_URI, sNestedDirectoryOptions);
     Assert.assertTrue(mTree.inodePathExists(NESTED_URI));
     Inode<?> nested = getInodeByPath(mTree, NESTED_URI);
     Assert.assertEquals(TEST_PATH, nested.getName());
