diff --git a/old_test.java b/new_test.java
index 897a593..73889e3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,6 @@
     String ufsRoot = ServerConfiguration.get(PropertyKey.MASTER_MOUNT_TABLE_ROOT_UFS);
     UnderFileSystem ufs = UnderFileSystem.Factory.createForRoot(ServerConfiguration.global());
     ufs.create(ufsRoot + "/xyz").close();
-    mFileSystem.loadMetadata(new AlluxioURI("/xyz"));
     URIStatus status = mFileSystem.getStatus(new AlluxioURI("/xyz"));
     mLocalAlluxioCluster.stopFS();
     loadMetadataTestUtil(status);
