diff --git a/old_product.java b/new_product.java
index 17dc461..b719385 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,5 @@
-@Deprecated
-  @Override
-  public void loadMetadata(AlluxioURI path)
-      throws FileDoesNotExistException, IOException, AlluxioException {
-    loadMetadata(path, LoadMetadataPOptions.getDefaultInstance());
+@Override
+  public void mount(AlluxioURI alluxioPath, AlluxioURI ufsPath)
+      throws IOException, AlluxioException {
+    mount(alluxioPath, ufsPath, MountPOptions.getDefaultInstance());
   }
\ No newline at end of file
