diff --git a/old_test.java b/new_test.java
index 651a090..f4ab15f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	public void transformPixelToRectNorm_F64() {
-		CameraPinholeRadial param = new CameraPinholeRadial().
+		CameraPinholeBrown param = new CameraPinholeBrown().
 						fsetK(300, 320, 0, 150, 130, width, height).fsetRadial(0.1,1e-4);
 
 		DMatrixRMaj rect = new DMatrixRMaj(3,3,true,1.1,0,0,0,2,0,0.1,0,3);
