diff --git a/old_product.java b/new_product.java
index 79182af..53ebd6b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static Point2Transform2_F64 transformPixelToRectNorm(CameraPinholeRadial param,
+public static Point2Transform2_F64 transformPixelToRectNorm(CameraPinholeBrown param,
 																DMatrixRMaj rectify,
 																DMatrixRMaj rectifyK) {
 		return ImplRectifyImageOps_F64.transformPixelToRectNorm(param, rectify, rectifyK);
