diff --git a/old_test.java b/new_test.java
index 576a893..b951df8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @TestInfo(
-            status = TestStatus.LGTM,
-            notes = "The test verifies the start() method.",
+            level = TestLevel.PARTIAL_OK,
+            purpose = "Verifies the start() method.",
             targets = { @TestTarget(methodName = "start", 
                                     methodArgs = {})                         
             }
