diff --git a/old_test.java b/new_test.java
index fecba1a..b33410c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
     public void testGet() throws IOException {
-        try (ClosableMapSupplier<Integer, String> supplier = map5()) {
+        try (@NotNull ClosableMapSupplier<Integer, String> supplier = map5()) {
             final Map map = supplier.get();
             assertEquals("A", map.get(one));
-            try (ClosableMapSupplier empty = newStrStrMap()) {
+            try (@NotNull ClosableMapSupplier empty = newStrStrMap()) {
                 writeMessage("example of get(<key>) returning null, when the keys is not " +
                         "present in the map");
 
