diff --git a/old_test.java b/new_test.java
index db60e14..99feb38 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,11 +2,11 @@
     public void testToMap() {
         String[] values = {"key1:value1", "key2:value2", "key3:value3"};
         String separator = ":";
-        Map expResult = new HashMap<String, String>();
+        Map<String, String> expResult = new HashMap<String, String>();
         expResult.put("key1", "value1");
         expResult.put("key2", "value2");
         expResult.put("key3", "value3");
 
-        Map result = OsgiPropertyUtil.toMap(values, separator);
+        Map<String, String> result = OsgiPropertyUtil.toMap(values, separator);
         assertEquals(expResult, result);
     }
\ No newline at end of file
