diff --git a/old_product.java b/new_product.java
index 4b04740..231969f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-public static <T> Map<String, T> toMap(JSONObject json, Class<T> klass) throws JSONException {
+@SuppressWarnings("unchecked")
+    public static <T> Map<String, T> toMap(JSONObject json, Class<T> klass) throws JSONException {
         final HashMap<String, T> map = new HashMap<String, T>();
         final List<?> keys = IteratorUtils.toList(json.keys());
 
