diff --git a/old_test.java b/new_test.java
index d15237e..8d16605 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(expected = IllegalArgumentException.class)
-	public void testGetConversationData() {
-		Map<String, Object> params = new HashMap<String,Object>();
-		service.getConversationData(params);
-	}
\ No newline at end of file
+  public void testGetConversationData() {
+    final Map<String, Object> params = new HashMap<String, Object>();
+    service.getConversationData(params);
+  }
\ No newline at end of file
