diff --git a/old_product.java b/new_product.java
index bd0fe01..1a6d29f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,6 @@ public void validate(Object obj, Errors errors) {
 			// for the following elements Order.hbm.xml says: not-null="true"
 			ValidationUtils.rejectIfEmpty(errors, "asNeeded", "error.null");
 			ValidationUtils.rejectIfEmpty(errors, "dosingType", "error.null");
-			//ValidationUtils.rejectIfEmpty(errors, "drug", "error.null");
 			
 			if (order.getDrug() != null)
 				ValidationUtils.rejectIfEmpty(errors, "drug.concept", "error.null");
@@ -17,8 +16,22 @@ public void validate(Object obj, Errors errors) {
 				if (!(order.getDrug() == null) && !(order.getDrug().getConcept().equals(order.getConcept()))) {
 					errors.rejectValue("drug", "error.general");
 					errors.rejectValue("concept", "error.concept");
-					
 				}
 			}
+			if (order.getDosingType() != null) {
+				if (order.getDosingType().equals(DrugOrder.DosingType.SIMPLE)) {
+					ValidationUtils.rejectIfEmpty(errors, "dose", "error.doseIsNullForDosingTypeSimple");
+					ValidationUtils.rejectIfEmpty(errors, "doseUnits", "error.doseUnitsIsNullForDosingTypeSimple");
+					ValidationUtils.rejectIfEmpty(errors, "route", "error.routeIsNullForDosingTypeSimple");
+					ValidationUtils.rejectIfEmpty(errors, "frequency", "error.frequencyIsNullForDosingTypeSimple");
+				} else if (order.getDosingType().equals(DrugOrder.DosingType.FREE_TEXT)) {
+					ValidationUtils.rejectIfEmpty(errors, "instructions", "error.instructionIsNullForDosingTypeFreeText");
+				} else {
+					ValidationUtils.rejectIfEmpty(errors, "dosingInstructions",
+					    "error.dosingInstructionsIsNullForDosingTypeOther");
+				}
+			}
+			validateFieldsForOutpatientCareSettingType(order, errors);
+			validatePairedFields(order, errors);
 		}
 	}
\ No newline at end of file
