diff --git a/old_product.java b/new_product.java
index 658d20d..35d3951 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,6 +4,8 @@
 	                               ModelMap model,
 	                               @RequestParam(required=false, value="action") String action,
 	                               @RequestParam(required=false, value="userFormPassword") String password,
+	                               @RequestParam(required=false, value="secretQuestion") String secretQuestion,
+	                               @RequestParam(required=false, value="secretAnswer") String secretAnswer,
 	                               @RequestParam(required=false, value="confirm") String confirm,
 	                               @RequestParam(required=false, value="forcePassword") Boolean forcePassword,
 	                               @RequestParam(required=false, value="roleStrings") String[] roles,
@@ -118,19 +120,22 @@
 				return showForm(user.getUserId(), createNewPerson, user, model);
 			}
 			
-			if (isNewUser(user))
+			if (isNewUser(user)){
 				us.saveUser(user, password);
-			else {
+            } else {
 				us.saveUser(user, null);
-				
+                
 				if (!password.equals("") && Context.hasPrivilege(OpenmrsConstants.PRIV_EDIT_USER_PASSWORDS)) {
 					if (log.isDebugEnabled())
 						log.debug("calling changePassword for user " + user + " by user " + Context.getAuthenticatedUser());
 					us.changePassword(user, password);
 				}
-				
 			}
-
+            
+            if (StringUtils.hasLength(secretQuestion) && StringUtils.hasLength(secretAnswer)) {
+            	us.changeQuestionAnswer(user, secretQuestion, secretAnswer);
+            }
+            
 			httpSession.setAttribute(WebConstants.OPENMRS_MSG_ATTR, "User.saved");
 		}
 		return "redirect:user.list";
