diff --git a/old_test.java b/new_test.java
index 5f3e363..074f801 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void handleAllServersAtOnceTest() throws MalformedObjectNameException {
-        JmxReadRequest request = new JmxRequestBuilder(READ, testBeanName).
+        JolokiaReadRequest request = new JmxRequestBuilder(READ, testBeanName).
                 attribute("attr").
                 build();
         assertFalse(handler.handleAllServersAtOnce(request));
