diff --git a/old_test.java b/new_test.java
index b67ec78..9273db6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,19 +1,26 @@
 @Test
     public void testPutBucketAccessMode() {
-    	Response response;
-        try {
-        	response = bucketManager.putBucketAccessMode(TestConfig.testBucket_z0, AclType.PRIVATE);
-        	Assert.assertEquals(200, response.statusCode);
-            BucketInfo info = bucketManager.getBucketInfo(TestConfig.testBucket_z0);
-            Assert.assertEquals(1, info.getPrivate());
+    	String[] buckets = new String[]{TestConfig.testBucket_z0, TestConfig.testBucket_na0};
+    	for (String bucket : buckets) {
+        	Response response;
+            try {
+            	// 测试转私有空间
+            	response = bucketManager.putBucketAccessMode(bucket, AclType.PRIVATE);
+            	Assert.assertEquals(200, response.statusCode);
+                BucketInfo info = bucketManager.getBucketInfo(bucket);
+                Assert.assertEquals(1, info.getPrivate());
 
-            response = bucketManager.putBucketAccessMode(TestConfig.testBucket_z0, AclType.PUBLIC);
-            Assert.assertEquals(200, response.statusCode);
-            info = bucketManager.getBucketInfo(TestConfig.testBucket_z0);
-            Assert.assertEquals(0, info.getPrivate());
-        } catch (QiniuException e) {
-        	Assert.fail(e.response.toString());
-        }
+                // 测试转公有空间
+                response = bucketManager.putBucketAccessMode(bucket, AclType.PUBLIC);
+                Assert.assertEquals(200, response.statusCode);
+                info = bucketManager.getBucketInfo(bucket);
+                Assert.assertEquals(0, info.getPrivate());
+                
+            } catch (QiniuException e) {
+            	Assert.fail(e.response.toString());
+            }
+    	}
+    	// 测试空间不存在情况
         try {
         	bucketManager.putBucketAccessMode(TestConfig.dummyBucket, AclType.PRIVATE);
         } catch (QiniuException e) {
