diff --git a/old_product.java b/new_product.java
index 8c641f3..51a2e62 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,8 +5,8 @@ public static void printRootCauseStackTrace(final Throwable throwable, final Pri
         if (writer == null) {
             throw new IllegalArgumentException("The PrintWriter must not be null");
         }
-        String trace[] = getRootCauseStackTrace(throwable);
-        for (String element : trace) {
+        final String trace[] = getRootCauseStackTrace(throwable);
+        for (final String element : trace) {
             writer.println(element);
         }
         writer.flush();
