diff --git a/old_product.java b/new_product.java
index 88c57a8..0c23266 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,13 @@
 private Item optimize(final QueryContext ctx) throws QueryException {
     checkWrite(ctx);
 
-    final boolean all = expr.length == 2 ? checkBln(expr[1], ctx) : false;
-    ctx.resource.data(checkStr(expr[0], ctx), input);
-    try {
-      (all ? new OptimizeAll() : new Optimize()).execute(ctx.context);
-    } catch(final BaseXException e) {
-      Util.errln(e);
-    }
+    // [DP] I've rewritten this a little; yet, the optimize code will
+    //   probably have to be moved to static methods to avoid that
+    //   the command is executed on the global data instance.
+
+    final boolean all = expr.length == 2 && checkBln(expr[1], ctx);
+    data(0, ctx);
+    final Command cmd = all ? new OptimizeAll() : new Optimize();
+    if(!cmd.run(ctx.context)) DBERR.thrw(input, cmd.info());
     return null;
   }
\ No newline at end of file
