diff --git a/old_test.java b/new_test.java
index 9260d10..8660ef2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
 	public void testGetStructureInfo() {
 		Structure structure = new StructureImpl();
 		Chain chain = new ChainImpl();
-		chain.setChainID("A");
+		chain.setId("A");
 		Map<String,Integer> testMap = new HashMap<>();
 		testMap.put("A", 0);
 		List<Chain> chainList = new ArrayList<>();
