diff --git a/old_product.java b/new_product.java
index 8935a91..767eaad 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,7 +13,7 @@ public static MmtfSummaryDataBean getStructureInfo(Structure structure) {
 			List<Chain> chains = structure.getModel(i);
 			allChains.addAll(chains);
 			for (Chain chain : chains) {
-				String idOne = chain.getChainID();
+				String idOne = chain.getId();
 				if (!chainIdToIndexMap.containsKey(idOne)) { 
 					chainIdToIndexMap.put(idOne, chainCounter);
 				}
