diff --git a/old_product.java b/new_product.java
index b944f6c..7e07ffc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,14 @@
 @Override
     public String readLine() throws IOException {
         synchronized (lock) {
-            lineNumber++;
-            return super.readLine();
+            if (lastWasCR) {
+                chompNewline();
+                lastWasCR = false;
+            }
+            String result = super.readLine();
+            if (result != null) {
+                lineNumber++;
+            }
+            return result;
         }
     }
\ No newline at end of file
