diff --git a/old_product.java b/new_product.java
index 9a9b216..f18a257 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public OffsetDateTime plus(Duration duration) {
-        LocalDateTime newDT = dateTime.plus(duration);
+        LocalDateTime newDT = dateTime.plusSeconds(duration.getSeconds()).plusNanos(duration.getNano());
         return (newDT == dateTime ? this : new OffsetDateTime(newDT, offset));
     }
\ No newline at end of file
