diff --git a/old_test.java b/new_test.java
index 7656954..33290d6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
-    public void test_plus_Duration() {
-        Duration dur = Duration.ofSeconds(62, 3);
-        LocalTime t = TEST_12_30_40_987654321.plus(dur);
+    public void test_plus_adjuster() {
+        Period p = Period.ofTime(0, 0, 62, 3);
+        LocalTime t = TEST_12_30_40_987654321.plus(p);
         assertEquals(t, LocalTime.of(12, 31, 42, 987654324));
     }
\ No newline at end of file
