diff --git a/old_test.java b/new_test.java
index 4f19586..f34e25d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,29 +12,17 @@
         assertEquals(expResult, result);
 
         ids = new HashSet<>();
-        ids.add(new Identifier("maven", "somegroup:something:1.0", ""));
-        expResult = "\"\"";
-        result = instance.csvCpeConfidence(ids);
-        assertEquals(expResult, result);
-
-        ids = new HashSet<>();
-        Identifier i1 = new Identifier("cpe", "cpe:/a:somegroup:something:1.0", "");
-        i1.setConfidence(Confidence.HIGH);
+        GenericIdentifier i1 = new GenericIdentifier("cpe:/a:somegroup:something:1.0", Confidence.HIGH);
         ids.add(i1);
         expResult = "HIGH";
         result = instance.csvCpeConfidence(ids);
         assertEquals(expResult, result);
 
         ids = new HashSet<>();
-        i1 = new Identifier("cpe", "cpe:/a:somegroup:something:1.0", "");
-        i1.setConfidence(Confidence.HIGH);
+        i1 = new GenericIdentifier("cpe:/a:somegroup:something:1.0", Confidence.HIGH);
         ids.add(i1);
-        Identifier i2 = new Identifier("cpe", "cpe:/a:somegroup:something2:1.0", "");
-        i2.setConfidence(Confidence.MEDIUM);
+        GenericIdentifier i2 = new GenericIdentifier("cpe:/a:somegroup:something2:1.0", Confidence.MEDIUM);
         ids.add(i2);
-        Identifier i3 = new Identifier("maven", "somegroup:something:1.0", "");
-        i3.setConfidence(Confidence.LOW);
-        ids.add(i3);
 
         expResult = "\"HIGH, MEDIUM\"";
         String expResult2 = "\"MEDIUM, HIGH\"";
