diff --git a/old_product.java b/new_product.java
index 8961faa..f971915 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,14 +5,12 @@ public String csvCpeConfidence(Set<Identifier> ids) {
         boolean addComma = false;
         final StringBuilder sb = new StringBuilder();
         for (Identifier id : ids) {
-            if ("cpe".equals(id.getType())) {
-                if (addComma) {
-                    sb.append(", ");
-                } else {
-                    addComma = true;
-                }
-                sb.append(id.getConfidence());
+            if (addComma) {
+                sb.append(", ");
+            } else {
+                addComma = true;
             }
+            sb.append(id.getConfidence());
         }
         if (sb.length() == 0) {
             return "\"\"";
