diff --git a/old_product.java b/new_product.java
index 84e1b50..97c6507 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static DateTimeAdjuster dayOfWeekInMonth(int ordinal, DayOfWeek dayOfWeek) {
+public static WithAdjuster dayOfWeekInMonth(int ordinal, DayOfWeek dayOfWeek) {
         if (dayOfWeek == null) {
             throw new NullPointerException("DayOfWeek must not be null");
         }
