diff --git a/old_test.java b/new_test.java
index 7906cfa..5c83c5a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(expectedExceptions=NullPointerException.class)
     public void test_print_nullAppendable() throws Exception {
-        Calendrical calendrical = Calendrical.calendrical(RULE_DOW, 3);
+        Calendrical calendrical = new Calendrical(RULE_DOW, 3);
         TextPrinterParser pp = new TextPrinterParser(RULE_DOW, TextStyle.FULL);
         pp.print(calendrical, (Appendable) null, symbols);
     }
\ No newline at end of file
