diff --git a/old_product.java b/new_product.java
index e2603a8..7bc7a93 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public void print(Calendrical calendrical, Appendable appendable, DateTimeFormatSymbols symbols) throws IOException {
-        int value = calendrical.getValue(fieldRule);
+        int value = calendrical.getValueInt(fieldRule);
         BigDecimal fraction = fieldRule.convertValueToFraction(value);
         if (fraction.scale() == 0) {  // scale is zero if value is zero
             if (minWidth > 0) {
