diff --git a/old_product.java b/new_product.java
index bf5e0e0..c2c863f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,21 +1,23 @@
 static protected Capabilities getGrblStatusCapabilities(final double version, final Character letter) {
         Capabilities ret = new Capabilities();
+        ret.addCapability(CapabilitiesConstants.JOGGING);
 
         // Check if real time commands are enabled.
         if (version==0.8 && (letter != null) && (letter >= 'c')) {
-            ret.REAL_TIME = true;
+            ret.addCapability(GrblCapabilitiesConstants.REAL_TIME);
         } else if (version >= 0.9) {
-            ret.REAL_TIME = true;
+            ret.addCapability(GrblCapabilitiesConstants.REAL_TIME);
         }
 
         // Check for V1.x features
         if (version >= 1.1) {
-            ret.REAL_TIME = true;
+            ret.addCapability(GrblCapabilitiesConstants.REAL_TIME);
 
             // GRBL 1.1
-            ret.V1_FORMAT = true;
-            ret.OVERRIDES = true;
-            ret.JOG_MODE = true;
+            ret.addCapability(GrblCapabilitiesConstants.V1_FORMAT);
+            ret.addCapability(CapabilitiesConstants.OVERRIDES);
+            ret.addCapability(GrblCapabilitiesConstants.HARDWARE_JOGGING);
+            ret.addCapability(CapabilitiesConstants.CONTINUOUS_JOGGING);
         }
 
         return ret;
