diff --git a/old_test.java b/new_test.java
index 64d8ede..4ef26a1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,31 +3,41 @@
         System.out.println("getGrblStatusCapabilities");
         double version;
         Character letter;
-        Capabilities expResult = new Capabilities();
         Capabilities result;
 
         version = 0.8;
         letter = 'c';
         result = GrblUtils.getGrblStatusCapabilities(version, letter);
-        assertEquals(true, result.REAL_TIME);
-        assertEquals(false, result.OVERRIDES);
-        
+        assertTrue(result.hasCapability(GrblCapabilitiesConstants.REAL_TIME));
+        assertFalse(result.hasCapability(GrblCapabilitiesConstants.V1_FORMAT));
+        assertFalse(result.hasCapability(GrblCapabilitiesConstants.HARDWARE_JOGGING));
+        assertFalse(result.hasOverrides());
+        assertFalse(result.hasContinuousJogging());
+
         version = 0.8;
         letter = 'a';
         result = GrblUtils.getGrblStatusCapabilities(version, letter);
-        assertEquals(false, result.REAL_TIME);
-        assertEquals(false, result.OVERRIDES);
-        
+        assertFalse(result.hasCapability(GrblCapabilitiesConstants.REAL_TIME));
+        assertFalse(result.hasCapability(GrblCapabilitiesConstants.V1_FORMAT));
+        assertFalse(result.hasCapability(GrblCapabilitiesConstants.HARDWARE_JOGGING));
+        assertFalse(result.hasOverrides());
+        assertFalse(result.hasContinuousJogging());
+
         version = 0.9;
         letter = null;
         result = GrblUtils.getGrblStatusCapabilities(version, letter);
-        assertEquals(true, result.REAL_TIME);
-        assertEquals(false, result.OVERRIDES);
+        assertTrue(result.hasCapability(GrblCapabilitiesConstants.REAL_TIME));
+        assertFalse(result.hasCapability(GrblCapabilitiesConstants.V1_FORMAT));
+        assertFalse(result.hasCapability(GrblCapabilitiesConstants.HARDWARE_JOGGING));
+        assertFalse(result.hasOverrides());
+        assertFalse(result.hasContinuousJogging());
 
         version = 1.1;
         letter = null;
         result = GrblUtils.getGrblStatusCapabilities(version, letter);
-        assertEquals(true, result.REAL_TIME);
-        assertEquals(true, result.OVERRIDES);
-        assertEquals(true, result.V1_FORMAT);
+        assertTrue(result.hasCapability(GrblCapabilitiesConstants.REAL_TIME));
+        assertTrue(result.hasCapability(GrblCapabilitiesConstants.V1_FORMAT));
+        assertTrue(result.hasCapability(GrblCapabilitiesConstants.HARDWARE_JOGGING));
+        assertTrue(result.hasOverrides());
+        assertTrue(result.hasContinuousJogging());
     }
\ No newline at end of file
