diff --git a/old_product.java b/new_product.java
index 0d64b08..1aac6b8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-protected ExceptionCollection scanArtifacts(MavenProject project, MavenEngine engine) {
+protected ExceptionCollection scanArtifacts(MavenProject project, Engine engine) {
         // <editor-fold defaultstate="collapsed" desc="old implementation">
         /*
             for (Artifact a : project.getArtifacts()) {
