diff --git a/old_test.java b/new_test.java
index 151a18e..45f9129 100644
--- a/old_test.java
+++ b/new_test.java
@@ -26,7 +26,7 @@
 
             boolean autoUpdate = Settings.getBoolean(Settings.KEYS.AUTO_UPDATE);
             Settings.setBoolean(Settings.KEYS.AUTO_UPDATE, false);
-            MavenEngine engine = new MavenEngine(null, null);
+            Engine engine = new Engine();
             Settings.setBoolean(Settings.KEYS.AUTO_UPDATE, autoUpdate);
 
             assertTrue(engine.getDependencies().isEmpty());
