diff --git a/old_test.java b/new_test.java
index b2e2158..aa1a4a2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,5 +18,5 @@
     // updates an existing entry
     error(_ARCHIVE_CREATE.args("X", "X",
         "<archive:options><archive:format value='gzip'/></archive:options>") + " ! " +
-        _ARCHIVE_UPDATE.args(" .", "X", "Y"), Err.ARCH_MODIFY);
+        _ARCHIVE_UPDATE.args(" .", "X", "Y"), Err.ARCH_MODIFY_X);
   }
\ No newline at end of file
