diff --git a/old_product.java b/new_product.java
index 99a9a14..41d6e68 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,12 +2,19 @@ private void object() throws QueryIOException {
     consumeWs('{', true);
     conv.openObject();
     if(!consumeWs('}', false)) {
+      final TokenSet set = new TokenSet();
       do {
-        conv.openPair(spec != JsonSpec.LIBERAL || curr() == '"' ? string() : unquoted());
+        final byte[] key = !liberal || curr() == '"' ? string() : unquoted();
+        final boolean dupl = set.contains(key);
+        if(dupl && duplicates == JsonDuplicates.REJECT)
+          throw error(BXJS_DUPLICATE_X, "Key '%' occurs more than once.", key);
+
+        conv.openPair(key);
         consumeWs(':', true);
         value();
-        conv.closePair();
-      } while(consumeWs(',', false) && !(spec == JsonSpec.LIBERAL && curr() == '}'));
+        conv.closePair(!dupl || duplicates == JsonDuplicates.USE_LAST);
+        set.put(key);
+      } while(consumeWs(',', false) && !(liberal && curr() == '}'));
       consumeWs('}', true);
     }
     conv.closeObject();
