diff --git a/old_product.java b/new_product.java
index 7f039a4..a86bf60 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,15 +12,18 @@ public void complete() throws SQLException
             // commit any changes made as part of the transaction
             if (isValid() && !isReadOnly())
             {
-                commit();
+                dispatchEvents();
             }
         }
         finally
         {
-            // Free the DB connection
-            // If connection is closed or null, this is a no-op
-            DatabaseManager.freeConnection(connection);
-            connection = null;
-            clearCache();
+            if(dbConnection != null)
+            {
+                //Commit our changes
+                dbConnection.commit();
+                // Free the DB connection
+                dbConnection.closeDBConnection();
+                dbConnection = null;
+            }
         }
     }
\ No newline at end of file
