diff --git a/old_test.java b/new_test.java
index 666f8fc..3862b70 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
         
         // By default, we should have a new DB connection, so let's make sure it is there
         assertThat("testComplete 0", instance.getDBConnection(), notNullValue());
-        assertThat("testComplete 1", instance.getDBConnection().isClosed(), equalTo(false));
+        assertThat("testComplete 1", instance.getDBConnection().isSessionAlive(), equalTo(true));
         assertThat("testComplete 2", instance.isValid(), equalTo(true));
         
         // Now, call complete(). This should set DB connection to null & invalidate context
