diff --git a/old_test.java b/new_test.java
index 672f2ce..b53edad 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,9 +5,9 @@
 				new CameraPinholeRadial().fsetK(300, 320, 0, 150, 130, width, height).fsetRadial(0.1,1e-4);
 
 		// do nothing rectification
-		DenseMatrix32F rect1 = CommonOps_D32.identity(3);
-		DenseMatrix32F rect2 = CommonOps_D32.identity(3);
-		DenseMatrix32F rectK = PerspectiveOps.calibrationMatrix(param, (DenseMatrix32F)null);
+		RowMatrix_F32 rect1 = CommonOps_D32.identity(3);
+		RowMatrix_F32 rect2 = CommonOps_D32.identity(3);
+		RowMatrix_F32 rectK = PerspectiveOps.calibrationMatrix(param, (RowMatrix_F32)null);
 
 		RectifyImageOps.fullViewLeft(param,rect1,rect2,rectK);
 
