diff --git a/old_product.java b/new_product.java
index c34efa0..d723b65 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-private static Str dirName(final File path) {
-    final String pth = path.getParent();
-    return Str.get(pth == null ? "." : pth);
+private Str dirName(final QueryContext ctx) throws QueryException {
+    String path = checkFile(0, ctx).getParent();
+    return Str.get(dir(path == null ? "." : path));
   }
\ No newline at end of file
