diff --git a/old_test.java b/new_test.java
index 6b1bdc2..f7ff2d9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
     assertEquals(norm(PATH),
         norm(query(_FILE_DIR_NAME.args(PATH1))).toLowerCase(Locale.ENGLISH));
     // check with an empty path
-    query(_FILE_DIR_NAME.args(""), ".");
+    query(_FILE_DIR_NAME.args(""), "." + File.separator);
     // check with a path without directory separators
-    query(_FILE_DIR_NAME.args(NAME), ".");
+    query(_FILE_DIR_NAME.args(NAME), "." + File.separator);
   }
\ No newline at end of file
