diff --git a/old_test.java b/new_test.java
index 2277e9e..cf56bd3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void getBlockLocation() {
     BlockStoreLocation expectedLocation =
-        new BlockStoreLocation(mTier.getTierAlias(), mDir.getDirIndex());
+        new BlockStoreLocation(mTier.getTierAlias(), mDir.getDirIndex(), mDir.getDirMedium());
     Assert.assertEquals(expectedLocation, mBlockMeta.getBlockLocation());
   }
\ No newline at end of file
