diff --git a/old_product.java b/new_product.java
index 4a3f1f6..f66906c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public BlockStoreLocation getBlockLocation() {
     StorageTier tier = mDir.getParentTier();
-    return new BlockStoreLocation(tier.getTierAlias(), mDir.getDirIndex());
+    return new BlockStoreLocation(tier.getTierAlias(), mDir.getDirIndex(), mDir.getDirMedium());
   }
\ No newline at end of file
