diff --git a/old_product.java b/new_product.java
index 5c98fc8..6b88a9b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,7 @@
 			
 			Concept concept = orderFrequency.getConcept();
 			if (concept != null) {
-				if (!OrderFrequency.CONCEPT_CLASS_UUID.equals(concept.getConceptClass().getUuid())) {
+				if (!ConceptClass.FREQUENCY_UUID.equals(concept.getConceptClass().getUuid())) {
 					errors.rejectValue("concept", "OrderFrequency.concept.shouldBeClassFrequency");
 				}
 				
