diff --git a/old_test.java b/new_test.java
index f1853b7..e031ba7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
 	public void validate_shouldFailValidationIfClassOfQuantityUnitsdoseUnitsOrDurationUnitsIsNotUnitsOfMeasureConceptClass()
 	        throws Exception {
 		Concept concept = Context.getConceptService().getConcept(3);
-		Assert.assertFalse(concept.getConceptClass().getUuid().equals(ConceptClass.UNITS_OF_MEASURE_UUID));
+		Assert.assertFalse(concept.getConceptClass().getUuid().equals(ConceptClass.UNIT_OF_MEASUREMENT_UUID));
 		
 		DrugOrder order = new DrugOrder();
 		order.setDosingType(DrugOrder.DosingType.FREE_TEXT);
