diff --git a/old_product.java b/new_product.java
index 7edecb4..8210376 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,10 @@
-public int get(int x, int y) {
+@Override
+	public int get(int x, int y) {
 		if (!isInBounds(x, y))
 			throw new ImageAccessException("Requested pixel is out of bounds");
 
-		return data[getIndex(x, y)];
+		if( signed )
+			return data[getIndex(x, y)];
+		else
+			return data[getIndex(x, y)] & 0xFFFF;
 	}
\ No newline at end of file
