diff --git a/old_test.java b/new_test.java
index eb5e63c..a84c1e4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,6 +14,10 @@
 		Number found = (Number) call(img, "get", 0, null, 1, 1);
 		if (GeneralizedImageOps.isFloatingPoint(img))
 			assertEquals(expected.doubleValue(), found.doubleValue(), 1e-4);
-		else
-			assertTrue(expected.intValue() == found.intValue());
+		else {
+			if( ((ImageInteger)img).isSigned() )
+				assertTrue(expected.intValue() == found.intValue());
+			else
+				assertTrue((expected.intValue() & 0xFFFF) == found.intValue());
+		}
 	}
\ No newline at end of file
