diff --git a/old_product.java b/new_product.java
index 58c509f..61c70a9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -51,6 +51,19 @@
           String.format("Invalid desired state %s. Should be either CURRENT or INSTALLED",
                   newStateStr));
       }
+
+      // Get a host name to populate the hostrolecommand table's hostEntity.
+      ClusterEntity cluster = clusterDAO.findByName(clName);
+      String defaultHostName = null;
+      List<HostEntity> hosts = new ArrayList(cluster.getHostEntities());
+      if (hosts != null && !hosts.isEmpty()) {
+        Collections.sort(hosts);
+        defaultHostName = hosts.get(0).getHostName();
+      }
+      if (defaultHostName == null) {
+        throw new AmbariException("Could not find at least one host to set the command for");
+      }
+
       args.put(FinalizeUpgradeAction.VERSION_KEY, desiredRepoVersion);
       args.put(FinalizeUpgradeAction.CLUSTER_NAME_KEY, clName);
 
@@ -58,7 +71,8 @@
       command.setCommandParams(args);
       command.setClusterName(clName);
       finalizeUpgradeAction.setExecutionCommand(command);
-      HostRoleCommand hostRoleCommand = new HostRoleCommand("none",
+      
+      HostRoleCommand hostRoleCommand = hostRoleCommandFactory.create(defaultHostName,
               Role.AMBARI_SERVER_ACTION, null, null);
       finalizeUpgradeAction.setHostRoleCommand(hostRoleCommand);
 
