diff --git a/old_product.java b/new_product.java
index b31210c..87c78ac 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public QueueConfig setName(String name) {
         this.name = name;
-        if (backingMapName == null) {
-            backingMapName = "q:" + name;
+        if (backingMapRef == null) {
+            backingMapRef = "q:" + name;
         }
         return this;
     }
\ No newline at end of file
