diff --git a/old_test.java b/new_test.java
index fb7a37b..e2875e0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
         String name = "a test name";
         QueueConfig queueConfig = new QueueConfig().setName(name);
         assertEquals(name, queueConfig.getName());
-        assertEquals("q:" + name, queueConfig.getBackingMapName());
-        queueConfig.setBackingMapName("backingMap");
-        assertEquals("backingMap", queueConfig.getBackingMapName());
+        assertEquals("q:" + name, queueConfig.getBackingMapRef());
+        queueConfig.setBackingMapRef("backingMap");
+        assertEquals("backingMap", queueConfig.getBackingMapRef());
     }
\ No newline at end of file
