diff --git a/old_product.java b/new_product.java
index 43d07de..cb36606 100644
--- a/old_product.java
+++ b/new_product.java
@@ -59,9 +59,11 @@ boolean parse() throws QueryException {
         }
       } else if(eq(uri, QueryText.OUTPUTURI)) {
         // serialization parameters
-        final Option opt = output.option(string(local));
-        if(opt == null) error(info, UNKNOWN_SER, local);
-        output.set(opt, toString(value, name));
+        try {
+          output.assign(string(local), toString(value, name));
+        } catch(final BaseXException ex) {
+          error(info, UNKNOWN_SER, local);
+        }
       }
       found |= rexq;
     }
