diff --git a/old_test.java b/new_test.java
index d918f42..b4c8db6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
   public void testClearRecursively() {
     FinishedTriggersProperties.verifyClearRecursively(
-        FinishedTriggersSet.fromSet(new HashSet<ExecutableTrigger<?>>()));
+        FinishedTriggersSet.fromSet(new HashSet<ExecutableTrigger>()));
   }
\ No newline at end of file
