diff --git a/old_product.java b/new_product.java
index bb14a03..612909e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,12 @@
-public static <T> void filter(Iterable<T> collection, Predicate<? super T> predicate) {
+public static <T> boolean filter(Iterable<T> collection, Predicate<? super T> predicate) {
+        boolean result = false;
         if (collection != null && predicate != null) {
             for (Iterator<T> it = collection.iterator(); it.hasNext();) {
                 if (!predicate.evaluate(it.next())) {
                     it.remove();
+                    result = true;
                 }
             }
         }
+        return result;
     }
\ No newline at end of file
