diff --git a/old_test.java b/new_test.java
index c19d47a..2d05aec 100644
--- a/old_test.java
+++ b/new_test.java
@@ -23,13 +23,14 @@
 
     Date start = new Date();
 
-    CreateSynonymOptions createOptions =
-            new CreateSynonymOptions.Builder(workspaceId, entity, entityValue, synonym).build();
+    CreateSynonymOptions createOptions = new CreateSynonymOptions.Builder(workspaceId, entity, entityValue, synonym)
+        .build();
     service.createSynonym(createOptions).execute();
 
     try {
-      GetSynonymOptions getOptions =
-              new GetSynonymOptions.Builder(workspaceId, entity, entityValue, synonym).build();
+      GetSynonymOptions getOptions = new GetSynonymOptions.Builder(workspaceId, entity, entityValue, synonym)
+          .includeAudit(true)
+          .build();
       Synonym response = service.getSynonym(getOptions).execute();
 
       assertNotNull(response);
@@ -46,8 +47,8 @@
     } catch (Exception ex) {
       fail(ex.getMessage());
     } finally {
-      DeleteSynonymOptions deleteOptions =
-              new DeleteSynonymOptions.Builder(workspaceId, entity, entityValue, synonym).build();
+      DeleteSynonymOptions deleteOptions = new DeleteSynonymOptions.Builder(workspaceId, entity, entityValue, synonym)
+          .build();
       service.deleteSynonym(deleteOptions).execute();
     }
   }
\ No newline at end of file
