diff --git a/old_product.java b/new_product.java
index 46d0bae..4e44c22 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,5 +4,8 @@ public ServiceCall<Synonym> getSynonym(GetSynonymOptions getSynonymOptions) {
         getSynonymOptions.workspaceId(), getSynonymOptions.entity(), getSynonymOptions.value(), getSynonymOptions
             .synonym()));
     builder.query(VERSION, versionDate);
+    if (getSynonymOptions.includeAudit() != null) {
+      builder.query("include_audit", String.valueOf(getSynonymOptions.includeAudit()));
+    }
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(Synonym.class));
   }
\ No newline at end of file
