diff --git a/old_test.java b/new_test.java
index c273593..bc1c974 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
         int operationCountPerType = 5;
 
         // Add to MTL + Add to ReadIndex (append; beginMerge).
-        MemoryOperationLog opLog = new MemoryOperationLog();
+        SequencedItemList<Operation> opLog = new SequencedItemList<>();
         ArrayList<TestReadIndex.MethodInvocation> methodInvocations = new ArrayList<>();
         TestReadIndex readIndex = new TestReadIndex(methodInvocations::add);
         @Cleanup
@@ -13,10 +13,9 @@
         ArrayList<Operation> operations = populate(updater, segmentCount, operationCountPerType);
 
         // Verify they were properly processed.
-        Assert.assertEquals("Unexpected size for MemoryOperationLog.", operations.size(), opLog.size());
         Assert.assertEquals("Unexpected number of items added to ReadIndex.", operations.size() - segmentCount * operationCountPerType, methodInvocations.size());
 
-        Iterator<Operation> logIterator = opLog.read(op -> true, opLog.size());
+        Iterator<Operation> logIterator = opLog.read(-1, operations.size());
         int currentIndex = -1;
         int currentReadIndex = -1;
         while (logIterator.hasNext()) {
