diff --git a/old_product.java b/new_product.java
index 256c0d1..c720448 100644
--- a/old_product.java
+++ b/new_product.java
@@ -17,8 +17,7 @@ void process(Operation operation) throws DataCorruptionException {
             }
         }
 
-        long seqNo = operation.getSequenceNumber();
-        boolean added = this.inMemoryOperationLog.addIf(operation, previous -> previous.getSequenceNumber() < seqNo);
+        boolean added = this.inMemoryOperationLog.add(operation);
         if (!added) {
             // This is a pretty nasty one. It's safer to shut down the container than continue.
             // We either recorded the Operation correctly, but invoked this callback out of order, or we really
