diff --git a/old_product.java b/new_product.java
index 7628ef5..2c79aa3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 protected AmqpSession createSession(String sessionName, Session session, SessionHandler handler) {
         return new ReactorSession(session, handler, sessionName, reactorProvider, handlerProvider, getCBSNode(),
-            tokenManagerProvider, connectionOptions.getRetry().getTryTimeout());
+            tokenManagerProvider, messageSerializer, connectionOptions.getRetry().getTryTimeout());
     }
\ No newline at end of file
