diff --git a/old_product.java b/new_product.java
index 3fd8f63..905bcf7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static <T extends ImageBase>
-	void features(T ii, int c_x, int c_y,
+	void features(T ii, double c_x, double c_y,
 				  double theta, Kernel2D_F64 weight,
 				  int widthLargeGrid, int widthSubRegion, int widthSample, double scale,
 				  boolean useHaar, boolean inBounds,
@@ -8,6 +8,5 @@ public static <T extends ImageBase>
 		SparseImageGradient<T,?> gradient = createGradient(inBounds,useHaar,widthSample,scale,(Class<T>)ii.getClass());
 		gradient.setImage(ii);
 
-		int regionSize = widthLargeGrid*widthSubRegion;
-		ImplSurfDescribeOps.features(c_x, c_y, theta, weight, regionSize, widthSubRegion, scale, gradient, features);
+		ImplSurfDescribeOps.features(c_x, c_y, theta, weight, widthLargeGrid, widthSubRegion, scale, gradient, features);
 	}
\ No newline at end of file
