diff --git a/old_test.java b/new_test.java
index a8e7b20..4c31a04 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,16 +1,15 @@
 @Test
 	public void features() {
-		fail("reimplement");
-//		double scale = 1.5;
-//		double []found = new double[ 64 ];
-//		double []expected = new double[ 64 ];
-//
-//		Kernel2D_F64 weight = FactoryKernelGaussian.gaussian(2,true,64,-1,10);
-//		SurfDescribeOps.features(inputF32,c_x,c_y,1.2,weight,20,5,scale,true,found);
-//
-//		SparseImageGradient<ImageFloat32,?> gradient = SurfDescribeOps.createGradient(true,false,2,scale,ImageFloat32.class);
-//		gradient.setImage(inputF32);
-//		ImplSurfDescribeOps.features(c_x,c_y,1.2,weight,20,5,scale,gradient,expected);
-//
-//		BoofTesting.assertEquals(expected,found,1e-4);
+		double scale = 1.5;
+		double []found = new double[ 64 ];
+		double []expected = new double[ 64 ];
+
+		Kernel2D_F64 weight = FactoryKernelGaussian.gaussian(2, true, 64, -1, 10);
+		SurfDescribeOps.features(inputF32,c_x,c_y,1.2,weight,4,5,2,scale,false,true,found);
+
+		SparseImageGradient<ImageFloat32,?> gradient = SurfDescribeOps.createGradient(true,false,2,scale,ImageFloat32.class);
+		gradient.setImage(inputF32);
+		ImplSurfDescribeOps.features(c_x,c_y,1.2,weight,4,5,scale,gradient,expected);
+
+		BoofTesting.assertEquals(expected,found,1e-4);
 	}
\ No newline at end of file
