diff --git a/old_product.java b/new_product.java
index fb10882..e63fa61 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,11 +7,18 @@
 		if (roles == null)
 			roles = new Vector<Role>();
 		
+		// add the requested roles and all child roles for consideration
+		Set<Role> allRoles = new HashSet<Role>();
+		for (Role r : roles) {
+			allRoles.add(r);
+			allRoles.addAll(r.getAllChildRoles());
+		}
+		
 		// if the authenticated role is in the list of searched roles, then all
 		// persons should be searched
 		Role auth_role = getRole(RoleConstants.AUTHENTICATED);
 		if (roles.contains(auth_role))
 			return dao.getUsers(name, new Vector<Role>(), includeRetired, start, length);
 		
-		return dao.getUsers(name, roles, includeRetired, start, length);
+		return dao.getUsers(name, new ArrayList<Role>(allRoles), includeRetired, start, length);
 	}
\ No newline at end of file
