diff --git a/old_test.java b/new_test.java
index 21a939b..6287081 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
 		
 		List<User> users = Context.getUserService().getUsers("Johnson", null, false);
 		Assert.assertEquals(3, users.size());
-		Assert.assertTrue(users.contains(new User(2)));
-		Assert.assertTrue(users.contains(new User(4)));
-		Assert.assertTrue(users.contains(new User(5)));
+		Assert.assertTrue(containsId(users, 2));
+		Assert.assertTrue(containsId(users, 4));
+		Assert.assertTrue(containsId(users, 5));
 	}
\ No newline at end of file
