diff --git a/old_product.java b/new_product.java
index 382d028..d881580 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,13 +14,13 @@
                 plaintext = decryptCipher.decrypt(getKeySet().getReadKey(localConEndType), decryptIv, Arrays
                         .copyOfRange(decryptionRequest.getCipherText(), decryptCipher.getBlocksize(),
                                 decryptionRequest.getCipherText().length));
-                usedIv = decryptCipher.getIv();
+                usedIv = decryptIv;
             } else {
                 byte[] decryptIv = getDecryptionIV();
                 LOGGER.debug("decryptionIV: " + ArrayConverter.bytesToHexString(decryptIv));
                 plaintext = decryptCipher.decrypt(getKeySet().getReadKey(localConEndType), decryptIv,
                         decryptionRequest.getCipherText());
-                usedIv = decryptCipher.getIv();
+                usedIv = decryptIv;
                 // Set next IV
             }
 
