diff --git a/old_test.java b/new_test.java
index 1cbbac6..97bf542 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testDecrypt() throws NoSuchAlgorithmException, CryptoException {
-        context.setSelectedProtocolVersion(ProtocolVersion.TLS13);
+        context.setSelectedProtocolVersion(ProtocolVersion.TLS13_DRAFT21);
         context.setSelectedCipherSuite(CipherSuite.TLS_AES_128_GCM_SHA256);
         context.setClientHandshakeTrafficSecret(ArrayConverter
                 .hexStringToByteArray("4B63051EABCD514D7CB6D1899F472B9F56856B01BDBC5B733FBB47269E7EBDC2"));
@@ -9,6 +9,7 @@
         context.setConnection(new OutboundConnection());
         record.setProtocolMessageBytes(ArrayConverter
                 .hexStringToByteArray("1BB3293A919E0D66F145AE830488E8D89BE5EC16688229"));
+        record.setLength(record.getProtocolMessageBytes().getValue().length);
         context.setActiveClientKeySetType(Tls13KeySetType.HANDSHAKE_TRAFFIC_SECRETS);
         recordCipher = new RecordAEADCipher(context, KeySetGenerator.generateKeySet(context));
         decryptor = new RecordDecryptor(recordCipher, context);
