diff --git a/old_test.java b/new_test.java
index 83f7038..a2abb20 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,6 @@
     query("[] cast as xs:integer?", "");
     query("xs:integer([1])", 1);
 
-    error("[] cast as xs:integer", INVCAST_X_X_X);
-    error("[1,2] cast as xs:integer", INVCAST_X_X_X);
+    error("[] cast as xs:integer", INVTYPE_X_X_X);
+    error("[1,2] cast as xs:integer", INVTYPE_X_X_X);
   }
\ No newline at end of file
