diff --git a/old_product.java b/new_product.java
index c643c3e..15aaf32 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public Value cast(final Value value, final QueryContext qc, final StaticContext
       final InputInfo ii) throws QueryException {
 
     final long vs = value.size();
-    if(!occ.check(vs)) throw INVCAST_X_X_X.get(ii, value.seqType(), this, value);
+    if(!occ.check(vs)) throw INVTYPE_X_X_X.get(ii, value.seqType(), this, value);
 
     if(value.isEmpty()) return Empty.SEQ;
     if(value instanceof Item) return cast((Item) value, qc, sc, ii, true);
