diff --git a/old_test.java b/new_test.java
index c956d83..b30bec0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test(description = "DELETE /vApp/{id}")
-   public void testDeleteVApp() {
-      // Create a temporary VApp to delete
+   public void testRemoveVApp() {
+      // Create a temporary VApp to remove
       VApp temp = instantiateVApp();
 
       // The method under test
-      Task deleteVApp = vAppApi.deleteVApp(temp.getHref());
-      assertTrue(retryTaskSuccess.apply(deleteVApp), String.format(TASK_COMPLETE_TIMELY, "deleteVApp"));
+      Task removeVApp = vAppApi.removeVApp(temp.getHref());
+      assertTrue(retryTaskSuccess.apply(removeVApp), String.format(TASK_COMPLETE_TIMELY, "removeVApp"));
 
-      VApp deleted = vAppApi.getVApp(temp.getHref());
-      assertNull(deleted, "The VApp "+temp.getName()+" should have been deleted");
+      VApp removed = vAppApi.getVApp(temp.getHref());
+      assertNull(removed, "The VApp " + temp.getName() + " should have been removed");
    }
\ No newline at end of file
