diff --git a/old_test.java b/new_test.java
index 015bc63..aa90f45 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void setAttributeTest() throws Exception {
-    TachyonURI file = new TachyonURI("/file");
+    AlluxioURI file = new AlluxioURI("/file");
     SetAttributeOptions setAttributeOptions = SetAttributeOptions.defaults();
     mFileSystem.setAttribute(file, setAttributeOptions);
     Mockito.verify(mFileSystemMasterClient).setAttribute(file, setAttributeOptions);
