diff --git a/old_product.java b/new_product.java
index 6660c1f..988379d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
-void setAttribute(TachyonURI path, SetAttributeOptions options)
-      throws FileDoesNotExistException, IOException, TachyonException;
\ No newline at end of file
+void setAttribute(AlluxioURI path, SetAttributeOptions options)
+      throws FileDoesNotExistException, IOException, AlluxioException;
\ No newline at end of file
