diff --git a/old_product.java b/new_product.java
index c0a8e00..e2e1001 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,6 +16,19 @@ public void runAsync() {
             while (canContinue) {
                 try {
                     this.runnable.run();
+                    this.consecutiveFailedAttempts.set(0);
+                } catch (Throwable ex) {
+                    int c = this.consecutiveFailedAttempts.incrementAndGet();
+                    if (ExceptionHelpers.mustRethrow(ex)) {
+                        close();
+                    }
+
+                    boolean retry = this.errorHandler.apply(ex, c);
+                    if (retry) {
+                        synchronized (this) {
+                            this.runAgain = true;
+                        }
+                    }
                 } finally {
                     // Determine if we need to run the task again. Otherwise release our spot.
                     synchronized (this) {
