diff --git a/old_test.java b/new_test.java
index 5733adf..e879bf9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
     {
         System.out.println("getSettingsPanel");
         BytesThroughputOverTimeGui instance = new BytesThroughputOverTimeGui();
-        JSettingsPanel result = instance.getSettingsPanel();
+        JSettingsPanel result = instance.createSettingsPanel();
         assertNotNull(result);
     }
\ No newline at end of file
