diff --git a/old_test.java b/new_test.java
index f4ef63d..d747360 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
-    public void testRegisterListeners() {
-        DynamicLoadManager dm = new DynamicLoadManager(ldServiceMock);
-        assertNotNull(dm);
-        dm.registerListeners();
-        
-        verify(pNodeMock, atLeast(1)).addListener((OperationsNodeListener)dm);
-        //verify(pNodeMock, times(1)).addListener((BootstrapNodeListener)dm);
-    }
\ No newline at end of file
+  public void testRegisterListeners() {
+    DynamicLoadManager dm = new DynamicLoadManager(ldServiceMock);
+    assertNotNull(dm);
+    dm.registerListeners();
+
+    verify(pNodeMock, atLeast(1)).addListener((OperationsNodeListener) dm);
+    //verify(pNodeMock, times(1)).addListener((BootstrapNodeListener)dm);
+  }
\ No newline at end of file
