diff --git a/old_product.java b/new_product.java
index 4713d03..0560023 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public void registerListeners() {
-        LOG.trace("DynamicLoadManager register listeners...");
-        ControlNode pm = getLoadDistributionService().getZkService().getControlZKNode();
-        pm.addListener((OperationsNodeListener) this);
-        pm.addListener((BootstrapNodeListener) this);
-    }
\ No newline at end of file
+    LOG.trace("DynamicLoadManager register listeners...");
+    ControlNode pm = getLoadDistributionService().getZkService().getControlZkNode();
+    pm.addListener((OperationsNodeListener) this);
+    pm.addListener((BootstrapNodeListener) this);
+  }
\ No newline at end of file
