diff --git a/old_test.java b/new_test.java
index 33dd57e..17c0b0a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,8 @@
   public void testRename() throws QueryException, BaseXException {
     final String fun = check(Function.RENAME);
 
-    // add documents with certain prefix
     new Add("etc/test/dir", "docs", "test").execute(CONTEXT);
-    query(fun + "('db', 'test', 'newtest')", "");
+
+    query(fun + "('db/test', 'newtest')", "");
     query("count(collection('db/newtest')) gt 0", "true");
   }
\ No newline at end of file
