diff --git a/old_product.java b/new_product.java
index a9b45ca..fa18b14 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,12 +6,12 @@
         }
 
         return connectionMono.map(connection -> sessionMap.computeIfAbsent(sessionName, key -> {
-            final SessionHandler handler =
-                handlerProvider.createSessionHandler(connectionId, getHost(), sessionName, connectionOptions.timeout());
+            final SessionHandler handler = handlerProvider.createSessionHandler(connectionId, getHost(), sessionName,
+                connectionOptions.retry().tryTimeout());
             final Session session = connection.session();
 
             BaseHandler.setHandler(session, handler);
-            return new ReactorSession(session, handler, sessionName, reactorProvider, handlerProvider,
-                this.getCBSNode(), tokenResourceProvider, connectionOptions.timeout());
+            return new ReactorSession(session, handler, sessionName, reactorProvider, handlerProvider, getCBSNode(),
+                tokenResourceProvider, connectionOptions.retry().tryTimeout());
         }));
     }
\ No newline at end of file
