diff --git a/old_test.java b/new_test.java
index 656d1e7..f1997fa 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testSizeEq() {
-        assertTrue(instance.sizeEq("id", 3).getQueryCriterions()
-                .contains(new SizeEqCriterion("id", 3)));
+    	assertEquals(new SizeEqCriterion("id", 3), instance.sizeEq("id", 3).getQueryCriterion());
     }
\ No newline at end of file
