diff --git a/old_product.java b/new_product.java
index 2f2c380..c7e9787 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
 public DocumentProcessingStatus getDocumentProcessingState(final Document document) {
-		String documentId = IDHelper.getDocumentId(document);
-		return executeRequest(API_VERSION + documentId + PROCESSING_STATE_PATH, null, DocumentProcessingStatus.class);
-	}
\ No newline at end of file
+    final String documentId = IDHelper.getDocumentId(document);
+    return executeRequest(API_VERSION + documentId + PROCESSING_STATE_PATH, null,
+        DocumentProcessingStatus.class);
+  }
\ No newline at end of file
