diff --git a/old_test.java b/new_test.java
index 5a08619..1fcf784 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,6 @@
 @Test
-	public void testGetDocumentProcessingState() {
-		DocumentProcessingStatus documentProcessingState = service.getDocumentProcessingState(EXAMPLE_DOCUMENT);
-		Assert.assertNotNull(documentProcessingState);
-	}
\ No newline at end of file
+  public void testGetDocumentProcessingState() {
+    final DocumentProcessingStatus documentProcessingState =
+        service.getDocumentProcessingState(EXAMPLE_DOCUMENT);
+    Assert.assertNotNull(documentProcessingState);
+  }
\ No newline at end of file
