diff --git a/old_test.java b/new_test.java
index 7922d05..c36cdf8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
     public void test_with_adjustment() {
-        final OffsetDate sample = OffsetDate.of(2012, 3, 4, OFFSET_PONE);
+        final OffsetDate sample = OffsetDate.of(LocalDate.of(2012, 3, 4), OFFSET_PONE);
         TemporalAdjuster adjuster = new TemporalAdjuster() {
             @Override
             public Temporal adjustInto(Temporal dateTime) {
