diff --git a/old_test.java b/new_test.java
index 353d532..89d09e2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,18 @@
 @Test
 	public void fullViewLeft_calibrated() {
 
-		fullViewLeft_calibrated(false);
-		fullViewLeft_calibrated(true);
+		IntrinsicParameters param =
+				new IntrinsicParameters().fsetK(300, 320, 0, 150, 130, width, height).fsetRadial(0.1,1e-4);
+
+		// do nothing rectification
+		DenseMatrix64F rect1 = CommonOps.identity(3);
+		DenseMatrix64F rect2 = CommonOps.identity(3);
+		DenseMatrix64F rectK = PerspectiveOps.calibrationMatrix(param, null);
+
+		RectifyImageOps.fullViewLeft(param,rect1,rect2,rectK);
+
+		// check left image
+		PointTransform_F32 tran = RectifyImageOps.transformPixelToRect_F32(param, rect1);
+		checkInside(tran);
+		// the right view is not checked since it is not part of the contract
 	}
\ No newline at end of file
