diff --git a/old_product.java b/new_product.java
index d507831..700610b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,6 @@ public static void fullViewLeft(IntrinsicParameters paramLeft,
 	{
 		// need to take in account the order in which image distort will remove rectification later on
 		paramLeft = new IntrinsicParameters(paramLeft);
-		paramLeft.flipY = false;
 
 		PointTransform_F32 tranLeft = transformPixelToRect_F32(paramLeft, rectifyLeft);
 
