diff --git a/old_test.java b/new_test.java
index c860581..2f47186 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test public void arrayTest() throws QueryException {
-    parse("[ ]", Spec.RFC4627);
-    parse("[]", "[ ]", Spec.RFC4627);
-    parse("[[[[[[42], {}]]]]]", "[ [ [ [ [ [ 42 ], { } ] ] ] ] ]", Spec.RFC4627);
-    parse("[ 1, 2, 3, 4, 5, 6, 7, 8 ]", Spec.RFC4627);
-    parse("[1,2,3,]", "[ 1, 2, 3 ]", Spec.LIBERAL);
+    parse("[ ]", JsonSpec.RFC4627);
+    parse("[]", "[ ]", JsonSpec.RFC4627);
+    parse("[[[[[[42], {}]]]]]", "[ [ [ [ [ [ 42 ], { } ] ] ] ] ]", JsonSpec.RFC4627);
+    parse("[ 1, 2, 3, 4, 5, 6, 7, 8 ]", JsonSpec.RFC4627);
+    parse("[1,2,3,]", "[ 1, 2, 3 ]", JsonSpec.LIBERAL);
 
-    error("[1,2,3,]", Spec.RFC4627);
-    error("[,42]", Spec.RFC4627);
-    error("[1, ", Spec.RFC4627);
+    error("[1,2,3,]", JsonSpec.RFC4627);
+    error("[,42]", JsonSpec.RFC4627);
+    error("[1, ", JsonSpec.RFC4627);
   }
\ No newline at end of file
