diff --git a/old_product.java b/new_product.java
index 65a2c4a..0e0fb8c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ private void array(final JsonHandler h) throws QueryException {
         h.openItem();
         value(h);
         h.closeItem();
-      } while(consumeWs(',', false) && !(spec == Spec.LIBERAL && curr() == ']'));
+      } while(consumeWs(',', false) && !(spec == JsonSpec.LIBERAL && curr() == ']'));
       consumeWs(']', true);
     }
     h.closeArray();
