diff --git a/old_test.java b/new_test.java
index 6adb2f8..eada7a1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testStringToClusterCriterias() throws GenieException {
+    public void testStringToClusterCriterias() throws GeniePreconditionException {
         final List<ClusterCriteria> criterias =
                 this.job.stringToClusterCriterias(EXPECTED_CLUSTER_CRITERIAS_STRING);
         Assert.assertEquals(CLUSTER_CRITERIAS.size(), criterias.size());
