diff --git a/old_product.java b/new_product.java
index 5b9f905..c6fa4ba 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,9 @@
-protected List<ClusterCriteria> stringToClusterCriterias(final String criteriaString) throws GenieException {
+protected List<ClusterCriteria> stringToClusterCriterias(final String criteriaString)
+            throws GeniePreconditionException {
         //Rebuild the cluster criteria objects
         final String[] criteriaSets = StringUtils.split(criteriaString, CRITERIA_SET_DELIMITER);
         if (criteriaSets == null || criteriaSets.length == 0) {
-            throw new GenieException(
-                    HttpURLConnection.HTTP_PRECON_FAILED,
-                    "No cluster criteria found. Unable to continue.");
+            throw new GeniePreconditionException("No cluster criteria found. Unable to continue.");
         }
         final List<ClusterCriteria> cc = new ArrayList<>();
         for (final String criteriaSet : criteriaSets) {
@@ -17,9 +16,7 @@ protected List<ClusterCriteria> stringToClusterCriterias(final String criteriaSt
             cc.add(new ClusterCriteria(c));
         }
         if (cc.isEmpty()) {
-            throw new GenieException(
-                    HttpURLConnection.HTTP_PRECON_FAILED,
-                    "No Cluster Criteria found. Unable to continue");
+            throw new GeniePreconditionException("No Cluster Criteria found. Unable to continue");
         }
         return cc;
     }
\ No newline at end of file
