diff --git a/old_test.java b/new_test.java
index f17a9fc..3a5cab2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,23 +1,23 @@
 @Test
     public void get() throws Exception {
-        GUID guid = new GUID();
-        SimpleMarshalledValue<GUID> mv = this.factory.createMarshalledValue(guid);
+        UUID uuid = UUID.randomUUID();
+        SimpleMarshalledValue<UUID> mv = this.factory.createMarshalledValue(uuid);
 
         assertNotNull(mv.peek());
-        assertSame(guid, mv.peek());
-        assertSame(guid, mv.get(this.context));
+        assertSame(uuid, mv.peek());
+        assertSame(uuid, mv.get(this.context));
 
-        SimpleMarshalledValue<GUID> copy = replicate(mv);
+        SimpleMarshalledValue<UUID> copy = replicate(mv);
 
         assertNull(copy.peek());
         
-        GUID guid2 = copy.get(this.context);
-        assertNotSame(guid, guid2);
-        assertEquals(guid, guid2);
+        UUID uuid2 = copy.get(this.context);
+        assertNotSame(uuid, uuid2);
+        assertEquals(uuid, uuid2);
 
         copy = replicate(copy);
-        guid2 = copy.get(this.context);
-        assertEquals(guid, guid2);
+        uuid2 = copy.get(this.context);
+        assertEquals(uuid, uuid2);
 
         mv = this.factory.createMarshalledValue(null);
         assertNull(mv.peek());
