diff --git a/old_test.java b/new_test.java
index b38211e..a5429b1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
     persistResource(Registry.get("lol").asBuilder().setTldType(Registry.TldType.TEST).build());
     Registrar registrar = persistResource(makeRegistrar(
         "evilregistrar", "Yes Virginia", ACTIVE));
-    persistResource(makeDomainResource(
+    persistResource(makeDomainBase(
         "cat.lol",
         persistResource(makeContactResource("5372808-ERL", "Goblin Market", "lol@cat.lol")),
         persistResource(makeContactResource("5372808-IRL", "Santa Claus", "BOFH@cat.lol")),
