diff --git a/old_product.java b/new_product.java
index c54f27a..890ab50 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@
     extensionManager.validate();
     validateClientIsLoggedIn(clientId);
     DateTime now = ofy().getTransactionTime();
-    DomainResource existingDomain = loadAndVerifyExistence(DomainResource.class, targetId, now);
+    DomainBase existingDomain = loadAndVerifyExistence(DomainBase.class, targetId, now);
     verifyOptionalAuthInfo(authInfo, existingDomain);
     verifyHasPendingTransfer(existingDomain);
     verifyResourceOwnership(clientId, existingDomain);
@@ -74,9 +74,9 @@
         .setParent(historyEntry)
         .build();
     // Construct the post-transfer domain.
-    DomainResource partiallyApprovedDomain =
+    DomainBase partiallyApprovedDomain =
         approvePendingTransfer(existingDomain, TransferStatus.CLIENT_APPROVED, now);
-    DomainResource newDomain =
+    DomainBase newDomain =
         partiallyApprovedDomain
             .asBuilder()
             // Update the transferredRegistrationExpirationTime here since approvePendingTransfer()
