diff --git a/old_test.java b/new_test.java
index 4c72084..9dc955b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void cellTest(){
         Object dest = Json.fromJson(Streams.fileInr("org/nutz/json/person.txt"));
-        assertEquals("dtri", MapListCell.cell(dest, "company.name"));
-        assertEquals("Dao", MapListCell.cell(dest, "company.creator.name"));
+        assertEquals("dtri", MapList.cell(dest, "company.name"));
+        assertEquals("Dao", MapList.cell(dest, "company.creator.name"));
     }
\ No newline at end of file
