diff --git a/old_test.java b/new_test.java
index fafa5d2..35d5bf7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,10 +2,10 @@
 	@Test
 	public void testGetTestList() {
 		ModelMap model = new ModelMap();
-		controller.getTestList(getTestUser(), model, false, null);
-		Page<PerfTest> testPage = (Page<PerfTest>)model.get("testListPage");
+		controller.getTestList(getTestUser(), null, false, null, model);
+		Page<PerfTest> testPage = (Page<PerfTest>) model.get("testListPage");
 		List<PerfTest> testList = testPage.getContent();
 
 		assertThat(testList.size(), is(1));
-		
+
 	}
\ No newline at end of file
