diff --git a/old_product.java b/new_product.java
index 8aaba4f..237e3f7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static <T> boolean addIgnoreNull(Collection<T> collection, T object) {
+public static <T> boolean addIgnoreNull(final Collection<T> collection, final T object) {
         if (collection == null) {
             throw new NullPointerException("The collection must not be null");
         }
