diff --git a/old_product.java b/new_product.java
index 72b7266..95504bf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,5 +11,5 @@
         action = new ContextualFunction<>(contextDescriptor, action);
 
         CompletableFuture<T> dependentStage = completableFuture.exceptionally(action);
-        return new ManagedCompletableFuture<T>(dependentStage, defaultExecutor);
+        return new ManagedCompletableFuture<T>(dependentStage, defaultExecutor, null);
     }
\ No newline at end of file
