diff --git a/old_product.java b/new_product.java
index 5b91a10..1ba9a31 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,16 +7,18 @@
 			User userEntity = BeanMapper.map(user, User.class);
 			BeanValidators.validateWithException(validator, userEntity);
 
-			accountDao.saveUser(userEntity);
+			accountService.saveUser(userEntity);
 
 			return new IdResponse(userEntity.getId());
 		} catch (ConstraintViolationException e) {
 			String message = StringUtils.join(BeanValidators.extractPropertyAndMessageAsList(e, " "), "\n");
 			return handleParameterError(response, e, message);
-		} catch (DataIntegrityViolationException e) {
-			String message = "新建用户参数存在唯一性冲突(用户:" + user + ")";
-			return handleParameterError(response, e, message);
 		} catch (RuntimeException e) {
-			return handleGeneralError(response, e);
+			if (Exceptions.isCausedBy(e, org.hibernate.exception.ConstraintViolationException.class)) {
+				String message = "新建用户参数存在唯一性冲突(用户:" + user + ")";
+				return handleParameterError(response, e, message);
+			} else {
+				return handleGeneralError(response, e);
+			}
 		}
 	}
\ No newline at end of file
