diff --git a/old_test.java b/new_test.java
index aa9fe00..3aa1a74 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,10 @@
 @Test
 	public void createUser() {
 		User user = UserData.randomUser();
+		UserDTO userDTO = BeanMapper.map(user, UserDTO.class);
 
-		UserDTO userDTO = new UserDTO();
-		userDTO.setLoginName(user.getLoginName());
-		userDTO.setName(user.getName());
-		userDTO.setEmail(user.getEmail());
-
-		IdResponse result = accountWebServiceClient.createUser(userDTO);
-		assertNotNull(result.getId());
+		IdResponse response = accountWebServiceClient.createUser(userDTO);
+		assertNotNull(response.getId());
+		GetUserResponse response2 = accountWebServiceClient.getUser(response.getId());
+		assertEquals(user.getLoginName(), response2.getUser().getLoginName());
 	}
\ No newline at end of file
