diff --git a/old_product.java b/new_product.java
index ef9ff46..7f48629 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static LocalDate from(TemporalAccessor temporal) {
-        LocalDate date = temporal.query(Queries.localDate());
+        LocalDate date = temporal.query(TemporalQuery.localDate());
         if (date == null) {
             throw new DateTimeException("Unable to obtain LocalDate from TemporalAccessor: " + temporal.getClass());
         }
