diff --git a/old_product.java b/new_product.java
index a8bb8e1..ff8a316 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
-public int addAll(Iterable<? extends KTypeCursor<? extends KType>> iterable)
-    {
-        int count = 0;
-        for (KTypeCursor<? extends KType> cursor : iterable)
-        {
-            if (add(cursor.value)) count++;
-        }
-        return count;
-    }
\ No newline at end of file
+public int addAll(Iterable<? extends KTypeCursor<? extends KType>> iterable) {
+    int count = 0;
+    for (KTypeCursor<? extends KType> cursor : iterable) {
+      if (add(cursor.value)) {
+        count++;
+      }
+    }
+    return count;
+  }
\ No newline at end of file
