diff --git a/old_product.java b/new_product.java
index f02fc88..a625ef0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,5 +8,5 @@ public static Locale toLocale(Object locale){
         if (locale instanceof String){
             return LocaleUtils.toLocale((String) locale);
         }
-        throw new UnsupportedOperationException("locale2 not support!");
+        throw new UnsupportedOperationException("input param [locale] type is:[" + locale.getClass().getName() + "] not support!");
     }
\ No newline at end of file
