diff --git a/old_product.java b/new_product.java
index 07deb2c..813916b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,3 @@
 void add(final Data d) {
-    if(size == data.length) {
-      data = Arrays.copyOf(data, size << 1);
-      pins = Arrays.copyOf(pins, size << 1);
-    }
-    data[size] = d;
-    pins[size++] = 1;
+    list.add(new PData(d));
   }
\ No newline at end of file
