diff --git a/old_test.java b/new_test.java
index 744ac83..b9c830a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,8 @@
         CacheBasedDataset<Integer, String, Long, AutoCloseable> dataset = builder.build(
             TestUtils.testEnvBuilder(),
             (env, upstream, upstreamSize) -> upstreamSize,
-            (env, upstream, upstreamSize, ctx) -> null
+            (env, upstream, upstreamSize, ctx) -> null,
+            TestUtils.testEnvBuilder().buildForTrainer()
         );
 
         Affinity<Integer> upstreamAffinity = ignite.affinity(upstreamCache.getName());
