diff --git a/old_product.java b/new_product.java
index cd690f8..fe38245 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
 public <C extends Serializable, D extends AutoCloseable> Dataset<C, D> build(
         LearningEnvironmentBuilder envBuilder,
         PartitionContextBuilder<K, V, C> partCtxBuilder,
-        PartitionDataBuilder<K, V, C, D> partDataBuilder);
\ No newline at end of file
+        PartitionDataBuilder<K, V, C, D> partDataBuilder,
+        LearningEnvironment localLearningEnv);
\ No newline at end of file
