diff --git a/old_test.java b/new_test.java
index 122bbcc..2e85fe3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,6 +16,8 @@
 		expect(mockProperties.getProperty(CONFIG_CONNECTION_DRIVER_CLASS)).andReturn(DRIVER).anyTimes();
 		expect(mockProperties.getProperty(CONFIG_CONNECTION_HOOK_CLASS)).andReturn("com.jolbox.bonecp.provider.CustomHook").anyTimes();
 		expect(mockProperties.getProperty(CONFIG_INIT_SQL)).andReturn(TEST_QUERY).anyTimes();
+		expect(mockProperties.getProperty(CONFIG_LOG_STATEMENTS_ENABLED)).andReturn("true").anyTimes();
+
 
 
 		BoneCPConnectionProvider partialTestClass = createNiceMock(BoneCPConnectionProvider.class, 
@@ -41,6 +43,7 @@
 		assertEquals(USERNAME, config.getUsername());
 		assertEquals(PASSWORD, config.getPassword());
 		assertEquals(TEST_QUERY, config.getInitSQL());
+		assertEquals(true, config.isLogStatementsEnabled());
 
 
 		verify(mockProperties, partialTestClass);
