diff --git a/old_product.java b/new_product.java
index 23b9db5..c06829a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -17,8 +17,10 @@ public void configure(Properties props)
 		String password = props.getProperty(CONFIG_CONNECTION_PASSWORD, "password not set");
 		String connectionHookClass = props.getProperty(CONFIG_CONNECTION_HOOK_CLASS);
 		String initSQL = props.getProperty(CONFIG_INIT_SQL);
+		boolean closeConnectionWatch = configParseBoolean(props, CONFIG_CLOSE_CONNECTION_WATCH, false);
+		boolean logStatementsEnabled = configParseBoolean(props, CONFIG_LOG_STATEMENTS_ENABLED, false);
 
-
+		
 		// Remember Isolation level
 		this.isolation = PropertiesHelper.getInteger(Environment.ISOLATION, props);
 		this.autocommit = PropertiesHelper.getBoolean(Environment.AUTOCOMMIT, props);
@@ -41,7 +43,8 @@ public void configure(Properties props)
 			this.config.setPreparedStatementsCacheSize(preparedStatementCacheSize);
 			this.config.setStatementsCachedPerConnection(statementsCachedPerConnection);
 			this.config.setInitSQL(initSQL);
-			
+			this.config.setCloseConnectionWatch(closeConnectionWatch);
+			this.config.setLogStatementsEnabled(logStatementsEnabled);
 			if (connectionHookClass != null){
 				Object hookClass = Class.forName(connectionHookClass).newInstance();
 				this.config.setConnectionHook((ConnectionHook) hookClass);
