diff --git a/old_test.java b/new_test.java
index e590dde..fc89443 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,11 +7,11 @@
 
     runError("geo:z(<gml:MultiPoint><gml:Point><gml:coordinates>1,1" +
             "</gml:coordinates></gml:Point><gml:Point><gml:coordinates>1,2" +
-            "</gml:coordinates></gml:Point></gml:MultiPoint>)", GeoErrors.qname(5));
+            "</gml:coordinates></gml:Point></gml:MultiPoint>)", GeoErrors.qname(3));
 
     runError("geo:y(" +
             "<gml:LinearRing><gml:coordinates>0,0 20,0 0,20 0,0" +
-            "</gml:coordinates></gml:LinearRing>)", GeoErrors.qname(5));
+            "</gml:coordinates></gml:LinearRing>)", GeoErrors.qname(3));
     runError("geo:y(<gml:Point><gml:coordinates></gml:coordinates></gml:Point>)",
         GeoErrors.qname(2));
     runError("geo:y(<gml:geo><gml:coordinates>2,1</gml:coordinates></gml:geo>)",
