diff --git a/old_product.java b/new_product.java
index cfc63c6..d56f3ed 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static PointTransform_F64 transformPixelToRectNorm_F64(PinholeRadial param,
+public static PointTransform_F64 transformPixelToRectNorm_F64(CameraPinholeRadial param,
 																  DenseMatrix64F rectify,
 																  DenseMatrix64F rectifyK) {
 		if (rectifyK.get(0, 1) != 0)
