diff --git a/old_product.java b/new_product.java
index 519b5b5..b4098e3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,7 @@ public void hugeFolder() throws Exception {
 
         for (int i=0; i < names.size(); i++) {
             String filename = names.get(i);
-            context.getUserRoot().get().mkdir(filename, context.network, false, context.crypto.random, hasher);
+            context.getUserRoot().get().mkdir(filename, context.network, false, context.crypto);
             Set<FileWrapper> children = context.getUserRoot().get().getChildren(context.network).get();
             Assert.assertTrue("All children present", children.size() == i + 3); // 3 due to .keystore and shared
         }
