diff --git a/old_test.java b/new_test.java
index f5d2c0f..1a31e59 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,7 +11,7 @@
         for (int i=0; i < names.size(); i++) {
             String filename = names.get(i);
             long t1 = System.currentTimeMillis();
-            userRoot = userRoot.mkdir(filename, context.network, false, crypto.random, crypto.hasher).join();
+            userRoot = userRoot.mkdir(filename, context.network, false, crypto).join();
             long duration = System.currentTimeMillis() - t1;
             worst = Math.max(worst, duration);
             best = Math.min(best, duration);
