diff --git a/old_test.java b/new_test.java
index b397026..cafddf3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,15 @@
 @Test
+  @SuppressWarnings("unchecked")
   public void execute()
     throws Exception
   {
-    String result = (String)Eclim.execute(new String[]{"settings"});
+    List<Map<String,String>> results = (List<Map<String,String>>)
+      Eclim.execute(new String[]{"settings"});
 
-    Properties properties = new Properties();
-    properties.load(new ByteArrayInputStream(result.getBytes()));
+    HashMap<String,String> properties = new HashMap<String,String>();
+    for (Map<String,String> result : results){
+      properties.put(result.get("name"), result.get("value"));
+    }
 
     assertTrue("Missing org.eclim.user.email",
         properties.containsKey("org.eclim.user.email"));
