diff --git a/old_product.java b/new_product.java
index 35de537..e2bb832 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,6 @@ public Object execute(CommandLine commandLine)
     throws Exception
   {
     Option[] options = getPreferences().getOptions();
-    return SettingsFilter.instance.filter(commandLine, options);
+    Arrays.sort(options);
+    return options;
   }
\ No newline at end of file
