diff --git a/old_test.java b/new_test.java
index 5b74821..0ae9dea 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,5 @@
         MailboxStoragePayload mailboxStoragePayload = buildMailboxStoragePayload(senderKeys.getPublic(), receiverKeys.getPublic());
         ProtectedStorageEntry protectedStorageEntry = buildProtectedMailboxStorageEntry(mailboxStoragePayload, senderKeys, senderKeys.getPublic(), 1);
 
-        // should be assertFalse
-        Assert.assertTrue(protectedStorageEntry.isValidForAddOperation());
+        Assert.assertFalse(protectedStorageEntry.isValidForAddOperation());
     }
\ No newline at end of file
