diff --git a/old_product.java b/new_product.java
index 90e508d..b398c0a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,6 +4,13 @@
             return false;
 
         MailboxStoragePayload mailboxStoragePayload = this.getMailboxStoragePayload();
+
+        // Verify the Entry.receiversPubKey matches the Payload.ownerPubKey. This is a requirement for removal
+        if (!mailboxStoragePayload.getOwnerPubKey().equals(this.receiversPubKey)) {
+            log.debug("Entry receiversPubKey does not match payload owner which is a requirement for adding MailboxStoragePayloads");
+            return false;
+        }
+
         boolean result = mailboxStoragePayload.getSenderPubKeyForAddOperation() != null &&
                 mailboxStoragePayload.getSenderPubKeyForAddOperation().equals(this.getOwnerPubKey());
 
