diff --git a/old_product.java b/new_product.java
index 4a48a79..f0a226b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -24,7 +24,7 @@ public SpanId startNewSpan(String component, String operation, long timestamp) {
         newSpan.setName(operation);
         newSpan.setTimestamp(timestamp);
         newSpan.addToBinary_annotations(
-            new BinaryAnnotation(LOCAL_COMPONENT, component, spanAndEndpoint().endpoint()));
+            BinaryAnnotation.create(LOCAL_COMPONENT, component, spanAndEndpoint().endpoint()));
         spanAndEndpoint().state().setCurrentLocalSpan(newSpan);
         return newSpanId;
     }
\ No newline at end of file
