diff --git a/old_test.java b/new_test.java
index 9671334..5e0abad 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,10 +11,10 @@
 
         Span started = state.getCurrentLocalSpan();
 
-        assertEquals(1000L, started.timestamp);
+        assertEquals(1000L, started.getTimestamp().longValue());
         assertEquals(500L, started.startTick.longValue());
-        assertEquals("lc", started.binary_annotations.get(0).getKey());
-        assertEquals(COMPONENT_NAME, new String(started.binary_annotations.get(0).getValue(), Util.UTF_8));
-        assertEquals(state.getClientEndpoint(), started.binary_annotations.get(0).host);
+        assertEquals("lc", started.getBinary_annotations().get(0).getKey());
+        assertEquals(COMPONENT_NAME, new String(started.getBinary_annotations().get(0).getValue(), Util.UTF_8));
+        assertEquals(state.getClientEndpoint(), started.getBinary_annotations().get(0).host);
         assertEquals(OPERATION_NAME, started.getName());
     }
\ No newline at end of file
