diff --git a/old_test.java b/new_test.java
index 50073ff..6353e27 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     mLineageMaster.createLineage(Lists.<TachyonURI>newArrayList(),
         Lists.newArrayList(new TachyonURI("/test1")), mJob);
     FileInfo fileInfo = new FileInfo();
-    fileInfo.isCompleted = false;
+    fileInfo.setIsCompleted(false);
     Mockito.when(mFileSystemMaster.getFileInfo(Mockito.any(Long.class))).thenReturn(fileInfo);
     mLineageMaster.reinitializeFile("/test1", 500L, 10L);
     Mockito.verify(mFileSystemMaster).reinitializeFile(new TachyonURI("/test1"), 500L, 10L);
