diff --git a/old_product.java b/new_product.java
index 8003839..f9d1687 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,8 @@
 public boolean remove(TraceContext context) {
     if (context == null) throw new NullPointerException("context == null");
-    PendingSpan last = delegate.remove(new LookupKey(context));
+    LookupKey lookupKey = getLookupKey();
+    lookupKey.set(context);
+    PendingSpan last = delegate.remove(lookupKey);
     reportOrphanedSpans(); // also clears the reference relating to the recent remove
     return last != null;
   }
\ No newline at end of file
