diff --git a/old_test.java b/new_test.java
index 8d305a7..1c23862 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,24 +1,23 @@
 @Test
   public void testDelete() throws MetaDataException {
     // add an entry with a text and binary field
-    MetaDataEntry meta = new MetaDataEntry("tbd", "whatever");
+    MetaDataEntry meta = new MetaDataEntry("u", "q", "tbd", "whatever");
     meta.addField("text", "some text");
     meta.addField("binary", new byte[] { 'b', 'i', 'n' });
     mds.add(meta);
 
     // verify it was written
-    Assert.assertEquals(meta, mds.get("tbd", "whatever"));
+    Assert.assertEquals(meta, mds.get("u", "q", "tbd", "whatever"));
 
     // delete it
-    mds.delete("tbd", "whatever");
+    mds.delete("u", "q", "tbd", "whatever");
 
     // verify it's gone
-    Assert.assertNull(mds.get("tbd", "whatever"));
-    // TODO can't list until every implementation supports it
-    // Assert.assertFalse(mds.list(null, null).contains(meta));
+    Assert.assertNull(mds.get("u", "q", "tbd", "whatever"));
+    Assert.assertFalse(mds.list("u", null, "tbd", null).contains(meta));
 
     // add another entry with same name and type
-    MetaDataEntry meta1 = new MetaDataEntry("tbd", "whatever");
+    MetaDataEntry meta1 = new MetaDataEntry("u", "q", "tbd", "whatever");
     meta1.addField("other", "other text");
     // add should succeed, update should fail
     try {
@@ -32,5 +31,5 @@
     // read back entry and verify that it does not contain spurious
     // fields from the old meta data entry
     // verify it was written
-    Assert.assertEquals(meta1, mds.get("tbd", "whatever"));
+    Assert.assertEquals(meta1, mds.get("u", "q", "tbd", "whatever"));
   }
\ No newline at end of file
