diff --git a/old_test.java b/new_test.java
index 4d6c4cc..ae35e4a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
-@Test(groups={"tck"})
-    public void test_plus_Period_positiveMonths() {
-        MockSimplePeriod period = MockSimplePeriod.of(7, ChronoUnit.MONTHS);
+@Test
+    public void test_plus_TemporalAmount_positiveMonths() {
+        MockSimplePeriod period = MockSimplePeriod.of(7, MONTHS);
         LocalDateTime t = TEST_2007_07_15_12_30_40_987654321.plus(period);
         assertEquals(t, LocalDateTime.of(2008, 2, 15, 12, 30, 40, 987654321));
     }
\ No newline at end of file
