diff --git a/old_test.java b/new_test.java
index 70ff0fe..f864633 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,9 +3,9 @@
 		executeDataSet(COHORT_XML);
 		
 		// make a patient, add it using the method
-		Patient patientToAdd = new Patient(4);
+		Patient patientToAdd = Context.getPatientService().getPatient(3);
 		service.addPatientToCohort(service.getCohort(2), patientToAdd);
-		assertTrue(service.getCohort(2).contains(4));
+		assertTrue(service.getCohort(2).contains(3));
 		
 		// do it again to see if it fails
 		try {
