diff --git a/old_product.java b/new_product.java
index 23d674b..ae308c1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,8 @@
 @Override
 	public Cohort addPatientToCohort(Cohort cohort, Patient patient) {
 		if (!cohort.contains(patient.getPatientId())) {
-			CohortMembership cohortMembership = new CohortMembership(patient);
+			CohortMembership cohortMembership = new CohortMembership(patient.getPatientId());
 			Context.getCohortService().addMembershipToCohort(cohort, cohortMembership);
-			Context.getCohortService().saveCohort(cohort);
 		}
 		return cohort;
 	}
\ No newline at end of file
