diff --git a/old_product.java b/new_product.java
index d107389..ef1a5f2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,6 @@
-public void initialize() throws IOException {
+@Override
+    @SneakyThrows(IOException.class)
+    public void initialize(long epoch) {
         Preconditions.checkState(this.fileSystem == null, "HDFSStorage has already been initialized.");
         Exceptions.checkNotClosed(this.closed.get(), this);
         Configuration conf = new Configuration();
