diff --git a/old_product.java b/new_product.java
index bec5fc1..9e99070 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public XmlParser getDelegate(final String namespaceUri) {
+XmlParser getDelegate(final String namespaceUri) {
       try {
         return (XmlParser) getDelegateMethod.invoke(this, namespaceUri);
       } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
