diff --git a/old_test.java b/new_test.java
index 4a5c392..08cfed9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,26 +1,24 @@
 @Test
   public void testGetDelegate() {
     final TestCacheXmlParser cacheXmlParser = new TestCacheXmlParser();
-
-    assertTrue("delegates should be empty.", cacheXmlParser.getDelegates().isEmpty());
+    assertThat(cacheXmlParser.getDelegates()).as("delegates should be empty.").isEmpty();
 
     final MockXmlParser delegate = (MockXmlParser) cacheXmlParser.getDelegate(NAMESPACE_URI);
-
-    assertNotNull("Delegate should be found in classpath.", delegate);
-
-    assertSame("Should have same stack as cacheXmlParser.", cacheXmlParser.stack, delegate.stack);
-    assertSame("Should have same stack as cacheXmlParser.", cacheXmlParser.documentLocator,
-        delegate.documentLocator);
-
-    assertEquals("Should be exactly 1 delegate.", 1, cacheXmlParser.getDelegates().size());
-    assertNotNull("There should be an entry in delegates cache.",
-        cacheXmlParser.getDelegates().get(NAMESPACE_URI));
-    assertSame("Cached delegate should match the one from get.", delegate,
-        cacheXmlParser.getDelegates().get(NAMESPACE_URI));
+    assertThat(delegate).as("Delegate should be found in classpath.").isNotNull();
+    assertThat(cacheXmlParser.stack).as("Should have same stack as cacheXmlParser.")
+        .isSameAs(delegate.stack);
+    assertThat(cacheXmlParser.documentLocator).as("Should have same stack as cacheXmlParser.")
+        .isSameAs(delegate.documentLocator);
+    assertThat(cacheXmlParser.getDelegates().size()).as("Should be exactly 1 delegate.")
+        .isEqualTo(1);
+    assertThat(cacheXmlParser.getDelegates().get(NAMESPACE_URI))
+        .as("There should be an entry in delegates cache.").isNotNull();
+    assertThat(cacheXmlParser.getDelegates().get(NAMESPACE_URI))
+        .as("Cached delegate should match the one from get.").isSameAs(delegate);
 
     final MockXmlParser delegate2 = (MockXmlParser) cacheXmlParser.getDelegate(NAMESPACE_URI);
-    assertSame("Delegate should be the same between gets.", delegate, delegate2);
-    assertEquals("Should still be exactly 1 delegate.", 1, cacheXmlParser.getDelegates().size());
-
-    assertNull(cacheXmlParser.getDelegate("--nothing-should-use-this-namespace--"));
+    assertThat(delegate2).as("Delegate should be the same between gets.").isSameAs(delegate);
+    assertThat(cacheXmlParser.getDelegates().size()).as("Should still be exactly 1 delegate.")
+        .isEqualTo(1);
+    assertThat(cacheXmlParser.getDelegate("--nothing-should-use-this-namespace--")).isNull();
   }
\ No newline at end of file
