diff --git a/old_test.java b/new_test.java
index eca9e24..43902e3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
 		assertEquals(2,alg.cornerList.get(idx).index); // 2 corner squares
 
 		// tell it to only consider only proper corners
-		alg.setAllowNoCorner(false);
+		alg.setRequireCornerSquares(true);
 		idx = alg.selectCorner(createGridInfo(2,2, false));  // 4 non corner squares
 		assertEquals(-1,idx);
 		idx = alg.selectCorner(createGridInfo(2,3, false));  // 2 corner squares
