diff --git a/old_product.java b/new_product.java
index 43084af..55e1500 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,7 +13,7 @@ int selectCorner( GridInfo info ) {
 
 			// If there are no corner points which are valid corners, then any corner can be the origin if
 			// allowNoCorner is true
-			if( corner || (allowNoCorner && !bestIsCornerSquare) ) {
+			if( corner || ( !requireCornerSquares && !bestIsCornerSquare) ) {
 				// sanity check the shape
 				if( checkShape != null ) {
 					if( i%2==0 ) {
