diff --git a/old_test.java b/new_test.java
index f6adc70..c66c55d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,8 +5,6 @@
         objectCollection.add(new User("关羽", 24));
         objectCollection.add(new User("刘备", 25));
 
-        List<User> removeAll = CollectionsUtil.removeAll(objectCollection, "name", "刘备");
-        LOGGER.info(JsonUtil.format(removeAll));
-
-        LOGGER.info(JsonUtil.format(CollectionsUtil.selectRejected(objectCollection, "name", "刘备")));
+        LOGGER.info(JsonUtil.format(CollectionsUtil.removeAll(objectCollection, "name", "刘备")));
+        LOGGER.info(JsonUtil.format(CollectionsUtil.removeAll(objectCollection, "name", "刘备", "关羽")));
     }
\ No newline at end of file
