diff --git a/old_product.java b/new_product.java
index 4e77b79..980ba6c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-public static <O, V> List<O> removeAll(Collection<O> objectCollection,String propertyName,V value){
-        Collection<O> removeCollection = select(objectCollection, propertyName, value);
+@SafeVarargs
+    public static <O, V> List<O> removeAll(Collection<O> objectCollection,String propertyName,V...propertyValues){
+        Collection<O> removeCollection = select(objectCollection, propertyName, propertyValues);
         return removeAll(objectCollection, removeCollection);
     }
\ No newline at end of file
