diff --git a/old_product.java b/new_product.java
index 3181d9e..6194875 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,18 +9,13 @@ public SpanId startNewSpan(String requestName) {
         SpanId newSpanId = getNewSpanId();
         if (sample == null) {
             // No sample indication is present.
-            if (!traceSampler().isSampled(newSpanId.getTraceId())) {
+            if (!traceSampler().isSampled(newSpanId.traceId)) {
                 spanAndEndpoint().state().setCurrentClientSpan(null);
                 return null;
             }
         }
 
-        Span newSpan = new Span();
-        newSpan.setId(newSpanId.getSpanId());
-        newSpan.setTrace_id(newSpanId.getTraceId());
-        if (newSpanId.getParentSpanId() != null) {
-            newSpan.setParent_id(newSpanId.getParentSpanId());
-        }
+        Span newSpan = newSpanId.toSpan();
         newSpan.setName(requestName);
         spanAndEndpoint().state().setCurrentClientSpan(newSpan);
         return newSpanId;
