diff --git a/old_product.java b/new_product.java
index 35936bb..16874ec 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-void aggregateGroupBySV(int length, int[] docIdToGroupKey, ResultHolder resultHolder, double[]... valueArray);
\ No newline at end of file
+void aggregateGroupBySV(int length, int[] docIdToGroupKey, GroupByResultHolder resultHolder, double[]... valueArray);
\ No newline at end of file
