diff --git a/old_test.java b/new_test.java
index d2c2d5a..034b3e3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,7 +20,8 @@
     }
 
     MinAggregationFunction minAggregationFunction = new MinAggregationFunction();
-    ResultHolder resultHolder = ResultHolderFactory.getResultHolder(minAggregationFunction, MAX_NUM_GROUP_KEYS);
+    GroupByResultHolder resultHolder =
+        ResultHolderFactory.getGroupByResultHolder(minAggregationFunction, MAX_NUM_GROUP_KEYS);
 
     minAggregationFunction.aggregateGroupBySV(NUM_VALUES_TO_AGGREGATE, groupKeysForValues, resultHolder,
         valuesToAggregate);
