diff --git a/old_product.java b/new_product.java
index 73ed4a4..d3fbdc5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -33,6 +33,6 @@ public String getText(Chronology chrono, TemporalField field, long value,
         } else {
             return null;
         }
-        return CalendarDataUtility.retrieveCldrFieldValueName(
-                chrono.getCalendarType(), fieldIndex, fieldValue, toStyle(style), locale);
+        return CalendarDataUtility.retrieveJavaTimeFieldValueName(
+                chrono.getCalendarType(), fieldIndex, fieldValue, style.toCalendarStyle(), locale);
     }
\ No newline at end of file
