diff --git a/old_product.java b/new_product.java
index ab78eaa..174ccf1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public static PointTransform_F64 rectifyNormalized_F64(IntrinsicParameters param,
-														   DenseMatrix64F rectify ,
-														   DenseMatrix64F rectifyK )
+public static PointTransform_F64 transformPixelToRectNorm_F64(IntrinsicParameters param,
+																  DenseMatrix64F rectify,
+																  DenseMatrix64F rectifyK)
 	{
 		if( rectifyK.get(0,1) != 0 )
 			throw new IllegalArgumentException("Skew should be zero in rectified images");
