diff --git a/old_product.java b/new_product.java
index f4b8ba5..4b6bbb8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery containsText(String propName, String value) {
-        addCriterion(criterionBuilder.containsText(propName, value));
+        criterion = criterion.and(criterionBuilder.containsText(propName, value));
         return this;
     }
\ No newline at end of file
