diff --git a/old_test.java b/new_test.java
index bda8d18..e951782 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testContainsText() {
-        assertTrue(instance.containsText("name", "a").getQueryCriterions()
-                .contains(new ContainsTextCriterion("name", "a")));
+    	assertEquals(new ContainsTextCriterion("name", "a"), instance.containsText("name", "a").getQueryCriterion());
     }
\ No newline at end of file
