diff --git a/old_test.java b/new_test.java
index 5d9bb20..e64286c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test(description = "GET /vApp/{id}/screen", dependsOnMethods = { "testInstallVMwareTools" })
    public void testGetScreenImage() {
       // Power on Vm
-      vm = powerOnVm(vm.getHref());
+      vm = powerOnVm(vmUrn);
 
       // The method under test
-      byte[] image = vmApi.getScreenImage(vm.getHref());
+      byte[] image = vmApi.getScreenImage(vmUrn);
 
       // Check returned bytes against PNG header magic number
       byte[] pngHeaderBytes = new byte[] { (byte) 0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A };
