diff --git a/old_test.java b/new_test.java
index e7f358c..5558c7c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,10 +16,10 @@
     query(_DB_STORE.args(NAME, "one", ""));
     query(_DB_RENAME.args(NAME, "one", "two"));
     query(_DB_RETRIEVE.args(NAME, "two"));
-    error(_DB_RETRIEVE.args(NAME, "one"), Err.WHICHRES);
+    error(_DB_RETRIEVE.args(NAME, "one"), Err.WHICHRES_X);
 
     // invalid target
-    error(_DB_RENAME.args(NAME, "x/input.xml", " ''"), Err.BXDB_RENAME);
-    error(_DB_RENAME.args(NAME, "x/input.xml", " '/'"), Err.BXDB_RENAME);
-    error(_DB_RENAME.args(NAME, "x/input.xml", " '.'"), Err.BXDB_RENAME);
+    error(_DB_RENAME.args(NAME, "x/input.xml", " ''"), Err.BXDB_RENAME_X);
+    error(_DB_RENAME.args(NAME, "x/input.xml", " '/'"), Err.BXDB_RENAME_X);
+    error(_DB_RENAME.args(NAME, "x/input.xml", " '.'"), Err.BXDB_RENAME_X);
   }
\ No newline at end of file
