diff --git a/old_product.java b/new_product.java
index 9711db9..1c79e50 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
-@NotNull private PreprocessingTrainer makePreprocessorTrainer(String preprocessorClassName) throws Exception {
+@NotNull private PreprocessingTrainer makePreprocessorTrainer(String preprocessorClsName) throws Exception {
         return new PreprocessingTrainer() {
             @Override public Preprocessor fit(LearningEnvironmentBuilder envBuilder, DatasetBuilder datasetBuilder,
                 Preprocessor basePreprocessor) {
                 try {
-                    return createPreprocessor(basePreprocessor, preprocessorClassName);
+                    return createPreprocessor(basePreprocessor, preprocessorClsName);
                 } catch (Exception e) {
                     throw new RuntimeException(e);
                 }
