diff --git a/old_product.java b/new_product.java
index 7cf2c1b..151e03e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public Constr add(final QueryContext qc, final Expr... exprs) throws QueryException {
-    final int s = sc.ns.size();
+    final int size = sc.ns.size();
     try {
       for(final Expr expr : exprs) {
         more = false;
@@ -9,6 +9,6 @@ public Constr add(final QueryContext qc, final Expr... exprs) throws QueryExcept
       if(!text.isEmpty()) children.add(new FTxt(text.toArray()));
       return this;
     } finally {
-      sc.ns.size(s);
+      sc.ns.size(size);
     }
   }
\ No newline at end of file
