diff --git a/old_test.java b/new_test.java
index 9e7793f..e2b0e20 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,7 +17,8 @@
         int qualityThreshold = 0;
         int maxLevels = 1000;
 
-        Map<String, List<AlignmentInterval.Row>> result = (new AlignmentPacker()).packAlignments(iter, end, false, new AlignmentTrack.RenderOptions(), 10000);
+        Map<String, List<AlignmentInterval.Row>> result = (new AlignmentPacker()).packAlignments(iter, end, false,
+                new AlignmentTrack.RenderOptions());
         assertEquals(1, result.size());
         for (List<AlignmentInterval.Row> alignmentrows : result.values()) {
             for (AlignmentInterval.Row alignmentrow : alignmentrows) {
