diff --git a/old_product.java b/new_product.java
index 6ae3fab..c07a61c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,15 @@
 public static Dataframe parseCSVFile(Reader reader, String yVariable, Map<String, TypeInference.DataType> headerDataTypes, 
-                                           char delimiter, char quote, String recordSeparator, DatabaseConfiguration dbConf) {
+                                           char delimiter, char quote, String recordSeparator, Long skip, Long limit, DatabaseConfiguration dbConf) {
             Logger logger = LoggerFactory.getLogger(Dataframe.Builder.class);
             
+            if(skip == null) {
+                skip = 0L;
+            }
+            
+            if(limit == null) {
+                limit = Long.MAX_VALUE;
+            }
+            
             logger.info("Parsing CSV file");
             
             if (!headerDataTypes.containsKey(yVariable)) {
@@ -22,7 +30,7 @@ public static Dataframe parseCSVFile(Reader reader, String yVariable, Map<String
                                 .withRecordSeparator(recordSeparator);
             
             try (final CSVParser parser = new CSVParser(reader, format)) { 
-                ThreadMethods.throttledExecution(StreamMethods.enumerate(StreamMethods.stream(parser.spliterator(), false)), e -> { 
+                ThreadMethods.throttledExecution(StreamMethods.enumerate(StreamMethods.stream(parser.spliterator(), false)).skip(skip).limit(limit), e -> { 
                     Integer rId = e.getKey();
                     CSVRecord row = e.getValue();
                 
