diff --git a/old_test.java b/new_test.java
index 1f7886a..09cbcea 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,7 +16,7 @@
         
         Dataframe dataset;
         try (Reader fileReader = new InputStreamReader(this.getClass().getClassLoader().getResourceAsStream("datasets/cities.csv"), "UTF-8")) {
-            dataset = Dataframe.Builder.parseCSVFile(fileReader, "metro_population", headerDataTypes, ',', '"', "\r\n", dbConf);
+            dataset = Dataframe.Builder.parseCSVFile(fileReader, "metro_population", headerDataTypes, ',', '"', "\r\n", null, null, dbConf);
         }
         catch(UncheckedIOException | IOException ex) {
             logger.warn("Unable to download datasets, skipping test.");
