diff --git a/old_product.java b/new_product.java
index 35b1367..6291140 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,3 @@
 FileInStream openFile(AlluxioURI path, OpenFilePOptions options)
-      throws FileDoesNotExistException, IOException, AlluxioException;
\ No newline at end of file
+      throws FileDoesNotExistException, OpenDirectoryException, FileIncompleteException,
+      IOException, AlluxioException;
\ No newline at end of file
