diff --git a/old_test.java b/new_test.java
index 25993bf..d60b329 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void openFile() throws Exception {
     AlluxioURI file = new AlluxioURI("/file");
-    URIStatus status = new URIStatus(new FileInfo());
+    URIStatus status = new URIStatus(new FileInfo().setCompleted(true));
     GetStatusPOptions getStatusOptions = getOpenOptions(GetStatusPOptions.getDefaultInstance());
     when(mFileSystemMasterClient.getStatus(file, getStatusOptions))
         .thenReturn(status);
