diff --git a/old_product.java b/new_product.java
index 78f0001..7a57a24 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,6 +5,7 @@ private byte[] string() throws QueryIOException {
     while(pos < length) {
       final int p = pos;
       int ch = consume();
+
       // string is closed..
       if(ch == '"') {
         // unpaired surrogate?
@@ -59,8 +60,8 @@ private byte[] string() throws QueryIOException {
         if(ch >= 0xDC00 && ch <= 0xDFFF) {
           // compute resulting codepoint
           ch = (high - 0xD800 << 10) + ch - 0xDC00 + 0x10000;
-        } else if(escape) {
-          // add invalid high surrogate, treat low surrogate as next character
+        } else {
+          // add invalid high surrogate, treat expected low surrogate as new character
           add(high, p, pos);
         }
         high = 0;
