diff --git a/old_test.java b/new_test.java
index 1af754a..d0e554e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
     public void testIsFlightFull() throws Exception {
-        BitSet seats = new BitSet(SEATSConstants.NUM_SEATS_PER_FLIGHT);
+        BitSet seats = new BitSet(SEATSConstants.FLIGHTS_NUM_SEATS);
         
-        int seatnum = rand.nextInt(SEATSConstants.NUM_SEATS_PER_FLIGHT);
+        int seatnum = rand.nextInt(SEATSConstants.FLIGHTS_NUM_SEATS);
         assertFalse(seats.get(seatnum));
         seats.set(seatnum);
         assertTrue(seats.get(seatnum));
-        assertFalse(SEATSClient.isFlightFull(seats));
+        assertFalse(client.isFlightFull(seats));
     }
\ No newline at end of file
