diff --git a/old_product.java b/new_product.java
index ad2f201..f35742f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,4 @@
-public static <R> Observable<R> fromFunc0(Func0<? extends R> function, Scheduler scheduler) {
-        return Observable.create(OperationFromFunctionals.fromFunc0(function)).subscribeOn(scheduler);
+@Deprecated
+    public static <R> Observable<R> fromFunc0(Func0<? extends R> function, Scheduler scheduler) {
+        return fromCallable(function, scheduler);
     }
\ No newline at end of file
