diff --git a/old_product.java b/new_product.java
index 4286932..cd07d7a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,22 +1,15 @@
 private Item rename(final QueryContext ctx) throws QueryException {
     checkWrite(ctx);
 
-    final String path = path(string(checkStr(expr[0], ctx)));
+    final Data data = data(0, ctx);
+    final byte[] source = path(checkStr(expr[1], ctx));
+    final byte[] target = path(checkStr(expr[2], ctx));
 
     // the first step of the path should be the database name
-    final int pos = path.indexOf('/');
-    if(pos <= 0) NODB.thrw(input, path);
-    final byte[] db = token(path.substring(0, pos));
-    final Data data = ctx.resource.data(db, input);
-
-    final byte[] src = token(path.substring(pos + 1));
-    final byte[] trg = token(path(string(checkStr(expr[1], ctx))));
-
-    final int[] docs = data.doc(string(src));
-    for(final int pre : docs) {
-      final byte[] nm = newName(data, pre, src, trg);
-      ctx.updates.add(new ReplaceValue(pre, data, input, nm), ctx);
+    for(final int pre : data.doc(string(source))) {
+      final byte[] trg = ACreate.newName(data, pre, source, target);
+      if(trg.length == 0) EMPTYPATH.thrw(input, this);
+      ctx.updates.add(new ReplaceValue(pre, data, input, trg), ctx);
     }
-
     return null;
   }
\ No newline at end of file
