diff --git a/old_product.java b/new_product.java
index 9e170d8..c849001 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,6 @@
-public ServiceCall<Void> deleteWord(String customizationId, String wordName) {
-    Validator.notNull(customizationId, "customizationId cannot be null");
-    Validator.notNull(wordName, "words cannot be null");
-
-    RequestBuilder requestBuilder = RequestBuilder.delete(String.format(PATH_WORD, customizationId, wordName));
-    return createServiceCall(requestBuilder.build(), ResponseConverterUtils.getVoid());
+public ServiceCall<Void> deleteWord(DeleteWordOptions deleteWordOptions) {
+    Validator.notNull(deleteWordOptions, "deleteWordOptions cannot be null");
+    RequestBuilder builder = RequestBuilder.delete(String.format("/v1/customizations/%s/words/%s", deleteWordOptions
+        .customizationId(), deleteWordOptions.wordName()));
+    return createServiceCall(builder.build(), ResponseConverterUtils.getVoid());
   }
\ No newline at end of file
