diff --git a/old_test.java b/new_test.java
index 9f6ee41..2549627 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,11 @@
 
     server.enqueue(new MockResponse().addHeader(CONTENT_TYPE, HttpMediaType.APPLICATION_JSON).setBody("{}"));
 
-    service.deleteWord(id, wordName).execute();
+    DeleteWordOptions deleteOptions = new DeleteWordOptions.Builder()
+        .customizationId(id)
+        .wordName(wordName)
+        .build();
+    service.deleteWord(deleteOptions).execute();
     final RecordedRequest request = server.takeRequest();
 
     assertEquals("DELETE", request.getMethod());
