diff --git a/old_product.java b/new_product.java
index fda79d8..80e69f7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -18,8 +18,7 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
             logger.warn("Unable to download datasets, skipping test.");
             throw new RuntimeException(ex);
         }
-        
-        TextClassifier instance = new TextClassifier(dbName, conf);
+
         TextClassifier.TrainingParameters trainingParameters = new TextClassifier.TrainingParameters();
         
         //Classifier configuration
@@ -40,8 +39,9 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
         exParams.setMaxDistanceBetweenKwds(2);
         exParams.setExaminationWindowLength(6);
         trainingParameters.setTextExtractorParameters(exParams);
-        
-        instance.fit(dataset, trainingParameters);
+
+        TextClassifier instance = new TextClassifier(dbName, conf, trainingParameters);
+        instance.fit(dataset);
 
 
         ClassificationMetrics vm = instance.validate(dataset);
