diff --git a/old_test.java b/new_test.java
index 934ea2b..db380a0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,8 +11,7 @@
         
         
         String dbName = this.getClass().getSimpleName();
-        
-        Modeler instance = new Modeler(dbName, conf);
+
         Modeler.TrainingParameters trainingParameters = new Modeler.TrainingParameters();
         
         
@@ -31,8 +30,13 @@
         //feature selection configuration
         trainingParameters.setFeatureSelectorClass(null);
         trainingParameters.setFeatureSelectorTrainingParameters(null);
-        
-        instance.fit(trainingData, trainingParameters);
+
+        Modeler instance = new Modeler(dbName, conf, trainingParameters);
+        instance.fit(trainingData);
+
+        instance.close();
+
+        instance = new Modeler(dbName, conf);
 
         instance.predict(trainingData);
 
