diff --git a/old_test.java b/new_test.java
index 5168838..d3c4d1c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-  public void createDB() throws BaseXException {
-    new CreateDB(NAME).execute(context);
+  public void createDB() {
+    execute(new CreateDB(NAME));
     // check if database name equals argument of create command
     assertEquals(db(), NAME);
   }
\ No newline at end of file
