diff --git a/old_product.java b/new_product.java
index c2fcf08..e971eb7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-static void createDB() throws BaseXException {
-    new CreateDB(COLL, DOCPATH + DOC1).execute(context);
-    new Close().execute(context);
+static void createDB() {
+    execute(new CreateDB(COLL, DOCPATH + DOC1));
+    execute(new Close());
   }
\ No newline at end of file
