diff --git a/old_test.java b/new_test.java
index b1a97e5..61929a7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,7 @@
 @Test(groups={"tck"})
     public void test_withZoneSameInstant() {
-        LocalDateTime ldt = LocalDateTime.of(2008, 6, 30, 23, 30, 59, 0);
-        ZonedDateTime base = ZonedDateTime.of(ldt, ZONE_0100);
+        ZonedDateTime base = ZonedDateTime.of(TEST_LOCAL_2008_06_30_11_30_59_500, ZONE_0100);
         ZonedDateTime test = base.withZoneSameInstant(ZONE_0200);
-        ZonedDateTime expected = ZonedDateTime.of(ldt.plusHours(1), ZONE_0200);
+        ZonedDateTime expected = ZonedDateTime.of(TEST_LOCAL_2008_06_30_11_30_59_500.plusHours(1), ZONE_0200);
         assertEquals(test, expected);
     }
\ No newline at end of file
