diff --git a/old_product.java b/new_product.java
index d7b66d2..6e77a3a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,6 @@
 @Override
     public ZonedDateTime withZoneSameInstant(ZoneId zone) {
-        return zone == this.zone ? this : ofInstant(dateTime, zone);
+        Objects.requireNonNull(zone, "zone");
+        return this.zone.equals(zone) ? this :
+            create(dateTime.toEpochSecond(offset), dateTime.getNano(), zone);
     }
\ No newline at end of file
