diff --git a/old_product.java b/new_product.java
index b3ad18d..0da5a31 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,8 +3,9 @@
 		
 		Criteria crit = sessionFactory.getCurrentSession().createCriteria(Order.class);
 		
-		if (orderType != null)
+		if (orderType != null) {
 			crit.add(Restrictions.eq("orderType", orderType));
+		}
 		
 		crit.add(Restrictions.eq("patient", patient));
 		if (careSetting != null) {
