diff --git a/old_test.java b/new_test.java
index 7e44d2e..4eb1b68 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,5 @@
 @Test
-    public void testCreateTenant() throws Exception {
-        TenantUserDto tenant = createTenant();
-        Assert.assertFalse(strIsEmpty(tenant.getId()));
-        client.deleteTenant(tenant.getId());
-    }
\ No newline at end of file
+  public void testCreateTenant() throws Exception {
+    TenantDto tenant = createTenant();
+    Assert.assertFalse(strIsEmpty(tenant.getId()));
+  }
\ No newline at end of file
