diff --git a/old_product.java b/new_product.java
index 1f074c0..3009be0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,7 @@
-protected TenantUserDto createTenant() throws Exception {
-        return createTenant(null);
-    }
\ No newline at end of file
+protected TenantDto createTenant() throws Exception {
+    loginKaaAdmin();
+    TenantDto tenantDto = new TenantDto();
+    tenantDto.setName(generateString(TENANT_NAME));
+    tenantDto = client.editTenant(tenantDto);
+    return tenantDto;
+  }
\ No newline at end of file
