diff --git a/old_product.java b/new_product.java
index 3c381e9..d5da7e0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public static RegressionDataSet loadR(Reader reader, double sparseRatio, int vectorLength) throws IOException
     {
-        return (RegressionDataSet) loadG(reader, sparseRatio, vectorLength, false);
+        return loadR(reader, sparseRatio, vectorLength, DataStore.DEFAULT_STORE);
     }
\ No newline at end of file
