diff --git a/old_test.java b/new_test.java
index 68887c2..c9803fb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,8 @@
 		ImageFloat32 derivX = new ImageFloat32(200,200);
 		ImageFloat32 derivY = new ImageFloat32(200,200);
 
-		DescribePointSiftLowe alg = new DescribePointSiftLowe(4,4,8,1.5,0.5,0.2);
+		DescribePointSiftLowe<ImageFloat32> alg =
+				new DescribePointSiftLowe<ImageFloat32>(4,4,8,1.5,0.5,0.2,ImageFloat32.class);
 		int r = alg.getCanonicalRadius();
 		ImageMiscOps.fillRectangle(derivX,5.0f,60,60,2*r,2*r);
 
