diff --git a/old_product.java b/new_product.java
index d9f608a..89fc8e7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-CompletableFuture<Void> write(SegmentHandle segmentHandle, long offset, InputStream data, int length, Duration timeout);
\ No newline at end of file
+CompletableFuture<Void> write(String streamSegmentName, long offset, InputStream data, int length, Duration timeout);
\ No newline at end of file
