diff --git a/old_product.java b/new_product.java
index 4318c16..c08ab85 100644
--- a/old_product.java
+++ b/new_product.java
@@ -19,7 +19,7 @@ public static String toPrettyDateString(Date inDate){
             case 2: // 间隔2天
                 return doWithTwoDaysInterval(inDate, nowDate, isSameYear);
             default://spaceDay > 2     // 间隔大于2天
-                return isSameYear ? date2String(inDate, DatePattern.COMMON_DATE_AND_TIME_WITHOUT_YEAR_AND_SECOND)
-                                : date2String(inDate, DatePattern.COMMON_DATE_AND_TIME_WITHOUT_SECOND);
+                return isSameYear ? DateUtil.toString(inDate, DatePattern.COMMON_DATE_AND_TIME_WITHOUT_YEAR_AND_SECOND)
+                                : DateUtil.toString(inDate, DatePattern.COMMON_DATE_AND_TIME_WITHOUT_SECOND);
         }
     }
\ No newline at end of file
