diff --git a/old_product.java b/new_product.java
index 34b9134..7ae662a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,6 +2,7 @@ private Iter search(final QueryContext ctx) throws QueryException {
     final Data data = checkData(ctx);
     final Value terms = ctx.value(expr[1]);
     final Item opt = expr.length > 2 ? expr[2].item(ctx, info) : null;
-    final TokenMap tm = new FuncParams(Q_OPTIONS, info).parse(opt);
-    return search(data, terms, tm, this, ctx);
+    final FTOptions opts = new FTOptions();
+    new FuncOptions(Q_OPTIONS, info).parse(opt, opts);
+    return search(data, terms, opts, this, ctx);
   }
\ No newline at end of file
