diff --git a/old_product.java b/new_product.java
index 9e85dac..23256b0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,5 @@
-public HMM<O> learn(int[][] observations, int iterations) {
-        HMM<O> hmm = this;
-
+public void update(int[][] observations, int iterations) {
         for (int iter = 0; iter < iterations; iter++) {
-            hmm = hmm.iterate(observations);
+            iterate(observations);
         }
-
-        return hmm;
     }
\ No newline at end of file
