diff --git a/old_product.java b/new_product.java
index 35d8a7e..380d6ff 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@ public void add(final I_M_ReceiptSchedule_Alloc rsa)
 		final IHUContext huContext = getHUContext();
 		final I_M_HU tuHU = rsa.getM_TU_HU();
 		final HUReceiptLinePartAttributes partAttributes = HUReceiptLinePartAttributes.newInstance(huContext, tuHU);
-		final HUReceiptLinePartCandidate receiptLinePartToAdd = new HUReceiptLinePartCandidate(partAttributes);
+		final HUReceiptLinePartCandidate receiptLinePartToAdd = new HUReceiptLinePartCandidate(partAttributes, productId, uomId);
 		receiptLinePartToAdd.add(rsa);
 
 		//
