diff --git a/old_test.java b/new_test.java
index 8cbb31c..5593ec8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,9 +3,9 @@
 	{
 		final ReceiptQtyExpectation<Object> expectationForOneTransaction = ReceiptQtyExpectation.newInstance()
 				.qtyPrecision(2)
-				.qty("7")
+				.qty(StockQtyAndUOMQtys.create(new BigDecimal("8"), productId, new BigDecimal("7"), uomId)) // note that at the end of the day, we care for the uomQty, i.e. the potential catch quantity
 				.qualityDiscountPercent("3")
-				.qtyWithIssues("0.21") // = 3 * 7% = 0.21
+				.qtyWithIssues(StockQtyAndUOMQtys.create(new BigDecimal("8"), productId, new BigDecimal("0.21"), uomId)) // = 3 * 7% = 0.21
 		;
 		test_add_sameQty_ExpectConstantPercent(expectationForOneTransaction);
 	}
\ No newline at end of file
