diff --git a/old_product.java b/new_product.java
index cc7e536..f58f5cd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,5 +3,5 @@ final DBNode open(final QueryContext qc, final boolean id) throws QueryException
     final int v = (int) toLong(exprs[1], qc);
     final int pre = id ? data.pre(v) : v;
     if(pre >= 0 && pre < data.meta.size) return new DBNode(data, pre);
-    throw BXDB_RANGE_X_X_X.get(info, data.meta.name, id ? "ID" : "pre", v);
+    throw DB_RANGE_X_X_X.get(info, data.meta.name, id ? "ID" : "pre", v);
   }
\ No newline at end of file
