diff --git a/old_product.java b/new_product.java
index 60c04d4..ce92fd0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @GET
         @Path("responseFilterSync")
         public Response responseFilterSync() {
-            return Response.created(getUriBuilder().segment("responseFilterSync").build()).build();
+            return Response.created(URI.create("responseFilterSync")).build();
         }
\ No newline at end of file
