diff --git a/old_test.java b/new_test.java
index 477b764..29eef00 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,13 +6,13 @@
             connection43.setShardingKey(shardingKey);
         }
         catch (SQLException e) {
-            assert (e.getMessage().contains(TestResource.getResource("R_notImplemented")));
+            assert (e.getMessage().contains(TestResource.getResource("R_operationNotSupported")));
         }
         try {
             connection43.setShardingKey(shardingKey, superShardingKey);
         }
         catch (SQLException e) {
-            assert (e.getMessage().contains(TestResource.getResource("R_notImplemented")));
+            assert (e.getMessage().contains(TestResource.getResource("R_operationNotSupported")));
         }
        
     }
\ No newline at end of file
