diff --git a/old_test.java b/new_test.java
index 464d0f6..f7bde62 100644
--- a/old_test.java
+++ b/new_test.java
@@ -24,7 +24,7 @@
         trainingParameters.setCategoricalEncoderTrainingParameters(ceParams);
         
         //feature selection configuration
-        trainingParameters.setFeatureSelectorTrainingParameters(null);
+        trainingParameters.setFeatureSelectorTrainingParametersList(Arrays.asList());
 
         //model Configuration
         MultinomialNaiveBayes.TrainingParameters modelTrainingParameters = new MultinomialNaiveBayes.TrainingParameters();
