diff --git a/old_product.java b/new_product.java
index 161fb7e..c106195 100644
--- a/old_product.java
+++ b/new_product.java
@@ -28,7 +28,7 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector, NS e
         trainingParameters.setCategoricalEncoderTrainingParameters(null);
         
         //feature selection configuration
-        trainingParameters.setFeatureSelectorTrainingParameters(featureSelectorTrainingParameters);
+        trainingParameters.setFeatureSelectorTrainingParametersList(Arrays.asList(featureSelectorTrainingParameters));
 
         //classifier configuration
         trainingParameters.setModelerTrainingParameters(modelerTrainingParameters);
