diff --git a/old_product.java b/new_product.java
index f1735f3..ea18ed1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,4 @@
 public static <U> CompletionStage<U> completedStage(U value) {
-        if (JAVA8)
-            throw new UnsupportedOperationException();
-
         ManagedExecutorService defaultExecutor = AccessController.doPrivileged(getDefaultManagedExecutorAction);
-        ManagedCompletionStage<U> stage = new ManagedCompletionStage<U>(defaultExecutor);
-        stage.super_complete(value);
-        return stage;
+        return completedStage(value, defaultExecutor);
     }
\ No newline at end of file
