diff --git a/old_test.java b/new_test.java
index 214ca92..79e2c0a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,6 @@
 	public void findPatients_shouldLoggedInUserShouldLoadTheirOwnPatientObject() throws Exception {
 		executeDataSet("org/openmrs/web/dwr/include/DWRPatientService-patientisauser.xml");
 		DWRPatientService dwrService = new DWRPatientService();
-		Collection<Object> resultObjects = dwrService.findPatients("Super", false, null, null);
+		Collection<Object> resultObjects = dwrService.findPatients("Super", false);
 		Assert.assertEquals(1, resultObjects.size());
 	}
\ No newline at end of file
