diff --git a/old_test.java b/new_test.java
index 7e2568d..d279f9b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public void delete() throws BaseXException {
     // try to delete a package which is not installed
     try {
-      new RepoManager(ctx.repo).delete("xyz", null);
+      new RepoManager(ctx).delete("xyz");
       fail("Not installed package not detected.");
     } catch(final QueryException ex) {
       check(ex, Err.PKGNOTEXIST);
@@ -35,7 +35,7 @@
 
     // try to delete pkg3
     try {
-      new RepoManager(ctx.repo).delete(pkg3Dir, null);
+      new RepoManager(ctx).delete(pkg3Dir);
       fail("Package involved in a dependency was deleted.");
     } catch(final QueryException ex) {
       check(ex, Err.PKGDEP);
