diff --git a/old_test.java b/new_test.java
index 4354cbe..c0fef86 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test(groups={"tck"})
     public void test_hashCode_ZOT() {
-        ZoneOffsetTransition zot = new ZoneOffsetTransition(OffsetDateTime.of(2010, 3, 31, 1, 0, OFFSET_0200), OFFSET_0300);
-        ZoneOffsetInfo test = new ZoneOffsetInfo(LocalDateTime.of(2010, 3, 31, 1, 0), null, zot);
+        ZoneOffsetTransition zot = ZoneOffsetTransition.of(OffsetDateTime.of(2010, 3, 31, 1, 0, OFFSET_0200), OFFSET_0300);
+        ZoneOffsetInfo test = make(null, zot);
         
         assertEquals(test.hashCode(), test.hashCode());
     }
\ No newline at end of file
