diff --git a/old_product.java b/new_product.java
index 5ee3d74..69c46b0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 @Override
     public int hashCode() {
-        return dateTime.hashCode() ^ (transition != null ? transition.hashCode() : offset.hashCode());
+        return (transition != null ? transition.hashCode() : offset.hashCode());
     }
\ No newline at end of file
