diff --git a/old_test.java b/new_test.java
index 3f28eb4..d49843f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -22,11 +22,11 @@
         xData4.put("4", "s");
         instance.add(new Record(xData4, null));
         
-        Map<Object, Dataset.ColumnType> expResult = new LinkedHashMap<>();
-        expResult.put("1", Dataset.ColumnType.DUMMYVAR);
-        expResult.put("2", Dataset.ColumnType.NUMERICAL);
-        expResult.put("3", Dataset.ColumnType.ORDINAL);
-        expResult.put("4", Dataset.ColumnType.CATEGORICAL);
-        Map<Object, Dataset.ColumnType> result = instance.getColumns();
+        Map<Object, TypeInference.DataType> expResult = new LinkedHashMap<>();
+        expResult.put("1", TypeInference.DataType.BOOLEAN);
+        expResult.put("2", TypeInference.DataType.NUMERICAL);
+        expResult.put("3", TypeInference.DataType.ORDINAL);
+        expResult.put("4", TypeInference.DataType.CATEGORICAL);
+        Map<Object, TypeInference.DataType> result = instance.getXDataTypes();
         assertEquals(expResult, result);
     }
\ No newline at end of file
