diff --git a/old_product.java b/new_product.java
index 1e0c044..7164736 100644
--- a/old_product.java
+++ b/new_product.java
@@ -37,9 +37,9 @@ public static RESTCmd get(final RESTSession session) throws IOException {
 
       // handle database options
       try(QueryProcessor qp = new QueryProcessor("*/*:option", ctx).context(doc)) {
-        for(final Item it : qp.value()) {
-          final String name = value("@name", it, ctx).toUpperCase(Locale.ENGLISH);
-          final String value = value("@value", it, ctx);
+        for(final Item item : qp.value()) {
+          final String name = value("@name", item, ctx).toUpperCase(Locale.ENGLISH);
+          final String value = value("@value", item, ctx);
           ctx.options.assign(name, value);
         }
       }
@@ -47,10 +47,10 @@ public static RESTCmd get(final RESTSession session) throws IOException {
       // handle variables
       final Map<String, String[]> vars = new HashMap<>();
       try(QueryProcessor qp = new QueryProcessor("*/*:variable", ctx).context(doc)) {
-        for(final Item it : qp.value()) {
-          final String name = value("@name", it, ctx);
-          final String value = value("@value", it, ctx);
-          final String type = value("@type", it, ctx);
+        for(final Item item : qp.value()) {
+          final String name = value("@name", item, ctx);
+          final String value = value("@value", item, ctx);
+          final String type = value("@type", item, ctx);
           vars.put(name, new String[] { value, type });
         }
       }
@@ -59,10 +59,10 @@ public static RESTCmd get(final RESTSession session) throws IOException {
       String val = null;
       try(QueryProcessor qp = new QueryProcessor("*/*:context/(*, text()[normalize-space()])",
           ctx).context(doc)) {
-        for(final Item it : qp.value()) {
+        for(final Item item : qp.value()) {
           if(val != null) throw HTTPCode.MULTIPLE_CONTEXT_X.get();
           // create main memory instance of the specified node
-          val = DataBuilder.stripNS((ANode) it, REST_URI, ctx).serialize().toString();
+          val = DataBuilder.stripNS((ANode) item, REST_URI, ctx).serialize().toString();
         }
       }
 
