diff --git a/old_test.java b/new_test.java
index 6486b28..4635d75 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,19 +2,19 @@
     // GET1 - just send a GET request
     try(QueryProcessor qp = new QueryProcessor(_HTTP_SEND_REQUEST.args(
         " <http:request method='get' href='" + REST_ROOT + "'/>"), ctx)) {
-      final Value v = qp.value();
-      checkResponse(v, 2, HttpURLConnection.HTTP_OK);
+      final Value value = qp.value();
+      checkResponse(value, 2, HttpURLConnection.HTTP_OK);
 
-      assertEquals(NodeType.DOC, v.itemAt(1).type);
+      assertEquals(NodeType.DOC, value.itemAt(1).type);
     }
 
     // GET2 - with override-media-type='text/plain'
     try(QueryProcessor qp = new QueryProcessor(_HTTP_SEND_REQUEST.args(
         " <http:request method='get' override-media-type='text/plain'/>", REST_ROOT), ctx)) {
-      final Value v = qp.value();
-      checkResponse(v, 2, HttpURLConnection.HTTP_OK);
+      final Value value = qp.value();
+      checkResponse(value, 2, HttpURLConnection.HTTP_OK);
 
-      assertEquals(AtomType.STR, v.itemAt(1).type);
+      assertEquals(AtomType.STR, value.itemAt(1).type);
     }
 
     // Get3 - with status-only='true'
