diff --git a/old_test.java b/new_test.java
index 547fb0e..3bf0f3e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
   public void attribute() throws BaseXException {
     // run function without and with index
-    new DropIndex(Commands.CmdIndex.ATTRIBUTE).execute(context);
+    new DropIndex(CmdIndex.ATTRIBUTE).execute(context);
     query(DATA.args(_DB_ATTRIBUTE.args(NAME, "0")), "0");
-    new CreateIndex(Commands.CmdIndex.ATTRIBUTE).execute(context);
+    new CreateIndex(CmdIndex.ATTRIBUTE).execute(context);
     query(DATA.args(_DB_ATTRIBUTE.args(NAME, "0")), "0");
     query(DATA.args(_DB_ATTRIBUTE.args(NAME, "0", "id")), "0");
     query(DATA.args(_DB_ATTRIBUTE.args(NAME, "0", "XXX")), "");
