diff --git a/old_product.java b/new_product.java
index c590790..5240c04 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,9 +6,9 @@ final Iter attribute(final IndexAccess ia, final QueryContext qc, final int a)
 
     // parse and compile the name test
     final QNm nm = new QNm(toToken(exprs[a], qc), sc);
-    if(!nm.hasPrefix()) nm.uri(sc.ns.uri(Token.EMPTY));
+    if(!nm.hasPrefix()) nm.uri(sc.ns.uri(EMPTY));
 
-    final NameTest nt = new NameTest(nm, Test.Kind.URI_NAME, true, sc.elemNS);
+    final NameTest nt = new NameTest(nm, Kind.URI_NAME, true, sc.elemNS);
     // return empty sequence if test will yield no results
     if(!nt.optimize(qc)) return Empty.ITER;
 
