diff --git a/old_test.java b/new_test.java
index af96f2b..3ac07a0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,7 +12,7 @@
 
     // first request
     RecordedRequest request = server.takeRequest();
-    String path = PATH_DETECT_FACES + "?" + VERSION_DATE + "=" + VisualRecognition.VERSION_DATE_2016_05_20 + "&api_key="
+    String path = PATH_DETECT_FACES + "?" + VERSION_DATE + "=2016-05-20&api_key="
         + API_KEY;
 
     assertEquals(path, request.getPath());
