diff --git a/old_product.java b/new_product.java
index 85503b3..83de7c4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public ServiceCall<DetectedFaces> detectFaces(DetectFacesOptions detectFacesOptions) {
     Validator.notNull(detectFacesOptions, "detectFacesOptions cannot be null");
-    Validator.isTrue((detectFacesOptions.imagesFile() != null) || (detectFacesOptions.parameters() != null),
-        "At least one of imagesFile or parameters must be supplied.");
+    Validator.isTrue((detectFacesOptions.imagesFile() != null) || (detectFacesOptions.url() != null),
+        "At least one of imagesFile or url must be supplied.");
     RequestBuilder builder = RequestBuilder.post("/v3/detect_faces");
     builder.query(VERSION, versionDate);
     MultipartBody.Builder multipartBuilder = new MultipartBody.Builder();
@@ -11,8 +11,8 @@ public ServiceCall<DetectedFaces> detectFaces(DetectFacesOptions detectFacesOpti
           .imagesFileContentType());
       multipartBuilder.addFormDataPart("images_file", detectFacesOptions.imagesFilename(), imagesFileBody);
     }
-    if (detectFacesOptions.parameters() != null) {
-      multipartBuilder.addFormDataPart("parameters", detectFacesOptions.parameters());
+    if (detectFacesOptions.url() != null) {
+      multipartBuilder.addFormDataPart("url", detectFacesOptions.url());
     }
     builder.body(multipartBuilder.build());
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(DetectedFaces.class));
