diff --git a/old_test.java b/new_test.java
index a0e61b7..0bcfa2f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test(description = "GET /vApp/{id}")
    public void testGetVApp() {
       // The method under test
-      vApp = vAppApi.getVApp(vAppURI);
+      vApp = vAppApi.get(vAppUrn);
 
       // Check the retrieved object is well formed
       checkVApp(vApp);
@@ -18,5 +18,5 @@
       // TODO source.href vAppTemplateURI
 
       // Check status
-      assertVAppStatus(vAppURI, Status.POWERED_OFF);
+      assertVAppStatus(vAppUrn, Status.POWERED_OFF);
    }
\ No newline at end of file
