diff --git a/old_product.java b/new_product.java
index 484603e..c4b74af 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,8 +4,8 @@ public static <T> double loocv(ClassifierTrainer<T> trainer, T[] x, int[] y, Cla
         
         LOOCV loocv = new LOOCV(n);
         for (int i = 0; i < n; i++) {
-            T[] trainx = Math.slice(x, loocv.train[i]);
-            int[] trainy = Math.slice(y, loocv.train[i]);
+            T[] trainx = MathEx.slice(x, loocv.train[i]);
+            int[] trainy = MathEx.slice(y, loocv.train[i]);
             
             Classifier<T> classifier = trainer.train(trainx, trainy);
 
