diff --git a/old_product.java b/new_product.java
index c5509a8..1a48e5a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,13 +8,14 @@
 	{
 		final CandidatesQuery segment = CandidatesQuery.builder()
 				.type(Type.STOCK)
-				.date(date)
-				.dateOperator(DateOperator.after)
-				.productId(productId)
-				.warehouseId(warehouseId)
+				.materialDescr(MaterialDescriptor.builderForQuery()
+						.date(date)
+						.productId(productId)
+						.warehouseId(warehouseId).build())
+				.dateOperator(DateOperator.AFTER)
 				.build();
 
-		final List<Candidate> candidatesToUpdate = candidateRepository.retrieveMatchesOrderByDateAndSeqNo(segment);
+		final List<Candidate> candidatesToUpdate = candidateRepository.retrieveOrderedByDateAndSeqNo(segment);
 		for (final Candidate candidate : candidatesToUpdate)
 		{
 			final BigDecimal newQty = candidate.getQuantity().add(delta);
