diff --git a/old_test.java b/new_test.java
index 942fb7e..263cba5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,5 +3,6 @@
         int[] int1 = { 2, 1 };
         assertEquals("2,1", ConvertUtil.toString(new ToStringConfig(","), int1));
         assertEquals("2", ConvertUtil.toString(new ToStringConfig(","), 2));
-        assertEquals("2", ConvertUtil.toString(new ToStringConfig(","), new Integer(2)));
+        assertEquals(",,,", ConvertUtil.toString(new ToStringConfig(",", true), ",", ","));
+        assertEquals("2,", ConvertUtil.toString(new ToStringConfig(",", true), new Integer(2), null));
     }
\ No newline at end of file
