diff --git a/old_test.java b/new_test.java
index 0ee2f02..e45de65 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,8 +10,8 @@
     array(_ARRAY_SUBARRAY.args(" array { 1 to 5 }", " 1", " 1"), "[1]");
     array(_ARRAY_SUBARRAY.args(" array { 1 to 5 }", " 2", " 3"), "[2, 3, 4]");
 
-    error(_ARRAY_SUBARRAY.args(" [1]", " 0", " 0"), Err.ARRAYBOUNDS);
-    error(_ARRAY_SUBARRAY.args(" [1]", " 1", " -1"), Err.ARRAYNEG);
-    error(_ARRAY_SUBARRAY.args(" []", " 1", " 1"), Err.ARRAYBOUNDS);
-    error(_ARRAY_SUBARRAY.args(" [1]", " 1", " 2"), Err.ARRAYBOUNDS);
+    error(_ARRAY_SUBARRAY.args(" [1]", " 0", " 0"), Err.ARRAYBOUNDS_X_X);
+    error(_ARRAY_SUBARRAY.args(" [1]", " 1", " -1"), Err.ARRAYNEG_X);
+    error(_ARRAY_SUBARRAY.args(" []", " 1", " 1"), Err.ARRAYBOUNDS_X_X);
+    error(_ARRAY_SUBARRAY.args(" [1]", " 1", " 2"), Err.ARRAYBOUNDS_X_X);
   }
\ No newline at end of file
