diff --git a/old_product.java b/new_product.java
index 027eb02..978a2d1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,9 @@ public ServiceCall<Workspace> updateWorkspace(UpdateWorkspaceOptions updateWorks
     RequestBuilder builder = RequestBuilder.post(String.format("/v1/workspaces/%s", updateWorkspaceOptions
         .workspaceId()));
     builder.query(VERSION, versionDate);
+    if (updateWorkspaceOptions.append() != null) {
+      builder.query("append", String.valueOf(updateWorkspaceOptions.append()));
+    }
     final JsonObject contentJson = new JsonObject();
     if (updateWorkspaceOptions.name() != null) {
       contentJson.addProperty("name", updateWorkspaceOptions.name());
