diff --git a/old_test.java b/new_test.java
index a1e0792..ac6c94b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -33,8 +33,10 @@
 
       String counterExampleText = "What are you drinking";
       CreateCounterexample counterexample2 = new CreateCounterexample.Builder(counterExampleText).build();
-      UpdateWorkspaceOptions updateOptions =
-              new UpdateWorkspaceOptions.Builder(workspaceId).addCounterexample(counterexample2).build();
+      UpdateWorkspaceOptions updateOptions = new UpdateWorkspaceOptions.Builder(workspaceId)
+          .addCounterexample(counterexample2)
+          .append(false)
+          .build();
 
       Workspace updateResponse = service.updateWorkspace(updateOptions).execute();
 
