diff --git a/old_product.java b/new_product.java
index d088195..d206008 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,11 +10,11 @@ static Iter search(final Data data, final Value terms, final FTOptions opts,
     final FTOpt opt = new FTOpt().copy(data.meta);
     FTMode mode = FTMode.ANY;
     if(opts != null) {
-      opt.set(FZ, opts.bool(FTOptions.FUZZY));
-      opt.set(WC, opts.bool(FTOptions.WILDCARDS));
-      final String md = opts.string(FTOptions.MODE);
+      opt.set(FZ, opts.get(FTOptions.FUZZY));
+      opt.set(WC, opts.get(FTOptions.WILDCARDS));
+      final String md = opts.get(FTOptions.MODE);
       mode = FTMode.get(md);
-      if(mode == null) ELMOPTION.thrw(info, md);
+      if(mode == null) INVALIDOPT.thrw(info, Util.info(Text.INVALID_VALUE_X_X, "mode", md));
     }
 
     ctx.ftOpt(opt);
