diff --git a/old_test.java b/new_test.java
index 28940db..2728d0f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -28,7 +28,7 @@
         "Exercise 1");
 
     // check buggy options
-    error(_FT_SEARCH.args(NAME, "x", " { 'x':'y' }"), Err.ELMOPTION);
-    error(_FT_SEARCH.args(NAME, "x", " { 'mode':'' }"), Err.ELMOPTION);
+    error(_FT_SEARCH.args(NAME, "x", " { 'x':'y' }"), Err.INVALIDOPT);
+    error(_FT_SEARCH.args(NAME, "x", " { 'mode':'' }"), Err.INVALIDOPT);
     error(_FT_SEARCH.args(NAME, "x", " 1"), Err.ELMMAPTYPE);
   }
\ No newline at end of file
