diff --git a/old_product.java b/new_product.java
index 429cf54..ad6a145 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,13 +10,13 @@
         // Execute script
         final DataSet dataSet = createShellTask(request);
         final StructType schema = dataSet.schema();
-        TransformResponse response = submitTransformJob(new ShellTransformStage(dataSet), getPolicies(request));
+        TransformResponse response = submitTransformJob(new ShellTransformStage(dataSet, converterService), getPolicies(request));
 
         // Build response
         if (response.getStatus() != TransformResponse.Status.SUCCESS) {
             final String table = response.getTable();
             final TransformQueryResult partialResult = new TransformQueryResult();
-            partialResult.setColumns(Arrays.<QueryResultColumn>asList(new QueryResultRowTransform(schema, table).columns()));
+            partialResult.setColumns(Arrays.<QueryResultColumn>asList(new QueryResultRowTransform(schema, table, converterService).columns()));
 
             response = new TransformResponse();
             response.setProgress(0.0);
