diff --git a/old_test.java b/new_test.java
index 4c646c1..acac17b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,7 +18,7 @@
         final TransformRequest request = new TransformRequest();
         request.setScript("sqlContext.range(1,10)");
 
-        final TransformService service = new TransformService(TransformScript.class, engine, sparkContextService, new MockJobTrackerService());
+        final TransformService service = new TransformService(TransformScript.class, engine, sparkContextService, new MockJobTrackerService(), Mockito.mock(DataSetConverterService.class));
         final TransformResponse response = service.execute(request);
         Assert.assertEquals(TransformResponse.Status.PENDING, response.getStatus());
 
