diff --git a/old_test.java b/new_test.java
index 4e6622d..822a50f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,8 @@
 @Test
     public void testParseHertz() {
+        assertEquals(-1L, ParseUtil.parseHertz("OneHz"));
+        assertEquals(-1L, ParseUtil.parseHertz("NotEvenAHertz"));
+        assertEquals(Long.MAX_VALUE, ParseUtil.parseHertz("10000000000000000000 Hz"));
         assertEquals(1L, ParseUtil.parseHertz("1Hz"));
         assertEquals(500L, ParseUtil.parseHertz("500 Hz"));
         assertEquals(1000L, ParseUtil.parseHertz("1kHz"));
