diff --git a/old_product.java b/new_product.java
index cfb849a..143b1cd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,9 @@
 public static long parseHertz(String hertz) {
         Matcher matcher = HERTZ.matcher(hertz.trim());
         if (matcher.find() && matcher.groupCount() == 3) {
-            try {
-                Double value = Double.valueOf(matcher.group(1)) * multipliers.getOrDefault(matcher.group(3), -1L);
-                return value < 0 ? -1L : value.longValue();
-            } catch (NumberFormatException e) {
-                LOG.trace("", e);
-            }
+            // Regexp enforces #(.#) format so no test for NFE required
+            Double value = Double.valueOf(matcher.group(1)) * multipliers.getOrDefault(matcher.group(3), -1L);
+            return value < 0d ? -1L : value.longValue();
         }
         return -1L;
     }
\ No newline at end of file
