diff --git a/old_test.java b/new_test.java
index 0a0077d..cab73d2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,15 +5,15 @@
 		};
 
 		Future<StackTraceSample> sampleFuture = coord.triggerStackTraceSample(
-				vertices, 1, new FiniteDuration(100, TimeUnit.MILLISECONDS), 0);
+				vertices, 1, Time.milliseconds(100L), 0);
 
-		assertFalse(sampleFuture.isCompleted());
+		assertFalse(sampleFuture.isDone());
 
 		// Cancel
 		coord.cancelStackTraceSample(0, null);
 
 		// Verify completed
-		assertTrue(sampleFuture.isCompleted());
+		assertTrue(sampleFuture.isDone());
 
 		// Verify no more pending samples
 		assertEquals(0, coord.getNumberOfPendingSamples());
