diff --git a/old_product.java b/new_product.java
index 8e22f71..900ae26 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
 public Language getLanguage(Map<String, Object> params) {
-		return executeRequest(params, AlchemyAPI.language, Language.class, "text", "html", "url");
-	}
\ No newline at end of file
+    return executeRequest(params, AlchemyAPI.language, Language.class, TEXT, HTML, URL);
+  }
\ No newline at end of file
