diff --git a/old_test.java b/new_test.java
index 48f15e4..c7228d5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
-	public void testGetLanguage() {
-		Map<String, Object> params = new HashMap<String, Object>();
-		params.put(AlchemyLanguage.URL, "http://news.google.fr/");
-		Language language = service.getLanguage(params);
-		Assert.assertNotNull(language);
-	}
\ No newline at end of file
+  public void testGetLanguage() {
+    final Map<String, Object> params = new HashMap<String, Object>();
+    params.put(AlchemyLanguage.URL, "http://news.google.fr/");
+    final Language language = service.getLanguage(params);
+    Assert.assertNotNull(language);
+  }
\ No newline at end of file
