diff --git a/old_product.java b/new_product.java
index 21d3fce..f342f29 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 private Item lookup(final QueryContext qc, final InputInfo ii) throws QueryException {
-    final QNm name = checkQNm(exprs[0], qc, sc);
-    final long arity = checkItr(exprs[1], qc);
-    if(arity < 0 || arity > Integer.MAX_VALUE) throw FUNCUNKNOWN.get(ii, name);
+    final QNm name = toQNm(exprs[0], qc, sc, false);
+    final long arity = toLong(exprs[1], qc);
+    if(arity < 0 || arity > Integer.MAX_VALUE) throw FUNCUNKNOWN_X.get(ii, name);
 
     try {
       final Expr lit = Functions.getLiteral(name, (int) arity, qc, sc, ii);
