diff --git a/old_test.java b/new_test.java
index 776119e..b44a125 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
 	{
 		try {
 			// create the resettable Iterator
-			SpillingResettableIterator<PactInteger> iterator = new SpillingResettableIterator<PactInteger>(
+			SpillingResettableIterator<IntValue> iterator = new SpillingResettableIterator<IntValue>(
 					this.reader, this.serializer, this.memman, this.ioman, 2 * 32 * 1024, this.memOwner);
 			// open the iterator
 			try {
@@ -12,7 +12,7 @@
 				Assert.fail("Could not open resettable iterator:" + e.getMessage());
 			}
 	
-			PactInteger record;
+			IntValue record;
 			int cnt = 0;
 			while (cnt < NUM_TESTRECORDS) {
 				record = iterator.next();
