diff --git a/old_test.java b/new_test.java
index f95ae41..4551788 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void createClass()
     {
-	//TODO: rewrite the test to not rely on the breadcrumb based on parent/child mechanism.
+	    //TODO: rewrite the test to not rely on the breadcrumb based on parent/child mechanism.
         getUtil().setHierarchyMode("parentchild");
         try {
             String spaceName = getTestClassName();
@@ -13,7 +13,7 @@
             getUtil().deletePage(spaceName, pageName);
 
             // Create the class document.
-            DataTypesPage dataTypesPage = DataTypesPage.gotoPage();
+            DataTypesPage dataTypesPage = DataTypesPage.gotoPage().waitUntilPageIsLoaded();
             String dataTypesPageTitle = dataTypesPage.getDocumentTitle();
             Assert.assertTrue(dataTypesPage.isClassListed("XWiki", "XWikiRights"));
             Assert.assertFalse(dataTypesPage.isClassListed(spaceName, classDocName));
@@ -59,7 +59,7 @@
             editPage.clickBreadcrumbLink(classTitle);
 
             // Create a document based on the class template.
-            Assert.assertEquals(spaceName, classSheetPage.getSpaceNameInput().getAttribute("value"));
+            Assert.assertEquals(spaceName, classSheetPage.getNewPagePicker().getParentInput().getAttribute("value"));
             editPage = classSheetPage.createNewDocument(spaceName, pageName);
 
             Assert.assertEquals(pageName, editPage.getDocumentTitle());
