diff --git a/old_product.java b/new_product.java
index 2f21618..0dceacb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,8 @@
 public ClassSheetPage createClass(String spaceName, String className)
     {
-        spaceNameInput.clear();
-        spaceNameInput.sendKeys(spaceName);
-        classNameInput.clear();
-        classNameInput.sendKeys(className);
-        createClassButton.click();
+        this.documentPicker.setParent(spaceName);
+        this.documentPicker.setName(className);
+        this.createClassButton.click();
         new WikiEditPage().clickSaveAndView();
         return new ClassSheetPage();
     }
\ No newline at end of file
