diff --git a/old_test.java b/new_test.java
index d5e7424..ba32f62 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,8 +12,6 @@
     ArrayList<String> invalidPaths = new ArrayList<>();
     invalidPaths.add(null);
     invalidPaths.add("");
-    invalidPaths.add(" /");
-    invalidPaths.add("/ ");
     for (String invalidPath : invalidPaths) {
       try {
         PathUtils.validatePath(invalidPath);
