diff --git a/old_product.java b/new_product.java
index 9d5eab2..546526d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static void validatePath(String path) throws InvalidPathException {
-    boolean invalid = (path == null || path.isEmpty() || path.contains(" "));
+    boolean invalid = (path == null || path.isEmpty());
     if (!OSUtils.isWindows()) {
       invalid = (invalid || !path.startsWith(AlluxioURI.SEPARATOR));
     }
