diff --git a/old_test.java b/new_test.java
index 0414a5e..7577e06 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,9 @@
 		expectLastCall().once();
 		expect(mockConnection.getOriginatingPartition()).andReturn(mockPartition).anyTimes();
 		expect(mockPartition.getFreeConnections()).andReturn(mockConnectionHandles).anyTimes();
+		AtomicInteger ai = new AtomicInteger(1);
+		expect(mockPartition.getAvailableConnections()).andReturn(ai).anyTimes();
+		
 		expect(mockConnectionHandles.offer(mockConnection)).andReturn(false).anyTimes();
 		mockConnectionHandles.put(mockConnection);
 		expectLastCall().once();
