diff --git a/old_product.java b/new_product.java
index 897345b..848e36e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,7 +14,7 @@ protected void internalReleaseConnection(ConnectionHandle connectionHandle) thro
 
 		connectionHandle.setConnectionLastUsed(System.currentTimeMillis());
 		if (!this.poolShuttingDown){ 
-			releaseInAnyFreePartition(connectionHandle, connectionHandle.getOriginatingPartition());
+			putConnectionBackInPartition(connectionHandle);
 		} else {
 			connectionHandle.internalClose();
 		}
