diff --git a/old_product.java b/new_product.java
index 8795dec..45789cb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public DocumentCollection getDocumentCollection(Map<String, Object> docListParams) {
-        Request request = Request.Get(DocumentConversion.DOCUMENTS_PATH);
+        Request request = Request.Get(ConfigConstants.DOCUMENTS_PATH);
 
         if(docListParams != null) {
             if (docListParams.get(DocumentConversion.TOKEN) != null) {
