diff --git a/old_test.java b/new_test.java
index ec3c163..e5113f7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,7 +13,7 @@
 
         // Call get documents without query parameters
         mockServer.reset().when(request()
-                .withPath(DocumentConversion.DOCUMENTS_PATH))
+                .withPath(DOCUMENTS_PATH))
                 .respond(response().withBody(GsonSingleton.getGson().toJson(docCollWithQueryResponse))
                 );
         DocumentCollection docColl = service.getDocumentCollection();
@@ -25,7 +25,7 @@
         queryParams.add(new Parameter("name", docName));
         queryParams.add(new Parameter("limit", "5"));
         mockServer.reset().when(request()
-                .withPath(DocumentConversion.DOCUMENTS_PATH).withQueryStringParameters(queryParams))
+                .withPath(DOCUMENTS_PATH).withQueryStringParameters(queryParams))
                 .respond(response().withBody(GsonSingleton.getGson().toJson(docCollWithQueryResponse))
                 );
         Map<String, Object> docListParams = new HashMap<String, Object>();
@@ -36,7 +36,7 @@
         Assert.assertEquals(docCollWithQuery.toString(), docCollWithQueryResponse.toString());
 
         // Call get document
-        mockServer.when(request().withPath(DocumentConversion.DOCUMENTS_PATH + "/" + docId)).respond(
+        mockServer.when(request().withPath(DOCUMENTS_PATH + "/" + docId)).respond(
                 response().withBody(docContent)
         );
         String document1 = ConversionUtils.writeInputStreamToString(service.getDocument(docId));
