diff --git a/old_product.java b/new_product.java
index 2edab1f..0372eaa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
-@Authorized(PrivilegeConstants.GET_ORDERS)
-	public Order getOrderByOrderNumber(String orderNumber);
\ No newline at end of file
+@Authorized(PrivilegeConstants.PURGE_ORDER_TYPES)
+	public void purgeOrderType(OrderType orderType) throws APIException;
\ No newline at end of file
