diff --git a/old_test.java b/new_test.java
index ae0ad27..5cb1310 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
-	@Verifies(value = "should find object given valid order number", method = "getOrderByOrderNumber(String)")
-	public void getOrderByOrderNumber_shouldFindObjectGivenValidOrderNumber() throws Exception {
-		Order order = Context.getOrderService().getOrderByOrderNumber("1");
-		Assert.assertNotNull(order);
-		Assert.assertEquals(1, (int) order.getOrderId());
+	@Verifies(value = "should find object given valid uuid", method = "getOrderTypeByUuid(String)")
+	public void getOrderTypeByUuid_shouldFindObjectGivenValidUuid() throws Exception {
+		String uuid = "84ce45a8-5e7c-48f7-a581-ca1d17d63a62";
+		OrderType orderType = Context.getOrderService().getOrderTypeByUuid(uuid);
+		Assert.assertEquals(1, (int) orderType.getOrderTypeId());
 	}
\ No newline at end of file
