diff --git a/old_product.java b/new_product.java
index 1dbe1d5..5ce7ac2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -20,7 +20,9 @@ public void repairSurvivors()
                 tmLog.debug(m_whoami + "repairing: " + CoreUtils.hsIdCollectionToString(m_survivors) + " with: " + TxnEgo.txnIdToString(li.getTxnId()) +
                         " " + repairMsg);
             }
-            m_mailbox.repairReplicasWith(m_survivors, repairMsg);
+            if (repairMsg != null) {
+                m_mailbox.repairReplicasWith(m_survivors, repairMsg);
+            }
         }
 
         m_promotionResult.set(new RepairResult(m_maxSeenTxnId));
