diff --git a/old_test.java b/new_test.java
index 4f494c8..be3ae03 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
         masters.add(2L);
         masters.add(3L);
 
-        MpPromoteAlgo algo = new MpPromoteAlgo(masters, mailbox, "Test");
+        MpPromoteAlgo algo = new MpPromoteAlgo(masters, mailbox, 0, "Test");
         long requestId = algo.getRequestId();
         Future<RepairResult> result = algo.start();
         verify(mailbox, times(1)).send(any(long[].class), any(Iv2RepairLogRequestMessage.class));
