diff --git a/old_test.java b/new_test.java
index a4499c1..80ab7e6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -26,4 +26,16 @@
 		assertNull("invalidated valid name with 0x4E01", Verifier.checkAttributeName("test" + (char)0x4E01));
 		assertNull("invalidated valid name with 0x0301", Verifier.checkAttributeName("test" + (char)0x0301));
 
-	}
\ No newline at end of file
+    	assertNull(Verifier.checkAttributeName("hi"));
+    	assertNull(Verifier.checkAttributeName("hi2you"));
+    	assertNull(Verifier.checkAttributeName("hi_you"));
+    	
+    	assertNotNull(Verifier.checkAttributeName(null));
+    	assertNotNull(Verifier.checkAttributeName(""));
+    	assertNotNull(Verifier.checkAttributeName("   "));
+    	assertNotNull(Verifier.checkAttributeName("  hi  "));
+    	assertNotNull(Verifier.checkAttributeName("hi "));
+    	assertNotNull(Verifier.checkAttributeName(" hi"));
+    	assertNotNull(Verifier.checkAttributeName("2bad"));
+
+    }
\ No newline at end of file
