diff --git a/old_test.java b/new_test.java
index 3fc0bb5..1fd344f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,7 @@
                 = (DataCiteIdentifierProvider)
                 sm.getServicesByType(DataCiteIdentifierProvider.class).get(0);
 
-        DSpaceObject dso = item;
+        DSpaceObject dso = newItem(context);
         String result = instance.mint(context, dso);
         assertNotNull("Null returned", result);
     }
\ No newline at end of file
