diff --git a/old_product.java b/new_product.java
index 5919c8a..60de064 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,15 +1,15 @@
 public EZIDResponse mint(Map<String, String> metadata)
-            throws IOException, IdentifierException
+            throws IOException, IdentifierException, URISyntaxException
     {
         // POST path [+metadata]
         HttpPost request;
-        request = new HttpPost(url);
+        URI uri = new URI(scheme, host, SHOULDER_PATH + authority, null);
+        request = new HttpPost(uri);
         if (null != metadata)
         {
-            request.setEntity(new StringEntity(formatMetadata(metadata), "UTF-8"));
+            request.setEntity(new StringEntity(formatMetadata(metadata), UTF_8));
         }
         HttpResponse response = client.execute(request);
         EZIDResponse myResponse = new EZIDResponse(response);
-        // TODO add the identifier to the path for subsequent operations?
         return myResponse;
     }
\ No newline at end of file
