diff --git a/old_test.java b/new_test.java
index 4e7e1eb..3bf0090 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
 		DslConverter<?, ?> compatibleDslConverter =
 				index2FieldSchemaNode.getConverter().getDslConverter();
 		DslConverter<?, ?> incompatibleDslConverter =
-				new DslConverter<>( new IncompatibleToDocumentFieldValueConverter() );
+				new DslConverter<>( typeDescriptor.getJavaType(), new IncompatibleToDocumentFieldValueConverter<>() );
 		ToDocumentFieldValueConvertContext toDocumentConvertContext =
 				new ToDocumentFieldValueConvertContextImpl( new JavaBeanBackendMappingContext() );
 
