diff --git a/old_product.java b/new_product.java
index aff6bd3..ed26f63 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-static ChronoLocalDate<?> from(TemporalAccessor temporal) {
+static ChronoLocalDate from(TemporalAccessor temporal) {
         if (temporal instanceof ChronoLocalDate) {
-            return (ChronoLocalDate<?>) temporal;
+            return (ChronoLocalDate) temporal;
         }
         Chronology chrono = temporal.query(TemporalQuery.chronology());
         if (chrono == null) {
