diff --git a/old_product.java b/new_product.java
index 7214b48..129754b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-protected static String post(final String query, final String request,
-      final String type) throws IOException {
+static String post(final String query, final String request,
+                     final String type) throws IOException {
 
     // create connection
     final URL url = new URL(root + query);
