diff --git a/old_test.java b/new_test.java
index 8fbc3b8..6080bc0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,13 +10,13 @@
 			RefinePolygonLineToImage alg = createAlg(input.size(),black, imageType);
 
 			Polygon2D_F64 output = new Polygon2D_F64(4);
-			alg.initialize(image);
+			alg.setImage(image);
 			assertTrue(alg.refine(input, output));
 
 			// do it again with a sub-image
 			Polygon2D_F64 output2 = new Polygon2D_F64(4);
 			image = BoofTesting.createSubImageOf_S(image);
-			alg.initialize(image);
+			alg.setImage(image);
 			assertTrue(alg.refine(input, output2));
 
 			assertTrue(UtilPolygons2D_F64.isIdentical(output, output2, 1e-8));
