diff --git a/old_product.java b/new_product.java
index 0948dbc..3100d80 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,8 +16,8 @@ public boolean fit( List<Point2D_I32> contour , GrowQueue_I32 corners )
 		for( int iteration = 0; iteration < maxIterations && change; iteration++ ) {
 			change = false;
 			for (int i = startCorner; i < endCorner; i++) {
-				int c0 = UtilShapePolygon.minus(i, 1, corners.size());
-				int c2 = UtilShapePolygon.plus(i,1,corners.size());
+				int c0 = UtilShapePolygon.minusPOffset(i, 1, corners.size());
+				int c2 = UtilShapePolygon.plusPOffset(i, 1, corners.size());
 
 				int improved = optimize(contour, corners.get(c0), corners.get(i), corners.get(c2));
 				if( improved != corners.get(i)) {
