diff --git a/old_test.java b/new_test.java
index aad66b3..0c63a6d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
         userStore.put(SyncTestUtils.createTestUser(Long.MIN_VALUE));
         userStore.put(SyncTestUtils.createTestUser(Long.MAX_VALUE));
 
-        Collection<SyncUser> users = SyncUser.all();
+        Map<String, SyncUser> users = SyncUser.all();
         assertEquals(1, users.size());
-        assertTrue(users.iterator().next().isValid());
+        assertTrue(users.get(users.keySet().iterator().next()).isValid());
     }
\ No newline at end of file
