diff --git a/old_product.java b/new_product.java
index cfadf72..3a0afd0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,11 @@
-public static Collection<SyncUser> all() {
+public static Map<String, SyncUser> all() {
         UserStore userStore = SyncManager.getUserStore();
         Collection<SyncUser> storedUsers = userStore.allUsers();
-        List<SyncUser> result = new ArrayList<SyncUser>(storedUsers.size());
+        Map<String, SyncUser> map = new HashMap<String, SyncUser>();
         for (SyncUser user : storedUsers) {
             if (user.isValid()) {
-                result.add(user);
+                map.put(user.getIdentity(), user);
             }
         }
-        return result;
+        return Collections.unmodifiableMap(map);
     }
\ No newline at end of file
