diff --git a/old_test.java b/new_test.java
index 558a8dc..80aed81 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,8 @@
     for (int i = 0; i < 10; i++) {
       mFileSystem.createDirectory(new AlluxioURI("/i" + i));
       for (int j = 0; j < 10; j++) {
-        CreateFileOptions option = CreateFileOptions.defaults().setBlockSizeBytes((i + j + 1) * 64);
+        CreateFilePOptions option =
+            CreateFilePOptions.newBuilder().setBlockSizeBytes((i + j + 1) * 64).build();
         AlluxioURI path = new AlluxioURI("/i" + i + "/j" + j);
         mFileSystem.createFile(path, option).close();
         mFileSystem.rename(path, new AlluxioURI("/i" + i + "/jj" + j));
