diff --git a/old_test.java b/new_test.java
index 3e83dfe..13fd708 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,9 @@
 @Test
 	@Verifies(value = "should return default location for the implementation", method = "getDefaultLocation()")
 	public void getDefaultLocation_shouldReturnDefaultLocationForTheImplementation() throws Exception {
-		Assert.assertNotNull(Context.getLocationService().getDefaultLocation());
+		//set the global property for default location to something other than Unknown Location
+		GlobalProperty gp = new GlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_DEFAULT_LOCATION_NAME,
+		        "Test Parent Location", "Testing default Location");
+		Context.getAdministrationService().saveGlobalProperty(gp);
+		Assert.assertEquals("Test Parent Location", Context.getLocationService().getDefaultLocation().getName());
 	}
\ No newline at end of file
