diff --git a/old_product.java b/new_product.java
index 71b9d2e..40574fa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,17 @@
 public Location getDefaultLocation() throws APIException {
+		Location location = null;
+		String locationGP = Context.getAdministrationService().getGlobalProperty(
+		    OpenmrsConstants.GLOBAL_PROPERTY_DEFAULT_LOCATION_NAME);
 		
-		// TODO The name of the default location should be configured using global properties 
-		Location location = getLocation("Unknown Location");
+		if (StringUtils.hasText(locationGP))
+			location = getLocation(locationGP);
 		
-		// If Unknown Location does not exist, try Unknown
-		if (location == null) {
+		//Try to look up 'Unknown Location' in case the global property is something else
+		if (location == null && (!StringUtils.hasText(locationGP) || !locationGP.equalsIgnoreCase("Unknown Location")))
+			location = getLocation("Unknown Location");
+		
+		// If Unknown Location does not exist, try Unknown if the global property was different
+		if (location == null && (!StringUtils.hasText(locationGP) || !locationGP.equalsIgnoreCase("Unknown"))) {
 			location = getLocation("Unknown");
 		}
 		
