diff --git a/old_product.java b/new_product.java
index 36bcf91..2013b44 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,11 @@
-private void timeoutRecords() {
-		Long currentTime = System.currentTimeMillis();
+List<String> timeoutRecords(Long currentTime) {
 
 		if (timeOfLastUpdate + TIMEOUT < currentTime) {
 
 			List<String> timedOutRecords = new LinkedList<String>();
 			Map<Long, Set<String>> timedOut = recordsByTime.subMap(0L, currentTime
 					- TIMEOUT);
+
 			for (Set<String> recordSet : timedOut.values()) {
 				if (!recordSet.isEmpty()) {
 					for (String recordID : recordSet) {
@@ -18,5 +18,10 @@ private void timeoutRecords() {
 			for (String recordID : timedOutRecords) {
 				failRecord(recordID);
 			}
+
+			timeOfLastUpdate = currentTime;
+			return timedOutRecords;
 		}
+
+		return null;
 	}
\ No newline at end of file
