diff --git a/old_test.java b/new_test.java
index 0ca1a6c..f602d37 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testInitialize() throws Exception {
         final TestCallable call = new TestCallable();
-        final CallableBackgroundInitializer<Integer> init = new CallableBackgroundInitializer<Integer>(
+        final CallableBackgroundInitializer<Integer> init = new CallableBackgroundInitializer<>(
                 call);
         assertEquals("Wrong result", RESULT, init.initialize());
         assertEquals("Wrong number of invocations", 1, call.callCount);
