diff --git a/old_product.java b/new_product.java
index 3609511..2391c62 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@
         Map<String, BackgroundInitializer<?>> inits;
         synchronized (this) {
             // create a snapshot to operate on
-            inits = new HashMap<String, BackgroundInitializer<?>>(
+            inits = new HashMap<>(
                     childInitializers);
         }
 
@@ -18,8 +18,8 @@
         }
 
         // collect the results
-        final Map<String, Object> results = new HashMap<String, Object>();
-        final Map<String, ConcurrentException> excepts = new HashMap<String, ConcurrentException>();
+        final Map<String, Object> results = new HashMap<>();
+        final Map<String, ConcurrentException> excepts = new HashMap<>();
         for (final Map.Entry<String, BackgroundInitializer<?>> e : inits.entrySet()) {
             try {
                 results.put(e.getKey(), e.getValue().get());
