diff --git a/old_product.java b/new_product.java
index 2d060d2..35047c7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,7 @@ public LinkedHashMap<String, List<AlignmentInterval.Row>> packAlignments(
             int end,
             boolean pairAlignments,
             AlignmentTrack.GroupOption groupBy,
+            String tag,
             int maxLevels) {
 
         LinkedHashMap<String, List<AlignmentInterval.Row>> packedAlignments = new LinkedHashMap<String, List<Row>>();
@@ -21,7 +22,7 @@ public LinkedHashMap<String, List<AlignmentInterval.Row>> packAlignments(
             HashMap<String, List<Alignment>> groupedAlignments = new HashMap();
             while (iter.hasNext()) {
                 Alignment alignment = iter.next();
-                String groupKey = getGroupValue(alignment, groupBy);
+                String groupKey = getGroupValue(alignment, groupBy, tag);
                 if (groupKey == null) nullGroup.add(alignment);
                 else {
                     List<Alignment> group = groupedAlignments.get(groupKey);
