diff --git a/old_product.java b/new_product.java
index e5755f7..0df686a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,6 +7,7 @@
 		newOrder.setPreviousOrder(this);
 		newOrder.setPatient(this.getPatient());
 		newOrder.setDrug(this.getDrug());
+		newOrder.setOrderType(this.getOrderType());
 		newOrder.setStartDate(this.getStartDate());
 		return newOrder;
 	}
\ No newline at end of file
