diff --git a/old_test.java b/new_test.java
index ad5ce21..bb30062 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,7 @@
 		anOrder.setPatient(new Patient());
 		anOrder.setCareSetting(new CareSetting());
 		anOrder.setConcept(new Concept());
+		anOrder.setOrderType(new OrderType());
 		
 		Order orderThatCanDiscontinueTheOrder = anOrder.cloneForDiscontinuing();
 		
@@ -18,4 +19,6 @@
 		    Order.Action.DISCONTINUE);
 		
 		assertEquals(anOrder.getCareSetting(), orderThatCanDiscontinueTheOrder.getCareSetting());
+		
+		assertEquals(anOrder.getOrderType(), orderThatCanDiscontinueTheOrder.getOrderType());
 	}
\ No newline at end of file
