diff --git a/old_test.java b/new_test.java
index 3913ee3..d08ee53 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,8 +16,8 @@
 		String filePath = conf.get("dfs.result.dir") + "/model-"
 				+ DriverClassUtil.getDriverName(UserKNNRecommender.class);
 		recommender.loadModel(filePath);
-		recommender.recommend(context);
-		List<RecommendedItem> recommendedItemList = recommender.getRecommendedList();
-		RecommendedFilter filter = new GenericRecommendedFilter();
-		recommendedItemList = filter.filter(recommendedItemList);
+//		recommender.recommend(context); // TODO: 21/04/2018
+//		List<RecommendedItem> recommendedItemList = recommender.getRecommendedList();
+//		RecommendedFilter filter = new GenericRecommendedFilter();
+//		recommendedItemList = filter.filter(recommendedItemList);
 	}
\ No newline at end of file
