diff --git a/old_product.java b/new_product.java
index a2354d8..12debd4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -21,15 +21,8 @@ public void validate(Object obj, Errors errors) {
 				}
 			}
 			if (order.getAction() != Order.Action.DISCONTINUE && order.getDosingType() != null) {
-				if (order.getDosingType().equals(DrugOrder.DosingType.SIMPLE)) {
-					ValidationUtils.rejectIfEmpty(errors, "dose", "DrugOrder.error.doseIsNullForDosingTypeSimple");
-					ValidationUtils.rejectIfEmpty(errors, "doseUnits", "DrugOrder.error.doseUnitsIsNullForDosingTypeSimple");
-					ValidationUtils.rejectIfEmpty(errors, "route", "DrugOrder.error.routeIsNullForDosingTypeSimple");
-					ValidationUtils.rejectIfEmpty(errors, "frequency", "DrugOrder.error.frequencyIsNullForDosingTypeSimple");
-				} else {
-					ValidationUtils.rejectIfEmpty(errors, "dosingInstructions",
-					    "DrugOrder.error.dosingInstructionsIsNullForDosingTypeOther");
-				}
+				DosingInstructions dosingInstructions = order.getDosingInstructionsInstance();
+				dosingInstructions.validate(order, errors);
 			}
 			validateFieldsForOutpatientCareSettingType(order, errors);
 			validatePairedFields(order, errors);
