diff --git a/old_test.java b/new_test.java
index b9af052..6512088 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
-	public void validate_shouldFailValidationIfDoseUnitsIsNullForSIMPLEDosingType() throws Exception {
+	public void validate_shouldFailValidationIfDoseUnitsIsNullForSimpleDosingInstructionsDosingType() throws Exception {
 		DrugOrder order = new DrugOrder();
-		order.setDosingType(DrugOrder.DosingType.SIMPLE);
+		order.setDosingType(SimpleDosingInstructions.class);
 		order.setDoseUnits(null);
 		Errors errors = new BindException(order, "order");
 		new DrugOrderValidator().validate(order, errors);
