diff --git a/old_test.java b/new_test.java
index 0d88069..b94bfe0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,14 @@
 @Override
   @SuppressWarnings("unchecked")
   public Description matchMethod(MethodTree methodTree, VisitorState state) {
-    if (!allOf(methodNameStartsWith("test"),
+    boolean matches = allOf(
+        methodNameStartsWith("test"),
         methodHasParameters(),
         methodHasModifier(Modifier.PUBLIC),
         not(hasJUnitAnnotation),
         enclosingClass(isJUnit4TestClass))
-        .matches(methodTree, state)) {
+        .matches(methodTree, state);
+    if (!matches) {
       return Description.NO_MATCH;
     }
 
