diff --git a/old_product.java b/new_product.java
index 88d5eef..67684a2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,7 @@
 @Override
   @SuppressWarnings("unchecked")    // matchers + varargs cause this
   public Description matchMethod(MethodTree methodTree, VisitorState state) {
-    if (!allOf(
-        methodHasVisibility(Visibility.PUBLIC),
-        methodIsNamed("equals"),
-        methodReturns(state.getSymtab().booleanType),
-        methodHasParameters(variableType(isSameType(findEnclosingClass(state)))),
-        enclosingClass(not(hasMethod(Matchers.<MethodTree>allOf(
-            methodIsNamed("equals"),
-            methodReturns(state.getSymtab().booleanType),
-            methodHasParameters(variableType(isSameType(state.getSymtab().objectType)))))))
-    ).matches(methodTree, state)) {
+    if (!MATCHER.matches(methodTree, state)) {
       return Description.NO_MATCH;
     }
 
