diff --git a/old_product.java b/new_product.java
index 4269c76..0e7519d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,7 +14,7 @@ private ANode entries(final QueryContext ctx) throws QueryException {
       createEntries(paths(zf).iterator(), root, "");
       return root;
     } catch(final IOException ex) {
-      throw ZIP_FAIL.thrw(info, ex.getMessage());
+      throw ZIP_FAIL.thrw(info, ex);
     } finally {
       if(zf != null) try { zf.close(); } catch(final IOException e) { }
     }
