diff --git a/old_test.java b/new_test.java
index f8bfec3..5d0df9e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testAAT() {
         System.out.println("AAT");
-        SparseMatrix c = SparseMatrix.AAT(sm, sm.transpose());
+        SparseMatrix c = sm.aat();
         assertEquals(c.nrows(), 3);
         assertEquals(c.ncols(), 3);
         assertEquals(c.size(), 9);
