diff --git a/old_test.java b/new_test.java
index 78615e3..6d6b06f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,23 +10,23 @@
 		Pageable page = new PageRequest(0, 10);
 
 		// search
-		userController.getUsers(model, null, page, "NewUserName");
+		userController.getAll(model, null, page, "NewUserName");
 		PageImpl userList = (PageImpl<User>) model.get("users");
 		assertThat(userList.getContent().size(), is(3));
 
 		// test to delete one
 		model.clear();
-		userController.deleteUser(model, "NewUserId1");
+		userController.delete(model, "NewUserId1");
 		model.clear();
-		userController.getUsers(model, Role.USER, page, "NewUserName");
+		userController.getAll(model, Role.USER, page, "NewUserName");
 		userList = (PageImpl<User>) model.get("users");
 		assertThat(userList.getContent().size(), is(2));
 
 		// test to delete more
 		model.clear();
-		userController.deleteUser(model, "NewUserId2,NewUserId3");
+		userController.delete(model, "NewUserId2,NewUserId3");
 		model.clear();
-		userController.getUsers(model, Role.USER, page, "NewUserName");
+		userController.getAll(model, Role.USER, page, "NewUserName");
 		userList = (PageImpl<User>) model.get("users");
 		assertThat(userList.getContent().size(), is(0));
 	}
\ No newline at end of file
