diff --git a/old_test.java b/new_test.java
index a8378c5..51e0759 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
-	public void testSupportsExtension() {
+	public void testSupportsFileExtension() {
 		assertTrue("Should support \"ac\" extension.",
-				analyzer.supportsExtension("ac"));
+				analyzer.accept(new File("configure.ac")));
 		assertTrue("Should support \"in\" extension.",
-				analyzer.supportsExtension("in"));
+				analyzer.accept(new File("configure.in")));
 		assertTrue("Should support \"configure\" extension.",
-				analyzer.supportsExtension("configure"));
+				analyzer.accept(new File("configure")));
 	}
\ No newline at end of file
