diff --git a/old_product.java b/new_product.java
index 6880a31..559bdc5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,12 @@
-public boolean supportsExtension(String ext) {
-        if (ext == null) {
+public boolean accept(File file) {
+        if (file == null) {
             return false;
         }
         boolean scan = false;
         for (FileTypeAnalyzer a : this.fileTypeAnalyzers) {
             /* note, we can't break early on this loop as the analyzers need to know if
              they have files to work on prior to initialization */
-            scan |= a.supportsExtension(ext);
+            scan |= a.accept(file);
         }
         return scan;
     }
\ No newline at end of file
