diff --git a/old_test.java b/new_test.java
index b33e6db..ed91207 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,7 @@
     Cluster cluster = createMock(Cluster.class);
 
     // requests
-    HostRequest request1 = new HostRequest("host1", "cluster1", Collections.<String, String>emptyMap());
+    HostRequest request1 = new HostRequest("host1", "cluster1");
     Set<HostRequest> setRequests = Collections.singleton(request1);
 
     // expectations
