diff --git a/old_product.java b/new_product.java
index c7a4ac2..88dbc2c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-TestResult full() throws IOException {
+TestResult full(final Object[] exp) throws IOException {
       final byte[] result = ((TestSession) cs).exec(ServerCmd.FULL, id);
       final TestResult tr = new TestResult();
       tr.type = result[0];
@@ -11,8 +11,11 @@ TestResult full() throws IOException {
         tr.result = Arrays.copyOfRange(result, 1, rl);
       }
 
-      final boolean uriResult = tr.uri != null, uriExpected = tr.uri != null;
-      if(uriResult && !uriExpected) fail("No URI expected for " + TYPES[tr.type][0]);
-      if(!uriResult && uriExpected) fail("URI expected for " + TYPES[tr.type][0]);
+      final boolean expected = exp.length > 3;
+      if(tr.uri == null) {
+        if(expected) fail("URI expected for " + TYPES[tr.type][0]);
+      } else if(!expected) {
+        fail("No URI expected for " + TYPES[tr.type][0] + ": " + Token.string(tr.result));
+      }
       return tr;
     }
\ No newline at end of file
