diff --git a/old_test.java b/new_test.java
index bc8436e..a01c621 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     for(final Object[] exp : TYPES) {
       if(exp.length < 2) continue;
       try(TestQuery tq = session.query(exp[1].toString())) {
-        final TestResult tr = tq.full();
+        final TestResult tr = tq.full(exp);
         final Object[] type = TYPES[tr.type];
         assertSame("Types are different.\nExpected: " + exp[0] + "\nFound: " + type[0], exp, type);
         assertEquals(Token.string(tr.result), type[2]);
