diff --git a/old_test.java b/new_test.java
index 4332fa7..ca69f39 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,11 +15,11 @@
         EigenValueDecomposition result = Math.eigen(A, 3);
         assertTrue(Math.equals(eigenValues, result.getEigenValues(), 1E-7));
 
-        assertEquals(eigenVectors.length, result.getEigenVectors().length);
-        assertEquals(eigenVectors[0].length, result.getEigenVectors()[0].length);
+        assertEquals(eigenVectors.length, result.getEigenVectors().nrows());
+        assertEquals(eigenVectors[0].length, result.getEigenVectors().ncols());
         for (int i = 0; i < eigenVectors.length; i++) {
             for (int j = 0; j < eigenVectors[i].length; j++) {
-                assertEquals(Math.abs(eigenVectors[i][j]), Math.abs(result.getEigenVectors()[i][j]), 1E-7);
+                assertEquals(Math.abs(eigenVectors[i][j]), Math.abs(result.getEigenVectors().get(i, j)), 1E-7);
             }
         }
     }
\ No newline at end of file
