diff --git a/old_product.java b/new_product.java
index eef72ce..7a527af 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,8 +2,8 @@ public static <T extends Serializable> T clone(final T object) {
         if (object == null) {
             return null;
         }
-        byte[] objectData = serialize(object);
-        ByteArrayInputStream bais = new ByteArrayInputStream(objectData);
+        final byte[] objectData = serialize(object);
+        final ByteArrayInputStream bais = new ByteArrayInputStream(objectData);
 
         ClassLoaderAwareObjectInputStream in = null;
         try {
@@ -15,19 +15,20 @@ public static <T extends Serializable> T clone(final T object) {
              * is of the same type as the original serialized object
              */
             @SuppressWarnings("unchecked") // see above
+            final
             T readObject = (T) in.readObject();
             return readObject;
 
-        } catch (ClassNotFoundException ex) {
+        } catch (final ClassNotFoundException ex) {
             throw new SerializationException("ClassNotFoundException while reading cloned object data", ex);
-        } catch (IOException ex) {
+        } catch (final IOException ex) {
             throw new SerializationException("IOException while reading cloned object data", ex);
         } finally {
             try {
                 if (in != null) {
                     in.close();
                 }
-            } catch (IOException ex) {
+            } catch (final IOException ex) {
                 throw new SerializationException("IOException on closing cloned object data InputStream.", ex);
             }
         }
