diff --git a/old_test.java b/new_test.java
index 3801f40..7a7f417 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
         assertTrue(Arrays.equals(original1, cloned1));
         assertTrue(original1 != cloned1);
         
-        StringBuffer buf = new StringBuffer("pick");
+        final StringBuffer buf = new StringBuffer("pick");
         original1 = new Object[] {buf, "a", new String[] {"stick"}};
         cloned1 = ArrayUtils.clone(original1);
         assertTrue(Arrays.equals(original1, cloned1));
