diff --git a/old_test.java b/new_test.java
index 014b335..4c6cb19 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,12 +8,13 @@
 
 		field = testClass.getClass().getDeclaredField("isolation");
 		field.setAccessible(true);
-		field.set(testClass, 0);
+		field.set(testClass, 8);
 
 
 		expect(mockPool.getConnection()).andReturn(mockConnection).once();
 		expect(mockConnection.getAutoCommit()).andReturn(false).once();
-		mockConnection.setTransactionIsolation(0);
+		expect(mockConnection.getTransactionIsolation()).andReturn(0).once();
+		mockConnection.setTransactionIsolation(8);
 		expectLastCall().once();
 		replay(mockPool, mockConnection);
 		testClass.getConnection();
