diff --git a/old_product.java b/new_product.java
index 905987a..26a00f4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,15 +3,27 @@ public Connection getConnection()
 		Connection connection = this.pool.getConnection();
 
 		// set the Transaction Isolation if defined
-		if (this.isolation !=null){
-			connection.setTransactionIsolation( this.isolation.intValue() );
-		}
-
-		// toggle autoCommit to false if set
-		if ( connection.getAutoCommit() != this.autocommit ){
-			connection.setAutoCommit(this.autocommit);
-		}
-		return connection;
+		boolean success = false;
+		try {
+			// set the Transaction Isolation if defined
+			if ((this.isolation != null) && (connection.getTransactionIsolation() != this.isolation.intValue())) {
+				connection.setTransactionIsolation (this.isolation.intValue());
+			}
 
+			// toggle autoCommit to false if set
+			if ( connection.getAutoCommit() != this.autocommit ){
+				connection.setAutoCommit(this.autocommit);
+			}
 
+			success = true;
+			return connection;
+		} finally {
+			if (!success) {
+				try {
+					connection.close();
+				} catch (Exception e) {
+					logger.warn("Failed to close a connection", e);
+				}
+			}
+		}
 	}
\ No newline at end of file
