diff --git a/old_test.java b/new_test.java
index 745c9ca..3f9c252 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
-    public void test_atStartOfDayInZone() {
+    public void test_atStartOfDay() {
         OffsetDate t = OffsetDate.of(2008, 6, 30, OFFSET_PTWO);
-        assertEquals(t.atStartOfDayInZone(ZONE_PARIS),
+        assertEquals(t.atStartOfDay(ZONE_PARIS),
                 ZonedDateTime.of(LocalDateTime.of(2008, 6, 30, 0, 0), ZONE_PARIS));
     }
\ No newline at end of file
