diff --git a/old_product.java b/new_product.java
index 2b89a53..ecb5fe0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1,3 @@
-public void register(String name, BooleanSupplier supplier, @Nullable String description);
\ No newline at end of file
+public void register(String name, BooleanSupplier supplier, @Nullable String desc) {
+        addMetric(name, new BooleanGauge(metricName(grpName, name), desc, nonThrowableSupplier(supplier, log)));
+    }
\ No newline at end of file
