diff --git a/old_test.java b/new_test.java
index 30f7a55..c7cc21c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -22,8 +22,14 @@
                     put("B", 0);
                 }
             }},
-            (k, v) -> v
-        );
+            (k, v) -> v,
+            new HashSet() {
+                {
+                    add(0);
+                    add(1);
+                    add(2);
+                }
+            });
 
         double[][] postProcessedData = new double[][]{
             {1.0, 0.0, 1.0},
