diff --git a/old_product.java b/new_product.java
index 1ce71f6..4ac8559 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-@Override public Double apply(double[] features) {
+@Override public Double apply(Vector features) {
             double[] newFeatures = new double[featuresMapping.size()];
-            featuresMapping.forEach((localId, featureVectorId) -> newFeatures[localId] = features[featureVectorId]);
-            return model.apply(newFeatures);
+            featuresMapping.forEach((localId, featureVectorId) -> newFeatures[localId] = features.get(featureVectorId));
+            return mdl.apply(new DenseLocalOnHeapVector(newFeatures));
         }
\ No newline at end of file
