diff --git a/old_product.java b/new_product.java
index 540a7a9..1f5b308 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,14 +9,12 @@ private MathTransformProvider getProvider(final String method) throws NoSuchIden
         if (provider!=null && provider.nameMatches(method)) {
             return provider;
         }
-        final Iterator<MathTransformProvider> providers =
-                registry.getServiceProviders(MathTransformProvider.class, null, HINTS);
-        while (providers.hasNext()) {
-            provider = providers.next();
-            if (provider.nameMatches(method)) {
-                return lastProvider = provider;
-            }
-        }
-        throw new NoSuchIdentifierException(Errors.format(
-                  ErrorKeys.NO_TRANSFORM_FOR_CLASSIFICATION_$1, method), method);
+
+        provider = registry.getFactories(MathTransformProvider.class, null, HINTS)
+                .filter(prov -> prov.nameMatches(method))
+                .findAny()
+                .orElseThrow(() -> new NoSuchIdentifierException(Errors
+                        .format(ErrorKeys.NO_TRANSFORM_FOR_CLASSIFICATION_$1, method), method));
+
+        return lastProvider = provider;
     }
\ No newline at end of file
