diff --git a/old_test.java b/new_test.java
index 07a5364..7cfd8a6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,18 +3,18 @@
     {
         list.add(asArray(0, 1, 0, 1, 0));
 
-        assertEquals(0, list.removeAllOccurrences(k2));
-        assertEquals(3, list.removeAllOccurrences(k0));
+        assertEquals(0, list.removeAll(k2));
+        assertEquals(3, list.removeAll(k0));
         assertListEquals(list.toArray(), 1, 1);
 
-        assertEquals(2, list.removeAllOccurrences(k1));
+        assertEquals(2, list.removeAll(k1));
         assertTrue(list.isEmpty());
 
         /*! #if ($TemplateOptions.KTypeGeneric) !*/
         list.clear();
         list.add(newArray(k0, null, k2, null, k0));
-        assertEquals(2, list.removeAllOccurrences(null));
-        assertEquals(0, list.removeAllOccurrences(null));
+        assertEquals(2, list.removeAll((KType) null));
+        assertEquals(0, list.removeAll((KType) null));
         assertListEquals(list.toArray(), 0, 2, 0);
         /*! #end !*/
     }
\ No newline at end of file
