diff --git a/old_test.java b/new_test.java
index 1ac94cc..c8d8b9f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
     public void isContainIgnoreCase(){
-        LOGGER.info(StringUtil.isContainIgnoreCase(null, "") + "");
-        LOGGER.info(StringUtil.isContainIgnoreCase(TEXT, null) + "");
-        LOGGER.info(StringUtil.isContainIgnoreCase(TEXT, "") + "");
-        LOGGER.info(StringUtil.isContainIgnoreCase(TEXT, "feilong") + "");
-        LOGGER.info(StringUtil.isContainIgnoreCase(TEXT, "feilong1") + "");
-        LOGGER.info(StringUtil.isContainIgnoreCase(TEXT, "feiLong") + "");
+        LOGGER.info(StringUtil.containsIgnoreCase(null, "") + "");
+        LOGGER.info(StringUtil.containsIgnoreCase(TEXT, null) + "");
+        LOGGER.info(StringUtil.containsIgnoreCase(TEXT, "") + "");
+        LOGGER.info(StringUtil.containsIgnoreCase(TEXT, "feilong") + "");
+        LOGGER.info(StringUtil.containsIgnoreCase(TEXT, "feilong1") + "");
+        LOGGER.info(StringUtil.containsIgnoreCase(TEXT, "feiLong") + "");
     }
\ No newline at end of file
