diff --git a/old_product.java b/new_product.java
index 9a6a6c2..6cf4f63 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,3 @@
-public static boolean isContainIgnoreCase(Object text,String beIncludedString){
-        if (null == text){
-            LOGGER.warn("the param [text] is null,default return false");
-            return false;
-        }
-        if (null == beIncludedString){
-            LOGGER.warn("the param [beIncludedString] is null,default return false");
-            return false;
-        }
-        return isContain(text.toString().toLowerCase(), beIncludedString.toLowerCase());
+public static boolean containsIgnoreCase(final CharSequence str,final CharSequence searchStr){
+        return org.apache.commons.lang3.StringUtils.containsIgnoreCase(str, searchStr);
     }
\ No newline at end of file
