diff --git a/old_test.java b/new_test.java
index 71cdb94..a9181dc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testDeriveSecret() {
+    public void testDeriveSecret() throws CryptoException {
         HKDFAlgorithm hkdfAlgorithm = HKDFAlgorithm.TLS_HKDF_SHA256;
         String hashAlgorithm = DigestAlgorithm.SHA256.getJavaName();
         byte[] prk = ArrayConverter
