diff --git a/old_test.java b/new_test.java
index 1927713..a14cf21 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,6 +7,6 @@
                .build();
 
       // The method under test
-      Task relocate = vmApi.relocate(vm.getHref(), params);
+      Task relocate = vmApi.relocate(vmUrn, params);
       assertTrue(retryTaskSuccess.apply(relocate), String.format(TASK_COMPLETE_TIMELY, "relocate"));
    }
\ No newline at end of file
