diff --git a/old_test.java b/new_test.java
index 9f3c879..a2022d2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
 			// XYZ
 			Patient p = patientService.getPatient(999);
 			Assert.assertNotNull(p);
-			Assert.assertTrue(p.isVoided());
+			Assert.assertTrue(p.getVoided());
 			boolean found = false;
 			for (PatientIdentifier id : p.getIdentifiers()) {
 				if (id.getIdentifier().equals("XYZ") && id.getIdentifierType().getId() == 2) {
