diff --git a/old_product.java b/new_product.java
index 5e54f65..60df792 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,7 +9,6 @@ private void activate(String nodeCipherSuite, String utilityCipherSuite, int exp
 
         List<String> params = new ArrayList<>();
 
-        params.add("--activate");
         params.add("--keystore");
         params.add(GridTestUtils.keyStorePath("node01"));
         params.add("--keystore-password");
@@ -20,6 +19,8 @@ private void activate(String nodeCipherSuite, String utilityCipherSuite, int exp
             params.add(utilityCipherSuite);
         }
 
+        params.add("--activate");
+
         assertEquals(expRes, execute(params));
 
         if (expRes == EXIT_CODE_OK)
