diff --git a/old_product.java b/new_product.java
index 43123f4..b105444 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public CreatePathResult createPath(AlluxioURI path, CreatePathOptions options)
+public CreatePathResult createPath(AlluxioURI path, CreatePathOptions<?> options)
       throws FileAlreadyExistsException, BlockInfoException, InvalidPathException, IOException {
     if (path.isRoot()) {
       LOG.info(ExceptionMessage.FILE_ALREADY_EXISTS.getMessage(path));
