diff --git a/old_test.java b/new_test.java
index 3403a56..cc77007 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,8 +8,8 @@
     // create nested directory
     InodeTree.CreatePathResult createResult =
         mTree.createPath(NESTED_URI, sNestedDirectoryOptions);
-    List<Inode> modified = createResult.getModified();
-    List<Inode> created = createResult.getCreated();
+    List<Inode<?>> modified = createResult.getModified();
+    List<Inode<?>> created = createResult.getCreated();
     // 1 modified directory
     Assert.assertEquals(1, modified.size());
     Assert.assertEquals("", modified.get(0).getName());
