diff --git a/old_test.java b/new_test.java
index 0f1ca82..b3b116e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
 	@Verifies(value = "should fail if name is a duplicate", method = "validate(Object,Errors)")
 	public void validate_shouldFailIfNameIsADuplicate() throws Exception {
 		OrderType orderType = new OrderType();
-		orderType.setName("Drug order");
+		orderType.setName(orderService.getOrderType(1).getName());
 		Errors errors = new BindException(orderType, "orderType");
 		new OrderTypeValidator().validate(orderType, errors);
 		Assert.assertEquals(true, errors.hasFieldErrors("name"));
