diff --git a/old_product.java b/new_product.java
index 7e70081..b4f8ad4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 private Str pathToNative(final QueryContext qc) throws QueryException, IOException {
-    final Path nat = checkPath(0, qc).toRealPath();
+    final Path nat = toPath(0, qc).toRealPath();
     return get(nat, Files.isDirectory(nat));
   }
\ No newline at end of file
