diff --git a/old_test.java b/new_test.java
index bec81d4..4bcdb68 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,5 +4,5 @@
     assertEquals(Paths.get(PATH1).toRealPath().toString(), query(_FILE_PATH_TO_NATIVE.args(PATH1)));
     query(_FILE_PATH_TO_NATIVE.args(PATH + "../" + NAME + '/' + NAME),
         Paths.get(PATH + "../" + NAME + '/' + NAME).toRealPath().toString());
-    error(_FILE_PATH_TO_NATIVE.args(PATH1 + NAME), Err.FILE_NOT_FOUND);
+    error(_FILE_PATH_TO_NATIVE.args(PATH1 + NAME), Err.FILE_NOT_FOUND_X);
   }
\ No newline at end of file
