diff --git a/old_product.java b/new_product.java
index 8712df9..1975518 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,8 +7,11 @@
       // We must verify further if the schedule is blocked completely or if it will fire
       if (willBlockSchedule(scheduleTrigger, blockOut)) {
 
+        Trigger blockOutTrigger = (Trigger) blockOut;
+
         // If recurrence intervals are the same, it will never fire
-        if (getRecurrenceInterval((Trigger) blockOut) == getRecurrenceInterval(scheduleTrigger)) {
+        if (!isComplexTrigger(blockOutTrigger) && !isComplexTrigger(scheduleTrigger)
+            && getRecurrenceInterval(blockOutTrigger) == getRecurrenceInterval(scheduleTrigger)) {
           return false;
         }
 
