diff --git a/old_product.java b/new_product.java
index b127186..2586638 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,8 @@
-public static CollectionProperty tableModelToCollectionProperty(PowerTableModel model, String propname) {
+public static CollectionProperty tableModelRowsToCollectionProperty(PowerTableModel model, String propname) {
         CollectionProperty rows = new CollectionProperty(propname, new ArrayList<Object>());
-        for (int col = 0; col < model.getColumnCount(); col++) {
-            rows.addItem(model.getColumnData(model.getColumnName(col)));
+        for (int row = 0; row < model.getRowCount(); row++) {
+            List<Object> item = getArrayListForArray(model.getRowData(row));
+            rows.addItem(item);
         }
         return rows;
     }
\ No newline at end of file
