diff --git a/old_test.java b/new_test.java
index 90b4489..f5db0cd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,12 @@
 @Test
-   public void testTableModelToCollectionProperty()
-   {
-      System.out.println("tableModelToCollectionProperty");
-        PowerTableModel model = new PowerTableModel();
-      CollectionProperty prop = JMeterPluginsUtils.tableModelToCollectionProperty(model, "");
-      assertTrue(prop instanceof CollectionProperty);
-   }
\ No newline at end of file
+    public void testCollectionPropertyToTableModelRows() {
+        System.out.println("collectionPropertyToTableModelRows");
+        String propname = "prop";
+        PowerTableModel modelSrc = getTestModel();
+        CollectionProperty propExp = JMeterPluginsUtils.tableModelRowsToCollectionProperty(modelSrc, propname);
+        PowerTableModel modelDst = getTestModel();
+        modelDst.clearData();
+        JMeterPluginsUtils.collectionPropertyToTableModelRows(propExp, modelDst);
+        CollectionProperty propRes = JMeterPluginsUtils.tableModelRowsToCollectionProperty(modelDst, propname);
+        assertEquals(propExp.toString(), propRes.toString());
+    }
\ No newline at end of file
