diff --git a/old_test.java b/new_test.java
index 9831fe5..b49d6bf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
   public final void export() {
-    final IO io = IO.get("export.xml");
+    final IO io = IO.get("input.xml");
     no(new Export(io.path()));
     ok(new CreateDB(NAME, FILE));
-    ok(new Export(".", io.name()));
+    ok(new Export("."));
     ok(io.exists());
     ok(io.delete());
   }
\ No newline at end of file
