diff --git a/old_test.java b/new_test.java
index bec9990..5dd61d0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,10 @@
 @Test
     public void testCommit() {
 
-	Result result = new Result(true, true, 0, System.currentTimeMillis(), new BranchTrace(), new WorkflowTrace(),
-		new WorkflowTrace(), "test.unit");
+	Result result = new Result(true, true, 0, System.currentTimeMillis(), new BranchTrace(), new TestVector(
+		new WorkflowTrace(), null, null), new TestVector(new WorkflowTrace(), null, null), "test.unit");// TODO
+														// Delete
+														// Testfiles
 	ResultContainer instance = ResultContainer.getInstance();
 	instance.commit(result);
 
