diff --git a/old_test.java b/new_test.java
index ae1891d..d8491e8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,8 +4,8 @@
         final SimpleFilterChain filterChain = new SimpleFilterChain();
         final SimpleHttpRequest request = new SimpleHttpRequest();
 
-        final String clientToken = BaseEncoding.base64()
-                .encode(WindowsAccountImpl.getCurrentUsername().getBytes(StandardCharsets.UTF_8));
+        final String clientToken = Base64.getEncoder()
+                .encodeToString(WindowsAccountImpl.getCurrentUsername().getBytes(StandardCharsets.UTF_8));
         request.addHeader("Authorization", securityPackage + " " + clientToken);
 
         final SimpleHttpResponse response = new SimpleHttpResponse();
