diff --git a/old_test.java b/new_test.java
index 92d5437..d816d25 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,6 @@
     public void testCost() {
         sut = new LatencyConstraint(Duration.of(10, ChronoUnit.MICROS));
 
-        assertThat(sut.cost(link1, linkResourceService), is(closeTo(Double.parseDouble(LATENCY1), 1.0e-6)));
-        assertThat(sut.cost(link2, linkResourceService), is(closeTo(Double.parseDouble(LATENCY2), 1.0e-6)));
+        assertThat(sut.cost(link1, resourceContext), is(closeTo(Double.parseDouble(LATENCY1), 1.0e-6)));
+        assertThat(sut.cost(link2, resourceContext), is(closeTo(Double.parseDouble(LATENCY2), 1.0e-6)));
     }
\ No newline at end of file
