diff --git a/old_product.java b/new_product.java
index d4dd75d..4d79b16 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-private long flushSnapshot(final SkipListFlusher flusher, Checkpoint checkpoint) throws IOException {
+long flushSnapshot(final SkipListFlusher flusher, Checkpoint checkpoint) throws IOException {
         long size = 0;
         if (this.snapshot.compareTo(checkpoint) < 0) {
             long ledger, ledgerGC = -1;
