diff --git a/old_test.java b/new_test.java
index f23f576..9534f84 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testFlushSnapshot() throws IOException {
         HashSet<EntryKeyValue> keyValues = new HashSet<EntryKeyValue>();
-        HashSet<EntryKeyValue> flushedKVs = new HashSet<EntryKeyValue>();
+        Set<EntryKeyValue> flushedKVs = Collections.newSetFromMap(new ConcurrentHashMap<EntryKeyValue, Boolean>());
         KVFLusher flusher = new KVFLusher(flushedKVs);
 
         byte[] data = new byte[10];
