diff --git a/old_test.java b/new_test.java
index 8c62abd..81e512f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
         // Array container, range is complete
         final short[] content = {1, 3, 5, 7, 9};
         final Container c = makeContainer(content);
-        final Container c1 = c.not(0, 65535);
+        final Container c1 = c.not(0, 65536);
         final short[] s = new short[65536 - content.length];
         int pos = 0;
         for (int i = 0; i < 65536; ++i)
