diff --git a/old_product.java b/new_product.java
index d03feee..af7dd71 100644
--- a/old_product.java
+++ b/new_product.java
@@ -37,7 +37,7 @@ public List<Object> findBatchOfConcepts(String phrase, boolean includeRetired, L
 		
 		try {
 			ConceptService cs = Context.getConceptService();
-			List<ConceptSearchResult> searchResults = new Vector<ConceptSearchResult>();
+			Set<ConceptSearchResult> searchResults = new HashSet<ConceptSearchResult>();
 			
 			if (phrase.matches("\\d+")) {
 				// user searched on a number. Insert concept with
