diff --git a/old_product.java b/new_product.java
index ba2e0ae..60844a4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 private static void run(final int clients, final int runs) throws Exception {
     // Create test database
-    Command cmd = new CreateDB(DB, INPUT);
-    cmd.execute(CONTEXT);
+    Command cmd = new CreateDB(NAME, INPUT);
+    cmd.execute(context);
 
     // Start clients
     final Client[] cl = new Client[clients];
@@ -9,6 +9,6 @@ private static void run(final int clients, final int runs) throws Exception {
     for(final Client c : cl) c.start();
     for(final Client c : cl) c.join();
     // Drop database
-    cmd = new DropDB(DB);
-    cmd.execute(CONTEXT);
+    cmd = new DropDB(NAME);
+    cmd.execute(context);
   }
\ No newline at end of file
