diff --git a/old_product.java b/new_product.java
index 61c3723..46bd7a8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
-Flux<ResourceResponse<Document>> createDocument(String collectionLink, Object document, RequestOptions options,
+Mono<ResourceResponse<Document>> createDocument(String collectionLink, Object document, RequestOptions options,
                                                     boolean disableAutomaticIdGeneration);
\ No newline at end of file
