diff --git a/old_product.java b/new_product.java
index 6aee5ef..0499346 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,7 @@
-public InodeTree.CreatePathResult mkdir(AlluxioURI path, CreatePathOptions options)
+public InodeTree.CreatePathResult mkdir(AlluxioURI path, CreateDirectoryOptions options)
       throws InvalidPathException, FileAlreadyExistsException, IOException, AccessControlException {
     LOG.debug("mkdir {} ", path);
     MasterContext.getMasterSource().incCreateDirectoriesOps(1);
-    options.setDirectory(true);
     synchronized (mInodeTree) {
       try {
         checkPermission(FileSystemAction.WRITE, path, true);
