diff --git a/old_product.java b/new_product.java
index 60ddc46..3431646 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,9 @@
 public static int ceilPowerOfTwo(int val) {
+        final int LARGEST_FITTING_POWER_OF_TWO = Integer.MIN_VALUE >>> 1;
+
+        if(val > LARGEST_FITTING_POWER_OF_TWO)
+            throw new IllegalArgumentException("ceilPowerOfTwo overflows on values > " + LARGEST_FITTING_POWER_OF_TWO);
+
         if(val == Integer.MIN_VALUE)
             return 0;
 
