diff --git a/old_test.java b/new_test.java
index 2d22eeb..dcd15fc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,28 @@
 @Test
     public void testCeilPowerOfTwo() {
-        assertEquals(8, TeraMath.ceilPowerOfTwo(8));
-        assertEquals(8, TeraMath.ceilPowerOfTwo(7));
-        assertEquals(0, TeraMath.ceilPowerOfTwo(-100));
+        List<Integer> powersOfTwo = generateAllPowersOfTwo();
+        for(int i = 1; i < powersOfTwo.size(); i++) {
+            //test inputs on and around powers of two. Skips tests on zero
+            testCeilPowerOfTwo(powersOfTwo.get(i-1), powersOfTwo.get(i));
+        }
+
+        int largestIntegerPowerOfTwo = powersOfTwo.get(powersOfTwo.size()-1);
+        //test other boundary values
+        assertEquals("0", 0, TeraMath.ceilPowerOfTwo(0));
+        assertEquals("-1", 0, TeraMath.ceilPowerOfTwo(0));
+        assertEquals("Integer.MIN_VALUE", 0, TeraMath.ceilPowerOfTwo(Integer.MIN_VALUE));
+
+        try {
+            TeraMath.ceilPowerOfTwo(largestIntegerPowerOfTwo + 1);
+            fail("Expected runtime exception because of overflow");
+        }
+        catch (RuntimeException expectedException)
+        { }
+
+        try {
+            TeraMath.ceilPowerOfTwo(Integer.MAX_VALUE);
+            fail("Expected runtime exception because of overflow");
+        }
+        catch (RuntimeException expectedException)
+        { }
     }
\ No newline at end of file
