diff --git a/old_product.java b/new_product.java
index ca6753f..bb1035a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -19,9 +19,7 @@
                             throw e;
                         }
                     } else {
-                        verifier.verifySignatures(
-                                (JarVerifier.VerifierEntry) verStream,
-                                jarEntry);
+                        ((JarVerifier.VerifierEntry) verStream).verify();
                     }
                 }
             } else {
