diff --git a/old_product.java b/new_product.java
index 4d14c92..32009d6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,6 @@
-static Pair<String, String> getAccountContainerName(AccountService accountService, short accountId,
+static Pair<Account, Container> getAccountContainer(AccountService accountService, short accountId,
       short containerId) {
     Account account = accountService.getAccountById(accountId);
-    String accountName = account == null ? null : account.getName();
     Container container = account == null ? null : account.getContainerById(containerId);
-    String containerName = container == null ? null : container.getName();
-    return new Pair<>(accountName, containerName);
+    return new Pair<>(account, container);
   }
\ No newline at end of file
