diff --git a/old_test.java b/new_test.java
index becbb5e..afd7372 100644
--- a/old_test.java
+++ b/new_test.java
@@ -25,7 +25,8 @@
 
     // This should not throw file exists exception those a/f1 is loaded.
     mFileSystemMaster.loadMetadata(new AlluxioURI("alluxio:/a"),
-        LoadMetadataOptions.defaults().setCreateAncestors(true).setLoadDescendantLevels(1));
+        LoadMetadataOptions.defaults().setCreateAncestors(true).setLoadDescendantType(
+            DescendantType.ONE));
 
     // TODO(peis): Avoid this hack by adding an option in getFileInfo to skip loading metadata.
     try {
@@ -38,5 +39,6 @@
     }
 
     mFileSystemMaster.loadMetadata(new AlluxioURI("alluxio:/a"),
-        LoadMetadataOptions.defaults().setCreateAncestors(true).setLoadDescendantLevels(1));
+        LoadMetadataOptions.defaults().setCreateAncestors(true)
+            .setLoadDescendantType(DescendantType.ONE));
   }
\ No newline at end of file
