diff --git a/old_product.java b/new_product.java
index 2926a4f..71f7d3a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,7 +7,8 @@
       @Override
       public LoadMetadataTResponse call() throws AlluxioException, IOException {
         return new LoadMetadataTResponse(mFileSystemMaster.loadMetadata(new AlluxioURI(alluxioPath),
-            LoadMetadataOptions.defaults().setCreateAncestors(true).setLoadDescendantLevels(1)));
+            LoadMetadataOptions.defaults().setCreateAncestors(true).setLoadDescendantType(
+                DescendantType.ONE)));
       }
 
       @Override
