diff --git a/old_test.java b/new_test.java
index fb623e9..0f675d7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -39,10 +39,10 @@
 		Time delayBetweenSamples = Time.milliseconds(100L);
 
 		BackPressureStatsTracker tracker = new BackPressureStatsTracker(
-				sampleCoordinator, 9999, numSamples, delayBetweenSamples);
+				sampleCoordinator, 9999, numSamples, Integer.MAX_VALUE, delayBetweenSamples);
 
-		// Trigger
-		Assert.assertTrue("Failed to trigger", tracker.triggerStackTraceSample(jobVertex));
+		// getOperatorBackPressureStats triggers stack trace sampling
+		Assert.assertFalse(tracker.getOperatorBackPressureStats(jobVertex).isPresent());
 
 		Mockito.verify(sampleCoordinator).triggerStackTraceSample(
 				Matchers.eq(taskVertices),
@@ -51,7 +51,7 @@
 				Matchers.eq(BackPressureStatsTracker.MAX_STACK_TRACE_DEPTH));
 
 		// Trigger again for pending request, should not fire
-		Assert.assertFalse("Unexpected trigger", tracker.triggerStackTraceSample(jobVertex));
+		Assert.assertFalse("Unexpected trigger", tracker.triggerStackTraceSampleInternal(jobVertex));
 
 		Assert.assertTrue(!tracker.getOperatorBackPressureStats(jobVertex).isPresent());
 
