diff --git a/old_test.java b/new_test.java
index 84bc30b..f74d828 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,14 +14,14 @@
         assertEquals(expected, WordUtils.wrap(input, 20));
 
         // long word at end
-        input = "Click here to jump to the commons website - http://commons.apache.org";
+        input = "Click here to jump to the commons website - https://commons.apache.org";
         expected = "Click here to jump" + systemNewLine + "to the commons" + systemNewLine
-            + "website -" + systemNewLine + "http://commons.apache.org";
+            + "website -" + systemNewLine + "https://commons.apache.org";
         assertEquals(expected, WordUtils.wrap(input, 20));
 
         // long word in middle
-        input = "Click here, http://commons.apache.org, to jump to the commons website";
-        expected = "Click here," + systemNewLine + "http://commons.apache.org," + systemNewLine
+        input = "Click here, https://commons.apache.org, to jump to the commons website";
+        expected = "Click here," + systemNewLine + "https://commons.apache.org," + systemNewLine
             + "to jump to the" + systemNewLine + "commons website";
         assertEquals(expected, WordUtils.wrap(input, 20));
 
