diff --git a/old_product.java b/new_product.java
index 66c5133..b50caa0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -18,7 +18,8 @@ public static String wrap(final String str, int wrapLength, String newLineStr, f
 
         while (offset < inputLineLength) {
             int spaceToWrapAt = -1;
-            Matcher matcher = patternToWrapOn.matcher(str.substring(offset, Math.min(offset + wrapLength + 1, inputLineLength)));
+            Matcher matcher = patternToWrapOn.matcher(
+                str.substring(offset, Math.min((int)Math.min(Integer.MAX_VALUE, offset + wrapLength + 1L), inputLineLength)));
             if (matcher.find()) {
                 if (matcher.start() == 0) {
                     offset += matcher.end();
