diff --git a/old_test.java b/new_test.java
index 913910a..9055c50 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test(groups={"tck"})
     public void test_query_chrono() {
-        assertEquals(TEST_2007_07_15_PONE.query(Query.CHRONO), ISOChrono.INSTANCE);
+        assertEquals(TEST_2007_07_15_PONE.query(TemporalQuery.CHRONO), ISOChrono.INSTANCE);
     }
\ No newline at end of file
