diff --git a/old_test.java b/new_test.java
index 4e068ad..4da792f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void sortWithTest() {
-    query("hof:sort-with(function($a, $b) { $a < $b }, ())", "");
-    query("hof:sort-with(function($a, $b) { $a > $b }, 1 to 5)", "5 4 3 2 1");
-    error("hof:sort-with(<x/>, 1 to 5)", Err.INVCAST);
+    query("hof:sort-with((), function($a, $b) { $a < $b })", "");
+    query("hof:sort-with(1 to 5, function($a, $b) { $a > $b })", "5 4 3 2 1");
+    error("hof:sort-with(1 to 5, <x/>)", Err.INVCAST);
   }
\ No newline at end of file
