diff --git a/old_product.java b/new_product.java
index cf4de31..1e4a431 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 private Value sortWith(final QueryContext ctx) throws QueryException {
-    final Value v = expr[1].value(ctx);
-    final Comparator<Item> cmp = getComp(0, ctx);
+    final Value v = expr[0].value(ctx);
+    final Comparator<Item> cmp = getComp(1, ctx);
     if(v.size() < 2) return v;
     final ValueBuilder vb = v.cache();
     try {
