diff --git a/old_product.java b/new_product.java
index c6ddff9..3a98bc7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public void hugeFolder() throws Exception {
         String username = generateUsername();
         String password = "test01";
         UserContext context = PeergosNetworkUtils.ensureSignedUp(username, password, network, crypto);
-        FileTreeNode userRoot = context.getUserRoot().get();
+        FileWrapper userRoot = context.getUserRoot().get();
         List<String> names = new ArrayList<>();
         IntStream.range(0, 2000).forEach(i -> names.add(randomString()));
 
