diff --git a/old_test.java b/new_test.java
index 503c314..2f6e525 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
         String username = generateUsername();
         String password = "test01";
         UserContext context = ensureSignedUp(username, password, network, crypto);
-        FileTreeNode userRoot = context.getUserRoot().get();
+        FileWrapper userRoot = context.getUserRoot().get();
         List<String> names = new ArrayList<>();
         IntStream.range(0, 100).forEach(i -> names.add(randomString()));
 
