diff --git a/old_test.java b/new_test.java
index 8f00f28..4f1a1ce 100644
--- a/old_test.java
+++ b/new_test.java
@@ -62,7 +62,7 @@
         
         
         String dbName = "JUnitClassifier";
-        BinarizedNaiveBayes instance = new BinarizedNaiveBayes(dbName);
+        BinarizedNaiveBayes instance = new BinarizedNaiveBayes(dbName, new InMemoryConfiguration());
         
         BinarizedNaiveBayes.TrainingParameters param = instance.getEmptyTrainingParametersObject();
         instance.initializeTrainingConfiguration(param);
