diff --git a/old_product.java b/new_product.java
index 735be04..8d71767 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,5 +4,5 @@
             System.out.println("kFoldCrossValidation()");
         }
         
-        return modelValidator.kFoldCrossValidation(trainingData, k, dbName, this.getClass(), knowledgeBase.getTrainingParameters());
+        return modelValidator.kFoldCrossValidation(trainingData, k, dbName, knowledgeBase.getDbConf(), this.getClass(), knowledgeBase.getTrainingParameters());
     }
\ No newline at end of file
