diff --git a/old_product.java b/new_product.java
index e6a6ffd..8ab51a5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ private Str serialize(final QueryContext ctx) throws QueryException {
     final CsvOptions opts = checkOptions(1, Q_OPTIONS, new CsvOptions(), ctx);
 
     final SerializerOptions sopts = new SerializerOptions();
-    sopts.set(S_METHOD, M_CSV);
-    sopts.set(S_CSV, opts.toString());
-    return Str.get(delete(serialize(node.iter(), sopts), '\r'));
+    sopts.set(SerializerOptions.METHOD, SerialMethod.CSV.toString());
+    sopts.set(SerializerOptions.CSV, opts.toString());
+    return Str.get(delete(serialize(node.iter(), sopts, INVALIDOPT), '\r'));
   }
\ No newline at end of file
