diff --git a/old_product.java b/new_product.java
index 7b39921..b5bffaa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,17 +1,33 @@
 public Color getNextColor() {
-        Color ret;
-        index++;
-        if (index < fixedColors.length) {
-            ret = fixedColors[index];
-        } else {
-            int rndIndex = index - fixedColors.length;
+        Color color = null;
+
+        doCycles();
 
-            if (randomColors.size() > rndIndex) {
-                ret = randomColors.get(rndIndex);
-            } else {
-                ret = new Color(rnd.nextInt(0xFFFFFF));
-                randomColors.add(ret);
-            }
+        int r = 0, g = 0, b = 0;
+        if ((bits & 1) == 1) {
+            r = level;
+        }
+        if ((bits & 2) == 2) {
+            g = level;
+        }
+        if ((bits & 4) == 4) {
+            b = level;
         }
-        return ret;
+        Color c = new Color(r, g, b);
+        if (assignedColors.contains(c)) {
+            System.out.println("Existing " + r + " " + g + " " + b);
+            color = getNextColor();
+        } else if ((r + g + b) / 3 < 32) {
+            log.debug("Too dark " + r + " " + g + " " + b);
+            color = getNextColor();
+        } else if ((r + g + b) / 3 > 256 - 64) {
+            log.debug("Too light " + r + " " + g + " " + b);
+            color = getNextColor();
+        } else {
+            log.debug("New " + r + " " + g + " " + b);
+            color = new Color(r, g, b);
+        }
+
+        assignedColors.add(color);
+        return color;
     }
\ No newline at end of file
