diff --git a/old_test.java b/new_test.java
index 27f9855..888a626 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,17 +14,17 @@
 
         String queryText = instance.buildSearch(vendor, product, version, null, null);
         String expResult = " product:( struts 2 core )  AND  vendor:( apache software foundation )  AND version:(2.1.2^0.7 )";
-        assertTrue(expResult.equals(queryText));
+        Assert.assertTrue(expResult.equals(queryText));
 
         queryText = instance.buildSearch(vendor, product, version, null, productWeightings);
         expResult = " product:(  struts^5 struts2^5 2 core )  AND  vendor:( apache software foundation )  AND version:(2.1.2^0.2 )";
-        assertTrue(expResult.equals(queryText));
+        Assert.assertTrue(expResult.equals(queryText));
 
         queryText = instance.buildSearch(vendor, product, version, vendorWeightings, null);
         expResult = " product:( struts 2 core )  AND  vendor:(  apache^5 software foundation )  AND version:(2.1.2^0.2 )";
-        assertTrue(expResult.equals(queryText));
+        Assert.assertTrue(expResult.equals(queryText));
 
         queryText = instance.buildSearch(vendor, product, version, vendorWeightings, productWeightings);
         expResult = " product:(  struts^5 struts2^5 2 core )  AND  vendor:(  apache^5 software foundation )  AND version:(2.1.2^0.2 )";
-        assertTrue(expResult.equals(queryText));
+        Assert.assertTrue(expResult.equals(queryText));
     }
\ No newline at end of file
