diff --git a/old_test.java b/new_test.java
index 5a56baf..44f1eb2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,10 +3,9 @@
     {
         context.turnOffAuthorisationSystem(); //must be an Admin to create a Group
         int step = 1;
-        Group g = Group.create(context);
-        context.commit();
+        Group g = groupService.create(context);
         context.restoreAuthSystemState();
-        c.setWorkflowGroup(step, g);
-        assertThat("testSetWorkflowGroup 0",c.getWorkflowGroup(step), notNullValue());
-        assertThat("testSetWorkflowGroup 1",c.getWorkflowGroup(step), equalTo(g));
+        collection.setWorkflowGroup(step, g);
+        assertThat("testSetWorkflowGroup 0",collectionService.getWorkflowGroup(collection, step), notNullValue());
+        assertThat("testSetWorkflowGroup 1",collectionService.getWorkflowGroup(collection, step), equalTo(g));
     }
\ No newline at end of file
