diff --git a/old_test.java b/new_test.java
index d847019..0c7e09f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,12 +3,19 @@
         Ignite ignite = startGrids(1);
 
         assertFalse(ignite.cluster().active());
+        assertEquals(INACTIVE, ignite.cluster().state());
 
-        ignite.cluster().active(true);
+        ignite.cluster().state(ACTIVE);
 
         assertTrue(ignite.cluster().active());
+        assertEquals(ACTIVE, ignite.cluster().state());
+
+        injectTestSystemOut();
 
         assertEquals(EXIT_CODE_OK, execute("--deactivate"));
 
         assertFalse(ignite.cluster().active());
+        assertEquals(INACTIVE, ignite.cluster().state());
+
+        assertContains(log, testOut.toString(), "Command deprecated. Use --set-state instead.");
     }
\ No newline at end of file
