diff --git a/old_product.java b/new_product.java
index 37f132f..37834b2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,8 @@
-public void add(String id, String url, String oldUrl, long timestamp, boolean needsUpdate) {
+public void add(String id, String url, long timestamp, boolean needsUpdate) {
         final NvdCveInfo item = new NvdCveInfo();
         item.setNeedsUpdate(needsUpdate); //the others default to true, to make life easier later this should default to false.
         item.setId(id);
         item.setUrl(url);
-        item.setOldSchemaVersionUrl(oldUrl);
         item.setTimestamp(timestamp);
         collection.put(id, item);
     }
\ No newline at end of file
