diff --git a/old_test.java b/new_test.java
index d06e5f9..e402723 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,20 +1,19 @@
 @Test
     public void testAdd() throws Exception {
         String id = "key";
-        String url = new File("target/test-classes/nvdcve-2.0-2012.xml").toURI().toString();
+        String url = new File("target/test-classes/nvdcve-1.0-2012.json.gz").toURI().toString();
         long timestamp = 42;
         UpdateableNvdCve instance = new UpdateableNvdCve();
-        instance.add(id, url, url, timestamp, false);
+        instance.add(id, url, timestamp, false);
 
         boolean expResult = false;
         boolean result = instance.isUpdateNeeded();
         assertEquals(expResult, result);
 
-        instance.add("nextId", url, url, 23, false);
+        instance.add("nextId", url, 23, false);
         NvdCveInfo results = instance.get(id);
 
         assertEquals(id, results.getId());
         assertEquals(url, results.getUrl());
-        assertEquals(url, results.getOldSchemaVersionUrl());
         assertEquals(timestamp, results.getTimestamp());
     }
\ No newline at end of file
