diff --git a/old_product.java b/new_product.java
index 3a4f558..3c414aa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
-Future<LogSegmentMetadata> changeSequenceNumber(LogSegmentMetadata segment,
+CompletableFuture<LogSegmentMetadata> changeSequenceNumber(LogSegmentMetadata segment,
                                                     long logSegmentSeqNo);
\ No newline at end of file
