diff --git a/old_test.java b/new_test.java
index 36717e8..6763e9a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,7 @@
 @Test
 	public void testPostDestroyConnection(){
+		Connection mockRealConnection = createNiceMock(Connection.class);
+
 		reset(mockConnection);
 		expect(mockConnection.getOriginatingPartition()).andReturn(mockPartition).anyTimes();
 		mockPartition.updateCreatedConnections(-1);
@@ -8,9 +10,11 @@
 		expectLastCall().once();
 		ConnectionHook mockConnectionHook = createNiceMock(ConnectionHook.class);
 		expect(mockConnection.getConnectionHook()).andReturn(mockConnectionHook).anyTimes();
+		expect(mockConnection.getInternalConnection()).andReturn(mockRealConnection).anyTimes();
+
 		mockConnectionHook.onDestroy(mockConnection);
 		expectLastCall().once();
-		replay(mockConnectionHook, mockConnection);
+		replay(mockRealConnection, mockConnectionHook, mockConnection);
 		testClass.postDestroyConnection(mockConnection);
 		verify(mockConnectionHook, mockConnection);
 	}
\ No newline at end of file
