diff --git a/old_test.java b/new_test.java
index c356039..285060a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,7 +16,7 @@
                 ByteArrayInputStream writeStream = new ByteArrayInputStream(writeData);
                 long seqNo = log.append(writeStream, Timeout).join();
 
-                AssertExtensions.assertLessThan("Sequence Number is not monotonically increasing.", prevSeqNo, seqNo);
+                AssertExtensions.assertGreaterThan("Sequence Number is not monotonically increasing.", prevSeqNo, seqNo);
                 prevSeqNo = seqNo;
             }
         }
