diff --git a/old_test.java b/new_test.java
index 6d675e6..450899d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,8 @@
         user2.setId(2L);
 
         User[] users = { user1, user2 };
+        LOGGER.info(JsonUtil.format(ConvertUtil.toList(users)));
 
-        List<User> list = ConvertUtil.toList(users);
-        LOGGER.info(JsonUtil.format(list));
+        users = null;
+        LOGGER.info(JsonUtil.format(ConvertUtil.toList(users)));
     }
\ No newline at end of file
