diff --git a/old_product.java b/new_product.java
index 6a6ecc2..2b7caac 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,4 @@
-public static <T> List<T> toList(final Enumeration<T> enumeration){
-        if (Validator.isNullOrEmpty(enumeration)){
-            return Collections.emptyList();
-        }
-        return Collections.list(enumeration);
+@SuppressWarnings("unchecked")
+    public static <T> List<T> toList(final Enumeration<T> enumeration){
+        return Validator.isNullOrEmpty(enumeration) ? (List<T>) Collections.emptyList() : Collections.list(enumeration);
     }
\ No newline at end of file
