diff --git a/old_test.java b/new_test.java
index 0f00886..53028de 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,10 +5,13 @@
 		initializeInMemoryDatabase();
 		executeDataSet(INITIAL_CONCEPTS_XML);
 		authenticate();
-		
+		//This will automatically add the given locale to the list of allowed locales
+		Context.setLocale(Locale.US);
 		// this tests saving a previously conceptnumeric as just a concept
 		Concept c2 = new Concept(2);
-		c2.addName(new ConceptName("not a numeric anymore", Locale.US));
+		ConceptName cn = new ConceptName("not a numeric anymore", Locale.US);
+		c2.addName(cn);
+		
 		c2.setDatatype(new ConceptDatatype(3));
 		conceptService.saveConcept(c2);
 		
