diff --git a/old_product.java b/new_product.java
index d3713b4..46fbac9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,19 +5,6 @@ public void validate(Object obj, Errors errors) {
 		if (order == null) {
 			errors.rejectValue("order", "error.general");
 		} else {
-			// for the following elements Order.hbm.xml says: not-null="true"
-			ValidationUtils.rejectIfEmpty(errors, "prn", "error.null");
-			
-			if (order.isSigned()) {
-				ValidationUtils.rejectIfEmpty(errors, "signedBy", "error.null");
-				ValidationUtils.rejectIfEmpty(errors, "dateSigned", "error.null");
-			}
-			
-			if (order.isActivated()) {
-				ValidationUtils.rejectIfEmpty(errors, "activatedBy", "error.null");
-				ValidationUtils.rejectIfEmpty(errors, "dateActivated", "error.null");
-			}
-			
 			if (order.getDuration() != null)
 				ValidationUtils.rejectIfEmpty(errors, "durationUnits", "DrugOrder.add.error.missingDurationUnits");
 			
@@ -27,5 +14,8 @@ public void validate(Object obj, Errors errors) {
 			if (order.getStrength() != null)
 				ValidationUtils.rejectIfEmpty(errors, "strengthUnits", "DrugOrder.add.error.missingStrengthUnits");
 			
+			if (order.getDose() != null)
+				ValidationUtils.rejectIfEmpty(errors, "doseUnits", "DrugOrder.add.error.missingDoseUnits");
+			
 		}
 	}
\ No newline at end of file
