diff --git a/old_test.java b/new_test.java
index 7470fef..9b7cc89 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
     query(_CONVERT_BINARY_TO_BYTES.args("xs:base64Binary(xs:hexBinary('4261736558'))"),
       "66 97 115 101 88");
     query(_CONVERT_BINARY_TO_BYTES.args("xs:base64Binary(<x>AAE=</x>)"), "0 1");
-    query(_CONVERT_BINARY_TO_BYTES.args("a"), 97);
-    query(COUNT.args(_CONVERT_BINARY_TO_BYTES.args("\u00e4")), 2);
-    query(COUNT.args(_CONVERT_BINARY_TO_BYTES.args(123)), 3);
+    query(_CONVERT_BINARY_TO_BYTES.args(_CONVERT_STRING_TO_BASE64.args("a")), 97);
+    query(COUNT.args(_CONVERT_BINARY_TO_BYTES.args(_CONVERT_STRING_TO_BASE64.args("\u00e4"))), 2);
+    query(COUNT.args(_CONVERT_BINARY_TO_BYTES.args(_CONVERT_STRING_TO_BASE64.args("123"))), 3);
   }
\ No newline at end of file
