diff --git a/old_product.java b/new_product.java
index b9d4747..d933031 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 private Value binaryToBytes(final QueryContext qc) throws QueryException {
     try {
-      return BytSeq.get(checkItem(exprs[0], qc).input(info).content());
+      return BytSeq.get(toBin(exprs[0], qc).input(info).content());
     } catch(final IOException ex) {
-      throw BXCO_STRING.get(info, ex);
+      throw BXCO_STRING_X.get(info, ex);
     }
   }
\ No newline at end of file
