diff --git a/old_test.java b/new_test.java
index 1ac2fbf..f14d0a7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,7 +17,7 @@
             .perform(
                 MockMvcRequestBuilders.post(APPLICATION_API_PATH)
                     .contentType(MediaType.APPLICATION_JSON)
-                    .content(new ObjectMapper().writeValueAsBytes(app))
+                    .content(objectMapper.writeValueAsBytes(app))
             )
             .andExpect(MockMvcResultMatchers.status().isCreated())
             .andExpect(MockMvcResultMatchers
