diff --git a/old_test.java b/new_test.java
index b9a8e5c..41ba19e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
-  public void testFindTargetByName() {
+  public void testFindTargetByName() throws Exception {
     List<AlertTargetEntity> targets = dao.findAllTargets();
-    Assert.assertNotNull(targets);
+    assertNotNull(targets);
     AlertTargetEntity target = targets.get(3);
 
     AlertTargetEntity actual = dao.findTargetByName(target.getTargetName());
-    Assert.assertEquals(target, actual);
+    assertEquals(target, actual);
   }
\ No newline at end of file
