diff --git a/old_product.java b/new_product.java
index 2175284..6efb776 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,6 +2,10 @@
 	public boolean dropTrack(PointTrack track) {
 		if( !tracksAll.remove(track) )
 			return false;
+
+		if( !sets[track.setId].tracks.remove(track) ) {
+			return false;
+		}
 		// the track may or may not be in the active list
 		tracksActive.remove(track);
 		tracksInactive.remove(track);
