diff --git a/old_test.java b/new_test.java
index 16d730a..56c0b75 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,11 +5,14 @@
 		dat.tracksAll.add(dat.getUnused());
 		dat.tracksAll.add(dat.getUnused());
 		dat.tracksAll.add(dat.getUnused());
+		for (int i = 0; i < dat.tracksAll.size(); i++) {
+			dat.sets[0].tracks.add(dat.tracksAll.get(i));
+		}
 
 		PointTrack a = dat.tracksAll.get(1);
 
 		assertEquals(0,dat.unused.size());
-		dat.dropTrack(a);
+		assertTrue(dat.dropTrack(a));
 		assertEquals(1,dat.unused.size());
 
 		assertEquals(2,dat.tracksAll.size());
