diff --git a/old_product.java b/new_product.java
index 87f8ef1..2e5c452 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,6 +11,11 @@ public boolean restartApp(int appId) {
     String restartRequest = String.format("%s/v2/apps/%s/%d/restart",
         this.marathonURI, this.topologyName, appId);
     HttpURLConnection conn = NetworkUtils.getHttpConnection(restartRequest);
+
+    if (this.marathonAuthToken != null) {
+      conn.setRequestProperty("Authorization", String.format("token=%s", this.marathonAuthToken));
+    }
+
     if (conn == null) {
       LOG.log(Level.SEVERE, "Failed to find marathon scheduler");
       return false;
@@ -30,6 +35,9 @@ public boolean restartApp(int appId) {
       if (success) {
         LOG.log(Level.INFO, "Successfully restarted container {0}", appId);
         return true;
+      } else if (NetworkUtils.checkHttpResponseCode(conn, HttpURLConnection.HTTP_UNAUTHORIZED)) {
+        LOG.log(Level.SEVERE, "Marathon requires authentication");
+        return false;
       } else {
         LOG.log(Level.SEVERE, "Failed to restart container {0}", appId);
         return false;
