diff --git a/old_test.java b/new_test.java
index 2340c11..bead84a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,8 @@
 
     // Failed to get connection
     PowerMockito.spy(NetworkUtils.class);
-    PowerMockito.doReturn(null).when(NetworkUtils.class, "getHttpConnection", Mockito.anyString());
+    PowerMockito.doReturn(httpURLConnection)
+        .when(NetworkUtils.class, "getHttpConnection", Mockito.anyString());
     Assert.assertFalse(controller.restartApp(appId));
     PowerMockito.verifyStatic();
     NetworkUtils.getHttpConnection(Mockito.anyString());
@@ -48,6 +49,31 @@
         Mockito.any(byte[].class));
     NetworkUtils.checkHttpResponseCode(Mockito.any(HttpURLConnection.class), Mockito.anyInt());
 
+    // Failed authentication
+    PowerMockito.spy(NetworkUtils.class);
+    PowerMockito.doReturn(httpURLConnection)
+        .when(NetworkUtils.class, "getHttpConnection", Mockito.anyString());
+    PowerMockito.doReturn(true)
+        .when(NetworkUtils.class, "sendHttpPostRequest",
+            Mockito.any(HttpURLConnection.class),
+            Mockito.anyString(),
+            Mockito.any(byte[].class));
+    PowerMockito.doReturn(false)
+        .when(NetworkUtils.class, "checkHttpResponseCode",
+            Mockito.any(HttpURLConnection.class), Mockito.anyInt());
+    PowerMockito.doReturn(true)
+        .when(NetworkUtils.class, "checkHttpResponseCode",
+            httpURLConnection, HttpURLConnection.HTTP_UNAUTHORIZED);
+    Assert.assertFalse(controller.restartApp(appId));
+    PowerMockito.verifyStatic();
+    NetworkUtils.getHttpConnection(Mockito.anyString());
+    NetworkUtils.sendHttpPostRequest(
+        Mockito.any(HttpURLConnection.class),
+        Mockito.anyString(),
+        Mockito.any(byte[].class));
+    NetworkUtils.checkHttpResponseCode(Mockito.any(HttpURLConnection.class), Mockito.anyInt());
+    NetworkUtils.checkHttpResponseCode(httpURLConnection, HttpURLConnection.HTTP_UNAUTHORIZED);
+
     // Success
     PowerMockito.spy(NetworkUtils.class);
     PowerMockito.doReturn(httpURLConnection)
