diff --git a/old_test.java b/new_test.java
index c796acf..888c9dc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,17 +1,6 @@
 @Test(groups={"tck"})
     public void test_toString_formatter() {
-        final OffsetDate date = OffsetDate.of(2010, 12, 3, OFFSET_PONE);
-        CalendricalFormatter f = new CalendricalFormatter() {
-            @Override
-            public String print(DateTimeAccessor accessor) {
-                assertEquals(accessor, date);
-                return "PRINTED";
-            }
-            @Override
-            public <T> T parse(CharSequence text, Class<T> type) {
-                throw new AssertionError();
-            }
-        };
-        String t = date.toString(f);
-        assertEquals(t, "PRINTED");
+        DateTimeFormatter f = DateTimeFormatters.pattern("y M d");
+        String t = OffsetDate.of(2010, 12, 3, OFFSET_PONE).toString(f);
+        assertEquals(t, "2010 12 3");
     }
\ No newline at end of file
