diff --git a/old_test.java b/new_test.java
index 101c3f8..6444706 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
 		StringResponse response = new StringResponse();
 		String url = "some/url;jsessionid=1234?with=parameters&p1=v1";
 		String media = "some&bad&media";
-		CssUtils.writeLinkUrl(response, url, media);
+		CssUtils.writeLinkUrl(response, url, media, "markupId");
 
-		assertEquals("<link rel=\"stylesheet\" type=\"text/css\" href=\"some/url;jsessionid=1234?with=parameters&amp;p1=v1\" media=\"some&amp;bad&amp;media\" />", response.toString());
+		assertEquals("<link rel=\"stylesheet\" type=\"text/css\" href=\"some/url;jsessionid=1234?with=parameters&amp;p1=v1\" media=\"some&amp;bad&amp;media\" id=\"markupId\" />", response.toString());
 	}
\ No newline at end of file
