diff --git a/old_product.java b/new_product.java
index 34dece6..c3351a1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,20 @@
-public static void writeLinkUrl(final Response response, final CharSequence url, final CharSequence media)
+public static void writeLinkUrl(final Response response, final CharSequence url, final CharSequence media,
+	                                final String markupId)
 	{
 		response.write("<link rel=\"stylesheet\" type=\"text/css\" href=\"");
 		response.write(Strings.escapeMarkup(url));
 		response.write("\"");
-		if (media != null)
+		if (Strings.isEmpty(media) == false)
 		{
 			response.write(" media=\"");
 			response.write(Strings.escapeMarkup(media));
 			response.write("\"");
 		}
+		if (Strings.isEmpty(markupId) == false)
+		{
+			response.write(" id=\"");
+			response.write(Strings.escapeMarkup(markupId));
+			response.write("\"");
+		}
 		response.write(" />");
 	}
\ No newline at end of file
