diff --git a/old_product.java b/new_product.java
index 8316b35..8bf6ace 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
-public static int getInt(String key, int defaultValue) {
+public int getInt(String key, int defaultValue) {
         int value;
         try {
-            value = Integer.parseInt(Settings.getString(key));
+            value = Integer.parseInt(getString(key));
         } catch (NumberFormatException ex) {
-            if (!Settings.getString(key, "").isEmpty()) {
-                LOGGER.debug("Could not convert property '{}={}' to an int; using {} instead.", key, Settings.getString(key), defaultValue);
+            if (!getString(key, "").isEmpty()) {
+                LOGGER.debug("Could not convert property '{}={}' to an int; using {} instead.", key, getString(key), defaultValue);
             }
             value = defaultValue;
         }
