diff --git a/old_test.java b/new_test.java
index a535c0d..eab0279 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testGetInt() throws InvalidSettingException {
         String key = "SomeNumber";
         int expResult = 85;
-        Settings.setString(key, "85");
-        int result = Settings.getInt(key);
+        getSettings().setString(key, "85");
+        int result = getSettings().getInt(key);
         Assert.assertEquals(expResult, result);
     }
\ No newline at end of file
