diff --git a/old_test.java b/new_test.java
index 90c00d1..31f9bc4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(expectedExceptions=NullPointerException.class, groups={"tck"})
-    public void test_parse_Rule_String_nullRule() throws Exception {
+    public void test_parse_Class_String_nullRule() throws Exception {
         DateTimeFormatter test = new DateTimeFormatter(compPP, Locale.ENGLISH, DateTimeFormatSymbols.STANDARD);
         test.parse("30", (Class<?>) null);
     }
\ No newline at end of file
