diff --git a/old_product.java b/new_product.java
index f3748f9..cccbe7b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,5 +7,6 @@ public static Instant parse(final CharSequence text) {
         if (text.charAt(length - 1) != 'Z' && text.charAt(length - 1) != 'z') {
             throw new CalendricalParseException("Instant could not be parsed: " + text, text, length - 1);
         }
-        return OffsetDateTime.of(LocalDateTime.parse(text.subSequence(0, length - 1)), ZoneOffset.UTC).toInstant();
+        String str = text.toString().replace(',', '.');  //TODO properly, the decimal point may be either a dot or a comma
+        return OffsetDateTime.of(LocalDateTime.parse(str.substring(0, length - 1)), ZoneOffset.UTC).toInstant();
     }
\ No newline at end of file
