diff --git a/old_test.java b/new_test.java
index 2656a3e..32d73bf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,20 +1,21 @@
 @Test
-	public void testIdentify() {
+  public void testIdentify() {
 
-		Map<String, Object> response = new HashMap<String, Object>();
-		List<IdentifiedLanguage> langs = new ArrayList<IdentifiedLanguage>();
-		langs.add(new IdentifiedLanguage("en", 0.877159));
-		langs.add(new IdentifiedLanguage("af", 0.0752636));
-		langs.add(new IdentifiedLanguage("nl", 0.0301593));
+    final Map<String, Object> response = new HashMap<String, Object>();
+    final List<IdentifiedLanguage> langs = new ArrayList<IdentifiedLanguage>();
+    langs.add(new IdentifiedLanguage("en", 0.877159));
+    langs.add(new IdentifiedLanguage("af", 0.0752636));
+    langs.add(new IdentifiedLanguage("nl", 0.0301593));
 
-		response.put("languages", langs);
+    response.put("languages", langs);
 
-		mockServer.when(request().withMethod("POST").withPath(IDENTITY_PATH).withBody(text)).respond(
-				response().withHeaders(new Header(HttpHeaders.Names.CONTENT_TYPE, MediaType.APPLICATION_JSON))
-						.withBody(GsonSingleton.getGson().toJson(response)));
+    mockServer.when(request().withMethod("POST").withPath(IDENTITY_PATH).withBody(text)).respond(
+        response().withHeaders(
+            new Header(HttpHeaders.Names.CONTENT_TYPE, HttpMediaType.APPLICATION_JSON)).withBody(
+            GsonSingleton.getGson().toJson(response)));
 
-		List<IdentifiedLanguage> identifiedLanguages = service.identify(text);
-		assertNotNull(identifiedLanguages);
-		assertFalse(identifiedLanguages.isEmpty());
-		assertNotNull(identifiedLanguages.containsAll(langs));
-	}
\ No newline at end of file
+    final List<IdentifiedLanguage> identifiedLanguages = service.identify(text);
+    assertNotNull(identifiedLanguages);
+    assertFalse(identifiedLanguages.isEmpty());
+    assertNotNull(identifiedLanguages.containsAll(langs));
+  }
\ No newline at end of file
