diff --git a/old_product.java b/new_product.java
index c8279b7..34f57ef 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public MemberList getMembers() {
-        final MemberList list = new MemberList(this.getClient(), this.getSid());
+        MemberList list = new MemberList(this.getClient(), this.getSid());
         list.setRequestAccountSid(this.getRequestAccountSid());
         return list;
     }
\ No newline at end of file
