diff --git a/old_test.java b/new_test.java
index 1eb3c26..0c897d1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,8 +7,8 @@
                         + ".json"), Matchers.eq("GET"), Matchers.any(Map.class)))
                 .toReturn(resp);
 
-        final TwilioRestResponse membersresp = mock(TwilioRestResponse.class);
-        final HashMap<String, Object> map = new HashMap<String, Object>();
+        TwilioRestResponse membersresp = mock(TwilioRestResponse.class);
+        HashMap<String, Object> map = new HashMap<String, Object>();
         stub(membersresp.toMap()).toReturn(map);
         stub(membersresp.getParser()).toReturn(new JsonResponseParser());
         
@@ -24,9 +24,9 @@
                 client.safeRequest(Matchers.eq("/2010-04-01/Accounts/" + accountSid + "/Queues/" + queueSid
                         + "/Members.json"), Matchers.eq("GET"), Matchers.any(Map.class)))
                 .toReturn(membersresp);
-        final Queue q = new Queue(client, queueSid);
+        Queue q = new Queue(client, queueSid);
         q.setRequestAccountSid(accountSid);
-        final MemberList m = q.getMembers();
-        final Iterator<Member> memiter = m.iterator();
+        MemberList m = q.getMembers();
+        Iterator<Member> memiter = m.iterator();
         assertTrue(memiter.hasNext());
     }
\ No newline at end of file
