diff --git a/old_test.java b/new_test.java
index 8d65e28..be0a66e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,10 @@
 @Test
 	void process_small() {
-		// go through features of different radius settings
-		fail("implement");
+		w = 5;
+		RenderCalibrationTargetsGraphics2D renderer = new RenderCalibrationTargetsGraphics2D(p,1);
+		renderer.chessboard(rows,cols,w);
+
+		DetectChessboardCorners<GrayF32,GrayF32> alg = new DetectChessboardCorners<>(GrayF32.class);
+		alg.setKernelRadius(1); // should fail if 2
+		checkSolution(renderer.getGrayF32(), alg);
 	}
\ No newline at end of file
