diff --git a/old_test.java b/new_test.java
index f68cff7..0e8a86b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,6 +7,9 @@
         .that(testLogHandler)
         .hasLogAtLevelWithMessage(
             Level.INFO,
-            "Cannot associate admin user badGaeUserId with configured client Id."
-                + " ClientId is null or empty.");
+            formatMessage(
+                "Cannot associate admin user {user} with configured client Id."
+                    + " ClientId is null or empty.",
+                UNAUTHORIZED_ADMIN,
+                null));
   }
\ No newline at end of file
