diff --git a/old_product.java b/new_product.java
index 5fb5277..3c1381d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
-public static List<SquareBlob> copy( List<SquareBlob> input ) {
-		List<SquareBlob> output = new ArrayList<SquareBlob>();
+public static List<QuadBlob> copy( List<QuadBlob> input ) {
+		List<QuadBlob> output = new ArrayList<QuadBlob>();
 
-		for( SquareBlob i : input ) {
-			SquareBlob o = new SquareBlob();
+		for( QuadBlob i : input ) {
+			QuadBlob o = new QuadBlob();
 			o.contour = i.contour;
 			o.corners = i.corners;
 			o.center = i.center;
@@ -15,10 +15,10 @@ public static List<SquareBlob> copy( List<SquareBlob> input ) {
 
 		// only add connections if they are in the sub-graph
 		for( int index = 0; index < input.size(); index++ ) {
-			SquareBlob in = input.get(index);
-			SquareBlob out = output.get(index);
+			QuadBlob in = input.get(index);
+			QuadBlob out = output.get(index);
 
-			for( SquareBlob c : in.conn ) {
+			for( QuadBlob c : in.conn ) {
 				int i = input.indexOf(c);
 				if( i >= 0) {
 					out.conn.add(output.get(i));
