diff --git a/old_test.java b/new_test.java
index 58796f8..5360b9f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	public void copy() {
-		List<SquareBlob> all = new ArrayList<SquareBlob>();
+		List<QuadBlob> all = new ArrayList<QuadBlob>();
 		all.add(createSquare(50,60,10,12));
 		all.add(createSquare(51,61,11,13));
 		all.add(createSquare(52, 62, 12, 14));
@@ -11,12 +11,12 @@
 		connect(1,3,all);
 		connect(2,3,all);
 
-		List<SquareBlob> sub = new ArrayList<SquareBlob>();
+		List<QuadBlob> sub = new ArrayList<QuadBlob>();
 		sub.add( all.get(0));
 		sub.add( all.get(1));
 		sub.add( all.get(2));
 
-		List<SquareBlob> found = ConnectGridSquares.copy(sub);
+		List<QuadBlob> found = ConnectGridSquares.copy(sub);
 
 		checkSquare(found.get(0),50,60,10,12,2);
 		checkSquare(found.get(1),51,61,11,13,1);
