diff --git a/old_test.java b/new_test.java
index d34a5e5..daf63b8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,6 +18,15 @@
         assertSame(sc, acceptKey.channel());
 
         //test that sc.register invokes Selector.register()
-        sc.register(acceptSelector2, SelectionKey.OP_READ, null);
+        acceptKey = sc.register(acceptSelector2, SelectionKey.OP_READ, null);
         assertTrue(((MockAbstractSelector)acceptSelector2).isRegisterCalled);
+
+        // Regression test to ensure acceptance of a selector with empty
+        // interest set.
+        SocketChannel channel = SocketChannel.open();
+        channel.configureBlocking(false);
+        Selector selector = Selector.open();
+        channel.register(selector, 0);
+        selector.close();
+        channel.close();
     }
\ No newline at end of file
