diff --git a/old_product.java b/new_product.java
index 075b363..9a752b1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-public final SelectionKey register(Selector selector, int interestSet,
+@Override
+    public final SelectionKey register(Selector selector, int interestSet,
             Object attachment) throws ClosedChannelException {
         if (!isOpen()) {
             throw new ClosedChannelException();
@@ -19,10 +20,6 @@ public final SelectionKey register(Selector selector, int interestSet,
                 // throw NPE exactly to keep consistency
                 throw new NullPointerException();
             }
-            if (0 == interestSet) {
-                // throw ISE exactly to keep consistency
-                throw new IllegalSelectorException();
-            }
             SelectionKey key = keyFor(selector);
             if (null == key) {
                 key = ((AbstractSelector) selector).register(this, interestSet,
