diff --git a/old_test.java b/new_test.java
index 1041183..8820612 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,11 @@
 @Test
   public void testUpdateHBaseMaster_Cluster() throws InterruptedException {
+    setUpPortState(true);
+    scaner.setDefaultScanTimeoutMsc(scanTimeOut);
+    scaner.setMaxAttempts(maxAttempts);
+    scaner.setRescanTimeoutMsc(reScanTimeOut);
     log.debug("updateHBaseMaster - pass Cluster");
     scaner.updateHBaseMaster(cluster);
-    Thread.sleep(2000);
+    Thread.sleep(1000);
     assertEquals("active", serviceComponentHost.convertToResponse().getHa_status());
   }
\ No newline at end of file
