diff --git a/old_product.java b/new_product.java
index 60c5e17..cbcd96a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
 public void updateHBaseMaster(Cluster cluster) {
-    if (LOG.isDebugEnabled()) {
-      LOG.debug("HBaseMasterPortScaner start scanning for cluster " + cluster.getClusterName());
-    }
     synchronized (wakeupSyncObject) {
       collectServiceComponentHostsForCluster(cluster);
-      activeAwakeRequest = true;
-      wakeupSyncObject.notify();
+      if(!componentHostMap.isEmpty()){
+        LOG.debug("HBaseMasterPortScanner start scanning for cluster " + cluster.getClusterName());
+        activeAwakeRequest = true;
+        wakeupSyncObject.notify();
+      } else LOG.debug("No for scan (with HBaseMaster component)");
     }
   }
\ No newline at end of file
