diff --git a/old_test.java b/new_test.java
index 32405ed..f81693c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
-@Test public void objectTest() throws QueryException {
+@Test public void objectTest() throws QueryIOException {
     parse("{ }", JsonSpec.RFC4627);
     parse("{ \"\": 42 }", JsonSpec.RFC4627);
     parse("{ a : 42, b: 23 }", "{ \"a\": 42, \"b\": 23 }", JsonSpec.LIBERAL);
