diff --git a/old_test.java b/new_test.java
index c317900..f3e439a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,5 +12,15 @@
         unsafe = null;
         result = MarkupChecker.stripAllTags(unsafe);
         assertNull(result);
-        
+
+        unsafe = "Johnson & Johnson <>";
+        expResult = "Johnson & Johnson <>";
+        result = MarkupChecker.stripAllTags(unsafe);
+        assertEquals(expResult, result);
+
+        unsafe = "Johnson && Johnson <&>&";
+        expResult = "Johnson && Johnson <&>&";
+        result = MarkupChecker.stripAllTags(unsafe);
+        assertEquals(expResult, result);
+
     }
\ No newline at end of file
