diff --git a/old_product.java b/new_product.java
index e7001b1..b300b3f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
-public static String stripAllTags(String unsafe){
-        
-        if (unsafe == null){
+public static String stripAllTags(String unsafe) {
+
+        if (unsafe == null) {
             return null;
-        }        
-        
-        return Jsoup.clean(unsafe, Whitelist.none());
-        
+        }
+
+        return Parser.unescapeEntities(Jsoup.clean(unsafe, Whitelist.none()), true);
+
     }
\ No newline at end of file
