diff --git a/old_product.java b/new_product.java
index a213ad1..331948c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 public void coverage() {
     BusinessDayAdjustment bda = BusinessDayAdjustment.of(BusinessDayConventions.FOLLOWING, HolidayCalendars.SAT_SUN);
     PeriodicScheduleDefn defn = PeriodicScheduleDefn.of(
-        AdjustableDate.of(date(2014, JUNE, 4), bda),
-        AdjustableDate.of(date(2014, SEPTEMBER, 17), bda),
+        date(2014, JUNE, 4),
+        date(2014, SEPTEMBER, 17),
         P1M, 
         bda,
         SHORT_INITIAL,
