diff --git a/old_test.java b/new_test.java
index 59abe68..50b074e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,11 +7,5 @@
         } catch (IOException ex) {
             fail(ex.getMessage());
         }
-        try {
-            instance.close();
-        } catch (CorruptIndexException ex) {
-            fail(ex.getMessage());
-        } catch (IOException ex) {
-            fail(ex.getMessage());
-        }
+        instance.close();
     }
\ No newline at end of file
