diff --git a/old_test.java b/new_test.java
index 593c65e..9c9c16d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,11 @@
 @Test(groups={"tck"})
     public void test_registerProvider() {
-        Set<String> pre = ZoneRulesProvider.getAvailableGroupIds();
-        assertEquals(pre.contains("TEMPMOCK.-_"), false);
+        Set<String> pre = ZoneRulesProvider.getAvailableZoneIds();
+        assertEquals(pre.contains("FooLocation"), false);
         ZoneRulesProvider.registerProvider(new MockTempProvider());
-        assertEquals(pre.contains("TEMPMOCK.-_"), false);
-        Set<String> post = ZoneRulesProvider.getAvailableGroupIds();
-        assertEquals(post.contains("TEMPMOCK.-_"), true);
+        assertEquals(pre.contains("FooLocation"), false);
+        Set<String> post = ZoneRulesProvider.getAvailableZoneIds();
+        assertEquals(post.contains("FooLocation"), true);
 
-        assertEquals(ZoneRulesProvider.getProvider("TEMPMOCK.-_").getGroupId(), "TEMPMOCK.-_");
-        assertEquals(ZoneRulesProvider.getProvider("TEMPMOCK.-_").getRules("World%@~.-_", "1.-_").isFixedOffset(), true);
+        assertEquals(ZoneRulesProvider.getRules("FooLocation"), ZoneOffset.of("+01:45").getRules());
     }
\ No newline at end of file
