diff --git a/old_product.java b/new_product.java
index 5a098b9..e0b90a7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 public Builder<T> antialiasing(Antialiasing config)
 		{
 			checkForNull(config, "Antialiasing is null.");
+			updateStatus(Properties.RESIZER_FACTORY, Status.CANNOT_SET);
 			updateStatus(Properties.ANTIALIASING, Status.ALREADY_SET);
 			antialiasing = config;
 			return this;
