diff --git a/old_product.java b/new_product.java
index a78b0a7..d291c18 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,22 +4,21 @@ public void bind(final String name, final Object val) throws QueryException {
 
     // remove optional $ prefix
     String loc = name.indexOf('$') == 0 ? name.substring(1) : name;
-    String uri = "";
+    byte[] uri = EMPTY;
 
     // check for namespace declaration
     final Matcher m = BIND.matcher(loc);
     if(m.find()) {
-      uri = m.group(3);
-      if(uri == null) uri = m.group(5);
+      String u = m.group(3);
+      if(u == null) u = m.group(5);
+      uri = token(u);
       loc = m.group(6);
     }
     final byte[] ln = token(loc);
     if(loc.isEmpty() || !XMLToken.isNCName(ln)) return;
 
     // bind variable
-    final QNm nm = new QNm(ln, token(uri));
-    if(nm.ns()) nm.uri(ns.uri(nm.pref()));
-
+    final QNm nm = uri.length == 0 ? new QNm(ln, this) : new QNm(ln, uri);
     final Var gl = vars.global().get(nm);
     if(gl == null) {
       // assign new variable
