diff --git a/old_test.java b/new_test.java
index 762f985..5f662a8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,5 +12,5 @@
     equals("8", "-b{}a=8", "-q$a");
     equals("9", "-b'':a=9", "-q$a");
     equals("A", "-b{URI}a=A", "-qdeclare namespace a='URI'; $a:a");
-    equals("B", "-b'URI':a=B", "-qdeclare namespace a='URI'; $a:a");
+    equals("B", "-b'URI':b=B", "-qdeclare namespace b='URI'; $b:b");
   }
\ No newline at end of file
