diff --git a/old_product.java b/new_product.java
index 33891bd..0b2377a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-private boolean add(final BasicUpdate u, final boolean merged) {
-    if(u == null) return false;
+private void add(final BasicUpdate u, final boolean merged) {
+    if(u == null) return;
 
     if(!merged) {
       if(recent instanceof StructuralUpdate)
@@ -7,5 +7,4 @@ private boolean add(final BasicUpdate u, final boolean merged) {
       else val.add(recent);
     }
     recent = u;
-    return true;
   }
\ No newline at end of file
