diff --git a/old_test.java b/new_test.java
index fa28b03..b4aaec1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,10 +2,9 @@
   public void testAdd() {
     for(int i = 0; i < 4000; ++i) {
       final byte[] key = token("keyAdd" + i);
-      final int size = i;
       final long pointer = i + 5000L;
 
-      cache.add(key, size, pointer);
-      assertCacheEntry(key, size, pointer);
+      cache.add(key, i, pointer);
+      assertCacheEntry(key, i, pointer);
     }
   }
\ No newline at end of file
