diff --git a/old_test.java b/new_test.java
index c96ba1c..c1d3d02 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test(description = "GET /vApp/{id}/question", dependsOnMethods = { "testDeployVm" })
    public void testGetPendingQuestion() {
       // Power on Vm
-      vm = powerOnVm(vm.getHref());
+      vm = powerOnVm(vmUrn);
 
       // TODO how to test?
 
       // The method under test
-      VmPendingQuestion question = vmApi.getPendingQuestion(vm.getHref());
+      VmPendingQuestion question = vmApi.getPendingQuestion(vmUrn);
 
       // Check the retrieved object is well formed
       checkVmPendingQuestion(question);
