diff --git a/old_product.java b/new_product.java
index 18b4de3..804c9b0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public void lockReadAndCheckFullPath(Inode parent, String name) throws InvalidPathException {
+public void lockReadAndCheckNameAndParent(Inode parent, String name) throws InvalidPathException {
     lockReadAndCheckParent(parent);
     if (!mName.equals(name)) {
       unlockRead();
