diff --git a/old_test.java b/new_test.java
index 1715298..a10916c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,18 +1,13 @@
 @Test
-    public void testCompareTo() {
-        VulnerableSoftware vs = new VulnerableSoftware();
-        vs.setCpe("cpe:/a:mortbay:jetty:6.1.0");
-        VulnerableSoftware instance = new VulnerableSoftware();
-        instance.setCpe("cpe:/a:mortbay:jetty:6.1");
-        int expResult = -2;
-        int result = instance.compareTo(vs);
-        assertEquals(expResult, result);
-
-        vs = new VulnerableSoftware();
-        vs.setCpe("cpe:/a:yahoo:toolbar:3.1.0.20130813024103");
-        instance = new VulnerableSoftware();
-        instance.setCpe("cpe:/a:yahoo:toolbar:3.1.0.20130813024104");
-        expResult = 1;
-        result = instance.compareTo(vs);
-        assertEquals(expResult, result);
+    public void testCompareTo() throws CpeValidationException {
+        VulnerableSoftwareBuilder builder = new VulnerableSoftwareBuilder();
+        VulnerableSoftware obj = builder.part(Part.APPLICATION).vendor("mortbay").product("jetty").version("6.1.0").build();
+        VulnerableSoftware instance = builder.part(Part.APPLICATION).vendor("mortbay").product("jetty").version("6.1").build();
+        int result = instance.compareTo(obj);
+        assertTrue(result<0);
+        
+        obj = builder.part(Part.APPLICATION).vendor("yahoo").product("toolbar").version("3.1.0.20130813024103").build();
+        instance = builder.part(Part.APPLICATION).vendor("yahoo").product("toolbar").version("3.1.0.20130813024104").build();
+        result = instance.compareTo(obj);
+        assertTrue(result>0);
     }
\ No newline at end of file
