diff --git a/old_test.java b/new_test.java
index 3aa1a74..4640219 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,8 +3,8 @@
 		User user = UserData.randomUser();
 		UserDTO userDTO = BeanMapper.map(user, UserDTO.class);
 
-		IdResponse response = accountWebServiceClient.createUser(userDTO);
+		IdResult response = accountWebServiceClient.createUser(userDTO);
 		assertNotNull(response.getId());
-		GetUserResponse response2 = accountWebServiceClient.getUser(response.getId());
+		GetUserResult response2 = accountWebServiceClient.getUser(response.getId());
 		assertEquals(user.getLoginName(), response2.getUser().getLoginName());
 	}
\ No newline at end of file
