diff --git a/old_test.java b/new_test.java
index 2e160ee..53c044b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,12 +15,14 @@
         {
             /* replaceIf:primitiveKType KType */ Object /* end:replaceIf */ k = 
                 /* intrinsic:ktypecast */ rnd.nextInt(modulo);
-            assertEquals(deque.removeLastOccurrence(k), sequence.removeLastOccurrence(k) >= 0);
+            assertEquals(
+                deque.removeLastOccurrence(k) >= 0, 
+                sequence.removeLastOccurrence(k) >= 0);
         }
 
         assertListEquals(deque.toArray(), sequence.toArray());
 
-        assertFalse(deque.removeLastOccurrence(/* intrinsic:ktypecast */ (modulo + 1)));
+        assertTrue(0 > deque.removeLastOccurrence(/* intrinsic:ktypecast */ (modulo + 1)));
         deque.addFirst(/* intrinsic:ktypecast */ (modulo + 1));
-        assertTrue(deque.removeLastOccurrence(/* intrinsic:ktypecast */ (modulo + 1)));
+        assertTrue(0 <= deque.removeLastOccurrence(/* intrinsic:ktypecast */ (modulo + 1)));
     }
\ No newline at end of file
