diff --git a/old_test.java b/new_test.java
index 448a16d..a5f22c1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
 		Provider provider = new Provider();
 		provider.setIdentifier("test");
 		provider.setName("test provider");
-		AttributeType<Provider> providerAttributeType = service.getProviderAttributeType(3);
+		ProviderAttributeType providerAttributeType = service.getProviderAttributeType(3);
 		provider.setAttribute(createProviderAttribute(providerAttributeType, "bangalore"));
 		provider.setAttribute(createProviderAttribute(providerAttributeType, "chennai"));
 		Assert.assertEquals(1, provider.getAttributes().size());
