diff --git a/old_product.java b/new_product.java
index 68c63e7..14f9d3b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,12 @@
-@SuppressWarnings("unchecked")
-	@Override
-	public void setAttribute(AttrClass attribute) {
-		if (getAttributes() == null)
-			setAttributes(new LinkedHashSet<AttrClass>());
-		// TODO validate
+public void setAttribute(A attribute) {
+		if (getAttributes() == null) {
+			addAttribute(attribute);
+			return;
+		}
+		
 		if (getActiveAttributes(attribute.getAttributeType()).size() == 1) {
-			AttrClass existing = getActiveAttributes(attribute.getAttributeType()).get(0);
-			if (existing.getSerializedValue().equals(attribute.getSerializedValue())) {
+			A existing = getActiveAttributes(attribute.getAttributeType()).get(0);
+			if (existing.getValueReference().equals(attribute.getValueReference())) {
 				// do nothing, since the value is already as-specified
 			} else {
 				if (existing.getId() != null)
@@ -16,8 +16,9 @@
 				getAttributes().add(attribute);
 				attribute.setOwner(this);
 			}
+			
 		} else {
-			for (AttrClass existing : getActiveAttributes(attribute.getAttributeType()))
+			for (A existing : getActiveAttributes(attribute.getAttributeType()))
 				if (existing.getAttributeType().equals(attribute.getAttributeType()))
 					if (existing.getId() != null)
 						existing.setVoided(true);
