diff --git a/old_product.java b/new_product.java
index 5ebe494..8fac8fe 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
-    public XmlWriter<T> createWriter(PipelineOptions options) throws Exception {
+    public XmlWriter<T> createWriter() throws Exception {
       JAXBContext context;
       Marshaller marshaller;
       context = JAXBContext.newInstance(getSink().spec.getRecordClass());
