diff --git a/old_test.java b/new_test.java
index b44897d..4971a5e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,14 +5,14 @@
         XmlIO.<Bird>write()
             .withRecordClass(Bird.class)
             .withRootElement(testRootElement)
-            .toFilenamePrefix(testFilePrefix)
+            .to(testFilePrefix)
             .createSink()
             .createWriteOperation();
-    XmlWriter<Bird> writer = writeOp.createWriter(options);
+    XmlWriter<Bird> writer = writeOp.createWriter();
     Path outputPath = new File(testFilePrefix).toPath();
-    Path tempPath = new File(writer.getWriteOperation().getTemporaryDirectory()).toPath();
-    assertEquals(outputPath.getParent(), tempPath.getParent());
-    assertThat(
-        tempPath.getFileName().toString(), containsString("temp-beam-" + outputPath.getFileName()));
+    Path tempPath = new File(writer.getWriteOperation().getTemporaryDirectory().toString())
+        .toPath();
+    assertThat(tempPath.getParent(), equalTo(outputPath.getParent()));
+    assertThat(tempPath.getFileName().toString(), containsString("temp-beam-"));
     assertNotNull(writer.marshaller);
   }
\ No newline at end of file
