diff --git a/old_test.java b/new_test.java
index 7558539..6f57bfa 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,7 +16,7 @@
             KType k = cast(rnd.nextInt(modulo));
             assertEquals(
                 deque.removeFirstOccurrence(k) >= 0, 
-                sequence.removeFirstOccurrence(k) >= 0);
+                sequence.removeFirst(k) >= 0);
         }
 
         assertListEquals(deque.toArray(), sequence.toArray());
