diff --git a/old_product.java b/new_product.java
index 2a09165..f4209e9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,8 @@
 public static String hexStringToString(String hexString) {
+        // Odd length strings won't parse, return
+        if (hexString.length() % 2 > 0) {
+            return hexString;
+        }
         int charAsInt;
         StringBuilder sb = new StringBuilder();
         try {
