diff --git a/old_test.java b/new_test.java
index b9d17b1..1591c9c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,5 +2,8 @@
     public void testHexStringToString() {
         assertEquals("ABC", ParseUtil.hexStringToString("414243"));
         assertEquals("ab00cd", ParseUtil.hexStringToString("ab00cd"));
-        assertEquals("not hex", ParseUtil.hexStringToString("not hex"));
+        assertEquals("ab88cd", ParseUtil.hexStringToString("ab88cd"));
+        assertEquals("notHex", ParseUtil.hexStringToString("notHex"));
+        assertEquals("320", ParseUtil.hexStringToString("320"));
+        assertEquals("0", ParseUtil.hexStringToString("0"));
     }
\ No newline at end of file
