diff --git a/old_product.java b/new_product.java
index f19a105..e7d5ab7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@ public static File[] getExtensionFilesToLoad(ExtensionsConfig config)
       throw new ISE("Root extensions directory [%s] is not a directory!?", rootExtensionsDir);
     }
     File[] extensionsToLoad;
-    final List<String> toLoad = config.getLoadList();
+    final LinkedHashSet<String> toLoad = config.getLoadList();
     if (toLoad == null) {
       extensionsToLoad = rootExtensionsDir.listFiles();
     } else {
