diff --git a/old_test.java b/new_test.java
index 1c5c987..b196258 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,9 +5,9 @@
     final ExtensionsConfig config = new ExtensionsConfig()
     {
       @Override
-      public List<String> getLoadList()
+      public LinkedHashSet<String> getLoadList()
       {
-        return Arrays.asList("mysql-metadata-storage", "druid-kafka-eight");
+        return Sets.newLinkedHashSet(Arrays.asList("mysql-metadata-storage", "druid-kafka-eight"));
       }
 
       @Override
