diff --git a/old_test.java b/new_test.java
index 1f7af9b..b06580f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,8 +3,8 @@
         MutableSparseVector simple = simpleVector();
         simple.addChannel(fooSymbol);
         simple.channel(fooSymbol).set(3, 4.5);
-        simple.alwaysAddChannel(fooSymbol);
-        simple.alwaysAddChannel(barSymbol);
+        simple.getOrAddChannel(fooSymbol);
+        simple.getOrAddChannel(barSymbol);
         assert(simple.channel(barSymbol).isEmpty());
         simple.channel(barSymbol).set(3, 33);
         assertThat(simple.channel(fooSymbol).get(3), closeTo(4.5));
