diff --git a/old_product.java b/new_product.java
index 3f6c2f1..b111899 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public MutableSparseVector alwaysAddChannel(Symbol channelSymbol) {
+public MutableSparseVector getOrAddChannel(Symbol channelSymbol) {
         MutableSparseVector chan = channelMap.get(channelSymbol);
         if (chan == null) {
             chan = addChannel(channelSymbol);
