diff --git a/old_product.java b/new_product.java
index 228616b..8b06ee5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,15 +5,20 @@ public String getS3SiteXmlsAsCsv() {
         if (s3CoreSiteXml != null) {
             csv.append(s3CoreSiteXml);
         }
-        csv.append(",");
-
+        
         if (s3MapredSiteXml != null) {
-            csv.append(s3MapredSiteXml);
+        	csv.append(",");
+        	csv.append(s3MapredSiteXml);
         }
-        csv.append(",");
-
+        
         if (s3HdfsSiteXml != null) {
-            csv.append(s3HdfsSiteXml);
+        	csv.append(",");
+        	csv.append(s3HdfsSiteXml);
+        }
+        
+        if (s3YarnSiteXml != null) {
+        	csv.append(",");
+        	csv.append(s3YarnSiteXml);
         }
 
         return csv.toString();
