diff --git a/old_test.java b/new_test.java
index c29d9dc..293aee6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,5 +3,6 @@
         cc.setS3CoreSiteXml("core");
         cc.setS3MapredSiteXml("mapred");
         cc.setS3HdfsSiteXml("hdfs");
-        Assert.assertEquals(cc.getS3SiteXmlsAsCsv(), "core,mapred,hdfs");
+        cc.setS3YarnSiteXml("yarn");
+        Assert.assertEquals(cc.getS3SiteXmlsAsCsv(), "core,mapred,hdfs,yarn");
     }
\ No newline at end of file
