diff --git a/old_test.java b/new_test.java
index 1efdf35..19ccc58 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testGetResource(){
-        LOGGER.info(ClassLoaderUtil.getResource("") + "");
-        LOGGER.info("" + ClassLoaderUtil.getResource("com"));
-        ClassLoaderUtil.getResource("jstl-1.2", this.getClass());
+        LOGGER.debug(ClassLoaderUtil.getResource("") + "");
+        LOGGER.debug("" + ClassLoaderUtil.getResource("com"));
+        ClassLoaderUtil.getResourceInAllClassLoader("jstl-1.2", this.getClass());
     }
\ No newline at end of file
