diff --git a/old_product.java b/new_product.java
index 5bed44f..ca13ce5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,3 @@
 public static URL getResource(String resourceName){
-        ClassLoader classLoader = getClassLoaderByClass(ClassLoaderUtil.class);
-        return getResource(classLoader, resourceName);
+        return getResource(getClassLoaderByClass(ClassLoaderUtil.class), resourceName);
     }
\ No newline at end of file
