diff --git a/old_test.java b/new_test.java
index 2bf5667..0ecf4f0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,22 +1,17 @@
 @Test(enabled = true, description = "Get accountNet by http")
   public void getAccountNet() {
-    response = HttpMethed.getAccountNet(httpnode,fromAddress);
-    try {
-      entity = response.getEntity();
-    } catch (Exception e) {
-      e.printStackTrace();
-    }
+    response = HttpMethed.getAccountNet(httpnode, fromAddress);
     logger.info("code is " + response.getStatusLine().getStatusCode());
-    Assert.assertEquals(response.getStatusLine().getStatusCode(),200);
+    Assert.assertEquals(response.getStatusLine().getStatusCode(), 200);
     responseContent = HttpMethed.parseResponseContent(response);
     HttpMethed.printJsonContent(responseContent);
-    for (String str:responseContent.keySet()) {
+    for (String str : responseContent.keySet()) {
       if (str.equals("freeNetLimit")) {
-        Assert.assertEquals(responseContent.get(str),5000);
+        Assert.assertEquals(responseContent.get(str), 5000);
       }
       if (str.equals("TotalNetLimit")) {
-        Assert.assertEquals(responseContent.get(str),43200000000L);
+        Assert.assertEquals(responseContent.get(str), 43200000000L);
       }
     }
-    Assert.assertTrue(responseContent.size() >= 4);
+    Assert.assertTrue(responseContent.size() >= 3);
   }
\ No newline at end of file
