diff --git a/old_test.java b/new_test.java
index 3f1b2da..7d33760 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,5 +5,5 @@
     error(_CLIENT_CLOSE.args("xs:anyURI('unknown')"), Err.BXCL_NOTAVL);
     // BXCL0002: session has already been closed
     error(conn() + " ! (" + _CLIENT_CLOSE.args(" .") + ", " +
-        _CLIENT_CLOSE.args(" .") + ")", Err.BXCL_NOTAVL);
+        _CLIENT_CLOSE.args(" .") + ')', Err.BXCL_NOTAVL);
   }
\ No newline at end of file
