diff --git a/old_product.java b/new_product.java
index 8739a3a..8007d99 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@ public void close() {
       if(obj == null) continue;
       try {
         if(obj instanceof Connection) ((Connection) obj).close();
-        else ((PreparedStatement) obj).close();
+        else ((Statement) obj).close();
       } catch(final SQLException ex) {
         Util.debug(ex);
       }
