diff --git a/old_product.java b/new_product.java
index 0c9f80c..90fe302 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public static List<String> getGroups(String userName) throws IOException {
-    CachedGroupMapping cachedGroupMapping = GroupMappingService.Factory.getCachedGroupMapping();
-    return cachedGroupMapping.getGroups(userName);
+    GroupMappingService groupMappingService = GroupMappingService.Factory.get();
+    return groupMappingService.getGroups(userName);
   }
\ No newline at end of file
