diff --git a/old_test.java b/new_test.java
index ec7ffba..6291769 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,8 +10,7 @@
     PowerMockito.when(cachedGroupService.getGroups(Mockito.anyString())).thenReturn(
         Lists.newArrayList(userGroup1, userGroup2));
     PowerMockito.mockStatic(GroupMappingService.Factory.class);
-    Mockito.when(GroupMappingService.Factory.getCachedGroupMapping())
-        .thenReturn(cachedGroupService);
+    Mockito.when(GroupMappingService.Factory.get()).thenReturn(cachedGroupService);
 
     List<String> groups = CommonUtils.getGroups(userName);
     Assert.assertEquals(Arrays.asList(userGroup1, userGroup2), groups);
