diff --git a/old_test.java b/new_test.java
index 888decd..ba096f1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void test_get_TemporalField() {
-        assertEquals(TEST_2008_06.get(ChronoField.YEAR), 2008);
-        assertEquals(TEST_2008_06.get(ChronoField.MONTH_OF_YEAR), 6);
-        assertEquals(TEST_2008_06.get(ChronoField.YEAR_OF_ERA), 2008);
-        assertEquals(TEST_2008_06.get(ChronoField.ERA), 1);
+        assertEquals(TEST_2008_06.get(YEAR), 2008);
+        assertEquals(TEST_2008_06.get(MONTH_OF_YEAR), 6);
+        assertEquals(TEST_2008_06.get(YEAR_OF_ERA), 2008);
+        assertEquals(TEST_2008_06.get(ERA), 1);
     }
\ No newline at end of file
