diff --git a/old_test.java b/new_test.java
index 0b4b711..d568d63 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,12 @@
     final String fun = check(Function.REPLACEDOC);
 
     new Add("etc/test/input.xml", null, "test").execute(CONTEXT);
-    query(fun + "('db', 'test/input.xml', document { <root/> })");
+
+    query(fun + "('db/test/input.xml', document { <root/> })");
     query("count(collection('db/test/input.xml')/html) eq 0", "true");
     query("count(collection('db/test/input.xml')/root) eq 1", "true");
+
+    query(fun + "('db/test/input.xml', 'etc/test/input.xml')");
+    query("count(collection('db/test/input.xml')/html) eq 1", "true");
+    query("count(collection('db/test/input.xml')/root) eq 0", "true");
   }
\ No newline at end of file
