diff --git a/old_test.java b/new_test.java
index 52b9035..072f71b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
     public void testParseDateWithLeniency() throws Exception {
-        GregorianCalendar cal = new GregorianCalendar(1998, 6, 30);
-        String dateStr = "02 942, 1996";
-        String[] parsers = new String[] {"MM DDD, yyyy"};
+        final GregorianCalendar cal = new GregorianCalendar(1998, 6, 30);
+        final String dateStr = "02 942, 1996";
+        final String[] parsers = new String[] {"MM DDD, yyyy"};
         
         Date date = DateUtils.parseDate(dateStr, parsers);
         assertEquals(cal.getTime(), date);
@@ -10,5 +10,5 @@
         try {
             date = DateUtils.parseDateStrictly(dateStr, parsers);
             fail();
-        } catch (ParseException ex) {}
+        } catch (final ParseException ex) {}
     }
\ No newline at end of file
