diff --git a/old_product.java b/new_product.java
index 65db2b2..cdcb0b9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,8 +12,8 @@ private static Date parseDateWithLeniency(
         }
         
         parser.setLenient(lenient);
-        ParsePosition pos = new ParsePosition(0);
-        for (String parsePattern : parsePatterns) {
+        final ParsePosition pos = new ParsePosition(0);
+        for (final String parsePattern : parsePatterns) {
 
             String pattern = parsePattern;
 
@@ -31,7 +31,7 @@ private static Date parseDateWithLeniency(
                 str2 = str.replaceAll("([-+][0-9][0-9]):([0-9][0-9])$", "$1$2"); 
             }
 
-            Date date = parser.parse(str2, pos);
+            final Date date = parser.parse(str2, pos);
             if (date != null && pos.getIndex() == str2.length()) {
                 return date;
             }
