diff --git a/old_test.java b/new_test.java
index cadd696..78658b9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,4 +17,8 @@
     error(_XQUERY_PARSE.args("1+"), CALCEXPR);
     query("\n\ntry {" + _XQUERY_PARSE.args("1+",
         " map{'pass':true()}") + "} catch * { $err:line-number }", "1");
+
+    query("contains(try {" + _XQUERY_PARSE.args("1+",
+        " map { 'base-uri': 'XXXX', 'pass': 'true' }") + "} catch * { $err:module }, 'XXXX')",
+        "true");
   }
\ No newline at end of file
