diff --git a/old_test.java b/new_test.java
index 0c5820f..25f3bb1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,6 @@
         AssociativeArray a2 = new AssociativeArray(getMap2());
         Map<Object, Double> columnWeights = getWeights();
         double expResult = 2.449489742783178;
-        double result = Distance.euclideanWeighhted(a1, a2, columnWeights);
+        double result = Distance.euclideanWeighted(a1, a2, columnWeights);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
