diff --git a/old_test.java b/new_test.java
index 0b1f7a5..1c533ce 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testOnCreateOrUpdateCommonEntityFields() throws GenieException {
+    public void testOnCreateOrUpdateCommonEntityFields() throws GeniePreconditionException {
         this.c = new CommonEntityFields(NAME, USER, VERSION);
         this.c.onCreateOrUpdateCommonEntityFields();
     }
\ No newline at end of file
