diff --git a/old_product.java b/new_product.java
index 4e094c9..8da4d23 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @PrePersist
     @PreUpdate
-    protected void onCreateOrUpdateCommonEntityFields() throws GenieException {
+    protected void onCreateOrUpdateCommonEntityFields() throws GeniePreconditionException {
         this.validate(this.name, this.user, this.version, null);
     }
\ No newline at end of file
