diff --git a/old_test.java b/new_test.java
index 1dc294f..89adc78 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,8 +3,7 @@
         NvdCveInfo cve = new NvdCveInfo();
         cve.setId("modified");
         cve.setNeedsUpdate(true);
-        cve.setUrl(getSettings().getString(Settings.KEYS.CVE_MODIFIED_20_URL));
-        cve.setOldSchemaVersionUrl(getSettings().getString(Settings.KEYS.CVE_MODIFIED_12_URL));
+        cve.setUrl(getSettings().getString(Settings.KEYS.CVE_MODIFIED_JSON));
         ExecutorService processExecutor = null;
         CveDB cveDB = null;
         DownloadTask instance = new DownloadTask(cve, processExecutor, cveDB, getSettings());
