diff --git a/old_product.java b/new_product.java
index 4b1e1da..3c15ab5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,7 +11,7 @@
             } catch (Throwable ex) {
                 LOGGER.warn("An unexpected error occurred during analysis of '{}' ({}): {}",
                         dependency.getActualFilePath(), analyzer.getName(), ex.getMessage());
-                LOGGER.debug("", ex);
+                LOGGER.error("", ex);
                 exceptions.add(ex);
             }
         }
