diff --git a/old_test.java b/new_test.java
index 6c7b849..64617ac 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,18 +16,21 @@
         queryParams.add(new Parameter("name", docName));
         queryParams.add(new Parameter("limit", "5"));
         mockServer.reset().when(request()
-                .withPath(DOCUMENTS_PATH).withQueryStringParameters(queryParams))
+                .withPath(DocumentConversion.DOCUMENTS_PATH).withQueryStringParameters(queryParams))
                 .respond(response().withBody(GsonSingleton.getGson().toJson(docCollWithQueryResponse))
                 );
-        DocumentCollection docCollWithQuery = service.getDocumentCollection(null, 5, docName, null, null);
+        Map<String, Object> docListParams = new HashMap<String, Object>();
+        docListParams.put(DocumentConversion.LIMIT, 5);
+        docListParams.put(DocumentConversion.NAME, docName);
+        DocumentCollection docCollWithQuery = service.getDocumentCollection(docListParams);
         Assert.assertNotNull(docCollWithQuery);
         Assert.assertEquals(docCollWithQuery.toString(), docCollWithQueryResponse.toString());
 
         // Call get document
-        mockServer.when(request().withPath(DOCUMENTS_PATH + "/" + docId)).respond(
+        mockServer.when(request().withPath(DocumentConversion.DOCUMENTS_PATH + "/" + docId)).respond(
                 response().withBody(docContent)
         );
-        String document1 = service.getDocument(docId);
+        String document1 = ConversionUtils.writeInputStreamToString(service.getDocument(docId));
         Assert.assertNotNull(document1);
         Assert.assertEquals(document1, docContent);
     }
\ No newline at end of file
