diff --git a/old_product.java b/new_product.java
index 34da70a..ec0db9d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,8 +5,8 @@
             LocationHeaderTest.executor.submit(new Runnable() {
                 @Override
                 public void run() {
-                    URI uri = getUriBuilder().segment("executorAsync").build();
-                    Response result = Response.created(uri).build();
+                    final URI uri = getUriBuilder().segment("executorAsync").build();
+                    final Response result = Response.created(uri).build();
                     asyncResponse.resume(result);
                     if (!uriInfo.getAbsolutePath().equals(result.getLocation())) {
                         executorComparisonFailed.set(true);
