diff --git a/old_product.java b/new_product.java
index 5ba4b08..cea9a82 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,15 +1,14 @@
-public boolean removeValue(String name,Object value)
+public boolean removeValue(String name,V value)
     {
-        Object lo = get(name);
-        Object ln=lo;
-        int s=LazyList.size(lo);
-        if (s>0)
-        {
-            ln=LazyList.remove(lo,value);
-            if (ln==null)
-                remove(name);
-            else
-                put(name, ln);
+        List<V> lo = get(name);
+        if((lo == null)||(lo.isEmpty())) {
+            return false;
         }
-        return LazyList.size(ln)!=s;
+        boolean ret = lo.remove(value);
+        if(lo.isEmpty()) {
+            remove(name);
+        } else {
+            put(name,lo);
+        }
+        return ret;
     }
\ No newline at end of file
