diff --git a/old_test.java b/new_test.java
index 2358a7c..4810cf0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,7 +13,7 @@
         @Cleanup
         TestContext context = new TestContext(config);
         context.storage.create(context.segmentAggregator.getMetadata().getName(), TIMEOUT).join();
-        context.segmentAggregator.initialize(TIMEOUT).join();
+        context.segmentAggregator.initialize(TIMEOUT, executorService()).join();
 
         @Cleanup
         ByteArrayOutputStream writtenData = new ByteArrayOutputStream();
@@ -57,7 +57,7 @@
         Assert.assertEquals("Unexpected number of bytes flushed to Storage.", expectedData.length, storageInfo.getLength());
         Assert.assertTrue("Segment is not sealed in storage post flush.", storageInfo.isSealed());
         Assert.assertTrue("Segment is not marked in metadata as sealed in storage post flush.", context.segmentAggregator.getMetadata().isSealedInStorage());
-        context.storage.read(context.segmentAggregator.getMetadata().getName(), 0, actualData, 0, actualData.length, TIMEOUT).join();
+        context.storage.read(InMemoryStorage.newHandle(context.segmentAggregator.getMetadata().getName(), false), 0, actualData, 0, actualData.length, TIMEOUT).join();
 
         Assert.assertArrayEquals("Unexpected data written to storage.", expectedData, actualData);
     }
\ No newline at end of file
