diff --git a/old_product.java b/new_product.java
index 8acf5af..3e42963 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,17 +1,4 @@
-protected String getNextNodeId( final String nodeId, final Collection<String> nodeIds ) {
-
-        /* create a list of nodeIds to check randomly
-         */
-        final List<String> otherNodeIds = new ArrayList<String>( nodeIds );
-        otherNodeIds.remove( nodeId );
-
-        while ( !otherNodeIds.isEmpty() ) {
-            final String nodeIdToCheck = otherNodeIds.get( _random.nextInt( otherNodeIds.size() ) );
-            if ( isNodeAvailable( nodeIdToCheck ) ) {
-                return nodeIdToCheck;
-            }
-            otherNodeIds.remove( nodeIdToCheck );
-        }
-
-        return null;
+@Nonnull
+    public String getNextNodeId( @Nonnull final String nodeId ) {
+        return _nodeIds.getNextNodeId( nodeId );
     }
\ No newline at end of file
