diff --git a/old_test.java b/new_test.java
index ebf8619..07c091a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
         final String nodeId1 = "n1";
         final String nodeId2 = "n2";
         final NodeIdService cut = new NodeIdService( createNodeAvailabilityCache(),
-                Arrays.asList( nodeId1 ), Arrays.asList( nodeId2 ) );
+                NodeIdList.create( nodeId1 ), Arrays.asList( nodeId2 ) );
 
         final String actual = cut.getAvailableNodeId( nodeId2 );
         assertEquals( nodeId1, actual, "The regular node is not chosen" );
