diff --git a/old_test.java b/new_test.java
index 1e92375..4f39574 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     @RunTestInLooperThread
-    public void removeChangeListeners() {
+    public void removeAllChangeListeners() {
         final Realm realm = looperThread.realm;
         realm.beginTransaction();
         Dog dog = realm.createObject(Dog.class);
@@ -12,7 +12,7 @@
                 assertTrue(false);
             }
         });
-        dog.removeChangeListeners();
+        dog.removeAllChangeListeners();
 
         realm.beginTransaction();
         Dog sameDog = realm.where(Dog.class).equalTo(Dog.FIELD_AGE, 13).findFirst();
