diff --git a/old_product.java b/new_product.java
index 4f3f76d..c5b7efe 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@ public Connection getConnection() throws SQLException {
 		if (!connectionPartition.isUnableToCreateMoreTransactions() ){
 			try{
 				this.connectionsObtainedLock.lock();
-	 				maybeSignalForMoreConnections(connectionPartition);
+	 			maybeSignalForMoreConnections(connectionPartition);
 			} finally {
 				this.connectionsObtainedLock.unlock(); 
 			}
@@ -51,5 +51,11 @@ public Connection getConnection() throws SQLException {
 		}
 		result.setOriginatingPartition(connectionPartition);
 		result.renewConnection();
+		
+		// Give an application the chance to do something with it.
+		if (result.getConnectionHook() != null){
+			result.getConnectionHook().onCheckOut(result);
+		}
+
 		return result;
 	}
\ No newline at end of file
