diff --git a/old_test.java b/new_test.java
index 07e322f..89a7c7e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     {
         System.out.println("cluster");
         NaiveKMeans kMeans = new NaiveKMeans(new EuclideanDistance(), SeedSelectionMethods.SeedSelection.FARTHEST_FIRST);
-        int[] assignment = new int[easyData10.getSampleSize()];
+        int[] assignment = new int[easyData10.size()];
         kMeans.cluster(easyData10, null, 10, seeds, assignment, true, true, true, null);
         List<List<DataPoint>> clusters = KClustererBase.createClusterListFromAssignmentArray(assignment, easyData10);
         assertEquals(10, clusters.size());
