diff --git a/old_product.java b/new_product.java
index 2344256..9f04617 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
 @ServiceMethod(returns = ReturnType.SINGLE)
-    public ConfigurationSetting setReadOnly(String key, String label) {
-        return setReadOnlyWithResponse(new ConfigurationSetting().setKey(key).setLabel(label), Context.NONE).getValue();
+    public ConfigurationSetting setReadOnly(String key, String label, boolean isReadOnly) {
+        return setReadOnlyWithResponse(new ConfigurationSetting().setKey(key).setLabel(label), isReadOnly, Context.NONE)
+            .getValue();
     }
\ No newline at end of file
