diff --git a/old_test.java b/new_test.java
index b74a88f..d38e565 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
         lockUnlockRunner((expected) -> {
             // read-only setting
             client.addConfigurationSettingWithResponse(expected, Context.NONE);
-            client.setReadOnly(expected.getKey(), expected.getLabel());
+            client.setReadOnly(expected.getKey(), expected.getLabel(), true);
 
             // unsuccessfully delete
             assertRestException(() ->
