diff --git a/old_product.java b/new_product.java
index 338a59a..7a382ee 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,8 @@
 @POST
   @Path(PATH_PARAM + OPEN_FILE)
-  @ReturnType("java.lang.Integer")
   @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Opens the given path for reading, use the id with the stream api",
+      response = java.lang.Integer.class)
   public Response openFile(@PathParam("path") final String path, final OpenFileOptions options) {
     return RestUtils.call(new RestUtils.RestCallable<Integer>() {
       @Override
