diff --git a/old_test.java b/new_test.java
index 4f1a469..788ab70 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
     query(_DB_EXISTS.args(dbname), "false");
 
     // invalid name
-    error(_DB_DROP.args(" ''"), Err.BXDB_NAME);
+    error(_DB_DROP.args(" ''"), Err.BXDB_NAME_X);
     // try to drop non-existing DB
-    error(_DB_DROP.args(dbname), Err.BXDB_WHICH);
+    error(_DB_DROP.args(dbname), Err.BXDB_WHICH_X);
   }
\ No newline at end of file
