diff --git a/old_product.java b/new_product.java
index 7bc7354..f29ac0e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 private Item drop(final QueryContext qc) throws QueryException {
-    final String name = string(checkStr(exprs[0], qc));
-    if(!Databases.validName(name)) throw BXDB_NAME.get(info, name);
-    if(!qc.context.globalopts.dbexists(name)) throw BXDB_WHICH.get(info, name);
+    final String name = string(toToken(exprs[0], qc));
+    if(!Databases.validName(name)) throw BXDB_NAME_X.get(info, name);
+    if(!qc.context.globalopts.dbexists(name)) throw BXDB_WHICH_X.get(info, name);
     qc.resources.updates().add(new DBDrop(name, info, qc), qc);
     return null;
   }
\ No newline at end of file
