diff --git a/old_test.java b/new_test.java
index 9d8ab31..d91c53f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,7 @@
 						"	}\n"+
 				"}\n");
 		int[] loc = findCompletionLocation(unit, "Objec");
-		CompletionList list = server.completion(JsonMessageHelper.getParams(createCompletionRequest(unit, loc[0], loc[1]))).join();
+		CompletionList list = server.completion(JsonMessageHelper.getParams(createCompletionRequest(unit, loc[0], loc[1]))).join().getRight();
 		assertNotNull(list);
 		assertFalse("No proposals were found",list.getItems().isEmpty());
 		for ( CompletionItem item : list.getItems()) {
