diff --git a/old_product.java b/new_product.java
index 7cd5389..100792b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,12 +8,12 @@ public List<SynonymResult> getSynonyms(SynonymOptions options) {
     final Request request =
         RequestBuilder
             .post(PATH_SYNONYM)
-            .withBodyContent(GsonSingleton.getGson().toJson(options),
+            .withBodyContent(GsonSingleton.getGsonWithoutPrettyPrinting().toJson(options),
                 HttpMediaType.APPLICATION_JSON).build();
 
     final Response response = execute(request);
     final String synonymResultJson = ResponseUtil.getString(response);
     final List<SynonymResult> synonyms =
-        GsonSingleton.getGson().fromJson(synonymResultJson, synonymListType);
+        GsonSingleton.getGsonWithoutPrettyPrinting().fromJson(synonymResultJson, synonymListType);
     return synonyms;
   }
\ No newline at end of file
