diff --git a/old_test.java b/new_test.java
index ee9b6a1..df9aec7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -119,7 +119,7 @@
         .respond(
             response().withHeaders(
                 new Header(HttpHeaders.Names.CONTENT_TYPE, HttpMediaType.APPLICATION_JSON))
-                .withBody(GsonSingleton.getGson().toJson(response)));
+                .withBody(GsonSingleton.getGsonWithoutPrettyPrinting().toJson(response)));
 
     final List<SynonymResult> synonyms = service.getSynonyms(options);
 
