diff --git a/old_test.java b/new_test.java
index 0be1d9b..90512b7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
     public void testLimit() {
         ew.where("name={0}", "'123'").orderBy("id", false);
-        ew.limit(0, 3);
+        ew.last("limit 1,2");
         String sqlSegment = ew.toString();
         System.err.println("testLimit = " + sqlSegment);
-        Assert.assertEquals("WHERE (name=?)\nORDER BY id DESC LIMIT 0, 3 ", sqlSegment);
+        Assert.assertEquals("WHERE (name=?)\nORDER BY id DESC limit 1,2", sqlSegment);
     }
\ No newline at end of file
