diff --git a/old_product.java b/new_product.java
index 7f834eb..0c6ca93 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,6 +7,7 @@ public int rebalanceNode(RebalancePartitionsInfo stealInfo) {
                                                                                                                 .addAllUnbalancedStore(stealInfo.getUnbalancedStoreList())
                                                                                                                 .addAllDeletePartitions(stealInfo.getDeletePartitionsList())
                                                                                                                 .addAllStealMasterPartitions(stealInfo.getStealMasterPartitions())
+                                                                                                                .addAllStoreToRODir(decodeROStoreVersionDirMap(stealInfo.getStoreToRODir()))
                                                                                                                 .build();
         VAdminProto.VoldemortAdminRequest adminRequest = VAdminProto.VoldemortAdminRequest.newBuilder()
                                                                                           .setType(VAdminProto.AdminRequestType.INITIATE_REBALANCE_NODE)
