diff --git a/old_test.java b/new_test.java
index de0ce7d..6969486 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,6 +20,7 @@
                                                                         rebalancePartitionList,
                                                                         rebalancePartitionList,
                                                                         Arrays.asList(testStoreName),
+                                                                        new HashMap<String, String>(),
                                                                         0);
         int asyncId = adminClient.rebalanceNode(stealInfo);
         assertNotSame("Got a valid rebalanceAsyncId", -1, asyncId);
