diff --git a/old_test.java b/new_test.java
index 842e5d8..7b46f1a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,8 @@
 @Test
-	public void isObsGrouping_shouldIgnoreVoidedObs() throws Exception {
+	public void isObsGrouping_shouldIncludeVoidedObs() throws Exception {
 		Obs parent = new Obs(5);
 		Obs child = new Obs(33);
 		child.setVoided(true);
 		parent.addGroupMember(child);
-		assertFalse("When checking for Obs grouping, should ignore voided Obs", parent.isObsGrouping());
-		child = new Obs(66); //new Child that is non-voided
-		parent.addGroupMember(child);
-		assertTrue("When there is at least 1 non-voided child, should return True", parent.isObsGrouping());
+		assertTrue("When checking for Obs grouping, should include voided Obs", parent.isObsGrouping());
 	}
\ No newline at end of file
