diff --git a/old_product.java b/new_product.java
index 403f12f..4984667 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,25 +1,25 @@
 private T deserializeNextRecord(final T target) throws IOException {
 
-		if (this.uncompressedDataBuffer == null) {
+		if (this.dataBuffer == null) {
 
 			if (this.ioException != null) {
 				throw this.ioException;
 			}
 
-			requestReadBuffersFromBroker();
+			requestReadBufferFromBroker();
 
-			if (this.uncompressedDataBuffer == null) {
+			if (this.dataBuffer == null) {
 				return null;
 			}
 
 			if (this.decompressor != null) {
-				this.decompressor.decompress();
+				this.dataBuffer = this.decompressor.decompress(this.dataBuffer);
 			}
 		}
 
-		final T nextRecord = this.deserializer.readData(target, this.uncompressedDataBuffer);
+		final T nextRecord = this.deserializer.readData(target, this.dataBuffer);
 
-		if (this.uncompressedDataBuffer.remaining() == 0) {
+		if (this.dataBuffer.remaining() == 0) {
 			releasedConsumedReadBuffer();
 		}
 
