diff --git a/old_product.java b/new_product.java
index 86f4db6..43ffc18 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,4 @@
-public static String joinSingleValueMap(Map<String, String> singleValueMap){
-        if (Validator.isNullOrEmpty(singleValueMap)){
-            return StringUtils.EMPTY;
-        }
-        Map<String, String[]> arrayValueMap = toArrayValueMap(singleValueMap);
-        return joinArrayValueMap(arrayValueMap);
+public static String toQueryStringUseSingleValueMap(Map<String, String> singleValueMap){
+        return Validator.isNullOrEmpty(singleValueMap) ? StringUtils.EMPTY
+                        : toQueryStringUseArrayValueMap(toArrayValueMap(singleValueMap));
     }
\ No newline at end of file
