diff --git a/old_test.java b/new_test.java
index bd7cc2f..5066cb0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,5 +7,5 @@
         //        map.put("a", "");
         //        map.put("b", null);
         //        map.put("c", "jim");
-        LOGGER.info(ParamUtil.joinSingleValueMap(map));
+        LOGGER.info(ParamUtil.toQueryStringUseSingleValueMap(map));
     }
\ No newline at end of file
