diff --git a/old_product.java b/new_product.java
index e98a465..e4c931f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,5 +11,5 @@ private Str contentType(final QueryContext qc) throws QueryException {
       final IOFile io = data.meta.binary(path);
       if(io.exists() && !io.isDir()) return Str.get(MimeTypes.get(path));
     }
-    throw WHICHRES.get(info, path);
+    throw WHICHRES_X.get(info, path);
   }
\ No newline at end of file
