diff --git a/old_test.java b/new_test.java
index ff0529f..81878c4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,5 +4,5 @@
     query(_DB_STORE.args(NAME, "raw", "bla"));
     query(_DB_CONTENT_TYPE.args(NAME, "xml"), MimeTypes.APP_XML);
     query(_DB_CONTENT_TYPE.args(NAME, "raw"), MimeTypes.APP_OCTET);
-    error(_DB_CONTENT_TYPE.args(NAME, "test"), Err.WHICHRES);
+    error(_DB_CONTENT_TYPE.args(NAME, "test"), Err.WHICHRES_X);
   }
\ No newline at end of file
