diff --git a/old_product.java b/new_product.java
index 0399965..61fb5c2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @Override
-	public GetUserResponse getUser(Long id) {
-		GetUserResponse response = new GetUserResponse();
+	public GetUserResult getUser(Long id) {
+		GetUserResult result = new GetUserResult();
 		try {
 
 			Validate.notNull(id, "id参数为空");
@@ -10,13 +10,13 @@
 			Validate.notNull(user, "用户不存在(id:" + id + ")");
 
 			UserDTO dto = BeanMapper.map(user, UserDTO.class);
-			response.setUser(dto);
+			result.setUser(dto);
 
-			return response;
+			return result;
 
 		} catch (IllegalArgumentException e) {
-			return handleParameterError(response, e);
+			return handleParameterError(result, e);
 		} catch (RuntimeException e) {
-			return handleGeneralError(response, e);
+			return handleGeneralError(result, e);
 		}
 	}
\ No newline at end of file
