diff --git a/old_test.java b/new_test.java
index f5ec368..adfdda7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,18 @@
 @Test
     public void testGetQueryParam_HttpServletRequest_String() {
-        MockSlingHttpServletRequest request = new MockSlingHttpServletRequest("/apple/macbookair", "show", "html", "simple", "ghz=2.4");
+        // MockSlingHttpServletRequest doesn't natively support getParameter
+        MockSlingHttpServletRequest request = new MockSlingHttpServletRequest("/apple/macbookair", "show", "html", "simple", null) {
+            public String getParameter(String name) {
+                if (name.equals("ghz")) {
+                    return "2.4";
+                } else {
+                    return null;
+                }
+            };
+        };
 
         String key = "ghz";
         String expResult = "2.4";
         String result = PathInfoUtil.getQueryParam(request, key);
-        //assertEquals(expResult, result);
+        assertEquals(expResult, result);
     }
\ No newline at end of file
