diff --git a/old_test.java b/new_test.java
index d0fcb6f..e5d7d84 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,7 +10,7 @@
         Dataframe validationData = data[1];
         
         
-        String dbName = this.getClass().getSimpleName();
+        String storageName = this.getClass().getSimpleName();
 
         Modeler.TrainingParameters trainingParameters = new Modeler.TrainingParameters();
         
@@ -30,11 +30,11 @@
 
         Modeler instance = MLBuilder.create(trainingParameters, conf);
         instance.fit(trainingData);
-        instance.save(dbName);
+        instance.save(storageName);
 
         instance.close();
 
-        instance = MLBuilder.load(Modeler.class, dbName, conf);
+        instance = MLBuilder.load(Modeler.class, storageName, conf);
 
         instance.predict(trainingData);
 
@@ -48,7 +48,7 @@
         //instance = null;
 
 
-        instance = MLBuilder.load(Modeler.class, dbName, conf);
+        instance = MLBuilder.load(Modeler.class, storageName, conf);
         
         instance.predict(validationData);
         
