diff --git a/old_product.java b/new_product.java
index 3c0bf24..48730e6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
         Configuration conf = Configuration.getConfiguration();
         
         
-        String dbName = this.getClass().getSimpleName();
+        String storageName = this.getClass().getSimpleName();
         
         Map<Object, URI> dataset = new HashMap<>();
         try {
@@ -36,7 +36,7 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
 
         TextClassifier instance = MLBuilder.create(trainingParameters, conf);
         instance.fit(dataset);
-        instance.save(dbName);
+        instance.save(storageName);
 
 
         ClassificationMetrics vm = instance.validate(dataset);
@@ -47,7 +47,7 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
         
         
         
-        instance = MLBuilder.load(TextClassifier.class, dbName, conf);
+        instance = MLBuilder.load(TextClassifier.class, storageName, conf);
         Dataframe validationData;
         try {
             validationData = instance.predict(this.getClass().getClassLoader().getResource("datasets/sentimentAnalysis.unlabelled.txt").toURI());
