diff --git a/old_test.java b/new_test.java
index b819eeb..223ada5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -32,9 +32,9 @@
                         .withBody(GsonSingleton.getGson().toJson(response)));
 
 
-        List<Classifier> classifiers = service.getClassifiers();
-        Assert.assertNotNull(classifiers);
-        Assert.assertFalse(classifiers.isEmpty());
-        Assert.assertFalse(classifiers.contains(classifiersResponse));
+        Classifiers classifiers = service.getClassifiers();
+        Assert.assertNotNull(classifiers.getClassifiers());
+        Assert.assertFalse(classifiers.getClassifiers().isEmpty());
+        Assert.assertFalse(classifiers.getClassifiers().contains(classifiersResponse));
 
     }
\ No newline at end of file
