diff --git a/old_test.java b/new_test.java
index becf5a1..324b2bd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-    public void test_toString_formatter() {
+    public void test_format_formatter() {
         DateTimeFormatter f = DateTimeFormatter.ofPattern("H m s");
-        String t = OffsetTime.of(LocalTime.of(11, 30), OFFSET_PONE).toString(f);
+        String t = OffsetTime.of(LocalTime.of(11, 30), OFFSET_PONE).format(f);
         assertEquals(t, "11 30 0");
     }
\ No newline at end of file
