diff --git a/old_product.java b/new_product.java
index fe48def..560217b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 @Override  // override for Javadoc
-    public String toString(DateTimeFormatter formatter) {
-        return super.toString(formatter);
+    public String format(DateTimeFormatter formatter) {
+        return super.format(formatter);
     }
\ No newline at end of file
