diff --git a/old_test.java b/new_test.java
index 88d8fdf..d61f8ed 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,16 +1,16 @@
 @Test
     public void testItemRatingVector() {
-        Collection<Rating> ratings = new ArrayList<Rating>();
+        Collection<Rating> ratings = new ArrayList<>();
         ratings.add(Rating.create(7, 5, 3.5));
         RatingBuilder rb = new RatingBuilder();
         ratings.add(Rating.create(3, 5, 1.5));
         ratings.add(Rating.create(8, 5, 2));
-        SparseVector v = Ratings.itemRatingVector(ratings);
+        Long2DoubleMap v = Ratings.itemRatingVector(ratings);
         assertEquals(3, v.size());
-        assertEquals(7, v.sum(), EPSILON);
+        assertEquals(7, Vectors.sum(v), EPSILON);
 
         long[] keys = {3, 7, 8};
         double[] values = {1.5, 3.5, 2};
-        SparseVector sv = MutableSparseVector.wrap(keys, values);
+        Long2DoubleSortedArrayMap sv = Long2DoubleSortedArrayMap.wrap(SortedKeyIndex.create(keys), values);
         assertEquals(sv, v);
     }
\ No newline at end of file
