diff --git a/old_test.java b/new_test.java
index d32d022..a2320c3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	public void testGenerateQuerySql() {
-		String sql = pagingQuerier.generateQuerySql();
+		String sql = pagingQuerier.generateQuerySql().getStatment();
 		assertEquals("select * from DATA_SOURCES limit 10 offset 0", sql);
 		
 		GeneralQuery generalQuery = new GeneralQuery("person");
