diff --git a/old_product.java b/new_product.java
index f2c1aaf..8a1451b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -31,5 +31,8 @@ public static DependencyVersion parseVersion(String text) {
                 return null;
             }
         }
+        if (version != null && version.endsWith("-py2") && version.length() > 4) {
+            version = version.substring(0, version.length() - 4);
+        }
         return new DependencyVersion(version);
     }
\ No newline at end of file
