diff --git a/old_test.java b/new_test.java
index 5ad21e1..211bb51 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,7 +10,7 @@
 
 			GrowQueue_I8 ecc = new GrowQueue_I8();
 			int nsyn = 10;
-			int syndromes[] = new int[nsyn];
+			GrowQueue_I8 syndromes = GrowQueue_I8.zeros(nsyn);
 
 			ReidSolomonCodes alg = new ReidSolomonCodes(8,primitive8);
 			alg.generator(nsyn);
@@ -23,7 +23,7 @@
 			GrowQueue_I32 whereList = new GrowQueue_I32(1);
 			whereList.add(where);
 
-			alg.findErrorLocatorPolynomialBM(syndromes,nsyn,found);
+			alg.findErrorLocatorPolynomialBM(syndromes,found);
 			alg.findErrorLocatorPolynomial(N+ecc.size,whereList,expected);
 
 			assertEquals(found.size,expected.size);
