diff --git a/old_test.java b/new_test.java
index 1b0eae3..f097707 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test
   public void changeLocalFilePermissionTest() throws IOException {
     File tempFile = mTestFolder.newFile("perm.txt");
-    FileUtils.changeLocalFilePermission(tempFile.getAbsolutePath(), "000");
+    FileUtils.changeLocalFilePermission(tempFile.getAbsolutePath(), "---------");
     Assert.assertFalse(tempFile.canRead() || tempFile.canWrite() || tempFile.canExecute());
-    FileUtils.changeLocalFilePermission(tempFile.getAbsolutePath(), "777");
+    FileUtils.changeLocalFilePermission(tempFile.getAbsolutePath(), "rwxrwxrwx");
     Assert.assertTrue(tempFile.canRead() && tempFile.canWrite() && tempFile.canExecute());
     // File deletion should fail, because we don't have write permissions
-    FileUtils.changeLocalFilePermission(tempFile.getAbsolutePath(), "444");
+    FileUtils.changeLocalFilePermission(tempFile.getAbsolutePath(), "r--r--r--");
     Assert.assertTrue(tempFile.canRead());
     Assert.assertFalse(tempFile.canWrite());
     Assert.assertFalse(tempFile.canExecute());
