diff --git a/old_product.java b/new_product.java
index 639757c..6f8cc6e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,3 @@
 public static void changeLocalFilePermission(String filePath, String perms) throws IOException {
-    // TODO(cc): Switch to java's Files.setPosixFilePermissions() when Java 6 support is dropped.
-    ShellUtils.execCommand(ShellUtils.getSetPermissionCommand(perms, filePath));
+    Files.setPosixFilePermissions(Paths.get(filePath), PosixFilePermissions.fromString(perms));
   }
\ No newline at end of file
