diff --git a/old_product.java b/new_product.java
index 1d25031..9238584 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,10 @@
 @Nonnull
-    protected String toScript(@Nonnull final TransformRequest request) {
-        StringBuilder script = new StringBuilder();
-        script.append("class Transform (destination: String, sendResults: Boolean, sqlContext: org.apache.spark.sql.SQLContext)");
-        script.append(" extends com.thinkbiganalytics.spark.metadata.TransformScript(destination, sendResults, sqlContext) {\n");
+    String toScript(@Nonnull final TransformRequest request) {
+        final StringBuilder script = new StringBuilder();
+        script.append(
+            "class Transform (destination: String, profiler: com.thinkbiganalytics.spark.dataprofiler.Profiler, sqlContext: org.apache.spark.sql.SQLContext, sparkContextService: com.thinkbiganalytics.spark.SparkContextService) extends ");
+        script.append(transformScriptClass.getName());
+        script.append("(destination, profiler, sqlContext, sparkContextService) {\n");
 
         script.append("override def dataFrame: org.apache.spark.sql.DataFrame = {");
         script.append(request.getScript());
@@ -15,12 +17,10 @@
             script.append("override def parentTable: String = {\"");
             script.append(StringEscapeUtils.escapeJava(request.getParent().getTable()));
             script.append("\"}\n");
-
-            this.cache.touch(request.getParent().getTable());
         }
 
         script.append("}\n");
-        script.append("new Transform(tableName, true, sqlContext).run()\n");
+        script.append("new Transform(tableName, profiler, sqlContext, sparkContextService).run()\n");
 
         return script.toString();
     }
\ No newline at end of file
