diff --git a/old_test.java b/new_test.java
index a049603..5775a20 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,7 @@
 @Test
 	@Verifies(value = "return distinct drugs", method = "getDrugs(String,Concept,boolean,boolean,boolean,Integer,Integer)")
 	public void getDrugs_shouldReturnDistinctDrugs() throws Exception {
-		Session session2 = sessionFactory.getCurrentSession();
-		session2.beginTransaction();
-		Concept concept1 = (Concept) session2.get(Concept.class, 14);
+		Concept concept1 = dao.getConcept(14);
 		
 		List<Drug> drugList = dao.getDrugs("TEST_DRUG", concept1, true, true, false, 0, 10);
 		Assert.assertEquals(1, drugList.size());
