diff --git a/old_product.java b/new_product.java
index 408f4c1..f862252 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
 @Authorized(PrivilegeConstants.GET_CONCEPTS)
-	public List<Drug> getDrugs(String drugName, Concept concept, boolean searchOnPhrase, boolean searchDrugConceptNames,
+	public List<Drug> getDrugs(String drugName, Concept concept, boolean searchKeywords, boolean searchDrugConceptNames,
 	        boolean includeRetired, Integer start, Integer length) throws APIException;
\ No newline at end of file
