diff --git a/old_test.java b/new_test.java
index b9b36d5..2ea15cb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -19,11 +19,7 @@
 		assertEquals(3, ZOOKEEPER.getClient().getChildren().forPath(CHECKPOINT_PATH).size());
 		assertEquals(3, checkpoints.getNumberOfRetainedCheckpoints());
 
-		resetCheckpoint(expected[0].getOperatorStates().values());
-		resetCheckpoint(expected[1].getOperatorStates().values());
-		resetCheckpoint(expected[2].getOperatorStates().values());
-
-		// Recover TODO!!! clear registry!
+		// Recover
 		checkpoints.recover();
 
 		assertEquals(3, ZOOKEEPER.getClient().getChildren().forPath(CHECKPOINT_PATH).size());
