diff --git a/old_product.java b/new_product.java
index cfe8bdd..804f07a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@
 		// Clear local handles in order to prevent duplicates on
 		// recovery. The local handles should reflect the state
 		// of ZooKeeper.
-		checkpointStateHandles.clear();
+		completedCheckpoints.clear();
 
 		// Get all there is first
 		List<Tuple2<RetrievableStateHandle<CompletedCheckpoint>, String>> initialCheckpoints;
@@ -29,6 +29,11 @@
 
 			try {
 				completedCheckpoint = retrieveCompletedCheckpoint(checkpointStateHandle);
+				if (completedCheckpoint != null) {
+					// Re-register all shared states in the checkpoint.
+					completedCheckpoint.registerSharedStates(sharedStateRegistry);
+					completedCheckpoints.add(completedCheckpoint);
+				}
 			} catch (Exception e) {
 				LOG.warn("Could not retrieve checkpoint. Removing it from the completed " +
 					"checkpoint store.", e);
@@ -36,10 +41,5 @@
 				// remove the checkpoint with broken state handle
 				removeBrokenStateHandle(checkpointStateHandle.f1, checkpointStateHandle.f0);
 			}
-
-			if (completedCheckpoint != null) {
-				completedCheckpoint.registerSharedStates(sharedStateRegistry);
-				checkpointStateHandles.add(checkpointStateHandle);
-			}
 		}
 	}
\ No newline at end of file
