diff --git a/old_test.java b/new_test.java
index 6157545..59dcc7a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,8 @@
 @Test
 	public void testQueryHistoryLog() {
-		List<HistoryLogVo> vos = getJBPMApplication().queryHistoryLog(1l);
+		long i = getJBPMApplication().startProcess("defaultPackage.Trade",
+				"aaa", null);
+		List<HistoryLogVo> vos = getJBPMApplication().queryHistoryLog(i);
 		Assert.assertTrue(vos.size() > 0);
+		getJBPMApplication().removeProcessInstance(i);
 	}
\ No newline at end of file
