diff --git a/old_test.java b/new_test.java
index ed0f7a5..a4f0f7d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
 
         // Test to get an existing stream
         when(mockControllerService.getStream(scope1, stream1)).thenReturn(streamConfigFuture);
-        Response response = client.target(resourceURI).request().buildGet().invoke();
+        Response response = addAuthHeaders(client.target(resourceURI).request()).buildGet().invoke();
         assertEquals("Get Stream Config Status", 200, response.getStatus());
         StreamProperty streamResponseActual = response.readEntity(StreamProperty.class);
         testExpectedVsActualObject(streamResponseExpected, streamResponseActual);
@@ -15,7 +15,7 @@
         when(mockControllerService.getStream(scope1, stream2)).thenReturn(CompletableFuture.supplyAsync(() -> {
             throw StoreException.create(StoreException.Type.DATA_NOT_FOUND, stream2);
         }));
-        response = client.target(resourceURI2).request().buildGet().invoke();
+        response = addAuthHeaders(client.target(resourceURI2).request()).buildGet().invoke();
         assertEquals("Get Stream Config Status", 404, response.getStatus());
         response.close();
     }
\ No newline at end of file
