diff --git a/old_product.java b/new_product.java
index 7f15cc5..987ddb5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,15 @@
             final AsyncResponse asyncResponse) {
         long traceId = LoggerHelpers.traceEnter(log, "getStream");
 
+        try {
+            authenticate(scopeName + "/" + streamName, READ);
+        } catch (AuthenticationException e) {
+            log.warn("Get stream for {} failed due to authentication failure.", scopeName + "/" + streamName);
+            asyncResponse.resume(Response.status(Status.UNAUTHORIZED).build());
+            LoggerHelpers.traceLeave(log, "getStream", traceId);
+            return;
+        }
+
         controllerService.getStream(scopeName, streamName)
                 .thenApply(streamConfig -> Response.status(Status.OK)
                         .entity(ModelHelper.encodeStreamResponse(streamConfig))
