diff --git a/old_test.java b/new_test.java
index fc6392d..38fad90 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,5 @@
         QueryCriterion criterion2 = new NotEqCriterion("name", "");
         QueryCriterion criterion3 = new AndCriterion(criterion1, criterion2);
         
-        assertTrue(instance.notBlank("name").getQueryCriterions()
-                .contains(criterion3));
+        assertEquals(criterion3, instance.notBlank("name").getQueryCriterion());
     }
\ No newline at end of file
