diff --git a/old_test.java b/new_test.java
index 96f34e5..97c0510 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
-  public void appendTest() throws IOException {
+  public void appendTest() throws IOException, StoreException {
     // buffer append
     doAppendTest(new Appender() {
       @Override
-      public void append(LogSegment segment, ByteBuffer buffer) throws IOException {
+      public void append(LogSegment segment, ByteBuffer buffer) throws StoreException {
         int writeSize = buffer.remaining();
         int written = segment.appendFrom(buffer);
         assertEquals("Size written did not match size of buffer provided", writeSize, written);
@@ -13,7 +13,7 @@
     // channel append
     doAppendTest(new Appender() {
       @Override
-      public void append(LogSegment segment, ByteBuffer buffer) throws IOException {
+      public void append(LogSegment segment, ByteBuffer buffer) throws StoreException {
         int writeSize = buffer.remaining();
         segment.appendFrom(Channels.newChannel(new ByteBufferInputStream(buffer)), writeSize);
         assertFalse("The buffer was not completely written", buffer.hasRemaining());
@@ -24,7 +24,7 @@
     if (Utils.isLinux()) {
       doAppendTest(new Appender() {
         @Override
-        public void append(LogSegment segment, ByteBuffer buffer) throws IOException {
+        public void append(LogSegment segment, ByteBuffer buffer) throws StoreException {
           int writeSize = buffer.remaining();
           segment.appendFromDirectly(buffer.array(), 0, writeSize);
         }
