diff --git a/old_test.java b/new_test.java
index dd61265..24464a4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,22 @@
 @Test
   public void testAddWord() {
     model = createVoiceModel();
-    final CustomTranslation expected = instantiateCustomTranslations().get(0);
+    final Word expected = instantiateWords().get(0);
 
-    service.addWord(model, expected).execute();
+    AddWordOptions addOptions = new AddWordOptions.Builder()
+        .word(expected.getWord())
+        .translation(expected.getTranslation())
+        .customizationId(model.getCustomizationId())
+        .build();
+    service.addWord(addOptions).execute();
 
-    final List<CustomTranslation> results = service.getWords(model).execute();
-    assertEquals(1, results.size());
+    ListWordsOptions listOptions = new ListWordsOptions.Builder()
+        .customizationId(model.getCustomizationId())
+        .build();
+    final Words results = service.listWords(listOptions).execute();
+    assertEquals(1, results.getWords().size());
 
-    final CustomTranslation result = results.get(0);
+    final Word result = results.getWords().get(0);
     assertEquals(expected, result);
     assertEquals(expected.getWord(), result.getWord());
     assertEquals(expected.getTranslation(), result.getTranslation());
