diff --git a/old_product.java b/new_product.java
index 73f1554..dcce5a8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,13 +5,13 @@ public static LPResult solve(double[] linearObjectiveFunction, List<LPSolver.LPC
         for(LPSolver.LPConstraint constraint : linearConstraintsList) {
             String sign = constraint.getSign();
             Relationship relationship = null;
-            if(">=".equals(sign)) {
+            if(LPSolver.GEQ.equals(sign)) {
                 relationship = Relationship.GEQ;
             }
-            else if("<=".equals(sign)) {
+            else if(LPSolver.LEQ.equals(sign)) {
                 relationship = Relationship.LEQ;
             }
-            else if("=".equals(sign)) {
+            else if(LPSolver.EQ.equals(sign)) {
                 relationship = Relationship.EQ;
             }
             constraints.add(
