diff --git a/old_test.java b/new_test.java
index 895a422..46e9ccd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,9 +5,9 @@
         //Example from http://lpsolve.sourceforge.net/5.5/PHP.htm
         double[] linearObjectiveFunction = {143.0, 60.0, 195.0};
         List<LPSolver.LPConstraint> linearConstraintsList = new ArrayList<>();
-        linearConstraintsList.add(new LPSolver.LPConstraint(new double[]{120.0, 210.0, 150.75}, "<=", 15000.0));
-        linearConstraintsList.add(new LPSolver.LPConstraint(new double[]{110.0, 30.0, 125.0}, "<=", 4000.0));
-        linearConstraintsList.add(new LPSolver.LPConstraint(new double[]{1.0, 1.0, 1.0}, "<=", 75.0));
+        linearConstraintsList.add(new LPSolver.LPConstraint(new double[]{120.0, 210.0, 150.75}, LPSolver.LEQ, 15000.0));
+        linearConstraintsList.add(new LPSolver.LPConstraint(new double[]{110.0, 30.0, 125.0}, LPSolver.LEQ, 4000.0));
+        linearConstraintsList.add(new LPSolver.LPConstraint(new double[]{1.0, 1.0, 1.0}, LPSolver.LEQ, 75.0));
 
         LPSolver.LPResult expResult = new LPSolver.LPResult();
         expResult.setObjectiveValue(6986.8421052632);
