diff --git a/old_test.java b/new_test.java
index 960b800..9ae4dd4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,13 +8,13 @@
     query(_XQUERY_EVAL.args("\"declare variable $local:a external; $local:a\"",
         " map { xs:QName('local:a'): 1 }"), "1");
     query(_XQUERY_EVAL.args(".", " map { '': 1 }"), "1");
-    error(_XQUERY_EVAL.args("1+"), Err.CALCEXPR);
-    error("declare variable $a:=1;" + _XQUERY_EVAL.args("\"$a\""), Err.VARUNDEF_X);
-    error("for $a in (1,2) return " + _XQUERY_EVAL.args("\"$a\""), Err.VARUNDEF_X);
+    error(_XQUERY_EVAL.args("1+"), CALCEXPR);
+    error("declare variable $a:=1;" + _XQUERY_EVAL.args("\"$a\""), VARUNDEF_X);
+    error("for $a in (1,2) return " + _XQUERY_EVAL.args("\"$a\""), VARUNDEF_X);
     // check updating expressions
-    error(_XQUERY_EVAL.args("delete node ()"), Err.BXXQ_UPDATING);
+    error(_XQUERY_EVAL.args("delete node ()"), BXXQ_UPDATING);
     error(_XQUERY_EVAL.args("declare %updating function local:x() {()}; local:x()"),
-        Err.BXXQ_UPDATING);
+        BXXQ_UPDATING);
     query(_XQUERY_EVAL.args("declare %updating function local:x() {()}; 1"));
     query(_XQUERY_EVAL.args('"' + DOC.args(PATH).replace('"', '\'') + '"'));
 
@@ -23,13 +23,13 @@
     query("try{ " + _XQUERY_EVAL.args("\"(1 to 10000000000000)[.=0]\"", " map{}",
         " map{ 'timeout':'1'}") + " } catch * { () }", "");
     error(_XQUERY_EVAL.args("\"doc('" + NAME + "')\"", " map{}", " map{ 'permission':'none'}"),
-        Err.BXXQ_PERM_X);
+        BXXQ_PERM_X);
     error(_XQUERY_EVAL.args("\"db:open('" + NAME + "')\"", " map{}", " map{ 'permission':'none'}"),
-        Err.BXDB_OPEN_X);
+        BXDB_OPEN_X);
     error(_XQUERY_EVAL.args("\"file:exists('x')\"", " map{}", " map{ 'permission':'none'}"),
-        Err.BXXQ_PERM_X);
+        BXXQ_PERM_X);
     error(_XQUERY_EVAL.args("\"(1 to 10000000000000)[.=0]\"", " map{}", " map{ 'timeout':'1'}"),
-        Err.BXXQ_STOPPED);
+        BXXQ_STOPPED);
     error(_XQUERY_EVAL.args("\"(1 to 10000000000000) ! <a/>\"", " map{}", " map{ 'memory':'10'}"),
-        Err.BXXQ_STOPPED);
+        BXXQ_STOPPED);
   }
\ No newline at end of file
