diff --git a/old_test.java b/new_test.java
index a013ed6..4c89b36 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test(groups = { "direct" }, timeOut = TIMEOUT)
     public void read() throws Exception {
         Document docDefinition = this.getDocumentDefinition();
-        Document document = client.createDocument(getCollectionLink(), docDefinition, null, false).toBlocking().single().getResource();
+        Document document = client.createDocument(getCollectionLink(), docDefinition, null, false).single().block().getResource();
 
         // give times to replicas to catch up after a write
         waitIfNeededForReplicasToCatchUp(clientBuilder());
