diff --git a/old_test.java b/new_test.java
index 8afaaac..274e771 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
     public void testBetween() {
-    	assertEquals(new BetweenCriterion("name", "a", "b"), instance.between("name", "a", "b").getQueryCriterion());
+    	assertEquals(Criteria.between("name", "a", "b"), instance.between("name", "a", "b").getQueryCriterion());
     }
\ No newline at end of file
