diff --git a/old_test.java b/new_test.java
index c2d1b39..c6b16e2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,5 +5,5 @@
         executor.sendMessages(protocolMessages);
         byte[] sendByte = ((FakeTransportHandler) context.getTransportHandler()).getSendByte();
         LOGGER.info(ArrayConverter.bytesToHexString(sendByte));
-        assertArrayEquals(sendByte, new byte[] { 21, 03, 03, 00, 02, 02, 51 });
+        assertArrayEquals(new byte[] { 21, 03, 03, 00, 02, 02, 51 }, sendByte);
     }
\ No newline at end of file
