diff --git a/old_product.java b/new_product.java
index 9b80857..019c1e0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,8 @@
 @Override
     public List<ProtocolMessage> sendMessages(List<ProtocolMessage> messages) {
+        if (!proceed) {
+            return null;
+        }
         MessageBytesCollector messageBytesCollector = new MessageBytesCollector();
         for (ProtocolMessage message : messages) {
             LOGGER.debug("Preparing " + message.toCompactString());
