diff --git a/old_test.java b/new_test.java
index e8ab083..214c5d9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,12 +2,12 @@
 	public void fromJson() throws Exception {
 		// Bean
 		String beanString = "{\"name\":\"A\"}";
-		TestBean bean = mapper.fromJson(beanString, TestBean.class);
+		TestBean bean = JsonMapper.INSTANCE.fromJson(beanString, TestBean.class);
 		System.out.println("Bean:" + bean);
 
 		// Map
 		String mapString = "{\"name\":\"A\",\"age\":2}";
-		Map<String, Object> map = mapper.fromJson(mapString, HashMap.class);
+		Map<String, Object> map = JsonMapper.INSTANCE.fromJson(mapString, HashMap.class);
 		System.out.println("Map:");
 		for (Entry<String, Object> entry : map.entrySet()) {
 			System.out.println(entry.getKey() + " " + entry.getValue());
@@ -15,7 +15,7 @@
 
 		// List<String>
 		String listString = "[\"A\",\"B\",\"C\"]";
-		List<String> stringList = mapper.getMapper().readValue(listString, List.class);
+		List<String> stringList = JsonMapper.INSTANCE.fromJson(listString, List.class);
 		System.out.println("String List:");
 		for (String element : stringList) {
 			System.out.println(element);
@@ -23,8 +23,8 @@
 
 		// List<Bean>
 		String beanListString = "[{\"name\":\"A\"},{\"name\":\"B\"}]";
-		List<TestBean> beanList = mapper.getMapper().readValue(beanListString, new TypeReference<List<TestBean>>() {
-		});
+		List<TestBean> beanList = JsonMapper.INSTANCE.fromJson(beanListString,
+				JsonMapper.INSTANCE.buildCollectionType(List.class, TestBean.class));
 		System.out.println("Bean List:");
 		for (TestBean element : beanList) {
 			System.out.println(element);
