diff --git a/old_product.java b/new_product.java
index 61bd839..330c6b9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,9 +2,6 @@
 	public List<Order> getActiveOrders(Patient patient, OrderType orderType, CareSetting careSetting, Date asOfDate) {
 		Criteria crit = createOrderCriteria(patient, careSetting, orderType, false, false);
 		
-		Criterion startDateEqualToOrBeforeAsOfDate = Restrictions.le("startDate", asOfDate);
-		crit.add(startDateEqualToOrBeforeAsOfDate);
-		
 		Disjunction dateStoppedAndAutoExpDateDisjunction = Restrictions.disjunction();
 		Criterion stopAndAutoExpDateAreBothNull = Restrictions.and(Restrictions.isNull("dateStopped"), Restrictions
 		        .isNull("autoExpireDate"));
