diff --git a/old_product.java b/new_product.java
index 186496b..0767eec 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,19 @@
-public static String getConstructor(String yangName, JavaAttributeInfo attr) {
+public static String getConstructor(String yangName, JavaAttributeInfo attr, int generatedJavaFiles) {
 
         String attributeName = getSmallCase(attr.getAttributeName());
+        String constructor;
 
-        String constructor = EIGHT_SPACE_INDENTATION + THIS + PERIOD + getCamelCase(attributeName, null) + SPACE + EQUAL
-                + SPACE + BUILDER.toLowerCase() + OBJECT + PERIOD + GET_METHOD_PREFIX
-                + getCaptialCase(getCamelCase(attributeName, null)) + OPEN_PARENTHESIS + CLOSE_PARENTHESIS + SEMI_COLAN
-                + NEW_LINE;
-
+        if ((generatedJavaFiles & GENERATE_SERVICE_AND_MANAGER) != 0) {
+            constructor = EIGHT_SPACE_INDENTATION + THIS + PERIOD + getCamelCase(attributeName, null) + SPACE + EQUAL
+                    + SPACE + BUILDER.toLowerCase() + OBJECT + PERIOD + GET_METHOD_PREFIX
+                    + getCaptialCase(getCamelCase(attributeName, null)) + OPEN_PARENTHESIS + CLOSE_PARENTHESIS +
+                    SEMI_COLAN
+                    + NEW_LINE;
+        } else {
+            constructor = EIGHT_SPACE_INDENTATION + THIS + PERIOD + getCamelCase(attributeName, null) + SPACE + EQUAL
+                    + SPACE + BUILDER.toLowerCase() + OBJECT + PERIOD + getCamelCase(attributeName, null) +
+                    OPEN_PARENTHESIS + CLOSE_PARENTHESIS + SEMI_COLAN
+                    + NEW_LINE;
+        }
         return constructor;
     }
\ No newline at end of file
