diff --git a/old_product.java b/new_product.java
index 35fab16..ffd60ea 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 protected void doRemove(final RuleRemovalContext context,
                             final ReteooBuilder builder,
                             final InternalWorkingMemory[] workingMemories) {
-        if ( !context.getRuleBase().getConfiguration().isPhreakEnabled() && context.getCleanupAdapter() != null ) {
+        if ( !context.getKnowledgeBase().getConfiguration().isPhreakEnabled() && context.getCleanupAdapter() != null ) {
             for ( InternalWorkingMemory workingMemory : workingMemories ) {
                 CleanupAdapter adapter = context.getCleanupAdapter();
                 final ObjectTypeNodeMemory memory = (ObjectTypeNodeMemory) workingMemory.getNodeMemory( this );
