diff --git a/old_test.java b/new_test.java
index 680de50..b3f581d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
 		GrowQueue_I32 corners = createSquareCorners(10, 12, 20, 30);
 		corners.add(corners.get(3)+4);
 
-		MinimizeEnergyPrune alg = new MinimizeEnergyPrune();
+		MinimizeEnergyPrune alg = new MinimizeEnergyPrune(1);
 
 		GrowQueue_I32 output = new GrowQueue_I32();
 		alg.fit(contours,corners,output);
