diff --git a/old_test.java b/new_test.java
index 70dba60..7eb9e10 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,33 +1,36 @@
 @Test
-	public void testCreateCoord() {
+	public void testCreateCoord() throws EntityAlreadyExistsException {
 		// SUCCESS
 		CoordData c = new CoordData();
-		c.id = "herp.derp";
-		c.name = "Herp McDerpson";
-		c.email = "ching@chang.com";
-		
-		try {
-			accountsDb.createCoord(c);
-		} catch (EntityAlreadyExistsException e) {
-			fail();
-		}
+		c.id = "valid.id";
+		c.name = "John Doe";
+		c.email = "john.doe@coordinator.com";
+		accountsDb.createCoord(c);
 		
 		// FAIL : duplicate
 		try {
 			accountsDb.createCoord(c);
 			fail();
 		} catch (EntityAlreadyExistsException e) {
-			
+			assertContains(AccountsDb.ERROR_CREATE_COORD_ALREADY_EXISTS, e.getMessage());
 		}
 		
 		// FAIL : invalid params
-		c.id = "herp mc derp";
+		c.id = "invalid id with spaces";
 		try {
 			accountsDb.createCoord(c);
 			fail();
 		} catch (AssertionError a) {
-			
+			assertEquals(a.getMessage(), CoordData.ERROR_FIELD_ID);
 		} catch (EntityAlreadyExistsException e) {
 			fail();
 		}
+		
+		// Null params check:
+		try {
+			accountsDb.createCoord(null);
+			fail();
+		} catch (AssertionError a) {
+			assertEquals(Common.ERROR_DBLEVEL_NULL_INPUT, a.getMessage());
+		}
 	}
\ No newline at end of file
