diff --git a/old_product.java b/new_product.java
index 1407c5d..ee99ec5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,10 @@
 public void createCoord(String coordId, String coordName, String coordEmail)
 			throws EntityAlreadyExistsException, InvalidParametersException {
 
+		Assumption.assertNotNull(ERROR_NULL_PARAMETER, coordId);
+		Assumption.assertNotNull(ERROR_NULL_PARAMETER, coordName);
+		Assumption.assertNotNull(ERROR_NULL_PARAMETER, coordEmail);
+
 		verifyAdminLoggedIn();
 
 		CoordData coordToAdd = new CoordData(coordId, coordName, coordEmail);
