diff --git a/old_product.java b/new_product.java
index 771bad5..79c599c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public CreateFileTOptions toThrift() {
     CreateFileTOptions options = new CreateFileTOptions();
     options.setBlockSizeBytes(mBlockSizeBytes);
-    options.setPersisted(mWriteType.getUnderStorageType().isSyncPersist());
+    options.setPersisted(mWriteType.isThrough());
     options.setRecursive(mRecursive);
     options.setTtl(mTtl);
     options.setTtlAction(ThriftUtils.toThrift(mTtlAction));
