diff --git a/old_test.java b/new_test.java
index 436ceef..3dd8f08 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,6 +14,7 @@
         assertEquals(10., binarizationTrainer.getThreshold(), 0);
 
         BinarizationPreprocessor<Integer, double[]> preprocessor = binarizationTrainer.fit(
+            TestUtils.testEnvBuilder(),
             datasetBuilder,
             (k, v) -> VectorUtils.of(v)
         );
