diff --git a/old_test.java b/new_test.java
index b28b0c6..5606a56 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
-	public void testCheckUserId() {
-		NGrinderBaseController ngridnerBaseController = new NGrinderBaseController();
+	public void testCheckDuplicatedID() {
+		NGrinderBaseController ngrinderBaseController = new NGrinderBaseController();
 		ModelMap model = new ModelMap();
-		String rtnStr = userController.checkUserId(model, "not-exist");
-		assertThat(rtnStr, is(ngridnerBaseController.returnSuccess()));
+		HttpEntity<String> rtnStr = userController.checkDuplication(model, "not-exist");
+		assertThat(rtnStr.getBody(), is(ngrinderBaseController.returnSuccess()));
 
-		rtnStr = userController.checkUserId(model, getTestUser().getUserId());
-		assertThat(rtnStr, is(ngridnerBaseController.returnError()));
+		rtnStr = userController.checkDuplication(model, getTestUser().getUserId());
+		assertThat(rtnStr.getBody(), is(ngrinderBaseController.returnError()));
 	}
\ No newline at end of file
