diff --git a/old_product.java b/new_product.java
index 866ea5b..d30f1b6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,11 +15,11 @@
         controllerService.listStreamsInScope(scopeName)
                 .thenApply(streamsList -> {
                     StreamsList streams = new StreamsList();
-                    streamsList.forEach(stream -> {
+                    streamsList.forEach((stream, config) -> {
                         // If internal streams are requested select only the ones that have the special stream names
                         // otherwise display the regular user created streams.
-                        if (!showOnlyInternalStreams ^ stream.getStreamName().startsWith(INTERNAL_NAME_PREFIX)) {
-                            streams.addStreamsItem(ModelHelper.encodeStreamResponse(stream));
+                        if (!showOnlyInternalStreams ^ stream.startsWith(INTERNAL_NAME_PREFIX)) {
+                            streams.addStreamsItem(ModelHelper.encodeStreamResponse(scopeName, stream, config));
                         }
                     });
                     log.info("Successfully fetched streams for scope: {}", scopeName);
