diff --git a/old_product.java b/new_product.java
index 86efa88..bf560dd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -19,7 +19,7 @@
 		OrderService os = Context.getOrderService();
 		List<Patient> patients = new ArrayList<Patient>();
 		patients.add(patient);
-		List<Order> orders = os.getOrders(Order.class, patients, null, null, null);
+		List<Order> orders = os.getOrders(null, patients, null, null, null);
 		if (CollectionUtils.isNotEmpty(orders)) {
 			for (Order order : orders) {
 				if (!order.isVoided()) {
