diff --git a/old_test.java b/new_test.java
index 1b21872..4222445 100644
--- a/old_test.java
+++ b/new_test.java
@@ -19,7 +19,7 @@
 		OrderService os = Context.getOrderService();
 		List<Patient> patients = new ArrayList<Patient>();
 		patients.add(patient);
-		List<Order> orders = os.getOrders(Order.class, patients, null, null, null);
+		List<Order> orders = os.getOrders(null, patients, null, null, null);
 		Assert.assertTrue(CollectionUtils.isNotEmpty(orders));
 		//all order void related fields should be null
 		for (Order order : orders) {
