diff --git a/old_product.java b/new_product.java
index b0e2ef4..44e7be4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,11 @@
 void flush() {
-        this.cacheUpdater.flush();
+        HashSet<Long> elements;
+        synchronized (this.readIndex) {
+            elements = this.recentStreamSegmentIds;
+            this.recentStreamSegmentIds = new HashSet<>();
+        }
+
+        this.readIndex.triggerFutureReads(elements);
         if (this.flushCallback != null) {
             this.flushCallback.run();
         }
