diff --git a/old_product.java b/new_product.java
index 816a084..86868f6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public LocalDateTime withSecondOfMinute(int secondOfMinute) {
-        LocalTime newTime = time.withSecondOfMinute(secondOfMinute);
+public LocalDateTime withSecond(int second) {
+        LocalTime newTime = time.withSecond(second);
         return with(date, newTime);
     }
\ No newline at end of file
