diff --git a/old_test.java b/new_test.java
index 6038e1c..3ed3719 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
-    public void test_withSecondOfMinute_normal() {
+    public void test_withSecond_normal() {
         OffsetDateTime base = OffsetDateTime.of(2008, 6, 30, 11, 30, 59, OFFSET_PONE);
-        OffsetDateTime test = base.withSecondOfMinute(15);
+        OffsetDateTime test = base.withSecond(15);
         assertEquals(test, OffsetDateTime.of(2008, 6, 30, 11, 30, 15, OFFSET_PONE));
     }
\ No newline at end of file
