diff --git a/old_product.java b/new_product.java
index 72b6099..f311be6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -33,7 +33,7 @@ private void run() throws Exception {
 
     // save log data
     Util.outln(NL + "Writing log file '" + TESTID + ".log'...");
-    try(PrintOutput po = new PrintOutput(TESTID + ".log")) {
+    try(PrintOutput po = new PrintOutput(new IOFile(TESTID + ".log"))) {
       po.println("QT3TS RESULTS __________________________" + NL);
       po.println(result.toString());
       po.println("WRONG __________________________________" + NL);
