diff --git a/old_product.java b/new_product.java
index fa79007..b466a3b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-public static Priority[] getAllPossiblePriorities() {
+@Deprecated
+    public static Priority[] getAllPossiblePriorities() {
         return new Priority[]{Priority.FATAL, Priority.ERROR, Level.WARN,
             Priority.INFO, Priority.DEBUG};
     }
\ No newline at end of file
