diff --git a/old_test.java b/new_test.java
index 64f308a..c71647e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,11 +3,14 @@
         final Context context = RuntimeEnvironment.application;
         final String versionName = context.getPackageManager().getPackageInfo(context.getPackageName(), 0).versionName;
 
+        final SupportUtils.SumoTopic testTopic = SupportUtils.SumoTopic.TRACKERS;
+        final String testTopicStr = testTopic.topicStr;
+
         Locale.setDefault(Locale.GERMANY);
-        assertEquals("https://support.mozilla.org/1/mobile/" + versionName + "/Android/de-DE/foobar",
-                SupportUtils.getSumoURLForTopic(RuntimeEnvironment.application, "foobar"));
+        assertEquals("https://support.mozilla.org/1/mobile/" + versionName +"/Android/de-DE/" + testTopicStr,
+                SupportUtils.getSumoURLForTopic(RuntimeEnvironment.application, testTopic));
 
         Locale.setDefault(Locale.CANADA_FRENCH);
-        assertEquals("https://support.mozilla.org/1/mobile/" + versionName + "/Android/fr-CA/foobar",
-                SupportUtils.getSumoURLForTopic(RuntimeEnvironment.application, "foobar"));
+        assertEquals("https://support.mozilla.org/1/mobile/" + versionName + "/Android/fr-CA/" + testTopicStr,
+                SupportUtils.getSumoURLForTopic(RuntimeEnvironment.application, testTopic));
     }
\ No newline at end of file
