diff --git a/old_test.java b/new_test.java
index b874748..d93d356 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,13 @@
 @Test
-  public void baseName() {
+  public void name() {
     // check with a simple path
-    query(_FILE_BASE_NAME.args(PATH1), NAME);
+    query(_FILE_NAME.args(PATH1), NAME);
     // check with a path ending with a directory separator
-    query(_FILE_BASE_NAME.args(PATH1 + File.separator), NAME);
+    query(_FILE_NAME.args(PATH1 + File.separator), NAME);
     // check with a path consisting only of directory separators
-    query(_FILE_BASE_NAME.args("//"), "");
+    query(_FILE_NAME.args("//"), "");
     // check with empty string path
-    query(_FILE_BASE_NAME.args(""), ".");
+    query(_FILE_NAME.args(""), "");
     // check using a suffix
-    query(_FILE_BASE_NAME.args(PATH1 + File.separator + "test.xml", ".xml"), "test");
+    query(_FILE_NAME.args(PATH1 + File.separator + "test.xml"), "test.xml");
   }
\ No newline at end of file
