diff --git a/old_product.java b/new_product.java
index bf23e6d..3c26432 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,3 @@
-private Str baseName(final QueryContext ctx) throws QueryException {
-    final File path = checkFile(0, ctx);
-    if(path.getPath().isEmpty()) return Str.get(".");
-    final String suf = expr.length < 2 ? null : string(checkStr(expr[1], ctx));
-    String pth = path.getName();
-    if(suf != null && pth.endsWith(suf))
-      pth = pth.substring(0, pth.length() - suf.length());
-    return Str.get(pth);
+private Str name(final QueryContext ctx) throws QueryException {
+    return Str.get(checkFile(0, ctx).getName());
   }
\ No newline at end of file
