diff --git a/old_test.java b/new_test.java
index 9c96ad5..0d3f871 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,8 @@
-@Test
+@Test(expected=UnsupportedOperationException.class)
     public void testClose() throws SQLException {
-        sqlSessionTemplate.close();
-        assertFalse(connection.isClosed());
-
-        connection.close();
+        try {
+            sqlSessionTemplate.close();
+        } finally {
+            connection.close();    
+        }
     }
\ No newline at end of file
