diff --git a/old_test.java b/new_test.java
index 2885b58..8f8c9c9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
 	public void testChangeProtectionKey() throws NoPeerConnectionException, IOException {
-		KeyPair keypair1 = H2HEncryptionUtil.generateProtectionKeys();
-		KeyPair keypair2 = H2HEncryptionUtil.generateProtectionKeys();
+		KeyPair keypair1 = EncryptionUtil.generateRSAKeyPair();
+		KeyPair keypair2 = EncryptionUtil.generateRSAKeyPair();
 
 		Number160 locationKey = Number160.createHash(NetworkTestUtil.randomString());
 		Number160 domainKey = H2HConstants.TOMP2P_DEFAULT_KEY;
