diff --git a/old_product.java b/new_product.java
index 5f9057e..63cf3b5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,11 +10,11 @@ public FuturePut changeProtectionKey(Number160 locationKey, Number160 domainKey,
 				return null;
 			} else {
 				// create dummy object to change the protection key
-				Data data = new Data("dummy");
+				Data data = new Data();
 				data.ttlSeconds(ttl);
 
 				// create a meta duplicate
-				data = data.setProtectedEntry().sign(newProtectionKey).duplicateMeta();
+				data = data.setProtectedEntry().sign(newProtectionKey, signatureFactory).duplicateMeta();
 
 				// the content will be protected after this put
 				if (oldProtectionKey == null) {
