diff --git a/old_test.java b/new_test.java
index 7cabf94..dd41048 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,8 @@
 
         try {
 
-            String valueString = fetcherMgr.downloadFileFromServer(requestUrl, RestfulMgrMock.defaultFileName);
+            String valueString = fetcherMgr.downloadFileFromServer(requestUrl, RestfulMgrMock.defaultFileName,
+                    "./disconf");
             Assert.assertEquals(RestfulMgrMock.defaultFileName, valueString);
 
         } catch (Exception e) {
