diff --git a/old_test.java b/new_test.java
index 8e1623a..d6e81ae 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,17 @@
 @Test
     public void testApplyDelegate() {
         Config config = Config.createConfig();
-        config.clearConnectionEnds();
+        int expectedDefaultTimeout = 390121;
+        config.getDefaultServerConnection().setTimeout(expectedDefaultTimeout);
         args = new String[2];
         args[0] = "-port";
         args[1] = "1234";
         jcommander.parse(args);
         delegate.applyDelegate(config);
-        ConnectionEnd newConEnd = config.getConnectionEnd();
-        assertNotNull(newConEnd);
-        assertTrue(newConEnd.getPort() == 1234);
-        assertTrue(newConEnd.getConnectionEndType() == ConnectionEndType.SERVER);
+        AliasedConnection actual = config.getDefaultServerConnection();
+        assertNotNull(actual);
+        assertThat(actual.getPort(), equalTo(1234));
+        assertThat(actual.getLocalConnectionEndType(), equalTo(ConnectionEndType.SERVER));
+        assertThat(actual.getTimeout(), equalTo(expectedDefaultTimeout));
+
     }
\ No newline at end of file
