diff --git a/old_product.java b/new_product.java
index e58ac32..a0784a9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -34,6 +34,8 @@ public static int getPrefixLength(final String filename) {
                         return 2;
                     }
                     return 3;
+                } else if (ch0 == UNIX_SEPARATOR) {
+                    return 1;
                 }
                 return NOT_FOUND;
 
