diff --git a/old_test.java b/new_test.java
index abb4f67..686cb46 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,9 +3,15 @@
         String segmentName = "foo_open";
         try (Storage s = createStorage()) {
             s.initialize(DEFAULT_EPOCH);
-            s.create(segmentName, null).join();
+            createSegment(segmentName, s);
+            Assert.assertTrue("Expected the segment to exist.", s.exists(segmentName, null).join());
             assertThrows("create() did not throw for existing StreamSegment.",
-                    s.create(segmentName, null),
+                    () -> createSegment(segmentName, s),
                     ex -> ex instanceof StreamSegmentExistsException);
+
+            // Delete and make sure it can be recreated.
+            s.openWrite(segmentName).thenCompose(handle -> s.delete(handle, null)).join();
+            createSegment(segmentName, s);
+            Assert.assertTrue("Expected the segment to exist.", s.exists(segmentName, null).join());
         }
     }
\ No newline at end of file
