diff --git a/old_product.java b/new_product.java
index 9998477..3bca6ce 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1,3 @@
-CompletableFuture<SegmentProperties> create(String streamSegmentName, Duration timeout);
\ No newline at end of file
+default CompletableFuture<SegmentProperties> create(String streamSegmentName, Duration timeout) {
+        return create(streamSegmentName, SegmentRollingPolicy.NO_ROLLING, timeout);
+    }
\ No newline at end of file
