diff --git a/old_test.java b/new_test.java
index 5c73860..0f54eed 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
         final OffsetDateTime sample = OffsetDateTime.of(2012, 3, 4, 23, 5, OFFSET_PONE);
         TemporalAdjuster adjuster = new TemporalAdjuster() {
             @Override
-            public Temporal doWithAdjustment(Temporal dateTime) {
+            public Temporal adjustInto(Temporal dateTime) {
                 return sample;
             }
         };
