diff --git a/old_test.java b/new_test.java
index 72e1e18..0e9efa7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test(groups={"tck"})
     public void test_with_MonthOfYear() {
-        assertEquals(MonthDay.of(6, 30).with(MonthOfYear.JANUARY), MonthDay.of(1, 30));
+        assertEquals(MonthDay.of(6, 30).with(Month.JANUARY), MonthDay.of(1, 30));
     }
\ No newline at end of file
