diff --git a/old_product.java b/new_product.java
index 13459d6..13e17ea 100644
--- a/old_product.java
+++ b/new_product.java
@@ -42,7 +42,7 @@ public static void equalizeLocalNaive( ImageUInt8 input , int radius , ImageUInt
 				localHistogram(input,x0,y0,x1,y1,histogram);
 
 				// only need to compute up to the value of the input pixel
-				int inputValue =  input.data[indexIn++] & 0xff;
+				int inputValue =  input.data[indexIn++] & 0xFF;
 				int sum = 0;
 				for( int i = 0; i <= inputValue; i++ ) {
 					sum += histogram[i];
