diff --git a/old_test.java b/new_test.java
index 74dc795..7981f4c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,6 @@
   public void getFqdnHostTest() throws UnknownHostException {
     Assert.assertEquals(NetworkAddressUtils.getFqdnHost(new InetSocketAddress("localhost", 0)),
         "localhost");
-    Assert.assertEquals(NetworkAddressUtils.getFqdnHost(new NetAddress("localhost", 0, 0, 0)),
-        "localhost");
+    Assert.assertEquals(
+        NetworkAddressUtils.getFqdnHost(new WorkerNetAddress().setHost("localhost")), "localhost");
   }
\ No newline at end of file
