diff --git a/old_test.java b/new_test.java
index 1d67f30..8c945db 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
     CircularBuffer<Integer> instance = new CircularBuffer<Integer>(0);
     try {
       instance.get();
-      fail("exception should be raised for getting from buffer which does not have data");
+      Assert.fail("exception should be raised for getting from buffer which does not have data");
     }
     catch (Exception bue) {
       assert (bue instanceof BufferUnderflowException);
@@ -15,7 +15,7 @@
     instance = new CircularBuffer<Integer>(10);
     try {
       instance.get();
-      fail("exception should be raised for getting from buffer which does not have data");
+      Assert.fail("exception should be raised for getting from buffer which does not have data");
     }
     catch (Exception bue) {
       assert (bue instanceof BufferUnderflowException);
@@ -27,10 +27,10 @@
 
     Integer i = instance.get();
     Integer j = instance.get();
-    assert(i == 0 && j == 1);
+    assert (i == 0 && j == 1);
 
     instance.add(10);
 
-    assert(instance.size() == 9);
-    assert(instance.get() == 2);
+    assert (instance.size() == 9);
+    assert (instance.get() == 2);
   }
\ No newline at end of file
