diff --git a/old_product.java b/new_product.java
index a7d72a7..295b5b6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -27,8 +27,8 @@ public void configure(Properties props)
 		this.autocommit = PropertiesHelper.getBoolean(Environment.AUTOCOMMIT, props);
 
 		try {
-			String driver = props.getProperty(CONFIG_CONNECTION_DRIVER_CLASS, "");
-			if (!driver.trim().equals("")){
+			String driver = props.getProperty(CONFIG_CONNECTION_DRIVER_CLASS);
+			if (driver != null && !driver.trim().equals("")){
 				loadClass(driver);
 			}
 			logger.debug(String.format(CONFIG_STATUS, url, username, minsize, maxsize, acquireIncrement, partcount, idleConnectionTestPeriod/1000, idleMaxAge/(60*1000)));
