diff --git a/old_test.java b/new_test.java
index e14b891..ed7f5ed 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
-@Test public void invoke() {
-    final Function func = _XQUERY_INVOKE;
+@Test public void evalURI() {
+    final Function func = _XQUERY_EVAL;
     // queries
-    query(func.args("src/test/resources/input.xq"), "XML");
-    error(func.args("src/test/resources/xxx.xq"), WHICHRES_X);
+    query(func.args(" xs:anyURI('src/test/resources/input.xq')"), "XML");
+    error(func.args(" xs:anyURI('src/test/resources/xxx.xq')"), WHICHRES_X);
   }
\ No newline at end of file
