diff --git a/old_product.java b/new_product.java
index c964e32..70a3d89 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
-public synchronized List<URIStatus> listStatus(final TachyonURI path)
-      throws IOException, TachyonException {
+public synchronized List<URIStatus> listStatus(final AlluxioURI path)
+      throws IOException, AlluxioException {
     return retryRPC(new RpcCallableThrowsTachyonTException<List<URIStatus>>() {
       @Override
-      public List<URIStatus> call() throws TachyonTException, TException {
+      public List<URIStatus> call() throws AlluxioTException, TException {
         List<URIStatus> result = new ArrayList<URIStatus>();
         for (alluxio.thrift.FileInfo fileInfo : mClient.listStatus(path.getPath())) {
           result.add(new URIStatus(ThriftUtils.fromThrift(fileInfo)));
