diff --git a/old_product.java b/new_product.java
index 622a1b9..6dc71ad 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-void onTopicCreated(TopicName topicName, Handler<AsyncResult<Void>> resultHandler) {
+void onTopicCreated(LogContext logContext, TopicName topicName, Handler<AsyncResult<Void>> resultHandler) {
         // XXX currently runs on the ZK thread, requiring a synchronized inFlight
         // is it better to put this check in the topic deleted event?
         Handler<Future<Void>> action = new Reconciliation("onTopicCreated") {
@@ -20,7 +20,7 @@ void onTopicCreated(TopicName topicName, Handler<AsyncResult<Void>> resultHandle
                                 // We now have the metadata we need to create the
                                 // resource...
                                 Topic kafkaTopic = TopicSerialization.fromTopicMetadata(metadataResult.result());
-                                reconcileOnTopicChange(topicName, kafkaTopic, fut);
+                                reconcileOnTopicChange(logContext, topicName, kafkaTopic, fut);
                             }
                         } else {
                             fut.handle(metadataResult.map((Void) null));
@@ -35,5 +35,5 @@ void onTopicCreated(TopicName topicName, Handler<AsyncResult<Void>> resultHandle
                 kafka.topicMetadata(topicName, handler);
             }
         };
-        executeWithTopicLockHeld(topicName, action).setHandler(resultHandler);
+        executeWithTopicLockHeld(logContext, topicName, action).setHandler(resultHandler);
     }
\ No newline at end of file
