diff --git a/old_test.java b/new_test.java
index d987a8b..67a3bb8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,9 +16,9 @@
             return Future.failedFuture("This should never happen");
         });
         mockK8s.setCreateResponse(resourceName, null);
-
+        LogContext logContext = LogContext.zkWatch("///", topicName.toString());
         Async async = context.async();
-        topicOperator.onTopicCreated(topicName, ar -> {
+        topicOperator.onTopicCreated(logContext, topicName, ar -> {
             assertSucceeded(context, ar);
             context.assertEquals(4, counter.get());
             mockK8s.assertExists(context, resourceName);
