diff --git a/old_product.java b/new_product.java
index df62c8c..7bc787f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -17,8 +17,7 @@ protected static History parseFileLog(Reader fileLog) throws IOException {
                 int year = Integer.parseInt(matcher.group(2));
                 int month = Integer.parseInt(matcher.group(3));
                 int day = Integer.parseInt(matcher.group(4));
-                Calendar calendar = new GregorianCalendar(year, month, day);
-                entry.setDate(calendar.getTime());
+                entry.setDate(newDate(year, month, day));
                 entry.setAuthor(matcher.group(5));
                 entry.setActive(true);
             } else {
