diff --git a/old_test.java b/new_test.java
index c366b28..3abf21a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -29,6 +29,19 @@
         assertEquals(0, e1.getFiles().size());
         assertTrue(e1.getMessage().contains("number 4"));
 
+        HistoryEntry e2 = result.getHistoryEntries().get(1);
+        assertNotNull(e2);
+
+        HistoryEntry e3 = result.getHistoryEntries().get(2);
+        assertNotNull(e3);
+
         HistoryEntry e4 = result.getHistoryEntries().get(3);
         assertEquals("1", e4.getRevision());
+
+        // Bug #16660: Months used to be off by one. Verify that they match
+        // the dates in the sample output above.
+        assertDate(e1, 2008, Calendar.AUGUST, 19);
+        assertDate(e2, 2008, Calendar.AUGUST, 19);
+        assertDate(e3, 2008, Calendar.AUGUST, 1);
+        assertDate(e4, 2008, Calendar.JULY, 30);
     }
\ No newline at end of file
