diff --git a/old_product.java b/new_product.java
index 0882396..c45808e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -55,7 +55,7 @@ public void validate(Object obj, Errors errors) {
 			if (OpenmrsUtil.nullSafeEquals(possibleDuplicate.getId(), personAddress.getId()))
 				continue;
 			
-			if (!possibleDuplicate.isBlank() && !personAddress.isBlank()
+			if (!possibleDuplicate.isVoided() && !possibleDuplicate.isBlank() && !personAddress.isBlank()
 			        && possibleDuplicate.toString().equalsIgnoreCase(personAddress.toString())) {
 				errors.reject("Patient.duplicateAddress", new Object[] { personAddress.toString() }, personAddress
 				        .toString()
