diff --git a/old_product.java b/new_product.java
index 17d53b0..44a4e3e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static void standardize(double[][] x) {
+public static java.util.function.Function<double[], double[]> standardize(double[][] x) {
         int n = x.length;
         int p = x[0].length;
 
@@ -9,17 +9,33 @@ public static void standardize(double[][] x) {
             }
         }
 
+        double[] scale = new double[p];
         for (int j = 0; j < p; j++) {
-            double scale = 0.0;
             for (int i = 0; i < n; i++) {
-                scale += Math.sqr(x[i][j]);
+                scale[j] += Math.sqr(x[i][j]);
             }
-            scale = Math.sqrt(scale / (n-1));
+            scale[j] = Math.sqrt(scale[j] / (n-1));
 
-            if (!Math.isZero(scale)) {
+            if (!Math.isZero(scale[j])) {
                 for (int i = 0; i < n; i++) {
-                    x[i][j] /= scale;
+                    x[i][j] /= scale[j];
                 }
             }
         }
+
+        return (double[] xi) -> {
+            if (xi.length != p)
+                throw new IllegalArgumentException(String.format("array size: %d, expected: %d", xi.length, p));
+
+            double[] y = new double[p];
+            for (int j = 0; j < p; j++) {
+                if (!Math.isZero(scale[j])) {
+                    y[j] = (xi[j] - center[j]) / scale[j];
+                } else {
+                    y[j] = 0.0;
+                }
+            }
+
+            return y;
+        };
     }
\ No newline at end of file
