diff --git a/old_test.java b/new_test.java
index b6200d5..96c44b7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,11 +6,6 @@
         assertEquals(r1, r2);
         assertEquals(r2, r1);
 
-        r1.setLinesVisible(true);
-        assertFalse(r1.equals(r2));
-        r2.setLinesVisible(true);
-        assertTrue(r1.equals(r2));
-
         r1.setSeriesLinesVisible(3, true);
         assertFalse(r1.equals(r2));
         r2.setSeriesLinesVisible(3, true);
@@ -26,11 +21,6 @@
         r2.setLegendLine(new Line2D.Double(1.0, 2.0, 3.0, 4.0));
         assertTrue(r1.equals(r2));
 
-        r1.setShapesVisible(true);
-        assertFalse(r1.equals(r2));
-        r2.setShapesVisible(true);
-        assertTrue(r1.equals(r2));
-
         r1.setSeriesShapesVisible(3, true);
         assertFalse(r1.equals(r2));
         r2.setSeriesShapesVisible(3, true);
@@ -41,11 +31,6 @@
         r2.setBaseShapesVisible(false);
         assertTrue(r1.equals(r2));
 
-        r1.setShapesFilled(true);
-        assertFalse(r1.equals(r2));
-        r2.setShapesFilled(true);
-        assertTrue(r1.equals(r2));
-
         r1.setSeriesShapesFilled(3, true);
         assertFalse(r1.equals(r2));
         r2.setSeriesShapesFilled(3, true);
