diff --git a/old_product.java b/new_product.java
index d0f33f1..2cc4c1f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,9 +10,6 @@
             return false;
         }
         XYLineAndShapeRenderer that = (XYLineAndShapeRenderer) obj;
-        if (!ObjectUtils.equal(this.linesVisible, that.linesVisible)) {
-            return false;
-        }
         if (!ObjectUtils.equal(
             this.seriesLinesVisible, that.seriesLinesVisible)
         ) {
@@ -24,9 +21,6 @@
         if (!ShapeUtils.equal(this.legendLine, that.legendLine)) {
             return false;
         }
-        if (!ObjectUtils.equal(this.shapesVisible, that.shapesVisible)) {
-            return false;
-        }
         if (!ObjectUtils.equal(
             this.seriesShapesVisible, that.seriesShapesVisible)
         ) {
@@ -35,9 +29,6 @@
         if (this.baseShapesVisible != that.baseShapesVisible) {
             return false;
         }
-        if (!ObjectUtils.equal(this.shapesFilled, that.shapesFilled)) {
-            return false;
-        }
         if (!ObjectUtils.equal(
             this.seriesShapesFilled, that.seriesShapesFilled)
         ) {
