diff --git a/old_product.java b/new_product.java
index 31587dd..ef64727 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public void removeBlocks(List<Long> blockIds) {
+public void removeBlocks(List<Long> blockIds, boolean deleteBlocksMetadata) {
     synchronized (mBlocks) {
       synchronized (mWorkers) {
         for (long blockId : blockIds) {
@@ -14,6 +14,9 @@ public void removeBlocks(List<Long> blockIds) {
             }
           }
           mLostBlocks.remove(blockId);
+          if (deleteBlocksMetadata) {
+            mBlocks.remove(blockId);
+          }
         }
       }
     }
