diff --git a/old_product.java b/new_product.java
index 83683b8..09eef49 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
-  public RequestStatus deleteResources(Predicate predicate)
+  public RequestStatus deleteResources(Request request, Predicate predicate)
       throws SystemException, UnsupportedPropertyException, NoSuchResourceException, NoSuchParentResourceException {
     Authentication authentication = AuthorizationHelper.getAuthentication();
 
@@ -9,5 +9,5 @@
       throw new AuthorizationException("The authenticated user does not have the appropriate authorizations to delete the requested resource(s)");
     }
 
-    return deleteResourcesAuthorized(predicate);
+    return deleteResourcesAuthorized(request, predicate);
   }
\ No newline at end of file
