diff --git a/old_product.java b/new_product.java
index dd1a1b4..43c7b39 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,18 @@
-public void updateProfile(final String dialogId, final List<NameValue> nameValues) {
+public void updateProfile(final String dialogId, final Integer clientId,
+      final Map<String, String> profile) {
     if (dialogId == null || dialogId.isEmpty())
       throw new IllegalArgumentException("dialogId cannot be null or empty");
 
-    if (nameValues == null || nameValues.isEmpty())
-      throw new IllegalArgumentException("nameValues cannot be null or empty");
+    if (profile == null || profile.isEmpty())
+      throw new IllegalArgumentException("profile cannot be null or empty");
 
     final JsonObject contentJson = new JsonObject();
+    if (clientId != null)
+      contentJson.addProperty(CLIENT_ID, clientId);
 
-    contentJson.add(NAME_VALUES, GsonSingleton.getGson().toJsonTree(nameValues));
+    contentJson.add(NAME_VALUES, GsonSingleton.getGson().toJsonTree(toNameValue(profile)));
 
     final Request request =
-        RequestBuilder.put("/v1/dialogs/" + dialogId + "/profile").withBodyJson(contentJson)
-            .build();
+        RequestBuilder.put(String.format(PATH_PROFILE, dialogId)).withBodyJson(contentJson).build();
     executeWithoutResponse(request);
   }
\ No newline at end of file
