diff --git a/old_product.java b/new_product.java
index df6b83c..010c65e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @Override
     public byte[] calculateMac(byte[] data) {
         writeMac.update(data);
-        LOGGER.debug("The MAC was caluculated over the following data: {}", ArrayConverter.bytesToHexString(data));
+        LOGGER.debug("The MAC was calculated over the following data: {}", ArrayConverter.bytesToHexString(data));
         byte[] result = writeMac.doFinal();
         LOGGER.debug("MAC result: {}", ArrayConverter.bytesToHexString(result));
         return result;
