diff --git a/old_product.java b/new_product.java
index e3be449..5ad4cd5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,9 +9,8 @@ public static void copyProperties(Object toObj,Object fromObj,String...includePr
                 BeanUtils.copyProperties(toObj, fromObj);
                 return;
             }catch (Exception e){
-                String pattern = "copyProperties exception,message:[{}],toObj:[{}],fromObj:[{}],includePropertyNames:[{}]";
-                String message = Slf4jUtil.format(pattern, e.getMessage(), toObj, fromObj, includePropertyNames);
-                throw new BeanOperationException(message, e);
+                String pattern = "copyProperties exception,toObj:[{}],fromObj:[{}],includePropertyNames:[{}]";
+                throw new BeanOperationException(Slf4jUtil.format(pattern, toObj, fromObj, includePropertyNames), e);
             }
         }
 
