diff --git a/old_test.java b/new_test.java
index 342d2e9..dbbed6d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,6 @@
         long sec = 0L;
         VariableThroughputTimer instance = new VariableThroughputTimer();
         int expResult = -1;
-        double result = instance.getRPSForSecond(sec);
+        double result = instance.getRPSForSecond(sec).getLeft();
         assertEquals(expResult, result, 0.01);
     }
\ No newline at end of file
