diff --git a/old_product.java b/new_product.java
index 162db69..1887ea9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-public String getPathRelativeToSourceRoot(File file, int stripCount) throws IOException {
+public String getPathRelativeToSourceRoot(File file, int stripCount)
+            throws IOException, ForbiddenSymlinkException {
         String sourceRoot = getSourceRootPath();
         if (sourceRoot == null) {
             throw new FileNotFoundException("Source Root Not Found");
