diff --git a/old_test.java b/new_test.java
index a10a1a4..eb5bcd9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
 		Map<String,Object> values = new HashMap<String,Object>();
 		values.put("creater", "abc");
 		getJBPMApplication().startProcess("defaultPackage.Trade", "aaa", XmlParseUtil.paramsToXml(values));
-		List<TaskVO> tasks = getJBPMApplication().queryTodoList("abc");
+		List<TaskVO> tasks = getJBPMApplication().queryTodoList("fhjl");
 		Assert.assertTrue(tasks.size() > 0);
 		//getJBPMApplication().removeProcessInstance(i);
 	}
\ No newline at end of file
