diff --git a/old_product.java b/new_product.java
index 9d080ba..68ed5e0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,19 @@
 public List<TaskVO> queryTodoList(String user) {
-		List<TaskVO> todos = new ArrayList<TaskVO>();
-		todos.addAll(this.queryTodoListCall(user, null, null));
-		todos.addAll(this.queryDelegateTodoList(user, null));
-		return todos;
+		UserTransaction owner = null;
+		try {
+			owner = this.startUserTransaction();
+			List<TaskVO> todos = new ArrayList<TaskVO>();
+			todos.addAll(this.queryTodoListCall(user, null, null));
+			todos.addAll(this.queryDelegateTodoList(user, null));
+			this.commitUserTransaction(owner);
+			return todos;
+		} catch (RuntimeException e) {
+			e.printStackTrace();
+			this.rollbackUserTransaction(owner);
+			throw e;
+		} catch (Exception e) {
+			e.printStackTrace();
+			this.rollbackUserTransaction(owner);
+			throw new RuntimeException(e.getCause());
+		}
 	}
\ No newline at end of file
