diff --git a/old_product.java b/new_product.java
index 0fc3527..0d5f67b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ public static CharSet getInstance(final String... setStrs) {
             return null;
         }
         if (setStrs.length == 1) {
-            CharSet common = COMMON.get(setStrs[0]);
+            final CharSet common = COMMON.get(setStrs[0]);
             if (common != null) {
                 return common;
             }
