diff --git a/old_test.java b/new_test.java
index b564b10..72d4f6d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void matchIcmpv6TypeTest() {
-        Criterion criterion = Criteria.matchIcmpv6Type((byte) 15);
+        Criterion criterion = Criteria.matchIcmpv6Type((byte) 250);
         ObjectNode result = criterionCodec.encode(criterion, context);
         assertThat(result.get("type").textValue(), is(criterion.type().toString()));
-        assertThat(result.get("icmpv6Type").asInt(), is(15));
+        assertThat(result.get("icmpv6Type").asInt(), is(250));
     }
\ No newline at end of file
