diff --git a/old_product.java b/new_product.java
index 33fe5d9..8aeb19f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,8 +9,8 @@ private ANode entries(final QueryContext ctx) throws QueryException {
     try {
       zf = new ZipFile(file);
       // create result node
-      final FElem root = new FElem(E_FILE, ZIP, ZIPURI);
-      root.atts.add(new FAttr(A_HREF, token(path.path()), root));
+      final FElem root = new FElem(E_FILE, new Atts().add(ZIP, ZIPURI), null);
+      root.add(new FAttr(A_HREF, token(path.path())));
       createEntries(paths(zf).iterator(), root, "");
       return root;
     } catch(final IOException ex) {
