diff --git a/old_test.java b/new_test.java
index b7c1256..f95e8dd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -32,7 +32,7 @@
         "'window':map {'size':3} }"), true);
 
     // check buggy options
-    error(_FT_CONTAINS.args("x", "x", " map { 'x':'y' }"), Err.INVALIDOPT);
-    error(_FT_CONTAINS.args("x", "x", " map { 'mode':'' }"), Err.INVALIDOPT);
-    error(_FT_CONTAINS.args("x", "x", " 1"), Err.ELMMAPTYPE);
+    error(_FT_CONTAINS.args("x", "x", " map { 'x':'y' }"), Err.INVALIDOPT_X);
+    error(_FT_CONTAINS.args("x", "x", " map { 'mode':'' }"), Err.INVALIDOPT_X);
+    error(_FT_CONTAINS.args("x", "x", " 1"), Err.ELMMAP_X_X_X);
   }
\ No newline at end of file
