diff --git a/old_product.java b/new_product.java
index 99a613f..ec97432 100644
--- a/old_product.java
+++ b/new_product.java
@@ -24,8 +24,8 @@ private int contains(final QueryContext qc) throws QueryException {
     }
 
     // check if occurrences are in valid range. if yes, return number of tokens
-    final long mn = occ != null ? checkItr(occ[0], qc) : 1;
-    final long mx = occ != null ? checkItr(occ[1], qc) : Long.MAX_VALUE;
+    final long mn = occ != null ? toLong(occ[0], qc) : 1;
+    final long mx = occ != null ? toLong(occ[1], qc) : Long.MAX_VALUE;
     if(mn == 0 && oc == 0) matches = FTNot.not(matches);
     return oc >= mn && oc <= mx ? Math.max(1, num) : 0;
   }
\ No newline at end of file
