diff --git a/old_product.java b/new_product.java
index 80087c0..5122eec 100644
--- a/old_product.java
+++ b/new_product.java
@@ -22,7 +22,7 @@ CompletableFuture<SegmentProperties> getStreamSegmentInfo(String streamSegmentNa
                 TimeoutTimer timer = new TimeoutTimer(timeout);
                 result = this.storage
                         .getStreamSegmentInfo(streamSegmentName, timer.getRemaining())
-                        .thenComposeAsync(si -> retrieveAttributes(si, timer.getRemaining()), this.executor)
+                        .thenComposeAsync(si -> retrieveState(si, timer.getRemaining()), this.executor)
                         .thenApply(si -> si.properties);
             }
         }
