diff --git a/old_test.java b/new_test.java
index 206bdbf..0ec8609 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-	public void validate_shouldBeInvalidIfBothPersonAndNameAreSet() throws Exception {
+	public void validate_shouldNeverHaveBothPersonAndNameSet() throws Exception {
 		//given
 		provider.setIdentifier("id");
 		provider.setPerson(new Person(1));
@@ -9,7 +9,6 @@
 		providerValidator.validate(provider, errors);
 		
 		//then
-		Assert.assertTrue(errors.hasErrors());
-		Assert.assertTrue(errors.hasFieldErrors("name"));
-		Assert.assertEquals("Provider.error.personOrName.required", errors.getFieldError("name").getCode());
+		Assert.assertFalse(errors.hasErrors());
+		Assert.assertFalse(errors.hasFieldErrors("name"));
 	}
\ No newline at end of file
