diff --git a/old_product.java b/new_product.java
index d69820a..c67d795 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,8 +11,7 @@ public void validate(Object obj, Errors errors) throws APIException {
 			errors.rejectValue("identifier", "Provider.error.identifier.required");
 		}
 		
-		if ((provider.getPerson() != null && StringUtils.isNotBlank(provider.getName()))
-		        || (provider.getPerson() == null && StringUtils.isBlank(provider.getName()))) {
+		if (provider.getPerson() == null && StringUtils.isBlank(provider.getName())) {
 			errors.rejectValue("name", "Provider.error.personOrName.required");
 		}
 		
