diff --git a/old_test.java b/new_test.java
index 642c294..56f7b5b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public void delete() throws BaseXException {
     // try to delete a package which is not installed
     try {
-      new RepoManager(context).delete("xyz");
+      new RepoManager(context).delete(token("xyz"));
       fail("Not installed package not detected.");
     } catch(final QueryException ex) {
       check(ex, Err.PKGNOTEXIST);
@@ -35,7 +35,7 @@
 
     // try to delete pkg3
     try {
-      new RepoManager(context).delete(PKG3ID);
+      new RepoManager(context).delete(token(PKG3ID));
       fail("Package involved in a dependency was deleted.");
     } catch(final QueryException ex) {
       check(ex, Err.PKGDEP);
