diff --git a/old_product.java b/new_product.java
index 5153d0e..4f54ee6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,10 @@
-public void delete(final String pkg) throws QueryException {
-    final byte[] pk = token(pkg);
+public void delete(final byte[] pkg) throws QueryException {
     boolean found = false;
     final TokenMap dict = repo.pkgDict();
     for(final byte[] nextPkg : dict) {
       if(nextPkg == null) continue;
       // a package can be deleted by its name or the name suffixed with its version
-      if(eq(nextPkg, pk) || eq(Package.name(nextPkg), pk)) {
+      if(eq(nextPkg, pkg) || eq(Package.name(nextPkg), pkg)) {
         // check if package to be deleted participates in a dependency
         final byte[] primPkg = primary(nextPkg);
         if(primPkg != null) PKGDEP.thrw(info, string(primPkg), pkg);
