diff --git a/old_test.java b/new_test.java
index 38fad90..831f623 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
     public void testNotBlank() {
-        QueryCriterion criterion1 = new NotNullCriterion("name");
-        QueryCriterion criterion2 = new NotEqCriterion("name", "");
-        QueryCriterion criterion3 = new AndCriterion(criterion1, criterion2);
+        QueryCriterion criterion1 = Criteria.notNull("name");
+        QueryCriterion criterion2 = Criteria.notEq("name", "");
+        QueryCriterion criterion3 = Criteria.and(criterion1, criterion2);
         
         assertEquals(criterion3, instance.notBlank("name").getQueryCriterion());
     }
\ No newline at end of file
