diff --git a/old_product.java b/new_product.java
index 2b8e31c..afc7241 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,8 +7,7 @@ public void process(NodeContainer nodeContainer) {
 					new Node(
 						inputEntity.getId(),
 						timestamp,
-						inputEntity.getUserName(),
-						inputEntity.getUserId(),
+						inputEntity.getUser(),
 						inputEntity.getVersion(),
 						inputEntity.getLatitude(),
 						inputEntity.getLongitude()
