diff --git a/old_product.java b/new_product.java
index 980bff3..87c0020 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,6 +4,11 @@ protected Connection obtainInternalConnection() throws SQLException {
 		
 		int acquireRetryAttempts = this.pool.getConfig().getAcquireRetryAttempts();
 		int acquireRetryDelay = this.pool.getConfig().getAcquireRetryDelay();
+		AcquireFailConfig acquireConfig = new AcquireFailConfig();
+		acquireConfig.setAcquireRetryAttempts(new AtomicInteger(acquireRetryAttempts));
+		acquireConfig.setAcquireRetryDelay(acquireRetryDelay);
+		acquireConfig.setLogMessage("Failed to acquire connection");
+
 		this.connectionHook = this.pool.getConfig().getConnectionHook();
 		do{ 
 			try { 
@@ -24,7 +29,7 @@ protected Connection obtainInternalConnection() throws SQLException {
 			} catch (Throwable t) {
 				// call the hook, if available.
 				if (this.connectionHook != null){
-					tryAgain = this.connectionHook.onAcquireFail(t);
+					tryAgain = this.connectionHook.onAcquireFail(t, acquireConfig);
 				} else {
 					logger.error("Failed to acquire connection. Sleeping for "+acquireRetryDelay+"ms. Attempts left: "+acquireRetryAttempts);
 
