diff --git a/old_test.java b/new_test.java
index dc4688b..06eb010 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,12 +3,12 @@
 		Pageable page = PageRequest.of(1, 10);
 
 		ModelMap model = new ModelMap();
-		userController.getAll(model, null, page, null);
+		userController.getAll(null, page, null);
 
 		model.clear();
-		userController.getAll(model, Role.ADMIN, page, null);
+		userController.getAll(Role.ADMIN, page, null);
 
 		model.clear();
-		userController.getAll(model, null, page, "user");
+		userController.getAll(null, page, "user");
 
 	}
\ No newline at end of file
