diff --git a/old_product.java b/new_product.java
index 9f5efe0..2c84a7d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,6 @@
 @Override
   public void setValueForKey(int groupKey, double newValue) {
-    if (_storageMode == StorageMode.ARRAY_STORAGE) {
+    if (groupKey != GroupKeyGenerator.INVALID_ID) {
       _resultArray[groupKey] = newValue;
-    } else {
-      _resultMap.put(groupKey, newValue);
-      _priorityQueue.put(groupKey, newValue);
     }
   }
\ No newline at end of file
