diff --git a/old_test.java b/new_test.java
index f76f03f..dd80aa2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-  public void testDescribeVersion() throws TTransportException, TException, UnknownHostException {
+  public void testDescribeThriftVersion() throws TTransportException, TException, UnknownHostException {
     CassandraCluster cassandraCluster = new CassandraClusterFactory(cassandraClient).create();
-    assertEquals("2.1.0",cassandraCluster.describeVersion());
+    assertEquals("2.1.0",cassandraCluster.describeThriftVersion());
   }
\ No newline at end of file
