diff --git a/old_test.java b/new_test.java
index b596390..408520a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,16 +1,16 @@
 @Test
-    public void testMarshall_Application() throws Exception {
-        Application app = new Application();
-        app.setApplicationId("app-id");
-        app.setOrganizationId("test-org");
-        app.setVersion("1.0");
+    public void testMarshall_Client() throws Exception {
+        Client client = new Client();
+        client.setClientId("client-id");
+        client.setOrganizationId("test-org");
+        client.setVersion("1.0");
 
         Assert.assertEquals("{"
                 + "\"organizationId\":\"test-org\","
-                + "\"applicationId\":\"app-id\","
+                + "\"clientId\":\"client-id\","
                 + "\"version\":\"1.0\","
                 + "\"contracts\":[]"
-            + "}", ESRegistryMarshalling.marshall(app).string());
+            + "}", ESRegistryMarshalling.marshall(client).string());
 
         Contract contract = new Contract();
         contract.setApiKey("12345");
@@ -18,11 +18,11 @@
         contract.setApiId("api-id");
         contract.setApiOrgId("api-test-org");
         contract.setApiVersion("1.7");
-        app.getContracts().add(contract);
+        client.getContracts().add(contract);
 
         Assert.assertEquals("{"
                 + "\"organizationId\":\"test-org\","
-                + "\"applicationId\":\"app-id\","
+                + "\"clientId\":\"client-id\","
                 + "\"version\":\"1.0\","
                 + "\"contracts\":["
                     + "{"
@@ -34,7 +34,7 @@
                         + "\"policies\":[]"
                     + "}"
                 + "]"
-            + "}", ESRegistryMarshalling.marshall(app).string());
+            + "}", ESRegistryMarshalling.marshall(client).string());
 
         Policy policy = new Policy();
         policy.setPolicyImpl("policy-1-impl");
@@ -48,7 +48,7 @@
 
         Assert.assertEquals("{"
                 + "\"organizationId\":\"test-org\","
-                + "\"applicationId\":\"app-id\","
+                + "\"clientId\":\"client-id\","
                 + "\"version\":\"1.0\","
                 + "\"contracts\":["
                     + "{"
@@ -69,5 +69,5 @@
                         + "]"
                     + "}"
                 + "]"
-            + "}", ESRegistryMarshalling.marshall(app).string());
+            + "}", ESRegistryMarshalling.marshall(client).string());
     }
\ No newline at end of file
