diff --git a/old_product.java b/new_product.java
index ef63bbe..bf6731f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static XContentBuilder marshall(Application bean) throws IOException {
+public static XContentBuilder marshall(Client bean) throws IOException {
         XContentBuilder builder = XContentFactory.jsonBuilder();
         marshallInto(bean, builder);
         return builder;
