diff --git a/old_product.java b/new_product.java
index e67050a..c6afdcd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-I_S_Resource findPlant(Properties ctx, final int adOrgId, final I_M_Warehouse warehouse, final int productId, int attributeSetInstanceId);
\ No newline at end of file
+I_S_Resource findPlant(final int adOrgId, final I_M_Warehouse warehouse, final int productId, int attributeSetInstanceId);
\ No newline at end of file
