diff --git a/old_test.java b/new_test.java
index e25a72f..867ec44 100644
--- a/old_test.java
+++ b/new_test.java
@@ -22,7 +22,7 @@
         long now = System.currentTimeMillis();
         SessionData data = store.newSessionData("1234", now - 20, now - 10, now - 20, TimeUnit.MINUTES.toMillis(10));
         Session session = cache.newSession(data);
-        cache.put("1234", session);
+        cache.add("1234", session);
         assertTrue(cache.contains("1234"));
 
         cache.renewSessionId("1234", "5678", "1234.foo", "5678.foo");
