diff --git a/old_product.java b/new_product.java
index 0693009..91645f0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,10 @@
 public void renewSessionId(String oldId, String oldExtendedId, String newId, String newExtendedId)
     {
+        Session session = null;
         try
         {
-            Session session = _sessionCache.renewSessionId(oldId, newId, oldExtendedId, newExtendedId); //swap the id over
+            //the use count for the session will be incremented in renewSessionId
+            session = _sessionCache.renewSessionId(oldId, newId, oldExtendedId, newExtendedId); //swap the id over
             if (session == null)
             {
                 //session doesn't exist on this context
@@ -16,4 +18,18 @@ public void renewSessionId(String oldId, String oldExtendedId, String newId, Str
         {
             LOG.warn(e);
         }
+        finally
+        {
+            if (session != null)
+            {
+                try
+                {
+                    _sessionCache.release(newId, session);
+                }
+                catch (Exception e)
+                {
+                    LOG.warn(e);
+                }
+            }
+        }
     }
\ No newline at end of file
