diff --git a/old_test.java b/new_test.java
index e3551d3..dcef5c3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,8 +3,8 @@
         final LocalDate date = LocalDate.of(2010, 12, 3);
         CalendricalFormatter f = new CalendricalFormatter() {
             @Override
-            public String print(DateTimeAccessor calendrical) {
-                assertEquals(calendrical, date);
+            public String print(DateTimeAccessor accessor) {
+                assertEquals(accessor, date);
                 return "PRINTED";
             }
             @Override
