diff --git a/old_test.java b/new_test.java
index 1dbf1ee..02d1783 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,7 @@
         Async async0 = context.async(2);
         mockK8s.setCreateResponse(mapName, null)
                 .createConfigMap(TopicSerialization.toConfigMap(kubeTopic, cmPredicate), ar -> async0.countDown());
-        mockK8s.setDeleteResponse(topicName, null);
+        mockK8s.setDeleteResponse(mapName, null);
         mockTopicStore.setCreateTopicResponse(topicName, null)
                 .create(privateTopic, ar-> async0.countDown());
         mockTopicStore.setDeleteTopicResponse(topicName, null);
