diff --git a/old_product.java b/new_product.java
index 17ef3d0..3a0a54e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -50,7 +50,7 @@ void reconcile(final HasMetadata involvedObject,
                     logger.debug("cm deleted in k8s => delete topic from kafka and from topicStore");
                     enqueue(new DeleteKafkaTopic(kafkaTopic.getTopicName(), involvedObject, ar -> {
                         if (ar.succeeded()) {
-                            enqueue(new DeleteFromTopicStore(kafkaTopic.getTopicName(), involvedObject, reconciliationResultHandler));
+                            enqueue(new DeleteFromTopicStore(privateTopic.getTopicName(), involvedObject, reconciliationResultHandler));
                         } else {
                             reconciliationResultHandler.handle(ar);
                         }
@@ -60,9 +60,9 @@ void reconcile(final HasMetadata involvedObject,
             } else if (kafkaTopic == null) {
                 // it was deleted in kafka so delete in k8s and privateState
                 logger.debug("topic deleted in kafkas => delete cm from k8s and from topicStore");
-                enqueue(new DeleteConfigMap(k8sTopic.getTopicName(), ar -> {
+                enqueue(new DeleteConfigMap(privateTopic.getOrAsMapName(), ar -> {
                     if (ar.succeeded()) {
-                        enqueue(new DeleteFromTopicStore(k8sTopic.getTopicName(), involvedObject, reconciliationResultHandler));
+                        enqueue(new DeleteFromTopicStore(privateTopic.getTopicName(), involvedObject, reconciliationResultHandler));
                     } else {
                         reconciliationResultHandler.handle(ar);
                     }
