diff --git a/old_product.java b/new_product.java
index e617ff5..5936e11 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,4 @@
-public void predict(Dataframe newData) {
-        logger.info("predict()");
-
+public void _predict(Dataframe newData) {
         //load all trainables on the bundles
         initBundle();
 
@@ -8,9 +6,13 @@ public void predict(Dataframe newData) {
         bundle.setParallelized(isParallelized());
 
         //run the pipeline
-        AbstractTransformer dataTransformer = (AbstractTransformer) bundle.get(DT_KEY);
-        if(dataTransformer != null) {
-            dataTransformer.transform(newData);
+        AbstractNumericalScaler numericalScaler = (AbstractNumericalScaler) bundle.get(NS_KEY);
+        if(numericalScaler != null) {
+            numericalScaler.transform(newData);
+        }
+        AbstractCategoricalEncoder categoricalEncoder = (AbstractCategoricalEncoder) bundle.get(CE_KEY);
+        if(categoricalEncoder != null) {
+            categoricalEncoder.transform(newData);
         }
         AbstractFeatureSelector featureSelector = (AbstractFeatureSelector) bundle.get(FS_KEY);
         if(featureSelector != null) {
@@ -18,7 +20,4 @@ public void predict(Dataframe newData) {
         }
         AbstractModeler modeler = (AbstractModeler) bundle.get(ML_KEY);
         modeler.predict(newData);
-        if(dataTransformer != null) {
-            dataTransformer.denormalize(newData);
-        }
     }
\ No newline at end of file
