diff --git a/old_product.java b/new_product.java
index 61f15c5..8c63d8c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public void process( double c_x , double c_y , double scale , double orientation ,
 						 int imageIndex ,
 						 double pixelScale ,
-						 SurfFeature desc )
+						 BrightFeature desc )
 	{
 		image = ss.getPyramidLayer(imageIndex);
 		derivX = ss.getDerivativeX(imageIndex);
