diff --git a/old_test.java b/new_test.java
index 6d81373..b8cf26b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void toAclActions() {
     for (Mode.Bits bits : Mode.Bits.values()) {
-      Assert.assertEquals(bits, new AclActions(bits.toAclActions()).toModeBits());
+      Assert.assertEquals(bits, new AclActions(bits.toAclActionSet()).toModeBits());
     }
   }
\ No newline at end of file
