diff --git a/old_test.java b/new_test.java
index 84739fb..e8585f4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,9 +3,17 @@
         @Cleanup
         TestContext context = new TestContext(DEFAULT_CONFIG);
 
-        // Check behavior for non-existent segments (in Storage).
+        // Check behavior for non-existent segments (in Storage) that are actually supposed to be empty.
         context.transactionAggregators[0].initialize(TIMEOUT).join();
-        Assert.assertTrue("isDeleted() flag not set on metadata for deleted segment.", context.transactionAggregators[0].getMetadata().isDeleted());
+        Assert.assertFalse("isDeleted() flag not set on metadata for deleted segment.", context.transactionAggregators[0].getMetadata().isDeleted());
+
+        // Check behavior for non-existent segments (in Storage) that are not supposed to be empty.
+        val sm3 = (UpdateableSegmentMetadata) context.transactionAggregators[3].getMetadata();
+        sm3.setLength(1L);
+        sm3.setStorageLength(1L);
+        context.transactionAggregators[3].initialize(TIMEOUT).join();
+        Assert.assertTrue("isDeleted() flag not set on metadata for deleted segment.", sm3.isDeleted());
+        Assert.assertTrue("isDeletedInStorage() flag not set on metadata for deleted segment.", sm3.isDeletedInStorage());
 
         // Check behavior for already-sealed segments (in storage, but not in metadata)
         context.storage.create(context.transactionAggregators[1].getMetadata().getName(), TIMEOUT).join();
