diff --git a/old_test.java b/new_test.java
index 2b07a39..868cfdc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
                     () -> log.truncate(0, TIMEOUT),
                     ex -> ex instanceof IllegalStateException);
 
-            log.initialize(TIMEOUT).join();
+            log.initialize(TIMEOUT);
             TreeMap<Long, byte[]> writeData = populate(log, WRITE_COUNT);
             ArrayList<Long> seqNos = new ArrayList<>(writeData.keySet());
 
