diff --git a/old_test.java b/new_test.java
index 8b4e74e..6a7c96c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,5 +4,5 @@
         now = new Date();
         Date date = DateUtil.string2Date("2012-12-03 00:00:00", DatePattern.COMMON_DATE_AND_TIME);
         LOGGER.debug(DateExtensionUtil.getIntervalForView(now, date));
-        LOGGER.debug(DateUtil.getIntervalTime(now, date) + "");
+        LOGGER.debug(DateExtensionUtil.getIntervalTime(now, date) + "");
     }
\ No newline at end of file
