diff --git a/old_test.java b/new_test.java
index 03e1cb7..1ceb9b7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,9 @@
-@Test(enabled = false)
+@Test(expectedExceptions = HttpResponseException.class)
    public void testCreateUser() throws Exception {
-      // TODO
+      adminConnection.getApi().deleteUser(PREFIX);
+      adminConnection.getApi().createUser(new User(PREFIX));
+      orgUser = adminConnection.getApi().getUser(PREFIX);
+      assertNotNull(orgUser);
+      assertEquals(orgUser.getUsername(), PREFIX);
+      assertNotNull(orgUser.getPrivateKey());
    }
\ No newline at end of file
