diff --git a/old_product.java b/new_product.java
index 918e26c..6448c47 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,30 +1,23 @@
 public static List<Date> getIntervalDayList(String fromDateString,String toDateString,String datePattern){
-        List<Date> dateList = new ArrayList<Date>();
-        //***************************************************************/
-        Date beginDate = DateUtil.string2Date(fromDateString, datePattern);
-        Date endDate = DateUtil.string2Date(toDateString, datePattern);
+
+        Date fromDate = DateUtil.string2Date(fromDateString, datePattern);
+        Date toDate = DateUtil.string2Date(toDateString, datePattern);
+
+        Date minDate = fromDate.before(toDate) ? fromDate : toDate;
+        Date maxDate = fromDate.before(toDate) ? toDate : fromDate;
+
         // ******重置时间********
-        Date beginDateReset = DateUtil.getFirstDateOfThisDay(beginDate);
-        Date endDateReset = DateUtil.getLastDateOfThisDay(endDate);
-        //***************************************************************/
+        Date beginDateReset = DateUtil.getFirstDateOfThisDay(minDate);
+        Date endDateReset = DateUtil.getLastDateOfThisDay(maxDate);
+
+        List<Date> dateList = new ArrayList<Date>();
+        dateList.add(beginDateReset);
+
         // 相隔的天数
         int intervalDay = DateUtil.getIntervalDay(beginDateReset, endDateReset);
-        //***************************************************************/
-        Date minDate = beginDateReset;
-        if (beginDateReset.equals(endDateReset)){
-            minDate = beginDateReset;
-        }else if (beginDateReset.before(endDateReset)){
-            minDate = beginDateReset;
-        }else{
-            minDate = endDateReset;
-        }
-        //***************************************************************/
-        dateList.add(minDate);
-        //***************************************************************/
-        if (intervalDay > 0){
-            for (int i = 0; i < intervalDay; ++i){
-                dateList.add(DateUtil.addDay(minDate, i + 1));
-            }
+        for (int i = 0; i < intervalDay; ++i){
+            dateList.add(DateUtil.addDay(beginDateReset, i + 1));
         }
+
         return dateList;
     }
\ No newline at end of file
