diff --git a/old_test.java b/new_test.java
index 1c584bc..c2c7fbb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
     public void testGetIntervalDayList(){
-        List<Date> dates = DateExtensionUtil.getIntervalDayList(FROMSTRING, TOSTRING, DatePattern.COMMON_DATE_AND_TIME);
-        LOGGER.debug(JsonUtil.format(dates));
+        LOGGER.debug(JsonUtil.format(DateExtensionUtil.getIntervalDayList(FROMSTRING, TOSTRING, DatePattern.COMMON_DATE_AND_TIME)));
+        LOGGER.debug(JsonUtil.format(DateExtensionUtil.getIntervalDayList(TOSTRING, FROMSTRING, DatePattern.COMMON_DATE_AND_TIME)));
     }
\ No newline at end of file
