diff --git a/old_test.java b/new_test.java
index 65769e9..273306f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
 		}
 
 		DebugModelStuff stuff = new DebugModelStuff((int) modelVal);
-		SimpleRansacCommon<Integer> ransac = new RandsacDebug(stuff, stuff);
+		SimpleRansacCommon<double[],Integer> ransac = new RandsacDebug(stuff, stuff);
 		double param[] = new double[]{modelVal};
 
 		ransac.selectMatchSet(dataSet, 4, 5, param);
