diff --git a/old_test.java b/new_test.java
index 7aeb355..32a84d3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
 	public void standard_F32() {
 		for( int i = 4; i <= 4; i += 2 ) {
 
-			WaveletDesc_F32 desc = FactoryWaveletDaub.standard_F32(i);
+			WaveletCoefficient_F32 desc = FactoryWaveletDaub.standard_F32(i);
 
 			double sumScaling = UtilWavelet.sumCoefficients(desc.scaling);
 			double sumWavelet = UtilWavelet.sumCoefficients(desc.wavelet);
