diff --git a/old_product.java b/new_product.java
index a27f591..2be734e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
-public static WaveletDesc_F32 standard_F32( int J ) {
+public static WaveletCoefficient_F32 standard_F32( int J ) {
 		if( J != 4 ) {
 			throw new IllegalArgumentException("Only 4 is currently supported");
 		}
 
-		WaveletDesc_F32 ret = new WaveletDesc_F32();
+		WaveletCoefficient_F32 ret = new WaveletCoefficient_F32();
 
 		ret.border = new BorderIndex1D_Wrap();
 		ret.offsetScaling = 0;
