diff --git a/old_test.java b/new_test.java
index 53ddf4d..741441d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,12 +8,12 @@
     Attribute[] attrs = DescriptorUtils.parseDescriptor(descriptor);
 
     // prepare the data
-    double[][] data = Utils.randomDoubles(rng, descriptor, datasize);
+    double[][] data = Utils.randomDoubles(rng, descriptor, false, datasize);
     Collection<Integer> missings = Lists.newArrayList();
     String[] sData = prepareData(data, attrs, missings);
-    Dataset expected = DataLoader.generateDataset(descriptor, sData);
+    Dataset expected = DataLoader.generateDataset(descriptor, false, sData);
 
-    Dataset dataset = DataLoader.generateDataset(descriptor, sData);
+    Dataset dataset = DataLoader.generateDataset(descriptor, false, sData);
     
     assertEquals(expected, dataset);
   }
\ No newline at end of file
