diff --git a/old_product.java b/new_product.java
index a6d80da..97ffdba 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static Dataset generateDataset(String descriptor, FileSystem fs, Path path) throws DescriptorException,
+public static Dataset generateDataset(String descriptor, boolean regression, FileSystem fs, Path path) throws DescriptorException,
                                                                                     IOException {
     Attribute[] attrs = DescriptorUtils.parseDescriptor(descriptor);
     
@@ -22,5 +22,5 @@ public static Dataset generateDataset(String descriptor, FileSystem fs, Path pat
     
     scanner.close();
     
-    return new Dataset(attrs, values, id);
+    return new Dataset(attrs, values, id, regression);
   }
\ No newline at end of file
