diff --git a/old_test.java b/new_test.java
index 22a5e7d..c9180eb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,4 +11,22 @@
 		testClass.terminateAllConnections();
 		verify(mockConnectionsScheduler, mockKeepAliveScheduler, mockPartition, mockConnectionHandles, mockConnection);
 
+		// same test but to cover the finally section
+		reset(mockConnectionsScheduler, mockKeepAliveScheduler, mockPartition, mockConnectionHandles, mockConnection);
+		expect(mockConnectionHandles.poll()).andReturn(mockConnection).anyTimes();
+		mockConnection.internalClose();
+		expectLastCall().once().andThrow(new RuntimeException()).once();
+		expect(mockPartition.getFreeConnections()).andReturn(mockConnectionHandles).anyTimes();
+		expect(mockConnection.getOriginatingPartition()).andReturn(mockPartition).anyTimes();
+		replay(mockConnectionsScheduler, mockKeepAliveScheduler, mockPartition, mockConnectionHandles, mockConnection);
+		
+		// test.
+		try{
+			testClass.terminateAllConnections();
+			fail("Should throw exception");
+		} catch (RuntimeException e){
+			// do nothing
+		}
+		verify(mockConnectionsScheduler, mockKeepAliveScheduler, mockPartition, mockConnectionHandles, mockConnection);
+		
 	}
\ No newline at end of file
