diff --git a/old_product.java b/new_product.java
index df3842a..4a9208e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,15 +1,21 @@
 protected void terminateAllConnections(){
-		// close off all connections.
-		for (int i=0; i < this.partitionCount; i++) {
-			ConnectionHandle conn;
-			while ((conn = this.partitions[i].getFreeConnections().poll()) != null){
-				postDestroyConnection(conn);
+		if (this.terminationLock.tryLock()){
+			try{
+				// close off all connections.
+				for (int i=0; i < this.partitionCount; i++) {
+					ConnectionHandle conn;
+					while ((conn = this.partitions[i].getFreeConnections().poll()) != null){
+						postDestroyConnection(conn);
 
-				try {
-					conn.internalClose();
-				} catch (SQLException e) {
-					logger.error(e);
+						try {
+							conn.internalClose();
+						} catch (SQLException e) {
+							logger.error(e);
+						}
+					}
 				}
+			} finally {
+				this.terminationLock.unlock();
 			}
 		}
 	}
\ No newline at end of file
