diff --git a/old_test.java b/new_test.java
index ee80625..8c79f83 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,7 @@
         if (!file.exists()) {
             fail("Unable to create a temporary file.");
         }
-        FileUtils.delete(file);
+        boolean status = FileUtils.delete(file);
+        assertTrue("delete returned a failed status", status);
         assertFalse("Temporary file exists after attempting deletion", file.exists());
     }
\ No newline at end of file
