diff --git a/old_product.java b/new_product.java
index 18b24c1..5a41df6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,6 @@
-protected void clearResultSetHandles(boolean internalClose) throws SQLException {
-		if (!internalClose){
-			this.resultSetHandles.clear();
-		} else {
-			ResultSet rs = null;
-			while ((rs=this.resultSetHandles.poll()) != null) {
-				rs.close();
-			}
+protected void closeAndClearResultSetHandles() throws SQLException {
+		ResultSet rs = null;
+		while ((rs=this.resultSetHandles.poll()) != null) {
+			rs.close();
 		}
 	}
\ No newline at end of file
