diff --git a/old_test.java b/new_test.java
index bb308e3..da6a339 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
 		ConcurrentLinkedQueue<ResultSet> mockResultSetHandles = createNiceMock(ConcurrentLinkedQueue.class);
 
 		// alternate constructor
-		StatementHandle handle = new StatementHandle(mockStatement, null);
+		StatementHandle handle = new StatementHandle(mockStatement, null, true);
 		Field field = handle.getClass().getDeclaredField("resultSetHandles");
 		field.setAccessible(true);
 		field.set(handle, mockResultSetHandles);
@@ -14,7 +14,7 @@
 		mockResultSet.close();
 		expectLastCall().once();
 		replay(mockResultSetHandles, mockResultSet, mockStatement);
-		handle.clearResultSetHandles(true);
+		handle.closeAndClearResultSetHandles();
 		verify(mockResultSet, mockResultSetHandles, mockStatement);
 		
 	}
\ No newline at end of file
