diff --git a/old_product.java b/new_product.java
index 2ae5723..db5996a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,6 +13,8 @@
             throw new GenieBadRequestException("Application id inconsistent with id passed in.");
         }
 
-        LOG.debug("Called with app " + updateApp.toString());
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Called with app " + updateApp.toString());
+        }
         return this.applicationRepo.save(updateApp);
     }
\ No newline at end of file
