diff --git a/old_test.java b/new_test.java
index 5bd84da..5616119 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testSetSubnetAddres() throws Exception {
-        reachability.setSubnetAddres(ip4Address);
-        result = reachability.getSubnetAddres();
+        reachability.setSubnetAddress(ip4Address);
+        result = reachability.getSubnetAddress();
         assertThat(result, is(ip4Address));
     }
\ No newline at end of file
