diff --git a/old_product.java b/new_product.java
index d77c00c..4a3f8fd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,16 +8,11 @@ private Value query(final QueryContext qc) throws QueryException {
       // bind variables and context item
       for(final Map.Entry<String, Value> binding : bindings(2, qc).entrySet()) {
         final String k = binding.getKey();
-        final Value v = binding.getValue();
-        if(!v.isItem()) throw BXCL_ITEM.get(info, v);
-        final Item it = (Item) v;
-        final Type t = v.type;
-        if(it instanceof FuncItem) throw FIVALUE.get(info, t);
-
-        final Object value = t instanceof NodeType ? v.serialize() : Token.string(it.string(info));
-        if(k.isEmpty()) cq.context(value, t.toString());
-        else cq.bind(k, value, t.toString());
+        final Value value = binding.getValue();
+        if(k.isEmpty()) cq.context(value);
+        else cq.bind(k, value);
       }
+
       // evaluate query
       while(cq.more()) {
         final String result = cq.next();
