diff --git a/old_test.java b/new_test.java
index a65e376..3c9f9dc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,12 @@
         _CLIENT_QUERY.args("$a", "1") + '+' + _CLIENT_QUERY.args("$b", "2"), "3");
     query(_CLIENT_QUERY.args(conn(), "\"declare variable $a external; $a*2\"",
         " map { 'a': 1 }"), "2");
+    query(_CLIENT_QUERY.args(conn(), "\"declare variable $a external; count($a)\"",
+        " map { 'a': () }"), "0");
+    query(_CLIENT_QUERY.args(conn(), "\"declare variable $a external; count($a)\"",
+        " map { 'a': (1 to 5) }"), "5");
+    query(_CLIENT_QUERY.args(conn(), "\"declare variable $a external; $a\"",
+        " map { 'a': (1,<a/>,'a') }"), "1<a/>a");
     // query errors
     error(_CLIENT_QUERY.args(conn(), "x"), Err.NOCTX);
-    error(_CLIENT_QUERY.args(conn(), "\"$a\"", " map { 'a': (1,2) }"), Err.BXCL_ITEM);
   }
\ No newline at end of file
