diff --git a/old_product.java b/new_product.java
index b1cbffc..5913e95 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public synchronized void addFeature(TransactionTrace txn, String key, Object val, Type type) {
         final boolean trace = LOG.isTraceEnabled();
         final boolean debug = LOG.isDebugEnabled();
-        String txn_id = txn.getTransactionId();
+        long txn_id = txn.getTransactionId();
         
         // Add the attribute if it's new
         if (!this.attributes.containsKey(key)) {
