diff --git a/old_product.java b/new_product.java
index 76707fb..b75c1ef 100644
--- a/old_product.java
+++ b/new_product.java
@@ -27,19 +27,4 @@
       prerequisiteCheck.setFailReason(getFailReason(prerequisiteCheck, request));
       return;
     }
-
-    // no failues so far, check to see if any hosts are in MM so that this check
-    // will produce a warning
-    for (Host host : hosts) {
-      MaintenanceState maintenanceState = host.getMaintenanceState(cluster.getClusterId());
-      if (maintenanceState != MaintenanceState.OFF) {
-        prerequisiteCheck.getFailedOn().add(host.getHostName());
-      }
-    }
-
-    if (!prerequisiteCheck.getFailedOn().isEmpty()) {
-      prerequisiteCheck.setStatus(PrereqCheckStatus.WARNING);
-      prerequisiteCheck.setFailReason(
-          getFailReason(KEY_HOSTS_IN_MM_WARNING, prerequisiteCheck, request));
-    }
   }
\ No newline at end of file
