diff --git a/old_product.java b/new_product.java
index f14a804..44cd859 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,7 +7,7 @@ protected XDOM buildPresentationXDOM(String html, DocumentReference targetDocume
 
             Map<String, String> galleryParameters = Collections.emptyMap();
             ExpandedMacroBlock gallery = new ExpandedMacroBlock("gallery", galleryParameters, renderer, false);
-            gallery.addChildren(xhtmlParser.parse(new StringReader(html)).getChildren());
+            gallery.addChild(xhtmlParser.parse(new StringReader(html)));
 
             return new XDOM(Collections.singletonList((Block) gallery));
         } catch (Exception e) {
