diff --git a/old_test.java b/new_test.java
index ec095b7..2a78cfa 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     {
         final DocumentReference reference = new DocumentReference("wiki", "Space", "Page");
         final DocumentModelBridge mockDocumentModelBridge = getMockery().mock(DocumentModelBridge.class);
-        final Block galleryContent = new GroupBlock();
+        final XDOM galleryContent = new XDOM(Collections.<Block> emptyList());
         getMockery().checking(new Expectations()
         {
             {
@@ -19,7 +19,7 @@
 
                 try {
                     oneOf(mockXHTMLParser).parse(with(aNonNull(StringReader.class)));
-                    will(returnValue(new XDOM(Collections.singletonList(galleryContent))));
+                    will(returnValue(galleryContent));
                 } catch (ParseException e) {
                     Assert.fail(e.getMessage());
                 }
