diff --git a/old_test.java b/new_test.java
index 72ec6d4..6175ff7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,5 +4,5 @@
     query(EXISTS.args(_MAP_ENTRY.args(1, 2)), true);
     query(EXISTS.args(_MAP_MERGE.args(_MAP_ENTRY.args(1, 2))), "true");
     error(EXISTS.args(_MAP_ENTRY.args("()", 2)), Err.EMPTYFOUND);
-    error(EXISTS.args(_MAP_ENTRY.args("(1,2)", 2)), Err.SEQFOUND);
+    error(EXISTS.args(_MAP_ENTRY.args("(1,2)", 2)), Err.SEQFOUND_X);
   }
\ No newline at end of file
