diff --git a/old_product.java b/new_product.java
index f3dcdb9..5b16981 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public static String formatDate(Date date) {
-        return date == null ? "null" : DATE_FORMAT.format(date);
+public static String formatDate(LocalDate date) {
+        return date == null ? "null" : date.format(DATE_FORMATTER);
     }
\ No newline at end of file
