diff --git a/old_test.java b/new_test.java
index ebe8b06..dee2855 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,8 @@
 @Test
     public void testFormatDate() {
         assertEquals("null", FormatUtil.formatDate(null));
-        try {
-            assertEquals("01-01-2017", FormatUtil.formatDate(DATE_FORMAT.parse("01-01-2017")));
-        } catch (ParseException e) {
-            assertEquals("01-01-2017", e.getMessage());
-        }
+        assertEquals("01/01/2017", FormatUtil.formatDate(LocalDate.parse("2017-01-01")));
+        assertEquals(null, FormatUtil.formatStringDate(null));
+        assertEquals(null, FormatUtil.formatStringDate("Unparseable"));
+        assertEquals(LocalDate.parse("2017-01-01"), FormatUtil.formatStringDate("01/01/2017"));
     }
\ No newline at end of file
