diff --git a/old_test.java b/new_test.java
index 8463d84..5dadf68 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,12 @@
 @Test
     public void hasDefinitionAt() {
         Definitions instance = new Definitions();
+        String[] type= new String[1];
+        type[0]="";
         instance.addTag(1, "found", "", "");
-        assertEquals(instance.hasDefinitionAt("found", 0), false);
-        assertEquals(instance.hasDefinitionAt("found", 1), true);
-        assertEquals(instance.hasDefinitionAt("found", 2), false);
-        assertEquals(instance.hasDefinitionAt("notFound", 0), false);
-        assertEquals(instance.hasDefinitionAt("notFound", 1), false);
+        assertEquals(instance.hasDefinitionAt("found", 0, type), false);
+        assertEquals(instance.hasDefinitionAt("found", 1, type), true);
+        assertEquals(instance.hasDefinitionAt("found", 2, type), false);
+        assertEquals(instance.hasDefinitionAt("notFound", 0, type), false);
+        assertEquals(instance.hasDefinitionAt("notFound", 1, type), false);
     }
\ No newline at end of file
