diff --git a/old_product.java b/new_product.java
index 992272b..c818e30 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,8 +2,8 @@ public int getTotalRunCount ()
     {
         int total = 0;
 
-        for (RunSequence seq : sequences) {
-            total += seq.size();
+        for (short[] seq : sequences) {
+            total += sequenceSize(seq);
         }
 
         return total;
