diff --git a/old_test.java b/new_test.java
index facf5ad..3ae4b11 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,15 +1,15 @@
 @Test
     public void testGetTotalRunCount ()
     {
-        System.out.println("getTotalRunCount");
+        System.out.println("\n+++ getTotalRunCount");
 
         RunTable instance = createHorizontalInstance();
-        int expResult = 11;
+        int expResult = 9;
         int result = instance.getTotalRunCount();
         assertEquals(expResult, result);
 
         instance = createVerticalInstance();
-        expResult = 19;
+        expResult = 16;
         result = instance.getTotalRunCount();
         assertEquals(expResult, result);
     }
\ No newline at end of file
