diff --git a/old_product.java b/new_product.java
index 6f5487f..667824b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 private Iter list(final QueryContext qc) throws QueryException, IOException {
-    final Path dir = checkPath(0, qc).toRealPath();
+    final Path dir = toPath(0, qc).toRealPath();
     final boolean rec = optionalBool(1, qc);
     final Pattern pat = exprs.length == 3 ? Pattern.compile(IOFile.regex(
-        string(checkStr(exprs[2], qc))), Prop.CASE ? 0 : Pattern.CASE_INSENSITIVE) : null;
+        string(toToken(exprs[2], qc))), Prop.CASE ? 0 : Pattern.CASE_INSENSITIVE) : null;
 
     final TokenList list = new TokenList();
     list(dir.getNameCount(), dir, list, rec, pat);
