diff --git a/old_test.java b/new_test.java
index 6bb0f42..f5c01a5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,17 @@
 @Test
   public void toThrift() throws IOException {
-    CreateUfsFileOptions options = CreateUfsFileOptions.defaults();
-    String owner = "test-owner";
-    String group = "test-group";
-    short mode = Constants.DEFAULT_FILE_SYSTEM_MODE;
+    Random random = new Random();
+    String owner = CommonUtils.randomAlphaNumString(10);
+    String group = CommonUtils.randomAlphaNumString(10);
+    Mode mode = new Mode((short) random.nextInt());
 
-    options.setPermission(new Permission(owner, group, mode));
+    CreateUfsFileOptions options = CreateUfsFileOptions.defaults();
+    options.setOwner(owner);
+    options.setGroup(group);
+    options.setMode(mode);
 
     CreateUfsFileTOptions thriftOptions = options.toThrift();
     Assert.assertEquals(owner, thriftOptions.getOwner());
     Assert.assertEquals(group, thriftOptions.getGroup());
-    Assert.assertEquals(mode, thriftOptions.getMode());
+    Assert.assertEquals(mode.toShort(), thriftOptions.getMode());
   }
\ No newline at end of file
