diff --git a/old_product.java b/new_product.java
index a63bcfe..fabc86a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,12 @@
 public CompleteUfsFileTOptions toThrift() {
     CompleteUfsFileTOptions options = new CompleteUfsFileTOptions();
-    if (!mPermission.getOwner().isEmpty()) {
-      options.setOwner(mPermission.getOwner());
+    if (!mOwner.isEmpty()) {
+      options.setOwner(mOwner);
     }
-    if (!mPermission.getGroup().isEmpty()) {
-      options.setGroup(mPermission.getGroup());
+    if (!mGroup.isEmpty()) {
+      options.setGroup(mGroup);
     }
-    short mode = mPermission.getMode().toShort();
+    short mode = mMode.toShort();
     if (mode != Constants.INVALID_MODE) {
       options.setMode(mode);
     }
