diff --git a/old_product.java b/new_product.java
index fa65ce0..7d75ba6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,9 +14,9 @@
       throw new SchedulerException(Messages.getInstance().getString(ERR_WRONG_BLOCKER_TYPE, blockoutName), ex);
     }
 
-    deleteBlockout(blockoutName);
     Trigger oldBlockoutTrigger = (Trigger) oldBlockout;
     JobDetail jd = scheduler.getJobDetail(oldBlockoutTrigger.getJobName(), oldBlockoutTrigger.getJobGroup());
+    deleteBlockout(blockoutName);
 
     newBlockoutTrigger.setJobName(jd.getName());
     newBlockoutTrigger.setJobGroup(jd.getGroup());
