diff --git a/old_product.java b/new_product.java
index 03da895..a081df0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,8 @@
 private void notifyCreditAvailable() {
 		checkState(partitionRequestClient != null, "Tried to send task event to producer before requesting a queue.");
 
-		partitionRequestClient.notifyCreditAvailable(this);
+		// We should skip the notification if this channel is already released.
+		if (!isReleased.get()) {
+			partitionRequestClient.notifyCreditAvailable(this);
+		}
 	}
\ No newline at end of file
