diff --git a/old_product.java b/new_product.java
index 4458cf4..96b2b0d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
 public ArrayList<ArrayList<NGram>> extract(Collection<String[]> sentences, int maxNGramSize, int minFrequency) {
-        ArrayList<Set<NGram>> features = new ArrayList<Set<NGram>>(maxNGramSize + 1);
-        features.add(new HashSet<NGram>());
+        ArrayList<Set<NGram>> features = new ArrayList<>(maxNGramSize + 1);
+        features.add(new HashSet<>());
         for (int n = 1; n <= maxNGramSize; n++) {
-            Map<NGram, Integer> candidates = new HashMap<NGram, Integer>();
-            Set<NGram> feature = new HashSet<NGram>();
+            Map<NGram, Integer> candidates = new HashMap<>();
+            Set<NGram> feature = new HashSet<>();
             features.add(feature);
             Set<NGram> feature_1 = features.get(n - 1);
             for (String[] sentence : sentences) {
@@ -44,9 +44,9 @@ public ArrayList<ArrayList<NGram>> extract(Collection<String[]> sentences, int m
         }
         
         // filter out stop words
-        ArrayList<ArrayList<NGram>> results = new ArrayList<ArrayList<NGram>>();
+        ArrayList<ArrayList<NGram>> results = new ArrayList<>();
         for (Set<NGram> ngrams : features) {
-            ArrayList<NGram> result = new ArrayList<NGram>();
+            ArrayList<NGram> result = new ArrayList<>();
             results.add(result);
             for (NGram ngram : ngrams) {
                 boolean stopWord = true;
