diff --git a/old_test.java b/new_test.java
index 76bce59..8a9c049 100644
--- a/old_test.java
+++ b/new_test.java
@@ -37,7 +37,7 @@
             }
         }
 
-        // Now, map all the parents (stand-alone segments).
+        // Now, map all the batches.
         for (String name : storageSegments) {
             String parentName = StreamSegmentNameUtils.getParentStreamSegmentName(name);
             if (parentName != null) {
