diff --git a/old_test.java b/new_test.java
index 80ff520..a0b188b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,7 @@
             assertTrue(e.getMessage().contains("Value cannot be null"));
         }
 
-        assertEquals("0.0015", bitcoinValueToPlainString(BigInteger.valueOf(150000)));
+        assertEquals("0.0015", bitcoinValueToPlainString(Coin.valueOf(150000)));
         assertEquals("1.23", bitcoinValueToPlainString(toNanoCoins("1.23")));
 
         assertEquals("0.1", bitcoinValueToPlainString(toNanoCoins("0.1")));
