diff --git a/old_product.java b/new_product.java
index d1212f2..5b9be4f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,5 @@
 @GET
   @Produces("text/plain")
   public Response getClusters(String body, @Context HttpHeaders headers, @Context UriInfo ui) {
-
-    hasPermission(Request.Type.GET, null);
     return handleRequest(headers, body, ui, Request.Type.GET, createClusterResource(null));
   }
\ No newline at end of file
