diff --git a/old_test.java b/new_test.java
index d32cb35..f4f1522 100644
--- a/old_test.java
+++ b/new_test.java
@@ -71,7 +71,7 @@
       query(_DB_CREATE.args(dbname, "()", "()", " { 'updindex':" + b + "() }"));
       query(_DB_INFO.args(dbname) + "//updindex/text()", b ? "ON" : "OFF");
     }
-    assertEquals(context.options.is(MainOptions.UPDINDEX), false);
+    assertEquals(context.options.bool(MainOptions.UPDINDEX), false);
 
     final String[] nopt = { "maxcats", "maxlen", "indexsplitsize", "ftindexsplitsize" };
     for(final String k : nopt) {
