diff --git a/old_test.java b/new_test.java
index f6c591b..537c1da 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     // large number of small files
     InodeDirectory inodeDirectory = createInodeDirectory();
     int nFiles = (int) 1E5;
-    Inode[] inodes = new Inode[nFiles];
+    Inode<?>[] inodes = new Inode[nFiles];
     for (int i = 0; i < nFiles; i++) {
       inodes[i] = createInodeFile(i + 1);
       inodeDirectory.addChild(inodes[i]);
