diff --git a/old_product.java b/new_product.java
index 1eba63f..91a848b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,15 +1,15 @@
-public LinkedHashMap<String, List<AlignmentInterval.Row>> packAlignments(
+public PackedAlignments packAlignments(
             Iterator<Alignment> iter,
             int end,
             AlignmentTrack.RenderOptions renderOptions) {
 
         if(renderOptions == null) renderOptions = new AlignmentTrack.RenderOptions();
 
-        LinkedHashMap<String, List<AlignmentInterval.Row>> packedAlignments = new LinkedHashMap<String, List<Row>>();
+        LinkedHashMap<String, List<Row>> packedAlignments = new LinkedHashMap<String, List<Row>>();
         boolean pairAlignments = renderOptions.isViewPairs() || renderOptions.isPairedArcView();
 
         if (iter == null || !iter.hasNext()) {
-            return packedAlignments;
+            return new PackedAlignments(packedAlignments, renderOptions);
         }
 
         if (renderOptions.groupByOption == null) {
@@ -49,6 +49,6 @@ public LinkedHashMap<String, List<AlignmentInterval.Row>> packAlignments(
             packedAlignments.put("", alignmentRows);
         }
 
-        return packedAlignments;
+        return new PackedAlignments(packedAlignments, renderOptions);
 
     }
\ No newline at end of file
