diff --git a/old_test.java b/new_test.java
index e1fd039..b8754ea 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testSetShortDescription() throws SQLException
     {
         String desc = "short";
-        bf.setShortDescription(desc);
+        bf.setShortDescription(context, desc);
 
         assertThat("testSetShortDescription 0", bf.getShortDescription(),
                 notNullValue());
