diff --git a/old_product.java b/new_product.java
index 476408b..cb5635a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public List<EcfInfoDto> getVacantEventClassFamiliesByApplicationToken(String applicationToken) throws Exception {
-        ParameterizedTypeReference<List<EcfInfoDto>> typeRef = new ParameterizedTypeReference<List<EcfInfoDto>>() {
-        };
-        ResponseEntity<List<EcfInfoDto>> entity = restTemplate.exchange(restTemplate.getUrl() + "vacantEventClassFamiliesByAppToken/" + applicationToken,
-                HttpMethod.GET, null, typeRef);
-        return entity.getBody();
-    }
\ No newline at end of file
+public List<EcfInfoDto> getVacantEventClassFamiliesByApplicationToken(String applicationToken)
+      throws Exception {
+    ResponseEntity<List<EcfInfoDto>> entity = restTemplate.exchange(
+        restTemplate.getUrl() + "vacantEventClassFamilies/" + applicationToken,
+        HttpMethod.GET, null, new ParameterizedTypeReference<List<EcfInfoDto>>() {});
+    return entity.getBody();
+  }
\ No newline at end of file
