diff --git a/old_test.java b/new_test.java
index f33ad47..2d0cc85 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testGetVacantEventClassFamiliesByApplicationToken() throws Exception {
         ApplicationDto application = createApplication(tenantAdminDto);
         EventClassFamilyDto eventClassFamily = createEventClassFamily(application.getTenantId());
-        createApplicationEventFamilyMap(application.getId(), eventClassFamily.getId(), 1);
+        createApplicationEventFamilyMap(application.getApplicationToken(), eventClassFamily.getId(), 1);
 
         loginTenantDeveloper(tenantDeveloperUser);
         List<EcfInfoDto> vacantEcfs = client.getVacantEventClassFamiliesByApplicationToken(application.getApplicationToken());
