diff --git a/old_test.java b/new_test.java
index 12dec1e..0c56c20 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,7 +18,7 @@
      * just not provide suggested fixes for these few cases when the javac infrastructure gets in
      * the way.
      */
-    if (methodHasModifier(Modifier.STATIC).matches(methodTree, state)) {
+    if (Matchers.<MethodTree>hasModifier(Modifier.STATIC).matches(methodTree, state)) {
       CharSequence methodSource = state.getSourceForNode((JCMethodDecl) methodTree);
       if (methodSource != null) {
         String methodString = "@Test\n" + methodSource.toString().replaceFirst(" static ", " ");
