diff --git a/old_product.java b/new_product.java
index a562d1e..51b9a2e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@
     @SuppressWarnings("unchecked")
     Matcher<MethodTree> methodMatcher = allOf(
         not(methodNameStartsWith("test")),
-        methodHasModifier(Modifier.PUBLIC),
+        Matchers.<MethodTree>hasModifier(Modifier.PUBLIC),
         methodReturns(VOID_TYPE),
         methodHasParameters(),
         enclosingClass(isJUnit3TestClass));
