diff --git a/old_test.java b/new_test.java
index 4b998d4..f3fb252 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,12 +6,14 @@
     Date start = new Date();
 
     String exampleText = "Howdy " + UUID.randomUUID().toString(); // gotta be unique
-    CreateExampleOptions createOptions =
-            new CreateExampleOptions.Builder(workspaceId, exampleIntent, exampleText).build();
+    CreateExampleOptions createOptions = new CreateExampleOptions.Builder(workspaceId, exampleIntent, exampleText)
+        .build();
     service.createExample(createOptions).execute();
 
     try {
-      GetExampleOptions getOptions = new GetExampleOptions.Builder(workspaceId, exampleIntent, exampleText).build();
+      GetExampleOptions getOptions = new GetExampleOptions.Builder(workspaceId, exampleIntent, exampleText)
+          .includeAudit(true)
+          .build();
       Example response = service.getExample(getOptions).execute();
       assertNotNull(response);
       assertNotNull(response.getExampleText());
@@ -29,8 +31,8 @@
       fail(ex.getMessage());
     } finally {
       // Clean up
-      DeleteExampleOptions deleteOptions =
-              new DeleteExampleOptions.Builder(workspaceId, exampleIntent, exampleText).build();
+      DeleteExampleOptions deleteOptions = new DeleteExampleOptions.Builder(workspaceId, exampleIntent, exampleText)
+          .build();
       service.deleteExample(deleteOptions).execute();
     }
   }
\ No newline at end of file
