diff --git a/old_product.java b/new_product.java
index 78d90a4..25c5ce6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,5 +3,8 @@ public ServiceCall<Example> getExample(GetExampleOptions getExampleOptions) {
     RequestBuilder builder = RequestBuilder.get(String.format("/v1/workspaces/%s/intents/%s/examples/%s",
         getExampleOptions.workspaceId(), getExampleOptions.intent(), getExampleOptions.text()));
     builder.query(VERSION, versionDate);
+    if (getExampleOptions.includeAudit() != null) {
+      builder.query("include_audit", String.valueOf(getExampleOptions.includeAudit()));
+    }
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(Example.class));
   }
\ No newline at end of file
