diff --git a/old_product.java b/new_product.java
index b6804d4..c4d64f3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery sizeGe(String propName, int size) {
-        criterion = criterion.and(criterionBuilder.sizeGe(propName, size));
+        criterion = criterion.and(Criteria.sizeGe(propName, size));
         return this;
     }
\ No newline at end of file
