diff --git a/old_test.java b/new_test.java
index 7fa7c6b..9260a2b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,9 @@
 @Test public void testUnlockAndFreeze() throws Exception {
         new File("lock_server_timestamp.dat").delete();
         server = SimulatingServerProxy.newProxyInstance(LockService.class, LockServiceImpl.create(
-                new LockServerOptions() {
-                    private static final long serialVersionUID = 1L;
-                    @Override public TimeDuration getMaxAllowedClockDrift() {
-                        return SimpleTimeDuration.of(0, TimeUnit.MILLISECONDS);
-                    }
-                }), 10);
+                LockServerOptions.builder()
+                        .maxAllowedClockDrift(SimpleTimeDuration.of(0, TimeUnit.MILLISECONDS))
+                        .build()), 10);
 
         LockRequest request = LockRequest.builder(ImmutableSortedMap.of(lock1, LockMode.WRITE))
                 .timeoutAfter(SimpleTimeDuration.of(1, TimeUnit.SECONDS)).doNotBlock().build();
