diff --git a/old_product.java b/new_product.java
index 3a0b60b..f89aaa8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1,5 @@
-@NonIdempotent boolean unlockAndFreeze(HeldLocksToken token);
\ No newline at end of file
+@POST
+    @Path("unlock-and-freeze")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    @NonIdempotent boolean unlockAndFreeze(HeldLocksToken token);
\ No newline at end of file
