diff --git a/old_product.java b/new_product.java
index 0651c50..c076bd1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
-@SuppressWarnings("unchecked")
-    protected static <K> void loadAll(Cache<K, ?> cache, final Set<K> keys, final boolean replaceExistingValues)
+protected static <K> void loadAll(Cache<K, ?> cache, final Set<K> keys, final boolean replaceExistingValues)
         throws Exception {
         IgniteCache<K, Object> cacheCp = (IgniteCache<K, Object>)cache;
 
