diff --git a/old_product.java b/new_product.java
index 6863f73..ae24afc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 private void refreshToken(String accessToken, String refreshToken, String authCode) {
-        Retrofit retrofit = GhostApiUtils.getRetrofit("http://blog.com", Helpers.getProdHttpClient());
+        Retrofit retrofit = GhostApiUtils.getRetrofit(BLOG_URL, Helpers.getProdHttpClient());
         MockRetrofit mockRetrofit = Helpers.getMockRetrofit(retrofit, Helpers.getIdealNetworkBehavior());
         BehaviorDelegate<GhostApiService> delegate = mockRetrofit.create(GhostApiService.class);
         GhostApiService api = new MockGhostApiService(delegate, true);
@@ -10,7 +10,7 @@ private void refreshToken(String accessToken, String refreshToken, String authCo
         token.setAccessToken(accessToken);
         token.setRefreshToken(refreshToken);
 
-        AuthService authService = new AuthService(api, credSource, credSink);
+        AuthService authService = new AuthService(BLOG_URL, api, credSource, credSink);
         authService.listen(listener);
         authService.refreshToken(token);
     }
\ No newline at end of file
