diff --git a/old_test.java b/new_test.java
index 54f3cf3..7cfa23f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void refreshToken_expiredAccessToken() {
         refreshToken("expired-access-token", "refresh-token", "auth-code");
 
-        verify(credSink).setLoggedIn(true);
+        verify(credSink).setLoggedIn(BLOG_URL, true);
         verify(listener).onNewAuthToken(argThat(hasProperty("accessToken", is("refreshed-access-token"))));
         verify(listener).onNewAuthToken(argThat(hasProperty("refreshToken", is("refresh-token"))));
     }
\ No newline at end of file
