diff --git a/old_product.java b/new_product.java
index 9ae1044..943ba6c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -21,7 +21,7 @@ void query(final String path) throws IOException {
         }
 
         // initializes the response with query serialization options
-        http.sopts().parse(xq.parameters(context));
+        http.sopts().assign(xq.parameters(context));
         http.initResponse();
       }
       // run command
