diff --git a/old_product.java b/new_product.java
index c4345e2..8216855 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,6 +13,8 @@
             throw new GenieBadRequestException("Command id inconsistent with id passed in.");
         }
 
-        LOG.debug("Called to update command with id " + id + " " + updateCommand.toString());
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Called to update command with id " + id + " " + updateCommand.toString());
+        }
         return this.commandRepo.save(updateCommand);
     }
\ No newline at end of file
