diff --git a/old_test.java b/new_test.java
index f3ab825..5998cad 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,16 +2,16 @@
   public void testGetConnectors() {
     MConnector connector;
 
-    when(requests.readConnector(null)).thenReturn(connectorBean(connector(1), connector(2)));
+    when(resourceRequests.readConnector(null)).thenReturn(connectorBean(connector(1), connector(2)));
     Collection<MConnector> connectors = client.getConnectors();
     assertEquals(2, connectors.size());
 
-    client.getResourceBundle(1);
+    client.getConnectorConfigResourceBundle(1);
     connector = client.getConnector(1);
     assertEquals(1, connector.getPersistenceId());
 
     connector = client.getConnector(2);
-    client.getResourceBundle(2);
+    client.getConnectorConfigResourceBundle(2);
     assertEquals(2, connector.getPersistenceId());
 
     connectors = client.getConnectors();
@@ -28,6 +28,6 @@
     connector = client.getConnector("A3");
     assertNull(connector);
 
-    verify(requests, times(1)).readConnector(null);
-    verifyNoMoreInteractions(requests);
+    verify(resourceRequests, times(1)).readConnector(null);
+    verifyNoMoreInteractions(resourceRequests);
   }
\ No newline at end of file
