diff --git a/old_product.java b/new_product.java
index 7da5595..133f71f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,14 +1,14 @@
 public Collection<MConnector> getConnectors() {
-    if(allConnectors) {
+    if(isAllConnectors) {
       return connectors.values();
     }
 
-    ConnectorBean bean = requests.readConnector(null);
-    allConnectors = true;
+    ConnectorBean bean = resourceRequests.readConnector(null);
+    isAllConnectors = true;
     for(MConnector connector : bean.getConnectors()) {
       connectors.put(connector.getPersistenceId(), connector);
     }
-    bundles = bean.getResourceBundles();
+    connectorConfigBundles = bean.getResourceBundles();
 
     return connectors.values();
   }
\ No newline at end of file
