diff --git a/old_test.java b/new_test.java
index f07af73..bbb941c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,6 +7,6 @@
         int y = 0;
         RunsTable instance = createHorizontalInstance();
         int expResult = level;
-        int result = instance.getPixel(x, y);
+        int result = instance.getValue(x, y);
         assertEquals(expResult, result);
     }
\ No newline at end of file
