diff --git a/old_product.java b/new_product.java
index 785903b..c417036 100644
--- a/old_product.java
+++ b/new_product.java
@@ -17,28 +17,38 @@ public static Item restoreItem(Context c, InProgressSubmission is,
         }
 
         // Even though we are restoring an item it may not have the proper dates. So let's
-        // double check that it has a date accessioned and date issued, and if either of those dates
-        // are not set then set them to today.
+        // double check its associated date(s)
         DCDate now = DCDate.getCurrent();
         
-        // If the item doesn't have a date.accessioned, create one.
+        // If the item doesn't have a date.accessioned, set it to today
         DCValue[] dateAccessioned = item.getDC("date", "accessioned", Item.ANY);
         if (dateAccessioned.length == 0)
         {
 	        item.addDC("date", "accessioned", null, now.toString());
         }
         
-        // create issue date if not present
+        // If issue date is set as "today" (literal string), then set it to current date
+        // In the below loop, we temporarily clear all issued dates and re-add, one-by-one,
+        // replacing "today" with today's date.
+        // NOTE: As of DSpace 4.0, DSpace no longer sets an issue date by default
         DCValue[] currentDateIssued = item.getDC("date", "issued", Item.ANY);
-        if (currentDateIssued.length == 0)
+        item.clearDC("date", "issued", Item.ANY);
+        for (DCValue dcv : currentDateIssued)
         {
-            DCDate issued = new DCDate(now.getYear(),now.getMonth(),now.getDay(),-1,-1,-1);
-            item.addDC("date", "issued", null, issued.toString());
+            if(dcv.value!=null && dcv.value.equalsIgnoreCase("today"))
+            {
+                DCDate issued = new DCDate(now.getYear(),now.getMonth(),now.getDay(),-1,-1,-1);
+                item.addDC(dcv.element, dcv.qualifier, dcv.language, issued.toString());
+            }
+            else if(dcv.value!=null)
+            {
+                item.addDC(dcv.element, dcv.qualifier, dcv.language, dcv.value);
+            }
         }
         
         // Record that the item was restored
-		String provDescription = "Restored into DSpace on "+ now + " (GMT).";
-		item.addDC("description", "provenance", "en", provDescription);
+        String provDescription = "Restored into DSpace on "+ now + " (GMT).";
+        item.addDC("description", "provenance", "en", provDescription);
 
         return finishItem(c, item, is);
     }
\ No newline at end of file
