diff --git a/old_test.java b/new_test.java
index 0300cc0..732d652 100644
--- a/old_test.java
+++ b/new_test.java
@@ -23,9 +23,9 @@
         assertThat("testRestoreItem 0", result, equalTo(is.getItem()));
 
         //Make sure that restore did NOT insert a new provenance message with today's date
-        DCValue[] provMsgValues = result.getMetadata("dc", "description", "provenance", Item.ANY);
+        Metadatum[] provMsgValues = result.getMetadata("dc", "description", "provenance", Item.ANY);
         int i = 1;
-        for(DCValue val : provMsgValues)
+        for(Metadatum val : provMsgValues)
         {
             assertFalse("testRestoreItem " + i, val.value.startsWith(provDescriptionBegins));
             i++;
