diff --git a/old_product.java b/new_product.java
index dc903e0..fbf8a1f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public boolean delete(TachyonURI path, boolean recursive) throws IOException {
+public boolean delete(AlluxioURI path, boolean recursive) throws IOException {
     try {
       mFileSystem.delete(path, DeleteOptions.defaults().setRecursive(recursive));
       return true;
