diff --git a/old_product.java b/new_product.java
index 3fc2fbc..b9b26ab 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public boolean optimizedParam( List<CalibrationObservation> observations ,
 								   List<Point2D_F64> grid ,
-								   Zhang99ParamAll initial ,
-								   Zhang99ParamAll found ,
+								   Zhang99AllParam initial ,
+								   Zhang99AllParam found ,
 								   UnconstrainedLeastSquares optimizer )
 	{
 		if( optimizer == null ) {
@@ -15,11 +15,10 @@ public boolean optimizedParam( List<CalibrationObservation> observations ,
 		initial.convertToParam(model);
 
 		Zhang99OptimizationFunction func = new Zhang99OptimizationFunction(
-				initial.createNew(), grid,observations);
+				initial.createLike(), grid,observations);
 
 		Zhang99OptimizationJacobian jacobian = new Zhang99OptimizationJacobian(
-				initial.assumeZeroSkew,initial.radial.length,initial.includeTangential,
-				observations,grid);
+				(CalibParamPinholeRadial)found.getIntrinsic(), observations,grid);
 
 		optimizer.setFunction(func,jacobian);
 		optimizer.initialize(model,1e-10,1e-25*observations.size());
