diff --git a/old_test.java b/new_test.java
index dca5685..c89b999 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,8 @@
     public void testSetStringIfNotNull() {
         String key = "nullableProperty";
         String value = "someValue";
-        Settings.setString(key, value);
-        Settings.setStringIfNotNull(key, null); // NO-OP
-        String expResults = Settings.getString(key);
+        getSettings().setString(key, value);
+        getSettings().setStringIfNotNull(key, null); // NO-OP
+        String expResults = getSettings().getString(key);
         Assert.assertEquals(expResults, value);
     }
\ No newline at end of file
