diff --git a/old_test.java b/new_test.java
index b48d49a..a5f3ea8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void concatTest() {
-    Array seq1 = Array.empty();
-    Array seq2 = Array.empty();
+    XQArray seq1 = XQArray.empty();
+    XQArray seq2 = XQArray.empty();
     final int n = 200_000;
     for(int i = 0; i < n; i++) {
       final Value value = Int.get(i);
@@ -11,7 +11,7 @@
 
     assertEquals(n, seq1.arraySize());
     assertEquals(n, seq2.arraySize());
-    final Array seq = seq1.concat(seq2);
+    final XQArray seq = seq1.concat(seq2);
     assertEquals(2 * n, seq.arraySize());
 
     for(int i = 0; i < 2 * n; i++) {
