diff --git a/old_product.java b/new_product.java
index a034bbc..0064365 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public static Expr[] concat(final Expr[] source, final Expr... add) {
     final int sl = source.length, al = add.length;
     final Expr[] tmp = new Expr[sl + al];
-    System.arraycopy(source, 0, tmp, 0, sl);
-    System.arraycopy(add, 0, tmp, sl, al);
+    Array.copy(source, sl, tmp);
+    Array.copyFromStart(add, al, tmp, sl);
     return tmp;
   }
\ No newline at end of file
