diff --git a/old_product.java b/new_product.java
index b63dcff..f1111d2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,11 +5,11 @@ default FloatColumn difference(DateColumn column1, DateColumn column2, ChronoUni
             int c1 = column1.getInt(r);
             int c2 = column2.getInt(r);
             if (c1 == FloatColumn.MISSING_VALUE || c2 == FloatColumn.MISSING_VALUE) {
-                newColumn.add(FloatColumn.MISSING_VALUE);
+                newColumn.append(FloatColumn.MISSING_VALUE);
             } else {
                 LocalDate value1 = PackedLocalDate.asLocalDate(c1);
                 LocalDate value2 = PackedLocalDate.asLocalDate(c2);
-                newColumn.add(unit.between(value1, value2));
+                newColumn.append(unit.between(value1, value2));
             }
         }
         return newColumn;
