diff --git a/old_product.java b/new_product.java
index 321b863..7ef6cbb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,16 +4,8 @@ public void add(long xVal, double yVal)
       {
          maxX = xVal;
       }
-      if (yVal > maxY)
-      {
-         maxY = yVal;
-      }
       if (xVal < minX)
       {
          minX = xVal;
       }
-      if (yVal < minY)
-      {
-         minY = yVal;
-      }
    }
\ No newline at end of file
