diff --git a/old_test.java b/new_test.java
index bc254af..ee758b7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,8 +5,11 @@
       long xVal = 10;
       double yVal = 5.0;
       instance.add(xVal, yVal);
+
+      double[] minMax = instance.getMinMaxY(-1);
+
       assertEquals(xVal, instance.getMinX());
       assertEquals(xVal, instance.getMaxX());
-      assertEquals(yVal, instance.getMinY(), 0.001);
-      assertEquals(yVal, instance.getMaxY(), 0.001);
+      assertEquals(yVal, minMax[0], 0.001);
+      assertEquals(yVal, minMax[1], 0.001);
    }
\ No newline at end of file
