diff --git a/old_test.java b/new_test.java
index 8df729b..bee2486 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,11 @@
-@Test(enabled = false)
+@Test(enabled = true)
    public void testListAsyncJobs() throws Exception {
       Set<AsyncJob<?>> response = client.getAsyncJobClient().listAsyncJobs();
       assert null != response;
+
       long asyncJobCount = response.size();
       assertTrue(asyncJobCount >= 0);
+
       for (AsyncJob<?> asyncJob : response) {
          assert asyncJob.getCmd() != null : asyncJob;
          assert asyncJob.getUserId() >= 0 : asyncJob;
