diff --git a/old_product.java b/new_product.java
index 751e15c..ffa2610 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @GET
-   @QueryParams(keys = "command", values = "listAsyncJobs")
+   @QueryParams(keys = { "command", "listAll" }, values = { "listAsyncJobs", "true" })
    @ResponseParser(ParseAsyncJobsFromHttpResponse.class)
    @ExceptionParser(ReturnEmptySetOnNotFoundOr404.class)
    ListenableFuture<Set<AsyncJob<?>>> listAsyncJobs(ListAsyncJobsOptions... options);
\ No newline at end of file
