diff --git a/old_test.java b/new_test.java
index 7b59a2a..37e8ccf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,7 +16,6 @@
     mMaster.commitBlock(worker2, 3L, "MEM", 3L, 1L);
     mMaster.removeBlocks(workerBlocks, false /* delete */);
     Assert.assertEquals(1L, mMaster.getBlockInfo(1L).getBlockId());
-    Assert.assertTrue(mMaster.getLostBlocks().contains(1L));
 
     // Test removeBlocks with delete
     mMaster.removeBlocks(workerBlocks, true /* delete */);
