diff --git a/old_product.java b/new_product.java
index b476789..40c7fa1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,11 +13,15 @@ public void removeBlocks(List<Long> blockIds, boolean delete) {
               worker.updateToRemovedBlock(true, blockId);
             }
           }
+          // Two cases here:
+          // 1) For delete: delete the block metadata.
+          // 2) For free: keep the block metadata. mLostBlocks will be changed in
+          // processWorkerRemovedBlocks
           if (delete) {
+            // Make sure blockId is removed from mLostBlocks when the block metadata is deleted.
+            // Otherwise blockId in mLostBlock can be dangling index if the metadata is gone.
             mLostBlocks.remove(blockId);
             mBlocks.remove(blockId);
-          } else {
-            mLostBlocks.add(blockId);
           }
         }
       }
