diff --git a/old_product.java b/new_product.java
index f1051c1..3908bcd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,14 @@
-public void attack() {
+public BigInteger attack() {
+        BigInteger result = null;
         long currentTime = System.currentTimeMillis();
         switch (server) {
             case NORMAL:
-                attackNormal();
+                result = attackNormal();
                 break;
             case ORACLE:
-                attackOracle();
+                result = attackOracle();
                 break;
         }
         LOGGER.info("Time needed for the attack: {} seconds", ((System.currentTimeMillis() - currentTime) / 1000));
+        return result;
     }
\ No newline at end of file
