diff --git a/old_product.java b/new_product.java
index b9a87ec..c577535 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ private Iter listDetails(final QueryContext qc) throws QueryException {
     if(exprs.length == 0) return listDBs(qc);
 
     final Data data = checkData(qc);
-    final String path = string(exprs.length == 1 ? Token.EMPTY : checkStr(exprs[1], qc));
+    final String path = string(exprs.length == 1 ? Token.EMPTY : toToken(exprs[1], qc));
     final IntList il = data.resources.docs(path);
     final TokenList tl = data.resources.binaries(path);
 
