diff --git a/old_product.java b/new_product.java
index e9f96e8..777ad68 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static double HypergeometricCdf(int k, int n, int Kp, int Np) {
+public static double hypergeometricCdf(int k, int n, int Kp, int Np) {
         if(k<0 || n<0 || Kp<0 || Np<0) {
             throw new IllegalArgumentException("All the parameters must be positive.");
         }
@@ -9,7 +9,7 @@ public static double HypergeometricCdf(int k, int n, int Kp, int Np) {
         //slow!
         $probabilitySum=0;
         for($i=0;$i<=$k;++$i) {
-            $probabilitySum+=self::Hypergeometric($i,$n,$Kp,$Np);
+            $probabilitySum+=self::hypergeometric($i,$n,$Kp,$Np);
         }
         */
 
