diff --git a/old_product.java b/new_product.java
index 0646da0..0aa872d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -47,9 +47,7 @@
 				}
 			}
 
-		} catch (InvocationTargetException e) {
-			throw new RuntimeException(e);
-		} catch (IllegalAccessException e) {
+		} catch (InvocationTargetException | IllegalAccessException e) {
 			throw new RuntimeException(e);
 		}
 	}
\ No newline at end of file
