diff --git a/old_test.java b/new_test.java
index bb82e1a..893e801 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
 	public void checkSubImage() {
-		List<Point2D_F64> original = new ArrayList<Point2D_F64>();
-		List<Point2D_F64> found = new ArrayList<Point2D_F64>();
+		List<Point2D_F64> original = new ArrayList<>();
+		List<Point2D_F64> found = new ArrayList<>();
 
 		detector.detect(image);
 
