diff --git a/old_product.java b/new_product.java
index 94d7cdb..d516cc0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,3 @@
-@Test
-    public void networkHostCoreLocal() throws IOException {
-        resolveGce("_local_", new NetworkService(Settings.EMPTY).resolveBindHostAddress(NetworkService.DEFAULT_NETWORK_HOST));
+public void networkHostCoreLocal() throws IOException {
+        resolveGce("_local_", new NetworkService(Settings.EMPTY).resolveBindHostAddresses(new String[] { NetworkService.DEFAULT_NETWORK_HOST }));
     }
\ No newline at end of file
