diff --git a/old_test.java b/new_test.java
index 95a7e1c..e655755 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,6 @@
 
         NetworkService networkService = new NetworkService(nodeSettings);
         networkService.addCustomNameResolver(new Ec2NameResolver(nodeSettings));
-        InetAddress[] addresses = networkService.resolveBindHostAddress(null);
-        assertThat(addresses, arrayContaining(networkService.resolveBindHostAddress("_local_")));
+        InetAddress[] addresses = networkService.resolveBindHostAddresses(null);
+        assertThat(addresses, arrayContaining(networkService.resolveBindHostAddresses(new String[] { "_local_" })));
     }
\ No newline at end of file
