diff --git a/old_test.java b/new_test.java
index c4f6956..f9c7297 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
-	public void testConceptualSearch() {
-		Map<String, Object> params = new HashMap<String, Object>();
-		List<String> ids = new ArrayList<String>();
-		ids.add(EXAMPLE_CONCEPT.getId());
-		params.put(ConceptInsights.IDS, ids);
-		params.put(ConceptInsights.LIMIT, 10);
-		params.put(ConceptInsights.CURSOR, 0);
-		QueryConcepts cp = service.conceptualSearch(Corpus.TED_TALKS, params);
-		Assert.assertNotNull(cp);
-	}
\ No newline at end of file
+  public void testConceptualSearch() {
+    final Map<String, Object> params = new HashMap<String, Object>();
+    final List<String> ids = new ArrayList<String>();
+    ids.add(EXAMPLE_CONCEPT.getId());
+    params.put(ConceptInsights.IDS, ids);
+    params.put(ConceptInsights.LIMIT, 10);
+    params.put(ConceptInsights.CURSOR, 0);
+    final QueryConcepts cp = service.conceptualSearch(Corpus.TED_TALKS, params);
+    Assert.assertNotNull(cp);
+  }
\ No newline at end of file
