diff --git a/old_test.java b/new_test.java
index ec43f70..2146a07 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,9 +3,11 @@
     AlluxioURI file = new AlluxioURI("/file");
     URIStatus status = new URIStatus(new FileInfo());
     GetStatusPOptions getStatusOptions = GetStatusPOptions.getDefaultInstance();
-    when(mFileSystemMasterClient.getStatus(file, getStatusOptions)).thenReturn(status);
+    when(mFileSystemMasterClient.getStatus(file, FileSystemOptions.getStatusDefaults(mConf)
+            .toBuilder().mergeFrom(getStatusOptions).build())).thenReturn(status);
     mFileSystem.openFile(file, OpenFilePOptions.getDefaultInstance());
-    verify(mFileSystemMasterClient).getStatus(file, getStatusOptions);
+    verify(mFileSystemMasterClient).getStatus(file,
+        FileSystemOptions.getStatusDefaults(mConf).toBuilder().mergeFrom(getStatusOptions).build());
 
     verifyFilesystemContextAcquiredAndReleased();
   }
\ No newline at end of file
