diff --git a/old_product.java b/new_product.java
index 9ef4f41..4ac1f66 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,7 @@
       public Integer call() throws Exception {
         FileInStream is;
         if (options == null) {
-          is = mFileSystem.openFile(new AlluxioURI(path), OpenFilePOptions.getDefaultInstance());
+          is = mFileSystem.openFile(new AlluxioURI(path));
         } else {
           is = mFileSystem.openFile(new AlluxioURI(path), options);
         }
