diff --git a/old_product.java b/new_product.java
index c1b3cd5..031963a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,7 +11,7 @@ final Item parse(final byte[] json, final boolean xml, final QueryContext qc,
         "Escaping cannot be combined with a fallback function.");
 
     try {
-      opts.set(JsonOptions.FORMAT, xml ? JsonFormat.BASIC : JsonFormat.MAP);
+      opts.set(JsonOptions.FORMAT, xml ? JsonFormat.BASIC : JsonFormat.XQUERY);
       final JsonConverter conv = JsonConverter.get(opts);
       if(!esc && fallback != null) conv.fallback(string -> {
         try {
