diff --git a/old_product.java b/new_product.java
index ceb2508..b5da9f6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -56,6 +56,14 @@
       }
     }
 
+    // Downgrade from local write lock to no write locks
+    synchronized(globalLock) {
+      if(write.isEmpty()) {
+        localWriters--;
+        globalLock.notifyAll();
+      }
+    }
+
     // Write back new locking lists
     writeLocked.put(thread, newWriteObjects);
     if (null != newReadObjects)
