diff --git a/old_test.java b/new_test.java
index 4bbcd7f..fc3bbcd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-	public void voidLastState_shouldVoidStateWithEndDateNullIfStartDatesEqual() {
+	public void voidLastState_shouldVoidStateWithEndDateNullIfStartDatesEqual() throws Exception {
 		//given
 		PatientProgram program = new PatientProgram();
 		ProgramWorkflow workflow = new ProgramWorkflow();
@@ -20,11 +20,14 @@
 		
 		program.getStates().add(state1);
 		program.getStates().add(state2);
+		state1.setPatientProgram(program);
+		state2.setPatientProgram(program);
 		
 		//when
 		program.voidLastState(workflow, new User(), new Date(), "");
 		
 		//then
-		Assert.assertTrue(state1.getVoided());
-		Assert.assertFalse(state2.getVoided());
+		Assert.assertTrue(state1.isVoided());
+		Assert.assertFalse(state2.isVoided());
+		Assert.assertTrue(state2.getEndDate() == null);
 	}
\ No newline at end of file
