diff --git a/old_product.java b/new_product.java
index 9dd5cd8..5daa755 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,15 @@
             final StreamState updateStreamStateRequest, SecurityContext securityContext, AsyncResponse asyncResponse) {
         long traceId = LoggerHelpers.traceEnter(log, "updateStreamState");
 
+        try {
+            authenticate(scopeName + "/" + streamName, READ_UPDATE);
+        } catch (AuthenticationException e) {
+            log.warn("Update stream for {} failed due to authentication failure.", scopeName + "/" + streamName);
+            asyncResponse.resume(Response.status(Status.UNAUTHORIZED).build());
+            LoggerHelpers.traceLeave(log, "Update stream", traceId);
+            return;
+        }
+
         // We only support sealed state now.
         if (updateStreamStateRequest.getStreamState() != StreamState.StreamStateEnum.SEALED) {
             log.warn("Received invalid stream state: {} from client for stream {}/{}",
