diff --git a/old_test.java b/new_test.java
index cf98024..468ed17 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
         when(mockControllerService.sealStream("scope1", "stream1")).thenReturn(CompletableFuture.completedFuture(
                 UpdateStreamStatus.newBuilder().setStatus(UpdateStreamStatus.Status.SUCCESS).build()));
         StreamState streamState = new StreamState().streamState(StreamState.StreamStateEnum.SEALED);
-        Response response = client.target(resourceURI).request().buildPut(Entity.json(streamState)).invoke();
+        Response response = addAuthHeaders(client.target(resourceURI).request()).buildPut(Entity.json(streamState)).invoke();
         assertEquals("Update Stream State response code", 200, response.getStatus());
         response.close();
 
@@ -14,7 +14,7 @@
         when(mockControllerService.sealStream(scope1, stream1)).thenReturn(CompletableFuture.completedFuture(
                 UpdateStreamStatus.newBuilder().setStatus(UpdateStreamStatus.Status.SCOPE_NOT_FOUND).build()));
         streamState = new StreamState().streamState(StreamState.StreamStateEnum.SEALED);
-        response = client.target(resourceURI).request().buildPut(Entity.json(streamState)).invoke();
+        response = addAuthHeaders(client.target(resourceURI).request()).buildPut(Entity.json(streamState)).invoke();
         assertEquals("Update Stream State response code", 404, response.getStatus());
         response.close();
 
@@ -22,7 +22,7 @@
         when(mockControllerService.sealStream(scope1, stream1)).thenReturn(CompletableFuture.completedFuture(
                 UpdateStreamStatus.newBuilder().setStatus(UpdateStreamStatus.Status.STREAM_NOT_FOUND).build()));
         streamState = new StreamState().streamState(StreamState.StreamStateEnum.SEALED);
-        response = client.target(resourceURI).request().buildPut(Entity.json(streamState)).invoke();
+        response = addAuthHeaders(client.target(resourceURI).request()).buildPut(Entity.json(streamState)).invoke();
         assertEquals("Update Stream State response code", 404, response.getStatus());
         response.close();
 
@@ -30,7 +30,7 @@
         when(mockControllerService.sealStream(scope1, stream1)).thenReturn(CompletableFuture.completedFuture(
                 UpdateStreamStatus.newBuilder().setStatus(UpdateStreamStatus.Status.FAILURE).build()));
         streamState = new StreamState().streamState(StreamState.StreamStateEnum.SEALED);
-        response = client.target(resourceURI).request().buildPut(Entity.json(streamState)).invoke();
+        response = addAuthHeaders(client.target(resourceURI).request()).buildPut(Entity.json(streamState)).invoke();
         assertEquals("Update Stream State response code", 500, response.getStatus());
         response.close();
     }
\ No newline at end of file
