diff --git a/old_test.java b/new_test.java
index 91fda74..dee4c2d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,10 @@
 @Test
-  public void applyExtraGroupBitsTest() {
+  public void setGroupBitsTest() {
     Mode mode = new Mode((short) 0640);
-    mode.applyExtraGroupBits(Mode.Bits.READ_EXECUTE);
+    mode.setGroupBits(Mode.Bits.READ_EXECUTE);
     Assert.assertEquals(Mode.Bits.READ_EXECUTE, mode.getGroupBits());
-    mode.applyExtraGroupBits(Mode.Bits.WRITE);
+    mode.setGroupBits(Mode.Bits.WRITE);
+    Assert.assertEquals(Mode.Bits.WRITE, mode.getGroupBits());
+    mode.setGroupBits(Mode.Bits.ALL);
     Assert.assertEquals(Mode.Bits.ALL, mode.getGroupBits());
   }
\ No newline at end of file
