diff --git a/old_product.java b/new_product.java
index c3adf80..87e86b1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -19,9 +19,7 @@ boolean centerOnSquare(QrCode.Alignment pattern, float guessY, float guessX) {
 
 			float r = (float)Math.sqrt(dx*dx + dy*dy);
 
-			if( r == 0 ) {
-				break;
-			} else if( bestMag > r ) {
+			if( bestMag > r ) {
 //				System.out.println("good step at "+i);
 				bestMag = r;
 				bestX = guessX;
@@ -31,8 +29,12 @@ boolean centerOnSquare(QrCode.Alignment pattern, float guessY, float guessX) {
 				step *= 0.75f;
 			}
 
-			guessX = bestX+step*dx/r;
-			guessY = bestY+step*dy/r;
+			if( r > 0 ) {
+				guessX = bestX + step * dx / r;
+				guessY = bestY + step * dy / r;
+			} else {
+				break;
+			}
 		}
 
 		pattern.moduleFound.x = bestX;
