diff --git a/old_product.java b/new_product.java
index 1fd4bc5..8a40f2e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,5 +11,5 @@ public HijrahDate plusMonths(int months) {
             years = MathUtils.safeDecrement(years);
         }
         int newYear = MathUtils.safeAdd(this.yearOfEra, years);
-        return HijrahDate.hijrahDate(this.era, newYear, newMonth + 1, this.dayOfMonth);
+        return HijrahDate.of(this.era, newYear, newMonth + 1, this.dayOfMonth);
     }
\ No newline at end of file
