diff --git a/old_product.java b/new_product.java
index d598c16..c8d0024 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public QNm module(final String qu, final String path) throws QueryException {
-    return new QueryParser(qu, path, this).parseModule(true);
+public QNm parseLibrary(final String qu, final String path) throws QueryException {
+    return new QueryParser(qu, path, this).parseLibrary(true);
   }
\ No newline at end of file
