diff --git a/old_product.java b/new_product.java
index 0eea1bf..2d79d3e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,20 @@
 boolean shutdownBlobStore(PartitionId id) {
-    BlobStore store = stores.get(id);
-    if (store == null || !running) {
-      return false;
-    } else if (!store.isStarted()) {
-      return true;
-    } else {
-      try {
+    rwLock.readLock().lock();
+    boolean succeed = false;
+    try {
+      BlobStore store = stores.get(id);
+      if (store == null || !running) {
+        logger.error("Failed to shut down store because {} is not found or DiskManager is not running", id);
+      } else if (!store.isStarted()) {
+        succeed = true;
+      } else {
         store.shutdown();
-      } catch (Exception e) {
-        logger.error("Exception while shutting down store {} on disk {}", id, disk, e);
-        return false;
+        succeed = true;
       }
-      return true;
+    } catch (Exception e) {
+      logger.error("Exception while shutting down store {} on disk {}", id, disk, e);
+    } finally {
+      rwLock.readLock().unlock();
     }
+    return succeed;
   }
\ No newline at end of file
