diff --git a/old_test.java b/new_test.java
index 6fc569a..dc94f56 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,7 +18,7 @@
     // test shutdown the store which is not started
     ReplicaId replica = replicas.get(replicas.size() - 1);
     PartitionId id = replica.getPartitionId();
-    Store store = storageManager.getStore(id);
+    Store store = storageManager.getStore(id, false);
     store.shutdown();
     assertTrue("Shutdown should succeed on the store which is not started", storageManager.shutdownBlobStore(id));
     // test shutdown the store whose DiskManager is not running
