diff --git a/old_product.java b/new_product.java
index 54fb5c5..566cebb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,10 +6,12 @@ private static int dropCookies(final HttpServletResponse response, final Cookie[
                 continue;
             }
 
-            cookie.setMaxAge(0);
-            cookie.setPath(cookiePath);
+            final Cookie responseCookie = (Cookie) cookie.clone();
+            responseCookie.setMaxAge(0);
+            responseCookie.setPath(cookiePath);
+            responseCookie.setValue("");
 
-            addCookie(cookie, response);
+            addCookie(responseCookie, response);
             count++;
         }
 
