diff --git a/old_product.java b/new_product.java
index 5d76197..1201565 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,9 @@
 public static boolean isSame(Table first, Table second) {
-        if (first.getName().equals(second.getName())) {
+        if (first.getId().equals(second.getId())) {
             final Schema firstParent  = first.getParentOrThrow();
             final Schema secondParent = second.getParentOrThrow();
             return isSame(firstParent, secondParent);
-        } else return false;
+        } else {
+            return false;
+        }
     }
\ No newline at end of file
