diff --git a/old_test.java b/new_test.java
index 016cfd0..6410194 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,7 +16,18 @@
         requestJSONObject.put(User.USER_NAME, "user2 new name");
 
         userMgmtService.updateUser(requestJSONObject);
-        
+
         Assert.assertEquals(getUserQueryService().getUser(id).getJSONObject(
                 User.USER).getString(User.USER_NAME), "user2 new name");
+
+        // Do not update password
+        requestJSONObject.put(Keys.OBJECT_ID, id);
+        requestJSONObject.put(User.USER_NAME, "user2 name");
+        requestJSONObject.put(User.USER_EMAIL, "test2@gmail.com");
+        requestJSONObject.put(User.USER_PASSWORD, "pass2");
+
+        userMgmtService.updateUser(requestJSONObject);
+
+        Assert.assertEquals(getUserQueryService().getUser(id).getJSONObject(
+                User.USER).getString(User.USER_PASSWORD), MD5.hash("pass2"));
     }
\ No newline at end of file
