diff --git a/old_product.java b/new_product.java
index 9699ffe..8e244f3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -24,20 +24,27 @@ public void updateUser(final JSONObject requestJSONObject) throws ServiceExcepti
 
             oldUser.put(User.USER_EMAIL, userNewEmail);
             oldUser.put(User.USER_NAME, userName);
-            oldUser.put(User.USER_PASSWORD, MD5.hash(userPassword));
-            
+
+            final boolean mybeHashed = HASHED_PASSWORD_LENGTH == userPassword.length();
+            final String newHashedPassword = MD5.hash(userPassword);
+            final String oldHashedPassword = oldUser.optString(User.USER_PASSWORD);
+
+            if (!mybeHashed || (!oldHashedPassword.equals(userPassword) && !oldHashedPassword.equals(newHashedPassword))) {
+                oldUser.put(User.USER_PASSWORD, newHashedPassword);
+            }
+
             final String userRole = requestJSONObject.optString(User.USER_ROLE);
 
             if (!Strings.isEmptyOrNull(userRole)) {
                 oldUser.put(User.USER_ROLE, userRole);
             }
-            
+
             final String userURL = requestJSONObject.optString(User.USER_URL);
 
             if (!Strings.isEmptyOrNull(userURL)) {
                 oldUser.put(User.USER_URL, userURL);
             }
-            
+
             userRepository.update(oldUserId, oldUser);
             transaction.commit();
         } catch (final RepositoryException e) {
