diff --git a/old_product.java b/new_product.java
index a2717d3..1ef1e89 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 private Item replace(final QueryContext qc) throws QueryException {
     final Data data = checkData(qc);
     final String path = path(1, qc);
-    final Item item = checkItem(exprs[2], qc);
-    final Options opts = checkOptions(3, Q_OPTIONS, new Options(), qc);
+    final Item item = toItem(exprs[2], qc);
+    final Options opts = toOptions(3, Q_OPTIONS, new Options(), qc);
 
     // remove old documents
     final Resources res = data.resources;
@@ -16,7 +16,7 @@ private Item replace(final QueryContext qc) throws QueryException {
     final IOFile bin = data.inMemory() ? null : data.meta.binary(path);
     if(bin != null) {
       if(bin.exists() || item instanceof Bin) {
-        if(bin.isDir()) throw BXDB_DIR.get(info, path);
+        if(bin.isDir()) throw BXDB_DIR_X.get(info, path);
         updates.add(new DBStore(data, path, item, info), qc);
       } else {
         updates.add(new DBAdd(data, checkInput(item, token(path)), opts, qc, info), qc);
