diff --git a/old_product.java b/new_product.java
index 61dd7a0..f281961 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,24 @@
-private B64Stream readBinary(final QueryContext ctx) throws QueryException {
+private B64 readBinary(final QueryContext ctx) throws QueryException, IOException {
     final File path = checkFile(0, ctx);
+    final long off = expr.length > 1 ? checkItr(expr[1], ctx) : 0;
+    long len = expr.length > 2 ? checkItr(expr[2], ctx) : 0;
+
     if(!path.exists()) FILE_WHICH.thrw(info, path.getAbsolutePath());
     if(path.isDirectory()) FILE_DIR.thrw(info, path.getAbsolutePath());
-    return new B64Stream(new IOFile(path), FILE_IO);
+
+    // read full file
+    if(expr.length == 1) return new B64Stream(new IOFile(path), FILE_IO);
+
+    // read file chunk
+    final DataAccess da = new DataAccess(new IOFile(path));
+    try {
+      final long dlen = da.length();
+      if(expr.length == 2) len = dlen - off;
+      if(off < 0 || off > dlen || len < 0 || off + len > dlen)
+        FILE_BOUNDS.thrw(info, off, off + len);
+      da.cursor(off);
+      return new B64(da.readBytes((int) len));
+    } finally {
+      da.close();
+    }
   }
\ No newline at end of file
