diff --git a/old_test.java b/new_test.java
index db42657..62ea549 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
     public void testMinusYears() {
-        assertEquals(testDate.minusYears(10), MinguoDate.minguoDate(testYear-10, testMonthOfYear, testDayOfMonth));
+        assertEquals(testDate.minusYears(10), MinguoDate.of(testYear-10, testMonthOfYear, testDayOfMonth));
     }
\ No newline at end of file
