diff --git a/old_product.java b/new_product.java
index 6729f8b..7abad38 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@ public HijrahDate minusYears(int years) {
         int newYear = 0;
         try {
             newYear = MathUtils.safeSubtract(this.yearOfEra, years);
-            return HijrahDate.hijrahDate(this.era, newYear, this.monthOfYear, this.dayOfMonth);
+            return HijrahDate.of(this.era, newYear, this.monthOfYear, this.dayOfMonth);
         } catch (ArithmeticException ae) {
             throw new CalendricalException("Year "
                     + (((long) this.yearOfEra) + years)
