diff --git a/old_product.java b/new_product.java
index 1e7175a..a7cf01d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,3 @@
 public static <O> List<O> removeDuplicate(Collection<O> objectCollection){
-        return Validator.isNullOrEmpty(objectCollection) ? Collections.<O> emptyList()
-                        : new ArrayList<O>(new LinkedHashSet<O>(objectCollection));
+        return isNullOrEmpty(objectCollection) ? Collections.<O> emptyList() : toList(new LinkedHashSet<O>(objectCollection));
     }
\ No newline at end of file
