diff --git a/old_test.java b/new_test.java
index 734ce4e..df64bc8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,16 +2,14 @@
 	public void process_simple() {
 		SquaresIntoCrossClusters alg = new SquaresIntoCrossClusters(0.05, -1);
 
-		List<Polygon2D_F64> squares = new ArrayList<>();
+		List<DetectPolygonFromContour.Info> squares = new ArrayList<>();
 		squares.add( createSquare(7,8));
 		squares.add( createSquare(9,8));
 		squares.add( createSquare(8,9));
 		squares.add( createSquare(7,10));
 		squares.add( createSquare(9,10));
 
-		List<BinaryPolygonDetector.Info> squareInfo = createInfo(squares);
-
-		List<List<SquareNode>> clusters = alg.process(squares,squareInfo);
+		List<List<SquareNode>> clusters = alg.process(squares);
 
 		assertEquals(1,clusters.size());
 
