diff --git a/old_product.java b/new_product.java
index d47f252..9e5ac3c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,11 +2,27 @@ public boolean process( T input , GrayU8 binary ) {
 		boundPolygon.vertexes.reset();
 
 		detectorSquare.process(input, binary);
+		List<DetectPolygonFromContour.Info> found = detectorSquare.getPolygonInfo();
 
-		FastQueue<Polygon2D_F64> found = detectorSquare.getFoundPolygons();
-		FastQueue<BinaryPolygonDetector.Info> foundInfo = detectorSquare.getPolygonInfo();
+		for (int i = 0; i < found.size(); i++) {
+			DetectPolygonFromContour.Info f = found.get(i);
+//			System.out.println("  touches "+f.contourTouchesBorder+"  size "+f.polygon.size());
 
-		clusters = s2c.process(found.toList(),foundInfo.toList());
+			// get a better fit onto the contour
+			detectorSquare.refineContour(f);
+
+			// expand the polygons to compensate for the binary image being dilated
+			adjustBeforeOptimize(f.polygon,f.borderCorners);
+
+			// now refine the polygon fit
+//			System.out.println("before "+f.polygon);
+			if( !detectorSquare.refineUsingEdge(f) ) {
+				System.out.println("Refine failed");
+			}
+//			System.out.println("       "+f.polygon.areaSimple());
+		}
+
+		clusters = s2c.process(found);
 
 		c2g.process(clusters);
 		List<SquareGrid> grids = c2g.getGrids().toList();
