diff --git a/old_test.java b/new_test.java
index 8621769..bb4d481 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
-    public void testGetIdentifiers() {
+    public void testGetSoftwareIdentifiers() {
         Dependency instance = new Dependency();
-        Set<Identifier> result = instance.getIdentifiers();
+        Set<Identifier> result = instance.getSoftwareIdentifiers();
 
         assertNotNull(result);
     }
\ No newline at end of file
