diff --git a/old_product.java b/new_product.java
index f859cee..4fed5b3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,9 +8,10 @@
             "/output/**"
         }
     )
-    public String getIndex(@NotNull final HttpServletResponse response, @Nullable final Authentication authentication) {
-        if (authentication != null) {
-            response.addCookie(new Cookie("genie.user", authentication.getName()));
+    public String getIndex(@NotNull final HttpServletRequest request, @NotNull final HttpServletResponse response) {
+        final Principal principal = request.getUserPrincipal();
+        if (principal != null) {
+            response.addCookie(new Cookie("genie.user", principal.getName()));
         } else {
             response.addCookie(new Cookie("genie.user", "user@genie"));
         }
