diff --git a/old_test.java b/new_test.java
index 16a1c0a..7a775f9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,17 +1,11 @@
 @Test
     public void testGetIndex() throws Exception {
         final String indexContent;
-        InputStream is = null;
 
-        try {
-            is = UIController.class.getResourceAsStream("/templates/index.html");
+        try (final InputStream is = UIController.class.getResourceAsStream("/templates/index.html")) {
             Assert.assertNotNull(is);
             Assert.assertTrue(is.available() > 0);
             indexContent = IOUtils.toString(is, StandardCharsets.UTF_8);
-        } finally {
-            if (is != null) {
-                is.close();
-            }
         }
 
         final List<String> validPaths = Arrays.asList(
