diff --git a/old_product.java b/new_product.java
index cca3b7e..faeba04 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ protected void computeHistogramBorder(T image, RectangleRotate_F32 region) {
 		for( int i = 0; i < samplePts.size(); i++ ) {
 			Point2D_F32 p = samplePts.get(i);
 
-			squareToImage(p.x, p.y, region);
+			squareToImageSample(p.x, p.y, region);
 
 			// make sure its inside the image
 			if( !BoofMiscOps.checkInside(image, imageX, imageY)) {
