diff --git a/old_test.java b/new_test.java
index 52da686..1741eb9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,7 +15,7 @@
 		int i = 0;
 		for( int y = 0; y < numSamples; y++ ) {
 			for( int x = 0; x < numSamples; x++ , i++ ) {
-				alg.squareToImage((float)x/(numSamples-1)-0.5f,(float)y/(numSamples-1)-0.5f,rect);
+				alg.squareToImageSample((float)x/(numSamples-1)-0.5f,(float)y/(numSamples-1)-0.5f,rect);
 
 				boolean inside = alg.imageX >= 0 && alg.imageX < 40 && alg.imageY >= 0 && alg.imageY < 50;
 
