diff --git a/old_product.java b/new_product.java
index c0618ee..3dd1225 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-public BytesRef get(int ord) {
+public BytesRef get(int ord, BytesRef ref) {
     assert bytesStart != null : "bytesStart is null - not initialized";
     assert ord < bytesStart.length: "ord exceeeds byteStart len: " + bytesStart.length;
-    return pool.setBytesRef(scratch1, bytesStart[ord]);
+    return pool.setBytesRef(ref, bytesStart[ord]);
   }
\ No newline at end of file
