diff --git a/old_test.java b/new_test.java
index e8a609a..dcafc95 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,7 @@
 @Test
   public void testGet() {
     BytesRef ref = new BytesRef();
+    BytesRef scratch = new BytesRef();
     for (int j = 0; j < 2 * RANDOM_MULTIPLIER; j++) {
       Map<String, Integer> strings = new HashMap<String, Integer>();
       for (int i = 0; i < 797; i++) {
@@ -22,7 +23,7 @@
       }
       for (Entry<String, Integer> entry : strings.entrySet()) {
         ref.copy(entry.getKey());
-        assertEquals(ref, hash.get(entry.getValue().intValue()));
+        assertEquals(ref, hash.get(entry.getValue().intValue(), scratch));
       }
       hash.clear();
       assertEquals(0, hash.size());
