diff --git a/old_product.java b/new_product.java
index c146e98..b9e412b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ private boolean instanceOf(final FuncType ft, final boolean coerce) {
     if(type.instanceOf(ft)) return true;
 
     final SeqType[] at = ft.argTypes;
-    if(at != null && (at.length != 1 || !at[0].instanceOf(SeqType.ITR))) return false;
+    if(at != null && (at.length != 1 || !at[0].instanceOf(SeqType.ITR_O))) return false;
 
     final SeqType ret = ft.declType;
     if(ft instanceof ArrayType) {
