diff --git a/old_product.java b/new_product.java
index 29350fc..a0fdcde 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,7 @@ public Builder<T> scalingMode(ScalingMode config)
 			checkForNull(config, "Scaling mode is null.");
 			updateStatus(Properties.SCALING_MODE, Status.ALREADY_SET);
 			updateStatus(Properties.RESIZER, Status.CANNOT_SET);
+			updateStatus(Properties.RESIZER_FACTORY, Status.CANNOT_SET);
 			scalingMode = config;
 			return this;
 		}
\ No newline at end of file
