diff --git a/old_test.java b/new_test.java
index 7a96d5d..48cb088 100644
--- a/old_test.java
+++ b/new_test.java
@@ -111,10 +111,6 @@
 		field.setAccessible(true);
 		field.set(testClass, mockLock);
 		
-		field = testClass.getClass().getDeclaredField("connectionsObtained");
-		field.setAccessible(true);
-		field.setInt(testClass, 0);
-	
 		reset(mockPartition, mockConnectionHandles, mockConnection);
 		expect(mockPartition.isUnableToCreateMoreTransactions()).andReturn(false).anyTimes();
 		expect(mockPartition.getFreeConnections()).andReturn(mockConnectionHandles).anyTimes();
