diff --git a/old_product.java b/new_product.java
index a41aa14..4f3f76d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,12 +4,8 @@ public Connection getConnection() throws SQLException {
 		ConnectionPartition connectionPartition = this.partitions[partition];
 		if (!connectionPartition.isUnableToCreateMoreTransactions() ){
 			try{
-				// don't bother checking for connections on every single connections.
 				this.connectionsObtainedLock.lock();
-				if (this.connectionsObtained++  % 2  == 0){ 
-					// it's possible for connectionsObtained variable to wrap around but this is not dangerous
 	 				maybeSignalForMoreConnections(connectionPartition);
-				}
 			} finally {
 				this.connectionsObtainedLock.unlock(); 
 			}
