diff --git a/old_product.java b/new_product.java
index fcb3f05..789f90b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,9 @@
 public static void updateLoader(URL[] urls) {
-        DynamicClassLoader loader = (DynamicClassLoader) Thread.currentThread().getContextClassLoader();
-        for (int i = 0; i < urls.length; i++) {
-            loader.addURL(urls[i]);
+        ClassLoader l = Thread.currentThread().getContextClassLoader();
+        if (l instanceof DynamicClassLoader) {
+            DynamicClassLoader loader = (DynamicClassLoader) l;
+            for (int i = 0; i < urls.length; i++) {
+                loader.addURL(urls[i]);
+            }
         }
     }
\ No newline at end of file
