diff --git a/old_test.java b/new_test.java
index a13fa2b..ffd32eb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,6 @@
 @Test
     public void testUpdateLoader() {
         System.out.println("updateLoader");
-        URL[] urls = null;
+        URL[] urls = new URL[0];
         DynamicClassLoader.updateLoader(urls);
-        // TODO review the generated test code and remove the default call to fail.
-        fail("The test case is a prototype.");
     }
\ No newline at end of file
