diff --git a/old_product.java b/new_product.java
index 7b24a22..fc074bf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public DateTime nextExecution(DateTime date) {
+public ZonedDateTime nextExecution(ZonedDateTime date) {
         Validate.notNull(date);
         try {
-            DateTime nextMatch = nextClosestMatch(date);
+            ZonedDateTime nextMatch = nextClosestMatch(date);
             if(nextMatch.equals(date)){
                 nextMatch = nextClosestMatch(date.plusSeconds(1));
             }
