diff --git a/old_test.java b/new_test.java
index 9ab1e86..be0a70c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,6 @@
     public void testNextExecution2016() {
         CronParser parser = new CronParser(CronDefinitionBuilder.instanceDefinitionFor(CronType.UNIX));
         ExecutionTime executionTime = ExecutionTime.forCron(parser.parse("1 0 * * tue"));
-        DateTime date = DateTime.parse("2016-05-24T01:02:50Z");
-        assertEquals(DateTime.parse("2016-05-31T00:01:00Z"), executionTime.nextExecution(date));
+        ZonedDateTime date = ZonedDateTime.parse("2016-05-24T01:02:50Z");
+        assertEquals(ZonedDateTime.parse("2016-05-31T00:01:00Z"), executionTime.nextExecution(date));
     }
\ No newline at end of file
