diff --git a/old_test.java b/new_test.java
index f3179b5..750a221 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void loadMetadataTest() throws Exception {
-    TachyonURI file = new TachyonURI("/file");
+    AlluxioURI file = new AlluxioURI("/file");
     LoadMetadataOptions loadMetadataOptions = LoadMetadataOptions.defaults().setRecursive(true);
     Mockito.doNothing().when(mFileSystemMasterClient).loadMetadata(file, loadMetadataOptions);
     mFileSystem.loadMetadata(file, loadMetadataOptions);
