diff --git a/old_product.java b/new_product.java
index 3cac667..b21cfd2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,8 @@
-public ServiceCall<SessionStatus> getRecognizeStatus(final SpeechSession session) {
-    if (session == null)
-      throw new IllegalArgumentException("Session was not specified");
-
+public ServiceCall<SpeechSessionStatus> getRecognizeStatus(final SpeechSession session) {
+    Validator.notNull(session, "session cannot be null");
+    Validator.notNull(session.getSessionId(), "session.sessionId cannot be null");
+    
     Request request = RequestBuilder.get(String.format(PATH_SESSION_RECOGNIZE, session.getSessionId())).build();
-    return createServiceCall(request, ResponseConverterUtils.getObject(SessionStatus.class));
+    ResponseConverter<SpeechSessionStatus> converter = ResponseConverterUtils.getGenericObject(TYPE_SESSION_STATUS, "session");
+    return createServiceCall(request,converter);
   }
\ No newline at end of file
