diff --git a/old_test.java b/new_test.java
index 98f9592..649ebb9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void testGetRecognizeStatus() {
-    SpeechSession session = service.createSession(SpeechModel.EN_BROADBAND16K).execute();
-    SessionStatus status = service.getRecognizeStatus(session).execute();
+    SpeechSession session = service.createSession(SpeechModel.EN_US_BROADBANDMODEL).execute();
+    SpeechSessionStatus status = service.getRecognizeStatus(session).execute();
     try {
       assertNotNull(status);
       assertNotNull(status.getModel());
