diff --git a/old_product.java b/new_product.java
index 8dbb978..4c94ad4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,15 @@
-public static void checkMapConfig(MapConfig mapConfig, NativeMemoryConfig nativeMemoryConfig,
+public static void checkMapConfig(MapConfig mapConfig,
+                                      NativeMemoryConfig nativeMemoryConfig,
                                       SplitBrainMergePolicyProvider mergePolicyProvider,
                                       HazelcastProperties properties) {
+
         checkNotNativeWhenOpenSource(mapConfig.getInMemoryFormat());
 
-        boolean enterprise = getBuildInfo().isEnterprise();
-        if (enterprise) {
-            checkNativeConfig(mapConfig, nativeMemoryConfig);
+        if (getBuildInfo().isEnterprise()) {
+            checkMapNativeConfig(mapConfig, nativeMemoryConfig);
             checkHotRestartSpecificConfig(mapConfig, properties);
         }
+
+        checkMapEvictionConfig(mapConfig.getEvictionConfig());
         checkMapMergePolicy(mapConfig, mergePolicyProvider);
     }
\ No newline at end of file
