diff --git a/old_test.java b/new_test.java
index cc5e7f3..1c9054b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,5 +5,5 @@
     query(_FILE_CREATE_DIR.args(PATH3));
     query(_FILE_WRITE.args(PATH4, "()"));
     query(_FILE_DELETE.args(PATH1, "true()"));
-    error(_FILE_DELETE.args(PATH1), Err.FILE_NF);
+    error(_FILE_DELETE.args(PATH1), Err.FILE_NOT_FOUND);
   }
\ No newline at end of file
