diff --git a/old_product.java b/new_product.java
index 01294ab..d0e1da3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,9 @@
-private synchronized Item delete(final QueryContext ctx) throws QueryException {
-    final File path = checkFile(0, ctx);
-    if(!path.exists()) throw FILE_NF.get(info, path.getAbsolutePath());
+private synchronized Item delete(final QueryContext ctx) throws QueryException, IOException {
+    final Path path = checkPath(0, ctx);
     if(optionalBool(1, ctx)) {
-      deleteRec(path);
-    } else if(!path.delete()) {
-      throw (path.isDirectory() ? FILE_ID_FULL : FILE_IE_DEL).get(info, path);
+      delete(path);
+    } else {
+      Files.delete(path);
     }
     return null;
   }
\ No newline at end of file
