diff --git a/old_product.java b/new_product.java
index 5d38b78..98f80a5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,9 @@ public static <K, V, C extends Serializable> void initContext(
         LearningEnvironmentBuilder envBuilder,
         int retries,
         int interval,
-        boolean isKeepBinary) {
+        boolean isKeepBinary,
+        DeployingContext deployingContext) {
+
         affinityCallWithRetries(ignite, Arrays.asList(datasetCacheName, upstreamCacheName), part -> {
             Ignite locIgnite = Ignition.localIgnite();
             LearningEnvironment env = envBuilder.buildForWorker(part);
@@ -49,5 +51,5 @@ public static <K, V, C extends Serializable> void initContext(
             datasetCache.put(part, ctx);
 
             return part;
-        }, retries, interval);
+        }, retries, interval, deployingContext);
     }
\ No newline at end of file
