diff --git a/old_product.java b/new_product.java
index 3c7ded6..2ba37b1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public default C build(Stream<UpstreamEntry<K, V>> upstreamData, long upstreamDataSize) {
-        return build(upstreamData.iterator(), upstreamDataSize);
+public default C build(LearningEnvironment env, Stream<UpstreamEntry<K, V>> upstreamData, long upstreamDataSize) {
+        return build(env, upstreamData.iterator(), upstreamDataSize);
     }
\ No newline at end of file
