diff --git a/old_product.java b/new_product.java
index 5b2db3c..d46be19 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,7 +16,18 @@
 
         closed = true;
 
-        cliIo.close();
+        if (affinityAwareness) {
+            for (JdbcThinTcpIo clioIo : ios.values())
+                clioIo.close();
+
+            ios.clear();
+
+            iosArr = null;
+        }
+        else {
+            if (singleIo != null)
+                singleIo.close();
+        }
 
         timer.cancel();
 
