diff --git a/old_test.java b/new_test.java
index 0d3332c..f18081e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     public void testClose() throws Exception {
         final Connection conn;
 
-        try (Connection conn0 = DriverManager.getConnection("jdbc:ignite:thin://127.0.0.1")) {
+        try (Connection conn0 = DriverManager.getConnection(urlWithAffinityAwarenessFlag)) {
             conn = conn0;
 
             assert conn != null;
