diff --git a/old_product.java b/new_product.java
index cd376a4..963b361 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,6 @@
 @Override public double compute(Vector a, Vector b)
         throws CardinalityException {
-        IgniteDoubleFunction<Double> fun = (value -> {
-            if (value == 0) return 0.0;
-            else return 1.0;
-        });
+        IgniteDoubleFunction<Double> fun = (value -> value == 0 ? 0.0 : 1.0);
+
         return MatrixUtil.localCopyOf(a).minus(b).foldMap(Functions.PLUS, fun, 0d);
     }
\ No newline at end of file
