diff --git a/old_test.java b/new_test.java
index 0cbfada..5b453bc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     public void testCompute2() {
         doReturn(42).when(dataset).compute(any(IgniteBiFunction.class), any(IgniteBinaryOperator.class), any());
 
-        Integer res = (Integer) wrapper.compute(mock(IgniteFunction.class), mock(IgniteBinaryOperator.class),
+        Integer res = wrapper.compute(mock(IgniteFunction.class), mock(IgniteBinaryOperator.class),
             null);
 
         assertEquals(42, res.intValue());
