diff --git a/old_product.java b/new_product.java
index c214f71..795fdc0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,11 @@
 public int analyzeAll(final InputStream input, final String location)
 			throws IOException {
-		final ContentTypeDetector detector = new ContentTypeDetector(input);
+		final ContentTypeDetector detector;
+		try {
+			detector = new ContentTypeDetector(input);
+		} catch (IOException e) {
+			throw analyzerError(location, e);
+		}
 		switch (detector.getType()) {
 		case ContentTypeDetector.CLASSFILE:
 			analyzeClass(detector.getInputStream(), location);
