diff --git a/old_test.java b/new_test.java
index e24a930..36ae361 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,15 @@
-@Test(expected = IOException.class)
-	public void testAnalyzeAll_BrokenZip() throws IOException {
-		File file = new File(folder.getRoot(), "broken.zip");
-		OutputStream out = new FileOutputStream(file);
-		ZipOutputStream zip = new ZipOutputStream(out);
-		zip.putNextEntry(new ZipEntry("brokenentry.txt"));
-		out.write(0x23); // Unexpected data here
-		zip.close();
-		analyzer.analyzeAll(file);
+@Test
+	public void testAnalyzeAll_BrokenZip() {
+		final byte[] buffer = new byte[30];
+		buffer[0] = 0x50;
+		buffer[1] = 0x4b;
+		buffer[2] = 0x03;
+		buffer[3] = 0x04;
+		Arrays.fill(buffer, 4, buffer.length, (byte) 0x42);
+		try {
+			analyzer.analyzeAll(new ByteArrayInputStream(buffer), "Test.zip");
+			fail("expected exception");
+		} catch (IOException e) {
+			assertEquals("Error while analyzing Test.zip.", e.getMessage());
+		}
 	}
\ No newline at end of file
