diff --git a/old_product.java b/new_product.java
index e495973..421195d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 private static void query(final String input, final String options, final String expected,
                             final Function function) {
     final String query = options.isEmpty() ? function.args(input) :
-      function.args(input, " {" + options + '}');
+      function.args(input, " map {" + options + '}');
     if(expected.startsWith("...")) {
       contains(query, expected.substring(3));
     } else {
