diff --git a/old_test.java b/new_test.java
index cbeba06..2ed5ab9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,8 +3,8 @@
     contains(_CLIENT_EXECUTE.args(conn(), new ShowUsers()), Text.S_USERINFO[0]);
     query("let $a := " + conn() + ", $b := " + conn() + " return " +
         _CLIENT_QUERY.args("$a", "1") + '+' + _CLIENT_QUERY.args("$b", "2"), "3");
-    query(_CLIENT_QUERY.args(conn(), "\"$a*2\"", " { 'a':1 }"), "2");
+    query(_CLIENT_QUERY.args(conn(), "\"$a*2\"", " map { 'a': 1 }"), "2");
     // query errors
     error(_CLIENT_QUERY.args(conn(), "x"), Err.NOCTX);
-    error(_CLIENT_QUERY.args(conn(), "\"$a\"", " { 'a':(1,2) }"), Err.BXCL_ITEM);
+    error(_CLIENT_QUERY.args(conn(), "\"$a\"", " map { 'a': (1,2) }"), Err.BXCL_ITEM);
   }
\ No newline at end of file
