diff --git a/old_test.java b/new_test.java
index 1f24e9e..dd1f1fb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,10 @@
 @Test
   public void testGetClassifier() throws InterruptedException {
     server.enqueue(jsonResponse(classifier));
-    final Classifier response = service.getClassifier(classifierId).execute();
+    GetClassifierOptions getOptions = new GetClassifierOptions.Builder()
+        .classifierId(classifierId)
+        .build();
+    final Classifier response = service.getClassifier(getOptions).execute();
     final RecordedRequest request = server.takeRequest();
 
     assertEquals(CLASSIFIERS_PATH + "/" + classifierId, request.getPath());
