diff --git a/old_product.java b/new_product.java
index 1d19157..d8b13f7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,8 @@
 public Builder ipv6(@Nullable byte[] ipv6) {
-      if (ipv6 == null) return this;
+      if (ipv6 == null) {
+        this.ipv6 = null;
+        return this;
+      }
       checkArgument(ipv6.length == 16, "ipv6 addresses are 16 bytes: " + ipv6.length);
       for (int i = 0; i < 10; i++) { // Embedded IPv4 addresses start with unset 80 bits
         if (ipv6[i] != 0) {
