diff --git a/old_product.java b/new_product.java
index fd442f4..53937e1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public LocalTime with(DateTimeField field, long newValue) {
-        if (field instanceof LocalDateTimeField) {
-            LocalDateTimeField f = (LocalDateTimeField) field;
+        if (field instanceof ChronoField) {
+            ChronoField f = (ChronoField) field;
             f.checkValidValue(newValue);
             switch (f) {
                 case NANO_OF_SECOND: return withNano((int) newValue);
