diff --git a/old_product.java b/new_product.java
index acf700e..448e8b1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,5 @@ public JapaneseDate minusDays(long days) {
         if (days == 0) {
             return this;
         }
-        return JapaneseDate.japaneseDate(date.minusDays(days));
+        return JapaneseDate.from(date.minusDays(days));
     }
\ No newline at end of file
