diff --git a/old_test.java b/new_test.java
index bbdb43b..450715a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
     public void testMinusDays() {
-        assertEquals(testDate.minusDays(2), JapaneseDate.japaneseDate(testYear, testMonthOfYear, testDayOfMonth-2));
+        assertEquals(testDate.minusDays(2), JapaneseDate.of(testYear, testMonthOfYear, testDayOfMonth-2));
     }
\ No newline at end of file
