diff --git a/old_test.java b/new_test.java
index 34f54bc..6471655 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,7 @@
 		expect(mockConfig.getIdleConnectionTestPeriod()).andReturn(10000L).anyTimes();
 		expect(mockConfig.getUsername()).andReturn(CommonTestUtils.username).anyTimes();
 		expect(mockConfig.getPassword()).andReturn(CommonTestUtils.password).anyTimes();
-		expect(mockConfig.getJdbcUrl()).andReturn("somethingbad").anyTimes();
+		expect(mockConfig.getJdbcUrl()).andReturn("invalid").anyTimes();
 		expect(mockConfig.getReleaseHelperThreads()).andReturn(0).anyTimes();
 		expect(mockConfig.getConnectionHook()).andReturn(hookClass).anyTimes();
 		replay(mockConfig);
