diff --git a/old_test.java b/new_test.java
index 87c9bdd..7320faa 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,6 +14,6 @@
     query(fun + "('db', 'etc/test/input.xml', 'test3.xml', 'test')");
     query("count(collection('db/test/test3.xml')/html) eq 1", "true");
 
-    query(fun + "('db', 'etc/test/dir', '', 'test/dir')");
-    query("count(collection('db/test/dir')) gt 0", "true");
+    query(fun + "('db', '" + FLDR + "', '', 'test/dir')");
+    query("count(collection('db/test/dir'))", NFLDR);
   }
\ No newline at end of file
