diff --git a/old_test.java b/new_test.java
index 8062812..977ce31 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testExpandLabel() {
+    public void testExpandLabel() throws CryptoException {
         HKDFAlgorithm hkdfAlgorithm = HKDFAlgorithm.TLS_HKDF_SHA256;
         byte[] prk = ArrayConverter
                 .hexStringToByteArray("E056D47C7DB9C04BBECE6AC9525163DE72B7D25B6B0899366F8FA741A5C01709");
