diff --git a/old_product.java b/new_product.java
index f5f1d04..4739971 100644
--- a/old_product.java
+++ b/new_product.java
@@ -86,7 +86,8 @@ public static PiPipelineModel parse(URL p4InfoUrl) throws P4InfoParserException
                             tableMsg.getSize(),
                             tableCounterMapBuilder.build(),
                             tableMeterMapBuilder.build(),
-                            tableMsg.getWithEntryTimeout(),
+                            !tableMsg.getIdleTimeoutBehavior()
+                                    .equals(Table.IdleTimeoutBehavior.NO_TIMEOUT),
                             tableFieldMapBuilder.build(),
                             tableActionMapBuilder.build(),
                             actionMap.get(tableMsg.getConstDefaultActionId()),
