diff --git a/old_product.java b/new_product.java
index 3601cdd..8123f13 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,4 @@
-public static double getLongitudeFromX01(final double pX01, boolean wrapEnabled) {
-		return MinLongitude + (MaxLongitude - MinLongitude) * (wrapEnabled ? Clip(pX01, 0, 1) : pX01);
+public double getLongitudeFromX01(final double pX01, boolean wrapEnabled) {
+        final double longitude = getLongitudeFromX01(wrapEnabled ? Clip(pX01, 0, 1) : pX01);
+        return wrapEnabled ? Clip(longitude, getMinLongitude(), getMaxLongitude()) : longitude;
 	}
\ No newline at end of file
