diff --git a/old_test.java b/new_test.java
index fb989d1..7d869c5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,10 +2,10 @@
     public void testGetLongitudeFromX01() {
         final int iterations = 10;
         for (int i = 0 ; i <= iterations ; i ++) {
-            final double longitude = TileSystem.MinLongitude + i * (TileSystem.MaxLongitude - TileSystem.MinLongitude) / iterations;
-            checkLongitude(longitude, TileSystem.getLongitudeFromX01(((double)i) / iterations, true));
+            final double longitude = tileSystem.getMinLongitude() + i * (tileSystem.getMaxLongitude() - tileSystem.getMinLongitude()) / iterations;
+            checkLongitude(longitude, tileSystem.getLongitudeFromX01(((double)i) / iterations, true));
         }
-        checkLongitude(TileSystem.MinLongitude, TileSystem.getLongitudeFromX01(0, true));
-        checkLongitude(0, TileSystem.getLongitudeFromX01(0.5, true));
-        checkLongitude(TileSystem.MaxLongitude, TileSystem.getLongitudeFromX01(1, true));
+        checkLongitude(tileSystem.getMinLongitude(), tileSystem.getLongitudeFromX01(0, true));
+        checkLongitude(0, tileSystem.getLongitudeFromX01(0.5, true));
+        checkLongitude(tileSystem.getMaxLongitude(), tileSystem.getLongitudeFromX01(1, true));
     }
\ No newline at end of file
