diff --git a/old_test.java b/new_test.java
index 41a9eb6..79d9823 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,9 +8,9 @@
       long id = oplog.getOplogId();
       oplog.close();
 
-      StatisticsFactory factory = cache.getDistributedSystem();
+      StatisticsFactory factory = cache.getDistributedSystem().getStatisticsFactory();
       Oplog newOplog =
-          new Oplog(id, dr.getOplogSet(), new DirectoryHolder(factory, dirs[0], 1000, 0));
+          new Oplog(id, dr.getOplogSet(), new DirectoryHolder(dirs[0], 1000, 0));
       dr.getOplogSet().setChild(newOplog);
       closeDown();
     }
@@ -29,9 +29,9 @@
       Oplog oplog = dr.testHook_getChild();
       long id = oplog.getOplogId();
       oplog.close();
-      StatisticsFactory factory = cache.getDistributedSystem();
+      StatisticsFactory factory = cache.getDistributedSystem().getStatisticsFactory();
       Oplog newOplog =
-          new Oplog(id, dr.getOplogSet(), new DirectoryHolder(factory, dirs[0], 1000, 2));
+          new Oplog(id, dr.getOplogSet(), new DirectoryHolder(dirs[0], 1000, 2));
       dr.setChild(newOplog);
       closeDown();
     }
