diff --git a/old_test.java b/new_test.java
index 5b9e820..dc56a5f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,14 +16,14 @@
     parse("\"\\b\\f\\t\\r\\n\"", "\"\uFFFD\uFFFD\\t\\r\\n\"", false);
     parse("\"\\u0000\\u001F\"", "\"\uFFFD\uFFFD\"", false);
 
-    unescape("\"\\b\\f\\t\\r\\n\"", "\"\\\\b\\\\f\\\\t\\\\r\\\\n\"");
+    escape("\"\\b\\f\\t\\r\\n\"", "\"\\\\b\\\\f\\\\t\\\\r\\\\n\"");
     // Unicode in JSON notation
-    unescape("\"\\uD853\\uDF5C\"", "\"\\\\uD853\\\\uDF5C\"");
-    unescape("\"\\uD853asdf\"", "\"\\\\uD853asdf\"");
-    unescape("\"\\uD853\"", "\"\\\\uD853\"");
+    escape("\"\\uD853\\uDF5C\"", "\"\\\\uD853\\\\uDF5C\"");
+    escape("\"\\uD853asdf\"", "\"\\\\uD853asdf\"");
+    escape("\"\\uD853\"", "\"\\\\uD853\"");
     // Unicode in Java notation
-    unescape("\"\u00E4\\t\"", "\"\u00E4\\\\t\"");
-    unescape("\"\u00E4\\u00E4\\t\"", "\"\u00E4\\\\u00E4\\\\t\"");
+    escape("\"\u00E4\\t\"", "\"\u00E4\\\\t\"");
+    escape("\"\u00E4\\u00E4\\t\"", "\"\u00E4\\\\u00E4\\\\t\"");
 
     error("\"\\u0A", false);
     error("\"\\uXX0A\"", false);
