diff --git a/old_product.java b/new_product.java
index be9a08c..1ab35db 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,7 +12,7 @@ private byte[] string() throws QueryIOException {
       }
 
       if(ch == '\\') {
-        if(!unescape) {
+        if(escape) {
           if(high != 0) {
             tb.add(high);
             high = 0;
@@ -28,32 +28,23 @@ private byte[] string() throws QueryIOException {
             ch = n;
             break;
           case 'b':
-            ch = unescape ? '\b' : 'b';
+            ch = escape ? 'b' : '\b';
             break;
           case 'f':
-            ch = unescape ? '\f' : 'f';
+            ch = escape ? 'f' : '\f';
             break;
           case 't':
-            ch = unescape ? '\t' : 't';
+            ch = escape ? 't' : '\t';
             break;
           case 'r':
-            ch = unescape ? '\r' : 'r';
+            ch = escape ? 'r' : '\r';
             break;
           case 'n':
-            ch = unescape ? '\n' : 'n';
+            ch = escape ? 'n' : '\n';
             break;
           case 'u':
             if(pos + 4 >= length) throw eof(", expected four-digit hex value");
-            if(unescape) {
-              ch = 0;
-              for(int i = 0; i < 4; i++) {
-                final char x = consume();
-                if(x >= '0' && x <= '9')      ch = 16 * ch + x      - '0';
-                else if(x >= 'a' && x <= 'f') ch = 16 * ch + x + 10 - 'a';
-                else if(x >= 'A' && x <= 'F') ch = 16 * ch + x + 10 - 'A';
-                else throw error("Illegal hexadecimal digit: '%'", x);
-              }
-            } else {
+            if(escape) {
               tb.add('u');
               for(int i = 0; i < 4; i++) {
                 final char x = consume();
@@ -63,6 +54,14 @@ private byte[] string() throws QueryIOException {
               }
               continue;
             }
+            ch = 0;
+            for(int i = 0; i < 4; i++) {
+              final char x = consume();
+              if(x >= '0' && x <= '9')      ch = 16 * ch + x      - '0';
+              else if(x >= 'a' && x <= 'f') ch = 16 * ch + x + 10 - 'a';
+              else if(x >= 'A' && x <= 'F') ch = 16 * ch + x + 10 - 'A';
+              else throw error("Illegal hexadecimal digit: '%'", x);
+            }
             break;
           default:
             throw error("Unknown character escape: '\\%'", n);
