diff --git a/old_product.java b/new_product.java
index 8e1ca4d..d11ac7a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public static void easy( ImageFloat32 image ) {
+public static void easy( GrayF32 image ) {
 		// create the detector and descriptors
-		DetectDescribePoint<ImageFloat32,BrightFeature> surf = FactoryDetectDescribe.
-				surfStable(new ConfigFastHessian(0, 2, 200, 2, 9, 4, 4), null, null,ImageFloat32.class);
+		DetectDescribePoint<GrayF32,BrightFeature> surf = FactoryDetectDescribe.
+				surfStable(new ConfigFastHessian(0, 2, 200, 2, 9, 4, 4), null, null,GrayF32.class);
 
 		 // specify the image to process
 		surf.detect(image);
