diff --git a/old_test.java b/new_test.java
index b7edcbd..a512cc9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,6 @@
 		clientConfig.setString(SecurityOptions.SSL_TRUSTSTORE, "src/test/resources/local127.truststore");
 		clientConfig.setString(SecurityOptions.SSL_TRUSTSTORE_PASSWORD, "password");
 
-		SSLContext clientContext = SSLUtils.createSSLClientContext(clientConfig);
+		SSLUtils.SSLContext clientContext = SSLUtils.createSSLClientContext(clientConfig);
 		Assert.assertNotNull(clientContext);
 	}
\ No newline at end of file
