diff --git a/old_test.java b/new_test.java
index e9cc229..8c14db3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-	public final void testGoToUrl() throws CrawljaxException {
+	public final void testGoToUrl() throws CrawljaxException, MalformedURLException {
 		// TODO Stefan; bug in WebDriver iff you specify bla:// will end up in NullPointer.
-		browser.goToUrl("http://non.exsisting.domain");
+		browser.goToUrl(new URL("http://non.exsisting.domain"));
 	}
\ No newline at end of file
