diff --git a/old_test.java b/new_test.java
index d147e1d..b612860 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
 		Assert.assertTrue(!order.isSigned());
 		Assert.assertTrue(order.getDateSigned() == null);
 		
-		Context.getOrderService().signOrder(order, provider);
+		Context.getOrderService().signOrder(order, provider, null);
 		
 		order = Context.getOrderService().getOrder(10);
 		Assert.assertTrue(order.isSigned());
