diff --git a/old_test.java b/new_test.java
index 4f6b7cc..14446dd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @SuppressWarnings("unchecked")
     @Test(expected = GenieServerException.class)
-    public void testSubmitJob() throws GenieException {
+    public void testSubmitJob() throws GenieException, IOException {
 
         final Set<CommandStatus> enumStatuses = EnumSet.noneOf(CommandStatus.class);
         enumStatuses.add(CommandStatus.ACTIVE);
@@ -77,6 +77,7 @@
         Mockito
             .when(this.clusterService.getCommandsForCluster(CLUSTER_ID, enumStatuses))
             .thenReturn(commandList);
+        Mockito.doThrow(new IOException("something bad")).when(this.task2).executeTask(Mockito.anyMap());
 
         final ArgumentCaptor<String> jobId1 = ArgumentCaptor.forClass(String.class);
         final ArgumentCaptor<String> clusterId = ArgumentCaptor.forClass(String.class);
