diff --git a/old_test.java b/new_test.java
index dff56e0..09ff43d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,11 +12,11 @@
     assertEquals("Did not receive an AsyncRequestResponseHandler instance",
         AsyncRequestResponseHandler.class.getCanonicalName(), restResponseHandler.getClass().getCanonicalName());
 
-    BlobStorageService blobStorageService =
-        new MockBlobStorageService(verifiableProperties, restResponseHandler, router);
+    RestRequestService restRequestService =
+        new MockRestRequestService(verifiableProperties, restResponseHandler, router);
     // Get request handler.
     AsyncRequestResponseHandlerFactory requestHandlerFactory =
-        new AsyncRequestResponseHandlerFactory(1, METRIC_REGISTRY, blobStorageService);
+        new AsyncRequestResponseHandlerFactory(1, METRIC_REGISTRY, restRequestService);
     RestRequestHandler restRequestHandler = requestHandlerFactory.getRestRequestHandler();
     assertNotNull("No RestRequestHandler returned", restRequestHandler);
     assertEquals("Did not receive an AsyncRequestResponseHandler instance",
