diff --git a/old_product.java b/new_product.java
index e568e5f..f64648f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,10 +2,10 @@ private static AsyncRequestResponseHandler getAsyncRequestResponseHandler(int re
     RequestResponseHandlerMetrics metrics = new RequestResponseHandlerMetrics(new MetricRegistry());
     AsyncRequestResponseHandler handler = new AsyncRequestResponseHandler(metrics);
     if (requestWorkers > 0) {
-      if (blobStorageService == null) {
-        blobStorageService = new MockBlobStorageService(verifiableProperties, handler, router);
+      if (restRequestService == null) {
+        restRequestService = new MockRestRequestService(verifiableProperties, handler, router);
       }
-      handler.setupRequestHandling(requestWorkers, blobStorageService);
+      handler.setupRequestHandling(requestWorkers, restRequestService);
     }
     return handler;
   }
\ No newline at end of file
