diff --git a/old_test.java b/new_test.java
index e5368e7..6aa3bdd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
   public void pinTest() throws Exception {
     SetStateOptions setPinned =
-        new SetStateOptions.Builder(new TachyonConf()).setPinned(true).build();
+        new SetStateOptions.Builder().setPinned(true).build();
     SetStateOptions setUnpinned =
-        new SetStateOptions.Builder(new TachyonConf()).setPinned(false).build();
+        new SetStateOptions.Builder().setPinned(false).build();
     mTfs.mkdir(new TachyonURI("/myFolder"));
     TachyonFile directory = mTfs.open(new TachyonURI("/myFolder"));
     mTfs.setState(directory, setPinned);
