diff --git a/old_product.java b/new_product.java
index 25c7de6..fa432eb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public void pin(TachyonURI path) throws IOException {
     try {
       TachyonFile fd = mTfs.open(path);
-      SetStateOptions options = new SetStateOptions.Builder(mTachyonConf).setPinned(true).build();
+      SetStateOptions options = new SetStateOptions.Builder().setPinned(true).build();
       mTfs.setState(fd, options);
       System.out.println("File '" + path + "' was successfully pinned.");
     } catch (TachyonException e) {
