diff --git a/old_product.java b/new_product.java
index d84b2df..f78c9c7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,4 @@
-@TestInfo(
-      level = TestLevel.COMPLETE,
-      purpose = "",
-      targets = {
-        @TestTarget(
-          methodName = "getInstance",
-          methodArgs = {String.class, String.class}
-        )
-    })
-    public void GetInstance02(int mode) throws NoSuchAlgorithmException,
+public void GetInstance02(int mode) throws NoSuchAlgorithmException,
             NoSuchProviderException, IllegalArgumentException,
             InvalidAlgorithmParameterException {
         try {
