diff --git a/old_product.java b/new_product.java
index 38c1288..2293710 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static CatalogResolver catalogResolver(CatalogFeatures features, String... path) {
-        Catalog catalog = catalog(features, path);
+public static CatalogResolver catalogResolver(CatalogFeatures features, String... paths) {
+        Catalog catalog = catalog(features, paths);
         return new CatalogResolverImpl(catalog);
     }
\ No newline at end of file
