diff --git a/old_test.java b/new_test.java
index 4c73596..e97d66e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -30,8 +30,8 @@
 				.builder()
 				.quantityAssignedToRefundCandidate(Quantity.of(SEVEN, uom)) // the assignable candidate with 10 is still completely assigned
 				.quantityOfRefundCandidate(Quantity.of(SEVEN, uom)) // used to have 14 and 4 of those belonged to the assignable candidate with 7
-				.moneyAssignedToRefundCandidate(Money.of(TWO, currentId))
-				.moneyOfRefundCandidate(Money.of(HUNDRED.add(TWO), currentId))
+				.moneyAssignedToRefundCandidate(Money.of(ONE, currentId))
+				.moneyOfRefundCandidate(Money.of(HUNDRED.add(ONE), currentId))
 				.build();
 
 		assertMoneyAndQuantityAssignments(reloadedAssignableCandidateWithSeven, expectation);
