diff --git a/old_test.java b/new_test.java
index 2f94c6b..8dac759 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,6 +8,7 @@
 		order.setOrderer(Context.getProviderService().getProvider(1));
 		order.setDosingType(DrugOrder.DosingType.FREE_TEXT);
 		order.setInstructions("Instructions");
+		order.setDosingInstructions("Test Instruction");
 		order.setPatient(patient);
 		encounter.setPatient(patient);
 		order.setEncounter(encounter);
