diff --git a/old_product.java b/new_product.java
index 445f037..704a4b6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -24,9 +24,6 @@ public void validate(Object obj, Errors errors) {
 					ValidationUtils.rejectIfEmpty(errors, "doseUnits", "DrugOrder.error.doseUnitsIsNullForDosingTypeSimple");
 					ValidationUtils.rejectIfEmpty(errors, "route", "DrugOrder.error.routeIsNullForDosingTypeSimple");
 					ValidationUtils.rejectIfEmpty(errors, "frequency", "DrugOrder.error.frequencyIsNullForDosingTypeSimple");
-				} else if (order.getDosingType().equals(DrugOrder.DosingType.FREE_TEXT)) {
-					ValidationUtils.rejectIfEmpty(errors, "instructions",
-					    "DrugOrder.error.instructionIsNullForDosingTypeFreeText");
 				} else {
 					ValidationUtils.rejectIfEmpty(errors, "dosingInstructions",
 					    "DrugOrder.error.dosingInstructionsIsNullForDosingTypeOther");
