diff --git a/old_test.java b/new_test.java
index 630fd1d..adc7026 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
    public void testListExtensions() throws Exception {
       for (String zoneId : novaContext.getApi().getConfiguredZones()) {
          ExtensionApi api = novaContext.getApi().getExtensionApiForZone(zoneId);
-         Set<Extension> response = api.listExtensions();
+         Set<? extends Extension> response = api.listExtensions();
          assert null != response;
          assertTrue(response.size() >= 0);
          for (Extension extension : response) {
