diff --git a/old_test.java b/new_test.java
index b409850..7aab6a0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,8 +9,10 @@
         List<ProfileSchemaDto> defaultProfileSchemas = client.getProfileSchemas(application.getId());
         profileSchemas.addAll(defaultProfileSchemas);
 
+        CTLSchemaInfoDto ctlSchema = this.createCTLSchema(this.ctlRandomFieldType(), CTL_DEFAULT_NAMESPACE, 1, CTLSchemaScopeDto.TENANT, null, null, null);
+
         for (int i=0;i<10;i++) {
-            ProfileSchemaDto profileSchema = createProfileSchema(application.getId());
+            ProfileSchemaDto profileSchema = createProfileSchema(application.getId(), ctlSchema.getId());
             profileSchemas.add(profileSchema);
         }
         
@@ -18,14 +20,14 @@
         
         SchemaVersions schemaVersions = client.getSchemaVersionsByApplicationId(application.getId());
         
-        List<SchemaDto> storedProfileSchemas = schemaVersions.getProfileSchemaVersions();
+        List<VersionDto> storedProfileSchemas = schemaVersions.getProfileSchemaVersions();
 
         Collections.sort(storedProfileSchemas, new IdComparator());
         
         Assert.assertEquals(profileSchemas.size(), storedProfileSchemas.size());
         for (int i=0;i<profileSchemas.size();i++) {
             ProfileSchemaDto profileSchema = profileSchemas.get(i);
-            SchemaDto storedProfileSchema = storedProfileSchemas.get(i);
+            VersionDto storedProfileSchema = storedProfileSchemas.get(i);
             assertSchemasEquals(profileSchema, storedProfileSchema);
         }
     }
\ No newline at end of file
