diff --git a/old_product.java b/new_product.java
index 1cd1f94..6a2a27c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 @Override
-    public List<SchemaDto> getProfileSchemaVersionsByApplicationId(String applicationId) throws ControlServiceException {
+    public List<VersionDto> getProfileSchemaVersionsByApplicationId(String applicationId) throws ControlServiceException {
         return profileService.findProfileSchemaVersionsByAppId(applicationId);
     }
\ No newline at end of file
