diff --git a/old_test.java b/new_test.java
index 5709f02..658f4f5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testSizeGe() {
-        assertTrue(instance.sizeGe("id", 3).getQueryCriterions()
-                .contains(new SizeGeCriterion("id", 3)));
+    	assertEquals(new SizeGeCriterion("id", 3), instance.sizeGe("id", 3).getQueryCriterion());
     }
\ No newline at end of file
