diff --git a/old_product.java b/new_product.java
index 6220003..a97da5f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,11 @@
 @Override
     public int indexOf(KType e1)
     {
-        for (int i = 0; i < elementsCount; i++)
-            if (Intrinsics.equalsKType(e1, buffer[i]))
+        for (int i = 0; i < elementsCount; i++) {
+            if (Intrinsics.equals(this, e1, buffer[i])) {
                 return i;
+            }
+        }
 
         return -1;
     }
\ No newline at end of file
