diff --git a/old_test.java b/new_test.java
index f637a94..0470967 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
     public void test_firstDayOfNextMonth_serialization() throws IOException, ClassNotFoundException {
-        DateTimeAdjuster firstDayOfMonth = DateTimeAdjusters.firstDayOfNextMonth();
+        WithAdjuster firstDayOfMonth = DateTimeAdjusters.firstDayOfNextMonth();
         assertTrue(firstDayOfMonth instanceof Serializable);
         
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
