diff --git a/old_product.java b/new_product.java
index 3e42ac5..84c9273 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,18 +1,4 @@
 @Override
 	public <T> Collection<T> getById(final Collection<? extends Serializable> ids, final Class<T> clazz) {
-
-		if (CollectionUtils.isEmpty(ids)) {
-			return Collections.emptyList();
-		}
-
-		return execute(new SolrCallback<Collection<T>>() {
-			@Override
-			public Collection<T> doInSolr(SolrClient solrClient) throws SolrServerException, IOException {
-
-				QueryResponse response = new SolrRealtimeGetRequest(ids).process(solrClient,
-						getSolrCoreOrBeanCollection(clazz));
-				return convertSolrDocumentListToBeans(response.getResults(), clazz);
-			}
-
-		});
+		return getById(getSolrCoreOrBeanCollection(clazz), ids, clazz);
 	}
\ No newline at end of file
