diff --git a/old_test.java b/new_test.java
index eb4aa15..746deb0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,16 +1,11 @@
 @Test
+	@SuppressWarnings({ "rawtypes", "unchecked" })
 	public void testGetById() throws SolrServerException, IOException {
 
-		ArgumentCaptor<SolrRequest> captor = ArgumentCaptor.forClass(SolrRequest.class);
-		QueryResponse responseMock = Mockito.mock(QueryResponse.class);
-		SolrDocumentList resultList = new SolrDocumentList();
-		Mockito.when(responseMock.getResults()).thenReturn(resultList);
-		Mockito.when(solrClientMock.request(captor.capture(), Matchers.anyString())).thenReturn(new NamedList<Object>());
+		ArgumentCaptor<List> captor = ArgumentCaptor.forClass(List.class);
 
-		DocumentWithIndexAnnotations result = solrTemplate.getById("myId", DocumentWithIndexAnnotations.class);
+		solrTemplate.getById("myId", DocumentWithIndexAnnotations.class);
 
-		Mockito.verify(solrClientMock, Mockito.times(1)).request(captor.capture(), Matchers.anyString());
-		Assert.assertNull(result);
-		Assert.assertEquals("myId", captor.getValue().getParams().get("ids"));
-		Assert.assertEquals("/get", captor.getValue().getPath());
+		Mockito.verify(solrClientMock, Mockito.times(1)).getById(eq("core1"), captor.capture());
+		Assert.assertThat((List<String>) captor.getValue(), IsCollectionContaining.hasItems("myId"));
 	}
\ No newline at end of file
