diff --git a/old_test.java b/new_test.java
index 6c256d0..9de263a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test(expected=OsmosisRuntimeException.class)
 	public final void testProcess7() {
 		Way testWay;
-		testWay = new Way(3456, new Date(), new OsmUser("OsmosisTest", 12), 0);
+		testWay = new Way(3456, 0, new Date(), new OsmUser(12, "OsmosisTest"));
 		testWay.addWayNode(new WayNode(1234));
 		testWay.addWayNode(new WayNode(1235));
 		testWay.addTag(new Tag("test_key1", "test_value1"));
