diff --git a/old_product.java b/new_product.java
index afc7241..45851a1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,9 +6,9 @@ public void process(NodeContainer nodeContainer) {
 				new NodeContainer(
 					new Node(
 						inputEntity.getId(),
+						inputEntity.getVersion(),
 						timestamp,
 						inputEntity.getUser(),
-						inputEntity.getVersion(),
 						inputEntity.getLatitude(),
 						inputEntity.getLongitude()
 					)
