diff --git a/old_test.java b/new_test.java
index e5d7d84..bc7f05a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,9 +2,9 @@
     public void testTrainAndValidate() {
         logger.info("testTrainAndValidate");
         
-        Configuration conf = Configuration.getConfiguration();
+        Configuration configuration = Configuration.getConfiguration();
         
-        Dataframe[] data = Datasets.carsNumeric(conf);
+        Dataframe[] data = Datasets.carsNumeric(configuration);
         Dataframe trainingData = data[0];
         
         Dataframe validationData = data[1];
@@ -28,13 +28,13 @@
         //feature selection configuration
         trainingParameters.setFeatureSelectorTrainingParameters(null);
 
-        Modeler instance = MLBuilder.create(trainingParameters, conf);
+        Modeler instance = MLBuilder.create(trainingParameters, configuration);
         instance.fit(trainingData);
         instance.save(storageName);
 
         instance.close();
 
-        instance = MLBuilder.load(Modeler.class, storageName, conf);
+        instance = MLBuilder.load(Modeler.class, storageName, configuration);
 
         instance.predict(trainingData);
 
@@ -48,7 +48,7 @@
         //instance = null;
 
 
-        instance = MLBuilder.load(Modeler.class, storageName, conf);
+        instance = MLBuilder.load(Modeler.class, storageName, configuration);
         
         instance.predict(validationData);
         
