diff --git a/old_product.java b/new_product.java
index 48730e6..4dc1993 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
             ML.AbstractTrainingParameters modelerTrainingParameters,
             FS.AbstractTrainingParameters featureSelectorTrainingParameters,
             double expectedF1score) {
-        Configuration conf = Configuration.getConfiguration();
+        Configuration configuration = Configuration.getConfiguration();
         
         
         String storageName = this.getClass().getSimpleName();
@@ -34,7 +34,7 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
         exParams.setExaminationWindowLength(6);
         trainingParameters.setTextExtractorParameters(exParams);
 
-        TextClassifier instance = MLBuilder.create(trainingParameters, conf);
+        TextClassifier instance = MLBuilder.create(trainingParameters, configuration);
         instance.fit(dataset);
         instance.save(storageName);
 
@@ -47,7 +47,7 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector> void
         
         
         
-        instance = MLBuilder.load(TextClassifier.class, storageName, conf);
+        instance = MLBuilder.load(TextClassifier.class, storageName, configuration);
         Dataframe validationData;
         try {
             validationData = instance.predict(this.getClass().getClassLoader().getResource("datasets/sentimentAnalysis.unlabelled.txt").toURI());
