diff --git a/old_product.java b/new_product.java
index 346056f..5ede5e1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -29,7 +29,7 @@ private static boolean isAssignable(final Type type, final Class<?> toClass) {
         if (type instanceof TypeVariable<?>) {
             // if any of the bounds are assignable to the class, then the
             // type is assignable to the class.
-            for (Type bound : ((TypeVariable<?>) type).getBounds()) {
+            for (final Type bound : ((TypeVariable<?>) type).getBounds()) {
                 if (isAssignable(bound, toClass)) {
                     return true;
                 }
