diff --git a/old_test.java b/new_test.java
index c7b1112..085fb56 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
-	public void getDrugDurationUnits_shouldReturnAListIfGPIsSet() throws Exception {
-		
-		assertThat(orderService.getDrugDurationUnits(), containsInAnyOrder(hasId(28)));
+	public void getDurationUnits_shouldReturnAListIfGPIsSet() throws Exception {
+		assertThat(orderService.getDurationUnits(), containsInAnyOrder(hasId(28)));
 	}
\ No newline at end of file
