diff --git a/old_test.java b/new_test.java
index 0fbc113..47b492c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,7 +18,8 @@
     List<CreateExample> intentExamples = new ArrayList<CreateExample>();
     intentExamples.add(new CreateExample.Builder().text(intentExample).build());
     workspaceIntents.add(
-        new CreateIntent.Builder().intent(intentName).description(intentDescription).examples(intentExamples).build());
+            new CreateIntent.Builder().intent(intentName).description(intentDescription).examples(intentExamples)
+                    .build());
 
     // entities
     List<CreateEntity> workspaceEntities = new ArrayList<CreateEntity>();
@@ -30,7 +31,8 @@
     List<CreateValue> entityValues = new ArrayList<CreateValue>();
     entityValues.add(new CreateValue.Builder().value(entityValue).addSynonym(entityValueSynonym).build());
     workspaceEntities
-        .add(new CreateEntity.Builder().entity(entityName).description(entityDescription).values(entityValues).build());
+            .add(new CreateEntity.Builder().entity(entityName).description(entityDescription).values(entityValues)
+                    .build());
 
     // counterexamples
     List<CreateCounterexample> workspaceCounterExamples = new ArrayList<CreateCounterexample>();
@@ -38,8 +40,8 @@
     workspaceCounterExamples.add(new CreateCounterexample.Builder().text(counterExampleText).build());
 
     CreateWorkspaceOptions createOptions = new CreateWorkspaceOptions.Builder().name(workspaceName)
-        .description(workspaceDescription).language(workspaceLanguage).metadata(workspaceMetadata)
-        .intents(workspaceIntents).entities(workspaceEntities).counterexamples(workspaceCounterExamples).build();
+            .description(workspaceDescription).language(workspaceLanguage).metadata(workspaceMetadata)
+            .intents(workspaceIntents).entities(workspaceEntities).counterexamples(workspaceCounterExamples).build();
 
     String workspaceId = null;
     try {
@@ -78,27 +80,27 @@
       // intents
       assertNotNull(exResponse.getIntents());
       assertTrue(exResponse.getIntents().size() == 1);
-      assertNotNull(exResponse.getIntents().get(0).getIntent());
-      assertEquals(exResponse.getIntents().get(0).getIntent(), intentName);
+      assertNotNull(exResponse.getIntents().get(0).getIntentName());
+      assertEquals(exResponse.getIntents().get(0).getIntentName(), intentName);
       assertNotNull(exResponse.getIntents().get(0).getDescription());
       assertEquals(exResponse.getIntents().get(0).getDescription(), intentDescription);
       assertNotNull(exResponse.getIntents().get(0).getExamples());
       assertTrue(exResponse.getIntents().get(0).getExamples().size() == 1);
       assertNotNull(exResponse.getIntents().get(0).getExamples().get(0));
-      assertNotNull(exResponse.getIntents().get(0).getExamples().get(0).getText());
-      assertEquals(exResponse.getIntents().get(0).getExamples().get(0).getText(), intentExample);
+      assertNotNull(exResponse.getIntents().get(0).getExamples().get(0).getExampleText());
+      assertEquals(exResponse.getIntents().get(0).getExamples().get(0).getExampleText(), intentExample);
 
       // entities
       assertNotNull(exResponse.getEntities());
       assertTrue(exResponse.getEntities().size() == 1);
-      assertNotNull(exResponse.getEntities().get(0).getEntity());
-      assertEquals(exResponse.getEntities().get(0).getEntity(), entityName);
+      assertNotNull(exResponse.getEntities().get(0).getEntityName());
+      assertEquals(exResponse.getEntities().get(0).getEntityName(), entityName);
       assertNotNull(exResponse.getEntities().get(0).getDescription());
       assertEquals(exResponse.getEntities().get(0).getDescription(), entityDescription);
       assertNotNull(exResponse.getEntities().get(0).getValues());
       assertTrue(exResponse.getEntities().get(0).getValues().size() == 1);
-      assertNotNull(exResponse.getEntities().get(0).getValues().get(0).getValue());
-      assertEquals(exResponse.getEntities().get(0).getValues().get(0).getValue(), entityValue);
+      assertNotNull(exResponse.getEntities().get(0).getValues().get(0).getValueText());
+      assertEquals(exResponse.getEntities().get(0).getValues().get(0).getValueText(), entityValue);
       assertNotNull(exResponse.getEntities().get(0).getValues().get(0).getSynonyms());
       assertTrue(exResponse.getEntities().get(0).getValues().get(0).getSynonyms().size() == 1);
       assertEquals(exResponse.getEntities().get(0).getValues().get(0).getSynonyms().get(0), entityValueSynonym);
