diff --git a/old_product.java b/new_product.java
index f0546c9..b4e57ed 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,9 +15,12 @@ public ServiceCall<Workspace> createWorkspace(CreateWorkspaceOptions createWorks
       if (createWorkspaceOptions.name() != null) {
         contentJson.addProperty("name", createWorkspaceOptions.name());
       }
+      if (createWorkspaceOptions.learningOptOut() != null) {
+        contentJson.addProperty("learning_opt_out", createWorkspaceOptions.learningOptOut());
+      }
       if (createWorkspaceOptions.counterexamples() != null) {
-        contentJson.add("counterexamples", GsonSingleton.getGson().toJsonTree(createWorkspaceOptions
-            .counterexamples()));
+        contentJson.add("counterexamples", GsonSingleton.getGson().toJsonTree(
+                createWorkspaceOptions.counterexamples()));
       }
       if (createWorkspaceOptions.description() != null) {
         contentJson.addProperty("description", createWorkspaceOptions.description());
