diff --git a/old_test.java b/new_test.java
index abba67c..0221654 100644
--- a/old_test.java
+++ b/new_test.java
@@ -25,7 +25,7 @@
     OperationHandle opHandle = client.executeStatement(sessHandle, queryString, opConf);
     assertNotNull(opHandle);
 
-    OperationStatus opStatus = client.getOperationStatus(opHandle);
+    OperationStatus opStatus = client.getOperationStatus(opHandle, false);
     assertNotNull(opStatus);
 
     OperationState state = opStatus.getState();
