diff --git a/old_product.java b/new_product.java
index 13ec081..b22f0ec 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,11 @@
 @Override
   public OperationHandle executeStatement(SessionHandle sessionHandle, String statement,
       Map<String, String> confOverlay) throws HiveSQLException {
-    OperationHandle opHandle =
-        sessionManager.getSession(sessionHandle).executeStatement(statement, confOverlay);
+    HiveSession session = sessionManager.getSession(sessionHandle);
+    // need to reset the monitor, as operation handle is not available down stream, Ideally the
+    // monitor should be associated with the operation handle.
+    session.getSessionState().updateProgressMonitor(null);
+    OperationHandle opHandle = session.executeStatement(statement, confOverlay);
     LOG.debug(sessionHandle + ": executeStatement()");
     return opHandle;
   }
\ No newline at end of file
