diff --git a/old_product.java b/new_product.java
index 2a33c99..5b0c4aa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,7 +9,8 @@
         if (!this.applicationRepo.exists(id)) {
             throw new GenieNotFoundException("No application information entered. Unable to update.");
         }
-        if (!id.equals(updateApp.getId())) {
+        final Optional<String> updateId = updateApp.getId();
+        if (updateId.isPresent() && !id.equals(updateId.get())) {
             throw new GenieBadRequestException("Application id inconsistent with id passed in.");
         }
 
