diff --git a/old_test.java b/new_test.java
index 675fd43..dcb1b51 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testGetChangeLog() throws ConnectionException {
 
-        System.out.println( "getChangeLog" );
+        LOG.info( "getChangeLog" );
 
         // create an entity and make a series of changes to it so that versions get created
         CollectionScope context = new CollectionScopeImpl(
@@ -50,7 +50,7 @@
         List<ChangeLogEntry> result = instance.getChangeLog( versions, e2.getVersion() );
 
         for (ChangeLogEntry cle : result) {
-            System.out.println( cle.toString() );
+            LOG.info( cle.toString() );
         }
         assertEquals(16, result.size());
     }
\ No newline at end of file
