diff --git a/old_test.java b/new_test.java
index e8b6452..b6a8199 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testOnCreateOrUpdate() throws GenieException {
+    public void testOnCreateOrUpdateApplication() throws GenieException {
         this.a = new Application(NAME, USER, ApplicationStatus.ACTIVE, VERSION);
-        this.a.onCreateOrUpdate();
+        this.a.onCreateOrUpdateApplication();
     }
\ No newline at end of file
