diff --git a/old_product.java b/new_product.java
index ab1d781..fee5c8b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 @PrePersist
     @PreUpdate
-    protected void onCreateOrUpdate() throws GenieException {
-        validate(this.getName(), this.getUser(), this.status);
-     // Add the id to the tags
+    protected void onCreateOrUpdateApplication() throws GenieException {
+        this.validate(this.status);
+        // Add the id to the tags
         if (this.tags == null) {
             this.tags = new HashSet<String>();
         }
