diff --git a/old_product.java b/new_product.java
index 8bf57d4..57e4285 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
-    public byte[] decrypt(byte[] data) throws CryptoException {
+    public byte[] decrypt(byte[] data) {
         try {
             byte[] plaintext;
             ConnectionEndType localConEndType = context.getConnection().getLocalConnectionEndType();
@@ -25,6 +25,7 @@
             return plaintext;
         } catch (BadPaddingException | IllegalBlockSizeException | InvalidAlgorithmParameterException
                 | InvalidKeyException | UnsupportedOperationException ex) {
-            throw new CryptoException(ex);
+            LOGGER.warn("Could not decrypt Data with the provided parameters. Returning undecrypted data.", ex);
+            return data;
         }
     }
\ No newline at end of file
