diff --git a/old_test.java b/new_test.java
index b91c356..31b8b00 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,10 @@
 @Test
   public void baseUri() throws Exception {
-    final String find =
-      "for $x in collection('" + NAME + '/' + DIR + "xmark.xml') " +
-      "return base-uri($x)";
+    final String find = "base-uri(collection('" + NAME + '/' + DIR + "xmark.xml'))";
     final QueryProcessor qp = new QueryProcessor(find, context);
-    assertEquals(NAME + '/' + FILES[1], qp.iter().next().toJava());
-    qp.close();
+    try {
+      assertEquals(NAME + '/' + FILES[1], qp.execute().toString());
+    } finally {
+      qp.close();
+    }
   }
\ No newline at end of file
