diff --git a/old_product.java b/new_product.java
index acb046e..56115f8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -64,8 +64,8 @@
 
             tlsContext.setServerDHParameters(publicKeyParameters);
 
-            if (tlsContext.getConfig().getProtocolVersion() == ProtocolVersion.DTLS12
-                    || tlsContext.getConfig().getProtocolVersion() == ProtocolVersion.TLS12) {
+            if (tlsContext.getSelectedProtocolVersion() == ProtocolVersion.DTLS12
+                    || tlsContext.getSelectedProtocolVersion() == ProtocolVersion.TLS12) {
                 currentPointer = nextPointer;
                 nextPointer++;
                 HashAlgorithm ha = HashAlgorithm.getHashAlgorithm(message[currentPointer]);
