diff --git a/old_test.java b/new_test.java
index dde6650..20cf6c3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testParseMessageAction() {
         AlertHandler handler = new AlertHandler(new TlsContext());
-        handler.setProtocolMessage(new AlertMessage());
+        handler.setProtocolMessage(new AlertMessage(new TlsConfig()));
         byte[] message = { 3, 3 };
         int pointer = handler.parseMessageAction(message, 0);
         assertEquals(2, pointer);
