diff --git a/old_test.java b/new_test.java
index 3d1d469..3bc06ca 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups = "samples", timeOut = TIMEOUT)
     public void createDatabase_toBlocking() {
-        Flux<ResourceResponse<Database>> createDatabaseObservable = client.createDatabase(getDatabaseDefinition(),
+        Mono<ResourceResponse<Database>> createDatabaseObservable = client.createDatabase(getDatabaseDefinition(),
                                                                                                      null);
 
         // toBlocking() converts to a blocking observable.
