diff --git a/old_product.java b/new_product.java
index 1bc2772..6a2862b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,12 +5,16 @@ public static Period parse(CharSequence text) {
             int negate = ("-".equals(matcher.group(1)) ? -1 : 1);
             String yearMatch = matcher.group(2);
             String monthMatch = matcher.group(3);
-            String dayMatch = matcher.group(4);
-            if (yearMatch != null || monthMatch != null || dayMatch != null) {
+            String weekMatch = matcher.group(4);
+            String dayMatch = matcher.group(5);
+            if (yearMatch != null || monthMatch != null || weekMatch != null || dayMatch != null) {
                 try {
-                    return create(parseNumber(text, yearMatch, negate),
-                            parseNumber(text, monthMatch, negate),
-                            parseNumber(text, dayMatch, negate));
+                    int years = parseNumber(text, yearMatch, negate);
+                    int months = parseNumber(text, monthMatch, negate);
+                    int weeks = parseNumber(text, weekMatch, negate);
+                    int days = parseNumber(text, dayMatch, negate);
+                    days = Jdk8Methods.safeAdd(days, Jdk8Methods.safeMultiply(weeks, 7));
+                    return create(years, months, days);
                 } catch (NumberFormatException ex) {
                     throw (DateTimeParseException) new DateTimeParseException("Text cannot be parsed to a Period", text, 0).initCause(ex);
                 }
