diff --git a/old_test.java b/new_test.java
index 74a860a..5c63694 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,14 +6,13 @@
         TemporalField womField = week.weekOfMonth();
 
         for (int i = 1; i <= 15; i++) {
-            // Test that with dayOfWeek and Week of month it computes the date
             DateTimeFormatter f = new DateTimeFormatterBuilder()
-                    .appendValue(YEAR).appendLiteral('-')
-                    .appendValue(MONTH_OF_YEAR).appendLiteral('-')
-                    .appendValue(womField).appendLiteral('-')
+                    .appendValue(YEAR).appendLiteral(':')
+                    .appendValue(MONTH_OF_YEAR).appendLiteral(':')
+                    .appendValue(womField).appendLiteral(':')
                     .appendValue(dowField).toFormatter();
-            String str = date.getYear() + "-" + date.getMonthValue() + "-" +
-                    date.get(womField) + "-" + date.get(dowField);
+            String str = date.getYear() + ":" + date.getMonthValue() + ":" +
+                    date.get(womField) + ":" + date.get(dowField);
             LocalDate parsed = LocalDate.parse(str, f);
             assertEquals(parsed, date, " :: " + str + " " + i);
 
