diff --git a/old_test.java b/new_test.java
index 65e6f48..4258bdf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
     }
 
     MaxAggregationFunction maxAggregationFunction = new MaxAggregationFunction();
-    ResultHolder resultHolder = ResultHolderFactory.getResultHolder(maxAggregationFunction, 1 /* maxNumResults */);
+    AggregationResultHolder resultHolder = ResultHolderFactory.getAggregationResultHolder(maxAggregationFunction);
 
     maxAggregationFunction.aggregate(NUM_VALUES_TO_AGGREGATE, resultHolder, valuesToAggregate);
     double actual = resultHolder.getDoubleResult();
