diff --git a/old_product.java b/new_product.java
index 96bb080..bfbb5d7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,7 +9,9 @@
 
       for (String column : aggregationFunctionContext.getAggregationColumns()) {
         double[] valuesToAggregate = _columnToValueArrayMap.get(column);
-        aggregationFunctionContext.aggregate(length, _resultHolderArray[i], valuesToAggregate);
+
+        AggregationFunction function = aggregationFunctionContext.getAggregationFunction();
+        function.aggregate(length, _resultHolderArray[i], valuesToAggregate);
       }
     }
   }
\ No newline at end of file
