diff --git a/old_test.java b/new_test.java
index 20f476f..12dec1e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -22,13 +22,13 @@
       CharSequence methodSource = state.getSourceForNode((JCMethodDecl) methodTree);
       if (methodSource != null) {
         String methodString = "@Test\n" + methodSource.toString().replaceFirst(" static ", " ");
-        SuggestedFix fix = new SuggestedFix()
+        Fix fix = new SuggestedFix()
             .addImport(JUNIT4_TEST_ANNOTATION)
             .replace(methodTree, methodString);
         return describeMatch(methodTree, fix);
       }
     }
-    SuggestedFix fix = new SuggestedFix()
+    Fix fix = new SuggestedFix()
         .addImport(JUNIT4_TEST_ANNOTATION)
         .prefixWith(methodTree, "@Test\n");
     return describeMatch(methodTree, fix);
