diff --git a/old_product.java b/new_product.java
index cd55cc3..a562d1e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -25,7 +25,7 @@
     // We don't have start position for a method symbol, so we replace everything between result
     // type and body.
     JCMethodDecl decl = (JCMethodDecl) methodTree;
-    SuggestedFix fix = new SuggestedFix().replace(
+    Fix fix = new SuggestedFix().replace(
         decl.restype.getStartPosition() + 4, decl.body.getStartPosition(), " " + fixedName + "() ");
     return describeMatch(methodTree, fix);
   }
\ No newline at end of file
