diff --git a/old_test.java b/new_test.java
index 1895705..2a6203d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,8 +3,8 @@
     {
         System.out.println("train");
         
-        RegressionDataSet trainSet =  FixedProblems.getSimpleRegression1(2000, new XORWOW());
-        RegressionDataSet testSet =  FixedProblems.getSimpleRegression1(200, new XORWOW());
+        RegressionDataSet trainSet =  FixedProblems.getSimpleRegression1(2000, RandomUtil.getRandom());
+        RegressionDataSet testSet =  FixedProblems.getSimpleRegression1(200, RandomUtil.getRandom());
         
         ExecutorService threadPool = Executors.newFixedThreadPool(SystemInfo.LogicalCores);
         
