diff --git a/old_product.java b/new_product.java
index 1c0ca2c..5c82afa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ private void train(Vec[] columns, double[] y)
         final double beta = loss.getDeriv2Max();
         double[] z = new double[y.length];///stores w.dot(x)
         w = new DenseVector(columns.length);
-        Random rand = new XORWOW();
+        Random rand = RandomUtil.getRandom();
         for (int iter = 0; iter < iterations; iter++)
         {
             final int j = rand.nextInt(columns.length);
