diff --git a/old_product.java b/new_product.java
index 2de743b..50db1fc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 private Array remove(final QueryContext qc) throws QueryException {
-    final Array array = array(0, qc);
-    final int p = checkPos(array, checkItr(exprs[1], qc));
+    final Array array = toArray(exprs[0], qc);
+    final int p = checkPos(array, toLong(exprs[1], qc));
     final int as = array.arraySize();
     if(p == 0) return Array.get(array, 1, as - 1);
     if(p + 1 == as) return Array.get(array, 0, as - 1);
