diff --git a/old_test.java b/new_test.java
index afb3c3a..0308c51 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
     array(_ARRAY_REMOVE.args(" array { 1 to 5 }", " 3"), "[1, 2, 4, 5]");
     array(_ARRAY_REMOVE.args(" array { 1 to 5 }", " 5"), "[1, 2, 3, 4]");
 
-    error(_ARRAY_REMOVE.args(" []", " 0"), Err.ARRAYBOUNDS);
-    error(_ARRAY_REMOVE.args(" [1]", " 0"), Err.ARRAYBOUNDS);
-    error(_ARRAY_REMOVE.args(" [1]", " 2"), Err.ARRAYBOUNDS);
+    error(_ARRAY_REMOVE.args(" []", " 0"), Err.ARRAYBOUNDS_X_X);
+    error(_ARRAY_REMOVE.args(" [1]", " 0"), Err.ARRAYBOUNDS_X_X);
+    error(_ARRAY_REMOVE.args(" [1]", " 2"), Err.ARRAYBOUNDS_X_X);
   }
\ No newline at end of file
