diff --git a/old_product.java b/new_product.java
index f23104a..86e6548 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,22 +10,23 @@ private void addCriticalityToVulnerability(String parentName, Vulnerability vuln
                     LOGGER.debug("Unable to look up vulnerability {}", vulnerability.getName());
                 }
             }
-            if (v != null) {
-                score = v.getCvssScore();
-                vulnerability.setCvssAccessComplexity(v.getCvssAccessComplexity());
-                vulnerability.setCvssAccessVector(v.getCvssAccessVector());
-                vulnerability.setCvssAuthentication(v.getCvssAuthentication());
-                vulnerability.setCvssAvailabilityImpact(v.getCvssAvailabilityImpact());
-                vulnerability.setCvssConfidentialityImpact(v.getCvssConfidentialityImpact());
-                vulnerability.setCvssIntegrityImpact(v.getCvssIntegrityImpact());
-            } else if ("High".equalsIgnoreCase(criticality)) {
-                score = 8.5f;
-            } else if ("Medium".equalsIgnoreCase(criticality)) {
-                score = 5.5f;
-            } else if ("Low".equalsIgnoreCase(criticality)) {
-                score = 2.0f;
+            if (v != null && (v.getCvssV2() != null || v.getCvssV3() != null)) {
+                if (v.getCvssV2() != null) {
+                    vulnerability.setCvssV2(v.getCvssV2());
+                }
+                if (v.getCvssV3() != null) {
+                    vulnerability.setCvssV3(v.getCvssV3());
+                }
+            } else {
+                if ("High".equalsIgnoreCase(criticality)) {
+                    score = 8.5f;
+                } else if ("Medium".equalsIgnoreCase(criticality)) {
+                    score = 5.5f;
+                } else if ("Low".equalsIgnoreCase(criticality)) {
+                    score = 2.0f;
+                }
+                vulnerability.setCvssV2(new CvssV2(score, "-", "-", "-", "-", "-", "-", criticality));
             }
-            vulnerability.setCvssScore(score);
         }
         LOGGER.debug("bundle-audit ({}): {}", parentName, nextLine);
     }
\ No newline at end of file
