diff --git a/old_test.java b/new_test.java
index 4a1c4b7..64be968 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
             analyzer.analyze(result, engine);
             Dependency dependency = engine.getDependencies()[0];
             Vulnerability vulnerability = dependency.getVulnerabilities(true).iterator().next();
-            assertEquals(vulnerability.getCvssScore(), 5.0f, 0.0);
+            assertEquals(5.0f, vulnerability.getCvssV2().getScore(), 0.0);
 
         } catch (InitializationException | DatabaseException | AnalysisException | UpdateException e) {
             LOGGER.warn("Exception setting up RubyBundleAuditAnalyzer. Make sure Ruby gem bundle-audit is installed. You may also need to set property \"analyzer.bundle.audit.path\".");
