diff --git a/old_test.java b/new_test.java
index fafb726..ae09668 100644
--- a/old_test.java
+++ b/new_test.java
@@ -36,7 +36,8 @@
                 expectedMaxSizeResult, maxSizeResult, Object::equals);
 
         //3. Complete a few writes, then mark a few as in progress.
-        writes.get(0).complete(new TestLogAddress(0));
+        writes.get(0).setEntryId(0);
+        writes.get(0).complete();
         writes.get(1).beginAttempt();
         val result1 = q.getWritesToExecute(Long.MAX_VALUE);
 
@@ -53,7 +54,8 @@
         //5. LedgerChange.
         int beginIndex = ledgerChangeIndex - MAX_PARALLELISM / 2;
         for (int i = 0; i < beginIndex; i++) {
-            writes.get(i).complete(new TestLogAddress(i));
+            writes.get(i).setEntryId(i);
+            writes.get(i).complete();
         }
 
         q.removeFinishedWrites();
@@ -61,7 +63,8 @@
         AssertExtensions.assertListEquals("Unexpected writes fetched when ledger changed.",
                 writes.subList(beginIndex, ledgerChangeIndex), result3, Object::equals);
 
-        result3.forEach(w -> w.complete(new TestLogAddress(0)));
+        result3.forEach(w -> w.setEntryId(0));
+        result3.forEach(Write::complete);
         q.removeFinishedWrites();
         val result4 = q.getWritesToExecute(Long.MAX_VALUE);
         AssertExtensions.assertListEquals("Unexpected writes fetched from the end, after ledger changed.",
