diff --git a/old_test.java b/new_test.java
index 85fea2f..03509d6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,5 +3,5 @@
         LocalDateTime ldt = LocalDateTime.of(2008, 6, 30, 23, 30, 59, 0);
         ZonedDateTime base = ZonedDateTime.of(ldt, ZONE_0100);
         ZonedDateTime test = base.withZoneSameLocal(ZONE_0200);
-        assertEquals(test.toLocalTime(), base.toLocalTime());
+        assertEquals(test.getTime(), base.getTime());
     }
\ No newline at end of file
