diff --git a/old_product.java b/new_product.java
index 45ab45d..751eb00 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,5 @@ public ChronoZonedDateTime<C> withZoneSameLocal(ZoneId zone, ZoneResolver resolv
         Objects.requireNonNull(zone, "ZoneId must not be null");
         Objects.requireNonNull(resolver, "ZoneResolver must not be null");
         return zone == this.zone ? this :
-            resolve(dateTime.toDateTime(), zone, dateTime, resolver);
+            resolve(dateTime.getDateTime(), zone, dateTime, resolver);
     }
\ No newline at end of file
