diff --git a/old_test.java b/new_test.java
index 44bbbd3..46071eb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
     public void cacheRecievedMessageHash() throws Exception {
-        messageCacheService.cacheRecievedMessageHash(blockMessage.getHash().getDigestHex());
+        messageCacheService.cacheRecievedMessageHash(blockMessage.getHash());
         assertNotNull(messageCacheService);
 
         Field cacheMapRecievedField = messageCacheService.getClass().getDeclaredField("cacheMapRecieved");
         cacheMapRecievedField.setAccessible(true);
-        CacheMap<String, Integer>  cacheMap = (CacheMap<String, Integer>)cacheMapRecievedField.get(messageCacheService);
-        assertTrue(cacheMap.get(blockMessage.getHash().getDigestHex()) == 1);
+        CacheMap<NulsDigestData, Integer> cacheMap = (CacheMap<NulsDigestData, Integer>) cacheMapRecievedField.get(messageCacheService);
+        assertTrue(cacheMap.get(blockMessage.getHash()) == 1);
     }
\ No newline at end of file
