diff --git a/old_product.java b/new_product.java
index 4ee2c17..82702b0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-String execute(Collector collector, String counterName, String sessionId, String threadId)
-			throws IOException {
+String execute(Collector collector, String counterName, String sessionId, String threadId,
+			String jobId) throws IOException {
 		// CHECKSTYLE:ON
 		String messageForReport;
 		switch (this) {
@@ -54,12 +54,12 @@ String execute(Collector collector, String counterName, String sessionId, String
 			messageForReport = killThread(threadId);
 			break;
 		case PAUSE_JOB:
-			pauseAllJobs();
-			messageForReport = I18N.getString("all_jobs_paused");
+			assert jobId != null;
+			messageForReport = pauseJob(jobId);
 			break;
 		case RESUME_JOB:
-			resumeAllJobs();
-			messageForReport = I18N.getString("all_jobs_resumed");
+			assert jobId != null;
+			messageForReport = resumeJob(jobId);
 			break;
 		default:
 			throw new IllegalStateException(toString());
