diff --git a/old_product.java b/new_product.java
index 564238d..cc9a62f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 protected String resolveVariable(final String variableName, final StrBuilder buf, final int startPos, final int endPos) {
-        StrLookup<?> resolver = getVariableResolver();
+        final StrLookup<?> resolver = getVariableResolver();
         if (resolver == null) {
             return null;
         }
