diff --git a/old_test.java b/new_test.java
index cca6412..b2d50ec 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
     public void testResolveVariable() {
         final StrBuilder builder = new StrBuilder("Hi ${name}!");
-        Map<String, String> map = new HashMap<String, String>();
+        final Map<String, String> map = new HashMap<String, String>();
         map.put("name", "commons");
-        StrSubstitutor sub = new StrSubstitutor(map) {
+        final StrSubstitutor sub = new StrSubstitutor(map) {
             @Override
             protected String resolveVariable(final String variableName, final StrBuilder buf, final int startPos, final int endPos) {
                 assertEquals("name", variableName);
