diff --git a/old_test.java b/new_test.java
index 40ac56f..4878f41 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test
   public void getPathTest() throws Exception {
-    Inode root = mTree.getInodeById(0);
+    Inode<?> root = mTree.getInodeById(0);
     // test root path
     Assert.assertEquals(new AlluxioURI("/"), mTree.getPath(root));
 
     // test one level
     InodeTree.CreatePathResult createResult = mTree.createPath(TEST_URI, sDirectoryOptions);
-    List<Inode> created = createResult.getCreated();
+    List<Inode<?>> created = createResult.getCreated();
     Assert.assertEquals(new AlluxioURI("/test"), mTree.getPath(created.get(created.size() - 1)));
 
     // test nesting
