diff --git a/old_test.java b/new_test.java
index 520f12a..c6cf684 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
-@Test(expected = IllegalArgumentException.class)
-	@Verifies(value = "should fail if the drug object is null", method = "validate(Object,Errors)")
-	public void validate_shouldFailIfTheDrugObjectIsNull() {
-		Errors errors = new BindException(new Drug(), "drug");
-		new DrugValidator().validate(null, errors);
+@Test
+	public void validate_shouldFailIfTheDrugObjectIsNull() throws Exception {
+		expectedException.expect(IllegalArgumentException.class);
+		expectedException.expectMessage("The parameter obj should not be null and must be of type" + Drug.class);
+		new DrugValidator().validate(null, new BindException(new Drug(), "drug"));
 	}
\ No newline at end of file
