diff --git a/old_product.java b/new_product.java
index 52e7b77..6b8ad3a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,10 +4,9 @@
 			throw new IllegalArgumentException("The parameter obj should not be null and must be of type" + Drug.class);
 		} else {
 			Drug drug = (Drug) obj;
-			
 			Set<DrugReferenceMap> drugReferenceMaps = drug.getDrugReferenceMaps();
+			Set<String> mappedTermUuids = new HashSet<String>();
 			int index = 0;
-			
 			for (DrugReferenceMap referenceMap : drugReferenceMaps) {
 				Drug mappedDrug = referenceMap.getDrug();
 				ConceptReferenceTerm referenceTerm = referenceMap.getConceptReferenceTerm();
@@ -28,6 +27,7 @@
 						errors.popNestedPath();
 					}
 				}
+				
 				if (mapType == null) {
 					errors.rejectValue("drugReferenceMaps[" + index + "].conceptMapType",
 					    "Drug.drugReferenceMap.conceptMapType");
@@ -40,6 +40,18 @@
 						errors.popNestedPath();
 					}
 				}
+				
+				//don't proceed to the next map
+				if (errors.hasErrors()) {
+					return;
+				}
+				
+				//if we already have a mapping to this term, reject it this map
+				if (!mappedTermUuids.add(referenceMap.getConceptReferenceTerm().getUuid())) {
+					errors.rejectValue("drugReferenceMaps[" + index + "].conceptReferenceTerm",
+					    "Drug.drugReferenceMap.termAlreadyMapped",
+					    "Cannot map a drug multiple times to the same reference term");
+				}
 				index++;
 			}
 		}
