diff --git a/old_test.java b/new_test.java
index 28dbd95..63bea15 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,21 @@
 @Test
   public void testShouldFireNow() {
-    fail("Not yet implemented");
+    try {
+      Date blockOutStartDate = new Date(System.currentTimeMillis());
+      SimpleBlockoutTrigger blockOutTrigger = new SimpleBlockoutTrigger("blockOut1", blockOutStartDate, null, //$NON-NLS-1$
+          -1, TIME.WEEK.time * 2, TIME.HOUR.time * 2);
+      this.blockOutManager.addBlockout(blockOutTrigger);
+
+      assertFalse(this.blockOutManager.shouldFireNow());
+
+      this.blockOutManager.deleteBlockout(blockOutTrigger.getName());
+      blockOutStartDate = new Date(System.currentTimeMillis() + TIME.HOUR.time);
+      blockOutTrigger = new SimpleBlockoutTrigger("blockOut1", blockOutStartDate, null, //$NON-NLS-1$
+          -1, TIME.WEEK.time * 2, TIME.HOUR.time * 2);
+      this.blockOutManager.addBlockout(blockOutTrigger);
+
+      assertTrue(this.blockOutManager.shouldFireNow());
+    } catch (SchedulerException e) {
+      throw new RuntimeException(e);
+    }
   }
\ No newline at end of file
