diff --git a/old_test.java b/new_test.java
index f90004f..6e49e04 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
 		Context.getUserService().saveUser(user, null);
 		
 		//when
-		List<Locale> searchLocales = Context.getAdministrationService().getSearchLocales(Context.getAuthenticatedUser());
+		List<Locale> searchLocales = Context.getAdministrationService().getSearchLocales();
 		
 		//then
 		Assert.assertTrue("en_US", searchLocales.contains(new Locale("en", "US")));
