diff --git a/old_product.java b/new_product.java
index 5a9491a..ebb0f1d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public FileRecords read(int position, int size) throws IOException {
+public FileRecords slice(int position, int size) throws IOException {
         if (position < 0)
             throw new IllegalArgumentException("Invalid position: " + position + " in read from " + file);
         if (size < 0)
