diff --git a/old_product.java b/new_product.java
index 398b9d1..42fff90 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,7 @@
 public void removeCommand(final Command command)
-            throws GenieException {
+            throws GeniePreconditionException {
         if (command == null) {
-            throw new GenieException(
-                    HttpURLConnection.HTTP_BAD_REQUEST,
-                    "No command entered unable to remove.");
+            throw new GeniePreconditionException("No command entered unable to remove.");
         }
 
         if (this.commands != null) {
