diff --git a/old_test.java b/new_test.java
index 4397101..c7a4e5e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,24 +3,24 @@
 	{
 		final CompositeSecurityRule rule = new CompositeSecurityRule();
 		final String tableName = "DummyTableName"; // does not matter
-		final boolean rw = false; // does not matter
+		final Access access = Access.READ; // does not matter
 
 		{
 			final Set<OrgId> expected = asOrgIdsSet();
 			final Set<OrgId> actual = asOrgIdsSet();
-			rule.filterOrgs(role, tableName, rw, actual);
+			rule.filterOrgs(role, tableName, access, actual);
 			Assert.assertEquals("Org IDs shall not be modified", expected, actual);
 		}
 		{
 			final Set<OrgId> expected = asOrgIdsSet(1);
 			final Set<OrgId> actual = asOrgIdsSet(1);
-			rule.filterOrgs(role, tableName, rw, actual);
+			rule.filterOrgs(role, tableName, access, actual);
 			Assert.assertEquals("Org IDs shall not be modified", expected, actual);
 		}
 		{
 			final Set<OrgId> expected = asOrgIdsSet(1, 2, 3);
 			final Set<OrgId> actual = asOrgIdsSet(1, 2, 3);
-			rule.filterOrgs(role, tableName, rw, actual);
+			rule.filterOrgs(role, tableName, access, actual);
 			Assert.assertEquals("Org IDs shall not be modified", expected, actual);
 		}
 	}
\ No newline at end of file
