diff --git a/old_test.java b/new_test.java
index 899b499..c2fde91 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,14 +16,14 @@
                 -(5 * ONE_HOUR), st1.getOffset(time2));
 
         // test on subclass Support_TimeZone, an instance with daylight savings
-        TimeZone tz1 = new Support_TimeZone(-5 * ONE_HOUR, true);
+        TimeZone tz1 = newSupportTimeZone(-5 * ONE_HOUR, true);
         assertEquals("T3. Incorrect offset returned, ",
                 -(5 * ONE_HOUR), tz1.getOffset(time1));
         assertEquals("T4. Incorrect offset returned, ",
                 -(4 * ONE_HOUR), tz1.getOffset(time2));
 
         // an instance without daylight savings
-        tz1 = new Support_TimeZone(3 * ONE_HOUR, false);
+        tz1 = newSupportTimeZone(3 * ONE_HOUR, false);
         assertEquals("T5. Incorrect offset returned, ",
                 (3 * ONE_HOUR), tz1.getOffset(time1));
         assertEquals("T6. Incorrect offset returned, ",
