diff --git a/old_product.java b/new_product.java
index e53a810..b5930c1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,8 @@
 List<String> timeoutRecords(Long currentTime) {
-		if (timeOfLastUpdate + TIMEOUT < currentTime) {
+		if (timeOfLastUpdate + timeout < currentTime) {
 			log.trace("Updating record buffer");
 			List<String> timedOutRecords = new LinkedList<String>();
-			Map<Long, Set<String>> timedOut = recordsByTime.subMap(0L,
-					currentTime - TIMEOUT);
+			Map<Long, Set<String>> timedOut = recordsByTime.subMap(0L, currentTime - timeout);
 
 			for (Set<String> recordSet : timedOut.values()) {
 				if (!recordSet.isEmpty()) {
