diff --git a/old_test.java b/new_test.java
index 676d372..8a8b37e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testInclude ()
     {
-        System.out.println("include");
+        System.out.println("\n+++ include");
 
         RunTable instance = createHorizontalInstance();
 
@@ -11,11 +11,11 @@
         GlobalFilter filter = new GlobalFilter(buffer, 127);
         RunTableFactory factory = new RunTableFactory(HORIZONTAL);
         RunTable that = factory.createTable("inverted", filter.filteredImage());
-        that.dumpSequences();
         System.out.println("inverted" + that.dumpOf());
+        that.dumpSequences();
 
         instance.include(that);
-        instance.dumpSequences();
         System.out.println("full" + instance.dumpOf());
+        instance.dumpSequences();
         assertEquals(5, instance.getTotalRunCount());
     }
\ No newline at end of file
