diff --git a/old_product.java b/new_product.java
index 92095e3..8c82c11 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,6 @@
     // remove NAGIOS to make way for the new embedded alert framework
     removeNagiosService();
     addNewConfigurationsFromXml();
-    updateDfsClusterAdmintistratorsProperty();
     updateHiveDatabaseType();
     setSecurityType();
   }
\ No newline at end of file
