diff --git a/old_test.java b/new_test.java
index 98de48b..590f854 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,23 +5,18 @@
     Method addNewConfigurationsFromXml = AbstractUpgradeCatalog.class.getDeclaredMethod
         ("addNewConfigurationsFromXml");
     Method setSecurityType = UpgradeCatalog200.class.getDeclaredMethod("setSecurityType");
-    Method updateDfsClusterAdmintistratorsProperty = UpgradeCatalog200.class.getDeclaredMethod("updateDfsClusterAdmintistratorsProperty");
 
     UpgradeCatalog200 upgradeCatalog = createMockBuilder(UpgradeCatalog200.class)
         .addMockedMethod(removeNagiosService)
         .addMockedMethod(updateHiveDatabaseType)
         .addMockedMethod(addNewConfigurationsFromXml)
         .addMockedMethod(setSecurityType)
-        .addMockedMethod(updateDfsClusterAdmintistratorsProperty)
         .createMock();
 
     upgradeCatalog.removeNagiosService();
     expectLastCall().once();
     upgradeCatalog.addNewConfigurationsFromXml();
     expectLastCall();
-
-    upgradeCatalog.updateDfsClusterAdmintistratorsProperty();
-    expectLastCall();
     
     upgradeCatalog.updateHiveDatabaseType();
     expectLastCall().once();
