diff --git a/old_product.java b/new_product.java
index b78c77f..8f11fe5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,15 @@
                          final AsyncResponse asyncResponse) {
         long traceId = LoggerHelpers.traceEnter(log, "getScope");
 
+        try {
+            authenticate(scopeName, READ);
+        } catch (AuthenticationException e) {
+            log.warn("Get scope for {} failed due to authentication failure.", scopeName);
+            asyncResponse.resume(Response.status(Status.UNAUTHORIZED).build());
+            LoggerHelpers.traceLeave(log, "getScope", traceId);
+            return;
+        }
+
         controllerService.getScope(scopeName)
                 .thenApply(scope -> {
                         return Response.status(Status.OK).entity(new ScopeProperty().scopeName(scope)).build();
