diff --git a/old_product.java b/new_product.java
index 1aafa2c..fb75631 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public RunnerImpl create(Class<?> klass) throws InvocationTargetException {
+public InternalRunner create(Class<?> klass) throws InvocationTargetException {
         return create(klass, new Supplier<MockitoTestListener>() {
             public MockitoTestListener get() {
                 return new NoOpTestListener();
