diff --git a/old_test.java b/new_test.java
index 48273bf..a842094 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
-public RunnerImpl create(Class<?> klass, Supplier<MockitoTestListener> listenerSupplier) throws InvocationTargetException {
+public InternalRunner create(Class<?> klass, Supplier<MockitoTestListener> listenerSupplier) throws InvocationTargetException {
         try {
             return new RunnerProvider().newInstance("org.mockito.internal.runners.SilentJUnitRunner", klass, listenerSupplier);
         } catch (InvocationTargetException e) {
