diff --git a/old_test.java b/new_test.java
index de45261..f3bed28 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,7 @@
 	public void validate_shouldPassValidationIfAllRequiredFieldsHaveProperValues() throws Exception {
 		Program prog = new Program();
 		prog.setName("Hypochondriasis program");
+        prog.setDescription("This is Hypochondriasis program");
 		prog.setConcept(Context.getConceptService().getConcept(3));
 		
 		Errors errors = new BindException(prog, "prog");
