diff --git a/old_product.java b/new_product.java
index a0b5d08..218002a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public static void fullViewLeft(CameraPinholeRadial paramLeft,
-									RowMatrix_F32 rectifyLeft, RowMatrix_F32 rectifyRight,
-									RowMatrix_F32 rectifyK)
+									FMatrixRMaj rectifyLeft, FMatrixRMaj rectifyRight,
+									FMatrixRMaj rectifyK)
 	{
 		ImplRectifyImageOps_F32.fullViewLeft(paramLeft, rectifyLeft, rectifyRight, rectifyK);
 	}
\ No newline at end of file
