diff --git a/old_test.java b/new_test.java
index 842f528..e089038 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,16 @@
 @Test
-	public void voidOrderGroup_shouldVoidOrderGroup() throws Exception {
+	public void voidOrderGroup_shouldVoidOrdersInGroup() throws Exception {
 		OrderGroup group = Context.getOrderService().getOrderGroup(2);
 		String reason = "because";
 		
 		group = Context.getOrderService().voidOrderGroup(group, reason);
+		Order order = (Order) group.getMembers().toArray()[0];
 		
+		// check if group is voided
 		Assert.assertNotNull(group);
 		Assert.assertTrue(group.isVoided());
+		
+		// check if group members are voided
+		Assert.assertNotNull(order);
+		Assert.assertTrue(order.isVoided());
 	}
\ No newline at end of file
