diff --git a/old_product.java b/new_product.java
index 6aa0b3b..3e2c5ca 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,11 +7,5 @@
 		if (!StringUtils.hasLength(voidReason))
 			throw new IllegalArgumentException("voidReason cannot be empty or null");
 		
-		group.setVoided(Boolean.TRUE);
-		group.setVoidReason(voidReason);
-		group.setVoidedBy(Context.getAuthenticatedUser());
-		if (group.getDateVoided() == null)
-			group.setDateVoided(new Date());
-		
 		return dao.saveOrderGroup(group);
 	}
\ No newline at end of file
