diff --git a/old_test.java b/new_test.java
index 725eeff..829a265 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,7 +17,7 @@
         this.service.createCluster(cluster);
         final Cluster created = this.service.getCluster(id);
         Assert.assertNotNull(created);
-        Assert.assertEquals(id, created.getId());
+        Assert.assertEquals(id, created.getId().orElseThrow(IllegalArgumentException::new));
         Assert.assertEquals(CLUSTER_1_NAME, created.getName());
         Assert.assertEquals(CLUSTER_1_USER, created.getUser());
         Assert.assertEquals(ClusterStatus.OUT_OF_SERVICE, created.getStatus());
