diff --git a/old_product.java b/new_product.java
index d258031..3dcbc87 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,13 +4,14 @@
         @Valid
         final Cluster cluster
     ) throws GenieException {
-        log.debug("Called to create cluster {}", cluster.toString());
-        if (StringUtils.isNotBlank(cluster.getId()) && this.clusterRepo.exists(cluster.getId())) {
-            throw new GenieConflictException("A cluster with id " + cluster.getId() + " already exists");
+        log.debug("Called to create cluster {}", cluster);
+        final Optional<String> clusterId = cluster.getId();
+        if (clusterId.isPresent() && this.clusterRepo.exists(clusterId.get())) {
+            throw new GenieConflictException("A cluster with id " + clusterId.get() + " already exists");
         }
 
         final ClusterEntity clusterEntity = new ClusterEntity();
-        clusterEntity.setId(StringUtils.isBlank(cluster.getId()) ? UUID.randomUUID().toString() : cluster.getId());
+        clusterEntity.setId(cluster.getId().orElse(UUID.randomUUID().toString()));
         this.updateAndSaveClusterEntity(clusterEntity, cluster);
         return clusterEntity.getId();
     }
\ No newline at end of file
