diff --git a/old_product.java b/new_product.java
index 684ecd3..2c4f0b5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,8 +8,12 @@ public void process( LookupSimilarImages similarImages ) {
 		FastQueue<AssociatedIndex> matches = new FastQueue<>(AssociatedIndex.class,true);
 		FastQueue<AssociatedPair> pairs = new FastQueue<>(AssociatedPair.class,true);
 
+		// map to quickly look up the ID of a view
+		Map<String,Integer> imageToindex = new HashMap<>();
+
 		// Create a node in the graph for each image
 		for (int idxTgt = 0; idxTgt < imageIds.size(); idxTgt++) {
+			imageToindex.put(imageIds.get(idxTgt),idxTgt);
 			graph.createNode(imageIds.get(idxTgt));
 		}
 
@@ -25,8 +29,14 @@ public void process( LookupSimilarImages similarImages ) {
 
 			for (int idxSimilar = 0; idxSimilar < similar.size(); idxSimilar++) {
 				String dst = similar.get(idxSimilar);
-				similarImages.lookupFeatures(dst,dstFeats);
 
+				// make sure it isn't considering the same motion twice
+				int dstIdx = imageToindex.get(dst);
+				if( dstIdx <= idxTgt )
+					continue;
+
+				// get information on the features and association
+				similarImages.lookupFeatures(dst,dstFeats);
 				similarImages.lookupMatches(src,dst,matches);
 
 				pairs.reset();
