diff --git a/old_product.java b/new_product.java
index b33b4dc..ed1c6b0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public static <T> PredicatedCollection<T> predicatedCollection(final Collection<T> coll,
                                                                    final Predicate<? super T> predicate) {
-        return new PredicatedCollection<T>(coll, predicate);
+        return new PredicatedCollection<>(coll, predicate);
     }
\ No newline at end of file
