diff --git a/old_product.java b/new_product.java
index 5ebc0fb..700ff83 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ public void completeFile(AlluxioURI path, CompleteFileOptions options)
       InvalidFileSizeException, FileAlreadyCompletedException, AccessControlException {
     MasterContext.getMasterSource().incCompleteFileOps(1);
     synchronized (mInodeTree) {
-      checkPermission(FileSystemAction.WRITE, path, false);
+      mPermissionChecker.checkPermission(FileSystemAction.WRITE, path);
       // Even readonly mount points should be able to complete a file, for UFS reads in CACHE mode.
       long opTimeMs = System.currentTimeMillis();
       Inode<?> inode = mInodeTree.getInodeByPath(path);
