diff --git a/old_test.java b/new_test.java
index bc5f555..24a7206 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void completeFileTest() throws Exception {
-    mFileSystemMaster.create(ROOT_FILE_URI, sNestedFileOptions);
+    mFileSystemMaster.createFile(ROOT_FILE_URI, sNestedFileOptions);
     writeBlockForFile(ROOT_FILE_URI);
     completeFile(ROOT_FILE_URI);
 
