diff --git a/old_test.java b/new_test.java
index 282d640..7b94cdf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,7 +10,19 @@
     } else {
       for(final IOFile f : new IOFile("etc").children()) {
         final String n = f.name();
-        if(n.endsWith(".bat")) libraries(n, libs);
+        final StringList missing = new StringList();
+        final StringList obsolete = new StringList();
+        if(n.endsWith(".bat")) libraries(n, libs, missing, obsolete);
+        if(!missing.isEmpty()) {
+          final StringBuilder sb = new StringBuilder();
+          for(final String l : missing) sb.append(';').append(l);
+          fail("Library not found in '" + n + "':\n" + sb.substring(1));
+        }
+        if(!obsolete.isEmpty()) {
+          final StringBuilder sb = new StringBuilder();
+          for(final String l : obsolete) sb.append(';').append(l);
+          fail("Library obsolete in '" + n + "':\n" + sb.substring(1));
+        }
       }
     }
   }
\ No newline at end of file
