diff --git a/old_product.java b/new_product.java
index 709e15a..9d23e74 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,6 @@
-private void libraries(final String name, final HashSet<String> libs) throws Exception {
+private void libraries(final String name, final HashSet<String> libs,
+      final StringList missing, final StringList obsolete) throws Exception {
+
     final HashSet<String> sl = new HashSet<String>();
     final NewlineInput nli = new NewlineInput(IO.get("etc/" + name));
     try {
@@ -13,9 +15,9 @@ private void libraries(final String name, final HashSet<String> libs) throws Exc
 
     for(final String l : libs) {
       if(l.contains("basex")) continue;
-      assertTrue("Library not found in '" + name + "': " + l, sl.remove(l));
+      if(!sl.remove(l)) missing.add(l);
+    }
+    for(final String l : sl) {
+      if(l.endsWith(".jar")) obsolete.add(l);
     }
-    final StringBuilder sb = new StringBuilder();
-    for(final String l : sl) sb.append("\n- ").append(l);
-    assertTrue("Libraries superfluous in '" + name + "':" + sb, sl.isEmpty());
   }
\ No newline at end of file
