diff --git a/old_test.java b/new_test.java
index 69d116b..2a9b3d0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-	@Verifies(value = "should matchStringsHashedWithCorrectAlgorithm", method = "hashMatches(String,String)")
-	public void hashMatches_shouldMatchStringsHashedWithCorrectAlgorithm() throws Exception {
+	@Verifies(value = "should match strings hashed with sha1 algorithm", method = "hashMatches(String,String)")
+	public void hashMatches_shouldMatchStringsHashedWithSha1Algorithm() throws Exception {
 		Assert.assertTrue(Security.hashMatches("4a1750c8607d0fa237de36c6305715c223415189", "test"
 		        + "c788c6ad82a157b712392ca695dfcf2eed193d7f"));
 	}
\ No newline at end of file
