diff --git a/old_product.java b/new_product.java
index 469345c..a499811 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,11 +2,7 @@ public static boolean hashMatches(String hashedPassword, String passwordToHash)
 		if (hashedPassword == null || passwordToHash == null)
 			throw new APIException("Neither the hashed password or the password to hash cannot be null");
 		
-		String newHashedPassword = encodeString(passwordToHash);
-		if (hashedPassword.equals(newHashedPassword))
-			return true;
-		else {
-			String incorrectlyHashedPassword = incorrectlyEncodeString(passwordToHash);
-			return hashedPassword.equals(incorrectlyHashedPassword);
-		}
+		return hashedPassword.equals(encodeString(passwordToHash))
+		        || hashedPassword.equals(encodeStringSHA1(passwordToHash))
+		        || hashedPassword.equals(incorrectlyEncodeString(passwordToHash));
 	}
\ No newline at end of file
