diff --git a/old_product.java b/new_product.java
index 4a3461f..731fbea 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,10 @@
 protected String getCurrentReleaseVersion() {
         HttpURLConnection conn = null;
         try {
-            final String str = Settings.getString(Settings.KEYS.ENGINE_VERSION_CHECK_URL, "http://jeremylong.github.io/DependencyCheck/current.txt");
+            final String str = settings.getString(Settings.KEYS.ENGINE_VERSION_CHECK_URL, "http://jeremylong.github.io/DependencyCheck/current.txt");
             final URL url = new URL(str);
-            conn = URLConnectionFactory.createHttpURLConnection(url);
+            URLConnectionFactory factory = new URLConnectionFactory(settings);
+            conn = factory.createHttpURLConnection(url);
             conn.connect();
             if (conn.getResponseCode() != 200) {
                 return null;
