diff --git a/old_test.java b/new_test.java
index 9cd6ad5..50ef542 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,16 +3,16 @@
         String str = "apache";
         String str2 = "owasp";
         Dependency instance = new Dependency();
-        instance.vendorEvidence.addEvidence("manifest", "something", "apache", Evidence.Confidence.HIGH);
-        instance.vendorEvidence.addEvidence("manifest", "something", "owasp", Evidence.Confidence.MEDIUM);
+        instance.getVendorEvidence().addEvidence("manifest", "something", "apache", Evidence.Confidence.HIGH);
+        instance.getVendorEvidence().addEvidence("manifest", "something", "owasp", Evidence.Confidence.MEDIUM);
         assertFalse(instance.containsUsedString(str));
         assertFalse(instance.containsUsedString(str2));
-        for (Evidence i : instance.vendorEvidence.iterator(Evidence.Confidence.HIGH)) {
+        for (Evidence i : instance.getVendorEvidence().iterator(Evidence.Confidence.HIGH)) {
             String readValue = i.getValue();
         }
         assertTrue(instance.containsUsedString(str));
         assertFalse(instance.containsUsedString(str2));
-        for (Evidence i : instance.vendorEvidence.iterator(Evidence.Confidence.MEDIUM)) {
+        for (Evidence i : instance.getVendorEvidence().iterator(Evidence.Confidence.MEDIUM)) {
             String readValue = i.getValue();
         }
         assertTrue(instance.containsUsedString(str));
