diff --git a/old_product.java b/new_product.java
index 44f174a..47b5853 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,10 +2,10 @@ public boolean containsUsedString(String text) {
         if (text == null) {
             return false;
         }
-        text = text.toLowerCase();
+        final String textToTest = text.toLowerCase();
 
         for (Evidence e : this.list) {
-            if (e.used && e.value.toLowerCase().contains(text)) {
+            if (e.isUsed() && e.getValue().toLowerCase().contains(textToTest)) {
                 return true;
             }
         }
