diff --git a/old_product.java b/new_product.java
index 1ca4bec..097881a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,8 @@ public Bln isClosed(final ANode node) throws QueryException {
 			
       // Check QName
       QNm[] line = {Q_GML_LINEARRING, Q_GML_LINESTRING, Q_GML_MULTILINESTRING};
-      QNm[] other = {Q_GML_POLYGON, Q_GML_MULTIPOINT, Q_GML_MULTIPOLYGON};
+      QNm[] other = {Q_GML_POLYGON, Q_GML_MULTIPOINT, Q_GML_MULTIPOLYGON,
+                      Q_GML_POINT};
       if(checkNode(node, line)) {
         Geometry geom = gmlReader(node);
 				if (geom instanceof LineString) 
@@ -16,5 +17,5 @@ public Bln isClosed(final ANode node) throws QueryException {
 			}
 			if (checkNode(node, other))
 			 throw GeoErrors.lineNeeded(node.qname().local());
-			return null;
+			throw GeoErrors.unrecognizedGeo(node.qname().local());
 		}
\ No newline at end of file
