diff --git a/old_test.java b/new_test.java
index 88ca511..69310f7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -22,6 +22,6 @@
 			  		new QNm("GEO0002"));
 	  runError("geo:isClosed()", new QNm("XPST0017"));
 	  runError("geo:isClosed(text {'gml:Point'})", new QNm("FORG0006"));
-	  runError("geo:isClosed(<Point><gml:coordinates>2,1</gml:coordinates></Point>)",
-		  		new QNm("GEO0001"));
+	  runError("geo:isClosed(<gml:Point><gml:coordinates>2,1</gml:coordinates></gml:Point>)",
+		  		new QNm("GEO0004"));
   }
\ No newline at end of file
