diff --git a/old_test.java b/new_test.java
index f15dc13..cd69cca 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
 //        Settings.setString(Settings.KEYS.CONNECTION_TIMEOUT, "1000");
 //        Settings.setString(Settings.KEYS.PROXY_PORT, "8080");
 //        Settings.setString(Settings.KEYS.PROXY_SERVER, "127.0.0.1");
-        URL url = new URL(getSettings().getString(Settings.KEYS.CVE_MODIFIED_20_URL));
+        URL url = new URL(getSettings().getString(Settings.KEYS.CVE_MODIFIED_JSON));
         File outputPath = new File("target/downloaded_cve.xml");
         Downloader downloader = new Downloader(getSettings());
         downloader.fetchFile(url, outputPath);
