diff --git a/old_product.java b/new_product.java
index fd94f68..b021abe 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 private Item zipFile(final QueryContext qc) throws QueryException {
     // check argument
-    final ANode elm = (ANode) checkType(exprs[0].item(qc, info), NodeType.ELM);
-    if(!elm.qname().eq(Q_FILE)) throw ZIP_UNKNOWN.get(info, elm.qname());
+    final ANode elm = toElem(exprs[0], qc);
+    if(!elm.qname().eq(Q_FILE)) throw ZIP_UNKNOWN_X.get(info, elm.qname());
     // get file
     final String file = attribute(elm, HREF, true);
 
@@ -12,7 +12,7 @@ private Item zipFile(final QueryContext qc) throws QueryException {
       create(zos, elm.children(), "", null, qc);
     } catch(final IOException ex) {
       ok = false;
-      throw ZIP_FAIL.get(info, ex);
+      throw ZIP_FAIL_X.get(info, ex);
     } finally {
       if(!ok) new IOFile(file).delete();
     }
