diff --git a/old_test.java b/new_test.java
index 85dce5b..6f18398 100644
--- a/old_test.java
+++ b/new_test.java
@@ -23,5 +23,5 @@
 
     // error: duplicate entry specified
     error(_ZIP_ZIP_FILE.args(params("<entry src='" + TMPFILE + "'/>" +
-        "<entry src='" + TMPFILE + "'/>")), Err.ZIP_FAIL);
+        "<entry src='" + TMPFILE + "'/>")), Err.ZIP_FAIL_X);
   }
\ No newline at end of file
