diff --git a/old_test.java b/new_test.java
index c3f8805..bdcf233 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
             .thenReturn(Collections.singletonList(PROXY));
 
         final WebSocketsProxyConnectionHandler handler = new WebSocketsProxyConnectionHandler(CONNECTION_ID, HOSTNAME,
-            PROXY_CONFIGURATION);
+            PROXY_CONFIGURATION, PRODUCT, CLIENT_VERSION);
 
         // Act and Assert
         Assertions.assertEquals(PROXY_ADDRESS.getHostName(), handler.getHostname());
