diff --git a/old_product.java b/new_product.java
index e5a7beb..e8a376c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,7 @@
 
         // Act
         final ConnectionHandler handler = provider.createConnectionHandler(CONNECTION_ID, hostname,
-            AmqpTransportType.AMQP_WEB_SOCKETS, configuration);
+            AmqpTransportType.AMQP_WEB_SOCKETS, configuration, PRODUCT, CLIENT_VERSION);
 
         // Act and Assert
         Assertions.assertEquals(PROXY_ADDRESS.getHostName(), handler.getHostname());
