diff --git a/old_product.java b/new_product.java
index b82e66a..95a4765 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,3 @@
-public void setConfigs(final Set<String> configs) throws GenieException {
-        if (configs == null || configs.isEmpty()) {
-            throw new GenieException(
-                    HttpURLConnection.HTTP_BAD_REQUEST,
-                    "At least one config required.");
-        }
+public void setConfigs(final Set<String> configs) {
         this.configs = configs;
     }
\ No newline at end of file
