diff --git a/old_test.java b/new_test.java
index 26817ee..d9dfce3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,8 @@
 @Test
     public void testSetConfigs() {
-        final Application a = new Application();
-        Assert.assertNull(a.getConfigs());
+        Assert.assertNull(this.a.getConfigs());
         final Set<String> configs = new HashSet<String>();
         configs.add("s3://netflix.configFile");
-        a.setConfigs(configs);
-        Assert.assertEquals(configs, a.getConfigs());
+        this.a.setConfigs(configs);
+        Assert.assertEquals(configs, this.a.getConfigs());
     }
\ No newline at end of file
