diff --git a/old_product.java b/new_product.java
index 30e36a2..0ef3516 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,7 @@ public String downloadFileFromServer(String url, String fileName, String targetF
 
         // 下载
         return restfulMgr
-                .downloadFromServer(remoteUrl, fileName, localDir, targetFileDir,
+                .downloadFromServer(remoteUrl, fileName, localDir, localDownloadDirTemp, targetFileDir,
                         enableLocalDownloadDirInClassPath,
                         retryTime,
                         retrySleepSeconds);
