diff --git a/old_product.java b/new_product.java
index a23fc2e..eb3ec09 100644
--- a/old_product.java
+++ b/new_product.java
@@ -23,8 +23,7 @@ public void process ()
 
         final RunsTable hugeHoriTable = new RunsTableFactory(
                 HORIZONTAL,
-                (PixelFilter) sheet.getPicture().getSource(
-                        Picture.SourceKey.BINARY),
+                sheet.getPicture().getSource(Picture.SourceKey.BINARY),
                 minRunLength).createTable("huge-hori", filter);
 
         final Lag lag = new BasicLag(
@@ -38,7 +37,8 @@ public void process ()
         sectionsBuilder.createSections(hugeHoriTable, true);
 
         sheet.setLag(Lags.FULL_HLAG, lag);
-        sheet.getSystemManager().dispatchHorizontalHugeSections();
+        sheet.getSystemManager()
+                .dispatchHorizontalHugeSections();
 
         if (Main.getGui() != null) {
             // Display a view on this lag
