diff --git a/old_test.java b/new_test.java
index a0818b4..51e9fac 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(groups={"tck"})
     public void test_print_basicIsoDate() {
-        CalendricalObject test = LocalDateTime.of(2008, 6, 3, 11, 5, 30);
+        DateTimeCalendrical test = LocalDateTime.of(2008, 6, 3, 11, 5, 30);
         assertEquals(DateTimeFormatters.basicIsoDate().print(test), "20080603");
     }
\ No newline at end of file
