diff --git a/old_product.java b/new_product.java
index 8b547f0..2ca9fc5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,9 @@
-public void addValues(String name, String[] values) 
+public void addValues(String name, V[] values) 
     {
-        Object lo = get(name);
-        Object ln = LazyList.addCollection(lo,Arrays.asList(values));
-        if (lo!=ln)
-            put(name,ln);
+        List<V> lo = get(name);
+        if(lo == null) {
+            lo = new ArrayList<>();
+        }
+        lo.addAll(Arrays.asList(values));
+        put(name,lo);
     }
\ No newline at end of file
