diff --git a/old_product.java b/new_product.java
index d17ee85..a51a98b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,11 +6,11 @@ private void query(final ServerCmd sc) throws IOException {
     String err = null;
     try {
       if(sc == QUERY) {
-        qp = new QueryProcess(arg, out, context);
+        final String query = arg;
+        qp = new QueryProcess(query, out, context);
         arg = Integer.toString(id++);
         queries.put(arg, qp);
-
-        log.write(this, arg, qp.query, OK);
+        log.write(this, arg, query, OK);
         // send {ID}0
         out.writeString(arg);
       } else {
@@ -25,11 +25,13 @@ private void query(final ServerCmd sc) throws IOException {
           qp.init();
         } else if(sc == NEXT) {
           qp.next();
+        } else if(sc == EXEC) {
+          qp.execute();
+        } else if(sc == INFO) {
+          qp.info();
         } else if(sc == CLOSE && qp != null) {
           qp.close(false);
           queries.remove(arg);
-        } else if(sc == EXEC) {
-          qp.execute();
         }
         // send 0 as end marker
         out.write(0);
