diff --git a/old_test.java b/new_test.java
index fcd5ec1..b91398a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test
   public void query() throws BaseXException {
     final ClientQuery cq = cs.query("1");
-    if(!cq.more()) fail("No result returned");
-    assertEquals("1", cq.next());
+    assertEquals("1", cq.execute());
     assertEquals("", cq.close());
   }
\ No newline at end of file
