diff --git a/old_test.java b/new_test.java
index 6e48976..2bdcaa0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testPrefop() {
         try {
         	String jobid = "z0.5c81361a38b9f349c8bb5288";
-            Configuration cfg = new Configuration();
+            Configuration cfg = new Configuration(Region.autoRegion());
             OperationManager operationManager = new OperationManager(TestConfig.testAuth, cfg);
             OperationStatus status = operationManager.prefop(jobid);
             Assert.assertEquals(0, status.code);
