diff --git a/old_product.java b/new_product.java
index c7d3b61..3a572a0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,21 +1,7 @@
 public <T> T prefop(String persistentId, Class<T> retClass) throws QiniuException {
         StringMap params = new StringMap().put("id", persistentId);
         byte[] data = StringUtils.utf8Bytes(params.formString());
-        String apiHost;
-
-        if (this.configuration.region != null) {
-            apiHost = this.configuration.region.getApiHost();
-            if (this.configuration.useHttpsDomains) {
-                apiHost = this.configuration.region.getApiHost();
-            }
-        } else {
-            apiHost = "http://api.qiniu.com";
-            if (this.configuration.useHttpsDomains) {
-                apiHost = "https://api.qiniu.com";
-            }
-        }
-
-        String url = String.format("%s/status/get/prefop", apiHost);
+        String url = String.format("%s/status/get/prefop", configuration.apiHost());
         Response response = this.client.post(url, data, null, Client.FormMime);
         return response.jsonToObject(retClass);
     }
\ No newline at end of file
