diff --git a/old_test.java b/new_test.java
index 9d129d9..fd47478 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,7 @@
 	@Verifies(value = "should return empty list for concept without orders", method = "getOrderHistoryByConcept(Concept)")
 	public void getOrderHistoryByConcept_shouldReturnEmptyListForConceptWithoutOrders() throws Exception {
 		Concept concept = Context.getConceptService().getConcept(21);
-		List<Order> orders = Context.getOrderService().getOrderHistoryByConcept(concept);
+		Patient patient = Context.getPatientService().getPatient(1);
+		List<Order> orders = Context.getOrderService().getOrderHistoryByConcept(patient, concept);
 		Assert.assertEquals(0, orders.size());
 	}
\ No newline at end of file
