diff --git a/old_product.java b/new_product.java
index d1aa3e1..3d70a0a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,6 @@
 CompletableFuture<Boolean> createStream(final String scopeName,
                                             final String streamName,
                                             final StreamConfiguration configuration,
-                                            final long createTimestamp);
\ No newline at end of file
+                                            final long createTimestamp,
+                                            final OperationContext context,
+                                            final Executor executor);
\ No newline at end of file
