diff --git a/old_test.java b/new_test.java
index 44fee2e..410abcf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -30,7 +30,7 @@
             .withDisableLogArchival(true)
             .build();
 
-        final JobRequestMetadata metadata = new JobRequestMetadata
+        final JobMetadata metadata = new JobMetadata
             .Builder()
             .withClientHost(clientHost)
             .withUserAgent(UUID.randomUUID().toString())
@@ -54,10 +54,13 @@
             jobExecutionArgumentCaptor.capture()
         );
 
-        Assert.assertEquals(JOB_1_ID, jobArgumentCaptor.getValue().getId());
+        Assert.assertEquals(JOB_1_ID, jobArgumentCaptor.getValue().getId().orElseThrow(IllegalArgumentException::new));
         Assert.assertEquals(JOB_1_NAME, jobArgumentCaptor.getValue().getName());
         Assert.assertEquals(JOB_1_USER, jobArgumentCaptor.getValue().getUser());
         Assert.assertEquals(JOB_1_VERSION, jobArgumentCaptor.getValue().getVersion());
         Assert.assertEquals(JobStatus.INIT, jobArgumentCaptor.getValue().getStatus());
-        Assert.assertEquals(description, jobArgumentCaptor.getValue().getDescription());
+        Assert.assertEquals(
+            description,
+            jobArgumentCaptor.getValue().getDescription().orElseThrow(IllegalArgumentException::new)
+        );
     }
\ No newline at end of file
