diff --git a/old_product.java b/new_product.java
index 46a9cf4..352ea08 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,10 @@
-public boolean handleFailure(IFailureHandlerPolicy failureHandlerPolicy,
-                              Layout currentLayout,
-                              CorfuRuntime corfuRuntime,
-                              Set<String> failedServers,
-                              Set<String> healedServers) {
+public boolean handleFailure(IReconfigurationHandlerPolicy failureHandlerPolicy,
+                                 Layout currentLayout,
+                                 CorfuRuntime corfuRuntime,
+                                 Set<String> failedServers) {
         try {
             corfuRuntime.getLayoutManagementView().handleFailure(failureHandlerPolicy,
-                    currentLayout, failedServers, healedServers);
+                    currentLayout, failedServers);
             return true;
         } catch (Exception e) {
             log.error("Error: handleFailure: {}", e);
