diff --git a/old_test.java b/new_test.java
index ca9505b..0db8f2d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,5 @@
     public void handleFailure()
             throws Exception {
         // Since the servers are started as single nodes thus already bootstrapped.
-        assertThat(
-                client.handleFailure(Collections.singleton("key"), Collections.emptySet()).get())
-                .isEqualTo(true);
+        assertThat(client.handleFailure(0L, Collections.singleton("key")).get()).isEqualTo(true);
     }
\ No newline at end of file
