diff --git a/old_product.java b/new_product.java
index 02760d6..ef8be17 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,6 +5,8 @@
 		if (encounter.getVisit() != null)
 			return;
 		
+		List<Patient> patients = new ArrayList<Patient>();
+		patients.add(encounter.getPatient());
 		List<Visit> visits = Context.getVisitService().getVisitsByPatient(encounter.getPatient(), true, false);
 		if (visits == null)
 			return;
@@ -22,7 +24,7 @@
 				continue;
 			}
 			
-			if (visit.getLocation() == null || visit.getLocation().equals(encounter.getLocation())) {
+			if (visit.getLocation() == null || Location.isInHierarchy(encounter.getLocation(), visit.getLocation())) {
 				encounter.setVisit(visit);
 				return;
 			}
