diff --git a/old_product.java b/new_product.java
index c52cc93..5bb23ef 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public ServiceCall<List<IdentifiableLanguage>> getIdentifiableLanguages() {
     final RequestBuilder requestBuilder = RequestBuilder.get(PATH_IDENTIFIABLE_LANGUAGES);
 
     ResponseConverter<List<IdentifiableLanguage>> converter =
-        ResponseConverterUtils.getGenericObject(TYPE_LIST_IDENTIFIABLE_LANGUAGE, "languages");
+        ResponseConverterUtils.getGenericObject(TYPE_LIST_IDENTIFIABLE_LANGUAGE, LANGUAGES);
 
     return createServiceCall(requestBuilder.build(), converter);
 
