diff --git a/old_test.java b/new_test.java
index c0bef17..cedc367 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,11 @@
 @Test
   public void testGetIdentifiableLanguages() {
 
-    mockServer.when(request().withPath(IDENTIFIABLE_LANGUAGES_PATH)).respond(
-        response().withHeader(APPLICATION_JSON).withBody(
-            GsonSingleton.getGsonWithoutPrettyPrinting().toJson(identifiableLanguages)));
+    mockServer.when(request().withPath(IDENTIFIABLE_LANGUAGES_PATH)).respond(response().withHeader(APPLICATION_JSON)
+        .withBody(GsonSingleton.getGsonWithoutPrettyPrinting().toJson(identifiableLanguages)));
 
     List<IdentifiableLanguage> languages = service.getIdentifiableLanguages().execute();
 
     assertEquals(GsonSingleton.getGsonWithoutPrettyPrinting().toJson(languages),
-        GsonSingleton.getGsonWithoutPrettyPrinting().toJson(identifiableLanguages.getLanguages()));
+        GsonSingleton.getGsonWithoutPrettyPrinting().toJson(identifiableLanguages.get("languages")));
   }
\ No newline at end of file
