diff --git a/old_product.java b/new_product.java
index a238fe6..7276f5e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,15 +16,17 @@ V put(final DirectBytes keyBytes, final V value, int hash2, boolean replaceIfPre
                             continue;
                         final long keyLength = keyBytes.remaining();
                         tmpBytes.skip(keyLength);
-                        final long alignPosition = align(tmpBytes.position());
-                        tmpBytes.position(alignPosition);
                         if (replaceIfPresent) {
                             if (putReturnsNull) {
                                 appendInstance(keyBytes, value);
                                 return null;
                             }
-                            final V v = readObjectUsing(null, offset + alignPosition);
+                            long valuePosition = tmpBytes.position();
+                            tmpBytes.readStopBit();
+                            final long alignPosition = align(tmpBytes.position());
                             tmpBytes.position(alignPosition);
+                            final V v = readObjectUsing(null, offset + alignPosition);
+                            tmpBytes.position(valuePosition);
                             appendInstance(keyBytes, value);
                             return v;
 
