diff --git a/old_test.java b/new_test.java
index 9875c0e..410cf2c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testPut() throws ExecutionException, InterruptedException {
-        int count = 4000000;
+        int count = 1000000;
         HugeConfig config = HugeConfig.DEFAULT.clone()
                 .setSegments(256)
                 .setSmallEntrySize(72) // TODO 64 corrupts the values !!
