diff --git a/old_product.java b/new_product.java
index bd7aef7..9ec333a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,9 @@ public String coordinateJob(
         @NotNull(message = "No jobRequest provided. Unable to submit job for execution.")
         @Valid
         final JobRequest jobRequest,
-        final String clientHost
+        @NotNull
+        @Valid
+        final JobRequestMetadata jobRequestMetadata
     ) throws GenieException {
         log.debug("Called with job request {}", jobRequest);
         if (StringUtils.isBlank(jobRequest.getId())) {
@@ -10,7 +12,7 @@ public String coordinateJob(
         }
 
         // Log the job request and optionally the client host
-        this.jobPersistenceService.createJobRequest(jobRequest, clientHost);
+        this.jobPersistenceService.createJobRequest(jobRequest, jobRequestMetadata);
 
         String archiveLocation = null;
         if (!jobRequest.isDisableLogArchival()) {
