diff --git a/old_product.java b/new_product.java
index 3d87507..8c145ee 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
-public static Configuration loadFromHadoopConfiguration(
-      org.apache.hadoop.conf.Configuration source) {
+public static void mergeHadoopConfiguration(org.apache.hadoop.conf.Configuration source) {
     // Load Alluxio configuration if any and merge to the one in Alluxio file system
     // Push Alluxio configuration to the Job configuration
     Properties alluxioConfProperties = new Properties();
@@ -22,6 +21,5 @@ public static Configuration loadFromHadoopConfiguration(
       }
     }
     LOG.info("Loading Alluxio properties from Hadoop configuration: {}", alluxioConfProperties);
-    // FIX return Configuration.fromMap(alluxioConfProperties);
-    return null;
+    Configuration.merge(alluxioConfProperties);
   }
\ No newline at end of file
