diff --git a/old_product.java b/new_product.java
index 2871637..df74087 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,16 @@
-public static Map<String, String> readPrefixAsMap(String baseName,String prefix,String spliter,Locale locale){
+public static Map<String, String> readPrefixAsMap(String baseName,String prefix,String delimiters,Locale locale){
         Map<String, String> propertyMap = readAllPropertiesToMap(baseName, locale);
         if (Validator.isNullOrEmpty(propertyMap)){
             return Collections.emptyMap();
         }
 
-        Map<String, String> result = new HashMap<String, String>();
+        Map<String, String> result = new TreeMap<String, String>();
         for (Map.Entry<String, String> entry : propertyMap.entrySet()){
             String key = entry.getKey();
             String value = entry.getValue();
             // 以 prefix 开头
             if (key.startsWith(prefix)){
-                String[] values = key.split(spliter);// 分隔
+                String[] values = StringUtil.tokenizeToStringArray(key, delimiters);
                 if (values.length >= 2){
                     result.put(values[1], value);
                 }
