diff --git a/old_product.java b/new_product.java
index 1a27d92..ff98bc8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1 @@
-Set<DeviceId> getPhysicalDevices(NetworkId networkId,
-                                             VirtualDevice virtualDevice);
\ No newline at end of file
+Set<DeviceId> getPhysicalDevices(NetworkId networkId, DeviceId deviceId);
\ No newline at end of file
