diff --git a/old_product.java b/new_product.java
index 929cbb8..11d4864 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,9 +4,11 @@
         if (action instanceof ManagedTask)
             throw new IllegalArgumentException(ManagedTask.class.getName());
 
-        ThreadContextDescriptor contextDescriptor = captureThreadContext(defaultExecutor);
-        if (contextDescriptor != null)
-            action = new ContextualConsumer<>(contextDescriptor, action);
+        if (!(action instanceof ContextualConsumer)) {
+            ThreadContextDescriptor contextDescriptor = captureThreadContext(defaultExecutor);
+            if (contextDescriptor != null)
+                action = new ContextualConsumer<>(contextDescriptor, action);
+        }
 
         if (JAVA8) {
             if (other instanceof ManagedCompletableFuture)
