diff --git a/old_product.java b/new_product.java
index e3bbdcb..da3248b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static boolean withinDateRange(long date, long compareTo, int dayRange) {
         // ms = dayRange x 24 hours/day x 60 min/hour x 60 sec/min x 1000 ms/sec
-        final long msRange = dayRange * 24L * 60L * 60L * 1000L;
+        final long msRange = dayRange * 24L * 60L * 60L;
         return (compareTo - date) < msRange;
     }
\ No newline at end of file
