diff --git a/old_test.java b/new_test.java
index b178e24..02186d7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,14 +2,14 @@
     public void testWithinDateRange() {
         Calendar c = Calendar.getInstance();
 
-        long current = c.getTimeInMillis();
-        long lastRun = c.getTimeInMillis() - (3 * (1000 * 60 * 60 * 24));
+        long current = c.getTimeInMillis() / 1000;
+        long lastRun = current - (3 * (60 * 60 * 24));
         int range = 7; // 7 days
         boolean expResult = true;
         boolean result = DateUtil.withinDateRange(lastRun, current, range);
         assertEquals(expResult, result);
 
-        lastRun = c.getTimeInMillis() - (8 * (1000 * 60 * 60 * 24));
+        lastRun = c.getTimeInMillis() / 1000 - (8 * (60 * 60 * 24));
         expResult = false;
         result = DateUtil.withinDateRange(lastRun, current, range);
         assertEquals(expResult, result);
