diff --git a/old_product.java b/new_product.java
index b92fb72..677d0b5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,3 @@
-protected static String get(final String root, final String query) throws IOException {
-    final URL url = new URL(root + query);
-    final HttpURLConnection conn = (HttpURLConnection) url.openConnection();
-    try {
-      return read(new BufferInput(conn.getInputStream())).replaceAll("\r?\n *", "");
-    } catch(final IOException ex) {
-      throw error(conn, ex);
-    } finally {
-      conn.disconnect();
-    }
+protected static String get(final String query) throws IOException {
+    return request(query, GET);
   }
\ No newline at end of file
