diff --git a/old_product.java b/new_product.java
index 2975a1c..e808222 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ static Iter search(final Data data, final Value terms, final TokenMap map,
       final StandardFunc fun, final QueryContext ctx) throws QueryException {
 
     final InputInfo info = fun.info;
-    final IndexContext ic = new IndexContext(ctx, data, null, true);
+    final IndexContext ic = new IndexContext(data, false);
     if(!data.meta.ftxtindex) BXDB_INDEX.thrw(info, data.meta.name,
         IndexType.FULLTEXT.toString().toLowerCase(Locale.ENGLISH));
 
@@ -28,7 +28,7 @@ static Iter search(final Data data, final Value terms, final TokenMap map,
     }
 
     ctx.ftOpt(opt);
-    final FTWords words = new FTWords(info, ic.data, terms, m, ctx);
+    final FTWords words = new FTWords(info, ic, terms, m, ctx);
     ctx.ftOpt(tmp);
-    return new FTIndexAccess(info, words, ic.data.meta.name, ic.iterable).iter(ctx);
+    return new FTIndexAccess(info, words, ic).iter(ctx);
   }
\ No newline at end of file
