diff --git a/old_test.java b/new_test.java
index 0c46cd9..9f30189 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,10 +10,9 @@
     // apply index options to query term
     new Set(Prop.STEMMING, true).execute(context);
     new CreateIndex("fulltext").execute(context);
-    contains(_FT_SEARCH.args(NAME, "Exercises") + "/..",
-        "<li>Exercise 1</li>");
+    contains(_FT_SEARCH.args(NAME, "Exercises") + "/..", "<li>Exercise 1</li>");
     new Set(Prop.STEMMING, false).execute(context);
-    new CreateIndex("fulltext").execute(context);
+    new CreateIndex(CmdIndex.FULLTEXT).execute(context);
 
     // check match options
     query(_FT_SEARCH.args(NAME, "Assignments", " map {}"), "Assignments");
