diff --git a/old_product.java b/new_product.java
index 3f0c53f..03e230b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-public Token createToken(String tokenType, Map<String, Object> tokenData) throws TokenCreationFailedException {
+@Override
+    public Token createToken(String tokenType, Map<String, Object> tokenData) throws TokenCreationFailedException {
         try {
             TokenService tokenService = getTokenServiceForType(tokenType);
             return tokenService.createToken(tokenData);
