diff --git a/old_test.java b/new_test.java
index a0fac85..a3901cd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,15 @@
 @Test
   public void tokens() {
+    final Function func = _FT_TOKENS;
     execute(new CreateIndex(IndexType.FULLTEXT));
 
-    String entries = _FT_TOKENS.args(NAME);
+    String entries = func.args(NAME);
     query("count(" + entries + ')', 7);
     query("exists(" + entries + "/self::entry)", true);
     query(entries + "/@count = 1", true);
     query(entries + "/@count = 2", true);
     query(entries + "/@count = 3", false);
 
-    entries = _FT_TOKENS.args(NAME, "a");
+    entries = func.args(NAME, "a");
     query("count(" + entries + ')', 2);
   }
\ No newline at end of file
