diff --git a/old_test.java b/new_test.java
index 80da829..dd9da28 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
     public void testGetConnectionString() throws Exception {
-        String value = Settings.getConnectionString(Settings.KEYS.DB_CONNECTION_STRING, Settings.KEYS.DB_FILE_NAME);
+        String value = getSettings().getConnectionString(Settings.KEYS.DB_CONNECTION_STRING, Settings.KEYS.DB_FILE_NAME);
         Assert.assertNotNull(value);
         String msg = null;
         try {
-            value = Settings.getConnectionString("invalidKey", null);
+            value = getSettings().getConnectionString("invalidKey", null);
         } catch (InvalidSettingException e) {
             msg = e.getMessage();
         }
