diff --git a/old_product.java b/new_product.java
index 5dfc5e0..7d36f6d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public static String getConnectionString(String connectionStringKey, String dbFileNameKey)
+public String getConnectionString(String connectionStringKey, String dbFileNameKey)
             throws IOException, InvalidSettingException {
-        final String connStr = Settings.getString(connectionStringKey);
+        final String connStr = getString(connectionStringKey);
         if (connStr == null) {
             final String msg = String.format("Invalid properties file; %s is missing.", connectionStringKey);
             throw new InvalidSettingException(msg);
@@ -9,7 +9,7 @@ public static String getConnectionString(String connectionStringKey, String dbFi
             final File directory = getDataDirectory();
             String fileName = null;
             if (dbFileNameKey != null) {
-                fileName = Settings.getString(dbFileNameKey);
+                fileName = getString(dbFileNameKey);
             }
             if (fileName == null) {
                 final String msg = String.format("Invalid properties file to get a file based connection string; '%s' must be defined.",
