diff --git a/old_product.java b/new_product.java
index 60758f5..16228a1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,11 @@
-public StateHandle<T> add(String pathInZooKeeper, T state, CreateMode createMode) throws Exception {
+public StateHandle<T> add(
+			String pathInZooKeeper,
+			T state,
+			CreateMode createMode) throws Exception {
 		checkNotNull(pathInZooKeeper, "Path in ZooKeeper");
 		checkNotNull(state, "State");
 
-		// Create the state handle. Nothing persisted yet.
-		StateHandle<T> stateHandle = stateHandleProvider.createStateHandle(state);
+		StateHandle<T> stateHandle = storage.store(state);
 
 		boolean success = false;
 
