diff --git a/old_test.java b/new_test.java
index 726fa84..a8cb1d4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,8 @@
 @Test
 	public void testAdd() throws Exception {
-		// Setup
-		LongStateHandleProvider stateHandleProvider = new LongStateHandleProvider();
-
-		ZooKeeperStateHandleStore<Long> store = new ZooKeeperStateHandleStore<>(
-				ZooKeeper.getClient(), stateHandleProvider);
+		LongStateStorage longStateStorage = new LongStateStorage();
+		ZooKeeperStateHandleStore<Long> store = new ZooKeeperStateHandleStore<Long>(
+				ZooKeeper.getClient(), longStateStorage);
 
 		// Config
 		final String pathInZooKeeper = "/testAdd";
@@ -15,8 +13,8 @@
 
 		// Verify
 		// State handle created
-		assertEquals(1, stateHandleProvider.getStateHandles().size());
-		assertEquals(state, stateHandleProvider.getStateHandles().get(0).getState(null));
+		assertEquals(1, store.getAll().size());
+		assertEquals(state, store.get(pathInZooKeeper).getState(null));
 
 		// Path created and is persistent
 		Stat stat = ZooKeeper.getClient().checkExists().forPath(pathInZooKeeper);
