diff --git a/old_test.java b/new_test.java
index a76c628..42a04eb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,22 +1,22 @@
 @Test
     public void testRecommend2() {
 
-        LongList recs = recommender.recommend(6, 4);
+        List<ScoredId> recs = recommender.recommend(6, 4);
         assertEquals(3, recs.size());
-        assertEquals(6, recs.getLong(0));
-        assertEquals(7, recs.getLong(1));
-        assertEquals(9, recs.getLong(2));
+        assertEquals(6, recs.get(0).getId());
+        assertEquals(7, recs.get(1).getId());
+        assertEquals(9, recs.get(2).getId());
 
         recs = recommender.recommend(6, 3);
         assertEquals(3, recs.size());
-        assertEquals(6, recs.getLong(0));
-        assertEquals(7, recs.getLong(1));
-        assertEquals(9, recs.getLong(2));
+        assertEquals(6, recs.get(0).getId());
+        assertEquals(7, recs.get(1).getId());
+        assertEquals(9, recs.get(2).getId());
 
         recs = recommender.recommend(6, 2);
         assertEquals(2, recs.size());
-        assertEquals(6, recs.getLong(0));
-        assertEquals(7, recs.getLong(1));
+        assertEquals(6, recs.get(0).getId());
+        assertEquals(7, recs.get(1).getId());
 
         recs = recommender.recommend(6, 1);
         assertEquals(1, recs.size());
@@ -27,7 +27,7 @@
 
         recs = recommender.recommend(6, -1);
         assertEquals(3, recs.size());
-        assertEquals(6, recs.getLong(0));
-        assertEquals(7, recs.getLong(1));
-        assertEquals(9, recs.getLong(2));
+        assertEquals(6, recs.get(0).getId());
+        assertEquals(7, recs.get(1).getId());
+        assertEquals(9, recs.get(2).getId());
     }
\ No newline at end of file
