diff --git a/old_product.java b/new_product.java
index 3423aa3..b4db4ca 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,11 +4,8 @@
     return RpcUtils.call(new RpcUtils.RpcCallableThrowsIOException<Long>() {
       @Override
       public Long call() throws AlluxioException, IOException {
-        String user = options.isSetOwner() ? options.getOwner() : "";
-        String group = options.isSetGroup() ? options.getGroup() : "";
-        short mode = options.isSetMode() ? options.getMode() : Constants.INVALID_MODE;
         return mWorker
-            .createUfsFile(sessionId, new AlluxioURI(ufsUri), new Permission(user, group, mode));
+            .createUfsFile(sessionId, new AlluxioURI(ufsUri), new CreateUfsFileOptions(options));
       }
     });
   }
\ No newline at end of file
