diff --git a/old_test.java b/new_test.java
index b323615..e50ee47 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void createUfsFile() throws Exception {
-    mManager.createFile(SESSION_ID, mUri, Permission.defaults());
+    mManager.createFile(SESSION_ID, mUri, CreateUfsFileOptions.defaults());
     Mockito.verify(mMockUfs).create(Mockito.contains(mUri.toString()),
         Mockito.any(CreateOptions.class));
     Mockito.verify(mMockUfs).connectFromWorker(Mockito.anyString());
