diff --git a/old_product.java b/new_product.java
index 376a19a..60c1e8c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,6 +11,15 @@
             return;
         }
 
+        try {
+            authenticate(createScopeRequest.getScopeName(), READ_UPDATE);
+        } catch (AuthenticationException e) {
+            log.warn("Create scope for {} failed due to authentication failure {}.", createScopeRequest.getScopeName(), e);
+            asyncResponse.resume(Response.status(Status.UNAUTHORIZED).build());
+            LoggerHelpers.traceLeave(log, "createScope", traceId);
+            return;
+        }
+
         controllerService.createScope(createScopeRequest.getScopeName()).thenApply(scopeStatus -> {
             if (scopeStatus.getStatus() == CreateScopeStatus.Status.SUCCESS) {
                 log.info("Successfully created new scope: {}", createScopeRequest.getScopeName());
