diff --git a/old_test.java b/new_test.java
index a247e0f..273cfdd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,5 +2,5 @@
 	@Verifies(value = "should return all unvoided visits if includeEnded is set to true", method = "getVisits(Collection<VisitType>,Collection<Patient>,Collection<Location>,Collection<Concept>,Date,Date,Date,Date,boolean,boolean)")
 	public void getVisits_shouldReturnAllUnvoidedVisitsIfIncludeEndedIsSetToTrue() throws Exception {
 		executeDataSet(VISITS_WITH_DATES_XML);
-		Assert.assertEquals(13, dao.getVisits(null, null, null, null, null, null, null, null, true, false).size());
+		Assert.assertEquals(13, dao.getVisits(null, null, null, null, null, null, null, null, null, true, false).size());
 	}
\ No newline at end of file
