diff --git a/old_test.java b/new_test.java
index dce5277..21a3498 100644
--- a/old_test.java
+++ b/new_test.java
@@ -19,13 +19,8 @@
     testCases.add(new TestCase("1", TestClassB.class, new Class[] {int.class}, 1));
 
     for (TestCase testCase : testCases) {
-      try {
-        Object o =
-            CommonUtils.createNewClassInstance(testCase.mCls, testCase.mCtorClassArgs,
-                testCase.mCtorArgs);
-        assertEquals(o.toString(), testCase.mExpected);
-      } catch (Exception e) {
-        e.printStackTrace();
-      }
+      Object o = CommonUtils.createNewClassInstance(testCase.mCls, testCase.mCtorClassArgs,
+          testCase.mCtorArgs);
+      assertEquals(o.toString(), testCase.mExpected);
     }
   }
\ No newline at end of file
