diff --git a/old_product.java b/new_product.java
index 94670fc..371c92f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,14 @@
 public static <T> T createNewClassInstance(Class<T> cls, Class<?>[] ctorClassArgs,
-      Object[] ctorArgs) throws InstantiationException, IllegalAccessException,
-      NoSuchMethodException, SecurityException, InvocationTargetException {
-    if (ctorClassArgs == null) {
-      return cls.newInstance();
+      Object[] ctorArgs) {
+    try {
+      if (ctorClassArgs == null) {
+        return cls.newInstance();
+      }
+      Constructor<T> ctor = cls.getConstructor(ctorClassArgs);
+      return ctor.newInstance(ctorArgs);
+    } catch (InvocationTargetException e) {
+      throw new RuntimeException(e.getCause());
+    } catch (ReflectiveOperationException e) {
+      throw new RuntimeException(e);
     }
-    Constructor<T> ctor = cls.getConstructor(ctorClassArgs);
-    return ctor.newInstance(ctorArgs);
   }
\ No newline at end of file
