diff --git a/old_test.java b/new_test.java
index d198f80..b5aa5f6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void toMap(){
-        String json = "{'data1':{'name':'get'},'data2':{'name':'set'},'null':{'name':'set'}}";
+        String json = "{'data1':{'name':'get'},'data2':{'name':'set'}}";
         Map<String, Person> map = JsonUtil.toMap(json, Person.class);
         LOGGER.info(JsonUtil.format(map));
     }
\ No newline at end of file
