diff --git a/old_product.java b/new_product.java
index f7385cb..13a949b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,25 @@
-public static <T> Map<String, T> toMap(String json,Class<T> clazz,Map<String, Class<?>> classMap){
+@SuppressWarnings("unchecked")
+    public static <T> Map<String, T> toMap(String json,Class<T> clazz,Map<String, Class<?>> classMap){
         LOGGER.debug("in json:{}", json);
 
+        if (Validator.isNullOrEmpty(json)){
+            return Collections.emptyMap();
+        }
+
         Map<String, T> map = new HashMap<String, T>();
 
         JSONObject jsonObject = toJSONObject(json);
-        @SuppressWarnings("unchecked")
         Iterator<String> keys = jsonObject.keys();
         while (keys.hasNext()){
             String key = keys.next();
             Object value = jsonObject.get(key);
             LOGGER.debug("key:[{}], value:{}", key, value);
-            map.put(key, toBean(value, clazz, classMap));
+
+            if (null != clazz){
+                map.put(key, toBean(value, clazz, classMap));
+            }else{//如果clazz是null,表示不需要转换
+                map.put(key, (T) value);
+            }
         }
         return map;
     }
\ No newline at end of file
