diff --git a/old_test.java b/new_test.java
index 3bd397c..310088a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -21,12 +21,17 @@
             StormMetricsRegistry metricRegistry = new StormMetricsRegistry();
             WorkerLogs stubbedWorkerLogs = new WorkerLogs(conf, null, metricRegistry) {
                 @Override
-                public Map<String, File> identifyWorkerLogDirs(Set<File> logDirs) {
-                    Map<String, File> ret = new HashMap<>();
-                    ret.put("42", unexpectedDir1);
-                    ret.put("007", expectedDir2);
-                    // this tests a directory with no yaml file thus no worker id
-                    ret.put("", expectedDir3);
+                public SortedSet<File> getLogDirs(Set<File> logDirs, Predicate<String> predicate) {
+                    TreeSet<File> ret = new TreeSet<>();
+                    if (predicate.test("42")) {
+                        ret.add(unexpectedDir1);
+                    }
+                    if (predicate.test("007")) {
+                        ret.add(expectedDir2);
+                    }
+                    if(predicate.test("")) {
+                        ret.add(expectedDir3);
+                    }
 
                     return ret;
                 }
