diff --git a/old_product.java b/new_product.java
index 1f9aca2..74967f0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,7 +13,7 @@
 		visit.setLocation(encounter.getLocation());
 		visit.setPatient(encounter.getPatient());
 		
-		visit.setVisitType(getVisitType(encounter.getEncounterType()));
+		visit.setVisitType(getVisitType(encounter));
 		
 		//set stop date time to last millisecond of the encounter day.
 		visit.setStopDatetime(OpenmrsUtil.getLastMomentOfDay(encounter.getEncounterDatetime()));
