diff --git a/old_test.java b/new_test.java
index 127d527..097c7db 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,14 +18,6 @@
 		
 		Assert.assertNotNull(encounter.getVisit());
 		
-		//The visit needs to be persisted, else the assert below will throw
-		//org.hibernate.TransientObjectException: object references an unsaved transient 
-		//instance - save the transient instance before flushing: org.openmrs.Visit
-		Visit visit = encounter.getVisit();
-		encounter.setVisit(null);
-		Context.getVisitService().saveVisit(visit);
-		encounter.setVisit(visit);
-		
 		//should be set according to: 1:2 encounterTypeId:visitTypeId
 		Assert.assertEquals(1, encounter.getEncounterType().getEncounterTypeId().intValue());
 		Assert.assertEquals(Context.getVisitService().getVisitType(2), encounter.getVisit().getVisitType());
