diff --git a/old_test.java b/new_test.java
index 07e674f..55ff309 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,7 @@
 		DrugOrder orderToDiscontinue = (DrugOrder) Context.getOrderService().getOrder(111);
 		assertTrue(OrderUtilTest.isActiveOrder(orderToDiscontinue, null));
 		DrugOrder discontinuationOrder = new DrugOrder();
+		discontinuationOrder.setDosingType(null);
 		discontinuationOrder.setCareSetting(orderToDiscontinue.getCareSetting());
 		discontinuationOrder.setConcept(orderToDiscontinue.getConcept());
 		discontinuationOrder.setAction(Order.Action.DISCONTINUE);
