diff --git a/old_product.java b/new_product.java
index 704a4b6..8a25c83 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,8 +7,9 @@ public void validate(Object obj, Errors errors) {
 		} else {
 			// for the following elements Order.hbm.xml says: not-null="true"
 			ValidationUtils.rejectIfEmpty(errors, "asNeeded", "error.null");
-			ValidationUtils.rejectIfEmpty(errors, "dosingType", "error.null");
-			
+			if (order.getAction() != Order.Action.DISCONTINUE) {
+				ValidationUtils.rejectIfEmpty(errors, "dosingType", "error.null");
+			}
 			if (order.getDrug() != null)
 				ValidationUtils.rejectIfEmpty(errors, "drug.concept", "error.null");
 			
