diff --git a/old_product.java b/new_product.java
index 84f747f..eb2bc42 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,8 +3,8 @@ private void set(final IndexType type, final ValueIndex index) {
     switch(type) {
       case TEXT:      textIndex = index; break;
       case ATTRIBUTE: attrIndex = index; break;
-      case ATTTOKEN:  attrTokenIndex = index; break;
-      case FULLTEXT:  ftxtIndex = index; break;
+      case TOKEN:     tokenIndex = index; break;
+      case FULLTEXT:  ftIndex = index; break;
       default:        break;
     }
   }
\ No newline at end of file
