diff --git a/old_test.java b/new_test.java
index 52ed8bb..8169ae2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,9 @@
 @Test
-    public void testGetMem()
-    {
-        System.out.println("getMem");
-        long result = instance.getMem();
-        System.out.println(result);
-        assertTrue(result >= 0);
-    }
\ No newline at end of file
+   public void testGetMem()
+   {
+      System.out.println("getMem");
+      prepareTestData(123);
+      long result = instance.getMem();
+      System.out.println(result);
+      assertTrue(result >= 0);
+   }
\ No newline at end of file
