diff --git a/old_product.java b/new_product.java
index 783beca..deb8d5f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,18 +1,16 @@
-public long getMem() {
-        long ret;
+public long getMem()
+   {
+      long ret;
 
-        try
-        {
-            String value = getData("mem");
-            if(value != null) {
-                ret = Long.parseLong(value);
-            } else {
-                ret = MetricsGetter.AGENT_ERROR;
-            }
-        } catch (IOException ioex)
-        {
-            ret = MetricsGetter.AGENT_ERROR;
-        }
+      String value = getData("mem");
+      if (value != null)
+      {
+         ret = Long.parseLong(value);
+      }
+      else
+      {
+         ret = MetricsGetter.AGENT_ERROR;
+      }
 
-        return ret;
-    }
\ No newline at end of file
+      return ret;
+   }
\ No newline at end of file
