diff --git a/old_product.java b/new_product.java
index 220f257..d7e772b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,6 @@ private Item rename(final QueryContext ctx) throws QueryException {
     final Data data = data(0, ctx);
     final String src = path(1, ctx);
     final String trg = path(2, ctx);
-    if(!new IOFile(trg).isValid()) RESINV.thrw(input, trg);
 
     // the first step of the path should be the database name
     final IntList il = data.docs(src);
@@ -15,10 +14,8 @@ private Item rename(final QueryContext ctx) throws QueryException {
       ctx.updates.add(new ReplaceValue(pre, data, input, token(target)), ctx);
     }
     // rename files
-    final IOFile source = data.meta.binary(src);
-    final IOFile target = data.meta.binary(trg);
-    if(source != null && target != null)
-      ctx.updates.add(new DBRename(data, source, target, input), ctx);
+    if(data.meta.binary(src) != null && data.meta.binary(trg) != null)
+      ctx.updates.add(new DBRename(data, src, trg, input), ctx);
 
     return null;
   }
\ No newline at end of file
