diff --git a/old_test.java b/new_test.java
index 9bc352b..e037ac7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,5 @@
-@Test(expectedExceptions=NullPointerException.class)
-    public void factory_date_objects_nullYear() {
-        OffsetDate.date(null, MonthOfYear.JULY, DayOfMonth.dayOfMonth(15), OFFSET_PONE);
+@Test(dataProvider="sampleDates")
+    public void test_toLocalYear(int year, int month, int day, ZoneOffset offset) {
+        OffsetDate t = OffsetDate.date(year, month, day, offset);
+        assertEquals(t.toYear(), Year.isoYear(year));
     }
\ No newline at end of file
