diff --git a/old_product.java b/new_product.java
index 9c646ce..d9a8c64 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-@Test(expectedExceptions=NullPointerException.class)
-    public void factory_date_objects_nullYear() {
-        LocalDate.date(null, MonthOfYear.JULY, DayOfMonth.dayOfMonth(15));
+public void test_atOffset() {
+        LocalDate t = LocalDate.date(2008, 6, 30);
+        assertEquals(t.atOffset(OFFSET_PTWO), OffsetDate.date(2008, 6, 30, OFFSET_PTWO));
     }
\ No newline at end of file
