diff --git a/old_test.java b/new_test.java
index eddcb27..f2a5fd2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,11 +12,11 @@
     }
 
     // invalid names
-    error(_DB_COPY.args("x", " ''"), Err.BXDB_NAME);
-    error(_DB_COPY.args(" ''", "x"), Err.BXDB_NAME);
+    error(_DB_COPY.args("x", " ''"), Err.BXDB_NAME_X);
+    error(_DB_COPY.args(" ''", "x"), Err.BXDB_NAME_X);
 
     // same name is disallowed
-    error(_DB_COPY.args(NAME, NAME), Err.BXDB_SAME);
+    error(_DB_COPY.args(NAME, NAME), Err.BXDB_SAME_X);
     // source database does not exist
-    error(_DB_COPY.args(NAME + "xx", NAME), Err.BXDB_WHICH);
+    error(_DB_COPY.args(NAME + "xx", NAME), Err.BXDB_WHICH_X);
   }
\ No newline at end of file
