diff --git a/old_product.java b/new_product.java
index 0835f8e..b828a92 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,6 @@
-public Criteria or(String fieldname) {
-		return or(new SimpleField(fieldname));
+@SuppressWarnings("unchecked")
+	public Crotch or(String fieldname) {
+		Criteria node = new Criteria(fieldname);
+		node.setPartIsOr(true);
+		return or(node);
 	}
\ No newline at end of file
