diff --git a/old_test.java b/new_test.java
index 1b164a7..362f1e1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,9 @@
 @Test
-	public void testOr() {
-		Criteria criteria = new Criteria("field_1").startsWith("start").or("field_2").endsWith("end").startsWith("start2");
-		Assert.assertEquals(" OR ", criteria.getConjunctionOperator());
-		Assert.assertEquals(2, criteria.getCriteriaChain().size());
+	public void testCirteriasJoindWithOrShouldBeSiblingsOfCreatedCrotch() {
+
+		Criteria c1 = new Criteria("field_1").startsWith("start").endsWith("end");
+		Criteria c2 = new Criteria("field_2").startsWith("2start");
+		Crotch crotch = c1.or(c2);
+
+		Assert.assertThat(crotch.getSiblings(), IsIterableContainingInOrder.<Node> contains(c1, c2));
 	}
\ No newline at end of file
