diff --git a/old_test.java b/new_test.java
index aba8ad6..e0c6437 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,18 +2,18 @@
     public void testGetGrblStatusCapabilities() {
         System.out.println("getGrblStatusCapabilities");
         double version;
-        String letter;
+        Character letter;
         Capabilities expResult = new Capabilities();
         Capabilities result;
 
         version = 0.8;
-        letter = "c";
+        letter = 'c';
         result = GrblUtils.getGrblStatusCapabilities(version, letter);
         assertEquals(true, result.REAL_TIME);
         assertEquals(false, result.OVERRIDES);
         
         version = 0.8;
-        letter = "a";
+        letter = 'a';
         result = GrblUtils.getGrblStatusCapabilities(version, letter);
         assertEquals(false, result.REAL_TIME);
         assertEquals(false, result.OVERRIDES);
@@ -29,4 +29,12 @@
         result = GrblUtils.getGrblStatusCapabilities(version, letter);
         assertEquals(true, result.REAL_TIME);
         assertEquals(true, result.OVERRIDES);
+        assertEquals(false, result.V1_FORMAT);
+
+        version = 1.0;
+        letter = 'c';
+        result = GrblUtils.getGrblStatusCapabilities(version, letter);
+        assertEquals(true, result.REAL_TIME);
+        assertEquals(true, result.OVERRIDES);
+        assertEquals(true, result.V1_FORMAT);
     }
\ No newline at end of file
