diff --git a/old_test.java b/new_test.java
index 9a54030..86d6642 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test
 	@Verifies(value = "should escape underscore character in identifier phrase", method = "getPatients(String,String,List<QPatientIdentifierType;>,null)")
 	public void getPatients_shouldEscapeUnderscoreCharacterInIdentifierPhrase() throws Exception {
-		deleteAllData();
 		baseSetupWithStandardDataAndAuthentication();
 		Patient patient2 = pService.getPatient(2);
 		PatientIdentifier patientIdentifier = new PatientIdentifier("_567", pService.getPatientIdentifierType(5), Context
