diff --git a/old_product.java b/new_product.java
index 05abb0e..dc2cc6f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,7 +16,7 @@ private Item zipFile(final QueryContext ctx) throws QueryException {
       zos.close();
     } catch(final IOException ex) {
       ok = false;
-      ZIP_FAIL.thrw(info, ex.getMessage());
+      ZIP_FAIL.thrw(info, ex);
     } finally {
       if(fos != null) {
         try { fos.close(); } catch(final IOException ex) { }
