diff --git a/old_product.java b/new_product.java
index 8a18b2c..4662bec 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public void delete(OperationContext context,
                      String account, String application,
                      String type, String name)
-      throws MetaDataException;
\ No newline at end of file
+      throws OperationException;
\ No newline at end of file
