diff --git a/old_test.java b/new_test.java
index db130b6..ba92147 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,6 @@
         method = "equals",
         args = {java.lang.Object.class}
     )
-    @KnownFailure("AccessControlContext.equals() doesn't compare the DomainCombiner")
     public void test_equals() {
         final Permission perm1 = new PropertyPermission("java.class.path",
                 "read");
