diff --git a/old_test.java b/new_test.java
index 4a88544..55d39cb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,7 @@
             mCallback.resetLocked();
             mSession.setShuffleModeEnabled(shuffleModeEnabled);
             mWaitLock.wait(TIME_OUT_MS);
-            assertTrue(mCallback.mOnShuffleModeChangedCalled);
+            assertTrue(mCallback.mOnShuffleModeChangedDeprecatedCalled);
             assertEquals(shuffleModeEnabled, mCallback.mShuffleModeEnabled);
             assertEquals(shuffleModeEnabled, controller.isShuffleModeEnabled());
         }
