diff --git a/old_test.java b/new_test.java
index a516a35..3306ec0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testGetSampleNumbers() throws Exception {
+    public void testGetSampleNumbers() {
         assertEquals(10, RDDSampleUtils.getSampleNumbers(2, 10,-1));
         assertEquals(100, RDDSampleUtils.getSampleNumbers(2, 100,-1));
         assertEquals(10, RDDSampleUtils.getSampleNumbers(5, 1000,-1));
@@ -7,4 +7,6 @@
         assertEquals(100, RDDSampleUtils.getSampleNumbers(5, 10001,-1));
         assertEquals(1000, RDDSampleUtils.getSampleNumbers(5, 100011,-1));
         assertEquals(99, RDDSampleUtils.getSampleNumbers(6, 100011,99));
+        assertEquals(999, RDDSampleUtils.getSampleNumbers(20, 999,-1));
+        assertEquals(40, RDDSampleUtils.getSampleNumbers(20, 1000,-1));
     }
\ No newline at end of file
