diff --git a/old_test.java b/new_test.java
index 0ba0e39..ec5de73 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,7 +12,9 @@
         Mockito.when(engine.eval(Mockito.anyString(), Mockito.any(List.class))).thenReturn(new Callable<TransformResponse>() {
             @Override
             public TransformResponse call() throws Exception {
-                return new TransformResponse();
+                TransformResponse response = new TransformResponse();
+                response.setStatus(TransformResponse.Status.SUCCESS);
+                return response;
             }
         });
         Mockito.when(engine.getSparkContext()).thenReturn(Mockito.mock(SparkContext.class));
@@ -33,7 +35,7 @@
             service.stopAsync();
         }
 
-        Assert.assertEquals(response.getStatus(), TransformResponse.Status.PENDING);
+        Assert.assertEquals(TransformResponse.Status.SUCCESS, response.getStatus());
 
         // Test eval arguments
         ArgumentCaptor<String> evalScript = ArgumentCaptor.forClass(String.class);
@@ -50,5 +52,5 @@
         Assert.assertEquals("spark_shell_temp", bindings.get(0).value());
         Assert.assertEquals("tableName", bindings.get(1).name());
         Assert.assertEquals("String", bindings.get(1).tpe());
-        Assert.assertEquals(response.getTable(), bindings.get(1).value());
+        Assert.assertTrue(((String)bindings.get(1).value()).matches("^[0-9a-f]{32}$"));
     }
\ No newline at end of file
