diff --git a/old_product.java b/new_product.java
index 034e805..ae30b2a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,7 @@
 @Override
 	public Double weighConceptWord(ConceptWord word) {
-		//TODO Add unit tests
 		Double weight = 0.0;
-		String conceptName = word.getConceptName().getName();
+		String conceptName = word.getConceptName().getName().toUpperCase();
 		String wordString = word.getWord();
 		//why is this the case, this seems like invalid data
 		if (conceptName.indexOf(wordString) < 0)
@@ -43,6 +42,6 @@
 			weight += computeBonusWeight(weightCoefficient, word);
 		}
 		
-		//round off to 2 decimal places
+		//round off to 3 decimal places
 		return Double.parseDouble(new DecimalFormat("0.000").format(weight));
 	}
\ No newline at end of file
