diff --git a/old_test.java b/new_test.java
index 6468fd6..5368b71 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,10 +4,10 @@
 		Concept c = new Concept();
 		
 		ConceptName synonymName = new ConceptName("my depot", Locale.ENGLISH);
-		ConceptWord synonymWord = new ConceptWord("my", c, synonymName, Locale.ENGLISH);
+		ConceptWord synonymWord = new ConceptWord("MY", c, synonymName, Locale.ENGLISH);
 		
 		ConceptName shortName = new ConceptName("my depot", Locale.ENGLISH);
-		ConceptWord shortWord = new ConceptWord("my", c, shortName, Locale.ENGLISH);
+		ConceptWord shortWord = new ConceptWord("MY", c, shortName, Locale.ENGLISH);
 		shortName.setConceptNameType(ConceptNameType.SHORT);
 		
 		Assert.assertTrue("A word for a synonym should weigh higher than that of a short name", dao
