diff --git a/old_product.java b/new_product.java
index 25acd60..dc4032f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -24,7 +24,7 @@ public static String wrap(final String str, int wrapLength, String newLineStr, f
                     offset += matcher.end();
                     continue;
                 }else {
-                    spaceToWrapAt = matcher.start();
+                    spaceToWrapAt = matcher.start() + offset;
                 }
             }
 
