diff --git a/old_product.java b/new_product.java
index 7270922..bd1763b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,11 @@
 private static URL getResource(ClassLoader classLoader,String resourceName){
         Validate.notNull(classLoader, "classLoader can't be null!");
         Validate.notNull(resourceName, "resourceName can't be null!");
-        return classLoader.getResource(resourceName);
+
+        boolean startsWithSlash = resourceName.startsWith("/");
+        String usePath = startsWithSlash ? StringUtil.substring(resourceName, 1) : resourceName;
+        URL result = classLoader.getResource(usePath);
+
+        LOGGER.info("search resource:[\"{}\"] in [{}],result:[{}]", resourceName, classLoader, result);
+        return result;
     }
\ No newline at end of file
