diff --git a/old_product.java b/new_product.java
index 5e6f342..f867cc9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public void restore(Date startTime, Date endTime) throws Exception
     {
         // Stop cassandra if its running and restoring all keyspaces
         if (config.getRestoreKeySpaces().size() == 0)
-            SystemUtils.stopCassandra(config);
+            cassProcess.stop();
 
         // Cleanup local data
         SystemUtils.cleanupDir(config.getDataFileLocation(), config.getRestoreKeySpaces());
