diff --git a/old_test.java b/new_test.java
index de5038f..9fa8de2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
     public void testEqProp() {
-    	assertEquals(new EqPropCriterion("id", "name"), instance.eqProp("id", "name").getQueryCriterion());
+    	assertEquals(Criteria.eqProp("id", "name"), instance.eqProp("id", "name").getQueryCriterion());
     }
\ No newline at end of file
