diff --git a/old_product.java b/new_product.java
index 23df243..c7e9bee 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery eqProp(String propName, String otherProp) {
-        criterion = criterion.and(criterionBuilder.eqProp(propName, otherProp));
+        criterion = criterion.and(Criteria.eqProp(propName, otherProp));
         return this;
     }
\ No newline at end of file
