diff --git a/old_product.java b/new_product.java
index bb86492..59b6cde 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,10 +4,10 @@ public static Rectangle2D_I32 boundBox( int srcWidth , int srcHeight ,
 	{
 		Rectangle2D_I32 ret = boundBox(srcWidth,srcHeight,transform);
 
-		int x0 = ret.tl_x;
-		int y0 = ret.tl_y;
-		int x1 = ret.tl_x + ret.width;
-		int y1 = ret.tl_y + ret.height;
+		int x0 = ret.x0;
+		int y0 = ret.y0;
+		int x1 = ret.x0 + ret.width;
+		int y1 = ret.y0 + ret.height;
 
 		if( x0 < 0 ) x0 = 0;
 		if( x1 > dstWidth) x1 = dstWidth;
