diff --git a/old_test.java b/new_test.java
index fc90d6c..c922803 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
-    public void test_withChrono() {
+    public void test_withChronology() {
         DateTimeFormatter test = fmt;
-        assertEquals(test.getChrono(), null);
-        test = test.withChrono(ISOChrono.INSTANCE);
-        assertEquals(test.getChrono(), ISOChrono.INSTANCE);
-        test = test.withChrono(null);
-        assertEquals(test.getChrono(), null);
+        assertEquals(test.getChronology(), null);
+        test = test.withChronology(IsoChronology.INSTANCE);
+        assertEquals(test.getChronology(), IsoChronology.INSTANCE);
+        test = test.withChronology(null);
+        assertEquals(test.getChronology(), null);
     }
\ No newline at end of file
