diff --git a/old_product.java b/new_product.java
index 16fcc19..f1eecf4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @Override
-    public UsagePerAppBean getUsagePerApp(String organizationId, String apiId, String version,
+    public UsagePerClientBean getUsagePerClient(String organizationId, String apiId, String version,
             DateTime from, DateTime to) {
-        UsagePerAppBean rval = new UsagePerAppBean();
+        UsagePerClientBean rval = new UsagePerClientBean();
         return rval;
     }
\ No newline at end of file
