diff --git a/old_test.java b/new_test.java
index ac81898..5024505 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,7 @@
 		MockHttpServletRequest request = new MockHttpServletRequest("POST", "");
 		request.setParameter("locationId", "1");
 		request.setParameter("retireReason", "");
+		request.setParameter("retired", "true");
 		request.setParameter("retireLocation", "true");
 		
 		HttpServletResponse response = new MockHttpServletResponse();
@@ -14,5 +15,5 @@
 		
 		// make sure an error is returned because of the empty retire reason
 		BeanPropertyBindingResult bindingResult = (BeanPropertyBindingResult)modelAndView.getModel().get("org.springframework.validation.BindingResult.location");
-		Assert.assertEquals("general.retiredReason.empty", bindingResult.getGlobalError().getDefaultMessage());
+		Assert.assertTrue(bindingResult.hasFieldErrors("retireReason"));
 	}
\ No newline at end of file
