diff --git a/old_product.java b/new_product.java
index d81a3bc..0155432 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,13 +16,6 @@ protected ModelAndView onSubmit(HttpServletRequest request, HttpServletResponse
 				}
 				//the 'retire this location' button was clicked
 				else if (request.getParameter("retireLocation") != null) {
-					//make sure a reason was filled in for retiring the location
-					// this isn't done in the validator because we only want it checked if the user
-					// clicked the "retired location" button
-					if (!StringUtils.hasLength(location.getRetireReason())) {
-						errors.reject("retireReason", "general.retiredReason.empty");
-						return showForm(request, response, errors);
-					}
 					locationService.retireLocation(location, location.getRetireReason());
 					httpSession.setAttribute(WebConstants.OPENMRS_MSG_ATTR, "Location.retired");
 				}
