diff --git a/old_product.java b/new_product.java
index 7aaea71..9150276 100644
--- a/old_product.java
+++ b/new_product.java
@@ -24,7 +24,7 @@ public RealmResults<E> findAllAsync() {
 
         final WeakReference<RealmResults<? extends RealmModel>> weakRealmResults = realm.handlerController.addToAsyncRealmResults(realmResults, this);
 
-        final Future<Long> pendingQuery = Realm.asyncQueryExecutor.submit(new Callable<Long>() {
+        final Future<Long> pendingQuery = Realm.asyncTaskExecutor.submit(new Callable<Long>() {
             @Override
             public Long call() throws Exception {
                 if (!Thread.currentThread().isInterrupted()) {
