diff --git a/old_product.java b/new_product.java
index 3e49ece..3ceca93 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,3 @@
 public long getLastAccessedTime() {
-        if (lastAccessedTime == 0)
-            return nativegetcreationtime();
-        else
-            return lastAccessedTime;
+        return (lastAccessedTime == 0) ? getCreationTime() : lastAccessedTime;
     }
\ No newline at end of file
