diff --git a/old_test.java b/new_test.java
index 28529ac..5ad037b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public void optimize() throws BaseXException {
     query(_DB_OPTIMIZE.args(NAME));
     query(_DB_OPTIMIZE.args(NAME));
-    error(_DB_OPTIMIZE.args(NAME, true), Err.UPDBOPTERR);
+    error(_DB_OPTIMIZE.args(NAME, true), Err.UPDBOPTERR_X);
     new Close().execute(context);
     query(_DB_OPTIMIZE.args(NAME, true));
 
@@ -24,11 +24,11 @@
     }
     assertEquals(context.options.get(MainOptions.TEXTINDEX), true);
 
-    error(_DB_OPTIMIZE.args(NAME, false, " map { 'xyz': 'abc' }"), Err.BASX_OPTIONS);
-    error(_DB_OPTIMIZE.args(NAME, false, " map { 'updindex': 1 }"), Err.BASX_OPTIONS);
-    error(_DB_OPTIMIZE.args(NAME, false, " map { 'maxlen': -1 }"), Err.BASX_VALUE);
-    error(_DB_OPTIMIZE.args(NAME, false, " map { 'maxlen': 'a' }"), Err.BASX_VALUE);
-    error(_DB_OPTIMIZE.args(NAME, false, " map { 'textindex':'nope' }"), Err.BASX_VALUE);
+    error(_DB_OPTIMIZE.args(NAME, false, " map { 'xyz': 'abc' }"), Err.BASX_OPTIONS_X);
+    error(_DB_OPTIMIZE.args(NAME, false, " map { 'updindex': 1 }"), Err.BASX_OPTIONS_X);
+    error(_DB_OPTIMIZE.args(NAME, false, " map { 'maxlen': -1 }"), Err.BASX_VALUE_X_X);
+    error(_DB_OPTIMIZE.args(NAME, false, " map { 'maxlen': 'a' }"), Err.BASX_VALUE_X_X);
+    error(_DB_OPTIMIZE.args(NAME, false, " map { 'textindex':'nope' }"), Err.BASX_VALUE_X_X);
 
     // check if optimize call preserves original options
     query(_DB_OPTIMIZE.args(NAME));
