diff --git a/old_product.java b/new_product.java
index 41c8218..5bf4023 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,9 +6,19 @@
       return null;
     }
 
-    String dateAsString = json.getAsJsonPrimitive().getAsString().replaceAll("Z$", "+0000");
+    String dateAsString = json.getAsJsonPrimitive().getAsString();
     ParseException e = null;
 
+    if (dateAsString.endsWith("Z")) {
+      for (SimpleDateFormat format : iso8601Formatters) {
+        try {
+          return format.parse(dateAsString);
+        } catch (ParseException e1) {
+          e = e1;
+        }
+      }
+    }
+
     for (SimpleDateFormat format : dateFormatters) {
       try {
         return format.parse(dateAsString);
