diff --git a/old_test.java b/new_test.java
index 2910c3f..da83878 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,11 @@
 @Test(groups={"tck"})
-    public void test_with() {
-        TimeAdjuster timeAdjuster = new TimeAdjuster() {
-            public LocalTime adjustTime(LocalTime time) {
-                return LocalTime.of(23, 5);
+    public void test_with_adjustment() {
+        final LocalTime sample = LocalTime.of(23, 5);
+        DateTimeAdjuster adjuster = new DateTimeAdjuster() {
+            @Override
+            public DateTimeObject adjustCalendrical(DateTimeObject calendrical) {
+                return sample;
             }
         };
-        assertEquals(TEST_12_30_40_987654321.with(timeAdjuster), LocalTime.of(23, 5));
+        assertEquals(TEST_12_30_40_987654321.with(adjuster), sample);
     }
\ No newline at end of file
