diff --git a/old_test.java b/new_test.java
index 1e32292..65c5265 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,7 +16,7 @@
 		isLocalAddress = true;
 
 		Assert.assertEquals(
-				"LOCAL: addr1\naddr2\n121212 City1",
+				"LOCAL: addr1\naddr2\n121212 City1\nCountry1",
 				builder
 						.setLanguage("de_CH")
 						.buildAddressString(location, isLocalAddress, bPartnerBlock, userBlock));
