diff --git a/old_product.java b/new_product.java
index 0b7dc1e..50c5521 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public String buildAddressString(final I_C_Location location, boolean isLocalAddress, String bPartnerBlock, String userBlock)
 	{
-		final String displaySequence =  getDisplaySequence(location.getC_Country(), isLocalAddress);
+		final String displaySequence = getDisplaySequence(location.getC_Country(), isLocalAddress);
 
 		String inStr = displaySequence;
 		final StringBuilder outStr = new StringBuilder();
