diff --git a/old_test.java b/new_test.java
index 14736c7..ca6af58 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,10 +16,15 @@
 				assertNull(dataSourceNotExist);
 			}
 			
-			DataSource dataSourceExist = DataSource.getSystemDataSource(systemDataSourceIdExist);
+			DataSource dataSourceExist = null;
+			try {
+				dataSourceExist = DataSource.getSystemDataSource(systemDataSourceIdExist);
+			} catch (Exception e) {
+				fail("获取存在的系统数据源失败！");
+			}
 			assertNotNull(dataSourceExist);
 			
-		} catch (SQLException e) {
+		} catch (Exception e) {
 			assertFalse(true);
 		}
 	}
\ No newline at end of file
