diff --git a/old_test.java b/new_test.java
index 2215aee..8c9b943 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,8 @@
     public void testGenerateInitialTableAttributes() {
         val updates = IndexWriter.generateInitialTableAttributes();
         val values = updates.stream().collect(Collectors.toMap(AttributeUpdate::getAttributeId, AttributeUpdate::getValue));
-        Assert.assertEquals("Unexpected number of updates generated.", 1, values.size());
+        Assert.assertEquals("Unexpected number of updates generated.", 3, values.size());
         Assert.assertEquals("Unexpected value for TableIndexOffset.", 0L, (long) values.get(Attributes.TABLE_INDEX_OFFSET));
+        Assert.assertEquals("Unexpected value for TableEntryCount.", 0L, (long) values.get(Attributes.TABLE_ENTRY_COUNT));
+        Assert.assertEquals("Unexpected value for TableBucketCount.", 0L, (long) values.get(Attributes.TABLE_BUCKET_COUNT));
     }
\ No newline at end of file
