diff --git a/old_test.java b/new_test.java
index 2ddce6c..3a17508 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(description = "DELETE /catalogItem/{id}", dependsOnMethods = "testDeleteCatalogItemMetadataValue")
    public void testDeleteCatalogItem() {
-      catalogApi.deleteCatalogItem(catalogItem.getHref());
-      catalogItem = catalogApi.getCatalogItem(catalogItem.getHref());
+      catalogApi.deleteItem(catalogItem.getId());
+      catalogItem = catalogApi.getItem(catalogItem.getId());
       assertNull(catalogItem);
    }
\ No newline at end of file
