diff --git a/old_product.java b/new_product.java
index 6d0a414..4e90a9d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,7 @@
-@ContextDependent
-  @Requires(Permission.NONE)
-  public Item attribute(final Str key, final Item def) throws QueryException {
+@Requires(Permission.NONE)
+  public Item get(final Str key, final Item def) throws QueryException {
     final Object o = session().getAttribute(key.toJava());
     if(o == null) return def;
-    if(!(o instanceof Item)) BXSE_GET.thrw(null, Util.name(o));
-    return (Item) o;
+    if(o instanceof Item) return (Item) o;
+    throw BXSE_GET.thrw(null, Util.name(o));
   }
\ No newline at end of file
