diff --git a/old_test.java b/new_test.java
index 35059ab..68701d9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -19,8 +19,8 @@
     for (final String paragraph : text1) {
       paragraphs.add(new JsonPrimitive(paragraph));
     }
-    contentJson.add(LanguageTranslation.TEXT, paragraphs);
-    contentJson.addProperty(LanguageTranslation.MODEL_ID, modelId);
+    contentJson.add("text", paragraphs);
+    contentJson.addProperty("model_id", modelId);
     mockServer.when(request().withMethod("POST").withPath(LANGUAGE_TRANSLATION_PATH)
 
     .withBody(contentJson.toString())
@@ -33,11 +33,7 @@
     TranslationResult translationResult = service.translate(text, modelId);
     testTranslationResult(text, translationResult);
 
-    final Map<String, Object> params = new HashMap<String, Object>();
-    params.put(LanguageTranslation.TEXT, new String[] {text});
-    params.put(LanguageTranslation.MODEL_ID, modelId);
-
-    translationResult = service.translate(params);
+    translationResult = service.translate(text, modelId);
     testTranslationResult(text, translationResult);
     assertNotNull(service.toString());
   }
\ No newline at end of file
