diff --git a/old_product.java b/new_product.java
index d751481..4c2905d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,5 +14,8 @@ public String csvIdentifiers(Set<Identifier> ids) {
                 sb.append(id.getValue());
             }
         }
+        if (sb.length()==0) {
+            return "\"\"";
+        }
         return StringEscapeUtils.escapeCsv(sb.toString());
     }
\ No newline at end of file
