diff --git a/old_product.java b/new_product.java
index 8a75e9e..9fef61d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public BitString[] learn(int size, int generation, BiFunction<double[][], double[], Regression<double[]>> trainer, RegressionMeasure measure, double[][] x, double[] y, double[][] testx, double[] testy) {
+public BitString[] learn(int size, int generation, double[][] x, double[] y, double[][] testx, double[] testy, RegressionMeasure measure, BiFunction<double[][], double[], Regression<double[]>> trainer) {
         if (size <= 0) {
             throw new IllegalArgumentException("Invalid population size: " + size);
         }
