diff --git a/old_test.java b/new_test.java
index 3656ca6..72be39c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,7 +13,9 @@
     out.write(BufferUtils.getIncreasingByteArray(blockSize));
     out.close();
 
-    String localPath = mBlockWorkerServiceHandler.lockBlock(blockId, SESSION_ID).getBlockPath();
+    String localPath =
+        mBlockWorkerServiceHandler.lockBlock(blockId, SESSION_ID, new LockBlockTOptions())
+            .getBlockPath();
 
     // The local path should exist
     Assert.assertNotNull(localPath);
