diff --git a/old_product.java b/new_product.java
index 59ac464..ca5711e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-public LockBlockResult lockBlock(long blockId, long sessionId) throws alluxio.thrift.AlluxioTException, org.apache.thrift.TException;
\ No newline at end of file
+public LockBlockResult lockBlock(long blockId, long sessionId, LockBlockTOptions options) throws alluxio.thrift.AlluxioTException, org.apache.thrift.TException;
\ No newline at end of file
