diff --git a/old_test.java b/new_test.java
index ad56947..4bb2485 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,16 +6,15 @@
         TestContainerFactory factory = new TestContainerFactory();
         StreamSegmentContainerRegistry registry = new StreamSegmentContainerRegistry(factory, executor.get());
 
-        HashSet<String> expectedContainerIds = new HashSet<>();
+        HashSet<Integer> expectedContainerIds = new HashSet<>();
         Collection<CompletableFuture<ContainerHandle>> handleFutures = new ArrayList<>();
-        for (int i = 0; i < containerCount; i++) {
-            String containerId = getContainerId(i);
+        for (int containerId = 0; containerId < containerCount; containerId++) {
             handleFutures.add(registry.startContainer(containerId, TIMEOUT));
             expectedContainerIds.add(containerId);
         }
 
         Collection<ContainerHandle> handles = FutureHelpers.allOfWithResults(handleFutures).join();
-        HashSet<String> actualHandleIds = new HashSet<>();
+        HashSet<Integer> actualHandleIds = new HashSet<>();
         for (ContainerHandle handle : handles) {
             actualHandleIds.add(handle.getContainerId());
             SegmentContainer container = registry.getContainer(handle.getContainerId());
@@ -28,6 +27,6 @@
 
         AssertExtensions.assertThrows(
                 "getContainer did not throw when passed an invalid container id.",
-                () -> registry.getContainer("foo"),
+                () -> registry.getContainer(containerCount + 1),
                 ex -> ex instanceof ContainerNotFoundException);
     }
\ No newline at end of file
