diff --git a/old_test.java b/new_test.java
index 3b0548b..f56a955 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
-	public void removeGroupMember_shouldReturnTrueWhenAnObsIsRemoved() throws Exception {
+	public void removeGroupMember_shouldReturnDirtyFalseWhenAnObsIsRemoved() throws Exception {
 		Obs obs = new Obs(2);
 		Obs member = new Obs();
 		obs.addGroupMember(member);
-		assertTrue(obs.isDirty());
+		assertFalse(obs.isDirty());
 		resetObs(obs);
 		obs.removeGroupMember(member);
-		assertTrue(obs.isDirty());
+		assertFalse(obs.isDirty());
 	}
\ No newline at end of file
