diff --git a/old_test.java b/new_test.java
index 9265391..dc76712 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,13 +5,11 @@
 
 	int endPointer = handler.parseMessage(serverHelloDoneMsg, 0);
 	ServerHelloDoneMessage message = handler.getProtocolMessage();
-	HandshakeMessageFields handshakeMessageFields = message.getMessageFields();
 
 	assertNotNull("Confirm that parseMessage didn't return 'NULL'", endPointer);
 	assertEquals("Confirm expected message type: \"ServerHelloDone\"", HandshakeMessageType.SERVER_HELLO_DONE,
 		message.getHandshakeMessageType());
-	assertEquals("Confirm expected message length of \"0\"", new Integer(0), handshakeMessageFields.getLength()
-		.getValue());
+	assertEquals("Confirm expected message length of \"0\"", new Integer(0), message.getLength().getValue());
 	assertEquals("Confirm the correct value of endPointer representing the " + "actual number of message bytes",
 		serverHelloDoneMsg.length, endPointer);
     }
\ No newline at end of file
