diff --git a/old_test.java b/new_test.java
index 8bf4476..b3df70e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,19 @@
 @Test
-	public void test_distanceTo_one() {
-		final GeoPoint target = new GeoPoint(1.0, 1.0);
-		final GeoPoint other = new GeoPoint(1.0, 1.0);
-		assertEquals("distance to self is zero", 0, target.distanceTo(other));
+	public void test_distanceTo_Equator_Smaller() {
+		final double ratioDelta = 1E-5;
+		final int iterations = 10;
+		final double latitude = 0;
+		double longitudeIncrement = 1;
+		for (int i = 0 ; i < iterations ; i ++) {
+			final double longitude1 = getRandomLongitude();
+			final double longitude2 = longitude1 + longitudeIncrement;
+			longitudeIncrement /= 10.;
+			final GeoPoint target = new GeoPoint(latitude, longitude1);
+			final GeoPoint other = new GeoPoint(latitude, longitude2);
+			final double diff = getCleanLongitudeDiff(longitude1, longitude2);
+			final double expected = GeoConstants.RADIUS_EARTH_METERS * diff * MathConstants.DEG2RAD;
+			final double delta = expected * ratioDelta;
+			assertEquals("distance between " + target + " and " + other,
+					expected, target.distanceToAsDouble(other), delta);
+		}
 	}
\ No newline at end of file
