diff --git a/old_test.java b/new_test.java
index 0b48376..206a942 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,8 +6,8 @@
     StreamQueueConsumer consumer = new StreamQueueConsumer(0, 0, 1,
                                                           new QueueConfig(QueuePartitioner.PartitionerType.FIFO, true));
 
-    TTQueueOnVCTable table = new TTQueueOnVCTable( new MemoryOVCTable(Bytes.toBytes("TestStreamMemoryQueue")),
-                                                   Bytes.toBytes("TestStream"),oracle,conf);
+    TTQueueTableOnVCTable table = new TTQueueTableOnVCTable( new MemoryOVCTable(Bytes.toBytes("TestStreamMemoryQueue")),
+                                                             oracle,conf);
     OrderedVersionedColumnarTable metaTable = new MemoryOVCTable(Bytes.toBytes("StreamMeta"));
     StreamTable streamTable = new StreamTable(Bytes.toBytes("StreamTest"),table, metaTable, -1);
 
