diff --git a/old_product.java b/new_product.java
index 87aa905..aaf01d8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public void unack(StreamEntryPointer entryPointer, StreamQueueConsumer consumer, ReadPointer readPointer)
     throws  OperationException {
-    queue.unack(entryPointer,consumer,readPointer);
+    queue.unack(name,entryPointer,consumer,readPointer);
   }
\ No newline at end of file
