diff --git a/old_test.java b/new_test.java
index 0af66ea..fe6964b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -36,11 +36,14 @@
     manager.persistFile(fileId, blockIds);
 
     // verify file persisted
-    List<Long> persistedFiles = (List<Long>) Whitebox.getInternalState(manager, "mPersistedFiles");
-    Assert.assertEquals(Lists.newArrayList(fileId), persistedFiles);
+    Set<Long> persistedFiles = (Set<Long>) Whitebox.getInternalState(manager, "mPersistedFiles");
+    Assert.assertEquals(Sets.newHashSet(fileId), persistedFiles);
 
     // verify fastCopy called twice, once per block
     PowerMockito.verifyStatic(Mockito.times(2));
     BufferUtils.fastCopy(Mockito.any(ReadableByteChannel.class),
         Mockito.any(WritableByteChannel.class));
+
+    // verify the file is not needed for another persistence
+    Assert.assertFalse(manager.needPersistence(fileId));
   }
\ No newline at end of file
