diff --git a/old_product.java b/new_product.java
index 6226053..eba9d51 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,13 +6,13 @@
 		
 		Criteria criteria = getCurrentSession().createCriteria(Visit.class);
 		
-		if (CollectionUtils.isNotEmpty(visitTypes))
+		if (visitTypes != null)
 			criteria.add(Restrictions.in("visitType", visitTypes));
-		if (CollectionUtils.isNotEmpty(patients))
+		if (patients != null)
 			criteria.add(Restrictions.in("patient", patients));
-		if (CollectionUtils.isNotEmpty(locations))
+		if (locations != null)
 			criteria.add(Restrictions.in("location", locations));
-		if (CollectionUtils.isNotEmpty(indications))
+		if (indications != null)
 			criteria.add(Restrictions.in("indication", indications));
 		
 		if (minStartDatetime != null)
