diff --git a/old_test.java b/new_test.java
index 2021d9a..aeb23b7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,9 +2,9 @@
 	public void testGetSource() {
 		final Operator<?> fixture = new OpImpl(0);
 
-		final Operator<?>.Output result = fixture.getSource();
+		final JsonStream result = fixture.getSource();
 
 		assertNotNull(result);
-		assertEquals(0, result.getIndex());
+		assertEquals(0, result.getSource().getIndex());
 		assertSame(fixture.getOutput(0), result);
 	}
\ No newline at end of file
