diff --git a/old_test.java b/new_test.java
index 02d40ac..533bd22 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,7 +20,7 @@
             .withAmountOfTrees(5)
             .withFeaturesCountSelectionStrgy(x -> 2);
 
-        ModelsComposition mdl = trainer.fit(ignite, data, new ArraysVectorizer<Integer>().labeled(1));
+        ModelsComposition mdl = trainer.fit(ignite, data, new DoubleArrayVectorizer<Integer>().labeled(1));
 
         assertTrue(mdl.getPredictionsAggregator() instanceof MeanValuePredictionsAggregator);
         assertEquals(5, mdl.getModels().size());
