diff --git a/old_test.java b/new_test.java
index 6b9637a..375d6e8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,4 +8,7 @@
     Assert.assertEquals("alluxio", new AlluxioURI("alluxio://localhost/").getScheme());
     Assert.assertEquals("hdfs", new AlluxioURI("hdfs://localhost/").getScheme());
     Assert.assertEquals("glusterfs", new AlluxioURI("glusterfs://localhost/").getScheme());
+    Assert.assertEquals("scheme:part1", new AlluxioURI("scheme:part1://localhost/").getScheme());
+    Assert.assertEquals("scheme:part1:part2",
+        new AlluxioURI("scheme:part1:part2://localhost/").getScheme());
   }
\ No newline at end of file
