diff --git a/old_test.java b/new_test.java
index 88f11b8..1c736c8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,6 +12,8 @@
 		((MockHttpServletRequest) pageContext.getRequest()).addHeader("Referer", referer);
 		tag.setPageContext(pageContext);
 		tag.setAllPrivileges("Manage WhirleyGigs,Manage Thingamajigs");
+		String redirect = "/myRedirect.html";
+		tag.setRedirect(redirect);
 		
 		Assert.assertEquals(Tag.SKIP_PAGE, tag.doStartTag());
 		Object missingPrivilegesObj = pageContext.getAttribute(WebConstants.FOUND_MISSING_PRIVILEGES,
@@ -24,7 +26,7 @@
 		Assert.assertTrue(errorMsg.toString().indexOf("general.authentication.unableToViewPage") > -1);
 		Assert.assertTrue(errorMsg.toString().indexOf("general.authentication.accountHasNoPrivilege") > -1);
 		
-		Assert.assertEquals(referer, pageContext.getAttribute(WebConstants.DENIED_PAGE, PageContext.SESSION_SCOPE)
+		Assert.assertEquals(redirect, pageContext.getAttribute(WebConstants.DENIED_PAGE, PageContext.SESSION_SCOPE)
 		        .toString());
 		
 		Context.logout();
