diff --git a/old_product.java b/new_product.java
index ab23652..5039dbc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -51,11 +51,9 @@ public int doStartTag() {
 					
 					if (StringUtils.hasText(redirect)) {
 						httpSession.setAttribute(WebConstants.DENIED_PAGE, redirect);
-					} else if (request.getAttribute("javax.servlet.error.request_uri") != null) {
+					} else if (StringUtils.hasText(referer)) {
 						//This is not exactly correct all the time
-						Object requestedUrl = request.getAttribute("javax.servlet.error.request_uri");
-						if (StringUtils.hasText(requestedUrl.toString()))
-							httpSession.setAttribute(WebConstants.DENIED_PAGE, requestedUrl.toString());
+						httpSession.setAttribute(WebConstants.DENIED_PAGE, referer);
 					}
 				} else {
 					//Why would there be no missing privileges yet the hasPrivileges(String priv) returned false
