diff --git a/old_test.java b/new_test.java
index 9f76045..cd2dd9d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test
     public void testGetRapidDefaultReadNameRegexSplit() {
-        final OpticalDuplicateFinder opticalDuplicateFinder = new OpticalDuplicateFinder();
         for (int i = 1; i < 10; i++) {
-            doTestGetRapidDefaultReadNameRegexSplit((i <= 5) ? i : 5, opticalDuplicateFinder);
+            doTestGetRapidDefaultReadNameRegexSplit(i);
         }
     }
\ No newline at end of file
