diff --git a/old_product.java b/new_product.java
index 29473e2..f64dbad 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,18 +1,24 @@
 public static int getRapidDefaultReadNameRegexSplit(final String readName, final char delim, final int[] tokens) {
         int tokensIdx = 0;
         int prevIdx = 0;
+        int numFields = 1;
+        for (int i = 0; i < readName.length(); i++) {
+            if (readName.charAt(i) == delim) numFields++;
+        }
+        int startOffset = numFields - 2 - 1; // zero-based (ex. 7 -> 4, 5 -> 2)
+        if (startOffset < 0) return -1;
+        int endOffset = startOffset + 2; // zero-based
         for (int i = 0; i < readName.length(); i++) {
             if (readName.charAt(i) == delim) {
-                if (1 < tokensIdx && tokensIdx < 5)
-                    tokens[tokensIdx] = rapidParseInt(readName.substring(prevIdx, i)); // only fill in 2-4 inclusive
+                if (startOffset <= tokensIdx && tokensIdx <= endOffset)
+                    tokens[tokensIdx - startOffset] = rapidParseInt(readName.substring(prevIdx, i)); // only fill in 2-4 inclusive for 5 fields
                 tokensIdx++;
-                if (4 < tokensIdx) return tokensIdx; // early return, only consider the first five tokens
                 prevIdx = i + 1;
             }
         }
         if (prevIdx < readName.length()) {
-            if (1 < tokensIdx && tokensIdx < 5)
-                tokens[tokensIdx] = rapidParseInt(readName.substring(prevIdx, readName.length())); // only fill in 2-4 inclusive
+            if (startOffset <= tokensIdx && tokensIdx <= endOffset)
+                tokens[tokensIdx - startOffset] = rapidParseInt(readName.substring(prevIdx, readName.length())); // only fill in 2-4 inclusive
             tokensIdx++;
         }
         return tokensIdx;
