diff --git a/old_product.java b/new_product.java
index 4511b3c..7996ffe 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 @Override
-    public List<PointSegment> addCommand(String command) throws GcodeParserException {
+    public List<GcodeMeta> addCommand(String command) throws GcodeParserException {
         return addCommand(command, ++this.state.commandNumber);
     }
\ No newline at end of file
