diff --git a/old_product.java b/new_product.java
index ba4e515..cfe6a2a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public <M extends SparseMatrix> int[] getNeighborsND(M poolerMem, int columnIndex, int radius, boolean wrapAround) {
+public <M extends SparseMatrix> TIntArrayList getNeighborsND(M poolerMem, int columnIndex, int radius, boolean wrapAround) {
 		int[] columnCoords = poolerMem.computeCoordinates(columnIndex);
 		List<int[]> dimensionCoords = new ArrayList<int[]>();
 		for(int i = 0;i < inputDimensions.length;i++) {
@@ -17,11 +17,11 @@ public <M extends SparseMatrix> int[] getNeighborsND(M poolerMem, int columnInde
 		}
 		
 		List<TIntList> neighborList = ArrayUtils.dimensionsToCoordinateList(dimensionCoords);
-		TIntList neighbors = new TIntArrayList(neighborList.size());
+		TIntArrayList neighbors = new TIntArrayList(neighborList.size());
 		for(int i = 0;i < neighborList.size();i++) {
 			int flatIndex = poolerMem.computeIndex(neighborList.get(i).toArray());
 			if(flatIndex == columnIndex) continue;
 			neighbors.add(flatIndex);
 		}
-		return neighbors.toArray();
+		return neighbors;
 	}
\ No newline at end of file
