diff --git a/old_test.java b/new_test.java
index ad5bfdd..afec59b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,5 +3,5 @@
       throws IOException, JSONException, RestServiceException, URISyntaxException {
     AdminBlobStorageService adminBlobStorageService = getAdminBlobStorageService();
     MessageInfo messageInfo = createMessageInfo(RestMethod.DELETE, "/", new JSONObject());
-    adminBlobStorageService.handleMessage(messageInfo);
+    adminBlobStorageService.handleDelete(messageInfo);
   }
\ No newline at end of file
