diff --git a/old_test.java b/new_test.java
index 3883df3..3b4590e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,8 @@
 		ConceptService cs = Context.getConceptService();
 		Concept concept = new Concept();
 		concept.addName(new ConceptName("test name", Context.getLocale()));
+		concept.setConceptClass(new ConceptClass());
+		concept.setDatatype(new ConceptDatatype());
 		ConceptMap map = new ConceptMap();
 		map.setSourceCode("unique code");
 		map.setSource(cs.getConceptSource(1));
