diff --git a/old_product.java b/new_product.java
index b576107..1c4c28c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -54,7 +54,6 @@ public void validate(Object obj, Errors errors) {
 					errors.rejectValue("username", "error.username.pattern");
 				}
 			}
-			ValidateUtil.validateFieldLengths(errors, obj.getClass(), "username", "systemId", "retireReason",
-			    "secretQuestion");
+			ValidateUtil.validateFieldLengths(errors, obj.getClass(), "username", "systemId", "retireReason");
 		}
 	}
\ No newline at end of file
