diff --git a/old_test.java b/new_test.java
index 20c7265..a3ce4db 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,8 +8,13 @@
         final ArgumentCaptor<JobEntity> argument = ArgumentCaptor.forClass(JobEntity.class);
 
         Mockito.when(this.jobRequestRepo.findOne(Mockito.eq(JOB_1_ID))).thenReturn(jobRequestEntity);
-
-        this.jobPersistenceService.createJob(job);
+        final JobExecution jobExecution = new JobExecution.Builder(
+            HOST_NAME,
+            PROCESS_ID,
+            CHECK_DELAY,
+            TIMEOUT
+        ).build();
+        this.jobPersistenceService.createJobAndJobExecution(job, jobExecution);
 
         Mockito.verify(jobRequestEntity).setJob(argument.capture());
         Assert.assertEquals(JOB_1_NAME, argument.getValue().getName());
@@ -17,4 +22,5 @@
         Assert.assertEquals(JOB_1_VERSION, argument.getValue().getVersion());
         Assert.assertEquals(JOB_1_ID, argument.getValue().getId());
         Assert.assertEquals(JOB_1_COMMAND_ARGS, argument.getValue().getCommandArgs());
+        Assert.assertNotNull(argument.getValue().getExecution());
     }
\ No newline at end of file
