diff --git a/old_product.java b/new_product.java
index 7cbf290..d4c8cd8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,9 @@
 @Override
-    public void createJob(
+    public void createJobAndJobExecution(
         @NotNull(message = "No Job provided to create")
-        final Job job
+        final Job job,
+        @NotNull(message = "No Job execution information provided. Unable to create.")
+        final JobExecution jobExecution
     ) throws GenieException {
         log.debug("Called with job: {}", job);
 
@@ -17,13 +19,11 @@
         }
 
         final JobRequestEntity jobRequestEntity = this.jobRequestRepo.findOne(job.getId());
-
         if (jobRequestEntity == null) {
             throw new GeniePreconditionException("Cannot find the job request for the id of the job specified.");
         }
 
         final JobEntity jobEntity = new JobEntity();
-
         jobEntity.setId(job.getId());
         jobEntity.setName(job.getName());
         jobEntity.setUser(job.getUser());
@@ -39,5 +39,14 @@
         jobEntity.setTags(job.getTags());
         jobEntity.setCommandArgs(job.getCommandArgs());
 
+        final JobExecutionEntity jobExecutionEntity = new JobExecutionEntity();
+        jobExecutionEntity.setId(jobExecution.getId());
+        jobExecutionEntity.setHostName(jobExecution.getHostName());
+        jobExecutionEntity.setProcessId(jobExecution.getProcessId());
+        jobExecutionEntity.setCheckDelay(jobExecution.getCheckDelay());
+        jobExecutionEntity.setTimeout(jobExecution.getTimeout());
+
+        // Persist the entities
+        jobEntity.setExecution(jobExecutionEntity);
         jobRequestEntity.setJob(jobEntity);
     }
\ No newline at end of file
