diff --git a/old_product.java b/new_product.java
index cc82978..d2b2992 100644
--- a/old_product.java
+++ b/new_product.java
@@ -61,8 +61,9 @@ public void persistFile(long fileId, List<Long> blockIds) throws AlluxioExceptio
     outputStream.flush();
     outputChannel.close();
     outputStream.close();
+    UfsFileStatus ufsFileStatus = ufs.getFileStatus(dstPath);
     synchronized (mLock) {
       mPersistingInProgressFiles.remove(fileId);
-      mPersistedFiles.add(fileId);
+      mPersistedFilesInfo.put(fileId, ufsFileStatus);
     }
   }
\ No newline at end of file
