diff --git a/old_test.java b/new_test.java
index 0988f99..8871f95 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,8 @@
     writeFileWithBlocks(fileId, blockIds);
 
     // verify file persisted
-    assertEquals(Arrays.asList(fileId), mManager.getPersistedFiles());
+    FileDataManager.PersistedFilesInfo info = mManager.getPersistedFilesInfo();
+    assertEquals(Arrays.asList(fileId), info.idList());
 
     // verify fastCopy called twice, once per block
     PowerMockito.verifyStatic(Mockito.times(2));
