diff --git a/old_test.java b/new_test.java
index b7709c9..da965bd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-	@Verifies(value = "should return null if no locale match and exact equals true", method = "getName(Locale,null)")
-	public void getName_shouldReturnNullIfNoLocaleMatchAndExactEqualsTrue() throws Exception {
+	@Verifies(value = "return null if no names are found in locale given exact equals true", method = "getName(Locale,null)")
+	public void getName_shouldReturnNullIfNoNamesAreFoundInLocaleGivenExactEqualsTrue() throws Exception {
 		Locale nonMatchingNameLocale = new Locale("en", "US");
 		Locale localeToSearch = new Locale("en");
 		
