diff --git a/old_test.java b/new_test.java
index f90287d..bd3412b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
     public void testGetProfileSchema() throws Exception {
-        ProfileSchemaDto profileSchema = createProfileSchema();
+        EndpointProfileSchemaDto profileSchema = createProfileSchema();
         
-        ProfileSchemaDto storedProfileSchema = client.getProfileSchema(profileSchema.getId());
+        EndpointProfileSchemaDto storedProfileSchema = client.getProfileSchema(profileSchema.getId());
         
         Assert.assertNotNull(storedProfileSchema);
         assertProfileSchemasEquals(profileSchema, storedProfileSchema);
