diff --git a/old_product.java b/new_product.java
index e15fc97..53813ac 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @RequestMapping(value="profileSchema/{profileSchemaId}", method=RequestMethod.GET)
     @ResponseBody
-    public ProfileSchemaDto getProfileSchema(@PathVariable String profileSchemaId) throws KaaAdminServiceException {
+    public EndpointProfileSchemaDto getProfileSchema(@PathVariable String profileSchemaId) throws KaaAdminServiceException {
         return kaaAdminService.getProfileSchema(profileSchemaId);
     }
\ No newline at end of file
