diff --git a/old_test.java b/new_test.java
index e7eb835..eb818c1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
-	public void testLoadResources_ZipFile_partial_lazy() throws FileNotFoundException, IOException {
+	public void testLoadResources_ZipFile_partial_lazy() throws IOException {
 		// given
 		ZipFile zipFile = new ZipFile(testBookFilename);
 		
 		// when
-		Resources resources = ResourcesLoader.loadResources(zipFile, encoding, Arrays.asList(MediatypeService.CSS));
+		Resources resources = ResourcesLoader.loadResources(zipFile, encoding, Collections.singletonList(MediatypeService.CSS));
 		
 		// then
 		verifyResources(resources);
