diff --git a/old_product.java b/new_product.java
index 8e6fd4e..9d32e3f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@ public static Resources loadResources(ZipInputStream zipInputStream, String defa
 		do {
 			// get next valid zipEntry
 			zipEntry = getNextZipEntry(zipInputStream);
-			if((zipEntry == null) || (zipEntry == ERROR_ZIP_ENTRY) || zipEntry.isDirectory()) {
+			if((zipEntry == null) || zipEntry.isDirectory()) {
 				continue;
 			}
 			
