diff --git a/old_product.java b/new_product.java
index 1b3aafb..259f545 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,8 +7,9 @@
 	        @RequestParam(required = false, value = "confirm") String confirm,
 	        @RequestParam(required = false, value = "forcePassword") Boolean forcePassword,
 	        @RequestParam(required = false, value = "roleStrings") String[] roles,
-	        @RequestParam(required = false, value = "createNewPerson") String createNewPerson,
-	        @ModelAttribute("user") User user, BindingResult errors) {
+	        @RequestParam(required = false, value = "createNewPerson") String createNewPerson, 
+			@RequestParam(required = false, value = "providerCheckBox") String addToProviderTableOption,
+			@ModelAttribute("user") User user, BindingResult errors) {
 		
 		UserService us = Context.getUserService();
 		MessageSourceService mss = Context.getMessageSourceService();
@@ -149,6 +150,13 @@
 				us.changeQuestionAnswer(user, secretQuestion, secretAnswer);
 			}
 			
+			//Check if admin wants the person associated with the user to be added to the Provider Table
+			if(addToProviderTableOption != null) {
+				Provider provider = new Provider();
+				provider.setPerson(user.getPerson());
+				provider.setIdentifier(user.getSystemId());
+				Context.getProviderService().saveProvider(provider);
+			}
 			httpSession.setAttribute(WebConstants.OPENMRS_MSG_ATTR, "User.saved");
 		}
 		return "redirect:users.list";
