diff --git a/old_product.java b/new_product.java
index ba0f6b3..e346e2a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,18 +1,15 @@
 @Override
-    public KTypeOpenHashSet<KType> clone()
-    {
-        try
-        {
-            /* #if ($TemplateOptions.KTypeGeneric) */
-            @SuppressWarnings("unchecked")
-            /* #end */
-            KTypeOpenHashSet<KType> cloned = (KTypeOpenHashSet<KType>) super.clone();
-            cloned.keys = keys.clone();
-            cloned.allocated = allocated.clone();
-            return cloned;
-        }
-        catch (CloneNotSupportedException e)
-        {
-            throw new RuntimeException(e);
-        }
-    }
\ No newline at end of file
+  public KTypeOpenHashSet<KType> clone() {
+    try {
+      /* #if ($TemplateOptions.KTypeGeneric) */
+      @SuppressWarnings("unchecked")
+      /* #end */
+      KTypeOpenHashSet<KType> cloned = (KTypeOpenHashSet<KType>) super.clone();
+      cloned.keys = keys.clone();
+      cloned.allocated = allocated.clone();
+      cloned.orderMixer = orderMixer.clone();
+      return cloned;
+    } catch (CloneNotSupportedException e) {
+      throw new RuntimeException(e);
+    }
+  }
\ No newline at end of file
