diff --git a/old_test.java b/new_test.java
index 40537af..eeb9cf7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,7 @@
 @Test
     public void testSetTags() {
-        Assert.assertNull(this.c.getTags());
+        Assert.assertNotNull(this.c.getTags());
+        Assert.assertTrue(this.c.getTags().isEmpty());
         final Set<String> tags = new HashSet<>();
         tags.add("tag1");
         tags.add("tag2");
