diff --git a/old_product.java b/new_product.java
index c6ac4be..c8f43eb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,6 @@
-public void setTags(final Set<String> tags) {
-        this.tags = tags;
+protected void setTags(final Set<String> tags) {
+        this.tags.clear();
+        if (tags != null) {
+            this.tags.addAll(tags);
+        }
     }
\ No newline at end of file
