diff --git a/old_product.java b/new_product.java
index 7dc0fa6..3aef2c1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
-private static <T extends Collection<File>> T filter(IOFileFilter filter,
-            Iterable<File> files, T acceptedFiles) {
+private static <T extends Collection<File>> T filter(final IOFileFilter filter,
+            final Iterable<File> files, final T acceptedFiles) {
         if (filter == null) {
             throw new IllegalArgumentException("file filter is null");
         }
         if (files != null) {
-            for (File file : files) {
+            for (final File file : files) {
                 if (file == null) {
                     throw new IllegalArgumentException("file collection contains null");
                 }
