diff --git a/old_product.java b/new_product.java
index b6594c8..64d8125 100644
--- a/old_product.java
+++ b/new_product.java
@@ -45,6 +45,7 @@ public void badLSReq(Channel ch) throws Exception {
             //setting destination ip
             ddPacket.setDestinationIp(neighborIpAddr());
             setLastSentDdPacket(ddPacket);
-            ch.write(ddPacket);
+            byte[] messageToWrite = getMessage(ddPacket);
+            ch.write(messageToWrite);
         }
     }
\ No newline at end of file
