diff --git a/old_product.java b/new_product.java
index 3a0ca04..729503b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
-public void deleteTenant(String userId) throws Exception {
-        MultiValueMap<String, Object> params = new LinkedMultiValueMap<>();
-        params.add("userId", userId);
-        restTemplate.postForLocation(restTemplate.getUrl() + "delTenant", params);
-    }
\ No newline at end of file
+public List<UserDto> getAllTenantAdminsByTenantId(String tenantId) {
+    ResponseEntity<List<UserDto>> entity = restTemplate.exchange(
+        restTemplate.getUrl() + "admins/" + tenantId,
+        HttpMethod.GET, null, new ParameterizedTypeReference<List<UserDto>>() {});
+    return entity.getBody();
+  }
\ No newline at end of file
