diff --git a/old_product.java b/new_product.java
index b87475b..bed1d7a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,18 +4,16 @@ public void get(HttpServletRequest req, HttpServletResponse res, WebContext ctx)
 		String operation = req.getParameter(WebConstants.OPERATION);
         String msgType = "success";
         String msg = null;
-        if (Util.hasLength(operation)) {
-          if (TOGGLE_AUDIT.equalsIgnoreCase(operation)) {
-              if (getFf4j().isEnableAudit()) {
-                  getFf4j().audit(false);
-                  msg = "Audit is now DISABLED";
-                  LOGGER.info("Audit has been disabled");
-              } else {
-                  getFf4j().audit(true);
-                  msg = "Audit is now ENABLED";
-                  LOGGER.info("Audit has been enabled");
-              }
-          }
+        if (Util.hasLength(operation) && TOGGLE_AUDIT.equalsIgnoreCase(operation)) {
+            if (getFf4j().isEnableAudit()) {
+                getFf4j().audit(false);
+                msg = "Audit is now DISABLED";
+                LOGGER.info("Audit has been disabled");
+            } else {
+                getFf4j().audit(true);
+                msg = "Audit is now ENABLED";
+                LOGGER.info("Audit has been enabled");
+            }
         }
         ctx.setVariable(KEY_AUDITENABLE, getFf4j().isEnableAudit());
         ctx.setVariable("msgType", msgType);
