diff --git a/old_product.java b/new_product.java
index 3b29a8e..9862c72 100644
--- a/old_product.java
+++ b/new_product.java
@@ -26,12 +26,8 @@ void commit(UpdateableContainerMetadata target) {
             txn.apply(targetSegmentMetadata);
         });
 
-        // We must first copy the Standalone Segments, and then the Transaction Segments. That's because
-        // the Transaction Segments may refer to one of these newly created Segments, and the metadata
-        // will reject the operation if it can't find the parent.
-        // We need this because HashMap does not necessarily preserve order when iterating via values().
-        copySegmentMetadata(this.newSegments.values(), s -> !s.isTransaction(), target);
-        copySegmentMetadata(this.newSegments.values(), SegmentMetadata::isTransaction, target);
+        // Copy all Segment Metadata
+        copySegmentMetadata(this.newSegments.values(), target);
 
         // Copy truncation points.
         this.newTruncationPoints.forEach(target::setValidTruncationPoint);
