diff --git a/old_product.java b/new_product.java
index 429c12d..2c5d6a1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,7 +14,8 @@ public void assertObject(final InternalFactHandle factHandle,
             this.sink.createAndPropagateAssertLeftTuple( factHandle,
                                                          context,
                                                          workingMemory,
-                                                         useLeftMemory );
+                                                         useLeftMemory, 
+                                                         this );
         } else {
             workingMemory.addLIANodePropagation( new LIANodePropagation( this,
                                                                          factHandle,
