diff --git a/old_test.java b/new_test.java
index 266ee7c..0813205 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testAssertObject() throws Exception {
         when( constraint.isAllowedCachedLeft(any(ContextEntry.class), any(InternalFactHandle.class))).thenReturn(true);
-        when( constraint.isAllowedCachedRight(any(LeftTuple.class), any(ContextEntry.class))).thenReturn(true);
+        when( constraint.isAllowedCachedRight(any(LeftTupleImpl.class), any(ContextEntry.class))).thenReturn(true);
 
         final DefaultFactHandle f0 = (DefaultFactHandle) this.workingMemory
                 .insert("test0");
@@ -18,7 +18,7 @@
         assertEquals(2, this.memory.getRightTupleMemory().size());
 
         RightTuple rightTuple = this.memory.getRightTupleMemory().getFirst(
-                new LeftTuple(f0, this.node, true), null);
+                new LeftTupleImpl(f0, this.node, true), null);
 
         final InternalFactHandle rf0 = rightTuple.getFactHandle();
         final InternalFactHandle rf1 = ((RightTuple) rightTuple.getNext())
