diff --git a/old_product.java b/new_product.java
index d58d836..9fbcd9a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,9 @@
 public void addProvider(EncounterRole role, Provider provider) {
+		// first, make sure the provider isn't already there
+		for (EncounterProvider ep : encounterProviders) {
+			if (ep.getEncounterRole().equals(role) && ep.getProvider().equals(provider))
+				return;
+		}
 		EncounterProvider encounterProvider = new EncounterProvider();
 		encounterProvider.setEncounter(this);
 		encounterProvider.setEncounterRole(role);
