diff --git a/old_test.java b/new_test.java
index 048b1c0..4885065 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,18 +1,6 @@
 @Test
   public void testOnElement() throws Exception {
     setUp(FixedWindows.of(Duration.millis(10)));
-
-    injectElement(1, TriggerResult.CONTINUE);
-    assertThat(tester.getLatestResult(), equalTo(TriggerResult.CONTINUE));
-
-    injectElement(2, TriggerResult.FIRE);
-    assertThat(tester.getLatestResult(), equalTo(TriggerResult.FIRE));
-
-    injectElement(3, TriggerResult.FIRE_AND_FINISH);
-    assertThat(tester.getLatestResult(), equalTo(TriggerResult.FIRE));
-
-    injectElement(4, TriggerResult.CONTINUE);
-    assertThat(tester.getLatestResult(), equalTo(TriggerResult.CONTINUE));
-
-    assertFalse(tester.isMarkedFinished(firstWindow));
+    tester.injectElements(37);
+    verify(mockTrigger).onElement(Mockito.<Trigger<IntervalWindow>.OnElementContext>any());
   }
\ No newline at end of file
