diff --git a/old_test.java b/new_test.java
index a1db83d..d4fb29b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,11 @@
 @Test
   public void AddCheckpointTest() throws Exception {
     TestUtils.createByteFile(mTfs, "/xyz", WriteType.THROUGH, 10);
-    ClientFileInfo fInfo = mLocalTachyonCluster.getMasterInfo().getClientFileInfo("/xyz");
+    ClientFileInfo fInfo =
+        mLocalTachyonCluster.getMasterInfo().getClientFileInfo(new TachyonURI("/xyz"));
     mTfs.createFile(new TachyonURI("/xyz_ck"), new TachyonURI(fInfo.getUfsPath()));
-    ClientFileInfo ckFileInfo = mLocalTachyonCluster.getMasterInfo().getClientFileInfo("/xyz_ck");
+    ClientFileInfo ckFileInfo =
+        mLocalTachyonCluster.getMasterInfo().getClientFileInfo(new TachyonURI("/xyz_ck"));
     mLocalTachyonCluster.stopTFS();
     AddCheckpointTestUtil(fInfo, ckFileInfo);
     String editLogPath = mLocalTachyonCluster.getEditLogPath();
