diff --git a/old_product.java b/new_product.java
index 52b9b85..a830af1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public synchronized void addCheckpoint(int fileId, long length, String checkpointPath,
+public synchronized void addCheckpoint(int fileId, long length, TachyonURI checkpointPath,
       long opTimeMs) {
     if (INACTIVE) {
       return;
@@ -7,6 +7,6 @@ public synchronized void addCheckpoint(int fileId, long length, String checkpoin
     EditLogOperation operation =
         new EditLogOperation(EditLogOperationType.ADD_CHECKPOINT, ++mTransactionId)
             .withParameter("fileId", fileId).withParameter("length", length)
-            .withParameter("path", checkpointPath).withParameter("opTimeMs", opTimeMs);
+            .withParameter("path", checkpointPath.getPath()).withParameter("opTimeMs", opTimeMs);
     writeOperation(operation);
   }
\ No newline at end of file
