diff --git a/old_product.java b/new_product.java
index 8862d26..33e8bee 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public List<PartionerConfigReference> getReferences(final String tableName)
+public List<PartitionerConfigReference> getReferences(final String tableName)
 	{
-		final List<PartionerConfigReference> references = getLines().stream()
+		final List<PartitionerConfigReference> references = getLines().stream()
 				.flatMap(line -> line.getReferences().stream()) // get a stream of all references
 				.filter(ref -> tableName.equalsIgnoreCase(ref.getReferencedTableName())) // filter those who refer to 'tablename'
 				.collect(Collectors.toList());
