diff --git a/old_product.java b/new_product.java
index 63d05f0..8dc9c0c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 private DBNode open(final QueryContext ctx, final boolean id)
       throws QueryException {
 
-    final Data data = ctx.resource.data(checkStr(expr[0], ctx), input);
+    final Data data = data(0, ctx);
     final int v = (int) checkItr(expr[1], ctx);
     final int pre = id ? data.pre(v) : v;
     if(pre < 0 || pre >= data.meta.size) IDINVALID.thrw(input, this, v);
