diff --git a/old_test.java b/new_test.java
index 2101c11..e243be0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public void retrieve() {
     error(_DB_RETRIEVE.args(NAME, "raw"), WHICHRES_X);
     query(_DB_STORE.args(NAME, "raw", "xs:hexBinary('41')"));
-    query("xs:hexBinary(" + _DB_RETRIEVE.args(NAME, "raw") + ')', "41");
+    query("xs:hexBinary(" + _DB_RETRIEVE.args(NAME, "raw") + ')', "A");
     query(_DB_DELETE.args(NAME, "raw"));
     error(_DB_RETRIEVE.args(NAME, "raw"), WHICHRES_X);
   }
\ No newline at end of file
