diff --git a/old_product.java b/new_product.java
index 754c41d..db7a345 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,10 +2,9 @@ void retrieve(final String db, final String path, final boolean raw, final Outpu
       throws IOException {
 
     session().setOutputStream(out);
-    final WebDAVQuery query = new WebDAVQuery(
-      "declare option output:" + (raw ?
-      "method 'raw'; " + _DB_RETRIEVE.args("$db", "$path") + "[1]" :
-      "use-character-maps '&#xA0;=&amp;#xA0;'; " + _DB_OPEN.args("$db", "$path")) + "[1]");
+    final String string = SerializerOptions.USE_CHARACTER_MAPS.arg("&#xA0;=&amp;#xA0;") +
+        (raw ? _DB_RETRIEVE : _DB_OPEN).args("$db", "$path") + "[1]";
+    final WebDAVQuery query = new WebDAVQuery(string);
     query.bind("db", db);
     query.bind("path", path);
     execute(query);
