diff --git a/old_test.java b/new_test.java
index eeba63c..0914265 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,9 +4,9 @@
 
         // tunnelId1 with device label store info
         assertThat(tunnelId1, is(notNullValue()));
-        assertThat(distrPceStore.getTunnelInfo(tunnelId1), is(pceccTunnelInfo1));
+        assertThat(distrPceStore.getTunnelInfo(tunnelId1), is(tunnelConsumerId1));
 
         // tunnelId2 with device label store info
         assertThat(tunnelId2, is(notNullValue()));
-        assertThat(distrPceStore.getTunnelInfo(tunnelId2), is(pceccTunnelInfo2));
+        assertThat(distrPceStore.getTunnelInfo(tunnelId2), is(tunnelConsumerId2));
     }
\ No newline at end of file
