diff --git a/old_product.java b/new_product.java
index f697d47..62aab6a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,7 @@
-public CriteriaQuery and(QueryCriterion... queryCriterions) {
-        criterion = criterion.and(criterionBuilder.and(queryCriterions));
+public CriteriaQuery and(QueryCriterion aCriterion) {
+        if (aCriterion == null) {
+            return this;
+        }
+        criterion = criterion.and(aCriterion);
         return this;
     }
\ No newline at end of file
