diff --git a/old_test.java b/new_test.java
index ae06635..253d4ff 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,5 @@
         QueryCriterion criterion1 = new EqCriterion("name", "abc");
         QueryCriterion criterion2 = new GtCriterion("id", 5);
         QueryCriterion criterion3 = new AndCriterion(criterion1, criterion2);
-        
-        assertEquals(criterion3, instance.and(criterion1, criterion2).getQueryCriterion());
+        assertEquals(criterion3, instance.and(criterion1).and(criterion2).getQueryCriterion());
     }
\ No newline at end of file
