diff --git a/old_product.java b/new_product.java
index 8839a2b..92d7a28 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,13 @@
-public static String getPasswordRequirements(int minLength, int maxLength, List<CharacterRule> characterRules, int numberOfCharacteristics, int numberOfRepeatingCharactersAllowed, int goodStrength, boolean dictionaryEnabled) {
+public static String getPasswordRequirements(int minLength, int maxLength, List<CharacterRule> characterRules, int numberOfCharacteristics, int numberOfRepeatingCharactersAllowed, int goodStrength, boolean dictionaryEnabled, List<String> errors) {
+        logger.info(errors.toString());
         String message = "Your password must contain:";
         message += "<ul>";
         String optionalGoodStrengthNote = "";
         if (goodStrength > 0) {
-            optionalGoodStrengthNote = "( passwords of at least " + goodStrength + " characters are exempt from all other requirements)";
+            optionalGoodStrengthNote = " (passwords of at least " + goodStrength + " characters are exempt from all other requirements)";
         }
-        message += "<li>At least " + minLength + " characters" + optionalGoodStrengthNote + "</li>";
-        message += "<li>At least " + numberOfCharacteristics + " of the following: " + getRequiredCharacters(characterRules) + "</li>";
+        message += "<li " + getColor(errors, ErrorType.TOO_SHORT) + ">" + getOkOrFail(errors, ErrorType.TOO_SHORT) + "At least " + minLength + " characters" + optionalGoodStrengthNote + "</li>";
+        message += "<li " + getColor(errors, ErrorType.INSUFFICIENT_CHARACTERISTICS) + ">" + getOkOrFail(errors, ErrorType.INSUFFICIENT_CHARACTERISTICS) + "At least " + numberOfCharacteristics + " of the following: " + getRequiredCharacters(characterRules) + "</li>";
         message += "</ul>";
         boolean repeatingDigitRuleEnabled = numberOfRepeatingCharactersAllowed > 0;
         boolean showMayNotBlock = repeatingDigitRuleEnabled || dictionaryEnabled;
@@ -23,5 +24,7 @@ public static String getPasswordRequirements(int minLength, int maxLength, List<
         if (showMayNotBlock) {
             message += "</ul>";
         }
+        // for debugging
+//        message += errors.toString();
         return message;
     }
\ No newline at end of file
