diff --git a/old_product.java b/new_product.java
index bbc532b..26b55b9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,7 @@
-public void add(KType e1, KType e2)
+public int add(KType e1, KType e2)
     {
-        add(e1);
-        add(e2);
+        int count = 0;
+        if (add(e1)) count++;
+        if (add(e2)) count++;
+        return count;
     }
\ No newline at end of file
