diff --git a/old_product.java b/new_product.java
index db21871..0f36646 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,12 @@
 public void shutdown() {
-        if (!running) {
-            return;
-        }
-        running = false;
-        LOG.info("Shutting down " + getName());
-        this.interrupt();
+        LOG.info("Shutting down auditor");
+        submitShutdownTask();
+
         try {
-            this.join();
+            while (!executor.awaitTermination(30, TimeUnit.SECONDS)) {
+                LOG.warn("Executor not shutting down, interrupting");
+                executor.shutdownNow();
+            }
         } catch (InterruptedException ie) {
             Thread.currentThread().interrupt();
             LOG.warn("Interrupted while shutting down auditor bookie", ie);
