diff --git a/old_test.java b/new_test.java
index 26f71d6..a81be69 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testShutdown() throws Exception {
         BookieServer auditor = verifyAuditor();
-        shudownBookie(auditor);
+        shutdownBookie(auditor);
 
         // waiting for new auditor
         BookieServer newAuditor = waitForNewAuditor(auditor);
