diff --git a/old_product.java b/new_product.java
index 9ff2757..1c5b39f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
 	public List<Provider> getProviders(String query, Integer start, Integer length,
 	        Map<ProviderAttributeType, Object> attributeValues) {
-		return getProviders(query, start, length, attributeValues, false);
+		return getProviders(query, start, length, attributeValues, true);
 	}
\ No newline at end of file
