diff --git a/old_product.java b/new_product.java
index d3e037f..6a31147 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,11 +12,15 @@ public static String wrap(final String str, int wrapLength, String newLineStr, f
         int offset = 0;
         final StringBuilder wrappedLine = new StringBuilder(inputLineLength + 32);
         
-        while (inputLineLength - offset > wrapLength) {
+        while (offset < inputLineLength) {
             if (str.charAt(offset) == ' ') {
                 offset++;
                 continue;
             }
+            // only last line without leading spaces is left
+            if(inputLineLength - offset <= wrapLength) {
+                break;
+            }
             int spaceToWrapAt = str.lastIndexOf(' ', wrapLength + offset);
 
             if (spaceToWrapAt >= offset) {
