diff --git a/old_test.java b/new_test.java
index f3bd419..796e3fc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -24,4 +24,10 @@
         expected = "Click here," + systemNewLine + "http://commons.apache.org," + systemNewLine 
             + "to jump to the" + systemNewLine + "commons website";
         assertEquals(expected, WordUtils.wrap(input, 20));
+
+        // leading spaces on a new line are stripped
+        // trailing spaces are not stripped
+        input = "word1             word2                        word3";
+        expected = "word1  " + systemNewLine + "word2  " + systemNewLine + "word3";
+        assertEquals(expected, WordUtils.wrap(input, 7));
     }
\ No newline at end of file
