diff --git a/old_test.java b/new_test.java
index 17f7697..4fc6cde 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,24 +1,19 @@
 @Test
-	public void testCreateEvaluation() {
+	public void testCreateEvaluation() throws EntityAlreadyExistsException {
 		// SUCCESS
 		EvaluationData e = new EvaluationData();
-		e.course = "Winzor101";
-		e.name = "Basic Herping Derping";
+		e.course = "Computing101";
+		e.name = "Basic Computing Evaluation1";
 		e.startTime = new Date();
 		e.endTime = new Date();
-		
-		try {
-			evaluationsDb.createEvaluation(e);
-		} catch (EntityAlreadyExistsException ex) {
-			fail();
-		}
+		evaluationsDb.createEvaluation(e);
 		
 		// FAIL : duplicate
 		try {
 			evaluationsDb.createEvaluation(e);
 			fail();
 		} catch (EntityAlreadyExistsException ex) {
-			
+			assertContains(EvaluationsDb.ERROR_CREATE_EVALUATION_ALREADY_EXISTS, ex.getMessage());
 		}
 		
 		// FAIL : invalid params
@@ -27,8 +22,16 @@
 			evaluationsDb.createEvaluation(e);
 			fail();
 		} catch (AssertionError a) {
-			
+			assertEquals(a.getMessage(), EvaluationData.ERROR_FIELD_STARTTIME);
 		} catch (EntityAlreadyExistsException ex) {
 			fail();
 		}
+		
+		// Null params check:
+		try {
+			evaluationsDb.createEvaluation(null);
+			fail();
+		} catch (AssertionError a) {
+			assertEquals(Common.ERROR_DBLEVEL_NULL_INPUT, a.getMessage());
+		}
 	}
\ No newline at end of file
