diff --git a/old_product.java b/new_product.java
index 61e90fa..207676c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-private static MutableSparseVector userRatingVector(ArrayList<Rating> ratings) {
+@Deprecated
+    private static MutableSparseVector userRatingVector(ArrayList<Rating> ratings) {
     	Rating rp = null;
     	for (Rating r: ratings) {
     		if (rp != null && ITEM_TIME_COMPARATOR.compare(rp, r) > 0) {
