diff --git a/old_product.java b/new_product.java
index f3d1958..61466f7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,7 +11,7 @@
 
         final CommandEntity commandEntity = this.findCommand(id);
         final List<ApplicationEntity> applicationEntities = new ArrayList<>();
-        applicationIds.stream().forEach(appId -> applicationEntities.add(this.appRepo.findOne(appId)));
+        applicationIds.forEach(appId -> applicationEntities.add(this.appRepo.findOne(appId)));
 
         commandEntity.setApplications(applicationEntities);
     }
\ No newline at end of file
