diff --git a/old_test.java b/new_test.java
index 2ec4fe9..289fa59 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
         Assert.assertEquals(1, preCommands.size());
         Assert.assertEquals(1, preCommands
             .stream()
-            .filter(command -> COMMAND_1_ID.equals(command.getId()))
+            .filter(command -> COMMAND_1_ID.equals(command.getId().orElseThrow(IllegalArgumentException::new)))
             .count()
         );
 
@@ -19,7 +19,7 @@
             1,
             this.service.getApplicationsForCommand(COMMAND_2_ID)
                 .stream()
-                .filter(application -> APP_1_ID.equals(application.getId()))
+                .filter(application -> APP_1_ID.equals(application.getId().orElseThrow(IllegalArgumentException::new)))
                 .count()
         );
     }
\ No newline at end of file
