diff --git a/old_test.java b/new_test.java
index 8671650..ca18641 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
       throws Exception {
     when(req.getMethod()).thenReturn("GET");
     when(req.getRequestURI()).thenReturn("/failAtConstruction");
-    handler.handleRequest(req, rsp, component);
+    handler.handleRequest(req, rsp);
     verify(rsp).sendError(503, "Fail at construction");
   }
\ No newline at end of file
