diff --git a/old_test.java b/new_test.java
index 9ea2879..1131633 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testGetConnection() throws Exception {
         String uri = "thecooldatabase";
         String user = "tryggve";
-        String password = "arne";
+        char[] password = "arne".toCharArray();
         final PoolableConnection result = instance.getConnection(uri, user, password);
         assertNotNull(result);
     }
\ No newline at end of file
