diff --git a/old_product.java b/new_product.java
index a886467..428fa12 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,13 @@
-public final ByteBuffer get(byte[] dest, int off, int len) {
+@Override
+    public final ByteBuffer get(byte[] dest, int off, int len) {
         int length = dest.length;
-        if ((off < 0 ) || (len < 0) || (long)off + (long)len > length) {
+        if ((off < 0) || (len < 0) || (long) off + (long) len > length) {
             throw new IndexOutOfBoundsException();
         }
         if (len > remaining()) {
             throw new BufferUnderflowException();
         }
-        getBaseAddress().getByteArray(offset+position, dest, off, len);
+        getBaseAddress().getByteArray(offset + position, dest, off, len);
         position += len;
         return this;
     }
\ No newline at end of file
