diff --git a/old_product.java b/new_product.java
index 60f52ee..5afedb5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-protected void scanArtifacts(MavenProject project, Engine engine) {
+protected void scanArtifacts(MavenProject project, MavenEngine engine) {
         for (Artifact a : project.getArtifacts()) {
             if (excludeFromScan(a)) {
                 continue;
