diff --git a/old_test.java b/new_test.java
index d35e32b..4397101 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,20 +6,20 @@
 		final boolean rw = false; // does not matter
 
 		{
-			final Set<Integer> expected = asHashSet();
-			final Set<Integer> actual = asHashSet();
+			final Set<OrgId> expected = asOrgIdsSet();
+			final Set<OrgId> actual = asOrgIdsSet();
 			rule.filterOrgs(role, tableName, rw, actual);
 			Assert.assertEquals("Org IDs shall not be modified", expected, actual);
 		}
 		{
-			final Set<Integer> expected = asHashSet(1);
-			final Set<Integer> actual = asHashSet(1);
+			final Set<OrgId> expected = asOrgIdsSet(1);
+			final Set<OrgId> actual = asOrgIdsSet(1);
 			rule.filterOrgs(role, tableName, rw, actual);
 			Assert.assertEquals("Org IDs shall not be modified", expected, actual);
 		}
 		{
-			final Set<Integer> expected = asHashSet(1, 2, 3);
-			final Set<Integer> actual = asHashSet(1, 2, 3);
+			final Set<OrgId> expected = asOrgIdsSet(1, 2, 3);
+			final Set<OrgId> actual = asOrgIdsSet(1, 2, 3);
 			rule.filterOrgs(role, tableName, rw, actual);
 			Assert.assertEquals("Org IDs shall not be modified", expected, actual);
 		}
