diff --git a/old_product.java b/new_product.java
index 04ee18e..d3615dd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-void filterOrgs(IUserRolePermissions rolePermissions, String tableName, boolean readWrite, Set<Integer> orgIds);
\ No newline at end of file
+void filterOrgs(IUserRolePermissions rolePermissions, String tableName, boolean readWrite, Set<OrgId> orgIds);
\ No newline at end of file
