diff --git a/old_product.java b/new_product.java
index 2211a38..504891d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,15 +1,14 @@
-public Criteria or(Criteria criteria) {
-		Assert.notNull(criteria, "Cannot chain 'null' criteria.");
+@SuppressWarnings("unchecked")
+	@Override
+	public Crotch and(Node node) {
 
-		Criteria orConnectedCriteria = new OrCriteria(this.criteriaChain, criteria.getField());
-		orConnectedCriteria.criteria.addAll(criteria.criteria);
-
-		for (Criteria c : criteria.getCriteriaChain()) {
-			if (c == criteria) {
-				continue;
-			}
-			orConnectedCriteria.and(c);
+		if (!(node instanceof Criteria)) {
+			throw new IllegalArgumentException("Can only add instances of Criteria");
 		}
 
-		return orConnectedCriteria;
+		Crotch crotch = new Crotch();
+		crotch.setParent(this.getParent());
+		crotch.add(this);
+		crotch.add((Criteria) node);
+		return crotch;
 	}
\ No newline at end of file
