diff --git a/old_product.java b/new_product.java
index 1aa3ce7..7d804fc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -19,7 +19,7 @@
             return;
         }
 
-        controllerService.getScaleRecords(scopeName, streamName).thenApply(listScaleMetadata -> {
+        controllerService.getScaleRecords(scopeName, streamName, from, to).thenApply(listScaleMetadata -> {
             Iterator<ScaleMetadata> metadataIterator = listScaleMetadata.iterator();
             List<ScaleMetadata> finalScaleMetadataList = new ArrayList<ScaleMetadata>();
 
