diff --git a/old_test.java b/new_test.java
index 08cc9cc..87ad161 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,6 +15,9 @@
 		// Video of a TV episode with the "Mysterious Wordplay" title
 		assertThat(FileUtil.getFileNameWithRewriting("Universal.Media.Server.S01E02.Mysterious.Wordplay.720p.mkv", null)).isEqualTo("Universal Media Server - 102 - Mysterious Wordplay");
 
+		// Video of an uncut TV episode
+		assertThat(FileUtil.getFileNameWithRewriting("Universal.Media.Server.S01E02.UNCUT.720p.mkv", null)).isEqualTo("Universal Media Server - 102 (Uncut)");
+
 		// Video of an extended cut of a TV episode
 		assertThat(FileUtil.getFileNameWithRewriting("Universal.Media.Server.S01E02.Extended.Cut.720p.mkv", null)).isEqualTo("Universal Media Server - 102 (Extended Cut)");
 
@@ -28,5 +31,6 @@
 		assertThat(FileUtil.getFileNameWithRewriting("Universal.Media.Server.2015.720p.mkv", null)).isEqualTo("Universal Media Server (2015)");
 
 		// Video of a special edition of a movie
-		assertThat(FileUtil.getFileNameWithRewriting("Universal.Media.Server.Special.Edition.2015.720p.mkv", null)).isEqualTo("Universal Media Server (Special Edition) (2015)");
+		assertThat(FileUtil.getFileNameWithRewriting("Universal.Media.Server.Special.Edition.2015.720p.mkv", null)).isEqualTo("Universal Media Server (2015) (Special Edition)");
+		assertThat(FileUtil.getFileNameWithRewriting("Universal.Media.Server.2015.Special.Edition.720p.mkv", null)).isEqualTo("Universal Media Server (2015) (Special Edition)");
 	}
\ No newline at end of file
