diff --git a/old_product.java b/new_product.java
index 984a9c1..5ab58d1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ public void module(final byte[] path, final byte[] uri) throws QueryException {
     // check if module has already been parsed
     final byte[] u = qc.modParsed.get(p);
     if(u != null) {
-      if(!eq(uri, u)) throw error(WRONGMODULE, uri,
+      if(!eq(uri, u)) throw error(WRONGMODULE_X_X, uri,
           qc.context.user.has(Perm.ADMIN) ? io.path() : io.name());
       if(!sc.xquery3() && qc.modStack.contains(p)) throw error(CIRCMODULE);
       return;
@@ -18,7 +18,7 @@ public void module(final byte[] path, final byte[] uri) throws QueryException {
     try {
       qu = string(io.read());
     } catch(final IOException ex) {
-      throw error(WHICHMODFILE, qc.context.user.has(Perm.ADMIN) ? io.path() : io.name());
+      throw error(WHICHMODFILE_X, qc.context.user.has(Perm.ADMIN) ? io.path() : io.name());
     }
 
     qc.modStack.push(p);
@@ -27,14 +27,14 @@ public void module(final byte[] path, final byte[] uri) throws QueryException {
     final byte[] muri = lib.name.uri();
 
     // check if import and declaration uri match
-    if(!eq(uri, muri)) throw error(WRONGMODULE, muri, file);
+    if(!eq(uri, muri)) throw error(WRONGMODULE_X_X, muri, file);
 
-    // check if context item declaration types are compatible to each other
+    // check if context value declaration types are compatible to each other
     if(sub.contextType != null) {
       if(sc.contextType == null) {
         sc.contextType = sub.contextType;
       } else if(!sub.contextType.eq(sc.contextType)) {
-        throw error(CITYPES, sub.contextType, sc.contextType);
+        throw error(CITYPES_X_X, sub.contextType, sc.contextType);
       }
     }
     qc.modStack.pop();
