diff --git a/old_product.java b/new_product.java
index 0a43d7c..2f07461 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
 @Path("/transactions/{tx-id}/invalidate")
-  @PUT
+  @POST
   public void invalidateTx(HttpRequest request, HttpResponder responder,
                            @PathParam("tx-id") final String txId) {
     try {
-      Long txIdLong = Long.valueOf(txId);
+      long txIdLong = Long.parseLong(txId);
       boolean success = txClient.invalidate(txIdLong);
       if (success) {
         LOG.info("Transaction {} successfully invalidated", txId);
