diff --git a/old_test.java b/new_test.java
index d0a9b94..4c38e5b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,7 +14,7 @@
         
         result = instance.insepct(file);
         boolean found = false;
-        for (Evidence e : result.getTitleEvidence()) {
+        for (Evidence e : result.getProductEvidence()) {
             if (e.getName().equals("package-title") && e.getValue().equals("org.mortbay.http")) {
                 found = true;
                 break;
