diff --git a/old_product.java b/new_product.java
index f9f30e9..5d2ae5e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,11 @@
 @Override
     public long get(DateTimeField field) {
-        if (field instanceof LocalDateField) {
-            switch ((LocalDateField) field) {
+        if (field instanceof LocalDateTimeField) {
+            switch ((LocalDateTimeField) field) {
                 case DAY_OF_MONTH: return day;
                 case MONTH_OF_YEAR: return month.getValue();
             }
             throw new CalendricalException(field.getName() + " not valid for MonthDay");
-        } else if (field instanceof LocalTimeField) {
-            throw new CalendricalException(field.getName() + " not valid for MonthDay");
         }
         return field.get(this);
     }
\ No newline at end of file
