diff --git a/old_test.java b/new_test.java
index a1f3816..3af89d8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test(groups={"tck"})
     public void test_get_DateField() {
         OffsetDate test = OffsetDate.of(2008, 6, 30, OFFSET_PONE);
-        assertEquals(test.get(LocalDateField.YEAR), 2008);
-        assertEquals(test.get(LocalDateField.MONTH_OF_YEAR), 6);
-        assertEquals(test.get(LocalDateField.DAY_OF_MONTH), 30);
-        assertEquals(test.get(LocalDateField.DAY_OF_WEEK), 1);
-        assertEquals(test.get(LocalDateField.DAY_OF_YEAR), 182);
+        assertEquals(test.get(LocalDateTimeField.YEAR), 2008);
+        assertEquals(test.get(LocalDateTimeField.MONTH_OF_YEAR), 6);
+        assertEquals(test.get(LocalDateTimeField.DAY_OF_MONTH), 30);
+        assertEquals(test.get(LocalDateTimeField.DAY_OF_WEEK), 1);
+        assertEquals(test.get(LocalDateTimeField.DAY_OF_YEAR), 182);
     }
\ No newline at end of file
