diff --git a/old_test.java b/new_test.java
index 1822668..760bb03 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,13 @@
 @Test
 	public void fullView() {
-		IntrinsicParameters param = new IntrinsicParameters(false).fsetK(300, 320, 0, 150, 130, width, height).
-				fsetRadial(0.1, 0.05);
+		IntrinsicParameters param = new IntrinsicParameters().
+				fsetK(300, 320, 0, 150, 130, width, height).fsetRadial(0.1, 0.05);
 
 		PointTransform_F32 adjusted = LensDistortionOps.fullView(param,null);
 		checkBorderOutside(adjusted);
 
-		param = new IntrinsicParameters(false).fsetK(300, 320, 0, 150, 130, width, height).fsetRadial(-0.1,-0.05);
+		param = new IntrinsicParameters().
+				fsetK(300, 320, 0, 150, 130, width, height).fsetRadial(-0.1,-0.05);
 		adjusted = LensDistortionOps.fullView(param,null);
 		checkBorderOutside(adjusted);
 
