diff --git a/old_product.java b/new_product.java
index fffcd64..4eeb770 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,7 @@
-public void setId(final String id) throws GenieException {
+public void setId(final String id) throws GeniePreconditionException {
         if (StringUtils.isBlank(this.id)) {
             this.id = id;
         } else {
-            throw new GenieException(
-                    HttpURLConnection.HTTP_BAD_REQUEST,
-                    "Id already set for this entity.");
+            throw new GeniePreconditionException("Id already set for this entity.");
         }
     }
\ No newline at end of file
