diff --git a/old_product.java b/new_product.java
index 2da71a0..9d5d59d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public Map<String, Collection<String>> getComponents() {
-    Map<String, Collection<String>> serviceComponents = new HashMap<String, Collection<String>>();
+    Map<String, Collection<String>> serviceComponents = new HashMap<>();
     for (String service : getServices()) {
-      Collection<String> components = new HashSet<String>();
+      Collection<String> components = new HashSet<>();
       components.addAll(getComponents(service));
       serviceComponents.put(service, components);
     }
