diff --git a/old_test.java b/new_test.java
index ae146ed..0b355a4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
     public void testGeProp() {
-    	assertEquals(new GePropCriterion("id", "name"), instance.geProp("id", "name").getQueryCriterion());
+    	assertEquals(Criteria.geProp("id", "name"), instance.geProp("id", "name").getQueryCriterion());
     }
\ No newline at end of file
