diff --git a/old_product.java b/new_product.java
index 74551cd..daeca4b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery geProp(String propName, String otherProp) {
-        criterion = criterion.and(criterionBuilder.geProp(propName, otherProp));
+        criterion = criterion.and(Criteria.geProp(propName, otherProp));
         return this;
     }
\ No newline at end of file
