diff --git a/old_test.java b/new_test.java
index 679628e..21c0aff 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testFromXContent() throws IOException {
         QB testQuery = createTestQueryBuilder();
-        QueryParseContext context = createContext();
+        QueryParseContext context = createParseContext();
         String contentString = testQuery.toString();
         XContentParser parser = XContentFactory.xContent(contentString).createParser(contentString);
         context.reset(parser);
