diff --git a/old_product.java b/new_product.java
index 4855d5a..c5b5c91 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
     public QueryBuilder fromXContent(QueryParseContext parseContext) throws IOException, QueryParsingException {
-        Query query = parse(parseContext);
+        Query query = parse(parseContext.shardContext());
         return new QueryWrappingQueryBuilder(query);
     }
\ No newline at end of file
