diff --git a/old_product.java b/new_product.java
index 927e97b..4dd1bd5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ void retrieve(final String db, final String path, final boolean raw, final Outpu
 
     session().setOutputStream(out);
     final String string = SerializerOptions.USE_CHARACTER_MAPS.arg(WEBDAV) +
-        (raw ? _DB_RETRIEVE : _DB_OPEN).args("$db", "$path") + "[1]";
+        (raw ? _DB_RETRIEVE : _DB_OPEN).args(" $db", " $path") + "[1]";
     final WebDAVQuery query = new WebDAVQuery(string);
     query.bind("db", db);
     query.bind("path", path);
