diff --git a/old_test.java b/new_test.java
index e243be0..837f1ee 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void retrieve() {
     error(_DB_RETRIEVE.args(NAME, "raw"), WHICHRES_X);
-    query(_DB_STORE.args(NAME, "raw", "xs:hexBinary('41')"));
+    query(_DB_STORE.args(NAME, "raw", " xs:hexBinary('41')"));
     query("xs:hexBinary(" + _DB_RETRIEVE.args(NAME, "raw") + ')', "A");
     query(_DB_DELETE.args(NAME, "raw"));
     error(_DB_RETRIEVE.args(NAME, "raw"), WHICHRES_X);
