diff --git a/old_test.java b/new_test.java
index f63116a..506aa60 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
     public void test_parseBest_firstOption() throws Exception {
         DateTimeFormatter test = DateTimeFormatters.pattern("yyyy-MM-dd[ZZZ]");
-        CalendricalObject result = test.parseBest("2011-06-30+03:00", OffsetDate.class, LocalDate.class);
+        DateTimeCalendrical result = test.parseBest("2011-06-30+03:00", OffsetDate.class, LocalDate.class);
         assertEquals(result, OffsetDate.of(2011, 6, 30, ZoneOffset.ofHours(3)));
     }
\ No newline at end of file
