diff --git a/old_product.java b/new_product.java
index ea54e4d..22bdc24 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public CalendricalObject parseBest(CharSequence text, Class<?>... types) {
+public DateTimeCalendrical parseBest(CharSequence text, Class<?>... types) {
         DateTimes.checkNotNull(text, "Text must not be null");
         DateTimes.checkNotNull(types, "Class array must not be null");
         if (types.length < 2) {
@@ -8,7 +8,7 @@ public CalendricalObject parseBest(CharSequence text, Class<?>... types) {
         try {
             DateTimeBuilder builder = parseToBuilder(str).resolve();
             for (Class<?> type : types) {
-                CalendricalObject cal = (CalendricalObject) builder.extract(type);
+                DateTimeCalendrical cal = (DateTimeCalendrical) builder.extract(type);
                 if (cal != null) {
                     return cal;
                 }
