diff --git a/old_test.java b/new_test.java
index e778d97..d33d3d7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,12 @@
 @Test
     public void testSetCreated() {
         final Auditable a = new Auditable();
-        Assert.assertNull(a.getCreated());
-        a.setCreated(new Date());
-        Assert.assertNull(a.getCreated());
+        Assert.assertNotNull(a.getCreated());
+        final Date date = new Date(0);
+        a.setCreated(date);
+        Assert.assertNotNull(a.getCreated());
+        Assert.assertEquals(date, a.getCreated());
         a.onCreateAuditable();
         Assert.assertNotNull(a.getCreated());
+        Assert.assertNotEquals(date, a.getCreated());
     }
\ No newline at end of file
