diff --git a/old_product.java b/new_product.java
index e0644bd..ea00da3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,6 @@
 public void setCreated(final Date created) {
         LOG.info("Tried to set created to " + created + " for entity " + this.id + ". Will not be persisted.");
-        this.created = new Date(created.getTime());
+        if (created.before(this.created)) {
+            this.created = new Date(created.getTime());
+        }
     }
\ No newline at end of file
