diff --git a/old_product.java b/new_product.java
index 28b3df9..9f41039 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,8 @@ public Builder<T> resizer(Resizer resizer)
 		{
 			checkForNull(resizer, "Resizer is null.");
 			updateStatus(Properties.RESIZER, Status.ALREADY_SET);
+			updateStatus(Properties.RESIZER_FACTORY, Status.CANNOT_SET);
 			updateStatus(Properties.SCALING_MODE, Status.CANNOT_SET);
-			this.resizer = resizer;
+			this.resizerFactory = new FixedResizerFactory(resizer);
 			return this;
 		}
\ No newline at end of file
