diff --git a/old_test.java b/new_test.java
index 294b619..f452dba 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,12 +17,16 @@
         assertEquals(CacheScope.INSTANCE, c1.getConfig().getCacheScope());
         assertTrue(c1 instanceof EhcacheCache);
 
+        requestService.startRequest();
+
         Cache rc1 = cachingService.getCache("org.dspace.request.cache1", new CacheConfig(CacheScope.REQUEST));
         assertNotNull(rc1);
         assertEquals("org.dspace.request.cache1", rc1.getName());
         assertEquals(CacheScope.REQUEST, rc1.getConfig().getCacheScope());
         assertTrue(rc1 instanceof MapCache);
 
+        requestService.endRequest(null);
+
         // try getting the same one twice
         Cache c2 = cachingService.getCache("org.dspace.aztest", null);
         assertNotNull(c2);
