diff --git a/old_product.java b/new_product.java
index f765987..961a195 100644
--- a/old_product.java
+++ b/new_product.java
@@ -32,14 +32,7 @@
 
         // We are the first/only ones requesting this id; go ahead and assign an id.
         if (needsAssignment) {
-            // Determine if given StreamSegmentName is actually a Transaction StreamSegmentName.
-            String parentStreamSegmentName = StreamSegmentNameUtils.getParentStreamSegmentName(streamSegmentName);
-            if (parentStreamSegmentName == null) {
-                // Stand-alone StreamSegment.
-                this.executor.execute(() -> assignStreamSegmentId(streamSegmentName, timeout));
-            } else {
-                this.executor.execute(() -> assignTransactionStreamSegmentId(streamSegmentName, parentStreamSegmentName, timeout));
-            }
+            this.executor.execute(() -> assignStreamSegmentId(streamSegmentName, timeout));
         }
 
         return queuedCallback.result;
