diff --git a/old_product.java b/new_product.java
index ed530a9..2e184b6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
 private synchronized Item writeBinary(final boolean append, final QueryContext qc)
       throws QueryException, IOException {
 
-    final Path path = checkParentDir(checkPath(0, qc));
-    final Bin bin = checkBin(exprs[1], qc);
-    final long off = exprs.length > 2 ? checkItr(exprs[2], qc) : 0;
+    final Path path = checkParentDir(toPath(0, qc));
+    final Bin bin = toBin(exprs[1], qc);
+    final long off = exprs.length > 2 ? toLong(exprs[2], qc) : 0;
 
     // write full file
     if(exprs.length == 2) {
@@ -15,7 +15,7 @@ private synchronized Item writeBinary(final boolean append, final QueryContext q
       // write file chunk
       try(final RandomAccessFile raf = new RandomAccessFile(path.toFile(), "rw")) {
         final long dlen = raf.length();
-        if(off < 0 || off > dlen) throw FILE_OUT_OF_RANGE.get(info, off, dlen);
+        if(off < 0 || off > dlen) throw FILE_OUT_OF_RANGE_X_X.get(info, off, dlen);
         raf.seek(off);
         raf.write(bin.binary(info));
       }
