diff --git a/old_test.java b/new_test.java
index 9b8adb5..9262d1d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,15 +2,15 @@
   public void writeBinary() {
     // check errors
     final String bin = "xs:base64Binary('MA==')";
-    error(_FILE_WRITE_BINARY.args(PATH, bin), Err.FILE_IS_DIR);
-    error(_FILE_WRITE_BINARY.args(PATH1, "NoBinary"), Err.BINARYTYPE);
+    error(_FILE_WRITE_BINARY.args(PATH, bin), Err.FILE_IS_DIR_X);
+    error(_FILE_WRITE_BINARY.args(PATH1, "NoBinary"), Err.BINARY_X);
     // write file and check size
     query(_FILE_WRITE_BINARY.args(PATH1, bin));
     query(_FILE_SIZE.args(PATH1), "1");
     query(_FILE_WRITE_BINARY.args(PATH1, bin));
     query(_FILE_SIZE.args(PATH1), "1");
     // write data to specific offset and check size
-    error(_FILE_WRITE_BINARY.args(PATH1, bin, 2), Err.FILE_OUT_OF_RANGE);
+    error(_FILE_WRITE_BINARY.args(PATH1, bin, 2), Err.FILE_OUT_OF_RANGE_X_X);
     query(_FILE_WRITE_BINARY.args(PATH1, bin, 0));
     query(_FILE_READ_TEXT.args(PATH1), "0");
     query(_FILE_WRITE_BINARY.args(PATH1, bin, 1));
