diff --git a/old_test.java b/new_test.java
index 2c38dd3..91def0a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,6 @@
 @Test
 	public void getProvidersByPerson_shouldNotReturnRetiredProvidersIfIncludeRetiredFalse() throws Exception {
-		Assert.assertEquals(1, service.getProvidersByPerson(personDao.getPerson(2), false).size());
+        Collection<Provider> providers = service.getProvidersByPerson(personDao.getPerson(2), false);
+		Assert.assertEquals(1, providers.size());
+        Assert.assertFalse(providers.iterator().next().isRetired());
 	}
\ No newline at end of file
