diff --git a/old_product.java b/new_product.java
index 21294bb..b85ec82 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery sizeLe(String propName, int size) {
-        criterion = criterion.and(criterionBuilder.sizeLe(propName, size));
+        criterion = criterion.and(Criteria.sizeLe(propName, size));
         return this;
     }
\ No newline at end of file
