diff --git a/old_test.java b/new_test.java
index 9d95abc..2073fa9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
         System.out.println("\n+++ setSequence");
 
         int index = 1;
-        short[] seq = new short[]{0, 3, 7};
+        RunSequence seq = new RunSequence(new short[]{0, 3, 7});
         RunTable instance = createHorizontalInstance();
         System.out.println("table before:" + instance.dumpOf());
         instance.setSequence(index, seq);
