diff --git a/old_test.java b/new_test.java
index 5051d99..a41afbe 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,15 @@
 @Test
-    public void testRegister() throws IOException, SQLException
+    public void testRegister() throws IOException, SQLException, AuthorizeException
     {
+        new NonStrictExpectations(authorizeService.getClass())
+        {{
+            authorizeService.authorizeAction((Context) any, (Bitstream) any,
+                    Constants.WRITE); result = null;
+        }};
         int assetstore = 0;
         File f = new File(testProps.get("test.bitstream").toString());
-        Bitstream registered = Bitstream.register(context,assetstore, f.getName());
+        Bitstream registered = bitstreamService.register(context,assetstore, f.getName());
         //the item created by default has no name nor type set
-        assertThat("testRegister 0", registered.getFormat().getMIMEType(), equalTo("application/octet-stream"));
+        assertThat("testRegister 0", registered.getFormat(context).getMIMEType(), equalTo("application/octet-stream"));
         assertThat("testRegister 1", registered.getName(), nullValue());
     }
\ No newline at end of file
