diff --git a/old_product.java b/new_product.java
index 1ee98b3..bdefacb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
-@RequiresSession
-  public List<AlertDefinitionEntity> findAll(long clusterId) {
+public List<AlertDefinitionEntity> findAll(long clusterId) {
     TypedQuery<AlertDefinitionEntity> query = entityManagerProvider.get().createNamedQuery(
         "AlertDefinitionEntity.findAllInCluster", AlertDefinitionEntity.class);
 
