diff --git a/old_product.java b/new_product.java
index 6ee6e79..b7c8dd9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -18,5 +18,8 @@ public ServiceCall<IntentCollection> listIntents(ListIntentsOptions listIntentsO
     if (listIntentsOptions.cursor() != null) {
       builder.query("cursor", listIntentsOptions.cursor());
     }
+    if (listIntentsOptions.includeAudit() != null) {
+      builder.query("include_audit", String.valueOf(listIntentsOptions.includeAudit()));
+    }
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(IntentCollection.class));
   }
\ No newline at end of file
