diff --git a/old_product.java b/new_product.java
index 16bccca..532c4d1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
-ZonedDateTime calculatePreparationDateOrNull(
-			IContextAware context,
-			SOTrx soTrx,
-			ZonedDateTime calculationTime,
-			ZonedDateTime datePromised,
-			BPartnerLocationId bpartnerLocationId);
\ No newline at end of file
+@NonNull
+	ImmutablePair<TourId, ZonedDateTime> calculateTourAndPreparationDate(
+			@NonNull IContextAware context,
+			@NonNull SOTrx soTrx,
+			@NonNull ZonedDateTime calculationTime,
+			@NonNull ZonedDateTime datePromised,
+			@NonNull BPartnerLocationId bpartnerLocationId);
\ No newline at end of file
