diff --git a/old_product.java b/new_product.java
index 1403073..d66bd3d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,18 +1,14 @@
-public int truncate(long upToSequenceNumber) {
-        int count = 0;
+public void truncate(long upToSequenceNumber) {
         synchronized (this.lock) {
             // We truncate by finding the new head and simply pointing our head reference to it, as well as disconnecting
             // its predecessor node from it. We also need to mark every truncated node as such - this will instruct ongoing
             // reads to stop serving truncated data.
             while (this.head != null && this.head.item.getSequenceNumber() <= upToSequenceNumber) {
                 this.head = trim(this.head);
-                count++;
             }
 
             if (this.head == null) {
                 this.tail = null;
             }
         }
-
-        return count;
     }
\ No newline at end of file
