diff --git a/old_test.java b/new_test.java
index 103e136..3dc2cee 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,8 +13,7 @@
 
             // Test truncating after each sequence number that we got back.
             for (LogAddress address : addresses) {
-                boolean truncated = log.truncate(address, TIMEOUT).join();
-                Assert.assertTrue("No truncation happened.", truncated);
+                log.truncate(address, TIMEOUT).join();
                 writeData.headMap(address, true).clear();
                 verifyReads(log, createLogAddress(-1), writeData);
             }
