diff --git a/old_test.java b/new_test.java
index 62a9c97..504fb89 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,7 +14,7 @@
     Files.write(
         folder.newFile("my/root/some/path/my_image.png").toPath(), "images".getBytes(UTF_8));
 
-    ReportFiles files = createReportFiles(destination, Optional.empty(), root);
+    FilesWithEntryPoint files = readFilesWithEntryPoint(destination, Optional.empty(), root);
 
     assertThat(files.entryPoint().toString()).isEqualTo("some/path/path.zip");
     assertThat(readAllFiles(files).keySet()).containsExactly("some/path/path.zip");
