diff --git a/old_product.java b/new_product.java
index 4cda0a3..c87454f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,10 +3,10 @@ protected Connection obtainInternalConnection() throws SQLException {
 		Connection result = null;
 
 		int acquireRetryAttempts = this.pool.getConfig().getAcquireRetryAttempts();
-		int acquireRetryDelay = this.pool.getConfig().getAcquireRetryDelay();
+		long acquireRetryDelayInMs = this.pool.getConfig().getAcquireRetryDelayInMs();
 		AcquireFailConfig acquireConfig = new AcquireFailConfig();
 		acquireConfig.setAcquireRetryAttempts(new AtomicInteger(acquireRetryAttempts));
-		acquireConfig.setAcquireRetryDelay(acquireRetryDelay);
+		acquireConfig.setAcquireRetryDelayInMs(acquireRetryDelayInMs);
 		acquireConfig.setLogMessage("Failed to acquire connection");
 
 		this.connectionHook = this.pool.getConfig().getConnectionHook();
@@ -31,10 +31,10 @@ protected Connection obtainInternalConnection() throws SQLException {
 				if (this.connectionHook != null){
 					tryAgain = this.connectionHook.onAcquireFail(e, acquireConfig);
 				} else {
-					logger.error("Failed to acquire connection. Sleeping for "+acquireRetryDelay+"ms. Attempts left: "+acquireRetryAttempts, e);
+					logger.error("Failed to acquire connection. Sleeping for "+acquireRetryDelayInMs+"ms. Attempts left: "+acquireRetryAttempts, e);
 
 					try {
-						Thread.sleep(acquireRetryDelay);
+						Thread.sleep(acquireRetryDelayInMs);
 						if (acquireRetryAttempts > 0){
 							tryAgain = (--acquireRetryAttempts) != 0;
 						}
