diff --git a/old_test.java b/new_test.java
index 2736b71..2486321 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
 		expect(this.mockPool.getConfig()).andReturn(this.config).anyTimes();
 
 		this.testClass.url = "jdbc:mock:driver";
-		this.config.setAcquireRetryDelay(1);
+		this.config.setAcquireRetryDelayInMs(1);
 		CustomHook testHook = new CustomHook();
 		this.config.setConnectionHook(testHook);
 		// make it fail the first time and succeed the second time
@@ -41,7 +41,7 @@
 		//	Test 4: Get signalled to interrupt fail delay
 		count=99;
 		this.config.setAcquireRetryAttempts(2);
-		this.config.setAcquireRetryDelay(7000);
+		this.config.setAcquireRetryDelayInMs(7000);
 		final Thread currentThread = Thread.currentThread();
 
 		try{
@@ -64,7 +64,7 @@
 		} catch (SQLException e){
 			// expected behaviour
 		}
-		this.config.setAcquireRetryDelay(10);
+		this.config.setAcquireRetryDelayInMs(10);
 		
 		
 	}
\ No newline at end of file
