diff --git a/old_test.java b/new_test.java
index 5cae46d..d183052 100644
--- a/old_test.java
+++ b/new_test.java
@@ -65,8 +65,8 @@
         
         MaximumEntropy.TrainingParameters param = new MaximumEntropy.TrainingParameters();
         param.setTotalIterations(10);
-        instance.initializeTrainingConfiguration(param);
-        MaximumEntropy.ValidationMetrics vm = instance.kFoldCrossValidation(trainingData, k);
+        
+        MaximumEntropy.ValidationMetrics vm = instance.kFoldCrossValidation(trainingData, param, k);
         
         double expResult = 0.6051098901098901;
         double result = vm.getMacroF1();
