diff --git a/old_product.java b/new_product.java
index d7930b2..b11df5d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
     @ResponseStatus(HttpStatus.CREATED)
-    public ResponseEntity<Void> createCluster(@RequestBody final Cluster cluster) throws GenieException {
+    public ResponseEntity<Void> createCluster(@RequestBody @Valid final Cluster cluster) throws GenieException {
         log.debug("Called to create new cluster {}", cluster);
-        final String id = this.clusterService.createCluster(cluster);
+        final String id = this.clusterService.createCluster(DtoAdapters.toV4ClusterRequest(cluster));
         final HttpHeaders httpHeaders = new HttpHeaders();
         httpHeaders.setLocation(
             ServletUriComponentsBuilder
