diff --git a/old_test.java b/new_test.java
index e2a271f..323a21f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,25 +3,27 @@
         final Set<String> configs = Sets.newHashSet("a config", "another config", "yet another config");
         final Set<String> dependencies = Sets.newHashSet("a dependency");
         final String id = UUID.randomUUID().toString();
-        final Cluster cluster = new Cluster.Builder(
-            CLUSTER_1_NAME,
-            CLUSTER_1_USER,
-            CLUSTER_1_VERSION,
-            ClusterStatus.OUT_OF_SERVICE
+        final ClusterRequest cluster = new ClusterRequest.Builder(
+            new ClusterMetadata.Builder(
+                CLUSTER_1_NAME,
+                CLUSTER_1_USER,
+                ClusterStatus.OUT_OF_SERVICE
+            )
+                .withVersion(CLUSTER_1_VERSION)
+                .build()
         )
-            .withId(id)
-            .withConfigs(configs)
-            .withDependencies(dependencies)
+            .withRequestedId(id)
+            .withResources(new ExecutionEnvironment(configs, dependencies, null))
             .build();
         this.service.createCluster(cluster);
         final Cluster created = this.service.getCluster(id);
         Assert.assertNotNull(created);
-        Assert.assertEquals(id, created.getId().orElseThrow(IllegalArgumentException::new));
-        Assert.assertEquals(CLUSTER_1_NAME, created.getName());
-        Assert.assertEquals(CLUSTER_1_USER, created.getUser());
-        Assert.assertEquals(ClusterStatus.OUT_OF_SERVICE, created.getStatus());
-        Assert.assertEquals(3, created.getConfigs().size());
-        Assert.assertEquals(1, created.getDependencies().size());
+        Assert.assertEquals(id, created.getId());
+        Assert.assertEquals(CLUSTER_1_NAME, created.getMetadata().getName());
+        Assert.assertEquals(CLUSTER_1_USER, created.getMetadata().getUser());
+        Assert.assertEquals(ClusterStatus.OUT_OF_SERVICE, created.getMetadata().getStatus());
+        Assert.assertEquals(3, created.getResources().getConfigs().size());
+        Assert.assertEquals(1, created.getResources().getDependencies().size());
         this.service.deleteCluster(id);
         try {
             this.service.getCluster(id);
