diff --git a/old_product.java b/new_product.java
index a315d00..3729c19 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,5 @@
-@Override
-	public boolean accept(Class<?> scope, String path)
+public boolean accept(Class<?> scope, String absolutePath)
 	{
-		String absolutePath = Packages.absolutePath(scope, path);
+		// path is already absolute
 		return acceptAbsolutePath(absolutePath);
 	}
\ No newline at end of file
