diff --git a/old_test.java b/new_test.java
index ac0bd51..c0de16c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,15 +3,16 @@
         final String resourceURI = getURI() + "v1/scopes";
 
         // Test to list scopes.
-        List<String> scopesList = Arrays.asList("scope1", "scope2");
+        List<String> scopesList = Arrays.asList("scope1", "scope2", "scope3");
         when(mockControllerService.listScopes()).thenReturn(CompletableFuture.completedFuture(scopesList));
         Response response = addAuthHeaders(client.target(resourceURI).request()).buildGet().invoke();
         assertEquals("List Scopes response code", 200, response.getStatus());
         assertTrue(response.bufferEntity());
         final ScopesList scopesList1 = response.readEntity(ScopesList.class);
-        assertEquals("List count", scopesList1.getScopes().size(), 2);
+        assertEquals("List count", scopesList1.getScopes().size(), 3);
         assertEquals("List element", scopesList1.getScopes().get(0).getScopeName(), "scope1");
         assertEquals("List element", scopesList1.getScopes().get(1).getScopeName(), "scope2");
+        assertEquals("List element", scopesList1.getScopes().get(2).getScopeName(), "scope3");
         response.close();
 
         // Test for list scopes failure.
