diff --git a/old_test.java b/new_test.java
index 44793d8..c9a71a6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testOnLoadJob() throws GenieException {
+    public void testOnLoadJob() throws GeniePreconditionException {
         this.job.onCreateOrUpdateJob();
         final String clusterCriteriasString = this.job.getClusterCriteriasString();
         final String commandCriteriaString = this.job.getCommandCriteriaString();
