diff --git a/old_product.java b/new_product.java
index 930439f..d12228a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @PostLoad
-    protected void onLoadJob() throws GenieException {
+    protected void onLoadJob() throws GeniePreconditionException {
         this.clusterCriterias = this.stringToClusterCriterias(this.clusterCriteriasString);
         this.commandCriteria = this.stringToCommandCriteria(this.commandCriteriaString);
     }
\ No newline at end of file
