diff --git a/old_product.java b/new_product.java
index e6ba259..4746c0a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public AssociativeArray estimateEfficiency(Map<Object, DeaRecord> id2DeaRecordMapDatabase, Map<Object, DeaRecord> id2DeaRecordMapEvaluation) {
+public AssociativeArray estimateEfficiency(Map<Object, DeaRecord> id2DeaRecordMapInput, Map<Object, DeaRecord> id2DeaRecordMapOutput) {
         AssociativeArray evaluatedResults = new AssociativeArray();
         
         List<LPSolver.LPConstraint> constraints = new ArrayList<>();
@@ -6,7 +6,7 @@ public AssociativeArray estimateEfficiency(Map<Object, DeaRecord> id2DeaRecordMa
         //initialize the constraints list
         Integer totalColumns = null;
         boolean hasInput = false;
-        for(Map.Entry<Object, DeaRecord> entry : id2DeaRecordMapDatabase.entrySet()) {
+        for(Map.Entry<Object, DeaRecord> entry : id2DeaRecordMapInput.entrySet()) {
             DeaRecord currentRecord = entry.getValue();
             int currentColumns = currentRecord.getInput().length; //add the size of input array
             boolean currentHasInput=(currentColumns > 0); //check if the input is defined
@@ -58,7 +58,7 @@ public AssociativeArray estimateEfficiency(Map<Object, DeaRecord> id2DeaRecordMa
         }
         
         
-        for(Map.Entry<Object, DeaRecord> entry : id2DeaRecordMapEvaluation.entrySet()) {
+        for(Map.Entry<Object, DeaRecord> entry : id2DeaRecordMapOutput.entrySet()) {
             Object currentRecordId = entry.getKey();
             DeaRecord currentRecord = entry.getValue();
             
