diff --git a/old_test.java b/new_test.java
index b07c803..c1001c6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @SuppressWarnings("cast")
     @Test
     public void filter() {
-        List<Integer> ints = new ArrayList<Integer>();
+        final List<Integer> ints = new ArrayList<Integer>();
         ints.add(1);
         ints.add(2);
         ints.add(3);
         ints.add(3);
-        Iterable<Integer> iterable = ints;
+        final Iterable<Integer> iterable = ints;
         assertTrue(CollectionUtils.filter(iterable, EQUALS_TWO));
         assertEquals(1, ints.size());
         assertEquals(2, (int) ints.get(0));
