diff --git a/old_product.java b/new_product.java
index 612909e..6893088 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public static <T> boolean filter(Iterable<T> collection, Predicate<? super T> predicate) {
+public static <T> boolean filter(final Iterable<T> collection, final Predicate<? super T> predicate) {
         boolean result = false;
         if (collection != null && predicate != null) {
-            for (Iterator<T> it = collection.iterator(); it.hasNext();) {
+            for (final Iterator<T> it = collection.iterator(); it.hasNext();) {
                 if (!predicate.evaluate(it.next())) {
                     it.remove();
                     result = true;
