diff --git a/old_test.java b/new_test.java
index 6a25efd..34ce5a3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void foldLeft1Test() {
     query("hof:fold-left1(1 to 10, function($x, $y) { $x + $y })", "55");
-    error("hof:fold-left1((), function($x, $y) { $x + $y })", Err.INVEMPTY);
+    error("hof:fold-left1((), function($x, $y) { $x + $y })", Err.SEQEMPTY);
     // should be unrolled and evaluated at compile time
     check("hof:fold-left1(1 to 9, function($a,$b) {$a+$b})",
         "45",
