diff --git a/old_product.java b/new_product.java
index 15ff8fa..1857a77 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 private Value foldLeft1(final QueryContext qc) throws QueryException {
     final FItem f = withArity(1, 2, qc);
-    final Iter xs = exprs[0].iter(qc);
+    final Iter iter = exprs[0].iter(qc);
 
-    Value sum = checkNoEmpty(xs.next());
-    for(Item x; (x = xs.next()) != null;) sum = f.invokeValue(qc, info, sum, x);
+    Value sum = checkNoEmpty(iter.next());
+    for(Item it; (it = iter.next()) != null;) sum = f.invokeValue(qc, info, sum, it);
     return sum;
   }
\ No newline at end of file
