diff --git a/old_test.java b/new_test.java
index 2a37e38..fe6cbd7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,10 +3,17 @@
       // Get URI for child VM
       URI vmURI = Iterables.getOnlyElement(vApp.getChildren().getVms()).getHref();
 
+      // Look up a network in the Vdc
+      Set<Reference> networks = vdc.getAvailableNetworks().getNetworks();
+      Reference network = Iterables.getLast(networks);
+
       // Copy existing section and update fields
       NetworkConnectionSection oldSection = vAppClient.getNetworkConnectionSection(vmURI);
       NetworkConnectionSection newSection = oldSection.toBuilder()
-            .info("Changed NetworkConnectionSection Info")
+            .networkConnection(NetworkConnection.builder()
+                  .ipAddressAllocationMode(IpAddressAllocationMode.DHCP.toString())
+                  .network(network.getName())
+                  .build())
             .build();
 
       // The method under test
@@ -19,8 +26,8 @@
       // Check the retrieved object is well formed
       checkNetworkConnectionSection(modified);
 
-      // Check the modified section fields are set correctly
-      assertEquals(modified.getInfo(), newSection.getInfo());
+      // Check the modified section has an extra network connection
+      assertEquals(modified.getNetworkConnections().size(), newSection.getNetworkConnections().size() + 1);
 
       // Check the section was modified correctly
       assertEquals(modified, newSection, String.format(ENTITY_EQUAL, "NetworkConnectionSection"));
