diff --git a/old_product.java b/new_product.java
index 07c2b0a..d97d9e3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-static <R extends Chrono<R>> ChronoLocalDateTimeImpl<R> of(ChronoLocalDate<R> date, LocalTime time) {
+static <R extends Chronology<R>> ChronoLocalDateTimeImpl<R> of(ChronoLocalDate<R> date, LocalTime time) {
         return new ChronoLocalDateTimeImpl<>(date, time);
     }
\ No newline at end of file
