diff --git a/old_product.java b/new_product.java
index cbfedbd..9cfe8e7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,17 +1,11 @@
-String get(final String query) throws IOException {
-    final URL url = new URL(ROOT + "?" + query);
+final String get(final String query) throws IOException {
+    final URL url = new URL(ROOT + query);
     final HttpURLConnection conn = (HttpURLConnection) url.openConnection();
-    final int code = conn.getResponseCode();
-    final InputStream is = conn.getInputStream();
-    final ByteArrayOutputStream bais = new ByteArrayOutputStream();
-    int i;
-    while((i = is.read()) != -1) bais.write(i);
-    is.close();
-    conn.disconnect();
-    final String result = bais.toString().replaceAll("\r?\n *", "");
-    if(code != HttpURLConnection.HTTP_OK) {
-      System.out.println("???????????");
-      throw new JaxRxException(code, result);
+    try {
+      return read(conn.getInputStream()).replaceAll("\r?\n *", "");
+    } catch(final IOException ex) {
+      throw new IOException(read(conn.getErrorStream()));
+    } finally {
+      conn.disconnect();
     }
-    return result;
   }
\ No newline at end of file
