diff --git a/old_product.java b/new_product.java
index f0dfc7d..a370c2b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,8 @@ private static String get(final String query) throws IOException {
     // create connection
     final HttpURLConnection conn = (HttpURLConnection) url.openConnection();
     try {
-      return read(conn.getInputStream()).replaceAll("\r?\n *", "");
+      final BufferInput bi = new BufferInput(conn.getInputStream());
+      return read(bi).replaceAll("\r?\n *", "");
     } catch(final IOException ex) {
       throw error(conn, ex);
     } finally {
