diff --git a/old_product.java b/new_product.java
index c4bbf7b..3d92009 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,5 @@
 private void parse() throws QueryIOException {
     skipWs();
-    if(spec == JsonSpec.RFC4627 && !(curr() == '{' || curr() == '['))
-      throw error("Expected '{' or '[', found %", rest());
     value();
     if(more()) throw error("Unexpected trailing content: %", rest());
   }
\ No newline at end of file
