diff --git a/old_test.java b/new_test.java
index d4ca5b5..fbcf3e2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-  public void parse() {
+  public void parseXml() {
     // default output
     parse("[]", "", "<json type=\"array\"/>");
     parse("{}", "", "<json type=\"object\"/>");
@@ -50,5 +50,5 @@
     parseError("{ \"\" : 0.1. }", "");
     parseError("{ \"\" : 0.1e }", "");
     parseError("{ \"a\" : 0 }}", "");
-    parseError("{ \"a\" : 0, }", "'format':'RFC4627'");
+    parseError("{ \"a\" : 0, }", "'liberal':false()");
   }
\ No newline at end of file
