diff --git a/old_product.java b/new_product.java
index cb458e6..f03c6e5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,9 @@
-private Item delete(final File path, final QueryContext ctx)
+private Str baseName(final File path, final QueryContext ctx)
       throws QueryException {
 
-    final boolean rec = optionalBool(1, ctx);
-    if(path.exists()) {
-      if(rec) recDelete(path);
-      else delete(path);
-    }
-    return null;
+    final String suf = expr.length < 2 ? null : string(checkStr(expr[1], ctx));
+    String pth = path.getName();
+    if(suf != null && pth.endsWith(suf))
+      pth = pth.substring(0, pth.length() - suf.length());
+    return Str.get(pth);
   }
\ No newline at end of file
