diff --git a/old_test.java b/new_test.java
index d01aafc..23ac124 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,9 +2,9 @@
   public void testDelete() throws QueryException {
     final String fun = check(FunDef.DELETE);
     query("file:create-directory('" + PATH3 + "')");
-    error(fun + "('" + PATH1 + "')", Err.DIRNOTEMPTY);
     query(fun + "('" + PATH3 + "')");
     query("file:create-directory('" + PATH3 + "')");
     query("file:write('" + PATH4 + "', ())");
-    query(fun + "('" + PATH1 + "', true())");
+    query(fun + "('" + PATH1 + "')");
+    error(fun + "('" + PATH1 + "')", Err.PATHNOTEXISTS);
   }
\ No newline at end of file
