diff --git a/old_product.java b/new_product.java
index 599178a..982dff1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,6 @@
 List<Command> addCommandsForCluster(
+            @NotBlank(message = "No cluster id entered. Unable to add commands.")
             final String id,
-            final List<Command> commands) throws GenieException;
\ No newline at end of file
+            @NotEmpty(message = "No commands entered. Unable to add commands.")
+            final List<Command> commands
+    ) throws GenieException;
\ No newline at end of file
