diff --git a/old_test.java b/new_test.java
index be93df0..5347b56 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,18 +4,18 @@
                 new Command(
                         "name",
                         "user",
+                        "23.1.0",
                         CommandStatus.ACTIVE,
-                        "pig",
-                        "23.1.0"
+                        "pig"
                 )
         );
         final Command command2 = this.commandService.createCommand(
                 new Command(
                         "name2",
                         "user2",
+                        "23.1.1",
                         CommandStatus.INACTIVE,
-                        "pig2",
-                        "23.1.1"
+                        "pig2"
                 )
         );
         final List<Command> newCommands = new ArrayList<>();
