diff --git a/old_product.java b/new_product.java
index 76ae536..0101074 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,14 @@
-public static Map<String,Object> omit(Map<String,Object> params, String... toOmit) {
-		if (params == null) return null;
-		if (toOmit == null || toOmit.length == 0) return params;
+public static Map<String, Object> omit(Map<String, Object> params, String... toOmit) {
+    if (params == null)
+      return null;
+    if (toOmit == null || toOmit.length == 0)
+      return params;
 
-		Map<String,Object> ret = new HashMap<String,Object>();
+    final Map<String, Object> ret = new HashMap<String, Object>();
 
-		for (String key : params.keySet()) {
-			if (!ArrayUtils.contains(toOmit, key))
-				ret.put(key, params.get(key));
-		}
-		return ret;
-	}
\ No newline at end of file
+    for (final String key : params.keySet()) {
+      if (!ArrayUtils.contains(toOmit, key))
+        ret.put(key, params.get(key));
+    }
+    return ret;
+  }
\ No newline at end of file
