diff --git a/old_test.java b/new_test.java
index c107eeb..1b5b37f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,14 @@
 @Test
-	public void testOmit() {
-		Map<String, Object> params = createMap();
-		
-		
-		Map<String,Object> omitted = RequestUtil.omit(params, "A");
-	
-		Assert.assertTrue(omitted.keySet().containsAll(Lists.newArrayList("B","C","D")));
-		Assert.assertTrue(omitted.values().containsAll(Lists.newArrayList(2,3,4)));
-		
-		omitted = RequestUtil.omit(params, "F");
-		Assert.assertTrue(omitted.keySet().containsAll(Lists.newArrayList("A","B","C","D")));
-		Assert.assertTrue(omitted.values().containsAll(Lists.newArrayList(1,2,3,4)));
-	}
\ No newline at end of file
+  public void testOmit() {
+    final Map<String, Object> params = createMap();
+
+
+    Map<String, Object> omitted = RequestUtil.omit(params, "A");
+
+    Assert.assertTrue(omitted.keySet().containsAll(Lists.newArrayList("B", "C", "D")));
+    Assert.assertTrue(omitted.values().containsAll(Lists.newArrayList(2, 3, 4)));
+
+    omitted = RequestUtil.omit(params, "F");
+    Assert.assertTrue(omitted.keySet().containsAll(Lists.newArrayList("A", "B", "C", "D")));
+    Assert.assertTrue(omitted.values().containsAll(Lists.newArrayList(1, 2, 3, 4)));
+  }
\ No newline at end of file
