diff --git a/old_product.java b/new_product.java
index a8f6f61..4c305ed 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-@RequestMapping(value = "CTL/saveSchema", params = { "body", "scope", "applicationId" }, method = RequestMethod.POST)
+@RequestMapping(value = "CTL/saveSchema", params = { "body" }, method = RequestMethod.POST)
     @ResponseBody
     public CTLSchemaInfoDto saveCTLSchema(@RequestParam String body, 
-            @RequestParam(required = false) CTLSchemaScopeDto scope,  
+            @RequestParam(required = false) String scope,  
             @RequestParam(required = false) String applicationId) throws KaaAdminServiceException {
-        return kaaAdminService.saveCTLSchema(body, scope, applicationId);
+        return kaaAdminService.saveCTLSchema(body, isEmpty(scope) ? null : CTLSchemaScopeDto.valueOf(scope.toUpperCase()), applicationId);
     }
\ No newline at end of file
