diff --git a/old_test.java b/new_test.java
index 1ec8ac8..d52e8ed 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,17 @@
     wordsAsMap.put("words", new Word[] {newWord});
     server.enqueue(new MockResponse().addHeader(CONTENT_TYPE, HttpMediaType.APPLICATION_JSON).setBody("{}"));
 
-    service.addWords(id, newWord).execute();
+    AddWordsOptions addOptions = new AddWordsOptions.Builder()
+        .customizationId(id)
+        .customWords(new CustomWords.Builder()
+            .addWords(new CustomWord.Builder()
+                .word(newWord.getWord())
+                .displayAs(newWord.getDisplayAs())
+                .soundsLike(newWord.getSoundsLike())
+                .build())
+            .build())
+        .build();
+    service.addWords(addOptions).execute();
     final RecordedRequest request = server.takeRequest();
 
     assertEquals("POST", request.getMethod());
