diff --git a/old_test.java b/new_test.java
index bfec8d4..295a16d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,5 +4,5 @@
     final ANode root = new DBNode(io, CONTEXT.prop);
     final QueryProcessor qp = new QueryProcessor("/*:a/*:b", root, CONTEXT);
     final ANode sub = (ANode) qp.iter().next();
-    DataBuilder.stripNS(sub, token("a"), qp.ctx);
+    DataBuilder.stripNS(sub, token("a"), CONTEXT);
   }
\ No newline at end of file
