diff --git a/old_product.java b/new_product.java
index 9c4855a..fd4e12d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,7 @@
-public static ANode stripNS(final ANode node, final byte[] ns,
-      final QueryContext ctx) {
+public static ANode stripNS(final ANode node, final byte[] ns, final Context ctx) {
     if(node.type != NodeType.ELM) return node;
 
-    final MemData md = new MemData(ctx.context.prop);
+    final MemData md = new MemData(ctx.prop);
     final DataBuilder db = new DataBuilder(md);
     db.build(node);
 
