diff --git a/old_product.java b/new_product.java
index d7c2f29..28b4912 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,13 +2,13 @@ public static TrifocalTensor createTrifocal( Se3_F64 P2 , Se3_F64 P3 , TrifocalT
 		if( ret == null )
 			ret = new TrifocalTensor();
 
-		RowMatrix_F64 R2 = P2.getR();
-		RowMatrix_F64 R3 = P3.getR();
+		DMatrixRMaj R2 = P2.getR();
+		DMatrixRMaj R3 = P3.getR();
 		Vector3D_F64 T2 = P2.getT();
 		Vector3D_F64 T3 = P3.getT();
 
 		for( int col = 0; col < 3; col++ ) {
-			RowMatrix_F64 T = ret.getT(col);
+			DMatrixRMaj T = ret.getT(col);
 
 			int index = 0;
 			for( int i = 0; i < 3; i++ ) {
