diff --git a/old_test.java b/new_test.java
index 052e498..567fce0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -22,6 +22,6 @@
 
 			SimpleMatrix expected = ai.mult(b4.transpose()).minus(a4.mult(bi.transpose()));
 
-			assertTrue(MatrixFeatures_R64.isIdentical(expected.matrix_F64(),found.getT(i),1e-8));
+			assertTrue(MatrixFeatures_DDRM.isIdentical(expected.matrix_F64(),found.getT(i),1e-8));
 		}
 	}
\ No newline at end of file
