diff --git a/old_test.java b/new_test.java
index 02276f2..3f0ffa4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -33,7 +33,7 @@
 		PersonName copy = PersonName.newInstance(pn);
 		
 		Assert.assertEquals(personNameId, copy.getPersonNameId());
-		Assert.assertEquals(preferred, copy.getPreferred().booleanValue());
+		Assert.assertEquals(preferred, copy.getPreferred());
 		Assert.assertEquals(prefix, copy.getPrefix());
 		Assert.assertEquals(person, copy.getPerson());
 		Assert.assertEquals(givenName, copy.getGivenName());
@@ -43,7 +43,7 @@
 		Assert.assertEquals(familyName2, copy.getFamilyName2());
 		Assert.assertEquals(familyNameSuffix, copy.getFamilyNameSuffix());
 		Assert.assertEquals(degree, copy.getDegree());
-		Assert.assertEquals(voided, copy.getVoided().booleanValue());
+		Assert.assertEquals(voided, copy.getVoided());
 		Assert.assertEquals(voidedBy, copy.getVoidedBy());
 		Assert.assertEquals(voidReason, copy.getVoidReason());
 	}
\ No newline at end of file
