diff --git a/old_product.java b/new_product.java
index 1b6e11a..3102bd3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -42,10 +42,10 @@ public static PersonName newInstance(PersonName pn) {
 		}
 		
 		if (pn.getPreferred() != null) {
-			newName.setPreferred(pn.getPreferred().booleanValue());
+			newName.setPreferred(pn.getPreferred());
 		}
 		if (pn.getVoided() != null) {
-			newName.setVoided(pn.getVoided().booleanValue());
+			newName.setVoided(pn.getVoided());
 		}
 		
 		newName.setPerson(pn.getPerson());
