diff --git a/old_product.java b/new_product.java
index 33a91c4..16bccca 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1,6 @@
-Timestamp calculatePreparationDateOrNull(IContextAware context, boolean isSOTrx, Timestamp dateOrdered, Timestamp datePromised, int bpartnerLocationId);
\ No newline at end of file
+ZonedDateTime calculatePreparationDateOrNull(
+			IContextAware context,
+			SOTrx soTrx,
+			ZonedDateTime calculationTime,
+			ZonedDateTime datePromised,
+			BPartnerLocationId bpartnerLocationId);
\ No newline at end of file
