diff --git a/old_test.java b/new_test.java
index bc4ad1d..dcd5204 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
 	public final void testProcess9() {
 		Relation testRelation;
 		
-		testRelation = new Relation(3456, 0, new Date(), new OsmUser(12, "OsmosisTest"), 0);
+		testRelation = new Relation(new CommonEntityData(3456, 0, new Date(), new OsmUser(12, "OsmosisTest"), 0));
 		testRelation.getMembers().add(new RelationMember(1234, EntityType.Node, "role1"));
 		testRelation.getTags().add(new Tag("test_key1", "test_value1"));
 		
