diff --git a/old_test.java b/new_test.java
index f08aa48..5e863b1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,15 +12,23 @@
     TachyonFile fileC = TachyonFSTestUtils.createByteFile(mTfs, "/testRoot/testFileC",
         CacheType.NO_CACHE, UnderStorageType.PERSIST, 30);
     files[3] = mTfs.getInfo(fileC);
-    mFsShell.run(new String[] {"ls", "/testRoot"});
+    mFsShell.run(new String[] {"lsr", "/testRoot"});
     String expected = "";
     String format = "%-10s%-25s%-15s%-5s\n";
-    expected += String.format(format, FormatUtils.getSizeFromBytes(10),
-        TFsShell.convertMsToDate(files[0].getCreationTimeMs()), "In Memory", "/testRoot/testFileA");
-    expected += String.format(format, FormatUtils.getSizeFromBytes(0),
-        TFsShell.convertMsToDate(files[1].getCreationTimeMs()), "", "/testRoot/testDir");
-    expected += String.format(format, FormatUtils.getSizeFromBytes(30),
-        TFsShell.convertMsToDate(files[3].getCreationTimeMs()), "Not In Memory",
-        "/testRoot/testFileC");
+    expected +=
+        String.format(format, FormatUtils.getSizeFromBytes(10),
+            TFsShell.convertMsToDate(files[0].getCreationTimeMs()), "In Memory",
+            "/testRoot/testFileA");
+    expected +=
+        String.format(format, FormatUtils.getSizeFromBytes(0),
+            TFsShell.convertMsToDate(files[1].getCreationTimeMs()), "", "/testRoot/testDir");
+    expected +=
+        String.format(format, FormatUtils.getSizeFromBytes(20),
+            TFsShell.convertMsToDate(files[2].getCreationTimeMs()), "In Memory",
+            "/testRoot/testDir/testFileB");
+    expected +=
+        String.format(format, FormatUtils.getSizeFromBytes(30),
+            TFsShell.convertMsToDate(files[3].getCreationTimeMs()), "Not In Memory",
+            "/testRoot/testFileC");
     Assert.assertEquals(expected, mOutput.toString());
   }
\ No newline at end of file
