diff --git a/old_product.java b/new_product.java
index 7bd0053..d326c08 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,5 @@
 public int lsr(TachyonURI path) throws IOException {
-    TachyonFS tachyonClient = createFS(path);
-    List<FileInfo> files = tachyonClient.listStatus(path);
-    Collections.sort(files);
+    List<FileInfo> files = listStatusSortedByIncreasingCreationTime(path);
     String format = "%-10s%-25s%-15s%-5s%n";
     for (FileInfo file : files) {
       String inMemory = "";
