diff --git a/old_product.java b/new_product.java
index caf3309..fbd0575 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ private Data create(final QueryInput input, final boolean single, final InputInf
 
     // do not check for existence of input if user has no read permissions
     if(!context.user().has(Perm.READ))
-      throw BXXQ_PERM_X.get(info, Util.info(Text.PERM_REQUIRED_X, Perm.READ));
+      throw XQUERY_PERMISSION1_X.get(info, Util.info(Text.PERM_REQUIRED_X, Perm.READ));
 
     // check if input points to a single file
     final IO io = input.io;
