diff --git a/old_product.java b/new_product.java
index 494e04c..4936edb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,6 @@
 public static void writeJavaScript(final Response response, final CharSequence text)
 	{
-		writeJavaScript(response, text, null);
+		AttributeMap attributes = new AttributeMap();
+		attributes.putAttribute(ATTR_TYPE, "text/javascript");
+		writeInlineScript(response, text, attributes);
 	}
\ No newline at end of file
