diff --git a/old_product.java b/new_product.java
index 08226fd..bf9501a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-void sendChallengeDuringNegotiate(final String protocol, final ServletResponse response, final byte[] out) {
+void sendChallengeDuringNegotiate(final String protocol, final ServletResponse response, final byte[] out)
+            throws IOException {
         final List<String> protocolsList = new ArrayList<>();
         protocolsList.add(protocol);
         this.sendChallenge(protocolsList, response, out);
