diff --git a/old_test.java b/new_test.java
index 2ef023a..6732a48 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testSendChallengeDuringNegotiate() {
+    public void testSendChallengeDuringNegotiate() throws IOException {
 
         final String myProtocol = "myProtocol";
 
@@ -13,8 +13,7 @@
 
         Assertions.assertEquals("keep-alive", this.response.headers.get("Connection"));
 
-        Assertions.assertEquals(HttpServletResponse.SC_UNAUTHORIZED, this.response.sc);
-        Assertions.assertEquals(0, this.response.errorCode);
+        Assertions.assertEquals(HttpServletResponse.SC_UNAUTHORIZED, this.response.errorCode);
 
         Assertions.assertFalse(this.response.isFlushed);
     }
\ No newline at end of file
