diff --git a/old_test.java b/new_test.java
index 74438d7..0b90b5c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,12 +2,16 @@
   public void testExecuteDMLUpdates() throws Exception {
     Method removeNagiosService = UpgradeCatalog200.class.getDeclaredMethod("removeNagiosService");
     Method updateHiveDatabaseType = UpgradeCatalog200.class.getDeclaredMethod("updateHiveDatabaseType");
+    Method addNewConfigurationsFromXml = AbstractUpgradeCatalog.class.getDeclaredMethod
+        ("addNewConfigurationsFromXml");
 
     UpgradeCatalog200 upgradeCatalog = createMockBuilder(
-        UpgradeCatalog200.class).addMockedMethod(removeNagiosService).addMockedMethod(updateHiveDatabaseType).createMock();
+        UpgradeCatalog200.class).addMockedMethod(removeNagiosService).addMockedMethod(updateHiveDatabaseType).addMockedMethod(addNewConfigurationsFromXml).createMock();
 
     upgradeCatalog.removeNagiosService();
     expectLastCall().once();
+    upgradeCatalog.addNewConfigurationsFromXml();
+    expectLastCall();
     upgradeCatalog.updateHiveDatabaseType();
     expectLastCall().once();
 
