diff --git a/old_test.java b/new_test.java
index 963d140..31feb63 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,7 @@
 @Test
     public void testSubscribe() {
-        Logger.getLogger(ControllerTest.class.getName()).info("subscribe");
+
+        LOG.info("subscribe");
 
         final String callback = "http://localhost/doNothing";
         final String topic = "http://feeds.feedburner.com/screaming-penguin";
@@ -28,7 +29,7 @@
             fail();
         } catch (final HttpStatusCodeException e) {
             assertEquals(400, e.getStatus());
-            Logger.getLogger(ControllerTest.class.getName()).info(e.getMessage());
+            LOG.info(e.getMessage());
         }
 
         try {
@@ -36,7 +37,7 @@
             fail();
         } catch (final HttpStatusCodeException e) {
             assertEquals(400, e.getStatus());
-            Logger.getLogger(ControllerTest.class.getName()).info(e.getMessage());
+            LOG.info(e.getMessage());
         }
 
         try {
@@ -44,7 +45,7 @@
             fail();
         } catch (final HttpStatusCodeException e) {
             assertEquals(400, e.getStatus());
-            Logger.getLogger(ControllerTest.class.getName()).info(e.getMessage());
+            LOG.info(e.getMessage());
         }
 
         // test general exception
