diff --git a/old_product.java b/new_product.java
index f3344de..6e999fe 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public Boolean subscribe(final String callback, final String topic, final String verify, final long lease_seconds, final String secret,
             final String verify_token) {
-        Logger.getLogger(Hub.class.getName()).log(Level.FINE, "{0} wants to subscribe to {1}", new Object[] { callback, topic });
+        LOG.debug("{} wants to subscribe to {}", callback, topic);
         try {
             try {
                 assert callback != null : "Callback URL is required.";
@@ -27,8 +27,7 @@ public Boolean subscribe(final String callback, final String topic, final String
                 @Override
                 public void onVerify(final boolean verified) {
                     if (verified) {
-                        Logger.getLogger(Hub.class.getName()).log(Level.FINE, "Verified {0} subscribed to {1}",
-                                new Object[] { subscriber.getCallback(), subscriber.getTopic() });
+                        LOG.debug("Verified {} subscribed to {}", subscriber.getCallback(), subscriber.getTopic());
                         dao.addSubscriber(subscriber);
                     }
                 }
