diff --git a/old_product.java b/new_product.java
index ad28c84..3657b1d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,7 @@ private static String add(final Parser parser, final Context ctx,
       throws BaseXException {
 
     final Performance p = new Performance();
+    final String input = name == null ? parser.src.path() : name;
     final String path = target + (target.isEmpty() ? "/" : "") +
         (name == null ? parser.src.name() : name);
 
@@ -32,7 +33,7 @@ private static String add(final Parser parser, final Context ctx,
         ctx.update();
         data.flush();
       }
-      return Util.info(PATHADDED, path, p);
+      return Util.info(parser.info() + PATHADDED, input, p);
     } catch(final IOException ex) {
       Util.debug(ex);
       throw new BaseXException(ex);
