diff --git a/old_product.java b/new_product.java
index e731165..fdef97c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public static Directory getDirectory() throws IOException {
+public Directory getDirectory() throws IOException {
         String fileName = Settings.getString(Settings.KEYS.CPE_INDEX);
         File path = new File(fileName);
-        Directory directory = FSDirectory.open(path);
+        Directory dir = FSDirectory.open(path);
 
-        return directory;
+        return dir;
     }
\ No newline at end of file
