diff --git a/old_test.java b/new_test.java
index d6da719..f91bd48 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,8 @@
 @Test
     public void testGetDirectory() throws Exception {
         System.out.println("getDirectory");
-        Directory result = Index.getDirectory();
+        Index index = new Index();
+        Directory result = index.getDirectory();
         String exp = "\\target\\store\\cpe";
         // TODO review the generated test code and remove the default call to fail.
         assertTrue(result.toString().contains(exp));
