diff --git a/old_test.java b/new_test.java
index 29d3186..1efd73f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void test_chrono_byName() {
-        Chrono<ISOChrono> c = ISOChrono.INSTANCE;
-        Chrono<?> test = Chrono.of("ISO");
+        Chronology<ISOChronology> c = ISOChronology.INSTANCE;
+        Chronology<?> test = Chronology.of("ISO");
         Assert.assertNotNull(test, "The ISO calendar could not be found byName");
         Assert.assertEquals(test.getId(), "ISO", "ID mismatch");
         Assert.assertEquals(test.getCalendarType(), "iso8601", "Type mismatch");
