diff --git a/old_product.java b/new_product.java
index a36f4f5..35afa32 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,6 +4,6 @@ private Str textEntry(final QueryContext ctx) throws QueryException {
     try {
       return Str.get(new NewlineInput(io).encoding(enc).content());
     } catch(final IOException ex) {
-      throw ZIPFAIL.thrw(info, ex.getMessage());
+      throw ZIP_FAIL.thrw(info, ex.getMessage());
     }
   }
\ No newline at end of file
