diff --git a/old_test.java b/new_test.java
index c5b40b6..e76e38a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     check(_ZIP_TEXT_ENTRY);
     query(_ZIP_TEXT_ENTRY.args(ZIP, ENTRY1));
     query(_ZIP_TEXT_ENTRY.args(ZIP, ENTRY1, "US-ASCII"));
-    error(_ZIP_TEXT_ENTRY.args(ZIP, ENTRY1, "xyz"), Err.ZIPFAIL);
+    error(_ZIP_TEXT_ENTRY.args(ZIP, ENTRY1, "xyz"), Err.ZIP_FAIL);
     // newlines are removed from the result..
     contains(_ZIP_TEXT_ENTRY.args(ZIP, ENTRY1), "aaboutab");
   }
\ No newline at end of file
