diff --git a/old_product.java b/new_product.java
index 61b53d3..e61d00d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,9 +10,9 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 		Concept concept = order.getConcept();
 		if (!isDiscontinueOrReviseOrder(order)) {
 			List<Order> activeOrders = getActiveOrders(order.getPatient(), null, order.getCareSetting(), null);
-			for (Order o : activeOrders) {
-				if (o.getConcept().equals(concept)) {
-					throw new APIException("Cannot have more than one active order for the same concept and care setting");
+			for (Order activeOrder : activeOrders) {
+				if (order.hasSameOrderableAs(activeOrder)) {
+					throw new APIException("Cannot have more than one active order for the same orderable and care setting");
 				}
 			}
 		}
