diff --git a/old_test.java b/new_test.java
index cbe1805..becca25 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,6 +16,6 @@
 		order.setCareSetting(duplicateOrder.getCareSetting());
 		
 		expectedException.expect(APIException.class);
-		expectedException.expectMessage("Cannot have more than one active order for the same concept and care setting");
+		expectedException.expectMessage("Cannot have more than one active order for the same orderable and care setting");
 		orderService.saveOrder(order, null);
 	}
\ No newline at end of file
