diff --git a/old_product.java b/new_product.java
index 61d3c92..c743b39 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static double flesch_kincaid_reading_ease(String strText) {
-        strText = clean_text(strText);
-        return PHPfunctions.round((206.835 - (1.015 * average_words_per_sentence(strText)) - (84.6 * average_syllables_per_word(strText))), 1);
+public static double fleschKincaidReadingEase(String strText) {
+        strText = cleanText(strText);
+        return PHPfunctions.round((206.835 - (1.015 * averageWordsPerSentence(strText)) - (84.6 * averageSyllablesPerWord(strText))), 1);
     }
\ No newline at end of file
