diff --git a/old_test.java b/new_test.java
index 7a7f96b..1477a3a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
         logger.info("flesch_kincaid_reading_ease");
         String strText = TEST_STRING;
         double expResult = -188.4;
-        double result = ReadabilityStatistics.flesch_kincaid_reading_ease(strText);
+        double result = ReadabilityStatistics.fleschKincaidReadingEase(strText);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
