diff --git a/old_product.java b/new_product.java
index 705c2de..7e1902b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
-public static <MONAD,T>  Monad<MONAD,T> sequence(Class c, List<?> seq){
-		return (Monad)AsGenericMonad.asMonad(new ComprehenderSelector().selectComprehender(c).of(1))
-				.flatMap(in-> asMonad(seq.stream()).flatMap(m-> m).unwrap()
-							);
+public static <T1>  AnyM<Stream<T1>> sequence(Stream<AnyM<T1>> seq){
+			return AsGenericMonad.asMonad(Stream.of(1))
+										.flatMap(in-> monad(seq.map(it->it.unwrap()))
+												.flatten().unwrap())
+												.anyM();
 	}
\ No newline at end of file
