diff --git a/old_test.java b/new_test.java
index c0d79e7..b059c00 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,7 @@
                 .collect(Collectors.toList());
        
         
-        AnyM<List<Integer>> futureList = Monads.sequence(CompletableFuture.class,futures).anyM();
+        AnyM<Stream<Integer>> futureList = Monads.sequence(completableFutureToAnyMList(futures));
         
  
         List<Integer> collected = futureList.<CompletableFuture<List<Integer>>>unwrap().join();
