diff --git a/old_product.java b/new_product.java
index e7710f4..171c733 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,13 +2,13 @@ public List<Operator> getInputOperators() {
 		return new AbstractList<Operator>() {
 
 			@Override
-			public Operator get(int index) {
+			public Operator get(final int index) {
 				return Operator.this.inputs.get(index) == null ? null : Operator.this.inputs.get(index).getOperator();
 			}
 
 			@Override
-			public int indexOf(Object o) {
-				ListIterator<Output> e = Operator.this.inputs.listIterator();
+			public int indexOf(final Object o) {
+				final ListIterator<Output> e = Operator.this.inputs.listIterator();
 				while (e.hasNext())
 					if (o == e.next())
 						return e.previousIndex();
