diff --git a/old_test.java b/new_test.java
index f892a12..f3c496a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
 	public void testGetInputOperators() {
-		Operator input1 = new OpImpl(0);
-		Operator input2 = new OpImpl(1);
-		Operator fixture = new OpImpl(0, input1, input2);
+		final Operator input1 = new OpImpl(0);
+		final Operator input2 = new OpImpl(1);
+		final Operator fixture = new OpImpl(0, input1, input2);
 
 		assertSame(input1, fixture.getInputOperators().get(0));
 		assertSame(input2, fixture.getInputOperators().get(1));
