diff --git a/old_test.java b/new_test.java
index 97695ce..152fae4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,5 +5,5 @@
 		Retireable retireable = new Location();
 		retireable.setRetired(true); // make sure isRetired is set
 		handler.handle(retireable, null, null, null);
-		Assert.assertFalse(retireable.isRetired());
+		Assert.assertFalse(retireable.getRetired());
 	}
\ No newline at end of file
