diff --git a/old_product.java b/new_product.java
index a16d84a..3e978e3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@
 	public void handle(Retireable retireableObject, User retiringUser, Date retireDate, String retireReason) {
 		
 		// skip over doing retire stuff if already retired
-		if (!retireableObject.isRetired() || retireableObject.getRetiredBy() == null) {
+		if (!retireableObject.getRetired() || retireableObject.getRetiredBy() == null) {
 			
 			retireableObject.setRetired(true);
 			retireableObject.setRetireReason(retireReason);
