diff --git a/old_product.java b/new_product.java
index 125e451..c9db7bd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @RequiresSession
   public UpgradeEntity findLastUpgradeForCluster(long clusterId) {
     TypedQuery<UpgradeEntity> query = entityManagerProvider.get().createNamedQuery(
-        "UpgradeEntity.findLatestForCluster", UpgradeEntity.class);
+        "UpgradeEntity.findLatestForClusterInDirection", UpgradeEntity.class);
     query.setMaxResults(1);
     query.setParameter("clusterId", clusterId);
     query.setParameter("direction", Direction.UPGRADE);
