diff --git a/old_product.java b/new_product.java
index d703e41..5a5f70c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,10 @@
 public SexGenotypeDataCollection inferSexGenotypes() {
         final SexGenotypeDataCollection col = new SexGenotypeDataCollection();
-        IntStream.range(0, processedReadCounts.columnNames().size())
-                .forEach(si -> col.add(calculateSexGenotypeData(si)));
+        IntStream.range(0, readCounts.columnNames().size())
+                .forEach(si -> {
+                    final SexGenotypeData sampleSexGenotype = calculateSexGenotypeData(si);
+                    logger.info(sampleSexGenotype);
+                    col.add(sampleSexGenotype);
+                });
         return col;
     }
\ No newline at end of file
