diff --git a/old_test.java b/new_test.java
index 63703e4..413fb84 100644
--- a/old_test.java
+++ b/new_test.java
@@ -24,7 +24,7 @@
             while (rs.next()) {
                 String name = rs.getString("COLUMN_NAME");
 
-                assert names.remove(name);
+                assert names.remove(name) : "Unexpected column name " + name;
 
                 if ("NAME".equals(name)) {
                     assert rs.getInt("DATA_TYPE") == VARCHAR;
