diff --git a/old_product.java b/new_product.java
index 5b747b1..da3bf9f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@
         throws SQLException {
         conn.ensureNotClosed();
 
-        final List<JdbcColumnMeta> meta = Arrays.asList(
+        final List<JdbcColumnMeta> meta = asList(
             new JdbcColumnMeta(null, null, "TABLE_CAT", String.class),      // 1
             new JdbcColumnMeta(null, null, "TABLE_SCHEM", String.class),    // 2
             new JdbcColumnMeta(null, null, "TABLE_NAME", String.class),     // 3
@@ -30,7 +30,7 @@
         );
 
         if (!isValidCatalog(catalog))
-            return new JdbcThinResultSet(Collections.<List<Object>>emptyList(), meta);
+            return new JdbcThinResultSet(Collections.emptyList(), meta);
 
         JdbcMetaColumnsResult res = conn.sendRequest(new JdbcMetaColumnsRequest(schemaPtrn, tblNamePtrn, colNamePtrn)).response();
 
