diff --git a/old_product.java b/new_product.java
index f82ab16..2d04f93 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,3 @@
 public List<List<Span>> getTraces() {
-    List<Long> traceIds = storage.spanStore().traceIds();
-    List<List<Span>> result = new ArrayList<>(traceIds.size());
-    for (long traceId : traceIds) {
-      result.add(storage.spanStore().getTrace(traceId));
-    }
-    return result;
+    return storage.spanStore().getTraces(QueryRequest.builder().limit(Integer.MAX_VALUE).build());
   }
\ No newline at end of file
