diff --git a/old_test.java b/new_test.java
index c344c71..6488431 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,6 +12,6 @@
     assertThat(store().getTraces((QueryRequest.builder().serviceName("service").build())))
         .containsExactly(asList(span));
 
-    assertThat(store().getTrace(1L))
+    assertThat(store().getTrace(span.traceIdHigh, span.traceId))
         .containsExactly(span);
   }
\ No newline at end of file
