diff --git a/old_product.java b/new_product.java
index 41352ff..79a0c8e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public QrCode kanji( String message ) {
+public QrCodeEncoder kanji( String message ) {
 		qr.mode = QrCode.Mode.KANJI;
 
 		int lengthBits = getLengthBitsKanji(qr.version);
@@ -10,9 +10,6 @@ public QrCode kanji( String message ) {
 			throw new IllegalArgumentException(ex);
 		}
 
-		packed.resize(lengthBits + 8*bytes.length); // predeclare memory
-		packed.size = 0; // set size to 0 so that append() starts from the front
-
 		// specify the mode
 		packed.append(0b1000,4,false);
 
@@ -35,10 +32,5 @@ public QrCode kanji( String message ) {
 			packed.append(encoded, 13, false);
 		}
 
-		// add the terminator to the bit stream
-		packed.append(0b0000,4,false);
-//		packed.print();
-		bitsToMessage(packed);
-
-		return qr;
+		return this;
 	}
\ No newline at end of file
