diff --git a/old_product.java b/new_product.java
index 3791601..eaa9add 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public CorpusStats getCorpusStats(Corpus corpus) {
-        ConceptInsightsId.validateGenarate(corpus, getAccountId());
-        return executeRequest(version + corpus.getId() + STATS_PATH, null, CorpusStats.class);
-    }
\ No newline at end of file
+public CorpusStats getCorpusStats(final Corpus corpus) {
+		String corpusId = IDValidator.getCorpusId(corpus, getAccountId());
+		return executeRequest(API_VERSION + corpusId + STATS_PATH, null, CorpusStats.class);
+	}
\ No newline at end of file
