diff --git a/old_test.java b/new_test.java
index 3cfe460..f112d1f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,8 +17,8 @@
 		c.setSectorIdentifierUri("https://other.example.net/file_of_redirect_uris.json");
 		c.setTokenEndpointAuthMethod(ClientDetailsEntity.AuthMethod.SECRET_BASIC);
 		c.setJwksUri("https://client.example.org/my_public_keys.jwks");
-		c.setUserInfoEncryptedResponseAlg(new JWEAlgorithmEmbed(JWEAlgorithm.RSA1_5));
-		c.setUserInfoEncryptedResponseEnc(new JWEEncryptionMethodEmbed(EncryptionMethod.A128CBC_HS256));
+		c.setUserInfoEncryptedResponseAlg(JWEAlgorithm.RSA1_5);
+		c.setUserInfoEncryptedResponseEnc(EncryptionMethod.A128CBC_HS256);
 		c.setContacts(ImmutableSet.of("ve7jtb@example.org", "mary@example.org"));
 		c.setRequestUris(ImmutableSet.of("https://client.example.org/rf.txt#qpXaRLh_n93TTR9F252ValdatUQvQiJi5BDub2BeznA"));
 
