diff --git a/old_test.java b/new_test.java
index 4c2ca38..1a6b045 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,9 +10,7 @@
     assertEquals(Arrays.asList(fileId), info.idList());
 
     // verify fastCopy called twice, once per block
-    PowerMockito.verifyStatic(times(2));
-    BufferUtils.fastCopy(any(ReadableByteChannel.class),
-        any(WritableByteChannel.class));
+    assertEquals(2, mCopyCounter.get());
 
     // verify the file is not needed for another persistence
     assertFalse(mManager.needPersistence(fileId));
