diff --git a/old_product.java b/new_product.java
index e2f91e3..ba199d6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -33,7 +33,7 @@ public void persistFile(long fileId, List<Long> blockIds) throws AlluxioExceptio
 
           // write content out
           ReadableByteChannel inputChannel = reader.getChannel();
-          BufferUtils.fastCopy(inputChannel, outputChannel);
+          mChannelCopier.copy(inputChannel, outputChannel);
           reader.close();
         }
       } catch (BlockDoesNotExistException | InvalidWorkerStateException e) {
