diff --git a/old_test.java b/new_test.java
index 80f4a6c..f1b3e3e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,9 @@
 	@Verifies(value = "should always return unique orderNumbers when called multiple times without saving orders", method = "getNewOrderNumber()")
 	public void getNewOrderNumber_shouldAlwaysReturnUniqueOrderNumbersWhenCalledMultipleTimesWithoutSavingOrders()
 	        throws Exception {
+		
+		executeDataSet("org/openmrs/api/include/OrderServiceTest-globalProperties.xml");
+		
 		int N = 50;
 		final Set<String> uniqueOrderNumbers = new HashSet<String>(50);
 		List<Thread> threads = new ArrayList<Thread>();
