diff --git a/old_test.java b/new_test.java
index 38b714b..7529bdc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,8 @@
 @Test
-   public void testGetInUserCount()
-   {
-      System.out.println("getInUserCount");
-      SteppingThreadGroup instance = new SteppingThreadGroup();
-      int expResult = 1;
-      int result = instance.getInUserCount();
-      assertEquals(expResult, result);
-   }
\ No newline at end of file
+    public void testGetInUserCount() {
+        System.out.println("getInUserCount");
+        SteppingThreadGroup instance = new SteppingThreadGroup();
+        String expResult = "";
+        String result = instance.getInUserCount();
+        assertEquals(expResult, result);
+    }
\ No newline at end of file
