diff --git a/old_product.java b/new_product.java
index 62b6006..10db03c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,13 +7,13 @@
             return false;
         }
         ChartRenderingInfo that = (ChartRenderingInfo) obj;
-        if (!ObjectUtilities.equal(this.chartArea, that.chartArea)) {
+        if (!ObjectUtils.equal(this.chartArea, that.chartArea)) {
             return false;
         }
-        if (!ObjectUtilities.equal(this.plotInfo, that.plotInfo)) {
+        if (!ObjectUtils.equal(this.plotInfo, that.plotInfo)) {
             return false;
         }
-        if (!ObjectUtilities.equal(this.entities, that.entities)) {
+        if (!ObjectUtils.equal(this.entities, that.entities)) {
             return false;
         }
         return true;
