diff --git a/old_test.java b/new_test.java
index e777a70..14cb4ac 100644
--- a/old_test.java
+++ b/new_test.java
@@ -60,11 +60,11 @@
                 3.0f, 4.0f, Color.green));
         assertTrue(plot1.equals(plot2));
 
-        // backgroundImage
-        plot1.setBackgroundImage(JFreeChart.INFO.getLogo());
-        assertFalse(plot1.equals(plot2));
-        plot2.setBackgroundImage(JFreeChart.INFO.getLogo());
-        assertTrue(plot1.equals(plot2));
+//        // backgroundImage
+//        plot1.setBackgroundImage(JFreeChart.INFO.getLogo());
+//        assertFalse(plot1.equals(plot2));
+//        plot2.setBackgroundImage(JFreeChart.INFO.getLogo());
+//        assertTrue(plot1.equals(plot2));
 
         // backgroundImageAlignment
         plot1.setBackgroundImageAlignment(Align.BOTTOM_RIGHT);
