diff --git a/old_test.java b/new_test.java
index 33c19c8..3e17f85 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
     InStreamOptions inStreamOptions = options.toInStreamOptions();
     Assert.assertEquals(options.getReadType().getAlluxioStorageType(),
         inStreamOptions.getAlluxioStorageType());
-    Assert.assertEquals(options.getLocationPolicy(), inStreamOptions.getLocationPolicy());
+    Assert.assertEquals(options.getCacheLocationPolicy(), inStreamOptions.getCacheLocationPolicy());
     Assert.assertEquals(options.getUfsReadLocationPolicy(),
         inStreamOptions.getUfsReadLocationPolicy());
     Assert.assertEquals(options.getMaxUfsReadConcurrency(),
