diff --git a/old_test.java b/new_test.java
index 24ef363..da499e2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
     public void test_firstInMonth_serialization() throws IOException, ClassNotFoundException {
-        DateTimeAdjuster firstInMonth = DateTimeAdjusters.firstInMonth(SUNDAY);
+        WithAdjuster firstInMonth = DateTimeAdjusters.firstInMonth(SUNDAY);
         assertTrue(firstInMonth instanceof Serializable);
 
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
