diff --git a/old_product.java b/new_product.java
index 4cab13d..c06d235 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,6 +12,7 @@ public void validate(Object obj, Errors errors) {
 			ValidationUtils.rejectIfEmpty(errors, "voided", "error.null");
 			ValidationUtils.rejectIfEmpty(errors, "concept", "Concept.noConceptSelected");
 			ValidationUtils.rejectIfEmpty(errors, "patient", "error.null");
+			ValidationUtils.rejectIfEmpty(errors, "orderer", "error.null");
 			
 			Date startDate = order.getStartDate();
 			if (startDate != null) {
