diff --git a/old_test.java b/new_test.java
index e667695..08cae63 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,16 +1,14 @@
 @Test(description = "PUT /vAppTemplate/{id}/leaseSettingsSection")
    public void testEditLeaseSettingsSection() throws Exception {
       // NOTE use smallish number for storageLeaseInSeconds; it seems to be capped at 5184000?
-      int storageLeaseInSeconds = random.nextInt(10000)+1;
+      int storageLeaseInSeconds = random.nextInt(10000) + 1;
 
-      LeaseSettingsSection leaseSettingSection = LeaseSettingsSection.builder()
-               .info("my info")
-               .storageLeaseInSeconds(storageLeaseInSeconds)
-               .build();
-      
-      final Task task = vAppTemplateApi.editLeaseSettingsSection(vAppTemplateURI, leaseSettingSection);
+      LeaseSettingsSection leaseSettingSection = LeaseSettingsSection.builder().info("my info")
+               .storageLeaseInSeconds(storageLeaseInSeconds).build();
+
+      final Task task = vAppTemplateApi.editLeaseSettingsSection(vAppTemplateUrn, leaseSettingSection);
       assertTaskSucceeds(task);
-      
-      LeaseSettingsSection newLeaseSettingsSection = vAppTemplateApi.getLeaseSettingsSection(vAppTemplateURI);
+
+      LeaseSettingsSection newLeaseSettingsSection = vAppTemplateApi.getLeaseSettingsSection(vAppTemplateUrn);
       assertEquals(newLeaseSettingsSection.getStorageLeaseInSeconds(), (Integer) storageLeaseInSeconds);
    }
\ No newline at end of file
