diff --git a/old_test.java b/new_test.java
index 2a648ba..953bc09 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testLeProp() {
-        assertTrue(instance.leProp("id", "name").getQueryCriterions()
-                .contains(new LePropCriterion("id", "name")));
+    	assertEquals(new LePropCriterion("id", "name"), instance.leProp("id", "name").getQueryCriterion());
     }
\ No newline at end of file
