diff --git a/old_test.java b/new_test.java
index e9776ed..2005e96 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,28 @@
 @Test
   public void testWillFire() {
-    fail("Not yet implemented");
+    try {
+      Calendar blockOutStartDate = new GregorianCalendar(2013, Calendar.JANUARY, 7);
+      SimpleBlockoutTrigger blockOutTrigger = new SimpleBlockoutTrigger("blockOut", blockOutStartDate.getTime(), null, //$NON-NLS-1$
+          -1, TIME.WEEK.time * 2, TIME.HOUR.time * 2);
+
+      Calendar scheduleStartDate = new GregorianCalendar(2013, Calendar.JANUARY, 7, 1, 0, 0);
+      SimpleTrigger trueScheduleTrigger = new SimpleTrigger("trueSchedule", "SCHEDULES", scheduleStartDate.getTime(),
+          null, -1, TIME.WEEK.time);
+      scheduleJob(trueScheduleTrigger);
+
+      SimpleTrigger falseScheduleTrigger = new SimpleTrigger("falseSchedule", "SCHEDULES", scheduleStartDate.getTime(),
+          null, -1, TIME.WEEK.time * 2);
+      scheduleJob(falseScheduleTrigger);
+
+      this.blockOutManager.addBlockout(blockOutTrigger);
+      assertTrue(this.blockOutManager.willFire(trueScheduleTrigger));
+      assertFalse(this.blockOutManager.willFire(falseScheduleTrigger));
+
+      // Clean up
+      deleteJob(falseScheduleTrigger);
+      deleteJob(trueScheduleTrigger);
+
+    } catch (SchedulerException e) {
+      throw new RuntimeException(e);
+    }
   }
\ No newline at end of file
