diff --git a/old_test.java b/new_test.java
index 4d1ecf3..6fab0b4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
     public void testRange() {
         System.out.println("range");
-        List<Neighbor<double[], double[]>> n1 = new ArrayList<Neighbor<double[], double[]>>();
-        List<Neighbor<double[], double[]>> n2 = new ArrayList<Neighbor<double[], double[]>>();
+        List<Neighbor<double[], double[]>> n1 = new ArrayList<>();
+        List<Neighbor<double[], double[]>> n2 = new ArrayList<>();
         for (int i = 0; i < data.length; i++) {
             kdtree.range(data[i], 0.5, n1);
             naive.range(data[i], 0.5, n2);
