diff --git a/old_product.java b/new_product.java
index 419d5e0..c5d068c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -33,7 +33,7 @@ public void range(double[] q, double radius, List<Neighbor<double[], E>> neighbo
                                 }
 
                                 if (!existed) {
-                                    neighbors.add(new Neighbor<double[], E>(e.key, e.data, e.index, distance));
+                                    neighbors.add(new Neighbor<>(e.key, e.data, e.index, distance));
                                 }
                             }
                         }
