diff --git a/old_product.java b/new_product.java
index 1dbecdc..7ce1de7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-public static <K, T> Map<K, T> getSubMapExcludeKeys(Map<K, T> map,K[] excludeKeys){
+@SafeVarargs
+    public static <K, T> Map<K, T> getSubMapExcludeKeys(Map<K, T> map,K...excludeKeys){
         if (Validator.isNullOrEmpty(map)){
             return Collections.emptyMap();
         }
@@ -6,13 +7,15 @@ public static <K, T> Map<K, T> getSubMapExcludeKeys(Map<K, T> map,K[] excludeKey
             return map;
         }
 
-        Map<K, T> returnMap = new HashMap<K, T>(map);
+        Map<K, T> returnMap = new LinkedHashMap<K, T>(map);
 
         for (K key : excludeKeys){
             if (map.containsKey(key)){
                 returnMap.remove(key);
             }else{
-                LOGGER.warn("map don't contains key:[{}]", key);
+                if (LOGGER.isWarnEnabled()){
+                    LOGGER.warn("map:{} don't contains key:[{}]", JsonUtil.format(map), key);
+                }
             }
         }
         return returnMap;
