diff --git a/old_test.java b/new_test.java
index ed13b59..9234c5c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,18 +8,18 @@
 
     error("geo:relate(<gml:Point><gml:coordinates>18,11</gml:coordinates></gml:Point>,"
         + "<gml:LineString><gml:coordinates>11,10 20,1 20,20</gml:coordinates>" +
-          "</gml:LineString>, \"0******\")", INVCAST_X_X_X.qname());
+          "</gml:LineString>, \"0******\")", GeoErrors.qname(6));
 
     error("geo:relate(" +
         "<gml:Point><gml:coordinates>18,11</gml:coordinates></gml:Point>," +
         "<gml:LineString><gml:coordinates>11,10 20,1 20,20</gml:coordinates>" +
-        "</gml:LineString>, \"0*******12*F\")", INVCAST_X_X_X.qname());
+        "</gml:LineString>, \"0*******12*F\")", GeoErrors.qname(6));
 
-    error("geo:relate()", FUNCARGNUM_X_X_X.qname());
+    error("geo:relate()", JAVAARITY_X_X_X_X_X);
     error("geo:relate(" +
         "<gml:Line><gml:coordinates>1,1 55,99 2,1</gml:coordinates></gml:Line>," +
         "<gml:LineString></gml:LineString>, \"0********\")", GeoErrors.qname(1));
     error("geo:relate(" +
         "<gml:Point><gml:coordinates>1,1</gml:coordinates></gml:Point>," +
-        " \"0********\")", FUNCARGNUM_X_X_X.qname());
+        " \"0********\")", JAVAARITY_X_X_X_X_X);
   }
\ No newline at end of file
