diff --git a/old_product.java b/new_product.java
index dcec748..e006eaf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,5 +3,9 @@
       throws QueryException {
     final Geometry geo1 = checkGeo(node1);
     final Geometry geo2 = checkGeo(node2);
-    return Bln.get(geo1.relate(geo2, intersectionMatrix.toJava()));
+    try {
+      return Bln.get(geo1.relate(geo2, intersectionMatrix.toJava()));
+    } catch(final IllegalArgumentException ex) {
+      throw GeoErrors.illegalArg(intersectionMatrix);
+    }
   }
\ No newline at end of file
