diff --git a/old_product.java b/new_product.java
index 717a412..b32b3ef 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static void vertical(Kernel1D_I32 kernel, GrayU8 image, GrayI8 dest , int skip ) {
+public static void vertical(Kernel1D_S32 kernel, GrayU8 image, GrayI8 dest , int skip ) {
 		checkParameters(image, dest, skip);
 
 		if( kernel.width >= image.width ) {
