diff --git a/old_product.java b/new_product.java
index a3f1678..1b3ea15 100644
--- a/old_product.java
+++ b/new_product.java
@@ -28,7 +28,7 @@ private ValueBuilder eval(final QueryContext ctx, final byte[] qu, final String
           }
         }, 500, 500);
       }
-      final long ms = opts.get(XQueryOptions.TIMEOUT) * 1000l;
+      final long ms = opts.get(XQueryOptions.TIMEOUT) * 1000L;
       if(ms != 0) {
         to.schedule(new TimerTask() {
           @Override
