diff --git a/old_test.java b/new_test.java
index d8b13f3..85d0f83 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,9 +17,7 @@
 		}
 		
 		OrderService os = Context.getOrderService();
-		List<Patient> patients = new ArrayList<Patient>();
-		patients.add(patient);
-		List<Order> orders = os.getOrders(null, patients, null, null, null);
+		List<Order> orders = os.getAllOrdersByPatient(patient);
 		Assert.assertFalse(orders.isEmpty());
 		//all order void related fields should be null
 		for (Order order : orders) {
