diff --git a/old_product.java b/new_product.java
index 27337b6..0c3543d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,8 @@
-public List<ApplicationEventFamilyMapDto> getApplicationEventFamilyMapsByApplicationToken(String applicationToken) throws Exception {
-        ParameterizedTypeReference<List<ApplicationEventFamilyMapDto>> typeRef = new ParameterizedTypeReference<List<ApplicationEventFamilyMapDto>>() {
-        };
-        ResponseEntity<List<ApplicationEventFamilyMapDto>> entity = restTemplate.exchange(restTemplate.getUrl() + "applicationEventMapsByAppToken/" +
-                        applicationToken, HttpMethod.GET, null, typeRef);
-        return entity.getBody();
-    }
\ No newline at end of file
+public List<ApplicationEventFamilyMapDto> getApplicationEventFamilyMapsByApplicationToken(
+      String applicationToken) throws Exception {
+    ResponseEntity<List<ApplicationEventFamilyMapDto>> entity = restTemplate.exchange(
+        restTemplate.getUrl() + "applicationEventMaps/" + applicationToken,
+        HttpMethod.GET, null,
+        new ParameterizedTypeReference<List<ApplicationEventFamilyMapDto>>() {});
+    return entity.getBody();
+  }
\ No newline at end of file
