diff --git a/old_test.java b/new_test.java
index 7812c12..03059cf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,8 @@
         File mockMetaFile = new MockFileBuilder().setFileName("worker.yaml").build();
 
         String expId = "id12345";
-        Map<String, File> expected = Collections.singletonMap(expId, port1Dir);
+        SortedSet<File> expected = new TreeSet<>();
+        expected.add(port1Dir);
 
         try {
             SupervisorUtils mockedSupervisorUtils = mock(SupervisorUtils.class);
@@ -24,7 +25,7 @@
             };
 
             when(mockedSupervisorUtils.readWorkerHeartbeatsImpl(anyMapOf(String.class, Object.class))).thenReturn(null);
-            assertEquals(expected, workerLogs.identifyWorkerLogDirs(Collections.singleton(port1Dir)));
+            assertEquals(expected, workerLogs.getLogDirs(Collections.singleton(port1Dir), (wid) -> true));
         } finally {
             SupervisorUtils.resetInstance();
         }
