diff --git a/old_product.java b/new_product.java
index f20c089..d28f6ac 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,7 @@ public List<Operator<?>> getInputOperators() {
 
 			@Override
 			public int indexOf(final Object o) {
-				final ListIterator<Output> e = Operator.this.inputs.listIterator();
+				final ListIterator<Operator<?>.Output> e = Operator.this.inputs.listIterator();
 				while (e.hasNext())
 					if (o == e.next())
 						return e.previousIndex();
