diff --git a/old_product.java b/new_product.java
index 70703d8..4664686 100644
--- a/old_product.java
+++ b/new_product.java
@@ -23,13 +23,13 @@ public CompletableFuture<Long> getOrAssignStreamSegmentId(String streamSegmentNa
         }
         // We are the first/only ones requesting this id; go ahead and assign an id.
         if (needsAssignment) {
-            // Determine if given StreamSegmentName is actually a batch StreamSegmentName.
+            // Determine if given StreamSegmentName is actually a Transaction StreamSegmentName.
             String parentStreamSegmentName = StreamSegmentNameUtils.getParentStreamSegmentName(streamSegmentName);
             if (parentStreamSegmentName == null) {
                 // Stand-alone StreamSegment.
                 this.executor.execute(() -> assignStreamSegmentId(streamSegmentName, timeout));
             } else {
-                this.executor.execute(() -> assignBatchStreamSegmentId(streamSegmentName, parentStreamSegmentName, timeout));
+                this.executor.execute(() -> assignTransactionStreamSegmentId(streamSegmentName, parentStreamSegmentName, timeout));
             }
         }
 
