diff --git a/old_product.java b/new_product.java
index 573cdb6..cdac3f9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -18,6 +18,10 @@
                                   this.displayAuthenticationRealm, orig.displayAuthenticationRealm);
         appendToBufferIfDifferent(buf, "httpOnlyCookies",
                                   this.httpOnlyCookies, orig.httpOnlyCookies);
+        appendToBufferIfDifferent(buf, "jaspicSessionCookieName",
+                                  this.jaspicSessionCookieName, orig.jaspicSessionCookieName);
+        appendToBufferIfDifferent(buf, "jaspicSessionForMechanismsEnabled",
+                                  this.jaspicSessionForMechanismsEnabled, orig.jaspicSessionForMechanismsEnabled);
         appendToBufferIfDifferent(buf, "logoutOnHttpSessionExpire",
                                   this.logoutOnHttpSessionExpire, orig.logoutOnHttpSessionExpire);
         appendToBufferIfDifferent(buf, "logoutPageRedirectDomainNames",
