diff --git a/old_product.java b/new_product.java
index 05dfc09..cac938a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public PackedAlignments packAlignments(
             AlignmentInterval interval,
             AlignmentTrack.RenderOptions renderOptions) {
 
-        if (renderOptions == null) renderOptions = new AlignmentTrack.RenderOptions();
+       // if (renderOptions == null) renderOptions = new AlignmentTrack.RenderOptions();
 
         LinkedHashMap<String, List<Row>> packedAlignments = new LinkedHashMap<String, List<Row>>();
 
