diff --git a/old_product.java b/new_product.java
index b635c00..639006a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,8 @@
 private Iter event(final QueryContext ctx) throws QueryException {
     String name = expr[0].toString();
-    String q2 = expr[2].toString();
     String msg = "";
     Value v = expr[1].value(ctx);
-    if(q2.replaceAll("\"", "").trim().isEmpty()) {
+    if(expr.length < 3) {
       ctx.updating = true;
       msg = ctx.value(expr[1]).toString();
     } else {
