diff --git a/old_test.java b/new_test.java
index 21e3545..c14a48a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
-	public void testGetCorpusRelationScores() {
-		List<Concept> concepts = new ArrayList<Concept>();
-		concepts.add(EXAMPLE_CONCEPT);
-		Scores scores = service.getCorpusRelationScores(Corpus.IBM_RESEARCHERS, concepts);
-		Assert.assertNotNull(scores);
-	}
\ No newline at end of file
+  public void testGetCorpusRelationScores() {
+    final List<Concept> concepts = new ArrayList<Concept>();
+    concepts.add(EXAMPLE_CONCEPT);
+    final Scores scores = service.getCorpusRelationScores(Corpus.IBM_RESEARCHERS, concepts);
+    Assert.assertNotNull(scores);
+  }
\ No newline at end of file
