diff --git a/old_product.java b/new_product.java
index 7981dff..ed641c1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,7 @@
 @RequestMapping(value = "CTL/deleteSchema", params = { "fqn", "version" }, method = RequestMethod.POST)
     @ResponseStatus(value = HttpStatus.OK)
-    public void deleteCTLSchemaByFqnAndVersion(@RequestParam String fqn, @RequestParam int version) throws KaaAdminServiceException {
-        kaaAdminService.deleteCTLSchemaByFqnAndVersion(fqn, version);
+    public void deleteCTLSchemaByFqnVersionTenantIdAndApplicationId(@RequestParam String fqn, @RequestParam int version, 
+            @RequestParam(required = false) String tenantId,
+            @RequestParam(required = false) String applicationId) throws KaaAdminServiceException {
+        kaaAdminService.deleteCTLSchemaByFqnVersionTenantIdAndApplicationId(fqn, version, tenantId, applicationId);
     }
\ No newline at end of file
