diff --git a/old_product.java b/new_product.java
index 5f8a31c..0f99ca4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,11 @@ private void run(final String[] args) throws Exception {
     final Performance perf = new Performance();
     ctx.options.set(MainOptions.CHOP, false);
     ctx.options.set(MainOptions.INTPARSE, false);
-    ctx.options.set(MainOptions.SERIALIZER, "omit-xml-declaration=no,indent=no");
+
+    final SerializerOptions sopts = new SerializerOptions();
+    sopts.set(SerializerOptions.INDENT, YesNo.NO);
+    sopts.set(SerializerOptions.OMIT_XML_DECLARATION, YesNo.NO);
+    ctx.options.set(MainOptions.SERIALIZER, sopts);
 
     final XdmValue doc = new XQuery("doc('" + file(false, CATALOG) + "')", ctx).value();
     final String version = asString("*:catalog/@version", doc);
