diff --git a/old_product.java b/new_product.java
index a297252..81e84bf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-public List<S> process() throws Exception {
+@SuppressWarnings("unchecked")
+    public List<S> process() throws Exception {
         ArrayList<S> sequences = new ArrayList<S>();
 
 
@@ -14,7 +15,7 @@ public List<S> process() throws Exception {
             if (line.length() != 0) {
                 if (line.startsWith(">")) {
                     if (sb.length() > 0) {
-                        S sequence = (S) sequenceCreator.getSequence(sb.toString(), sequenceIndex);
+                        S sequence = (S)sequenceCreator.getSequence(sb.toString(), sequenceIndex);
                         headerParser.parseHeader(header, sequence);
                         sequences.add(sequence);
                         if (maxSequenceLength < sb.length()) {
@@ -35,7 +36,7 @@ public List<S> process() throws Exception {
             fileIndex = br.getBytesRead();
             line = br.readLine();
             if (line == null) {
-                S sequence = (S) sequenceCreator.getSequence(sb.toString(), fileIndex);
+                S sequence = (S)sequenceCreator.getSequence(sb.toString(), fileIndex);
                 headerParser.parseHeader(header, sequence);
                 sequences.add(sequence);
                 keepGoing = false;
