diff --git a/old_product.java b/new_product.java
index fe76424..678a224 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static Rectangle2D_I32 boundBox( int srcWidth , int srcHeight ,
+public static RectangleLength2D_I32 boundBox( int srcWidth , int srcHeight ,
 											PixelTransform_F32 transform )
 	{
 		int x0,y0,x1,y1;
@@ -25,5 +25,5 @@ public static Rectangle2D_I32 boundBox( int srcWidth , int srcHeight ,
 				y1 = (int)transform.distY;
 		}
 
-		return new Rectangle2D_I32(x0,y0,x1-x0,y1-y0);
+		return new RectangleLength2D_I32(x0,y0,x1-x0,y1-y0);
 	}
\ No newline at end of file
