diff --git a/old_test.java b/new_test.java
index 8be7d05..6a7125e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
 		// basic sanity check
 		Affine2D_F32 affine = new Affine2D_F32(1,0,0,1,2,3);
 		PixelTransformAffine_F32 transform = new PixelTransformAffine_F32(affine);
-		Rectangle2D_I32 found = DistortImageOps.boundBox(10,20,30,40,transform);
+		RectangleLength2D_I32 found = DistortImageOps.boundBox(10,20,30,40,transform);
 		
 		assertEquals(2,found.x0);
 		assertEquals(3,found.y0);
