diff --git a/old_test.java b/new_test.java
index f2b1fe2..84df3d2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test(timeout = 10000)
   public void testSetCapacity() throws InterruptedException {
     TaskExecutorServiceForTest taskExecutorService =
-        new TaskExecutorServiceForTest(2, 3, ShortestJobFirstComparator.class.getName(), true);
+        new TaskExecutorServiceForTest(2, 3, ShortestJobFirstComparator.class.getName(), true, mockMetrics);
 
     // Fourth is lower priority as a result of canFinish being set to false.
     MockRequest r1 = createMockRequest(1, 1, 1, 100, 200, true, 20000L, true);
