diff --git a/old_product.java b/new_product.java
index 0127f73..1ba8f7d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static <T> CompletableFuture<Collection<T>> allOfWithResults(Collection<CompletableFuture<T>> futures) {
+public static <T> CompletableFuture<List<T>> allOfWithResults(List<CompletableFuture<T>> futures) {
         CompletableFuture<Void> allDoneFuture = CompletableFuture.allOf(futures.toArray(new CompletableFuture[futures.size()]));
         return allDoneFuture.thenApply(v -> futures.stream().map(CompletableFuture::join).collect(Collectors.toList()));
     }
\ No newline at end of file
