diff --git a/old_product.java b/new_product.java
index 3f46802..2cfd73f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public static void remove(BValue json, String fieldName) {
-        if (json == null || json.getType().getTag() != TypeTags.MAP_TAG) {
+        if (json == null || json.getType().getTag() != TypeTags.JSON_TAG) {
             return;
         }
 
-        ((BMap) json).remove(fieldName);
+        ((BMap<String, ?>) json).remove(fieldName);
     }
\ No newline at end of file
