diff --git a/old_product.java b/new_product.java
index 627f27b..12536c1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-public void setBlockingState(BlockingState state, InternalCallContext context);
\ No newline at end of file
+public void setBlockingStateAndPostBlockingTransitionEvent(BlockingState state, UUID bundleId, InternalCallContext context);
\ No newline at end of file
