diff --git a/old_test.java b/new_test.java
index bc31cc2..80aa6c9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,7 +14,7 @@
     String [] expectedResTopDir2 = new String[] {"/testDirNonEmpty2", "/testDirNonEmptyF"};
     Arrays.sort(expectedResTopDir);
     Arrays.sort(expectedResTopDir2);
-    String [] resTopDir = mUfs.list(testDirNonEmpty);
+    UnderFileStatus [] resTopDir = mUfs.list(testDirNonEmpty);
     Arrays.sort(resTopDir);
     Assert.assertTrue(Arrays.equals(expectedResTopDir, resTopDir)
         || Arrays.equals(expectedResTopDir2, resTopDir));
