diff --git a/old_product.java b/new_product.java
index 5244d9e..52d05e1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 private void maybeSignalForMoreConnections(ConnectionPartition connectionPartition) {
 
 		if (!this.poolShuttingDown && !connectionPartition.isUnableToCreateMoreTransactions() && connectionPartition.getFreeConnections().size()*100/connectionPartition.getMaxConnections() < this.poolAvailabilityThreshold){
-			this.poolWatchThreadSignalQueue.offer(new Object()); // item being pushed is not important.
+			connectionPartition.getPoolWatchThreadSignalQueue().offer(new Object()); // item being pushed is not important.
 		}
 	}
\ No newline at end of file
