diff --git a/old_test.java b/new_test.java
index 55e7be5..b5d5e23 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,8 @@
 		expect(mockPartition.getFreeConnections()).andReturn(mockConnectionHandles).anyTimes();
 		expect(mockConnectionHandles.size()).andReturn(1).anyTimes();
 		expect(mockPartition.getMaxConnections()).andReturn(10).anyTimes();
+		BlockingQueue<Object> bq = new ArrayBlockingQueue<Object>(1);
+		expect(mockPartition.getPoolWatchThreadSignalQueue()).andReturn(bq).anyTimes();
 //		mockPartition.lockAlmostFullLock();
 //		expectLastCall().once();
 //		mockPartition.almostFullSignal();
@@ -18,6 +20,7 @@
 		
 		// Test 2, same test but fake an exception
 		reset(mockPartition, mockConnectionHandles);
+		expect(mockPartition.getPoolWatchThreadSignalQueue()).andReturn(bq).anyTimes();
 		expect(mockPartition.isUnableToCreateMoreTransactions()).andReturn(false).anyTimes();
 		expect(mockPartition.getFreeConnections()).andReturn(mockConnectionHandles).anyTimes();
 		expect(mockConnectionHandles.size()).andReturn(1).anyTimes();
