diff --git a/old_test.java b/new_test.java
index 8699a07..cd7c1f5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,10 +4,10 @@
         PolicyDefinitionBean def = new PolicyDefinitionBean();
         def.setId("missingvar"); //$NON-NLS-1$
         PolicyDefinitionTemplateBean template = new PolicyDefinitionTemplateBean();
-        template.setTemplate("Templates includes a missing var: @{missingVar}"); //$NON-NLS-1$
+        template.setTemplate("Template includes a missing var: @{missingVar}"); //$NON-NLS-1$
         def.getTemplates().add(template);
         policy.setDefinition(def);
         policy.setConfiguration("{}"); //$NON-NLS-1$
         PolicyTemplateUtil.generatePolicyDescription(policy);
-        Assert.assertEquals("Templates includes a missing var: @{missingVar}", policy.getDescription()); //$NON-NLS-1$
+        Assert.assertEquals("Template includes a missing var: null", policy.getDescription()); //$NON-NLS-1$
     }
\ No newline at end of file
