diff --git a/old_product.java b/new_product.java
index 6b6aa20..dbbf027 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,6 +14,7 @@ public static void generatePolicyDescription(PolicyBean policy) throws Exception
             String jsonConfig = policy.getConfiguration();
             @SuppressWarnings("unchecked")
             Map<String, Object> configMap = mapper.readValue(jsonConfig, Map.class);
+            configMap = new PolicyConfigMap(configMap);
             String desc = (String) TemplateRuntime.execute(template, configMap);
             policy.setDescription(desc);
         } catch (Exception e) {
