diff --git a/old_product.java b/new_product.java
index 57a5f31..41c8218 100644
--- a/old_product.java
+++ b/new_product.java
@@ -21,12 +21,12 @@
     Matcher foundMatch = isJustNumber.matcher(dateAsString);
     if (foundMatch.find()) {
       Long timeAsLong = Long.parseLong(dateAsString);
-      Long msCheck = 1000000000000L;
+      Long msCheck = 100000000000L;
 
       // are we ms or seconds maybe?
-      if (timeAsLong > msCheck) {
-        // assuming milliseconds
-        timeAsLong = timeAsLong / 1000;
+      if (timeAsLong < msCheck) {
+        // assuming in seconds
+        timeAsLong = timeAsLong * 1000;
       }
       return new Date(timeAsLong);
     }
