diff --git a/old_test.java b/new_test.java
index 0e67150..9845398 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,8 +6,8 @@
             "2016-06-20T04:25:16.218Z",
             "2015-05-28T18:01:57Z",
             "2016-06-20T04:25:16.218+0000",
-            "1477796981",
-            "1477796981000"
+            "1478097789",
+            "1478097789000"
         };
 
         String jsonStr = "[\"" + StringUtils.join(dateStrings, "\",\"") + "\"]";
