diff --git a/old_test.java b/new_test.java
index 27c0dec..e8278fe 100644
--- a/old_test.java
+++ b/new_test.java
@@ -23,7 +23,7 @@
                 for (int i = 0; i < k; i++) {
                     priori[i] = 1.0 / k;
                     for (int j = 0; j < p; j++) {
-                        ArrayList<Double> axi = new ArrayList<Double>();
+                        ArrayList<Double> axi = new ArrayList<>();
                         for (int m = 0; m < trainx.length; m++) {
                             if (trainy[m] == i) {
                                 axi.add(trainx[m][j]);
