diff --git a/old_product.java b/new_product.java
index c1bec1b..466ed68 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @Override
     public int predict(T x) {
-        List<Neighbor<T,T>> neighbors = new ArrayList<Neighbor<T,T>>();
+        List<Neighbor<T,T>> neighbors = new ArrayList<>();
         nns.range(x, radius, neighbors);
         
         if (neighbors.size() < minPts) {
