diff --git a/old_product.java b/new_product.java
index c85da24..535e795 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 private Item write(final QueryContext qc) throws QueryException {
-    final java.nio.file.Path path = checkPath(0, qc);
-    final B64 archive = (B64) checkType(checkItem(exprs[1], qc), AtomType.B64);
+    final java.nio.file.Path path = toPath(0, qc);
+    final B64 archive = toB64(exprs[1], qc, false);
     final TokenSet hs = entries(2, qc);
 
     final ArchiveIn in = ArchiveIn.get(archive.input(info), info);
@@ -19,7 +19,7 @@ private Item write(final QueryContext qc) throws QueryException {
         }
       }
     } catch(final IOException ex) {
-      throw ARCH_FAIL.get(info, ex);
+      throw ARCH_FAIL_X.get(info, ex);
     } finally {
       in.close();
     }
