diff --git a/old_product.java b/new_product.java
index e5c067e..8626317 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
 public List<T> topologicalSort() throws CycleException {
 		
 		Set<T> queue = getNodesWithNoIncomingEdges();
-		List<T> result = new ArrayList<T>();
+		List<T> result = new ArrayList<>();
 		
 		// The initial edges are stored.
-		List<Edge> initialEdges = new ArrayList<Edge>();
+		List<Edge> initialEdges = new ArrayList<>();
 		initialEdges.addAll(edges);
 		while (!queue.isEmpty()) {
 			T node = queue.iterator().next();
