diff --git a/old_test.java b/new_test.java
index 4ce8930..1fffa9d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,25 +1,30 @@
 @Test
     public void testPutBucketAccessStyleMode() {
-    	Client client = new Client();
-    	Response response;
-    	try {
-    		response = bucketManager.putBucketAccessStyleMode(TestConfig.testBucket_z0, AccessStyleMode.OPEN);
-    		Assert.assertEquals(200, response.statusCode);
-    		
-    		try {
-        		client.get(TestConfig.testUrl_z0);
-    		} catch (QiniuException e) {
-    			e.printStackTrace();
-    			Assert.assertEquals(401, e.response.statusCode);
+    	String[] buckets = new String[]{TestConfig.testBucket_z0, TestConfig.testBucket_na0};
+    	String[] urls = new String[]{TestConfig.testUrl_z0, TestConfig.testUrl_na0};
+    	for (int i = 0; i < buckets.length; i ++) {
+    		String bucket = buckets[i];
+    		String url = urls[i];
+        	Client client = new Client();
+        	Response response;
+        	try {
+        		// 测试开启原图保护
+        		response = bucketManager.putBucketAccessStyleMode(bucket, AccessStyleMode.OPEN);
+        		Assert.assertEquals(200, response.statusCode);
+        		try {
+            		client.get(url);
+        		} catch (QiniuException e) {
+        			Assert.assertEquals(401, e.response.statusCode);
+        		}
+        		
+        		// 测试关闭原图保护
+        		response = bucketManager.putBucketAccessStyleMode(bucket, AccessStyleMode.CLOSE);
+        		Assert.assertEquals(200, response.statusCode);
+        		response = client.get(url);
+        		Assert.assertEquals(200, response.statusCode);
+        		
+        	} catch (QiniuException e) {
+        		Assert.fail(e.response.toString());
     		}
-    		
-    		response = bucketManager.putBucketAccessStyleMode(TestConfig.testBucket_z0, AccessStyleMode.CLOSE);
-    		Assert.assertEquals(200, response.statusCode);
-    		
-    		response = client.get(TestConfig.testUrl_z0);
-    		Assert.assertEquals(200, response.statusCode);
-    		
-    	} catch (QiniuException e) {
-    		Assert.fail(e.response.toString());
-		}
+    	}
     }
\ No newline at end of file
