diff --git a/old_test.java b/new_test.java
index 68db0a7..eb5d5a9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,8 @@
 @Test
 	@Verifies(value = "return a drug if phrase match drug_name No need to match both concept_name and drug_name", method = "getDrugs(String,Concept,boolean,boolean,boolean,Integer,Integer)")
 	public void getDrugs_shouldReturnDrugIfPhraseMatchDrugNameNoNeedToMatchBothConceptNameAndDrugName() throws Exception {
-		Session session3 = sessionFactory.getCurrentSession();
-		session3.beginTransaction();
 		// This concept does not contain concept_name with "Triomune"
-		Concept concept2 = (Concept) session3.get(Concept.class, 3);
+		Concept concept2 = dao.getConcept(3);
 		
 		// In this test there is no any concept_name match with "Triomune" but
 		// Drug name match with "Trimonue" so no need to match both drug_name
