diff --git a/old_test.java b/new_test.java
index 860a852..a463e43 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-    public void testRowMean() {
-        System.out.println("rowMean");
+    public void testRowMeans() {
+        System.out.println("rowMeans");
         double[][] A = {
             {0.7220180, 0.07121225, 0.6881997},
             {-0.2648886, -0.89044952, 0.3700456},
@@ -8,7 +8,7 @@
         };
         double[] r = {0.4938100, -0.2617642, 0.1447914};
 
-        double[] result = Math.rowMean(A);
+        double[] result = Math.rowMeans(A);
         for (int i = 0; i < r.length; i++) {
             assertEquals(result[i], r[i], 1E-7);
         }
