diff --git a/old_product.java b/new_product.java
index d256234..a7b5ec4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static void delete(Context ctx, int type, int id, String handle, String[] identifiers)
+public static void delete(Context ctx, int type, UUID id, String handle, List<String> identifiers)
             throws SQLException, RDFMissingIdentifierException
     {
         String uri = RDFConfiguration.getURIGenerator().generateIdentifier(ctx,
