diff --git a/old_test.java b/new_test.java
index ed60aee..c9d1111 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testValidate() {
-        TestConfiguration.getLogger().debug("validate"); 
+        TestUtils.log(this.getClass(), "validate"); 
         
         Dataset trainingData = KmeansTest.generateDataset();
         Dataset validationData = new Dataset();
