diff --git a/old_test.java b/new_test.java
index 979902c..90b50f4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,7 @@
 	@Verifies(value = "should not fail validation if drug is null", method = "validate(Object,Errors)")
 	public void validate_shouldNotFailValidationIfDrugIsNull() throws Exception {
 		DrugOrder order = new DrugOrder();
+		order.setDrug(null);
 		
 		Errors errors = new BindException(order, "order");
 		new DrugOrderValidator().validate(order, errors);
