diff --git a/old_product.java b/new_product.java
index 88a3165..3d4fcd1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,5 +9,16 @@ public void validate(Object obj, Errors errors) {
 			ValidationUtils.rejectIfEmpty(errors, "prn", "error.null");
 			ValidationUtils.rejectIfEmpty(errors, "complex", "error.null");
 			//ValidationUtils.rejectIfEmpty(errors, "drug", "error.null");
+			
+			if (order.getDrug() != null)
+				ValidationUtils.rejectIfEmpty(errors, "drug.concept", "error.null");
+			
+			if (!(order.getConcept() == null)) {
+				if (!(order.getDrug() == null) && !(order.getDrug().getConcept().equals(order.getConcept()))) {
+					errors.rejectValue("drug", "error.general");
+					errors.rejectValue("concept", "error.concept");
+					
+				}
+			}
 		}
 	}
\ No newline at end of file
