diff --git a/old_test.java b/new_test.java
index 189ae0d..20b1317 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,6 @@
             super.cloneSheet();
             fail("expected exception");
         } catch (RuntimeException e){
-            assertEquals("NotImplemented", e.getMessage());
+            assertEquals("Not Implemented", e.getMessage());
         }
     }
\ No newline at end of file
