diff --git a/old_product.java b/new_product.java
index 0edc8f5..6d9bc1f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,5 @@
 @Override
     @NotImplemented
-    public Sheet cloneSheet(int sheetNum)
-    {
-        throw new RuntimeException("NotImplemented");
+    public Sheet cloneSheet(int sheetNum) {
+        throw new RuntimeException("Not Implemented");
     }
\ No newline at end of file
