diff --git a/old_product.java b/new_product.java
index b8742d1..401c69f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@ final Item parse(final byte[] json, final boolean xml, final QueryContext qc,
     final JsonParserOptions opts = new JsonParserOptions();
     if(exprs.length > 1) {
       final Map options = toMap(exprs[1], qc);
-      new FuncOptions(null, info).acceptUnknown().parse(options, opts);
+      new FuncOptions(null, info).acceptUnknown().assign(options, opts);
     }
 
     final boolean esc = opts.get(JsonParserOptions.ESCAPE);
