diff --git a/old_product.java b/new_product.java
index bc23c41..e4f73c7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -42,6 +42,10 @@ public boolean isValidCCN(String rawCCN) {
                     return false;
                 }
                 break;
+            default:
+                if (false == isValidOtherDigitGrouping(splitCCN)) {
+                    return false;
+                }
         }
 
         return CREDIT_CARD_NUM_LUHN_CHECK.isValid(cannonicalCCN);
