diff --git a/old_test.java b/new_test.java
index 3f82404..d85a5ad 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,8 +9,8 @@
 		DenseMatrix64F rectK_inv = new DenseMatrix64F(3,3);
 		CommonOps.invert(rectK,rectK_inv);
 
-		PointTransform_F32 tranRect = RectifyImageOps.transformPixelToRect_F32(param, rect);
-		PointTransform_F64 alg = RectifyImageOps.transformPixelToRectNorm_F64(param, rect, rectK);
+		Point2Transform2_F32 tranRect = RectifyImageOps.transformPixelToRect_F32(param, rect);
+		Point2Transform2_F64 alg = RectifyImageOps.transformPixelToRectNorm_F64(param, rect, rectK);
 
 		double x=10,y=20;
 
