diff --git a/old_test.java b/new_test.java
index 7431532..7c69273 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-    public void testRecieveRawBytes() throws Exception {
+    public void testReceiveRawBytes() throws Exception {
         transportHandler.setFetchableByte(new byte[] { 1, 2, 3 });
-        byte[] received = socket.recieveRawBytes();
+        byte[] received = socket.receiveRawBytes();
         assertArrayEquals(new byte[] { 1, 2, 3 }, received);
     }
\ No newline at end of file
