diff --git a/old_product.java b/new_product.java
index a3e09a8..d0fcc98 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,11 @@
-public Set<Obs> getAllFlattenObs(boolean includeVoided) {
+public Set<Obs> getAllFlattenedObs(boolean includeVoided) {
 
 		Set<Obs> ret = new LinkedHashSet<>();
 
 		if (this.obs != null) {
 			for (Obs o : this.obs) {
-				if (includeVoided || (!includeVoided && !o.getVoided())) {
-					ret.add(o);
-					ret.addAll(getFlattenObsLeaves(o, includeVoided));
+				if (includeVoided || (!o.getVoided())) {
+					ret.addAll(getFlattenedObsLeaves(o, includeVoided));
 				}
 			}
 		}
