diff --git a/old_test.java b/new_test.java
index c1d6841..d289bb9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-	public void getAllFlattenObs_shouldGetAllFlattenObs() {
+	public void getAllFlattenObs_shouldGetAllFlattenedObs() {
 		Encounter enc = new Encounter();
 
 		//create and add an Obs
@@ -15,10 +15,32 @@
 		// add second top level obs
 		Obs thirdObs = new Obs();
 		thirdObs.setValueText("thirdObs");
+		thirdObs.setVoided(true);
 		enc.addObs(thirdObs);
 
+		// add three-level obs
+		Obs fourthObs = new Obs();
+		fourthObs.setValueText("fourthObs");
+
+		Obs fifthObs = new Obs();
+		fifthObs.setValueText("fifthObs");
+		Obs sixthObs = new Obs();
+		sixthObs.setValueText("sixthObs");
+		sixthObs.setVoided(true);
+		Obs seventhObs = new Obs();
+		seventhObs.setValueText("seventhObs");
+		fifthObs.addGroupMember(sixthObs);
+		fifthObs.addGroupMember(seventhObs);
+		fourthObs.addGroupMember(fifthObs);
+
+		enc.addObs(fourthObs);
+
 		// do the check
-		assertEquals(3, enc.getAllFlattenObs(true).size());
-		assertEquals(3, enc.getAllFlattenObs(false).size());
+		assertEquals(7, enc.getAllFlattenedObs(true).size());
+		assertTrue(enc.getAllFlattenedObs(true).contains(thirdObs));
+		assertTrue(enc.getAllFlattenedObs(true).contains(seventhObs));
 
+		assertEquals(5, enc.getAllFlattenedObs(false).size());
+		assertFalse(enc.getAllFlattenedObs(false).contains(sixthObs));
+		assertTrue(enc.getAllFlattenedObs(false).contains(secondObs));
 	}
\ No newline at end of file
