diff --git a/old_product.java b/new_product.java
index a3584e0..1730261 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,8 @@
-public ServiceCall<UtterancesTone> getChatTone(String jsonText) {
-    Validator.notNull(jsonText, "text cannot be null");
+public ServiceCall<UtterancesTone> getChatTone(ToneChatInput chatInput) {
+    Validator.notNull(chatInput, "text cannot be null");
 
     RequestBuilder requestBuilder = RequestBuilder.post(PATH_CHAT_TONE).query(VERSION_DATE, versionDate);
-    JsonObject contentJson = new JsonParser().parse(jsonText).getAsJsonObject();
-    requestBuilder.bodyJson(contentJson);
+    requestBuilder.bodyJson(GsonSingleton.getGson().toJsonTree(chatInput).getAsJsonObject());
 
     return createServiceCall(requestBuilder.build(), ResponseConverterUtils.getObject(UtterancesTone.class));
   }
\ No newline at end of file
