diff --git a/old_test.java b/new_test.java
index 804e9b9..bac9142 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,5 @@
 @Test
+	@Verifies(value = "should fetch number of provider matching given query", method = "getCountOfProviders(String,null)")
 	public void getCountOfProviders_shouldFetchNumberOfProviderMatchingGivenQuery() throws Exception {
 		assertEquals(1, service.getCountOfProviders("Hippo").intValue());
 		Person person = Context.getPersonService().getPerson(502);
