diff --git a/old_test.java b/new_test.java
index db04c31..d2d2bde 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test(groups={"tck"})
-    public void test_getFirstMonthOfQuarter() {
-        assertEquals(QuarterOfYear.Q1.getFirstMonthOfQuarter(), Month.JANUARY);
-        assertEquals(QuarterOfYear.Q2.getFirstMonthOfQuarter(), Month.APRIL);
-        assertEquals(QuarterOfYear.Q3.getFirstMonthOfQuarter(), Month.JULY);
-        assertEquals(QuarterOfYear.Q4.getFirstMonthOfQuarter(), Month.OCTOBER);
+    public void test_firstMonth() {
+        assertEquals(QuarterOfYear.Q1.firstMonth(), Month.JANUARY);
+        assertEquals(QuarterOfYear.Q2.firstMonth(), Month.APRIL);
+        assertEquals(QuarterOfYear.Q3.firstMonth(), Month.JULY);
+        assertEquals(QuarterOfYear.Q4.firstMonth(), Month.OCTOBER);
     }
\ No newline at end of file
