diff --git a/old_product.java b/new_product.java
index 4f8845a..1befcdb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ public Authentication authenticate(Authentication authentication) throws Authent
 
         SAMLAuthenticationToken token = (SAMLAuthenticationToken) authentication;
         SAMLMessageStorage store = token.getMessageStore();
-        BasicSAMLMessageContext context = token.getCredentials();
+        SAMLMessageContext context = token.getCredentials();
         SAMLCredential credential;
 
         try {
