diff --git a/old_test.java b/new_test.java
index f085ef9..db75431 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testAuthenticate() throws Exception {
-        BasicSAMLMessageContext context = new BasicSAMLMessageContext();
+        SAMLMessageContext context = new SAMLMessageContext();
 
         SAMLAuthenticationToken token = new SAMLAuthenticationToken(context, messageStorage);
         SAMLMessageStorage store = token.getMessageStore();
