diff --git a/old_product.java b/new_product.java
index 92ba1b3..092b4f1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,9 @@
             @Valid
             final Command command
     ) throws GenieException {
-        LOG.debug("Called to create command " + command.toString());
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Called to create command " + command.toString());
+        }
         if (StringUtils.isEmpty(command.getId())) {
             command.setId(UUID.randomUUID().toString());
         }
