diff --git a/old_product.java b/new_product.java
index f7d98e9..1b992db 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@
             List<LabeledVector> neighbors = findKNearestNeighbors(v);
 
             return classify(neighbors, v, stgy);
-        } else {
-            throw new IllegalStateException("The train kNN dataset is null");
         }
+        else
+            throw new IllegalStateException("The train kNN dataset is null");
     }
\ No newline at end of file
