diff --git a/old_test.java b/new_test.java
index 274b34e..0a44e80 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,7 +12,7 @@
             (k, v) -> v
         );
 
-        double[][] expectedData = new double[][] {
+        double[][] expData = new double[][] {
             {.5, 4. / 22, 1. / 300},
             {.25, 8. / 22, 22. / 300},
             {-1., 10. / 22, 100. / 300},
@@ -20,5 +20,5 @@
         };
 
         for (int i = 0; i < data.length; i++)
-            assertArrayEquals(expectedData[i], preprocessor.apply(i, VectorUtils.of(data[i])).asArray(), 1e-8);
+            assertArrayEquals(expData[i], preprocessor.apply(i, VectorUtils.of(data[i])).asArray(), 1e-8);
     }
\ No newline at end of file
