diff --git a/old_test.java b/new_test.java
index a0ada70..2a8e5e6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,10 @@
-@Test(expected = APIException.class)
-	@Verifies(value = "should fail if the preferred name is an empty string", method = "validate(Concept)")
-	public void validate_shouldFailIfThePreferredNameIsAnEmptyString() throws Exception {
-		
+@Test
+	@Verifies(value = "should fail if any name is an empty string", method = "validate(Object,Errors)")
+	public void validate_shouldFailIfAnyNameIsAnEmptyString() throws Exception {
 		Concept concept = new Concept();
-		concept.setPreferredName(Context.getLocale(), new ConceptName("", Context.getLocale()));
+		concept.addName(new ConceptName("name", Context.getLocale()));
+		concept.addName(new ConceptName("", Context.getLocale()));
 		Errors errors = new BindException(concept, "concept");
 		new ConceptValidator().validate(concept, errors);
-		Assert.assertTrue(errors.hasErrors());
-		
+		Assert.assertEquals(true, errors.hasErrors());
 	}
\ No newline at end of file
