diff --git a/old_test.java b/new_test.java
index 46514f8..7fa49db 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,10 @@
             {-0.2648886, -0.89044952, 0.3700456},
             {-0.6391588, 0.44947578, 0.6240573}
         };
-        double[][] B = Math.inverse(A);
-        assertTrue(Math.equals(B, Math.transpose(A), 1E-7));
+        DenseMatrix B = Math.inverse(A);
+        for (int i = 0; i < A.length; i++) {
+            for (int j = 0; j < A[i].length; j++) {
+                assertEquals(A[j][i], B.get(i, j), 1E-7);
+            }
+        }
     }
\ No newline at end of file
