diff --git a/old_product.java b/new_product.java
index 140efb6..f13381e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,9 @@
-public static double[][] inverse(double[][] A) {
-        double[][] inv = eye(A[0].length, A.length);
-
+public static DenseMatrix inverse(double[][] A) {
         if (A.length == A[0].length) {
             LUDecomposition lu = new LUDecomposition(A);
-            lu.solve(inv);
+            return lu.inverse();
         } else {
             QRDecomposition qr = new QRDecomposition(A);
-            qr.solve(inv);
+            return qr.inverse();
         }
-
-        return inv;
     }
\ No newline at end of file
