diff --git a/old_test.java b/new_test.java
index ce44451..aabf307 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,7 +14,7 @@
 
         assert !conn.isValid(2) : "Connection must be closed";
 
-        GridTestUtils.assertThrows(log, new Callable<Object>() {
+        assertThrows(log, new Callable<Object>() {
             @Override public Object call() throws Exception {
                 conn.isValid(-2);
 
