diff --git a/old_test.java b/new_test.java
index 4ee51db..a34b03a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,11 @@
 @Test
   public void applyUMaskTest() {
-    Mode umaskPermission = new Mode((short) 0022);
-    Permission permission =
-        new Permission("user1", "group1", Mode.getDefault());
-    permission.applyUMask(umaskPermission);
+    Mode umaskMode = new Mode((short) 0022);
+    Permission permission = new Permission("user1", "group1", Mode.getDefault());
+    permission.applyUMask(umaskMode);
 
-    Assert.assertEquals(Mode.Bits.ALL, permission.getMode().getUserMode());
-    Assert.assertEquals(Mode.Bits.READ_EXECUTE,
-        permission.getMode().getGroupMode());
-    Assert.assertEquals(Mode.Bits.READ_EXECUTE,
-        permission.getMode().getOtherMode());
-    verifyPermissionStatus("user1", "group1", (short) 0755, permission);
+    Assert.assertEquals(Mode.Bits.ALL, permission.getMode().getUserBits());
+    Assert.assertEquals(Mode.Bits.READ_EXECUTE, permission.getMode().getGroupBits());
+    Assert.assertEquals(Mode.Bits.READ_EXECUTE, permission.getMode().getOtherBits());
+    verifyPermission("user1", "group1", (short) 0755, permission);
   }
\ No newline at end of file
