diff --git a/old_product.java b/new_product.java
index 9da1b44..2bf8b31 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public Mode applyUMask(Mode umask) {
-    return new Mode(mUserMode.and(umask.mUserMode.not()), mGroupMode.and(umask.mGroupMode.not()),
-        mOtherMode.and(umask.mOtherMode.not()));
+    return new Mode(mUserBits.and(umask.mUserBits.not()), mGroupBits.and(umask.mGroupBits.not()),
+        mOtherBits.and(umask.mOtherBits.not()));
   }
\ No newline at end of file
