diff --git a/old_test.java b/new_test.java
index 6c7ef2e..c9c1cce 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,16 +1,15 @@
 @Test
     public void testMutate() {
         System.out.println("mutate");
-        int[] father = {1,1,1,0,1,0,0,1,0,0,0};
-        BitString instance = new BitString(father.clone(), null, BitString.Crossover.SINGLE_POINT, 1.0, 1.0);
+
+        MathEx.setSeed(19650218); // to get repeatable results.
+
+        byte[] father = {1,1,1,0,1,0,0,1,0,0,0};
+        BitString instance = new BitString(father.clone(), null, Crossover.SINGLE_POINT, 1.0, 0.1);
         instance.mutate();
-        assertEquals(father.length, instance.length);
-        assertEquals(father.length, instance.bits().length);
-        boolean same = true;
+
+        byte[] mutant = {1,1,1,1,1,0,0,1,0,1,0};
         for (int i = 0; i < father.length; i++) {
-            if (father[i] != instance.bits()[i]) {
-                same = false;
-            }
+            assertEquals(mutant[i], instance.bits()[i]);
         }
-        assertFalse(same);
     }
\ No newline at end of file
