diff --git a/old_test.java b/new_test.java
index cd68b15..b1a4a86 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,4 @@
-@TestTargetNew(
-        level = TestLevel.COMPLETE,
-        method = "getNextEntry",
-        args = {}
-    )
-    public void test_getNextEntry() throws Exception {
+public void test_getNextEntry() throws Exception {
         assertNotNull("getNextEntry failed", zis.getNextEntry());
 
         File resources = Support_Resources.createTempFolder();
