diff --git a/old_product.java b/new_product.java
index 3d200bc..f3a379a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -81,8 +81,5 @@ public ZipEntry getNextEntry() throws IOException {
             }
             currentEntry.setExtra(e);
         }
-        // BEGIN android-added
-        eof = false;
-        // END android-added
         return currentEntry;
     }
\ No newline at end of file
