diff --git a/old_product.java b/new_product.java
index 3ab0ff2..546dfa0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public void setClientReceived() {
-        if (submitEndAnnotation(Constants.CLIENT_RECV, spanCollector())) {
+        if (submitEndAnnotation(Constants.CLIENT_RECV, reporter())) {
             spanAndEndpoint().state().setCurrentClientSpan(null);
         }
     }
\ No newline at end of file
