diff --git a/old_test.java b/new_test.java
index df2543a..467292f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,8 @@
   public void getStatus() throws Exception {
     AlluxioURI file = new AlluxioURI("/file");
     URIStatus status = new URIStatus(new FileInfo());
-    Mockito.when(mFileSystemMasterClient.getStatus(file)).thenReturn(status);
     GetStatusOptions getStatusOptions = GetStatusOptions.defaults();
+    Mockito.when(mFileSystemMasterClient.getStatus(file, getStatusOptions)).thenReturn(status);
     Assert.assertSame(status, mFileSystem.getStatus(file, getStatusOptions));
-    Mockito.verify(mFileSystemMasterClient).getStatus(file);
+    Mockito.verify(mFileSystemMasterClient).getStatus(file, getStatusOptions);
   }
\ No newline at end of file
