diff --git a/old_product.java b/new_product.java
index da96bb5..0e5e3e1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static OSClient getConnectedClient(OpenstackNode osNode) {
-        OpenstackAuth auth = osNode.authentication();
+        OpenstackAuth auth = osNode.keystoneConfig().authentication();
         String endpoint = buildEndpoint(osNode);
         Perspective perspective = auth.perspective();
 
