diff --git a/old_test.java b/new_test.java
index 954a50c..8f0f412 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
 		
 		expect(mockPartition.isUnableToCreateMoreTransactions()).andReturn(true).once();
 		expect(mockPartition.getFreeConnections()).andReturn(mockConnectionHandles).anyTimes();
-		expect(mockPartition.getAvailableConnections()).andReturn(new AtomicInteger(1)).anyTimes();
+		expect(mockPartition.getAvailableConnections()).andReturn(1).anyTimes();
 		expect(mockConnectionHandles.poll()).andReturn(mockConnection).once();
 		mockConnection.setOriginatingPartition(mockPartition);
 		expectLastCall().once();
