diff --git a/old_test.java b/new_test.java
index b7490bf..5f3320d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,7 @@
 @Test
     public void testSetCommands() throws GeniePreconditionException {
-        Assert.assertNull(this.c.getCommands());
+        Assert.assertNotNull(this.c.getCommands());
+        Assert.assertTrue(this.c.getCommands().isEmpty());
         final Command one = new Command();
         one.setId("one");
         final Command two = new Command();
