diff --git a/old_test.java b/new_test.java
index 001e09c..f829224 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,6 @@
         controller.start(ospfAgent, driverService);
         ospfRouter = new OspfRouterImpl();
         controller.addLinkDetails(ospfRouter, new OspfLinkTedImpl());
-        controller.removeLinkDetails(ospfRouter);
+        controller.removeLinkDetails(ospfRouter, new OspfLinkTedImpl());
         assertThat(controller, is(notNullValue()));
     }
\ No newline at end of file
