diff --git a/old_product.java b/new_product.java
index d2561b6..c460037 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public void removeLinkDetails(OspfRouter ospfRouter) {
-        agent.deleteLink(ospfRouter);
+public void removeLinkDetails(OspfRouter ospfRouter, OspfLinkTed ospfLinkTed) {
+        agent.deleteLink(ospfRouter, ospfLinkTed);
     }
\ No newline at end of file
