diff --git a/old_product.java b/new_product.java
index 329028d..48195e1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public StrBuilder appendWithSeparators(final Iterable<?> iterable, String separator) {
         if (iterable != null) {
             separator = ObjectUtils.toString(separator);
-            Iterator<?> it = iterable.iterator();
+            final Iterator<?> it = iterable.iterator();
             while (it.hasNext()) {
                 append(it.next());
                 if (it.hasNext()) {
