diff --git a/old_product.java b/new_product.java
index e976698..90b5d83 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,10 @@ public static double combination(int n, int k) {
         if(n<k) {
             throw new IllegalArgumentException("The n can't be smaller than k.");
         }
-        return factorial(n)/(factorial(k)*factorial(n-k));
+        double combinations=1.0;
+        double lowerBound = n-k;
+        for(int i=n;i>lowerBound;i--) {
+            combinations *= i/(i-lowerBound);
+        }
+        return combinations;
     }
\ No newline at end of file
