diff --git a/old_test.java b/new_test.java
index a0963be..6740a60 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
     public void testCombination() {
         logger.info("combination");
-        int n = 10;
-        int k = 3;
-        double expResult = 120.0;
+        int n = 52;
+        int k = 5;
+        double expResult = 2598960.0;
         double result = Arithmetics.combination(n, k);
         assertEquals(expResult, result, Constants.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
