diff --git a/old_test.java b/new_test.java
index fb00064..467cca6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
 
         data = Math.transpose(data);
 
-        assertEquals(2.73861, Math.colSd(data)[0], 1E-5);
-        assertEquals(2.73861, Math.colSd(data)[1], 1E-5);
-        assertEquals(2.73861, Math.colSd(data)[2], 1E-5);
+        assertEquals(2.73861, Math.colSds(data)[0], 1E-5);
+        assertEquals(2.73861, Math.colSds(data)[1], 1E-5);
+        assertEquals(2.73861, Math.colSds(data)[2], 1E-5);
     }
\ No newline at end of file
