diff --git a/old_product.java b/new_product.java
index ad1917e..3757e31 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,9 +2,9 @@ int literal() throws QueryException {
     final int ch = next();
     if(ch == '[') { // check begin of variable marker
       if(!more()) PICDATE.thrw(info, token); // [$
-      if(!consume(ch)) return -1; // [...
+      if(!consume('[')) return -1; // [...
     } else if(ch == ']') { // check end of variable marker
-      if(!consume(ch)) PICDATE.thrw(info, token); // ]$ or ]...
+      if(!consume(']')) PICDATE.thrw(info, token); // ]$ or ]...
     }
     return ch;
   }
\ No newline at end of file
