diff --git a/old_test.java b/new_test.java
index 46a8ece..0baba4f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -45,6 +45,13 @@
         when(macroDescriptorTranslator.translate(any(org.xwiki.gwt.wysiwyg.client.plugin.macro.MacroDescriptor.class)))
             .thenReturn(translatedDescriptor);
 
+        XWikiContext xcontext = mock(XWikiContext.class);
+        when(xcontext.getWikiId()).thenReturn("xwiki");
+
+        Provider<XWikiContext> contextProvider =
+            mocker.getInstance(new DefaultParameterizedType(null, Provider.class, XWikiContext.class));
+        when(contextProvider.get()).thenReturn(xcontext);
+
         Assert.assertSame(translatedDescriptor, mocker.getComponentUnderTest().getMacroDescriptor(macroId, syntaxId));
 
         ArgumentCaptor<org.xwiki.gwt.wysiwyg.client.plugin.macro.MacroDescriptor> argument =
