diff --git a/old_product.java b/new_product.java
index a72ea52..2343b4c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,17 +1,15 @@
-public Annotations annotateText(Graph graph, final String text) {
-        ConceptInsightsId.validateGenarate(graph, getAccountId());
-        Validate.notNull(text, "text can't be null");
+public Annotations annotateText(final Graph graph, final String text) {
+		IDValidator.getGraphId(graph, getAccountId());
+		Validate.notNull(text, "text can't be null");
 
-        HttpRequestBase request = Request.Post(version + graph.getId() + ANNOTATE_TEXT_PATH)
-                .withContent(text, MediaType.TEXT_PLAIN)
-                .withHeader(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON).build();
+		HttpRequestBase request = Request.Post(API_VERSION + graph.getId() + ANNOTATE_TEXT_PATH)
+				.withContent(text, MediaType.TEXT_PLAIN).withHeader(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON)
+				.build();
 
-        try {
-            HttpResponse response = execute(request);
-            Annotations annotations = GsonSingleton.getGson().fromJson(
-                    ResponseUtil.getString(response), Annotations.class);
-            return annotations;
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
-    }
\ No newline at end of file
+		try {
+			HttpResponse response = execute(request);
+			return ResponseUtil.getObject(response, Annotations.class);
+		} catch (IOException e) {
+			throw new RuntimeException(e);
+		}
+	}
\ No newline at end of file
