diff --git a/old_test.java b/new_test.java
index df71a43..c19d47a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -24,17 +24,17 @@
     Date start = new Date();
 
     CreateSynonymOptions createOptions =
-        new CreateSynonymOptions.Builder(workspaceId, entity, entityValue, synonym).build();
+            new CreateSynonymOptions.Builder(workspaceId, entity, entityValue, synonym).build();
     service.createSynonym(createOptions).execute();
 
     try {
       GetSynonymOptions getOptions =
-          new GetSynonymOptions.Builder(workspaceId, entity, entityValue, synonym).build();
+              new GetSynonymOptions.Builder(workspaceId, entity, entityValue, synonym).build();
       Synonym response = service.getSynonym(getOptions).execute();
 
       assertNotNull(response);
-      assertNotNull(response.getSynonym());
-      assertEquals(response.getSynonym(), synonym);
+      assertNotNull(response.getSynonymText());
+      assertEquals(response.getSynonymText(), synonym);
       assertNotNull(response.getCreated());
       assertNotNull(response.getUpdated());
 
@@ -47,7 +47,7 @@
       fail(ex.getMessage());
     } finally {
       DeleteSynonymOptions deleteOptions =
-          new DeleteSynonymOptions.Builder(workspaceId, entity, entityValue, synonym).build();
+              new DeleteSynonymOptions.Builder(workspaceId, entity, entityValue, synonym).build();
       service.deleteSynonym(deleteOptions).execute();
     }
   }
\ No newline at end of file
