diff --git a/old_product.java b/new_product.java
index ec78f5e..05abb0e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 private Item zipFile(final QueryContext ctx) throws QueryException {
     // check argument
     final ANode elm = (ANode) checkType(expr[0].item(ctx, info), NodeType.ELM);
-    if(!elm.qname().eq(E_FILE)) ZIPUNKNOWN.thrw(info, elm.qname());
+    if(!elm.qname().eq(E_FILE)) ZIP_UNKNOWN.thrw(info, elm.qname());
     // get file
     final String file = attribute(elm, A_HREF, true);
 
@@ -16,7 +16,7 @@ private Item zipFile(final QueryContext ctx) throws QueryException {
       zos.close();
     } catch(final IOException ex) {
       ok = false;
-      ZIPFAIL.thrw(info, ex.getMessage());
+      ZIP_FAIL.thrw(info, ex.getMessage());
     } finally {
       if(fos != null) {
         try { fos.close(); } catch(final IOException ex) { }
