diff --git a/old_test.java b/new_test.java
index 155f5ba..bc3993d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -23,8 +23,8 @@
     checkZipEntry("a/b", new byte[] { '!' });
 
     // error: no entry specified
-    error(_ZIP_ZIP_FILE.args(zipParams("")), Err.ZIPFAIL);
+    error(_ZIP_ZIP_FILE.args(zipParams("")), Err.ZIP_FAIL);
     // error: duplicate entry specified
     error(_ZIP_ZIP_FILE.args(zipParams("<entry src='" + TMPFILE + "'/>" +
-        "<entry src='" + TMPFILE + "'/>")), Err.ZIPFAIL);
+        "<entry src='" + TMPFILE + "'/>")), Err.ZIP_FAIL);
   }
\ No newline at end of file
