diff --git a/old_test.java b/new_test.java
index 7247c74..ff0af4e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,8 @@
     public void isAttached() {
         VplsConfig vpls = createNewVpls();
 
+        assertTrue("Interface not correctly attached to the VPLS",
+                   vpls.isAttached(IF4));
         assertTrue("Interface not correctly attached to the VPLS",
                    vpls.isAttached(IF4));
         assertFalse("Unexpected interface attached to the VPLS",
