diff --git a/old_product.java b/new_product.java
index 56f1e37..cbd7486 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,8 +3,6 @@ public void respond(IRequestCycle requestCycle)
 		WebRequest webRequest = (WebRequest)requestCycle.getRequest();
 		HttpServletRequest request = ((ServletWebRequest)webRequest).getHttpServletRequest();
 
-		final HttpsConfig httpsConfig = Application.get().getSecuritySettings().getHttpsConfig();
-
 		Integer port = null;
 		if (protocol == Protocol.HTTP)
 		{
@@ -33,12 +31,5 @@ public void respond(IRequestCycle requestCycle)
 
 		WebResponse response = (WebResponse)requestCycle.getResponse();
 
-		// an attempt to rewrite a secure jsessionid into nonsecure, doesnt seem to work
-		// Session session = Session.get();
-		// if (!session.isTemporary())
-		// {
-		// response.addCookie(new Cookie("JSESSIONID", session.getId()));
-		// }
-
 		response.sendRedirect(url);
 	}
\ No newline at end of file
