diff --git a/old_test.java b/new_test.java
index 707347e..f08d72f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,20 +15,15 @@
 		final IRequestCycle requestCycle = Mockito.mock(IRequestCycle.class);
 		Mockito.when(requestCycle.getRequest()).thenReturn(webRequest);
 
+		HttpsConfig httpsConfig = new HttpsConfig(80, 1443);
+
 		// request secure communication (over https)
 		final SwitchProtocolRequestHandler handler = new SwitchProtocolRequestHandler(
-			Protocol.HTTPS);
+			Protocol.HTTPS, httpsConfig);
 
 		final WebResponse webResponse = Mockito.mock(WebResponse.class);
 		Mockito.when(requestCycle.getResponse()).thenReturn(webResponse);
 
-		final WebApplication application = new MockApplication();
-
-		// needed to be able to call Application.get().getSecuritySettings().getHttpsConfig()
-		final WicketTester tester = new WicketTester(application);
-		ThreadContext.setApplication(application);
-		application.getSecuritySettings().setHttpsConfig(new HttpsConfig(80, 1443));
-
 		handler.respond(requestCycle);
 
 		Mockito.verify(webResponse).sendRedirect(httpsUrl.toString());
