diff --git a/old_test.java b/new_test.java
index 2bf4069..5c38ada 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
   public void testSlice() {
-    Array arr = Array.empty();
+    XQArray arr = XQArray.empty();
     for(int len = 0; len < 180; len++) {
       assertEquals(len, arr.arraySize());
       for(int pos = 0; pos < len; pos++) {
         for(int k = 0; k <= len - pos; k++) {
-          final Array sub = arr.subArray(pos, k, qc);
+          final XQArray sub = arr.subArray(pos, k, qc);
           assertEquals(k, sub.arraySize());
           sub.checkInvariants();
           final Iterator<Value> iter = sub.iterator(0);
