diff --git a/old_product.java b/new_product.java
index d214925..057a61f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,6 +2,6 @@ static Item[] slice(final Item[] items, final int from, final int to) {
     final Item[] out = new Item[to - from];
     final int in0 = Math.max(0, from), in1 = Math.min(to, items.length);
     final int out0 = Math.max(-from, 0);
-    System.arraycopy(items, in0, out, out0, in1 - in0);
+    Array.copy(items, in0, in1 - in0, out, out0);
     return out;
   }
\ No newline at end of file
