diff --git a/old_product.java b/new_product.java
index 20b7994..bf38b18 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,7 +14,7 @@
 
       try {
 
-        scanner = connector.createScanner(tablename, new Authorizations());
+        scanner = accumuloClient.createScanner(tablename, new Authorizations());
 
         // Never start readahead
         scanner.setReadaheadThreshold(Long.MAX_VALUE);
