diff --git a/old_product.java b/new_product.java
index 2360704..c06b5c3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,7 +16,11 @@
 			}
 		}
 
-		// if patient is voided, we set the membership containing the patient to be voided
-		CohortService cs = Context.getCohortService();
-		cs.patientVoided(patient);
+		Context.addProxyPrivilege(PrivilegeConstants.EDIT_COHORTS);
+		try {
+			Context.getCohortService().notifyPatientVoided(patient);
+		}
+		finally {
+			Context.removeProxyPrivilege(PrivilegeConstants.EDIT_COHORTS);
+		}
 	}
\ No newline at end of file
