diff --git a/old_test.java b/new_test.java
index 8369d02..db20d86 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,6 +12,6 @@
         assertTrue(entryPoint.processFilter(request));
         assertFalse(entryPoint.processFilter(request));
         assertFalse(entryPoint.processFilter(request));
-        assertFalse(entryPoint.processFilter(request));
+        assertTrue(entryPoint.processFilter(request));
         verifyMock();
     }
\ No newline at end of file
