diff --git a/old_test.java b/new_test.java
index d99efc2..f5cec36 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,13 +7,13 @@
 
             final ChronicleQueue chronicle = new SingleChronicleQueueBuilder(getTmpDir())
                     .wireType(this.wireType)
-                    .epoc(System.currentTimeMillis())
+                    .epoch(System.currentTimeMillis())
                     .rollCycle(RollCycles.HOURS)
                     .build();
 
             final ExcerptAppender appender = chronicle.createAppender();
             appender.writeDocument(wire -> wire.write(() -> "key").text("value=v"));
-            Assert.assertTrue(appender.cycle() >= 403476);
+            Assert.assertTrue(appender.cycle() == 0);
 
         } catch (Exception e) {
             e.printStackTrace();
