diff --git a/old_product.java b/new_product.java
index 015bd6d..0bf4918 100644
--- a/old_product.java
+++ b/new_product.java
@@ -33,6 +33,8 @@ public void handleRequest(HttpServletRequest req, HttpServletResponse rsp) throw
         .requestModule(new RequestModule(req, rsp, authResult.get()))
         .build();
     // Apply the selected Route to the component to produce an Action instance, and run it.
+    boolean success = true;
+    DateTime startTime = clock.nowUtc();
     try {
       route.get().instantiator().apply(component).run();
       if (route.get().action().automaticallyPrintOk()) {
@@ -41,5 +43,13 @@ public void handleRequest(HttpServletRequest req, HttpServletResponse rsp) throw
       }
     } catch (HttpException e) {
       e.send(rsp);
+      success = false;
+    } finally {
+      requestMetrics.record(
+          new Duration(startTime, clock.nowUtc()),
+          path,
+          method,
+          authResult.get().authLevel(),
+          success);
     }
   }
\ No newline at end of file
