diff --git a/old_test.java b/new_test.java
index fbffcee..324eb4a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -32,10 +32,13 @@
             .build();
 
         Mockito.when(this.jobPersistenceService.createJobRequest(jobRequest, clientHost)).thenReturn(jobRequest);
+        final Future<?> task = Mockito.mock(Future.class);
+        Mockito.doReturn(task).when(this.taskExecutor).submit(Mockito.any(JobLauncher.class));
 
         this.jobCoordinatorService.coordinateJob(jobRequest, clientHost);
 
-        Mockito.verify(this.taskExecutor, Mockito.times(1)).execute(Mockito.any(JobLauncher.class));
+        Mockito.verify(this.taskExecutor, Mockito.times(1)).submit(Mockito.any(JobLauncher.class));
+        Mockito.verify(this.eventPublisher, Mockito.times(1)).publishEvent(Mockito.any(JobScheduledEvent.class));
 
         final ArgumentCaptor<Job> argument = ArgumentCaptor.forClass(Job.class);
         Mockito.verify(this.jobPersistenceService).createJob(argument.capture());
