diff --git a/old_test.java b/new_test.java
index e80c966..9628e0d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,12 +15,14 @@
         {
             /* replaceIf:primitiveKType KType */ Object /* end:replaceIf */ k = 
                 /* intrinsic:ktypecast */ rnd.nextInt(modulo);
-            assertEquals(deque.removeFirstOccurrence(k), sequence.removeFirstOccurrence(k) >= 0);
+            assertEquals(
+                deque.removeFirstOccurrence(k) >= 0, 
+                sequence.removeFirstOccurrence(k) >= 0);
         }
 
         assertListEquals(deque.toArray(), sequence.toArray());
 
-        assertFalse(deque.removeFirstOccurrence(/* intrinsic:ktypecast */ (modulo + 1)));
+        assertTrue(0 > deque.removeFirstOccurrence(/* intrinsic:ktypecast */ (modulo + 1)));
         deque.addLast(/* intrinsic:ktypecast */ (modulo + 1));
-        assertTrue(deque.removeFirstOccurrence(/* intrinsic:ktypecast */ (modulo + 1)));
+        assertTrue(0 <= deque.removeFirstOccurrence(/* intrinsic:ktypecast */ (modulo + 1)));
     }
\ No newline at end of file
