diff --git a/old_product.java b/new_product.java
index b910b14..24af888 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
-public boolean removeFirstOccurrence(KType e1)
+@Override
+    public int removeFirstOccurrence(KType e1)
     {
         final int index = bufferIndexOf(e1);
         if (index >= 0) removeAtBufferIndex(index);
-        return index >= 0;
+        return index;
     }
\ No newline at end of file
