diff --git a/old_test.java b/new_test.java
index 53b74e4..d10e43e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void testGet() {
-    final Request request = RequestBuilder.get(urlWithQuery).build();
+    final Request request = RequestBuilder.get(HttpUrl.parse(urlWithQuery)).build();
     assertEquals("GET", request.method());
     assertEquals(urlWithQuery, request.url().toString());
   }
\ No newline at end of file
