diff --git a/old_product.java b/new_product.java
index 2ce5752..d9e07f8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,9 @@
-public static boolean getBoolean(String key) {
-        return Boolean.parseBoolean(Settings.getString(key));
+public static boolean getBoolean(String key) throws InvalidSettingException {
+        boolean value;
+        try {
+            value = Boolean.parseBoolean(Settings.getString(key));
+        } catch (NumberFormatException ex) {
+            throw new InvalidSettingException("Could not convert property '" + key + "' to an int.", ex);
+        }
+        return value;
     }
\ No newline at end of file
