diff --git a/old_test.java b/new_test.java
index c1a389c..4b85acc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,6 +9,7 @@
         service.setServicePolicies(null);
         service.setEndpoint("http://host/path/to/svc");
         service.setEndpointType("REST");
+        service.setEndpointContentType("json");
         service.setOrganizationId("test-org");
         service.setPublicService(true);
         service.setServiceId("service-id");
@@ -43,6 +44,7 @@
                 + "\"service\":{"
                     + "\"endpoint\":\"http://host/path/to/svc\","
                     + "\"endpointType\":\"REST\","
+                    + "\"endpointContentType\":\"json\","
                     + "\"publicService\":true,"
                     + "\"organizationId\":\"test-org\","
                     + "\"serviceId\":\"service-id\","
