diff --git a/old_product.java b/new_product.java
index 7ea1fba..568f92e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -21,6 +21,7 @@ public static XContentBuilder marshall(ServiceVersionBean bean) throws StorageEx
                     .field("publicService", bean.isPublicService())
                     .field("endpoint", bean.getEndpoint())
                     .field("endpointType", bean.getEndpointType())
+                    .field("endpointContentType", bean.getEndpointContentType())
                     .field("definitionType", bean.getDefinitionType());
             Set<ServiceGatewayBean> gateways = bean.getGateways();
             if (gateways != null) {
