diff --git a/old_product.java b/new_product.java
index 409af0d..74e507f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,7 +12,7 @@ private Dbl mb(final QueryContext ctx) throws QueryException {
     final Value v = (c ? ItemIter.get(ir) : ir).finish();
 
     // measure resulting memory consumption
-    Performance.gc(3);
+    Performance.gc(2);
     final double d = Performance.mem() - l;
 
     // loop through all results to avoid premature result disposal
