diff --git a/old_test.java b/new_test.java
index c3395b9..45a6e00 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,7 @@
 @Test
   public void testMB() throws BaseXException {
-    // wrong arguments
-    error("util:mb()", "XPST0017");
-    error("util:mb('a','b')", "XPTY0004");
-    error("util:mb('a','b','c')", "XPST0017");
-    error("util:mb(1+)", "XPST0003");
-
-    // measure memory (will always yield different results)
-    query("util:mb(())");
-    query("util:mb(1 to 10000, false())");
-    query("util:mb(1 to 10000, true())");
+    final String fun = check(FunDef.MB, (Class<?>) null, Boolean.class);
+    query(fun + "(())");
+    query(fun + "(1 to 1000, false())");
+    query(fun + "(1 to 1000, true())");
   }
\ No newline at end of file
