diff --git a/old_product.java b/new_product.java
index c8e19b9..1000b5d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,3 @@
-public PagedList<CertificateItem> listCertificates(final String vaultBaseUrl, final Integer maxresults)
-            throws KeyVaultErrorException, IOException, IllegalArgumentException {
+public PagedList<CertificateItem> listCertificates(final String vaultBaseUrl, final Integer maxresults) {
         return innerKeyVaultClient.getCertificates(vaultBaseUrl, maxresults);
     }
\ No newline at end of file
