diff --git a/old_test.java b/new_test.java
index 34f9d2b..8c50f41 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,20 +1,13 @@
 @Test
-    public void test_getJavaScriptHtmlCookieString_withCookieSettings() throws Exception {
+    public void test_getUnencodedJavaScriptHtmlCookieString_htmlCharsInName_htmlCharsInValue() throws Exception {
         try {
-            utils = getUtilsWithHtmlCookieStringMethodMocked();
+            String name = "<cookie name\"'>";
+            String value = ">cookie'<\" value ";
 
-            String name = "some value";
-            String value = "some value";
-            Map<String, String> cookieProps = new HashMap<String, String>();
-            cookieProps.put("key1", "value1");
-            cookieProps.put("key2", "value2");
+            String result = utils.getUnencodedJavaScriptHtmlCookieString(name, value);
 
-            final String mockedCookieString = name + "=" + value + "; key1=value1; key2=value2";
-            setMockedCookieStringExpectation(mockedCookieString);
-
-            String result = utils.getJavaScriptHtmlCookieString(name, value, cookieProps);
-
-            verifyCaseInsensitiveQuotedPatternMatches(result, DOCUMENT_COOKIE_START + mockedCookieString + DOCUMENT_COOKIE_END, "Cookie string did not match expected pattern.");
+            String expectedCookieString = name + "=" + value + ";";
+            verifyCaseInsensitiveQuotedPatternMatches(result, DOCUMENT_COOKIE_START + expectedCookieString + DOCUMENT_COOKIE_END, "Cookie string did not match expected pattern.");
 
         } catch (Throwable t) {
             outputMgr.failWithThrowable(testName.getMethodName(), t);
