diff --git a/old_test.java b/new_test.java
index d61ff99..fc040e4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
   public void saveCTLSchemaTest() throws Exception {
     this.loginKaaAdmin();
-    CtlSchemaDto beta = client.saveCTLSchemaWithAppToken(getResourceAsString(TEST_CTL_SCHEMA_BETA), null, null);
+    CTLSchemaDto beta = client.saveCTLSchemaWithAppToken(getResourceAsString(TEST_CTL_SCHEMA_BETA), null, null);
     Assert.assertNotNull(beta.getId());
 
     this.loginTenantDeveloper(tenantDeveloperUser);
-    CtlSchemaDto alpha = client.saveCTLSchemaWithAppToken(getResourceAsString(TEST_CTL_SCHEMA_ALPHA), tenantDeveloperDto.getTenantId(), null);
+    CTLSchemaDto alpha = client.saveCTLSchemaWithAppToken(getResourceAsString(TEST_CTL_SCHEMA_ALPHA), tenantDeveloperDto.getTenantId(), null);
     Assert.assertNotNull(alpha.getId());
   }
\ No newline at end of file
