diff --git a/old_product.java b/new_product.java
index 1a598fe..9c2e78e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -17,10 +17,15 @@ public static void createThumbnails(
 			throw new NullPointerException("Rename is null.");
 		}
 		
+		ThumbnailParameter param = 
+			new ThumbnailParameterBuilder()
+				.size(width, height)
+				.build();
+		
 		for (File inFile : files)
 		{
 			File outFile = 
-				new File(inFile.getParent(), rename.apply(inFile.getName()));
+				new File(inFile.getParent(), rename.apply(inFile.getName(), param));
 			
 			createThumbnail(inFile, outFile, width, height);
 		}
