diff --git a/old_product.java b/new_product.java
index d2a1aee..248db5c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,10 +5,10 @@ public final void delete(final int pre) {
     int k = kind(pre);
     int s = size(pre, k);
     final boolean empty = pre == 0 && s == meta.size;
-
     // update document index: delete specified entry
-    docindex.delete(this, pre, s);
+    if(!empty) docindex.delete(pre, s);
 
+    // update namespaces
     ns.delete(pre, s);
 
     // reduce size of ancestors
