diff --git a/old_test.java b/new_test.java
index 9a0031e..afcd8e7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
       new RepoManager(context).delete("xyz");
       fail("Not installed package not detected.");
     } catch(final QueryException ex) {
-      error(null, ex, BXRE_WHICH_X);
+      error(null, ex, REPO_NOTFOUND_X);
     }
     // install a package without dependencies (pkg3)
     execute(new RepoInstall(REPO + "pkg3.xar", null));
@@ -38,7 +38,7 @@
       new RepoManager(context).delete(PKG3ID);
       fail("Package involved in a dependency was deleted.");
     } catch(final QueryException ex) {
-      error(null, ex, BXRE_DEP_X_X);
+      error(null, ex, REPO_DELETE_X_X);
     }
     // try to delete pkg4 (use package name)
     execute(new RepoDelete(PKG4, null));
