diff --git a/old_product.java b/new_product.java
index 29f581a..f32aa06 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,5 +5,17 @@
 		Assert.notNull(members, "Members must not be null!");
 		Assert.noNullElements(members, "Members must not contain null!");
 
-		throw new UnsupportedOperationException("Lettuce does currently not supprt GEOHASH.");
+		try {
+			if (isPipelined()) {
+				pipeline(new LettuceResult(getAsyncConnection().geohash(key, members)));
+				return null;
+			}
+			if (isQueueing()) {
+				transaction(new LettuceTxResult(getConnection().geohash(key, members)));
+				return null;
+			}
+			return getConnection().geohash(key, members);
+		} catch (Exception ex) {
+			throw convertLettuceAccessException(ex);
+		}
 	}
\ No newline at end of file
