diff --git a/old_product.java b/new_product.java
index b1a2d62..f430b52 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,8 @@
 public static <T> void shuffle(T[] array) {
         //Implementing Fisher-Yates shuffle
-        RandomSingleton rnd = RandomSingleton.getInstance();
         T tmp;
         for (int i = array.length - 1; i > 0; --i) {
-            int index = rnd.nextInt(i + 1);
+            int index = RandomGenerator.nextInt(i + 1);
             
             tmp = array[index];
             array[index] = array[i];
