diff --git a/old_product.java b/new_product.java
index 02aa0f2..530f393 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,12 @@
 @Override
 	public PatientProgram savePatientProgram(PatientProgram patientProgram) throws DAOException {
-		if (patientProgram.getPatientProgramId() == null) {
+                CustomDatatypeUtil.saveAttributesIfNecessary(patientProgram);
+
+                if (patientProgram.getPatientProgramId() == null) {
 			sessionFactory.getCurrentSession().save(patientProgram);
 		} else {
 			sessionFactory.getCurrentSession().merge(patientProgram);
 		}
+                
 		return patientProgram;
 	}
\ No newline at end of file
