diff --git a/old_product.java b/new_product.java
index 6ada4df..8f40983 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ public static String add(final String name, final String target,
     if(data == null) return PROCNODB;
 
     String trg = path(target);
-    if(!trg.isEmpty()) trg = trg + '/';
+    if(!trg.isEmpty()) trg += '/';
 
     final SAXSource sax = new SAXSource(input);
     final Parser parser = new SAXWrapper(sax, name, trg, ctx.prop);
