diff --git a/old_test.java b/new_test.java
index 8f86e0c..87c9bdd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void testAdd() throws QueryException {
-    final String fun = check(Function.ADD);
+    final String fun = check(Function.DBADD);
 
     query(fun + "('db', document { <root/> }, 'test1.xml')");
     query("count(collection('db/test1.xml')/root) eq 1", "true");
