diff --git a/old_product.java b/new_product.java
index f688c3a..5c2672d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static void convolve(Kernel2D_I32 kernel, GrayS16 image, GrayI16 dest , int skip ) {
+public static void convolve(Kernel2D_S32 kernel, GrayS16 image, GrayI16 dest , int skip ) {
 		checkParameters(image, dest, skip);
 
 		if( kernel.width >= image.width ) {
