diff --git a/old_product.java b/new_product.java
index d95de50..98b9715 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,12 @@
-private Item write(final File path, final QueryContext ctx)
-      throws QueryException {
+private Item write(final File path, final QueryContext ctx,
+      final boolean append) throws QueryException {
 
-    final boolean append = optionalBool(3, ctx);
     if(path.isDirectory()) PATHISDIR.thrw(input, path);
 
     final Iter ir = expr[1].iter(ctx);
     try {
-      final PrintOutput out = new PrintOutput(
-          new BufferedOutputStream(new FileOutputStream(path, append)));
+      final PrintOutput out = PrintOutput.get(
+          new FileOutputStream(path, append));
       try {
         final XMLSerializer xml = new XMLSerializer(out,
             FNGen.serialPar(this, 2, ctx));
