diff --git a/old_product.java b/new_product.java
index 89d9574..968d011 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,8 @@
 public static RESTExec get(final RESTSession rs) throws IOException {
     // create new database or update resource
     final HTTPContext http = rs.http;
-    if(http.depth() == 0) throw HTTPCode.NO_PATH.get();
+    final String db = http.db();
+    if(db.isEmpty()) throw HTTPCode.NO_PATH.get();
 
     RESTCmd.parseOptions(rs);
 
@@ -29,8 +30,8 @@ public static RESTExec get(final RESTSession rs) throws IOException {
     if(parser != null) rs.context.options.set(MainOptions.PARSER, parser);
 
     // store data as XML or raw file, depending on content type
-    final String db = http.db();
-    if(http.depth() == 1) {
+    final String path = http.dbpath();
+    if(path.isEmpty()) {
       if(xml) {
         rs.add(new CreateDB(db), is);
       } else {
@@ -39,7 +40,6 @@ public static RESTExec get(final RESTSession rs) throws IOException {
       }
     } else {
       rs.add(new Open(db));
-      final String path = http.dbpath();
       if(xml) {
         rs.add(new Replace(path), is);
       } else {
