diff --git a/old_test.java b/new_test.java
index 565a6e2..0900339 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,11 +3,4 @@
     parseBadParentId("holy 💩");
     parseBadParentId("-");
     parseBadParentId("");
-
-    assertThat(messages).containsExactly(
-        "parent-id: 463acL$c9f6413ad is not a lower-hex string",
-        "parent-id: holy 💩 is not a lower-hex string",
-        "parent-id should be a 1 to 16 character lower-hex string with no prefix",
-        "parent-id should be a 1 to 16 character lower-hex string with no prefix"
-    );
   }
\ No newline at end of file
