diff --git a/old_product.java b/new_product.java
index 9a010b8..f36980e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,9 +5,7 @@ final <C, K> boolean parseParentId(Propagation.Getter<C, K> getter, C carrier, K
       if (invalidIdLength(key, length, 16)) return false;
 
       parentId = lenientLowerHexToUnsignedLong(parentIdString, 0, length);
-      if (parentId == 0) {
-        maybeLogNotLowerHex(key, parentIdString);
-        return false;
-      }
-      return true;
+      if (parentId != 0) return true;
+      maybeLogNotLowerHex(key, parentIdString);
+      return false;
     }
\ No newline at end of file
