diff --git a/old_product.java b/new_product.java
index 9fdcb4c..7a3928d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,10 +15,10 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 				drugOrder.setConcept(concept);
 			}
 		}
-		if (isDrugOrder && !isDiscontinueOrReviseOrder(order) && order.getAutoExpireDate() == null) {
-			DrugOrder drugOrder = (DrugOrder) order;
-			drugOrder.setAutoExpireDate(drugOrder.getDosingInstructionsInstance().getAutoExpireDate(drugOrder));
+		if (isDrugOrder) {
+			((DrugOrder) order).setAutoExpireDateBasedOnDuration();
 		}
+		
 		if (concept == null) {
 			throw new APIException("concept is required for an order");
 		}
