diff --git a/old_test.java b/new_test.java
index 60d8dc2..b239515 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
 	public void saveOrder_shouldSaveARevisedOrder() throws Exception {
 		Order originalOrder = orderService.getOrder(111);
-		assertTrue(originalOrder.isCurrent());
+		assertTrue(originalOrder.isActive());
 		final Patient patient = originalOrder.getPatient();
 		List<Order> originalActiveOrders = orderService.getActiveOrders(patient, null, null, null);
 		final int originalOrderCount = originalActiveOrders.size();
@@ -19,5 +19,5 @@
 		Thread.sleep(1);
 		List<Order> activeOrders = orderService.getActiveOrders(patient, null, null, null);
 		assertEquals(originalOrderCount, activeOrders.size());
-		assertFalse(originalOrder.isCurrent());
+		assertFalse(originalOrder.isActive());
 	}
\ No newline at end of file
