diff --git a/old_product.java b/new_product.java
index 4d72a43..4adc8fc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,23 +1,23 @@
-public void setState(TachyonURI path, SetAttributeOptions options)
+public void setAttribute(TachyonURI path, SetAttributeOptions options)
       throws FileDoesNotExistException, AccessControlException, InvalidPathException {
-    MasterContext.getMasterSource().incSetStateOps(1);
+    MasterContext.getMasterSource().incSetAttributeOps(1);
     synchronized (mInodeTree) {
       checkPermission(FileSystemAction.WRITE, path, false);
       long fileId = mInodeTree.getInodeByPath(path).getId();
       long opTimeMs = System.currentTimeMillis();
-      setStateInternal(fileId, opTimeMs, options);
-      SetStateEntry.Builder setState =
-          SetStateEntry.newBuilder().setId(fileId).setOpTimeMs(opTimeMs);
-      if (options.hasPinned()) {
-        setState.setPinned(options.getPinned());
+      setAttributeInternal(fileId, opTimeMs, options);
+      SetAttributeEntry.Builder builder =
+          SetAttributeEntry.newBuilder().setId(fileId).setOpTimeMs(opTimeMs);
+      if (options.getPinned() != null) {
+        builder.setPinned(options.getPinned());
       }
-      if (options.hasTtl()) {
-        setState.setTtl(options.getTtl());
+      if (options.getTtl() != null) {
+        builder.setTtl(options.getTtl());
       }
-      if (options.hasPersisted()) {
-        setState.setPersisted(options.getPersisted());
+      if (options.getPersisted() != null) {
+        builder.setPersisted(options.getPersisted());
       }
-      writeJournalEntry(JournalEntry.newBuilder().setSetState(setState).build());
+      writeJournalEntry(JournalEntry.newBuilder().setSetAttribute(builder).build());
       flushJournal();
     }
   }
\ No newline at end of file
