diff --git a/old_test.java b/new_test.java
index 6150f65..91b59ae 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,6 +12,7 @@
         Set<Cell> currentPrediction = new LinkedHashSet<>(); currentPrediction.add(new Cell(new Column(4, 0), 3));
         Classification<Object> classification = new Classification<>();
         double anomalyScore = 0.48d;
+        ComputeCycle cc = new ComputeCycle();
         Object customObject = new Network("CopyTest", NetworkTestHarness.getNetworkDemoTestEncoderParams());
         
         ManualInput mi = new ManualInput()
@@ -27,6 +28,7 @@
         .classifiers(classifiers)
         .storeClassification("foo", classification)
         .anomalyScore(anomalyScore)
+        .computeCycle(cc)
         .customObject(customObject);
         
         ManualInput copy = mi.copy();
@@ -50,6 +52,8 @@
         assertTrue(copy.getPredictiveCells().equals(currentPrediction));
         assertFalse(copy.getPredictiveCells() == currentPrediction);
         
+        assertTrue(copy.getPreviousPredictiveCells().equals(previousPrediction));
+        
         assertTrue(copy.getActiveCells().equals(activeCells));
         assertFalse(copy.getActiveCells() == activeCells);
         
@@ -63,5 +67,10 @@
         
         assertEquals(copy.getAnomalyScore(), anomalyScore, 0.0); // zero deviation
         
+        assertTrue(copy.getComputeCycle().equals(cc));
+        
         assertEquals(copy.getCustomObject(), customObject);
+        
+        assertTrue(mi.equals(copy));
+        assertTrue(mi.hashCode() == copy.hashCode());
     }
\ No newline at end of file
