diff --git a/old_product.java b/new_product.java
index 4749a54..7c035b6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,6 +12,7 @@ ManualInput copy() {
         retVal.classification = new HashMap<>(this.classification);
         retVal.anomalyScore = this.anomalyScore;
         retVal.customObject = this.customObject;
+        retVal.computeCycle = this.computeCycle;
         retVal.activeCells = new LinkedHashSet<Cell>(this.activeCells);
         
         return retVal;
