diff --git a/old_test.java b/new_test.java
index 818664a..78be670 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,7 +15,7 @@
             .atLeastOnce();
       
       for (List<MetricDatum> slice : Iterables.partition(metrics, 10)) {
-         metricClient.putMetricData(slice, namespace);
+         metricClient.putMetricsInNamespace(slice, namespace);
       }
 
       EasyMock.replay(client, metricClient);
