diff --git a/old_product.java b/new_product.java
index 43e6333..933efce 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
 @PrePersist
-    protected void onCreateJob() throws GenieException {
+    @PreUpdate
+    protected void onCreateOrUpdateJob() throws GenieException {
         this.validate(this.commandCriteria, this.commandArgs, this.clusterCriterias, null);
         this.clusterCriteriasString = clusterCriteriasToString(this.clusterCriterias);
         this.commandCriteriaString = commandCriteriaToString(this.commandCriteria);
