diff --git a/old_test.java b/new_test.java
index 92b8516..9122926 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test
-    public void testOnCreateJob() throws GenieException {
+    public void testOnCreateOrUpdateJob() throws GenieException {
         Assert.assertNull(this.job.getId());
         Assert.assertNull(this.job.getClusterCriteriasString());
         Assert.assertNull(this.job.getCommandCriteriaString());
         //Simulate the call stack JPA will make on persist
         this.job.onCreateAuditable();
         this.job.onCreateOrUpdateCommonEntityFields();
-        this.job.onCreateJob();
+        this.job.onCreateOrUpdateJob();
         Assert.assertNotNull(this.job.getId());
         Assert.assertNotNull(this.job.getClusterCriteriasString());
         Assert.assertNotNull(this.job.getCommandCriteriaString());
