diff --git a/old_test.java b/new_test.java
index 9b9ee4c..3e52974 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
     List<RequestInfo> requestInfoList = new ArrayList<RequestInfo>();
     networkClient.close();
     try {
-      networkClient.sendAndPoll(requestInfoList);
+      networkClient.sendAndPoll(requestInfoList, 100);
       Assert.fail("Polling after close should throw");
     } catch (IllegalStateException e) {
     }
