diff --git a/old_product.java b/new_product.java
index 0c313b9..14b96ea 100644
--- a/old_product.java
+++ b/new_product.java
@@ -77,7 +77,9 @@ public boolean divideInternally(double coverageThreshold,
 				Atom[] reprAtoms = Arrays.copyOfRange(subunits.get(s)
 						.getRepresentativeAtoms(), start, end + 1);
 
-				newSubunits.add(new Subunit(reprAtoms));
+				newSubunits.add(new Subunit(reprAtoms, subunits.get(s)
+						.getName(), subunits.get(s).getIdentifier(), subunits
+						.get(s).getStructure()));
 
 				// Recalculate equivalent residues
 				List<Integer> eqr = new ArrayList<Integer>();
@@ -98,7 +100,7 @@ public boolean divideInternally(double coverageThreshold,
 				representative = s;
 		}
 
-		method = SubunitClustererMethod.INTERNAL_SYMMETRY;
+		method = SubunitClustererMethod.STRUCTURE;
 
 		return true;
 	}
\ No newline at end of file
