diff --git a/old_test.java b/new_test.java
index f59fb4f..efd1cc9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,7 @@
 @Test
 	public void testDivideInternally() throws StructureException, IOException {
 
-		AtomCache cache = new AtomCache();
-		cache.setUseMmCif(true);
-		Structure s = cache.getStructure("4e3e");
+		Structure s = StructureIO.getStructure("4e3e");
 
 		// Create a SubunitCluster for the chain
 		SubunitCluster sc1 = new SubunitCluster(
