diff --git a/old_product.java b/new_product.java
index a587d46..259eeb4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public Document getDocument(Document document,String corpusName) {
-        ConceptInsightsId.validateGenarate(document, getAccountId(), corpusName);
-        return executeRequest(version + document.getId(), null, Document.class);
-    }
\ No newline at end of file
+public Document getDocument(final Document document) {
+		String documentId = IDValidator.getDocumentId(document);
+		return executeRequest(API_VERSION + documentId, null, Document.class);
+	}
\ No newline at end of file
