diff --git a/old_test.java b/new_test.java
index e94380f..990a6f4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,5 @@
 @Test
-    public void testGetDocument() {
-        Map<String,Object> params = new HashMap<String, Object>();
-        params.put(ConceptInsights.ACCOUNT_ID, "public");
-        params.put(ConceptInsights.CORPUS, "ibmresearcher");
-        params.put(ConceptInsights.DOCUMENT, "il-AHARONA");
-        Document document = service.getDocument("public", "ibmresearcher", "il-AHARONA");
-        Assert.assertNotNull(document);
-    }
\ No newline at end of file
+	public void testGetDocument() {
+		Document document = service.getDocument(EXAMPLE_DOCUMENT);
+		Assert.assertNotNull(document);
+	}
\ No newline at end of file
