diff --git a/old_product.java b/new_product.java
index d097030..15640c9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
 @Override
-  public void add(T toAdd)
+  public boolean add(T e)
   {
     if (head - tail <= buffermask) {
-      buffer[(int)(head & buffermask)] = toAdd;
+      buffer[(int)(head & buffermask)] = e;
       head++;
-      return;
+      return true;
     }
 
     throw new BufferOverflowException();
