diff --git a/old_product.java b/new_product.java
index 771a3f5..8b5b5d6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,19 +4,19 @@ public boolean changeState(StateVertex nextState) {
 			return false;
 		}
 		LOGGER.debug("Trying to change to state: '{}' from: '{}'", nextState.getName(),
-		        currentState.getName());
+				currentState.getName());
 
 		if (stateFlowGraph.canGoTo(currentState, nextState)) {
 
 			LOGGER.debug("Changed to state: '{}' from: '{}'", nextState.getName(),
-			        currentState.getName());
+					currentState.getName());
 
-			currentState = nextState;
+			setCurrentState(nextState);
 
 			return true;
 		} else {
 			LOGGER.info("Cannot go to state: '{}' from: '{}'", nextState.getName(),
-			        currentState.getName());
+					currentState.getName());
 			return false;
 		}
 	}
\ No newline at end of file
