diff --git a/old_test.java b/new_test.java
index d756039..c5c1979 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,7 @@
 
 		// Add index.
 		Eventable c = new Eventable(new Identification(How.xpath, "/bla"), EventType.click);
-		assertTrue(sm.swithToStateAndCheckIfClone(c, state2, context));
+		assertTrue(sm.switchToStateAndCheckIfClone(c, state2, context));
 
 		// Current index is the new index
 		assertEquals(sm.getCurrentState(), state2);
