diff --git a/old_product.java b/new_product.java
index 720e5af..53a07da 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ public boolean removePartition(String hcatURI) throws MetadataServiceException {
         catch (URISyntaxException e) {
             throw new MetadataServiceException(ErrorCode.E1025, e.getMessage());
         }
-        PartitionWrapper partition = new PartitionWrapper(uri.getServerEndPoint(), uri.getDb(), uri.getTable(),
+        PartitionWrapper partition = new PartitionWrapper(uri.getServer(), uri.getDb(), uri.getTable(),
                 uri.getPartitionMap());
         return removePartition(partition, true);
     }
\ No newline at end of file
