diff --git a/old_test.java b/new_test.java
index bc0381d..b0d1cc0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,9 +5,7 @@
     String group = "test-group";
     short mode = Constants.DEFAULT_FILE_SYSTEM_MODE;
 
-    options.setOwner(owner);
-    options.setGroup(group);
-    options.setMode(mode);
+    options.setPermission(new Permission(owner, group, mode));
 
     CreateUfsFileTOptions thriftOptions = options.toThrift();
     Assert.assertEquals(owner, thriftOptions.getOwner());
