diff --git a/old_product.java b/new_product.java
index 85cb62e..a63bcfe 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,14 @@
 public CompleteUfsFileTOptions toThrift() {
     CompleteUfsFileTOptions options = new CompleteUfsFileTOptions();
-    if (hasGroup()) {
-      options.setGroup(mGroup);
+    if (!mPermission.getOwner().isEmpty()) {
+      options.setOwner(mPermission.getOwner());
     }
-    if (hasOwner()) {
-      options.setOwner(mOwner);
+    if (!mPermission.getGroup().isEmpty()) {
+      options.setGroup(mPermission.getGroup());
     }
-    if (hasMode()) {
-      options.setMode(mMode);
+    short mode = mPermission.getMode().toShort();
+    if (mode != Constants.INVALID_MODE) {
+      options.setMode(mode);
     }
     return options;
   }
\ No newline at end of file
