diff --git a/old_product.java b/new_product.java
index c2c5a11..9f36843 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public static Model readPom(File file) throws AnalysisException {
         Model model = null;
         try {
             PomParser parser = new PomParser();
-            parser.parse(file);
+            model = parser.parse(file);
         } catch (PomParseException ex) {
             final String msg = String.format("Unable to parse pom '%s'", file.getPath());
             LOGGER.log(Level.WARNING, msg);
