diff --git a/old_test.java b/new_test.java
index c3791a1..b7b7ac8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test
     public void testReadPom_File() throws Exception {
-        File file = BaseTest.getResourceAsFile(this, "dwr-xml.pom");
-
+        File file = BaseTest.getResourceAsFile(this, "dwr-pom.xml");
         String expResult = "Direct Web Remoting";
         Model result = PomUtils.readPom(file);
         assertEquals(expResult, result.getName());
