diff --git a/old_test.java b/new_test.java
index 63d731f..a56f0cc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,5 +6,5 @@
         Cron cron = parser.parse(crontab);
         ZonedDateTime date = ZonedDateTime.parse("2014-11-30T00:00:00Z");
         ExecutionTime executionTime = ExecutionTime.forCron(cron);
-        assertEquals(ZonedDateTime.parse("2014-12-01T08:00:00Z"), executionTime.nextExecution(date));
+        assertEquals(ZonedDateTime.parse("2014-12-01T08:00:00Z"), executionTime.nextExecution(date).get());
     }
\ No newline at end of file
