diff --git a/old_product.java b/new_product.java
index 8975e2f..0c004af 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,12 @@
-public ZonedDateTime nextExecution(ZonedDateTime date) {
+public Optional<ZonedDateTime> nextExecution(ZonedDateTime date) {
         Preconditions.checkNotNull(date);
         try {
             ZonedDateTime nextMatch = nextClosestMatch(date);
             if(nextMatch.equals(date)){
                 nextMatch = nextClosestMatch(date.plusSeconds(1));
             }
-            return nextMatch;
+            return Optional.of(nextMatch);
         } catch (NoSuchValueException e) {
-            throw new IllegalArgumentException(e);
+            return Optional.empty();
         }
     }
\ No newline at end of file
