diff --git a/old_test.java b/new_test.java
index 21c791a..ab9c081 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-  public void testClear() throws MetaDataException {
+  public void testClear() throws OperationException {
     testOneAddGet(false, "a", "p", "a", "x", "a", "1", null, null);
     testOneAddGet(false, "a", "q", "b", "y", "a", "2", null, null);
     testOneAddGet(false, "a", null, "c", "z", "a", "1", null, null);
