diff --git a/old_product.java b/new_product.java
index 9cb36ed..17dc461 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,5 @@
   @Override
   public void loadMetadata(AlluxioURI path)
       throws FileDoesNotExistException, IOException, AlluxioException {
-    loadMetadata(path, LoadMetadataOptions.defaults());
+    loadMetadata(path, LoadMetadataPOptions.getDefaultInstance());
   }
\ No newline at end of file
