diff --git a/old_test.java b/new_test.java
index 76ccd67..a45b65f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,17 +1,17 @@
 @Test
-  public void localEndpoint_provisionsOnce() throws Exception {
+  public void linkLocalIp_provisionsOnce() throws Exception {
     // create all the tasks up front so that they are executed with no delay
-    List<Callable<Endpoint>> tasks = new ArrayList<>();
+    List<Callable<String>> tasks = new ArrayList<>();
     for (int i = 0; i < 10; i++) {
-      tasks.add(() -> platform.endpoint());
+      tasks.add(() -> platform.linkLocalIp());
     }
 
     ExecutorService executor = Executors.newFixedThreadPool(tasks.size());
-    List<Future<Endpoint>> futures = executor.invokeAll(tasks);
+    List<Future<String>> futures = executor.invokeAll(tasks);
 
     // check there's only a single unique endpoint returned
     Set<Object> results = Sets.newIdentityHashSet();
-    for (Future<Endpoint> future : futures) {
+    for (Future<String> future : futures) {
       results.add(future.get());
     }
     assertThat(results).hasSize(1);
