diff --git a/old_product.java b/new_product.java
index afcedbc..6418ede 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,6 @@
-public void localEndpoint(Endpoint localEndpoint) {
-    if (localEndpoint == null) throw new NullPointerException("localEndpoint == null");
-    this.localEndpoint = localEndpoint;
+public void remoteServiceName(String remoteServiceName) {
+    if (remoteServiceName == null || remoteServiceName.isEmpty()) {
+      throw new NullPointerException("remoteServiceName is empty");
+    }
+    this.remoteServiceName = remoteServiceName.toLowerCase(Locale.ROOT);
   }
\ No newline at end of file
