diff --git a/old_test.java b/new_test.java
index 3bce614..f568a16 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,9 @@
 @Test
     public void testParseDHMSOrDefault() {
-        assertEquals(93784L, ParseUtil.parseDHMSOrDefault("1-02:03:04", 0L));
-        assertEquals(7384L, ParseUtil.parseDHMSOrDefault("02:03:04", 0L));
-        assertEquals(184L, ParseUtil.parseDHMSOrDefault("03:04", 0L));
+        assertEquals(93784050L, ParseUtil.parseDHMSOrDefault("1-02:03:04.05", 0L));
+        assertEquals(93784000L, ParseUtil.parseDHMSOrDefault("1-02:03:04", 0L));
+        assertEquals(7384000L, ParseUtil.parseDHMSOrDefault("02:03:04", 0L));
+        assertEquals(184050L, ParseUtil.parseDHMSOrDefault("03:04.05", 0L));
+        assertEquals(184000L, ParseUtil.parseDHMSOrDefault("03:04", 0L));
         assertEquals(0L, ParseUtil.parseDHMSOrDefault("04", 0L));
     }
\ No newline at end of file
