diff --git a/old_product.java b/new_product.java
index 08b8870..b9291e7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,15 +1,16 @@
 public static long parseDHMSOrDefault(String s, long defaultLong) {
         Matcher m = DHMS.matcher(s);
-        long seconds = 0L;
+        long milliseconds = 0L;
         if (m.matches()) {
             if (m.group(1) != null) {
-                seconds += parseLongOrDefault(m.group(1), 0L) * 86400L;
+                milliseconds += parseLongOrDefault(m.group(1), 0L) * 86400000L;
             }
             if (m.group(2) != null) {
-                seconds += parseLongOrDefault(m.group(2), 0L) * 3600L;
+                milliseconds += parseLongOrDefault(m.group(2), 0L) * 3600000L;
             }
-            seconds += parseLongOrDefault(m.group(3), 0L) * 60L;
-            seconds += parseLongOrDefault(m.group(4), 0L);
+            milliseconds += parseLongOrDefault(m.group(3), 0L) * 60000L;
+            milliseconds += parseLongOrDefault(m.group(4), 0L) * 1000L;
+            milliseconds += 1000 * parseDoubleOrDefault("0." + m.group(5), 0d);
         }
-        return seconds;
+        return milliseconds;
     }
\ No newline at end of file
