diff --git a/old_test.java b/new_test.java
index dfbb0de..2b0542e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testGetTemporaryFile() throws Exception
     {
-        getContext().setDatabase("wiki");
+        getContext().setWikiId("wiki");
         createEmptyFile("temp/module/wiki/Space/Page/file.txt");
         Assert.assertNotNull(action.getTemporaryFile("/xwiki/bin/temp/Space/Page/module/file.txt", getContext()));
     }
\ No newline at end of file
