diff --git a/old_product.java b/new_product.java
index 5a98295..30b0e45 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,6 +12,9 @@ protected File getTemporaryFile(String uri, XWikiContext context)
             String space = withMinimalURLEncoding(matcher.group(1));
             String page = withMinimalURLEncoding(matcher.group(2));
             String module = withMinimalURLEncoding(matcher.group(3));
+            // The file path is used as is, without any modifications (no decoding/encoding is performed). The modules
+            // that create the temporary files and the corresponding URLs used to access them are responsible for
+            // encoding the file path components so that they don't contain invalid characters.
             String filePath = matcher.group(4);
             String prefix = String.format("temp/%s/%s/%s/%s/", module, wiki, space, page);
             String path = URI.create(prefix + filePath).normalize().toString();
