diff --git a/old_product.java b/new_product.java
index 0bb5d10..07218f3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery gtProp(String propName, String otherProp) {
-        criterion = criterion.and(criterionBuilder.gtProp(propName, otherProp));
+        criterion = criterion.and(Criteria.gtProp(propName, otherProp));
         return this;
     }
\ No newline at end of file
