diff --git a/old_test.java b/new_test.java
index 2a7efc0..d40b170 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
     public void testGtProp() {
-    	assertEquals(new GtPropCriterion("id", "name"), instance.gtProp("id", "name").getQueryCriterion());
+    	assertEquals(Criteria.gtProp("id", "name"), instance.gtProp("id", "name").getQueryCriterion());
     }
\ No newline at end of file
