diff --git a/old_test.java b/new_test.java
index 7098733..22a7165 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,10 +6,11 @@
 		UserService us = Context.getUserService();
 		
 		User u = new User();
+		u.setPerson(new Person());
 		
 		u.addName(new PersonName("Benjamin", "A", "Wolfe"));
 		u.setUsername("bwolfe");
-		u.setGender("M");
+		u.getPerson().setGender("M");
 		
 		User createdUser = us.saveUser(u, "some arbitrary password to use");
 		
