diff --git a/old_product.java b/new_product.java
index 2954aab..2d666a3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,6 @@
 public User saveUser(User user, String password) {
 		
-		insertUserStubIfNeeded(user);
-		
 		sessionFactory.getCurrentSession().saveOrUpdate(user);
-		
 		if (password != null) {
 			//update the new user with the password
 			String salt = Security.getRandomToken();
@@ -12,6 +9,5 @@ public User saveUser(User user, String password) {
 			updateUserPassword(hashedPassword, salt, Context.getAuthenticatedUser().getUserId(), new Date(), user
 			        .getUserId());
 		}
-		
 		return user;
 	}
\ No newline at end of file
