diff --git a/old_product.java b/new_product.java
index d2d29bd..fef9c67 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
 @Authorized(PrivilegeConstants.ADD_ORDERS)
-	public Order discontinueOrder(Order orderToDiscontinue, String reasonNonCoded, Date discontinueDate, Provider orderer)
-	        throws Exception;
\ No newline at end of file
+	public Order discontinueOrder(Order orderToDiscontinue, String reasonNonCoded, Date discontinueDate, Provider orderer,
+	        Encounter encounter) throws Exception;
\ No newline at end of file
