diff --git a/old_test.java b/new_test.java
index 4a5d892..adc0955 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,15 +1,15 @@
 @Test(groups={"tck"})
-    public void test_getMonthStartDayOfYear_notLeapYear() {
-        assertEquals(Month.JANUARY.getMonthStartDayOfYear(false), 1);
-        assertEquals(Month.FEBRUARY.getMonthStartDayOfYear(false), 1 + 31);
-        assertEquals(Month.MARCH.getMonthStartDayOfYear(false), 1 + 31 + 28);
-        assertEquals(Month.APRIL.getMonthStartDayOfYear(false), 1 + 31 + 28 + 31);
-        assertEquals(Month.MAY.getMonthStartDayOfYear(false), 1 + 31 + 28 + 31 + 30);
-        assertEquals(Month.JUNE.getMonthStartDayOfYear(false), 1 + 31 + 28 + 31 + 30 + 31);
-        assertEquals(Month.JULY.getMonthStartDayOfYear(false), 1 + 31 + 28 + 31 + 30 + 31 + 30);
-        assertEquals(Month.AUGUST.getMonthStartDayOfYear(false), 1 + 31 + 28 + 31 + 30 + 31 + 30 + 31);
-        assertEquals(Month.SEPTEMBER.getMonthStartDayOfYear(false), 1 + 31 + 28 + 31 + 30 + 31 + 30 + 31 + 31);
-        assertEquals(Month.OCTOBER.getMonthStartDayOfYear(false), 1 + 31 + 28 + 31 + 30 + 31 + 30 + 31 + 31 + 30);
-        assertEquals(Month.NOVEMBER.getMonthStartDayOfYear(false), 1 + 31 + 28 + 31 + 30 + 31 + 30 + 31 + 31 + 30 + 31);
-        assertEquals(Month.DECEMBER.getMonthStartDayOfYear(false), 1 + 31 + 28 + 31 + 30 + 31 + 30 + 31 + 31 + 30 + 31 + 30);
+    public void test_firstDayOfYear_notLeapYear() {
+        assertEquals(Month.JANUARY.firstDayOfYear(false), 1);
+        assertEquals(Month.FEBRUARY.firstDayOfYear(false), 1 + 31);
+        assertEquals(Month.MARCH.firstDayOfYear(false), 1 + 31 + 28);
+        assertEquals(Month.APRIL.firstDayOfYear(false), 1 + 31 + 28 + 31);
+        assertEquals(Month.MAY.firstDayOfYear(false), 1 + 31 + 28 + 31 + 30);
+        assertEquals(Month.JUNE.firstDayOfYear(false), 1 + 31 + 28 + 31 + 30 + 31);
+        assertEquals(Month.JULY.firstDayOfYear(false), 1 + 31 + 28 + 31 + 30 + 31 + 30);
+        assertEquals(Month.AUGUST.firstDayOfYear(false), 1 + 31 + 28 + 31 + 30 + 31 + 30 + 31);
+        assertEquals(Month.SEPTEMBER.firstDayOfYear(false), 1 + 31 + 28 + 31 + 30 + 31 + 30 + 31 + 31);
+        assertEquals(Month.OCTOBER.firstDayOfYear(false), 1 + 31 + 28 + 31 + 30 + 31 + 30 + 31 + 31 + 30);
+        assertEquals(Month.NOVEMBER.firstDayOfYear(false), 1 + 31 + 28 + 31 + 30 + 31 + 30 + 31 + 31 + 30 + 31);
+        assertEquals(Month.DECEMBER.firstDayOfYear(false), 1 + 31 + 28 + 31 + 30 + 31 + 30 + 31 + 31 + 30 + 31 + 30);
     }
\ No newline at end of file
