diff --git a/old_product.java b/new_product.java
index 471a0f7..71f88df 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,10 +7,9 @@ protected List<Entry> searchCPE(String vendor, String product, String version,
         if (searchString == null) {
             return ret;
         }
-        Query query = queryParser.parse(searchString);
-        TopDocs docs = indexSearcher.search(query, MAX_QUERY_RESULTS);
+        TopDocs docs = cpe.search(searchString, MAX_QUERY_RESULTS);
         for (ScoreDoc d : docs.scoreDocs) {
-            Document doc = indexSearcher.doc(d.doc);
+            Document doc = cpe.getDocument(d.doc);
             Entry entry = Entry.parse(doc);
             entry.setSearchScore(d.score);
             if (!ret.contains(entry)) {
