diff --git a/old_test.java b/new_test.java
index 945b055..815ea91 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-    public void testSearchCPE_5args() throws Exception {
-        System.out.println("searchCPE - 5 args");
+    public void testSearchCPE() throws Exception {
+        System.out.println("searchCPE");
         String vendor = "apache software foundation";
         String product = "struts 2 core";
         String version = "2.1.2";
