diff --git a/old_test.java b/new_test.java
index fd8a7c2..9ee9e6b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,15 @@
 @Test
-    public void testEquals() {
-        VulnerableSoftware obj = new VulnerableSoftware();
-        obj.setCpe("cpe:/a:mortbay:jetty:6.1.0");
-        VulnerableSoftware instance = new VulnerableSoftware();
-        instance.setCpe("cpe:/a:mortbay:jetty:6.1");
+    public void testEquals() throws CpeValidationException {
+        VulnerableSoftwareBuilder builder = new VulnerableSoftwareBuilder();
+        VulnerableSoftware obj = null;
+        VulnerableSoftware instance = builder.part(Part.APPLICATION).vendor("mortbay").product("jetty").version("6.1").build();
         assertFalse(instance.equals(obj));
+
+        obj = builder.part(Part.APPLICATION).vendor("mortbay").product("jetty").version("6.1.0").build();
+        instance = builder.part(Part.APPLICATION).vendor("mortbay").product("jetty").version("6.1").build();
+        assertFalse(instance.equals(obj));
+
+        obj = builder.part(Part.APPLICATION).vendor("mortbay").product("jetty").version("6.1.0").build();
+        instance = builder.part(Part.APPLICATION).vendor("mortbay").product("jetty").version("6.1.0").build();
+        assertTrue(instance.equals(obj));
     }
\ No newline at end of file
