diff --git a/old_product.java b/new_product.java
index ff43fc3..2bed9fd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@
         if (this == other) {
             return true;
         }
-        if (other == null || other.getClass() != this.getClass()) {
+        if (other == null || !(other instanceof NugetPackageReference)) {
             return false;
         }
         final NugetPackageReference o = (NugetPackageReference) other;
