diff --git a/old_test.java b/new_test.java
index b6b153d..b158a54 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(dataProvider="validVariantContexts")
-    public void testCreate(final VariantContext vc, @SuppressWarnings("unused") final ReferenceMultiSource reference) {
-        final StructuralVariantContext svc = StructuralVariantContext.create(vc);
+    public void testOf(final VariantContext vc, @SuppressWarnings("unused") final ReferenceMultiSource reference) {
+        final StructuralVariantContext svc = StructuralVariantContext.of(vc);
         Assert.assertNotNull(svc);
     }
\ No newline at end of file
