diff --git a/old_product.java b/new_product.java
index d0ec8ae..1544710 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static StructuralVariantContext create(final VariantContext vc) {
+public static StructuralVariantContext of(final VariantContext vc) {
         if (vc instanceof StructuralVariantContext) {
             return (StructuralVariantContext) vc;
         } else {
