diff --git a/old_product.java b/new_product.java
index 24399c2..c06bd8b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
 @Override
   public ListenableFuture<List<List<Span>>> getTraces(final QueryRequest request) {
+    checkArgument(request.minDuration == null, "getTraces with duration is unsupported");
     // Over fetch on indexes as they don't return distinct (trace id, timestamp) rows.
     final int traceIndexFetchSize = request.limit * indexFetchMultiplier;
     ListenableFuture<Map<Long, Long>> traceIdToTimestamp;
