diff --git a/old_product.java b/new_product.java
index a0c5d00..7521d0b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,8 +7,8 @@ void computeHistogram(int c_x, int c_y, double sigma) {
 		bound.x1 = c_x + r + 1;
 		bound.y1 = c_y + r + 1;
 
-		ImageSingleBand rawDX = derivX.getImage();
-		ImageSingleBand rawDY = derivY.getImage();
+		ImageGray rawDX = derivX.getImage();
+		ImageGray rawDY = derivY.getImage();
 
 		// make sure it is contained in the image bounds
 		BoofMiscOps.boundRectangleInside(rawDX,bound);
