diff --git a/old_test.java b/new_test.java
index badc8ce..7b4267d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
 	public void computeHistogram() {
-		ImageFloat32 derivX = new ImageFloat32(width, height);
-		ImageFloat32 derivY = new ImageFloat32(width, height);
+		GrayF32 derivX = new GrayF32(width, height);
+		GrayF32 derivY = new GrayF32(width, height);
 		int N = 36;
-		OrientationHistogramSift<ImageFloat32> alg = new OrientationHistogramSift<ImageFloat32>(N, 1.5,ImageFloat32.class);
+		OrientationHistogramSift<GrayF32> alg = new OrientationHistogramSift<GrayF32>(N, 1.5,GrayF32.class);
 		alg.setImageGradient(derivX, derivY);
 
 		for (int degrees = 5; degrees < 360; degrees+=10) {
