diff --git a/old_product.java b/new_product.java
index a50f232..5bd651b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,9 +15,12 @@ protected static History parseChanges(Reader fileHistory) throws IOException {
                 int year = Integer.parseInt(matcher.group(2));
                 int month = Integer.parseInt(matcher.group(3));
                 int day = Integer.parseInt(matcher.group(4));
-                entry.setDate(newDate(year, month, day));
-                entry.setAuthor(matcher.group(5));
-                entry.setMessage(matcher.group(6).trim());
+                int hour = Integer.parseInt(matcher.group(5));
+                int minute = Integer.parseInt(matcher.group(6));
+                int second = Integer.parseInt(matcher.group(7));
+                entry.setDate(newDate(year, month, day, hour, minute, second));
+                entry.setAuthor(matcher.group(8));
+                entry.setMessage(matcher.group(9).trim());
                 entry.setActive(true);
                 entries.add(entry);
             }
