diff --git a/old_test.java b/new_test.java
index 88ec7c1..c2bfd37 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
     public void parseChanges() throws Exception {
-        String output = "Change 1234 on 2008/10/13 by ADMIN@UserWorkspaceName 'Comment given to changelist within single qoutes, this is change one'\n" +
-                "Change 6543 on 2008/10/08 by USER@USER_WS 'Comment given to changelist within single qoutes'\n" +
-                "Change 7654 on 2008/09/30 by USER@USER_WS 'Comment given to changelist within single qoutes'\n" +
-                "Change 2345 on 2008/09/30 by ADMIN@Workspace2 'Comment given to changelist within single qoutes'\n";
+        String output = "Change 1234 on 2008/10/13 11:30:00 by ADMIN@UserWorkspaceName 'Comment given to changelist within single qoutes, this is change one'\n" +
+                "Change 6543 on 2008/10/08 18:25:38 by USER@USER_WS 'Comment given to changelist within single qoutes'\n" +
+                "Change 7654 on 2008/09/30 01:00:01 by USER@USER_WS 'Comment given to changelist within single qoutes'\n" +
+                "Change 2345 on 2008/09/30 17:45:33 by ADMIN@Workspace2 'Comment given to changelist within single qoutes'\n";
         History result = PerforceHistoryParser.parseChanges(new StringReader(output));
 
         assertNotNull(result);
