diff --git a/old_test.java b/new_test.java
index afa7066..6077169 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,8 @@
 
 		ZooKeeperStateHandleStore<Long> zkStore = new ZooKeeperStateHandleStore<>(
 			ZOOKEEPER.getClient(),
-			longStateStorage);
+			longStateStorage,
+			Executors.directExecutor());
 
 		final Collection<String> paths = Arrays.asList("/state1", "/state2", "/state3");
 
