diff --git a/old_test.java b/new_test.java
index bbbc603..c06afc0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void mountTest() throws Exception {
-    TachyonURI tachyonPath = new TachyonURI("/t");
-    TachyonURI ufsPath = new TachyonURI("/u");
+    AlluxioURI tachyonPath = new AlluxioURI("/t");
+    AlluxioURI ufsPath = new AlluxioURI("/u");
     MountOptions mountOptions = MountOptions.defaults();
     Mockito.doNothing().when(mFileSystemMasterClient).mount(tachyonPath, ufsPath);
     mFileSystem.mount(tachyonPath, ufsPath, mountOptions);
