diff --git a/old_test.java b/new_test.java
index c13d8e5..d9046f1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,21 +11,19 @@
                     int expectedLength = (int) l;
                     for (long a : addresses) {
                         int expectedAddress = (int) a;
-                        long m = layout().newBlockMetadata(first, expectedBlockId, expectedLength, expectedAddress);
+                        long m = layout().newBlockMetadata(expectedBlockId, expectedLength, expectedAddress);
                         Assert.assertTrue(layout().isUsedBlock(m));
-                        Assert.assertEquals(first, layout().isFirstBlock(m));
                         Assert.assertEquals(expectedBlockId, layout().getNextFreeBlockId(m));
                         Assert.assertEquals(expectedLength, layout().getLength(m));
-                        Assert.assertEquals(expectedAddress, layout().getSuccessorAddress(m));
+                        Assert.assertEquals(expectedAddress, layout().getPredecessorAddress(m));
                     }
                 }
             }
         }
 
         // Test the empty metadata.
-        Assert.assertFalse(layout().isFirstBlock(layout().emptyBlockMetadata()));
         Assert.assertFalse(layout().isUsedBlock(layout().emptyBlockMetadata()));
         Assert.assertEquals(CacheLayout.NO_BLOCK_ID, layout().getNextFreeBlockId(layout().emptyBlockMetadata()));
         Assert.assertEquals(0, layout().getLength(layout().emptyBlockMetadata()));
-        Assert.assertEquals(CacheLayout.NO_ADDRESS, layout().getSuccessorAddress(layout().emptyBlockMetadata()));
+        Assert.assertEquals(CacheLayout.NO_ADDRESS, layout().getPredecessorAddress(layout().emptyBlockMetadata()));
     }
\ No newline at end of file
