diff --git a/old_product.java b/new_product.java
index e94de5d..3fc2fbc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public boolean optimizedParam( List<List<Point2D_F64>> observations ,
+public boolean optimizedParam( List<CalibrationObservation> observations ,
 								   List<Point2D_F64> grid ,
 								   Zhang99ParamAll initial ,
 								   Zhang99ParamAll found ,
@@ -19,7 +19,7 @@ public boolean optimizedParam( List<List<Point2D_F64>> observations ,
 
 		Zhang99OptimizationJacobian jacobian = new Zhang99OptimizationJacobian(
 				initial.assumeZeroSkew,initial.radial.length,initial.includeTangential,
-				observations.size(),grid);
+				observations,grid);
 
 		optimizer.setFunction(func,jacobian);
 		optimizer.initialize(model,1e-10,1e-25*observations.size());
