diff --git a/old_test.java b/new_test.java
index 02aeaec..3d83ba8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,11 +2,11 @@
 	public void testToString()
 	{
 		assertEquals("1 bytes", Bytes.bytes(1).toString());
-		assertEquals("1K", Bytes.bytes(1024).toString());
-		assertEquals("1M", Bytes.bytes(1024 * 1024L).toString());
-		assertEquals("1G", Bytes.bytes(1024 * 1024 * 1024L).toString());
-		assertEquals("1T", Bytes.bytes(1024 * 1024 * 1024 * 1024L).toString());
-		assertEquals("1.5K", Bytes.bytes(1024 * 1.5).toString());
+		assertEquals("1KB", Bytes.bytes(1024).toString());
+		assertEquals("1MB", Bytes.bytes(1024 * 1024L).toString());
+		assertEquals("1GB", Bytes.bytes(1024 * 1024 * 1024L).toString());
+		assertEquals("1TB", Bytes.bytes(1024 * 1024 * 1024 * 1024L).toString());
+		assertEquals("1.5KB", Bytes.bytes(1024 * 1.5).toString());
 
 		assertEquals("1 bytes", Bytes.bytes(1).toString(Locale.GERMAN));
 	}
\ No newline at end of file
