diff --git a/old_product.java b/new_product.java
index 068cf77..7cce4d7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,22 +2,22 @@ public String toString(final Locale locale)
 	{
 		if (terabytes() >= 1.0)
 		{
-			return unitString(terabytes(), "T", locale);
+			return unitString(terabytes(), "TB", locale);
 		}
 
 		if (gigabytes() >= 1.0)
 		{
-			return unitString(gigabytes(), "G", locale);
+			return unitString(gigabytes(), "GB", locale);
 		}
 
 		if (megabytes() >= 1.0)
 		{
-			return unitString(megabytes(), "M", locale);
+			return unitString(megabytes(), "MB", locale);
 		}
 
 		if (kilobytes() >= 1.0)
 		{
-			return unitString(kilobytes(), "K", locale);
+			return unitString(kilobytes(), "KB", locale);
 		}
 
 		return Long.toString(value) + " bytes";
