diff --git a/old_product.java b/new_product.java
index bf4c92c..72fcada 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,21 +1,21 @@
 @Override
 	public boolean hasNext() {
 
-		if (this.traversalStack.isEmpty()) {
+		if (traversalStack.isEmpty()) {
 
-			if (this.numVisitedEntryVertices < 0) {
+			if (numVisitedEntryVertices < 0) {
 				// User chose a specific starting vertex
 				return false;
 			}
 
-			++this.numVisitedEntryVertices;
+			numVisitedEntryVertices++;
 
-			if (this.forward) {
-				if (this.managementGraph.getNumberOfInputVertices(this.startStage) <= this.numVisitedEntryVertices) {
+			if (forward) {
+				if (managementGraph.getNumberOfInputVertices(this.startStage) <= numVisitedEntryVertices) {
 					return false;
 				}
 			} else {
-				if (this.managementGraph.getNumberOfOutputVertices(this.startStage) <= this.numVisitedEntryVertices) {
+				if (managementGraph.getNumberOfOutputVertices(this.startStage) <= numVisitedEntryVertices) {
 					return false;
 				}
 			}
