diff --git a/old_test.java b/new_test.java
index b434a7e..4a1960c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
-@Test
+@Test @Config(minSdk = VERSION_CODES.N)
   public void applyRules_rtlScript() throws Exception {
     applyQualifiers("he");
     DeviceConfig.applyRules(configuration, displayMetrics, apiLevel);
 
     assertThat(asQualifierString())
-        .isEqualTo("iw-ldrtl-sw320dp-w320dp-h470dp-normal-notlong-notround-port-notnight-mdpi-finger-keyssoft-nokeys-navhidden-nonav");
+        .isEqualTo("iw-ldrtl-sw320dp-w320dp-h470dp-normal-notlong-notround-" + optsForO + "port-notnight-mdpi-finger-keyssoft-nokeys-navhidden-nonav");
   }
\ No newline at end of file
