diff --git a/old_product.java b/new_product.java
index 6733812..b7f285b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,7 +15,7 @@
                 case NANO_OF_SECOND: return (newValue != nanos ? create(seconds, (int) newValue) : this);
                 case INSTANT_SECONDS: return (newValue != seconds ? create(newValue, nanos) : this);
             }
-            throw new DateTimeException("Unsupported field: " + field.getName());
+            throw new UnsupportedTemporalTypeException("Unsupported field: " + field.getName());
         }
         return field.adjustInto(this, newValue);
     }
\ No newline at end of file
