diff --git a/old_product.java b/new_product.java
index 9c38731..01e5093 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,6 +11,8 @@ public void validate(Object obj, Errors errors) {
 		
 		Patient patient = (Patient) obj;
 		
+		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "gender", "Person.gender.required");
+		
 		if (PatientIdentifierValidator.hasMoreThanOneIdentifierForSameIdentifierType(patient.getActiveIdentifiers())) {
 			errors.reject("error.duplicateIdentifierTypes");
 		}
