diff --git a/old_test.java b/new_test.java
index b8b20bf..718184a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,12 @@
 @Test
   public void serialize() {
     serialError("<a/>", ""); // invalid tag
-    serialError("<json/>", ""); // no type specified
     serialError("<json type='o'/>", ""); // invalid type
     serial("<json type='object'/>", "", "{}");
     serial("<json objects='json'/>", "", "{}");
     serial("<json type='array'/>", "", "[]");
     serial("<json arrays='json'/>", "", "[]");
-    serialError("<json type='number'>1</json>", ""); // no text allowed in json tag
+    serial("<json type='number'>1</json>", "", "1"); // no text allowed in json tag
     serial("<json type='array'><_ type='null'/></json>", "", "[null]");
     serialError("<json type='array'><_ type='number'/></json>", ""); // value needed
     serialError("<json type='array'><_ type='boolean'/></json>", ""); // value needed
