diff --git a/old_product.java b/new_product.java
index e94b08d..64476fe 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,9 @@
-private void serialize(final String query, final String expected, final JsonFormat format) {
-    serialize(query, expected, format, JsonSpec.RFC4627);
+private static void serialize(final String query, final String expected,
+      final JsonFormat format) {
+    try {
+      final String actual = serialize(query, format);
+      assertEquals("\n[E] " + expected + "\n[F] " + actual + '\n', expected, actual);
+    } catch(final Exception ex) {
+      fail(ex.toString());
+    }
   }
\ No newline at end of file
