diff --git a/old_product.java b/new_product.java
index f27f529..efcc61d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
-    public ManagedCompletableFuture<T> exceptionally(Function<Throwable, ? extends T> action) {
+    public CompletableFuture<T> exceptionally(Function<Throwable, ? extends T> action) {
         // Reject ManagedTask so that we have the flexibility to decide later how to handle ManagedTaskListener and execution properties
         if (action instanceof ManagedTask)
             throw new IllegalArgumentException(ManagedTask.class.getName());
@@ -14,6 +14,6 @@
             CompletableFuture<T> dependentStage = completableFuture.exceptionally(action);
             return new ManagedCompletableFuture<T>(dependentStage, defaultExecutor, null);
         } else {
-            return (ManagedCompletableFuture<T>) super.exceptionally(action);
+            return super.exceptionally(action);
         }
     }
\ No newline at end of file
