diff --git a/old_product.java b/new_product.java
index a204af3..5515062 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public DateTimeFormatterBuilder appendFraction(
-            DateTimeField field, int minWidth, int maxWidth) {
-        appendInternal(new FractionPrinterParser(field, minWidth, maxWidth));
+            DateTimeField field, int minWidth, int maxWidth, boolean decimalPoint) {
+        appendInternal(new FractionPrinterParser(field, minWidth, maxWidth, decimalPoint));
         return this;
     }
\ No newline at end of file
