diff --git a/old_test.java b/new_test.java
index c0b5cb8..ab5b985 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
     public void test_appendFraction_3arg() throws Exception {
-        builder.appendFraction(MINUTE_OF_HOUR, 1, 9);
+        builder.appendFraction(MINUTE_OF_HOUR, 1, 9, false);
         DateTimeFormatter f = builder.toFormatter();
         assertEquals(f.toString(), "Fraction(MinuteOfHour,1,9)");
     }
\ No newline at end of file
