diff --git a/old_product.java b/new_product.java
index 728e879..5dc3c8b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,18 +1,7 @@
 public void finishSpan() {
-        long endTick = System.nanoTime();
-
         Span span = spanAndEndpoint().span();
         if (span == null) return;
 
-        Long startTick;
-        synchronized (span) {
-            startTick = span.startTick;
-        }
-        final long duration;
-        if (startTick != null) {
-            duration = Math.max(1L, (endTick - startTick) / 1000L);
-        } else {
-            duration = Math.max(1L, clock().currentTimeMicroseconds() - span.getTimestamp());
-        }
+        long duration = Math.max(1L, clock().currentTimeMicroseconds() - span.getTimestamp());
         internalFinishSpan(span, duration);
     }
\ No newline at end of file
