diff --git a/old_test.java b/new_test.java
index bc4a310..c2501e9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,9 @@
 @Test
     public void finishSpan_lessThanMicrosRoundUp() {
-        Span finished = new Span().setName("foo").setTimestamp(1000L); // set in start span
-        finished.startTick = 500L; // set in start span
+        Span finished = new Span().setName("foo").setTimestamp(START_TIME_MICROSECONDS);
         state.setCurrentLocalSpan(finished);
 
-        PowerMockito.when(System.nanoTime()).thenReturn(1000L);
+        PowerMockito.when(System.nanoTime()).thenReturn(50L);
 
         localTracer.finishSpan();
 
