diff --git a/old_test.java b/new_test.java
index 55afaad..4b9823c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,15 +1,17 @@
 @Test
   public void addKeyTest() {
-    String key = "key";
+    PropertyKey key = PropertyKey.HOME;
     org.apache.hadoop.conf.Configuration conf = new org.apache.hadoop.conf.Configuration();
     Configuration.set(key, "alluxioKey");
 
-    System.setProperty(key, "systemKey");
+    System.setProperty(key.toString(), "systemKey");
+    ConfigurationTestUtils.resetConfiguration();
     HdfsUnderFileSystemUtils.addKey(conf, key);
-    Assert.assertEquals("systemKey", conf.get(key));
+    Assert.assertEquals("systemKey", conf.get(key.toString()));
 
-    System.clearProperty(key);
+    System.clearProperty(key.toString());
+    ConfigurationTestUtils.resetConfiguration();
     HdfsUnderFileSystemUtils.addKey(conf, key);
-    Assert.assertEquals("alluxioKey", conf.get(key));
+    Assert.assertEquals("alluxioKey", conf.get(key.toString()));
     ConfigurationTestUtils.resetConfiguration();
   }
\ No newline at end of file
