diff --git a/old_product.java b/new_product.java
index 9b44406..da670d8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,5 @@
-public static void addKey(org.apache.hadoop.conf.Configuration hadoopConf, String key) {
-    if (System.getProperty(key) != null) {
-      hadoopConf.set(key, System.getProperty(key));
-    } else if (Configuration.get(key) != null) {
-      hadoopConf.set(key, Configuration.get(key));
+public static void addKey(org.apache.hadoop.conf.Configuration hadoopConf, PropertyKey key) {
+    if (Configuration.containsKey(key)) {
+      hadoopConf.set(key.toString(), Configuration.get(key));
     }
   }
\ No newline at end of file
