diff --git a/old_product.java b/new_product.java
index 606ea6a..0095e75 100644
--- a/old_product.java
+++ b/new_product.java
@@ -28,18 +28,16 @@ private void serialize(final DBNode node) throws IOException {
         openDoc(data.text(p++, true));
         doc = true;
       } else if(k == Data.TEXT) {
-        final FTPos ftd = ft != null ? ft.get(data, p) : null;
-        if(ftd != null) text(data.text(p++, true), ftd);
-        else text(data.text(p++, true));
+        prepareText(data.text(p++, true), ft != null ? ft.get(data, p) : null);
       } else if(k == Data.COMM) {
-        comment(data.text(p++, true));
+        prepareComment(data.text(p++, true));
       } else {
         if(k == Data.PI) {
-          pi(data.name(p, Data.PI), data.atom(p++));
+          preparePi(data.name(p, Data.PI), data.atom(p++));
         } else {
           // add element node
           final byte[] name = data.name(p, k);
-          startElement(name);
+          openElement(name);
 
           // add namespace definitions
           if(nsp != null) {
