diff --git a/old_test.java b/new_test.java
index 3ced249..9126aca 100644
--- a/old_test.java
+++ b/new_test.java
@@ -21,6 +21,6 @@
         "return validate:dtd($doc, $dtd) "), "true");
 
     // invalid arguments
-    error(_VALIDATE_DTD_INFO.args("unknown"), Err.WHICHRES);
-    error(_VALIDATE_DTD_INFO.args(FILE, "unknown.dtd"), Err.WHICHRES);
+    error(_VALIDATE_DTD_INFO.args("unknown"), Err.WHICHRES_X);
+    error(_VALIDATE_DTD_INFO.args(FILE, "unknown.dtd"), Err.WHICHRES_X);
   }
\ No newline at end of file
