diff --git a/old_product.java b/new_product.java
index 1482dfb..747fb32 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@ private Value dtdInfo(final QueryContext qc) throws QueryException {
       void process(final ErrorHandler handler)
           throws IOException, ParserConfigurationException, SAXException, QueryException {
 
-        final Item it = checkItem(exprs[0], qc);
+        final Item it = toItem(exprs[0], qc);
         SerializerOptions sp = null;
 
         // integrate doctype declaration via serialization parameters
