diff --git a/old_product.java b/new_product.java
index 24ddb75..6c52a7c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,7 @@
 @Override
     public String toString() {
-        StringBuilder buf = new StringBuilder();
+        StringBuilder buf = new StringBuilder(32);
         buf.append("OffsetInfo[")
-            .append(dateTime)
-            .append(' ')
             .append(isTransition() ? transition : offset)
             .append(']');
         return buf.toString();
