diff --git a/old_test.java b/new_test.java
index dc3077d..f5262c6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
-@Test(groups={"tck"})
+@Test(groups={"implementation"})
     public void test_toString_normal() {
-        ZoneOffsetInfo test = new ZoneOffsetInfo(LocalDateTime.of(2010, 3, 31, 1, 0), OFFSET_0200, null);
-        assertEquals(test.toString(), "OffsetInfo[2010-03-31T01:00 +02:00]");
+        ZoneOffsetInfo test = make(OFFSET_0200, null);
+        assertEquals(test.toString(), "OffsetInfo[+02:00]");
     }
\ No newline at end of file
