diff --git a/old_test.java b/new_test.java
index a8e8c36..bfe72ac 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,7 @@
 @Test
     public void testAuthenticate() throws Exception {
         SAMLMessageContext context = new SAMLMessageContext();
+        context.setCommunicationProfileId(SAMLConstants.SAML2_WEBSSO_PROFILE_URI);
 
         SAMLAuthenticationToken token = new SAMLAuthenticationToken(context, messageStorage);
         SAMLMessageStorage store = token.getMessageStore();
