diff --git a/old_product.java b/new_product.java
index fa18cf3..0fbc926 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,8 +10,16 @@ public Authentication authenticate(Authentication authentication) throws Authent
         SAMLCredential credential;
 
         try {
-            // TODO hok - use correct processing filter based on profile
-            credential = consumer.processAuthenticationResponse(context, store);
+            if (SAMLConstants.SAML2_WEBSSO_PROFILE_URI.equals(context.getCommunicationProfileId())) {
+                credential = consumer.processAuthenticationResponse(context, store);
+            } else if (SAMLConstants.SAML2_HOK_WEBSSO_PROFILE_URI.equals(context.getCommunicationProfileId())) {
+                credential = hokConsumer.processAuthenticationResponse(context, store);
+            } else {
+                throw new SAMLException("Unsupported profile encountered in the context " + context.getCommunicationProfileId());
+            }
+        } catch (SAMLRuntimeException e) {
+            samlLogger.log(SAMLConstants.AUTH_N_RESPONSE, SAMLConstants.FAILURE, context, e);
+            throw new AuthenticationServiceException("Error validating SAML message", e);
         } catch (SAMLException e) {
             samlLogger.log(SAMLConstants.AUTH_N_RESPONSE, SAMLConstants.FAILURE, context, e);
             throw new AuthenticationServiceException("Error validating SAML message", e);
