diff --git a/old_test.java b/new_test.java
index b36fcbe..ad4e7b8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,8 +15,15 @@
     Mockito.when(clusterVersionEntity.getRepositoryVersion()).thenReturn(repositoryVersionEntity);
     Mockito.when(repositoryVersionEntity.getVersion()).thenReturn("2.2.4.2");
 
+    Map<String, String> checkProperties = new HashMap<String, String>();
+    checkProperties.put("min-applicable-stack-version","HDP-2.2.4.2");
+    PrerequisiteCheckConfig prerequisiteCheckConfig = Mockito.mock(PrerequisiteCheckConfig.class);
+    Mockito.when(prerequisiteCheckConfig.getCheckProperties(
+        m_check.getClass().getName())).thenReturn(checkProperties);
+
     PrereqCheckRequest request = new PrereqCheckRequest("cluster");
     request.setRepositoryVersion("2.3.0.0");
+    request.setPrerequisiteCheckConfig(prerequisiteCheckConfig);
 
     // YARN not installed
     Assert.assertFalse(m_check.isApplicable(request));
