diff --git a/old_product.java b/new_product.java
index c135847..847e25e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,14 +10,19 @@
       return false;
     }
 
-    // not applicable if not HDP 2.2.4.2 or later
+    if(null == _configuration)
+      _configuration = new Configuration();
+    
+    String rollingUpgradeStack = _configuration.getRollingUpgradeStack(); 
+    // not applicable if not HDP 2.2.4.2 or later    
     String stackName = cluster.getCurrentStackVersion().getStackName();
-    if (!"HDP".equals(stackName)) {
+    if (!rollingUpgradeStack.equals(stackName)) {
       return false;
     }
 
+    String rollingUpgradeVersion = _configuration.getRollingUpgradeVersion();
     String currentClusterRepositoryVersion = cluster.getCurrentClusterVersion().getRepositoryVersion().getVersion();
-    if (VersionUtils.compareVersions(currentClusterRepositoryVersion, "2.2.4.2") < 0) {
+    if (VersionUtils.compareVersions(currentClusterRepositoryVersion, rollingUpgradeVersion) < 0) {
       return false;
     }
 
