diff --git a/old_test.java b/new_test.java
index 4d2605c..5bf34f5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testGetLatitudeFromY01() {
-        checkLatitude(TileSystem.MaxLatitude, TileSystem.getLatitudeFromY01(0, true));
-        checkLatitude(0, TileSystem.getLatitudeFromY01(0.5, true));
-        checkLatitude(TileSystem.MinLatitude, TileSystem.getLatitudeFromY01(1, true));
+        checkLatitude(tileSystem.getMaxLatitude(), tileSystem.getLatitudeFromY01(0, true));
+        checkLatitude(0, tileSystem.getLatitudeFromY01(0.5, true));
+        checkLatitude(tileSystem.getMinLatitude(), tileSystem.getLatitudeFromY01(1, true));
     }
\ No newline at end of file
