diff --git a/old_product.java b/new_product.java
index b77e2f6..30da370 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static double getLatitudeFromY01(final double pY01, boolean wrapEnabled) {
-		double latitude = 90 - 360 * Math.atan(Math.exp((pY01 - 0.5) * 2 * Math.PI)) / Math.PI;
-		return wrapEnabled ? Clip(latitude, MinLatitude, MaxLatitude) : latitude;
+public double getLatitudeFromY01(final double pY01, boolean wrapEnabled) {
+		final double latitude = getLatitudeFromY01(wrapEnabled ? Clip(pY01, 0, 1) : pY01);
+		return wrapEnabled ? Clip(latitude, getMinLatitude(), getMaxLatitude()) : latitude;
 	}
\ No newline at end of file
