diff --git a/old_product.java b/new_product.java
index 3fd9e4e..584c4cd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-static void makeVersionInfoBits(int version, BitArray bits) throws WriterException {
-    bits.appendBits(version, 6);
-    int bchCode = calculateBCHCode(version, VERSION_INFO_POLY);
+static void makeVersionInfoBits(Version version, BitArray bits) throws WriterException {
+    bits.appendBits(version.getVersionNumber(), 6);
+    int bchCode = calculateBCHCode(version.getVersionNumber(), VERSION_INFO_POLY);
     bits.appendBits(bchCode, 12);
 
     if (bits.getSize() != 18) {  // Just in case.
