diff --git a/old_test.java b/new_test.java
index ca84e4c..386da57 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,6 @@
   public void testMakeVersionInfoBits() throws WriterException {
     // From Appendix D in JISX0510:2004 (p 68)
     BitArray bits = new BitArray();
-    MatrixUtil.makeVersionInfoBits(7, bits);
+    MatrixUtil.makeVersionInfoBits(Version.getVersionForNumber(7), bits);
     assertEquals(" ...XXXXX ..X..X.X ..", bits.toString());
   }
\ No newline at end of file
