diff --git a/old_test.java b/new_test.java
index cfe7b0d..f9bbbcf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,6 @@
 		OrderType orderType = orderService.getOrderType(1);
 		assertNotNull(orderType);
 		expectedException.expect(APIException.class);
-		expectedException.expectMessage("This order type cannot be deleted because it is already in use");
+		expectedException.expectMessage("Order.type.cannot.delete");
 		orderService.purgeOrderType(orderType);
 	}
\ No newline at end of file
