diff --git a/old_product.java b/new_product.java
index dc3fa9f..667ae3d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @Override
 	public void purgeOrderType(OrderType orderType) {
 		if (dao.isOrderTypeInUse(orderType)) {
-			throw new APIException("This order type cannot be deleted because it is already in use");
+			throw new APIException("Order.type.cannot.delete", (Object[]) null);
 		}
 		dao.purgeOrderType(orderType);
 	}
\ No newline at end of file
