diff --git a/old_product.java b/new_product.java
index 38a0c2f..f36fbc4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -27,8 +27,10 @@ public static List<Pkcs1Vector> generatePlainPkcs1Vectors(int publicKeyBitLength
                 publicKeyByteLength, keyBytes, 0)));
         pkcs1Vectors.add(new Pkcs1Vector("0x00 on the next to last position (|PMS| = 1)", getEK_SymmetricKeyOfSize(
                 publicKeyByteLength, keyBytes, 1)));
-        pkcs1Vectors.add(new Pkcs1Vector("0x00 on the 9th position from the rigth (|PMS| = 8)",
-                getEK_SymmetricKeyOfSize(publicKeyByteLength, keyBytes, 8)));
+        pkcs1Vectors.add(new Pkcs1Vector("Correctly formatted PKCS#1 message, (|PMS| = 47)", getPaddedKey(
+                publicKeyByteLength, Arrays.copyOf(keyBytes, HandshakeByteLength.PREMASTER_SECRET - 1))));
+        pkcs1Vectors.add(new Pkcs1Vector("Correctly formatted PKCS#1 message, (|PMS| = 49)", getPaddedKey(
+                publicKeyByteLength, Arrays.copyOf(keyBytes, HandshakeByteLength.PREMASTER_SECRET + 1))));
 
         if (type == BleichenbacherCommandConfig.Type.FULL) {
             List<Pkcs1Vector> additionalVectors = getEK_DifferentPositionsOf0x00(publicKeyByteLength, keyBytes);
