diff --git a/old_test.java b/new_test.java
index 13b3065..b0c4230 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,5 +3,5 @@
         List<Pkcs1Vector> vectors = Pkcs1VectorGenerator.generatePlainPkcs1Vectors(2048,
                 BleichenbacherCommandConfig.Type.FAST, ProtocolVersion.TLS12);
         Assert.assertNotNull(vectors);
-        Assert.assertEquals("11 PKCS#1 vectors should be generated", 11, vectors.size());
+        Assert.assertEquals("11 PKCS#1 vectors should be generated", 12, vectors.size());
     }
\ No newline at end of file
