diff --git a/old_product.java b/new_product.java
index 538c14e..121978b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @Override
     public YearMonth plus(long period, PeriodUnit unit) {
-        if (unit instanceof LocalDateUnit) {
-            switch ((LocalDateUnit) unit) {
+        if (unit instanceof LocalDateTimeUnit) {
+            switch ((LocalDateTimeUnit) unit) {
                 case MONTHS: return plusMonths(period);
                 case QUARTER_YEARS: return plusYears(period / 256).plusMonths((period % 256) * 3);  // no overflow (256 is multiple of 4)
                 case HALF_YEARS: return plusYears(period / 256).plusMonths((period % 256) * 6);  // no overflow (256 is multiple of 2)
@@ -11,8 +11,6 @@
                 case MILLENIA: return plusYears(DateTimes.safeMultiply(period, 1000));
             }
             throw new CalendricalException(unit.getName() + " not valid for YearMonth");
-        } else if (unit instanceof LocalTimeUnit) {
-            throw new CalendricalException(unit.getName() + " not valid for YearMonth");
         }
         return unit.add(this, period);
     }
\ No newline at end of file
