diff --git a/old_product.java b/new_product.java
index 76ef42e..9d6bab2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,4 +2,6 @@ CompletableFuture<RegistrationResponse> registerTaskExecutor(
 		String taskExecutorAddress,
 		ResourceID resourceId,
 		SlotReport slotReport,
+		int dataPort,
+		HardwareDescription hardwareDescription,
 		@RpcTimeout Time timeout);
\ No newline at end of file
