diff --git a/old_product.java b/new_product.java
index e0a1002..aa5d02e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -18,5 +18,6 @@ public void sendLsa(LsaHeader lsa, Ip4Address destination, Channel ch) {
 
         //setting the destination.
         responseLsUpdate.setDestinationIp(destination);
-        ch.write(responseLsUpdate);
+        byte[] messageToWrite = getMessage(responseLsUpdate);
+        ch.write(messageToWrite);
     }
\ No newline at end of file
