diff --git a/old_test.java b/new_test.java
index cb1e453..3ba9cf4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-	public void validate_shouldFailIfDurationUnitsHasNoMappingToISO8601Source() throws Exception {
+	public void validate_shouldFailIfDurationUnitsHasNoMappingToSNOMEDCTSource() throws Exception {
 		Patient patient = Context.getPatientService().getPatient(7);
 		CareSetting careSetting = Context.getOrderService().getCareSetting(2);
 		OrderType orderType = Context.getOrderService().getOrderTypeByName("Drug order");
@@ -22,6 +22,6 @@
 		order.setDurationUnits(cs.getConcept(28));
 		Errors errors = new BindException(order, "order");
 		new DrugOrderValidator().validate(order, errors);
-		assertEquals("DrugOrder.error.durationUnitsNotMappedToISO8601DurationCode", errors.getFieldError("durationUnits")
+		assertEquals("DrugOrder.error.durationUnitsNotMappedToSnomedCtDurationCode", errors.getFieldError("durationUnits")
 		        .getCode());
 	}
\ No newline at end of file
