diff --git a/old_product.java b/new_product.java
index ee323ab..c67c139 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
 Flux<String> getPartitionIds() {
-        return getProperties().flatMapMany(properties -> Flux.fromArray(properties.getPartitionIds()));
+        return getProperties().flatMapMany(properties -> Flux.fromIterable(properties.getPartitionIds()));
     }
\ No newline at end of file
