diff --git a/old_test.java b/new_test.java
index 0482df6..31b3dd3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,6 @@
     public void getPartitionIds() {
         final EventHubConsumerClient consumer = createBuilder()
             .consumerGroup(DEFAULT_CONSUMER_GROUP_NAME)
-            .startingPosition(EventPosition.earliest())
             .buildConsumer();
 
         // Act & Assert
