diff --git a/old_product.java b/new_product.java
index 62bb0d6..7dc5954 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,10 +9,14 @@ public InodeTree.CreatePathResult createDirectory(AlluxioURI path, CreateDirecto
         mMountTable.checkUnderWritableMountPoint(path);
       }
       InodeTree.CreatePathResult createResult = createDirectoryInternal(path, options);
-      writeJournalEntry(mDirectoryIdGenerator.toJournalEntry());
-      journalCreatePathResult(createResult);
-      flushJournal();
-      MasterContext.getMasterSource().incDirectoriesCreated(1);
-      return createResult;
+      try {
+        writeJournalEntry(mDirectoryIdGenerator.toJournalEntry());
+        journalCreatePathResult(createResult);
+        flushJournal();
+        MasterContext.getMasterSource().incDirectoriesCreated(1);
+        return createResult;
+      } finally {
+        createResult.unlock();
+      }
     }
   }
\ No newline at end of file
