diff --git a/old_test.java b/new_test.java
index feb9335..80a4aef 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
-    public void testAllStoredEventsBetween() {
-        System.out.println("allStoredEventsBetween");
+    public void testFindStoredEventsBetween() {
+        System.out.println("findStoredEventsBetween");
         Date occurredFrom = null;
         Date occurredTo = null;
         HibernateEventStore instance = new HibernateEventStore();
         List<StoredEvent> expResult = null;
-        List<StoredEvent> result = instance.allStoredEventsBetween(occurredFrom, occurredTo);
+        List<StoredEvent> result = instance.findStoredEventsBetween(occurredFrom, occurredTo);
         assertEquals(expResult, result);
         // TODO review the generated test code and remove the default call to fail.
         fail("The test case is a prototype.");
