diff --git a/old_product.java b/new_product.java
index fbc390f..edce736 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,28 +1,30 @@
-public synchronized void restart(String prefix) {
+public synchronized void restart(String prefix, File certificateFile, File keyFile) {
 	if (!this.isFree()) {
 	    if (p != null) {
-		p.destroy();
+		stop();
 	    }
 	    try {
 		id = LogFileIDManager.getInstance().getID();
 		String command = (prefix + restartServerCommand).replace("[id]", "" + id);
 		command = command.replace("[output]", traces.getAbsolutePath());
 		command = command.replace("[port]", "" + port);
-		// System.out.println(command);
+		command = command.replace("[cert]", "" + certificateFile.getAbsolutePath());
+		command = command.replace("[key]", "" + keyFile.getAbsolutePath());
+		LOG.log(Level.INFO, "Starting Server:" + command);
 		long time = System.currentTimeMillis();
 		Runtime rt = Runtime.getRuntime();
-		Process proc = rt.exec(command);
+		p = rt.exec(command);
 
 		// any error message?
-		errorGobbler = new StreamGobbler(proc.getErrorStream(), "ERR", accepted);
+		errorGobbler = new StreamGobbler(p.getErrorStream(), "ERR", accepted);
 
 		// any output?
-		outputGobbler = new StreamGobbler(proc.getInputStream(), "OUT", accepted);
+		outputGobbler = new StreamGobbler(p.getInputStream(), "OUT", accepted);
 
 		// kick them off
 		errorGobbler.start();
 		outputGobbler.start();
-		procmon = ProcMon.create(proc);
+		procmon = ProcMon.create(p);
 		while (!outputGobbler.accepted()) {
 
 		    try {
