diff --git a/old_product.java b/new_product.java
index 71aa8ba..28329ac 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,8 +10,6 @@ public void print(Calendrical calendrical, Appendable appendable) {
                 printerParser.print(calendrical, buf, symbols);
                 appendable.append(buf);
             }
-        } catch (UnsupportedRuleException ex) {
-            throw new CalendricalPrintFieldException(ex);
         } catch (IOException ex) {
             throw new CalendricalPrintException(ex.getMessage(), ex);
         }
