diff --git a/old_test.java b/new_test.java
index 10481f0..00beb17 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
-@Test(expectedExceptions=CalendricalPrintException.class)
+@Test(expectedExceptions=UnsupportedRuleException.class)
     public void test_print_emptyCalendrical() throws Exception {
         ZoneOffsetPrinterParser pp = new ZoneOffsetPrinterParser("Z", "+HH:MM:ss");
         pp.print(emptyCalendrical, buf, symbols);
