diff --git a/old_product.java b/new_product.java
index 5755798..0bb04a2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,5 @@ private static void insertPerformance(final IdPreMap m) {
       d[i][1] = id;
     }
     // perform the actual test:
-    final Performance p = new Performance();
     for(final int[] a : d) m.insert(a[0], a[1], 1);
-    if(VERBOSE) Util.errln(d.length + " records inserted in: " + p);
   }
\ No newline at end of file
