diff --git a/old_product.java b/new_product.java
index dc4032f..9fc7e7a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ public static String wrap(final String str, int wrapLength, String newLineStr, f
             return null;
         }
         if (newLineStr == null) {
-            newLineStr = SystemUtils.LINE_SEPARATOR;
+            newLineStr = System.lineSeparator();
         }
         if (wrapLength < 1) {
             wrapLength = 1;
