diff --git a/old_product.java b/new_product.java
index 3692e0f..724d3ee 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1 @@
-public <T> LearningEnvironmentBuilder withParallelismStrategy(ParallelismStrategy stgy) {
-        this.parallelismStgy = stgy;
-
-        return this;
-    }
\ No newline at end of file
+public LearningEnvironmentBuilder withParallelismStrategyDependency(IgniteFunction<Integer, ParallelismStrategy> stgy);
\ No newline at end of file
