diff --git a/old_product.java b/new_product.java
index f8c9869..6c288c1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,24 +1,9 @@
-private void handleGetReplicasForBlobId(MessageInfo messageInfo, AdminExecutionData executionData)
-      throws RestServiceException {
-    logger.trace("Handling getReplicas");
+private AdminOperationType getOperationType(RestRequestMetadata restRequestMetadata) {
+    String path = restRequestMetadata.getPath();
+    path = path.startsWith("/") ? path.substring(1, path.length()) : path;
     try {
-      RestResponseHandler responseHandler = messageInfo.getResponseHandler();
-      if (messageInfo.getRestObject() instanceof RestRequest) {
-        //TODO: Reconsider this model of execution
-        TaskExecutor executor = new GetReplicasForBlobIdExecutor(clusterMap);
-        String replicaStr = executor.execute(executionData).toString();
-        if (replicaStr != null) {
-          responseHandler.setContentType("application/json");
-          responseHandler.finalizeResponse();
-          responseHandler.addToBodyAndFlush(replicaStr.getBytes(), true);
-        } else {
-          throw new RestServiceException("Did not get a result for the GetReplicasForBlobId operation",
-              RestServiceErrorCode.ResponseBuildingFailure);
-        }
-      } else {
-        responseHandler.close();
-      }
-    } finally {
-      messageInfo.getRestObject().release();
+      return AdminOperationType.valueOf(path);
+    } catch (IllegalArgumentException e) {
+      return AdminOperationType.unknown;
     }
   }
\ No newline at end of file
