diff --git a/old_test.java b/new_test.java
index 4ec34fd..f5fe231 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,17 @@
 @Test
   public void handleGetReplicasForBlobIdTest()
-      throws InstantiationException, IOException, JSONException, RestServiceException, URISyntaxException {
-    ClusterMap clusterMap = new MockClusterMap();
-    AdminBlobStorageService adminBlobStorageService = getAdminBlobStorageService(clusterMap);
+      throws InstantiationException, JSONException, RestServiceException, URISyntaxException {
     List<PartitionId> partitionIds = clusterMap.getWritablePartitionIds();
     for (PartitionId partitionId : partitionIds) {
-      createBlobIdAndTest(partitionId, adminBlobStorageService);
+      String originalReplicaStr = partitionId.getReplicaIds().toString().replace(", ", ",");
+      BlobId blobId = new BlobId(partitionId);
+      RestRequestInfo restRequestInfo = createGetReplicasForBlobIdRestRequestInfo(blobId.getID());
+      adminBlobStorageService.handleGet(restRequestInfo);
+      finishGetRequest(adminBlobStorageService, restRequestInfo);
+      String returnedReplicasStr =
+          getJsonizedResponseBody(restRequestInfo).getString(GetReplicasForBlobIdHandler.REPLICAS_KEY)
+              .replace("\"", "");
+      assertEquals("Replica IDs returned for the BlobId do no match with the replicas IDs of partition",
+          originalReplicaStr, returnedReplicasStr);
     }
   }
\ No newline at end of file
