diff --git a/old_test.java b/new_test.java
index 3ecdf5f..8ea86e3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-	public void setGroupMembers_shouldMarkTheObsAsDirtyWhenTheSetIsChangedFromNullToANonEmptyOne() throws Exception {
-		Obs obs = new Obs();
+	public void setGroupMembers_shouldMarkTheExistingObsAsDirtyWhenTheSetIsChangedFromNullToANonEmptyOne() throws Exception {
+		Obs obs = new Obs(5);
 		assertNull(Obs.class.getDeclaredField("groupMembers").get(obs));
 		Set members = new HashSet<>();
 		members.add(new Obs());
