diff --git a/old_test.java b/new_test.java
index 3ae1ebc..b512a62 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
     public void testCreateProfileSchema() throws Exception {
-        ProfileSchemaDto profileSchema = createProfileSchema();
+        EndpointProfileSchemaDto profileSchema = createProfileSchema();
         Assert.assertFalse(strIsEmpty(profileSchema.getId()));
     }
\ No newline at end of file
