diff --git a/old_product.java b/new_product.java
index c3aec35..ae83d9f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,5 +10,5 @@ public void fireEvent(Object event, Annotation... bindings)
       // parameterized, this
       // method is not, so we have to use Observer<Object> for observers.
       Set<Observer<Object>> observers = this.resolveObservers(event, bindings);
-      this.eventBus.notifyObservers(observers, event);
+      this.eventManager.notifyObservers(observers, event);
    }
\ No newline at end of file
