diff --git a/old_product.java b/new_product.java
index 58b2fa3..bd7579f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,14 +1,14 @@
-public boolean deleteFile(AlluxioURI path, boolean recursive)
+public boolean delete(AlluxioURI path, boolean recursive)
       throws IOException, FileDoesNotExistException, DirectoryNotEmptyException,
           InvalidPathException, AccessControlException {
     MasterContext.getMasterSource().incDeletePathOps(1);
     synchronized (mInodeTree) {
-      checkPermission(FileSystemAction.WRITE, path, true);
+      mPermissionChecker.checkParentPermission(FileSystemAction.WRITE, path);
       mMountTable.checkUnderWritableMountPoint(path);
       Inode<?> inode = mInodeTree.getInodeByPath(path);
       long fileId = inode.getId();
       long opTimeMs = System.currentTimeMillis();
-      boolean ret = deleteFileInternal(fileId, recursive, false, opTimeMs);
+      boolean ret = deleteInternal(fileId, recursive, false, opTimeMs);
       DeleteFileEntry deleteFile = DeleteFileEntry.newBuilder()
           .setId(fileId)
           .setRecursive(recursive)
