diff --git a/old_test.java b/new_test.java
index d54ea14..c55bc62 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test
   public void deleteFileTest() throws Exception {
     // cannot delete root
-    Assert.assertFalse(mFileSystemMaster.deleteFile(ROOT_URI, true));
+    Assert.assertFalse(mFileSystemMaster.delete(ROOT_URI, true));
 
     // delete the file
     long blockId = createFileWithSingleBlock(NESTED_FILE_URI);
     Assert.assertTrue(
-        mFileSystemMaster.deleteFile(NESTED_FILE_URI, false));
+        mFileSystemMaster.delete(NESTED_FILE_URI, false));
 
     mThrown.expect(BlockInfoException.class);
     mBlockMaster.getBlockInfo(blockId);
