diff --git a/old_test.java b/new_test.java
index 4cb5270..4805cea 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
         final OffsetTime sample = OffsetTime.of(23, 5, OFFSET_PONE);
         WithAdjuster adjuster = new WithAdjuster() {
             @Override
-            public DateTime doAdjustment(DateTime calendrical) {
+            public DateTime doWithAdjustment(DateTime calendrical) {
                 return sample;
             }
         };
