diff --git a/old_product.java b/new_product.java
index c98456f..32efa55 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,5 +10,8 @@ public static CipherType getCipherType(CipherSuite cipherSuite) {
                 || cipher.contains("CHACHA")) {
             return CipherType.STREAM;
         }
+        if (cipherSuite == CipherSuite.TLS_FALLBACK_SCSV || cipherSuite == CipherSuite.TLS_EMPTY_RENEGOTIATION_INFO_SCSV || cipherSuite == CipherSuite.TLS_UNKNOWN_CIPHER) {
+            throw new IllegalArgumentException("The CipherSuite:" + cipherSuite.name() + " does not specify a CipherType");
+        }
         throw new UnsupportedOperationException("Cipher suite " + cipherSuite + " is not supported yet.");
     }
\ No newline at end of file
