diff --git a/old_product.java b/new_product.java
index 4d4b9c7..ed1a1d0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,5 +4,5 @@
    @Produces(VCloudDirectorMediaType.CATALOG_ITEM)
    @JAXBResponseParser
    @ExceptionParser(ThrowVCloudErrorOn4xx.class)
-   ListenableFuture<CatalogItem> addCatalogItem(@EndpointParam(parser = URISupplierToEndpoint.class) URISupplier catalogRef,
+   ListenableFuture<CatalogItem> addCatalogItem(@EndpointParam URI catalogUri,
          @BinderParam(BindToXMLPayload.class) CatalogItem catalogItem);
\ No newline at end of file
