diff --git a/old_test.java b/new_test.java
index 493dcaa..6e57720 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,8 @@
 @Test
-	public void encodeFormatMessage() {
+	public void encodeFormatBits() {
 
-		int found = QrCodePolynomialMath.encodeFormatMessage(QrCode.ErrorCorrectionLevel.M,0b101);
+		int found = QrCodePolynomialMath.encodeFormatBits(QrCode.ErrorCorrectionLevel.M,0b101);
+		found ^= QrCodePolynomialMath.FORMAT_MASK;
 		int expected = 0b100000011001110;
 
 		assertEquals(expected,found);
