diff --git a/old_test.java b/new_test.java
index 1e1b527..fcdf799 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,9 +4,12 @@
     Properties properties = new Properties();
     VerifiableProperties verifiableProperties = new VerifiableProperties(properties);
     RestRequestHandler restRequestHandler = new MockRestRequestResponseHandler();
+    PublicAccessLogger publicAccessLogger = new PublicAccessLogger(new String[]{}, new String[]{});
+    VIPHealthCheckService vipHealthCheckService = new VIPHealthCheckService("/ambry-frontend/admin");
 
     NettyServerFactory nettyServerFactory =
-        new NettyServerFactory(verifiableProperties, new MetricRegistry(), restRequestHandler);
+        new NettyServerFactory(verifiableProperties, new MetricRegistry(), restRequestHandler, publicAccessLogger,
+            vipHealthCheckService);
     NioServer nioServer = nettyServerFactory.getNioServer();
     assertNotNull("No NioServer returned", nioServer);
     assertEquals("Did not receive a NettyServer instance", NettyServer.class.getCanonicalName(),
