diff --git a/old_product.java b/new_product.java
index fa65e37..a3aa971 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,5 +8,7 @@ private NettyServer getNettyServer(Properties properties)
     NettyConfig nettyConfig = new NettyConfig(verifiableProperties);
     NettyMetrics nettyMetrics = new NettyMetrics(new MetricRegistry());
     RestRequestHandler requestHandler = new MockRestRequestResponseHandler();
-    return new NettyServer(nettyConfig, nettyMetrics, requestHandler);
+    PublicAccessLogger publicAccessLogger = new PublicAccessLogger(new String[]{}, new String[]{});
+    VIPHealthCheckService vipHealthCheckService = new VIPHealthCheckService("/ambry-frontend/admin");
+    return new NettyServer(nettyConfig, nettyMetrics, requestHandler, publicAccessLogger, vipHealthCheckService);
   }
\ No newline at end of file
