diff --git a/old_test.java b/new_test.java
index 3090142..dd48346 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,13 +2,13 @@
   public void testConfigure() throws Exception {
     OffHeapIdentifierRegistry registry = new OffHeapIdentifierRegistry(32, MemoryUnit.MEGABYTES);
 
-    final EhcacheActiveEntity activeEntity = new EhcacheActiveEntity(registry, ENTITY_ID);
+    final EhcacheActiveEntity activeEntity = new EhcacheActiveEntity(registry, ENTITY_ID, RESPONSE_FACTORY);
     ClientDescriptor client = new TestClientDescriptor();
     activeEntity.connected(client);
     assertThat(activeEntity.getConnectedClients().keySet(), contains(client));
 
     assertSuccess(activeEntity.invoke(client,
-        MESSAGE_FACTORY.configureCacheManager(new ServerSideConfigBuilder()
+        MESSAGE_FACTORY.configureStoreManager(new ServerSideConfigBuilder()
             .defaultResource("defaultServerResource")
             .sharedPool("primary", "serverResource1", 4, MemoryUnit.MEGABYTES)
             .sharedPool("secondary", "serverResource2", 8, MemoryUnit.MEGABYTES)
