diff --git a/old_product.java b/new_product.java
index 39e49ba..e1c3085 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-private EhcacheEntityResponse configure(ClientDescriptor clientDescriptor, ConfigureCacheManager message) {
+private EhcacheEntityResponse configure(ClientDescriptor clientDescriptor, ConfigureStoreManager message) {
     ClientState clientState = this.clientStateMap.get(clientDescriptor);
     if (clientState == null) {
       return responseFactory.failure(new IllegalStateException("Client " + clientDescriptor + " is not connected to the Clustered Store Manager"));
