diff --git a/old_product.java b/new_product.java
index 83940fb..9e90670 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
 private B64 encodeString(final QueryContext qc) throws QueryException {
     final byte[] str = str(0, qc);
-    final String enc = checkEncoding(1, BIN_UE_X, qc);
+    final String enc = toEncoding(1, BIN_UE_X, qc);
     if(str == null) return null;
     try {
       return new B64(enc == null || enc == UTF8 ? str : FNConvert.toBinary(str, enc));
     } catch(final CharacterCodingException ex) {
-      throw BIN_CE.get(info, ex);
+      throw BIN_CE_X.get(info, ex);
     }
   }
\ No newline at end of file
