diff --git a/old_product.java b/new_product.java
index 9d58f66..989a305 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,9 @@
             @NotEmpty(message = "No configs entered. Unable to update.")
             final Set<String> configs
     ) throws GenieException {
-        LOG.debug("called with id " + id + " and configs " + configs);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("called with id " + id + " and configs " + configs);
+        }
         final Cluster cluster = this.clusterRepo.findOne(id);
         if (cluster != null) {
             cluster.setConfigs(configs);
