diff --git a/old_test.java b/new_test.java
index 0d59fc2..80da829 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,18 +1,12 @@
 @Test
     public void testGetConnectionString() throws Exception {
-        String value = Settings.getConnectionString(Settings.KEYS.DB_CONNECTION_STRING, Settings.KEYS.DB_FILE_NAME, Settings.KEYS.DB_VERSION);
+        String value = Settings.getConnectionString(Settings.KEYS.DB_CONNECTION_STRING, Settings.KEYS.DB_FILE_NAME);
         Assert.assertNotNull(value);
         String msg = null;
         try {
-            value = Settings.getConnectionString(Settings.KEYS.DB_CONNECTION_STRING, Settings.KEYS.DB_FILE_NAME, null);
+            value = Settings.getConnectionString("invalidKey", null);
         } catch (InvalidSettingException e) {
             msg = e.getMessage();
         }
-        Assert.assertNotNull(msg, msg);
-        try {
-            value = Settings.getConnectionString("invalidKey", null, null);
-        } catch (InvalidSettingException e) {
-            msg = e.getMessage();
-        }
-        Assert.assertNotNull(msg, msg);
+        Assert.assertNotNull(msg);
     }
\ No newline at end of file
