diff --git a/old_product.java b/new_product.java
index 5b1b807..69f3bef 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
-public static String getConnectionString(String connectionStringKey, String dbFileNameKey, String dbVersionKey)
+public static String getConnectionString(String connectionStringKey, String dbFileNameKey)
             throws IOException, InvalidSettingException {
         final String connStr = Settings.getString(connectionStringKey);
         if (connStr == null) {
-            final String msg = String.format("Invalid properties file to get the connection string; '%s' must be defined.",
+            final String msg = String.format("Invalid properties file; data.connection_string is missing.",
                     connectionStringKey);
             throw new InvalidSettingException(msg);
         }
@@ -17,18 +17,6 @@ public static String getConnectionString(String connectionStringKey, String dbFi
                         dbFileNameKey);
                 throw new InvalidSettingException(msg);
             }
-            if (fileName.contains("%s")) {
-                String version = null;
-                if (dbVersionKey != null) {
-                    version = Settings.getString(dbVersionKey);
-                }
-                if (version == null) {
-                    final String msg = String.format("Invalid properties file to get a file based connection string; '%s' must be defined.",
-                            dbFileNameKey);
-                    throw new InvalidSettingException(msg);
-                }
-                fileName = String.format(fileName, version);
-            }
             if (connStr.startsWith("jdbc:h2:file:") && fileName.endsWith(".h2.db")) {
                 fileName = fileName.substring(0, fileName.length() - 6);
             }
