diff --git a/old_product.java b/new_product.java
index 280e38a..8181e7c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,7 +10,7 @@
         // Execute script
         final DataSet dataSet = createShellTask(request);
         final StructType schema = dataSet.schema();
-        TransformResponse response = submitTransformJob(new ShellTransformStage(dataSet, converterService), getPolicies(request), request.getPageSpec());
+        TransformResponse response = submitTransformJob(new ShellTransformStage(dataSet, converterService),request);
 
         // Build response
         if (response.getStatus() != TransformResponse.Status.SUCCESS) {
