diff --git a/old_test.java b/new_test.java
index acac17b..21a8bcc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,6 +16,7 @@
 
         // Test executing a request
         final TransformRequest request = new TransformRequest();
+        request.setDoProfile(true);
         request.setScript("sqlContext.range(1,10)");
 
         final TransformService service = new TransformService(TransformScript.class, engine, sparkContextService, new MockJobTrackerService(), Mockito.mock(DataSetConverterService.class));
