diff --git a/old_product.java b/new_product.java
index 2a172ac..d67b1d3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public static Evaluator getDateFormatEvaluator() {
     return new Evaluator() {
-      public String evaluate(VariableResolver resolver, String expression) {
+      public String evaluate(String expression, Context context) {
         CacheEntry e = getCachedData(expression);
         String expr = e.key;
         SimpleDateFormat fmt = e.format;
@@ -16,7 +16,7 @@ public static Evaluator getDateFormatEvaluator() {
                     "Invalid expression for date", exp);
           }
         } else {
-          Object o = resolver.resolve(expr);
+          Object o = context.getVariableResolver().resolve(expr);
           if (o == null)
             return "";
           Date date = null;
