diff --git a/old_test.java b/new_test.java
index ea75635..b367a05 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,13 +2,15 @@
   @Ignore
   public void testGetDateFormatEvaluator() {
     Evaluator dateFormatEval = EvaluatorBag.getDateFormatEvaluator();
+    resolver.context = new ContextImpl(null, resolver, null, 0, Collections.EMPTY_MAP, null, null);
+
     assertEquals(new SimpleDateFormat("yyyy-MM-dd").format(new Date()),
-            dateFormatEval.evaluate(resolver, "'NOW',yyyy-MM-dd HH:mm"));
+            dateFormatEval.evaluate("'NOW',yyyy-MM-dd HH:mm", resolver.context));
 
     Map<String, Object> map = new HashMap<String, Object>();
     map.put("key", new Date());
     resolver.addNamespace("A", map);
 
     assertEquals(new SimpleDateFormat("yyyy-MM-dd HH:mm").format(new Date()),
-            dateFormatEval.evaluate(resolver, "A.key, yyyy-MM-dd HH:mm"));
+            dateFormatEval.evaluate("A.key, yyyy-MM-dd HH:mm", resolver.context));
   }
\ No newline at end of file
