diff --git a/old_test.java b/new_test.java
index b4bada1..aa2239a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,9 +2,9 @@
     public void testGetExtensionByAlias() throws Exception {
        for (String zoneId : zones) {
            ExtensionApi api = novaContext.getApi().getExtensionApiForZone(zoneId);
-           Set<? extends Extension> response = api.listExtensions();
+           Set<? extends Extension> response = api.list();
            for (Extension extension : response) {
-              Extension details = api.getExtensionByAlias(extension.getId());
+              Extension details = api.get(extension.getId());
               assertNotNull(details);
               assertEquals(details.getId(), extension.getId());
               assertEquals(details.getName(), extension.getName());
