diff --git a/old_test.java b/new_test.java
index 49013fb..e919ea1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,14 +7,9 @@
 				EWAHCompressedBitmap32[] x = i.next();
 				EWAHCompressedBitmap32 tanswer = EWAHCompressedBitmap32.or(x);
 				EWAHCompressedBitmap32 x1 = IteratorUtil32
-						.materialize(IteratorAggregation32.or(IteratorUtil32
+						.materialize(IteratorAggregation32.bufferedor(IteratorUtil32
 								.toIterators(x)));
-				//EWAHCompressedBitmap x2 = IteratorUtil
-					//	.materialize(IteratorAggregation
-						//		.bufferedor(IteratorUtil.toIterators(x)));
 				assertTrue(x1.equals(tanswer));
-			//	assertTrue(x2.equals(tanswer));
-				//assertTrue(x1.equals(x2));
 			}
 			System.gc();
 		}
