diff --git a/old_product.java b/new_product.java
index 38cf364..2f704be 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,7 +7,7 @@ public static double root(Function func, double x1, double x2, double tol, int m
             throw new IllegalArgumentException("Invalid maximum number of iterations: " + maxIter);            
         }
         
-        double a = x1, b = x2, c = x2, d = 0, e = 0, fa = func.f(a), fb = func.f(b), fc, p, q, r, s, xm;
+        double a = x1, b = x2, c = x2, d = 0, e = 0, fa = func.apply(a), fb = func.apply(b), fc, p, q, r, s, xm;
         if ((fa > 0.0 && fb > 0.0) || (fa < 0.0 && fb < 0.0)) {
             throw new IllegalArgumentException("Root must be bracketed.");
         }
@@ -79,7 +79,7 @@ public static double root(Function func, double x1, double x2, double tol, int m
             } else {
                 b += Math.copySign(tol, xm);
             }
-            fb = func.f(b);
+            fb = func.apply(b);
         }
 
         logger.error("Brent's method exceeded the maximum number of iterations.");
