diff --git a/old_test.java b/new_test.java
index 000191b..1ece1e9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test
-    @Ignore
-    public void testGetFilePostfixName(){
-        assertEquals("png", FileUtil.getFilePostfixName(fileName1));
+    public void testGetExtension(){
+        assertEquals("png", FileUtil.getExtension(fileName1));
         LOGGER.info(fileName1.substring(fileName1.lastIndexOf(".")));
         LOGGER.info(fileName1.substring(fileName1.lastIndexOf("\\") + 1));
     }
\ No newline at end of file
