diff --git a/old_product.java b/new_product.java
index 4842da6..9ac50ed 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,17 +10,29 @@ void putGridIntoCanonical(Grid g ) {
 
 		// select the best corner for canonical
 		if( g.columns%2 == 1 && g.rows%2 == 1) {
-			int numRotationsCCW = closestCorner4(g);
+			// first make sure orientation constraint is maintained
+			if( isClockWise(g)) {
+				flipHorizontal(g);
+			}
 
-			for (int i = 0; i < numRotationsCCW; i++) {
+			int numRotationsCCW = closestCorner4(g);
+			if( g.columns == g.rows ) {
+				for (int i = 0; i < numRotationsCCW; i++) {
+					rotateGridCCW(g);
+				}
+			} else if( numRotationsCCW == 2 ){
+				// only two valid solutions.  rotate only if the other valid solution is better
+				rotateGridCCW(g);
 				rotateGridCCW(g);
 			}
 		} else if( g.columns%2 == 1 ) {
-			if( g.get(0,0).center.normSq() > g.get(0,g.columns-1).center.normSq() ) {
+			// only two solutions.  Go with the one which maintains orientation constraint
+			if( isClockWise(g)) {
 				flipHorizontal(g);
 			}
 		} else if( g.rows%2 == 1 ) {
-			if( g.get(0,0).center.normSq() > g.get(g.rows-1,0).center.normSq() ) {
+			// only two solutions.  Go with the one which maintains orientation constraint
+			if( isClockWise(g)) {
 				flipVertical(g);
 			}
 		}
