diff --git a/old_test.java b/new_test.java
index a350e5f..303a0eb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testParseApiRequestQueryParams() {
-        Map<String, String> paramMap = GatewayServlet.parseApiRequestQueryParams(null);
+        QueryMap paramMap = GatewayServlet.parseApiRequestQueryParams(null);
         Assert.assertNotNull(paramMap);
 
         paramMap = GatewayServlet.parseApiRequestQueryParams("param1");
