diff --git a/old_product.java b/new_product.java
index 3996599..b61a150 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-protected static final Map<String, String> parseApiRequestQueryParams(String queryString) {
-        Map<String, String> rval = new LinkedHashMap<>();
+protected static final QueryMap parseApiRequestQueryParams(String queryString) {
+        QueryMap rval = new QueryMap();
         if (queryString != null) {
             try {
                 queryString = URLDecoder.decode(queryString, "UTF-8"); //$NON-NLS-1$
@@ -12,9 +12,9 @@ protected static final Map<String, String> parseApiRequestQueryParams(String que
                 if (idx != -1) {
                     String key = paramPair.substring(0, idx);
                     String val = paramPair.substring(idx + 1);
-                    rval.put(key, val);
+                    rval.add(key, val);
                 } else {
-                    rval.put(paramPair, null);
+                    rval.add(paramPair, null);
                 }
             }
         }
