diff --git a/old_product.java b/new_product.java
index b9d131d..cb9d666 100644
--- a/old_product.java
+++ b/new_product.java
@@ -56,20 +56,17 @@
 
             final String version = xpath.evaluate("/assembly/version", doc);
             if (version != null) {
-                dependency.getVersionEvidence().addEvidence(new Evidence("grokassembly", "version",
-                        version, Confidence.HIGHEST));
+                dependency.addEvidence(EvidenceType.VERSION, "grokassembly", "version", version, Confidence.HIGHEST);
             }
 
             final String vendor = xpath.evaluate("/assembly/company", doc);
             if (vendor != null) {
-                dependency.getVendorEvidence().addEvidence(new Evidence("grokassembly", "vendor",
-                        vendor, Confidence.HIGH));
+                dependency.addEvidence(EvidenceType.VENDOR, "grokassembly", "vendor", vendor, Confidence.HIGH);
             }
 
             final String product = xpath.evaluate("/assembly/product", doc);
             if (product != null) {
-                dependency.getProductEvidence().addEvidence(new Evidence("grokassembly", "product",
-                        product, Confidence.HIGH));
+                dependency.addEvidence(EvidenceType.PRODUCT, "grokassembly", "product", product, Confidence.HIGH);
             }
 
         } catch (ParserConfigurationException pce) {
