diff --git a/old_test.java b/new_test.java
index d6a2d9e..a6f433f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void testDelete() {
-    final Request request = RequestBuilder.delete(urlWithQuery).build();
+    final Request request = RequestBuilder.delete(HttpUrl.parse(urlWithQuery)).build();
     assertEquals("DELETE", request.method());
     assertEquals(urlWithQuery, request.url().toString());
   }
\ No newline at end of file
