diff --git a/old_test.java b/new_test.java
index f04bf96..b431663 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	public final void testGetInputWithRandomValue() {
 		assertNull("Wrong Xpath so null as result of InputWithRandomValue",
-		        browser.getInputWithRandomValue(new FormInput("text", new Identification(
-		                How.xpath, "/RUBISH"), "abc")));
+				browser.getInputWithRandomValue(new FormInput(InputType.TEXT,
+						new Identification(How.xpath, "/RUBISH"), "abc")));
 	}
\ No newline at end of file
