diff --git a/old_product.java b/new_product.java
index bd9ccd1..76348fe 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,15 +5,14 @@ public Authentication authenticate(Authentication authentication) throws Authent
         }
 
         SAMLAuthenticationToken token = (SAMLAuthenticationToken) authentication;
-        SAMLMessageStorage store = token.getMessageStore();
         SAMLMessageContext context = token.getCredentials();
         SAMLCredential credential;
 
         try {
             if (SAMLConstants.SAML2_WEBSSO_PROFILE_URI.equals(context.getCommunicationProfileId())) {
-                credential = consumer.processAuthenticationResponse(context, store);
+                credential = consumer.processAuthenticationResponse(context);
             } else if (SAMLConstants.SAML2_HOK_WEBSSO_PROFILE_URI.equals(context.getCommunicationProfileId())) {
-                credential = hokConsumer.processAuthenticationResponse(context, store);
+                credential = hokConsumer.processAuthenticationResponse(context);
             } else {
                 throw new SAMLException("Unsupported profile encountered in the context " + context.getCommunicationProfileId());
             }
