diff --git a/old_test.java b/new_test.java
index bfe72ac..de315af 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,11 +3,10 @@
         SAMLMessageContext context = new SAMLMessageContext();
         context.setCommunicationProfileId(SAMLConstants.SAML2_WEBSSO_PROFILE_URI);
 
-        SAMLAuthenticationToken token = new SAMLAuthenticationToken(context, messageStorage);
-        SAMLMessageStorage store = token.getMessageStore();
+        SAMLAuthenticationToken token = new SAMLAuthenticationToken(context);
         SAMLCredential result = new SAMLCredential(nameID, assertion, "IDP", "testSP");
 
-        expect(consumer.processAuthenticationResponse(context, store)).andReturn(result);
+        expect(consumer.processAuthenticationResponse(context)).andReturn(result);
         expect(nameID.getValue()).andReturn("Name");
 
         DateTime expiry = new DateTime().plusHours(4);
