diff --git a/old_test.java b/new_test.java
index 9ea7788..1df55bd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
     public void testGetDBConnection() throws SQLException
     {
-        Connection connection = context.getDBConnection();
+        DBConnection connection = context.getDBConnection();
         
         assertThat("testGetDBConnection 0", connection, notNullValue());
-        assertThat("testGetDBConnection 1", connection.isClosed(), equalTo(false));
+        assertThat("testGetDBConnection 1", connection.isSessionAlive(), equalTo(true));
     }
\ No newline at end of file
