diff --git a/old_product.java b/new_product.java
index b5d9caa..5bb73f1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public static String getAbbreviatedName(final Class<?> cls, final int len) {
+public static String getAbbreviatedName(final Class<?> cls, final int lengthHint) {
       if (cls == null) {
         return StringUtils.EMPTY;
       }
-      return getAbbreviatedName(cls.getName(), len);
+      return getAbbreviatedName(cls.getName(), lengthHint);
     }
\ No newline at end of file
