diff --git a/old_product.java b/new_product.java
index 094acdd..b62ea91 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public static <C> Collection<C> predicatedCollection(Collection<C> collection, Predicate<? super C> predicate) {
+public static <C> Collection<C> predicatedCollection(final Collection<C> collection, final Predicate<? super C> predicate) {
         return PredicatedCollection.predicatedCollection(collection, predicate);
     }
\ No newline at end of file
