diff --git a/old_test.java b/new_test.java
index 73d11db..73a392f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testToMap(){
-        Map<String, String> map = toMap(//
+        Map<String, String> map = toMapUseEntrys(//
                         Pair.of("张飞", "丈八蛇矛"),
                         Pair.of("关羽", "青龙偃月刀"),
                         Pair.of("赵云", "龙胆枪"),
