diff --git a/old_product.java b/new_product.java
index 3eb8434..b6eaee9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,5 @@
-@SafeVarargs
-    public static <V, K> Map<K, V> toMap(Map.Entry<K, V>...mapEntrys){
-        if (null == mapEntrys){
-            return emptyMap();
-        }
-        Validate.noNullElements(mapEntrys, "mapEntrys can't has null elememt!");
-
-        Map<K, V> map = newLinkedHashMap(mapEntrys.length);
-        for (Map.Entry<K, V> entry : mapEntrys){
-            map.put(entry.getKey(), entry.getValue());
-        }
+public static <K, V> Map<K, V> toMap(K key1,V value1,K key2,V value2){
+        Map<K, V> map = toMap(key1, value1);
+        map.put(key2, value2);
         return map;
     }
\ No newline at end of file
