diff --git a/old_test.java b/new_test.java
index 2f47186..6ab1ca7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
-@Test public void arrayTest() throws QueryException {
+@Test public void arrayTest() throws QueryIOException {
     parse("[ ]", JsonSpec.RFC4627);
     parse("[]", "[ ]", JsonSpec.RFC4627);
     parse("[[[[[[42], {}]]]]]", "[ [ [ [ [ [ 42 ], { } ] ] ] ] ]", JsonSpec.RFC4627);
