diff --git a/old_test.java b/new_test.java
index 6969486..2feb666 100644
--- a/old_test.java
+++ b/new_test.java
@@ -21,6 +21,7 @@
                                                                         rebalancePartitionList,
                                                                         Arrays.asList(testStoreName),
                                                                         new HashMap<String, String>(),
+                                                                        new HashMap<String, String>(),
                                                                         0);
         int asyncId = adminClient.rebalanceNode(stealInfo);
         assertNotSame("Got a valid rebalanceAsyncId", -1, asyncId);
