diff --git a/old_test.java b/new_test.java
index 3a862f3..163d7a2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,12 +6,12 @@
         int pos = 3;
         Run run = new Run(4, 1);
         RunTable instance = createHorizontalInstance();
-        short[] seq1 = instance.getSequence(pos);
-        System.out.println("sequence before: " + Arrays.toString(seq1));
+        RunSequence seq1 = instance.getSequence(pos);
+        System.out.println("sequence before: " + seq1);
         System.out.println("table before:" + instance.dumpOf());
         instance.removeRun(pos, run);
 
-        short[] seq2 = instance.getSequence(pos);
-        System.out.println("sequence  after: " + Arrays.toString(seq2));
+        RunSequence seq2 = instance.getSequence(pos);
+        System.out.println("sequence  after: " + seq2);
         System.out.println("table after:" + instance.dumpOf());
     }
\ No newline at end of file
