diff --git a/old_test.java b/new_test.java
index d2b8902..e6ff08e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,9 @@
     public void testExtract() {
         logger.info("extract");
         String text = "In publishing and graphic design, lorem ipsum[1] is a placeholder text (filler text) commonly used to demonstrate the graphic elements of a document or visual presentation, such as font, typography, and layout, by removing the distraction of meaningful content. The lorem ipsum text is typically a section of a Latin text by Cicero with words altered, added, and removed that make it nonsensical and not proper Latin.[1] In publishing and graphic design, lorem ipsum[1] is a placeholder text (filler text) commonly used to demonstrate the graphic elements of a document or visual presentation, such as font, typography, and layout, by removing the distraction of meaningful content. The lorem ipsum text is typically a section of a Latin text by Cicero with words altered, added, and removed that make it nonsensical and not proper Latin.[1]";
-        NgramsExtractor instance = new NgramsExtractor(new NgramsExtractor.Parameters());
+        NgramsExtractor.Parameters p = new NgramsExtractor.Parameters();
+        p.setMaxDistanceBetweenKwds(0);
+        NgramsExtractor instance = new NgramsExtractor(p);
         
         Map<String, Double> expResult = new HashMap<>();
         
