diff --git a/old_product.java b/new_product.java
index d3dc567..bdfed26 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @Override
-    public AppUsagePerApiBean getAppUsagePerApi(String organizationId, String applicationId,
+    public ClientUsagePerApiBean getClientUsagePerApi(String organizationId, String clientId,
             String version, DateTime from, DateTime to) {
-        AppUsagePerApiBean rval = new AppUsagePerApiBean();
+        ClientUsagePerApiBean rval = new ClientUsagePerApiBean();
         return rval;
     }
\ No newline at end of file
