diff --git a/old_test.java b/new_test.java
index d66ff27..7a6c7be 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
     mockServer.when(request().withPath(GET_MODELS_PATH + "/" + speechModel.getName())).respond(
         response().withHeaders(
             new Header(HttpHeaders.Names.CONTENT_TYPE, HttpMediaType.APPLICATION_JSON)).withBody(
-            GsonSingleton.getGsonWithoutPrettyPrinting().toJson(speechModel)));
+            GSON.toJson(speechModel)));
 
 
     SpeechModel model = service.getModel("not-a-real-Model").execute();
