diff --git a/old_product.java b/new_product.java
index 41c93a0..210c636 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,7 @@
 @Override
 	@SuppressWarnings("unchecked")
 	public List<User> getAllUsers() throws DAOException {
-		return sessionFactory.getCurrentSession().createQuery("from User u order by u.userId").list();
+		return sessionFactory.getCurrentSession().createQuery("from User where not uuid = :daemonUserUuid order by userId")
+				                                                     .setString("daemonUserUuid", Daemon.getDaemonUserUuid()).list();
+		
 	}
\ No newline at end of file
