diff --git a/old_test.java b/new_test.java
index 9f1daa4..ddfeae7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
 	public void getAllUsers_shouldNotContainsAnyDuplicateUsers() {
 		executeDataSet(XML_FILENAME);
 		List<User> users = userService.getAllUsers();
-		Assert.assertEquals(12, users.size());
+		assertEquals(11, users.size());
 		// TODO Need to test with duplicate data in the dataset (not sure if that's possible)
 		
 	}
\ No newline at end of file
