diff --git a/old_product.java b/new_product.java
index 4d4f574..55bd7fd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static double Poisson(int k, double lamda) {
+public static double poisson(int k, double lamda) {
         if(k<0 || lamda<0) {
             throw new IllegalArgumentException("All the parameters must be positive.");
         }
@@ -10,9 +10,9 @@ public static double Poisson(int k, double lamda) {
 
         //fast and can handle large numbers
         //Cdf(k)-Cdf(k-1)
-        double probability = PoissonCdf(k,lamda);
+        double probability = poissonCdf(k,lamda);
         if(k>0) {
-            probability -= PoissonCdf(k-1,lamda);
+            probability -= poissonCdf(k-1,lamda);
         }
         
         return probability;
