diff --git a/old_test.java b/new_test.java
index 53ad793..82a648c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -34,7 +34,7 @@
             double[] trainy = Math.slice(y, loocv.train[i]);
             
             try {
-                RandomForest forest = new RandomForest(trainx, trainy, 300, 2, 3);
+                RandomForest forest = new RandomForest(trainx, trainy, 300, 2, 3, n);
 
                 double r = y[loocv.test[i]] - forest.predict(longley[loocv.test[i]]);
                 rss += r * r;
