diff --git a/old_product.java b/new_product.java
index d8b2686..0d1a1ab 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-Map<String, String> crosswalkMetadata(DSpaceObject dso)
+Map<String, String> crosswalkMetadata(Context context, DSpaceObject dso)
     {
         if ((null == dso) || !(dso instanceof Item))
         {
@@ -10,10 +10,10 @@ Map<String, String> crosswalkMetadata(DSpaceObject dso)
 
         for (Entry<String, String> datum : crosswalk.entrySet())
         {
-            Metadatum[] values = item.getMetadataByMetadataString(datum.getValue());
+            List<MetadataValue> values = itemService.getMetadataByMetadataString(item, datum.getValue());
             if (null != values)
             {
-                for (Metadatum value : values)
+                for (MetadataValue value : values)
                 {
                     String key = datum.getKey();
                     String mappedValue;
@@ -21,11 +21,11 @@ Map<String, String> crosswalkMetadata(DSpaceObject dso)
                     if (null != xfrm)
                     {
                         try {
-                            mappedValue = xfrm.transform(value.value);
+                            mappedValue = xfrm.transform(value.getValue());
                         } catch (Exception ex) {
                             log.error("Unable to transform '{}' from {} to {}:  {}",
                                     new String[] {
-                                        value.value,
+                                        value.getValue(),
                                         value.toString(),
                                         key,
                                         ex.getMessage()
@@ -35,7 +35,7 @@ Map<String, String> crosswalkMetadata(DSpaceObject dso)
                     }
                     else
                     {
-                        mappedValue = value.value;
+                        mappedValue = value.getValue();
                     }
                     mapped.put(key, mappedValue);
                 }
@@ -46,7 +46,7 @@ Map<String, String> crosswalkMetadata(DSpaceObject dso)
         {
             DataCiteXMLCreator xmlGen = new DataCiteXMLCreator();
             xmlGen.setDisseminationCrosswalkName(DATACITE_XML_CROSSWALK);
-            String xmlString = xmlGen.getXMLString(dso);
+            String xmlString = xmlGen.getXMLString(context, dso);
             log.debug("Generated DataCite XML:  {}", xmlString);
             mapped.put("datacite", xmlString);
         }
@@ -74,7 +74,7 @@ Map<String, String> crosswalkMetadata(DSpaceObject dso)
         if (null == handle)
         {
             log.warn("{} #{} has no handle -- location not set.",
-                    dso.getTypeText(), dso.getID());
+                    contentServiceFactory.getDSpaceObjectService(dso).getTypeText(dso), dso.getID());
         }
         else
         {
