diff --git a/old_test.java b/new_test.java
index a086b10..3a3df2d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(expected = SqoopException.class)
-  public void testNewConnection() {
-    when(requests.readConnector(null)).thenReturn(connectorBean(connector(1)));
-    client.newConnection("non existing connector");
+  public void testCreateLink() {
+    when(resourceRequests.readConnector(null)).thenReturn(connectorBean(connector(1)));
+    client.createLink("non existing connector");
   }
\ No newline at end of file
