diff --git a/old_product.java b/new_product.java
index 68147c0..c1b6205 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public MConnection newConnection(long cid) {
-    return new MConnection(
-      cid,
-      getConnector(cid).getConnectionForms(),
-      getFramework().getConnectionForms()
+public MLink createLink(long connectorId) {
+    return new MLink(
+      connectorId,
+      getConnector(connectorId).getConnectionForms(),
+      getDriverConfig().getConnectionForms()
     );
   }
\ No newline at end of file
