diff --git a/old_test.java b/new_test.java
index 2436b83..033b8af 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
 		OrderFrequency orderFrequency = orderService.getOrderFrequency(1);
 		assertNotNull(orderFrequency);
 		
-		expectedException.expect(APIException.class);
+		expectedException.expect(CannotDeleteOrderPropertyInUseException.class);
 		expectedException.expectMessage("Order.frequency.cannot.delete");
 		orderService.purgeOrderFrequency(orderFrequency);
 	}
\ No newline at end of file
