diff --git a/old_test.java b/new_test.java
index 94732a9..b8b42f1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,16 @@
 @Test(dependsOnMethods = "releaseTheSessionBlockingTheGrid")
 	public void validate() throws InterruptedException {
-		Thread.sleep(250);
+		int cpt=0;
+		while(cpt <10){
+			Thread.sleep(250);
+			cpt++;
+			try {
+				requests.get(0).getTestSession();
+			}catch (IllegalAccessError iae){
+				//ignore
+			}
+		}
+		System.out.println(">"+requests);
 		Assert.assertNotNull(requests.get(0).getTestSession());	
 		Assert.assertEquals(requests.get(0).getDesiredCapabilities().get("_priority"), 1);
 	}
\ No newline at end of file
