diff --git a/old_test.java b/new_test.java
index 2eaf345..bc290cc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,16 +4,14 @@
         BlockHeader header = new BlockHeader();
         NulsDigestData hash = NulsDigestData.calcDigestData("abcdefg".getBytes());
         header.setHash(hash);
-        smallBlock.setHeader(header);
-        NulsDigestData requestHash = NulsDigestData.calcDigestData(new byte[]{0});
-        manager.cacheSmallBlock(requestHash, smallBlock);
+        manager.cacheSmallBlock(smallBlock);
         assertTrue(true);
 
         this.getSmallBlock(hash, smallBlock);
 
         this.removeSmallBlock(hash);
 
-        manager.cacheSmallBlock(requestHash, smallBlock);
+        manager.cacheSmallBlock(smallBlock);
 
         this.clear();
     }
\ No newline at end of file
