diff --git a/old_test.java b/new_test.java
index 7770398..927e1f9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,12 @@
 @Test
 	public void testResourceAllocated() throws Exception {
 
-		QueueScheduler toTest = spy(new QueueScheduler(this.instanceManager));
-		JobID jobid = mock(JobID.class);
-		AllocatedResource resource = mock(AllocatedResource.class);
-		InstanceType instanceType = new InstanceType();
+		final DeploymentManager deploymentManager = new TestDeploymentManager();
+
+		final QueueScheduler toTest = spy(new QueueScheduler(deploymentManager, this.instanceManager));
+		final JobID jobid = mock(JobID.class);
+		final AllocatedResource resource = mock(AllocatedResource.class);
+		final InstanceType instanceType = new InstanceType();
 		InstanceConnectionInfo instanceConnectionInfo = mock(InstanceConnectionInfo.class);
 		when(instanceConnectionInfo.toString()).thenReturn("");
 		LocalInstance instance = spy(new LocalInstance(instanceType, instanceConnectionInfo, null, null, null));
