diff --git a/old_product.java b/new_product.java
index c0f748d..c5eecc1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,4 @@
-public static void deleteFileOrDirectory(String fileName){
+public static boolean deleteFileOrDirectory(String fileName){
         File file = new File(fileName);
-        if (file.exists()){
-            deleteFileOrDirectory(file);
-        }
-        throw new IllegalArgumentException("file:[" + fileName + "] not exists,please check it!");
+        return deleteFileOrDirectory(file);
     }
\ No newline at end of file
