diff --git a/old_test.java b/new_test.java
index 51ee59f..1c8cc6f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
   public void run() throws Exception {
-    context.prop.set(Prop.TEXTINDEX, false);
-    context.prop.set(Prop.ATTRINDEX, false);
-    context.prop.set(Prop.AUTOFLUSH, false);
+    context.options.set(Options.TEXTINDEX, false);
+    context.options.set(Options.ATTRINDEX, false);
+    context.options.set(Options.AUTOFLUSH, false);
 
     // create test database
     new CreateDB(NAME).execute(context);
