diff --git a/old_product.java b/new_product.java
index 97eb482..2fe99d7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 public boolean run(final Context ctx, final OutputStream os) {
     perf = new Performance();
     context = ctx;
-    prop = ctx.prop;
-    mprop = ctx.mprop;
+    options = ctx.options;
+    globalopts = ctx.globalopts;
     out = PrintOutput.get(os);
 
     try {
