diff --git a/old_product.java b/new_product.java
index 075b685..656390b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 String execute(Collector collector, CollectorServer collectorServer, String counterName, // NOPMD
-			String sessionId, String threadId, String jobId) throws IOException {
+			String sessionId, String threadId, String jobId, String cacheId) throws IOException {
 		// CHECKSTYLE:ON
 		String messageForReport;
 		switch (this) {
@@ -58,6 +58,10 @@ String execute(Collector collector, CollectorServer collectorServer, String coun
 			clearCaches();
 			messageForReport = I18N.getString("caches_purges");
 			break;
+		case CLEAR_CACHE:
+			clearCache(cacheId);
+			messageForReport = I18N.getFormattedString("cache_purge", cacheId);
+			break;
 		case KILL_THREAD:
 			assert threadId != null;
 			messageForReport = killThread(threadId);
