diff --git a/old_product.java b/new_product.java
index 57790d6..76b08f8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-@Override
-    public Set<Map.Entry<K, V>> entrySet() {
-        return new HashMapEntrySet<K, V>(this);
+public Set<Entry<K, V>> entrySet() {
+        Set<Entry<K, V>> es = entrySet;
+        return (es != null) ? es : (entrySet = new EntrySet());
     }
\ No newline at end of file
