diff --git a/old_product.java b/new_product.java
index 04b0c2a..ebbbf81 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public Response listLogFiles(String user, Integer port, String topologyId, String callback, String origin) throws IOException {
-        List<File> fileResults = null;
+        List<Path> fileResults = null;
         if (topologyId == null) {
             if (port == null) {
                 fileResults = workerLogs.getAllLogsForRootDir();
@@ -13,7 +13,7 @@ public Response listLogFiles(String user, Integer port, String topologyId, Strin
                         if (topoDirFiles != null) {
                             for (File portDir : topoDirFiles) {
                                 if (portDir.getName().equals(port.toString())) {
-                                    fileResults.addAll(directoryCleaner.getFilesForDir(portDir));
+                                    fileResults.addAll(directoryCleaner.getFilesForDir(portDir.toPath()));
                                 }
                             }
                         }
@@ -29,7 +29,7 @@ public Response listLogFiles(String user, Integer port, String topologyId, Strin
                     File[] topoDirFiles = topoDir.listFiles();
                     if (topoDirFiles != null) {
                         for (File portDir : topoDirFiles) {
-                            fileResults.addAll(directoryCleaner.getFilesForDir(portDir));
+                            fileResults.addAll(directoryCleaner.getFilesForDir(portDir.toPath()));
                         }
                     }
                 }
@@ -37,7 +37,7 @@ public Response listLogFiles(String user, Integer port, String topologyId, Strin
             } else {
                 File portDir = ConfigUtils.getWorkerDirFromRoot(logRoot, topologyId, port);
                 if (portDir.exists()) {
-                    fileResults = directoryCleaner.getFilesForDir(portDir);
+                    fileResults = directoryCleaner.getFilesForDir(portDir.toPath());
                 }
             }
         }
