diff --git a/old_test.java b/new_test.java
index 922659b..100cb3d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,19 @@
 @Test
 	public void testCanBeSubsumed() throws Exception {
-		PendingCheckpoint pending = createPendingCheckpoint();
+		// Forced checkpoints cannot be subsumed
+		CheckpointProperties forced = new CheckpointProperties(true, true, false, false, false, false, false);
+		PendingCheckpoint pending = createPendingCheckpoint(forced, "ignored");
+		assertFalse(pending.canBeSubsumed());
+
+		try {
+			pending.abortSubsumed();
+			fail("Did not throw expected Exception");
+		} catch (IllegalStateException ignored) {
+			// Expected
+		}
+
+		// Non-forced checkpoints can be subsumed
+		CheckpointProperties subsumed = new CheckpointProperties(false, true, false, false, false, false, false);
+		pending = createPendingCheckpoint(subsumed, "ignored");
 		assertTrue(pending.canBeSubsumed());
 	}
\ No newline at end of file
