diff --git a/old_product.java b/new_product.java
index 822608f..d86d6cf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,8 +14,9 @@ private void write(final Graphics g) {
     if(y > 0 && y < h) {
       // mark selected text
       if(text.selectStart()) {
-        int xx = x, cw = 0;
+        int xx = x;
         while(!text.inSelect() && text.more()) xx += charW(g, text.next());
+        int cw = 0;
         while(text.inSelect() && text.more()) cw += charW(g, text.next());
         g.setColor(GUIConstants.color(3));
         g.fillRect(xx, y - fontH * 4 / 5, cw, fontH);
@@ -25,8 +26,8 @@ private void write(final Graphics g) {
       // mark found text
       int xx = x;
       while(text.more() && text.searchStart()) {
-        int cw = 0;
         while(!text.inSearch() && text.more()) xx += charW(g, text.next());
+        int cw = 0;
         while(text.inSearch() && text.more()) cw += charW(g, text.next());
         g.setColor(GUIConstants.color2A);
         g.fillRect(xx, y - fontH * 4 / 5, cw, fontH);
