diff --git a/old_test.java b/new_test.java
index 9cc31ce..76f1231 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,7 +11,7 @@
 
     query("let $a := " + _ARCHIVE_ENTRIES.args(
       _FILE_READ_BINARY.args(ZIP)) + "/string() " +
-      "let $f := " + _FILE_LIST.args(tmp, "true()") + "[" +
+      "let $f := " + _FILE_LIST.args(tmp, "true()") + '[' +
       _FILE_IS_FILE.args(" '" + tmp + "/'||.") + "] ! replace(., '\\\\', '/') " +
       "return (every $e in $a satisfies $e = $f) and (every $e in $f satisfies $e =$ a)",
       "true");
