diff --git a/old_test.java b/new_test.java
index 48c2e46..4b5df10 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,5 +2,5 @@
     public void test_getDate(int year, int month, int day) {
         LocalDate d = LocalDate.of(year, month, day);
         LocalDateTime dt = LocalDateTime.of(d, LocalTime.MIDNIGHT);
-        assertSame(dt.getDate(), d);
+        assertSame(dt.toLocalDate(), d);
     }
\ No newline at end of file
