diff --git a/old_test.java b/new_test.java
index a4b13d7..2425c0c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(groups={"tck"})
     public void test_print_isoOrdinalDate() {
-        DateTime test = LocalDateTime.of(2008, 6, 3, 11, 5, 30);
+        DateTimeAccessor test = LocalDateTime.of(2008, 6, 3, 11, 5, 30);
         assertEquals(DateTimeFormatters.isoOrdinalDate().print(test), "2008-155");
     }
\ No newline at end of file
