diff --git a/old_product.java b/new_product.java
index 739974e..691d237 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,7 +9,7 @@ private void processServiceComponentHosts(Cluster cluster, KerberosDescriptor ke
       // Create the context to use for filtering Kerberos Identities based on the state of the cluster
       Map<String, Object> filterContext = new HashMap<>();
       filterContext.put("configurations", currentConfigurations);
-      filterContext.put("services", cluster.getServices().keySet());
+      filterContext.put("services", cluster.getServicesByName().keySet());
 
       try {
         Map<String, Set<String>> propertiesToIgnore = null;
