diff --git a/old_test.java b/new_test.java
index 8daa0f5..b2fd095 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testAreaAddres() throws Exception {
-        result1 = IsisUtil.areaAddres(Bytes.toArray(
-                IsisUtil.areaAddressToBytes(areaAddres)));
-        assertThat(result1, is(areaAddres));
+        result1 = IsisUtil.areaAddress(Bytes.toArray(
+                IsisUtil.areaAddressToBytes(areaAddress)));
+        assertThat(result1, is(areaAddress));
     }
\ No newline at end of file
