diff --git a/old_product.java b/new_product.java
index cdac3f9..e4a6bb4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,7 +10,7 @@
 
         StringBuffer buf = new StringBuffer();
         WebAppSecurityConfigImpl orig = (WebAppSecurityConfigImpl) original;
-        appendToBufferIfDifferent(buf, "allowFailOverToBasicAuth",
+        appendToBufferIfDifferent(buf, CFG_KEY_FAIL_OVER_TO_BASICAUTH,
                                   this.allowFailOverToBasicAuth, orig.allowFailOverToBasicAuth);
         appendToBufferIfDifferent(buf, "allowLogoutPageRedirectToAnyHost",
                                   this.allowLogoutPageRedirectToAnyHost, orig.allowLogoutPageRedirectToAnyHost);
@@ -48,11 +48,11 @@
                                   this.useAuthenticationDataForUnprotectedResource, orig.useAuthenticationDataForUnprotectedResource);
         appendToBufferIfDifferent(buf, "webAlwaysLogin",
                                   this.webAlwaysLogin, orig.webAlwaysLogin);
-        appendToBufferIfDifferent(buf, "loginFormURL",
+        appendToBufferIfDifferent(buf, CFG_KEY_LOGIN_FORM_URL,
                                   this.loginFormURL, orig.loginFormURL);
-        appendToBufferIfDifferent(buf, "loginErrorURL",
+        appendToBufferIfDifferent(buf, CFG_KEY_LOGIN_ERROR_URL,
                                   this.loginErrorURL, orig.loginErrorURL);
-        appendToBufferIfDifferent(buf, "allowFailOverToAuthMethod",
+        appendToBufferIfDifferent(buf, CFG_KEY_ALLOW_FAIL_OVER_TO_AUTH_METHOD,
                                   this.allowFailOverToAuthMethod, orig.allowFailOverToAuthMethod);
         appendToBufferIfDifferent(buf, "includePathInWASReqURL",
                                   this.includePathInWASReqURL, orig.includePathInWASReqURL);
@@ -62,11 +62,11 @@
                                   this.useOnlyCustomCookieName, orig.useOnlyCustomCookieName);
         appendToBufferIfDifferent(buf, "wasReqURLRedirectDomainNames",
                                   this.wasReqURLRedirectDomainNames, orig.wasReqURLRedirectDomainNames);
-        appendToBufferIfDifferent(buf, "overrideHttpAuthenticationMechanism",
+        appendToBufferIfDifferent(buf, CFG_KEY_OVERRIDE_HAM,
                                   this.overrideHttpAuthenticationMechanism, orig.overrideHttpAuthenticationMechanism);
-        appendToBufferIfDifferent(buf, "loginFormContextRoot",
+        appendToBufferIfDifferent(buf, CFG_KEY_LOGIN_FORM_CONTEXT_ROOT,
                                   this.loginFormContextRoot, orig.loginFormContextRoot);
-        appendToBufferIfDifferent(buf, "basicAuthRealmName",
+        appendToBufferIfDifferent(buf, CFG_KEY_BASIC_AUTH_REALM_NAME,
                                   this.basicAuthRealmName, orig.basicAuthRealmName);
         return buf.toString();
     }
\ No newline at end of file
