diff --git a/old_product.java b/new_product.java
index b81594e..75d1158 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,10 @@
-CompletableFuture<Void> initialize(Duration timeout) {
+CompletableFuture<Void> initialize(Duration timeout, Executor executor) {
         Exceptions.checkNotClosed(isClosed(), this);
         Preconditions.checkState(this.state.get() == AggregatorState.NotInitialized, "SegmentAggregator has already been initialized.");
+        assert this.handle.get() == null : "non-null handle but state == " + this.state.get();
         long traceId = LoggerHelpers.traceEnterWithContext(log, this.traceObjectId, "initialize");
 
-        return this.storage
-                .open(this.metadata.getName())
-                .thenCompose(v -> this.storage.getStreamSegmentInfo(this.metadata.getName(), timeout))
+        return openWrite(this.metadata.getName(), this.handle, executor, timeout)
                 .thenAccept(segmentInfo -> {
                     // Check & Update StorageLength in metadata.
                     if (this.metadata.getStorageLength() != segmentInfo.getLength()) {
