diff --git a/old_product.java b/new_product.java
index b2057a1..b024570 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 boolean union(final CmpG g, final QueryContext qc, final VarScope scp) throws QueryException {
     if(op != g.op || !exprs[0].sameAs(g.exprs[0])) return false;
     exprs[1] = new List(info, exprs[1], g.exprs[1]).compile(qc, scp);
-    atomic = atomic && exprs[1].type().zeroOrOne();
+    atomic = atomic && exprs[1].seqType().zeroOrOne();
     return true;
   }
\ No newline at end of file
