diff --git a/old_test.java b/new_test.java
index 38bc334..8a8e46e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,5 +20,5 @@
             "<gml:Point><gml:coordinates>2,3</gml:coordinates></gml:Point>)",
             GeoErrors.qname(2));
     error("geo:union(text {'a'}, <gml:Point><gml:coordinates>2,3" +
-            "</gml:coordinates></gml:Point>)", FUNCMP.qname());
+            "</gml:coordinates></gml:Point>)", FUNTYPE.qname());
   }
\ No newline at end of file
