diff --git a/old_test.java b/new_test.java
index ca91362..25247f9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
 		assertNotNull(orderFrequency);
 		
 		orderFrequency.setFrequencyPerDay(4d);
-		expectedException.expect(CannotEditOrderPropertyInUseException.class);
+		expectedException.expect(CannotUpdateObjectInUseException.class);
 		expectedException.expectMessage("Order.frequency.cannot.edit");
 		orderService.saveOrderFrequency(orderFrequency);
 	}
\ No newline at end of file
