diff --git a/old_product.java b/new_product.java
index bf73cfc..7d3b68e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static void validate(Object obj) throws ValidationException {
-		BindException errors = new BindException(obj, "");
+		Errors errors = new BindException(obj, "");
 		
 		Context.getAdministrationService().validate(obj, errors);
 		
@@ -16,6 +16,6 @@ public static void validate(Object obj) throws ValidationException {
 			
 			String exceptionMessage = "'" + obj + "' failed to validate with reason: ";
 			exceptionMessage += StringUtils.join(uniqueErrorMessages, ", ");
-			throw new ValidationException(exceptionMessage, errors.getCause());
+			throw new ValidationException(exceptionMessage, errors);
 		}
 	}
\ No newline at end of file
