diff --git a/old_test.java b/new_test.java
index 61c3608..af80264 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,14 @@
-@Test(expected = APIException.class)
-	@Verifies(value = "should throw APIException if errors occur during validation", method = "validate(Object)")
+@Test
+	@Verifies(value = "should return Spring errors in ValidationException", method = "validate(Object)")
 	public void validate_shouldThrowAPIExceptionIfErrorsOccurDuringValidation() throws Exception {
 		Location loc = new Location();
-		ValidateUtil.validate(loc);
+		
+		try {
+			ValidateUtil.validate(loc);
+		}
+		catch (ValidationException validationException) {
+			assertNotNull(validationException.getErrors());
+			assertTrue(validationException.getErrors().hasErrors());
+		}
+		
 	}
\ No newline at end of file
