diff --git a/old_product.java b/new_product.java
index 390d4db..d1e12fc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public boolean isObserverInterested(MetaDataCache mdc, Annotation... bindings)
+public boolean isObserverInterested(Annotation... bindings)
    {
       // Simply check that all event bindings specified by the observer are
       // in the list provided.
@@ -14,7 +14,7 @@ public boolean isObserverInterested(MetaDataCache mdc, Annotation... bindings)
             boolean found = false;
             for (Annotation y: bindings)
             {
-        	if ( mdc.getBindingTypeModel(x.annotationType()).isEqual(x, y) ) {
+        	if ( metaDataCache.getBindingTypeModel(x.annotationType()).isEqual(x, y) ) {
                found = true;
             }
             }
