diff --git a/old_test.java b/new_test.java
index c8bcae1..0a6c66c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,6 +9,6 @@
         if (expectedPos > 0) {
             assertParsed(parseContext, DAY_OF_MONTH, expectedValue);
         } else {
-            assertEquals(parseContext.toCalendricalMerger().getInputMap().containsKey(DAY_OF_MONTH), false);
+            assertEquals(parseContext.getParsed().size(), 0);
         }
     }
\ No newline at end of file
