diff --git a/old_product.java b/new_product.java
index b391dbb..44a56eb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public CalendricalMerger parse(CharSequence text) {
+public CalendricalNormalizer parse(CharSequence text) {
         DateTimeFormatter.checkNotNull(text, "Text must not be null");
         String str = text.toString();
         ParsePosition pos = new ParsePosition(0);
