diff --git a/old_product.java b/new_product.java
index 6ea8502..6350ecf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,7 +10,7 @@ public RequestStatus createResources(final Set<ConfigGroupRequest> requests)thro
           }
         });
 
-    Set<Resource> associatedResources = new HashSet<Resource>();
+    Set<Resource> associatedResources = new HashSet<>();
     for (ConfigGroupResponse response : responses) {
       Resource resource = new ResourceImpl(Resource.Type.ConfigGroup);
       resource.setProperty(CONFIGGROUP_ID_PROPERTY_ID, response.getId());
