diff --git a/old_test.java b/new_test.java
index 3552877..b602e6c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
 	public final void testProcess3() {
-		testOsmWriter.process(new NodeContainer(new Node(1234, 0, new Date(), new OsmUser(12, "OsmosisTest"), 20, 20)));
+		testOsmWriter.process(new NodeContainer(new Node(1234, 0, new Date(), new OsmUser(12, "OsmosisTest"), new ArrayList<Tag>(), 20, 20)));
 		// Nothing to assert; just expect no exception
 	}
\ No newline at end of file
