diff --git a/old_product.java b/new_product.java
index 45851a1..77976a4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,17 +1,11 @@
 public void process(NodeContainer nodeContainer) {
-		Node inputEntity = nodeContainer.getEntity();
+		nodeBuilder.initialize(nodeContainer.getEntity());
+		nodeBuilder.setTimestamp(timestamp);
 		
 		changeSink.process(
 			new ChangeContainer(
 				new NodeContainer(
-					new Node(
-						inputEntity.getId(),
-						inputEntity.getVersion(),
-						timestamp,
-						inputEntity.getUser(),
-						inputEntity.getLatitude(),
-						inputEntity.getLongitude()
-					)
+					nodeBuilder.buildEntity()
 				),
 				action
 			)
