diff --git a/old_product.java b/new_product.java
index 38c728f..8379262 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,7 @@
 public static boolean isInterface(Class<?> ownerClass){
-        int flag = ownerClass.getModifiers();// 返回此类或接口以整数编码的 Java 语言修饰符
-        return Modifier.isInterface(flag);// 对类和成员访问修饰符进行解码
+        if (null == ownerClass){
+            return false;
+        }
+        int modifiers = ownerClass.getModifiers();// 返回此类或接口以整数编码的 Java 语言修饰符
+        return Modifier.isInterface(modifiers);// 对类和成员访问修饰符进行解码
     }
\ No newline at end of file
