diff --git a/old_test.java b/new_test.java
index f03a259..4c8f17e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,6 @@
 @Test
     public void testIsInterface(){
+        assertEquals(false, ClassUtil.isInterface(null));
         assertEquals(false, ClassUtil.isInterface(this.getClass()));
         assertEquals(false, ClassUtil.isInterface(DatePattern.class));
     }
\ No newline at end of file
