diff --git a/old_test.java b/new_test.java
index 5be541d..93a5bd3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,16 @@
 @Test
   public void getAuthorityTests() {
     String[] authorities =
-        new String[] {"localhost", "localhost:8080", "127.0.0.1", "127.0.0.1:8080", "localhost",
-            null};
+        new String[] {"localhost", "localhost:8080", "127.0.0.1", "127.0.0.1:8080", "localhost"};
     for (String authority : authorities) {
-      AlluxioURI uri = new AlluxioURI("file", authority, "/a/b");
-      assertEquals(authority, uri.getAuthority());
+      AlluxioURI uri = new AlluxioURI("file", Authority.fromString(authority), "/a/b");
+      assertEquals(authority, uri.getAuthority().toString());
     }
 
-    assertEquals(null, new AlluxioURI("file", "", "/b/c").getAuthority());
-    assertEquals(null, new AlluxioURI("file", null, "/b/c").getAuthority());
-    assertEquals(null, new AlluxioURI("file:///b/c").getAuthority());
+    assertEquals("",
+        new AlluxioURI("file", Authority.fromString(""), "/b/c").getAuthority().toString());
+    assertEquals("", new AlluxioURI("file", null, "/b/c").getAuthority().toString());
+    assertEquals("",
+        new AlluxioURI("file", Authority.fromString(null), "/b/c").getAuthority().toString());
+    assertEquals("", new AlluxioURI("file:///b/c").getAuthority().toString());
   }
\ No newline at end of file
