diff --git a/old_test.java b/new_test.java
index 3ea45a3..f494401 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,7 +11,7 @@
 
         MpPromoteAlgo algo = new MpPromoteAlgo(masters, mailbox, "Test");
         long requestId = algo.getRequestId();
-        Future<Pair<Long, Long>> result = algo.start();
+        Future<RepairResult> result = algo.start();
         verify(mailbox, times(1)).send(any(long[].class), any(Iv2RepairLogRequestMessage.class));
 
         final long uid = uig.getNextUniqueId();
@@ -43,9 +43,8 @@
         needsRepair.add(2L);
         needsRepair.add(3L);
         verify(mailbox, times(1)).repairReplicasWith(eq(needsRepair), any(Iv2RepairLogResponseMessage.class));
-        Long real_result = result.get().getFirst();
-        assertEquals(txnEgo(1000L), (long)real_result);
-        assertEquals(new Long(uid), result.get().getSecond());
+        assertEquals(txnEgo(1000L), result.get().m_txnId);
+        assertEquals(uid, result.get().m_uniqueId);
 
         // check if the hashinator was updated to the newer version
         assertEquals(torv3.getFirst(), TheHashinator.getCurrentVersionedConfig().getFirst());
