diff --git a/old_product.java b/new_product.java
index 379f65e..f7312d7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -31,5 +31,8 @@ public void repairSurvivors()
         }
         tmLog.debug(m_whoami + "finished queuing " + queued + " replica repair messages.");
 
-        m_promotionResult.set(Pair.of(m_maxSeenTxnId, m_maxSeenUniqueId));
+        m_promotionResult.set(new RepairResult(
+                m_maxSeenTxnId,
+                m_maxSeenUniqueId,
+                m_maxSeenBinaryLogUniqueId));
     }
\ No newline at end of file
