diff --git a/old_test.java b/new_test.java
index d32ac29..8ae2334 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,29 +2,25 @@
     @Test
     public void testGetShuffleMode() throws InterruptedException {
         final int testShuffleMode = MediaPlaylistAgent.SHUFFLE_MODE_GROUP;
-        final MediaPlaylistAgent agent = new MediaPlaylistAgent() {
-            @Override
-            public int getShuffleMode() {
-                return testShuffleMode;
-            }
-        };
+        mMockAgent.setShuffleMode(testShuffleMode);
+
         final CountDownLatch latch = new CountDownLatch(1);
         final SessionCallback sessionCallback = new SessionCallback() {
             @Override
             public void onShuffleModeChanged(MediaSession2 session,
                     MediaPlaylistAgent playlistAgent, int shuffleMode) {
-                assertEquals(agent, playlistAgent);
+                assertEquals(mMockAgent, playlistAgent);
                 assertEquals(testShuffleMode, shuffleMode);
                 latch.countDown();
             }
         };
         try (MediaSession2 session = new MediaSession2.Builder(mContext)
                 .setPlayer(mPlayer)
-                .setPlaylistAgent(agent)
+                .setPlaylistAgent(mMockAgent)
                 .setId("testGetShuffleMode")
                 .setSessionCallback(sHandlerExecutor, sessionCallback)
                 .build()) {
-            agent.notifyShuffleModeChanged();
+            mMockAgent.notifyShuffleModeChanged();
             assertTrue(latch.await(TIMEOUT_MS, TimeUnit.MILLISECONDS));
         }
     }
\ No newline at end of file
