diff --git a/old_test.java b/new_test.java
index 9c7642f..afd1393 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,4 @@
-@Test(groups={"tck"})
-    public void test_plusSeconds() {
-        LocalDateTime ldt = LocalDateTime.of(2008, 6, 30, 23, 30, 59, 0);
-        ZonedDateTime base = ZonedDateTime.of(ldt, ZONE_0100);
-        ZonedDateTime test = base.plusSeconds(1);
-        assertEquals(test, ZonedDateTime.of(ldt.plusSeconds(1), ZONE_0100));
+@Test(groups={"tck"}, dataProvider="plusTime")
+    public void test_plusSeconds(ZonedDateTime base, long amount, ZonedDateTime expected) {
+        assertEquals(base.plusSeconds(amount * 3600), expected);
     }
\ No newline at end of file
