diff --git a/old_test.java b/new_test.java
index 62cbc0e..b5a6c4e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -47,7 +47,7 @@
             .mapToObj(i -> c.getCell(i))
             .collect(Collectors.toList());
         
-        Activity activity = c.newComputeActivity(inputVec, .5, 2, .1, 1, true);
+        Activity activity = c.computeActivity(inputVec, .5, 2, .1, 1, true);
         List<SegmentOverlap> active = activity.activeSegments;
         List<SegmentOverlap> matching = activity.matchingSegments;
         
