diff --git a/old_product.java b/new_product.java
index 7e12f48..6712347 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-@Deprecated
-    public Tuple computeActivity(Set<Cell> activeInput, double activePermanenceThreshold,
-        int activeSynapseThreshold, double matchingPermananceThreshold, int matchingSynapseThreshold) {
+public Activity computeActivity(Collection<Cell> activeInput, double activePermanenceThreshold,
+        int activeSynapseThreshold, double matchingPermananceThreshold, int matchingSynapseThreshold,
+            boolean recordIteration) {
         
         int nextSegmentIdx = getSegmentCount();
         
@@ -28,25 +28,31 @@
             }
         }
         
-        List<DistalDendrite> activeSegments = new ArrayList<>();
-        List<DistalDendrite> matchingSegments = new ArrayList<>();
+        if(recordIteration) {
+            tmIteration++;
+        }
+        
+        List<SegmentOverlap> activeSegments = new ArrayList<>();
+        List<SegmentOverlap> matchingSegments = new ArrayList<>();
         for(int i = 0;i < nextSegmentIdx;i++) {
             if(((int)numActiveSynapsesForSegment[i][1]) >= activeSynapseThreshold) {
-                activeSegments.add(((DistalDendrite)numActiveSynapsesForSegment[i][0]));
+                activeSegments.add(new SegmentOverlap(((DistalDendrite)numActiveSynapsesForSegment[i][0]),
+                    (int)numActiveSynapsesForSegment[i][1]));
+                
+                if(recordIteration) {
+                    ((DistalDendrite)numActiveSynapsesForSegment[i][0]).setLastUsedIteration(tmIteration);
+                }
             }
         }
         
         for(int i = 0;i < nextSegmentIdx;i++) {
             if(((int)numMatchingSynapsesForSegment[i][1]) >= matchingSynapseThreshold) {
-                matchingSegments.add(((DistalDendrite)numMatchingSynapsesForSegment[i][0]));
+                matchingSegments.add(new SegmentOverlap(((DistalDendrite)numMatchingSynapsesForSegment[i][0]),
+                    (int)numMatchingSynapsesForSegment[i][1]));
             }
         }
         
-        return new Tuple(
-            (Object)activeSegments.stream()
-                .sorted()
-                .collect(Collectors.toCollection(LinkedHashSet<DistalDendrite>::new)),
-            (Object)matchingSegments.stream()
-                .sorted()
-                .collect(Collectors.toCollection(LinkedHashSet<DistalDendrite>::new)));
+        Collections.sort(activeSegments, (as1, as2) -> as1.segment.getIndex() - as2.segment.getIndex());
+        Collections.sort(matchingSegments, (ms1, ms2) -> ms1.segment.getIndex() - ms2.segment.getIndex());
+        return new Activity(activeSegments, matchingSegments);
     }
\ No newline at end of file
