diff --git a/old_test.java b/new_test.java
index cf69a80..746f93f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(expected = IllegalArgumentException.class)
-	@Verifies(value = "fail if null passed in", method = "getOrderables(String)")
+	@Verifies(value = "shoudl fail if null passed in", method = "getOrderables(String)")
 	public void getOrderables_shouldFailIfNullPassedIn() throws Exception {
 		executeDataSet(simpleOrderEntryDatasetFilename);
 		
