diff --git a/old_test.java b/new_test.java
index 18c1aa4..f90fb13 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
 	public void testGetInput() {
-		final Operator input1 = new OpImpl(0);
-		final Operator input2 = new OpImpl(1);
-		final Operator fixture = new OpImpl(0).withInputs(input1, input2);
+		final Operator<?> input1 = new OpImpl(0);
+		final Operator<?> input2 = new OpImpl(1);
+		final Operator<?> fixture = new OpImpl(0).withInputs(input1, input2);
 
 		assertSame(input1.getOutput(0), fixture.getInput(0));
 		assertSame(input2.getOutput(0), fixture.getInput(1));
