diff --git a/old_test.java b/new_test.java
index 50b8c0b..ecfd3f9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
 		RelationshipType type = new RelationshipType();
 		type.setaIsToB("A is To B");
 		type.setbIsToA("B is To A");
-		
+		type.setDescription("Description");
 		Errors errors = new BindException(type, "type");
 		new RelationshipTypeValidator().validate(type, errors);
 		Assert.assertFalse(errors.hasErrors());
