diff --git a/old_product.java b/new_product.java
index a4d45b1..73fcddd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,6 +5,7 @@ public void validate(Object obj, Errors errors) {
 		} else {
 			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "aIsToB", "RelationshipType.aIsToB.required");
 			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "bIsToA", "RelationshipType.bIsToA.required");
+			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "description", "RelationshipType.description.required");
 			RelationshipType exist = Context.getPersonService().getRelationshipTypeByName(
 			    relationshipType.getaIsToB() + "/" + relationshipType.getbIsToA());
 			if (exist != null && !exist.isRetired()
