diff --git a/old_test.java b/new_test.java
index 23672cb..a84fd23 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,7 +12,7 @@
 		Assert.assertTrue("There should be some active visits for this test to be valid", activeVisitCount > 0);
 		
 		//close any unvoided open visits
-		service.stopVisits();
+		service.stopVisits(null);
 		
 		activeVisitCount = Context.getAdministrationService().executeSQL(openVisitsQuery, true).size();
 		
