diff --git a/old_product.java b/new_product.java
index 139c4c5..34e5946 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static void pixelToNormalized(Point2D_F64 orig, Point2D_F64 normed, DenseMatrix64F N) {
+public static void pixelToNormalized(DenseMatrix64F N, Point2D_F64 orig, Point2D_F64 normed) {
 		normed.x = orig.x * N.data[0] + N.data[2];
 		normed.y = orig.y * N.data[4] + N.data[5];
 	}
\ No newline at end of file
