diff --git a/old_product.java b/new_product.java
index fb65e66..46005db 100644
--- a/old_product.java
+++ b/new_product.java
@@ -17,8 +17,12 @@
         }
 
         String jobArchivalLocation = null;
-        if (!jobRequestWithId.isDisableLogArchival() && baseArchiveLocation != null) {
-           jobArchivalLocation = baseArchiveLocation + "/" + jobRequestWithId.getId();
+        if (!jobRequestWithId.isDisableLogArchival()) {
+            if(baseArchiveLocation == null) {
+                throw new
+                    GeniePreconditionException("Job archival is enabled but base location for archival is null.");
+            }
+            jobArchivalLocation = baseArchiveLocation + "/" + jobRequestWithId.getId();
         }
         // create the job object in the database with status INIT
         final Job job  = new Job.Builder(
