diff --git a/old_product.java b/new_product.java
index 49872c3..68b4016 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
-public static WaveletDesc_F32 generate_F32( int I ) {
+public static WaveletCoefficient_F32 generate_F32( int I ) {
 		if( I != 6 ) {
 			throw new IllegalArgumentException("Only 6 is currently supported");
 		}
 
-		WaveletDesc_F32 ret = new WaveletDesc_F32();
+		WaveletCoefficient_F32 ret = new WaveletCoefficient_F32();
 
 		ret.offsetScaling = -2;
 		ret.offsetWavelet = -2;
