diff --git a/old_product.java b/new_product.java
index fafef61..82e800f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,8 @@
 public void setInputs(final List<? extends JsonStream> inputs) {
 		if (inputs == null)
 			throw new NullPointerException("inputs must not be null");
+		if (this.minInputs > inputs.size() || inputs.size() > this.maxInputs)
+			throw new IndexOutOfBoundsException();
 
 		this.inputs.clear();
 		for (final JsonStream input : inputs)
