diff --git a/old_test.java b/new_test.java
index 9820f3c..9af55d0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
             AttributeDataset data = parser.parse(smile.data.parser.IOUtils.getTestDataFile("weka/regression/abalone.arff"));
             double[][] x = data.toArray(new double[data.size()][]);
             
-            FeatureSet<double[]> features = new FeatureSet<double[]>();
+            FeatureSet<double[]> features = new FeatureSet<>();
             features.add(new Nominal2Binary(data.attributes()));
             features.add(new NumericAttributeFeature(data.attributes(), 0.05, 0.95, x));
             
