diff --git a/old_product.java b/new_product.java
index 11035a5..3bb6770 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@ public AttributeDataset f(Dataset<T> data) {
 
         for (int i = 0; i < data.size(); i++) {
             Datum<T> datum = data.get(i);
-            Datum<double[]> x = new Datum<double[]>(f(datum.x), datum.y, datum.weight);
+            Datum<double[]> x = new Datum<>(f(datum.x), datum.y, datum.weight);
             x.name = datum.name;
             x.description = datum.description;
             x.timestamp = datum.timestamp;
