diff --git a/old_product.java b/new_product.java
index 1134b9e..87b4119 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@
       @Override
       public Void call() throws Exception {
         if (options == null) {
-          mFileSystem.delete(new AlluxioURI(path), DeletePOptions.getDefaultInstance());
+          mFileSystem.delete(new AlluxioURI(path));
         } else {
           mFileSystem.delete(new AlluxioURI(path), options);
         }
