diff --git a/old_test.java b/new_test.java
index e8edaaa..bdb68f5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,8 @@
     AlluxioURI file = new AlluxioURI("/file");
     DeletePOptions deleteOptions = DeletePOptions.newBuilder().setRecursive(true).build();
     mFileSystem.delete(file, deleteOptions);
-    verify(mFileSystemMasterClient).delete(file, deleteOptions);
+    verify(mFileSystemMasterClient).delete(file,
+        FileSystemOptions.deleteDefaults(mConf).toBuilder().mergeFrom(deleteOptions).build());
 
     verifyFilesystemContextAcquiredAndReleased();
   }
\ No newline at end of file
