diff --git a/old_product.java b/new_product.java
index 3ff434a..5f5a0a5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@ private <ML extends AbstractClassifier, FS extends AbstractFeatureSelector, NS e
             NS.AbstractTrainingParameters numericalScalerTrainingParameters,
             double expectedF1score,
             int testId) {
-        Configuration configuration = Configuration.getConfiguration();
+        Configuration configuration = getConfiguration();
         
         
         String storageName = this.getClass().getSimpleName() + testId;
