diff --git a/old_test.java b/new_test.java
index 3d8665f..dfd4c23 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testTrainAndValidate() {
         logger.info("testTrainAndValidate");
         
-        Configuration configuration = Configuration.getConfiguration();
+        Configuration configuration = getConfiguration();
 
         Dataframe[] data = Datasets.heartDiseaseClusters(configuration);
 
