diff --git a/old_product.java b/new_product.java
index c362663..92180af 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public static <O> List<O> sort(List<O> list,String...propertyNames){
         if (null == list){
-            return null;
+            return emptyList();
         }
         Validate.notEmpty(propertyNames, "propertyNames can't be null/empty!");
         Validate.noNullElements(propertyNames, "propertyName:%s has empty value", propertyNames);
