diff --git a/old_test.java b/new_test.java
index c7197f8..0b6cc74 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
         map.put("a13", 123);
         map.put("a2", 345);
 
-        Map<String, Integer> sortByKeyAsc = SortUtil.sortByKeyAsc(map);
+        Map<String, Integer> sortByKeyAsc = sortByKeyAsc(map);
         assertThat(sortByKeyAsc.keySet(), contains("a13", "a2", "a8"));
 
         Map<String, Integer> sort = sort(
