diff --git a/old_test.java b/new_test.java
index 2bb9f74..7ee6b73 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,5 @@
         QueryCriterion criterion2 = new GtCriterion("id", 5);
         QueryCriterion criterion3 = new OrCriterion(criterion1, criterion2);
         
-        assertTrue(instance.or(criterion1, criterion2).getQueryCriterions()
-                .contains(criterion3));
+        assertEquals(criterion3, instance.or(criterion1, criterion2).getQueryCriterion());
     }
\ No newline at end of file
