diff --git a/old_test.java b/new_test.java
index b022943..7308f1b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,8 @@
     ExecutorLoader loader = createLoader();
     List<Executor> executors = addTestExecutors(loader);
 
-    loader.inactivateExecutor(executors.get(0).getId());
+    executors.get(0).setActive(false);
+    loader.updateExecutor(executors.get(0));
 
     List<Executor> fetchedExecutors = loader.fetchActiveExecutors();
     Assert.assertEquals(executors.size(), fetchedExecutors.size() + 1);
