diff --git a/old_product.java b/new_product.java
index 3197d79..3f74f9d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,7 @@
         runner.query(FetchExecutorHandler.FETCH_ACTIVE_EXECUTORS,
           executorHandler);
       return executors;
-    } catch (SQLException e) {
+    } catch (Exception e) {
       throw new ExecutorManagerException("Error fetching active executors", e);
     }
   }
\ No newline at end of file
