diff --git a/old_product.java b/new_product.java
index c0b375b..4017e5a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,6 +12,8 @@
 		if (user != null) {
 			List<Locale> proficientLocales = user.getProficientLocales();
 			if (proficientLocales != null) {
+				locales.addAll(proficientLocales);
+				
 				//limit proficient locales to only allowed locales
 				List<Locale> allowedLocales = getAllowedLocales();
 				if (allowedLocales != null) {
@@ -22,10 +24,8 @@
 						retainLocales.add(new Locale(allowedLocale.getLanguage()));
 					}
 					
-					proficientLocales.retainAll(retainLocales);
+					locales.retainAll(retainLocales);
 				}
-				
-				locales.addAll(proficientLocales); //allowed user proficient locales
 			}
 		}
 		
