diff --git a/old_product.java b/new_product.java
index 839793f..a2388e3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,7 +13,7 @@ private void list(final int root, final File dir, final StringList list,
     }
     // parse files. ignore directories if a pattern is specified
     for(final File f : ch) {
-      if(pat == null || pat.matcher(f.getName()).matches() && !f.isDirectory()) {
+      if(pat == null || pat.matcher(f.getName()).matches()) {
         list.add(f.getPath().substring(root));
       }
     }
