diff --git a/old_product.java b/new_product.java
index 13d04cc..d61f4ff 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-protected boolean linearEstimate(List<CalibrationObservation> observations , Zhang99AllParam param )
+protected boolean linearEstimate(List<CalibrationObservation> observations  )
 	{
 		status("Estimating Homographies");
 		List<DMatrixRMaj> homographies = new ArrayList<>();
@@ -28,6 +28,6 @@ protected boolean linearEstimate(List<CalibrationObservation> observations , Zha
 
 		double distort[] = computeRadial.getParameters();
 
-		convertIntoZhangParam(motions, K,distort, param);
+		convertIntoBundleStructure(motions, K,distort,observations);
 		return true;
 	}
\ No newline at end of file
