diff --git a/old_product.java b/new_product.java
index d7bdb7a..7d08f0e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 public Builder<T> dithering(Dithering config)
 		{
 			checkForNull(config, "Dithering is null.");
+			updateStatus(Properties.RESIZER_FACTORY, Status.CANNOT_SET);
 			updateStatus(Properties.DITHERING, Status.ALREADY_SET);
 			dithering = config;
 			return this;
