diff --git a/old_test.java b/new_test.java
index 1d0bcc7..2a07005 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,8 @@
     CommonUtils.sleepMs(10);
 
     // create nested directory
-    InodeTree.CreatePathResult createResult = mTree.createPath(NESTED_URI, sNestedDirectoryOptions);
+    InodeTree.CreatePathResult createResult =
+        createPath(mTree, NESTED_URI, sNestedDirectoryOptions);
     List<Inode<?>> modified = createResult.getModified();
     List<Inode<?>> created = createResult.getCreated();
 
@@ -25,7 +26,7 @@
 
     // creating the directory path again results in no new inodes.
     try {
-      mTree.createPath(NESTED_URI, sNestedDirectoryOptions);
+      createPath(mTree, NESTED_URI, sNestedDirectoryOptions);
       Assert.assertTrue("createPath should throw FileAlreadyExistsException", false);
     } catch (FileAlreadyExistsException e) {
       Assert.assertEquals(e.getMessage(),
@@ -35,7 +36,7 @@
     // create a file
     CreateFileOptions options =
         CreateFileOptions.defaults().setBlockSizeBytes(Constants.KB).setRecursive(true);
-    createResult = mTree.createPath(NESTED_FILE_URI, options);
+    createResult = createPath(mTree, NESTED_FILE_URI, options);
     modified = createResult.getModified();
     created = createResult.getCreated();
 
