diff --git a/old_test.java b/new_test.java
index e7209f8..1feed5c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testShuffle() {
         TestUtils.log(this.getClass(), "shuffle");
-        RandomValue.randomGenerator = new Random(42);
+        RandomValue.setRandomGenerator(new Random(42));
         Integer[] result = {1,2,3,4,5};
         Integer[] expResult = {2,3,4,5,1};
         PHPfunctions.shuffle(result);
