diff --git a/old_product.java b/new_product.java
index 10224c0..ce05adc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1,4 @@
-public <T> void register(String name, Supplier<T> supplier, Class<T> type, @Nullable String description);
\ No newline at end of file
+public <T> void register(String name, Supplier<T> supplier, Class<T> type, @Nullable String desc) {
+        addMetric(name, new ObjectGauge<>(metricName(grpName, name), desc,
+            nonThrowableSupplier(supplier, log), type));
+    }
\ No newline at end of file
