diff --git a/old_test.java b/new_test.java
index 3fc3582..cdafd5a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testSetEnvPropFile() {
-        Assert.assertNull(this.a.getEnvPropFile());
+        Assert.assertNull(this.a.getSetupFile());
         final String propFile = "s3://netflix.propFile";
-        this.a.setEnvPropFile(propFile);
-        Assert.assertEquals(propFile, this.a.getEnvPropFile());
+        this.a.setSetupFile(propFile);
+        Assert.assertEquals(propFile, this.a.getSetupFile());
     }
\ No newline at end of file
