diff --git a/old_product.java b/new_product.java
index 4bebe14..7cd60c3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,6 +4,15 @@
             final AsyncResponse asyncResponse) {
         long traceId = LoggerHelpers.traceEnter(log, "updateStream");
 
+        try {
+            authenticate(scopeName + "/" + streamName, READ_UPDATE);
+        } catch (AuthenticationException e) {
+            log.warn("Update stream for {} failed due to authentication failure.", scopeName + "/" + streamName);
+            asyncResponse.resume(Response.status(Status.UNAUTHORIZED).build());
+            LoggerHelpers.traceLeave(log, "Update stream", traceId);
+            return;
+        }
+
         StreamConfiguration streamConfiguration = ModelHelper.getUpdateStreamConfig(
                 updateStreamRequest, scopeName, streamName);
         controllerService.updateStream(streamConfiguration).thenApply(streamStatus -> {
