diff --git a/old_product.java b/new_product.java
index 2aa79cd..8e89aa3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,8 +12,8 @@ protected void determineCPE(Dependency dependency) throws CorruptIndexException,
                 LOGGER.debug("product search: {}", products);
             }
             if (!vendors.isEmpty() && !products.isEmpty()) {
-                final List<IndexEntry> entries = searchCPE(vendors, products, dependency.getProductEvidence().getWeighting(),
-                        dependency.getVendorEvidence().getWeighting());
+                final List<IndexEntry> entries = searchCPE(vendors, products, dependency.getVendorEvidence().getWeighting(),
+                        dependency.getProductEvidence().getWeighting());
                 if (entries == null) {
                     continue;
                 }
