diff --git a/old_test.java b/new_test.java
index cc77007..9eb14e1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,6 +10,8 @@
         mTree.createPath(NESTED_URI, sNestedDirectoryOptions);
     List<Inode<?>> modified = createResult.getModified();
     List<Inode<?>> created = createResult.getCreated();
+    createResult.unlock();
+
     // 1 modified directory
     Assert.assertEquals(1, modified.size());
     Assert.assertEquals("", modified.get(0).getName());
@@ -25,7 +27,7 @@
 
     // creating the directory path again results in no new inodes.
     try {
-      createResult = mTree.createPath(NESTED_URI, sNestedDirectoryOptions);
+      createAndUnlockPath(NESTED_URI, sNestedDirectoryOptions);
       Assert.assertTrue("createPath should throw FileAlreadyExistsException", false);
     } catch (FileAlreadyExistsException e) {
       Assert.assertEquals(e.getMessage(),
@@ -38,6 +40,8 @@
     createResult = mTree.createPath(NESTED_FILE_URI, options);
     modified = createResult.getModified();
     created = createResult.getCreated();
+    createResult.unlock();
+
     // test directory was modified
     Assert.assertEquals(1, modified.size());
     Assert.assertEquals("test", modified.get(0).getName());
