diff --git a/old_test.java b/new_test.java
index 1408992..ad2094f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,13 +9,13 @@
 
 		Assert.assertEquals(
 				"addr1\naddr2\n121212 City1\nRegion Region1 Country1",
-				builder.buildAddressString(location, isLocalAddress, bPartnerBlock, userBlock)
+				builder.buildAddressString(org, location,"de_CH", isLocalAddress, bPartnerBlock, userBlock)
 				);
 
 		isLocalAddress = true;
 
 		Assert.assertEquals(
 				"LOCAL: addr1\naddr2\n121212 City1\nRegion Region1",
-				builder.buildAddressString(location, isLocalAddress, bPartnerBlock, userBlock)
+				builder.buildAddressString(org, location,"de_CH", isLocalAddress, bPartnerBlock, userBlock)
 				);
 	}
\ No newline at end of file
