diff --git a/old_product.java b/new_product.java
index ca3acb2..be66d83 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,7 @@
-public String buildAddressString(final I_C_Location location, boolean isLocalAddress, String bPartnerBlock, String userBlock)
+public String buildAddressString(final I_AD_Org org, final I_C_Location location, final String language, boolean isLocalAddress, String bPartnerBlock, String userBlock)
 	{
-		final I_C_Country country = location.getC_Country();
+		final String displaySequence =  getDisplaySequence(org, location.getC_Country(), isLocalAddress, language);
 
-		final String displaySequence = isLocalAddress ? country.getDisplaySequenceLocal() : country.getDisplaySequence();
 		String inStr = displaySequence;
 		final StringBuilder outStr = new StringBuilder();
 
