diff --git a/old_product.java b/new_product.java
index 810ba72..605375c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,11 @@
-public boolean isConnectionHandleAlive(Connection connection) {
+public boolean isConnectionHandleAlive(ConnectionHandle connection) {
 		Statement stmt = null;
 		boolean result = false; 
+		boolean logicallyClosed = connection.logicallyClosed;
 		try {
+			if (logicallyClosed){
+				connection.logicallyClosed = false; // avoid checks later on if it's marked as closed.
+			}
 			String testStatement = this.config.getConnectionTestStatement();
 			ResultSet rs = null;
 
@@ -23,6 +27,7 @@ public boolean isConnectionHandleAlive(Connection connection) {
 			// connection must be broken!
 			result = false;
 		} finally {
+			connection.logicallyClosed = logicallyClosed;
 			result = closeStatement(stmt, result);
 		}
 		return result;
