diff --git a/old_test.java b/new_test.java
index d566351..555f9fb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,7 +12,7 @@
                 .request(MediaType.APPLICATION_JSON_TYPE)
                 .delete();
 
-        assertThat(response.getStatus(), is(HttpURLConnection.HTTP_OK));
+        assertThat(response.getStatus(), is(HttpURLConnection.HTTP_NO_CONTENT));
 
         verify(mockVnetAdminService);
     }
\ No newline at end of file
