diff --git a/old_product.java b/new_product.java
index b62b834..7fe97e3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,6 +4,5 @@
     public void delete(@PathParam("subjectClassKey") String subjectClassKey) {
         NetworkConfigService service = get(NetworkConfigService.class);
         service.getSubjects(service.getSubjectFactory(subjectClassKey).subjectClass())
-                .forEach(subject -> service.getConfigs(subject)
-                        .forEach(config -> service.removeConfig(subject, config.getClass())));
+                .forEach(subject -> service.removeConfig(subject));
     }
\ No newline at end of file
