diff --git a/old_product.java b/new_product.java
index bca7140..5bda047 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ public Map<Instance, Long> degrees() {
         ComputerResult result = computer.compute(new DegreeVertexProgram(allTypes));
         result.graph().traversal().V().forEachRemaining(v -> {
             if (v.keys().contains(DegreeVertexProgram.DEGREE)) {
-                Instance instance = graph.getTransaction().getInstance(v.value(ITEM_IDENTIFIER.name()));
+                Instance instance = graph.getInstance(v.value(ITEM_IDENTIFIER.name()));
                 allDegrees.put(instance, v.value(DegreeVertexProgram.DEGREE));
             }
         });
