diff --git a/old_product.java b/new_product.java
index 10df768..3680962 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,5 @@
-JobInfo convert(WorkSpec workSpec) {
+JobInfo convert(WorkSpec workSpec, int jobId) {
         Constraints constraints = workSpec.constraints;
-        int jobId = mIdGenerator.nextJobSchedulerIdWithRange(
-                mConfiguration.getMinJobSchedulerID(),
-                mConfiguration.getMaxJobSchedulerID());
         // TODO(janclarin): Support newer required network types if unsupported by API version.
         int jobInfoNetworkType = convertNetworkType(constraints.getRequiredNetworkType());
         PersistableBundle extras = new PersistableBundle();
@@ -40,12 +37,11 @@ JobInfo convert(WorkSpec workSpec) {
             for (ContentUriTriggers.Trigger trigger : constraints.getContentUriTriggers()) {
                 builder.addTriggerContentUri(convertContentUriTrigger(trigger));
             }
-        } else {
-            // Jobs with Content Uri Triggers cannot be persisted
-            builder.setPersisted(true);
         }
 
-        // TODO(janclarin): Support requires[Battery|Storage]NotLow for versions older than 26.
+        // We don't want to persist these jobs because we reschedule these jobs on BOOT_COMPLETED.
+        // That way ForceStopRunnable correctly reschedules Jobs when necessary.
+        builder.setPersisted(false);
         if (Build.VERSION.SDK_INT >= 26) {
             builder.setRequiresBatteryNotLow(constraints.requiresBatteryNotLow());
             builder.setRequiresStorageNotLow(constraints.requiresStorageNotLow());
