diff --git a/old_test.java b/new_test.java
index 94728f2..95333ea 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
     @SmallTest
     public void testConvert_noConstraints_doesNotThrowException() {
-        mConverter.convert(new WorkSpec("id", TestWorker.class.getName()));
+        mConverter.convert(new WorkSpec("id", TestWorker.class.getName()), JOB_ID);
     }
\ No newline at end of file
