diff --git a/old_test.java b/new_test.java
index 40ccf80..14ae669 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,9 +15,14 @@
             }
 
             @Override
-            public ExecutorService getExecutor() {
+            public ExecutorService getExecutor(CorfuMsgType corfuMsgType) {
                 return executor;
             }
+
+            @Override
+            public List<ExecutorService> getExecutors() {
+                return Collections.singletonList(executor);
+            }
         };
 
         server.shutdown();
