diff --git a/old_test.java b/new_test.java
index 24934a7..643f741 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,7 +11,7 @@
             "</gml:coordinates></gml:LineString>)", "false");
     runError("geo:isClosed(<gml:Polygon><gml:outerBoundaryIs><gml:LinearRing>" +
             "<gml:coordinates>1,1 2,1 5,3 1,1</gml:coordinates></gml:LinearRing>" +
-            "</gml:outerBoundaryIs></gml:Polygon>)", GeoErrors.qname(4));
+            "</gml:outerBoundaryIs></gml:Polygon>)", GeoErrors.qname(3));
 
     runError("geo:isClosed(" +
             "<gml:LineString><gml:coordinates>1,1</gml:coordinates></gml:LineString>)",
@@ -20,5 +20,5 @@
     runError("geo:isClosed(text {'gml:Point'})", FUNCMP.qname());
     runError("geo:isClosed(" +
             "<gml:Point><gml:coordinates>2,1</gml:coordinates></gml:Point>)",
-            GeoErrors.qname(4));
+            GeoErrors.qname(3));
   }
\ No newline at end of file
