diff --git a/old_test.java b/new_test.java
index 7b41ac5..22bf823 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,8 +20,8 @@
         ComputeUtils.<Integer, Integer, Integer>initContext(
             ignite,
             upstreamCacheName,
-            (k, v) -> true,
             UpstreamTransformerBuilder.identity(),
+            (k, v) -> true,
             datasetCacheName,
             (env, upstream, upstreamSize) -> {
 
@@ -31,7 +31,9 @@
                 return e.getKey() + e.getValue();
             },
             TestUtils.testEnvBuilder(),
-            0
+            0,
+            0,
+            false
         );
 
         assertEquals(1, datasetCache.size());
