diff --git a/old_product.java b/new_product.java
index c2975b8..5d38b78 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,13 +7,17 @@ public static <K, V, C extends Serializable> void initContext(
         PartitionContextBuilder<K, V, C> ctxBuilder,
         LearningEnvironmentBuilder envBuilder,
         int retries,
-        int interval) {
+        int interval,
+        boolean isKeepBinary) {
         affinityCallWithRetries(ignite, Arrays.asList(datasetCacheName, upstreamCacheName), part -> {
             Ignite locIgnite = Ignition.localIgnite();
             LearningEnvironment env = envBuilder.buildForWorker(part);
 
             IgniteCache<K, V> locUpstreamCache = locIgnite.cache(upstreamCacheName);
 
+            if (isKeepBinary)
+                locUpstreamCache = locUpstreamCache.withKeepBinary();
+
             ScanQuery<K, V> qry = new ScanQuery<>();
             qry.setLocal(true);
             qry.setPartition(part);
