diff --git a/old_product.java b/new_product.java
index 97d2d91..deb8ea6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,9 @@ public static SimpleDateFormat getDateFormat(Locale locale) {
 		if (dateFormatCache.containsKey(locale))
 			return (SimpleDateFormat) dateFormatCache.get(locale).clone();
 		
-		SimpleDateFormat sdf = (SimpleDateFormat) DateFormat.getDateInstance(DateFormat.SHORT, locale);
+		// note that we are using the custom OpenmrsDateFormat class here which prevents erroneous parsing of 2-digit years
+		SimpleDateFormat sdf = new OpenmrsDateFormat(
+		        (SimpleDateFormat) DateFormat.getDateInstance(DateFormat.SHORT, locale), locale);
 		String pattern = sdf.toPattern();
 		
 		if (!pattern.contains("yyyy")) {
