diff --git a/old_test.java b/new_test.java
index df86109..929f14e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public final void set() {
-    ok(new Set(Options.CHOP, false));
-    ok(new Set(Options.CHOP, true));
+    ok(new Set(MainOptions.CHOP, false));
+    ok(new Set(MainOptions.CHOP, true));
     ok(new Set("chop", true));
     ok(new Set("runs", 1));
     no(new Set("runs", true));
