diff --git a/old_product.java b/new_product.java
index 2ce66c5..3695259 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-private void set(final AOptions options, final Option opt, final Object val) {
+private void set(final Options options, final Option opt, final Object val) {
    if(!options.sameAs(opt, val)) {
      final Set cmd = new Set(opt, val);
      cmd.run(context);
