diff --git a/old_product.java b/new_product.java
index d7c0e98..7872f17 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,7 +12,7 @@ public void add(Term term) {
         }
         output.writeVInt(suffix);
         output.writeBytes(term.bytes.bytes, term.bytes.offset + prefix, suffix);
-        lastTerm.bytes.copy(term.bytes);
+        lastTerm.bytes.copyBytes(term.bytes);
         lastTerm.field = term.field;
       } catch (IOException e) {
         throw new RuntimeException(e);
