diff --git a/old_test.java b/new_test.java
index 0d7b1d3..e1e666b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,21 +1,11 @@
 @Test
     public void createTableTest() throws SQLException {
-        try {
-            pstmt1 = con.prepareStatement("create table x (col1 int)");
-            pstmt2 = con.prepareStatement("drop table x");
+        try (PreparedStatement pstmt1 = con.prepareStatement("create table x (col1 int)");
+                PreparedStatement pstmt2 = con.prepareStatement("drop table x")) {
             pstmt1.execute();
             pstmt2.execute();
         } catch (SQLException e) {
             fail(TestResource.getResource("R_createDropTableFailed") + TestResource.getResource("R_errorMessage")
                     + e.getMessage());
         }
-
-        finally {
-            if (null != pstmt1) {
-                pstmt1.close();
-            }
-            if (null != pstmt2) {
-                pstmt2.close();
-            }
-        }
     }
\ No newline at end of file
