diff --git a/old_product.java b/new_product.java
index 4d33f93..e69c4a2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
-@SuppressWarnings("unchecked")
-    public <K, V, T> IgniteInternalFuture<GridCacheReturn> invokeAsync(
+public <K, V, T> IgniteInternalFuture<GridCacheReturn> invokeAsync(
         GridCacheContext cacheCtx,
         @Nullable AffinityTopologyVersion entryTopVer,
         @Nullable Map<? extends K, ? extends EntryProcessor<K, V, Object>> map,
