diff --git a/old_test.java b/new_test.java
index 2caa5f2..078547d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
                 .hexStringToByteArray("4B63051EABCD514D7CB6D1899F472B9F56856B01BDBC5B733FBB47269E7EBDC2"));
         context.setServerHandshakeTrafficSecret(ArrayConverter
                 .hexStringToByteArray("ACC9DB33EE0968FAE7E06DAA34D642B146092CE7F9C9CF47670C66A0A6CE1C8C"));
-        context.setConnectionEndType(ConnectionEndType.CLIENT);
+        context.setConnectionEnd(new ClientConnectionEnd());
         record.setProtocolMessageBytes(ArrayConverter
                 .hexStringToByteArray("1BB3293A919E0D66F145AE830488E8D89BE5EC16688229"));
         recordCipher = new RecordAEADCipher(context);
