diff --git a/old_test.java b/new_test.java
index 744221e..e629342 100644
--- a/old_test.java
+++ b/new_test.java
@@ -26,7 +26,7 @@
                     break;
                 } catch (KeyVaultErrorException e) {
                     ++failureCount;
-                    if (e.getBody().error().code().equals("Throttled")) {
+                    if (e.body().error().code().equals("Throttled")) {
                         System.out.println("Waiting to avoid throttling");
                         Thread.sleep(failureCount * 1500);
                         continue;
@@ -37,7 +37,7 @@
         }
 
         PagedList<CertificateItem> listResult = keyVaultClient.listCertificateVersions(getVaultUri(), certificateName, PAGELIST_MAX_CERTS);
-        Assert.assertTrue(PAGELIST_MAX_CERTS >= listResult.currentPage().getItems().size());
+        Assert.assertTrue(PAGELIST_MAX_CERTS >= listResult.currentPage().items().size());
 
         listResult = keyVaultClient.listCertificateVersions(getVaultUri(), certificateName);
 
