diff --git a/old_product.java b/new_product.java
index ebf9b21..8d77a69 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
-public static boolean optimizedParam( List<List<Point2D_F64>> observations ,
-										  List<Point2D_F64> grid ,
-										  Zhang99Parameters initial ,
-										  Zhang99Parameters found ,
-										  UnconstrainedLeastSquares optimizer )
+public boolean optimizedParam( List<List<Point2D_F64>> observations ,
+								   List<Point2D_F64> grid ,
+								   Zhang99Parameters initial ,
+								   Zhang99Parameters found ,
+								   UnconstrainedLeastSquares optimizer )
 	{
 		if( optimizer == null ) {
 //			optimizer = FactoryOptimization.leastSquaresTrustRegion(1,
@@ -25,8 +25,13 @@ public static boolean optimizedParam( List<List<Point2D_F64>> observations ,
 		optimizer.setFunction(func,null);
 		optimizer.initialize(model,1e-10,1e-25*observations.size());
 
-		if( !UtilOptimize.process(optimizer,500) ) {
-			return false;
+		for( int i = 0; i < 500; i++ ) {
+			if( optimizer.iterate() ) {
+				break;
+			} else {
+				if( i % 25 == 0 )
+					status("Progress "+(100*i/500.0)+"%");
+			}
 		}
 
 		double param[] = optimizer.getParameters();
