diff --git a/old_test.java b/new_test.java
index 97e09d1..ec52bb9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,8 @@
 
 		List<List<Point2D_F64>> observations = GenericCalibrationGrid.createObservations(initial,grid);
 
-		assertTrue(CalibrationPlanarGridZhang99.optimizedParam(observations, grid, initial, found,null));
+		CalibrationPlanarGridZhang99 alg = new CalibrationPlanarGridZhang99(config,true,2);
+		assertTrue(alg.optimizedParam(observations, grid, initial, found,null));
 
 		checkEquals(initial, found,initial);
 	}
\ No newline at end of file
