diff --git a/old_product.java b/new_product.java
index 5ef5c38..5518cd6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public default DataStreamGenerator mapVectors(IgniteFunction<Vector, Vector> f) {
         return new DataStreamGenerator() {
-            @Override public Stream<LabeledVector<Vector, Double>> labeled() {
+            @Override public Stream<LabeledVector<Double>> labeled() {
                 return DataStreamGenerator.this.labeled()
                     .map(v -> new LabeledVector<>(f.apply(v.features()), v.label()));
             }
