diff --git a/old_product.java b/new_product.java
index fe8091e..f87bba2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,8 @@
-boolean disableCompactionForBlobStore(BlobStore store) {
-      storesDisabledCompaction.add(store);
+boolean controlCompactionForBlobStore(BlobStore store, boolean enable) {
+      if (enable) {
+        storesDisabledCompaction.remove(store);
+      } else {
+        storesDisabledCompaction.add(store);
+      }
       return true;
     }
\ No newline at end of file
