diff --git a/old_test.java b/new_test.java
index d74f011..033c460 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
-  public void testDisableCompactionForBlobStore() throws Exception {
+  public void testControlCompactionForBlobStore() throws Exception {
     // without compaction enabled.
     compactionManager.enable();
     assertTrue("Disable compaction on BlobStore should be true when compaction executor is not instantiated",
-        compactionManager.disableCompactionForBlobStore(blobStore));
+        compactionManager.controlCompactionForBlobStore(blobStore, false));
     compactionManager.disable();
     compactionManager.awaitTermination();
     // with compaction enabled.
@@ -13,9 +13,13 @@
     compactionManager = new CompactionManager(MOUNT_PATH, config, Collections.singleton(blobStore), metrics, time);
     compactionManager.enable();
     assertTrue("Disable compaction on given BlobStore should succeed",
-        compactionManager.disableCompactionForBlobStore(blobStore));
+        compactionManager.controlCompactionForBlobStore(blobStore, false));
     assertFalse("BlobStore should not be scheduled after compaction is disabled on it",
         compactionManager.scheduleNextForCompaction(blobStore));
+    assertTrue("Enable compaction on given BlobStore should succeed",
+        compactionManager.controlCompactionForBlobStore(blobStore, true));
+    assertTrue("BlobStore should be scheduled after compaction is enabled on it",
+        compactionManager.scheduleNextForCompaction(blobStore));
     compactionManager.disable();
     compactionManager.awaitTermination();
   }
\ No newline at end of file
