diff --git a/old_product.java b/new_product.java
index 514c3f9..50e96b5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,3 @@
 public static Map<String, Object> toMap(String json){
-        LOGGER.debug("in json:{}", json);
-
-        Map<String, Object> map = new HashMap<String, Object>();
-
-        JSONObject jsonObject = toJSONObject(json);
-        @SuppressWarnings("unchecked")
-        Iterator<String> keys = jsonObject.keys();
-        while (keys.hasNext()){
-            String key = keys.next();
-            Object value = jsonObject.get(key);
-            LOGGER.debug("key:[{}], value:{}", key, value);
-            map.put(key, value);
-        }
-        return map;
+        return toMap(json, null);
     }
\ No newline at end of file
