diff --git a/old_product.java b/new_product.java
index f6deb0d..ebe025d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,10 @@
 @Override public double[] apply(K k, V v) {
         double[] res = basePreprocessor.apply(k, v);
 
-        assert res.length == min.length;
-        assert res.length == max.length;
+        double pNorm = Math.pow(foldMap(res, Functions.PLUS, Functions.pow(p), 0d), 1.0 / p);
 
         for (int i = 0; i < res.length; i++)
-            res[i] = (res[i] - min[i]) / (max[i] - min[i]);
+            res[i] /= pNorm;
 
         return res;
     }
\ No newline at end of file
