diff --git a/old_test.java b/new_test.java
index 6d60d5e..49c0403 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-	public void testDeletePoints() {
-		String dbName = "deletepoints-unittest-" + System.currentTimeMillis();
+	public void testDeleteSeries() {
+		String dbName = "deleteseries-unittest-" + System.currentTimeMillis();
 		this.influxDB.createDatabase(dbName);
 
 		Serie serie = new Serie.Builder("testSeries")
@@ -13,10 +13,13 @@
 		Assert.assertNotNull(result);
 		Assert.assertEquals(result.size(), 1);
 
-		this.influxDB.deletePoints(dbName, "testSeries");
-		result = this.influxDB.query(dbName, "select value1 from testSeries", TimeUnit.MILLISECONDS);
-		Assert.assertNotNull(result);
-		Assert.assertEquals(result.size(), 0);
+		result = this.influxDB.query(dbName, "list series", TimeUnit.MILLISECONDS);
+		Assert.assertTrue(result.get(0).getRows().get(0).containsKey("name"));
+		Assert.assertEquals(result.get(0).getRows().get(0).get("name"), "testSeries");
+
+		this.influxDB.deleteSeries(dbName, "testSeries");
+		result = this.influxDB.query(dbName, "list series", TimeUnit.MILLISECONDS);
+		Assert.assertEquals(result.get(0).getRows().size(), 0);
 
 		this.influxDB.deleteDatabase(dbName);
 	}
\ No newline at end of file
