diff --git a/old_product.java b/new_product.java
index 15394c6..04b0c2a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,7 +13,7 @@ public Response listLogFiles(String user, Integer port, String topologyId, Strin
                         if (topoDirFiles != null) {
                             for (File portDir : topoDirFiles) {
                                 if (portDir.getName().equals(port.toString())) {
-                                    fileResults.addAll(DirectoryCleaner.getFilesForDir(portDir));
+                                    fileResults.addAll(directoryCleaner.getFilesForDir(portDir));
                                 }
                             }
                         }
@@ -29,7 +29,7 @@ public Response listLogFiles(String user, Integer port, String topologyId, Strin
                     File[] topoDirFiles = topoDir.listFiles();
                     if (topoDirFiles != null) {
                         for (File portDir : topoDirFiles) {
-                            fileResults.addAll(DirectoryCleaner.getFilesForDir(portDir));
+                            fileResults.addAll(directoryCleaner.getFilesForDir(portDir));
                         }
                     }
                 }
@@ -37,7 +37,7 @@ public Response listLogFiles(String user, Integer port, String topologyId, Strin
             } else {
                 File portDir = ConfigUtils.getWorkerDirFromRoot(logRoot, topologyId, port);
                 if (portDir.exists()) {
-                    fileResults = DirectoryCleaner.getFilesForDir(portDir);
+                    fileResults = directoryCleaner.getFilesForDir(portDir);
                 }
             }
         }
