diff --git a/old_test.java b/new_test.java
index fb8406b..6907565 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
     public void test_lastDayOfMonth_serialization() throws IOException, ClassNotFoundException {
-        DateTimeAdjuster lastDayOfMonth = DateTimeAdjusters.lastDayOfMonth();
+        WithAdjuster lastDayOfMonth = DateTimeAdjusters.lastDayOfMonth();
         assertTrue(lastDayOfMonth instanceof Serializable);
         
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
