diff --git a/old_test.java b/new_test.java
index 31d04f8..ceebcc5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,9 +3,9 @@
 
 		PlanarCalibrationTarget config = GenericCalibrationGrid.createStandardConfig();
 		List<Point2D_F64> grid = config.points;
-		ParametersZhang99 initial = GenericCalibrationGrid.createStandardParam(true,2,3,rand);
-		ParametersZhang99 expected = initial.copy();
-		ParametersZhang99 found = new ParametersZhang99(2,3);
+		Zhang99Parameters initial = GenericCalibrationGrid.createStandardParam(true,2,3,rand);
+		Zhang99Parameters expected = initial.copy();
+		Zhang99Parameters found = new Zhang99Parameters(true,2,3);
 
 		List<List<Point2D_F64>> observations = GenericCalibrationGrid.createObservations(initial,grid);
 
@@ -20,7 +20,7 @@
 			initial.distortion[i] = rand.nextGaussian()*expected.distortion[i]*0.1;
 		}
 
-		assertTrue(CalibrationPlanarGridZhang99.optimizedParam(observations, grid, true, initial, found,null));
+		assertTrue(CalibrationPlanarGridZhang99.optimizedParam(observations, grid, initial, found,null));
 
 		checkEquals(expected, found, initial);
 	}
\ No newline at end of file
