diff --git a/old_product.java b/new_product.java
index 6f9a385..184622a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,16 @@
 @Override
     public boolean first() throws SQLException {
         checkClosed();
-        if (rowCount == 0)
+        if (rowCount == 0) {
             return false;
+        }
+        if (cursorPosition != Position.middle) {
+            cursorPosition = Position.middle;
+        }
         try {
-            if (table.getActiveRowIndex() != 0)
+            if (table.getActiveRowIndex() != 0){
                 table.resetRowPosition();
+            }
             return table.advanceToRow(0);
         } catch (Exception x) {
             throw SQLError.get(x);
