diff --git a/old_test.java b/new_test.java
index be01bbf..ca18f80 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,7 @@
 @Test
     public void testFirst() throws Exception {
 
+        assertEquals(hsql3RowRS.absolute(2), volt3RowRS.absolute(2));
         assertTrue(hsql3RowRS.first() == volt3RowRS.first());
         assertTrue(hsql3RowRS.getRow() == volt3RowRS.getRow());
         assertEquals(hsql3RowRS.getInt(1), volt3RowRS.getInt(1));
