diff --git a/old_test.java b/new_test.java
index da31605..112195a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,14 +2,16 @@
 	public void set()
 	{
 		PageParameters parameters = new PageParameters();
-		parameters.add("named1", "value1").add("named2", "value2");
+		parameters
+				.add("named1", "value1", INamedParameters.Type.MANUAL)
+				.add("named2", "value2", INamedParameters.Type.MANUAL);
 
 		assertEquals(0, parameters.getPosition("named1"));
 		assertEquals(1, parameters.getPosition("named2"));
 
 		// overwrite it
-		parameters.set("named1", "newValue");
-		parameters.set("named3", "value3");
+		parameters.set("named1", "newValue", INamedParameters.Type.MANUAL);
+		parameters.set("named3", "value3", INamedParameters.Type.MANUAL);
 		assertEquals(0, parameters.getPosition("named1"));
 		assertEquals("newValue", parameters.get("named1").toString());
 		assertEquals(1, parameters.getPosition("named2"));
