diff --git a/old_test.java b/new_test.java
index c53bd6a..5167685 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
     public void test_plus_Adjuster_positiveHours() {
-        PlusAdjuster period = MockSimplePeriod.of(7, ChronoUnit.HOURS);
+        TemporalAdder period = MockSimplePeriod.of(7, ChronoUnit.HOURS);
         LocalTime t = TEST_12_30_40_987654321.plus(period);
         assertEquals(t, LocalTime.of(19, 30, 40, 987654321));
     }
\ No newline at end of file
