diff --git a/old_test.java b/new_test.java
index b06580f..5b72a54 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,14 @@
 @Test
     public void testAlwaysAddChannel() {
         MutableSparseVector simple = simpleVector();
-        simple.addChannel(fooSymbol);
-        simple.channel(fooSymbol).set(3, 4.5);
-        simple.getOrAddChannel(fooSymbol);
-        simple.getOrAddChannel(barSymbol);
-        assert(simple.channel(barSymbol).isEmpty());
-        simple.channel(barSymbol).set(3, 33);
-        assertThat(simple.channel(fooSymbol).get(3), closeTo(4.5));
-        assertThat(simple.channel(barSymbol).get(3, -1.0), closeTo(33));
-        simple.channel(fooSymbol).unset(8);
-        assertThat(simple.channel(fooSymbol).get(8, 45.0), closeTo(45.0));
+        simple.addChannelVector(fooSymbol);
+        simple.getChannelVector(fooSymbol).set(3, 4.5);
+        simple.getOrAddChannelVector(fooSymbol);
+        simple.getOrAddChannelVector(barSymbol);
+        assert(simple.getChannelVector(barSymbol).isEmpty());
+        simple.getChannelVector(barSymbol).set(3, 33);
+        assertThat(simple.getChannelVector(fooSymbol).get(3), closeTo(4.5));
+        assertThat(simple.getChannelVector(barSymbol).get(3, -1.0), closeTo(33));
+        simple.getChannelVector(fooSymbol).unset(8);
+        assertThat(simple.getChannelVector(fooSymbol).get(8, 45.0), closeTo(45.0));
     }
\ No newline at end of file
