diff --git a/old_test.java b/new_test.java
index 99aafa5..468cc5f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,5 +4,5 @@
         ZonedDateTime base = ZonedDateTime.of(ldt, ZONE_0100);
         LocalDateTime dt = LocalDateTime.of(2008, 6, 30, 11, 31, 0);
         ZonedDateTime test = base.withDateTime(dt);
-        assertEquals(test.toLocalDateTime(), dt);
+        assertEquals(test.getDateTime(), dt);
     }
\ No newline at end of file
