diff --git a/old_product.java b/new_product.java
index 8568b2a..9a093dd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public <R extends Chronology<R>> ChronoZonedDateTime<R> withDateTime(ChronoDateTime<R> newDateTime, ZoneResolver resolver) {
         Objects.requireNonNull(newDateTime, "ChronoDateTime must not be null");
         Objects.requireNonNull(resolver, "ZoneResolver must not be null");
-        if (dateTime.toDateTime().equals(newDateTime)) {
+        if (dateTime.getDateTime().equals(newDateTime)) {
             return (ChronoZonedDateTime<R>)this;
         } else {
             return resolve(newDateTime, zone, this.dateTime, resolver);
