diff --git a/old_test.java b/new_test.java
index ed66af8..0ce8c99 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
     try {
       v1 = parse_test_file("smalldata/junit/mixcat_train.csv");
       v2 = parse_test_file("smalldata/junit/mixcat_test.csv");
-      CategoricalWrappedVec vv = v2.vecs()[0].adaptTo(v1.vecs()[0].domain());
+      CategoricalWrappedVec vv = (CategoricalWrappedVec) v2.vecs()[0].adaptTo(v1.vecs()[0].domain());
       Assert.assertArrayEquals("Mapping differs",new int[]{0,1,3},vv._map);
       Assert.assertArrayEquals("Mapping differs",new String[]{"A","B","C","D"},vv.domain());
       vv.remove();
