diff --git a/old_test.java b/new_test.java
index 7ab736c..ce77775 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testUpdateJobStatus() throws GenieException {
+    public void testUpdateJobStatusForStatusRunning() throws GenieException {
         final String id = UUID.randomUUID().toString();
         final JobEntity jobEntity = new JobEntity();
         final ArgumentCaptor<JobEntity> argument = ArgumentCaptor.forClass(JobEntity.class);
@@ -12,4 +12,7 @@
         Assert.assertEquals(JobStatus.RUNNING, argument.getValue().getStatus());
         Assert.assertEquals(JOB_1_STATUS_MSG, argument.getValue().getStatusMsg());
         Assert.assertEquals(new Date(0), argument.getValue().getFinished());
+
+        // Started should not be null as the status is being set to STARTED
+        Assert.assertNotNull(argument.getValue().getStarted());
     }
\ No newline at end of file
