diff --git a/old_product.java b/new_product.java
index b71d48e..bf64899 100644
--- a/old_product.java
+++ b/new_product.java
@@ -18,11 +18,15 @@
         jobEntity.setStatus(jobStatus);
         jobEntity.setStatusMsg(statusMsg);
 
-        // If the status is either failed, killed or succeeded then set the finish time of the job as well
-        if (jobStatus.equals(JobStatus.KILLED)
+        if (jobStatus.equals(JobStatus.RUNNING)) {
+            // Status being changed to running so set start date.
+            jobEntity.setStarted(new Date());
+        } else if (jobEntity.getStarted() != null && (jobStatus.equals(JobStatus.KILLED)
             || jobStatus.equals(JobStatus.FAILED)
-            || jobStatus.equals(JobStatus.SUCCEEDED)) {
+            || jobStatus.equals(JobStatus.SUCCEEDED))) {
 
+            // Since start date is set the job was running previously and now has finished
+            // with status killed, failed or succeeded. So we set the job finish time.
             jobEntity.setFinished(new Date());
         }
         this.jobRepo.save(jobEntity);
