diff --git a/old_product.java b/new_product.java
index 07b8dda..9c0a01c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public static ImageUInt8 threshold( ImageSInt16 input , ImageUInt8 output ,
-										int threshold , boolean down )
+public static GrayU8 threshold(GrayS16 input , GrayU8 output ,
+								   int threshold , boolean down )
 	{
-		output = InputSanityCheck.checkDeclare(input,output,ImageUInt8.class);
+		output = InputSanityCheck.checkDeclare(input,output,GrayU8.class);
 
 		if( down ) {
 			for( int y = 0; y < input.height; y++ ) {
