diff --git a/old_product.java b/new_product.java
index 0b8f8bc..e4ab29e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
-public boolean addAsEqual(T existingObj, T newObj) {
+public boolean addAsEqual(final T existingObj, final T newObj) {
         checkLocked();
-        Integer position = map.get(existingObj);
+        final Integer position = map.get(existingObj);
         if (position == null) {
             throw new IllegalArgumentException(existingObj + " not known to " + this);
         }
-        Integer result = map.put(newObj, position);
+        final Integer result = map.put(newObj, position);
         return result == null;
     }
\ No newline at end of file
