diff --git a/old_product.java b/new_product.java
index a6ac295..c6245c9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public void setDatasetOverlays(final Dataset ds, final ImagePlus imp) {
+public void setDisplayOverlays(final Display display, final ImagePlus imp) {
 		final OverlayManager overlayManager = ImageJ.get(OverlayManager.class);
-		ShapeRoi oldROI = createROI(overlayManager.getOverlays(ds));
+		ShapeRoi oldROI = createROI(overlayManager.getOverlays(display));
 		if (oldROI != null) {
 			float [] oldPath = oldROI.getShapeAsArray();
 			Roi newROI = imp.getRoi();
@@ -18,10 +18,10 @@ public void setDatasetOverlays(final Dataset ds, final ImagePlus imp) {
 				}
 			}
 		}
-		final List<Overlay> overlaysToRemove = overlayManager.getOverlays(ds);
+		final List<Overlay> overlaysToRemove = overlayManager.getOverlays(display);
 		for (Overlay overlay:overlaysToRemove) {
-			overlayManager.removeOverlay(ds, overlay);
+			overlayManager.removeOverlay(display, overlay);
 		}
 		final List<Overlay> overlays = getOverlays(imp);
-		overlayManager.setOverlays(ds, overlays);
+		overlayManager.addOverlays(display, overlays);
 	}
\ No newline at end of file
