diff --git a/old_product.java b/new_product.java
index dab9bc5..84c220a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,15 @@
             final AsyncResponse asyncResponse) {
         long traceId = LoggerHelpers.traceEnter(log, "deleteStream");
 
+        try {
+            authenticate(scopeName + "/" + streamName, READ_UPDATE);
+        } catch (AuthenticationException e) {
+            log.warn("Delete stream for {} failed due to authentication failure.", streamName);
+            asyncResponse.resume(Response.status(Status.UNAUTHORIZED).build());
+            LoggerHelpers.traceLeave(log, "deleteStream", traceId);
+            return;
+        }
+
         controllerService.deleteStream(scopeName, streamName).thenApply(deleteStreamStatus -> {
           if (deleteStreamStatus.getStatus() == DeleteStreamStatus.Status.SUCCESS) {
               log.info("Successfully deleted stream: {}", streamName);
