diff --git a/old_product.java b/new_product.java
index a07d97b..14f33c4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-protected KnowledgesEntity insertKnowledge(String title, LoginedUser loginedUser) throws Exception {
+protected KnowledgesEntity insertKnowledge(String title, AccessUser loginedUser) throws Exception {
         int publicFlag = KnowledgeLogic.PUBLIC_FLAG_PUBLIC;
         int typeId = TemplateLogic.TYPE_ID_KNOWLEDGE;
         return insertKnowledge(title, loginedUser, typeId, publicFlag);
