diff --git a/old_product.java b/new_product.java
index c59bf49..a55ae39 100644
--- a/old_product.java
+++ b/new_product.java
@@ -22,7 +22,7 @@ public void handle(Order order, User creator, Date dateCreated, String other) {
 			}
 			
 			String nextOrderNumber = Context.getAdministrationService().getGlobalProperty(
-			    OpenmrsConstants.GP_ORDER_ENTRY_ORDER_NUMBER_PREFIX, OpenmrsConstants.ORDER_NUMBER_DEFAULT_PERFIX)
+			    OpenmrsConstants.GP_ORDER_ENTRY_ORDER_NUMBER_PREFIX, OpenmrsConstants.ORDER_NUMBER_DEFAULT_PREFIX)
 			        + orderNumber.toString();
 			order.setOrderNumber(nextOrderNumber);
 		}
