diff --git a/old_test.java b/new_test.java
index 5344ff0..66257bb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public void getStatus() throws Exception {
     AlluxioURI file = new AlluxioURI("/file");
     URIStatus status = new URIStatus(new FileInfo());
-    GetStatusOptions getStatusOptions = GetStatusOptions.defaults();
+    GetStatusPOptions getStatusOptions = GetStatusPOptions.getDefaultInstance();
     when(mFileSystemMasterClient.getStatus(file, getStatusOptions)).thenReturn(status);
     assertSame(status, mFileSystem.getStatus(file, getStatusOptions));
     verify(mFileSystemMasterClient).getStatus(file, getStatusOptions);
