diff --git a/old_product.java b/new_product.java
index 76261a2..14b9bfb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 @POST
   @Path(PATH_PARAM + GET_STATUS)
-  @ReturnType("alluxio.client.file.URIStatus")
+  @ApiOperation(value = "Get the file status of the path",
+      response = alluxio.client.file.URIStatus.class)
   public Response getStatus(@PathParam("path") final String path, final GetStatusOptions options) {
     return RestUtils.call(new RestUtils.RestCallable<URIStatus>() {
       @Override
