diff --git a/old_test.java b/new_test.java
index 4a1960c..5479236 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,5 +4,5 @@
     DeviceConfig.applyRules(configuration, displayMetrics, apiLevel);
 
     assertThat(asQualifierString())
-        .isEqualTo("iw-ldrtl-sw320dp-w320dp-h470dp-normal-notlong-notround-" + optsForO + "port-notnight-mdpi-finger-keyssoft-nokeys-navhidden-nonav");
+        .isEqualTo("iw-ldrtl-sw320dp-w320dp-h470dp-normal-notlong-notround-port-notnight-mdpi-finger-keyssoft-nokeys-navhidden-nonav");
   }
\ No newline at end of file
