diff --git a/old_product.java b/new_product.java
index 9ad2f09..2ad0c0f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -80,10 +80,6 @@ static void applyRules(Configuration configuration, DisplayMetrics displayMetric
       swapXY(configuration);
     }
 
-    if (getUiModeType(configuration) == Configuration.UI_MODE_TYPE_UNDEFINED) {
-      setUiModeType(configuration, Configuration.UI_MODE_TYPE_NORMAL);
-    }
-
     if (getUiModeNight(configuration) == Configuration.UI_MODE_NIGHT_UNDEFINED) {
       setUiModeNight(configuration, Configuration.UI_MODE_NIGHT_NO);
     }
@@ -117,14 +113,4 @@ static void applyRules(Configuration configuration, DisplayMetrics displayMetric
     if (configuration.navigation == Configuration.NAVIGATION_UNDEFINED) {
       configuration.navigation = Configuration.NAVIGATION_NONAV;
     }
-
-    if (apiLevel >= VERSION_CODES.O) {
-      if (getColorModeGamut(configuration) == Configuration.COLOR_MODE_WIDE_COLOR_GAMUT_UNDEFINED) {
-        setColorModeGamut(configuration, Configuration.COLOR_MODE_WIDE_COLOR_GAMUT_NO);
-      }
-
-      if (getColorModeHdr(configuration) == Configuration.COLOR_MODE_HDR_UNDEFINED) {
-        setColorModeHdr(configuration, Configuration.COLOR_MODE_HDR_NO);
-      }
-    }
   }
\ No newline at end of file
