diff --git a/old_product.java b/new_product.java
index 98d70ec..cbe3ba1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public List<TenantUserDto> getTenants() throws Exception {
-        ParameterizedTypeReference<List<TenantUserDto>> typeRef = new ParameterizedTypeReference<List<TenantUserDto>>() {
-        };
-        ResponseEntity<List<TenantUserDto>> entity = restTemplate.exchange(restTemplate.getUrl() + "tenants", HttpMethod.GET, null, typeRef);
-        return entity.getBody();
-    }
\ No newline at end of file
+public List<TenantDto> getTenants() throws Exception {
+    ResponseEntity<List<TenantDto>> entity = restTemplate.exchange(
+        restTemplate.getUrl() + "tenants",
+        HttpMethod.GET, null, new ParameterizedTypeReference<List<TenantDto>>() {});
+    return entity.getBody();
+  }
\ No newline at end of file
