diff --git a/old_test.java b/new_test.java
index db1cb63..79503aa 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,6 @@
 @Test
   public void testRun() throws BaseXException {
-    // test wrong arguments
-    args("util:run", String.class);
-
-    // dynamically run query files
-    query("util:run('etc/xml/input.xq')", "XML");
-    error("util:run('etc/xml/xxx.xq')", "FODC0002");
+    final String fun = check(FunDef.RUN, String.class);
+    query(fun + "('etc/xml/input.xq')", "XML");
+    error(fun + "('etc/xml/xxx.xq')", Err.UNDOC);
   }
\ No newline at end of file
