diff --git a/old_product.java b/new_product.java
index 919b73f..04226ed 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,10 @@
 public double norm() {
-        if (norm == null) {
-            double ssq = 0;
-            DoubleIterator iter = values().iterator();
-            while (iter.hasNext()) {
-                double v = iter.nextDouble();
-                ssq += v * v;
-            }
-            norm = Math.sqrt(ssq);
+        double ssq = 0;
+        DoubleIterator iter = values().iterator();
+        while (iter.hasNext()) {
+            double v = iter.nextDouble();
+            ssq += v * v;
         }
-        return norm;
+        double result = Math.sqrt(ssq);
+        return result;
     }
\ No newline at end of file
