diff --git a/old_test.java b/new_test.java
index f5dc918..1b5ccdb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,5 +2,7 @@
     public void testNorm() {
         assertThat(emptyVector().norm(), closeTo(0));
         assertThat(singleton().norm(), closeTo(Math.PI));
-        assertThat(simpleVector().norm(), closeTo(4.301162634));
+        SparseVector sv = simpleVector();
+        assertThat(sv.norm(), closeTo(4.301162634));
+        assertThat(sv.norm(), closeTo(4.301162634));  // doubled, to check caching
     }
\ No newline at end of file
