diff --git a/old_product.java b/new_product.java
index 03e230b..3f0c53f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
-@Override
-    public Token createToken(String tokenType, Map<String, Object> tokenData) throws TokenCreationFailedException {
+public Token createToken(String tokenType, Map<String, Object> tokenData) throws TokenCreationFailedException {
         try {
             TokenService tokenService = getTokenServiceForType(tokenType);
             return tokenService.createToken(tokenData);
