diff --git a/old_test.java b/new_test.java
index af6b4fc..4b829b4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,6 +7,6 @@
 		userDTO.setName(user.getName());
 		userDTO.setEmail(user.getEmail());
 
-		IdResult result = accountWebServiceClient.createUser(userDTO);
+		IdResponse result = accountWebServiceClient.createUser(userDTO);
 		assertNotNull(result.getId());
 	}
\ No newline at end of file
