diff --git a/old_product.java b/new_product.java
index bf122e7..21b2cd6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
-public ServiceCall<VisualClassifier> createClassifier(CreateClassifierOptions options) {
+public ServiceCall<VisualClassifier> createClassifier(ClassifierOptions options) {
     Validator.notNull(options, " options cannot be null");
+    validateClassifierOptions(options);
 
     Builder bodyBuilder = new MultipartBody.Builder().setType(MultipartBody.FORM);
     bodyBuilder.addFormDataPart(PARAM_NAME, options.classifierName());
