diff --git a/old_test.java b/new_test.java
index fec3c39..1a6d10c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
     // execute request
     File images = new File(IMAGE_FILE);
     String class1 = "class1"; 
-    CreateClassifierOptions options = new CreateClassifierOptions.Builder()
+    ClassifierOptions options = new ClassifierOptions.Builder()
         .classifierName(class1)
         .addClass(class1, images)
         .negativeExamples(images)
