diff --git a/old_test.java b/new_test.java
index bfbae47..7d8586b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
     public void testCimDateTimeToDate() {
-        assertEquals(ParseUtil.cimDateTimeToDate("20160513072950.782000-420").getTime(), 1463149790782L);
+        assertEquals(1463149790782L, ParseUtil.cimDateTimeToMillis("20160513072950.782000-420"));
     }
\ No newline at end of file
