diff --git a/old_product.java b/new_product.java
index 512fd99..5df6e88 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,15 @@
-public static Date cimDateTimeToDate(String cimDate) {
-        // Keep first 18 digits; ignore next 3
-        // Keep + or - sign of timezone
-        // Parse last 3 digits from minutes to HH:mm
-        int tzMinutes = Integer.parseInt(cimDate.substring(22));
-        return cimDateFormat.parse(String.format("%s%c%02d%02d", cimDate.substring(0, 18), cimDate.charAt(21),
-                tzMinutes / 60, tzMinutes % 60), new ParsePosition(0));
+public static long cimDateTimeToMillis(String cimDate) {
+        // Keep first 22 characters: digits, decimal, and + or - sign of
+        // time zone. Parse last 3 digits from minutes to HH:mm
+        try {
+            int tzInMinutes = Integer.parseInt(cimDate.substring(22));
+            LocalTime offsetAsLocalTime = LocalTime.MIN.plusMinutes(tzInMinutes);
+            OffsetDateTime dateTime = OffsetDateTime.parse(
+                    cimDate.substring(0, 22) + offsetAsLocalTime.format(DateTimeFormatter.ISO_LOCAL_TIME), CIM_FORMAT);
+            return dateTime.toInstant().toEpochMilli();
+        } catch (IndexOutOfBoundsException // if cimDate not 22+ chars
+                | NumberFormatException // if TZ minutes doesn't parse
+                | DateTimeParseException e) {
+            return 0L;
+        }
     }
\ No newline at end of file
