diff --git a/old_test.java b/new_test.java
index 3b43bcd..a8f6c48 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
-@Test
-    public void test_atMonthDay() {
-        Year test = Year.of(2008);
-        assertEquals(test.atMonthDay(MonthDay.of(6, 30)), LocalDate.of(2008, 6, 30));
+@Test(dataProvider="atMonthDay")
+    public void test_atMonthDay(Year year, MonthDay monthDay, LocalDate expected) {
+        assertEquals(year.atMonthDay(monthDay), expected);
     }
\ No newline at end of file
