diff --git a/old_product.java b/new_product.java
index d7a16a4..a22acd5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-List<SchemaDto> getConfigurationSchemaVersionsByApplicationId(String applicationId) throws ControlServiceException;
\ No newline at end of file
+List<VersionDto> getConfigurationSchemaVersionsByApplicationId(String applicationId) throws ControlServiceException;
\ No newline at end of file
