diff --git a/old_test.java b/new_test.java
index 6ddd2fd..390958b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,8 @@
 @Test
     public void getParameterizedClassArgumentsTest() {
-
         ReflectionHelper.DeclaringClassInterfacePair dcip = ReflectionHelper.getClass(TestNoInterface.class, I.class);
         Class[] arguments = ReflectionHelper.getParameterizedClassArguments(dcip);
-        Class aClass = arguments[0];
+        final Class aClass = arguments[0];
 
         dcip = ReflectionHelper.getClass(TestInterface.class, I.class);
         arguments = ReflectionHelper.getParameterizedClassArguments(dcip);
