diff --git a/old_product.java b/new_product.java
index ca8fa86..4322977 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public synchronized String lockBlock(long blockId) throws IOException {
     mustConnect();
 
     try {
-      return mClient.lockBlock(blockId, mMasterClient.getUserId());
+      return mClient.lockBlock(blockId, mUserId);
     } catch (FileDoesNotExistException e) {
       return null;
     } catch (TException e) {
