diff --git a/old_test.java b/new_test.java
index bafece3..60d9c51 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,14 @@
 @Test
   public void lockBlockTest() throws Exception {
-    final int fileId = mTfs.createFile(new TachyonURI("/testFile"));
     final int blockSize = (int) WORKER_CAPACITY_BYTES / 2;
-    final long blockId = mTfs.getBlockId(fileId, 0);
 
-    OutStream out = mTfs.getFile(fileId).getOutStream(WriteType.MUST_CACHE);
+    ClientOptions options = new ClientOptions.Builder(new TachyonConf()).setBlockSize(blockSize)
+        .setTachyonStoreType(TachyonStorageType.STORE).build();
+    FileOutStream out = mTfs.getOutStream(new TachyonURI("/testFile"), options);
+    TachyonFile file = mTfs.open(new TachyonURI("/testFile"));
+
+    final long blockId = BlockId.createBlockId(BlockId.getContainerId(file.getFileId()), 0);
+
     out.write(BufferUtils.getIncreasingByteArray(blockSize));
     out.close();
 
