diff --git a/old_product.java b/new_product.java
index c93069e..1d1ff22 100644
--- a/old_product.java
+++ b/new_product.java
@@ -21,11 +21,10 @@
                 ScaleMetadata scaleMetadata = metadataIterator.next();
                 if (scaleMetadata.getTimestamp() >= from && scaleMetadata.getTimestamp() <= to) {
                     finalScaleMetadataList.add(scaleMetadata);
-                } else if (scaleMetadata.getTimestamp() < from) {
+                } else if ((scaleMetadata.getTimestamp() < from) &&
+                            !(referenceEvent != null && referenceEvent.getTimestamp() > scaleMetadata.getTimestamp())) {
                     // This check is required to store a reference event i.e. an event before the 'from' datetime
                     referenceEvent = scaleMetadata;
-                } else {
-                    break;
                 }
             }
 
