diff --git a/old_test.java b/new_test.java
index 9eb2238..4a267b3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,7 @@
         String path1 = "/a/b/test_mk_dir";
         BinaryRawWriterEx message = createMessage(reqId, ModelStorateThinClientProcessor.PROCESSOR_ID, ModelStorateThinClientProcessor.Method.MKDIR);
         message.writeString(path1);
+        message.writeBoolean(false);
         ClientCustomQueryRequest req = new ClientCustomQueryRequest(toReader(message));
         ClientResponse resp = req.process(connCtx);
         BinaryRawWriterEx out = createWriter();
