diff --git a/old_product.java b/new_product.java
index fc57059..7af38c2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,12 @@
 private ClientResponse mkdir(long reqId, BinaryRawReader reader) {
         String path = reader.readString();
+        boolean onlyIfNotExists = reader.readBoolean();
 
         return modelStorage.lockPaths(() -> {
-            if (modelStorage.exists(path))
+            if (onlyIfNotExists && modelStorage.exists(path))
                 return error(reqId, "Directory already exists [path=" + path + "]");
 
-            modelStorage.mkdir(path);
+            modelStorage.mkdir(path, onlyIfNotExists);
             return new ClientResponse(reqId);
         }, path);
     }
\ No newline at end of file
