diff --git a/old_product.java b/new_product.java
index 50db495..f11d53d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,8 +2,18 @@
     public List<ChangeLogEntry> getChangeLog( List<MvccEntity> mvccEntities, UUID minVersion ) {
 
         Map<String, ChangeLogEntry> writeMap = new HashMap<String, ChangeLogEntry>();
-
         List<ChangeLogEntry> changeLog = new ArrayList<ChangeLogEntry>();
+        Entity keeper = null;
+
+        for (MvccEntity mvccEntity : mvccEntities) {
+
+            Entity entity = mvccEntity.getEntity().get();
+            int compare = mvccEntity.getVersion().compareTo(minVersion);
+
+            if (compare == 0) {
+                keeper = entity;
+            }
+        }
 
         for (MvccEntity mvccEntity : mvccEntities) {
 
@@ -13,10 +23,17 @@
             if (compare == -1) { // less than minVersion
 
                 for (Field field : entity.getFields()) {
-                    String key = field.getName();
-                    ChangeLogEntry cle = new ChangeLogEntry( entity.getId(), null, 
-                                                 ChangeLogEntry.ChangeType.PROPERTY_DELETE, field );
-                    changeLog.add( 0, cle );
+
+                    // only delete field if it is not in the keeper
+                    Field keeperField = keeper.getField( field.getName() );
+                    if (       keeperField == null 
+                            || keeperField.getValue() == null
+                            || !keeperField.getValue().equals( field.getValue() )) {
+
+                        ChangeLogEntry cle = new ChangeLogEntry( entity.getId(), null, 
+                            ChangeLogEntry.ChangeType.PROPERTY_DELETE, field );
+                        changeLog.add( cle );
+                    }
                 }
 
             } else { // greater than or equal to minVersion
@@ -27,9 +44,9 @@
                     ChangeLogEntry cle = writeMap.get( key );
                     if ( cle == null ) {
                         cle = new ChangeLogEntry( entity.getId(), mvccEntity.getVersion(), 
-                                      ChangeLogEntry.ChangeType.PROPERTY_WRITE, field );
+                            ChangeLogEntry.ChangeType.PROPERTY_WRITE, field );
                         writeMap.put( key, cle );
-                        changeLog.add( 0, cle );
+                        changeLog.add( cle );
                     } else {
                         cle.addVersion( mvccEntity.getVersion() );
                     } 
