diff --git a/old_test.java b/new_test.java
index 346feff..3e7c032 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,14 @@
 @Test
-    public void testGetChangeLog2() throws ConnectionException {
+    public void testBasicOperation() throws ConnectionException {
 
-        LOG.info( "getChangeLog2" );
+        LOG.info( "getChangeLog1" );
 
         // create an entity and make a series of changes to it so that versions get created
         CollectionScope context = new CollectionScopeImpl(
                 new SimpleId( "organization" ), new SimpleId( "test" ), "test" );
 
+        // Todd's example:
+        //
         // V1 : { "name" : "name1" , "count": 1}
         // V2:  { "name" : "name2" , "count": 2, "nickname" : "buddy"}
         // V3:  { "name" : "name3" , "count": 2}
@@ -43,7 +45,7 @@
             for (ChangeLogEntry cle : result) {
                 LOG.info( cle.toString() );
             }
-            assertEquals(7, result.size() );
+            assertEquals(6, result.size() );
         }
        
         LOG.info("===================");
