diff --git a/old_product.java b/new_product.java
index a240226..5622237 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public static boolean getBoolean(String key) throws InvalidSettingException {
-        return Boolean.parseBoolean(Settings.getString(key));
+public boolean getBoolean(String key) throws InvalidSettingException {
+        return Boolean.parseBoolean(getString(key));
     }
\ No newline at end of file
