diff --git a/old_test.java b/new_test.java
index 7a2e8b6..28dc927 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,13 @@
 @Test
     public void testGetBoolean() throws InvalidSettingException {
         String key = "SomeBoolean";
-        Settings.setString(key, "false");
+        getSettings().setString(key, "false");
         boolean expResult = false;
-        boolean result = Settings.getBoolean(key);
+        boolean result = getSettings().getBoolean(key);
         Assert.assertEquals(expResult, result);
 
         key = "something that does not exist";
         expResult = true;
-        result = Settings.getBoolean(key, true);
+        result = getSettings().getBoolean(key, true);
         Assert.assertEquals(expResult, result);
     }
\ No newline at end of file
