diff --git a/old_product.java b/new_product.java
index 1e7b36a..91e2589 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 private <T> void deferEvent(T event, Observer<T> observer)
    {
-      TransactionListener transactionListener = ManagerImpl.instance().getInstanceByType(TransactionListener.class);
+      TransactionListener transactionListener = manager.getInstanceByType(TransactionListener.class);
       DeferredEventNotification<T> deferredEvent = new DeferredEventNotification<T>(event, observer);
       transactionListener.registerSynhronization(deferredEvent);
    }
\ No newline at end of file
