diff --git a/old_test.java b/new_test.java
index 59b55d4..0a1afed 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,7 @@
 		Random rnd = new Random(42);
 		m1.type(Message.Type.REQUEST_4);
 		m1.setHintSign();
+		m1.verified();
 
 		KeyPairGenerator gen = KeyPairGenerator.getInstance("DSA");
 		KeyPair pair1 = gen.generateKeyPair();
@@ -30,10 +31,10 @@
 		m1.keyMap640Keys(new KeyMap640Keys(keysMap));
 
 		Message m2 = encodeDecode(m1);
-		m1.sender(m1.sender().withSkipIPv4(true));
+		m1.sender(m1.sender().withSkipIP(true));
 		Assert.assertEquals(true, m2.publicKey(0) != null);
 		Assert.assertEquals(false, m2.dataMap(0) == null);
 		Assert.assertEquals(false, m2.keyMap640Keys(0) == null);
-		Assert.assertEquals(true, m2.verified());
+		Assert.assertEquals(true, m2.isVerified());
 		compareMessage(m1, m2);
 	}
\ No newline at end of file
