diff --git a/old_product.java b/new_product.java
index c364465..0b9d3ba 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,9 @@
 private Message encodeDecode(final Message m1) throws Exception {
-		AtomicReference<Message> m2 = new AtomicReference<Message>();
 		final CompositeByteBuf buf = Unpooled.compositeBuffer();
 		Encoder encoder = new Encoder(new DSASignatureFactory());
 		encoder.write(buf, m1, null);
-		ChannelHandlerContext ctx = mockChannelHandlerContext(buf, m2);
 		Decoder decoder = new Decoder(new DSASignatureFactory());
-		decoder.decode(ctx, buf, m1.recipient().ipv4Socket().createTCPSocket(), m1
-				.sender().ipv4Socket().createTCPSocket());
-		//buf.release();
+		decoder.decode(buf, m1.recipient().ipv4Socket().createUDPSocket(), m1
+				.sender().ipv4Socket().createUDPSocket());
 		return decoder.message();
 	}
\ No newline at end of file
