diff --git a/old_product.java b/new_product.java
index 2c79aa3..28fe355 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
 protected AmqpSession createSession(String sessionName, Session session, SessionHandler handler) {
-        return new ReactorSession(session, handler, sessionName, reactorProvider, handlerProvider, getCBSNode(),
-            tokenManagerProvider, messageSerializer, connectionOptions.getRetry().getTryTimeout());
+        return new ReactorSession(session, handler, sessionName, reactorProvider, handlerProvider,
+            getClaimsBasedSecurityNode(), tokenManagerProvider, messageSerializer,
+            connectionOptions.getRetry().getTryTimeout());
     }
\ No newline at end of file
