diff --git a/old_product.java b/new_product.java
index ce99342..f8b3ade 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,25 +10,25 @@ public int[] sort(final Comparator<BytesRef> comp) {
       
       @Override
       protected int compare(int i, int j) {
-        final int ord1 = compact[i], ord2 = compact[j];
-        assert bytesStart.length > ord1 && bytesStart.length > ord2;
-        pool.setBytesRef(scratch1, bytesStart[ord1]);
-        pool.setBytesRef(scratch2, bytesStart[ord2]);
+        final int id1 = compact[i], id2 = compact[j];
+        assert bytesStart.length > id1 && bytesStart.length > id2;
+        pool.setBytesRef(scratch1, bytesStart[id1]);
+        pool.setBytesRef(scratch2, bytesStart[id2]);
         return comp.compare(scratch1, scratch2);
       }
 
       @Override
       protected void setPivot(int i) {
-        final int ord = compact[i];
-        assert bytesStart.length > ord;
-        pool.setBytesRef(pivot, bytesStart[ord]);
+        final int id = compact[i];
+        assert bytesStart.length > id;
+        pool.setBytesRef(pivot, bytesStart[id]);
       }
   
       @Override
       protected int comparePivot(int j) {
-        final int ord = compact[j];
-        assert bytesStart.length > ord;
-        pool.setBytesRef(scratch2, bytesStart[ord]);
+        final int id = compact[j];
+        assert bytesStart.length > id;
+        pool.setBytesRef(scratch2, bytesStart[id]);
         return comp.compare(pivot, scratch2);
       }
       
