diff --git a/old_test.java b/new_test.java
index dd471a6..622b180 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,5 +10,7 @@
 		//then
 		Assert.assertTrue(errors.hasErrors());
 		Assert.assertTrue(errors.hasFieldErrors("name"));
+		Assert.assertTrue(errors.hasFieldErrors("person"));
 		Assert.assertEquals("Provider.error.personOrName.required", errors.getFieldError("name").getCode());
+		Assert.assertEquals("Provider.error.personOrName.required", errors.getFieldError("person").getCode());
 	}
\ No newline at end of file
