diff --git a/old_product.java b/new_product.java
index d4dc016..5f5aa0f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,6 +11,7 @@ public void validate(Object obj, Errors errors) throws APIException {
 		
 		if (provider.getPerson() == null && StringUtils.isBlank(provider.getName())) {
 			errors.rejectValue("name", "Provider.error.personOrName.required");
+			errors.rejectValue("person", "Provider.error.personOrName.required");
 		}
 		
 		if (provider.isRetired() && StringUtils.isEmpty(provider.getRetireReason())) {
