diff --git a/old_test.java b/new_test.java
index 0437b12..f66aff5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
         result = Zeroin.root(eps, -6, 6, polyF);
         assertEquals(-4.8790576334840479813, result, eps);
         
-        result = Zeroin.root(eps, -6, 6, polyF, 0);
+        result = Zeroin.root(eps, -6, 6, 0, polyF);
         assertEquals(-4.8790576334840479813, result, eps);
         
         
