diff --git a/old_test.java b/new_test.java
index b95f9f4..b7a4264 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     {
         System.out.println("cluster");
         MiniBatchKMeans kMeans = new MiniBatchKMeans(new EuclideanDistance(), 50, 50, SeedSelectionMethods.SeedSelection.FARTHEST_FIRST);
-        List<List<DataPoint>> clusters = kMeans.cluster(easyData10, 10, ex);
+        List<List<DataPoint>> clusters = kMeans.cluster(easyData10, 10, true);
         assertEquals(10, clusters.size());
         Set<Integer> seenBefore = new IntSet();
         for(List<DataPoint> cluster :  clusters)
