diff --git a/old_test.java b/new_test.java
index 454fa44..9623f84 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
 		encoder.setVersion(1).
 				setError(QrCode.ErrorLevel.H).
 				setMask(new QrCodeMaskPattern.NONE(0b011)).
-				alphanumeric("AC-42").fixate();
+				addAlphanumeric("AC-42").fixate();
 
 		byte[] expected = new byte[]{0b00100000, 0b00101001, (byte)0b11001110, (byte)0b11100111, 0b00100001,0};
 
