diff --git a/old_product.java b/new_product.java
index 285ece1..9d618d1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,6 @@
-public Period minus(Period other) {
+public Period minus(Period amountToSubtract) {
         return create(
-                Jdk8Methods.safeSubtract(years, other.years),
-                Jdk8Methods.safeSubtract(months, other.months),
-                Jdk8Methods.safeSubtract(days, other.days),
-                Jdk8Methods.safeSubtract(nanos, other.nanos));
+                Jdk8Methods.safeAdd(years, amountToSubtract.years),
+                Jdk8Methods.safeAdd(months, amountToSubtract.months),
+                Jdk8Methods.safeAdd(days, amountToSubtract.days));
     }
\ No newline at end of file
