diff --git a/old_product.java b/new_product.java
index d6ea0da..4691d66 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public void registerJobMaster(MockJobMaster jobMaster) throws Exception  {
-			Future<RegistrationResponse> registration = resourceManager.registerJobManager(
+			CompletableFuture<RegistrationResponse> registration = resourceManager.registerJobManager(
 				rmServices.rmLeaderSessionId, jobMaster.leaderSessionID, jobMaster.resourceID, jobMaster.address, jobMaster.jobID);
 			assertTrue(registration.get() instanceof JobMasterRegistrationSuccess);
 		}
\ No newline at end of file
