diff --git a/old_product.java b/new_product.java
index 6b990f3..21a1444 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,11 +7,11 @@ public static OnLineStatistics add(OnLineStatistics A, OnLineStatistics B)
        else if(A.n == 0)
            return new OnLineStatistics(B.n, B.mean, B.m2, B.m3, B.m4, B.min, B.max);
        
-       int nX = B.n + A.n;
-       int nXsqrd = nX*nX;
-       int nAnB = B.n*A.n;
-       int AnSqrd = A.n*A.n;
-       int BnSqrd = B.n*B.n;
+       double nX = B.n + A.n;
+       double nXsqrd = nX*nX;
+       double nAnB = B.n*A.n;
+       double AnSqrd = A.n*A.n;
+       double BnSqrd = B.n*B.n;
        
        double delta = B.mean - A.mean;
        double deltaSqrd = delta*delta;
