diff --git a/old_test.java b/new_test.java
index 439e4a9..12ce981 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,8 @@
 @Test
   public void testCreateTimeOfDay() throws Exception {
-    int hour = rand.nextInt(24);
-    int minute = rand.nextInt(60);
-    int second = rand.nextInt(60);
-    String timeString =
-        Integer.toString(hour) + ":" + Integer.toString(minute) + ":" + Integer.toString(second);
+    TimeOfDay time = TransferJobUtils.createTimeOfDay("17:00:42");
+    assertThat(time).isEqualTo(new TimeOfDay().setHours(17).setMinutes(0).setSeconds(42));
 
-    TimeOfDay time = TransferJobUtils.createTimeOfDay(timeString);
-
-    assertEquals(time,
-        TimeOfDay.class.newInstance().setHours(hour).setMinutes(minute).setSeconds(second));
+    time = TransferJobUtils.createTimeOfDay("08:09:08");
+    assertThat(time).isEqualTo(new TimeOfDay().setHours(8).setMinutes(9).setSeconds(8));
   }
\ No newline at end of file
