diff --git a/old_product.java b/new_product.java
index c8d51af..56ca53e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-public static boolean createStorageDirPath(String path) throws IOException {
+public static boolean createStorageDirPath(String path, String workerDataFolderPermissions)
+      throws IOException {
     if (Files.exists(Paths.get(path))) {
       return false;
     }
@@ -9,8 +10,7 @@ public static boolean createStorageDirPath(String path) throws IOException {
       throw new IOException("Failed to create folder " + path, e);
     }
     String absolutePath = storagePath.toAbsolutePath().toString();
-    String perms = Configuration.get(PropertyKey.WORKER_DATA_FOLDER_PERMISSIONS);
-    changeLocalFilePermission(absolutePath, perms);
+    changeLocalFilePermission(absolutePath, workerDataFolderPermissions);
     setLocalDirStickyBit(absolutePath);
     return true;
   }
\ No newline at end of file
