diff --git a/old_product.java b/new_product.java
index 85e7e0b..13c90ce 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,14 @@ public Order saveOrder(Order order, OrderContext orderContext) throws APIExcepti
 			throw new APIException("Cannot edit an existing order, you need to revise it instead");
 		}
 		
+		if (order.getOrderType() == null) {
+			OrderType orderType = getOrderTypeByConcept(order.getConcept());
+			if (orderType == null) {
+				throw new APIException("No order type matches the concept class");
+			}
+			order.setOrderType(orderType);
+		}
+		
 		if (Order.Action.REVISE.equals(order.getAction())) {
 			Order previousOrder = order.getPreviousOrder();
 			if (previousOrder == null) {
