diff --git a/old_product.java b/new_product.java
index d326c08..2a7b480 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public int lsr(TachyonURI path) throws IOException {
+public void lsr(TachyonURI path) throws IOException {
     List<FileInfo> files = listStatusSortedByIncreasingCreationTime(path);
     String format = "%-10s%-25s%-15s%-5s%n";
     for (FileInfo file : files) {
@@ -16,5 +16,4 @@ public int lsr(TachyonURI path) throws IOException {
         lsr(new TachyonURI(path.getScheme(), path.getAuthority(), file.getPath()));
       }
     }
-    return 0;
   }
\ No newline at end of file
