diff --git a/old_test.java b/new_test.java
index 9e0f181..2ec699a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,10 +6,10 @@
                .entry(MetadataEntry.builder().entry(metadataKey, metadataValue).build())
                .build();
       
-      Task task = metadataApi.mergeMetadata(adminVdcUri, metadata);
+      Task task = metadataApi.merge(adminVdcUri, metadata);
       assertTaskSucceeds(task);
       
-      MetadataValue modified = metadataApi.getMetadataValue(adminVdcUri, metadataKey);
+      MetadataValue modified = metadataApi.getValue(adminVdcUri, metadataKey);
       Checks.checkMetadataValueFor("AdminVdc", modified, metadataValue);
       Checks.checkMetadata(metadata);
    }
\ No newline at end of file
