diff --git a/old_test.java b/new_test.java
index ea7c3e7..e097f48 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,8 @@
         files.add("file1.file");
         files.add("file2.file");
         instance.setFiles(files);
+        instance.setTags("test tag");
         instance.strip();
         assertEquals(0, instance.getFiles().size());
+        assertEquals(null, instance.getTags());
     }
\ No newline at end of file
