diff --git a/old_test.java b/new_test.java
index 3215a42..8eb9132 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
 		as.saveGlobalProperty(new GlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_USER_REQUIRE_EMAIL_AS_USERNAME, "true"));
 		
 		Errors errors = new BindException(user, "user");
-		new UserValidator().validate(user, errors);
+		validator.validate(user, errors);
 		
 		Assert.assertFalse(errors.hasFieldErrors("username"));
 	}
\ No newline at end of file
