diff --git a/old_product.java b/new_product.java
index a197b5d..b576107 100644
--- a/old_product.java
+++ b/new_product.java
@@ -23,6 +23,13 @@ public void validate(Object obj, Errors errors) {
 				if (person.getPersonName() == null || StringUtils.isEmpty(person.getPersonName().getFullName())) {
 					errors.rejectValue("person", "Person.names.length");
 				}
+				errors.pushNestedPath("person");
+				try {
+					personValidator.validate(person, errors);
+				} finally {
+					errors.popNestedPath();
+				}
+					
 			}
 			
 			AdministrationService as = Context.getAdministrationService();
