diff --git a/old_product.java b/new_product.java
index 41ca185..27d904d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -52,7 +52,7 @@ private void runStreamed(MemoryManager memoryManager, IOManager ioManager,
 		try {
 			innerInput = new SpillingResettableIterator<KeyValuePair<Key, Value>>(memoryManager, ioManager,
 				innerReader, MEMORY_IO, new KeyValuePairDeserializer<Key, Value>(stub.getFirstInKeyType(), stub
-					.getFirstInValueType()));
+					.getFirstInValueType()), this);
 		} catch (MemoryAllocationException mae) {
 			throw new RuntimeException("Unable to obtain SpillingResettable iterator for inner side.", mae);
 		}
