diff --git a/old_test.java b/new_test.java
index 65c1063..2cfd1e0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,12 @@
 @Test
 	public void testHasNext() throws ServiceException, InterruptedException  {
+		final AbstractInvokable memOwner = new DummyInvokable();
 		
 		// create the reader
 		reader = new CollectionReader<PactInteger>(objects);
 		// create the resettable Iterator
 		SpillingResettableIterator<PactInteger> iterator = new SpillingResettableIterator<PactInteger>(memman, ioman,
-			reader, 1000, deserializer);
+			reader, 1000, deserializer, memOwner);
 		// open the iterator
 		try {
 			iterator.open();
