diff --git a/old_product.java b/new_product.java
index 2f69078..f24ccc4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ public boolean isObserverInterested(Annotation... bindings)
       // TODO This logic needs to be in injectable
       boolean result = true;
       // Check each binding specified by this observer against those provided
-      if (this.eventBindings != null)
+      if (this.eventBindings.length > 0)
       {
          if ((bindings != null) && (bindings.length > 0))
          {
@@ -13,14 +13,17 @@ public boolean isObserverInterested(Annotation... bindings)
                int eventBindingIndex = bindingsArray.indexOf(annotation);
                if (eventBindingIndex >= 0)
                {
-                  // TODO Use annotation equality
-                  result = annotationsMatch(annotation, bindingsArray.get(eventBindingIndex));
+                  //result = annotationsMatch(annotation, bindingsArray.get(eventBindingIndex));
+                  result = annotation.equals(bindingsArray.get(eventBindingIndex));
                } else
                {
                   result = false;
                   break;
                }
             }
+         } else
+         {
+            result = false;
          }
       }
       return result;
