diff --git a/old_test.java b/new_test.java
index 3268ec5..c8f94cb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,6 +7,7 @@
       assert wrappedObserver.getEventBindings().length == 1;
       assert wrappedObserver.isObserverInterested(new TameAnnotationLiteral());
       assert !wrappedObserver.isObserverInterested(new AnimalStereotypeAnnotationLiteral());
+      assert !wrappedObserver.isObserverInterested();
       assert wrappedObserver.isObserverInterested(new TameAnnotationLiteral(), new RoleBinding("Admin"));
       
       // Perform some tests with binding values (7.7.1)
