diff --git a/old_product.java b/new_product.java
index 5c2c1a0..e3f93e6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,7 +14,10 @@ public void buildIndex(String indexType) {
 				STRtree rt = new STRtree();
 				GeometryFactory geometryFactory = new GeometryFactory();
 				for (Envelope e : envelopes)
-					rt.insert(e, geometryFactory.toGeometry(e));
+					try {
+						rt.insert(e, geometryFactory.toGeometry(e));
+					} catch (ClassCastException e1) {
+					}
 				ArrayList<STRtree> result = new ArrayList<STRtree>();
 				result.add(rt);
 				return result;
