diff --git a/old_test.java b/new_test.java
index 74ec651..0adb265 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,7 +10,7 @@
         Batch response = createMockBatch(batchId, name, currentTime, currentTime, propertyList);
 
         mockServer.when(
-                request().withMethod("POST").withPath(DocumentConversion.BATCHES_PATH)
+                request().withMethod("POST").withPath(BATCHES_PATH)
         ).respond((response().withBody(GsonSingleton.getGson().toJson(response))));
 
         // Call create batch
