diff --git a/old_test.java b/new_test.java
index a535745..7f0523b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void testPost() {
-    final Request request = RequestBuilder.post(url).build();
+    final Request request = RequestBuilder.post(HttpUrl.parse(url)).build();
     assertEquals("POST", request.method());
     assertEquals(url, request.url().toString());
   }
\ No newline at end of file
