diff --git a/old_product.java b/new_product.java
index 88e8dea..d216269 100644
--- a/old_product.java
+++ b/new_product.java
@@ -28,5 +28,5 @@
       Tree parent = state.getPath().getParentPath().getLeaf();
       fix = new SuggestedFix().delete(parent);
     }
-    return new Description(methodInvocationTree, diagnosticMessage, fix);
+    return new Description(methodInvocationTree, getDiagnosticMessage(), fix);
   }
\ No newline at end of file
