diff --git a/old_test.java b/new_test.java
index cfb5696..7e58772 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,11 +7,11 @@
         SuggestedFix fix = new SuggestedFix()
             .addImport(JUNIT4_TEST_ANNOTATION)
             .replace(methodTree, methodString);
-        return new Description(methodTree, diagnosticMessage, fix);
+        return new Description(methodTree, getDiagnosticMessage(), fix);
       }
     }
     SuggestedFix fix = new SuggestedFix()
         .addImport(JUNIT4_TEST_ANNOTATION)
         .prefixWith(methodTree, "@Test\n");
-    return new Description(methodTree, diagnosticMessage, fix);
+    return new Description(methodTree, getDiagnosticMessage(), fix);
   }
\ No newline at end of file
