diff --git a/old_test.java b/new_test.java
index 63efcf5..4ab511d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,7 @@
 @Test
     public void testOnCreateOrUpdateCommand() throws GeniePreconditionException {
-        this.c = new Command(NAME, USER, CommandStatus.ACTIVE, EXECUTABLE, VERSION);
+        this.c = new Command(NAME, USER, VERSION, CommandStatus.ACTIVE, EXECUTABLE);
+        Assert.assertNull(this.c.getTags());
         this.c.onCreateOrUpdateCommand();
+        Assert.assertEquals(2, this.c.getTags().size());
     }
\ No newline at end of file
