diff --git a/old_product.java b/new_product.java
index eb68192..56000c0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,6 @@
 @PrePersist
     @PreUpdate
     protected void onCreateOrUpdateCommand() throws GeniePreconditionException {
-        validate(this.status, this.executable, null);
         // Add the id to the tags
         if (this.tags == null) {
            this.tags = new HashSet<>();
