diff --git a/old_product.java b/new_product.java
index ae4df52..1a86c68 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public void factoryUTC_ofInstant_InstantProvider() {
+public void factory_ofInstantUTC() {
         Instant instant = Instant.ofEpochSecond(86400 + 5 * 3600 + 10 * 60 + 20);
         OffsetDateTime test = OffsetDateTime.ofInstantUTC(instant);
         check(test, 1970, 1, 2, 5, 10, 20, 0, ZoneOffset.UTC);
