diff --git a/old_product.java b/new_product.java
index a2abd6c..9f29d6e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,5 +6,5 @@ public List<List<Object>> executeSQL(String sql, boolean selectOnly) throws DAOE
 		if (HibernateUtil.isHSQLDialect(sessionFactory)) {
 			sql = sql.replace("`", "");
 		}
-		return DatabaseUtil.executeSQLWithSession(sessionFactory.getCurrentSession(), sql, selectOnly);
+		return DatabaseUtil.executeSQL(sessionFactory.getCurrentSession(), sql, selectOnly);
 	}
\ No newline at end of file
