diff --git a/old_test.java b/new_test.java
index a183c68..4092ce9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
     public void testMinusMonths() {
-        assertEquals(testDate.minusMonths(1), MinguoDate.minguoDate(testYear, testMonthOfYear.previous(), testDayOfMonth));
+        assertEquals(testDate.minusMonths(1), MinguoDate.of(testYear, testMonthOfYear.previous(), testDayOfMonth));
     }
\ No newline at end of file
