diff --git a/old_product.java b/new_product.java
index 536da60..2f30fb9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,5 +13,5 @@ public HijrahDate minusMonths(int months) {
             years = MathUtils.safeIncrement(years);
         }
         int newYear = MathUtils.safeAdd(this.yearOfEra, years);
-        return HijrahDate.hijrahDate(this.era, newYear, (newMonth + 1), this.dayOfMonth);
+        return HijrahDate.of(this.era, newYear, (newMonth + 1), this.dayOfMonth);
     }
\ No newline at end of file
