diff --git a/old_test.java b/new_test.java
index 8b4a43a..0083fc5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,6 @@
 		e.setVisit(visit);
 		Context.getEncounterService().saveEncounter(e);
 		//sanity check
-		Assert.assertTrue(Context.getEncounterService().getEncountersByVisit(visit).size() > 0);
+		Assert.assertTrue(Context.getEncounterService().getEncountersByVisit(visit, false).size() > 0);
 		Context.getVisitService().purgeVisit(visit);
 	}
\ No newline at end of file
