diff --git a/old_product.java b/new_product.java
index 14b5567..73b5fe4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@
 		if (visit.getVisitId() == null)
 			return;
 		//TODO there is a ticket for adding includeVoided argument to getEncountersByVisit for this not to fail
-		if (Context.getEncounterService().getEncountersByVisit(visit).size() > 0)
+		if (Context.getEncounterService().getEncountersByVisit(visit, true).size() > 0)
 			throw new APIException(Context.getMessageSourceService().getMessage("Visit.purge.inUse", null,
 			    "Cannot purge a visit that has encounters associated to it", Context.getLocale()));
 		dao.deleteVisit(visit);
