diff --git a/old_product.java b/new_product.java
index 91a848b..7c807f7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public PackedAlignments packAlignments(
             Iterator<Alignment> iter,
-            int end,
+            List<? extends Range> ranges,
             AlignmentTrack.RenderOptions renderOptions) {
 
         if(renderOptions == null) renderOptions = new AlignmentTrack.RenderOptions();
@@ -9,9 +9,10 @@ public PackedAlignments packAlignments(
         boolean pairAlignments = renderOptions.isViewPairs() || renderOptions.isPairedArcView();
 
         if (iter == null || !iter.hasNext()) {
-            return new PackedAlignments(packedAlignments, renderOptions);
+            return new PackedAlignments(ranges, packedAlignments, renderOptions);
         }
 
+        int end = ranges.get(ranges.size()-1).getEnd();
         if (renderOptions.groupByOption == null) {
             List<Row> alignmentRows = new ArrayList<Row>(10000);
             pack(iter, end, pairAlignments, alignmentRows);
@@ -49,6 +50,6 @@ public PackedAlignments packAlignments(
             packedAlignments.put("", alignmentRows);
         }
 
-        return new PackedAlignments(packedAlignments, renderOptions);
+        return new PackedAlignments(ranges, packedAlignments, renderOptions);
 
     }
\ No newline at end of file
