diff --git a/old_test.java b/new_test.java
index 897a2c4..de5e9af 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
         */
         CloseableIterator<Alignment> iter = getAlignments();
 
-        Map<String, List<Row>> result = (new AlignmentPacker()).packAlignments(iter, end,
+        Map<String, List<Row>> result = (new AlignmentPacker()).packAlignments(iter, ranges,
                 new AlignmentTrack.RenderOptions());
         assertEquals(1, result.size());
         for (List<Row> alignmentrows : result.values()) {
