diff --git a/old_test.java b/new_test.java
index 2f84d7d..9b9bb08 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testOnCreateApplication() throws GenieException {
-        final Application a = new Application(NAME, USER, ApplicationStatus.ACTIVE);
-        a.onCreateApplication();
+    public void testOnCreateOrUpdate() throws GenieException {
+        this.a = new Application(NAME, USER, ApplicationStatus.ACTIVE);
+        this.a.onCreateOrUpdate();
     }
\ No newline at end of file
