diff --git a/old_product.java b/new_product.java
index e7b4268..4985f48 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,8 @@
 public AlluxioURI join(String suffix) {
-    return new AlluxioURI(getScheme(), getAuthority(), getPath() + AlluxioURI.SEPARATOR + suffix,
-        mUri.getQuery());
+    // TODO(gpang): there should be other usage of join() which can use joinUnsafe() instead.
+    String path = getPath();
+    StringBuilder sb = new StringBuilder(path.length() + 1 + suffix.length());
+
+    return new AlluxioURI(this,
+        sb.append(path).append(AlluxioURI.SEPARATOR).append(suffix).toString(), true);
   }
\ No newline at end of file
