diff --git a/old_test.java b/new_test.java
index 07c5ad6..27c6bb4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,16 +8,9 @@
         val sm = context.containerMetadata.getStreamSegmentMetadata(SEGMENT_ID);
         val idx = context.index.forSegment(SEGMENT_ID, TIMEOUT).join();
 
-        AssertExtensions.assertThrows(
-                "delete() allowed deleting data for a non-deleted segment.",
-                () -> context.index.delete(sm, TIMEOUT),
-                ex -> ex instanceof IllegalArgumentException);
-
-        sm.markDeleted();
-
         // We intentionally delete twice to make sure the operation is idempotent.
-        context.index.delete(sm, TIMEOUT).join();
-        context.index.delete(sm, TIMEOUT).join();
+        context.index.delete(sm.getName(), TIMEOUT).join();
+        context.index.delete(sm.getName(), TIMEOUT).join();
 
         AssertExtensions.assertThrows(
                 "put() worked after delete().",
