diff --git a/old_test.java b/new_test.java
index b3f8fc5..e979111 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,4 +4,14 @@
     responseContent = HttpMethed.parseResponseContent(response);
     HttpMethed.printJsonContent(responseContent);
     Assert.assertEquals("20", responseContent.getString("brokerage"));
+
+    response = HttpMethed.getBrokerageOnVisible(httpnode, witnessAddress2, "true");
+    responseContent = HttpMethed.parseResponseContent(response);
+    HttpMethed.printJsonContent(responseContent);
+    Assert.assertEquals("20", responseContent.getString("brokerage"));
+
+    response = HttpMethed.getBrokerageOnVisible(httpnode, fromAddress, "false");
+    responseContent = HttpMethed.parseResponseContent(response);
+    HttpMethed.printJsonContent(responseContent);
+    Assert.assertEquals("20", responseContent.getString("brokerage"));
   }
\ No newline at end of file
