diff --git a/old_product.java b/new_product.java
index 533b8f7..5b2db3c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,18 +16,7 @@
 
         closed = true;
 
-        if (bestEffortAffinity) {
-            for (JdbcThinTcpIo clioIo : ios.values())
-                clioIo.close();
-
-            ios.clear();
-
-            iosArr = null;
-        }
-        else {
-            if (singleIo != null)
-                singleIo.close();
-        }
+        cliIo.close();
 
         timer.cancel();
 
