diff --git a/old_test.java b/new_test.java
index 98e81cf..6711a80 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testParseCSVFile() {
         logger.info("parseCSVFile");
         
-        DatabaseConfiguration dbConf = TestUtils.getDBConfig();
+        Configuration conf = TestUtils.getConfig();
         
         
         LinkedHashMap<String, TypeInference.DataType> headerDataTypes = new LinkedHashMap<>(); 
@@ -16,14 +16,14 @@
         
         Dataframe dataset;
         try (Reader fileReader = new InputStreamReader(this.getClass().getClassLoader().getResourceAsStream("datasets/cities.csv"), "UTF-8")) {
-            dataset = Dataframe.Builder.parseCSVFile(fileReader, "metro_population", headerDataTypes, ',', '"', "\r\n", null, null, dbConf);
+            dataset = Dataframe.Builder.parseCSVFile(fileReader, "metro_population", headerDataTypes, ',', '"', "\r\n", null, null, conf);
         }
         catch(UncheckedIOException | IOException ex) {
             logger.warn("Unable to download datasets, skipping test.");
             throw new RuntimeException(ex);
         }
         
-        Dataframe expResult = new Dataframe(dbConf);
+        Dataframe expResult = new Dataframe(conf);
         
         AssociativeArray xData1 = new AssociativeArray();
         xData1.put("city", "Athens");
