diff --git a/old_product.java b/new_product.java
index 739306f..9dc38fb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static Dataframe parseCSVFile(Reader reader, String yVariable, LinkedHashMap<String, TypeInference.DataType> headerDataTypes, 
-                                           char delimiter, char quote, String recordSeparator, Long skip, Long limit, DatabaseConfiguration dbConf) {
+                                           char delimiter, char quote, String recordSeparator, Long skip, Long limit, Configuration conf) {
             Logger logger = LoggerFactory.getLogger(Dataframe.Builder.class);
             
             if(skip == null) {
@@ -19,7 +19,7 @@ public static Dataframe parseCSVFile(Reader reader, String yVariable, LinkedHash
             TypeInference.DataType yDataType = headerDataTypes.get(yVariable);
             Map<String, TypeInference.DataType> xDataTypes = new HashMap<>(headerDataTypes); //copy header types
             xDataTypes.remove(yVariable); //remove the response variable from xDataTypes
-            Dataframe dataset = new Dataframe(dbConf, yDataType, xDataTypes); //use the private constructor to pass DataTypes directly and avoid updating them on the fly
+            Dataframe dataset = new Dataframe(conf, yDataType, xDataTypes); //use the private constructor to pass DataTypes directly and avoid updating them on the fly
             
             
             CSVFormat format = CSVFormat
@@ -66,7 +66,7 @@ public static Dataframe parseCSVFile(Reader reader, String yVariable, LinkedHash
                             }
                         }
                     }
-                });
+                }, conf.getConcurrencyConfig());
             } 
             catch (IOException ex) {
                 throw new RuntimeException(ex);
