diff --git a/old_test.java b/new_test.java
index 6912bb7..ef8381c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
     LayoutState current = mComponentTree.getMainThreadLayoutState();
     DefaultInternalNode layout = (DefaultInternalNode) current.getLayoutRoot();
 
-    mComponentTree.updateStateAsync("root,B,C", createStateUpdate(), "test");
+    mComponentTree.updateStateAsync("root,B,C", createStateUpdate(), "test", false);
     mLayoutThreadShadowLooper.runToEndOfTasks();
 
     Set<String> set = new HashSet<>();
