diff --git a/old_product.java b/new_product.java
index 8323d42..1289619 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,5 +5,6 @@ public void updateStateAsync(StateUpdate stateUpdate, String attribution) {
       return;
     }
 
-    mComponentTree.updateStateAsync(mComponentScope.getGlobalKey(), stateUpdate, attribution);
+    mComponentTree.updateStateAsync(
+        mComponentScope.getGlobalKey(), stateUpdate, attribution, isCreateLayoutInProgress());
   }
\ No newline at end of file
