diff --git a/old_product.java b/new_product.java
index 6a0823a..86dd71a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -172,9 +172,11 @@ public void validate(Object obj, Errors errors) throws APIException, DuplicateCo
 					mappedTermIds = new HashSet<Integer>();
 				
 				//if we already have a mapping to this term, reject it this map
-				if (!mappedTermIds.add(map.getConceptReferenceTerm().getId())) {
-					errors.rejectValue("conceptMappings[" + index + "]", "ConceptReferenceTerm.term.alreadyMapped",
-					    "Cannot map a reference term multiple times to the same concept");
+				if (map.getConceptReferenceTerm().getId() != null) {
+					if (!mappedTermIds.add(map.getConceptReferenceTerm().getId())) {
+						errors.rejectValue("conceptMappings[" + index + "]", "ConceptReferenceTerm.term.alreadyMapped",
+						    "Cannot map a reference term multiple times to the same concept");
+					}
 				}
 				
 				index++;
