diff --git a/old_test.java b/new_test.java
index a9b51fc..ca197fd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -23,7 +23,7 @@
                 theDate = Date.valueOf(element);
                 fail("Should throw IllegalArgumentException.");
             } catch (IllegalArgumentException e) {
-                //expected
+                // expected
             } // end try
         } // end for
 
