diff --git a/old_product.java b/new_product.java
index 90ac332..7a05531 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-@RequestMapping(value = "vacantEventClassFamiliesByAppToken/{applicationToken}", method = RequestMethod.GET)
+@RequestMapping(value = "vacantEventClassFamilies/{applicationToken}", method = RequestMethod.GET)
     @ResponseBody
     public List<EcfInfoDto> getVacantEventClassFamiliesByApplicationToken(@PathVariable String applicationToken) throws KaaAdminServiceException {
         return kaaAdminService.getVacantEventClassFamiliesByApplicationToken(applicationToken);
