diff --git a/old_test.java b/new_test.java
index 62321dc..c01a15b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,9 +8,9 @@
         Number160 key2 = new Number160(5667);
         m1.setKey(key1);
         m1.setKey(key2);
-        List<Number480> tmp2 = new ArrayList<Number480>();
-        Number480 n1 = new Number480(rnd);
-        Number480 n2 = new Number480(rnd);
+        List<Number640> tmp2 = new ArrayList<Number640>();
+        Number640 n1 = new Number640(rnd);
+        Number640 n2 = new Number640(rnd);
         tmp2.add(n1);
         tmp2.add(n2);
         m1.setKeyCollection(new KeyCollection(tmp2));
