diff --git a/old_product.java b/new_product.java
index 710c916..1df3f97 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public void setUpToClass(final Class<?> clazz) {
         if (clazz != null) {
-            Object object = getObject();
+            final Object object = getObject();
             if (object != null && clazz.isInstance(object) == false) {
                 throw new IllegalArgumentException("Specified class is not a superclass of the object");
             }
