diff --git a/old_test.java b/new_test.java
index 94b49fa..b18bb22 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test(expected=IllegalArgumentException.class)
     public void test_setUpToClass_invalid() {
-        Integer val = Integer.valueOf(5);
-        ReflectionToStringBuilder test = new ReflectionToStringBuilder(val);
+        final Integer val = Integer.valueOf(5);
+        final ReflectionToStringBuilder test = new ReflectionToStringBuilder(val);
         try {
             test.setUpToClass(String.class);
         } finally {
