diff --git a/old_test.java b/new_test.java
index c281996..6340d9d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,14 @@
 @Test(testName = "GET /catalog/{id}/metadata/{key}", dependsOnMethods = { "testGetCatalogMetadata" }, enabled = false)
-   public void testGetCatalogMetadataEntry() {
+   public void testGetCatalogMetadataValue() {
       MetadataEntry existingMetadataEntry = Iterables.find(catalogMetadata.getMetadataEntries(), new Predicate<MetadataEntry>() {
          @Override
          public boolean apply(MetadataEntry input) {
             return input.getKey().equals("KEY");
          }
       });
-      MetadataEntry metadataEntry = catalogClient.getCatalogMetadataEntry(catalogRef, "KEY");
-      assertEquals(existingMetadataEntry.getValue(), metadataEntry.getValue());
-      checkMetadataEntry(metadataEntry);
+      MetadataValue metadataValue = catalogClient.getCatalogMetadataValue(catalogRef, "KEY");
+      // XXX
+      assertEquals(metadataValue.getValue(), existingMetadataEntry.getValue(),
+            "The MetadataValue for KEY should have the same Value as the existing MetadataEntry");
+      checkMetadataValue(metadataValue);
    }
\ No newline at end of file
