diff --git a/old_product.java b/new_product.java
index 323db63..f9d734e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,9 +4,6 @@
         if (closed) {
             throw new IOException(Messages.getString("archive.1E")); //$NON-NLS-1$
         }
-        if (currentEntry == null) {
-            return 1;
-        }
-        return super.available();
+        return (currentEntry == null || inRead < currentEntry.size) ? 1 : 0;
         // END android-changed
     }
\ No newline at end of file
