diff --git a/old_product.java b/new_product.java
index 51e29e8..6b06e3f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -35,9 +35,9 @@ public void clearStorage() throws BackendException {
                 client.system_drop_keyspace(keySpaceName);
                 pool.clear();
             } else {
-                final List<CfDef> cfDefs = ksDef.getCf_defs();
+                final List<CfDef> columnFamilyDefinitions = ksDef.getCf_defs();
 
-                if (null == cfDefs) {
+                if (null == columnFamilyDefinitions) {
                     log.debug(lp + "Received empty CfDef list for keyspace {}; not truncating CFs", keySpaceName);
                     return;
                 }
