diff --git a/old_product.java b/new_product.java
index 9592280..86cf1dd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @Override
-    public ResponseStatsPerAppBean getResponseStatsPerApp(String organizationId, String apiId,
+    public ResponseStatsPerClientBean getResponseStatsPerClient(String organizationId, String apiId,
             String version, DateTime from, DateTime to) {
-        ResponseStatsPerAppBean rval = new ResponseStatsPerAppBean();
+        ResponseStatsPerClientBean rval = new ResponseStatsPerClientBean();
         return rval;
     }
\ No newline at end of file
