diff --git a/old_product.java b/new_product.java
index 4717809..557235f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
-public <T extends Client<I, O>, R extends Client<I, O>, I extends HttpRequest, O extends HttpResponse>
-    B decorator(Function<T, R> decorator) {
+public B decorator(Function<? super HttpClient, ? extends HttpClient> decorator) {
         decoration.add(decorator);
         return self();
     }
\ No newline at end of file
