diff --git a/old_test.java b/new_test.java
index 8de326c..221cbfc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,8 @@
 @Test
   public void parent() {
     // check with a simple path
-    assertEquals(norm(PATH), norm(query(_FILE_PARENT.args(PATH1))).toLowerCase(Locale.ENGLISH));
+    assertEquals(Paths.get(PATH1).getParent().toString() + File.separator,
+        query(_FILE_PARENT.args(PATH1)));
     // check with an empty path
     query(EMPTY.args(_FILE_PARENT.args("")), "false");
     // check with a path without directory separators
