diff --git a/old_product.java b/new_product.java
index fd5bfea..3244786 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 private Str parent(final QueryContext ctx) throws QueryException {
-    final String parent = checkFile(0, ctx).getAbsoluteFile().getParent();
-    return parent == null ? null : Str.get(dir(parent));
+    final Path parent = absolute(checkPath(0, ctx)).getParent();
+    return parent == null ? null : get(parent, true);
   }
\ No newline at end of file
