diff --git a/old_product.java b/new_product.java
index 044fded..2a40832 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,18 @@
 @Override
-	public synchronized void resourceAllocated(JobID jobID, AllocatedResource allocatedResource) {
+	public synchronized void resourcesAllocated(final JobID jobID, final List<AllocatedResource> allocatedResources) {
 
-		List<AllocatedResource> allocatedResources = this.resourcesOfJobs.get(jobID);
-		if (allocatedResources == null) {
-			allocatedResources = new ArrayList<AllocatedResource>();
-			this.resourcesOfJobs.put(jobID, allocatedResources);
+		List<AllocatedResource> allocatedResourcesOfJob = this.resourcesOfJobs.get(jobID);
+		if (allocatedResourcesOfJob == null) {
+			allocatedResourcesOfJob = new ArrayList<AllocatedResource>();
+			this.resourcesOfJobs.put(jobID, allocatedResourcesOfJob);
 		}
 
-		if (allocatedResources.contains(allocatedResource)) {
-			throw new IllegalStateException("Resource " + allocatedResource.getAllocationID()
-				+ " is already allocated by job " + jobID);
-		}
+		for (final AllocatedResource allocatedResource : allocatedResources) {
+			if (allocatedResourcesOfJob.contains(allocatedResource)) {
+				throw new IllegalStateException("Resource " + allocatedResource.getAllocationID()
+					+ " is already allocated by job " + jobID);
+			}
 
-		allocatedResources.add(allocatedResource);
+			allocatedResourcesOfJob.add(allocatedResource);
+		}
 	}
\ No newline at end of file
