diff --git a/old_product.java b/new_product.java
index f38e158..e115415 100644
--- a/old_product.java
+++ b/new_product.java
@@ -20,14 +20,30 @@
                     message = "Genie Server Error due to Unknown Exception"
             )
     })
-    public Set<Command> getCommandsForApplication(
+    public List<Command> getCommandsForApplication(
             @ApiParam(
                     value = "Id of the application to get the commands for.",
                     required = true
             )
             @PathParam("id")
-            final String id
+            final String id,
+            @ApiParam(
+                    value = "The statuses of the commands to find.",
+                    allowableValues = "ACTIVE, DEPRECATED, INACTIVE"
+            )
+            @QueryParam("status")
+            final Set<String> statuses
     ) throws GenieException {
         LOG.info("Called with id " + id);
-        return this.applicationConfigService.getCommandsForApplication(id);
+
+        Set<CommandStatus> enumStatuses = null;
+        if (!statuses.isEmpty()) {
+            enumStatuses = EnumSet.noneOf(CommandStatus.class);
+            for (final String status : statuses) {
+                if (StringUtils.isNotBlank(status)) {
+                    enumStatuses.add(CommandStatus.parse(status));
+                }
+            }
+        }
+        return this.applicationConfigService.getCommandsForApplication(id, enumStatuses);
     }
\ No newline at end of file
