diff --git a/old_product.java b/new_product.java
index 81d117d..53e0f7b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -17,7 +17,7 @@ private Message encodeDecode(final Message m1) throws Exception {
 		ChannelHandlerContext ctx = mockChannelHandlerContext(buf, m2);
 		encoder.write(ctx, m1, null);
 		Decoder decoder = new Decoder(new DSASignatureFactory());
-		decoder.decode(ctx, buf, m1.getRecipient().createSocketTCP(), m1
-				.getSender().createSocketTCP());
+		decoder.decode(ctx, buf, m1.recipient().createSocketTCP(), m1
+				.sender().createSocketTCP());
 		return decoder.message();
 	}
\ No newline at end of file
