diff --git a/old_product.java b/new_product.java
index 7673853..1dbd99a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public static synchronized File getTempDirectory() throws IOException {
+public synchronized File getTempDirectory() throws IOException {
         if (tempDirectory == null) {
-            final File baseTemp = new File(Settings.getString(Settings.KEYS.TEMP_DIRECTORY, System.getProperty("java.io.tmpdir")));
+            final File baseTemp = new File(getString(Settings.KEYS.TEMP_DIRECTORY, System.getProperty("java.io.tmpdir")));
             tempDirectory = FileUtils.createTempDirectory(baseTemp);
         }
         return tempDirectory;
