diff --git a/old_test.java b/new_test.java
index a7889ea..178433f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
         double[] B = {0.5, 0.5, 0.5};
         double[] X = {-0.2027027, 0.8783784, 0.4729730};
 
-        SVD result = Matrix.newInstance(A).svd();
+        SVD result = Matrix.of(A).svd();
         double[] x = new double[B.length];
         result.solve(B, x);
         assertEquals(X.length, x.length);
