diff --git a/old_product.java b/new_product.java
index 2f4921f..b04cdf0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,16 +3,16 @@
     public synchronized T get(MarshallingContext context) throws IOException, ClassNotFoundException {
         if (this.object == null) {
             this.context = context;
-            if (this.bytes != null) {
-                ByteArrayInputStream input = new ByteArrayInputStream(this.bytes);
+            if (this.buffer != null) {
+                ByteArrayInputStream input = new ByteArrayInputStream(this.buffer.array(), this.buffer.arrayOffset(), this.buffer.limit() - this.buffer.arrayOffset());
                 ClassLoader loader = setThreadContextClassLoader(this.context.getClassLoader());
                 try (SimpleDataInput data = new SimpleDataInput(Marshalling.createByteInput(input))) {
-                    int version = IndexSerializer.VARIABLE.readInt(data);
+                    int version = IndexSerializer.UNSIGNED_BYTE.readInt(data);
                     try (Unmarshaller unmarshaller = context.createUnmarshaller(version)) {
                         unmarshaller.start(data);
                         this.object = (T) unmarshaller.readObject();
                         unmarshaller.finish();
-                        this.bytes = null; // Free up memory
+                        this.buffer = null; // Free up memory
                     }
                 } finally {
                     setThreadContextClassLoader(loader);
