diff --git a/old_product.java b/new_product.java
index f3b2c7b..0acf4ce 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,10 +5,9 @@
                 throw new IOException(Msg.getString("K005b")); //$NON-NLS-1$
             }
             /* Are there buffered characters available? */
-            if (pos < count || fillbuf() != -1) {
+            if (pos < end || fillBuf() != -1) {
                 return buf[pos++];
             }
-            markpos = -1;
             return -1;
         }
     }
\ No newline at end of file
