diff --git a/old_product.java b/new_product.java
index e2f4f3a..d2a1aee 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,14 @@
 public final void delete(final int pre) {
-    meta.update(true);
+    meta.update();
 
     // size of the subtree to delete
     int k = kind(pre);
     int s = size(pre, k);
+    final boolean empty = pre == 0 && s == meta.size;
+
+    // update document index: delete specified entry
+    docindex.delete(this, pre, s);
+
     ns.delete(pre, s);
 
     // reduce size of ancestors
@@ -25,12 +30,11 @@ public final void delete(final int pre) {
 
     // preserve empty root node
     int p = pre;
-    final boolean empty = p == 0 && s == meta.size;
     if(empty) {
       ++p;
       --s;
-    } else {
-      if(kind(p) == DOC) --meta.ndocs;
+    } else if(kind(p) == DOC) {
+      --meta.ndocs;
     }
 
     // delete node from table structure and reduce document size
