diff --git a/old_test.java b/new_test.java
index 39f7964..8a9b70a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public void testDelete() throws QueryException, BaseXException {
     final String fun = check(Function.DBDELETE);
 
-    new Add("etc/test/dir", "docs", "test").execute(CONTEXT);
+    new Add(FLDR, "docs", "test").execute(CONTEXT);
 
     query(fun + "('db', 'test')", "");
     query("count(collection('db/test')) eq 0", "true");
