diff --git a/old_test.java b/new_test.java
index cc456fe..cfe5155 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,10 +2,10 @@
   public final void testWriteNum() throws IOException {
     long pos = RANDOM_POS;
     da.cursor(pos);
-    da.writeNum(da.cursor(), CINT5);
-    da.writeNum(da.cursor(), CINT4);
-    da.writeNum(da.cursor(), CINT2);
-    da.writeNum(da.cursor(), CINT1);
+    da.writeNum(CINT5);
+    da.writeNum(CINT4);
+    da.writeNum(CINT2);
+    da.writeNum(CINT1);
     da.flush();
 
     assertContent(pos, CINT5_BIN);
