diff --git a/old_test.java b/new_test.java
index ca18641..14237f8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,10 @@
       throws Exception {
     when(req.getMethod()).thenReturn("GET");
     when(req.getRequestURI()).thenReturn("/failAtConstruction");
+    when(requestAuthenticator.authorize(AUTH_PUBLIC.authSettings(), req))
+        .thenReturn(Optional.of(AuthResult.create(AuthLevel.NONE)));
+
     handler.handleRequest(req, rsp);
+
     verify(rsp).sendError(503, "Fail at construction");
   }
\ No newline at end of file
