diff --git a/old_product.java b/new_product.java
index 5082227..1119aac 100644
--- a/old_product.java
+++ b/new_product.java
@@ -22,7 +22,7 @@ public void handleRequest(HttpServletRequest req, HttpServletResponse rsp) throw
       return;
     }
     Optional<AuthResult> authResult =
-        requestAuthenticator.authorize(route.get().action().auth(), req);
+        requestAuthenticator.authorize(route.get().action().auth().authSettings(), req);
     if (!authResult.isPresent()) {
       rsp.sendError(SC_FORBIDDEN, "Not authorized");
       return;
