diff --git a/old_product.java b/new_product.java
index 4af1399..88a3165 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,6 +8,6 @@ public void validate(Object obj, Errors errors) {
 			// for the following elements Order.hbm.xml says: not-null="true"
 			ValidationUtils.rejectIfEmpty(errors, "prn", "error.null");
 			ValidationUtils.rejectIfEmpty(errors, "complex", "error.null");
-			ValidationUtils.rejectIfEmpty(errors, "drug", "error.null");
+			//ValidationUtils.rejectIfEmpty(errors, "drug", "error.null");
 		}
 	}
\ No newline at end of file
