diff --git a/old_test.java b/new_test.java
index 4e55399..0aa78ff 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
-	@Verifies(value = "should fail validation if drug is null", method = "validate(Object,Errors)")
+	@Verifies(value = "should not fail validation if drug is null", method = "validate(Object,Errors)")
 	public void validate_shouldFailValidationIfDrugIsNull() throws Exception {
 		DrugOrder order = new DrugOrder();
 		
 		Errors errors = new BindException(order, "order");
 		new DrugOrderValidator().validate(order, errors);
 		
-		Assert.assertTrue(errors.hasFieldErrors("drug"));
+		Assert.assertFalse(errors.hasFieldErrors("drug"));
 	}
\ No newline at end of file
