diff --git a/old_product.java b/new_product.java
index f9e1e2c..2803c5a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,20 +1,16 @@
 @RequestMapping(method = RequestMethod.POST, value = "/gateways/receivers", params = "op=start")
   @ResponseBody
   public String startGatewayReceiver(
-      @RequestParam(value = CliStrings.START_GATEWAYRECEIVER__GROUP,
-          required = false) final String[] groups,
-      @RequestParam(value = CliStrings.START_GATEWAYRECEIVER__MEMBER,
-          required = false) final String[] members) {
+      @RequestParam(value = CliStrings.GROUP, required = false) final String[] groups,
+      @RequestParam(value = CliStrings.MEMBER, required = false) final String[] members) {
     CommandStringBuilder command = new CommandStringBuilder(CliStrings.START_GATEWAYRECEIVER);
 
     if (hasValue(groups)) {
-      command.addOption(CliStrings.START_GATEWAYRECEIVER__GROUP,
-          StringUtils.join(groups, StringUtils.COMMA_DELIMITER));
+      command.addOption(CliStrings.GROUP, StringUtils.join(groups, StringUtils.COMMA_DELIMITER));
     }
 
     if (hasValue(members)) {
-      command.addOption(CliStrings.START_GATEWAYRECEIVER__MEMBER,
-          StringUtils.join(members, StringUtils.COMMA_DELIMITER));
+      command.addOption(CliStrings.MEMBER, StringUtils.join(members, StringUtils.COMMA_DELIMITER));
     }
 
     return processCommand(command.toString());
