diff --git a/old_test.java b/new_test.java
index 8bb1fc5..13effa0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void toBlockStoreLocation() {
     StorageTier tier = mDir.getParentTier();
-    assertEquals(new BlockStoreLocation(tier.getTierAlias(), mDir.getDirIndex()),
-        mDir.toBlockStoreLocation());
+    assertEquals(new BlockStoreLocation(tier.getTierAlias(), mDir.getDirIndex(),
+        mDir.getDirMedium()), mDir.toBlockStoreLocation());
   }
\ No newline at end of file
