diff --git a/old_product.java b/new_product.java
index 69af245..647bb7b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 List<WebDAVResource> list(final String db, final String path) throws IOException {
     final WebDAVQuery query = new WebDAVQuery(STRING_JOIN.args(
-      _DB_LIST_DETAILS.args("$db", "$path") + " ! (" +
-      "@raw,@content-type,@modified-date,@size," + SUBSTRING_AFTER.args("text()", "$path") + ')',
+      _DB_LIST_DETAILS.args(" $db", " $path") + " ! (" +
+      "@raw,@content-type,@modified-date,@size," + SUBSTRING_AFTER.args(" text()", " $path") + ')',
       "out:tab()"));
     query.bind("db", db);
     query.bind("path", path);
