diff --git a/old_test.java b/new_test.java
index 4c74b04..34f10ba 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testGetMBR() throws Exception {
-        Circle circle = new Circle(0.0, 0.0, 0.1);
+        Circle circle = new Circle(geomFact.createPoint(new Coordinate(0.0,0.0)), 0.1);
 
-        assertEquals(circle.getMBR().getMinX(), circle.getCenter().getX() - circle.getRadius(), 0.01);
+        assertEquals(circle.getEnvelopeInternal().getMinX(), circle.getCenterPoint().x - circle.getRadius(), 0.01);
     }
\ No newline at end of file
