diff --git a/old_test.java b/new_test.java
index cc0da34..6d3bdbd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	public void horizontal() {
-		Kernel1D_I32 kernel = new Kernel1D_I32(new int[]{1,2,3,4,5,6},4,6);
+		Kernel1D_I32 kernel = new Kernel1D_I32(new int[]{1,2,3,4,5,6}, 6, 4);
 
 		ImageUInt8 input = new ImageUInt8(15,16);
 		ImageMiscOps.fillUniform(input, rand, 0, 50);
