diff --git a/old_test.java b/new_test.java
index e2d2cca..93b2506 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,5 @@
 @Test
     public void testReadPoints() throws Exception {
-        String namedCurve = "secp192r1";
-        List<ICEPoint> result = ICEPointReader.readPoints(namedCurve);
-
+        List<ICEPoint> result = ICEPointReader.readPoints(NamedGroup.SECP192R1);
         assertEquals(5, result.get(0).getOrder());
     }
\ No newline at end of file
