diff --git a/old_product.java b/new_product.java
index 113ac81..a39b656 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public static DependencyGraph generate(Database catalog_db) {
-        DependencyGraph dgraph = new DependencyGraph(catalog_db);
-        DesignerInfo info = new DesignerInfo(catalog_db, new Workload(catalog_db.getCatalog()));
+public static DependencyGraph generate(CatalogContext catalogContext) {
+        DependencyGraph dgraph = new DependencyGraph(catalogContext.database);
+        DesignerInfo info = new DesignerInfo(catalogContext, new Workload(catalogContext.catalog));
         try {
             new DependencyGraphGenerator(info).generate(dgraph);
         } catch (Exception ex) {
