diff --git a/old_product.java b/new_product.java
index c0ccff8..5368464 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,12 @@
 private boolean installJAR(final byte[] content, final String path)
       throws QueryException, IOException {
 
-    final IOContent mf = new IOContent(new Zip(new IOContent(content)).read(MANIFEST_MF));
-    final NewlineInput nli = new NewlineInput(mf);
+    final IOContent manifest = new IOContent(new Zip(new IOContent(content)).read(MANIFEST_MF));
+    final NewlineInput nli = new NewlineInput(manifest);
     for(String s; (s = nli.readLine()) != null;) {
       // write file to rewritten file path
-      final Matcher m = MAIN_CLASS.matcher(s);
-      if(m.find()) return write(m.group(1).replace('.', '/') + IO.JARSUFFIX, content);
+      final Matcher main = MAIN_CLASS.matcher(s);
+      if(main.find()) return write(main.group(1).replace('.', '/') + IO.JARSUFFIX, content);
     }
     throw BXRE_MAIN_X.get(info, path);
   }
\ No newline at end of file
