diff --git a/old_test.java b/new_test.java
index 948b902..a3491fb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,11 @@
     final String path = String.format(CLASSIFY_PATH, classifierId);
 
     server.enqueue(jsonResponse(classification));
-    final Classification result = service.classify(classifierId, classification.getText()).execute();
+    ClassifyOptions classifyOptions = new ClassifyOptions.Builder()
+        .classifierId(classifierId)
+        .text(classification.getText())
+        .build();
+    final Classification result = service.classify(classifyOptions).execute();
     final RecordedRequest request = server.takeRequest();
 
     assertEquals(path, request.getPath());
