diff --git a/old_test.java b/new_test.java
index cba42e1..1f6a7d4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testGetFieldId()
     {
         MetadataValue instance = new MetadataValue();
-        assertThat("testGetFieldId 0", instance.getFieldId(), equalTo(0));
+        assertThat("testGetFieldId 0", instance.getValueId(), equalTo(0));
 
-        assertThat("testGetFieldId 1", mv.getFieldId(), equalTo(mf.getFieldID()));
+        assertThat("testGetFieldId 1", mv.getMetadataField().getFieldID(), equalTo(mf.getFieldID()));
     }
\ No newline at end of file
