diff --git a/old_test.java b/new_test.java
index a603685..1e32292 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,13 +9,15 @@
 
 		Assert.assertEquals(
 				"addr1\naddr2\n121212 City1\nCountry1",
-				builder.buildAddressString(org, location,"de_CH", isLocalAddress, bPartnerBlock, userBlock)
-				);
+				builder
+						.setLanguage("de_CH")
+						.buildAddressString(location, isLocalAddress, bPartnerBlock, userBlock));
 
 		isLocalAddress = true;
 
 		Assert.assertEquals(
 				"LOCAL: addr1\naddr2\n121212 City1",
-				builder.buildAddressString(org, location,"de_CH", isLocalAddress, bPartnerBlock, userBlock)
-				);
+				builder
+						.setLanguage("de_CH")
+						.buildAddressString(location, isLocalAddress, bPartnerBlock, userBlock));
 	}
\ No newline at end of file
