diff --git a/old_product.java b/new_product.java
index 627ab80..29a0ab0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,11 @@
 @SuppressWarnings("unchecked")
 	public List<EncounterType> findEncounterTypes(String name) throws DAOException {
-		return sessionFactory.getCurrentSession().createCriteria(EncounterType.class)
-		// 'ilike' case insensitive search
-		.add(Expression.ilike("name", name, MatchMode.START)).addOrder(Order.asc("name")).addOrder(
-			Order.asc("retired")).list();
+		// define query orders
+		LinkedHashMap<String, String> orders = new LinkedHashMap<String, String>();
+		orders.put("localizedName", "asc");
+		orders.put("retired", "asc");
+		
+		return HibernateUtil.findMetadataInexactlyInLocalizedColumn(name, "name", "localizedName", EncounterType.class,
+		    orders,
+		    sessionFactory);
 	}
\ No newline at end of file
