diff --git a/old_test.java b/new_test.java
index 1aabc6c..5ccd030 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
 
         RunsTable instance = createHorizontalInstance();
         RunsTable expResult = instance;
-        RunsTable result = instance.clone();
+        RunsTable result = instance.copy();
 
         if (!expResult.isIdentical(result)) {
             fail("Clone not identical to original");
