diff --git a/old_product.java b/new_product.java
index 527f15c..d123c49 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,9 @@
-public void add(@NonNull final Candidate candidate)
+public Candidate addOrReplace(@NonNull final Candidate candidate)
 	{
-		final I_MD_Candidate candidadteRecord = retrieveExact(candidate);
-		syncToRecord(candidadteRecord, candidate);
-		InterfaceWrapperHelper.save(candidadteRecord);
+		final Optional<I_MD_Candidate> oldCandidateRecord = retrieveExact(candidate);
+
+		final I_MD_Candidate synchedRecord = syncToRecord(oldCandidateRecord, candidate);
+		InterfaceWrapperHelper.save(synchedRecord);
+
+		return candidate.withId(synchedRecord.getMD_Candidate_ID());
 	}
\ No newline at end of file
