diff --git a/old_test.java b/new_test.java
index 30850c8..a383aad 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,13 @@
 @Test
     public void testSetRole() {
-        // TODO: will be added when CompletableFuture is removed
+        mockAdminService.setRoleSync(anyObject(), anyObject(), anyObject());
+        expectLastCall();
+        replay(mockAdminService);
+
+        final WebTarget wt = target();
+        final InputStream jsonStream = MetersResourceTest.class
+                .getResourceAsStream("put-set-roles.json");
+        final Response response = wt.path("mastership")
+                                    .request().put(Entity.json(jsonStream));
+        assertThat(response.getStatus(), is(HttpURLConnection.HTTP_OK));
     }
\ No newline at end of file
