diff --git a/old_test.java b/new_test.java
index 0f70e81..de0ce7d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,8 +20,7 @@
                                                                         rebalancePartitionList,
                                                                         rebalancePartitionList,
                                                                         Arrays.asList(testStoreName),
-                                                                        0,
-                                                                        true);
+                                                                        0);
         int asyncId = adminClient.rebalanceNode(stealInfo);
         assertNotSame("Got a valid rebalanceAsyncId", -1, asyncId);
 
