diff --git a/old_product.java b/new_product.java
index 7d9eb4a..48557ba 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,8 +3,9 @@ public static Dataset generateDataset(CharSequence descriptor,
                                         String[] data) throws DescriptorException {
     Attribute[] attrs = DescriptorUtils.parseDescriptor(descriptor);
     
-    // used to convert CATEGORICAL and LABEL attributes to Integer
-    List<String>[] values = new List[attrs.length];
+    // used to convert CATEGORICAL attributes to Integer
+    @SuppressWarnings("unchecked")
+    Set<String>[] valsets = new Set[attrs.length];
     
     int size = 0;
     for (String aData : data) {
@@ -12,10 +13,18 @@ public static Dataset generateDataset(CharSequence descriptor,
         continue;
       }
       
-      if (parseString(attrs, values, aData) != null) {
+      if (parseString(attrs, valsets, aData, regression)) {
         size++;
       }
     }
+
+    @SuppressWarnings("unchecked")
+    List<String>[] values = new List[attrs.length];
+    for (int i = 0; i < valsets.length; i++) {
+      if (valsets[i] != null) {
+        values[i] = Lists.newArrayList(valsets[i]);
+      }
+    }
     
     return new Dataset(attrs, values, size, regression);
   }
\ No newline at end of file
