diff --git a/old_test.java b/new_test.java
index 741441d..745ca39 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,4 +16,14 @@
     Dataset dataset = DataLoader.generateDataset(descriptor, false, sData);
     
     assertEquals(expected, dataset);
-  }
\ No newline at end of file
+
+    // regression
+    data = Utils.randomDoubles(rng, descriptor, true, datasize);
+    missings = Lists.newArrayList();
+    sData = prepareData(data, attrs, missings);
+    expected = DataLoader.generateDataset(descriptor, true, sData);
+
+    dataset = DataLoader.generateDataset(descriptor, true, sData);
+    
+    assertEquals(expected, dataset);
+}
\ No newline at end of file
