diff --git a/old_test.java b/new_test.java
index 403f155..d18fc08 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,10 +20,10 @@
 		for (Key<Trivial> k: q.keys())
 			assert keys.contains(k);
 		
-		Key<Trivial> first = q.keysOnly().first().key();
+		Key<Trivial> first = q.keys().first().key();
 		assert first.equals(this.keys.get(0));
 		
 		q = q.offset(1);
-		Key<Trivial> second = q.keysOnly().first().key();
+		Key<Trivial> second = q.keys().first().key();
 		assert second.equals(this.keys.get(1));
 	}
\ No newline at end of file
