diff --git a/old_test.java b/new_test.java
index 25914f7..fe4df77 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void test_LatLongToPixelXY() {
-        final PointL point = TileSystem.getMercatorFromGeo(60, 60, TileSystem.MapSize((double)10), null, true);
+        final PointL point = tileSystem.getMercatorFromGeo(60, 60, TileSystem.MapSize((double)10), null, true);
         Assert.assertEquals(174762, point.x);
         Assert.assertEquals(76126, point.y);
     }
\ No newline at end of file
