diff --git a/old_product.java b/new_product.java
index 64ee135..84fbb92 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ private void train(Vec[] featureVals, double[] target)
         final double[] z = new double[m];
         final double beta = loss.beta();
 
-        Random rand = new Random();
+        Random rand = RandomUtil.getRandom();
         for (int t = 1; t <= epochs; t++)
         {
             int j = rand.nextInt(d + 1);//+1 for the bias term
