diff --git a/old_test.java b/new_test.java
index 506b4fa..8bd3032 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
     new Set(Prop.ATTRINDEX, false).execute(CONTEXT);
 
     // Create test database
-    new CreateDB(DBNAME, "<X><A>1</A><A>1</A></X>").execute(CONTEXT);
+    new CreateDB(DB, "<X><A>1</A><A>1</A></X>").execute(CONTEXT);
 
     long len1 = CONTEXT.data().meta.dbfile(DataText.DATATXT).length();
 
@@ -26,6 +26,6 @@
     assertEquals(len1, len2);
 
     // Drop database
-    new DropDB(DBNAME).execute(CONTEXT);
+    new DropDB(DB).execute(CONTEXT);
     CONTEXT.close();
   }
\ No newline at end of file
