diff --git a/old_product.java b/new_product.java
index e5c91c9..0cd97f7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 private void run(final int clients, final int runs) throws Exception {
     // Create test database
-    Command cmd = new CreateDB(DBNAME, INPUT);
+    Command cmd = new CreateDB(DB, INPUT);
     cmd.execute(CONTEXT);
 
     // Start clients
@@ -9,6 +9,6 @@ private void run(final int clients, final int runs) throws Exception {
     for(final Client c : cl) c.start();
     for(final Client c : cl) c.join();
     // Drop database
-    cmd = new DropDB(DBNAME);
+    cmd = new DropDB(DB);
     cmd.execute(CONTEXT);
   }
\ No newline at end of file
