diff --git a/old_test.java b/new_test.java
index e2dc206..aba8ad6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,24 +3,30 @@
         System.out.println("getGrblStatusCapabilities");
         double version;
         String letter;
-        Capabilities expResult;
+        Capabilities expResult = new Capabilities();
         Capabilities result;
 
         version = 0.8;
         letter = "c";
-        expResult = Capabilities.STATUS_C;
         result = GrblUtils.getGrblStatusCapabilities(version, letter);
-        assertEquals(expResult, result);
+        assertEquals(true, result.REAL_TIME);
+        assertEquals(false, result.OVERRIDES);
         
         version = 0.8;
         letter = "a";
-        expResult = null;
         result = GrblUtils.getGrblStatusCapabilities(version, letter);
-        assertEquals(expResult, result);
+        assertEquals(false, result.REAL_TIME);
+        assertEquals(false, result.OVERRIDES);
         
         version = 0.9;
         letter = null;
-        expResult = Capabilities.STATUS_C;
         result = GrblUtils.getGrblStatusCapabilities(version, letter);
-        assertEquals(expResult, result);
+        assertEquals(true, result.REAL_TIME);
+        assertEquals(false, result.OVERRIDES);
+
+        version = 1.0;
+        letter = null;
+        result = GrblUtils.getGrblStatusCapabilities(version, letter);
+        assertEquals(true, result.REAL_TIME);
+        assertEquals(true, result.OVERRIDES);
     }
\ No newline at end of file
