diff --git a/old_product.java b/new_product.java
index 1bcad9f..882aaca 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,17 @@
 static protected Capabilities getGrblStatusCapabilities(final double version, final String letter) {
-        if (version >= 0.8) {
+        Capabilities ret = new Capabilities();
+
+        if (version >= 1.0) {
+            ret.REAL_TIME = true;
+            ret.OVERRIDES = true;
+        }
+
+        else if (version >= 0.8) {
             if (version==0.8 && (letter != null) && letter.equals("c")) {
-                return Capabilities.STATUS_C;
+                ret.REAL_TIME = true;
             } else if (version >= 0.9) {
-                return Capabilities.STATUS_C;
+                ret.REAL_TIME = true;
             }
         }
-        return null;
+        return ret;
     }
\ No newline at end of file
