diff --git a/old_product.java b/new_product.java
index e6f4f48..2971282 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public default long toEpochSecond(ZoneOffset offset) {
+default long toEpochSecond(ZoneOffset offset) {
         Objects.requireNonNull(offset, "offset");
         long epochDay = toLocalDate().toEpochDay();
         long secs = epochDay * 86400 + toLocalTime().toSecondOfDay();
