diff --git a/old_test.java b/new_test.java
index fe5325c..31d04f8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,7 +20,7 @@
 			initial.distortion[i] = rand.nextGaussian()*expected.distortion[i]*0.1;
 		}
 
-		assertTrue(CalibrationPlanarGridZhang99.optimizedParam(observations, grid, true, initial, found));
+		assertTrue(CalibrationPlanarGridZhang99.optimizedParam(observations, grid, true, initial, found,null));
 
 		checkEquals(expected, found, initial);
 	}
\ No newline at end of file
