diff --git a/old_product.java b/new_product.java
index 8102660..450d061 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public void updateIndexFromWeb() throws MalformedURLException, ParserConfigurationException, SAXException, IOException {
+public void update() throws MalformedURLException, ParserConfigurationException, SAXException, IOException {
         long timeStamp = updateNeeded();
         if (timeStamp > 0) {
             URL url = new URL(Settings.getString(Settings.KEYS.CPE_URL));
