diff --git a/old_test.java b/new_test.java
index 9f04b2c..d05f320 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,6 +7,6 @@
     public void test_hashCode() throws Exception {
         URL url = new java.net.URL("file:///test");
         CodeSource cs = new CodeSource(url, (Certificate[]) null);
-        assertTrue("Did not get expected hashCode!", cs.hashCode() == url
+        assertEquals("Did not get expected hashCode!", cs.hashCode(), url
                 .hashCode());
     }
\ No newline at end of file
