diff --git a/old_product.java b/new_product.java
index c7ecd92..95462e7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static void attachInvocationIdIfAsync(URL url, Invocation inv) {
         if (isAttachInvocationId(url, inv) && getInvocationId(inv) == null && inv instanceof RpcInvocation) {
-            ((RpcInvocation) inv).setAttachment(Constants.ID_KEY, String.valueOf(INVOKE_ID.getAndIncrement()));
+            ((RpcInvocation) inv).setAttachment(ID_KEY, String.valueOf(INVOKE_ID.getAndIncrement()));
         }
     }
\ No newline at end of file
