diff --git a/old_product.java b/new_product.java
index b9be6e7..4b78b3b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,12 @@
 public double logp(int[] o) {
-        double[][] alpha = new double[o.length][numStates];
+        double[][] alpha = new double[o.length][a.nrows()];
         double[] scaling = new double[o.length];
 
         forward(o, alpha, scaling);
 
         double p = 0.0;
         for (int t = 0; t < o.length; t++) {
-            p += java.lang.Math.log(scaling[t]);
+            p += Math.log(scaling[t]);
         }
 
         return p;
