diff --git a/old_test.java b/new_test.java
index 4c9afec..a6ac8c6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,8 @@
 @Test
-    public void testLogp_intArr2() {
+    public void testLogp() {
         System.out.println("logp");
-        String[] symbols = {"0", "1"};
-        HMM<String> hmm = new HMM<>(pi, a, b, symbols);
-
-        String[] o = {"0", "0", "1", "1", "0", "1", "1", "0"};
+        HMM hmm = new HMM(pi, Matrix.of(a), Matrix.of(b));
+        int[] o = {0, 0, 1, 1, 0, 1, 1, 0};
         double expResult = -5.609373;
         double result = hmm.logp(o);
         assertEquals(expResult, result, 1E-6);
