diff --git a/old_test.java b/new_test.java
index f80bb19..b48d49a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,9 +4,9 @@
     Array seq2 = Array.empty();
     final int n = 200_000;
     for(int i = 0; i < n; i++) {
-      final Value val = Int.get(i);
-      seq1 = seq1.cons(val);
-      seq2 = seq2.snoc(val);
+      final Value value = Int.get(i);
+      seq1 = seq1.cons(value);
+      seq2 = seq2.snoc(value);
     }
 
     assertEquals(n, seq1.arraySize());
