diff --git a/old_product.java b/new_product.java
index a888f26..510e22b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,9 @@
-public static String getCategory(int type) {
-    int i, n;
-    for (i = 0, n = CATEGORY_VALUES.length - 1; i < n; ++i) {
-      if (CATEGORY_VALUES[i] == type) break;
+@Nullable
+  public static String getCategory(int type) {
+    for (int i = 0, n = CATEGORY_VALUES.length - 1; i < n; ++i) {
+      if (CATEGORY_VALUES[i] == type) {
+        return CATEGORY_STRINGS[i][0];
+      }
     }
-    return CATEGORY_STRINGS[i][0];
+    return null;
   }
\ No newline at end of file
