diff --git a/old_product.java b/new_product.java
index 62eb30a..618a26d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,7 +9,10 @@ protected XDOM buildPresentationXDOM(String html, DocumentReference targetDocume
             ExpandedMacroBlock gallery = new ExpandedMacroBlock("gallery", galleryParameters, renderer, false);
             gallery.addChild(this.xhtmlParser.parse(new StringReader(html)));
 
-            return new XDOM(Collections.singletonList((Block) gallery));
+            XDOM xdom = new XDOM(Collections.singletonList((Block) gallery));
+            // Make sure (image) references are resolved relative to the target document reference.
+            xdom.getMetaData().addMetaData(MetaData.BASE, entityReferenceSerializer.serialize(targetDocumentReference));
+            return xdom;
         } catch (Exception e) {
             throw new OfficeImporterException("Failed to build presentation XDOM.", e);
         }
