diff --git a/old_test.java b/new_test.java
index 6c0bfa8..344dc09 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void testSize() throws Exception {
-    assertEquals("Unexpected size!", size, data.size);
+    assertEquals("Unexpected size!", size, data.meta.size);
     reload();
-    assertEquals("Unexpected size!", size, data.size);
+    assertEquals("Unexpected size!", size, data.meta.size);
   }
\ No newline at end of file
