diff --git a/old_product.java b/new_product.java
index 7be0fd9..98be1bb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,6 +2,6 @@
         L modelAns = model.predict(vector.features());
         L truth = vector.label();
         if (modelAns.equals(truth))
-            validAnswersCount++;
+            validAnswersCnt++;
         totalNumberOfExamples++;
     }
\ No newline at end of file
