diff --git a/old_test.java b/new_test.java
index d8393ed..3913ee3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Ignore
 	@Test
-	public void test2LoadModel() throws ClassNotFoundException, LibrecException, IOException {
+	public void test02LoadModel() throws ClassNotFoundException, LibrecException, IOException {
 		Configuration conf = new Configuration();
 		Configuration.Resource resource = new Resource("rec/cf/userknn-test.properties");
 		conf.addResource(resource);
