diff --git a/old_product.java b/new_product.java
index 1587618..0e72583 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1,2 @@
-LockBlockResult lockBlock(final long blockId) throws IOException, AlluxioException;
\ No newline at end of file
+LockBlockResource lockBlock(final long blockId, final LockBlockOptions options)
+      throws IOException, AlluxioException;
\ No newline at end of file
