diff --git a/old_product.java b/new_product.java
index c457f1c..b53f98f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public List<Scorecard> getScorecards() {
-    Response response = execute(RequestBuilder.get(PATH_SCORECARD).build());
+    final Response response = execute(RequestBuilder.get(PATH_SCORECARD).build());
     final JsonObject jsonObject = ResponseUtil.getJsonObject(response);
     final List<Scorecard> scorecards =
         GsonSingleton.getGson().fromJson(jsonObject.get(SCORECARDS), scorecardListType);
