diff --git a/old_product.java b/new_product.java
index a9b796f..2f8bfba 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@ public void prunePoints(int count ) {
 			SceneObservations.View v = observations.views[viewIndex];
 
 			for(int pointIndex = v.point.size-1; pointIndex >= 0; pointIndex-- ) {
-				SceneStructureMetric.Point p = structure.points[v.getPointId(pointIndex)];
+				SceneStructureMetric.Point p = structure.points.data[v.getPointId(pointIndex)];
 
 				if( p.views.size < count ) {
 					v.remove(pointIndex);
@@ -13,12 +13,12 @@ public void prunePoints(int count ) {
 		}
 
 		// Create a look up table containing from old to new indexes for each point
-		int oldToNew[] = new int[ structure.points.length ];
+		int[] oldToNew = new int[ structure.points.size ];
 		Arrays.fill(oldToNew,-1); // crash is bug
 
 		GrowQueue_I32 prune = new GrowQueue_I32(); // List of point ID's which are to be removed.
-		for (int i = 0; i < structure.points.length; i++) {
-			if( structure.points[i].views.size < count ) {
+		for (int i = 0; i < structure.points.size; i++) {
+			if( structure.points.data[i].views.size < count ) {
 				prune.add(i);
 			} else {
 				oldToNew[i] = i-prune.size;
