diff --git a/old_test.java b/new_test.java
index 8446dbd..b7b8481 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
-	public void prunePoints_neighbors() {
+	void prunePoints_neighbors() {
 		createPerfectScene();
-		int countPoints0 = structure.points.length;
+		int countPoints0 = structure.points.size;
 		int countObservations0 = observations.getObservationCount();
 
 		PruneStructureFromSceneMetric alg = new PruneStructureFromSceneMetric(structure,observations);
@@ -9,14 +9,14 @@
 		// This should just prune the outliers far from the center
 		alg.prunePoints(2,0.5);
 
-		int countPoints1 = structure.points.length;
+		int countPoints1 = structure.points.size;
 		int countObservations1 = observations.getObservationCount();
 		assertTrue(countPoints0>countPoints1 && countPoints1>0.95*countPoints0);
 		assertTrue(countObservations0>countObservations1 && countObservations1>0.95*countObservations0);
 
 		// If run a second time it should have very similar results
 		alg.prunePoints(2,0.5);
-		assertEquals(countPoints1, structure.points.length,5);
+		assertEquals(countPoints1, structure.points.size,5);
 		assertEquals(countObservations1, observations.getObservationCount(),countObservations1*0.005);
 
 		// sanity check the modifications
