diff --git a/old_product.java b/new_product.java
index 748b038..0167c48 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,5 @@
 public static <E> Iterable<E> filter(
       final Iterable<?> iterable,
       final Class<E> includeFilter) {
-    return new Iterable<E>() {
-      public Iterator<E> iterator() {
-        return new Filterator<>(iterable.iterator(), includeFilter);
-      }
-    };
+    return () -> new Filterator<>(iterable.iterator(), includeFilter);
   }
\ No newline at end of file
