diff --git a/old_test.java b/new_test.java
index 373e065..9d462eb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,30 +2,15 @@
     final List<String> abc = Arrays.asList("A", "B", "C", "D");
     // a miss, then a hit
     Assert.assertEquals("[A, C, D]",
-        Functions.filter(abc,
-            new Predicate1<String>() {
-              public boolean apply(String v1) {
-                return !v1.equals("B");
-              }
-            }).toString());
+        Functions.filter(abc, v1 -> !v1.equals("B")).toString());
     // a hit, then all misses
     Assert.assertEquals("[A]",
-        Functions.filter(abc,
-            new Predicate1<String>() {
-              public boolean apply(String v1) {
-                return v1.equals("A");
-              }
-            }).toString());
+        Functions.filter(abc, v1 -> v1.equals("A")).toString());
     // two hits, then a miss
     Assert.assertEquals("[A, B, D]",
-        Functions.filter(abc,
-            new Predicate1<String>() {
-              public boolean apply(String v1) {
-                return !v1.equals("C");
-              }
-            }).toString());
+        Functions.filter(abc, v1 -> !v1.equals("C")).toString());
     Assert.assertSame(Collections.emptyList(),
-        Functions.filter(abc, Functions.<String>falsePredicate1()));
+        Functions.filter(abc, Functions.falsePredicate1()));
     Assert.assertSame(abc,
-        Functions.filter(abc, Functions.<String>truePredicate1()));
+        Functions.filter(abc, Functions.truePredicate1()));
   }
\ No newline at end of file
