diff --git a/old_test.java b/new_test.java
index 68cc7a4..3a5952d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
 
       Job job = Job.getInstance();
       AccumuloInputFormat.setInputTableName(job, table);
-      AccumuloInputFormat.setClientInfo(job, getClientInfo());
+      AccumuloInputFormat.setClientProperties(job, getClientProperties());
 
       // split table
       TreeSet<Text> splitsToAdd = new TreeSet<>();
