diff --git a/old_test.java b/new_test.java
index 0cd6255..a445c47 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,16 +20,21 @@
         ComputeUtils.<Integer, Integer, Integer>initContext(
             ignite,
             upstreamCacheName,
+            UpstreamTransformerBuilder.identity(),
             (k, v) -> true,
             datasetCacheName,
-            (upstream, upstreamSize) -> {
+            (env, upstream, upstreamSize) -> {
 
                 assertEquals(1, upstreamSize);
 
                 UpstreamEntry<Integer, Integer> e = upstream.next();
                 return e.getKey() + e.getValue();
             },
-            0
+            TestUtils.testEnvBuilder(),
+            0,
+            0,
+            false,
+            DeployingContext.unitialized()
         );
 
         assertEquals(1, datasetCache.size());
