diff --git a/old_product.java b/new_product.java
index 74c55c2..f7acbab 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 public CreateDirectoryTOptions toThrift() {
     CreateDirectoryTOptions options = new CreateDirectoryTOptions();
     options.setAllowExists(mAllowExists);
+    options.setMode(mMode.toShort());
     options.setRecursive(mRecursive);
     options.setPersisted(mUnderStorageType.isSyncPersist());
     return options;
