diff --git a/old_test.java b/new_test.java
index ad7a072..03e4435 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,4 +4,6 @@
     query(_FILE_READ_TEXT.args(FILE), "<");
     query(func.args(_FILE_READ_BINARY.args(FILE)), "<");
     query(func.args(_FILE_READ_TEXT.args(FILE)), "<");
+    query(func.args(_FILE_READ_BINARY.args(FILE), " true()"), "<");
+    query(func.args(_FILE_READ_TEXT.args(FILE), " true()"), "<");
   }
\ No newline at end of file
