diff --git a/old_product.java b/new_product.java
index 36ea3c9..dec76d8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -30,7 +30,7 @@ Value cache(final int max) throws QueryException {
 
     // use standard iterator
     while((item = next(iter)) != null && items.size() < mx) {
-      item.cache(null);
+      item.cache(null, false);
       items.add(item);
     }
     return items.value();
