diff --git a/old_test.java b/new_test.java
index e33a1eb..70ba278 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
     public void labelInSortedOrderTest() {
-        assertEquals(features.length, index.rowsCount());
-        assertEquals(features[0].length, index.columnsCount());
+        assertEquals(features.length, idx.rowsCount());
+        assertEquals(features[0].length, idx.columnsCount());
 
-        for (int k = 0; k < index.rowsCount(); k++) {
-            for (int featureId = 0; featureId < index.columnsCount(); featureId++)
-                assertEquals(labelsInSortedOrder[k][featureId], index.labelInSortedOrder(k, featureId), 0.01);
+        for (int k = 0; k < idx.rowsCount(); k++) {
+            for (int featureId = 0; featureId < idx.columnsCount(); featureId++)
+                assertEquals(labelsInSortedOrder[k][featureId], idx.labelInSortedOrder(k, featureId), 0.01);
         }
     }
\ No newline at end of file
