diff --git a/old_test.java b/new_test.java
index 0f74098..ff3e62b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,9 @@
-@TestInfo(
-      level = TestLevel.COMPLETE,
-      purpose = "",
-      targets = {
-        @TestTarget(
-          methodName = "getTimestamp",
-          methodArgs = {}
-        )
-    })
+@TestTargetNew(
+        level = TestLevel.COMPLETE,
+        notes = "",
+        method = "getTimestamp",
+        args = {}
+    )
     public void testGetTimestamp() {
         assertNull(new CodeSigner(cpath, null).getTimestamp());
         assertSame(new CodeSigner(cpath, ts).getTimestamp(), ts);
