diff --git a/old_product.java b/new_product.java
index 002ea4e..069534b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,5 @@
 public void createDB(final String name) {
     final CreateDB c = new CreateDB(name, "<" + DeepFS.S_FSML + "/>");
-    if(!c.run(ctx)) Util.notexpected(
-        "Failed to create file system database (%).", c.info());
+    if(!c.run(ctx)) Util.notexpected(c.info());
     ctx.data.meta.deepfs = true;
   }
\ No newline at end of file
