diff --git a/old_test.java b/new_test.java
index f088f0e..3fab611 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
                     s.seal("foo", TIMEOUT),
                     ex -> ex instanceof StreamSegmentNotExistsException);
 
-            HashMap<String, ByteArrayOutputStream> appendData = populate(s, 10, 10);
+            HashMap<String, ByteArrayOutputStream> appendData = populate(s);
             for (String segmentName : appendData.keySet()) {
                 s.seal(segmentName, TIMEOUT).join();
                 AssertExtensions.assertThrows(
