diff --git a/old_test.java b/new_test.java
index fb4f624..a29d6e0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
 	public void testSetInput() {
 		final Operator input1 = new OpImpl(0);
 		final Operator input2 = new OpImpl(1);
-		final Operator fixture = new OpImpl(0, input1, input2);
+		final Operator fixture = new OpImpl(0).withInputs(input1, input2);
 
 		final Operator newInput2 = new OpImpl(2);
 
