diff --git a/old_test.java b/new_test.java
index aee2cab..73afbc1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,9 +4,9 @@
 		executeDataSet(XML_FILENAME);
 		Context.logout();
 		Context.authenticate("incorrectlyhashedSha1", "test");
-		
+
 		UserService us = Context.getUserService();
-		us.changePassword("test", "test2");
-		
+		us.changePassword("test", "Tester12");
+
 		Context.logout(); // so that the next test reauthenticates
 	}
\ No newline at end of file
