diff --git a/old_test.java b/new_test.java
index 6da363a..e52aac8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
     public void test_eraOf() {
-        assertEquals(IsoChronology.INSTANCE.eraOf(0), ERA_BCE);
-        assertEquals(IsoChronology.INSTANCE.eraOf(1), ERA_CE);
+        assertEquals(IsoChronology.INSTANCE.eraOf(0), IsoEra.BCE);
+        assertEquals(IsoChronology.INSTANCE.eraOf(1), IsoEra.CE);
     }
\ No newline at end of file
