diff --git a/old_product.java b/new_product.java
index 80ece19..b837cc1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public HijrahDate withDayOfYear(int dayOfYear) {
         HijrahChronology.dayOfYearRule().checkValue(dayOfYear);
-        return HijrahDate.hijrahDate(this.era, this.yearOfEra, 1, 1).plusDays(dayOfYear).plusDays(-1);
+        return HijrahDate.of(this.era, this.yearOfEra, 1, 1).plusDays(dayOfYear).plusDays(-1);
     }
\ No newline at end of file
