diff --git a/old_test.java b/new_test.java
index 8f59698..17369d4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
     public void testWithDayOfYear() {
         JapaneseDate date = testDate.withDayOfYear(15);
-        assertEquals(date, JapaneseDate.japaneseDate(testYear, MonthOfYear.JANUARY, 15));
+        assertEquals(date, JapaneseDate.of(testYear, MonthOfYear.JANUARY, 15));
     }
\ No newline at end of file
