diff --git a/old_product.java b/new_product.java
index df9baea..b5f6525 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
-private static void optimize(final IndexType type, final Data d,
-      final boolean create, final boolean old, final Optimize c) throws IOException {
+private static void optimize(final IndexType type, final Data d, final boolean create,
+      final boolean old, final boolean rebuild, final Optimize cmd) throws IOException {
 
     // check if flags are nothing has changed
-    if(create == old) return;
+    if(!rebuild && create == old) return;
 
     // create or drop index
-    if(create) create(type, d, c);
+    if(create) create(type, d, cmd);
     else drop(type, d);
   }
\ No newline at end of file
