diff --git a/old_product.java b/new_product.java
index 5b96fa9..3d684c3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,11 @@
 public ServiceCall<DialogNode> updateDialogNode(UpdateDialogNodeOptions updateDialogNodeOptions) {
     Validator.notNull(updateDialogNodeOptions, "updateDialogNodeOptions cannot be null");
     RequestBuilder builder = RequestBuilder.post(String.format("/v1/workspaces/%s/dialog_nodes/%s",
-        updateDialogNodeOptions.workspaceId(), updateDialogNodeOptions.dialogNode()));
+            updateDialogNodeOptions.workspaceId(), updateDialogNodeOptions.dialogNode()));
     builder.query(VERSION, versionDate);
     final JsonObject contentJson = new JsonObject();
-    if (updateDialogNodeOptions.newType() != null) {
-      contentJson.addProperty("type", updateDialogNodeOptions.newType());
+    if (updateDialogNodeOptions.nodeType() != null) {
+      contentJson.addProperty("type", updateDialogNodeOptions.nodeType());
     }
     if (updateDialogNodeOptions.newActions() != null) {
       contentJson.add("actions", GsonSingleton.getGson().toJsonTree(updateDialogNodeOptions.newActions()));
