diff --git a/old_test.java b/new_test.java
index 97c8a9e..73549df 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
     String dialogNodeDescription = "Description of " + dialogNodeName;
 
     CreateDialogNodeOptions createOptions =
-        new CreateDialogNodeOptions.Builder(workspaceId, dialogNodeName).description(dialogNodeDescription).build();
+            new CreateDialogNodeOptions.Builder(workspaceId, dialogNodeName).description(dialogNodeDescription).build();
     service.createDialogNode(createOptions).execute();
 
     String dialogNodeName2 = "Test2" + UUID.randomUUID().toString();
@@ -13,12 +13,12 @@
       String dialogNodeDescription2 = "Updated description of " + dialogNodeName;
       Date start = new Date();
       UpdateDialogNodeOptions updateOptions =
-          new UpdateDialogNodeOptions.Builder(workspaceId, dialogNodeName, dialogNodeName2)
-              .newDescription(dialogNodeDescription2).build();
+              new UpdateDialogNodeOptions.Builder(workspaceId, dialogNodeName, dialogNodeName2)
+                      .newDescription(dialogNodeDescription2).build();
       DialogNode response = service.updateDialogNode(updateOptions).execute();
       assertNotNull(response);
-      assertNotNull(response.getDialogNode());
-      assertEquals(response.getDialogNode(), dialogNodeName2);
+      assertNotNull(response.getDialogNodeId());
+      assertEquals(response.getDialogNodeId(), dialogNodeName2);
       assertNotNull(response.getDescription());
       assertEquals(response.getDescription(), dialogNodeDescription2);
       assertNotNull(response.getCreated());
