diff --git a/old_product.java b/new_product.java
index 9481955..3a4eccb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -27,7 +27,7 @@ public static String getLicenseText(Locale locale, Collection collection,
             }
         }
 
-        String licenseTemplate = collection.getLicense();
+        String licenseTemplate = collectionService.getLicense(collection);
 
         return formatter.format(licenseTemplate, args).toString();
     }
\ No newline at end of file
