diff --git a/old_product.java b/new_product.java
index 1edd1cf..f0c3517 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public <E> CriteriaQuery between(String propName, Comparable<E> from, Comparable<E> to) {
-        addCriterion(criterionBuilder.between(propName, from, to));
+        criterion = criterion.and(criterionBuilder.between(propName, from, to));
         return this;
     }
\ No newline at end of file
