diff --git a/old_test.java b/new_test.java
index 269c8b0..8afaaac 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testBetween() {
-        assertTrue(instance.between("name", "a", "b").getQueryCriterions()
-                .contains(new BetweenCriterion("name", "a", "b")));
+    	assertEquals(new BetweenCriterion("name", "a", "b"), instance.between("name", "a", "b").getQueryCriterion());
     }
\ No newline at end of file
