diff --git a/old_product.java b/new_product.java
index 1c8e760..a60353c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public LocalDateTime withMinuteOfHour(int minuteOfHour) {
-        LocalTime newTime = time.withMinuteOfHour(minuteOfHour);
+public LocalDateTime withMinute(int minute) {
+        LocalTime newTime = time.withMinute(minute);
         return with(date, newTime);
     }
\ No newline at end of file
