diff --git a/old_product.java b/new_product.java
index f387929..31aca48 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
-@RpcMethod
-	public void acceptOffers(AcceptOffers msg) {
+public void acceptOffers(AcceptOffers msg) {
 		try {
 			List<TaskMonitor.TaskGoalStateUpdated> toMonitor = new ArrayList<>(msg.operations().size());
 
@@ -29,8 +28,7 @@
 
 			// send the acceptance message to Mesos
 			schedulerDriver.acceptOffers(msg.offerIds(), msg.operations(), msg.filters());
-		}
-		catch(Exception ex) {
+		} catch (Exception ex) {
 			onFatalError(new ResourceManagerException("unable to accept offers", ex));
 		}
 	}
\ No newline at end of file
