diff --git a/old_product.java b/new_product.java
index a9c5168..d17e858 100644
--- a/old_product.java
+++ b/new_product.java
@@ -44,23 +44,23 @@ public boolean init(String[] args) throws ParseException, IOException, Exception
     //b.addFromProperties(tplgProperties);
     //this.topology = b.getTopology();
 
-    FileInputStream fis = new FileInputStream("./" + Topology.SER_FILE_NAME);
-    this.topology = Topology.read(fis);
+    FileInputStream fis = new FileInputStream("./" + DAG.SER_FILE_NAME);
+    this.topology = DAG.read(fis);
     fis.close();
     // "debug" simply dumps all data using LOG.info
     if (topology.isDebug()) {
       dumpOutDebugInfo();
     }
 
-    this.dnmgr = new DNodeManager(topology);
+    this.dnmgr = new ModuleManager(topology);
 
     // start RPC server
-    rpcImpl = new StreamingNodeParent(this.getClass().getName(), dnmgr);
+    rpcImpl = new StreamingContainerParent(this.getClass().getName(), dnmgr);
     rpcImpl.init(conf);
     rpcImpl.start();
     LOG.info("Container callback server listening at " + rpcImpl.getAddress());
 
-    LOG.info("Initializing logical topology with {} nodes in {} containers", topology.getAllNodes().size(), dnmgr.getNumRequiredContainers());
+    LOG.info("Initializing logical topology with {} nodes in {} containers", topology.getAllOperators().size(), dnmgr.getNumRequiredContainers());
 
     // start buffer server
     com.malhartech.bufferserver.Server s = new Server(0);
