diff --git a/old_test.java b/new_test.java
index effead6..c32580a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testOnDecline() {
-        Result result = new Result(false, false, 0, 2, null, null, null);
+        AgentResult result = new AgentResult(false, false, 0, 2, null, null, null, null);
         assertNull(result.isGoodTrace());
         rule.onDecline(result);
         assertFalse(result.isGoodTrace());
