diff --git a/old_test.java b/new_test.java
index a9488c7..e97c6bb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,5 +9,5 @@
         jcommander.parse(args);
         delegate.applyDelegate(config);
         assertTrue(config.getHost().equals("123456"));
-        assertTrue(config.getConnectionEndType() == ConnectionEndType.CLIENT);
+        assertTrue(config.getDefaultConnectionEndType() == ConnectionEndType.CLIENT);
     }
\ No newline at end of file
