diff --git a/old_product.java b/new_product.java
index 6806732..44b92da 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,7 @@ protected boolean requestSchedulerService(Command command, byte[] data) {
     // now, we have a valid connection
     try {
       // send the actual http request
-      if (!HttpUtils.sendHttpPostRequest(connection, data)) {
+      if (!NetworkUtils.sendHttpPostRequest(connection, data)) {
         LOG.log(Level.SEVERE, "Failed to send http request to scheduler");
         return false;
       }
@@ -19,7 +19,7 @@ protected boolean requestSchedulerService(Command command, byte[] data) {
       LOG.fine("Receiving response from scheduler...");
       try {
         statusCode = Scheduler.SchedulerResponse.newBuilder()
-            .mergeFrom(HttpUtils.readHttpResponse(connection))
+            .mergeFrom(NetworkUtils.readHttpResponse(connection))
             .build().getStatus().getStatus();
       } catch (InvalidProtocolBufferException e) {
         LOG.log(Level.SEVERE, "Failed to parse response", e);
