diff --git a/old_test.java b/new_test.java
index 8eb9de5..2ba82b7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,15 @@
 @Test
 	public void testConnect() throws SQLException {
 		final Properties info = new Properties();
-		assertNull("connect", driver.connect(null, info));
-		info.put("driver", "sun.jdbc.odbc.JdbcOdbcDriver");
 		try {
 			driver.connect(null, info);
 		} catch (final SQLException e) {
 			// SQLException normale : The url cannot be null
 			assertNotNull("connect", e);
 		}
+		driver.connect("jdbc:h2:mem:?driver=org.h2.Driver", info);
+		info.put("driver", "org.h2.Driver");
+		driver.connect("jdbc:h2:mem:", info);
 		info.put("driver", "nimporte.quoi");
 		try {
 			driver.connect(null, info);
