diff --git a/old_product.java b/new_product.java
index 913ed88..ffcb470 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,16 @@
 @NotNull
-    public static LinkedHashMap<VariableElement, TypeMirror> getConcreteMembers(@NotNull TypeMirror concreteInherited,
-                                                                        @NotNull TypeElement genericInherited,
-                                                                        @NotNull Map<VariableElement, TypeMirror> members) {
+    public static LinkedHashMap<FieldAccessor, TypeMirror> getConcreteMembers(@NotNull TypeMirror concreteInherited,
+                                                                              @NotNull TypeElement genericInherited,
+                                                                              @NotNull Map<FieldAccessor, TypeMirror> members) {
 
         DebugLog.log(TAG, "Inherited concrete type: " + concreteInherited.toString());
         DebugLog.log(TAG, "Inherited generic type: " + genericInherited.asType().toString());
         List<? extends TypeMirror> concreteTypes = getParameterizedTypes(concreteInherited);
         List<? extends TypeMirror> inheritedTypes = getParameterizedTypes(genericInherited);
 
-        LinkedHashMap<VariableElement, TypeMirror> map = new LinkedHashMap<>();
+        LinkedHashMap<FieldAccessor, TypeMirror> map = new LinkedHashMap<>();
 
-        for (Entry<VariableElement, TypeMirror> member : members.entrySet()) {
+        for (Entry<FieldAccessor, TypeMirror> member : members.entrySet()) {
 
             DebugLog.log(TAG, "\t\tEvaluating member - " + member.getValue().toString());
 
