diff --git a/old_product.java b/new_product.java
index 46a2d44..2cf0e9b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1,5 @@
-Job createJob(final Job job) throws GenieException;
\ No newline at end of file
+Job createJob(
+            @NotNull(message = "No job entered. Unable to create.")
+            @Valid
+            final Job job
+    ) throws GenieException;
\ No newline at end of file
