diff --git a/old_test.java b/new_test.java
index c35f73c..6109a24 100644
--- a/old_test.java
+++ b/new_test.java
@@ -26,10 +26,10 @@
                 new Job(
                         user,
                         name,
+                        version,
                         commandArgs,
                         commandCriteria,
-                        clusterCriterias,
-                        version
+                        clusterCriterias
                 )
         );
 
