diff --git a/old_test.java b/new_test.java
index 7a4dee4..6233873 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,22 +1,22 @@
-@Test(description = "PUT /vApp/{id}/networkConnectionSection", dependsOnMethods = { "testModifyGuestCustomizationSection" })
-   public void testModifyNetworkConnectionSection() {
+@Test(description = "PUT /vApp/{id}/networkConnectionSection", dependsOnMethods = { "testEditGuestCustomizationSection" })
+   public void testEditNetworkConnectionSection() {
       powerOffVm(vm.getHref());
       // Look up a network in the Vdc
       Set<Reference> networks = vdc.getAvailableNetworks();
       Reference network = Iterables.getLast(networks);
 
-      // Copy existing section and update fields
+      // Copy existing section and edit fields
       NetworkConnectionSection oldSection = vmApi.getNetworkConnectionSection(vm.getHref());
-      NetworkConnectionSection newSection = oldSection.toBuilder()
-            .networkConnection(NetworkConnection.builder()
-                  .ipAddressAllocationMode(IpAddressAllocationMode.DHCP.toString())
-                  .network(network.getName())
-                  .build())
-            .build();
+      NetworkConnectionSection newSection = oldSection
+               .toBuilder()
+               .networkConnection(
+                        NetworkConnection.builder().ipAddressAllocationMode(IpAddressAllocationMode.DHCP.toString())
+                                 .network(network.getName()).build()).build();
 
       // The method under test
-      Task modifyNetworkConnectionSection = vmApi.modifyNetworkConnectionSection(vm.getHref(), newSection);
-      assertTrue(retryTaskSuccess.apply(modifyNetworkConnectionSection), String.format(TASK_COMPLETE_TIMELY, "modifyNetworkConnectionSection"));
+      Task editNetworkConnectionSection = vmApi.editNetworkConnectionSection(vm.getHref(), newSection);
+      assertTrue(retryTaskSuccess.apply(editNetworkConnectionSection),
+               String.format(TASK_COMPLETE_TIMELY, "editNetworkConnectionSection"));
 
       // Retrieve the modified section
       NetworkConnectionSection modified = vmApi.getNetworkConnectionSection(vm.getHref());
