diff --git a/old_product.java b/new_product.java
index 0b99361..90d6702 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,7 +7,7 @@
       @Override
       public Void call() throws Exception {
         if (options == null) {
-          mFileSystem.setAttribute(new AlluxioURI(path), SetAttributePOptions.getDefaultInstance());
+          mFileSystem.setAttribute(new AlluxioURI(path));
         } else {
           mFileSystem.setAttribute(new AlluxioURI(path), options);
         }
