diff --git a/old_product.java b/new_product.java
index e9bb71a..cbaf0e3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public AccountInfo getCachedAccountInfo() {
+public Wallet getCachedAccountInfo() {
 
-    return exchangeEventListener.getAdapterInstance().getCachedAccountInfo();
+    return exchangeEventListener.getAdapterInstance().getCachedWallet();
   }
\ No newline at end of file
