diff --git a/old_test.java b/new_test.java
index 8fc1ebf..30a4f7a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test(expected = ExchangeException.class)
   public void getCachedAccountInfo() {
 
-    if (cachedAccountInfo == null) {
+    if (cachedWallet == null) {
       throw new ExchangeException("getBalances method has not been called yet, or balance data has not been recieved!");
     }
 
