diff --git a/old_test.java b/new_test.java
index d2b7514..0682def 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,5 @@
 @Test
-    public final void getIntervalHour(){
-        //        Date startDate = DateUtil.string2Date("2013-01-01 00:00:00", DatePattern.COMMON_DATE_AND_TIME);
-        //        log.debug(DateUtil.getIntervalHour(startDate, NOW) + "");
-        //        log.debug(DateUtil.getIntervalHour(startDate, DateUtil.string2Date("2113-01-01 00:00:00", DatePattern.COMMON_DATE_AND_TIME)) + "");
-        //        log.debug(DateUtil.getIntervalHour(startDate, DateUtil.string2Date("3113-01-01 00:00:00", DatePattern.COMMON_DATE_AND_TIME)) + "");
+    public void testGetIntervalHour1(){
         log.debug(StringUtil.format(
                         "%05d",
                         DateUtil.getIntervalHour(
