diff --git a/old_product.java b/new_product.java
index c935c74..13d36fc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,3 @@
 public static final int getIntervalHour(long spaceMillisecond){
-        // 相差小时数
-        return (int) (spaceMillisecond / (TimeInterval.SECONDS_PER_HOUR * 1000));
+        return (int) (spaceMillisecond / (TimeInterval.MILLISECOND_PER_HOUR));
     }
\ No newline at end of file
