diff --git a/old_test.java b/new_test.java
index 12acceb..9bfc4b2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,20 +1,20 @@
 @Test
 	public void getMemberships_shouldReturnUnvoidedMemberships() throws Exception {
 		executeDataSet(COHORT_XML);
-
+		
 		Cohort cohort = Context.getCohortService().getCohort(1);
-
+		
 		CohortMembership nonVoidedMembership = new CohortMembership(4);
 		CohortMembership voidedMembership = new CohortMembership(7);
 		voidedMembership.setVoided(true);
 		voidedMembership.setVoidedBy(Context.getAuthenticatedUser());
 		voidedMembership.setDateVoided(new Date());
 		voidedMembership.setVoidReason("Void reason");
-
+		
 		cohort.addMembership(nonVoidedMembership);
 		cohort.addMembership(voidedMembership);
-
+		
 		Context.getCohortService().saveCohort(cohort);
-		List<CohortMembership> unvoidedMemberships = cohort.getMemberships(false);
-		assertEquals(unvoidedMemberships.size(), 2);
+		Collection<CohortMembership> unvoidedMemberships = cohort.getMemberships(false);
+		assertEquals(2, unvoidedMemberships.size());
 	}
\ No newline at end of file
