diff --git a/old_product.java b/new_product.java
index 4ac8a94..68f46e3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
-public List<CohortMembership> getMemberships(boolean includeVoided) {
-		return getMemberships().stream()
-				.filter(m -> m.getVoided() == includeVoided)
-				.collect(Collectors.toList());
+public Collection<CohortMembership> getMemberships(boolean includeVoided) {
+		if (includeVoided) {
+			return getMemberships();
+		}
+		return getMemberships().stream().filter(m -> m.getVoided() == includeVoided).collect(Collectors.toList());
 	}
\ No newline at end of file
