diff --git a/old_test.java b/new_test.java
index e16479a..def973b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
     public void test_lastDayOfYear_serialization() throws IOException, ClassNotFoundException {
-        DateTimeAdjuster lastDayOfYear = DateTimeAdjusters.lastDayOfYear();
+        WithAdjuster lastDayOfYear = DateTimeAdjusters.lastDayOfYear();
         assertTrue(lastDayOfYear instanceof Serializable);
         
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
