diff --git a/old_product.java b/new_product.java
index 5af8c1b..ba49203 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public List<I_M_HU> cuToExistingTU(
 			@NonNull final I_M_HU sourceCuHU,
-			@NonNull final Quantity qtyCU,
+			@NonNull final BigDecimal qtyCU,
 			@NonNull final I_M_HU targetTuHU)
 	{
 		// NOTE: because this method does several non-atomic operations it is important to glue them together in a transaction
