diff --git a/old_test.java b/new_test.java
index fbd7a88..10f5b23 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,7 +13,7 @@
 		}
 
 		final HUProducerDestination producer = HUProducerDestination.ofVirtualPI();
-		data.helper.load(producer, data.helper.pSalad, new BigDecimal("3"), data.helper.uomKg);
+		data.helper.load(producer, data.helper.pSaladProductId, new BigDecimal("3"), data.helper.uomKg);
 		final I_M_HU cu2 = producer.getCreatedHUs().get(0);
 		final I_M_ReceiptSchedule rs2 = create_receiptSchedule_for_CU(cu2, "3");
 		final TableRecordReference rs2TableRef = TableRecordReference.of(rs2);
