diff --git a/old_product.java b/new_product.java
index 2e187b1..c5303bc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -20,6 +20,10 @@ public boolean rename(long fileId, TachyonURI dstPath)
           || (srcMount != null && dstMount != null && !srcMount.equals(dstMount))) {
         return false;
       }
+      // Renaming onto a mount point is not allowed.
+      if (mMountTable.isMountPoint(dstPath)) {
+        return false;
+      }
 
       // Rename a path to one of its subpaths is not allowed. Check for that, by making sure
       // srcComponents isn't a prefix of dstComponents.
