diff --git a/old_product.java b/new_product.java
index e8bc7f4..6e91618 100644
--- a/old_product.java
+++ b/new_product.java
@@ -44,7 +44,7 @@
 			if (i.hasNext())
 				hql += "and ";
 		}
-		hql += " order by username asc";
+		hql += " order by user.username asc";
 		Query query = sessionFactory.getCurrentSession().createQuery(hql);
 		for (Map.Entry<String, String> e : namesMap.entrySet())
 			query.setString(e.getKey(), e.getValue());
