diff --git a/old_test.java b/new_test.java
index 24cd84c..3b3fab2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,10 +4,10 @@
 		EquirectangularTools_F64 tools = new EquirectangularTools_F64();
 		tools.configure(width,height);
 
-		testCoordinate(tools, width/2, height/2);
-		testCoordinate(tools, 0, height/2);
-		testCoordinate(tools, width-1, height/2);
-		testCoordinate(tools, width/2, 0);
-		testCoordinate(tools, width/2, height-1);
+		equiToLonlat_reverse(tools, width/2, height/2);
+		equiToLonlat_reverse(tools, 0, height/2);
+		equiToLonlat_reverse(tools, width-1, height/2);
+		equiToLonlat_reverse(tools, width/2, 0);
+		equiToLonlat_reverse(tools, width/2, height-1);
 
 	}
\ No newline at end of file
