diff --git a/old_product.java b/new_product.java
index dbf1279..294d107 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,7 +11,7 @@
       String tableName = tableConfigEntry.getKey();
       InputTableConfig tableConfig = tableConfigEntry.getValue();
 
-      ClientContext context = new ClientContext(getClientProperties(job));
+      ClientContext context = new ClientContext(InputConfigurator.getClientInfo(CLASS, job));
       Table.ID tableId;
       // resolve table name to id once, and use id from this point forward
       try {
