diff --git a/old_product.java b/new_product.java
index 292654a..3bbbad2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@ public Classifiers getClassifiers(final String filterName) {
         if(filterName!=null && !filterName.isEmpty()) {
             Map<String, Object> queryParameters = new HashMap<String, Object>();
             queryParameters.put(FILTER_NAME,filterName);
-            request.withQuery(queryParameters);
+            request.withQueryMap(queryParameters);
         }
 
         return executeRequest(request, Classifiers.class);
