diff --git a/old_test.java b/new_test.java
index 223ada5..53a19dc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,23 +18,18 @@
         c1.setStatusDescription("The classifier instance is now available and is ready to take classifier requests.");
         classifiersResponse.add(c1);
 
-        Classifier c2 = new Classifier();
-        c2.setId("testId2");
-        c2.setStatus(Classifier.Status.AVAILABLE);
-        c2.setUrl("http://gateway.watson.net/");
-        c2.setStatusDescription("The classifier instance is now available and is ready to take classifier requests.");
-        classifiersResponse.add(c2);
+
 
         response.put("classifiers", classifiersResponse);
 
-        mockServer.when(request().withPath(LANGUAGE_CLASSIFIERS_PATH)).respond(
+        mockServer.when(request().withPath(CLASSIFIERS_PATH)).respond(
                 response().withHeaders(new Header(HttpHeaders.Names.CONTENT_TYPE, MediaType.APPLICATION_JSON))
                         .withBody(GsonSingleton.getGson().toJson(response)));
 
 
         Classifiers classifiers = service.getClassifiers();
-        Assert.assertNotNull(classifiers.getClassifiers());
-        Assert.assertFalse(classifiers.getClassifiers().isEmpty());
-        Assert.assertFalse(classifiers.getClassifiers().contains(classifiersResponse));
+        assertNotNull(classifiers.getClassifiers());
+        assertFalse(classifiers.getClassifiers().isEmpty());
+        assertFalse(classifiers.getClassifiers().contains(classifiersResponse));
 
     }
\ No newline at end of file
