diff --git a/old_product.java b/new_product.java
index 9fee446..7589285 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,12 +2,12 @@ public static String swapCase(final String str) {
         if (StringUtils.isEmpty(str)) {
             return str;
         }
-        char[] buffer = str.toCharArray();
+        final char[] buffer = str.toCharArray();
 
         boolean whitespace = true;
 
         for (int i = 0; i < buffer.length; i++) {
-            char ch = buffer[i];
+            final char ch = buffer[i];
             if (Character.isUpperCase(ch)) {
                 buffer[i] = Character.toLowerCase(ch);
                 whitespace = false;
