diff --git a/old_test.java b/new_test.java
index 9ccfcff..07cf27f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test(groups={"tck"})
     public void test_plusMinutes_one() {
         LocalDateTime t = TEST_2007_07_15_12_30_40_987654321.with(LocalTime.MIDNIGHT);
-        LocalDate d = t.toLocalDate();
+        LocalDate d = t.getDate();
 
         int hour = 0;
         int min = 0;
@@ -14,7 +14,7 @@
                 min = 0;
             }
 
-            assertEquals(t.toLocalDate(), d);
+            assertEquals(t.getDate(), d);
             assertEquals(t.getHour(), hour);
             assertEquals(t.getMinute(), min);
         }
