diff --git a/old_test.java b/new_test.java
index 1f20c10..ef5d251 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,14 @@
 @Test
-    public void testOnUpdateAuditable() throws InterruptedException {
+    public void testOnUpdateBaseEntity() throws InterruptedException {
         final BaseEntity a = new BaseEntity();
         Assert.assertNull(a.getId());
         Assert.assertNotNull(a.getCreated());
         Assert.assertNotNull(a.getUpdated());
-        a.onCreateAuditable();
+        a.onCreateBaseEntity();
         final Date originalCreate = a.getCreated();
         final Date originalUpdate = a.getUpdated();
         Thread.sleep(1);
-        a.onUpdateAuditable();
+        a.onUpdateBaseEntity();
         Assert.assertEquals(originalCreate, a.getCreated());
         Assert.assertNotEquals(originalUpdate, a.getUpdated());
     }
\ No newline at end of file
