diff --git a/old_product.java b/new_product.java
index 5412a70..c434db2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static CachedFileHandle createSourceFiles(String pkg, String yangName, GeneratedFileType types)
+public static CachedFileHandle createSourceFiles(String pkg, String yangName, int types)
             throws IOException {
         yangName = JavaIdentifierSyntax.getCamelCase(yangName);
         CachedFileHandle handler = new CachedJavaFileHandle(pkg, yangName, types);
