diff --git a/old_test.java b/new_test.java
index a00b378..e1765ab 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
     query(conn());
     query(EXISTS.args(' ' + conn()));
     // BXCL0001: connection errors
-    error(_CLIENT_CONNECT.args(Text.S_LOCALHOST, 9999, ADMIN, ""), BXCL_CONN_X);
-    error(_CLIENT_CONNECT.args("xxx", 9999, ADMIN, ADMIN), BXCL_CONN_X);
+    error(_CLIENT_CONNECT.args(Text.S_LOCALHOST, DB_PORT, ADMIN, ""), BXCL_CONN_X);
+    error(_CLIENT_CONNECT.args("xxx", DB_PORT, ADMIN, ADMIN), BXCL_CONN_X);
   }
\ No newline at end of file
