diff --git a/old_product.java b/new_product.java
index c7c0e6e..d9ff5d8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-public List<PointSegment> addCommand(String command, int line) throws GcodeParserException {
+@Override
+    public List<PointSegment> addCommand(String command, int line) throws GcodeParserException {
         //String stripped = GcodePreprocessorUtils.removeComment(command);
         List<String> commands = this.preprocessCommand(command);
         List<PointSegment> results = new ArrayList<>();
