diff --git a/old_test.java b/new_test.java
index dca8df1..ace7cbf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
     public void testWithMonthOfYear() {
         ThaiBuddhistDate date = testDate.withMonthOfYear(MonthOfYear.APRIL);
-        assertEquals(date, ThaiBuddhistDate.thaiBuddhistDate(testYear, MonthOfYear.APRIL, testDayOfMonth));
+        assertEquals(date, ThaiBuddhistDate.of(testYear, MonthOfYear.APRIL, testDayOfMonth));
     }
\ No newline at end of file
