diff --git a/old_test.java b/new_test.java
index 7f86ab9..631cbd3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
 		Zhang99ParamAll initial = GenericCalibrationGrid.createEasierParam(true, 2, false, 3, rand);
 		// tangential can't be linearly estimated
 
-		List<List<Point2D_F64>> observations = GenericCalibrationGrid.createObservations(initial,grid);
+		List<CalibrationObservation> observations = GenericCalibrationGrid.createObservations(initial,grid);
 
 		Helper alg = new Helper(grid,true,2,false);
 
