diff --git a/old_product.java b/new_product.java
index 301745f..a5cd120 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
-protected Zhang99ParamAll initialParam( List<List<Point2D_F64>> observations )
+protected Zhang99ParamAll initialParam( List<CalibrationObservation> observations )
 	{
 		status("Estimating Homographies");
 		List<DenseMatrix64F> homographies = new ArrayList<DenseMatrix64F>();
 		List<Se3_F64> motions = new ArrayList<Se3_F64>();
 
-		for( List<Point2D_F64> obs : observations ) {
+		for( CalibrationObservation obs : observations ) {
 			if( !computeHomography.computeHomography(obs) )
 				return null;
 
