diff --git a/old_product.java b/new_product.java
index c36c420..70c4c7a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,8 @@
 public void createCourse(String coordId, String courseId, String courseName)
 			throws EntityAlreadyExistsException, InvalidParametersException {
+		Assumption.assertNotNull(ERROR_NULL_PARAMETER, coordId);
+		Assumption.assertNotNull(ERROR_NULL_PARAMETER, courseId);
+		Assumption.assertNotNull(ERROR_NULL_PARAMETER, courseName);
 
 		verifyCoordUsingOwnIdOrAbove(coordId);
 
