diff --git a/old_test.java b/new_test.java
index 295db2f..0814343 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,33 +1,36 @@
 @Test
-	public void testCreateCourse() {
+	public void testCreateCourse() throws EntityAlreadyExistsException {
 		// SUCCESS
 		CourseData c = new CourseData();
-		c.id = "Winzor101";
-		c.name = "Basic Herping Derping";
-		c.coord = "herp.derp";
-		
-		try {
-			coursesDb.createCourse(c);
-		} catch (EntityAlreadyExistsException e) {
-			fail();
-		}
+		c.id = "Computing101";
+		c.name = "Basic Computing";
+		c.coord = "valid.id";
+		coursesDb.createCourse(c);
 		
 		// FAIL : duplicate
 		try {
 			coursesDb.createCourse(c);
 			fail();
 		} catch (EntityAlreadyExistsException e) {
-			
+			assertContains(CoursesDb.ERROR_CREATE_COURSE_ALREADY_EXISTS, e.getMessage());
 		}
 		
 		// FAIL : invalid params
-		c.id = "herp mc derp";
+		c.id = "invalid id spaces";
 		try {
 			coursesDb.createCourse(c);
 			fail();
 		} catch (AssertionError a) {
-			
+			assertEquals(CourseData.ERROR_ID_INVALIDCHARS, a.getMessage());
 		} catch (EntityAlreadyExistsException e) {
 			fail();
 		}
+		
+		// Null params check:
+		try {
+			coursesDb.createCourse(null);
+			fail();
+		} catch (AssertionError a) {
+			assertEquals(Common.ERROR_DBLEVEL_NULL_INPUT, a.getMessage());
+		}
 	}
\ No newline at end of file
