diff --git a/old_test.java b/new_test.java
index e4b8918..85db001 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,7 +17,8 @@
 							.clientId(CLIENT_ID)
 							.orgId(ORG_ID)
 							.materialDescriptor(earlierMaterialDescriptor)
-							.build());
+							.build())
+					.getCandidate();
 
 			final MaterialDescriptor laterMaterialDescriptor = materialDescriptor.withDate(t3);
 
@@ -66,7 +67,7 @@
 				.orgId(ORG_ID)
 				.materialDescriptor(materialDescriptor)
 				.groupId(earlierCandidate.getGroupId()).build();
-		stockCandidateService.applyDeltaToMatchingLaterStockCandidates(candidateWithDelta);
+		stockCandidateService.applyDeltaToMatchingLaterStockCandidates(SaveResult.builder().candidate(candidateWithDelta).build());
 
 		// assert that every stock record got some groupId
 		assertThat(DispoTestUtils.retrieveAllRecords()).allSatisfy(r -> assertThatModel(r).hasValueGreaterThanZero(I_MD_Candidate.COLUMN_MD_Candidate_GroupId));
