diff --git a/old_product.java b/new_product.java
index 4b08f2a..39ffbc7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,8 +3,6 @@ public ImmutableMap<String, String> getBasePathToAliasMap() {
     if (aliases == null) {
       return ImmutableMap.of();
     }
-    Preconditions.checkNotNull(buildTargetParser,
-        "buildTargetParser should be set for all instances of BuckConfig except EMPTY_INSTANCE");
 
     // Build up the Map with an ordinary HashMap because we need to be able to check whether the Map
     // already contains the key before inserting.
