diff --git a/old_test.java b/new_test.java
index 8be3595..b4561e5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,7 +10,7 @@
         "[alias]",
         "fb4a   =   //java/com/example:fbandroid",
         "katana =   //java/com/example:fbandroid"));
-    BuckConfig config1 = BuckConfig.createFromReader(reader1, parser);
+    BuckConfig config1 = createWithDefaultFilesystem(reader1, parser);
     assertEquals(ImmutableMap.of("java/com/example", "fb4a"), config1.getBasePathToAliasMap());
     assertEquals(
         ImmutableMap.of(
@@ -22,7 +22,7 @@
         "[alias]",
         "katana =   //java/com/example:fbandroid",
         "fb4a   =   //java/com/example:fbandroid"));
-    BuckConfig config2 = BuckConfig.createFromReader(reader2, parser);
+    BuckConfig config2 = createWithDefaultFilesystem(reader2, parser);
     assertEquals(ImmutableMap.of("java/com/example", "katana"), config2.getBasePathToAliasMap());
     assertEquals(
         ImmutableMap.of(
@@ -31,7 +31,7 @@
         config2.getEntriesForSection("alias"));
 
     Reader noAliasesReader = new StringReader("");
-    BuckConfig noAliasesConfig = BuckConfig.createFromReader(noAliasesReader, parser);
+    BuckConfig noAliasesConfig = createWithDefaultFilesystem(noAliasesReader, parser);
     assertEquals(ImmutableMap.of(), noAliasesConfig.getBasePathToAliasMap());
     assertEquals(ImmutableMap.of(), noAliasesConfig.getEntriesForSection("alias"));
 
