diff --git a/old_test.java b/new_test.java
index 138a5fe..2021d9a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
 	public void testGetSource() {
-		final Operator fixture = new OpImpl(0);
+		final Operator<?> fixture = new OpImpl(0);
 
-		final Operator.Output result = fixture.getSource();
+		final Operator<?>.Output result = fixture.getSource();
 
 		assertNotNull(result);
 		assertEquals(0, result.getIndex());
