diff --git a/old_test.java b/new_test.java
index 09cd4f6..de97a07 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,5 +6,5 @@
         String name = "NOTANONYMOUS";
         GroupService groupService = EPersonServiceFactory.getInstance().getGroupService();
         Group group = groupService.findByName(context, Group.ANONYMOUS);
-        groupService.setName(context, group, name);
+        groupService.setName(group, name);
     }
\ No newline at end of file
