diff --git a/old_product.java b/new_product.java
index ca9cec7..7e55541 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-void setName(Context context, String name) throws SQLException
+void setName(String name) throws SQLException
     {
-        if(!StringUtils.equals(this.name, name)) {
+        if(!StringUtils.equals(this.name, name) && !isPermanent()) {
             this.name = name;
             groupsChanged = true;
         }
