diff --git a/old_test.java b/new_test.java
index 5bc77bb..7c62cce 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void testGetWords() {
-    List<Word> result = service.getWords(customizationId, Type.ALL).execute();
+    List<WordData> result = service.getWords(customizationId, Type.ALL).execute();
     assertNotNull(result);
     assertTrue(!result.isEmpty());
   }
\ No newline at end of file
