diff --git a/old_product.java b/new_product.java
index a000e58..d989799 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
-public ServiceCall<List<Word>> getWords(String customizationId, Word.Type type) {
+public ServiceCall<List<WordData>> getWords(String customizationId, Word.Type type) {
     Validator.notNull(customizationId, "customizationId cannot be null");
     RequestBuilder requestBuilder = RequestBuilder.get(String.format(PATH_WORDS, customizationId));
     if (type != null) {
       requestBuilder.query(WORD_TYPE, type.toString().toLowerCase());
     }
 
-    ResponseConverter<List<Word>> converter = ResponseConverterUtils.getGenericObject(TYPE_WORDS, WORDS);
+    ResponseConverter<List<WordData>> converter = ResponseConverterUtils.getGenericObject(TYPE_WORDS, WORDS);
     return createServiceCall(requestBuilder.build(), converter);
   }
\ No newline at end of file
