diff --git a/old_test.java b/new_test.java
index bebbb71..1e8e557 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(expected = BrowserConnectionException.class)
-	public void testFireEvent() throws CrawljaxException {
+	public void testFireEvent() throws Exception {
 		browser.fireEventAndWait(new Eventable(new Identification(How.xpath, "/HTML"),
 		        EventType.click));
 	}
\ No newline at end of file
