diff --git a/old_product.java b/new_product.java
index 5bae047..a66e4dc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-CompletableFuture<Void> concat(String targetStreamSegmentName, long offset, String sourceStreamSegmentName, Duration timeout);
\ No newline at end of file
+CompletableFuture<Void> concat(SegmentHandle targetHandle, long offset, SegmentHandle sourceHandle, Duration timeout);
\ No newline at end of file
