diff --git a/old_product.java b/new_product.java
index 3860c10..57605f9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -25,9 +25,15 @@ public void parse(File file) {
                 //facts occur more often, do them first.
                 Matcher matcherFact = rxFact.matcher(str);
                 if (matcherFact.matches()) {
-                    String cpe = matcherFact.group(1);
+                    String cpe = matcherFact.group(1).trim();
                     if (cpe != null && cpe.startsWith("cpe:/a:")) {
                         skipEntry = false;
+
+                        //TODO deal with other possible :-: scenarios. do we need to be concerned about those?
+                        if (cpe.endsWith(":-")) {
+                            cpe = cpe.substring(0, cpe.length() - 2);
+                        }
+
                         addVulnerableCpe(cpe, doc);
                     }
                     continue;
