diff --git a/old_test.java b/new_test.java
index df175cb..2e4b646 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
     public void testSupports() {
         assertFalse(this.legacyTrustAdaptor.supports(null));
-        LegacyCasTrustedCredentials goodCred = new LegacyCasTrustedCredentials();
+        LegacyCasTrustedCredential goodCred = new LegacyCasTrustedCredential();
         assertTrue(this.legacyTrustAdaptor.supports(goodCred));
-        LegacyCasCredentials badCred = new LegacyCasCredentials();
+        LegacyCasCredential badCred = new LegacyCasCredential();
         assertFalse(this.legacyTrustAdaptor.supports(badCred));
     }
\ No newline at end of file
