diff --git a/old_product.java b/new_product.java
index 1a5517b..e2421f9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public final boolean supports(final Credentials credentials) {
-        return credentials != null
-            && (this.classToSupport.equals(credentials.getClass()) || (this.classToSupport
-                .isAssignableFrom(credentials.getClass()))
+public final boolean supports(final Credential credential) {
+        return credential != null
+            && (this.classToSupport.equals(credential.getClass()) || (this.classToSupport
+                .isAssignableFrom(credential.getClass()))
                 && this.supportSubClasses);
     }
\ No newline at end of file
