diff --git a/old_product.java b/new_product.java
index 00fa5b4..974c616 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,5 @@
 protected final void execute(final Command cmd, final boolean info) throws IOException {
     final Session ss = session();
     ss.execute(cmd);
-    if(newline && cmd instanceof XQuery) out.write(token(NL));
     if(info) Util.out(ss.info());
   }
\ No newline at end of file
