diff --git a/old_test.java b/new_test.java
index 7af9e74..068e5e4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,7 +20,7 @@
 		List<LocationTag> tagsNotRetiredAfter = ls.getAllLocationTags(false);
 		
 		// Make sure that all the values were filled in
-		assertTrue(retiredTag.isRetired());
+		assertTrue(retiredTag.getRetired());
 		assertNotNull(retiredTag.getDateRetired());
 		assertEquals(Context.getAuthenticatedUser(), retiredTag.getRetiredBy());
 		assertEquals("Just Testing", retiredTag.getRetireReason());
