diff --git a/old_product.java b/new_product.java
index 5f15343..2f0d0c6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,3 @@
-public static void throwIfIllegalArgument(boolean validCondition, String argName, String message, Object... args) throws IllegalArgumentException {
-        if (!validCondition) {
-            throw new IllegalArgumentException(argName + ": " + String.format(message, args));
-        }
+private static String badArgumentMessage(String argName, String message, Object... args) {
+        return argName + ": " + String.format(message, args);
     }
\ No newline at end of file
