diff --git a/old_test.java b/new_test.java
index ab28fcb..8801c3a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,16 +1,10 @@
 @Test
-    public void testThrowIfIllegalArgument() {
+    public void testCheckArgument() {
         AssertExtensions.assertThrows(
-                "Unexpected behavior for throwIfIllegalArgument(arg) with valid=false argument.",
-                () -> Exceptions.throwIfIllegalArgument(false, "invalid-arg"),
-                ex -> ex instanceof IllegalArgumentException);
-
-        AssertExtensions.assertThrows(
-                "Unexpected behavior for throwIfIllegalArgument(arg, msg) with valid=false argument.",
-                () -> Exceptions.throwIfIllegalArgument(false, "invalid-arg", "format msg %s", "foo"),
+                "Unexpected behavior for checkArgument(arg, msg) with valid=false argument.",
+                () -> Exceptions.checkArgument(false, "invalid-arg", "format msg %s", "foo"),
                 ex -> ex instanceof IllegalArgumentException);
 
         // These should not throw.
-        Exceptions.throwIfIllegalArgument(true, "valid-arg");
-        Exceptions.throwIfIllegalArgument(true, "valid-arg", "format msg %s", "foo");
+        Exceptions.checkArgument(true, "valid-arg", "format msg %s", "foo");
     }
\ No newline at end of file
