diff --git a/old_product.java b/new_product.java
index 832535d..ae29bfb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,5 +3,5 @@
    @Produces(RECOMPOSE_VAPP_PARAMS)
    @Consumes(TASK)
    @JAXBResponseParser
-   ListenableFuture<Task> recomposeVApp(@EndpointParam URI vAppURI,
-                                        @BinderParam(BindToXMLPayload.class) RecomposeVAppParams params);
\ No newline at end of file
+   ListenableFuture<Task> recompose(@EndpointParam URI vAppURI,
+                                    @BinderParam(BindToXMLPayload.class) RecomposeVAppParams params);
\ No newline at end of file
