diff --git a/old_test.java b/new_test.java
index 0e4a750..3021474 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
       RecomposeVAppParams params = RecomposeVAppParams.builder().build();
 
       // The method under test
-      Task recomposeVApp = vAppClient.recomposeVApp(vApp.getHref(), params);
+      Task recomposeVApp = vAppClient.recompose(vApp.getHref(), params);
       assertTrue(retryTaskSuccess.apply(recomposeVApp), String.format(TASK_COMPLETE_TIMELY, "recomposeVApp"));
    }
\ No newline at end of file
