diff --git a/old_test.java b/new_test.java
index 7a8bd64..bde89f1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	public void getAllUser() {
-		List<UserDTO> userList = client.getAllUser();
+		List<UserDTO> userList = client.getAllUser("admin", "admin");
 		assertTrue(userList.size() >= 6);
 		UserDTO admin = userList.iterator().next();
 		assertEquals("admin", admin.getLoginName());
