diff --git a/old_product.java b/new_product.java
index 4a54b9c..f7e9985 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-public List<UserDTO> getAllUser() {
-		String authentication = Servlets.encodeHttpBasic("admin", "admin");
+public List<UserDTO> getAllUser(String userName, String password) {
+		String authentication = Servlets.encodeHttpBasic(userName, password);
 		return client.path("/users").header(Servlets.AUTHENTICATION_HEADER, authentication)
 				.accept(MediaType.APPLICATION_JSON).get(new GenericType<List<UserDTO>>() {
 				});
