diff --git a/old_test.java b/new_test.java
index 3607ef1..503c314 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,7 +11,7 @@
         for (int i=0; i < names.size(); i++) {
             String filename = names.get(i);
             long t1 = System.currentTimeMillis();
-            userRoot.mkdir(filename, context.network, false, context.crypto.random, context.entrie).get();
+            userRoot.mkdir(filename, context.network, false, context.crypto.random).get();
             long duration = System.currentTimeMillis() - t1;
             worst = Math.max(worst, duration);
             best = Math.min(best, duration);
