diff --git a/old_product.java b/new_product.java
index e7ca573..af964ec 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,6 +11,16 @@
             LoggerHelpers.traceLeave(log, "createStream", traceId);
             return;
         }
+
+        try {
+            authenticate(scopeName + "/" + createStreamRequest.getStreamName(), READ_UPDATE);
+        } catch (AuthenticationException e) {
+            log.warn("Create stream for {} failed due to authentication failure.", createStreamRequest.getStreamName());
+            asyncResponse.resume(Response.status(Status.UNAUTHORIZED).build());
+            LoggerHelpers.traceLeave(log, "createStream", traceId);
+            return;
+        }
+
         StreamConfiguration streamConfiguration = ModelHelper.getCreateStreamConfig(createStreamRequest, scopeName);
         controllerService.createStream(streamConfiguration, System.currentTimeMillis())
                 .thenApply(streamStatus -> {
