diff --git a/old_product.java b/new_product.java
index ad5067e..eeb2d43 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public List<AssociationRule> learn(double confidence) {
-        List<AssociationRule> list = new ArrayList<AssociationRule>();
+        List<AssociationRule> list = new ArrayList<>();
         ttree = fim.buildTotalSupportTree();
         for (int i = 0; i < ttree.root.children.length; i++) {
             if (ttree.root.children[i] != null) {
