diff --git a/old_test.java b/new_test.java
index 3056b64..3becd94 100644
--- a/old_test.java
+++ b/new_test.java
@@ -35,7 +35,7 @@
         double[] pi2 = {0.55, 0.45};
         double[][] a2 = {{0.7, 0.3}, {0.15, 0.85}};
         double[][] b2 = {{0.45, 0.55}, {0.3, 0.7}};
-        HMM<String> init = new HMM<String>(pi2, a2, b2, symbols);
+        HMM<String> init = new HMM<>(pi2, a2, b2, symbols);
         HMM<String> result = init.learn(sequences, 100);
         System.out.println(result);
     }
\ No newline at end of file
