diff --git a/old_product.java b/new_product.java
index 2ef2640..38f4140 100644
--- a/old_product.java
+++ b/new_product.java
@@ -50,8 +50,8 @@ public static void optimize(final Data data, final boolean enforceText, final bo
     }
 
     // rebuild value indexes
-    optimize(IndexType.TEXT, data, md.createtext, md.textindex, enforceText, cmd);
-    optimize(IndexType.ATTRIBUTE, data, md.createattr, md.attrindex, enforceAttr, cmd);
-    optimize(IndexType.TOKEN, data, md.createtoken, md.tokenindex, enforceToken, cmd);
-    optimize(IndexType.FULLTEXT, data, md.createft, md.ftindex, enforceFt, cmd);
+    optimize(IndexType.TEXT, data, md.createtext, enforceText, cmd);
+    optimize(IndexType.ATTRIBUTE, data, md.createattr, enforceAttr, cmd);
+    optimize(IndexType.TOKEN, data, md.createtoken, enforceToken, cmd);
+    optimize(IndexType.FULLTEXT, data, md.createft, enforceFt, cmd);
   }
\ No newline at end of file
