diff --git a/old_product.java b/new_product.java
index 8d117a5..f879fa7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,6 +8,8 @@ public static PRFAlgorithm getPRFAlgorithm(ProtocolVersion protocolVersion, Ciph
             result = PRFAlgorithm.TLS_PRF_LEGACY;
         } else if (cipherSuite.usesSHA384()) {
             result = PRFAlgorithm.TLS_PRF_SHA384;
+        } else if (cipherSuite.usesGOSTR3411()) {
+            result = PRFAlgorithm.TLS_PRF_GOSTR3411;
         } else {
             result = PRFAlgorithm.TLS_PRF_SHA256;
         }
