diff --git a/old_test.java b/new_test.java
index 16cf029..113c064 100644
--- a/old_test.java
+++ b/new_test.java
@@ -19,5 +19,6 @@
         assertTrue(AlgorithmResolver.getPRFAlgorithm(ProtocolVersion.DTLS12, CipherSuite.TLS_DHE_PSK_WITH_AES_256_CCM) == PRFAlgorithm.TLS_PRF_SHA256);
         assertTrue(AlgorithmResolver.getPRFAlgorithm(ProtocolVersion.TLS12,
                 CipherSuite.TLS_DH_anon_EXPORT_WITH_RC4_40_MD5) == PRFAlgorithm.TLS_PRF_SHA256);
-
+        assertTrue(AlgorithmResolver.getPRFAlgorithm(ProtocolVersion.TLS12,
+                CipherSuite.TLS_GOSTR341001_WITH_28147_CNT_IMIT) == PRFAlgorithm.TLS_PRF_GOSTR3411);
     }
\ No newline at end of file
