diff --git a/old_product.java b/new_product.java
index 5c7ebcb..7a1a2c9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -34,8 +34,8 @@ protected void updateTrackLocation(T image) {
 		float deltaY = (peakY+offY) - templateNew.height/2;
 
 		// convert peak location into image coordinate system
-		regionTrack.tl_x = regionTrack.tl_x + deltaX*stepX;
-		regionTrack.tl_y = regionTrack.tl_y + deltaY*stepY;
+		regionTrack.x0 = regionTrack.x0 + deltaX*stepX;
+		regionTrack.y0 = regionTrack.y0 + deltaY*stepY;
 
 		updateRegionOut();
 	}
\ No newline at end of file
