diff --git a/old_test.java b/new_test.java
index 3e8e4f7..a3217f9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,21 +18,21 @@
 
 		// No motion motion
 		Rectangle2D_F32 r = alg.getTargetLocation();
-		assertEquals(5,r.tl_x,tolerance);
-		assertEquals(6,r.tl_y,tolerance);
+		assertEquals(5,r.x0,tolerance);
+		assertEquals(6,r.y0,tolerance);
 
 		// check estimated motion
 		GImageMiscOps.fillUniform(b,rand,0,200);
 		shiftCopy(-3,2,a,b);
 		alg.updateTrackLocation(b);
 		r = alg.getTargetLocation();
-		assertEquals(5-3,r.tl_x,tolerance);
-		assertEquals(6+2,r.tl_y,tolerance);
+		assertEquals(5-3,r.x0,tolerance);
+		assertEquals(6+2,r.y0,tolerance);
 
 		// try out of bounds case
 		GImageMiscOps.fillUniform(b,rand,0,200);
 		shiftCopy(-6,0,a,b);
 		alg.updateTrackLocation(b);
-		assertEquals(5-6,r.tl_x,tolerance);
-		assertEquals(6,r.tl_y,tolerance);
+		assertEquals(5-6,r.x0,tolerance);
+		assertEquals(6,r.y0,tolerance);
 	}
\ No newline at end of file
