diff --git a/old_test.java b/new_test.java
index 1fd0521..ac52071 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-	@Verifies(value = "should encodeStringsToFortyCharacters", method = "encodeString(String)")
-	public void encodeString_shouldEncodeStringsToFortyCharacters() throws Exception {
+	@Verifies(value = "should encodeStringsTo128Characters", method = "encodeString(String)")
+	public void encodeString_shouldEncodeStringsTo128Characters() throws Exception {
 		String hash = Security.encodeString("test" + "c788c6ad82a157b712392ca695dfcf2eed193d7f");
-		Assert.assertEquals(40, hash.length());
+		Assert.assertEquals(HASH_LENGTH, hash.length());
 	}
\ No newline at end of file
