diff --git a/old_test.java b/new_test.java
index b367a05..2625da6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,12 +5,12 @@
     resolver.context = new ContextImpl(null, resolver, null, 0, Collections.EMPTY_MAP, null, null);
 
     assertEquals(new SimpleDateFormat("yyyy-MM-dd").format(new Date()),
-            dateFormatEval.evaluate("'NOW',yyyy-MM-dd HH:mm", resolver.context));
+            dateFormatEval.evaluate("'NOW','yyyy-MM-dd HH:mm'", resolver.context));
 
     Map<String, Object> map = new HashMap<String, Object>();
     map.put("key", new Date());
     resolver.addNamespace("A", map);
 
     assertEquals(new SimpleDateFormat("yyyy-MM-dd HH:mm").format(new Date()),
-            dateFormatEval.evaluate("A.key, yyyy-MM-dd HH:mm", resolver.context));
+            dateFormatEval.evaluate("A.key, 'yyyy-MM-dd HH:mm'", resolver.context));
   }
\ No newline at end of file
