diff --git a/old_test.java b/new_test.java
index 32da25e..789c552 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,7 @@
 @Test
     public void testInstantiateMapCache() {
+        requestService.startRequest();
+
         MapCache cache = cachingService.instantiateMapCache("aaronz-map", null);
         assertNotNull(cache);
         assertEquals("aaronz-map", cache.getName());
@@ -11,4 +13,6 @@
         MapCache cache2 = cachingService.instantiateMapCache("aaronz-map", null);
         assertNotNull(cache2);
         assertEquals(cache2, cache);
+
+        requestService.endRequest(null);
     }
\ No newline at end of file
