diff --git a/old_product.java b/new_product.java
index d36f5a9..02bae67 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,16 +12,21 @@ protected MapCache instantiateMapCache(String cacheName, CacheConfig cacheConfig
         if (cacheConfig != null) {
             scope = cacheConfig.getCacheScope();
         }
-        if (CacheScope.REQUEST.equals(scope)) {
-            cache = getRequestMap().get(cacheName);
-        }
 
-        if (cache == null) {
-            cache = new MapCache(cacheName, cacheConfig);
-            // place cache into the right TL
+        Map<String, MapCache> caches = getRequestCaches();
+        if (caches != null) {
             if (CacheScope.REQUEST.equals(scope)) {
-                getRequestMap().put(cacheName, cache);
+                cache = caches.get(cacheName);
+            }
+
+            if (cache == null) {
+                cache = new MapCache(cacheName, cacheConfig);
+                // place cache into the right TL
+                if (CacheScope.REQUEST.equals(scope)) {
+                    caches.put(cacheName, cache);
+                }
             }
         }
+
         return cache;
     }
\ No newline at end of file
