diff --git a/old_test.java b/new_test.java
index 7856ce2..b596390 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,9 +15,9 @@
         Contract contract = new Contract();
         contract.setApiKey("12345");
         contract.setPlan("Silver");
-        contract.setServiceId("service-id");
-        contract.setServiceOrgId("service-test-org");
-        contract.setServiceVersion("1.7");
+        contract.setApiId("api-id");
+        contract.setApiOrgId("api-test-org");
+        contract.setApiVersion("1.7");
         app.getContracts().add(contract);
 
         Assert.assertEquals("{"
@@ -28,9 +28,9 @@
                     + "{"
                         + "\"apiKey\":\"12345\","
                         + "\"plan\":\"Silver\","
-                        + "\"serviceOrgId\":\"service-test-org\","
-                        + "\"serviceId\":\"service-id\","
-                        + "\"serviceVersion\":\"1.7\","
+                        + "\"apiOrgId\":\"api-test-org\","
+                        + "\"apiId\":\"api-id\","
+                        + "\"apiVersion\":\"1.7\","
                         + "\"policies\":[]"
                     + "}"
                 + "]"
@@ -54,9 +54,9 @@
                     + "{"
                         + "\"apiKey\":\"12345\","
                         + "\"plan\":\"Silver\","
-                        + "\"serviceOrgId\":\"service-test-org\","
-                        + "\"serviceId\":\"service-id\","
-                        + "\"serviceVersion\":\"1.7\","
+                        + "\"apiOrgId\":\"api-test-org\","
+                        + "\"apiId\":\"api-id\","
+                        + "\"apiVersion\":\"1.7\","
                         + "\"policies\":["
                             + "{"
                                 + "\"policyImpl\":\"policy-1-impl\","
