diff --git a/old_test.java b/new_test.java
index edcce1d..59ecf16 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test(expected = APIException.class)
 	public void saveOrder_shouldFailIfTheExistingDrugOrderMatchesTheConceptAndNotDrugOfTheDCOrder() throws Exception {
 		executeDataSet("org/openmrs/api/include/OrderServiceTest-globalProperties.xml");
-		OrderService orderService = Context.getOrderService();
 		final DrugOrder orderToDiscontinue = (DrugOrder) orderService.getOrder(5);
 		assertTrue(isOrderActive(orderToDiscontinue, null));
 		
