diff --git a/old_product.java b/new_product.java
index 186d8cc..905d97c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,8 @@
 public Order saveOrder(Order order) throws APIException {
-		//TODO reject existing orders
+		if (order.getOrderId() != null) {
+			throw new APIException("Cannot edit an existing order, see OrderService.reviseOrder");
+		}
+
 		discontinueExistingOrdersIfNecessary(order);
 		
 		return saveOrderInternal(order);
