diff --git a/old_product.java b/new_product.java
index dd17d62..09d680a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public DateTime parseBest(CharSequence text, Class<?>... types) {
+public DateTimeAccessor parseBest(CharSequence text, Class<?>... types) {
         DateTimes.checkNotNull(text, "Text must not be null");
         DateTimes.checkNotNull(types, "Class array must not be null");
         if (types.length < 2) {
@@ -9,7 +9,7 @@ public DateTime parseBest(CharSequence text, Class<?>... types) {
             DateTimeBuilder builder = parseToBuilder(str).resolve();
             for (Class<?> type : types) {
                 try {
-                    return (DateTime) builder.build(type);
+                    return (DateTimeAccessor) builder.build(type);
                 } catch (RuntimeException ex) {
                     // continue
                 }
