diff --git a/old_product.java b/new_product.java
index efb8373..e34fb42 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 @Override
-  public BlockoutTrigger getBlockout(String blockoutName) throws SchedulerException {
-    return (BlockoutTrigger) scheduler.getTrigger(blockoutName, BLOCK_GROUP);
+  public IBlockoutTrigger getBlockout(String blockoutName) throws SchedulerException {
+    return (IBlockoutTrigger) scheduler.getTrigger(blockoutName, BLOCK_GROUP);
   }
\ No newline at end of file
