diff --git a/old_product.java b/new_product.java
index cf79c83..64bb6c6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -126,8 +126,7 @@
 			
 			new UserProperties(user.getUserProperties()).setSupposedToChangePassword(forcePassword);
 			
-			UserValidator uv = new UserValidator();
-			uv.validate(user, errors);
+			userValidator.validate(user, errors);
 			
 			if (errors.hasErrors()) {
 				return showForm(user.getUserId(), createNewPerson, user, model);
