diff --git a/old_product.java b/new_product.java
index 5122eec..3b7fbc0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@ CompletableFuture<SegmentProperties> getStreamSegmentInfo(String streamSegmentNa
             // Looks like the Segment is active and we have it in our Metadata. Return the result from there.
             SegmentMetadata sm = this.containerMetadata.getStreamSegmentMetadata(streamSegmentId);
             if (sm.isDeleted()) {
-                result = FutureHelpers.failedFuture(new StreamSegmentNotExistsException(streamSegmentName));
+                result = Futures.failedFuture(new StreamSegmentNotExistsException(streamSegmentName));
             } else {
                 result = CompletableFuture.completedFuture(sm.getSnapshot());
             }
