diff --git a/old_test.java b/new_test.java
index 59a215d..837f5b1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,7 +15,7 @@
         val segmentState = context.stateStore.get(segmentName, TIMEOUT).join();
         Map<UUID, Long> expectedAttributes = segmentState == null ? null : segmentState.getAttributes();
 
-        AssertExtensions.assertThrows(
+        AssertExtensions.assertSuppliedFutureThrows(
                 "getStreamSegmentInfo did not throw correct exception when segment does not exist in Metadata or Storage.",
                 () -> context.mapper.getStreamSegmentInfo(segmentName, TIMEOUT),
                 ex -> ex instanceof StreamSegmentNotExistsException);
@@ -46,7 +46,7 @@
 
         // Segment exists in Metadata, but is marked as deleted.
         sm.markDeleted();
-        AssertExtensions.assertThrows(
+        AssertExtensions.assertSuppliedFutureThrows(
                 "getStreamSegmentInfo did not throw correct exception when segment is marked as Deleted in metadata.",
                 () -> context.mapper.getStreamSegmentInfo(segmentName, TIMEOUT),
                 ex -> ex instanceof StreamSegmentNotExistsException);
