diff --git a/old_test.java b/new_test.java
index ba6c0b1..1df3511 100644
--- a/old_test.java
+++ b/new_test.java
@@ -48,14 +48,20 @@
         instance.close();
 
         String expResult = "cpe:/a:apache:struts:2.1.2";
-        Identifier expIdentifier = new Identifier("cpe", expResult, expResult);
 
         for (Identifier i : commonValidator.getIdentifiers()) {
             assertFalse("Apache Common Validator - found a CPE identifier?", "cpe".equals(i.getType()));
         }
 
         assertTrue("Incorrect match size - struts", struts.getIdentifiers().size() >= 1);
-        assertTrue("Incorrect match - struts", struts.getIdentifiers().contains(expIdentifier));
+        boolean found = false;
+        for (Identifier i : struts.getIdentifiers()) {
+            if (expResult.equals(i.getValue())) {
+                found = true;
+                break;
+            }
+        }
+        assertTrue("Incorrect match - struts", found);
         assertTrue("Incorrect match size - spring3 - " + spring3.getIdentifiers().size(), spring3.getIdentifiers().size() >= 1);
 
         jarAnalyzer.close();
