diff --git a/old_product.java b/new_product.java
index e0902d6..2832cec 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,17 +2,17 @@ protected void determineCPE(Dependency dependency) throws CorruptIndexException,
         String vendors = "";
         String products = "";
         for (Confidence confidence : Confidence.values()) {
-            if (dependency.getVendorEvidence().contains(confidence)) {
-                vendors = addEvidenceWithoutDuplicateTerms(vendors, dependency.getVendorEvidence(), confidence);
+            if (dependency.contains(EvidenceType.VENDOR, confidence)) {
+                vendors = addEvidenceWithoutDuplicateTerms(vendors, dependency.getIterator(EvidenceType.VENDOR, confidence));
                 LOGGER.debug("vendor search: {}", vendors);
             }
-            if (dependency.getProductEvidence().contains(confidence)) {
-                products = addEvidenceWithoutDuplicateTerms(products, dependency.getProductEvidence(), confidence);
+            if (dependency.contains(EvidenceType.PRODUCT, confidence)) {
+                products = addEvidenceWithoutDuplicateTerms(products, dependency.getIterator(EvidenceType.PRODUCT, confidence));
                 LOGGER.debug("product search: {}", products);
             }
             if (!vendors.isEmpty() && !products.isEmpty()) {
-                final List<IndexEntry> entries = searchCPE(vendors, products, dependency.getVendorEvidence().getWeighting(),
-                        dependency.getProductEvidence().getWeighting());
+                final List<IndexEntry> entries = searchCPE(vendors, products, dependency.getVendorWeightings(),
+                        dependency.getProductWeightings());
                 if (entries == null) {
                     continue;
                 }
