diff --git a/old_product.java b/new_product.java
index 829a421..a8edd00 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-public void completeUfsFile(long tempUfsFileId, UFSCompleteFileTOptions options) throws alluxio.thrift.AlluxioTException, alluxio.thrift.ThriftIOException, org.apache.thrift.TException;
\ No newline at end of file
+public void completeUfsFile(long tempUfsFileId, CompleteUfsFileTOptions options) throws alluxio.thrift.AlluxioTException, alluxio.thrift.ThriftIOException, org.apache.thrift.TException;
\ No newline at end of file
