diff --git a/old_product.java b/new_product.java
index d567d17..182cba0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-public static void mergeHadoopConfiguration(org.apache.hadoop.conf.Configuration source) {
+public static void mergeHadoopConfiguration(org.apache.hadoop.conf.Configuration source,
+      AlluxioConfiguration alluxioConfiguration) {
     // Load Alluxio configuration if any and merge to the one in Alluxio file system
     // Push Alluxio configuration to the Job configuration
     Properties alluxioConfProperties = new Properties();
@@ -12,6 +13,6 @@ public static void mergeHadoopConfiguration(org.apache.hadoop.conf.Configuration
     LOG.info("Loading Alluxio properties from Hadoop configuration: {}", alluxioConfProperties);
     // Merge the relevant Hadoop configuration into Alluxio's configuration.
     // TODO(jiri): support multiple client configurations (ALLUXIO-2034)
-    Configuration.merge(alluxioConfProperties, Source.RUNTIME);
-    Configuration.validate();
+    alluxioConfiguration.merge(alluxioConfProperties, Source.RUNTIME);
+    alluxioConfiguration.validate();
   }
\ No newline at end of file
