diff --git a/old_product.java b/new_product.java
index 6cf0398..d73aed4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,7 @@ public void validate(Object obj, Errors errors) {
 		if (fieldType == null) {
 			errors.rejectValue("fieldType", "error.general");
 		} else {
-			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "name", "error.name");
+			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "localizedName.unlocalizedValue",
+			    "LocalizedName.unlocalizedName.empty");
 		}
 	}
\ No newline at end of file
