diff --git a/old_product.java b/new_product.java
index 4b7a0f7..a2eb4fe 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,22 +1,23 @@
-public void solve(double[][] B, double[][] X) {
+public void solve(DenseMatrix B) {
         if (!full) {
             throw new IllegalStateException("This is not a FULL singular value decomposition.");
         }
 
-        if (B.length != n || X.length != n || B[0].length != X[0].length) {
+        if (B.nrows() != m) {
             throw new IllegalArgumentException("Dimensions do not agree.");
         }
 
-        double[] xx = new double[n];
-        int p = B[0].length;
+        double[] b = new double[m];
+        double[] x = new double[n];
+        int p = B.ncols();
         for (int j = 0; j < p; j++) {
-            for (int i = 0; i < n; i++) {
-                xx[i] = B[i][j];
+            for (int i = 0; i < m; i++) {
+                b[i] = B.get(i, j);
             }
 
-            solve(xx, xx);
+            solve(b, x);
             for (int i = 0; i < n; i++) {
-                X[i][j] = xx[i];
+                B.set(i, j, x[i]);
             }
         }
     }
\ No newline at end of file
