diff --git a/old_product.java b/new_product.java
index 6024480..7e2eb2f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -75,7 +75,16 @@ public InitializeResult initialize() throws WIMApplicationException {
         if (iConnectTimeout != null) {
             iEnvironment.put(LDAP_ENV_PROP_CONNECT_TIMEOUT, iConnectTimeout.toString());
         } else {
-            iEnvironment.put(LDAP_ENV_PROP_CONNECT_TIMEOUT, String.valueOf(DEFAULT_TIMEOUT));
+            iEnvironment.put(LDAP_ENV_PROP_CONNECT_TIMEOUT, String.valueOf(DEFAULT_CONNECT_TIMEOUT));
+        }
+
+        /*
+         * Set the LDAP read time out.
+         */
+        if (iReadTimeout != null) {
+            iEnvironment.put(LDAP_ENV_PROP_READ_TIMEOUT, iReadTimeout.toString());
+        } else {
+            iEnvironment.put(LDAP_ENV_PROP_READ_TIMEOUT, String.valueOf(DEFAULT_READ_TIMEOUT));
         }
 
         /*
