diff --git a/old_product.java b/new_product.java
index 1d2a8d6..fd9cb5d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,12 @@
 public InputStream getInputStreamAtPosition(long tempUfsFileId, long position)
       throws FileDoesNotExistException, IOException {
-    InputStreamAgent stream = mInputAgents.get(tempUfsFileId);
-    if (stream == null) {
+    InputStreamAgent agent;
+    synchronized (mInputStreamAgents) {
+      agent = mInputStreamAgents.getFirstByField(mInputIdIndex, tempUfsFileId);
+    }
+    if (agent == null) {
       throw new FileDoesNotExistException(
           ExceptionMessage.BAD_WORKER_FILE_ID.getMessage(tempUfsFileId));
     }
-    return stream.openAtPosition(position);
+    return agent.openAtPosition(position);
   }
\ No newline at end of file
