diff --git a/old_test.java b/new_test.java
index 7a9765a..5563aef 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
     Mockito.when(mMockUfs.exists(uniqPath)).thenReturn(true);
     Mockito.when(mMockInputStream.skip(position)).thenReturn(position);
     UnderFileSystemManager manager = new UnderFileSystemManager();
-    long id = manager.openFile(new AlluxioURI(uniqPath));
+    long id = manager.openFile(mSessionId, new AlluxioURI(uniqPath));
     InputStream in = manager.getInputStreamAtPosition(id, position);
     Assert.assertEquals(mMockInputStream, in);
     Mockito.verify(mMockInputStream).skip(position);
