diff --git a/old_product.java b/new_product.java
index 144ab07..0addec8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -38,7 +38,7 @@
 
         currentPointer = nextPointer;
         int sessionIdLength = message[currentPointer] & 0xFF;
-        currentPointer = currentPointer + HandshakeByteLength.SESSION_ID_LENGTH;
+        currentPointer += HandshakeByteLength.SESSION_ID_LENGTH;
         nextPointer = currentPointer + sessionIdLength;
         byte[] sessionId = Arrays.copyOfRange(message, currentPointer, nextPointer);
         protocolMessage.setSessionId(sessionId);
