diff --git a/old_product.java b/new_product.java
index 6313756..d3c3f16 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery leProp(String propName, String otherProp) {
-        criterion = criterion.and(criterionBuilder.leProp(propName, otherProp));
+        criterion = criterion.and(Criteria.leProp(propName, otherProp));
         return this;
     }
\ No newline at end of file
