diff --git a/old_product.java b/new_product.java
index 14b31d9..aaf5962 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,11 @@
 private void init() {
-        Calendar definingCalendar = Calendar.getInstance(timeZone, locale);
+        final Calendar definingCalendar = Calendar.getInstance(timeZone, locale);
         thisYear= definingCalendar.get(Calendar.YEAR);
 
-        StringBuilder regex= new StringBuilder();
-        List<Strategy> collector = new ArrayList<Strategy>();
+        final StringBuilder regex= new StringBuilder();
+        final List<Strategy> collector = new ArrayList<Strategy>();
 
-        Matcher patternMatcher= formatPattern.matcher(pattern);
+        final Matcher patternMatcher= formatPattern.matcher(pattern);
         if(!patternMatcher.lookingAt()) {
             throw new IllegalArgumentException(
                     "Illegal pattern character '" + pattern.charAt(patternMatcher.regionStart()) + "'");
@@ -19,7 +19,7 @@ private void init() {
                 nextStrategy = null;
                 break;
             }
-            String nextFormatField= patternMatcher.group();
+            final String nextFormatField= patternMatcher.group();
             nextStrategy = getStrategy(nextFormatField, definingCalendar);
             if(currentStrategy.addRegex(this, regex)) {
                 collector.add(currentStrategy);
