diff --git a/old_test.java b/new_test.java
index 24bfa4c..badaf14 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
     public void testInit() {
-        ScheduledExecutorService service = EasyMock
+        final ScheduledExecutorService service = EasyMock
                 .createMock(ScheduledExecutorService.class);
         EasyMock.replay(service);
-        TimedSemaphore semaphore = new TimedSemaphore(service, PERIOD, UNIT,
+        final TimedSemaphore semaphore = new TimedSemaphore(service, PERIOD, UNIT,
                 LIMIT);
         EasyMock.verify(service);
         assertEquals("Wrong service", service, semaphore.getExecutorService());
