diff --git a/old_test.java b/new_test.java
index a22fffc..25a122c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
         logger.info("GaussCdf");
         double z = 3.0;
         double expResult = 0.9986501025724;
-        double result = ContinuousDistributions.GaussCdf(z);
+        double result = ContinuousDistributions.gaussCdf(z);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
