diff --git a/old_test.java b/new_test.java
index bc40011..4b8f6c7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,8 +10,8 @@
     query(_CONVERT_INTEGER_FROM_BASE.args("4", 10), 4);
     query(_CONVERT_INTEGER_FROM_BASE.args("65535", 10), 65535);
     query(_CONVERT_INTEGER_FROM_BASE.args("65536", 10), 65536);
-    error(_CONVERT_INTEGER_FROM_BASE.args("1", 1), Err.INVBASE);
-    error(_CONVERT_INTEGER_FROM_BASE.args("1", 100), Err.INVBASE);
-    error(_CONVERT_INTEGER_FROM_BASE.args("abc", 10), Err.INVDIG);
-    error(_CONVERT_INTEGER_FROM_BASE.args("012", 2), Err.INVDIG);
+    error(_CONVERT_INTEGER_FROM_BASE.args("1", 1), Err.INVBASE_X);
+    error(_CONVERT_INTEGER_FROM_BASE.args("1", 100), Err.INVBASE_X);
+    error(_CONVERT_INTEGER_FROM_BASE.args("abc", 10), Err.INVBASEDIG_X_X);
+    error(_CONVERT_INTEGER_FROM_BASE.args("012", 2), Err.INVBASEDIG_X_X);
   }
\ No newline at end of file
