diff --git a/old_product.java b/new_product.java
index cfb87f4..874bddf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @Override
-	public GetTeamDetailResponse getTeamDetail(Long id) {
-		GetTeamDetailResponse response = new GetTeamDetailResponse();
+	public GetTeamDetailResult getTeamDetail(Long id) {
+		GetTeamDetailResult result = new GetTeamDetailResult();
 		try {
 
 			Validate.notNull(id, "id参数为空");
@@ -10,12 +10,12 @@
 			Validate.notNull(team, "项目不存在(id:" + id + ")");
 
 			TeamDTO dto = BeanMapper.map(team, TeamDTO.class);
-			response.setTeam(dto);
+			result.setTeam(dto);
 
-			return response;
+			return result;
 		} catch (IllegalArgumentException e) {
-			return handleParameterError(response, e);
+			return handleParameterError(result, e);
 		} catch (RuntimeException e) {
-			return handleGeneralError(response, e);
+			return handleGeneralError(result, e);
 		}
 	}
\ No newline at end of file
