diff --git a/old_test.java b/new_test.java
index a992886..5e052d5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
 	public void getTeamDetail() {
 		AccountWebService accountWebService = creatClient();
 
-		GetTeamDetailResponse response = accountWebService.getTeamDetail(1L);
-		assertEquals("Dolphin", response.getTeam().getName());
-		assertEquals("Admin", response.getTeam().getMaster().getName());
+		GetTeamDetailResult result = accountWebService.getTeamDetail(1L);
+		assertEquals("Dolphin", result.getTeam().getName());
+		assertEquals("Admin", result.getTeam().getMaster().getName());
 	}
\ No newline at end of file
