diff --git a/old_product.java b/new_product.java
index 9a9a3a9..2f38562 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ private static void error(final String query, final QNm error, final QueryError
     final String q = "import module namespace geo='http://expath.org/ns/geo'; " +
         "declare namespace gml='http://www.opengis.net/gml';" + query;
 
-    try(final QueryProcessor qp = new QueryProcessor(q, context)) {
+    try(QueryProcessor qp = new QueryProcessor(q, context)) {
       final String res = qp.value().serialize().toString().replaceAll("(\\r|\\n) *", "");
       fail("Query did not fail:\n" + query + "\n[E] " + error + "...\n[F] " + res);
     } catch(final QueryException ex) {
