diff --git a/old_test.java b/new_test.java
index ca3b7f9..f40f4d9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,9 @@
 @Test
-	public void testAnd() {
-		Criteria criteria = new Criteria("field_1").startsWith("start").endsWith("end").and("field_2").startsWith("2start")
-				.endsWith("2end");
-		Assert.assertEquals("field_2", criteria.getField().getName());
-		Assert.assertEquals(" AND ", criteria.getConjunctionOperator());
-		Assert.assertEquals(2, criteria.getCriteriaChain().size());
+	public void testCirteriasJoindWihtAndShouldBeSiblingsOfCreatedCrotch() {
+
+		Criteria c1 = new Criteria("field_1").startsWith("start").endsWith("end");
+		Criteria c2 = new Criteria("field_2").startsWith("2start");
+		Crotch crotch = c1.and(c2);
+
+		Assert.assertThat(crotch.getSiblings(), IsIterableContainingInOrder.<Node> contains(c1, c2));
 	}
\ No newline at end of file
