diff --git a/old_product.java b/new_product.java
index 96165dd..1be0659 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,6 @@
-public Criteria and(String fieldname) {
-		return new Criteria(this.criteriaChain, fieldname);
+@SuppressWarnings("unchecked")
+	public Crotch and(String fieldname) {
+
+		Criteria node = new Criteria(fieldname);
+		return and(node);
 	}
\ No newline at end of file
