diff --git a/old_test.java b/new_test.java
index 5a1c083..302b4ce 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,8 @@
     public void testEquals() {
         final String text = "equals";
         MatcherAssert.assertThat(
-            "Envelope value does not match its represented String value",
+            "Envelope does not match text representing the same value",
             new TextEnvelopeDummy(text),
-            new IsEqual<>(text)
+            new IsEqual<>(new TextOf(text))
         );
     }
\ No newline at end of file
