diff --git a/old_product.java b/new_product.java
index 16e4fc9..139b003 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,14 @@
 @Override
+    @SuppressWarnings("PMD.OnlyOneReturn")
     public final boolean equals(final Object obj) {
-        return new UncheckedScalar<>(this.origin).value().equals(obj);
+        if (this == obj) {
+            return true;
+        }
+        if (!(obj instanceof TextEnvelope)) {
+            return false;
+        }
+        final TextEnvelope that = (TextEnvelope) obj;
+        return new UncheckedText(this).asString().equals(
+            new UncheckedText(that).asString()
+        );
     }
\ No newline at end of file
