diff --git a/old_product.java b/new_product.java
index 2708e9d..9f6e7cb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,9 @@
 public static RectangleLength2D_I32 boundBox( int srcWidth , int srcHeight ,
 												  int dstWidth , int dstHeight ,
-												  PixelTransform2_F32 transform )
+												  Point2D_F32 work,
+												  PixelTransform<Point2D_F32> transform )
 	{
-		RectangleLength2D_I32 ret = boundBox(srcWidth,srcHeight,transform);
+		RectangleLength2D_I32 ret = boundBox(srcWidth,srcHeight,work,transform);
 
 		int x0 = ret.x0;
 		int y0 = ret.y0;
