diff --git a/old_test.java b/new_test.java
index 89d89c3..271c04e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,7 @@
 @Test
 	public void testFuzzy() {
 		Criteria criteria = new Criteria("field_1").fuzzy("value_1");
-		assertCriteriaEntry(criteria.getCriteriaEntries(), 0, "$fuzzy#NaN", "value_1");
+		CriteriaEntry entry = getCriteriaEntryByPosition(criteria.getCriteriaEntries(), 0);
+		Assert.assertEquals("value_1", ((Object[]) entry.getValue())[0]);
+		Assert.assertEquals(Float.NaN, ((Object[]) entry.getValue())[1]);
 	}
\ No newline at end of file
