diff --git a/old_product.java b/new_product.java
index acc3809..34fcc40 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,6 +2,6 @@ public Criteria fuzzy(String s, float levenshteinDistance) {
 		if (!Float.isNaN(levenshteinDistance) && (levenshteinDistance < 0 || levenshteinDistance > 1)) {
 			throw new InvalidDataAccessApiUsageException("Levenshtein Distance has to be within its bounds (0.0 - 1.0).");
 		}
-		criteria.add(new CriteriaEntry("$fuzzy#" + levenshteinDistance, s));
+		criteria.add(new CriteriaEntry(OperationKey.FUZZY, new Object[] { s, levenshteinDistance }));
 		return this;
 	}
\ No newline at end of file
