diff --git a/old_test.java b/new_test.java
index a17b108..27dbe9c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,9 +15,9 @@
     assertEntry("gateway-conflict-resolver", order++, entries.next());
     assertEntry("async-event-queue", order++, entries.next());
     assertEntry("cache-server", order++, entries.next());
-    assertEntry("bridge-server", order++, entries.next());
     assertEntry("pool", order++, entries.next());
     assertEntry("disk-store", order++, entries.next());
+    assertEntry("hdfs-store", order++, entries.next());
     assertEntry("pdx", order++, entries.next());
     assertEntry("region-attributes", order++, entries.next());
     assertEntry("jndi-bindings", order++, entries.next());
