diff --git a/old_product.java b/new_product.java
index 0c5d5f0..14d7c11 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@ public static LinkedHashMap<String, CacheElement> buildElementMap(final Document
 
     final LinkedHashMap<String, CacheElement> elementMap = new LinkedHashMap<String, CacheElement>();
 
-    buildElementMapCacheType(elementMap, resolveSchema(schemaLocationMap, CacheXml.NAMESPACE));
+    buildElementMapCacheType(elementMap, resolveSchema(schemaLocationMap, CacheXml.GEODE_NAMESPACE));
 
     // if we are ever concerned with the order of extensions or children process them here.
 
