diff --git a/old_product.java b/new_product.java
index 7ebacdb..e9ef347 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,8 @@
 @Override
-    public int removeLastOccurrence(KType e1)
-    {
-        final int index = lastBufferIndexOf(e1);
-        if (index >= 0) removeAtBufferIndex(index);
-        return index;
-    }
\ No newline at end of file
+  public int removeLast(KType e1) {
+    final int index = lastBufferIndexOf(e1);
+    if (index >= 0) {
+      removeAtBufferIndex(index);
+    }
+    return index;
+  }
\ No newline at end of file
