diff --git a/old_test.java b/new_test.java
index 4754280..e50d886 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,13 +15,13 @@
         {
             KType k = cast(rnd.nextInt(modulo));
             assertEquals(
-                deque.removeLastOccurrence(k) >= 0, 
+                deque.removeLast(k) >= 0, 
                 sequence.removeLast(k) >= 0);
         }
 
         assertListEquals(deque.toArray(), sequence.toArray());
 
-        assertTrue(0 > deque.removeLastOccurrence(cast(modulo + 1)));
+        assertTrue(0 > deque.removeLast(cast(modulo + 1)));
         deque.addFirst(cast(modulo + 1));
-        assertTrue(0 <= deque.removeLastOccurrence(cast(modulo + 1)));
+        assertTrue(0 <= deque.removeLast(cast(modulo + 1)));
     }
\ No newline at end of file
