diff --git a/old_product.java b/new_product.java
index 4377f44..6579ea1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-String execute(Collector collector, String counterName, String sessionId, String threadId, // NOPMD
-			String jobId) throws IOException {
+String execute(Collector collector, CollectorServer collectorServer, String counterName, // NOPMD
+			String sessionId, String threadId, String jobId) throws IOException {
 		// CHECKSTYLE:ON
 		String messageForReport;
 		switch (this) {
@@ -8,6 +8,10 @@ String execute(Collector collector, String counterName, String sessionId, String
 			assert counterName != null;
 			messageForReport = clearCounter(collector, counterName);
 			break;
+		case MAIL_TEST:
+			assert collector != null;
+			messageForReport = mailTest(collector, collectorServer);
+			break;
 		case GC:
 			if (GC_ENABLED) {
 				// garbage collector
