diff --git a/old_test.java b/new_test.java
index e1b8d32..1536aea 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,24 +11,24 @@
 		final String jobId = "jobId";
 
 		assertNotNull("message GC",
-				Action.GC.execute(collector, counterName, sessionId, threadId, jobId));
-		assertNotNull("message CLEAR_COUNTER",
-				Action.CLEAR_COUNTER.execute(collector, counterName, sessionId, threadId, jobId));
+				Action.GC.execute(collector, null, counterName, sessionId, threadId, jobId));
+		assertNotNull("message CLEAR_COUNTER", Action.CLEAR_COUNTER.execute(collector, null,
+				counterName, sessionId, threadId, jobId));
 		assertNotNull("message CLEAR_COUNTER",
-				Action.CLEAR_COUNTER.execute(collector, ALL, sessionId, threadId, jobId));
+				Action.CLEAR_COUNTER.execute(collector, null, ALL, sessionId, threadId, jobId));
 		if (CacheManager.getInstance().getCache("test clear") == null) {
 			CacheManager.getInstance().addCache("test clear");
 		}
-		assertNotNull("message CLEAR_CACHES",
-				Action.CLEAR_CACHES.execute(collector, counterName, sessionId, threadId, jobId));
-		final String heapDump1 = Action.HEAP_DUMP.execute(collector, counterName, sessionId,
+		assertNotNull("message CLEAR_CACHES", Action.CLEAR_CACHES.execute(collector, null,
+				counterName, sessionId, threadId, jobId));
+		final String heapDump1 = Action.HEAP_DUMP.execute(collector, null, counterName, sessionId,
 				threadId, jobId);
 		assertNotNull("message HEAP_DUMP", heapDump1);
 		String heapDump2;
 		// on le refait une deuxième fois dans la même seconde pour tester le nom du fichier
 		do {
-			heapDump2 = Action.HEAP_DUMP
-					.execute(collector, counterName, sessionId, threadId, jobId);
+			heapDump2 = Action.HEAP_DUMP.execute(collector, null, counterName, sessionId, threadId,
+					jobId);
 			assertNotNull("message HEAP_DUMP", heapDump2);
 		} while (heapDump1.equals(heapDump2));
 		for (final File file : Parameters.TEMPORARY_DIRECTORY.listFiles()) {
@@ -37,9 +37,9 @@
 			}
 		}
 		assertNotNull("message INVALIDATE_SESSIONS", Action.INVALIDATE_SESSIONS.execute(collector,
-				counterName, sessionId, threadId, jobId));
+				null, counterName, sessionId, threadId, jobId));
 		assertNotNull("message INVALIDATE_SESSION", Action.INVALIDATE_SESSION.execute(collector,
-				counterName, sessionId, threadId, jobId));
+				null, counterName, sessionId, threadId, jobId));
 
 		killThread(collector, counterName, sessionId, threadId, jobId);
 
