diff --git a/old_test.java b/new_test.java
index 5b1a713..54c7e39 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test(dataProvider="sampleTimes", groups={"tck"})
     public void test_get(int h, int m, int s, int ns) {
         LocalTime a = LocalTime.of(h, m, s, ns);
-        assertEquals(a.getHourOfDay(), h);
-        assertEquals(a.getMinuteOfHour(), m);
-        assertEquals(a.getSecondOfMinute(), s);
-        assertEquals(a.getNanoOfSecond(), ns);
+        assertEquals(a.getHour(), h);
+        assertEquals(a.getMinute(), m);
+        assertEquals(a.getSecond(), s);
+        assertEquals(a.getNano(), ns);
     }
\ No newline at end of file
