diff --git a/old_product.java b/new_product.java
index 78a3491..bc2f931 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
 public Publisher<Payload> requestChannel(final Publisher<Payload> payloadStream) {
-		return startStream(nextStreamId(), FrameType.REQUEST_CHANNEL, payloadStream);
+		return startChannel(nextStreamId(), FrameType.REQUEST_CHANNEL, payloadStream);
 	}
\ No newline at end of file
