diff --git a/old_test.java b/new_test.java
index 98555c6..9badefd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,7 +11,7 @@
             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
         };
-        MCCMeasure instance = new MCCMeasure();
+        MCC instance = new MCC();
         double expResult = 0.83068;
         double result = instance.measure(truth, prediction);
         assertEquals(expResult, result, 1E-5);
