diff --git a/old_product.java b/new_product.java
index cbbe044..5f9cc32 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static double measure(int[] truth, double[] probability) {
+public static double apply(int[] truth, double[] probability) {
         if (truth.length != probability.length) {
             throw new IllegalArgumentException(String.format("The vector sizes don't match: %d != %d.", truth.length, probability.length));
         }
