diff --git a/old_product.java b/new_product.java
index b860006..b102e5d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@
 	public List<EncounterType> findEncounterTypes(String name) throws DAOException {
 		List<EncounterType> results = null;
 		Criteria crit = sessionFactory.getCurrentSession().createCriteria(EncounterType.class);
-		HibernateUtil.addLikeCriterionForLocalizedColumn(name, "name", crit, false, MatchMode.START);
+		HibernateUtil.addLikeCriterionForLocalizedColumn(name, "localizedName", crit, false, MatchMode.START);
 		results = crit.list();
 		Collections.sort(results, new MetadataComparator(Context.getLocale()));
 		return results;
