diff --git a/old_test.java b/new_test.java
index b7e9738..03808fe 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,5 +5,5 @@
 
         assertTrue("should have two mappings",
                 Lists.newLinkedList(
-                        service.getMappingEntriesByAddId(MAP_DATABASE, appId)).size() == 2);
+                        service.getMappingEntriesByAppId(MAP_DATABASE, appId)).size() == 2);
     }
\ No newline at end of file
