diff --git a/old_test.java b/new_test.java
index b2de604..bd86ca5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,8 @@
         File file = File.createTempFile("DiskCheck", "test");
         long usableSpace = file.getUsableSpace();
         long totalSpace = file.getTotalSpace();
-        diskChecker
-                .setDiskSpaceThreshold((1f - ((float) usableSpace / (float) totalSpace)) - 0.05f);
+        float threshold =
+                (1f - ((float) usableSpace / (float) totalSpace)) - 0.05f;
+        diskChecker.setDiskSpaceThreshold(threshold, threshold);
         diskChecker.checkDiskFull(file);
     }
\ No newline at end of file
