diff --git a/old_product.java b/new_product.java
index 4397c8c..8369aa8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @VisibleForTesting
-    void checkDiskFull(File dir) throws DiskOutOfSpaceException {
+    void checkDiskFull(File dir) throws DiskOutOfSpaceException, DiskWarnThresholdException {
         if (null == dir) {
             return;
         }
@@ -10,7 +10,12 @@
             float used = 1f - free;
             if (used > diskUsageThreshold) {
                 throw new DiskOutOfSpaceException("Space left on device "
-                        + usableSpace + " < threshhold " + diskUsageThreshold);
+                        + usableSpace + " Used space fraction:" + used + " < threshhold " + diskUsageThreshold);
+            }
+            // Warn should be triggered only if disk usage threshold doesn't trigger first.
+            if (used > diskUsageWarnThreshold) {
+                throw new DiskWarnThresholdException("Space left on device:"
+                        + usableSpace + " Used space fraction:" + used +" < WarnThreshold:" + diskUsageWarnThreshold);
             }
         } else {
             checkDiskFull(dir.getParentFile());
