diff --git a/old_test.java b/new_test.java
index b9bac5f..64efe34 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,10 @@
 @Test
-  public void restore() {
+  public void restore() throws BaseXException {
+    new Close().execute(context);
+
+    // backup and restore file
+    query(_DB_BACKUP.args(NAME));
+    query(_DB_RESTORE.args(NAME));
+
     // [LK] more tests! E.g. combination of restore, create, ...
-//    query(_DB_RESTORE.args(NAME));
   }
\ No newline at end of file
