diff --git a/old_test.java b/new_test.java
index d024d1e..dd70607 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test(dataProvider = "calendarsystemtype")
-    public <C extends Chronology<C>> void test_chronoSerializationSingleton(C chrono, String calendarType) throws Exception {
-        C orginal = chrono;
+    public void test_chronoSerializationSingleton(Chronology chrono, String calendarType) throws Exception {
+        Chronology orginal = chrono;
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
         ObjectOutputStream out = new ObjectOutputStream(baos);
         out.writeObject(orginal);
         out.close();
         ByteArrayInputStream bais = new ByteArrayInputStream(baos.toByteArray());
         ObjectInputStream in = new ObjectInputStream(bais);
-        C ser = (C)in.readObject();
+        Chronology ser = (Chronology) in.readObject();
         assertSame(ser, chrono, "Deserialized Chrono is not the singleton serialized");
     }
\ No newline at end of file
