diff --git a/old_test.java b/new_test.java
index 2024802..a7ec9c2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,8 @@
     public void testSetFormat() throws SQLException
     {
         int id = 3;
-        BitstreamFormat format = BitstreamFormat.find(context, id);
+        BitstreamFormat format = bitstreamFormatService.find(context, id);
         bs.setFormat(format);
-        assertThat("testSetFormat 0", bs.getFormat(), notNullValue());
-        assertThat("testSetFormat 1", bs.getFormat(), equalTo(BitstreamFormat.find(context, id)));
+        assertThat("testSetFormat 0", bs.getFormat(context), notNullValue());
+        assertThat("testSetFormat 1", bs.getFormat(context), equalTo(bitstreamFormatService.find(context, id)));
     }
\ No newline at end of file
