diff --git a/old_product.java b/new_product.java
index 0860c7f..f06ec2b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,9 @@
-protected long getNextTimeoutIntervalMs() {
-    long nextTimeoutIntervalMs = INFINITE_FUTURE;
+protected Duration getNextTimeoutInterval() {
+    Duration nextTimeoutInterval = INFINITE_FUTURE;
     if (!timers.isEmpty()) {
       // The time recorded in timer is in nano-seconds. We have to convert it to milli-seconds
       // We need to ceil the result to avoid early wake up
-      nextTimeoutIntervalMs =
-          (timers.peek().getExpirationNs() - System.nanoTime()
-          + Constants.MILLISECONDS_TO_NANOSECONDS) / Constants.MILLISECONDS_TO_NANOSECONDS;
+      nextTimeoutInterval = timers.peek().expirationTime.minusNanos(System.nanoTime());
     }
-    return nextTimeoutIntervalMs;
+    return nextTimeoutInterval;
   }
\ No newline at end of file
