diff --git a/old_test.java b/new_test.java
index c9b135b..a676cc6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,16 +8,16 @@
       }
     };
 
-    long intervalNs = 6L * 1000 * 1000 * 1000;
-    slaveLooper.registerTimerEventInNanoSeconds(intervalNs, r);
+    Duration interval = Duration.ofSeconds(6);
+    slaveLooper.registerTimerEvent(interval, r);
 
-    long res = 1000;
+    Duration res = Duration.ofNanos(1000);
 
     try {
       Method method =
-          slaveLooper.getClass().getSuperclass().getDeclaredMethod("getNextTimeoutIntervalMs");
+          slaveLooper.getClass().getSuperclass().getDeclaredMethod("getNextTimeoutInterval");
       method.setAccessible(true);
-      res = (Long) method.invoke(slaveLooper) * 1000 * 1000;
+      res = (Duration) method.invoke(slaveLooper);
     } catch (NoSuchMethodException e) {
       e.printStackTrace();
     } catch (IllegalAccessException e) {
@@ -28,5 +28,5 @@
 
     Assert.assertNotNull(res);
 
-    Assert.assertTrue(res <= intervalNs && res > intervalNs / 2);
+    Assert.assertTrue(res.compareTo(interval) <= 0 && res.compareTo(interval.dividedBy(2)) > 0);
   }
\ No newline at end of file
