diff --git a/old_product.java b/new_product.java
index b6ecfe3..4cec8e1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @Override
 	public void purgeOrderType(OrderType orderType) {
 		if (dao.isOrderTypeInUse(orderType)) {
-			throw CannotDeleteOrderPropertyInUseException.withProperty("type");
+			throw new CannotDeleteObjectInUseException("Order.type.cannot.delete", (Object[]) null);
 		}
 		dao.purgeOrderType(orderType);
 	}
\ No newline at end of file
