diff --git a/old_test.java b/new_test.java
index 828353e..921d2dd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -29,7 +29,7 @@
             Assert.assertEquals("Exactly one Data Frame was expected so far.", 1, commitFrames.size());
 
             //Read all entries in the Log and interpret them as DataFrames, then verify the records can be reconstructed.
-            List<DataFrame> frames = dataLog.getAllEntries(readItem -> new DataFrame(readItem.getPayload(), readItem.getLength()));
+            List<DataFrame> frames = dataLog.getAllEntries(readItem -> DataFrame.from(readItem.getPayload(), readItem.getLength()));
             Assert.assertEquals("Unexpected number of frames generated.", commitFrames.size(), frames.size());
             DataFrameTestHelpers.checkReadRecords(frames, records, r -> new ByteArraySegment(r.getFullSerialization()));
         }
