diff --git a/old_test.java b/new_test.java
index a671f58..9e6b365 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,8 @@
 @Test
 	@Verifies(value = "should save properties with case difference only", method = "saveGlobalProperties(List<QGlobalProperty;>)")
 	public void saveGlobalProperties_shouldSavePropertiesWithCaseDifferenceOnly() throws Exception {
+		int originalSize = adminService.getAllGlobalProperties().size();
+		
 		List<GlobalProperty> props = new ArrayList<GlobalProperty>();
 		props.add(new GlobalProperty("a.property.key", "something"));
 		props.add(new GlobalProperty("a.property.KEY", "somethingelse"));
@@ -8,5 +10,5 @@
 		
 		// make sure that we now have two properties
 		props = adminService.getAllGlobalProperties();
-		Assert.assertEquals(2, props.size());
+		Assert.assertEquals(originalSize + 2, props.size());
 	}
\ No newline at end of file
