diff --git a/old_test.java b/new_test.java
index 929278a..93c2a39 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
-@Test
-    public void testCopy() {
+public void testCopy() {
         Map<String, String> bucketKeyMap = new HashMap<String, String>();
         bucketKeyMap.put(TestConfig.testBucket_z0, TestConfig.testKey_z0);
         bucketKeyMap.put(TestConfig.testBucket_na0, TestConfig.testKey_na0);
