diff --git a/old_test.java b/new_test.java
index 3aee424..943e3eb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,7 +18,7 @@
             .withAmountOfTrees(5)
             .withFeaturesCountSelectionStrgy(x -> 2);
 
-        ModelsComposition mdl = trainer.fit(sample, parts, (k, v) -> VectorUtils.of(v), (k, v) -> k);
+        ModelsComposition mdl = trainer.fit(sample, parts, FeatureLabelExtractorWrapper.wrap((k, v) -> VectorUtils.of(v), (k, v) -> k));
         assertTrue(mdl.getPredictionsAggregator() instanceof MeanValuePredictionsAggregator);
         assertEquals(5, mdl.getModels().size());
     }
\ No newline at end of file
