diff --git a/old_product.java b/new_product.java
index 915c937..9f3bf0a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -68,7 +68,7 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 			if (previousOrder == null) {
 				throw new APIException("Order.previous.required", (Object[]) null);
 			}
-			stopOrder(previousOrder, order.getDateActivated());
+			stopOrder(previousOrder, aMomentBefore(order.getDateActivated()));
 		} else if (DISCONTINUE == order.getAction()) {
 			discontinueExistingOrdersIfNecessary(order);
 		}
