diff --git a/old_test.java b/new_test.java
index 1ea55ab..d39dada 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
         assertEquals("", WordUtils.wrap("", -1));
         
         // normal
-        String systemNewLine = System.getProperty("line.separator");
+        final String systemNewLine = System.getProperty("line.separator");
         String input = "Here is one line of text that is going to be wrapped after 20 columns.";
         String expected = "Here is one line of" + systemNewLine + "text that is going" 
             + systemNewLine + "to be wrapped after" + systemNewLine + "20 columns.";
