diff --git a/old_product.java b/new_product.java
index bd66a4c..d3e037f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,9 +8,9 @@ public static String wrap(final String str, int wrapLength, String newLineStr, f
         if (wrapLength < 1) {
             wrapLength = 1;
         }
-        int inputLineLength = str.length();
+        final int inputLineLength = str.length();
         int offset = 0;
-        StringBuilder wrappedLine = new StringBuilder(inputLineLength + 32);
+        final StringBuilder wrappedLine = new StringBuilder(inputLineLength + 32);
         
         while (inputLineLength - offset > wrapLength) {
             if (str.charAt(offset) == ' ') {
