diff --git a/old_test.java b/new_test.java
index 69b1492..26e716a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,6 @@
         int r = 4;
         double p = 0.5;
         double expResult = 0.08203125;
-        double result = DiscreteDistributions.NegativeBinomial(n, r, p);
+        double result = DiscreteDistributions.negativeBinomial(n, r, p);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
