diff --git a/old_product.java b/new_product.java
index e2cb09b..f5c6926 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static double NegativeBinomial(int n, int r, double p) {
+public static double negativeBinomial(int n, int r, double p) {
         //tested its validity with http://www.mathcelebrity.com/binomialneg.php
         if(n<0 || r<0 || p<0) {
             throw new IllegalArgumentException("All the parameters must be positive.");
