diff --git a/old_product.java b/new_product.java
index c7bc991..89f2ae3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,6 @@
-protected byte[] createUrl(final byte[] url, final XQMap map) throws QueryException {
+protected byte[] createUrl(final byte[] url, final XQMap map, final byte[] anchor)
+      throws QueryException {
+
     final TokenBuilder tb = new TokenBuilder().add(url);
     int c = 0;
     for(final Item key : map.keys()) {
@@ -8,5 +10,6 @@ protected byte[] createUrl(final byte[] url, final XQMap map) throws QueryExcept
         tb.add('=').add(Token.encodeUri(value.string(info), false));
       }
     }
+    if(anchor.length > 0) tb.add('#').add(Token.encodeUri(anchor, false));
     return tb.finish();
   }
\ No newline at end of file
