diff --git a/old_test.java b/new_test.java
index d01a749..833016b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,8 @@
     query(func.args("url", " map { 'a': ('b','c') }"), "url?a=b&a=c");
     query(func.args("url", " map { 12: true() }"), "url?12=true");
 
+    query(func.args("url", " map { }", "a"), "url#a");
+
     error(func.args("url", " map { (): 'a' }"), EMPTYFOUND);
     error(func.args("url", " map { ('a','b'): () }"), SEQFOUND_X);
     error(func.args("url", " map { 'a': true#0 }"), FIATOM_X);
