diff --git a/old_product.java b/new_product.java
index 3730353..a6aadeb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Deprecated
-	public static GeoPoint PixelXYToLatLong(
+	public GeoPoint PixelXYToLatLong(
 			final int pixelX, final int pixelY, final double zoomLevel, final GeoPoint reuse) {
 		return getGeoFromMercator(pixelX, pixelY, MapSize(zoomLevel), reuse, true, true);
 	}
\ No newline at end of file
