diff --git a/old_test.java b/new_test.java
index 11f0e96..7841a9c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
         final int levelOfDetail = 8;
         final double delta = 1E-3;
 
-        final GeoPoint point = TileSystem.getGeoFromMercator(pixelX, pixelY, TileSystem.MapSize((double)levelOfDetail), null, true, true);
+        final GeoPoint point = tileSystem.getGeoFromMercator(pixelX, pixelY, TileSystem.MapSize((double)levelOfDetail), null, true, true);
 
         Assert.assertEquals(-179.752807617187, point.getLongitude(), delta);
         Assert.assertEquals(85.0297584051224, point.getLatitude(), delta);
