diff --git a/old_product.java b/new_product.java
index 3b1be34..79d608c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,7 +10,9 @@
             final String id,
             final int pid
     ) throws GenieException {
-        LOG.debug("Setting the id of process for job with id " + id + " to " + pid);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Setting the id of process for job with id " + id + " to " + pid);
+        }
         final Job job = this.jobRepo.findOne(id);
         if (job != null) {
             job.setProcessHandle(pid);
