diff --git a/old_product.java b/new_product.java
index 7f57e86..0a53ecb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
 public DocumentAuthors getAuthors(Map<String, Object> params) {
-		return executeRequest(params, AlchemyAPI.authors, DocumentAuthors.class, "html", "url");
-	}
\ No newline at end of file
+    return executeRequest(params, AlchemyAPI.authors, DocumentAuthors.class, HTML, URL);
+  }
\ No newline at end of file
