diff --git a/old_test.java b/new_test.java
index 9bb9067..bd0c290 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,5 +3,5 @@
       throws IOException, JSONException, RestServiceException, URISyntaxException {
     AdminBlobStorageService adminBlobStorageService = getAdminBlobStorageService();
     MessageInfo messageInfo = createMessageInfo(RestMethod.GET, "/", new JSONObject());
-    adminBlobStorageService.handleMessage(messageInfo);
+    adminBlobStorageService.handleGet(messageInfo);
   }
\ No newline at end of file
