diff --git a/old_test.java b/new_test.java
index 61ef08b..9588e52 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,19 +10,19 @@
         // Act & Assert
         StepVerifier.create(connection.createSession(SESSION_NAME))
             .assertNext(s -> {
-                Assert.assertNotNull(s);
-                Assert.assertEquals(SESSION_NAME, s.getSessionName());
-                Assert.assertTrue(s instanceof ReactorSession);
-                Assert.assertSame(session, ((ReactorSession) s).session());
+                Assertions.assertNotNull(s);
+                Assertions.assertEquals(SESSION_NAME, s.getSessionName());
+                Assertions.assertTrue(s instanceof ReactorSession);
+                Assertions.assertSame(session, ((ReactorSession) s).session());
             }).verifyComplete();
 
         // Assert that the same instance is obtained and we don't get a new session with the same name.
         StepVerifier.create(connection.createSession(SESSION_NAME))
             .assertNext(s -> {
-                Assert.assertNotNull(s);
-                Assert.assertEquals(SESSION_NAME, s.getSessionName());
-                Assert.assertTrue(s instanceof ReactorSession);
-                Assert.assertSame(session, ((ReactorSession) s).session());
+                Assertions.assertNotNull(s);
+                Assertions.assertEquals(SESSION_NAME, s.getSessionName());
+                Assertions.assertTrue(s instanceof ReactorSession);
+                Assertions.assertSame(session, ((ReactorSession) s).session());
             }).verifyComplete();
 
         verify(record, Mockito.times(1)).set(Handler.class, Handler.class, sessionHandler);
