diff --git a/old_product.java b/new_product.java
index fa18b14..2f5096e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,11 +7,11 @@
 
         return connectionMono.map(connection -> sessionMap.computeIfAbsent(sessionName, key -> {
             final SessionHandler handler = handlerProvider.createSessionHandler(connectionId, getHost(), sessionName,
-                connectionOptions.retry().tryTimeout());
+                connectionOptions.getRetry().getTryTimeout());
             final Session session = connection.session();
 
             BaseHandler.setHandler(session, handler);
             return new ReactorSession(session, handler, sessionName, reactorProvider, handlerProvider, getCBSNode(),
-                tokenResourceProvider, connectionOptions.retry().tryTimeout());
+                tokenResourceProvider, connectionOptions.getRetry().getTryTimeout());
         }));
     }
\ No newline at end of file
