diff --git a/old_product.java b/new_product.java
index c03c4e2..5ed34d8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
 void delete(final int p, final int s) {
+    final int sz = size;
     // find the node to deleted
     int i = find(p);
     // if the node is not directly contained as a child, either start at array index 0 or
@@ -9,7 +10,7 @@ void delete(final int p, final int s) {
     // number of nodes to be deleted
     int num = 0;
     // determine number of nodes to be deleted
-    for(int n = i; n < size && nodes[n].pre < upper; ++n, ++num);
+    for(int n = i; n < sz && nodes[n].pre < upper; ++n, ++num);
     // new size of child array
     size -= num;
 
@@ -18,7 +19,7 @@ void delete(final int p, final int s) {
       nodes = new NSNode[0];
     } else if(num > 0) {
       // otherwise remove nodes from the child array
-      System.arraycopy(nodes, i + num, nodes, i, size - i);
-      for(int n = size; n < size + num; n++) nodes[n] = null;
+      Array.remove(nodes, i, num, sz);
+      for(int n = size; n < sz; n++) nodes[n] = null;
     }
   }
\ No newline at end of file
