diff --git a/old_product.java b/new_product.java
index c9b78e5..a2138f2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-private B64Stream contentBinary(final QueryContext ctx) throws QueryException {
+private B64Stream binary(final QueryContext ctx) throws QueryException {
     final byte[] uri = checkStr(expr[0], ctx);
     return new B64Stream(IO.get(Token.string(uri)), BXFE_IO);
   }
\ No newline at end of file
