diff --git a/old_test.java b/new_test.java
index e5bb2dd..4e7d8b4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-  public void contentBinary() {
-    query(_FETCH_CONTENT_BINARY.args(FILE));
-    error(_FETCH_CONTENT_BINARY.args(FILE + 'x'), Err.BXFE_IO);
+  public void binary() {
+    query(_FETCH_BINARY.args(FILE));
+    error(_FETCH_BINARY.args(FILE + 'x'), Err.BXFE_IO);
   }
\ No newline at end of file
