diff --git a/old_test.java b/new_test.java
index 0f0f99a..2c3c533 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,6 @@
         Iterator<Relevance> hits = corpus.search(new BM25(), "romantic");
         while (hits.hasNext()) {
             Relevance hit = hits.next();
-            System.out.println(hit.doc() + "\t" + hit.score());
+            System.out.println(hit.text + "\t" + hit.score);
         }
     }
\ No newline at end of file
