diff --git a/old_test.java b/new_test.java
index 6b0f421..4d644c1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-	public void saveConcept_shouldLeavePreferredNamePreferredIfSet() throws Exception {
+	public void saveConcept_shouldLeavePreferredNamePreferredIfSet() {
 		Locale loc = new Locale("fr", "CA");
 		ConceptName fullySpecifiedName = new ConceptName("fully specified", loc);
 		fullySpecifiedName.setConceptNameType(ConceptNameType.FULLY_SPECIFIED); //be explicit for test case
