diff --git a/old_product.java b/new_product.java
index 7de06ea..3ab52d3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -114,6 +114,11 @@ public Concept saveConcept(Concept concept) throws APIException {
 		concept.setDateChanged(new Date());
 		concept.setChangedBy(Context.getAuthenticatedUser());
 		
+		// force isSet when concept has members
+		if (!concept.isSet() && (concept.getSetMembers().size() > 0)) {
+                    concept.setSet(true);
+		}
+
 		Concept conceptToReturn = dao.saveConcept(concept);
 		
 		// add/remove entries in the concept_word table (used for searching)
