diff --git a/old_product.java b/new_product.java
index 456981d..afc36b2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,11 @@
 public static String substringWithoutLast(final CharSequence text,final String lastString){
-        if (Validator.isNullOrEmpty(text)){
+        if (null == text){
             return StringUtils.EMPTY;
         }
 
-        //由于上面的循环中,最后一个元素可能是null或者empty,判断加还是不加拼接符有点麻烦,因此,循环中统一拼接,但是循环之后做截取处理
-        String returnValue = text.toString();
-        if (Validator.isNullOrEmpty(lastString)){
-            return returnValue;
+        String textString = text.toString();
+        if (null == lastString){
+            return textString;
         }
-        return returnValue.endsWith(lastString) ? substringWithoutLast(returnValue, lastString.length()) : returnValue;
+        return textString.endsWith(lastString) ? substringWithoutLast(textString, lastString.length()) : textString;
     }
\ No newline at end of file
