diff --git a/old_test.java b/new_test.java
index f909159..eef59c3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
-	public void isCurrent_shouldReturnFalseForAVoidedOrder() throws Exception {
+	public void isActive_shouldReturnFalseForAVoidedOrder() throws Exception {
 		Order order = new Order();
 		order.setVoided(true);
 		order.setDateActivated(DateUtils.parseDate("2014-11-01 11:11:10", DATE_FORMAT));
 		assertNull(order.getDateStopped());
 		assertNull(order.getAutoExpireDate());
-		assertFalse(order.isCurrent(DateUtils.parseDate("2014-11-01 11:11:11", DATE_FORMAT)));
+		assertFalse(order.isActive(DateUtils.parseDate("2014-11-01 11:11:11", DATE_FORMAT)));
 	}
\ No newline at end of file
