diff --git a/old_test.java b/new_test.java
index 780324c..85171ba 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
         StepVerifier.create(client.getProperties())
             .assertNext(properties -> {
                 Assert.assertNotNull(properties);
-                Assert.assertEquals(eventHubPath, properties.path());
+                Assert.assertEquals(eventHubName, properties.name());
                 Assert.assertEquals(expectedPartitionIds.length, properties.partitionIds().length);
             }).verifyComplete();
     }
\ No newline at end of file
