diff --git a/old_test.java b/new_test.java
index a93f958..4ffd540 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
   public void completeUfsFileTest() throws Exception {
     long id = mManager.createFile(SESSION_ID, mUri,
-        new PermissionStatus("", "", Constants.DEFAULT_FS_FULL_PERMISSION));
+        new Permission("", "", Constants.DEFAULT_FILE_SYSTEM_MODE));
     mManager.completeFile(SESSION_ID, id,
-        new PermissionStatus("", "", Constants.DEFAULT_FS_FULL_PERMISSION));
+        new Permission("", "", Constants.DEFAULT_FILE_SYSTEM_MODE));
     Mockito.verify(mMockUfs).rename(Mockito.contains(mUri.toString()), Mockito.eq(mUri.toString()));
   }
\ No newline at end of file
