diff --git a/old_product.java b/new_product.java
index 87463f4..583cfe7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,5 +14,5 @@
         action = new ContextualBiConsumer<>(contextDescriptor, action);
 
         CompletableFuture<Void> dependentStage = completableFuture.thenAcceptBoth(other, action);
-        return new ManagedCompletableFuture<Void>(dependentStage, defaultExecutor);
+        return new ManagedCompletableFuture<Void>(dependentStage, defaultExecutor, null);
     }
\ No newline at end of file
