diff --git a/old_product.java b/new_product.java
index 2d0e916..b617ee6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public List<Obs> getObservations(List<Person> whom, List<Encounter> encounters, List<Concept> questions,
 	        List<Concept> answers, List<PERSON_TYPE> personTypes, List<Location> locations, List<String> sort,
-	        Integer mostRecentN, Integer obsGroupId, Date fromDate, Date toDate, boolean includeVoidedObs)
-	        throws DAOException;
\ No newline at end of file
+	        Integer mostRecentN, Integer obsGroupId, Date fromDate, Date toDate, boolean includeVoidedObs,
+	        String accessionNumber) throws DAOException;
\ No newline at end of file
