diff --git a/old_test.java b/new_test.java
index f1c2daa..0200a13 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,17 +10,17 @@
 		obsListExpected = session.createCriteria(Obs.class, "obs").addOrder(Order.desc("id")).list();
 		
 		obsListActual = dao.getObservations(null, null, null, null, null, null, Arrays.asList("id"), null, null, null, null,
-		    false);
+		    false, null);
 		Assert.assertArrayEquals(obsListExpected.toArray(), obsListActual.toArray());
 		
 		obsListActual = dao.getObservations(null, null, null, null, null, null, Arrays.asList("id desc"), null, null, null,
-		    null, false);
+		    null, false, null);
 		Assert.assertArrayEquals(obsListExpected.toArray(), obsListActual.toArray());
 		
 		//Order by id asc
 		obsListExpected = session.createCriteria(Obs.class, "obs").addOrder(Order.asc("id")).list();
 		obsListActual = dao.getObservations(null, null, null, null, null, null, Arrays.asList("id asc"), null, null, null,
-		    null, false);
+		    null, false, null);
 		Assert.assertArrayEquals(obsListExpected.toArray(), obsListActual.toArray());
 		
 		//Order by person_id asc and id desc
@@ -28,7 +28,7 @@
 		    Order.desc("id")).list();
 		
 		obsListActual = dao.getObservations(null, null, null, null, null, null, Arrays.asList("person.id asc", "id"), null,
-		    null, null, null, false);
+		    null, null, null, false, null);
 		Assert.assertArrayEquals(obsListExpected.toArray(), obsListActual.toArray());
 		
 		//Order by person_id asc and id asc
@@ -36,6 +36,6 @@
 		        .addOrder(Order.asc("id")).list();
 		
 		obsListActual = dao.getObservations(null, null, null, null, null, null, Arrays.asList("person.id asc", "id asc"),
-		    null, null, null, null, false);
+		    null, null, null, null, false, null);
 		Assert.assertArrayEquals(obsListExpected.toArray(), obsListActual.toArray());
 	}
\ No newline at end of file
