diff --git a/old_product.java b/new_product.java
index 06623a0..5175221 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 @Deterministic @Requires(Permission.NONE)
-  public ANode buffer(final ANode node, final ANum distance) throws QueryException {
+  public ANode buffer(final ANode node, final Dbl distance) throws QueryException {
     return gmlWriter(checkGeo(node).buffer(distance.dbl()));
   }
\ No newline at end of file
