diff --git a/old_test.java b/new_test.java
index bce4f47..9b259c1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,6 @@
             throws IOException
     {
         // would crash with java.lang.IllegalArgumentException: Points of LinearRing do not form a closed linestring if Invalid syntax is not skipped
-        SpatialRDD geojsonRDD = GeometryReader.readToGeometryRDD(sc, invalidSyntaxGeoJsonGeomWithFeatureProperty,
-                FileDataSplitter.GEOJSON, false, true);
+        SpatialRDD geojsonRDD = GeoJsonReader.readToGeometryRDD(sc, invalidSyntaxGeoJsonGeomWithFeatureProperty, false, true);
         assertEquals(geojsonRDD.rawSpatialRDD.count(), 1);
     }
\ No newline at end of file
