diff --git a/old_product.java b/new_product.java
index 16287db..bcc8384 100644
--- a/old_product.java
+++ b/new_product.java
@@ -19,6 +19,7 @@ void usage(boolean error) {
     out.println("  --hex                   To use hexadecimal encoding for key and value");
     out.println("  --ascii                 To use ASCII encoding for key and value");
     out.println("  --url                   To use URL encoding for key and value");
+    out.println("  --counter               To interpret value as a long counter");
     out.println("  --start <n>             To start at the nth element - only for list");
     out.println("  --limit <k>             To list at most k elements - only for list");
     out.println("  --all                   To format all data");
