diff --git a/old_test.java b/new_test.java
index fdeba6b..e7e0052 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,30 +1,6 @@
 @Test
   public void testUsage() throws Exception {
 
-    // configure a gateway
-    final String name = "access.rest";
-    final String prefix = "/continuuity";
-    final String path = "/table/";
-    final int port = Util.findFreePort();
-
-    CConfiguration configuration = new CConfiguration();
-    configuration.set(Constants.CONFIG_CONNECTORS, name);
-    configuration.set(Constants.buildConnectorPropertyName(name,
-        Constants.CONFIG_CLASSNAME), RestAccessor.class.getCanonicalName());
-    configuration.setInt(Constants.buildConnectorPropertyName(name,
-        Constants.CONFIG_PORT), port);
-    configuration.set(Constants.buildConnectorPropertyName(name,
-        Constants.CONFIG_PATH_PREFIX), prefix);
-    configuration.set(Constants.buildConnectorPropertyName(name,
-        Constants.CONFIG_PATH_MIDDLE), path);
-
-    // Now create our Gateway with a dummy consumer (we don't run collectors)
-    // and make sure to pass the data fabric executor to the gateway.
-    Gateway gateway = new Gateway();
-    gateway.setExecutor(this.executor);
-    gateway.setConsumer(new Util.NoopConsumer());
-    gateway.start(null, configuration);
-
     // argument combinations that should return success
     String[][] goodArgsList = {
         { "--help" }, // print help
@@ -97,7 +73,4 @@
       LOG.info("Testing: " + Arrays.toString(args));
       Assert.assertNull(new DataClient().execute(args, configuration));
     }
-
-    // and shut down
-    gateway.stop(false);
   }
\ No newline at end of file
