diff --git a/old_product.java b/new_product.java
index 5fa233b..1676d7b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -47,6 +47,6 @@ public void trainC(ClassificationDataSet dataSet, Set<Integer> categoriesToUse)
                     cordinate[j] = dp.getCategoricalValue(realIndexToCatIndex[j]);
                 else
                     cordinate[j] = dataSet.getDataPointCategory(i);
-            countArray[cordToIndex(cordinate)]+= dp.getWeight();
+            countArray[cordToIndex(cordinate)]+= dataSet.getWeight(i);
         }
     }
\ No newline at end of file
