diff --git a/old_test.java b/new_test.java
index 9aeeb85..5ecf32e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,7 @@
 @Test
   public void createUfsFileTest() throws Exception {
-    UnderFileSystem mockUfs = Mockito.mock(UnderFileSystem.class);
-    OutputStream mockOutputStream = Mockito.mock(OutputStream.class);
-    Mockito.when(mockUfs.create(Mockito.anyString())).thenReturn(mockOutputStream);
-    PowerMockito.mockStatic(UnderFileSystem.class);
-    BDDMockito.given(UnderFileSystem.get(Mockito.anyString(), Mockito.any(Configuration.class)))
-        .willReturn(mockUfs);
-
-    String testFile = "test";
+    String uniqPath = PathUtils.uniqPath();
     UnderFileSystemManager manager = new UnderFileSystemManager();
-    manager.createFile(testFile);
-    Mockito.verify(mockUfs).create(Mockito.contains(testFile));
+    manager.createFile(uniqPath);
+    Mockito.verify(mMockUfs).create(Mockito.contains(uniqPath));
   }
\ No newline at end of file
