diff --git a/old_product.java b/new_product.java
index 96b7fe4..d44e4dc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-public long createUfsFile(String ufsPath, UFSCreateFileTOptions options) throws alluxio.thrift.AlluxioTException, alluxio.thrift.ThriftIOException, org.apache.thrift.TException;
\ No newline at end of file
+public long createUfsFile(String ufsPath, CreateUfsFileTOptions options) throws alluxio.thrift.AlluxioTException, alluxio.thrift.ThriftIOException, org.apache.thrift.TException;
\ No newline at end of file
