diff --git a/old_test.java b/new_test.java
index 3c20109..2a50db3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void testGetProcessDiagramLayout() throws Exception {
-    Map<String, Bounds> processDiagramLayout;
+    DiagramLayout processDiagramLayout;
     if (1 == processDefinitionQuery.count()) {
       ProcessDefinition processDefinition = processDefinitionQuery.singleResult();
       assertNotNull(processDefinition);
@@ -12,7 +12,7 @@
       InputStream imageStream = new FileInputStream("src/test/resources/org/activiti/engine/test/api/repository/diagram/" + imageFileName);
       assertNotNull(bpmnXmlStream);
       assertNotNull(imageStream);
-      processDiagramLayout = new ProcessDiagramLayout().getProcessDiagramLayout(bpmnXmlStream, imageStream);
+      processDiagramLayout = new ProcessDiagramLayoutFactory().getProcessDiagramLayout(bpmnXmlStream, imageStream);
     }
     assertLayoutCorrect(processDiagramLayout);
   }
\ No newline at end of file
