diff --git a/old_product.java b/new_product.java
index ecce39e..e166cbd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,7 +10,7 @@ protected void splitPixels( int indexStart , int indexStop ) {
 		line.p.set(a.x,a.y);
 		line.slope.set(c.x-a.x,c.y-a.y);
 
-		int indexSplit = selectSplitOffset(indexStart,indexStop);
+		int indexSplit = selectSplitBetween(indexStart, indexStop);
 
 		if( indexSplit >= 0 ) {
 			splitPixels(indexStart, indexSplit);
