diff --git a/old_product.java b/new_product.java
index 96f4c13..a8c3e1b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -18,18 +18,22 @@ public void process() {
         if (index == -1) {
             index = 0;
         }
-        if (log.isDebugEnabled()) {
-            log.debug("Check for matches against: " + sampler.toString());
-        }
+        /*
+         * if (log.isDebugEnabled()) { log.debug("Check for matches against: " +
+         * sampler.toString()); }
+         *
+         */
         html = (Document) HtmlParsingUtils.getDOM(responseText.substring(index));
         addAnchorUrls(html, result, sampler);
         addFormUrls(html, result, sampler);
         addFramesetUrls(html, result, sampler);
         if (hasNotVisited()) {
             HTTPSamplerBase url = getNextLink();
-            if (log.isDebugEnabled()) {
-                log.debug("Selected: " + url.toString());
-            }
+            /*
+             * if (log.isDebugEnabled()) { log.debug("Selected: " +
+             * url.toString()); }
+             *
+             */
             sampler.setDomain(url.getDomain());
             sampler.setPath(url.getPath());
             if (url.getMethod().equals(HTTPConstants.POST)) {
@@ -44,6 +48,7 @@ public void process() {
             }
             sampler.setProtocol(url.getProtocol());
         } else {
-            log.debug("No matches found");
+            log.info("No further matches found, stopping test");
+            context.getEngine().askThreadsToStop();
         }
     }
\ No newline at end of file
