diff --git a/old_test.java b/new_test.java
index 801d093..5affc81 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,8 +5,9 @@
         JMeterContext context = JMeterContextService.getContext();
         Sampler s = new HTTPSampler();
         HTTPSampleResult res = new HTTPSampleResult();
-        res.setURL(new URL("http://localhost/"));
-        String data="<a href='http://test'/><a href='http://test2'/><a href='http://test'/>";
+        res.setURL(new URL("http://test/"));
+        String data = "<a href='http://test'/><a href='http://test/2'/><a href='http://test'/>";
+        data += "<a href='testsub'/><a href='http://test/2'/><a href='http://test'/>";
         res.setResponseData(data.getBytes());
         context.setPreviousResult(res);
         context.setCurrentSampler(s);
