diff --git a/old_product.java b/new_product.java
index 8b28b4b..c79456c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public static RxDocumentServiceRequest create(OperationType operation,
             ResourceType resourceType,
             String relativePath,
-            Observable<byte[]> content,
+            Flux<byte[]> content,
             Map<String, String> headers) {
         return new RxDocumentServiceRequest(operation, resourceType, relativePath, content, headers, AuthorizationTokenType.PrimaryMasterKey);
     }
\ No newline at end of file
