diff --git a/old_test.java b/new_test.java
index 852cd4d..8fc0c93 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void create() {
         final Document docDefinition = getDocumentDefinition();
 
-        Observable<ResourceResponse<Document>> createObservable = client.createDocument(
+        Flux<ResourceResponse<Document>> createObservable = client.createDocument(
             this.getCollectionLink(), docDefinition, null, false);
 
         ResourceResponseValidator<Document> validator = new ResourceResponseValidator.Builder<Document>()
