diff --git a/old_product.java b/new_product.java
index 194cbc8..8d86840 100644
--- a/old_product.java
+++ b/new_product.java
@@ -66,7 +66,7 @@ private byte[] string() throws QueryException {
           default:
             throw error("Unknown character escape: '\\%'", n);
         }
-      } else if(spec != Spec.LIBERAL && cp <= 0x1F) {
+      } else if(spec != JsonSpec.LIBERAL && cp <= 0x1F) {
         throw error("Non-escaped control character: '\\%'", CTRL[cp]);
       }
 
