diff --git a/old_product.java b/new_product.java
index 783d50f..65a5f25 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
-AppUsagePerApiBean getAppUsagePerApi(String organizationId, String applicationId, String version,
+ClientUsagePerApiBean getClientUsagePerApi(String organizationId, String clientId, String version,
             DateTime from, DateTime to);
\ No newline at end of file
