diff --git a/old_product.java b/new_product.java
index f7968bc..e10041a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static int bitPolyDivide(int data , int generator , int totalBits, int dataBits) {
+public static int bitPolyModulus(int data , int generator , int totalBits, int dataBits) {
 		int errorBits = totalBits-dataBits;
 		for (int i = dataBits-1; i >= 0; i--) {
 			if( (data & (1 << (i+errorBits))) != 0 ) {
