diff --git a/old_product.java b/new_product.java
index 89fc8e7..590203e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-CompletableFuture<Void> write(String streamSegmentName, long offset, InputStream data, int length, Duration timeout);
\ No newline at end of file
+CompletableFuture<Void> write(SegmentHandle handle, long offset, InputStream data, int length, Duration timeout);
\ No newline at end of file
