diff --git a/old_test.java b/new_test.java
index 59ea416..bd83d01 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
     public void testCollectResults() {
-	WorkflowTrace t = new WorkflowTrace();
+	TestVector t = new TestVector(null, null, null);
 	agent.collectResults(new File("../resources/testsuite/EvolutionaryFuzzer/PinTest/test.trace"), t, t);
     }
\ No newline at end of file
