diff --git a/old_product.java b/new_product.java
index 94b6cb9..dda38cf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-public abstract Result collectResults(File branchTrace, WorkflowTrace trace, WorkflowTrace executedTrace);
\ No newline at end of file
+public abstract Result collectResults(File branchTrace, TestVector vector, TestVector executedVector);
\ No newline at end of file
