diff --git a/old_test.java b/new_test.java
index 27ffd79..bcb3574 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test
 	public void testGetOne() {
-		ModelMap model = new ModelMap();
-		userController.getOne(getTestUser().getUserId(), model);
-		User user = (User) model.get("user");
+		Map<String, Object> result = userApiController.getOne(getTestUser().getUserId());
+		User user = (User) result.get("user");
 		assertThat(user.getId(), is(getTestUser().getId()));
 	}
\ No newline at end of file
