diff --git a/old_product.java b/new_product.java
index a0b56e4..796e6cf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @RestAPI
 	@GetMapping("/profile")
-	public HttpEntity<String> getOne(User user) {
+	public Map<String, Object> getOne(User user) {
 		checkNotEmpty(user.getUserId(), "UserID should not be NULL!");
 		Map<String, Object> model = new HashMap<>();
 		User one = userService.getOneWithFollowers(user.getUserId());
@@ -9,5 +9,5 @@
 		model.put("allowRoleChange", false);
 		model.put("showPasswordByDefault", false);
 		attachCommonAttribute(one, model);
-		return toJsonHttpEntity(model);
+		return model;
 	}
\ No newline at end of file
