diff --git a/old_product.java b/new_product.java
index 8fac2f5..11678b2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static String combineQueryString(Map<String, String[]> paramsMap,String charsetType){
+public static String parseArrayValueMapToQueryString(Map<String, String[]> paramsMap,String charsetType){
         if (Validator.isNullOrEmpty(paramsMap)){
             return StringUtils.EMPTY;
         }
