diff --git a/old_test.java b/new_test.java
index 0b54dbf..1966407 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,8 +3,8 @@
         Map<String, String[]> keyAndArrayMap = new HashMap<String, String[]>();
         keyAndArrayMap.put("a", new String[] { "aaaa", "bbbb" });
         String charsetType = CharsetType.UTF8;
-        LOGGER.info(ParamUtil.combineQueryString(keyAndArrayMap, charsetType));
-        LOGGER.info(ParamUtil.combineQueryString(null, charsetType));
-        LOGGER.info(ParamUtil.combineQueryString(null, null));
-        LOGGER.info(ParamUtil.combineQueryString(keyAndArrayMap, null));
+        LOGGER.info(ParamUtil.parseArrayValueMapToQueryString(keyAndArrayMap, charsetType));
+        LOGGER.info(ParamUtil.parseArrayValueMapToQueryString(null, charsetType));
+        LOGGER.info(ParamUtil.parseArrayValueMapToQueryString(null, null));
+        LOGGER.info(ParamUtil.parseArrayValueMapToQueryString(keyAndArrayMap, null));
     }
\ No newline at end of file
