diff --git a/old_product.java b/new_product.java
index 8a1a7d2..8d84981 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,10 +8,10 @@ private void load(AlluxioURI filePath, int replication)
         load(newPath, replication);
       }
     } else {
-      Thread thread = JobThriftClientUtils.createProgressThread(System.out);
+      Thread thread = JobGrpcClientUtils.createProgressThread(System.out);
       thread.start();
       try {
-        JobThriftClientUtils.run(new LoadConfig(filePath.getPath(), replication), 3);
+        JobGrpcClientUtils.run(new LoadConfig(filePath.getPath(), replication), 3);
       } finally {
         thread.interrupt();
       }
