diff --git a/old_test.java b/new_test.java
index d8aef87..370d772 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     // write a file outside of Alluxio
     AlluxioURI filePath = new AlluxioURI(TEST_URI);
     FileOutStream os = mFileSystem.createFile(filePath,
-        CreateFileOptions.defaults().setWriteType(WriteType.THROUGH));
+        CreateFilePOptions.newBuilder().setWriteType(WritePType.THROUGH).build());
     os.write((byte) 0);
     os.write((byte) 1);
     os.close();
