diff --git a/old_test.java b/new_test.java
index eac0257..5a567f5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,9 @@
-@TestInfo(
-      level = TestLevel.COMPLETE,
-      purpose = "",
-      targets = {
-        @TestTarget(
-          methodName = "clear",
-          methodArgs = {}
-        )
-    })
+@TestTargetNew(
+        level = TestLevel.COMPLETE,
+        notes = "",
+        method = "clear",
+        args = {}
+    )
     public void test_clear() {
         // Test for method void java.util.TreeSet.clear()
         ts.clear();
