diff --git a/old_product.java b/new_product.java
index ea6cc35..bac2747 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,26 +1,30 @@
-protected void generateReport(String templateName, OutputStream outputStream) throws ReportException {
+protected void processTemplate(String templateName, OutputStream outputStream) throws ReportException {
         InputStream input = null;
-        String templatePath = null;
+        String logTag = null;
         final File f = new File(templateName);
         try {
-            if (f.exists() && f.isFile()) {
+            if (f.isFile()) {
                 try {
-                    templatePath = templateName;
+                    logTag = templateName;
                     input = new FileInputStream(f);
                 } catch (FileNotFoundException ex) {
                     throw new ReportException("Unable to locate template file: " + templateName, ex);
                 }
             } else {
-                templatePath = "templates/" + templateName + ".vsl";
-                input = this.getClass().getClassLoader().getResourceAsStream(templatePath);
+                logTag = "templates/" + templateName + ".vsl";
+                input = this.getClass().getClassLoader().getResourceAsStream(logTag);
             }
             if (input == null) {
-                throw new ReportException("Template file doesn't exist: " + templatePath);
+                logTag = templateName;
+                input = this.getClass().getClassLoader().getResourceAsStream(templateName);
+            }
+            if (input == null) {
+                throw new ReportException("Template file doesn't exist: " + logTag);
             }
 
             try (InputStreamReader reader = new InputStreamReader(input, "UTF-8");
                     OutputStreamWriter writer = new OutputStreamWriter(outputStream, "UTF-8")) {
-                if (!velocityEngine.evaluate(context, writer, templatePath, reader)) {
+                if (!velocityEngine.evaluate(context, writer, logTag, reader)) {
                     throw new ReportException("Failed to convert the template into html.");
                 }
                 writer.flush();
