diff --git a/old_test.java b/new_test.java
index ef2ab63..2b2e0e7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,9 +5,9 @@
 
         int index = 1;
         RunTable instance = createHorizontalInstance();
-        short[] expResult = new short[]{1, 3, 2};
-        System.out.println("expResult: " + Arrays.toString(expResult));
+        RunSequence expResult = new RunSequence(new short[]{1, 3, 2});
+        System.out.println("expResult: " + expResult);
 
-        short[] result = instance.getSequence(index);
-        Assert.assertArrayEquals(expResult, result);
+        RunSequence result = instance.getSequence(index);
+        assertEquals(expResult, result);
     }
\ No newline at end of file
