diff --git a/old_product.java b/new_product.java
index 9f28c12..b1cbffc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -21,6 +21,11 @@ public synchronized void addFeature(TransactionTrace txn, String key, Object val
             if (debug) LOG.debug("Adding new attribute " + key + " [" + type + "]");
             this.attributes.put(key, type);
             this.attribute_histograms.put(key, new Histogram());
+            this.attribute_types.put(key, VoltType.NULL);
+        }
+        // HACK
+        if (val != null && (val.getClass().equals(int.class) || val.getClass().equals(Integer.class))) {
+            val = new Long((Integer)val);
         }
         
         // Always store the values in a histogram so we can normalize them later on
@@ -51,5 +56,10 @@ public synchronized void addFeature(TransactionTrace txn, String key, Object val
             if (trace) LOG.trace("Increased FeatureSet size to " + this.last_num_attributes + " attributes");
         }
         this.txn_values.get(txn_id).set(idx, val);
+        
+        if (val != null && this.attribute_types.get(key) == VoltType.NULL) {
+            this.attribute_types.put(key, VoltType.typeFromClass(val.getClass()));
+        }
+        
         if (trace) LOG.trace(txn_id + ": " + key + " => " + val);
     }
\ No newline at end of file
