diff --git a/old_test.java b/new_test.java
index ec9ae48..150827e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,8 +17,8 @@
         // Message for open and close.
         mockMessageService.dispatchMessage(anyObject(), anyString());
         expect(expectLastCall()).times(2);
-        expect(mockCommunicator.openCommPort(ConnectionDriver.JSERIALCOMM, port, baud)).andReturn(true).once();
-        mockCommunicator.closeCommPort();
+        mockCommunicator.connect(ConnectionDriver.JSERIALCOMM, port, baud);
+        mockCommunicator.disconnect();
         expect(expectLastCall()).once();
         replay(instance, mockCommunicator, mockListener);
 
