diff --git a/old_product.java b/new_product.java
index 61ee33d..2c777fa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public static void deriv_I8(ImageInt8 orig,
 								ImageInt16 derivX,
 								ImageInt16 derivY) {
-		InputSanityCheck.checkSameShape(orig, derivX, derivY);
+		InputSanityCheck.checkSameShape(orig, false, derivX, true, derivY, true);
 		GradientThree_Standard.deriv_I8(orig, derivX, derivY);
 	}
\ No newline at end of file
