diff --git a/old_product.java b/new_product.java
index db4f581..b3b738a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,8 @@
 public static RectangleLength2D_F32 centerBoxInside(int srcWidth, int srcHeight,
-														PixelTransform2_F32 transform) {
+														PixelTransform<Point2D_F32> transform ,
+														Point2D_F32 work ) {
 
-		List<Point2D_F32> points = computeBoundingPoints(srcWidth, srcHeight, transform);
+		List<Point2D_F32> points = computeBoundingPoints(srcWidth, srcHeight, transform, work);
 
 		Point2D_F32 center = new Point2D_F32();
 		UtilPoint2D_F32.mean(points,center);
