diff --git a/old_product.java b/new_product.java
index 9e9917b..3fc542c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
 public List<Encounter> getEncounters(Patient patient, Location location, Date fromDate, Date toDate,
 	                                     Collection<Form> enteredViaForms, Collection<EncounterType> encounterTypes,
-	                                     boolean includeVoided);
\ No newline at end of file
+	                                     Collection<User> providers, boolean includeVoided);
\ No newline at end of file
