diff --git a/old_product.java b/new_product.java
index 655b583..b742170 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,7 +13,7 @@ public static Specification<ClusterEntity> findByClusterAndCommandCriteria(
             if (commandCriteria != null && !commandCriteria.isEmpty()) {
                 predicates.add(
                         cb.like(
-                                commands.get(CommandEntity_.sortedTags),
+                                commands.get(CommandEntity_.tags),
                                 JpaSpecificationUtils.getTagLikeString(commandCriteria)
                         )
                 );
@@ -22,7 +22,7 @@ public static Specification<ClusterEntity> findByClusterAndCommandCriteria(
             if (clusterCriteria != null && clusterCriteria.getTags() != null && !clusterCriteria.getTags().isEmpty()) {
                 predicates.add(
                         cb.like(
-                                root.get(ClusterEntity_.sortedTags),
+                                root.get(ClusterEntity_.tags),
                                 JpaSpecificationUtils.getTagLikeString(clusterCriteria.getTags())
                         )
                 );
