diff --git a/old_test.java b/new_test.java
index 3f47610..a2df6e2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,7 +11,7 @@
         Mockito.verify(this.cb, Mockito.times(1))
                 .equal(Mockito.eq(this.root.get(ClusterEntity_.status)), Mockito.eq(ClusterStatus.UP));
         Mockito.verify(this.cb, Mockito.times(1))
-                .like(Mockito.eq(this.root.get(ClusterEntity_.sortedTags)), Mockito.eq(this.tagLikeStatement));
+                .like(Mockito.eq(this.root.get(ClusterEntity_.tags)), Mockito.eq(this.tagLikeStatement));
         Mockito.verify(this.cb, Mockito.times(1))
-                .like(Mockito.eq(this.commands.get(CommandEntity_.sortedTags)), Mockito.eq(this.commandLikeStatement));
+                .like(Mockito.eq(this.commands.get(CommandEntity_.tags)), Mockito.eq(this.commandLikeStatement));
     }
\ No newline at end of file
