diff --git a/old_test.java b/new_test.java
index 6533038..d99d3a4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void startNewSpan() {
-        brave.serverSpanThreadBinder().setCurrentSpan(ServerSpan.create(PARENT_CONTEXT, "name"));
+        brave.serverTracer().setStateCurrentTrace(PARENT_CONTEXT, "name");
 
         PowerMockito.when(System.nanoTime()).thenReturn(500L);
 
