diff --git a/old_product.java b/new_product.java
index f732c19..d7dee6d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,13 +9,13 @@ public SpanId startNewSpan(@Nullable String requestName) {
             return null;
         }
 
-        SpanId nextContext = spanIdFactory().next(maybeParent());
+        Span newSpan = spanFactory().newSpan(maybeParent());
+        SpanId nextContext = Brave.context(newSpan);
         if (Boolean.FALSE.equals(nextContext.sampled())) {
             currentSpan().setCurrentSpan(null);
             return null;
         }
 
-        Span newSpan = Brave.newSpan(nextContext).setName(requestName);
-        currentSpan().setCurrentSpan(newSpan);
+        currentSpan().setCurrentSpan(newSpan.setName(requestName));
         return nextContext;
     }
\ No newline at end of file
