diff --git a/old_product.java b/new_product.java
index 04b5b92..21b0690 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,14 +10,12 @@
 				return super.rename(Mono.just(command));
 			}
 
-			Observable<Boolean> result = cmd.dump(command.getKey().array())
-					.switchIfEmpty(Observable.error(new RedisSystemException("Cannot rename key that does not exist",
+			Flux<Boolean> result = cmd.dump(command.getKey())
+					.otherwiseIfEmpty(Mono.error(new RedisSystemException("Cannot rename key that does not exist",
 							new RedisException("ERR no such key."))))
-					.concatMap(value -> cmd.restore(command.getNewName().array(), 0, value)
-							.concatMap(res -> cmd.del(command.getKey().array())))
+					.flatMap(value -> cmd.restore(command.getNewName(), 0, value).flatMap(res -> cmd.del(command.getKey())))
 					.map(LettuceConverters.longToBooleanConverter()::convert);
 
-			return LettuceReactiveRedisConnection.<Boolean> monoConverter().convert(result)
-					.map(val -> new ReactiveRedisConnection.BooleanResponse<>(command, val));
+			return result.map(val -> new ReactiveRedisConnection.BooleanResponse<>(command, val));
 		}));
 	}
\ No newline at end of file
