diff --git a/old_product.java b/new_product.java
index e9a7207..6656d6e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -19,8 +19,8 @@
                     }
                 }
             }
-            for (Dependency d : dependenciesToRemove) {
+            dependenciesToRemove.forEach((d) -> {
                 engine.removeDependency(d);
-            }
+            });
         }
     }
\ No newline at end of file
