diff --git a/old_test.java b/new_test.java
index 9ce4b3a..ce54bb5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,11 +3,13 @@
         Dependency dependency = new Dependency();
         dependency.setFileName("pom.xml");
         dependency.setFilePath("pom.xml");
-        dependency.addIdentifier("cpe", "cpe:/a:file:file:1.2.1", "http://some.org/url");
+        Cpe cpe = builder.part(Part.APPLICATION).vendor("file").product("file").version("1.2.1").build();
+        CpeIdentifier id = new CpeIdentifier(cpe, "http://some.org/url", Confidence.HIGHEST);
+        dependency.addVulnerableSoftwareIdentifier(id);
         Engine engine = null;
         FalsePositiveAnalyzer instance = new FalsePositiveAnalyzer();
-        int before = dependency.getIdentifiers().size();
+        int before = dependency.getVulnerableSoftwareIdentifiers().size();
         instance.analyze(dependency, engine);
-        int after = dependency.getIdentifiers().size();
+        int after = dependency.getVulnerableSoftwareIdentifiers().size();
         assertTrue(before > after);
     }
\ No newline at end of file
