diff --git a/old_test.java b/new_test.java
index fe1e5a1..fbffcee 100644
--- a/old_test.java
+++ b/new_test.java
@@ -32,12 +32,12 @@
             .build();
 
         Mockito.when(this.jobPersistenceService.createJobRequest(jobRequest, clientHost)).thenReturn(jobRequest);
-        final ArgumentCaptor<Job> argument = ArgumentCaptor.forClass(Job.class);
 
         this.jobCoordinatorService.coordinateJob(jobRequest, clientHost);
 
-        Mockito.verify(this.jobSubmitterService, Mockito.times(1)).submitJob(jobRequest);
+        Mockito.verify(this.taskExecutor, Mockito.times(1)).execute(Mockito.any(JobLauncher.class));
 
+        final ArgumentCaptor<Job> argument = ArgumentCaptor.forClass(Job.class);
         Mockito.verify(this.jobPersistenceService).createJob(argument.capture());
 
         Assert.assertEquals(JOB_1_ID, argument.getValue().getId());
