diff --git a/old_product.java b/new_product.java
index 40176b5..f6d5341 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,9 +7,9 @@ public void abort()
         try
         {
             // Rollback if we have a database connection, and it is NOT Read Only
-            if (isValid() && !connection.isClosed() && !isReadOnly())
+            if (isValid() && !isReadOnly())
             {
-                connection.rollback();
+                dbConnection.rollback();
             }
         }
         catch (SQLException se)
@@ -20,16 +20,15 @@ public void abort()
         {
             try
             {
-                // Free the DB connection
-                // If connection is closed or null, this is a no-op
-                DatabaseManager.freeConnection(connection);
+                if (!dbConnection.isSessionAlive())
+                {
+                    dbConnection.closeDBConnection();
+                }
             }
             catch (Exception ex)
             {
                 log.error("Exception aborting context", ex);
             }
-            connection = null;
             events = null;
-            clearCache();
         }
     }
\ No newline at end of file
