diff --git a/old_product.java b/new_product.java
index d7d1c56..df5f2cd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -113,8 +113,8 @@ FindInfo findEntriesSince(FindToken token, long maxTotalSizeOfEntries) throws St
         newToken.setBytesRead(totalBytesRead);
         return new FindInfo(messageEntries, newToken);
       }
-    } catch (IOException e) {
-      throw new StoreException("IOError when finding entries for index " + dataDir, e, StoreErrorCodes.IOError);
+    } catch (StoreException e) {
+      throw e;
     } catch (Exception e) {
       throw new StoreException("Unknown error when finding entries for index " + dataDir, e,
           StoreErrorCodes.Unknown_Error);
