diff --git a/old_test.java b/new_test.java
index 86a9cfc..006b3d1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,4 +6,10 @@
       keysPresent.add(info.getStoreKey());
     }
     assertEquals("All keys were not present in the return from findEntriesSince()", allKeys.keySet(), keysPresent);
+
+    // Extra Test: findEntriesSince method can correctly capture disk related IO error and shutdown store if needed.
+    store.shutdown();
+    catchStoreExceptionAndVerifyErrorCode(
+        (blobStore) -> blobStore.findEntriesSince(new StoreFindToken(), Long.MAX_VALUE));
+    reloadStore();
   }
\ No newline at end of file
