diff --git a/old_product.java b/new_product.java
index 9ffb4aa..603c109 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
 @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<PartitionProperties> getPartitionProperties(String partitionId) {
-        return connection.getManagementNode().flatMap(node -> node.getPartitionProperties(partitionId));
+        return connectionProcessor.flatMap(connection -> connection.getManagementNode())
+            .flatMap(node -> node.getPartitionProperties(partitionId));
     }
\ No newline at end of file
