diff --git a/old_product.java b/new_product.java
index b4f4a7b..88397f6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,7 +10,8 @@
         if (!this.clusterRepo.exists(id)) {
             throw new GenieNotFoundException("No cluster exists with the given id. Unable to update.");
         }
-        if (!id.equals(updateCluster.getId())) {
+        final Optional<String> updateId = updateCluster.getId();
+        if (updateId.isPresent() && !id.equals(updateId.get())) {
             throw new GenieBadRequestException("Cluster id inconsistent with id passed in.");
         }
 
