diff --git a/old_test.java b/new_test.java
index 488d510..5a41242 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,4 +17,9 @@
         Err.BXXQ_UPDATING);
     query(_XQUERY_EVAL.args("declare %updating function local:x() {()}; 1"));
     query(_XQUERY_EVAL.args("\"" + DOC.args(PATH).replace('"', '\'') + "\""));
+    // check updating expressions
+    error(_XQUERY_EVAL.args("\"file:exists('x')\"", " map{}", " map{ 'permission':'none'}"),
+        Err.BXXQ_PERM);
+    error(_XQUERY_EVAL.args("\"(1 to 10000000000000)[.=0]\"", " map{}", " map{ 'timeout':'1'}"),
+        Err.BXXQ_STOPPED);
   }
\ No newline at end of file
