diff --git a/old_test.java b/new_test.java
index fc1f683..e63d95b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,5 +3,5 @@
       throws IOException, JSONException, RestServiceException, URISyntaxException {
     AdminBlobStorageService adminBlobStorageService = getAdminBlobStorageService();
     MessageInfo messageInfo = createMessageInfo(RestMethod.POST, "/", new JSONObject());
-    adminBlobStorageService.handleMessage(messageInfo);
+    adminBlobStorageService.handlePost(messageInfo);
   }
\ No newline at end of file
