diff --git a/old_test.java b/new_test.java
index 9122926..5e2da4c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testOnCreateOrUpdateJob() throws GenieException {
+    public void testOnCreateOrUpdateJob() throws GeniePreconditionException {
         Assert.assertNull(this.job.getId());
         Assert.assertNull(this.job.getClusterCriteriasString());
         Assert.assertNull(this.job.getCommandCriteriaString());
