diff --git a/old_product.java b/new_product.java
index 4279a13..d567d17 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,6 +12,6 @@ public static void mergeHadoopConfiguration(org.apache.hadoop.conf.Configuration
     LOG.info("Loading Alluxio properties from Hadoop configuration: {}", alluxioConfProperties);
     // Merge the relevant Hadoop configuration into Alluxio's configuration.
     // TODO(jiri): support multiple client configurations (ALLUXIO-2034)
-    Configuration.merge(alluxioConfProperties, Configuration.Source.HADOOP_CONF);
+    Configuration.merge(alluxioConfProperties, Source.RUNTIME);
     Configuration.validate();
   }
\ No newline at end of file
