diff --git a/old_test.java b/new_test.java
index e0c7703..4644d99 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,10 +7,9 @@
 
     // This hadoop config will not be loaded into Alluxio configuration.
     hadoopConfig.set("hadoop.config.parameter", "hadoop config value");
-    long beforeSize = Configuration.toMap().size();
     HadoopConfigurationUtils.mergeHadoopConfiguration(hadoopConfig);
-    long afterSize = Configuration.toMap().size();
-    Assert.assertEquals(beforeSize + 2, afterSize);
     Assert.assertEquals(TEST_S3_ACCCES_KEY, Configuration.get(PropertyKey.S3A_ACCESS_KEY));
     Assert.assertEquals(TEST_S3_SECRET_KEY, Configuration.get(PropertyKey.S3A_SECRET_KEY));
+    Assert.assertEquals(Source.RUNTIME, Configuration.getSource(PropertyKey.S3A_ACCESS_KEY));
+    Assert.assertEquals(Source.RUNTIME, Configuration.getSource(PropertyKey.S3A_SECRET_KEY));
   }
\ No newline at end of file
