diff --git a/old_product.java b/new_product.java
index 9035a0a..390d4db 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,25 @@
-public boolean isObserverInterested(Annotation... bindings)
+public boolean isObserverInterested(MetaDataCache mdc, Annotation... bindings)
    {
       // Simply check that all event bindings specified by the observer are
       // in the list provided.
-      List<Annotation> bindingsArray = Arrays.asList(bindings);
-      boolean result = true;
-      if (!this.eventBindings.isEmpty())
+      if (this.eventBindings.isEmpty())
       {
-         result = bindingsArray.containsAll(this.eventBindings);
+         return true;
+      }
+      else
+      {
+	     //List<Annotation> bindingsArray = Arrays.asList(bindings);
+         //return bindingsArray.containsAll(this.eventBindings);
+         for (Annotation x: eventBindings) {
+            boolean found = false;
+            for (Annotation y: bindings)
+            {
+        	if ( mdc.getBindingTypeModel(x.annotationType()).isEqual(x, y) ) {
+               found = true;
+            }
+            }
+            if (!found) return false;
+         }
+         return true;
       }
-      return result;
    }
\ No newline at end of file
