diff --git a/old_test.java b/new_test.java
index 75419d0..3a95424 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,14 +5,14 @@
       Observer<DangerCall> observer = new AnObserver<DangerCall>();
       EventObserver<DangerCall> wrappedObserver = new EventObserver<DangerCall>(observer, DangerCall.class, new TameAnnotationLiteral());
       assert wrappedObserver.getEventBindings().size() == 1;
-      assert wrappedObserver.isObserverInterested(new TameAnnotationLiteral());
-      assert !wrappedObserver.isObserverInterested(new AnimalStereotypeAnnotationLiteral());
-      assert !wrappedObserver.isObserverInterested();
-      assert wrappedObserver.isObserverInterested(new TameAnnotationLiteral(), new RoleBinding("Admin"));
+      assert wrappedObserver.isObserverInterested(new MetaDataCache(), new TameAnnotationLiteral());
+      assert !wrappedObserver.isObserverInterested(new MetaDataCache(), new AnimalStereotypeAnnotationLiteral());
+      assert !wrappedObserver.isObserverInterested(new MetaDataCache());
+      assert wrappedObserver.isObserverInterested(new MetaDataCache(), new TameAnnotationLiteral(), new RoleBinding("Admin"));
       
       // Perform some tests with binding values (7.7.1)
       wrappedObserver = new EventObserver<DangerCall>(observer, DangerCall.class, new RoleBinding("Admin"));
       assert wrappedObserver.getEventBindings().size() == 1;
-      assert wrappedObserver.isObserverInterested(new RoleBinding("Admin"));
-      assert !wrappedObserver.isObserverInterested(new RoleBinding("User"));
+      assert wrappedObserver.isObserverInterested(new MetaDataCache(), new RoleBinding("Admin"));
+      assert !wrappedObserver.isObserverInterested(new MetaDataCache(), new RoleBinding("User"));
    }
\ No newline at end of file
