diff --git a/old_product.java b/new_product.java
index 3e591b4..c9fd102 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,5 @@ public JapaneseDate plusDays(long days) {
         if (days == 0) {
             return this;
         }
-        return JapaneseDate.japaneseDate(date.plusDays(days));
+        return JapaneseDate.from(date.plusDays(days));
     }
\ No newline at end of file
