diff --git a/old_test.java b/new_test.java
index b559e69..b6d0c42 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
     public void testPlusDays() {
-        assertEquals(testDate.plusDays(2), MinguoDate.minguoDate(testYear, testMonthOfYear, testDayOfMonth+2));
+        assertEquals(testDate.plusDays(2), MinguoDate.of(testYear, testMonthOfYear, testDayOfMonth+2));
     }
\ No newline at end of file
