diff --git a/old_product.java b/new_product.java
index b84ace6..72eafe3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,14 +9,14 @@
         }
 
         LineAndShapeRenderer that = (LineAndShapeRenderer) obj;
-        if (this.baseLinesVisible != that.baseLinesVisible) {
+        if (this.defaultLinesVisible != that.defaultLinesVisible) {
             return false;
         }
         if (!ObjectUtils.equal(this.seriesLinesVisible,
                 that.seriesLinesVisible)) {
             return false;
         }
-        if (this.baseShapesVisible != that.baseShapesVisible) {
+        if (this.defaultShapesVisible != that.defaultShapesVisible) {
             return false;
         }
         if (!ObjectUtils.equal(this.seriesShapesVisible,
@@ -27,7 +27,7 @@
                 that.seriesShapesFilled)) {
             return false;
         }
-        if (this.baseShapesFilled != that.baseShapesFilled) {
+        if (this.defaultShapesFilled != that.defaultShapesFilled) {
             return false;
         }
         if (this.useOutlinePaint != that.useOutlinePaint) {
