diff --git a/old_test.java b/new_test.java
index e9ffb2e..11e689e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
   public void createFileTest() throws Exception {
     Mockito.doNothing().when(mFileSystemMasterClient)
-        .createFile(Mockito.any(TachyonURI.class), Mockito.any(CreateFileOptions.class));
-    TachyonURI file = new TachyonURI("/file");
+        .createFile(Mockito.any(AlluxioURI.class), Mockito.any(CreateFileOptions.class));
+    AlluxioURI file = new AlluxioURI("/file");
     CreateFileOptions options = CreateFileOptions.defaults();
     FileOutStream out = mFileSystem.createFile(file, options);
     Mockito.verify(mFileSystemMasterClient).createFile(file, options);
