diff --git a/old_test.java b/new_test.java
index 9013bbb..28853dd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,5 +9,5 @@
 		order.setEncounter(encounterService.getEncounter(6));
 		order.setStartDate(new Date());
 		order = orderService.saveOrder(order, null);
-		assertTrue(order.getOrderType().getOrderTypeId() == 1);
+		assertEquals(2, order.getOrderType().getOrderTypeId().intValue());
 	}
\ No newline at end of file
