diff --git a/old_product.java b/new_product.java
index 13c90ce..ce2d90b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,11 +5,21 @@ public Order saveOrder(Order order, OrderContext orderContext) throws APIExcepti
 		
 		if (order.getOrderType() == null) {
 			OrderType orderType = getOrderTypeByConcept(order.getConcept());
+			if (orderType == null && orderContext != null) {
+				orderType = orderContext.getOrderType();
+			}
 			if (orderType == null) {
 				throw new APIException("No order type matches the concept class");
 			}
 			order.setOrderType(orderType);
 		}
+		if (order.getCareSetting() == null) {
+			if (orderContext != null && orderContext.getCareSetting() != null) {
+				order.setCareSetting(orderContext.getCareSetting());
+			} else {
+				throw new APIException("CareSetting is required for an order");
+			}
+		}
 		
 		if (Order.Action.REVISE.equals(order.getAction())) {
 			Order previousOrder = order.getPreviousOrder();
