diff --git a/old_test.java b/new_test.java
index 5b93ce6..ce81003 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,7 @@
     Method addNewConfigurationsFromXml = AbstractUpgradeCatalog.class.getDeclaredMethod
         ("addNewConfigurationsFromXml");
     Method setSecurityType = UpgradeCatalog200.class.getDeclaredMethod("setSecurityType");
+    Method updateTezConfiguration = UpgradeCatalog200.class.getDeclaredMethod("updateTezConfiguration");
     Method updateConfigurationProperties = AbstractUpgradeCatalog.class.getDeclaredMethod
             ("updateConfigurationProperties", String.class, Map.class, boolean.class, boolean.class);
 
@@ -13,6 +14,7 @@
         .addMockedMethod(updateHiveDatabaseType)
         .addMockedMethod(addNewConfigurationsFromXml)
         .addMockedMethod(setSecurityType)
+        .addMockedMethod(updateTezConfiguration)
         .addMockedMethod(updateConfigurationProperties)
         .createMock();
 
@@ -26,6 +28,9 @@
     upgradeCatalog.setSecurityType();
     expectLastCall().once();
 
+    upgradeCatalog.updateTezConfiguration();
+    expectLastCall().once();
+
     upgradeCatalog.updateConfigurationProperties("hive-site",
             Collections.singletonMap("hive.server2.transport.mode", "binary"), false, false);
     expectLastCall();
