diff --git a/old_product.java b/new_product.java
index 6d51f2d..ba65492 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1,2 @@
-public List<User> getUsers(String nameSearch, List<Role> roles, boolean includeVoided) throws DAOException;
\ No newline at end of file
+public List<User> getUsers(String name, List<Role> roles, boolean includeRetired, Integer start, Integer length)
+	                                                                                                                throws DAOException;
\ No newline at end of file
