diff --git a/old_test.java b/new_test.java
index 5f79f12..83501f9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,7 +13,7 @@
     out.write(BufferUtils.getIncreasingByteArray(blockSize));
     out.close();
 
-    String localPath = mWorkerServiceHandler.lockBlock(blockId, SESSION_ID);
+    String localPath = mWorkerServiceHandler.lockBlock(blockId, SESSION_ID).blockPath;
 
     // The local path should exist
     Assert.assertNotNull(localPath);
