diff --git a/old_product.java b/new_product.java
index 8076d19..691a083 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
-public synchronized String lockBlock(final long blockId) throws IOException {
+public synchronized LockBlockResult lockBlock(final long blockId) throws IOException {
     // TODO(jiri) Would be nice to have a helper method to execute this try-catch logic
     try {
-      return retryRPC(new RpcCallableThrowsTachyonTException<String>() {
+      return retryRPC(new RpcCallableThrowsTachyonTException<LockBlockResult>() {
         @Override
-        public String call() throws TachyonTException, TException {
+        public LockBlockResult call() throws TachyonTException, TException {
           return mClient.lockBlock(blockId, mSessionId);
         }
       });
