diff --git a/old_product.java b/new_product.java
index a63d159..ccec200 100644
--- a/old_product.java
+++ b/new_product.java
@@ -22,6 +22,16 @@
 				}
 			}
 		}
-		
+		// Makes sure that the end dates of most recent states in each workflow
+		// and the program end date are consistent
+		if (patientProgram.getDateCompleted() != null) {
+			for (PatientState state : patientProgram.getMostRecentStateInEachWorkflow()) {
+				// The EndDate of active states only should be updated
+				if (state.getEndDate() == null) {
+					state.setEndDate(patientProgram.getDateCompleted());
+				}
+			}
+		}
+
 		return dao.savePatientProgram(patientProgram);
 	}
\ No newline at end of file
