diff --git a/old_test.java b/new_test.java
index 2285c63..c4f6956 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,11 @@
 @Test
-    public void testConceptualSearch() {
-        Map<String,Object> params = new HashMap<String, Object>();
-        params.put(ConceptInsights.ACCOUNT_ID, "public");
-        params.put(ConceptInsights.CORPUS, "ibmresearcher");
-        List<String> ids = new ArrayList<String>();
-        ids.add("/graphs/wikipedia/en-20120601/concepts/IBM_Watson");
-        params.put(ConceptInsights.IDS, ids);
-        params.put(ConceptInsights.LIMIT, 10);
-        params.put(ConceptInsights.CURSOR, 0);
-        QueryConcepts cp =  service.conceptualSearch(params);
-        Assert.assertNotNull(cp);
-    }
\ No newline at end of file
+	public void testConceptualSearch() {
+		Map<String, Object> params = new HashMap<String, Object>();
+		List<String> ids = new ArrayList<String>();
+		ids.add(EXAMPLE_CONCEPT.getId());
+		params.put(ConceptInsights.IDS, ids);
+		params.put(ConceptInsights.LIMIT, 10);
+		params.put(ConceptInsights.CURSOR, 0);
+		QueryConcepts cp = service.conceptualSearch(Corpus.TED_TALKS, params);
+		Assert.assertNotNull(cp);
+	}
\ No newline at end of file
