diff --git a/old_test.java b/new_test.java
index b44d88b..5005484 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,10 @@
-@Test(expected = IllegalArgumentException.class)
+@Test
 	public void setTo_mismatch() {
 		DummyImage a = new DummyImage(10, 20);
-		DummyImage b = new DummyImage(11, 20);
+		DummyImage b = new DummyImage(11, 21);
 
 		a.setTo(b);
+
+		assertEquals(a.width, 11);
+		assertEquals(b.height, 21);
 	}
\ No newline at end of file
