diff --git a/old_product.java b/new_product.java
index 14a5450..af6bd36 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@
 	@Override
 	public void setTo(T orig) {
 		if (orig.width != width || orig.height != height)
-			throw new IllegalArgumentException("The width and/or height of 'orig' is not the same as this class");
+			reshape(orig.width,orig.height);
 		if (orig.numBands != numBands)
 			throw new IllegalArgumentException("The two images have different number of bands");
 
