diff --git a/old_test.java b/new_test.java
index 7b4969a..527c664 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,7 +10,7 @@
 		testOrder.setCareSetting(orderService.getCareSetting(1));
 		Encounter encounter = encounterService.getEncounter(3);
 		testOrder.setEncounter(encounter);
-		testOrder.setStartDate(encounter.getEncounterDatetime());
+		testOrder.setDateActivated(encounter.getEncounterDatetime());
 		testOrder.setClinicalHistory("Patient had a negative reaction to the test in the past");
 		testOrder.setFrequency(orderService.getOrderFrequency(3));
 		testOrder.setSpecimenSource(conceptService.getConcept(22));
