diff --git a/old_product.java b/new_product.java
index 57dbf6c..9b616cf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,8 +2,8 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 		if (order.getOrderId() != null) {
 			throw new APIException("Cannot edit an existing order, you need to revise it instead");
 		}
-		if (order.getStartDate() == null) {
-			order.setStartDate(new Date());
+		if (order.getDateActivated() == null) {
+			order.setDateActivated(new Date());
 		}
 		//Reject if there is an active order for the same orderable
 		boolean isDrugOrder = DrugOrder.class.isAssignableFrom(getActualType(order));
@@ -73,7 +73,7 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 			if (previousOrder == null) {
 				throw new APIException("Previous Order is required for a revised order");
 			}
-			stopOrder(previousOrder, order.getStartDate());
+			stopOrder(previousOrder, order.getDateActivated());
 		} else if (DISCONTINUE == order.getAction()) {
 			discontinueExistingOrdersIfNecessary(order);
 		}
