diff --git a/old_product.java b/new_product.java
index de594e2..b13cc20 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,24 +12,24 @@
                 that.itemLabelGeneratorMap)) {
             return false;
         }
-        if (!ObjectUtils.equal(this.baseItemLabelGenerator,
-                that.baseItemLabelGenerator)) {
+        if (!ObjectUtils.equal(this.defaultItemLabelGenerator,
+                that.defaultItemLabelGenerator)) {
             return false;
         }
         if (!ObjectUtils.equal(this.toolTipGeneratorMap,
                 that.toolTipGeneratorMap)) {
             return false;
         }
-        if (!ObjectUtils.equal(this.baseToolTipGenerator,
-                that.baseToolTipGenerator)) {
+        if (!ObjectUtils.equal(this.defaultToolTipGenerator,
+                that.defaultToolTipGenerator)) {
             return false;
         }
         if (!ObjectUtils.equal(this.itemURLGeneratorMap,
                 that.itemURLGeneratorMap)) {
             return false;
         }
-        if (!ObjectUtils.equal(this.baseItemURLGenerator,
-                that.baseItemURLGenerator)) {
+        if (!ObjectUtils.equal(this.defaultItemURLGenerator,
+                that.defaultItemURLGenerator)) {
             return false;
         }
         if (!ObjectUtils.equal(this.legendItemLabelGenerator,
