diff --git a/old_product.java b/new_product.java
index bdab9db..a19eddd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery notIn(String propName, Collection<? extends Object> value) {
-        addCriterion(criterionBuilder.notIn(propName, value));
+        criterion = criterion.and(criterionBuilder.notIn(propName, value));
         return this;
     }
\ No newline at end of file
