diff --git a/old_test.java b/new_test.java
index 3bc4fb0..749fcde 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,5 @@
 @Test
-    public void testNotIn_Array() {
+    public void testNotInArray() {
         Object[] criterions = new Object[] {"a", "b"};
-        assertTrue(instance.notIn("name", criterions).getQueryCriterions()
-                .contains(new NotInCriterion("name", criterions)));
+    	assertEquals(new NotInCriterion("name", criterions), instance.notIn("name", criterions).getQueryCriterion());
     }
\ No newline at end of file
