diff --git a/old_product.java b/new_product.java
index b4d13c4..f5684bc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,20 +4,16 @@
             byte[] plaintext;
             if (useExplicitIv) {
                 decryptIv = new IvParameterSpec(Arrays.copyOf(data, decryptCipher.getBlockSize()));
-            }
-            if (tlsContext.getConfig().getConnectionEndType() == ConnectionEndType.CLIENT) {
-                decryptCipher.init(Cipher.DECRYPT_MODE, new SecretKeySpec(serverWriteKey, bulkCipherAlg.getJavaName()),
-                        decryptIv);
-            } else {
-                decryptCipher.init(Cipher.DECRYPT_MODE, new SecretKeySpec(clientWriteKey, bulkCipherAlg.getJavaName()),
-                        decryptIv);
-            }
-            if (useExplicitIv) {
+                if (tlsContext.getConfig().getConnectionEndType() == ConnectionEndType.CLIENT) {
+                    decryptCipher.init(Cipher.DECRYPT_MODE,
+                            new SecretKeySpec(serverWriteKey, bulkCipherAlg.getJavaName()), decryptIv);
+                } else {
+                    decryptCipher.init(Cipher.DECRYPT_MODE,
+                            new SecretKeySpec(clientWriteKey, bulkCipherAlg.getJavaName()), decryptIv);
+                }
                 plaintext = decryptCipher.doFinal(Arrays.copyOfRange(data, decryptCipher.getBlockSize(), data.length));
             } else {
-                plaintext = decryptCipher.doFinal(data);
-                decryptIv = new IvParameterSpec(Arrays.copyOfRange(data, data.length - decryptCipher.getBlockSize(),
-                        data.length));
+                plaintext = decryptCipher.update(data);
             }
             return plaintext;
         } catch (BadPaddingException | IllegalBlockSizeException | InvalidAlgorithmParameterException
