diff --git a/old_test.java b/new_test.java
index c1bc5d8..b0c858e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,3 +1,20 @@
 @Test
-    public void testDecrypt() {
+    public void testDecrypt10() {
+        context.getConfig().setConnectionEndType(ConnectionEndType.CLIENT);
+        context.setSelectedCipherSuite(CipherSuite.TLS_RSA_WITH_3DES_EDE_CBC_SHA);
+        context.setSelectedProtocolVersion(ProtocolVersion.TLS10);
+        context.setClientRandom(ArrayConverter
+                .hexStringToByteArray("03c08c3460b420bb3851d9d47acb933dbe70399bf6c92da33af01d4fb770e98c"));
+        context.setServerRandom(ArrayConverter
+                .hexStringToByteArray("78f0c84e04d3c23cad94aad61ccae23ce79bcd9d2d6953f8ccbe0e528c63a238"));
+        context.setMasterSecret(ArrayConverter
+                .hexStringToByteArray("F81015161244782B3541E6020140556E4FFEA98C57FCF6CEC172CD8B577DC73CCDE4B724E07DB8687DDF327CD8A68891"));
+        byte[] data = ArrayConverter
+                .hexStringToByteArray("BCD644DF7E82BF0097E1B0C16CDD53199733EE70629FA82DAC7B0B4F6100B602ACBA3B8EA6A7741B");
+
+        cipher = new RecordBlockCipher(context);
+        byte[] plaintext = cipher.decrypt(data);
+        byte[] correctPlaintext = ArrayConverter
+                .hexStringToByteArray("1400000CC84350158844FE559EC327B77F44B9791ECB11453B7FC40ED27C35DDDC7C250603030303");
+        assertArrayEquals(plaintext, correctPlaintext);
     }
\ No newline at end of file
