diff --git a/old_test.java b/new_test.java
index 9665f86..fed6800 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,27 +5,57 @@
 		InputStream inStream = this.getClass().getResourceAsStream("/BondFeature.gb");
 		assertNotNull(inStream);
 		
-		GenbankReader<ProteinSequence,AminoAcidCompound> GenbankProtein = 
+		GenbankReader<ProteinSequence,AminoAcidCompound> genbankProtein = 
 				new GenbankReader<ProteinSequence,AminoAcidCompound>(
 						inStream, 
 						new GenericGenbankHeaderParser<ProteinSequence,AminoAcidCompound>(), 
 						new ProteinSequenceCreator(AminoAcidCompoundSet.getAminoAcidCompoundSet())
 						);
 		@SuppressWarnings("unused")
-		LinkedHashMap<String,ProteinSequence> proteinSequences = GenbankProtein.process();
+		LinkedHashMap<String,ProteinSequence> proteinSequences = genbankProtein.process();
 		inStream.close();
 
 		logger.info("process DNA");
 		inStream = this.getClass().getResourceAsStream("/NM_000266.gb");
 		assertNotNull(inStream);
 
-		GenbankReader<DNASequence,NucleotideCompound> GenbankDNA = 
+		GenbankReader<DNASequence,NucleotideCompound> genbankDNA = 
 				new GenbankReader<DNASequence,NucleotideCompound>(
 						inStream,
 						new GenericGenbankHeaderParser<DNASequence,NucleotideCompound>(), 
 						new DNASequenceCreator(DNACompoundSet.getDNACompoundSet())
 						);
 		@SuppressWarnings("unused")
-		LinkedHashMap<String,DNASequence> dnaSequences = GenbankDNA.process();
+		LinkedHashMap<String,DNASequence> oneDNASequence = genbankDNA.process();
 		inStream.close();
+
+		logger.info("process 2 DNAs");
+		inStream = this.getClass().getResourceAsStream("/two-dnaseqs.gb");
+		assertNotNull(inStream);
+
+		GenbankReader<DNASequence,NucleotideCompound> readTwoDNAs = 
+				new GenbankReader<DNASequence,NucleotideCompound>(
+						inStream,
+						new GenericGenbankHeaderParser<DNASequence,NucleotideCompound>(), 
+						new DNASequenceCreator(DNACompoundSet.getDNACompoundSet())
+						);
+		@SuppressWarnings("unused")
+		LinkedHashMap<String,DNASequence> twoDNAs = readTwoDNAs.process();
+		inStream.close();
+
+		logger.info("process coli genome");
+		inStream = this.getClass().getResourceAsStream("/NC_000913.gb");
+		assertNotNull(inStream);
+
+		
+		GenbankReader<DNASequence,NucleotideCompound> readColiGenome = 
+				new GenbankReader<DNASequence,NucleotideCompound>(
+						inStream,
+						new GenericGenbankHeaderParser<DNASequence,NucleotideCompound>(), 
+						new DNASequenceCreator(DNACompoundSet.getDNACompoundSet())
+						);
+		@SuppressWarnings("unused")
+		LinkedHashMap<String,DNASequence> coliGenome = readColiGenome.process();
+		assertNotNull(inStream);
+		inStream.close();	
 	}
\ No newline at end of file
