diff --git a/old_test.java b/new_test.java
index 1655d06..33c19c8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,4 +5,8 @@
     Assert.assertEquals(options.getReadType().getAlluxioStorageType(),
         inStreamOptions.getAlluxioStorageType());
     Assert.assertEquals(options.getLocationPolicy(), inStreamOptions.getLocationPolicy());
+    Assert.assertEquals(options.getUfsReadLocationPolicy(),
+        inStreamOptions.getUfsReadLocationPolicy());
+    Assert.assertEquals(options.getMaxUfsReadConcurrency(),
+        inStreamOptions.getMaxUfsReadConcurrency());
   }
\ No newline at end of file
