diff --git a/old_test.java b/new_test.java
index 0083965..9f33a6b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -25,7 +25,7 @@
     mockServer.when(request().withPath(GET_VOICES_PATH)).respond(
         response().withHeaders(
             new Header(HttpHeaders.Names.CONTENT_TYPE, HttpMediaType.APPLICATION_JSON)).withBody(
-            GsonSingleton.getGson().toJson(response)));
+            GsonSingleton.getGsonWithoutPrettyPrinting().toJson(response)));
 
     final List<Voice> result = service.getVoices();
     Assert.assertNotNull(result);
