diff --git a/old_product.java b/new_product.java
index f1211f8..1e60657 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,6 @@ public List<Voice> getVoices() {
     final Response response = execute(request);
     final JsonObject jsonObject = ResponseUtil.getJsonObject(response);
     final List<Voice> voices =
-        GsonSingleton.getGson().fromJson(jsonObject.get("voices"), listVoiceType);
+        GsonSingleton.getGsonWithoutPrettyPrinting().fromJson(jsonObject.get("voices"), listVoiceType);
     return voices;
   }
\ No newline at end of file
