diff --git a/old_test.java b/new_test.java
index 6873b21..9ccba5d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,7 @@
     InternalDistributedSystem system = getSystem(config);
     try {
 
-      DM dm = system.getDistributionManager();
+      DistributionManager dm = system.getDistributionManager();
       DistributedMember member = dm.getDistributionManagerId();
 
       assertEquals(member.getId(), system.getMemberId());
