diff --git a/old_product.java b/new_product.java
index 2784ed0..900f8a0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-CompletableFuture<Boolean> createStream(final StreamConfiguration streamConfig);
\ No newline at end of file
+CompletableFuture<Boolean> createStream(final String scope, final String streamName, final StreamConfiguration streamConfig);
\ No newline at end of file
