diff --git a/old_test.java b/new_test.java
index d7f5b02..64ff042 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,10 +3,10 @@
 		CameraPinholeRadial param = new CameraPinholeRadial().
 						fsetK(300, 320, 0, 150, 130, width, height).fsetRadial(0.1,1e-4);
 
-		DenseMatrix64F rect = new DenseMatrix64F(3,3,true,1.1,0,0,0,2,0,0.1,0,3);
-		DenseMatrix64F rectK = PerspectiveOps.calibrationMatrix(param, (DenseMatrix64F)null);
+		RowMatrix_F64 rect = new RowMatrix_F64(3,3,true,1.1,0,0,0,2,0,0.1,0,3);
+		RowMatrix_F64 rectK = PerspectiveOps.calibrationMatrix(param, (RowMatrix_F64)null);
 
-		DenseMatrix64F rectK_inv = new DenseMatrix64F(3,3);
+		RowMatrix_F64 rectK_inv = new RowMatrix_F64(3,3);
 		CommonOps_D64.invert(rectK,rectK_inv);
 
 		Point2Transform2_F64 tranRect = RectifyImageOps.transformPixelToRect(param, rect);
