diff --git a/old_product.java b/new_product.java
index 192be66..cc9f57e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static Point2Transform2_F64 transformPixelToRectNorm(CameraPinholeRadial param,
-																DenseMatrix64F rectify,
-																DenseMatrix64F rectifyK) {
+																RowMatrix_F64 rectify,
+																RowMatrix_F64 rectifyK) {
 		return ImplRectifyImageOps_F64.transformPixelToRectNorm(param, rectify, rectifyK);
 	}
\ No newline at end of file
