diff --git a/old_test.java b/new_test.java
index b6dfb65..89b8502 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testOnRender_overlaid_drawsBarsInOrder() throws Exception {
+    public void onRender_overlaid_drawsSmallestBarsLast() {
         XYSeries s1 = new SimpleXYSeries(SimpleXYSeries.ArrayFormat.Y_VALS_ONLY, "s1", -1, -2, 1, 2);
         XYSeries s2 = new SimpleXYSeries(SimpleXYSeries.ArrayFormat.Y_VALS_ONLY, "s2", -2, -1, 2, 1);
 
