diff --git a/old_product.java b/new_product.java
index 48a3830..d22c75d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
 protected abstract void onRender(Canvas canvas, RectF plotArea, SeriesType series,
-                                  SeriesFormatterType formatter, RenderStack stack) throws PlotRenderException;
\ No newline at end of file
+                                  SeriesFormatterType formatter, RenderStack stack);
\ No newline at end of file
