diff --git a/old_product.java b/new_product.java
index afb3260..4b7eb3e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static double manhattanWeighhted(AssociativeArray a1, AssociativeArray a2, Map<Object, Double> columnWeights) {
+public static double manhattanWeighted(AssociativeArray a1, AssociativeArray a2, Map<Object, Double> columnWeights) {
         Map<Object, Double> columnDistances = columnDistances(a1, a2, columnWeights.keySet());
         
         double distance = 0.0;
