diff --git a/old_product.java b/new_product.java
index 492a53c..86fbf1c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public ServiceCall<Value> updateValue(UpdateValueOptions updateValueOptions) {
     Validator.notNull(updateValueOptions, "updateValueOptions cannot be null");
     RequestBuilder builder = RequestBuilder.post(String.format("/v1/workspaces/%s/entities/%s/values/%s",
-        updateValueOptions.workspaceId(), updateValueOptions.entity(), updateValueOptions.value()));
+            updateValueOptions.workspaceId(), updateValueOptions.entity(), updateValueOptions.value()));
     builder.query(VERSION, versionDate);
     final JsonObject contentJson = new JsonObject();
     if (updateValueOptions.newSynonyms() != null) {
@@ -13,6 +13,9 @@ public ServiceCall<Value> updateValue(UpdateValueOptions updateValueOptions) {
     if (updateValueOptions.newValue() != null) {
       contentJson.addProperty("value", updateValueOptions.newValue());
     }
+    if (updateValueOptions.newPatterns() != null) {
+      contentJson.add("patterns", GsonSingleton.getGson().toJsonTree(updateValueOptions.newPatterns()));
+    }
     builder.bodyJson(contentJson);
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(Value.class));
   }
\ No newline at end of file
