diff --git a/old_test.java b/new_test.java
index 4bd69e5..0f35125 100644
--- a/old_test.java
+++ b/new_test.java
@@ -31,19 +31,19 @@
     Date start = new Date();
 
     UpdateValueOptions updateOptions = new UpdateValueOptions.Builder()
-        .workspaceId(workspaceId)
-        .entity(entity)
-        .value(entityValue1)
-        .newValue(entityValue2)
-        .newSynonyms(new ArrayList<String>(Arrays.asList(synonym1, synonym2)))
-        .newMetadata(valueMetadata)
-        .build();
+            .workspaceId(workspaceId)
+            .entity(entity)
+            .value(entityValue1)
+            .newValue(entityValue2)
+            .newSynonyms(new ArrayList<String>(Arrays.asList(synonym1, synonym2)))
+            .newMetadata(valueMetadata)
+            .build();
     Value response = service.updateValue(updateOptions).execute();
 
     try {
       assertNotNull(response);
-      assertNotNull(response.getValue());
-      assertEquals(response.getValue(), entityValue2);
+      assertNotNull(response.getValueText());
+      assertEquals(response.getValueText(), entityValue2);
       assertNotNull(response.getCreated());
       assertNotNull(response.getUpdated());
 
@@ -54,14 +54,14 @@
       assertTrue(fuzzyAfter(response.getUpdated(), start));
 
       GetValueOptions getOptions =
-          new GetValueOptions.Builder(workspaceId, entity, entityValue2)
-              .export(true)
-              .build();
+              new GetValueOptions.Builder(workspaceId, entity, entityValue2)
+                      .export(true)
+                      .build();
       ValueExport vResponse = service.getValue(getOptions).execute();
 
       assertNotNull(vResponse);
-      assertNotNull(vResponse.getValue());
-      assertEquals(vResponse.getValue(), entityValue2);
+      assertNotNull(vResponse.getValueText());
+      assertEquals(vResponse.getValueText(), entityValue2);
       assertNotNull(vResponse.getCreated());
       assertNotNull(vResponse.getUpdated());
 
@@ -83,7 +83,7 @@
     } finally {
       // Clean up
       DeleteValueOptions deleteOptions =
-          new DeleteValueOptions.Builder(workspaceId, entity, entityValue2).build();
+              new DeleteValueOptions.Builder(workspaceId, entity, entityValue2).build();
       service.deleteValue(deleteOptions).execute();
     }
   }
\ No newline at end of file
