diff --git a/old_product.java b/new_product.java
index ac1f509..6cd68ad 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@
 	public Collection<Provider> getProvidersByPerson(Person person, boolean includeRetired) {
 		Criteria criteria = getSession().createCriteria(Provider.class);
 		if (!includeRetired) {
-			criteria.add(Expression.eq("retired", false));
+			criteria.add(Restrictions.eq("retired", false));
 		} else {
 			//push retired Provider to the end of the returned list
 			criteria.addOrder(Order.asc("retired"));
