diff --git a/old_test.java b/new_test.java
index 13440e4..7b44964 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,16 +1,20 @@
 @Test
-    public void testSpatialJoinQueryUsingIndex() throws Exception {
+    public void testSpatialJoinQueryWithPolygonRDDUsingQuadTreeIndex() throws Exception {
+        sc = new JavaSparkContext(conf);
+    	
+        PolygonRDD queryRDD = new PolygonRDD(sc, InputLocationQueryPolygon, offset, splitter, numPartitions);
 
-    	PolygonRDD polygonRDD = new PolygonRDD(sc, InputLocation, offset, splitter, numPartitions);
-
-    	PolygonRDD objectRDD = new PolygonRDD(sc, InputLocation, offset, splitter, gridType, numPartitions);
-
-    	objectRDD.buildIndex(IndexType.RTREE);
-
-        JoinQuery joinQuery = new JoinQuery(sc,objectRDD,polygonRDD);
+        PolygonRDD spatialRDD = new PolygonRDD(sc, InputLocation, offset, splitter, numPartitions);
+        
+        spatialRDD.spatialPartitioning(gridType);
+        
+        spatialRDD.buildIndex(IndexType.QUADTREE, true);
+        
+        queryRDD.spatialPartitioning(spatialRDD.grids);
+        
+        List<Tuple2<Polygon, HashSet<Polygon>>> result = JoinQuery.SpatialJoinQuery(spatialRDD,queryRDD,false).collect();
         
-        List<Tuple2<Polygon, HashSet<Polygon>>> result = joinQuery.SpatialJoinQueryUsingIndex(objectRDD,polygonRDD).collect();
-        assert result.get(0)._1().getUserData()!=null;
+        assert result.get(1)._1().getUserData()!=null;
         for(int i=0;i<result.size();i++)
         {
         	if(result.get(i)._2().size()!=0)
@@ -18,4 +22,5 @@
         		assert result.get(i)._2().iterator().next().getUserData()!=null;
         	}
         }
+        sc.close();
     }
\ No newline at end of file
