diff --git a/old_product.java b/new_product.java
index 2293710..7f7bce4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static CatalogResolver catalogResolver(CatalogFeatures features, String... paths) {
-        Catalog catalog = catalog(features, paths);
+public static CatalogResolver catalogResolver(CatalogFeatures features, URI... uris) {
+        Catalog catalog = catalog(features, uris);
         return new CatalogResolverImpl(catalog);
     }
\ No newline at end of file
