diff --git a/old_product.java b/new_product.java
index 35eca19..f23fcdc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,17 +6,16 @@
 			throw new InsufficientIdentifiersException("At least one nonvoided Patient Identifier is required");
 		}
 		
-		List<PatientIdentifier> identifiers = new Vector<PatientIdentifier>();
-		identifiers.addAll(patient.getIdentifiers());
-		List<String> identifiersUsed = new Vector<String>();
-		List<PatientIdentifierType> requiredTypes = Context.getPatientService().getPatientIdentifierTypes(null, null, true,
-		    null);
+		List<PatientIdentifier> patientIdentifiers = new ArrayList<>();
+		patientIdentifiers.addAll(patient.getIdentifiers());
+		List<PatientIdentifierType> requiredTypes = this.getPatientIdentifierTypes(null, null, true, null);
 		if (requiredTypes == null) {
 			requiredTypes = new ArrayList<PatientIdentifierType>();
 		}
-		List<PatientIdentifierType> foundRequiredTypes = new ArrayList<PatientIdentifierType>();
-		
-		for (PatientIdentifier pi : identifiers) {
+
+		List<String> identifiersUsed = new ArrayList<>();
+
+		for (PatientIdentifier pi : patientIdentifiers) {
 			if (pi.getVoided()) {
 				continue;
 			}
@@ -32,7 +31,6 @@
 			// check if this is a required identifier
 			for (PatientIdentifierType requiredType : requiredTypes) {
 				if (pi.getIdentifierType().equals(requiredType)) {
-					foundRequiredTypes.add(requiredType);
 					requiredTypes.remove(requiredType);
 					break;
 				}
