diff --git a/old_test.java b/new_test.java
index 8434595..ac238a6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,15 +4,15 @@
         System.out.println("root");
         double eps = 1e-13;
         int maxIterations = 1000;
-        double result = Zeroin.root(eps, maxIterations, -PI/2, PI/2, 0, sinF);
+        double result = Zeroin.root(eps, maxIterations, -PI/2, PI/2, sinF);
         assertEquals(0, result, eps);
         
-        result = Zeroin.root(eps, maxIterations, -PI/2, PI/2, 0, sinFp1, 0.0, 1.0);
+        result = Zeroin.root(eps, maxIterations, -PI/2, PI/2, (x)->sin(x+1));
         assertEquals(-1.0, result, eps);
         
-        result = Zeroin.root(eps, maxIterations, -PI/2, PI/2, 1, sinFp1, 3.0, 0.0);
+        result = Zeroin.root(eps, maxIterations, -PI/2, PI/2, (x)->sin(x+3));
         assertEquals(PI-3.0, result, eps);
         
-        result = Zeroin.root(eps, maxIterations, -6, 6, 0, polyF);
+        result = Zeroin.root(eps, maxIterations, -6, 6, polyF);
         assertEquals(-4.8790576334840479813, result, eps);
     }
\ No newline at end of file
