diff --git a/old_test.java b/new_test.java
index 99e2d84..8d4e2d8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void sendHeartbeatRequest()
             throws Exception {
-        NodeView nodeView = client.sendHeartbeatRequest().get();
-        assertThat(nodeView).isNotNull();
+        ClusterState clusterState = client.sendHeartbeatRequest().get();
+        assertThat(clusterState).isNotNull();
     }
\ No newline at end of file
