diff --git a/old_test.java b/new_test.java
index aa7e7b4..39a235b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void execute ()
   {
-    String result = Eclim.execute(new String[]{"-command", "ping"});
+    String result = Eclim.execute(new String[]{"ping"});
     assertEquals("Unexpected result",
         "eclim " + System.getProperty("eclim.version"), result);
   }
\ No newline at end of file
