diff --git a/old_product.java b/new_product.java
index 3d8aeeb..7b3b504 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,11 @@
 public static String execute (String[] _args, long _timeout)
   {
-    String[] args = new String[_args.length + 1];
-    System.arraycopy(_args, 0, args, 1, _args.length);
+    String[] args = new String[_args.length + 2];
+    System.arraycopy(_args, 0, args, 2, _args.length);
     args[0] = ECLIM;
+    args[1] = COMMAND;
+
+    System.out.println("Command: " + StringUtils.join(args, ' '));
 
     CommandExecutor process = null;
     try{
@@ -17,8 +20,8 @@ public static String execute (String[] _args, long _timeout)
     }
 
     if(process.getReturnCode() != 0){
-      System.out.println("ERR: " + process.getErrorMessage());
       System.out.println("OUT: " + process.getResult());
+      System.out.println("ERR: " + process.getErrorMessage());
       throw new RuntimeException("Command failed: " + process.getReturnCode());
     }
 
