diff --git a/old_product.java b/new_product.java
index db26daf..f86ae2c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 @Override
-	public synchronized boolean fireEvent(Eventable eventable) throws ElementNotVisibleException,
-	        NoSuchElementException {
+	public synchronized boolean fireEventAndWait(Eventable eventable)
+	        throws ElementNotVisibleException,
+	        NoSuchElementException, InterruptedException {
 		try {
 
 			boolean handleChanged = false;
