diff --git a/old_test.java b/new_test.java
index fa03cea..c1be96f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,6 @@
 @Test
-	public final void testFireEvent() throws CrawljaxException, MalformedURLException {
+	public final void testFireEvent() throws Exception {
 		browser.goToUrl(new URL(SERVER.getSiteUrl() + "simple.html"));
-		browser.fireEvent(new Eventable(new Identification(How.xpath, "//H1"), EventType.click));
+		browser.fireEventAndWait(new Eventable(new Identification(How.xpath, "//H1"),
+		        EventType.click));
 	}
\ No newline at end of file
