diff --git a/old_product.java b/new_product.java
index 50b63f1..4cc7783 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-<T extends OperationProcessor> long getHighestCommittedSequenceNumber(Iterable<T> processors) {
+<T extends WriterSegmentProcessor> long getHighestCommittedSequenceNumber(Iterable<T> processors) {
         long lowestUncommittedSeqNo = Long.MAX_VALUE;
-        for (OperationProcessor a : processors) {
+        for (WriterSegmentProcessor a : processors) {
             if (!a.isClosed()) {
                 long firstSeqNo = a.getLowestUncommittedSequenceNumber();
                 if (firstSeqNo >= 0) {
