diff --git a/old_product.java b/new_product.java
index f4084f8..05ce213 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public int size() {
         int count = 0;
-        for (Node<E> p = first(); p != null; p = p.getNext()) {
+        for (Node<E> p = first(); p != null; p = succ(p)) {
             if (p.getItem() != null) {
                 // Collections.size() spec says to max out
                 if (++count == Integer.MAX_VALUE)
