diff --git a/old_product.java b/new_product.java
index fccb7d4..41685ba 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery gtProp(String propName, String otherProp) {
-        addCriterion(criterionBuilder.gtProp(propName, otherProp));
+    	criterion = criterion.and(criterionBuilder.gtProp(propName, otherProp));
         return this;
     }
\ No newline at end of file
