diff --git a/old_test.java b/new_test.java
index 1476988..2a7efc0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testGtProp() {
-        assertTrue(instance.gtProp("id", "name").getQueryCriterions()
-                .contains(new GtPropCriterion("id", "name")));
+    	assertEquals(new GtPropCriterion("id", "name"), instance.gtProp("id", "name").getQueryCriterion());
     }
\ No newline at end of file
