diff --git a/old_test.java b/new_test.java
index 63bbf69..b649a2f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testGetSchemaID() 
+    public void testGetSchemaID() throws SQLException
     {
-        assertThat("testGetSchemaID 0",ms.getSchemaID(), equalTo(MetadataSchema.DC_SCHEMA_ID));
+        assertThat("testGetSchemaID 0",ms.getSchemaID(), equalTo(metadataSchemaService.find(context, MetadataSchema.DC_SCHEMA).getSchemaID()));
     }
\ No newline at end of file
