diff --git a/old_test.java b/new_test.java
index e1e666b..36fe653 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,10 @@
 @Test
     public void createTableTest() throws SQLException {
-        try (PreparedStatement pstmt1 = con.prepareStatement("create table x (col1 int)");
-                PreparedStatement pstmt2 = con.prepareStatement("drop table x")) {
+        try (Connection con = DriverManager.getConnection(connectionString);
+                PreparedStatement pstmt1 = con.prepareStatement(
+                        "create table " + AbstractSQLGenerator.escapeIdentifier(tableName) + " (col1 int)");
+                PreparedStatement pstmt2 = con
+                        .prepareStatement("drop table " + AbstractSQLGenerator.escapeIdentifier(tableName))) {
             pstmt1.execute();
             pstmt2.execute();
         } catch (SQLException e) {
