diff --git a/old_product.java b/new_product.java
index 745533b..3208b82 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,24 +1,24 @@
 private void createTable(Connection connection, Statement stmt) throws SQLException {
-        String createQuery = "CREATE TABLE " + tableName + " (" + "[Id] [int] IDENTITY(1,1) NOT NULL,"
-                + "[CompanyName] [nvarchar](40) NOT NULL," + "[ContactName] [nvarchar](50) NULL,"
-                + "[ContactTitle] [nvarchar](40) NULL," + "[City] [nvarchar](40) NULL,"
-                + "[CountryName] [nvarchar](40) NULL,"
+        String createQuery = "CREATE TABLE " + AbstractSQLGenerator.escapeIdentifier(tableName) + " ("
+                + "[Id] [int] IDENTITY(1,1) NOT NULL," + "[CompanyName] [nvarchar](40) NOT NULL,"
+                + "[ContactName] [nvarchar](50) NULL," + "[ContactTitle] [nvarchar](40) NULL,"
+                + "[City] [nvarchar](40) NULL," + "[CountryName] [nvarchar](40) NULL,"
                 + "[Phone] [nvarchar](30) MASKED WITH (FUNCTION = 'default()') NULL,"
                 + "[Fax] [nvarchar](30) MASKED WITH (FUNCTION = 'default()') NULL)";
         stmt.execute(createQuery);
 
-        stmt.execute("ADD SENSITIVITY CLASSIFICATION TO " + tableName
+        stmt.execute("ADD SENSITIVITY CLASSIFICATION TO " + AbstractSQLGenerator.escapeIdentifier(tableName)
                 + ".CompanyName WITH (LABEL='PII', LABEL_ID='L1', INFORMATION_TYPE='Company name', INFORMATION_TYPE_ID='COMPANY')");
-        stmt.execute("ADD SENSITIVITY CLASSIFICATION TO " + tableName
+        stmt.execute("ADD SENSITIVITY CLASSIFICATION TO " + AbstractSQLGenerator.escapeIdentifier(tableName)
                 + ".ContactName WITH (LABEL='PII', LABEL_ID='L1', INFORMATION_TYPE='Person name', INFORMATION_TYPE_ID='NAME')");
-        stmt.execute("ADD SENSITIVITY CLASSIFICATION TO " + tableName
+        stmt.execute("ADD SENSITIVITY CLASSIFICATION TO " + AbstractSQLGenerator.escapeIdentifier(tableName)
                 + ".Phone WITH (LABEL='PII', LABEL_ID='L1', INFORMATION_TYPE='Contact Information', INFORMATION_TYPE_ID='CONTACT')");
-        stmt.execute("ADD SENSITIVITY CLASSIFICATION TO " + tableName
+        stmt.execute("ADD SENSITIVITY CLASSIFICATION TO " + AbstractSQLGenerator.escapeIdentifier(tableName)
                 + ".Fax WITH (LABEL='PII', LABEL_ID='L1', INFORMATION_TYPE='Contact Information', INFORMATION_TYPE_ID='CONTACT')");
 
         // INSERT ROWS OF DATA
-        try (PreparedStatement ps = connection
-                .prepareStatement("INSERT INTO " + tableName + " VALUES (?,?,?,?,?,?,?)")) {
+        try (PreparedStatement ps = connection.prepareStatement(
+                "INSERT INTO " + AbstractSQLGenerator.escapeIdentifier(tableName) + " VALUES (?,?,?,?,?,?,?)")) {
 
             ps.setString(1, "Exotic Liquids");
             ps.setString(2, "Charlotte Cooper");
