diff --git a/old_product.java b/new_product.java
index b9063ec..8a7db33 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-CompletableFuture<Boolean> updateStream(final StreamConfiguration streamConfig);
\ No newline at end of file
+CompletableFuture<Boolean> updateStream(final String scope, final String streamName, final StreamConfiguration streamConfig);
\ No newline at end of file
