diff --git a/old_test.java b/new_test.java
index 60eeb17..fa54260 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
-    public void testGetCombinedFragment() {
+    public void testbuildCombinedFragment() {
         byte[] original = ArrayConverter.hexStringToByteArray("123456789A123456789A");
         collector.addFragment(fragmentOfMsg(0, 0, 3, original));
         collector.addFragment(fragmentOfMsg(0, 3, 5, original));
         collector.addFragment(fragmentOfMsg(0, 8, 2, original));
-        DtlsHandshakeMessageFragment fragment = collector.getCombinedFragment();
+        DtlsHandshakeMessageFragment fragment = collector.buildCombinedFragment();
         checkFragment(fragment, 0, 10, original);
     }
\ No newline at end of file
