diff --git a/old_product.java b/new_product.java
index 4428f81..baa81d4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
-public DtlsHandshakeMessageFragment getCombinedFragment() {
+public DtlsHandshakeMessageFragment buildCombinedFragment() {
         if (!isMessageComplete()) {
             LOGGER.warn("Returning incompletely received message! Missing pieces are replaced by 0 in content.");
         }
-        if (type == null) {
-            throw new WorkflowExecutionException("DtlsFragmentedMessage does not have type!");
+        if (isEmpty()) {
+            throw new WorkflowExecutionException("The FragmentCollector is empty, cannot build combined fragment!");
         }
 
         DtlsHandshakeMessageFragment message = new DtlsHandshakeMessageFragment();
