diff --git a/old_test.java b/new_test.java
index 0ec3572..f4c2f48 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     List<VectorWritable> points = getPointsWritable(REFERENCE);
     
     Path pointsPath = getTestTempDirPath("points");
-    conf = new Configuration();
+    Configuration conf = new Configuration();
     ClusteringTestUtils.writePointsToFile(points, new Path(pointsPath, "file1"), fs, conf);
     ClusteringTestUtils.writePointsToFile(points, new Path(pointsPath, "file2"), fs, conf);
     
