diff --git a/old_product.java b/new_product.java
index acfe381..d39c3be 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public void removeAllCommands() throws GenieException {
         if (this.commands != null) {
-            final List<Command> locCommands = new ArrayList<Command>();
+            final List<Command> locCommands = new ArrayList<>();
             locCommands.addAll(this.commands);
             for (final Command command : locCommands) {
                 this.removeCommand(command);
