diff --git a/old_test.java b/new_test.java
index 7396df0..a401d97 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
         one.setId("one");
         final Command two = new Command();
         two.setId("two");
-        final List<Command> commands = new ArrayList<Command>();
+        final List<Command> commands = new ArrayList<>();
         commands.add(one);
         commands.add(two);
         this.c.setCommands(commands);
