diff --git a/old_test.java b/new_test.java
index 5bfa1db..88d6ffd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testGetProcedureHistogram() throws Exception {
-        Histogram h = WorkloadUtil.getProcedureHistogram(workload_file);
+        Histogram<String> h = WorkloadUtil.getProcedureHistogram(workload_file);
         assertNotNull(h);
         assert(h.getSampleCount() > 0);
         
