diff --git a/old_product.java b/new_product.java
index 219b166..46e070e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@ public static Histogram<String> getProcedureHistogram(File workload_path) throws
                              "\":[\\s]*\"([\\w\\d]+)\"[\\s]*,[\\s]*.*";
         final Pattern p = Pattern.compile(regex);
 
-        if (debug.get()) LOG.debug("Generating Procedure Histogram from Workload '" + workload_path.getAbsolutePath() + "'");
+        if (debug.get()) LOG.debug("Fast generation of Procedure Histogram from Workload '" + workload_path.getAbsolutePath() + "'");
         BufferedReader reader = FileUtil.getReader(workload_path.getAbsolutePath());
         int line_ctr = 0;
         while (reader.ready()) {
@@ -28,6 +28,6 @@ public static Histogram<String> getProcedureHistogram(File workload_path) throws
         } // WHILE
         reader.close();
         
-        if (debug.get()) LOG.debug("Processed " + line_ctr + " workload trace records");
+        if (debug.get()) LOG.debug("Processed " + line_ctr + " workload trace records for histogram\n" + h);
         return (h);
     }
\ No newline at end of file
