diff --git a/old_product.java b/new_product.java
index e6f7bff..8775c2e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,9 @@ public String csv(String text) {
         if (text == null || text.isEmpty()) {
             return "\"\"";
         }
-        return StringEscapeUtils.escapeCsv(text.trim().replace("\n", " "));
+        final String str = text.trim().replace("\n", " ");
+        if (str.length()==0) {
+            return "\"\"";
+        }
+        return StringEscapeUtils.escapeCsv(str);
     }
\ No newline at end of file
