diff --git a/old_product.java b/new_product.java
index 17b7952..1b24949 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,12 @@
 private void ls(AlluxioURI path, boolean recursive, boolean forceLoadMetadata, boolean dirAsFile,
-                  boolean hSize, boolean pinnedOnly, String sortField, boolean reverse)
+      boolean hSize, boolean pinnedOnly, String sortField, boolean reverse, String timestampOption)
       throws AlluxioException, IOException {
     URIStatus pathStatus = mFileSystem.getStatus(path);
     if (dirAsFile) {
       if (pinnedOnly && !pathStatus.isPinned()) {
         return;
       }
-      printLsString(pathStatus, hSize);
+      printLsString(pathStatus, hSize, timestampOption);
       return;
     }
 
@@ -33,7 +33,7 @@ private void ls(AlluxioURI path, boolean recursive, boolean forceLoadMetadata, b
     List<URIStatus> sorted = sortByFieldAndOrder(statuses, sortField, reverse);
     for (URIStatus status : sorted) {
       if (!pinnedOnly || status.isPinned()) {
-        printLsString(status, hSize);
+        printLsString(status, hSize, timestampOption);
       }
     }
   }
\ No newline at end of file
