diff --git a/old_product.java b/new_product.java
index 3c9080e..52f0db1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public List<AlignmentInterval.Row> packAlignments(
-            Iterator<Alignment> iter, int maxLevels, int end, boolean pairAlignments,
+            Iterator<Alignment> iter, int end, boolean pairAlignments,
             AlignmentTrack.SortOption groupBy) {
 
         // Compares 2 alignments by length.
@@ -17,7 +17,7 @@ public List<AlignmentInterval.Row> packAlignments(
         }
 
         if (groupBy == null) {
-            pack(iter, maxLevels, end, pairAlignments, lengthComparator, alignmentRows);
+            pack(iter, end, pairAlignments, lengthComparator, alignmentRows);
         } else {
             // Separate by group
             List<Alignment> nullGroup = new ArrayList();
@@ -39,9 +39,9 @@ public List<AlignmentInterval.Row> packAlignments(
             Collections.sort(keys);
             for (String key : keys) {
                 List<Alignment> group = groupedAlignments.get(key);
-                pack(group.iterator(), maxLevels, end, pairAlignments, lengthComparator, alignmentRows);
+                pack(group.iterator(), end, pairAlignments, lengthComparator, alignmentRows);
             }
-            pack(nullGroup.iterator(), maxLevels, end, pairAlignments, lengthComparator, alignmentRows);
+            pack(nullGroup.iterator(), end, pairAlignments, lengthComparator, alignmentRows);
         }
 
         return alignmentRows;
