diff --git a/old_test.java b/new_test.java
index 19e23d1..c92f63f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -24,6 +24,6 @@
         int maxLevels = 1000;
 
 
-        List<AlignmentInterval.Row> result = (new AlignmentPacker()).packAlignments(iter, maxLevels, end, false);
+        List<AlignmentInterval.Row> result = (new AlignmentPacker()).packAlignments(iter, end, false);
 
     }
\ No newline at end of file
