diff --git a/old_test.java b/new_test.java
index c92057f..517801d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     {
         System.out.println("supports Class");
 
-        EZIDIdentifierProvider instance = new EZIDIdentifierProvider();
+        EZIDIdentifierProvider instance = new DSpace().getServiceManager().getServiceByName(EZIDIdentifierProvider.class.getName(), EZIDIdentifierProvider.class);
 
         Class<? extends Identifier> identifier = DOI.class;
         boolean result = instance.supports(identifier);
