diff --git a/old_product.java b/new_product.java
index 79d608c..62d2d64 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,10 +13,5 @@
         if (LOG.isDebugEnabled()) {
             LOG.debug("Setting the id of process for job with id " + id + " to " + pid);
         }
-        final Job job = this.jobRepo.findOne(id);
-        if (job != null) {
-            job.setProcessHandle(pid);
-        } else {
-            throw new GenieNotFoundException("No job with id " + id + " exists");
-        }
+        this.findJob(id).setProcessHandle(pid);
     }
\ No newline at end of file
