diff --git a/old_test.java b/new_test.java
index c780d8d..94df28d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,5 +10,5 @@
 		Locale lu_UG = new Locale("lu", "UG");
 		Context.setLocale(lu_UG);
 		//note that unique list of locales should be lu_UG, lu, sw_KE, en_US, en
-		Assert.assertEquals(5, LocaleUtility.getLocalesInOrder().size());
+		Assert.assertEquals(6, LocaleUtility.getLocalesInOrder().size());
 	}
\ No newline at end of file
