diff --git a/old_product.java b/new_product.java
index c9ead91..0ecaa55 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,6 +10,7 @@ public static Set<Locale> getLocalesInOrder() {
 			locales.addAll(localesAllowedListCache);
 		
 		locales.add(Locale.ENGLISH);
+		locales.add(fromSpecification(OpenmrsConstants.GLOBAL_PROPERTY_DEFAULT_LOCALE_DEFAULT_VALUE));
 		
 		return locales;
 	}
\ No newline at end of file
