diff --git a/old_test.java b/new_test.java
index e1ddefb..0e973a4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
-    public void testGetFile() throws IOException {
+    public void testGetDataFile() throws IOException {
         String key = Settings.KEYS.CVE_DATA_DIRECTORY;
         String expResult = "data" + File.separator + "cve";
-        File result = Settings.getFile(key);
+        File result = Settings.getDataFile(key);
         Assert.assertTrue(result.getAbsolutePath().endsWith(expResult));
 
-        result = Settings.getFile(Settings.KEYS.DATA_DIRECTORY);
+        result = Settings.getDataFile(Settings.KEYS.DATA_DIRECTORY);
         String path = result.getPath();
         Assert.assertTrue(path.endsWith("data") || path.endsWith("data" + File.separator));
     }
\ No newline at end of file
