diff --git a/old_product.java b/new_product.java
index 609a077..9b1fb78 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public void setUserFormatDescription(String desc) throws SQLException {
-        setFormat(null);
-        setMetadataSingleValue(MetadataSchema.DC_SCHEMA, "format", null, null, desc);
+public void setUserFormatDescription(Context context, String desc) throws SQLException
+    {
+        getBitstreamService().setUserFormatDescription(context, this, desc);
     }
\ No newline at end of file
