diff --git a/old_test.java b/new_test.java
index e565e05..c93ad75 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testSetUserFormatDescription() throws SQLException
     {
         String userdescription = "user format description";
-        bs.setUserFormatDescription(userdescription);
+        bs.setUserFormatDescription(context, userdescription);
         assertThat("testSetUserFormatDescription 0", bs.getUserFormatDescription()
                 , notNullValue());
         assertThat("testSetUserFormatDescription 1", bs.getUserFormatDescription()
