diff --git a/old_product.java b/new_product.java
index 62149f0..6e0fd14 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,7 +10,7 @@
                     if (streamStatus == UpdateStreamStatus.SUCCESS) {
                         return Response.status(Status.CREATED).
                                 entity(ModelHelper.encodeStreamResponse(streamConfiguration)).build();
-                    } else if (streamStatus == UpdateStreamStatus.STREAM_NOT_FOUND) {
+                    } else if (streamStatus == UpdateStreamStatus.STREAM_NOT_FOUND || streamStatus == UpdateStreamStatus.SCOPE_NOT_FOUND) {
                         return Response.status(Status.NOT_FOUND).build();
                     } else {
                         return Response.status(Status.INTERNAL_SERVER_ERROR).build();
