diff --git a/old_test.java b/new_test.java
index b915e51..3f28fd9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,5 +15,12 @@
         // Test for validation of request object
         when(mockControllerService.alterStream(any())).thenReturn(updateStreamStatus3);
         response = target(resourceURI).request().async().put(Entity.json(updateStreamRequest3));
+        // TODO: Server should be returning 400 here, change this once issue
+        // https://github.com/pravega/pravega/issues/531 is fixed.
         assertEquals("Update Stream Status", 500, response.get().getStatus());
+
+        // Test to update stream for non-existent scope
+        when(mockControllerService.alterStream(any())).thenReturn(updateStreamStatus4);
+        response = target(resourceURI).request().async().put(Entity.json(updateStreamRequest));
+        assertEquals("Update Stream Status", 404, response.get().getStatus());
     }
\ No newline at end of file
