diff --git a/old_test.java b/new_test.java
index cf58af5..c19b516 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,9 @@
 @Test
-    public void getSystemCTLSchemas() throws Exception {
-        CTLSchemaDto saved = this.createCTLSchema(null, null, CTLSchemaScope.SYSTEM);
-        List<CTLSchemaDto> loaded = client.getSystemCTLSchemas();
+    public void getSystemCTLSchemaByFqnAndVersionTest() throws Exception {
+        this.loginKaaAdmin();
+        CTLSchemaDto saved = this.createCTLSchema(this.randomFieldType(), DEFAULT_NAMESPACE, 1, CTLSchemaScope.SYSTEM, null, null);
+        this.loginTenantDeveloper(tenantDeveloperUser);
+        CTLSchemaDto loaded = client.getCTLSchemaByFqnAndVersion(saved.getFqn(), saved.getVersion());
         Assert.assertNotNull(loaded);
-        Assert.assertEquals(1, loaded.size());
-        Assert.assertEquals(saved, loaded.get(0));
+        Assert.assertEquals(saved, loaded);
     }
\ No newline at end of file
