diff --git a/old_product.java b/new_product.java
index 45054fc..963ea05 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,6 +4,7 @@
   @ReturnType("alluxio.wire.LockBlockResult")
   public Response lockBlock(@QueryParam("sessionId") final Long sessionId,
       @QueryParam("blockId") final Long blockId) {
+    // NOTE: the logic here does not match the thrift lockBlock interface anymore since 1.5.0.
     return RestUtils.call(new RestUtils.RestCallable<LockBlockResult>() {
       @Override
       public LockBlockResult call() throws Exception {
@@ -11,7 +12,8 @@
         Preconditions.checkNotNull(sessionId, "required 'sessionId' parameter is missing");
         long lockId = mBlockWorker.lockBlock(sessionId, blockId);
         return new LockBlockResult().setLockId(lockId)
-            .setBlockPath(mBlockWorker.readBlock(sessionId, blockId, lockId));
+            .setBlockPath(mBlockWorker.readBlock(sessionId, blockId, lockId)).setLockBlockStatus(
+                LockBlockResult.LockBlockStatus.ALLUXIO_BLOCK_LOCKED);
       }
     });
   }
\ No newline at end of file
