diff --git a/old_test.java b/new_test.java
index 8ac2120..a3bdc4f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
-	public void getProviders_shouldReturnAllProvidersIfQueryIsEmpty() throws Exception {
+	public void getProviders_shouldReturnAllProvidersIfQueryIsEmptyAndIncludeRetiredTrue() throws Exception {
 		//given
 		List<Provider> allProviders = service.getAllProviders();
 		
 		//when
-		List<Provider> providers = service.getProviders("", null, null, null);
+		List<Provider> providers = service.getProviders("", null, null, null, true);
 		
 		//then
 		Assert.assertEquals(allProviders.size(), providers.size());
