diff --git a/old_product.java b/new_product.java
index 7bd0e3e..a105fb3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,3 @@
-@Transactional(readOnly = true)
-	@Authorized( { PrivilegeConstants.VIEW_PROVIDERS })
+@Authorized( { PrivilegeConstants.VIEW_PROVIDERS })
 	public List<Provider> getProviders(String query, Integer start, Integer length,
 	        Map<ProviderAttributeType, Object> attributes);
\ No newline at end of file
