diff --git a/old_product.java b/new_product.java
index 2a59207..21f8155 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,15 @@
-public static Map<String, Object> describe(Object bean){
+public static Map<String, Object> describe(Object bean,String...propertyNames){
         Validate.notNull(bean, "bean can't be null!");
-        try{
-            return PropertyUtils.describe(bean);
-        }catch (Exception e){
-            throw new BeanUtilException(e);
+        if (isNullOrEmpty(propertyNames)){
+            try{
+                return PropertyUtils.describe(bean);
+            }catch (Exception e){
+                throw new BeanUtilException(e);
+            }
         }
+        Map<String, Object> map = newLinkedHashMap(propertyNames.length);
+        for (String propertyName : propertyNames){
+            map.put(propertyName, getProperty(bean, propertyName));
+        }
+        return map;
     }
\ No newline at end of file
