diff --git a/old_product.java b/new_product.java
index b7cfc83..e539c1b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,5 +6,8 @@ public ServiceCall<IntentExport> getIntent(GetIntentOptions getIntentOptions) {
     if (getIntentOptions.export() != null) {
       builder.query("export", String.valueOf(getIntentOptions.export()));
     }
+    if (getIntentOptions.includeAudit() != null) {
+      builder.query("include_audit", String.valueOf(getIntentOptions.includeAudit()));
+    }
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(IntentExport.class));
   }
\ No newline at end of file
