diff --git a/old_test.java b/new_test.java
index 4062b89..df53d66 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,12 +10,16 @@
     Date start = new Date();
 
     CreateIntentOptions createOptions = new CreateIntentOptions.Builder().workspaceId(workspaceId).intent(intentName)
-            .description(intentDescription).examples(intentExamples).build();
+        .description(intentDescription).examples(intentExamples).build();
     service.createIntent(createOptions).execute();
 
     try {
-      GetIntentOptions getOptions =
-              new GetIntentOptions.Builder().workspaceId(workspaceId).intent(intentName).export(true).build();
+      GetIntentOptions getOptions = new GetIntentOptions.Builder()
+          .workspaceId(workspaceId)
+          .intent(intentName)
+          .export(true)
+          .includeAudit(true)
+          .build();
       IntentExport response = service.getIntent(getOptions).execute();
       assertNotNull(response);
       assertNotNull(response.getIntentName());
