diff --git a/old_test.java b/new_test.java
index 8c62615..aa7fc59 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,9 @@
-@TestInfo(
-              level = TestLevel.PARTIAL,
-              purpose = "",
-              targets = {
-                @TestTarget(
-                  methodName = "connect",
-                  methodArgs = {java.net.SocketAddress.class}
-                )
-            })
+@TestTargetNew(
+        level = TestLevel.PARTIAL_COMPLETE,
+        notes = "",
+        method = "connect",
+        args = {java.net.SocketAddress.class}
+    )
     public void testConnect_NonBlockWithServer() throws IOException {
         // Non blocking mode
         this.channel1.configureBlocking(false);
