diff --git a/old_test.java b/new_test.java
index c19ee35..87c2876 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
 	public void naive8_equal() {
-		ImageFloat32 intensity = new ImageFloat32(3,3);
-		ImageFloat32 output = new ImageFloat32(3,3);
-		ImageSInt8 direction = new ImageSInt8(3,3);
+		GrayF32 intensity = new GrayF32(3,3);
+		GrayF32 output = new GrayF32(3,3);
+		GrayS8 direction = new GrayS8(3,3);
 
 		GImageMiscOps.fill(intensity, 2);
 
