diff --git a/old_product.java b/new_product.java
index 647ccc9..ba6cc61 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,7 @@
 		if (getPerson() != null && getPerson().getPersonName() != null) {
 			return getPerson().getPersonName().getFullName();
 		} else {
-			return super.getName();
+			log.warn("We no longer support providers who are not linked to person. Set the name on the linked person");
+			return null;
 		}
 	}
\ No newline at end of file
