diff --git a/old_test.java b/new_test.java
index 6740d15..7eae10e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,7 @@
 @Test
-	@Verifies(value = "return person full name if person is not null", method = "getName()")
-	public void getName_shouldReturnPersonFullNameIfPersonIsNotNull() throws Exception {
-		final String providerName = "Provider Name";
-		
+	@Verifies(value = "return person full name if person is not null or null otherwise", method = "getName()")
+	public void getName_shouldReturnPersonFullNameIfPersonIsNotNullOrNullOtherwise() throws Exception {
 		Provider provider = new Provider();
-		provider.setName(providerName);
-		Assert.assertEquals(providerName, provider.getName());
 		
 		Person person = new Person(1);
 		person.addName(new PersonName("givenName", "middleName", "familyName"));
