diff --git a/old_test.java b/new_test.java
index ea02ca7..f64a083 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void writeTextLines() {
-    error(_FILE_WRITE_TEXT_LINES.args(PATH, "x"), Err.FILE_IS_DIR);
-    error(_FILE_WRITE_TEXT_LINES.args(PATH1, " 123"), Err.INVCASTEX);
+    error(_FILE_WRITE_TEXT_LINES.args(PATH, "x"), Err.FILE_IS_DIR_X);
+    error(_FILE_WRITE_TEXT_LINES.args(PATH1, " 123"), Err.INVCAST_X_X_X);
 
     query(_FILE_WRITE_TEXT_LINES.args(PATH1, "x"));
     query(_FILE_SIZE.args(PATH1), 1 + Prop.NL.length());
