diff --git a/old_test.java b/new_test.java
index 3d268ac..717a6d4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,13 @@
-@Test
-  public void soundexDifference() {
-    soundexDifference("Smith", "Smythe", 4);
-    soundexDifference("Ann", "Andrew", 2);
-    soundexDifference("Margaret", "Andrew", 1);
-    soundexDifference("Janet", "Margaret", 0);
+@Test public void soundexDifference() {
+    soundexDiff("Smith", "Smythe", 4);
+    soundexDiff("Ann", "Andrew", 2);
+    soundexDiff("Margaret", "Andrew", 1);
+    soundexDiff("Janet", "Margaret", 0);
 
-    soundexDifference("Green", "Greene", 4);
-    soundexDifference("Blotchet-Halls", "Greene", 0);
+    soundexDiff("Green", "Greene", 4);
+    soundexDiff("Blotchet-Halls", "Greene", 0);
 
-    soundexDifference("Smith", "Smythe", 4);
-    soundexDifference("Smithers", "Smythers", 4);
-    soundexDifference("Anothers", "Brothers", 2);
+    soundexDiff("Smith", "Smythe", 4);
+    soundexDiff("Smithers", "Smythers", 4);
+    soundexDiff("Anothers", "Brothers", 2);
   }
\ No newline at end of file
