diff --git a/old_test.java b/new_test.java
index f877d3c..fe6c8e6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(expectedExceptions=CalendricalRuleException.class)
     public void test_checkIntValue_valid() {
-        if (rule().getRange().isIntValue()) {
+        if (rule().getValueRange().isIntValue()) {
             rule().checkIntValue();
             throw new CalendricalRuleException("Dummy", rule());
         } else {
