diff --git a/old_product.java b/new_product.java
index 3dfbdc1..1318abf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public void checkIntValue() {
-        DateTimeRuleRange range = getRange();
+        DateTimeRuleRange range = getValueRange();
         if (range.isIntValue() == false) {
             throw new CalendricalRuleException("Rule does not specify an int value: " + getName(), this);
         }
