diff --git a/old_product.java b/new_product.java
index d9f1178..047a3c6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,9 +8,9 @@ public final Run getRunAt (int x,
             return null;
         }
 
-        RunSequence seq = getSequence(oPt.y);
+        for (Itr it = new Itr(oPt.y); it.hasNext();) {
+            Run run = it.next();
 
-        for (Run run : seq) {
             if (run.getStart() > oPt.x) {
                 return null;
             }
