diff --git a/old_test.java b/new_test.java
index 10c7fcc..e1d18ea 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,8 @@
 @Test
-    public void testExcludePorts() throws Exception {
-        Set<String> excludePorts = config.excludePorts();
-        assertThat(excludePorts.size(), is(2));
-        assertTrue(excludePorts.contains(PORT_NAME_1));
-        assertTrue(excludePorts.contains(PORT_NAME_2));
+    public void testSuppressHost() throws Exception {
+        Set<ConnectPoint> suppressHost = config.suppressHost();
+        assertNotNull("suppressHost should not be null", suppressHost);
+        assertThat(suppressHost.size(), is(2));
+        assertTrue(suppressHost.contains(PORT_1));
+        assertTrue(suppressHost.contains(PORT_2));
     }
\ No newline at end of file
