diff --git a/old_test.java b/new_test.java
index e5dbb9e..9df17ba 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
         CountDownLatch blocker2 = new CountDownLatch(1);
 
         try {
-            ManagedCompletableFuture<Boolean> cf1 = ManagedCompletableFuture.supplyAsync(() -> {
+            CompletableFuture<Boolean> cf1 = ManagedCompletableFuture.supplyAsync(() -> {
                 System.out.println("> supplyAsync[1] from testAcceptEither");
                 try {
                     boolean result = blocker1.await(TIMEOUT_NS * 2, TimeUnit.NANOSECONDS);
@@ -29,7 +29,7 @@
             });
 
             LinkedBlockingQueue<Object> results = new LinkedBlockingQueue<Object>();
-            ManagedCompletableFuture<Void> cf3 = cf1.acceptEither(cf2, (b) -> {
+            CompletableFuture<Void> cf3 = cf1.acceptEither(cf2, (b) -> {
                 System.out.println("> lookup from testAcceptEither");
                 results.add(b);
                 results.add(Thread.currentThread().getName());
