diff --git a/old_test.java b/new_test.java
index 5883e6c..68bb387 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test(expectedExceptions=NullPointerException.class, groups={"tck"})
-    public void test_with_null_adjustDate() {
-        TEST_2007_07_15_12_30_40_987654321.with(new MockDateAdjusterReturnsNull());
+    public void test_with_null() {
+        TEST_2007_07_15_12_30_40_987654321.with(null);
     }
\ No newline at end of file
