diff --git a/old_product.java b/new_product.java
index 343998c..6267598 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,10 @@
-public LocalTime with(TimeAdjuster adjuster) {
-        LocalTime time = adjuster.adjustTime(this);
-        if (time == null) {
-            throw new NullPointerException("TimeAdjuster implementation must not return null");
+@Override
+    public LocalTime with(CalendricalAdjuster adjuster) {
+        if (adjuster instanceof TimeAdjuster) {
+            return ((TimeAdjuster) adjuster).adjustTime(this);
+        } else if (adjuster instanceof LocalTime) {
+            return ((LocalTime) adjuster);
         }
-        return time;
+        DateTimes.checkNotNull(adjuster, "Adjuster must not be null");
+        throw new CalendricalException("Unable to adjust LocalTime with " + adjuster.getClass().getSimpleName());
     }
\ No newline at end of file
