diff --git a/old_product.java b/new_product.java
index bbcdd30..9d0d20b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
-public static void putMetricData(CloudWatchClient cloudWatchClient, String region, Iterable<MetricDatum> metrics,
+public static void putMetricData(CloudWatchApi cloudWatchApi, String region, Iterable<MetricDatum> metrics,
             String namespace) {
-      MetricClient metricClient = cloudWatchClient.getMetricClientForRegion(region);
+      MetricApi metricApi = cloudWatchApi.getMetricApiForRegion(region);
 
       for (List<MetricDatum> slice : Iterables.partition(metrics, 10)) {
-         metricClient.putMetricsInNamespace(slice, namespace);
+         metricApi.putMetricsInNamespace(slice, namespace);
       }
    }
\ No newline at end of file
