diff --git a/old_test.java b/new_test.java
index 78be670..0db24eb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
    public void testPutMetricData() throws Exception {
-      CloudWatchClient client = createMock(CloudWatchClient.class);
-      MetricClient metricClient = createMock(MetricClient.class);
+      CloudWatchApi api = createMock(CloudWatchApi.class);
+      MetricApi metricApi = createMock(MetricApi.class);
       Set<MetricDatum> metrics = Sets.newLinkedHashSet();
       String namespace = "JCLOUDS/Test";
 
@@ -10,17 +10,17 @@
       }
 
       // Using EasyMock.eq("") because EasyMock makes it impossible to pass null as a String value here
-      expect(client.getMetricClientForRegion(EasyMock.eq("")))
-            .andReturn(metricClient)
+      expect(api.getMetricApiForRegion(EasyMock.eq("")))
+            .andReturn(metricApi)
             .atLeastOnce();
       
       for (List<MetricDatum> slice : Iterables.partition(metrics, 10)) {
-         metricClient.putMetricsInNamespace(slice, namespace);
+         metricApi.putMetricsInNamespace(slice, namespace);
       }
 
-      EasyMock.replay(client, metricClient);
+      EasyMock.replay(api, metricApi);
 
-      CloudWatch.putMetricData(client, "", metrics, namespace);
+      CloudWatch.putMetricData(api, "", metrics, namespace);
 
-      EasyMock.verify(metricClient);
+      EasyMock.verify(metricApi);
    }
\ No newline at end of file
