diff --git a/old_product.java b/new_product.java
index d8b5f8b..d05ed5c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
-boolean process(Operation operation) throws DataCorruptionException {
+void process(Operation operation) throws DataCorruptionException {
         if (!operation.canSerialize()) {
             // Nothing to do.
-            return false;
+            return;
         }
 
-        // Add entry to MemoryTransactionLog and ReadIndex/Cache. This callback is invoked from the QueueProcessor,
+        // Add entry to MemoryTransactionLog and ReadIndex/Cache. This callback is invoked from the OperationProcessor,
         // which always acks items in order of Sequence Number - so the entries should be ordered (but always check).
         if (operation instanceof StorageOperation) {
             addToReadIndex((StorageOperation) operation);
@@ -30,6 +30,4 @@ boolean process(Operation operation) throws DataCorruptionException {
             // while serving reads, so better stop now than later.
             throw new DataCorruptionException("About to have added a Log Operation to InMemoryOperationLog that was out of order.");
         }
-
-        return true;
     }
\ No newline at end of file
