diff --git a/old_test.java b/new_test.java
index 8a7fd47..be46e1c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,7 @@
 
 		// preparations
 		{
-			final MaterialDescriptor materialDescr = MaterialDescriptor.builder()
+			final MaterialDescriptor materialDescriptor = MaterialDescriptor.builder()
 					.complete(true)
 					.productDescriptor(createProductDescriptor())
 					.warehouseId(WAREHOUSE_ID)
@@ -20,31 +20,31 @@
 					.type(CandidateType.STOCK)
 					.clientId(org.getAD_Client_ID())
 					.orgId(org.getAD_Org_ID())
-					.materialDescr(materialDescr)
+					.materialDescriptor(materialDescriptor)
 					.build();
 			candidateRepositoryCommands.addOrUpdateOverwriteStoredSeqNo(candidate);
 
-			final MaterialDescriptor earlierMaterialDescr = materialDescr.withDate(t1);
+			final MaterialDescriptor earlierMaterialDescriptor = materialDescriptor.withDate(t1);
 
 			earlierCandidate = candidateRepositoryCommands
 					.addOrUpdateOverwriteStoredSeqNo(Candidate.builder()
 							.type(CandidateType.STOCK)
 							.clientId(org.getAD_Client_ID())
 							.orgId(org.getAD_Org_ID())
-							.materialDescr(earlierMaterialDescr)
+							.materialDescriptor(earlierMaterialDescriptor)
 							.build());
 
-			final MaterialDescriptor laterMaterialDescr = materialDescr.withDate(t3);
+			final MaterialDescriptor laterMaterialDescriptor = materialDescriptor.withDate(t3);
 
 			final Candidate laterCandidate = Candidate.builder()
 					.type(CandidateType.STOCK)
 					.clientId(org.getAD_Client_ID())
 					.orgId(org.getAD_Org_ID())
-					.materialDescr(laterMaterialDescr)
+					.materialDescriptor(laterMaterialDescriptor)
 					.build();
 			candidateRepositoryCommands.addOrUpdateOverwriteStoredSeqNo(laterCandidate);
 
-			final MaterialDescriptor evenLatermaterialDescr = materialDescr
+			final MaterialDescriptor evenLatermaterialDescriptor = materialDescriptor
 					.withQuantity(new BigDecimal("12"))
 					.withDate(t4);
 
@@ -52,18 +52,18 @@
 					.type(CandidateType.STOCK)
 					.clientId(org.getAD_Client_ID())
 					.orgId(org.getAD_Org_ID())
-					.materialDescr(evenLatermaterialDescr)
+					.materialDescriptor(evenLatermaterialDescriptor)
 					.build();
 			candidateRepositoryCommands.addOrUpdateOverwriteStoredSeqNo(evenLaterCandidate);
 
-			final MaterialDescriptor evenLatermaterialDescrWithDifferentWarehouse = evenLatermaterialDescr
+			final MaterialDescriptor evenLatermaterialDescrWithDifferentWarehouse = evenLatermaterialDescriptor
 					.withWarehouseId(OTHER_WAREHOUSE_ID);
 
 			evenLaterCandidateWithDifferentWarehouse = Candidate.builder()
 					.type(CandidateType.STOCK)
 					.clientId(org.getAD_Client_ID())
 					.orgId(org.getAD_Org_ID())
-					.materialDescr(evenLatermaterialDescrWithDifferentWarehouse)
+					.materialDescriptor(evenLatermaterialDescrWithDifferentWarehouse)
 					.build();
 			candidateRepositoryCommands.addOrUpdateOverwriteStoredSeqNo(evenLaterCandidateWithDifferentWarehouse);
 		}
