diff --git a/old_product.java b/new_product.java
index 6710a81..9c85a7c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,7 @@
 	{
 		final CandidatesQuery segment = CandidatesQuery.builder()
 				.type(CandidateType.STOCK)
-				.materialDescr(MaterialDescriptor.builderForQuery()
+				.materialDescriptor(MaterialDescriptor.builderForQuery()
 						.date(date)
 						.productDescriptor(productDescriptor)
 						.warehouseId(warehouseId)
