diff --git a/old_test.java b/new_test.java
index ea12bf1..5ca2c0f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,35 +5,21 @@
         String expResult = "\"\"";
         String result = instance.csvIdentifiers(ids);
         assertEquals(expResult, result);
-        
+
         ids = new HashSet<>();
         expResult = "\"\"";
         result = instance.csvIdentifiers(ids);
         assertEquals(expResult, result);
-        
+
         ids = new HashSet<>();
-        ids.add(new Identifier("cpe", "cpe:/a:somegroup:something:1.0", ""));
-        expResult = "\"\"";
-        result = instance.csvIdentifiers(ids);
-        assertEquals(expResult, result);
-        
-        ids = new HashSet<>();
-        ids.add(new Identifier("maven", "somegroup:something:1.0", ""));
-        expResult = "somegroup:something:1.0";
-        result = instance.csvIdentifiers(ids);
-        assertEquals(expResult, result);
-        
-        ids = new HashSet<>();
-        ids.add(new Identifier("cpe", "cpe:/a:somegroup:something:1.0", ""));
-        ids.add(new Identifier("maven", "somegroup:something:1.0", ""));
+        ids.add(new GenericIdentifier("somegroup:something:1.0", Confidence.HIGH));
         expResult = "somegroup:something:1.0";
         result = instance.csvIdentifiers(ids);
         assertEquals(expResult, result);
-        
+
         ids = new HashSet<>();
-        ids.add(new Identifier("cpe", "cpe:/a:somegroup:something:1.0", ""));
-        ids.add(new Identifier("maven", "somegroup:something:1.0", ""));
-        ids.add(new Identifier("maven", "somegroup2:something:1.2", ""));
+        ids.add(new GenericIdentifier("somegroup:something:1.0", Confidence.HIGH));
+        ids.add(new GenericIdentifier("somegroup2:something:1.2", Confidence.HIGH));
         expResult = "\"somegroup:something:1.0, somegroup2:something:1.2\"";
         String expResult2 = "\"somegroup2:something:1.2, somegroup:something:1.0\"";
         result = instance.csvIdentifiers(ids);
