diff --git a/old_product.java b/new_product.java
index 3791a07..fd04351 100644
--- a/old_product.java
+++ b/new_product.java
@@ -56,8 +56,7 @@ public WeldContainer initialize() {
         bootstrap.endInitialization();
 
         // Set up the ShutdownManager for later
-        this.shutdownManager = getInstanceByType(bootstrap.getManager(deployment.loadBeanDeploymentArchive(ShutdownManager.class)), ShutdownManager.class);
-        this.shutdownManager.setBootstrap(bootstrap);
+        this.shutdownManager = new ShutdownManager(bootstrap);
 
         WeldContainer container = getInstanceByType(bootstrap.getManager(deployment.loadBeanDeploymentArchive(WeldContainer.class)), WeldContainer.class);
 
