diff --git a/old_test.java b/new_test.java
index 7b4f058..22a43c5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,11 @@
 @Test(enabled = true, description = "Get now block by http")
-  public void getNowBlock() {
+  public void get1NowBlock() {
     response = HttpMethed.getNowBlock(httpnode);
-    try {
-      entity = response.getEntity();
-    } catch (Exception e) {
-      e.printStackTrace();
-    }
     logger.info("code is " + response.getStatusLine().getStatusCode());
     Assert.assertEquals(response.getStatusLine().getStatusCode(),200);
     responseContent = HttpMethed.parseResponseContent(response);
+    blockContent = responseContent;
+    blockId = responseContent.get("blockID").toString();
     HttpMethed.printJsonContent(responseContent);
     Assert.assertTrue(responseContent.size() >= 2);
     responseContent = HttpMethed.parseStringContent(responseContent.get("block_header").toString());
@@ -18,6 +15,7 @@
     responseContent = HttpMethed.parseStringContent(responseContent.get("raw_data").toString());
     HttpMethed.printJsonContent(responseContent);
     Assert.assertTrue(Integer.parseInt(responseContent.get("number").toString()) > 0);
+    currentBlockNum = Integer.parseInt(responseContent.get("number").toString());
     Assert.assertTrue(Long.parseLong(responseContent.get("timestamp").toString()) > 1550724114000L);
     Assert.assertFalse(responseContent.get("witness_address").toString().isEmpty());
   }
\ No newline at end of file
