diff --git a/old_product.java b/new_product.java
index d216269..d8549c2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
-@Override
-  public Description describe(MethodInvocationTree methodInvocationTree, VisitorState state) {
+public Description describe(MethodInvocationTree methodInvocationTree, VisitorState state) {
     // Find the root of the field access chain, i.e. a.intern().trim() ==> a.
     ExpressionTree identifierExpr = ASTHelpers.getRootAssignable(methodInvocationTree);
     String identifierStr = null;
@@ -28,5 +27,5 @@
       Tree parent = state.getPath().getParentPath().getLeaf();
       fix = new SuggestedFix().delete(parent);
     }
-    return new Description(methodInvocationTree, getDiagnosticMessage(), fix);
+    return describeMatch(methodInvocationTree, fix);
   }
\ No newline at end of file
