diff --git a/old_product.java b/new_product.java
index fc6683b..604f507 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,5 @@
     @ResponseStatus(HttpStatus.OK)
     public ClusterResource getCluster(@PathVariable("id") final String id) throws GenieException {
         log.debug("Called with id: {}", id);
-        return this.clusterResourceAssembler.toResource(this.clusterService.getCluster(id));
+        return this.clusterResourceAssembler.toResource(DtoAdapters.toV3Cluster(this.clusterService.getCluster(id)));
     }
\ No newline at end of file
