diff --git a/old_test.java b/new_test.java
index f1d755f..b948f39 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
     PowerMockito.whenNew(FileOutputStream.class).withArguments(mFile).thenReturn(outputStream);
     FileInputStream inputStream = PowerMockito.mock(FileInputStream.class);
     PowerMockito.whenNew(FileInputStream.class).withArguments(mFile).thenReturn(inputStream);
-    KodoOutputStream stream = new KodoOutputStream("testKey", mKodoClient);
+    KodoOutputStream stream = new KodoOutputStream("testKey", mKodoClient, sTmpDirs);
     stream.close();
     Mockito.verify(mFile).delete();
   }
\ No newline at end of file
