diff --git a/old_test.java b/new_test.java
index 83c0766..9268733 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-    public void testLearn() {
-        System.out.println("learn");
+    public void test() {
+        System.out.println("GHA");
 
         int k = 3;
         double[] mu = MathEx.colMeans(USArrests);
@@ -14,7 +14,7 @@
         for (int iter = 1, t = 0; iter <= 1000; iter++) {
             double error = 0.0;
             for (int i = 0; i < USArrests.length; i++, t++) {
-                error += gha.learn(USArrests[i]);
+                error += gha.update(USArrests[i]);
             }
             error /= USArrests.length;
 
