diff --git a/old_test.java b/new_test.java
index d5611db..c27c4c0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -24,7 +24,14 @@
                 Assert.assertEquals(200, response.statusCode);
 
             } catch (QiniuException e) {
+                e.printStackTrace();
                 Assert.fail(e.response.toString());
+            } finally {
+                try {
+                    bucketManager.putBucketAccessStyleMode(bucket, AccessStyleMode.CLOSE);
+                } catch (QiniuException e) {
+                    // do nothing
+                }
             }
         }
     }
\ No newline at end of file
