diff --git a/old_product.java b/new_product.java
index e99e88a..a88fab0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public Response putBucketAccessStyleMode(String bucket, AccessStyleMode mode) throws QiniuException {
-        String url = String.format("%s/accessMode/%s/mode/%d", configuration.ucHost(), bucket, mode.getType());
+        String url = String.format("%s/accessMode/%s/mode/%d", configHelper.ucHost(), bucket, mode.getType());
         Response res = post(url, null);
         if (!res.isOK()) {
             throw new QiniuException(res);
