diff --git a/old_product.java b/new_product.java
index 942c96b..c3012c8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
 public static PointTransform_F32 fullView( IntrinsicParameters param,
-											   boolean applyLeftToRight ,
 											   IntrinsicParameters paramAdj ) {
 
 		RemoveRadialPtoP_F32 removeDistort = new RemoveRadialPtoP_F32();
@@ -24,7 +23,7 @@ public static PointTransform_F32 fullView( IntrinsicParameters param,
 
 		PointTransform_F32 tranAdj = UtilIntrinsic.adjustIntrinsic_F32(addDistort, false, param, A, paramAdj);
 
-		if( applyLeftToRight) {
+		if( param.leftHanded ) {
 			PointTransform_F32 l2r = new LeftToRightHanded_F32(param.height);
 			return new SequencePointTransform_F32(l2r,tranAdj,l2r);
 		} else
