diff --git a/old_product.java b/new_product.java
index 5020bdc..aecb1f4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,7 +9,8 @@
         if (!this.commandRepo.exists(id)) {
             throw new GenieNotFoundException("No command exists with the given id. Unable to update.");
         }
-        if (!id.equals(updateCommand.getId())) {
+        final Optional<String> updateId = updateCommand.getId();
+        if (updateId.isPresent() && !id.equals(updateId.get())) {
             throw new GenieBadRequestException("Command id inconsistent with id passed in.");
         }
 
