diff --git a/old_product.java b/new_product.java
index 7a078cd..05e5f7a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,6 +4,7 @@ public void validate(Object obj, Errors errors) throws IllegalArgumentException
 			        + ConceptSource.class);
 		} else {
 			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "name", "error.name");
+			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "description", "error.null");
 			ValidateUtil.validateFieldLengths(errors, obj.getClass(), "name", "hl7Code", "uniqueId", "description",
 			    "retireReason");
 		}
