diff --git a/old_test.java b/new_test.java
index 75d25b0..1942efc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,21 +1,21 @@
 @Test
-	@Verifies(value = "should pass validation if description is null or empty or whitespace", method = "validate(Object,Errors)")
-	public void validate_shouldPassValidationIfDescriptionIsNullOrEmptyOrWhitespace() throws Exception {
+	public void validate_shouldFailValidationIfDescriptionIsNullOrEmptyOrWhitespace() throws Exception {
+
 		ConceptSource conceptSource = new ConceptSource();
 		conceptSource.setName("New name");
+
 		conceptSource.setDescription(null);
-		
 		Errors errors = new BindException(conceptSource, "conceptSource");
 		new ConceptSourceValidator().validate(conceptSource, errors);
-		Assert.assertFalse(errors.hasFieldErrors("description"));
+		Assert.assertTrue(errors.hasFieldErrors("description"));
 		
 		conceptSource.setDescription("");
 		errors = new BindException(conceptSource, "conceptSource");
 		new ConceptSourceValidator().validate(conceptSource, errors);
-		Assert.assertFalse(errors.hasFieldErrors("description"));
+		Assert.assertTrue(errors.hasFieldErrors("description"));
 		
 		conceptSource.setDescription("   ");
 		errors = new BindException(conceptSource, "conceptSource");
 		new ConceptSourceValidator().validate(conceptSource, errors);
-		Assert.assertFalse(errors.hasFieldErrors("description"));
+		Assert.assertTrue(errors.hasFieldErrors("description"));
 	}
\ No newline at end of file
