diff --git a/old_product.java b/new_product.java
index 3470309..3d96baa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,12 @@
-public static int parseString(String s, int i) {
+public static int parseLastElementOfStringToInt(String s, int i) {
         String[] ss = s.split("\\s+");
         if (ss.length < 2) {
             return i;
         } else {
-            return Integer.valueOf(ss[ss.length - 1]);
+            try {
+                return Integer.parseInt(ss[ss.length - 1]);
+            } catch (NumberFormatException nfe) {
+                return i;
+            }
         }
     }
\ No newline at end of file
