diff --git a/old_test.java b/new_test.java
index 8adcf70..0a8d072 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
     public void testParseString() {
-        assertEquals(1, ParseUtil.parseString("foo : 1", 0));
-        assertEquals(2, ParseUtil.parseString("foo", 2));
+        assertEquals(1, ParseUtil.parseLastElementOfStringToInt("foo : 1", 0));
+        assertEquals(2, ParseUtil.parseLastElementOfStringToInt("foo", 2));
     }
\ No newline at end of file
