diff --git a/old_product.java b/new_product.java
index a660571..632f93f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public final int addAllLast(Iterable<? extends ObjectCursor<? extends KType>> iterable)
+public final int addLast(Iterable<? extends ObjectCursor<? extends KType>> iterable)
     {
-        return addAllLast(iterable.iterator());
+        return addLast(iterable.iterator());
     }
\ No newline at end of file
