diff --git a/old_product.java b/new_product.java
index 2353e94..d87eb42 100644
--- a/old_product.java
+++ b/new_product.java
@@ -31,7 +31,7 @@ private void add(CharsRef input, int numInputWords, CharsRef output, int numOutp
       MapEntry e = workingSet.get(input);
       if (e == null) {
         e = new MapEntry();
-        workingSet.put(new CharsRef(input), e); // make a copy, since we will keep around in our map    
+        workingSet.put(CharsRef.deepCopyOf(input), e); // make a copy, since we will keep around in our map    
       }
       
       e.ords.add(ord);
