diff --git a/old_product.java b/new_product.java
index 75d1158..38af62c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -34,7 +34,7 @@ CompletableFuture<Void> initialize(Duration timeout, Executor executor) {
                     setState(AggregatorState.Writing);
                 })
                 .exceptionally(ex -> {
-                    ex = ExceptionHelpers.getRealException(ex);
+                    ex = Exceptions.unwrap(ex);
                     if (ex instanceof StreamSegmentNotExistsException) {
                         // Segment does not exist anymore. This is a real possibility during recovery, in the following cases:
                         // * We already processed a Segment Deletion but did not have a chance to checkpoint metadata
