diff --git a/old_product.java b/new_product.java
index 459e1e8..f006947 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,6 @@
-public byte[] fromMessage(Message<?> message) throws Exception {
-		return getPayloadAsBytes(message);
+public Object fromMessage(Message<?> message) throws Exception {
+		if (this.messageFormat < MessageFormats.FORMAT_IMPLICIT) {
+			return getPayloadAsBytes(message);
+		}
+		return message.getPayload();
 	}
\ No newline at end of file
