diff --git a/old_test.java b/new_test.java
index 0c81b68..3f2872e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
 		String s = "test";
 		Message<String> message = MessageBuilder.withPayload(s).build();
 		SocketMessageMapper mapper = new SocketMessageMapper();
-		byte[] bArray = mapper.fromMessage(message);
+		byte[] bArray = (byte[]) mapper.fromMessage(message);
 		assertEquals(s, new String(bArray));
 		
 	}
\ No newline at end of file
