diff --git a/old_product.java b/new_product.java
index cb183dd..67757d6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ public static AnalyzerResponse containsAlertAfterModifiedMessage(TlsContext tlsC
 	if (position == -1) {
 	    return AnalyzerResponse.NO_MODIFICATION;
 	} else {
-	    ProtocolMessage pm = getNextProtocolMessageFromPeer(tlsContext, position);
+	    ProtocolMessage pm = getNextReceiveProtocolMessage(tlsContext, position);
 	    if (pm != null && pm.getProtocolMessageType() == ProtocolMessageType.ALERT) {
 		return AnalyzerResponse.ALERT;
 	    } else {
