diff --git a/old_test.java b/new_test.java
index a307807..11f9a77 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     // dud properties. server should pick up defaults
     Properties properties = new Properties();
     VerifiableProperties verifiableProperties = new VerifiableProperties(properties);
-    RestRequestHandler restRequestHandler = new MockRequestResponseHandler();
+    RestRequestHandler restRequestHandler = new MockRestRequestResponseHandler();
 
     NettyServerFactory nioServerFactory =
         new NettyServerFactory(verifiableProperties, new MetricRegistry(), restRequestHandler);
