diff --git a/old_product.java b/new_product.java
index 27c8fa5..fa65e37 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,6 +7,6 @@ private NettyServer getNettyServer(Properties properties)
     VerifiableProperties verifiableProperties = new VerifiableProperties(properties);
     NettyConfig nettyConfig = new NettyConfig(verifiableProperties);
     NettyMetrics nettyMetrics = new NettyMetrics(new MetricRegistry());
-    RestRequestHandler requestHandler = new MockRequestResponseHandler();
+    RestRequestHandler requestHandler = new MockRestRequestResponseHandler();
     return new NettyServer(nettyConfig, nettyMetrics, requestHandler);
   }
\ No newline at end of file
