diff --git a/old_product.java b/new_product.java
index 538088b..24ca136 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-private ValueBuilder eval(final QueryContext ctx) throws QueryException {
-    return eval(ctx, checkStr(expr[0], ctx), null);
+private ValueBuilder eval(final QueryContext ctx, final boolean openDB) throws QueryException {
+    return eval(ctx, checkStr(expr[0], ctx), null, openDB);
   }
\ No newline at end of file
