diff --git a/old_test.java b/new_test.java
index 0aa0f5b..2021307 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,4 +16,5 @@
     error(_XQUERY_EVAL.args("declare %updating function local:x() {()}; local:x()"),
         Err.BXXQ_UPDATING);
     query(_XQUERY_EVAL.args("declare %updating function local:x() {()}; 1"));
+    query(_XQUERY_EVAL.args("\"" + DOC.args(PATH).replace('"', '\'') + "\""));
   }
\ No newline at end of file
