diff --git a/old_test.java b/new_test.java
index bc805b5..3c089a9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,9 +6,9 @@
             final String header = "^/**\\n * Licensed to the Apache Software Foundation (ASF)";
             instance.setHeader(header);
             fail(String.format(Locale.ROOT, "%s should have been thrown",
-                    ConversionException.class));
+                    IllegalArgumentException.class));
         }
-        catch (ConversionException ex) {
+        catch (IllegalArgumentException ex) {
             assertEquals("Unable to parse format: ^/**\\n *"
                     + " Licensed to the Apache Software Foundation (ASF)",
                     ex.getMessage());
