diff --git a/old_product.java b/new_product.java
index 9864c5a..53e3030 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,7 +10,7 @@ public void setHeader(String header) {
                 loadHeader(headerReader);
             }
             catch (final IOException ex) {
-                throw new ConversionException("unable to load header", ex);
+                throw new IllegalArgumentException("unable to load header", ex);
             }
             finally {
                 Closeables.closeQuietly(headerReader);
