diff --git a/old_test.java b/new_test.java
index 4431113..5460384 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
   public void run() throws Exception {
-    context.options.set(Options.TEXTINDEX, false);
-    context.options.set(Options.ATTRINDEX, false);
-    context.options.set(Options.AUTOFLUSH, false);
+    context.options.set(MainOptions.TEXTINDEX, false);
+    context.options.set(MainOptions.ATTRINDEX, false);
+    context.options.set(MainOptions.AUTOFLUSH, false);
 
     // create test database
     new CreateDB(NAME, "<X>" +
