diff --git a/old_product.java b/new_product.java
index d27d039..5f8a31c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,9 +4,9 @@ private void run(final String[] args) throws Exception {
     init();
 
     final Performance perf = new Performance();
-    ctx.options.set(Options.CHOP, false);
-    ctx.options.set(Options.INTPARSE, false);
-    ctx.options.set(Options.SERIALIZER, "omit-xml-declaration=no,indent=no");
+    ctx.options.set(MainOptions.CHOP, false);
+    ctx.options.set(MainOptions.INTPARSE, false);
+    ctx.options.set(MainOptions.SERIALIZER, "omit-xml-declaration=no,indent=no");
 
     final XdmValue doc = new XQuery("doc('" + file(false, CATALOG) + "')", ctx).value();
     final String version = asString("*:catalog/@version", doc);
