diff --git a/old_test.java b/new_test.java
index 353d13a..da9c258 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,6 +9,6 @@
 
         QueryBuilder newQuery = queryParserService.queryParser(testQuery.getName()).fromXContent(context);
         assertNotSame(newQuery, testQuery);
-        assertEquals("Queries should be equal: " + newQuery + " vs. " + testQuery, newQuery, testQuery);
-        assertEquals("Queries should have equal hashcodes: " + newQuery + " vs. " + testQuery, newQuery.hashCode(), testQuery.hashCode());
+        assertEquals("Queries should be equal", testQuery, newQuery);
+        assertEquals("Queries should have equal hashcodes", testQuery.hashCode(), newQuery.hashCode());
     }
\ No newline at end of file
