diff --git a/old_product.java b/new_product.java
index 45c8a7b..71198f7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@
         XContentParser parser = parseContext.parser();
         List<String> ids = new ArrayList<>();
         List<String> types = new ArrayList<>();
-        float boost = 1.0f;
+        float boost = AbstractQueryBuilder.DEFAULT_BOOST;
         String queryName = null;
 
         String currentFieldName = null;
