diff --git a/old_test.java b/new_test.java
index 8d19134..c4a5217 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testOnApply() {
-        Result result = new Result(false, true, 9, 10, new BranchTrace(), new TestVector(new WorkflowTrace(), null,
-                null, ExecutorType.TLS, null), "unit3.test");
+        AgentResult result = new AgentResult(false, true, 9, 10, new BranchTrace(), new TestVector(new WorkflowTrace(), null,
+                null, ExecutorType.TLS, null), "unit3.test", null);
         rule.onApply(result);
         assertTrue(new File(config.getOutputFolder() + rule.getConfig().getOutputFolder()).listFiles().length == 1);
 
