diff --git a/old_product.java b/new_product.java
index 80fa8a6..b6bb4cd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static void regionPixelId_to_Compact(ImageSInt32 graph, GrowQueue_I32 segmentId, ImageSInt32 output) {
+public static void regionPixelId_to_Compact(GrayS32 graph, GrowQueue_I32 segmentId, GrayS32 output) {
 
 		InputSanityCheck.checkSameShape(graph,output);
 
