diff --git a/old_test.java b/new_test.java
index d28f08c..55de142 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
 	public void regionPixelId_to_Compact() {
-		ImageSInt32 graph = new ImageSInt32(4,5);
-		ImageSInt32 output = new ImageSInt32(4,5);
+		GrayS32 graph = new GrayS32(4,5);
+		GrayS32 output = new GrayS32(4,5);
 
 		regionPixelId_to_Compact(graph, output);
 		regionPixelId_to_Compact(BoofTesting.createSubImageOf(graph), output);
