diff --git a/old_test.java b/new_test.java
index e7d75a9..7877a19 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
     when(mockTrigger.shouldFire(anyTriggerContext())).thenReturn(true);
     assertTrue(tester.shouldFire(new IntervalWindow(new Instant(0), new Instant(10))));
 
-    when(mockTrigger.shouldFire(Mockito.<Trigger<IntervalWindow>.TriggerContext>any()))
+    when(mockTrigger.shouldFire(Mockito.<Trigger.TriggerContext>any()))
         .thenReturn(false);
     assertFalse(tester.shouldFire(new IntervalWindow(new Instant(0), new Instant(10))));
   }
\ No newline at end of file
