diff --git a/old_product.java b/new_product.java
index 2f45304..4910ecd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @Override
   public boolean shouldFire(TriggerContext context) throws Exception {
-    for (ExecutableTrigger<W> subtrigger : context.trigger().subTriggers()) {
+    for (ExecutableTrigger subtrigger : context.trigger().subTriggers()) {
       if (!context.forTrigger(subtrigger).trigger().isFinished()
           && !subtrigger.invokeShouldFire(context)) {
         return false;
