diff --git a/old_test.java b/new_test.java
index 32a4d66..95fadbd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,6 @@
 @Test
   public void testNode() throws BaseXException {
-    // wrong arguments
-    args("db:node-id", (Class<?>) null);
-    args("db:node-pre", (Class<?>) null);
-    error("db:node-id(1)", "BASX0002");
-    error("db:node-pre(1)", "BASX0002");
-
-    // test results
-    query("db:node-id(/html)", "1");
-    query("db:node-pre(/html)", "1");
-    query("db:node-pre(/ | /html)", "0 1");
+    final String fun = check(FunDef.NODEID, DBNode.class);
+    query(fun + "(/html)", "1");
+    query(fun + "(/ | /html)", "0 1");
   }
\ No newline at end of file
