diff --git a/old_product.java b/new_product.java
index 3a3ea2e..e7fa11c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -24,7 +24,7 @@ final int node(final Data data, final int pre, final FTPosData ft)
       if(k == Data.DOC) {
         ++p;
       } else if(k == Data.TEXT) {
-        final FTPos ftd = ft != null ? ft.get(p) : null;
+        final FTPos ftd = ft != null ? ft.get(data, p) : null;
         if(ftd != null) text(data.text(p++, true), ftd);
         else text(data.text(p++, true));
       } else if(k == Data.COMM) {
