diff --git a/old_test.java b/new_test.java
index 2be3811..d1e4c4e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
         for (String partitionId : expectedPartitionIds) {
             StepVerifier.create(client.getPartitionProperties(partitionId))
                 .assertNext(properties -> {
-                    Assert.assertEquals(eventHubPath, properties.eventHubPath());
+                    Assert.assertEquals(eventHubName, properties.eventHubName());
                     Assert.assertEquals(partitionId, properties.id());
                 })
                 .verifyComplete();
