diff --git a/old_product.java b/new_product.java
index a47a657..d4502aa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@
     public Mono<PartitionProperties> getPartitionProperties(String partitionId) {
         final Map<String, Object> properties = new HashMap<>();
         properties.put(MANAGEMENT_ENTITY_TYPE_KEY, MANAGEMENT_PARTITION_ENTITY_TYPE);
-        properties.put(MANAGEMENT_ENTITY_NAME_KEY, eventHubPath);
+        properties.put(MANAGEMENT_ENTITY_NAME_KEY, eventHubName);
         properties.put(MANAGEMENT_PARTITION_NAME_KEY, partitionId);
         properties.put(MANAGEMENT_OPERATION_KEY, READ_OPERATION_VALUE);
 
