diff --git a/old_product.java b/new_product.java
index 445f803..6b9a1c1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery ltProp(String propName, String otherProp) {
-        criterion = criterion.and(criterionBuilder.ltProp(propName, otherProp));
+        criterion = criterion.and(Criteria.ltProp(propName, otherProp));
         return this;
     }
\ No newline at end of file
