diff --git a/old_test.java b/new_test.java
index 86010f1..3e710de 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
     public void testLtProp() {
-    	assertEquals(new LtPropCriterion("id", "name"), instance.ltProp("id", "name").getQueryCriterion());
+    	assertEquals(Criteria.ltProp("id", "name"), instance.ltProp("id", "name").getQueryCriterion());
     }
\ No newline at end of file
