diff --git a/old_product.java b/new_product.java
index de78d50..05c54cb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-SearchUserResponse searchUser(@WebParam(name = "loginName") String loginName, @WebParam(name = "name") String name);
\ No newline at end of file
+SearchUserResult searchUser(@WebParam(name = "loginName") String loginName, @WebParam(name = "name") String name);
\ No newline at end of file
