diff --git a/old_test.java b/new_test.java
index f3382d3..3e3608f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,4 +11,9 @@
         when(mockControllerService.createStream(any(), anyLong())).thenReturn(createStreamStatus2);
         response = target(streamResourceURI).request().async().post(Entity.json(createStreamRequest));
         assertEquals("Create Stream Status", 409, response.get().getStatus());
+
+        // Test for validation of create stream request object
+        when(mockControllerService.createStream(any(), anyLong())).thenReturn(createStreamStatus3);
+        response = target(streamResourceURI).request().async().post(Entity.json(createStreamRequest2));
+        assertEquals("Create Stream Status", 400, response.get().getStatus());
     }
\ No newline at end of file
