diff --git a/old_product.java b/new_product.java
index fef7024..dd23792 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,5 @@
         @Path("/scopes/{scope}/streams")
         @Produces(MediaType.APPLICATION_JSON)
         @Consumes(MediaType.APPLICATION_JSON)
-        public void createStream(@PathParam("scope") final String scope, final CreateStreamRequest createStreamRequest,
+        public void createStream(@PathParam("scope") final String scope, @Valid final CreateStreamRequest createStreamRequest,
                                  @Suspended final AsyncResponse asyncResponse);
\ No newline at end of file
