diff --git a/old_product.java b/new_product.java
index 500610f..a9ba010 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,7 +16,7 @@
             ps.executeUpdate();
             
         } catch (SQLException sqlEX) {
-            throw new FeatureAccessException("Cannot check feature existence, error related to database", sqlEX);
+            throw new FeatureAccessException(CANNOT_CHECK_FEATURE_EXISTENCE_ERROR_RELATED_TO_DATABASE, sqlEX);
         } finally {
             closeStatement(ps);
             closeConnection(sqlConn);
