diff --git a/old_test.java b/new_test.java
index 8b3bc1e..09a0027 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
     public void testClear() {
         // Given
-        cacheManager.putFeature(new Feature("ff", false, "Description"));
-        cacheManager.putFeature(new Feature("ff2", false, "Description"));
-        cacheManager.putFeature(new Feature("ff3", false, "Description"));
+        cacheManager.putFeature(new Feature("ff", false, DESCRIPTION));
+        cacheManager.putFeature(new Feature("ff2", false, DESCRIPTION));
+        cacheManager.putFeature(new Feature("ff3", false, DESCRIPTION));
         Assert.assertEquals(3, cacheManager.listCachedFeatureNames().size());
         // When
         cacheManager.clearFeatures();
