diff --git a/old_product.java b/new_product.java
index 036f93b..d13b182 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,8 +2,10 @@ public static String decodePath(String path)
     {
         if (path==null)
             return null;
+        // Array to hold all converted characters
         char[] chars=null;
         int n=0;
+        // Array to hold a sequence of %encodings
         byte[] bytes=null;
         int b=0;
         
@@ -25,14 +27,26 @@ public static String decodePath(String path)
                 i+=2;
                 continue;
             }
+            else if (c==';')
+            {
+                if (chars==null)
+                {
+                    chars=new char[len];
+                    path.getChars(0,i,chars,0);
+                    n=i;
+                }
+                break;
+            }
             else if (bytes==null)
             {
                 n++;
                 continue;
             }
             
+            // Do we have some bytes to convert?
             if (b>0)
             {
+                // convert series of bytes and add to chars
                 String s;
                 try
                 {
@@ -53,8 +67,10 @@ public static String decodePath(String path)
         if (chars==null)
             return path;
 
+        // if we have a remaining sequence of bytes
         if (b>0)
         {
+            // convert series of bytes and add to chars
             String s;
             try
             {
