diff --git a/old_test.java b/new_test.java
index 6c6dd40..83bc5fd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,17 +2,17 @@
     public void testGenerateAgent() throws IllegalAgentException {
         EvolutionaryFuzzerConfig config = new EvolutionaryFuzzerConfig();
         config.setAgent("BLIND");
-        Agent agent = AgentFactory.generateAgent(config, null);
+        Agent agent = AgentFactory.generateAgent(config, null,null);
         assertTrue(agent instanceof BlindAgent);
         config.setAgent("PIN");
-        agent = AgentFactory.generateAgent(config, null);
+        agent = AgentFactory.generateAgent(config, null,null);
         assertTrue(agent instanceof PINAgent);
         config.setAgent("AFL");
-        agent = AgentFactory.generateAgent(config, null);
+        agent = AgentFactory.generateAgent(config, null,null);
         assertTrue(agent instanceof AFLAgent);
         config.setAgent("NOT A REAL AGENT");
         try {
-            agent = AgentFactory.generateAgent(config, null);
+            agent = AgentFactory.generateAgent(config, null, null);
             fail("Undefined Agent did not throw an Exception");
         } catch (IllegalAgentException E) {
             assertTrue(E != null);
