diff --git a/old_test.java b/new_test.java
index b97376c..faa508a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
 	public void testToMessage() throws Exception {
 		SocketMessageMapper mapper = new SocketMessageMapper();
 		Message<byte[]> message = mapper.toMessage(new StubSocketReader());
-		assertEquals(TEST_PAYLOAD, new String((byte[]) message.getPayload()));
+		assertEquals(TEST_PAYLOAD, new String(message.getPayload()));
 		assertEquals(InetAddress.getLocalHost().getHostName(), message
 				.getHeaders().get(IpHeaders.HOSTNAME));
 		assertEquals(InetAddress.getLocalHost().getHostAddress(), message
