diff --git a/old_test.java b/new_test.java
index 8b53edb..dfa694a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
-    @SuppressWarnings("static-method")
     public void testNewDynaBean(){
         Map<String, Object> map = newHashMap();
 
@@ -12,12 +11,4 @@
         assertEquals(map, newDynaBean.get("address"));
         assertEquals("Fred", newDynaBean.get("firstName"));
         assertEquals("Flintstone", newDynaBean.get("lastName"));
-
-        //        assertThat(newDynaBean, allOf(//
-        //                        hasProperty("address", is((Object) new HashMap())),
-        //                        hasProperty("firstName", is((Object) "Fred")),
-        //                        hasProperty("lastName", is((Object) "Flintstone"))
-        //
-        //        //
-        //        ));
     }
\ No newline at end of file
