diff --git a/old_product.java b/new_product.java
index 8e8e870..785b686 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,13 @@
-public static DynaBean newDynaBean(Map<String, ?> valueMap){
+public static DynaBean newDynaBean(Map<?, ?> valueMap){
         Validate.notNull(valueMap, "valueMap can't be null!");
 
         //---------------------------------------------------------------
-
         LazyDynaBean lazyDynaBean = new LazyDynaBean();
-        for (Map.Entry<String, ?> entry : valueMap.entrySet()){
-            Validate.notBlank(entry.getKey(), "entry.getKey() can't be blank!");
-            lazyDynaBean.set(entry.getKey(), entry.getValue());
+        for (Map.Entry<?, ?> entry : valueMap.entrySet()){
+            String key = ConvertUtil.toString(entry.getKey());
+
+            Validate.notBlank(key, "entry.getKey() can't be blank!");
+            lazyDynaBean.set(key, entry.getValue());
         }
         return lazyDynaBean;
     }
\ No newline at end of file
