diff --git a/old_product.java b/new_product.java
index 9ab537a..cdc3bc7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
-public static void syncLocalCoreWithSharedStore(String collectionName, String coreName, String shardName, CoreContainer coreContainer) throws SolrException
-  {
+public static void syncLocalCoreWithSharedStore(String collectionName, String coreName, String shardName, CoreContainer coreContainer) throws SolrException {
     assertTrue(coreContainer.isZooKeeperAware());
 
     ZkController zkController = coreContainer.getZkController();
@@ -7,12 +6,11 @@ public static void syncLocalCoreWithSharedStore(String collectionName, String co
     DocCollection collection = zkController.getClusterState().getCollection(collectionName);
     CoreStorageClient blobClient = coreContainer.getSharedStoreManager().getBlobStorageProvider().getDefaultClient();
     log.info("sync intialized for collection=" + collectionName + " shard=" + shardName + " coreName=" + coreName);
-
-    CoreSyncStatus syncStatus = CoreSyncStatus.FAILURE;
-
+    
     Slice shard = collection.getSlicesMap().get(shardName);
     if (shard != null) {
       try {
+        sharedMetadataController.ensureMetadataNodeExists(collectionName, shardName);
         String sharedStoreName = (String)shard.get(ZkStateReader.SHARED_SHARD_NAME);
         // Fetch the latest metadata from ZK.
         // TODO: this can be optimized, depends on correct handling of leadership change.
