diff --git a/old_product.java b/new_product.java
index 9370b19..14e4a87 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,5 +4,6 @@
         removeBadMatches(dependency);
         removeWrongVersionMatches(dependency);
         removeSpuriousCPE(dependency);
+        removeDuplicativePOMEntries(dependency, engine);
         addFalseNegativeCPEs(dependency);
     }
\ No newline at end of file
