diff --git a/old_test.java b/new_test.java
index 3cc1a50..1db9be5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testNot() {
-        QueryCriterion criterion1 = new EqCriterion("name", "abc");
-        QueryCriterion criterion2 = new NotCriterion(criterion1);
+        QueryCriterion criterion1 = Criteria.eq("name", "abc");
+        QueryCriterion criterion2 = Criteria.not(criterion1);
         
         assertEquals(criterion2, instance.not(criterion1).getQueryCriterion());
     }
\ No newline at end of file
