diff --git a/old_product.java b/new_product.java
index 595a103..fc7185d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
 	@Transactional(readOnly = true)
 	public List<Concept> getDrugRoutes() {
-		return getSetMembersOfConceptSetFromGP(OpenmrsConstants.GP_DRUG_ROUTE_CONCEPT_UUID);
+		return getSetMembersOfConceptSetFromGP(OpenmrsConstants.GP_DRUG_ROUTES_CONCEPT_UUID);
 	}
\ No newline at end of file
