diff --git a/old_product.java b/new_product.java
index 59f05c3..31c796c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,15 +15,4 @@
 				}
 			}
 		}
-		//void all the orders associated with this patient
-		OrderService os = Context.getOrderService();
-		List<Order> orders = os.getAllOrdersByPatient(patient);
-		if (CollectionUtils.isNotEmpty(orders)) {
-			for (Order order : orders) {
-				if (!order.isVoided()) {
-					order.setDateVoided(patient.getDateVoided());
-					os.voidOrder(order, voidReason);
-				}
-			}
-		}
 	}
\ No newline at end of file
