diff --git a/old_product.java b/new_product.java
index 91969c1..3f52ec2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,7 +12,9 @@
             final JobStatus status,
             final String msg
     ) throws GenieException {
-        LOG.debug("Setting job with id " + id + " to status " + status + " for reason " + msg);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Setting job with id " + id + " to status " + status + " for reason " + msg);
+        }
         final Job job = this.jobRepo.findOne(id);
         if (job != null) {
             job.setJobStatus(status, msg);
