diff --git a/old_product.java b/new_product.java
index 51d7c4b..6db0ad9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -26,5 +26,5 @@ public static String getCamelCase(String yangIdentifier, YangToJavaNamingConflic
             }
             strArray = stringArrangement.toArray(new String[stringArrangement.size()]);
         }
-        return applyCamelCaseRule(strArray);
+        return upperCaseConflictResolver(strArray);
     }
\ No newline at end of file
