diff --git a/old_product.java b/new_product.java
index 344e4f7..3fcf08f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-RelationshipType findByTypesAndLabels(Context context, EntityType leftType,EntityType rightType,
+RelationshipType findbyTypesAndTypeName(Context context, EntityType leftType,EntityType rightType,
                                                                               String leftwardType,
                                                                               String rightwardType)
                                                                                     throws SQLException;
\ No newline at end of file
