diff --git a/old_test.java b/new_test.java
index 0e5e1c2..fcd6be4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testFindByTypesAndLabels() throws Exception {
         assertEquals("TestFindbyTypesAndLabels 0", relationshipType, relationshipTypeService
-                .findbyTypesAndLabels(context, entityTypeTwo, entityTypeOne, "isAuthorOfPublication",
+                .findbyTypesAndTypeName(context, entityTypeTwo, entityTypeOne, "isAuthorOfPublication",
                         "isPublicationOfAuthor"));
     }
\ No newline at end of file
