diff --git a/old_test.java b/new_test.java
index e33a9c4..e24e594 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
 	public void testConformance() {
 		final Conformance conf = client.fetchConformance().ofType(Conformance.class).execute();
-		assertEquals(conf.getRest().get(0).getResource().get(0).getType().toString(), "Patient");
+		assertEquals(conf.getRest().get(0).getResource().get(0).getType(), "Patient");
 	}
\ No newline at end of file
