diff --git a/old_test.java b/new_test.java
index a1966a3..f5256c4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
     public void testMergeProperties_String() throws IOException, URISyntaxException {
         String key = Settings.KEYS.PROXY_PORT;
-        String expResult = Settings.getString(key);
+        String expResult = getSettings().getString(key);
         File f = new File(this.getClass().getClassLoader().getResource("test.properties").toURI());
         //InputStream in = this.getClass().getClassLoader().getResourceAsStream("test.properties");
-        Settings.mergeProperties(f.getAbsolutePath());
-        String result = Settings.getString(key);
+        getSettings().mergeProperties(f.getAbsolutePath());
+        String result = getSettings().getString(key);
         Assert.assertTrue("setting didn't change?", (expResult == null && result != null) || !expResult.equals(result));
     }
\ No newline at end of file
