diff --git a/old_test.java b/new_test.java
index ed35dac..d4b6545 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-    public void testGetSummary() {
-        File images = new File("src/test/resources/images.zip");
-        Summary summary = service.getSummary(images);
-        Assert.assertNotNull(summary);
-    }
\ No newline at end of file
+  public void testGetSummary() {
+    final File images = new File("src/test/resources/images.zip");
+    final Summary summary = service.getSummary(images);
+    Assert.assertNotNull(summary);
+  }
\ No newline at end of file
