diff --git a/old_product.java b/new_product.java
index 95484ba..4d651b1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,19 +1,18 @@
-public static ArrayList<Container> getContainers(PackingPlan currentPackingPlan,
-                                                   int paddingPercentage) {
-    ArrayList<Container> containers = new ArrayList<>();
+static Map<Integer, Container> getContainers(PackingPlan currentPackingPlan,
+                                               int paddingPercentage) {
+    Map<Integer, Container> containers = new HashMap<>();
 
     //sort containers based on containerIds;
-    PackingPlan.ContainerPlan[] currentContainers =
+    PackingPlan.ContainerPlan[] currentContainerPlans =
         PackingUtils.sortOnContainerId(currentPackingPlan.getContainers());
 
     Resource capacity = currentPackingPlan.getMaxContainerResources();
-    for (int i = 0; i < currentContainers.length; i++) {
-      int containerId = PackingUtils.allocateNewContainer(
-          containers, capacity, paddingPercentage);
-      for (PackingPlan.InstancePlan instancePlan
-          : currentContainers[i].getInstances()) {
-        containers.get(containerId - 1).add(instancePlan);
+    for (PackingPlan.ContainerPlan currentContainerPlan : currentContainerPlans) {
+      Container container = new Container(capacity, paddingPercentage);
+      for (PackingPlan.InstancePlan instancePlan : currentContainerPlan.getInstances()) {
+        container.add(instancePlan);
       }
+      containers.put(currentContainerPlan.getId(), container);
     }
     return containers;
   }
\ No newline at end of file
