diff --git a/old_test.java b/new_test.java
index d7b93ec..0f35d61 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,7 @@
 @Test
     public void testJoinWhileActivate1_WithCache_Server() throws Exception {
+        if (MvccFeatureChecker.forcedMvcc())
+            fail("https://issues.apache.org/jira/browse/IGNITE-10583");
+
         joinWhileActivate1(false, true);
     }
\ No newline at end of file
