diff --git a/old_test.java b/new_test.java
index 6e54b07..60df33f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,12 +15,13 @@
 		assertThat(pathLocale.path, is(equalTo("some.min.ext")));
 		assertThat(pathLocale.locale, is(Locale.ENGLISH));
 
-		pathLocale = ResourceUtils.getLocaleFromFilename("some.min_fr_CA.ext");
+		pathLocale = ResourceUtils.getLocaleFromFilename("some_fr_CA.min.ext");
 		assertThat(pathLocale.path, is(equalTo("some.min.ext")));
 		assertThat(pathLocale.locale, is(Locale.CANADA_FRENCH));
 
 		String localeVariant = "blah";
-		pathLocale = ResourceUtils.getLocaleFromFilename("some.min_fr_CA_"+localeVariant+".ext");
+		pathLocale = ResourceUtils
+			.getLocaleFromFilename("some_fr_CA_" + localeVariant + ".min.ext");
 		assertThat(pathLocale.path, is(equalTo("some.min.ext")));
 		assertThat(pathLocale.locale.getLanguage(), is("fr"));
 		assertThat(pathLocale.locale.getCountry(), is("CA"));
