diff --git a/old_product.java b/new_product.java
index 07f0ae5..126e57b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,8 @@
 public static PathLocale getLocaleFromFilename(String path)
 	{
 		String extension = "";
-		int pos = path.lastIndexOf('.');
+
+		final int pos = path.lastIndexOf('.');
 		if (pos != -1)
 		{
 			extension = path.substring(pos);
@@ -9,12 +10,13 @@ public static PathLocale getLocaleFromFilename(String path)
 		}
 
 		String filename = Strings.lastPathComponent(path, '/');
-		Matcher matcher = LOCALE_PATTERN.matcher(filename);
+		Matcher matcher = LOCALE_MIN_PATTERN.matcher(filename);
 		if (matcher.find())
 		{
 			String language = matcher.group(1);
 			String country = matcher.group(3);
 			String variant = matcher.group(5);
+			String min = matcher.group(6);
 
 			// did we find a language?
 			if (language != null)
@@ -39,8 +41,9 @@ public static PathLocale getLocaleFromFilename(String path)
 
 			if (language != null)
 			{
-				pos = path.length() - filename.length() + matcher.start();
-				String basePath = path.substring(0, pos) + extension;
+				int languagePos = path.length() - filename.length() + matcher.start();
+				String basePath = path.substring(0, languagePos) + (min == null ? "" : min) +
+					extension;
 
 				Locale locale = new Locale(language, country != null ? country : "",
 					variant != null ? variant : "");
