diff --git a/old_test.java b/new_test.java
index ab10641..785bb2e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void testDequeue() throws IOException, ExecutorManagerException {
-    QueuedExecutions queue = new QueuedExecutions(5);
-    List<Pair<ExecutionReference, ExecutableFlow>> dataList = getDummyData();
+    final QueuedExecutions queue = new QueuedExecutions(5);
+    final List<Pair<ExecutionReference, ExecutableFlow>> dataList = getDummyData();
     queue.enqueueAll(dataList);
     queue.dequeue(dataList.get(0).getFirst().getExecId());
     Assert.assertEquals(queue.size(), 1);
