diff --git a/old_test.java b/new_test.java
index 9467a13..98bfb38 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     ExecutorService service =
         (ExecutorService) Whitebox.getInternalState(mFileSystemMaster, "mExecutorService");
     Future<?> ttlThread =
-        (Future<?>) Whitebox.getInternalState(mFileSystemMaster, "mTTLCheckerService");
+        (Future<?>) Whitebox.getInternalState(mFileSystemMaster, "mTtlCheckerService");
     Assert.assertFalse(ttlThread.isDone());
     Assert.assertFalse(service.isShutdown());
     mFileSystemMaster.stop();
