diff --git a/old_product.java b/new_product.java
index ca47f2b..e137fba 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public void stop() throws IOException {
+@Override
+  public void stop() throws IOException {
     mDataServer.close();
-    mThriftServer.stop();
-    mThriftServerSocket.close();
+
     mSessionCleanerThread.stop();
     mBlockMasterClient.close();
     if (mSpaceReserver != null) {
@@ -17,11 +17,8 @@ public void stop() throws IOException {
       LOG.error("Failed to stop web server", e);
     }
     mBlockDataManager.stop();
-    while (!mDataServer.isClosed() || mThriftServer.isServing()) {
-      // The reason to stop and close again is due to some issues in Thrift.
+    while (!mDataServer.isClosed()) {
       mDataServer.close();
-      mThriftServer.stop();
-      mThriftServerSocket.close();
       CommonUtils.sleepMs(100);
     }
   }
\ No newline at end of file
