diff --git a/old_test.java b/new_test.java
index 71a1cef..d2f6a09 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
     Assert.assertEquals(1, mBlockMaster.getBlockInfo(blockId).getLocations().size());
 
     // free the file
-    Assert.assertTrue(mFileSystemMaster.free(NESTED_FILE_URI, false));
+    Assert.assertTrue(mFileSystemMaster.free(NESTED_FILE_URI, false, true));
     // Update the heartbeat of removedBlockId received from worker 1
     Command heartbeat2 =
         mBlockMaster.workerHeartbeat(mWorkerId1, ImmutableMap.of("MEM", (long) Constants.KB),
