diff --git a/old_test.java b/new_test.java
index c739b88..99972a1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -21,7 +21,7 @@
             Assert.assertEquals(1, mapper.insert(userInfo));
 
             Assert.assertNotNull(userInfo.getId());
-            Assert.assertEquals(6, (int)userInfo.getId());
+            Assert.assertTrue((int) userInfo.getId() >= 6);
 
             Assert.assertEquals(1,mapper.deleteByPrimaryKey(userInfo));
         } finally {
