diff --git a/old_product.java b/new_product.java
index 20e50f3..fedf99b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,6 +7,9 @@ public String insert(MappedStatement ms) {
         Boolean hasIdentityKey = false;
         //先处理cache或bind节点
         for (EntityColumn column : columnList) {
+            if (!column.isInsertable()) {
+                continue;
+            }
             if (StringUtil.isNotEmpty(column.getSequenceName())) {
             } else if (column.isIdentity()) {
                 //这种情况下,如果原先的字段有值,需要先缓存起来,否则就一定会使用自动增长
@@ -33,6 +36,9 @@ public String insert(MappedStatement ms) {
         sql.append(SqlHelper.insertColumns(entityClass, false, false, false));
         sql.append("<trim prefix=\"VALUES(\" suffix=\")\" suffixOverrides=\",\">");
         for (EntityColumn column : columnList) {
+            if (!column.isInsertable()) {
+                continue;
+            }
             //优先使用传入的属性值,当原属性property!=null时，用原属性
             //自增的情况下,如果默认有值,就会备份到property_cache中,所以这里需要先判断备份的值是否存在
             if (column.isIdentity()) {
