diff --git a/old_test.java b/new_test.java
index 8562a95..7bb2e32 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,17 +20,17 @@
     Date start = new Date();
 
     CreateValueOptions createOptions = new CreateValueOptions.Builder()
-        .workspaceId(workspaceId)
-        .entity(entity)
-        .value(entityValue)
-        .metadata(valueMetadata)
-        .build();
+            .workspaceId(workspaceId)
+            .entity(entity)
+            .value(entityValue)
+            .metadata(valueMetadata)
+            .build();
     Value response = service.createValue(createOptions).execute();
 
     try {
       assertNotNull(response);
-      assertNotNull(response.getValue());
-      assertEquals(response.getValue(), entityValue);
+      assertNotNull(response.getValueText());
+      assertEquals(response.getValueText(), entityValue);
       assertNotNull(response.getCreated());
       assertNotNull(response.getUpdated());
       assertNotNull(response.getMetadata());
@@ -50,7 +50,7 @@
     } finally {
       // Clean up
       DeleteValueOptions deleteOptions =
-          new DeleteValueOptions.Builder(workspaceId, entity, entityValue).build();
+              new DeleteValueOptions.Builder(workspaceId, entity, entityValue).build();
       service.deleteValue(deleteOptions).execute();
     }
   }
\ No newline at end of file
