diff --git a/old_product.java b/new_product.java
index 2c5bce3..be67b37 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public ServiceCall<Value> createValue(CreateValueOptions createValueOptions) {
     Validator.notNull(createValueOptions, "createValueOptions cannot be null");
     RequestBuilder builder = RequestBuilder.post(String.format("/v1/workspaces/%s/entities/%s/values",
-        createValueOptions.workspaceId(), createValueOptions.entity()));
+            createValueOptions.workspaceId(), createValueOptions.entity()));
     builder.query(VERSION, versionDate);
     final JsonObject contentJson = new JsonObject();
     if (createValueOptions.metadata() != null) {
@@ -10,6 +10,9 @@ public ServiceCall<Value> createValue(CreateValueOptions createValueOptions) {
     if (createValueOptions.synonyms() != null) {
       contentJson.add("synonyms", GsonSingleton.getGson().toJsonTree(createValueOptions.synonyms()));
     }
+    if (createValueOptions.patterns() != null) {
+      contentJson.add("patterns", GsonSingleton.getGson().toJsonTree(createValueOptions.patterns()));
+    }
     contentJson.addProperty("value", createValueOptions.value());
     builder.bodyJson(contentJson);
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(Value.class));
