diff --git a/old_product.java b/new_product.java
index 7441e74..3b4cd64 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public static double logChoose(int n, int k) {
+public static double lchoose(int n, int k) {
         if (n < 0 || k < 0 || k > n) {
             throw new IllegalArgumentException(String.format("Invalid n = %d, k = %d", n, k));
         }
 
-        return logFactorial(n) - logFactorial(k) - logFactorial(n - k);
+        return lfactorial(n) - lfactorial(k) - lfactorial(n - k);
     }
\ No newline at end of file
