diff --git a/old_product.java b/new_product.java
index 5291976..6351d63 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,6 @@
-public List<CohortMembership> getMemberships(Date asOf) {
-		return getMembers().stream()
-				.filter(m -> !m.getStartDate().before(asOf)).collect(Collectors.toList());
+public CohortMembership getActiveMembership(Patient patient) {
+		return getMemberships().stream()
+				.filter(m -> m.isActive() && m.getPatientId().equals(patient.getPatientId()))
+				.collect(Collectors.toList())
+				.get(0);
 	}
\ No newline at end of file
