diff --git a/old_product.java b/new_product.java
index 1b26e1e..08ca699 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public boolean onAcquireFail(Throwable t, AcquireFailConfig acquireConfig) {
 		boolean tryAgain = false;
-		String log = acquireConfig.getLogMessage() == null ? "" : acquireConfig.getLogMessage();
+		String log = acquireConfig.getLogMessage();
 		logger.error(log+" Sleeping for "+acquireConfig.getAcquireRetryDelay()+"ms and trying again. Attempts left: "+acquireConfig.getAcquireRetryAttempts()+". Exception: "+t.getCause());
 
 		try {
@@ -8,9 +8,9 @@ public boolean onAcquireFail(Throwable t, AcquireFailConfig acquireConfig) {
 			if (acquireConfig.getAcquireRetryAttempts().get() > 0){
 				tryAgain = (acquireConfig.getAcquireRetryAttempts().decrementAndGet()) > 0;
 			}
-		} catch (InterruptedException e) {
+		} catch (Exception e) {
 			tryAgain=false;
 		}
-
-		return tryAgain; 
+ 
+		return tryAgain;  
 	}
\ No newline at end of file
