diff --git a/old_test.java b/new_test.java
index 2f7c119..10a0875 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,7 @@
         assertEquals("test",CompressedHelper.getFileName("test.rar"));
         assertEquals("test",CompressedHelper.getFileName("test.tar"));
         assertEquals("test",CompressedHelper.getFileName("test.tar.gz"));
+        assertEquals("test",CompressedHelper.getFileName("test.tgz"));
         assertEquals("test",CompressedHelper.getFileName("test.jar"));
         assertEquals("test",CompressedHelper.getFileName("test.apk"));
 
