diff --git a/old_test.java b/new_test.java
index 8579873..8eab134 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testEncrypt() {
+    public void testEncryptTLS13() throws NoSuchAlgorithmException {
         context.setSelectedProtocolVersion(ProtocolVersion.TLS13);
         context.setSelectedCipherSuite(CipherSuite.TLS_AES_128_GCM_SHA256);
         context.setEncryptActive(true);
@@ -11,7 +11,7 @@
         record.setCleanProtocolMessageBytes(ArrayConverter.hexStringToByteArray("080000020000"));
         record.setContentMessageType(ProtocolMessageType.HANDSHAKE);
         record.setPaddingLength(0);
-        recordCipher = new RecordAEADCipher(context);
+        recordCipher = new RecordAEADCipher(context, KeySetGenerator.generateKeySet(context));
         encryptor = new RecordEncryptor(recordCipher, context);
         encryptor.encrypt(record);
         assertTrue(record.getProtocolMessageBytes().getValue().length == 23);
