diff --git a/old_product.java b/new_product.java
index a2683f3..41aeec3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,19 @@
 @Override
-    public byte[] encrypt(byte[] data) throws CryptoException {
+    public EncryptionResult encrypt(EncryptionRequest request) throws CryptoException {
         try {
             byte[] ciphertext;
             if (useExplicitIv) {
-                // Note: here we always use the same IV that was generated from
-                // the master secret
-                ciphertext = ArrayConverter.concatenate(encryptIv.getIV(), encryptCipher.doFinal(data));
+                encryptIv = new IvParameterSpec(request.getInitialisationVector());
+                encryptCipher.init(Cipher.ENCRYPT_MODE, new SecretKeySpec(getKeySet().getKey(context.getTalkingConnectionEndType()),bulkCipherAlg.getJavaName()),encryptIv);
+                ciphertext = encryptCipher.doFinal(request.getPlainText());
+                return new EncryptionResult(encryptIv.getIV(), ciphertext, useExplicitIv);
+
             } else {
-                ciphertext = encryptCipher.update(data);
+                byte[] iv = encryptCipher.getIV();
+                ciphertext = encryptCipher.update(request.getPlainText());
+                return new EncryptionResult(ciphertext, iv, false);
             }
-            return ciphertext;
-        } catch (BadPaddingException | IllegalBlockSizeException ex) {
+        } catch (InvalidKeyException | InvalidAlgorithmParameterException | BadPaddingException | IllegalBlockSizeException ex) {
             throw new CryptoException(ex);
         }
     }
\ No newline at end of file
