diff --git a/old_test.java b/new_test.java
index 9dac05a..709c721 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testGt() {
-        assertTrue(instance.gt("id", 5).getQueryCriterions()
-                .contains(new GtCriterion("id", 5)));
+    	assertEquals(new GtCriterion("id", 5), instance.gt("id", 5).getQueryCriterion());
     }
\ No newline at end of file
