diff --git a/old_test.java b/new_test.java
index edb64dd..ac30263 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testRemove() throws Exception {
-        MetricRegistry mreg = newMetricRegistry();
+        MetricRegistry mreg = new MetricRegistry("group", null);
 
         LongMetricImpl cntr = mreg.metric("my.name", null);
         LongMetricImpl cntr2 = mreg.metric("my.name.x", null);
