diff --git a/old_product.java b/new_product.java
index fcb3479..f54129e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public VM kFoldCrossValidation(Dataframe dataset, int k, String dbName, DatabaseConfiguration dbConf, Class<? extends AbstractAlgorithm> aClass, TP trainingParameters) {
+public VM kFoldCrossValidation(Dataframe dataset, int k, String dbName, DatabaseConfiguration dbConf, Class<? extends AbstractModeler> aClass, TP trainingParameters) {
         int n = dataset.size();
         if(k<=0 || n<=k) {
             throw new IllegalArgumentException("Invalid number of folds.");
@@ -16,7 +16,7 @@ public VM kFoldCrossValidation(Dataframe dataset, int k, String dbName, Database
         }
         PHPMethods.shuffle(ids);
         
-        AbstractAlgorithm<MP, TP, VM> mlmodel;
+        AbstractModeler<MP, TP, VM> mlmodel;
         
         String foldDBname=dbName+dbConf.getDBnameSeparator()+DB_INDICATOR;
         
@@ -54,7 +54,7 @@ public VM kFoldCrossValidation(Dataframe dataset, int k, String dbName, Database
             
             
             //initialize mlmodel
-            mlmodel = AbstractAlgorithm.newInstance(aClass, foldDBname+(fold+1), dbConf);
+            mlmodel = Trainable.newInstance(aClass, foldDBname+(fold+1), dbConf);
             
             
             Dataframe trainingData = dataset.getSubset(foldTrainingIds);
