diff --git a/old_test.java b/new_test.java
index 2ad09e2..5fcbae2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,7 +17,7 @@
         GaussianDPMM.TrainingParameters param = new GaussianDPMM.TrainingParameters();
         param.setAlpha(0.01);
         param.setMaxIterations(100);
-        param.setInitializationMethod(AbstractDPMM.TrainingParameters.Initialization.ONE_CLUSTER_PER_RECORD);
+        param.setInitializationMethod(AbstractDPMM.AbstractTrainingParameters.Initialization.ONE_CLUSTER_PER_RECORD);
         param.setKappa0(0);
         param.setNu0(1);
         param.setMu0(new double[]{0.0, 0.0});
