diff --git a/old_test.java b/new_test.java
index de348e4..eb973ee 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void writeTextLines() {
-    error(_FILE_WRITE_TEXT_LINES.args(PATH, "x"), Err.FILE_ID);
+    error(_FILE_WRITE_TEXT_LINES.args(PATH, "x"), Err.FILE_IS_DIR);
     error(_FILE_WRITE_TEXT_LINES.args(PATH1, " 123"), Err.INVCAST);
 
     query(_FILE_WRITE_TEXT_LINES.args(PATH1, "x"));
