diff --git a/old_test.java b/new_test.java
index 6c301df..5487357 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,34 +1,23 @@
-@TestTargetNew(
-        level = TestLevel.COMPLETE,
-        notes = "",
-        method = "read",
-        args = {}
-    )
-    public void test_read() throws IOException {
-        int result = 0;
-        int buffer[] = new int[500];
-        byte orgBuffer[] = {1, 3, 4, 7, 8};
-        InputStream infile = Support_Resources.getStream("hyts_constru_OD.txt");
-        Inflater inflate = new Inflater();
-        InflaterInputStream inflatIP = new InflaterInputStream(infile, inflate);
+public void test_read() throws IOException {
+		int result = 0;
+		int buffer[] = new int[500];
+		byte orgBuffer[] = { 1, 3, 4, 7, 8 };
+		InputStream infile = Support_Resources
+				.getStream("hyts_constru_OD.txt"); // android-changed
+		Inflater inflate = new Inflater();
+		InflaterInputStream inflatIP = new InflaterInputStream(infile,
+				inflate);
 
-        int i = 0;
-        while ((result = inflatIP.read()) != -1) {
-            buffer[i] = result;
-            i++;
-        }
-        inflatIP.close();
+		int i = 0;
+		while ((result = inflatIP.read()) != -1) {
+			buffer[i] = result;
+			i++;
+		}
+		inflatIP.close();
 
-        for (int j = 0; j < orgBuffer.length; j++) {
-            assertTrue(
-                    "original compressed data did not equal decompressed data",
-                    buffer[j] == orgBuffer[j]);
-        }
-        inflatIP.close();
-        try {
-            inflatIP.read();
-            fail("IOException expected");
-        } catch (IOException ee) {
-            // expected.
-        }
-    }
\ No newline at end of file
+		for (int j = 0; j < orgBuffer.length; j++) {
+			assertTrue(
+				"original compressed data did not equal decompressed data",
+				buffer[j] == orgBuffer[j]);
+		}
+	}
\ No newline at end of file
