diff --git a/old_test.java b/new_test.java
index e76918c..d07307f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,5 +4,5 @@
 		List<Location> locations = new ArrayList<Location>();
 		locations.add(new Location(1));
 		Assert.assertEquals(1, Context.getVisitService().getVisits(null, null, locations, null, null, null, null, null,
-		    null, false).size());
+		    null, true, false).size());
 	}
\ No newline at end of file
