diff --git a/old_product.java b/new_product.java
index 02efe69..ca05ac2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,5 @@
 	@Authorized(PrivilegeConstants.VIEW_VISITS)
 	public List<Visit> getVisits(Collection<VisitType> visitTypes, Collection<Patient> patients,
 	        Collection<Location> locations, Collection<Concept> indications, Date minStartDatetime, Date maxStartDatetime,
-	        Date minEndDatetime, Date maxEndDatetime, Map<VisitAttributeType, Object> attributeValues, boolean includeVoided)
-	        throws APIException;
\ No newline at end of file
+	        Date minEndDatetime, Date maxEndDatetime, Map<VisitAttributeType, Object> attributeValues,
+	        boolean includeInactive, boolean includeVoided) throws APIException;
\ No newline at end of file
