diff --git a/old_test.java b/new_test.java
index 77100d6..61c9405 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,7 +10,7 @@
 
 		this.influxDB.write(dbName, TimeUnit.MILLISECONDS, serie);
 
-		List<Serie> result = this.influxDB.Query(dbName, "select value2 from testSeries", TimeUnit.MILLISECONDS);
+		List<Serie> result = this.influxDB.query(dbName, "select value2 from testSeries", TimeUnit.MILLISECONDS);
 		Assert.assertNotNull(result);
 		Assert.assertEquals(result.size(), 1);
 		// [{"name":"testSeries","columns":["time","sequence_number","value"],"points":[[1398412802823,160001,5]]}]
