diff --git a/old_product.java b/new_product.java
index e6ba5c0..08d6ffa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,18 +2,15 @@ public static void export(final Prop prop, final Data data,
       final String target) throws IOException {
 
     final SerializerProp sp = new SerializerProp(prop.get(Prop.EXPORTER));
-    final IO root = IO.get(target);
-    if(!(root instanceof IOFile))
-      throw new IOException(Util.info(DBNOTEXPORTED, target));
-
-    if(!root.exists()) root.md();
+    final IOFile root = new IOFile(target);
+    root.md();
 
     final HashSet<String> exported = new HashSet<String>();
     for(final int pre : data.doc()) {
       // create file path
       final IO file = root.merge(Token.string(data.text(pre, true)));
       // create dir if necessary
-      final IO dir = IO.get(file.dir());
+      final IOFile dir = new IOFile(file.dir());
       if(!dir.exists()) dir.md();
 
       // attach counter to duplicate file names
