diff --git a/old_test.java b/new_test.java
index ccc377b..66b17d8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,4 @@
-@TestTargetNew(
-        level = TestLevel.COMPLETE,
-        notes = "",
-        method = "getManifest",
-        args = {}
-    )
-    public void test_getManifest() {
+public void test_getManifest() {
         // Test for method java.util.jar.Manifest
         // java.util.jar.JarFile.getManifest()
         try {
