diff --git a/old_product.java b/new_product.java
index b96c4b0..beb3802 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,8 +5,8 @@ public static synchronized Data create(final String name, final Parser parser,
     if(!ctx.user.has(Perm.CREATE)) throw new BaseXException(PERM_REQUIRED_X, Perm.CREATE);
 
     // create main memory database instance
-    final Options opts = ctx.options;
-    if(opts.is(Options.MAINMEM)) return MemBuilder.build(name, parser);
+    final MainOptions opts = ctx.options;
+    if(opts.is(MainOptions.MAINMEM)) return MemBuilder.build(name, parser);
 
     // database is currently locked by another process
     if(ctx.pinned(name)) throw new BaseXException(DB_PINNED_X, name);
