diff --git a/old_test.java b/new_test.java
index bcb3574..2f9c0d3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	public void testGetOne() {
-		Map<String, Object> result = userApiController.getOne(getTestUser().getUserId());
+		Map<String, Object> result = userApiController.getOneDetail(getTestUser().getUserId());
 		User user = (User) result.get("user");
 		assertThat(user.getId(), is(getTestUser().getId()));
 	}
\ No newline at end of file
