diff --git a/old_product.java b/new_product.java
index 06cc0f9..50a203a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,14 +1,6 @@
-@GetMapping("/{userId}/detail")
+@RestAPI
 	@PreAuthorize("hasAnyRole('A')")
-	public Map<String, Object> getOne(@PathVariable final String userId) {
-		User one = userService.getOneWithFollowers(userId);
-		Map<String, Object> model = buildMap(
-			"user", one,
-			"allowPasswordChange", true,
-			"allowRoleChange", true,
-			"roleSet", EnumSet.allOf(Role.class),
-			"showPasswordByDefault", false
-		);
-		attachCommonAttribute(one, model);
-		return model;
+	@RequestMapping(value = "/{userId}", method = RequestMethod.GET)
+	public HttpEntity<String> getOne(@PathVariable("userId") String userId) {
+		return toJsonHttpEntity(userService.getOne(userId));
 	}
\ No newline at end of file
