diff --git a/old_product.java b/new_product.java
index 6e397e0..344236d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,7 +12,7 @@
 		
 		// set the number of samples
 		this.numLineSamples = DEFAULT_NUM_SAMPLES;
-		String samplesString = parameters.getString(NUM_STATISTICS_SAMPLES, null);
+		final String samplesString = parameters.getString(NUM_STATISTICS_SAMPLES, null);
 		
 		if (samplesString != null) {
 			try {
