diff --git a/old_test.java b/new_test.java
index 2d6d49b..f3e32b5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
         ESMetricsAccessor metrics = new ESMetricsAccessor();
         metrics.setEsClient(client);
 
-        UsageHistogramBean usage = metrics.getUsage("JBossOverlord", "s-ramp-api", "1.0", UsageHistogramIntervalType.day,
+        UsageHistogramBean usage = metrics.getUsage("JBossOverlord", "s-ramp-api", "1.0", HistogramIntervalType.day,
                 parseDate("2015-01-01"), DateTime.now().withZone(DateTimeZone.UTC));
         List<UsageDataPoint> data = usage.getData();
         Assert.assertTrue(data.size() > 0);
@@ -12,7 +12,7 @@
         Assert.assertEquals(46L, usage.getData().get(169).getCount());
 
 
-        usage = metrics.getUsage("JBossOverlord", "s-ramp-api", "1.0", UsageHistogramIntervalType.hour,
+        usage = metrics.getUsage("JBossOverlord", "s-ramp-api", "1.0", HistogramIntervalType.hour,
                 parseDate("2015-06-15"), parseDate("2015-06-22"));
         data = usage.getData();
         Assert.assertEquals(168, data.size());
@@ -20,7 +20,7 @@
         Assert.assertEquals(46L, usage.getData().get(111).getCount());
 
 
-        usage = metrics.getUsage("JBossOverlord", "s-ramp-api", "1.0", UsageHistogramIntervalType.minute,
+        usage = metrics.getUsage("JBossOverlord", "s-ramp-api", "1.0", HistogramIntervalType.minute,
                 parseDate("2015-06-19"), parseDate("2015-06-20"));
         data = usage.getData();
         Assert.assertEquals(1440, data.size());
@@ -30,5 +30,4 @@
         Assert.assertEquals(14L, usage.getData().get(913).getCount());
         Assert.assertEquals(15L, usage.getData().get(914).getCount());
         Assert.assertEquals(17L, usage.getData().get(915).getCount());
-
     }
\ No newline at end of file
