diff --git a/old_product.java b/new_product.java
index ccd6093..d4ca692 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @Override
     public UsageHistogramBean getUsage(String organizationId, String serviceId, String version,
-            UsageHistogramIntervalType interval, DateTime from, DateTime to) {
+            HistogramIntervalType interval, DateTime from, DateTime to) {
         UsageHistogramBean rval = new UsageHistogramBean();
         return rval;
     }
\ No newline at end of file
