diff --git a/old_product.java b/new_product.java
index 85bfa44..d232ae5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,7 +15,7 @@
             List<LayerInfo> coverages = new ArrayList<LayerInfo>();
 
             for (String encodedCoverageId : (List<String>)request.getCoverageId()) {
-                LayerInfo layer = getCoverage(encodedCoverageId);
+                LayerInfo layer = NCNameResourceCodec.getCoverage(catalog, encodedCoverageId);
                 if(layer != null) {
                     coverages.add(layer);
                 } else {
