diff --git a/old_product.java b/new_product.java
index ce90323..b4137c7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,12 +2,14 @@ static RESTExec get(final RESTSession rs) throws IOException {
     RESTCmd.parseOptions(rs);
 
     final HTTPContext http = rs.http;
-    if(http.depth() == 0) throw HTTPCode.NO_PATH.get();
+    final String db = http.db();
+    if(db.isEmpty()) throw HTTPCode.NO_PATH.get();
 
     // open database to ensure it exists
-    rs.add(new Open(http.db()));
-    if(http.depth() == 1) rs.add(new DropDB(http.db()));
-    else rs.add(new Delete(http.dbpath()));
+    rs.add(new Open(db));
+    final String path = http.dbpath();
+    if(path.isEmpty()) rs.add(new DropDB(db));
+    else rs.add(new Delete(path));
 
     return new RESTExec(rs);
   }
\ No newline at end of file
