diff --git a/old_test.java b/new_test.java
index 69cda43..5a89959 100644
--- a/old_test.java
+++ b/new_test.java
@@ -24,7 +24,7 @@
     query(_DB_OPEN.args(NAME) + "/root()", "<dummy/>");
 
     // try to create DB twice during same query
-    error(_DB_CREATE.args(NAME) + ',' + _DB_CREATE.args(NAME), Err.BXDB_ONCE);
+    error(_DB_CREATE.args(NAME) + ',' + _DB_CREATE.args(NAME), Err.BXDB_ONCE_X_X);
 
     // create DB from file
     query(_DB_CREATE.args(NAME, FILE, "in/"));
@@ -38,19 +38,19 @@
     query(_DB_CREATE.args(NAME, "(<a/>,<b/>)", "('1.xml','2.xml')"));
     query(_DB_CREATE.args(NAME, "(<a/>,'" + FILE + "')", "('1.xml','2.xml')"));
 
-    error(_DB_CREATE.args(NAME, "()", "1.xml"), Err.BXDB_CREATEARGS);
-    error(_DB_CREATE.args(NAME, "(<a/>,<b/>)", "1.xml"), Err.BXDB_CREATEARGS);
+    error(_DB_CREATE.args(NAME, "()", "1.xml"), Err.BXDB_CREATEARGS_X_X);
+    error(_DB_CREATE.args(NAME, "(<a/>,<b/>)", "1.xml"), Err.BXDB_CREATEARGS_X_X);
 
     // create and drop more than one database
     query("for $i in 1 to 5 return " + _DB_CREATE.args(" '" + NAME + "' || $i"));
     query("for $i in 1 to 5 return " + _DB_DROP.args(" '" + NAME + "' || $i"));
 
     // create DB with initial EMPTY content
-    error(_DB_CREATE.args(""), Err.BXDB_NAME);
+    error(_DB_CREATE.args(""), Err.BXDB_NAME_X);
 
     // try to access non-existing DB
     query(_DB_DROP.args(NAME));
-    error(_DB_CREATE.args(NAME) + ',' + _DB_DROP.args(NAME), Err.BXDB_WHICH);
+    error(_DB_CREATE.args(NAME) + ',' + _DB_DROP.args(NAME), Err.BXDB_WHICH_X);
 
     // run update on existing DB then drop it and create a new one
     query(_DB_CREATE.args(NAME, "<a/>", "a.xml"));
@@ -93,8 +93,8 @@
     query(_DB_OPEN.args(NAME), "<a> </a>");
 
     // specify unknown or invalid options
-    error(_DB_CREATE.args(NAME, "()", "()", " map { 'xyz':'abc' }"), Err.BASX_OPTIONS);
-    error(_DB_CREATE.args(NAME, "()", "()", " map { 'maxlen':-1 }"), Err.BASX_VALUE);
-    error(_DB_CREATE.args(NAME, "()", "()", " map { 'maxlen':'a' }"), Err.BASX_VALUE);
-    error(_DB_CREATE.args(NAME, "()", "()", " map { 'textindex':'nope' }"), Err.BASX_VALUE);
+    error(_DB_CREATE.args(NAME, "()", "()", " map { 'xyz':'abc' }"), Err.BASX_OPTIONS_X);
+    error(_DB_CREATE.args(NAME, "()", "()", " map { 'maxlen':-1 }"), Err.BASX_VALUE_X_X);
+    error(_DB_CREATE.args(NAME, "()", "()", " map { 'maxlen':'a' }"), Err.BASX_VALUE_X_X);
+    error(_DB_CREATE.args(NAME, "()", "()", " map { 'textindex':'nope' }"), Err.BASX_VALUE_X_X);
   }
\ No newline at end of file
