diff --git a/old_product.java b/new_product.java
index c6ca1a8..b0e06de 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,11 +6,11 @@ private Data create(final QueryInput input, final boolean single, final IO baseI
 
     // do not check input if no read permissions are given
     if(!qc.context.user.has(Perm.READ))
-      throw BXXQ_PERM.get(info, Util.info(Text.PERM_REQUIRED_X, Perm.READ));
+      throw BXXQ_PERM_X.get(info, Util.info(Text.PERM_REQUIRED_X, Perm.READ));
 
     // check if input is an existing file
     final IO source = checkPath(input, baseIO, info);
-    if(single && source.isDir()) WHICHRES.get(info, baseIO);
+    if(single && source.isDir()) WHICHRES_X.get(info, baseIO);
 
     // overwrite parsing options with default values
     final MainOptions opts = context.options;
@@ -27,7 +27,7 @@ private Data create(final QueryInput input, final boolean single, final IO baseI
       final boolean fc = context.options.get(MainOptions.FORCECREATE);
       return addData(CreateDB.create(source.dbname(), new DirParser(source, opts), context, !fc));
     } catch(final IOException ex) {
-      throw IOERR.get(info, ex);
+      throw IOERR_X.get(info, ex);
     } finally {
       // reset original values
       for(int o = 0; o < ol; o++) opts.put(options[o], values[o]);
