diff --git a/old_product.java b/new_product.java
index 7e7931a..d837cac 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,8 +7,13 @@
 
     // HDFS needs to actually prevent client retry since that causes them to try too long and not failover quickly.
     if (services.containsKey("HDFS")) {
-      String clientRetryPolicyEnabled = getProperty(request, "hdfs-site", "dfs.client.retry.policy.enabled");
+      String clientRetryPolicyEnabled = getProperty(request, "hdfs-site", HDFS_CLIENT_RETRY_PROPERTY);
       if (null != clientRetryPolicyEnabled && Boolean.parseBoolean(clientRetryPolicyEnabled)) {
+        MissingClientRetryProperty missingProperty = new MissingClientRetryProperty("HDFS",
+            "hdfs-site", HDFS_CLIENT_RETRY_PROPERTY);
+
+        prerequisiteCheck.getFailedDetail().add(missingProperty);
+
         errorMessages.add(getFailReason(HDFS_CLIENT_RETRY_DISABLED_KEY, prerequisiteCheck, request));
         prerequisiteCheck.getFailedOn().add("HDFS");
       }
@@ -16,8 +21,13 @@
 
     // check hive client properties
     if (services.containsKey("HIVE")) {
-      String hiveClientRetryCount = getProperty(request, "hive-site", "hive.metastore.failure.retries");
+      String hiveClientRetryCount = getProperty(request, "hive-site", HIVE_CLIENT_RETRY_PROPERTY);
       if (null != hiveClientRetryCount && Integer.parseInt(hiveClientRetryCount) <= 0) {
+        MissingClientRetryProperty missingProperty = new MissingClientRetryProperty("HIVE",
+            "hive-site", HIVE_CLIENT_RETRY_PROPERTY);
+
+        prerequisiteCheck.getFailedDetail().add(missingProperty);
+
         errorMessages.add(getFailReason(HIVE_CLIENT_RETRY_MISSING_KEY, prerequisiteCheck, request));
         prerequisiteCheck.getFailedOn().add("HIVE");
       }
@@ -25,7 +35,12 @@
 
     if (services.containsKey("OOZIE")) {
       String oozieClientRetry = getProperty(request, "oozie-env", "content");
-      if (null == oozieClientRetry || !oozieClientRetry.contains("-Doozie.connection.retry.count")) {
+      if (null == oozieClientRetry || !oozieClientRetry.contains(OOZIE_CLIENT_RETRY_PROPERTY)) {
+        MissingClientRetryProperty missingProperty = new MissingClientRetryProperty("OOZIE",
+            "oozie-env", OOZIE_CLIENT_RETRY_PROPERTY);
+
+        prerequisiteCheck.getFailedDetail().add(missingProperty);
+
         errorMessages.add(getFailReason(OOZIE_CLIENT_RETRY_MISSING_KEY, prerequisiteCheck, request));
         prerequisiteCheck.getFailedOn().add("OOZIE");
       }
