diff --git a/old_product.java b/new_product.java
index cad0a1a..f764d98 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ final B64 hash(final String algo, final QueryContext qc) throws QueryException {
     final Item it = exprs[0].atomItem(qc, info);
     try {
       final MessageDigest md = MessageDigest.getInstance(algo);
-      if(it instanceof B64Stream) {
+      if(it instanceof B64Lazy) {
         try(BufferInput bi = it.input(info)) {
           final byte[] tmp = new byte[IO.BLOCKSIZE];
           do {
@@ -19,6 +19,6 @@ final B64 hash(final String algo, final QueryContext qc) throws QueryException {
       return B64.get(md.digest(toBytes(it)));
     } catch(final NoSuchAlgorithmException ex) {
       Util.debug(ex);
-      throw HASH_ALG_X.get(info, algo);
+      throw HASH_ALGORITHM_X.get(info, algo);
     }
   }
\ No newline at end of file
