diff --git a/old_test.java b/new_test.java
index a5bb45f..a34e25d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,15 +1,10 @@
 @Test
-	@Ignore
 	public void testTerminateAllConnections2() throws SQLException {
 		Connection mockRealConnection = EasyMock.createNiceMock(Connection.class);
 
 		// same test but to cover the finally section
 		reset(mockRealConnection, mockConnectionsScheduler, mockKeepAliveScheduler, mockPartition, mockConnectionHandles, mockConnection);
-		expect(mockConnectionHandles.poll()).andReturn(mockConnection).anyTimes();
-		expect(mockConnection.getInternalConnection()).andReturn(mockRealConnection).anyTimes();
-		mockConnection.internalClose();
-		expectLastCall().once().andThrow(new RuntimeException()).once();
-		expect(mockPartition.getFreeConnections()).andReturn(mockConnectionHandles).anyTimes();
+		expect(mockPartition.getFreeConnections()).andThrow(new RuntimeException()).once();
 		expect(mockConnection.getOriginatingPartition()).andReturn(mockPartition).anyTimes();
 		replay(mockRealConnection, mockConnectionsScheduler, mockKeepAliveScheduler, mockPartition, mockConnectionHandles, mockConnection);
 
