diff --git a/old_test.java b/new_test.java
index fb06230..67f3320 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,7 +17,7 @@
         this.service.createCluster(testCluster);
 
         // Shouldn't delete any clusters as all are UP or OOS
-        Assert.assertThat(this.service.deleteTerminatedClusters(), Matchers.is(0));
+        Assert.assertThat(this.service.deleteTerminatedClusters(), Matchers.is(0L));
 
         // Change status to UP
         String patchString = "[{ \"op\": \"replace\", \"path\": \"/status\", \"value\": \"UP\" }]";
@@ -26,7 +26,7 @@
         Assert.assertThat(this.service.getCluster(testClusterId).getStatus(), Matchers.is(ClusterStatus.UP));
 
         // All clusters are UP/OOS or attached to jobs
-        Assert.assertThat(this.service.deleteTerminatedClusters(), Matchers.is(0));
+        Assert.assertThat(this.service.deleteTerminatedClusters(), Matchers.is(0L));
 
         // Change status to terminated
         patchString = "[{ \"op\": \"replace\", \"path\": \"/status\", \"value\": \"TERMINATED\" }]";
@@ -35,7 +35,7 @@
         Assert.assertThat(this.service.getCluster(testClusterId).getStatus(), Matchers.is(ClusterStatus.TERMINATED));
 
         // All clusters are UP/OOS or attached to jobs
-        Assert.assertThat(this.service.deleteTerminatedClusters(), Matchers.is(1));
+        Assert.assertThat(this.service.deleteTerminatedClusters(), Matchers.is(1L));
 
         // Make sure it didn't delete any of the clusters we wanted
         Assert.assertTrue(this.clusterRepository.existsByUniqueId(CLUSTER_1_ID));
