diff --git a/old_product.java b/new_product.java
index cd7ce19..ca94e44 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,2 @@
-@Modifying
-    @Query(value = DELETE_TERMINATED_CLUSTERS_SQL, nativeQuery = true)
-    int deleteTerminatedClusters();
\ No newline at end of file
+@Query(value = FIND_TERMINATED_CLUSTERS_SQL, nativeQuery = true)
+    Set<Number> findTerminatedUnusedClusters();
\ No newline at end of file
