diff --git a/old_product.java b/new_product.java
index ce07ac0..9230b96 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,9 +2,8 @@
 	public void writeRecord(T record) throws IOException, InterruptedException {
 
 		// Get a write buffer from the broker
-		if (this.uncompressedDataBuffer == null) {
-
-			requestWriteBuffersFromBroker();
+		if (this.dataBuffer == null) {
+			requestWriteBufferFromBroker();
 		}
 
 		if (this.closeRequested) {
@@ -16,22 +15,22 @@
 		if (this.compressor != null) {
 			while (this.serializationBuffer.dataLeftFromPreviousSerialization()) {
 
-				this.serializationBuffer.read(this.uncompressedDataBuffer);
-				if (this.uncompressedDataBuffer.remaining() == 0) {
+				this.serializationBuffer.read(this.dataBuffer);
+				if (this.dataBuffer.remaining() == 0) {
 
-					this.compressor.compress();
+					this.dataBuffer = this.compressor.compress(this.dataBuffer);
 					// this.leasedWriteBuffer.flip();
-					releaseWriteBuffers();
-					requestWriteBuffersFromBroker();
+					releaseWriteBuffer();
+					requestWriteBufferFromBroker();
 				}
 			}
 		} else {
 			while (this.serializationBuffer.dataLeftFromPreviousSerialization()) {
 
-				this.serializationBuffer.read(this.uncompressedDataBuffer);
-				if (this.uncompressedDataBuffer.remaining() == 0) {
-					releaseWriteBuffers();
-					requestWriteBuffersFromBroker();
+				this.serializationBuffer.read(this.dataBuffer);
+				if (this.dataBuffer.remaining() == 0) {
+					releaseWriteBuffer();
+					requestWriteBufferFromBroker();
 				}
 			}
 		}
@@ -43,17 +42,17 @@
 		this.serializationBuffer.serialize(record);
 
 		if (this.compressor != null) {
-			this.serializationBuffer.read(this.uncompressedDataBuffer);
+			this.serializationBuffer.read(this.dataBuffer);
 
-			if (this.uncompressedDataBuffer.remaining() == 0) {
-				this.compressor.compress();
+			if (this.dataBuffer.remaining() == 0) {
+				this.dataBuffer = this.compressor.compress(this.dataBuffer);
 				// this.leasedWriteBuffer.flip();
-				releaseWriteBuffers();
+				releaseWriteBuffer();
 			}
 		} else {
-			this.serializationBuffer.read(this.uncompressedDataBuffer);
-			if (this.uncompressedDataBuffer.remaining() == 0) {
-				releaseWriteBuffers();
+			this.serializationBuffer.read(this.dataBuffer);
+			if (this.dataBuffer.remaining() == 0) {
+				releaseWriteBuffer();
 			}
 		}
 	}
\ No newline at end of file
