diff --git a/old_test.java b/new_test.java
index c163e87..a1b6992 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,6 @@
 		final StringRecord record = new StringRecord("abc");
 		final Decompressor decompressorMock = mock(Decompressor.class);
 		this.uncompressedDataBuffer = mock(Buffer.class);
-		BufferPairResponse bufferPair = new BufferPairResponse(this.uncompressedDataBuffer, this.uncompressedDataBuffer);
 		// BufferPairResponse bufferPair = mock(BufferPairResponse.class);
 		// when(bufferPair.getUncompressedDataBuffer()).thenReturn(this.uncompressedDataBuffer,
 		// this.uncompressedDataBuffer, this.uncompressedDataBuffer,null);
@@ -20,7 +19,7 @@
 		@SuppressWarnings("unchecked")
 		final OutputGate<StringRecord> outGate = mock(OutputGate.class);
 		final ByteBufferedOutputChannelBroker outputBroker = mock(ByteBufferedOutputChannelBroker.class);
-		when(outputBroker.requestEmptyWriteBuffers()).thenReturn(bufferPair);
+		when(outputBroker.requestEmptyWriteBuffer()).thenReturn(this.uncompressedDataBuffer);
 
 		when(outputBroker.hasDataLeftToTransmit()).thenReturn(true);
 
