diff --git a/old_product.java b/new_product.java
index 0a6507d..6aee5ef 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,17 +1,12 @@
-public InodeTree.CreatePathResult mkdir(AlluxioURI path, CreateDirectoryOptions options)
+public InodeTree.CreatePathResult mkdir(AlluxioURI path, CreatePathOptions options)
       throws InvalidPathException, FileAlreadyExistsException, IOException, AccessControlException {
     LOG.debug("mkdir {} ", path);
     MasterContext.getMasterSource().incCreateDirectoriesOps(1);
+    options.setDirectory(true);
     synchronized (mInodeTree) {
       try {
         checkPermission(FileSystemAction.WRITE, path, true);
-        CreatePathOptions createPathOptions =
-            CreatePathOptions.defaults().setAllowExists(options.isAllowExists()).setDirectory(true)
-                .setPersisted(options.isPersisted()).setRecursive(options.isRecursive())
-                .setOperationTimeMs(options.getOperationTimeMs())
-                .setPermissionStatus(PermissionStatus.get(MasterContext.getConf(), true))
-                .setMountPoint(mMountTable.isMountPoint(path));
-        InodeTree.CreatePathResult createResult = mInodeTree.createPath(path, createPathOptions);
+        InodeTree.CreatePathResult createResult = mInodeTree.createPath(path, options);
 
         LOG.debug("writing journal entry for mkdir {}", path);
         writeJournalEntry(mDirectoryIdGenerator.toJournalEntry());
