diff --git a/old_test.java b/new_test.java
index b619703..3cc6d92 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,26 @@
 @Test
-	public void testAssignToNode(){
-		//[id=2, name=分行经理审批]
-		getJBPMApplication().assignToNode(1l, 2l);
-		testQueryTodoList();
+	public void testAssignToNode() {
+		long i = getJBPMApplication().startProcess("defaultPackage.Trade",
+				"aaa", null);
+		
+		Map<String, Object> data = new HashMap<String, Object>();
+		data.put("approveStatus", "1");
+		getJBPMApplication().completeTask(i, 1l, "fhjl",
+				XmlParseUtil.paramsToXml(data), null);
+		
+		// [id=2, name=分行经理审批]
+		getJBPMApplication().assignToNode(i, 2l);
+		
+		List<TaskVO> tasks = getJBPMApplication().queryTodoList("fhjl");
+		Assert.assertTrue(tasks.size() > 0);
+		
+		data = new HashMap<String, Object>();
+		data.put("approveStatus", "1");
+		getJBPMApplication().completeTask(i, tasks.get(0).getTaskId(), "fhjl",
+				XmlParseUtil.paramsToXml(data), null);
+		
+		tasks = getJBPMApplication().queryTodoList("fwzy");
+		Assert.assertTrue(tasks.size() > 0);
+//		
+		getJBPMApplication().removeProcessInstance(i);
 	}
\ No newline at end of file
