diff --git a/old_test.java b/new_test.java
index 7be8af1..00b084d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,10 @@
-@Test void check() {
-    server.enqueue(HEALTH_RESPONSE);
+@Test void check_fail_onNoContent() {
+    storage.indexTemplates = mock(IndexTemplates.class); // assume index templates called before
 
-    assertThat(storage.check()).isEqualTo(CheckResult.OK);
+    server.enqueue(SUCCESS_RESPONSE); // empty instead of success response
+
+    CheckResult result = storage.check();
+    assertThat(result.ok()).isFalse();
+    assertThat(result.error().getMessage())
+      .isEqualTo("No content reading cluster health");
   }
\ No newline at end of file
