diff --git a/old_test.java b/new_test.java
index 78034ee..9b035eb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
   public void testIsFull() throws IOException, ExecutorManagerException,
-    InterruptedException {
-    QueuedExecutions queue = new QueuedExecutions(2);
-    List<Pair<ExecutionReference, ExecutableFlow>> dataList = getDummyData();
+      InterruptedException {
+    final QueuedExecutions queue = new QueuedExecutions(2);
+    final List<Pair<ExecutionReference, ExecutableFlow>> dataList = getDummyData();
     queue.enqueueAll(dataList);
     Assert.assertTrue(queue.isFull());
   }
\ No newline at end of file
