diff --git a/old_product.java b/new_product.java
index c9bdd09..8ee6dc5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,7 +16,9 @@ public void handle(OpenmrsObject openmrsObject, User creator, Date dateCreated,
 			try {
 				Object value = PropertyUtils.getProperty(openmrsObject, field.getName());
 				if (value != null && value.toString().isEmpty()) {
-					PropertyUtils.setProperty(openmrsObject, field.getName(), null);
+					//TODO Commenting this out until we resolve the cases where some non
+					//nullable fields have emptry string holders.
+					//PropertyUtils.setProperty(openmrsObject, field.getName(), null);
 				}
 			}
 			catch (Exception ex) {
