diff --git a/old_test.java b/new_test.java
index 2e6f1e4..2e61027 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,7 +14,7 @@
 					try {
 						Context.openSession();
 						Context.addProxyPrivilege(PrivilegeConstants.ADD_ORDERS);
-						uniqueOrderNumbers.add(((OrderNumberGenerator) Context.getOrderService()).getNewOrderNumber());
+						uniqueOrderNumbers.add(((OrderNumberGenerator) Context.getOrderService()).getNewOrderNumber(null));
 					}
 					finally {
 						Context.removeProxyPrivilege(PrivilegeConstants.ADD_ORDERS);
