diff --git a/old_product.java b/new_product.java
index f002c2f..df686ba 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,7 +13,9 @@
             @NotBlank(message = "No app name entered. unable to update app info for job.")
             final String appName
     ) throws GenieException {
-        LOG.debug("Setting the application info for job with id " + id);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Setting the application info for job with id " + id);
+        }
         final Job job = this.jobRepo.findOne(id);
         if (job != null) {
             job.setApplicationId(appId);
