diff --git a/old_test.java b/new_test.java
index a23c91c..0eb6af8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
 
     CircularBuffer<Integer> instance = new CircularBuffer<Integer>(0);
     try {
-      instance.get();
+      instance.remove();
       Assert.fail("exception should be raised for getting from buffer which does not have data");
     }
     catch (Exception bue) {
@@ -14,7 +14,7 @@
 
     instance = new CircularBuffer<Integer>(10);
     try {
-      instance.get();
+      instance.remove();
       Assert.fail("exception should be raised for getting from buffer which does not have data");
     }
     catch (Exception bue) {
@@ -25,12 +25,12 @@
       instance.add(i);
     }
 
-    Integer i = instance.get();
-    Integer j = instance.get();
+    Integer i = instance.remove();
+    Integer j = instance.remove();
     assert (i == 0 && j == 1);
 
     instance.add(10);
 
     assert (instance.size() == 9);
-    assert (instance.get() == 2);
+    assert (instance.remove() == 2);
   }
\ No newline at end of file
