diff --git a/old_product.java b/new_product.java
index 9c71156..f34b5fd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static byte[] expandLabel(String macAlgorithm, byte[] prk, String labelIn, byte[] hashValue, int outLen) {
+public static byte[] expandLabel(HKDFAlgorithm hkdfAlgortihm, byte[] prk, String labelIn, byte[] hashValue, int outLen) {
         byte[] info = labelEncoder(hashValue, labelIn, outLen);
-        return expand(macAlgorithm, prk, info, outLen);
+        return expand(hkdfAlgortihm, prk, info, outLen);
     }
\ No newline at end of file
