diff --git a/old_product.java b/new_product.java
index 4b44716..4a4932c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,6 +14,8 @@ private Item write(final File path, final QueryContext ctx)
         Item it;
         while((it = ir.next()) != null) it.serialize(xml);
         xml.close();
+      } catch(final SerializerException ex) {
+        throw new QueryException(input, ex);
       } finally {
         out.close();
       }
