diff --git a/old_test.java b/new_test.java
index d3e4b91..fa94ecb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,10 +17,11 @@
     query("file:delete('" + PATH1 + "')");
 
     query(fun + "('" + PATH1 + "', 'a\u00e4'," +
-      "<encoding>CP1252</encoding>)");
+      serialParams("<encoding>CP1252</encoding>") + ")");
     query("file:read('" + PATH1 + "', 'CP1252')", "a\u00e4");
 
-    query(fun + "('" + PATH1 + "', '<a/>'," + "<method>text</method>)");
+    query(fun + "('" + PATH1 + "', '<a/>'," +
+        serialParams("<method>text</method>") + ")");
     query("file:read('" + PATH1 + "')", "&amp;lt;a/&amp;gt;");
     query("file:delete('" + PATH1 + "')");
   }
\ No newline at end of file
