diff --git a/old_test.java b/new_test.java
index 5e3276b..9979e7e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,15 +9,15 @@
         try {
             ArrayUtils.toMap(new String[][] {{"foo", "bar"}, {"short"}});
             fail("exception expected");
-        } catch (IllegalArgumentException ex) {}
+        } catch (final IllegalArgumentException ex) {}
         try {
             ArrayUtils.toMap(new Object[] {new Object[] {"foo", "bar"}, "illegal type"});
             fail("exception expected");
-        } catch (IllegalArgumentException ex) {}
+        } catch (final IllegalArgumentException ex) {}
         try {
             ArrayUtils.toMap(new Object[] {new Object[] {"foo", "bar"}, null});
             fail("exception expected");
-        } catch (IllegalArgumentException ex) {}
+        } catch (final IllegalArgumentException ex) {}
         
         map = ArrayUtils.toMap(new Object[] {new Map.Entry<Object, Object>() {
             @Override
