diff --git a/old_test.java b/new_test.java
index bba9e41..3e75de0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,11 @@
 @Test
-    public void testGetFormatDescription()
-    {
+    public void testGetFormatDescription() throws SQLException {
         //format is unknown by default
         String format = "Unknown";
-        assertThat("testGetFormatDescription 0", bs.getFormatDescription(),
+        assertThat("testGetFormatDescription 0", bs.getFormatDescription(context),
                 notNullValue());
-        assertThat("testGetFormatDescription 1", bs.getFormatDescription(),
+        assertThat("testGetFormatDescription 1", bs.getFormatDescription(context),
                 not(equalTo("")));
-        assertThat("testGetFormatDescription 2", bs.getFormatDescription(),
+        assertThat("testGetFormatDescription 2", bs.getFormatDescription(context),
                 equalTo(format));
     }
\ No newline at end of file
