diff --git a/old_product.java b/new_product.java
index f6fa56a..ce97f52 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,18 +1,4 @@
-public String getFormatDescription()
+public String getFormatDescription(Context context) throws SQLException
     {
-        if (bitstreamFormat.getShortDescription().equals("Unknown"))
-        {
-            // Get user description if there is one
-            String desc = getUserFormatDescription();
-
-            if (desc == null)
-            {
-                return "Unknown";
-            }
-
-            return desc;
-        }
-
-        // not null or Unknown
-        return bitstreamFormat.getShortDescription();
+        return getBitstreamService().getFormatDescription(context, this);
     }
\ No newline at end of file
