diff --git a/old_product.java b/new_product.java
index d7d9b91..1aa0d0e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,12 @@
 protected final FElem parse(final QueryContext qc, final byte[] query, final String path)
       throws QueryException {
 
-    boolean compile = false, plan = true;
+    boolean compile = false, plan = true, pass = false;
     if(exprs.length > 1) {
       final Options opts = toOptions(1, Q_OPTIONS, new XQueryOptions(), qc);
       compile = opts.get(XQueryOptions.COMPILE);
       plan = opts.get(XQueryOptions.PLAN);
+      pass = opts.get(XQueryOptions.PASS);
     }
 
     try(final QueryContext qctx = new QueryContext(qc.context)) {
@@ -27,6 +28,7 @@ protected final FElem parse(final QueryContext qc, final byte[] query, final Str
       if(plan) root.add(qctx.plan());
       return root;
     } catch(final QueryException ex) {
-      throw ex.info(info);
+      if(!pass) ex.info(info);
+      throw ex;
     }
   }
\ No newline at end of file
