diff --git a/old_test.java b/new_test.java
index ce6aa7a..64d7b4e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
-@Ignore
+@Ignore 
     @Test
     public void testDeploy() {
         DeploymentFacility df = DeploymentFacilityFactory.getDeploymentFacility();
@@ -6,13 +6,12 @@
         sci.setHostName("localhost");
         sci.setHostPort(4848); // 8080 for the REST client
         sci.setUserName("admin");
-        sci.setPassword("adminadmin");
+        sci.setPassword("");
         
         df.connect(sci);
         
-//        File archive = new File("C:\\tim\\asgroup\\dev-9p-fcs\\glassfish\\appserv-tests\\devtests\\deployment\\build\\servletonly.war");
-        File archive = new File("/Users/Tim/asgroup/v3/warWithPlan/servletonly-portable.war");
-        File plan = new File("/Users/Tim/asgroup/v3/warWithPlan/servletonly-deployplan.jar");
+        File archive = new File("/home/hzhang/deployment/apps/jsr88/servletonly-portable.war");
+        File plan = new File("/home/hzhang/deployment/apps/jsr88/servletonly-deployplan.jar");
         DFDeploymentProperties options = new DFDeploymentProperties();
         options.setForce(true);
         options.setUpload(true);
@@ -21,8 +20,11 @@
         props.setProperty("keepSessions", "true");
         props.setProperty("foo", "bar");
         options.setProperties(props);
+
+        try {
+        Target[] targets = df.listTargets(); 
         DFProgressObject prog = df.deploy(
-                new Target[0] /* ==> deploy to the default target */, 
+                targets /* ==> deploy to the default target */, 
                 archive.toURI(), 
                 plan.toURI(),
                 options);
@@ -31,5 +33,9 @@
         if (ds.getStatus() == DFDeploymentStatus.Status.FAILURE) {
             fail(ds.getAllStageMessages());
         }
+        } catch (Exception e) {
+          e.printStackTrace();
+          fail(e.getMessage());
+        }
         
     }
\ No newline at end of file
