diff --git a/old_product.java b/new_product.java
index d678ffc..929a8a9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public void read(final File poifsFile) throws IOException {
-        try (NPOIFSFileSystem poifs = new NPOIFSFileSystem(poifsFile, true)) {
+        try (POIFSFileSystem poifs = new POIFSFileSystem(poifsFile, true)) {
             read(poifs);
         }
     }
\ No newline at end of file
