diff --git a/old_test.java b/new_test.java
index b731997..463ea23 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,11 +2,8 @@
     public void read() throws IOException, NoPropertySetStreamException, MarkUnsupportedException {
         /* Read the POI filesystem's property set streams: */
         for (POIFile pf : Util.readPropertySets(file)) {
-            final InputStream in = new ByteArrayInputStream(pf.getBytes());
-            try {
+            try (InputStream in = new ByteArrayInputStream(pf.getBytes())) {
                 PropertySetFactory.create(in);
-            } finally {
-                in.close();
             }
         }
     }
\ No newline at end of file
