diff --git a/old_product.java b/new_product.java
index 54d98d9..cad78db 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,8 +2,8 @@ public void rename(AlluxioURI srcPath, AlluxioURI dstPath) throws FileAlreadyExi
       FileDoesNotExistException, InvalidPathException, IOException, AccessControlException {
     MasterContext.getMasterSource().incRenamePathOps(1);
     synchronized (mInodeTree) {
-      checkPermission(FileSystemAction.WRITE, srcPath, true);
-      checkPermission(FileSystemAction.WRITE, dstPath, true);
+      mPermissionChecker.checkParentPermission(FileSystemAction.WRITE, srcPath);
+      mPermissionChecker.checkParentPermission(FileSystemAction.WRITE, dstPath);
       mMountTable.checkUnderWritableMountPoint(srcPath);
       mMountTable.checkUnderWritableMountPoint(dstPath);
       Inode<?> srcInode = mInodeTree.getInodeByPath(srcPath);
