diff --git a/old_product.java b/new_product.java
index a917118..6ed3d1b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 static StreamManager create(URI controller, PravegaCredentials credentials) {
-        boolean enableTls = Boolean.parseBoolean(System.getProperty("io.pravega.auth.enabled"));
+        boolean enableTls = Boolean.parseBoolean(System.getProperty("io.pravega.tls.enabled"));
         String tlsCertFile = System.getProperty("io.pravega.auth.certfile");
         return new StreamManagerImpl(controller, credentials, enableTls, tlsCertFile);
     }
\ No newline at end of file
