diff --git a/old_test.java b/new_test.java
index 8221207..8e0783b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
         String streamName = "Stream";
         int numSegments = 10;
 
-        System.setProperty("io.pravega.auth.enabled", String.valueOf(authEnabled));
+        System.setProperty("io.pravega.tls.enabled", String.valueOf(tlsEnabled));
         System.setProperty("io.pravega.auth.certFile",  "../config/cert.pem");
 
         @Cleanup
