diff --git a/old_test.java b/new_test.java
index 7a6d4c0..d78e082 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
 	public void testReplace() throws Exception {
 		// Setup
-		LongStateHandleProvider stateHandleProvider = new LongStateHandleProvider();
+		LongStateStorage stateHandleProvider = new LongStateStorage();
 
 		ZooKeeperStateHandleStore<Long> store = new ZooKeeperStateHandleStore<>(
 				ZooKeeper.getClient(), stateHandleProvider);
