diff --git a/old_product.java b/new_product.java
index 1250683..54bdac3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@ public void replace(String pathInZooKeeper, int expectedVersion, T state) throws
 
 		StateHandle<T> oldStateHandle = get(pathInZooKeeper);
 
-		StateHandle<T> stateHandle = stateHandleProvider.createStateHandle(state);
+		StateHandle<T> stateHandle = storage.store(state);
 
 		boolean success = false;
 
