diff --git a/old_product.java b/new_product.java
index 2ce8277..6148025 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 @PUT
-   @Produces(VCloudDirectorMediaType.VAPP)
-   @Consumes(VCloudDirectorMediaType.TASK)
+   @Produces(VAPP)
+   @Consumes(TASK)
    @JAXBResponseParser
    @ExceptionParser(ThrowVCloudErrorOn4xx.class)
-   ListenableFuture<Task> modifyVApp(@EndpointParam URI vAppURI, @BinderParam(BindToXMLPayload.class) VApp vApp);
\ No newline at end of file
+   ListenableFuture<Task> modifyVApp(@EndpointParam URI vAppURI,
+                                     @BinderParam(BindToXMLPayload.class) VApp vApp);
\ No newline at end of file
