diff --git a/old_test.java b/new_test.java
index 0d055e0..721c20b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,15 @@
 @Test(testName = "PUT /vApp/{id}", dependsOnMethods = { "testGetVApp" })
    public void testModifyVApp() {
-      vApp.setName("new-name");
-      vApp.setDescription("New Description");
+      VApp newVApp = VApp.builder()
+		      .name("new-name")
+		      .description("New Description")
+		      .build();
 
       // The method under test
-      Task modifyVApp = vAppClient.modifyVApp(vApp.getHref(), vApp);
+      Task modifyVApp = vAppClient.modifyVApp(vApp.getHref(), newVApp);
       assertTrue(retryTaskSuccess.apply(modifyVApp), String.format(TASK_COMPLETE_TIMELY, "modifyVApp"));
 
       vApp = vAppClient.getVApp(vApp.getHref());
-      assertEquals(vApp.getName(), "new-name", String.format(OBJ_FIELD_EQ, VAPP, "name", "new-name", "modified"));
-      assertEquals(vApp.getDescription(), "New Description", String.format(OBJ_FIELD_EQ, VAPP, "description", "New Description", "modified"));
+      assertEquals(vApp.getName(), newVApp.getName(), String.format(OBJ_FIELD_EQ, VAPP, "Name", newVApp.getName(), vApp.getName()));
+      assertEquals(vApp.getDescription(), newVApp.getDescription(), String.format(OBJ_FIELD_EQ, VAPP, "Description", newVApp.getDescription(), vApp.getDescription()));
    }
\ No newline at end of file
