diff --git a/old_product.java b/new_product.java
index 8c04a4c..4f8aa33 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,16 +13,14 @@ public RunTable purge (Predicate<Run> predicate,
         }
 
         for (int i = 0, size = getSize(); i < size; i++) {
-            RunSequence seq = getSequence(i);
-
-            for (Iterator<Run> it = seq.iterator(); it.hasNext();) {
+            for (Itr it = new Itr(i); it.hasNext();) {
                 Run run = it.next();
 
                 if (predicate.check(run)) {
                     it.remove();
 
                     if (removed != null) {
-                        removed.getSequence(i).add(run);
+                        removed.addRun(i, run);
                     }
                 }
             }
