diff --git a/old_test.java b/new_test.java
index 9b3279e..5ff1798 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testPurge_Predicate_RunTable ()
     {
-        System.out.println("purge_predicate_removed");
+        System.out.println("\n+++ purge_predicate_removed");
 
         Predicate<Run> predicate1 = new Predicate<Run>()
         {
@@ -13,16 +13,16 @@
         };
 
         RunTable instance = createHorizontalInstance();
-        instance.dumpSequences();
         System.out.println("table before1:" + instance.dumpOf());
+        instance.dumpSequences();
 
         RunTable removed = new RunTable("purged", HORIZONTAL, dim.width, dim.height);
         instance.purge(predicate1, removed);
-        instance.dumpSequences();
         System.out.println("table after1:" + instance.dumpOf());
+        instance.dumpSequences();
 
-        removed.dumpSequences();
         System.out.println("purge after1:" + removed.dumpOf());
+        removed.dumpSequences();
 
         Predicate<Run> predicate2 = new Predicate<Run>()
         {
@@ -34,9 +34,9 @@
         };
 
         instance.purge(predicate2, removed);
-        instance.dumpSequences();
         System.out.println("table after2:" + instance.dumpOf());
+        instance.dumpSequences();
 
-        removed.dumpSequences();
         System.out.println("purge after2:" + removed.dumpOf());
+        removed.dumpSequences();
     }
\ No newline at end of file
