diff --git a/old_test.java b/new_test.java
index 98dec54..a876459 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,17 +18,17 @@
 
 		PerspectiveOps.splitAssociated(list,list1,list2,list3);
 
-		DenseMatrix64F expected1 = new DenseMatrix64F(3,3);
-		DenseMatrix64F expected2 = new DenseMatrix64F(3,3);
-		DenseMatrix64F expected3 = new DenseMatrix64F(3,3);
+		RowMatrix_F64 expected1 = new RowMatrix_F64(3,3);
+		RowMatrix_F64 expected2 = new RowMatrix_F64(3,3);
+		RowMatrix_F64 expected3 = new RowMatrix_F64(3,3);
 
 		LowLevelMultiViewOps.computeNormalization(list1, expected1);
 		LowLevelMultiViewOps.computeNormalization(list2, expected2);
 		LowLevelMultiViewOps.computeNormalization(list3, expected3);
 
-		DenseMatrix64F found1 = new DenseMatrix64F(3,3);
-		DenseMatrix64F found2 = new DenseMatrix64F(3,3);
-		DenseMatrix64F found3 = new DenseMatrix64F(3,3);
+		RowMatrix_F64 found1 = new RowMatrix_F64(3,3);
+		RowMatrix_F64 found2 = new RowMatrix_F64(3,3);
+		RowMatrix_F64 found3 = new RowMatrix_F64(3,3);
 
 		LowLevelMultiViewOps.computeNormalization(list, found1, found2, found3);
 
