diff --git a/old_product.java b/new_product.java
index 6b5fb0b..ddb533a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public TemporalAccessor parse(CharSequence text, ParsePosition position) {
         Objects.requireNonNull(text, "text");
         Objects.requireNonNull(position, "position");
         try {
-            return parseToBuilder(text, position).resolve();
+            return parseToBuilder(text, position).resolve(resolverStyle);
         } catch (DateTimeParseException | IndexOutOfBoundsException ex) {
             throw ex;
         } catch (RuntimeException ex) {
