diff --git a/old_test.java b/new_test.java
index 0bb0922..80f4a6c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,9 +12,11 @@
 				public void run() {
 					try {
 						Context.openSession();
+						Context.addProxyPrivilege(PrivilegeConstants.ADD_ORDERS);
 						uniqueOrderNumbers.add(((OrderNumberGenerator) Context.getOrderService()).getNewOrderNumber());
 					}
 					finally {
+						Context.removeProxyPrivilege(PrivilegeConstants.ADD_ORDERS);
 						Context.closeSession();
 					}
 				}
