diff --git a/old_test.java b/new_test.java
index 03b8a19..5ccaaac 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
     public void testSetLicense() throws SQLException {
         String license = "license for test";
-        c.setLicense(license);
-        assertThat("testSetLicense 0", c.getLicense(), notNullValue());
-        assertThat("testSetLicense 1", c.getLicense(), equalTo(license));
-        assertThat("testSetLicense 2", c.getLicenseCollection(), notNullValue());
-        assertThat("testSetLicense 3", c.getLicenseCollection(), equalTo(license));
+        collection.setLicense(context, license);
+        assertThat("testSetLicense 0", collectionService.getLicense(collection), notNullValue());
+        assertThat("testSetLicense 1", collectionService.getLicense(collection), equalTo(license));
+        assertThat("testSetLicense 2", collection.getLicenseCollection(), notNullValue());
+        assertThat("testSetLicense 3", collection.getLicenseCollection(), equalTo(license));
     }
\ No newline at end of file
