diff --git a/old_product.java b/new_product.java
index 12cd212..49facdc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public void setLicense(String license) {
-        setMetadata("license",license);
+public void setLicense(Context context, String license) throws SQLException {
+        getCollectionService().setMetadata(context, this, "license", license);
     }
\ No newline at end of file
