diff --git a/old_test.java b/new_test.java
index bef9619..e206394 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
   public void testGetTrimmedWords() {
     a.setWords(new long[] {-1L, 0L}, 64);
-    assertTrue("Array not trimmed correctly", a.getTrimmedWords().length == 1);
+    assertTrue("Array not trimmed correctly", a.toArray().length == 1);
   }
\ No newline at end of file
