diff --git a/old_product.java b/new_product.java
index f266b8a..ce2f7dc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,9 @@
-public long[] getTrimmedWords() {
+public long[] toArray() {
     // find the last index of a word which is different from 0:
-    int i;
-    for(i = words.length - 1; i >= 0; i--) if(words[i] != 0) break;
+    int i = words.length;
+    while(--i >= 0 && words[i] == 0);
 
     final long[] result = new long[++i];
     System.arraycopy(words, 0, result, 0, i);
-
     return result;
   }
\ No newline at end of file
