diff --git a/old_test.java b/new_test.java
index e7d9344..fe8f227 100644
--- a/old_test.java
+++ b/new_test.java
@@ -50,7 +50,6 @@
             public void run() {
                 // Manually call refresh, so the did_change will be triggered.
                 foregroundRealm.sharedRealm.refresh();
-                foregroundRealm.setAutoRefresh(true);
             }
         });
     }
\ No newline at end of file
