diff --git a/old_product.java b/new_product.java
index 4a14a09..66091d6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static Instant now(TimeSource timeSource) {
-        MathUtils.checkNotNull(timeSource, "TimeSource must not be null");
-        return of(timeSource.instant());
+public static Instant now(Clock clock) {
+        MathUtils.checkNotNull(clock, "Clock must not be null");
+        return clock.instant();
     }
\ No newline at end of file
