diff --git a/old_test.java b/new_test.java
index 90762f8..0036e71 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test(groups={"tck"})
     public void now() {
-        Instant expected = Instant.now(TimeSource.system());
+        Instant expected = Instant.now(Clock.systemUTC());
         Instant test = Instant.now();
         BigInteger diff = test.toEpochNano().subtract(expected.toEpochNano()).abs();
         if (diff.compareTo(BigInteger.valueOf(100000000)) >= 0) {
             // may be date change
-            expected = Instant.now(TimeSource.system());
+            expected = Instant.now(Clock.systemUTC());
             test = Instant.now();
             diff = test.toEpochNano().subtract(expected.toEpochNano()).abs();
         }
