diff --git a/old_test.java b/new_test.java
index 1ca20ac..7507852 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,9 @@
 @Test
 	public void testGetTotalFree() {
 		expect(mockPartition.getFreeConnections()).andReturn(mockConnectionHandles).anyTimes();
-		expect(mockConnectionHandles.size()).andReturn(3).anyTimes();
+		expect(mockPartition.getAvailableConnections()).andReturn(new AtomicInteger(3)).anyTimes();
+		
+		// expect(mockConnectionHandles.size()).andReturn(3).anyTimes();
 		replay(mockPartition, mockConnectionHandles);
 		assertEquals(6, testClass.getTotalFree());
 		verify(mockPartition, mockConnectionHandles);
