diff --git a/old_test.java b/new_test.java
index 2367106..3aee424 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,7 +15,7 @@
         for(int i = 0; i < 4; i++)
             meta.add(new FeatureMeta("", i, false));
         RandomForestRegressionTrainer trainer = new RandomForestRegressionTrainer(meta)
-            .withCountOfTrees(5)
+            .withAmountOfTrees(5)
             .withFeaturesCountSelectionStrgy(x -> 2);
 
         ModelsComposition mdl = trainer.fit(sample, parts, (k, v) -> VectorUtils.of(v), (k, v) -> k);
