diff --git a/old_product.java b/new_product.java
index 9317827..8cdb402 100644
--- a/old_product.java
+++ b/new_product.java
@@ -34,12 +34,12 @@
                     if (b == null)
                         return a;
 
-                    double[] result = new double[a.length];
+                    double[] res = new double[a.length];
 
-                    for (int i = 0; i < result.length; i++) {
-                        result[i] = Math.max(Math.abs(a[i]), Math.abs(b[i]));
-                    }
-                    return result;
+                    for (int i = 0; i < res.length; i++)
+                        res[i] = Math.max(Math.abs(a[i]), Math.abs(b[i]));
+
+                    return res;
                 });
             return new MaxAbsScalerPreprocessor<>(maxAbs, basePreprocessor);
         }
