diff --git a/old_test.java b/new_test.java
index 5292b92..109b522 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
-  public void applyExtraOtherBitsTest() {
+  public void setOtherBitsTest() {
     Mode mode = new Mode((short) 0640);
-    mode.applyExtraOtherBits(Mode.Bits.READ);
-    Assert.assertEquals(Mode.Bits.READ, mode.getOtherBits());
-    mode.applyExtraOtherBits(Mode.Bits.EXECUTE);
+    mode.setOtherBits(Mode.Bits.READ_EXECUTE);
     Assert.assertEquals(Mode.Bits.READ_EXECUTE, mode.getOtherBits());
-    mode.applyExtraOtherBits(Mode.Bits.WRITE_EXECUTE);
+    mode.setOtherBits(Mode.Bits.WRITE);
+    Assert.assertEquals(Mode.Bits.WRITE, mode.getOtherBits());
+    mode.setOtherBits(Mode.Bits.ALL);
     Assert.assertEquals(Mode.Bits.ALL, mode.getOtherBits());
   }
\ No newline at end of file
