diff --git a/old_product.java b/new_product.java
index 4b28cce..6388b51 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @SuppressWarnings("FutureReturnValueIgnored")
-    public void logout() {
+    public void logOut() {
         // Acquire lock to prevent users creating new instances
         synchronized (Realm.class) {
             if (!SyncManager.getUserStore().isActive(identity, authenticationUrl.toString())) {
