diff --git a/old_test.java b/new_test.java
index 65cf3e6..a2b508f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void logout_sameSyncUserMultipleSessions() {
         String uniqueName = UUID.randomUUID().toString();
         SyncCredentials credentials = SyncCredentials.usernamePassword(uniqueName, "password", true);
-        SyncUser user =  SyncUser.login(credentials, Constants.AUTH_URL);
+        SyncUser user =  SyncUser.logIn(credentials, Constants.AUTH_URL);
 
         SyncConfiguration syncConfiguration1 = configFactory
                 .createSyncConfigurationBuilder(user, Constants.SYNC_SERVER_URL)
@@ -28,13 +28,13 @@
 
         assertEquals(session1.getUser(), session2.getUser());
 
-        user.logout();
+        user.logOut();
 
         assertEquals(SyncSession.State.INACTIVE, session1.getState());
         assertEquals(SyncSession.State.INACTIVE, session2.getState());
 
         credentials = SyncCredentials.usernamePassword(uniqueName, "password", false);
-        SyncUser.login(credentials, Constants.AUTH_URL);
+        SyncUser.logIn(credentials, Constants.AUTH_URL);
 
         // reviving the sessions. The state could be changed concurrently.
         assertTrue(session1.getState() == SyncSession.State.WAITING_FOR_ACCESS_TOKEN ||
