diff --git a/old_test.java b/new_test.java
index b3ee627..d59bf9e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
-    public void testParseSuppressionRules() throws Exception {
+    public void testParseSuppressionRulesV1_0() throws Exception {
         //File file = new File(this.getClass().getClassLoader().getResource("suppressions.xml").getPath());
         File file = BaseTest.getResourceAsFile(this, "suppressions.xml");
         SuppressionParser instance = new SuppressionParser();
         List<SuppressionRule> result = instance.parseSuppressionRules(file);
-        assertTrue(result.size() > 3);
+        Assert.assertEquals(5, result.size());
     }
\ No newline at end of file
