diff --git a/old_product.java b/new_product.java
index 6cad76f..a27266d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,15 +1,15 @@
-public Scores getDocumentRelationScores(Document document,String corpusName,List<String> concepts) {
-        ConceptInsightsId.validateGenarate(document, getAccountId(), corpusName);
-        Validate.notNull(concepts, "concepts can't be null");
+public Scores getDocumentRelationScores(final Document document, final List<Concept> concepts) {
+		String documentId = IDValidator.getDocumentId(document);
+		Validate.notNull(concepts, "concepts can't be null");
 
-        Map<String, Object> queryParams = new HashMap<String, Object>();
-        JsonObject contentJson = new JsonObject();
-        JsonArray conceptsJson = new JsonArray();
+		Map<String, Object> queryParams = new HashMap<String, Object>();
+		JsonObject contentJson = new JsonObject();
+		JsonArray conceptsJson = new JsonArray();
 
-        for (String value : concepts) {
-            conceptsJson.add(new JsonPrimitive(value));
-        }
-        contentJson.add(CONCEPTS, conceptsJson);
-        queryParams.put(CONCEPTS, conceptsJson.toString());
-        return executeRequest(version + document.getId() + RELATION_SCORES_PATH, queryParams, Scores.class);
-    }
\ No newline at end of file
+		for (Concept con : concepts) {
+			conceptsJson.add(new JsonPrimitive(con.getId()));
+		}
+		contentJson.add(CONCEPTS, conceptsJson);
+		queryParams.put(CONCEPTS, conceptsJson.toString());
+		return executeRequest(API_VERSION + documentId + RELATION_SCORES_PATH, queryParams, Scores.class);
+	}
\ No newline at end of file
