diff --git a/old_product.java b/new_product.java
index 1040eb9..258c523 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,9 @@
             @NotBlank(message = "No job id entered. Unable to continue.")
             final String jobId
     ) throws GenieException {
-        LOG.debug("Called");
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Called");
+        }
         final Job job = this.jobRepo.findOne(jobId);
         if (job == null) {
             throw new GenieNotFoundException("No job with id " + jobId + " exists. Unable to continue."
