diff --git a/old_product.java b/new_product.java
index 0cbaf71..fc23000 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @Override
     public void enableGroup(String groupName) {
+        getTarget().enableGroup(groupName);
         // Cannot know wich feature to work with (exceptional event) : flush cache
         getCacheManager().clear();
-        getTarget().enableGroup(groupName);
     }
\ No newline at end of file
