diff --git a/old_test.java b/new_test.java
index de4a2b8..f53ae28 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,11 +2,11 @@
     @Transactional
     public void testEnableGroup() {
         // Given
-        assertFf4j.assertDisable(F2);
+        assertFf4j.assertThatFeatureIsDisabled(F2);
         // When
         testedStore.enableGroup(G0);
         // Then
-        assertFf4j.assertEnable(F2);
+        assertFf4j.assertThatFeatureIsEnabled(F2);
         // Reinit
         testedStore.disable(F2);
     }
\ No newline at end of file
