diff --git a/old_test.java b/new_test.java
index 8feff1c..5c34fa9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,16 +2,14 @@
   public void execute()
     throws Exception
   {
-    String result = Eclim.execute(new String[]{
+    String result = (String)Eclim.execute(new String[]{
       "history_clear", "-p", Eclim.TEST_PROJECT, "-f", TEST_FILE
     });
-    System.out.println(result);
     assertEquals("Wrong result.", result, "History Cleared.");
 
-    result = Eclim.execute(new String[]{
+    result = (String)Eclim.execute(new String[]{
       "history_list", "-p", Eclim.TEST_PROJECT, "-f", TEST_FILE
     });
-    System.out.println(result);
     assertEquals("Wrong result.", result, "[]");
 
     assertEquals("Wrong file contents.",
@@ -43,10 +41,9 @@
       "project_refresh_file", "-p", Eclim.TEST_PROJECT, "-f", TEST_FILE
     });
 
-    result = Eclim.execute(new String[]{
+    result = (String)Eclim.execute(new String[]{
       "history_list", "-p", Eclim.TEST_PROJECT, "-f", TEST_FILE
     });
-    System.out.println(result);
 
     Pattern pattern = Pattern.compile("^\\[" + ENTRY + "," + ENTRY + "\\]$");
     Matcher matcher = pattern.matcher(result);
@@ -59,9 +56,8 @@
     String ts = matcher.group(1);
     System.out.println(ts);
 
-    result = Eclim.execute(new String[]{
+    result = (String)Eclim.execute(new String[]{
       "history_revision", "-p", Eclim.TEST_PROJECT, "-f", TEST_FILE, "-r", ts
     });
-    System.out.println(result);
     assertEquals("Wrong result.", result, "line 1\n");
   }
\ No newline at end of file
