diff --git a/old_test.java b/new_test.java
index ecdbf1d..fde897d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,7 @@
 		final Instant eventTime = Instant.now();
 
 		huTraceRepository.addEvent(HUTraceEvent.builder()
+				.orgId(13)
 				.eventTime(eventTime)
 				.topLevelHuId(2)
 				.productId(23)
@@ -13,17 +14,20 @@
 				.type(HUTraceType.TRANSFORM_LOAD)
 				.build());
 
-		final HUTraceSpecification query = HUTraceSpecification.builder()
+		final HUTraceEventQuery query = HUTraceEventQuery.builder()
 				.vhuId(12)
 				.recursionMode(RecursionMode.NONE)
 				.build();
 
 		final List<HUTraceEvent> result = huTraceRepository.query(query);
 		assertThat(result.size(), is(1));
+		assertThat(result.get(0).getHuTraceEventId().isPresent(), is(true));
 		assertThat(result.get(0).getVhuId(), is(12));
+		assertThat(result.get(0).getOrgId(), is(13));
 
 		// add an equal event, again
 		huTraceRepository.addEvent(HUTraceEvent.builder()
+				.orgId(13)
 				.eventTime(eventTime)
 				.topLevelHuId(2)
 				.productId(23)
@@ -35,4 +39,5 @@
 
 		final List<HUTraceEvent> result2 = huTraceRepository.query(query);
 		assertThat(result2.size(), is(1)); // still just one..
+		assertThat(result.get(0), is(result2.get(0)));
 	}
\ No newline at end of file
