diff --git a/old_product.java b/new_product.java
index 4066cae..f72430d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public boolean addEvent(@NonNull final HUTraceEvent huTraceEvent)
 	{
-		final HUTraceSpecification query = huTraceEvent.asQuery();
+		final HUTraceEventQuery query = huTraceEvent.asQuery();
 
 		final I_M_HU_Trace dbRecord;
 		final List<I_M_HU_Trace> existingDbRecords = queryDbRecord(query);
