diff --git a/old_test.java b/new_test.java
index bcbdfbc..e6f776e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,13 @@
 @Test
     @RunTestInLooperThread
-    public void findAllSortedAsync() throws Throwable {
+    public void sort_async() throws Throwable {
         final Realm realm = looperThread.getRealm();
         populateTestRealm(realm, 10);
 
         final RealmResults<AllTypes> results = realm.where(AllTypes.class)
                 .between("columnLong", 0, 4)
-                .findAllSortedAsync("columnString", Sort.DESCENDING);
+                .sort("columnString", Sort.DESCENDING)
+                .findAllAsync();
 
         assertFalse(results.isLoaded());
         assertEquals(0, results.size());
