diff --git a/old_test.java b/new_test.java
index b6dadae..923bebf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,8 +3,8 @@
         String key = "key That Doesn't Exist";
         String defaultValue = "blue bunny";
         String expResult = "blue bunny";
-        String result = Settings.getString(key);
+        String result = getSettings().getString(key);
         Assert.assertTrue(result == null);
-        result = Settings.getString(key, defaultValue);
+        result = getSettings().getString(key, defaultValue);
         Assert.assertEquals(expResult, result);
     }
\ No newline at end of file
