diff --git a/old_product.java b/new_product.java
index c213fb2..50da5d0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,11 @@
 @Override
 	public boolean hasSameOrderableAs(Order otherOrder) {
-		if (!super.hasSameOrderableAs(otherOrder))
+		if (!super.hasSameOrderableAs(otherOrder)) {
 			return false;
-		if (!(otherOrder instanceof DrugOrder))
+		}
+		if (!(otherOrder instanceof DrugOrder)) {
 			return false;
+		}
 		DrugOrder otherDrugOrder = (DrugOrder) otherOrder;
 		return OpenmrsUtil.nullSafeEquals(this.getDrug(), otherDrugOrder.getDrug());
 	}
\ No newline at end of file
