diff --git a/old_test.java b/new_test.java
index 2523555..d3a4673 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,14 @@
 @Test
-	public void hasSameOrderableAs_shouldBeFalseIfConceptsDoNotMatch() {
+	public void hasSameOrderableAs_shouldReturnFalseIfTheConceptsMatchAndOnlyTheOtherHasADrug() throws Exception {
 		DrugOrder order = new DrugOrder();
-		Drug drug1 = new Drug();
-		drug1.setConcept(new Concept(123));
-		order.setDrug(drug1);
-
+		Concept concept = new Concept();
+		order.setConcept(concept);
+		
 		DrugOrder otherOrder = new DrugOrder();
-		Drug drug2 = new Drug();
-		drug2.setConcept(new Concept(456));
-		otherOrder.setDrug(drug2);
-
+		Drug drug1 = new Drug();
+		drug1.setConcept(concept);
+		otherOrder.setDrug(drug1); //should set the concept
+		assertEquals(order.getConcept(), otherOrder.getConcept());//sanity check
+		
 		assertFalse(order.hasSameOrderableAs(otherOrder));
 	}
\ No newline at end of file
