diff --git a/old_test.java b/new_test.java
index 94281fa..93f533c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test(description = "PUT /admin/network/{id}" )
    public void testUpdateNetwork() {
       //TODO: ensure network instanceof OrgNetwork, may require queries
-      assertTrue(network instanceof OrgNetwork, String.format(REF_REQ_LIVE, "OrgNetwork"));
+      assertTrue(network instanceof OrgNetwork, String.format(URN_REQ_LIVE, "OrgNetwork"));
       
       OrgNetwork oldNetwork = Network.<OrgNetwork>toSubType(network)
             .toBuilder()
@@ -11,10 +11,10 @@
       OrgNetwork updateNetwork = getMutatedOrgNetwork(oldNetwork);
       
       try {
-         Task updateNetworkTask = networkApi.updateNetwork(network.getHref(), updateNetwork);
+         Task updateNetworkTask = networkApi.update(networkUrn, updateNetwork);
          Checks.checkTask(updateNetworkTask);
          assertTrue(retryTaskSuccess.apply(updateNetworkTask), String.format(TASK_COMPLETE_TIMELY, "updateNetworkTask"));
-         network = networkApi.getNetwork(network.getHref());
+         network = networkApi.get(networkUrn);
          
          Checks.checkOrgNetwork(Network.<OrgNetwork>toSubType(network));
          
@@ -48,9 +48,9 @@
 //            updateNetwork.getAllowedExternalIpAddresses()), 
 //            String.format(OBJ_FIELD_UPDATABLE, NETWORK, "allowedExternalIpAddresses"));
       } finally {
-         Task updateNetworkTask = networkApi.updateNetwork(network.getHref(), oldNetwork);
+         Task updateNetworkTask = networkApi.update(networkUrn, oldNetwork);
          Checks.checkTask(updateNetworkTask);
          assertTrue(retryTaskSuccess.apply(updateNetworkTask), String.format(TASK_COMPLETE_TIMELY, "updateNetworkTask"));
-         network = networkApi.getNetwork(network.getHref());
+         network = networkApi.get(networkUrn);
       }
    }
\ No newline at end of file
