diff --git a/old_test.java b/new_test.java
index b9fbb9b..3cc1a50 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,5 @@
         QueryCriterion criterion1 = new EqCriterion("name", "abc");
         QueryCriterion criterion2 = new NotCriterion(criterion1);
         
-        assertTrue(instance.not(criterion1).getQueryCriterions()
-                .contains(criterion2));
+        assertEquals(criterion2, instance.not(criterion1).getQueryCriterion());
     }
\ No newline at end of file
