diff --git a/old_product.java b/new_product.java
index c4f85bd..a517310 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,9 +4,9 @@
         if (str == null) {
             str = nullText;
         }
-        int strLen = (str == null ? 0 : str.length());
+        final int strLen = (str == null ? 0 : str.length());
         if (strLen > 0) {
-            int newSize = size + strLen;
+            final int newSize = size + strLen;
             ensureCapacity(newSize);
             System.arraycopy(buffer, index, buffer, index + strLen, size - index);
             size = newSize;
