diff --git a/old_product.java b/new_product.java
index 816e5c1..80beb53 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
-@Deprecated
-    public static Date[] getExtentToday(){
+public static Date[] getResetTodayAndTomorrow(){
         Calendar calendar = CalendarUtil.resetCalendarByDay(new Date());
         Date today = calendar.getTime();
         // ***************************
