diff --git a/old_test.java b/new_test.java
index 4be4297..491507a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,5 +4,5 @@
         params.put("name", "abc");
         params.put("created", new Date());
         instance.setParameters(params);
-        assertEquals(MapParameters.create(params), instance.getParameters());
+        assertEquals(NamedParameters.create(params), instance.getParameters());
     }
\ No newline at end of file
