diff --git a/old_test.java b/new_test.java
index 2314d87..8d198c5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,14 @@
 @Test
     public void testReadGroup() {
+        // Given
+        assertFf4j.assertThatGroupExist(G1);
+        assertFf4j.assertThatFeatureExist(F3);
+        assertFf4j.assertThatFeatureExist(F4);
+        assertFf4j.assertThatFeatureIsInGroup(F3, G1);
+        assertFf4j.assertThatFeatureIsInGroup(F4, G1);
+        // When
         Map<String, Feature> group = testedStore.readGroup(G1);
+        // Then
         Assert.assertEquals(2, group.size());
         Assert.assertTrue(group.containsKey(F3));
         Assert.assertTrue(group.containsKey(F4));
