diff --git a/old_test.java b/new_test.java
index ba47d3e..cb83ab9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,9 +12,9 @@
     error("declare variable $a:=1;" + _XQUERY_EVAL.args("$a"), VARUNDEF_X);
     error("for $a in (1,2) return " + _XQUERY_EVAL.args("$a"), VARUNDEF_X);
     // check updating expressions
-    error(_XQUERY_EVAL.args("delete node ()"), BXXQ_UPDATING);
+    error(_XQUERY_EVAL.args("delete node ()"), XQUERY_UPDATE1);
     error(_XQUERY_EVAL.args("declare %updating function local:x() {()}; local:x()"),
-        BXXQ_UPDATING);
+        XQUERY_UPDATE1);
     query(_XQUERY_EVAL.args("declare %updating function local:x() {()}; 1"));
     query(_XQUERY_EVAL.args(DOC.args(PATH).trim()));
 
@@ -26,13 +26,13 @@
         " map { 'base-uri': 'http://x.x/' }"), "http://x.x/");
 
     error(_XQUERY_EVAL.args(DOC.args(NAME).trim(), " map { }",
-        " map { 'permission': 'none' }"), BXXQ_PERM_X);
+        " map { 'permission': 'none' }"), XQUERY_PERMISSION1_X);
     error(_XQUERY_EVAL.args(_DB_OPEN.args(NAME).trim(), " map { }",
-        " map { 'permission': 'none' }"), BXDB_OPEN_X);
+        " map { 'permission': 'none' }"), DB_OPEN2_X);
     error(_XQUERY_EVAL.args(_FILE_EXISTS.args("x").trim(), " map { }",
-        " map { 'permission': 'none' }"), BXXQ_PERM_X);
+        " map { 'permission': 'none' }"), XQUERY_PERMISSION1_X);
     error(_XQUERY_EVAL.args("(1 to 10000000000000)[. = 0]", " map { }",
-        " map { 'timeout': 1 }"), BXXQ_TIMEOUT);
+        " map { 'timeout': 1 }"), XQUERY_TIMEOUT);
     error(_XQUERY_EVAL.args("(1 to 10000000000000) ! <a/>", " map { }",
-        " map { 'memory': 10 }"), BXXQ_MEMORY);
+        " map { 'memory': 10 }"), XQUERY_MEMORY);
   }
\ No newline at end of file
