diff --git a/old_product.java b/new_product.java
index 856f656..94098fd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,12 @@
 public void createSubmission(SubmissionData submissionToAdd) throws EntityAlreadyExistsException {
-
+		Assumption.assertNotNull(Common.ERROR_DBLEVEL_NULL_INPUT, submissionToAdd);
+		
 		Assumption.assertTrue(submissionToAdd.getInvalidStateInfo(), submissionToAdd.isValid());
 		
 		if (getSubmissionEntity(submissionToAdd.course,
 				submissionToAdd.evaluation, submissionToAdd.reviewee,
 				submissionToAdd.reviewer) != null) {
-			String error = "Trying to create a Submission that exists: "
+			String error = ERROR_CREATE_SUBMISSION_ALREADY_EXISTS
 					+ "course: " + submissionToAdd.course + ", evaluation: "
 					+ submissionToAdd.evaluation + ", toStudent: "
 					+ submissionToAdd.reviewee + ", fromStudent: "
