diff --git a/old_test.java b/new_test.java
index 5d852f8..668f2cd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -103,6 +103,7 @@
 
     expect(cluster.getCurrentStackVersion()).andReturn(stackId);
     expect(cluster.getServiceComponentHosts(anyObject(String.class))).andReturn(schs).anyTimes();
+    expect(cluster.getHosts()).andReturn(Arrays.asList(new Host[]{host1, host2}));
 
     expect(sch.getServiceName()).andReturn("HIVE").anyTimes();
 
