diff --git a/old_test.java b/new_test.java
index ad1e253..4d75c07 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,10 @@
 @Test
     public void testGetAnalyzers() {
         AnalyzerService instance = new AnalyzerService(Thread.currentThread().getContextClassLoader());
-        Iterator<Analyzer> result = instance.getAnalyzers();
+        List<Analyzer> result = instance.getAnalyzers();
 
         boolean found = false;
-        while (result.hasNext()) {
-            Analyzer a = result.next();
+        for (Analyzer a : result) {
             if ("Jar Analyzer".equals(a.getName())) {
                 found = true;
             }
