diff --git a/old_test.java b/new_test.java
index 9e36fa6..838de64 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,7 +12,7 @@
             classifier.setC(1);
             classifier.train(trainSet, true);
 
-            for (int i = 0; i < testSet.getSampleSize(); i++)
+            for (int i = 0; i < testSet.size(); i++)
                 assertEquals(testSet.getDataPointCategory(i), classifier.classify(testSet.getDataPoint(i)).mostLikely());
         }
     }
\ No newline at end of file
