diff --git a/old_product.java b/new_product.java
index d0249ae..5fa233b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -39,7 +39,7 @@ public void trainC(ClassificationDataSet dataSet, Set<Integer> categoriesToUse)
         Arrays.fill(countArray, 1);//Laplace correction
         
         int[] cordinate = new int[dimSize.length];
-        for(int i = 0; i < dataSet.getSampleSize(); i++)
+        for(int i = 0; i < dataSet.size(); i++)
         {
             DataPoint dp = dataSet.getDataPoint(i);
             for (int j = 0; j < realIndexToCatIndex.length; j++)
