diff --git a/old_test.java b/new_test.java
index 2b8bc59..6be2340 100644
--- a/old_test.java
+++ b/new_test.java
@@ -99,4 +99,7 @@
     List<ArtifactSummary> summaries = readResponse(response, new TypeToken<List<ArtifactSummary>>() {
     }.getType());
     Assert.assertFalse(summaries.isEmpty());
+
+    // cleanup
+    deleteNamespace(NamespaceId.DEFAULT.getNamespace());
   }
\ No newline at end of file
