diff --git a/old_product.java b/new_product.java
index 2d447de..6435117 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public void delete(final Id.Artifact artifactId) throws IOException {
+public void delete(final Id.Artifact artifactId) throws ArtifactNotFoundException, IOException {
 
     // delete everything in a transaction
     try {
@@ -10,13 +10,12 @@ public void delete(final Id.Artifact artifactId) throws IOException {
           Table metaTable = getMetaTable(context);
           byte[] detailBytes = metaTable.get(artifactCell.rowkey, artifactCell.column);
           if (detailBytes == null) {
-            // ok there is nothing to delete, we're done
-            return;
+            throw new ArtifactNotFoundException(artifactId.toEntityId());
           }
           deleteMeta(metaTable, artifactId, detailBytes);
         }
       });
     } catch (TransactionFailureException e) {
-      throw Transactions.propagate(e, IOException.class);
+      throw Transactions.propagate(e, IOException.class, ArtifactNotFoundException.class);
     }
   }
\ No newline at end of file
