diff --git a/old_product.java b/new_product.java
index 629e8cd..3d439b1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static double Binomial(int k, double p, int n) {
+public static double binomial(int k, double p, int n) {
         if(k<0 ||  p<0 || n<1) {
             throw new IllegalArgumentException("All the parameters must be positive and n larger than 1.");
         }
