diff --git a/old_test.java b/new_test.java
index 5a1f7a7..fa3f544 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,6 @@
         double p = 0.5;
         int n = 10;
         double expResult = 0.11718750001462;
-        double result = DiscreteDistributions.Binomial(k, p, n);
+        double result = DiscreteDistributions.binomial(k, p, n);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
