diff --git a/old_test.java b/new_test.java
index 738d5f3..4be38ba 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,14 +3,14 @@
         when(mockControllerService.getStreamStore()).thenReturn(mockStreamStore);
 
         // Test to get an existing stream
-        when(mockStreamStore.getConfiguration(stream1)).thenReturn(streamConfigFuture);
+        when(mockStreamStore.getConfiguration(scope1, stream1)).thenReturn(streamConfigFuture);
         response = target(resourceURI).request().async().get();
         streamResponseActual = response.get().readEntity(StreamProperty.class);
         assertEquals("Get Stream Config Status", 200, response.get().getStatus());
         testExpectedVsActualObject(streamResponseExpected, streamResponseActual);
 
         // Get a non-existent stream
-        when(mockStreamStore.getConfiguration(stream2)).thenReturn(CompletableFuture.supplyAsync(() -> {
+        when(mockStreamStore.getConfiguration(scope1, stream2)).thenReturn(CompletableFuture.supplyAsync(() -> {
             throw new DataNotFoundException("Stream Not Found");
         }));
         response = target(resourceURI2).request().async().get();
