diff --git a/old_product.java b/new_product.java
index b1e86cc..e7d66de 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@
         long traceId = LoggerHelpers.traceEnter(log, "getStreamConfig");
 
         StreamMetadataStore streamStore = controllerService.getStreamStore();
-        streamStore.getConfiguration(stream)
+        streamStore.getConfiguration(scope, stream)
                 .thenApply(streamConfig -> {
                     return Response.status(Status.OK).entity(ModelHelper.encodeStreamResponse(streamConfig)).build();
                 })
