diff --git a/old_product.java b/new_product.java
index 6305bdf..cbba922 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
-public static PatchOperationBuilder rollback(final String patchId, final boolean rollbackTo, final boolean restoreConfiguration) {
+public static PatchOperationBuilder rollback(final String patchId, final boolean rollbackTo, final boolean resetConfiguration) {
             return new AbstractOperationBuilder() {
                 @Override
                 public ModelNode execute(PatchOperationTarget target) throws IOException {
-                    return target.rollback(patchId, this, rollbackTo, restoreConfiguration);
+                    return target.rollback(patchId, this, rollbackTo, resetConfiguration);
                 }
             };
         }
\ No newline at end of file
