diff --git a/old_test.java b/new_test.java
index 9da8f78..6ce375c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,5 +2,5 @@
   public void testCreateUser_NoDuplicates() throws Exception {
     initForCreateUser(null);
     Users users = injector.getInstance(Users.class);
-    users.createUser(SERVICEOP_USER_NAME, "qwert");
+    users.createUser(SERVICEOP_USER_NAME, SERVICEOP_USER_NAME, SERVICEOP_USER_NAME);
   }
\ No newline at end of file
