diff --git a/old_product.java b/new_product.java
index 216422a..a5a938e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,3 @@
-public static boolean hasPostfixName(String fileName){
-        String fileNameString = getFileName(fileName);
-        int lastIndexOf = fileNameString.lastIndexOf(".");
-        return -1 != lastIndexOf;
+public static boolean hasExtension(String fileName){
+        return -1 != FilenameUtils.indexOfExtension(fileName);
     }
\ No newline at end of file
