diff --git a/old_product.java b/new_product.java
index 70700b9..c89780e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static void setInputs(Contract contract, Contract[] inputs) {
+public static void setInputs(final Contract contract, final Contract[] inputs) {
 		if (contract instanceof DataSinkContract<?, ?>) {
 			if (inputs.length != 1)
 				throw new IllegalArgumentException("wrong number of inputs");
