diff --git a/old_test.java b/new_test.java
index 1b373bc..c8f9a29 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,5 @@
-@Test public void testFit() {
+@Test
+    public void testFit() {
         int sampleSize = 1000;
         Map<double[], Double> sample = new HashMap<>();
         for (int i = 0; i < sampleSize; i++) {
@@ -11,12 +12,14 @@
         }
 
         RandomForestClassifierTrainer trainer = new RandomForestClassifierTrainer(4, 3, 5, 0.3, 4, 0.1);
-        ModelsComposition model = trainer.fit(sample, parts, (k, v) -> VectorUtils.of(k), (k, v) -> v);
-        model.getModels().forEach(m -> {
+
+        ModelsComposition mdl = trainer.fit(sample, parts, (k, v) -> VectorUtils.of(k), (k, v) -> v);
+
+        mdl.getModels().forEach(m -> {
             assertTrue(m instanceof ModelOnFeaturesSubspace);
             assertTrue(((ModelOnFeaturesSubspace) m).getMdl() instanceof DecisionTreeConditionalNode);
         });
 
-        assertTrue(model.getPredictionsAggregator() instanceof OnMajorityPredictionsAggregator);
-        assertEquals(5, model.getModels().size());
+        assertTrue(mdl.getPredictionsAggregator() instanceof OnMajorityPredictionsAggregator);
+        assertEquals(5, mdl.getModels().size());
     }
\ No newline at end of file
