diff --git a/old_product.java b/new_product.java
index adb879e..ef5bc88 100644
--- a/old_product.java
+++ b/new_product.java
@@ -20,7 +20,7 @@
             int iteration = 0;
 
             while (iteration < maxIterations && !converged) {
-                Vector[] newCentroids = new DenseLocalOnHeapVector[k];
+                Vector[] newCentroids = new DenseVector[k];
 
                 TotalCostAndCounts totalRes = calcDataForNewCentroids(centers, dataset, cols);
 
