diff --git a/old_test.java b/new_test.java
index 9ca164d..d0fcb6f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -43,7 +43,7 @@
         double expResult2 = 0.8;
         assertEquals(expResult2, vm.getMacroF1(), Constants.DOUBLE_ACCURACY_HIGH);
 
-        trainingData.delete();
+        trainingData.close();
         instance.close();
         //instance = null;
 
@@ -66,5 +66,5 @@
         
         instance.delete();
 
-        validationData.delete();
+        validationData.close();
     }
\ No newline at end of file
