diff --git a/old_test.java b/new_test.java
index 5b44364..1388c05 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,16 @@
 @Test
 	public void testPutConnectionBackInPartition() throws InterruptedException {
 		expect(mockPartition.getFreeConnections()).andReturn(mockConnectionHandles).anyTimes();
+		AtomicInteger ai = new AtomicInteger(1);
+		expect(mockPartition.getAvailableConnections()).andReturn(ai).anyTimes();
+		
 		expect(mockConnection.getOriginatingPartition()).andReturn(mockPartition).anyTimes();
+		expect(mockConnectionHandles.tryTransfer(mockConnection)).andReturn(false).anyTimes();
 		mockConnectionHandles.put(mockConnection);
 		expectLastCall().once();
 		replay(mockPartition, mockConnectionHandles, mockConnection);
 		testClass.putConnectionBackInPartition(mockConnection);
+		assertEquals(2, ai.get());
 		verify(mockPartition, mockConnectionHandles);
 		
 	}
\ No newline at end of file
