diff --git a/old_product.java b/new_product.java
index aa40a5e..497543c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,10 @@
-protected void putConnectionBackInPartition(ConnectionHandle connectionHandle) throws InterruptedException {
+protected void putConnectionBackInPartition(ConnectionHandle connectionHandle) {
 		LinkedTransferQueue<ConnectionHandle> queue = connectionHandle.getOriginatingPartition().getFreeConnections();
-		
+
 		if (!queue.tryTransfer(connectionHandle)){
 			queue.put(connectionHandle);
 		}
+		
+		connectionHandle.getOriginatingPartition().getAvailableConnections().incrementAndGet();
+		
 	}
\ No newline at end of file
