diff --git a/old_test.java b/new_test.java
index e5d2f77..9ec2ad1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,8 @@
         if (!p.getValidationErrors().isEmpty()) {
             for (Iterator e = p.getValidationErrors().iterator(); e.hasNext(); ) {
                 SAXParseException ex = (SAXParseException) e.next();
-                System.out.println(
+                LOGGER.log(
+                        Level.SEVERE,
                         ex.getLineNumber() + "," + ex.getColumnNumber() + " -" + ex.toString());
             }
             fail("Document did not validate.");
