diff --git a/old_product.java b/new_product.java
index 8162320..e9d7a08 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,9 +7,9 @@ public boolean validate(
             line = getDefaultLineString(feature);
         } catch (ClassCastException unLine) {
             results.error(feature, "Geometry is required to be a LineString");
-            System.out.println(feature.getID() + "  name: " + getName());
-            System.out.println(feature.getID() + "   ref: " + getTypeRef());
-            System.out.println(feature.getID() + "   ref: " + getTypeRefs());
+            // System.out.println(feature.getID() + "  name: " + getName());
+            // System.out.println(feature.getID() + "   ref: " + getTypeRef());
+            // System.out.println(feature.getID() + "   ref: " + getTypeRefs());
         }
         if (line == null) {
             // Ignore null geometry (user can check with nullZero )
