diff --git a/old_test.java b/new_test.java
index c677e57..abe87fd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,15 +10,36 @@
         final String app2 = UUID.randomUUID().toString();
         final String app3 = UUID.randomUUID().toString();
         final List<Application> applications = Lists.newArrayList(
-            new Application.Builder(placeholder, placeholder, placeholder, ApplicationStatus.ACTIVE)
-                .withId(app3)
-                .build(),
-            new Application.Builder(placeholder, placeholder, placeholder, ApplicationStatus.ACTIVE)
-                .withId(app1)
-                .build(),
-            new Application.Builder(placeholder, placeholder, placeholder, ApplicationStatus.ACTIVE)
-                .withId(app2)
-                .build()
+            new Application(
+                app3,
+                Instant.now(),
+                Instant.now(),
+                new ExecutionEnvironment(null, null, null),
+                new ApplicationMetadata
+                    .Builder(placeholder, placeholder, ApplicationStatus.ACTIVE)
+                    .withVersion(placeholder)
+                    .build()
+            ),
+            new Application(
+                app1,
+                Instant.now(),
+                Instant.now(),
+                new ExecutionEnvironment(null, null, null),
+                new ApplicationMetadata
+                    .Builder(placeholder, placeholder, ApplicationStatus.ACTIVE)
+                    .withVersion(placeholder)
+                    .build()
+            ),
+            new Application(
+                app2,
+                Instant.now(),
+                Instant.now(),
+                new ExecutionEnvironment(null, null, null),
+                new ApplicationMetadata
+                    .Builder(placeholder, placeholder, ApplicationStatus.ACTIVE)
+                    .withVersion(placeholder)
+                    .build()
+            )
         );
 
         final JobRequest jobRequest = new JobRequest.Builder(
@@ -32,25 +53,24 @@
             .withApplications(Lists.newArrayList(app3, app1, app2))
             .build();
 
-        final Cluster cluster = new Cluster.Builder(
-            CLUSTER_NAME,
-            USER,
-            VERSION,
-            ClusterStatus.UP
-        )
-            .withId(CLUSTER_ID)
-            .build();
+        final Cluster cluster = new Cluster(
+            CLUSTER_ID,
+            Instant.now(),
+            Instant.now(),
+            new ExecutionEnvironment(null, null, null),
+            new ClusterMetadata.Builder(CLUSTER_NAME, USER, ClusterStatus.UP).withVersion(VERSION).build()
+        );
 
-        final Command command = new Command.Builder(
-            COMMAND_NAME,
-            USER,
-            VERSION,
-            CommandStatus.ACTIVE,
-            "foo",
+        final Command command = new Command(
+            COMMAND_ID,
+            Instant.now(),
+            Instant.now(),
+            new ExecutionEnvironment(null, null, null),
+            new CommandMetadata.Builder(COMMAND_NAME, USER, CommandStatus.ACTIVE).withVersion(VERSION).build(),
+            Lists.newArrayList("foo"),
+            null,
             5000L
-        )
-            .withId(COMMAND_ID)
-            .build();
+        );
 
         final int memory = 2438;
 
