diff --git a/old_test.java b/new_test.java
index b119f2e..0866e91 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,6 +16,9 @@
     query(_JOBS_SCHEDULE.args("delete node <a/>"));
 
     // errors
-    error(_JOBS_SCHEDULE.args("1+"), CALCEXPR);
-    error(_JOBS_SCHEDULE.args("1, delete node <a/>"), UPALL);
+    query(_JOBS_SCHEDULE.args("1+"));
+    query(_JOBS_SCHEDULE.args("1, delete node <a/>"));
+    error(_JOBS_SCHEDULE.args("1+<a/>", "()", " map{ 'start':'12345' }"), DATEFORMAT_X_X_X);
+    error(_JOBS_SCHEDULE.args("1+<a/>", "()", " map{ 'repeat':'12345' }"), DATEFORMAT_X_X_X);
+    error(_JOBS_SCHEDULE.args("1+<a/>", "()", " map{ 'repeat':'-PT1S' }"), JOBS_RANGE);
   }
\ No newline at end of file
