diff --git a/old_test.java b/new_test.java
index 70da6c5..11b702a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,10 @@
 @Test
     public void test_extractProperties() {
-        final String jsonString = "{\"key1\": \"value1\", \"key2\": \"value2\"}";
+        final String jsonString
+                = "{\"key1\": \"value1\", \"key2\": \"value2\"}";
         FormData formData = new FormData(1);
         formData.add(BodyInjectorHandler.PROPERTIES, jsonString);
-        DBObject result = BodyInjectorHandler.extractProperties(formData);
-        DBObject expected = (DBObject) JSON.parse(jsonString);
+        BsonDocument result = BodyInjectorHandler.extractProperties(formData);
+        BsonDocument expected = BsonDocument.parse(jsonString);
         assertEquals(expected, result);
     }
\ No newline at end of file
