diff --git a/old_product.java b/new_product.java
index 3e42649..01f6c16 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,14 @@
-protected static DBObject extractProperties(final FormData formData) throws JSONParseException {
-        DBObject properties = new BasicDBObject();
+protected static BsonDocument extractProperties(
+            final FormData formData)
+            throws JSONParseException {
+        BsonDocument properties = new BsonDocument();
 
         final String propsString = formData.getFirst(PROPERTIES) != null
                 ? formData.getFirst(PROPERTIES).getValue()
                 : null;
 
         if (propsString != null) {
-            properties = (DBObject) JSON.parse(propsString);
+            properties = BsonDocument.parse(propsString);
         }
 
         return properties;
