diff --git a/old_test.java b/new_test.java
index 07922ad..7711f00 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
 	public void test_CUToExistingTU_create_mixed_TU_completeCU()
 	{
-		final I_M_HU cu1 = mkRealCUWithTUToSplit("5");
+		final I_M_HU cu1 = mkRealCUWithTUToSplit("5", create_receiptSchedule_for_realCUWithTU);
 
 		final HUProducerDestination producer = HUProducerDestination.ofVirtualPI();
 		data.helper.load(producer, data.helper.pSalad, new BigDecimal("4"), data.helper.uomKg);
