diff --git a/old_test.java b/new_test.java
index 445cf03..a99f1a9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,11 +4,11 @@
       vAppNames.add(newVApp.getName());
 
       // The method under test
-      Task editVApp = vAppApi.editVApp(vApp.getHref(), newVApp);
+      Task editVApp = vAppApi.edit(vAppUrn, newVApp);
       assertTrue(retryTaskSuccess.apply(editVApp), String.format(TASK_COMPLETE_TIMELY, "editVApp"));
 
       // Get the edited VApp
-      vApp = vAppApi.getVApp(vApp.getHref());
+      vApp = vAppApi.get(vAppUrn);
 
       // Check the required fields are set
       assertEquals(vApp.getName(), newVApp.getName(),
