diff --git a/old_test.java b/new_test.java
index b621ac0..e087074 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,6 @@
    public void testSetResponseTime()
    {
       System.out.println("setResponseTime");
-      int time = 0;
       DummySampler instance = new DummySampler();
-      instance.setResponseTime(time);
+      instance.setResponseTime("10");
    }
\ No newline at end of file
