diff --git a/old_test.java b/new_test.java
index f8dfd36..6b66c22 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
   public void retrieve() {
-    error(_DB_RETRIEVE.args(NAME, "raw"), Err.WHICHRES);
+    error(_DB_RETRIEVE.args(NAME, "raw"), Err.WHICHRES_X);
     query(_DB_STORE.args(NAME, "raw", "xs:hexBinary('41')"));
     query("xs:hexBinary(" + _DB_RETRIEVE.args(NAME, "raw") + ')', "41");
     query(_DB_DELETE.args(NAME, "raw"));
-    error(_DB_RETRIEVE.args(NAME, "raw"), Err.WHICHRES);
+    error(_DB_RETRIEVE.args(NAME, "raw"), Err.WHICHRES_X);
   }
\ No newline at end of file
