diff --git a/old_product.java b/new_product.java
index 7385849..b1c1dfe 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery startsWithText(String propName, String value) {
-        criterion = criterion.and(criterionBuilder.startsWithText(propName, value));
+        criterion = criterion.and(Criteria.startsWithText(propName, value));
         return this;
     }
\ No newline at end of file
