diff --git a/old_test.java b/new_test.java
index 533bead..3c08f6a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,7 +11,7 @@
         BufferedReader result = null;
         try (BufferedInputStream stream = new BufferedInputStream(
                 new FileInputStream(testFile))) {
-            CSVFileReader instance = new CSVFileReader(new CSVFileReaderSpi());
+            CSVFileReader instance = new CSVFileReader(new CSVFileReaderSpi(), ',');
             File outFile = instance.read(stream, null).getTabDelimitedFile();
             result = new BufferedReader(new FileReader(outFile));
             logger.fine("Final pass: " + outFile.getPath());
