diff --git a/old_product.java b/new_product.java
index e7c5880..d852779 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,7 +15,7 @@
 
         int lineCount = readFile(localBufferedReader, dataTable, tabFileWriter);
 
-        dbglog.fine("Tab file produced: " + tabFileDestination.getAbsolutePath());
+        logger.fine("Tab file produced: " + tabFileDestination.getAbsolutePath());
 
         dataTable.setUnf("UNF:6:NOTCALCULATED");
 
