diff --git a/old_product.java b/new_product.java
index 91e2589..4e87a61 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,5 @@
 private <T> void deferEvent(T event, Observer<T> observer)
    {
-      TransactionListener transactionListener = manager.getInstanceByType(TransactionListener.class);
       DeferredEventNotification<T> deferredEvent = new DeferredEventNotification<T>(event, observer);
-      transactionListener.registerSynhronization(deferredEvent);
+      userTransaction.registerSynchronization(deferredEvent);
    }
\ No newline at end of file
