diff --git a/old_test.java b/new_test.java
index b515878..4f395af 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test(expected = IllegalArgumentException.class)
-    public void testIsOneTrue_primitive_nullInput() {
-        BooleanUtils.isOneTrue((boolean[]) null);
+    public void testIsExactlyOneTrue_primitive_nullInput() {
+        BooleanUtils.isExactlyOneTrue((boolean[]) null);
     }
\ No newline at end of file
