diff --git a/old_product.java b/new_product.java
index a497cb4..19154b7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static boolean isOneTrue(final boolean... array) {
+public static boolean isExactlyOneTrue(final boolean... array) {
         // Validates input
         if (array == null) {
             throw new IllegalArgumentException("The Array must not be null");
@@ -11,7 +11,7 @@ public static boolean isOneTrue(final boolean... array) {
         int trueCount = 0;
         for (final boolean element : array) {
             // If item is true, and trueCount is < 1, increments count
-            // Else, isOneTrue fails
+            // Else, isExactlyOneTrue fails
             if (element) {
                 if (trueCount < 1) {
                     trueCount++;
