diff --git a/old_product.java b/new_product.java
index 51f62b8..975c4e2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,8 +5,7 @@
 			return (this.getConceptId().equals(c.getConceptId()));
 		} else if (obj instanceof Concept) {
 			// use the reverse .equals in case we have hibernate proxies - #1511
-			if (obj.equals(this))
-				return true;
+			return OpenmrsUtil.nullSafeEquals(((Concept) obj).getConceptId(), this.getConceptId());
 		}
 		
 		// fall back to object equality
