diff --git a/old_test.java b/new_test.java
index fbe3275..21a706b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,10 +17,4 @@
 		Assert.assertNull(first.getIdentifierType());
 		
 		Assert.assertFalse(first + " and " + second + " should not equal." , second.equals(first));
-		
-		first.setIdentifier("EXAMPLE IDENTIFIER");
-		first.setPatient(patient);
-		first.setIdentifierType(new PatientIdentifierType(1));
-		
-		Assert.assertTrue(first + " and " + second + " should be equal." , second.equals(first));
 	}
\ No newline at end of file
