diff --git a/old_product.java b/new_product.java
index 141e2bb..584468c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,6 +11,7 @@
                 if (addresses != null && addresses.contains(ipAddress)) {
                     addresses = new HashSet<>(existingHost.ipAddresses());
                     addresses.remove(ipAddress);
+                    removeIpFromHostsByIp(existingHost, ipAddress);
                     return new DefaultHost(existingHost.providerId(),
                             hostId,
                             existingHost.mac(),
