diff --git a/old_test.java b/new_test.java
index 8e768e5..488a7c2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,8 +14,8 @@
     String [] expectedResTopDir2 = new String[] {"/testDirNonEmpty2", "/testDirNonEmptyF"};
     Arrays.sort(expectedResTopDir);
     Arrays.sort(expectedResTopDir2);
-    UnderFileStatus [] resTopDirStatus = mUfs.listStatus(testDirNonEmpty);
-    String [] resTopDir = UnderFileStatus.convertToNames(resTopDirStatus);
+    UfsStatus[] resTopDirStatus = mUfs.listStatus(testDirNonEmpty);
+    String [] resTopDir = UfsStatus.convertToNames(resTopDirStatus);
     Arrays.sort(resTopDir);
     Assert.assertTrue(Arrays.equals(expectedResTopDir, resTopDir)
         || Arrays.equals(expectedResTopDir2, resTopDir));
