diff --git a/old_test.java b/new_test.java
index 98ad2d7..6f3793a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,12 +7,6 @@
         LOGGER.info(URIUtil.encode(value, CharsetType.UTF8));
         value = "景儿,么么哒";
         LOGGER.info(URIUtil.encode(value, CharsetType.UTF8));
-
-        value = "白色/黑色/纹理浅麻灰";
-        result = URIUtil.encode(value, CharsetType.UTF8);
-        LOGGER.info(result);
-        LOGGER.info(URIUtil.encode(value, CharsetType.UTF8));
-
-        result = URIUtil.encode("Lifestyle / Graphic,", CharsetType.UTF8);
-        LOGGER.info(result);
+        LOGGER.info(URIUtil.encode("白色/黑色/纹理浅麻灰", CharsetType.UTF8));
+        LOGGER.info(URIUtil.encode("Lifestyle / Graphic,", CharsetType.UTF8));
     }
\ No newline at end of file
