diff --git a/old_test.java b/new_test.java
index 564812f..19efc9b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,14 +15,19 @@
         final AttachmentReference attachmentName2 = new AttachmentReference("filename2",
             new DocumentReference("wiki", "space", "page"));
 
-        getMockery().checking(new Expectations() {{
-            allowing(arf).resolve("page@filename1"); will(returnValue(attachmentName1));
-            oneOf(dab).getAttachmentContent(attachmentName1);
+        getMockery().checking(new Expectations()
+        {
+            {
+                allowing(URIClassLoaderTest.this.arf).resolve("page@filename1");
+                will(returnValue(attachmentName1));
+                oneOf(URIClassLoaderTest.this.dab).getAttachmentContent(attachmentName1);
                 will(returnValue(new ByteArrayInputStream(createJarFile("/nomatch"))));
-            allowing(arf).resolve("filename2"); will(returnValue(attachmentName2));
-            oneOf(dab).getAttachmentContent(attachmentName2);
+                allowing(URIClassLoaderTest.this.arf).resolve("filename2");
+                will(returnValue(attachmentName2));
+                oneOf(URIClassLoaderTest.this.dab).getAttachmentContent(attachmentName2);
                 will(returnValue(new ByteArrayInputStream(createJarFile("/something"))));
-        }});
+            }
+        });
 
         Assert.assertEquals("jar:attachmentjar://filename2!/something", cl.findResource("/something").toString());
     }
\ No newline at end of file
