diff --git a/old_product.java b/new_product.java
index ba0bd99..4c219d7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,11 @@
-public URL findResource(final String name)
+@Override
+    public URL findResource(final String name)
     {
-        return (URL) AccessController.doPrivileged(new PrivilegedAction()
+        return AccessController.doPrivileged(new PrivilegedAction<URL>()
         {
-            public Object run()
+            public URL run()
             {
-                return finder.findResource(name);
+                return URIClassLoader.this.finder.findResource(name);
             }
-        }, acc);
+        }, this.acc);
     }
\ No newline at end of file
