diff --git a/old_test.java b/new_test.java
index 4354e89..960b800 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,8 +9,8 @@
         " map { xs:QName('local:a'): 1 }"), "1");
     query(_XQUERY_EVAL.args(".", " map { '': 1 }"), "1");
     error(_XQUERY_EVAL.args("1+"), Err.CALCEXPR);
-    error("declare variable $a:=1;" + _XQUERY_EVAL.args("\"$a\""), Err.VARUNDEF);
-    error("for $a in (1,2) return " + _XQUERY_EVAL.args("\"$a\""), Err.VARUNDEF);
+    error("declare variable $a:=1;" + _XQUERY_EVAL.args("\"$a\""), Err.VARUNDEF_X);
+    error("for $a in (1,2) return " + _XQUERY_EVAL.args("\"$a\""), Err.VARUNDEF_X);
     // check updating expressions
     error(_XQUERY_EVAL.args("delete node ()"), Err.BXXQ_UPDATING);
     error(_XQUERY_EVAL.args("declare %updating function local:x() {()}; local:x()"),
@@ -23,11 +23,11 @@
     query("try{ " + _XQUERY_EVAL.args("\"(1 to 10000000000000)[.=0]\"", " map{}",
         " map{ 'timeout':'1'}") + " } catch * { () }", "");
     error(_XQUERY_EVAL.args("\"doc('" + NAME + "')\"", " map{}", " map{ 'permission':'none'}"),
-        Err.BXXQ_PERM);
+        Err.BXXQ_PERM_X);
     error(_XQUERY_EVAL.args("\"db:open('" + NAME + "')\"", " map{}", " map{ 'permission':'none'}"),
-        Err.BXDB_OPEN);
+        Err.BXDB_OPEN_X);
     error(_XQUERY_EVAL.args("\"file:exists('x')\"", " map{}", " map{ 'permission':'none'}"),
-        Err.BXXQ_PERM);
+        Err.BXXQ_PERM_X);
     error(_XQUERY_EVAL.args("\"(1 to 10000000000000)[.=0]\"", " map{}", " map{ 'timeout':'1'}"),
         Err.BXXQ_STOPPED);
     error(_XQUERY_EVAL.args("\"(1 to 10000000000000) ! <a/>\"", " map{}", " map{ 'memory':'10'}"),
