diff --git a/old_product.java b/new_product.java
index d4c0b68..dd1afec 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,11 +2,7 @@
 	public SearchUserResponse searchUser(String loginName, String name) {
 		SearchUserResponse response = new SearchUserResponse();
 		try {
-
-			Map<String, Object> parameters = Maps.newHashMap();
-			parameters.put("loginName", loginName);
-			parameters.put("name", name);
-			List<User> userList = accountDao.searchUser(parameters);
+			List<User> userList = accountService.searchUser(loginName, name);
 
 			List<UserDTO> dtoList = BeanMapper.mapList(userList, UserDTO.class);
 			response.setUserList(dtoList);
