diff --git a/old_test.java b/new_test.java
index 4099923..a992886 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,9 @@
 @Test
 	@Category(Smoke.class)
-	public void searchUser() {
+	public void getTeamDetail() {
 		AccountWebService accountWebService = creatClient();
 
-		SearchUserResponse result = accountWebService.searchUser(null, null);
-
-		assertTrue(result.getUserList().size() >= 4);
-		assertEquals("Jack", result.getUserList().get(0).getName());
+		GetTeamDetailResponse response = accountWebService.getTeamDetail(1L);
+		assertEquals("Dolphin", response.getTeam().getName());
+		assertEquals("Admin", response.getTeam().getMaster().getName());
 	}
\ No newline at end of file
