diff --git a/old_product.java b/new_product.java
index d13a013..a01977d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,4 @@
 @Override
 	public List<Order> getOrdersByPatient(Patient patient) throws APIException {
-		if (patient == null)
-			throw new APIException("Unable to get orders if I am not given a patient");
-		
-		List<Patient> patients = new Vector<Patient>();
-		patients.add(patient);
-		
-		return getOrders(Order.class, patients, null, null, null, null, null, null);
+		return getOrdersByPatient(patient, false);
 	}
\ No newline at end of file
