diff --git a/old_test.java b/new_test.java
index f6fc74d..d052f90 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,11 @@
 @Test
-	@Ignore
 	@Verifies(value = "return list of orders for patient with respect to the include voided flag", method = "getOrdersByPatient(Patient, boolean)")
 	public void getOrdersByPatient_shouldReturnListOfOrdersForPatientWithRespectToTheIncludeVoidedFlag() throws Exception {
 		executeDataSet(ORDERS_DATASET_XML);
 		Patient p = Context.getPatientService().getPatient(2);
 		List<Order> orders = Context.getOrderService().getOrdersByPatient(p, true);
-		Assert.assertEquals(8, orders.size());
+		Assert.assertEquals(12, orders.size());
 		
 		orders = Context.getOrderService().getOrdersByPatient(p, false);
-		Assert.assertEquals(4, orders.size());
+		Assert.assertEquals(8, orders.size());
 	}
\ No newline at end of file
