diff --git a/old_test.java b/new_test.java
index 450899d..e3b1285 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,9 +5,7 @@
         User user2 = new User();
         user2.setId(2L);
 
-        User[] users = { user1, user2 };
-        LOGGER.info(JsonUtil.format(ConvertUtil.toList(users)));
+        LOGGER.info(JsonUtil.format(ConvertUtil.toList(user1, user2)));
 
-        users = null;
-        LOGGER.info(JsonUtil.format(ConvertUtil.toList(users)));
+        LOGGER.info(JsonUtil.format(ConvertUtil.toList((User) null)));
     }
\ No newline at end of file
