diff --git a/old_product.java b/new_product.java
index 24af888..0dc1b84 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @Override
-    public int removeFirstOccurrence(KType e1)
-    {
-        final int index = bufferIndexOf(e1);
-        if (index >= 0) removeAtBufferIndex(index);
-        return index;
-    }
\ No newline at end of file
+  public int removeFirst(KType e1) {
+    final int index = bufferIndexOf(e1);
+    if (index >= 0)
+      removeAtBufferIndex(index);
+    return index;
+  }
\ No newline at end of file
