diff --git a/old_test.java b/new_test.java
index 6f57bfa..644789c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,13 +15,13 @@
         {
             KType k = cast(rnd.nextInt(modulo));
             assertEquals(
-                deque.removeFirstOccurrence(k) >= 0, 
+                deque.removeFirst(k) >= 0, 
                 sequence.removeFirst(k) >= 0);
         }
 
         assertListEquals(deque.toArray(), sequence.toArray());
 
-        assertTrue(0 > deque.removeFirstOccurrence(cast(modulo + 1)));
+        assertTrue(0 > deque.removeFirst(cast(modulo + 1)));
         deque.addLast(cast(modulo + 1));
-        assertTrue(0 <= deque.removeFirstOccurrence(cast(modulo + 1)));
+        assertTrue(0 <= deque.removeFirst(cast(modulo + 1)));
     }
\ No newline at end of file
