diff --git a/old_product.java b/new_product.java
index 70e941a..677557f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public static Set<File> filterSet(final IOFileFilter filter, final File... files) {
         final File[] acceptedFiles = filter(filter, files);
-        return new HashSet<File>(Arrays.asList(acceptedFiles));
+        return new HashSet<>(Arrays.asList(acceptedFiles));
     }
\ No newline at end of file
