diff --git a/old_test.java b/new_test.java
index 7aa45fe..d74c595 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testFilterSet() throws Exception {
         final File fileA = TestUtils.newFile(getTestDirectory(), "A");
         final File fileB = TestUtils.newFile(getTestDirectory(), "B");
-        final Set<File> fileList = new HashSet<File>(Arrays.asList(fileA, fileB));
+        final Set<File> fileList = new HashSet<>(Arrays.asList(fileA, fileB));
 
         final IOFileFilter filter = FileFilterUtils.nameFileFilter("A");
 
