diff --git a/old_product.java b/new_product.java
index b83ea5e..0d32708 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,17 @@
 @Override
-    public boolean equals(Object o){
-        if (!(o instanceof Hit)) return false;
-        
-        return o.hashCode() == this.hashCode();
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        final Hit other = (Hit) obj;
+        if (this.hitLen != other.hitLen) {
+            return false;
+        }
+        if (this.hsps != other.hsps && (this.hsps == null || !this.hsps.equals(other.hsps))) {
+            return false;
+        }
+        return true;
     }
\ No newline at end of file
