diff --git a/old_test.java b/new_test.java
index b4a096e..0f84236 100644
--- a/old_test.java
+++ b/new_test.java
@@ -23,7 +23,7 @@
         
         assertEquals(o, instance);
         
-        // example of Hsp retrieved from uncomplete report. The result is null, hance false
+        // example of Hsp retrieved from uncomplete report.
         // (Those HSP may come from a tabular format, for example)
         o = new BlastHspBuilder()
                 .setPercentageIdentity(100.00/100)
@@ -38,6 +38,5 @@
                 .setHspBitScore(95.6)
                 .createBlastHsp();
         
-        // At now, check will return false as it could be not determined
-        assertFalse(uncompleteHsp.equals(o));
+        assertEquals(uncompleteHsp, o);
     }
\ No newline at end of file
