diff --git a/old_product.java b/new_product.java
index 0ef0cd2..a71f22f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,3 @@
-public synchronized long addToTableSize(String table_name, long size) {
-        Long orig_size = this.table_sizes.get(table_name);
-        if (orig_size == null) {
-            orig_size = 0l;
-        }
-        long new_size = orig_size + size;
-        this.setTableSize(table_name, new_size);
-        return (new_size);
+public synchronized void addToTableSize(String table_name, long size) {
+        this.table_sizes.put(table_name, size);
     }
\ No newline at end of file
