diff --git a/old_test.java b/new_test.java
index 5c8f48b..a350755 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,5 @@
 		Url baseUrl = Url.parse("bar/baz");
 		baseUrl.resolveRelative(relative);
 
-		assertEquals("bar?a=b", baseUrl.toString());
-		assertEquals("no empty segment", 1, baseUrl.getSegments().size());
+		assertEquals("bar/?a=b", baseUrl.toString());
 	}
\ No newline at end of file
