diff --git a/old_product.java b/new_product.java
index 4edc53b..f2a39a1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public void resolveRelative(final Url relative)
 	{
 		if (getSegments().size() > 0)
 		{
-			// strip the first non-folder segment
+			// strip the first non-folder segment (if it is not empty)
 			getSegments().remove(getSegments().size() - 1);
 		}
 
@@ -32,6 +32,11 @@ public void resolveRelative(final Url relative)
 			}
 		}
 
+		if (!getSegments().isEmpty() && relative.getSegments().isEmpty())
+		{
+			getSegments().add("");
+		}
+
 		// append the remaining relative segments
 		getSegments().addAll(relative.getSegments());
 
