diff --git a/old_test.java b/new_test.java
index 618ba6b..1dc294f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,11 +3,11 @@
         NvdCveInfo cve = new NvdCveInfo();
         cve.setId("modified");
         cve.setNeedsUpdate(true);
-        cve.setUrl(Settings.getString(Settings.KEYS.CVE_MODIFIED_20_URL));
-        cve.setOldSchemaVersionUrl(Settings.getString(Settings.KEYS.CVE_MODIFIED_12_URL));
+        cve.setUrl(getSettings().getString(Settings.KEYS.CVE_MODIFIED_20_URL));
+        cve.setOldSchemaVersionUrl(getSettings().getString(Settings.KEYS.CVE_MODIFIED_12_URL));
         ExecutorService processExecutor = null;
         CveDB cveDB = null;
-        DownloadTask instance = new DownloadTask(cve, processExecutor, cveDB, Settings.getInstance());
+        DownloadTask instance = new DownloadTask(cve, processExecutor, cveDB, getSettings());
         Future<ProcessTask> result = instance.call();
         assertNull(result);
     }
\ No newline at end of file
