diff --git a/old_test.java b/new_test.java
index cc2a16e..30d1304 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,7 +15,8 @@
 
     check("if(<a>X</a>[text()]) then 1 else 2", null, "//@axis = 'child'");
     check("<a>X</a>[text()] and <a/>", null, "//@axis = 'child'");
-    check("<a>X</a>[text()] or <a/>", null, "//@axis = 'child'");
+    check("<a>X</a>[text()] or <a/>", null, "//Bln/@value = 'true'");
+    check("<a>X</a>[text()] or <a/>[text()]", null, "//@axis = 'child'");
     check("for $a in <a>X</a> where $a[text()] return $a", null, "//@axis = 'child'");
 
     check("empty(<a>X</a>/.[text()])", null, "//@axis = 'child'");
