diff --git a/old_test.java b/new_test.java
index 1daee0e..d711ea0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,11 @@
 @Test
-	@Verifies(value = "should not fail if cohort already contains patient", method = "addPatientToCohort(Cohort,Patient)")
 	public void addPatientToCohort_shouldNotFailIfCohortAlreadyContainsPatient() throws Exception {
 		executeDataSet(COHORT_XML);
 		
 		// make a patient, add it using the method
-		Patient patientToAdd = new Patient(4);
+		Patient patientToAdd = Context.getPatientService().getPatient(3);
 		service.addPatientToCohort(service.getCohort(2), patientToAdd);
-		assertTrue(service.getCohort(2).contains(4));
+		assertTrue(service.getCohort(2).contains(3));
 		
 		// do it again to see if it fails
 		try {
