diff --git a/old_test.java b/new_test.java
index d99f595..8eb8bf4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,7 +14,7 @@
         Assert.assertTrue(one.getClusters().contains(this.c));
         Assert.assertTrue(two.getClusters().contains(this.c));
         this.c.setCommands(null);
-        Assert.assertNull(this.c.getCommands());
+        Assert.assertThat(this.c.getCommands(), Matchers.empty());
         Assert.assertFalse(one.getClusters().contains(this.c));
         Assert.assertFalse(two.getClusters().contains(this.c));
     }
\ No newline at end of file
