diff --git a/old_product.java b/new_product.java
index d1c5c9a..737c6b3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,7 @@
 public static Matrix forwardAlgorithm(HmmModel model, int[] observations,
                                         boolean scaled) {
 
-    DenseMatrix alpha = new DenseMatrix(observations.length, model
-        .getNrOfHiddenStates());
+    Matrix alpha = new DenseMatrix(observations.length, model.getNrOfHiddenStates());
 
     forwardAlgorithm(alpha, model, observations, scaled);
 
