diff --git a/old_test.java b/new_test.java
index ac52281..14a1099 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,6 @@
 @Test
     public void testGetHostgroupId() {
-        System.out.println("getHostgroupId");
-        HostgroupHost instance = null;
-        int expResult = 0;
-        int result = instance.getHostgroupId();
-        assertEquals(expResult, result);
-        // TODO review the generated test code and remove the default call to fail.
-        fail("The test case is a prototype.");
+        DBObject m = this.buildMongoHostgroupHost();
+        HostgroupHost hh = new HostgroupHost(m);
+        assertEquals(m.get("hostgroup_id"), hh.getHostgroupId());
     }
\ No newline at end of file
