diff --git a/old_test.java b/new_test.java
index 406c7b5..d4122ae 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
         mAnim1.setRepeatCount(ValueAnimator.INFINITE);
         Animator[] animatorArray = {mAnim1, mAnim2};
 
-        TimeInterpolator interpolator = new AccelerateDecelerateInterpolator();
+        Interpolator interpolator = new AccelerateDecelerateInterpolator();
         mAnimatorSet = new AnimatorSet();
         mAnimatorSet.playTogether(animatorArray);
         mAnimatorSet.setInterpolator(interpolator);
