diff --git a/old_product.java b/new_product.java
index 8ceb8d8..95d4a55 100644
--- a/old_product.java
+++ b/new_product.java
@@ -30,7 +30,7 @@ public Collection[] getCollections() throws SQLException
         {
             while (tri.hasNext())
             {
-                TableRow row = tri.next();
+                TableRow row = tri.next(ourContext);
 
                 // First check the cache
                 Collection fromCache = (Collection) ourContext.fromCache(
