diff --git a/old_test.java b/new_test.java
index 6fc9b2d..73ecaa7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,22 +1,22 @@
 @Test
     public void testGetCollections() throws Exception
     {
-        new NonStrictExpectations(AuthorizeManager.class)
+        new NonStrictExpectations(authorizeService.getClass())
         {{
             // Allow current Community ADD perms
-            AuthorizeManager.authorizeAction((Context) any, (Community) any,
+            authorizeService.authorizeAction((Context) any, (Community) any,
                     Constants.ADD); result = null;
             // Allow current Community WRITE perms
-            AuthorizeManager.authorizeAction((Context) any, (Community) any,
+            authorizeService.authorizeAction((Context) any, (Community) any,
                     Constants.WRITE); result = null;
         }};
 
         //empty by default
         assertThat("testGetCollections 0",c.getCollections(), notNullValue());
-        assertTrue("testGetCollections 1", c.getCollections().length == 0);
+        assertTrue("testGetCollections 1", c.getCollections().size() == 0);
 
-        Collection result = c.createCollection();
+        Collection result = collectionService.create(context, c);
         assertThat("testGetCollections 2",c.getCollections(), notNullValue());
-        assertTrue("testGetCollections 3", c.getCollections().length == 1);
-        assertThat("testGetCollections 4",c.getCollections()[0], equalTo(result));
+        assertTrue("testGetCollections 3", c.getCollections().size() == 1);
+        assertThat("testGetCollections 4",c.getCollections().get(0), equalTo(result));
     }
\ No newline at end of file
