diff --git a/old_test.java b/new_test.java
index 97a73ec..fc89102 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,8 +6,9 @@
       hashtree.add(new LoopController());
       JMeterThread thread = new JMeterThread(hashtree, null, null);
 
-      CollectionProperty prop = JMeterPluginsUtils.tableModelToCollectionProperty(dataModel, UltimateThreadGroup.DATA_PROPERTY);
+      CollectionProperty prop = JMeterPluginsUtils.tableModelRowsToCollectionProperty(dataModel, UltimateThreadGroup.DATA_PROPERTY);
       instance.setData(prop);
+      instance.testStarted();
 
       instance.scheduleThread(thread);
 
