diff --git a/old_test.java b/new_test.java
index cd26256..f37000b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-    public void testColMean() {
-        System.out.println("colMean");
+    public void testColMeans() {
+        System.out.println("colMeans");
         double[][] A = {
             {0.7220180, 0.07121225, 0.6881997},
             {-0.2648886, -0.89044952, 0.3700456},
@@ -8,7 +8,7 @@
         };
         double[] r = {-0.06067647, -0.12325383, 0.56076753};
 
-        double[] result = Math.colMean(A);
+        double[] result = Math.colMeans(A);
         for (int i = 0; i < r.length; i++) {
             assertEquals(result[i], r[i], 1E-7);
         }
