diff --git a/old_product.java b/new_product.java
index 7941097..b15e72f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -23,16 +23,21 @@ public void validate(Object obj, Errors errors) {
 				Set<String> statesAndStartDates = new HashSet<String>();
 				PatientState latestState = null;
 				boolean foundCurrentPatientState = false;
+				boolean foundStateWithNullStartDate = false;
 				for (PatientState patientState : patientStates) {
 					if (patientState.isVoided())
 						continue;
 					
 					String missingRequiredFieldCode = null;
-					//only the intial state can have a null start date
-					if (patientState.getStartDate() == null && !patientState.getState().getInitial())
-						missingRequiredFieldCode = "general.dateStart";
-					else if (patientState.getState() == null)
+					//only the initial state can have a null start date
+					if (patientState.getStartDate() == null) {
+						if (foundStateWithNullStartDate)
+							missingRequiredFieldCode = "general.dateStart";
+						else
+							foundStateWithNullStartDate = true;
+					} else if (patientState.getState() == null) {
 						missingRequiredFieldCode = "State.state";
+					}
 					
 					if (missingRequiredFieldCode != null) {
 						errors.rejectValue("states", "PatientState.error.requiredField", new Object[] { mss
