diff --git a/old_test.java b/new_test.java
index b796496..3632f84 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,7 @@
 @Test
   public void parse() throws Exception {
     context.options.set(MainOptions.STRIPNS, true);
-    final SerializerOptions sopts = new SerializerOptions();
-    sopts.set(SerializerOptions.INDENT, YesNo.NO);
-    context.options.set(MainOptions.SERIALIZER, sopts);
+    context.options.set(MainOptions.SERIALIZER, SerializerOptions.get(false));
 
     final String doc = "<e xmlns='A'><b:f xmlns:b='B'/></e>";
     for(final boolean b : new boolean[] { false, true }) {
