diff --git a/old_test.java b/new_test.java
index e94127b..84a80cd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,8 +4,8 @@
         ZonedDateTime base = ZonedDateTime.of(ldt, ZONE_0100);
         ZonedDateTime test = base.with(new TimeAdjuster() {
             public LocalTime adjustTime(LocalTime time) {
-                return time.withHourOfDay(1);
+                return time.withHour(1);
             }
         }, ZoneResolvers.retainOffset());
-        assertEquals(test, ZonedDateTime.of(ldt.withHourOfDay(1), ZONE_0100));
+        assertEquals(test, ZonedDateTime.of(ldt.withHour(1), ZONE_0100));
     }
\ No newline at end of file
