diff --git a/old_test.java b/new_test.java
index fb71856..aa125cf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
-    public void test_withHourOfDay_normal() {
+    public void test_withHour_normal() {
         OffsetTime base = OffsetTime.of(11, 30, 59, OFFSET_PONE);
-        OffsetTime test = base.withHourOfDay(15);
+        OffsetTime test = base.withHour(15);
         assertEquals(test, OffsetTime.of(15, 30, 59, OFFSET_PONE));
     }
\ No newline at end of file
