diff --git a/old_product.java b/new_product.java
index c81e0a5..ab69214 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public LocalDateTime withHourOfDay(int hourOfDay) {
-        LocalTime newTime = time.withHourOfDay(hourOfDay);
+public LocalDateTime withHour(int hour) {
+        LocalTime newTime = time.withHour(hour);
         return with(date, newTime);
     }
\ No newline at end of file
