diff --git a/old_product.java b/new_product.java
index 94fb3fe..8dbc678 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static double smog_index(String strText) {
-        strText = clean_text(strText);
-        return PHPfunctions.round(1.043 * Math.sqrt((words_with_three_syllables(strText) * (30.0 / sentence_count(strText))) + 3.1291), 1);
+public static double smogIndex(String strText) {
+        strText = cleanText(strText);
+        return PHPfunctions.round(1.043 * Math.sqrt((wordsWithThreeSyllables(strText) * (30.0 / sentenceCount(strText))) + 3.1291), 1);
     }
\ No newline at end of file
