diff --git a/old_test.java b/new_test.java
index 8a3a1f9..467ee85 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,6 @@
         logger.info("smog_index");
         String strText = TEST_STRING;
         double expResult = 14.1;
-        double result = ReadabilityStatistics.smog_index(strText);
+        double result = ReadabilityStatistics.smogIndex(strText);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
