diff --git a/old_product.java b/new_product.java
index 2c2d726..a4f745f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static void horizontal(Kernel1D_I32 kernel, GrayS16 image, GrayI16 dest , int skip ) {
+public static void horizontal(Kernel1D_S32 kernel, GrayS16 image, GrayI16 dest , int skip ) {
 		checkParameters(image, dest, skip);
 
 		if( kernel.width >= image.width ) {
