diff --git a/old_test.java b/new_test.java
index c021ff8..49c1dc9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
     public void collectMigrations() {
         MigrationResolver migrationResolver = new MigrationResolver() {
-            public List<ExecutableMigration> resolveMigrations() {
-                List<ExecutableMigration> migrations = new ArrayList<ExecutableMigration>();
+            public List<MigrationInfoImpl> resolveMigrations() {
+                List<MigrationInfoImpl> migrations = new ArrayList<MigrationInfoImpl>();
 
                 migrations.add(createTestMigration(com.googlecode.flyway.core.api.MigrationType.JAVA, "1", "Description", "Migration1", 123));
                 migrations.add(createTestMigration(com.googlecode.flyway.core.api.MigrationType.JAVA, "1", "Description", "Migration1", 123));
@@ -13,6 +13,6 @@
         Collection<MigrationResolver> migrationResolvers = new ArrayList<MigrationResolver>();
         migrationResolvers.add(migrationResolver);
 
-        Collection<ExecutableMigration> migrations = CompositeMigrationResolver.collectMigrations(migrationResolvers);
+        Collection<MigrationInfoImpl> migrations = CompositeMigrationResolver.collectMigrations(migrationResolvers);
         assertEquals(2, migrations.size());
     }
\ No newline at end of file
