diff --git a/old_product.java b/new_product.java
index 753a7fe..5f5012f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,5 @@
 @Override
     public void addToGroup(String featureId, String groupName) {
-        getCacheManager().evict(featureId);
         getTarget().addToGroup(featureId, groupName);
-
+        getCacheManager().evict(featureId);
     }
\ No newline at end of file
