diff --git a/old_test.java b/new_test.java
index 09153cd..50e34e5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
 		LongStateStorage stateHandleProvider = new LongStateStorage();
 
 		ZooKeeperStateHandleStore<Long> store = new ZooKeeperStateHandleStore<>(
-				ZooKeeper.getClient(), stateHandleProvider);
+				ZooKeeper.getClient(), stateHandleProvider, Executors.directExecutor());
 
 		// Config
 		final String pathInZooKeeper = "/testRemove";
