diff --git a/old_product.java b/new_product.java
index 4765488..de869ce 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,5 @@ public void remove(String pathInZooKeeper, BackgroundCallback callback) throws E
 		checkNotNull(pathInZooKeeper, "Path in ZooKeeper");
 		checkNotNull(callback, "Background callback");
 
-		client.delete().deletingChildrenIfNeeded().inBackground(callback).forPath(pathInZooKeeper);
+		client.delete().deletingChildrenIfNeeded().inBackground(callback, executor).forPath(pathInZooKeeper);
 	}
\ No newline at end of file
