diff --git a/old_product.java b/new_product.java
index be6c2e0..f6b4529 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,4 @@
-public void setCreated(final Date created) {
-        //This is to prevent the create time from being updated after
-        //an entity has been persisted and someone is just trying to
-        //update another field in the entity
-        if (created.before(this.created)) {
-            this.created.setTime(created.getTime());
-        }
+@JsonIgnore
+    public void setCreated(final Date created) {
+        LOG.info("Tried to set created time to " + created + " for entity " + this.id + ". Ignoring.");
     }
\ No newline at end of file
