diff --git a/old_test.java b/new_test.java
index ede3c87..481eb8e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,7 +15,7 @@
       Assert.assertEquals(200,
                           GatewayFastTestsSuite.doPost("/v2/apps/WordCount/flows/WordCounter/stop", null)
                             .getStatusLine().getStatusCode());
-      Assert.assertEquals(200, GatewayFastTestsSuite.doDelete("/v2/apps/WordCount").getStatusLine().getStatusCode());
+     // Assert.assertEquals(200, GatewayFastTestsSuite.doDelete("/v2/apps/WordCount").getStatusLine().getStatusCode());
 
       response = GatewayFastTestsSuite.doGet("/v2/apps/WordCount/flows/WordCounter/history");
       Assert.assertEquals(200, response.getStatusLine().getStatusCode());
