diff --git a/old_test.java b/new_test.java
index b3fd430..953f3d0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,15 +2,14 @@
     public void toHTML() throws Exception
     {
         String source = "wiki syntax";
-        String syntaxId = "syntax/x.y";
+        Syntax syntax = new Syntax(new SyntaxType("syntax", "Syntax"), "x.y");
 
         // The source should be parsed.
-        Parser parser = this.mocker.registerMockComponent(Parser.class, syntaxId);
-
+        ContentParser contentParser = this.mocker.getInstance(ContentParser.class);
         XDOM xdom = new XDOM(Collections.<Block>emptyList());
-        when(parser.parse(any(StringReader.class))).thenReturn(xdom);
+        when(contentParser.parse(source, syntax, null)).thenReturn(xdom);
 
-        Assert.assertEquals("", mocker.getComponentUnderTest().toHTML(source, syntaxId));
+        Assert.assertEquals("", mocker.getComponentUnderTest().toHTML(source, syntax.toIdString()));
 
         // Verify that the macro transformations have been executed.
         Transformation macroTransformation = mocker.getInstance(Transformation.class, "macro");
