diff --git a/old_test.java b/new_test.java
index 4c46999..e173351 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,12 +3,12 @@
 		Pageable page = new PageRequest(1, 10);
 
 		ModelMap model = new ModelMap();
-		userController.getUserList(model, null, page, null);
+		userController.getUsers(model, null, page, null);
 
 		model.clear();
-		userController.getUserList(model, "ADMIN", page, null);
+		userController.getUsers(model, Role.ADMIN, page, null);
 
 		model.clear();
-		userController.getUserList(model, null, page, "user");
+		userController.getUsers(model, null, page, "user");
 
 	}
\ No newline at end of file
