diff --git a/old_test.java b/new_test.java
index 8d0f102..c6caab3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
     public void testRun_exception() throws Exception {
-        conn.exceptionOnRead(new IOException("done"));
-        try {
-            peer.run();
-            fail("did not throw");
-        } catch (PeerException e) {
-            assertTrue(e.getCause() instanceof IOException);
-        }
+        expect(channel.close()).andReturn(null);
+        control.replay();
+        
+        handler.exceptionCaught(ctx,
+                new DefaultExceptionEvent(channel, new IOException("proto")));
+
+        control.verify();
     }
\ No newline at end of file
