diff --git a/old_test.java b/new_test.java
index 80ec1f0..588ffa5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,12 @@
 @Test
 	public void saveVisit_shouldBeAbleToAddAnAttributeToAVisit() throws Exception {
+		Date now = new Date();
 		Visit visit = service.getVisit(1);
 		VisitAttributeType attrType = service.getVisitAttributeType(1);
 		VisitAttribute attr = new VisitAttribute();
 		attr.setAttributeType(attrType);
-		attr.setValue(new Date());
+		attr.setValue(now);
 		visit.addAttribute(attr);
 		service.saveVisit(visit);
+		Assert.assertEquals(new SimpleDateFormat("yyyy-MM-dd").format(now), attr.getValueReference());
 	}
\ No newline at end of file
