diff --git a/old_product.java b/new_product.java
index 18069e0..2030a86 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public int removeAllOccurrences(KType e1)
+public int removeAll(KType e1)
     {
         int removed = 0;
         final int last = tail;
@@ -12,16 +12,16 @@ public int removeAllOccurrences(KType e1)
                 removed++;
                 continue;
             }
-
+    
             if (to != from)
             {
                 buffer[to] = buffer[from];
                 buffer[from] = Intrinsics.<KType>defaultKTypeValue();
             }
-
+    
             to = oneRight(to, bufLen);
         }
-
+    
         tail = to;
         return removed;
     }
\ No newline at end of file
