diff --git a/old_product.java b/new_product.java
index e79b8cd..02b3d5c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
-public static <K> Map<K, String> toSingleValueMap(Map<K, String[]> arrayValueMap){
+public static <K, V> Map<K, V> toSingleValueMap(Map<K, V[]> arrayValueMap){
         if (Validator.isNullOrEmpty(arrayValueMap)){
             return Collections.emptyMap();
         }
-        Map<K, String> singleValueMap = newLinkedHashMap(arrayValueMap.size());//保证顺序和 参数 arrayValueMap 顺序相同
-        for (Map.Entry<K, String[]> entry : arrayValueMap.entrySet()){
-            singleValueMap.put(entry.getKey(), Validator.isNotNullOrEmpty(entry.getValue()) ? entry.getValue()[0] : StringUtils.EMPTY);
+        Map<K, V> singleValueMap = newLinkedHashMap(arrayValueMap.size());//保证顺序和 参数 arrayValueMap 顺序相同
+        for (Map.Entry<K, V[]> entry : arrayValueMap.entrySet()){
+            singleValueMap.put(entry.getKey(), null == entry.getValue() ? null : entry.getValue()[0]);
         }
         return singleValueMap;
     }
\ No newline at end of file
