diff --git a/old_test.java b/new_test.java
index a745816..2e08d55 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
     public void testToSingleValueMap(){
-        Map<String, String[]> keyAndArrayMap = new LinkedHashMap<String, String[]>();
-        keyAndArrayMap.put("province", toArray("浙江省", "江苏省"));
-        keyAndArrayMap.put("city", toArray("南通市"));
+        Map<String, String[]> arrayValueMap = MapUtil.newLinkedHashMap(2);
+        arrayValueMap.put("province", toArray("浙江省", "江苏省"));
+        arrayValueMap.put("city", toArray("南通市"));
 
-        Map<String, String> singleValueMap = MapUtil.toSingleValueMap(keyAndArrayMap);
+        Map<String, String> singleValueMap = MapUtil.toSingleValueMap(arrayValueMap);
         assertThat(singleValueMap, allOf(hasEntry("province", "浙江省"), hasEntry("city", "南通市")));
     }
\ No newline at end of file
