diff --git a/old_product.java b/new_product.java
index fa04018..9f21b61 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,6 +5,5 @@ public List<Order> getOrdersByPatient(Patient patient, boolean includeVoided) th
 		List<Patient> patients = new ArrayList<Patient>();
 		patients.add(patient);
 		
-		return getOrders(Order.class, patients, null, includeVoided ? ORDER_STATUS.ANY : ORDER_STATUS.NOTVOIDED, null, null,
-		    null);
+		return getOrders(Order.class, patients, null, null, null, null, null, null, includeVoided);
 	}
\ No newline at end of file
