diff --git a/old_product.java b/new_product.java
index f5ab497..482db1f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -94,16 +94,10 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 			}
 			
 			//concept should be the same as on previous order, same applies to drug for drug orders
-			boolean isDrugOrderAndHasADrug = isDrugOrder && ((DrugOrder) order).getDrug() != null;
-			if (!OpenmrsUtil.nullSafeEquals(order.getConcept(), previousOrder.getConcept())) {
-				throw new APIException("Order.previous.concept");
-			} else if (isDrugOrderAndHasADrug) {
-				DrugOrder drugOrder1 = (DrugOrder) order;
-				DrugOrder drugOrder2 = (DrugOrder) previousOrder;
-				if (!OpenmrsUtil.nullSafeEquals(drugOrder1.getDrug(), drugOrder2.getDrug())) {
-					throw new APIException("Order.previous.drug");
-				}
-			} else if (!order.getOrderType().equals(previousOrder.getOrderType())) {
+			if(!order.hasSameOrderableAs(previousOrder)){
+				throw new APIException("The orderable of the previous order and the new one order don't match");
+			}
+			else if (!order.getOrderType().equals(previousOrder.getOrderType())) {
 				throw new APIException("Order.type.does.not.match");
 			} else if (!order.getCareSetting().equals(previousOrder.getCareSetting())) {
 				throw new APIException("Order.care.setting.does.not.match");
