diff --git a/old_test.java b/new_test.java
index ff959e1..380aef3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public void removeByFieldTest() {
     Assert.assertEquals(3, mSet.getByField(mIntIndex, 1).size());
     Assert.assertEquals(9, mSet.size());
-    Assert.assertTrue(mSet.removeByField(mIntIndex, 1));
+    Assert.assertEquals(3, mSet.removeByField(mIntIndex, 1));
     Assert.assertEquals(6, mSet.size());
     Assert.assertEquals(0, mSet.getByField(mIntIndex, 1).size());
     Assert.assertEquals(3, mSet.getByField(mIntIndex, 0).size());
