diff --git a/old_product.java b/new_product.java
index a6f7258..b2db928 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,15 +1,13 @@
-public boolean removeByField(FieldIndex<T> index, Object value) {
-    synchronized (mLock) {
-      Set<T> toRemove = getByFieldInternal(index, value);
-      if (toRemove == null) {
-        return false;
-      }
-      // Copy the set so that no ConcurrentModificationException happens
-      toRemove = ImmutableSet.copyOf(toRemove);
-      boolean success = true;
-      for (T o : toRemove) {
-        success = success && remove(o);
+public int removeByField(FieldIndex<T> index, Object value) {
+    ConcurrentHashSet<T> toRemove = getByFieldInternal(index, value);
+    if (toRemove == null) {
+      return 0;
+    }
+    int removed = 0;
+    for (T o : toRemove) {
+      if (remove(o)) {
+        removed++;
       }
-      return success;
     }
+    return removed;
   }
\ No newline at end of file
