diff --git a/old_test.java b/new_test.java
index 134b3dd..3eea486 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,18 @@
 @Test
     public void testRequestRoleFor() {
-        // TODO: will be added when CompletableFuture is removed
+        expect(mockService.requestRoleForSync(anyObject())).andReturn(role1).anyTimes();
+        replay(mockService);
+
+        final WebTarget wt = target();
+        final String response = wt.path("mastership/" + deviceId1.toString() +
+                "/request").request().get(String.class);
+        final JsonObject result = Json.parse(response).asObject();
+        assertThat(result, notNullValue());
+
+        assertThat(result.names(), hasSize(1));
+        assertThat(result.names().get(0), is("role"));
+
+        final String role = result.get("role").asString();
+        assertThat(role, notNullValue());
+        assertThat(role, is("MASTER"));
     }
\ No newline at end of file
