diff --git a/old_test.java b/new_test.java
index bb80072..0eec7be 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,7 +20,9 @@
 
 			try {
 
-				cm.requestInstance(jobID, conf, cm.getInstanceTypeByName(LARGE_INSTANCE_TYPE_NAME));
+				Map<InstanceType, Integer> instancem = new HashMap<InstanceType, Integer>();
+				instancem.put(cm.getInstanceTypeByName(LARGE_INSTANCE_TYPE_NAME), 1);
+				cm.requestInstance(jobID, conf, instancem, null);
 
 			} catch (InstanceException ie) {
 				fail(ie.getMessage());
