diff --git a/old_product.java b/new_product.java
index db3c51e..39a8abd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@
       throws IndexingException {
     HashSet<String> set = null;
     
-    for (Entry<Utf8, Utf8> e : page.getInlinks().entrySet()) {
+    for (Entry<CharSequence, CharSequence> e : page.getInlinks().entrySet()) {
       String anchor = TableUtil.toString(e.getValue());
       
       if(anchor.equals(""))
