diff --git a/old_test.java b/new_test.java
index 237fa4e..e2d987f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void testFilter() {
     // check that we get the same values
-    Map<Utf8, ByteBuffer> meta = parseMetaTags(sampleFile, false);
+    Map<CharSequence, ByteBuffer> meta = parseMetaTags(sampleFile, false);
 
     assertEquals(description,
         getMeta(meta, MetaTagsParser.PARSE_META_PREFIX + "description"));
