diff --git a/old_test.java b/new_test.java
index b7965ac..18e0595 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(dataProvider="calendars")
-    public void test_date_comparisons(Chrono chrono) {
+    public void test_date_comparisons(Chronology chrono) {
         List<ChronoLocalDate<?>> dates = new ArrayList<>();
 
         ChronoLocalDate<?> date = chrono.date(LocalDate.of(1900, 1, 1));
@@ -22,9 +22,9 @@
         dates.add(date.plus(1000, ChronoUnit.YEARS));
 
         // Check these dates against the corresponding dates for every calendar
-        for (Chrono[] clist : data_of_calendars()) {
+        for (Chronology[] clist : data_of_calendars()) {
             List<ChronoLocalDate<?>> otherDates = new ArrayList<>();
-            Chrono chrono2 = clist[0];
+            Chronology chrono2 = clist[0];
             for (ChronoLocalDate<?> d : dates) {
                 otherDates.add(chrono2.date(d));
             }
