diff --git a/old_test.java b/new_test.java
index a2d43ee..8ad83b4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,17 +1,18 @@
 @Test
-    public void validate_shouldFailIfGroupDoesntHaveAnyMembers() throws Exception {
-    	User provider = Context.getUserService().getUser(501);
-		Patient patient = Context.getPatientService().getPatient(6);
+	public void validate_shouldFailIfGroupDoesntHaveAnyMembers() throws Exception {
+		User provider = new User();
+		provider.setUsername("test");
+		Patient patient = new Patient(6);
 		
 		OrderGroup group = new OrderGroup(null, patient);
 		group.setCreator(provider);
 		group.setDateCreated(new Date());
 		
 		Errors errors = new BindException(group, "group");
-		validator.validate(group, errors);
+		getValidator().validate(group, errors);
 		
 		Assert.assertTrue(errors.hasErrors());
 		Assert.assertFalse(errors.hasFieldErrors("creator"));
 		Assert.assertFalse(errors.hasFieldErrors("patient"));
 		Assert.assertFalse(errors.hasFieldErrors("dateCreated"));
-    }
\ No newline at end of file
+	}
\ No newline at end of file
