diff --git a/old_product.java b/new_product.java
index ca40b0f..740d281 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @Override
-	public void validate(Object arg0, Errors errors) {
-		OrderGroup group = (OrderGroup) arg0;
+	public void validate(Object target, Errors errors) {
+		OrderGroup group = (OrderGroup) target;
 		if (group == null) {
 			errors.reject("group", "error.general");
 		} else {
