diff --git a/old_test.java b/new_test.java
index a932ff3..d0d7a44 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,10 +2,13 @@
     public void testActivate() throws Exception {
         Ignite ignite = startGrids(1);
 
-        assertFalse(ignite.cluster().active());
+        injectTestSystemOut();
+
+        assertEquals(INACTIVE, ignite.cluster().state());
 
         assertEquals(EXIT_CODE_OK, execute("--activate"));
 
-        assertTrue(ignite.cluster().active());
-        assertFalse(ignite.cluster().readOnly());
+        assertEquals(ACTIVE, ignite.cluster().state());
+
+        assertContains(log, testOut.toString(), "Command deprecated. Use --set-state instead.");
     }
\ No newline at end of file
