diff --git a/old_product.java b/new_product.java
index 53d32c6..a01a1a7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
 public boolean activate() {
-        return stateChangeReq != null && stateChangeReq.activeChanged() && stateChangeReq.activate();
+        return stateChangeReq != null && stateChangeReq.activeChanged() && active(stateChangeReq.state());
     }
\ No newline at end of file
