diff --git a/old_product.java b/new_product.java
index a0a7d2b..1add11c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -43,7 +43,12 @@
 			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "voidReason", "error.null");
 		}
 		if (person.getDead()) {
-			ValidationUtils.rejectIfEmpty(errors, "causeOfDeath", "Person.dead.causeOfDeathNull");
+			if(person.getCauseOfDeath() != null && person.getCauseOfDeathNonCoded() != null) {
+				errors.rejectValue("causeOfDeath", "Person.dead.shouldHaveOnlyOneCauseOfDeathOrCauseOfDeathNonCodedSet");
+			}
+			else if(person.getCauseOfDeath() == null && person.getCauseOfDeathNonCoded() == null) {
+				errors.rejectValue("causeOfDeath", "Person.dead.causeOfDeathAndCauseOfDeathNonCodedNull");
+			}
 		}
 		
 		ValidateUtil.validateFieldLengths(errors, Person.class, "gender", "personVoidReason");
