diff --git a/old_product.java b/new_product.java
index 2fb8311..379f65e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -31,5 +31,5 @@ public void repairSurvivors()
         }
         tmLog.debug(m_whoami + "finished queuing " + queued + " replica repair messages.");
 
-        m_promotionResult.set(m_maxSeenTxnId);
+        m_promotionResult.set(Pair.of(m_maxSeenTxnId, m_maxSeenUniqueId));
     }
\ No newline at end of file
