diff --git a/old_test.java b/new_test.java
index c6a269d..2fd6619 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testNotEqProp() {
-        assertTrue(instance.notEqProp("id", "name").getQueryCriterions()
-                .contains(new NotEqPropCriterion("id", "name")));
+    	assertEquals(new NotEqPropCriterion("id", "name"), instance.notEqProp("id", "name").getQueryCriterion());
     }
\ No newline at end of file
