diff --git a/old_test.java b/new_test.java
index 929f14e..0fc10e1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,8 @@
     ok(new Set(MainOptions.CHOP, true));
     ok(new Set("chop", true));
     ok(new Set("runs", 1));
+    ok(new Set(MainOptions.ATTRTOKENIZE, ""));
+    ok(new Set(MainOptions.ATTRTOKENIZE, "id"));
     no(new Set("runs", true));
     no(new Set(NAME2, NAME2));
   }
\ No newline at end of file
