diff --git a/old_product.java b/new_product.java
index f7033a9..84f747f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,7 @@ private void set(final IndexType type, final ValueIndex index) {
     switch(type) {
       case TEXT:      textIndex = index; break;
       case ATTRIBUTE: attrIndex = index; break;
+      case ATTTOKEN:  attrTokenIndex = index; break;
       case FULLTEXT:  ftxtIndex = index; break;
       default:        break;
     }
