diff --git a/old_test.java b/new_test.java
index 75413c8..50120b2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,5 +5,5 @@
         SdkGenerator generator = factory.createSdkGenerator(SdkPlatform.JAVA);
         Assert.assertNotNull(generator);
         generator = factory.createSdkGenerator(SdkPlatform.CPP);
-        Assert.assertNull(generator);
+        Assert.assertNotNull(generator);
     }
\ No newline at end of file
