diff --git a/old_product.java b/new_product.java
index 1b0e030..2dc9786 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,10 @@
 public static SdkGenerator createSdkGenerator(SdkPlatform sdkPlatform) {
         switch (sdkPlatform) {
         case JAVA:
-            return new JavaSdkGenerator(sdkPlatform);
         case ANDROID:
             return new JavaSdkGenerator(sdkPlatform);
+        case CPP:
+            return new CppSdkGenerator();
         default:
             return null;
         }
