diff --git a/old_test.java b/new_test.java
index 0b54504..472fc79 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,17 @@
 @Test
-	public void setTransform() {
-		fail("Implement");
+	public void setTransform_input() {
+		PointTransformHomography_F32 H = new PointTransformHomography_F32();
+		PixelTransform_F32 transform = new PointToPixelTransform_F32(H);
+
+		RefinePolygonLineToImage alg = createAlg(4,true, ImageUInt8.class);
+
+		// correct example
+		alg.setTransform(20,30,transform);
+
+		// bad example
+		try {
+			H.getModel().a11 = 2;
+			alg.setTransform(20,30,transform);
+			fail("Should have thrown exception");
+		} catch( RuntimeException ignore ) {}
 	}
\ No newline at end of file
