diff --git a/old_product.java b/new_product.java
index ae0fd22..352813e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@ public void setTransform( int width , int height , PixelTransform_F32 transform
 		float y1 = rect.y0 + rect.height;
 
 		if( rect.getX() < 0 || rect.getY() < 0 || x1 > width || y1 > height ) {
-			throw new IllegalArgumentException("You failed the idiot proof test! RTFM! The undistorted image "+
+			throw new IllegalArgumentException("You failed the idiot test! RTFM! The undistorted image "+
 					"must be contained by the same bounds as the input distorted image");
 		}
 
