diff --git a/old_product.java b/new_product.java
index 69c1f87..57dbf6c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -21,9 +21,9 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 		
 		if (!isDiscontinueOrReviseOrder(order)) {
 			List<Order> activeOrders = getActiveOrders(order.getPatient(), null, order.getCareSetting(), null);
-			for (Order o : activeOrders) {
-				if (o.getConcept().equals(concept)) {
-					throw new APIException("Cannot have more than one active order for the same concept and care setting");
+			for (Order activeOrder : activeOrders) {
+				if (order.hasSameOrderableAs(activeOrder)) {
+					throw new APIException("Cannot have more than one active order for the same orderable and care setting");
 				}
 			}
 		}
