diff --git a/old_product.java b/new_product.java
index 4f8c8aa..2ac142a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public OffsetDateTime minus(Duration duration) {
-        LocalDateTime newDT = dateTime.minus(duration);
+        LocalDateTime newDT = dateTime.minusSeconds(duration.getSeconds()).minusNanos(duration.getNano());
         return (newDT == dateTime ? this : new OffsetDateTime(newDT, offset));
     }
\ No newline at end of file
