diff --git a/old_test.java b/new_test.java
index e19ff0d..48f1b0f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -25,7 +25,7 @@
     query(_DB_OPEN.args(dbname) + "/root()", "<dummy/>");
 
     // try to create DB twice during same query
-    error(_DB_CREATE.args(dbname) + "," + _DB_CREATE.args(dbname), Err.BXDB_CREATE);
+    error(_DB_CREATE.args(dbname) + ',' + _DB_CREATE.args(dbname), Err.BXDB_CREATE);
 
     // create DB from file
     query(_DB_CREATE.args(dbname, FILE, "in/"));
@@ -53,13 +53,13 @@
 
     // try to access non-existing DB (create is supposed to be called last)
     query(_DB_DROP.args(dbname));
-    error(_DB_CREATE.args(dbname) + "," + _DB_DROP.args(dbname), Err.BXDB_OPEN);
+    error(_DB_CREATE.args(dbname) + ',' + _DB_DROP.args(dbname), Err.BXDB_OPEN);
 
     // run update on existing DB then drop it and create a new one
     query(_DB_CREATE.args(dbname, "<a/>", "a.xml"));
     query("insert node <dummy/> into " + _DB_OPEN.args(dbname));
     query(_DB_CREATE.args(dbname, "<dummy/>", "t1.xml") +
-        ", insert node <dummy/> into " + _DB_OPEN.args(dbname) + "," +
+        ", insert node <dummy/> into " + _DB_OPEN.args(dbname) + ',' +
         _DB_DROP.args(dbname));
     query(_DB_OPEN.args(dbname) + "/root()", "<dummy/>");
 
