diff --git a/old_test.java b/new_test.java
index 55af3bf..77ddea9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,10 @@
 @Test
     public void getRememberedPrincipals() {
         HttpServletRequest mockRequest = createMock(HttpServletRequest.class);
-        WebUtils.bind(mockRequest);
         HttpServletResponse mockResponse = createMock(HttpServletResponse.class);
-        WebUtils.bind(mockResponse);
+        Map<String,Object> context = new HashMap<String,Object>();
+        context.put(SubjectFactory.SERVLET_REQUEST, mockRequest);
+        context.put(SubjectFactory.SERVLET_RESPONSE, mockResponse);
 
         expect(mockRequest.getAttribute(ShiroHttpServletRequest.IDENTITY_REMOVED_KEY)).andReturn(null);
 
@@ -26,7 +27,7 @@
         replay(mockRequest);
 
         WebRememberMeManager mgr = new WebRememberMeManager();
-        PrincipalCollection collection = mgr.getRememberedPrincipals();
+        PrincipalCollection collection = mgr.getRememberedPrincipals(context);
 
         verify(mockRequest);
 
