diff --git a/old_product.java b/new_product.java
index e2db29a..a49c4f3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,26 +1,8 @@
-public PrincipalCollection getRememberedPrincipals() {
+public PrincipalCollection getRememberedPrincipals(Map subjectContext) {
         try {
-
-            PrincipalCollection principals = null;
-            byte[] bytes = getSerializedRememberedIdentity();
-            if (bytes != null) {
-                if (getCipher() != null) {
-                    bytes = decrypt(bytes);
-                }
-                try {
-                    principals = deserialize(bytes);
-                } catch (SerializationException e) {
-                    if (log.isWarnEnabled()) {
-                        log.warn("Unable to deserialize stored identity byte array.  Remembered identity " +
-                                "cannot be reconstituted!  This is a non fatal exception as RememberMe identity services " +
-                                "are not considered critical and execution can continue as normal, but please " +
-                                "investigate and resolve to prevent seeing this message again.", e);
-                    }
-                }
-            }
-            return principals;
-
-        } catch (Exception e) {
-            return onRememberedPrincipalFailure(e);
+            byte[] bytes = getRememberedSerializedIdentity(subjectContext);
+            return convertBytesToPrincipals(bytes, subjectContext);
+        } catch (RuntimeException re) {
+            return onRememberedPrincipalFailure(re, subjectContext);
         }
     }
\ No newline at end of file
