diff --git a/old_test.java b/new_test.java
index 7aedc56..cb24d45 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,20 +1,18 @@
-@Test(description = "PUT /admin/vdc/{id}", enabled=false)
+@Test(description = "PUT /admin/vdc/{id}", enabled = false)
    public void testEditVdc() throws Exception {
-      String origName = vdcApi.getVdc(adminVdcUri).getName();
+      String origName = lazyGetVdc().getName();
       String newName = name("a");
       Exception exception = null;
-      
-      AdminVdc vdc = AdminVdc.builder()
-               .name(newName)
-               .build();
-      
+
+      AdminVdc vdc = AdminVdc.builder().name(newName).build();
+
       try {
-         Task task = vdcApi.editVdc(adminVdcUri, vdc);
+         Task task = vdcApi.update(vdcUrn, vdc);
          assertTaskSucceeds(task);
-         
-         AdminVdc modified = vdcApi.getVdc(adminVdcUri);
+
+         AdminVdc modified = vdcApi.get(vdcUrn);
          assertEquals(modified.getName(), newName);
-          
+
          // parent type
          Checks.checkAdminVdc(vdc);
       } catch (Exception e) {
@@ -22,7 +20,7 @@
       } finally {
          try {
             AdminVdc restorableVdc = AdminVdc.builder().name(origName).build();
-            Task task = vdcApi.editVdc(adminVdcUri, restorableVdc);
+            Task task = vdcApi.update(vdcUrn, restorableVdc);
             assertTaskSucceeds(task);
          } catch (Exception e) {
             if (exception != null) {
