diff --git a/old_test.java b/new_test.java
index 0dd4ebf..c97d644 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,13 @@
 @Test
-	@Verifies(value = "should remove membership from cohort", method = "removeMemberShipFromCohort(Cohort, CohortMembership)")
 	public void removeMembershipFromCohort_shouldRemoveMembershipFromCohort() throws Exception {
 		executeDataSet(COHORT_XML);
 
-		CohortMembership memberToAddThenRemove = new CohortMembership(new Patient(4));
+		CohortMembership memberToAddThenRemove = new CohortMembership(4);
 		service.addMembershipToCohort(service.getCohort(1), memberToAddThenRemove);
-		assertTrue(service.getCohort(1).contains(memberToAddThenRemove.getPatient()));
+		assertTrue(service.getCohort(1).contains(memberToAddThenRemove.getPatientId()));
 		assertNull(memberToAddThenRemove.getEndDate());
 
-		service.removeMemberShipFromCohort(service.getCohort(1), memberToAddThenRemove);
+		service.removeMembershipFromCohort(service.getCohort(1), memberToAddThenRemove);
 		assertNotNull(memberToAddThenRemove.getEndDate());
+		assertFalse(service.getCohort(1).contains(memberToAddThenRemove.getPatientId()));
 	}
\ No newline at end of file
