diff --git a/old_product.java b/new_product.java
index 61edc8b..87f6f15 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
 @Authorized( {PrivilegeConstants.EDIT_COHORTS })
-	Cohort removeMemberShipFromCohort(Cohort cohort, CohortMembership cohortMembership);
\ No newline at end of file
+	Cohort removeMembershipFromCohort(Cohort cohort, CohortMembership cohortMembership);
\ No newline at end of file
