diff --git a/old_test.java b/new_test.java
index d733643..d8eb55a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,14 +2,11 @@
     public void recreateTokenFromBytes() throws Exception {
         mock.checking(new Expectations() {
             {
-                allowing(testTokenService).recreateTokenFromBytes(tokenBytes);
-                allowing(testTokenService).recreateTokenFromBytes(tokenBytes, removeAttrs);
+                one(testTokenService).recreateTokenFromBytes(tokenBytes);
             }
         });
 
         assertNotNull("Mock should return a non-null token",
-                      tokenManager.recreateTokenFromBytes(tokenBytes));
-        assertNotNull("Mock should return a non-null token",
-                      tokenManager.recreateTokenFromBytes(tokenBytes, removeAttrs));
+                          tokenManager.recreateTokenFromBytes(tokenBytes));
 
     }
\ No newline at end of file
