diff --git a/old_product.java b/new_product.java
index 9658c93..5e27a12 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
 @FFDCIgnore(InvalidTokenException.class)
-    public abstract Token recreateTokenFromBytes(byte[] tokenBytes, String... removeAttributes) throws InvalidTokenException, TokenExpiredException;
\ No newline at end of file
+    public abstract Token recreateTokenFromBytes(byte[] tokenBytes) throws InvalidTokenException, TokenExpiredException;
\ No newline at end of file
