diff --git a/old_product.java b/new_product.java
index 934e925..cf0e004 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-private static void intersect(final SeqType a, final SeqType b, final SeqType r) {
-    assertTrue(a + ", " + b, r != null ? a.intersect(b).eq(r) : a.intersect(b) == null);
-    assertTrue(b + ", " + a, r != null ? b.intersect(a).eq(r) : b.intersect(a) == null);
+private static void intersect(final SeqType st1, final SeqType st2, final SeqType expected) {
+    eq(st1.intersect(st2), expected);
+    eq(st2.intersect(st1), expected);
   }
\ No newline at end of file
