diff --git a/old_test.java b/new_test.java
index da8f880..36797a9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-  public void defaultsTest() {
+  public void defaultsTest() throws IOException {
     ClientContext.getConf().set(Constants.SECURITY_AUTHENTICATION_TYPE, "SIMPLE");
     ClientContext.getConf().set(Constants.SECURITY_LOGIN_USERNAME, "foo");
     // Use IdentityUserGroupMapping to map user "foo" to group "foo".
