diff --git a/old_product.java b/new_product.java
index 1e1c663..dd4af26 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,10 @@
-public static CompleteUfsFileOptions defaults() {
+public static CompleteUfsFileOptions defaults() throws IOException {
     PermissionStatus ps = PermissionStatus.defaults();
-    try {
-      // Set user and group from user login module, apply default file UMask.
-      ps.setUserFromLoginModule(ClientContext.getConf()).applyFileUMask(ClientContext.getConf());
-      // TODO(chaomin): set permission based on the alluxio file. Not needed for now since the
-      // file is always created with default permission.
-    } catch (IOException e) {
-      throw Throwables.propagate(e);
-    }
+    // Set user and group from user login module, apply default file UMask.
+    ps.setUserFromLoginModule(ClientContext.getConf()).applyFileUMask(ClientContext.getConf());
+    // TODO(chaomin): set permission based on the alluxio file. Not needed for now since the
+    // file is always created with default permission.
+
     return new CompleteUfsFileOptions(ps.getUserName(), ps.getGroupName(),
         ps.getPermission().toShort());
   }
\ No newline at end of file
