diff --git a/old_test.java b/new_test.java
index 5f893ce..0414a5e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -21,7 +21,7 @@
 		// make it fail to return false
 		expect(mockConnection.getMetaData()).andThrow(new SQLException()).once();
 
-		expect(mockConnection.getOriginatingPartition()).andReturn(mockPartition).once();
+		expect(mockConnection.getOriginatingPartition()).andReturn(mockPartition).anyTimes();
 		// we're about to destroy this connection, so we can create new ones.
 		mockPartition.setUnableToCreateMoreTransactions(false);
 		expectLastCall().once();
