diff --git a/old_product.java b/new_product.java
index 991611f..d125378 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,12 @@
-protected void internalReleaseConnection(Connection conn) throws InterruptedException, SQLException {
+protected void internalReleaseConnection(ConnectionHandle connectionHandle) throws InterruptedException, SQLException {
 
-		ConnectionHandle connectionHandle = (ConnectionHandle)conn;
 		connectionHandle.clearStatementHandles(false);
 		if (connectionHandle.isPossiblyBroken() && !isConnectionHandleAlive(connectionHandle)){
 
 			ConnectionPartition connectionPartition = connectionHandle.getOriginatingPartition();
-			connectionPartition.setUnableToCreateMoreTransactions(false);
 			maybeSignalForMoreConnections(connectionPartition);
-			connectionPartition.updateCreatedConnections(-1);
+
+			postDestroyConnection(connectionHandle);
 			return; // don't place back in queue - connection is broken!
 		}
 
