diff --git a/old_test.java b/new_test.java
index 2efc5f5..e110f66 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,11 +12,11 @@
     }
 
     // invalid names
-    error(_DB_COPY.args("x", " ''"), BXDB_NAME_X);
-    error(_DB_COPY.args(" ''", "x"), BXDB_NAME_X);
+    error(_DB_COPY.args("x", " ''"), DB_NAME_X);
+    error(_DB_COPY.args(" ''", "x"), DB_NAME_X);
 
     // same name is disallowed
-    error(_DB_COPY.args(NAME, NAME), BXDB_SAME_X);
+    error(_DB_COPY.args(NAME, NAME), DB_CONFLICT4_X);
     // source database does not exist
-    error(_DB_COPY.args(NAME + "copy", NAME), BXDB_WHICH_X);
+    error(_DB_COPY.args(NAME + "copy", NAME), DB_OPEN1_X);
   }
\ No newline at end of file
