diff --git a/old_product.java b/new_product.java
index 7f91117..9e0e4dd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,4 @@
 public void setGroupMembers(Set<Obs> groupMembers) {
-		if (CollectionUtils.isNotEmpty(this.groupMembers) && CollectionUtils.isNotEmpty(groupMembers)) {
-			setDirty(!CollectionUtils.disjunction(this.groupMembers, groupMembers).isEmpty());
-		} else if (CollectionUtils.isEmpty(this.groupMembers) && CollectionUtils.isNotEmpty(groupMembers)) {
-			setDirty(true);
-		} else if (CollectionUtils.isNotEmpty(this.groupMembers) && CollectionUtils.isEmpty(groupMembers)) {
-			setDirty(true);
-		}
 		this.groupMembers = new HashSet<Obs>(groupMembers); //Copy over the entire list
 		
 	}
\ No newline at end of file
