diff --git a/old_test.java b/new_test.java
index 8ea86e3..7dabe56 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
-	public void setGroupMembers_shouldMarkTheExistingObsAsDirtyWhenTheSetIsChangedFromNullToANonEmptyOne() throws Exception {
+	public void setGroupMembers_shouldNotMarkTheExistingObsAsDirtyWhenTheSetIsChangedFromNullToANonEmptyOne() throws Exception {
 		Obs obs = new Obs(5);
 		assertNull(Obs.class.getDeclaredField("groupMembers").get(obs));
 		Set members = new HashSet<>();
 		members.add(new Obs());
 		obs.setGroupMembers(members);
-		assertTrue(obs.isDirty());
+		assertFalse(obs.isDirty());
 	}
\ No newline at end of file
