diff --git a/old_product.java b/new_product.java
index 6e057fd..b2c0aea 100644
--- a/old_product.java
+++ b/new_product.java
@@ -117,6 +117,10 @@ ClientCacheConfiguration cacheConfiguration(BinaryInputStream in, ProtocolVersio
                                 }
                             ));
                     }
-                ).toArray(new QueryEntity[0]));
+                ).toArray(new QueryEntity[0]))
+                .setExpiryPolicy(
+                    ver.compareTo(V1_6_0) < 0 ? null : reader.readBoolean() ?
+                        new PlatformExpiryPolicy(reader.readLong(), reader.readLong(), reader.readLong()) : null
+                );
         }
     }
\ No newline at end of file
