diff --git a/old_product.java b/new_product.java
index 0291af3..12efec3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,8 @@
 protected static String eval(final String query) throws QueryException, IOException {
     final ArrayOutput ao = new ArrayOutput();
     try(QueryProcessor qp = new QueryProcessor(query, BASEURI, context)) {
+      // update flag will be set in parsing step
+      qp.parse();
       qp.register(context);
       try(Serializer ser = qp.getSerializer(ao)) {
         qp.value().serialize(ser);
