diff --git a/old_test.java b/new_test.java
index 34685b9..2b04d87 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-	@Verifies(value = "should return null if no object found with given uuid", method = "getOrderTypeByUuid(String)")
-	public void getOrderTypeByUuid_shouldReturnNullIfNoObjectFoundWithGivenUuid() throws Exception {
-		Assert.assertNull(Context.getOrderService().getOrderTypeByUuid("some invalid uuid"));
+	@Verifies(value = "should return null if no object found with given order number", method = "getOrderByOrderNumber(String)")
+	public void getOrderByOrderNumber_shouldReturnNullIfNoObjectFoundWithGivenOrderNumber() throws Exception {
+		Assert.assertNull(Context.getOrderService().getOrderByOrderNumber("some invalid order number"));
 	}
\ No newline at end of file
