diff --git a/old_test.java b/new_test.java
index 6d933de..9be91ce 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testValidate() throws GenieException {
+    public void testValidate() throws GeniePreconditionException {
         this.a = new Application(NAME, USER, ApplicationStatus.ACTIVE, VERSION);
         this.a.validate();
     }
\ No newline at end of file
