diff --git a/old_product.java b/new_product.java
index fbb36c1..acfcf8b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,9 @@
 private void validate(
-            final ApplicationStatus status) throws GenieException {
+            final ApplicationStatus status, final String error) throws GeniePreconditionException {
         final StringBuilder builder = new StringBuilder();
+        if (StringUtils.isNotBlank(error)) {
+            builder.append(error);
+        }
         if (status == null) {
             builder.append("No application status entered and is required.\n");
         }
@@ -9,6 +12,6 @@ private void validate(
             builder.insert(0, "Application configuration errors:\n");
             final String msg = builder.toString();
             LOG.error(msg);
-            throw new GenieException(HttpURLConnection.HTTP_BAD_REQUEST, msg);
+            throw new GeniePreconditionException(msg);
         }
     }
\ No newline at end of file
