diff --git a/old_test.java b/new_test.java
index fb36b7a..c9a23c6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,11 @@
 @Test
     public void testNearest() {
         System.out.println("nearest");
+
+        double[][] data = Matrix.randn(1000, 10).toArray();
+        CoverTree<double[]> coverTree = new CoverTree<>(data, new EuclideanDistance());
+        LinearSearch<double[]> naive = new LinearSearch<>(data, new EuclideanDistance());
+
         for (int i = 0; i < data.length; i++) {
             Neighbor n1 = coverTree.nearest(data[i]);
             Neighbor n2 = naive.nearest(data[i]);
