diff --git a/old_product.java b/new_product.java
index 20b9254..c53af79 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public static <T> T defaultIfNull(final T object, final Supplier<T> defaultSupplier) {
+public static <T> T getIfNull(final T object, final Supplier<T> defaultSupplier) {
         return object != null ? object : defaultSupplier == null ? null : defaultSupplier.get();
     }
\ No newline at end of file
