diff --git a/old_test.java b/new_test.java
index 09e24c7..406e029 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void test_convertClassNamesToClasses_List() {
-        List<String> list = new ArrayList<String>();
+        final List<String> list = new ArrayList<String>();
         List<Class<?>> result = ClassUtils.convertClassNamesToClasses(list);
         assertEquals(0, result.size());
 
@@ -14,11 +14,12 @@
         assertEquals(Object.class, result.get(2));
 
         @SuppressWarnings("unchecked") // test what happens when non-generic code adds wrong type of element
+        final
         List<Object> olist = (List<Object>)(List<?>)list;
         olist.add(new Object());
         try {
             ClassUtils.convertClassNamesToClasses(list);
             fail("Should not have been able to convert list");
-        } catch (ClassCastException expected) {}
+        } catch (final ClassCastException expected) {}
         assertEquals(null, ClassUtils.convertClassNamesToClasses(null));
     }
\ No newline at end of file
