diff --git a/old_product.java b/new_product.java
index 21695f0..c49af71 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,11 +2,11 @@ public static List<Class<?>> convertClassNamesToClasses(final List<String> class
         if (classNames == null) {
             return null;
         }
-        List<Class<?>> classes = new ArrayList<Class<?>>(classNames.size());
-        for (String className : classNames) {
+        final List<Class<?>> classes = new ArrayList<Class<?>>(classNames.size());
+        for (final String className : classNames) {
             try {
                 classes.add(Class.forName(className));
-            } catch (Exception ex) {
+            } catch (final Exception ex) {
                 classes.add(null);
             }
         }
