diff --git a/old_product.java b/new_product.java
index 6724947..c72ed93 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,5 +4,5 @@ private StrStream readText(final File path, final QueryContext ctx)
     final String enc = encoding(1, FILE_ENCODING, ctx);
     if(!path.exists()) FILE_WHICH.thrw(info, path.getAbsolutePath());
     if(path.isDirectory()) FILE_DIR.thrw(info, path.getAbsolutePath());
-    return new StrStream(new IOFile(path), enc, FILE_IO);
+    return new StrStream(new IOFile(path), enc, FILE_IO, ctx);
   }
\ No newline at end of file
