diff --git a/old_test.java b/new_test.java
index c83dd1a..9084214 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,5 +16,5 @@
     // new file has no entries
     list = list.replaceAll("<zip:dir.*</zip:dir>", "");
     error(_ZIP_UPDATE_ENTRIES.args(list,
-        new File(TMPZIP).getCanonicalPath()), Err.ZIPFAIL);
+        new File(TMPZIP).getCanonicalPath()), Err.ZIP_FAIL);
   }
\ No newline at end of file
