diff --git a/old_test.java b/new_test.java
index 55c88a0..4130886 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void test_getJavaScriptForRedirect_nonEmptyCookieName_secureCookie() throws Exception {
+    public void test_getJavaScriptForRedirect_secureCookie() throws Exception {
         try {
             String requestUrlCookieName = "some cookie name";
             String redirectUrl = VALID_URL;
@@ -13,12 +13,11 @@
 
             String result = utils.getJavaScriptForRedirect(requestUrlCookieName, redirectUrl);
 
-            verifyValidJavaScriptForRedirectBlock(result, requestUrlCookieName, redirectUrl);
-
             Map<String, String> cookieProps = new HashMap<String, String>();
             cookieProps.put("path", "/");
             cookieProps.put("secure", null);
-            verifyCookiePropertyStrings(result, cookieProps);
+
+            verifyValidJavaScriptForRedirectBlock(result, requestUrlCookieName, redirectUrl, cookieProps);
 
         } catch (Throwable t) {
             outputMgr.failWithThrowable(testName.getMethodName(), t);
