diff --git a/old_product.java b/new_product.java
index 5368464..4122758 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,5 +8,5 @@ private boolean installJAR(final byte[] content, final String path)
       final Matcher main = MAIN_CLASS.matcher(s);
       if(main.find()) return write(main.group(1).replace('.', '/') + IO.JARSUFFIX, content);
     }
-    throw BXRE_MAIN_X.get(info, path);
+    throw REPO_PARSE_X_X.get(info, path, MANIFEST);
   }
\ No newline at end of file
