diff --git a/old_test.java b/new_test.java
index 55fe467..6c82926 100644
--- a/old_test.java
+++ b/new_test.java
@@ -28,6 +28,6 @@
     try {
       new RepoDelete("org.basex.modules.Hello", null).execute(context);
     } catch(final BaseXException ex) {
-      assertTrue(ex.toString().contains(BXRE_WHICH_X.code));
+      assertTrue(ex.toString().contains(REPO_NOTFOUND_X.code));
     }
   }
\ No newline at end of file
