diff --git a/old_product.java b/new_product.java
index 0b9d3ba..4908cd9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
-private Message encodeDecode(final Message m1) throws Exception {
-		final CompositeByteBuf buf = Unpooled.compositeBuffer();
-		Encoder encoder = new Encoder(new DSASignatureFactory());
-		encoder.write(buf, m1, null);
-		Decoder decoder = new Decoder(new DSASignatureFactory());
-		decoder.decode(buf, m1.recipient().ipv4Socket().createUDPSocket(), m1
-				.sender().ipv4Socket().createUDPSocket());
-		return decoder.message();
-	}
\ No newline at end of file
+private Message encodeDecode(final Message m1, final byte[] privateKeySender, final byte[] privateKeyRecipient, int bufferSize)
+            throws GeneralSecurityException, IOException {
+        ByteBuffer buffer = ByteBuffer.wrap(new byte[bufferSize]);
+        Codec.encode(buffer, m1, create(m1.sender(), privateKeySender), true);
+        Message m2 = new Message();
+        buffer.flip();
+        Codec.decode(buffer, m2, create(m1.recipient(), privateKeyRecipient), m1.recipient().createSocket(m1.sender()), m1.sender().createSocket(m1.recipient()));
+        return m2;
+    }
\ No newline at end of file
