diff --git a/old_product.java b/new_product.java
index 685484a..38da33d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,7 +13,7 @@ public void parse(File file) throws UpdateException {
             }
             reader.beginArray();
             while (reader.hasNext()) {
-                final CVEItem cve = gson.fromJson(reader, CVEItem.class);
+                final DefCveItem cve = gson.fromJson(reader, DefCveItem.class);
 
                 //cve.getCve().getCVEDataMeta().getSTATE();
                 if (testCveCpeStartWithFilter(cve)) {
