diff --git a/old_test.java b/new_test.java
index 8f7683f..5fbb06a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,7 @@
         CliParser instance = new CliParser(getSettings());
         try {
             instance.parse(args);
+            Assert.fail("An exception should have been thrown");
         } catch (FileNotFoundException ex) {
             Assert.assertTrue(ex.getMessage().contains("Invalid 'scan' argument"));
         }
