diff --git a/old_test.java b/new_test.java
index ea4bf96..57686a1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,9 +3,6 @@
     long sessionId = mRandom.nextLong();
     long metricIncrease = 3;
 
-    mBlockWorker.sessionHeartbeat(sessionId, null);
+    mBlockWorker.sessionHeartbeat(sessionId);
     verify(mSessions).sessionHeartbeat(sessionId);
-    Counter counter = WorkerContext.getWorkerSource().getMetricRegistry().getCounters()
-        .get(WorkerSource.BLOCKS_READ_LOCAL);
-    assertEquals(metricIncrease, counter.getCount());
   }
\ No newline at end of file
