diff --git a/old_product.java b/new_product.java
index e490fb6..92c2f12 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,11 +7,11 @@ public void process() {
 
 		computeCellHistograms();
 
-		int cellRowMax = cellRows - widthBlock;
-		int cellColMax = cellCols - widthBlock;
+		int cellRowMax = (cellRows - (widthBlock-1));
+		int cellColMax = (cellCols - (widthBlock-1));
 
-		for (int i = 0; i < cellRowMax; i++) {
-			for (int j = 0; j < cellColMax; j++) {
+		for (int i = 0; i < cellRowMax; i += stepBlock) {
+			for (int j = 0; j < cellColMax; j += stepBlock) {
 				computeDescriptor(i,j);
 			}
 		}
