diff --git a/old_test.java b/new_test.java
index 5787b83..c1a872b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,6 +10,6 @@
 		order.setConcept(newConcept);
 		
 		expectedException.expect(APIException.class);
-		expectedException.expectMessage("The concept of the previous order and the new one order don't match");
+		expectedException.expectMessage("The orderable of the previous order and the new one order don't match");
 		orderService.saveOrder(order, null);
 	}
\ No newline at end of file
