diff --git a/old_test.java b/new_test.java
index 3ab0f62..7a0d124 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
     query(COUNT.args(_DB_BACKUPS.args(NAME)), "1");
 
     // invalid name
-    error(_DB_CREATE_BACKUP.args(" ''"), Err.BXDB_NAME);
+    error(_DB_CREATE_BACKUP.args(" ''"), Err.BXDB_NAME_X);
     // try to backup non-existing database
-    error(_DB_CREATE_BACKUP.args(NAME + 'x'), Err.BXDB_WHICH);
+    error(_DB_CREATE_BACKUP.args(NAME + 'x'), Err.BXDB_WHICH_X);
   }
\ No newline at end of file
