diff --git a/old_product.java b/new_product.java
index 2c95157..6c2b137 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 private Item createBackup(final QueryContext qc) throws QueryException {
-    final String name = string(checkStr(exprs[0], qc));
-    if(!Databases.validName(name)) throw BXDB_NAME.get(info, name);
-    if(!qc.context.globalopts.dbexists(name)) throw BXDB_WHICH.get(info, name);
+    final String name = string(toToken(exprs[0], qc));
+    if(!Databases.validName(name)) throw BXDB_NAME_X.get(info, name);
+    if(!qc.context.globalopts.dbexists(name)) throw BXDB_WHICH_X.get(info, name);
 
     qc.resources.updates().add(new BackupCreate(name, info, qc), qc);
     return null;
