diff --git a/old_product.java b/new_product.java
index 89e414c..311c66e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
-public synchronized boolean drop(final User user, final String db) {
-    if(db == null) {
+public synchronized boolean drop(final User user, final String pattern) {
+    if(pattern == null) {
       if(users.remove(user.name()) == null) return false;
     } else {
-      user.remove(db);
+      user.remove(pattern);
     }
     return true;
   }
\ No newline at end of file
