diff --git a/old_test.java b/new_test.java
index 0f7ba6b..dce8967 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
 	@Verifies(value = "should return true for any privilegeName if super user", method = "hasPrivilege(String)")
 	public void hasPrivilege_shouldReturnTrueForAnyPrivilegeNameIfSuperUser() throws Exception {
 		// check super user "super" status
-		Role role = new Role(OpenmrsConstants.SUPERUSER_ROLE);
+		Role role = new Role(RoleConstants.SUPERUSER);
 		
 		assertTrue("Super users are super special and should have all privileges", role
 		        .hasPrivilege("Some weird privilege name that shouldn't be there"));
