diff --git a/old_product.java b/new_product.java
index 318f4ee..a0dbf13 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @SuppressWarnings("deprecation")
-	public static void stopQuetly(Thread thread, String stopMaessage) {
+	public static void stopQuietly(Thread thread, String stopMessage) {
 		if (thread == null) {
 			return;
 		}
@@ -25,7 +25,7 @@
 		try {
 			// Force to Stop
 			if (thread.isAlive()) {
-				LOGGER.error(stopMaessage);
+				LOGGER.error(stopMessage);
 				thread.stop();
 			}
 		} catch (Exception e) {
