diff --git a/old_test.java b/new_test.java
index e1ac427..ec43f70 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,10 +2,9 @@
   public void openFile() throws Exception {
     AlluxioURI file = new AlluxioURI("/file");
     URIStatus status = new URIStatus(new FileInfo());
-    GetStatusOptions getStatusOptions = GetStatusOptions.defaults();
+    GetStatusPOptions getStatusOptions = GetStatusPOptions.getDefaultInstance();
     when(mFileSystemMasterClient.getStatus(file, getStatusOptions)).thenReturn(status);
-    OpenFileOptions openOptions = OpenFileOptions.defaults();
-    mFileSystem.openFile(file, openOptions);
+    mFileSystem.openFile(file, OpenFilePOptions.getDefaultInstance());
     verify(mFileSystemMasterClient).getStatus(file, getStatusOptions);
 
     verifyFilesystemContextAcquiredAndReleased();
