diff --git a/old_test.java b/new_test.java
index 87b0b5b..6a7b942 100644
--- a/old_test.java
+++ b/new_test.java
@@ -22,10 +22,10 @@
     for(final String k : sopt) {
       query(_DB_OPTIMIZE.args(NAME, "false()", " map { '" + k + "':='' }"));
     }
-    assertEquals(context.options.is(Options.TEXTINDEX), true);
+    assertEquals(context.options.is(MainOptions.TEXTINDEX), true);
 
-    error(_DB_OPTIMIZE.args(NAME, "false()", " map { 'updindex':=1 }"), Err.BASX_OPTIONS);
     error(_DB_OPTIMIZE.args(NAME, "false()", " map { 'xyz':='abc' }"), Err.BASX_OPTIONS);
+    error(_DB_OPTIMIZE.args(NAME, "false()", " map { 'updindex':=1 }"), Err.BASX_OPTIONS);
     error(_DB_OPTIMIZE.args(NAME, "false()", " map { 'maxlen':=-1 }"), Err.BASX_VALUE);
     error(_DB_OPTIMIZE.args(NAME, "false()", " map { 'maxlen':='a' }"), Err.BASX_VALUE);
 
