diff --git a/old_product.java b/new_product.java
index a09dd2e..c45bb39 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,8 +3,9 @@ private Item optimize(final QueryContext ctx) throws QueryException {
     final boolean all = expr.length > 1 && checkBln(expr[1], ctx);
 
     final Item opt = expr.length > 2 ? expr[2].item(ctx, info) : null;
-    final TokenMap map = new FuncParams(Q_OPTIONS, info).parse(opt);
+    final Options opts = new Options();
+    new FuncOptions(Q_OPTIONS, info).parse(opt, opts);
     // check database options
-    ctx.updates.add(new DBOptimize(data, ctx, all, map, info), ctx);
+    ctx.updates.add(new DBOptimize(data, ctx, all, opts, info), ctx);
     return null;
   }
\ No newline at end of file
