diff --git a/old_product.java b/new_product.java
index 6b1aeba..32be267 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,6 +3,7 @@ public static HttpResponse getBrokerageFromSolidity(String httpSolidityNode, byt
       String requestUrl = "http://" + httpSolidityNode + "/walletsolidity/getBrokerage";
       JsonObject userBaseObj2 = new JsonObject();
       userBaseObj2.addProperty("address", ByteArray.toHexString(address));
+      logger.info("userBaseObj2:" + userBaseObj2);
       response = createConnect(requestUrl, userBaseObj2);
     } catch (Exception e) {
       e.printStackTrace();
