diff --git a/old_product.java b/new_product.java
index e46c394..8195f87 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,11 +7,7 @@ private Int find(final QueryContext ctx) throws QueryException {
     final byte[] bytes = b64.binary(info);
     final int bl = bytes.length;
     final byte[] search = srch.binary(info);
-    final int sl = search.length;
-
     final int[] bounds = bounds(off, null, bl);
-    if(sl == 0) throw BIN_ESI.get(info, off);
-
     final int pos = indexOf(bytes, search, bounds[0]);
     return pos == -1 ? null : Int.get(pos);
   }
\ No newline at end of file
