diff --git a/old_test.java b/new_test.java
index 6dc4a25..9da56ef 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,8 +3,8 @@
     query(_BIN_FIND.args(base64("1122"),   0, base64("11")), 0);
     query(_BIN_FIND.args(base64("1122"),   1, base64("11")), "");
     query(_BIN_FIND.args(base64("112233"), 0, base64("22")), 1);
+    query(_BIN_FIND.args(base64(""), 0, base64("")), 0);
     // errors
-    error(_BIN_FIND.args(base64(""), -1, base64("11")), BIN_NO_X);
-    error(_BIN_FIND.args(base64(""), 1, base64("11")),  BIN_OBE_X_X);
-    error(_BIN_FIND.args(base64(""), 0, base64("")),    BIN_ESI);
+    error(_BIN_FIND.args(base64(""), -1, base64("11")), BIN_IOOR_X_X);
+    error(_BIN_FIND.args(base64(""), 1, base64("11")),  BIN_IOOR_X_X);
   }
\ No newline at end of file
