diff --git a/old_product.java b/new_product.java
index 0a0a86e..2aaadd7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 private Iter list(final QueryContext ctx) {
     final ItemIter ii = new ItemIter();
-    for(final String s : List.list(ctx.resource.context)) ii.add(Str.get(s));
+    for(final String s : List.list(ctx.context)) ii.add(Str.get(s));
     return ii;
   }
\ No newline at end of file
