diff --git a/old_test.java b/new_test.java
index 78e5b1c..bcd57b6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,11 @@
 @Test
   public void testList() throws BaseXException {
-    // wrong arguments
-    args("db:list");
+    final String fun = check(FunDef.LIST);
 
     // create two other database and compare substring
     new CreateDB("daz").execute(CTX);
     new CreateDB("dba").execute(CTX);
-    contains("db:list()", "daz db dba");
+    contains(fun + "()", "daz db dba");
     new DropDB("daz").execute(CTX);
     new DropDB("dba").execute(CTX);
   }
\ No newline at end of file
