diff --git a/old_test.java b/new_test.java
index d795416..ce5a2e2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,9 +6,9 @@
         IgniteEx ignite_3 = startGrid(3);
 
         List<String> ids = Lists.newArrayList(
-            cluster.localNode().id().toString(),
-            ignite_2.cluster().localNode().id().toString(),
-            ignite_3.cluster().localNode().id().toString()
+            cluster.localNode().consistentId().toString(),
+            ignite_2.cluster().localNode().consistentId().toString(),
+            ignite_3.cluster().localNode().consistentId().toString()
         );
 
         Request req = new Request()
