diff --git a/old_product.java b/new_product.java
index c5adcd0..511f2a9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,5 @@
-public void setBaselineTopology(Collection<String> ids) {
+public void setBaselineTopology(Collection<String> consIds) {
         authorizeIfNeeded(ctx.security(), SecurityPermission.ADMIN_OPS);
 
-        Set<UUID> uuids = ids.stream().map(UUID::fromString).collect(Collectors.toSet());
-
-        Collection<ClusterNode> nodes = ctx.grid().cluster().forNodeIds(uuids).forServers().nodes();
-        ctx.grid().cluster().setBaselineTopology(nodes);
+        ctx.grid().cluster().setBaselineTopology(baselineNodesForIds(consIds));
     }
\ No newline at end of file
