diff --git a/old_product.java b/new_product.java
index 9748d87..0c5a913 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,13 @@
-public static File getFile(String key) {
+public static File getFile(String key) throws IOException {
+        final String file = getString(key);
         final String baseDir = getString(Settings.KEYS.DATA_DIRECTORY);
-        final String tmp = getString(key);
         if (baseDir != null) {
-            return new File(baseDir, tmp);
+            if (baseDir.startsWith("[JAR]/")) {
+                final File jarPath = getJarPath();
+                final File newBase = new File(jarPath.getCanonicalPath(), baseDir.substring(6));
+                return new File(newBase, file);
+            }
+            return new File(baseDir, file);
         }
-        return new File(tmp);
+        return new File(file);
     }
\ No newline at end of file
