diff --git a/old_test.java b/new_test.java
index 3e17f85..1655d06 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,9 +4,5 @@
     InStreamOptions inStreamOptions = options.toInStreamOptions();
     Assert.assertEquals(options.getReadType().getAlluxioStorageType(),
         inStreamOptions.getAlluxioStorageType());
-    Assert.assertEquals(options.getCacheLocationPolicy(), inStreamOptions.getCacheLocationPolicy());
-    Assert.assertEquals(options.getUfsReadLocationPolicy(),
-        inStreamOptions.getUfsReadLocationPolicy());
-    Assert.assertEquals(options.getMaxUfsReadConcurrency(),
-        inStreamOptions.getMaxUfsReadConcurrency());
+    Assert.assertEquals(options.getLocationPolicy(), inStreamOptions.getLocationPolicy());
   }
\ No newline at end of file
