diff --git a/old_test.java b/new_test.java
index f0cd119..f086928 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,7 @@
 	public void validate_shouldPassValidationIfAllFieldsAreCorreect() throws Exception {
 		PersonAttributeType type = new PersonAttributeType();
 		type.setName("Zodiac");
+        type.setDescription("Zodiac Description");
 		
 		Errors errors = new BindException(type, "patObj");
 		new PersonAttributeTypeValidator().validate(type, errors);
