diff --git a/old_product.java b/new_product.java
index 88b680d..8e75b95 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public void validate(Object obj, Errors errors) {
 		PersonAttributeType patObj = (PersonAttributeType) obj;
 		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "name", "PersonAttributeType.error.nameEmpty");
-		
+		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "description", "error.description");
 		PersonService ps = Context.getPersonService();
 		PersonAttributeType pat = ps.getPersonAttributeTypeByName(patObj.getName());
 		if (pat != null && !pat.getPersonAttributeTypeId().equals(patObj.getPersonAttributeTypeId())) {
