diff --git a/old_test.java b/new_test.java
index 8ea0a76..6c53fd3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,6 @@
 @Test
     public void handleTask_anyOrder() throws Exception {
+        san.add("SAN=dns:localhost");
         String[] args = new String[] { task.getTaskName(),
                                        "--validity=" + VALIDITY,
                                        "--password=" + PLAINTEXT,
@@ -22,7 +23,7 @@
                                                          SUBJECT_DN,
                                                          DefaultSSLCertificateCreator.DEFAULT_SIZE,
                                                          DefaultSSLCertificateCreator.SIGALG,
-                                                         null);
+                                                         san);
             }
         });
 
