diff --git a/old_product.java b/new_product.java
index 295f9ad..ded3ca8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -74,7 +74,7 @@
         String subjectDN = getArgumentValue(ARG_SUBJECT, args, new DefaultSubjectDN(null, ou_name).getSubjectDN());
         int keySize = Integer.valueOf(getArgumentValue(ARG_KEYSIZE, args, String.valueOf(DefaultSSLCertificateCreator.DEFAULT_SIZE)));
         String sigAlg = getArgumentValue(ARG_SIGALG, args, DefaultSSLCertificateCreator.SIGALG);
-        String extInfo = getArgumentValue(ARG_EXT, args, null);
+        List<String> extInfo = getExtInfoArgumentValues(ARG_EXT, args);
 
         try {
             String encoding = getArgumentValue(ARG_ENCODING, args, PasswordUtil.getDefaultEncoding());
