diff --git a/old_product.java b/new_product.java
index a916982..676cca7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,10 @@
-public static Date createDate(String dateString) throws NumberFormatException,
-    InstantiationException, IllegalAccessException {
-    Date date = Date.class.newInstance().setYear(Integer.decode(dateString.split("-")[0]))
-        .setMonth(Integer.decode(dateString.split("-")[1]))
-        .setDay(Integer.decode(dateString.split("-")[2]));
+public static Date createDate(String dateString)
+      throws NumberFormatException, InstantiationException, IllegalAccessException {
+    Date date =
+        Date.class
+            .newInstance()
+            .setYear(Integer.parseInt(dateString.split("-")[0], BASE_10))
+            .setMonth(Integer.parseInt(dateString.split("-")[1], BASE_10))
+            .setDay(Integer.parseInt(dateString.split("-")[2], BASE_10));
     return date;
   }
\ No newline at end of file
