diff --git a/old_test.java b/new_test.java
index 10028ea..33f2f79 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
     public void testNotEq() {
-    	assertEquals(new NotEqCriterion("name", "abc"), instance.notEq("name", "abc").getQueryCriterion());
+    	assertEquals(Criteria.notEq("name", "abc"), instance.notEq("name", "abc").getQueryCriterion());
     }
\ No newline at end of file
