diff --git a/old_product.java b/new_product.java
index ef85720..702f4af 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,10 +3,11 @@ private boolean consumes(final HTTPContext http) {
     if(consumes.isEmpty()) return true;
     // return true if no content type is specified by the user
     final String ct = http.contentType();
-    if(ct == null) return true;
+    if(ct.isEmpty()) return true;
+
     // check if any combination matches
-    for(final String c : consumes) {
-      if(MimeTypes.matches(c, ct)) return true;
+    for(final String consume : consumes) {
+      if(MimeTypes.matches(consume, ct)) return true;
     }
     return false;
   }
\ No newline at end of file
