diff --git a/old_test.java b/new_test.java
index 14cec1c..4cb5fe6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,7 @@
     // correct syntax
     get("declare %R:path('') %R:consumes('text/plain') function m:f() {1};", "", "1");
     get("declare %R:path('') %R:consumes('*/*') function m:f() {1};", "", "1");
+    get("declare %R:path('') %R:consumes('text/plain;bla=blu') function m:f() {1};", "", "1");
     // multiple types
     get("declare %R:path('') %R:consumes('text/plain','*/*') function m:f() {1};", "", "1");
     get("declare %R:path('') %R:consumes('text/plain') %R:consumes('*/*') " +
