diff --git a/old_test.java b/new_test.java
index 3a03e1d..a049d87 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,15 +14,13 @@
 
         LinearRegressionSGDTrainer<?> trainer = new LinearRegressionSGDTrainer<>(new UpdatesStrategy<>(
             new RPropUpdateCalculator(),
-            RPropParameterUpdate::sumLocal,
-            RPropParameterUpdate::avg
+            RPropParameterUpdate.SUM_LOCAL,
+            RPropParameterUpdate.AVG
         ), 100000, 10, 100, 0L);
 
         LinearRegressionModel originalMdl = trainer.withSeed(0).fit(
-            data,
-            parts,
-            (k, v) -> VectorUtils.of(Arrays.copyOfRange(v, 0, v.length - 1)),
-            (k, v) -> v[4]
+            data, parts,
+            new ArraysVectorizer<Integer>().labeled(Vectorizer.LabelCoordinate.LAST)
         );
 
 
@@ -30,16 +28,14 @@
             originalMdl,
             data,
             parts,
-            (k, v) -> VectorUtils.of(Arrays.copyOfRange(v, 0, v.length - 1)),
-            (k, v) -> v[4]
+            new ArraysVectorizer<Integer>().labeled(Vectorizer.LabelCoordinate.LAST)
         );
 
         LinearRegressionModel updatedOnEmptyDS = trainer.withSeed(0).update(
             originalMdl,
-            new HashMap<Integer, double[]>(),
+            new HashMap<>(),
             parts,
-            (k, v) -> VectorUtils.of(Arrays.copyOfRange(v, 0, v.length - 1)),
-            (k, v) -> v[4]
+            new ArraysVectorizer<Integer>().labeled(Vectorizer.LabelCoordinate.LAST)
         );
 
         assertArrayEquals(
