diff --git a/old_product.java b/new_product.java
index 7c7eb95..e3dfc5f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,7 @@
         URI uri = URI.create(s);
 
         if (uri.getScheme() == null) {
-            uri = ((conf != null) ? FileSystem.getDefaultUri(conf) : URI.create("file:///")).resolve(uri);
+            uri = ((conf != null) ? FileSystem.getDefaultUri(conf) : URI.create("file:///").resolve(new File("").getAbsolutePath() + "/")).resolve(uri);
         }
         if (!IGNORED_PROTOCOLS.contains(uri.getScheme())) {
             uri = URI.create("hadoop:" + uri);
