diff --git a/old_test.java b/new_test.java
index 7071c18..1ccc8e0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,4 +7,9 @@
 		boolean removed = Files.remove(file);
 		assertFalse("The just removed file should not exist!", file.exists());
 		assertTrue("Files.remove(file) should remove the file", removed);
+
+		// try to remove non-existing file
+		removed = Files.remove(file);
+		assertFalse("The just removed file should not exist!", file.exists());
+		assertFalse("Files.remove(file) should not remove the file", removed);
 	}
\ No newline at end of file
