diff --git a/old_product.java b/new_product.java
index 2be7651..a6b353e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -48,6 +48,9 @@ protected final CharSequence renderAjaxAttributes(final Component component,
 					CharSequence beforeHandler = ajaxCallListener.getBeforeHandler(component);
 					appendListenerHandler(beforeHandler, attributesJson, "bh", BEFORE_HANDLER_FUNCTION_TEMPLATE);
 
+					CharSequence beforeSendHandler = ajaxCallListener.getBeforeSendHandler(component);
+					appendListenerHandler(beforeSendHandler, attributesJson, "bsh", BEFORE_SEND_HANDLER_FUNCTION_TEMPLATE);
+
 					CharSequence afterHandler = ajaxCallListener.getAfterHandler(component);
 					appendListenerHandler(afterHandler, attributesJson, "ah", AFTER_HANDLER_FUNCTION_TEMPLATE);
 
