diff --git a/old_product.java b/new_product.java
index 9279e45..38aa9e5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,14 +1,14 @@
 @Override
-  public List<byte[]> rowKeys(String profile, String entity, List<Object> groups, long durationAgo, TimeUnit unit) {
+  public List<byte[]> rowKeys(String profile, String entity, List<Object> groups, long start, long end) {
     List<byte[]> rowKeys = new ArrayList<>();
 
-    // find the time horizon
-    long endTime = System.currentTimeMillis();
-    long startTime = endTime - unit.toMillis(durationAgo);
+    // be forgiving of out-of-order start and end times; order is critical to this algorithm
+    end = Math.max(start, end);
+    start = Math.min(start, end);
 
     // find the starting period and advance until the end time is reached
-    ProfilePeriod period = new ProfilePeriod(startTime, periodDurationMillis, TimeUnit.MILLISECONDS);
-    while(period.getStartTimeMillis() <= endTime) {
+    ProfilePeriod period = new ProfilePeriod(start, periodDurationMillis, TimeUnit.MILLISECONDS);
+    while(period.getStartTimeMillis() <= end) {
 
       byte[] k = rowKey(profile, entity, period, groups);
       rowKeys.add(k);
