diff --git a/old_test.java b/new_test.java
index 1472746..2a6a954 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,8 @@
 @Test
-    public void testSetParams() {
-        System.out.println("setParams");
+    public void testAddMetric() {
+        System.out.println("addMetric");
+        String metric = "";
         String params = "";
         PerfMonAgentConnector instance = new PerfMonAgentConnectorImpl();
-        instance.setParams(params);
+        instance.addMetric(metric, params, null);
     }
\ No newline at end of file
