diff --git a/old_product.java b/new_product.java
index 192188c..128b234 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,11 @@
 @Override
-   public void setValue(Object value)
-   {
-      setText((value == null) ? "" : dateFormatter.format(value));
-   }
\ No newline at end of file
+    public void setValue(Object value)
+    {
+        if (!(value instanceof Long))
+            setText(EMPTY);
+        else
+        {
+            long val = (Long) value;
+            setText(dateFormatter.format(val - relativeStartTime));
+        }
+    }
\ No newline at end of file
