diff --git a/old_test.java b/new_test.java
index 5428d8c..fcc22a6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
    public void testSetValue()
    {
       System.out.println("setValue");
-      DateTimeRenderer instance = new DateTimeRenderer("HH:mm:ss");
+      DateTimeRenderer instance = new DateTimeRenderer(HHMMSS);
 
       Calendar test = Calendar.getInstance();
 
@@ -11,9 +11,6 @@
       test.set(Calendar.SECOND, 40);
       test.set(Calendar.MILLISECOND, 0);
       
-      instance.setValue(null);
-      assertEquals("", instance.getText());
-
       instance.setValue(test.getTimeInMillis());
       String text = instance.getText();
       assertEquals("03:16:40", text);
