diff --git a/old_test.java b/new_test.java
index 66b4d33..0940722 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,32 +1,24 @@
 @SuppressWarnings("unchecked")
 	@Test
 	public void testDelete() {
-		ModelMap model = new ModelMap();
 		// save new user for test
 		saveTestUser("NewUserId1", "NewUserName1");
 		saveTestUser("NewUserId2", "NewUserName2");
 		saveTestUser("NewUserId3", "NewUserName3");
 
-		Pageable page = new PageRequest(0, 10);
+		Pageable page = PageRequest.of(0, 10);
 
 		// search
-		userController.getAll(model, null, page, "NewUserName");
-		PageImpl userList = (PageImpl<User>) model.get("users");
+		Page<User> userList = userController.getAll(null, page, "NewUserName");
 		assertThat(userList.getContent().size(), is(3));
 
 		// test to delete one
-		model.clear();
-		userController.delete(testUser, "NewUserId1", model);
-		model.clear();
-		userController.getAll(model, Role.USER, page, "NewUserName");
-		userList = (PageImpl<User>) model.get("users");
+		userController.delete(testUser, "NewUserId1");
+		userList = userController.getAll(Role.USER, page, "NewUserName");
 		assertThat(userList.getContent().size(), is(2));
 
 		// test to delete more
-		model.clear();
-		userController.delete(testUser, "NewUserId2,NewUserId3", model);
-		model.clear();
-		userController.getAll(model, Role.USER, page, "NewUserName");
-		userList = (PageImpl<User>) model.get("users");
+		userController.deleteUsers(testUser, "NewUserId2,NewUserId3");
+		userList = userController.getAll(Role.USER, page, "NewUserName");
 		assertThat(userList.getContent().size(), is(0));
 	}
\ No newline at end of file
