diff --git a/old_product.java b/new_product.java
index b7ce1cd..91393d7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,11 @@
 @PreAuthorize("hasAnyRole('A')")
-	@RequestMapping("/delete")
-	public String delete(User user, @RequestParam String userIds, ModelMap model) {
+	@DeleteMapping({"/api", "/api/"})
+	public HttpEntity<String> deleteUsers(User user, @RequestParam String userIds) {
 		String[] ids = userIds.split(",");
-		for (String eachId : Arrays.asList(ids)) {
+		for (String eachId : ids) {
 			if (!user.getUserId().equals(eachId)) {
 				userService.delete(eachId);
 			}
 		}
-		model.clear();
-		return "redirect:/user/";
+		return successJsonHttpEntity();
 	}
\ No newline at end of file
