diff --git a/old_test.java b/new_test.java
index 2b590bc..4d08d98 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
         
         Dataset trainingData = generateDataset();
         
-        MemoryConfiguration memoryConfiguration = new MemoryConfiguration();
+        
         
         
         String dbName = "JUnitRegressor";
@@ -25,7 +25,7 @@
         param.setNu0(1);
         param.setMu0(new double[]{0.0, 0.0});
         param.setPsi0(new double[][]{{1.0,0.0},{0.0,1.0}});
-        instance.initializeTrainingConfiguration(memoryConfiguration, param);
+        instance.initializeTrainingConfiguration(param);
         GaussianDPMM.ValidationMetrics vm = instance.kFoldCrossValidation(trainingData, k);
 
         
