diff --git a/old_product.java b/new_product.java
index e7cbc81..735be04 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,5 +4,5 @@
             System.out.println("kFoldCrossValidation()");
         }
         
-        return modelValidator.kFoldCrossValidation(trainingData, k, dbName, this.getClass(), knowledgeBase.getTrainingParameters(), knowledgeBase.getMemoryConfiguration());
+        return modelValidator.kFoldCrossValidation(trainingData, k, dbName, this.getClass(), knowledgeBase.getTrainingParameters());
     }
\ No newline at end of file
