diff --git a/old_product.java b/new_product.java
index dccfd57..045c8a0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,12 @@
 public ServiceCall<AcousticModel> createAcousticModel(CreateAcousticModelOptions createAcousticModelOptions) {
     Validator.notNull(createAcousticModelOptions, "createAcousticModelOptions cannot be null");
     RequestBuilder builder = RequestBuilder.post("/v1/acoustic_customizations");
-    builder.header("Content-Type", createAcousticModelOptions.contentType());
-    if (createAcousticModelOptions.contentType().equalsIgnoreCase(
-        CreateAcousticModelOptions.ContentType.APPLICATION_JSON)) {
-      builder.bodyJson(GsonSingleton.getGson().toJsonTree(createAcousticModelOptions.createAcousticModel())
-          .getAsJsonObject());
+    final JsonObject contentJson = new JsonObject();
+    contentJson.addProperty("name", createAcousticModelOptions.name());
+    contentJson.addProperty("base_model_name", createAcousticModelOptions.baseModelName());
+    if (createAcousticModelOptions.description() != null) {
+      contentJson.addProperty("description", createAcousticModelOptions.description());
     }
+    builder.bodyJson(contentJson);
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(AcousticModel.class));
   }
\ No newline at end of file
