diff --git a/old_test.java b/new_test.java
index bb9fecb..d6042b5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,26 +1,22 @@
 @Test
-    public void testAddEventClassFamilySchema() throws Exception {
-        EventClassFamilyDto eventClassFamily = createEventClassFamily();
-        Schema expectedSchema = new Schema.Parser().parse(getResourceAsString(TEST_EVENT_CLASS_FAMILY_SCHEMA));
-
-        client.addEventClassFamilySchema(eventClassFamily.getId(), TEST_EVENT_CLASS_FAMILY_SCHEMA);
-        EventClassFamilyDto storedEventClassFamily = client.getEventClassFamilyById(eventClassFamily.getId());
-        List<EventSchemaVersionDto> schemas = storedEventClassFamily.getSchemas();
+    public void testAddEventClassFamilyVersion() throws Exception {
+        tenantAdminDto = createTenant(tenantAdminUser);
+        loginTenantAdmin(tenantAdminUser);
+        EventClassFamilyDto eventClassFamily = createEventClassFamily(tenantAdminDto.getId());
+        EventClassFamilyVersionDto eventClassFamilyVersion = createEventClassFamilyVersion(eventClassFamily.getId());
+        client.addEventClassFamilyVersion(eventClassFamily.getId(), eventClassFamilyVersion);
+        List<EventClassFamilyVersionDto> schemas = eventClassService.findEventClassFamilyVersionsById(eventClassFamily.getId());
         Assert.assertNotNull(schemas);
         Assert.assertEquals(1, schemas.size());
-        EventSchemaVersionDto eventSchema = schemas.get(0);
+        EventClassFamilyVersionDto eventSchema = schemas.get(0);
         Assert.assertNotNull(eventSchema);
         Assert.assertEquals(1, eventSchema.getVersion());
-        
-        Assert.assertEquals(expectedSchema, new Schema.Parser().parse(eventSchema.getSchema()));
-        
-        client.addEventClassFamilySchema(eventClassFamily.getId(), TEST_EVENT_CLASS_FAMILY_SCHEMA);
-        storedEventClassFamily = client.getEventClassFamilyById(eventClassFamily.getId());
-        schemas = storedEventClassFamily.getSchemas();
+
+        client.addEventClassFamilyVersion(eventClassFamily.getId(), eventClassFamilyVersion);
+        schemas = eventClassService.findEventClassFamilyVersionsById(eventClassFamily.getId());
         Assert.assertNotNull(schemas);
         Assert.assertEquals(2, schemas.size());
         eventSchema = schemas.get(1);
         Assert.assertNotNull(eventSchema);
         Assert.assertEquals(2, eventSchema.getVersion());
-        Assert.assertEquals(expectedSchema, new Schema.Parser().parse(eventSchema.getSchema()));
     }
\ No newline at end of file
