diff --git a/old_product.java b/new_product.java
index 09c7595..834fec7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,8 @@
-public void addEventClassFamilySchema(String eventClassFamilyId, String schemaResource) throws Exception {
-        MultiValueMap<String, Object> params = new LinkedMultiValueMap<>();
-        params.add("eventClassFamilyId", eventClassFamilyId);
-        params.add("file", getFileResource(schemaResource));
-        restTemplate.postForLocation(restTemplate.getUrl() + "addEventClassFamilySchema", params);
-    }
\ No newline at end of file
+public void addEventClassFamilyVersion(String eventClassFamilyId,
+                                         EventClassFamilyVersionDto eventClassFamilyVersion)
+      throws Exception {
+    MultiValueMap<String, Object> params = new LinkedMultiValueMap<>();
+    params.add("eventClassFamilyId", eventClassFamilyId);
+    params.add("eventClassFamilyVersion", eventClassFamilyVersion);
+    restTemplate.postForLocation(restTemplate.getUrl() + "addEventClassFamilyVersion", params);
+  }
\ No newline at end of file
