diff --git a/old_test.java b/new_test.java
index ae9f749..ce7f546 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void testEnqueueAll() throws IOException, ExecutorManagerException {
-    QueuedExecutions queue = new QueuedExecutions(5);
-    List<Pair<ExecutionReference, ExecutableFlow>> dataList = getDummyData();
+    final QueuedExecutions queue = new QueuedExecutions(5);
+    final List<Pair<ExecutionReference, ExecutableFlow>> dataList = getDummyData();
     queue.enqueueAll(dataList);
     Assert.assertTrue(queue.getAllEntries().containsAll(dataList));
     Assert.assertTrue(dataList.containsAll(queue.getAllEntries()));
