diff --git a/old_product.java b/new_product.java
index af0b8a3..ab6d70a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public void enqueueAll(
-    Collection<Pair<ExecutionReference, ExecutableFlow>> collection)
-    throws ExecutorManagerException {
-    for (Pair<ExecutionReference, ExecutableFlow> pair : collection) {
+      final Collection<Pair<ExecutionReference, ExecutableFlow>> collection)
+      throws ExecutorManagerException {
+    for (final Pair<ExecutionReference, ExecutableFlow> pair : collection) {
       enqueue(pair.getSecond(), pair.getFirst());
     }
   }
\ No newline at end of file
