diff --git a/old_test.java b/new_test.java
index 64e9467..5289865 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,4 +4,9 @@
 		PersonName pname = Context.getPersonService().parsePersonName("Doe, John");
 		assertEquals("Doe", pname.getFamilyName());
 		assertEquals("John", pname.getGivenName());
+		
+		// try without a space
+		pname = Context.getPersonService().parsePersonName("Doe,John");
+		assertEquals("Doe", pname.getFamilyName());
+		assertEquals("John", pname.getGivenName());
 	}
\ No newline at end of file
