diff --git a/old_test.java b/new_test.java
index 8fb4384..c32ceba 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,18 +11,18 @@
                 "line 8 with \u2029 char\n" +
                 "line 9\n";
 
-        bug15890LineCount(new CXref(new StringReader(fileContents)));
-        bug15890LineCount(new CxxXref(new StringReader(fileContents)));
-        bug15890LineCount(new LispXref(new StringReader(fileContents)));
-        bug15890LineCount(new JavaXref(new StringReader(fileContents)));
-        bug15890LineCount(new ScalaXref(new StringReader(fileContents)));
-        bug15890LineCount(new FortranXref(new StringReader(fileContents)));
-        bug15890LineCount(new HaskellXref(new StringReader(fileContents)));
-        bug15890LineCount(new XMLXref(new StringReader(fileContents)));
-        bug15890LineCount(new ShXref(new StringReader(fileContents)));
-        bug15890LineCount(new TclXref(new StringReader(fileContents)));
-        bug15890LineCount(new SQLXref(new StringReader(fileContents)));
+        bug15890LineCount(new JFlexXref(new CXref(new StringReader(fileContents))));
+        bug15890LineCount(new JFlexXref(new CxxXref(new StringReader(fileContents))));
+        bug15890LineCount(new JFlexXref(new LispXref(new StringReader(fileContents))));
+        bug15890LineCount(new JFlexXref(new JavaXref(new StringReader(fileContents))));
+        bug15890LineCount(new JFlexXref(new ScalaXref(new StringReader(fileContents))));
+        bug15890LineCount(new JFlexXref(new FortranXref(new StringReader(fileContents))));
+        bug15890LineCount(new JFlexXref(new HaskellXref(new StringReader(fileContents))));
+        bug15890LineCount(new JFlexXref(new XMLXref(new StringReader(fileContents))));
+        bug15890LineCount(new JFlexXref(new ShXref(new StringReader(fileContents))));
+        bug15890LineCount(new JFlexXref(new TclXref(new StringReader(fileContents))));
+        bug15890LineCount(new JFlexXref(new SQLXref(new StringReader(fileContents))));
         bug15890LineCount(new TroffXref(new StringReader(fileContents)));
-        bug15890LineCount(new PlainXref(new StringReader(fileContents)));
-        bug15890LineCount(new PerlXref(new StringReader(fileContents)));
+        bug15890LineCount(new JFlexXref(new PlainXref(new StringReader(fileContents))));
+        bug15890LineCount(new JFlexXref(new PerlXref(new StringReader(fileContents))));
     }
\ No newline at end of file
