diff --git a/old_product.java b/new_product.java
index 0bc15b4..fa37868 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-abstract void calculateIndexUpdates(List<SepEvent> events, SolrUpdateCollector updateCollector) throws IOException;
\ No newline at end of file
+abstract void calculateIndexUpdates(List<RowData> rowDataList, SolrUpdateCollector updateCollector) throws IOException;
\ No newline at end of file
