diff --git a/old_test.java b/new_test.java
index 1229ee1..f35249b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
     public void testCalculateIndexUpdates_DeleteRow_NoRowFieldDefinedForIndexer() throws IOException {
         KeyValue toDelete = new KeyValue("_row_".getBytes(), "_cf_".getBytes(), "_qual_".getBytes(), 0L, Type.Delete);
-        SepEvent event = createSepEvent("_row_", toDelete);
+        RowData eventRowData = createEventRowData("_row_", toDelete);
         
-        indexer.calculateIndexUpdates(Lists.newArrayList(event), updateCollector);
+        indexer.calculateIndexUpdates(ImmutableList.of(eventRowData), updateCollector);
 
         assertTrue(updateCollector.getDeleteQueries().isEmpty());
         assertTrue(updateCollector.getIdsToDelete().isEmpty());
