diff --git a/old_product.java b/new_product.java
index 9362211..6274494 100644
--- a/old_product.java
+++ b/new_product.java
@@ -17,8 +17,14 @@ public boolean process( I frame ) {
 			return false;
 		}
 
+		// refine the motion estimate
+		if( modelRefiner == null ||
+				!modelRefiner.fitModel(modelMatcher.getMatchSet(),modelMatcher.getModel(),keyToCurr))
+		{
+			keyToCurr.set(modelMatcher.getModel());
+		}
+
 		// Update the motion
-		keyToCurr.set(modelMatcher.getModel());
 		worldToKey.concat(keyToCurr, worldToCurr);
 
 		return true;
