diff --git a/old_product.java b/new_product.java
index c3407a7..a3a5498 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-public void mergeSimilarSegments(final double sigmaThresholdSegmentMean,
-                                     final double sigmaThresholdMinorAlleleFraction,
+public void mergeSimilarSegments(final double intervalThresholdSegmentMean,
+                                     final double intervalThresholdMinorAlleleFraction,
                                      final int numSamplesCopyRatio, final int numBurnInCopyRatio,
                                      final int numSamplesAlleleFraction, final int numBurnInAlleleFraction) {
         logger.info("Starting similar-segment merging...");
@@ -18,7 +18,7 @@ public void mergeSimilarSegments(final double sigmaThresholdSegmentMean,
             prevNumSegments = mergedSegments.size();
             mergedSegments =
                     SegmentMergeUtils.mergeSimilarSegments(segments,
-                            sigmaThresholdSegmentMean, sigmaThresholdMinorAlleleFraction);
+                            intervalThresholdSegmentMean, intervalThresholdMinorAlleleFraction);
             logger.info("Number of segments after merging: " + mergedSegments.size());
             segmentedModel = new SegmentedModel(toUnmodeledSegments(mergedSegments), segmentedModel.getGenome());
             if (mergedSegments.size() == prevNumSegments) {
