diff --git a/old_test.java b/new_test.java
index bd3bc49..e170ff9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,8 @@
 @Test
     public void testMergeSimilarSegments() {
+        final JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
+        LoggingUtils.setLoggingLevel(Log.LogLevel.INFO);
+
         final String tempDir = publicTestDir + "similar-segment-test";
         final File tempDirFile = createTempDir(tempDir);
 
@@ -10,8 +13,8 @@
         //run MCMC
         final ACNVModeller modeller =
                 new ACNVModeller(segmentedModel, tempDirFile.getAbsolutePath() + "/test",
-                        NUM_SAMPLES, NUM_BURN_IN, NUM_SAMPLES, NUM_BURN_IN);
-        modeller.mergeSimilarSegments(SIGMA_THRESHOLD, SIGMA_THRESHOLD,
+                        NUM_SAMPLES, NUM_BURN_IN, NUM_SAMPLES, NUM_BURN_IN, ctx);
+        modeller.mergeSimilarSegments(INTERVAL_THRESHOLD, INTERVAL_THRESHOLD,
                 NUM_SAMPLES, NUM_BURN_IN, NUM_SAMPLES, NUM_BURN_IN);
 
         //check equality of segments
