diff --git a/old_test.java b/new_test.java
index 32d6ea6..b92ebc9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
-	public void testCreateDialog() throws URISyntaxException {
-		File dialogFile = new File("src/test/resources/pizza_sample.xml");
-		String dialogName = ""+UUID.randomUUID().toString().substring(0, 15);
-		Dialog newDialog = service.createDialog(dialogName, dialogFile);
-		
-		assertNotNull(newDialog.getId());
-		newDialog = service.updateDialog(newDialog.getId(), dialogFile);
-		assertNotNull(newDialog.getId());
-		service.deleteDialog(newDialog.getId());
-	}
\ No newline at end of file
+  public void testCreateDialog() throws URISyntaxException {
+    final File dialogFile = new File("src/test/resources/pizza_sample.xml");
+    final String dialogName = "" + UUID.randomUUID().toString().substring(0, 15);
+    Dialog newDialog = service.createDialog(dialogName, dialogFile);
+
+    assertNotNull(newDialog.getId());
+    newDialog = service.updateDialog(newDialog.getId(), dialogFile);
+    assertNotNull(newDialog.getId());
+    service.deleteDialog(newDialog.getId());
+  }
\ No newline at end of file
