diff --git a/old_product.java b/new_product.java
index 184f43d..4ec0b12 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-public void syn(SynInfo synInfo, StreamFrameListener listener, long timeout, TimeUnit unit, Callback<Stream> callback);
\ No newline at end of file
+public void syn(SynInfo synInfo, StreamFrameListener listener, long timeout, TimeUnit unit, Promise<Stream> callback);
\ No newline at end of file
