diff --git a/old_test.java b/new_test.java
index 9da3c06..4fe7a1d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,12 +2,12 @@
     @Test
     public void testSyn()
     {
-        Stream stream = new StandardStream(synStreamFrame,session,0,null);
+        Stream stream = new StandardStream(synStreamFrame.getStreamId(), synStreamFrame.getPriority(), session, null);
         Set<Stream> streams = new HashSet<>();
         streams.add(stream);
         when(synStreamFrame.isClose()).thenReturn(false);
         SynInfo synInfo = new SynInfo(false);
         when(session.getStreams()).thenReturn(streams);
         stream.syn(synInfo);
-        verify(session).syn(argThat(new PushSynInfoMatcher(stream.getId(),synInfo)),any(StreamFrameListener.class),anyLong(),any(TimeUnit.class),any(Callback.class));
+        verify(session).syn(argThat(new PushSynInfoMatcher(stream.getId(), synInfo)), any(StreamFrameListener.class), anyLong(), any(TimeUnit.class), any(Handler.class));
     }
\ No newline at end of file
