diff --git a/old_test.java b/new_test.java
index aba3525..683b2ff 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,15 +1,18 @@
 @Test
-    public void testMultiply_Vec_ExecutorService()
+    public void testMultiply_Vec_Double_Vec()
     {
         DenseVector b = new DenseVector(Arrays.asList(4.0, 5.0, 2.0, 6.0, 7.0));
         
         DenseVector z = new DenseVector(Arrays.asList(2.0, 1.0, 2.0, 3.0, 4.0, 5.0, 0.0));
+                
+        DenseVector store = b.deepCopy();
         
-        DenseVector Ab = new DenseVector(Arrays.asList(148.0, 110.0, 103.0, 94.0, 149.0));
-        
-        assertEquals(Ab, A.multiply(b, threadpool));
+        A.multiply(b, 3.0, store);
+        assertEquals(new DenseVector(new double[]{ 448, 335, 311, 288, 454}), store);
         
         DenseVector Cz = new DenseVector(Arrays.asList(62.0, 100.0, 88.0, 74.0, 68.0));
         
-        assertEquals(Cz, C.multiply(z, threadpool));
+        store.zeroOut();
+        C.multiply(z, 1.0, store);
+        assertEquals(Cz, store);
     }
\ No newline at end of file
