diff --git a/old_test.java b/new_test.java
index 66e68f2..21fa0be 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
     public void test_withEra_BE() {
         ChronoLocalDate base = ThaiBuddhistChrono.INSTANCE.date(2555, 8, 29);
-        ChronoLocalDate test = base.with(LocalDateTimeField.ERA, ThaiBuddhistChrono.ERA_BE.getValue());
+        ChronoLocalDate test = base.with(ChronoField.ERA, ThaiBuddhistChrono.ERA_BE.getValue());
         assertEquals(test, ThaiBuddhistChrono.INSTANCE.date(2555, 8, 29));
     }
\ No newline at end of file
