diff --git a/old_product.java b/new_product.java
index 4da8fa7..23899a1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ public static CameraPinhole estimatePinhole(Point2Transform2_F64 pixelToNorm , i
 		Vector3D_F64 vectorB = new Vector3D_F64();
 
 		pixelToNorm.compute(0,height/2,normA);
-		pixelToNorm.compute(width-1,height/2,normB);
+		pixelToNorm.compute(width,height/2,normB);
 		vectorA.set(normA.x,normA.y,1);
 		vectorB.set(normB.x,normB.y,1);
 		double hfov = UtilVector3D_F64.acute(vectorA,vectorB);
