diff --git a/old_test.java b/new_test.java
index cd64149..7a68e15 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,15 @@
 @Test
 	public void estimatePinhole() {
-		fail("Implement");
+		CameraPinhole expected = new CameraPinhole(500,550,0,600,700,1200,1400);
+		Point2Transform2_F64 pixelToNorm = new LensDistortionPinhole(expected).distort_F64(true,false);
+
+		CameraPinhole found = PerspectiveOps.estimatePinhole(pixelToNorm,expected.width,expected.height);
+
+		assertEquals(expected.fx,found.fx,UtilEjml.TEST_F64);
+		assertEquals(expected.fy,found.fy,UtilEjml.TEST_F64);
+		assertEquals(expected.cx,found.cx,UtilEjml.TEST_F64);
+		assertEquals(expected.cy,found.cy,UtilEjml.TEST_F64);
+		assertEquals(expected.skew,found.skew,UtilEjml.TEST_F64);
+		assertEquals(expected.width,found.width);
+		assertEquals(expected.height,found.height);
 	}
\ No newline at end of file
