diff --git a/old_product.java b/new_product.java
index f6be583..35e8df4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -17,7 +17,7 @@
 		}
 		//void all the orders associated with this patient
 		OrderService os = Context.getOrderService();
-		List<Order> orders = os.getOrdersByPatient(patient);
+		/*List<Order> orders = os.getOrdersByPatient(patient);
 		if (CollectionUtils.isNotEmpty(orders)) {
 			for (Order order : orders) {
 				if (!order.isVoided()) {
@@ -25,5 +25,5 @@
 					os.voidOrder(order, voidReason);
 				}
 			}
-		}
+		}*/
 	}
\ No newline at end of file
