diff --git a/old_product.java b/new_product.java
index d02b76a..9f6770f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,19 @@ public IoBuffer clear() {
         position = 0;
         mark = UNSET_MARK;
 
+        BufferNode node = buffers.head;
+        int offset = 0;
+
+        while (node != null) {
+            node.buffer.clear();
+            node.offset = offset;
+            offset += node.buffer.limit();
+            node = node.next;
+        }
+
+        limit = offset;
+        buffers.length = 0;
+        buffers.current = buffers.head;
+
         return this;
     }
\ No newline at end of file
