diff --git a/old_test.java b/new_test.java
index cd79138..b5abcf6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,15 +1,17 @@
 @Test
-    public void testClear() {
+    public void testClearEmptyBuffer() {
         ByteBuffer bb1 = ByteBuffer.allocate(4);
-        bb1.put("0123".getBytes());
+        bb1.put("012".getBytes());
         bb1.flip();
 
         ByteBuffer bb2 = ByteBuffer.allocate(4);
-        bb2.put("4567".getBytes());
+        bb2.put("345".getBytes());
         bb2.flip();
 
         IoBuffer ioBuffer = new IoBuffer(bb1, bb2);
 
+        assertEquals(6, ioBuffer.limit());
+
         // Move forward a bit
         ioBuffer.get();
         ioBuffer.get();
@@ -19,5 +21,7 @@
 
         // We should be back to the origin
         assertEquals(0, ioBuffer.position());
+
+        // The limit must have grown
         assertEquals(8, ioBuffer.limit());
     }
\ No newline at end of file
