diff --git a/old_product.java b/new_product.java
index 5b836ed..274f9a5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 @Sensitive
     public static String getCookieValue(Cookie[] cookies, String cookieName) {
-
-        if (cookies == null)
+        if (cookies == null) {
             return null;
+        }
 
         String retVal = null;
         for (int i = 0; i < cookies.length; ++i) {
