diff --git a/old_test.java b/new_test.java
index b795420..25699fc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,4 +4,5 @@
         String value = "cookieValue";
         Cookie[] cookies = new Cookie[] { new Cookie(name, value) };
         assertEquals(value, CookieHelper.getCookieValue(cookies, "COOKIENAME"));
+        assertTrue(CookieHelper.hasCookie(cookies, "COOKIENAME"));
     }
\ No newline at end of file
