diff --git a/old_test.java b/new_test.java
index 9f2fd4f..d25c523 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,6 @@
 	public void toString_shouldIncludeUuidIfNotNull() throws Exception {
 		BaseOpenmrsObject o = new BaseOpenmrsObjectMock();
 		
-		assertEquals("BaseOpenmrsObjectMock{hashCode=" + Integer.toHexString(o.hashCode()) + ", uuid=" + o.getUuid() + "}",
-		    o.toString());
+		assertEquals("BaseOpenmrsObjectTest.BaseOpenmrsObjectMock[hashCode=" + Integer.toHexString(o.hashCode()) + ",uuid="
+		        + o.getUuid() + "]", o.toString());
 	}
\ No newline at end of file
