diff --git a/old_product.java b/new_product.java
index 23b3989..d343fb8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
 	public String toString() {
-		return Objects.toStringHelper(this).add("hashCode", Integer.toHexString(hashCode())).add("uuid", getUuid())
-		        .omitNullValues().toString();
+		return new ToStringBuilder(this, ToStringStyle.SHORT_PREFIX_STYLE).append("hashCode",
+		    Integer.toHexString(hashCode())).append("uuid", getUuid()).build();
 	}
\ No newline at end of file
