diff --git a/old_test.java b/new_test.java
index f90759e..6175c9b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,18 +1,18 @@
 @Test
   public void getPathTest() throws Exception {
-    try (InodePath inodePath = mTree.lockFullInodePath(0, InodeTree.LockMode.READ)) {
+    try (LockedInodePath inodePath = mTree.lockFullInodePath(0, InodeTree.LockMode.READ)) {
       Inode<?> root = inodePath.getInode();
       // test root path
       Assert.assertEquals(new AlluxioURI("/"), mTree.getPath(root));
     }
 
     // test one level
-    InodeTree.CreatePathResult createResult = mTree.createPath(TEST_URI, sDirectoryOptions);
+    InodeTree.CreatePathResult createResult = createPath(mTree, TEST_URI, sDirectoryOptions);
     List<Inode<?>> created = createResult.getCreated();
     Assert.assertEquals(new AlluxioURI("/test"), mTree.getPath(created.get(created.size() - 1)));
 
     // test nesting
-    createResult = mTree.createPath(NESTED_URI, sNestedDirectoryOptions);
+    createResult = createPath(mTree, NESTED_URI, sNestedDirectoryOptions);
     created = createResult.getCreated();
     Assert.assertEquals(new AlluxioURI("/nested/test"),
         mTree.getPath(created.get(created.size() - 1)));
