diff --git a/old_product.java b/new_product.java
index f7d9394..fc61e11 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
 public AlluxioURI getPath(long fileId) throws FileDoesNotExistException {
-    try (InodePath inodePath = mInodeTree.lockFullInodePath(fileId, InodeTree.LockMode.READ)) {
+    try (
+        LockedInodePath inodePath = mInodeTree.lockFullInodePath(fileId, InodeTree.LockMode.READ)) {
       // the path is already locked.
       return mInodeTree.getPath(inodePath.getInode());
     }
