diff --git a/old_test.java b/new_test.java
index 2f3ca5b..6c0fba4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -32,19 +32,17 @@
 
 		CharSequence json = behavior.renderAjaxAttributes(component, attributes);
 
-		String expected = "{\"" + AjaxAttributeName.COMPLETE_HANDLER +
-			"\":[function(attrs, jqXHR, textStatus){alert('Complete!');}],\"" +
-			AjaxAttributeName.URL + "\":\"some/url\",\"" +
-			AjaxAttributeName.PRECONDITION +
-			"\":[function(attrs){return somePrecondition();}],\"" +
+		String expected = "{\"" + AjaxAttributeName.URL + "\":\"some/url\",\"" +
+			AjaxAttributeName.BEFORE_HANDLER + "\":[function(attrs){alert('Before!');}],\"" +
+			AjaxAttributeName.AFTER_HANDLER + "\":[function(attrs){alert('After!');}],\"" +
+			AjaxAttributeName.SUCCESS_HANDLER +
+			"\":[function(attrs, jqXHR, data, textStatus){alert('Success!');}],\"" +
 			AjaxAttributeName.FAILURE_HANDLER +
 			"\":[function(attrs, jqXHR, errorMessage, textStatus){alert('Failure!');}],\"" +
+			AjaxAttributeName.COMPLETE_HANDLER +
+			"\":[function(attrs, jqXHR, textStatus){alert('Complete!');}],\"" +
+			AjaxAttributeName.PRECONDITION + "\":[function(attrs){return somePrecondition();}],\"" +
 			AjaxAttributeName.EXTRA_PARAMETERS +
-			"\":[{\"name\":\"param1\",\"value\":123},{\"name\":\"param2\",\"value\":\"fr_CA\"}],\"" +
-			AjaxAttributeName.BEFORE_HANDLER +
-			"\":[function(attrs){alert('Before!');}],\"" +
-			AjaxAttributeName.SUCCESS_HANDLER +
-			"\":[function(attrs, jqXHR, data, textStatus){alert('Success!');}],\"" +
-			AjaxAttributeName.AFTER_HANDLER + "\":[function(attrs){alert('After!');}]}";
+			"\":[{\"name\":\"param1\",\"value\":123},{\"name\":\"param2\",\"value\":\"fr_CA\"}]}";
 		assertEquals(expected, json);
 	}
\ No newline at end of file
