diff --git a/old_product.java b/new_product.java
index a87e972..a2d59e0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,8 @@
 public static void checkMapConfig(MapConfig mapConfig) {
         checkMergePolicy(mapConfig.isStatisticsEnabled(), mapConfig.getMergePolicyConfig().getPolicy());
         checkNotNative(mapConfig.getInMemoryFormat());
+        checkMergePolicySupportsNative(mapConfig.getName(),
+                mapConfig.getMergePolicyConfig().getPolicy(), mapConfig.getInMemoryFormat());
 
         logIgnoredConfig(mapConfig);
     }
\ No newline at end of file
