diff --git a/old_test.java b/new_test.java
index 5fb8175..8738a52 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,6 @@
 @Test
     public void testGetStaticProperty(){
-        assertEquals(MimeType.JPG.getExtension(), FieldUtil.getStaticProperty("com.feilong.core.io.ImageType", "JPG"));
-        assertEquals(TimeInterval.SECONDS_PER_WEEK, FieldUtil.getStaticProperty("com.feilong.core.date.TimeInterval", "SECONDS_PER_WEEK"));
+        assertEquals(MimeType.JPG.getExtension(), FieldUtil.getStaticFieldValue("com.feilong.core.io.ImageType", "JPG"));
+        assertEquals(TimeInterval.SECONDS_PER_WEEK, FieldUtil.getStaticFieldValue("com.feilong.core.date.TimeInterval", "SECONDS_PER_WEEK"));
+        assertEquals(-1699987643831455524L, FieldUtil.getStaticFieldValue("com.feilong.core.bean.BeanUtilException", "serialVersionUID"));
     }
\ No newline at end of file
