diff --git a/old_product.java b/new_product.java
index f169374..df22dcb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,13 @@
 @SuppressWarnings("unchecked")
-    public static <T> T getStaticProperty(String className,String fieldName){
+    public static <T> T getStaticFieldValue(String className,String fieldName){
         try{
             Class<?> ownerClass = ClassUtil.loadClass(className);
+            // Field field = org.apache.commons.lang3.reflect.FieldUtils.getField(ownerClass, fieldName);
             Field field = ownerClass.getField(fieldName);
             return (T) field.get(ownerClass);
         }catch (Exception e){
-            LOGGER.error(e.getClass().getName(), e);
-            throw new ReflectException(e);
+            String formatMessage = Slf4jUtil.formatMessage("className:[{}],fieldName:[{}]", className, fieldName);
+            LOGGER.error(formatMessage + e.getClass().getName(), e);
+            throw new ReflectException(formatMessage, e);
         }
     }
\ No newline at end of file
