diff --git a/old_test.java b/new_test.java
index 5045a8c..6a58f94 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testEncrypt() throws NoSuchAlgorithmException {
+    public void testEncrypt() throws NoSuchAlgorithmException, CryptoException {
         context.setActiveServerKeySetType(Tls13KeySetType.HANDSHAKE_TRAFFIC_SECRETS);
         context.setConnection(new InboundConnection());
         this.cipher = new RecordAEADCipher(context, KeySetGenerator.generateKeySet(context));
