diff --git a/old_product.java b/new_product.java
index 7b40592..e89dd14 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
-    public EncryptionResult encrypt(EncryptionRequest request) throws CryptoException {
+    public EncryptionResult encrypt(EncryptionRequest request) {
         try {
             byte[] ciphertext;
             encryptIv = new IvParameterSpec(request.getInitialisationVector());
@@ -15,6 +15,7 @@
             return new EncryptionResult(encryptIv.getIV(), ciphertext, useExplicitIv);
 
         } catch (InvalidKeyException | InvalidAlgorithmParameterException ex) {
-            throw new CryptoException(ex);
+            LOGGER.warn("Could not decrypt Data with the provided parameters. Returning unencrypted data.", ex);
+            return new EncryptionResult(request.getPlainText());
         }
     }
\ No newline at end of file
