diff --git a/old_product.java b/new_product.java
index 05feb66..8454fed 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,6 +6,8 @@
 
 		retry:
 		while (!success) {
+			stateHandles.clear();
+
 			Stat stat = client.checkExists().forPath("/");
 			if (stat == null) {
 				break; // Node does not exist, done.
@@ -24,6 +26,9 @@
 					} catch (KeeperException.NoNodeException ignored) {
 						// Concurrent deletion, retry
 						continue retry;
+					} catch (IOException ioException) {
+						LOG.warn("Could not get all ZooKeeper children. Node {} contained " +
+							"corrupted data. Ignoring this node.", path, ioException);
 					}
 				}
 
