diff --git a/old_product.java b/new_product.java
index 72ce4ed..a27f9b2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -14,5 +14,5 @@
         action = new ContextualBiFunction<>(contextDescriptor, action);
 
         CompletableFuture<R> dependentStage = completableFuture.thenCombine(other, action);
-        return new ManagedCompletableFuture<R>(dependentStage, defaultExecutor);
+        return new ManagedCompletableFuture<R>(dependentStage, defaultExecutor, null);
     }
\ No newline at end of file
