diff --git a/old_test.java b/new_test.java
index 7ea3eff..6f07fd7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,17 +1,15 @@
 @Test
-    public void testEvolve() {
-        System.out.println("evolve");
+    public void test() {
+        System.out.println("Genetic Algorithm");
         BitString[] seeds = new BitString[100];
         
         // The mutation parameters are set higher than usual to prevent premature convergence. 
         for (int i = 0; i < seeds.length; i++) {
-            seeds[i] = new BitString(15, new Knapnack(), BitString.Crossover.UNIFORM, 1.0, 0.2);
+            seeds[i] = new BitString(15, new Knapnack(), Crossover.UNIFORM, 1.0, 0.2);
         }
         
-        GeneticAlgorithm<BitString> instance = new GeneticAlgorithm<>(seeds, GeneticAlgorithm.Selection.TOURNAMENT);
-        instance.setElitism(2);
-        instance.setTournament(3, 0.95);
-        
+        GeneticAlgorithm<BitString> instance = new GeneticAlgorithm<>(seeds, Selection.Tournament(3, 0.95), 2);
+
         BitString result = instance.evolve(1000, 18);
         assertEquals(18, result.fitness(), 1E-7);
 
@@ -19,5 +17,4 @@
         for (int i = 0; i < best.length; i++) {
             assertEquals(best[i], result.bits()[i]);
         }
-        
     }
\ No newline at end of file
