diff --git a/old_test.java b/new_test.java
index db03b2e..c758aa7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,6 @@
   public void getOutputStreamTest() throws Exception {
     String uniqPath = PathUtils.uniqPath();
     UnderFileSystemManager manager = new UnderFileSystemManager();
-    long id = manager.createFile(new AlluxioURI(uniqPath));
+    long id = manager.createFile(mSessionId, new AlluxioURI(uniqPath));
     Assert.assertEquals(mMockOutputStream, manager.getOutputStream(id));
   }
\ No newline at end of file
