diff --git a/old_product.java b/new_product.java
index 124322d..df7d3fa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,11 @@
 public OutputStream getOutputStream(long tempUfsFileId) throws FileDoesNotExistException {
-    OutputStreamAgent stream = mOutputAgents.get(tempUfsFileId);
-    if (stream == null) {
+    OutputStreamAgent agent;
+    synchronized (mOutputStreamAgents) {
+      agent = mOutputStreamAgents.getFirstByField(mOutputIdIndex, tempUfsFileId);
+    }
+    if (agent == null) {
       throw new FileDoesNotExistException(
           ExceptionMessage.BAD_WORKER_FILE_ID.getMessage(tempUfsFileId));
     }
-    return stream.getStream();
+    return agent.getStream();
   }
\ No newline at end of file
