diff --git a/old_test.java b/new_test.java
index f30cab2..c768112 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
     cs.execute("create trigger " + TRIGGER_NAME);
     // attach half of the clients to the trigger
     for(int i = ccs.length / 2; i < ccs.length; i++) {
-      ccs[i].attachTrigger(TRIGGER_NAME, new TriggerNotification() {
+      ccs[i].attachTrigger(TRIGGER_NAME, new TriggerEvent() {
         @Override
         public void update(final String data) {
           assertEquals(RETURN_VALUE, data);
@@ -14,7 +14,7 @@
     }
 
     // release a trigger
-    cs.trigger("1 to 10", TRIGGER_NAME, RETURN_VALUE);
+    cs.trigger("1 to 10", TRIGGER_NAME, RETURN_VALUE, "m");
 
     // detach all clients attached to trigger beforehand
     for(int i = ccs.length / 2; i < ccs.length; i++) {
