diff --git a/old_product.java b/new_product.java
index 7b71ed4..06b257c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,10 @@
 private Iter trigger(final QueryContext ctx) throws QueryException {
-    ctx.updating = true;
+    String msg = expr[3].toString().equals("m") ? expr[2].toString() : null;
     Value v = expr[0].value(ctx);
-    ctx.context.triggers.notify(ctx.context.session, expr[1].toString(),
-        checkStr(expr[2], ctx));
+    if(msg == null) {
+      ctx.updating = true;
+      msg = ItemCache.get(v.iter()).toString();
+    }
+    ctx.context.triggers.notify(ctx.context.session, expr[1].toString(), msg);
     return v.iter();
   }
\ No newline at end of file
