diff --git a/old_product.java b/new_product.java
index 0a2b42b..d1ff887 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@ private void add(final String head, final StringList list) {
     final int is = list.size();
     for(int i = 0; i < is; ++i) {
       String line = list.get(i);
-      if(list == strings) line = " " + QUERYSEP + line + ":  " +
+      if(list == strings) line = ' ' + QUERYSEP + line + ":  " +
         Performance.getTimer(stat.get(i) * 10000L * runs, runs);
       text.add(line).nline();
     }
