diff --git a/old_product.java b/new_product.java
index 905fe47..0807ee0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,10 +7,6 @@ public static HighAvailabilityMode fromConfig(Configuration config) {
 			// Map old default to new default
 			return HighAvailabilityMode.NONE;
 		} else {
-			try {
-				return HighAvailabilityMode.valueOf(haMode.toUpperCase());
-			} catch (IllegalArgumentException e) {
-				return FACTORY_CLASS;
-			}
+			return HighAvailabilityMode.valueOf(haMode.toUpperCase());
 		}
 	}
\ No newline at end of file
