diff --git a/old_test.java b/new_test.java
index 8cbf1b3..16786c0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,8 +8,4 @@
 		// Check not equals default
 		config.setString(HighAvailabilityOptions.HA_MODE, HighAvailabilityMode.ZOOKEEPER.name().toLowerCase());
 		assertEquals(HighAvailabilityMode.ZOOKEEPER, HighAvailabilityMode.fromConfig(config));
-
-		// Check factory class
-		config.setString(HighAvailabilityOptions.HA_MODE, "factory.class.FQN");
-		assertEquals(HighAvailabilityMode.FACTORY_CLASS, HighAvailabilityMode.fromConfig(config));
 	}
\ No newline at end of file
