diff --git a/old_test.java b/new_test.java
index 1f4e707..2616764 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,6 @@
     public void testFindByProperties() {
         List list = Collections.singletonList(instance);
         Map props = Collections.singletonMap("name", "abc");
-        when(repository.findByProperties(MyEntity.class, props)).thenReturn(list);
+        when(repository.findByProperties(MyEntity.class, MapParameters.create(props))).thenReturn(list);
         assertEquals(list, AbstractEntity.findByProperties(MyEntity.class, props));
     }
\ No newline at end of file
