diff --git a/old_product.java b/new_product.java
index 9f940f1..b1cee37 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,10 +8,12 @@
             return false;
         SimpleTypedNameDeclaration other = (SimpleTypedNameDeclaration) obj;
         if (type == null) {
+            if (other.type == Object.class)
+                return true;
             if (other.type != null)
                 return false;
         }
-        if (type != null && type.equals(other.type))
+        if (type != null && (type.equals(other.type) || type == Object.class))
             return true;
 
         // if the type is given, only compare the type and don't care about the type image
