diff --git a/old_test.java b/new_test.java
index c1b3eba..e3e4d4b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -35,4 +35,7 @@
         Assert.assertEquals(by(Long.TYPE, "long"), by(null, "char"));
         Assert.assertEquals(by(Long.TYPE, "long"), by(Character.TYPE, "char"));
         Assert.assertEquals(by(Long.TYPE, "long"), by(Character.class, "Character"));
+
+        // should always equal to Object
+        Assert.assertEquals(by(Object.class, "Object"), by(null, "Something"));
     }
\ No newline at end of file
