diff --git a/old_product.java b/new_product.java
index 8cbc3de..fa058e6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,6 @@
 public void open() throws IOException, DatabaseException {
         if (!isOpen()) {
-            cve = new CveDB();
-            cve.open();
+            cve = CveDB.getInstance();
             cpe = CpeMemoryIndex.getInstance();
             try {
                 final long creationStart = System.currentTimeMillis();
