diff --git a/old_product.java b/new_product.java
index 2bd8920..2433d3e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,9 @@
             @Valid
             final Cluster cluster
     ) throws GenieException {
-        LOG.debug("Called to create cluster " + cluster.toString());
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Called to create cluster " + cluster.toString());
+        }
         if (StringUtils.isEmpty(cluster.getId())) {
             cluster.setId(UUID.randomUUID().toString());
         }
