diff --git a/old_product.java b/new_product.java
index d1fffa3..ad9f63e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -18,9 +18,9 @@ void writeToZooKeeper(ZooKeeper zk, ServerConfiguration conf, Version version)
             zk.create(zkPath, data,
                     zkAcls, CreateMode.PERSISTENT);
         } else {
-            if (!(version instanceof ZkVersion)) {
+            if (!(version instanceof LongVersion)) {
                 throw new IllegalArgumentException("Invalid version type, expected ZkVersion type");
             }
-            zk.setData(zkPath, data, ((ZkVersion) version).getZnodeVersion());
+            zk.setData(zkPath, data, (int) ((LongVersion) version).getLongVersion());
         }
     }
\ No newline at end of file
