diff --git a/old_test.java b/new_test.java
index f29436d..1a4971c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,5 +5,5 @@
         // EQUALS
         resultFilter = CompilerUtil.parseFilter(language, "EQUALS(ATTR1, POINT(1 2))");
 
-        Assert.assertTrue("not an instance of Equals", resultFilter instanceof Equals);
+        assertTrue("not an instance of Equals", resultFilter instanceof Equals);
     }
\ No newline at end of file
