diff --git a/old_product.java b/new_product.java
index 31a22f7..6a3b583 100644
--- a/old_product.java
+++ b/new_product.java
@@ -21,8 +21,12 @@ public boolean equals(Object obj) {
 
             // do the conversion dance
             int expressionType = getExpressionType(this);
-            if (expressionType == ExpressionType.LITERAL_GEOMETRY) {
+            if (expressionType == ExpressionType.LITERAL_GEOMETRY
+                    && this.literal instanceof Geometry) {
                 return ((Geometry) this.literal).equalsExact(expLit.evaluate(null, Geometry.class));
+            } else if (expressionType == ExpressionType.LITERAL_GEOMETRY
+                    && this.literal instanceof Envelope) {
+                return ((Envelope) this.literal).equals(expLit.evaluate(null, Envelope.class));
             } else if (expressionType == ExpressionType.LITERAL_INTEGER) {
                 return ((Integer) this.literal).equals(expLit.evaluate(null, Integer.class));
             } else if (expressionType == ExpressionType.LITERAL_STRING) {
