diff --git a/old_product.java b/new_product.java
index 4ee553b..f9f49b1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,11 @@
-public static byte[] deriveSecret(String macAlgorithm, String hashAlgorithm, byte[] prk, String labelIn,
+public static byte[] deriveSecret(HKDFAlgorithm hkdfAlgortihm, String hashAlgorithm, byte[] prk, String labelIn,
             byte[] toHash) {
         try {
             MessageDigest hashFunction = MessageDigest.getInstance(hashAlgorithm);
             hashFunction.update(toHash);
             byte[] hashValue = hashFunction.digest();
-            int outLen = Mac.getInstance(macAlgorithm).getMacLength();
-            return expandLabel(macAlgorithm, prk, labelIn, hashValue, outLen);
+            int outLen = Mac.getInstance(hkdfAlgortihm.getMacAlgorithm().getJavaName()).getMacLength();
+            return expandLabel(hkdfAlgortihm, prk, labelIn, hashValue, outLen);
         } catch (NoSuchAlgorithmException ex) {
             throw new CryptoException("Could not initialize HKDF", ex);
         }
