diff --git a/old_test.java b/new_test.java
index c7787f3..9457bf8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,11 @@
 @Test
-    public void testPrint ()
+    public void testDataString ()
     {
-        System.out.println("print");
-
-        PrintStream        stream = System.out;
-        Histogram<Integer> instance = createHistogram();
-        instance.print(stream);
+        //        System.out.println("dataString");
+        //
+        //        Histogram instance = new Histogram();
+        //        String    expResult = "";
+        //        String    result = instance.dataString();
+        //        assertEquals(expResult, result);
+        //        fail("The test case is a prototype.");
     }
\ No newline at end of file
