diff --git a/old_test.java b/new_test.java
index ad139f4..30e13d3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testStartsWithText() {
-        assertTrue(instance.startsWithText("name", "a").getQueryCriterions()
-                .contains(new StartsWithTextCriterion("name", "a")));
+    	assertEquals(new StartsWithTextCriterion("name", "a"), instance.startsWithText("name", "a").getQueryCriterion());
     }
\ No newline at end of file
