diff --git a/old_test.java b/new_test.java
index 5193df8..b7d9ad6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,8 +20,8 @@
 
         NodeSplit catSplit = catFeatureSmpl1.findBestSplit().get();
         NodeSplit contSplit = contFeatureSmpl1.findBestSplit().get();
-        assertEquals(1.0, catSplit.getValue(), 0.01);
-        assertEquals(-0.5, contSplit.getValue(), 0.01);
+        assertEquals(1.0, catSplit.getVal(), 0.01);
+        assertEquals(-0.5, contSplit.getVal(), 0.01);
         assertFalse(emptyHist.findBestSplit().isPresent());
         assertFalse(catFeatureSmpl2.findBestSplit().isPresent());
     }
\ No newline at end of file
