diff --git a/old_product.java b/new_product.java
index 2d10963..7d8ab4c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,23 +1,22 @@
 @Override
     public VType remove(KType key)
     {
-        final int slot = slotFor(key);
+        final int mask = states.length - 1;
+        int slot = keyHashFunction.hash(key) & mask; 
 
-        final VType value = values[slot];
-        final byte state = states[slot]; 
-        if (state == ASSIGNED)
+        while (states[slot] == ASSIGNED)
         {
-            deleted++;
-            assigned--;
-
-            keys[slot] = Intrinsics.<KType>defaultKTypeValue();
-            values[slot] = Intrinsics.<VType>defaultVTypeValue();
-            states[slot] = DELETED;
-        }
-        else
-        {
-            assert (Intrinsics.defaultVTypeValue() == value) : "Default value expected.";
+            if (/* replaceIf:primitiveKType 
+                (keys[slot] == key) */ 
+                 keyComparator.compare(keys[slot], key) == 0 /* end:replaceIf */ )
+             {
+                assigned--;
+                VType v = values[slot];
+                shiftConflictingKeys(slot);
+                return v;
+             }
+             slot = (slot + 1) & mask;
         }
 
-        return value;
+        return Intrinsics.<VType> defaultVTypeValue();
     }
\ No newline at end of file
