diff --git a/old_test.java b/new_test.java
index b7b0a89..84f0711 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,14 +5,15 @@
 
             // Check pre-create concat.
             String firstSegmentName = getSegmentName(0);
+            long firstSegmentLength = s.getStreamSegmentInfo(firstSegmentName, TIMEOUT).join().getLength();
             AssertExtensions.assertThrows(
                     "concat() did not throw for non-existent target StreamSegment.",
-                    s.concat("foo1", firstSegmentName, TIMEOUT),
+                    s.concat("foo1", 0, firstSegmentName, TIMEOUT),
                     ex -> ex instanceof StreamSegmentNotExistsException);
 
             AssertExtensions.assertThrows(
                     "concat() did not throw for non-existent source StreamSegment.",
-                    s.concat(firstSegmentName, "foo2", TIMEOUT),
+                    s.concat(firstSegmentName, firstSegmentLength, "foo2", TIMEOUT),
                     ex -> ex instanceof StreamSegmentNotExistsException);
 
             ArrayList<String> concatOrder = new ArrayList<>();
@@ -25,7 +26,7 @@
 
                 AssertExtensions.assertThrows(
                         "Concat allowed when source segment is not sealed.",
-                        () -> s.concat(firstSegmentName, segmentName, TIMEOUT),
+                        () -> s.concat(firstSegmentName, firstSegmentLength, segmentName, TIMEOUT),
                         ex -> ex instanceof IllegalStateException);
 
                 // Seal the source segment and then re-try the concat
@@ -33,7 +34,7 @@
                 SegmentProperties preConcatTargetProps = s.getStreamSegmentInfo(firstSegmentName, TIMEOUT).join();
                 SegmentProperties sourceProps = s.getStreamSegmentInfo(segmentName, TIMEOUT).join();
 
-                s.concat(firstSegmentName, segmentName, TIMEOUT).join();
+                s.concat(firstSegmentName, firstSegmentLength, segmentName, TIMEOUT).join();
                 concatOrder.add(segmentName);
                 SegmentProperties postConcatTargetProps = s.getStreamSegmentInfo(firstSegmentName, TIMEOUT).join();
                 AssertExtensions.assertThrows(
