diff --git a/old_test.java b/new_test.java
index 1aa6734..41a7163 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,11 +15,6 @@
         r2.setSeriesLinesVisible(1, true);
         assertTrue(r1.equals(r2));
 
-        r1.setLinesVisible(false);
-        assertFalse(r1.equals(r2));
-        r2.setLinesVisible(false);
-        assertTrue(r1.equals(r2));
-
         r1.setBaseShapesVisible(!r1.getBaseShapesVisible());
         assertFalse(r1.equals(r2));
         r2.setBaseShapesVisible(r1.getBaseShapesVisible());
@@ -30,16 +25,6 @@
         r2.setSeriesShapesVisible(1, true);
         assertTrue(r1.equals(r2));
 
-        r1.setShapesVisible(false);
-        assertFalse(r1.equals(r2));
-        r2.setShapesVisible(false);
-        assertTrue(r1.equals(r2));
-
-        r1.setShapesFilled(false);
-        assertFalse(r1.equals(r2));
-        r2.setShapesFilled(false);
-        assertTrue(r1.equals(r2));
-
         r1.setSeriesShapesFilled(1, true);
         assertFalse(r1.equals(r2));
         r2.setSeriesShapesFilled(1, true);
