diff --git a/old_product.java b/new_product.java
index 7f1f627..b84ace6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,9 +16,6 @@
                 that.seriesLinesVisible)) {
             return false;
         }
-        if (!ObjectUtils.equal(this.linesVisible, that.linesVisible)) {
-            return false;
-        }
         if (this.baseShapesVisible != that.baseShapesVisible) {
             return false;
         }
@@ -26,12 +23,6 @@
                 that.seriesShapesVisible)) {
             return false;
         }
-        if (!ObjectUtils.equal(this.shapesVisible, that.shapesVisible)) {
-            return false;
-        }
-        if (!ObjectUtils.equal(this.shapesFilled, that.shapesFilled)) {
-            return false;
-        }
         if (!ObjectUtils.equal(this.seriesShapesFilled,
                 that.seriesShapesFilled)) {
             return false;
