diff --git a/old_test.java b/new_test.java
index fc3896c..f64c504 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,4 +14,9 @@
         expResult = "Main ø modified to test issue #710 and #801 (&amps;)";
         result = PomUtils.readPom(file);
         assertEquals(expResult, result.getName());
+
+        file = BaseTest.getResourceAsFile(this, "pom/mailapi-1.4.3_projectcomment.pom");
+        expResult = "JavaMail API jar";
+        result = PomUtils.readPom(file);
+        assertEquals(expResult, result.getName());
     }
\ No newline at end of file
