diff --git a/old_product.java b/new_product.java
index fc8dcd2..3d5a54f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,9 @@ private void readObject(java.io.ObjectInputStream in) throws IOException,
         ObjectInputStream.GetField fields = in.readFields();
 
         items = new HashMap<String, Permission>();
-        synchronized (items) {
+        // BEGIN android-changed
+        // copied from a newer version of harmony
+        synchronized (this) {
             permClass = (Class<? extends Permission>)fields.get("permClass", null); //$NON-NLS-1$
             items.putAll((Hashtable<String, Permission>) fields.get(
                     "permissions", new Hashtable<String, Permission>())); //$NON-NLS-1$
@@ -17,4 +19,5 @@ private void readObject(java.io.ObjectInputStream in) throws IOException,
                 throw new InvalidObjectException(Messages.getString("security.25")); //$NON-NLS-1$
             }
         }
+        // END android-changed
     }
\ No newline at end of file
