diff --git a/old_test.java b/new_test.java
index 965a3fc..2efc63f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,9 @@
-@TestInfo(
-      level = TestLevel.COMPLETE,
-      purpose = "",
-      targets = {
-        @TestTarget(
-          methodName = "!Serialization",
-          methodArgs = {}
-        )
-    })
+@TestTargetNew(
+        level = TestLevel.COMPLETE,
+        notes = "",
+        method = "!Serialization",
+        args = {}
+    )
     public void testReadObject() throws Exception {
         String secret = "secret string";
         SealedObject so = new SealedObject(secret, new NullCipher());
