diff --git a/old_product.java b/new_product.java
index a65cfc2..0a00092 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,17 +1,17 @@
-public void bind(final String n, final Object o, final String t)
+public void bind(final String name, final Object val, final String type)
       throws QueryException {
 
-    Object obj = o;
-    if(t != null && !t.isEmpty()) {
-      if(t.equals(QueryText.JSONSTR)) {
-        obj = JsonMapConverter.parse(token(o.toString()), null);
+    Object obj = val;
+    if(type != null && !type.isEmpty()) {
+      if(type.equals(QueryText.JSONSTR)) {
+        obj = JsonMapConverter.parse(token(val.toString()), null);
       } else {
-        final QNm type = new QNm(token(t));
-        if(type.ns()) type.uri(ns.uri(type.pref(), false, null));
-        final Type typ = Types.find(type, true);
+        final QNm nm = new QNm(token(type));
+        if(nm.ns()) nm.uri(ns.uri(nm.pref(), false, null));
+        final Type typ = Types.find(nm, true);
         if(typ != null) obj = typ.e(obj, null);
-        else NOTYPE.thrw(null, type);
+        else NOTYPE.thrw(null, nm);
       }
     }
-    bind(n, obj);
+    bind(name, obj);
   }
\ No newline at end of file
