diff --git a/old_product.java b/new_product.java
index 45c5766..162e76e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,8 @@
 public double sum() {
-        if (sum == null) {
-            double s = 0;
-            DoubleIterator iter = values().iterator();
-            while (iter.hasNext()) {
-                s += iter.nextDouble();
-            }
-            sum = s;
+        double result = 0;
+        DoubleIterator iter = values().iterator();
+        while (iter.hasNext()) {
+            result += iter.nextDouble();
         }
-        return sum;
+        return result;
     }
\ No newline at end of file
