diff --git a/old_test.java b/new_test.java
index f798636..4499dff 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,5 +2,7 @@
     public void testSum() {
         assertThat(emptyVector().sum(), closeTo(0));
         assertThat(singleton().sum(), closeTo(Math.PI));
-        assertThat(simpleVector().sum(), closeTo(7));
+        SparseVector sv = simpleVector();
+        assertThat(sv.sum(), closeTo(7));
+        assertThat(sv.sum(), closeTo(7)); // doubled, to check caching
     }
\ No newline at end of file
