diff --git a/old_test.java b/new_test.java
index b529105..eabdbcc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,11 @@
 @Test
-   public void testUpdateGui()
+   public void testUpdateGui_Graphs()
    {
       System.out.println("updateGui");
+      instance.setSelectedIndex(0);
+      instance.updateGui();
+      instance.setSelectedIndex(1);
+      instance.updateGui();
+      instance.setSelectedIndex(2);
       instance.updateGui();
    }
\ No newline at end of file
