diff --git a/old_test.java b/new_test.java
index 5d1afec..fbb03a7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void testAddWords() throws InterruptedException {
     final VoiceModel model = instantiateVoiceModel();
-    final List<CustomWord> expected = instantiateCustomWords();
+    final List<Word> expected = instantiateWords();
 
     server.enqueue(new MockResponse().setResponseCode(201));
     AddWordsOptions addOptions = new AddWordsOptions.Builder()
