diff --git a/old_product.java b/new_product.java
index da98b6e..9e2b377 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,9 @@
 public ServiceCall<Void> addWords(AddWordsOptions addWordsOptions) {
     Validator.notNull(addWordsOptions, "addWordsOptions cannot be null");
-    RequestBuilder builder = RequestBuilder.post(String.format("/v1/customizations/%s/words", addWordsOptions
-        .customizationId()));
+    List<String> pathSegments = Arrays.asList("v1/customizations", "words");
+    List<String> pathParameters = Arrays.asList(addWordsOptions.customizationId());
+    RequestBuilder builder = RequestBuilder.post(RequestBuilder.constructHttpUrl(getEndPoint(), pathSegments,
+        pathParameters));
     final JsonObject contentJson = new JsonObject();
     contentJson.add("words", GsonSingleton.getGson().toJsonTree(addWordsOptions.words()));
     builder.bodyJson(contentJson);
