diff --git a/old_product.java b/new_product.java
index b87812a..1a81c0a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ private void array() throws QueryIOException {
         conv.openItem();
         value();
         conv.closeItem();
-      } while(consumeWs(',', false) && !(spec == JsonSpec.LIBERAL && curr() == ']'));
+      } while(consumeWs(',', false) && !(liberal && curr() == ']'));
       consumeWs(']', true);
     }
     conv.closeArray();
