diff --git a/old_test.java b/new_test.java
index 9145081..002cb49 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test public void arrayTest() throws QueryIOException {
-    parse("[ ]", RFC4627);
-    parse("[]", "[ ]", RFC4627);
-    parse("[[[[[[42], {}]]]]]", "[ [ [ [ [ [ 42 ], { } ] ] ] ] ]", RFC4627);
-    parse("[ 1, 2, 3, 4, 5, 6, 7, 8 ]", RFC4627);
-    parse("[1,2,3,]", "[ 1, 2, 3 ]", LIBERAL);
+    parse("[ ]", false);
+    parse("[]", "[ ]", false);
+    parse("[[[[[[42], {}]]]]]", "[ [ [ [ [ [ 42 ], { } ] ] ] ] ]", false);
+    parse("[ 1, 2, 3, 4, 5, 6, 7, 8 ]", false);
+    parse("[1,2,3,]", "[ 1, 2, 3 ]", true);
 
-    error("[1,2,3,]", RFC4627);
-    error("[,42]", RFC4627);
-    error("[1,", RFC4627);
+    error("[1,2,3,]", false);
+    error("[,42]", false);
+    error("[1,", false);
   }
\ No newline at end of file
