diff --git a/old_product.java b/new_product.java
index 206f799..6e319c8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,5 +3,5 @@
     final Object o = session().getAttribute(key.toJava());
     if(o == null) return def;
     if(o instanceof Item) return (Item) o;
-    throw SessionErrors.noAttribute(Util.name(o));
+    throw SessionErrors.noAttribute(Util.className(o));
   }
\ No newline at end of file
