diff --git a/old_product.java b/new_product.java
index 8812b4b..e24570c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,12 +7,6 @@ public void skipToEnd() {
             throw new AndroidRuntimeException("Animations may only be started on the main thread");
         }
         if (mRunning) {
-            if (mPendingPosition != UNSET) {
-                mSpring.setFinalPosition(mPendingPosition);
-                mPendingPosition = UNSET;
-            }
-            mValue = mSpring.getFinalPosition();
-            mVelocity = 0;
-            cancel();
+            mEndRequested = true;
         }
     }
\ No newline at end of file
