diff --git a/old_product.java b/new_product.java
index 0a65ceb..26d6256 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,17 +7,17 @@
         int from, to;
         for (from = to = head; from != last; from = oneRight(from, bufLen))
         {
-            if (Intrinsics.equalsKType(e1, buffer[from]))
+            if (Intrinsics.equals(this, e1, buffer[from]))
             {
-                buffer[from] = Intrinsics.<KType>defaultKTypeValue();
+                buffer[from] = Intrinsics.empty();
                 removed++;
                 continue;
             }
-    
+
             if (to != from)
             {
                 buffer[to] = buffer[from];
-                buffer[from] = Intrinsics.<KType>defaultKTypeValue();
+                buffer[from] = Intrinsics.empty();
             }
     
             to = oneRight(to, bufLen);
