diff --git a/old_test.java b/new_test.java
index b1290b2..e0d5fe7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,27 @@
 @Test
-    public void testGetFrequentItemsets_0args() {
-        System.out.println("getFrequentItemsets");
-        FPGrowth fpgrowth = new FPGrowth(itemsets, 3);
-        TotalSupportTree ttree = fpgrowth.buildTotalSupportTree();
-        List<ItemSet> results = ttree.getFrequentItemsets();
+    public void testTTree() {
+        System.out.println("T-Tree");
+
+        FPTree tree = FPTree.build(3, itemsets);
+        TotalSupportTree ttree = new TotalSupportTree(tree);
+
+        int[][] items = {
+            {3, 2, 1},
+            {3},
+            {3, 1},
+            {3, 2},
+            {4},
+            {2}
+        };
+        
+        assertEquals(3, ttree.getSupport(items[0]));
+        assertEquals(8, ttree.getSupport(items[1]));
+        assertEquals(5, ttree.getSupport(items[2]));
+        assertEquals(6, ttree.getSupport(items[3]));
+        assertEquals(3, ttree.getSupport(items[4]));
+        assertEquals(7, ttree.getSupport(items[5]));
+
+        List<ItemSet> results = ttree.stream().collect(Collectors.toList());
         assertEquals(8, results.size());
         
         assertEquals(8, results.get(0).support);
