diff --git a/old_product.java b/new_product.java
index 960e794..0b175c3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,9 @@
-public void add(String name, Object value) 
+public void add(String name, V value) 
     {
-        Object lo = get(name);
-        Object ln = LazyList.add(lo,value);
-        if (lo!=ln)
-            put(name,ln);
+        List<V> lo = get(name);
+        if(lo == null) {
+            lo = new ArrayList<>();
+        }
+        lo.add(value);
+        super.put(name,lo);
     }
\ No newline at end of file
