diff --git a/old_product.java b/new_product.java
index 6ce90b7..ca0725e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,11 @@
-public List<List<Point2D_I32>> fitPolygon( boolean external , double toleranceDist, double toleranceAngle ) {
+public List<List<Point2D_I32>> fitPolygon( boolean external , double splitFraction, double minimumSplitPixels ) {
 		List<List<Point2D_I32>> polygons = new ArrayList<List<Point2D_I32>>();
 
 		int numIterations = 20;
 
 		if( external ) {
 			List<PointIndex_I32> output = ShapeFittingOps.
-					fitPolygon(contour.external, true, toleranceDist, toleranceAngle, numIterations);
+					fitPolygon(contour.external, true, splitFraction, minimumSplitPixels, numIterations);
 
 			List<Point2D_I32> poly = new ArrayList<Point2D_I32>();
 			for( PointIndex_I32 p : output ) {
@@ -15,7 +15,7 @@ public List<List<Point2D_I32>> fitPolygon( boolean external , double toleranceDi
 		} else {
 			for( List<Point2D_I32> i : contour.internal ) {
 				List<PointIndex_I32> output = ShapeFittingOps.
-						fitPolygon(i, true, toleranceDist, toleranceAngle, numIterations);
+						fitPolygon(i, true, splitFraction, minimumSplitPixels, numIterations);
 
 				List<Point2D_I32> poly = new ArrayList<Point2D_I32>();
 				for (PointIndex_I32 p : output) {
