diff --git a/old_test.java b/new_test.java
index bb70735..4629e0f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-	public void updatesRequired_shouldAlwaysHaveAValidUpdateToLatestFile() throws Exception {
+	public void updatesRequired_shouldAlwaysHaveAValidUpdateToLatestFile() throws LockException {
 		// expects /metadata/model to be on the classpath so that
 		// the liquibase-update-to-latest.xml can be found.
 		try {
