diff --git a/old_test.java b/new_test.java
index d052f90..120c63c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,8 +4,8 @@
 		executeDataSet(ORDERS_DATASET_XML);
 		Patient p = Context.getPatientService().getPatient(2);
 		List<Order> orders = Context.getOrderService().getOrdersByPatient(p, true);
-		Assert.assertEquals(12, orders.size());
+		Assert.assertEquals(8, orders.size());
 		
 		orders = Context.getOrderService().getOrdersByPatient(p, false);
-		Assert.assertEquals(8, orders.size());
+		Assert.assertEquals(4, orders.size());
 	}
\ No newline at end of file
