diff --git a/old_product.java b/new_product.java
index da0321c..ad38ba7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,10 @@
-public long getMem()
+public long getMem() throws PerfMonException
    {
-      long ret;
+      long ret = -1;
 
       String value = getData(MEMORY);
-      if (value != null)
-      {
-         ret = Long.parseLong(value);
-      }
-      else
-      {
-         ret = AGENT_ERROR;
-      }
+      if (value != null) ret = Long.parseLong(value);
+      if(ret <= 0) throwNotSupportedMetricException("memory");
 
       return ret;
    }
\ No newline at end of file
