diff --git a/old_product.java b/new_product.java
index 6945799..aba8e33 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static void read(Graph graph, StringReader in, String base, Lang lang)
     {
         StreamRDF dest = StreamRDFLib.graph(graph) ;
-        processTriples(dest, base, in, lang, null) ;
+        process(dest, base, in, lang, null) ;
     }
\ No newline at end of file
