diff --git a/old_test.java b/new_test.java
index 3e83835..49090fc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(groups={"tck"})
-    public void test_atOffset() {
+    public void test_atDate() {
         LocalTime t = LocalTime.of(11, 30);
-        assertEquals(t.atOffset(OFFSET_PTWO), OffsetTime.of(11, 30, OFFSET_PTWO));
+        assertEquals(t.atDate(LocalDate.of(2012, 6, 30)), LocalDateTime.of(2012, 6, 30, 11, 30));
     }
\ No newline at end of file
