diff --git a/old_test.java b/new_test.java
index 78db9ed..963a8bf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
 			"/somethingThatDoesntExistsOnMostMachines-111111111111111111111111111111")));
 
 		java.io.File file = java.io.File.createTempFile("wicket-test--", ".tmp");
-		assertTrue("The just created file should exist!", file.exists());
+		assertTrue("The just created file should exist!", file.isFile());
 
 		boolean removed = Files.remove(file);
 		assertFalse("The just removed file should not exist!", file.exists());
