diff --git a/old_product.java b/new_product.java
index 8735b8e..62eb67d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,33 +1,27 @@
 public static boolean remove(final java.io.File file)
 	{
-		if (file == null || file.exists() == false || file.isDirectory())
+		if (file != null && file.isFile())
 		{
-			return false;
-		}
-
-		int retries = 10;
-
-		boolean deleted = false;
-
-		while ((deleted = file.delete()) == false && retries > 0)
-		{
-			retries--;
-			try
+			for (int j = 0; j < 5; ++j)
 			{
-				Thread.sleep(100);
-			}
-			catch (InterruptedException ignored)
-			{
-			}
-		}
+				for (int i = 0; i < 10; ++i)
+				{
+					if (file.delete())
+					{
+						return true;
+					}
 
-		if (deleted == false && logger.isWarnEnabled())
-		{
-			logger.warn(
-				"Cannot delete file '{}' for unknown reason. The file will be scheduled for deletion at JVM exit time.",
-				file);
-			file.deleteOnExit();
+					try
+					{
+						Thread.sleep(100);
+					}
+					catch (InterruptedException ix)
+					{
+						Thread.currentThread().interrupt();
+					}
+				}
+			}
 		}
 
-		return deleted;
+		return false;
 	}
\ No newline at end of file
