diff --git a/old_test.java b/new_test.java
index ec54646..82a8582 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,6 @@
         versions.add(ProtocolVersion.TLS11);
         versions.add(ProtocolVersion.TLS12);
         versions.add(ProtocolVersion.TLS13);
-        ProtocolVersion highestProtocolVersion = ProtocolVersion.gethighestProtocolVersion(versions);
+        ProtocolVersion highestProtocolVersion = ProtocolVersion.getHighestProtocolVersion(versions);
         assertArrayEquals(ProtocolVersion.TLS13.getValue(), highestProtocolVersion.getValue());
     }
\ No newline at end of file
