diff --git a/old_product.java b/new_product.java
index 348f68b..d643af7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static ProtocolVersion gethighestProtocolVersion(List<ProtocolVersion> list) {
+public static ProtocolVersion getHighestProtocolVersion(List<ProtocolVersion> list) {
         ProtocolVersion highestProtocolVersion = list.get(0);
         for (ProtocolVersion pv : list) {
             if (ArrayConverter.bytesToInt(pv.getValue()) > ArrayConverter.bytesToInt(highestProtocolVersion.getValue())) {
