diff --git a/old_test.java b/new_test.java
index 33b997a..fc7c320 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	public void splitPixels() {
-		SplitMergeLineFitLoop alg = new SplitMergeLineFitLoop(0.01,1,100);
+		SplitMergeLineFitLoop alg = new SplitMergeLineFitLoop(0.01, MINIMUM_SPLIT_FRACTION,100);
 		alg.contour = new ArrayList<Point2D_I32>();
 		for( int i = 0; i < 10; i++ )
 			alg.contour.add( new Point2D_I32(i,0));
