diff --git a/old_product.java b/new_product.java
index 91d9047..7b9fd28 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 static void create(final int... db) throws BaseXException {
     for(final int d : db) {
       final String[] doc = DOCS[d - 1];
-      new CreateDB(doc[0], doc[1]).execute(CONTEXT);
+      new CreateDB(doc[0], doc[1]).execute(context);
     }
   }
\ No newline at end of file
