diff --git a/old_product.java b/new_product.java
index 68f444d..68186d4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,4 @@
-public String getName(){
-        return getMetadataFirstValue(MetadataSchema.DC_SCHEMA, "title", null, Item.ANY);
+@Override
+    public String getName(){
+        return getBitstreamService().getMetadataFirstValue(this, MetadataSchema.DC_SCHEMA, "title", null, Item.ANY);
     }
\ No newline at end of file
