diff --git a/old_product.java b/new_product.java
index 2205ac5..7a5fcf6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
-public static void minus(double[] y, double[] x) {
+public static void add(double[] y, double[] x) {
         if (x.length != y.length) {
             throw new IllegalArgumentException(String.format("Arrays have different length: x[%d], y[%d]", x.length, y.length));
         }
 
         for (int i = 0; i < x.length; i++) {
-            y[i] -= x[i];
+            y[i] += x[i];
         }
     }
\ No newline at end of file
