diff --git a/old_product.java b/new_product.java
index c1fa199..81d117d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ private Message encodeDecode(final Message m1) throws Exception {
 		AtomicReference<Message> m2 = new AtomicReference<Message>();
 		final AlternativeCompositeByteBuf buf = AlternativeCompositeByteBuf.compBuffer();
 		TomP2POutbound encoder = new TomP2POutbound(true,
-				new DefaultSignatureFactory(), new CompByteBufAllocator() {
+				new DSASignatureFactory(), new CompByteBufAllocator() {
 					@Override
 					public AlternativeCompositeByteBuf compBuffer() {
 						return buf;
@@ -16,7 +16,7 @@ private Message encodeDecode(final Message m1) throws Exception {
 		buf.retain();
 		ChannelHandlerContext ctx = mockChannelHandlerContext(buf, m2);
 		encoder.write(ctx, m1, null);
-		Decoder decoder = new Decoder(new DefaultSignatureFactory());
+		Decoder decoder = new Decoder(new DSASignatureFactory());
 		decoder.decode(ctx, buf, m1.getRecipient().createSocketTCP(), m1
 				.getSender().createSocketTCP());
 		return decoder.message();
