diff --git a/old_product.java b/new_product.java
index ad3a6bc..6ddddd3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,5 @@
 @Override
     public void unregisterService(ServiceInstance<T> service) throws Exception
     {
-        clean();
-
-        internalUnregisterService(getOrMakeHolder(service, null));
+        internalUnregisterService(services.remove(service.getId()));
     }
\ No newline at end of file
