diff --git a/old_test.java b/new_test.java
index dbf289f..e737bc1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
   public void getCTLSchemaByIdTest() throws Exception {
     this.loginTenantDeveloper(tenantDeveloperUser);
-    CTLSchemaDto saved = this.createCTLSchema(this.ctlRandomFieldType(), CTL_DEFAULT_NAMESPACE, 1, tenantDeveloperDto.getTenantId(), null, null, null);
-    CTLSchemaDto loaded = client.getCTLSchemaById(saved.getId());
+    CtlSchemaDto saved = this.createCTLSchema(this.ctlRandomFieldType(), CTL_DEFAULT_NAMESPACE, 1, tenantDeveloperDto.getTenantId(), null, null, null);
+    CtlSchemaDto loaded = client.getCTLSchemaById(saved.getId());
     Assert.assertNotNull(loaded);
     Assert.assertEquals(saved, loaded);
   }
\ No newline at end of file
