diff --git a/old_test.java b/new_test.java
index a1d6d4c..cc1d041 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,7 +20,7 @@
 
     // call replenishConnections to warm up connections
     assertCounts(totalConnectionsCount, availableCount);
-    connectionTracker.replenishConnections(this::mockNewConnection);
+    connectionTracker.replenishConnections(this::mockNewConnection, Integer.MAX_VALUE);
     totalConnectionsCount += minActiveConnectionsCount;
     assertCounts(totalConnectionsCount, availableCount);
     List<String> newConnections = getNewlyEstablishedConnections();
@@ -37,7 +37,7 @@
     assertCounts(totalConnectionsCount, availableCount);
 
     // replenish connections again
-    connectionTracker.replenishConnections(this::mockNewConnection);
+    connectionTracker.replenishConnections(this::mockNewConnection, Integer.MAX_VALUE);
     totalConnectionsCount += 2;
     assertCounts(totalConnectionsCount, availableCount);
     newConnections = getNewlyEstablishedConnections();
@@ -59,7 +59,24 @@
     totalConnectionsCount -= 1;
     availableCount += 1;
     assertCounts(totalConnectionsCount, availableCount);
-    connectionTracker.replenishConnections(this::mockNewConnection);
+    connectionTracker.replenishConnections(this::mockNewConnection, Integer.MAX_VALUE);
+    totalConnectionsCount += 1;
+    assertCounts(totalConnectionsCount, availableCount);
+
+    // check out and destroy all host2 connections
+    String connId;
+    while ((connId = connectionTracker.checkOutConnection("host2", sslPort, dataNodeId2)) != null) {
+      connectionTracker.removeConnection(connId);
+    }
+    totalConnectionsCount -= 2;
+    availableCount -= 2;
+    assertCounts(totalConnectionsCount, availableCount);
+
+    // Replenish connection with rate limit of 1 connection per host per call.
+    connectionTracker.replenishConnections(this::mockNewConnection, 1);
+    totalConnectionsCount += 1;
+    assertCounts(totalConnectionsCount, availableCount);
+    connectionTracker.replenishConnections(this::mockNewConnection, 1);
     totalConnectionsCount += 1;
     assertCounts(totalConnectionsCount, availableCount);
   }
\ No newline at end of file
