diff --git a/old_product.java b/new_product.java
index 9a8b764..0a374e5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,12 @@
 private void replenishConnections() {
-    if (networkConfig.networkClientEnableConnectionReplenishment) {
-      int connectionsInitiated = connectionTracker.replenishConnections(this::connect);
-      networkMetrics.connectionReplenished.inc(connectionsInitiated);
+    long currentTimeMs = time.milliseconds();
+    if (networkConfig.networkClientEnableConnectionReplenishment && currentTimeMs >= nextReplenishMs) {
+      int connectionsInitiated = connectionTracker.replenishConnections(this::connect,
+          networkConfig.networkClientMaxReplenishmentPerHostPerSecond);
+      if (connectionsInitiated > 0) {
+        networkMetrics.connectionReplenished.inc(connectionsInitiated);
+        logger.debug("replenishConnections initiated {} connections", connectionsInitiated);
+        nextReplenishMs = currentTimeMs + Time.MsPerSec;
+      }
     }
   }
\ No newline at end of file
