diff --git a/old_test.java b/new_test.java
index a01cc3e..bbb6699 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,7 +12,7 @@
 		// similarity.buildSimilarityMatrix(dataModel, true);
 		context.setSimilarity(similarity);
 		Recommender recommender = new UserKNNRecommender();
-		recommender.recommend(context);
+//		recommender.recommend(context); // TODO: 21/04/2018
 		String filePath = conf.get("dfs.result.dir") + "/model-"
 				+ DriverClassUtil.getDriverName(UserKNNRecommender.class);
 		recommender.saveModel(filePath);
