diff --git a/old_product.java b/new_product.java
index 94439af..9236bfe 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1 @@
-void releaseConnection(
-            HttpClientConnection conn, Object newState, long validDuration, TimeUnit timeUnit);
\ No newline at end of file
+void release(ConnectionEndpoint endpoint, Object newState, long validDuration, TimeUnit timeUnit);
\ No newline at end of file
