diff --git a/old_test.java b/new_test.java
index 10fdf2a..7a538a1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,9 +7,9 @@
         assertNotNull(cid);
 
         // Checks path computation is called as expected
-        assertEquals(1, pathService.edges.size());
-        assertEquals(CP12.deviceId(), pathService.edges.get(0).getKey());
-        assertEquals(CP71.deviceId(), pathService.edges.get(0).getValue());
+        assertEquals(1, topologyService.edges.size());
+        assertEquals(CP12.deviceId(), topologyService.edges.get(0).getKey());
+        assertEquals(CP71.deviceId(), topologyService.edges.get(0).getValue());
 
         // Checks intents are installed as expected
         assertEquals(1, intentService.submitted.size());
