diff --git a/old_test.java b/new_test.java
index ccbf9db..2525bd7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,12 +13,7 @@
         DecisionTreeClassificationTrainer trainer = new DecisionTreeClassificationTrainer(1, 0)
             .withUseIndex(useIdx == 1);
 
-        DecisionTreeNode tree = trainer.fit(
-            data,
-            parts,
-            (k, v) -> VectorUtils.of(Arrays.copyOf(v, v.length - 1)),
-            (k, v) -> v[v.length - 1]
-        );
+        DecisionTreeNode tree = trainer.fit(data, parts, new ArraysVectorizer<Integer>().labeled(1));
 
         assertTrue(tree instanceof DecisionTreeConditionalNode);
 
