diff --git a/old_product.java b/new_product.java
index bb16bd6..cd1e826 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-@Override public <K, V> MultiClassModel<M> fit(DatasetBuilder<K, V> datasetBuilder,
-        FeatureLabelExtractor<K, V, Double> extractor) {
+@Override public <K, V, C extends Serializable> MultiClassModel<M> fit(DatasetBuilder<K, V> datasetBuilder,
+        Vectorizer<K, V, C, Double> extractor) {
 
         return updateModel(null, datasetBuilder, extractor);
     }
\ No newline at end of file
