diff --git a/old_product.java b/new_product.java
index 3d85b9d..358d485 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,4 @@
 static void open(final RESTSession rs) {
     final String db = rs.http.db();
-    if(db == null) return;
-    final String dbpath = rs.http.dbpath();
-
-    rs.add(new Open(db));
-    if(!dbpath.isEmpty()) rs.add(new Cs(_DB_OPEN.args(db, dbpath)));
+    if(db != null) rs.add(new Open(db, rs.http.dbpath()));
   }
\ No newline at end of file
