diff --git a/old_product.java b/new_product.java
index 18d51ea..4bebe14 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@
 
         StreamConfiguration streamConfiguration = ModelHelper.getUpdateStreamConfig(
                 updateStreamRequest, scopeName, streamName);
-        controllerService.alterStream(streamConfiguration).thenApply(streamStatus -> {
+        controllerService.updateStream(streamConfiguration).thenApply(streamStatus -> {
             if (streamStatus.getStatus() == UpdateStreamStatus.Status.SUCCESS) {
                 log.info("Successfully updated stream config for: {}/{}", scopeName, streamName);
                 return Response.status(Status.OK)
