diff --git a/old_test.java b/new_test.java
index 3ffe74a..dd4c538 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,22 +2,9 @@
     public void testApplyFunction()
     {
         System.out.println("applyFunction");
-        Function f = new FunctionBase() {
+       
+        x.applyFunction((x) -> -x);
 
-            /**
-			 * 
-			 */
-			private static final long serialVersionUID = 5260043973153571531L;
-
-			@Override
-            public double f(Vec x)
-            {
-                return -x.get(0);
-            }
-        };
-        
-        x.applyFunction(f);
-        
         DenseVector truth = new DenseVector(new double[]{
             -2.00000000000000e+00,
             -0.00000000000000e+00,
