diff --git a/old_product.java b/new_product.java
index 89128ff..d044a10 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,11 +7,11 @@
     ret.setLength(0);
     ret.setBlockSizeBytes(0);
     ret.setCreationTimeMs(getCreationTimeMs());
-    ret.setIsCompleted(true);
-    ret.setIsFolder(true);
-    ret.setIsPinned(isPinned());
-    ret.setIsCacheable(false);
-    ret.setIsPersisted(isPersisted());
+    ret.setCompleted(true);
+    ret.setFolder(true);
+    ret.setPinned(isPinned());
+    ret.setCacheable(false);
+    ret.setPersisted(isPersisted());
     ret.setBlockIds(null);
     ret.setLastModificationTimeMs(getLastModificationTimeMs());
     ret.setTtl(Constants.NO_TTL);
