diff --git a/old_test.java b/new_test.java
index 80410c2..1bd3442 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,10 +10,10 @@
     Assert.assertEquals(0, info.getLength());
     Assert.assertEquals(0, info.getBlockSizeBytes());
     Assert.assertEquals(inodeDirectory.getCreationTimeMs(), info.getCreationTimeMs());
-    Assert.assertTrue(info.isIsCompleted());
-    Assert.assertTrue(info.isIsFolder());
-    Assert.assertEquals(inodeDirectory.isPinned(), info.isIsPinned());
-    Assert.assertFalse(info.isIsCacheable());
+    Assert.assertTrue(info.isCompleted());
+    Assert.assertTrue(info.isFolder());
+    Assert.assertEquals(inodeDirectory.isPinned(), info.isPinned());
+    Assert.assertFalse(info.isCacheable());
     Assert.assertNull(info.getBlockIds());
     Assert.assertEquals(inodeDirectory.getLastModificationTimeMs(),
         info.getLastModificationTimeMs());
