diff --git a/old_product.java b/new_product.java
index 9a562cb..bcc812f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static List<Contour> contour(ImageUInt8 input, int rule, ImageSInt32 output) {
+public static List<Contour> contour(ImageUInt8 input, ConnectRule rule, ImageSInt32 output) {
 		if( output == null ) {
 			output = new ImageSInt32(input.width,input.height);
 		} else {
