diff --git a/old_test.java b/new_test.java
index 3c272da..bec7857 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,4 +10,8 @@
         Assert.assertFalse(byClass(Map.class).equals(byClass(List.class)));
         Assert.assertFalse(byName("A").equals(byName("B")));
         Assert.assertFalse(byClass(String.class).equals(byName("A")));
+
+        Assert.assertEquals(by(Double.TYPE, "double"), by(null, "double"));
+        Assert.assertEquals(by(Double.class, "Double"), by(null, "double"));
+        Assert.assertEquals(by(Character.class, "Character"), by(null, "char"));
     }
\ No newline at end of file
