diff --git a/old_test.java b/new_test.java
index a9eeb95..d1ef229 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
     public void testShutdown() {
-        final CorfuMsgHandler handler = mock(CorfuMsgHandler.class);
+        final HandlerMethods handler = mock(HandlerMethods.class);
         final ExecutorService executor = mock(ExecutorService.class);
 
         AbstractServer server = new AbstractServer() {
             @Override
-            public CorfuMsgHandler getHandler() {
+            public HandlerMethods getHandler() {
                 return handler;
             }
 
@@ -13,16 +13,6 @@
             public boolean isServerReadyToHandleMsg(CorfuMsg msg) {
                 return getState() == ServerState.READY;
             }
-
-            @Override
-            public ExecutorService getExecutor(CorfuMsgType corfuMsgType) {
-                return executor;
-            }
-
-            @Override
-            public List<ExecutorService> getExecutors() {
-                return Collections.singletonList(executor);
-            }
         };
 
         server.shutdown();
