diff --git a/old_product.java b/new_product.java
index d9b8508..e717503 100644
--- a/old_product.java
+++ b/new_product.java
@@ -23,7 +23,7 @@ NodeBuilder update(Object key)
             }
             else
             {
-                i = find(comparator, key, copyFrom, i + 1, copyFromKeyEnd);
+                i = Arrays.binarySearch(copyFrom, i + 1, copyFromKeyEnd, key, comparator);
                 found = i >= 0;
                 if (!found)
                     i = -i - 1;
@@ -39,7 +39,7 @@ NodeBuilder update(Object key)
                 return null;
             key = next;
         }
-        else if (i == copyFromKeyEnd && compare(comparator, key, upperBound) >= 0)
+        else if (i == copyFromKeyEnd && compareUpperBound(comparator, key, upperBound) >= 0)
             owns = false;
 
         if (isLeaf(copyFrom))
