diff --git a/old_product.java b/new_product.java
index e0e5e55..a3ce5b7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-static String createBinaryString(IpPrefix ip4Prefix) {
+static String createBinaryString(Ip4Prefix ip4Prefix) {
         if (ip4Prefix.prefixLength() == 0) {
             return "";
         }
@@ -6,7 +6,7 @@ static String createBinaryString(IpPrefix ip4Prefix) {
         StringBuilder result = new StringBuilder(ip4Prefix.prefixLength());
         long value = ip4Prefix.address().toInt() & 0xffffffffL;
         for (int i = 0; i < ip4Prefix.prefixLength(); i++) {
-            long mask = 1 << (IpPrefix.MAX_INET_MASK_LENGTH - 1 - i);
+            long mask = 1 << (Ip4Prefix.MAX_MASK_LENGTH - 1 - i);
             result.append(((value & mask) == 0) ? "0" : "1");
         }
         return result.toString();
