diff --git a/old_test.java b/new_test.java
index 9466645..a806e24 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
-    public void testString2Date(){
-        logDate(DateUtil.string2Date(FROMSTRING, DatePattern.COMMON_DATE));
+    public void testToDate(){
+        logDate(DateUtil.toDate(FROMSTRING, DatePattern.COMMON_DATE));
 
         // 商品上线时间
-        logDate(DateUtil.string2Date("20130102140806000", DatePattern.TIMESTAMP_WITH_MILLISECOND));
+        logDate(DateUtil.toDate("20130102140806000", DatePattern.TIMESTAMP_WITH_MILLISECOND));
     }
\ No newline at end of file
