diff --git a/old_product.java b/new_product.java
index aacad21..4b992b6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,4 @@
-public static Date string2Date(String dateString,String datePattern){
-        return DateFormatUtil.parse(dateString, datePattern);
+@Deprecated
+    public static Date string2Date(String dateString,String datePattern){
+        return toDate(dateString, datePattern);
     }
\ No newline at end of file
