diff --git a/old_test.java b/new_test.java
index 4f34f75..b656d09 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,7 +18,7 @@
         try (Reader fileReader = new FileReader(Paths.get(TestUtils.getRemoteFile(new URL("http://www.datumbox.com/files/datasets/cities.csv"))).toFile())) {
             dataset = Dataframe.Builder.parseCSVFile(fileReader, "metro_population", headerDataTypes, ',', '"', "\r\n", dbConf);
         }
-        catch(IOException ex) {
+        catch(UncheckedIOException | IOException ex) {
             logger.warn("Unable to download datasets, skipping test.");
             return;
         }
