diff --git a/old_product.java b/new_product.java
index d16a6fd..a6ab4c4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -48,7 +48,7 @@ public static Dataframe parseCSVFile(Reader reader, String yVariable, Map<String
             } 
             catch (IOException ex) {
                 dataset.delete();
-                throw new RuntimeException(ex);
+                throw new UncheckedIOException(ex);
             }
             return dataset;
         }
\ No newline at end of file
