diff --git a/old_test.java b/new_test.java
index b9625fa..290c18d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
-    public void testGetPropertiesMapForLog(){
-        LOGGER.debug(JsonUtil.format(SystemUtil.getPropertiesMapForLog()));
+    public void testGetPropertiesMap(){
+        LOGGER.debug(JsonUtil.format(SystemUtil.getPropertiesMap()));
     }
\ No newline at end of file
