diff --git a/old_product.java b/new_product.java
index 0b9ad97..4bd9a7f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,3 @@
-public static Map<String, String> getPropertiesMapForLog(){
-        Properties properties = System.getProperties();
-        return new TreeMap<String, String>(ConvertUtil.toMap(properties));
+public static Map<String, String> getPropertiesMap(){
+        return new TreeMap<String, String>(toMap(System.getProperties()));
     }
\ No newline at end of file
