diff --git a/old_test.java b/new_test.java
index 7687f75..a2c4e86 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testLt() {
-        assertTrue(instance.lt("id", 5).getQueryCriterions()
-                .contains(new LtCriterion("id", 5)));
+    	assertEquals(new LtCriterion("id", 5), instance.lt("id", 5).getQueryCriterion());
     }
\ No newline at end of file
