diff --git a/old_product.java b/new_product.java
index f119fd2..68c3e40 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ public Value cast(final Item it, final QueryContext qc, final StaticContext sc,
       if(!error && info != null) info.check(true);
       final Value v = type.cast(it, qc, sc, info);
       if(kind != null) {
-        for(final Item i : v) if(!kind.eq(it)) throw Err.castError(info, i, type);
+        for(final Item i : v) if(!kind.eq(it)) throw castError(info, i, type);
       }
       return v;
     } catch(final QueryException ex) {
