diff --git a/old_test.java b/new_test.java
index f371b07..0363482 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,8 @@
         ApplicationDto application = createApplication(tenantAdminDto);
         EventClassFamilyDto eventClassFamily = createEventClassFamily(application.getTenantId());
         for (int i=0;i<10;i++) {
-            ApplicationEventFamilyMapDto applicationEventFamilyMap = createApplicationEventFamilyMap(application.getId(), eventClassFamily.getId(), (i+1));
+            ApplicationEventFamilyMapDto applicationEventFamilyMap = createApplicationEventFamilyMap(
+                    application.getApplicationToken(), eventClassFamily.getId(), (i+1));
             applicationEventFamilyMaps.add(applicationEventFamilyMap);
         }
 
