diff --git a/old_test.java b/new_test.java
index 2dbf1c7..24cac31 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,8 @@
 		concept.addName(new ConceptName("same name", Context.getLocale()));
 		concept.addName(new ConceptName("same name", Context.getLocale()));
 		concept.addDescription(new ConceptDescription("some description",null));
+		concept.setConceptClass(new ConceptClass());
+		concept.setDatatype(new ConceptDatatype());
 		Errors errors = new BindException(concept, "concept");
 		new ConceptValidator().validate(concept, errors);
 	}
\ No newline at end of file
