diff --git a/old_product.java b/new_product.java
index 92a8d2c..3483243 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,6 +15,9 @@ public void validate(Object obj, Errors errors) throws APIException, DuplicateCo
 			return;
 		}
 
+		ValidationUtils.rejectIfEmpty(errors, "datatype", "Concept.datatype.empty");
+		ValidationUtils.rejectIfEmpty(errors, "conceptClass", "Concept.conceptClass.empty");
+
 		boolean hasFullySpecifiedName = false;
 		for (Locale conceptNameLocale : conceptToValidate.getAllConceptNameLocales()) {
 			boolean fullySpecifiedNameForLocaleFound = false;
