diff --git a/old_test.java b/new_test.java
index 59a26c7..fd62eea 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,15 +1,18 @@
 @Test
     public void testSeal() throws Exception {
+        final String context = "Seal";
         try (Storage s = createStorage()) {
             // Check invalid handle.
             assertThrows("seal() did not throw for invalid handle.",
                     () -> s.seal(createInvalidHandle("foo"), TIMEOUT),
                     ex -> ex instanceof InvalidSegmentHandleException);
 
-            HashMap<String, ByteArrayOutputStream> appendData = populate(s);
+            HashMap<String, ByteArrayOutputStream> appendData = populate(s, context);
             for (String segmentName : appendData.keySet()) {
                 val handle = s.open(segmentName, TIMEOUT).join();
-                s.seal(handle, TIMEOUT).join();
+                val segmentInfo = s.seal(handle, TIMEOUT).join();
+                Assert.assertTrue("seal() did not return a segmentInfo with isSealed == true", segmentInfo.isSealed());
+
                 assertThrows("seal() did not throw for an already sealed StreamSegment.",
                         () -> s.seal(handle, TIMEOUT),
                         ex -> ex instanceof StreamSegmentSealedException);
