diff --git a/old_test.java b/new_test.java
index 30d58f0..95315d0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
     public void testPlusYears() {
-        assertEquals(testDate.plusYears(10), JapaneseDate.japaneseDate(testYear+10, testMonthOfYear, testDayOfMonth));
+        assertEquals(testDate.plusYears(10), JapaneseDate.of(testYear+10, testMonthOfYear, testDayOfMonth));
     }
\ No newline at end of file
