diff --git a/old_test.java b/new_test.java
index 72ea043..aeb5d85 100644
--- a/old_test.java
+++ b/new_test.java
@@ -21,7 +21,7 @@
         List<Integer> result = IteratorUtils.toList(it);
         assertEquals(12, result.size());
 
-        List<Integer> combinedList = new ArrayList<Integer>();
+        List<Integer> combinedList = new ArrayList<>();
         combinedList.addAll(collectionOdd);
         combinedList.addAll(collectionEven);
         Collections.sort(combinedList);
