diff --git a/old_product.java b/new_product.java
index f47481f..795c9b1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,5 +3,5 @@ public static <E> Iterator<E> collatedIterator(final Comparator<? super E> compa
         @SuppressWarnings("unchecked")
         final Comparator<E> comp =
             comparator == null ? ComparatorUtils.NATURAL_COMPARATOR : (Comparator<E>) comparator;
-        return new CollatingIterator<E>(comp, iterators);
+        return new CollatingIterator<>(comp, iterators);
     }
\ No newline at end of file
