diff --git a/old_product.java b/new_product.java
index dbabbf4..a817432 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,7 @@
     Command cmdFromMaster = null;
     try {
       cmdFromMaster = mMasterClient
-          .heartbeat(WorkerIdRegistry.getWorkerId(), storeMeta.getUsedBytesOnTiers(),
+          .heartbeat(mWorkerId.get(), storeMeta.getUsedBytesOnTiers(),
               blockReport.getRemovedBlocks(), blockReport.getAddedBlocks());
       handleMasterCommand(cmdFromMaster);
       mLastSuccessfulHeartbeatMs = System.currentTimeMillis();
