diff --git a/old_product.java b/new_product.java
index ea7b267..7aad963 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,18 +1,18 @@
 public static String stripJSessionId(final String url)
 	{
-		if (url == null)
+		if (Strings.isEmpty(url))
 		{
-			return null;
+			return url;
 		}
 
 		// http://.../abc;jsessionid=...?param=...
-		int ixSemiColon = url.indexOf(";");
+		int ixSemiColon = url.toLowerCase(Locale.ENGLISH).indexOf(";jsessionid=");
 		if (ixSemiColon == -1)
 		{
 			return url;
 		}
 
-		int ixQuestionMark = url.indexOf("?");
+		int ixQuestionMark = url.indexOf('?');
 		if (ixQuestionMark == -1)
 		{
 			// no query paramaters; cut off at ";"
