diff --git a/old_test.java b/new_test.java
index 1ab5e6d..5feaa5d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,4 +11,10 @@
 		assertEquals(url + "?param", Strings.stripJSessionId(url + ";jsessionid=12345?param"));
 		assertEquals(url + "?param=a;b",
 			Strings.stripJSessionId(url + ";jsessionid=12345?param=a;b"));
+
+		// WICKET-4816
+		assertEquals(url + ";a=b;c=d",
+				Strings.stripJSessionId(url + ";a=b;c=d;jsessionid=12345"));
+		assertEquals(url + ";a=b;c=d?param=a;b",
+				Strings.stripJSessionId(url + ";a=b;c=d;jsessionid=12345?param=a;b"));
 	}
\ No newline at end of file
