diff --git a/old_test.java b/new_test.java
index 9a2af9f..76b7bfc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,7 @@
         // Ensure that 'id' attribute is optional
         String field = null; // not used
         String text = "north 40";
-        int collection = 0;
+        Collection collection = null;
         int start = 0;
         int limit = 0;
         String locale = null;
