diff --git a/old_product.java b/new_product.java
index b2c1fad..ca2e9dd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-public Choices getMatches(String field, String text, int collection, int start, int limit, String locale);
\ No newline at end of file
+public Choices getMatches(String field, String text, Collection collection, int start, int limit, String locale);
\ No newline at end of file
