diff --git a/old_product.java b/new_product.java
index 3d2be28..6169d83 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,4 @@
-public void clear() {
-        this.readIndex.clear();
+void clear() {
+        this.cacheUpdater.clear();
         this.inMemoryOperationLog.clear();
-        synchronized (this.readIndex) {
-            this.recentStreamSegmentIds = new HashSet<>();
-        }
     }
\ No newline at end of file
