diff --git a/old_product.java b/new_product.java
index 08ffccc..6b364f3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -17,7 +17,7 @@ public void validate(Object obj, Errors errors) {
 			
 			validateSamePatientInOrderAndEncounter(order, errors);
 			validateOrderTypeClass(order, errors);
-			validateStartDate(order, errors);
+			validateDateActivated(order, errors);
 			validateScheduledDate(order, errors);
 		}
 	}
\ No newline at end of file
