diff --git a/old_product.java b/new_product.java
index 3dd772d..b96c360 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1,10 @@
-public void configure(Configuration config);
\ No newline at end of file
+public static <T extends Value> Class<FieldParser<T>> getParserForType(Class<T> type) {
+		Class<? extends FieldParser<?>> parser = PARSERS.get(type);
+		if (parser == null) {
+			return null;
+		} else {
+			@SuppressWarnings("unchecked")
+			Class<FieldParser<T>> typedParser = (Class<FieldParser<T>>) parser;
+			return typedParser;
+		}
+	}
\ No newline at end of file
