diff --git a/old_test.java b/new_test.java
index 8637f67..bf83cda 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,14 +3,14 @@
         CompositeByteBuf buf = (CompositeByteBuf) Unpooled.wrappedBuffer(new byte[] { 1, 2, 3, 4, 5 }, new byte[] {4, 5, 6, 7, 8, 9, 26});
     
         //Ensure that a random place will be fine
-        assertEquals(buf.getBufferFor(2).capacity(), 5);
+        assertEquals(buf.getBuffer(2).capacity(), 5);
         
         //Loop through each byte
         
         byte index = 0;
         
         while (index < buf.capacity()) {
-            ByteBuf _buf = buf.getBufferFor(index++);
+            ByteBuf _buf = buf.getBuffer(index++);
             assertNotNull(_buf);
             assertTrue(_buf.capacity() > 0);
             assertNotNull(_buf.getByte(0));
