diff --git a/old_product.java b/new_product.java
index e585892..b4295c1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public ByteBuf getBufferFor(int index) throws IndexOutOfBoundsException {
+public ByteBuf getBuffer(int index) throws IndexOutOfBoundsException {
         if (index < 0 || index >= capacity()) {
             throw new IndexOutOfBoundsException("Invalid index: " + index
                     + " - Bytes needed: " + index + ", maximum is "
