diff --git a/old_product.java b/new_product.java
index 4bdb177..83bc2e0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ public static DigestAlgorithm getDigestAlgorithm(ProtocolVersion protocolVersion
         if (cipherSuite.usesGOSTR3411()) {
             result = DigestAlgorithm.GOSTR3411;
         } else if (cipherSuite.usesGOSTR34112012()) {
-            result = DigestAlgorithm.GOSTR3411_2012_256;
+            result = DigestAlgorithm.GOSTR34112012_256;
         } else if (protocolVersion == ProtocolVersion.TLS10 || protocolVersion == ProtocolVersion.TLS11
                 || protocolVersion == ProtocolVersion.DTLS10) {
             result = DigestAlgorithm.LEGACY;
