diff --git a/old_product.java b/new_product.java
index b30f3ab..ecce7f1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public final void bug15375(int num) {
-        Workbook wb = _testDataProvider.createWorkbook();
-        Sheet sheet = wb.createSheet();
-        CreationHelper factory = wb.getCreationHelper();
+public final void bug15375(int num) throws Exception {
+        Workbook wb1 = _testDataProvider.createWorkbook();
+        Sheet sheet = wb1.createSheet();
+        CreationHelper factory = wb1.getCreationHelper();
 
         String tmp1 = null;
         String tmp2 = null;
@@ -21,7 +21,10 @@ public final void bug15375(int num) {
             cell = row.createCell(2);
             cell.setCellValue(factory.createRichTextString(tmp3));
         }
-        wb = _testDataProvider.writeOutAndReadBack(wb);
+        Workbook wb2 = _testDataProvider.writeOutAndReadBack(wb1);
+        wb1.close();
+
+        sheet = wb2.getSheetAt(0);
         for (int i = 0; i < num; i++) {
             tmp1 = "Test1" + i;
             tmp2 = "Test2" + i;
@@ -33,4 +36,5 @@ public final void bug15375(int num) {
             assertEquals(tmp2, row.getCell(1).getStringCellValue());
             assertEquals(tmp3, row.getCell(2).getStringCellValue());
         }
+        wb2.close();
     }
\ No newline at end of file
