diff --git a/old_product.java b/new_product.java
index 2cff499..2ea4ef9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,11 +3,12 @@ public CompleteUfsFileTOptions toThrift() {
     if (hasGroup()) {
       options.setGroup(mGroup);
     }
-    if (hasUser()) {
-      options.setUser(mUser);
+    if (hasOwner()) {
+      options.setUser(mOwner); // remove in 2.0
+      options.setOwner(mOwner);
     }
-    if (hasPermission()) {
-      options.setPermission(mPermission);
+    if (hasMode()) {
+      options.setMode(mMode);
     }
     return options;
   }
\ No newline at end of file
