diff --git a/old_test.java b/new_test.java
index b28c46d..d49763c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,16 +1,16 @@
 @Test
   public void toThriftTest() throws IOException {
     CompleteUfsFileOptions options = CompleteUfsFileOptions.defaults();
-    String user = "test-user";
+    String owner = "test-owner";
     String group = "test-group";
-    short permission = Constants.DEFAULT_FS_FULL_PERMISSION;
+    short permission = Constants.DEFAULT_FILE_SYSTEM_MODE;
 
-    options.setUser(user);
+    options.setOwner(owner);
     options.setGroup(group);
-    options.setPermission(permission);
+    options.setMode(permission);
 
     CompleteUfsFileTOptions thriftOptions = options.toThrift();
-    Assert.assertEquals(user, thriftOptions.getUser());
+    Assert.assertEquals(owner, thriftOptions.getOwner());
     Assert.assertEquals(group, thriftOptions.getGroup());
-    Assert.assertEquals(permission, thriftOptions.getPermission());
+    Assert.assertEquals(permission, thriftOptions.getMode());
   }
\ No newline at end of file
