diff --git a/old_product.java b/new_product.java
index f68d329..5f314d1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,6 @@
 public static char boolsToHexDigit(boolean[] src, int srcPos) {
+        if (src.length == 0)
+            throw new IllegalArgumentException("Cannot convert empty array.");
         if (src.length > srcPos + 3 && src[srcPos + 3]) {
             if (src.length > srcPos + 2 && src[srcPos + 2]) {
                 if (src.length > srcPos + 1 && src[srcPos + 1]) {
