diff --git a/old_test.java b/new_test.java
index b7ab2be..e22390d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -19,9 +19,9 @@
         assertEquals('1', Conversion.boolsToHexDigit(new boolean[]{true}));
         assertEquals('f', Conversion.boolsToHexDigit(new boolean[]{true, true, true, true, true}));
         try {
-            assertEquals('0', Conversion.boolsToHexDigit(new boolean[]{}));
-            fail("Thrown " + ArrayIndexOutOfBoundsException.class.getName() + " expected");
-        } catch (final ArrayIndexOutOfBoundsException e) {
+            Conversion.boolsToHexDigit(new boolean[]{});
+            fail("Thrown " + IllegalArgumentException.class.getName() + " expected");
+        } catch (final IllegalArgumentException e) {
             // OK
         }
     }
\ No newline at end of file
