diff --git a/old_test.java b/new_test.java
index 014f2a7..f1dd1df 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
             @Override
             public Boolean answer(final InvocationOnMock invocation) throws Throwable {
                 final String sql = (String)invocation.getArguments()[0];
-                if (sql.equals("DROP TABLE IF EXISTS invalid")) {
+                if (sql.equals("DROP TABLE IF EXISTS `invalid`")) {
                     throw new SQLException();
                 }
                 return true;
@@ -17,9 +17,9 @@
 
         // Test dropping table with success
         final TableRegisterSupport support = new TableRegisterSupport(connection);
-        Assert.assertTrue(support.dropTable("feed"));
-        Mockito.verify(statement).execute("DROP TABLE IF EXISTS feed");
+        Assert.assertTrue(support.dropTable("`feed`"));
+        Mockito.verify(statement).execute("DROP TABLE IF EXISTS `feed`");
 
         // Test dropping table with exception
-        Assert.assertFalse(support.dropTable("invalid"));
+        Assert.assertFalse(support.dropTable("`invalid`"));
     }
\ No newline at end of file
