diff --git a/old_product.java b/new_product.java
index ff5d616..8040494 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public void dropTable(String table) {
+public void dropTable(@Nonnull final String schema, @Nonnull final String table) {
         // Make managed to remove the old data
-        String makeManagedSQL = "alter table " + table + " SET TBLPROPERTIES ('EXTERNAL'='FALSE')";
+        String makeManagedSQL = "alter table " + HiveUtils.quoteIdentifier(schema, table) + " SET TBLPROPERTIES ('EXTERNAL'='FALSE')";
         doExecuteSQL(makeManagedSQL);
-        String sql = "DROP TABLE " + table;
+        String sql = "DROP TABLE " + HiveUtils.quoteIdentifier(schema, table);
         doExecuteSQL(sql);
     }
\ No newline at end of file
