diff --git a/old_product.java b/new_product.java
index 1e788c7..4728dc6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static File getMuleAppsFile()
+public static File getMuleAppsDir()
     {
-        return isStandalone() ? new File(getMuleHomeFile(), MULE_APPS_FILENAME) : null;
+        return isStandalone() ? new File(getMuleHome(), MULE_APPS_FILENAME) : null;
     }
\ No newline at end of file
