diff --git a/old_test.java b/new_test.java
index f7736e4..9b74f9a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testGetMuleAppsFile()
     {
-        File muleApps = MuleContainerBootstrapUtils.getMuleAppsFile(); 
+        File muleApps = MuleContainerBootstrapUtils.getMuleAppsDir();
         assertNotNull(muleApps.getAbsolutePath());
     }
\ No newline at end of file
