diff --git a/old_test.java b/new_test.java
index 41bdb52..c54d1ba 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,9 +8,8 @@
             newValues.put("field" + i, new StringByteIterator("newvalue" + i));
         }
 
-        int expResult = 0;
-        int result = instance.update(MOCK_TABLE, MOCK_KEY1, newValues);
-        assertEquals(expResult, result);
+        Status result = instance.update(MOCK_TABLE, MOCK_KEY1, newValues);
+        assertEquals(Status.OK, result);
 
         //validate that the values changed
         HashMap<String, ByteIterator> resultParam = new HashMap<String, ByteIterator>(10);
