diff --git a/old_test.java b/new_test.java
index a806e24..19e5aed 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,8 @@
 @Test
-    public void testToDate(){
-        logDate(DateUtil.toDate(FROMSTRING, DatePattern.COMMON_DATE));
+    public void testToDate() throws ParseException{
+        //logDate(DateUtils.parseDate("2016-02-33", DatePattern.COMMON_DATE));
+        Date date = DateUtil.toDate("2016-02-33", DatePattern.COMMON_DATE);
+        logDate(date);
 
         // 商品上线时间
         logDate(DateUtil.toDate("20130102140806000", DatePattern.TIMESTAMP_WITH_MILLISECOND));
