diff --git a/old_product.java b/new_product.java
index e166cbd..88ce1fa 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,13 +3,6 @@ protected void splitPixels( int indexStart , int indexStop ) {
 		if( indexStart+1 >= indexStop )
 			return;
 
-		// end points of the line
-		Point2D_I32 a = contour.get(indexStart);
-		Point2D_I32 c = contour.get(indexStop);
-
-		line.p.set(a.x,a.y);
-		line.slope.set(c.x-a.x,c.y-a.y);
-
 		int indexSplit = selectSplitBetween(indexStart, indexStop);
 
 		if( indexSplit >= 0 ) {
