diff --git a/old_test.java b/new_test.java
index fc7c320..a7ceb19 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
 	public void splitPixels() {
 		SplitMergeLineFitLoop alg = new SplitMergeLineFitLoop(0.01, MINIMUM_SPLIT_FRACTION,100);
-		alg.contour = new ArrayList<Point2D_I32>();
+		alg.contour = new ArrayList<>();
 		for( int i = 0; i < 10; i++ )
 			alg.contour.add( new Point2D_I32(i,0));
 		alg.N = alg.contour.size();
