diff --git a/old_test.java b/new_test.java
index 1be58ad..6eed6cc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,12 +2,15 @@
   public void testGetExtensionFilesToLoad_with_load_list() throws IOException
   {
     final File extensionsDir = temporaryFolder.newFolder();
+
+    final File absolutePathExtension = temporaryFolder.newFolder();
+
     final ExtensionsConfig config = new ExtensionsConfig()
     {
       @Override
       public List<String> getLoadList()
       {
-        return Arrays.asList("mysql-metadata-storage", "druid-kafka-eight");
+        return Arrays.asList("mysql-metadata-storage", "druid-kafka-eight", absolutePathExtension.getAbsolutePath());
       }
 
       @Override
@@ -19,12 +22,12 @@
     final File mysql_metadata_storage = new File(extensionsDir, "mysql-metadata-storage");
     final File druid_kafka_eight = new File(extensionsDir, "druid-kafka-eight");
     final File random_extension = new File(extensionsDir, "random-extensions");
+
     mysql_metadata_storage.mkdir();
     druid_kafka_eight.mkdir();
     random_extension.mkdir();
 
-    final File[] expectedFileList = new File[]{druid_kafka_eight, mysql_metadata_storage};
+    final File[] expectedFileList = new File[]{mysql_metadata_storage, druid_kafka_eight, absolutePathExtension};
     final File[] actualFileList = Initialization.getExtensionFilesToLoad(config);
-    Arrays.sort(actualFileList);
     Assert.assertArrayEquals(expectedFileList, actualFileList);
   }
\ No newline at end of file
