diff --git a/old_product.java b/new_product.java
index 1883ce6..f4c881c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,7 +12,11 @@ public static File[] getExtensionFilesToLoad(ExtensionsConfig config)
       int i = 0;
       extensionsToLoad = new File[toLoad.size()];
       for (final String extensionName : toLoad) {
-        final File extensionDir = new File(rootExtensionsDir, extensionName);
+        File extensionDir = new File(extensionName);
+        if (!extensionDir.isAbsolute()) {
+          extensionDir = new File(rootExtensionsDir, extensionName);
+        }
+
         if (!extensionDir.isDirectory()) {
           throw new ISE(
               String.format(
