diff --git a/old_product.java b/new_product.java
index 4028911..f543fdb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,8 @@ public void validate(Object obj, Errors errors) {
 		if (orderType == null) {
 			errors.rejectValue("orderType", "error.general");
 		} else {
-			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "name", "error.name");
+			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "localizedName.unlocalizedValue",
+			    "LocalizedName.unlocalizedName.empty");
 			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "description", "error.description");
 		}
 		//log.debug("errors: " + errors.getAllErrors().toString());
