diff --git a/old_product.java b/new_product.java
index c1beb7f..b4ed298 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @RequestMapping(value="profileSchemas/{applicationId}", method=RequestMethod.GET)
     @ResponseBody
-    public List<ProfileSchemaDto> getProfileSchemasByApplicationId(@PathVariable String applicationId) throws KaaAdminServiceException {
+    public List<EndpointProfileSchemaDto> getProfileSchemasByApplicationId(@PathVariable String applicationId) throws KaaAdminServiceException {
         return kaaAdminService.getProfileSchemasByApplicationId(applicationId);
     }
\ No newline at end of file
