diff --git a/old_test.java b/new_test.java
index c2e712f..302b819 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,31 +1,31 @@
 @Test
     public void testGetProfileSchemasByApplicationId() throws Exception {
         
-        List<ProfileSchemaDto> profileSchemas  = new ArrayList<>(11);
+        List<EndpointProfileSchemaDto> profileSchemas  = new ArrayList<>(11);
         ApplicationDto application = createApplication(tenantAdminDto);
         
         loginTenantDeveloper(tenantDeveloperDto.getUsername());
         
         CTLSchemaInfoDto ctlSchema = this.createCTLSchema(this.ctlRandomFieldType(), CTL_DEFAULT_NAMESPACE, 1, CTLSchemaScopeDto.TENANT, null, null, null);
 
-        List<ProfileSchemaDto> defaultProfileSchemas = client.getProfileSchemas(application.getId());
+        List<EndpointProfileSchemaDto> defaultProfileSchemas = client.getProfileSchemas(application.getId());
         profileSchemas.addAll(defaultProfileSchemas);
 
         for (int i=0;i<10;i++) {
-            ProfileSchemaDto profileSchema = createProfileSchema(application.getId(), ctlSchema.getId());
+            EndpointProfileSchemaDto profileSchema = createProfileSchema(application.getId(), ctlSchema.getId());
             profileSchemas.add(profileSchema);
         }
         
         Collections.sort(profileSchemas, new IdComparator());
         
-        List<ProfileSchemaDto> storedProfileSchemas = client.getProfileSchemas(application.getId());
+        List<EndpointProfileSchemaDto> storedProfileSchemas = client.getProfileSchemas(application.getId());
 
         Collections.sort(storedProfileSchemas, new IdComparator());
         
         Assert.assertEquals(profileSchemas.size(), storedProfileSchemas.size());
         for (int i=0;i<profileSchemas.size();i++) {
-            ProfileSchemaDto profileSchema = profileSchemas.get(i);
-            ProfileSchemaDto storedProfileSchema = storedProfileSchemas.get(i);
+            EndpointProfileSchemaDto profileSchema = profileSchemas.get(i);
+            EndpointProfileSchemaDto storedProfileSchema = storedProfileSchemas.get(i);
             assertProfileSchemasEquals(profileSchema, storedProfileSchema);
         }
     }
\ No newline at end of file
