diff --git a/old_test.java b/new_test.java
index be77bfa..3bd7ae9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,16 +1,9 @@
 @Test
   public void cancelUfsFileTest() throws Exception {
-    UnderFileSystem mockUfs = Mockito.mock(UnderFileSystem.class);
-    OutputStream mockOutputStream = Mockito.mock(OutputStream.class);
-    Mockito.when(mockUfs.create(Mockito.anyString())).thenReturn(mockOutputStream);
-    PowerMockito.mockStatic(UnderFileSystem.class);
-    BDDMockito.given(UnderFileSystem.get(Mockito.anyString(), Mockito.any(Configuration.class)))
-        .willReturn(mockUfs);
-
-    String testFile = "test";
+    String uniqPath = PathUtils.uniqPath();
     UnderFileSystemManager manager = new UnderFileSystemManager();
-    long id = manager.createFile(testFile);
-    Mockito.verify(mockUfs).create(Mockito.contains(testFile));
+    long id = manager.createFile(uniqPath);
+    Mockito.verify(mMockUfs).create(Mockito.contains(uniqPath));
     manager.cancelFile(id);
-    Mockito.verify(mockUfs).delete(Mockito.contains(testFile), Mockito.eq(false));
+    Mockito.verify(mMockUfs).delete(Mockito.contains(uniqPath), Mockito.eq(false));
   }
\ No newline at end of file
