diff --git a/old_product.java b/new_product.java
index e482523..8c1af0b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-public void cancelUfsFile(long tempUfsFileId, UFSCancelFileTOptions options) throws alluxio.thrift.AlluxioTException, alluxio.thrift.ThriftIOException, org.apache.thrift.TException;
\ No newline at end of file
+public void cancelUfsFile(long tempUfsFileId, CancelUfsFileTOptions options) throws alluxio.thrift.AlluxioTException, alluxio.thrift.ThriftIOException, org.apache.thrift.TException;
\ No newline at end of file
