diff --git a/old_product.java b/new_product.java
index ce70362..367db30 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
 private FElem parse(final QueryContext ctx) throws QueryException {
+    final byte[] input = checkStr(expr[0], ctx);
     final Item opt = expr.length > 1 ? expr[1].item(ctx, info) : null;
     final TokenMap map = new FuncParams(Q_OPTIONS, info).parse(opt);
 
@@ -9,10 +10,8 @@ private FElem parse(final QueryContext ctx) throws QueryException {
       if(s.length != 1) BXCS_SEP.thrw(info);
       sep = s[0];
     }
-    final CsvParser parser = new CsvParser(sep, header);
-
     try {
-      return parser.convert(checkStr(expr[0], ctx));
+      return new CsvParser(sep, header).convert(input);
     } catch(final IOException ex) {
       throw BXCS_ERROR.thrw(info, ex);
     }
