diff --git a/old_product.java b/new_product.java
index 3991538..47c6cba 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,7 +9,9 @@
             @NotBlank(message = "No id entered unable to kill job.")
             final String id
     ) throws GenieException {
-        LOG.debug("called for id: " + id);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("called for id: " + id);
+        }
         final Job job = this.jobRepo.findOne(id);
 
         // do some basic error handling
@@ -38,12 +40,16 @@
         final String localURI = getEndPoint() + "/" + this.jobResourcePrefix + "/" + id;
 
         if (!killURI.equals(localURI)) {
-            LOG.debug("forwarding kill request to: " + killURI);
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("forwarding kill request to: " + killURI);
+            }
             return forwardJobKill(killURI);
         }
 
         // if we get here, killURI == localURI, and job should be killed here
-        LOG.debug("killing job on same instance: " + id);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("killing job on same instance: " + id);
+        }
         this.jobManagerFactory.getJobManager(job).kill();
 
         job.setJobStatus(JobStatus.KILLED, "Job killed on user request");
@@ -52,7 +58,9 @@
         // increment counter for killed jobs
         this.stats.incrGenieKilledJobs();
 
-        LOG.debug("updating job status to KILLED for: " + id);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("updating job status to KILLED for: " + id);
+        }
         if (!job.isDisableLogArchival()) {
             job.setArchiveLocation(this.netUtil.getArchiveURI(id));
         }
