diff --git a/old_product.java b/new_product.java
index b87dc8e..f969226 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,13 +3,8 @@ CompletableFuture<Void> initialize(Duration timeout) {
         Preconditions.checkState(this.state.get() == AggregatorState.NotInitialized, "SegmentAggregator has already been initialized.");
         long traceId = LoggerHelpers.traceEnter(log, this.traceObjectId, "initialize");
 
-        TimeoutTimer timer = new TimeoutTimer(timeout);
-        return this.storage
-                .open(this.metadata.getName(), timer.getRemaining())
-                .thenCompose(handle -> {
-                    this.segmentHandle.set(handle);
-                    return this.storage.getStreamSegmentInfo(handle, timer.getRemaining());
-                })
+        return this.storage.acquireLockForSegment(this.metadata.getName()).
+                thenCompose( bool -> this.storage.getStreamSegmentInfo(this.metadata.getName(), timeout))
                 .thenAccept(segmentInfo -> {
                     // Check & Update StorageLength in metadata.
                     if (this.metadata.getStorageLength() != segmentInfo.getLength()) {
