diff --git a/old_test.java b/new_test.java
index 8191bdd..b00cf4d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,18 +3,18 @@
     {
         list.add(newArray(list.buffer, 0, 1, 0, 1, 0));
 
-        assertEquals(0, list.removeAll(/* intrinsic:ktypecast */ 2));
-        assertEquals(3, list.removeAll(/* intrinsic:ktypecast */ 0));
+        assertEquals(0, list.removeAllOccurrences(/* intrinsic:ktypecast */ 2));
+        assertEquals(3, list.removeAllOccurrences(/* intrinsic:ktypecast */ 0));
         assertListEquals(list.toArray(), 1, 1);
 
-        assertEquals(2, list.removeAll(/* intrinsic:ktypecast */ 1));
+        assertEquals(2, list.removeAllOccurrences(/* intrinsic:ktypecast */ 1));
         assertTrue(list.isEmpty());
 
         /* removeIf:primitive */
         list.clear();
         list.add(newArray(list.buffer, 0, null, 2, null, 0));
-        assertEquals(2, list.removeAll((Object) null));
-        assertEquals(0, list.removeAll((Object) null));
+        assertEquals(2, list.removeAllOccurrences((Object) null));
+        assertEquals(0, list.removeAllOccurrences((Object) null));
         assertListEquals(list.toArray(), 0, 2, 0);
         /* end:removeIf */
     }
\ No newline at end of file
