diff --git a/old_test.java b/new_test.java
index b4a6415..4495882 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,6 @@
                .media(Reference.builder().href(mediaURI).type(MEDIA).build()).build();
 
       // The method under test
-      Task insertMedia = vmApi.insertMedia(vm.getHref(), params);
+      Task insertMedia = vmApi.insertMedia(vmUrn, params);
       assertTrue(retryTaskSuccess.apply(insertMedia), String.format(TASK_COMPLETE_TIMELY, "insertMedia"));
    }
\ No newline at end of file
