diff --git a/old_test.java b/new_test.java
index e3e4d4b..d6131e2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -38,4 +38,7 @@
 
         // should always equal to Object
         Assert.assertEquals(by(Object.class, "Object"), by(null, "Something"));
+
+        Assert.assertEquals(withNext(byName("Foo.I"), "Foo.B"), byName("Foo.I"));
+        Assert.assertEquals(byName("Foo.I"), withNext(byName("Foo.I"), "Foo.B"));
     }
\ No newline at end of file
