diff --git a/old_product.java b/new_product.java
index 7fb02b7..9e1f1c7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-void suspend() {
-        throw new UnsupportedOperationException(
-                "Deprecated Thread methods are not supported.");
+@SuppressWarnings("ThrowableInstanceNeverThrown")
+    void suspend() {
+        Logger.global.log(Level.SEVERE, UNSUPPORTED_THREAD_METHOD,
+                new UnsupportedOperationException());
     }
\ No newline at end of file
