diff --git a/old_product.java b/new_product.java
index 96c1298..24a8bb8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,9 +2,7 @@
     public void destroy() {
         super.destroy();
         try {
-            if (cleanFuture != null) {
-                cleanFuture.cancel(true);
-            }
+            ExecutorUtil.cancelScheduledFuture(cleanFuture);
         } catch (Throwable t) {
             logger.warn(t.getMessage(), t);
         }
