diff --git a/old_test.java b/new_test.java
index 34895f4..7ba7c7e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,11 +12,9 @@
     newReplica = collection.getReplicas().get(0);
     cc = getCoreContainer(newReplica.getNodeName());
     
-    SharedShardMetadataController sharedMetadataController = cc.getSharedStoreManager().getSharedShardMetadataController();
-    sharedMetadataController.ensureMetadataNodeExists(collectionName, shardName);
-    sharedMetadataController.updateMetadataValueWithVersion(collectionName, shardName, UUID.randomUUID().toString(), -1);
     try {
-      BlobStoreUtils.syncLocalCoreWithSharedStore(collectionName, newReplica.getCoreName(), shardName, cc);
+      SharedShardVersionMetadata shardVersionMetadata = new SharedShardVersionMetadata(0, UUID.randomUUID().toString());
+      BlobStoreUtils.syncLocalCoreWithSharedStore(collectionName, newReplica.getCoreName(), shardName, cc, shardVersionMetadata, true);
       fail("syncLocalCoreWithSharedStore should throw exception if shared store doesn't have the core.metadata file.");
     } catch (Exception ex){
       String expectedException = "cannot get core.metadata file from shared store";
