diff --git a/old_product.java b/new_product.java
index a1bf81d..3a42064 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
-@RequiresSession
-  public AlertDefinitionEntity findByName(long clusterId, String definitionName) {
+public AlertDefinitionEntity findByName(long clusterId, String definitionName) {
     TypedQuery<AlertDefinitionEntity> query = entityManagerProvider.get().createNamedQuery(
         "AlertDefinitionEntity.findByName", AlertDefinitionEntity.class);
 
