diff --git a/old_test.java b/new_test.java
index 040416a..d2502fa 100644
--- a/old_test.java
+++ b/new_test.java
@@ -22,10 +22,10 @@
     assertSame(result, entity);
     verify(mockEntityManagerProvider, entityManager);
 
-    List<AlertDefinitionEntity> definitions = realDAO.findAll();
+    List<AlertDefinitionEntity> definitions = dao.findAll();
     Assert.assertNotNull(definitions);
     AlertDefinitionEntity definition = definitions.get(2);
-    AlertDefinitionEntity retrieved = realDAO.findByName(
+    AlertDefinitionEntity retrieved = dao.findByName(
         definition.getClusterId(), definition.getDefinitionName());
 
     Assert.assertEquals(definition, retrieved);
