diff --git a/old_product.java b/new_product.java
index e2e7f8a..d8ab846 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public boolean instanceOf(final SeqType st) {
-    final Type t1 = type, t2 = st.type;
-    final Test k1 = kind, k2 = st.kind;
-    return (t2 == AtomType.ITEM || t1.instanceOf(t2)) && occ.instanceOf(st.occ) &&
+    final Type type1 = type, type2 = st.type;
+    final Test kind1 = kind, kind2 = st.kind;
+    return (type2 == AtomType.ITEM || type1.instanceOf(type2)) && occ.instanceOf(st.occ) &&
       // [LW] complete kind check
-      (k2 == null || k1 != null && k1.intersect(k2) != null);
+      (kind2 == null || kind1 != null && kind1.intersect(kind2) != null);
   }
\ No newline at end of file
