diff --git a/old_test.java b/new_test.java
index b77bbee..5d967e6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,8 +9,8 @@
     assertFalse(f.instanceOf(ITR_O));
     assertTrue(f.instanceOf(ITEM_O));
     assertTrue(f.instanceOf(f));
-    assertTrue(f.instanceOf(FUN_ZO));
-    assertFalse(FUN_O.instanceOf(f));
+    assertTrue(f.instanceOf(FUNC_ZO));
+    assertFalse(FUNC_O.instanceOf(f));
     assertFalse(f.instanceOf(FuncType.get(DEC_ZO, BLN_O, ITR_O).seqType()));
     assertFalse(f.instanceOf(FuncType.get(DEC_ZO, AAT_O).seqType()));
     assertFalse(f.instanceOf(FuncType.get(BLN_O, BLN_O).seqType()));
@@ -19,7 +19,7 @@
     final MapType m = MapType.get(AtomType.STR, ITR_O);
     assertTrue(m.instanceOf(m));
     assertTrue(m.instanceOf(AtomType.ITEM));
-    assertTrue(m.instanceOf(ANY_FUN));
+    assertTrue(m.instanceOf(ANY_FUNC));
     assertTrue(m.instanceOf(ANY_MAP));
     assertTrue(m.instanceOf(MapType.get(AtomType.ITEM, ITR_O)));
     assertTrue(m.instanceOf(MapType.get(AtomType.STR, ITR_O)));
@@ -31,7 +31,7 @@
     final ArrayType a = ArrayType.get(ITR_O);
     assertTrue(a.instanceOf(a));
     assertTrue(a.instanceOf(AtomType.ITEM));
-    assertTrue(a.instanceOf(ANY_FUN));
+    assertTrue(a.instanceOf(ANY_FUNC));
     assertTrue(a.instanceOf(ANY_ARRAY));
     assertTrue(a.instanceOf(ArrayType.get(ITR_O)));
     assertTrue(a.instanceOf(ArrayType.get(ITR_O)));
