diff --git a/old_product.java b/new_product.java
index 8ddd837..c1c0393 100644
--- a/old_product.java
+++ b/new_product.java
@@ -29,7 +29,7 @@ private ValueBuilder eval(final QueryContext ctx, final byte[] qu, final String
             // limit reached: perform garbage collection and check again
             if(Performance.memory() > limit) {
               Performance.gc(1);
-              if(Performance.memory() > limit) qc.stop();
+              if(Performance.memory() > limit) ctx.stop();
             }
           }
         }, 500, 500);
@@ -38,9 +38,7 @@ private ValueBuilder eval(final QueryContext ctx, final byte[] qu, final String
       if(ms != 0) {
         to.schedule(new TimerTask() {
           @Override
-          public void run() {
-            qc.stop();
-          }
+          public void run() { ctx.stop(); }
         }, ms);
       }
     }
