diff --git a/old_product.java b/new_product.java
index 0293421..6ef91f7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 private Iter children(final QueryContext qc) throws QueryException, IOException {
     final TokenList children = new TokenList();
-    try(DirectoryStream<Path> paths = Files.newDirectoryStream(checkPath(0, qc))) {
+    try(DirectoryStream<Path> paths = Files.newDirectoryStream(toPath(0, qc))) {
       for(final Path child : paths) children.add(get(child, Files.isDirectory(child)).string());
     }
     return StrSeq.get(children).iter();
