diff --git a/old_product.java b/new_product.java
index c95a02f..adff832 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,10 @@
 public boolean pin(@NonNull VoidAggregation aggregation) {
-        VoidAggregation existing = clipboard.get(aggregation.getTaskId());
+        RequestDescriptor descriptor = RequestDescriptor.createDescriptor(aggregation.getOriginatorId(), aggregation.getTaskId());
+        VoidAggregation existing = clipboard.get(descriptor);
         if (existing == null) {
             existing = aggregation;
             trackingCounter.incrementAndGet();
-            clipboard.put(aggregation.getTaskId(), aggregation);
+            clipboard.put(descriptor, aggregation);
         }
 
         existing.accumulateAggregation(aggregation);
@@ -15,9 +16,6 @@ public boolean pin(@NonNull VoidAggregation aggregation) {
         if (missing == 0) {
             completedQueue.add(existing);
             completedCounter.incrementAndGet();
-
-            // TODO: delete it from tracking table probably?
-
             return true;
         } else return false;
     }
\ No newline at end of file
