diff --git a/old_test.java b/new_test.java
index 9b66b8b..c6a8df6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,6 +8,6 @@
         InitializationAggregation aggregation = new InitializationAggregation(1, 0);
         clipboard.pin(aggregation);
 
-        assertTrue(clipboard.isTracking(aggregation.getTaskId()));
-        assertTrue(clipboard.isReady(aggregation.getTaskId()));
+        assertTrue(clipboard.isTracking(0L, aggregation.getTaskId()));
+        assertTrue(clipboard.isReady(0L, aggregation.getTaskId()));
     }
\ No newline at end of file
