diff --git a/old_test.java b/new_test.java
index 9d95f9b..3cf56a1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -32,8 +32,8 @@
 		assertEquals(5, config.getAcquireIncrement());
 		assertEquals(5, config.getPartitionCount());
 		assertEquals(3, config.getReleaseHelperThreads());
-		assertEquals(60*60*1000, config.getIdleConnectionTestPeriod());
-		assertEquals(240*60*1000, config.getIdleMaxAge()); 
+		assertEquals(60, config.getIdleConnectionTestPeriod());
+		assertEquals(240, config.getIdleMaxAge()); 
 		assertEquals(URL, config.getJdbcUrl());
 		assertEquals(USERNAME, config.getUsername());
 		assertEquals(PASSWORD, config.getPassword());
