diff --git a/old_product.java b/new_product.java
index adf5d03..8e6f7f4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ public void persistFile(long fileId, List<Long> blockIds) throws IOException {
     synchronized (mLock) {
       blockIdToLockId = mPersistingInProgressFiles.get(fileId);
       if (blockIdToLockId == null || !blockIdToLockId.keySet().equals(new HashSet<>(blockIds))) {
-        throw new IOException("Not all the blocks of file " + fileId + " are blocked");
+        throw new IOException("Not all the blocks of file " + fileId + " are locked");
       }
     }
 
