diff --git a/old_product.java b/new_product.java
index da6fe86..0b564cf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
-private long learn(PrintStream out, List<AssociationRule> list, int[] itemset, int support, double confidence) {
-        long n = 0;
+private void generate(int[] itemset, int support) {
         // Determine combinations
         int[][] combinations = getPowerSet(itemset);
 
@@ -11,20 +10,10 @@ private long learn(PrintStream out, List<AssociationRule> list, int[] itemset, i
             if (complement != null) {
                 double arc = getConfidence(combinations[i], support);
                 if (arc >= confidence) {
-                    double supp = (double) support / fim.size();
+                    double supp = (double) support / size;
                     AssociationRule ar = new AssociationRule(combinations[i], complement, supp, arc);
-                    n++;
-
-                    if (out != null) {
-                        out.println(ar);
-                    }
-
-                    if (list != null) {
-                        list.add(ar);
-                    }
+                    buffer.offer(ar);
                 }
             }
         }
-        
-        return n;
     }
\ No newline at end of file
