diff --git a/old_test.java b/new_test.java
index 60fda8f..a1308d9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,8 @@
 @Test
-    public void testLearn() {
-        System.out.println("learn");
-        ARM instance = new ARM(itemsets, 3);
-        instance.learn(0.5, System.out);
-        List<AssociationRule> rules = instance.learn(0.5);
+    public void testARM() {
+        System.out.println("ARM");
+        FPTree tree = FPTree.build(3, itemsets);
+        List<AssociationRule> rules = ARM.apply(0.5, tree).collect(Collectors.toList());
         assertEquals(9, rules.size());
         
         assertEquals(0.6, rules.get(0).support, 1E-2);
