diff --git a/old_test.java b/new_test.java
index bb7b00e..aa5d3d8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-  public void reportOrphanedSpans_afterGC() throws Exception {
+  public void reportOrphanedSpans_afterGC() {
     TraceContext context1 = context.toBuilder().traceId(1).spanId(1).build();
     PendingSpan span = pendingSpans.getOrCreate(context1, false);
     span.state.name("foo");
