diff --git a/old_test.java b/new_test.java
index 9601e6d..28940db 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,10 +8,10 @@
     query(_FT_SEARCH.args(NAME, "XXX"), "");
 
     // apply index options to query term
-    new Set(Options.STEMMING, true).execute(context);
+    new Set(MainOptions.STEMMING, true).execute(context);
     new CreateIndex("fulltext").execute(context);
     contains(_FT_SEARCH.args(NAME, "Exercises") + "/..", "<li>Exercise 1</li>");
-    new Set(Options.STEMMING, false).execute(context);
+    new Set(MainOptions.STEMMING, false).execute(context);
     new CreateIndex(CmdIndex.FULLTEXT).execute(context);
 
     // check match options
