diff --git a/old_product.java b/new_product.java
index 4623a58..f6823d4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static char boolArrayToHexDigit(boolean[] src, int srcPos) {
+public static char binaryToHexDigit(boolean[] src, int srcPos) {
         if (src.length == 0) {
             throw new IllegalArgumentException("Cannot convert an empty array.");
         }
