diff --git a/old_product.java b/new_product.java
index eb08d80..809df17 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
-public static double UniformCdf(int k, int n) {
+public static double uniformCdf(int k, int n) {
         if(k<0 || n<1) {
             throw new IllegalArgumentException("All the parameters must be positive and n larger than 1.");
         }
         k = Math.min(k, n);
         
-        double probabilitySum = k*Uniform(n);
+        double probabilitySum = k*uniform(n);
         
         return probabilitySum;
     }
\ No newline at end of file
