diff --git a/old_product.java b/new_product.java
index 456718e..00d131c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,3 @@
-public LearningEnvironmentBuilder withLoggingFactory(MLLogger.Factory loggingFactory) {
-        this.loggingFactory = loggingFactory;
-        return this;
+public default <T extends MLLogger.Factory & Serializable> LearningEnvironmentBuilder withLoggingFactory(T loggingFactory) {
+        return withLoggingFactoryDependency(constant(loggingFactory));
     }
\ No newline at end of file
