diff --git a/old_test.java b/new_test.java
index 4f6b763..8f3c9a3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,7 @@
 @Test
   public void testEncode() throws IOException
   {
+    assertRoundTrip("arrays.AnonArray");
     assertRoundTrip("arrays.WithPrimitivesArray");
     assertRoundTrip("denormalized.WithNamespacedDeclarations");
     assertRoundTrip("denormalized.WithIncludeDeclaration");
