diff --git a/old_product.java b/new_product.java
index 56f8411..f5348f4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -25,22 +25,22 @@
         if (hasPackage)
         {
           writeLine("package " + escapeIdentifier(namedSchema.getPackage()));
+          _package = namedSchema.getPackage();
         }
         newline();
       }
-
-      if (_importsByLocalName.size() > 0)
+    }
+    if (_importsByLocalName.size() > 0)
+    {
+      // Sort imports by fully qualified name
+      for (Name importName : new TreeSet<>(_importsByLocalName.values()))
       {
-        // Sort imports by fully qualified name
-        for (Name importName : new TreeSet<>(_importsByLocalName.values()))
+        if (!importName.getNamespace().equals(_namespace))
         {
-          if (!importName.getNamespace().equals(_namespace))
-          {
-            writeLine("import " + escapeIdentifier(importName.getFullName()));
-          }
+          writeLine("import " + escapeIdentifier(importName.getFullName()));
         }
-        newline();
       }
+      newline();
     }
     writeInlineSchema(schema);
   }
\ No newline at end of file
