diff --git a/old_product.java b/new_product.java
index 2b8dd7d..c4242dc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
 private void degreesAndPersist(String resourceType) {
-        MindmapsTransaction transaction= graph.getTransaction();
         insertOntology(resourceType, Data.LONG);
         ComputerResult result = computer.compute(new DegreeAndPersistVertexProgram(keySpace,allTypes));
 
@@ -7,13 +6,13 @@ private void degreesAndPersist(String resourceType) {
         // collect relation ids and delete them in a single thread
         result.graph().traversal().V().forEachRemaining(v -> {
             if (v.keys().contains(DegreeAndPersistVertexProgram.OLD_ASSERTION_ID)) {
-                Relation relation = transaction.getRelation(v.value(DegreeAndPersistVertexProgram.OLD_ASSERTION_ID));
+                Relation relation = graph.getRelation(v.value(DegreeAndPersistVertexProgram.OLD_ASSERTION_ID));
                 relation.delete();
             }
         });
 
         try {
-            transaction.commit();
+            graph.commit();
         } catch (MindmapsValidationException e) {
             e.printStackTrace();
         }
