diff --git a/old_product.java b/new_product.java
index be8ddc3..2123b1c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -57,7 +57,7 @@
           // tablets... so clear it
           tl.invalidateCache();
 
-          ClientContext context = new ClientContext(getConnectionInfo(job));
+          ClientContext context = new ClientContext(getClientInfo(job));
           while (!tl.binRanges(context, ranges, binnedRanges).isEmpty()) {
             String tableIdStr = tableId.canonicalID();
             if (!Tables.exists(instance, tableId))
