diff --git a/old_test.java b/new_test.java
index 0446d12..fb4f624 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
 	public void testSetInput() {
-		Operator input1 = new OpImpl(0);
-		Operator input2 = new OpImpl(1);
-		Operator fixture = new OpImpl(0, input1, input2);
+		final Operator input1 = new OpImpl(0);
+		final Operator input2 = new OpImpl(1);
+		final Operator fixture = new OpImpl(0, input1, input2);
 
-		Operator newInput2 = new OpImpl(2);
+		final Operator newInput2 = new OpImpl(2);
 
 		fixture.setInput(1, newInput2);
 
