diff --git a/old_product.java b/new_product.java
index d8b0b22..4af6313 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-public ClassifierValidator.ValidationMetrics validate(Dataframe testingData) {
+public ClassificationMetrics validate(Dataframe testingData) {
         logger.info("validate()");
 
         predict(testingData);
 
-        return new ClassifierValidator().validate(testingData);
+        return new ClassificationMetrics(testingData);
     }
\ No newline at end of file
