diff --git a/old_test.java b/new_test.java
index f460d31..3ac4d12 100644
--- a/old_test.java
+++ b/new_test.java
@@ -28,7 +28,7 @@
         //instance = null;
         instance = new GaussianDPMM(dbName, conf);
 
-        ClustererValidator.ValidationMetrics vm = instance.validate(validationData);
+        ClusteringMetrics vm = instance.validate(validationData);
 
         double expResult = 1.0;
         double result = vm.getPurity();
