diff --git a/old_test.java b/new_test.java
index e2c34ed..d51f4a9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test(expected = APIException.class)
 	@Verifies(value = "should not merge patient with itself", method = "mergePatients(Patient,Patient)")
-	public void mergePatients_shouldNotMergePatientWithItself() {
+	public void mergePatients_shouldNotMergePatientWithItself() throws Exception {
 		Context.getPatientService().mergePatients(new Patient(2), new Patient(2));
 	}
\ No newline at end of file
