diff --git a/old_product.java b/new_product.java
index 644ca8d..002a7c7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
 @Authorized( { PrivilegeConstants.EDIT_PATIENTS })
-	public void mergePatients(Patient preferred, Patient notPreferred) throws APIException;
\ No newline at end of file
+	public void mergePatients(Patient preferred, Patient notPreferred) throws APIException, SerializationException;
\ No newline at end of file
