diff --git a/old_test.java b/new_test.java
index fad244e..592b79b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test
     public void startNewSpan() {
-        state.setCurrentServerSpan(ServerSpan.create(PARENT_SPAN_ID, "name"));
+        state.setCurrentServerSpan(ServerSpan.create(PARENT_CONTEXT, "name"));
 
         PowerMockito.when(System.nanoTime()).thenReturn(500L);
 
-        SpanId expectedSpanId = PARENT_SPAN_ID.toBuilder().spanId(555L).parentId(PARENT_SPAN_ID.spanId).build();
+        SpanId expectedContext = PARENT_CONTEXT.toBuilder().spanId(555L).parentId(PARENT_CONTEXT.spanId).build();
 
-        assertEquals(expectedSpanId, localTracer.startNewSpan(COMPONENT_NAME, OPERATION_NAME));
+        assertEquals(expectedContext, localTracer.startNewSpan(COMPONENT_NAME, OPERATION_NAME));
 
         Span started = state.getCurrentLocalSpan();
 
