diff --git a/old_product.java b/new_product.java
index 2db1eb8..cb54631 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,17 +6,16 @@ public SpanId startNewSpan(String requestName) {
             return null;
         }
 
-        SpanId newSpanId = getNewSpanId();
+        SpanId nextContext = nextContext(maybeParent());
         if (sample == null) {
             // No sample indication is present.
-            if (!traceSampler().isSampled(newSpanId.traceId)) {
+            if (!traceSampler().isSampled(nextContext.traceId)) {
                 spanAndEndpoint().state().setCurrentClientSpan(null);
                 return null;
             }
         }
 
-        Span newSpan = Span.fromSpanId(newSpanId);
-        newSpan.setName(requestName);
+        Span newSpan = Span.create(nextContext).setName(requestName);
         spanAndEndpoint().state().setCurrentClientSpan(newSpan);
-        return newSpanId;
+        return nextContext;
     }
\ No newline at end of file
