diff --git a/old_test.java b/new_test.java
index 5fec04c..cce618d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,12 +5,12 @@
 		new UserProperties(user.getUserProperties()).setSupposedToChangePassword(true);
 		
 		UserService us = Context.getUserService();
-		us.saveUser(user, "Openmr5xy");
+		us.saveUser(user);
 		
 		ChangePasswordFormController controller = new ChangePasswordFormController();
 		BindException errors = new BindException(controller.formBackingObject(), "user");
 		
-		controller.handleSubmission(new MockHttpSession(), "Passw0rd", "Passw0rd", "", "", "", Context
+		controller.handleSubmission(new MockHttpSession(), null, "Passw0rd", "Passw0rd", "", "", "", Context
 		        .getAuthenticatedUser(), errors);
 		
 		User modifiedUser = us.getUser(user.getId());
