diff --git a/old_product.java b/new_product.java
index f2c22da..f777569 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
 @RequestMapping(method = RequestMethod.POST)
 	public String handleSubmission(HttpSession httpSession,
+			@RequestParam(required = true, value = "oldPassword") String oldPassword,
 	        @RequestParam(required = true, value = "password") String password,
 	        @RequestParam(required = true, value = "confirmPassword") String confirmPassword,
 	        @RequestParam(required = false, value = "question") String question,
@@ -16,7 +17,7 @@
 			return showForm(httpSession);
 		}
 		
-		changeUserPasswordAndQuestion(user, newPassword, newQuestionAnswer);
+		changeUserPasswordAndQuestion(user, oldPassword, newPassword, newQuestionAnswer);
 		httpSession.removeAttribute(WebConstants.OPENMRS_MSG_ATTR);
 		return "redirect:/index.htm";
 		
