diff --git a/old_product.java b/new_product.java
index 36a9f72..de68156 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
-public static long createRandom(Number minValue,Number maxValue){
-        Validate.notNull(minValue, "min can't be null!");
-        Validate.notNull(maxValue, "max can't be null!");
+public static long createRandom(Number minInclusiveValue,Number maxExclusiveValue){
+        Validate.notNull(minInclusiveValue, "minInclusiveValue can't be null!");
+        Validate.notNull(maxExclusiveValue, "maxExclusiveValue can't be null!");
 
-        long minLong = minValue.longValue();
-        long maxLong = maxValue.longValue();
+        long minLong = minInclusiveValue.longValue();
+        long maxLong = maxExclusiveValue.longValue();
 
-        Validate.isTrue(maxLong >= minLong, Slf4jUtil.format("maxLong:[{}] can not < minLong:[{}]", maxLong, minLong));
+        Validate.isTrue(maxLong >= minLong, Slf4jUtil.format("minInclusiveValue:[{}] can not < maxExclusiveValue:[{}]", maxLong, minLong));
         return RandomUtils.nextLong(minLong, maxLong);
     }
\ No newline at end of file
