diff --git a/old_test.java b/new_test.java
index bd86617..5a93d34 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test
+    @Repeat(20000)
     public void testCreateRandom2(){
-        LOGGER.debug("" + RandomUtil.createRandom(10, 20));
-        LOGGER.debug("" + RandomUtil.createRandom(0, 800));
-
-        assertEquals(800L, RandomUtil.createRandom(800, 800));
+        assertThat(RandomUtil.createRandom(10, 20), allOf(greaterThanOrEqualTo(10L), lessThan(20L)));
+        assertThat(RandomUtil.createRandom(0, 800), allOf(greaterThanOrEqualTo(0L), lessThan(800L)));
     }
\ No newline at end of file
