diff --git a/old_test.java b/new_test.java
index faa508a..98fd8d9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
 	public void testToMessage() throws Exception {
 		SocketMessageMapper mapper = new SocketMessageMapper();
-		Message<byte[]> message = mapper.toMessage(new StubSocketReader());
-		assertEquals(TEST_PAYLOAD, new String(message.getPayload()));
+		Message<Object> message = mapper.toMessage(new StubSocketReader());
+		assertEquals(TEST_PAYLOAD, new String((byte[]) message.getPayload()));
 		assertEquals(InetAddress.getLocalHost().getHostName(), message
 				.getHeaders().get(IpHeaders.HOSTNAME));
 		assertEquals(InetAddress.getLocalHost().getHostAddress(), message
