diff --git a/old_test.java b/new_test.java
index 7a3ce7d..7f0a61e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,14 @@
 @Test
-	@Verifies(value = "should return null if no name is explicitly marked as locale preferred", method = "getPreferredName(Locale)")
-	public void getPreferredName_shouldReturnNullIfNoNameIsExplicitlyMarkedAsLocalePreferred() throws Exception {
+	@Verifies(value = "should return the fully specified name if no name is explicitly marked as locale preferred", method = "getPreferredName(Locale)")
+	public void getPreferredName_shouldReturnTheFullySpecifiedNameIfNoNameIsExplicitlyMarkedAsLocalePreferred()
+	                                                                                                           throws Exception {
 		Concept testConcept = createMockConcept(1, Locale.US);
 		//preferred name in en_US
 		ConceptName preferredNameEN_US = ConceptNameTest.createMockConceptName(3, Locale.US, null, false);
 		testConcept.addName(preferredNameEN_US);
+		String fullySpecName = testConcept.getFullySpecifiedName(Locale.US).getName();
 		//preferred name in en
 		ConceptName preferredNameEN = ConceptNameTest.createMockConceptName(4, new Locale("en"), null, false);
 		testConcept.addName(preferredNameEN);
-		Assert.assertNull(testConcept.getPreferredName(Locale.US));
-		
+		Assert.assertEquals(fullySpecName, testConcept.getPreferredName(Locale.US).getName());
 	}
\ No newline at end of file
