diff --git a/old_product.java b/new_product.java
index 7f29933..586439b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,5 @@ public static <K, V> Map<K, V> sortByKeyAsc(Map<K, V> map){
         if (null == map){
             return emptyMap();
         }
-        return new TreeMap<K, V>(map);
+        return sort(map, new PropertyComparator<Map.Entry<K, V>>("key"));
     }
\ No newline at end of file
