diff --git a/old_test.java b/new_test.java
index df887eb..5fddde1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,5 +2,5 @@
     public void test_adjustDate() {
         MonthDay test = MonthDay.of(6, 30);
         LocalDate date = LocalDate.of(2007, 1, 1);
-        assertEquals(test.adjustDate(date), LocalDate.of(2007, 6, 30));
+        assertEquals(test.adjustCalendrical(date), LocalDate.of(2007, 6, 30));
     }
\ No newline at end of file
