diff --git a/old_product.java b/new_product.java
index e6f8874..1e8909f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,4 @@
-public LocalDate adjustDate(LocalDate date) {
-        return date.withDayOfMonth(dayOfMonth);
+@Override
+    public DateTimeObject adjustCalendrical(DateTimeObject calendrical) {
+        return calendrical.with(DAY_OF_MONTH, dayOfMonth);
     }
\ No newline at end of file
