diff --git a/old_product.java b/new_product.java
index 5393cf8..6e31b39 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,7 +4,8 @@ protected boolean acceptAbsolutePath(String path)
 		int len = path.length();
 		final String ext;
 		if (ixExtension <= 0 || ixExtension == len ||
-			(path.lastIndexOf(File.separator) + 1) == ixExtension)
+			(path.lastIndexOf('/') + 1) == ixExtension ||
+			(path.lastIndexOf('\\') + 1) == ixExtension)
 		{
 			ext = null;
 		}
