diff --git a/old_test.java b/new_test.java
index c4bf3aa..57f6e03 100644
--- a/old_test.java
+++ b/new_test.java
@@ -30,12 +30,12 @@
 			// this should be ignored since the allocation id does not match
 			slotManager.freeSlot(slotId, new AllocationID());
 
-			assertTrue(slot.isAllocated());
+			assertTrue(slot.getState() == TaskManagerSlot.State.ALLOCATED);
 			assertEquals("The slot has not been allocated to the expected allocation id.", allocationId, slot.getAllocationId());
 
 			slotManager.freeSlot(slotId, allocationId);
 
-			assertTrue(slot.isFree());
+			assertTrue(slot.getState() == TaskManagerSlot.State.FREE);
 			assertNull(slot.getAllocationId());
 		}
 	}
\ No newline at end of file
