diff --git a/old_test.java b/new_test.java
index 41c0557..a2e4cc3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,8 @@
         DateTime from = parseDate("2015-01-01T00:00:00Z");
         DateTime to = parseDate("2015-01-10T00:00:00Z");
         UsageHistogramBean histogram = new UsageHistogramBean();
-        Map<String, UsageDataPoint> index = ESMetricsAccessor.generateHistogramSkeleton(histogram, from, to, UsageHistogramIntervalType.day);
+        Map<String, UsageDataPoint> index = ESMetricsAccessor.generateHistogramSkeleton(histogram, from, to,
+                HistogramIntervalType.day, UsageDataPoint.class);
         Assert.assertEquals(9, index.size());
         Assert.assertEquals(9, histogram.getData().size());
         Assert.assertEquals("2015-01-01T00:00:00.000Z", histogram.getData().get(0).getLabel());
@@ -14,7 +15,8 @@
         from = parseDate("2015-01-01T00:00:00Z");
         to = parseDate("2015-01-03T00:00:00Z");
         histogram = new UsageHistogramBean();
-        index = ESMetricsAccessor.generateHistogramSkeleton(histogram, from, to, UsageHistogramIntervalType.hour);
+        index = ESMetricsAccessor.generateHistogramSkeleton(histogram, from, to, HistogramIntervalType.hour,
+                UsageDataPoint.class);
         Assert.assertEquals(48, index.size());
         Assert.assertEquals(48, histogram.getData().size());
         Assert.assertEquals("2015-01-01T00:00:00.000Z", histogram.getData().get(0).getLabel());
@@ -26,7 +28,8 @@
         from = parseDate("2015-01-01");
         to = parseDate("2015-01-03");
         histogram = new UsageHistogramBean();
-        index = ESMetricsAccessor.generateHistogramSkeleton(histogram, from, to, UsageHistogramIntervalType.hour);
+        index = ESMetricsAccessor.generateHistogramSkeleton(histogram, from, to, HistogramIntervalType.hour,
+                UsageDataPoint.class);
         Assert.assertEquals(48, index.size());
         Assert.assertEquals(48, histogram.getData().size());
         Assert.assertEquals("2015-01-01T00:00:00.000Z", histogram.getData().get(0).getLabel());
@@ -38,7 +41,8 @@
         from = parseDate("2015-01-01T00:00:00Z");
         to = parseDate("2015-01-02T00:00:00Z");
         histogram = new UsageHistogramBean();
-        index = ESMetricsAccessor.generateHistogramSkeleton(histogram, from, to, UsageHistogramIntervalType.minute);
+        index = ESMetricsAccessor.generateHistogramSkeleton(histogram, from, to,
+                HistogramIntervalType.minute, UsageDataPoint.class);
         Assert.assertEquals(1440, index.size());
         Assert.assertEquals(1440, histogram.getData().size());
         Assert.assertEquals("2015-01-01T00:00:00.000Z", histogram.getData().get(0).getLabel());
@@ -49,7 +53,8 @@
         from = parseDate("2015-01-01");
         to = parseDate("2015-12-31");
         histogram = new UsageHistogramBean();
-        index = ESMetricsAccessor.generateHistogramSkeleton(histogram, from, to, UsageHistogramIntervalType.month);
+        index = ESMetricsAccessor.generateHistogramSkeleton(histogram, from, to, HistogramIntervalType.month,
+                UsageDataPoint.class);
         Assert.assertEquals(12, index.size());
         Assert.assertEquals(12, histogram.getData().size());
         Assert.assertEquals("2015-01-01T00:00:00.000Z", histogram.getData().get(0).getLabel());
@@ -60,7 +65,8 @@
         from = parseDate("2015-01-01");
         to = parseDate("2015-12-30");
         histogram = new UsageHistogramBean();
-        index = ESMetricsAccessor.generateHistogramSkeleton(histogram, from, to, UsageHistogramIntervalType.week);
+        index = ESMetricsAccessor.generateHistogramSkeleton(histogram, from, to, HistogramIntervalType.week,
+                UsageDataPoint.class);
         Assert.assertEquals(53, index.size());
         Assert.assertEquals(53, histogram.getData().size());
         Assert.assertEquals("2014-12-28T00:00:00.000Z", histogram.getData().get(0).getLabel());
