diff --git a/old_product.java b/new_product.java
index 22c2f1c..10730a1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public static Map<String, UsageDataPoint> generateHistogramSkeleton(UsageHistogramBean rval, DateTime from, DateTime to,
-            UsageHistogramIntervalType interval) {
-        Map<String, UsageDataPoint> index = new HashMap<>();
+public static <T extends HistogramDataPoint> Map<String, T> generateHistogramSkeleton(HistogramBean<T> rval, DateTime from, DateTime to,
+            HistogramIntervalType interval, Class<T> dataType) {
+        Map<String, T> index = new HashMap<>();
 
         Calendar fromCal = from.toGregorianCalendar();
         Calendar toCal = to.toGregorianCalendar();
@@ -41,8 +41,14 @@ public static Map<String, UsageDataPoint> generateHistogramSkeleton(UsageHistogr
 
         while (fromCal.before(toCal)) {
             String label = formatDateWithMillis(fromCal);
-            UsageDataPoint point = new UsageDataPoint(label, 0L);
-            rval.getData().add(point);
+            T point;
+            try {
+                point = dataType.newInstance();
+            } catch (InstantiationException | IllegalAccessException e) {
+                throw new RuntimeException(e);
+            }
+            point.setLabel(label);
+            rval.addDataPoint(point);
             index.put(label, point);
             switch (interval) {
                 case day:
