diff --git a/old_test.java b/new_test.java
index ada947a..727e715 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,7 +14,7 @@
 		
 		AdministrationService as = Context.getAdministrationService();
 		List<GlobalProperty> globalProperties = as.getAllGlobalProperties();
-		globalProperties.add(new GlobalProperty(OpenmrsConstants.DRUG_ROUTE_CONCEPT_UUID, drugRoutes.getUuid(), "test"));
+		globalProperties.add(new GlobalProperty(OpenmrsConstants.GP_DRUG_ROUTE_CONCEPT_UUID, drugRoutes.getUuid(), "test"));
 		as.saveGlobalProperties(globalProperties);
 		List<Concept> drugRoutesList = orderService.getDrugRoutes();
 		assertEquals(3, drugRoutesList.size());
