diff --git a/old_product.java b/new_product.java
index d48375c..4d1d2e7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,6 +5,6 @@ public boolean execute() throws SQLServerException {
         }
         checkClosed();
         executeStatement(new PrepStmtExecCmd(this, EXECUTE));
-        loggerExternal.exiting(getClassNameLogging(), "execute", Boolean.valueOf(null != resultSet));
+        loggerExternal.exiting(getClassNameLogging(), "execute", null != resultSet);
         return null != resultSet;
     }
\ No newline at end of file
