diff --git a/old_test.java b/new_test.java
index ad19869..3833405 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @TestFactory
     Stream<DynamicTest> generateBulkCopyOptionsTest() {
-        List<BulkCopyTestWrapper> testData = createTestData_testBulkCopyOption();
+        List<BulkCopyTestWrapper> testData = createTestDatatestBulkCopyOption();
         return testData.stream().map(new Function<BulkCopyTestWrapper, DynamicTest>() {
             @Override
             public DynamicTest apply(final BulkCopyTestWrapper datum) {
