diff --git a/old_product.java b/new_product.java
index dc2d291..2baf712 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,20 +1,16 @@
-public static void executeCustomGcode(final String str, BackendAPI backend) {
+public static void executeCustomGcode(final String str, BackendAPI backend) throws Exception {
         if (str == null) {
             return;
         }
         String command = MacroHelper.substituteValues(str, backend);
         command = command.replaceAll("(\\r\\n|\\n\\r|\\r|\\n)", "");
         final String[] parts = command.split(";");
-        EventQueue.invokeLater(new Runnable() {
-            @Override
-            public void run() {
-                try {
-                    for (String cmd : parts) {
-                        backend.sendGcodeCommand(cmd);
-                    }
-                } catch (Exception ex) {
-                    GUIHelpers.displayErrorDialog(ex.getMessage());
-                }
-            }
-        });
+
+        /* 
+         * specifically NOT catching exceptions on gCode commands, let them pass to the invoking method
+         * so the error handling is aligned to the UI that triggered it (i.e. GUI button press versus Pendant UI http request)
+         */
+		for (String cmd : parts) {
+			backend.sendGcodeCommand(cmd);
+		}
     }
\ No newline at end of file
