diff --git a/old_test.java b/new_test.java
index dc2468a..3c46fa4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
-	@Ignore
-	public void testExtract() {
-		service.setDataset(RelationshipExtractionDataset.ENGLISH_NEWS);
-		String response = service.extract("IBM Watson Developer Cloud");
-		Assert.assertNotNull(response);
-	}
\ No newline at end of file
+  @Ignore
+  public void testExtract() {
+    service.setDataset(RelationshipExtractionDataset.ENGLISH_NEWS);
+    final String response = service.extract("IBM Watson Developer Cloud");
+    Assert.assertNotNull(response);
+  }
\ No newline at end of file
