diff --git a/old_product.java b/new_product.java
index 0b6e400..8b7d72d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,18 +1,12 @@
 public String extract(final String text) {
-		if (dataset == null)
-			throw new IllegalArgumentException("dataset can not be null");
-		if (text == null)
-			throw new IllegalArgumentException("text can not be null");
+    if (dataset == null)
+      throw new IllegalArgumentException("dataset cannot be null");
+    if (text == null)
+      throw new IllegalArgumentException("text cannot be null");
 
-		HttpRequestBase request = Request.Post("/v1/sire/0")
-				.withForm("sid", dataset.getId(), "rt", "xml", "txt", text).build();
-		String relations = null;
-		try {
-			HttpResponse response = execute(request);
-			relations = ResponseUtil.getString(response);
-		} catch (IOException e) {
-			throw new RuntimeException(e);
-		}
-
-		return relations;
-	}
\ No newline at end of file
+    final Request request =
+        RequestBuilder.post("/v1/sire/0")
+            .withForm("sid", dataset.getId(), "rt", "xml", "txt", text).build();
+    final Response response = execute(request);
+    return ResponseUtil.getString(response);
+  }
\ No newline at end of file
