diff --git a/old_product.java b/new_product.java
index e5b39b1..8191fcb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-@RequestMapping(value = "eventClassFamiliesByAppToken/{applicationToken}", method = RequestMethod.GET)
+@RequestMapping(value = "eventClassFamilies/{applicationToken}", method = RequestMethod.GET)
     @ResponseBody
     public List<AefMapInfoDto> getEventClassFamiliesByApplicationToken(@PathVariable String applicationToken) throws KaaAdminServiceException {
         return kaaAdminService.getEventClassFamiliesByApplicationToken(applicationToken);
