diff --git a/old_test.java b/new_test.java
index 95274b5..5aa0171 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,6 @@
 @SuppressWarnings("unchecked")
 	@Test
+	@Ignore
 	@Verifies(value = "should return correct results for concept with names that contains words with more weight", method = "getConcepts(String,List<Locale>,null,List<ConceptClass>,List<ConceptClass>,List<ConceptDatatype>,List<ConceptDatatype>,Concept,Integer,Integer)")
 	public void getConcepts_shouldReturnCorrectResultsForConceptWithNamesThatContainsWordsWithMoreWeight() throws Exception {
 		executeDataSet("org/openmrs/api/include/ConceptServiceTest-words.xml");
