diff --git a/old_product.java b/new_product.java
index fff1631..5e11247 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,3 @@
-public void setUser(final String user) throws GenieException {
-        if (StringUtils.isBlank(user)) {
-            throw new GenieException(
-                    HttpURLConnection.HTTP_BAD_REQUEST,
-                    "No user Entered.");
-        }
+public void setUser(final String user) {
         this.user = user;
     }
\ No newline at end of file
