diff --git a/old_test.java b/new_test.java
index b16fc50..0ff872a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test
     public void testSetUser() {
-        final Application a = new Application();
-        Assert.assertNull(a.getUser());
-        a.setUser(USER);
-        Assert.assertEquals(USER, a.getUser());
+        Assert.assertNull(this.a.getUser());
+        this.a.setUser(USER);
+        Assert.assertEquals(USER, this.a.getUser());
     }
\ No newline at end of file
