diff --git a/old_test.java b/new_test.java
index 04dc5cc..a151838 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,31 +20,39 @@
         LOGGER.info("Running loadCache test");
 
         try (Ignite ignite = Ignition.start("org/apache/ignite/tests/persistence/pojo/ignite-config.xml")) {
-            IgniteCache<PersonId, Person> personCache3 = ignite.getOrCreateCache(new CacheConfiguration<PersonId, Person>("cache3"));
+            CacheConfiguration<PersonId, Person> ccfg = new CacheConfiguration<>("cache3");
+
+            IgniteCache<PersonId, Person> personCache3 = ignite.getOrCreateCache(ccfg);
+
             int size = personCache3.size(CachePeekMode.ALL);
 
             LOGGER.info("Initial cache size " + size);
 
             LOGGER.info("Loading cache data from Cassandra table");
 
-            personCache3.loadCache(null, new String[] {"select * from test1.pojo_test3 limit 3"});
+            String qry = "select * from test1.pojo_test3 limit 3";
+
+            personCache3.loadCache(null, qry);
+
+            size = personCache3.size(CachePeekMode.ALL);
+            Assert.assertEquals("Cache data was incorrectly loaded from Cassandra table by '" + qry + "'", 3, size);
+
+            personCache3.clear();
+
+            personCache3.loadCache(null, new SimpleStatement(qry));
 
             size = personCache3.size(CachePeekMode.ALL);
-            if (size != 3) {
-                throw new RuntimeException("Cache data was incorrectly loaded from Cassandra. " +
-                    "Expected number of records is 3, but loaded number of records is " + size);
-            }
+            Assert.assertEquals("Cache data was incorrectly loaded from Cassandra table by statement", 3, size);
 
             personCache3.clear();
 
             personCache3.loadCache(null);
 
             size = personCache3.size(CachePeekMode.ALL);
-            if (size != TestsHelper.getBulkOperationSize()) {
-                throw new RuntimeException("Cache data was incorrectly loaded from Cassandra. " +
+            Assert.assertEquals("Cache data was incorrectly loaded from Cassandra. " +
                     "Expected number of records is " + TestsHelper.getBulkOperationSize() +
-                    ", but loaded number of records is " + size);
-            }
+                    ", but loaded number of records is " + size,
+                TestsHelper.getBulkOperationSize(), size);
 
             LOGGER.info("Cache data loaded from Cassandra table");
         }
