diff --git a/old_test.java b/new_test.java
index 78b7e11..bb62dce 100644
--- a/old_test.java
+++ b/new_test.java
@@ -85,7 +85,6 @@
 
             @Override
             protected IDelegateFactory createLoggerFactory(IPluginRegistry pluginRegistry) {
-                // TODO Auto-generated method stub
                 return null;
             }
 
@@ -93,6 +92,10 @@
             protected IApiRequestPathParser createRequestPathParser(IPluginRegistry pluginRegistry) {
                 return new DefaultRequestPathParser(null);
             }
+
+            @Override
+            protected void complete() {
+            }
         };
 
         IEngine engine = factory.createEngine();
