diff --git a/old_product.java b/new_product.java
index 0dfb866..6382fd6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,3 @@
-private Signature getSigner() throws NoSuchAlgorithmException{
-		if(signer == null){
-			signer = Signature.getInstance(JwsAlgorithm.getByName(super.getAlgorithm()).getStandardName());
-		}
-		return signer;
+public void initializeSigner() throws NoSuchAlgorithmException{
+		signer = Signature.getInstance(JwsAlgorithm.getByName(super.getAlgorithm()).getStandardName());
 	}
\ No newline at end of file
