diff --git a/old_test.java b/new_test.java
index 21dae60..7643c13 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @TestInfo(
-            status = TestStatus.LGTM,
-            notes = "The test verifies the basic functionality of end() method.",
+            level = TestLevel.PARTIAL_OK,
+            purpose = "Verifies the basic functionality of end() method. ",
             targets = { @TestTarget(methodName = "end", 
                                     methodArgs = {})                         
             }
