diff --git a/old_product.java b/new_product.java
index 80e7386..99f0621 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,9 @@
 @Override
-	protected final byte[] crypt(final byte[] input, final int mode)
+	protected byte[] crypt(final byte[] input, final int mode)
 		throws GeneralSecurityException
 	{
 		SecretKey key = generateSecretKey();
-		PBEParameterSpec spec = new PBEParameterSpec(salt, COUNT);
-		Cipher ciph = Cipher.getInstance(CRYPT_METHOD);
-		ciph.init(mode, key, spec);
+		AlgorithmParameterSpec spec = createParameterSpec();
+		Cipher ciph = createCipher(key, spec, mode);
 		return ciph.doFinal(input);
 	}
\ No newline at end of file
