diff --git a/old_test.java b/new_test.java
index 5b5f3b5..5f35fdf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,7 +20,7 @@
 			intrinsic.radial[i] = rand.nextGaussian()*intrinsic.radial[i]*0.1;
 		}
 
-		CalibrationPlanarGridZhang99 alg = new CalibrationPlanarGridZhang99(grid,true,2,true);
+		CalibrationPlanarGridZhang99 alg = new CalibrationPlanarGridZhang99(grid,new CalibParamPinholeRadial(true,2,true));
 		assertTrue(alg.optimizedParam(observations, grid, initial, found,null));
 
 		checkEquals(expected, found, initial);
