diff --git a/old_product.java b/new_product.java
index b9b26ab..336d5c0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -17,13 +17,15 @@ public boolean optimizedParam( List<CalibrationObservation> observations ,
 		Zhang99OptimizationFunction func = new Zhang99OptimizationFunction(
 				initial.createLike(), grid,observations);
 
-		Zhang99OptimizationJacobian jacobian = new Zhang99OptimizationJacobian(
-				(CalibParamPinholeRadial)found.getIntrinsic(), observations,grid);
+		Zhang99OptimizationJacobian jacobian = initial.getIntrinsic().createJacobian(observations,grid);
 
 		optimizer.setFunction(func,jacobian);
 		optimizer.initialize(model,1e-10,1e-25*observations.size());
 
+//		System.out.println("Error before = "+optimizer.getFunctionValue());
+
 		for( int i = 0; i < 500; i++ ) {
+//			System.out.println("i = "+i);
 			if( optimizer.iterate() ) {
 				break;
 			} else {
@@ -35,5 +37,7 @@ public boolean optimizedParam( List<CalibrationObservation> observations ,
 		double param[] = optimizer.getParameters();
 		found.setFromParam(param);
 
+//		System.out.println("Error after = "+optimizer.getFunctionValue());
+
 		return true;
 	}
\ No newline at end of file
