diff --git a/old_test.java b/new_test.java
index 50de9d6..96ed617 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,7 +12,7 @@
 
 		Element<Corner> expected = alg.list.getHead().next;
 		PolylineSplitMerge.ErrorValue foundError = new PolylineSplitMerge.ErrorValue();
-		Element<Corner> found = alg.selectCornerToRemove(contour,foundError);
+		Element<Corner> found = alg.selectCornerToRemove(contour,foundError,true);
 
 		assertTrue(expected == found);
 		assertEquals(0,foundError.value, GrlConstants.TEST_F64);
