diff --git a/old_product.java b/new_product.java
index c0b33a8..0126749 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,5 @@
-@Nullable
-  public static String resolveIpAddress(String hostname) throws UnknownHostException {
-    if (hostname == null || hostname.isEmpty()) {
-      return null;
-    }
-
+public static String resolveIpAddress(String hostname) throws UnknownHostException {
+    Preconditions.checkNotNull(hostname);
+    Preconditions.checkArgument(!hostname.isEmpty());
     return InetAddress.getByName(hostname).getHostAddress();
   }
\ No newline at end of file
