diff --git a/old_test.java b/new_test.java
index 7813e87..a255065 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,12 +2,12 @@
     public void testGetStateFromStatusString() {
         System.out.println("getStateFromStatusString");
         String status;
-        Capabilities version;
+        Capabilities version = new Capabilities();
         String expResult;
         String result;
 
         status = "<Idle,MPos:5.529,0.560,7.000,WPos:1.529,-5.440,-0.000>";
-        version = Capabilities.STATUS_C;
+        version.REAL_TIME = true;
         expResult = "Idle";
         result = GrblUtils.getStateFromStatusString(status, version);
         assertEquals(expResult, result);
