diff --git a/old_product.java b/new_product.java
index 0fc9dd2..dae44cf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 @Override
-    public MessageActionResult sendMessages(List<ProtocolMessage> messages, List<Record> records) {
+    public MessageActionResult sendMessages(List<ProtocolMessage> messages, List<AbstractRecord> records) {
         if (!proceed) {
-            return new MessageActionResult(new LinkedList<Record>(), new LinkedList<ProtocolMessage>());
+            return new MessageActionResult(new LinkedList<AbstractRecord>(), new LinkedList<ProtocolMessage>());
         }
         if (records == null) {
             records = new LinkedList<>();
