diff --git a/old_test.java b/new_test.java
index b277221..427ea65 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testSendMessages() {
         List<ProtocolMessage> protocolMessages = new LinkedList<>();
         protocolMessages.add(message);
-        List<Record> records = new LinkedList<>();
+        List<AbstractRecord> records = new LinkedList<>();
         records.add(record);
         executor.sendMessages(protocolMessages, records);
         byte[] sendByte = ((FakeTransportHandler) context.getTransportHandler()).getSendByte();
