diff --git a/old_product.java b/new_product.java
index 6d22566..002bf78 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-synchronized EnumSet<CleanupStatus> removeFinishedWrites() {
+synchronized CleanupStatus removeFinishedWrites() {
         Exceptions.checkNotClosed(this.closed, this);
         long currentTime = this.timeSupplier.get();
         long totalElapsed = 0;
@@ -16,6 +16,7 @@ synchronized EnumSet<CleanupStatus> removeFinishedWrites() {
             this.lastDurationMillis = (int) (totalElapsed / removedCount / AbstractTimer.NANOS_TO_MILLIS);
         }
 
-        CleanupStatus empty = this.writes.isEmpty() ? CleanupStatus.QueueEmpty : CleanupStatus.QueueNotEmpty;
-        return failedWrite ? EnumSet.of(CleanupStatus.WriteFailed, empty) : EnumSet.of(empty);
+        return failedWrite
+                ? CleanupStatus.WriteFailed
+                : this.writes.isEmpty() ? CleanupStatus.QueueEmpty : CleanupStatus.QueueNotEmpty;
     }
\ No newline at end of file
