diff --git a/old_test.java b/new_test.java
index d16a286..c3a3307 100644
--- a/old_test.java
+++ b/new_test.java
@@ -21,8 +21,8 @@
             val write = writes.pollFirst();
             if (!write.isDone()) {
                 val result1 = q.removeFinishedWrites();
-                AssertExtensions.assertContainsSameElements("Unexpected value from removeFinishedWrites when there were writes left in the queue.",
-                        EnumSet.of(WriteQueue.CleanupStatus.QueueNotEmpty), result1);
+                Assert.assertEquals("Unexpected value from removeFinishedWrites when there were writes left in the queue.",
+                        WriteQueue.CleanupStatus.QueueNotEmpty, result1);
                 val stats1 = q.getStatistics();
                 Assert.assertEquals("Unexpected size after removeFinishedWrites with no effect.", writes.size() + 1, stats1.getSize());
 
@@ -41,8 +41,8 @@
             expectedElapsed = (time.get() * removed - expectedElapsed) / AbstractTimer.NANOS_TO_MILLIS / removed;
 
             val result2 = q.removeFinishedWrites();
-            val expectedResult = EnumSet.of(writes.isEmpty() ? WriteQueue.CleanupStatus.QueueEmpty : WriteQueue.CleanupStatus.QueueNotEmpty);
-            AssertExtensions.assertContainsSameElements("Unexpected result from removeFinishedWrites.", expectedResult, result2);
+            val expectedResult = writes.isEmpty() ? WriteQueue.CleanupStatus.QueueEmpty : WriteQueue.CleanupStatus.QueueNotEmpty;
+            Assert.assertEquals("Unexpected result from removeFinishedWrites.", expectedResult, result2);
             val stats2 = q.getStatistics();
             Assert.assertEquals("Unexpected size after removeFinishedWrites.", writes.size(), stats2.getSize());
             Assert.assertEquals("Unexpected getExpectedProcessingTimeMillis after clear.", expectedElapsed, stats2.getExpectedProcessingTimeMillis());
@@ -53,7 +53,7 @@
         q.add(w3);
         w3.fail(new IntentionalException(), true);
         val result3 = q.removeFinishedWrites();
-        AssertExtensions.assertContainsSameElements("Unexpected value from removeFinishedWrites when there were failed writes.",
-                EnumSet.of(WriteQueue.CleanupStatus.QueueEmpty, WriteQueue.CleanupStatus.WriteFailed), result3);
+        Assert.assertEquals("Unexpected value from removeFinishedWrites when there were failed writes.",
+                WriteQueue.CleanupStatus.WriteFailed, result3);
 
     }
\ No newline at end of file
