diff --git a/old_test.java b/new_test.java
index e705498..880363e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testNewConnection() throws Exception {
         String uri = "someurl";
         String user = "a";
-        String password = "b";
+        char[] password = "b".toCharArray();
         Connection result = instance.newConnection(uri, user, password);
         assertNotNull(result);
         assertFalse(result.isClosed());
