diff --git a/old_product.java b/new_product.java
index ac7ca00..61e2ed2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-public boolean kownTheMessage(String hashHex) {
-        boolean b = this.cacheMapRecieved.containsKey(hashHex) ||
-                this.cacheMapSended.containsKey(hashHex);
+public boolean kownTheMessage(NulsDigestData hash) {
+        boolean b = this.cacheMapRecieved.containsKey(hash) ||
+                this.cacheMapSended.containsKey(hash);
         return b;
     }
\ No newline at end of file
