diff --git a/old_test.java b/new_test.java
index 5470817..044315d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
     public void kownTheMessage() {
-        if (null == messageCacheService.getSendMessage(blockMessage.getHash().getDigestHex())){
+        if (null == messageCacheService.getSendMessage(blockMessage.getHash())) {
             messageCacheService.cacheSendedMessage(blockMessage);
-            assertTrue(messageCacheService.kownTheMessage(blockMessage.getHash().getDigestHex()));
-        }else{
-            assertTrue(messageCacheService.kownTheMessage(blockMessage.getHash().getDigestHex()));
+            assertTrue(messageCacheService.kownTheMessage(blockMessage.getHash()));
+        } else {
+            assertTrue(messageCacheService.kownTheMessage(blockMessage.getHash()));
         }
-        messageCacheService.cacheRecievedMessageHash(blockMessage.getHash().getDigestHex());
-        assertTrue(messageCacheService.kownTheMessage(blockMessage.getHash().getDigestHex()));
+        messageCacheService.cacheRecievedMessageHash(blockMessage.getHash());
+        assertTrue(messageCacheService.kownTheMessage(blockMessage.getHash()));
     }
\ No newline at end of file
