diff --git a/old_product.java b/new_product.java
index 01142f9..b5d2b62 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,7 +9,7 @@
             RealmObjectProxy proxy = (RealmObjectProxy) copyToRealmIfNeeded(object);
             view.insert(location, proxy.realmGet$proxyState().getRow$realm().getIndex());
         } else {
-            nonManagedList.add(location, object);
+            unmanagedList.add(location, object);
         }
         modCount++;
     }
\ No newline at end of file
