diff --git a/old_product.java b/new_product.java
index 9758aac..c4642b2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static int nextInt(int startInclusive, int endExclusive) {
+public static int nextInt(final int startInclusive, final int endExclusive) {
         Validate.isTrue(endExclusive >= startInclusive,
                 "Start value must be smaller or equal to end value.");
         Validate.isTrue(startInclusive >= 0, "Both range values must be non-negative.");
