diff --git a/old_product.java b/new_product.java
index dd53a98..bbd8f59 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 protected static ITestNGMethod[] findDependedUponMethods(ITestNGMethod m, List<ITestNGMethod> methods) {
-    ITestNGMethod[] methodsArray = methods.toArray(new ITestNGMethod[methods.size()]);
+    ITestNGMethod[] methodsArray = methods.toArray(new ITestNGMethod[0]);
     return findDependedUponMethods(m, methodsArray);
   }
\ No newline at end of file
