diff --git a/old_product.java b/new_product.java
index de29223..574acfb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-List<SchemaDto> getUserNotificationSchemasByAppId(String applicationId) throws ControlServiceException;
\ No newline at end of file
+List<VersionDto> getUserNotificationSchemasByAppId(String applicationId) throws ControlServiceException;
\ No newline at end of file
