diff --git a/old_test.java b/new_test.java
index fe1f8a0..d80c14f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
     public void testSetUpdated() {
         final Auditable a = new Auditable();
-        final Date ou = a.getUpdated();
-        final Date newer = new Date(ou.getTime() + 1);
+        Assert.assertNull(a.getUpdated());
+        final Date newer = new Date();
         a.setUpdated(newer);
-        Assert.assertEquals(newer, a.getUpdated());
+        Assert.assertNull(a.getUpdated());
     }
\ No newline at end of file
