diff --git a/old_product.java b/new_product.java
index 2fdfdd9..d60d5d2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,4 @@
-public void setUpdated(final Date updated) {
-        this.updated.setTime(updated.getTime());
+@JsonIgnore
+    public void setUpdated(final Date updated) {
+        LOG.info("Tried to set updated time to " + updated + " for entity " + this.id + ". Ignoring.");
     }
\ No newline at end of file
