diff --git a/old_product.java b/new_product.java
index 147d5bd..7a9a7ca 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 @Override
   public int putAll(KTypeVTypeAssociativeContainer<? extends KType, ? extends VType> container) {
-    final int count = this.assigned;
+    final int count = size();
     for (KTypeVTypeCursor<? extends KType, ? extends VType> c : container) {
       put(c.key, c.value);
     }
-    return this.assigned - count;
+    return size() - count;
   }
\ No newline at end of file
