diff --git a/old_product.java b/new_product.java
index cf6ffc8..ebfbb3a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public static String createBinaryString(IpPrefix ipPrefix) {
         if (ipPrefix.prefixLength() == 0) {
-            return "";
+            return "0";
         }
 
         byte[] octets = ipPrefix.address().toOctets();
@@ -13,5 +13,6 @@ public static String createBinaryString(IpPrefix ipPrefix) {
             boolean isSet = ((value & mask) != 0);
             result.append(isSet ? "1" : "0");
         }
-        return result.toString();
+
+        return "0" + result.toString();
     }
\ No newline at end of file
