diff --git a/old_test.java b/new_test.java
index 2256655..a724193 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,11 @@
 @Test
   public void shift() {
-    hexQuery(_BIN_SHIFT.args("()", 1),          "");
-    hexQuery(_BIN_SHIFT.args(base64("77"), 0),  "77");
-    hexQuery(_BIN_SHIFT.args(base64("33"), 1),  "66");
-    hexQuery(_BIN_SHIFT.args(base64("66"), -1), "33");
+    hexQuery(_BIN_SHIFT.args("()", 1),                 "");
+    hexQuery(_BIN_SHIFT.args(base64("77"), 0),         "77");
+    hexQuery(_BIN_SHIFT.args(base64("33"), 1),         "66");
+    hexQuery(_BIN_SHIFT.args(base64("66"), -1),        "33");
+    hexQuery(_BIN_SHIFT.args(base64("0066"), 8),       "6600");
+    hexQuery(_BIN_SHIFT.args(base64("6600"), -8),      "0066");
+    hexQuery(_BIN_SHIFT.args(base64("12345678"), 16),  "56780000");
+    hexQuery(_BIN_SHIFT.args(base64("12345678"), -16), "00001234");
   }
\ No newline at end of file
