diff --git a/old_product.java b/new_product.java
index b8f86c1..327ac88 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
-void commitBlock(long workerId, long usedBytesOnTier, String tierAlias, long blockId, long
-      length) throws NotFoundException, UnavailableException;
\ No newline at end of file
+void commitBlock(long workerId, long usedBytesOnTier, String tierAlias, String mediumType,
+      long blockId, long length) throws NotFoundException, UnavailableException;
\ No newline at end of file
