diff --git a/old_product.java b/new_product.java
index 5bda047..730b22b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,8 @@
 public Map<Instance, Long> degrees() {
         Map<Instance, Long> allDegrees = new HashMap<>();
+        MindmapsComputer computer = MindmapsClient.getGraphComputer(keySpace);
         ComputerResult result = computer.compute(new DegreeVertexProgram(allTypes));
+        MindmapsGraph graph = MindmapsClient.getGraph(keySpace);
         result.graph().traversal().V().forEachRemaining(v -> {
             if (v.keys().contains(DegreeVertexProgram.DEGREE)) {
                 Instance instance = graph.getInstance(v.value(ITEM_IDENTIFIER.name()));
