diff --git a/old_test.java b/new_test.java
index 55327d5..1e72d55 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,10 @@
 	public void validate_shouldPassIfAllFieldsAreCorrect() throws Exception {
 		OrderType orderType = new OrderType();
 		orderType.setName("unique name");
+		orderType.setJavaClassName("org.openmrs.TestDrugOrder");
+		Collection<ConceptClass> col = new HashSet<ConceptClass>();
+		col.add(Context.getConceptService().getConceptClass(2));
+		orderType.setConceptClasses(col);
 		Errors errors = new BindException(orderType, "orderType");
 		new OrderTypeValidator().validate(orderType, errors);
 		
