diff --git a/old_product.java b/new_product.java
index 3101de9..9d590e6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,9 +9,25 @@ public void validate(Object obj, Errors errors) {
 				errors.rejectValue("name", "error.name");
 			}
 			
-			OrderType ot = Context.getOrderService().getOrderTypeByName(name);
-			if (ot != null && !orderType.equals(ot)) {
+			OrderType duplicate = Context.getOrderService().getOrderTypeByName(name);
+			if (duplicate != null && !orderType.equals(duplicate)) {
 				errors.rejectValue("name", "OrderType.duplicate.name", "Duplicate order type name: " + name);
 			}
+			
+			for (OrderType ot : Context.getOrderService().getOrderTypes(true)) {
+				if (ot != null) {
+					if (OpenmrsUtil.nullSafeEquals(orderType.getJavaClassName(), ot.getJavaClassName())) {
+						errors.rejectValue("javaClassName", "OrderType.duplicate.javaClass",
+						    "Duplicate order type java class: " + ot.getJavaClassName());
+					} else {
+						for (ConceptClass cc : ot.getConceptClasses()) {
+							if (cc != null && orderType.getConceptClasses().contains(cc)) {
+								errors.rejectValue("conceptClasses", "OrderType.duplicate.conceptClass",
+								    "Duplicate order type concept class: " + cc.getName());
+							}
+						}
+					}
+				}
+			}
 		}
 	}
\ No newline at end of file
