diff --git a/old_product.java b/new_product.java
index d1535de..7e7d92f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-protected static MediaType contentType(final String query) throws IOException {
+protected static MediaType mediaType(final String query) throws IOException {
     final IOUrl url = new IOUrl(REST_ROOT + query);
     final HttpURLConnection conn = (HttpURLConnection) url.connection();
     try {
