diff --git a/old_test.java b/new_test.java
index 6d7717f..f899db6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,5 +5,5 @@
         instance.addAsEvidence("pom", mavenArtifact, Confidence.HIGH);
         assertTrue(instance.contains(EvidenceType.VENDOR, Confidence.HIGH));
         assertEquals(3, instance.size());
-        assertFalse(instance.getIdentifiers().isEmpty());
+        assertFalse(instance.getSoftwareIdentifiers().isEmpty());
     }
\ No newline at end of file
