diff --git a/old_test.java b/new_test.java
index dc207a3..d3d1c23 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,11 @@
 @Test
     public void testSpatialRangeQueryUsingIndex() throws Exception {
-    	PointRDD pointRDD = new PointRDD(sc, InputLocation, offset, splitter);
-    	pointRDD.buildIndex("rtree");
+    	PointRDD spatialRDD = new PointRDD(sc, InputLocation, offset, splitter, true);
+    	spatialRDD.buildIndex(IndexType.RTREE,false);
     	for(int i=0;i<loopTimes;i++)
     	{
-    		long resultSize = RangeQuery.SpatialRangeQueryUsingIndex(pointRDD, queryEnvelope, 0).getRawPointRDD().count();
+    		long resultSize = RangeQuery.SpatialRangeQuery(spatialRDD, queryEnvelope, 0,true).count();
     		assert resultSize>-1;
     	}
-    	assert RangeQuery.SpatialRangeQueryUsingIndex(pointRDD, queryEnvelope, 0).getRawPointRDD().take(10).get(1).getUserData().toString() !=null;
-        
+    	assert RangeQuery.SpatialRangeQuery(spatialRDD, queryEnvelope, 0,true).take(10).get(1).getUserData().toString() !=null;
     }
\ No newline at end of file
