diff --git a/old_product.java b/new_product.java
index b7e815e..50d3b99 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,18 +8,20 @@ void scoreDirection(int dx , int dy ) {
 			}
 		} else if( dx < 0 ) {
 			for (int y = 0; y < lengthY; y++) {
-				trajectories.grow().set(lengthX-1,y,dx,dy);
+				trajectories.grow().set(effectiveLengthX-1,y,dx,dy);
 			}
 		}
 		if( dy > 0 ) {
-			int x0 = dx > 0 ? 1 : 0;
-			int x1 = dx < 0 ? lengthX-1 : lengthX;
+			int x0 = 0, x1 = effectiveLengthX;
+			if( dx > 0) x0 += 1;
+			if( dx < 0) x1 -= 1;
 			for (int x = x0; x < x1; x++) {
 				trajectories.grow().set(x,0,dx,dy);
 			}
 		} else if( dy < 0 ) {
-			int x0 = dx > 0 ? 1 : 0;
-			int x1 = dx < 0 ? lengthX-1 : lengthX;
+			int x0 = 0, x1 = effectiveLengthX;
+			if( dx > 0) x0 += 1;
+			if( dx < 0) x1 -= 1;
 			for (int x = x0; x < x1; x++) {
 				trajectories.grow().set(x,lengthY-1,dx,dy);
 			}
