diff --git a/old_product.java b/new_product.java
index 2a099fb..9e3a4ef 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,10 @@
-private void initialize(Resource storeFile, String storePass, String storeType) {
+private KeyStore initialize(Resource storeFile, String storePass, String storeType) {
         InputStream inputStream = null;
         try {
             inputStream = storeFile.getInputStream();
-            ks = KeyStore.getInstance(storeType);
+            KeyStore ks = KeyStore.getInstance(storeType);
             ks.load(inputStream, storePass.toCharArray());
+            return ks;
         } catch (Exception e) {
             log.error("Error initializing key store", e);
             throw new RuntimeException("Error initializing keystore", e);
