diff --git a/old_test.java b/new_test.java
index ef09a7a..6e80a96 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test(dataProvider="calendars")
     public void test_date_comparisons(Chronology chrono) {
-        List<ChronoLocalDate<?>> dates = new ArrayList<>();
+        List<ChronoLocalDate> dates = new ArrayList<>();
 
-        ChronoLocalDate<?> date = chrono.date(LocalDate.of(1900, 1, 1));
+        ChronoLocalDate date = chrono.date(LocalDate.of(1900, 1, 1));
 
         // Insert dates in order, no duplicates
         if (chrono != JapaneseChronology.INSTANCE) {
@@ -25,20 +25,20 @@
 
         // Check these dates against the corresponding dates for every calendar
         for (Chronology[] clist : data_of_calendars()) {
-            List<ChronoLocalDate<?>> otherDates = new ArrayList<>();
+            List<ChronoLocalDate> otherDates = new ArrayList<>();
             Chronology chrono2 = clist[0];
             if (chrono2 == JapaneseChronology.INSTANCE) {
                 continue;
             }
-            for (ChronoLocalDate<?> d : dates) {
+            for (ChronoLocalDate d : dates) {
                 otherDates.add(chrono2.date(d));
             }
 
             // Now compare  the sequence of original dates with the sequence of converted dates
             for (int i = 0; i < dates.size(); i++) {
-                ChronoLocalDate<?> a = dates.get(i);
+                ChronoLocalDate a = dates.get(i);
                 for (int j = 0; j < otherDates.size(); j++) {
-                    ChronoLocalDate<?> b = otherDates.get(j);
+                    ChronoLocalDate b = otherDates.get(j);
                     int cmp = ChronoLocalDate.DATE_COMPARATOR.compare(a, b);
                     if (i < j) {
                         assertTrue(cmp < 0, a + " compare " + b);
