diff --git a/old_product.java b/new_product.java
index d25631c..68735c6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -50,7 +50,8 @@ public void negotiationDone(OspfMessage ospfMessage,
                 setLastSentDdPacket(ddPacket);
                 getIsMoreBit();
 
-                ch.write(lastSentDdPacket());
+                byte[] messageToWrite = getMessage(lastSentDdPacket);
+                ch.write(messageToWrite);
             } else {
                 // process LSA Vector's List, Add it to LSRequestList.
                 processLsas(payload);
@@ -78,7 +79,8 @@ public void negotiationDone(OspfMessage ospfMessage,
                 setLastSentDdPacket(ddPacket);
                 getIsMoreBit();
                 ddPacket.setDestinationIp(packet.sourceIp());
-                ch.write(lastSentDdPacket());
+                byte[] messageToWrite = getMessage(lastSentDdPacket);
+                ch.write(messageToWrite);
                 startRxMtDdTimer(ch);
             }
         }
