diff --git a/old_test.java b/new_test.java
index 0d23ae4..6c66097 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testIndexOf_StrMatcher() {
-        StrBuilder sb = new StrBuilder();
+        final StrBuilder sb = new StrBuilder();
         assertEquals(-1, sb.indexOf((StrMatcher) null));
         assertEquals(-1, sb.indexOf(StrMatcher.charMatcher('a')));
         
