diff --git a/old_product.java b/new_product.java
index f82b4a3..1b81762 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,8 +5,8 @@ public static int indexOf(final double[] array, final double valueToFind, int st
         if (startIndex < 0) {
             startIndex = 0;
         }
-        double min = valueToFind - tolerance;
-        double max = valueToFind + tolerance;
+        final double min = valueToFind - tolerance;
+        final double max = valueToFind + tolerance;
         for (int i = startIndex; i < array.length; i++) {
             if (array[i] >= min && array[i] <= max) {
                 return i;
