diff --git a/old_product.java b/new_product.java
index 9eabf0b..d4542bd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,7 @@
 public void clearPersistedFiles(List<Long> persistedFiles) {
     synchronized (mLock) {
-      mPersistedFiles.removeAll(persistedFiles);
+      for (long persistedId : persistedFiles) {
+        mPersistedFilesInfo.remove(persistedId);
+      }
     }
   }
\ No newline at end of file
