diff --git a/old_test.java b/new_test.java
index 14a759c..f421cbe 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,8 +3,8 @@
 		String x = "<form>BL</form>";
 		String y = "<form>blabla</form>";
 
-		LOG.debug(StringUtils.getLevenshteinDistance(x, y) + " Thesh: "
-		        + comparator.getThreshold(x, y, 0.7));
+		LOG.debug(new LevenshteinDistance().apply(x, y) + " Threshold: " + comparator
+				.getThreshold(x, y, 0.7));
 		assertTrue(comparator.isClone(x, y, 0.0));
 		assertTrue(comparator.isClone(x, y, 0.5));
 		assertTrue(comparator.isClone(x, y, 0.7));
