diff --git a/old_product.java b/new_product.java
index 901bbf0..e80d1a0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,8 @@
-boolean isClone(String str1, String str2, double thresholdCoef) {
-		if ((thresholdCoef < 0.0) || (thresholdCoef > 1.0)) {
+boolean isClone(String str1, String str2, double thresholdCoefficient) {
+		if ((thresholdCoefficient < 0.0) || (thresholdCoefficient > 1.0)) {
 			throw new IllegalArgumentException(
 					"Threshold Coefficient must be between 0.0 and 1.0!");
-		} else if (StringUtils.getLevenshteinDistance(str1, str2) <= getThreshold(
-				str1, str2, thresholdCoef)) {
-			return true;
-		} else {
-			return false;
-		}
+		} else
+			return LevenshteinDistance.getDefaultInstance().apply(str1, str2) <= getThreshold(
+					str1, str2, thresholdCoefficient);
 	}
\ No newline at end of file
