diff --git a/old_product.java b/new_product.java
index c26731d..163478d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 @SuppressWarnings("unchecked")
-    public static <K, T extends Number> T getMinValue(Map<K, T> map,K[] keys){
+    public static <K, T extends Number> T getMinValue(Map<K, T> map,K...keys){
         Map<K, T> subMap = getSubMap(map, keys);
 
-        if (null == subMap){
+        if (Validator.isNullOrEmpty(subMap)){
             return null;
         }
 
