diff --git a/old_test.java b/new_test.java
index 3220c0e..5e56cdf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,5 @@
 @Test
     public void testGetMinValue(){
-
         Map<String, Integer> map = new HashMap<String, Integer>();
 
         map.put("a", 3007);
@@ -11,7 +10,5 @@
         map.put("f", 3005);
         map.put("g", -1005);
 
-        String[] keys = { "a", "b", "d", "g", "m" };
-
-        LOGGER.info(MapUtil.getMinValue(map, keys) + "");
+        LOGGER.info("" + MapUtil.getMinValue(map, "a", "b", "d", "g", "m"));
     }
\ No newline at end of file
