diff --git a/old_test.java b/new_test.java
index f895297..309ede2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,5 +3,5 @@
         System.out.println("queueStringForComm");
         String input = "input";
         instance.queueStringForComm(input);
-        assertEquals(input, cb.getFirst());
+        assertEquals(input, cb.getFirst().getCommandString());
     }
\ No newline at end of file
