diff --git a/old_test.java b/new_test.java
index 93c16cc..3b394b0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,6 +4,7 @@
         assertIdFromName("EricWittmann", "Eric Wittmann");
         assertIdFromName("DeloitteTouche", "Deloitte & Touche");
         assertIdFromName("JBoss_Overlord", "JBoss_Overlord");
-        assertIdFromName("RedHat", "!Red--Hat?");
-        assertIdFromName("OrgWithPeriods", "Org.With.Periods");
+        assertIdFromName("Red--Hat", "!Red--Hat?");
+        assertIdFromName("Org.With.Periods", "Org.With.Periods");
+        assertIdFromName("my-project", "my-project");
     }
\ No newline at end of file
