diff --git a/old_product.java b/new_product.java
index c41e613..887f8ce 100644
--- a/old_product.java
+++ b/new_product.java
@@ -22,7 +22,7 @@ public static ImmutableList<HasAndroidResourceDeps> getAndroidResourceDeps(
     Predicate<BuildRule> inclusionPredicate = new Predicate<BuildRule>() {
       @Override
       public boolean apply(BuildRule rule) {
-        return androidResourceDeps.contains(rule.getBuildable());
+        return androidResourceDeps.contains(rule);
       }
     };
     ImmutableList<BuildRule> sortedAndroidResourceRules = TopologicalSort.sort(mutableGraph,
