diff --git a/old_test.java b/new_test.java
index 841ea55..9157a64 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,6 @@
 @Test(groups={"tck"})
     public void test_from_Accessor() {
-        assertEquals(LocalDateTime.from(LocalDateTime.of(2007, 7, 15, 17, 30)), LocalDateTime.of(2007, 7, 15, 17, 30));
-        assertEquals(LocalDateTime.from(OffsetDateTime.of(2007, 7, 15, 17, 30, ZoneOffset.ofHours(2))), LocalDateTime.of(2007, 7, 15, 17, 30));
+        LocalDateTime base = LocalDateTime.of(2007, 7, 15, 17, 30);
+        assertEquals(LocalDateTime.from(base), base);
+        assertEquals(LocalDateTime.from(ZonedDateTime.of(base, ZoneOffset.ofHours(2))), base);
     }
\ No newline at end of file
