diff --git a/old_product.java b/new_product.java
index a102538..3465a0e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,8 +3,6 @@ public static LocalDate from(DateTimeAccessor dateTime) {
             return (LocalDate) dateTime;
         } else if (dateTime instanceof LocalDateTime) {
             return ((LocalDateTime) dateTime).getDate();
-        } else if (dateTime instanceof OffsetDateTime) {
-            return ((OffsetDateTime) dateTime).getDate();
         } else if (dateTime instanceof ZonedDateTime) {
             return ((ZonedDateTime) dateTime).getDate();
         }
