diff --git a/old_product.java b/new_product.java
index 7c36bac..b987bcb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,5 @@
 public double mean() {
-        if (mean == null) {
-            final int sz = size();
-            mean = sz > 0 ? sum() / sz : 0;
-        }
-        return mean;
+        final int sz = size();
+        double result = sz > 0 ? sum() / sz : 0;
+        return result;
     }
\ No newline at end of file
