diff --git a/old_test.java b/new_test.java
index b6ff1b5..eb1d082 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,5 +2,7 @@
     public void testMean() {
         assertThat(emptyVector().mean(), closeTo(0));
         assertThat(singleton().mean(), closeTo(Math.PI));
-        assertThat(simpleVector().mean(), closeTo(7.0 / 3));
+        SparseVector sv = simpleVector();
+        assertThat(sv.mean(), closeTo(7.0 / 3));
+        assertThat(sv.mean(), closeTo(7.0 / 3));  // doubled, to check caching
     }
\ No newline at end of file
