diff --git a/old_product.java b/new_product.java
index 0e3daa5..5cd0abc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,8 +10,8 @@ public static <T, O> Map<T, O> groupOne(Collection<O> objectCollection,String pr
             if (!map.containsKey(key)){
                 map.put(key, o);
             }else{
-                if (LOGGER.isDebugEnabled()){
-                    LOGGER.debug("map:{} already has the key:{},ignore!", JsonUtil.format(map.keySet()), key);
+                if (LOGGER.isTraceEnabled()){
+                    LOGGER.trace("map:{} already has the key:[{}],ignore!", JsonUtil.format(map.keySet()), key);
                 }
             }
         }
