diff --git a/old_test.java b/new_test.java
index 8648014..16fad8b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,14 @@
 @Test
     public void testGroupOne(){
-        List<User> list = toList(//
-                        new User("张飞", 23),
-                        new User("刘备", 25),
-                        new User("刘备", 25));
+        User zhangfei = new User("张飞", 23);
+        User liubei25 = new User("刘备", 25);
+        User liubei30 = new User("刘备", 30);
+        List<User> list = toList(zhangfei, liubei25, liubei30);
         Map<String, User> map = CollectionsUtil.groupOne(list, "name");
-        LOGGER.debug(JsonUtil.format(map));
+
+        assertThat(map.keySet(), is(hasSize(2)));
+        assertThat(map, allOf(//
+                        hasEntry("张飞", zhangfei),
+                        hasEntry("刘备", liubei25),
+                        not(hasEntry("刘备", liubei30))));
     }
\ No newline at end of file
