diff --git a/old_test.java b/new_test.java
index a29b510..f79b730 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testEq() {
-        assertTrue(instance.eq("name", "abc").getQueryCriterions()
-                .contains(new EqCriterion("name", "abc")));
+    	assertEquals(new EqCriterion("name", "abc"), instance.eq("name", "abc").getQueryCriterion());
     }
\ No newline at end of file
