diff --git a/old_product.java b/new_product.java
index 9b00187..a879faf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,5 @@
 static String openPre(final Nodes n, final int i) {
-    return "db:open-pre('" + n.data.meta.name + "', " + n.list[i] + ")";
+    System.out.println("? ");
+    return Function.DBOPENPRE.get(null, Str.get(n.data.meta.name),
+        Itr.get(n.list[i])).toString();
   }
\ No newline at end of file
