diff --git a/old_test.java b/new_test.java
index 7552365..dc13427 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void testOpenPre() throws QueryException {
-    final String fun = check(Function.OPENPRE);
+    final String fun = check(Function.DBOPENPRE);
     query(fun + "('db', 0)//title/text()", "XML");
     error(fun + "('db', -1)", Err.IDINVALID);
   }
\ No newline at end of file
