diff --git a/old_product.java b/new_product.java
index 46fbac9..49750e6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -17,5 +17,9 @@ public void validate(Object obj, Errors errors) {
 			if (order.getDose() != null)
 				ValidationUtils.rejectIfEmpty(errors, "doseUnits", "DrugOrder.add.error.missingDoseUnits");
 			
+			// for the following elements Order.hbm.xml says: not-null="true"
+			ValidationUtils.rejectIfEmpty(errors, "drug", "error.null");
+			if (order.getDrug() != null)
+				ValidationUtils.rejectIfEmpty(errors, "drug.concept", "error.null");
 		}
 	}
\ No newline at end of file
