diff --git a/old_product.java b/new_product.java
index 766db6f..7ca58b4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,22 +1,14 @@
 public Tone getTone(final String text, final Scorecard scorecard) {
 
-		if (text == null || text.isEmpty())
-			throw new IllegalArgumentException("text can not be null or empty");
+    if (text == null || text.isEmpty())
+      throw new IllegalArgumentException("text cannot be null or empty");
 
-		JsonObject contentJson = new JsonObject();
-		contentJson.addProperty(TEXT, text);
+    final JsonObject contentJson = new JsonObject();
+    contentJson.addProperty(TEXT, text);
 
-		if (scorecard != null)
-			contentJson.addProperty(SCORECARD, scorecard.getId());
+    if (scorecard != null)
+      contentJson.addProperty(SCORECARD, scorecard.getId());
 
-		HttpRequestBase request = Request.Post("/v1/tone").withContent(contentJson).build();
-
-		try {
-			HttpResponse response = execute(request);
-			String toneAsJson = ResponseUtil.getString(response);
-			Tone tone = GsonSingleton.getGson().fromJson(toneAsJson, Tone.class);
-			return tone;
-		} catch (IOException e) {
-			throw new RuntimeException(e);
-		}
-	}
\ No newline at end of file
+    final Request request = RequestBuilder.post(PATH_TONE).withBodyJson(contentJson).build();
+    return executeRequest(request, Tone.class);
+  }
\ No newline at end of file
