diff --git a/old_product.java b/new_product.java
index 4543b44..a69111e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static double Hypergeometric(int k, int n, int Kp, int Np) {
+public static double hypergeometric(int k, int n, int Kp, int Np) {
         if(k<0 || n<0 || Kp<0 || Np<0) {
             throw new IllegalArgumentException("All the parameters must be positive.");
         }
