diff --git a/old_test.java b/new_test.java
index 9e248c9..ddb7963 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,6 @@
         int Kp = 30;
         int Np = 100;
         double expResult = 0.28116339430254;
-        double result = DiscreteDistributions.Hypergeometric(k, n, Kp, Np);
+        double result = DiscreteDistributions.hypergeometric(k, n, Kp, Np);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
