diff --git a/old_test.java b/new_test.java
index f200e19..3f043f1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
         System.err.println(est.toString());
         
         MarkovGraph markov = markovs.get(BASE_PARTITION, this.catalog_proc);
-        List<Vertex> initial_path = state.getEstimatedPath();
+        List<Vertex> initial_path = state.getInitialPath();
         assertFalse(initial_path.isEmpty());
         
         System.err.println("# of Vertices: " + markov.getVertexCount());
@@ -24,7 +24,7 @@
 //        MarkovUtil.exportGraphviz(markov, false, markov.getPath(multip_path)).writeToTempFile(this.catalog_proc, 1);
         
         assertFalse(est.isSinglePartition(this.thresholds));
-        assertFalse(est.isUserAbort(this.thresholds));
+        assertFalse(est.isAbortable(this.thresholds));
         
         Set<Integer> partitions = p_estimator.getAllPartitions(multip_trace);
         
