diff --git a/old_product.java b/new_product.java
index 2aa9f10..166a7eb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -63,7 +63,7 @@ public State startTransaction(long txn_id, int base_partition, Procedure catalog
             assert(estimator.getEstimate().isValid()) :
                 String.format("Invalid MarkovEstimate for cache Estimator used by %s [hashCode=%d]",
                               TransactionState.formatTxnName(catalog_proc, txn_id), estimator.getEstimate().hashCode());
-            estimator.getEstimate().reused++;
+            estimator.getEstimate().incrementReusedCounter();
         }
         assert(estimator != null);
         if (t) {
