diff --git a/old_product.java b/new_product.java
index c0e6aa4..228a276 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,7 +7,7 @@ public static <T extends ImageBase>
 				   double []features )
 	{
 		int regionSize = widthLargeGrid*widthSubRegion;
-		if( weight.width != regionSize+1 ) {
+		if( weight.width != regionSize ) {
 			throw new IllegalArgumentException("Weighting kernel has an unexpected size");
 		}
 
