diff --git a/old_test.java b/new_test.java
index 4c31a04..bc31fb4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
 		double []found = new double[ 64 ];
 		double []expected = new double[ 64 ];
 
-		Kernel2D_F64 weight = FactoryKernelGaussian.gaussian(2, true, 64, -1, 10);
+		Kernel2D_F64 weight = FactoryKernelGaussian.gaussianWidth(-1, 20);
 		SurfDescribeOps.features(inputF32,c_x,c_y,1.2,weight,4,5,2,scale,false,true,found);
 
 		SparseImageGradient<ImageFloat32,?> gradient = SurfDescribeOps.createGradient(true,false,2,scale,ImageFloat32.class);
