diff --git a/old_product.java b/new_product.java
index 5b16981..9a63c4f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,8 @@
-public static String formatDate(LocalDate date) {
-        return date == null ? "null" : date.format(DATE_FORMATTER);
+public static String formatDate(Date date) {
+        Calendar c = Calendar.getInstance();
+        if (date == null) {
+            return "null";
+        }
+        c.setTime(date);
+        return String.format("%02d/%02d/%04d", c.get(Calendar.MONTH) + 1, c.get(Calendar.DATE), c.get(Calendar.YEAR));
     }
\ No newline at end of file
