diff --git a/old_test.java b/new_test.java
index dee2855..d54b4c6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,9 @@
 @Test
-    public void testFormatDate() {
+    public void testFormatDate() throws ParseException {
         assertEquals("null", FormatUtil.formatDate(null));
-        assertEquals("01/01/2017", FormatUtil.formatDate(LocalDate.parse("2017-01-01")));
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd");
+        assertEquals("01/01/2017", FormatUtil.formatDate(formatter.parse("2017-01-01")));
         assertEquals(null, FormatUtil.formatStringDate(null));
         assertEquals(null, FormatUtil.formatStringDate("Unparseable"));
-        assertEquals(LocalDate.parse("2017-01-01"), FormatUtil.formatStringDate("01/01/2017"));
+        assertEquals(formatter.parse("2017-01-01"), FormatUtil.formatStringDate("01/01/2017"));
     }
\ No newline at end of file
