diff --git a/old_test.java b/new_test.java
index 3834b56..86a2390 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
         ConnectionFactoryResource pool = new ConnectionFactoryResource(1, connectionFactory);
         pool.fillPool();
         assertNotNull(pool);
-        ActiveMQConnection connection = (ActiveMQConnection)pool.borrowObject();
+        ActiveMQConnection connection = (ActiveMQConnection)pool.makeObject();
         assertNotNull(connection);
         assertTrue(connection.isStarted());
         pool.drainPool();
