diff --git a/old_product.java b/new_product.java
index ea36a58..7559d5e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @Override
-        protected void destroyObject(MessageConsumerResources model) throws Exception {
+        public void destroyObject(MessageConsumerResources model) throws Exception {
             if (model != null) {
                 // First clean up our message consumer
                 if (model.getMessageConsumer() != null) {
