diff --git a/old_test.java b/new_test.java
index b20996c..f1e1e1f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
-  public void testDescribeThriftVersion() throws TTransportException, TException, UnknownHostException {
+  public void testDescribeThriftVersion() throws Exception {
     assertEquals("2.1.0",cassandraCluster.describeThriftVersion());
   }
\ No newline at end of file
