diff --git a/old_test.java b/new_test.java
index f50d209..4754280 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,7 +16,7 @@
             KType k = cast(rnd.nextInt(modulo));
             assertEquals(
                 deque.removeLastOccurrence(k) >= 0, 
-                sequence.removeLastOccurrence(k) >= 0);
+                sequence.removeLast(k) >= 0);
         }
 
         assertListEquals(deque.toArray(), sequence.toArray());
