diff --git a/old_product.java b/new_product.java
index c7c9f2a..6ec3cdf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,6 +2,5 @@
   protected boolean isReusable(Bitmap value) {
     Preconditions.checkNotNull(value);
     return !value.isRecycled() &&
-        value.isMutable() &&
-        Bitmaps.BITMAP_CONFIG.equals(value.getConfig());
+        value.isMutable();
   }
\ No newline at end of file
