diff --git a/old_product.java b/new_product.java
index 20de3cc..2c4697c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public static OutStreamOptions defaults(AlluxioConfiguration alluxioConf) {
-    return new OutStreamOptions(alluxioConf);
+public static OutStreamOptions defaults(ClientContext context, AlluxioConfiguration alluxioConf) {
+    return new OutStreamOptions(context, alluxioConf);
   }
\ No newline at end of file
