diff --git a/old_product.java b/new_product.java
index defe9b2..e4353ca 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,8 +8,8 @@ public void addAsEvidence(String source, MavenArtifact mavenArtifact, Confidence
         if (mavenArtifact.getVersion() != null && !mavenArtifact.getVersion().isEmpty()) {
             this.addEvidence(EvidenceType.VERSION, source, "version", mavenArtifact.getVersion(), confidence);
         }
+        boolean found = false;
         if (mavenArtifact.getArtifactUrl() != null && !mavenArtifact.getArtifactUrl().isEmpty()) {
-            boolean found = false;
             synchronized (this) {
                 for (Identifier i : this.identifiers) {
                     if ("maven".equals(i.getType()) && i.getValue().equals(mavenArtifact.toString())) {
@@ -23,9 +23,9 @@ public void addAsEvidence(String source, MavenArtifact mavenArtifact, Confidence
                     }
                 }
             }
-            if (!found) {
-                LOGGER.debug("Adding new maven identifier {}", mavenArtifact);
-                this.addIdentifier("maven", mavenArtifact.toString(), mavenArtifact.getArtifactUrl(), Confidence.HIGHEST);
-            }
+        }
+        if (!found && mavenArtifact.getGroupId() != null && mavenArtifact.getArtifactId() != null && mavenArtifact.getVersion() != null) {
+            LOGGER.debug("Adding new maven identifier {}", mavenArtifact);
+            this.addIdentifier("maven", mavenArtifact.toString(), mavenArtifact.getArtifactUrl(), Confidence.HIGHEST);
         }
     }
\ No newline at end of file
