diff --git a/old_test.java b/new_test.java
index 6756b5d..53cbc6f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,8 @@
 	{
 		SecurePackageResourceGuard guard = new SecurePackageResourceGuard();
 		guard.addPattern("+*.gif");
-		assertTrue(guard.acceptAbsolutePath("test.gif"));
+		assertFalse(guard.acceptAbsolutePath("test.gif"));
+		assertFalse(guard.acceptAbsolutePath("/test.gif"));
 		assertTrue(guard.acceptAbsolutePath("mydir/test.gif"));
 		assertTrue(guard.acceptAbsolutePath("/root/mydir/test.gif"));
 	}
\ No newline at end of file
