diff --git a/old_test.java b/new_test.java
index 516433f..5553e98 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test public void writeVarint_64() {
     long number = 300;
 
-    Buffer buffer = new Buffer(Buffer.varintSizeInBytes(number));
+    Buffer buffer = Buffer.allocate(Buffer.varintSizeInBytes(number));
     buffer.writeVarint(number);
 
     assertThat(buffer.toByteArray())
