diff --git a/old_product.java b/new_product.java
index 5d93694..f36b0d6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 void writeVarint(int v) {
     while ((v & ~0x7f) != 0) {
-      buf[pos++] = (byte) ((v & 0x7f) | 0x80);
+      writeByte((byte) ((v & 0x7f) | 0x80));
       v >>>= 7;
     }
-    buf[pos++] = (byte) v;
+    writeByte((byte) v);
   }
\ No newline at end of file
