diff --git a/old_product.java b/new_product.java
index b6d46ab..426c4dc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -46,6 +46,11 @@ protected boolean acceptAbsolutePath(String path)
 			}
 		}
 
+		//
+		// for windows we have to check both File.separator ('\') and the usual '/' since both can
+		// be used and are used interchangeably
+		//
+
 		if (!allowAccessToRootResources)
 		{
 			String absolute = path;
@@ -53,18 +58,23 @@ protected boolean acceptAbsolutePath(String path)
 			{
 				// handle a windows path which may have a drive letter in it
 
-				if (absolute.indexOf(":\\") > 0)
+				int drive = absolute.indexOf(":\\");
+				if (drive < 0)
+				{
+					drive = absolute.indexOf(":/");
+				}
+				if (drive > 0)
 				{
 					// strip the drive letter off the path
-					absolute = absolute.substring(absolute.indexOf(":\\") + 2);
+					absolute = absolute.substring(drive + 2);
 				}
 			}
 
-			if (absolute.startsWith(File.separator))
+			if (absolute.startsWith(File.separator) || absolute.startsWith("/"))
 			{
 				absolute = absolute.substring(1);
 			}
-			if (!absolute.contains(File.separator))
+			if (!absolute.contains(File.separator) && !absolute.contains("/"))
 			{
 				log.warn("Access to root directory is by default disabled for shared resources: " +
 					path);
