diff --git a/old_test.java b/new_test.java
index 6bc6d8a..26a94d0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,6 +13,13 @@
 			assertTrue(guard.acceptAbsolutePath("\\test\\org\\apache\\test.js"));
 			assertFalse(guard.acceptAbsolutePath("c:\\test.js"));
 			assertFalse(guard.acceptAbsolutePath("\\test.js"));
+
+			// java also generates file paths with '/' on windows
+			assertTrue(guard.acceptAbsolutePath("c:/test/org/apache/test.js"));
+			assertTrue(guard.acceptAbsolutePath("/test/org/apache/test.js"));
+			assertFalse(guard.acceptAbsolutePath("c:/test.js"));
+			assertFalse(guard.acceptAbsolutePath("/test.js"));
 		}
 
+
 	}
\ No newline at end of file
