diff --git a/old_product.java b/new_product.java
index d6e376b..321bc1a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,7 @@
 @VisibleForTesting
-  static String createSnapshotFileName(String serviceName, long serviceId, long index) {
-    return String.format("%s-%d-%d.%s",
-        serviceName,
-        serviceId,
+  static String createSnapshotFileName(String serverName, long index) {
+    return String.format("%s-%d.%s",
+        serverName,
         index,
         EXTENSION);
   }
\ No newline at end of file
