diff --git a/old_test.java b/new_test.java
index 4c85ed0..6a20d9e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
   public void testCreateSnapshotFileName() throws Exception {
-    assertEquals(SnapshotFile.createSnapshotFileName("foo", 1, 2), "foo-1-2.snapshot");
-    assertEquals(SnapshotFile.createSnapshotFileName("foo-bar", 1, 2), "foo-bar-1-2.snapshot");
+    assertEquals(SnapshotFile.createSnapshotFileName("test", 1), "test-1.snapshot");
+    assertEquals(SnapshotFile.createSnapshotFileName("test", 2), "test-2.snapshot");
   }
\ No newline at end of file
