diff --git a/old_product.java b/new_product.java
index f92824d..fae1cef 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,4 @@
-public void addOpsServerLoad(int registeredUsersCount, int processedRequestCount, int deltaCalculationCount) {
+public void addOpsServerLoad(LoadInfo load) {
         removeOldHistory();
-        OperationsServerLoad snap = new OperationsServerLoad();
-        snap.setRegisteredUsersCount(registeredUsersCount);
-        snap.setProcessedRequestCount(processedRequestCount);
-        snap.setDeltaCalculationCount(deltaCalculationCount);
-        history.add(snap);
+        history.add(new OperationsServerLoad(load));
     }
\ No newline at end of file
