diff --git a/old_test.java b/new_test.java
index f81b883..169987f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testAddOpsServerLoad() {
         OperationsServerLoadHistory hist = new OperationsServerLoadHistory(MAX_HISTORY_TIME_LIVE);
-        hist.addOpsServerLoad(1, 2, 3);
+        hist.addOpsServerLoad(new LoadInfo(2));
         fillOutHistory(hist,1000,5);
         assertNotNull(hist.getHistory());
         if (hist.getHistory().size() >= 5) {
