diff --git a/old_product.java b/new_product.java
index df6cbe5..80248c5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
-@Override
-    public SingleSignonToken createSSOToken(Map<String, Object> tokenData) throws TokenCreationFailedException {
+public SingleSignonToken createSSOToken(Map<String, Object> tokenData) throws TokenCreationFailedException {
         try {
             TokenService tokenService = getTokenServiceForType(ssoTokenType);
             SingleSignonTokenImpl ssoToken = new SingleSignonTokenImpl(tokenService);
