diff --git a/old_test.java b/new_test.java
index d08343b..868064f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,6 +3,7 @@
     final IO io = IO.get("<a xmlns:a='a'><b><c/><c/><c/></b></a>");
     try(QueryProcessor qp = new QueryProcessor("/*:a/*:b", context).context(new DBNode(io))) {
       final ANode sub = (ANode) qp.iter().next();
-      DataBuilder.stripNS(sub, token("a"), context);
+      query(DataBuilder.stripNS(sub, token("a"), context).serialize().toString(),
+          "<b><c/><c/><c/></b>");
     }
   }
\ No newline at end of file
