diff --git a/old_product.java b/new_product.java
index 4f049c2..85a3a49 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public static ANode stripNS(final ANode node, final byte[] ns, final Context ctx) {
-    if(node.type != NodeType.ELM) return node;
+    if(node.type != NodeType.ELM && node.type != NodeType.DOC) return node;
 
     final MemData data = new MemData(ctx.options);
     final DataBuilder db = new DataBuilder(data);
