diff --git a/old_test.java b/new_test.java
index cbe4e78..b3d8590 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,7 +20,7 @@
     // test add store onto a new disk, which should succeed
     assertTrue("Add new store should succeed", storageManager.addBlobStore(newPartition1.getReplicaIds().get(0)));
     assertNotNull("The store shouldn't be null because new store is successfully added",
-        storageManager.getStore(newPartition1));
+        storageManager.getStore(newPartition1, false));
     // test add store whose diskManager is not running, which should fail
     PartitionId newPartition2 =
         new MockPartitionId(11L, MockClusterMap.DEFAULT_PARTITION_CLASS, clusterMap.getDataNodes(), 0);
@@ -42,6 +42,6 @@
 
     assertFalse("Add store should fail if store couldn't start due to initializePool failure",
         storageManager2.addBlobStore(newPartition2.getReplicaIds().get(0)));
-    assertNull("New store shouldn't be in in-memory data structure", storageManager2.getStore(newPartition2));
+    assertNull("New store shouldn't be in in-memory data structure", storageManager2.getStore(newPartition2, false));
     shutdownAndAssertStoresInaccessible(storageManager2, localReplicas);
   }
\ No newline at end of file
