diff --git a/old_product.java b/new_product.java
index b135ca9..0e641d5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,8 @@
 @SuppressWarnings("unchecked")
 	public EncounterType getEncounterType(String name) throws DAOException {
 		Criteria crit = sessionFactory.getCurrentSession().createCriteria(EncounterType.class);
-		HibernateUtil.addEqCriterionForLocalizedColumn(name, "name", crit);
+		crit.add(Expression.eq("retired", false));
+		HibernateUtil.addEqCriterionForLocalizedColumn(name, "localizedName", crit);
 		List<EncounterType> types = crit.list();
 		if (null == types || types.isEmpty()) {
 			return null;
