diff --git a/old_product.java b/new_product.java
index ea8a431..35248b5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ public static JavaRDD<Point> SpatialRangeQuery(PointRDD spatialRDD, Polygon quer
 		if(useIndex==true)
 		{
 	        if(spatialRDD.indexedRawRDD == null) {
-	            throw new Exception("[RangeQuery][SpatialRangeQuery] Index doesn't exist. Please build index.");
+	            throw new Exception("[RangeQuery][SpatialRangeQuery] Index doesn't exist. Please build index on rawSpatialRDD.");
 	        }
 			JavaRDD<Object> result = spatialRDD.indexedRawRDD.mapPartitions(new RangeFilterUsingIndex(queryWindow));
 			return result.map(new Function<Object, Point>()
