diff --git a/old_test.java b/new_test.java
index 5ebc052..2f51f81 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testSpatialRangeQuery() throws Exception {
-    	RectangleRDD spatialRDD = new RectangleRDD(sc, InputLocation, offset, splitter);
+    	RectangleRDD spatialRDD = new RectangleRDD(sc, InputLocation, offset, splitter, true);
     	for(int i=0;i<loopTimes;i++)
     	{
     		long resultSize = RangeQuery.SpatialRangeQuery(spatialRDD, queryEnvelope, 0,false).count();
