diff --git a/old_product.java b/new_product.java
index 8d77a69..7976e14 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public boolean optimizedParam( List<List<Point2D_F64>> observations ,
 								   List<Point2D_F64> grid ,
-								   Zhang99Parameters initial ,
-								   Zhang99Parameters found ,
+								   Zhang99ParamAll initial ,
+								   Zhang99ParamAll found ,
 								   UnconstrainedLeastSquares optimizer )
 	{
 		if( optimizer == null ) {
@@ -11,7 +11,7 @@ public boolean optimizedParam( List<List<Point2D_F64>> observations ,
 //			optimizer = FactoryOptimization.leastSquareLevenberg(1e-3);
 		}
 
-		double model[] = new double[ initial.size() ];
+		double model[] = new double[ initial.numParameters() ];
 		initial.convertToParam(model);
 
 		Zhang99OptimizationFunction func = new Zhang99OptimizationFunction(
