diff --git a/old_test.java b/new_test.java
index 94cdf84..0536607 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,13 +3,13 @@
 
 		PlanarCalibrationTarget config = GenericCalibrationGrid.createStandardConfig();
 		List<Point2D_F64> grid = config.points;
-		Zhang99Parameters initial = GenericCalibrationGrid.createStandardParam(true,2,3,rand);
-		Zhang99Parameters found = new Zhang99Parameters(true,2,false,3);// TODO add in includeTangential?
+		Zhang99ParamAll initial = GenericCalibrationGrid.createStandardParam(true,2,true,3,rand);
+		Zhang99ParamAll found = new Zhang99ParamAll(true,2,true,3);
 
 		List<List<Point2D_F64>> observations = GenericCalibrationGrid.createObservations(initial,grid);
 
-		CalibrationPlanarGridZhang99 alg = new CalibrationPlanarGridZhang99(config,true,2);
+		CalibrationPlanarGridZhang99 alg = new CalibrationPlanarGridZhang99(config,true,2,true);
 		assertTrue(alg.optimizedParam(observations, grid, initial, found,null));
 
-		checkEquals(initial, found,initial);
+		checkEquals(initial, found, initial);
 	}
\ No newline at end of file
