diff --git a/old_test.java b/new_test.java
index 45f7804..e8edaaa 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void delete() throws Exception {
     AlluxioURI file = new AlluxioURI("/file");
-    DeleteOptions deleteOptions = DeleteOptions.defaults().setRecursive(true);
+    DeletePOptions deleteOptions = DeletePOptions.newBuilder().setRecursive(true).build();
     mFileSystem.delete(file, deleteOptions);
     verify(mFileSystemMasterClient).delete(file, deleteOptions);
 
