diff --git a/old_product.java b/new_product.java
index 0e487b9..7aae644 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
-public static FuncType get(final SeqType type, final SeqType... args) {
-    return get(new AnnList(), type, args);
+public static FuncType get(final SeqType declType, final SeqType... args) {
+    return get(new AnnList(), declType, args);
   }
\ No newline at end of file
