diff --git a/old_product.java b/new_product.java
index 25ca525..bd162a4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public ServiceCall<DialogNode> createDialogNode(CreateDialogNodeOptions createDialogNodeOptions) {
     Validator.notNull(createDialogNodeOptions, "createDialogNodeOptions cannot be null");
-    RequestBuilder builder = RequestBuilder.post(String.format("/v1/workspaces/%s/dialog_nodes", createDialogNodeOptions
-        .workspaceId()));
+    RequestBuilder builder = RequestBuilder.post(String.format("/v1/workspaces/%s/dialog_nodes",
+            createDialogNodeOptions.workspaceId()));
     builder.query(VERSION, versionDate);
     final JsonObject contentJson = new JsonObject();
     if (createDialogNodeOptions.parent() != null) {
@@ -13,8 +13,8 @@ public ServiceCall<DialogNode> createDialogNode(CreateDialogNodeOptions createDi
     if (createDialogNodeOptions.description() != null) {
       contentJson.addProperty("description", createDialogNodeOptions.description());
     }
-    if (createDialogNodeOptions.type() != null) {
-      contentJson.addProperty("type", createDialogNodeOptions.type());
+    if (createDialogNodeOptions.nodeType() != null) {
+      contentJson.addProperty("type", createDialogNodeOptions.nodeType());
     }
     if (createDialogNodeOptions.title() != null) {
       contentJson.addProperty("title", createDialogNodeOptions.title());
