diff --git a/old_test.java b/new_test.java
index 8e06c26..d3b8975 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,13 +6,13 @@
     Date start = new Date();
 
     CreateDialogNodeOptions createOptions =
-        new CreateDialogNodeOptions.Builder(workspaceId, dialogNodeName).description(dialogNodeDescription).build();
+            new CreateDialogNodeOptions.Builder(workspaceId, dialogNodeName).description(dialogNodeDescription).build();
     DialogNode response = service.createDialogNode(createOptions).execute();
 
     try {
       assertNotNull(response);
-      assertNotNull(response.getDialogNode());
-      assertEquals(response.getDialogNode(), dialogNodeName);
+      assertNotNull(response.getDialogNodeId());
+      assertEquals(response.getDialogNodeId(), dialogNodeName);
       assertNotNull(response.getDescription());
       assertEquals(response.getDescription(), dialogNodeDescription);
       assertNotNull(response.getCreated());
