diff --git a/old_product.java b/new_product.java
index 08be02e..0a257f8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,7 @@
 public TopicConfig setGlobalOrderingEnabled(boolean globalOrderingEnabled) {
+        if (this.multiThreadingEnabled && globalOrderingEnabled) {
+            throw new IllegalArgumentException("Global ordering can not be enabled when multi-threading is used.");
+        }
         this.globalOrderingEnabled = globalOrderingEnabled;
         return this;
     }
\ No newline at end of file
