diff --git a/old_test.java b/new_test.java
index f7313d6..378dbff 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,4 +2,11 @@
     public void testSetGlobalOrderingEnabled() {
         TopicConfig topicConfig = new TopicConfig().setGlobalOrderingEnabled(true);
         assertTrue(topicConfig.isGlobalOrderingEnabled());
+        try {
+        	topicConfig.setMultiThreadingEnabled(true);
+        	assertTrue("multi-threading must be disabled when global-ordering is enabled", false);
+        } catch (IllegalArgumentException e) {
+        	// anticipated..
+        }
+        assertFalse(topicConfig.isMultiThreadingEnabled());
     }
\ No newline at end of file
