diff --git a/old_test.java b/new_test.java
index d8bb607..98f9592 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,12 @@
 @Test
   public void testGetRecognizeStatus() {
-    SpeechSession session = service.createSession(SpeechModel.EN_BROADBAND16K);
-    SessionStatus status = service.getRecognizeStatus(session);
+    SpeechSession session = service.createSession(SpeechModel.EN_BROADBAND16K).execute();
+    SessionStatus status = service.getRecognizeStatus(session).execute();
     try {
       assertNotNull(status);
       assertNotNull(status.getModel());
       assertNotNull(status.getState());
     } finally {
-      service.deleteSession(session);
+      service.deleteSession(session).execute();
     }
   }
\ No newline at end of file
