diff --git a/old_product.java b/new_product.java
index 86fd074..d8ce666 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,7 @@
-public SessionStatus getRecognizeStatus(final SpeechSession session) {
+public ServiceCall<SessionStatus> getRecognizeStatus(final SpeechSession session) {
     if (session == null)
-      throw new IllegalArgumentException("session was not specified");
+      throw new IllegalArgumentException("Session was not specified");
 
-    final Request request =
-        RequestBuilder.get(String.format(PATH_SESSION_RECOGNIZE, session.getSessionId())).build();
-    final Response response = execute(request);
-    final JsonObject jsonObject = ResponseUtil.getJsonObject(response);
-    return GsonSingleton.getGsonWithoutPrettyPrinting().fromJson(jsonObject.get(SESSION), SessionStatus.class);
+    final okhttp3.Request request = RequestBuilder.get(String.format(PATH_SESSION_RECOGNIZE, session.getSessionId())).build3();
+    return createServiceCall(createCall(request), ResponseUtil.getObjectConverter(SessionStatus.class));
   }
\ No newline at end of file
