diff --git a/old_product.java b/new_product.java
index 2f4077b..59aa7e4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
-@Override
-    public Object clone() {
+public Object clone() {
         try {
-            CopyOnWriteArrayList thisClone = (CopyOnWriteArrayList) super.clone();
-            thisClone.setData(this.getData());
-            return thisClone;
+            CopyOnWriteArrayList c = (CopyOnWriteArrayList)(super.clone());
+            c.resetLock();
+            return c;
         } catch (CloneNotSupportedException e) {
-            throw new RuntimeException("CloneNotSupportedException is not expected here");
+            // this shouldn't happen, since we are Cloneable
+            throw new InternalError();
         }
     }
\ No newline at end of file
