diff --git a/old_test.java b/new_test.java
index 4f302d1..e71e274 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,12 +10,9 @@
 		m1.setPublicKeyAndSign(pair1);
 
 		Map<Number640, Data> dataMap = new HashMap<Number640, Data>();
-		dataMap.put(new Number640(rnd), new Data(new byte[] { 3, 4, 5 }, true,
-				true));
-		dataMap.put(new Number640(rnd), new Data(new byte[] { 4, 5, 6 }, true,
-				true));
-		dataMap.put(new Number640(rnd), new Data(new byte[] { 5, 6, 7 }, true,
-				true));
+		dataMap.put(new Number640(rnd), new Data(new byte[] { 3, 4, 5 }));
+		dataMap.put(new Number640(rnd), new Data(new byte[] { 4, 5, 6, 7 }));
+		dataMap.put(new Number640(rnd), new Data(new byte[] { 5, 6, 7, 8, 9 }));
 		m1.setDataMap(new DataMap(dataMap));
 		NavigableMap<Number640, Number160> keysMap = new TreeMap<Number640, Number160>();
 		keysMap.put(new Number640(rnd), new Number160(rnd));
@@ -28,5 +25,6 @@
 		Assert.assertEquals(true, m2.getPublicKey() != null);
 		Assert.assertEquals(false, m2.getDataMap(0) == null);
 		Assert.assertEquals(false, m2.getKeyMap640(0) == null);
+		Assert.assertEquals(true, m2.verified());
 		compareMessage(m1, m2);
 	}
\ No newline at end of file
