diff --git a/old_product.java b/new_product.java
index e336b52..c1fa199 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,6 +13,7 @@ private Message encodeDecode(final Message m1) throws Exception {
 					}
 				});
 		
+		buf.retain();
 		ChannelHandlerContext ctx = mockChannelHandlerContext(buf, m2);
 		encoder.write(ctx, m1, null);
 		Decoder decoder = new Decoder(new DefaultSignatureFactory());
