diff --git a/old_product.java b/new_product.java
index 324633c..8167d9e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,9 +2,16 @@ public ServiceCall<Void> addWord(AddWordOptions addWordOptions) {
     Validator.notNull(addWordOptions, "addWordOptions cannot be null");
     RequestBuilder builder = RequestBuilder.put(String.format("/v1/customizations/%s/words/%s", addWordOptions
         .customizationId(), addWordOptions.wordName()));
-    builder.header("Content-Type", addWordOptions.contentType());
-    if (addWordOptions.contentType().equalsIgnoreCase(AddWordOptions.ContentType.APPLICATION_JSON)) {
-      builder.bodyJson(GsonSingleton.getGson().toJsonTree(addWordOptions.customWord()).getAsJsonObject());
+    final JsonObject contentJson = new JsonObject();
+    if (addWordOptions.word() != null) {
+      contentJson.addProperty("word", addWordOptions.word());
     }
+    if (addWordOptions.soundsLike() != null) {
+      contentJson.add("sounds_like", GsonSingleton.getGson().toJsonTree(addWordOptions.soundsLike()));
+    }
+    if (addWordOptions.displayAs() != null) {
+      contentJson.addProperty("display_as", addWordOptions.displayAs());
+    }
+    builder.bodyJson(contentJson);
     return createServiceCall(builder.build(), ResponseConverterUtils.getVoid());
   }
\ No newline at end of file
