diff --git a/old_test.java b/new_test.java
index 0851c11..11b45cb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,11 +7,9 @@
     AddWordOptions addOptions = new AddWordOptions.Builder()
         .wordName(newWord.getWord())
         .customizationId(id)
-        .customWord(new CustomWord.Builder()
-            .word(newWord.getWord())
-            .displayAs(newWord.getDisplayAs())
-            .soundsLike(newWord.getSoundsLike())
-            .build())
+        .word(newWord.getWord())
+        .displayAs(newWord.getDisplayAs())
+        .soundsLike(newWord.getSoundsLike())
         .build();
     service.addWord(addOptions).execute();
     final RecordedRequest request = server.takeRequest();
