diff --git a/old_product.java b/new_product.java
index 2a4d4ec..9434e32 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
-@RequestMapping("/save")
+@ResponseBody
+	@RequestMapping("/api/save")
 	@PreAuthorize("hasAnyRole('A') or #user.id == #updatedUser.id")
-	public String save(User user, @ModelAttribute User updatedUser) {
+	public String save(User user, @RequestBody User updatedUser) {
 		checkArgument(updatedUser.validate());
 		if (user.getRole() == Role.USER) {
 			// General user can not change their role.
@@ -10,12 +11,8 @@
 
 			// prevent user to modify with other user id
 			checkArgument(updatedUserInDb.getId().equals(updatedUser.getId()), "Illegal request to update user:%s",
-					updatedUser);
+				updatedUser);
 		}
 		save(updatedUser);
-		if (user.getId().equals(updatedUser.getId())) {
-			return "redirect:/";
-		} else {
-			return "redirect:/user/";
-		}
+		return returnSuccess();
 	}
\ No newline at end of file
