diff --git a/old_test.java b/new_test.java
index 7912e2b..8c15044 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,8 +3,10 @@
 
         ExtendedMetadata extendedMetadata;
 
-        generator.setEntityAlias("testAlias");
+        generator.setExtendedMetadata(new ExtendedMetadata());
         generator.setEntityBaseURL("http://localhost:8080");
+        generator.getExtendedMetadata().setAlias("testAlias");
+        generator.getExtendedMetadata().setIdpDiscoveryEnabled(true);
 
         // Default generation
         extendedMetadata = generator.generateExtendedMetadata();
@@ -15,7 +17,7 @@
         assertEquals("http://localhost:8080/saml/login/alias/testAlias?disco=true", extendedMetadata.getIdpDiscoveryResponseURL());
 
         // Disabled discovery
-        generator.setIncludeDiscovery(false);
+        generator.getExtendedMetadata().setIdpDiscoveryEnabled(false);
         generator.setIncludeDiscoveryExtension(false);
         extendedMetadata = generator.generateExtendedMetadata();
         assertFalse(extendedMetadata.isIdpDiscoveryEnabled());
@@ -23,15 +25,15 @@
         assertNull(extendedMetadata.getIdpDiscoveryResponseURL());
 
         // Default extended metadata
-        ExtendedMetadata defaultMetadata = new ExtendedMetadata();
-        defaultMetadata.setRequireLogoutResponseSigned(true);
-        generator.setExtendedMetadata(defaultMetadata);
-        generator.setIncludeDiscovery(true);
+        generator.setExtendedMetadata(new ExtendedMetadata());
+        generator.getExtendedMetadata().setRequireLogoutResponseSigned(true);
+        generator.getExtendedMetadata().setIdpDiscoveryEnabled(true);
+        generator.getExtendedMetadata().setIdpDiscoveryResponseURL("http://testDisco.com/response");
         generator.setIncludeDiscoveryExtension(true);
-        generator.setCustomDiscoveryResponseURL("http://testDisco.com/response");
+
         extendedMetadata = generator.generateExtendedMetadata();
         assertTrue(extendedMetadata.isIdpDiscoveryEnabled());
-        assertEquals("http://localhost:8080/saml/discovery/alias/testAlias", extendedMetadata.getIdpDiscoveryURL());
+        assertEquals("http://localhost:8080/saml/discovery", extendedMetadata.getIdpDiscoveryURL());
         assertEquals("http://testDisco.com/response", extendedMetadata.getIdpDiscoveryResponseURL());
         assertTrue(extendedMetadata.isRequireLogoutResponseSigned());
 
