diff --git a/old_product.java b/new_product.java
index 2483e87..dfe32f0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,8 +11,6 @@ public ExtendedMetadata generateExtendedMetadata() {
         String entityBaseURL = getEntityBaseURL();
         String entityAlias = getEntityAlias();
 
-        metadata.setIdpDiscoveryEnabled(isIncludeDiscovery());
-
         if (isIncludeDiscovery()) {
             metadata.setIdpDiscoveryURL(getDiscoveryURL(entityBaseURL, entityAlias));
             metadata.setIdpDiscoveryResponseURL(getDiscoveryResponseURL(entityBaseURL, entityAlias));
@@ -21,11 +19,6 @@ public ExtendedMetadata generateExtendedMetadata() {
             metadata.setIdpDiscoveryResponseURL(null);
         }
 
-        metadata.setSignMetadata(signMetadata);
-        metadata.setEncryptionKey(encryptionKey);
-        metadata.setSigningKey(signingKey);
-        metadata.setAlias(entityAlias);
-        metadata.setTlsKey(tlsKey);
         metadata.setLocal(true);
 
         return metadata;
