diff --git a/old_test.java b/new_test.java
index 72b6dd2..2d23051 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,11 @@
 @Test
   public void testBuild() {
     final String xToken = X_TOKEN;
-    final Request request =
-        RequestBuilder.post(urlWithQuery)
-          .bodyContent("body1", HttpMediaType.TEXT_PLAIN).header(X_TOKEN, "token1").build();
+    final RequestBuilder builder =  RequestBuilder.post(urlWithQuery)
+      .bodyContent("body1", HttpMediaType.TEXT_PLAIN).header(X_TOKEN, "token1");
+    final Request request = builder.build();
 
     assertEquals("POST", request.method());
     assertEquals("token1", request.header(xToken));
+    assertNotNull(builder.toString());
   }
\ No newline at end of file
