diff --git a/old_test.java b/new_test.java
index 9a825e6..9e8d2f5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
 	public void getPatientIdentifiers_shouldNotGetVoidedPatientIdentifiers() {
 		
-		List<PatientIdentifier> patientIdentifiers = dao.getPatientIdentifiers(null, new ArrayList<PatientIdentifierType>(),
-		    new ArrayList<Location>(), new ArrayList<Patient>(), null);
+		List<PatientIdentifier> patientIdentifiers = dao.getPatientIdentifiers(null, new ArrayList<>(),
+				new ArrayList<>(), new ArrayList<>(), null);
 		
 		// standartTestDataset.xml contains 5 non-voided identifiers
 		//
