diff --git a/old_product.java b/new_product.java
index 2c01e09..d88c19d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public List<PatientIdentifier> getPatientIdentifiers(PatientIdentifierType pit) {
-		List<PatientIdentifier> ids = new ArrayList<PatientIdentifier>();
+		List<PatientIdentifier> ids = new ArrayList<>();
 		for (PatientIdentifier pi : getIdentifiers()) {
 			if (!pi.getVoided() && pit.equals(pi.getIdentifierType())) {
 				ids.add(pi);
