diff --git a/old_product.java b/new_product.java
index 896227f..15bc971 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,17 +2,23 @@ public boolean matchesAtLeastThreeLevels(DependencyVersion version) {
         if (version == null) {
             return false;
         }
+        if (Math.abs(this.versionParts.size() - version.versionParts.size()) >= 3) {
+            return false;
+        }
 
-        boolean ret = true;
-        int max = (this.versionParts.size() < version.versionParts.size())
+        final int max = (this.versionParts.size() < version.versionParts.size())
                 ? this.versionParts.size() : version.versionParts.size();
 
-        if (max > 3) {
-            max = 3;
-        }
-
+        boolean ret = true;
         for (int i = 0; i < max; i++) {
-            if (this.versionParts.get(i) == null || !this.versionParts.get(i).equals(version.versionParts.get(i))) {
+            String thisVersion = this.versionParts.get(i);
+            String otherVersion = version.getVersionParts().get(i);
+            if (i >= 3) {
+                if (thisVersion.compareToIgnoreCase(otherVersion) >= 0) {
+                    ret = false;
+                    break;
+                }
+            } else if (!thisVersion.equals(otherVersion)) {
                 ret = false;
                 break;
             }
