diff --git a/old_test.java b/new_test.java
index 4d3d7c5..5a08619 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,5 @@
 @Test
-    public void testGetDocumentProcessingState() {
-        Map<String,Object> params = new HashMap<String, Object>();
-        params.put(ConceptInsights.ACCOUNT_ID, "public");
-        params.put(ConceptInsights.CORPUS, "ibmresearcher");
-        params.put(ConceptInsights.DOCUMENT, "il-AHARONA");
-        DocumentProcessingStatus documentProcessingState = service.getDocumentProcessingState("public", "ibmresearcher", "il-AHARONA");
-        Assert.assertNotNull(documentProcessingState);
-    }
\ No newline at end of file
+	public void testGetDocumentProcessingState() {
+		DocumentProcessingStatus documentProcessingState = service.getDocumentProcessingState(EXAMPLE_DOCUMENT);
+		Assert.assertNotNull(documentProcessingState);
+	}
\ No newline at end of file
