diff --git a/old_product.java b/new_product.java
index 4200474..bfbdad4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public DocumentProcessingStatus getDocumentProcessingState(final Document document, final String corpusName) {
-        ConceptInsightsId.validateGenarate(document, getAccountId(), corpusName);
-        return executeRequest(version + document.getId() + PROCESSING_STATE_PATH, null, DocumentProcessingStatus.class);
-    }
\ No newline at end of file
+public DocumentProcessingStatus getDocumentProcessingState(final Document document) {
+		String documentId = IDValidator.getDocumentId(document);
+		return executeRequest(API_VERSION + documentId + PROCESSING_STATE_PATH, null, DocumentProcessingStatus.class);
+	}
\ No newline at end of file
