diff --git a/old_test.java b/new_test.java
index 5dcfe3a..f01a7d9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,8 @@
 @Test
 	@Verifies(value = "should fail validation if a preferred patient identifier is not chosen for voided patients", method = "validate(Object,Errors)")
 	public void validate_shouldFailValidationIfAPreferredPatientIdentifierIsNotChosenForVoidedPatients() throws Exception {
-		Patient pa = Context.getPatientService().getPatient(999);
+		Patient pa = Context.getPatientService().getPatient(432);
+		
 		Assert.assertTrue(pa.isVoided());//sanity check
 		Assert.assertNotNull(pa.getPatientIdentifier());
 		for (PatientIdentifier id : pa.getIdentifiers())
