diff --git a/old_product.java b/new_product.java
index c53a76a..a4bf86f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -18,5 +18,8 @@ public ServiceCall<EntityCollection> listEntities(ListEntitiesOptions listEntiti
     if (listEntitiesOptions.cursor() != null) {
       builder.query("cursor", listEntitiesOptions.cursor());
     }
+    if (listEntitiesOptions.includeAudit() != null) {
+      builder.query("include_audit", String.valueOf(listEntitiesOptions.includeAudit()));
+    }
     return createServiceCall(builder.build(), ResponseConverterUtils.getObject(EntityCollection.class));
   }
\ No newline at end of file
