diff --git a/old_product.java b/new_product.java
index dc891f1..e88947f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@
     if (location.equals(BlockStoreLocation.anyTier())) {
       // When any tier is ok, loop over all tier views and dir views,
       // and return a temp block meta from the first available dirview.
-      for (StorageTierView tierView : mManagerView.getTierViews()) {
+      for (StorageTierView tierView : mMetadataView.getTierViews()) {
         for (StorageDirView dirView : tierView.getDirViews()) {
           if (dirView.getAvailableBytes() >= blockSize) {
             return dirView;
@@ -18,7 +18,7 @@
     String mediumType = location.mediumType();
     if (!mediumType.equals(BlockStoreLocation.ANY_MEDIUM)
         && location.equals(BlockStoreLocation.anyDirInTierWithMedium(mediumType))) {
-      for (StorageTierView tierView : mManagerView.getTierViews()) {
+      for (StorageTierView tierView : mMetadataView.getTierViews()) {
         for (StorageDirView dirView : tierView.getDirViews()) {
           if (dirView.getMediumType().equals(mediumType)
               && dirView.getAvailableBytes() >= blockSize) {
@@ -30,7 +30,7 @@
     }
 
     String tierAlias = location.tierAlias();
-    StorageTierView tierView = mManagerView.getTierView(tierAlias);
+    StorageTierView tierView = mMetadataView.getTierView(tierAlias);
     if (location.equals(BlockStoreLocation.anyDirInTier(tierAlias))) {
       // Loop over all dir views in the given tier
       for (StorageDirView dirView : tierView.getDirViews()) {
