diff --git a/old_product.java b/new_product.java
index 75091dd..787f775 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,4 @@
-public static String replace(CharSequence content,String target,Object replacement){
-        if (null == content){
-            return StringUtils.EMPTY;
-        }
-        // 替换序列是null
-        String useReplacement = null == replacement ? StringUtils.EMPTY : replacement.toString();
-        return content.toString().replace(target, useReplacement);
+public static String replace(CharSequence content,CharSequence target,CharSequence replacement){
+        return null == content ? StringUtils.EMPTY
+                        : content.toString().replace(target, null == replacement ? StringUtils.EMPTY : replacement);
     }
\ No newline at end of file
