diff --git a/old_test.java b/new_test.java
index 59a5339..562495e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,4 @@
 @Test
     public void replace(){
-        String target = "/";
-        Object replacement = "_";
-        LOGGER.info(StringUtil.replace("黑色/黄色/蓝色", target, replacement));
+        assertEquals("黑色_黄色_蓝色", StringUtil.replace("黑色/黄色/蓝色", "/", "_"));
     }
\ No newline at end of file
