diff --git a/old_test.java b/new_test.java
index b63cdba..9427bc2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -22,7 +22,7 @@
                         "23.1.1"
                 )
         );
-        final List<Command> newCommands = new ArrayList<Command>();
+        final List<Command> newCommands = new ArrayList<>();
         newCommands.add(command1);
         newCommands.add(command2);
         Assert.assertEquals(
