diff --git a/old_product.java b/new_product.java
index adabfb7..0896b19 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,7 +15,7 @@
         }
         final Cluster cluster = this.clusterRepo.findOne(id);
         if (cluster != null) {
-            final List<Command> cmds = new ArrayList<Command>();
+            final List<Command> cmds = new ArrayList<>();
             for (final Command detached : commands) {
                 final Command cmd = this.commandRepo.findOne(detached.getId());
                 if (cmd != null) {
