diff --git a/old_test.java b/new_test.java
index 21904ea..313d4d3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     @Tag("AzureDWTest")
     public void createTableTest() throws SQLException {
-        try (Connection con = DriverManager.getConnection(connectionString);
+        try (Connection con = getConnection();
                 PreparedStatement pstmt1 = con.prepareStatement(
                         "create table " + AbstractSQLGenerator.escapeIdentifier(tableName) + " (col1 int)");
                 PreparedStatement pstmt2 = con
