diff --git a/old_test.java b/new_test.java
index c390ee4..782698d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,35 +1,14 @@
 @Test
-  public void testUploadPackage() throws Exception {
-    // Insert mock Controller
+  public void testUploadPackage() {
     ScpUploader uploader = Mockito.spy(new ScpUploader());
+    Mockito.doReturn(true).when(uploader).isLocalFileExists(Mockito.anyString());
     ScpController controller = Mockito.mock(ScpController.class);
     Mockito.doReturn(controller).when(uploader).getScpController();
-    uploader.initialize(config);
-
-    // Local file not exist
-    Mockito.doReturn(false).when(uploader).isLocalFileExists(Mockito.anyString());
-    Assert.assertNull(uploader.uploadPackage());
-    Mockito.verify(controller, Mockito.never()).copyFromLocalFile(
-        Mockito.anyString(), Mockito.anyString());
-
-    // Failed to create folder on remote
-    Mockito.doReturn(true).when(uploader).isLocalFileExists(Mockito.anyString());
-    Mockito.doReturn(false).when(controller).mkdirsIfNotExists(Mockito.anyString());
-    Assert.assertNull(uploader.uploadPackage());
-    Mockito.verify(controller, Mockito.never()).copyFromLocalFile(
-        Mockito.anyString(), Mockito.anyString());
-
-    // Failed to copy file from local to remote
     Mockito.doReturn(true).when(controller).mkdirsIfNotExists(Mockito.anyString());
-    Mockito.doReturn(false).when(controller).copyFromLocalFile(
-        Mockito.anyString(), Mockito.anyString());
-    Assert.assertNull(uploader.uploadPackage());
-    Mockito.verify(controller).copyFromLocalFile(Mockito.anyString(), Mockito.anyString());
-
-    // Happy path
+    uploader.initialize(config);
     Mockito.doReturn(true).when(controller).copyFromLocalFile(
         Mockito.anyString(), Mockito.anyString());
-    Assert.assertNotNull(uploader.uploadPackage());
+    uploader.uploadPackage();
     Mockito.verify(controller, Mockito.atLeastOnce()).copyFromLocalFile(
         Mockito.anyString(), Mockito.anyString());
   }
\ No newline at end of file
