diff --git a/old_product.java b/new_product.java
index 56000c0..6c5d934 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,5 @@
 @PrePersist
     @PreUpdate
     protected void onCreateOrUpdateCommand() throws GeniePreconditionException {
-        // Add the id to the tags
-        if (this.tags == null) {
-           this.tags = new HashSet<>();
-        }
         this.addAndValidateSystemTags(this.tags);
     }
\ No newline at end of file
