diff --git a/old_test.java b/new_test.java
index 4ab511d..ba3aec7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,8 @@
 @Test
     public void testOnCreateOrUpdateCommand() throws GeniePreconditionException {
         this.c = new Command(NAME, USER, VERSION, CommandStatus.ACTIVE, EXECUTABLE);
-        Assert.assertNull(this.c.getTags());
+        Assert.assertNotNull(this.c.getTags());
+        Assert.assertTrue(this.c.getTags().isEmpty());
         this.c.onCreateOrUpdateCommand();
         Assert.assertEquals(2, this.c.getTags().size());
     }
\ No newline at end of file
