diff --git a/old_product.java b/new_product.java
index e65067c..7f4f1b6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,5 @@
-private static void createTable() throws SQLException {
-        try (Connection connection = getConnection(); Statement stmt = connection.createStatement()) {
-            String sql = "create table " + AbstractSQLGenerator.escapeIdentifier(tableName)
-                    + " ( col1 int, col2 varchar(50), col3 varchar(10), col4 int)";
-            stmt.execute(sql);
-        }
+private static void createTable(Statement stmt) throws SQLException {
+        String sql = "create table " + AbstractSQLGenerator.escapeIdentifier(tableName)
+                + " ( col1 int, col2 varchar(50), col3 varchar(10), col4 int)";
+        stmt.execute(sql);
     }
\ No newline at end of file
