diff --git a/old_test.java b/new_test.java
index 313d4d3..b747e27 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,8 @@
 @Test
-    @Tag("AzureDWTest")
     public void createTableTest() throws SQLException {
-        try (Connection con = getConnection();
-                PreparedStatement pstmt1 = con.prepareStatement(
-                        "create table " + AbstractSQLGenerator.escapeIdentifier(tableName) + " (col1 int)");
-                PreparedStatement pstmt2 = con
+        try (PreparedStatement pstmt1 = connection
+                .prepareStatement("create table " + AbstractSQLGenerator.escapeIdentifier(tableName) + " (col1 int)");
+                PreparedStatement pstmt2 = connection
                         .prepareStatement("drop table " + AbstractSQLGenerator.escapeIdentifier(tableName))) {
             pstmt1.execute();
             pstmt2.execute();
