diff --git a/old_product.java b/new_product.java
index 44b92da..236a1b0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
 protected boolean requestSchedulerService(Command command, byte[] data) {
-    final HttpURLConnection connection = createHttpConnection(command);
+    String endpoint = getCommandEndpoint(schedulerHttpEndpoint, command);
+    final HttpURLConnection connection = NetworkUtils.getHttpConnection(endpoint);
     if (connection == null) {
       LOG.severe("Scheduler not found.");
       return false;
@@ -8,7 +9,7 @@ protected boolean requestSchedulerService(Command command, byte[] data) {
     // now, we have a valid connection
     try {
       // send the actual http request
-      if (!NetworkUtils.sendHttpPostRequest(connection, data)) {
+      if (!NetworkUtils.sendHttpPostRequest(connection, NetworkUtils.URL_ENCODE_TYPE, data)) {
         LOG.log(Level.SEVERE, "Failed to send http request to scheduler");
         return false;
       }
