diff --git a/old_test.java b/new_test.java
index 29a0a91..0699366 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,19 +4,23 @@
         Mockito.spy(new HttpServiceSchedulerClient(config, runtime, SCHEDULER_HTTP_ENDPOINT));
 
     // Failed to create new http connection
-    Mockito.doReturn(null).when(client).createHttpConnection(Mockito.any(Command.class));
+    PowerMockito.spy(NetworkUtils.class);
+    PowerMockito.doReturn(null).
+        when(NetworkUtils.class, "getHttpConnection", Mockito.any(String.class));
+    Mockito.doReturn(COMMAND_ENDPOINT).
+        when(client).getCommandEndpoint(Mockito.any(String.class), Mockito.any(Command.class));
     Assert.assertFalse(
         client.requestSchedulerService(
             Mockito.any(Command.class), Mockito.any(byte[].class)));
 
     HttpURLConnection connection = Mockito.mock(HttpURLConnection.class);
-    Mockito.doReturn(connection).when(client).createHttpConnection(Mockito.any(Command.class));
+    PowerMockito.doReturn(connection).
+        when(NetworkUtils.class, "getHttpConnection", Mockito.any(String.class));
 
     // Failed to send http post request
-    PowerMockito.spy(NetworkUtils.class);
     PowerMockito.doReturn(false).
         when(NetworkUtils.class, "sendHttpPostRequest",
-            Mockito.eq(connection), Mockito.any(byte[].class));
+            Mockito.eq(connection), Mockito.any(String.class), Mockito.any(byte[].class));
     Assert.assertFalse(
         client.requestSchedulerService(
             Mockito.any(Command.class), Mockito.any(byte[].class)));
@@ -25,7 +29,7 @@
     // Received non-ok response
     PowerMockito.doReturn(true).
         when(NetworkUtils.class, "sendHttpPostRequest",
-            Mockito.eq(connection), Mockito.any(byte[].class));
+            Mockito.eq(connection), Mockito.any(String.class), Mockito.any(byte[].class));
     Scheduler.SchedulerResponse notOKResponse = SchedulerUtils.constructSchedulerResponse(false);
     PowerMockito.doReturn(notOKResponse.toByteArray()).
         when(NetworkUtils.class, "readHttpResponse", Mockito.eq(connection));
