diff --git a/old_test.java b/new_test.java
index 019f01a..adceaeb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -24,6 +24,6 @@
         .containsExactlyInAnyOrder(context3, context4);
 
     // We also expect the spans to have been reported
-    assertThat(spans).flatExtracting(s -> s.annotations).extracting(a -> a.value)
+    assertThat(spans).flatExtracting(Span::annotations).extracting(Annotation::value)
         .containsExactly("brave.flush", "brave.flush");
   }
\ No newline at end of file
