diff --git a/old_product.java b/new_product.java
index 7b99e58..3609511 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,8 +8,8 @@
         }
 
         // start the child initializers
-        ExecutorService exec = getActiveExecutor();
-        for (BackgroundInitializer<?> bi : inits.values()) {
+        final ExecutorService exec = getActiveExecutor();
+        for (final BackgroundInitializer<?> bi : inits.values()) {
             if (bi.getExternalExecutor() == null) {
                 // share the executor service if necessary
                 bi.setExternalExecutor(exec);
@@ -18,12 +18,12 @@
         }
 
         // collect the results
-        Map<String, Object> results = new HashMap<String, Object>();
-        Map<String, ConcurrentException> excepts = new HashMap<String, ConcurrentException>();
-        for (Map.Entry<String, BackgroundInitializer<?>> e : inits.entrySet()) {
+        final Map<String, Object> results = new HashMap<String, Object>();
+        final Map<String, ConcurrentException> excepts = new HashMap<String, ConcurrentException>();
+        for (final Map.Entry<String, BackgroundInitializer<?>> e : inits.entrySet()) {
             try {
                 results.put(e.getKey(), e.getValue().get());
-            } catch (ConcurrentException cex) {
+            } catch (final ConcurrentException cex) {
                 excepts.put(e.getKey(), cex);
             }
         }
