diff --git a/old_test.java b/new_test.java
index 1a9c6dc..61b2765 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testInitialize() {
-        BackgroundInitializerTestImpl init = new BackgroundInitializerTestImpl();
+        final BackgroundInitializerTestImpl init = new BackgroundInitializerTestImpl();
         init.start();
         checkInitialize(init);
     }
\ No newline at end of file
