diff --git a/old_product.java b/new_product.java
index bfb02bd..118d74a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public void compute(Connections c, int[] inputVector, boolean learn, int[] activeArray) {
+public void compute(Connections c, int[] inputVector, int[] activeArray, boolean learn) {
         if(inputVector.length != c.getNumInputs()) {
             throw new IllegalArgumentException("Input array must be same size as the defined number of inputs");
         }
