diff --git a/old_test.java b/new_test.java
index dcbe0c9..10028ea 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testNotEq() {
-        assertTrue(instance.notEq("name", "abc").getQueryCriterions()
-                .contains(new NotEqCriterion("name", "abc")));
+    	assertEquals(new NotEqCriterion("name", "abc"), instance.notEq("name", "abc").getQueryCriterion());
     }
\ No newline at end of file
