diff --git a/old_test.java b/new_test.java
index 8516f79..4460536 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,6 @@
         double p = 0.5;
         int n = 10;
         double expResult = 0.17187500002003;
-        double result = DiscreteDistributions.BinomialCdf(k, p, n);
+        double result = DiscreteDistributions.binomialCdf(k, p, n);
         assertEquals(expResult, result, TestConfiguration.DOUBLE_ACCURACY_HIGH);
     }
\ No newline at end of file
