diff --git a/old_product.java b/new_product.java
index ecce7f1..9f9b558 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public final void bug15375(int num) throws Exception {
+public final void bug15375(int num) throws IOException {
         Workbook wb1 = _testDataProvider.createWorkbook();
         Sheet sheet = wb1.createSheet();
         CreationHelper factory = wb1.getCreationHelper();
