diff --git a/old_product.java b/new_product.java
index f79a9fa..71465eb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,4 @@
-private Str pathToNative(final QueryContext ctx) throws QueryException {
-    final File path = checkFile(0, ctx);
-    try {
-      final String nat = path.getCanonicalFile().getPath();
-      return Str.get(path.isDirectory() ? dir(nat) : nat);
-    } catch(final IOException ex) {
-      throw FILE_IE_PATH.get(info, path);
-    }
+private Str pathToNative(final QueryContext ctx) throws QueryException, IOException {
+    final Path nat = checkPath(0, ctx).toRealPath();
+    return get(nat, Files.isDirectory(nat));
   }
\ No newline at end of file
