diff --git a/old_test.java b/new_test.java
index 87945d2..bec81d4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,8 @@
 @Test
   public void pathToNative() throws IOException {
-    assertEquals(norm(new File(PATH1).getCanonicalPath()),
-        norm(query(_FILE_PATH_TO_NATIVE.args(PATH1))));
-    query(_FILE_PATH_TO_NATIVE.args(PATH + ".." + "/test.xml"),
-        new File(PATH + ".." + "/test.xml").getCanonicalPath());
+    query(_FILE_WRITE.args(PATH1, "()"));
+    assertEquals(Paths.get(PATH1).toRealPath().toString(), query(_FILE_PATH_TO_NATIVE.args(PATH1)));
+    query(_FILE_PATH_TO_NATIVE.args(PATH + "../" + NAME + '/' + NAME),
+        Paths.get(PATH + "../" + NAME + '/' + NAME).toRealPath().toString());
+    error(_FILE_PATH_TO_NATIVE.args(PATH1 + NAME), Err.FILE_NOT_FOUND);
   }
\ No newline at end of file
