diff --git a/old_product.java b/new_product.java
index bb77cd9..0b76d45 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
 private Iter text(final QueryContext ctx) throws QueryException {
-    final Data data = ctx.resource.data(checkStr(expr[0], ctx), input);
-    final IndexContext ic = new IndexContext(ctx, data, null, true);
+    final IndexContext ic = new IndexContext(ctx, data(0, ctx), null, true);
     return new IndexAccess(input, expr[1], IndexType.TEXT, ic).iter(ctx);
   }
\ No newline at end of file
