diff --git a/old_product.java b/new_product.java
index 2624b13..08c9094 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,7 +9,7 @@ public void process( T gray ) {
 		found.reset();
 
 		if( verbose ) System.out.println("---------- Got Polygons! "+candidates.size);
-		Quadrilateral_F64 q = new Quadrilateral_F64(); // todo predeclare
+
 		for (int i = 0; i < candidates.size; i++) {
 			// compute the homography from the input image to an undistorted square image
 			Polygon2D_F64 p = candidates.get(i);
