diff --git a/old_test.java b/new_test.java
index 2091513..e588c9d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,13 @@
 @Test
-  public void testCheckRegistrarConsoleLogin_authedWithValidSession_doesNothing() throws Exception {
-    when(session.getAttribute("clientId")).thenReturn("TheRegistrar");
-    assertThat(sessionUtils.checkRegistrarConsoleLogin(req, jart)).isTrue();
-    verify(session).getAttribute("clientId");
-    verifyNoMoreInteractions(session);
+  public void testCheckRegistrarConsoleLogin_noSession_noAccess_isAdmin() throws Exception {
+    assertThat(sessionUtils.checkRegistrarConsoleLogin(req, UNAUTHORIZED_ADMIN)).isTrue();
+    verify(session).setAttribute(eq("clientId"), eq(ADMIN_CLIENT_ID));
+    assertAboutLogs()
+        .that(testLogHandler)
+        .hasLogAtLevelWithMessage(
+            Level.INFO,
+            String.format(
+                "User %s is an admin with no associated registrar."
+                    + " Automatically associating the user with configured client Id %s.",
+                UNAUTHORIZED_ADMIN.user().getUserId(), ADMIN_CLIENT_ID));
   }
\ No newline at end of file
