diff --git a/old_test.java b/new_test.java
index 95c5a0e..41a79e0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,18 +1,25 @@
 @Test
     public void testCreateApplication() throws GenieException {
         final String id = UUID.randomUUID().toString();
-        final Application app = new Application
-            .Builder(APP_1_NAME, APP_1_USER, APP_1_VERSION, ApplicationStatus.ACTIVE)
-            .withId(id)
+        final ApplicationRequest app = new ApplicationRequest.Builder(
+            new ApplicationMetadata.Builder(
+                APP_1_NAME,
+                APP_1_USER,
+                ApplicationStatus.ACTIVE
+            )
+                .withVersion(APP_1_VERSION)
+                .build()
+        )
+            .withRequestedId(id)
             .build();
         final String createdId = this.appService.createApplication(app);
         Assert.assertThat(createdId, Matchers.is(id));
         final Application created = this.appService.getApplication(id);
         Assert.assertNotNull(created);
-        Assert.assertEquals(id, created.getId().orElseThrow(IllegalArgumentException::new));
-        Assert.assertEquals(APP_1_NAME, created.getName());
-        Assert.assertEquals(APP_1_USER, created.getUser());
-        Assert.assertEquals(ApplicationStatus.ACTIVE, created.getStatus());
+        Assert.assertEquals(id, created.getId());
+        Assert.assertEquals(APP_1_NAME, created.getMetadata().getName());
+        Assert.assertEquals(APP_1_USER, created.getMetadata().getUser());
+        Assert.assertEquals(ApplicationStatus.ACTIVE, created.getMetadata().getStatus());
         this.appService.deleteApplication(id);
         try {
             this.appService.getApplication(id);
