diff --git a/old_product.java b/new_product.java
index 559c40c..c8b01e2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@
     @ResponseStatus(HttpStatus.CREATED)
     public ResponseEntity<Void> createApplication(@RequestBody final Application app) throws GenieException {
         log.debug("Called to create new application");
-        final String id = this.applicationService.createApplication(app);
+        final String id = this.applicationService.createApplication(DtoAdapters.toV4ApplicationRequest(app));
         final HttpHeaders httpHeaders = new HttpHeaders();
         httpHeaders.setLocation(
             ServletUriComponentsBuilder
