diff --git a/old_product.java b/new_product.java
index de1b0b3..480cb35 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,21 @@
 @Override
     public void go(AcceptsOneWidget container) {
 
+        if (Strings.isNullOrEmpty(dataObject.getName()) && Strings.isNullOrEmpty(dataObject.getSource().getLocation())) {
+            ignoreChanges = true;
+
+            view.setProjectUrlErrorHighlight(false);
+            view.setProjectNameErrorHighlight(false);
+            view.setURLErrorMessage(null);
+        }
+
         view.setProjectName(dataObject.getName());
         view.setProjectDescription(dataObject.getDescription());
         view.setProjectUrl(dataObject.getSource().getLocation());
 
         container.setWidget(view);
 
-        view.setUrlTextBoxFocused();
+        view.setInputsEnableState(true);
+
+        ignoreChanges = false;
     }
\ No newline at end of file
