diff --git a/old_test.java b/new_test.java
index dab7f98..19ab693 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,5 +12,5 @@
         verify(view).setProjectName(anyString());
         verify(view).setProjectDescription(anyString());
         verify(view).setProjectUrl(anyString());
-        verify(view).setUrlTextBoxFocused();
+        verify(view).setInputsEnableState(eq(true));
     }
\ No newline at end of file
