diff --git a/old_product.java b/new_product.java
index 930a12d..2acc56c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,7 @@ final byte[] transform(final QueryContext qc) throws QueryException {
     final ArrayOutput ao = new ArrayOutput();
     try {
       System.setErr(new PrintStream(ao));
-      return transform(in, xsl, opts.free(), xopts);
+      return transform(in, xsl, opts.free(), xopts, qc);
     } catch(final TransformerException ex) {
       Util.debug(ex);
       throw XSLT_ERROR_X.get(info, trim(utf8(ao.finish(), Prop.ENCODING)));
