diff --git a/old_product.java b/new_product.java
index 82c83ca..0da65df 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,6 @@
 @SafeVarargs
     public static <O, V> List<O> select(Collection<O> objectCollection,String propertyName,V...propertyValues){
-        return Validator.isNullOrEmpty(objectCollection) ? Collections.<O> emptyList()
-                        : select(objectCollection, new ArrayContainsPredicate<O>(propertyName, propertyValues));
+        return isNullOrEmpty(objectCollection) ? Collections.<O> emptyList() : select(
+                        objectCollection,
+                        BeanPredicateUtil.<O, V> containsPredicate(propertyName, propertyValues));
     }
\ No newline at end of file
