diff --git a/old_test.java b/new_test.java
index 29aa9db..0881601 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,10 @@
 @Test
     public void testSelect(){
-        List<User> list = toList(//
-                        new User("张飞", 23),
-                        new User("关羽", 24),
-                        new User("刘备", 25));
-
-        LOGGER.debug(JsonUtil.format(CollectionsUtil.select(list, "name", toList("张飞", "刘备"))));
+        User zhangfei = new User("张飞", 23);
+        User guanyu = new User("关羽", 24);
+        User liubei = new User("刘备", 25);
+        List<User> list = toList(zhangfei, guanyu, liubei);
+        assertThat(
+                        CollectionsUtil.select(list, "name", toList("张飞", "刘备")),
+                        allOf(hasItem(zhangfei), hasItem(liubei), not(hasItem(guanyu))));
     }
\ No newline at end of file
