diff --git a/old_test.java b/new_test.java
index 62ba7f0..a4b9de7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,9 +3,7 @@
         final long segmentId = 0L;
         final long baseOffset = 1000;
         @Cleanup
-        val cacheFactory = new InMemoryCacheFactory();
-        @Cleanup
-        val keyCache = new ContainerKeyCache(CONTAINER_ID, cacheFactory);
+        val keyCache = new ContainerKeyCache(this.cacheStorage);
         val expectedResult = new HashMap<TestKey, CacheBucketOffset>();
 
         // Insert some pre-existing values.
