diff --git a/old_product.java b/new_product.java
index 1cd665d..50d9839 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ void includeTailCache(long segmentId, Map<UUID, CacheBucketOffset> keyOffsets) {
         int generation;
         synchronized (this.segmentCaches) {
             generation = this.currentCacheGeneration;
-            cache = this.segmentCaches.computeIfAbsent(segmentId, s -> new SegmentKeyCache(s, this.cache));
+            cache = this.segmentCaches.computeIfAbsent(segmentId, s -> new SegmentKeyCache(s, this.cacheStorage));
         }
 
         cache.includeTailCache(keyOffsets, generation);
