diff --git a/old_test.java b/new_test.java
index 0821b0f..ddd9142 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,8 +6,8 @@
         ObjectArrayList<Object> list2 = new ObjectArrayList<Object>();
         list2.add(newArray(list2.buffer, 0, 2));
 
-        assertEquals(3, deque.removeAllIn(list2));
-        assertEquals(0, deque.removeAllIn(list2.iterator()));
+        assertEquals(3, deque.removeAll(list2));
+        assertEquals(0, deque.removeAll(list2.iterator()));
 
         assertListEquals(deque.toArray(), 1, 1);
     }
\ No newline at end of file
