diff --git a/old_test.java b/new_test.java
index 003bdf0..2c037f7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,6 +10,6 @@
     error(_BIN_DECODE_STRING.args(base64(""), "UTF-8", 1, 0),  BIN_IOOR_X_X);
     error(_BIN_DECODE_STRING.args(base64(""), "UTF-8", 0, 1),  BIN_IOOR_X_X);
     error(_BIN_DECODE_STRING.args(base64(""), "X"),            BIN_UE_X);
-    error(_BIN_DECODE_STRING.args(base64("FF"), "UTF-8"),      BIN_CE);
-    error(_BIN_DECODE_STRING.args(_BIN_HEX.args("03")),        BIN_CE);
+    error(_BIN_DECODE_STRING.args(base64("FF"), "UTF-8"),      BIN_CE_X);
+    error(_BIN_DECODE_STRING.args(_BIN_HEX.args("03")),        BIN_CE_X);
   }
\ No newline at end of file
