diff --git a/old_product.java b/new_product.java
index 398786a..493c9c1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 private Str decodeString(final QueryContext qc) throws QueryException {
-    final B64 b64 = b64(exprs[0], true, qc);
-    final String enc = checkEncoding(1, BIN_UE_X, qc);
-    final Long off = exprs.length > 2 ? checkItr(exprs[2], qc) : null;
-    final Long len = exprs.length > 3 ? checkItr(exprs[3], qc) : null;
+    final B64 b64 = toB64(exprs[0], qc, true);
+    final String enc = toEncoding(1, BIN_UE_X, qc);
+    final Long off = exprs.length > 2 ? toLong(exprs[2], qc) : null;
+    final Long len = exprs.length > 3 ? toLong(exprs[3], qc) : null;
     if(b64 == null) return null;
 
     byte[] bytes = b64.binary(info);
@@ -18,6 +18,6 @@ private Str decodeString(final QueryContext qc) throws QueryException {
     try {
       return Str.get(FNConvert.toString(new IOContent(bytes).inputStream(), enc, true));
     } catch(final IOException ex) {
-      throw BIN_CE.get(info, ex);
+      throw BIN_CE_X.get(info, ex);
     }
   }
\ No newline at end of file
