diff --git a/old_test.java b/new_test.java
index d9a30e3..b9b1000 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,5 +4,7 @@
         Index instance = new Index();
         String exp = File.separatorChar + "target" + File.separatorChar + "data" + File.separatorChar + "cve";
         Directory result = instance.getDirectory();
-        assertTrue(result.toString().contains(exp));
+        
+        assertTrue("Recieved '" + result.toString() + "' and excpected '" + exp + "'.",
+                result.toString().contains(exp));
     }
\ No newline at end of file
