diff --git a/old_product.java b/new_product.java
index e409aab..f3b0a52 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,5 @@
 public Directory getDirectory() throws IOException {
-        String fileName = Settings.getString(Settings.KEYS.CPE_INDEX);
-        String filePath = Index.class.getProtectionDomain().getCodeSource().getLocation().getPath();
-        String decodedPath = URLDecoder.decode(filePath, "UTF-8");
-
-        File path = new File(decodedPath + File.separator + fileName);
-        path = new File(path.getCanonicalPath());
+        File path = getDataDirectory();
         Directory dir = FSDirectory.open(path);
 
         return dir;
