diff --git a/old_product.java b/new_product.java
index 07d7156..0572f16 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,4 @@
-@Test
-	public void setAsText_shouldSetUsingUuid() {
-		CohortEditor editor = new CohortEditor();
-		editor.setAsText("h9a9m0i6-15e6-467c-9d4b-mbi7teu9lf0f");
-		Assert.assertNotNull(editor.getValue());
+@Override
+	protected Cohort getExistingObject() {
+		return cohortService.getCohort(EXISTING_ID);
 	}
\ No newline at end of file
