diff --git a/old_test.java b/new_test.java
index 4a9ba49..8e84c9e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
-@Test
-	public void setAsText_shouldSetUsingUuid() {
-		LocationTagEditor editor = new LocationTagEditor();
-		editor.setAsText("001e503a-47ed-11df-bc8b-001e378eb67e");
-		Assert.assertNotNull(editor.getValue());
+@Override
+	@Test
+	public void shouldReturnEmptyStringIfValueIsNull() {
+		
+		assertThat(editor.getAsText(), is(nullValue()));
 	}
\ No newline at end of file
