diff --git a/old_product.java b/new_product.java
index 2ab8f40..021bba6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
 @Override
-    public boolean containsValue(Object value) {
+    public boolean containsValue(final Object value) {
         if (value == null) {
             return containsNullValue();
         }
 
-        Entry[] tab = table;
+        final Entry[] tab = table;
         for (int i = 0; i < tab.length; i++) {
             for (Entry e = tab[i]; e != null; e = e.next) {
                 if (value.equals(e.value)) {
