diff --git a/old_test.java b/new_test.java
index 7ad0133..b962005 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testContainsValue() throws Exception {
-        List<ParameterizedHeader> values = new ArrayList<ParameterizedHeader>();
+        List<ParameterizedHeader> values = new ArrayList<>();
         values.add(new ParameterizedHeader("bar"));
         values.add(new ParameterizedHeader("bop"));
 
