diff --git a/old_product.java b/new_product.java
index ec10bd3..70b474b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -29,17 +29,17 @@ public TranslationResult translate(final Map<String, Object> params) {
     }
     contentJson.add(TEXT, paragraphs);
 
-    final RequestBuilder requestBuilder =
-        RequestBuilder.post("/v2/translate").withBodyJson(contentJson);
+    final RequestBuilder requestBuilder = RequestBuilder.post("/v2/translate");
 
     if (source != null && !source.isEmpty())
-      requestBuilder.withQuery(SOURCE, source);
+      contentJson.addProperty(SOURCE, source);
 
     if (target != null && !target.isEmpty())
-      requestBuilder.withQuery(TARGET, target);
+      contentJson.addProperty(TARGET, target);
 
     if (modelId != null && !modelId.isEmpty())
-      requestBuilder.withQuery(MODEL_ID, modelId);
+      contentJson.addProperty(MODEL_ID, modelId);
 
+    requestBuilder.withBodyJson(contentJson);
     return executeRequest(requestBuilder.build(), TranslationResult.class);
   }
\ No newline at end of file
