diff --git a/old_test.java b/new_test.java
index c8ed6dc..35059ab 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,10 +20,10 @@
       paragraphs.add(new JsonPrimitive(paragraph));
     }
     contentJson.add(LanguageTranslation.TEXT, paragraphs);
-    mockServer.when(
-        request().withMethod("POST").withPath(LANGUAGE_TRANSLATION_PATH)
-            .withQueryStringParameter(new Parameter(LanguageTranslation.MODEL_ID, modelId))
-            .withBody(contentJson.toString())
+    contentJson.addProperty(LanguageTranslation.MODEL_ID, modelId);
+    mockServer.when(request().withMethod("POST").withPath(LANGUAGE_TRANSLATION_PATH)
+
+    .withBody(contentJson.toString())
 
     ).respond(
         response().withHeaders(
