diff --git a/old_test.java b/new_test.java
index 4dfcd3d..867a82c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
     assertFalse(mTree.inodeIdExists(1));
 
     createPath(mTree, TEST_URI, sFileOptions);
-    Inode<?> inode = getInodeByPath(mTree, TEST_URI);
+    InodeView inode = getInodeByPath(mTree, TEST_URI);
     assertTrue(mTree.inodeIdExists(inode.getId()));
 
     deleteInodeByPath(mTree, TEST_URI);
