diff --git a/old_product.java b/new_product.java
index 6cd2c4a..c44425a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,6 @@
 @Override
     public void adjustTLSContext(ExtendedMasterSecretExtensionMessage message) {
-        context.setIsExtendedMasterSecretExtension(true);
+        if (context.getTalkingConnectionEnd() == ConnectionEnd.SERVER) {
+            context.setIsExtendedMasterSecretExtension(true);
+        }
     }
\ No newline at end of file
