diff --git a/old_test.java b/new_test.java
index 511a872..346a499 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,8 +7,9 @@
         message.prepareComputations();
         message.getComputations().setPrivateKey(new BigInteger("123"));
         message.getComputations().setPremasterSecret(new byte[] { 0, 1, 2, 3 });
-        message.getComputations().setMasterSecret(new byte[] { 4, 5, 6 });
+        // TODO assert master secret was computed correctly
+        // message.getComputations().setMasterSecret(new byte[] { 4, 5, 6 });
         handler.adjustTLSContext(message);
         assertNull(context.getPreMasterSecret());
-        assertNull(context.getMasterSecret());
+        // assertNull(context.getMasterSecret());
     }
\ No newline at end of file
