diff --git a/old_test.java b/new_test.java
index 9e26313..d088a6c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,13 @@
 @Test
 	public void setDescription_partial() {
-		fail("implement tests");
+		// now tell it to set a description near the edge
+		// only the first layer should be set
+		PyramidKltFeature feature = new PyramidKltFeature(pyramid.getNumLayers(),featureReadius);
+		feature.setPosition(featureReadius,featureReadius);
+		tracker.setImage(pyramid,derivX,derivY);
+		tracker.setDescription(feature);
+		assertTrue( feature.desc[0].x != 0 );
+		for( int i = 1; i < pyramid.getNumLayers(); i++ ) {
+			assertTrue( feature.desc[i].Gxx == 0.0f );
+		}
 	}
\ No newline at end of file
