diff --git a/old_product.java b/new_product.java
index 68c252e..baa4887 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @SuppressWarnings({"SuspiciousNameCombination"})
-	public void setDescription(KltFeature feature) {
+	public boolean setDescription(KltFeature feature) {
 		if (derivX == null || derivY == null)
 			throw new IllegalArgumentException("Image derivatives must be set");
 
@@ -8,7 +8,7 @@
 		int size = regionWidth * regionWidth;
 
 		if (!isFullyInside(feature.x, feature.y))
-			throw new IllegalArgumentException("Feature is too close to the image's border");
+			return false;
 
 		float tl_x = feature.x - feature.radius;
 		float tl_y = feature.y - feature.radius;
@@ -32,4 +32,6 @@
 		feature.Gxx = Gxx;
 		feature.Gyy = Gyy;
 		feature.Gxy = Gxy;
+
+		return true;
 	}
\ No newline at end of file
