diff --git a/old_product.java b/new_product.java
index 9c85a7c..8b373c6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,12 @@
 @VisibleForTesting
-	/* package */ void applyDeltaToLaterStockCandidates(
+	/* package */ void applyDeltaToMatchingLaterStockCandidates(
 			@NonNull final ProductDescriptor productDescriptor,
 			@NonNull final Integer warehouseId,
 			@NonNull final Date date,
 			@NonNull final Integer groupId,
 			@NonNull final BigDecimal delta)
 	{
-		final CandidatesQuery segment = CandidatesQuery.builder()
+		final CandidatesQuery query = CandidatesQuery.builder()
 				.type(CandidateType.STOCK)
 				.materialDescriptor(MaterialDescriptor.builderForQuery()
 						.date(date)
@@ -14,9 +14,10 @@
 						.warehouseId(warehouseId)
 						.dateOperator(DateOperator.AFTER)
 						.build())
+				.matchExactStorageAttributesKey(true)
 				.build();
 
-		final List<Candidate> candidatesToUpdate = candidateRepository.retrieveOrderedByDateAndSeqNo(segment);
+		final List<Candidate> candidatesToUpdate = candidateRepository.retrieveOrderedByDateAndSeqNo(query);
 		for (final Candidate candidate : candidatesToUpdate)
 		{
 			final BigDecimal newQty = candidate.getQuantity().add(delta);
