diff --git a/old_test.java b/new_test.java
index 9a99403..e13c9a0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-    public void testRowSum() {
-        System.out.println("rowSum");
+    public void testRowSums() {
+        System.out.println("rowSums");
         double[][] A = {
             {0.7220180, 0.07121225, 0.6881997},
             {-0.2648886, -0.89044952, 0.3700456},
@@ -8,7 +8,7 @@
         };
         double[] r = {1.4814300, -0.7852925, 0.4343743};
 
-        double[] result = Math.rowSum(A);
+        double[] result = Math.rowSums(A);
         for (int i = 0; i < r.length; i++) {
             assertEquals(result[i], r[i], 1E-7);
         }
