diff --git a/old_test.java b/new_test.java
index a16723a..4eb1382 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,4 +9,5 @@
         assertSame(cyclicCause.getCause(), ExceptionUtils.getCause(cyclicCause));
         assertSame(((ExceptionWithCause) cyclicCause.getCause()).getCause(), ExceptionUtils.getCause(cyclicCause.getCause()));
         assertSame(cyclicCause.getCause(), ExceptionUtils.getCause(((ExceptionWithCause) cyclicCause.getCause()).getCause()));
+        assertSame(withoutCause, ExceptionUtils.getCause(notVisibleException));
     }
\ No newline at end of file
