diff --git a/old_test.java b/new_test.java
index 9167633..3a6184b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public final void testStart()
     {
     Map<String, Double> context = new HashMap<String, Double>();
-    average.start( context );
+    average.start( context, null );
 
     TupleEntryCollector resultEntryCollector = new TupleEntryCollector( new Fields( "field" ) );
     average.complete( context, resultEntryCollector.iterator() );
