diff --git a/old_product.java b/new_product.java
index c56d23a..943ff1a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public int getTotalLeased(){
 		int total=0;
 		for (int i=0; i < this.partitionCount; i++){
-			total+=this.partitions[i].getCreatedConnections()-this.partitions[i].getFreeConnections().size();
+			total+=this.partitions[i].getCreatedConnections()-this.partitions[i].getAvailableConnections().get();
 		}
 		return total;
 	}
\ No newline at end of file
