diff --git a/old_test.java b/new_test.java
index 8167ba4..583ed40 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
 	public void testGetTotalLeased() {
 		expect(mockPartition.getCreatedConnections()).andReturn(5).anyTimes();
 		expect(mockPartition.getFreeConnections()).andReturn(mockConnectionHandles).anyTimes();
-		expect(mockConnectionHandles.size()).andReturn(3).anyTimes();
+		expect(mockPartition.getAvailableConnections()).andReturn(new AtomicInteger(3)).anyTimes();
 		replay(mockPartition, mockConnectionHandles);
 		assertEquals(4, testClass.getTotalLeased());
 		verify(mockPartition, mockConnectionHandles);
