diff --git a/old_product.java b/new_product.java
index d5fe087..5ba6765 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,5 @@
 public static String createRandomFromString(String str,int length){
-        Validate.notEmpty(str, "str can't be null/empty!");
+        Validate.notBlank(str, "str can't be null/empty!");
         Validate.isTrue(length > 0, Slf4jUtil.formatMessage("length:[{}] can not <=0", length));
-
-        char[] ch = new char[length];
-        int j = str.length();
-        for (int i = 0; i < length; ++i){
-            int index = JVM_RANDOM.nextInt(j);// 随机取个字符
-            ch[i] = str.charAt(index);
-        }
-        return new String(ch);
+        return RandomStringUtils.random(length, str);
     }
\ No newline at end of file
