diff --git a/old_product.java b/new_product.java
index 8e81cf4..6680fc0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
 public final boolean containsAll(final StringList l) {
-    if (isEmpty() && !l.isEmpty()) return false;
+    if(isEmpty() && !l.isEmpty()) return false;
     int i = 0;
-    for (String e : l) {
+    for(final String e : l) {
       int result;
-      while (0 != (result = list[i].compareTo(e))) {
-        if (++i >= size() || result > 0) return false;
+      while(0 != (result = list[i].compareTo(e))) {
+        if(++i >= size() || result > 0) return false;
       }
     }
     return true;
