diff --git a/old_product.java b/new_product.java
index 6857438..824d9fd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,12 @@
-public final int addAll(Iterator<? extends ObjectCursor<? extends KType>> iterator)
+public final int addAll(ObjectContainer<? extends KType> container)
     {
-        int count = 0;
-        while (iterator.hasNext())
+        final int size = container.size();
+        ensureBufferSpace(size);
+
+        for (ObjectCursor<? extends KType> cursor : container)
         {
-            add(iterator.next().value);
-            count++;
+            add(cursor.value);
         }
 
-        return count;
+        return size;
     }
\ No newline at end of file
