diff --git a/old_test.java b/new_test.java
index 68a6854..f5b1161 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
         set.add(newArray(set2.keys, 0, 1));
 
         assertEquals(1, set.addAll(set2));
-        assertEquals(0, set.addAll(set2.iterator()));
+        assertEquals(0, set.addAll(set2));
 
         assertEquals(3, set.size());
         assertSortedListEquals(set.toArray(), 0, 1, 2);
