diff --git a/old_test.java b/new_test.java
index 4ac4724..7c19c6a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -11,7 +11,8 @@
         // least two TLS-Actions with exactly one message for the first
         // TLS-Action and at least one message for the last TLS-Action, which
         // would be the basic Client/Server-Hello:
-        helloWorkflow = workflowConfigurationFactory.createHelloWorkflow();
+        WorkflowConfigurationFactory factory = new WorkflowConfigurationFactory(config);
+        helloWorkflow = factory.createWorkflowTrace(WorkflowTraceType.HELLO, RunningModeType.CLIENT);
 
         Assert.assertThat(helloWorkflow.getMessageActions().size(), Matchers.greaterThanOrEqualTo(2));
 
@@ -33,7 +34,8 @@
         config.setHighestProtocolVersion(ProtocolVersion.DTLS10);
         config.setClientAuthentication(false);
         workflowConfigurationFactory = new WorkflowConfigurationFactory(config);
-        helloWorkflow = workflowConfigurationFactory.createHelloWorkflow();
+        helloWorkflow = workflowConfigurationFactory.createWorkflowTrace(WorkflowTraceType.HELLO,
+                RunningModeType.CLIENT);
 
         firstAction = helloWorkflow.getMessageActions().get(0);
         clientHelloMessage = (ClientHelloMessage) firstAction.getMessages().get(0);
@@ -60,7 +62,8 @@
         config.setClientAuthentication(true);
         config.setDefaultSelectedCipherSuite(CipherSuite.TLS_DHE_DSS_WITH_AES_256_GCM_SHA384);
         workflowConfigurationFactory = new WorkflowConfigurationFactory(config);
-        helloWorkflow = workflowConfigurationFactory.createHelloWorkflow();
+        helloWorkflow = workflowConfigurationFactory.createWorkflowTrace(WorkflowTraceType.HELLO,
+                RunningModeType.CLIENT);
 
         lastAction = (ReceiveAction) helloWorkflow.getLastMessageAction();
         Assert.assertNotNull(lastAction.getExpectedMessages().get(2));
