diff --git a/old_product.java b/new_product.java
index 7ae414a..eeaf2ba 100644
--- a/old_product.java
+++ b/new_product.java
@@ -47,14 +47,20 @@ private WorkflowTrace createHelloWorkflow(ConnectionEnd ourConnectionEnd) {
             messages.add(new CertificateVerifyMessage(config));
             messages.add(new FinishedMessage(config));
         } else {
-            if (ourConnectionEnd.getConnectionEndType() == ConnectionEndType.CLIENT) {
-                messages.add(new CertificateMessage());
-            } else {
-                messages.add(new CertificateMessage(config));
+            if (config.getDefaultSelectedCipherSuite().isPSK_DHPSK() == false
+                    && config.getDefaultSelectedCipherSuite().isSRP() == false) {
+                if (ourConnectionEnd.getConnectionEndType() == ConnectionEndType.CLIENT) {
+                    messages.add(new CertificateMessage());
+                } else {
+                    messages.add(new CertificateMessage(config));
+                }
             }
             if (config.getDefaultSelectedCipherSuite().isEphemeral()) {
                 addServerKeyExchangeMessage(messages);
             }
+            if (config.getDefaultSelectedCipherSuite().isSRP()) {
+                messages.add(new SRPServerKeyExchangeMessage(config));
+            }
             if (config.isClientAuthentication()) {
                 CertificateRequestMessage certRequest = new CertificateRequestMessage(config);
                 messages.add(certRequest);
