diff --git a/old_product.java b/new_product.java
index 55c0cdf..3975ad5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,10 +2,12 @@ private Item store(final QueryContext ctx) throws QueryException {
     checkWrite(ctx);
 
     final Data data = data(0, ctx);
-    final String key = path(1, ctx);
-    if(!new IOFile(key).valid()) RESINV.thrw(input, key);
+    final String path = path(1, ctx);
+    final IOFile file = data.meta.binary(path);
+    if(file == null || file.isDir() || !file.isValid())
+      RESINV.thrw(input, path);
 
     final byte[] val = checkBin(expr[2], ctx);
-    ctx.updates.add(new DBPut(data, token(key), val, input), ctx);
+    ctx.updates.add(new DBStore(data, token(path), val, input), ctx);
     return null;
   }
\ No newline at end of file
