diff --git a/old_test.java b/new_test.java
index edc45a7..0027d54 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,7 @@
         42,
         KeyStatusType.ENABLED,
         OutputPrefixType.TINK));
-    KeysetHandle handle = CleartextKeysetHandleFactory.fromBinaryFormat(keyset.toByteArray());
+    KeysetHandle handle = CleartextKeysetHandle.fromBinaryFormat(keyset.toByteArray());
     assertEquals(keyset, handle.getKeyset());
 
     String keysetInfo = handle.toString();
