diff --git a/old_product.java b/new_product.java
index 97dfee2..ae61bd6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 private Item xsd(final QueryContext qc) throws QueryException {
     final Value seq = xsdInfo(qc);
     if(seq == Empty.SEQ) return null;
-    throw BXVA_FAIL.get(info, seq.iter().next());
+    throw BXVA_FAIL_X.get(info, seq.iter().next());
   }
\ No newline at end of file
