diff --git a/old_test.java b/new_test.java
index 0126b16..d2b09f8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,13 +15,13 @@
       "return validate:xsd($doc, $schema)", "");
 
     // invalid arguments
-    error(_VALIDATE_XSD.args("unknown"), Err.WHICHRES);
-    error(_VALIDATE_XSD.args(FILE, "unknown.xsd"), Err.WHICHRES);
-    error(_VALIDATE_XSD.args(FILE), Err.BXVA_FAIL);
+    error(_VALIDATE_XSD.args("unknown"), Err.WHICHRES_X);
+    error(_VALIDATE_XSD.args(FILE, "unknown.xsd"), Err.WHICHRES_X);
+    error(_VALIDATE_XSD.args(FILE), Err.BXVA_FAIL_X);
     error(
         "let $doc := <root/> " +
         "let $schema := <xs:schema xmlns:xs='http://www.w3.org/2001/XMLSchema'> " +
         "<xs:element name='unknown'/> " +
         "</xs:schema> " +
-        "return validate:xsd($doc, $schema)", Err.BXVA_FAIL);
+        "return validate:xsd($doc, $schema)", Err.BXVA_FAIL_X);
   }
\ No newline at end of file
