diff --git a/old_test.java b/new_test.java
index 2d6b6cb..9777b50 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,24 +2,24 @@
     public void testKFoldCrossValidation() {
         logger.info("kFoldCrossValidation");
         
-        DatabaseConfiguration dbConf = TestUtils.getDBConfig();
+        Configuration conf = TestUtils.getConfig();
         
         int k = 5;
         
-        Dataframe[] data = Datasets.heartDiseaseClusters(dbConf);
+        Dataframe[] data = Datasets.heartDiseaseClusters(conf);
         Dataframe trainingData = data[0];
         data[1].delete();
         
         
         String dbName = this.getClass().getSimpleName();
-        DummyXYMinMaxNormalizer df = new DummyXYMinMaxNormalizer(dbName, dbConf);
+        DummyXYMinMaxNormalizer df = new DummyXYMinMaxNormalizer(dbName, conf);
         df.fit_transform(trainingData, new DummyXYMinMaxNormalizer.TrainingParameters());
         
 
         
         
         
-        Kmeans instance = new Kmeans(dbName, dbConf);
+        Kmeans instance = new Kmeans(dbName, conf);
         
         Kmeans.TrainingParameters param = new Kmeans.TrainingParameters();
         param.setK(2);
