diff --git a/old_product.java b/new_product.java
index 0475dc4..af15714 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public VM kFoldCrossValidation(Dataframe trainingData, TP trainingParameters, int k) {
         logger.info("kFoldCrossValidation()");
         
-        return modelValidator.kFoldCrossValidation(trainingData, k, dbName, kb().getDbConf(), this.getClass(), trainingParameters);
+        return modelValidator.kFoldCrossValidation(trainingData, k, dbName, kb().getConf(), this.getClass(), trainingParameters);
     }
\ No newline at end of file
