diff --git a/old_test.java b/new_test.java
index cbddedc..a447aec 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,8 +5,8 @@
         instance.setYHigh(yhigh);
         instance.setYLow(ylow);
         Properties expResult = new Properties();
-        expResult.setProperty("transferFunction.yHigh", new Double(yhigh).toString());
-        expResult.setProperty("transferFunction.yLow",new Double(ylow).toString());
+        expResult.setProperty("transferFunction.yHigh", String.valueOf(yhigh));
+        expResult.setProperty("transferFunction.yLow", String.valueOf(ylow));
         Properties result = instance.getProperties();
         assertEquals(expResult, result);
         }
\ No newline at end of file
