diff --git a/old_product.java b/new_product.java
index 431a53c..e6a4ab8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 public Properties getProperties() {
 		Properties properties = new Properties();
-		properties.setProperty("transferFunction.yHigh", new Double(yHigh).toString());
-		properties.setProperty("transferFunction.yLow", new Double(yLow).toString());
+		properties.setProperty("transferFunction.yHigh", String.valueOf(yHigh));
+		properties.setProperty("transferFunction.yLow", String.valueOf(yLow));
 		return properties;
 	}
\ No newline at end of file
