diff --git a/old_test.java b/new_test.java
index 04c7278..63a8730 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,4 +20,11 @@
                 CreateScopeStatus.newBuilder().setStatus(CreateScopeStatus.Status.FAILURE).build()));
         response = target(resourceURI).request().async().post(Entity.json(createScopeRequest));
         assertEquals("Create Scope response code", 500, response.get().getStatus());
+
+        // Test to create an invalid scope name.
+        when(mockControllerService.createScope(scope1)).thenReturn(CompletableFuture.completedFuture(
+                CreateScopeStatus.newBuilder().setStatus(CreateScopeStatus.Status.SCOPE_EXISTS).build()));
+        createScopeRequest.setScopeName("_system");
+        response = target(resourceURI).request().async().post(Entity.json(createScopeRequest));
+        assertEquals("Create Scope response code", 400, response.get().getStatus());
     }
\ No newline at end of file
