diff --git a/old_product.java b/new_product.java
index 8334283..19f4b21 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,6 +2,14 @@
     public void createScope(final CreateScopeRequest createScopeRequest, final SecurityContext securityContext,
             final AsyncResponse asyncResponse) {
         long traceId = LoggerHelpers.traceEnter(log, "createScope");
+        try {
+            NameUtils.validateUserScopeName(createScopeRequest.getScopeName());
+        } catch (IllegalArgumentException | NullPointerException e) {
+            log.warn("Create scope failed due to invalid scope name {}", createScopeRequest.getScopeName());
+            asyncResponse.resume(Response.status(Status.BAD_REQUEST).build());
+            LoggerHelpers.traceLeave(log, "createScope", traceId);
+            return;
+        }
 
         controllerService.createScope(createScopeRequest.getScopeName()).thenApply(scopeStatus -> {
             if (scopeStatus.getStatus() == CreateScopeStatus.Status.SUCCESS) {
