diff --git a/old_test.java b/new_test.java
index d657640..803deff 100644
--- a/old_test.java
+++ b/new_test.java
@@ -24,6 +24,6 @@
         int maxLevels = 1000;
 
 
-        List<AlignmentInterval.Row> result = (new AlignmentPacker()).packAlignments(iter, end, false, null, 10000);
+        Map<String, List<AlignmentInterval.Row>> result = (new AlignmentPacker()).packAlignments(iter, end, false, null, 10000);
 
     }
\ No newline at end of file
