diff --git a/old_test.java b/new_test.java
index 4fe7a1d..711591a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,5 +9,5 @@
         SynInfo synInfo = new SynInfo(false);
         when(session.getStreams()).thenReturn(streams);
         stream.syn(synInfo);
-        verify(session).syn(argThat(new PushSynInfoMatcher(stream.getId(), synInfo)), any(StreamFrameListener.class), anyLong(), any(TimeUnit.class), any(Handler.class));
+        verify(session).syn(argThat(new PushSynInfoMatcher(stream.getId(), synInfo)), any(StreamFrameListener.class), anyLong(), any(TimeUnit.class), any(Callback.class));
     }
\ No newline at end of file
