diff --git a/old_product.java b/new_product.java
index 5b659c7..588aa95 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public static MonthDay now(Clock clock) {
         final LocalDate now = LocalDate.now(clock);  // called once
-        return MonthDay.of(now.getMonthOfYear(), now.getDayOfMonth());
+        return MonthDay.of(now.getMonth(), now.getDayOfMonth());
     }
\ No newline at end of file
