diff --git a/old_test.java b/new_test.java
index 461dcdc..0ab3b43 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,5 +4,5 @@
         Clock clock = Clock.fixed(instant, ZoneId.UTC);
         YearMonth test = YearMonth.now(clock);
         assertEquals(test.getYear(), 2010);
-        assertEquals(test.getMonthOfYear(), Month.DECEMBER);
+        assertEquals(test.getMonth(), Month.DECEMBER);
     }
\ No newline at end of file
