diff --git a/old_test.java b/new_test.java
index 8c0c1d7..27d3605 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,14 @@
 @Test
     public void testHashCode() {
-        StrBuilder sb = new StrBuilder();
-        int hc1a = sb.hashCode();
-        int hc1b = sb.hashCode();
+        final StrBuilder sb = new StrBuilder();
+        final int hc1a = sb.hashCode();
+        final int hc1b = sb.hashCode();
         assertEquals(0, hc1a);
         assertEquals(hc1a, hc1b);
         
         sb.append("abc");
-        int hc2a = sb.hashCode();
-        int hc2b = sb.hashCode();
+        final int hc2a = sb.hashCode();
+        final int hc2b = sb.hashCode();
         assertTrue(hc2a != 0);
         assertEquals(hc2a, hc2b);
     }
\ No newline at end of file
