diff --git a/old_test.java b/new_test.java
index adc9acb..4e28700 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,17 +16,17 @@
 
 		PerspectiveOps.splitAssociated(list,list1,list2);
 
-		DMatrixRMaj expected1 = new DMatrixRMaj(3,3);
-		DMatrixRMaj expected2 = new DMatrixRMaj(3,3);
+		NormalizationPoint2D expected1 = new NormalizationPoint2D();
+		NormalizationPoint2D expected2 = new NormalizationPoint2D();
 
 		LowLevelMultiViewOps.computeNormalization(list1, expected1);
 		LowLevelMultiViewOps.computeNormalization(list2, expected2);
 
-		DMatrixRMaj found1 = new DMatrixRMaj(3,3);
-		DMatrixRMaj found2 = new DMatrixRMaj(3,3);
+		NormalizationPoint2D found1 = new NormalizationPoint2D();
+		NormalizationPoint2D found2 = new NormalizationPoint2D();
 
 		LowLevelMultiViewOps.computeNormalization(list, found1, found2);
 
-		assertTrue(MatrixFeatures_DDRM.isIdentical(expected1, found1, 1e-8));
-		assertTrue(MatrixFeatures_DDRM.isIdentical(expected2,found2,1e-8));
+		assertTrue(expected1.isEquals(found1,1e-8));
+		assertTrue(expected2.isEquals(found2,1e-8));
 	}
\ No newline at end of file
