diff --git a/old_test.java b/new_test.java
index 9f27efa..d6d9668 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testGetAnalyzers_SpecificPhases() throws Exception {
-        AnalyzerService instance = new AnalyzerService(Thread.currentThread().getContextClassLoader());
+        AnalyzerService instance = new AnalyzerService(Thread.currentThread().getContextClassLoader(), false);
         List<Analyzer> result = instance.getAnalyzers(INITIAL, FINAL);
 
         for (Analyzer a : result) {
