diff --git a/old_product.java b/new_product.java
index cc7bf7c..ecc6d76 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,10 +2,12 @@ private List<Analyzer> getAnalyzers(List<AnalysisPhase> phases) {
         final List<Analyzer> analyzers = new ArrayList<>();
         final Iterator<Analyzer> iterator = service.iterator();
         boolean experimentalEnabled = false;
+        boolean retiredEnabled = false;
         try {
             experimentalEnabled = Settings.getBoolean(Settings.KEYS.ANALYZER_EXPERIMENTAL_ENABLED, false);
+            retiredEnabled = Settings.getBoolean(Settings.KEYS.ANALYZER_RETIRED_ENABLED, false);
         } catch (InvalidSettingException ex) {
-            LOGGER.error("invalid experimental setting", ex);
+            LOGGER.error("invalid experimental or retired setting", ex);
         }
         while (iterator.hasNext()) {
             final Analyzer a = iterator.next();
@@ -15,6 +17,9 @@ private List<Analyzer> getAnalyzers(List<AnalysisPhase> phases) {
             if (!experimentalEnabled && a.getClass().isAnnotationPresent(Experimental.class)) {
                 continue;
             }
+            if (!retiredEnabled && a.getClass().isAnnotationPresent(Retired.class)) {
+                continue;
+            }
             LOGGER.debug("Loaded Analyzer {}", a.getName());
             analyzers.add(a);
         }
