diff --git a/old_test.java b/new_test.java
index 54c2cbe..62a9c97 100644
--- a/old_test.java
+++ b/new_test.java
@@ -16,6 +16,6 @@
 
     ReportFiles files = createReportFiles(destination, Optional.empty(), root);
 
-    assertThat(files.entryPoint().toString()).isEqualTo("some/path/unimplemented.txt");
-    assertThat(readAllFiles(files).keySet()).containsExactly("some/path/unimplemented.txt");
+    assertThat(files.entryPoint().toString()).isEqualTo("some/path/path.zip");
+    assertThat(readAllFiles(files).keySet()).containsExactly("some/path/path.zip");
   }
\ No newline at end of file
