diff --git a/old_test.java b/new_test.java
index c6a8695..a81f172 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,6 @@
 @Test
-	public void saveOrder_shouldFailIfAnActiveDrugOrderForTheSameDrugFormulationExists() throws Exception {
+	public void saveOrder_shouldThrowAmbiguousOrderExceptionIfAnActiveDrugOrderForTheSameDrugFormulationExists()
+	        throws Exception {
 		executeDataSet("org/openmrs/api/include/OrderServiceTest-drugOrdersWithSameConceptAndDifferentFormAndStrength.xml");
 		final Patient patient = patientService.getPatient(2);
 		//sanity check that we have an active order for the same concept
@@ -19,7 +20,8 @@
 		order.setQuantityUnits(conceptService.getConcept(51));
 		order.setNumRefills(2);
 		
-		expectedException.expect(APIException.class);
-		expectedException.expectMessage("Cannot have more than one active order for the same orderable and care setting");
+		expectedException.expect(AmbiguousOrderException.class);
+		expectedException
+		        .expectMessage("Cannot have more than one active order for the same orderable and care setting at same time");
 		orderService.saveOrder(order, null);
 	}
\ No newline at end of file
