diff --git a/old_product.java b/new_product.java
index aec9f0e..f2c1821 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 private Iter eval(final QueryContext ctx, final byte[] qu)
       throws QueryException {
-    final QueryContext qt = new QueryContext(ctx.resource.context);
+    final QueryContext qt = new QueryContext(ctx.context);
     qt.parse(string(qu));
     qt.compile();
     return ItemIter.get(qt.iter());
