diff --git a/old_product.java b/new_product.java
index bc141a7..2b97319 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,5 +2,5 @@
     @ResponseStatus(HttpStatus.OK)
     public CommandResource getCommand(@PathVariable("id") final String id) throws GenieException {
         log.debug("Called to get command with id {}", id);
-        return this.commandResourceAssembler.toResource(this.commandService.getCommand(id));
+        return this.commandResourceAssembler.toResource(DtoAdapters.toV3Command(this.commandService.getCommand(id)));
     }
\ No newline at end of file
