diff --git a/old_test.java b/new_test.java
index 21b0e26..cb4ccf9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,9 @@
 @Test
-    public void testGetCpu()
-    {
-        System.out.println("getCpu");
-        double result = instance.getCpu();
-        System.out.println(result);
-        assertTrue(result >= 0);
-    }
\ No newline at end of file
+   public void testGetCpu()
+   {
+      System.out.println("getCpu");
+      prepareTestData(123);
+      double result = instance.getCpu();
+      System.out.println(result);
+      assertTrue(result >= 0);
+   }
\ No newline at end of file
