diff --git a/old_product.java b/new_product.java
index c1bd73a..897a2f7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,18 +1,16 @@
-public double getCpu() {
-        double ret;
+public double getCpu()
+   {
+      double ret;
 
-        try
-        {
-            String value = getData("cpu");
-            if(value != null) {
-                ret = Double.parseDouble(value);
-            } else {
-                ret = MetricsGetter.AGENT_ERROR;
-            }
-        } catch (IOException ioex)
-        {
-            ret = MetricsGetter.AGENT_ERROR;
-        }
+      String value = getData("cpu");
+      if (value != null)
+      {
+         ret = Double.parseDouble(value);
+      }
+      else
+      {
+         ret = MetricsGetter.AGENT_ERROR;
+      }
 
-        return ret;
-    }
\ No newline at end of file
+      return ret;
+   }
\ No newline at end of file
