diff --git a/old_test.java b/new_test.java
index 6939e76..454fa44 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
 		encoder.setVersion(1).
 				setError(QrCode.ErrorLevel.H).
 				setMask(new QrCodeMaskPattern.NONE(0b011)).
-				alphanumeric("AC-42");
+				alphanumeric("AC-42").fixate();
 
 		byte[] expected = new byte[]{0b00100000, 0b00101001, (byte)0b11001110, (byte)0b11100111, 0b00100001,0};
 
@@ -12,6 +12,6 @@
 
 		assertEquals(encoder.packed.size/8,expected.length);
 		for (int i = 0; i < expected.length; i++) {
-			assertEquals(encoder.packed.data[i],expected[i]);
+			assertEquals(expected[i],encoder.packed.data[i]);
 		}
 	}
\ No newline at end of file
