diff --git a/old_test.java b/new_test.java
index 752d006..5a7be37 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,22 +2,22 @@
     public void testSelectByExample() {
         SqlSession sqlSession = MybatisHelper.getSqlSession();
         try {
-            CommonMapper entityMapper = sqlSession.getMapper(CommonMapper.class);
-            EntityMapper baseMapper = new EntityMapper(entityMapper);
+            CommonMapper commonMapper = sqlSession.getMapper(CommonMapper.class);
+            EntityMapper entityMapper = new EntityMapper(commonMapper);
 
             Example example = new Example(Country.class);
-            example.createCriteria().andEqualTo("countryname","China");
+            example.createCriteria().andEqualTo("countryname", "China");
 
-            List<Country> countries = baseMapper.selectByExample(example);
-            Assert.assertEquals(1,countries.size());
-            Assert.assertEquals("CN",countries.get(0).getCountrycode());
+            List<Country> countries = entityMapper.selectByExample(example);
+            Assert.assertEquals(1, countries.size());
+            Assert.assertEquals("CN", countries.get(0).getCountrycode());
 
             example = new Example(Country.class);
-            example.createCriteria().andEqualTo("id",100);
+            example.createCriteria().andEqualTo("id", 100);
 
-            countries = baseMapper.selectByExample(example);
-            Assert.assertEquals(1,countries.size());
-            Assert.assertEquals("MY",countries.get(0).getCountrycode());
+            countries = entityMapper.selectByExample(example);
+            Assert.assertEquals(1, countries.size());
+            Assert.assertEquals("MY", countries.get(0).getCountrycode());
         } finally {
             sqlSession.close();
         }
