diff --git a/old_product.java b/new_product.java
index d31b112..f389b2f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public static int getDayOfYear(Date date){
-        Date firstDateOfThisYear = getFirstDateOfThisYear(date);
-        return getIntervalDay(date, firstDateOfThisYear) + 1;
+        Validate.notNull(date, "date can't be null!");
+        return CalendarUtil.getFieldValue(date, Calendar.DAY_OF_YEAR);
     }
\ No newline at end of file
