diff --git a/old_test.java b/new_test.java
index 2a53ba2..a2d98b8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,9 +7,6 @@
 
     style = wrap("<xsl:param name='t'/><xsl:template match='/'>" +
         "<X><xsl:value-of select='$t'/></X></xsl:template>");
-    String param = "<xslt:parameters><xslt:t>1</xslt:t></xslt:parameters>";
-    query(_XSLT_TRANSFORM.args(doc, style, param), "<X>1</X>");
-
-    param = " map { 't' : text { '1' } }";
-    query(_XSLT_TRANSFORM.args(doc, style, param), "<X>1</X>");
+    query(_XSLT_TRANSFORM.args(doc, style, " map { 't': '1' }"), "<X>1</X>");
+    query(_XSLT_TRANSFORM.args(doc, style, " map { 't' : text { '1' } }"), "<X>1</X>");
   }
\ No newline at end of file
