diff --git a/old_product.java b/new_product.java
index 4335dcb..569b53e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@ final byte[] transform(final QueryContext qc) throws QueryException {
     checkCreate(qc);
     final IO in = read(exprs[0], qc);
     final IO xsl = read(exprs[1], qc);
-    final Options opts = toOptions(2, Q_PARAMETERS, new Options(), qc);
+    final Options opts = toOptions(2, new Options(), qc);
 
     final PrintStream tmp = System.err;
     final ArrayOutput ao = new ArrayOutput();
