diff --git a/old_test.java b/new_test.java
index 9cfadb8..fd63734 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
 
     // Get response handler.
     AsyncRequestResponseHandlerFactory responseHandlerFactory =
-        new AsyncRequestResponseHandlerFactory(1, restServerMetrics);
+        new AsyncRequestResponseHandlerFactory(1, METRIC_REGISTRY);
     RestResponseHandler restResponseHandler = responseHandlerFactory.getRestResponseHandler();
     assertNotNull("No RestResponseHandler returned", restResponseHandler);
     assertEquals("Did not receive an AsyncRequestResponseHandler instance",
@@ -17,7 +17,7 @@
         new MockBlobStorageService(verifiableProperties, restResponseHandler, router);
     // Get request handler.
     AsyncRequestResponseHandlerFactory requestHandlerFactory =
-        new AsyncRequestResponseHandlerFactory(1, restServerMetrics, blobStorageService);
+        new AsyncRequestResponseHandlerFactory(1, METRIC_REGISTRY, blobStorageService);
     RestRequestHandler restRequestHandler = requestHandlerFactory.getRestRequestHandler();
     assertNotNull("No RestRequestHandler returned", restRequestHandler);
     assertEquals("Did not receive an AsyncRequestResponseHandler instance",
