diff --git a/old_product.java b/new_product.java
index f53b381..8620e9e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-protected ParametersZhang99 initialParam( List<List<Point2D_F64>> observations )
+protected Zhang99Parameters initialParam( List<List<Point2D_F64>> observations )
 	{
 		List<DenseMatrix64F> homographies = new ArrayList<DenseMatrix64F>();
 		List<Se3_F64> motions = new ArrayList<Se3_F64>();
@@ -25,5 +25,5 @@ protected ParametersZhang99 initialParam( List<List<Point2D_F64>> observations )
 
 		double distort[] = computeRadial.getParameters();
 
-		return convertIntoZhangParam(motions, K, distort);
+		return convertIntoZhangParam(motions, K,assumeZeroSkew, distort);
 	}
\ No newline at end of file
