diff --git a/old_test.java b/new_test.java
index 2c9e3f3..1d21098 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,12 @@
 @Test
   public void defaults() throws IOException {
-    MkdirsOptions options = MkdirsOptions.defaults();
+    MkdirsOptions options = MkdirsOptions.defaults(mConfiguration);
 
     // Verify the default createParent is true.
     assertTrue(options.getCreateParent());
     // Verify that the owner and group are not set.
     assertNull(options.getOwner());
     assertNull(options.getGroup());
-    assertEquals(ModeUtils.applyDirectoryUMask(Mode.defaults()), options.getMode());
+    String umask = mConfiguration.get(PropertyKey.SECURITY_AUTHORIZATION_PERMISSION_UMASK);
+    assertEquals(ModeUtils.applyDirectoryUMask(Mode.defaults(), umask), options.getMode());
   }
\ No newline at end of file
