diff --git a/old_product.java b/new_product.java
index b857b8a..2d4c146 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static SerializerOptions parse(final Item it, final SerializerOptions sopts,
+public static SerializerOptions serializer(final Item it, final SerializerOptions sopts,
       final InputInfo info) throws QueryException {
     new FuncOptions(Q_SPARAM, info).parse(it, sopts, Err.SEROPT);
     return sopts;
