diff --git a/old_test.java b/new_test.java
index 2b3d2f7..ebc632a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,9 +2,9 @@
     public void testGetMemory() {
         SystemInfo si = new SystemInfo();
         HardwareAbstractionLayer hal = si.getHardware();
-        Memory memory = hal.getMemory();
-        assertNotNull(memory);
-        assertTrue(memory.getTotal() > 0);
-        assertTrue(memory.getAvailable() >= 0);
-        assertTrue(memory.getAvailable() <= memory.getTotal());
+        GlobalMemory globalMemory = hal.getMemory();
+        assertNotNull(globalMemory);
+        assertTrue(globalMemory.getTotal() > 0);
+        assertTrue(globalMemory.getAvailable() >= 0);
+        assertTrue(globalMemory.getAvailable() <= globalMemory.getTotal());
     }
\ No newline at end of file
