diff --git a/old_product.java b/new_product.java
index 6102301..d1e28d6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,7 @@
-public ZoneId toZoneId() {
-        return ZoneId.of(this);
+@Override
+    public <R> R query(Query<R> query) {
+        if (query == Query.ZONE_ID || query == Query.CHRONO) {
+            return null;
+        }
+        return query.doQuery(this);
     }
\ No newline at end of file
