diff --git a/old_test.java b/new_test.java
index 238a598..7ac5c1b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
     public void testNextDouble() throws Exception {
-        double result = RandomUtils.nextDouble(33d, 42d);
+        final double result = RandomUtils.nextDouble(33d, 42d);
         assertTrue(result >= 33d && result <= 42d);
     }
\ No newline at end of file
