diff --git a/old_test.java b/new_test.java
index ced5575..ce4e75b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testAdjustTLSContext() {
-        context.getConfig().setConnectionEndType(ConnectionEndType.CLIENT);
+        context.setConnectionEndType(ConnectionEndType.CLIENT);
         context.setTalkingConnectionEndType(ConnectionEndType.SERVER);
         AlertMessage message = new AlertMessage();
         message.setDescription(AlertDescription.ACCESS_DENIED.getValue());
