diff --git a/old_test.java b/new_test.java
index 1fa5dde..e70d18d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,14 +2,13 @@
 	public void testPersist() throws Exception {
 		assumeTrue(!ConnectionUtils.isJredis(template.getConnectionFactory()));
 
-		populateBoundKey();
-
 		keyOps.persist();
 
 		assertEquals(keyOps.getClass().getName() + " -> " + keyOps.getKey(), Long.valueOf(-1), keyOps.getExpire());
 		if (keyOps.expire(10, TimeUnit.SECONDS)) {
 			assertTrue(keyOps.getExpire().longValue() > 0);
 		}
+
 		keyOps.persist();
 		assertEquals(keyOps.getClass().getName() + " -> " + keyOps.getKey(), -1, keyOps.getExpire().longValue());
 	}
\ No newline at end of file
