diff --git a/old_product.java b/new_product.java
index b679d89..6da3159 100644
--- a/old_product.java
+++ b/new_product.java
@@ -11,9 +11,9 @@ public void parse(final Item item, final Options options, final boolean ignore)
         if(!(it instanceof AStr)) FUNCMP.thrw(info, map.description(), AtomType.STR, it.type);
         final Value v = map.get(it, info);
         if(!v.isItem()) FUNCMP.thrw(info, map.description(), AtomType.ITEM, v);
-        final String key = Token.string(it.string(null));
-        final String val = Token.string(((Item) v).string(info));
-        if(!options.set(key, val, true) && options.predefined()) ELMOPTION.thrw(info, key);
+        final String key = string(it.string(null));
+        final String val = string(((Item) v).string(info));
+        if(!options.set(key, val) && options.predefined()) ELMOPTION.thrw(info, key);
       }
     } else {
       if(!test.eq(item)) ELMMAPTYPE.thrw(info, root, item.type);
@@ -31,8 +31,7 @@ public void parse(final Item item, final Options options, final boolean ignore)
         final String key = string(qn.local());
         byte[] val = n.attribute(VALUE);
         if(val == null) val = n.string();
-        if(!options.set(key, string(val), true) && options.predefined())
-          ELMOPTION.thrw(info, key);
+        if(!options.set(key, string(val)) && options.predefined()) ELMOPTION.thrw(info, key);
       }
     }
   }
\ No newline at end of file
