diff --git a/old_test.java b/new_test.java
index 0506956..c32da4c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
     public void testAdd2()
     {
-        set.add(key1, key1);
+        set.addAll(key1, key1);
         assertEquals(1, set.size());
-        assertEquals(1, set.add(key1, key2));
+        assertEquals(1, set.addAll(key1, key2));
         assertEquals(2, set.size());
     }
\ No newline at end of file
