diff --git a/old_product.java b/new_product.java
index 152d0c5..7971c2a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
 @Authorized(PrivilegeConstants.ADD_ORDERS)
-	public Order discontinueOrder(Order orderToDiscontinue, Concept reasonCoded, Date discontinueDate) throws Exception;
\ No newline at end of file
+	public Order discontinueOrder(Order orderToDiscontinue, Concept reasonCoded, Date discontinueDate, Provider orderer) throws Exception;
\ No newline at end of file
