diff --git a/old_test.java b/new_test.java
index a993afe..6687200 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,5 +2,5 @@
 	public void discontinueOrder_shouldFailForAStoppedOrder() throws Exception {
 		Order orderToDiscontinue = orderService.getOrder(1);
 		assertNotNull(orderToDiscontinue.getDateStopped());
-		orderService.discontinueOrder(orderToDiscontinue, Context.getConceptService().getConcept(1), null);
+		orderService.discontinueOrder(orderToDiscontinue, Context.getConceptService().getConcept(1), null, null);
 	}
\ No newline at end of file
