diff --git a/old_test.java b/new_test.java
index 08ee9b7..b293653 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,5 @@
     count(_MAP_PUT.args(" map{ 'a': 'b' }", "c", "d"), 2);
     count(_MAP_PUT.args(" map{ 'a': 'b' }", "c", "d"), 2);
 
-    error(_MAP_PUT.args(" map{ xs:time('01:01:01'):'b' }", "xs:time('01:01:02+01:00')", "1"),
-        MAP_TZ);
+    query(_MAP_PUT.args(" map{ xs:time('01:01:01'):'b' }", "xs:time('01:01:02+01:00')", "1"));
   }
\ No newline at end of file
