diff --git a/old_product.java b/new_product.java
index 45f4087..6475147 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,11 +2,11 @@ public final void serialize(final Item item) throws IOException {
     openResult();
     if(item instanceof ANode) {
       final Type type = item.type;
-      if(type == NodeType.ATT) SERATTR.thrwSerial(item);
-      if(type == NodeType.NSP) SERNS.thrwSerial(item);
+      if(type == NodeType.ATT) SERATTR.thrwIO(item);
+      if(type == NodeType.NSP) SERNS.thrwIO(item);
       serialize((ANode) item);
     } else if(item instanceof FItem) {
-      SERFUNC.thrwSerial(item.description());
+      SERFUNC.thrwIO(item.description());
     } else {
       finishElement();
       atomic(item);
