diff --git a/old_test.java b/new_test.java
index 97fda79..bb2a55f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void testSetString() {
         String key = "newProperty";
         String value = "someValue";
-        Settings.setString(key, value);
-        String expResults = Settings.getString(key);
+        getSettings().setString(key, value);
+        String expResults = getSettings().getString(key);
         Assert.assertEquals(expResults, value);
     }
\ No newline at end of file
