diff --git a/old_product.java b/new_product.java
index bbf78a3..8152d85 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static void setString(String key, String value) {
-        LOCAL_SETTINGS.get().props.setProperty(key, value);
+public void setString(String key, String value) {
+        props.setProperty(key, value);
         LOGGER.debug("Setting: {}='{}'", key, value);
     }
\ No newline at end of file
