diff --git a/old_test.java b/new_test.java
index 6b17e73..00d4ff9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testOnCreateOrUpdateApplication() throws GeniePreconditionException {
+    public void testOnCreateOrUpdateApplication() throws GenieException {
         this.a = new ApplicationEntity(NAME, USER, VERSION, ApplicationStatus.ACTIVE);
         Assert.assertNotNull(this.a.getTags());
         this.a.onCreateOrUpdateApplication();
