diff --git a/old_test.java b/new_test.java
index e951782..fa82a5d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test
     public void testContainsText() {
-    	assertEquals(new ContainsTextCriterion("name", "a"), instance.containsText("name", "a").getQueryCriterion());
+    	assertEquals(Criteria.containsText("name", "a"), instance.containsText("name", "a").getQueryCriterion());
     }
\ No newline at end of file
