diff --git a/old_product.java b/new_product.java
index 4b6bbb8..6cd9271 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public CriteriaQuery containsText(String propName, String value) {
-        criterion = criterion.and(criterionBuilder.containsText(propName, value));
+        criterion = criterion.and(Criteria.containsText(propName, value));
         return this;
     }
\ No newline at end of file
