diff --git a/old_product.java b/new_product.java
index 898707d..a41b6c1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -59,7 +59,7 @@
             try {
                 MulticoreExecutor.run(tasks);
             } catch (Exception e) {
-                System.err.println(e.getMessage());
+                e.printStackTrace();
             }
         } else {
             List<TrainingTask> tasks = new ArrayList<TrainingTask>(k * (k - 1) / 2);
