diff --git a/old_test.java b/new_test.java
index 051ddfa..778a4eb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,12 @@
 @Test
   public void testDelete() throws BaseXException {
+    // Try to delete a package which is not installed
+    try {
+      new RepoManager(ctx).delete("xyz", null);
+      fail("Not installed package not detected.");
+    } catch(QueryException ex) {
+      check(ex, Err.PKGNOTINST);
+    }
     // Install a package without dependencies (pkg3)
     new RepoInstall(REPO + "pkg3.xar", null).execute(ctx);
     // Check if pkg3 is registered in the repo
