diff --git a/old_product.java b/new_product.java
index bfd1fae..3803c3d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,14 @@
 public void delete(final String pkg, final InputInfo ii)
       throws QueryException {
+    boolean found = false;
     for(final byte[] nextPkg : ctx.repo.pkgDict()) {
       if(nextPkg != null) {
         final byte[] dir = ctx.repo.pkgDict().get(nextPkg);
         if(eq(Package.getName(nextPkg), token(pkg)) || eq(dir, token(pkg))) {
           // A package can be deleted either by its name or by its directory
           // name
+          found = true;
+          // Check if package to be deleted participates in a dependency
           final byte[] primPkg = getPrimary(nextPkg, ii);
           if(primPkg == null) {
             // Clean package repository
@@ -14,10 +17,9 @@ public void delete(final String pkg, final InputInfo ii)
             ctx.repo.remove(new PkgParser(ctx, ii).parse(new IOFile(desc)));
             // Package does not participate in a dependency => delete it
             deleteFromDisc(f, ii);
-          } else {
-            PKGDEP.thrw(ii, string(primPkg), pkg);
-          }
+          } else PKGDEP.thrw(ii, string(primPkg), pkg);
         }
       }
     }
+    if(!found) PKGNOTINST.thrw(ii, pkg);
   }
\ No newline at end of file
