diff --git a/old_product.java b/new_product.java
index b11a034..716eaf3 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,14 +13,6 @@ public void validate(Object obj, Errors errors) throws APIException {
 			errors.rejectValue("name", "Provider.error.personOrName.required");
 		}
 		
-		//if this is a retired existing provider, skip this
-		//check if this provider has a unique identifier
-		boolean isUnique = Context.getProviderService().isProviderIdentifierUnique(provider);
-		if (!isUnique) {
-			errors.rejectValue("identifier", "Provider.error.duplicateIdentifier",
-			    new Object[] { provider.getIdentifier() }, null);
-		}
-		
 		if (provider.isRetired() && StringUtils.isEmpty(provider.getRetireReason())) {
 			errors.rejectValue("retireReason", "Provider.error.retireReason.required");
 		}
