diff --git a/old_test.java b/new_test.java
index 0cb0f86..7199341 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
-	@Verifies(value = "should reject a duplicate identifier for a new provider", method = "validate(Object,Errors)")
-	public void validate_shouldRejectADuplicateIdentifierForANewProvider() throws Exception {
+	@Verifies(value = "should not reject a duplicate identifier for a new provider", method = "validate(Object,Errors)")
+	public void validate_shouldNotRejectADuplicateIdentifierForANewProvider() throws Exception {
 		Provider duplicateProvider = providerService.getProvider(1);
 		Assert.assertFalse(duplicateProvider.isRetired());
 		
@@ -9,5 +9,5 @@
 		provider.setName("name");
 		
 		providerValidator.validate(provider, errors);
-		Assert.assertTrue(errors.hasFieldErrors("identifier"));
+		Assert.assertFalse(errors.hasFieldErrors("identifier"));
 	}
\ No newline at end of file
