diff --git a/old_product.java b/new_product.java
index 8c87b5d..441788a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@ public static INDArray pca(INDArray in, double variance) {
         // unit vector components
         INDArray[] pce = principalComponents(covmean[0]);
         // calculate the variance of each component
-        INDArray vars = Transforms.pow(Transforms.sqrt(pce[1], true), -1, true);
+        INDArray vars = Transforms.pow(pce[1], -0.5, true);
         double res = vars.sumNumber().doubleValue();
         double total = 0.0;
         int ndims = 0;
