diff --git a/old_product.java b/new_product.java
index 813cdda..896a4bf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public Document getDocument(final Document document) {
-		String documentId = IDHelper.getDocumentId(document);
-		return executeRequest(API_VERSION + documentId, null, Document.class);
-	}
\ No newline at end of file
+    final String documentId = IDHelper.getDocumentId(document);
+    return executeRequest(API_VERSION + documentId, null, Document.class);
+  }
\ No newline at end of file
