diff --git a/old_test.java b/new_test.java
index 990a6f4..a1091c6 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-	public void testGetDocument() {
-		Document document = service.getDocument(EXAMPLE_DOCUMENT);
-		Assert.assertNotNull(document);
-	}
\ No newline at end of file
+  public void testGetDocument() {
+    final Document document = service.getDocument(EXAMPLE_DOCUMENT);
+    Assert.assertNotNull(document);
+  }
\ No newline at end of file
