diff --git a/old_product.java b/new_product.java
index d16875c..714f530 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,6 +2,7 @@
   @Override
   public void loadMetadata(AlluxioURI path, LoadMetadataOptions options)
       throws FileDoesNotExistException, IOException, AlluxioException {
+    checkUri(path);
     FileSystemMasterClient masterClient = mFileSystemContext.acquireMasterClient();
     try {
       masterClient.loadMetadata(path, options);
