diff --git a/old_test.java b/new_test.java
index 6b60768..86bb08f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,4 +5,6 @@
     doNothing().when(mFileSystemMasterClient).loadMetadata(file, loadMetadataOptions);
     mFileSystem.loadMetadata(file, loadMetadataOptions);
     verify(mFileSystemMasterClient).loadMetadata(file, loadMetadataOptions);
+
+    verifyFilesystemContextAcquiredAndReleased();
   }
\ No newline at end of file
