diff --git a/old_product.java b/new_product.java
index 722d7b1..1c1e620 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static void computeNormalization(List<AssociatedPair> points, RowMatrix_F64 N1, RowMatrix_F64 N2)
+public static void computeNormalization(List<AssociatedPair> points, DMatrixRMaj N1, DMatrixRMaj N2)
 	{
 		double meanX1 = 0; double meanY1 = 0;
 		double meanX2 = 0; double meanY2 = 0;
