diff --git a/old_test.java b/new_test.java
index 23c88e4..1059fcd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
     public void testFilter() {
-        List<File> results = new TestFileFinder(dirsAndFilesFilter, -1).find(javaDir);
+        final List<File> results = new TestFileFinder(dirsAndFilesFilter, -1).find(javaDir);
         assertEquals("Result Size", 1 + dirs.length + ioFiles.length + outputFiles.length, results.size());
         assertTrue("Start Dir", results.contains(javaDir));
         checkContainsFiles("Dir", dirs, results);
