diff --git a/old_product.java b/new_product.java
index 784d17f..7b4b252 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,12 @@
-public static File[] filter(IOFileFilter filter, File... files) {
+public static File[] filter(final IOFileFilter filter, final File... files) {
         if (filter == null) {
             throw new IllegalArgumentException("file filter is null");
         }
         if (files == null) {
             return new File[0];
         }
-        List<File> acceptedFiles = new ArrayList<File>();
-        for (File file : files) {
+        final List<File> acceptedFiles = new ArrayList<File>();
+        for (final File file : files) {
             if (file == null) {
                 throw new IllegalArgumentException("file array contains null");
             }
