diff --git a/old_product.java b/new_product.java
index e4f8979..617e486 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,22 +1,12 @@
 List<PaddingVector> createMissingMacByteVectors(CipherSuite suite, ProtocolVersion version) {
         List<PaddingVector> vectorList = new LinkedList<>();
         int macSize = AlgorithmResolver.getMacAlgorithm(version, suite).getSize();
-        byte[] padding = createPaddingBytes(DEFAULT_CIPHERTEXT_LENGTH - macSize + 1 - 1);
+        byte[] padding = createPaddingBytes(DEFAULT_CIPHERTEXT_LENGTH - macSize);
+        // Missing first MAC byte because of overlong valid padding
         vectorList.add(new TrippleVector(new ByteArrayExplicitValueModification(new byte[0]),
                 new ByteArrayDeleteModification(0, 1), new ByteArrayExplicitValueModification(padding)));
-        padding = createPaddingBytes(DEFAULT_CIPHERTEXT_LENGTH - macSize + 1 - 1);
-        padding[0] ^= 0x80; // flip first padding byte last bit
+        // Missing last MAC byte because of overlong valid padding
         vectorList.add(new TrippleVector(new ByteArrayExplicitValueModification(new byte[0]),
-                new ByteArrayDeleteModification(0, 1), new ByteArrayExplicitValueModification(padding)));
-        padding = createPaddingBytes(DEFAULT_CIPHERTEXT_LENGTH - macSize + 1 - 1);
-        padding[(DEFAULT_CIPHERTEXT_LENGTH - macSize + 1) / 2] ^= 0x8; // flip middle padding byte
-        // middle bit
-        vectorList.add(new TrippleVector(new ByteArrayExplicitValueModification(new byte[0]),
-                new ByteArrayDeleteModification(0, 1), new ByteArrayExplicitValueModification(padding)));
-        padding = createPaddingBytes(DEFAULT_CIPHERTEXT_LENGTH - macSize + 1 - 1);
-        padding[(DEFAULT_CIPHERTEXT_LENGTH - macSize + 1 - 1)] ^= 0x01; // flip last padding byte first
-        // bit
-        vectorList.add(new TrippleVector(new ByteArrayExplicitValueModification(new byte[0]),
-                new ByteArrayDeleteModification(0, 1), new ByteArrayExplicitValueModification(padding)));
+                new ByteArrayDeleteModification((macSize - 1), 1), new ByteArrayExplicitValueModification(padding)));
         return vectorList;
     }
\ No newline at end of file
