diff --git a/old_product.java b/new_product.java
index 2481ab4..a69f8bd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 @RequiresSession
-  public List<HostVersionEntity> findByClusterHostAndState(String  clusterName, String hostName, UpgradeState state) {
+  public List<HostVersionEntity> findByClusterHostAndState(String  clusterName, String hostName, RepositoryVersionState state) {
     final TypedQuery<HostVersionEntity> query = entityManagerProvider.get()
         .createNamedQuery("hostVersionByClusterHostnameAndState", HostVersionEntity.class);
     query.setParameter("clusterName", clusterName);
