diff --git a/old_product.java b/new_product.java
index 2504146..802699d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
 public LinkedHashMap<String, List<AlignmentInterval.Row>> packAlignments(
             Iterator<Alignment> iter,
             int end,
-            boolean pairAlignments,
             AlignmentTrack.RenderOptions renderOptions) {
 
         LinkedHashMap<String, List<AlignmentInterval.Row>> packedAlignments = new LinkedHashMap<String, List<Row>>();
+        boolean pairAlignments = renderOptions.isViewPairs() || renderOptions.isPairedArcView();
 
         if (iter == null || !iter.hasNext()) {
             return packedAlignments;
