diff --git a/old_test.java b/new_test.java
index e2b0e20..d5a3112 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,7 +17,7 @@
         int qualityThreshold = 0;
         int maxLevels = 1000;
 
-        Map<String, List<AlignmentInterval.Row>> result = (new AlignmentPacker()).packAlignments(iter, end, false,
+        Map<String, List<AlignmentInterval.Row>> result = (new AlignmentPacker()).packAlignments(iter, end,
                 new AlignmentTrack.RenderOptions());
         assertEquals(1, result.size());
         for (List<AlignmentInterval.Row> alignmentrows : result.values()) {
