diff --git a/old_product.java b/new_product.java
index 91988ca..99e2a6e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -29,19 +29,19 @@ public boolean manageTopology() {
         LOG.log(Level.FINE, "Topology: {0} to be {1}ed", new Object[]{topologyName, command});
 
         // build the runtime config
-        runtime = Config.newBuilder()
+        Config runtime = Config.newBuilder()
             .put(Keys.topologyName(), Context.topologyName(config))
             .put(Keys.schedulerStateManagerAdaptor(), adaptor)
             .build();
 
         // Create a ISchedulerClient basing on the config
-        ISchedulerClient schedulerClient = getSchedulerClient();
+        ISchedulerClient schedulerClient = getSchedulerClient(runtime);
         if (schedulerClient == null) {
           LOG.severe("Failed to initialize scheduler client");
           return false;
         }
 
-        isSuccessful = callRuntimeManagerRunner(schedulerClient);
+        isSuccessful = callRuntimeManagerRunner(runtime, schedulerClient);
       }
     } finally {
       // 3. Do post work basing on the result
