diff --git a/old_product.java b/new_product.java
index 3bbe386..c949d42 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,11 @@
-public static double GeometricCdf(int k, double p) {
+public static double geometricCdf(int k, double p) {
         if(k<=0 || p<0) {
             throw new IllegalArgumentException("All the parameters must be positive.");
         }
         
         double probabilitySum = 0.0;
         for(int i=1;i<=k;++i) {
-            probabilitySum += Geometric(i, p);
+            probabilitySum += geometric(i, p);
         }
         
         return probabilitySum;
