diff --git a/old_test.java b/new_test.java
index 877f70a..e5512dd 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void maximizeConcurrency() {
     Cache<Integer, Integer> c = Cache2kBuilder.of(Integer.class, Integer.class)
-      .maximizeConcurrency(true)
+      .boostConcurrency(true)
       .expireAfterWrite(5, TimeUnit.MINUTES)
       .loader(new CacheLoader<Integer, Integer>() {
         @Override
