diff --git a/old_product.java b/new_product.java
index 2dbdffa..cbb4915 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public final Cache2kBuilder<K,V> maximizeConcurrency(boolean f) {
-    config().setMaximizeConcurrency(f);
+public final Cache2kBuilder<K,V> boostConcurrency(boolean f) {
+    config().setBoostConcurrency(f);
     return this;
   }
\ No newline at end of file
