diff --git a/old_test.java b/new_test.java
index 2faa577..68981ba 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void testUpdate() throws Exception {
-    List<Metric> metrics = new ArrayList<Metric>(NUM_METRICS);
+    List<Metric> metrics = new ArrayList<>(NUM_METRICS);
     for (int i = 0; i < NUM_METRICS; i++) {
       metrics.add(new Metric("test", BasicTagList.EMPTY, System.currentTimeMillis(), VALUE));
     }
