diff --git a/old_product.java b/new_product.java
index 39581d9..cc1ba77 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public void update(List<Metric> metrics) {
     Preconditions.checkNotNull(metrics, "metrics");
     LOGGER.debug("received {} metrics", metrics.size());
-    final List<Metric> newMetrics = new ArrayList<Metric>(metrics.size());
+    final List<Metric> newMetrics = new ArrayList<>(metrics.size());
     for (Metric m : metrics) {
       if (isCounter(m)) {
         final MonitorConfig rateConfig = toRateConfig(m.getConfig());
