diff --git a/old_product.java b/new_product.java
index 20e6dfb..d0e42e6 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1 +1 @@
-String handleRequest(Map<String, String> pathParams, Map<String, String> queryParams, ActorGateway jobManager) throws Exception;
\ No newline at end of file
+FullHttpResponse handleRequest(Map<String, String> pathParams, Map<String, String> queryParams, ActorGateway jobManager) throws Exception;
\ No newline at end of file
