diff --git a/old_test.java b/new_test.java
index 05a294c..d294336 100644
--- a/old_test.java
+++ b/new_test.java
@@ -25,7 +25,7 @@
         val w = new CustomLocateBucketIndexer(KeyHashers.DEFAULT_HASHER, executorService(), hashToBuckets);
         val allKeyUpdates = new ArrayList<BucketUpdate.KeyUpdate>();
         bucketsToKeys.values().forEach(allKeyUpdates::addAll);
-        val bucketUpdates = w.groupByBucket(allKeyUpdates, null, new TimeoutTimer(TIMEOUT)).join();
+        val bucketUpdates = w.groupByBucket(null, allKeyUpdates, new TimeoutTimer(TIMEOUT)).join();
 
         Assert.assertEquals("Unexpected number of Bucket Updates.", bucketCount, bucketUpdates.size());
         for (BucketUpdate bu : bucketUpdates) {
