diff --git a/old_test.java b/new_test.java
index dcc81ff..a6536a8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
 	public void searchUser() {
 
-		SearchUserResponse response = accountWebServiceClient.searchUser(null, null);
+		SearchUserResult response = accountWebServiceClient.searchUser(null, null);
 
 		assertTrue(response.getUserList().size() >= 4);
 		assertEquals("Admin", response.getUserList().get(0).getName());
