diff --git a/old_test.java b/new_test.java
index 23c60b3..dd05c86 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,12 +2,11 @@
 	@Verifies(value = "should fail if any patient state has an end date before its start date", method = "validate(Object,Errors)")
 	public void validate_shouldFailIfAnyPatientStateHasAnEndDateBeforeItsStartDate() throws Exception {
 		PatientProgram program = Context.getProgramWorkflowService().getPatientProgram(1);
-		PatientState patientSate = program.getStates().iterator().next();
-		Assert.assertNotNull(patientSate);
+		PatientState patientState = program.getStates().iterator().next();
 		Calendar c = Calendar.getInstance();
-		patientSate.setStartDate(c.getTime());
+		patientState.setStartDate(c.getTime());
 		c.set(1970, 2, 1);//set to an old date
-		patientSate.setEndDate(c.getTime());
+		patientState.setEndDate(c.getTime());
 		
 		BindException errors = new BindException(program, "");
 		new PatientProgramValidator().validate(program, errors);
