diff --git a/old_product.java b/new_product.java
index 0aa65c7..bf2cfa5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public Dbl x(final ANode node) throws QueryException {
     final Geometry geo = geo(node, Q_GML_POINT);
     if(geo == null && checkGeo(node) != null)
-      throw GeoErrors.pointNeeded(node.qname().local());
+      throw GeoErrors.geoType(node.qname().local(), "Point");
 
     return Dbl.get(geo.getCoordinate().x);
   }
\ No newline at end of file
