diff --git a/old_test.java b/new_test.java
index fb710d3..959b0c4 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,5 +4,5 @@
     mMockHdfsUnderFileSystem.prepareConfiguration("", conf);
     Assert.assertEquals("org.apache.hadoop.hdfs.DistributedFileSystem", conf.get("fs.hdfs.impl"));
     Assert.assertFalse(conf.getBoolean("fs.hdfs.impl.disable.cache", false));
-    Assert.assertNotNull(conf.get(Constants.UNDERFS_HDFS_CONFIGURATION));
+    Assert.assertNotNull(conf.get(PropertyKey.UNDERFS_HDFS_CONFIGURATION));
   }
\ No newline at end of file
