diff --git a/old_test.java b/new_test.java
index bea7b55..e54ce0a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -20,14 +20,16 @@
         service.startAsync();
         service.awaitRunning();
 
-        String table = null;
+        TransformResponse response = null;
         try {
-            table = service.execute(request);
+            response = service.execute(request);
         }
         finally {
             service.stopAsync();
         }
 
+        Assert.assertEquals(response.getStatus(), TransformResponse.Status.SUCCESS);
+
         // Test eval arguments
         ArgumentCaptor<String> evalScript = ArgumentCaptor.forClass(String.class);
         ArgumentCaptor<List> evalBindings = ArgumentCaptor.forClass(List.class);
@@ -44,5 +46,5 @@
         Assert.assertEquals("spark_shell_temp", bindings.get(0).value());
         Assert.assertEquals("tableName", bindings.get(1).name());
         Assert.assertEquals("String", bindings.get(1).tpe());
-        Assert.assertEquals(table, bindings.get(1).value());
+        Assert.assertEquals(response.getTable(), bindings.get(1).value());
     }
\ No newline at end of file
