diff --git a/old_product.java b/new_product.java
index 8e6a051..23b9db5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -16,6 +16,7 @@ public void configure(Properties props)
 		String username = props.getProperty(CONFIG_CONNECTION_USERNAME, "username not set");
 		String password = props.getProperty(CONFIG_CONNECTION_PASSWORD, "password not set");
 		String connectionHookClass = props.getProperty(CONFIG_CONNECTION_HOOK_CLASS);
+		String initSQL = props.getProperty(CONFIG_INIT_SQL);
 
 
 		// Remember Isolation level
@@ -39,6 +40,7 @@ public void configure(Properties props)
 			this.config.setConnectionTestStatement(connectionTestStatement);
 			this.config.setPreparedStatementsCacheSize(preparedStatementCacheSize);
 			this.config.setStatementsCachedPerConnection(statementsCachedPerConnection);
+			this.config.setInitSQL(initSQL);
 			
 			if (connectionHookClass != null){
 				Object hookClass = Class.forName(connectionHookClass).newInstance();
