diff --git a/old_test.java b/new_test.java
index 683b2f7..e5bb7c9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,7 @@
     public void testSetItemId()
     {
         int itemId = 55;
-        mv.setItemId(itemId);
-        assertThat("testSetItemId 0", mv.getItemId(), equalTo(itemId));
+        mv.setResourceId(itemId);
+        mv.setResourceTypeId(Constants.ITEM);
+        assertThat("testSetItemId 0", mv.getResourceId(), equalTo(itemId));
     }
\ No newline at end of file
