diff --git a/old_test.java b/new_test.java
index d82b047..78cfaf2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,7 @@
         requestJSONObject.put(User.USER_EMAIL, "test@gmail.com");
         requestJSONObject.put(User.USER_NAME, "Admin");
         requestJSONObject.put(User.USER_PASSWORD, "pass");
+        requestJSONObject.put(Keys.LOCALE, Preference.Default.DEFAULT_LANGUAGE);
 
         initService.init(requestJSONObject);
 
