diff --git a/old_product.java b/new_product.java
index e655e8f..e29cbf7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,6 +5,7 @@ public void init() throws Exception {
         requestJSONObject.put(User.USER_EMAIL, "test@b3log.org");
         requestJSONObject.put(User.USER_NAME, "Admin");
         requestJSONObject.put(User.USER_PASSWORD, "pass");
+        requestJSONObject.put(Keys.LOCALE, Preference.Default.DEFAULT_LANGUAGE);
 
         initService.init(requestJSONObject);
 
