diff --git a/old_product.java b/new_product.java
index 317f730..3d829c4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -12,7 +12,7 @@ void onNodeMetrics(UUID nodeId, Object msg, byte plc) {
                 headers.setContentType(MimeTypeUtils.APPLICATION_OCTET_STREAM);
                 headers.setDestination(buildMetricsDest(res.clusterId()));
 
-                mgr.getSession().send(headers, res.body());
+                mgr.send(headers, res.body());
             }
             catch (Throwable e) {
                 log.error("Failed to send metrics to GMC", e);
