diff --git a/old_test.java b/new_test.java
index 863ed62..92a4b3c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,14 +1,14 @@
 @Test
     public void onNodeMetrics() {
-        MetricsService srvc = new MetricsService(getMockContext(), getMockWebSocketManager());
+        MetricsService srvc = new MetricsService(getMockContext(), mgr);
 
-        MetricResponse metricResponse = getMetricResponse();
-        srvc.onNodeMetrics(UUID.randomUUID(), metricResponse, SYSTEM_POOL);
+        MetricResponse metricRes = getMetricResponse();
+        srvc.onNodeMetrics(UUID.randomUUID(), metricRes, SYSTEM_POOL);
 
         ArgumentCaptor<StompHeaders> destCaptor = ArgumentCaptor.forClass(StompHeaders.class);
         ArgumentCaptor<Object> payloadCaptor = ArgumentCaptor.forClass(Object.class);
-        verify(ses, times(1)).send(destCaptor.capture(), payloadCaptor.capture());
+        verify(mgr, times(1)).send(destCaptor.capture(), payloadCaptor.capture());
 
-        Assert.assertEquals(StompDestinationsUtils.buildMetricsDest(metricResponse.clusterId()), destCaptor.getValue().getDestination());
+        Assert.assertEquals(StompDestinationsUtils.buildMetricsDest(metricRes.clusterId()), destCaptor.getValue().getDestination());
         Assert.assertArrayEquals(new byte[]{1, 2, 3, 4}, (byte[]) payloadCaptor.getValue());
     }
\ No newline at end of file
