diff --git a/old_product.java b/new_product.java
index f5b0ff7..a50f232 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,8 +15,7 @@ protected static History parseChanges(Reader fileHistory) throws IOException {
                 int year = Integer.parseInt(matcher.group(2));
                 int month = Integer.parseInt(matcher.group(3));
                 int day = Integer.parseInt(matcher.group(4));
-                Calendar calendar = new GregorianCalendar(year, month, day);
-                entry.setDate(calendar.getTime());
+                entry.setDate(newDate(year, month, day));
                 entry.setAuthor(matcher.group(5));
                 entry.setMessage(matcher.group(6).trim());
                 entry.setActive(true);
