diff --git a/old_test.java b/new_test.java
index e872d88..eb0c714 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,7 @@
 @Test
   public void testTranslate() {
-    final String result = "El equipo es increíble IBM Watson";
-    testTranslationResult(text, result, service.translate(text, ENGLISH_TO_SPANISH).execute());
-    testTranslationResult(text, result, service.translate(text, Language.ENGLISH, Language.SPANISH).execute());
+    for(String text : texts) {
+      testTranslationResult(text, translations.get(text), service.translate(text, ENGLISH_TO_SPANISH).execute());
+      testTranslationResult(text, translations.get(text), service.translate(text, ENGLISH, SPANISH).execute());
+    }
   }
\ No newline at end of file
