diff --git a/old_product.java b/new_product.java
index dc870c9..48e3dd8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,3 @@
 public ServiceCall<TranslationResult> translate(final String text, final String modelId) {
-    Validator.isTrue(modelId != null && !modelId.isEmpty(), "modelId cannot be null or empty");
-    return translateRequest(text, modelId, null, null);
+    return translate(Collections.singletonList(text), modelId);
   }
\ No newline at end of file
