diff --git a/old_test.java b/new_test.java
index 0239914..3bc770c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
 	public void fullViewLeft_calibrated() {
 
-		CameraPinholeRadial param =
-				new CameraPinholeRadial().fsetK(300, 320, 0, 150, 130, width, height).fsetRadial(0.1,1e-4);
+		CameraPinholeBrown param =
+				new CameraPinholeBrown().fsetK(300, 320, 0, 150, 130, width, height).fsetRadial(0.1,1e-4);
 
 		// do nothing rectification
 		FMatrixRMaj rect1 = CommonOps_FDRM.identity(3);
