diff --git a/old_test.java b/new_test.java
index 8fc40c1..4b9a14b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,7 +14,7 @@
         instance.setMaximumIterations(500);
 
         Vec w = new DenseVector(x0.length());
-        instance.optimize(1e-8, w, x0, f, fp, null);
+        instance.optimize(1e-8, w, x0, f, fp);
 
         for (int i = 0; i < w.length(); i++)
             assertEquals(1.0, w.get(i), 1e-2);
