diff --git a/old_product.java b/new_product.java
index 46bd25c..c959403 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,27 +1,35 @@
-public static int getSampleNumbers(Integer numPartitions, long totalNumberOfRecords) throws Exception {
-    	long sampleNumbers;
-    	/*
-    	 * If the input RDD is too small, Geospark will use the entire RDD instead of taking samples.
-    	 */
-    	if(totalNumberOfRecords>=1000)
+public static int getSampleNumbers(Integer numPartitions, long totalNumberOfRecords, long givenSampleNumbers) throws Exception{
+    	Long sampleNumber = new Long(0);
+
+    	if(givenSampleNumbers>0)
     	{
-    		sampleNumbers = totalNumberOfRecords / 100;
+    		// This means that the user manually specifies the sample number
+    		sampleNumber = givenSampleNumbers;
+    		return sampleNumber.intValue();
     	}
     	else
     	{
-    		sampleNumbers = totalNumberOfRecords;
+    		// Follow GeoSpark internal sampling rule
+        	/*
+        	 * If the input RDD is too small, Geospark will use the entire RDD instead of taking samples.
+        	 */
+        	if(totalNumberOfRecords>=1000)
+        	{
+        		sampleNumber = totalNumberOfRecords / 100;
+        	}
+        	else
+        	{
+        		sampleNumber = totalNumberOfRecords;
+        	}
+        	
+    		if(sampleNumber > Integer.MAX_VALUE) {
+    			sampleNumber = new Long(Integer.MAX_VALUE);
+    		}
+            if(sampleNumber < 2*numPartitions ) {
+                // Partition size is too big. Should throw exception for this.
+                throw new Exception("[RDDSampleUtils][getSampleNumbers] Too many RDD partitions. Call SpatialRDD.setSampleNumber() to manually increase sample or make partitionNum less than "+sampleNumber/2);
+            }
+            return sampleNumber.intValue();
     	}
-    	
-		if(sampleNumbers > Integer.MAX_VALUE) {
-			sampleNumbers = Integer.MAX_VALUE;
-		}
-        int result=(int)sampleNumbers;
-        // Partition size is too big. Should throw exception for this.
-        
-        if(sampleNumbers < 2*numPartitions ) {
-            throw new Exception("[RDDSampleUtils][getSampleNumbers] Too many RDD partitions. Please make this RDD's partitions less than "+sampleNumbers/2);
-        }
-        
-        return result;
 
 	}
\ No newline at end of file
