diff --git a/old_test.java b/new_test.java
index 4176064..a516a35 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
     public void testGetSampleNumbers() throws Exception {
-        assertEquals(10, RDDSampleUtils.getSampleNumbers(2, 10));
-        assertEquals(100, RDDSampleUtils.getSampleNumbers(2, 100));
-        assertEquals(10, RDDSampleUtils.getSampleNumbers(5, 1000));
-        assertEquals(100, RDDSampleUtils.getSampleNumbers(5, 10000));
-        assertEquals(100, RDDSampleUtils.getSampleNumbers(5, 10001));
-        assertEquals(1000, RDDSampleUtils.getSampleNumbers(5, 100011));
-        assertEquals(1000, RDDSampleUtils.getSampleNumbers(6, 100011));
+        assertEquals(10, RDDSampleUtils.getSampleNumbers(2, 10,-1));
+        assertEquals(100, RDDSampleUtils.getSampleNumbers(2, 100,-1));
+        assertEquals(10, RDDSampleUtils.getSampleNumbers(5, 1000,-1));
+        assertEquals(100, RDDSampleUtils.getSampleNumbers(5, 10000,-1));
+        assertEquals(100, RDDSampleUtils.getSampleNumbers(5, 10001,-1));
+        assertEquals(1000, RDDSampleUtils.getSampleNumbers(5, 100011,-1));
+        assertEquals(99, RDDSampleUtils.getSampleNumbers(6, 100011,99));
     }
\ No newline at end of file
