diff --git a/old_test.java b/new_test.java
index 0c56c20..43ba545 100644
--- a/old_test.java
+++ b/new_test.java
@@ -22,14 +22,16 @@
       CharSequence methodSource = state.getSourceForNode((JCMethodDecl) methodTree);
       if (methodSource != null) {
         String methodString = "@Test\n" + methodSource.toString().replaceFirst(" static ", " ");
-        Fix fix = new SuggestedFix()
+        Fix fix = SuggestedFix.builder()
             .addImport(JUNIT4_TEST_ANNOTATION)
-            .replace(methodTree, methodString);
+            .replace(methodTree, methodString)
+            .build();
         return describeMatch(methodTree, fix);
       }
     }
-    Fix fix = new SuggestedFix()
+    Fix fix = SuggestedFix.builder()
         .addImport(JUNIT4_TEST_ANNOTATION)
-        .prefixWith(methodTree, "@Test\n");
+        .prefixWith(methodTree, "@Test\n")
+        .build();
     return describeMatch(methodTree, fix);
   }
\ No newline at end of file
