diff --git a/old_product.java b/new_product.java
index 5989cbc..5057d9c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@
       return Description.NO_MATCH;
     }
 
-    SuggestedFix fix = new SuggestedFix();
+    SuggestedFix.Builder fix = SuggestedFix.builder();
     JCClassDecl cls = (JCClassDecl) state.findEnclosing(ClassTree.class);
 
     if ((cls.getModifiers().flags & ENUM) != 0) {
@@ -48,5 +48,5 @@
       }
     }
 
-    return describeMatch(methodTree, fix);
+    return describeMatch(methodTree, fix.build());
   }
\ No newline at end of file
