diff --git a/old_product.java b/new_product.java
index be5ea14..0a7017f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @Deprecated
     public static String toHexStringUpperCase(String original,String charsetName){
         String hexStringUpperCase = ByteUtil.bytesToHexStringUpperCase(getBytes(original, charsetName));
-        LOGGER.debug("original:{},hexStringUpperCase:{}", original, hexStringUpperCase);
+        LOGGER.debug("original:[{}],hexStringUpperCase:[{}]", original, hexStringUpperCase);
         return hexStringUpperCase;
     }
\ No newline at end of file
