diff --git a/old_product.java b/new_product.java
index d0c18e8..d1535de 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,9 @@
-protected static String contentType(final String query) throws IOException {
+protected static MediaType contentType(final String query) throws IOException {
     final IOUrl url = new IOUrl(REST_ROOT + query);
     final HttpURLConnection conn = (HttpURLConnection) url.connection();
     try {
       read(conn.getInputStream());
-      return conn.getContentType();
+      return new MediaType(conn.getContentType());
     } catch(final IOException ex) {
       throw error(conn, ex);
     } finally {
