diff --git a/old_test.java b/new_test.java
index c91c972..aba4ba5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public void contentType() {
     query(_DB_ADD.args(NAME, "\"<a/>\"", "xml"));
     query(_DB_STORE.args(NAME, "raw", "bla"));
-    query(_DB_CONTENT_TYPE.args(NAME, "xml"), MimeTypes.APP_XML);
-    query(_DB_CONTENT_TYPE.args(NAME, "raw"), MimeTypes.APP_OCTET);
+    query(_DB_CONTENT_TYPE.args(NAME, "xml"), MediaType.APPLICATION_XML.toString());
+    query(_DB_CONTENT_TYPE.args(NAME, "raw"), MediaType.APPLICATION_OCTET_STREAM.toString());
     error(_DB_CONTENT_TYPE.args(NAME, "test"), WHICHRES_X);
   }
\ No newline at end of file
