diff --git a/old_product.java b/new_product.java
index c4aa6f7..73f9d91 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public synchronized User[] users(final Users users) {
     final ArrayList<User> al = new ArrayList<>();
     for(final User user : list) {
-      if(users == null || users.get(user.name) != null) al.add(user);
+      if(users == null || users.get(user.name()) != null) al.add(user);
     }
     return al.toArray(new User[al.size()]);
   }
\ No newline at end of file
