diff --git a/old_test.java b/new_test.java
index 842aece..f5122f8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
 	public void contour() {
-		ImageUInt8 input = new ImageUInt8(10,12);
+		GrayU8 input = new GrayU8(10,12);
 		ImageMiscOps.fillRectangle(input,1,2,3,4,5);
 		input.set(9,11,1);
 
-		ImageSInt32 output = new ImageSInt32(10,12);
-		ImageSInt32 expected = new ImageSInt32(10,12);
+		GrayS32 output = new GrayS32(10,12);
+		GrayS32 expected = new GrayS32(10,12);
 		ImageMiscOps.fillRectangle(expected,1,2,3,4,5);
 		expected.set(9,11,2);
 
