diff --git a/old_product.java b/new_product.java
index a4f85e6..751dd89 100644
--- a/old_product.java
+++ b/new_product.java
@@ -23,7 +23,7 @@
 
                     //TODO - can we get more evidence from the parent? EAR contains module name, etc.
                     //analyze the dependency (i.e. extract files) if it is a supported type.
-                    if (this.accept(d.getActualFile()) && scanDepth < MAX_SCAN_DEPTH) {
+                    if (this.accept(d.getActualFile()) && scanDepth < maxScanDepth) {
                         scanDepth += 1;
                         analyze(d, engine);
                         scanDepth -= 1;
