diff --git a/old_test.java b/new_test.java
index d4f72aa..aac4453 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,6 +8,7 @@
         versions.addEvidence("other", "Implementation-Version", "1.2.3", Confidence.HIGHEST);
 
         VersionFilterAnalyzer instance = new VersionFilterAnalyzer();
+        instance.initializeSettings(getSettings());
 
         instance.analyzeDependency(dependency, null);
         assertEquals(3, versions.size());
