diff --git a/old_test.java b/new_test.java
index 3fe3dbc..fd98f68 100644
--- a/old_test.java
+++ b/new_test.java
@@ -17,6 +17,6 @@
 		Thread.sleep(10);
 		
 		expectedException.expect(APIException.class);
-		expectedException.expectMessage("Cannot unvoid a discontinuation order if the previous order is no longer active");
+		expectedException.expectMessage("Order.action.cannot.unvoid");
 		orderService.unvoidOrder(order);
 	}
\ No newline at end of file
