diff --git a/old_product.java b/new_product.java
index 014fac9..a97b071 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,7 @@ public Order unvoidOrder(Order order) throws APIException {
 				final String action = DISCONTINUE == order.getAction() ? "discontinuation" : "revision";
 				throw new APIException("Cannot unvoid a " + action + " order if the previous order is no longer active");
 			}
-			stopOrder(previousOrder, order.getDateActivated());
+			stopOrder(previousOrder, aMomentBefore(order.getDateActivated()));
 		}
 		
 		return saveOrderInternal(order, null);
