diff --git a/old_test.java b/new_test.java
index 674e617..2458c87 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,6 +2,7 @@
 	public void testIsConnectionHandleAlive() throws SQLException {
 		// Test 1: Normal case (+ without connection test statement)
 		expect(mockConfig.getConnectionTestStatement()).andReturn(null).once();
+		mockConnection.logicallyClosed=new AtomicBoolean();
 		expect(mockConnection.getMetaData()).andReturn(mockDatabaseMetadata).once();
 		expect(mockDatabaseMetadata.getTables((String)anyObject(), (String)anyObject(), (String)anyObject(), (String[])anyObject())).andReturn(mockResultSet).once();
 		mockResultSet.close();
