diff --git a/old_product.java b/new_product.java
index 3db4658..069e323 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,9 @@
 public boolean isConnectionHandleAlive(ConnectionHandle connection) {
 		Statement stmt = null;
 		boolean result = false;
-		boolean logicallyClosed = connection.logicallyClosed;
+		boolean logicallyClosed = connection.logicallyClosed.get();
 		try {
-			if (logicallyClosed){
-				connection.logicallyClosed = false; // avoid checks later on if it's marked as closed.
-			}
+			connection.logicallyClosed.compareAndSet(true, false); // avoid checks later on if it's marked as closed.
 			String testStatement = this.config.getConnectionTestStatement();
 			ResultSet rs = null;
 
@@ -27,7 +25,7 @@ public boolean isConnectionHandleAlive(ConnectionHandle connection) {
 			// connection must be broken!
 			result = false;
 		} finally {
-			connection.logicallyClosed = logicallyClosed;
+			connection.logicallyClosed.set(logicallyClosed);
 			connection.setConnectionLastResetInMs(System.currentTimeMillis());
 			result = closeStatement(stmt, result);
 		}
