diff --git a/old_test.java b/new_test.java
index 3f28fd9..49c342e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,8 +3,14 @@
         // Test to update an existing stream
         when(mockControllerService.alterStream(any())).thenReturn(updateStreamStatus);
         response = target(resourceURI).request().async().put(Entity.json(updateStreamRequest));
+        assertEquals("Update Stream Status", 200, response.get().getStatus());
+        streamResponseActual = response.get().readEntity(StreamProperty.class);
+        testExpectedVsActualObject(streamResponseExpected, streamResponseActual);
+
+        // Test sending extra fields in the request object to check if json parser can handle it.
+        response = target(resourceURI).request().async().put(Entity.json(createStreamRequest));
+        assertEquals("Update Stream Status", 200, response.get().getStatus());
         streamResponseActual = response.get().readEntity(StreamProperty.class);
-        assertEquals("Update Stream Status", 201, response.get().getStatus());
         testExpectedVsActualObject(streamResponseExpected, streamResponseActual);
 
         // Test to update an non-existing stream
