diff --git a/old_test.java b/new_test.java
index 7911820..9e405d2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,12 +1,7 @@
 @Test(groups={"tck"})
-    public void test_getRules() {
-        ZoneRulesProvider group = ZoneRulesProvider.getProvider("TZDB");
-        ZoneRules rules = group.getRules("Europe/London", "2008i");
-        assertEquals(rules.getTransitionRules().size(), 2);
-        assertEquals(rules.getTransitionRules().get(0).getDayOfWeek(), DayOfWeek.SUNDAY);
-        assertEquals(rules.getTransitionRules().get(0).getDayOfMonthIndicator(), 25);
-        assertEquals(rules.getTransitionRules().get(0).getMonth(), Month.MARCH);
-        assertEquals(rules.getTransitionRules().get(1).getDayOfWeek(), DayOfWeek.SUNDAY);
-        assertEquals(rules.getTransitionRules().get(1).getDayOfMonthIndicator(), 25);
-        assertEquals(rules.getTransitionRules().get(1).getMonth(), Month.OCTOBER);
+    public void test_getRules_String() {
+        ZoneRules rules = ZoneRulesProvider.getRules("Europe/London");
+        assertNotNull(rules);
+        ZoneRules rules2 = ZoneRulesProvider.getRules("Europe/London");
+        assertEquals(rules2, rules);
     }
\ No newline at end of file
