diff --git a/old_test.java b/new_test.java
index 98aedd6..39f7964 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,9 +1,9 @@
 @Test
   public void testDelete() throws QueryException, BaseXException {
-    final String fun = check(Function.DELETE);
+    final String fun = check(Function.DBDELETE);
 
     new Add("etc/test/dir", "docs", "test").execute(CONTEXT);
 
-    query(fun + "('db/test')", "");
+    query(fun + "('db', 'test')", "");
     query("count(collection('db/test')) eq 0", "true");
   }
\ No newline at end of file
