diff --git a/old_product.java b/new_product.java
index c2b609a..fb5edd7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public EncounterType getEncounterType(String name) throws DAOException {
 		Criteria crit = sessionFactory.getCurrentSession().createCriteria(EncounterType.class);
-		crit.add(Expression.eq("retired", false));
-		crit.add(Expression.eq("name", name));
+		crit.add(Restrictions.eq("retired", false));
+		crit.add(Restrictions.eq("name", name));
 		EncounterType encounterType = (EncounterType) crit.uniqueResult();
 		
 		return encounterType;
