diff --git a/old_product.java b/new_product.java
index 22e5c92..e29a71a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,6 +15,11 @@ public static void regionPixelId_to_Compact(ImageSInt32 graph, GrowQueue_I32 seg
 		}
 		// need to do some clean up since the above approach doesn't work for the roots
 		for( int i = 0; i < segmentId.size; i++ ) {
-			output.data[segmentId.data[i]] = i;
+			int indexGraph = segmentId.data[i] - graph.startIndex;
+
+			int x = indexGraph%graph.stride;
+			int y = indexGraph/graph.stride;
+
+			output.data[output.startIndex + y*output.stride + x] = i;
 		}
 	}
\ No newline at end of file
