diff --git a/old_test.java b/new_test.java
index 8108832..ac71905 100644
--- a/old_test.java
+++ b/new_test.java
@@ -35,7 +35,7 @@
         "'window':map {'size':3} }"), "Databases and XML");
 
     // check buggy options
-    error(_FT_SEARCH.args(NAME, "x", " map { 'x':'y' }"), Err.INVALIDOPT);
-    error(_FT_SEARCH.args(NAME, "x", " map { 'mode':'' }"), Err.INVALIDOPT);
-    error(_FT_SEARCH.args(NAME, "x", " 1"), Err.ELMMAPTYPE);
+    error(_FT_SEARCH.args(NAME, "x", " map { 'x':'y' }"), Err.INVALIDOPT_X);
+    error(_FT_SEARCH.args(NAME, "x", " map { 'mode':'' }"), Err.INVALIDOPT_X);
+    error(_FT_SEARCH.args(NAME, "x", " 1"), Err.ELMMAP_X_X_X);
   }
\ No newline at end of file
