diff --git a/old_product.java b/new_product.java
index cae747a..f8c3e16 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,10 @@
 private Iter search(final QueryContext qc) throws QueryException {
     final Data data = checkData(qc);
     final Value terms = qc.value(exprs[1]);
-    final FTOptions opts = checkOptions(2, Q_OPTIONS, new FTOptions(), qc);
+    final FTOptions opts = toOptions(2, Q_OPTIONS, new FTOptions(), qc);
 
     final IndexContext ic = new IndexContext(data, false);
-    if(!data.meta.ftxtindex) throw BXDB_INDEX.get(info, data.meta.name,
+    if(!data.meta.ftxtindex) throw BXDB_INDEX_X.get(info, data.meta.name,
         IndexType.FULLTEXT.toString().toLowerCase(Locale.ENGLISH));
 
     final FTOpt opt = new FTOpt().copy(data.meta);
