diff --git a/old_product.java b/new_product.java
index 74075d1..a8e226f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -2,7 +2,7 @@
     public boolean containsKey(Object key)
     {
         if (key==null)
-            return _nullEntry!=null;
+            return false;
         return
             getEntry(key.toString(),0,key==null?0:key.toString().length())!=null;
     }
\ No newline at end of file
