diff --git a/old_test.java b/new_test.java
index 2503fc4..9c96198 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,7 +6,6 @@
         assertTrue(m5.containsKey("bbb"));
         assertTrue(!m5.containsKey("xyz"));
         
-        assertTrue(m5i.containsKey(null));
         assertTrue(m5i.containsKey("abc"));
         assertTrue(m5i.containsKey("aBc"));
         assertTrue(m5i.containsKey("ABC"));
