diff --git a/old_product.java b/new_product.java
index 34440e5..a10aa2e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,11 @@
 @Override
-  public void distribute(JSONObject message, MessageRoute route, Context context) throws ExecutionException {
-    getBuilder(route, context).apply(message);
+  public void distribute(JSONObject message, long timestamp, MessageRoute route, Context context) {
+    try {
+      ProfileBuilder builder = getBuilder(route, context);
+      builder.apply(message, timestamp);
+
+    } catch(ExecutionException e) {
+      LOG.error("Unexpected error", e);
+      throw new RuntimeException(e);
+    }
   }
\ No newline at end of file
