diff --git a/old_product.java b/new_product.java
index 35aad70..a1c6ffe 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1 @@
-@IgniteAsyncSupported
-    public void loadCache(@Nullable IgniteBiPredicate<K, V> p, @Nullable Object... args) throws CacheException;
\ No newline at end of file
+public void loadCache(IgniteBiPredicate<K, V> p, Object... args) throws CacheException;
\ No newline at end of file
