diff --git a/old_product.java b/new_product.java
index f06efae..44ee92a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -37,9 +37,10 @@ private GridSqlMerge parseMerge(Merge merge) {
 
             int colId = intoTbl.getColumn(colName).getColumnId();
 
-            if (!rowDesc.isKeyColumn(colId) && !F.eq(colName, rowDesc.type().affinityKey()))
+            if (!rowDesc.isKeyColumn(colId) && !F.eq(colName, rowDesc.type().affinityKey())) {
                 throw new IgniteSQLException("Invalid column name in KEYS clause of MERGE - it may include only " +
                     "key and/or affinity columns: " + colName, IgniteQueryErrorCode.PARSING);
+            }
 
             keys[i] = new GridSqlColumn(srcKeys[i], tbl, null, null, colName);
         }
@@ -52,9 +53,15 @@ private GridSqlMerge parseMerge(Merge merge) {
             for (Expression[] srcRow : srcRows) {
                 GridSqlElement[] row = new GridSqlElement[srcRow.length];
 
-                for (int i = 0; i < srcRow.length; i++)
+                for (int i = 0; i < srcRow.length; i++) {
                     row[i] = parseExpression(srcRow[i], false);
 
+                    if (row[i] == null) {
+                        throw new IgniteSQLException("DEFAULT values are unsupported for MERGE.",
+                            IgniteQueryErrorCode.UNSUPPORTED_OPERATION);
+                    }
+                }
+
                 rows.add(row);
             }
             res.rows(rows);
