diff --git a/old_product.java b/new_product.java
index 99d25bc..747b0a0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,23 +6,9 @@ public void validate(Object obj, Errors errors) {
 			errors.rejectValue("order", "error.general");
 		} else {
 			ValidationUtils.rejectIfEmpty(errors, "concept", "error.null");
-			if (order.getDuration() != null) {
-				ValidationUtils.rejectIfEmpty(errors, "durationUnits", "DrugOrder.add.error.missingDurationUnits");
-			}
-			
-			if (order.getQuantity() != null) {
-				ValidationUtils.rejectIfEmpty(errors, "quantityUnits", "DrugOrder.add.error.missingQuantityUnits");
-			}
-			
-			if (order.getStrength() != null) {
-				ValidationUtils.rejectIfEmpty(errors, "strengthUnits", "DrugOrder.add.error.missingStrengthUnits");
-			}
-			
-			if (order.getDose() != null) {
-				ValidationUtils.rejectIfEmpty(errors, "doseUnits", "DrugOrder.add.error.missingDoseUnits");
-			}
-			
 			// for the following elements Order.hbm.xml says: not-null="true"
+			ValidationUtils.rejectIfEmpty(errors, "prn", "error.null");
+			ValidationUtils.rejectIfEmpty(errors, "complex", "error.null");
 			ValidationUtils.rejectIfEmpty(errors, "drug", "error.null");
 			if (order.getDrug() != null) {
 				ValidationUtils.rejectIfEmpty(errors, "drug.concept", "error.null");
@@ -35,5 +21,8 @@ public void validate(Object obj, Errors errors) {
 					
 				}
 			}
+			if (order.getDose() != null || order.getEquivalentDailyDose() != null || order.getQuantity() != null) {
+				ValidationUtils.rejectIfEmpty(errors, "units", "DrugOrder.error.unitsNotSetWhenDoseOrQuantitySpecified");
+			}
 		}
 	}
\ No newline at end of file
