diff --git a/old_test.java b/new_test.java
index 54024a5..fb56b64 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,6 +5,7 @@
 		;
 		order.setConcept(Context.getConceptService().getConcept(88));
 		order.setPatient(Context.getPatientService().getPatient(2));
+		order.setOrderType(Context.getOrderService().getOrderType(1));
 		Calendar cal = Calendar.getInstance();
 		cal.set(Calendar.DAY_OF_MONTH, cal.get(Calendar.DAY_OF_MONTH) - 1);
 		order.setStartDate(new Date());
