diff --git a/old_product.java b/new_product.java
index f6d5341..6685f26 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,7 +7,7 @@ public void abort()
         try
         {
             // Rollback if we have a database connection, and it is NOT Read Only
-            if (isValid() && !isReadOnly())
+            if (isValid())
             {
                 dbConnection.rollback();
             }
