diff --git a/old_product.java b/new_product.java
index f430b52..e7e2beb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,9 @@
 public static <T> void shuffle(T[] array) {
         //Implementing Fisher-Yates shuffle
         T tmp;
+        Random rnd = RandomGenerator.getThreadLocalRandom();
         for (int i = array.length - 1; i > 0; --i) {
-            int index = RandomGenerator.nextInt(i + 1);
+            int index = rnd.nextInt(i + 1);
             
             tmp = array[index];
             array[index] = array[i];
