diff --git a/old_test.java b/new_test.java
index a5002ac..0be0c3c 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
 		final AbstractInvokable memOwner = new DummyInvokable();
 
 		// create the reader
-		reader = new CollectionReader<PactInteger>(objects);
+		reader = new CollectionIterator<PactInteger>(objects);
 		// create the resettable Iterator
 		SpillingResettableIterator<PactInteger> iterator = new SpillingResettableIterator<PactInteger>(memman, ioman,
 			reader, SpillingResettableIterator.MIN_BUFFER_SIZE * SpillingResettableIterator.MINIMUM_NUMBER_OF_BUFFERS,
