diff --git a/old_product.java b/new_product.java
index 9847bc9..1500685 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,7 @@
 public void addCommand(final Command command)
-            throws GenieException {
+            throws GeniePreconditionException {
         if (command == null) {
-            throw new GenieException(
-                    HttpURLConnection.HTTP_BAD_REQUEST,
-                    "No command entered unable to add.");
+            throw new GeniePreconditionException("No command entered unable to add.");
         }
 
         if (this.commands == null) {
