diff --git a/old_product.java b/new_product.java
index fa63ea3..3548f94 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,5 +3,5 @@
    @Produces(RELOCATE_VM_PARAMS)
    @Consumes(TASK)
    @JAXBResponseParser
-   ListenableFuture<Task> relocate(@EndpointParam URI vAppURI,
-                                   @BinderParam(BindToXMLPayload.class) RelocateParams params);
\ No newline at end of file
+   ListenableFuture<Task> relocateVm(@EndpointParam URI vAppURI,
+                                     @BinderParam(BindToXMLPayload.class) RelocateParams params);
\ No newline at end of file
