diff --git a/old_test.java b/new_test.java
index 60e3e3e..a287ea7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,6 @@
       RelocateParams params = RelocateParams.builder().datastore(Reference.builder().href(datastore.getHref()).build()).build();
 
       // The method under test
-      Task relocate = vAppClient.relocate(vApp.getHref(), params);
+      Task relocate = vAppClient.relocateVm(vApp.getHref(), params);
       assertTrue(retryTaskSuccess.apply(relocate), String.format(TASK_COMPLETE_TIMELY, "relocate"));
    }
\ No newline at end of file
