diff --git a/old_product.java b/new_product.java
index a71bcde..04c96df 100644
--- a/old_product.java
+++ b/new_product.java
@@ -47,7 +47,9 @@ public static int getPrefixLength(final String fileName) {
             }
             posUnix = posUnix == NOT_FOUND ? posWin : posUnix;
             posWin = posWin == NOT_FOUND ? posUnix : posWin;
-            return Math.min(posUnix, posWin) + 1;
+            int pos = Math.min(posUnix, posWin) + 1;
+            String hostnamePart = fileName.substring(2, pos - 1);
+            return isValidHostName(hostnamePart) ? pos : NOT_FOUND;
         } else {
             return isSeparator(ch0) ? 1 : 0;
         }
