diff --git a/old_test.java b/new_test.java
index cafddf3..2f6a27b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -23,8 +23,6 @@
 
     assertTrue("Missing org.eclim.java.logging.impl",
         properties.containsKey("org.eclim.java.logging.impl"));
-    assertTrue("Missing org.eclim.java.validation.ignore.warnings",
-        properties.containsKey("org.eclim.java.validation.ignore.warnings"));
     assertTrue("Missing org.eclipse.jdt.core.compiler.source",
         properties.containsKey("org.eclipse.jdt.core.compiler.source"));
 
