diff --git a/old_product.java b/new_product.java
index 9a0293f..bfd5d2a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,6 @@
-public Object execute(CommandLine commandLine)
+@Override
+  @SuppressWarnings("unchecked")
+  public Object execute(CommandLine commandLine)
     throws Exception
   {
     String file = commandLine.getValue(Options.FILE_OPTION);
@@ -38,6 +40,13 @@ public Object execute(CommandLine commandLine)
             problems[ii].isWarning()));
       }
 
+      boolean checkstyle = "true".equals(getPreferences()
+          .getValue(project, "org.eclim.java.checkstyle.onvalidate"));
+      if (checkstyle){
+        errors.addAll((List<Error>)
+            Services.getCommand("java_checkstyle").execute(commandLine));
+      }
+
       if(commandLine.hasOption(Options.BUILD_OPTION)){
         project.build(
             IncrementalProjectBuilder.INCREMENTAL_BUILD,
