diff --git a/old_product.java b/new_product.java
index 6ab27a7..1a39823 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,8 @@
 @Override
     public void disableGroup(String groupName) {
+        if (groupName == null || groupName.isEmpty()) {
+            throw new IllegalArgumentException("Groupname cannot be null nor empty");
+        }
         if (!existGroup(groupName)) {
             throw new GroupNotFoundException(groupName);
         }
