diff --git a/old_test.java b/new_test.java
index 15c66c7..9819f23 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,11 +2,12 @@
     public void testDisableGroup() {
         // Given
         assertFf4j.assertThatFeatureIsEnabled(F4);
+        assertFf4j.assertThatFeatureIsInGroup(F4, G1);
         // When
         testedStore.disableGroup(G1);
         // Then
         assertFf4j.assertThatFeatureIsDisabled(F4);
-        // Cancel modifications
+        // Rollback modifications
         testedStore.enable(F4);
         assertFf4j.assertThatFeatureIsEnabled(F4);
     }
\ No newline at end of file
