diff --git a/old_test.java b/new_test.java
index 868cfdc..aa4e558 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
             // Check Read pre-initialization.
             AssertExtensions.assertThrows(
                     "truncate() worked before initialize()",
-                    () -> log.truncate(0, TIMEOUT),
+                    () -> log.truncate(new InMemoryDurableDataLog.InMemoryLogAddress(0), TIMEOUT),
                     ex -> ex instanceof IllegalStateException);
 
             log.initialize(TIMEOUT);
@@ -13,7 +13,7 @@
 
             // Test truncating after each sequence number that we got back.
             for (long seqNo : seqNos) {
-                log.truncate(seqNo, TIMEOUT).join();
+                log.truncate(new InMemoryDurableDataLog.InMemoryLogAddress(seqNo), TIMEOUT).join();
                 writeData.remove(seqNo);
                 testRead(log, -1, writeData);
             }
