diff --git a/old_product.java b/new_product.java
index 220b2b0..2ea9100 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,8 @@
-public CompletableFuture<Boolean> truncate(long upToTransactionId, java.time.Duration timeout) {
+CompletableFuture<Boolean> truncate(DLSNAddress upToAddress, java.time.Duration timeout) {
         ensureNotClosed();
         Preconditions.checkState(this.logManager != null, "LogHandle is not initialized.");
 
-        log.info("{}: Truncate (TransactionId = {}.", this.logName, upToTransactionId);
-        throw new NotImplementedException("Truncate has not yet been implemented.");
+        log.info("{}: Truncate (LogAddress = {}.", this.logName, upToAddress);
+        Future<Boolean> truncateFuture = this.logWriter.truncate(upToAddress.getDLSN());
+        return toCompletableFuture(truncateFuture, b -> b);
     }
\ No newline at end of file
