diff --git a/old_product.java b/new_product.java
index a99ab6f..ef8cecb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,7 +9,7 @@ protected void read(POIXMLFactory factory, Map<PackagePart, POIXMLDocumentPart>
         if (!pp.hasRelationships()) return;
 
         PackageRelationshipCollection rels = packagePart.getRelationships();
-        List<POIXMLDocumentPart> readLater = new ArrayList<POIXMLDocumentPart>();
+        List<POIXMLDocumentPart> readLater = new ArrayList<>();
 
         // scan breadth-first, so parent-relations are hopefully the shallowest element
         for (PackageRelationship rel : rels) {
