diff --git a/old_test.java b/new_test.java
index 2e7c63d..12ece9e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     public void read() throws IOException {
         HSLFSlideShow ppt = HSLFTestDataSamples.getSlideShow("text_shapes.ppt");
 
-        List<String> lst1 = new ArrayList<String>();
+        List<String> lst1 = new ArrayList<>();
         HSLFSlide slide = ppt.getSlides().get(0);
         for (HSLFShape shape : slide.getShapes()) {
             assertTrue("Expected TextShape but found " + shape.getClass().getName(), shape instanceof HSLFTextShape);
@@ -40,7 +40,7 @@
             lst1.add(rawText);
         }
 
-        List<String> lst2 = new ArrayList<String>();
+        List<String> lst2 = new ArrayList<>();
         for (List<HSLFTextParagraph> paras : slide.getTextParagraphs()) {
             lst2.add(HSLFTextParagraph.getRawText(paras));
         }
