diff --git a/old_test.java b/new_test.java
index 21270ae..2d4f6f1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void getSendMessage() {
-        if (null == messageCacheService.getSendMessage(blockMessage.getHash().getDigestHex())){
+        if (null == messageCacheService.getSendMessage(blockMessage.getHash())) {
             messageCacheService.cacheSendedMessage(blockMessage);
         }
-        assertNotNull(messageCacheService.getSendMessage(blockMessage.getHash().getDigestHex()));
+        assertNotNull(messageCacheService.getSendMessage(blockMessage.getHash()));
     }
\ No newline at end of file
