diff --git a/old_test.java b/new_test.java
index b960ce0..73ffb86 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
   public void testIter() throws IOException {
     for(final Object[] exp : TYPES) {
       if(exp.length < 2) continue;
-      try(final TestQuery tq = session.query(exp[1].toString())) {
+      try(TestQuery tq = session.query(exp[1].toString())) {
         final TestResult tr = tq.iter();
         final Object[] type = TYPES[tr.type];
         assertSame("Types are different.\nExpected: " + exp[0] + "\nFound: " + type[0], exp, type);
