diff --git a/old_product.java b/new_product.java
index 3460683..32c9bc9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@
       @Override
       public Void call() throws Exception {
         if (options == null) {
-          mFileSystem.free(new AlluxioURI(path), FreePOptions.getDefaultInstance());
+          mFileSystem.free(new AlluxioURI(path));
         } else {
           mFileSystem.free(new AlluxioURI(path), options);
         }
