diff --git a/old_test.java b/new_test.java
index 4ba26ee..6230536 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
     assertEquals(1, mBlockMaster.getBlockInfo(blockId).getLocations().size());
 
     // free the file
-    mFileSystemMaster.free(NESTED_FILE_URI, FreeContext.defaults(FreePOptions.newBuilder()
+    mFileSystemMaster.free(NESTED_FILE_URI, FreeContext.mergeFrom(FreePOptions.newBuilder()
         .setForced(false).setRecursive(false)));
     // Update the heartbeat of removedBlockId received from worker 1.
     Command heartbeat2 = mBlockMaster.workerHeartbeat(mWorkerId1, null,
