diff --git a/old_test.java b/new_test.java
index f37c340..d97c2ba 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,13 +12,13 @@
     mFsShell.run("ls", "/testRoot");
     String expected = "";
     expected += getLsResultStr("/testRoot/testFileA", files[0].getCreationTimeMs(), 10,
-        LsCommand.STATE_FILE_IN_MEMORY, testUser, testUser, files[0].getMode(),
+        LsCommand.STATE_FILE_IN_ALLUXIO, testUser, testUser, files[0].getMode(),
         files[0].isFolder());
     expected +=
         getLsResultStr("/testRoot/testDir", files[1].getCreationTimeMs(), 1, LsCommand.STATE_FOLDER,
             testUser, testUser, files[1].getMode(), files[1].isFolder());
     expected += getLsResultStr("/testRoot/testFileC", files[3].getCreationTimeMs(), 30,
-        LsCommand.STATE_FILE_NOT_IN_MEMORY, testUser, testUser, files[3].getMode(),
+        LsCommand.STATE_FILE_NOT_IN_ALLUXIO, testUser, testUser, files[3].getMode(),
         files[3].isFolder());
     Assert.assertEquals(expected, mOutput.toString());
   }
\ No newline at end of file
