diff --git a/old_test.java b/new_test.java
index 9de81ee..686b974 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,18 +2,18 @@
     public void testCsvGav() {
         EscapeTool instance = new EscapeTool();
         Set<Identifier> ids = null;
-        String expResult = "";
+        String expResult = "\"\"";
         String result = instance.csvGav(ids);
         assertEquals(expResult, result);
 
         ids = new HashSet<>();
-        expResult = "";
+        expResult = "\"\"";
         result = instance.csvGav(ids);
         assertEquals(expResult, result);
 
         ids = new HashSet<>();
         ids.add(new Identifier("cpe", "somegroup:something:1.0", ""));
-        expResult = "";
+        expResult = "\"\"";
         result = instance.csvGav(ids);
         assertEquals(expResult, result);
 
