diff --git a/old_product.java b/new_product.java
index fb9600c..0ad76ac 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,10 @@
-public EZIDResponse delete()
-            throws IOException, IdentifierException
+public EZIDResponse delete(String name)
+            throws IOException, IdentifierException, URISyntaxException
     {
         // DELETE path
         HttpDelete request;
-        request = new HttpDelete(url);
+        URI uri = new URI(scheme, host, ID_PATH + authority + name, null);
+        request = new HttpDelete(uri);
         HttpResponse response = client.execute(request);
         return new EZIDResponse(response);
     }
\ No newline at end of file
