diff --git a/old_product.java b/new_product.java
index afcedd8..88ad4f1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
 public Point findPoint (Point point)
     {
         Rectangle window = new Rectangle(point);
-        window.grow(stickyDistance, stickyDistance);
+        window.grow(getStickyDistance(), getStickyDistance());
 
         for (Point pt : points) {
             if (window.contains(pt)) {
