diff --git a/old_test.java b/new_test.java
index 077cde2..b42addc 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,11 +6,5 @@
         System.out.println("before: " + instance.getSequenceString());
 
         Point point = new Point(11, 2);
-        instance.setStickyDistance(1);
-
-        Point result = instance.findPoint(point);
-        assertEquals(expResult, result);
-
-        instance.setStickyDistance(0);
-        assertEquals(null, instance.findPoint(point));
+        assertEquals(p1, instance.findPoint(point));
     }
\ No newline at end of file
