diff --git a/old_product.java b/new_product.java
index 6500e54..805df1e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -60,7 +60,7 @@ void reconcile(final HasMetadata involvedObject,
             } else if (kafkaTopic == null) {
                 // it was deleted in kafka so delete in k8s and privateState
                 LOGGER.debug("topic deleted in kafkas => delete KafkaTopic from k8s and from topicStore");
-                enqueue(new DeleteResource(privateTopic.getOrAsMapName(), ar -> {
+                enqueue(new DeleteResource(privateTopic.getOrAsKubeName(), ar -> {
                     if (ar.succeeded()) {
                         enqueue(new DeleteFromTopicStore(privateTopic.getTopicName(), involvedObject, reconciliationResultHandler));
                     } else {
