diff --git a/old_product.java b/new_product.java
index 85278d3..7946800 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,6 @@
 @GetMapping(value = "/{id}/tags", produces = MediaType.APPLICATION_JSON_VALUE)
     @ResponseStatus(HttpStatus.OK)
-    public Set<String> getTagsForCluster(
-        @PathVariable("id") final String id
-    ) throws GenieException {
+    public Set<String> getTagsForCluster(@PathVariable("id") final String id) throws GenieException {
         log.debug("Called with id {}", id);
-        return this.clusterService.getTagsForCluster(id);
+        return DtoAdapters.toV3Cluster(this.clusterService.getCluster(id)).getTags();
     }
\ No newline at end of file
