diff --git a/old_test.java b/new_test.java
index 07a0a8f..141ddc5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testGetTagsForCluster() throws GenieException {
-        Assert.assertEquals(5,
-            this.service.getTagsForCluster(CLUSTER_1_ID).size());
+        Assert.assertEquals(3, this.service.getTagsForCluster(CLUSTER_1_ID).size());
     }
\ No newline at end of file
