diff --git a/old_product.java b/new_product.java
index 749eece..44d6c5d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,7 @@
 public RealmResults<DynamicRealmObject> distinct(String className, String fieldName) {
-        checkNotNullFieldName(fieldName);
         checkIfValid();
         Table table = schema.getTable(className);
-        long columnIndex = table.getColumnIndex(fieldName);
-        if (columnIndex == -1) {
-            throw new IllegalArgumentException(String.format("Field name '%s' does not exist.", fieldName));
-        }
-
+        long columnIndex = RealmQuery.getAndValidateDistinctColumnIndex(fieldName, table);
         TableView tableView = table.getDistinctView(columnIndex);
         return RealmResults.createFromDynamicTableOrView(this, tableView, className);
     }
\ No newline at end of file
