diff --git a/old_test.java b/new_test.java
index 97d8866..0891f79 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,14 +2,12 @@
     public void distinct() {
         final long numberOfBlocks = 25;
         final long numberOfObjects = 10; // must be greater than 1
-
         populateForDistinct(realm, numberOfBlocks, numberOfObjects, false);
 
-        RealmResults<DynamicRealmObject> distinctBool = realm.distinct(AnnotationIndexTypes.CLASS_NAME, "indexBoolean");
+        RealmResults<DynamicRealmObject> distinctBool = realm.distinct(AnnotationIndexTypes.CLASS_NAME, AnnotationIndexTypes.FIELD_INDEX_BOOL);
         assertEquals(2, distinctBool.size());
-
-        for (String fieldName : new String[]{"Long", "Date", "String"}) {
-            RealmResults<DynamicRealmObject> distinct = realm.distinct(AnnotationIndexTypes.CLASS_NAME, "index" + fieldName);
-            assertEquals("index" + fieldName, numberOfBlocks, distinct.size());
+        for (String field : new String[]{AnnotationIndexTypes.FIELD_INDEX_LONG, AnnotationIndexTypes.FIELD_INDEX_DATE, AnnotationIndexTypes.FIELD_INDEX_STRING}) {
+            RealmResults<DynamicRealmObject> distinct = realm.distinct(AnnotationIndexTypes.CLASS_NAME, field);
+            assertEquals(field, numberOfBlocks, distinct.size());
         }
     }
\ No newline at end of file
