diff --git a/old_test.java b/new_test.java
index d53b7f6..0b86e7d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,6 +12,6 @@
         networkSet.forEach(network -> NETWORK_METRICS.forEach(cmt ->
                 testUpdateMetricWithResource(cmt, mv, network)));
 
-        assertThat(monitor.availableResources(Type.DISK), is(diskSet));
-        assertThat(monitor.availableResources(Type.NETWORK), is(networkSet));
+        assertThat(monitor.availableResourcesSync(nodeId, Type.DISK), is(diskSet));
+        assertThat(monitor.availableResourcesSync(nodeId, Type.NETWORK), is(networkSet));
     }
\ No newline at end of file
