diff --git a/old_test.java b/new_test.java
index 9c001c2..9f9529e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
         String jobId = UUID.randomUUID().toString();
         String jobName = "testJob";
 
-        CreateJobResponse response = new CreateJobResponse();
+        JobResponse response = new JobResponse();
         response.setId(jobId);
         response.setName(jobName);
         response.setLinks(Arrays.asList(createLink("self", JOBS_PATH + "/" + jobId)));
@@ -15,7 +15,7 @@
         ).respond((response().withBody(GsonSingleton.getGson().toJson(response))));
 
         // Call create batch
-        CreateJobResponse job = service.createJob(jobName, batchId, ConversionTarget.ANSWER_UNITS);
+        JobResponse job = service.createJob(jobName, batchId, ConversionTarget.ANSWER_UNITS);
         Assert.assertNotNull(job);
         Assert.assertEquals(job.toString(), response.toString());
     }
\ No newline at end of file
