diff --git a/old_product.java b/new_product.java
index 34f9b47..aba047e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,16 +1,16 @@
-public Value geometryN(final ANode node, Int geoNumber) throws QueryException {
+public Value geometryN(final ANode node, final Int geoNumber) throws QueryException {
 
-		// Check QName
-      if(checkNode(node, new QNm[0])) {
-        Geometry geom = gmlReader(node);
-			  
-        int temp = geom.getNumGeometries();
-			  if (((BigInteger) geoNumber.toJava()).intValue() < 1 || ((BigInteger) geoNumber.toJava()).intValue() > temp)
-			    throw GeoErrors.outOfRangeIdx(geoNumber); 	  
-			  
-			  Geometry geo = geom.getGeometryN(((BigInteger) geoNumber.toJava()).intValue() - 1);
-			  // Write the Geometry in GML2 format
-			  return gmlWriter(geo);
-			}
-			return null;
-		}
\ No newline at end of file
+    // Check QName
+    if(checkNode(node, new QNm[0])) {
+      Geometry geom = gmlReader(node);
+      int temp = geom.getNumGeometries();
+      if (((BigInteger) geoNumber.toJava()).intValue() < 1
+          || ((BigInteger) geoNumber.toJava()).intValue() > temp)
+        throw GeoErrors.outOfRangeIdx(geoNumber);
+      Geometry geo = geom.getGeometryN(
+          ((BigInteger) geoNumber.toJava()).intValue() - 1);
+      //Write the Geometry in GML2 format
+      return gmlWriter(geo);
+      }
+    return null;
+    }
\ No newline at end of file
