diff --git a/old_test.java b/new_test.java
index 8e8dcc9..b40ad90 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test(groups={"tck"})
     public void test_minus_PeriodProvider() {
-        PeriodProvider provider = Period.ofDateFields(1, 2, 3);
-        OffsetDate t = TEST_2007_07_15_PONE.minus(provider);
-        assertEquals(t, OffsetDate.of(2006, 5, 12, OFFSET_PONE));
+        Period period = Period.of(7, LocalDateUnit.MONTHS);
+        OffsetDate t = TEST_2007_07_15_PONE.minus(period);
+        assertEquals(t, OffsetDate.of(2006, 12, 15, OFFSET_PONE));
     }
\ No newline at end of file
