diff --git a/old_product.java b/new_product.java
index 978b47c..fb51aed 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,23 +1,5 @@
-public static boolean equivalent(
+@Deprecated
+  public static boolean equivalent(
       @Nullable Network<?, ?> networkA, @Nullable Network<?, ?> networkB) {
-    if (networkA == networkB) {
-      return true;
-    }
-    if (networkA == null || networkB == null) {
-      return false;
-    }
-
-    if (networkA.isDirected() != networkB.isDirected()
-        || !networkA.nodes().equals(networkB.nodes())
-        || !networkA.edges().equals(networkB.edges())) {
-      return false;
-    }
-
-    for (Object edge : networkA.edges()) {
-      if (!networkA.incidentNodes(edge).equals(networkB.incidentNodes(edge))) {
-        return false;
-      }
-    }
-
-    return true;
+    return Objects.equal(networkA, networkB);
   }
\ No newline at end of file
