diff --git a/old_product.java b/new_product.java
index ec8fcf9..301745f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-protected Zhang99Parameters initialParam( List<List<Point2D_F64>> observations )
+protected Zhang99ParamAll initialParam( List<List<Point2D_F64>> observations )
 	{
 		status("Estimating Homographies");
 		List<DenseMatrix64F> homographies = new ArrayList<DenseMatrix64F>();
@@ -24,9 +24,10 @@ protected Zhang99Parameters initialParam( List<List<Point2D_F64>> observations )
 		}
 
 		status("Estimating Radial Distortion");
-		computeRadial.process(K,homographies,observations);
+		computeRadial.process(K, homographies, observations);
 
 		double distort[] = computeRadial.getParameters();
 
-		return convertIntoZhangParam(motions, K,assumeZeroSkew, distort);
+		return convertIntoZhangParam(motions, K,optimized.assumeZeroSkew, distort,
+				optimized.includeTangential);
 	}
\ No newline at end of file
