diff --git a/old_test.java b/new_test.java
index 502a998..5d5cf91 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,6 @@
         assertEquals(ctx.getSecondClassLbl(), null);
 
         ctx.aggregate(VectorUtils.of().labeled(0.0));
-        assertEquals(ctx.getFirstClassLbl(), 1., 0.);
-        assertEquals(ctx.getSecondClassLbl(), 0., 0.);
+        assertEquals(ctx.getFirstClassLbl(), 0., 0.);
+        assertEquals(ctx.getSecondClassLbl(), 1., 0.);
     }
\ No newline at end of file
