diff --git a/old_test.java b/new_test.java
index 99736d2..6a08602 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,10 @@
 @Test
-    public void isValidForAddOperation() throws NoSuchAlgorithmException {
+    public void isValidForAddOperation() throws NoSuchAlgorithmException, CryptoException {
         KeyPair senderKeys = TestUtils.generateKeyPair();
         KeyPair receiverKeys = TestUtils.generateKeyPair();
 
         MailboxStoragePayload mailboxStoragePayload = buildMailboxStoragePayload(senderKeys.getPublic(), receiverKeys.getPublic());
-        ProtectedStorageEntry protectedStorageEntry = buildProtectedMailboxStorageEntry(mailboxStoragePayload, senderKeys.getPublic(), receiverKeys.getPublic());
+        ProtectedStorageEntry protectedStorageEntry = buildProtectedMailboxStorageEntry(mailboxStoragePayload, senderKeys, receiverKeys.getPublic());
 
         Assert.assertTrue(protectedStorageEntry.isValidForAddOperation());
     }
\ No newline at end of file
