diff --git a/old_product.java b/new_product.java
index 784b5ef..00c4b08 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,8 @@
 @Override
     public boolean isValidForAddOperation() {
+        if (!this.isSignatureValid())
+            return false;
+
         MailboxStoragePayload mailboxStoragePayload = this.getMailboxStoragePayload();
         boolean result = mailboxStoragePayload.getSenderPubKeyForAddOperation() != null &&
                 mailboxStoragePayload.getSenderPubKeyForAddOperation().equals(this.getOwnerPubKey());
