diff --git a/old_product.java b/new_product.java
index fcfa61a..b62d8c8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public PermissionStatus applyUMask(FileSystemPermission umask) {
-    FileSystemPermission newFileSystemPermission = mPermission.applyUMask(umask);
-    return new PermissionStatus(mUserName, mGroupName, newFileSystemPermission);
+    mPermission = mPermission.applyUMask(umask);
+    return this;
   }
\ No newline at end of file
