diff --git a/old_product.java b/new_product.java
index f2fd81e..5c11bf7 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,4 @@
 public int[] sort() {
-    final Comparator<BytesRef> comp = BytesRef.getUTF8SortedAsUnicodeComparator();
     final int[] compact = compact();
     new IntroSorter() {
       @Override
@@ -15,7 +14,7 @@ public int[] sort() {
         assert bytesStart.length > id1 && bytesStart.length > id2;
         pool.setBytesRef(scratch1, bytesStart[id1]);
         pool.setBytesRef(scratch2, bytesStart[id2]);
-        return comp.compare(scratch1, scratch2);
+        return scratch1.compareTo(scratch2);
       }
 
       @Override
@@ -30,7 +29,7 @@ public int[] sort() {
         final int id = compact[j];
         assert bytesStart.length > id;
         pool.setBytesRef(scratch2, bytesStart[id]);
-        return comp.compare(pivot, scratch2);
+        return pivot.compareTo(scratch2);
       }
       
       private final BytesRef pivot = new BytesRef(),
