diff --git a/old_test.java b/new_test.java
index 3aa321a..f5e4e93 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,17 +1,17 @@
 @Test
 	public void retrieveStockAt()
 	{
-		final CandidatesSegment earlierQuery = mkQuery(earlier);
+		final CandidatesSegment earlierQuery = mkQueryWithLocator(earlier);
 		final Optional<Candidate> earlierStock = candidateRepository.retrieveStockAt(earlierQuery);
 		assertThat(earlierStock.isPresent(), is(false));
 
-		final CandidatesSegment sameTimeQuery = mkQuery(now);
+		final CandidatesSegment sameTimeQuery = mkQueryWithLocator(now);
 		final Optional<Candidate> sameTimeStock = candidateRepository.retrieveStockAt(sameTimeQuery);
 		assertThat(sameTimeStock.isPresent(), is(true));
-		assertThat(sameTimeStock.get(), is(stockCandidate));
+		assertThat(toCandidateWithoutIds(sameTimeStock.get()), is(stockCandidate));
 
-		final CandidatesSegment laterQuery = mkQuery(later);
+		final CandidatesSegment laterQuery = mkQueryWithLocator(later);
 		final Optional<Candidate> laterStock = candidateRepository.retrieveStockAt(laterQuery);
 		assertThat(laterStock.isPresent(), is(true));
-		assertThat(laterStock.get(), is(laterStockCandidate));
+		assertThat(toCandidateWithoutIds(laterStock.get()), is(laterStockCandidate));
 	}
\ No newline at end of file
