diff --git a/old_test.java b/new_test.java
index 18a0ab7..0694367 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testOnCreateAuditable() throws InterruptedException, GeniePreconditionException {
+    public void testOnCreateBaseEntity() throws InterruptedException, GeniePreconditionException {
         final BaseEntity a = new BaseEntity();
         Assert.assertNull(a.getId());
         Assert.assertNotNull(a.getCreated());
@@ -7,7 +7,7 @@
         final Date originalCreated = a.getCreated();
         final Date originalUpdated = a.getUpdated();
         Thread.sleep(1);
-        a.onCreateAuditable();
+        a.onCreateBaseEntity();
         Assert.assertNotNull(a.getId());
         Assert.assertNotNull(a.getCreated());
         Assert.assertNotNull(a.getUpdated());
@@ -19,6 +19,6 @@
         final BaseEntity baseEntity = new BaseEntity();
         final String id = UUID.randomUUID().toString();
         baseEntity.setId(id);
-        baseEntity.onCreateAuditable();
+        baseEntity.onCreateBaseEntity();
         Assert.assertEquals(id, baseEntity.getId());
     }
\ No newline at end of file
