diff --git a/old_product.java b/new_product.java
index 7fbeb33..b3798a1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-@Override public void aggregate(IgniteModel<Vector, L> model, LabeledVector<L> vector) {
-        L modelAns = model.predict(vector.features());
+@Override public void aggregate(IgniteModel<Vector, L> mdl, LabeledVector<L> vector) {
+        L modelAns = mdl.predict(vector.features());
         L realAns = vector.label();
 
         if (modelAns.equals(falseLabel) && realAns.equals(falseLabel))
