diff --git a/old_test.java b/new_test.java
index 0794f88..9528ac3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,10 +12,10 @@
         assertEquals(0, aggregator.getTruePositive());
         assertEquals(0, aggregator.getFalsePositive());
 
-        aggregator.aggregate(model, VectorUtils.of(0.).labeled(0.));
-        aggregator.aggregate(model, VectorUtils.of(1.).labeled(0.));
-        aggregator.aggregate(model, VectorUtils.of(1.).labeled(1.));
-        aggregator.aggregate(model, VectorUtils.of(0.).labeled(1.));
+        aggregator.aggregate(mdl, VectorUtils.of(0.).labeled(0.));
+        aggregator.aggregate(mdl, VectorUtils.of(1.).labeled(0.));
+        aggregator.aggregate(mdl, VectorUtils.of(1.).labeled(1.));
+        aggregator.aggregate(mdl, VectorUtils.of(0.).labeled(1.));
 
         assertEquals(1, aggregator.getTrueNegative());
         assertEquals(1, aggregator.getFalseNegative());
