diff --git a/old_product.java b/new_product.java
index 95439c6..def3603 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,9 +9,12 @@
                     .matchSubset(that.context, context))) {
                 return false;
             }
-            // 'combiner' is not taken into account - see the test 
-            // AccessControllerTest.testEqualsObject_01
-            return true;
+            // BEGIN android-changed
+            if(combiner != null) {
+                return combiner.equals(that.combiner);
+            }
+            return that.combiner == null;
+            // END android-changed
         }
         return false;
     }
\ No newline at end of file
