diff --git a/old_product.java b/new_product.java
index 165f9b8..e9ec22f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ private Item store(final QueryContext ctx) throws QueryException {
     final IOFile file = data.meta.binary(path);
     if(file == null || file.isDir()) RESINV.thrw(input, path);
 
-    final byte[] val = checkBin(expr[2], ctx);
-    ctx.updates.add(new DBStore(data, token(path), val, input), ctx);
+    final Item it = checkItem(expr[2], ctx);
+    ctx.updates.add(new DBStore(data, token(path), it, input), ctx);
     return null;
   }
\ No newline at end of file
