diff --git a/old_test.java b/new_test.java
index e9736ee..b680a57 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,6 +6,7 @@
     session.store("X", new ArrayInput(""));
     check("", session.query(_DB_RETRIEVE.args(DB, "X")).execute());
     session.store("X", new ArrayInput(new byte[] { 0, 1, -1 }));
-    check("0001FF", session.query(_DB_RETRIEVE.args(DB, "X")).execute());
+    check("0001FF", session.query(
+        "xs:hexBinary(" + _DB_RETRIEVE.args(DB, "X") + ")").execute());
     session.execute("drop db " + DB);
   }
\ No newline at end of file
