diff --git a/old_product.java b/new_product.java
index ce56060..3468525 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,8 @@
 @Override
-    public Set<Macro> getMacros(MachineImpl devMachine) {
+    public Set<Macro> getMacros(MachineImpl machine) {
         final Set<Macro> macros = Sets.newHashSet();
 
-        for (Map.Entry<String, ServerImpl> entry : devMachine.getServers().entrySet()) {
+        for (Map.Entry<String, ServerImpl> entry : machine.getServers().entrySet()) {
 
             if (!entry.getValue().getUrl().contains(":")) {
                 continue;
