diff --git a/old_product.java b/new_product.java
index 58e3750..39cb956 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-@PostPersist
-    @PostUpdate
-    protected void onCreateOrUpdateCluster() throws GeniePreconditionException {
-        this.addAndValidateSystemTags(this.tags);
+@PrePersist
+    @PreUpdate
+    protected void onCreateOrUpdateCluster() throws GenieException {
+        this.setClusterTags(this.getFinalTags());
     }
\ No newline at end of file
