diff --git a/old_test.java b/new_test.java
index a2fb73b..dbca7c7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
-    public void testOnCreateOrUpdateCluster() throws GeniePreconditionException {
+    public void testOnCreateOrUpdateCluster() throws GenieException {
         this.c = new ClusterEntity(NAME, USER, VERSION, ClusterStatus.UP, CLUSTER_TYPE);
         Assert.assertNotNull(this.c.getTags());
         Assert.assertTrue(this.c.getTags().isEmpty());
