diff --git a/old_product.java b/new_product.java
index fc68f79..de594e2 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,10 +8,6 @@
         }
         AbstractCategoryItemRenderer that = (AbstractCategoryItemRenderer) obj;
 
-        if (!ObjectUtils.equal(this.itemLabelGenerator,
-                that.itemLabelGenerator)) {
-            return false;
-        }
         if (!ObjectUtils.equal(this.itemLabelGeneratorMap,
                 that.itemLabelGeneratorMap)) {
             return false;
@@ -20,10 +16,6 @@
                 that.baseItemLabelGenerator)) {
             return false;
         }
-        if (!ObjectUtils.equal(this.toolTipGenerator,
-                that.toolTipGenerator)) {
-            return false;
-        }
         if (!ObjectUtils.equal(this.toolTipGeneratorMap,
                 that.toolTipGeneratorMap)) {
             return false;
@@ -32,10 +24,6 @@
                 that.baseToolTipGenerator)) {
             return false;
         }
-        if (!ObjectUtils.equal(this.itemURLGenerator,
-                that.itemURLGenerator)) {
-            return false;
-        }
         if (!ObjectUtils.equal(this.itemURLGeneratorMap,
                 that.itemURLGeneratorMap)) {
             return false;
