diff --git a/old_test.java b/new_test.java
index 194d045..eb951d2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,4 +1,4 @@
 @Test(expected = ContextAuthenticationException.class)
-	public void authenticate_shouldNotAuthenticateGivenNullPasswordWhenPasswordInDatabaseIsNull() throws Exception {
+	public void authenticate_shouldNotAuthenticateGivenNullPasswordWhenPasswordInDatabaseIsNull() {
 		dao.authenticate("nullpassword", null);
 	}
\ No newline at end of file
