diff --git a/old_product.java b/new_product.java
index db5760d..a0d6229 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,5 +4,16 @@ public void validate(Object obj, Errors errors) {
 			errors.rejectValue("encounterType", "error.general");
 		} else {
 			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "name", "error.name");
+			
+			if (!errors.hasErrors()) {
+				EncounterType duplicate = Context.getEncounterService().getEncounterType(encounterType.getName().trim());
+				if (duplicate != null) {
+					if (duplicate.getUuid() != null
+					        && !OpenmrsUtil.nullSafeEquals(encounterType.getUuid(), duplicate.getUuid())) {
+						errors.rejectValue("name", "encounterType.duplicate.name",
+						    "Specified Encounter Type name already exists, please specify another ");
+					}
+				}
+			}
 		}
 	}
\ No newline at end of file
