diff --git a/old_test.java b/new_test.java
index 8b8f6a2..247c2c8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,7 +8,7 @@
             /* no-op */
         }
     
-        Iterator<Long2DoubleMap.Entry> iter = singleton().fastIterator();
+        Iterator<VectorEntry> iter = singleton().fastIterator();
         assertTrue(iter.hasNext());
         Long2DoubleMap.Entry e = iter.next();
         assertFalse(iter.hasNext());
