diff --git a/old_product.java b/new_product.java
index ec6cf48..fd4f872 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,3 +1,3 @@
 public static <T, O> Map<T, Integer> groupCount(Collection<O> objectCollection,String propertyName){
-        return groupCount(objectCollection, null, propertyName);
+        return groupCount(objectCollection, propertyName, null);
     }
\ No newline at end of file
