diff --git a/old_product.java b/new_product.java
index 0807ee0..905fe47 100644
--- a/old_product.java
+++ b/new_product.java
@@ -7,6 +7,10 @@ public static HighAvailabilityMode fromConfig(Configuration config) {
 			// Map old default to new default
 			return HighAvailabilityMode.NONE;
 		} else {
-			return HighAvailabilityMode.valueOf(haMode.toUpperCase());
+			try {
+				return HighAvailabilityMode.valueOf(haMode.toUpperCase());
+			} catch (IllegalArgumentException e) {
+				return FACTORY_CLASS;
+			}
 		}
 	}
\ No newline at end of file
