diff --git a/old_product.java b/new_product.java
index 0c01d0f..d95922c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,22 +1,23 @@
-public IRequestHandler checkSecureIncoming(IRequestHandler handler)
+IRequestHandler checkSecureIncoming(IRequestHandler requestHandler,
+		final HttpsConfig httpsConfig)
 	{
 
-		if (handler instanceof SwitchProtocolRequestHandler)
+		if (requestHandler instanceof SwitchProtocolRequestHandler)
 		{
-			return handler;
+			return requestHandler;
 		}
 
-		Class<?> pageClass = getPageClass(handler);
+		Class<?> pageClass = getPageClass(requestHandler);
 		if (pageClass != null)
 		{
 			IRequestHandler redirect = null;
 			if (hasSecureAnnotation(pageClass))
 			{
-				redirect = SwitchProtocolRequestHandler.requireProtocol(Protocol.HTTPS);
+				redirect = SwitchProtocolRequestHandler.requireProtocol(Protocol.HTTPS, httpsConfig);
 			}
 			else
 			{
-				redirect = SwitchProtocolRequestHandler.requireProtocol(Protocol.HTTP);
+				redirect = SwitchProtocolRequestHandler.requireProtocol(Protocol.HTTP, httpsConfig);
 			}
 
 			if (redirect != null)
@@ -25,5 +26,5 @@ public IRequestHandler checkSecureIncoming(IRequestHandler handler)
 			}
 
 		}
-		return handler;
+		return requestHandler;
 	}
\ No newline at end of file
