diff --git a/old_test.java b/new_test.java
index 3b6a91d..e1ab3ac 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,12 +12,16 @@
 		IPageRequestHandler httpsPageRequestHandler = Mockito.mock(IPageRequestHandler.class);
 		Mockito.when(httpsPageRequestHandler.getPage()).thenReturn(new HttpsPage());
 
-		IRequestHandler httpsPageSecureIncoming = checker.checkSecureIncoming(httpsPageRequestHandler);
+		HttpsConfig httpsConfig = new HttpsConfig();
+
+		IRequestHandler httpsPageSecureIncoming = checker.checkSecureIncoming(
+			httpsPageRequestHandler, httpsConfig);
 		assertTrue(httpsPageSecureIncoming instanceof SwitchProtocolRequestHandler);
 
 		IPageRequestHandler httpPageRequestHandler = Mockito.mock(IPageRequestHandler.class);
 		Mockito.when(httpPageRequestHandler.getPage()).thenReturn(new HttpPage());
 
-		IRequestHandler httpPageSecureIncoming = checker.checkSecureIncoming(httpPageRequestHandler);
+		IRequestHandler httpPageSecureIncoming = checker.checkSecureIncoming(
+			httpPageRequestHandler, httpsConfig);
 		assertSame(httpPageRequestHandler, httpPageSecureIncoming);
 	}
\ No newline at end of file
