diff --git a/old_test.java b/new_test.java
index 8314753..1005723 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
 	public void getLastModified() throws Exception
 	{
 		URL url = new URL("http://wicket.apache.org/learn/books/wia.png");
-		Time lastModified = Connections.getLastModified(url);
+		Instant lastModified = Connections.getLastModified(url);
 		assertNotNull(lastModified);
-		assertNotEquals(lastModified.getMilliseconds(), 0L);
+		assertNotEquals(lastModified.toEpochMilli(), 0L);
 	}
\ No newline at end of file
