diff --git a/old_product.java b/new_product.java
index c7960ac..cf23960 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@ public void shouldComplyEqualsContract(T first, T second, T... more) {
 			.suppress(Warning.NULL_FIELDS)
 			.suppress(Warning.NONFINAL_FIELDS)
 			.withPrefabValues(BitSet.class, new BitSet(), blackBitSet);
-			
-		initVerifier(equalVerifier);
-			equalVerifier.usingGetClass().verify();
+
+		this.initVerifier(equalVerifier);
+		equalVerifier.usingGetClass().verify();
 	}
\ No newline at end of file
