diff --git a/old_test.java b/new_test.java
index 374c041..6472755 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testStreamCommands() {
         System.out.println("streamCommands");
-        GrblCommunicator instance = new GrblCommunicator(mg.in, mg.out, gcb, acl);
+        GrblCommunicator instance = new GrblCommunicator(mg.in, mg.out, cb, asl);
         String thirtyNineCharString = "thirty-nine character command here.....";
 
         boolean result;
