diff --git a/old_test.java b/new_test.java
index bedb0f3..055973d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,9 @@
 @Test
   public void testConfigure() throws Exception {
     OffHeapIdentifierRegistry registry = new OffHeapIdentifierRegistry(32, MemoryUnit.MEGABYTES);
+    registry.addResource("defaultServerResource", 8, MemoryUnit.MEGABYTES);
+    registry.addResource("serverResource1", 8, MemoryUnit.MEGABYTES);
+    registry.addResource("serverResource2", 8, MemoryUnit.MEGABYTES);
 
     final EhcacheActiveEntity activeEntity = new EhcacheActiveEntity(registry, ENTITY_ID);
     ClientDescriptor client = new TestClientDescriptor();
