diff --git a/old_product.java b/new_product.java
index 06d0354..8dcf777 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,9 +9,9 @@ private void configure(ClientDescriptor clientDescriptor, ConfigureStoreManager
 
       this.defaultServerResource = configuration.getDefaultServerResource();
       if (this.defaultServerResource != null) {
-        OffHeapResource source = services.getService(OffHeapResourceIdentifier.identifier(this.defaultServerResource));
-        if (source == null) {
-          throw new ResourceConfigurationException("Default server resource '" + this.defaultServerResource + "' is not defined");
+        if (!offHeapResourceIdentifiers.contains(this.defaultServerResource)) {
+          throw new ResourceConfigurationException("Default server resource '" + this.defaultServerResource
+              + "' is not defined. Available resources are: " + offHeapResourceIdentifiers);
         }
       }
 
