diff --git a/old_product.java b/new_product.java
index 155bcc2..7862017 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public static void adjustForDisplay( ImageSingleBand transform , int numLevels , double valueRange ) {
-		if( transform instanceof ImageFloat32 )
-			adjustForDisplay((ImageFloat32)transform,numLevels,(float)valueRange);
+public static void adjustForDisplay(ImageGray transform , int numLevels , double valueRange ) {
+		if( transform instanceof GrayF32)
+			adjustForDisplay((GrayF32)transform,numLevels,(float)valueRange);
 		else
-			adjustForDisplay((ImageInteger)transform,numLevels,(int)valueRange);
+			adjustForDisplay((GrayI)transform,numLevels,(int)valueRange);
 	}
\ No newline at end of file
