diff --git a/old_product.java b/new_product.java
index 6943e3a..3a42119 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,6 @@
 public boolean instanceOf(final SeqType st) {
     // empty sequence: only check cardinality
-    if(zero()) return st.mayBeEmpty();
-
-    final Type type1 = type, type2 = st.type;
-    final Test kind1 = kind, kind2 = st.kind;
-    return (type2 == AtomType.ITEM || type1.instanceOf(type2)) && occ.instanceOf(st.occ) &&
-      (kind2 == null || kind1 != null && kind1.intersect(kind2) != null);
+    return zero() ? st.mayBeEmpty() :
+      (st.type == AtomType.ITEM || type.instanceOf(st.type)) && occ.instanceOf(st.occ) &&
+      (st.kind == null || kind != null && kind.intersect(st.kind) != null);
   }
\ No newline at end of file
