diff --git a/old_test.java b/new_test.java
index d9be2fa..220f22a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -8,6 +8,7 @@
     final SeqType f = FuncType.get(DEC_ZO, BLN_O).seqType();
     assertFalse(f.instanceOf(ITR_O));
     assertTrue(f.instanceOf(ITEM_O));
+    assertTrue(f.instanceOf(FUNC_O));
     assertTrue(f.instanceOf(f));
     assertTrue(f.instanceOf(FUNC_ZO));
     assertFalse(FUNC_O.instanceOf(f));
@@ -24,7 +25,7 @@
     assertTrue(m.instanceOf(MapType.get(AtomType.AAT, ITR_O)));
     assertTrue(m.instanceOf(MapType.get(AtomType.STR, ITR_O)));
     assertTrue(m.instanceOf(MapType.get(AtomType.STR, ITR_ZO)));
-    assertTrue(m.instanceOf(MapType.get(AtomType.ITR, ITEM_ZM)));
+    assertFalse(m.instanceOf(MapType.get(AtomType.ITR, ITEM_ZM)));
     assertFalse(m.instanceOf(ANY_ARRAY));
     assertFalse(m.instanceOf(MapType.get(AtomType.STR, BLN_O)));
 
