diff --git a/old_test.java b/new_test.java
index ab1b021..dd4cef2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,7 +13,7 @@
         AssertExtensions.assertThrows(
                 "initialize() succeeded on a Segment is sealed in Storage but not in the metadata.",
                 () -> context.batchesAggregators[1].initialize(TIMEOUT),
-                ex -> ex instanceof RuntimeStreamingException && ex.getCause() instanceof DataCorruptionException);
+                ex -> ex instanceof DataCorruptionException);
 
         // Check behavior for already-sealed segments (in storage, in metadata, but metadata does not reflect Sealed in storage.)
         context.storage.create(context.batchesAggregators[2].getMetadata().getName(), TIMEOUT).join();
