diff --git a/old_test.java b/new_test.java
index a5e38a6..4f04ef2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test(groups={"tck"})
     public void test_plusNanos_halfABillion() {
         LocalDateTime t = TEST_2007_07_15_12_30_40_987654321.with(LocalTime.MIDNIGHT);
-        LocalDate d = t.toLocalDate();
+        LocalDate d = t.getDate();
 
         int hour = 0;
         int min = 0;
@@ -24,7 +24,7 @@
                 min = 0;
             }
 
-            assertEquals(t.toLocalDate(), d, String.valueOf(i));
+            assertEquals(t.getDate(), d, String.valueOf(i));
             assertEquals(t.getHour(), hour);
             assertEquals(t.getMinute(), min);
             assertEquals(t.getSecond(), sec);
