diff --git a/old_product.java b/new_product.java
index 54d8d48..9e4b139 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,7 @@
-private ValueBuilder contextFunctions(final QueryContext ctx) throws QueryException {
+private ValueBuilder functions(final QueryContext ctx) throws QueryException {
     final ValueBuilder vb = new ValueBuilder();
     for(final StaticFunc sf : ctx.funcs.funcs()) {
-      vb.add(Functions.getUser(sf, ctx, sc, info));
+      vb.add(Functions.getUser(sf, ctx, sf.sc, info));
     }
     return vb;
   }
\ No newline at end of file
