diff --git a/old_test.java b/new_test.java
index de950a5..041c256 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,11 @@
 @Test
-  public void contextFunctions() {
+  public void functions() {
     query("declare function local:x() { 1 }; " +
         COUNT.args(_INSPECT_FUNCTIONS.args()), "1");
     query("declare function local:x() { 2 }; " +
         _INSPECT_FUNCTIONS.args() + "()", "2");
+    query("import module namespace hello='world' at 'src/test/resources/hello.xqm';" +
+        "inspect:functions()[last()] instance of function(*)",
+        "true"
+        );
   }
\ No newline at end of file
