diff --git a/old_product.java b/new_product.java
index 7e0ca97..0b65147 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public static ZoneRules getRules(String zoneId) {
+public static ZoneRules getRules(String zoneId, boolean forCaching) {
         Objects.requireNonNull(zoneId, "zoneId");
-        return getProvider(zoneId).provideRules(zoneId);
+        return getProvider(zoneId).provideRules(zoneId, forCaching);
     }
\ No newline at end of file
