diff --git a/old_product.java b/new_product.java
index a7af4a3..849a55b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public default DatasetBuilder<Vector, Double> asDatasetBuilder(int datasetSize, IgniteBiPredicate<Vector, Double> filter,
-        int partitions, UpstreamTransformerBuilder<Vector, Double> upstreamTransformerBuilder) {
+        int partitions, UpstreamTransformerBuilder upstreamTransformerBuilder) {
 
         return new DatasetBuilderAdapter(this, datasetSize, filter, partitions, upstreamTransformerBuilder);
     }
\ No newline at end of file
