diff --git a/old_product.java b/new_product.java
index 710ff21..0c822e1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,9 @@
 @RequestMapping(value = "CTL/getSchema", params = { "fqn", "version" }, method = RequestMethod.GET)
     @ResponseBody
-    public CTLSchemaInfoDto getCTLSchemaByFqnAndVersion(@RequestParam String fqn, @RequestParam int version)
+    public CTLSchemaDto getCTLSchemaByFqnVersionTenantIdAndApplicationId(@RequestParam String fqn, 
+            @RequestParam int version,
+            @RequestParam(required = false) String tenantId,
+            @RequestParam(required = false) String applicationId)
             throws KaaAdminServiceException {
-        return kaaAdminService.getCTLSchemaByFqnAndVersion(fqn, version);
+        return kaaAdminService.getCTLSchemaByFqnVersionTenantIdAndApplicationId(fqn, version, tenantId, applicationId);
     }
\ No newline at end of file
