diff --git a/old_test.java b/new_test.java
index 19fb48d..5a78efe 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	public void testGetNumber() throws Exception {
-		suppress(constructor(SuppressConstructorHierarchyParent.class));
-		SuppressConstructorHierarchyParent tested = new SuppressConstructorHierarchyParent("message");
+		suppress(constructor(SuppressConstructorHierarchy.class));
+		SuppressConstructorHierarchy tested = new SuppressConstructorHierarchy("message");
 		assertEquals(42, tested.getNumber());
 	}
\ No newline at end of file
