diff --git a/old_test.java b/new_test.java
index 075e61a..428df0a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -13,12 +13,9 @@
 		Date twoDate = dateFormat.parse("2017-01-31 00:00:00");
 		
 		firstMembership.setStartDate(oneDate);
-		secondMembership.setStartDate(twoDate);
-		
-		assertEquals(-1, firstMembership.compareTo(secondMembership));
-		
 		secondMembership.setStartDate(oneDate);
 		secondMembership.setEndDate(twoDate);
 		
-		assertEquals(-1, firstMembership.compareTo(secondMembership));
+		assertEquals(1, firstMembership.compareTo(secondMembership));
+		assertEquals(-1, secondMembership.compareTo(firstMembership));
 	}
\ No newline at end of file
