diff --git a/old_product.java b/new_product.java
index 9ccb5a6..d264df0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,25 +1,19 @@
 @Override
 	public int compareTo(CohortMembership o) {
-		if ((this.getVoided() && !o.getVoided()) || (!this.isActive() && o.isActive())) {
-			return 1;
-		} else if ((!this.getVoided() && o.getVoided()) || (this.isActive() && !o.isActive())) {
-			return -1;
+		int ret = this.getVoided().compareTo(o.getVoided());
+		if (ret == 0) {
+			ret = OpenmrsUtil.compareWithNullAsLowest(
+					getCohort() == null ? null : getCohort().getId(),
+					o.getCohort() == null ? null : o.getCohort().getId());
 		}
-		
-		int ret = OpenmrsUtil.compareWithNullAsGreatest(this.getCohort().getCohortId(), o.getCohort().getCohortId());
-		if (ret != 0) {
-			return ret;
+		if (ret == 0) {
+			ret = this.getPatientId().compareTo(o.getPatientId());
 		}
-		
-		ret = this.getPatientId().compareTo(o.getPatientId());
-		if (ret != 0) {
-			return ret;
+		if (ret == 0) {
+			ret = OpenmrsUtil.compare(this.getStartDate(), o.getStartDate());
 		}
-		
-		ret = OpenmrsUtil.compareWithNullAsEarliest(this.getEndDate(), o.getEndDate());
-		if (ret != 0) {
-			return ret;
+		if (ret == 0) {
+			ret = OpenmrsUtil.compareWithNullAsLatest(this.getEndDate(), o.getEndDate());
 		}
-		
-		return OpenmrsUtil.compare(this.getStartDate(), o.getStartDate());
+		return ret;
 	}
\ No newline at end of file
