diff --git a/old_test.java b/new_test.java
index e9fd18d..818664a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -6,18 +6,17 @@
       String namespace = "JCLOUDS/Test";
 
       for (int i = 0; i < 11; i++) {
-         metrics.add(MetricDatum.builder().build());
+         metrics.add(MetricDatum.builder().metricName("foo").build());
       }
 
       // Using EasyMock.eq("") because EasyMock makes it impossible to pass null as a String value here
       expect(client.getMetricClientForRegion(EasyMock.eq("")))
             .andReturn(metricClient)
             .atLeastOnce();
-
-
-      metricClient.putMetricData(metrics, namespace);
-
-      expectLastCall().times(2);
+      
+      for (List<MetricDatum> slice : Iterables.partition(metrics, 10)) {
+         metricClient.putMetricData(slice, namespace);
+      }
 
       EasyMock.replay(client, metricClient);
 
