diff --git a/old_product.java b/new_product.java
index 48915ea..41297fd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,5 @@
-public Candidate addOrReplace(@NonNull final Candidate candidate, final boolean preserveExistingSeqNo)
+@VisibleForTesting
+	Candidate addOrUpdate0(@NonNull final Candidate candidate, final boolean preserveExistingSeqNo)
 	{
 		final Optional<I_MD_Candidate> oldCandidateRecord = retrieveExact(candidate);
 
@@ -25,8 +26,17 @@ public Candidate addOrReplace(@NonNull final Candidate candidate, final boolean
 			addOrRecplaceProductionDetail(candidate, synchedRecord);
 		}
 
+		if (candidate.getDemandDetail() != null)
+		{
+			// we do this independently of the type; the demand info might be needed by many records, not just by the "first" demand record
+			addOrRecplaceDemandDetail(candidate, synchedRecord);
+		}
+
+		final Integer parentId = synchedRecord.getMD_Candidate_Parent_ID() > 0 ? synchedRecord.getMD_Candidate_Parent_ID() : null;
+
 		return candidate
 				.withId(synchedRecord.getMD_Candidate_ID())
+				.withParentId(parentId)
 				.withGroupId(synchedRecord.getMD_Candidate_GroupId())
 				.withSeqNo(synchedRecord.getSeqNo())
 				.withQuantity(qtyDelta);
