diff --git a/old_test.java b/new_test.java
index fbefea0..6a0cf87 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
 		assertThat(stockBeforeReplacement.stream().collect(Collectors.toList()), contains(stockCandidate, laterStockCandidate));
 
 		final Candidate replacementCandidate = stockCandidate.withQuantity(BigDecimal.ONE);
-		candidateRepository.addOrReplace(replacementCandidate);
+		candidateRepository.addOrUpdate(replacementCandidate);
 
 		assertThat(candidateRepository.retrieveLatestMatch(mkStockUntilSegment(now)).get(), is(replacementCandidate));
 		final List<Candidate> stockAfterReplacement = candidateRepository.retrieveMatches(mkStockFromSegment(now));
