diff --git a/old_test.java b/new_test.java
index 3da6003..25042b9 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
   public void testCount() throws QueryException {
-    final String fun = check(FunDef.COUNT);
+    final String fun = check(Function.COUNT);
     query(fun + "(())", "0");
     query(fun + "(//*[text() contains text '1'])", "1");
     query(fun + "(//li[text() contains text 'exercise'])", "2");
