diff --git a/old_product.java b/new_product.java
index c4f7253..ed0d02a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-protected Expr count(final Op o) throws QueryException {
+protected final Expr count(final Op o) throws QueryException {
     // evaluate argument
     final Expr a = expr[1];
     if(!a.item()) return this;
@@ -14,10 +14,10 @@ protected Expr count(final Op o) throws QueryException {
        o == Op.EQ && v != (int) v) return Bln.FALSE;
     // EXISTS: c > (v<1), c >= (v<=1), c != (v=0)
     if(o == Op.GT && v < 1 || o == Op.GE && v <= 1 || o == Op.NE && v == 0)
-      return FunDef.EXISTS.get(input, ((Fun) expr[0]).expr);
+      return Function.EXISTS.get(input, ((FuncCall) expr[0]).expr);
     // EMPTY: c < (v<=1), c <= (v<1), c = (v=0)
     if(o == Op.LT && v <= 1 || o == Op.LE && v < 1 || o == Op.EQ && v == 0)
-      return FunDef.EMPTY.get(input, ((Fun) expr[0]).expr);
+      return Function.EMPTY.get(input, ((FuncCall) expr[0]).expr);
 
     return this;
   }
\ No newline at end of file
