diff --git a/old_test.java b/new_test.java
index b6da221..6ab5f90 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,12 @@
 @Test
   public void testDelete() {
-
-    FNFile fnFile = new FNFile();
-    Expr[] expr = new Expr[1];
+    final FNFile fnFile = new FNFile();
+    final Expr[] expr = new Expr[1];
     expr[0] = Str.get("test2");
     fnFile.init(FunDef.DELETE, expr);
     try {
-      fnFile.atomic(null);
-    } catch(QueryException e) {
+      fnFile.atomic(qc);
+    } catch(final QueryException e) {
       throw new RuntimeException(e);
     }
 
