diff --git a/old_product.java b/new_product.java
index 31a6de9..7c264a1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -9,12 +9,16 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 		boolean isDrugOrder = DrugOrder.class.isAssignableFrom(getActualType(order));
 		Concept concept = order.getConcept();
 		if (concept == null && isDrugOrder) {
-			DrugOrder dOrder = (DrugOrder) order;
-			if (dOrder.getDrug() != null) {
-				concept = dOrder.getDrug().getConcept();
-				dOrder.setConcept(concept);
+			DrugOrder drugOrder = (DrugOrder) order;
+			if (drugOrder.getDrug() != null) {
+				concept = drugOrder.getDrug().getConcept();
+				drugOrder.setConcept(concept);
 			}
 		}
+		if (isDrugOrder && !isDiscontinueOrReviseOrder(order) && order.getAutoExpireDate() == null) {
+			DrugOrder drugOrder = (DrugOrder) order;
+			drugOrder.setAutoExpireDate(drugOrder.getDosingInstructionsInstance().getAutoExpireDate(drugOrder));
+		}
 		if (concept == null) {
 			throw new APIException("concept is required for an order");
 		}
