diff --git a/old_test.java b/new_test.java
index 19ed8c5..3d7b7d1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -19,8 +19,6 @@
 		drugOrder.setDoseUnits(conceptService.getConcept(50));
 		drugOrder.setQuantity(20.0);
 		drugOrder.setQuantityUnits(conceptService.getConcept(51));
-		drugOrder.setDuration(20.0);
-		drugOrder.setDurationUnits(conceptService.getConcept(28));
 		drugOrder.setFrequency(orderService.getOrderFrequency(3));
 		drugOrder.setRoute(conceptService.getConcept(22));
 		drugOrder.setNumRefills(10);
