diff --git a/old_product.java b/new_product.java
index 9682875..588a0cd 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,13 @@
 public synchronized void invalidateCachedControllerLeader() {
-    long now = System.currentTimeMillis();
-    long millisSinceLastInvalidate = now - _lastCacheInvalidateMillis;
-    if (millisSinceLastInvalidate < MILLIS_BETWEEN_INVALIDATE) {
+    long now = getCurrentTimeMs();
+    long millisSinceLastInvalidate = now - _lastCacheInvalidationTimeMs;
+    if (millisSinceLastInvalidate < MIN_INVALIDATE_INTERVAL_MS) {
       LOGGER.info(
           "Millis since last controller cache value invalidate {} is less than allowed frequency {}. Skipping invalidate.",
-          millisSinceLastInvalidate, MILLIS_BETWEEN_INVALIDATE);
+          millisSinceLastInvalidate, MIN_INVALIDATE_INTERVAL_MS);
     } else {
       LOGGER.info("Invalidating cached controller leader value");
-      _cachedControllerLeaderInvalid = true;
-      _lastCacheInvalidateMillis = now;
+      _cachedControllerLeaderValid = false;
+      _lastCacheInvalidationTimeMs = now;
     }
   }
\ No newline at end of file
