diff --git a/old_test.java b/new_test.java
index b6067e6..ec4bda3 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,10 +1,25 @@
 @Test
- 	public void testInvalidateQuery() throws Exception {
-
- 		SFSB2LC sfsb = lookup("SFSB2LC", SFSB2LC.class);
- 		String message = sfsb.invalidateQuery();
- 		
- 		if (!message.equals("OK")){
- 			fail(message);
- 		}
- 	}
\ No newline at end of file
+    public void testInvalidateQuery() throws Exception {
+
+        SFSB2LC sfsb = lookup("SFSB2LC", SFSB2LC.class);
+        String id = "2";
+
+        String message = sfsb.queryCacheCheck(id);
+        
+        if (!message.equals("OK")){
+            fail(message);
+        }
+    
+        // invalidate the cache
+        sfsb.createEmployee("Newman", "Paul", 400);
+        
+        // the nextTimestamp from infinispan is "return System.currentTimeMillis() / 100;"
+        Thread.sleep(1000);
+        
+        message = sfsb.queryCacheCheck(id);
+
+        if (!message.equals("OK")){
+            fail(message);
+        }
+
+    }
\ No newline at end of file
