diff --git a/old_product.java b/new_product.java
index 42f464f..8cfa3c9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public void trace(Marker marker, Object message, Throwable t) {
+public void trace(final Marker marker, final Object message, final Throwable t) {
         if (isEnabled(Level.TRACE, marker, message, t)) {
             log(marker, FQCN, Level.TRACE, messageFactory.newMessage(message), t);
         }
