diff --git a/old_product.java b/new_product.java
index a103f68..0f02f09 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,25 +1,23 @@
-public Quantity computeAssignableQuantity()
+public Quantity computeAssignableQuantity(@NonNull final RefundConfig refundCondig)
 	{
-		if(RefundMode.ALL_MAX_SCALE.equals(refundConfig.getRefundMode()))
-		{
-			return Quantity.infinite(assignedQuantity.getUOM());
-		}
-
 		final Optional<RefundConfig> nextRefundConfig = getRefundContract()
 				.getRefundConfigs()
 				.stream()
-				.filter(config -> config.getMinQty().compareTo(getRefundConfig().getMinQty()) > 0)
+				.filter(config -> config.getMinQty().compareTo(refundCondig.getMinQty()) > 0)
 				.min(Comparator.comparing(RefundConfig::getMinQty));
 
+		final I_C_UOM uomRecord = assignedQuantity.getUOM();
+
 		if (!nextRefundConfig.isPresent())
 		{
-			return Quantity.infinite(assignedQuantity.getUOM());
+			return Quantity.infinite(uomRecord);
 		}
 
 		return Quantity
 				.of(
 						nextRefundConfig.get().getMinQty(),
-						assignedQuantity.getUOM())
+						uomRecord)
 				.subtract(getAssignedQuantity())
-				.subtract(ONE);
+				.subtract(ONE)
+				.max(Quantity.zero(uomRecord));
 	}
\ No newline at end of file
