diff --git a/old_test.java b/new_test.java
index 73fc6ab..7527f9a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -18,7 +18,7 @@
 		final RefundInvoiceCandidate refundInvoiceCandidate = RefundInvoiceCandidate
 				.builder()
 				.assignedQuantity(Quantity.of(TWENTY, refundTestTools.getUomRecord()))
-				.refundConfig(refundConfig1)
+				.refundConfigs(ImmutableList.of(refundConfig1))
 				.bpartnerId(BPartnerId.ofRepoId(10))
 				.invoiceableFrom(NOW)
 				.money(Money.of(ONE, refundTestTools.getCurrency().getId()))
@@ -26,7 +26,7 @@
 				.build();
 
 		// invoke the method under test
-		final Quantity result = refundInvoiceCandidate.computeAssignableQuantity();
+		final Quantity result = refundInvoiceCandidate.computeAssignableQuantity(refundConfig1);
 
 		assertThat(result.getAsBigDecimal()).isEqualByComparingTo("9");
 	}
\ No newline at end of file
