diff --git a/old_product.java b/new_product.java
index 83e1428..28c0510 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,7 +1,6 @@
 @Override
                 public void read(InputStream in, String baseURI, ContentType ct, RDFParserOutput output, Context context)
                 {
-                    Lang lang = RDFLanguages.convert(language) ;
                     LangRIOT parser = RiotReader.createParserQuads(in, lang, baseURI, output) ;
                     parser.parse() ;
                 }
\ No newline at end of file
