diff --git a/old_test.java b/new_test.java
index 5a09856..fb0a1e8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -5,7 +5,7 @@
         {
             StringReader s = new StringReader(x) ;
             Model m = ModelFactory.createDefaultModel() ;
-            WebReader2.read(m, s, null, RDFLanguages.NTriples) ;
+            RDFDataMgr.read(m, s, null, RDFLanguages.NTriples) ;
         }
         
         StringReader s1 = new StringReader("<s> <p> <p> .") ;
