diff --git a/old_product.java b/new_product.java
index 9ff1c70..7fbeb33 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-@Override public void aggregate(IgniteModel<Vector, Double> model, LabeledVector<Double> vector) {
-        Double modelAns = model.predict(vector.features());
-        Double realAns = vector.label();
+@Override public void aggregate(IgniteModel<Vector, L> model, LabeledVector<L> vector) {
+        L modelAns = model.predict(vector.features());
+        L realAns = vector.label();
 
         if (modelAns.equals(falseLabel) && realAns.equals(falseLabel))
             trueNegative += 1;
