diff --git a/old_test.java b/new_test.java
index 431c63d..502a998 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,11 +1,11 @@
 @Test
     public void testAggregate() {
-        BinaryClassificationEvaluationContext ctx = new BinaryClassificationEvaluationContext();
+        BinaryClassificationEvaluationContext<Double> ctx = new BinaryClassificationEvaluationContext<>();
         ctx.aggregate(VectorUtils.of().labeled(1.0));
         assertEquals(ctx.getFirstClassLbl(), 1., 0.);
-        assertEquals(ctx.getSecondClassLbl(), Double.NaN, 0.);
+        assertEquals(ctx.getSecondClassLbl(), null);
 
         ctx.aggregate(VectorUtils.of().labeled(0.0));
-        assertEquals(ctx.getFirstClassLbl(), 0., 0.);
-        assertEquals(ctx.getSecondClassLbl(), 1., 0.);
+        assertEquals(ctx.getFirstClassLbl(), 1., 0.);
+        assertEquals(ctx.getSecondClassLbl(), 0., 0.);
     }
\ No newline at end of file
