diff --git a/old_test.java b/new_test.java
index f651bba..8e5c261 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,9 +10,9 @@
         assertThat(result, notNullValue());
 
         assertThat(result.names(), hasSize(1));
-        assertThat(result.names().get(0), is("node"));
+        assertThat(result.names().get(0), is("nodeId"));
 
-        final String node = result.get("node").asString();
+        final String node = result.get("nodeId").asString();
         assertThat(node, notNullValue());
         assertThat(node, is("node:1"));
     }
\ No newline at end of file
