diff --git a/old_product.java b/new_product.java
index 268eb48..0e2dffc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,6 +6,6 @@
                     DeviceId.deviceId(deviceId)), NODE_ID_NOT_FOUND);
 
         ObjectNode root = mapper().createObjectNode();
-        root.put(NODE, id.id());
+        root.put(NODE_ID, id.id());
         return ok(root).build();
     }
\ No newline at end of file
