diff --git a/old_test.java b/new_test.java
index 2728d0f..4f737d1 100644
--- a/old_test.java
+++ b/new_test.java
@@ -27,6 +27,13 @@
     query(_FT_SEARCH.args(NAME, "1 Exercise", " { 'mode':'all words' }"),
         "Exercise 1");
 
+    query(_FT_SEARCH.args(NAME, "databases xml", " { 'mode':'all words'," +
+        "'distance':{'min':0,'max':1} }"), "Databases and XML");
+    query(_FT_SEARCH.args(NAME, "databases xml", " { 'mode':'all words'," +
+        "'distance':{'max':0} }"), "");
+    query(_FT_SEARCH.args(NAME, "databases xml", " { 'mode':'all words'," +
+        "'window':{'size':3} }"), "Databases and XML");
+
     // check buggy options
     error(_FT_SEARCH.args(NAME, "x", " { 'x':'y' }"), Err.INVALIDOPT);
     error(_FT_SEARCH.args(NAME, "x", " { 'mode':'' }"), Err.INVALIDOPT);
