diff --git a/old_test.java b/new_test.java
index c3e65a3..39b8faf 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
-@Test(expected = ContextAuthenticationException.class)
-	@Verifies(value = "should lockoutUserAfterFiveFailedAttempts", method = "authenticate(String,String)")
-	public void authenticate_shouldLockoutUserAfterFiveFailedAttempts() throws Exception {
+@Test
+    @Verifies(value = "should lockout user after eight failed attempts", method = "authenticate(String,String)")
+    public void authenticate_shouldLockoutUserAfterEightFailedAttempts() throws Exception {
 		// logout after the base setup
 		Context.logout();
 		
@@ -15,7 +15,7 @@
 		}
 		Context.logout();
 		
-		for (int x = 1; x <= 5; x++) {
+		for (int x = 1; x <= 7; x++) {
 			// try to authenticate with a proper 
 			try {
 				dao.authenticate("admin", "not the right password");
@@ -26,7 +26,7 @@
 			}
 		}
 		
-		// those were the first five, now the sixth request 
+		// those were the first seven, now the eigth request 
 		// (with the same user and right pw) should fail
 		dao.authenticate("admin", "test");
 	}
\ No newline at end of file
