diff --git a/old_product.java b/new_product.java
index dc47e45..0eed118 100644
--- a/old_product.java
+++ b/new_product.java
@@ -40,9 +40,11 @@ public User authenticate(String login, String password) throws ContextAuthentica
 			if (lockoutTime != null) {
 				// unlock them after 5 mins, otherwise reset the timestamp 
 				// to now and make them wait another 5 mins 
-				if (new Date().getTime() - lockoutTime > 300000)
+				if (new Date().getTime() - lockoutTime > 300000) {
 					candidateUser.setUserProperty(OpenmrsConstants.USER_PROPERTY_LOGIN_ATTEMPTS, "0");
-				else {
+					candidateUser.removeUserProperty(OpenmrsConstants.USER_PROPERTY_LOCKOUT_TIMESTAMP);
+					saveUserProperties(candidateUser);
+				} else {
 					candidateUser.setUserProperty(OpenmrsConstants.USER_PROPERTY_LOCKOUT_TIMESTAMP, String
 					        .valueOf(new Date().getTime()));
 					throw new ContextAuthenticationException(
@@ -81,7 +83,7 @@ public User authenticate(String login, String password) throws ContextAuthentica
 				
 				attempts++;
 				
-				if (attempts >= 5) {
+				if (attempts >= 8) {
 					// set the user as locked out at this exact time
 					candidateUser.setUserProperty(OpenmrsConstants.USER_PROPERTY_LOCKOUT_TIMESTAMP, String
 					        .valueOf(new Date().getTime()));
