diff --git a/old_product.java b/new_product.java
index 2764127..779bd57 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,7 +5,9 @@
         long traceId = LoggerHelpers.traceEnter(log, "updateStream");
 
         try {
-            authenticateAuthorize(scopeName + "/" + streamName, READ_UPDATE);
+            restAuthHelper.authenticateAuthorize(getAuthorizationHeader(),
+                    AuthResourceRepresentation.ofStreamInScope(scopeName, streamName),
+                    READ_UPDATE);
         } catch (AuthException e) {
             log.warn("Update stream for {} failed due to authentication failure.", scopeName + "/" + streamName);
             asyncResponse.resume(Response.status(Status.fromStatusCode(e.getResponseCode())).build());
