diff --git a/old_test.java b/new_test.java
index c741298..5f532bb 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
     public void testGetPathFromRoot() {
         System.out.println("getPathToRoot");
-        LinkedList<Concept> expResult = new LinkedList<Concept>();
+        LinkedList<Concept> expResult = new LinkedList<>();
         expResult.add(instance.getRoot());
         expResult.add(ad);
         expResult.add(a);
