diff --git a/old_test.java b/new_test.java
index 0bff997..934b694 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,6 @@
 @Test
 	@Verifies(value = "set this care setting to new order", method = "cloneForDiscontinuing(Order)")
-	public void cloneForDiscontinuing_shouldSetThisCareSettingToNewOrder() {
+	public void cloneForDiscontinuing_shouldSetThisCareSettingToNewOrder() throws Exception {
 		Order anOrder = new Order();
 		CareSetting careSetting = new CareSetting();
 		anOrder.setCareSetting(careSetting);
