diff --git a/old_test.java b/new_test.java
index 0ea4792..ee7d946 100644
--- a/old_test.java
+++ b/new_test.java
@@ -24,7 +24,7 @@
             } else if (item.getItemNo() == EventsLogic.ITEM_NO_END) {
                 item.setItemValue("13:00");
             } else if (item.getItemNo() == EventsLogic.ITEM_NO_TIMEZONE) {
-                item.setItemValue("Asia/Tokyo");
+                item.setItemValue("GMT");
             } else if (item.getItemNo() == EventsLogic.ITEM_NO_THE_NUMBER_TO_BE_ACCEPTED) {
                 item.setItemValue("3");
             } else if (item.getItemNo() == EventsLogic.ITEM_NO_PLACE) {
