diff --git a/old_product.java b/new_product.java
index 1381ae4..3bd8507 100644
--- a/old_product.java
+++ b/new_product.java
@@ -44,9 +44,9 @@
                 continue;
             }
             List<KnowledgeItemValuesEntity> values = KnowledgeItemValuesDao.get().selectOnKnowledgeId(event.getKnowledgeId());
-            notifyWeek(event, knowledge, values, now, mailConfigsEntity);
-            notifyDate(event, knowledge, values, now, mailConfigsEntity);
             notifyMin(event, knowledge, values, now, mailConfigsEntity);
+            notifyDate(event, knowledge, values, now, mailConfigsEntity);
+            notifyWeek(event, knowledge, values, now, mailConfigsEntity);
         }
 
     }
\ No newline at end of file
