diff --git a/old_test.java b/new_test.java
index ed43a4b..5125ec5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -30,8 +30,7 @@
 
 		// setup test-object
 		final FileInputChannel<StringRecord> fileInputChannel = new FileInputChannel<StringRecord>(inGate, 1,
-			this.deserializer,
-			null, CompressionLevel.NO_COMPRESSION);
+			this.deserializer, new ChannelID(), new ChannelID(), CompressionLevel.NO_COMPRESSION);
 		fileInputChannel.setInputChannelBroker(inputBroker);
 
 		Whitebox.setInternalState(fileInputChannel, "deserializationBuffer", this.deserializationBuffer);
