diff --git a/old_product.java b/new_product.java
index f81beb2..403f12f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -17,7 +17,7 @@ private T deserializeNextRecord(final T target) throws IOException {
 			}
 		}
 
-		final T nextRecord = this.deserializationBuffer.readData(target, this.uncompressedDataBuffer);
+		final T nextRecord = this.deserializer.readData(target, this.uncompressedDataBuffer);
 
 		if (this.uncompressedDataBuffer.remaining() == 0) {
 			releasedConsumedReadBuffer();
