diff --git a/old_product.java b/new_product.java
index e1cf515..2da8139 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,7 @@
-public static <KType> KTypeOpenHashSet<KType> from(KType... elements)
-    {
-        final KTypeOpenHashSet<KType> set = new KTypeOpenHashSet<KType>(
-            // NOCOMMIT: LEFTOVER!
-            (int) (elements.length * (1 + DEFAULT_LOAD_FACTOR)));
-        set.add(elements);
-        return set;
-    }
\ No newline at end of file
+@SafeVarargs
+  /* #end */
+  public static <KType> KTypeOpenHashSet<KType> from(KType... elements) {
+    final KTypeOpenHashSet<KType> set = new KTypeOpenHashSet<KType>(elements.length);
+    set.addAll(elements);
+    return set;
+  }
\ No newline at end of file
