diff --git a/old_product.java b/new_product.java
index 83520d1..e9c3f19 100644
--- a/old_product.java
+++ b/new_product.java
@@ -22,6 +22,7 @@
         commandEntity.setStatus(updateCommand.getStatus());
         commandEntity.setDescription(updateCommand.getDescription());
         commandEntity.setExecutable(updateCommand.getExecutable());
+        commandEntity.setCheckDelay(updateCommand.getCheckDelay());
         commandEntity.setSetupFile(updateCommand.getSetupFile());
         commandEntity.setConfigs(updateCommand.getConfigs());
         commandEntity.setTags(updateCommand.getTags());
