diff --git a/old_test.java b/new_test.java
index 27769e5..5e97819 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,20 +9,21 @@
         tags.add("hadoop");
 
         final Command updateCommand = new Command.Builder(
-                command.getName(),
-                COMMAND_2_USER,
-                command.getVersion(),
-                CommandStatus.INACTIVE,
-                command.getExecutable()
+            command.getName(),
+            COMMAND_2_USER,
+            command.getVersion(),
+            CommandStatus.INACTIVE,
+            command.getExecutable(),
+            command.getCheckDelay()
         )
-                .withId(command.getId())
-                .withCreated(command.getCreated())
-                .withUpdated(command.getUpdated())
-                .withDescription(command.getDescription())
-                .withTags(tags)
-                .withConfigs(command.getConfigs())
-                .withSetupFile(command.getSetupFile())
-                .build();
+            .withId(command.getId())
+            .withCreated(command.getCreated())
+            .withUpdated(command.getUpdated())
+            .withDescription(command.getDescription())
+            .withTags(tags)
+            .withConfigs(command.getConfigs())
+            .withSetupFile(command.getSetupFile())
+            .build();
 
         this.service.updateCommand(COMMAND_1_ID, updateCommand);
 
