diff --git a/old_product.java b/new_product.java
index eb3da1d..fb4828b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,5 @@
 @Override
 	public Long getCountOfProviders(String name, boolean includeRetired) {
-		Criteria criteria = prepareProviderCriteria(name, includeRetired);
-		criteria.setProjection(Projections.countDistinct("providerId"));
-		return (Long) criteria.uniqueResult();
+	  Criteria criteria = prepareProviderCriteria(name, includeRetired);
+	  return (long) criteria.list().size();
 	}
\ No newline at end of file
