diff --git a/old_test.java b/new_test.java
index 6c4cb61..f57db03 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
             throws Exception {
 
         // Since the servers are started as single nodes thus already bootstrapped.
-        Set<String> set = new HashSet<>();
-        set.add("Key");
-        assertThat(client.handleFailure(set).get()).isEqualTo(true);
+        assertThat(
+                client.handleFailure(Collections.singleton("key"), Collections.emptySet()).get())
+                .isEqualTo(true);
     }
\ No newline at end of file
