diff --git a/old_product.java b/new_product.java
index 7dcb5c7..4a8e906 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
-public CompletableFuture<Boolean> handleFailure(Set nodes) {
+public CompletableFuture<Boolean> handleFailure(Set failedNodes, Set healedNodes) {
         return router.sendMessageAndGetCompletable(CorfuMsgType.MANAGEMENT_FAILURE_DETECTED
-                .payloadMsg(new FailureDetectorMsg(nodes)));
+                .payloadMsg(new FailureDetectorMsg(failedNodes, healedNodes)));
     }
\ No newline at end of file
