diff --git a/old_test.java b/new_test.java
index 3866ed8..8ce2207 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
         DHCPInstance instance = initInstance();
         IPv4Address clientIP = IPv4Address.NONE;
 
-        OFPacketOut dhcpOffer = handler.handleDHCPDiscover(sw, OFPort.of(1), instance, clientIP, dhcpPayload);
+        OFPacketOut dhcpOffer = handler.handleDHCPDiscover(sw, OFPort.of(1), instance, clientIP, dhcpPayload, false);
 
         OFActionOutput output = sw.getOFFactory().actions().buildOutput()
                                 .setMaxLen(0xffFFffFF)
