diff --git a/old_product.java b/new_product.java
index 2bda72c..f156395 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public ServiceCall<SpeechRecognitionResults> recognize(RecognizeOptions recognizeOptions) {
-    Validator.notNull(recognizeOptions, "recognizeOptions cannot be null");
+    Validator.notNull(recognizeOptions, "recognizeSessionlessOptions cannot be null");
     RequestBuilder builder = RequestBuilder.post("/v1/recognize");
     if (recognizeOptions.transferEncoding() != null) {
       builder.header("Transfer-Encoding", recognizeOptions.transferEncoding());
@@ -54,7 +54,7 @@ public ServiceCall<SpeechRecognitionResults> recognize(RecognizeOptions recogniz
       builder.query("speaker_labels", String.valueOf(recognizeOptions.speakerLabels()));
     }
     if (recognizeOptions.audio() != null) {
-      builder.body(RequestBody.create(MediaType.parse(recognizeOptions.contentType()),
+      builder.body(InputStreamRequestBody.create(MediaType.parse(recognizeOptions.contentType()),
           recognizeOptions.audio()));
     }
     MultipartBody.Builder multipartBuilder = new MultipartBody.Builder();
