diff --git a/old_test.java b/new_test.java
index 863db4a..34e3c88 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,7 +9,7 @@
       MockRestResponseChannel restResponseChannel = new MockRestResponseChannel();
       ReadableStreamChannel channel = getReplicasHandler.getReplicas(blobId.getID(), restResponseChannel);
       assertEquals("Unexpected response status", ResponseStatus.Ok, restResponseChannel.getStatus());
-      assertEquals("Unexpected Content-Type", "application/json",
+      assertEquals("Unexpected Content-Type", RestUtils.JSON_CONTENT_TYPE,
           restResponseChannel.getHeader(RestUtils.Headers.CONTENT_TYPE));
       String returnedReplicasStr = RestTestUtils.getJsonizedResponseBody(channel)
           .get(GetReplicasHandler.REPLICAS_KEY)
