diff --git a/old_product.java b/new_product.java
index 38aac9a..418929a 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,10 +4,9 @@ ReadableStreamChannel getReplicas(String blobId, RestResponseChannel restRespons
     long startTime = System.currentTimeMillis();
     ReadableStreamChannel channel = null;
     try {
-      byte[] replicasResponseBytes = getReplicas(blobId).toString().getBytes();
-      restResponseChannel.setHeader(RestUtils.Headers.CONTENT_TYPE, "application/json");
-      restResponseChannel.setHeader(RestUtils.Headers.CONTENT_LENGTH, replicasResponseBytes.length);
-      channel = new ByteBufferReadableStreamChannel(ByteBuffer.wrap(replicasResponseBytes));
+      channel = FrontendUtils.serializeJsonToChannel(getReplicas(blobId));
+      restResponseChannel.setHeader(RestUtils.Headers.CONTENT_TYPE, RestUtils.JSON_CONTENT_TYPE);
+      restResponseChannel.setHeader(RestUtils.Headers.CONTENT_LENGTH, channel.getSize());
     } finally {
       metrics.getReplicasProcessingTimeInMs.update(System.currentTimeMillis() - startTime);
     }
