diff --git a/old_product.java b/new_product.java
index 3c2c2ef..407c5e5 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 @POST
   @Path(PATH_PARAM + LIST_STATUS)
-  @ReturnType("java.util.List<alluxio.client.file.URIStatus>")
+  @ApiOperation(value = "List the URIStatuses of the path's children",
+      response = java.util.List.class)
   public Response listStatus(@PathParam("path") final String path,
       final ListStatusOptions options) {
     return RestUtils.call(new RestUtils.RestCallable<List<URIStatus>>() {
