diff --git a/old_product.java b/new_product.java
index 01b94c1..ada1057 100644
--- a/old_product.java
+++ b/new_product.java
@@ -31,6 +31,7 @@ public static ClientDetailsEntity parse(String jsonString) {
 			}
 
 			c.setGrantTypes(getAsStringSet(o, "grant_types"));
+			c.setResponseTypes(getAsStringSet(o, "response_types"));
 			c.setPolicyUri(getAsString(o, "policy_uri"));
 			c.setJwksUri(getAsString(o, "jwks_uri"));
 
