diff --git a/old_product.java b/new_product.java
index 0006378..b635c00 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,14 @@
 private Iter event(final QueryContext ctx) throws QueryException {
-    String msg = expr[3].toString().equals("m") ? expr[2].toString() : null;
-    Value v = expr[0].value(ctx);
-    if(msg == null) {
+    String name = expr[0].toString();
+    String q2 = expr[2].toString();
+    String msg = "";
+    Value v = expr[1].value(ctx);
+    if(q2.replaceAll("\"", "").trim().isEmpty()) {
       ctx.updating = true;
-      msg = ctx.value(expr[0]).toString();
+      msg = ctx.value(expr[1]).toString();
+    } else {
+      msg = ctx.value(expr[2]).toString();
     }
-    ctx.context.events.notify(ctx.context.session, expr[1].toString(), msg);
+    ctx.context.events.notify(ctx.context.session, name, msg);
     return v.iter();
   }
\ No newline at end of file
