diff --git a/old_product.java b/new_product.java
index a4a28ec..add47c8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,2 +1,2 @@
 @Authorized( { PrivilegeConstants.GET_PATIENT_COHORTS })
-	public List<Cohort> getCohortsContainingPatient(Patient patient, Boolean voided) throws APIException;
\ No newline at end of file
+	public List<Cohort> getCohortsContainingPatient(Patient patient, boolean includeVoided, Date asOfDate);
\ No newline at end of file
