diff --git a/old_product.java b/new_product.java
index 6bc7a6b..291d050 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,7 @@
 		for (int i = 0; i < clusters.size(); i++) {
 			List<Node> cluster = clusters.get(i);
 			int clusterSize = cluster.size();
-			if( clusterSize < 5 ) // 3 x 3 grid has 5 elements
+			if( clusterSize < 6 ) // 3 x 4 grid has 6 elements
 				continue;
 
 			computeNodeInfo(ellipses, cluster);
@@ -29,9 +29,10 @@
 			List<List<NodeInfo>> grid = new ArrayList<>();
 
 			// traverse along the axis with closely spaced circles
-			double distLeft = corner.distance(corner.left);
-			double distRight = corner.distance(corner.right);
-			NodeInfo next = distLeft < distRight ? corner.left : corner.right;
+//			double distLeft = corner.distance(corner.left);
+//			double distRight = corner.distance(corner.right);
+//			NodeInfo next = distLeft < distRight ? corner.left : corner.right;
+			NodeInfo next = corner.left;
 			next.marked = true;
 
 //			System.out.println("corner "+corner.ellipse.center);
