diff --git a/old_product.java b/new_product.java
index 9b9398f..ae09074 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,14 @@
 @GET
     @Produces(MediaType.APPLICATION_JSON)
-    @Path("ids/name")
-    public Response getAppIdByName(@QueryParam("name") String name) {
+    @Path("ids/entry")
+    public Response getAppIdByName(@QueryParam("id") Short id,
+                                   @QueryParam("name") String name) {
         CoreService service = get(CoreService.class);
-        ApplicationId appId = service.getAppId(name);
+        ApplicationId appId = null;
+        if (id != null) {
+            appId = service.getAppId(id);
+        } else if (name != null) {
+            appId = service.getAppId(name);
+        }
         return response(appId);
     }
\ No newline at end of file
