diff --git a/old_test.java b/new_test.java
index 3fe02e0..9c5e32a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,7 +4,7 @@
         replay(coreService);
 
         WebTarget wt = target();
-        String response = wt.path("applications/ids/name")
+        String response = wt.path("applications/ids/entry")
                 .queryParam("name", "app2").request().get(String.class);
 
         JsonObject result = Json.parse(response).asObject();
