diff --git a/old_product.java b/new_product.java
index 1824154..159a002 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-static Collection<MigrationInfoImpl> collectMigrations(Collection<MigrationResolver> migrationResolvers) {
-        Set<MigrationInfoImpl> migrations = new HashSet<MigrationInfoImpl>();
+static Collection<ResolvedMigration> collectMigrations(Collection<MigrationResolver> migrationResolvers) {
+        Set<ResolvedMigration> migrations = new HashSet<ResolvedMigration>();
         for (MigrationResolver migrationResolver : migrationResolvers) {
             migrations.addAll(migrationResolver.resolveMigrations());
         }
