diff --git a/old_test.java b/new_test.java
index 49c1dc9..241aa61 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,18 +1,18 @@
 @Test
     public void collectMigrations() {
         MigrationResolver migrationResolver = new MigrationResolver() {
-            public List<MigrationInfoImpl> resolveMigrations() {
-                List<MigrationInfoImpl> migrations = new ArrayList<MigrationInfoImpl>();
+            public List<ResolvedMigration> resolveMigrations() {
+                List<ResolvedMigration> migrations = new ArrayList<ResolvedMigration>();
 
-                migrations.add(createTestMigration(com.googlecode.flyway.core.api.MigrationType.JAVA, "1", "Description", "Migration1", 123));
-                migrations.add(createTestMigration(com.googlecode.flyway.core.api.MigrationType.JAVA, "1", "Description", "Migration1", 123));
-                migrations.add(createTestMigration(com.googlecode.flyway.core.api.MigrationType.SQL, "2", "Description2", "Migration2", 1234));
+                migrations.add(createTestMigration(MigrationType.JAVA, "1", "Description", "Migration1", 123));
+                migrations.add(createTestMigration(MigrationType.JAVA, "1", "Description", "Migration1", 123));
+                migrations.add(createTestMigration(MigrationType.SQL, "2", "Description2", "Migration2", 1234));
                 return migrations;
             }
         };
         Collection<MigrationResolver> migrationResolvers = new ArrayList<MigrationResolver>();
         migrationResolvers.add(migrationResolver);
 
-        Collection<MigrationInfoImpl> migrations = CompositeMigrationResolver.collectMigrations(migrationResolvers);
+        Collection<ResolvedMigration> migrations = CompositeMigrationResolver.collectMigrations(migrationResolvers);
         assertEquals(2, migrations.size());
     }
\ No newline at end of file
