diff --git a/old_test.java b/new_test.java
index 9c21849..304961a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,7 @@
 @Test
   public void createFileTest() throws Exception {
     // created nested file
-    mTree.createPath(NESTED_FILE_URI, sNestedFileOptions);
+    createPath(mTree, NESTED_FILE_URI, sNestedFileOptions);
     Inode<?> nestedFile = getInodeByPath(mTree, NESTED_FILE_URI);
     Assert.assertEquals("file", nestedFile.getName());
     Assert.assertEquals(2, nestedFile.getParentId());
