diff --git a/old_product.java b/new_product.java
index ff3e079..be0f3b8 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ public long createFile(AlluxioURI path, CreateFileOptions options)
           BlockInfoException, IOException, FileDoesNotExistException {
     MasterContext.getMasterSource().incCreateFileOps(1);
     long flushCounter = AsyncJournalWriter.INVALID_FLUSH_COUNTER;
-    try (InodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.WRITE)) {
+    try (LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.WRITE)) {
       mPermissionChecker.checkParentPermission(FileSystemAction.WRITE, inodePath);
       mMountTable.checkUnderWritableMountPoint(path);
       flushCounter = createFileAndJournal(inodePath, options);
