diff --git a/old_product.java b/new_product.java
index 6d8a8a5..40d5ac4 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public AssociativeArray estimateEfficiency(Map<Object, DeaRecord> id2DeaRecordMapDatabase, Map<Object, DeaRecord> id2DeaRecordMapEvaluation) {        
-        AssociativeArray evaluatedResults = new AssociativeArray(new LinkedHashMap<>());
+        AssociativeArray evaluatedResults = new AssociativeArray();
         
         List<LPSolver.LPConstraint> constraints = new ArrayList<>();
         
