diff --git a/old_test.java b/new_test.java
index 2524cc0..ba2744d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -27,7 +27,7 @@
         Map<Object, DataEnvelopmentAnalysis.DeaRecord> id2DeaRecordMapEvaluation = new LinkedHashMap<>(id2DeaRecordMapDatabase);
         
         DataEnvelopmentAnalysis instance = new DataEnvelopmentAnalysis();
-        AssociativeArray expResult = new AssociativeArray(new LinkedHashMap<>());
+        AssociativeArray expResult = new AssociativeArray();
         
         expResult.put("Depot1", 0.82038345105954);
         expResult.put("Depot2", 0.94174174174174);
