diff --git a/old_product.java b/new_product.java
index 9674c59..95e5e7e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 public BootstrapBuilder peerAddress(final PeerAddress peerAddress) {
-        if (peerAddress != null && peerAddress.peerId().equals(Number160.ZERO)) {
+        if (peerAddress != null && peerAddress.peerId().equals(Number256.ZERO)) {
             logger.warn("Peer address with peer ID zero provided. Bootstrapping is impossible, because no peer with peer ID set to zero is allowed to exist.");
             return this;
         }
