diff --git a/old_test.java b/new_test.java
index 3c345f3..4ebc61b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,8 +1,8 @@
 @Test
     public void testPeerAddress4() throws UnknownHostException {
-        PeerAddress pa1 = Utils2.createPeerAddress(new Number160("0x857e35a42e444522456"),
+        PeerAddress pa1 = Utils2.createPeerAddress(new Number256("0x857e35a42e444522456"),
                 InetAddress.getByName("0123:4567:89ab:cdef:0123:4567:89ab:cdef"), RND.nextInt(BIT_16));
-        PeerAddress pa2 = Utils2.createPeerAddress(new Number160("0x657435a424444522456"),
+        PeerAddress pa2 = Utils2.createPeerAddress(new Number256("0x657435a424444522456"),
                 InetAddress.getByName("f123:4567:89ab:cdef:0123:4567:89ab:cdef"), RND.nextInt(BIT_16));
 
         final int length = 200;
