diff --git a/old_product.java b/new_product.java
index 7aaa742..99d4bad 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,8 +1,9 @@
 @Override
-    public Long geoAdd(byte[] key, Map<byte[], GeoCoordinate> memberCoordinateMap) {
-        Long result = delegate.geoAdd(key, memberCoordinateMap);
-        if (isFutureConversion()){
-            addResultConverter(identityConverter);
-        }
-        return result;
-    }
\ No newline at end of file
+	public Long geoAdd(byte[] key, Map<byte[], Point> memberCoordinateMap) {
+
+		Long result = delegate.geoAdd(key, memberCoordinateMap);
+		if (isFutureConversion()) {
+			addResultConverter(identityConverter);
+		}
+		return result;
+	}
\ No newline at end of file
