diff --git a/old_test.java b/new_test.java
index 772f4d5..bf04f0d 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,8 @@
 @Test
-    public void testGeoAdd(){
-        doReturn(1l).when(nativeConnection).geoAdd(fooBytes, 1.23232, 34.2342434, barBytes);
-        actual.add(connection.geoAdd(foo, 1.23232, 34.2342434, bar));
-        verifyResults(Arrays.asList(new Object[] { 1l }));
-    }
\ No newline at end of file
+	public void testGeoAdd() {
+
+		doReturn(1l).when(nativeConnection).geoAdd(fooBytes, new Point(1.23232, 34.2342434), barBytes);
+
+		actual.add(connection.geoAdd(foo, new Point(1.23232, 34.2342434), bar));
+		verifyResults(Collections.singletonList(1L));
+	}
\ No newline at end of file
