diff --git a/old_product.java b/new_product.java
index 2f54fef..ed98f04 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,11 +1,10 @@
 @Override
     public ZonedDateTime withEarlierOffsetAtOverlap() {
-        ZoneOffsetTransition trans = getZone().getRules().getTransition(getDateTime());
-        if (trans != null) {
-            ZoneOffset offset = trans.getOffsetBefore();
-            if (offset.equals(getOffset()) == false) {
-                OffsetDateTime newDT = dateTime.withOffsetSameLocal(offset);
-                return new ZonedDateTime(newDT, zone);
+        ZoneOffsetTransition trans = getZone().getRules().getTransition(dateTime);
+        if (trans != null && trans.isOverlap()) {
+            ZoneOffset earlierOffset = trans.getOffsetBefore();
+            if (earlierOffset.equals(offset) == false) {
+                return new ZonedDateTime(dateTime, earlierOffset, zone);
             }
         }
         return this;
