diff --git a/old_test.java b/new_test.java
index 9e7c1f1..08e423a 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,7 +1,6 @@
 @Test(groups={"tck"})
     public void test_withEarlierOffsetAtOverlap_notAtOverlap() {
-        OffsetDateTime odt = OffsetDateTime.of(TEST_LOCAL_2008_06_30_11_30_59_500, OFFSET_0200);
-        ZonedDateTime base = ZonedDateTime.of(odt, ZONE_PARIS);
+        ZonedDateTime base = ZonedDateTime.ofStrict(TEST_LOCAL_2008_06_30_11_30_59_500, OFFSET_0200, ZONE_PARIS);
         ZonedDateTime test = base.withEarlierOffsetAtOverlap();
         assertEquals(test, base);  // not changed
     }
\ No newline at end of file
