diff --git a/old_product.java b/new_product.java
index d9309e9..33dc474 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,10 @@
 public boolean process( ImageUInt8 thresholded )
 	{
+		// discard old results
+		interestPoints = new ArrayList<Point2D_F64>();
+		interestSquares = new ArrayList<QuadBlob>();
+
+		// initialize data structures
 		binaryA.reshape(thresholded.width,thresholded.height);
 		binaryB.reshape(thresholded.width,thresholded.height);
 
