diff --git a/old_test.java b/new_test.java
index 31720ad..ec1e579 100644
--- a/old_test.java
+++ b/new_test.java
@@ -14,13 +14,13 @@
 		// specify an initial transform
 		alg.setInitialTransform(initial);
 
-		// the first time it processes an image it should always return true since no motion is estimated
-		assertTrue(alg.process(input));
+		// the first time it processes an image it should always return false since no motion can be estimated
+		assertFalse(alg.process(input));
+		assertEquals(0, tracker.numSpawn);
 
-		// the transform should be the same as the initial one
-		// and requested that new tracks be spawned
-		Se2_F32 found = alg.getWorldToCurr();
-		assertEquals(initial.getX(), found.getX(), 1e-8);
+		// make the current frame into the keyframe
+		// request that the current frame is a keyframe
+		alg.changeKeyFrame();
 		assertEquals(1, tracker.numSpawn);
 
 		// now it should compute some motion
@@ -35,6 +35,6 @@
 
 		// see if this transform was correctly computed
 		Se2_F32 worldToCurr = initial.concat(keyToCurr, null);
-		found = alg.getWorldToCurr();
+		Se2_F32 found = alg.getWorldToCurr();
 		assertEquals(worldToCurr.getX(), found.getX(), 1e-8);
 	}
\ No newline at end of file
