diff --git a/old_product.java b/new_product.java
index 31fee5f..74b2ed1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -13,7 +13,7 @@ public void process(SimpleImageSequence<T> sequence) {
 			tracker.process(frame);
 
 			// if there are too few tracks spawn more
-			if( tracker.getActiveTracks().size() < 100 )
+			if( tracker.getActiveTracks(null).size() < 100 )
 				tracker.spawnTracks();
 
 			// visualize tracking results
