diff --git a/old_test.java b/new_test.java
index dbed011..1f0ec76 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,4 @@
 @Test
     public void testIsEmpty() {
-        assertTrue(instance.isEmpty("name").getQueryCriterions()
-                .contains(new IsEmptyCriterion("name")));
+    	assertEquals(new IsEmptyCriterion("name"), instance.isEmpty("name").getQueryCriterion());
     }
\ No newline at end of file
