diff --git a/old_product.java b/new_product.java
index 281b71b..02436e9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
-public void setApplications(final List<ApplicationEntity> applications) throws GeniePreconditionException {
+public void setApplications(@Nullable final List<ApplicationEntity> applications)
+        throws GeniePreconditionException {
         if (applications != null
-            && applications.stream().map(ApplicationEntity::getId).distinct().count() != applications.size()) {
+            && applications.stream().map(ApplicationEntity::getUniqueId).distinct().count() != applications.size()) {
             throw new GeniePreconditionException("List of applications to set cannot contain duplicates");
         }
 
