diff --git a/old_test.java b/new_test.java
index 59b83a7..a0937ea 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,9 +4,9 @@
         Assert.assertTrue(this.c.getApplications().isEmpty());
         final List<ApplicationEntity> applicationEntities = new ArrayList<>();
         final ApplicationEntity one = new ApplicationEntity();
-        one.setId("one");
+        one.setUniqueId("one");
         final ApplicationEntity two = new ApplicationEntity();
-        two.setId("two");
+        two.setUniqueId("two");
         applicationEntities.add(one);
         applicationEntities.add(two);
         this.c.setApplications(applicationEntities);
