diff --git a/old_test.java b/new_test.java
index 8c3a508..03f3ae5 100644
--- a/old_test.java
+++ b/new_test.java
@@ -37,9 +37,9 @@
     query(_DB_INFO.args(NAME) + "//ftindex/text()", "false");
 
     new Open(NAME).execute(context);
-    new CreateIndex(Commands.CmdIndex.TEXT).execute(context);
-    new CreateIndex(Commands.CmdIndex.ATTRIBUTE).execute(context);
-    new CreateIndex(Commands.CmdIndex.FULLTEXT).execute(context);
+    new CreateIndex(CmdIndex.TEXT).execute(context);
+    new CreateIndex(CmdIndex.ATTRIBUTE).execute(context);
+    new CreateIndex(CmdIndex.FULLTEXT).execute(context);
     new Close().execute(context);
 
     query(_DB_OPTIMIZE.args(NAME));
@@ -48,9 +48,9 @@
     query(_DB_INFO.args(NAME) + "//ftindex/text()", "true");
 
     new Open(NAME).execute(context);
-    new DropIndex(Commands.CmdIndex.TEXT).execute(context);
-    new DropIndex(Commands.CmdIndex.ATTRIBUTE).execute(context);
-    new DropIndex(Commands.CmdIndex.FULLTEXT).execute(context);
+    new DropIndex(CmdIndex.TEXT).execute(context);
+    new DropIndex(CmdIndex.ATTRIBUTE).execute(context);
+    new DropIndex(CmdIndex.FULLTEXT).execute(context);
     new Close().execute(context);
 
     query(_DB_OPTIMIZE.args(NAME));
