diff --git a/old_product.java b/new_product.java
index 9185538..203a452 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,9 +1,8 @@
-public static CipherSuite forJavaName(String javaName) {
+public static synchronized CipherSuite forJavaName(String javaName) {
     CipherSuite result = INSTANCES.get(javaName);
     if (result == null) {
-      CipherSuite sample = new CipherSuite(javaName);
-      CipherSuite canonical = INSTANCES.putIfAbsent(javaName, sample);
-      result = (canonical == null) ? sample : canonical;
+      result = new CipherSuite(javaName);
+      INSTANCES.put(javaName, result);
     }
     return result;
   }
\ No newline at end of file
