diff --git a/old_product.java b/new_product.java
index 3702db3..a0f9b8d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,5 @@
 @Override
   protected int getBucketedSizeForValue(Bitmap value) {
     Preconditions.checkNotNull(value);
-    final int allocationByteCount = value.getAllocationByteCount();
-    return allocationByteCount / Bitmaps.BYTES_PER_PIXEL;
+    return value.getAllocationByteCount();
   }
\ No newline at end of file
