diff --git a/old_product.java b/new_product.java
index 6b53034..d0372d0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -15,7 +15,7 @@ public static void writeJavaScriptUrl(final Response response, final CharSequenc
 			response.write("charset=\"" + Strings.escapeMarkup(charset) + "\" ");
 		}
 		response.write("src=\"");
-		response.write(Strings.escapeMarkup(url));
+		response.write(url);
 		response.write("\"></script>");
 		response.write("\n");
 	}
\ No newline at end of file
