diff --git a/old_product.java b/new_product.java
index d7013b1..907cfc0 100644
--- a/old_product.java
+++ b/new_product.java
@@ -5,18 +5,18 @@ private synchronized Item createDir(final QueryContext ctx) throws QueryExceptio
     try {
       f = path.getCanonicalFile();
     } catch(final IOException ex) {
-      throw FILE_PATH.get(info, path);
+      throw FILE_IE_PATH.get(info, path);
     }
 
     // find lowest existing path
     while(!f.exists()) {
       f = f.getParentFile();
-      if(f == null) throw FILE_PATH.get(info, path);
+      if(f == null) throw FILE_IE_PATH.get(info, path);
     }
     // warn if lowest path points to a file
-    if(f.isFile()) throw FILE_EXISTS.get(info, path);
+    if(f.isFile()) throw FILE_E.get(info, path);
 
     // only create directories if path does not exist yet
-    if(!path.exists() && !path.mkdirs()) throw FILE_CREATE.get(info, path);
+    if(!path.exists() && !path.mkdirs()) throw FILE_IE_DIR.get(info, path);
     return null;
   }
\ No newline at end of file
