diff --git a/old_test.java b/new_test.java
index 4abd45c..6990904 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,9 +3,8 @@
         final Command command2 = this.service.getCommand(COMMAND_2_ID);
         Assert.assertTrue(command2.getApplications().isEmpty());
 
-        final Application app = this.appService.getApplication(APP_1_ID);
-        final Set<Application> apps = new HashSet<>();
-        apps.add(app);
+        final Set<String> appIds = new HashSet<>();
+        appIds.add(APP_1_ID);
         final List<Command> preCommands = this.appService.getCommandsForApplication(APP_1_ID, null);
         Assert.assertEquals(1, preCommands.size());
         Assert.assertEquals(1, preCommands
@@ -14,7 +13,7 @@
                         .count()
         );
 
-        this.service.setApplicationsForCommand(COMMAND_2_ID, apps);
+        this.service.setApplicationsForCommand(COMMAND_2_ID, appIds);
 
         final List<Command> savedCommands = this.appService.getCommandsForApplication(APP_1_ID, null);
         Assert.assertEquals(2, savedCommands.size());
