diff --git a/old_product.java b/new_product.java
index fce26e8..0473595 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,10 +1,24 @@
-public static <T> Collection<List<T>> combinations(List<T> elements, int elementCount) {
-        Set<Set<T>> combinations = combinationsSet(elements, elementCount);
-
-        Collection<List<T>> result = new ArrayList<>();
-        for(Set<T> linkedset : combinations) {
-            result.add(new ArrayList<>(linkedset));
+public static <T> Set<Set<T>> combinations(Set<T> elements, int subsetSize) {
+        Set<Set<T>> resultingCombinations = new HashSet<> ();
+        int totalSize=elements.size();
+        if (subsetSize == 0) {
+            resultingCombinations.add(new HashSet<>());
         }
+        else if (subsetSize <= totalSize) {
+            Set<T> remainingElements = elements;
+
+            Iterator<T> it = remainingElements.iterator();
+            T X = it.next();
+            it.remove();
+
+            resultingCombinations.addAll(combinations(remainingElements, subsetSize));
 
-        return result;
+            for (Set<T> combination : combinations(remainingElements, subsetSize-1)) {
+                combination.add(X);
+                resultingCombinations.add(combination);
+            }
+
+            remainingElements.add(X);
+        }
+        return resultingCombinations;
     }
\ No newline at end of file
