diff --git a/old_product.java b/new_product.java
index f2ad31d..e60c547 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
-public static String getOwnerFromLoginModule() {
+public static String getOwnerFromLoginModule(AlluxioConfiguration conf) {
     try {
-      return LoginUser.get().getName();
+      return LoginUser.get(conf).getName();
     } catch (UnauthenticatedException | UnsupportedOperationException e) {
       return "";
     }
