diff --git a/old_test.java b/new_test.java
index a927db1..dbd58be 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,13 @@
 @Test
   public void getOwnerFromLoginModule() throws Exception {
     // When security is not enabled, user and group are not set
-    Configuration.set(PropertyKey.SECURITY_AUTHENTICATION_TYPE, AuthType.NOSASL.getAuthName());
-    Assert.assertEquals("", SecurityUtils.getOwnerFromLoginModule());
+    mConfiguration.set(PropertyKey.SECURITY_AUTHENTICATION_TYPE, AuthType.NOSASL.getAuthName());
+    Assert.assertEquals("", SecurityUtils.getOwnerFromLoginModule(mConfiguration));
 
     // When authentication is enabled, user and group are inferred from login module
-    Configuration.set(PropertyKey.SECURITY_AUTHENTICATION_TYPE, AuthType.SIMPLE.getAuthName());
-    Configuration.set(PropertyKey.SECURITY_LOGIN_USERNAME, "test_login_user");
-    Configuration.set(PropertyKey.SECURITY_GROUP_MAPPING_CLASS,
+    mConfiguration.set(PropertyKey.SECURITY_AUTHENTICATION_TYPE, AuthType.SIMPLE.getAuthName());
+    mConfiguration.set(PropertyKey.SECURITY_LOGIN_USERNAME, "test_login_user");
+    mConfiguration.set(PropertyKey.SECURITY_GROUP_MAPPING_CLASS,
         IdentityUserGroupsMapping.class.getName());
-    Assert.assertEquals("test_login_user", SecurityUtils.getOwnerFromLoginModule());
+    Assert.assertEquals("test_login_user", SecurityUtils.getOwnerFromLoginModule(mConfiguration));
   }
\ No newline at end of file
