diff --git a/old_test.java b/new_test.java
index 49406a9..7c9021b 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,5 @@
 @Test
     public void testNewColumn() {
-        DoubleColumn column = (DoubleColumn) TypeUtils.newColumn("test", DOUBLE);
+        NumberColumn column = (NumberColumn) TypeUtils.newColumn("test", NUMBER);
         assertThat(column, notNullValue());
     }
\ No newline at end of file
