diff --git a/old_product.java b/new_product.java
index 5c200c7..02ab890 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,6 +4,5 @@ public void validate(Object obj, Errors errors) {
 			errors.rejectValue("conceptClass", "error.general");
 		} else {
 			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "name", "error.name");
-			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "description", "error.description");
 		}
 	}
\ No newline at end of file
