diff --git a/old_test.java b/new_test.java
index b6af76f..bca1609 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,10 +2,10 @@
     public void testOnTopicCreated_retryTimeout(TestContext context) {
         TopicMetadata topicMetadata = getTopicMetadata();
 
-        mockKafka.setTopicMetadataResponse(topicName, null, new UnknownTopicOrPartitionException());
+        mockKafka.setTopicMetadataResponse(topicName, null, null);
 
         Async async = context.async();
-        op.onTopicCreated(topicName, ar -> {
+        controller.onTopicCreated(topicName, ar -> {
             assertFailed(context, ar);
             context.assertEquals(ar.cause().getClass(), MaxAttemptsExceededException.class);
             mockK8s.assertNotExists(context, mapName);
