diff --git a/old_test.java b/new_test.java
index fa951eb..085c259 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,8 +2,8 @@
     public void testStatusStringListener() {
         System.out.println("statusStringListener");
         String state = "";
-        Point3d machineCoord = null;
-        Point3d workCoord = null;
+        Position machineCoord = null;
+        Position workCoord = null;
         GUIBackend instance = new GUIBackend();
         instance.statusStringListener(state, machineCoord, workCoord);
         // TODO review the generated test code and remove the default call to fail.
