diff --git a/old_product.java b/new_product.java
index aed83de..1b3392d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,6 @@
 @Override
     public void createStream(final String scope, final CreateStreamRequest createStreamRequest,
-                             final AsyncResponse asyncResponse) {
+            final SecurityContext securityContext, final AsyncResponse asyncResponse) {
         long traceId = LoggerHelpers.traceEnter(log, "createStream");
 
         StreamConfiguration streamConfiguration = ModelHelper.getCreateStreamConfig(createStreamRequest, scope);
