diff --git a/old_test.java b/new_test.java
index fa89f6d..31db4d7 100644
--- a/old_test.java
+++ b/new_test.java
@@ -3,7 +3,7 @@
     final QueryContext qc = new QueryContext(context);
     try {
       qc.parseLibrary(
-          "module namespace m='foo'; declare function m:foo() { m:bar() }; ", "");
+          "module namespace m='foo'; declare function m:foo() { m:bar() }; ", "", null);
       fail("Unknown function 'm:bar()' was not detected.");
     } catch(final QueryException e) {
       assertSame(Err.FUNCUNKNOWN, e.err());
