diff --git a/old_product.java b/new_product.java
index f31a3e7..b5ee4eb 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,7 @@ public static FitData<EllipseRotated_F64> fitEllipse_F64( List<Point2D_F64> poin
 															  FitData<EllipseRotated_F64> outputStorage ) {
 
 		if( outputStorage == null ) {
-			outputStorage = new FitData<EllipseRotated_F64>(new EllipseRotated_F64());
+			outputStorage = new FitData<>(new EllipseRotated_F64());
 		}
 
 		// Compute the optimal algebraic error
