diff --git a/old_test.java b/new_test.java
index 03ab2f4..17e8abe 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
 	public void fitEllipse_F64() {
 		EllipseRotated_F64 rotated = new EllipseRotated_F64(1,2,3,2,-0.05);
 
-		List<Point2D_F64> points = new ArrayList<Point2D_F64>();
+		List<Point2D_F64> points = new ArrayList<>();
 		for( int i = 0; i < 20; i++ ) {
 			double theta = 2.0*(double)Math.PI*i/20;
 			points.add(UtilEllipse_F64.computePoint(theta, rotated, null));
