diff --git a/old_test.java b/new_test.java
index f30f072..534e945 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,5 @@
 @Test
-    public void testReadPropertiesToAliasBean(){
-        DangaMemCachedConfig dangaMemCachedConfig = ResourceBundleUtil
-                        .readPropertiesToAliasBean("messages.memcached", DangaMemCachedConfig.class);
+    public void testReadToAliasBean(){
+        DangaMemCachedConfig dangaMemCachedConfig = ResourceBundleUtil.readToAliasBean("messages.memcached", DangaMemCachedConfig.class);
         LOGGER.debug(JsonUtil.format(dangaMemCachedConfig));
     }
\ No newline at end of file
