diff --git a/old_product.java b/new_product.java
index 46dc738..ee4a44b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-public static <T> T readPropertiesToAliasBean(String baseName,Class<T> aliasBeanClass){
+public static <T> T readToAliasBean(String baseName,Class<T> aliasBeanClass){
         Validate.notBlank(baseName, "baseName can't be null/empty!");
         Validate.notNull(aliasBeanClass, "aliasBeanClass can't be null!");
-        return BeanUtil.populateAliasBean(newInstance(aliasBeanClass), readPropertiesToMap(baseName));
+        return BeanUtil.populateAliasBean(newInstance(aliasBeanClass), readToMap(baseName));
     }
\ No newline at end of file
