diff --git a/old_product.java b/new_product.java
index ea79f62..cd98649 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,15 +3,19 @@ protected Object readResolve() throws InvalidObjectException {
                 // text.0C=cannot resolve subclasses
                 throw new InvalidObjectException(Messages.getString("text.0C")); //$NON-NLS-1$
             }
-            if (this.name.equals(INPUT_METHOD_SEGMENT.name)) {
+            // BEGIN android-changed
+            // call getName() only once
+            String name = this.getName();
+            if (name.equals(INPUT_METHOD_SEGMENT.getName())) {
                 return INPUT_METHOD_SEGMENT;
             }
-            if (this.name.equals(LANGUAGE.name)) {
+            if (name.equals(LANGUAGE.getName())) {
                 return LANGUAGE;
             }
-            if (this.name.equals(READING.name)) {
+            if (name.equals(READING.getName())) {
                 return READING;
             }
+            // END android-changed
             // text.02=Unknown attribute
             throw new InvalidObjectException(Messages.getString("text.02")); //$NON-NLS-1$
         }
\ No newline at end of file
