diff --git a/old_product.java b/new_product.java
index d17e858..a8c311d 100644
--- a/old_product.java
+++ b/new_product.java
@@ -52,7 +52,7 @@ public boolean init(String[] args) throws ParseException, IOException, Exception
       dumpOutDebugInfo();
     }
 
-    this.dnmgr = new ModuleManager(topology);
+    this.dnmgr = new StreamingContainerManager(topology);
 
     // start RPC server
     rpcImpl = new StreamingContainerParent(this.getClass().getName(), dnmgr);
@@ -60,7 +60,7 @@ public boolean init(String[] args) throws ParseException, IOException, Exception
     rpcImpl.start();
     LOG.info("Container callback server listening at " + rpcImpl.getAddress());
 
-    LOG.info("Initializing logical topology with {} nodes in {} containers", topology.getAllOperators().size(), dnmgr.getNumRequiredContainers());
+    LOG.info("Initializing logical topology with {} operators in {} containers", topology.getAllOperators().size(), dnmgr.getNumRequiredContainers());
 
     // start buffer server
     com.malhartech.bufferserver.Server s = new Server(0);
