diff --git a/old_test.java b/new_test.java
index 7ff2a97..308c033 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,9 @@
 @Test
     public void substring(){
-        LOGGER.info(StringUtil.substring(TEXT, "jinxin".length()));
-        String text1 = "Index: src/main/java/com/jumbo/shop/web/command/PageCacheCommand.java";
-        LOGGER.info(StringUtil.substring(text1, "Index: ".length()));
+        assertEquals(
+                        "src/main/java/com/jumbo/shop/web/command/PageCacheCommand.java",
+                        StringUtil.substring("Index: src/main/java/com/jumbo/shop/web/command/PageCacheCommand.java", "Index: ".length()));
+        assertEquals(".feilong", StringUtil.substring(TEXT, "jinxin".length()));
+        assertEquals(".feilong", StringUtil.substring(TEXT, 6));
+        assertEquals("ng", StringUtil.substring(TEXT, -2));
     }
\ No newline at end of file
