diff --git a/old_product.java b/new_product.java
index 4662bec..25a5dea 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public void delete(OperationContext context,
                      String account, String application,
-                     String type, String name)
+                     String type, String id)
       throws OperationException;
\ No newline at end of file
