diff --git a/old_product.java b/new_product.java
index 492f5f5..a8754ae 100644
--- a/old_product.java
+++ b/new_product.java
@@ -32,7 +32,7 @@ public static double root(DifferentiableFunction func, double x1, double x2, dou
         double dxold = Math.abs(x2 - x1);
         double dx = dxold;
         double f = func.apply(rts);
-        double df = func.df(rts);
+        double df = func.g(rts);
         for (int iter = 1; iter <= maxIter; iter++) {
             if ((((rts - xh) * df - f) * ((rts - xl) * df - f) > 0.0) || (Math.abs(2.0 * f) > Math.abs(dxold * df))) {
                 dxold = dx;
@@ -63,7 +63,7 @@ public static double root(DifferentiableFunction func, double x1, double x2, dou
             }
 
             f = func.apply(rts);
-            df = func.df(rts);
+            df = func.g(rts);
             if (f < 0.0) {
                 xl = rts;
             } else {
