diff --git a/old_test.java b/new_test.java
index 2254b56..c1bb208 100644
--- a/old_test.java
+++ b/new_test.java
@@ -4,12 +4,12 @@
         Function func = new DifferentiableFunction() {
 
             @Override
-            public double apply(double x) {
+            public double f(double x) {
                 return x * x * x + x * x - 5 * x + 3;
             }
 
             @Override
-            public double df(double x) {
+            public double g(double x) {
                 return 3 * x * x + 2 * x - 5;
             }
         };
