diff --git a/old_product.java b/new_product.java
index 7e9036b..7d7266c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,7 @@
 public static String replace(final String p, final InputSource input,
       final Context ctx, final boolean lock) throws BaseXException {
-    final Data data = ctx.data;
+
+    final Data data = ctx.data();
     String path = path(p);
     if(path.isEmpty()) return Util.info(DIRERR, path);
 
