diff --git a/old_product.java b/new_product.java
index f059727..aeee19f 100644
--- a/old_product.java
+++ b/new_product.java
@@ -19,6 +19,9 @@ protected Connection obtainInternalConnection() throws SQLException {
 				if (acquireRetryAttempts != this.pool.getConfig().getAcquireRetryAttempts()){
 					logger.info("Successfully re-established connection to DB");
 				}
+				
+				this.pool.getDbIsDown().set(false);
+				
 				// call the hook, if available.
 				if (this.connectionHook != null){
 					this.connectionHook.onAcquire(this);
