diff --git a/old_product.java b/new_product.java
index 9b8e56b..d886e0e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 private ANode doc(final QueryContext ctx) throws QueryException {
-    final Item it = expr[0].item(ctx, info);
+    final Item it = exprs[0].item(ctx, info);
     if(it == null) return null;
     final byte[] in = checkEStr(it);
     if(!Uri.uri(in).isValid()) throw INVDOC.get(info, in);
