diff --git a/old_product.java b/new_product.java
index 37bacf5..6c5f5c1 100644
--- a/old_product.java
+++ b/new_product.java
@@ -17,15 +17,5 @@
             return false;
         }
 
-        // check the deprecated fields
-        if (this.categoryIndex != that.categoryIndex) {
-            return false;
-        }
-        if (this.series != that.series) {
-            return false;
-        }
-        if (!ObjectUtils.equal(this.category, that.category)) {
-            return false;
-        }
         return super.equals(obj);
     }
\ No newline at end of file
