diff --git a/old_product.java b/new_product.java
index f8420fc..af4a857 100644
--- a/old_product.java
+++ b/new_product.java
@@ -4,6 +4,8 @@ private FElem parse(final QueryContext ctx) throws QueryException {
 
     try {
       return new CsvConverter(opts).convert(input);
+    } catch(final QueryIOException ex) {
+      throw ex.getCause(info);
     } catch(final IOException ex) {
       throw BXCS_PARSE.thrw(info, ex);
     }
