diff --git a/old_product.java b/new_product.java
index 6ad2daf..f9b3887 100644
--- a/old_product.java
+++ b/new_product.java
@@ -3,7 +3,8 @@ public List<JobCommand> workerHeartbeat(long workerId, List<TaskInfo> taskInfoLi
     try (LockResource workersLockShared = new LockResource(mWorkerRWLock.readLock())) {
       MasterWorkerInfo worker = mWorkers.getFirstByField(mIdIndex, workerId);
       if (worker == null) {
-        return Collections.singletonList(JobCommand.registerCommand(new RegisterCommand()));
+        return Collections.singletonList(JobCommand.newBuilder()
+            .setRegisterCommand(RegisterCommand.getDefaultInstance()).build());
       }
       // Update last-update-time of this particular worker under lock
       // to prevent lost worker detector clearing it under race
