diff --git a/old_product.java b/new_product.java
index 4f93c56..d02687c 100644
--- a/old_product.java
+++ b/new_product.java
@@ -10,10 +10,7 @@
                     streamsList.forEach(stream -> {
                         // If internal streams are requested select only the ones that have the special stream names
                         // otherwise display the regular user created streams.
-                        // TODO: Remove the 200 size limit once issue - https://github.com/pravega/pravega/issues/926
-                        // is fixed.
-                        if ((!showOnlyInternalStreams ^ stream.getStreamName().startsWith(INTERNAL_NAME_PREFIX)) &&
-                                streams.getStreams().size() < 200) {
+                        if (!showOnlyInternalStreams ^ stream.getStreamName().startsWith(INTERNAL_NAME_PREFIX)) {
                             streams.addStreamsItem(ModelHelper.encodeStreamResponse(stream));
                         }
                     });
