diff --git a/old_test.java b/new_test.java
index 3973792..9ba26ed 100644
--- a/old_test.java
+++ b/new_test.java
@@ -74,12 +74,12 @@
         response.close();
 
         // Test to list large number of streams.
-        streamsList = Collections.nCopies(1000, streamConfiguration1);
+        streamsList = Collections.nCopies(50000, streamConfiguration1);
         when(mockControllerService.listStreamsInScope("scope1")).thenReturn(CompletableFuture.completedFuture(streamsList));
         response = client.target(resourceURI).request().buildGet().invoke();
         assertEquals("List Streams response code", 200, response.getStatus());
         assertTrue(response.bufferEntity());
         final StreamsList streamsList2 = response.readEntity(StreamsList.class);
-        assertEquals("List count", 200, streamsList2.getStreams().size());
+        assertEquals("List count", 50000, streamsList2.getStreams().size());
         response.close();
     }
\ No newline at end of file
