diff --git a/old_product.java b/new_product.java
index bb5b569..cc1cd08 100644
--- a/old_product.java
+++ b/new_product.java
@@ -8,7 +8,7 @@
         Long2ObjectMap<SparseVector> itemVectors = new Long2ObjectOpenHashMap<SparseVector>(1000);
         Cursor<ItemEventCollection<Event>> itemCursor = itemEventDAO.streamEventsByItem();
         try {
-            for (ItemEventCollection<Event> item: itemCursor.fast()) {
+            for (ItemEventCollection<Event> item: itemCursor) {
                 if (logger.isTraceEnabled()) {
                     logger.trace("processing {} ratings for item {}", item.size(), item);
                 }
@@ -17,7 +17,7 @@
                                                      .toList();
                 MutableSparseVector vector = Ratings.itemRatingVector(ratings);
                 normalizer.normalize(item.getItemId(), vector, vector);
-                for (VectorEntry e: vector.fast()) {
+                for (VectorEntry e: vector) {
                     long user = e.getKey();
                     LongList uis = userItems.get(user);
                     if (uis == null) {
