diff --git a/old_test.java b/new_test.java
index 8bf5f68..7236e2e 100644
--- a/old_test.java
+++ b/new_test.java
@@ -10,7 +10,7 @@
         allowed_types = CollectionUtil.addAll(new HashSet<PartitionMethodType>(), PartitionMethodType.HASH,
                                                                                   PartitionMethodType.REPLICATION,
                                                                                   PartitionMethodType.MAP);
-        for (Entry<Table,PartitionEntry> e : pplan.getTableEntries().entrySet()) {
+        for (Entry<Table,TableEntry> e : pplan.getTableEntries().entrySet()) {
             assertNotNull(e.getKey());
             assertNotNull("Null PartitionEntry for " + e.getKey(), e.getValue());
             assert(allowed_types.contains(e.getValue().getMethod())) : "Unexpected: " + e.getValue().getMethod();
@@ -20,7 +20,7 @@
         // PROCEDURES
         allowed_types = CollectionUtil.addAll(new HashSet<PartitionMethodType>(), PartitionMethodType.HASH,
                                                                                   PartitionMethodType.NONE);
-        for (Entry<Procedure, PartitionEntry> e : pplan.getProcedureEntries().entrySet()) {
+        for (Entry<Procedure, ProcedureEntry> e : pplan.getProcedureEntries().entrySet()) {
             assertNotNull(e.getKey());
             assertNotNull("Null PartitionEntry for " + e.getKey(), e.getValue());
             assert(allowed_types.contains(e.getValue().getMethod())) : "Unexpected: " + e.getValue().getMethod();
