diff --git a/old_test.java b/new_test.java
index 583c294..687e3e2 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,6 +1,5 @@
 @Test
   public void add() {
     assertEquals(SIZE, set.size());
-    for(final byte[] t : LIST)
-      assertTrue("Token is missing in list.", set.id(t) != 0);
+    for(final byte[] t : LIST) assertTrue("Token is missing.", set.contains(t));
   }
\ No newline at end of file
