diff --git a/old_product.java b/new_product.java
index 07162ea..aa3b89b 100644
--- a/old_product.java
+++ b/new_product.java
@@ -38,16 +38,16 @@ void add(final QueryContext ctx) throws QueryException  {
     if(ngl == 0) return;
 
     // adds the current non-grouping variable bindings to the p-th partition.
-    if(p == items.size()) items.add(new ItemCache[ngl]);
-    final ItemCache[] sq = items.get(p);
+    if(p == items.size()) items.add(new ValueBuilder[ngl]);
+    final ValueBuilder[] sq = items.get(p);
 
     for(int i = 0; i < ngl; ++i) {
-      ItemCache ic = sq[i];
+      ValueBuilder vb = sq[i];
       final Value result = ctx.value(ctx.vars.get(ngv[0][i]));
-      if(ic == null) {
-        ic = new ItemCache();
-        sq[i] = ic;
+      if(vb == null) {
+        vb = new ValueBuilder();
+        sq[i] = vb;
       }
-      ic.add(result);
+      vb.add(result);
     }
   }
\ No newline at end of file
