diff --git a/old_test.java b/new_test.java
index c802592..c1b83c0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -7,7 +7,7 @@
         new MockResponse().addHeader(CONTENT_TYPE, HttpMediaType.APPLICATION_JSON).setBody(GSON.toJson(customization)));
 
     Customization result = service
-        .createCustomization(customization.getName(), SpeechModel.EN_UK_BROADBANDMODEL, customization.getDescription())
+        .createCustomization(customization.getName(), SpeechModel.EN_GB_BROADBANDMODEL, customization.getDescription())
         .execute();
     final RecordedRequest request = server.takeRequest();
 
