diff --git a/old_product.java b/new_product.java
index ebc879f..8757175 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,13 +1,3 @@
 public ServiceCall<Customization> createCustomization(String name, SpeechModel baseModel, String description) {
-    Validator.notNull(name, "name cannot be null");
-    Validator.notNull(baseModel, "baseModel cannot be null");
-
-    RequestBuilder requestBuilder = RequestBuilder.post(PATH_CUSTOMIZATIONS);
-
-    Customization newCustomization = new Customization();
-    newCustomization.setBaseModelName(baseModel.getName());
-    newCustomization.setDescription(description);
-    newCustomization.setName(name);
-    requestBuilder.bodyContent(GSON.toJson(newCustomization), HttpMediaType.APPLICATION_JSON);
-    return createServiceCall(requestBuilder.build(), ResponseConverterUtils.getObject(Customization.class));
+    return createCustomization(name, baseModel, description, null);
   }
\ No newline at end of file
