diff --git a/old_product.java b/new_product.java
index 9e4fcd2..5204033 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
 private static byte[] transform(final IO in, final IO xsl, final HashMap<String, String> params,
-      final XsltOptions xopts) throws TransformerException {
+      final XsltOptions xopts, final QueryContext qc) throws TransformerException {
 
     // retrieve new or cached transformer
     final Transformer tr = transformer(xsl.streamSource(), xopts.get(XsltOptions.CACHE));
@@ -7,7 +7,7 @@ private static byte[] transform(final IO in, final IO xsl, final HashMap<String,
     params.forEach(tr::setParameter);
 
     // set URI resolver
-    final CatalogWrapper cw = CatalogWrapper.get(xopts.get(XsltOptions.CATALOG));
+    final CatalogWrapper cw = CatalogWrapper.get(qc.context.options.get(MainOptions.CATFILE));
     if(cw != null) tr.setURIResolver(cw.getURIResolver());
 
     // do transformation and return result
