diff --git a/old_test.java b/new_test.java
index 4e787fb..245ad19 100644
--- a/old_test.java
+++ b/new_test.java
@@ -9,4 +9,19 @@
         "<X><xsl:value-of select='$t'/></X></xsl:template>");
     query(func.args(doc, ' ' + style, " map { 't': '1' }"), "<X>1</X>");
     query(func.args(doc, ' ' + style, " map { 't' : text { '1' } }"), "<X>1</X>");
+
+    // catalog manager (via option declaration); requires resolver in lib/ directory
+    final String dir = "src/test/resources/catalog/";
+    query("declare option db:catfile '" + dir + "catalog.xml';" +
+        func.args(" <dummy/>", dir + "document.xsl"),
+        "<x>X</x>");
+    query("declare option db:catfile '" + dir + "catalog.xml';" +
+        func.args(" <dummy/>", " doc('" + dir + "document.xsl')"),
+        "<x>X</x>");
+
+    // catalog manager (via pragma); requires resolver in lib/ directory
+    query("(# db:catfile " + dir + "catalog.xml #) { " +
+        func.args(" <dummy/>", dir + "document.xsl") + " }",
+        "<x>X</x>");
+
   }
\ No newline at end of file
