diff --git a/old_product.java b/new_product.java
index f1d57c8..5ed04fc 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,12 +1,12 @@
 private Value query(final QueryContext qc) throws QueryException {
     final ClientSession cs = session(qc, false);
-    final String query = Token.string(checkStr(exprs[1], qc));
+    final String query = Token.string(toToken(exprs[1], qc));
     final ValueBuilder vb = new ValueBuilder();
     ClientQuery cq = null;
     try {
       cq = cs.query(query);
-      // bind variables and context item
-      for(final Map.Entry<String, Value> binding : bindings(2, qc).entrySet()) {
+      // bind variables and context value
+      for(final Map.Entry<String, Value> binding : toBindings(2, qc).entrySet()) {
         final String k = binding.getKey();
         final Value value = binding.getValue();
         if(k.isEmpty()) cq.context(value);
@@ -27,9 +27,9 @@ private Value query(final QueryContext qc) throws QueryException {
         final QueryException exc = get(m.group(1), m.group(2), info);
         throw exc == null ? new QueryException(info, new QNm(m.group(1)), m.group(2)) : exc;
       }
-      throw BXCL_QUERY.get(info, ex);
+      throw BXCL_QUERY_X.get(info, ex);
     } catch(final IOException ex) {
-      throw BXCL_COMM.get(info, ex);
+      throw BXCL_COMM_X.get(info, ex);
     } finally {
       if(cq != null) try { cq.close(); } catch(final IOException ignored) { }
     }
