diff --git a/old_test.java b/new_test.java
index 3c9f9dc..292131f 100644
--- a/old_test.java
+++ b/new_test.java
@@ -12,5 +12,5 @@
     query(_CLIENT_QUERY.args(conn(), "\"declare variable $a external; $a\"",
         " map { 'a': (1,<a/>,'a') }"), "1<a/>a");
     // query errors
-    error(_CLIENT_QUERY.args(conn(), "x"), Err.NOCTX);
+    error(_CLIENT_QUERY.args(conn(), "x"), Err.NOCTX_X);
   }
\ No newline at end of file
