diff --git a/old_product.java b/new_product.java
index 1f1e969..194950e 100644
--- a/old_product.java
+++ b/new_product.java
@@ -6,7 +6,7 @@
     Set<Resource> resources = getResources(
         new RequestImpl(null, null, null, null), predicate);
 
-    Set<Long> definitionIds = new HashSet<Long>();
+    Set<Long> definitionIds = new HashSet<>();
 
     for (final Resource resource : resources) {
       Long id = (Long) resource.getPropertyValue(ALERT_DEF_ID);
