diff --git a/old_test.java b/new_test.java
index 2aa3725..a8f77e8 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,5 +1,18 @@
 @Test
   public void testGetChatTone() {
+    List<Utterance> utterances = new ArrayList<>();
+    for (int i = 0; i < texts.length; i++) {
+      Utterance utterance = new Utterance.Builder()
+          .text(texts[i])
+          .user(users[i])
+          .build();
+      utterances.add(utterance);
+    }
+    ToneChatRequest toneChatInput = new ToneChatRequest.Builder()
+        .utterances(utterances)
+        .build();
+
+
     UtterancesTone utterancesTone = service.getChatTone(toneChatInput).execute();
 
     Assert.assertNotNull(utterancesTone);
