diff --git a/old_product.java b/new_product.java
index 7c880de..7ba48d9 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,5 +1,5 @@
-public ServiceCall<UtterancesTone> getChatTone(ToneChatInput chatInput) {
-    Validator.notNull(chatInput.getUtterances(), "Input utterances cannot be null");
+public ServiceCall<UtterancesTone> getChatTone(ToneChatRequest chatInput) {
+    Validator.notNull(chatInput.utterances(), "chatInput.utterances cannot be null");
 
     RequestBuilder requestBuilder = RequestBuilder.post(PATH_CHAT_TONE).query(VERSION_DATE, versionDate);
     requestBuilder.bodyJson(GsonSingleton.getGson().toJsonTree(chatInput).getAsJsonObject());
