diff --git a/old_product.java b/new_product.java
index 6740ac1..ce57fcf 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,4 +1,4 @@
 public void equiToNorm(float x , float y , Vector3D_F32 norm ) {
-		equiToLatlon(x,y, temp);
-		ConvertCoordinates3D_F32.latlonToUnitVector(temp.x,temp.y, norm);
+		equiToLonlat(x,y, temp);
+		ConvertCoordinates3D_F32.latlonToUnitVector(temp.y,temp.x, norm);
 	}
\ No newline at end of file
