diff --git a/old_product.java b/new_product.java
index 0436adc..97d8cff 100644
--- a/old_product.java
+++ b/new_product.java
@@ -1,6 +1,9 @@
 private Item backup(final QueryContext ctx) throws QueryException {
     checkCreate(ctx);
-    final Data data = checkData(ctx);
-    ctx.updates.add(new DBBackup(data, info, ctx), ctx);
+    final String name = string(checkStr(expr[0], ctx));
+    if(!Databases.validName(name)) throw BXDB_NAME.get(info, name);
+    if(!ctx.context.globalopts.dbexists(name)) throw BXDB_WHICH.get(info, name);
+
+    ctx.updates.add(new DBBackup(name, info, ctx), ctx);
     return null;
   }
\ No newline at end of file
