diff --git a/old_test.java b/new_test.java
index 9033235..88ed1b0 100644
--- a/old_test.java
+++ b/new_test.java
@@ -1,13 +1,17 @@
 @Test
-	public void allInside() {
-		IntrinsicParameters param = new IntrinsicParameters().fsetK(300, 320, 0, 150, 130, width, height).fsetRadial(0.1,1e-4);
+	public void allInside_Transform() {
+		IntrinsicParameters param =
+				new IntrinsicParameters().fsetK(300, 320, 0, 150, 130, width, height).fsetRadial(0.1, 1e-4);
 
-		PointTransform_F32 adjusted = LensDistortionOps.allInside(param,null,true);
-		checkInside(adjusted);
+		PointTransform_F32 adjToDist = LensDistortionOps.allInside(param, null, true);
+		PointTransform_F32 distToAdj = LensDistortionOps.allInside(param, null, false);
+		checkInside(adjToDist, distToAdj);
 
 		// distort it in the other direction
 		param = new IntrinsicParameters().fsetK(300, 320, 0, 150, 130, width, height).fsetRadial(-0.1,-1e-4);
 
-		adjusted = LensDistortionOps.allInside(param, null,true);
-		checkInside(adjusted);
+		adjToDist = LensDistortionOps.allInside(param, null, true);
+		distToAdj = LensDistortionOps.allInside(param, null, false);
+
+		checkInside(adjToDist, distToAdj);
 	}
\ No newline at end of file
