diff --git a/old_test.java b/new_test.java
index 63190f1..99a3178 100644
--- a/old_test.java
+++ b/new_test.java
@@ -15,9 +15,18 @@
         assertThat("testGetSubcommunities 0",c.getSubcommunities(), notNullValue());
         assertTrue("testGetSubcommunities 1", c.getSubcommunities().size() == 0);
 
-        //community with parent
-        Community son = communityService.create(c, context);
-        assertThat("testGetSubcommunities 2",c.getSubcommunities(), notNullValue());
-        assertTrue("testGetSubcommunities 3", c.getSubcommunities().size() == 1);
-        assertThat("testGetSubcommunities 4", c.getSubcommunities().get(0), equalTo(son));
+        context.turnOffAuthorisationSystem();
+        Community community = communityService.create(c, context);
+        communityService.setMetadataSingleValue(context, community, MetadataSchema.DC_SCHEMA, "title", null, Item.ANY, "subcommunity B");
+        community = communityService.create(c, context);
+        communityService.setMetadataSingleValue(context, community, MetadataSchema.DC_SCHEMA, "title", null, Item.ANY, "subcommunity A");
+        community = communityService.create(c, context);
+        communityService.setMetadataSingleValue(context, community, MetadataSchema.DC_SCHEMA, "title", null, Item.ANY, "subcommunity C");
+        //we need to commit the changes so we don't block the table for testing
+        context.restoreAuthSystemState();
+
+        //get Subcommunities sorted
+        assertTrue("testGetCollections 2",c.getSubcommunities().get(0).getName().equals("subcommunity A"));
+        assertTrue("testGetCollections 3",c.getSubcommunities().get(1).getName().equals("subcommunity B"));
+        assertTrue("testGetCollections 4",c.getSubcommunities().get(2).getName().equals("subcommunity C"));
     }
\ No newline at end of file
