diff --git a/old_test.java b/new_test.java
index 27990e6..0cbdf49 100644
--- a/old_test.java
+++ b/new_test.java
@@ -2,7 +2,7 @@
     for(final String[] f : TOJSON) {
       final String qu = _JSON_SERIALIZE.args(f[0]);
       if(f.length == 1) {
-        error(qu, Err.BXJS_SERIAL);
+        error(qu, Err.BXJS_SERIAL, Err.BXJS_CONFIG);
       } else if(f[1].startsWith("...")) {
         contains(qu, f[1].substring(3));
       } else {
